<!DOCTYPE html>
<html>





  <head>
    <meta charset="utf-8">
    <title>Options for using Galaxy</title>
    <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/slides.css">
    <link rel="stylesheet" href="https://galaxyproject.github.io/training-material//assets/css/font-awesome.css" id="theme">
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-footer"><span>

<img src="https://galaxyproject.github.io/training-material//assets/images/GTN.png" style="height: 40px;"/>

</span></div>
---

## Welcome!

This material is the result of a collaborative work. Thanks the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors !


<img src="https://galaxyproject.github.io/training-material//assets/images/GTN.png" style="height: 100px;"/>


---

# Options for using Galaxy



---

# Options for using Galaxy

---

### Options for using Galaxy

Galaxy is available in many ways, each with different configurations and capabilities

Here are some options

???

Many workshops use Galaxy instances that go away once the workshop is done.
This deck lists options for using Galaaxy outside of a training context.

---

### usegalaxy.org

- [usegalaxy.org](https://usegalaxy.org/) is the Galaxy Project's public server
- Hosted at the [Texas Advanced Computing Center (TACC)](https://www.tacc.utexas.edu/)
- Lots of
  - tools
  - reference genomes
  - compute power
  - storage
- But also very busy
  - Averages 250,000 jobs *per month* in 2016
  - A single resource cannot scale to meet the demand of the entire world, nor can it support all wanted tools, genomes, ...

---

### Public Galaxy Servers

[.image-25[![](../../images/public_server_galaxeast.png)]](https://galaxyproject.org/public-galaxy-servers/#galaxeast)
[.image-25[![](../../images/public_servers_odometer_count.png)]](https://galaxyproject.org/public-galaxy-servers/)
[.image-25[![](../../images/public_server_gigagalaxy.png)]](https://galaxyproject.org/public-galaxy-servers/#gigagalaxy)
[.image-25[![](../../images/public_server_genomichyperbrowser.png)]](https://hyperbrowser.uio.no/hb/)
[.image-25[![](../../images/public_server_lappsgrid.png)]](https://galaxy.lappsgrid.org/)

---

### Public Galaxy Servers

1. General purpose genomics servers
1. Domain specific servers&lt;br /&gt;
   Can be specific to:
   - *Research methods*: ChIP-Seq, RNA-Seq, repeats, ...
   - *Organisms*: Pathogens, phage, rice, poplars (!), ...
   - *Non-genomic domains*: image analysis, social science, natural language processing, ...
1. Tool servers
   - Host specific tools to make them easy to access and run

Complete list: [bit.ly/gxyServers ](https://bit.ly/gxyServers)

---

### Semi-public Galaxy services

Access based on geography or community membership

| Geography  | Resource |
| ---------- | -------- |
| Australia  | [Genomics Virtual Lab (GVL)](https://launch.genome.edu.au/launch)  |
| Canada | [GenAP](https://www.genap.ca/) |
| Norway | [Norwegian e-Infrastructure for Life Sciences (NeLS)](https://nels.bioinfo.no/) |
| US | [Jetstream](https://galaxyproject.org/cloud/jetstream/) |

Complete list: [bit.ly/gxysemipublic ](https://bit.ly/gxysemipublic)

---

.pull-left[
### Public clouds via CloudLaunch

.left[[CloudLaunch](https://launch.usegalaxy.org/) currently only supports:]
- Jetstream (covered above)
- **Amazon Web Services** (AWS)
  - You need to create an AWS account (which requires a
    credit card) and then provide your public and secret keys to CloudLaunch

]

.pull-right[
[.image-75[![](../../images/cloudlaunch_empty.png)]](https://launch.usegalaxy.org/)
]
---

.pull-left[
### CloudLaunch / CloudMan instances

- Comes with 100's of tools and many pre-defined reference genomes
- CloudLaunch starts a CloudMan-based Galaxy instance
  - **CloudMan** is a cloud management tool that moves much / all of the cloud management work into a Graphical user interface
]

.pull-right[
[.image-75[![](../../../../shared/images/cloudman_based_server_landing_page.png)]](https://wiki.galaxyproject.org/CloudMan)
]
---

### CloudMan

[.image-25[![](../../../../shared/images/cloudman_management_console_showing_autoscaled_w_1_worker.png)]](https://wiki.galaxyproject.org/CloudMan)

CloudMan enables you to statically or dynamically scale compute power

- This example shows a Galaxy Server with a head node and one worker node, and that is configured to scale up to 5 worker nodes, if demand justifies it
- Cloud instances can be incredibly cost effective, but only if you are actually using them when they are up
  - Paying to have a cloud instance up for two weeks that you only use for 8 hours is not cost effective
- **It is very important to shut down and remove everything when you are done**

---

### Docker (and maybe other appliances?)

Stuff about docker...

---

### Run your own Galaxy locally

- Galaxy is [open source software](https://getgalaxy.org/) and can be installed on local compute infrastructure, from lab servers to institutional compute clusters
- Installing Galaxy locally is relatively easy, but
  - the initial install does not include reference genomes and only has a few tools
  - installing tools and genomes, setting up authentication, and connecting to institutional compute resources all takes work
- There are hundreds of local Galaxy installs around the world
  - Installing tools and genomes has become much easier in recent years, and can now often be done with the Galaxy Admin GUI
  - Authentication and connecting to institutional compute resources is still heavy lifting

---

### Commercial support

- You can buy a [preconfigured Galaxy server](https://bioteam.net/products/galaxy-appliance/) from [BioTeam](https://bioteam.net/)
- [Globus Genomics](http://globusgenomics.org/) and [GenomeCloud](https://www.genome-cloud.com/) provide cloud-based Galaxy servers
- [Commercially provided consulting and training](https://galaxyproject.org/support/commercial/) are also available

---

### Which Galaxy instance to use?

- Choose based on
  - size of datasets, available storage, backup
  - data security
  - computational requirements
  - tools installed


---



## Tutorials



    

    
        
- [Galaxy 101](https://galaxyproject.github.io/training-material//topics/introduction/tutorials/galaxy-intro-101/tutorial.html)
        
    

    
        
- [From peaks to genes](https://galaxyproject.github.io/training-material//topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html)
        
    

    
        
- [Multisample Analysis](https://galaxyproject.github.io/training-material//topics/introduction/tutorials/processing-many-samples-at-once/tutorial.html)
        
    

    
        
- [Options for using Galaxy](https://galaxyproject.github.io/training-material//topics/introduction/slides/options-for-using-galaxy)
        
    

    
        
- [IGV Introduction](https://galaxyproject.github.io/training-material//topics/introduction/tutorials/igv-introduction/tutorial.html)
        
    

    
        
- [Visualization](https://galaxyproject.github.io/training-material//topics/introduction/tutorials/galaxy-intro-vis/tutorial.html)
        
    

    
        
- [Getting data into Galaxy](https://galaxyproject.github.io/training-material//topics/introduction/slides/galaxy-intro-get-data)
        
    

    
        
- [Collections](https://galaxyproject.github.io/training-material//topics/introduction/tutorials/galaxy-intro-collections/tutorial.html)
        
    



.footnote[Found a typo? Something is wrong in this tutorial? <br/>Edit it on [GitHub](https://github.com/galaxyproject/training-material/tree/master/introduction/slides/slides.html)]

    </textarea>
    <script src="https://galaxyproject.github.io/training-material//assets/js/remark-latest.min.js" type="text/javascript">
    </script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
    </script>
  </body>
</html>
