<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Using dataset collections</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="A collection of microtutorials explaining various feature..." />
        <meta property="og:title" content="Galaxy Training: Using dataset collections" />
        <meta property="og:description" content="A collection of microtutorials explaining various feature..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/galaxy-interface" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> User Interface and Data Manipulation
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fgalaxy-interface%2Ftutorials%2Fcollections%2Ftutorial.html&edit-text=&act=url" title="">
                                FranÃ§ais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fgalaxy-interface%2Ftutorials%2Fcollections%2Ftutorial.html&edit-text=&act=url" title="">
                                æ—¥æœ¬èªž
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fgalaxy-interface%2Ftutorials%2Fcollections%2Ftutorial.html&edit-text=&act=url" title="">
                                EspaÃ±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fgalaxy-interface%2Ftutorials%2Fcollections%2Ftutorial.html&edit-text=&act=url" title="">
                                PortuguÃªs
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fgalaxy-interface%2Ftutorials%2Fcollections%2Ftutorial.html&edit-text=&act=url" title="">
                                Ø§Ù„Ø¹Ø±Ø¨ÙŠØ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fgalaxy-interface%2Ftutorials%2Fcollections%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> ðŸŽƒ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/galaxy-interface/tutorials/collections/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Using dataset collections",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "User Interface and Data Manipulation",
    "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
    "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/"
  },
  "courseCode": "galaxy-interface / collections / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Using dataset collections' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/tutorials/collections/tutorial.html",
  "timeRequired": "PT30M",
  "keywords": "collections",
  "description": "The questions this  addresses are:\n - How to manipulate large numbers of datasets at once?\n\n\\nThe objectives are:\n - Understand and master dataset collections\n\n",
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "User Interface and Data Manipulation",
      "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
      "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/"
    }
  ],
  "skillLevel": "Intermediate"
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Using dataset collections</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nekrut" alt="Anton Nekrutenko">Anton Nekrutenko</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to manipulate large numbers of datasets at once?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Understand and master dataset collections</p>
</li>
            
            </ul>

            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 30 minutes</p>
            

            
            <p><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> Level: </strong>
             Intermediate 
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span>
</span>

</p>
            

            
            

            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <p>Here we will show <span class="notranslate">Galaxy</span> features designed to help with the analysis of large numbers of samples. When you have just a few samples - clicking through them is easy. But once youâ€™ve got hundreds - it becomes very annoying. In <span class="notranslate">Galaxy</span> we have introduced <strong>Dataset collections</strong> that allow you to combine numerous datasets in a single entity that can be easily manipulated.</p>

<h1 id="getting-data">Getting data</h1>
<p><a href="https://usegalaxy.org/library/folders/Fab5f788f07073c11">Here</a> is a library containing a few datasets we will be practicing with (as with any <span class="notranslate">Galaxy</span> tutorial, you can upload your own data and play with it instead of the provided datasets):</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">M117-bl_1</code> - family 117, mother, 1-st (<strong>F</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117-bl_2</code> - family 117, mother, 2-nd (<strong>R</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117-ch_1</code> - family 117, mother, 1-st (<strong>F</strong>) read from <strong>cheek</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117-ch_1</code> - family 117, mother, 2-nd (<strong>R</strong>) read from <strong>cheek</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-bl_1</code>- family 117, child, 1-st (<strong>F</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-bl_2</code>- family 117, child, 2-nd (<strong>R</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-ch_1</code>- family 117, child, 1-st (<strong>F</strong>) read from <strong>cheek</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-ch_2</code>- family 117, child, 2-nd (<strong>R</strong>) read from <strong>cheek</strong></li>
</ul>

<p>These datasets represent genomic <span class="notranslate">DNA</span> (enriched for mitochondria via a long range PCR) isolated from blood (<code class="language-plaintext highlighter-rouge">bl</code>) and cheek (buccal swab, <code class="language-plaintext highlighter-rouge">ch</code>) of mother (<code class="language-plaintext highlighter-rouge">M117</code>) and her child (<code class="language-plaintext highlighter-rouge">M117C1</code>) that was sequenced on an Illumina miSeq machine as paired-read library (250-bp <span class="notranslate">reads</span>; see our <a href="http://www.pnas.org/content/111/43/15474.abstract">2014</a> manuscript for <strong>Methods</strong>).</p>

<h2 id="load-data-from-galaxy-library">Load data from <span class="notranslate">Galaxy</span> library</h2>

<p>Right click (or Ctrl-click) on <a href="https://usegalaxy.org/library/folders/Fab5f788f07073c11">this link</a> to open a new browser window and position this window side-by-side with the window displaying this tutorial. You will see this:</p>

<p class="img-responsive"><img src="../../images/grab_data.png" alt="Data in Galaxy library" /></p>

<p>Select all datasets, click <strong>to History</strong> button, and select <strong>as Datasets</strong>. In the pop-up window, either select a history or create a new one. This will import all datasets into a history. The screen will look like this:</p>

<p class="img-responsive"><img src="../../images/manyDatasets.png" alt="manyDatasets" /></p>

<h2 id="creating-a-paired-dataset-collection">Creating a paired dataset collection</h2>

<p>Now click the checkbox in <img src="../../images/historyItemControls.png" alt="historyItemControls" /> and you will see your history changing like this:</p>

<p class="img-responsive"><img src="../../images/historyWithCheckboxes.png" alt="historyWithCheckboxes" /></p>

<p>Letâ€™s click <strong>All</strong>, which will select all datasets in the history, then click <strong>For all selectedâ€¦</strong> and finally select <strong>Build List of Dataset Pairs</strong> from the following menu:</p>

<p class="img-responsive"><img src="../../images/buildPairs.png" alt="buildPairs" /></p>

<p>The following wizard will appear:</p>

<p class="img-responsive"><img src="../../images/collectionCreation.png" alt="collectionCreation" /></p>

<p>In this case <span class="notranslate">Galaxy</span> automatically assigned pairs using the <code class="language-plaintext highlighter-rouge">_1</code> and <code class="language-plaintext highlighter-rouge">_2</code> endings of dataset names. Letâ€™s however pretend that this did not happen. Click on <strong>Unpair all</strong> (highlighted in red in the figure above) link and then on <strong>Clear</strong> link (highlighted in blue in the figure above). The interface will change into its unpaired state:</p>

<p class="img-responsive"><img src="../../images/collectionCreationClean.png" alt="collectionCreationClean" /></p>

<p>These datasets contain paired-end <span class="notranslate">reads</span>. Datasets containing the first (forward) and the second (reverse) read are differentiated by having <code class="language-plaintext highlighter-rouge">_1</code> and <code class="language-plaintext highlighter-rouge">_2</code> in the filename. We can use this feature in dataset collection wizard to pair our datasets.  Type <code class="language-plaintext highlighter-rouge">_1</code> in the left <strong>Filter this list</strong> text box and <code class="language-plaintext highlighter-rouge">_2</code> in the right:</p>

<p class="img-responsive"><img src="../../images/1and2.png" alt="1and2" /></p>

<p>You will see that the dataset collection wizard will automatically filter lists on each side of the interface:</p>

<p class="img-responsive"><img src="../../images/collectionPrefiltered.png" alt="collectionPrefiltered" /></p>

<p>Now you can either click <strong>Auto pair</strong> if pairs look good to you (proper combinations of datasets are listed in each line) or pair each forward/reverse group individually by pressing <strong>Pair these datasets</strong> button separating each pair:</p>

<p class="img-responsive"><img src="../../images/collectionCreation.png" alt="collectionCreation" /></p>

<p>Now it is time to name the collection:</p>

<p><img src="../../images/collectionNaming.png" alt="collectionNaming" /></p>

<p>and create the collection by clicking <strong>Create list</strong>. A new item will appear in the history as you can see on the panel <strong>A</strong> below. Clicking on collection will expand it to show four pairs it contains (panel <strong>B</strong>). Clicking individual pairs will expand them further to reveal <strong>forward</strong> and <strong>reverse</strong> datasets (panel <strong>C</strong>). Expanding these further will enable one to see individual datasets (panel <strong>D</strong>).</p>

<p class="img-responsive"><img src="../../images/collection_ABCD.png" alt="collection_ABCD" /></p>

<h2 id="using-collections">Using collections</h2>

<p>By now we see that a collection can be used to bundle a large number of items into a single history item. This means that many <span class="notranslate">Galaxy</span> tools will be able to process all datasets in a collection transparently to you. Letâ€™s try to map these datasets to human genome using <code class="language-plaintext highlighter-rouge">bwa-mem</code> mapper. In the Tools pane on the left, use the <code class="language-plaintext highlighter-rouge">search-tools</code> to find <code class="language-plaintext highlighter-rouge">bwa-mem</code>. Click on <code class="language-plaintext highlighter-rouge">Map with BWA-MEM</code> in the search list.</p>

<p class="img-responsive"><img src="../../images/bwa_mem_collection_readGroups.png" alt="bwa_mem_collection_readGroups" /></p>

<p>In the central panel, provide the following parameters for <code class="language-plaintext highlighter-rouge">Map with BWA-MEM</code>:</p>

<ul>
  <li>set <strong>Using reference genome</strong> to <code class="language-plaintext highlighter-rouge">hg38</code> (red outline);</li>
  <li>set <strong>Single or Paired-end <span class="notranslate">reads</span></strong> to <code class="language-plaintext highlighter-rouge">Paired collection</code> (blue outline);</li>
  <li>select <code class="language-plaintext highlighter-rouge">M177-collection</code> from <strong>Select a paired collection</strong> dropdown (magenta outline);</li>
  <li>In <strong>Set read groups information</strong> select <code class="language-plaintext highlighter-rouge">Automatically assign ID</code> (green outline);</li>
  <li>scroll down and click <strong>Execute</strong>.</li>
</ul>

<p>You will see jobs being submitted and new datasets appearing in the history. IN particular below you can see that <span class="notranslate">Galaxy</span> has started four jobs (two yellow and two gray). This is because we have eight paired datasets with each pair being processed separately by <code class="language-plaintext highlighter-rouge">bwa-mem</code>. As a result we have four <code class="language-plaintext highlighter-rouge">bwa-mem</code> runs:</p>

<p class="img-responsive"><img src="../../images/bwa_memCollectionRunning.png" alt="bwa_memCollectionRunning" /></p>

<p>Once these jobs are finished they will disappear from the history and all results will be represented as a new collection:</p>

<p class="img-responsive"><img src="../../images/bwa_memCollectionDone.png" alt="bwa_memCollectionDone" /></p>

<p>Letâ€™s look at this collection by clicking on it (panel <strong>A</strong> in the figure below). You can see that now this collection is no longer paired (compared to the collection we created in the beginning of this tutorial). This is because <code class="language-plaintext highlighter-rouge">bwa-mem</code> takes forward and reverse data as input, but produces only a single BAM dataset as the output. So what we have in the result is a <em>list</em> of four dataset (BAM files; panels <strong>B</strong> and <strong>C</strong>).</p>

<p class="img-responsive"><img src="../../images/bwa_memCollection_ABC.png" alt="bwa_memCollection_ABC" /></p>

<h2 id="processing-collection-as-a-single-entity">Processing collection as a single entity</h2>

<p>Now that <code class="language-plaintext highlighter-rouge">bwa-mem</code> has finished and generated a collection of BAM datasets we can continue to analyze the entire collection as a single <span class="notranslate">Galaxy</span> â€˜<em>item</em>â€™.</p>

<h3 id="ensuring-consistency-of-bam-dataset">Ensuring consistency of BAM dataset</h3>

<p>Letâ€™s perform cleanup of our BAM files with <code class="language-plaintext highlighter-rouge">cleanSam</code> utility from the <strong>Picard</strong> package:</p>

<p class="img-responsive"><img src="../../images/cleanSam.png" alt="cleanSam" /></p>

<p>If you look at the picture above carefully, you will see that the <strong>Select SAM/BAM dataset or dataset collection</strong> parameter is empty (it says <code class="language-plaintext highlighter-rouge">No sam or bam datasets available.</code>). This is because we do not have single SAM or BAM datasets in the history. Instead we have a collection. So all you need to do is to click on the <img src="../../images/folder.png" alt="folder" /> button and you will get our BAM collection selected:</p>

<p class="img-responsive"><img src="../../images/cleanSam_closeup.png" alt="cleanSam_closeup" /></p>

<p>Click <strong>Execute</strong>. As an output this tool will produce a collection contained cleaned data.</p>

<h3 id="retaining-proper-pairs">Retaining â€˜proper pairsâ€™</h3>

<p>Now letâ€™s clean the dataset further by only preserving truly paired <span class="notranslate">reads</span> (<span class="notranslate">reads</span> satisfying two requirements: (1) read is paired, and (2) it is mapped as a proper pair). For this we will use <code class="language-plaintext highlighter-rouge">Filter SAM or BAM</code> tools from <strong>SAMTools</strong> collection:</p>

<p class="img-responsive"><img src="../../images/filter.png" alt="filter" /></p>

<p>parameters should be set as shown below. By setting <span class="notranslate">mapping</span> quality to <code class="language-plaintext highlighter-rouge">20</code> we avoid <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to multiple locations and by using <strong>Filter on bitwise flag</strong> option we ensure that the resulting dataset will contain only properly paired <span class="notranslate">reads</span>. This operation will produce yet another collection containing now filtered datasets.</p>

<p class="img-responsive"><img src="../../images/filter_closeup.png" alt="filter_closeup" /></p>

<h3 id="merging-collection-into-a-single-dataset">Merging collection into a single dataset</h3>

<p>The beauty of BAM datasets is that they can be combined in a single entity using so called <em>Read group</em>. This allows to bundle <span class="notranslate">reads</span> from multiple experiments into a single dataset where read identity is maintained by labelling every sequence with <em>read group</em> tags. So letâ€™s finally reduce this collection to a single BAM dataset. For this we will use <code class="language-plaintext highlighter-rouge">MergeSamFiles</code> tool for the <code class="language-plaintext highlighter-rouge">Picard</code> suite:</p>

<p class="img-responsive"><img src="../../images/merge.png" alt="merge" /></p>

<p>Here we select the collection generated by the filtering tool described above:</p>

<p class="img-responsive"><img src="../../images/merge_closeup.png" alt="merge_closeup" /></p>

<p>This operation will <strong>not</strong> generate a collection. Instead, it will generate a single BAM dataset containing mapped <span class="notranslate">reads</span> from our four samples (<code class="language-plaintext highlighter-rouge">M117-bl</code>, <code class="language-plaintext highlighter-rouge">M117-ch</code>, <code class="language-plaintext highlighter-rouge">M117C1-bl</code>, and <code class="language-plaintext highlighter-rouge">M117C1-ch</code>).</p>

<h2 id="lets-look-at-what-weve-got">Letâ€™s look at what weâ€™ve got!</h2>

<p>So we have one BAM dataset combining everything weâ€™ve done so far. Letâ€™s look at the contents of this dataset using a genome browser. First, we will need to downsample the dataset to avoiding overwhelming the browser. For this we will use <code class="language-plaintext highlighter-rouge">Downsample SAM/BAM</code> tool:</p>

<p class="img-responsive"><img src="../../images/downsample.png" alt="downsample" /></p>

<p>Set <strong>Probability (between 0 and 1) that any given read will be kept</strong> to roughly <code class="language-plaintext highlighter-rouge">5%</code> (or <code class="language-plaintext highlighter-rouge">0.05</code>) using the slider control:</p>

<p><img src="../../images/downsample_closeup.png" alt="downsample_closeup" /></p>

<p>This will generate another BAM dataset containing only 5% of the original <span class="notranslate">reads</span> and much smaller as a result. Click on this dataset and you will see links to various genome browsers:</p>

<p class="img-responsive"><img src="../../images/browserLinks.png" alt="browserLinks" /></p>

<p>Click the <strong>Human hg38</strong> link in the <strong>display with IGV</strong> line as highlighted above (learn more about displaying <span class="notranslate">Galaxy</span> data in IGV with this <a href="https://vimeo.com/123442619#t=4m16s">movie</a>). Below is an example generated with IGV on these data. In this screenshot, <span class="notranslate">reads</span> are colored by read group (four distinct colors). A yellow inset displays additional information about a single read. One can see that this read corresponds to read group <code class="language-plaintext highlighter-rouge">M117-bl</code>.</p>

<p class="img-responsive"><img src="../../images/igv.png" alt="igv" /></p>

<h2 id="we-did-not-fake-this">We did not fake this:</h2>
<p>The history described in this page is accessible directly from here:</p>

<ul>
  <li>History <a href="https://usegalaxy.org/u/aun1/h/collections-full-analysis"><strong>Collections (full analysis)</strong></a></li>
</ul>

<p>From there you can import histories to make them your own.</p>

<h2 id="if-things-dont-work">If things donâ€™t workâ€¦</h2>
<p>â€¦you need to complain. Use <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span>â€™s Help Channel</a> to do this.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Jobs running on data collections use the same settings for each dataset in the collection.</p>
</li>
                            
                            <li><p>For example, FASTQ files can be combined into lists of dataset pairs. Each pair is made up of the forward reads dataset and the reverse reads dataset.</p>
</li>
                            
                            <li><p>In this tutorial, several tools were run on the list of dataset pairs, such as <code class="language-plaintext highlighter-rouge">bwa-mem</code>, <code class="language-plaintext highlighter-rouge">cleanSam</code>, <code class="language-plaintext highlighter-rouge">Filter SAM or BAM</code>, etc.</p>
</li>
                            
                            <li><p>When using collections, you have to click on the <code class="language-plaintext highlighter-rouge">batch input mode</code> button, to select one of the collections available in the history. Otherwise, the collections are not available in the drop-down list.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Using dataset collections (User Interface and Data Manipulation)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Anton Nekrutenko, 2021 <b>Using dataset collections (Galaxy Training Materials)</b>. <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{galaxy-interface-collections,
    author = "Anton Nekrutenko",
    title = "Using dataset collections (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and BjÃ¶rn GrÃ¼ning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Anton Nekrutenko)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/galaxy-interface/tutorials/collections/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
