---
username: bebatut
layout: training_philosophy
redirect_from:
  - /topics/instructors/philosophies/2018-10-19-bebatut
---

When I am giving a training using Galaxy (e.g. ref-based RNA-seq), I usually give a 10-20 min introduction to RNA, RNA-seq using the slides that in the transcriptomics topic.

Then I switch to the tutorial. I show the participants where to find the tutorial on the training material website. I introduce the tutorial story: explain that we want to reproduce an analysis from a paper, introduce this paper, talk about the datasets/samples we will use.

After that, I open Galaxy and I go step by step through the analysis with them:
- introduce the step: give an idea why we need this step
- talk a bit about the "theory" of this step
   - using the schemas in the tutorial to explain for example RNA-seq mapping
   - using legos to explain mapping (and let them try to map lego reads on a reference genome)
- open the tools in Galaxy and going through the parameters with them, explaining the main parameters and their potential effect
- run the tool
- inspect the generated output with them, insist on each new format
- ask the questions that are in the question boxes and/or to do an extra step (identify a new tool, identify and use different parameters, etc)

At some point, I introduce the step and explain the purpose of the step, but let them do it by themselves.

At the end of the tutorial, I spent some time to go through again each step, asking them again why this step, what were the inputs/outputs and in which format, repeating again :smile:

Here is the way I usually give a data analysis training using Galaxy. What is your way?
