---
layout: introduction_slides
topic_name: NGS-QC
logo: "GTN"
---

# Why Quality Control?

---

### Where my data come from?

![](../images/ecker_2012.png)

<small>[*Ecker et al, Nature, 2012*](http://www.nature.com/nature/journal/v489/n7414/full/489052a.html)</small>

---

### From experiments to data

![](../images/from_exp_to_data.png)

Quality control = First step of the bioinformatics analyses

---

### My sequences? <br>Fasta

```
> Identifiant (Commentaire)
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
> Identifiant2 (Commentaire)
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
XX
```

---

### Quality score

Measure of the quality of the identification of the nucleobases <br>generated by automated DNA sequencing

<small>

Phred Quality Score | Probability of incorrect base call | Base call accuracy
--- | --- | ---
10 | 1 in 10 | 90%
20 | 1 in 100 | 99%
30 | 1 in 1000 | 99.9%
40 | 1 in 10,000 | 99.99%
50 | 1 in 100,000 | 99.999%
60 | 1 in 1,000,000 | 99.9999%

</small>

---

### Quality score

![](../images/quality_score.png)

---

### Quality score encoding

![](../images/quality_score_encoding.png)

---

### My sequences? <br>FastQ

```
@ Identifiant (Commentaire)
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ
@ Identifiant2 (Commentaire)
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+
QQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQQ
```       

---

## How to check the quality of my sequences?

---

### Quality score<br>Per base sequence quality

![](../images/per_base_sequence_quality_good.png)

<i class="fa fa-thumbs-up"></i> Good quality score

---

### Quality score<br>Per base sequence quality

![](../images/per_base_sequence_quality_bad.png)

<i class="fa fa-thumbs-down"></i> Bad quality score

---

### Quality score<br>Per base sequence quality

![](../images/per_base_sequence_quality_medium.png)

<i class="fa fa-thumbs-up"></i> <i class="fa fa-thumbs-down"></i> Intermediate quality score

---

### Quality score<br>Per sequence quality scores

![](../images/per_sequence_quality.png)

---

### Quality score<br>Per tile sequence quality

![](../images/per_tile_quality.png)

???
In Illumina library, the original sequence identifiant is retained.Encoded in these is the flowcell tile from which each read came.

There could be transient problems such as bubbles going through the flowcell, or they could be more permanent problems such as smudges on the flowcell or debris inside the flowcell lane.

This graph will only appear with Illumina library which retains its original sequence identifiers. The graph allows to look at the quality scores from each tile across all bases to see if there was a loss in quality associated with only one part of the flowcell. The plot shows the deviation from the average quality for each tile. The colours are on a cold to hot scale, with cold colours being positions where the quality was at or below the average for that base in the run, and hotter colours indicate that a tile had worse qualities than other tiles for that base. In the example below you can see that certain tiles show consistently poor quality. A good plot should be blue all over.

---

### Also to check: Sequence content<br>Per base sequence content

![](../images/per_base_sequence_content.png)

???
Per Base Sequence Content checks out the proportion of each base position in a sequence file for which each of the four normal DNA bases has been called. In a random library there would be little to no difference between the different bases of a sequence run. The relative amount of each base should reflect the overall amount of these bases, but in any case they should not be hugely imbalanced from each other. It’s worth noting that some types of library will always produce biased sequence composition, normally at the start of the read. Libraries produced by priming using random hexamers (including nearly all RNA-Seq libraries) and those which were fragmented using transposases inherit an intrinsic bias in the positions at which reads start. This bias does not concern an absolute sequence, but instead provides enrichment of a number of different K-mers at the 5’ end of the reads. Whilst this is a true technical bias, it isn’t something which can be corrected by trimming and in most cases doesn’t seem to adversely affect the downstream analysis. It will however produce a warning or error in this module.

There are a number of common scenarios for these issues:

- Overrepresented sequences:
- Biased fragmentation
- Biased composition libraries
- Aggressive adapter trimming

---

### Also to check: Sequence content<br>Per sequence GC content

![](../images/per_sequence_GC_content.png)

???
The GC content distribution of most samples should follow a normal distribution. In some cases, a bi-modal distribution can be observed, especially for metagenomic data sets. An unusually shaped distribution could indicate a contaminated library or some other kinds of biased subset. A normal distribution which is shifted indicates some systematic bias which is independent of base position. If there is a systematic bias which creates a shifted normal distribution then this won’t be flagged as an error by the module since it doesn’t know what your genome’s GC content should be.

Issues in the GC content distribution usually indicate a problem with the library. Sharp peaks on an otherwise smooth distribution are normally the result of a specific contaminant (adapter dimers for example), which may well be picked up by the overrepresented sequences module. Broader peaks may represent contamination with a different species.

---

### Also to check: Sequence content<br>Per base N content

![](../images/per_base_N_content.png)

???
Sequences can contain the ambiguous base N for positions that could not be identified as a particular base. A high number of Ns can be a sign for a low quality sequence or even dataset. If no quality scores are available, the sequence quality can be inferred from the percent of Ns found in a sequence or dataset.

If a sequencer is unable to make a base call with sufficient confidence then it will normally substitute an N rather than a conventional base call. It’s not unusual to see a very low proportion of Ns appearing in a sequence, especially nearer the end of a sequence. However, if this proportion rises above a few percent it suggests that the analysis pipeline was unable to interpret the data well enough to make valid base calls.

---

### Also to check: Sequence content<br>Per base N content

![](../images/per_base_N_content.png)

???
Sequences can contain the ambiguous base N for positions that could not be identified as a particular base. A high number of Ns can be a sign for a low quality sequence or even dataset. If no quality scores are available, the sequence quality can be inferred from the percent of Ns found in a sequence or dataset.

If a sequencer is unable to make a base call with sufficient confidence then it will normally substitute an N rather than a conventional base call. It’s not unusual to see a very low proportion of Ns appearing in a sequence, especially nearer the end of a sequence. However, if this proportion rises above a few percent it suggests that the analysis pipeline was unable to interpret the data well enough to make valid base calls.

---

### Also to check: Sequence length<br>Sequence Length Distribution

![](../images/sequence_length_distribution.png)

???
Some high throughput sequencers generate sequence fragments of uniform length, but others can contain reads of wildly varying lengths. The length distribution can be then used as quality measure. You would expect a normal distribution for the best result. However, most sequencing results show a slowly increasing and then a steep falling distribution.

FastQC generates a graph showing the distribution of fragment sizes in the file which was analysed. In many cases this will produce a simple graph showing a peak only at one size, but for variable length FastQ files this will show the relative amounts of each different size of sequence fragment.

This module will raise a warning if all sequences are not the same length. This module will raise an error if any of the sequences have zero length.

---

### Also to check: Duplicated sequences

![](../images/sequence_duplication_levels.png)

???
In genomic projects, sequence duplication is investigated. Duplicated can arise when there are too few fragments present at any stage prior to sequencing. 

---

### Also to check: Tag sequences<br>Adapter contamination

![](../images/adapter_content.png)

???
Tag sequences are artifacts at the ends of sequence reads such as multiplex identifiers, adapters, and primer sequences that were introduced during pre-amplification with primer-based methods. The base frequencies across the reads present an easy way to check for tag sequences. If the distribution seems uneven (high frequencies for certain bases over several positions), it could indicate some residual tag sequences. This doesn’t indicate a problem as such - just that the sequences will need to be adapter trimmed before proceeding with any downstream analysis.

To investigate tag or adapter content, FastQC generates a plot showing a cumulative percentage count of the proportion of the library which has seen each of the adapter sequences at each position. Once a sequence has been seen in a read it is counted as being present right through to the end of the read so the percentages you see will only increase as the read length goes on.

---

### Also to check: Tag sequences<br>Kmer Content

![](../images/kmer_content.png)

???
An other way to find tag sequences is to look at kmer content and find those which do not have even coverage through the length of your reads and could correspond to tag sequences.

---

## How to improve the quality of my sequences?

---

### Sequence quality improvements

- Filtering of sequences
    - with small mean quality score
    - too small
    - with too many N bases
    - based on their GC content
    - ...
- Cutting/Trimming sequences
    - from low quality score parts
    - tails
    - ...



