<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>RNA Seq Counts to Viz in R</title>
        
        <script>
            (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
            (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
            m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
            })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
            ga('create', 'UA-45719423-18' , 'auto');
            ga('send', 'pageview');
        </script>
        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: RNA Seq Counts to Viz in R" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        













<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="EN">
                            EN
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-counts-to-viz-in-r%2Ftutorial.html&edit-text=&act=url" title="">
                                FR
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-counts-to-viz-in-r%2Ftutorial.html&edit-text=&act=url" title="">
                                JA
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-counts-to-viz-in-r%2Ftutorial.html&edit-text=&act=url" title="">
                                ES
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-counts-to-viz-in-r%2Ftutorial.html&edit-text=&act=url" title="">
                                PT
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-counts-to-viz-in-r%2Ftutorial.html&edit-text=&act=url" title="">
                                AR
                            </a>
                            
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Chat on Gitter
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "RNA Seq Counts to Viz in R",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "/blob//LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "courseCode": "transcriptomics / rna-seq-counts-to-viz-in-r / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'RNA Seq Counts to Viz in R' tutorial",
  "isPartOf": {
    "@type": "Course",
    "name": "RNA Seq Counts to Viz in R",
    "description": "RNA Seq Counts to Viz in R",
    "learningResourceType": "tutorial",
    "interactivityType": "expositive",
    "provider": {
      "@type": "Organization",
      "email": "galaxytrainingnetwork@gmail.com",
      "name": "Galaxy Training Network",
      "url": "https://galaxyproject.org/teach/gtn/"
    }
  },
  "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html",
  "timeRequired": "PT3H",
  "description": "RNA Seq Counts to Viz in R\\nThe questions this  addresses are:\n - How can I manipulate data using R in Galaxy?\n - How can I create neat visualizations of the data?\n - How do I get started with tabular data from Galaxy in R?\n\n\\nThe objectives are:\n - Be able to load and explore the shape and contents of a tabular dataset using base R functions.\n - Understand factors and how they can be used to store and work with categorical data.\n - Apply common `dplyr` functions to manipulate data in R.\n - Employ the ‘pipe’ operator to link together a sequence of functions.\n - Describe the role of data, aesthetics, geoms, and layers in `ggplot` functions.\n - Customize plot scales, titles, subtitles, themes, fonts, layout, and orientation.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/galaxy-ui/tutorials/rstudio/tutorial.html",
      "name": "RStudio in Galaxy",
      "description": "Hands-on for 'RStudio in Galaxy' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/introduction/tutorials/r-basics/tutorial.html",
      "name": "R basics in Galaxy",
      "description": "Hands-on for 'R basics in Galaxy' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html",
      "name": "Reference-based RNA-Seq data analysis",
      "description": "Hands-on for 'Reference-based RNA-Seq data analysis' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/3477564/",
      "name": "Training data for RNA Seq Counts to Viz in R tutorial"
    }
  ],
  "author": [
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Fotis E. Psomopoulos"
    },
    {
      "@type": "Person",
      "name": "Toby Hodges"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Bérénice Batut"
    },
    {
      "@type": "Person",
      "name": "Fotis E. Psomopoulos"
    },
    {
      "@type": "Person",
      "name": "Toby Hodges"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://galaxyproject.github.io//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>RNA Seq Counts to Viz in R</h1>
        
        <div class="alert alert-warning" role="alert">
            <h4 class="alert-heading">Under Development!</h4>
            <p>This tutorial is not in its final state. The content may change a lot in the next months.
            Because of this status, it is also not listed in the topic pages.</p>
        </div>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame#bebatut" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="Bérénice Batut">Bérénice Batut</a>, <a href="/training-material/hall-of-fame#fpsom" class="contributor-badge"><img src="https://avatars.githubusercontent.com/fpsom" alt="Fotis E. Psomopoulos">Fotis E. Psomopoulos</a>, <a href="/training-material/hall-of-fame#tobyhodges" class="contributor-badge"><img src="https://avatars.githubusercontent.com/tobyhodges" alt="Toby Hodges">Toby Hodges</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How can I manipulate data using R in Galaxy?</p>
</li>
            
            <li><p>How can I create neat visualizations of the data?</p>
</li>
            
            <li><p>How do I get started with tabular data from Galaxy in R?</p>
</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Be able to load and explore the shape and contents of a tabular dataset using base R functions.</p>
</li>
            
            <li><p>Understand factors and how they can be used to store and work with categorical data.</p>
</li>
            
            <li><p>Apply common <code class="highlighter-rouge">dplyr</code> functions to manipulate data in R.</p>
</li>
            
            <li><p>Employ the ‘pipe’ operator to link together a sequence of functions.</p>
</li>
            
            <li><p>Describe the role of data, aesthetics, geoms, and layers in <code class="highlighter-rouge">ggplot</code> functions.</p>
</li>
            
            <li><p>Customize plot scales, titles, subtitles, themes, fonts, layout, and orientation.</p>
</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/galaxy-ui">User Interface and Features</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Reference-based RNA-Seq data analysis:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"><i class="fa fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <div>
                <div><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></div>
                <div id="supporting_materials">
                    <ul><li><div>
                        
                        <div>
                            <a class="topic-icon" href="/training-material/topics/transcriptomics/slides/introduction.html" title="Introduction slides">
                                <i class="fa fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Introduction Slides
                            </a>
                        </div>
                        

                        

                        
                        


                        
                    </div></li></ul>
                </div>

                <p><strong> <i class="fa fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Oct 8, 2019 </p>
            </div>
            

        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>This tutorial is significantly based on <a href="https://carpentries.org">the Carpentries</a> <a href="https://datacarpentry.org/genomics-r-intro/">“Intro to R and RStudio for Genomics”</a> lesson</p>

</blockquote>

<p>With <span class="notranslate">RNA</span>-Seq data analyis, we generated tables containing list of DE genes, their <span class="notranslate">expression</span>, some statistics, etc. We can manipulate these tables using <span class="notranslate">Galaxy</span>, as we saw in some tutorials, e.g. <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">“Reference-based <span class="notranslate">RNA</span>-Seq data analysis”</a>, and create some visualisations.</p>

<p>Sometimes we want to have some customizations on visualization, some complex table manipulations or some statistical analysis. If we can not find a <span class="notranslate">Galaxy</span> tools for that or the right parameters, we may need to use programming languages as R or Python.</p>

<p>R is one of the most widely-used and powerful programming languages in bioinformatics. R especially shines where a variety of statistical tools are required (e.g. <span class="notranslate">RNA</span>-Seq, population genomics, etc.) and in the generation of publication-quality graphs and figures. Rather than get into an R vs. Python debate (both are useful), keep in mind that many of the concepts you will learn apply to Python and other programming languages.
At more than 20 years old, R is fairly mature and <a href="https://www.tiobe.com/tiobe-index/r/">growing in popularity</a>. However, programming isn’t a popularity contest. Here are key advantages of analyzing data in R:</p>

<ul>
  <li>
    <p><strong>R is free and <a href="https://en.wikipedia.org/wiki/Open-source_software">open source</a></strong></p>

    <p>R is free - an advantage if you are at an institution where you have to pay for your own MATLAB or SAS license. It is important to your colleagues in parts of the world where expensive software is inaccessible.</p>

    <p>Open source means that you can see and reuse the source. Thanks to that, R is actively developed by a community (see <a href="https://www.r-project.org/">r-project.org</a>), and there are regular updates.</p>
  </li>
  <li>
    <p><strong>R is widely used</strong></p>

    <p>Ok, maybe programming is a popularity contest. Because, R is used in many areas (not just bioinformatics), you are more likely to find help online when you need it. Chances are, almost any error message you run into, someone else has already experienced.</p>
  </li>
  <li>
    <p><strong>R is powerful</strong></p>

    <p>R runs on multiple platforms (Windows/MacOS/Linux). It can work with much larger datasets than popular sp<span class="notranslate">reads</span>heet programs like Microsoft Excel, and because of its scripting capabilities is far more reproducible. Also, there are thousands of available software packages for science, including genomics and other areas of life science.</p>
  </li>
</ul>

<blockquote class="comment">
  <h3 id="comment-on-the-content-of-the-tutorial"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> On the content of the tutorial</h3>

  <p>We believe that every learner can <strong>achieve competency with R</strong>. You have reached competency when you find that you are able to <strong>use R to handle common analysis challenges in a reasonable amount of time</strong> (which includes time needed to look at learning materials, search for answers online, and ask colleagues for help). As you spend more time using R (there is no substitute for regular use and practice) you will find yourself gaining competency and even expertise. The more familiar you get, the more complex the analyses you will be able to carry out, with less frustration, and in less time - the fantastic world of R awaits you!</p>

  <p>Nobody wants to learn how to use R. People want to learn how to use R to answer their own research questions! Ok, maybe some folks learn R for R’s sake, but these lessons assume that you want to start analyzing genomic data as soon as possible. Given this, there are many valuable pieces of information about R that we simply won’t have time to cover. Hopefully, we will clear the hurdle of giving you just enough knowledge to be dangerous, which can be a high bar in R! We suggest you look into the additional learning materials in the box below.</p>

  <p><strong>Some R skills we will <em>not</em> cover in these lessons</strong></p>

  <ul>
    <li>How to create and work with R matrices and R lists</li>
    <li>How to create and work with loops and conditional statements, and the “apply” of functions (which are super useful, read more <a href="https://www.r-bloggers.com/r-tutorial-on-the-apply-family-of-functions/">here</a>)</li>
    <li>How to do basic string manipulations (e.g. finding patterns in text using grep, replacing text)</li>
    <li>How to plot using the default R graphic tools (we <em>will</em> cover plot creation, but will do so using the popular plotting package <code class="highlighter-rouge">ggplot2</code>)</li>
    <li>How to use advanced R statistical functions</li>
  </ul>

  <blockquote class="tip">
    <h3 id="tip-tip-where-to-learn-more"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Where to learn more</h3>

    <p>The following are good resources for learning more about R. Some of them can be quite technical, but if you are a regular R user you may ultimately need this technical knowledge.</p>
    <ul>
      <li><a href="https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf">R for Beginners</a>, by Emmanuel Paradis: a great starting point</li>
      <li><a href="https://cran.r-project.org/manuals.html">The R Manuals</a>, by the R project people</li>
      <li><a href="https://cran.r-project.org/other-docs.html">R contributed documentation</a>, also linked to the R project, with materials available in several languages</li>
      <li><a href="http://r4ds.had.co.nz/">R for Data Science</a>: a wonderful collection by noted R educators and developers Garrett Grolemund and Hadley Wickham</li>
      <li><a href="https://peerj.com/collections/50-practicaldatascistats/">Practical Data Science for Stats</a>: not exclusively about R usage, but a nice collection of pre-prints on data science and applications for R</li>
      <li><a href="https://software-carpentry.org/lessons/">Programming in R Software Carpentry lesson</a>: several Software Carpentry lessons in R to choose from</li>
      <li><a href="https://www.datacamp.com/courses/free-introduction-to-r">Data Camp Introduction to R</a>: a fun online learning platform for Data Science, including R.</li>
    </ul>
  </blockquote>
</blockquote>

<p>In this tutorial, we will take the list of DE genes extracted from DESEq2’s output that we generated in the <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">“Reference-based <span class="notranslate">RNA</span>-Seq data analysis” tutorial</a>, manipulate it and create some visualizations.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#get-familiar-with-the-annotated-de-genes-table-in-r" id="markdown-toc-get-familiar-with-the-annotated-de-genes-table-in-r">Get familiar with the annotated DE genes table in R</a>    <ol>
      <li><a href="#import-tabular-data-into-r" id="markdown-toc-import-tabular-data-into-r">Import tabular data into R</a></li>
      <li><a href="#summarize-and-determine-the-structure-of-a-data-frame" id="markdown-toc-summarize-and-determine-the-structure-of-a-data-frame">Summarize and determine the structure of a data frame</a></li>
      <li><a href="#factors" id="markdown-toc-factors">Factors</a></li>
      <li><a href="#subset-a-data-frame" id="markdown-toc-subset-a-data-frame">Subset a data frame</a></li>
      <li><a href="#change-type-of-columns" id="markdown-toc-change-type-of-columns">Change type of columns</a></li>
      <li><a href="#inspect-a-numerical-column-using-math-sorting-renaming" id="markdown-toc-inspect-a-numerical-column-using-math-sorting-renaming">Inspect a numerical column using math, sorting, renaming</a></li>
      <li><a href="#save-the-data-frame-to-a-file" id="markdown-toc-save-the-data-frame-to-a-file">Save the data frame to a file</a></li>
    </ol>
  </li>
  <li><a href="#aggregate-and-analyze-with-dplyr" id="markdown-toc-aggregate-and-analyze-with-dplyr">Aggregate and Analyze with dplyr</a>    <ol>
      <li><a href="#select-columns-and-filter-rows" id="markdown-toc-select-columns-and-filter-rows">Select columns and filter rows</a></li>
      <li><a href="#create-columns" id="markdown-toc-create-columns">Create columns</a></li>
      <li><a href="#group-and-summarize-data" id="markdown-toc-group-and-summarize-data">Group and summarize data</a></li>
      <li><a href="#reshape-data-frames" id="markdown-toc-reshape-data-frames">Reshape data frames</a></li>
    </ol>
  </li>
  <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a>    <ol>
      <li><a href="#volcano-plot" id="markdown-toc-volcano-plot">Volcano plot</a></li>
      <li><a href="#volcano-plot-by-chromosomes" id="markdown-toc-volcano-plot-by-chromosomes">Volcano plot by chromosomes</a></li>
      <li><a href="#barplot-of-the-number-of-de-genes" id="markdown-toc-barplot-of-the-number-of-de-genes">Barplot of the number of DE genes</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<p>Before diving in the tutorial, we need to open RStudio. If you do not know how or never interacted with RStudio, please follow the <a href="/training-material/topics/galaxy-ui/tutorials/rstudio/tutorial.html">dedicated tutorial</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-open-rstudio"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Open RStudio</h3>

  <ol>
    <li>
      <p>Start RStudio in <span class="notranslate">Galaxy</span></p>

      <blockquote class="tip">

        <h3 id="tip-tip-launch-rstudio-in-galaxy"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio in <span class="notranslate">Galaxy</span></h3>
        <ol>
          <li>Open the Rstudio tool <i class="fa fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here</a></li>
          <li>Click Execute
<!-- TODO: update to tool panel search --></li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-launch-rstudio-cloud-if-not-available-on-galaxy"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio Cloud if not available on <span class="notranslate">Galaxy</span></h3>

        <p>If RStudio is not available on the <span class="notranslate">Galaxy</span> instance:</p>
        <ol>
          <li>Register for <a href="https://login.rstudio.cloud/login?redirect=%2Foauth%2Fauthorize%3Fredirect_uri%3Dhttps%253A%252F%252Frstudio.cloud%252Flogin%26client_id%3Drstudio-cloud%26response_type%3Dcode%26show_auth%3D0%26show_login%3D1%26show_setup%3D1&amp;setup=True">RStudio Cloud</a> or log in with your Github or Google account</li>
          <li>Create a new project</li>
        </ol>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="get-familiar-with-the-annotated-de-genes-table-in-r">Get familiar with the annotated DE genes table in R</h1>

<p>A substantial amount of the data we work with in science is tabular data, i.e. data arranged in rows and columns - also known as sp<span class="notranslate">reads</span>heets.</p>

<blockquote class="comment">
  <h3 id="comment-few-principles-when-working-with-data"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Few principles when working with data</h3>
  <p>We want to remind you of a few principles before we work with our first set of example data:</p>

  <ol>
    <li>
      <p><strong>Keep raw data separate from analyzed data</strong></p>

      <p>This is principle number one, because if you can’t tell which files are the original raw data, you risk making some serious mistakes (e.g. drawing conclusion from data which have been manipulated in some unknown way).</p>

      <p>When you work with data in R, you are not changing the original file which you loaded the data from. This is different than (for example) working with a sp<span class="notranslate">reads</span>heet program where changing the value of the cell leaves you one “save”-click away from overwriting the original file. You have to purposely use a writing function (e.g. <code class="highlighter-rouge">write.csv()</code>) to save data loaded into R. In that case, be sure to save the manipulated data into a new file. More on this later in the lesson.</p>
    </li>
    <li>
      <p><strong>Keep sp<span class="notranslate">reads</span>heet data Tidy</strong></p>

      <p>The simplest principle of <em>Tidy</em> data is that we have one row in our sp<span class="notranslate">reads</span>heet for each observation or sample, and one column for every variable that we measure or report on. As simple as this sounds, it’s very easily violated. Most data scientists agree that significant amounts of their time is spent tidying data for analysis.</p>
    </li>
    <li>
      <p><strong>Trust but verify</strong></p>

      <p>Finally, while you don’t need to be paranoid about data, you should have a plan for how you will prepare it for analysis. <strong>This a focus of this lesson</strong>. You probably already have a lot of intuition, expectations, assumptions about your data - the range of values you expect, how many values should have been recorded, etc. Of course, as the data get larger our human ability to keep track will start to fail (and yes, it can fail for small data sets too). R will help you to examine your data so that you can have greater confidence in your analysis, and its reproducibility.</p>
    </li>
  </ol>

</blockquote>

<h2 id="import-tabular-data-into-r">Import tabular data into R</h2>

<p>There are several ways to import data into R. For our purpose here, we will focus on using the tools every R installation comes with (so called “base” R) to import a comma-delimited file containing the results of our variant calling <span class="notranslate">workflow</span>. We will need to load the sheet using a function called <code class="highlighter-rouge">read.csv()</code>.</p>

<blockquote class="question">
  <h3 id="question-questions-review-the-arguments-of-the-readcsv-function"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions: Review the arguments of the <code class="highlighter-rouge">read.csv()</code> function</h3>

  <p>Before using the <code class="highlighter-rouge">read.csv()</code> function, use R’s help feature to answer the following questions.</p>

  <blockquote class="tip">
    <h3 id="tip-getting-help-in-r"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Getting help in R</h3>
    <p>Entering <code class="highlighter-rouge">?</code> before the function name and then running that line will bring up the help documentation. Also, when reading this particular help be careful to pay attention to the <code class="highlighter-rouge">read.csv</code> <span class="notranslate">expression</span> under the <strong>Usage</strong> heading. Other answers will be in the <strong>Arguments</strong> heading.</p>
  </blockquote>

  <ol>
    <li>What is the default parameter for <code class="highlighter-rouge">header</code> in the <code class="highlighter-rouge">read.csv()</code> function?</li>
    <li>What argument would you have to change to read a file that was delimited by semicolons (<code class="highlighter-rouge">;</code>) rather than commas?</li>
    <li>What argument would you have to change to read file in which numbers used commas for decimal separation (i.e. 1,00)?</li>
    <li>What argument would you have to change to read in only the first 10,000 rows of a very large file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The <code class="highlighter-rouge">read.csv() </code>function has the argument <code class="highlighter-rouge">header</code> set to <code class="highlighter-rouge">TRUE</code> by default, this means the function always assumes the first row is header information, (i.e. column names)</li>
      <li>The <code class="highlighter-rouge">read.csv()</code> function has the argument ‘sep’ set to “,”. This means the function assumes commas are used as delimiters, as you would expect. Changing this parameter (e.g. <code class="highlighter-rouge">sep=";"</code>) would now interpret semicolons as delimiters.</li>
      <li>Although it is not listed in the <code class="highlighter-rouge">read.csv()</code> usage, <code class="highlighter-rouge">read.csv()</code> is a “version” of the function <code class="highlighter-rouge">read.table()</code> and accepts all its arguments. If you set <code class="highlighter-rouge">dec=","</code> you could change the decimal operator. We’d probably assume the delimiter is some other character.</li>
      <li>You can set <code class="highlighter-rouge">nrow</code> to a numeric value (e.g. <code class="highlighter-rouge">nrow=10000</code>) to choose how many rows of a file you read in. This may be useful for very large files where not all the data is needed to test some data cleaning steps you are applying.</li>
    </ol>

    <p>Hopefully, this exercise gets you thinking about using the provided help documentation in R. There are many arguments that exist, but which we wont have time to cover. Look here to get familiar with functions you use frequently, you may be surprised at what you find they can do.</p>

  </blockquote>
</blockquote>

<p>Now, let’s read the file with the annotated differentially expressed genes that was produced in the “<a href="https://galaxyproject.github.io/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">Reference-based <span class="notranslate">RNA</span>-Seq data analysis</a>” tutorial.</p>

<!--
> ### <i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Import dataset into Galaxy
> 1. In Galaxy, upload the following file from Zenodo:
>    ```
>    https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular
>    ```
> 2. Remember the dataset number (e.g. `2` in the example below)
>    ![](../../images/rna-seq-counts-to-viz-in-r/dataset_number.png)
>
{: .hands_on}
-->

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-the-annotated-differentially-expressed-genes"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read the annotated differentially expressed genes</h3>
  <ol>
    <li>Create a new script (if needed)</li>
    <li>
      <p>Read the tabular file in an object called <code class="highlighter-rouge">annotatedDEgenes</code>. We can import directly by URL:</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## read in a CSV file and save it as 'annotatedDEgenes'</span><span class="w">
</span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p>The first argument to pass to our <code class="highlighter-rouge">read.csv()</code> function is the file path for our data. The file path must be in quotes and now is a good time to remember to use tab autocompletion. If you use tab autocompletion you avoid typos and errors in file paths. Use it!</p>
    </li>
    <li>Inspect the <strong>Environment</strong> panel</li>
  </ol>

  <blockquote class="tip">
    <h3 id="tip-upload-files-from-a-galaxy-history-to-rstudio"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Upload files from a <span class="notranslate">Galaxy</span> history to RStudio</h3>
    <p>Rstudio in <span class="notranslate">Galaxy</span> provides some special functions to import and export from your history.</p>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">gx_get</span><span class="p">(</span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="c1"># will import dataset number 2 from your history</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>In the <strong>Environment</strong> panel, you should have the <code class="highlighter-rouge">annotatedDEgenes</code> object, listed as 130 obs. (observations/rows) of 1 variable (column) - so the command worked (sort of)!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-the-annotateddegenes-file"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect the annotatedDEgenes file</h3>
  <ol>
    <li>Double-click on the name of the object in the <strong>Environment</strong> panel</li>
  </ol>

</blockquote>

<p>It will open a view of the data in a new tab on the top-left panel.</p>

<p>As you can see, there is a problem with how the data has been loaded. The table should contain 130 observations of 13 variables.</p>

<blockquote class="question">
  <h3 id="question-question-identify-the-issue-with-data"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question: Identify the issue with data</h3>

  <ol>
    <li>What is wrong?</li>
    <li>How should you adjust the parameters for <code class="highlighter-rouge">read.csv()</code> in order to produce the intended output?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The data file was not delimited by commas (<code class="highlighter-rouge">,</code>) which is the default expected delimiter for <code class="highlighter-rouge">read.csv()</code>. Instead it seems like it’s delimited by “white space”, i.e. spaces and/or tabs.</li>
      <li>We need to set the correct delimiter using the parameter <code class="highlighter-rouge">sep</code>. Given that we are not sure what the actual delimiter is, we could try both options, i.e. <code class="highlighter-rouge">sep=" "</code> (<em>space</em>) and <code class="highlighter-rouge">sep="\t"</code> (<em>tab</em>).</li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-correctly-tabular-file"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read correctly tabular file</h3>
  <ol>
    <li>
      <p>Read the file given it’s a tabular file</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>Inspect the object in the <strong>Environment</strong> panel</li>
    <li>
      <p>Open the object</p>

      <p>You should have now a nice table</p>
    </li>
    <li>
      <p>Check the column names</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s2">"GeneID"</span><span class="w">     </span><span class="s2">"Base.mean"</span><span class="w">  </span><span class="s2">"log2.FC."</span><span class="w">   </span><span class="s2">"StdErr"</span><span class="w">     </span><span class="s2">"Wald.Stats"</span><span class="w">
</span><span class="p">[</span><span class="m">6</span><span class="p">]</span><span class="w"> </span><span class="s2">"P.value"</span><span class="w">    </span><span class="s2">"P.adj"</span><span class="w">      </span><span class="s2">"Chromosome"</span><span class="w"> </span><span class="s2">"Start"</span><span class="w">      </span><span class="s2">"End"</span><span class="w">
</span><span class="p">[</span><span class="m">11</span><span class="p">]</span><span class="w"> </span><span class="s2">"Strand"</span><span class="w">     </span><span class="s2">"Feature"</span><span class="w">    </span><span class="s2">"Gene.name"</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>Congratulations! You’ve successfully loaded your data into RStudio!</p>

<h2 id="summarize-and-determine-the-structure-of-a-data-frame">Summarize and determine the structure of a data frame</h2>

<p>Tabular data are stored in R using <strong>data frame</strong>. A data frame could also be thought of as a collection of vectors, all of which have the same length.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-learn-more-about-our-data-frame"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Learn more about our data frame</h3>
  <ol>
    <li>
      <p>Get summary statistics of <code class="highlighter-rouge">annotatedDEgenes</code> using <code class="highlighter-rouge">summary</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## get summary statistics on a data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">summary</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span><span class="n">GeneID</span><span class="w">            </span><span class="n">Base.mean</span><span class="w">          </span><span class="n">log2.FC.</span><span class="w">          </span><span class="n">StdErr</span><span class="w">            </span><span class="n">Wald.Stats</span><span class="w">
</span><span class="n">FBgn0000071</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">   </span><span class="m">19.15</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-4.1485</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.08433</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-30.741</span><span class="w">
</span><span class="n">FBgn0000079</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">100.29</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-1.3363</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.12849</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-10.005</span><span class="w">
</span><span class="n">FBgn0000116</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w">  </span><span class="m">237.99</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">-1.0272</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.16370</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">-4.982</span><span class="w">
</span><span class="n">FBgn0000406</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">1911.27</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">-0.2074</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">0.16432</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">-1.901</span><span class="w">
</span><span class="n">FBgn0000567</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">948.66</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">1.2203</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.19866</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">7.692</span><span class="w">
</span><span class="n">FBgn0001137</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">65114.84</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">2.6999</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">0.23292</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">27.566</span><span class="w">
</span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="w">    </span><span class="o">:</span><span class="m">124</span><span class="w">
</span><span class="n">P.value</span><span class="w">             </span><span class="n">P.adj</span><span class="w">               </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">              </span><span class="n">End</span><span class="w">                </span><span class="n">Strand</span><span class="w">
</span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2L</span><span class="o">:</span><span class="m">24</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">127448</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">140340</span><span class="w">   </span><span class="o">-:</span><span class="m">58</span><span class="w">
</span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2R</span><span class="o">:</span><span class="m">31</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7277516</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7279063</span><span class="w">   </span><span class="o">+:</span><span class="m">72</span><span class="w">
</span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr3L</span><span class="o">:</span><span class="m">27</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">13161546</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">13166253</span><span class="w">
</span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">4.201e-07</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">9.321e-06</span><span class="w">   </span><span class="n">chr3R</span><span class="o">:</span><span class="m">32</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">13436843</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">13446444</span><span class="w">
</span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">7.000e-12</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">3.700e-10</span><span class="w">   </span><span class="n">chrX</span><span class="w"> </span><span class="o">:</span><span class="m">16</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">19250429</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">19284637</span><span class="w">
</span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">9.418e-06</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">1.951e-04</span><span class="w">              </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31196915</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31203722</span><span class="w">

</span><span class="n">Feature</span><span class="w">              </span><span class="n">Gene.Name</span><span class="w">
</span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w">       </span><span class="o">:</span><span class="w">  </span><span class="m">3</span><span class="w">   </span><span class="n">Ama</span><span class="w">        </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
</span><span class="n">protein_coding</span><span class="o">:</span><span class="m">126</span><span class="w">   </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w">      </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
</span><span class="n">pseudogene</span><span class="w">    </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Ant2</span><span class="w">       </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="n">Argk</span><span class="w">       </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="n">bou</span><span class="w">        </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="w">    </span><span class="o">:</span><span class="m">124</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Get the structure of <code class="highlighter-rouge">annotatedDEgenes</code> using <code class="highlighter-rouge">str</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## get the structure of a data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">str</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span><span class="s1">'data.frame'</span><span class="o">:</span><span class="w">	</span><span class="m">130</span><span class="w"> </span><span class="n">obs.</span><span class="w"> </span><span class="n">of</span><span class="w">  </span><span class="m">13</span><span class="w"> </span><span class="n">variables</span><span class="o">:</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">GeneID</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">130</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"FBgn0000071"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">87</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">26</span><span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">96</span><span class="w"> </span><span class="m">65</span><span class="w"> </span><span class="m">62</span><span class="w"> </span><span class="m">125</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Base.mean</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">1087</span><span class="w"> </span><span class="m">6410</span><span class="w"> </span><span class="m">65115</span><span class="w"> </span><span class="m">2192</span><span class="w"> </span><span class="m">5430</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w">  </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">-4.15</span><span class="w"> </span><span class="m">-3</span><span class="w"> </span><span class="m">-2.38</span><span class="w"> </span><span class="m">2.7</span><span class="w"> </span><span class="m">-2.11</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">StdErr</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">0.1349</span><span class="w"> </span><span class="m">0.1043</span><span class="w"> </span><span class="m">0.0843</span><span class="w"> </span><span class="m">0.0979</span><span class="w"> </span><span class="m">0.0925</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">-30.7</span><span class="w"> </span><span class="m">-28.7</span><span class="w"> </span><span class="m">-28.2</span><span class="w"> </span><span class="m">27.6</span><span class="w"> </span><span class="m">-22.7</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">P.value</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">1.62e-207</span><span class="w"> </span><span class="m">9.42e-182</span><span class="w"> </span><span class="m">2.85e-175</span><span class="w"> </span><span class="m">2.85e-167</span><span class="w"> </span><span class="m">1.57e-114</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">P.adj</span><span class="w">     </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">1.39e-203</span><span class="w"> </span><span class="m">4.04e-178</span><span class="w"> </span><span class="m">8.15e-172</span><span class="w"> </span><span class="m">6.10e-164</span><span class="w"> </span><span class="m">2.70e-111</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Seqname</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"chr2L"</span><span class="p">,</span><span class="s2">"chr2R"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Start</span><span class="w">     </span><span class="o">:</span><span class="w"> </span><span class="n">int</span><span class="w">  </span><span class="m">24141394</span><span class="w"> </span><span class="m">10780892</span><span class="w"> </span><span class="m">26869237</span><span class="w"> </span><span class="m">10778953</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">820758</span><span class="w"> </span><span class="o">&gt;</span><span class="w">    </span><span class="m">6762592</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">End</span><span class="w">       </span><span class="o">:</span><span class="w"> </span><span class="n">int</span><span class="w">  </span><span class="m">24147490</span><span class="w"> </span><span class="m">10786958</span><span class="w"> </span><span class="m">26871995</span><span class="w"> </span><span class="m">10786907</span><span class="w"> </span><span class="m">13860001</span><span class="w"> </span><span class="m">31203722</span><span class="w"> </span><span class="m">24946636</span><span class="w"> </span><span class="m">22552113</span><span class="w"> </span><span class="m">821512</span><span class="w"> </span><span class="o">&gt;</span><span class="w">    </span><span class="m">6765261</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Strand</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"-"</span><span class="p">,</span><span class="s2">"+"</span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Feature</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"linc<span class="notranslate">RNA</span>"</span><span class="p">,</span><span class="s2">"protein_coding"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">130</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"Ama"</span><span class="p">,</span><span class="s2">"Amy-p"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">88</span><span class="w"> </span><span class="m">113</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">84</span><span class="w"> </span><span class="m">26</span><span class="w"> </span><span class="m">41</span><span class="w"> </span><span class="m">53</span><span class="w"> </span><span class="m">69</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>Ok, thats a lot to unpack! Some things to notice:</p>

<ul>
  <li>the object type <code class="highlighter-rouge">data.frame</code> is displayed in the first row along with its dimensions, in this case 130 observations (rows) and 13 variables (columns)</li>
  <li>Each variable (column) has a name (e.g. <code class="highlighter-rouge">GeneID</code>). This is followed by the object mode (e.g. <code class="highlighter-rouge">factor</code>, <code class="highlighter-rouge">int</code>, <code class="highlighter-rouge">num</code>, etc.). Notice that before each variable name there is a <code class="highlighter-rouge">$</code> - this will be important later.</li>
</ul>

<p>So from both <code class="highlighter-rouge">summary</code> and <code class="highlighter-rouge">str</code>, we know that our data frame had 13 variables that summarize the data.</p>

<ul>
  <li>
    <p><code class="highlighter-rouge">Base.mean</code>, <code class="highlighter-rouge">log2.FC.</code> and <code class="highlighter-rouge">P.value</code> variables (and several others) are numerical data</p>

    <p>We get summary statistics on the min and max values for these columns, as well as mean, median, and interquartile ranges</p>
  </li>
  <li>
    <p>Many of the other variables (e.g. <code class="highlighter-rouge">Strand</code>) are treated as categorical data</p>

    <p>This type has special treatment in R - more on this in a bit. The most frequent 6 different categories and the number of times they appear (e.g. the <code class="highlighter-rouge">Strand</code> called <code class="highlighter-rouge">-</code> appeared 58 times) are displayed. Another example is the <code class="highlighter-rouge">protein_coding</code> value for <code class="highlighter-rouge">Feature</code> which appeared in 126 observations.</p>
  </li>
</ul>

<h2 id="factors">Factors</h2>

<p>Factors, the final major data structure we will introduce, can be thought of as vectors which are specialized for categorical data. Given R’s specialization for statistics, this make sense since categorial and continuous variables usually have different treatments. Sometimes you may want to have data treated as a factor, but in other cases, this may be undesirable.</p>

<p>Since some of the data in our data frame are factors, lets see how factors work.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-a-factor-object"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect a factor object</h3>
  <ol>
    <li>
      <p>Extract the <code class="highlighter-rouge">feature</code> column to a new object called <code class="highlighter-rouge">feature</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## extract the "feature" column to a new object</span><span class="w">
</span><span class="n">feature</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Feature</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Inspect the first few items in our  using <code class="highlighter-rouge">head</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">
</span><span class="p">[</span><span class="m">5</span><span class="p">]</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">
</span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">pseudogene</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>What we get back are the items in our factor, and also something called “Levels”. <strong>Levels are the different categories contained in a factor</strong>. By default, R will organize the levels in a factor in alphabetical order. So the first level in this factor is “linc<span class="notranslate">RNA</span>”.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-the-structure-of-a-factor-object"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get the structure of a factor object</h3>
  <ol>
    <li>
      <p>Get the structure of the <code class="highlighter-rouge">feature</code> object</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">str</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span><span class="w">
</span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"linc<span class="notranslate">RNA</span>"</span><span class="p">,</span><span class="s2">"protein_coding"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>For the sake of efficiency, R stores the content of a factor as a vector of integers, which an integer is assigned to each of the possible levels. Recall levels are assigned in alphabetical order. In this case, the first item in our “Feature” object is “protein_coding”, which happens to be the 2nd level of our factor, ordered alphabeticaly. The 9th item in the list is “linc<span class="notranslate">RNA</span>”, which is the 1st level of our factor.</p>

<p>One of the most common uses for factors is to plot categorical values. For example, suppose we want to know how many of our differentially expressed genes correspond to different features.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-a-factor-object"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot a factor object</h3>
  <ol>
    <li>
      <p>Plot the number of levels in the <code class="highlighter-rouge">feature</code> object using <code class="highlighter-rouge">plot</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plot</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/rna-seq-counts-to-viz-in-r/featurePlot.png" alt="feature plot" /></p>

<p>This isn’t a particularly pretty example of a plot. We’ll be learning much more about creating nice, publication-quality graphics later.</p>

<h2 id="subset-a-data-frame">Subset a data frame</h2>

<p>Now we would like to know how to get specific values from data frames, and where necessary, change the mode of a column of values.</p>

<p>The first thing to remember is that a data frame is two-dimensional (rows and columns). Therefore, to select a specific value we will once again use <code class="highlighter-rouge">[]</code> (bracket) notation, but we will specify more than one value (except in some cases where we are taking a range).</p>

<blockquote class="question">
  <h3 id="question-subsetting-a-data-frame"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Subsetting a data frame</h3>

  <p>Run the following indices and functions to figure out what they return</p>

  <ol>
    <li><code class="highlighter-rouge">annotatedDEgenes[1, 1]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[2, 4]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[130, 13]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[2, ]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[-1, ]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[1:4, 1]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[1:10, c("Feature", "Gene.Name")]</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[, c("Gene.Name")]</code></li>
    <li><code class="highlighter-rouge">head(annotatedDEgenes)</code></li>
    <li><code class="highlighter-rouge">tail(annotatedDEgenes)</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes$GeneID</code></li>
    <li><code class="highlighter-rouge">annotatedDEgenes[annotatedDEgenes$Feature == "pseudogene", ]</code></li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[1, 1]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[2, 4]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">0.1043451</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[130, 13]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">130</span><span class="p">,</span><span class="m">13</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">CG10440</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">Ama</span><span class="w"> </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="n">Ant2</span><span class="w"> </span><span class="n">Argk</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w"> </span><span class="n">bou</span><span class="w"> </span><span class="n">bves</span><span class="w"> </span><span class="n">CAHbeta</span><span class="w"> </span><span class="n">cbt</span><span class="w"> </span><span class="n">CG10063</span><span class="w"> </span><span class="n">CG10365</span><span class="w"> </span><span class="n">CG10440</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">zye</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[2, ]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w">
        </span><span class="n">GeneID</span><span class="w"> </span><span class="n">Base.mean</span><span class="w">   </span><span class="n">log2.FC.</span><span class="w">   </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.577</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.1043451</span><span class="w">  </span><span class="m">-28.74863</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">    </span><span class="n">chrX</span><span class="w"> </span><span class="m">10780892</span><span class="w">
        </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">sesB</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[-1, ]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">-1</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w">
         </span><span class="n">GeneID</span><span class="w">   </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">
 </span><span class="m">2</span><span class="w">  </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.57713</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w"> </span><span class="m">-28.748628</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">    </span><span class="n">chrX</span><span class="w">
 </span><span class="m">3</span><span class="w">  </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.84056</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w"> </span><span class="m">-28.225437</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">   </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">4</span><span class="w">  </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.32237</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">  </span><span class="m">27.565988</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">    </span><span class="n">chrX</span><span class="w">
 </span><span class="m">5</span><span class="w">  </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.06728</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w"> </span><span class="m">-22.745964</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">   </span><span class="n">chr3L</span><span class="w">
 </span><span class="m">6</span><span class="w">  </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.90178</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w"> </span><span class="m">-21.889786</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">   </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">7</span><span class="w">  </span><span class="n">FBgn0035085</span><span class="w">   </span><span class="m">928.26381</span><span class="w"> </span><span class="m">-2.414074</span><span class="w"> </span><span class="m">0.11518516</span><span class="w"> </span><span class="m">-20.958204</span><span class="w">  </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w">   </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">8</span><span class="w">  </span><span class="n">FBgn0034736</span><span class="w">   </span><span class="m">330.38302</span><span class="w"> </span><span class="m">-3.018179</span><span class="w"> </span><span class="m">0.15815418</span><span class="w"> </span><span class="m">-19.083774</span><span class="w">  </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w">   </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">9</span><span class="w">  </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">955.45445</span><span class="w"> </span><span class="m">-2.334486</span><span class="w"> </span><span class="m">0.12423003</span><span class="w"> </span><span class="m">-18.791643</span><span class="w">  </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">   </span><span class="n">chr3L</span><span class="w">
 </span><span class="m">10</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w">   </span><span class="m">468.05793</span><span class="w">  </span><span class="m">2.360017</span><span class="w"> </span><span class="m">0.13564397</span><span class="w">  </span><span class="m">17.398615</span><span class="w">  </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">   </span><span class="n">chr3R</span><span class="w">
 </span><span class="c1"># ... with 120 more rows</span><span class="w">

       </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">2</span><span class="w">  </span><span class="m">10780892</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w">
 </span><span class="m">3</span><span class="w">  </span><span class="m">26869237</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
 </span><span class="m">4</span><span class="w">  </span><span class="m">10778953</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w">
 </span><span class="m">5</span><span class="w">  </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w">
 </span><span class="m">6</span><span class="w">  </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w">
 </span><span class="m">7</span><span class="w">  </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG3770</span><span class="w">
 </span><span class="m">8</span><span class="w">  </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG6018</span><span class="w">
 </span><span class="m">9</span><span class="w">    </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">        </span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w">     </span><span class="n">CR43883</span><span class="w">
 </span><span class="m">10</span><span class="w">  </span><span class="m">6762592</span><span class="w">  </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Ama</span><span class="w">
 </span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[1:4,1]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p>annotatedDEgenes[1:10,c(“Feature”,”Gene.Name”)]</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Feature"</span><span class="p">,</span><span class="s2">"Gene.Name"</span><span class="p">)]</span><span class="w">
           </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">1</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Kal1</span><span class="w">
 </span><span class="m">2</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w">
 </span><span class="m">3</span><span class="w">  </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
 </span><span class="m">4</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w">
 </span><span class="m">5</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w">
 </span><span class="m">6</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w">
 </span><span class="m">7</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG3770</span><span class="w">
 </span><span class="m">8</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG6018</span><span class="w">
 </span><span class="m">9</span><span class="w">         </span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w">     </span><span class="n">CR43883</span><span class="w">
 </span><span class="m">10</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Ama</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[,c("Gene.Name")]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Gene.Name"</span><span class="p">)]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">Kal1</span><span class="w">         </span><span class="n">sesB</span><span class="w">         </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">  </span><span class="n">Ant2</span><span class="w">         </span><span class="n">Hml</span><span class="w">          </span><span class="n">CG1544</span><span class="w">       </span><span class="n">CG3770</span><span class="w">
 </span><span class="p">[</span><span class="m">8</span><span class="p">]</span><span class="w"> </span><span class="n">CG6018</span><span class="w">       </span><span class="n">CR43883</span><span class="w">      </span><span class="n">Ama</span><span class="w">          </span><span class="n">CG3168</span><span class="w">       </span><span class="n">CG15695</span><span class="w">      </span><span class="n">CG8500</span><span class="w">       </span><span class="n">Rgk1</span><span class="w">
 </span><span class="p">[</span><span class="m">15</span><span class="p">]</span><span class="w"> </span><span class="n">Sesn</span><span class="w">         </span><span class="n">CG9119</span><span class="w">       </span><span class="n">Sema</span><span class="m">-2</span><span class="n">a</span><span class="w">      </span><span class="n">l</span><span class="p">(</span><span class="m">1</span><span class="p">)</span><span class="n">G0196</span><span class="w">    </span><span class="n">lectin</span><span class="m">-28</span><span class="n">C</span><span class="w">   </span><span class="n">Sox100B</span><span class="w">      </span><span class="n">Hsp27</span><span class="w">
 </span><span class="p">[</span><span class="m">22</span><span class="p">]</span><span class="w"> </span><span class="n">CG34330</span><span class="w">      </span><span class="n">Ugt58Fa</span><span class="w">      </span><span class="n">Src64B</span><span class="w">       </span><span class="n">Tequila</span><span class="w">      </span><span class="n">CG32407</span><span class="w">      </span><span class="n">LpR2</span><span class="w">         </span><span class="n">CG12290</span><span class="w">
 </span><span class="p">[</span><span class="m">29</span><span class="p">]</span><span class="w"> </span><span class="n">ps</span><span class="w">           </span><span class="n">CG8157</span><span class="w">       </span><span class="n">nemy</span><span class="w">         </span><span class="n">bves</span><span class="w">         </span><span class="n">CG8468</span><span class="w">       </span><span class="n">CR43855</span><span class="w">      </span><span class="n">CG13315</span><span class="w">
 </span><span class="p">[</span><span class="m">36</span><span class="p">]</span><span class="w"> </span><span class="n">CG1124</span><span class="w">       </span><span class="n">Reg</span><span class="m">-2</span><span class="w">        </span><span class="n">CG5261</span><span class="w">       </span><span class="n">Treh</span><span class="w">         </span><span class="n">Hsp23</span><span class="w">        </span><span class="n">Gale</span><span class="w">         </span><span class="n">PPO1</span><span class="w">
 </span><span class="p">[</span><span class="m">43</span><span class="p">]</span><span class="w"> </span><span class="n">Picot</span><span class="w">        </span><span class="n">RpS24</span><span class="w">        </span><span class="n">CG14946</span><span class="w">      </span><span class="n">CG10365</span><span class="w">      </span><span class="n">NimC4</span><span class="w">        </span><span class="n">CG42694</span><span class="w">      </span><span class="n">CG33307</span><span class="w">
 </span><span class="p">[</span><span class="m">50</span><span class="p">]</span><span class="w"> </span><span class="n">CG6006</span><span class="w">       </span><span class="n">grk</span><span class="w">          </span><span class="n">CG42369</span><span class="w">      </span><span class="n">GILT1</span><span class="w">        </span><span class="n">CG5001</span><span class="w">       </span><span class="n">Hsp26</span><span class="w">        </span><span class="n">MtnA</span><span class="w">
 </span><span class="p">[</span><span class="m">57</span><span class="p">]</span><span class="w"> </span><span class="n">CG6330</span><span class="w">       </span><span class="n">l</span><span class="p">(</span><span class="m">3</span><span class="p">)</span><span class="n">neo38</span><span class="w">    </span><span class="n">Cyt</span><span class="o">-</span><span class="n">b5</span><span class="o">-</span><span class="n">r</span><span class="w">     </span><span class="n">CG12116</span><span class="w">      </span><span class="n">PPO2</span><span class="w">         </span><span class="n">zye</span><span class="w">          </span><span class="n">tnc</span><span class="w">
 </span><span class="p">[</span><span class="m">64</span><span class="p">]</span><span class="w"> </span><span class="n">CG14257</span><span class="w">      </span><span class="n">GstE2</span><span class="w">        </span><span class="n">olf413</span><span class="w">       </span><span class="n">CG31642</span><span class="w">      </span><span class="n">Nxf3</span><span class="w">         </span><span class="n">CG8292</span><span class="w">       </span><span class="n">CG5397</span><span class="w">
 </span><span class="p">[</span><span class="m">71</span><span class="p">]</span><span class="w"> </span><span class="n">Mctp</span><span class="w">         </span><span class="n">CG7777</span><span class="w">       </span><span class="n">CG4019</span><span class="w">       </span><span class="n">ChLD3</span><span class="w">        </span><span class="n">Argk</span><span class="w">         </span><span class="n">bou</span><span class="w">          </span><span class="n">Mct1</span><span class="w">
 </span><span class="p">[</span><span class="m">78</span><span class="p">]</span><span class="w"> </span><span class="n">CG9416</span><span class="w">       </span><span class="n">CG12512</span><span class="w">      </span><span class="n">CG32625</span><span class="w">      </span><span class="n">Npc2b</span><span class="w">        </span><span class="n">CG30463</span><span class="w">      </span><span class="n">Eip74EF</span><span class="w">      </span><span class="n">CAHbeta</span><span class="w">
 </span><span class="p">[</span><span class="m">85</span><span class="p">]</span><span class="w"> </span><span class="n">CG14326</span><span class="w">      </span><span class="n">Cyp6w1</span><span class="w">       </span><span class="n">CG3457</span><span class="w">       </span><span class="n">wdp</span><span class="w">          </span><span class="n">cbt</span><span class="w">          </span><span class="n">CrebA</span><span class="w">        </span><span class="n">Cyp4d20</span><span class="w">
 </span><span class="p">[</span><span class="m">92</span><span class="p">]</span><span class="w"> </span><span class="n">CR45973</span><span class="w">      </span><span class="n">Sr</span><span class="o">-</span><span class="n">CIV</span><span class="w">       </span><span class="n">SPH93</span><span class="w">        </span><span class="n">Cyp9h1</span><span class="w">       </span><span class="n">sug</span><span class="w">          </span><span class="n">CG14516</span><span class="w">      </span><span class="n">Ggamma30A</span><span class="w">
 </span><span class="p">[</span><span class="m">99</span><span class="p">]</span><span class="w"> </span><span class="n">CG4297</span><span class="w">       </span><span class="n">CG9503</span><span class="w">       </span><span class="n">CG10063</span><span class="w">      </span><span class="n">CG8501</span><span class="w">       </span><span class="n">CG43336</span><span class="w">      </span><span class="n">stet</span><span class="w">         </span><span class="n">CG8852</span><span class="w">
 </span><span class="p">[</span><span class="m">106</span><span class="p">]</span><span class="w"> </span><span class="n">CG15080</span><span class="w">      </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w">        </span><span class="n">sn</span><span class="w">           </span><span class="n">CG9989</span><span class="w">       </span><span class="n">mlt</span><span class="w">          </span><span class="n">CG2444</span><span class="w">       </span><span class="n">CG17189</span><span class="w">
 </span><span class="p">[</span><span class="m">113</span><span class="p">]</span><span class="w"> </span><span class="n">RhoGAP1A</span><span class="w">     </span><span class="n">CG13743</span><span class="w">      </span><span class="n">TwdlU</span><span class="w">        </span><span class="n">Galphaf</span><span class="w">      </span><span class="n">CG32335</span><span class="w">      </span><span class="n">E</span><span class="p">(</span><span class="n">spl</span><span class="p">)</span><span class="n">m3</span><span class="o">-</span><span class="n">HLH</span><span class="w"> </span><span class="n">CG16749</span><span class="w">
 </span><span class="p">[</span><span class="m">120</span><span class="p">]</span><span class="w"> </span><span class="n">CG34434</span><span class="w">      </span><span class="n">CG43799</span><span class="w">      </span><span class="n">Pde6</span><span class="w">         </span><span class="n">Sr</span><span class="o">-</span><span class="n">CIII</span><span class="w">      </span><span class="n">NimC2</span><span class="w">        </span><span class="n">CG5895</span><span class="w">       </span><span class="n">SP1173</span><span class="w">
 </span><span class="p">[</span><span class="m">127</span><span class="p">]</span><span class="w"> </span><span class="n">CG1311</span><span class="w">       </span><span class="n">CG14856</span><span class="w">      </span><span class="n">CG6356</span><span class="w">       </span><span class="n">CG10440</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">Ama</span><span class="w"> </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="n">Ant2</span><span class="w"> </span><span class="n">Argk</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w"> </span><span class="n">bou</span><span class="w"> </span><span class="n">bves</span><span class="w"> </span><span class="n">CAHbeta</span><span class="w"> </span><span class="n">cbt</span><span class="w"> </span><span class="n">CG10063</span><span class="w"> </span><span class="n">CG10365</span><span class="w"> </span><span class="n">CG10440</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">zye</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">head(annotatedDEgenes)</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">  </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">      </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.9743</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w">  </span><span class="m">-30.74090</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24141394</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.5771</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w">  </span><span class="m">-28.74863</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">    </span><span class="n">chrX</span><span class="w"> </span><span class="m">10780892</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.8406</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w">  </span><span class="m">-28.22544</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">26869237</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.3224</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">   </span><span class="m">27.56599</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">    </span><span class="n">chrX</span><span class="w"> </span><span class="m">10778953</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.0673</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w">  </span><span class="m">-22.74596</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">   </span><span class="n">chr3L</span><span class="w"> </span><span class="m">13846053</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.9018</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w">  </span><span class="m">-21.88979</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">31196915</span><span class="w">

 </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Kal1</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">tail(annotatedDEgenes)</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">tail</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
          </span><span class="n">GeneID</span><span class="w"> </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">    </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">      </span><span class="n">P.value</span><span class="w">        </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">
 </span><span class="m">125</span><span class="w"> </span><span class="n">FBgn0036560</span><span class="w">  </span><span class="m">27.71424</span><span class="w">  </span><span class="m">1.089000</span><span class="w"> </span><span class="m">0.2327527</span><span class="w">   </span><span class="m">4.678786</span><span class="w"> </span><span class="m">2.885785e-06</span><span class="w"> </span><span class="m">6.671531e-05</span><span class="w">   </span><span class="n">chr3L</span><span class="w"> </span><span class="m">16035484</span><span class="w">
 </span><span class="m">126</span><span class="w"> </span><span class="n">FBgn0035710</span><span class="w">  </span><span class="m">26.16177</span><span class="w">  </span><span class="m">1.048979</span><span class="w"> </span><span class="m">0.2329221</span><span class="w">   </span><span class="m">4.503559</span><span class="w"> </span><span class="m">6.682472e-06</span><span class="w"> </span><span class="m">1.436480e-04</span><span class="w">   </span><span class="n">chr3L</span><span class="w">  </span><span class="m">6689326</span><span class="w">
 </span><span class="m">127</span><span class="w"> </span><span class="n">FBgn0035523</span><span class="w">  </span><span class="m">70.99820</span><span class="w">  </span><span class="m">1.004819</span><span class="w"> </span><span class="m">0.2237625</span><span class="w">   </span><span class="m">4.490560</span><span class="w"> </span><span class="m">7.103600e-06</span><span class="w"> </span><span class="m">1.523189e-04</span><span class="w">   </span><span class="n">chr3L</span><span class="w">  </span><span class="m">4277961</span><span class="w">
 </span><span class="m">128</span><span class="w"> </span><span class="n">FBgn0038261</span><span class="w">  </span><span class="m">44.27058</span><span class="w">  </span><span class="m">1.006264</span><span class="w"> </span><span class="m">0.2241243</span><span class="w">   </span><span class="m">4.489756</span><span class="w"> </span><span class="m">7.130482e-06</span><span class="w"> </span><span class="m">1.525141e-04</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">14798985</span><span class="w">
 </span><span class="m">129</span><span class="w"> </span><span class="n">FBgn0039178</span><span class="w">  </span><span class="m">23.55006</span><span class="w">  </span><span class="m">1.040917</span><span class="w"> </span><span class="m">0.2326264</span><span class="w">   </span><span class="m">4.474631</span><span class="w"> </span><span class="m">7.654326e-06</span><span class="w"> </span><span class="m">1.629061e-04</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24283954</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">FBgn0034636</span><span class="w">  </span><span class="m">24.77052</span><span class="w"> </span><span class="m">-1.028531</span><span class="w"> </span><span class="m">0.2321678</span><span class="w">  </span><span class="m">-4.430119</span><span class="w"> </span><span class="m">9.418101e-06</span><span class="w"> </span><span class="m">1.951185e-04</span><span class="w">   </span><span class="n">chr2R</span><span class="w"> </span><span class="m">21560245</span><span class="w">

 </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">125</span><span class="w"> </span><span class="m">16037227</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">CG5895</span><span class="w">
 </span><span class="m">126</span><span class="w">  </span><span class="m">6703521</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">SP1173</span><span class="w">
 </span><span class="m">127</span><span class="w">  </span><span class="m">4281585</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">CG1311</span><span class="w">
 </span><span class="m">128</span><span class="w"> </span><span class="m">14801163</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">   </span><span class="n">CG14856</span><span class="w">
 </span><span class="m">129</span><span class="w"> </span><span class="m">24288617</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">CG6356</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="m">21576035</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">   </span><span class="n">CG10440</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes$GeneID</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">GeneID</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="n">FBgn0034736</span><span class="w">
 </span><span class="p">[</span><span class="m">9</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0264475</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0029896</span><span class="w"> </span><span class="n">FBgn0038832</span><span class="w"> </span><span class="n">FBgn0037754</span><span class="w"> </span><span class="n">FBgn0264753</span><span class="w"> </span><span class="n">FBgn0034897</span><span class="w"> </span><span class="n">FBgn0035189</span><span class="w">
 </span><span class="p">[</span><span class="m">17</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0011260</span><span class="w"> </span><span class="n">FBgn0027279</span><span class="w"> </span><span class="n">FBgn0040099</span><span class="w"> </span><span class="n">FBgn0024288</span><span class="w"> </span><span class="n">FBgn0001226</span><span class="w"> </span><span class="n">FBgn0085359</span><span class="w"> </span><span class="n">FBgn0040091</span><span class="w"> </span><span class="n">FBgn0262733</span><span class="w">
 </span><span class="p">[</span><span class="m">25</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0023479</span><span class="w"> </span><span class="n">FBgn0052407</span><span class="w"> </span><span class="n">FBgn0051092</span><span class="w"> </span><span class="n">FBgn0039419</span><span class="w"> </span><span class="n">FBgn0261552</span><span class="w"> </span><span class="n">FBgn0034010</span><span class="w"> </span><span class="n">FBgn0261673</span><span class="w"> </span><span class="n">FBgn0031150</span><span class="w">
 </span><span class="p">[</span><span class="m">33</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0033913</span><span class="w"> </span><span class="n">FBgn0264437</span><span class="w"> </span><span class="n">FBgn0040827</span><span class="w"> </span><span class="n">FBgn0037290</span><span class="w"> </span><span class="n">FBgn0016715</span><span class="w"> </span><span class="n">FBgn0031912</span><span class="w"> </span><span class="n">FBgn0003748</span><span class="w"> </span><span class="n">FBgn0001224</span><span class="w">
 </span><span class="p">[</span><span class="m">41</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0035147</span><span class="w"> </span><span class="n">FBgn0261362</span><span class="w"> </span><span class="n">FBgn0024315</span><span class="w"> </span><span class="n">FBgn0261596</span><span class="w"> </span><span class="n">FBgn0032405</span><span class="w"> </span><span class="n">FBgn0039109</span><span class="w"> </span><span class="n">FBgn0260011</span><span class="w"> </span><span class="n">FBgn0261584</span><span class="w">
 </span><span class="p">[</span><span class="m">49</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0053307</span><span class="w"> </span><span class="n">FBgn0063649</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">FBgn0259715</span><span class="w"> </span><span class="n">FBgn0038149</span><span class="w"> </span><span class="n">FBgn0031322</span><span class="w"> </span><span class="n">FBgn0001225</span><span class="w"> </span><span class="n">FBgn0002868</span><span class="w">
 </span><span class="p">[</span><span class="m">57</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039464</span><span class="w"> </span><span class="n">FBgn0265276</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0030041</span><span class="w"> </span><span class="n">FBgn0033367</span><span class="w"> </span><span class="n">FBgn0036985</span><span class="w"> </span><span class="n">FBgn0039257</span><span class="w"> </span><span class="n">FBgn0039479</span><span class="w">
 </span><span class="p">[</span><span class="m">65</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0063498</span><span class="w"> </span><span class="n">FBgn0037153</span><span class="w"> </span><span class="n">FBgn0051642</span><span class="w"> </span><span class="n">FBgn0263232</span><span class="w"> </span><span class="n">FBgn0032004</span><span class="w"> </span><span class="n">FBgn0031327</span><span class="w"> </span><span class="n">FBgn0034389</span><span class="w"> </span><span class="n">FBgn0033635</span><span class="w">
 </span><span class="p">[</span><span class="m">73</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0034885</span><span class="w"> </span><span class="n">FBgn0032598</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0261284</span><span class="w"> </span><span class="n">FBgn0023549</span><span class="w"> </span><span class="n">FBgn0034438</span><span class="w"> </span><span class="n">FBgn0031703</span><span class="w"> </span><span class="n">FBgn0052625</span><span class="w">
 </span><span class="p">[</span><span class="m">81</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0038198</span><span class="w"> </span><span class="n">FBgn0050463</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0037646</span><span class="w"> </span><span class="n">FBgn0038528</span><span class="w"> </span><span class="n">FBgn0033065</span><span class="w"> </span><span class="n">FBgn0024984</span><span class="w"> </span><span class="n">FBgn0034718</span><span class="w">
 </span><span class="p">[</span><span class="m">89</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0043364</span><span class="w"> </span><span class="n">FBgn0004396</span><span class="w"> </span><span class="n">FBgn0035344</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w"> </span><span class="n">FBgn0031547</span><span class="w"> </span><span class="n">FBgn0032638</span><span class="w"> </span><span class="n">FBgn0033775</span><span class="w"> </span><span class="n">FBgn0033782</span><span class="w">
 </span><span class="p">[</span><span class="m">97</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039640</span><span class="w"> </span><span class="n">FBgn0267252</span><span class="w"> </span><span class="n">FBgn0031258</span><span class="w"> </span><span class="n">FBgn0030598</span><span class="w"> </span><span class="n">FBgn0035727</span><span class="w"> </span><span class="n">FBgn0033724</span><span class="w"> </span><span class="n">FBgn0263041</span><span class="w"> </span><span class="n">FBgn0020248</span><span class="w">
 </span><span class="p">[</span><span class="m">105</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0031548</span><span class="w"> </span><span class="n">FBgn0034391</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0003447</span><span class="w"> </span><span class="n">FBgn0039593</span><span class="w"> </span><span class="n">FBgn0265512</span><span class="w"> </span><span class="n">FBgn0030326</span><span class="w"> </span><span class="n">FBgn0039485</span><span class="w">
 </span><span class="p">[</span><span class="m">113</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0025836</span><span class="w"> </span><span class="n">FBgn0033368</span><span class="w"> </span><span class="n">FBgn0037223</span><span class="w"> </span><span class="n">FBgn0010223</span><span class="w"> </span><span class="n">FBgn0063667</span><span class="w"> </span><span class="n">FBgn0002609</span><span class="w"> </span><span class="n">FBgn0037678</span><span class="w"> </span><span class="n">FBgn0250904</span><span class="w">
 </span><span class="p">[</span><span class="m">121</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0264343</span><span class="w"> </span><span class="n">FBgn0038237</span><span class="w"> </span><span class="n">FBgn0020376</span><span class="w"> </span><span class="n">FBgn0028939</span><span class="w"> </span><span class="n">FBgn0036560</span><span class="w"> </span><span class="n">FBgn0035710</span><span class="w"> </span><span class="n">FBgn0035523</span><span class="w"> </span><span class="n">FBgn0038261</span><span class="w">
 </span><span class="p">[</span><span class="m">129</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039178</span><span class="w"> </span><span class="n">FBgn0034636</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="highlighter-rouge">annotatedDEgenes[annotatedDEgenes$Feature == "pseudogene",]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Feature</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"pseudogene"</span><span class="p">,]</span><span class="w">
</span><span class="n">GeneID</span><span class="w"> </span><span class="n">Base.mean</span><span class="w">   </span><span class="n">log2.FC.</span><span class="w">    </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">      </span><span class="n">P.value</span><span class="w">        </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">   </span><span class="n">Start</span><span class="w">
</span><span class="m">115</span><span class="w"> </span><span class="n">FBgn0037223</span><span class="w">  </span><span class="m">37.00783</span><span class="w"> </span><span class="m">-1.161706</span><span class="w"> </span><span class="m">0.2297393</span><span class="w">  </span><span class="m">-5.056627</span><span class="w"> </span><span class="m">4.267378e-07</span><span class="w"> </span><span class="m">1.192225e-05</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">4243605</span><span class="w">
</span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">    </span><span class="n">Feature</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w">
</span><span class="m">115</span><span class="w"> </span><span class="m">4245511</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">pseudogene</span><span class="w">     </span><span class="n">TwdlU</span><span class="w">
</span></code></pre></div>        </div>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>The subsetting notation is very similar to what we learned for vectors. The key differences include:</p>
<ul>
  <li>2 values, separated by commas: <code class="highlighter-rouge">data.frame[row, column]</code></li>
  <li>Use of a colon between the numbers (<code class="highlighter-rouge">start:stop, inclusive</code>) for a continuous range of numbers</li>
  <li>Passing a vector using <code class="highlighter-rouge">c()</code> for a non continuous set of numbers, pass a vector</li>
  <li>Index using the name of a column(s) by passing them as vectors using <code class="highlighter-rouge">c()</code></li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-a-new-data-frame-by-subsetting-an-existing-one"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create a new data frame by subsetting an existing one</h3>
  <ol>
    <li>
      <p>Create a new data frame containing only observations from the “+” strand</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># create a new data frame containing only observations from the "+" strand</span><span class="w">
</span><span class="n">annotatedDEgenes_plusStrand</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">,]</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Check the dimension of the new data frame</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># check the dimension of the data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="nf">dim</span><span class="p">(</span><span class="n">annotatedDEgenes_plusStrand</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">72</span><span class="w"> </span><span class="m">13</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Get a summary of the data frame</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># get a summary of the data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">summary</span><span class="p">(</span><span class="n">annotatedDEgenes_plusStrand</span><span class="p">)</span><span class="w">
</span><span class="n">GeneID</span><span class="w">           </span><span class="n">Base.mean</span><span class="w">           </span><span class="n">log2.FC.</span><span class="w">         </span><span class="n">StdErr</span><span class="w">            </span><span class="n">Wald.Stats</span><span class="w">
</span><span class="n">FBgn0000071</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">   </span><span class="m">23.55</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-4.1485</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.08603</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-30.741</span><span class="w">
</span><span class="n">FBgn0001224</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">   </span><span class="m">96.04</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-1.3664</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.12523</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-11.223</span><span class="w">
</span><span class="n">FBgn0001226</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w">  </span><span class="m">275.91</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">-1.0739</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.15648</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">-5.290</span><span class="w">
</span><span class="n">FBgn0002609</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">1365.34</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">-0.3133</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">0.16087</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">-2.729</span><span class="w">
</span><span class="n">FBgn0003447</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">988.33</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">1.1856</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.19743</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">7.592</span><span class="w">
</span><span class="n">FBgn0003748</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">28915.90</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">2.3600</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">0.23275</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">17.399</span><span class="w">
</span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="w">    </span><span class="o">:</span><span class="m">66</span><span class="w">

</span><span class="n">P.value</span><span class="w">             </span><span class="n">P.adj</span><span class="w">              </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">              </span><span class="n">End</span><span class="w">               </span><span class="n">Strand</span><span class="w">
</span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2L</span><span class="o">:</span><span class="m">12</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">127448</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">140340</span><span class="w">   </span><span class="o">-:</span><span class="w"> </span><span class="m">0</span><span class="w">
</span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2R</span><span class="o">:</span><span class="m">14</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7377082</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7379907</span><span class="w">   </span><span class="o">+:</span><span class="m">72</span><span class="w">
</span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr3L</span><span class="o">:</span><span class="m">15</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">14068548</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">14070644</span><span class="w">
</span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">4.556e-07</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">1.013e-05</span><span class="w">   </span><span class="n">chr3R</span><span class="o">:</span><span class="m">24</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">14422245</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">14431849</span><span class="w">
</span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">6.000e-12</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">3.200e-10</span><span class="w">   </span><span class="n">chrX</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">7</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">21339535</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">21378634</span><span class="w">
</span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">7.654e-06</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">1.629e-04</span><span class="w">              </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31196915</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31203722</span><span class="w">

</span><span class="n">Feature</span><span class="w">             </span><span class="n">Gene.Name</span><span class="w">
</span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w">       </span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w">   </span><span class="n">Ama</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">protein_coding</span><span class="o">:</span><span class="m">69</span><span class="w">   </span><span class="n">bou</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">pseudogene</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">bves</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="n">CAHbeta</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="n">CG10365</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="n">CG1124</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="o">:</span><span class="m">66</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<h2 id="change-type-of-columns">Change type of columns</h2>

<p>Sometimes, it is possible that R will misinterpret the type of data represented in a data frame, or store that data in a mode which prevents you from operating on the data the way you wish. For example, a long list of gene names isn’t usually thought of as a categorical variable, the way that your experimental condition (e.g. control, treatment) might be. More importantly, some R packages you use to analyze your data may expect characters as input, not factors. At other times (such as plotting or some statistical analyses) a factor may be more appropriate. Ultimately, you should know how to change the mode of an object.</p>

<p>First, its very important to recognize that coercion happens in R all the time. This can be a good thing when R gets it right, or a bad thing when the result is not what you expect.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-format-gene-names"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Format gene names</h3>
  <ol>
    <li>
      <p>Check the structure of the <code class="highlighter-rouge">Gene.name</code> column</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">str</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">130</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"Ama"</span><span class="p">,</span><span class="s2">"Amy-p"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">88</span><span class="w"> </span><span class="m">113</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">84</span><span class="w"> </span><span class="m">26</span><span class="w"> </span><span class="m">41</span><span class="w"> </span><span class="m">53</span><span class="w"> </span><span class="m">69</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Try to coerce the column into numeric</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w">  </span><span class="m">88</span><span class="w"> </span><span class="m">113</span><span class="w">   </span><span class="m">5</span><span class="w">   </span><span class="m">3</span><span class="w">  </span><span class="m">84</span><span class="w">  </span><span class="m">26</span><span class="w">  </span><span class="m">41</span><span class="w">  </span><span class="m">53</span><span class="w">  </span><span class="m">69</span><span class="w">   </span><span class="m">1</span><span class="w">  </span><span class="m">33</span><span class="w">  </span><span class="m">27</span><span class="w">  </span><span class="m">60</span><span class="w"> </span><span class="m">109</span><span class="w"> </span><span class="m">114</span><span class="w">  </span><span class="m">63</span><span class="w"> </span><span class="m">112</span><span class="w">
</span><span class="p">[</span><span class="m">18</span><span class="p">]</span><span class="w">  </span><span class="m">89</span><span class="w">  </span><span class="m">91</span><span class="w"> </span><span class="m">116</span><span class="w">  </span><span class="m">87</span><span class="w">  </span><span class="m">38</span><span class="w"> </span><span class="m">128</span><span class="w"> </span><span class="m">121</span><span class="w"> </span><span class="m">124</span><span class="w">  </span><span class="m">35</span><span class="w">  </span><span class="m">92</span><span class="w">  </span><span class="m">15</span><span class="w"> </span><span class="m">107</span><span class="w">  </span><span class="m">57</span><span class="w">  </span><span class="m">97</span><span class="w">   </span><span class="m">7</span><span class="w">  </span><span class="m">59</span><span class="w">  </span><span class="m">68</span><span class="w">
</span><span class="p">[</span><span class="m">35</span><span class="p">]</span><span class="w">  </span><span class="m">18</span><span class="w">  </span><span class="m">13</span><span class="w"> </span><span class="m">108</span><span class="w">  </span><span class="m">49</span><span class="w"> </span><span class="m">126</span><span class="w">  </span><span class="m">85</span><span class="w">  </span><span class="m">78</span><span class="w"> </span><span class="m">105</span><span class="w"> </span><span class="m">104</span><span class="w"> </span><span class="m">111</span><span class="w">  </span><span class="m">24</span><span class="w">  </span><span class="m">11</span><span class="w">  </span><span class="m">99</span><span class="w">  </span><span class="m">44</span><span class="w">  </span><span class="m">37</span><span class="w">  </span><span class="m">52</span><span class="w">  </span><span class="m">82</span><span class="w">
</span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>

      <p>Strangely, it works! Almost. Instead of giving an error message, R returns numeric values, which in this case are the integers assigned to the levels in this factor. This kind of behavior can lead to hard-to-find bugs, for example when we do have numbers in a factor, and we get numbers from a coercion. If we don’t look carefully, we may not notice a problem.</p>
    </li>
    <li>
      <p>Coerce the column into character (which makes sense for gene names)</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>R will sometimes coerce without you asking for it (implicit coercion). You can explicitly change it (explicit coercion).</p>

<blockquote class="tip">
  <h3 id="tip-avoiding-needless-coercion"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Avoiding needless coercion</h3>
  <p>One way to avoid needless coercion when importing a data frame using functions such as <code class="highlighter-rouge">read.csv()</code> is to set the argument <code class="highlighter-rouge">StringsAsFactors</code> to <code class="highlighter-rouge">FALSE</code>. By default, this argument is <code class="highlighter-rouge">TRUE</code>. Setting it to <code class="highlighter-rouge">FALSE</code> will treat any non-numeric column to a character type. Going through the <code class="highlighter-rouge">read.csv() </code>documentation, you will also see you can explicitly type your columns using the <code class="highlighter-rouge">colClasses</code> argument. Other R packages (such as the Tidyverse <code class="highlighter-rouge">readr</code>) don’t have this particular conversion issue, but many packages will still try to guess a data type.</p>
</blockquote>

<h2 id="inspect-a-numerical-column-using-math-sorting-renaming">Inspect a numerical column using math, sorting, renaming</h2>

<p>Here are a few operations that don’t need much explanation, but which are good to know. There are lots of arithmetic functions you may want to apply to your data frame. For example, you can use functions like <code class="highlighter-rouge">mean()</code>, <code class="highlighter-rouge">min()</code>, <code class="highlighter-rouge">max()</code> on an individual column.</p>

<p>Let’s look at the <code class="highlighter-rouge">Base.mean</code> column, i.e. the mean of normalized counts of all samples, normalizing for sequencing depth and library composition.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-the-basemean-column-and-sort"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect the <code class="highlighter-rouge">Base.mean</code> column and sort</h3>
  <ol>
    <li>
      <p>Get the maximum value of <code class="highlighter-rouge">Base.mean</code> column</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">max</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">65114.84</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Sort the data frame by values in the <code class="highlighter-rouge">Base.mean</code> column</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">sorted_by_BaseMean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">sorted_by_BaseMean</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">19.15076</span><span class="w"> </span><span class="m">23.55006</span><span class="w"> </span><span class="m">24.77052</span><span class="w"> </span><span class="m">26.16177</span><span class="w"> </span><span class="m">27.71424</span><span class="w"> </span><span class="m">31.58769</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="question">
        <h3 id="question-changing-the-order"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Changing the order</h3>
        <p>The <code class="highlighter-rouge">order()</code> function lists values in increasing order by default. Look at the documentation for this function and change <code class="highlighter-rouge">sorted_by_BaseMean</code> to start with differentially expressed means with the greatest base mean.</p>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">sorted_by_BaseMean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">sorted_by_BaseMean</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">65114.841</span><span class="w"> </span><span class="m">28915.903</span><span class="w"> </span><span class="m">24584.306</span><span class="w"> </span><span class="m">22435.886</span><span class="w">  </span><span class="m">8903.630</span><span class="w">  </span><span class="m">8573.815</span><span class="w">
</span></code></pre></div>          </div>

        </blockquote>
      </blockquote>
    </li>
    <li>
      <p>Coerce the column into character (which makes sense for gene names)</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<h2 id="save-the-data-frame-to-a-file">Save the data frame to a file</h2>

<p>We can save data frame as table in our <span class="notranslate">Galaxy</span> history to use it as input for other tools.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-save-data-frame-in-galaxy-history"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Save data frame in <span class="notranslate">Galaxy</span> history</h3>

  <ol>
    <li>
      <p>Save the <code class="highlighter-rouge">annotatedDEgenes_plusStrand</code> object to a <code class="highlighter-rouge">.csv</code> file using the <code class="highlighter-rouge">write.csv()</code> function:</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">write.csv</span><span class="p">(</span><span class="n">annotatedDEgenes_plusStrand</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"annotatedDEgenes_plusStrand.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p><strong><em>TODO</em></strong>: <em>Description on how to save a file from RStudio to <span class="notranslate">Galaxy</span></em></p>
    </li>
  </ol>
</blockquote>

<p>The <code class="highlighter-rouge">write.csv()</code> function has some additional arguments listed in the help, but at a minimum you need to tell it what data frame to write to file, and give a path to a file name in quotes (if you only provide a file name, the file will be written in the current working directory).</p>

<h1 id="aggregate-and-analyze-with-dplyr">Aggregate and Analyze with dplyr</h1>

<p>Selecting columns and rows using bracket subsetting is handy, but it can be cumbersome and difficult to read, especially for complicated operations.</p>

<p>The functions we’ve been using so far, like <code class="highlighter-rouge">str()</code>, come built into R. By using packages, we can access more functions. Here, we would like to use <a href="https://cran.r-project.org/package=dplyr"><code class="highlighter-rouge">dplyr</code></a> package provides a number of very useful functions for manipulating data frames in a way that will reduce repetition, reduce the probability of making errors, and probably even save you some typing. As an added bonus, you might even find the <code class="highlighter-rouge">dplyr</code> grammar easier to read.</p>

<blockquote class="comment">
  <h3 id="comment-what-is-dplyr"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> What is dplyr?</h3>

  <p>The package <code class="highlighter-rouge">dplyr</code> is a fairly new (2014) package that tries to provide easy tools for the most common data manipulation tasks. It is built to work directly with data frames. The thinking behind it was largely inspired by the package <code class="highlighter-rouge">plyr</code> which has been in use for some time but suffered from being slow in some cases.<code class="highlighter-rouge"> dplyr</code> addresses this by porting much of the computation to C++. An additional feature is the ability to work with data stored directly in an exte<span class="notranslate">rna</span>l database. The benefits of doing this are that the data can be managed natively in a relational database, queries can be conducted on that database, and only the results of the query returned.</p>

  <p>This addresses a common problem with R in that all operations are conducted in memory and thus the amount of data you can work with is limited by available memory. The database connections essentially remove that limitation in that you can have a database of many 100s GB, conduct queries on it directly and pull back just what you need for analysis in R.</p>
</blockquote>

<p>To use <code class="highlighter-rouge">dplyr</code>, we need first to install the package and then load it to be able to use it.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-install-and-load-dyplr"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Install and load dyplr</h3>

  <ol>
    <li>
      <p>Load dyplr</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## load dyplr package</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="s2">"dplyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="tip">
        <h3 id="tip-tip-didnt-work"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Didn’t work?</h3>
        <p>You might need to install dyplr first</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"dplyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>        </div>

        <p>You might get asked to choose a CRAN mirror – this is asking you to choose a site to download the package from. The choice doesn’t matter too much.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We only need to install a package once per computer, but we need to load it every time we open a new R session and want to use that package.</p>

<h2 id="select-columns-and-filter-rows">Select columns and filter rows</h2>

<p>To select columns, instead with bracket, we could use the <code class="highlighter-rouge">select</code> function of <code class="highlighter-rouge">dplyr</code></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-column"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select column</h3>

  <ol>
    <li>
      <p>Select <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">Start</code>, <code class="highlighter-rouge">End</code>, <code class="highlighter-rouge">Strand</code> columns in <code class="highlighter-rouge">annotatedDEgenes</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="p">)</span><span class="w">
         </span><span class="n">GeneID</span><span class="w">    </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">
</span><span class="m">1</span><span class="w">   </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">2</span><span class="w">   </span><span class="n">FBgn0003360</span><span class="w"> </span><span class="m">10780892</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w">
</span><span class="m">3</span><span class="w">   </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">26869237</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w">
</span><span class="m">4</span><span class="w">   </span><span class="n">FBgn0025111</span><span class="w"> </span><span class="m">10778953</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w">
</span><span class="m">5</span><span class="w">   </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">6</span><span class="w">   </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">7</span><span class="w">   </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">8</span><span class="w">   </span><span class="n">FBgn0034736</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">9</span><span class="w">   </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">10</span><span class="w">  </span><span class="n">FBgn0000071</span><span class="w">  </span><span class="m">6762592</span><span class="w">  </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>

      <p>The first argument to this function is the data frame (<code class="highlighter-rouge">annotatedDEgenes</code>), and the subsequent arguments are the columns to keep.</p>
    </li>
    <li>
      <p>Select all columns <em>except</em> certain ones</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
         </span><span class="n">GeneID</span><span class="w">   </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w">    </span><span class="n">Start</span><span class="w"> </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">    </span><span class="n">Feature</span><span class="w">    </span><span class="n">Gene.Name</span><span class="w">
</span><span class="m">1</span><span class="w">   </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.97430</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w"> </span><span class="m">-30.740902</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w"> </span><span class="m">24141394</span><span class="w">  </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Kal1</span><span class="w">
</span><span class="m">2</span><span class="w">   </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.57713</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w"> </span><span class="m">-28.748628</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w"> </span><span class="m">10780892</span><span class="w">  </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">sesB</span><span class="w">
</span><span class="m">3</span><span class="w">   </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.84056</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w"> </span><span class="m">-28.225437</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w"> </span><span class="m">26869237</span><span class="w">  </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">  </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
</span><span class="m">4</span><span class="w">   </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.32237</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">  </span><span class="m">27.565988</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w"> </span><span class="m">10778953</span><span class="w">  </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Ant2</span><span class="w">
</span><span class="m">5</span><span class="w">   </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.06728</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w"> </span><span class="m">-22.745964</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w"> </span><span class="m">13846053</span><span class="w">  </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Hml</span><span class="w">
</span><span class="m">6</span><span class="w">   </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.90178</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w"> </span><span class="m">-21.889786</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w"> </span><span class="m">31196915</span><span class="w">  </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG1544</span><span class="w">
</span><span class="m">7</span><span class="w">   </span><span class="n">FBgn0035085</span><span class="w">   </span><span class="m">928.26381</span><span class="w"> </span><span class="m">-2.414074</span><span class="w"> </span><span class="m">0.11518516</span><span class="w"> </span><span class="m">-20.958204</span><span class="w">  </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w"> </span><span class="m">24945138</span><span class="w">   </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG3770</span><span class="w">
</span><span class="m">8</span><span class="w">   </span><span class="n">FBgn0034736</span><span class="w">   </span><span class="m">330.38302</span><span class="w"> </span><span class="m">-3.018179</span><span class="w"> </span><span class="m">0.15815418</span><span class="w"> </span><span class="m">-19.083774</span><span class="w">  </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w"> </span><span class="m">22550093</span><span class="w">   </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG6018</span><span class="w">
</span><span class="m">9</span><span class="w">   </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">955.45445</span><span class="w"> </span><span class="m">-2.334486</span><span class="w"> </span><span class="m">0.12423003</span><span class="w"> </span><span class="m">-18.791643</span><span class="w">  </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">   </span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w">      </span><span class="n">CR43883</span><span class="w">
</span><span class="m">10</span><span class="w">  </span><span class="n">FBgn0000071</span><span class="w">   </span><span class="m">468.05793</span><span class="w">  </span><span class="m">2.360017</span><span class="w"> </span><span class="m">0.13564397</span><span class="w">  </span><span class="m">17.398615</span><span class="w">  </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">  </span><span class="m">6762592</span><span class="w">   </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Ama</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><code class="highlighter-rouge">dplyr</code> also provides useful functions to select columns based on their names. For instance, <code class="highlighter-rouge">starts_with()</code> allows you to select columns that starts with specific letters.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-columns-that-start-with-the-letter-p"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select columns that start with the letter “P”</h3>

  <ol>
    <li>
      <p>Select columns that start with the letter “P”</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">starts_with</span><span class="p">(</span><span class="s2">"P."</span><span class="p">))</span><span class="w">

          </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w">
</span><span class="m">1</span><span class="w">   </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">
</span><span class="m">2</span><span class="w">   </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">
</span><span class="m">3</span><span class="w">   </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">
</span><span class="m">4</span><span class="w">   </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">
</span><span class="m">5</span><span class="w">   </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">
</span><span class="m">6</span><span class="w">   </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">
</span><span class="m">7</span><span class="w">    </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w">
</span><span class="m">8</span><span class="w">    </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w">
</span><span class="m">9</span><span class="w">    </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">
</span><span class="m">10</span><span class="w">   </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-selecting-on-multiple-conditions"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Selecting on multiple conditions</h3>

  <p>Create a table that contains all the columns with the letter “s” in their name except for the column “Wald.Stats”, and that contains the column “End”.</p>

  <p>Hint: look at the help for the function <code class="highlighter-rouge">starts_with()</code> we’ve just covered.</p>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">contains</span><span class="p">(</span><span class="s2">"s"</span><span class="p">),</span><span class="w"> </span><span class="o">-</span><span class="n">Wald.Stats</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>Similarly to choose rows, we can use the <code class="highlighter-rouge">filter()</code> function.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-rows"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter rows</h3>

  <ol>
    <li>
      <p>Filter for rows on the forward strand</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">)</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">   </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">    </span><span class="n">Gene.Name</span><span class="w">
</span><span class="m">1</span><span class="w">  </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.97430</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w"> </span><span class="m">-30.740902</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Kal1</span><span class="w">
</span><span class="m">2</span><span class="w">  </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.06728</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w"> </span><span class="m">-22.745964</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">      </span><span class="n">chr3L</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Hml</span><span class="w">
</span><span class="m">3</span><span class="w">  </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.90178</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w"> </span><span class="m">-21.889786</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG1544</span><span class="w">
</span><span class="m">4</span><span class="w">  </span><span class="n">FBgn0035085</span><span class="w">   </span><span class="m">928.26381</span><span class="w"> </span><span class="m">-2.414074</span><span class="w"> </span><span class="m">0.11518516</span><span class="w"> </span><span class="m">-20.958204</span><span class="w">  </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w">      </span><span class="n">chr2R</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG3770</span><span class="w">
</span><span class="m">5</span><span class="w">  </span><span class="n">FBgn0034736</span><span class="w">   </span><span class="m">330.38302</span><span class="w"> </span><span class="m">-3.018179</span><span class="w"> </span><span class="m">0.15815418</span><span class="w"> </span><span class="m">-19.083774</span><span class="w">  </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w">      </span><span class="n">chr2R</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG6018</span><span class="w">
</span><span class="m">6</span><span class="w">  </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">955.45445</span><span class="w"> </span><span class="m">-2.334486</span><span class="w"> </span><span class="m">0.12423003</span><span class="w"> </span><span class="m">-18.791643</span><span class="w">  </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">      </span><span class="n">chr3L</span><span class="w"> </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">        </span><span class="n">linc<span class="notranslate">RNA</span></span><span class="w">      </span><span class="n">CR43883</span><span class="w">
</span><span class="m">7</span><span class="w">  </span><span class="n">FBgn0000071</span><span class="w">   </span><span class="m">468.05793</span><span class="w">  </span><span class="m">2.360017</span><span class="w"> </span><span class="m">0.13564397</span><span class="w">  </span><span class="m">17.398615</span><span class="w">  </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">6762592</span><span class="w">  </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Ama</span><span class="w">
</span><span class="m">8</span><span class="w">  </span><span class="n">FBgn0038832</span><span class="w">   </span><span class="m">429.85033</span><span class="w"> </span><span class="m">-2.340711</span><span class="w"> </span><span class="m">0.15483651</span><span class="w"> </span><span class="m">-15.117308</span><span class="w">  </span><span class="m">1.245307e-51</span><span class="w">  </span><span class="m">8.900828e-49</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">20842139</span><span class="w"> </span><span class="m">20844981</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG15695</span><span class="w">
</span><span class="m">9</span><span class="w">  </span><span class="n">FBgn0037754</span><span class="w">   </span><span class="m">394.46410</span><span class="w"> </span><span class="m">-2.071994</span><span class="w"> </span><span class="m">0.14494069</span><span class="w"> </span><span class="m">-14.295459</span><span class="w">  </span><span class="m">2.335613e-46</span><span class="w">  </span><span class="m">1.540966e-43</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">9784652</span><span class="w">  </span><span class="m">9789323</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG8500</span><span class="w">
</span><span class="m">10</span><span class="w"> </span><span class="n">FBgn0034897</span><span class="w">  </span><span class="m">1340.88978</span><span class="w"> </span><span class="m">-1.795107</span><span class="w"> </span><span class="m">0.12629203</span><span class="w"> </span><span class="m">-14.213934</span><span class="w">  </span><span class="m">7.508655e-46</span><span class="w">  </span><span class="m">4.293449e-43</span><span class="w">      </span><span class="n">chr2R</span><span class="w"> </span><span class="m">23713899</span><span class="w"> </span><span class="m">23734846</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Sesn</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="comment">
        <h3 id="comment-equivalent-in-base-r-code"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Equivalent in base R code</h3>
        <p>Note that this is equivalent to the base R code below, but is easier to read!</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">,]</span><span class="w">
</span></code></pre></div>        </div>
      </blockquote>

      <p><code class="highlighter-rouge">filter()</code> keeps all the rows that match the conditions that are provided</p>
    </li>
    <li>
      <p>Filter for rows corresponding to genes in Chromosome X or 2R</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## rows for genes in Chromosome X or 2R</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"chrX"</span><span class="p">,</span><span class="w"> </span><span class="s2">"chr2R"</span><span class="p">))</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Filter for rows where the log2 fold change is greater than 2</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## rows where the log2 fold change is greater than 2</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><code class="highlighter-rouge">filter()</code> allows to combine multiple conditions. They need to be separated using a <code class="highlighter-rouge">,</code> as arguments to the function, they will be combined using the <code class="highlighter-rouge">&amp;</code> (AND) logical operator. If you need to use the <code class="highlighter-rouge">|</code> (OR) logical operator, you can specify it explicitly:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-rows-for-multiple-conditions"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter rows for multiple conditions</h3>

  <ol>
    <li>
      <p>Filter for genes on chrX and with adjusted p-value &lt;= 1e-100</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## both are equivalent</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">P.adj</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">1e-100</span><span class="p">)</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="p">,</span><span class="w"> </span><span class="n">P.adj</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">1e-100</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Filter for genes on chrX and with log2 Fold Change smaller than -2 or larger than 2</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="p">,</span><span class="w"> </span><span class="p">(</span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-2</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span><span class="p">))</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-practising-with-conditionals"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Practising with conditionals</h3>

  <p>Select all the rows for genes that:</p>
  <ul>
    <li>start after position <script type="math/tex">1e^6</script> (one million) and before position <script type="math/tex">2e^6</script> (included) in their chromosome</li>
    <li>have a log2 fold change greater than 1 or an adjusted p-value less than <script type="math/tex">1e^{-75}</script>.</li>
  </ul>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1e6</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">End</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">2e6</span><span class="p">,</span><span class="w"> </span><span class="p">(</span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">P.adj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">1e-75</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>But what if you wanted to select and filter?</p>

<p>We can do this with <strong>pipes</strong>. Pipes, are a fairly recent addition to R. They let you take the output of one function and send it directly to the next, which is useful when you need to do many things to the same data set. It was possible to do this before pipes were added to R, but it was much messier and more difficult. Pipes in R look like <code class="highlighter-rouge">%&gt;%</code> and are made available via the <code class="highlighter-rouge">magrittr</code> package, which is installed as part of <code class="highlighter-rouge">dplyr</code>.</p>

<blockquote class="comment">
  <h3 id="comment-pipes-on-rstudio"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Pipes on RStudio</h3>

  <p>With RStudio, you can type the pipe with</p>
  <ul>
    <li><kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> if you’re using a PC</li>
    <li><kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> if you’re using a Mac</li>
  </ul>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-and-filter"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select and filter</h3>

  <ol>
    <li>
      <p>Filter for genes on forward strand, select <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">Start</code>, <code class="highlighter-rouge">End</code>, <code class="highlighter-rouge">Chromosome</code> and display the first results using pipes</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">filter</span><span class="p">(</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">head</span><span class="p">()</span><span class="w">

       </span><span class="n">GeneID</span><span class="w">    </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Chromosome</span><span class="w">
</span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
</span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
</span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
</span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0034736</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
</span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>In the above code, we use the pipe to</p>
<ol>
  <li>Send the <code class="highlighter-rouge">annotatedDEgenes</code> dataset through <code class="highlighter-rouge">filter()</code></li>
  <li>Keep rows for genes on the ‘+’ strand of their chromosome</li>
  <li>Select through <code class="highlighter-rouge">select()</code> to keep only the <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">Start</code>, <code class="highlighter-rouge">End</code>, and <code class="highlighter-rouge">Chromosome</code> columns</li>
  <li>Show only the first 6 rows with <code class="highlighter-rouge">head</code></li>
</ol>

<blockquote class="comment">
  <h3 id="comment-no-need-to-give-1st-argument-to-the-functions"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> No need to give 1st argument to the functions</h3>

  <p>Since <code class="highlighter-rouge">%&gt;%</code> takes the object on its left and passes it as the first argument to the function on its right, we don’t need to explicitly include the data frame as an argument to the <code class="highlighter-rouge">filter()</code> and <code class="highlighter-rouge">select()</code> functions any more.</p>

</blockquote>

<p>Some may find it helpful to read the pipe like the word “then”. For instance, in the above example, we took the data frame <code class="highlighter-rouge">annotatedDEgenes</code>, <em>then</em> we <code class="highlighter-rouge">filter</code>ed for rows where <code class="highlighter-rouge">Strand</code> was +, <em>then</em> we <code class="highlighter-rouge">select</code>ed the <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">Start</code>, <code class="highlighter-rouge">End</code>, and <code class="highlighter-rouge">Chromosome</code> columns, <em>then</em> we showed only the first six rows.</p>

<p>The <code class="highlighter-rouge">dplyr</code> functions by themselves are somewhat simple, but by combining them into linear workflows with the pipe, we can accomplish complex manipulations of data frames.</p>

<p>If we want to create a new object with this smaller version of the data we can do so by assigning it to a new object.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-and-filter-to-create-new-object"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select and filter to create new object</h3>

  <ol>
    <li>
      <p>Create an <code class="highlighter-rouge">plus_strand_genes</code> from <code class="highlighter-rouge">annotatedDEgenes</code> after filter for genes on forward strand and selection of <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">Start</code>, <code class="highlighter-rouge">End</code>, <code class="highlighter-rouge">Chromosome</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plus_strand_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
 </span><span class="n">filter</span><span class="p">(</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
 </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Check the first 6 rows</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">plus_strand_genes</span><span class="p">)</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">    </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Chromosome</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0034736</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>This new object includes all of the data from this sample.</p>

<blockquote class="question">
  <h3 id="question-pipes-and-up-regulation"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Pipes and up-regulation</h3>

  <p>Starting with the <code class="highlighter-rouge">annotatedDEgenes</code> data frame, use pipes to</p>
  <ul>
    <li>subset the data to include only observations from chromosome 3L,with the log2 fold change is at least 2</li>
    <li>retain only the columns <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">P.adj</code>, and <code class="highlighter-rouge">log2.FC.</code>.</li>
  </ul>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">filter</span><span class="p">(</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chr3L"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">P.adj</span><span class="p">,</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<h2 id="create-columns">Create columns</h2>

<p>Frequently we want to create new columns based on the values in existing columns, e.g. to do unit conversions or find the ratio of values in two columns. For this we can use the <code class="highlighter-rouge">dplyr</code> function <code class="highlighter-rouge">mutate()</code>.</p>

<p>In the data frame, there is a column titled <code class="highlighter-rouge">log2.FC.</code>. This is the logarithmically-adjusted representation of the fold-change observed in <span class="notranslate">expression</span> of the genes in the transcriptomic experiment from which this data is derived. We can calculate the observed <span class="notranslate">expression</span> level relative to the reference according to the formula: <script type="math/tex">FC = 2^{log2.FC.}</script></p>

<p>Let’s create a column <code class="highlighter-rouge">FC</code> to our <code class="highlighter-rouge">annotatedDEgenes</code> data frame that show the observed <span class="notranslate">expression</span> as a multiple of the reference level.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-select-and-filter-to-create-new-object-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Select and filter to create new object</h3>

  <ol>
    <li>
      <p>Create a a column <code class="highlighter-rouge">FC</code> that show the observed <span class="notranslate">expression</span> as a multiple of the reference level</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">mutate</span><span class="p">(</span><span class="n">FC</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">**</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">head</span><span class="p">()</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">  </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Chromosome</span><span class="w">  </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">   </span><span class="n">FC</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.9743</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w">  </span><span class="m">-30.74090</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Kal1</span><span class="w"> </span><span class="m">0.05638871</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.5771</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w">  </span><span class="m">-28.74863</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">       </span><span class="n">chrX</span><span class="w"> </span><span class="m">10780892</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w"> </span><span class="m">0.12501930</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.8406</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w">  </span><span class="m">-28.22544</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">26869237</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w"> </span><span class="m">0.19208755</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.3224</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">   </span><span class="m">27.56599</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">       </span><span class="n">chrX</span><span class="w"> </span><span class="m">10778953</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w"> </span><span class="m">6.49774366</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.0673</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w">  </span><span class="m">-22.74596</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">      </span><span class="n">chr3L</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w"> </span><span class="m">0.23244132</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.9018</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w">  </span><span class="m">-21.88979</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w"> </span><span class="m">0.08820388</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-selected-mutation"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Selected mutation</h3>

  <p>Create a column with the gene length and just look at the <code class="highlighter-rouge">GeneID</code>, <code class="highlighter-rouge">Chromosome</code>, <code class="highlighter-rouge">Start</code>, <code class="highlighter-rouge">End</code>, and <code class="highlighter-rouge">Length</code> columns</p>

  <blockquote class="solution">
    <h3 id="solution-solution-7"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">mutate</span><span class="p">(</span><span class="n">Length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">End</span><span class="o">-</span><span class="n">Start</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Length</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<h2 id="group-and-summarize-data">Group and summarize data</h2>

<p>Many data analysis tasks can be approached using the “split-apply-combine” paradigm: split the data into groups, apply some analysis to each group, and
then combine the results.</p>

<p><code class="highlighter-rouge">dplyr</code> makes this very easy through the use of the <code class="highlighter-rouge">group_by()</code> function, which splits the data into groups. When the data is grouped in this way <code class="highlighter-rouge">summarize()</code> can be used to collapse each group into a single-row summary, by applying an aggregating or summary function to each group.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-group-and-summarize-data"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Group and summarize data</h3>

  <ol>
    <li>
      <p>Group by <code class="highlighter-rouge">Chromosome</code> and extract the number rows of data for each chromosome</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">summarize</span><span class="p">(</span><span class="n">n</span><span class="p">())</span><span class="w">
 </span><span class="c1"># A tibble: 5 x 2</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">`n()`</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">         </span><span class="m">24</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">         </span><span class="m">31</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">         </span><span class="m">27</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">         </span><span class="m">32</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">          </span><span class="m">16</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>Here the summary function used was <code class="highlighter-rouge">n()</code> to find the count for each group. Because it’s a common operation, the <code class="highlighter-rouge">dplyr</code> verb, <code class="highlighter-rouge">count()</code> is a “shortcut” that combines both <code class="highlighter-rouge">group_by</code> and <code class="highlighter-rouge">summarize</code>.</p>

<p>We can also apply many other functions to individual columns to get other summary statistics.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-group-and-summarize-data-1"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Group and summarize data</h3>

  <ol>
    <li>
      <p>View the highest fold change (<code class="highlighter-rouge">log2.FC.</code>) for each chromsome</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">summarize</span><span class="p">(</span><span class="nf">max</span><span class="p">(</span><span class="n">log2.FC.</span><span class="p">))</span><span class="w">
 </span><span class="c1"># A tibble: 5 x 2</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">`max(log2.FC.)`</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">               </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">                </span><span class="m">2.15</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">                </span><span class="m">2.41</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">                </span><span class="m">2.43</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">                </span><span class="m">2.36</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">                 </span><span class="m">2.70</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-summarizing-groups"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Summarizing groups</h3>

  <p>What are the longest genes in each chromosome?</p>

  <p>Hint: the function <code class="highlighter-rouge">abs()</code> returns the absolute value.</p>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">mutate</span><span class="p">(</span><span class="n">GeneLength</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">End</span><span class="o">-</span><span class="n">Start</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">summarize</span><span class="p">(</span><span class="n">max_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">GeneLength</span><span class="p">)))</span><span class="w">
</span><span class="c1"># A tibble: 5 x 2</span><span class="w">
  </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">max_length</span><span class="w">
  </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">           </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
</span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">           </span><span class="m">45932</span><span class="w">
</span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">           </span><span class="m">59682</span><span class="w">
</span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">          </span><span class="m">111613</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">           </span><span class="m">44404</span><span class="w">
</span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">            </span><span class="m">28429</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-missing-data-and-base-functions"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Missing data and base functions</h3>

  <p>R has many base functions like <code class="highlighter-rouge">mean()</code>, <code class="highlighter-rouge">median()</code>, <code class="highlighter-rouge">min()</code>, and <code class="highlighter-rouge">max()</code> that are useful to compute summary statistics. These are also called “built-in functions” because they come with R and don’t require that you install any additional packages.</p>

  <p>By default, all <strong>R functions operating on vectors that contains missing data will return NA</strong>. It’s a way to make sure that users know they have missing data, and make a conscious decision on how to deal with it. When dealing with simple statistics like the mean, the easiest way to ignore <code class="highlighter-rouge">NA</code> (the missing data) is to use <code class="highlighter-rouge">na.rm = TRUE</code> (<code class="highlighter-rouge">rm</code> stands for remove).</p>
</blockquote>

<p><code class="highlighter-rouge">group_by()</code> can also take multiple column names.</p>

<blockquote class="question">
  <h3 id="question-counting"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Counting</h3>

  <p>How many genes are found on each strand of each chromosome?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-9"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">count</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="p">)</span><span class="w">
</span><span class="c1"># A tibble: 10 x 3</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">Strand</span><span class="w">     </span><span class="n">n</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">  </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">12</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">17</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">14</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">15</span><span class="w">
 </span><span class="m">7</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">-</span><span class="w">          </span><span class="m">8</span><span class="w">
 </span><span class="m">8</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">24</span><span class="w">
 </span><span class="m">9</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">-</span><span class="w">          </span><span class="m">9</span><span class="w">
</span><span class="m">10</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">+</span><span class="w">          </span><span class="m">7</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<h2 id="reshape-data-frames">Reshape data frames</h2>

<p>While the tidy format is useful to analyze and plot data in R, it can sometimes be useful to transform the “long” tidy format, into the wide format. This transformation can be done with the <code class="highlighter-rouge">spread()</code> function provided by the <code class="highlighter-rouge">tidyr</code> package (also part of the <code class="highlighter-rouge">tidyverse</code>).</p>

<p><code class="highlighter-rouge">spread()</code> takes a data frame as the first argument, and two subsequent arguments:</p>
<ol>
  <li>the name of the column whose values will become the column names</li>
  <li>the name of the column whose values will fill the cells in the wide data</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-spread-a-key-value-pair-across-multiple-columns"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Spread a key-value pair across multiple columns</h3>

  <ol>
    <li>
      <p>Install and load <code class="highlighter-rouge">tidyr</code> library</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## load dyplr package</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="s2">"tidyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="tip">
        <h3 id="tip-tip-didnt-work-1"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Didn’t work?</h3>
        <p>You might need to install tidyr first</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"tidyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>        </div>
      </blockquote>
    </li>
    <li>
      <p>Create <code class="highlighter-rouge">annotatedDEgenes_wide</code> with chromosomes as row, strand as column and number of genes as values</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes_wide</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">summarize</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">spread</span><span class="p">(</span><span class="n">Strand</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">)</span><span class="w">
 </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes_wide</span><span class="w">
 </span><span class="c1"># A tibble: 5 x 3</span><span class="w">
 </span><span class="c1"># Groups:   Chromosome [5]</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w">   </span><span class="n">`+`</span><span class="w">   </span><span class="n">`-`</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w"> </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">         </span><span class="m">12</span><span class="w">    </span><span class="m">12</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">         </span><span class="m">14</span><span class="w">    </span><span class="m">17</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">         </span><span class="m">15</span><span class="w">    </span><span class="m">12</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">         </span><span class="m">24</span><span class="w">     </span><span class="m">8</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">           </span><span class="m">7</span><span class="w">     </span><span class="m">9</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>The opposite operation of <code class="highlighter-rouge">spread()</code> is taken care by <code class="highlighter-rouge">gather()</code>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-gather-columns-into-key-value-pairs"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Gather columns into key-value pairs</h3>

  <ol>
    <li>
      <p>Create a data frame with chromosomes, strand and number as columns from <code class="highlighter-rouge">annotatedDEgenes_wide</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes_wide</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">gather</span><span class="p">(</span><span class="n">Strand</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
 </span><span class="c1"># A tibble: 10 x 3</span><span class="w">
 </span><span class="c1"># Groups:   Chromosome [5]</span><span class="w">
    </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">Strand</span><span class="w">     </span><span class="n">n</span><span class="w">
    </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">  </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
  </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
  </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">17</span><span class="w">
  </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
  </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">-</span><span class="w">          </span><span class="m">8</span><span class="w">
  </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">-</span><span class="w">          </span><span class="m">9</span><span class="w">
  </span><span class="m">6</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">12</span><span class="w">
  </span><span class="m">7</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">14</span><span class="w">
  </span><span class="m">8</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">15</span><span class="w">
  </span><span class="m">9</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">24</span><span class="w">
 </span><span class="m">10</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">+</span><span class="w">          </span><span class="m">7</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>We specify the names of the new columns, and here add <code class="highlighter-rouge">-Chromosome</code> as this column shouldn’t be affected by the reshaping:</p>

<blockquote class="question">
  <h3 id="question-categorising-expression-levels"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Categorising <span class="notranslate">expression</span> levels</h3>

  <ol>
    <li>
      <p>Classify each gene as either “up-regulated” (fold change &gt; 1) or “down-regulated” (fold change &lt; 1) and create a table with <code class="highlighter-rouge">Chromosome</code> as rows, the two new labels as columns, and the number of genes in the cells.</p>

      <blockquote class="solution">
        <h3 id="solution-solution-10"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">mutate</span><span class="p">(</span><span class="n">exp_cat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">case_when</span><span class="p">(</span><span class="w">
     </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"up-regulated"</span><span class="p">,</span><span class="w">
     </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"down-regulated"</span><span class="w">
   </span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">count</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">exp_cat</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">spread</span><span class="p">(</span><span class="n">exp_cat</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">)</span><span class="w">
</span><span class="c1"># A tibble: 5 x 3</span><span class="w">
  </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">`down-regulated`</span><span class="w"> </span><span class="n">`up-regulated`</span><span class="w">
  </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">                 </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">          </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
</span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">                    </span><span class="m">17</span><span class="w">              </span><span class="m">7</span><span class="w">
</span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">                    </span><span class="m">16</span><span class="w">             </span><span class="m">15</span><span class="w">
</span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">                    </span><span class="m">11</span><span class="w">             </span><span class="m">16</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">                    </span><span class="m">18</span><span class="w">             </span><span class="m">14</span><span class="w">
</span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">                      </span><span class="m">8</span><span class="w">              </span><span class="m">8</span><span class="w">
</span></code></pre></div>        </div>

      </blockquote>
    </li>
    <li>
      <p>How could the code above be improved?</p>

      <p>Hint: think about the assumptions that we made about the data when writing this solution.</p>
    </li>
  </ol>
</blockquote>

<h1 id="visualization">Visualization</h1>

<p>In <span class="notranslate">RNA</span>-seq data analysis and other omics experiments, visualization are an important step to check the data, their quality and represent the results found for publication.</p>

<p><code class="highlighter-rouge">ggplot2</code> is a plotting package that makes it simple to create complex plots from data in a data frame. It provides a more programmatic interface for specifying what variables to plot, how they are displayed, and general visual properties. Therefore, we only need minimal changes if the underlying data change or if we decide to change from a bar plot to a scatter plot. This helps in creating publication quality plots with minimal amounts of adjustments and tweaking.</p>

<p><code class="highlighter-rouge">ggplot2</code> graphics are built step by step by adding new elements, i.e. layers:</p>

<div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">&lt;</span><span class="n">DATA</span><span class="o">&gt;</span><span class="p">,</span><span class="w"> </span><span class="n"><span class="notranslate">mapping</span></span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="o">&lt;</span><span class="n"><span class="notranslate">MAPPING</span>S</span><span class="o">&gt;</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="o">&lt;</span><span class="n">GEOM_FUNCTION</span><span class="o">&gt;</span><span class="p">()</span><span class="w">
</span></code></pre></div></div>

<p>Adding layers in this fashion allows for extensive flexibility and customization of plots.</p>

<h2 id="volcano-plot">Volcano plot</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-first-plot-step-by-step"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: First plot, step by step</h3>

  <ol>
    <li>
      <p>Load <code class="highlighter-rouge">ggplot2</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="n">library</span><span class="p">(</span><span class="n">ggplot2</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="tip">
        <h3 id="tip-tip-didnt-work-2"><i class="fa fa-lightbulb-o" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Didn’t work?</h3>
        <p>You might need to install ggplot2 first</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"ggplot2"</span><span class="p">)</span><span class="w">
</span></code></pre></div>        </div>
      </blockquote>
    </li>
    <li>
      <p>Bind a plot to a specific data frame</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p>A new window will open in the bottom-right panel</p>
    </li>
    <li>
      <p>Define a <span class="notranslate">mapping</span> (using the <code class="highlighter-rouge">aes</code> aesthetic function), by selecting the variables to be plotted and specifying how to present them in the graph, e.g. as x/y positions or characteristics such as size, shape, color, etc.</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">P.value</span><span class="p">))</span><span class="w">
</span></code></pre></div>      </div>

      <p>X-axis is now the log2 FC and and Y-axis the p-value</p>

      <blockquote class="comment">
        <h3 id="comment-data-format"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Data format</h3>

        <p>Some <code class="highlighter-rouge">ggplot2</code> functions need data in the ‘long’ format:</p>
        <ul>
          <li>a column for every dimension</li>
          <li>a row for every observation</li>
        </ul>

        <p>Well-structured data will save you lots of time when making figures with <code class="highlighter-rouge">ggplot2</code>.</p>
      </blockquote>
    </li>
    <li>
      <p>Add information about the graphical representations of the data in the plot (points, lines, bars) using <code class="highlighter-rouge">geoms</code> function added via the <code class="highlighter-rouge">+</code> operator</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">P.value</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
      </span><span class="n">geom_point</span><span class="p">()</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="comment">
        <h3 id="comment-graphical-representations-of-the-data-in-the-plot"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Graphical representations of the data in the plot</h3>

        <p><code class="highlighter-rouge">ggplot2</code> offers many different <code class="highlighter-rouge">geoms</code>; we will use some common ones today, including:</p>
        <ul>
          <li><code class="highlighter-rouge">geom_point()</code> for scatter plots, dot plots, etc.</li>
          <li><code class="highlighter-rouge">geom_boxplot()</code> for, well, boxplots!</li>
          <li><code class="highlighter-rouge">geom_line()</code> for trend lines, time series, etc.</li>
        </ul>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>We should obtain our first ggplot2 plot:</p>

<p><img src="../../images/rna-seq-counts-to-viz-in-r/firstPlot.png" alt="Our first ggplot2 plot" /></p>

<p>This plot is called a volcano plot, a type of scatterplot that shows statistical significance (P value) versus magnitude of change (fold change). The most upregulated genes are towards the right, the most downregulated genes are towards the left, and the most statistically significant genes are towards the top. With this plot, we can then quickly identify genes with large fold changes that are also statistically significant, i.e. probably the most biologically significant genes.</p>

<p>The current version of the plot is not really informative, mostly due to the high number of p-values close to zero. In order to resolve this, we will apply <code class="highlighter-rouge">-log10()</code> to the y-axis values.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-volcano-plot-with-log-values-on-y-axis"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Volcano plot with log values on y-axis</h3>

  <ol>
    <li>
      <p>Create volcano plot with log values on the y-axis</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span><span class="w">
     </span><span class="n">geom_point</span><span class="p">()</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v0.png" alt="Volcano Plot version 0" /></p>

<blockquote class="question">
  <h3 id="question-categorising-expression-levels-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Categorising <span class="notranslate">expression</span> levels</h3>

  <ol>
    <li>Why are there no points with log2 FC between -1 and 1?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-11"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The input file we used are the 130 genes with a significant adjusted p-values (below &lt; 0.05) and an absolute fold change higher than 2 (log2 FC &lt; -1 or log 2 FC &gt; 1).</p>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-the--sign"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> The <code class="highlighter-rouge">+</code> sign</h3>

  <p>The <code class="highlighter-rouge">+</code> in the <code class="highlighter-rouge">ggplot2</code> package is particularly useful because it allows you to modify existing <code class="highlighter-rouge">ggplot</code> objects. This means you can easily set up plot templates and conveniently explore different types of plots, so the above plot can also be generated with code like this:</p>

  <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Assign plot to a variable</span><span class="w">
</span><span class="n">de_genes_plot</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">)))</span><span class="w">

</span><span class="c1"># Draw the plot</span><span class="w">
</span><span class="n">de_genes_plot</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_point</span><span class="p">()</span><span class="w">
</span></code></pre></div>  </div>

  <p>The <code class="highlighter-rouge">+</code> sign used to add new layers must be placed at the end of the line containing the previous layer. If, instead, the <code class="highlighter-rouge">+</code> sign is added at the beginning of the line containing the new layer, <code class="highlighter-rouge">ggplot2</code> will not add the new layer and will return an error message.</p>

  <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># This is the correct syntax for adding layers</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">de_genes_plot</span><span class="w"> </span><span class="o">+</span><span class="w">
      </span><span class="n">geom_point</span><span class="p">()</span><span class="w">

</span><span class="c1"># This will not add the new layer and will return an error message</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">de_genes_plot</span><span class="w">
    </span><span class="o">+</span><span class="w"> </span><span class="n">geom_point</span><span class="p">()</span><span class="w">
</span></code></pre></div>  </div>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-some-extra-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i><span class="visually-hidden">comment</span> Some extra comments</h3>

  <p>Anything you put in the <code class="highlighter-rouge">ggplot()</code> function can be seen by any <code class="highlighter-rouge">geom</code> layers that you add (i.e., these are universal plot settings). This includes the <code class="highlighter-rouge">x-</code> and <code class="highlighter-rouge">y-axis</code> <span class="notranslate">mapping</span> you set up in <code class="highlighter-rouge">aes()</code>.</p>

  <p>You can also specify <span class="notranslate">mapping</span>s for a given <code class="highlighter-rouge">geom</code> independently of the <span class="notranslate">mapping</span>s defined globally in the <code class="highlighter-rouge">ggplot()</code> function.</p>
</blockquote>

<p>Building plots with <code class="highlighter-rouge">ggplot2</code> is typically an iterative process. We start by defining the dataset we’ll use, lay out the axes, and choose a <code class="highlighter-rouge">geom</code>. We can now modify this plot to extract more information from it.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-format-plot"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Format plot</h3>

  <ol>
    <li>
      <p>Add transparency (<code class="highlighter-rouge">alpha</code>) to avoid overplotting</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span><span class="w">
     </span><span class="n">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v1.png" alt="Volcano Plot version 1" /></p>
    </li>
    <li>
      <p>Add colors for all the points</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">)))</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">,</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"blue"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v2.png" alt="Volcano Plot version 2" /></p>
    </li>
    <li>
      <p>Color point based on their strand</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Strand</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
     </span><span class="n">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v3.png" alt="Volcano Plot version 3" /></p>

      <p>We use a vector as an input to the argument color and <code class="highlighter-rouge">ggplot2</code> provides a different color corresponding to different values in the vector.</p>
    </li>
    <li>
      <p>Add axis labels</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Strand</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
    </span><span class="n">geom_point</span><span class="p">(</span><span class="n">alpha</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">0.5</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
      </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"log2(Fold Change)"</span><span class="p">,</span><span class="w">
           </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"-log10(P-Value)"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>We have now a nice Volcano plot:</p>

<p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v5.png" alt="Volcano Plot version 5" /></p>

<blockquote class="question">
  <h3 id="question-standard-error-over-the--log10-of-the-adjusted-p-value"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Standard error over the <code class="highlighter-rouge">-log10</code> of the adjusted <code class="highlighter-rouge">P-value</code></h3>

  <p>Create a scatter plot of the standard error over the <code class="highlighter-rouge">-log10</code> of the adjusted <code class="highlighter-rouge">P-value</code> with the chromosomes showing in different colors. Make sure to give your plot relevant axis labels.</p>

  <blockquote class="solution">
    <h3 id="solution-solution-12"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ggplot(data = annotatedDEgenes, aes(x = -log10(P.adj), y = StdErr, color = Chromosome)) +
  geom_point() +
     labs(x = "-log10(P.adj)",
          y = "StdErr")
</code></pre></div>    </div>

    <p><img src="../../images/rna-seq-counts-to-viz-in-r/ggplotExercise01.png" alt="Exercise Plot 1" /></p>
  </blockquote>
</blockquote>

<h2 id="volcano-plot-by-chromosomes">Volcano plot by chromosomes</h2>

<p><code class="highlighter-rouge">ggplot2</code> has a special technique called faceting that allows us to split one plot into multiple plots based on a factor included in the dataset. We will use it to split our volcano plot into five panels, one for each chromosome.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-volcano-plot-by-chromosomes"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Volcano plot by chromosomes</h3>

  <ol>
    <li>
      <p>Split volcano plot by chromosome using <code class="highlighter-rouge">facet_grid</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Strand</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"log2(Fold Change)"</span><span class="p">,</span><span class="w">
       </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"-log10(P-Value)"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">facet_grid</span><span class="p">(</span><span class="n">.</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v6.png" alt="Volcano Plot version 6" /></p>
    </li>
    <li>
      <p>Split volcano plot by chromosome using <code class="highlighter-rouge">facet_grid</code> with plots stacked vertically</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Strand</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"log2(Fold Change)"</span><span class="p">,</span><span class="w">
       </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"-log10(P-Value)"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">facet_grid</span><span class="p">(</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">.</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v7.png" alt="Volcano Plot version 7" /></p>

      <p>The <code class="highlighter-rouge">facet_grid</code> geometry allows to explicitly specify how we want your plots to be arranged via formula notation (<code class="highlighter-rouge">rows ~ columns</code>). The <code class="highlighter-rouge">.</code> can be used as a placeholder that indicates only one row or column).</p>
    </li>
    <li>
      <p>Add white background using <code class="highlighter-rouge">theme_bw()</code> to make the plot more readable when printed</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Strand</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"log2(Fold Change)"</span><span class="p">,</span><span class="w">
       </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"-log10(P-Value)"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">facet_grid</span><span class="p">(</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">.</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_bw</span><span class="p">()</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="details">
        <h3 id="details-ggplot2-themes"><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> <code class="highlighter-rouge">ggplot2</code> themes</h3>

        <p>In addition to <code class="highlighter-rouge">theme_bw()</code>, which changes the plot background to white, <code class="highlighter-rouge">ggplot2</code> comes with several other themes which can be useful to quickly change the look of your visualization.</p>

        <ul>
          <li><code class="highlighter-rouge">theme_minimal()</code> and <code class="highlighter-rouge">theme_light() </code>are popular</li>
          <li><code class="highlighter-rouge">theme_void()</code> can be useful as a starting point to create a new hand-crafted theme</li>
        </ul>

        <p>The complete list of themes is available on the <a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">ggplot2 website</a>. The <code class="highlighter-rouge">ggthemes</code> package provides a wide variety of options (including an Excel 2003 theme). The <a href="https://www.ggplot2-exts.org/"><code class="highlighter-rouge">ggplot2</code> extensions website</a> provides a list of packages that extend the capabilities of <code class="highlighter-rouge">ggplot2</code>, including additional themes.</p>
      </blockquote>
    </li>
    <li>
      <p>Remove the grid</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">,</span><span class="w"> </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="o">-</span><span class="n">log10</span><span class="p">(</span><span class="n">P.value</span><span class="p">),</span><span class="w"> </span><span class="n">color</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Strand</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">geom_point</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">labs</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"log2(Fold Change)"</span><span class="p">,</span><span class="w">
       </span><span class="n">y</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"-log10(P-Value)"</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">facet_grid</span><span class="p">(</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">.</span><span class="p">)</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme_bw</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
  </span><span class="n">theme</span><span class="p">(</span><span class="n">panel.grid</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">element_blank</span><span class="p">())</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/rna-seq-counts-to-viz-in-r/volcanoPlot_v8.png" alt="Volcano Plot version 8" /></p>

<blockquote class="question">
  <h3 id="question-standard-error-over-the--log10-of-the-adjusted-p-value-1"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Standard error over the <code class="highlighter-rouge">-log10</code> of the adjusted <code class="highlighter-rouge">P-value</code></h3>

  <p>Create a scatter plot of the standard error over the <code class="highlighter-rouge">-log10</code> of the adjusted <code class="highlighter-rouge">P-value</code> with the chromosomes showing in different colors and one facet per strand. Make sure to give your plot relevant axis labels.</p>

  <blockquote class="solution">
    <h3 id="solution-solution-13"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ggplot(data = annotatedDEgenes, aes(x = -log10(P.adj), y = StdErr, color = Chromosome)) +
 geom_point() +
 labs(x = "-log10(P.adj)",
      y = "StdErr") +
 facet_grid(Strand ~ .)
</code></pre></div>    </div>

    <p><img src="../../images/rna-seq-counts-to-viz-in-r/ggplotExercise02.png" alt="Exercise Plot 2" /></p>

  </blockquote>
</blockquote>

<h2 id="barplot-of-the-number-of-de-genes">Barplot of the number of DE genes</h2>

<p>We would like now to make a barplot showing the number of differentially expressed genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-barplot-of-the-number-of-de-genes"><i class="fa fa-pencil" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Barplot of the number of DE genes</h3>

  <ol>
    <li>
      <p>Create a barplot with <code class="highlighter-rouge">geom_bar</code> function of the number of DE genes for each feature with one plot per chromosome</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">ggplot</span><span class="p">(</span><span class="n">data</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">aes</span><span class="p">(</span><span class="n">x</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Feature</span><span class="p">,</span><span class="w"> </span><span class="n">fill</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">))</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">geom_bar</span><span class="p">()</span><span class="w"> </span><span class="o">+</span><span class="w">
</span><span class="n">facet_grid</span><span class="p">(</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="n">.</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/rna-seq-counts-to-viz-in-r/barPlot_v1.png" alt="Bar plot 1" /></p>

<blockquote class="question">
  <h3 id="question-remove-the-legend"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Remove the legend</h3>

  <p>The chromosome is labeled on the individual plot facets so we don’t need the legend. Use the help file for <code class="highlighter-rouge">geom_bar</code> and any other online resources to remove the legend from the plot.</p>

  <blockquote class="solution">
    <h3 id="solution-solution-14"><i class="fa fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ggplot(data = annotatedDEgenes, aes(x = Feature, fill = Chromosome)) +
 geom_bar(show.legend = F) +
 facet_grid(Chromosome ~ .)
</code></pre></div>    </div>
    <p><img src="../../images/rna-seq-counts-to-viz-in-r/barPlot_v2.png" alt="Bar plot 2" /></p>

  </blockquote>
</blockquote>

<blockquote class="question">
  <h3 id="question-create-a-r-script-with-plots"><i class="fa fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Create a R script with plots</h3>

  <p>Take another few minutes to either improve one of the plots generated in this exercise or create a beautiful graph of your own (using the <a href="https://github.com/rstudio/cheatsheets/raw/master/data-visualization-2.1.pdf">RStudio <code class="highlighter-rouge">ggplot2</code> cheat sheet</a> for inspiration).</p>

  <p>Here are some ideas:</p>
  <ol>
    <li>Change the size or shape of the plotting symbol</li>
    <li>Change the name of the legend</li>
    <li>Change the label of the legend</li>
    <li>Use a different color palette (see the <a href="http://www.cookbook-r.com/Graphs/Colors_(ggplot2)/">cookbook here</a>).</li>
  </ol>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Data manipulation and visualization are important parts of any <span class="notranslate">RNA</span>-Seq analysis. <span class="notranslate">Galaxy</span> provides several tools for that as explained in several tutorials:</p>

<ul>
  <li><a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">Reference-based <span class="notranslate">RNA</span>-Seq</a></li>
  <li><a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">Visualization of <span class="notranslate">RNA</span>-seq results with heatmap2</a></li>
  <li><a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html">Visualization of <span class="notranslate">RNA</span>-seq results with Volcano Plot</a></li>
</ul>

<p>But sometimes we need more customization and then need to use programming languages as R or Python.</p>

<p>Working with a programming language (especially if it’s your first time) often feels intimidating, but the rewards outweigh any frustrations. An important secret of coding is that even experienced programmers find it difficult and frustrating at times – so if even the best feel that way, why let intimidation stop you? Given time and practice you will soon find it easier and easier to accomplish what you want.</p>

<p>Finally, we won’t lie; R is not the easiest-to-learn programming language ever created. So, don’t get discouraged! The truth is that even with the modest amount of R covered today, you can start using some sophisticated R software packages, and have a general sense of how to interpret an R script. Get through these lessons, and you are on your way to being an accomplished R user!</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fa fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>It is easy to import data into R from tabular formats. However, you still need to check that R has imported and interpreted your data correctly.</p>
</li>
                            
                            <li><p>Just like Galaxy, base R’s capabilities can be further enhanced by software packages developed by the user community.</p>
</li>
                            
                            <li><p>Use the <code class="highlighter-rouge">dplyr</code> package to manipulate data frames.</p>
</li>
                            
                            <li><p>Pipes can be used to combine simple operations into complex procedures.</p>
</li>
                            
                            <li><p>When creating plots with <code class="highlighter-rouge">ggplot2</code>, think about the graphics in layers (aesthetics, geometry, statistics, scale transformation, and grouping).</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Bérénice Batut, Fotis E. Psomopoulos, Toby Hodges, 2019 <b>RNA Seq Counts to Viz in R (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html">/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-rna-seq-counts-to-viz-in-r,
    author = "Bérénice Batut and Fotis E. Psomopoulos and Toby Hodges",
    title = "RNA Seq Counts to Viz in R (Galaxy Training Materials)",
    year = "2019",
    month = "10",
    day = "08"
    url = "\url{/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Björn Grüning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        

        <hr>
        <br>

        <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>
        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=RNA Seq Counts to Viz in R (Transcriptomics)">Loading...
        </iframe>

        <blockquote class="feedback">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Give us even more feedback on this content!</h3>
            <p>To give us more detailed feedback about these materials, please take a moment to fill in the extended <a href="https://docs.google.com/forms/d/e/1FAIpQLSc_GLZRpxZGAL9tN6DA1bE6WkNhmQdXT7B16TpOb-X4YwKUsQ/viewform?entry.1548889262=RNA Seq Counts to Viz in R (Transcriptomics)">Feedback Form</a>.</p>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Bérénice Batut, Fotis E. Psomopoulos, Toby Hodges)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    
</html>
