<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Compute and analyze biodiversity metrics with PAMPA toolsuite / Compute and analyze biodiversity metrics with PAMPA toolsuite / Ecology</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@649e2e9bca72de907031a6b13a02758bbb8b427b",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.20cf1f17.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Compute and analyze biodiversity metrics with PAMPA toolsuite">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-06-27 14:43:42 +0000">
<meta name="DC.creator" content="Coline Royaux">
<meta name="DC.creator" content="Yvan Le Bras">
<meta name="DC.creator" content="Dominique Pelletier">
<meta name="DC.creator" content="Jean-Baptiste Mihoub"><meta name="description" content="Learn to analyse Ecological data through Galaxy.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Ecology / Compute and analyze biodiversity metrics with PAMPA toolsuite / Hands-on: Compute and analyze biodiversity metrics with PAMPA toolsuite">
        <meta property="og:description" content="Learn to analyse Ecological data through Galaxy.">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Compute and analyze biodiversity metrics with PAMPA toolsuite",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-06-27 14:43:42 +0000",
  "datePublished": "2020-11-19 16:31:09 +0000",
  "identifier": "https://gxy.io/GTN:T00125",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Ecology",
    "description": "Learn to analyse Ecological data through Galaxy.",
    "url": "https://training.galaxyproject.org/training-material/topics/ecology/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Compute and analyze biodiversity metrics with PAMPA toolsuite",
  "url": "https://training.galaxyproject.org/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html",
  "version": 21,
  "timeRequired": "PT2H",
  "teaches": "Upload data from DATRAS portal of ICES\n - Pre-process population data with Galaxy\n - Learning how to use an ecological analysis workflow from raw data to graphical representations\n - Learning how to construct a Generalized Linear (Mixed) Model from a usual ecological question\n - Learning how to interpret a Generalized Linear (Mixed) Model",
  "keywords": [
    "Ecology",
    "Species population EBV class",
    "Community composition EBV class",
    "EBV dataset",
    "EBV workflow",
    "modeling"
  ],
  "description": "The questions this  addresses are:\n - How to evaluate properly populations and communities biological state with abundance data?\n - How does trawl exploited populations of Baltic sea, Southern Atlantic and Scotland are doing over time?\n - How to compute and analyze biodiversity metrics from abundance data?\n\n\nThe objectives are:\n - Upload data from DATRAS portal of ICES\n - Pre-process population data with Galaxy\n - Learning how to use an ecological analysis workflow from raw data to graphical representations\n - Learning how to construct a Generalized Linear (Mixed) Model from a usual ecological question\n - Learning how to interpret a Generalized Linear (Mixed) Model\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/colineroyaux/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/colineroyaux/",
      "name": "Coline Royaux",
      "image": "https://avatars.githubusercontent.com/colineroyaux",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/pndb/",
          "name": "Pôle National de Données de Biodiversité",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.pndb.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "This project ([`2020-1-NL01-KA203-064717`](https://erasmus-plus.ec.europa.eu/projects/search/details/2020-1-NL01-KA203-064717)) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.\n![eu flag with the text: with the support of the erasmus programme of the european union](https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg)\n![erasmus plus logo](https://www.erasmusplus.nl/assets/images/logo.png)\n",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4308-5617",
      "orcid": "https://orcid.org/0000-0003-4308-5617"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/yvanlebras/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/yvanlebras/",
      "name": "Yvan Le Bras",
      "image": "https://avatars.githubusercontent.com/yvanlebras",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/pndb/",
          "name": "Pôle National de Données de Biodiversité",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.pndb.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "This project ([`2020-1-NL01-KA203-064717`](https://erasmus-plus.ec.europa.eu/projects/search/details/2020-1-NL01-KA203-064717)) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.\n![eu flag with the text: with the support of the erasmus programme of the european union](https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg)\n![erasmus plus logo](https://www.erasmusplus.nl/assets/images/logo.png)\n",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/fairease/",
          "name": "Fair-Ease",
          "description": "FAIR-EASE is a RIA project funded under HORIZON-INFRA-2021-EOSC-01-04, and it involves a consortium of 25 partners from all over Europe.\n",
          "url": "https://fairease.eu/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/fnso2019/",
          "name": "OpenMetaPaper",
          "description": "This project ([`AAPFNSO2019OpenMetaPaper-14026`](https://www.ouvrirlascience.fr/submit-your-data-its-published/)) is funded with the support of the national Fund for Open Science programme of the French Minister of Higher Education, Research and Innovation. Their funding has supported development of data/metadata management tools and related tutorial for Ecology and FAIR topics.\n![French research ministry flag with the text: with the support of the National Fund for Open Science programme of french Minister of Higher Education, Research and Innovation](https://lamy-environnement.com/wp-content/uploads/2023/03/Logo-MESR.png)\n![FNSO logo](/training-material/shared/images/fnso.png)\n",
          "url": "https://www.ouvrirlascience.fr/national-fund-for-open-science/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-8504-068X",
      "orcid": "https://orcid.org/0000-0002-8504-068X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/dpelletier/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/dpelletier/",
      "name": "Dominique Pelletier",
      "image": "https://avatars.githubusercontent.com/dpelletier",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/jbmihoub/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/jbmihoub/",
      "name": "Jean-Baptiste Mihoub",
      "image": "https://avatars.githubusercontent.com/jbmihoub",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    }
  ],
  "contributor": [

  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Ecology",
      "description": "Learn to analyse Ecological data through Galaxy.",
      "url": "https://training.galaxyproject.org/training-material/topics/ecology/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0610",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0610",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0610"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.4264936",
      "name": "Associated Training Datasets"
    }
  ],
  "abstract": "This tutorial aims to present the PAMPA Galaxy workflow, how to use it to compute common"
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/ecology" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Ecology
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/ecology/faqs/" title="Check our FAQs for the Ecology topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-ecology">
    <h1 data-toc-skip>Compute and analyze biodiversity metrics with PAMPA toolsuite</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/colineroyaux/" class="contributor-badge contributor-colineroyaux"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/colineroyaux?s=36" alt="Coline Royaux avatar" width="36" class="avatar" />
    Coline Royaux</a><a href="/training-material/hall-of-fame/yvanlebras/" class="contributor-badge contributor-yvanlebras"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/yvanlebras?s=36" alt="Yvan Le Bras avatar" width="36" class="avatar" />
    Yvan Le Bras</a><a href="/training-material/hall-of-fame/dpelletier/" class="contributor-badge contributor-dpelletier"><img src="https://avatars.githubusercontent.com/dpelletier?s=36" alt="Dominique Pelletier avatar" width="36" class="avatar" />
    Dominique Pelletier</a><a href="/training-material/hall-of-fame/jbmihoub/" class="contributor-badge contributor-jbmihoub"><img src="https://avatars.githubusercontent.com/jbmihoub?s=36" alt="Jean-Baptiste Mihoub avatar" width="36" class="avatar" />
    Jean-Baptiste Mihoub</a>
		</td>
	</tr>
	

	

	

	

	

	

	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to evaluate properly populations and communities biological state with abundance data?</p>
</li>
        
        <li><p>How does trawl exploited populations of Baltic sea, Southern Atlantic and Scotland are doing over time?</p>
</li>
        
        <li><p>How to compute and analyze biodiversity metrics from abundance data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Upload data from DATRAS portal of ICES</p>
</li>
        
        <li><p>Pre-process population data with Galaxy</p>
</li>
        
        <li><p>Learning how to use an ecological analysis workflow from raw data to graphical representations</p>
</li>
        
        <li><p>Learning how to construct a Generalized Linear (Mixed) Model from a usual ecological question</p>
</li>
        
        <li><p>Learning how to interpret a Generalized Linear (Mixed) Model</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.4264936">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/workflows/" title="Compute and analyze biodiversity metrics with PAMPA toolsuite workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











<li class="btn btn-default supporting_material">


  <!-- dropdown with all recordings -->
  <a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/recordings/" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this material in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
  </a>


  <ul class="dropdown-menu">
    

    
      
      
      
    <li><a class="dropdown-item" href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/recordings/index.html#tutorial-recording-3-march-2022" title="View the recording for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (March 2022) - 2h</a>
    </li>
      
    
    <li><a class="dropdown-item" href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/recordings/" title="View all recordings for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> View All</a>
    </li>

  </ul>

  
</li>
  





            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/ecology/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Nov 19, 2020 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 27, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00125">https://gxy.io/GTN:T00125</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">5.0</a> (0 recent ratings, 2 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 21 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>This tutorial aims to present the PAMPA Galaxy workflow, how to use it to compute common
biodiversity metrics from species abundance data and analyse it through generalized
linear (mixed) models (GLM and GLMM). This workflow made up of 5 tools will allow you to process
temporal series data that include at least year, location and species sampled along with
abundance value and, finally, generate article-ready data products.</p>

<p>The PAMPA workflow is an ecological analysis workflow, like so, it is divided as to do a Community
analysis and a Population analysis separately in accordance with the biodiversity level described.
Only the last tool to create a plot from your GLM results is common to the two parts of this workflow.
Thus, in this tutorial, we’ll start by the Community analysis followed by the Population analysis.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-ecological-definitions-of-population-and-community"><button class="gtn-boxify-button details" type="button" aria-controls="details-ecological-definitions-of-population-and-community" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Ecological definitions of Population and Community</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Population: group of individuals of the same species interacting with each other.</p>

  <p>Community: group of several species interacting with each other.</p>

</blockquote>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/PAMPA-toolsuite-tutorial/PAMPA_workflow.png" alt="PAMPA toolsuite workflow. " width="3026" height="900" loading="lazy" /><a target="_blank" href="../../images/PAMPA-toolsuite-tutorial/PAMPA_workflow.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> PAMPA toolsuite workflow</figcaption></figure>

<p>Each part of this workflow has the same elementary steps :</p>
<ul>
  <li><strong>A first tool</strong> to compute metrics:
    <ul>
      <li>Community metrics (<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_communitymetrics/pampa_communitymetrics/0.0.2" title="Calculate community metrics tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Calculate community metrics</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span>):
Total abundance, Species richness, Shannon index, Simpson index, Hill index and Pielou index</li>
      <li>Population metrics (<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_presabs/pampa_presabs/0.0.2" title="Calculate presence absence table tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Calculate presence absence table</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span>):
Abundance, Presence-absence</li>
    </ul>
  </li>
  <li><strong>A second tool</strong> to compute Generalized Linear (Mixed) Models, testing the effect of <code class="language-plaintext highlighter-rouge">'site'</code>, <code class="language-plaintext highlighter-rouge">'year'</code> and/or
<code class="language-plaintext highlighter-rouge">'habitat'</code> on a metric selected by the user:
    <ul>
      <li>Community analysis (<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_glmcomm/pampa_glmcomm/0.0.2" title="Compute GLM on community data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute GLM on community data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span>):
The GLM(M) is computed on the whole dataset and, if a separation factor is selected by the user, GLM(M)s are computed on subsets
of the dataset (example: one GLM(M) per ecoregion)</li>
      <li>Population analysis (<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_glmsp/pampa_glmsp/0.0.2" title="Compute GLM on population data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute GLM on population data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span>):
One GLM(M) is computed on each species separately</li>
    </ul>
  </li>
  <li><strong>A third tool</strong> common for Community and Population analyses with at least <code class="language-plaintext highlighter-rouge">'year'</code> as a fixed effect to
create time-series plots from your GLM(M) results (<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_plotglm/pampa_plotglm/0.0.2" title="Create a plot from GLM data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Create a plot from GLM data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span>).
Two plots will be created for each GLM(M): one from the <em>“Estimate”</em> values of the GLM(M) and one from raw values from metrics file.</li>
</ul>

<blockquote class="details">
  <div class="box-title details-title" id="details-details-about-generalized-linear-mixed-models"><button class="gtn-boxify-button details" type="button" aria-controls="details-details-about-generalized-linear-mixed-models" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Details about Generalized Linear (Mixed) Models</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Generalized Linear Models (GLMs) and Generalized Linear Mixed Models (GLMMs) are two extensions of Linear Models (LMs).
As LMs uses only Gaussian distribution (for continuous variables), GLMs permits to analyse a wide spectrum of data
with various probability distributions (including Poisson distribution for count data and binomial distribution for
presence-absence data). To go further, GLMMs permits to handle pseudo-replication generated by non-independant
repeated observations by defining one or more random effect in your model.
Here, GLM(M) tools allows you to define <code class="language-plaintext highlighter-rouge">'year'</code> and/or <code class="language-plaintext highlighter-rouge">'site'</code> as random effects to take account of temporal
and/or spatial replicas. The other effects taken into account in the model that aren’t random are defined as ‘fixed’.</p>

</blockquote>

<p>In this tutorial, we’ll be working on data extracted from the data portal DATRAS (Database of Trawls Surveys) of the
International Council for the Exploration of the Sea (ICES). After pre-processing to fit the input format of the tools,
we’ll see how to calculate biodiversity metrics and construct properly, easily and with good practice, several GLMMs
to test the effect of <code class="language-plaintext highlighter-rouge">'year'</code> and <code class="language-plaintext highlighter-rouge">'site'</code> on the species richness of each survey studied and on the abundance
of each species.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#upload-and-pre-processing-of-the-data" id="markdown-toc-upload-and-pre-processing-of-the-data">Upload and pre-processing of the data</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#prepare-the-data" id="markdown-toc-prepare-the-data">Prepare the data</a></li>
    </ol>
  </li>
  <li><a href="#compute-community-and-population-metrics" id="markdown-toc-compute-community-and-population-metrics">Compute Community and Population metrics</a>    <ol>
      <li><a href="#creation-of-the-observation-unit-file" id="markdown-toc-creation-of-the-observation-unit-file">Creation of the observation unit file</a></li>
    </ol>
  </li>
  <li><a href="#glmms-and-plots-on-community-and-population-metrics" id="markdown-toc-glmms-and-plots-on-community-and-population-metrics">GLMMs and plots on Community and Population metrics</a>    <ol>
      <li><a href="#compute-glmms-and-create-plots-on-community-metrics" id="markdown-toc-compute-glmms-and-create-plots-on-community-metrics">Compute GLMMs and create plots on Community metrics</a></li>
      <li><a href="#compute-glmms-and-create-plots-on-population-metrics" id="markdown-toc-compute-glmms-and-create-plots-on-population-metrics">Compute GLMMs and create plots on Population metrics</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="upload-and-pre-processing-of-the-data">Upload and pre-processing of the data</h1>

<p>This first step consist of downloading and properly prepare the data to use it in PAMPA toolsuite.</p>

<h2 id="data-upload">Data upload</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a name (example: “DATRAS data analysis through PAMPA workflow”)
for you to find it again later if needed.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Import the CSV files from <a href="https://doi.org/10.5281/zenodo.4264936">Zenodo</a> via link with the three following links
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4264936/files/CPUE%20per%20length%20per%20area_A.csv?download=1
https://zenodo.org/record/4264936/files/CPUE%20per%20length%20per%20area_B.csv?download=1
https://zenodo.org/record/4264936/files/CPUE%20per%20length%20per%20area_C.csv?download=1
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>You may as well use the shared data library or directly download the data from the DATRAS data portal but the previous
method is preferred.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>The files used for this tutorial have been imported from DATRAS data portal in October 2020. Those files are
often updated on the data portal. So if you choose to download files directly from DATRAS there may be
some differences with the files on Zenodo and the final results of this tutorial.</p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-importing-data-from-datras-data-portal"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-datras-data-portal" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Importing data from DATRAS data portal</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Go to link: https://datras.ices.dk/Data_products/Download/Download_Data_public.aspx</li>
          <li>In the field ‘Data products’ choose ‘CPUE per length per area’</li>
          <li>In the field ‘Survey’ choose the survey of interest. Here, we chose to use ‘EVHOE’, ‘BITS’
and ‘SWC-IBTS’ surveys but you can’t select the three surveys at once in this field.
So, you’ll have to select one survey at a time (‘EVHOE’, ‘BITS’ and ‘SWC-IBTS’) and then
repeat the six following steps for each of the three surveys.</li>
          <li>In the field ‘Quarter(s)’ we chose to tick ‘All’</li>
          <li>In the field ‘Year(s)’ we chose to tick ‘All’</li>
          <li>In the field ‘Area(s)’ we chose to tick ‘All’</li>
          <li>In the field ‘Species’ we chose to tick ‘Standard species’ and ‘All’</li>
          <li>Click on ‘Submit’</li>
          <li>Tick ‘I accept the documents and their associated restrictions and agree to acknowledge the data source’</li>
          <li>Open the Galaxy Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</li>
          <li>Drop files on the dedicated area or click ‘Choose local files’</li>
          <li>Click on ‘Start’</li>
          <li>Click on ‘Close’ to return to Galaxy tools interface</li>
        </ul>

      </blockquote>
    </li>
    <li>
      <p>Make sure that each datasets has the following header <code class="language-plaintext highlighter-rouge">"Survey","Year","Quarter","Area","AphiaID",
"Species","LngtClass","CPUE_number_per_hour"</code> and that separator are “,”</p>
    </li>
    <li>In order to avoid mixing up the data files, we’ll add a tag to each file:
      <ul>
        <li>tag <code class="language-plaintext highlighter-rouge">#EVHOE</code> to the tabular data file of the EVHOE survey : “CPUE per length per area_A.csv”</li>
        <li>tag <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> to the tabular data file of the SWC-IBTS survey : “CPUE per length per area_B.csv”</li>
        <li>tag <code class="language-plaintext highlighter-rouge">#BITS</code> to the tabular data file of the BITS survey : “CPUE per length per area_C.csv”</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>If tags doesn’t display directly on your history, refresh the page.</p>
    </li>
    <li>
      <p>Convert datatype CSV to tabular for each data file</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-converting-the-file-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-converting-the-file-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Converting the file format</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the upper part <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong>, select <code class="language-plaintext highlighter-rouge">Convert CSV to tabular</code></li>   <li>Click the <strong>Create dataset</strong> button to start the conversion.</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>These three data files gather Catch Per Unit Effort (CPUE) observations, a relative abundance measure mostly used
to assess exploited fish species populations :</p>
<ul>
  <li>EVHOE (<span class="citation"><a href="#EVHOE">MAHE Jean-Claude 1987</a></span>) is a French survey on Southern Atlantic Bottom Trawl, the extracted file available on Zenodo goes from
1997 to 2018 and records CPUE for seven exploited fish species</li>
  <li>BITS (<span class="citation"><a href="#BITS">“<b>Fish trawl survey: ICES Baltic International Trawl Survey for commercial fish species. ICES Database of trawl surveys (DATRAS)</b>” 2010</a></span>) is an international survey on Trawl in Baltic sea, the extracted file available on Zenodo goes from 1991
to 2020 and records CPUE for three exploited fish species</li>
  <li>SWC-IBTS (<span class="citation"><a href="#SWCIBTS">“<b>DATRAS Scottish West Coast Bottom Trawl Survey (SWC-IBTS)</b>” 2013</a></span>) is a Scottish survey on bottom trawl in the west coast, the extracted file available on Zenodo goes
from 1985 to 2010 and records CPUE for six exploited fish species</li>
</ul>

<h2 id="prepare-the-data">Prepare the data</h2>

<p>Before starting the preparation of the data, we need to identify which inputs we need for the tools and their format
(same for Community or Population analyses except for a few details):</p>
<ul>
  <li>Input for the tools to compute metrics: Observation data file containing at least <code class="language-plaintext highlighter-rouge">'year'</code> and “location” sampled
that might be represented in one field “observation.unit”, “species.code” to name the species sampled,
and <code class="language-plaintext highlighter-rouge">'number'</code> as the abundance value</li>
  <li>Supplementary input for the tools to compute a GLM(M) analysis: Observation unit data file containing at least
“observation.unit” and <code class="language-plaintext highlighter-rouge">'year'</code>. It might also contain <code class="language-plaintext highlighter-rouge">'habitat'</code>, <code class="language-plaintext highlighter-rouge">'site'</code> and any other information
about the location at a given <code class="language-plaintext highlighter-rouge">'year'</code> that seems useful (examples: “temperature”, “substrate type”, “plant cover”)</li>
</ul>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-is-the-observation-unit-field"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-the-observation-unit-field" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What is the 'observation.unit' field?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The ‘observation.unit’ field is a primary key linking the observation data file and the metrics data file to an observation
unit data file. An unique ‘observation.unit’ identifier represents an unique sampling event at a given year and location.
Such a file permits to reduce the size of data files, if all informations about the sampling event (examples:
“temperature”,”substrate type”,”plant cover”) are gathered in the same dataframe as the observations data
(“species sampled”,”count”) same informations will be unnecessarily repeated in many lines.</p>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-the-observation-unit-nomenclature"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-the-observation-unit-nomenclature" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: The 'observation.unit' nomenclature</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>For this workflow, two nomenclatures are advised :</p>
  <ul>
    <li>
      <p>The PAMPA file nomenclature:
<code class="language-plaintext highlighter-rouge">[Two first letters of 'site'][Two last numbers of 'year'][Identifier of the precise sampled location in four numbers]</code></p>

      <p>The fields <code class="language-plaintext highlighter-rouge">'year'</code> and “location” will be automatically infered from this nomenclature in the first tools to
compute metrics</p>
    </li>
    <li>
      <p>The automatic nomenclature: <code class="language-plaintext highlighter-rouge">['year' in four numbers]_[Identifier of the precise sampled location in any character chain]</code></p>

      <p>This nomenclature will be automatically constructed in the first tools to compute metrics if the input file doesn’t already
have a “observation.unit” field but only <code class="language-plaintext highlighter-rouge">'year'</code> and “location”</p>
    </li>
  </ul>

  <p>If you already have your own nomenclature for the primary key linking your files, it will remain untouched by the tools but the
output data table from the first tools to compute metrics won’t contain <code class="language-plaintext highlighter-rouge">'year'</code> and “location” fields.
In this case and if you want to add such informations to your metrics data file, you can use the tool
<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.2" title="Join tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Join</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.2)</span>
with the metrics data file and the observation unit data file.</p>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-the-difference-between-fields-site-and-quot-location-quot"><button class="gtn-boxify-button details" type="button" aria-controls="details-the-difference-between-fields-site-and-quot-location-quot" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: The difference between fields ```'site'``` and &quot;location&quot;?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Those two fields both inform a geographical indication.</p>

  <p>“location” is the finest geographical level and <code class="language-plaintext highlighter-rouge">'site'</code> may represent a wider geographical range but can also
represent the same geographical level as “location”. This choice is up to your discretion, depending which geographical
level you want to take into account in your statistical analysis.</p>

</blockquote>

<h3 id="concatenation-of-data-files">Concatenation of data files</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-files-concatenation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data files concatenation</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0" title="Concatenate datasets tail-to-head (cat) tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets tail-to-head (cat)</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.1.0)</span> select the three <code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code>
datasets in <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Datasets to concatenate”</em></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Suppress the <code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> tags and add the <code class="language-plaintext highlighter-rouge">#Concatenate</code> tag to the Concatenated data file</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Verify if the three header lines have been included in the concatenate with <span class="tool" data-tool="Count1" title="Count occurrences of each record tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Count occurrences of each record</strong></span>
and following parameters :
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from dataset”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Concatenated data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Count occurrences of values in column(s)”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“How should the results be sorted?”</em>: <code class="language-plaintext highlighter-rouge">With the rarest value first</code> so multiple occurences can be seen directly at first lines of the resulting file</li>
      </ul>
    </li>
    <li>If the value <code class="language-plaintext highlighter-rouge">Survey</code> has more than <code class="language-plaintext highlighter-rouge">1</code> occurrence use <span class="tool" data-tool="Filter1" title="Filter data on any column using simple expressions tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter data on any column using simple expressions</strong></span>
with following parameters :
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Concatenated data file</li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c1!='Survey'</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Number of header lines to skip”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>How many header line(s) is left?</p>

        <p>Use <span class="tool" data-tool="Count1" title="Count occurrences of each record tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Count occurrences of each record</strong></span> and following parameters :</p>
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“from dataset”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Filtered data file</li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Count occurrences of values in column(s)”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“How should the results be sorted?”</em>: <code class="language-plaintext highlighter-rouge">With the rarest value first</code></li>
        </ul>

        <p><strong>Tip:</strong> Hit the rerun button <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> on the output of the previous run of <span class="tool" data-tool="Count1" title="Count occurrences of each record tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Count occurrences of each record</strong></span>,
and change the input dataset.</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>One header line is left as the value <code class="language-plaintext highlighter-rouge">Survey</code> has only <code class="language-plaintext highlighter-rouge">1</code> occurrence in the first column.</p>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h3 id="formating-of-data-files">Formating of data files</h3>

<p>Here we have data from the three surveys in one datatable (Filtered data file) with the following fields <code class="language-plaintext highlighter-rouge">"Survey",
"Year","Quarter","Area", "AphiaID","Species","LngtClass","CPUE_number_per_hour"</code>. We now need to define the proper
columns needed for the PAMPA workflow considering there is no primary key close to “observation.unit” :</p>
<ul>
  <li>“location” from “Area” associated with “Survey” to avoid mixing up areas from different surveys</li>
  <li><code class="language-plaintext highlighter-rouge">'year'</code> from “Year”</li>
  <li>“species.code” from “Species”</li>
  <li><code class="language-plaintext highlighter-rouge">'number'</code> from “CPUE_number_per_hour”</li>
</ul>

<p>This <code class="language-plaintext highlighter-rouge">#Concatenate</code> Filtered data file containing the three surveys will be used to create the observation unit data file and
compute Community metrics as the Community analysis can be automatically processed on each survey separately.
However, this file can’t be used as it is to compute Population metrics as it may contain data for several
populations of the same species that do not interact with each other. Therefore, we have to work on the three
<code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> original tabular files for the Population analysis.</p>

<p>Hence, the following manipulations will be applied not only on <code class="language-plaintext highlighter-rouge">#Concatenate</code> Filtered data file but also on
the three <code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> original tabular files used for the concatenation.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-the-quot-location-quot-field"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create the &quot;location&quot; field</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.0" title="Column Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.0)</span> with following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select cells from”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Filtered data file +
three <code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> tabular data files</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“using column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">([A-Z]{3}[A-Z]+)</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">\1-</code></li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>What did we do with this tool?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>We searched for every character chain matching the extended regular expression (Regex) <code class="language-plaintext highlighter-rouge">([A-Z]{3}[A-Z]+)</code>
in the first column “Survey” meaning every character chain of 4 or more capital letters. With the replacement <code class="language-plaintext highlighter-rouge">\1-</code>
we indicated to add <code class="language-plaintext highlighter-rouge">-</code> after the bracketed character chain. This manipulation is made in order to merge
properly columns “Area” and “Survey” with a separation.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/merge_cols/mergeCols1/1.0.2" title="Merge Columns tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge Columns</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.2)</span> with following parameters :
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select data”</em>: four <code class="language-plaintext highlighter-rouge">#Concatenate</code>, <code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code>
Column Regex Find And Replace data files</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Merge column”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“with column”</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
      </ul>

      <p>Check if the output data files has a supplementary column called “SurveyArea” and values with the survey ID and area ID
separated with <code class="language-plaintext highlighter-rouge">-</code>.</p>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-change-column-names"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Change column names</div>

  <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.0" title="Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.0)</span> with following parameters :</p>
  <ul>
    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select lines from”</em>: four <code class="language-plaintext highlighter-rouge">#Concatenate</code>, <code class="language-plaintext highlighter-rouge">#BITS</code>, <code class="language-plaintext highlighter-rouge">#EVHOE</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> Merged data files</li>
    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">Year</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">year</code></li>
      </ul>
    </li>
    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">Species</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">species.code</code></li>
      </ul>
    </li>
    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">CPUE_number_per_hour</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">number</code></li>
      </ul>
    </li>
    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">SurveyArea</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">location</code></li>
      </ul>
    </li>
  </ul>

  <p>Check if the columns are corresponding to <code class="language-plaintext highlighter-rouge">"Survey","year","Quarter","Area","AphiaID","species.code","LngtClass",
"number","location"</code> (pay attention to case).</p>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-other-tools-that-could-be-useful-to-pre-process-data"><button class="gtn-boxify-button details" type="button" aria-controls="details-other-tools-that-could-be-useful-to-pre-process-data" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Other tools that could be useful to pre-process data</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><span><strong>Add input name as column</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; <span><strong>Compute an expression on every row</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>
; <span><strong>Table Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; <span><strong>Join two files on column allowing a small difference</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>
; <span><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; <span><strong>Filter Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; <span><strong>melt</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>
; <span><strong>Unfold columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; <span><strong>Remove columns by heading</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>
; <span><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; <span><strong>Filter data on any column using simple expressions</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>
; <span><strong>Transpose rows/columns in a tabular file</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>; …</p>

</blockquote>

<h1 id="compute-community-and-population-metrics">Compute Community and Population metrics</h1>

<p>In this part of the tutorial, we’re starting to use tools from the PAMPA toolsuite to compute Community
and Population metrics. Then, we’ll be creating the observation unit file.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-compute-community-metrics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Compute Community metrics</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_communitymetrics/pampa_communitymetrics/0.0.2" title="Calculate community metrics tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Calculate community metrics</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Regex Find And Replace data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Choose the community metrics you want to compute”</em>: <code class="language-plaintext highlighter-rouge">All</code></li>
      </ul>

      <p>Check if the output Community metrics data file has the following column names <code class="language-plaintext highlighter-rouge">"year","location","number",
"species_richness","simpson","simpson_l","shannon","pielou","hill","observation.unit"</code>.</p>
    </li>
    <li>Tag the output Community metrics with <code class="language-plaintext highlighter-rouge">#Community</code></li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-compute-population-metrics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Compute Population metrics</div>

  <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_presabs/pampa_presabs/0.0.2" title="Calculate presence absence table tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Calculate presence absence table</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">#EVHOE</code>, <code class="language-plaintext highlighter-rouge">#BITS</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> Regex Find And Replace data files</p>

  <p>Check if the outputs Population metrics data files has the following column names <code class="language-plaintext highlighter-rouge">"year","location",
"species.code","number","presence_absence","observation.unit"</code>.</p>

</blockquote>

<h2 id="creation-of-the-observation-unit-file">Creation of the observation unit file</h2>

<p>Now that we have the “observation.unit” fields automatically created in the Community and Population
metrics files we can create the observation unit file from the <code class="language-plaintext highlighter-rouge">#Concatenate</code> Regex Find And Replace data file with the same
nomenclature for the “observation.unit” fields, namely:</p>

<p><code class="language-plaintext highlighter-rouge">['year' in four numbers]_[Identifier of the precise sampled location in any character chain]</code></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-the-observation-unit-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create the observation unit file</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.0" title="Column Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.0)</span> with following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select cells from”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Regex Find And Replace data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“using column”</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">([0-9]{4})</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">\1_</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/merge_cols/mergeCols1/1.0.2" title="Merge Columns tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge Columns</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.2)</span> with following parameters :
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select data”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Column Regex Find And Replace data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Merge column”</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“with column”</em>: <code class="language-plaintext highlighter-rouge">Column: 9</code></li>
      </ul>

      <p>Check if the output data file has a supplementary column called “yearlocation” and values with the <code class="language-plaintext highlighter-rouge">'year'</code> and “location” ID
separated with <code class="language-plaintext highlighter-rouge">_</code>.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.0" title="Column Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.0)</span> with following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select cells from”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Merge Columns data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“using column”</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">([0-9]{4})_</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">\1</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>What did we do with this tool?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>We removed the <code class="language-plaintext highlighter-rouge">_</code> we added in the first part of this “Hands_on” from the values of the second column
<code class="language-plaintext highlighter-rouge">'year'</code> to get proper numeric values back.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Cut columns from a table (cut) tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut columns from a table (cut)</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span>
with following parameters :
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“File to cut”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Column Regex Find And Replace data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Keep</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code> <code class="language-plaintext highlighter-rouge">Column: 2</code> <code class="language-plaintext highlighter-rouge">Column: 3</code> <code class="language-plaintext highlighter-rouge">Column: 4</code> <code class="language-plaintext highlighter-rouge">Column: 9</code> <code class="language-plaintext highlighter-rouge">Column: 10</code></li>
          </ul>
        </li>
      </ul>

      <p>Check if the output <code class="language-plaintext highlighter-rouge">#Concatenate</code> Advanced Cut data file has the following column names <code class="language-plaintext highlighter-rouge">"Survey","year","Quarter",
"Area","location","yearlocation"</code>.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1" title="Sort data in ascending or descending order tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Sort data in ascending or descending order</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span>
 with following parameters :
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Sort Query”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Advanced Cut data file</li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Number of header lines”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“1: Column selections”</em>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 6</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“in”</em>: <code class="language-plaintext highlighter-rouge">Ascending order</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Flavor”</em>: <code class="language-plaintext highlighter-rouge">Alphabetical sort</code></li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Output unique values”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Ignore case”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <p>The output <code class="language-plaintext highlighter-rouge">#Concatenate</code> Sort data file must have fewer lines than the <code class="language-plaintext highlighter-rouge">#Concatenate</code> Advanced Cut data file.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.0" title="Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.0)</span> with following parameters :
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate</code> Sort data file</li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">yearlocation</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">observation.unit</code></li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click <em>”+ Insert Check”</em>:
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">location</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">site</code></li>
          </ul>
        </li>
      </ul>

      <p>Here, we define <code class="language-plaintext highlighter-rouge">'site'</code> represents the same geographical level as “location”.</p>
    </li>
    <li>
      <p>Tag the output <code class="language-plaintext highlighter-rouge">#Concatenate</code> Regex Find And Replace data file with <code class="language-plaintext highlighter-rouge">#unitobs</code></p>

      <p>Check if the output <code class="language-plaintext highlighter-rouge">#Concatenate #unitobs</code> data file has the following column names <code class="language-plaintext highlighter-rouge">"Survey","year","Quarter",
"Area","site","observation.unit"</code>.</p>
    </li>
  </ol>

</blockquote>

<h1 id="glmms-and-plots-on-community-and-population-metrics">GLMMs and plots on Community and Population metrics</h1>

<p>Now that we have all our input files for GLM(M) tools ready, we can start computing statistical models to make
conclusions on our surveys. But before starting to use the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_glmcomm/pampa_glmcomm/0.0.2" title="Compute GLM on community data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute GLM on community data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span>
and the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_glmsp/pampa_glmsp/0.0.2" title="Compute GLM on population data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute GLM on population data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> tools, we have to think about the model we want to compute.
The tools allows to test the effects <code class="language-plaintext highlighter-rouge">Year'</code>, <code class="language-plaintext highlighter-rouge">Site</code> and/or <code class="language-plaintext highlighter-rouge">Habitat</code> on one interest variable at a time.
As we don’t have data about the <code class="language-plaintext highlighter-rouge">Habitat</code> we can’t test its effect. It is always better to take a temporal and a
geographical variable into account in an ecological model so we’ll test the effect of <code class="language-plaintext highlighter-rouge">Year</code> and <code class="language-plaintext highlighter-rouge">Site</code> with
a random effect of <code class="language-plaintext highlighter-rouge">Site</code> to take geographical pseudo-replication into account.</p>

<h2 id="compute-glmms-and-create-plots-on-community-metrics">Compute GLMMs and create plots on Community metrics</h2>

<p>For the Community analysis we have the choice to test the effect of <code class="language-plaintext highlighter-rouge">Year</code> and <code class="language-plaintext highlighter-rouge">Site</code> on:</p>
<ul>
  <li>“Total abundance”</li>
  <li>or “Species richness”</li>
  <li>or “Simpson index”</li>
  <li>or “1 - Simpson index”</li>
  <li>or “Shannon index”</li>
  <li>or “Pielou index”</li>
  <li>or “Hill index”</li>
</ul>

<p>As a GLM(M) permits to take into account only one interest variable at a time, we choose to take “Species richness”
as the interest variable, namely the quantity of different species at a given time and location.
This metric is located on the fourth column of the <code class="language-plaintext highlighter-rouge">#Concatenate #Community</code> metrics data file.</p>

<p>However, if you’re interested to test the effects of <code class="language-plaintext highlighter-rouge">Year</code> and <code class="language-plaintext highlighter-rouge">Site</code> on another metric you may compute
the GLM(M) tool again and choose any column of the <code class="language-plaintext highlighter-rouge">#Concatenate #Community</code> metrics data file containing one of
the metrics cited above.</p>

<p>As said above, analyses will be separated by the field “Survey” (first column) to avoid bias.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-compute-glmm-on-community-metrics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Compute GLMM on Community metrics</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_glmcomm/pampa_glmcomm/0.0.2" title="Compute GLM on community data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute GLM on community data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input metrics file”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #Community</code> metrics data file</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Unitobs informations file”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #unitobs</code> data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Interest variable from metrics file”</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Separation factor of your analysis from unitobs file”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Response variables”</em>: <code class="language-plaintext highlighter-rouge">Year</code> <code class="language-plaintext highlighter-rouge">Site</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Random effect?”</em>: <code class="language-plaintext highlighter-rouge">Site</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify advanced parameters”</em>: <code class="language-plaintext highlighter-rouge">No, use program defaults</code></li>
      </ul>
    </li>
  </ol>

  <p>The three outputs must have three tags <code class="language-plaintext highlighter-rouge">#Concatenate #Community #unitobs</code> and the first one named ‘GLM - results
from your community analysis’ must contain five lines. This file is formated to have one line per GLM processed,
it has been formated this way to make GLM results easier to represent and modify if needed. However, this format is
often not suited to read directly the results on Galaxy as Galaxy handles better tabular files with a lot of lines
rather than files with a lot of columns. So, if your file doesn’t display properly on Galaxy as a data table, we advise
to do the following optional step.</p>

  <ol>
    <li>(optional) <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0" title="Transpose rows/columns in a tabular file tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Transpose rows/columns in a tabular file</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span> with <code class="language-plaintext highlighter-rouge">#Concatenate #Community #unitobs</code> ‘GLM - results from your community analysis’ data file</li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-details-on-advanced-parameters-of-glm-tools"><button class="gtn-boxify-button details" type="button" aria-controls="details-details-on-advanced-parameters-of-glm-tools" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Details on advanced parameters of GLM tools</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>For now, there are two advanced parameters you can setup in the GLM tools:</p>
  <ul>
    <li>
      <p><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Distribution for model”</em> permits you to choose a probability distribution for your
model. When <code class="language-plaintext highlighter-rouge">Auto</code> is selected, the tool will automatically set the distribution on <code class="language-plaintext highlighter-rouge">Gaussian</code>, <code class="language-plaintext highlighter-rouge">Binomial</code>
or <code class="language-plaintext highlighter-rouge">Poisson</code> depending on the type of your interest variable but it may exist a better probability distribution
to fit your model. Mind that if you choose to use a random effect, “quasi” distribution can’t be used for your model.</p>
    </li>
    <li>
      <p><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“GLM object(s) as .Rdata output?”</em> permits you to get .Rdata file of your GLM(M)s if you
want to use it in your own R console.</p>
    </li>
  </ul>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-how-to-choose-a-proper-distribution-family-for-your-model"><button class="gtn-boxify-button details" type="button" aria-controls="details-how-to-choose-a-proper-distribution-family-for-your-model" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: How to choose a proper distribution family for your model?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The distribution family for a model is mainly selected by looking at the interest variable of the model:</p>
  <ul>
    <li>If your interest variable is a continuous numeric variable (weigth, size, mean, …) it follows a
<code class="language-plaintext highlighter-rouge">Gaussian</code> distribution so you should select <code class="language-plaintext highlighter-rouge">Gaussian</code>.</li>
    <li>If your interest variable is a discrete numeric variable (count) it follows a <code class="language-plaintext highlighter-rouge">Poisson</code> distribution
so you should select <code class="language-plaintext highlighter-rouge">Poisson</code>.</li>
    <li>If your interest variable is a binary variable (0-1, presence-absence) it follows a <code class="language-plaintext highlighter-rouge">Binomial</code> distribution
so you should select <code class="language-plaintext highlighter-rouge">Binomial</code>.</li>
  </ul>

  <p>These three distributions are the most commonly used, the other distributions are used when the interest variable
, the data and/or the model has specifics:</p>
  <ul>
    <li>When the model is over-dispersed, use a <code class="language-plaintext highlighter-rouge">Quasi-Poisson</code> (when interest variable is discrete) or a <code class="language-plaintext highlighter-rouge">Quasi-Binomial</code>
(when interest variable is binary) distribution.</li>
    <li>If your model doesn’t fit well and your interest variable is a <strong>positive</strong> continuous variable, you may try
with <code class="language-plaintext highlighter-rouge">Gamma</code> or <code class="language-plaintext highlighter-rouge">Inverse-Gaussian</code> distribution.</li>
  </ul>

</blockquote>

<h3 id="read-and-interpret-raw-glmm-outputs">Read and interpret raw GLMM outputs</h3>

<p>The GLM tool has three outputs:</p>
<ul>
  <li>The first output, ‘GLM - results from your community analysis’ data file, contains the summary of each GLMMs
on each lines.</li>
  <li>The second output, ‘Simple statistics on chosen variables’ file, contains descriptive statistics (mean, median, …)
of the interest variable for the whole input data table and for each selected factor levels combinations.
Here, each <code class="language-plaintext highlighter-rouge">Site</code> at each <code class="language-plaintext highlighter-rouge">Year</code>.</li>
  <li>The third output, ‘Your analysis rating file’, contains a notation for all analyses performed and notations for
each GLMMs processed. This rate is based on nine criterias commonly used in ecology to estimate wether a Linear
Model has been constructed properly or not. At the end of the file, red flags and advices to improve your model
are displayed also depending on the same criterias. This file is really useful to look at the bigger picture</li>
</ul>

<blockquote class="details">
  <div class="box-title details-title" id="details-details-on-criterias-used-in-your-analysis-rating-file"><button class="gtn-boxify-button details" type="button" aria-controls="details-details-on-criterias-used-in-your-analysis-rating-file" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Details on criterias used in 'Your analysis rating file'</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>For now, there are nine criterias taken into account to rate GLM(M)s in the Galaxy tools:</p>
  <ul>
    <li><em>Plan completeness</em>: If part of factor levels combinations are inexistant, bias can be induced. (0.5)</li>
    <li><em>Plan balance</em>: If factor levels combinations aren’t present in same quantity, bias can be induced. (0.5)</li>
    <li><em>NA proportion</em>: Lines with one or more NA value are ignored in Linear Models. If many NA values, there are
less observations taken in account in the model than expected. (1)</li>
    <li><em>Residuals dispersion</em> (DHARMa package): If the dispersion of residuals isn’t regular, the model is either under- or
over-dispersed. The distribution selected for the model isn’t right. (1.5)
To know if it is under- or over-dispersion, compute <em>“deviance”</em>/<em>“df.resid”</em> : if the result is &gt; 1 it is
over-dispersion, if the result is &lt; 1 it is under-dispersion.</li>
    <li><em>Residuals uniformity</em> (DHARMa package): If the residuals aren’t uniform, the distribution selected for the model
isn’t right. (1)</li>
    <li><em>Outliers proportion</em> (DHARMa package): If too much outliers, the distribution selected for the model isn’t right
or data is biased. (0.5)</li>
    <li><em>Zero-inflation</em> (DHARMa package): If too much zeros in data, bias can be induced. (1)</li>
    <li><em>Observations / factors ratio</em>: If the number of factors taken into account in the model exceeds 10% of the number
of observations (ex: 10 factors for 90 observations), bias can be induced. (1)</li>
    <li><em>Number of levels of the random effect</em>: If no random effect in the model or if the factor selected as random effect
has less than 10 levels. (1)</li>
  </ul>

</blockquote>

<h4 id="first-output-glm---results-from-your-community-analysis">First output: ‘GLM - results from your community analysis’</h4>

<p>First, on the ‘GLM - results from your community analysis’ data file, we see four analyses: <code class="language-plaintext highlighter-rouge">global</code> on the whole dataset
and one analysis for each of the three surveys <code class="language-plaintext highlighter-rouge">BITS</code>, <code class="language-plaintext highlighter-rouge">EVHOE</code> and <code class="language-plaintext highlighter-rouge">SWC-IBTS</code>. We also see the <code class="language-plaintext highlighter-rouge">Poisson</code> distribution has been
automatically selected by the tool. It seems right as the interest variable “Species richness” is a count.</p>

<p>Then, we see five fields that gives indexes to evaluate the quality of a model in comparison to another model performed
on the same data, so they can’t be used to compare the four models performed here :</p>
<ul>
  <li><em>“AIC”</em> Akaike information criterion. It is based on the number of factors. The lower AIC is better.</li>
  <li><em>“BIC”</em> Bayesian information criterion, is a derivate of AIC. It is based on the number of factors selected and on the
sample size. The lower BIC is better.</li>
  <li><em>“logLik”</em> log-likelihood, is an index that estimates the likelihood of obtaining the sample of observations used
for the model in the selected distribution. The higher logLik is better.</li>
  <li><em>“deviance”</em>, evaluates the model’s goodness of fit. The lower deviance is better.</li>
  <li><em>“df.resid”</em>, degrees of freedom in residuals used to calculate the adjusted R-square that indicates the fit quality.</li>
</ul>

<p>Three statistics about the random effect <code class="language-plaintext highlighter-rouge">Site</code> are given in this first file: standard deviation, number of observations taken
into account in the model and number of levels in the random effect.
As <code class="language-plaintext highlighter-rouge">Site</code> have been set as a random effect, pseudo-replication caused by the factor <code class="language-plaintext highlighter-rouge">Site</code> variable is taken into account to
fit the model as well as its effects on the interest variable but it is not tested so we can’t make any conclusion regarding
its effects on species richness.</p>

<p>At this point on, elements on fixed effects are displayed :</p>
<ul>
  <li><em>“Estimate”</em> is the mean value of the interest variable estimated by the model for a given factor.</li>
  <li><em>“Std.Err”</em> standard error, is the standard deviation of the <em>“Estimate”</em>. It measures uncertainty of the <em>“Estimate”</em>.</li>
  <li><em>“Zvalue”</em> (or <em>“Tvalue”</em> depending on the distribution) is the <em>“Estimate”</em> divided by the <em>“Std.Err”</em>.</li>
  <li><em>“Pvalue”</em> is the probability that the interest variable is not impacted by a given factor. If this probability is under a
certain value (commonly 5% in ecology so a p-value under 0.05), we determine the factor has a significant effect on the
interest variable.</li>
  <li><em>“IC_up”</em> and <em>“IC_inf”</em> are the upper and lower values of the confidence interval at 95% of the <em>“Estimate”</em>. The confidence
interval is the range of values that your <em>“Estimate”</em> may take 95% of the time if you re-sample your data.</li>
  <li><em>“signif”</em> significativity, “yes” if the <em>“Pvalue”</em> is inferior to 0.05. Tells you wether the tested factor has a significant
effect on your interest variable.</li>
</ul>

<p>If the factor used as fixed effect is defined as a qualitative variable, those seven fields will be provided for each factor’s level
and if it is a continuous variable they will be provided only once for the whole factor.</p>

<p>Here, the factor <code class="language-plaintext highlighter-rouge">'year'</code> may be considered a continuous variable as well as a qualitative variable. Only, it doesn’t provide the same
informations :</p>
<ul>
  <li>If we consider <code class="language-plaintext highlighter-rouge">'year'</code> as a continuous variable the model will inform on the trend of the interest variable on the whole time-series.
Making it more interesting to know about the global behaviour of the interest variable over time.</li>
  <li>If we consider <code class="language-plaintext highlighter-rouge">'year'</code> as a qualitative variable the model will inform on the relative mean of interest variable for each year
individually. Making it more interesting to know about each year individually and plot a detailed curve.</li>
</ul>

<p>As both these solutions are interesting, the tool is computing two GLMMs when <code class="language-plaintext highlighter-rouge">Year</code> is selected as a fixed effect:</p>
<ul>
  <li>one GLMM with <code class="language-plaintext highlighter-rouge">Year</code> as a continuous variable with results on fields <em>“year Estimate”</em>, <em>“year Std.Err”</em>, <em>“year Zvalue”</em>,
<em>“year Pvalue”</em>, <em>“year IC_up”</em>, <em>“year IC_inf”</em> and <em>“year signif”</em>.</li>
  <li>one GLMM with <code class="language-plaintext highlighter-rouge">Year</code> as a qualitative variable in the same fields with each <code class="language-plaintext highlighter-rouge">Year</code>’s levels as <em>“YYYY Estimate”</em>,
<em>“YYYY Std.Err”</em>, <em>“YYYY Zvalue”</em>, <em>“YYYY Pvalue”</em>, <em>“YYYY IC_up”</em>, <em>“YYYY IC_inf”</em> and <em>“YYYY signif”</em>.</li>
</ul>

<p>The <em>“(Intercept)”</em> value represents the reference value, it is rarely used for interpretations.
We see no significant effect on the ‘global’ model which seems logical as it considers the three surveys that are in
three geographical areas so three distinct communities. However, we observe the same thing with models on each of
the three surveys separately.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>What does this result with no significant effect found mean? How can we explain it?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>This result would mean that the communities sampled in this dataset haven’t change in the whole
time series. However, this result can’t be really trusted because the interest variable we selected is
the species richness and for the need of this tutorial we downloaded only part of the dataset with
a limited number of species. Hence, this dataset isn’t really suited to perform a community analysis
as only a little part of the community is considered.</p>

  </blockquote>

</blockquote>

<h4 id="second-output-simple-statistics-on-chosen-variables">Second output: ‘Simple statistics on chosen variables’</h4>

<p>The second output file doesn’t tell much about the statistic models, it contains details for you to have an outlook on
the interest variable of the input data file containing the community metrics. We can see in the ‘Base statistics’
part that the maximum species richness in the data file is 7 which is very low for trawl surveys and confirms this dataset
isn’t suited to perform community analyses.</p>

<h4 id="third-output-your-analysis-rating-file">Third output: ‘Your analysis rating file’</h4>

<p>Here, we already pointed out a major issue with the original data file for community analyses and we know they
aren’t robust. The third output will help us get a supplementary layer of understanding and take a step back on
the quality of the models performed.
We see the global rate is 3.5 out of 10 so it seems the analyses aren’t of good quality but we need to see
each analysis individually to make real conclusions. The best rate is for the analysis on <code class="language-plaintext highlighter-rouge">EVHOE</code> survey 4 out
of 8 which is a medium rate, the model isn’t necessarily bad but it can be improved. Indeed, data for this model
contains few NA values and it has a complete plan but not balanced. However, two very important criterias aren’t
checked: residuals haven’t a regular dispersion and aren’t uniform so it seems Poisson distribution doesn’t fit.
Besides, there isn’t enough factor’s levels in <code class="language-plaintext highlighter-rouge">'site'</code> to use it as a random effect. To get a relevant model we
should redo the model without <code class="language-plaintext highlighter-rouge">Site</code> as a random effect but in order to shorten this tutorial we won’t develop this
manipulation here.</p>

<p>You can see in the last part of the file ‘Red flags - advice’, some advices to improve your models but the issues
detected in models can come from different reasons (like here, the data isn’t representative enough of the whole
community) and may be hard to correct. Hence, you can try with another probability distribution but this issue may
come from the fact that the data isn’t suited to make an analysis on species richness.
Overall, the four models aren’t really trust-worthy.</p>

<h4 id="summary">Summary</h4>

<p>With all this interpretation made from raw results, we still don’t really visualize these results clearly. For this, it
could be nice to get a graphical representation of the GLMMs results and the interest variable.
Here, for these models, we know these plots won’t be trust-worthy but it can still be interesting.</p>

<h3 id="create-and-read-plots">Create and read plots</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-plots-from-community-metrics-file-and-glmm-results"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create plots from Community metrics file and GLMM results</div>

  <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_plotglm/pampa_plotglm/0.0.2" title="Create a plot from GLM data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Create a plot from GLM data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input glm results file”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #Community #unitobs</code> ‘GLM - results
from your community analysis’ data file</li>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Data table used for glm”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #Community</code> metrics data file</li>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Unitobs table used for glm”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #unitobs</code> data file</li>
  </ul>

  <p>The output must be a data collection with four PNG files.</p>

</blockquote>

<figure id="figure-2" style="max-width: 90%;"><img src="../../images/PAMPA-toolsuite-tutorial/Community.png" alt="Community analysis plots. " width="2401" height="2243" loading="lazy" /><a target="_blank" href="../../images/PAMPA-toolsuite-tutorial/Community.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Community analysis plots</figcaption></figure>

<p>Each PNG file contains two plots :</p>
<ul>
  <li>In the upper plot (blue), the line represents the <em>“Estimate”</em> values for each year and the confidence interval
is the light blue ribbon. There is no significantly different year here but, usually, it is represented by a white
dot inside the blue point of the <em>“Estimate”</em> value of the year. The writing on top of the plot is the <em>“Estimate”</em>
value for the whole time-series (year as continuous variable), if nothing else is written it isn’t significant.</li>
  <li>In the lower plot (purple), the line represents the observed mean of the species richness for each year (calculated
directly from the input metrics data table). If the interest variable had been a presence-absence, it would have been
a presence proportion and if it had been an abundance it would have been a raw abundance (sum of all abundances each year).</li>
</ul>

<p>We’ll ignore the ‘global’ analysis plot as it associate data from different communities. We see on both plots of each of the
three surveys that, as predicted, species richness haven’t varied much as few species are taken into account in the analysed
data set and these species are very commonly found as we selected only ‘Standard species’ when downloading the datasets.
Hence, mean species richness are globally around the maximum number of species taken into account in the surveys (7 for
<code class="language-plaintext highlighter-rouge">SWC-IBTS</code>, 6 for <code class="language-plaintext highlighter-rouge">EVHOE</code> and 3 for <code class="language-plaintext highlighter-rouge">BITS</code>). Moreover, confidence intervals are very wide and global trends are around 1.</p>

<h2 id="compute-glmms-and-create-plots-on-population-metrics">Compute GLMMs and create plots on Population metrics</h2>

<p>For the Population analysis we have the choice to test the effect of <code class="language-plaintext highlighter-rouge">Year</code> and <code class="language-plaintext highlighter-rouge">Site</code> on:</p>
<ul>
  <li>“Abundance”</li>
  <li>or “Presence-absence”</li>
</ul>

<p>As a GLM(M) permits to take into account only one interest variable at a time, we choose to take
“Abundance” as the interest variable, namely the quantity of a given species at a given time and location.
This metric is located on the fourth column of the <code class="language-plaintext highlighter-rouge">#EVHOE</code>, <code class="language-plaintext highlighter-rouge">#BITS</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> Population metrics data files.</p>

<p>However, if you’re interested to test the effects of <code class="language-plaintext highlighter-rouge">Year</code> and <code class="language-plaintext highlighter-rouge">Site</code> on “Presence-absence” you may compute
the GLM(M) tool again and choose the fifth column of the <code class="language-plaintext highlighter-rouge">#EVHOE</code>, <code class="language-plaintext highlighter-rouge">#BITS</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> Population metrics data
files.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-compute-glmm-on-population-metrics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Compute GLMM on Population metrics</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_glmsp/pampa_glmsp/0.0.2" title="Compute GLM on population data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute GLM on population data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input metrics file”</em>: <code class="language-plaintext highlighter-rouge">#EVHOE</code>, <code class="language-plaintext highlighter-rouge">#BITS</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> metrics (here presence absence) data files</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Unitobs informations file”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #unitobs</code> data file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Interest variable from metrics file”</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Response variables”</em>: <code class="language-plaintext highlighter-rouge">Year</code> <code class="language-plaintext highlighter-rouge">Site</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Random effect?”</em>: <code class="language-plaintext highlighter-rouge">Site</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Specify advanced parameters”</em>: <code class="language-plaintext highlighter-rouge">No, use program defaults</code></li>
      </ul>

      <p>The three outputs for each survey must have three tags <code class="language-plaintext highlighter-rouge">#Concatenate #SurveyID #unitobs</code>.</p>

      <p>The first files of each survey named ‘GLM - results from your population analysis’ must contain one line
per GLM processed, it has been formated this way to make GLM results easier to represent and modify if needed.
However, this format is often not suited to read directly the results on Galaxy as Galaxy handles better tabular
files with a lot of lines rather than files with a lot of columns. So, if your files doesn’t display properly
on Galaxy as a data table, we advise to do the following optional step.</p>
    </li>
    <li>(optional) <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0" title="Transpose rows/columns in a tabular file tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Transpose rows/columns in a tabular file</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span>
with <code class="language-plaintext highlighter-rouge">#Concatenate #EVHOE #unitobs</code> <code class="language-plaintext highlighter-rouge">#BITS #Concatenate #unitobs</code> <code class="language-plaintext highlighter-rouge">#Concatenate #SWCIBTS #unitobs</code>
‘GLM - results from your population analysis’ data files</li>
  </ol>

</blockquote>

<h3 id="read-and-interpret-raw-glm-outputs">Read and interpret raw GLM outputs</h3>

<p>The three outputs for each time the population GLM tool runs are the same as in the community GLM tool described in
the previous part 3. 1. 1. Read and interpret raw GLM outputs.
For the need of this tutorial, we’ll be analyzing only the <code class="language-plaintext highlighter-rouge">BITS</code> survey analysis results.</p>

<h4 id="first-output-glm---results-from-your-population-analysis">First output: ‘GLM - results from your population analysis’</h4>

<p>First, on the <code class="language-plaintext highlighter-rouge">#BITS #Concatenate #unitobs</code> ‘GLM - results from your population analysis’ data file, we see three analyses
one for each species in the <code class="language-plaintext highlighter-rouge">BITS</code> survey: <code class="language-plaintext highlighter-rouge">Gadus morhua</code> (Atlantic cod), <code class="language-plaintext highlighter-rouge">Platichthys flesus</code> (European flounder) and
<code class="language-plaintext highlighter-rouge">Pleuronectes platessa</code> (European plaice).</p>

<figure id="figure-3" style="max-width: 90%;"><img src="../../images/PAMPA-toolsuite-tutorial/Fish-BITS.png" alt="BITS survey standard fish species. " width="955" height="233" loading="lazy" /><a target="_blank" href="../../images/PAMPA-toolsuite-tutorial/Fish-BITS.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> BITS survey standard fish species</figcaption></figure>

<p>We also see the <code class="language-plaintext highlighter-rouge">Gaussian</code> distribution has been automatically selected by the tool. It seems right even if the interest variable
“Abundance” is usually a count, here it is a Catch Per Unit Effort abundance so it isn’t a round value.</p>

<p>Then, the details of what contains each field in this file has been developped in the previous part on community analyses 3. 1. 1. 1. First output: ‘GLM - results from your community analysis’.</p>

<p>Three statistics about the random effect <code class="language-plaintext highlighter-rouge">Site</code> are given in this first file: standard deviation, number of observations taken
into account in the model and number of levels in the random effect.
As <code class="language-plaintext highlighter-rouge">'site'</code> have been set as a random effect, pseudo-replication caused by the factor <code class="language-plaintext highlighter-rouge">Site</code> variable is taken into account to
fit the model as well as its effects on the “Abundance” but it is not tested so we can’t make any conclusion regarding
its effects on CPUE abundance.</p>

<p>As said above, the tool is computing two GLMMs when <code class="language-plaintext highlighter-rouge">Year</code> is selected as a fixed effect:</p>
<ul>
  <li>one GLMM with <code class="language-plaintext highlighter-rouge">Year</code> as a continuous variable with results on fields <em>“year Estimate”</em>, <em>“year Std.Err”</em>, <em>“year Zvalue”</em>,
<em>“year Pvalue”</em>, <em>“year IC_up”</em>, <em>“year IC_inf”</em> and <em>“year signif”</em>.</li>
  <li>one GLMM with <code class="language-plaintext highlighter-rouge">Year</code> as a qualitative variable in the same fields with each <code class="language-plaintext highlighter-rouge">Year</code>’s levels as <em>“YYYY Estimate”</em>,
<em>“YYYY Std.Err”</em>, <em>“YYYY Zvalue”</em>, <em>“YYYY Pvalue”</em>, <em>“YYYY IC_up”</em>, <em>“YYYY IC_inf”</em> and <em>“YYYY signif”</em>.</li>
</ul>

<p>The <em>“(Intercept)”</em> value represents the reference value, it is rarely used for interpretations.</p>

<p>Regarding the significance of the effect <code class="language-plaintext highlighter-rouge">Year</code> as continuous variable:</p>
<ul>
  <li>It isn’t significant in the <code class="language-plaintext highlighter-rouge">Gadus morhua</code> model.</li>
  <li>It is significant in the <code class="language-plaintext highlighter-rouge">Platichthys flesus</code> model.</li>
  <li>It is significant in the <code class="language-plaintext highlighter-rouge">Pleuronectes platessa</code> model.</li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>What does these results on the effect <code class="language-plaintext highlighter-rouge">Year</code> as continuous variable mean?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The question behind the test of this effect is: “Is the global temporal trend different from 1?”
When it isn’t significant as in the <code class="language-plaintext highlighter-rouge">Gadus morhua</code> model, it means the <em>“year Estimate”</em> value isn’t
significantly different from 1 so the interest variable “Abundance” hasn’t varied much through time.
When it is significant as in the two other models, the <em>“year Estimate”</em> value is significantly different
from 1 so the interest variable “Abundance” has varied through time.</p>

  </blockquote>

</blockquote>

<p>These results are consistent with the results on the significance of the effects <code class="language-plaintext highlighter-rouge">Year</code> as qualitative variable:</p>
<ul>
  <li>Only one year out of 30 considered significantly different in the <code class="language-plaintext highlighter-rouge">Gadus morhua</code> model. On year 1995.</li>
  <li>14 years out of 30 considered significantly different in the <code class="language-plaintext highlighter-rouge">Platichthys flesus</code> model. On years 2001, 2002, 2005,
2007, 2008, 2010 to 2017 and 2019.</li>
  <li>8 years out of 30 considered significantly different in the <code class="language-plaintext highlighter-rouge">Pleuronectes platessa</code> model. From year 2012 to 2019.</li>
</ul>

<h4 id="second-output-simple-statistics-on-chosen-variables-1">Second output: ‘Simple statistics on chosen variables’</h4>

<p>The second output file doesn’t tell much about the statistic models, it contains details for you to have an outlook on
the interest variable of the input data file containing the population metrics. We can see for example in the ‘Statistics
per combination of factor levels’ part that the median value for the CPUE abundance in the location “22” in 1991 is 6.862473.</p>

<h4 id="third-output-your-analysis-rating-file-1">Third output: ‘Your analysis rating file’</h4>

<p>The third output will help us get a supplementary layer of understanding and take a step back on the quality of
the models performed.
We see the global rate is 5.5 out of 10 so it seems the analyses are of better quality than the community analysis but
we need to see each analysis individually to make real conclusions.
The best rates are for the analyses on <code class="language-plaintext highlighter-rouge">Gadus morhua</code> and <code class="language-plaintext highlighter-rouge">Platichthys flesus</code> 5.5 out of 8 which is a nice rate. However,
the models can still be improved by removing the random effect. As residuals aren’t uniform, we could select another
distribution but dispersion of residuals and residuals outliers are regular so we don’t advise it.
All analyses checked an important critera: the dispersion of residuals is regular so, more broadly we don’t
advise changing the distribution for any of these models.</p>

<p>You can see in the last part of the file ‘Red flags - advice’, some advices to improve your models but the issues
detected in models can come from different reasons and may be hard to correct. Hence, if you really want to improve
your models, you can try to get a tighter geographical resolution in order to get more locations and sites to keep
the random effect on <code class="language-plaintext highlighter-rouge">'site'</code>. You can also try to remove biased observations from your original dataset by finding out
more about the sampling methods used (years with different methods or different fleet for example).</p>

<h4 id="summary-1">Summary</h4>

<p>With all this interpretation made from raw results, we still don’t really visualize these results clearly. For this, it
could be nice to get a graphical representation of the GLMMs results and the interest variable.
Here, for these models, the plots we’ll create will certainly be more trust-worthy than the previous Community analyses.
Hence, more interesting to look at.</p>

<h3 id="create-and-read-plots-1">Create and read plots</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-plots-from-population-metrics-file-and-glmm-results"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create plots from Population metrics file and GLMM results</div>

  <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/pampa_plotglm/pampa_plotglm/0.0.2" title="Create a plot from GLM data tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Create a plot from GLM data</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with following parameters:</p>
  <ul>
    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input glm results file”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #EVHOE #unitobs</code> <code class="language-plaintext highlighter-rouge">#BITS #Concatenate #unitobs</code>
<code class="language-plaintext highlighter-rouge">#Concatenate #SWCIBTS #unitobs</code> ‘GLM - results from your population analysis’ data files</li>
    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Data table used for glm”</em>: <code class="language-plaintext highlighter-rouge">#EVHOE</code>, <code class="language-plaintext highlighter-rouge">#BITS</code> and <code class="language-plaintext highlighter-rouge">#SWCIBTS</code> metrics data files</li>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Unitobs table used for glm”</em>: <code class="language-plaintext highlighter-rouge">#Concatenate #unitobs</code> data file</li>
  </ul>

  <p>The three outputs must be a data collection.</p>

</blockquote>

<p>See the previous part 3. 1. 2. Create and read plots to get details on output PNG files.
As the interest variable is “Abundance”, the plot with a purple line represents raw abundance: the sum of
all abundances of the studied species for one year.</p>

<figure id="figure-4" style="max-width: 90%;"><img src="../../images/PAMPA-toolsuite-tutorial/BITS.png" alt="BITS population analysis plots. " width="2354" height="2243" loading="lazy" /><a target="_blank" href="../../images/PAMPA-toolsuite-tutorial/BITS.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> BITS population analysis plots</figcaption></figure>

<p>On the <code class="language-plaintext highlighter-rouge">Gadus morhua</code> plots, only small variations of the abundance occured between 1991 and 2020 apart for year 1995
where we see a huge augmentation for the estimated value and the raw abundance (value almost tripled). As the values
are coming back to an average range of values the following year, there should be some bias or even a count or
typing mistake in the dataset.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>How can we see if there is a mistake in the dataframe? How can we deal with it?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>To see if there is an abnormal value in the <code class="language-plaintext highlighter-rouge">'number'</code> field representing the CPUE abundance we’ll use
the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.1.1" title="Sort data in ascending or descending order tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Sort data in ascending or descending order</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span>
tool in the “Text Manipulation” tool collection with following parameters :</p>
    <ul>
      <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Sort Query”</em>: <code class="language-plaintext highlighter-rouge">#BITS</code> ‘Calculate presence absence table’ data file</li>
      <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>“Number of header lines”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      <li><em>“1: Column selections”</em>
        <ul>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
          <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“in”</em>: <code class="language-plaintext highlighter-rouge">Descending order</code></li>
          <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Flavor”</em>: <code class="language-plaintext highlighter-rouge">Fast numeric sort (-n)</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Output unique values”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Ignore case”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        </ul>
      </li>
    </ul>

    <p>The first value should be a measure of <code class="language-plaintext highlighter-rouge">Gadus morhua</code> on year “1995” and location “BITS-23”, this
value is of <em>‘13253.12023’</em> which is way higher than the next biggest value <em>‘4709.504184’</em>.
This value may create bias in the statistical model so, even if we won’t develop it here, you can
remove this line from the <code class="language-plaintext highlighter-rouge">#BITS</code> ‘Calculate presence absence table’ data file and re-run the model.</p>

  </blockquote>

</blockquote>

<p>We see on both plots of <code class="language-plaintext highlighter-rouge">Platichthys flesus</code> and <code class="language-plaintext highlighter-rouge">Pleuronectes platessa</code> that CPUE abundance has significantly increased
since 30 years. Both species had a first CPUE abundance increase around 2000 that can be explained by a fleet change in
the trawl survey around this time. Then, for <code class="language-plaintext highlighter-rouge">Platichthys flesus</code>, after a decrease in 2003 had a more or less stable
CPUE abundance. For <code class="language-plaintext highlighter-rouge">Pleuronectes platessa</code> a second wave of increase occured between 2008 and 2013 to decrease after 2018.</p>

<figure id="figure-5" style="max-width: 90%;"><img src="../../images/PAMPA-toolsuite-tutorial/EVHOE.png" alt="EVHOE population analysis plots. " width="3177" height="1515" loading="lazy" /><a target="_blank" href="../../images/PAMPA-toolsuite-tutorial/EVHOE.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> EVHOE population analysis plots</figcaption></figure>
<figure id="figure-6" style="max-width: 90%;"><img src="../../images/PAMPA-toolsuite-tutorial/SWCIBTS.png" alt="SWCIBTS population analysis plots. " width="3177" height="1991" loading="lazy" /><a target="_blank" href="../../images/PAMPA-toolsuite-tutorial/SWCIBTS.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> SWCIBTS population analysis plots</figcaption></figure>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial, you analyzed abundance data from three trawl surveys with an ecological analysis workflow.
You learned how to pre-process abundance data with Galaxy and compute Community and Population metrics.
You know now how to construct a proper GL(M)M easily and interpret its results. The construction of a Generalized
Linear (Mixed) Model has to be well thought out and may need corrections a posteriori by the use of common tests
that helps evaluate the quality of your GL(M)M (found in the rating file).</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(plausible !== undefined){
                            plausible('TutorialComplete', {callback: tutorial_finish_finish, props: {path: document.location.pathname}})
                          } else {
                            tutorial_finish_finish();
                          }
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Pre-process abundance data</p>
</li>
                        
                        <li><p>Compute Community and Population metrics</p>
</li>
                        
                        <li><p>Construct a GL(M)M with common biodiversity metrics</p>
</li>
                        
                        <li><p>Interpret (and correct if needed) GL(M)M results</p>
</li>
                        
                        <li><p>Learn about and interpret common tests to evaluate the quality of your GL(M)M</p>
</li>
                        
                        <li><p>Use an Essential Biodiversity Variables workflow</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/ecology/faqs/">FAQ page for the Ecology topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/ecology#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="EVHOE">MAHE Jean-Claude, 1987 <b>EVHOE EVALUATION HALIEUTIQUE OUEST DE L’EUROPE</b>. <a href="https://doi.org/10.18142/8">10.18142/8</a> <a href="https://campagnes.flotteoceanographique.fr/series/8/">https://campagnes.flotteoceanographique.fr/series/8/</a></li>
<li id="BITS"><b>Fish trawl survey: ICES Baltic International Trawl Survey for commercial fish species. ICES Database of trawl surveys (DATRAS)</b>, 2010. <a href="http://ecosystemdata.ices.dk">http://ecosystemdata.ices.dk</a></li>
<li id="SWCIBTS"><b>DATRAS Scottish West Coast Bottom Trawl Survey (SWC-IBTS)</b>, 2013. <a href="https://gis.ices.dk/geonetwork/srv/api/records/a38fb537-0ec4-40bb-a3de-750b5dd3829a">https://gis.ices.dk/geonetwork/srv/api/records/a38fb537-0ec4-40bb-a3de-750b5dd3829a</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=ecology/PAMPA-toolsuite-tutorial"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=ecology/PAMPA-toolsuite-tutorial">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Coline Royaux, Yvan Le Bras, Dominique Pelletier, Jean-Baptiste Mihoub,  <b>Compute and analyze biodiversity metrics with PAMPA toolsuite (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html">https://training.galaxyproject.org/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{ecology-PAMPA-toolsuite-tutorial,
author = "Coline Royaux and Yvan Le Bras and Dominique Pelletier and Jean-Baptiste Mihoub",
	title = "Compute and analyze biodiversity metrics with PAMPA toolsuite (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                
                
<div class="d-flex flex-wrap">

	<a href="/training-material/hall-of-fame/gallantries/" class="funder-badge">
		
		<div class="avatar">
			
			<img class="funder-avatar" src="https://gallantries.github.io/assets/images/gallantries-logo.png" alt="Logo">
			
		</div>

		<div class="info">
			<div class="name">Gallantries</div>
			<div class="description">
			This project (2020-1-NL01-KA203-064717) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.



            
			</div>
		</div>
	</a>

	<a href="/training-material/hall-of-fame/pndb/" class="funder-badge">
		
		<div class="avatar">
			
			<img class="funder-avatar" src="/training-material/shared/images/PNDB_sub.png" alt="Logo">
			
		</div>

		<div class="info">
			<div class="name">Pôle National de Données de Biodiversité</div>
			<div class="description">
			

            
			</div>
		</div>
	</a>

</div>


                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: ddf54b12c295
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: merge_cols
  owner: devteam
  revisions: dd40b1e9eebe
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pampa_communitymetrics
  owner: ecology
  revisions: c67ece97d4b1
  tool_panel_section_label: Species abundance
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pampa_glmcomm
  owner: ecology
  revisions: b40fecef616b
  tool_panel_section_label: Species abundance
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pampa_glmsp
  owner: ecology
  revisions: 7a69e1c8ba46
  tool_panel_section_label: Species abundance
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pampa_plotglm
  owner: ecology
  revisions: c448e026f122
  tool_panel_section_label: Species abundance
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pampa_presabs
  owner: ecology
  revisions: 1ebb70b59037
  tool_panel_section_label: Species abundance
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: regex_find_replace
  owner: galaxyp
  revisions: 209b7c5ee9d7
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: regex_find_replace
  owner: galaxyp
  revisions: 209b7c5ee9d7
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: regex_find_replace
  owner: galaxyp
  revisions: ae8c4b2488e7
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: datamash_transpose
  owner: iuc
  revisions: 22c2a1ac7ae3
  tool_panel_section_label: Join, Subtract and Group
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">2</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>January 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Very clear
					<b>Disliked</b>: Where to find the tools (not always easy in the toolbar)
				</li>
				
		    </ul>
		    
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00125">gxy.io/GTN:T00125</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/649e2e9bca72de907031a6b13a02758bbb8b427b">649e2e9</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.20cf1f17.js'></script>

	
	

    </body>
</html>