<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Hands-on: Biomarker candidate identification / Proteomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@6d58dcda159f1edca14f9076e5437988d499318d",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.f1f2de89.js?v=1705882549' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.03aefefa.js?v=1705882549' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Hands-on: Hands-on: Biomarker candidate identification">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-03 14:30:27 +0000">
<meta name="DC.creator" content="Florence Combes">
<meta name="DC.creator" content="David Christiany">
<meta name="DC.creator" content="Valentin Loux">
<meta name="DC.creator" content="Yves Vandenbrouck"><meta name="description" content="Training material for proteomics workflows in Galaxy">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Proteomics / Hands-on: Biomarker candidate identification">
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "dateModified": "2023-11-03 14:30:27 +0000",
  "datePublished": "2019-06-13 19:56:33 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Hands-on: Hands-on: Biomarker candidate identification",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "identifier": "https://gxy.io/GTN:T00213",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on: Hands-on: Biomarker candidate identification",
  "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html",
  "version": 30,
  "timeRequired": "PT3H",
  "teaches": "Build a workflow implementing a strategy for the selection of tissue-leakage biomarkers using ProteoRE",
  "keywords": [
    "Proteomics",
    "DDA",
    "human"
  ],
  "description": "The questions this  addresses are:\n - How to mine public databases to retrieve info?\n - How to build a selection strategy by applying successive biochemical/cellular criteria to a list of gene/protein?\n - How to select biomarkers candidates using experimental information (transcriptomics & proteomics) and annotation from public databases?\n\n\nThe objectives are:\n - Build a workflow implementing a strategy for the selection of tissue-leakage biomarkers using ProteoRE\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/combesf/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/combesf/",
      "name": "Florence Combes",
      "image": "https://avatars.githubusercontent.com/combesf",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/davidchristiany/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/davidchristiany/",
      "name": "David Christiany",
      "image": "https://avatars.githubusercontent.com/davidchristiany",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/vloux/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/vloux/",
      "name": "Valentin Loux",
      "image": "https://avatars.githubusercontent.com/vloux",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/yvandenb/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/yvandenb/",
      "name": "Yves Vandenbrouck",
      "image": "https://avatars.githubusercontent.com/yvandenb",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "DDA"
    },
    {
      "@type": "Thing",
      "name": "human"
    }
  ],
  "abstract": "A biomarker is a measurable biological component that can be routinely detected in clinical practice and reflects a disease state,"
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.f1f2de89.js?v=1705882549'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Proteomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/proteomics/faqs/" title="Check our FAQs for the Proteomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/biomarker_selection/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/smorgasbord2023/topics/proteomics/tutorials/biomarker_selection/tutorial.html">smorgasbord2023</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2024-01-01/topics/proteomics/tutorials/biomarker_selection/tutorial.html">2024-01-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-12-01/topics/proteomics/tutorials/biomarker_selection/tutorial.html">2023-12-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-proteomics">
    <h1 data-toc-skip>Biomarker candidate identification</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/combesf/" class="contributor-badge contributor-combesf"><img src="https://avatars.githubusercontent.com/combesf?s=36" alt="Florence Combes avatar" width="36" class="avatar" />
    Florence Combes</a><a href="/training-material/hall-of-fame/davidchristiany/" class="contributor-badge contributor-davidchristiany"><img src="https://avatars.githubusercontent.com/davidchristiany?s=36" alt="David Christiany avatar" width="36" class="avatar" />
    David Christiany</a><a href="/training-material/hall-of-fame/vloux/" class="contributor-badge contributor-vloux"><img src="https://avatars.githubusercontent.com/vloux?s=36" alt="Valentin Loux avatar" width="36" class="avatar" />
    Valentin Loux</a><a href="/training-material/hall-of-fame/yvandenb/" class="contributor-badge contributor-yvandenb"><img src="https://avatars.githubusercontent.com/yvandenb?s=36" alt="Yves Vandenbrouck avatar" width="36" class="avatar" />
    Yves Vandenbrouck</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to mine public databases to retrieve info?</p>
</li>
        
        <li><p>How to build a selection strategy by applying successive biochemical/cellular criteria to a list of gene/protein?</p>
</li>
        
        <li><p>How to select biomarkers candidates using experimental information (transcriptomics &amp; proteomics) and annotation from public databases?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Build a workflow implementing a strategy for the selection of tissue-leakage biomarkers using ProteoRE</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/proteomics/tutorials/biomarker_selection/workflows/" title="Biomarker candidate identification workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Jun 13, 2019 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 3, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00213">https://gxy.io/GTN:T00213</a> </div>
        

	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 30 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>A biomarker is a measurable biological component that can be routinely detected in clinical practice and reflects a disease state,
response to therapeutic treatment, or other relevant biological state.</p>

<p><a href="http://www.proteore.org">ProteoRE Galaxy instance</a> provides necessary tools to execute a complete biomarkers selection pipeline.
In this tutorial we introduce successively the tools of this pipeline, and guide you to execute them in order to complete the entire
pipeline on a concrete example. This strategy is described by <span class="citation"><a href="#Nguyen2019">Nguyen <i>et al.</i> 2019</a></span>.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#global-view-of-the-strategy" id="markdown-toc-global-view-of-the-strategy">Global view of the strategy</a></li>
  <li><a href="#selection-of-tissue-specific-proteins" id="markdown-toc-selection-of-tissue-specific-proteins">Selection of tissue-specific proteins</a></li>
  <li><a href="#annotation-with-biochemical-and-cellular-features" id="markdown-toc-annotation-with-biochemical-and-cellular-features">Annotation with biochemical and cellular features</a></li>
  <li><a href="#check-for-previous-detection-by-lc-msms-experiments" id="markdown-toc-check-for-previous-detection-by-lc-msms-experiments">Check for previous detection by LC-MS/MS experiments</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="global-view-of-the-strategy">Global view of the strategy</h1>

<p>For this tutorial, no input data are required as the first steps will be to select data from
public databases with ProteoRE tools.</p>

<p>The strategy consists in selecting, one step after another, the most interesting candidates biomarkers.
Our use-case here is to identify candidate biomarkers for myocardial infarction tissue-leakage.</p>

<p>Criteria candidate biomarkers have to fulfill through this pipeline are:</p>
<ul>
  <li>heart-specificity</li>
  <li>cytoplasmic localization</li>
  <li>detection in LC-MS/MS experiments already done</li>
</ul>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/pipeline.png" alt="Overview of the pipeline described in this tutorial. " width="920" height="745" loading="lazy" /><a target="_blank" href="../../images/pipeline.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Pipeline of the tutorial, from Nguyen et al., as of in 2019.</figcaption></figure>

<!--SNIPPET-->
<blockquote class="comment">   <div class="box-title comment-title" id="comment-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Results may vary</div>   <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, external databases, or because of stochastic processes in the algorithms.</p> </blockquote>
<p><!--END_SNIPPET--></p>

<h1 id="selection-of-tissue-specific-proteins">Selection of tissue-specific proteins</h1>

<p>We will start by selecting proteins associated with heart muscle tissue based on experimental data available from
<a href="https://www.proteinatlas.org/">The Human Protein Atlas</a> (HPA) (Publication: <span class="citation"><a href="#Uhln2005">Uhlén <i>et al.</i> 2005</a></span>)</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-build-tissue-specific-expression-dataset-based-on-immunohistochemistry"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Build tissue-specific expression dataset based on ImmunoHistoChemistry</div>

  <ol>
    <li>
      <p><strong>Create a new history</strong> and give it a name.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><strong>Build tissue-specific expression dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
      <ul>
        <li><em>“Experimental data source (antibody- or RNAseq-based)”</em>: <code class="language-plaintext highlighter-rouge">Expression profiles based on immunohistochemistry</code></li>
        <li><em>“Select tissue”</em>: <code class="language-plaintext highlighter-rouge">Heart muscle</code></li>
        <li><em>“Expression level”</em>: <code class="language-plaintext highlighter-rouge">High</code> and <code class="language-plaintext highlighter-rouge">Medium</code></li>
        <li><em>“Reliability score”</em>: <code class="language-plaintext highlighter-rouge">Enhanced</code> and <code class="language-plaintext highlighter-rouge">Supported</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output">Output</h3>
    <ul>
      <li><strong>Tissue-specific expression from IHC</strong> (1596 lines): List of the selected proteins.
6 columns: ‘Gene’, ‘Gene name’ and the retrieved info from HPA.</li>
    </ul>
  </blockquote>
</blockquote>

<p>We will now rerun the same tool but to select transcripts according to their expression profile.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-build-tissue-specific-expression-dataset-based-on-rnaseq"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Build tissue-specific expression dataset based on RNAseq</div>

  <ol>
    <li><strong>Build tissue-specific expression dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Experimental data source (antibody- or RNAseq-based)”</em>: <code class="language-plaintext highlighter-rouge">RNA levels based on RNA-seq experiments</code></li>
        <li><em>“Select tissue”</em>: <code class="language-plaintext highlighter-rouge">Heart muscle</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-1">Output</h3>
    <ul>
      <li><strong>Tissue-specific expression from RNAseq</strong> (19613 lines): List of the selected transcripts.
4 columns: ‘Gene’, ‘Gene name’ and the retrieved info from HPA.</li>
    </ul>
  </blockquote>
</blockquote>

<p>This second list must be reduced by removing transcripts that are not highly enriched in heart muscle.
To do so, a filter is applied on the expression value provided by HPA and measured in TPM (last column of the output file).
In ProteoRE we’ll use the “Filter by keywords and/or numerical value” tool.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-on-expression-value-criterion"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter on expression value criterion</div>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Input File”</em>: <code class="language-plaintext highlighter-rouge">Tissue-specific expression from RNASeq</code> from the previous step</li>
        <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
        <li>In <em>“Filter by numerical value”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter by numerical value”</em>
              <ul>
                <li><em>“Column number on which to apply the filter”</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
                <li><em>“Select operator”</em>: <code class="language-plaintext highlighter-rouge">&lt;=</code></li>
                <li><em>“Value”</em>: <code class="language-plaintext highlighter-rouge">10.0</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Sort by column ?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Sort result files by”</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>How many lines are there in the file of heart transcripts
with a TPM value &gt;10 ?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>5257 lines.</p>
    </blockquote>
  </blockquote>

</blockquote>

<p>We have now 2 datasets of heart-muscle proteins/transcripts, based on IHC (Immonohistochemistry) data or <a href="https://statquest.org/2015/07/09/rpkm-fpkm-and-tpm-clearly-explained/">TPM</a> (Expression level expressed in Transcripts Per Million) value.</p>

<p>We want now to select candidate biomarkers that are expressed in the heart muscle according to <strong>both</strong> IHC and RNA-seq data, using the Jvenn tool.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-venn-diagram"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Venn diagram</div>

  <ol>
    <li><strong>Venn diagram</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>“List to compare”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert List to compare”</em>
              <ul>
                <li><em>“Enter your list”</em>: <code class="language-plaintext highlighter-rouge">Input file containing your list</code>
                  <ul>
                    <li><em>“Select your file”</em>: <code class="language-plaintext highlighter-rouge">Tissue-specific expression from IHC</code></li>
                    <li><em>“Enter the name of this list”</em>: <code class="language-plaintext highlighter-rouge">heart IHC</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert List to compare”</em>
              <ul>
                <li><em>“Enter your list”</em>: <code class="language-plaintext highlighter-rouge">Input file containing your list</code>
                  <ul>
                    <li><em>“Select your file”</em>: <code class="language-plaintext highlighter-rouge">Filtered tissue-specific expression from RNAseq</code></li>
                    <li><em>“Enter the name of this list”</em>: <code class="language-plaintext highlighter-rouge">heart RNAseq</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Examine <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> the Venn diagram output:</p>

      <p><a href="../../images/biomarker-selection/jVenn_chart-tuto2.png" rel="noopener noreferrer"><img src="../../images/biomarker-selection/jVenn_chart-tuto2.png" alt="Venn diagram output. " width="500" height="695" loading="lazy" /></a></p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>How many IDs are common to both IHC and RNA-seq data?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>931 candidate biomarkers are detected by both IHC and RNA-seq data.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>For greater clarity we’ll keep only the column with those 931 candidate biomarkers detected in both IHC and RNA-seq data in the
rest of our pipeline.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-obtain-list-of-candidate-biomarkers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Obtain list of candidate biomarkers</div>

  <ol>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“File to cut”</em>: <code class="language-plaintext highlighter-rouge">Venn diagram text output</code></li>
        <li><em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code>
          <ul>
            <li><em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 3</code></li>
          </ul>
        </li>
      </ul>

      <p>Now we’ll filter this dataset not to keep the ‘NA’ lines.</p>
    </li>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Input file”</em>: output from <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
        <li>In <em>“Filter by keywords”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter by keywords”</em>
              <ul>
                <li><em>“Search for exact match?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                <li><em>“Enter keywords”</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>“Copy/paste keywords to find (keep or discard)”</em>: <code class="language-plaintext highlighter-rouge">NA</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Sort by column ?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Sort result files by”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Let’s rename the 931 IDs dataset in “heart931” for simplification.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>The pipeline will then continue based on these 931 candidates, from which we have to select biomarkers that are
highly specific to the heart using additional expression data (still from HPA).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-add-expression-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Add expression data</div>

  <ol>
    <li><strong>Add expression data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Enter your IDs (Ensembl gene IDs only, e.g. ENSG00000064787)”</em>: <code class="language-plaintext highlighter-rouge">Input file containing your IDs</code>
          <ul>
            <li><em>“Select your file”</em>: <code class="language-plaintext highlighter-rouge">Heart931</code> output we created in the previous step</li>
          </ul>
        </li>
        <li>In <em>“RNAseq/Ab-based expression data”</em>:
          <ul>
            <li><em>“Select information to add to your list”</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Gene name</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Gene description</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">RNA tissue category</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">RNA tissue specificity abundance in Transcript Per Million</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We wish to focus on transcripts that have been classed as (according to the HPA definition):</p>
<ul>
  <li><strong>Tissue enriched</strong> (expression in one tissue at least fivefold higher than all other tissues),</li>
  <li><strong>Group enriched</strong> (fivefold higher average TPM in a group of two to to seven tissues compared to all other tissues) and</li>
  <li><strong>Tissue enhanced</strong> (fivefold higher average TPM in one or more tissues/cell lines compared to the mean TPM for all tissues)</li>
</ul>

<p>This information is listed in the column 4 : <em>“RNA tissue category”</em> of the result dataset.</p>

<p>Let’s use <strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to select the candidate biomarkers based on this
<em>“RNA tissue category”</em> criterion.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-by-keywords-and-or-numerical-value"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter by keywords and/or numerical value</div>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Input file”</em>: output from <strong>Add expression data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>“Filter by keywords”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter by keywords”</em>
              <ul>
                <li><em>“Column number on which to apply the filter”</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
                <li><em>“Enter keywords”</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>“Copy/paste keywords to find (keep or discard)”</em>: <code class="language-plaintext highlighter-rouge">enriched enhanced</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Sort by column ?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Sort result files by”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="output-2">Output</h3>
        <ul>
          <li>
            <p><strong>Filtered Add_expression_data</strong>: output list of the heart biomarkers with RNA tissue category
containing “enriched” or “enhanced” (115 lines = what we are interested in)</p>
          </li>
          <li>
            <p><strong>Filtered Add_expression_data - discarded lines</strong>: output list of the heart biomarkers with
RNA tissue category NOT containing “enriched” or “enhanced” (not what we are interested in)</p>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Let’s rename the filtered dataset <strong>heart115</strong></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>We now have identified 115 candidates considered to have significantly higher expression in heart muscle according to HPA criteria.</p>

<h1 id="annotation-with-biochemical-and-cellular-features">Annotation with biochemical and cellular features</h1>

<p>Candidate biomarkers we want to identify have to be cytoplasmic and without transmembrane domains (TMD).
Thus we will retrieve protein features from neXtProt to retrieve those informations.</p>

<p>Since HPA only considers <a href="https://www.ensembl.org">Ensemble</a> <code class="language-plaintext highlighter-rouge">ENSG</code> identifiers (related to the gene), although <a href="https://www.nextprot.org/">neXtProt</a> uses <a href="https://www.uniprot.org/">UniProt</a> identifiers (related to proteins), first thing to do is to map the Ensembl identifiers contained our list of (115)
candidates to their corresponding UniProt accession number. The tool <strong>ID Converter</strong> is what we need to do so.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-id-converter"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: ID Converter</div>

  <ol>
    <li><strong>ID Converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Enter IDs”</em>: <code class="language-plaintext highlighter-rouge">Input file containing IDs</code></li>
        <li><em>“Select your file”</em>: <code class="language-plaintext highlighter-rouge">heart115</code></li>
        <li><em>“Species”</em>: <code class="language-plaintext highlighter-rouge">Human (Homo sapiens)</code>
          <ul>
            <li><em>“Type/source of IDs”</em>: <code class="language-plaintext highlighter-rouge">Ensembl gene ID (e.g. ENSG00000166913)</code></li>
            <li>In <em>“Target type”</em>:
              <ul>
                <li><em>“Target type of IDs you would like to map to”</em>:
                  <ul>
                    <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Uniprot accession number</code></li>
                    <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Uniprot ID</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-3">Output</h3>
    <p><strong>ID converter</strong>: In this dataset, 2 columns (columns 6 and 7, at the end) which contain
UniProt accession number and ID are added.
Note: We used here the release “Homo sapiens 08/05/2019”; results can be slightly different
according to the version number of Uniprot.</p>
  </blockquote>
</blockquote>

<p>We have now UniProt IDs for the 115 candidate biomarkers: we are able to collect protein features from <a href="https://www.nextprot.org/">neXtProt</a>.
For this purpose, we use the <strong>Add protein features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> ProteoRE tool.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-add-protein-features"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Add protein features</div>

  <ol>
    <li><strong>Add protein features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Enter your IDs (neXtProt or UniProt)”</em>: <code class="language-plaintext highlighter-rouge">Input file containing your IDs </code>
          <ul>
            <li><em>“Select your file”</em>: the output from <strong>ID Converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><em>“Column IDs (e.g : Enter c1 for column n°1)”</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
          </ul>
        </li>
        <li>In <em>“Select features”</em>:
          <ul>
            <li><em>“Physico-Chemical Features”</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Number of transmembrane domains</code></li>
              </ul>
            </li>
            <li><em>“Localization”</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Subcellular Location</code></li>
              </ul>
            </li>
            <li><em>“Disease information”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-4">Output</h3>
    <p><strong>Add information from NextProt</strong>: In this file (115 lines), 3 columns (columns 8, 9 and 10)
were added (at the end). These columns present TMDomains, Subcell Location and Diseases info.</p>
  </blockquote>
</blockquote>

<p>With this dataset, we can select proteins reported as localized in the cytoplasm and having
no transmembrane domains by running the Filter by keywords and/or numerical value tool.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-by-keywords-and-or-numerical-value-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter by keywords and/or numerical value</div>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Input file”</em>: the output from <strong>Add protein features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Select an operator to combine your filters (if more than one)”</em>: <code class="language-plaintext highlighter-rouge">AND</code></li>
        <li>In <em>“Filter by keywords”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter by keywords”</em>
              <ul>
                <li><em>“Column number on which to apply the filter”</em>: <code class="language-plaintext highlighter-rouge">c9</code></li>
                <li><em>“Enter keywords”</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>“Copy/paste keywords to find (keep or discard)”</em>: <code class="language-plaintext highlighter-rouge">cytoplasm cytosol</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Filter by numerical value”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter by numerical value”</em>
              <ul>
                <li><em>“Column number on which to apply the filter”</em>: <code class="language-plaintext highlighter-rouge">c8</code></li>
                <li><em>“Value”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Sort by column ?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Sort result files by:”</em>: <code class="language-plaintext highlighter-rouge">c5</code></li>
            <li><em>“Sort in descending order ?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-5">Output</h3>
    <ul>
      <li><strong>Filtered Add_information_from_neXtProt</strong>: output list of the proteins having a cytoplasmic
location and no TMD (48 proteins)</li>
      <li><strong>Filtered Add_information_from_neXtProt - discarded lines</strong>: output list of the proteins NOT
cytoplasmic and having at least 1 TMD.</li>
    </ul>
  </blockquote>
</blockquote>

<p>We have now 45 proteins. The next step is to identify those proteins already previously observed in
LS MS/MS experiments.</p>

<h1 id="check-for-previous-detection-by-lc-msms-experiments">Check for previous detection by LC-MS/MS experiments</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-ms-ms-observations-in-tissue-fluid"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get MS/MS observations in tissue/fluid</div>

  <ol>
    <li><strong>Get MS/MS observations in tissue/fluid</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Enter your IDs (UniProt Accession number only)”</em>: <code class="language-plaintext highlighter-rouge">Input file containing your IDs </code>
          <ul>
            <li><em>“Select your file”</em>: the filtered output from the previous step</li>
            <li><em>“Column of IDs”</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
          </ul>
        </li>
        <li><em>“Proteomics dataset (biological sample)”</em>:
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Human Heart</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Human Plasma non glyco</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-6">Output</h3>
    <p><strong>Get MS/MS observations in tissue/fluid</strong>: In this file, 2 columns (11 and 12, at the end)
were added with the info of number of times peptides were seen by MS/MS.</p>
  </blockquote>
</blockquote>

<p>Let’s now keep only proteins that have already been seen by MS/MS in the plasma (last column of the file).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-for-proteins-seen-in-the-plasma"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter for proteins seen in the plasma</div>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Input file”</em>: the MA/MS observations output from the previous step</li>
        <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
        <li>In <em>“Filter by keywords”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Filter by keywords”</em>
              <ul>
                <li><em>“Column number on which to apply the filter”</em>: <code class="language-plaintext highlighter-rouge">c12</code></li>
                <li><em>“Search for exact match?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                <li><em>“Enter keywords”</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>“Copy/paste keywords to find (keep or discard)”</em>: <code class="language-plaintext highlighter-rouge">NA</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Sort by column ?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Sort results file by”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-7">Output</h3>
    <ul>
      <li><strong>Filtered Get MS/MS observations in tissue/fluid</strong>: output list of the proteins
whose some peptides have been seen in plasma (21 proteins)</li>
      <li><strong>Filtered Get MS/MS observations in tissue/fluid - discarded lines</strong>:
output list of proteins with no peptides seen in the plasma</li>
    </ul>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>At the end of the process we end up with a list of 22 biomarkers that are <strong>highly enriched in heart muscle</strong>, localized
<strong>in the cytosol</strong> and <strong>detectable by MS in the plasma</strong>.
<em>NB: Please be aware that, due to databases update, number of biomarkers you end up with can be different from 21 that we obtain here.</em></p>

<p>Briefly and from a biological point of view, 3 of these proteins exhibit a relative low detection level in the plasma
compared to heart muscle tissue, and are reported with a very high heart-muscle-specific RNA abundance.
These potential mechanistic biomarkers of myocardial infarction include (i) cardiac Troponin I type 3 (TNNI3 (P19429))
that is routinely used as the most specific marker of myocardial injury and (ii) the heart-type fatty acid-binding protein
(FABP3 (P05413)) that has been proposed as a diagnostic and prognostic marker for acute and chronic cardiac injury.</p>

<p>Extraction of a workflow from your history is valuable for analyses reproducibility and traceability.
Moreover, a workflow can also be made reusable with modifiable parameters, and as a result this strategy can
be applied to other types of tissue injury (e.g. brain, liver, kidney).</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Public proteomics databases are useful resources for annotation of biomarker candidates</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/proteomics/faqs/">FAQ page for the Proteomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Uhln2005">Uhlén, M., E. Björling, C. Agaton, C. A.-K. Szigyarto, B. Amini <i>et al.</i>, 2005 <b>A Human Protein Atlas for Normal and Cancer Tissues Based on Antibody Proteomics</b>. Molecular &amp; Cellular Proteomics 4: 1920–1932. <a href="https://doi.org/10.1074/mcp.m500279-mcp200">10.1074/mcp.m500279-mcp200</a></li>
<li id="Nguyen2019">Nguyen, L., V. Brun, F. Combes, V. Loux, and Y. Vandenbrouck, 2019 <b>Designing an In Silico Strategy to Select Tissue-Leakage Biomarkers Using the Galaxy Framework</b>, pp. 275–289 <span style="font-style: normal">in</span> <i>Methods in Molecular Biology</i>, <i>Springer New York</i>. <a href="https://doi.org/10.1007/978-1-4939-9164-8_18">10.1007/978-1-4939-9164-8_18</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Biomarker candidate identification (Proteomics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Florence Combes, David Christiany, Valentin Loux, Yves Vandenbrouck,  <b>Biomarker candidate identification (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{proteomics-biomarker_selection,
author = "Florence Combes and David Christiany and Valentin Loux and Yves Vandenbrouck",
	title = "Biomarker candidate identification (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/proteomics/tutorials/biomarker_selection/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: 288462ec2630
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_expression_rnaseq_abbased
  owner: proteore
  revisions: dbeabf9bf091
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_filter_keywords_values
  owner: proteore
  revisions: 33ca9ba2495a
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_id_converter
  owner: proteore
  revisions: b6607b7e683f
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_ms_observation_pepatlas
  owner: proteore
  revisions: 299eb4a62e0c
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_prot_features
  owner: proteore
  revisions: 8df559ad14a7
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_tissue_specific_expression_data
  owner: proteore
  revisions: 3e65e0249976
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: proteore_venn_diagram
  owner: proteore
  revisions: d1fd04dcb13a
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/6d58dcda159f1edca14f9076e5437988d499318d">6d58dcd</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00213">https://gxy.io/GTN:T00213</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/biomarker_selection/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


        <script src='/training-material/assets/js/bundle.03aefefa.js?v=1705882549'></script>
        
    </body>
</html>