<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Biomarker candidate identification</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Biomarker candidate identification" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fbiomarker_selection%2Ftutorial.html&edit-text=&act=url" title="">
                                FranÃ§ais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fbiomarker_selection%2Ftutorial.html&edit-text=&act=url" title="">
                                æ—¥æœ¬èª
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fbiomarker_selection%2Ftutorial.html&edit-text=&act=url" title="">
                                EspaÃ±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fbiomarker_selection%2Ftutorial.html&edit-text=&act=url" title="">
                                PortuguÃªs
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fbiomarker_selection%2Ftutorial.html&edit-text=&act=url" title="">
                                Ø§Ù„Ø¹Ø±Ø¨ÙŠØ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fbiomarker_selection%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> ğŸƒ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/biomarker_selection/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Biomarker candidate identification",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
  },
  "courseCode": "proteomics / biomarker_selection / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Biomarker candidate identification' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html",
  "timeRequired": "PT3H",
  "description": "The questions this  addresses are:\n - How to mine public databases to retrieve info?\n - How to build a selection strategy by applying successive biochemical/cellular criteria to a list of gene/protein?\n - How to select biomarkers candidates using experimental information (transcriptomics & proteomics) and annotation from public databases?\n\n\\nThe objectives are:\n - Build a workflow implementing a strategy for the selection of tissue-leakage biomarkers using ProteoRE\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Florence Combes"
    },
    {
      "@type": "Person",
      "name": "David Christiany"
    },
    {
      "@type": "Person",
      "name": "Valentin Loux"
    },
    {
      "@type": "Person",
      "name": "Yves Vandenbrouck"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Florence Combes"
    },
    {
      "@type": "Person",
      "name": "David Christiany"
    },
    {
      "@type": "Person",
      "name": "Valentin Loux"
    },
    {
      "@type": "Person",
      "name": "Yves Vandenbrouck"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Biomarker candidate identification</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/combesf/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/combesf" alt="Florence Combes">Florence Combes</a>, <a href="/training-material/hall-of-fame/davidchristiany/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/davidchristiany" alt="David Christiany">David Christiany</a>, <a href="/training-material/hall-of-fame/vloux/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/vloux" alt="Valentin Loux">Valentin Loux</a>, <a href="/training-material/hall-of-fame/yvandenb/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/yvandenb" alt="Yves Vandenbrouck">Yves Vandenbrouck</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to mine public databases to retrieve info?</p>
</li>
            
            <li><p>How to build a selection strategy by applying successive biochemical/cellular criteria to a list of gene/protein?</p>
</li>
            
            <li><p>How to select biomarkers candidates using experimental information (transcriptomics &amp; proteomics) and annotation from public databases?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Build a workflow implementing a strategy for the selection of tissue-leakage biomarkers using ProteoRE</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                

                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/biomarker_selection/workflows/" title="Workflows" alt="Biomarker candidate identification workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="http://www.proteore.org/" title="ProteoRE">
            ProteoRE
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>A biomarker is a measurable biological component that can be routinely detected in clinical practice and reflects a disease state,
response to therapeutic treatment, or other relevant biological state.</p>

<p><a href="http://www.proteore.org">ProteoRE <span class="notranslate">Galaxy</span> instance</a> provides necessary tools to execute a complete biomarkers selection pipeline.
In this tutorial we introduce successively the tools of this pipeline, and guide you to execute them in order to complete the entire
pipeline on a concrete example. This strategy is described by <a class="citation" href="#Nguyen2019">Nguyen <i>et al.</i> 2019</a>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#selection-of-tissue-specific-proteins" id="markdown-toc-selection-of-tissue-specific-proteins">Selection of tissue-specific proteins</a></li>
  <li><a href="#annotation-with-biochemical-and-cellular-features" id="markdown-toc-annotation-with-biochemical-and-cellular-features">Annotation with biochemical and cellular features</a></li>
  <li><a href="#check-for-previous-detection-by-lc-msms-experiments" id="markdown-toc-check-for-previous-detection-by-lc-msms-experiments">Check for previous detection by LC-MS/MS experiments</a></li>
</ol>

</blockquote>

<h1 class="no_toc" id="global-view-of-the-strategy">Global view of the strategy</h1>

<p>For this tutorial, no input data are required as the first steps will be to select data from
public databases with ProteoRE tools.</p>

<p>The strategy consists in selecting, one step after another, the most interesting candidates biomarkers.
Our use-case here is to identify candidate biomarkers for myocardial infarction tissue-leakage.</p>

<p>Criteria candidate biomarkers have to fulfill through this pipeline are:</p>
<ul>
  <li>heart-specificity</li>
  <li>cytoplasmic localization</li>
  <li>detection in LC-MS/MS experiments already done</li>
</ul>

<figure id="figure-1"><img src="../../images/pipeline.png" alt="Overview of the pipeline described in this tutorial" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Pipeline of the tutorial, from Nguyen et al., as of in 2019.</figcaption></figure>

<blockquote class="comment">
  <h3 id="comment-note-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: results may vary</h3>

  <p>Your results may be slightly different from the ones presented in this tutorial
due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or
because of stochastic processes in the algorithms.</p>

</blockquote>

<h1 id="selection-of-tissue-specific-proteins">Selection of tissue-specific proteins</h1>

<p>We will start by selecting proteins associated with heart muscle tissue based on experimental data available from
<a href="https://www.proteinatlas.org/">The Human Protein Atlas</a> (HPA) (Publication: <a class="citation" href="#Uhln2005">UhlÃ©n <i>et al.</i> 2005</a>)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-build-tissue-specific-expression-dataset-based-on-immunohistochemistry"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Build tissue-specific <span class="notranslate">expression</span> dataset based on ImmunoHistoChemistry</h3>

  <ol>
    <li><strong>Create a new history</strong> and give it a name.
      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li><strong>Build tissue-specific <span class="notranslate">expression</span> dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œExperimental data source (antibody- or <span class="notranslate">RNA</span>seq-based)â€</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Expression</span> profiles based on immunohistochemistry</code></li>
        <li><em>â€œSelect tissueâ€</em>: <code class="language-plaintext highlighter-rouge">Heart muscle</code></li>
        <li><em>â€œ<span class="notranslate">Expression</span> levelâ€</em>: <code class="language-plaintext highlighter-rouge">High</code> and <code class="language-plaintext highlighter-rouge">Medium</code></li>
        <li><em>â€œReliability scoreâ€</em>: <code class="language-plaintext highlighter-rouge">Enhanced</code> and <code class="language-plaintext highlighter-rouge">Supported</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output">Output</h3>
    <ul>
      <li><strong>Tissue-specific <span class="notranslate">expression</span> from IHC</strong> (1596 lines): List of the selected proteins.
6 columns: â€˜Geneâ€™, â€˜Gene nameâ€™ and the retrieved info from HPA.</li>
    </ul>
  </blockquote>
</blockquote>

<p>We will now rerun the same tool but to select transcripts according to their <span class="notranslate">expression</span> profile.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-build-tissue-specific-expression-dataset-based-on-rnaseq"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Build tissue-specific <span class="notranslate">expression</span> dataset based on <span class="notranslate">RNA</span>seq</h3>

  <ol>
    <li><strong>Build tissue-specific <span class="notranslate">expression</span> dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œExperimental data source (antibody- or <span class="notranslate">RNA</span>seq-based)â€</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span> levels based on <span class="notranslate">RNA</span>-seq experiments</code></li>
        <li><em>â€œSelect tissueâ€</em>: <code class="language-plaintext highlighter-rouge">Heart muscle</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-1">Output</h3>
    <ul>
      <li><strong>Tissue-specific <span class="notranslate">expression</span> from <span class="notranslate">RNA</span>seq</strong> (19613 lines): List of the selected transcripts.
4 columns: â€˜Geneâ€™, â€˜Gene nameâ€™ and the retrieved info from HPA.</li>
    </ul>
  </blockquote>
</blockquote>

<p>This second list must be reduced by removing transcripts that are not highly enriched in heart muscle.
To do so, a filter is applied on the <span class="notranslate">expression</span> value provided by HPA and measured in TPM (last column of the output file).
In ProteoRE weâ€™ll use the â€œFilter by keywords and/or numerical valueâ€ tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-on-expression-value-criterion"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter on <span class="notranslate">expression</span> value criterion</h3>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œInput Fileâ€</em>: <code class="language-plaintext highlighter-rouge">Tissue-specific <span class="notranslate">expression</span> from <span class="notranslate">RNA</span>Seq</code> from the previous step</li>
        <li><em>â€œOperationâ€</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
        <li>In <em>â€œFilter by numerical valueâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert Filter by numerical valueâ€</em>
              <ul>
                <li><em>â€œColumn number on which to apply the filterâ€</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
                <li><em>â€œSelect operatorâ€</em>: <code class="language-plaintext highlighter-rouge">&lt;=</code></li>
                <li><em>â€œValueâ€</em>: <code class="language-plaintext highlighter-rouge">10.0</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>â€œSort by column ?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>â€œSort result files byâ€</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>How many lines are there in the file of heart transcripts
with a TPM value &gt;10 ?</p>

    <blockquote class="solution">
      <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <p>5257 lines.</p>
    </blockquote>
  </blockquote>

</blockquote>

<p>We have now 2 datasets of heart-muscle proteins/transcripts, based on IHC (Immonohistochemistry) data or <a href="https://statquest.org/2015/07/09/rpkm-fpkm-and-tpm-clearly-explained/">TPM</a> (<span class="notranslate">Expression</span> level expressed in Transcripts Per Million) value.</p>

<p>We want now to select candidate biomarkers that are expressed in the heart muscle according to <strong>both</strong> IHC and <span class="notranslate">RNA</span>-seq data, using the Jvenn tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-venn-diagram"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Venn diagram</h3>

  <ol>
    <li><strong>Venn diagram</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>In <em>â€œList to compareâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert List to compareâ€</em>
              <ul>
                <li><em>â€œEnter your listâ€</em>: <code class="language-plaintext highlighter-rouge">Input file containing your list</code>
                  <ul>
                    <li><em>â€œSelect your fileâ€</em>: <code class="language-plaintext highlighter-rouge">Tissue-specific <span class="notranslate">expression</span> from IHC</code></li>
                    <li><em>â€œEnter the name of this listâ€</em>: <code class="language-plaintext highlighter-rouge">heart IHC</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert List to compareâ€</em>
              <ul>
                <li><em>â€œEnter your listâ€</em>: <code class="language-plaintext highlighter-rouge">Input file containing your list</code>
                  <ul>
                    <li><em>â€œSelect your fileâ€</em>: <code class="language-plaintext highlighter-rouge">Filtered tissue-specific <span class="notranslate">expression</span> from <span class="notranslate">RNA</span>seq</code></li>
                    <li><em>â€œEnter the name of this listâ€</em>: <code class="language-plaintext highlighter-rouge">heart <span class="notranslate">RNA</span>seq</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Examine <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the Venn diagram output:</p>

      <p><img src="../../images/biomarker-selection/jVenn_chart-tuto2.png" alt="Venn diagram output" /></p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
        <p>How many IDs are common to both IHC and <span class="notranslate">RNA</span>-seq data?</p>
        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>931 candidate biomarkers are detected by both IHC and <span class="notranslate">RNA</span>-seq data.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>For greater clarity weâ€™ll keep only the column with those 931 candidate biomarkers detected in both IHC and <span class="notranslate">RNA</span>-seq data in the
rest of our pipeline.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-obtain-list-of-candidate-biomarkers"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Obtain list of candidate biomarkers</h3>

  <ol>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œFile to cutâ€</em>: <code class="language-plaintext highlighter-rouge">Venn diagram text output</code></li>
        <li><em>â€œCut byâ€</em>: <code class="language-plaintext highlighter-rouge">fields</code>
          <ul>
            <li><em>â€œList of Fieldsâ€</em>: <code class="language-plaintext highlighter-rouge">Column: 3</code></li>
          </ul>
        </li>
      </ul>

      <p>Now weâ€™ll filter this dataset not to keep the â€˜NAâ€™ lines.</p>
    </li>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œInput fileâ€</em>: output from <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>â€œOperationâ€</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
        <li>In <em>â€œFilter by keywordsâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert Filter by keywordsâ€</em>
              <ul>
                <li><em>â€œSearch for exact match?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                <li><em>â€œEnter keywordsâ€</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>â€œCopy/paste keywords to find (keep or discard)â€</em>: <code class="language-plaintext highlighter-rouge">NA</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>â€œSort by column ?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>â€œSort result files byâ€</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Letâ€™s rename the 931 IDs dataset in â€œheart931â€ for simplification.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The pipeline will then continue based on these 931 candidates, from which we have to select biomarkers that are
highly specific to the heart using additional <span class="notranslate">expression</span> data (still from HPA).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-expression-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add <span class="notranslate">expression</span> data</h3>

  <ol>
    <li><strong>Add <span class="notranslate">expression</span> data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œEnter your IDs (Ensembl gene IDs only, e.g. ENSG00000064787)â€</em>: <code class="language-plaintext highlighter-rouge">Input file containing your IDs</code>
          <ul>
            <li><em>â€œSelect your fileâ€</em>: <code class="language-plaintext highlighter-rouge">Heart931</code> output we created in the previous step</li>
          </ul>
        </li>
        <li>In <em>â€œ<span class="notranslate">RNA</span>seq/Ab-based <span class="notranslate">expression</span> dataâ€</em>:
          <ul>
            <li><em>â€œSelect information to add to your listâ€</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Gene name</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Gene description</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span> tissue category</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span> tissue specificity abundance in Transcript Per Million</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We wish to focus on transcripts that have been classed as (according to the HPA definition):</p>
<ul>
  <li><strong>Tissue enriched</strong> (<span class="notranslate">expression</span> in one tissue at least fivefold higher than all other tissues),</li>
  <li><strong>Group enriched</strong> (fivefold higher average TPM in a group of two to to seven tissues compared to all other tissues) and</li>
  <li><strong>Tissue enhanced</strong> (fivefold higher average TPM in one or more tissues/cell lines compared to the mean TPM for all tissues)</li>
</ul>

<p>This information is listed in the column 4 : <em>â€œ<span class="notranslate">RNA</span> tissue categoryâ€</em> of the result dataset.</p>

<p>Letâ€™s use <strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to select the candidate biomarkers based on this
<em>â€œ<span class="notranslate">RNA</span> tissue categoryâ€</em> criterion.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-by-keywords-andor-numerical-value"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter by keywords and/or numerical value</h3>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œInput fileâ€</em>: output from <strong>Add <span class="notranslate">expression</span> data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>â€œFilter by keywordsâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert Filter by keywordsâ€</em>
              <ul>
                <li><em>â€œColumn number on which to apply the filterâ€</em>: <code class="language-plaintext highlighter-rouge">c4</code></li>
                <li><em>â€œEnter keywordsâ€</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>â€œCopy/paste keywords to find (keep or discard)â€</em>: <code class="language-plaintext highlighter-rouge">enriched enhanced</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>â€œSort by column ?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>â€œSort result files byâ€</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="output-2">Output</h3>
        <ul>
          <li>
            <p><strong>Filtered Add_<span class="notranslate">expression</span>_data</strong>: output list of the heart biomarkers with <span class="notranslate">RNA</span> tissue category
containing â€œenrichedâ€ or â€œenhancedâ€ (115 lines = what we are interested in)</p>
          </li>
          <li>
            <p><strong>Filtered Add_<span class="notranslate">expression</span>_data - discarded lines</strong>: output list of the heart biomarkers with
<span class="notranslate">RNA</span> tissue category NOT containing â€œenrichedâ€ or â€œenhancedâ€ (not what we are interested in)</p>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Letâ€™s rename the filtered dataset <strong>heart115</strong></p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset-1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We now have identified 115 candidates considered to have significantly higher <span class="notranslate">expression</span> in heart muscle according to HPA criteria.</p>

<h1 id="annotation-with-biochemical-and-cellular-features">Annotation with biochemical and cellular features</h1>

<p>Candidate biomarkers we want to identify have to be cytoplasmic and without transmembrane domains (TMD).
Thus we will retrieve protein features from neXtProt to retrieve those informations.</p>

<p>Since HPA only considers <a href="https://www.ensembl.org">Ensemble</a> <code class="language-plaintext highlighter-rouge">ENSG</code> identifiers (related to the gene), although <a href="https://www.nextprot.org/">neXtProt</a> uses <a href="https://www.uniprot.org/">UniProt</a> identifiers (related to proteins), first thing to do is to map the Ensembl identifiers contained our list of (115)
candidates to their corresponding UniProt accession number. The tool <strong>ID Converter</strong> is what we need to do so.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-id-converter"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: ID Converter</h3>

  <ol>
    <li><strong>ID Converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œEnter IDsâ€</em>: <code class="language-plaintext highlighter-rouge">Input file containing IDs</code></li>
        <li><em>â€œSelect your fileâ€</em>: <code class="language-plaintext highlighter-rouge">heart115</code></li>
        <li><em>â€œSpeciesâ€</em>: <code class="language-plaintext highlighter-rouge">Human (Homo sapiens)</code>
          <ul>
            <li><em>â€œType/source of IDsâ€</em>: <code class="language-plaintext highlighter-rouge">Ensembl gene ID (e.g. ENSG00000166913)</code></li>
            <li>In <em>â€œTarget typeâ€</em>:
              <ul>
                <li><em>â€œTarget type of IDs you would like to map toâ€</em>:
                  <ul>
                    <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Uniprot accession number</code></li>
                    <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Uniprot ID</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-3">Output</h3>
    <p><strong>ID converter</strong>: In this dataset, 2 columns (columns 6 and 7, at the end) which contain
UniProt accession number and ID are added.
Note: We used here the release â€œHomo sapiens 08/05/2019â€; results can be slightly different 
according to the version number of Uniprot.</p>
  </blockquote>
</blockquote>

<p>We have now UniProt IDs for the 115 candidate biomarkers: we are able to collect protein features from <a href="https://www.nextprot.org/">neXtProt</a>.
For this purpose, we use the <strong>Add protein features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> ProteoRE tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-protein-features"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add protein features</h3>

  <ol>
    <li><strong>Add protein features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œEnter your IDs (neXtProt or UniProt)â€</em>: <code class="language-plaintext highlighter-rouge">Input file containing your IDs </code>
          <ul>
            <li><em>â€œSelect your fileâ€</em>: the output from <strong>ID Converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><em>â€œColumn IDs (e.g : Enter c1 for column nÂ°1)â€</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
          </ul>
        </li>
        <li>In <em>â€œSelect featuresâ€</em>:
          <ul>
            <li><em>â€œPhysico-Chemical Featuresâ€</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Number of transmembrane domains</code></li>
              </ul>
            </li>
            <li><em>â€œLocalizationâ€</em>:
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Subcellular Location</code></li>
              </ul>
            </li>
            <li><em>â€œDisease informationâ€</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-4">Output</h3>
    <p><strong>Add information from NextProt</strong>: In this file (115 lines), 3 columns (columns 8, 9 and 10)
were added (at the end). These columns present TMDomains, Subcell Location and Diseases info.</p>
  </blockquote>
</blockquote>

<p>With this dataset, we can select proteins reported as localized in the cytoplasm and having
no transmembrane domains by running the Filter by keywords and/or numerical value tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-by-keywords-andor-numerical-value-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter by keywords and/or numerical value</h3>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œInput fileâ€</em>: the output from <strong>Add protein features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>â€œSelect an operator to combine your filters (if more than one)â€</em>: <code class="language-plaintext highlighter-rouge">AND</code></li>
        <li>In <em>â€œFilter by keywordsâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert Filter by keywordsâ€</em>
              <ul>
                <li><em>â€œColumn number on which to apply the filterâ€</em>: <code class="language-plaintext highlighter-rouge">c9</code></li>
                <li><em>â€œEnter keywordsâ€</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>â€œCopy/paste keywords to find (keep or discard)â€</em>: <code class="language-plaintext highlighter-rouge">cytoplasm cytosol</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>â€œFilter by numerical valueâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert Filter by numerical valueâ€</em>
              <ul>
                <li><em>â€œColumn number on which to apply the filterâ€</em>: <code class="language-plaintext highlighter-rouge">c8</code></li>
                <li><em>â€œValueâ€</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>â€œSort by column ?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>â€œSort result files by:â€</em>: <code class="language-plaintext highlighter-rouge">c5</code></li>
            <li><em>â€œSort in descending order ?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-5">Output</h3>
    <ul>
      <li><strong>Filtered Add_information_from_neXtProt</strong>: output list of the proteins having a cytoplasmic
location and no TMD (48 proteins)</li>
      <li><strong>Filtered Add_information_from_neXtProt - discarded lines</strong>: output list of the proteins NOT
cytoplasmic and having at least 1 TMD.</li>
    </ul>
  </blockquote>
</blockquote>

<p>We have now 45 proteins. The next step is to identify those proteins already previously observed in
LS MS/MS experiments.</p>

<h1 id="check-for-previous-detection-by-lc-msms-experiments">Check for previous detection by LC-MS/MS experiments</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-msms-observations-in-tissuefluid"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get MS/MS observations in tissue/fluid</h3>

  <ol>
    <li><strong>Get MS/MS observations in tissue/fluid</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œEnter your IDs (UniProt Accession number only)â€</em>: <code class="language-plaintext highlighter-rouge">Input file containing your IDs </code>
          <ul>
            <li><em>â€œSelect your fileâ€</em>: the filtered output from the previous step</li>
            <li><em>â€œColumn of IDsâ€</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
          </ul>
        </li>
        <li><em>â€œProteomics dataset (biological sample)â€</em>:
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Human Heart</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <code class="language-plaintext highlighter-rouge">Human Plasma non glyco</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-6">Output</h3>
    <p><strong>Get MS/MS observations in tissue/fluid</strong>: In this file, 2 columns (11 and 12, at the end)
were added with the info of number of times peptides were seen by MS/MS.</p>
  </blockquote>
</blockquote>

<p>Letâ€™s now keep only proteins that have already been seen by MS/MS in the plasma (last column of the file).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-for-proteins-seen-in-the-plasma"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter for proteins seen in the plasma</h3>

  <ol>
    <li><strong>Filter by keywords and/or numerical value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>â€œInput fileâ€</em>: the MA/MS observations output from the previous step</li>
        <li><em>â€œOperationâ€</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
        <li>In <em>â€œFilter by keywordsâ€</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>â€œInsert Filter by keywordsâ€</em>
              <ul>
                <li><em>â€œColumn number on which to apply the filterâ€</em>: <code class="language-plaintext highlighter-rouge">c12</code></li>
                <li><em>â€œSearch for exact match?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                <li><em>â€œEnter keywordsâ€</em>: <code class="language-plaintext highlighter-rouge">copy/paste</code>
                  <ul>
                    <li><em>â€œCopy/paste keywords to find (keep or discard)â€</em>: <code class="language-plaintext highlighter-rouge">NA</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>â€œSort by column ?â€</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>â€œSort results file byâ€</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="output-7">Output</h3>
    <ul>
      <li><strong>Filtered Get MS/MS observations in tissue/fluid</strong>: output list of the proteins
whose some peptides have been seen in plasma (21 proteins)</li>
      <li><strong>Filtered Get MS/MS observations in tissue/fluid - discarded lines</strong>:
output list of proteins with no peptides seen in the plasma</li>
    </ul>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>
<p>At the end of the process we end up with a list of 22 biomarkers that are <strong>highly enriched in heart muscle</strong>, localized
<strong>in the cytosol</strong> and <strong>detectable by MS in the plasma</strong>.
<em>NB: Please be aware that, due to databases update, number of biomarkers you end up with can be different from 21 that we obtain here.</em></p>

<p>Briefly and from a biological point of view, 3 of these proteins exhibit a relative low detection level in the plasma
compared to heart muscle tissue, and are reported with a very high heart-muscle-specific <span class="notranslate">RNA</span> abundance.
These potential mechanistic biomarkers of myocardial infarction include (i) cardiac Troponin I type 3 (TNNI3 (P19429))
that is routinely used as the most specific marker of myocardial injury and (ii) the heart-type fatty acid-binding protein
(FABP3 (P05413)) that has been proposed as a diagnostic and prognostic marker for acute and chronic cardiac injury.</p>

<p>Extraction of a <span class="notranslate">workflow</span> from your history is valuable for analyses reproducibility and traceability.
Moreover, a <span class="notranslate">workflow</span> can also be made reusable with modifiable parameters, and as a result this strategy can
be applied to other types of tissue injury (e.g. brain, liver, kidney).</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Public proteomics databases are useful resources for annotation of biomarker candidates</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Uhln2005">UhlÃ©n, M., E. BjÃ¶rling, C. Agaton, C. A.-K. Szigyarto, B. Amini <i>et al.</i>, 2005 <b>A Human Protein Atlas for Normal and Cancer Tissues Based on Antibody Proteomics</b>. Molecular &amp; Cellular Proteomics 4: 1920â€“1932.</span>
    
    
    <a href="https://doi.org/10.1074/mcp.m500279-mcp200">10.1074/mcp.m500279-mcp200</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Nguyen2019">Nguyen, L., V. Brun, F. Combes, V. Loux, and Y. Vandenbrouck, 2019 <b>Designing an In Silico Strategy to Select Tissue-Leakage Biomarkers Using the Galaxy Framework</b>, pp. 275â€“289 <span style="font-style: normal">in</span> <i>Methods in Molecular Biology</i>, <i>Springer New York</i>.</span>
    
    
    <a href="https://doi.org/10.1007/978-1-4939-9164-8_18">10.1007/978-1-4939-9164-8_18</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Biomarker candidate identification (Proteomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Florence Combes, David Christiany, Valentin Loux, Yves Vandenbrouck, 2021 <b>Biomarker candidate identification (Galaxy Training Materials)</b>. <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html">/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{proteomics-biomarker_selection,
    author = "Florence Combes and David Christiany and Valentin Loux and Yves Vandenbrouck",
    title = "Biomarker candidate identification (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and BjÃ¶rn GrÃ¼ning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Florence Combes, David Christiany, Valentin Loux, Yves Vandenbrouck)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/biomarker_selection/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
