<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: 2: RNA-seq counts to genes / 2: RNA-seq counts to genes / Transcriptomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@e493512ea65ff8e53e843f82cb09968f36ef7fff",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="2: RNA-seq counts to genes">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-04-10 11:49:01 +0000">
<meta name="DC.creator" content="Maria Doyle">
<meta name="DC.creator" content="Belinda Phipson">
<meta name="DC.creator" content="Jovana Maksimovic">
<meta name="DC.creator" content="Anna Trigos">
<meta name="DC.creator" content="Matt Ritchie">
<meta name="DC.creator" content="Harriet Dashnow">
<meta name="DC.creator" content="Shian Su">
<meta name="DC.creator" content="Charity Law"><meta name="description" content="Training material for all kinds of transcriptomics analysis.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Transcriptomics / 2: RNA-seq counts to genes / Hands-on: 2: RNA-seq counts to genes">
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis.">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "2: RNA-seq counts to genes",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-04-10 11:49:01 +0000",
  "datePublished": "2018-12-31 00:41:48 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00298",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
  },
  "abstract": "Measuring gene expression on a genome-wide scale has become common practice over the last two decades or so, with microarrays predominantly used pre-2008. With the advent of next generation sequencing technology in 2008, an increasing number of scientists use this technology to measure and understand changes in gene expression in often complex systems. As sequencing costs have decreased, using RNA-Seq to simultaneously measure the expression of tens of thousands of genes for multiple samples has never been easier. The cost of these experiments has now moved from generating the data to storing and analysing it.",
  "learningResourceType": "e-learning",
  "name": "2: RNA-seq counts to genes",
  "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html",
  "version": 21,
  "timeRequired": "PT2H",
  "teaches": "- Analysis of RNA-seq count data using limma-voom\n- QC of count data\n- Visualisation and interactive exploration of count data\n- Identification of differentially expressed genes",
  "keywords": [
    "Transcriptomics",
    "limma-voom",
    "mouse",
    "QC"
  ],
  "description": "## Abstract\n\nMeasuring gene expression on a genome-wide scale has become common practice over the last two decades or so, with microarrays predominantly used pre-2008. With the advent of next generation sequencing technology in 2008, an increasing number of scientists use this technology to measure and understand changes in gene expression in often complex systems. As sequencing costs have decreased, using RNA-Seq to simultaneously measure the expression of tens of thousands of genes for multiple samples has never been easier. The cost of these experiments has now moved from generating the data to storing and analysing it.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - What are the differentially expressed genes in the mammary gland of pregnant versus lactating mice?\n - How to analyze RNA count data using limma-voom?\n - How to perform quality control (QC) of RNA-seq count data?\n\n\n## Learning Objectives\n\n- Analysis of RNA-seq count data using limma-voom\n- QC of count data\n- Visualisation and interactive exploration of count data\n- Identification of differentially expressed genes\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html",
      "name": "1: RNA-Seq reads to counts",
      "description": "Hands-on for '1: RNA-Seq reads to counts' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/mblue9/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/mblue9/",
      "name": "Maria Doyle",
      "image": "https://avatars.githubusercontent.com/mblue9",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4847-8436",
      "orcid": "https://orcid.org/0000-0003-4847-8436"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bphipson/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bphipson/",
      "name": "Belinda Phipson",
      "image": "https://avatars.githubusercontent.com/bphipson",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/JovMaksimovic/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/JovMaksimovic/",
      "name": "Jovana Maksimovic",
      "image": "https://avatars.githubusercontent.com/JovMaksimovic",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annatrigos/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annatrigos/",
      "name": "Anna Trigos",
      "image": "https://avatars.githubusercontent.com/annatrigos",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/mritchie/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/mritchie/",
      "name": "Matt Ritchie",
      "image": "https://avatars.githubusercontent.com/mritchie",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/hdashnow/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/hdashnow/",
      "name": "Harriet Dashnow",
      "image": "https://avatars.githubusercontent.com/hdashnow",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/Shians/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/Shians/",
      "name": "Shian Su",
      "image": "https://avatars.githubusercontent.com/Shians",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/charitylaw/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/charitylaw/",
      "name": "Charity Law",
      "image": "https://avatars.githubusercontent.com/charitylaw",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/4273218",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>2: RNA-seq counts to genes</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/mblue9/" class="contributor-badge contributor-mblue9"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/mblue9?s=36" alt="Maria Doyle avatar" width="36" class="avatar" />
    Maria Doyle</a><a href="/training-material/hall-of-fame/bphipson/" class="contributor-badge contributor-bphipson"><img src="https://avatars.githubusercontent.com/bphipson?s=36" alt="Belinda Phipson avatar" width="36" class="avatar" />
    Belinda Phipson</a><a href="/training-material/hall-of-fame/JovMaksimovic/" class="contributor-badge contributor-JovMaksimovic"><img src="https://avatars.githubusercontent.com/JovMaksimovic?s=36" alt="Jovana Maksimovic avatar" width="36" class="avatar" />
    Jovana Maksimovic</a><a href="/training-material/hall-of-fame/annatrigos/" class="contributor-badge contributor-annatrigos"><img src="/training-material/assets/images/avatar.png" alt="Anna Trigos avatar" width="36" class="avatar" />
    Anna Trigos</a><a href="/training-material/hall-of-fame/mritchie/" class="contributor-badge contributor-mritchie"><img src="https://avatars.githubusercontent.com/mritchie?s=36" alt="Matt Ritchie avatar" width="36" class="avatar" />
    Matt Ritchie</a><a href="/training-material/hall-of-fame/hdashnow/" class="contributor-badge contributor-hdashnow"><img src="https://avatars.githubusercontent.com/hdashnow?s=36" alt="Harriet Dashnow avatar" width="36" class="avatar" />
    Harriet Dashnow</a><a href="/training-material/hall-of-fame/Shians/" class="contributor-badge contributor-Shians"><img src="https://avatars.githubusercontent.com/Shians?s=36" alt="Shian Su avatar" width="36" class="avatar" />
    Shian Su</a><a href="/training-material/hall-of-fame/charitylaw/" class="contributor-badge contributor-charitylaw"><img src="https://avatars.githubusercontent.com/charitylaw?s=36" alt="Charity Law avatar" width="36" class="avatar" />
    Charity Law</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are the differentially expressed genes in the mammary gland of pregnant versus lactating mice?</p>
</li>
        
        <li><p>How to analyze RNA count data using limma-voom?</p>
</li>
        
        <li><p>How to perform quality control (QC) of RNA-seq count data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Analysis of RNA-seq count data using limma-voom</p>
</li>
        
        <li><p>QC of count data</p>
</li>
        
        <li><p>Visualisation and interactive exploration of count data</p>
</li>
        
        <li><p>Identification of differentially expressed genes</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Quality Control</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Quality Control</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mapping</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mapping</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: 1: RNA-Seq reads to counts</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4273218">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/workflows/" title="2: RNA-seq counts to genes workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial"></abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy"></abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial"></abbr>
		</a>
	</li>
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Dec 31, 2018 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Apr 10, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00298">https://gxy.io/GTN:T00298</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">4.8</a> (5 recent ratings, 21 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 21 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Measuring gene expression on a genome-wide scale has become common practice over the last two decades or so, with microarrays predominantly used pre-2008. With the advent of next generation sequencing technology in 2008, an increasing number of scientists use this technology to measure and understand changes in gene expression in often complex systems. As sequencing costs have decreased, using RNA-Seq to simultaneously measure the expression of tens of thousands of genes for multiple samples has never been easier. The cost of these experiments has now moved from generating the data to storing and analysing it.</p>

<p>There are many steps involved in analysing an RNA-Seq experiment. The analysis begins with sequencing reads (FASTQ files). These are usually aligned to a reference genome, if available. Then the number of reads mapped to each gene can be counted. This results in a table of counts, which is what we perform statistical analyses on to determine differentially expressed genes and pathways. The purpose of this tutorial is to demonstrate how to perform differential expression on count data with <strong>limma-voom</strong>. How to generate counts from reads (FASTQs) is covered in the accompanying tutorial <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">RNA-seq reads to counts</a>.</p>

<h2 id="mouse-mammary-gland-dataset">Mouse mammary gland dataset</h2>

<p>The data for this tutorial comes from a Nature Cell Biology paper by <span class="citation"><a href="#Fu2015">Fu <i>et al.</i> 2015</a></span>. Both the raw data (sequence reads) and processed data (counts) can be downloaded from Gene Expression Omnibus database (GEO) under accession number <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60450">GSE60450</a>.</p>

<p>This study examined the expression profiles of basal and luminal cells in the mammary gland of virgin, pregnant and lactating mice. Six groups are present, with one for each combination of cell type and mouse status. Note that two biological replicates are used here, two independent sorts of cells from the mammary glands of virgin, pregnant or lactating mice, however three replicates is usually recommended as a minimum requirement for RNA-seq. In this tutorial we will use the GEO counts file as a starting point for our analysis. Alternatively, you could create a count matrix from the raw sequence reads, as demonstrated in the <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">RNA-seq reads to counts tutorial</a>. The GEO count file was generated from aligning the reads to the mouse <code class="language-plaintext highlighter-rouge">mm10</code> genome with the Rsubread aligner (<span class="citation"><a href="#Liao2019">Liao <i>et al.</i> 2019</a></span>), followed by counting reads mapped to RefSeq genes with featureCounts (<span class="citation"><a href="#liao2013featurecounts">Liao <i>et al.</i> 2013</a></span>), see the <a href="https://www.nature.com/articles/ncb3117">Fu paper</a> for details.</p>

<p>We will use <strong>limma-voom</strong> (<span class="citation"><a href="#Law2014">Law <i>et al.</i> 2014</a></span>) for identifying differentially expressed genes here. Other popular alternatives are edgeR (<span class="citation"><a href="#robinson2010edger">Robinson <i>et al.</i> 2010</a></span>) and DESeq2 (<span class="citation"><a href="#love2014moderated">Love <i>et al.</i> 2014</a></span>). Limma-voom has been shown to be perform well in terms of precision, accuracy and sensitivity (<span class="citation"><a href="#CostaSilva2017">Costa-Silva <i>et al.</i> 2017</a></span>) and, due to its speed, its particularly recommended for large-scale datasets with 100s of samples (<span class="citation"><a href="#Chen2016">Chen <i>et al.</i> 2016</a></span>).</p>

<p>This is a Galaxy tutorial based on material from the <a href="http://combine-australia.github.io/RNAseq-R/06-rnaseq-day1.html">COMBINE R RNAseq workshop</a>, first taught <a href="http://combine-australia.github.io/2016-05-11-RNAseq/">in this course</a>.</p>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/rna-seq-reads-to-counts/mouse_exp.png" alt="Tutorial Dataset. " width="567" height="402" loading="lazy" /><a target="_blank" href="../../images/rna-seq-reads-to-counts/mouse_exp.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Tutorial Dataset</figcaption></figure>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#mouse-mammary-gland-dataset" id="markdown-toc-mouse-mammary-gland-dataset">Mouse mammary gland dataset</a></li>
  <li><a href="#preparing-the-inputs" id="markdown-toc-preparing-the-inputs">Preparing the inputs</a>    <ol>
      <li><a href="#import-data" id="markdown-toc-import-data">Import data</a></li>
      <li><a href="#get-gene-annotations" id="markdown-toc-get-gene-annotations">Get gene annotations</a></li>
    </ol>
  </li>
  <li><a href="#differential-expression-with-limma-voom" id="markdown-toc-differential-expression-with-limma-voom">Differential expression with limma-voom</a>    <ol>
      <li><a href="#filtering-to-remove-lowly-expressed-genes" id="markdown-toc-filtering-to-remove-lowly-expressed-genes">Filtering to remove lowly expressed genes</a></li>
      <li><a href="#normalization-for-composition-bias" id="markdown-toc-normalization-for-composition-bias">Normalization for composition bias</a></li>
      <li><a href="#specify-contrasts-of-interest" id="markdown-toc-specify-contrasts-of-interest">Specify Contrast(s) of interest</a></li>
    </ol>
  </li>
  <li><a href="#qc-of-count-data" id="markdown-toc-qc-of-count-data">QC of count data</a>    <ol>
      <li><a href="#multidimensional-scaling-plot" id="markdown-toc-multidimensional-scaling-plot">Multidimensional scaling plot</a></li>
      <li><a href="#density-plots" id="markdown-toc-density-plots">Density plots</a></li>
      <li><a href="#box-plots" id="markdown-toc-box-plots">Box plots</a></li>
      <li><a href="#voom-variance-plot" id="markdown-toc-voom-variance-plot">Voom variance plot</a></li>
      <li><a href="#md-and-volcano-plots-for-de-results" id="markdown-toc-md-and-volcano-plots-for-de-results">MD and Volcano plots for DE results</a></li>
    </ol>
  </li>
  <li><a href="#testing-relative-to-a-threshold-treat" id="markdown-toc-testing-relative-to-a-threshold-treat">Testing relative to a threshold (TREAT)</a></li>
  <li><a href="#visualising-results" id="markdown-toc-visualising-results">Visualising results</a>    <ol>
      <li><a href="#heatmap-of-top-genes" id="markdown-toc-heatmap-of-top-genes">Heatmap of top genes</a></li>
      <li><a href="#stripcharts-of-top-genes" id="markdown-toc-stripcharts-of-top-genes">Stripcharts of top genes</a></li>
      <li><a href="#interactive-de-plots-glimma" id="markdown-toc-interactive-de-plots-glimma">Interactive DE plots (Glimma)</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<!--SNIPPET-->
<blockquote class="comment">   <div class="box-title comment-title" id="comment-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Results may vary</div>   <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, external databases, or because of stochastic processes in the algorithms.</p> </blockquote>
<p><!--END_SNIPPET--></p>

<h1 id="preparing-the-inputs">Preparing the inputs</h1>

<p>We will use three files for this analysis:</p>

<ul>
  <li><strong>Count matrix</strong> (genes in rows, samples in columns)</li>
  <li><strong>Sample information</strong> file (sample id, group)</li>
  <li><strong>Gene annotation</strong> file (gene id, symbol, description)</li>
</ul>

<h2 id="import-data">Import data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this RNA-seq exercise e.g. <code class="language-plaintext highlighter-rouge">RNA-seq with limma-voom</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is Unnamed history)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> Cancel button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the mammary gland counts table and the associated sample information file.</p>

      <p>To import the files, there are two options:</p>
      <ul>
        <li>Option 1: From a shared data library if available (<code class="language-plaintext highlighter-rouge">GTN - Material -&gt; transcriptomics -&gt; 2: RNA-seq counts to genes</code>)</li>
        <li>
          <p>Option 2: From <a href="https://zenodo.org/record/4273218">Zenodo</a></p>

          <!--SNIPPET-->
          <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
          <p><!--END_SNIPPET--></p>
        </li>
        <li>You can paste both links below into the <strong>Paste/Fetch</strong> box:</li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4273218/files/countdata.tsv
https://zenodo.org/record/4273218/files/factordata.tsv
</code></pre></div>      </div>
    </li>
    <li>Rename the counts dataset as <code class="language-plaintext highlighter-rouge">countdata</code> and the sample information dataset as <code class="language-plaintext highlighter-rouge">factordata</code> using the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon.</li>
    <li>
      <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">tabular</code>.
If the datatype is not <code class="language-plaintext highlighter-rouge">tabular</code>, please change the file type to <code class="language-plaintext highlighter-rouge">tabular</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">tabular</code> from <em>New type</em> dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>Lets take a look at the data. The <code class="language-plaintext highlighter-rouge">countdata</code> file contains information about genes (one gene per row), the first column has the Entrez gene id and the remaining columns contain information about the number of reads aligning to the gene in each experimental sample. There are two replicates for each cell type and time point (detailed sample info can be found in file GSE60450_series_matrix.txt from the GEO website). The first few rows and columns of the seqdata file are shown below.</p>

<figure id="figure-2" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/countdata.png" alt="countdata file. " width="1019" height="481" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/countdata.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Count file</figcaption></figure>

<p>The <code class="language-plaintext highlighter-rouge">factordata</code> file contains basic information about the samples that we will need for the analysis. See below. Note that the sample ids must match exactly with the ids in the counts file.</p>

<figure id="figure-3" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/factordata.png" alt="factordata file. " width="227" height="314" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/factordata.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Sample information file</figcaption></figure>

<blockquote class="details">
  <div class="box-title details-title" id="details-formatting-the-data"><button class="gtn-boxify-button details" type="button" aria-controls="details-formatting-the-data" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Formatting the data</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The files above have been formatted for you. If you are interested to know how they were formatted the information is below.</p>

  <p>The original files are available at</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4273218/files/GSE60450_Lactation-GenewiseCounts.txt
https://zenodo.org/record/4273218/files/SampleInfo.txt
</code></pre></div>  </div>

  <figure id="figure-4" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/seqdata.png" alt="seqdata file. " width="567" height="306" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/seqdata.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Count file (before formatting)</figcaption></figure>

  <figure id="figure-5" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/sampleinfo.png" alt="sampleinfo file. " width="454" height="250" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/sampleinfo.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Sample information file (before formatting)</figcaption></figure>

  <p>To create the file, <code class="language-plaintext highlighter-rouge">countdata</code>, that contains only the counts for the 12 samples i.e. well remove the gene length column with the <strong>Cut columns from a table (cut)</strong> tool. The sample names are also pretty long so well use the <strong>&gt;Replace Text in entire line</strong> tool to shorten these to contain only the relevant information about each sample. We will also replace the hyphen in the sample names with a dot so they match the names in the sample information file.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-format-the-counts-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Format the counts data</div>

    <ol>
      <li><strong>Cut columns from a table (cut)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>File to cut</em>: <code class="language-plaintext highlighter-rouge">seqdata</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Operation</em>: <code class="language-plaintext highlighter-rouge">Discard</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>List of fields</em>: Select <code class="language-plaintext highlighter-rouge">Column:2</code></li>
        </ul>
      </li>
      <li><strong>Replace Text in entire line</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>File to process</em>: output of <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Find pattern</em>: <code class="language-plaintext highlighter-rouge">_B[A-Z0-9_]+</code></li>
        </ul>
      </li>
      <li><strong>Replace Text in entire line</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>File to process</em>: output of <strong>Replace Text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Find pattern</em>: <code class="language-plaintext highlighter-rouge">-</code></li>
          <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Replace with</em>: <code class="language-plaintext highlighter-rouge">.</code></li>
        </ul>
      </li>
      <li>Rename file as <code class="language-plaintext highlighter-rouge">countdata</code> using the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon.</li>
    </ol>
  </blockquote>

  <p>To create the file, <code class="language-plaintext highlighter-rouge">factordata</code>, that contains the groups information that we need for the limma-voom tool. Well combine the cell type and mouse status to make 6 groups e.g. well combine the CellType <code class="language-plaintext highlighter-rouge">basal</code> with the Status <code class="language-plaintext highlighter-rouge">pregnant</code> for the group <code class="language-plaintext highlighter-rouge">basalpregnant</code>. Well use the <strong>Merge Columns</strong> tool to combine the cell type and mouse status columns in the sample information file, making a column with the 6 group names.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-format-the-sample-information-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Format the sample information file</div>

    <ol>
      <li><strong>Merge Columns together</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Select data</em>: <code class="language-plaintext highlighter-rouge">sampleinfo</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Merge column</em>: <code class="language-plaintext highlighter-rouge">Column: 3</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>with column</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
        </ul>
      </li>
      <li><strong>Cut columns from a table (cut)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>File to cut</em>: output of <strong>Merge Columns</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Operation</em>: <code class="language-plaintext highlighter-rouge">Keep</code></li>
          <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>List of fields</em>: Select <code class="language-plaintext highlighter-rouge">Column:2</code> and <code class="language-plaintext highlighter-rouge">Column:5</code></li>
        </ul>
      </li>
      <li>Rename file as <code class="language-plaintext highlighter-rouge">factordata</code> using the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon.</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="get-gene-annotations">Get gene annotations</h2>

<p>Optionally, gene annotations can be provided to the limma-voom tool and if provided the annotation will be available in the output files. Well get gene symbols and descriptions for these genes using the Galaxy <strong>annotateMyIDs</strong> tool, which provides annotations for human, mouse, fruitfly and zebrafish.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-gene-annotations"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get gene annotations</div>

  <ol>
    <li><strong>annotateMyIDs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>File with IDs</em>: <code class="language-plaintext highlighter-rouge">countdata</code></li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>File has header</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Organism</em>: <code class="language-plaintext highlighter-rouge">Mouse</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>ID Type</em>: <code class="language-plaintext highlighter-rouge">Entrez</code></li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>Output columns</em>: tick
          <ul>
            <li><code class="language-plaintext highlighter-rouge">ENTREZID</code></li>
            <li><code class="language-plaintext highlighter-rouge">SYMBOL</code></li>
            <li><code class="language-plaintext highlighter-rouge">GENENAME</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename file as <code class="language-plaintext highlighter-rouge">annodata</code> using the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (pencil) icon. The file should look like below.
      <figure id="figure-6" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/annodata.png" alt="annodata file. " width="510" height="360" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/annodata.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Gene annotation file</figcaption></figure>
    </li>
    <li>Check the number of lines shown on the datasets in the history, there should be 27,180 lines in both. There must be the same number of lines (rows) in the counts and annotation.</li>
  </ol>
</blockquote>

<h1 id="differential-expression-with-limma-voom">Differential expression with limma-voom</h1>

<h2 id="filtering-to-remove-lowly-expressed-genes">Filtering to remove lowly expressed genes</h2>

<p>It is recommended to filter for lowly expressed genes when running the limma-voom tool. Genes with very low counts across all samples provide little evidence for differential expression and they interfere with some of the statistical approximations that are used later in the pipeline. They also add to the multiple testing burden when estimating false discovery rates, reducing power to detect differentially expressed genes. These genes should be filtered out prior to further analysis.</p>

<p>There are a few ways to filter out lowly expressed genes. When there are biological replicates in each group, in this case we have a sample size of 2 in each group, we favour filtering on a minimum counts-per-million (CPM) threshold present in at least 2 samples. Two represents the smallest sample size for each group in our experiment. In this dataset, we choose to retain genes if they are expressed at a CPM above 0.5 in at least two samples. The CPM threshold selected can be compared to the raw count with the CpmPlots (see below).</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-details-on-filtering"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-details-on-filtering" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: More details on filtering</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The limma tool uses the <code class="language-plaintext highlighter-rouge">cpm</code> function from the edgeR package (<span class="citation"><a href="#robinson2010edger">Robinson <i>et al.</i> 2010</a></span>) to generate the CPM values which can then be filtered. Note that by converting to CPMs we are normalizing for the different sequencing depths for each sample. A CPM of 0.5 is used as it corresponds to a count of 10-15 for the library sizes in this data set. If the count is any smaller, it is considered to be very low, indicating that the associated gene is not expressed in that sample. A requirement for expression in two or more libraries is used as each group contains two replicates. This ensures that a gene will be retained if it is only expressed in one group. Smaller CPM thresholds are usually appropriate for larger libraries. As a general rule, a good threshold can be chosen by identifying the CPM that corresponds to a count of 10, which in this case is about 0.5. You should filter with CPMs rather than filtering on the counts directly, as the latter does not account for differences in library sizes between samples.</p>
</blockquote>

<h2 id="normalization-for-composition-bias">Normalization for composition bias</h2>

<p>In an RNA-seq analysis, the counts are normalized for different sequencing depths between samples. Normalizing to eliminate composition biases between samples is also typically performed. Composition biases can occur, for example, if there are a few highly expressed genes dominating in some samples, leading to less reads from other genes. By default, TMM normalization (<span class="citation"><a href="#robinson2010scaling">Robinson and Oshlack 2010</a></span>) is performed by the limma tool using the edgeR <code class="language-plaintext highlighter-rouge">calcNormFactors</code> function (this can be changed under <strong>Advanced Options</strong>). TMM stands for Trimmed Mean of M values, where a weighted trimmed mean of the log expression ratios is used to scale the counts for the samples. See the figure from the TMM paper below. Note the plot (Figure 1c) that shows how a few highly expressed genes in the liver sample (where the arrow is) results in the majority of other genes in the sample having the appearance of being expressed lower in liver. The mid-line through the points is offset from the expected zero and the TMM normalization factor (red line) scales the counts to adjust for this.</p>

<figure id="figure-7" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/TMM.png" alt="TMM normalization. " width="1130" height="944" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/TMM.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> TMM normalization (Robinson and Oshlack 2010)</figcaption></figure>

<h2 id="specify-contrasts-of-interest">Specify Contrast(s) of interest</h2>

<p>Since we are interested in differences between groups, we need to specify which comparisons we want to test. For example, if we are interested in knowing which genes are differentially expressed between the pregnant and lactating group in the basal cells we specify <code class="language-plaintext highlighter-rouge">basalpregnant-basallactate</code> for the <em>Contrast of Interest</em>. Note that the group names in the contrast must exactly match the names of the groups in the <code class="language-plaintext highlighter-rouge">factordata</code> file. More than one contrast can be specified using the <code class="language-plaintext highlighter-rouge">Insert Contrast</code> button, so we could look at more comparisons of the groups here, but first well take a look at <code class="language-plaintext highlighter-rouge">basalpregnant-basallactate</code>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-differential-expression-with-limma-voom"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Differential expression with limma-voom</div>

  <ol>
    <li><strong>limma</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Differential Expression Method</em>: <code class="language-plaintext highlighter-rouge">limma-voom</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Count Files or Matrix?</em>: <code class="language-plaintext highlighter-rouge">Single Count Matrix</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Count Matrix</em>: Select <code class="language-plaintext highlighter-rouge">countdata</code></li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Input factor information from file?</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Factor File</em>: Select <code class="language-plaintext highlighter-rouge">factordata</code></li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Use Gene Annotations?</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>Factor File</em>: Select <code class="language-plaintext highlighter-rouge">annodata</code></li>
          </ul>
        </li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Contrast of Interest</em>: <code class="language-plaintext highlighter-rouge">basalpregnant-basallactate</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Filter lowly expressed genes?</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>Filter on CPM or Count values?</em>: <code class="language-plaintext highlighter-rouge">CPM</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Minimum CPM</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Minimum Samples</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Report</code> produced by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon</li>
  </ol>
</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-multiple-factors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-multiple-factors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Multiple factors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If we need to account for additional sources of variation, for example, batch, sex, genotype etc, we can input that information as additional factors. For example, if we were interested in the genes differentially expressed between the luminal and basal cell types, we could include an additional column to account for the variation due to the different stages.</p>
  <figure id="figure-8" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/multiplefactors_factordata.png" alt="Multiple factors. " width="252" height="330" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/multiplefactors_factordata.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span> Multiple factors</figcaption></figure>

</blockquote>

<h1 id="qc-of-count-data">QC of count data</h1>

<p>Before we check out the differentially expressed genes, we can look at the <code class="language-plaintext highlighter-rouge">Report</code> information to check that the data is good quality and that the samples are as we would expect.</p>

<h2 id="multidimensional-scaling-plot">Multidimensional scaling plot</h2>

<p>By far, one of the most important plots we make when we analyse RNA-Seq data are MDS plots. An MDS plot is a visualisation of a principal components analysis, which determines the greatest sources of variation in the data. A principal components analysis is an example of an unsupervised analysis, where we dont need to specify the groups. If your experiment is well controlled and has worked well, what we hope to see is that the greatest sources of variation in the data are the treatments/groups we are interested in. It is also an incredibly useful tool for quality control and checking for outliers. This Galaxy limma tool outputs an MDS plot by default in the <code class="language-plaintext highlighter-rouge">Report</code> and a link is also provided to a PDF version (<code class="language-plaintext highlighter-rouge">MDSPlot_CellTypeStatus.pdf</code>). A scree plot is also produced that shows how much variation is attributed to each dimension. If there was a batch effect for example, you may see high values for additional dimensions, and you may choose to include batch as an additional factor in the differential expression analysis. The limma tool plots the first two dimensions by default (1 vs 2), however you can also plot additional dimensions 2 vs 3 and 3 vs 4 using under <strong>Output Options</strong> <em>Additional Plots</em> <code class="language-plaintext highlighter-rouge">MDS Extra</code>. These are displayed in the <code class="language-plaintext highlighter-rouge">Report</code> along with a link to a PDF version (MDSPlot_extra.pdf). Selecting the <code class="language-plaintext highlighter-rouge">Glimma Interactive Plots</code> will generate an interactive version of the MDS plot, see the plots section of the report below. If outlier samples are detected you may decide to remove them. Alternatively, you could downweight them by choosing the option in the limma tool <em>Apply voom with sample quality weights?</em>. The voom sample quality weighting is described in the paper <em>Why weight? Modelling sample and observational level variability improves power in RNA-seq analyses (<span class="citation"><a href="#Liu2015">Liu <i>et al.</i> 2015</a></span>)</em>.</p>

<figure id="figure-9" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/mdsscree.png" alt="MDS Plot. " width="1000" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/mdsscree.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> MDS Plot</figcaption></figure>

<p>Take a look at the MDS plot coloured by group.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Do you notice anything about the samples in this plot?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Two samples dont appear to be in the right place.</p>

  </blockquote>
</blockquote>

<p>It turns out that there has been a mix-up with two samples, they have been mislabelled in the sample information file. This shows how the MDS plot can also be useful to help identify if sample mix-ups may have occurred. We need to redo the limma-voom analysis with the correct sample information.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-use-the-rerun-button-to-redo-steps"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Use the Rerun button to redo steps</div>

  <ol>
    <li>Import the correct sample information file from <code class="language-plaintext highlighter-rouge">https://zenodo.org/record/4273218/files/factordata_fixed.tsv</code> and name as <code class="language-plaintext highlighter-rouge">factordata_fixed</code></li>
    <li>Use the <strong>Rerun</strong> <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> button in the History to run <strong>limma</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> as before with the <code class="language-plaintext highlighter-rouge">factordata_fixed</code> file and adding the following parameters:
      <ul>
        <li><strong>Output Options</strong>
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>Additional Plots</em> tick:
              <ul>
                <li><code class="language-plaintext highlighter-rouge">Glimma Interactive Plots</code></li>
                <li><code class="language-plaintext highlighter-rouge">Density Plots (if filtering)</code></li>
                <li><code class="language-plaintext highlighter-rouge">CpmsVsCounts Plots (if filtering on cpms)</code></li>
                <li><code class="language-plaintext highlighter-rouge">Box Plots (if normalising)</code></li>
                <li><code class="language-plaintext highlighter-rouge">MDS Extra (Dims 2vs3 and 3vs4)</code></li>
                <li><code class="language-plaintext highlighter-rouge">MD Plots for individual samples</code></li>
                <li><code class="language-plaintext highlighter-rouge">Heatmaps (top DE genes)</code></li>
                <li><code class="language-plaintext highlighter-rouge">Stripcharts (top DE genes)</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Delete the incorrect <code class="language-plaintext highlighter-rouge">factordata</code> file and its limma outputs to avoid any confusion.</li>
  </ol>
</blockquote>

<p>In the <code class="language-plaintext highlighter-rouge">Report</code> you should then see the correct MDS plot as below.</p>

<figure id="figure-10" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/mdsscree_corr.png" alt="MDS Plot. " width="1000" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/mdsscree_corr.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> MDS Plot (correct samples)</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What is the greatest source of variation in the data (i.e. what does dimension 1 represent)?
What is the second greatest source of variation in the data?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Dimension 1 represents the variation due to cell type, basal vs luminal. Dimension 2 represents the variation due to the stages, virgin, pregnant or lactating.</p>

  </blockquote>
</blockquote>

<p>Next, scroll down the <code class="language-plaintext highlighter-rouge">Report</code> to take a look at the <strong>Additional information</strong> and <strong>Summary of experimental data</strong> sections near the bottom. It should look similar to below. Here you can check that the correct samples have been assigned to the correct groups, what settings were used (e.g. filters, normalization method) and also how many genes were filtered out due to low expression.</p>

<p><a href="../../images/rna-seq-counts-to-genes/report_summary.png" rel="noopener noreferrer"><img src="../../images/rna-seq-counts-to-genes/report_summary.png" alt="Report summary. " width="1106" height="1018" loading="lazy" /></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many genes have been filtered out for low expression?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>11375 genes were filtered out as insignificant as they were without more than 0.5 CPM in at least 2 samples.</p>

  </blockquote>
</blockquote>

<h2 id="density-plots">Density plots</h2>

<p>Density plots can be output in the <code class="language-plaintext highlighter-rouge">Report</code> if <em>Filter lowly expressed genes</em> is selected. A link is also provided in the <code class="language-plaintext highlighter-rouge">Report</code> to a PDF version (<code class="language-plaintext highlighter-rouge">DensityPlots.pdf</code>). These plots allow comparison of the counts distributions before and after filtering. The samples are coloured by the groups. Count data is not normally distributed, so if we want to examine the distributions of the raw counts we need to log the counts. We typically check the distribution of the read counts on the log2 scale. A CPM value of 1 is equivalent to a log-CPM value of 0 and the CPM we used of 0.5 is equivalent to a log-CPM of -1. It can be seen in the Raw counts (before filtering) plot below, that a large proportion of genes within each sample are not expressed or lowly-expressed, and the Filtered counts plot shows our filter of CPM of 0.5 (in at least 2 samples) removes a lot of these uninformative genes.</p>

<figure id="figure-11" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/densityplots.png" alt="Density Plots. " width="1000" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/densityplots.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 11</strong>:</span> Density Plots</figcaption></figure>

<p>The <code class="language-plaintext highlighter-rouge">Report</code> provides links to PDFs of all plots shown in the <code class="language-plaintext highlighter-rouge">Report</code> and also to the rest of the additional plots selected to be output.</p>

<figure id="figure-12" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/report_plots.png" alt="Report Outputs. " width="352" height="492" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/report_plots.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 12</strong>:</span> Report outputs</figcaption></figure>

<blockquote class="details">
  <div class="box-title details-title" id="details-cpm-plots"><button class="gtn-boxify-button details" type="button" aria-controls="details-cpm-plots" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Cpm plots</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>We can also have a look more closely to see whether our threshold of 0.5 CPM does indeed correspond to a count of about 10-15 reads in each sample with the plots of CPM versus raw counts.</p>

  <p>Click on the <code class="language-plaintext highlighter-rouge">CpmPlots.pdf</code> link in the <code class="language-plaintext highlighter-rouge">Report</code>. You should see 12 plots, one for each sample. Two of the plots are shown below. From these plots we can see that 0.5 CPM is equivalent to ~10 counts in each of the 12 samples, so 0.5 seems to be an appropriate threshold for this dataset (these samples all have sequencing depth of 20-30 million, see the <code class="language-plaintext highlighter-rouge">Library information</code> file below, so a CPM value of 0.5 would be ~10 counts).</p>

  <figure id="figure-13" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/cpmsvscounts.png" alt="CPM threshold Plots. " width="1232" height="660" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/cpmsvscounts.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 13</strong>:</span> CPM vs Raw Counts Plots</figcaption></figure>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-thresholds"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Thresholds</div>

    <ul>
      <li>A threshold of 1 CPM in at least minimum group sample size is a good rule of thumb for samples with about 10 million reads. For larger library sizes increase the CPM theshold and for smaller library sizes decrease it. Check the CpmPlots to see if the selected threshold looks appropriate for the samples (equivalent to ~10 reads).</li>
    </ul>

  </blockquote>
</blockquote>

<h2 id="box-plots">Box plots</h2>

<p>We can also use box plots to check the distributions of counts in the samples. Box plots can be selected to be output by the Galaxy limma-voom tool if normalization is applied (TMM is applied by default). The plots are output in the <code class="language-plaintext highlighter-rouge">Report</code> and a link is also provided to a PDF version (<code class="language-plaintext highlighter-rouge">BoxPlots.pdf</code>). The samples are coloured by the groups. With the box plots for these samples we can see that overall the distributions are not identical but still not very different. If a sample is really far above or below the blue horizontal line we may need to investigate that sample further.</p>

<figure id="figure-14" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/boxplots.png" alt="Box Plots. " width="1000" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/boxplots.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 14</strong>:</span> Box Plots</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Compare the box plots before and after TMM normalisation. Can you see any differences?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>After the normalization more of the samples are closer to the median horizontal line.</p>

  </blockquote>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-normalization-factors"><button class="gtn-boxify-button details" type="button" aria-controls="details-normalization-factors" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Normalization factors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The TMM normalization generates normalization factors, where the product of these factors and the library sizes defines the effective library size. TMM normalization (and most scaling normalization methods) scale relative to one sample. The normalization factors multiply to unity across all libraries. A normalization factor below one indicates that the library size will be scaled down, as there is more suppression (i.e., composition bias) in that library relative to the other libraries. This is also equivalent to scaling the counts upwards in that sample. Conversely, a factor above one scales up the library size and is equivalent to downscaling the counts. We can see the normalization factors for these samples in the <code class="language-plaintext highlighter-rouge">Library information</code> file if we select to output it with <em>Output Library information file?</em>: <code class="language-plaintext highlighter-rouge">Yes</code>. Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon to view.</p>

  <figure id="figure-15" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/libinfo.png" alt="Library Info file. " width="567" height="300" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/libinfo.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 15</strong>:</span> Library information file</figcaption></figure>

  <blockquote class="question">
    <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>Which sample has the largest normalization factor? Which sample has the smallest?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <p>MCL1.LA has the largest normalization factor and MCL1.LE the smallest.</p>

    </blockquote>
  </blockquote>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-md-plots-for-samples"><button class="gtn-boxify-button details" type="button" aria-controls="details-md-plots-for-samples" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: MD plots for samples</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>It is considered good practice to make mean-difference (MD) plots for all the samples as a quality check, as described in this <a href="https://f1000research.com/articles/5-1438/v2">edgeR workflow article</a>. These plots allow expression profiles of individual samples to be explored more closely. An MD plot shows the log-fold change between a sample against the average expression across all the other samples. This visualisation can help you see if there are genes highly upregulated or downregulated in a sample. If we look at mean difference plots for these samples, we should be able to see the composition bias problem. The mean-difference plots show average expression (mean: x-axis) against log-fold-changes (difference: y-axis).</p>

  <p>Click on the <code class="language-plaintext highlighter-rouge">MDPlots_Samples.pdf</code> link in the <code class="language-plaintext highlighter-rouge">Report</code>. You should see 12 MD plots, one for each sample. Lets take a look at the plots for the two samples MCL1.LA and MCL1.LE that had the largest and smallest normalization factors. The MD plots on the left below show the counts normalized for library size and the plots on the right show the counts after the TMM normalization has been applied. MCL1.LA had the largest normalization factor and was above the median line in the unnormalized by TMM box plots. MCL1.LE had the smallest normalization factor and was below the median line in the box plots. These MD plots help show the composition bias problem has been addressed.</p>
  <figure id="figure-16" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/mdsampleLA.png" alt="MD Plot LA. " width="1208" height="618" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/mdsampleLA.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 16</strong>:</span> MD Plots for MCL1.LA before and after TMM normalization</figcaption></figure>
  <figure id="figure-17" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/mdsampleLE.png" alt="MD Plot LE. " width="1210" height="606" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/mdsampleLE.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 17</strong>:</span> MD Plots for MCL1.LE before and after TMM normalization</figcaption></figure>
</blockquote>

<h2 id="voom-variance-plot">Voom variance plot</h2>

<p>This plot is generated by the voom method and displayed in the <code class="language-plaintext highlighter-rouge">Report</code> along with a link to a PDF version (<code class="language-plaintext highlighter-rouge">VoomPlot.pdf</code>). Each dot represents a gene and it shows the mean-variance relationship of the genes in the dataset. This plot can help show if low counts have been filtered adequately and if there is a lot of variation in the data, as shown in the <strong>More details on Voom variance plots</strong> box below.</p>

<figure id="figure-18" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/voomplot.png" alt="Voom Plot. " width="500" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/voomplot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 18</strong>:</span> Voom Plot</figcaption></figure>

<p>The SA plot below plots log2 residual standard deviations against mean log-CPM values. The average log2 residual standard deviation is marked by a horizontal blue line. This plot shows how the dependence between the means and variances has been removed after the voom weights are applied to the data.</p>

<figure id="figure-19" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/saplot.png" alt="SA Plot. " width="500" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/saplot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 19</strong>:</span> SA Plot</figcaption></figure>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-details-on-voom-variance-plots"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-details-on-voom-variance-plots" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: More details on Voom variance plots</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If we didnt filter this dataset for the lowly expressed genes the variance plot would look like below.</p>

  <figure id="figure-20" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/voomplot_nofilt.png" alt="Voom Nofilter Plot. " width="500" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/voomplot_nofilt.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 20</strong>:</span> Voom Plot unfiltered counts</figcaption></figure>

  <p>If we look at the plot generated with the two samples mixed up we can see theres more variation.</p>

  <figure id="figure-21" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/voomplot_mixed.png" alt="Voom Mixedup Plot. " width="500" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/voomplot_mixed.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 21</strong>:</span> Voom Plot mixed-up samples</figcaption></figure>

  <p>More examples of the variation this plot can show can be seen in Figure 1 from the limma-voom paper (<span class="citation"><a href="#Law2014">Law <i>et al.</i> 2014</a></span>), shown below.</p>

  <figure id="figure-22" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/voom_variance_examples.jpg" alt="Voom Plot Examples. " width="785" height="532" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/voom_variance_examples.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 22</strong>:</span> Voom Plot more examples</figcaption></figure>

  <p><em>Figure 1: Mean-variance relationships. Gene-wise means and variances of RNA-seq data are represented by black points with a LOWESS trend. Plots are ordered by increasing levels of biological variation in datasets. (a) voom trend for HBRR and UHRR genes for Samples A, B, C and D of the SEQC project; technical variation only. (b) C57BL/6J and DBA mouse experiment; low-level biological variation. (c) Simulation study in the presence of 100 upregulating genes and 100 downregulating genes; moderate-level biological variation. (d) Nigerian lymphoblastoid cell lines; high-level biological variation. (e) Drosophila melanogaster embryonic developmental stages; very high biological variation due to systematic differences between samples. (f) LOWESS voom trends for datasets (a)(e). HBRR, Ambions Human Brain Reference RNA; LOWESS, locally weighted regression; UHRR, Stratagenes Universal Human Reference RNA</em>.</p>
</blockquote>

<h2 id="md-and-volcano-plots-for-de-results">MD and Volcano plots for DE results</h2>

<p>Genome-wide plots that are useful for checking differentially expressed (DE) results are MD plots (or MA plots) and Volcano plots. There are functions in limma for generating these plots and they are used by this tool. These plots are output by default and shown in the <code class="language-plaintext highlighter-rouge">Report</code> along with a link to PDF versions (<code class="language-plaintext highlighter-rouge">MDPlot_basalpregnant-basallactate.pdf</code> and <code class="language-plaintext highlighter-rouge">VolcanoPlot_basalpregnant-basallactate.pdf</code>). In the volcano plot the top genes (by adjusted p-value) are highlighted. The number of top genes is 10 by default and the user can specify the number of top genes to view (up to 100) under <strong>Advanced Options</strong>. For further information on volcano plots see the <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html">Volcano Plot tutorial</a></p>

<figure id="figure-23" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/mdvolplot_basalpregnant-basallactate.png" alt="MDVol Plot. " width="1000" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/mdvolplot_basalpregnant-basallactate.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 23</strong>:</span> MD Plot and Volcano Plot</figcaption></figure>

<p>The MD Plot highlighted genes are significant at an adjusted p-value (adj.P) threshold of 0.05 and exhibit log2-fold-change (lfc) of at least 0. These thresholds can be changed under <strong>Advanced Options</strong>.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many genes are differentially expressed at the default thresholds of adj.P=0.05 and lfc=0?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The number of DE genes at these adj.P and lfc thresholds is shown in the table in the <code class="language-plaintext highlighter-rouge">Report</code> as below.</p>

    <p><a href="../../images/rna-seq-counts-to-genes/decounts.png" rel="noopener noreferrer"><img src="../../images/rna-seq-counts-to-genes/decounts.png" alt="DE counts. " width="425" height="139" loading="lazy" /></a></p>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-on-adjusted-p-value"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: on adjusted P value</div>

  <p><strong>A note about deciding how many genes are significant</strong>: In order to decide which genes are differentially expressed, we usually take a cut-off (e.g. 0.05 or 0.01) on the adjusted p-value, NOT the raw p-value. This is because we are testing many genes (more than 15000 genes here), and the chances of finding differentially expressed genes is very high when you do that many tests. Hence we need to control the false discovery rate, which is the adjusted p-value column in the results table. What this means is that, if we choose an adjusted p-value cut-off of 0.05, and if 100 genes are significant at a 5% false discovery rate, we are willing to accept that 5 will be false positives.</p>
</blockquote>

<h1 id="testing-relative-to-a-threshold-treat">Testing relative to a threshold (TREAT)</h1>

<p>When there is a lot of differential expression, sometimes we may want to cut-off on a fold change threshold, as well as a p-value threshold, so that we follow up on the most biologically significant genes. However, it is not recommended to simply rank by p-value and then discard genes with small logFCs, as this has been shown to increase the false discovery rate. In other words, you are not controlling the false discovery rate at 5% any more. There is a function called <code class="language-plaintext highlighter-rouge">treat</code> in limma that performs this style of analysis correctly (<span class="citation"><a href="#McCarthy2009">McCarthy and Smyth 2009</a></span>). TREAT will simply take a user-specified log fold change cut-off and recalculate the moderated t-statistics and p-values with the new information about logFC. There are thousands of genes differentially expressed in this <code class="language-plaintext highlighter-rouge">basalpregnant-basallactate</code> comparison, so lets rerun the analysis applying TREAT and similar thresholds to what was used in the Fu paper: an adjusted P value of 0.01 (1% false discovery rate) and a log-fold-change cutoff of 0.58 (equivalent to a fold change of 1.5).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-testing-relative-to-a-threshold-treat"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Testing relative to a threshold (TREAT)</div>

  <ol>
    <li>Use the <strong>Rerun</strong> <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> button in the History to rerun <strong>limma</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em><strong>Advanced Options</strong></em>
          <ul>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Minimum Log2 Fold Change</em>: <code class="language-plaintext highlighter-rouge">0.58</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>P-Value Adjusted Threshold</em>: <code class="language-plaintext highlighter-rouge">0.01</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>Test significance relative to a fold-change threshold (TREAT)</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Add a tag <code class="language-plaintext highlighter-rouge">#treat</code> to the <code class="language-plaintext highlighter-rouge">Report</code> output and inspect the report</li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain plus data versus minus data. For example, does dataset 10 contain plus data or minus data? Probably minus but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
  <p><!--END_SNIPPET--></p>

</blockquote>

<p>We can see that much fewer genes are now highlighted in the MD plot and identified as differentially expressed.</p>

<figure id="figure-24" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/TREAT_mdvolplot_basalpregnant-basallactate.png" alt="TREAT MDVol Plot. " width="1000" height="500" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/TREAT_mdvolplot_basalpregnant-basallactate.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 24</strong>:</span> TREAT MD and Volcano plots</figcaption></figure>

<figure id="figure-25" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/TREATdecounts.png" alt="TREAT DE counts. " width="397" height="143" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/TREATdecounts.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 25</strong>:</span> TREAT differentially expressed genes</figcaption></figure>

<h1 id="visualising-results">Visualising results</h1>

<p>In addition to the plots already discussed, it is recommended to have a look at the expression levels of the individual samples for the genes of interest, before following up on the DE genes with further lab work. The Galaxy limma tool can auto-generate heatmaps of the top genes to show the expression levels across the <em>samples</em>. This enables a quick view of the expression of the top differentially expressed genes and can help show if expression is consistent amongst replicates in the groups.</p>

<h2 id="heatmap-of-top-genes">Heatmap of top genes</h2>

<p>Click on the <code class="language-plaintext highlighter-rouge">Heatmap_basalpregnant-basallactate.pdf</code> link in the <code class="language-plaintext highlighter-rouge">Report</code>. You should see a plot like below.</p>

<figure id="figure-26" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/heatmap.png" alt="Heatmap. " width="495" height="511" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/heatmap.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 26</strong>:</span> Heatmap of top genes</figcaption></figure>

<h2 id="stripcharts-of-top-genes">Stripcharts of top genes</h2>

<p>The limma-voom tool can also auto-generate stripcharts to view the expression of the top genes across the groups. Click on the <code class="language-plaintext highlighter-rouge">Stripcharts_basalpregnant-basallactate.pdf</code> link in the <code class="language-plaintext highlighter-rouge">Report</code>. You should see 10 plots, one for each top gene. Four are shown below. Note that here you can see if replicates tend to group together and how the expression compares to the other groups.</p>

<figure id="figure-27" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/stripcharts.png" alt="Stripchart Plot. " width="1312" height="918" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/stripcharts.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 27</strong>:</span> Stripcharts of top genes</figcaption></figure>

<h2 id="interactive-de-plots-glimma">Interactive DE plots (Glimma)</h2>

<p>Interactive versions of the MD and Volcano plots can be output by the limma-voom tool via the Glimma package (<span class="citation"><a href="#Su2017">Su <i>et al.</i> 2017</a></span>), if a gene annotation file is provided and <code class="language-plaintext highlighter-rouge">Glimma Interactive Plots</code> is selected. Links to the Glimma html pages are generated in the <code class="language-plaintext highlighter-rouge">Report</code>.</p>

<p>Lets take a look at the interactive MD plot. Click on the <code class="language-plaintext highlighter-rouge">Glimma_MDPlot_basalpregnant-basallactate.html</code> link in the <code class="language-plaintext highlighter-rouge">Report</code>. You should see a two-panel interactive MD plot like below. The left plot shows the log-fold-change vs average expression. The right plot shows the expression levels of a particular gene of each sample by groups (similar to the stripcharts). Hovering over points on left plot will plot expression level for corresponding gene, clicking on points will fix the expression plot to gene. Clicking on rows on the table has the same effect as clicking on the corresponding gene in the plot.</p>

<iframe src="../../images/rna-seq-counts-to-genes/glimma/MD-Plot.html" width="100%" height="1000"></iframe>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-search-for-a-gene-of-interest"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Search for a gene of interest</div>

  <p><code class="language-plaintext highlighter-rouge">Egf</code> was a gene identifed as very highly expressed in the Fu paper and confirmed with qRT-PCR, see Fig. 6c from the paper below.</p>
  <figure id="figure-28" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/fu_egf.png" alt="Fu EGF gene. " width="276" height="324" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/fu_egf.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 28</strong>:</span> Fu et al, Nat Cell Biol 2015</figcaption></figure>

  <p>Search for <code class="language-plaintext highlighter-rouge">Egf</code> in the Glimma interactive table. You should see something similar to below.</p>

  <figure id="figure-29" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/glimma_egf.png" alt="Glimma EGF gene. " width="1578" height="870" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/glimma_egf.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 29</strong>:</span> Glimma EGF gene</figcaption></figure>

  <p>Notice that in the plot on the right above, showing <code class="language-plaintext highlighter-rouge">Egf</code> expression for all samples, we can see it is highly expressed in the luminal lactate group but not the other samples.</p>
</blockquote>

<p>Multiple contrasts can be run with the limma tool. For example, we can compare the pregnant and lactating conditions for both the basal and luminal cells. So lets rerun the limma-voom TREAT analysis (adj.P &lt;0.01 and lfc=0.58) and this time use the <code class="language-plaintext highlighter-rouge">Insert Contrast</code> button to include the additional contrast <code class="language-plaintext highlighter-rouge">luminalpregnant - luminallactate</code>. We can then see how the number of differentially expressed genes in the luminal cells compares to the basal cells.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-multiple-contrasts"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run multiple contrasts</div>

  <ol>
    <li>Use the <strong>Rerun</strong> <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> button in the History to rerun the <strong>limma</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <code class="language-plaintext highlighter-rouge">#treat</code> analysis adding the following parameters <em>(i.e. run with 2 contrasts)</em>:
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Contrast of Interest</em>: <code class="language-plaintext highlighter-rouge">basalpregnant-basallactate</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>Contrast of Interest</em>: <code class="language-plaintext highlighter-rouge">luminalpregnant-luminallactate</code></li>
      </ul>
    </li>
    <li>Add a tag <code class="language-plaintext highlighter-rouge">#multiple-contrasts</code> to the <code class="language-plaintext highlighter-rouge">Report</code> output and inspect the report</li>
  </ol>

  <p>You should find that there are more genes differentially expressed for the luminal cells than the basal. There are ~274 genes DE in basal cells versus ~ 1610 in the luminal cells.</p>
  <figure id="figure-30" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/basal_luminal_decounts.png" alt="Basal Luminal DE counts. " width="425" height="171" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/basal_luminal_decounts.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 30</strong>:</span> Basal vs Luminal DE counts</figcaption></figure>

  <p>This is similar to what Fu et al found, many more genes differentially expressed in the luminal cells on lactation, compared to the basal cells.</p>
  <figure id="figure-31" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/fu_degenes.png" alt="Fu DE genes. " width="1666" height="226" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/fu_degenes.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 31</strong>:</span> Fu et al, Nat Cell Biol 2015</figcaption></figure>
</blockquote>

<p>The tables of differentially expressed genes are output as links in the <code class="language-plaintext highlighter-rouge">Report</code> (<code class="language-plaintext highlighter-rouge">limma-voom_basalpregnant-basallactate.tsv</code> and <code class="language-plaintext highlighter-rouge">limma-voom_luminalpregnant-luminallactate.tsv</code>), see below, and also as datasets in the history (<code class="language-plaintext highlighter-rouge">DE tables</code>). With multiple contrasts, a plot for each contrast is generated for relevant plots, as shown below. This enables a quick and easy visual comparison of the contrasts.</p>

<figure id="figure-32" style="max-width: 90%;"><img src="../../images/rna-seq-counts-to-genes/multiple_contrasts.png" alt="Multiple contrasts. " width="491" height="887" loading="lazy" /><a target="_blank" href="../../images/rna-seq-counts-to-genes/multiple_contrasts.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 32</strong>:</span> Multiple contrasts output</figcaption></figure>

<blockquote class="details">
  <div class="box-title details-title" id="details-r-code-in-detail"><button class="gtn-boxify-button details" type="button" aria-controls="details-r-code-in-detail" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: R code in detail</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ol>
    <li>The <code class="language-plaintext highlighter-rouge">Report</code> with all the plots and tables can be downloaded by clicking on the floppy disk icon on the dataset in the history.</li>
    <li>If you want to see the R code used, or to explore some of the outputs in R, you can select to output the Rscript and/or RData files in the tool form.</li>
  </ol>
</blockquote>

<p>To see some methods for identifying differentially expressed pathways in this dataset, see the follow-on tutorial <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html">RNA-seq genes to pathways</a>. To see how to create a heatmap of custom genes using this dataset, see the tutorial <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html">Visualization of RNA-Seq results with heatmap2</a></p>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial we have seen how counts files can be converted into differentially expressed genes with limma-voom. This follows on from the accompanying tutorial, <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html">RNA-seq reads to counts</a>, that showed how to generate counts from the raw reads (FASTQs) for this dataset. In this part we have seen ways to visualise the count data, and QC checks that can be performed to help assess the quality and results. We have also reproduced results similar to what the authors found in the original paper with this dataset. For further reading on analysis of RNA-seq count data and the methods used here, see the articles; <em>RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR (<span class="citation"><a href="#Law2018">Law <i>et al.</i> 2018</a></span>)</em>  and <em>From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline (<span class="citation"><a href="#Chen2016">Chen <i>et al.</i> 2016</a></span>)</em>.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial </button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! '
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>The limma-voom tool can be used to perform differential expression and output useful plots</p>
</li>
                        
                        <li><p>Multiple comparisons can be input and compared</p>
</li>
                        
                        <li><p>Results can be interactively explored with limma-voom via Glimma</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="McCarthy2009">McCarthy, D. J., and G. K. Smyth, 2009 <b>Testing significance relative to a fold-change threshold is a TREAT</b>. Bioinformatics 25: 765771. <a href="https://doi.org/10.1093/bioinformatics/btp053">10.1093/bioinformatics/btp053</a></li>
<li id="robinson2010edger">Robinson, M. D., D. J. McCarthy, and G. K. Smyth, 2010 <b>edgeR: a Bioconductor package for differential expression analysis of digital gene expression data</b>. Bioinformatics 26: 139140.</li>
<li id="robinson2010scaling">Robinson, M. D., and A. Oshlack, 2010 <b>A scaling normalization method for differential expression analysis of RNA-seq data</b>. Genome biology 11: 19.</li>
<li id="liao2013featurecounts">Liao, Y., G. K. Smyth, and W. Shi, 2013 <b>featureCounts: an efficient general purpose program for assigning sequence reads to genomic features</b>. Bioinformatics 30: 923930. <a href="https://academic.oup.com/bioinformatics/article/30/7/923/232889">https://academic.oup.com/bioinformatics/article/30/7/923/232889</a></li>
<li id="Law2014">Law, C. W., Y. Chen, W. Shi, and G. K. Smyth, 2014 <b>voom: precision weights unlock linear model analysis tools for RNA-seq read counts</b>. Genome Biology 15: R29. <a href="https://doi.org/10.1186/gb-2014-15-2-r29">10.1186/gb-2014-15-2-r29</a></li>
<li id="love2014moderated">Love, M. I., W. Huber, and S. Anders, 2014 <b>Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2</b>. Genome Biology 15: 550. <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0550-8</a></li>
<li id="Fu2015">Fu, N. Y., A. C. Rios, B. Pal, R. Soetanto, A. T. L. Lun <i>et al.</i>, 2015 <b>EGF-mediated induction of Mcl-1 at the switch to lactation is essential for alveolar cell survival</b>. Nature Cell Biology 17: 365375. <a href="https://doi.org/10.1038/ncb3117">10.1038/ncb3117</a></li>
<li id="Liu2015">Liu, R., A. Z. Holik, S. Su, N. Jansz, K. Chen <i>et al.</i>, 2015 <b>Why weight? Modelling sample and observational level variability improves power in RNA-seq analyses</b>. Nucleic Acids Research 43: e97e97. <a href="https://doi.org/10.1093/nar/gkv412">10.1093/nar/gkv412</a></li>
<li id="Chen2016">Chen, Y., A. T. L. Lun, and G. K. Smyth, 2016 <b>From reads to genes to pathways: differential expression analysis of RNA-Seq experiments using Rsubread and the edgeR quasi-likelihood pipeline</b>. F1000Research 5: 1438. <a href="https://doi.org/10.12688/f1000research.8987.2">10.12688/f1000research.8987.2</a></li>
<li id="CostaSilva2017">Costa-Silva, J., D. Domingues, and F. M. Lopes, 2017 <b>RNA-Seq differential expression analysis: An extended review and a software tool</b> (Z. Wei, Ed.). PLOS ONE 12: e0190152. <a href="https://doi.org/10.1371/journal.pone.0190152">10.1371/journal.pone.0190152</a></li>
<li id="Su2017">Su, S., C. W. Law, C. Ah-Cann, M.-L. Asselin-Labat, M. E. Blewitt <i>et al.</i>, 2017 <b>Glimma: interactive graphics for gene expression analysis</b> (B. Berger, Ed.). Bioinformatics 33: 20502052. <a href="https://doi.org/10.1093/bioinformatics/btx094">10.1093/bioinformatics/btx094</a></li>
<li id="Law2018">Law, C. W., M. Alhamdoosh, S. Su, X. Dong, L. Tian <i>et al.</i>, 2018 <b>RNA-seq analysis is easy as 1-2-3 with limma,  Glimma and edgeR</b>. F1000Research 5: 1408. <a href="https://doi.org/10.12688/f1000research.9005.3">10.12688/f1000research.9005.3</a></li>
<li id="Liao2019">Liao, Y., G. K. Smyth, and W. Shi, 2019 <b>The R package Rsubread is easier,  faster,  cheaper and better for alignment and quantification of RNA sequencing reads</b>. Nucleic Acids Research 47: e47e47. <a href="https://doi.org/10.1093/nar/gkz114">10.1093/nar/gkz114</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=transcriptomics/rna-seq-counts-to-genes"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=transcriptomics/rna-seq-counts-to-genes">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Maria Doyle, Belinda Phipson, Jovana Maksimovic, Anna Trigos, Matt Ritchie, Harriet Dashnow, Shian Su, Charity Law,  <b>2: RNA-seq counts to genes (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{transcriptomics-rna-seq-counts-to-genes,
author = "Maria Doyle and Belinda Phipson and Jovana Maksimovic and Anna Trigos and Matt Ritchie and Harriet Dashnow and Shian Su and Charity Law",
	title = "2: RNA-seq counts to genes (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Fll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Bjrn Grning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: 3: RNA-seq genes to pathways</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Visualization of RNA-Seq results with heatmap2</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Visualization of RNA-Seq results with Volcano Plot</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: RNA-Seq analysis with AskOmics Interactive Tool</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: RNA-Seq analysis with AskOmics Interactive Tool</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


                    </ul>
                </blockquote>
                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: 74a8bef53a00
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: merge_cols
  owner: devteam
  revisions: f2aac0c5c60d
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: annotatemyids
  owner: iuc
  revisions: 1a125daea0d8
  tool_panel_section_label: Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: limma_voom
  owner: iuc
  revisions: d5a940112511
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">17</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.23529411764705882">4</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>November 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: making different plots :)
					
				</li>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: the method of normalization
					
				</li>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: help full tutor
					<b>Disliked</b>: A video of this tutorial available before the work shop would be the great help for many pupils especially who are international students and have different educational system background.
				</li>
				
		    </ul>
		    
		    <b>October 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: It was very easy to follow
					<b>Disliked</b>: More background information of specific details available if wanted by the user.
				</li>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: easy to follow
					
				</li>
				
		    </ul>
		    
		    <b>October 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Easy to follow
					<b>Disliked</b>: -
				</li>
				
		    </ul>
		    
		    <b>December 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Everything was well-annotated. Thank you so much!
					<b>Disliked</b>: I want to see ''How to DESeq2 Results File (DEG List) with Tabular format open in Excel?''.  
				</li>
				
		    </ul>
		    
		    <b>September 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Good explanation!
					<b>Disliked</b>: Explanation on how to understand the figures in depth
				</li>
				
		    </ul>
		    
		    <b>April 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: easy to follow
					<b>Disliked</b>: interactive plots could not be opened https://usegalaxy.org/datasets/bbd44e69cb8906b52089f52da217dea4/display/glimma_basalpregnant-basallactate/MD-Plot.html
				</li>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: I very much enjoy this tutorial! coming back from extensive R experience using limma, DEseq, still find this Galaxy thing makes things easier
					
				</li>
				
		    </ul>
		    
		    <b>March 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: It was really clear and easy to follow. 
					
				</li>
				
		    </ul>
		    
		    <b>August 2019</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Easy to follow and showed what our plots/table should look like
					
				</li>
				
		    </ul>
		    
		    <b>April 2019</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Clear, comprehensive, great screenshots
					<b>Disliked</b>: Nothing so far
				</li>
				
		    </ul>
		    
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00298">gxy.io/GTN:T00298</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/e493512ea65ff8e53e843f82cb09968f36ef7fff">e493512</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>