<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Hi-C analysis of Drosophila melanogaster cells using HiCExplorer</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:title" content="Galaxy Training: Hi-C analysis of Drosophila melanogaster cells using HiCE..." />
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Epigenetics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fhicexplorer%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fhicexplorer%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fhicexplorer%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fhicexplorer%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fhicexplorer%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fhicexplorer%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/epigenetics/tutorials/hicexplorer/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
  },
  "courseCode": "epigenetics / hicexplorer / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Hi-C analysis of Drosophila melanogaster cells using HiCExplorer' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - Why is a Hi-C analysis useful?\n - What is 'chromosome conformation capture'?\n - What are main steps in order to generate and plot a Hi-C contact matrix?\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Joachim Wolff"
    },
    {
      "@type": "Person",
      "name": "Fidel Ramirez"
    },
    {
      "@type": "Person",
      "name": "Vivek Bhardwaj"
    },
    {
      "@type": "Person",
      "name": "Ekaterina Polkh"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Joachim Wolff"
    },
    {
      "@type": "Person",
      "name": "Fidel Ramirez"
    },
    {
      "@type": "Person",
      "name": "Vivek Bhardwaj"
    },
    {
      "@type": "Person",
      "name": "Ekaterina Polkh"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Hi-C analysis of Drosophila melanogaster cells using HiCExplorer</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/joachimwolff/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/joachimwolff" alt="Joachim Wolff">Joachim Wolff</a>, <a href="/training-material/hall-of-fame/fidelram/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/fidelram" alt="Fidel Ramirez">Fidel Ramirez</a>, <a href="/training-material/hall-of-fame/vivekbhr/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/vivekbhr" alt="Vivek Bhardwaj">Vivek Bhardwaj</a>, <a href="/training-material/hall-of-fame/polkhe/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/polkhe" alt="Ekaterina Polkh">Ekaterina Polkh</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>Why is a Hi-C analysis useful?</p>
</li>
            
            <li><p>What is ‚Äòchromosome conformation capture‚Äô?</p>
</li>
            
            <li><p>What are main steps in order to generate and plot a Hi-C contact matrix?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/epigenetics/slides/introduction-dna-methylation.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1183661">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/hicexplorer/workflows/" title="Workflows" alt="Hi-C analysis of Drosophila melanogaster cells using HiCExplorer workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/hicexplorer/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In this HiCExplorer tutorial we will generate and plot a Hi-C contact matrix.
For this the following steps are necessary to be performed:</p>
<ol>
  <li>Map the Hi-C <span class="notranslate">reads</span> to the reference genome</li>
  <li>Creation of a Hi-C matrix</li>
  <li>Plotting the Hi-C matrix</li>
  <li>Correction of Hi-C matrix</li>
  <li>TAD Calling</li>
  <li>A/B compartments computation</li>
  <li>pyGenomeTracks visualization</li>
  <li>Loop detection</li>
</ol>

<p>After a corrected Hi-C matrix is created other tools can be used to visualize it, call TADS or compare it with other matrices.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#reads-mapping" id="markdown-toc-reads-mapping"><span class="notranslate">Reads</span> <span class="notranslate">mapping</span></a></li>
  <li><a href="#creation-of-a-hi-c-matrix" id="markdown-toc-creation-of-a-hi-c-matrix">Creation of a Hi-C matrix</a></li>
  <li><a href="#plotting-the-hi-c-matrix" id="markdown-toc-plotting-the-hi-c-matrix">Plotting the Hi-C matrix</a></li>
  <li><a href="#correction-of-hi-c-matrix" id="markdown-toc-correction-of-hi-c-matrix">Correction of Hi-C matrix</a></li>
  <li><a href="#tad-calling" id="markdown-toc-tad-calling">TAD calling</a></li>
  <li><a href="#ab-compartments-computation" id="markdown-toc-ab-compartments-computation">A/B compartments computation</a></li>
  <li><a href="#integrating-hi-c-and-other-data" id="markdown-toc-integrating-hi-c-and-other-data">Integrating Hi-C and other data</a></li>
  <li><a href="#loop-detection" id="markdown-toc-loop-detection">Loop detection</a></li>
</ol>

</blockquote>

<h1 id="data-upload">Data upload</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import from <a href="https://doi.org/10.5281/zenodo.1183661">Zenodo</a>.</p>

      <blockquote class="tip">
        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ol>
          <li>Copy the link location</li>
          <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
          <li>
            <p>Select <strong>Paste/Fetch Data</strong></p>

            <p>Below are the links to the read files that can be copied and pasted in the upload manager.</p>

            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1183661/files/HiC_S2_1p_10min_lowU_R1.fastq.gz
https://zenodo.org/record/1183661/files/HiC_S2_1p_10min_lowU_R2.fastq.gz
</code></pre></div>            </div>
          </li>
          <li>Paste the links into the text field</li>
          <li>Press <strong>Start</strong></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Rename the data set to something meaningful, e.g. <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R1</code> and <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R2</code>.
By default, when data is imported via its link, <span class="notranslate">Galaxy</span> names it with its URL.</p>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-get-data-from-public-sources"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Get data from public sources</h3>
    <p>HiCExplorer needs as input the forward and reverse strand of a pair end read which are mapped independently. A usual start point for a typical analysis is the given GSE number of a publication, e.g. GSE63525 for Rao 2014. To get the actual data, go to <a href="https://www.ncbi.nlm.nih.gov/geo/">NCBI</a> and search for the <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63525">GSE number</a>. In the section ‚ÄòSamples‚Äô the GSM numbers of all samples are given. Select the correct one for you, and go to the <a href="https://www.ebi.ac.uk/ena">European Nucleotide Archive</a> and enter the GSM number. Select a matching result e.g. <a href="https://www.ebi.ac.uk/ena/data/view/SRX764936">SRX764936</a> and download the data given in the row ‚ÄòFASTQ files (FTP)‚Äô the forward and reverse strand.
It is important to have the forward and reverse strand individual as a FASTQ file and to map it individually, HiCExplorer can not work with interleaved files.</p>
  </blockquote>

</blockquote>

<h1 id="reads-mapping"><span class="notranslate">Reads</span> <span class="notranslate">mapping</span></h1>

<p>Mates have to be mapped individually to avoid mapper specific heuristics designed for standard paired-end libraries.</p>

<p>We have used the HiCExplorer successfully with bwa, bowtie2 and hisat2. In this tutorial we will be using <strong>Map with BWA-MEM</strong> tool. It is important to remember to:</p>
<ul>
  <li>use local <span class="notranslate">mapping</span>, in contrast to end-to-end. A fraction of Hi-C <span class="notranslate">reads</span> are chimeric and will not map end-to-end thus, local <span class="notranslate">mapping</span> is important to increase the number of mapped <span class="notranslate">reads</span></li>
  <li>tune the aligner parameters to penalize deletions and insertions. This is important to avoid aligned <span class="notranslate">reads</span> with gaps if they happen to be chimeric.</li>
  <li>If bowtie2 or hisat2 are used, <code class="language-plaintext highlighter-rouge">--reorder</code> option and as a file format <code class="language-plaintext highlighter-rouge">bam_native</code> needs to be used. Regular <code class="language-plaintext highlighter-rouge">bam</code> files are sorted by <span class="notranslate">Galaxy</span> and can not be used as an input for HiCExplorer.</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span> <span class="notranslate">reads</span></h3>

  <ol>
    <li><strong>Map with Bowtie</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run Bowtie on both strands <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R1</code> and <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R2</code> with:
      <ul>
        <li>‚ÄúIs this single or paired library‚Äù to <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li>Set multiple data sets</li>
        <li>‚ÄúFASTQ file‚Äù to <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R1</code>and <code class="language-plaintext highlighter-rouge">HiC_S2_1p_10min_lowU_R2</code></li>
        <li>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù to <code class="language-plaintext highlighter-rouge">Use a built-in index</code></li>
        <li>‚ÄúSelect a reference genome‚Äù to <code class="language-plaintext highlighter-rouge">dm3</code></li>
        <li>‚ÄúDo you want to tweak SAM/BAM Options?‚Äù to <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>‚ÄúReorder output to reflect order of the input file‚Äù to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Rename the output of the tool according to the corresponding files: <code class="language-plaintext highlighter-rouge">R1.bam</code> and <code class="language-plaintext highlighter-rouge">R2.bam</code></li>
  </ol>

</blockquote>

<h1 id="creation-of-a-hi-c-matrix">Creation of a Hi-C matrix</h1>

<p>Once the <span class="notranslate">reads</span> have been mapped the Hi-C matrix can be built.</p>

<p>For this step we will use <a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicBuildMatrix.html#hicbuildmatrix">hicBuildMatrix</a> tool, which builds the matrix of read counts over the bins in the genome, considering the sites around the given restriction site.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hicbuildmatrix"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: hicBuildMatrix</h3>

  <ol>
    <li><strong>hicBuildMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicBuildMatrix on the <code class="language-plaintext highlighter-rouge">R1.bam</code> and <code class="language-plaintext highlighter-rouge">R2.bam</code> from previous step with modifying the following parameters:
      <ul>
        <li>‚Äú1: Sam/Bam files to process‚Äù to <code class="language-plaintext highlighter-rouge">R1.bam</code></li>
        <li>‚Äú2: Sam/Bam files to process‚Äù to <code class="language-plaintext highlighter-rouge">R2.bam</code></li>
        <li>‚ÄúChoose to use a restriction cut file or a bin size‚Äù to <code class="language-plaintext highlighter-rouge">Bin size</code></li>
        <li>‚ÄúBin size in bp‚Äù to <code class="language-plaintext highlighter-rouge">10000</code></li>
        <li>
          <p>‚ÄúSequence of the restriction site‚Äù to <code class="language-plaintext highlighter-rouge">GATC</code></p>

          <blockquote class="comment">
            <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

            <p><em>hicBuildMatrix</em> creates two files, a bam file containing only the valid Hi-C read pairs and a matrix containing the Hi-C contacts at the given resolution. The bam file is useful to check the quality of the Hi-C library on the genome browser. A good Hi-C library should contain piles of <span class="notranslate">reads</span> near the restriction fragment sites. In the QC folder a html file is saved with plots containing useful information for the quality control of the Hi-C sample like the number of valid pairs, duplicated pairs, self-ligations etc. Usually, only 25%-40% of the <span class="notranslate">reads</span> are valid and used to build the Hi-C matrix mostly because of the <span class="notranslate">reads</span> that are on repetitive regions that need to be discarded.</p>
          </blockquote>

          <blockquote class="comment">
            <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

            <p>Normally 25% of the total <span class="notranslate">reads</span> are selected. The output matrices have counts for the genomic regions. The extension of output matrix files is .h5.
A quality report is created in e.g. <code class="language-plaintext highlighter-rouge">hicMatrix/R1_10kb_QC</code>, have a look at the report hicQC.html.</p>
          </blockquote>
        </li>
      </ul>
    </li>
    <li>Rename the output to <code class="language-plaintext highlighter-rouge">10 kb contact matrix</code>.</li>
  </ol>

</blockquote>

<h1 id="plotting-the-hi-c-matrix">Plotting the Hi-C matrix</h1>

<p>A 10kb bin matrix is too large to plot, it‚Äôs better to reduce the resolution. We usually run out of memory for a 1 kb or a 10 kb bin matrix and the time to plot it is very long (minutes instead of seconds). In order to reduce the resolution we use the tool <a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicMergeMatrixBins.html#hicmergematrixbins">hicMergeMatrixBins</a>.</p>

<p><a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicMergeMatrixBins.html#hicmergematrixbins">hicMergeMatrixBins</a> merges the bins into larger bins of given number (specified by ‚ÄìnumBins). We will merge 100 bins in the original (uncorrected) matrix and then correct it. The new bin size is going to be 10.000 bp * 100 = 1.000.000 bp = 1 Mb</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hicmergematrixbins"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: hicMergeMatrixBins</h3>

  <ol>
    <li><strong>hicMergeMatrixBins</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicMergeMatrixBins on the output from previous step setting the following parameter:
      <ul>
        <li>‚ÄúNumber of bins to merge‚Äù to <code class="language-plaintext highlighter-rouge">100</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output to <code class="language-plaintext highlighter-rouge">1 MB contact matrix</code>.</p>
    </li>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on the output from hicMergeMatrixBins <code class="language-plaintext highlighter-rouge">1 MB contact matrix</code> adjusting the parameters:
      <ul>
        <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">Hi-C matrix for dm3</code></li>
        <li>‚ÄúRemove masked bins from the matrix‚Äù to <code class="language-plaintext highlighter-rouge">True</code></li>
        <li>‚ÄúPlot the log1p of the matrix values: <code class="language-plaintext highlighter-rouge">True</code></li>
        <li>‚ÄúChromosomes to include (and order to plot in)‚Äù to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr2R</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr3L</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr3R</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-tip-log1p"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: log1p</h3>

        <p>Because of the large differences in counts found in the matrix, it is better to plot the counts using the <em>‚Äìlog1p</em> option.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The resulting plot of the 1 Mb contact matrix should look like:
<img src="../../images/plotMatrix.png" alt="TAD plot" /></p>

<h1 id="correction-of-hi-c-matrix">Correction of Hi-C matrix</h1>

<p><a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicCorrectMatrix.html#hiccorrectmatrix">hicCorrectMatrix</a> corrects the matrix counts in an iterative manner. For correcting the matrix, it‚Äôs important to remove the unassembled scaffolds (e.g. NT_) and keep only chromosomes, as scaffolds create problems with matrix correction. Therefore we use the chromosome names (chr2R, chr2L, chr3R, chr3L, chrX) here.</p>

<p>Matrix correction works in two steps: first a histogram containing the sum of contact per bin (row sum) is produced. This plot needs to be inspected to decide the best threshold for removing bins with lower number of <span class="notranslate">reads</span>. The second steps removes the low scoring bins and does the correction.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-matrix-diagnostic"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Matrix diagnostic</h3>

  <ol>
    <li><strong>hicCorrectMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicCorrectMatrix on the output from hicBuildMatrix <code class="language-plaintext highlighter-rouge">10 kb contact matrix</code> adjusting the parameters:
      <ul>
        <li>‚ÄúRange restriction (in bp)‚Äù to <code class="language-plaintext highlighter-rouge">Diagnostic plot</code></li>
        <li>‚ÄúChromosomes to include (and order to plot in)‚Äù to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr2R</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr3L</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr3R</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output of the program prints a threshold suggestion that is usually accurate but is better to revise the histogram plot. The threshold is visualized in the plot as a black vertical line.</p>

<p>In our case the distribution describes the counts per bin of a genomic distance. To remove all bins with a z-score threshold less / more than X means to remove all bins which have less / more counts than X of mean of their specific distribution in units of the standard deviation. Looking at the distribution, we can select the value of -1.6 (lower end) and 1.8 (upper end) to remove. This is given by the ‚ÄìfilterThreshold option in hicCorrectMatrix set to ‚Äòcorrect matrix‚Äô mode.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-matrix-correction"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Matrix correction</h3>

  <ol>
    <li><strong>hicCorrectMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicCorrectMatrix on the original matrix <code class="language-plaintext highlighter-rouge">10 kb contact matrix</code> adjusting the parameters:
      <ul>
        <li>‚ÄúRange restriction (in bp)‚Äù to <code class="language-plaintext highlighter-rouge">Correct matrix</code></li>
        <li>‚ÄúNormalize each chromosome separately‚Äù to <code class="language-plaintext highlighter-rouge">True</code></li>
        <li>‚ÄúRemove bins of low coverage‚Äù to <code class="language-plaintext highlighter-rouge">-1.6</code></li>
        <li>‚ÄúRemove bins of large coverage‚Äù to <code class="language-plaintext highlighter-rouge">1.8</code></li>
        <li>‚ÄúChromosomes to include (and order to plot in)‚Äù to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr2R</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr3L</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chr3R</code></li>
        <li>‚Äù+ Insert Chromosomes to include (and order to plot in):‚Äù to <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>
    </li>
    <li>Rename the corrected matrix to <code class="language-plaintext highlighter-rouge">10 kb corrected contact matrix</code>.</li>
  </ol>

</blockquote>

<p>It can happen that the correction stops with:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ERROR:iterative correction:*Error* matrix correction produced extremely large values.
This is often caused by bins of low counts. Use a more stringent filtering of bins.
</code></pre></div></div>

<p>This can be solved by a more stringent z-score values for the filter threshold or by a look at the plotted matrix. For example, chromosomes with 0 <span class="notranslate">reads</span> in its bins can be excluded from the correction by not defining it for the set of chromosomes that should be corrected (parameter ‚ÄòInclude chromosomes‚Äô).</p>

<h3 id="plotting-the-corrected-hi-c-matrix">Plotting the corrected Hi-C matrix</h3>

<p>We can now plot chromosome 2L with the corrected matrix.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plotting-the-corrected-hi-c-matrix"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plotting the corrected Hi-C matrix</h3>

  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on <code class="language-plaintext highlighter-rouge">10 kb corrected contact matrix</code> adjusting the parameters:
      <ul>
        <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">Hi-C matrix for dm3</code></li>
        <li>‚ÄúPlot per chromosome‚Äù to <code class="language-plaintext highlighter-rouge">False</code></li>
        <li>‚ÄúPlot only this region‚Äù to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>‚ÄúPlot the log1p of the matrix values‚Äù to <code class="language-plaintext highlighter-rouge">True</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="load-new-data">Load new data</h3>

<p>The steps so far would have led to long run times if real data would have been used. We therefore prepared a new matrix for you, <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code>. Please load it into your history.</p>

<h1 id="tad-calling">TAD calling</h1>

<p>‚ÄúThe partitioning of chromosomes into topologically associating domains (TADs) is an emerging concept that is reshaping our understanding of gene regulation in the context of physical organization of the genome‚Äù <a href="https://doi.org/10.1101/115063">Ramirez et al. 2017</a>.</p>

<p>TAD calling works in two steps: First HiCExplorer computes a TAD-separation score based on a z-score matrix for all bins. Then those bins having a local minimum of the TAD-separation score are evaluated with respect to the surrounding bins to assign a p-value. Then a cutoff is applied to select the bins more likely to be TAD boundaries.</p>

<p><a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicFindTADs.html#hicfindtads">hicFindTADs</a> tries to identify sensible parameters but those can be change to identify more stringent set of boundaries.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-finding-tads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Finding TADs</h3>

  <ol>
    <li><strong>hicFindTADs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicFindTADs on <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code> adjusting the parameters:
      <ul>
        <li>‚ÄúMinimum window length (in bp) to be considered to the left and to the right of each Hi-C bin.‚Äù to <code class="language-plaintext highlighter-rouge">30000</code></li>
        <li>‚ÄúMaximum window length (in bp) to be considered to the left and to the right of each Hi-C bin.‚Äù to <code class="language-plaintext highlighter-rouge">100000</code></li>
        <li>‚ÄúStep size when moving from minDepth to maxDepth‚Äù to <code class="language-plaintext highlighter-rouge">10000</code></li>
        <li>‚ÄúMultiple Testing Corrections‚Äù to <code class="language-plaintext highlighter-rouge">False discovery rate</code></li>
        <li>‚Äúq-value‚Äù to <code class="language-plaintext highlighter-rouge">0.05</code></li>
        <li>‚ÄúMinimum threshold of the difference between the TAD-separation score of a putative boundary and the mean of the TAD-sep. score of surrounding bins.‚Äù to <code class="language-plaintext highlighter-rouge">0.001</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>As an output we get the boundaries, domains and scores separated files. We will use in the plot later only the TAD-score file.</p>

<h1 id="ab-compartments-computation">A/B compartments computation</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-computing-a--b-compartments"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Computing A / B compartments</h3>

  <ol>
    <li><strong>hicPCA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPCA adjusting the parameters:
      <ul>
        <li>‚ÄúMatrix to compute on‚Äù to <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code></li>
        <li>‚ÄúOutput file format‚Äù to <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li>‚ÄúReturn inte<span class="notranslate">rna</span>lly used Pearson matrix‚Äù to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plotting-the-pearson-matrix-and-pca-track"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plotting the pearson matrix and PCA track</h3>

  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on <code class="language-plaintext highlighter-rouge">pearson_matrix from PCA computation</code> adjusting the parameters:
      <ul>
        <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">Pearson matrix and PC1</code></li>
        <li>‚ÄúChromosomes to include‚Äù to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>‚ÄúColor map to use for the heatmap‚Äù to <code class="language-plaintext highlighter-rouge">gist_heat</code></li>
        <li>‚ÄúDatatype of eigenvector file‚Äù to <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li>‚ÄúEigenvector file‚Äù to <code class="language-plaintext highlighter-rouge">hicPCA on [...] PC1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix on <code class="language-plaintext highlighter-rouge">pearson_matrix from PCA computation</code> adjusting the parameters:
      <ul>
        <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">Pearson matrix and PC2</code></li>
        <li>‚ÄúChromosomes to include‚Äù to <code class="language-plaintext highlighter-rouge">chr2L</code></li>
        <li>‚ÄúColor map to use for the heatmap‚Äù to <code class="language-plaintext highlighter-rouge">gist_heat</code></li>
        <li>‚ÄúDatatype of eigenvector file‚Äù to <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li>‚ÄúEigenvector file‚Äù to <code class="language-plaintext highlighter-rouge">hicPCA on [...] PC2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/pearson_pc1.png" alt="Pearson PC1" /></p>

<p><img src="../../images/pearson_pc2.png" alt="Pearson PC2" /></p>

<p>The first principal component correlates with the chromosome arms, while the second component correlates with A/B compartments.</p>

<h1 id="integrating-hi-c-and-other-data">Integrating Hi-C and other data</h1>

<p>We can plot the TADs for a given chromosomal region. For this we will use <a href="http://hicexplorer.readthedocs.io/en/latest/content/tools/hicPlotTADs.html">pyGenomeTracks</a>.</p>

<p>For the next step we need additional data tracks. Please load <code class="language-plaintext highlighter-rouge">dm3_genes.<span class="notranslate">bed</span></code>, <code class="language-plaintext highlighter-rouge">H3K27me3.bw</code>, <code class="language-plaintext highlighter-rouge">H3K36me3.bw</code> and <code class="language-plaintext highlighter-rouge">H4K16ac.bw</code> to your history.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plotting-tads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plotting TADs</h3>

  <ol>
    <li><strong>pyGenomeTracks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run pyGenomeTracks adjusting the parameters:
      <ul>
        <li>‚ÄúRegion of the genome to limit the operation‚Äù to <code class="language-plaintext highlighter-rouge">chr2L:14500000-16500000</code></li>
        <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">TAD visualization</code>
          <ul>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">HiC dm3 chr2L:14500000-16500000</code></li>
            <li>‚ÄúMatrix to compute on.‚Äù to the corrected matrix from hicCorrectMatrix step</li>
            <li>‚ÄúDepth‚Äù to <code class="language-plaintext highlighter-rouge">750000</code></li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">4</code></li>
            <li>‚ÄúBoundaries file‚Äù to <code class="language-plaintext highlighter-rouge">hicFindTads on data XX: TAD domains</code></li>
            <li>‚ÄúShow x labels‚Äù to <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge"><span class="notranslate">Bed</span>graph matrix track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">TAD separation score</code></li>
            <li>‚ÄúTrack file <span class="notranslate">bed</span>graph format‚Äù to <code class="language-plaintext highlighter-rouge">hicFindTads on data XX: TAD information in bm file</code></li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">4</code></li>
            <li>‚ÄúSet type to lines‚Äù to ‚ÄòTrue‚Äô</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">PC1</code></li>
            <li>‚ÄúTrack file bigwig format‚Äù the first computed <code class="language-plaintext highlighter-rouge">hicPCA</code> result</li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>‚ÄúColor of track‚Äù to a color of your choice</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">PC2</code></li>
            <li>‚ÄúTrack file bigwig format‚Äù the second computed <code class="language-plaintext highlighter-rouge">hicPCA</code> result</li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>‚ÄúColor of track‚Äù to a color of your choice</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">H3K36me3</code></li>
            <li>‚ÄúTrack file bigwig format‚Äù to <code class="language-plaintext highlighter-rouge">H3K36me3</code></li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>‚ÄúColor of track‚Äù to a color of your choice</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">H3K27me3</code></li>
            <li>‚ÄúTrack file bigwig format‚Äù to <code class="language-plaintext highlighter-rouge">H3K27me3</code></li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>‚ÄúColor of track‚Äù to a color of your choice</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Bigwig track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">H4K16ac</code></li>
            <li>‚ÄúTrack file bigwig format‚Äù to <code class="language-plaintext highlighter-rouge">H4K16ac</code></li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">1.5</code></li>
            <li>‚ÄúColor of track‚Äù to a color of your choice</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Gene track</code></li>
            <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">dm3 genes</code></li>
            <li>‚ÄúTrack file <span class="notranslate">bed</span>graph format‚Äù the imported .<span class="notranslate">bed</span> file</li>
            <li>‚ÄúHeight‚Äù to <code class="language-plaintext highlighter-rouge">3</code></li>
            <li>‚ÄúType‚Äù to <code class="language-plaintext highlighter-rouge">genes</code></li>
            <li>‚ÄúGene rows‚Äù to <code class="language-plaintext highlighter-rouge">15</code></li>
            <li>‚ÄúColor of track‚Äù to a color of your choice</li>
          </ul>
        </li>
        <li>‚Äú+Insert Include tracks in your plot‚Äù
          <ul>
            <li>‚ÄúChoose style of the track‚Äù to <code class="language-plaintext highlighter-rouge">Vlines track</code></li>
            <li>‚ÄúTrack file <span class="notranslate">bed</span> format‚Äù to <code class="language-plaintext highlighter-rouge">hicFindTads on data XX: TAD domains</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The resulting image should look like this one:
<img src="../../images/plotTADs.png" alt="TAD plot" /></p>

<h1 id="loop-detection">Loop detection</h1>

<p>In Hi-C data, the term <code class="language-plaintext highlighter-rouge">loop</code> refers to a 3D structure which represents enhancer-promoter, gene, architectural or polycomb-mediated interactions. These interactions have the characteristics to be enriched in a single region compared to the local background. These loops are also called long-range interactions with an expected maximum distance of 2 MB (see <a href="https://doi.org/10.1016/j.cell.2014.11.021">Rao et al. 2014</a>).</p>

<p><img src="../../images/loops_bonev_cavalli.png" alt="Loops visualization" /></p>

<p>To compute loops, we have to import a new data set from the shared library to our history: <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code> or download it via the <a href="ftp://cooler.csail.mit.edu/coolers/hg19/Rao2014-GM12878-MboI-allreps-filtered.10kb.cool">FTP cool files server</a>. (FTP links seems to be blocked, therefore the full URL: <code class="language-plaintext highlighter-rouge">ftp://cooler.csail.mit.edu/coolers/hg19/Rao2014-GM12878-MboI-allreps-filtered.10kb.cool</code>)</p>

<p>This dataset is from the human cell GM12878, mapped to hg19 and of 10 kb resolution. We use a new file because to detect loop structures the read coverage is required to be in the hunderts of million; this was not the case for the previous used drosophila dataset.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-matrix-information"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Matrix information</h3>

  <ol>
    <li><strong>hicInfo</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicInfo adjusting the parameters:
      <ul>
        <li>‚ÄúSelect‚Äù <code class="language-plaintext highlighter-rouge">Multiple datasets</code></li>
        <li>‚ÄúMatrix to compute on‚Äù to <code class="language-plaintext highlighter-rouge">corrected contact matrix dm3 large</code> and <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>We now investigate the result of hicInfo and see that the new imported file is having 1.2 billion non-zero elements, while the drosophila Hi-C interaction matrix has around 12 million non-zero elements.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-computing-loops"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Computing loops</h3>

  <ol>
    <li><strong>hicDetectLoops</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicDetectLoops adjusting the parameters:
      <ul>
        <li>‚ÄúMatrix to compute on‚Äù to <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code></li>
        <li>‚ÄúPeak width‚Äù to <code class="language-plaintext highlighter-rouge">6</code></li>
        <li>‚ÄúWindow size‚Äù to <code class="language-plaintext highlighter-rouge">10</code></li>
        <li>‚ÄúP-value preselection‚Äù to <code class="language-plaintext highlighter-rouge">0.01</code></li>
        <li>‚ÄúP-value‚Äù to <code class="language-plaintext highlighter-rouge">0.01</code></li>
        <li>‚ÄúChromosomes to include‚Äù to <code class="language-plaintext highlighter-rouge">chr1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The detection of the loops is based on a pre-selection of interactions, a p-value given a continuous negative binomial distribution over all interactions of a relative distance is computed. All interactions are filtered with a threshold (<code class="language-plaintext highlighter-rouge">p-value preselection</code>) to retrieve loop candidates. In a second step, the selected peak candidate is compared against its background using a Wilcoxon rank-sum test.</p>

<p>As an output we get a loop file containing the positions of both anchor points of the loop and the p-value of the used statistical test.</p>

<p><img src="../../images/loops_result.png" alt="Loops computed_file" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plotting-of-loops"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plotting of loops</h3>

  <ol>
    <li><strong>hicPlotMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run hicPlotMatrix adjusting the parameters:
      <ul>
        <li>‚ÄúMatrix to compute on‚Äù to <code class="language-plaintext highlighter-rouge">GM12878-MboI-allreps-filtered.10kb.cool</code></li>
        <li>‚ÄúPlot title‚Äù to <code class="language-plaintext highlighter-rouge">Loops</code></li>
        <li>‚ÄúPlot only this region‚Äù to <code class="language-plaintext highlighter-rouge">chr1:18000000-22000000</code></li>
        <li>‚ÄúPlot the log1p of the matrix values‚Äù to <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>‚ÄúAdd detected loops‚Äù to <code class="language-plaintext highlighter-rouge">Computed loops</code> of the previous step</li>
        <li>‚ÄúDPI for image‚Äù to <code class="language-plaintext highlighter-rouge">300</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/loops_plot.png" alt="Loops result_plot" /></p>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial we used HiCExplorer to analyze drosophila melanogaster cells. We mapped chimeric <span class="notranslate">reads</span> and created a contact matrix, to reduce noise this contact matrix was normalized. We showed how to visualize a contact matrix and how we can investigate topological associating domains and relate them to additional data like gene tracks. Moreover, we used a human Hi-C interaction matrix to compute loop structures.</p>

<p>To improve your learned skills we offer an additional tutorial based on mouse stem cells: <a href="http://hicexplorer.readthedocs.io/en/latest/content/example_usage.html">following work</a>.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Hi-C helps to investigate the 3D structure of the genome and to uncover folding principles of chromatin.</p>
</li>
                            
                            <li><p>In order to build a Hi-C contact matrix the reads have to be mapped to the reference genome.</p>
</li>
                            
                            <li><p>Based on a contact matrix open and closed chromatin (A/B compartments) and topological associating domains (TADs) can be computed.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Epigenetics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Joachim Wolff, Fidel Ramirez, Vivek Bhardwaj, Ekaterina Polkh, 2021 <b>Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Galaxy Training Materials)</b>. <a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{epigenetics-hicexplorer,
    author = "Joachim Wolff and Fidel Ramirez and Vivek Bhardwaj and Ekaterina Polkh",
    title = "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Joachim Wolff, Fidel Ramirez, Vivek Bhardwaj, Ekaterina Polkh)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/hicexplorer/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
