<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Visualization of RNA-Seq results with CummeRbund</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Visualization of RNA-Seq results with CummeRbund" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-viz-with-cummerbund%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-viz-with-cummerbund%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-viz-with-cummerbund%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-viz-with-cummerbund%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-viz-with-cummerbund%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frna-seq-viz-with-cummerbund%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Visualization of RNA-Seq results with CummeRbund",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / rna-seq-viz-with-cummerbund / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Visualization of RNA-Seq results with CummeRbund' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - How are RNA-Seq results stored?\n - Why are visualization techniques needed?\n - How to select genes for visualizing meaningful results of differential gene expression analysis?\n\n\\nThe objectives are:\n - Manage RNA-Seq results\n - Extract genes for producing differential gene expression analysis visualizations\n - Visualize meaningful information\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Andrea Bagnacani"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Andrea Bagnacani"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Visualization of RNA-Seq results with CummeRbund</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/bagnacan/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bagnacan" alt="Andrea Bagnacani">Andrea Bagnacani</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How are RNA-Seq results stored?</p>
</li>
            
            <li><p>Why are visualization techniques needed?</p>
</li>
            
            <li><p>How to select genes for visualizing meaningful results of differential gene expression analysis?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Manage RNA-Seq results</p>
</li>
            
            <li><p>Extract genes for producing differential gene expression analysis visualizations</p>
</li>
            
            <li><p>Visualize meaningful information</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/1001880">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p><span class="notranslate">RNA</span>-Seq analysis helps researchers annotate new genes and splice variants, and provides cell- and context-specific quantification of gene <span class="notranslate">expression</span>. <span class="notranslate">RNA</span>-Seq data, however, are complex and require both computer science and mathematical knowledge to be managed and interpreted.</p>

<p>Visualization techniques are key to overcome the complexity of <span class="notranslate">RNA</span>-Seq data, and represent valuable tools to gather information and insights.</p>

<p>In this tutorial we will visualize <span class="notranslate">RNA</span>-seq data from the <a href="http://cole-trapnell-lab.github.io/cufflinks/cuffdiff/">CuffDiff</a> tool.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#reasons-for-visualizing-rna-seq-results" id="markdown-toc-reasons-for-visualizing-rna-seq-results">Reasons for visualizing <span class="notranslate">RNA</span>-Seq results</a></li>
  <li><a href="#importing-rna-seq-result-data" id="markdown-toc-importing-rna-seq-result-data">Importing <span class="notranslate">RNA</span>-Seq result data</a></li>
  <li><a href="#filtering-and-sorting" id="markdown-toc-filtering-and-sorting">Filtering and sorting</a></li>
  <li><a href="#cummerbund" id="markdown-toc-cummerbund">CummeRbund</a></li>
</ol>

</blockquote>

<h1 id="reasons-for-visualizing-rna-seq-results">Reasons for visualizing <span class="notranslate">RNA</span>-Seq results</h1>

<p>To make sense of the available <span class="notranslate">RNA</span>-Seq data, and overview the condition-specific gene <span class="notranslate">expression</span> levels of the provided samples, we need to visualize our results. Here we will use <a href="http://compbio.mit.edu/cummeRbund/">CummeRbund</a>.</p>

<p>CummeRbund is an open-source tool that simplifies the analysis of a CuffDiff <span class="notranslate">RNA</span>-Seq output. In particular, it helps researchers with:</p>
<ul>
  <li>managing, integrating, and visualizing the data produced by CuffDiff</li>
  <li>simplifying data exploration</li>
  <li>providing a bird‚Äôs-eye view of the <span class="notranslate">expression</span> analysis by describing relationships betweeen genes, transcripts, transcription start sites, and protein-coding regions</li>
  <li>exploring subfeatures of individual genes or gene-sets</li>
  <li>creating publication-ready plots</li>
</ul>

<p>A typical <span class="notranslate">workflow</span> for the visualization of <span class="notranslate">RNA</span>-Seq data involving CummeRbund:</p>

<p><img src="../../images/cummerbund-rna-seq-viz-with-cummerbund.png" alt="workflow" /></p>

<p>CummeRbund <span class="notranslate">reads</span> your <span class="notranslate">RNA</span>-Seq results from a <a href="https://www.sqlite.org/">SQLite</a> database. This database has to be created using CuffDiff‚Äôs SQLite output option.</p>

<blockquote class="details">
  <h3 id="details-sqlite-output-with-cuffdiff"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> SQLite output with CuffDiff</h3>

  <p>Instruct CuffDiff to organize its output in a SQLite database to be read CummeRbund.</p>

  <p><img src="../../images/cummerbund-cuffdiff-set-sqlite.png" alt="SQLite output" /></p>
</blockquote>

<h1 id="importing-rna-seq-result-data">Importing <span class="notranslate">RNA</span>-Seq result data</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import the <a href="https://zenodo.org/record/1001880/files/CuffDiff_SQLite_database.sqlite"><code class="language-plaintext highlighter-rouge">CuffDiff SQLite</code></a> dataset</p>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <p>Rename the dataset to ‚Äú<span class="notranslate">RNA</span>-Seq SQLite result data‚Äù</p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field  to <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>-Seq SQLite result data</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

  <p>By default, when data is imported via its link, <span class="notranslate">Galaxy</span> names it with its URL.</p>
</blockquote>

<p>CuffDiff‚Äôs output data is organized in a SQLite database, so we need to extract it to be able to see what it looks like.</p>

<p>For this tutorial, we are interested in CuffDiff‚Äôs tested transcripts for differential <span class="notranslate">expression</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-cuffdiff-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract CuffDiff results</h3>

  <ol>
    <li><strong>Extract CuffDiff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúSelect tables to output‚Äù</em>: <code class="language-plaintext highlighter-rouge">Transcript differential <span class="notranslate">expression</span> testing</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the table</p>

      <blockquote class="tip">
        <h3 id="tip-tip-inspecting-the-content-of-a-file-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Inspecting the content of a file in <span class="notranslate">Galaxy</span></h3>

        <ul>
          <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon (‚ÄúView data‚Äù) on the right of the file name in the history</li>
          <li>Inspect the content of the file</li>
        </ul>
      </blockquote>
    </li>
  </ol>

  <p>Each entry is a differentially expressed gene, which is described in terms of the following attributes.</p>

  <ul>
    <li><em>test_id</em>: A unique identifier describing the transcript being tested</li>
    <li><em>gene_id</em>: The identifier of the gene being tested</li>
    <li><em>gene</em>: The name of the gene being tested</li>
    <li><em>locus</em>: The genomic coordinates of the gene transcript being tested</li>
    <li><em>sample_1</em>: Label of the 1st sample</li>
    <li><em>sample_2</em>: Label of the 2nd sample</li>
    <li><em>status</em>: The test‚Äôs status:
      <ul>
        <li>‚ÄúOK‚Äù if test successful</li>
        <li>‚ÄúNOTEST‚Äù if not enough alignments for testing</li>
        <li>‚ÄúLOWDATA‚Äù if too complex or shallowly sequenced</li>
        <li>‚ÄúHIDATA‚Äù if too many fragments in locus</li>
        <li>‚ÄúFAIL‚Äù if a numerical exception prevented testing</li>
      </ul>
    </li>
    <li><em>value_1</em>: The gene‚Äôs FPKM in sample_1</li>
    <li><em>value_2</em>: The gene‚Äôs FPKM in sample_2</li>
    <li><em>log2(fold_change)</em>: The log<sub>2</sub> of the fold change (sample_1/sample_2). This reports the <span class="notranslate">expression</span> difference between condition one (sample_1) and condition two (sample_2)</li>
    <li><em>test_stat</em>: The test statistic‚Äôs value, used to determine the significance of the observed change in FPKM</li>
    <li><em>p_value</em>: The uncorrected p-value of the test statistic</li>
    <li><em>q_value</em>: The False-discovery-rate-adjusted p-value of the test statistic</li>
    <li><em>significant</em>: ‚Äúyes‚Äù or ‚Äúno‚Äù, depending on whether the p_value is greater then the FDR after Benjamini-Hochberg correction for multiple-testing. This tells whether the difference between the <span class="notranslate">expression</span> levels in condition one (sample_1) and condition two (sample_2) is significant</li>
  </ul>

  <p>We want to keep only the <em>significant differentially expressed</em> genes.</p>

  <blockquote class="question">
    <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>How to retain only the significant differentially expressed genes?</li>
      <li>Which column stores this information?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>We need to filter on the column storing the record‚Äôs significance</li>
        <li>Column 14</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<h1 id="filtering-and-sorting">Filtering and sorting</h1>

<p>We now want to highlight the transcripts whose <span class="notranslate">expression</span> difference, the log<sub>2</sub>(fold_change), has been statistically assessed as both high and significant.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-cuffdiffs-significant-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract CuffDiff‚Äôs significant differentially expressed genes</h3>

  <ol>
    <li><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúFilter‚Äù</em>: the extracted table from the previous step</li>
        <li><em>‚ÄúWith following condition‚Äù</em>: an appropriate filter over the target column (see questions below when in doubt)</li>
        <li><em>‚ÄúNumber of header lines to skip‚Äù</em>: the number of rows used for the table‚Äôs header</li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
        <ol>
          <li>What column stores the information of significance for each record?</li>
          <li>Which conditional <span class="notranslate">expression</span> has to be set to filter all records on the selected column?</li>
          <li>How many rows are allocated for the table‚Äôs header?</li>
          <li>How many entries were originally stored in the table? And how many after the filtering operation?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>Column 14</li>
            <li>c14==‚Äôyes‚Äô</li>
            <li>1</li>
            <li>~140,000 (before filtering) vs. 219 (after filtering)</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

  <p>Review the meaning of each column, and look at your data.</p>

  <ul>
    <li>The differential <span class="notranslate">expression</span> values are stored in column 10</li>
    <li>The statistical score, assessing the differential <span class="notranslate">expression</span> significance, is stored in column 13
 We will sort all records on the basis of their Q-score (column 13) and log<sub>2</sub>(fold_change).</li>
  </ul>

  <ol>
    <li><strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters
      <ul>
        <li><em>‚ÄúSort Dataset‚Äù</em>: the filtered table</li>
        <li><em>‚Äúon column‚Äù</em>: <code class="language-plaintext highlighter-rouge">13</code></li>
        <li><em>‚Äúwith flavor‚Äù</em>: <code class="language-plaintext highlighter-rouge">Numerical sort</code></li>
        <li><em>‚Äúeverything in‚Äù</em>: <code class="language-plaintext highlighter-rouge">Ascending order</code></li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Column selection</strong>, and parameterize the Sort tool to sort on column 10. Be careful of the sorting order!</li>
        <li>Are there any rows allocated for the table‚Äôs header? In that case, set ‚ÄúNumber of header lines to skip‚Äù accordingly!</li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
        <ol>
          <li>Which gene transcript has been statistically assessed as both high and significant?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>1 LIMCH1</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="cummerbund">CummeRbund</h1>

<p>CummeRbund generates two outputs:</p>
<ul>
  <li>The plot, which visualizes our <span class="notranslate">RNA</span>-Seq results of interest</li>
  <li>The <a href="http://had.co.nz/ggplot2/">ggplot</a> object responsible for generating the plot</li>
</ul>

<p>In this section we will parametrize CummeRbund to create different kinds of plots from our input data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization</h3>

  <ol>
    <li><strong>CummeRbund</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span>  <strong>Insert plots</strong>
          <ul>
            <li><em>‚ÄúWidth‚Äù</em>: <code class="language-plaintext highlighter-rouge">800</code></li>
            <li><em>‚ÄúHeight‚Äù</em>: <code class="language-plaintext highlighter-rouge">600</code></li>
            <li><em>‚ÄúPlot type‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Expression</span> Plot</code>
              <ul>
                <li><em>‚Äú<span class="notranslate">Expression</span> levels to plot‚Äù</em>:<code class="language-plaintext highlighter-rouge">Isoforms</code></li>
                <li><em>‚ÄúGene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">NDUFV1</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The input data used to create the visualization comprise 3 conditions: <em>hits7</em> (Patient 1), <em>hits8</em> (Patient 2), and <em>hits9</em> (Control).</p>

<p>Our first CummeRbund plot is the ‚Äú<span class="notranslate">Expression</span> Plot‚Äù of the isoforms of gene NDUFV1, which shows the <span class="notranslate">expression</span> differences of isoforms NM_001166102 and NM_007103 among the three conditions. Error bars capture the variability of the distribution of FPKM values: the broader the distribution of FPKM values, the larger the corresponding error bar.</p>

<p><img src="../../images/cummerbund-expression-plot.png" alt="Expression plot" /></p>

<p>Our plot has a modest number of isoforms, and is therefore easy to read. However, with a high number of isoforms and <span class="notranslate">expression</span> variability among different conditions, the plot can look very busy. We can therefore change the visualization type by selecting another type of plot. Let‚Äôs change visualization.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization</h3>

  <ol>
    <li><strong>CummeRbund</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> Click on <em>‚ÄúInsert plots‚Äù</em>
          <ul>
            <li><em>‚ÄúWidth‚Äù</em>: <code class="language-plaintext highlighter-rouge">800</code></li>
            <li><em>‚ÄúHeight‚Äù</em>: <code class="language-plaintext highlighter-rouge">600</code></li>
            <li><em>‚ÄúPlot type‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Expression</span> Bar Plot</code>
              <ul>
                <li><em>‚Äú<span class="notranslate">Expression</span> levels to plot‚Äù</em>:<code class="language-plaintext highlighter-rouge">Isoforms</code></li>
                <li><em>‚ÄúGene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">NDUFV1</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/cummerbund-expression-bar-plot.png" alt="Expression bar plot" /></p>

<p>The <span class="notranslate">Expression</span> Bar Plot of gene NDUFV1‚Äôs replicates NM_001166102 and NM_007103, shows the <span class="notranslate">expression</span> changes across the three aforementioned conditions.</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>These plots are shown also in <a href="https://vimeo.com/channels/884356/128265982">this</a> <span class="notranslate">Galaxy</span> video tutorial.</p>
</blockquote>

<p>Let‚Äôs now create a heatmap to plot the <span class="notranslate">expression</span> levels of the significant differentially expressed gene isoforms obtained from our filter and sort operations.
As a showcase example, let‚Äôs consider only the top 5 differentially expressed genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization</h3>

  <ol>
    <li><strong>CummeRbund</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span>  <strong>Insert plots</strong>
          <ul>
            <li><em>‚ÄúWidth‚Äù</em>: <code class="language-plaintext highlighter-rouge">800</code></li>
            <li><em>‚ÄúHeight‚Äù</em>: <code class="language-plaintext highlighter-rouge">600</code></li>
            <li><em>‚ÄúPlot type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Heatmap</code>
              <ul>
                <li><em>‚Äú<span class="notranslate">Expression</span> levels to plot‚Äù</em>: <code class="language-plaintext highlighter-rouge">Isoforms</code></li>
                <li><em>‚ÄúGene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">LIMCH1</code></li>
                <li><em>‚ÄúGene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">IFNL2</code></li>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Genes</strong>
                  <ul>
                    <li><em>‚ÄúGene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">CXCL11</code></li>
                  </ul>
                </li>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Genes</strong>
                  <ul>
                    <li><em>‚ÄúGene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">NUB1</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúCluster by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Both</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/cummerbund-heatmap.png" alt="Expression bar plot" /></p>

<p>Heatmap of significant differentially expressed isoforms of genes LIMCH1, IFNL2, CXCL11, NUB1. Differences in up- and down-regulation of certain gene isoforms across Patient 1 (hits7), Patient 2 (hits8), and Control (hits9), are visible in the lower and upper part of the heatmap.</p>

<blockquote class="comment">
  <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>For more sophisticated visualizations of your <span class="notranslate">RNA</span>-Seq analysis results, try selecting different CummeRbund plot options and parametrizations. Have a look also at CummeRbund‚Äôs <a href="http://compbio.mit.edu/cummeRbund/manual_2_0.html">manual</a>. Alte<span class="notranslate">rna</span>tively, you can modify a plot‚Äôs style by changing CummeRbund‚Äôs R output! CummeRbund‚Äôs R outputs are <em>ggplot</em> objects. Look <a href="https://github.com/tidyverse/ggplot2">here</a> to learn how to change fonts, colors, error bars, and more.</p>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Visualization tools help researchers making sense of data, providing a bird‚Äôs-eye view of the underlying analysis results.
In this tutorial we overviewed the advantages of visualizing <span class="notranslate">RNA</span>-Seq results with CummeRbund, and gained insights on CuffDiff‚Äôs big-data output by plotting information relative to the most significant differentially expressed genes in our <span class="notranslate">RNA</span>-Seq analysis.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Extract information from a SQLite CuffDiff database</p>
</li>
                            
                            <li><p>Filter and sort results to highlight differential expressed genes of interest</p>
</li>
                            
                            <li><p>Generate publication-ready visualizations of RNA-Seq analysis results</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Visualization of RNA-Seq results with CummeRbund (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Andrea Bagnacani, 2021 <b>Visualization of RNA-Seq results with CummeRbund (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html">/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-rna-seq-viz-with-cummerbund,
    author = "Andrea Bagnacani",
    title = "Visualization of RNA-Seq results with CummeRbund (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Andrea Bagnacani)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
