<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Pre-processing of Single-Cell RNA Data</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Pre-processing of Single-Cell RNA Data" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/galaxy-single-cell'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-preprocessing%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-preprocessing%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-preprocessing%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-preprocessing%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-preprocessing%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-preprocessing%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Pre-processing of Single-Cell RNA Data",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / scrna-preprocessing / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Pre-processing of Single-Cell RNA Data' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html",
  "timeRequired": "PT3H",
  "keywords": "single-cell",
  "description": "The questions this  addresses are:\n - What is single-cell?\n - How do we process a batch?\n - How do we process multiple batches?\n - How do we check for cross-contamination?\n\n\\nThe objectives are:\n - Demultiplex FASTQ data via UMI-tools\n - Understand and validate the extraction of barcodes\n - Obtain an overview of general alignment and quantification techniques\n - Generate a count matrix for downstream single-cell RNA analysis\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Pre-processing of Single-Cell RNA Data</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/mtekman/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/mtekman" alt="Mehmet Tekman">Mehmet Tekman</a>, <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="B√©r√©nice Batut">B√©r√©nice Batut</a>, <a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/erxleben" alt="Anika Erxleben">Anika Erxleben</a>, <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/wm75" alt="Wolfgang Maier">Wolfgang Maier</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What is single-cell?</p>
</li>
            
            <li><p>How do we process a batch?</p>
</li>
            
            <li><p>How do we process multiple batches?</p>
</li>
            
            <li><p>How do we check for cross-contamination?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Demultiplex FASTQ data via UMI-tools</p>
</li>
            
            <li><p>Understand and validate the extraction of barcodes</p>
</li>
            
            <li><p>Obtain an overview of general alignment and quantification techniques</p>
</li>
            
            <li><p>Generate a count matrix for downstream single-cell RNA analysis</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/3253142">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/workflows/" title="Workflows" alt="Pre-processing of Single-Cell RNA Data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<h4 class="no_toc" id="why-do-single-cell-sequencing">Why do <span class="notranslate">Single Cell</span> sequencing?</h4>

<p><span class="notranslate">Single-cell</span> <span class="notranslate">RNA</span> (sc<span class="notranslate">RNA</span>) sequencing is the technological successor to classical ‚Äúbulk‚Äù <span class="notranslate">RNA</span>-seq, where samples are no longer defined at the tissue level but at the individual cell level. The bulk <span class="notranslate">RNA</span>-seq methods seen in <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">previous hands-on material</a> would give the average <span class="notranslate">expression</span> of genes in a sample, whilst overlooking the distinct <span class="notranslate">expression</span> profiles given by the cell sub-populations due to their heterogeneity.</p>

<p>The rise of sc<span class="notranslate">RNA</span> sequencing provides the means to explore this heterogeneity by examining samples at the individual cell level, enabling a greater understanding of the development and function of such samples, by the characteristics of their constituent cells.</p>

<p>Consider the heterogeneity of cells sampled from bone marrow, where hematopoietic stem cells can give rise to many different cell types within the same tissue:</p>

<figure id="figure-1"><img src="/training-material/topics/transcriptomics/images/scrna_hematopoiesis.png" alt="Cell Differentiation of Hematopoietic Stem cells" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Cell Differentiation of Hematopoietic Stem cells. Image Source: https://commons.wikimedia.org/wiki/File:Hematopoiesis_simple.svg</figcaption></figure>

<p>The genes expressed by these cells at different developmental time points can be subtle, but generally can be classified into discrete cell sub-populations or under statistical clustering methods such as PCA or tSNE. Cells in the same cluster exhibit similar profiles of differential <span class="notranslate">expression</span> in the same set of related genes, compared to cells in other clusters. By identifying significant genes in each cluster, cell types and cell developmental hierarchies can be inferred based on the proximity of these clusters to one another.</p>

<p>Other than cell development, there are many more factors that can shape the level of gene <span class="notranslate">expression</span> exhibited by a given cell. Intercellular cell-signalling can block or enhance specific transcripts, the total amount of transcripts of a cell increases with the cell-cycle, or the proximity of a cell within a tissue to nutrients or oxygen.</p>

<figure id="figure-2"><img src="/training-material/topics/transcriptomics/images/scrna_nbt3711.png" alt="Facets of Cellular Identity" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Revealing the vectors of cellular identity with <span class="notranslate">single-cell</span> genomics, Nature Biotechnology, 2016</figcaption></figure>

<p>This tutorial is in part inspired by aspects of the <a href="https://hemberg-lab.github.io/scRNA.seq.course/">Hemberg <span class="notranslate">workflow</span></a> at the Sanger institute, as well as the <a href="https://github.com/CGATOxford/UMI-tools">CGATOxford <span class="notranslate">workflow</span></a> which provides the <strong>UMI-tools</strong> suite that we make use of. The barcoding follows the <a href="https://doi.org/10.1186/s13059-016-0938-8">CEL-Seq2 protocol</a> mentioned in the <a href="/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html"><em>Understanding Barcodes</em></a> hands-on, and uses the same lane configuration as utilised by the <a href="https://www.ie-freiburg.mpg.de/gruen">Freiburg MPI Gr√ºn lab</a>.</p>

<h1 class="no_toc" id="analysis-strategy">Analysis Strategy</h1>

<p>Most sc<span class="notranslate">RNA</span> sequencing techniques use pooled-sequencing approaches to generate a higher throughput of data by performing amplification and sequencing upon multiple cells in the same ‚Äúpool‚Äù. From a bioinformatics standpoint, this means that the output FASTQ data from the sequencer is batch-specific and contains all the sequences from multiple cells, where one sample of cells is equal to one batch.</p>

<p>In this tutorial, we will perform pre-processing upon sc<span class="notranslate">RNA</span> FASTQ batch data to generate an <em>N</em>-by-<em>M</em>  count matrix of <em>N</em> cells and <em>M</em> genes, with each element indicating the level of <span class="notranslate">expression</span> of that gene in a particular cell.</p>

<p>This count matrix is crucial for performing the <span class="notranslate">downstream</span> analysis, where differential gene analysis is performed between cells in order to cluster them into groups denoting their cell type and lineage.</p>

<p>The tutorial is structured in two parts:</p>

<ol>
  <li><strong>Single-Batch Processing</strong></li>
  <li><strong>Multi-Batch Processing</strong></li>
</ol>

<figure id="figure-3"><img src="/training-material/topics/transcriptomics/images/scrna_workflow.svg" alt="Overview of workflow" /><figcaption><span class="figcaption-prefix">Figure 3:</span> An overview of the entire <span class="notranslate">workflow</span></figcaption></figure>

<p>The first part of this tutorial will deal with batches, and use example <em>FASTQ</em> data from a single batch, which we will perform <a href="#barcode-extraction">barcode extraction</a> and annotation upon. Alignment and quality control will also be performed, and we will see how to construct a rudimentary count matrix.</p>

<blockquote class="comment">
  <h3 id="comment-10x-datasets"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> 10x Datasets</h3>

  <p>10x Genomics datasets can be processed in a much easier way that is outlined in the <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html"><em>Pre-processing of 10X <span class="notranslate">Single-Cell</span> <span class="notranslate">RNA</span> Datasets</em></a> tutorial.</p>

  <p>However, much of the essential concepts for sc<span class="notranslate">RNA</span>-seq pre-processing are not explained, so it is a good idea to familiarise yourself with them in this tutorial.</p>

</blockquote>

<p>The second part of this tutorial will deal with merging several output count matrices from multiple single batches generated in the first portion. Here, a set of example count matrices are <a href="#merging-count-matrices">merged together</a> and quality control performed. This produces a final count matrix valid for <span class="notranslate">downstream</span> analysis.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#single-batch-processing" id="markdown-toc-single-batch-processing">Single-Batch Processing</a>    <ol>
      <li><a href="#barcode-extraction" id="markdown-toc-barcode-extraction">Barcode Extraction</a></li>
      <li><a href="#mapping--alignment" id="markdown-toc-mapping--alignment"><span class="notranslate">Mapping</span> / Alignment</a></li>
      <li><a href="#filtering" id="markdown-toc-filtering">Filtering</a></li>
      <li><a href="#quantification" id="markdown-toc-quantification">Quantification</a></li>
    </ol>
  </li>
  <li><a href="#multi-batch-processing" id="markdown-toc-multi-batch-processing">Multi-Batch Processing</a>    <ol>
      <li><a href="#merging-count-matrices" id="markdown-toc-merging-count-matrices">Merging Count Matrices</a></li>
      <li><a href="#guarding-against-cross-contamination" id="markdown-toc-guarding-against-cross-contamination">Guarding against Cross-Contamination</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="single-batch-processing">Single-Batch Processing</h1>

<h3 id="data-upload-and-organization">Data upload and organization</h3>

<p>In this tutorial we will be analysing sc<span class="notranslate">RNA</span>-seq data of bone marrow cells taken from a single C57 mouse by <em>Herman et al.</em> (<a href="https://doi.org/10.1038/nmeth.4662">2018</a>) and producing a count matrix that we can use for later analysis.</p>

<p>The size of sc<span class="notranslate">RNA</span> files (.fastq) are typically in the gigabyte range and are somewhat impractical for training purposes, so we will expedite the analysis by using a smaller subset of actual batch data. We will also be using <em>Mus Musculus</em> annotation data (.gtf) from the  <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Mus_musculus/">NCBI RefSeq</a> track, as well as a barcodes file (.tsv).</p>

<figure id="figure-4"><img src="/training-material/topics/transcriptomics/images/scrna_workflow_upper.png" alt="Workflow Upper" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Single batch processing</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload and organization</h3>

  <ol>
    <li>
      <p>Create a new history and rename it (e.g. sc<span class="notranslate">RNA</span>-seq single batch tutorial)</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import the subset FASTQ paired data from <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3253142/files/SRR5683689_1.subset.fastq
https://zenodo.org/record/3253142/files/SRR5683689_2.subset.fastq
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>
            <p>Click on <strong>Collection</strong> on the top</p>
          </li>
          <li>
            <p>Click on <strong>Collection Type</strong> and select <code class="language-plaintext highlighter-rouge">Paired</code></p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Change <strong>Genome</strong> to <code class="language-plaintext highlighter-rouge">GRCm38/mm10</code></p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li>
            <p>Click on <strong>Build</strong> when available</p>
          </li>
          <li>Ensure that the forward and reverse <span class="notranslate">reads</span> are set to <code class="language-plaintext highlighter-rouge">SRR5683689_1</code> and <code class="language-plaintext highlighter-rouge">SRR5683689_2</code>, respectively.
            <ul>
              <li>Click <strong>Swap</strong> otherwise</li>
            </ul>
          </li>
          <li>
            <p>Enter a name for the collection</p>

            <ul>
              <li>
                <p>A useful naming convention is to use <code class="language-plaintext highlighter-rouge">&lt;name&gt;_&lt;plate&gt;_&lt;batch&gt;</code> to preserve the sample names, sequencing plate number and batch number.</p>
              </li>
              <li>
                <p>Here we will write <code class="language-plaintext highlighter-rouge">C57_P1_B1</code></p>
              </li>
            </ul>
          </li>
          <li>Click on <strong>Create list</strong> (and wait a bit)</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>
    </li>
    <li>
      <p>Import the Gene Annotations and Barcodes from <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3253142/files/Mus_musculus.GRCm38.93.mm10.UCSC.ncbiRefSeq.gtf
https://zenodo.org/record/3253142/files/celseq_barcodes.192.tabular
</code></pre></div>      </div>
    </li>
    <li>
      <p>Set the datatype of the <code class="language-plaintext highlighter-rouge">celseq_barcodes.192.tabular</code> to <code class="language-plaintext highlighter-rouge">tabular</code></p>
    </li>
  </ol>

</blockquote>

<h2 id="barcode-extraction">Barcode Extraction</h2>

<blockquote class="comment">
  <h3 id="comment-note"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note</h3>

  <p>Before performing the barcode extraction process, it is recommended that you familiarise yourself with the concepts of designing cell barcodes as given by the <a href="/training-material/topics/transcriptomics/tutorials/scrna-plates-batches-barcodes/slides.html"><em>Plates, Batches, and Barcodes</em></a>, as well as the <a href="/training-material/topics/transcriptomics/tutorials/scrna-umis/tutorial.html"><em>Understanding Barcodes</em></a> hands-on material for an introduction into transcript barcodes.</p>

</blockquote>

<p>We will be demultiplexing our FASTQ batch data by performing barcode extraction whilst also making use of the provided barcodes file to filter for <em>specific</em> cell barcodes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-barcode-extraction"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Barcode Extraction</h3>

  <ol>
    <li><strong>UMI-tools extract</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúLibrary type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end Dataset Collection</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚Äú<span class="notranslate">Reads</span> in FASTQ format‚Äù</em>: <code class="language-plaintext highlighter-rouge">C57_P1_B1</code> (Our paired set)</li>
            <li><em>‚ÄúBarcode on both <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcode on first read only</code></li>
          </ul>
        </li>
        <li><em>‚ÄúUse Known Barcodes?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBarcode File‚Äù</em>: <code class="language-plaintext highlighter-rouge">celseq_barcodes.192.tabular</code> (Input dataset)</li>
          </ul>
        </li>
        <li><em>‚ÄúBarcode pattern for first read‚Äù</em>: <code class="language-plaintext highlighter-rouge">NNNNNNCCCCCC</code></li>
        <li><em>‚ÄúEnable quality filter?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Verifying that the desired UMI and cell barcodes have been extracted from the sequence of the Forward <span class="notranslate">reads</span> and inserted into the header of the Reverse <span class="notranslate">reads</span> is encouraged, using the method outlined in the above hands-on material.</p>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>How many <span class="notranslate">reads</span> were filtered out, and why?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The input FASTQ files contained <span class="notranslate">reads</span> from all barcodes, including those with sequencing errors, resulting in a larger pool of detected barcodes than those desired. (e.g. Cell barcode <code class="language-plaintext highlighter-rouge">AAATTT</code> could have single base-pair sequencing errors that could modify it into <code class="language-plaintext highlighter-rouge">ATATTT</code> or <code class="language-plaintext highlighter-rouge">AAACTT</code>, etc).</p>

    <p>This information is included in the Log file of <strong>UMI-tools extract</strong> which contains all the parameters used to run, as well as <em>INFO</em> lines that indicate how many <span class="notranslate">reads</span> were read, and how many output. In this case: 134431 <span class="notranslate">reads</span> were retained (&gt;90% of input <span class="notranslate">reads</span>).</p>
  </blockquote>
</blockquote>

<h2 id="mapping--alignment"><span class="notranslate">Mapping</span> / Alignment</h2>

<p>FASTQ files contain sequence information that we wish to map to genes in a genome. <span class="notranslate">Mapping</span> is a relatively straightforward process, and is covered more extensively in the <a href="/training-material/topics/sequence-analysis/"><em>Sequence Analysis</em></a> tutorials:</p>

<ol>
  <li>Select your genome</li>
  <li>Select your gene annotation file</li>
  <li>Run the alignment program</li>
  <li>(Optional) Run MultiQC to assess the quality of the <span class="notranslate">mapping</span></li>
</ol>

<p>The FASTQ data was sequenced from mouse data, so to perform the alignment we will need to gather all data relevant to that genome. We will use the latest version (mm10).</p>

<p>The annotation GTF file must match the genome version used, since both use physical coordinates. Each GTF contains all the gene, exon, intron, and other regions of interest that we will use to annotate our <span class="notranslate">reads</span>, should our <span class="notranslate">reads</span> align to any of the regions specified in this file.</p>

<p>For alignment, we will use <span class="notranslate">RNA</span>-STAR for performance and splice-awareness.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-performing-the-alignment"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Performing the Alignment</h3>

  <ol>
    <li><strong><span class="notranslate">RNA</span>-STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúSingle-end or paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">RNA</span>-Seq FASTQ/FASTA file‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span>2</code> (output of <strong>UMI-tools extract</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>‚ÄúCustom or built-in reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in index</code>
          <ul>
            <li><em>‚ÄúReference genome with or without an annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">use genome reference without builtin gene-model</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mus Musculus (mm10)</code> (Mouse)</li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGene model (gff3,gtf) file for splice junctions‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mus_musculus.GRCm38.93.mm10.UCSC.ncbiRefSeq</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúResults‚Äù</em>:
          <ul>
            <li><em>‚Äú1: Results‚Äù</em>:
              <ul>
                <li><em>‚ÄúWhich tool was used to generate logs?‚Äù</em>:<code class="language-plaintext highlighter-rouge">STAR</code></li>
                <li><em>‚ÄúSTAR output‚Äù</em>:
                  <ul>
                    <li><em>‚Äú1: STAR output‚Äù</em>
                      <ul>
                        <li><em>‚ÄúType of STAR output?‚Äù</em>:<code class="language-plaintext highlighter-rouge">Log</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li><em>‚ÄúSTAR log output‚Äù</em> :(Select the STAR output file that ends in ‚Äúlog‚Äù)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol on the MultiQC Webpage</li>
  </ol>

</blockquote>

<p>The purpose of MultiQC is to observe how well our <span class="notranslate">reads</span> were mapped against the reference genome. Many <span class="notranslate">reads</span> are discarded due to being of too low quality, or having ambiguous sequence content that can map them to multiple locations.</p>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>What percentage of our <span class="notranslate">reads</span> are uniquely mapped? How many millions of <span class="notranslate">reads</span> is this percentage?</li>
    <li>What percentage of our <span class="notranslate">reads</span> are mapped to more than one locus?</li>
    <li>Is our overall <span class="notranslate">mapping</span> ‚Äògood‚Äô ?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li><code class="language-plaintext highlighter-rouge">59.5%</code> or ~80k <span class="notranslate">reads</span> were successfully mapped</li>
      <li><code class="language-plaintext highlighter-rouge">13.6%</code> are multiply mapped, and <code class="language-plaintext highlighter-rouge">3.7%</code> were mapped to too many loci
        <ul>
          <li>Multiply mapped means that a read was aligned to more than one gene</li>
          <li>Mapped to too many loci means that a read was aligned to 10 or more loci, and should be ignored.</li>
        </ul>
      </li>
      <li>It depends on how good the sequencing protocol is, and how many <span class="notranslate">reads</span> in total were mapped.
        <ul>
          <li><code class="language-plaintext highlighter-rouge">90%</code> is amazing, reserved for bulk <span class="notranslate">RNA</span>-seq which typically has high coverage</li>
          <li><code class="language-plaintext highlighter-rouge">70%</code> is weak for bulk <span class="notranslate">RNA</span>-seq, but good for <span class="notranslate">single-cell</span> <span class="notranslate">RNA</span>-seq</li>
          <li>This a small subset of a real dataset, but one would expect that 6 million mapped <span class="notranslate">reads</span> would be enough to generate a <span class="notranslate">downstream</span> analysis.</li>
        </ul>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="filtering">Filtering</h2>

<p>Before continuing let us first look back on some of the previous stages:</p>

<blockquote class="comment">
  <h3 id="comment-recap-of-previous-stages"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Recap of previous stages</h3>

  <ol>
    <li>
      <p><em>Barcode Extraction</em>:</p>

      <p>Here we used <code class="language-plaintext highlighter-rouge">umi_tools extract</code> on our input forward and reverse FASTQ files, and extracted the UMI and cell barcode from the forward read <em>sequence</em>, and placed it into the <em>header</em> of both forward and reverse FASTQ files. i.e. FASTQ files ‚Üí Modified FASTQ files</p>
    </li>
    <li>
      <p><em><span class="notranslate">Mapping</span></em>:</p>

      <p>We took the sequencing data from the reverse FASTQ file (with modified headers) and aligned it to the mouse genome, using annotations presented in the GTF file for that genome. i.e. Modified FASTQ file (reverse) ‚Üí BAM file</p>
    </li>
  </ol>

</blockquote>

<h4 id="confirming-reads-in-the-bam-file">Confirming <span class="notranslate">Reads</span> in the BAM file</h4>

<p>We now have a BAM file of our aligned <span class="notranslate">reads</span>, with cell and UMI barcodes embedded in the read headers. We also have the chromosome and base-pair positions of where these <span class="notranslate">reads</span> are aligned. The can be confirmed by peeking into the BAM file:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-confirming-the-alignment-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Confirming the Alignment Data</h3>

  <ol>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol of the BAM output from STAR.</li>
    <li>There are many header lines that begin with <code class="language-plaintext highlighter-rouge">@</code> which we are not interested in.</li>
    <li>
      <p>Look at 10th read directly below the header lines:</p>

      <p>SRR5683689.38437_GCATTC_CTTCGT	16	chr1	3439991	255	70M	*	0	0	CTTTGAATCTCTTCTTCCCAGCTAGTCATCTTCCTGCTTTTCTCTCTGTCTGTCTGTCTGTCTGTCTGTC	‚Äò0‚Äô&lt;B&lt;‚Äô‚ÄòB77&lt;BFBBBBB7‚ÄôFBFB0F7FBB&lt;B‚Äô‚Äô‚Äò&lt;IFFBF&lt;FBFB&lt;FBBFBB0&lt;BFFFBB0BBFFB¬´	NH:i:1 HI:i:1 AS:i:66 nM:i:1</p>
    </li>
  </ol>

</blockquote>

<p>The fields of the BAM file can be better explained at section 1.4 of <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">the SAM specification</a>, but we will summarise the main fields of interest here:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">SRR568..._GCATTC_CTTCGT</code>: The <em>rea<span class="notranslate">dna</span>me</em> appended by <code class="language-plaintext highlighter-rouge">_</code>, the cell barcode, another <code class="language-plaintext highlighter-rouge">_</code>, and then the UMI barcode.</li>
  <li><code class="language-plaintext highlighter-rouge">16</code>: The FLAG value</li>
</ul>

<blockquote class="question">
  <h3 id="question-what-does-the-alignment-flag-value-of-16-tell-us-about-this-read"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> What does the alignment flag value of 16 tell us about this read?</h3>

  <p><!-- TODO This information needs to be integrated into an actual tool -->
  We can interactively see what the different FLAG values mean by feeding values into the SAM specification to the <a href="https://broadinstitute.github.io/picard/explain-flags.html">Picard web tool</a></p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The read aligns to the reverse strand</p>

  </blockquote>

</blockquote>

<ul>
  <li><code class="language-plaintext highlighter-rouge">chr1</code> <code class="language-plaintext highlighter-rouge">3439991</code>: The position and base-pair of alignment of the first base of the sequence.</li>
  <li>A series of quality fields, with the main contributors being  the sequence and sequence quality strings.</li>
  <li><code class="language-plaintext highlighter-rouge">NH</code>: The number of hits for  this read. If it is multiply mapped, then the number of multiples will be shown (here <code class="language-plaintext highlighter-rouge">1</code>, so not multiply mapped).</li>
  <li><code class="language-plaintext highlighter-rouge">HI</code>: Which number this particular read is in the series of (potentially) multi-mapped <span class="notranslate">reads</span> (here <code class="language-plaintext highlighter-rouge">1</code>, not necessarily meaning the first or ‚Äòbetter‚Äô).</li>
  <li><code class="language-plaintext highlighter-rouge">nM</code>: The number of base mismatches in the alignment of this read to the reference (here <code class="language-plaintext highlighter-rouge">1</code>).</li>
</ul>

<h4 id="filtering-the-bam-file">Filtering the BAM File</h4>

<p>If we perform counting on the current BAM file we will be counting all <span class="notranslate">reads</span>, even the undesirable ones such as those that did not align so optimally.</p>

<p>The main filtering steps performed on our <span class="notranslate">reads</span> so far have been relatively silent due to the ‚Äòdefault‚Äô parameters used.</p>
<ul>
  <li><em>UMI-tools Extract</em> - Filters <span class="notranslate">reads</span> for those only with matching barcodes given by our barcodes file.</li>
  <li><em><span class="notranslate">RNA</span>-STAR</em> - As seen in the log, we lose 25% of our <span class="notranslate">reads</span> for being too short or being multiply mapped.</li>
</ul>

<p>Another filtering measure we can apply is to keep <span class="notranslate">reads</span> that we are confident about, e.g those with a minimum number of mismatches to the reference within an acceptable range. Specifically, we want to keep all <span class="notranslate">reads</span> that align to the forward or reverse strand that also have less that 3 mismatches to the reference, and are also mapped only once to the reference.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>Filter BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM dataset(s) to filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_bam</code> (output of <strong><span class="notranslate">RNA</span> STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚ÄúCondition‚Äù</em>:
          <ul>
            <li>In <em>‚Äú1: Condition‚Äù</em>:
              <ul>
                <li>In <em>‚ÄúFilter‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">alignmentFlag</code>
                          <ul>
                            <li><em>‚ÄúFilter on this alignment flag‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <em>‚ÄúInsert Condition‚Äù</em>:</li>
            <li>In <em>‚Äú2: Condition‚Äù</em>:
              <ul>
                <li>In <em>‚ÄúFilter‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">alignmentFlag</code>
                          <ul>
                            <li><em>‚ÄúFilter on this alignment flag‚Äù</em>: <code class="language-plaintext highlighter-rouge">16</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <em>‚ÄúInsert Condition‚Äù</em>:</li>
            <li>In <em>‚Äú3: Condition‚Äù</em>:
              <ul>
                <li>In <em>‚ÄúFilter‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">tag</code>
                          <ul>
                            <li><em>‚ÄúFilter on a particular tag‚Äù</em>: <code class="language-plaintext highlighter-rouge">nM:&lt;3</code>
<small>(<strong>Attention! please use a lowercase ‚Äòn‚Äô here!</strong>)</small></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <em>‚ÄúInsert Condition‚Äù</em>:</li>
            <li>In <em>‚Äú4: Condition‚Äù</em>:
              <ul>
                <li>In <em>‚ÄúFilter‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">tag</code>
                          <ul>
                            <li><em>‚ÄúFilter on a particular tag‚Äù</em>: <code class="language-plaintext highlighter-rouge">NH:&lt;2</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúWould you like to set rules?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúEnter rules here‚Äù</em>: <code class="language-plaintext highlighter-rouge">(1 | 2) &amp; 3 &amp; 4</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>Why are we filtering only for alignment flags <code class="language-plaintext highlighter-rouge">0</code> and <code class="language-plaintext highlighter-rouge">16</code>?</li>
    <li>What do the tag filters <code class="language-plaintext highlighter-rouge">nM:&lt;3</code> and <code class="language-plaintext highlighter-rouge">NH:&lt;2</code> do?</li>
    <li>What is happening at the rules stage?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Alignment flags <code class="language-plaintext highlighter-rouge">0</code> and <code class="language-plaintext highlighter-rouge">16</code> specify that we wish to only keep <span class="notranslate">reads</span> that align to the forward and reverse strands.</li>
      <li>We are keeping <span class="notranslate">reads</span> that have a number of mismatches (<code class="language-plaintext highlighter-rouge">nM</code>) to the reference of less than 3, and has a number of hits (<code class="language-plaintext highlighter-rouge">NH</code>) across the reference of less than 2 (i.e. it is not a multiply-mapped read).</li>
      <li><a href="https://en.wikipedia.org/wiki/Boolean_expression">Boolean <span class="notranslate">expression</span>s</a> are applied that state that <em>either</em> conditions 1 or 2 can happen, in conjunction with rules 3 and 4 happening.</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="quantification">Quantification</h2>

<p>Once we have the name of the gene for a specific read, we can count how many of those <span class="notranslate">reads</span> fall into that gene and generate a count matrix.</p>

<p>Counting <span class="notranslate">reads</span> is performed by two commonly-used tools:</p>
<ol>
  <li><span class="notranslate">RNA</span>-STAR</li>
  <li>FeatureCounts</li>
</ol>

<p>The <strong><span class="notranslate">RNA</span>-STAR</strong> tools has the ability to count <span class="notranslate">reads</span> as it maps them. <strong>FeatureCounts</strong> performs the same task, but is capable of counting not just at the Read level, but also at the UMI level too, such that 10 duplicate <span class="notranslate">reads</span> will be counted only once.</p>

<p>Unfortunately, both are currently limited to counting without being able to distinguish between different cells.</p>

<blockquote class="details">
  <h3 id="details-example"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Example</h3>

  <p>If we consider the number of <span class="notranslate">reads</span> that align to <em>GeneA</em>, the output given by these two tools is as follows:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>(<span class="notranslate">reads</span>)</th>
          <th><span class="notranslate">RNA</span> STAR</th>
          <th>FeatureCounts</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GeneA</td>
          <td>12</td>
          <td>12</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

  <p>But what we actually require is:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>(<span class="notranslate">reads</span>)</th>
          <th>C1</th>
          <th>C2</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GeneA</td>
          <td>10</td>
          <td>2</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

  <p>or more specifically:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>(UMIs)</th>
          <th>C1</th>
          <th>C2</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GeneA</td>
          <td>2</td>
          <td>1</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

</blockquote>

<p>In order to obtain this desired format, we must use <strong>UMI-tools count</strong> to perform the counting. However, this tool is dependent on <strong>FeatureCounts</strong> to annotate our <span class="notranslate">reads</span> with the one crucial piece of information that is missing from our BAM file: the name of the gene.</p>

<blockquote class="comment">
  <h3 id="comment-verifying-missing-gene-name"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Verifying missing gene name</h3>
  <p>You can check this yourself by examining the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> of the BAM file <em>‚ÄúSTAR Alignment file‚Äù</em></p>
</blockquote>

<h3 id="annotating-gene-name-with-featurecounts">Annotating Gene name with FeatureCounts</h3>

<p>Let us annotate our BAM file with desired gene tags.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quantification-assist-via-featurecounts"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quantification assist via FeatureCounts</h3>

  <ol>
    <li><strong>FeatureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAlignment file‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapped_<span class="notranslate">reads</span></code> (output of <strong>Filter BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúSpecify strand information‚Äù</em>:<code class="language-plaintext highlighter-rouge">Stranded (Forward)</code></li>
        <li><em>‚ÄúGene annotation file‚Äù</em>: <code class="language-plaintext highlighter-rouge">in your history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGene annotation file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mus_musculus.GRCm38.93.mm10.UCSC.ncbiRefSeq.gtf</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúAdvanced options‚Äù</em>:
          <ul>
            <li><em>‚ÄúCount multi-<span class="notranslate">mapping</span> <span class="notranslate">reads</span>/fragments‚Äù</em>: <code class="language-plaintext highlighter-rouge">Disabled; multi-<span class="notranslate">mapping</span> <span class="notranslate">reads</span> are excluded (default)</code></li>
            <li><em>‚ÄúExon-exon junctions‚Äù</em>: <code class="language-plaintext highlighter-rouge">No (default)</code></li>
            <li><em>‚ÄúAnnotates the alignment file with ‚ÄòXS:Z:‚Äô-tags to described per read or read-pair the corresponding assigned feature(s).‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Examine the output BAM file
      <ul>
        <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> for the ‚ÄúFeature Counts: Alignment File‚Äù</li>
        <li>Scroll down past the header lines</li>
        <li>Scroll horizontally to the tags, observe the new <code class="language-plaintext highlighter-rouge">XT</code> tag.</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">XS</code> and <code class="language-plaintext highlighter-rouge">XT</code> tags in the BAM file will now form the basis for counting <span class="notranslate">reads</span>.
With all the relevant data now in our BAM file, we can actually perform the counting via <code class="language-plaintext highlighter-rouge">UMI-tools count</code>.</p>

<blockquote class="comment">
  <h3 id="comment-verifying-added-gene-name"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Verifying added gene name</h3>
  <p>You can once again check this yourself by examining the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> of the BAM file <em>‚ÄúSTAR Alignment file‚Äù</em></p>
</blockquote>

<h3 id="counting-genes--cell">Counting Genes / Cell</h3>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-final-quantification"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Final Quantification</h3>

  <ol>
    <li><strong>UMI-tools counts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSorted BAM file‚Äù</em>: <code class="language-plaintext highlighter-rouge">out_file1</code> (output of <strong>FeatureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúUMI Extract Method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcodes are contained at the end of the read separated by a delimiter</code></li>
        <li><em>‚ÄúBam is paired-end‚Äù</em>:<code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúMethod to identify group of <span class="notranslate">reads</span>‚Äù</em>:<code class="language-plaintext highlighter-rouge">Unique</code></li>
        <li><em>‚ÄúExtra Parameters‚Äù</em>:
          <ul>
            <li><em>‚ÄúDeduplicate per gene.‚Äù</em>:<code class="language-plaintext highlighter-rouge">XT</code></li>
            <li><em>‚ÄúGroup <span class="notranslate">reads</span> only if they have the same cell barcode.‚Äù</em>:<code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>‚ÄúPrepend a label to all column headers‚Äù</em>:<code class="language-plaintext highlighter-rouge">No modifications</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The important parameters to take note of are those given in the <em>Extra Parameters</em> where we specify that each of the <span class="notranslate">reads</span> with a <code class="language-plaintext highlighter-rouge">XT:Z</code> tag in the BAM file will be counted on a per cell basis. <span class="notranslate">Reads</span> sharing the same UMI and cell Barcodes will be de-duplicated into a single count, reducing PCR duplicate bias from the analysis.</p>

<p>At this stage, we now have a tabular file containing genes/features as rows, and cell labels as headers.</p>

<blockquote class="question">
  <h3 id="question-question-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>How many genes do we have in the matrix?</li>
    <li>How many cells?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>2,140 lines
This information can be seen in the file preview window by clicking on the name of the file (<strong>NOT</strong> the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> symbol).</p>
      </li>
      <li>
        <p>180 columns (not including the first column of gene names)
The number of columns can be seen by scrolling the file preview window completely to the right. 192 cell barcodes were given, but due to the subsetted data, only 180 were detected.</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>The generation of a single count matrix is now complete, with the emphasis on the word <em>single</em> due to the fact that we often deal in multiple batches of sequencing data.</p>

<blockquote class="comment">
  <h3 id="comment-recap-of-previous-stages-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Recap of previous stages</h3>

  <ol>
    <li>
      <p><em>Barcode Extraction</em>:</p>

      <p>Here we used <code class="language-plaintext highlighter-rouge">UMI_tools extract</code> on our input forward and reverse FASTQ files, and extracted the UMI and cell barcode from the forward read <em>sequence</em>, and placed it into the <em>header</em> of both forward and reverse FASTQ files. i.e. FASTQ files ‚Üí Modified FASTQ files</p>
    </li>
    <li>
      <p><em><span class="notranslate">Mapping</span></em>:</p>

      <p>We took the sequencing data from the reverse FASTQ file (with modified headers) and aligned it to the mouse genome, using annotations presented in the GTF file for that genome. i.e. Modified FASTQ file (reverse) ‚Üí BAM file</p>
    </li>
    <li>
      <p><em>Quality Filtering</em>:</p>

      <p><span class="notranslate">Reads</span> with alignment mismatches greater than 2 were discarded, and only non multi-mapped <span class="notranslate">reads</span> that mapped to the forward or reverse strand were kept</p>
    </li>
    <li>
      <p><em>Quantification</em>:</p>

      <p>Gene tags were added to our alignment file, and <span class="notranslate">reads</span> were grouped according those sharing the same gene tag, with further reduction performed by collapsing all <span class="notranslate">reads</span> sharing the same cell and UMI barcode to be counted only once.</p>
    </li>
  </ol>

</blockquote>

<p>This concludes the first part of the tutorial which focused on the transformation of raw FASTQ data from a single batch into a count matrix. The second part of this tutorial guides us through the process of merging multiple processed batches from the first stage, and performing qualitative filtering.</p>

<blockquote class="details">
  <h3 id="details-workflows-for-all-steps"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Workflows For All Steps</h3>

  <p><span class="notranslate">Galaxy</span> provides a <span class="notranslate">workflow</span> that captures the process of all the above stages for a <a href="workflows/scrna_pp_celseq.ga">single pair of FASTQ data</a>:</p>

  <figure id="figure-5"><img src="/training-material/topics/transcriptomics/images/scrna_workflow_single.png" alt="Single Batch Workflow" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Single Batch <span class="notranslate">Workflow</span></figcaption></figure>

  <p>For multiple batch processing, <span class="notranslate">Galaxy</span> can make use of <em>Nested Workflows</em> that in this case can take in a <a href="workflows/scrna_mp_celseq.ga">list of input paired FASTQ data</a> and process them in parallel.</p>

  <figure id="figure-6"><img src="/training-material/topics/transcriptomics/images/scrna_workflow_nested.png" alt="Multiple Batch Workflow" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Nested Multiple Batch <span class="notranslate">Workflow</span></figcaption></figure>

</blockquote>

<p>For the processing of 10x Genomics datasets, please refer to the <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html"><em>Pre-processing of 10X <span class="notranslate">Single-Cell</span> <span class="notranslate">RNA</span> Datasets</em></a> tutorial.</p>

<h1 id="multi-batch-processing">Multi-Batch Processing</h1>

<figure id="figure-7"><img src="/training-material/topics/transcriptomics/images/scrna_workflow_lower.png" alt="Workflow Lower" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Multiple batch processing</figcaption></figure>

<p>Handling more than one batch of sequencing data is rather trivial when we take into account our main goals and requirements:</p>

<ol>
  <li>For each batch, convert FASTQ <span class="notranslate">reads</span> from into a count matrix.</li>
  <li>Merge all count matrices into a single count matrix</li>
</ol>

<p>The first step merely requires us to run the same <span class="notranslate">workflow</span> on each of our batches, using the exact same inputs except for the FASTQ paired data. The second step requires a minimal level of interaction from us; namely using a merge tool and selecting our matrices.</p>

<h3 id="data-upload-and-organisation">Data upload and organisation</h3>

<p>The count matrix we have generated in the previous section is too sparse to perform any reasonable analysis upon, and constitutes data only of a single batch. Here we will use more populated count matrices from multiple batches, under the assumption that we now know how to generate each individual one of them using the steps provided in the previous section. This data is available at <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a>.</p>

<p>Once again, file naming is important, and so we will rename our matrix files appropriately to the plate and batch they are supposed to originate from.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organisation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload and organisation</h3>

  <ol>
    <li>
      <p>Create a new history and rename it (<em>e.g.</em> sc<span class="notranslate">RNA</span>-seq multiple-batch tutorial)</p>
    </li>
    <li>Import the eight matrices (<code class="language-plaintext highlighter-rouge">P1_B1.tsv</code>, <code class="language-plaintext highlighter-rouge">P1_B2.tsv</code>, etc.) and the barcodes file from <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)
      <ul>
        <li>Set the datatype of the files to <strong>tabular</strong></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3253142/files/P1_B1.tsv
https://zenodo.org/record/3253142/files/P1_B2.tsv
https://zenodo.org/record/3253142/files/P1_B3.tsv
https://zenodo.org/record/3253142/files/P1_B4.tsv
https://zenodo.org/record/3253142/files/P2_B5.tsv
https://zenodo.org/record/3253142/files/P2_B6.tsv
https://zenodo.org/record/3253142/files/P2_B7.tsv
https://zenodo.org/record/3253142/files/P2_B8.tsv
https://zenodo.org/record/3253142/files/celseq_barcodes.192.tabular
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links-1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>Rename a matrix
      <ul>
        <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> of the <em><code class="language-plaintext highlighter-rouge">P1_B1.tsv</code></em> file</li>
        <li>Set the Name field such that it is affixed with ‚Äú_P1_B1‚Äù (e.g. ‚Äòmultibatch_P1_B1‚Äô)</li>
        <li>Click <em>Save</em></li>
      </ul>
    </li>
    <li>Repeat for all matrices
<strong>Pay attention to the Plate number which changes after Batch 4</strong></li>
  </ol>

</blockquote>

<h2 id="merging-count-matrices">Merging Count Matrices</h2>

<p>Before we begin, we must consider that our matrices are not equal ‚Äì e.g. Batch1 has 3 cells that describe <em>Genes{A,B,C,D}</em> whereas Batch2 has 4 cells that describe <em>Genes{C,D,E}</em>.</p>

<p>We have the problem that only <em>GeneC</em> and <em>GeneD</em> appear in both batches, whilst describing 7 different cells in total.</p>

<p>To resolve this we can perform a ‚ÄúFull Table Join‚Äù where the missing data for <em>GeneE</em> and <em>GeneA</em> in Batch1 and Batch2 respectively are replaced with zeroes:</p>

<figure id="figure-8"><img src="/training-material/topics/transcriptomics/images/scrna_fulltable.svg" alt="Table Join" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Full Table Join</figcaption></figure>

<blockquote class="comment">
  <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>For more information on table joins, see <a href="http://www.sql-join.com/sql-join-types/">here</a></p>
</blockquote>

<blockquote class="question">
  <h3 id="question-question-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>Why is it required to change the column headers in the Full matrix?</li>
    <li>Why were the cell labels in B1 and B2 the same, if they were labelling completely different cells?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Although the cell headers in each batch matrix is the same, the cells they label are <em>not</em> the same and need to be relabelled in the final matrix to tell us which batch they originated from.</li>
      <li>The reason the cell headers are the same is because the cells use the same barcodes, due to fact that the <em>same</em> barcodes are sometimes used across <em>different</em> batches.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Let us now merge our matrices from different batches. In order to ensure that our batches are merged in the order that we wish, we should first create a list of datasets so that our matrices are merged in the order given by the list.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-table-merge"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Table Merge</h3>

  <ol>
    <li><strong>Create a Dataset List</strong>
      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a dataset collection</h3>

        <ul>
          <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="../../../galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>
          <li>Check all the datasets in your history you would like to include</li>
          <li>
            <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>

            <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>
          </li>
          <li>Enter a name for your collection</li>
          <li>Click <strong>Create List</strong> to build your collection</li>
          <li>Click on the checkmark icon at the top of your history again</li>
        </ul>
      </blockquote>
    </li>
    <li><strong>Column Join on Collections</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúTabular Files‚Äù</em>: (Select the Dataset Collection icon <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span>, and select the Collection from the previous step each of the matrices that you wish to join)</li>
        <li><em>‚ÄúIdentifier column‚Äù</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>‚ÄúNumber of Header lines in each item‚Äù</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>‚ÄúAdd column name to header‚Äù</em>:<code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFill character‚Äù</em>:<code class="language-plaintext highlighter-rouge">0</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The identifier column refers to the column where the gene names are listed. A 1:1 correspondence between matrices is checked, so that the merge does not concatenate the wrong rows between matrices. The <em>Fill character</em> provides a default value of 0 for cases where a Gene appears only in one of the matrices as per our example earlier.</p>

<p>Once the merge is complete, we can now peek at our full combined matrix by once again clicking on the file name to see a small summary.</p>

<blockquote class="question">
  <h3 id="question-question-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Each of these matrices/batches come from the same organism.</p>

  <ol>
    <li>How much overlap in their detected genes did you expect?</li>
    <li>How much overlap in their detected genes did you observe?</li>
    <li>Why is this?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>Given that they come from the same sample, and each matrix has ~15,000 genes, we would have expected a high overlap between matrices, yielding ~18,000 genes in the combined matrix if we assume a difference of +/- 500 genes per batch.</p>
      </li>
      <li>
        <p>We observe 20,800 genes, giving less overlap between batches of the same organism than we originally thought.</p>
      </li>
      <li>
        <p>The batches were sequenced at different time points along the organisms development, and therefore different genes were expressed/detected at different time points. For early development data, the overlap can be very sparse.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<p>In the new combined matrix we see that we have 1536 cells, but this number is greatly overestimated.  This is because <em>not all batches use the same barcodes</em>, and yet we applied the full set of 192 barcodes against our FASTQ data during the <a href="#barcode-extraction"><em>Barcode Extraction</em></a> stage previously.</p>

<p>The reason we do this is to test for cross-contamination between batches, the details of which are better explained in the <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/slides.html"><em>accompanying slides</em></a>.</p>

<h2 id="guarding-against-cross-contamination">Guarding against Cross-Contamination</h2>

<p>There are multiple possible ways to configure a plate for sequencing multiple batches. Thankfully, <span class="notranslate">Galaxy</span> provides a tool that caters for this, and checks for cross-contamination in any experimental setup. It requires only the following information:</p>

<ol>
  <li>A full list of barcodes</li>
  <li>Which barcodes apply to which batches</li>
  <li>Which batches apply to which plates</li>
</ol>

<p>Since the plating protocol we are using is that designed by the Freiburg MPI Gr√ºn lab, we will follow their structure.</p>

<blockquote class="details">
  <h3 id="details-details-plating-protocol"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Details: Plating protocol</h3>

  <ul>
    <li>
      <p>Barcodes:
  These are each 8bp long, with an edit distance of 2, and there 192 of them.</p>

      <blockquote class="matrix">
        <table>
          <tbody>
            <tr>
              <td>001-006</td>
              <td>AACACC AACCTC AACGAG AACTGG AAGCAC AAGCCA</td>
            </tr>
            <tr>
              <td>007-012</td>
              <td>AAGGTG AAGTGC ACAAGC ACAGAC ACAGGA ACAGTG</td>
            </tr>
            <tr>
              <td>.</td>
              <td>.</td>
            </tr>
            <tr>
              <td>.</td>
              <td>.</td>
            </tr>
            <tr>
              <td>180-186</td>
              <td>TTACGC TTCACC TTCCAG TTCGAC TTCTCG TTGCAC</td>
            </tr>
            <tr>
              <td>187-192</td>
              <td>TTGCGA TTGCTG TTGGAG TTGGCA TTGGTC TTGTGC</td>
            </tr>
          </tbody>
        </table>
      </blockquote>
    </li>
    <li>
      <p>Plates:
  Here we have 8 batches spread out over 2 plates, with alte<span class="notranslate">rna</span>te barcode striping.</p>

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th>¬†</th>
              <th>001-096</th>
              <th>097-192</th>
              <th>001-096</th>
              <th>097-192</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Plate 1</td>
              <td>B1</td>
              <td>B2</td>
              <td>B3</td>
              <td>B4</td>
            </tr>
            <tr>
              <td>Plate 2</td>
              <td>B5</td>
              <td>B6</td>
              <td>B7</td>
              <td>B8</td>
            </tr>
          </tbody>
        </table>
      </blockquote>
    </li>
  </ul>

</blockquote>

<p>This plating protocol can be converted into a more textual format, which allows for many variable setups (see <em>Help</em> section of <em>Cross-contamination Barcode Filter</em> tool).</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[Barcodes ‚Üí Batches]
001-096: B1 , B3 , B5 , B7
097-192: B2 , B4 , B6 , B8

[Plates ‚Üí Batches]
1: B1 , B2 , B3 , B4
2: B5 , B6 , B7 , B8
</code></pre></div></div>

<p>Let us now apply this protocol to our count matrix, and look for any cross-contamination.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-barcode-filtering"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Barcode Filtering</h3>

  <p>Select <strong>Cross-contamination Barcode Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
  <ul>
    <li><em>‚ÄúInput Matrix‚Äù</em>:<code class="language-plaintext highlighter-rouge">Column Join output</code> (merged matrices)</li>
    <li><em>‚ÄúComplete Barcodes‚Äù</em>:<code class="language-plaintext highlighter-rouge">celseq_barcodes.192.tabular</code> (barcodes file)</li>
    <li><em>‚ÄúPlate Protocol‚Äù</em>:<code class="language-plaintext highlighter-rouge">Custom</code>
      <ul>
        <li><em>‚ÄúUnder ‚ÄòBarcode Format‚Äô‚Äú</em>:
          <ul>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Barcode Format</code>:
              <ul>
                <li><em>‚Äú1: Barcode Format‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúBarcode Range: Start‚Äù</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
                    <li><em>‚ÄúBarcode Range: End‚Äù</em>:<code class="language-plaintext highlighter-rouge">96</code></li>
                    <li><em>‚ÄúBatches utilizing this Range‚Äù</em>:<code class="language-plaintext highlighter-rouge">1,3,5,7</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Barcode Format</code>:
              <ul>
                <li><em>‚Äú2: Barcode Format‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúBarcode Range: Start‚Äù</em>:<code class="language-plaintext highlighter-rouge">97</code></li>
                    <li><em>‚ÄúBarcode Range: End‚Äù</em>:<code class="language-plaintext highlighter-rouge">192</code></li>
                    <li><em>‚ÄúBatches utilizing this Range‚Äù</em>:<code class="language-plaintext highlighter-rouge">2,4,6,8</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúUnder ‚ÄòPlate Format‚Äô‚Äú</em>:
          <ul>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Plate Format</code>:
              <ul>
                <li><em>‚Äú1: Plate Format‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúPlate Number‚Äù</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
                    <li><em>‚ÄúBatches within this Plate Number‚Äù</em>:<code class="language-plaintext highlighter-rouge">1,2,3,4</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Plate Format</code>:
              <ul>
                <li><em>‚Äú2: Plate Format‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúPlate Number‚Äù</em>:<code class="language-plaintext highlighter-rouge">2</code></li>
                    <li><em>‚ÄúBatches within this Plate Number‚Äù</em>:<code class="language-plaintext highlighter-rouge">5,6,7,8</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Expand the <em>‚ÄúRegEx Parameters‚Äù</em> section:
      <ul>
        <li><em>‚ÄúRegEx to extract Plate, Batch, and Barcodes from headers‚Äù</em>:<code class="language-plaintext highlighter-rouge">.*P(\\d)_B(\\d)_([ACTG]+)</code></li>
        <li><em>‚ÄúRegEx to replace Plate, Batch, and Barcodes from headers‚Äù</em>:<code class="language-plaintext highlighter-rouge">P\\1_B\\2_\\3</code></li>
      </ul>
    </li>
  </ul>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-regular-expressions"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Regular <span class="notranslate">Expression</span>s</h3>
  <p>The <a href="https://www.regular-expressions.info/quickstart.html">regular <span class="notranslate">expression</span></a> (RegEx) used in the final steps of the above <em>Hands-On</em> is required in order to tell us how to capture the important information in the cell headers contained in brackets <code class="language-plaintext highlighter-rouge">(</code> <code class="language-plaintext highlighter-rouge">)</code>, where <code class="language-plaintext highlighter-rouge">\\d</code> denotes an expected digit, and <code class="language-plaintext highlighter-rouge">[ACTG]+</code> denotes 1 or more characters matching A or C or T or G.</p>

  <p>The information captured in the brackets <code class="language-plaintext highlighter-rouge">(</code> <code class="language-plaintext highlighter-rouge">)</code> can then be placed in the desired arrangement, where <code class="language-plaintext highlighter-rouge">Place \\1 Matches \\2 Here \\3</code> would place the first <code class="language-plaintext highlighter-rouge">\\d</code> after ‚ÄúPlace ‚Äú, the second after ‚ÄúMatches ‚Äú, and so on.</p>

</blockquote>

<p>The plot that follows tells us everything we need to know about each of our batches. Each batch is essentially tested against the full set of barcodes in order to assert that only the desired or ‚ÄòReal‚Äô barcodes have been sequenced.</p>

<h4 id="cross-contamination-plot">Cross-contamination Plot</h4>

<figure id="figure-9"><img src="/training-material/topics/transcriptomics/images/scrna_crosscontamination.png" alt="Contamination Plots" /><figcaption><span class="figcaption-prefix">Figure 9:</span> The Pre-filter and Post-filter plots</figcaption></figure>

<p>Two things to take note of:</p>

<ol>
  <li>In the pre-filter plot, we can see how only half of the sequences in each batch map to half the barcodes. This shows very little cross-contamination, and proves that our data is real.</li>
  <li>The post-filter plot essentially removes the false barcodes from each batch and retains only the ‚ÄòReal‚Äô barcodes.</li>
</ol>

<blockquote class="question">
  <h3 id="question-question-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>The count matrix that is output from this tool has only half the number of cells as the original input count matrix. Why is this?</li>
    <li>Which batches yield worrying levels of cross-contamination?</li>
    <li>Which batches should we remove from all further analysis?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Because only half the barcodes in each batch were real. The <em>UMI-tools extract</em> took the entire barcodes file to filter against each batch, and the <em>UMI-tools count</em> also took the entire barcodes file to count against each batch. Naturally, each batch produced 192 cells, even though 96 were real. As a result of joining each of these matrices we ended up with a count-matrix of \(8 * 192 = 1536\) cells. The cross-contamination tool removes the false barcodes (50% in each batch), resulting in \(768\) cells.</li>
      <li>Batch 4 and Batch 6 both appear to have a significant number of mid-to-high range counts in cells under the <em>False Positives</em> section in these batches. This means that the cell barcodes that we should <em>not</em> be detecting in those batches, did in fact detect cells.</li>
      <li>Batch 4 does still have a good number of <em>True Positives</em>, despite the many mid-level <em>False Positives</em> so we could still perhaps use the cells from that Batch. Batch 6 on the other hand appears to derive most of its counts from the <em>False Positives</em>, and therefore is likely not so suitable for further analysis.</li>
    </ol>
  </blockquote>
</blockquote>

<p>All <em>False Positives</em> from all batches are filtered out, leaving only the <em>True Positives</em> (or ‚ÄòReal‚Äô barcodes) in the remaining filtered matrix. It is up to the user to filter further based on the contamination information that they derive from this step. With this, a QC filtered count-matrix is produced that can then be used for further <span class="notranslate">downstream</span> analysis.</p>

<!--
Factoid: We can convert the number of UMIs to the number of molecules using a transformation script.
-->

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial we have learned the importance of barcoding; namely how to define, extract, and annotate them from our <span class="notranslate">reads</span> and into the read headers, in order to preserve them during <span class="notranslate">mapping</span>. We have discovered how these barcoded <span class="notranslate">reads</span> are transformed into counts, where the cell barcode and UMI barcode are used to denote individual cells and to correct against <span class="notranslate">reads</span> that are PCR duplicates. Finally, we have learned how to combine separate batch data as well as being able to check and correct against cross-contamination.</p>

<figure id="figure-10"><img src="/training-material/topics/transcriptomics/images/scrna_workflow.svg" alt="Recap of workflow" /><figcaption><span class="figcaption-prefix">Figure 10:</span> A recap of the entire <span class="notranslate">workflow</span></figcaption></figure>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Verifying the distribution of barcodes via a FASTQC plot</p>
</li>
                            
                            <li><p>Relocating barcodes into headers</p>
</li>
                            
                            <li><p>Merging matrices from different batches together</p>
</li>
                            
                            <li><p>Removing unwanted barcodes</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Pre-processing of Single-Cell RNA Data (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Mehmet Tekman, B√©r√©nice Batut, Anika Erxleben, Wolfgang Maier, 2021 <b>Pre-processing of Single-Cell RNA Data (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html">/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-scrna-preprocessing,
    author = "Mehmet Tekman and B√©r√©nice Batut and Anika Erxleben and Wolfgang Maier",
    title = "Pre-processing of Single-Cell RNA Data (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Mehmet Tekman, B√©r√©nice Batut, Anika Erxleben, Wolfgang Maier)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
