<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Interval-Wise Testing for omics data</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Statistical Analyses for omics data and machine learning ..." />
        <meta property="og:title" content="Galaxy Training: Interval-Wise Testing for omics data" />
        <meta property="og:description" content="Statistical Analyses for omics data and machine learning ..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/statistics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Statistics and machine learning
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fstatistics%2Ftutorials%2Fiwtomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fstatistics%2Ftutorials%2Fiwtomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fstatistics%2Ftutorials%2Fiwtomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fstatistics%2Ftutorials%2Fiwtomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fstatistics%2Ftutorials%2Fiwtomics%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fstatistics%2Ftutorials%2Fiwtomics%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/statistics/tutorials/iwtomics/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Interval-Wise Testing for omics data",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Statistics and machine learning",
    "description": "Statistical Analyses for omics data and machine learning using Galaxy tools",
    "url": "https://training.galaxyproject.org//training-material/topics/statistics/"
  },
  "courseCode": "statistics / iwtomics / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Interval-Wise Testing for omics data' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/statistics/tutorials/iwtomics/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - How to visualize high-resolution omics data in different groups of genomic regions?\n - How to evaluate differences in high-resolution omics data between groups of genomic regions?\n - How to detect locations and scales at which the significant effects unfold?\n\n\\nThe objectives are:\n - Pre-process high-resolution omics data\n - Create pointwise boxplots corresponding to different groups of genomic regions\n - Perform the Interval-Wise Testing between two groups of genomic regions\n - Visualize and inspect test results\n - Detect relevant locations and scales\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Marzia A Cremona"
    },
    {
      "@type": "Person",
      "name": "Fabio Cumbo"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Marzia A Cremona"
    },
    {
      "@type": "Person",
      "name": "Fabio Cumbo"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Statistics and machine learning",
      "description": "Statistical Analyses for omics data and machine learning using Galaxy tools",
      "url": "https://training.galaxyproject.org//training-material/topics/statistics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_2269",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_2269",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_2269"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Interval-Wise Testing for omics data</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/marziacremona/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/marziacremona" alt="Marzia A Cremona">Marzia A Cremona</a>, <a href="/training-material/hall-of-fame/fabio-cumbo/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/fabio-cumbo" alt="Fabio Cumbo">Fabio Cumbo</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to visualize high-resolution omics data in different groups of genomic regions?</p>
</li>
            
            <li><p>How to evaluate differences in high-resolution omics data between groups of genomic regions?</p>
</li>
            
            <li><p>How to detect locations and scales at which the significant effects unfold?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Pre-process high-resolution omics data</p>
</li>
            
            <li><p>Create pointwise boxplots corresponding to different groups of genomic regions</p>
</li>
            
            <li><p>Perform the Interval-Wise Testing between two groups of genomic regions</p>
</li>
            
            <li><p>Visualize and inspect test results</p>
</li>
            
            <li><p>Detect relevant locations and scales</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1288429">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/statistics/tutorials/iwtomics/workflows/" title="Workflows" alt="Interval-Wise Testing for omics data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/statistics/tutorials/iwtomics/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/statistics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://ecology.usegalaxy.eu/" title="Galaxy for Ecology">
            Galaxy for Ecology
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>IWTomics (<a class="citation" href="#Cremona2018">Cremona <i>et al.</i> 2018</a>) implements the Interval-Wise Testing (IWT; <a class="citation" href="#Pini2017">Pini and Vantini 2017</a>) for omics data. This
inferential procedure tests for differences in ‚ÄúOmics‚Äù data between two groups
of genomic regions, and does not require fixing location and scale at the
outset.</p>

<p>In this tutorial we use real data about endogenous retroviruses (ERVs) in mouse.
In particular, we compare recombination hotspots in the flanking regions of fixed ETn versus control regions.
The dataset contains two region datasets ‚ÄúETn fixed‚Äù, ‚ÄúControl‚Äù and one feature ‚ÄúRecombination hotspots content‚Äù. The region dataset ‚ÄúETn fixed‚Äù contains 1296 genomic regions of 64 kb surrounding fixed ETns elements (32-kb flanking sequences <span class="notranslate">upstream</span> and 32-kb flanking sequences <span class="notranslate">downstream</span> of each element). The region dataset ‚ÄúControl‚Äù contains 1142 regions of 64 kb without elements, used as control in the test. The regions are aligned around their center (i.e. around the ETn integration sites).</p>

<p>Recombination hotspots measurements are associated to each ‚ÄúETn fixed‚Äù and ‚ÄúControl‚Äù region. In particular, this feature is measured in 1-kb windows, so that each region is associated to a recombination hotspots curve made of 64 values. The measurement used is the feature content, i.e. the fraction of the 1-kb window that is covered by recombination hotspots</p>

<p>The data we use in this tutorial is available at <a href="https://doi.org/10.5281/zenodo.1184682">Zenodo</a>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>IWTomics is composed of three steps (corresponding to three tools):</p>

<ol id="markdown-toc">
  <li><a href="#step-1-loading-and-pre-processing" id="markdown-toc-step-1-loading-and-pre-processing">Step 1: Loading and pre-processing</a></li>
  <li><a href="#step-2-performing-interval-wise-testing" id="markdown-toc-step-2-performing-interval-wise-testing">Step 2: Performing Interval-Wise Testing</a></li>
  <li><a href="#step-3-selecting-test-scale" id="markdown-toc-step-3-selecting-test-scale">Step 3: Selecting test scale</a></li>
</ol>

</blockquote>

<h1 id="step-1-loading-and-pre-processing">Step 1: Loading and pre-processing</h1>

<p>The first tool (IWTomics Load Smooth and Plot) imports a collection of genomic region datasets, and associates to each region multiple genomic feature measurements. It allows to align the regions in multiple ways (center, left, right or scale alignment), to smooth the feature curves (possibly filling gaps in the measurements) and to create a graphical representation of the feature measurements in each region datasets (aligned curves or pointwise quantile curves).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get the data</h3>
  <ol>
    <li>
      <p>Create a new history for this tutorial</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>Import datasets:
      <ul>
        <li>Fixed ETn (<code class="language-plaintext highlighter-rouge">ETn_fixed.<span class="notranslate">bed</span></code>) regions</li>
        <li>Control (<code class="language-plaintext highlighter-rouge">Control.<span class="notranslate">bed</span></code>) regions</li>
        <li>Recombination Hotspot (<code class="language-plaintext highlighter-rouge">Recombination_hotspots.txt</code>) content</li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1288429/files/ETn_fixed.<span class="notranslate">bed</span>
https://zenodo.org/record/1288429/files/Control.<span class="notranslate">bed</span>
https://zenodo.org/record/1288429/files/Recombination_hotspots.txt
</code></pre></div>      </div>
    </li>
    <li>Import header files for regions and features:
      <ul>
        <li><code class="language-plaintext highlighter-rouge">regions_header.tabular</code></li>
        <li><code class="language-plaintext highlighter-rouge">features_header.tabular</code></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1288429/files/regions_header.tabular
https://zenodo.org/record/1288429/files/features_header.tabular
</code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-pre-process-data-and-create-pointwise-boxplot"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Pre-process data and create pointwise boxplot</h3>
  <p><strong>Load, Smooth and Plot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Load, Smooth and Plot</strong> with:</p>
  <ul>
    <li>Select the two region datasets <code class="language-plaintext highlighter-rouge">ETn_fixed.<span class="notranslate">bed</span></code> and <code class="language-plaintext highlighter-rouge">Control.<span class="notranslate">bed</span></code>.</li>
    <li>Select the region header file <code class="language-plaintext highlighter-rouge">regions_header.tabular</code>.</li>
    <li>Select the feature file <code class="language-plaintext highlighter-rouge">Recombination_hotspot.txt</code>.</li>
    <li>Select the feature header file <code class="language-plaintext highlighter-rouge">features_header.tabular</code>.</li>
    <li>Start positions are 0-based in this example; all region alignments are equivalent in this case.</li>
    <li>You can choose to smooth the feature curves.</li>
    <li>Select the graphical representation type <strong>Pointwise quantile curves (boxplot)</strong>.</li>
  </ul>

  <p><img src="../../images/load_smooth_and_plot_output.png" alt="Recombination hotspot pointwise boxplot" /></p>
</blockquote>

<h1 id="step-2-performing-interval-wise-testing">Step 2: Performing Interval-Wise Testing</h1>

<p>The second tool (IWTomics Test and Plot) statistically evaluates differences in genomic features between groups of regions along the genome. In particular, it implements the Interval-Wise Testing for omics data, an extended version of the Interval-Wise Testing for functional data presented in <a href="https://doi.org/10.1080/10485252.2017.1306627">Pini and Vantini (2017)</a>.</p>

<p>It allows to perform multiple two sample permutation tests between pairs of region datasets, on several features. It returns the adjusted p-value curves for every test and all possible scales. Moreover, it creates a graphical representation of the Interval-Wise Testing results and a summary plot (optional) with p-values at the maximum scale. The tool IWTomics Plot with Threshold on Test Scale permits to select the scale to be used in the plots.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-test-for-difference-between-etn-and-control-regions"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Test for difference between ETn and Control regions</h3>
  <p><strong>Test and Plot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Test and Plot</strong> with:</p>
  <ul>
    <li>Select the <code class="language-plaintext highlighter-rouge">IWTomicsData Object</code>, <code class="language-plaintext highlighter-rouge">Region Dataset IDs</code>, and <code class="language-plaintext highlighter-rouge">Feature IDs</code> created during <strong>Step 1</strong>.</li>
    <li>Select pairs of genomic region datasets to be compared using two-sample Interval-Wise Testing. In this example we consider one two-sample test contrasting <code class="language-plaintext highlighter-rouge">ETn_fixed</code> (select it as <strong>Region 1</strong>) to <code class="language-plaintext highlighter-rouge">Control</code> (select it as <strong>Region 2</strong>).</li>
    <li>Select feature measurements to be tested. In this example, select <code class="language-plaintext highlighter-rouge">Recomb_hot</code> feature.</li>
    <li>You can choose the test statistic, the number of permutations, and the significance level of the test (alpha).</li>
    <li>Select the graphical representation type <strong>Pointwise quantile curves (boxplot)</strong>.</li>
  </ul>

  <p><img src="../../images/test_and_plot_output.png" alt="Recombination hotspot IWT results" /></p>
</blockquote>

<h1 id="step-3-selecting-test-scale">Step 3: Selecting test scale</h1>

<p>The third tool (IWTomics Plot with Threshold on Test Scale) allows to select the scale for the Interval-Wise Testing results. In particular it returns the p-value curves for the different tests performed at the selected scale, and it creates a graphical representation of the Interval-Wise Testing results and a summary plot (optional) at the selected scale.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-change-scale-for-test-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Change scale for test results</h3>
  <p><strong>Plot with Threshold on Test Scale</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>Plot with Threshold on Test Scale</strong> with:</p>
  <ul>
    <li>Select the <code class="language-plaintext highlighter-rouge">IWTomicsData Object with Test Results</code>, <code class="language-plaintext highlighter-rouge">Test IDs</code>, and <code class="language-plaintext highlighter-rouge">Feature IDs</code> created during <strong>Step 2</strong>.</li>
    <li>Select the scale to be used in Interval-Wise Testing results and plot. Select the test identifier <code class="language-plaintext highlighter-rouge">ETn_fixed vs Control</code>, the feature identifier <code class="language-plaintext highlighter-rouge">Recomb_hot</code>, and the scale <code class="language-plaintext highlighter-rouge">8</code> (8-kb scale).</li>
    <li>Select the graphical representation type <strong>Pointwise quantile curves (boxplot)</strong>.</li>
  </ul>

  <p><img src="../../images/plot_with_threshold_on_test_scale_output.png" alt="Recombination hotspot IWT results scale 8" /></p>
</blockquote>

<blockquote class="details">
  <h3 id="details-additional-resources"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Additional resources</h3>

  <p>Read more about <strong>IWTomics</strong> <a href="https://bioconductor.org/packages/release/bioc/vignettes/IWTomics/inst/doc/IWTomics.pdf">here</a>.</p>
</blockquote>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>High-resolution omics data can be represented as curves along the genome</p>
</li>
                            
                            <li><p>IWTomics permits to visualize these data</p>
</li>
                            
                            <li><p>The Interval-Wise Testing detects statistically significant differences between pairs of genomic regions</p>
</li>
                            
                            <li><p>The Interval-Wise Testing corrects p-values taking into consideration the ordered nature of measurements along the genome</p>
</li>
                            
                            <li><p>Locations and scales do not need to be specified at the outset, indeed IWTomics permits to detect them</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Pini2017">Pini, A., and S. Vantini, 2017 <b>Interval-wise testing for functional data</b>. Journal of Nonparametric Statistics 29: 407‚Äì424.</span>
    
    
    <a href="https://doi.org/10.1080/10485252.2017.1306627">10.1080/10485252.2017.1306627</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Cremona2018">Cremona, M. A., A. Pini, F. Cumbo, K. D. Makova, F. Chiaromonte <i>et al.</i>, 2018 <b>IWTomics: testing high-resolution sequence-based ‚ÄòOmics‚Äô data at multiple locations and scales</b> (I. Birol, Ed.). Bioinformatics 34: 2289‚Äì2291.</span>
    
    
    <a href="https://doi.org/10.1093/bioinformatics/bty090">10.1093/bioinformatics/bty090</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Interval-Wise Testing for omics data (Statistics and machine learning)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Marzia A Cremona, Fabio Cumbo, 2021 <b>Interval-Wise Testing for omics data (Galaxy Training Materials)</b>. <a href="/training-material/topics/statistics/tutorials/iwtomics/tutorial.html">/training-material/topics/statistics/tutorials/iwtomics/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{statistics-iwtomics,
    author = "Marzia A Cremona and Fabio Cumbo",
    title = "Interval-Wise Testing for omics data (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/statistics/tutorials/iwtomics/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Marzia A Cremona, Fabio Cumbo)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/statistics/tutorials/iwtomics/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
