<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation / Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition / Microbiome</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@43bed0043c7b744f5a6074fbdafe6dd7d31c4834",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="1970-01-01 00:00:00 +0000"><meta name="description" content="A microbiome is the community of microorganisms that can usually be found living together in any given habitat. Microbiome research has grown substantially over the past decade in terms of the range of biomes sampled, identified taxa, and the volume of data derived from the samples. ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Microbiome / Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition / Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation">
        <meta property="og:description" content="A microbiome is the community of microorganisms that can usually be found living together in any given habitat. Microbiome research has grown substantially over the past decade in terms of the range of biomes sampled, identified taxa, and the volume of data derived from the samples. ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.png">
	</head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/microbiome/faqs/" title="Check our FAQs for the Microbiome topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<style>
hgroup {margin-bottom: 2rem; border-bottom: 1px solid var(--border-light);}
hgroup p { font-family: monospace; font-size: 0.8rem;}
.meta ul { padding-left: 1em;}
</style>

<section>
	<hgroup>
		<h1 data-toc-skip>Pathogen Detection PathoGFAIR Samples Aggregation and Visualisation</h1>
		<p>microbiome-pathogen-detection-from-nanopore-foodborne-data/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation</p>
	</hgroup>
	
	<div class="row">
		<div class="col-md-3 meta">
			
			<b>Author(s)</b>
			<br/>
			
			Engy Nasr, 
			
			Bérénice Batut, 
			
			Paul Zierep
			
			
			<br/>

			<i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span>
			<b>Version</b>
			<br/>
			2

			<br/>

			<i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span>
			<b>Last updated</b>
			<br/>
			Jun 7, 2024

			<br/>

			<i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">license</span>
			<b>License</b>
			<br/>

			
			<a href="https://spdx.org/licenses/MIT">MIT</a>
			
			<br/>

			
			<i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>
			<b>Tags</b>
			<br/>

			
			<div class="label label-default tutorial_tag" style="--color-primary: #8cc2ff; --color-darker: #52afff; --color-dimmed: #73b9ff;">name:Collection</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #00d7b0; --color-darker: #00bf9d; --color-dimmed: #00cba7;">name:microGalaxy</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #ffa988; --color-darker: #ff8a46; --color-dimmed: #ff9a6c;">name:PathoGFAIR</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #ffa6ad; --color-darker: #ff848f; --color-dimmed: #ff959e;">name:IWC</div>
			
			</br>
			


			<b>Features</b>
			<ul>
			
			
				<li>Includes a <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/workflows_report_view.html">Galaxy Workflow Report</a></li>
			
			
				<li>Uses <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/workflows_comments.html">Galaxy Workflow Comments</a></li>
			
			
			</ul>

			<hr/>


			<b>Tutorial</b>
			<br/>
			<i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> <a href="../tutorial.html">Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition</a>
			</br>

			
			
			<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> <a href="index.html">Other workflows associated with this material</a>
			


			<hr/>
			<b>Workflow Testing</b>
			<br/>

			Tests: ❌ <br/>
			Results: Not yet automated

			<hr/>

			<b>FAIRness</b>

			<i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span>
			<b><abbr title="Persistent URL">PURL</abbr></b>
			<br/>
			<a href="https://gxy.io/GTN:W00144">https://gxy.io/GTN:W00144</a>
			<br/>

			
			<a style="margin-bottom: 0.2rem" href="/training-material/api/workflows/microbiome/pathogen-detection-from-nanopore-foodborne-data/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation/rocrate.zip" class="btn btn-secondary"><img src="/training-material/assets/images/ro-crate.png" style="height: 32px" alt="RO-Crate logo with flask"/> Download Workflow RO-Crate</a>
			

			
			<a style="margin-bottom: 0.2rem" href="https://dev.workflowhub.eu/workflows/1077" class="btn btn-secondary"><img src="/training-material/assets/images/workflowhub-icon.svg" style="height: 32px" alt="Workflowhub cloud with gears logo"/> View on (Dev) WorkflowHub</a>
			

		</div>
		<div class="col-md-9">
			<span tabindex="0" role="button" class="workflow show-when-galaxy-proxy-active" data-workflow="https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/microbiome-pathogen-detection-from-nanopore-foodborne-data/versions/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation">Launch in Tutorial Mode</span>
			<a href="/training-material/faqs/galaxy/tutorial_mode.html" class="show-when-galaxy-proxy-active"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span></a>
			<br/>

			<span class="dropdown">
				<button class="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown" aria-expanded="false">
					<img src="/training-material/assets/images/galaxylogo.png" style="height: 16px" alt="galaxy logo"/> 
					Run Workflow in Galaxy
				</button>
				<ul class="dropdown-menu">
					
					
					<li class="dropdown-item"><a href="https://usegalaxy.org/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/microbiome-pathogen-detection-from-nanopore-foodborne-data/versions/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation">Galaxy Main</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.org.au/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/microbiome-pathogen-detection-from-nanopore-foodborne-data/versions/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation">Galaxy Australia</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.fr/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/microbiome-pathogen-detection-from-nanopore-foodborne-data/versions/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation">Galaxy France</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.eu/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/microbiome-pathogen-detection-from-nanopore-foodborne-data/versions/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation">Galaxy Europe</a></li>
					
					<li class="dropdown-item hide-when-galaxy-proxy-active">
						<a href="https://my.galaxy.training/?path=/workflows/trs_import%3Frun_form=true%26trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/microbiome-pathogen-detection-from-nanopore-foodborne-data/versions/pathogen-detection-pathogfair-samples-aggrtion-and-visualisation">Import to another server (≥23.0+ only!)</a>
					</li>
				</ul>
			</span>
			<a class="btn btn-primary" href="https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.ga">
				<i class="fas fa-download" aria-hidden="true"></i><span class="visually-hidden">galaxy-download</span>
				Download
			</a>
			<pre class="mermaid">flowchart TD
  0["ℹ️ Input Collection\namr_identified_by_ncbi"];
  style 0 stroke:#2c3143,stroke-width:4px;
  1["ℹ️ Input Collection\nvfs_of_genes_identified_by_vfdb"];
  style 1 stroke:#2c3143,stroke-width:4px;
  2["ℹ️ Input Dataset\nmetadata"];
  style 2 stroke:#2c3143,stroke-width:4px;
  3["ℹ️ Input Dataset\nremoved_hosts_percentage_tabular"];
  style 3 stroke:#2c3143,stroke-width:4px;
  4["ℹ️ Input Dataset\nmapping_mean_depth_per_sample"];
  style 4 stroke:#2c3143,stroke-width:4px;
  5["ℹ️ Input Collection\namrs"];
  style 5 stroke:#2c3143,stroke-width:4px;
  6["ℹ️ Input Dataset\nmapping_coverage_percentage_per_sample"];
  style 6 stroke:#2c3143,stroke-width:4px;
  7["ℹ️ Input Dataset\nnumber_of_variants_per_sample"];
  style 7 stroke:#2c3143,stroke-width:4px;
  8["ℹ️ Input Collection\ncontigs"];
  style 8 stroke:#2c3143,stroke-width:4px;
  9["ℹ️ Input Collection\nvfs"];
  style 9 stroke:#2c3143,stroke-width:4px;
  10["Filter failed datasets"];
  0 -->|output| 10;
  11["Filter failed datasets"];
  1 -->|output| 11;
  12["Bar chart"];
  3 -->|output| 12;
  0961166e-38b6-44e3-93db-100d821b0070["Output\nnumber_of_reads_before_host_removal_and_number_of_host_reads_found_per_sample_fig"];
  12 --> 0961166e-38b6-44e3-93db-100d821b0070;
  style 0961166e-38b6-44e3-93db-100d821b0070 stroke:#2c3143,stroke-width:4px;
  13["Bar chart"];
  3 -->|output| 13;
  548975e5-4619-49fc-9e95-4d7f4d761dfd["Output\nremoved_host_percentage_fig"];
  13 --> 548975e5-4619-49fc-9e95-4d7f4d761dfd;
  style 548975e5-4619-49fc-9e95-4d7f4d761dfd stroke:#2c3143,stroke-width:4px;
  14["Bar chart"];
  4 -->|output| 14;
  a71ebb67-1154-4f25-a62d-e8fa2b839e2e["Output\nmapping_mean_depth_per_sample_fig"];
  14 --> a71ebb67-1154-4f25-a62d-e8fa2b839e2e;
  style a71ebb67-1154-4f25-a62d-e8fa2b839e2e stroke:#2c3143,stroke-width:4px;
  15["Filter failed datasets"];
  5 -->|output| 15;
  16["Bar chart"];
  6 -->|output| 16;
  2567360e-d39f-4316-84f5-77aaf8e8198c["Output\nmapping_coverage_percentage_per_sample_fig"];
  16 --> 2567360e-d39f-4316-84f5-77aaf8e8198c;
  style 2567360e-d39f-4316-84f5-77aaf8e8198c stroke:#2c3143,stroke-width:4px;
  17["Bar chart"];
  7 -->|output| 17;
  68c76f06-d2ea-4280-9e6f-a7b4f1568389["Output\nnumber_of_Variants_and_SNPs_indentified_fig"];
  17 --> 68c76f06-d2ea-4280-9e6f-a7b4f1568389;
  style 68c76f06-d2ea-4280-9e6f-a7b4f1568389 stroke:#2c3143,stroke-width:4px;
  18["Filter failed datasets"];
  8 -->|output| 18;
  19["Filter failed datasets"];
  9 -->|output| 19;
  20["Remove beginning"];
  10 -->|output| 20;
  21["Remove beginning"];
  11 -->|output| 21;
  22["Remove beginning"];
  15 -->|output| 22;
  23["Collapse Collection"];
  18 -->|output| 23;
  86910e39-57bf-4a76-ac4c-739340fd2387["Output\nall_samples_contigs_in_one_fasta_file"];
  23 --> 86910e39-57bf-4a76-ac4c-739340fd2387;
  style 86910e39-57bf-4a76-ac4c-739340fd2387 stroke:#2c3143,stroke-width:4px;
  24["Collapse Collection"];
  19 -->|output| 24;
  02b996c6-a912-4e4e-b3ec-49601faaa452["Output\nall_vfs_in_one_tabular"];
  24 --> 02b996c6-a912-4e4e-b3ec-49601faaa452;
  style 02b996c6-a912-4e4e-b3ec-49601faaa452 stroke:#2c3143,stroke-width:4px;
  25["Remove beginning"];
  19 -->|output| 25;
  26["Count"];
  20 -->|out_file1| 26;
  27["Count"];
  21 -->|out_file1| 27;
  28["Group"];
  21 -->|out_file1| 28;
  29["Unique"];
  22 -->|out_file1| 29;
  30["Split by group"];
  24 -->|output| 30;
  59f8cb09-424b-47b1-b94b-a612c2610cab["Output\nsplit_by_group_collection"];
  30 --> 59f8cb09-424b-47b1-b94b-a612c2610cab;
  style 59f8cb09-424b-47b1-b94b-a612c2610cab stroke:#2c3143,stroke-width:4px;
  31["Unique"];
  25 -->|out_file1| 31;
  32["Cut"];
  26 -->|out_file1| 32;
  33["Cut"];
  27 -->|out_file1| 33;
  34["Filter empty datasets"];
  28 -->|out_file1| 34;
  35["Cut"];
  29 -->|outfile| 35;
  36["Cut"];
  30 -->|split_output| 36;
  eeb25a51-ea21-4a19-a196-55d5bd919b10["Output\nadjusted_abricate_vfs_tabular_part1"];
  36 --> eeb25a51-ea21-4a19-a196-55d5bd919b10;
  style eeb25a51-ea21-4a19-a196-55d5bd919b10 stroke:#2c3143,stroke-width:4px;
  37["Cut"];
  31 -->|outfile| 37;
  38["Collapse Collection"];
  32 -->|out_file1| 38;
  39["Collapse Collection"];
  33 -->|out_file1| 39;
  40["Column join"];
  34 -->|output| 40;
  41["bedtools getfasta"];
  23 -->|output| 41;
  35 -->|out_file1| 41;
  42["Remove beginning"];
  36 -->|out_file1| 42;
  aaaa4446-0817-4e5c-aa1b-9ec384f2a363["Output\nadjusted_abricate_vfs_tabular_part2"];
  42 --> aaaa4446-0817-4e5c-aa1b-9ec384f2a363;
  style aaaa4446-0817-4e5c-aa1b-9ec384f2a363 stroke:#2c3143,stroke-width:4px;
  43["bedtools getfasta"];
  23 -->|output| 43;
  37 -->|out_file1| 43;
  44["Column Regex Find And Replace"];
  38 -->|output| 44;
  4809c36b-31ef-4664-8e4e-47f0f72152de["Output\namrs_count"];
  44 --> 4809c36b-31ef-4664-8e4e-47f0f72152de;
  style 4809c36b-31ef-4664-8e4e-47f0f72152de stroke:#2c3143,stroke-width:4px;
  45["Column Regex Find And Replace"];
  39 -->|output| 45;
  87efc81d-4d84-4af3-831f-dfe033c59f78["Output\nvfs_count"];
  45 --> 87efc81d-4d84-4af3-831f-dfe033c59f78;
  style 87efc81d-4d84-4af3-831f-dfe033c59f78 stroke:#2c3143,stroke-width:4px;
  46["Column Regex Find And Replace"];
  40 -->|tabular_output| 46;
  f5c221e3-00ef-4834-9a5f-a94c97fd6764["Output\nheatmap_table"];
  46 --> f5c221e3-00ef-4834-9a5f-a94c97fd6764;
  style f5c221e3-00ef-4834-9a5f-a94c97fd6764 stroke:#2c3143,stroke-width:4px;
  47["Regex Find And Replace"];
  41 -->|output| 47;
  48["bedtools getfasta"];
  23 -->|output| 48;
  42 -->|out_file1| 48;
  82ce2107-89a3-438c-95bb-dc871b5258b7["Output\nfiltered_sequences_with_vfs_fasta"];
  48 --> 82ce2107-89a3-438c-95bb-dc871b5258b7;
  style 82ce2107-89a3-438c-95bb-dc871b5258b7 stroke:#2c3143,stroke-width:4px;
  49["Regex Find And Replace"];
  43 -->|output| 49;
  50["Multi-Join"];
  45 -->|out_file1| 50;
  44 -->|out_file1| 50;
  51["Heatmap w ggplot"];
  46 -->|out_file1| 51;
  c0417c91-a513-4c6a-9a62-3aac2f1f8e85["Output\nheatmap_pdf"];
  51 --> c0417c91-a513-4c6a-9a62-3aac2f1f8e85;
  style c0417c91-a513-4c6a-9a62-3aac2f1f8e85 stroke:#2c3143,stroke-width:4px;
  97816bc2-fd0c-4077-a721-8dd1470879d1["Output\nheatmap_png"];
  51 --> 97816bc2-fd0c-4077-a721-8dd1470879d1;
  style 97816bc2-fd0c-4077-a721-8dd1470879d1 stroke:#2c3143,stroke-width:4px;
  52["Filter empty datasets"];
  47 -->|out_file1| 52;
  53["ClustalW"];
  48 -->|output| 53;
  9b7bd78c-f112-480b-a7df-c10711af254c["Output\nclustalw_on_input_dnd"];
  53 --> 9b7bd78c-f112-480b-a7df-c10711af254c;
  style 9b7bd78c-f112-480b-a7df-c10711af254c stroke:#2c3143,stroke-width:4px;
  6af20322-24da-4036-80de-37bed2d25848["Output\nclustalw_on_input_clustal"];
  53 --> 6af20322-24da-4036-80de-37bed2d25848;
  style 6af20322-24da-4036-80de-37bed2d25848 stroke:#2c3143,stroke-width:4px;
  54["Filter empty datasets"];
  49 -->|out_file1| 54;
  55["Replace Text"];
  50 -->|outfile| 55;
  91745da0-0b8d-4a7a-b927-b36107f17ec5["Output\nvfs_amrs_count_table"];
  55 --> 91745da0-0b8d-4a7a-b927-b36107f17ec5;
  style 91745da0-0b8d-4a7a-b927-b36107f17ec5 stroke:#2c3143,stroke-width:4px;
  56["FASTA-to-Tabular"];
  52 -->|output| 56;
  57["Filter empty datasets"];
  53 -->|output| 57;
  32d00c1d-68c4-4069-8d5d-023aabdfadbe["Output\nfiltered_empty_datasets"];
  57 --> 32d00c1d-68c4-4069-8d5d-023aabdfadbe;
  style 32d00c1d-68c4-4069-8d5d-023aabdfadbe stroke:#2c3143,stroke-width:4px;
  58["FASTA-to-Tabular"];
  54 -->|output| 58;
  59["Cut"];
  56 -->|output| 59;
  60["FASTTREE"];
  57 -->|output| 60;
  aacdfe45-eb0c-4f6e-a479-eeb170774757["Output\nfasttree_nhx"];
  60 --> aacdfe45-eb0c-4f6e-a479-eeb170774757;
  style aacdfe45-eb0c-4f6e-a479-eeb170774757 stroke:#2c3143,stroke-width:4px;
  61["Cut"];
  58 -->|output| 61;
  62["Group"];
  59 -->|out_file1| 62;
  63["Newick Display"];
  60 -->|output| 63;
  0c22178c-dc85-4137-80e2-f3040b92bd20["Output\nnewick_genes_tree_graphs_collection"];
  63 --> 0c22178c-dc85-4137-80e2-f3040b92bd20;
  style 0c22178c-dc85-4137-80e2-f3040b92bd20 stroke:#2c3143,stroke-width:4px;
  64["Group"];
  61 -->|out_file1| 64;
  65["Tabular-to-FASTA"];
  62 -->|out_file1| 65;
  66["Tabular-to-FASTA"];
  64 -->|out_file1| 66;
  67["FASTA Merge Files and Filter Unique Sequences"];
  65 -->|output| 67;
  68["FASTA Merge Files and Filter Unique Sequences"];
  66 -->|output| 68;
  69["ClustalW"];
  67 -->|output| 69;
  70["ClustalW"];
  68 -->|output| 70;
  71["FASTTREE"];
  69 -->|output| 71;
  72["FASTTREE"];
  70 -->|output| 72;
  73["Newick Display"];
  71 -->|output| 73;
  1f9cb2cf-219f-48de-8058-d6d45f3b3158["Output\nall_samples_phylogenetic_tree_based_amrs"];
  73 --> 1f9cb2cf-219f-48de-8058-d6d45f3b3158;
  style 1f9cb2cf-219f-48de-8058-d6d45f3b3158 stroke:#2c3143,stroke-width:4px;
  74["Newick Display"];
  72 -->|output| 74;
  6bb4b32b-7cca-4e04-8120-be9f64ccba39["Output\nall_samples_phylogenetic_tree_based_vfs"];
  74 --> 6bb4b32b-7cca-4e04-8120-be9f64ccba39;
  style 6bb4b32b-7cca-4e04-8120-be9f64ccba39 stroke:#2c3143,stroke-width:4px;</pre>
			<h2>Inputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Input</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>Input dataset collection</td>
					<td>amr_identified_by_ncbi</td>
				</tr>
				
				<tr>
					<td>Input dataset collection</td>
					<td>vfs_of_genes_identified_by_vfdb</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>metadata</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>removed_hosts_percentage_tabular</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>mapping_mean_depth_per_sample</td>
				</tr>
				
				<tr>
					<td>Input dataset collection</td>
					<td>amrs</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>mapping_coverage_percentage_per_sample</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>number_of_variants_per_sample</td>
				</tr>
				
				<tr>
					<td>Input dataset collection</td>
					<td>contigs</td>
				</tr>
				
				<tr>
					<td>Input dataset collection</td>
					<td>vfs</td>
				</tr>
				
				</tbody>
			</table>

			<h2>Outputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>From</th>
						<th>Output</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>barchart_gnuplot</td>
					<td>Bar chart</td>
					<td></td>
				</tr>
				
				<tr>
					<td>barchart_gnuplot</td>
					<td>Bar chart</td>
					<td></td>
				</tr>
				
				<tr>
					<td>barchart_gnuplot</td>
					<td>Bar chart</td>
					<td></td>
				</tr>
				
				<tr>
					<td>barchart_gnuplot</td>
					<td>Bar chart</td>
					<td></td>
				</tr>
				
				<tr>
					<td>barchart_gnuplot</td>
					<td>Bar chart</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</td>
					<td>Collapse Collection</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</td>
					<td>Collapse Collection</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6</td>
					<td>Split by group</td>
					<td></td>
				</tr>
				
				<tr>
					<td>Cut1</td>
					<td>Cut</td>
					<td></td>
				</tr>
				
				<tr>
					<td>Remove beginning1</td>
					<td>Remove beginning</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3</td>
					<td>Column Regex Find And Replace</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3</td>
					<td>Column Regex Find And Replace</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3</td>
					<td>Column Regex Find And Replace</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1</td>
					<td>bedtools getfasta</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0</td>
					<td>Heatmap w ggplot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1</td>
					<td>ClustalW</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1</td>
					<td>Replace Text</td>
					<td></td>
				</tr>
				
				<tr>
					<td>__FILTER_EMPTY_DATASETS__</td>
					<td>Filter empty datasets</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1</td>
					<td>FASTTREE</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1</td>
					<td>Newick Display</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1</td>
					<td>Newick Display</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1</td>
					<td>Newick Display</td>
					<td></td>
				</tr>
				
				</tbody>
			</table>


			<h2>Tools</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Tool</th>
						<th>Links</th>
					</tr>
				</thead>
				<tbody>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Count1">Count1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Cut1">Cut1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Grouping1">Grouping1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Remove beginning1">Remove beginning1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=__FILTER_EMPTY_DATASETS__">__FILTER_EMPTY_DATASETS__</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=__FILTER_FAILED_DATASETS__">__FILTER_FAILED_DATASETS__</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=barchart_gnuplot">barchart_gnuplot</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6">toolshed.g2.bx.psu.edu/repos/bgruening/split_file_on_column/tp_split_on_column/0.6</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/bgruening/split_file_on_column/ff2a81aa3f08">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/9.3+galaxy1">toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/9.3+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/bgruening/text_processing/86755160afbf">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1">toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/9.3+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/bgruening/text_processing/86755160afbf">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1">toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/9.3+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/bgruening/text_processing/86755160afbf">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1">toolshed.g2.bx.psu.edu/repos/devteam/clustalw/clustalw/2.1+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/devteam/clustalw/5b097fde95dd">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1">toolshed.g2.bx.psu.edu/repos/devteam/fasta_to_tabular/fasta2tab/1.1.1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/devteam/fasta_to_tabular/e7ed3c310b74">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1">toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/devteam/tabular_to_fasta/0a7799698fe5">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0">toolshed.g2.bx.psu.edu/repos/galaxyp/fasta_merge_files_and_filter_unique_sequences/fasta_merge_files_and_filter_unique_sequences/1.2.0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/galaxyp/fasta_merge_files_and_filter_unique_sequences/f546e7278f04">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3">toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/galaxyp/regex_find_replace/503bcd6ebe4b">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3">toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/galaxyp/regex_find_replace/503bcd6ebe4b">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.30.0+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/bedtools/a1a923cd89e8">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3">toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/collection_column_join/3ddd99c7efee">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10+galaxy1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0">toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_heatmap/ggplot2_heatmap/3.4.0+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/ggplot2_heatmap/10515715c940">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/newick_utils/newick_display/1.6+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/newick_utils/b4163d2f64ab">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0">toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/5.1.0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/nml/collapse_collections/90981f86000f">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
				</tbody>
			</table>


			<p>To use these workflows in Galaxy you can either click the links to download the workflows, or you can right-click and copy the link to the workflow which can be used in the Galaxy form to import workflows.</p>


			<h2>Importing into Galaxy</h2>
			Below are the instructions for importing these workflows directly into your Galaxy server of choice to start using them!

			<!--SNIPPET--><blockquote class="hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-a-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing a workflow</div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote> <!--END_SNIPPET-->

			<h2>Version History</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Version</th>
						<th>Commit</th>
						<th>Time</th>
						<th>Comments</th>
					</tr>
				</thead>
				<tbody>
					
					<tr>
						<td>3</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/cdd93376adc6ce80742881c25d6810bcce85fc05">cdd93376a</a></td>
						<td>2024-06-06 12:00:29</td>
						<td>adding tags to some of the workflow outputs, updating the training with the latest PathoGFAIR  workflows updates</td>
					</tr>
					
					<tr>
						<td>2</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/211b69394d6ef4d129eaa18cd76b93c8dece8cf3">211b69394</a></td>
						<td>2024-05-26 09:45:27</td>
						<td>adding workflow reports to the workflows of the training to match the latest version of the IWC PR</td>
					</tr>
					
					<tr>
						<td>1</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/d320748c5c8ea425a77aa03ac78a5956eb75b005">d320748c5</a></td>
						<td>2024-05-20 18:17:48</td>
						<td>Foodborne training update 2024</td>
					</tr>
					
				</tbody>
			</table>


			<h2>For Admins</h2>
			<h3>Installing the workflow tools</h3>

			<pre>
wget https://training.galaxyproject.org/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.ga -O workflow.ga
workflow-to-tools -w workflow.ga -o tools.yaml
shed-tools install -g GALAXY -a API_KEY -t tools.yaml
workflow-install -g GALAXY -a API_KEY -w workflow.ga --publish-workflows
			</pre>
		</div>
	</div>
</section>


</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:W00144">gxy.io/GTN:W00144</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/workflows/pathogen_detection_pathoGFAIR_samples_aggregation_and_visualisation.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/43bed0043c7b744f5a6074fbdafe6dd7d31c4834">43bed00</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	
	<script src="/training-material/assets/js/mermaid@10.6.1.js" async></script>
	<link rel="preload" href="/training-material/assets/js/mermaid@10.6.1.js" as="script" crossorigin>
	

    </body>
</html>
