<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>FAQs</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@715a5f0aefd15f73c5a3222b576a5f6c21ada9c5",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/all-faq.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2021-03-24 13:50:23 +0000"><meta name="description" content="Collection of tutorials developed and maintained by the worldwide Galaxy community">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="FAQs">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the worldwide Galaxy community">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/all-faq.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/all-faq.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<section class="faqs">

   <nav aria-label="breadcrumb">
    <ol class="breadcrumb">

     <li class="breadcrumb-item "><a href="/training-material/faqs">All FAQs</a></li>
     
     <li class="breadcrumb-item "><a href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
     
     

     
     </ol>
    </nav>

    <h1 data-toc-skip>Frequently Asked Questions</h1>
    
    <div class="container">
        <div class="row">

            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h3 id="tutorial-faqs"> <a href="#tutorial-faqs"> Tutorial Questions </a></h3>

<hr/>



<h4 class="faq-area"> Additional resources to learn more about proteomic data analysis <a href="/training-material/topics/proteomics/faqs/resources.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-additional-resources-to-learn-more-about-proteomic-data-analysis"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-additional-resources-to-learn-more-about-proteomic-data-analysis" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Additional resources to learn more about proteomic data analysis</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To learn more about proteomic data analysis, we suggest you look at:</p>   <ul>   <li><a href="https://video.ucdavis.edu/playlist/details/0_4jkc4swu">UC Davis Proteomics Online Short Course </a></li>   <li><a href="https://www.youtube.com/c/MayInstituteNEU/videos">May Institute YouTube channel</a></li>   <li><a href="http://www.kilgourlab.com/teaching-resources/">Kilgour Lab Training Resources</a></li>   <li><a href="https://timms.uni-tuebingen.de/List/List?id=UT_201%5B45%5D_____00%5B12%5D_cpm_000_&amp;Headline=Computational%20Proteomics%20and%20Metabolomics">Tübingen University Computational Proteomics and Metabolomics Lessons</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> After sequencing with MinKNOW software, we get many fastq files, do these files need to be combined into one file before uploading or is it possible to upload them all at once? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/compining_fastq_files_of_one_sample.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-after-sequencing-with-minknow-software-we-get-many-fastq-files-do-these-files-need-to-be-combined-into-one-file-before-uploading-or-is-it-possible-to-upload-them-all-at-once"><i class="far fa-question-circle" aria-hidden="true"></i> Question: After sequencing with MinKNOW software, we get many fastq files, do these files need to be combined into one file before uploading or is it possible to upload them all at once?</div>   <p>After sequencing with MinKNOW software, it is a good approach to combine the files from the same run before processing them. You could create a collection per run with all fastq files and then use the collection operation to concatenate all files in a collection.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> AnnData Import/ AnnData Manipulate not working? <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/anndata_not_working.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-anndata-import-anndata-manipulate-not-working"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-anndata-import-anndata-manipulate-not-working" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: AnnData Import/ AnnData Manipulate not working?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This is a known issue, please do not use version <code class="language-plaintext highlighter-rouge">0.7.4</code> of the tool, and use version <code class="language-plaintext highlighter-rouge">0.6.2</code> instead. The  Inspect AnnData tool should work fine however.</p>   <p><a href="/training-material/faqs/galaxy/tools_change_version.html">Switching tool versions</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Are Barcodes always on R1 and Sequence data on R2? <a href="/training-material/topics/single-cell/tutorials/scrna-umis/faqs/r1r2.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-are-barcodes-always-on-r1-and-sequence-data-on-r2"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Are Barcodes always on R1 and Sequence data on R2?</div>   <p>No, it really depends on the protocol. In some protocols this convention is swapped, in others the barcodes can be distributed across both reads.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Are these data free to use and download? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/terms-of-usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-are-these-data-free-to-use-and-download"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Are these data free to use and download?</div>   <p>Yes, the metadata, aligned reads, and other SARS-CoV-2 data that is mentioned in this training are free to download and have no associated egress charges.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Automatically trim adapters (without providing custom sequences) <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/adapter_trimming.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-automatically-trim-adapters-without-providing-custom-sequences"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-automatically-trim-adapters-without-providing-custom-sequences" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Automatically trim adapters (without providing custom sequences)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are many tools for this: Trimmomatic, Trim Galore, and a few others (search: “Trim”). In some of these there are options to automatically trim adaptors, but they are not so specific to the sequence you are working on necessarily.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can EncyclopeDIA be run on a DIA-MS dataset without a spectral library? <a href="/training-material/topics/proteomics/tutorials/encyclopedia/faqs/without_spectral_library.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-encyclopedia-be-run-on-a-dia-ms-dataset-without-a-spectral-library"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can EncyclopeDIA be run on a DIA-MS dataset without a spectral library?</div>   <p>Yes. In this GTN, the workflow presented is the Standard EncyclopeDIA workflow; however, there is a variation upon the Standard EncyclopeDIA workflow, named the WALNUT EncyclopeDIA workflow in which a spectral library is not required. Simply, the WALNUT variation of the workflow omits the DLIB spectral/PROSIT library input, hence requiring just the GPF DIA dataset collection, Experimental DIA dataset collection, and the FASTA Protein Database file. Therefore, the Chromatogram Library is generated using the GPF DIA dataset collection and the FASTA Protein Database alone. This method does generate fewer searches than if a spectral library is used. The Galaxy-P team tested the efficacy of the WALNUT workflow compared to the Standard EncyclopeDIA workflow, and more information on that comparison and those results can be found at this <a href="http://galaxyp.org/wp-content/uploads/2020/10/BCC_2020-Pratik.pdf">link</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can I use alternative tools for the Quantification step? <a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/quantification_alternatives.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-i-use-alternative-tools-for-the-quantification-step"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can I use alternative tools for the Quantification step?</div>   <p>There are some alternatives to Salmon for reference transcriptome-based RNA quantification. <strong>Kallisto</strong> and <strong>Sailfish</strong> use a similar approach, known as pseudoalignment.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/reusing_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can I use these workflows on datasets generated from our laboratory?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can this ASaiM workflow be used for single-end data? <a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/single_end.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-this-asaim-workflow-be-used-for-single-end-data"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can this ASaiM workflow be used for single-end data?</div>   <p>Yes, the inputs have to be changed to a single-end file rather than a paired-end.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can we also use this workflow on Illumina raw reads? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/worflow_and_sequencing_techniques.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-we-also-use-this-workflow-on-illumina-raw-reads"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can we also use this workflow on Illumina raw reads?</div>   <p>Yes, some tools would need to be changed or removed:</p>   <ul>   <li>For the Preprocessing workflow, plotting with Nanoplot shall be removed and keep only FastQC, MultiQC and Fastp.</li>   <li>For the mapping in the SNP based pathogen detection workflow, instead of Minimap2, Bowtie can be used.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can we polish the assembly with long reads too? <a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/polish_longreads.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-we-polish-the-assembly-with-long-reads-too"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-we-polish-the-assembly-with-long-reads-too" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can we polish the assembly with long reads too?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Yes. In this tutorial, we only polish the assembly with the short reads. This may be enough for bacterial genomes. However, for an even better polish (usually), a common approach is to also polish the assembly with the long reads. A typical workflow for this would assemble with long reads, then polish with long reads (x 4 rounds, with Racon), polish with long reads again (x 1 round, with Medaka), then polish with short reads (x2 rounds with Pilon).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can we use snippy pipeline instead for the phylogenetic analysis? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/variant_calling_quality.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-we-use-snippy-pipeline-instead-for-the-phylogenetic-analysis"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can we use snippy pipeline instead for the phylogenetic analysis?</div>   <p>On principle yes. We did not try yet. Snippy is available in Galaxy</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can we use the ASaiM-MT workflow on multiple input files at the same time? <a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/multiple_inputs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-we-use-the-asaim-mt-workflow-on-multiple-input-files-at-the-same-time"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can we use the ASaiM-MT workflow on multiple input files at the same time?</div>   <p>Currently, that is one of its limitations. However, Galaxy offers a workflow within workflow feature which can help process multiple files at the same time and this output can be combined into one using the <code class="language-plaintext highlighter-rouge">MT2MQ</code> tool.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the heatmap colours <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/heatmap_colours.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-heatmap-colours"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-heatmap-colours" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the heatmap colours</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can change the heatmap color, by expanding the <strong>Show advanced options</strong> section. There are many options here, including setting the colors.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Could I use a different p-adj value for filtering differentially expressed genes? <a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/different_padj_value.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-could-i-use-a-different-p-adj-value-for-filtering-differentially-expressed-genes"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Could I use a different p-adj value for filtering differentially expressed genes?</div>   <p>Yes, you can modify this value, to perform a more rigorous analysis, or extend the range of genes selected. A higher p-value will significantly increase the number of genes selected, at the expense of including possible false positives.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Defining a Learning Pathway <a href="/training-material/topics/contributing/faqs/learning-pathways.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-defining-a-learning-pathway"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Defining a Learning Pathway</div>   <p><a href="/training-material/learning-pathways/">Learning Pathways</a> are sets of tutorials curated by community experts to form a coherent set of lessons around a topic, building up knowledge step by step.</p>   <p>To define a learning pathway, create a file in the <code class="language-plaintext highlighter-rouge">learning-pathways/</code> folder. An example file is also given in this folder (<a href="/training-material/learning-pathways/pathway-example.html">pathway-example.md</a>). It should look something like this:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>---<br>layout: learning-pathway<br><br>title: Title of your pathway<br>description: |<br>  Description of the pathway. What will be covered, what are the learning objectives, etc?<br>  Make this as thorough as possible, 1-2 paragraphs. This appears on the index page that<br>  lists all the learning paths, and at the top of the pathway page<br>tags: [some, keywords, here ]<br><br>cover-image: path/to/image.png # optional cover image, defaults to GTN logo<br>cover-image-alt: alt text for this image<br><br>pathway:<br>  - section: "Module 1: Title"<br>    description: |<br>      description of the module. What will be covered, what should learners expect, etc.<br>    tutorials:<br>      - name: galaxy-intro-short<br>        topic: introduction<br>      - name: galaxy-intro-101<br>        topic: introduction<br><br>  - section: "Module 2: Title"<br>    description: |<br>      description of the tutorial<br>      will be shown under the section title<br>    tutorials:<br>      - name: quality-control<br>        topic: sequence-analysis<br>      - name: mapping<br>        topic: sequence-analysis<br>      - name: general-introduction<br>        topic: assembly<br>      - name: chloroplast-assembly<br>        topic: assembly<br>      - name: "My non-GTN session"<br>        external: true<br>        link: "https://example.com"<br>        type: hands_on  # or 'slides'<br><br># you can make as many sections as you want, with as many tutorials as you want<br><br>---<br><br>You can put some extra information here. Markdown syntax can be used. This is shown after the description on the pathway page, but not on the cards on the index page.<br><br></code></pre></div></div>   <p>And that’s it!</p>   <p>We are happy to receive contributions of learning pathways! Did you teach a workshop around a topic using GTN materials? Capture the program as a learning pathways for others to reuse!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Do I have to run the tools in the order of the tutorial? <a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/tool_order.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-i-have-to-run-the-tools-in-the-order-of-the-tutorial"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do I have to run the tools in the order of the tutorial?</div>   <p>The tools are presented in the order that a typical analysis would use. If you want to run some tools in parallel (to save time) you can do so. <a href="https://usegalaxy.eu/u/pvanheus/w/m-tuberculosis-variant-analysis-tutorial-imported-from-uploaded-file">This workflow</a> illustrates the analysis done in the tutorial and shows that there are multiple “paths” leading to outputs that have some steps that could be run at the same time: MultiQC, Kraken2, JBrowse and TB Variant Report.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Do the pipelines work with both isolates and direct from raw meat? or only isolate? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/workflow_and_isolations.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-the-pipelines-work-with-both-isolates-and-direct-from-raw-meat-or-only-isolate"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do the pipelines work with both isolates and direct from raw meat? or only isolate?</div>   <p>The workflow can work with both isolates and raw meat. The workflow is designed to remove hosts before detecting any pathogen, so both isolates and raw meat samples are pre-processed equaliy before the analysis starts.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Do you have resources to help me get started working in the cloud? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/cloud.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-you-have-resources-to-help-me-get-started-working-in-the-cloud"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do you have resources to help me get started working in the cloud?</div>   <p>Yes, we have a number of documents and videos to help you start working with SRA data in the cloud:</p>   <ul>   <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud/">Cloud landing page</a></li>   <li><a href="https://www.youtube.com/watch?v=DkNz-RCCm-M">Exploring SRA metadata in BigQuery</a></li>   <li><a href="https://www.youtube.com/watch?v=_F4FhcDWSJg">Exploring SARS-CoV-2 metadata in Athena</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading the files from the NCBI server fails or takes too long. <a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/upload_slow.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-the-files-from-the-ncbi-server-fails-or-takes-too-long"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-the-files-from-the-ncbi-server-fails-or-takes-too-long" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading the files from the NCBI server fails or takes too long.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Download the data from Zenodo instead (see overview box at top of tutorial). This method uses Galaxy’s generic data import functionality, and is more reliable and faster than the download from NCBI.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> First job I submitted remains grey or running for a long time - is it broken? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/job_grey.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-first-job-i-submitted-remains-grey-or-running-for-a-long-time-is-it-broken"><i class="far fa-question-circle" aria-hidden="true"></i> Question: First job I submitted remains grey or running for a long time - is it broken?</div>   <ul>   <li>Check with <code class="language-plaintext highlighter-rouge">top</code> or your system monitor - if Conda is running, things are working but it’s slow the first time a dependency is installed.</li>   <li>The first run generally takes a while to install all the needed dependencies.</li>   <li>Subsequent runs should start immediately with all dependencies already in place.</li>   <li>Installing new Conda dependencies just takes time so tools that have new Conda packages will take longer to run the first time if they must be installed.</li>   <li>In general, a <code class="language-plaintext highlighter-rouge">planemo_test</code> job usually takes around a minute - planemo has to build and tear down a new Galaxy for generating test results and then again for testing properly. Longer if the tool has Conda dependencies.</li>   <li>The very first test in a fresh appliance may take 6 minutes so be patient.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> For preprocessing part with host removal: Where do you find the abbreviations for each host species available (e.g. bos is cow, homo is human..)? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_removal.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-for-preprocessing-part-with-host-removal-where-do-you-find-the-abbreviations-for-each-host-species-available-e-g-bos-is-cow-homo-is-human"><i class="far fa-question-circle" aria-hidden="true"></i> Question: For preprocessing part with host removal: Where do you find the abbreviations for each host species available (e.g. bos is cow, homo is human..)?</div>   <p>The abbreviation (i.e. the genus) is the first word in the <a href="https://github.com/lskatz/Kalamari/blob/master/src/Kalamari_v3.9.1.tsv">list of possible hosts</a>. The names are the scientific names for species, which would be shown on the taxonomy tree if you would look up the common name (i.e. bovine) on Wikipedia.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> From where can I import other genomes? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/importing_genomes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-from-where-can-i-import-other-genomes"><i class="far fa-question-circle" aria-hidden="true"></i> Question: From where can I import other genomes?</div>   <p>In this tutorial, we used kalamari DB with the <a href="https://github.com/lskatz/Kalamari/blob/master/src/Kalamari_v3.9.1.tsv">full list of possible host sequences</a> that can be removed. Reads are either tagged to map one of those species or are left unassigned. If the task at hand in the real world cannot be covered by those, you can also try another DB for Kraken2 that includes your species (or maybe retain unmapped reads from a read aligner such as Bowtie2, Minimap2…).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I plan meetings across timezones? <a href="/training-material/topics/community/faqs/meeting_times.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-plan-meetings-across-timezones"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-plan-meetings-across-timezones" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I plan meetings across timezones?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Go to a <a href="https://24timezones.com/difference/us-ca/brisbane#2024-06-30T00:00">timezone website</a> to see equivalent times across the globe.</li>   <li>Select multiple times that capture at least 2/3 of the globe (we recommend three timezones)</li>   <li>Alternate meetings across those timezones to enable global participation.</li>   <li>Share your meeting time by going to <a href="https://arewemeetingyet.com/#form">this timezone website</a> and inputting your timezone and meeting time. This will give you a URL you can link to any communications that will automatically convert that time to the local time of anyone opening the URL. You can also include your meeting notes link there for ease.</li> </ol>   <p><em>Time-saving tip</em>: If you meet every 2 months, you can set up 3 recurring calendar events for each time chosen to recur every 6 months. It’s automatic, it’s inclusive, and it’s less effort!</p>   <p><img src="/training-material/topics/community/images/timezones.png" alt="Time zone map showing England, California, and eastern Australia timezones, with three circles around times that include 2 of the 3 timezones during working hours" style="height:400px;" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I add my community to the Galaxy CoDex? <a href="/training-material/topics/community/faqs/codex.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-add-my-community-to-the-galaxy-codex"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-add-my-community-to-the-galaxy-codex" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I add my community to the Galaxy CoDex?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You need to create a new folder in the <code class="language-plaintext highlighter-rouge">data/community</code> folder within <a href="https://github.com/galaxyproject/galaxy_codex">Galaxy Codex code source</a>.</p>   <blockquote class="notranslate hands_on">   <div class="box-title hands-on-title" id="hands-on-create-a-folder-for-your-community"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create a folder for your community</div>   <ol>     <li>If not already done, fork the <a href="https://github.com/galaxyproject/galaxy_codex">Galaxy Codex repository</a></li>     <li>Go to the <code class="language-plaintext highlighter-rouge">communities</code> folder</li>     <li>Click on <strong>Add file</strong> in the drop-down menu at the top</li>     <li>Select <strong>Create a new file</strong></li>     <li>       <p>Fill in the <code class="language-plaintext highlighter-rouge">Name of your file</code> field with:  name of your community + <code class="language-plaintext highlighter-rouge">metadata/categories</code></p>       <p>This will create a new folder for your community and add a categories file to this folder.</p>     </li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I join the Galaxy Community Board? <a href="/training-material/topics/community/faqs/governance_gcb_join.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-join-the-galaxy-community-board"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-join-the-galaxy-community-board" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I join the Galaxy Community Board?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>SIG</strong> representatives should join our:</p>   <ul>   <li>📪 <a href="https://lists.galaxyproject.org/lists/sig-chairs.lists.galaxyproject.org/">Mailing list</a></li>   <li>📝 <a href="https://docs.google.com/document/d/19zv4rata-uVhFW43S8v_Fw83RVYq6jDGXy-1T8Aju4s/edit?usp=sharing">Rolling meeting notes</a></li>   <li>☕ <a href="https://matrix.to/usegalaxy_commreps:matrix.org#/!JHcKrPhyfJoqHESrdD:matrix.org">Chatroom</a></li>   <li>🗓️<a href="https://calendar.google.com/calendar/embed?height=600&amp;wkst=1&amp;ctz=Europe%2FLondon&amp;bgcolor=%23ffffff&amp;mode=AGENDA&amp;showPrint=0&amp;title=Galaxy%20Community%20Board&amp;src=MDQwNDY2MDRhNGYxODE2NDk0MjBkYTQzMzUzMTBkN2E1MmQxMGJmNDkxNDgwMGEyZjNhYjEzZWE0ZWY3MzEyY0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t&amp;color=%23F6BF26%22%20style=%22border:solid%201px%20#777%22%20width=%22800%22%20height=%22600%22%20frameborder=%220%22%20scrolling=%22no%22">Community Board Google-Calendar</a></li>   <li>🗓️<a href="https://calendar.google.com/calendar/ical/04046604a4f181649420da4335310d7a52d10bf4914800a2f3ab13ea4ef7312c%40group.calendar.google.com/public/basic.ics">Community Board iCalendar</a></li>   <li>📁<a href="https://drive.google.com/drive/folders/190O6bbfXQ7fX3kVUiObTUj4nKojGw8Cu?usp=share_link">Googlefolder of useful docs</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I know what protocol my data was sequenced with? <a href="/training-material/topics/single-cell/tutorials/scrna-umis/faqs/protocol.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-how-do-i-know-what-protocol-my-data-was-sequenced-with"><i class="far fa-question-circle" aria-hidden="true"></i> Question: How do I know what protocol my data was sequenced with?</div>   <p>If you have 10x data, then you just need to count the length of the R1 reads to guess the Chromium version (see <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html">this tutorial</a>). For other types of data, you must know the protocol in advance, and even then you must also know the multiplexing strategy and the list of expected (whitelisted) barcodes. The whitelist may vary from sequencing lab to sequencing lab, so always ask the wetlab people how the FASTQ data was generated.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How does one compare metaproteomics measurements from two experimental conditions? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/comparing_measurement.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-does-one-compare-metaproteomics-measurements-from-two-experimental-conditions"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-does-one-compare-metaproteomics-measurements-from-two-experimental-conditions" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How does one compare metaproteomics measurements from two experimental conditions?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>For comparing taxonomy composition or functional content of two conditions in metaproteomics or metatranscriptomics studies, users are recommended to use metaQuantome. GTN tutorials for metaQuantome are available <a href="/training-material/topics/proteomics/index.html">in the proteomics topic.</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How does one convert RAW files to MGF peak lists within Galaxy? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/convert_to_MGF.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-does-one-convert-raw-files-to-mgf-peak-lists-within-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-does-one-convert-raw-files-to-mgf-peak-lists-within-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How does one convert RAW files to MGF peak lists within Galaxy?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy has implemented msconvert tool so that RAW files from Thermo instruments can be converted into MGF or mzML formats.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How many search engines can you use in SearchGUI? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/search_gui.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-how-many-search-engines-can-you-use-in-searchgui"><i class="far fa-question-circle" aria-hidden="true"></i> Question: How many search engines can you use in SearchGUI?</div>   <p>SearchGUI has options to use upto 9 search algorithms. However, running all at the same time can be time consuming. According to our initial test, upto 4 search engines can give you good results.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to enable the Activity Bar <a href="/training-material/faqs/galaxy/activity_bar_how_to_enable.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This FAQ demonstrates how to enable the activity bar within the Galaxy interface
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-enable-the-activity-bar"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-enable-the-activity-bar" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to enable the Activity Bar</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you do not see the Activity Bar it can be enabled as follows:</p>   <ol>   <li>Click on the “<strong>User</strong>” link at the top of the Galaxy interface</li>   <li>Select “<strong>Preferences</strong>”</li>   <li>Scroll down and click on “<strong>Manage Activity Bar</strong>”</li>   <li>     <p>Toggle the “<strong>Enable Activity Bar</strong>” switch and voila!</p>     <p><img src="/training-material/shared/images/activity_bar_how_to_enable.svg" alt="The four steps described above are shown visually" /></p>   </li> </ol>   <!-- Original drawing = https://docs.google.com/drawings/d/1yynqEiV66F8idKH5JwTG-4ASuqK62thIwHeCMHRuupg/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> I cannot run client tests because yarn is not installed. <a href="/training-material/topics/dev/tutorials/debugging/faqs/no_tests_no_yarn.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-i-cannot-run-client-tests-because-yarn-is-not-installed"><i class="far fa-question-circle" aria-hidden="true"></i> Question: I cannot run client tests because yarn is not installed.</div>   <p>Make sure you have executed <code class="language-plaintext highlighter-rouge">scripts/common_startup.sh</code> and have activated the virtual environment (<code class="language-plaintext highlighter-rouge">. .venv/vin/activate</code>) in your <em>current terminal session</em>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/converting_FASTA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-i-have-fastq-files-from-metagenomics-or-metatranscriptomics-datasets-how-can-i-convert-them-into-a-protein-fasta-file-for-metaproteomics-searches"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-i-have-fastq-files-from-metagenomics-or-metatranscriptomics-datasets-how-can-i-convert-them-into-a-protein-fasta-file-for-metaproteomics-searches" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy has a tool named Sixgill that can be used to convert the nucleic acid sequences to ‘metapeptide’ sequences. There are other options available within Galaxy such as the GalaxyGraph approach and Metagenome Binning, Assembly and Annotation Workflow. Please <a href="https://help.galaxyproject.org/">contact us</a>, if you need assistance.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_strategies.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-i-have-a-really-large-search-database-what-search-strategies-do-you-recommend-for-searching-my-mass-spectrometry-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-i-have-a-really-large-search-database-what-search-strategies-do-you-recommend-for-searching-my-mass-spectrometry-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Readers are encouraged to use the database sectioning approach described by <a href="https://pubmed.ncbi.nlm.nih.gov/32396365/">Praveen Kumar et al</a> and available within Galaxy. Readers are also encouraged to consider other approaches such as MetaNovo (not yet available in Galaxy). In absence of any database or taxonomic information about the microbiome dataset, other methods such as COMPIL 2.0  and De novo search methods can also be considered.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> I want to use a collection for outputs but it always passes the test even when the script fails. Why? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/collections_test.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-i-want-to-use-a-collection-for-outputs-but-it-always-passes-the-test-even-when-the-script-fails-why"><i class="far fa-question-circle" aria-hidden="true"></i> Question: I want to use a collection for outputs but it always passes the test even when the script fails. Why?</div>   <ul>   <li>Collections are tricky for generating tests.     <ul>       <li>The contents appear only after the tool has been run and even then may vary with settings.</li>     </ul>   </li>   <li>A manual test override is currently the only way to test collections properly.</li>   <li>Automation is hard. If you can help, pull requests are welcomed.</li>   <li>Until it’s automated, please take a look at the plotter sample.</li>   <li>It is recommended that you modify the test over-ride that appears in that sample form. Substitute one or more of the file names you expect to see after the collection is filled by your new tool for the <code class="language-plaintext highlighter-rouge">&lt;element.../&gt;</code> used in the plotter sample’s tool test.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> In bowtie 2 parameters, in place of 1000 for other experiments, should we mention the median fragment length observed in our library? <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/bowtie_parameters.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-in-bowtie-2-parameters-in-place-of-1000-for-other-experiments-should-we-mention-the-median-fragment-length-observed-in-our-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-in-bowtie-2-parameters-in-place-of-1000-for-other-experiments-should-we-mention-the-median-fragment-length-observed-in-our-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: In bowtie 2 parameters, in place of 1000 for other experiments, should we mention the median fragment length observed in our library?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Not the median fragment length but the maximum fragment length you expect. However, you will see that in illumina sequencers, the longer the fragments are the less efficiently they are sequenced so long fragment length pairs are not very numerous.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> In the MVP platform, is it possible to view the genomic location of all the peptides? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/mvp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-in-the-mvp-platform-is-it-possible-to-view-the-genomic-location-of-all-the-peptides"><i class="far fa-question-circle" aria-hidden="true"></i> Question: In the MVP platform, is it possible to view the genomic location of all the peptides?</div>   <p>Not really, you can only view the genomic localization of the peptides that were present in the genomic mapping file (output from the first workflow).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is it possible to replace the existing alignment tools such as HISAT and Freebayes with other tools? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/alignment_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-is-it-possible-to-replace-the-existing-alignment-tools-such-as-hisat-and-freebayes-with-other-tools"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Is it possible to replace the existing alignment tools such as HISAT and Freebayes with other tools?</div>   <p>The tools in this workflow are customizable, however, the user has to ensure that the inputs are in the correct format, while using the same reference genome database.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is it possible to subsample some samples if you have more reads? <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/subsampling.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-is-it-possible-to-subsample-some-samples-if-you-have-more-reads"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Is it possible to subsample some samples if you have more reads?</div>   <p>Yes, we would recommend to process all reads and just before the peak calling. You can use <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Samtools view</strong> to sample the BAM file.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is it possible to use alternative tools to those proposed in the tutorial? <a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/alternative_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-is-it-possible-to-use-alternative-tools-to-those-proposed-in-the-tutorial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-is-it-possible-to-use-alternative-tools-to-those-proposed-in-the-tutorial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Is it possible to use alternative tools to those proposed in the tutorial?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Yes! There are many tools whose functionality are similar (e.g. Illumina reads can be mapped by using HISAT2 instead of Bowtie2).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is the ToolFactory a complete replacement for manual tool building? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/replace_manual_install.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-is-the-toolfactory-a-complete-replacement-for-manual-tool-building"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Is the ToolFactory a complete replacement for manual tool building?</div>   <ul>   <li>No, except where all the requirements for the package or script can be satisfied by the limited automated functions of the code generator, or where there is a script with all the complex logic that might otherwise go into XML</li>   <li>Many advanced XML features are not available such as output filters.</li>   <li>Adding DIY output filters, XML macros and some other advanced features is possible if anyone is sufficiently enthusiastic - some features in the galaxyxml package would be relatively straightforward to add.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is there a way to filter on the Kalimari database? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/kalamri_database_filtering.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-is-there-a-way-to-filter-on-the-kalimari-database"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Is there a way to filter on the Kalimari database?</div>   <p>To filter the Kalamari database, e.g. leaving out milk bacteria only to detect spoilers or contaminants, but the Kalimera list contains a lot more than that, you can:</p>   <ol>   <li>Look at a publication etc. to find a list of bacteria to remove.</li>   <li>Change the regex <code class="language-plaintext highlighter-rouge">^.*Gallus|Homo|Bos.*$</code>  to <code class="language-plaintext highlighter-rouge">^.*Gallus|Homo|Bos|Bacterium1|Bacterium2...|BacteriumN.*$</code></li> </ol>   <p>Milk pathogens are somewhat known, <em>Salmonella</em>, <em>Escherichia</em>… It might be easier to retain reads only mapping to pathogens instead</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Isn't it awkward to find so many humans sequences there, since we filter for them before? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/remaining_human_sequences.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-isn-t-it-awkward-to-find-so-many-humans-sequences-there-since-we-filter-for-them-before"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Isn't it awkward to find so many humans sequences there, since we filter for them before?</div>   <p>We see a lot that Kraken tends to assign many reads to human, despite they do not map to human genome. Due to resemblance between organisms and the limited species coverage of Kraken databases sometimes does happen that reads corresponding to higher organisms get mapped to humans. It was a very severe problem for the standard databases, because yeast genes were mis-assigned to human.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> It says I already have an account when registering for ecology.usegalaxy.eu <a href="/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/account_exists.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-it-says-i-already-have-an-account-when-registering-for-ecology-usegalaxy-eu"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-it-says-i-already-have-an-account-when-registering-for-ecology-usegalaxy-eu" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: It says I already have an account when registering for ecology.usegalaxy.eu</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The <a href="https://usegalaxy.eu">ecology.usegalaxy.eu</a> (and any other Galaxy server ending in <code class="language-plaintext highlighter-rouge">usegalaxy.eu</code>) is the SAME server as the regular <a href="https://usegalaxy.eu">usegalaxy.eu</a> server, just modified for Ecology analyses.</p>   <p>You can use the <strong>SAME credentials</strong> you used to register on usegalaxy.eu to log into the ecology server.</p>   <p>If you do not have an account on Galaxy EU yet, will need to create one.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> JBrowse is taking a long time to complete? <a href="/training-material/topics/sequence-analysis/tutorials/mapping/faqs/jbrowse_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-jbrowse-is-taking-a-long-time-to-complete"><i class="far fa-question-circle" aria-hidden="true"></i> Question: JBrowse is taking a long time to complete?</div>   <p>Normally this should be done in around 3 minutes. However, it might be busy on the servers, so please be patient and come back to it later.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Most tools seem to have options for assembly using long and short reads, what are the pros and cons of the different tools? <a href="/training-material/topics/assembly/faqs/hybrid_assembly.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-most-tools-seem-to-have-options-for-assembly-using-long-and-short-reads-what-are-the-pros-and-cons-of-the-different-tools"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Most tools seem to have options for assembly using long and short reads, what are the pros and cons of the different tools?</div>   <p>In our experience, when both long and short reads are allowed as input, the difference comes down to the order in which set is assembled first. For example, Unicycler assembles the short reads first (which can be good, because they are more accurate), and then scaffolds these into larger contigs using long reads. Other tools (or workflows) often assemble long reads first (which can also be good because these can span repeat regions), then correct this assembly with information from the more accurate short reads. There may also be other variations on long/short read assembly, and/or iterations of these types of steps (assemble, correct). My preference is to assemble long reads first, but that’s because I’m really interested in covering repeat regions. If accuracy was the aim, rather than contig length, the short-reads-first approach may be better. For even more complexity … I think some tools now allow input of “trusted contigs” - i.e. contigs assembled from other tools. Ryan Wick has a new tool called <a href="https://github.com/rrwick/Trycycler/wiki">Trycyler</a> that can take in multiple assemblies to make a consensus (bacterial genomes).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> MultiQC error for your FastQC reports? <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/multiqc_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-multiqc-error-for-your-fastqc-reports"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-multiqc-error-for-your-fastqc-reports" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: MultiQC error for your FastQC reports?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Please double-check that:</p>   <ol>   <li>You selected <code class="language-plaintext highlighter-rouge">FastQC</code> tool as the source of the log files in MultiQC.</li>   <li>And you provided the <strong>Raw Data</strong> of FastQC and not the HTML reports.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> My Rscript tool generates a strange R error on STDOUT about an invalid operation on a closure called 'args' ? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/error_closure_args.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-my-rscript-tool-generates-a-strange-r-error-on-stdout-about-an-invalid-operation-on-a-closure-called-args"><i class="far fa-question-circle" aria-hidden="true"></i> Question: My Rscript tool generates a strange R error on STDOUT about an invalid operation on a closure called 'args' ?</div>   <p>Did your code declare the <code class="language-plaintext highlighter-rouge">args vector</code> with something like <code class="language-plaintext highlighter-rouge">args = commandArgs(trailingOnly=TRUE)</code> before it tried to access args[1] ? See the plotter tool for a sample</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> My Scanpy FindMarkers step is giving me an empty table <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/findmarkers_empty.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-my-scanpy-findmarkers-step-is-giving-me-an-empty-table"><i class="far fa-question-circle" aria-hidden="true"></i> Question: My Scanpy FindMarkers step is giving me an empty table</div>   <p>Try selecting: “Use programme defaults: Yes” and see if that fixes it.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> My snippy is running for a very long time. Is this normal? <a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-my-snippy-is-running-for-a-very-long-time-is-this-normal"><i class="far fa-question-circle" aria-hidden="true"></i> Question: My snippy is running for a very long time. Is this normal?</div>   <p>As this tutorial uses real world data some of the tools can run for quite a while. During a course we can expected longer run times as the Galaxy servers are heavily used. Typically expected runtimes are approximately:</p>   <table>   <thead>     <tr>       <th>Tool name</th>       <th>Runtime</th>     </tr>   </thead>   <tbody>     <tr>       <td>FastQC</td>       <td>2 minutes</td>     </tr>     <tr>       <td>MultiQC</td>       <td>5 minutes</td>     </tr>     <tr>       <td>Trimmomatic</td>       <td>5 minutes</td>     </tr>     <tr>       <td>kraken2</td>       <td>5 - 12 minutes</td>     </tr>     <tr>       <td>snippy</td>       <td>15 - 25 minutes</td>     </tr>     <tr>       <td>TB Variant Filter</td>       <td>2 minutes</td>     </tr>     <tr>       <td>TB-Profiler</td>       <td>5 minutes</td>     </tr>     <tr>       <td>Text transformation</td>       <td>Less than 1 minute</td>     </tr>     <tr>       <td>TB Variant Report</td>       <td>1 minute</td>     </tr>     <tr>       <td>JBrowse</td>       <td>5 minutes</td>     </tr>     <tr>       <td>Samtools stats (optional)</td>       <td>1 minute</td>     </tr>     <tr>       <td>BAM Coverage plotter (optional)</td>       <td>1 minute</td>     </tr>   </tbody> </table> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> On Scanpy PlotEmbed, the tool is failing <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_fails.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-on-scanpy-plotembed-the-tool-is-failing"><i class="far fa-question-circle" aria-hidden="true"></i> Question: On Scanpy PlotEmbed, the tool is failing</div>   <p>Try selecting “Use raw attributes if present: NO”</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> On the Scanpy PlotEmbed step, my object doesn’t have Il2ra or Cd8b1 or Cd8a etc. <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_results_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-on-the-scanpy-plotembed-step-my-object-doesn-t-have-il2ra-or-cd8b1-or-cd8a-etc"><i class="far fa-question-circle" aria-hidden="true"></i> Question: On the Scanpy PlotEmbed step, my object doesn’t have Il2ra or Cd8b1 or Cd8a etc.</div>   <p>Check your <code class="language-plaintext highlighter-rouge">Anndata</code> object - it should be <code class="language-plaintext highlighter-rouge">7874 x 14832</code>, i.e. 7874 cells x 14832 genes. Is it actually 2000 genes only (i.e. and therefore missing the above markers)? You may have selected to remove genes at the Scanpy FindVariableGenes step (last toggle, ‘Remove genes not marked as highly variable’ &lt; Select NO.) (Most likely you did this correctly the first time, but later in investigating how many got marked as highly variable, may have run this tool again and removed the nonvariable ones. We’ve updated the text to more clearly prevent this, but you may have gotten caught out!)</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Only one Planemo test runs at a time. Why doesn't the server allow more than one at once? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/planemo_multiple.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-only-one-planemo-test-runs-at-a-time-why-doesn-t-the-server-allow-more-than-one-at-once"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Only one Planemo test runs at a time. Why doesn't the server allow more than one at once?</div>   <ul>   <li>When a new dependency is being installed in the Planemo Conda repository, there is no locking to prevent a second process from overwriting or otherwise interfering with it’s own independent repository update.</li>   <li>The result is not pretty.</li>   <li>Allowing two tests to run at once has proven to be unstable so the Appliance is currently limited to one.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Preparing materials for asynchronous learning: CYOA <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-cyoa.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-preparing-materials-for-asynchronous-learning-cyoa"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-preparing-materials-for-asynchronous-learning-cyoa" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Preparing materials for asynchronous learning: CYOA</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you are running a remote training, and expect your users to follow a specific path, be <strong>certain</strong> to include the URL parameter to select the pathway to avoid student confusion. Please note that all tutorials using a CYOA <em>should</em> be tagged which will give you a heads up as a trainer.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Preparing materials for asynchronous learning: FAQs <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-faqs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-preparing-materials-for-asynchronous-learning-faqs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-preparing-materials-for-asynchronous-learning-faqs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Preparing materials for asynchronous learning: FAQs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When you are running a remote, asynchronous lesson, you’ll want to be sure you collect all student questions and add them back to your tutorial afterwards, as FAQs. This will help other learners as they progress through the materials, and can give you a very easy URL to point your learners to if they get stuck on a particular task.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Preparing materials for asynchronous learning: Self-Study <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-selfstudy.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-preparing-materials-for-asynchronous-learning-self-study"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-preparing-materials-for-asynchronous-learning-self-study" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Preparing materials for asynchronous learning: Self-Study</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In the context of remote trainings, where a teacher isn’t synchronously available, ensuring that you have questions throughout your materials for students to check their understanding is incredibly key.</p>   <p>Additionally ensuring that solutions are provided, and are correct and up-to-date (or use a snippet explaining data variability along with with ways to check the results) is mandatory. Students will then use these questions to self-check their understanding against what you expected them to learn.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Preparing materials for asynchronous learning: Tips <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-preparing-materials-for-asynchronous-learning-tips"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-preparing-materials-for-asynchronous-learning-tips" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Preparing materials for asynchronous learning: Tips</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The use of snippets is <strong>extremely</strong> important for asynchronous, remote learning. In this situation as students do not have a teacher immediately on hand, and likely do not have friends or colleagues sitting working with them, they will rely on these boxes to refresh their knowledge and know what to do.</p>   <p>Please ensure you test your learning materials with a learner or colleague not familiar the material, and if possible, (<strong>silently</strong>) watch them go through your lesson. You’ll easily identify which portions need more explanations and details.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Running more than one round of Pilon polishing <a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/pilon.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-running-more-than-one-round-of-pilon-polishing"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-more-than-one-round-of-pilon-polishing" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Running more than one round of Pilon polishing</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Include the most recent polished assembly as input to the next round. You will also need to make a new bam file (here, we have <code class="language-plaintext highlighter-rouge">round1.bam</code> and <code class="language-plaintext highlighter-rouge">round2.bam</code>).</p>   <p>Round 1</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>assembly.fasta + illumina reads =&gt; BWA MEM =&gt; round1.bam<br></code></pre></div></div>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>round1.bam + assembly.fasta =&gt; pilon =&gt; polished.fasta<br></code></pre></div></div>   <p>Round 2</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>polished.fasta + illumina reads =&gt; BWA MEM =&gt; round2.bam<br></code></pre></div></div>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>round2.bam + polished.fasta =&gt; pilon =&gt; polished2.fasta<br></code></pre></div></div>   <p><strong>How to know when enough polishing iterations have run?</strong></p>   <p>There is no single answer, but a common way is to see when pilon stops making many polishing changes between rounds. So if round1 made 100 changes, and round2 made only 3, this seems like there would not be much more polishing to do.</p>   <p><strong>How can I see how many changes Pilon has made?</strong></p>   <p>There are two ways that I know of to see how many changes that Pilon made:</p>   <p>The first is to look at the tool <em>standard output</em> (<code class="language-plaintext highlighter-rouge">stdout</code>) from Pilon (<a href="/training-material/faqs/galaxy/tools_logs.html">instructions</a>).</p>   <p>Somewhere near the top of this log file will be a line that says how many corrections (changes) were made.</p>   <p>The second way is to count the number of lines in the changes file. To do this, use the tool called <strong>Line/Word/Character count</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, and select the line count option.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> TB Variant Report crashes (with an error about KeyError: 'protein') <a href="/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/key-error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-tb-variant-report-crashes-with-an-error-about-keyerror-protein"><i class="far fa-question-circle" aria-hidden="true"></i> Question: TB Variant Report crashes (with an error about KeyError: 'protein')</div>   <p>This is a bug present in TB Variant Report (aka tbvcfreport) version 0.1.8 and earlier. In this case it is triggered by the presence of variants in Rv3798. You only see this bug, however, if you forget to run <code class="language-plaintext highlighter-rouge">tb_variant_filter</code> (TB Variant Filter). Rv3798 is a suspected transposase and any variants in this gene region would be filtered out by <code class="language-plaintext highlighter-rouge">tb_variant_filter</code>, so if you see this crash, make sure you have run the filter step before the TB Variant Report step.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The Build tissue-specific expression dataset tool (step one) exits with an error code. <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/build_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-the-build-tissue-specific-expression-dataset-tool-step-one-exits-with-an-error-code"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-the-build-tissue-specific-expression-dataset-tool-step-one-exits-with-an-error-code" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: The Build tissue-specific expression dataset tool (step one) exits with an error code.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>For the HPS source files version select HPA normal tissue 23/10/2018 rather than the version from 01/04/2020.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The UMAP Plots errors out sometimes? <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/umap_plot_errors.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-the-umap-plots-errors-out-sometimes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-the-umap-plots-errors-out-sometimes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: The UMAP Plots errors out sometimes?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Try a different colour palette. For upstream code reasons, the default color palette sometimes causes the tool to error out.</p>   <ul>   <li>Under <strong>Plot attributes</strong>, do     <ul>       <li><em>“Colour map to use for continuous variables”</em>: <code class="language-plaintext highlighter-rouge">viridis</code></li>       <li><em>“Colors to use for plotting categorical annotation groups”</em>: <code class="language-plaintext highlighter-rouge">plasma</code></li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The folder `recipes/belerophon/` and the file `meta.yaml` already exist in bioconda? <a href="/training-material/topics/dev/tutorials/tool-from-scratch/faqs/bioconda_files_exist.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-the-folder-recipes-belerophon-and-the-file-meta-yaml-already-exist-in-bioconda"><i class="far fa-question-circle" aria-hidden="true"></i> Question: The folder `recipes/belerophon/` and the file `meta.yaml` already exist in bioconda?</div>   <p>The recipe has already been added previously. If you want to create the recipe from scratch you may just do this in another directory below <code class="language-plaintext highlighter-rouge">recipes/</code>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The input for a tool is not listed in the dropdown <a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/faqs/input_not_found.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-the-input-for-a-tool-is-not-listed-in-the-dropdown"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-the-input-for-a-tool-is-not-listed-in-the-dropdown" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: The input for a tool is not listed in the dropdown</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This tutorial uses collections, some tools will require collections as input (e.g. Taxonomy-to-Krona). To select a collection as in put to a file, click on the <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> button in front of the input parameter you want to supply the collection to.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The input for a tool is not listed in the dropdown <a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/input_not_found.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-the-input-for-a-tool-is-not-listed-in-the-dropdown-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-the-input-for-a-tool-is-not-listed-in-the-dropdown-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: The input for a tool is not listed in the dropdown</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This tutorial uses collections, some tools will require collections as input (e.g. Taxonomy-to-Krona). To select a collection as in put to a file, click on the <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> button in front of the input parameter you want to supply the collection to.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> UCSC import: what should my file look like? <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/ucsc_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-ucsc-import-what-should-my-file-look-like"><i class="far fa-question-circle" aria-hidden="true"></i> Question: UCSC import: what should my file look like?</div>   <p>~2020 lines, with the following header line:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>bin    name    chrom   strand  txStart txEnd   cdsStart        cdsEnd  exonCount       exonStarts      exonEnds        score   name2   cdsStartStat    cdsEndStat      exonFrames<br></code></pre></div></div>   <p>Where:</p>   <ul>   <li><code class="language-plaintext highlighter-rouge">txStart</code>: Transcript start site</li>   <li><code class="language-plaintext highlighter-rouge">cdsStart</code>: CodingSequence start site</li> </ul>   <p><strong>Note:</strong> UCSC is updated frequently, you might get a slightly different number of lines. If you only get one row in this file, make sure you requested the entire chr22, not just one position.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What advantages does a Chromatogram Library have over a DDA-generated library or predicted spectral library? <a href="/training-material/topics/proteomics/tutorials/encyclopedia/faqs/advantages.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-advantages-does-a-chromatogram-library-have-over-a-dda-generated-library-or-predicted-spectral-library"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What advantages does a Chromatogram Library have over a DDA-generated library or predicted spectral library?</div>   <p>While generating a Chromatogram Library is the most time consuming step of the EncyclopeDIA workflow, it is beneficial to DIA data analysis. DIA is a novel technique and methods for DIA data analysis are still being developed. One method commonly used includes searching DIA data against DDA-generated libraries. However, there are limitations in this method. Firstly, DDA-generated libraries are not always an accurate representation of DIA data: differences in the methods of data collection play an important role in the efficacy of the library. Secondly, DDA-generated libraries often require labs to run completely separate DDA experiments to simply generate a library with which to analyze their DIA data. Chromatogram Libraries mitigate some of the previous shortcomings mentioned. DIA data is incorporated into the generation of the Chromatogram Library and therefore provides context to the DIA data being analyzed. Secondly, the ELIB format of the Chromatogram Library allows for extra data to be included in the analysis of the DIA data, including intensity, m/z ratio, and retention time compared to the use of a DDA-generated DLIB library. Lastly, a Chromatogram Library can be generated without the use of a spectral library (as mentioned in the last question). Therefore, it is possible to forgo DDA data collection as the DLIB DDA-generated library is not strictly needed for Chromatogram Library generation and to run the EncyclopeDIA workflow (saving time and resources).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What does `^.*Gallus|Homo|Bos.*$` mean? <a href="/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_filtering_filter.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-does-gallus-homo-bos-mean"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What does `^.*Gallus|Homo|Bos.*$` mean?</div>   <p><code class="language-plaintext highlighter-rouge">^.*Gallus|Homo|Bos.*$</code> is a <a href="https://www.regular-expressions.info/">regular expression</a> that matches a string containing the words Gallus <strong>OR</strong> Homo <strong>OR</strong> Bos.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What file/data formats are defined for I/O in Galaxy? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/formats.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-file-data-formats-are-defined-for-i-o-in-galaxy"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What file/data formats are defined for I/O in Galaxy?</div>   <ul>   <li><a href="https://galaxyproject.org/learn/datatypes">Galaxy Datatypes</a></li>   <li><code class="language-plaintext highlighter-rouge">[galaxy-root]/config/datatypes_conf.xml</code> is read at startup so new datatypes can be defined.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is Gene Ontology (GO)? <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/gene_ontology.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-is-gene-ontology-go"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What is Gene Ontology (GO)?</div>   <p>A very commonly used way of specifying these sets is to gather genes/proteins that share the same Gene Ontology (GO) term, as specified by the Gene Ontology Consortium.</p>   <p>The GO project provides an ontology that describes gene products and their relations in three non-overlapping domains of molecular biology, namely “Molecular Function”, “Biological Process”, and “Cellular Component”. Genes/proteins are annotated by one or several GO terms, each composed of a label, a definition and a unique identifier. GO terms are organized within a classification scheme that supports relationships, and formalized by a hierarchical structure that forms a directed acyclic graph (DAG). In such a graph is used the notions of child and parent, where a child inherits from one or multiple parents, child class having a more specific annotation than parent class (e.g. “glucose metabolic process” inherits from “hexose metabolic” parent term which itself inherits from “monosaccharide metabolic process” etc.). In this graph, each node corresponds to a GO term composed of genes/proteins sharing the same annotation, while directed edges between nodes represents their relation (e.g. ‘is a’, ‘part of’) and their roles in the hierarchy (i.e. parent and child).</p>   <p><a href="http://www.geneontology.org/page/introduction-go-resource">Further reading</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is a SNP? <a href="/training-material/topics/sequence-analysis/tutorials/mapping/faqs/snp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-is-a-snp"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What is a SNP?</div>   <p>SNP (pronounced “snip”) stands for Single Nucleotide Polymorphism. This means a single nucleotide change as compared to the reference genome.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is the Galaxy Governance Structure? <a href="/training-material/topics/community/faqs/governance_structure.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="details">   <div class="box-title details-title" id="details-what-is-the-galaxy-governance-structure"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-the-galaxy-governance-structure" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> <span>Details: What is the Galaxy Governance Structure?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy Governance consists of a <a href="https://galaxyproject.org/community/governance/geb/">Galaxy Executive Board</a> that provides global direction, working with a <a href="https://galaxyproject.org/community/governance/gtb/">Galaxy Technical Board</a> that represents <a href="https://galaxyproject.org/community/wg/">Working Groups</a> and a <a href="https://galaxyproject.org/community/governance/gcb/">Galaxy Community Board</a> that represents <a href="https://galaxyproject.org/community/sig/">Special Interest Groups</a>.</p>   <p><img src="/training-material/topics/community/images/structure.png" alt="Galaxy Executive Board is in a rectangle over top of two rectangles, the Galaxy Technical Board and Galaxy Community Board, which are themselves over top of Working Groups and Special Interest Groups, respectively. A Project Management Office rectangle spans the image across the bottom" title="Galaxy Community Governance" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is the principle of an enrichment analysis? <a href="/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/enrichment.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-is-the-principle-of-an-enrichment-analysis"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What is the principle of an enrichment analysis?</div>   <p>Enrichment analysis approach (also called over-representation analysis (ORA)) was introduced to test whether pre-specified sets of proteins (e.g. those acting together in a given biological process), change in abundance more systematically than as expected by chance. This type of analysis investigates hypotheses that are more directly relevant to the biological function, and can also help highlight a process over-represented within a subset of proteins.</p>   <p>Further reading: <a href="https://pubmed.ncbi.nlm.nih.gov/19033363/">“Huang DW, Sherman BT, and Lempicki RA (2009) Bioinformatics enrichment tools: Paths toward the comprehensive functional analysis of large gene lists. Nucleic Acids Res 37:1–13”</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What other methods are available to study the functional state of the microbiome within Galaxy? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/studying_microbiome_state.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-other-methods-are-available-to-study-the-functional-state-of-the-microbiome-within-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-other-methods-are-available-to-study-the-functional-state-of-the-microbiome-within-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What other methods are available to study the functional state of the microbiome within Galaxy?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Other software such as EggNOG Mapper, MEGAN5, MetaGOmics, MetaProteomeAnalyzer (MPA) and  ProPHAnE also generate <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241503">functional outputs</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What should I do special if on usegalaxy.be? <a href="/training-material/topics/galaxy-interface/tutorials/workflow-automation/faqs/sars-bot-usegalaxy-be.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-should-i-do-special-if-on-usegalaxy-be"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-should-i-do-special-if-on-usegalaxy-be" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What should I do special if on usegalaxy.be?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Note for anyone trying to follow the tutorial on usegalaxy.be:</strong></p>   <p>In step 3 of the hands-on section of setting up the sars-cov-2 analysis bot, when suggested to run</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>planemo run vcf2lineage.ga vcf2lineage-job.yml --profile planemo-tutorial --history_name "vcf2lineage test"<br></code></pre></div></div>   <p>please use directly the workflow ID <code class="language-plaintext highlighter-rouge">814dd8d1c056bc54</code> instead of <code class="language-plaintext highlighter-rouge">vcf2lineage.ga</code>. This ID points to a public workflow that’s using the version of the pangolin tool installed on usegalaxy.be`.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What software tools are available to determine taxonomic composition from mass spectrometry data? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/determine_taxonomic_composition.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-software-tools-are-available-to-determine-taxonomic-composition-from-mass-spectrometry-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-software-tools-are-available-to-determine-taxonomic-composition-from-mass-spectrometry-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What software tools are available to determine taxonomic composition from mass spectrometry data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Within the Galaxy framework we recommend the use of Unipept software that uses NCBI taxonomy and UniProt databases to detect unique peptides for taxonomy. Other software tools such as MetaTryp 2.0 (PMID: 32897080) can also be used to determine the taxonomic composition of the metaproteomics datasets.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What's the Galaxy Community Board? <a href="/training-material/topics/community/faqs/governance_gcb.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="details">   <div class="box-title details-title" id="details-what-s-the-galaxy-community-board"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-s-the-galaxy-community-board" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> <span>Details: What's the Galaxy Community Board?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The Galaxy Community Board provides a supportive virtual forum for the exchange of ideas, and a governance body to represent <a href="https://galaxyproject.org/community/sig/"><strong>Special Interest Groups</strong> (SIGs)</a> in Galaxy.</p>   <p>The goals of the GCB are to:</p>   <ul>   <li>share resources, tips &amp; best practices to make running <strong>SIGs</strong> easier;</li>   <li>discuss scientist (user) feedback to help guide Galaxy platform development;</li>   <li>communicate scientist (user) needs to the Galaxy Governance structure; and</li>   <li>develop proposals to advance scientist (user) goals in the Galaxy community.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> When I get a warning for base per sequence content, what should I do? <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/perbase_warning.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-when-i-get-a-warning-for-base-per-sequence-content-what-should-i-do"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-when-i-get-a-warning-for-base-per-sequence-content-what-should-i-do" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: When I get a warning for base per sequence content, what should I do?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>So far it does not mean that your data is bad. Your protocol or your data might have a bias that you normally expect. Check first the following things:</p>   <ul>   <li>Adapter content (maybe some adapters are still in your data)</li>   <li>Kmer content/Over represented sequences (this would indicate a contamination or a protocol/sequence bias)</li>   <li>Per base quality plot. If the overall quality is not good, then probably the sequencing was poorly performed.</li>   <li>Read about your protocol, e.g., ChIP-Seq and ATAC-Seq typically have a nucleotide bias. For example this article about <a href="https://mobilednajournal.biomedcentral.com/articles/10.1186/1759-8753-3-3">ATAC-Seq</a>.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> When I try to run a Selenium test, I get an error <a href="/training-material/topics/dev/tutorials/debugging/faqs/selenium_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-when-i-try-to-run-a-selenium-test-i-get-an-error"><i class="far fa-question-circle" aria-hidden="true"></i> Question: When I try to run a Selenium test, I get an error</div>   <p>If you get the following error:</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>selenium.common.exceptions.SessionNotCreatedException <span class="o">(</span>...This version of ChromeDriver only supports Chrome version...<span class="o">)</span><br></code></pre></div></div>   <p>Make sure that (a) the version of your ChromeDriver is the same as the version of Chrome:</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>chromedriver <span class="nt">--version</span><br><span class="nv">$ </span>chrome <span class="nt">--version</span><br></code></pre></div></div>   <p>If they are not the same:</p>   <ul>   <li>download the appropriate version of <a href="https://chromedriver.chromium.org/downloads">ChromeDriver</a>.</li>   <li>unzip the file</li>   <li>move the chromedriver file into the appropriate location.     <ul>       <li>On Linux, that could be <code class="language-plaintext highlighter-rouge">/usr/bin</code>, <code class="language-plaintext highlighter-rouge">$HOME/.local/bin</code>, etc.</li>       <li>Use the <code class="language-plaintext highlighter-rouge">which</code> command to check the location: <code class="language-plaintext highlighter-rouge">$ which chromedriver</code></li>     </ul>   </li>   <li>Make sure the permissions are correct (755).</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> When will aligned read objects be available for other data types? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/other-datatypes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-when-will-aligned-read-objects-be-available-for-other-data-types"><i class="far fa-question-circle" aria-hidden="true"></i> Question: When will aligned read objects be available for other data types?</div>   <p>We hope to have these constructed for long read SARS-CoV-2 data in the near future. If there is strong community interest we may expand this offering to other organisms or data types such as metagenome submissions. If you would like this format for other datasets, write to the SRA helpdesk (<a href="mailto:sra@ncbi.nlm.nih">sra@ncbi.nlm.nih</a>) and let us know!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where can I find example queries for use in the cloud and elsewhere? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/example-queries.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-where-can-i-find-example-queries-for-use-in-the-cloud-and-elsewhere"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Where can I find example queries for use in the cloud and elsewhere?</div>   <p>We have examples on our website for Athena (<a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-athena-examples/">link</a>) and BigQuery (<a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-bigquery-examples/">link</a>) which can be easily adapted to other environments.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where can I find the full listing and description of the columns in each metadata table? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/metadata-description.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-where-can-i-find-the-full-listing-and-description-of-the-columns-in-each-metadata-table"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Where can I find the full listing and description of the columns in each metadata table?</div>   <p>Table definitions are available here:</p>   <ul>   <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/sra-cloud-based-examples/">SRA Cloud-based Examples</a></li>   <li><a href="https://www.ncbi.nlm.nih.gov/sra/docs/aligned-metadata-tables/">Aligned metadata Tables</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where can I get planemo? <a href="/training-material/topics/dev/tutorials/tool-from-scratch/faqs/get_planemo.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-where-can-i-get-planemo"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Where can I get planemo?</div>   <p>Plese see <a href="/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html#installing-planemo">the installation section</a>. Essentially you can <code class="language-plaintext highlighter-rouge">pip install planemo</code>. If you don’t have pip, you need to install this first.</p>   <p>On windows you’ll need <code class="language-plaintext highlighter-rouge">WSL2</code> and then you can <code class="language-plaintext highlighter-rouge">apt-get install python3-pip</code>, same for ubuntu. For OSX users it is probably present.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where can I read more about Quality Control of data? <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/further_learning.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-where-can-i-read-more-about-quality-control-of-data"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Where can I read more about Quality Control of data?</div>   <p>I really like <a href="https://sequencing.qcfail.com/">QCFAIL</a>, It has some nice user stories of quality control issues encountered in real data and experiments</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Which icons are available to use in my tutorial? <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-which-icons-are-available-to-use-in-my-tutorial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-which-icons-are-available-to-use-in-my-tutorial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Which icons are available to use in my tutorial?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To use icons in your tutorial, take the name of the icon, ‘details’ in this example, and write something like this in your tutorial:</p>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{% icon details %}<br></code></pre></div></div>   <p>Some icons have multiple aliases, any may be used, but we’d suggest trying to choose the most semantically appropriate one in case Galaxy later decides to change the icon.</p>   <p>The following icons are currently available:</p>   <div class="row"> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-bullhorn" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>announcement </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fa fa-arrows" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>arrow-keys </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>code-in </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-laptop-code" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>code-out </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>cofest, hall-of-fame, pref-permissions </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>comment </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>congratulations </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>copy, param-files, zenodo_link </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>curriculum, level </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>details, galaxy-info, dataset-info </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>docker_image </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-download" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>download, galaxy-download </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-cloud-arrow-down" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>download-cloud </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>dropdown, galaxy-dropdown </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-envelope" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>email </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>exchange, switch-histories </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-external-link-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>external-link, galaxy_instance </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>event, event-date, last_modification </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-location-dot" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>event-location </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-tag" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>event-cost </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>feedback </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-angle-double-down" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-advanced-search </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fa fa-map-marker" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-show-active </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-barchart, galaxy-visualise, galaxy-visualize </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-angle-double-left" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-vis-config, galaxy-viz-config </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-bug </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-chart-select-data, galaxy-history-size </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-clear </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-columns, galaxy-multihistory, galaxy-history </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-cross </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-dataset-map, galaxy-workflows-activity, dataset-related-datasets </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-delete </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-history-options </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-eye, solution </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-gear, galaxy-wf-options </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-histories-activity </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fa fa-compress" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-dataset-collapse </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-archive" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-history-archive </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-hard-drive" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-history-storage-choice </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-arrows-rotate" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-history-refresh </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-sign-in-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-history-input </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-sign-out-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-history-answer </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-home" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-home </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-lab, subdomain </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-library, param-collection, topic </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-link, dataset-link </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-panelview, pref-list </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-pencil, hands_on, param-text </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-refresh </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-undo </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-rulebuilder-history </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-save, save </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-scratchbook </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-selector, param-check </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-show-hidden </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-star, rating </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-tags </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-toggle, param-toggle </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-upload </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-wand-magic-sparkles" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-wf-best-practices </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-arrows-alt-h" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-wf-connection </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-wf-edit </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-wf-new, new-history, plus </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>galaxy-wf-report-download </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>github </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>gitter </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>gtn-theme, pref-palette </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>help, question </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-comment" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>history-annotate </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>history-share, workflow </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>history-select-multiple </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>instances </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>interactive_tour </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>keypoints, pref-apikey </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>language </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>license </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-linkedin" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>linkedin </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-file-code" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>notebook </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>objectives </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-orcid" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>orcid </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>param-file </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>param-repeat, pref-notifications </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>param-select, pref-toolboxfilters </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-hand-point-right" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>point-right </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-user" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-info </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-unlock-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-password </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-id-card" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-identities </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-dataprivate </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-cloud" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-cloud </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-custombuilds, tool-versions </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fa fa-sign-out" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-signout </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-radiation" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>pref-delete </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>purl </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>references </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>requirements </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-rss" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>rss-feed </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa fa-search" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>search </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>slides </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-sticky-note" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>sticky-note </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>time </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-file-alt" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>text-document </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>tip </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>tool </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>trophy </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>tutorial </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fab fa-twitter" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>twitter </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-recycle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>upgrade_workflow </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>warning </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-chevron-circle-right" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>wf-input </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>workflow-runtime-toggle </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>workflow-run </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>video </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="far fa-play-circle" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>video-slides </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>version </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>dataset-rerun </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>dataset-visualize </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>dataset-save </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-question" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>dataset-question </div> 	</div> 	<div class="col-md-2 col-sm-3" style="text-align: center"> 		<div style="font-size: 400%"><i class="fas fa-trash-can-arrow-up" aria-hidden="true"></i><span class="visually-hidden">icon[0][0]</span></div> 		<div>dataset-undelete </div> 	</div> </div>   <p>New icons can be added in <code class="language-plaintext highlighter-rouge">_config.yaml</code>, and you can search for the corresponding icons at https://fontawesome.com/v4/icons/</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Which search algorithms are recommended for searching the metaproteomics data? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_algorithms.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-which-search-algorithms-are-recommended-for-searching-the-metaproteomics-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-which-search-algorithms-are-recommended-for-searching-the-metaproteomics-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Which search algorithms are recommended for searching the metaproteomics data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>SearchGUI supports search using nine search algorithms (X! Tandem. MS-GF+. OMSSA, Comet, Tide, MyriMatch, MS_Amanda, DirecTag and Novor). For this tutorial, we have used the first two search algorithms in the list. In our hands, the first four search algorithms have given us the most optimal results.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Which version of SearchGUI and PeptideShaker shall I use for this tutorial? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/tools_versions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-which-version-of-searchgui-and-peptideshaker-shall-i-use-for-this-tutorial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-which-version-of-searchgui-and-peptideshaker-shall-i-use-for-this-tutorial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Which version of SearchGUI and PeptideShaker shall I use for this tutorial?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>We highly recommend the usage of SearchGUI Galaxy version 3.3.10.1 and PeptideShaker version Galaxy Version 1.16.36.3. The newer versions of SearchGUI and PeptideShaker have not yet been tested for this workflow.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do I need that big (~5GB!) complicated Docker thing - can I just install the ToolFactory into our local galaxy server from the toolshed? <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/local_install.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-do-i-need-that-big-5gb-complicated-docker-thing-can-i-just-install-the-toolfactory-into-our-local-galaxy-server-from-the-toolshed"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why do I need that big (~5GB!) complicated Docker thing - can I just install the ToolFactory into our local galaxy server from the toolshed?</div>   <p>You can but it can’t really be very useful. The ToolFactory is a Galaxy tool, but it installs newly generated tools automatically into the local Galaxy server. This is not normally possible because a tool cannot escape Galaxy’s job execution environment isolation. The ToolFactory needs to write to the normally forbidden server’s configuration so the new tool appears in the tool menu and is installed in the TFtools directory which is a subdirectory of the Galaxy tools directory. The Appliance is configured so the ToolFactory and the Planemo test tool use remote procedure calls (RPC using rpyc) to do what tools cannot normally do. The rpyc server runs in a separate container. Without it, tool installation and testing are difficult to do inside Galaxy tools. Known good tools can be uploaded to a local toolshed from your private appliance for installation to that server of yours. Debugging tools on a production server is not secure SOP. You just never know what might break. <em>That’s why a desktop disposable appliance is a better choice.</em></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we change the chromosome names in the Ensembl GTF to match the UCSC genome reference? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/chromosome_names.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-do-we-change-the-chromosome-names-in-the-ensembl-gtf-to-match-the-ucsc-genome-reference"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why do we change the chromosome names in the Ensembl GTF to match the UCSC genome reference?</div>   <p>UCSC chromosome names begin with the prefix <code class="language-plaintext highlighter-rouge">chr</code>, but Ensembl chromosome names do not. For example, chromosome 19 would be denoted as <code class="language-plaintext highlighter-rouge">chr19</code> in UCSC, and as <code class="language-plaintext highlighter-rouge">19</code> in Ensemble. Most tools would view those as different when looking for matches/overlaps. Therefore it is always a good idea to make sure these match before you perform any downstream analysis.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/functional_analysis_tools.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Within the Galaxy framework we recommend the use of Unipept software that uses UniProt databases and annotation to detect proteins (EC terms) and functional groups such as GO Ontology and InterPro terms. Other software tools such as <a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0241503">EggNOG Mapper</a> are also available within the Galaxy platform. Other software such as MEGAN5, MetaGOmics, MetaProteomeAnalyzer (MPA), ProPHAnE also generate functional outputs.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we have a variant mapping file when it is not being used in the workflow? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/variant_mapping_file.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-do-we-have-a-variant-mapping-file-when-it-is-not-being-used-in-the-workflow"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why do we have a variant mapping file when it is not being used in the workflow?</div>   <p>We are working on updating the existing annotation tool to include the variant mapping file. Once that is done, the variant mapping file will also be an input for those tools.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we use FASTQ interlacer and not the FASTQ joiner? <a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/fastq_interlacer_vs_joiner.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-do-we-use-fastq-interlacer-and-not-the-fastq-joiner"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why do we use FASTQ interlacer and not the FASTQ joiner?</div>   <p>The reason ASaiM-MT uses FASTQ-interlacer than FASTQ-joiner for combining forward and reverse reads is because the joiner tool combines the forward and reverse read sequence together while the interlacer puts the forward and reverse read sequences in the same file while retaining the entity of each read along with an additional file with unpaired sequences and it maintains the integrity of the reads while helping us distinguish between the forward and reverse reads.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why does my assembly graph in Bandage look different to the one pictured in the tutorial? <a href="/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/bandage_different.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-does-my-assembly-graph-in-bandage-look-different-to-the-one-pictured-in-the-tutorial"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why does my assembly graph in Bandage look different to the one pictured in the tutorial?</div>   <p>The assembly process in Flye is heuristic, and the resulting assembly will not necessarily be exactly the same each time. This may happen even if running the same data with the same version of Flye. It can also happen with a different version of Flye.</p>   <p>To make things more complicated (stop reading now if you would like!)… the chloroplast genome has a structure that includes repeats (the inverted repeats), and, the small-single-copy region of the chloroplast exists in two orientations between these repeats. So, sometimes the assembly will be a perfect circle, sometimes the inverted repeats will be collapsed into one piece, and sometimes the small-single-copy region will be attached ambiguously.  To make things even more complicated…the chloroplast genome may even be a dynamic structure, due to flip flop recombination.</p>   <p>For more see <a href="https://academic.oup.com/gbe/article/11/12/3372/5637229">this article</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why does the query `SRR11772204 OR SRR11597145 OR SRR11667145` in the Run Selector not return any results? <a href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/faqs/query_no_results.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-does-the-query-srr11772204-or-srr11597145-or-srr11667145-in-the-run-selector-not-return-any-results"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why does the query `SRR11772204 OR SRR11597145 OR SRR11667145` in the Run Selector not return any results?</div>   <p>The query for sars-cov-2 in SRA Entrez returns over 250K results, but only the first 20k are sent to the Run Selector.  Enter the above query in Entrez directly to find the three runs used for the tutorial and send them to the Run Selector to send to Galaxy.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why don't the aligned read files have quality scores? <a href="/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/quality-scores.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-don-t-the-aligned-read-files-have-quality-scores"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why don't the aligned read files have quality scores?</div>   <p>Quality scores take up the majority of space in our compressed sequence files, so removing them makes the files much smaller (~80% or more). In addition, many uses don’t require per-base quality scores to successfully complete their work (some pipelines even require fastq format but don’t actually use the quality scores), so these files represent a faster route to completing many analyses. The full quality scores are still available in the original SRA Runs for anyone that requires them, using the SRA Tools available in Galaxy.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why don't we perform the V-Search dereplication step of ASaiM for metatrascriptomic data? <a href="/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/vsearch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-why-don-t-we-perform-the-v-search-dereplication-step-of-asaim-for-metatrascriptomic-data"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Why don't we perform the V-Search dereplication step of ASaiM for metatrascriptomic data?</div>   <p>In the metatranscriptomics data, duplicated reads are expected. And to keep the integrity of the sample, we would like to retain the reverse reads.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why is Alevin is not working? <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/faqs/alevin_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><p>Check your tool version, you need to use <code class="language-plaintext highlighter-rouge">1.3.0+galaxy2</code></p> <p>Follow <a href="/training-material/faqs/galaxy/tools_change_version.html">these instructions</a> to switch between tool versions.</p> <!--END_SNIPPET-->




<h4 class="faq-area"> Why is Alevin is not working? <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/faqs/alevin_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><p>Check your tool version, you need to use <code class="language-plaintext highlighter-rouge">1.3.0+galaxy2</code></p> <p>Follow <a href="/training-material/faqs/galaxy/tools_change_version.html">these instructions</a> to switch between tool versions.</p> <!--END_SNIPPET-->




<h4 class="faq-area"> `docker-compose up` fails with error `/usr/bin/start.sh: line 133: /galaxy/.venv/bin/uwsgi: No such file or directory` <a href="/training-material/topics/dev/tutorials/tool-generators/faqs/error_compose.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-docker-compose-up-fails-with-error-usr-bin-start-sh-line-133-galaxy-venv-bin-uwsgi-no-such-file-or-directory"><i class="far fa-question-circle" aria-hidden="true"></i> Question: `docker-compose up` fails with error `/usr/bin/start.sh: line 133: /galaxy/.venv/bin/uwsgi: No such file or directory`</div>   <ul>   <li>This is why it’s useful to watch the boot process without detaching</li>   <li>This can happen if a container has become corrupt on disk after being interrupted     <ul>       <li>cured by a complete cleanup.</li>     </ul>   </li>   <li>Make sure no docker galaxy-server related processes are running - use docker ps to check and stop them manually</li>   <li>delete the <code class="language-plaintext highlighter-rouge">..compose/export directory</code> with <code class="language-plaintext highlighter-rouge">sudo rm -rf export/*</code> to clean out any corrupted files</li>   <li>run <code class="language-plaintext highlighter-rouge">docker system prune</code> to clear out any old corrupted containers, images or networks. Then run <code class="language-plaintext highlighter-rouge">docker volume prune</code> in the same way to remove the shared volumes.</li>   <li>run <code class="language-plaintext highlighter-rouge">docker-compose pull</code> again to ensure the images are correct</li>   <li>run <code class="language-plaintext highlighter-rouge">docker-compose up</code> to completely rebuild the appliance from scratch. <strong>Please be patient</strong>.</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Analysis"> <a href="#Analysis"> Analysis </a></h3>

<hr/>



<h4 class="faq-area"> Are UMIs not actually unique? <a href="/training-material/topics/single-cell/faqs/umi.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-are-umis-not-actually-unique"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-are-umis-not-actually-unique" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Are UMIs not actually unique?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Not strictly, but unique enough. The distribution of UMIs should ideally be uniform so that the chance of any two same UMIs capturing the same transcript (via different amplicons) is small. As barcodes have increased in size, the number of UMIs has also increased allowing for UMIs to reach more or less the same numbers of transcripts.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can RNA-seq techniques be applied to scRNA-seq? <a href="/training-material/topics/single-cell/faqs/techniques.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-rna-seq-techniques-be-applied-to-scrna-seq"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-rna-seq-techniques-be-applied-to-scrna-seq" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can RNA-seq techniques be applied to scRNA-seq?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The short answer is ‘no, but yes’. At the beginning this was impossible due to the over-prevalence of dropout events (“zeroes”) in the data complicating the normalisation techniques, but this is not so much of a problem any more with newer methods.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Notebook-based tutorials can give different outputs <a href="/training-material/topics/single-cell/faqs/notebook_warning.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="warning">   <div class="box-title warning-title" id="warning-notebook-based-tutorials-can-give-different-outputs"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i> Warning: Notebook-based tutorials can give different outputs</div>   <p>The nature of coding pulls the most recent tools to perform tasks. This can - and often does - change the outputs of an analysis. Be prepared, as you are unlikely to get outputs identical to a tutorial if you are running it in a programming environment like a Jupyter Notebook or R-Studio. That’s ok! The outputs should still be pretty close.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/single-cell/faqs/dimension_reduction.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The actual data has tens of thousands of genes, and so tens of thousands of variables to consider. Even after selecting for the most variable genes and the most high quality genes, we can still be left with &gt; 1000 genes. Performing clustering on a dataset with 1000s of variables is possible, but computationally expensive. It is therefore better to perform dimension reduction to reduce the number of variables to a latent representation of these variables. These latent variables are ideally more than 10 but less than 50 to capture the variability in the data to perform clustering upon.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we only consider highly variable genes? <a href="/training-material/topics/single-cell/faqs/variable_genes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-do-we-only-consider-highly-variable-genes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-do-we-only-consider-highly-variable-genes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why do we only consider highly variable genes?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The non-variable genes are likely housekeeping genes, which are expressed everywhere and are not so useful for distinguishing one cell type from another. However background genes are important to the analysis and are used to generate a background baseline model for measuring the variability of the other genes.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why is my tool erroring as 'Above error raised while reading key '/layers' of type <class 'h5py._hl.group.Group'> from /.' <a href="/training-material/topics/single-cell/faqs/sc_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-is-my-tool-erroring-as-above-error-raised-while-reading-key-layers-of-type-class-h5py-hl-group-group-from"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-is-my-tool-erroring-as-above-error-raised-while-reading-key-layers-of-type-class-h5py-hl-group-group-from" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why is my tool erroring as 'Above error raised while reading key '/layers' of type &lt;class 'h5py._hl.group.Group'&gt; from /.'</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Are you getting the following error, or similar?</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Traceback (most recent call last):<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/utils.py", line 177, in func_wrapper<br>return func(elem, *args, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/h5ad.py", line 527, in read_group<br>EncodingVersions[encoding_type].check(<br>File "/usr/local/lib/python3.9/enum.py", line 432, in __getitem__<br>return cls._member_map_[name]<br>KeyError: 'dict'<br><br>During handling of the above exception, another exception occurred:<br><br>Traceback (most recent call last):<br>File "/usr/local/bin/scanpy-cli", line 10, in &lt;module&gt;<br>sys.exit(cli())<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 829, in __call__<br>return self.main(*args, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 782, in main<br>rv = self.invoke(ctx)<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1259, in invoke<br>return _process_result(sub_ctx.command.invoke(sub_ctx))<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1259, in invoke<br>return _process_result(sub_ctx.command.invoke(sub_ctx))<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1066, in invoke<br>return ctx.invoke(self.callback, **ctx.params)<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 610, in invoke<br>return callback(*args, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/scanpy_scripts/cmd_utils.py", line 45, in cmd<br>adata = _read_obj(input_obj, input_format=input_format)<br>File "/usr/local/lib/python3.9/site-packages/scanpy_scripts/cmd_utils.py", line 87, in _read_obj<br>adata = sc.read(input_obj, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/scanpy/readwrite.py", line 112, in read<br>return _read(<br>File "/usr/local/lib/python3.9/site-packages/scanpy/readwrite.py", line 713, in _read<br>return read_h5ad(filename, backed=backed)<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/h5ad.py", line 421, in read_h5ad<br>d[k] = read_attribute(f[k])<br>File "/usr/local/lib/python3.9/functools.py", line 877, in wrapper<br>return dispatch(args[0].__class__)(*args, **kw)<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/utils.py", line 183, in func_wrapper<br>raise AnnDataReadError(<br>anndata._io.utils.AnnDataReadError: Above error raised while reading key '/layers' of type &lt;class 'h5py._hl.group.Group'&gt; from /.<br></code></pre></div></div>   <p>This is likely a Tool Version error. If you use a newer version of a tool with an AnnData object, and then try and use an older version of the tool or other tool in the same toolsuite (Scanpy) later, this will often fail with the above error message. The Scanpy toolsuite is not ‘backwards compatable’ - few toolsuites are. If this happened while performing a tutorial, we recommend Tutorial Mode as this embeds the correct tool version in each tool button.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>   <p><!--END_SNIPPET--></p>   <p>To fix this in your current history, try re-running the tool with the newer tool version. Or, re-run the prior dataset with an older version.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-tool-version"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-tool-version" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the tool version</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool.</p>   <p><strong>Switching to a different version</strong> of a tool:</p>   <ul>   <li>Open the tool</li>   <li>Click on the <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> versions logo at the <strong>top right</strong></li>   <li>Select the desired version from the dropdown list</li> </ul> </blockquote>   <p><!--END_SNIPPET--></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Community"> <a href="#Community"> Community </a></h3>

<hr/>



<h4 class="faq-area"> How can I talk with other users? <a href="/training-material/topics/single-cell/faqs/user_community_join.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-talk-with-other-users"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-talk-with-other-users" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I talk with other users?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> To discuss with like-minded scientists, join our <a href="http://gxy.io/gtn-slack">Galaxy Training Network chatspace in Slack</a> and discuss with fellow users of Galaxy single cell analysis tools on <a href="https://gtnsmrgsbord.slack.com/archives/C06PBRR40D7">#single-cell-users</a></p>   <p>We also post new tutorials / workflows there from time to time, as well as any other news.</p>   <p><i class="far fa-hand-point-right" aria-hidden="true"></i><span class="visually-hidden">point-right</span> If you’d like to contribute ideas, requests or feedback as part of the wider community building single-cell and spatial resources within Galaxy, you can also join our <a href="https://galaxyproject.org/community/sig/singlecell/">Single cell &amp; sPatial Omics Community of Practice</a>.</p>   <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> You can request tools here on our <a href="https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing">Single Cell and Spatial Omics Community Tool Request Spreadsheet</a></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="DESeq2"> <a href="#DESeq2"> Deseq2 </a></h3>

<hr/>



<h4 class="faq-area"> The tutorial uses the normalised count table for visualisation. What about using VST normalised counts or rlog normalised counts? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-the-tutorial-uses-the-normalised-count-table-for-visualisation-what-about-using-vst-normalised-counts-or-rlog-normalised-counts"><i class="far fa-question-circle" aria-hidden="true"></i> Question: The tutorial uses the normalised count table for visualisation. What about using VST normalised counts or rlog normalised counts?</div>   <p>this depends on what you would like to do with the table. The DESeq2 wrapper in Galaxy can output all of these, and there is a nice discussion in the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#count-data-transformations">DESeq2 vignette</a> about this topic.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="De novo transcriptome reconstruction with RNA-Seq"> <a href="#De novo transcriptome reconstruction with RNA-Seq"> De novo transcriptome reconstruction with rna-seq </a></h3>

<hr/>



<h4 class="faq-area"> I’m using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why? <a href="/training-material/topics/transcriptomics/tutorials/de-novo/faqs/transcript.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-i-m-using-the-same-training-data-tools-and-parameters-as-the-tutorial-but-i-get-a-different-number-of-transcripts-with-a-significant-change-in-gene-expression-between-the-g1e-and-megakaryocyte-cellular-states-why"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-i-m-using-the-same-training-data-tools-and-parameters-as-the-tutorial-but-i-get-a-different-number-of-transcripts-with-a-significant-change-in-gene-expression-between-the-g1e-and-megakaryocyte-cellular-states-why" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: I’m using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This is okay! Many aspects of the tutorial can potentially affect the exact results you obtain. For example, the reference genome version used and versions of tools. It’s less important to get the exact results shown in the tutorial, and more important to understand the concepts so you can apply them to your own data.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Interpretation"> <a href="#Interpretation"> Interpretation </a></h3>

<hr/>



<h4 class="faq-area"> What exactly is a ‘Gene profile’? <a href="/training-material/topics/single-cell/faqs/gene_profile.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-exactly-is-a-gene-profile"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-exactly-is-a-gene-profile" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What exactly is a ‘Gene profile’?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Think of it like a fingerprint that some cells exhibit and others don’t. It’s a small collection of genes which are up or down regulated in relation to one another. Their differences are not absolute, but relative. So if CellA has 100 counts of Gene1 and 50 counts of Gene2, this creates a relation of 2:1 between Gene1 and Gene2. If CellB has a 20 counts of Gene1 and 10 counts of Gene2, then they share the same relation. If CellA and CellB share other relations with other genes than this might be enough to say that they share a Gene profile, and will therefore likely cluster together as they describe the same cell type.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Resources"> <a href="#Resources"> Resources </a></h3>

<hr/>



<h4 class="faq-area"> Use our Single Cell Omics Lab <a href="/training-material/topics/single-cell/faqs/single_cell_omics.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-use-our-single-cell-omics-lab"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-use-our-single-cell-omics-lab" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Use our Single Cell Omics Lab</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Did you know we have a unique Single Cell Omics Lab with all our single cell tools highlighted to make it easier to use on Galaxy? We recommend this site for all your single cell analysis needs, particularly for newer users.</p>   <p>The Single Cell Omics Lab currently uses the main European Galaxy infrastructure and power, it’s just organised better for users of particular analyses…like single cell!</p>   <p>Try it out!</p>   <ul>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.eu">Europe: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org">USA: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org.au">Australia: Single Cell Omics Lab</a></li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Single-Cell RNA"> <a href="#Single-Cell RNA"> Single-cell rna </a></h3>

<hr/>



<h4 class="faq-area"> Why is amplification more of an issue in scRNA-seq than RNA-seq? <a href="/training-material/topics/single-cell/faqs/amplification.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-is-amplification-more-of-an-issue-in-scrna-seq-than-rna-seq"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-is-amplification-more-of-an-issue-in-scrna-seq-than-rna-seq" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why is amplification more of an issue in scRNA-seq than RNA-seq?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Due to the extremely small amount of starting material, the initial amplification is likely to be uneven due to the first cycle of amplified products being overrepresented in the second cycle of amplification leading to further bias. In Bulk RNA-seq, the larger selection of RNA molecules to amplify, evens out the odds that any one transcript will be amplified more than others.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="account"> <a href="#account"> Account </a></h3>

<hr/>



<h4 class="faq-area"> Can I create multiple Galaxy accounts? <a href="/training-material/faqs/galaxy/account_multiple.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-i-create-multiple-galaxy-accounts"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-i-create-multiple-galaxy-accounts" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can I create multiple Galaxy accounts?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The account registration form and activation email include a terms of service statement.</p>   <ul>   <li>You ARE NOT allowed to create more than 1 account per Galaxy server.</li>   <li>You ARE allowed to have accounts on different servers.</li> </ul>   <p>For example, you are allowed to have 1 account on Galaxy US, and another account on Galaxy EU, but never 2 accounts on the same Galaxy.</p>   <p><strong>WARNING: Having multiple accounts is a violation of the terms of service, and may result in deletion of your accounts.</strong></p>   <p><br /> Need more disk space?</p>   <ul>   <li>Review your <a href="https://my.galaxy.training/?path=/storage"><strong>User -&gt; Preferences -&gt; Storage Dashboard</strong></a> to find and manage all of your data.</li>   <li>Read about <a href="/training-material/faqs/galaxy/account_reduce_quota_usage.html">more ways to free up space in your account</a></li>   <li><a href="/training-material/faqs/galaxy/support_admins.html">Contact the admins of your Galaxy server</a> to ask about possibilities for temporarily increasing your quota.</li> </ul>   <p><br /> Other tips:</p>   <ul>   <li>Forgot your password? You can request a reset link in on the login page.</li>   <li>If you want to associate your account with a different email address, you can do so under <strong>User -&gt; Preferences</strong> in the top menu bar.</li>   <li>To start over with a new account, delete your existing account(s) first before creating your new account. This can be done in <strong>User -&gt; Preferences</strong> menu in the top bar.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing acount email or password <a href="/training-material/faqs/galaxy/user_preferences_manage_information.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-acount-email-or-password"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-acount-email-or-password" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing acount email or password</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Make sure you are logged in to Galaxy.</li>   <li>Go to <strong>User &gt; Preferences</strong> in the top menu bar.</li>   <li>To change email and public name, click on <strong>Manage Information</strong> and to change password, click on <strong>Change Password</strong>.</li>   <li>Make the changes and click on the <strong>Save</strong> button at the bottom.</li>   <li>To change email successfully, verify your account by email through the activation link sent by Galaxy.</li> </ol>   <p>Note: Don’t open another account if your email changes, update the existing account email instead. Creating a new account will be detected as a duplicate and will get your account disabled and deleted.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I reduce quota usage while still retaining prior work (data, tools, methods)? <a href="/training-material/faqs/galaxy/account_reduce_quota_usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-reduce-quota-usage-while-still-retaining-prior-work-data-tools-methods"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-reduce-quota-usage-while-still-retaining-prior-work-data-tools-methods" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I reduce quota usage while still retaining prior work (data, tools, methods)?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li><a href="/training-material/faqs/galaxy/datasets_download_datasets.html"><strong>Download</strong></a> Datasets as individual files or entire Histories as an archive. Then <em>purge</em> them from the public server.</li>   <li><strong>Transfer/Move</strong> <a href="/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html">Datasets</a> or <a href="/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html">Histories</a> to another Galaxy server, including your own Galaxy. Then purge.</li>   <li><a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><strong>Copy</strong></a> your most important Datasets into a new/other History (inputs, results), then purge the original full History.</li>   <li><a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><strong>Extract</strong></a> a <strong>Workflow</strong> from the History, then purge it.</li>   <li><strong>Back-up your work</strong>. It is a best practice to download an archive of your FULL original Histories periodically, even those still in use, as a backup.</li> </ul>   <p><strong>Resources</strong> Much discussion about all of the above options can be found at the <a href="https://help.galaxyproject.org/">Galaxy Help forum</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I create an account on a public Galaxy instance? <a href="/training-material/faqs/galaxy/account_create.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-create-an-account-on-a-public-galaxy-instance"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-create-an-account-on-a-public-galaxy-instance" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I create an account on a public Galaxy instance?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p>To create an account at any public Galaxy instance, choose your server from the available <a href="https://galaxyproject.org/use/">list of Galaxy Platforms</a>.</p>     <p>There are several UseGalaxy servers:</p>     <ul>       <li>         <p><a href="https://usegalaxy.org.au">UseGalaxy.org.au</a> (<strong>AU</strong>)</p>       </li>       <li>         <p><a href="https://usegalaxy.fr">UseGalaxy.fr</a> (<strong>FR</strong>)</p>       </li>       <li>         <p><a href="https://usegalaxy.org">UseGalaxy.org</a> (<strong>US</strong>)</p>       </li>       <li>         <p><a href="https://usegalaxy.eu">UseGalaxy.eu</a> (<strong>EU</strong>)</p>       </li>     </ul>   </li>   <li>     <p>Click on <strong>“Login or Register”</strong> in the masthead on the server.</p>     <p><img src="/training-material/faqs/galaxy/images/login_register.png" alt="Login or Register on the top panel" /></p>   </li>   <li>     <p>On the login page, find the <strong>Register here</strong> link and click on it.</p>   </li>   <li>     <p>Fill in the the registration form, then click on <strong>Create</strong>.</p>     <p>Your account should now get created, but will remain inactive until you verify the email address you provided in the registration form.</p>     <p><img src="/training-material/faqs/galaxy/images/unverified_account.png" alt="Banner warning about account with unverified email address" /></p>   </li>   <li>     <p>Check for a <strong>Confirmation Email</strong> in the email you used for account creation.</p>     <p>Missing? Check your <strong>Trash</strong> and <strong>Spam</strong> folders.</p>   </li>   <li>     <p>Click on the <strong>Email confirmation link</strong> to fully activate your account.</p>     <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> Delivery of the confimation email is blocked by your email provider or you mistyped the email address in the registration form?</p>     <p>Please do <strong>not</strong> register again, but follow the <a href="/training-material/faqs/galaxy/user_preferences_manage_information.html">instructions to change the email address</a> registered with your account! The confirmation email will be resent to your new address once you have changed it.</p>     <p>Trouble logging in later? Account email addresses and public names are caSe-sensiTive. Check your activation email for formats.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to update account preferences? <a href="/training-material/faqs/galaxy/account_update_preference.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-update-account-preferences"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-update-account-preferences" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to update account preferences?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Log in to Galaxy</li>   <li>Navigate to <a href="https://my.galaxy.training/?path=/user"><strong>User -&gt; Preferences</strong></a> on the top menu bar</li>   <li>Here you can update various preferences, such as:     <ul>       <li><i class="fas fa-user" aria-hidden="true"></i><span class="visually-hidden">pref-info</span> Manage Information (change your registered email addresses or public name)</li>       <li><i class="fas fa-unlock-alt" aria-hidden="true"></i><span class="visually-hidden">pref-password</span> Change Password (change your login credentials)</li>       <li><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">pref-permissions</span> Set Dataset Permissions for New Histories (grant others default access to newly created histories)</li>       <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">pref-toolboxfilters</span> Manage Toolbox Filters (customize your Toolbox by displaying or omitting sets of Tools)</li>       <li><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">pref-apikey</span> Manage API Key (access your current API key or create a new one)</li>       <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">pref-notifications</span> Manage Notifications (allow push and tab notifcations on job completion)</li>       <li><i class="fas fa-cloud" aria-hidden="true"></i><span class="visually-hidden">pref-cloud</span> Manage Cloud Authorization (grants Galaxy to access your cloud-based resources)</li>       <li><i class="far fa-id-card" aria-hidden="true"></i><span class="visually-hidden">pref-identities</span> Manage Third-Party Identities (connect or disconnect access to your third-party identities)</li>       <li><i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">pref-custombuilds</span> Manage Custom Builds (custom databases based on fasta datasets)</li>       <li><i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden">pref-list</span> Manage Activity Bar (a bonus navigation bar)</li>       <li><i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">pref-palette</span> Pick a Color Theme (interface color theme)</li>       <li><i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Make All Data Private (disable all data sharing)</li>       <li><i class="fas fa-radiation" aria-hidden="true"></i><span class="visually-hidden">pref-delete</span> Delete Account (on this Galaxy server)</li>       <li><i class="fa fa-sign-out" aria-hidden="true"></i><span class="visually-hidden">pref-signout</span> Sign out of Galaxy (signs you out of all sessions)</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="analysis"> <a href="#analysis"> Analysis </a></h3>

<hr/>



<h4 class="faq-area"> Adding a custom database/build (dbkey) <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy may have several reference genomes built-in, but you can also create your own.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-custom-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-custom-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a custom database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Navigate to the History that contains your fasta for the reference genome</li>   <li>Standarize the <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html">fasta format</a></li>   <li>In the top menu bar, go to <strong>User -&gt; Preferences -&gt; Manage Custom Builds</strong></li>   <li>Create a unique <strong>Name</strong> for your reference build</li>   <li>Create a unique <strong>Database</strong> (dbkey) for your reference build</li>   <li>Under <strong>Definition</strong>, select the option <code class="language-plaintext highlighter-rouge">FASTA-file from history</code></li>   <li>Under <strong>FASTA-file</strong>, select your fasta file</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Beware of Cuts <a href="/training-material/faqs/galaxy/analysis_cut.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy has several different cut tools
<!--SNIPPET--><blockquote class="warning">   <div class="box-title warning-title" id="warning-beware-of-cuts"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i> Warning: Beware of Cuts</div>   <p>The section below uses <strong>Cut</strong> tool. There are two <strong>cut</strong> tools in Galaxy due to historical reasons. This example uses tool with the full name <strong>Cut columns from a table (cut)</strong>. However, the same logic applies to the other tool. It simply has a slightly different interface.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Does MaxQuant in Galaxy support TMT, iTRAQ, etc.? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/tmt_itraq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-does-maxquant-in-galaxy-support-tmt-itraq-etc"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Does MaxQuant in Galaxy support TMT, iTRAQ, etc.?</div>   <p>Yes, iTRAQ 4 and 8 plex;  TMT 2,6,8,10,11 plex; iodoTMT6plex</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Extended Help for Differential Expression Analysis Tools <a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-extended-help-for-differential-expression-analysis-tools"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-extended-help-for-differential-expression-analysis-tools" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Extended Help for Differential Expression Analysis Tools</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The error and usage help in this FAQ applies to most if not all Bioconductor tools.</p>   <ul>   <li>DEseq2</li>   <li>Limma</li>   <li>edgeR</li>   <li>goseq</li>   <li>Diffbind</li>   <li>StringTie</li>   <li>Featurecounts</li>   <li>HTSeq-count</li>   <li>HTseq-clip</li>   <li>Kalisto</li>   <li>Salmon</li>   <li>Sailfish</li>   <li>DEXSeq</li>   <li>DEXSeq-count</li>   <li>IsoformSwitchAnalyzeR</li> </ul>   <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span>  Review your error messages and you’ll find some clues about what may be going wrong and what needs to be adjusted in your rerun. If you are getting a message from <code class="language-plaintext highlighter-rouge">R</code>, that usually means the underlying tool could not read in or understand your inputs. This can be a labeling problem (what was typed on the form) or a content problem (data within the files).</p>   <p>Expect odd errors or content problems if any of the usage requirements below are not met.</p>   <p>General</p>   <ul>   <li>Are your reference genome, reference transcriptome, and reference annotation all based on the same genome assembly?     <ul>       <li>Check the identifiers in all inputs and adjust as needed.</li>       <li>These all may mean the same thing to a person but not to a computer or tool: chr1, Chr1, 1, chr1.1</li>     </ul>   </li>   <li>Differential expression tools all require sample count replicates. <a href="https://www.seqanswers.com/forum/bioinformatics/bioinformatics-aa/26388-deseq2-without-biol-replicates">Rationale from two of the DEseq tool authors</a>.     <ul>       <li>At least two factor levels/groups/conditions with two samples each.</li>       <li>All must all contain unique content for valid scientific results.</li>     </ul>   </li>   <li>Factor/Factor level names should only contain alphanumeric characters and optionally underscores.     <ul>       <li>Avoid starting these with a number and do not include spaces.</li>       <li>Galaxy <em>may</em> be able to normalize these values for you, but if you are getting an error: standardize the format yourself.</li>     </ul>   </li>   <li><strong>DEXSeq</strong> additionally requires that the first Condition is labeled as <code class="language-plaintext highlighter-rouge">Condition</code>.</li>   <li>If your count inputs have a header, the option <strong>Files have header?</strong> is set to <strong>Yes</strong>. If no headers, set to <strong>No</strong>.     <ul>       <li>If your files have more than one header line: keep the sample header line, remove all extra line(s).</li>     </ul>   </li>   <li>Make sure that tool form settings match your annotation content or the tool cannot match up the inputs!     <ul>       <li>If you are counting by <strong>gene_id</strong>, your annotation should contain gene_id attributes (9th column)</li>       <li>If you are summarizing by <strong>exon</strong>, your annotation should contain exon features (3rd column)</li>     </ul>   </li>   <li>Sometimes these tools do not understand <code class="language-plaintext highlighter-rouge">transcript_id.N</code> and <code class="language-plaintext highlighter-rouge">gene_id.N</code> notation (where N is a version number).     <ul>       <li>This notation could be in fasta or tabular inputs.</li>       <li>Try <a href="/training-material/search2.html?query=olympics">removing <code class="language-plaintext highlighter-rouge">.N</code> from all inputs</a>, and check for the accidential creation of new duplicates!</li>     </ul>   </li>   <li>Errors? <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Understanding the job log messages</a> can be confusing! But are accessible and worth reviewing.     <ul>       <li>The good news is that usage in Galaxy produces the same error messages as direct usage.</li>       <li>This means that a search at the <a href="https://support.bioconductor.org/">Bioconductor Support</a> website can provide useful clues! Come back to the <a href="https://help.galaxyproject.org/">Galaxy Help</a> forum with any remaining questions.</li>     </ul>   </li> </ul>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Remember, for any value in your inputs that is not a number, using only alphanumeric characters and optionally underscores <code class="language-plaintext highlighter-rouge">_</code> with no spaces is what the authors recommend. Check your factor names, sample names, gene identifiers, transcript identifiers, and header lines in files.</p>   <p>Reference genome (fasta)</p>   <ul>   <li>Can be a server reference genome (hosted index in the pull down menu) or a custom reference genome (fasta from the history).</li>   <li>Custom reference genomes must be <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html">formatted correctly</a>.</li>   <li>If you are using <strong>Salmon</strong> or <strong>Kalisto</strong>, you probably don’t need a reference genome but a reference transcriptome instead!</li>   <li>More about understanding and <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">working with large fasta datasets</a>.</li> </ul>   <p>Reference transcriptome (fasta)</p>   <ul>   <li>Fasta file containing assembled transcripts.</li>   <li>Unassembled short or long reads will not work as a substitute.</li>   <li>The transcript identifiers on the <code class="language-plaintext highlighter-rouge">&gt;seq</code> fasta lines must exactly match the <code class="language-plaintext highlighter-rouge">transcript_id</code> values in your annotation or tabular mapping file.</li> </ul>   <p>Reference annotation (tabular, GTF, GFF3)</p>   <ul>   <li>Reference annotation <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">in GTF format</a> works best.</li>   <li>If a GTF dataset is not available for your genome, a two-column tabular dataset containing <code class="language-plaintext highlighter-rouge">transcript &lt;tab&gt; gene</code> can be used instead with most of these tools.</li>   <li><strong>HTseq-count</strong> requires GTF attributes. Featurecounts is an alternative tool choice.</li>   <li>Sometimes the tool <strong>gffread</strong> is used to transform GFF3 data to GTF.</li>   <li>DO use UCSC’s reference annotation (GTF) and reference transcriptome (fasta) data from their <a href="https://hgdownload.soe.ucsc.edu/downloads.html">Downloads</a> area.     <ul>       <li>These are a match for the UCSC genomes indexed at public Galaxy servers.</li>       <li>Links can be directly copy/pasted into the Upload tool.</li>       <li>Allow Galaxy to <em>autodetect the datatype</em> to produce an uncompressed dataset in your history ready to use with tools.</li>     </ul>   </li>   <li>Avoid GTF data from the UCSC Table Browser: this leads to scientific problems. GTFs will have the same content populated for both the transcript_id and gene_id values. See the note at UCSC for more about why.</li>   <li>Still have problems? Try removing all GTF header lines with the tool <strong>Remove beginning of a file</strong>.</li>   <li>More about understanding and <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">working with GTF/GFF/GFF3 reference annotation</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> For the “quantitation method” what is the default if I just leave it as “None”? Label free? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/quantitation_default.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-for-the-quantitation-method-what-is-the-default-if-i-just-leave-it-as-none-label-free"><i class="far fa-question-circle" aria-hidden="true"></i> Question: For the “quantitation method” what is the default if I just leave it as “None”? Label free?</div>   <p>It will report raw intensity (NON-normalized) values which were not normalized like e.g. the LFQ intensities.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I adapt this tutorial to my own data? <a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-how-can-i-adapt-this-tutorial-to-my-own-data"><i class="far fa-question-circle" aria-hidden="true"></i> Question: How can I adapt this tutorial to my own data?</div>   <p>If you would like to run this analysis on your own data, make sure to check which V-region was sequenced. In this tutorial, we sequenced the V4 region, and used a corresponding reference for just this region. If you sequenced another V-region, please use an appropriate reference (either the full SILVA reference, or the SILVA reference specific for your region). Similarly, the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step after the alignment filtered on start and end coordinates of the alignments. These will have to be adjusted to your V-region.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I adapt this tutorial to my own data? <a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-how-can-i-adapt-this-tutorial-to-my-own-data-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question: How can I adapt this tutorial to my own data?</div>   <p>If you would like to run this analysis on your own data, make sure to check which V-region was sequenced. In this tutorial, we sequenced the V4 region, and used a corresponding reference for just this region. If you sequenced another V-region, please use an appropriate reference (either the full SILVA reference, or the SILVA reference specific for your region). Similarly, the <code class="language-plaintext highlighter-rouge">Screen.seqs</code> step after the alignment filtered on start and end coordinates of the alignments. These will have to be adjusted to your V-region.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I do analysis X? - Getting help <a href="/training-material/faqs/galaxy/analysis_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-do-analysis-x-getting-help"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-do-analysis-x-getting-help" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I do analysis X? - Getting help</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you don’t know how to perform a certain analysis, you can ask the Galaxy community for help.</p>   <h4 id="where-to-ask">Where to ask</h4>   <p>The best places to ask your analysis questions are:</p>   <ul>   <li><a href="https://help.galaxyproject.org">Galaxy Help forum</a></li>   <li><a href="https://matrix.to/#/#Galaxy-Training-Network_Lobby:gitter.im">GTN Matrix chat</a></li> </ul>   <p><strong>Note:</strong> For questions about errors you’ve encountered in Galaxy, please see our <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">troubleshooting page</a>.</p>   <h4 id="how-to-ask">How to ask</h4>   <p>The more detail you provide, the better we can help you. Please provide information about:</p>   <ul>   <li><strong>Your data and experiment</strong> e.g. “paired-end RNASeq, mouse, 16 triplicates, 2 timepoints”, etc</li>   <li><strong>Your goal and research question</strong> e.g. “I want to detect diffentially expressed genes between these two groups and generate a volcano plot”</li>   <li><strong>What you have already tried?</strong> Do you already know which tools you want to use? Did you already try some but they didn’t work? Why not? Did you find good papers describing something similiar to what you want to do? etc.</li>   <li><strong>Which Galaxy are you using?</strong> And if you have already tried some steps, please <a href="/training-material/faqs/galaxy/histories_sharing.html">share your Galaxy history via URL</a> and provide this along with your question.</li>   <li><strong>Examples</strong>     <ul>       <li><strong>Bad Question:</strong> <em>“Help!!! How to perform metagenomics analysis. I need it urgent!”</em></li>       <li><strong>Good Question:</strong> <em>“Hello everybody, I have 16S rRNA sequencing data from Illumina, it was paired-end with 150bp reads. I want to perform a taxonomy analysis similar to this paper (provide link). I have followed this GTN tutorial (provide link), but my data is different because (reason) . How can I adapt this step of the analysis for my data? I read about a tool called X, but I cannot find it in Galaxy. I am using Galaxy EU, and here is a link to my history. Any help would be greatly appreciated!”</em></li>     </ul>   </li> </ul>   <h4 id="before-you-ask">Before you ask</h4>   <ul>   <li>Check the <a href="https://help.galaxyproject.org/">Galaxy Help forum</a> to see if others have already asked a similar question before.</li>   <li><a href="/training-material/search2.html">Search</a> the <a href="https://training.galaxyproject.org">GTN website</a> for a tutorial that matches what you want to do, and work your way through that. Even if it doesn’t doe exactly what you need, you usually learn a lot along the way that will help you adapt it to your own data or research question.</li> </ul>   <h4 id="be-patient">Be patient</h4>   <p>Please remember that most of the people answering questions on Matrix chat and the help forum are <strong>volunteers</strong> from the community. They take time out of their busy days to help you. They may also be in a different time zone, so it may take some time to get answers. Please always be patient and kind to each other, and adhere to our <a href="https://galaxyproject.org/community/coc/">code of conduct</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How many proteins can be identified and quantified in shotgun proteomics? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/shotgun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-how-many-proteins-can-be-identified-and-quantified-in-shotgun-proteomics"><i class="far fa-question-circle" aria-hidden="true"></i> Question: How many proteins can be identified and quantified in shotgun proteomics?</div>   <p>This is depending on the sample, the used technique(s) and the mass spectrometer. Routinely most labs obtain 4000 proteins, but with more effort  10.000 proteins could be analyzed in a single run.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> I got slightly different numbers than were in the tutorial <a href="/training-material/topics/introduction/faqs/ucsc.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-i-got-slightly-different-numbers-than-were-in-the-tutorial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-i-got-slightly-different-numbers-than-were-in-the-tutorial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: I got slightly different numbers than were in the tutorial</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This tutorial uses UCSC which is constantly updating it’s data! As a result it gets outdated very quickly before we can update it :( But it’s ok! It’s expected here to get different numbers.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> If you use a mqpar file, can you include modifications that are not in the Galaxy version? For instance, propionamide (Cys alkylation by acrylamide). <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/mqpar.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-if-you-use-a-mqpar-file-can-you-include-modifications-that-are-not-in-the-galaxy-version-for-instance-propionamide-cys-alkylation-by-acrylamide"><i class="far fa-question-circle" aria-hidden="true"></i> Question: If you use a mqpar file, can you include modifications that are not in the Galaxy version? For instance, propionamide (Cys alkylation by acrylamide).</div>   <p>No, one is limited to the modifications which are installed in MaxQuant. The mqpar only contains more parameters / options than the GUI in galaxy. Note: one must use an mqpar from the same version like MaxQuant!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Including custom modifications into MaxQuant in Galaxy? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-including-custom-modifications-into-maxquant-in-galaxy"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Including custom modifications into MaxQuant in Galaxy?</div>   <p>Unfortunately the inclusion of custom modifications is not possible by the user because it requires profound changes in the underlying code. Please let us know the modification you need by creating a new issue: https://github.com/galaxyproteomics/tools-galaxyp/issues entitled MaxQuant new modification request.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> MSStats: what does ‘compare groups = yes’ mean? And the comparison matrix to define the contrast between the 2 groups? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/stats.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-msstats-what-does-compare-groups-yes-mean-and-the-comparison-matrix-to-define-the-contrast-between-the-2-groups"><i class="far fa-question-circle" aria-hidden="true"></i> Question: MSStats: what does ‘compare groups = yes’ mean? And the comparison matrix to define the contrast between the 2 groups?</div>   <p>MSstats consists of three parts:</p>   <ul>   <li>Reading the input files and converting them into an MSstats compatible format, doing some processing of the data at the same time</li>   <li>Data processing: such as protein inference (summary), log2 transformation, normalization and missing value imputation</li>   <li><code class="language-plaintext highlighter-rouge">compare groups = yes</code>, means that the third step is performed, which is statistical analysis: Statistical modelling to find differentially abundant protein between different groups. The groups should be specified as “condition” in the annotation file and the group comparison matrix file specifies which groups to compare against each other. In the example this is quite simple because there are only 2 groups, with 3 or more groups the comparison matrix could become more complex.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> My jobs aren't running! <a href="/training-material/faqs/galaxy/analysis_jobs_not_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-my-jobs-aren-t-running"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-my-jobs-aren-t-running" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: My jobs aren't running!</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p><strong>Please make sure you are logged in</strong>. At the top menu bar, you should see a section labeled “User”. If you see “Login/Register” here you are not logged in.</p>   </li>   <li><strong>Activate your account</strong>. If you have recently registered your account, you may first have to activate it. You will receive an e-mail with an activation link.     <ul>       <li>Make sure to check your spam folder!</li>     </ul>   </li>   <li>     <p><strong>Be patient</strong>. Galaxy is a free service, when a lot of people are using it, you may have to wait longer than usual (especially for ‘big’ jobs, e.g. alignments).</p>   </li>   <li><strong>Contact Support</strong>. If you really think something is wrong with the server, you can <a href="/training-material/faqs/galaxy/support.html">ask for support</a></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Pick the right Concatenate tool <a href="/training-material/faqs/galaxy/analysis_concatenate.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Most Galaxy servers will have two Concatenate tools installed - know which one to pick!
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-pick-the-right-concatenate-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-pick-the-right-concatenate-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Pick the right Concatenate tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>On most Galaxy servers you will find two <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Concatenate datasets</strong> tools installed:</p>   <ol>   <li><strong>Concatenate datasets</strong> tail-to-head</li>   <li><strong>Concatenate datasets</strong> tail-to-head (cat)</li> </ol>   <p>The two tools have nearly identical interfaces, but behave differently in certain situations, specifically:</p>   <ul>   <li>     <p>The second tool, the one with “(cat)” in its name, simply concatenates everything you give to it into a single output dataset.</p>     <p>Whether you give it multiple datasets or a collection as the first parameter, or some datasets as the first and some others as the second parameter, it will always concatenate them all. In fact, the only reason for having multiple parameters for this tool is that by providing inputs through multiple parameters, you can make sure they are concatenated in the order you pass them in.</p>   </li>   <li>     <p>The first tool, on the other hand, will only ever concatenate inputs provided through <em>different</em> parameters.</p>     <p>This tool allows you to specify an arbitrary number of <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, but if you also want to use <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> multiple datasets or <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> a collection for some of the Dataset parameters, then all of these need to be of the same type (multiple datasets <em>or</em> collections) and have the same number of inputs.</p>     <p>Now depending on the inputs, one of the following behaviors will occur:</p>     <ul>       <li>If all the different inputs are <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, the tool will concatenate them all and produce a single output dataset.</li>       <li>If all the different inputs are specified <em>either</em> as <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> multiple datasets <em>or</em> as <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span>, and all have the same number of datasets, then the tool will concatenate the first datasets of each input parameter, the second datasets of each input parameter, the third, etc., and produce an output collection with as many elements as there are inputs per Dataset parameter.</li>       <li>In extension of the above, if <em>some</em> additional inputs are provided as <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, the content of these will be recycled and be reused in the concatenation of all the nth elements of the other parameters.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Reporting usage problems, security issues, and bugs <a href="/training-material/faqs/galaxy/analysis_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-reporting-usage-problems-security-issues-and-bugs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-reporting-usage-problems-security-issues-and-bugs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Reporting usage problems, security issues, and bugs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>For reporting <strong>Usage Problems</strong>, related to tools and functions, head to the <a href="https://help.galaxyproject.org/">Galaxy Help</a> site.     <ul>       <li><strong>Red Error Datasets</strong>:         <ul>           <li>Refer to the <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting errors</a> FAQ for red error in datasets.</li>         </ul>       </li>       <li><strong>Unexpected results in Green Success Dataset</strong>:         <ul>           <li>To resolve it you may be asked to send in a shared history link and possibly a shared workflow link. For sharing your history, refer to this <a href="/training-material/faqs/galaxy/histories_sharing.html">these instructions</a>.</li>           <li>To reach our support team, visit <a href="/training-material/faqs/galaxy/#support">Support FAQs</a>.</li>         </ul>       </li>       <li><strong>Functionality problems</strong>:         <ul>           <li>Using <a href="https://help.galaxyproject.org/">Galaxy Help</a> is the best way to get help in most cases.</li>           <li>If the problem is more complex, email a description of the problem and how to reproduce it.</li>         </ul>       </li>       <li><strong>Administrative problems</strong>:         <ul>           <li>If the problem is present in your own Galaxy, the administrative configuration may be a factor.</li>           <li>For the fastest help directly from the development community, admin issues can be alternatively reported to the mailing list or the GalaxyProject <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a>.</li>         </ul>       </li>     </ul>   </li>   <li>For <strong>Security Issues</strong>, do not report them via GitHub. Kindly disclose these as explained in this <a href="https://github.com/galaxyproject/galaxy/blob/dev/SECURITY.md">document</a>.</li>   <li>For <strong>Bug Reporting</strong>, create a Github issue. Include the steps mentioned <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">in these instructions</a>.</li>   <li>Search the <a href="https://training.galaxyproject.org/search">GTN Search</a> to find prior Q &amp; A, FAQs, tutorials, and other documentation across all Galaxy resources, to verify in case your issue was already faced by someone.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Results may vary <a href="/training-material/faqs/galaxy/analysis_results_may_vary.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Results may vary</div>   <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, external databases, or because of stochastic processes in the algorithms.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html"><i class="fas fa-external-link-alt"></i></a> </h4>
When you get a red dataset in your history, it means something went wrong. But how can you find out what it was? And how can you report errors?
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Troubleshooting errors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When something goes wrong in Galaxy, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your Galaxy errors.</p>   <ol>   <li><strong>Expand</strong> the red history dataset by clicking on it.     <ul>       <li>Sometimes you can already see an error message here</li>     </ul>   </li>   <li>     <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></p>   </li>   <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:     <ul>       <li>Expand the history item</li>       <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>       <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:         <ul>           <li>Tool Standard Output</li>           <li>Tool Standard Error</li>         </ul>       </li>       <li>For more information about specific tool errors, please see the <a href="/training-material/faqs/galaxy/index.html#troubleshooting">Troubleshooting section</a></li>     </ul>   </li>   <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.     <ul>       <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></li>       <li>Write down any information you think might help solve the problem         <ul>           <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>         </ul>       </li>       <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> Report</strong> button</li>     </ul>   </li>   <li><strong>Ask for help!</strong>     <ul>       <li>Where?         <ul>           <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Matrix Channel</a></li>           <li>In the <a href="https://gitter.im/galaxyproject/Lobby">Galaxy Matrix Channel</a></li>           <li>Browse the <a href="https://help.galaxyproject.org/">Galaxy Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>         </ul>       </li>       <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What does it mean to normalize the LFQ intensities? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/normalize_lfq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-does-it-mean-to-normalize-the-lfq-intensities"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What does it mean to normalize the LFQ intensities?</div>   <p>Median normalization typically refers to subtracting the median of all intensities within one sample from all of the intensities (e.g. Intensity of Protein A - Median of all intensities from Sample 1) , to account for measurement variations. Before normalization log2 transformation is required since many statistical tests demand that the data is actually normal distributed. (Non log intensities show very high values but have a minimum (limit of quantification) leading to a somehow right skewed distribution, after log-transformation the intensity distribution is more like a gaussian distribution. Beside the median (or median-polish) normalization there is also other e.g. the quantile  normalization.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is the advantage of breaking down protein to peptides before mass spec? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/breakdown.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-is-the-advantage-of-breaking-down-protein-to-peptides-before-mass-spec"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What is the advantage of breaking down protein to peptides before mass spec?</div>   <p>Mass spectrometry works better for peptides: LC separation and ionization is working better on peptides than on proteins and proteins generate too complex and overlaying mass spectra due to their isotopes and their mass might be shifted due to posttranslational modifications or point mutations.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> When can you use (or cannot use) Match between runs in MaxQuant? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/match.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-when-can-you-use-or-cannot-use-match-between-runs-in-maxquant"><i class="far fa-question-circle" aria-hidden="true"></i> Question: When can you use (or cannot use) Match between runs in MaxQuant?</div>   <p>No golden rule here. For quantitative comparison of different sample groups it can be valuable to use MBR to increase the number of identified + quantified proteins in all samples and then have more proteins that occur in most of the samples to compare them.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Which isobaric labeled quantification methods does MaxQuant in Galaxy support? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-which-isobaric-labeled-quantification-methods-does-maxquant-in-galaxy-support"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Which isobaric labeled quantification methods does MaxQuant in Galaxy support?</div>   <p>The current MaxQuant version supports: iTRAQ 4 and 8 plex; TMT 2,6,8,10,11 plex; iodoTMT6plex. Includion of TMT16 plex is in preparation.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Will my jobs keep running? <a href="/training-material/faqs/galaxy/analysis_jobs_keep_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-will-my-jobs-keep-running"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-will-my-jobs-keep-running" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Will my jobs keep running?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy is a fantastic system, but some users find themselves wondering:</p>   <blockquote>   <p>Will my jobs keep running once I’ve closed the tab? Do I need to keep my browser open?</p> </blockquote>   <p>No, you don’t! You can safely:</p>   <ol>   <li>Start jobs</li>   <li>Shut down your computer</li> </ol>   <p>and your jobs will keep running in the background! Whenever you next visit Galaxy, you can check if your jobs are still running or completed.</p>   <p><strong>However</strong>, this is not true for uploading data from your computer. You must wait for uploading a dataset <em>from your computer</em> to finish. (Uploading via URL is not affected by this, if you’re uploading from URL you can close your computer.)</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="ansible"> <a href="#ansible"> Ansible </a></h3>

<hr/>



<h4 class="faq-area"> Debugging Memory Leaks <a href="/training-material/topics/admin/faqs/python-memory-leaks.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-debugging-memory-leaks"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-debugging-memory-leaks" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Debugging Memory Leaks</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://github.com/bloomberg/memray">memray</a> is a great memory profiler for debugging memory issues.</p>   <p>In the context of Galaxy, this is significantly easier for job handlers. Install it in your <code class="language-plaintext highlighter-rouge">virtualenv</code> and</p>   <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">memray run  --trace-python-allocators -o the_dump &lt;your_handler_startup_command_here&gt;</span><span class="w"><br></span></code></pre></div></div>   <p>Once you’ve collected enough data,</p>   <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="go">memray flamegraph --leaks --temporal the_dump -o the_dump.html<br></span></code></pre></div></div>   <p>would then produce a report that shows allocation made but not freed over time.</p>   <p>It might also be useful to just check what the process is doing with py-spy dump.</p>   <p>You can follow web workers in gunicorn with</p>   <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="go">memray run --follow-fork -o the_dump gunicorn 'galaxy.webapps.galaxy.fast_factory:factory()' --timeout 600 --pythonpath lib -k galaxy.webapps.galaxy.workers.Worker -b localhost:8082 --config python:galaxy.web_stack.gunicorn_config -w 1 --preload<br></span></code></pre></div></div>   <p>the traced app will run on port 8082, you can then for instance in an upstream nginx section direct a portion of the traffic to your profiled app.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Define once, reference many times <a href="/training-material/topics/admin/faqs/ansible_define-once.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-define-once-reference-many-times"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-define-once-reference-many-times" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Define once, reference many times</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Using variables, either by defining them ahead of time, or simply accessing them via existing data structures that have been defined, e.g.:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># defining a variable that gets reused is great!</span><br><span class="na">galaxy_user</span><span class="pi">:</span> <span class="s">galaxy</span><br><br><span class="na">galaxy_config</span><span class="pi">:</span><br>  <span class="na">galaxy</span><span class="pi">:</span><br>    <span class="c1"># Re-using the galaxy_config_dir variable saves time and ensures everything</span><br>    <span class="c1"># is in sync!</span><br>    <span class="na">datatypes_config_file</span><span class="pi">:</span> <span class="s2">"</span><span class="s">{{</span><span class="nv"> </span><span class="s">galaxy_config_dir</span><span class="nv"> </span><span class="s">}}/datatypes_conf.xml"</span><br><br><span class="c1"># and now we can re-use "{{ galaxy_config.galaxy.datatypes_config_file }}"</span><br><span class="c1"># in other places!</span><br><br><span class="na">galaxy_config_templates</span><span class="pi">:</span><br>  <span class="pi">-</span> <span class="na">src</span><span class="pi">:</span> <span class="s">templates/galaxy/config/datatypes_conf.xml</span><br>    <span class="na">dest</span><span class="pi">:</span> <span class="s2">"</span><span class="s">{{</span><span class="nv"> </span><span class="s">galaxy_config.galaxy.datatypes_config_file</span><span class="nv"> </span><span class="s">}}"</span><br></code></pre></div></div>   <p>Practices like those shown above help to avoid problems caused when paths are defined differently in multiple places. The datatypes config file will be copied to the same path as Galaxy is configured to find it in, because that path is only defined in one place. Everything else is a reference to the original definition! If you ever need to update that definition, everything else will be updated accordingly.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Error: "skipping: no hosts matched" <a href="/training-material/topics/admin/faqs/ansible_error-no-hosts-matched.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-error-quot-skipping-no-hosts-matched-quot"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-error-quot-skipping-no-hosts-matched-quot" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Error: &quot;skipping: no hosts matched&quot;</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There can be multiple reasons this happens, so we’ll step through all of them. We’ll start by assuming you’re running the command</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ansible-playbook galaxy.yml<br></code></pre></div></div>   <p>The following things can cause issues:</p>   <ol>   <li>Within your <code class="language-plaintext highlighter-rouge">galaxy.yml</code>, you’ve referred to a host group that doesn’t exist or is misspelled. Check the <code class="language-plaintext highlighter-rouge">hosts: galaxyservers</code> to ensure it matches the host group defined in the <code class="language-plaintext highlighter-rouge">hosts</code> file.</li>   <li>Vice-versa, the group in your <code class="language-plaintext highlighter-rouge">hosts</code> file should match the hosts selected in the playbook, <code class="language-plaintext highlighter-rouge">galaxy.yml</code>.</li>   <li>If neither of these are the issue, it’s possible Ansible doesn’t know to check the <code class="language-plaintext highlighter-rouge">hosts</code> file for the inventory. Make sure you’ve specified <code class="language-plaintext highlighter-rouge">inventory = hosts</code> in your <code class="language-plaintext highlighter-rouge">ansible.cfg</code>.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Failing all jobs from a specific user <a href="/training-material/topics/admin/faqs/maintain_fail-users-jobs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-failing-all-jobs-from-a-specific-user"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-failing-all-jobs-from-a-specific-user" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Failing all jobs from a specific user</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This command will let you quickly fail every job from the user ‘service-account’ (replace with your preferred user)</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>gxadmin tsvquery jobs --user=service-account --nonterminal | awk '{print $1}' |  xargs -I {} -n 1 gxadmin mutate fail-job {} --commit<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Galaxy Admin Training Path <a href="/training-material/topics/admin/faqs/git-gat-path.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-galaxy-admin-training-path"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Galaxy Admin Training Path</div>   <p>The yearly Galaxy Admin Training follows a specific ordering of tutorials. Use this timeline to help keep track of where you are in Galaxy Admin Training.</p>   <ol id="git-gat-timeline">                    <li class="disabled">         <a href="/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html">             <div>Step 1</div>             <div>ansible-galaxy</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/backup-cleanup/tutorial.html">             <div>Step 2</div>             <div>backup-cleanup</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/customization/tutorial.html">             <div>Step 3</div>             <div>customization</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/tus/tutorial.html">             <div>Step 4</div>             <div>tus</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/cvmfs/tutorial.html">             <div>Step 5</div>             <div>cvmfs</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/apptainer/tutorial.html">             <div>Step 6</div>             <div>apptainer</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/tool-management/tutorial.html">             <div>Step 7</div>             <div>tool-management</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/reference-genomes/tutorial.html">             <div>Step 8</div>             <div>reference-genomes</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/data-library/tutorial.html">             <div>Step 9</div>             <div>data-library</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/dev/tutorials/bioblend-api/tutorial.html">             <div>Step 10</div>             <div>dev/bioblend-api</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html">             <div>Step 11</div>             <div>connect-to-compute-cluster</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/job-destinations/tutorial.html">             <div>Step 12</div>             <div>job-destinations</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/pulsar/tutorial.html">             <div>Step 13</div>             <div>pulsar</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/celery/tutorial.html">             <div>Step 14</div>             <div>celery</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/gxadmin/tutorial.html">             <div>Step 15</div>             <div>gxadmin</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/reports/tutorial.html">             <div>Step 16</div>             <div>reports</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/monitoring/tutorial.html">             <div>Step 17</div>             <div>monitoring</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/tiaas/tutorial.html">             <div>Step 18</div>             <div>tiaas</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/sentry/tutorial.html">             <div>Step 19</div>             <div>sentry</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/ftp/tutorial.html">             <div>Step 20</div>             <div>ftp</div>         </a>     </li>               <span aria-hidden="true">         <i class="fas fa-arrow-right" aria-hidden="true"></i>     </span>                         <li class="disabled">         <a href="/training-material/topics/admin/tutorials/beacon/tutorial.html">             <div>Step 21</div>             <div>beacon</div>         </a>     </li>           </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I know what I can do with a role? What variables are available? <a href="/training-material/topics/admin/faqs/ansible_role-properties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-know-what-i-can-do-with-a-role-what-variables-are-available"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-know-what-i-can-do-with-a-role-what-variables-are-available" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I know what I can do with a role? What variables are available?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You don’t. There is no standard way for reporting this, but well written roles by trusted authors (e.g. geerlingguy, galaxyproject) do it properly and write all of the variables in the README file of the repository. We try to pick sensible roles for you in this course, but, in real life it may not be that simple.</p>   <p>So, definitely check there first, but if they aren’t there, then you’ll need to read through <code class="language-plaintext highlighter-rouge">defaults/</code> and <code class="language-plaintext highlighter-rouge">tasks/</code> and <code class="language-plaintext highlighter-rouge">templates/</code> to figure out what the role does and how you can control and modify it to accomplish your goals.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I see what variables are set for a host? <a href="/training-material/topics/admin/faqs/ansible_see_inventory.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-see-what-variables-are-set-for-a-host"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-see-what-variables-are-set-for-a-host" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I see what variables are set for a host?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you are using a simple <code class="language-plaintext highlighter-rouge">group_vars</code> file only, per group, and no other variable sources, then it’s relatively easy to tell what variables are getting set for your host! Just look at that one file.</p>   <p>But if you have graduated into using a more complex setup, perhaps with multiple sets of variables, like for example:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>├── group_vars<br>│   ├── all<br>│   │   ├── all.yml<br>│   │   └── secret.yml<br>│   ├── galaxyservers.yml<br>│   └── pulsarservers.yml<br>├── hosts<br>├── host_vars<br>│   ├── galaxy.example.org<br>│   │   ├── all.yml<br>│   │   └── secret.yml<br>│   ├── pulsar.example.org<br>│   │   ├── all.yml<br>│   │   ├── pulsar.yml<br>│   │   └── secret.yml<br>...<br></code></pre></div></div>   <p>Then it might be harder to figure out what variables are being set, in full. This is where <a href="https://docs.ansible.com/ansible/latest/cli/ansible-inventory.html"><code class="language-plaintext highlighter-rouge">ansible-inventory</code></a> command can be useful.</p>   <p><strong>Graph</strong> shows you the structure of your host groups:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ ansible-inventory --graph<br>@all:<br>  |--@cluster:<br>  |  |--allie.example.com<br>  |  |--bob.example.com<br>  |  |--charlie.example.com<br>[...]<br></code></pre></div></div>   <p>Here is a relatively simple, flat example, but this can be more complicated if you nest sub-groups of hosts:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@all:<br>  |--@local:<br>  |  |--localhost<br>  |--@ungrouped:<br>  |--@workshop_instances:<br>  |  |--@workshop_eu:<br>  |  |  |--gat-0.eu.training.galaxyproject.eu<br>  |  |  |--gat-1.eu.training.galaxyproject.eu<br>  |  |--@workshop_oz:<br>  |  |--@workshop_us:<br></code></pre></div></div>   <p><strong>List</strong> shows you all defined variables:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ ansible-inventory --host galaxy.example.com | head<br>[WARNING]: While constructing a mapping from<br>/group_vars/galaxyservers.yml, line 3, column<br>1, found a duplicate dict key (tiaas_templates_dir). Using last defined value<br>only.<br>{<br>    "ansible_connection": "local",<br>    "ansible_user": "ubuntu",<br>    "certbot_agree_tos": "--agree-tos",<br>    "certbot_auth_method": "--webroot",<br>    "certbot_auto_renew": true,<br>    "certbot_auto_renew_hour": "{{ 23 |random(seed=inventory_hostname)  }}",<br>    "certbot_auto_renew_minute": "{{ 59 |random(seed=inventory_hostname)  }}",<br><br></code></pre></div></div>   <p>And, helpfully, if variables are overridden in precedence you can see that as well with the above warnings.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is YAML sensitive to True/true/False/false <a href="/training-material/topics/admin/faqs/ansible_yaml.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-is-yaml-sensitive-to-true-true-false-false"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-is-yaml-sensitive-to-true-true-false-false" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Is YAML sensitive to True/true/False/false</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>By <a href="https://yaml.org/refcard.html">this reference</a>, YAML doesn’t really care:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{ Y, true, Yes, ON   }    : Boolean true<br>{ n, FALSE, No, off  }    : Boolean false<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Mapping Jobs to Specific Storage By User <a href="/training-material/topics/admin/faqs/mapping_by_user.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-mapping-jobs-to-specific-storage-by-user"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-mapping-jobs-to-specific-storage-by-user" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Mapping Jobs to Specific Storage By User</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>It is possible to map your jobs to use specific storage backends based on user! If you have e.g. specific user groups that need their data stored separately from other users, for whatever political reasons, then in your dynamic destination you can do something like:</p>   <div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">job_destination</span> <span class="o">=</span> <span class="n">app</span><span class="p">.</span><span class="n">job_config</span><span class="p">.</span><span class="nf">get_destination</span><span class="p">(</span><span class="n">destination_id</span><span class="p">)</span><br><span class="k">if</span> <span class="n">user</span> <span class="o">==</span> <span class="sh">"</span><span class="s">alice</span><span class="sh">"</span><span class="p">:</span><br>    <span class="n">job_destination</span><span class="p">.</span><span class="n">params</span><span class="p">[</span><span class="sh">'</span><span class="s">object_store_id</span><span class="sh">'</span><span class="p">]</span> <span class="o">=</span> <span class="sh">'</span><span class="s">foo</span><span class="sh">'</span> <span class="c1"># Maybe lookup the ID from a mapping somewhere<br></span></code></pre></div></div>   <p>If you manage to do this in production, please let us know and we can update this FAQ with any information you encounter.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Operating system compatibility <a href="/training-material/topics/admin/faqs/admin-testing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-operating-system-compatibility"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-operating-system-compatibility" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Operating system compatibility</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>These Ansible roles and training materials were last tested on Centos 7 and Ubuntu 18.04, but will probably work on other RHEL and Debian variants.</p>   <p>The roles that are used in these training are currently used by <code class="language-plaintext highlighter-rouge">usegalaxy.*</code>, and other, servers in maintaining their infrastructure. (<a href="https://github.com/galaxyproject/infrastructure-playbook/">US</a>, <a href="https://github.com/usegalaxy-eu/infrastructure-playbook">EU</a>, both are running CentOS 7)</p>   <p>If you have an issue running these trainings on your OS flavour, please report <a href="https://github.com/galaxyproject/training-material/issues/new">the issue</a> in the training material and we can see if it is possible to solve.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Running Ansible on your remote machine <a href="/training-material/topics/admin/faqs/ansible_local.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-running-ansible-on-your-remote-machine"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-ansible-on-your-remote-machine" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Running Ansible on your remote machine</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>It is possible to have ansible installed on the remote machine and run it there, not just from your local machine connecting to the remote machine.</p>   <p>Your hosts file will need to use <code class="language-plaintext highlighter-rouge">localhost</code>,  and whenever you run playbooks with <code class="language-plaintext highlighter-rouge">ansible-playbook -i hosts playbook.yml</code>, you will need to add <code class="language-plaintext highlighter-rouge">-c local</code> to your command.</p>   <p>Be <strong>certain</strong> that the playbook that you’re writing on the remote machine is stored somewhere safe, like your user home directory, or backed up on your local machine. The cloud can be unreliable and things can disappear at any time.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Updating from 22.01 to 23.0 with Ansible <a href="/training-material/topics/admin/faqs/galaxy-update-22.05.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-updating-from-22-01-to-23-0-with-ansible"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-updating-from-22-01-to-23-0-with-ansible" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Updating from 22.01 to 23.0 with Ansible</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy introduced a number of changes in 22.05 and 23.0 that are extremely important to be aware of during the upgrade process. Namely a new database migration system, and a new required running environment (gunicorn instead of uwsgi).</p>   <p>The scripts to migrate to the new database migration system are only compatible with release 22.05, and then were subsequently removed, so it is <strong>mandatory</strong> to upgrade to 22.05 if you want to go further.</p>   <p>Here is the recommended update procedure with ansible:</p>   <ol>   <li>Update to 22.01 normally</li>   <li>Change the release to 22.05, and run the upgrade     <ol>       <li>Galaxy will probably not start correctly here, ignore it (even if the build fail, this if fine, just ignore).</li>       <li>         <p>Run the database migration manually (with the galaxy user with the venv activated)</p>         <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>GALAXY_CONFIG_FILE=/srv/galaxy/config/galaxy.yml sh /srv/galaxy/server/manage_db.sh -c /srv/galaxy/config/galaxy.yml upgrade<br></code></pre></div>        </div>       </li>     </ol>   </li>   <li>Update your system’s ansible, you probably need something with a major version of at least 2.</li>   <li>     <p>Set the release to <code class="language-plaintext highlighter-rouge">23.0</code> and make other required changes. There are a lot of useful changes, but the easiest procedure is probably something like:</p>     <ol>       <li>git clone https://github.com/hexylena/git-gat/</li>       <li>cd git-gat</li>       <li>git checkout c2e7bf6d3584fbf3281fb57d8024a9189f957e0e (this corresponds to the version of the repo after the 23.0 integration without too much customization and after potential bug fixes)</li>       <li>         <p>Diff and sync (e.g. <code class="language-plaintext highlighter-rouge">vimdiff group_vars/galaxyservers.yml git-gat/group_vars/galaxyservers.yml</code>) for the main configuration files:</p>         <ul>           <li>group_vars/all.yml</li>           <li>group_vars/dbservers.yml</li>           <li>galaxy.yml</li>           <li>requirements.yml (and don’t forget to install the new role versions)</li>           <li>hosts</li>           <li>templates/nginx/galaxy.j2</li>         </ul>       </li>     </ol>     <p>But the main change is the swap from uwsgi to gravity+gunicorn</p>     <div class="language-diff highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gd">-  uwsgi:<br>-    socket: 127.0.0.1:8080<br>-    buffer-size: 16384<br>-    processes: 1<br>-    threads: 4<br>-    offload-threads: 2<br>-    static-map:<br>-      - /static=/static<br>-      - /favicon.ico=/static/favicon.ico<br>-    static-safe: client/galaxy/images<br>-    master: true<br>-    virtualenv: ""<br>-    pythonpath: "/lib"<br>-    module: galaxy.webapps.galaxy.buildapp:uwsgi_app()<br>-    thunder-lock: true<br>-    die-on-term: true<br>-    hook-master-start:<br>-      - unix_signal:2 gracefully_kill_them_all<br>-      - unix_signal:15 gracefully_kill_them_all<br>-    py-call-osafterfork: true<br>-    enable-threads: true<br>-    mule:<br>-      - lib/galaxy/main.py<br>-      - lib/galaxy/main.py<br>-    farm: job-handlers:1,2<br></span><span class="gi">+  gravity:<br>+    process_manager: systemd<br>+    galaxy_root: "/server"<br>+    galaxy_user: ""<br>+    virtualenv: ""<br>+    gunicorn:<br>+      # listening options<br>+      bind: "unix:/gunicorn.sock"<br>+      # performance options<br>+      workers: 2<br>+      # Other options that will be passed to gunicorn<br>+      # This permits setting of 'secure' headers like REMOTE_USER (and friends)<br>+      # https://docs.gunicorn.org/en/stable/settings.html#forwarded-allow-ips<br>+      extra_args: '--forwarded-allow-ips="*"'<br>+      # This lets Gunicorn start Galaxy completely before forking which is faster.<br>+      # https://docs.gunicorn.org/en/stable/settings.html#preload-app<br>+      preload: true<br>+    celery:<br>+      concurrency: 2<br>+      loglevel: DEBUG<br>+    handlers:<br>+      handler:<br>+        processes: 2<br>+        pools:<br>+          - job-handlers<br>+          - workflow-schedulers<br></span></code></pre></div>    </div>     <p>Some other important changes include:</p>     <ul>       <li>uchida.miniconda is replaced with galaxyproject.conda</li>       <li>usegalaxy_eu.systemd is no longer needed</li>       <li><code class="language-plaintext highlighter-rouge">galaxy_user_name</code> is defined in all.yml in the latest git-gat</li>       <li>the <code class="language-plaintext highlighter-rouge">galaxy_job_config</code> needs to have a database <code class="language-plaintext highlighter-rouge">handling</code> specified - assign set to <code class="language-plaintext highlighter-rouge">db-skip-locked</code></li>       <li>git-gat also separates out the DB serving into a <code class="language-plaintext highlighter-rouge">dbservers.yml</code> host group</li>     </ul>   </li>   <li>Backup your <code class="language-plaintext highlighter-rouge">venv</code>, <code class="language-plaintext highlighter-rouge">mv /srv/galaxy/venv/ /srv/galaxy/venv-old/</code>, as your NodeJS is probably out of date and Galaxy doesn’t handle that gracefully</li>   <li>Do any local customs for luck (knocking on wood, etc.)</li>   <li>Run the playbook</li>   <li>Things might go wrong with systemd units     <ul>       <li>try running <code class="language-plaintext highlighter-rouge">galaxyctl -c /srv/galaxy/config/galaxy.yml update</code> as root</li>       <li>you may also need to <code class="language-plaintext highlighter-rouge">rm /etc/systemd/system/galaxy.service</code> which is then no longer needed</li>       <li>you’ll have a <code class="language-plaintext highlighter-rouge">galaxy.target</code> and you can instead <code class="language-plaintext highlighter-rouge">systemctl daemon-reload</code> and <code class="language-plaintext highlighter-rouge">systemctl start galaxy.target</code></li>     </ul>   </li>   <li>You may need to restart galaxy manually with <code class="language-plaintext highlighter-rouge">sudo galaxyctl restart</code></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Variable connection <a href="/training-material/topics/admin/faqs/ansible-connection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-variable-connection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-variable-connection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Variable connection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When the playbook runs, as part of the setup, it collects any variables that are set. For a playbook affecting a group of hosts named <code class="language-plaintext highlighter-rouge">my_hosts</code>, it checks many different places for variables, including “group_vars/my_hosts.yml”. If there are variables there, they’re added to the collection of current variables. It also checks “group_vars/all.yml” (for the built-in host group <code class="language-plaintext highlighter-rouge">all</code>). There is a precedence order, but then these variables are available for roles and tasks to consume.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What if you forget `--diff`? <a href="/training-material/topics/admin/faqs/ansible_diff.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-if-you-forget-diff"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-if-you-forget-diff" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What if you forget `--diff`?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you forget to use <code class="language-plaintext highlighter-rouge">--diff</code>, it is not easy to see what has changed. Some modules like the <code class="language-plaintext highlighter-rouge">copy</code> and <code class="language-plaintext highlighter-rouge">template</code> modules have a <code class="language-plaintext highlighter-rouge">backup</code> option. If you set this option, then it will keep a backup copy next to the destination file.</p>   <p>However, most modules do not have such an option, so if you want to know what changes, always use <code class="language-plaintext highlighter-rouge">--diff</code>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is the difference between the roles with `role:` prefix and without? <a href="/training-material/topics/admin/faqs/ansible_role-prefix.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-is-the-difference-between-the-roles-with-role-prefix-and-without"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-is-the-difference-between-the-roles-with-role-prefix-and-without" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What is the difference between the roles with `role:` prefix and without?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The bare role name is just simplified syntax for the roles, you could equally specifiy <code class="language-plaintext highlighter-rouge">role: &lt;name&gt;</code> every time but it’s only necessary if you want to set additional variables like <code class="language-plaintext highlighter-rouge">become_user</code></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="ansible-galaxy"> <a href="#ansible-galaxy"> Ansible-galaxy </a></h3>

<hr/>



<h4 class="faq-area"> Customising the welcome page <a href="/training-material/topics/admin/tutorials/ansible-galaxy/faqs/custom-welcome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-customising-the-welcome-page"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-customising-the-welcome-page" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Customising the welcome page</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Customising the <code class="language-plaintext highlighter-rouge">welcome.html</code> page is very easy. Simply follow the <a href="/training-material/topics/admin/tutorials/customization/tutorial.html">Customising Galaxy Tutorial</a>!</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="collections"> <a href="#collections"> Collections </a></h3>

<hr/>



<h4 class="faq-area"> Adding a tag to a collection <a href="/training-material/faqs/galaxy/collections_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-to-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-to-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag to a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection in your history to view it</li>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name at the top of the history panel</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add a tag  starting with <code class="language-plaintext highlighter-rouge">#</code>     <ul>       <li>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs any tools using this dataset.</li>     </ul>   </li>   <li>Click <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span></li>   <li>Check that the tag appears below the collection name</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the datatype of a collection <a href="/training-material/faqs/galaxy/collections_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This will set the datatype for all files in your collection. Does not change the files themselves.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-of-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-of-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype of a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name in your history</li>   <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>Under <strong>new type</strong>, select  your desired datatype     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ol>   <p><strong>Cannot find the feature?</strong></p>   <p>If you are on a smaller Galaxy server, i.e. not one of the large (multi)national public servers, you may not be able to find this operation, and there is no indication it is missing or why it is disabled.</p>   <p>Galaxy has recently started putting <a href="https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks">more features behind a setting and deployment configuration</a> that needs to be enabled by the server administrator. Your administrator will need to deploy Celery and potentially additionally flower and redis to their stack to enable changing the datatype of a collection. Consider sending your Galaxy administrator the link to the <a href="/training-material/topics/admin/tutorials/celeryless/tutorial.html">simpler deployment option</a> or more complex <a href="/training-material/topics/admin/tutorials/celery/tutorial.html">GTN tutorial for setting up redis and flower</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Converting the datatype of a collection <a href="/training-material/faqs/galaxy/collections_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This will convert all files in your collection to a different format. This will change the files themselves and create a new collection.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-converting-the-datatype-of-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-converting-the-datatype-of-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Converting the datatype of a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name in your history</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>Under <strong>Converter Tool</strong>, select  your desired conversion</li>   <li>Click the <strong>Convert Collection</strong> button</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a dataset collection <a href="/training-material/faqs/galaxy/collections_build_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a dataset collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create collection</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a paired collection <a href="/training-material/faqs/galaxy/collections_build_list_paired.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-paired-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-paired-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a paired collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build List of Dataset Pairs</strong></p>   </li>   <li>Change the text of <em>unpaired forward</em> to a common selector for the forward reads</li>   <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse reads</li>   <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create List</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming a collection <a href="/training-material/faqs/galaxy/collections_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection</li>   <li>Click on the name of the collection at the top</li>   <li>Change the name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="collections, histories"> <a href="#collections, histories"> Collections, histories </a></h3>

<hr/>



<h4 class="faq-area"> Datasets versus collections <a href="/training-material/faqs/galaxy/histories_datasets_vs_collections.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explanation of why collections are needed and what they are
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-datasets-versus-collections"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-datasets-versus-collections" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Datasets versus collections</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Datasets versus collections</strong></p>   <p>In Galaxy’s history datasets can be present as individual entries or they can be combined into <em>Collections</em>. Why do we need collections? Collections combine multiple individual datasets into a single entity which is easy to manage. Galaxy tools can use collections directly as inputs. Collection can be <strong>simple</strong> or <strong>nested</strong>.</p>   <p><strong>Simple collections</strong></p>   <p>Imagine that you’ve uploaded a hundred FASTQ files corresponding to a hundred samples. These will appear as a hundred individual datasets in your history making it very long. But the chances are that when you analyze these data you will do the same thing on each dataset.</p>   <p>To simplify this process you can combine all hundred datasets into a single entity called a <em>dataset collection</em> (or simply a <em>collection</em> or a <em>list</em>). It will appear as a single box in your history making it much easier to understand. Galaxy tools are designed to take collections as inputs. So, for example, if you want to map each of these datasets against a reference genome using, say, <span><strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>, you will need to provide <code class="language-plaintext highlighter-rouge">minmap2</code> with just one input, the collection, and it will automatically start 100 jobs behind the scenes and will combine all outputs into a single collection containing BAM files.</p>   <p><img src="/training-material/shared/images/simple_collection.svg" alt="A simple collection is a container containing individual datasets" /></p>   <p>There is a number of situations when simple collections are not sufficient to reflect the complexity of the data. To deal with this situation Galaxy allows for <strong>nested</strong> collections.</p>   <p><strong>Nested collections</strong></p>   <p>Probably the most common example of this is pared end data when each sample is represented by two files: one containing forward reads and another containing reverse reads. In Galaxy you can create <strong>nested</strong> collection that reflects the hierarchy of the data. In the case of paired data Galaxy supports <strong>paired</strong> collections.</p>   <p><img src="/training-material/shared/images/paired_collection.svg" alt="A paired collection is a container containing individual datasets and preserving their hierarchy" /></p>   <!-- Original editable image for simple collections = https://docs.google.com/drawings/d/1A-tRerNLzC4FJfShUFT327wMvSSX4Y8AAdaxD_Fwaa0/edit?usp=sharing -->   <!-- Original editable image for paired collection = https://docs.google.com/drawings/d/1Bbx4UmIYdDAqK3KSm6LtLQ8zwEXxDglHmVbb0MK-mbQ/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="contributing"> <a href="#contributing"> Contributing </a></h3>

<hr/>



<h4 class="faq-area"> How to Contribute to Galaxy <a href="/training-material/topics/dev/faqs/contributing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Contributing to Galaxy is a multi-step process, this will guide you through it.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-contribute-to-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-contribute-to-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to Contribute to Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To contribute to galaxy, a GitHub account is required. Changes are proposed via a <a href="https://docs.github.com/en/github/collaborating-with-pull-requests">pull request</a>. This allows the project maintainers to review the changes and suggest improvements.</p>   <p>The general steps are as follows:</p>   <ol>   <li>Fork the Galaxy repository</li>   <li>Clone your fork</li>   <li>Make changes in a new branch</li>   <li>Commit your changes, push branch to your fork</li>   <li>Open a pull request for this branch in the upstream Galaxy repository</li> </ol>   <blockquote class="details">   <div class="box-title details-title" id="details-git-github-and-galaxy-core"><button class="gtn-boxify-button details" type="button" aria-controls="details-git-github-and-galaxy-core" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Git, Github, and Galaxy Core</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>For a lot more information about Git branching and managing a repository on Github,  see the <a href="/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html">Contributing with GitHub via command-line</a> tutorial.</p>   <p>The <a href="/training-material/topics/dev/tutorials/architecture/slides.html">Galaxy Core Architecture slides</a> have a lot of important Galaxy core-related information related to branches,  project management, and contributing to Galaxy - under the Project Management section of the slides.</p> </blockquote> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="contributors"> <a href="#contributors"> Contributors </a></h3>

<hr/>



<h4 class="faq-area"> Adding workflow tests with Planemo <a href="/training-material/faqs/gtn/gtn_workflow_testing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-workflow-tests-with-planemo"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-workflow-tests-with-planemo" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding workflow tests with Planemo</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <h3 data-toc-skip id="ensuring-a-tutorial-has-a-workflow">Ensuring a Tutorial has a Workflow</h3>   <ol>   <li>     <p>Find a tutorial that you’re interested in, that <strong>doesn’t currently have tests.</strong></p>     <p>This tutorial has a workflow (<code class="language-plaintext highlighter-rouge">.ga</code>) and a test, notice the <code class="language-plaintext highlighter-rouge">-test.yml</code> that has the same name as the workflow <code class="language-plaintext highlighter-rouge">.ga</code> file.</p>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>machinelearning/workflows/machine_learning.ga<br>machinelearning/workflows/machine_learning-test.yml<br></code></pre></div>    </div>     <p>You want to find tutorials without the <code class="language-plaintext highlighter-rouge">-test.yml</code> file. The workflow file might also be missing.</p>   </li>   <li>Check if it has a workflow (if it does, skip to step 5.)</li>   <li>Follow the tutorial</li>   <li>Extract a workflow from the history</li>   <li>Run that workflow in a new history to test</li> </ol>   <h3 data-toc-skip id="extract-tests-online-version">Extract Tests (Online Version)</h3>   <p>If you are on UseGalaxy.org or another server running 24.2 or later, you can use <a href="https://pwdk.apps.galaxyproject.eu/">PWDK</a>, a version of planemo running online to generate the workflow tests.</p>   <p>However if you are on an older version of Galaxy, or a private Galaxy server, then you’ll need to do the following:</p>   <h3 data-toc-skip id="extract-tests-manual-version">Extract Tests (Manual Version)</h3>   <ol>   <li>     <p>Obtain the workflow invocation ID, and your API key (User → Preferences → Manage API Key)</p>     <p><img src="/training-material/faqs/gtn/images/invocation.png" alt="screenshot of the workflow invocation page. The user drop down shows where to find this page, and a red box circles a field named &quot;Invocation ID&quot;" /></p>   </li>   <li>     <p>Install the latest version of <code class="language-plaintext highlighter-rouge">planemo</code></p>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># In a virtualenv<br>pip install planemo<br></code></pre></div>    </div>   </li>   <li>     <p>Run the command to initialise a workflow test from the <code class="language-plaintext highlighter-rouge">workflows/</code> subdirectory - if it doesn’t exist, you might need to create it first.</p>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>planemo workflow_test_init --from_invocation &lt;INVOCATION ID&gt; --galaxy_url &lt;GALAXY SERVER URL&gt; --galaxy_user_key &lt;GALAXY API KEY&gt;<br></code></pre></div>    </div>     <p>This will produce a folder of files, for example from a testing workflow:</p>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ tree<br>.<br>├── test-data<br>│   ├── input dataset(s).shapefile.shp<br>│   └── shapefile.shp<br>├── testing-openlayer.ga<br>└── testing-openlayer-tests.yml<br></code></pre></div>    </div>   </li> </ol>   <h3 data-toc-skip id="adding-your-tests-to-the-gtn">Adding Your Tests to the GTN</h3>   <ol>   <li>     <p>You will need to check the <code class="language-plaintext highlighter-rouge">-tests.yml</code> file, it has some automatically generated comparisons. Namely it tests that output data matches the test-data exactly, however, you might want to replace that with assertions that check for e.g. correct file size, or specific text content you expect to see.</p>   </li>   <li>     <p>If the files in test-data are already uploaded to Zenodo, to save disk space, you should delete them from the <code class="language-plaintext highlighter-rouge">test-data</code> dir and use their URL in the <code class="language-plaintext highlighter-rouge">-tests.yml</code> file, as in this example:</p>     <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="pi">-</span> <span class="na">doc</span><span class="pi">:</span> <span class="s">Test the M. Tuberculosis Variant Analysis workflow</span><br>  <span class="na">job</span><span class="pi">:</span><br>     <span class="s1">'</span><span class="s">Read</span><span class="nv"> </span><span class="s">1'</span><span class="err">:</span><br>        <span class="na">location</span><span class="pi">:</span> <span class="s">https://zenodo.org/record/3960260/files/004-2_1.fastq.gz</span><br>        <span class="na">class</span><span class="pi">:</span> <span class="s">File</span><br>        <span class="na">filetype</span><span class="pi">:</span> <span class="s">fastqsanger.gz</span><br></code></pre></div>    </div>   </li>   <li>     <p>Add tests on the outputs! Check the <a href="https://planemo.readthedocs.io/en/latest/test_format.html">planemo reference if you need more detail</a>.</p>     <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="pi">-</span> <span class="na">doc</span><span class="pi">:</span> <span class="s">Test the M. Tuberculosis Variant Analysis workflow</span><br>  <span class="na">job</span><span class="pi">:</span><br>     <span class="c1"># Simple explicit Inputs</span><br>     <span class="s1">'</span><span class="s">Read</span><span class="nv"> </span><span class="s">1'</span><span class="err">:</span><br>        <span class="na">location</span><span class="pi">:</span> <span class="s">https://zenodo.org/record/3960260/files/004-2_1.fastq.gz</span><br>        <span class="na">class</span><span class="pi">:</span> <span class="s">File</span><br>        <span class="na">filetype</span><span class="pi">:</span> <span class="s">fastqsanger.gz</span><br>  <span class="na">outputs</span><span class="pi">:</span><br>    <span class="na">jbrowse_html</span><span class="pi">:</span><br>      <span class="na">asserts</span><span class="pi">:</span><br>        <span class="na">has_text</span><span class="pi">:</span><br>          <span class="na">text</span><span class="pi">:</span> <span class="s2">"</span><span class="s">JBrowseDefaultMainPage"</span><br>    <span class="na">snippy_fasta</span><span class="pi">:</span><br>      <span class="na">asserts</span><span class="pi">:</span><br>        <span class="na">has_line</span><span class="pi">:</span><br>          <span class="na">line</span><span class="pi">:</span> <span class="s1">'</span><span class="s">&gt;Wildtype</span><span class="nv"> </span><span class="s">Staphylococcus</span><span class="nv"> </span><span class="s">aureus</span><span class="nv"> </span><span class="s">strain</span><span class="nv"> </span><span class="s">WT.'</span><br>    <span class="na">snippy_tabular</span><span class="pi">:</span><br>      <span class="na">asserts</span><span class="pi">:</span><br>        <span class="na">has_n_columns</span><span class="pi">:</span><br>          <span class="na">n</span><span class="pi">:</span> <span class="m">2</span><br></code></pre></div>    </div>   </li>   <li>     <p>Contribute all of those files to the GTN in a PR, adding them to the <code class="language-plaintext highlighter-rouge">workflows/</code> folder of your tutorial.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Adding your recording to a tutorial or slide deck <a href="/training-material/faqs/gtn/recordings_add.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-your-recording-to-a-tutorial-or-slide-deck"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-your-recording-to-a-tutorial-or-slide-deck" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding your recording to a tutorial or slide deck</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>We welcome anybody to submit their recordings!</strong> Your videos can be used in (online) training events, or for self-study by learners on the GTN.</p>   <p>For some tips and tricks about recording the video itself, please ensure your recording conforms to our recommendations:</p>   <p><a href="/training-material/faqs/gtn/recordings_create.html" class="btn btn-info">Recording Tips &amp; Tricks</a> <a href="https://forms.gle/qNG8FkTN1yRZPNZY6" class="btn btn-info">Submit a Recording</a></p>   <h4 id="submission-process">Submission process</h4>   <p>The process of adding recordings to the GTN is as follows:</p>   <ol>   <li><strong>Instructor:</strong> Record video (<a href="/training-material/faqs/gtn/recordings_create.html">tips &amp; tricks</a>)</li>   <li><strong>Instructor:</strong> Submit your video using this <a href="https://forms.gle/qNG8FkTN1yRZPNZY6">Google Form</a></li>   <li><strong>GTN:</strong> A <a href="https://github.com/galaxyproject/training-material/pulls">GTN GitHub pull request (PR)</a> will be made by our bot based on the form.</li>   <li><strong>GTN:</strong>: We will upload your video to the <a href="https://www.youtube.com/c/galaxyproject">GalaxyProject YouTube channel</a></li>   <li><strong>GTN:</strong>: We will put the auto-generated captions from YouTube into a Google Doc</li>   <li><strong>Instructor:</strong>: Check and fix the auto-generated captions</li>   <li><strong>GTN:</strong> Upload the fixed captions to YouTube</li>   <li><strong>GTN:</strong> Merge the Pull Request on GitHub</li>   <li>Done! Your recording will now show up on the tutorial for anybody to use and re-use</li> </ol>   <p><strong>Note:</strong> If you are submitting a video to use in an event, please submit your recording <strong>2 weeks before the start of your course</strong> to allow ample time to complete the submission process.</p>   <h4 id="recordings-metadata">Recordings Metadata</h4>   <p>Our bot will add some metadata about your recording to the tutorial or slide deck in question, and looks as follows:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>recordings:<br>  - speakers:       # speakers must be defined in the CONTRIBUTORS.yaml file<br>    - shiltemann<br>    - hexylena<br>    captioners:     # captioners must also be present in the CONTRIBUTORS.yaml file<br>    - bebatut<br>    type:           # optional, will default to Tutorial or Lecture, but if you do something different, set it here (e.g. Demo, Lecture &amp; Tutorial, Background, Webinar)<br>    date: '2024-06-12'         # date on which you recorded the video<br>    galaxy_version: '24.0'     # version of Galaxy you used during the recording, can be found under 'Help-&gt;About' in Galaxy<br>    length: 1H17M              # length of your video, in format: 17M or 2H34M  etc<br>    youtube_id: "dQw4w9WgXcQ"  # the bit of the YouTube URL after youtube.com/watch?v=<br><br>  - speakers:<br>    - shiltemann<br>    captioners:<br>    - hexylena<br>    - bebatut<br>    date: '2020-06-12'<br>    galaxy_version: '20.05'<br>    length: 51M<br>    youtube_id: "oAVjF_7ensg"<br><br></code></pre></div></div>   <h4 id="misc">Misc</h4>   <p><strong>Note:</strong> If your videos are already uploaded to YouTube, for example as part of a different project’s account, you can add this metadata to the tutorial or slides manually, without using our submission form. Note that we do require all videos to have good-quality English captions, and we will not be able to help you configure these on other YouTube accounts.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can the FAIR-by-Design Methodology be used for FAIR development of other types of resources? <a href="/training-material/topics/contributing/tutorials/fair-by-design/faqs/FAQ-C2.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-the-fair-by-design-methodology-be-used-for-fair-development-of-other-types-of-resources"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-the-fair-by-design-methodology-be-used-for-fair-development-of-other-types-of-resources" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can the FAIR-by-Design Methodology be used for FAIR development of other types of resources?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The FAIR-by-Design Methodology stages can be relatively easily adapted to the processes for designing other FAIR objects.</p>   <p>For an example, the FAIR-by-Design Methodology can be adapted to create FAIR-by-Design software objects. This has been demonstrated on the IDCC24 W6 - FAIR-by-Design: introducing Skills4EOSC and FAIR-IMPACT workshop. The information regarding this example is available at <a href="https://fair-by-design-methodology.github.io/IDCC24workshop/latest/">https://fair-by-design-methodology.github.io/IDCC24workshop/latest/</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can this tutorial be adapted to other instructional development platforms? <a href="/training-material/topics/contributing/tutorials/fair-by-design/faqs/FAQ-C1.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-this-tutorial-be-adapted-to-other-instructional-development-platforms"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-this-tutorial-be-adapted-to-other-instructional-development-platforms" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can this tutorial be adapted to other instructional development platforms?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The tutorial has been specifically adapted to the rules and options available in GTN. However, the general FAIR-by-Design Methodology is platform agnostic and can be applied to any environment.</p>   <p>Thus, the tutorial can be reused and carefully adapted to be applicable to other platforms, as long as the adaptation is based on the originally published FAIR-by-Design Methodology.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a GTN Event <a href="/training-material/faqs/gtn/gtn_event_create.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-gtn-event"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-gtn-event" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a GTN Event</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To add your event to the GTN, you will need to supply your course information (dates, location, program, etc). You will then get <a href="/training-material/events/2024-04-01-example-event.html">an event page like this</a> which you can use during your training. This page includes a course overview, course handbook (full program with links to tutorials) and setup instructions for participants.</p>   <p>Your event will also be shown on the <a href="/training-material/events/index.html">GTN event horizon</a> and on the homepage. We are also happy to advertise your event on social media and Matrix channels.</p>   <p><strong>Already have your own event page?</strong> No problem! You can add your event as and external event (see below) and we will simply link to your page!</p>   <p>To add your event to the GTN:</p>   <ol>   <li>Create a page in the <code class="language-plaintext highlighter-rouge">events/</code> folder of the <a href="https://github.com/galaxyproject/training-material">GTN repository</a></li>   <li>Have a look at example event definitions in this folder:     <ul>       <li><a href="https://github.com/galaxyproject/training-material/blob/main/events/2024-04-01-example-event.md">2024-04-01-example-event.md</a></li>       <li>or <a href="https://github.com/galaxyproject/training-material/blob/main/events/2024-04-01-example-event-external.md">2024-04-01-example-event-external.md</a> if you already have an event page elsewhere</li>     </ul>   </li>   <li>Adapt one of these example pages to fit your event</li>   <li>Create a pull request on the GTN</li> </ol>   <p><strong>We are also happy to help you</strong> to add your event, please <a href="https://matrix.to/#/#Galaxy-Training-Network_Lobby:gitter.im">contact us on Matrix</a> to discuss the details of your course with us.</p>   <p>For a full list of metadata fields for events, please have a look at our <a href="https://github.com/galaxyproject/training-material/blob/main/_layouts/event.html">schema documentation page</a></p>   <p>Please also feel free to contact us with ideas for improvements! We know that training comes in many different forms, so if something in your event is not yet supported, let us know and we are happy to add it!</p>   <h3 data-toc-skip id="external-events">External events</h3>   <p>Already have a course webpage? Great! In this case, you only have to provide the most basic information about your course (title, desciption, dates, location).</p>   <p>The easiest method is to fill in our Google Form:</p>   <p><a href="https://forms.gle/4KjCKKrZ6kamg81o7" class="btn btn-success">Events Google Form!</a></p>   <p>Or you can create the event file manually. See also <a href="https://github.com/galaxyproject/training-material/blob/main/events/2024-04-01-example-event-external.md">2024-04-01-example-event-external.md</a> for an example definition.</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>---<br>layout: event-external<br>title: My External Training Event Title<br><br>external: "https://galaxyproject.org/events/"<br>description:<br><br>date_start:<br>date_end:  # optional, for multi-day events<br><br>location:<br>  name:<br>  city:<br>  country:<br><br>contributions:<br>  organisers:<br>    - name1<br>    - name2<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a GTN FAQ <a href="/training-material/faqs/gtn/gtn_faq_create.html"><i class="fas fa-external-link-alt"></i></a> </h4>
If you have a snippet of knowledge that is reusable, we recommend you to share with the GTN community, and we encourage you to create an FAQ for it!
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-gtn-faq"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-gtn-faq" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a GTN FAQ</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you have a snippet of knowledge that is reusable, we recommend you to share with the GTN community, and we encourage you to create an FAQ for it!</p>   <p><strong>Creating the FAQ: The Easy Way</strong></p>   <p><a href="https://forms.gle/2JVMfd1AgtenZPvv9">Fill out this Google Form</a>. Every day our bot will import the FAQs submitted via this Google Form, and we will process them, perhaps requesting small changes, so we recommend that you have a GitHub account already.</p>   <p><strong>For Advanced Users</strong></p>   <p>Have a look at the existing FAQs in the <a href="https://github.com/galaxyproject/training-material/tree/main/faqs/galaxy"><code class="language-plaintext highlighter-rouge">faqs/galaxy/</code> folder</a> of the GTN repository for some examples.</p>   <p>A news post is a markdown file that looks as follows:</p>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nn">---</span><br><span class="na">title</span><span class="pi">:</span> <span class="s">Finding Datasets</span><br><span class="na">area</span><span class="pi">:</span> <span class="s">datasets</span><br><span class="na">box_type</span><span class="pi">:</span> <span class="s">tip</span><br><span class="na">layout</span><span class="pi">:</span> <span class="s">faq</span><br><span class="na">contributors</span><span class="pi">:</span> <span class="pi">[</span><span class="nv">jennaj</span><span class="pi">,</span> <span class="nv">Melkeb</span><span class="pi">]</span><br><span class="nn">---</span><br><br><br><span class="p">-</span> To review all active Datasets in your account, go to <span class="gs">**User &gt; Datasets**</span>.<br><br>Notes:<br><span class="p">-</span> Logging out of Galaxy while the Upload tool is still loading data can cause uploads to abort. This is most likely to occur when a dataset is loaded by browsing local files.<br><span class="p">-</span> If you have more than one browser window open, each with a different Galaxy History loaded, the Upload tool will load data into the most recently used history.<br><span class="p">-</span> Click on refresh icon {% icon galaxy-refresh %} at the top of the History panel to display the current active History with the datasets.<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a GTN News post <a href="/training-material/faqs/gtn/gtn_news_create_post.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-gtn-news-post"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-gtn-news-post" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a GTN News post</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you have created a new tutorial, running an event, published a paper around training, or have anything else interesting to share with the GTN community, we encourage you to write a <strong>News item</strong> about it!</p>   <p>News items will show up on the <a href="/training-material/index.html">GTN homepage</a> and in the <a href="/training-material/news.html">GTN news feed</a>.</p>   <p><strong>Creating the news post: The Easy Way</strong></p>   <p><a href="https://forms.gle/6SRAVcppzieef7Zz9">Fill out this Google Form</a>. Every day our bot will import the news posts submitted via this Google Form, and we will process them, perhaps requesting small changes, so we recommend that you have a GitHub account already.</p>   <p><strong>For Advanced Users</strong></p>   <p>Have a look at the existing news items in the <a href="https://github.com/galaxyproject/training-material/tree/main/news/_posts"><code class="language-plaintext highlighter-rouge">news/_posts/</code> folder</a> of the GTN repository for some examples.</p>   <p>A news post is a markdown file that looks as follows:</p>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nn">---</span><br><span class="na">layout</span><span class="pi">:</span> <span class="s">news</span><br><br><span class="na">title</span><span class="pi">:</span> <span class="s2">"</span><span class="s">New</span><span class="nv"> </span><span class="s">Tutorial:</span><span class="nv"> </span><span class="s">My</span><span class="nv"> </span><span class="s">tutorial</span><span class="nv"> </span><span class="s">title"</span><br><span class="na">tags</span><span class="pi">:</span><br>  <span class="pi">-</span> <span class="s">new tutorial</span><br>  <span class="pi">-</span> <span class="s">transcriptomics</span><br><span class="na">contributors</span><span class="pi">:</span><br>  <span class="pi">-</span> <span class="s">shiltemann</span><br>  <span class="pi">-</span> <span class="s">hexylena</span><br><br><span class="na">tutorial</span><span class="pi">:</span> <span class="s2">"</span><span class="s">topics/introduction/tutorials/data-manipulation-olympics/tutorial.html"</span><br><span class="na">cover</span><span class="pi">:</span> <span class="s2">"</span><span class="s">path/to/cover-image.jpg"</span>  <span class="c1"># usually an image from your tutorial</span><br><span class="na">coveralt</span><span class="pi">:</span> <span class="s2">"</span><span class="s">description</span><span class="nv"> </span><span class="s">of</span><span class="nv"> </span><span class="s">the</span><span class="nv"> </span><span class="s">cover</span><span class="nv"> </span><span class="s">image"</span><br><br><span class="nn">---</span><br><br>A bit of text containing your news, this is all markdown formatted,<br>so you can do <span class="gs">**bold**</span> and <span class="ge">*italic*</span> text like this, and links look<br>like <span class="p">[</span><span class="nv">this</span><span class="p">](</span><span class="sx">https://example.com</span><span class="p">)</span> etc.<br><br>Describe everything you want to convey here, can be as long as you<br>need.<br></code></pre></div></div>   <p>Make sure the filename is structured as follows: <code class="language-plaintext highlighter-rouge">year-month-day-title.md</code>, so for example: <code class="language-plaintext highlighter-rouge">2022-10-28-my-new-tutorial.md</code></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I contribute in "advanced" mode? <a href="/training-material/faqs/gtn/contributors_git_advanced.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-contribute-in-quot-advanced-quot-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-contribute-in-quot-advanced-quot-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I contribute in &quot;advanced&quot; mode?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Most of the content is written in <a href="https://guides.github.com/features/mastering-markdown/">GitHub Flavored Markdown</a> with some metadata (or variables) found in <a href="http://yaml.org/">YAML</a> files. Everything is stored on our <a href="https://github.com/galaxyproject/training-material">GitHub repository</a>. Each training material is related to a topic. All training materials (slides, tutorials, etc) related to a topic are found in a dedicated directory (<em>e.g.</em> <code class="language-plaintext highlighter-rouge">transcriptomics</code> directory contains the material related to transcriptomic analysis). Each topic has the following structure:</p>   <p><img src="/training-material/shared/images/repo_organization.png" alt="Structure of the repository" width="400px" /></p>   <ul>   <li>a metadata file in YAML format</li>   <li>a directory with the topic introduction slide deck in Markdown with introductions to the topic</li>   <li>     <p>a directory with the tutorials:</p>     <p>Inside the tutorials directory, each tutorial related to the topic has its own subdirectory with several files:</p>     <ul>       <li>a tutorial file written in Markdown with hands-on</li>       <li>an optional slides file in Markdown with slides to support the tutorial</li>       <li>a directory with Galaxy Interactive Tours to reproduce the tutorial</li>       <li>a directory with workflows extracted from the tutorial</li>       <li>a YAML file with the links to the input data needed for the tutorial</li>       <li>a YAML file with the description of needed tools to run the tutorial</li>     </ul>   </li>   <li>a directory with the Dockerfile describing the details to build a container for the topic (self-study environments).</li> </ul>   <p>To manage changes, we use <a href="https://guides.github.com/introduction/flow/">GitHub flow</a> based on Pull Requests (check our <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">tutorial</a>):</p>   <ol>   <li><a href="https://help.github.com/articles/fork-a-repo/">Create a fork</a> of this repository on GitHub</li>   <li>Clone your fork of this repository to create a local copy on your computer and initialize the required submodules (<code class="language-plaintext highlighter-rouge">git submodule init</code> and <code class="language-plaintext highlighter-rouge">git submodule update</code>)</li>   <li>Create a new branch in your local copy for each significant change</li>   <li>Commit the changes in that branch</li>   <li>Push that branch to your fork on GitHub</li>   <li>Submit a pull request from that branch to the <a href="https://github.com/galaxyproject/training-material">original repository</a></li>   <li>If you receive feedback, make changes in your local clone and push them to your branch on GitHub: the pull request will update automatically</li>   <li>Pull requests will be merged by the training team members after at least one other person has reviewed the Pull request and approved it.</li> </ol>   <p>Globally, the process of development of new content is open and transparent:</p>   <ol>   <li>Creation of a branch derived from the main branch of the GitHub repository</li>   <li>Initialization of a new directory for the tutorial</li>   <li>Filling of the metadata with title, questions, learning objectives, etc</li>   <li>Generation of the input dataset for the tutorial</li>   <li>Filling of the tutorial content</li>   <li>Extraction of the workflows of the tutorial</li>   <li>Automatic extraction of the required tools to populate the tool file</li>   <li>Automatic annotation of the public Galaxy servers</li>   <li>Generation of an interactive tour for the tutorial with the <a href="https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/">Tourbuilder</a> web-browser extension</li>   <li>Upload of the datasets to Zenodo and addition of the links in the data library file.</li>   <li>Once ready, opening a Pull Request</li>   <li>Automatic checks of the changes are automatically checked for the right format and working links using continuous integration testing on Travis CI</li>   <li>Review of the content by several other instructors via discussions</li>   <li>After the review process, merge of the content into the main branch, starting a series of automatic steps triggered by Travis CI</li>   <li>Regeneration of the website and publication on <a href="/training-material/">https://training.galaxyproject.org/training-material/</a></li>   <li>Generation of PDF artifacts of the tutorials and slides and upload on the FTP server</li>   <li>Population of <a href="https://tess.elixir-europe.org/">TeSS</a>, the ELIXIR’s Training Portal, via the metadata</li> </ol>   <p><img src="/training-material/shared/images/development_process.png" alt="Development process" /></p>   <p>To learn how to add new content, check out our <a href="/training-material/topics/contributing/index.html">series of tutorials on creating new content</a>:</p>   <ol>         <li><a href="/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html">Contributing with GitHub via command-line</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/introduction/slides.html">Overview of the Galaxy Training Material</a></li>                  <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-tours/tutorial.html">Creating Interactive Galaxy Tours</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/github-interface-contribution/tutorial.html">Contributing with GitHub via its interface</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/generating-pdf/tutorial.html">Generating PDF artefacts of the website</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html">Tools, Data, and Workflows for tutorials</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/python/tutorial.html">Teaching Python</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/running-codespaces/tutorial.html">Running the GTN website online using GitHub CodeSpaces</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/design/tutorial.html">Design and plan session, course, materials</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/schemas/tutorial.html">GTN Metadata</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/gitpod/tutorial.html">Running the GTN website online using GitPod</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/fair-by-design/tutorial.html">FAIR-by-Design methodology</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/slides-with-video/tutorial.html">Adding auto-generated video to your slides</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/create-new-topic/tutorial.html">Including a new topic</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-quiz/tutorial.html">Adding Quizzes to your Tutorial</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/learning-principles/tutorial.html">Principles of learning and how they apply to training and teaching</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/admin-knitting/tutorial.html">Updating diffs in admin training</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/running-jekyll/tutorial.html">Running the GTN website locally using the command line</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">Creating a new tutorial</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/meta-analysis-plot/tutorial.html">Single Cell Publication - Data Plotting</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/fair-gtn/tutorial.html">FAIR Galaxy Training Material</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/meta-analysis-data/tutorial.html">Single Cell Publication - Data Analysis</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html">Creating content in Markdown</a></li>                 <li><a href="/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html">Creating Slides</a></li>                  <li><a href="/training-material/topics/contributing/tutorials/updating_tutorial/tutorial.html">Updating tool versions in a tutorial</a></li>          </ol>   <p>We also strongly recommend you read and follow <a href="https://carpentries.org/">The Carpentries</a> recommendations on <a href="https://carpentries.github.io/lesson-example/01-design/">lesson design</a> and <a href="https://carpentries.github.io/instructor-training/15-lesson-study/">lesson writing</a> if you plan to add or change some training materials, and also to check the structure of the training material above.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I create new content without dealing with git? <a href="/training-material/faqs/gtn/contributors_without_git.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-create-new-content-without-dealing-with-git"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-create-new-content-without-dealing-with-git" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I create new content without dealing with git?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you feel uncomfortable with using the git and the <a href="https://guides.github.com/introduction/flow/">GitHub flow</a>, you can write a new tutorial with any text editor and then contact us (via <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a> or <a href="mailto:galaxytrainingnetwork@gmail.com">email</a>). We will work together to integrate the new content.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I fix mistakes or expand an existing tutorial using the GitHub interface? <a href="/training-material/faqs/gtn/contributors_github_interface.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-fix-mistakes-or-expand-an-existing-tutorial-using-the-github-interface" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I fix mistakes or expand an existing tutorial using the GitHub interface?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Check our tutorial to learn how to use the GitHub interface (soon…)</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I get started with contributing? <a href="/training-material/faqs/gtn/contributors_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-get-started-with-contributing"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-get-started-with-contributing" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I get started with contributing?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you would like to get involved in the project but are unsure where to start, there are some easy ways to contribute which will also help you familiarize yourself with the project!</p>   <p>A great way to help out the project is to test/edit existing tutorials. Pick a tutorial and check the contents. Does everything work as expected? Are there things that could be improved?</p>   <p>Below is a checklist of things to look out for to help you get started. If you feel confident in making changes yourself, please open a pull request, otherwise please file an issue with any problems you run into or suggestions for improvements.</p>   <p>Basic:</p>   <ul>   <li><strong>Test</strong> the tutorial on a running Galaxy instance     <ul>       <li>For example <a href="https://usegalaxy.org.au">UseGalaxy.org.au</a>, <a href="https://usegalaxy.org">UseGalaxy.org</a>, <a href="https://usegalaxy.fr">UseGalaxy.fr</a>, <a href="https://usegalaxy.eu">UseGalaxy.eu</a></li>       <li>Report any issues you run into</li>     </ul>   </li>   <li><strong>Language</strong> editing     <ul>       <li>Fix spelling and grammar mistakes</li>       <li>Simplify the English (to make it more accessible)</li>     </ul>   </li> </ul>   <p>Intermediate:</p>   <ul>   <li><strong>Metadata</strong>     <ul>       <li>Are the objectives, keypoints and time estimate filled in?</li>       <li>Do they fit with the contents of the tutorial?</li>     </ul>   </li>   <li><strong>Content</strong>     <ul>       <li>Is there enough background information provided in the introduction section and throughout the tutorial?</li>       <li><strong>Question boxes</strong>         <ul>           <li>Add questions or question boxes where you think they might be useful (make people think about results they got, test their understanding, etc)</li>           <li>Check that answers are still up-to-date</li>         </ul>       </li>       <li><strong>Screenshots and Videos</strong>         <ul>           <li>Make sure there is also a textual description of the image/video contents</li>           <li>Does the screenshot add value to the tutorial or can it be removed?</li>         </ul>       </li>     </ul>   </li> </ul>   <p>Advanced:</p>   <ul>   <li><strong>Workflows</strong>     <ul>       <li>Add a workflow definition file <code class="language-plaintext highlighter-rouge">.ga</code> if none is present</li>       <li>Check that the existing workflow is up-to-date with the tutorial contents</li>       <li>Enable <a href="https://github.com/usegalaxy-eu/workflow-testing">workflow testing</a></li>     </ul>   </li>   <li><strong>Tours</strong>     <ul>       <li>Add a tour if none exists</li>       <li>Run the existing tour and check that it is up-to-date with the tutorial contents</li>     </ul>   </li>   <li><strong>Datasets</strong>     <ul>       <li>Check that all datasets used in the tutorial are present in Zenodo</li>       <li>Add a data-library.yaml file if none exists</li>     </ul>   </li> </ul>   <p>Another great way to help out the project is by reviewing <a href="https://github.com/galaxyproject/training-material/pulls?q=is%3Apr+is%3Aopen+sort%3Aupdated-desc">open pull requests</a>. You can use the above checklist as a guide for your review. Some documentation about how to add your review in the GitHub interface can be found <a href="https://help.github.com/articles/about-pull-request-reviews/">in GitHub’s PR Reviewing Documentation</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I give feedback? <a href="/training-material/faqs/gtn/contributors_giving_feedback.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-give-feedback"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-give-feedback" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I give feedback?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>At the end of each tutorial, there is a link to <a href="https://docs.google.com/forms/d/1BILuBp8bPv42odpZ55Kcf2rqzR-k0Czx46agVVSRgm4/edit">a feedback form</a>. We use this information to improve our tutorials.</p>   <p>For global feedbacks, you can open an <a href="https://github.com/galaxyproject/training-material/issues/new">issue on GitHub</a>, write us on <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a> or send us an <a href="mailto:galaxytrainingnetwork@gmail.com">email</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I report mistakes or errors? <a href="/training-material/faqs/gtn/contributors_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-report-mistakes-or-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-report-mistakes-or-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I report mistakes or errors?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The easiest way to start contributing is to <a href="https://github.com/galaxyproject/training-material/issues/new">file an issue</a> to tell us about a problem such as a typo, spelling mistake, or a factual error. You can then introduce yourself and meet some of our community members.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I test an Interactive Tour? <a href="/training-material/faqs/gtn/contributors_tours.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-test-an-interactive-tour"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-test-an-interactive-tour" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I test an Interactive Tour?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Perhaps you’ve been asked to review an interactive tour, or maybe you just want to try one out. The easiest way to run an interactive tour is to use the <a href="https://tailordev.fr/blog/2017/07/19/the-galaxy-tour-builder-extension/">Tour builder</a> browser extension.</p>   <ol>   <li>Install the Tour Builder extension to your browser (<a href="https://chrome.google.com/webstore/detail/galaxy-tour-builder/mdfbapknmcpnbmggahhaegehbbbmhmgg">Chrome Web Store</a>, <a href="https://addons.mozilla.org/en-US/firefox/addon/galaxy-tour-builder/">Firefox add-on</a>).</li>   <li>Navigate to a Galaxy instance supporting the tutorial. To find which Galaxy instances support each tutorial, please see the dropdown menu next to the tutorial on the training website. Using one of the usegalaxy.* instances (<a href="https://usegalaxy.org.au">UseGalaxy.org.au</a>, <a href="https://usegalaxy.fr">UseGalaxy.fr</a>, <a href="https://usegalaxy.org">UseGalaxy.org</a>, <a href="https://usegalaxy.eu">UseGalaxy.eu</a>) ) is usually a good bet.</li>   <li>Start the Tour Builder plugin by clicking on the icon in your browser menu bar</li>   <li>Copy the contents of the <code class="language-plaintext highlighter-rouge">tour.yaml</code> file into the Tour builder editor window</li>   <li>Click <code class="language-plaintext highlighter-rouge">Save</code> and then <code class="language-plaintext highlighter-rouge">Run</code></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How does the GTN ensure accessibility? <a href="/training-material/faqs/gtn/a11y.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><p>We are committed to an accessible training experience regardless of disability. Please see our <a href="/training-material/accessibility.html">accessibility page</a> for more information.</p> <!--END_SNIPPET-->




<h4 class="faq-area"> How does the GTN ensure our training materials are FAIR? <a href="/training-material/faqs/gtn/fair_training.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-does-the-gtn-ensure-our-training-materials-are-fair"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-does-the-gtn-ensure-our-training-materials-are-fair" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How does the GTN ensure our training materials are FAIR?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This infrastructure has been developed in accordance with the FAIR (Findable, Accessible, Interoperable, Reusable) principles for training materials <span class="citation"><a href="#Garcia2020">Garcia <i>et al.</i> 2020</a></span>. Following these principles enables trainers and trainees to find, reuse, adapt, and improve the available tutorials.</p>   <p>The GTN receives a 100% score on the <a href="https://fair-checker.france-bioinformatique.fr/check">FAIR Checker</a>, as noted in our recent <a href="/training-material/news/2024/05/13/fair.html">news post</a></p>   <table>   <thead>     <tr>       <th>10 Simple Rules</th>       <th>Implementation in GTN framework</th>     </tr>   </thead>   <tbody>     <tr>       <td>Plan to share your training materials online</td>       <td>Online <a href="https://training.galaxyproject.org/">training material portfolio</a>, managed via <a href="https://github.com/galaxyproject/training-material">a public GitHub repository</a></td>     </tr>     <tr>       <td>Improve findability of your training materials by properly describing them</td>       <td>Rich metadata associated with each tutorial that are visible and accessible via schema.org on each tutorial webpage.</td>     </tr>     <tr>       <td>Give your training materials a unique identity</td>       <td>URL persistency with redirection in case of renaming of tutorials. Data used for tutorials stored on Zenodo and associated with a Digital Object Identifiers (DOI)</td>     </tr>     <tr>       <td>Register your training materials online</td>       <td>Tutorials automatically registered on TeSS, the <a href="https://tess.elixir-europe.org/">ELIXIR’s Training e-Support System</a></td>     </tr>     <tr>       <td>If appropriate, define access rules for your training materials</td>       <td>Online and free to use without registration</td>     </tr>     <tr>       <td>Use an interoperable format for your training materials</td>       <td>Content of the tutorials and slides written in Markdown. Metadata associated with tutorials stored in YAML, and workflows in JSON. All of this metadata is available <a href="/training-material/api/index.html">from the GTN’s API</a></td>     </tr>     <tr>       <td>Make your training materials (re-)usable for trainers</td>       <td>Online. Rich metadata associated with each tutorial: title, contributor details, license, description, learning outcomes, audience, requirements, tags/keywords, duration, date of last revision. Strong technical support for each tutorial: workflow, data on Zenodo and also available as data libraries on UseGalaxy.*, tools installable via the Galaxy Tool Shed, list of possible Galaxy instances with the needed tools.</td>     </tr>     <tr>       <td>Make your training materials (re-)usable for trainees</td>       <td>Online and easy to follow hands-on tutorials. Rich metadata with “Specific, Measurable, Attainable, Realistic and Time bound” (SMART) learning outcomes following Bloom’s taxonomy. Requirements and follow-up tutorials to build learning path. List of Galaxy instances offering needed tools, data on Zenodo and also available as data libraries on UseGalaxy.*. Support chat embedded in tutorial pages.</td>     </tr>     <tr>       <td>Make your training materials contribution friendly and citable</td>       <td>Open and collaborative infrastructure with contribution guidelines, a CONTRIBUTING file and a chat. Details to cite tutorials and give credit to contributors available at the end of each tutorial.</td>     </tr>     <tr>       <td>Keep your training materials up-to-date</td>       <td>Open, collaborative and transparent peer-review and curation process. Short time between updates.</td>     </tr>   </tbody> </table> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How does the GTN implement the "Ten simple rules for collaborative lesson development" <a href="/training-material/faqs/gtn/collaborative_development.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-does-the-gtn-implement-the-quot-ten-simple-rules-for-collaborative-lesson-development-quot"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-does-the-gtn-implement-the-quot-ten-simple-rules-for-collaborative-lesson-development-quot" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How does the GTN implement the &quot;Ten simple rules for collaborative lesson development&quot;</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The GTN framework is inherently collaborative and community-driven, and comprises a growing number of contributors with expertise in a wide range of scientific and technical domains. Given this highly collaborative nature of a community with very different skill sets, the GTN framework has evolved over the years to facilitate the contribution and maintenance of the tutorials. We aim to adhere to best-practice guidelines for collaborative lesson development described in <span class="citation"><a href="#Devenyi_2018">Devenyi <i>et al.</i> 2018</a></span>. The structure of the tutorials and repository has been made modular with unified syntax and use of snippets enabling easy access for authors to add common tips and tricks new users might need to know. This system allows for easy updating of all tutorials, if there is a change in tools or interface. More generally, we continually strive to lower contribution barriers for content creators by providing a framework that is easy to use for training developers regardless of their level of knowledge of the underlying technical framework.</p>   <p>Implementation of the “Ten simple rules for collaborative lesson development” (<span class="citation"><a href="#Devenyi_2018">Devenyi <i>et al.</i> 2018</a></span>) in the training material:</p>   <table>   <thead>     <tr>       <th>Rules</th>       <th>Implementation in the GTN framework</th>     </tr>   </thead>   <tbody>     <tr>       <td>Clarify audience</td>       <td>Tutorial metadata includes level indicators (introductory, intermediate, advanced) and a list of prerequisite tutorials as recommended prior knowledge. This information is rendered at the top of each tutorial.</td>     </tr>     <tr>       <td>Make lessons modular</td>       <td>Development of small tutorials linked together via learning paths</td>     </tr>     <tr>       <td>Teach best practice lesson development</td>       <td>We maintain the topic <a href="/training-material/topics/contributing/index.html"><em>Contributing to the Galaxy Training Material</em></a> including numerous tutorials describing how to create new content. Furthermore, quarterly online collaboration fest (CoFests) are organized, where contributors can get direct support. Development of a Train the Trainer program and a mentoring program for instructors, in which lesson development is taught</td>     </tr>     <tr>       <td>Encourage and empower contributors</td>       <td>Involve them in reviews. Mentor them. Encourage them to become maintainers.</td>     </tr>     <tr>       <td>Build community around lessons</td>       <td>Quarterly online collaboration fest (CoFests) and Community calls. Chat on our Gitter/Matrix channel.</td>     </tr>     <tr>       <td>Publish periodically and recognize contributions</td>       <td>Author listed on tutorials. Hall of fame listing all contributors. Full tutorial citation at the end of the tutorial. Tweet about new or updated tutorials. List of new or updated tutorials in Galaxy Community newsletter. Soon: publication of tutorials via article</td>     </tr>     <tr>       <td>Evaluate lessons at several scales</td>       <td>Tutorial change (Pull Request) review. Embedded feedback form in tutorials for trainee feedback. Instructor feedback. Automatic workflow testing</td>     </tr>     <tr>       <td>Reduce, re-use, recycle</td>       <td>Sharing content between tutorials, specially using snippets. Development of small modular tutorials linked by learning paths</td>     </tr>     <tr>       <td>Link to other resources</td>       <td>Links to original paper, documentation, external tutorials and other material</td>     </tr>     <tr>       <td>You can’t please everyone</td>       <td>but we can try (several different Galaxy introduction tutorials for different audience). Aim to clearly state what the tutorial does and does not cover, at the start.</td>     </tr>   </tbody> </table> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Recording a video tutorial <a href="/training-material/faqs/gtn/recordings_create.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-recording-a-video-tutorial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-recording-a-video-tutorial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Recording a video tutorial</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This FAQ describes some <strong>general guidelines</strong> for recording your video</p>   <p><strong>Anybody is welcome to record one of the GTN tutorials</strong>, even if another recording already exists! Both the GTN tutorial and Galaxy itself change significantly over time, and having regular and/or multiple recordings of tutorials is great!</p>   <p><strong>Done with your recording?</strong> Check out the instructions for adding it to the GTN:</p>   <p><a href="/training-material/faqs/gtn/recordings_add.html" class="btn btn-info">Submitting Recordings to the GTN</a></p>   <h4 id="video-content">Video content</h4>   <ol>   <li><strong>Start of video</strong>     <ul>       <li><strong>Introduce yourself</strong></li>       <li>Discuss the <strong>questions and learning objectives</strong> of the tutorial</li>       <li>Give a <strong>basic introducion about the topic</strong>, many participants will be novices</li>     </ul>   </li>   <li><strong>Guide the learners through the tutorial step by step</strong>     <ul>       <li>Explain the scientific background of the analysis</li>       <li>Explain where you are clicking in Galaxy</li>       <li>Explain what tool parameters mean</li>       <li>Explain what the tool does</li>       <li>Discuss the output files</li>       <li>Discuss how to interpret the results</li>       <li>Discuss question boxes from the tutorial</li>     </ul>   </li>   <li><strong>Speak slowly and clearly</strong>     <ul>       <li>Take your time, we are not in a hurry</li>       <li>It is often a lot of new information for participants, give them a chance to process all of it</li>       <li>Speaking slowly and clearly will improve the quality of the auto-generated captions, and will be less work for you to fix captions.</li>     </ul>   </li>   <li><strong>If things go wrong that is OK!</strong>     <ul>       <li>It’s a great teaching moment!</li>       <li>Explain the steps you are taking to determine what went wrong, and how you are fixing it.</li>       <li>It makes participants feel less bad if things go wrong for them</li>     </ul>   </li>   <li><strong>If your tutorial is long</strong>     <ul>       <li>Indicate good places for people to take a break</li>       <li>e.g. when a tool takes a while to run</li>     </ul>   </li>   <li><strong>End of video</strong>     <ul>       <li>Go over some of the <strong>take-home messages (key-points)</strong> of the tutorial</li>       <li>Remind viewers about the <strong>feedback form</strong> embedded at the end of the tutorial</li>       <li>Share your recommendations for <strong>follow-up tutorials</strong></li>       <li>Share any other tips for where to learn more about the topic</li>       <li>Share <strong>how to connect with the community</strong> (e.g. Matrix, Help Forum, social media, etc)</li>     </ul>   </li>   <li>If you are doing both a lecture and a hands-on training, please create 2 separate videos</li> </ol>   <h4 id="technical-guidelines">Technical Guidelines</h4>   <ol>   <li>Start a <a href="https://zoom.us/">Zoom</a> call with yourself, record that.     <ul>       <li>For Mac users, QuickTime Player is also a nice option.</li>       <li>Have another preference like OBS? Totally OK too!</li>       <li>We recommend zoom to folks new to video production as it is the easiest to get started and produces quite small file sizes.</li>     </ul>   </li>   <li>Do a short <strong>test recording</strong> first     <ul>       <li>Is the <strong>audio quality</strong> good enough?         <ul>           <li>Wearing a headset often improves the audio quality.</li>         </ul>       </li>       <li><strong>Screen sharing:</strong> is your screen readable?         <ul>           <li>Make sure you <strong>zoom in</strong> enough for it to be clearly visible what you are doing in Galaxy.</li>           <li>Test watching the video in a non-maximised window. Is it still legible?</li>           <li>If the participant is using 50% of their screen for the video, 50% for Galaxy, will it be legible?</li>         </ul>       </li>     </ul>   </li>   <li>Need to edit your video after recording?     <ul>       <li>For example to merge multiple videos together?</li>       <li>Software like <a href="https://kdenlive.org/en/">KDEnlive</a> can help here.</li>       <li>Feel free to ask us for help if you need!</li>     </ul>   </li> </ol>   <h4 id="standards">Standards</h4>   <ol>   <li>     <p><strong>Zoom in</strong>, in every interface you’re covering! Many people will be watching the video while they’re doing the activity, and won’t have significant monitor space. Which video below would you rather be trying to follow?</p>     <table>       <thead>         <tr>           <th>Bad</th>           <th>Good 😍</th>         </tr>       </thead>       <tbody>         <tr>           <td><img src="/training-material/faqs/gtn/images/bad.png" alt="default size screenshot of usegalaxy.eu" /></td>           <td><img src="/training-material/faqs/gtn/images/good.png" alt="zoomed in screenshot of usegalaxy.eu, now much more legible" /></td>         </tr>       </tbody>     </table>     <table>       <thead>         <tr>           <th>Bad</th>           <th>Good 🤩</th>         </tr>       </thead>       <tbody>         <tr>           <td><img src="/training-material/faqs/gtn/images/bad-console.png" alt="green text on black background console with tiny font" /></td>           <td><img src="/training-material/faqs/gtn/images/good-console.png" alt="zoomed in screenshot of a console with high contrast black and white content" /></td>         </tr>       </tbody>     </table>   </li>   <li>     <p>(Especially for introductory videos!) Clearly call out what you’re doing, especially on the first occurrence</p>     <table>       <thead>         <tr>           <th>Bad</th>           <th>Good</th>         </tr>       </thead>       <tbody>         <tr>           <td>“Re-run the job”</td>           <td>“We need to re-run the job which we can do by first clicking to expand the dataset, and then using the re-run job button which looks like a refresh icon.”</td>         </tr>       </tbody>     </table>     <table>       <thead>         <tr>           <th>Bad</th>           <th>Good</th>         </tr>       </thead>       <tbody>         <tr>           <td>“As you can see here the report says X”</td>           <td>“I’m going to view the output of this tool, click on the eyeball icon, and as you can see the report says X.”</td>         </tr>       </tbody>     </table>     <p>But the same goes for terminal aliases, please disable all of your favourite terminal aliases and quick shortcuts that you’re used to using, disable your bashrc, etc. These are all things students will try and type, and will fail in doing so. We need to be very clear and explicit because people will type exactly what is on the screen, and their environment should at minimum match yours.</p>     <table>       <thead>         <tr>           <th>Bad</th>           <th>Good</th>         </tr>       </thead>       <tbody>         <tr>           <td><code class="language-plaintext highlighter-rouge">lg file</code></td>           <td><code class="language-plaintext highlighter-rouge">ls -al | grep file</code></td>         </tr>         <tr>           <td><code class="language-plaintext highlighter-rouge">z galaxy</code></td>           <td><code class="language-plaintext highlighter-rouge">cd path/to/the/galaxy</code></td>         </tr>       </tbody>     </table>   </li>   <li>     <p>Consider using a pointer that is more visually highlighted.</p>     <p><img src="/training-material/faqs/gtn/images/mouse.png" alt="mouse pointer with circle around it that follows it around" /></p>     <p>There are themes available for your mouse pointer that you can temporarily use while recording that can make it easier for watchers to see what you’re doing.</p>     <ul>       <li><a href="https://www.microsoft.com/en-us/p/mouse-pointer-highlight/9p7sb9s4rq7z?activetab=pivot:overviewtab">Windows</a></li>       <li><a href="https://askubuntu.com/questions/777896/how-do-i-highlight-my-mouse-pointer-while-screen-recording/917587#917587">Linux</a></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Supporting Tutorial Mode (GTN-in-Galaxy) in a tutorial <a href="/training-material/faqs/gtn/gtn_tutorial_mode.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-supporting-tutorial-mode-gtn-in-galaxy-in-a-tutorial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-supporting-tutorial-mode-gtn-in-galaxy-in-a-tutorial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Supporting Tutorial Mode (GTN-in-Galaxy) in a tutorial</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>GTN tutorials can be viewed directly within Galaxy, we call this <em>Tutorial mode</em> (<a href="/training-material/faqs/galaxy/tutorial_mode.html">read news post</a>)</p>   <p>In this mode, tool names in hands-on boxes become clickable, directly opening the tool in Galaxy, at the right version.</p>   <p>To enable this feature, a bit of metadata needs to be added to the tool names in hands-on boxes as follows:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{% tool [Name](Toolshed ID) %}<br></code></pre></div></div>   <p>For example:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>{% tool [bedtools intersect intervals](toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.30.0+galaxy1) %}<br></code></pre></div></div>   <p>To find the toolshed ID of a tool:</p>   <ol>   <li>Open the tool in Galaxy</li>   <li>Click on the <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">dropdown</span> options menu (dropdown icon)</li>   <li>Select <strong>Copy Tool ID</strong></li> </ol>   <p><img src="/training-material/faqs/gtn/images/tool_toolshed_link.png" alt="screenshot of menu with toolshed link for a tool" /></p>   <p>Example of a hands-on box using this feature:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt; &lt;hands-on-title&gt; Counting SNPs &lt;/hands-on-title&gt;<br>&gt;<br>&gt; 1. {% tool [Datamash](toolshed.g2.bx.psu.edu/repos/iuc/datamash_ops/datamash_ops/1.8+galaxy0) %} (operations on tabular data):<br>&gt;<br>&gt;    - *"Input tabular dataset"*: select the output dataset from **bedtools intersect intervals** {% icon tool %}<br>&gt;    - *"Group by fields"*: `Column: 4` (the column with the exon IDs)<br>&gt;<br>{: .hands_on}<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Thanks! <a href="/training-material/faqs/gtn/contributors_appreciated.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-thanks"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-thanks" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Thanks!</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>First off, thanks for your interest in contributing to the Galaxy training materials!</p>   <p>Individual learners and instructors can make these training more effective by contributing back to them. You can report mistakes and errors, create more content, etc. Whatever is your background, there is a way to contribute: via the GitHub website, via command-line or even without dealing with GitHub.</p>   <p>We will address your issues and/or assess your change proposal as promptly as we can, and help you become a member of our community. You can also check our <a href="/training-material/topics/contributing/index.html">tutorials</a> for more details.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What can I do to help the project? <a href="/training-material/faqs/gtn/contributors_helping_project.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-can-i-do-to-help-the-project"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-can-i-do-to-help-the-project" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What can I do to help the project?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In <a href="https://github.com/galaxyproject/training-material/issues">issues</a>, you will find lists of issues to fix and features to implement (with the <a href="https://github.com/galaxyproject/training-material/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Anewcomer-friendly">“newcomer-friendly”</a> label for example). Feel free to work on them!</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="data upload"> <a href="#data upload"> Data upload </a></h3>

<hr/>



<h4 class="faq-area"> Data retrieval with “NCBI SRA Tools” (fastq-dump) <a href="/training-material/faqs/galaxy/dataupload_NCBI_SRA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-data-retrieval-with-ncbi-sra-tools-fastq-dump"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-data-retrieval-with-ncbi-sra-tools-fastq-dump" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Data retrieval with “NCBI SRA Tools” (fastq-dump)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This section will guide you through downloading experimental metadata, organizing the metadata to short lists corresponding to conditions and replicates, and finally importing the data from NCBI SRA in collections reflecting the experimental design.</p>   <p><strong>Downloading metadata</strong></p>   <ul>   <li>It is critical to understand the condition/replicate structure of an experiment before working with the data so that it can be imported as collections ready for analysis. Direct your browser to <a href="https://www.ncbi.nlm.nih.gov/Traces/study/">SRA Run Selector</a> and in the search box enter GEO data set identifier (for example: GSE72018). Once the study appears, click the box to download the “RunInfo Table”.</li> </ul>   <p><strong>Organizing metadata</strong></p>   <ul>   <li>The “RunInfo Table” provides the experimental condition and replicate structure of all of the samples. Prior to importing the data, we need to parse this file into individual files that contain the sample IDs of the replicates in each condition. This can be achieved by using a combination of the ‘group’, ‘compare two datasets’, ‘filter’, and ‘cut’ tools <strong>to end up with single column lists of sample IDs (SRRxxxxx) corresponding to each condition.</strong></li> </ul>   <p><strong>Importing data</strong></p>   <ul>   <li>Provide the files with SRR IDs to NCBI SRA Tools (fastq-dump) to import the data from SRA to Galaxy. By organizing the replicates of each condition in separate lists, <strong>the data will be imported as “collections” that can be directly loaded to a workflow or analysis pipeline.</strong></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Directly obtaining UCSC sourced *genome* identifiers <a href="/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-directly-obtaining-ucsc-sourced-genome-identifiers"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-directly-obtaining-ucsc-sourced-genome-identifiers" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Directly obtaining UCSC sourced *genome* identifiers</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Option 1</strong></p>   <ol>   <li>Go to <a href="http://genome.ucsc.edu/">UCSC Genome Browser</a>, navigate to “genomes”, then the species of interest.</li>   <li>On the home page for the genome build, immediately under the top navigation box, in the blue bar next to the full genome build name, you will find <strong>View sequences</strong> button.</li>   <li>Click on the <strong>View sequences</strong> button and it will take you to a detail page with a table listing out the contents.</li> </ol>   <p><strong>Option 2</strong></p>   <ol>   <li>Use the tool <strong>Get Data -&gt; UCSC Main</strong>.</li>   <li>In the Table Browser, choose the target genome and build.</li>   <li>For “group” choose the last option “All Tables”.</li>   <li>For “table” choose “chromInfo”.</li>   <li>Leave all other options at default and send the output to Galaxy.</li>   <li>This new dataset will load as a tabular dataset into your history.</li>   <li>It will list out the contents of the genome build, including the chromosome identifiers (in the first column).</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I  upload data using EBI-SRA? <a href="/training-material/faqs/galaxy/dataupload_EBI-SRA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-upload-data-using-ebi-sra"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-upload-data-using-ebi-sra" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I  upload data using EBI-SRA?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Search for your data directly in the tool and use the Galaxy links.</li>   <li>Be sure to check your sequence data for correct quality score formats and the metadata “datatype” assignment.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importación por medio de enlaces <a href="/training-material/faqs/galaxy-es/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importación-por-medio-de-enlaces"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importación-por-medio-de-enlaces" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importación por medio de enlaces</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copia los enlaces</li>   <li>     <p>Abre el manejador de carga de datos de Galaxy (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> en la parte superior derecha del panel de herramientas)</p>   </li>   <li>Selecciona ‘Pegar/Traer datos’ <strong>Paste/Fetch Data</strong></li>   <li>     <p>Copia los enlaces en el campo de textos</p>   </li>   <li>     <p>Presiona ‘Iniciar’ <strong>Start</strong></p>   </li>   <li>     <p><strong>Close</strong> Cierra la ventana.</p>   </li>   <li>Galaxy utiliza los URLs como nombres de forma predeterminada , así que los tendrás que cambiar a algunos que sean más útiles o informativos. the window</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importer via un lien <a href="/training-material/faqs/galaxy-fr/datasets_importa_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importer-via-un-lien"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importer-via-un-lien" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importer via un lien</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copier le lien</li>   <li>     <p>Ouvrez le gestionnaire de téléchargement Galaxy (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> en haut à droite du panneau d’outils)</p>   </li>   <li>Selectionnez <strong>Coller/Récupérer les données</strong></li>   <li>     <p>Collez le lien dans le champ de texte</p>   </li>   <li>     <p>Appuyez sur Start**</p>   </li>   <li>     <p>Ferme la fenêtre</p>   </li>   <li>Galaxy utilise les URL comme noms par défaut, vous devrez donc les remplacer par des URL plus utiles ou informatives. the window</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing data from Sierra LIMS <a href="/training-material/faqs/galaxy/importing-data-from-sierra-lims.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-sierra-lims"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-sierra-lims" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from Sierra LIMS</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This section will guide you through generating external links to your data stored in the Sierra LIMS system to be downloaded directly into Galaxy.</p>   <ol>   <li>Go to the <a href="https://www.bioinformatics.babraham.ac.uk/sierra/sierra.pl">Sierra portal</a> and login to your account.</li>   <li>Click on the <strong>Sample ID</strong> of the sample you want to download data from.</li>   <li>Click on the <strong>Edit Sample Details</strong> button.</li>   <li>At the bottom of the page there will be an input box for creating a link, enter a description for the link in the <strong>Reason for link</strong> section, and click <strong>Create link</strong>. This will reload the page and add a new link to the sample under <strong>Authorised links to this sample</strong>.</li>   <li>Go back to the sample page or click on the hyperlink called <strong>link</strong> to take you back.</li>   <li>In the <strong>Results</strong> section select the lane you want to access your data from.</li>   <li>The bottom of the page, under the <strong>Links</strong> section, will now contain a list of <code class="language-plaintext highlighter-rouge">wget</code> commands with links for accessing all the files within that sample/lane.</li>   <li>Since this list is for <code class="language-plaintext highlighter-rouge">wget</code> commands, you need to extract out the links from the command. You can copy the link in the first set of double quotes for each line and <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> them directly into Galaxy to download the files.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing data from a data library <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing data from remote files <a href="/training-material/faqs/galaxy/datasets_import_from_remote_files.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-remote-files"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-remote-files" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from remote files</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>Choose remote files</em>:</p>   <ol>   <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>   <li>     <p>Click on <strong>Choose remote files</strong> and scroll down to find your data folder or type the folder name in the <code class="language-plaintext highlighter-rouge">search box</code> on the top.</p>   </li>   <li>click on <strong>OK</strong></li>   <li>Click on <strong>Start</strong></li>   <li>Click on <strong>Close</strong></li>   <li>You can find the dataset has begun loading in you history.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing via links <a href="/training-material/faqs/galaxy/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> NCBI SRA sourced fastq data <a href="/training-material/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-ncbi-sra-sourced-fastq-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-ncbi-sra-sourced-fastq-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: NCBI SRA sourced fastq data</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In these FASTQ data:</p>   <ul>   <li>The quality score identifier (+) is sometimes not a match for the sequence identifier (@).</li>   <li>The forward and reverse reads may be interlaced and need to be separated into distinct datasets.</li>   <li>Both may be present in a dataset. Correct the first, then the second, as explained below.</li>   <li>Format problems of any kind can cause tool failures and/or unexpected results.</li>   <li>Fix the problems before running any other tools (including <strong>FastQC</strong>, <strong>Fastq Groomer</strong>, or other QA tools)</li> </ul>   <p><strong>For inconsistent sequence (@) and quality (+) identifiers</strong></p>   <ul>   <li>     <p>Correct the format by running the tool <strong>Replace Text in entire line</strong> with these options:</p>     <ul>       <li>Find pattern: <code class="language-plaintext highlighter-rouge">^\+SRR.+</code></li>       <li>Replace with: <code class="language-plaintext highlighter-rouge">+</code></li>     </ul>   </li> </ul>   <p>Note: If the quality score line is named like “+ERR” instead (or other valid options), modify the pattern search to match.</p>   <p><strong>For interlaced forward and reverse reads</strong></p>   <p><strong>Solution 1 (reads named /1 and /2)</strong></p>   <ul>   <li>Use the tool <strong>FASTQ de-interlacer on paired end reads</strong></li> </ul>   <p><strong>Solution 2 (reads named /1 and /2)</strong></p>   <ul>   <li>Create distinct datasets from an interlaced fastq dataset by running the tool <strong>Manipulate FASTQ reads on various attributes</strong> on the original dataset. It will run twice.</li> </ul>   <p>Note: The solution does NOT use the <strong>FASTQ Splitter</strong> tool. The data to be manipulated are interlaced sequences. This is different in format from data that are joined into a single sequence.</p>   <ul>   <li>     <p>Use the Manipulate FASTQ settings to produce a dataset that contains the <code class="language-plaintext highlighter-rouge">/1</code> reads**</p>     <p><em>Match Reads</em></p>     <ul>       <li>Match Reads by <code class="language-plaintext highlighter-rouge">Name/Identifier</code></li>       <li>Identifier Match Type <code class="language-plaintext highlighter-rouge">Regular Expression</code></li>       <li>Match by <code class="language-plaintext highlighter-rouge">.+/2</code></li>     </ul>     <p><em>Manipulate Reads</em></p>     <ul>       <li>Manipulate Reads by <code class="language-plaintext highlighter-rouge">Miscellaneous Actions</code></li>       <li>Miscellaneous Manipulation Type <code class="language-plaintext highlighter-rouge">Remove Read</code></li>     </ul>   </li>   <li>     <p>Use these Manipulate FASTQ settings to produce a dataset that contains the <code class="language-plaintext highlighter-rouge">/2</code> reads**</p>     <ul>       <li>Exact same settings as above except for this change: Match by <code class="language-plaintext highlighter-rouge">.+/1</code></li>     </ul>   </li> </ul>   <p><strong>Solution 3 (reads named /1 and /3)</strong></p>   <ul>   <li>Use the same operations as in Solution 2 above, except change the first <strong>Manipulate FASTQ</strong> query term to be:</li>   <li>Match by <code class="language-plaintext highlighter-rouge">.+/3</code></li> </ul>   <p><strong>Solution 4 (reads named without /N)</strong></p>   <ul>   <li>If your data has differently formatted sequence identifiers, the “Match by” expression from <em>Solution 2</em> above can be modified to suit your identifiers.</li> </ul>   <p>Alternative identifiers such as:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@M00946:180:000000000-ANFB2:1:1107:14919:14410 1:N:0:1<br>@M00946:180:000000000-ANFB2:1:1107:14919:14410 2:N:0:1<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload datasets from GenomeArk <a href="/training-material/faqs/galaxy/datasets_upload_from_genomeark.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-datasets-from-genomeark"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-datasets-from-genomeark" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload datasets from GenomeArk</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>   <li>Click on <strong>Choose remote files</strong> tab</li>   <li>Click on the <strong>Genome Ark</strong> button and then click on <strong>species</strong></li> </ol>   <p>You can find the data by following this path: <code class="language-plaintext highlighter-rouge">/species/${Genus}_${species}/${specimen_code}/genomic_data</code>. Inside a given datatype directory (<em>e.g.</em> <code class="language-plaintext highlighter-rouge">pacbio</code>), select all the relevant files individually until all the desired files are highlighted and click the <kbd>Ok</kbd> button. Note that there may be multiple pages of files listed. Also note that you may not want every file listed.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload fasta datasets via links <a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/faqs/dataset_upload_fasta_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-fasta-datasets-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-fasta-datasets-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload fasta datasets via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Uploading <code class="language-plaintext highlighter-rouge">fasta</code> or <code class="language-plaintext highlighter-rouge">fasta.gz</code> datasets via URL.</p>   <p><img src="/training-material/faqs/galaxy/images/upload_fasta_via_url.png" alt="UploadAnimatedPng" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload fasta datasets via links <a href="/training-material/topics/assembly/tutorials/vgp_workflow_training/faqs/dataset_upload_fasta_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-fasta-datasets-via-links-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-fasta-datasets-via-links-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload fasta datasets via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Uploading <code class="language-plaintext highlighter-rouge">fasta</code> or <code class="language-plaintext highlighter-rouge">fasta.gz</code> datasets via URL.</p>   <p><img src="/training-material/faqs/galaxy/images/upload_fasta_via_url.png" alt="UploadAnimatedPng" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload fastqsanger datasets via links <a href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/faqs/dataset_upload_fastqsanger_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-fastqsanger-datasets-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-fastqsanger-datasets-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload fastqsanger datasets via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Uploading <code class="language-plaintext highlighter-rouge">fastqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> datasets via URL.</p>   <ol>   <li>     <p>Click on <strong>Upload Data</strong> on the top of the left panel:</p>     <p><img src="/training-material/faqs/galaxy/images/upload_data_button.png" alt="UploadDataButton" /></p>   </li>   <li>     <p>Click on <strong>Paste/Fetch</strong>:</p>     <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_button.png" alt="PasteFetchButton" /></p>   </li>   <li>     <p>Paste URL into text box that would appear:</p>     <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_modal.png" alt="PasteFetchModal" /></p>   </li>   <li>     <p>Set <strong>Type (set all)</strong> to <code class="language-plaintext highlighter-rouge">fastqsanger</code> or, if your data is compressed as in URLs above (they have <code class="language-plaintext highlighter-rouge">.gz</code> extensions), to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>     <p><img src="/training-material/faqs/galaxy/images/paste_fetch_set_data_type.png" alt="ChangeTypeDropDown" />:</p>   </li> </ol>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-danger-make-sure-you-choose-corect-format"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Danger: Make sure you choose corect format!</div>   <p>When selecting datatype in “<strong>Type (set all)</strong>” dropdown, make sure you select <code class="language-plaintext highlighter-rouge">fastaqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> BUT NOT <code class="language-plaintext highlighter-rouge">fastqcssanger</code> or anything else!</p> </blockquote>   <p><img src="/training-material/faqs/galaxy/images/upload_fastqsanger_via_url.png" alt="UploadAnimatedPng" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload fastqsanger datasets via links <a href="/training-material/topics/assembly/tutorials/vgp_workflow_training/faqs/dataset_upload_fastqsanger_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-fastqsanger-datasets-via-links-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-fastqsanger-datasets-via-links-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload fastqsanger datasets via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Uploading <code class="language-plaintext highlighter-rouge">fastqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> datasets via URL.</p>   <ol>   <li>     <p>Click on <strong>Upload Data</strong> on the top of the left panel:</p>     <p><img src="/training-material/faqs/galaxy/images/upload_data_button.png" alt="UploadDataButton" /></p>   </li>   <li>     <p>Click on <strong>Paste/Fetch</strong>:</p>     <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_button.png" alt="PasteFetchButton" /></p>   </li>   <li>     <p>Paste URL into text box that would appear:</p>     <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_modal.png" alt="PasteFetchModal" /></p>   </li>   <li>     <p>Set <strong>Type (set all)</strong> to <code class="language-plaintext highlighter-rouge">fastqsanger</code> or, if your data is compressed as in URLs above (they have <code class="language-plaintext highlighter-rouge">.gz</code> extensions), to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>     <p><img src="/training-material/faqs/galaxy/images/paste_fetch_set_data_type.png" alt="ChangeTypeDropDown" />:</p>   </li> </ol>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-danger-make-sure-you-choose-corect-format-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Danger: Make sure you choose corect format!</div>   <p>When selecting datatype in “<strong>Type (set all)</strong>” dropdown, make sure you select <code class="language-plaintext highlighter-rouge">fastaqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> BUT NOT <code class="language-plaintext highlighter-rouge">fastqcssanger</code> or anything else!</p> </blockquote>   <p><img src="/training-material/faqs/galaxy/images/upload_fastqsanger_via_url.png" alt="UploadAnimatedPng" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload few files (1-10) <a href="/training-material/faqs/galaxy/datasets_upload.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-few-files-1-10"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-few-files-1-10" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload few files (1-10)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>   <li>Click on <strong>Choose local file</strong> and select the files or drop the files in the <strong>Drop files here</strong> part</li>   <li>Click on <strong>Start</strong></li>   <li>Click on <strong>Close</strong></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload many files (>10) via FTP <a href="/training-material/faqs/galaxy/datasets_upload_ftp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-many-files-10-via-ftp"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-many-files-10-via-ftp" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload many files (&gt;10) via FTP</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some Galaxies offer FTP upload for very large datasets.</p>   <p><strong>Note:</strong> the <em>“Big Three”</em> Galaxies (Galaxy Main, Galaxy EU, and Galaxy Australia) no longer support FTP upload, due to the recent improvements of the default web upload, which should now support large file uploads and almost all use cases. For situations where uploading via the web interface is too tedious, the <a href="https://github.com/galaxyproject/galaxy-upload">galaxy-upload commandline utility</a> is also available as an alternative to FTP.</p>   <p>To upload files via FTP, please</p>   <ol>   <li>     <p>Check that your Galaxy supports FTP upload and look up the FTP settings.</p>   </li>   <li>     <p>Make sure to have an FTP client installed</p>     <p>There are many options. We can recommend <a href="https://filezilla-project.org/">FileZilla</a>, a free FTP client that is available on Windows, MacOS, and Linux.</p>   </li>   <li>Establish FTP connection to the Galaxy server     <ol>       <li>Provide the Galaxy server’s FTP server name (e.g. <code class="language-plaintext highlighter-rouge">ftp.mygalaxy.com</code>)</li>       <li>Provide the <strong>username</strong> (usually the e-mail address) and the <strong>password</strong> on the Galaxy server</li>       <li>Connect</li>     </ol>   </li>   <li>     <p>Add the files to the FTP server by dragging/dropping them or right clicking on them and uploading them</p>     <p>The FTP transfer will start. We need to wait until they are done.</p>   </li>   <li>Open the Upload menu on the Galaxy server</li>   <li>Click on <strong>Choose FTP file</strong> on the bottom</li>   <li>Select files to import into the history</li>   <li>Click on <strong>Start</strong></li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="data-libraries"> <a href="#data-libraries"> Data-libraries </a></h3>

<hr/>



<h4 class="faq-area"> Library Permission Issues <a href="/training-material/topics/admin/faqs/library-permissions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-library-permission-issues"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-library-permission-issues" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Library Permission Issues</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When running <code class="language-plaintext highlighter-rouge">setup-data-libraries</code> it imports the library with the permissions of the admin user, rather locked down to the account that handled the importing.</p>   <p>Due to how data libraries have been implemented, it isn’t sufficient to share the folder with another user, instead you must also share individual items within this folder. This is an unfortunate issue with Galaxy that we hope to fix someday.</p>   <p>Until then, we can recommend you install the latest version of Ephemeris which includes the <code class="language-plaintext highlighter-rouge">set-library-permissions</code> command which let’s you recursively correct the permissions on a data library. Simply run:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>set-library-permissions -g https://galaxy.example.com -a $API_KEY LIBRARY --roles ROLES role1,role2,role3<br></code></pre></div></div>   <p>Where <code class="language-plaintext highlighter-rouge">LIBRARY</code> is the id of the library you wish to correct.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="datasets"> <a href="#datasets"> Datasets </a></h3>

<hr/>



<h4 class="faq-area"> Adding a tag <a href="/training-material/faqs/galaxy/datasets_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tags can help you to better organize your history and track datasets.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Cambiar el tipo de datos <a href="/training-material/faqs/galaxy-es/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-cambiar-el-tipo-de-datos"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-cambiar-el-tipo-de-datos" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Cambiar el tipo de datos</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Selecciona sobre el <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>icono del lápiz</strong> para editar los atributos del conjunto de datos</li>   <li>Selecciona en la pestaña <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> en la parte superior del panel central</li>   <li>Selecciona  tu tipo de datos</li>   <li>Da clic en el botón <strong>Change datatype</strong></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changer le type de données <a href="/training-material/faqs/galaxy-fr/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changer-le-type-de-données"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changer-le-type-de-données" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changer le type de données</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Cliquez sur l’icône <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>icône crayon</strong> pour modifier les attributs du jeu de données</li>   <li>Sélectionnez l’onglet <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Types de données</strong> en haut du volet central</li>   <li>Sélectionnez   votre type de données</li>   <li>Cliquez sur le bouton <strong>Modifier le type de données</strong></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing database/build (dbkey) <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can tell Galaxy which dbkey (e.g. reference genome) your dataset is associated with. This may be used by tools to automatically use the correct settings.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click the desired dataset’s name to expand it.</li>   <li>     <p>Click on the “?” next to database indicator:</p>     <p><img src="/training-material/shared/images/datasets_dbkey.svg" alt="UI for changing dbkey" /></p>   </li>   <li>In the central panel, change the <strong>Database/Build</strong> field</li>   <li>Select your desired database key from the dropdown list</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the datatype <a href="/training-material/faqs/galaxy/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select  your desired datatype  from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Converting the file format <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Some datasets can be transformed into a different format. Galaxy has some built-in file conversion options depending on the type of data you have.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-converting-the-file-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-converting-the-file-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Converting the file format</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the upper part <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong>, select the appropriate datatype from the list</li>   <li>Click the <strong>Create dataset</strong> button to start the conversion.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a new file <a href="/training-material/faqs/galaxy/datasets_create_new_file.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy allows you to create new files from the upload menu. You can supply the contents of the file.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-file"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-file" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>   <li>Paste the file contents into the text field</li>   <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Datasets not downloading at all <a href="/training-material/faqs/galaxy/datasets_not_downloading_at_all.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-datasets-not-downloading-at-all"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-datasets-not-downloading-at-all" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Datasets not downloading at all</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Check to see if pop-ups are blocked by your web browser. Where to check can vary by browser and extensions.</li>   <li>Double check your API key, if used. Go to <strong>User &gt; Preferences &gt; Manage API key.</strong></li>   <li>Check the sharing/permission status of the Datasets. Go to <strong>Dataset &gt; Pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> &gt; Edit attributes &gt; Permissions.</strong> If you do not see a “Permissions” tab, then you are not the owner of the data.</li> </ol>   <p>Notes:</p>   <ul>   <li>If the data was shared with you by someone else from a Shared History, or was copied from a Published History, be aware that there are multiple levels of data sharing permissions.</li>   <li>All data are set to <strong>not shared</strong> by default.</li>   <li>Datasets sharing permissions for a new history can be set before creating a new history. Go to <strong>User &gt; Preferences &gt; Set Dataset Permissions for New Histories.</strong></li>   <li><strong>User &gt; Preferences &gt; Make all data private is a “one click” option to unshare ALL data (Datasets, Histories).</strong> Note that once confirmed and all data is unshared, the action cannot be “undone” in batch, even by an administrator. You will need to re-share data again and/or reset your global sharing preferences as wanted.</li>   <li>Only the data owner has control over sharing/permissions.</li>   <li>Any data you upload or create yourself is automatically owned by you with full access.</li>   <li>You may not have been granted full access if the data were shared or imported, and someone else is the data owner (your copy could be “view only”).</li>   <li>After you have a fully shared copy of any shared/published data from someone else, then you become the owner of that data copy. If the other person or you make changes, it applies to each person’s copy of the data, individually and only.</li>   <li>Histories can be shared with included Datasets. Datasets can be downloaded/manipulated by others or viewed by others.</li>   <li>Share access to Datasets is distinct but it relates to Histories’ access.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Detecting the datatype (file format) <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-detecting-the-datatype-file-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-detecting-the-datatype-file-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Detecting the datatype (file format)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>Click the <strong>Auto-detect</strong> button to have Galaxy try to autodetect it.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Different dataset icons and their usage <a href="/training-material/faqs/galaxy/datasets_icons.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Icons provide a visual experience for objects, actions, and ideas
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-different-dataset-icons-and-their-usage"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-different-dataset-icons-and-their-usage" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Different dataset icons and their usage</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Dataset icons and their usage:</p>   <ul>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <strong>“Eye icon”</strong>: Display dataset contents.</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>“Pencil icon”</strong>: Edit attributes of dataset metadata: labels, datatype, database.</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> <strong>“Trash icon”</strong>: Delete the dataset.</li>   <li><i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> <strong>“Disc icon”</strong>: Download the dataset.</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> <strong>“Copy link”</strong>: Copy link URL to the dataset.</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> <strong>“Info icon”</strong>: Dataset details and job runtime information: inputs, parameters, logs.</li>   <li><i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> <strong>“Refresh/Rerun icon”</strong>: Run this (selected) job again or examine original submitted form.</li>   <li><i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> <strong>“Visualize icon”</strong>: External display links (UCSC, IGV, NPL, PV); Charts and graphing; Editor (manually edit text).</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">galaxy-dataset-map</span> <strong>“Dataset Map icon”</strong>: Filter the history for related Input/Output Datasets. Click again to clear the filter.</li>   <li><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> <strong>“Bug icon”</strong>: Review subset of logs (review all under <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span>), and optionally submit a bug report.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading datasets <a href="/training-material/faqs/galaxy/datasets_download_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the dataset in your history to expand it</li>   <li>Click on the Download icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> to save the dataset to your computer.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading datasets using command line <a href="/training-material/faqs/galaxy/datasets_download_using_command_line.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-datasets-using-command-line"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-datasets-using-command-line" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading datasets using command line</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>From the terminal window on your computer, you can use <strong>wget</strong> or <strong>curl</strong>.</p>   <ol>   <li>Make sure you have <strong>wget</strong> or <strong>curl</strong> installed.</li>   <li>Click on the Dataset name, then click on the copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span>. This is the direct-downloadable dataset link.</li>   <li>Once you have the link, use any of the following commands:     <ul>       <li>For <strong>wget</strong>         <blockquote>           <p><strong><code class="language-plaintext highlighter-rouge">wget '&lt;link&gt;'</code> <br /> <code class="language-plaintext highlighter-rouge">wget -O '&lt;link&gt;'</code><br /> <code class="language-plaintext highlighter-rouge">wget -O --no-check-certificate '&lt;link&gt;'  # ignore SSL certificate warnings</code> <br /> <code class="language-plaintext highlighter-rouge">wget -c '&lt;link&gt;' # continue an interrupted download</code></strong></p>         </blockquote>       </li>       <li>For <strong>curl</strong>         <blockquote>           <p><strong><code class="language-plaintext highlighter-rouge">curl -o outfile '&lt;link&gt;'</code><br /> <code class="language-plaintext highlighter-rouge">curl -o outfile --insecure '&lt;link&gt;'     # ignore SSL certificate warnings</code><br /> <code class="language-plaintext highlighter-rouge">curl -C - -o outfile '&lt;link&gt;'           # continue an interrupted download</code></strong></p>         </blockquote>       </li>     </ul>   </li>   <li>For dataset collections and datasets within collections you have to supply your <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html">API key</a> with the request     <ul>       <li>Sample commands for <strong>wget</strong> and <strong>curl</strong> respectively are:         <blockquote>           <p><strong><code class="language-plaintext highlighter-rouge">wget https://usegalaxy.org/api/dataset_collections/d20ad3e1ccd4595de/download?key=MYSECRETAPIKEY</code></strong></p>           <p><strong><code class="language-plaintext highlighter-rouge">curl -o myfile.txt https://usegalaxy.org/api/dataset_collections/d20ad3e1ccd4595de/download?key=MYSECRETAPIKEY</code></strong></p>         </blockquote>       </li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding BAM dataset identifiers <a href="/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Quickly learn what the identifiers are in any **BAM** dataset that is the result from mapping
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-bam-dataset-identifiers"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-bam-dataset-identifiers" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding BAM dataset identifiers</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Run <strong>Samtools: IdxStats</strong> on the aligned data (<code class="language-plaintext highlighter-rouge">bam</code> dataset).</li>   <li>The “index header” chromosome names and lengths will be listed in the output (along with read counts).</li>   <li>Compare the chromosome identifiers to the chromosome (aka “chrom”) field in all other inputs: VCF, GTF, GFF(3), BED, Interval, etc.</li> </ol>   <p><strong>Note:</strong></p>   <ul>   <li>The original mapping target may have been a built-in genome index, custom genome (transcriptome, exome, other) – the same <code class="language-plaintext highlighter-rouge">bam</code> data will still be summarized.</li>   <li>This method will <em>not</em> work for “sequence-only” <code class="language-plaintext highlighter-rouge">bam</code> datasets, as these usually have no header.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding Datasets <a href="/training-material/faqs/galaxy/datasets_finding_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding Datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>To review all active Datasets in your account, go to <strong>User &gt; Datasets</strong>.</li> </ul>   <p>Notes:</p>   <ul>   <li>Logging out of Galaxy while the Upload tool is still loading data can cause uploads to abort. This is most likely to occur when a dataset is loaded by browsing local files.</li>   <li>If you have more than one browser window open, each with a different Galaxy History loaded, the Upload tool will load data into the most recently used history.</li>   <li>Click on refresh icon <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> at the top of the History panel to display the current active History with the datasets.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to delete datasets? <a href="/training-material/faqs/galaxy/datasets_deleting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-delete-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-delete-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to delete datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Deleting datasets individually</strong></p>   <p>To delete datasets individually simply click the <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> button with dataset’s box. That’s it! This action is reversible: datasets can be undeleted.</p>   <p><strong>Deleting datasets in bulk</strong></p>   <p>To delete multiple datasets at once:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to delete;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Delete”</strong> option.</li> </ul>   <p>This action is also reversible: datasets can be undeleted.</p>   <p><img src="/training-material/shared/images/datasets_deleting.gif" alt="An animated gif showing how to delete datasets" /></p>   <p><strong>Deleting datasets permanently</strong> <font color="red"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> <b>Danger zone!</b></font></p>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-permanent-is-permanent"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Permanent is ... PERMANENT!</div>   <p>Datasets deleted in this fashion CANNOT be undeleted!</p> </blockquote>   <p>To delete multiple datasets <font color="red">PERMANENTLY</font>:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to delete;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Delete (permanently)”</strong> option.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to hide datasets? <a href="/training-material/faqs/galaxy/datasets_hidden.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-hide-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-hide-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to hide datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To hide datasets:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to hide;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Hide”</strong> option.</li> </ul>   <p><img src="/training-material/shared/images/datasets_hide.gif" alt="An animated gif showing how to hide datasets" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to un-delete datasets? <a href="/training-material/faqs/galaxy/datasets_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-un-delete-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-un-delete-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to un-delete datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If your history contains deleted datasets you will see <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> <strong>“Include deleted”</strong> button directly above dataset display.</p>   <p>To un-delete datasets:</p>   <ul>   <li>Type <code class="language-plaintext highlighter-rouge">deleted:true</code> in the search box</li>   <li>Select datasets you want to un-delete</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Undelete”</strong> option.</li> </ul>   <p><img src="/training-material/shared/images/datasets_undeleting.gif" alt="An animated gif showing how to undelete datasets" /></p>   <p>Alternatively, you can:</p>   <ul>   <li>click <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> <strong>“Include deleted”</strong> button directly above dataset display. This will cause deleted datasets to appear in history along with normal (un-deleted) datasets;</li>   <li>deleted datasets are distinguished by having <i class="fas fa-trash-can-arrow-up" aria-hidden="true"></i><span class="visually-hidden">dataset-undelete</span> within dataset box. Clicking on this icon will un-delete a given dataset;</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to un-hide datasets? <a href="/training-material/faqs/galaxy/datasets_unhidden.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-un-hide-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-un-hide-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to un-hide datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If your history contains hidden datasets you will see <i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">galaxy-show-hidden</span> <strong>“Include hidden”</strong> button directly above the dataset display.</p>   <p>To un-hide datasets:</p>   <ul>   <li>Type <code class="language-plaintext highlighter-rouge">visible:hidden</code> in the search box</li>   <li>Select datasets you want to un-hide</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Unhide”</strong> option.</li> </ul>   <p><img src="/training-material/shared/images/datasets_unhide.gif" alt="An animated gif showing how to unhide datasets" /></p>   <p>Alternatively, you can:</p>   <ul>   <li>click <i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">galaxy-show-hidden</span> <strong>“Include hidden”</strong> button directly above dataset display. This will cause hidden datasets to appear in history along with normal (un-hidden) datasets;</li>   <li>hidden datasets are distinguished by having <i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">galaxy-show-hidden</span> within dataset box. Clicking on this icon will un-hide a given dataset;</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Mismatched Chromosome identifiers and how to avoid them <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-mismatched-chromosome-identifiers-and-how-to-avoid-them"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-mismatched-chromosome-identifiers-and-how-to-avoid-them" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Mismatched Chromosome identifiers and how to avoid them</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Reference data mismatches are similiar to bad reagents in a wet lab experiment: all sorts of odd problems can come up!</p>   <p>You inputs must be all based on an identical genome assembly build to achieve correct scientific results.</p>   <p>There are two areas to review for data to be considered <strong>identical</strong>.</p>   <ol>   <li>The data are based on the same exact genome <strong>assembly</strong> (or “assembly release”).     <ul>       <li>The “assembly” refers to the nucleotide sequence of the genome.</li>       <li>If the base order and length of the chromosomes are not the same, then your coordinates will have scientific problems.</li>       <li>Converting coordinates between assemblies may be possible. Search tool panel with <code class="language-plaintext highlighter-rouge">CrossMap</code>.</li>     </ul>   </li>   <li>The data are based on the same exact genome assembly <strong>build</strong>.     <ul>       <li>The “build” refers to the labels used inside the file. In this context, pay attention to the chromosome identifiers.</li>       <li>These all may mean the same thing to a person but not to a computer or tool: chr1, Chr1, 1, chr1.1</li>       <li>Converting identifiers between builds may be possible. Search tool panel with <code class="language-plaintext highlighter-rouge">Replace</code>.</li>     </ul>   </li> </ol>   <p>The methods listed below help to identify and correct errors or unexpected results when the underlying genome assembly build for all inputs are <strong>not identical</strong>.</p>   <p><strong>Method 1</strong>: <a href="/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html">Finding BAM dataset identifiers</a></p>   <p><strong>Method 2</strong>: <a href="/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html">Directly obtaining UCSC sourced <em>genome</em> identifiers</a></p>   <p><strong>Method 3</strong>: <a href="https://galaxyproject.org/support/chrom-identifiers/#adjusting-identifiers-or-input-source">Adjusting identifiers for UCSC sourced data used with other sourced data</a></p>   <p><strong>Method 4</strong>: <a href="https://galaxyproject.org/support/chrom-identifiers/#any-mixed-sourced-data">Adjusting identifiers or input source for any mixed sourced data</a></p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Reference data is self referential. <a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html">More help for your genome, transcriptome, and annotation</a></p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Genome not available as a native index? <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html">Use a custom genome fasta</a> and <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">create a custom build database</a> instead.</p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> More notes on Native Reference Genomes</p>   <ul>   <li>Native <strong>reference genomes</strong> (FASTA) are built as pre-computed indexes on the Galaxy server where you are working.</li>   <li>Different servers host both common <em>and</em> different reference genome data.</li>   <li>Most <strong>reference annotation</strong> (tabular, GTF, GFF3) is supplied from the history by the user, even when the genome is indexed.</li>   <li>Public Galaxy servers source reference genomes preferentially from <a href="https://hgdownload.soe.ucsc.edu/downloads.html">UCSC</a>.</li>   <li>A <strong>reference transcriptome</strong> (FASTA) is supplied from the history by the user.</li>   <li>Many experiements use a combination of all three types of reference data. Consider pre-preparing your files at the start!</li>   <li>The default variant for a native genome index is “Full”. Defined as: all primary chromosomes (or scaffolds/contigs) including mitochondrial plus associated unmapped, plasmid, and other segments.</li>   <li>When only one version of a genome is available for a tool, it represents the default “Full” variant.</li>   <li>Some genomes will have more than one variant available.</li>   <li>The “Canonical Male” or sometimes simply “Canonical” variant contains the primary chromosomes for a genome. For example a human “Canonical” variant contains chr1-chr22, chrX, chrY, and chrM.</li>   <li>The “Canonical Female” variant contains the primary chromosomes excluding chrY.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Moving datasets between Galaxy servers <a href="/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-moving-datasets-between-galaxy-servers"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-moving-datasets-between-galaxy-servers" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Moving datasets between Galaxy servers</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>On the origin Galaxy server:</p>   <ol>   <li>Click on the name of the dataset to expand the info.</li>   <li>Click on the <strong>Copy link icon</strong> <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span>.</li> </ol>   <p>On the destination Galaxy server:</p>   <ol>   <li>Click on <strong>Upload data &gt; Paste / Fetch Data</strong> and paste the link. Select attributes, such as genome assembly, if required. Hit the <strong>Start</strong> button.</li> </ol>   <p>Note: The copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> cannot be used to move HTML datasets (but this can be downloaded using the download button <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>) and SQLite datasets.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Purging datasets <a href="/training-material/faqs/galaxy/datasets_purging_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-purging-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-purging-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Purging datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>All account Datasets can be reviewed under <strong>User &gt; Datasets</strong>.</li>   <li>To permanently delete: use the link from within the dataset, or use the Operations on Multiple Datasets functions, or use the Purge Deleted Datasets option in the History menu.</li> </ol>   <p>Notes:</p>   <ul>   <li>Within a History, deleted/permanently deleted Datasets can be reviewed by toggling the deleted link at the top of the History panel, found immediately under the History name.</li>   <li>Both active (shown by default) and hidden (the other toggle link, next to the deleted link) datasets can be reviewed the same way.</li>   <li>Click on the far right “X” to delete a dataset.</li>   <li>Datasets in a deleted state are still part of your quota usage.</li>   <li>Datasets must be purged (permanently deleted) to not count toward quota.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Quotas for datasets and histories <a href="/training-material/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-quotas-for-datasets-and-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-quotas-for-datasets-and-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Quotas for datasets and histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li><em>Deleted datasets</em> and <em>deleted histories</em> containing datasets <strong>are considered when calculating quotas</strong>.</li>   <li><em>Permanently deleted datasets</em> and <em>permanently deleted histories</em> containing datasets <strong>are not considered</strong>.</li>   <li><em>Histories/datasets that are shared with you</em> are only <strong>partially considered</strong> unless you import them.</li> </ul>   <p><strong>Note:</strong> To reduce quota usage, refer to <a href="/training-material/faqs/galaxy/account_reduce_quota_usage.html">How can I reduce quota usage while still retaining prior work (data, tools, methods)?</a> FAQ.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming a dataset <a href="/training-material/faqs/galaxy/datasets_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding job statuses <a href="/training-material/faqs/galaxy/datasets_job_status.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Job statuses will help you understand the stages of your work.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-job-statuses"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-job-statuses" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding job statuses</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Compare the color of your datasets to these job processing stages.</p>   <ul>   <li><strong>Grey</strong>: The job is queued. Allow this to complete!</li>   <li><strong>Yellow</strong>: The job is executing. Allow this to complete!</li>   <li><strong>Green</strong>: The job has completed successfully.</li>   <li><strong>Red</strong>: The job has failed. Check your inputs and parameters with Help examples and GTN tutorials. Scroll to the bottom of the tool form to find these.</li>   <li><strong>Light Blue</strong>: The job is paused. This indicates either an input has a problem or that you have exceeded the disk quota set by the administrator of the Galaxy instance you are working on.</li>   <li><strong>Grey, Yellow, Grey again</strong>: The job is waiting to run due to admin re-run or an automatic fail-over to a longer-running cluster.</li> </ul>   <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span>  Don’t lose your queue placement! <strong>It is <em>essential</em> to allow queued jobs to remain queued, and to never interrupt an executing job.</strong> If you delete/re-run jobs, they are added back to the <em>end of the queue again</em>.</p>   <p>Related FAQs</p>   <ul>   <li><a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting errors</a></li>   <li><a href="/training-material/faqs/galaxy/analysis_jobs_not_running.html">My jobs aren’t running!</a></li>   <li><a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html">Extended Help for Differential Expression Analysis Tools</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with GFF GFT GTF2 GFF3 reference annotation <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-gff-gft-gtf2-gff3-reference-annotation"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-gff-gft-gtf2-gff3-reference-annotation" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with GFF GFT GTF2 GFF3 reference annotation</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>All annotation datatypes have a distinct format and content specification.     <ul>       <li>Data providers may release variations of any, and tools may produce variations.</li>       <li>GFF3 data may be labeled as GFF.</li>       <li>Content can overlap but is generally not understood by tools that are expecting just one of these specific formats.</li>     </ul>   </li>   <li>Best practices     <ul>       <li>The sequence identifiers must exactly match between reference annotation and reference genomes transcriptomes exomes.</li>       <li>Most tools expect GFT format unless the tool form specifically notes otherwise.         <ul>           <li>Get the GTF version from the data providers if it is available.</li>           <li>If only GFF3 is available, you can attempt to transform it with the tool <strong>gffread</strong>.</li>         </ul>       </li>       <li>Was GTF data detected as GFF during Upload? It probably has headers.   -Remove the headers (lines that start with a “#”) with the <strong>Select</strong> tool using the option “NOT Matching” with the regular expression: ^#         <ul>           <li><a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">Redetect the datatype</a>. It should be GTF once corrected.</li>         </ul>       </li>       <li>UCSC annotation         <ul>           <li>Find annotation under their Downloads area. The path will be similar to: <code class="language-plaintext highlighter-rouge">https://hgdownload.soe.ucsc.edu/goldenPath/&lt;database&gt;/bigZips/genes/</code></li>           <li>Copy the URL from UCSC and paste it into the Upload tool, allowing Galaxy to detect the datatype.</li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with deleted datasets <a href="/training-material/faqs/galaxy/datasets_working_with_deleted_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-deleted-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-deleted-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with deleted datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Deleted datasets and histories can be recovered by users as they are retained in Galaxy for a time period set by the instance administrator. Deleted datasets can be undeleted or permanently deleted within a History. <em>Links to show/hide deleted (and hidden) datasets are at the top of the History panel.</em></p>   <ul>   <li>To review or adjust an individual dataset:     <ol>       <li>Click on the name to expand it.</li>       <li>If it is only deleted, but not permanently deleted, you’ll see a message with links to recover or to purge.         <ul>           <li>Click on <em>Undelete</em> it to recover the dataset, making it active and accessible to tools again.</li>           <li>Click on <em>Permanently remove it from disk</em> to purge the dataset and remove it from the account quota calculation.</li>         </ul>       </li>     </ol>   </li>   <li>To review or adjust multiple datasets in batch:     <ol>       <li>Click on the checked box icon <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> near the top left of the history panel (Select Items) to switch into “Operations on Multiple Datasets” mode.</li>       <li>Accordingly for each individual dataset, choose the selection box. Check the datasets you want to modify and choose your option (show, hide, delete, undelete, purge, and group datasets).</li>     </ol>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with very large fasta datasets <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-very-large-fasta-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-very-large-fasta-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with very large fasta datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Run <strong>FastQC</strong> on your data to make sure the format/content is what you expect. Run more QA as needed.     <ul>       <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword “qa-qc” for examples.</li>       <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “qa-qc” and “fasta” for more help.</li>     </ul>   </li>   <li>Assembly result?     <ul>       <li>Consider filtering by length to remove reads that did not assemble.</li>       <li>Formatting criteria:         <ul>           <li>All sequence identifiers must be unique.</li>           <li>Some tools will require that there is no description line content, only identifiers, in the fasta title line (“&gt;” line). Use <strong>NormalizeFasta</strong> to remove the description (all content after the first whitespace) and wrap the sequences to 80 bases.</li>         </ul>       </li>     </ul>   </li>   <li><a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">Custom genome</a>, transcriptome exome?     <ul>       <li>Only appropriate for smaller genomes (bacterial, viral, most insects).</li>       <li>Not appropriate for any mammalian genomes, or some plants/fungi.</li>       <li>Sequence identifiers must be an exact match with all other inputs or expect problems. See <strong>GFF GFT GFF3</strong>.</li>       <li>Formatting criteria:         <ul>           <li>All sequence identifiers must be unique.</li>           <li>ALL tools will require that there is no description content, only identifiers, in the fasta title line (“&gt;” line). Use <strong>NormalizeFasta</strong> to remove the description (all content after the first whitespace) and wrap the sequences to 80 bases.</li>           <li>The only exception is when executing the <strong>MakeBLASTdb</strong> tool and when the input fasta is in NCBI BLAST format (see the tool form).</li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with very large fastq datasets <a href="/training-material/faqs/galaxy/datasets_working_with_fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-very-large-fastq-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-very-large-fastq-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with very large fastq datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Run <strong>FastQC</strong> on your data to make sure the format/content is what you expect. Run more QA as needed.     <ul>       <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword “qa-qc” for examples.</li>       <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “qa-qc” and “fastq” for more help.</li>     </ul>   </li>   <li>How to create a single smaller input. Search the tool panel with the keyword “subsample” for tool choices.</li>   <li>How to create multiple smaller inputs. Start with <strong>Split file to dataset collection</strong>, then merge the results back together using a tool specific for the datatype. Example: BAM results? Use <strong>MergeSamFiles</strong>.</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="datatypes"> <a href="#datatypes"> Datatypes </a></h3>

<hr/>



<h4 class="faq-area"> Best practices for loading fastq data into Galaxy <a href="/training-material/faqs/galaxy/datatypes_best_practices_fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-best-practices-for-loading-fastq-data-into-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-best-practices-for-loading-fastq-data-into-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Best practices for loading fastq data into Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>As of release <code class="language-plaintext highlighter-rouge">17.09</code>, <code class="language-plaintext highlighter-rouge">fastq</code> data will have the datatype <code class="language-plaintext highlighter-rouge">fastqsanger</code> auto-detected when that quality score scaling is detected and “autodetect” is used within the Upload tool. Compressed <code class="language-plaintext highlighter-rouge">fastq</code> data will be converted to uncompressed in the history.</li>   <li>To preserve <code class="language-plaintext highlighter-rouge">fastq</code> compression, directly assign the appropriate datatype (eg: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>).</li>   <li>If the data is close to or over 2 GB in size, be sure to use FTP.</li>   <li>If the data was already loaded as <code class="language-plaintext highlighter-rouge">fastq.gz</code>, don’t worry! Just test the data for correct format (as needed) and assign the <a href="/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html">metadata type</a>.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Compressed FASTQ files, (`*.gz`) <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_compressed.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-compressed-fastq-files-gz"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-compressed-fastq-files-gz" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Compressed FASTQ files, (`*.gz`)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Files ending in <code class="language-plaintext highlighter-rouge">.gz</code> are compressed (zipped) files.     <ul>       <li>The <code class="language-plaintext highlighter-rouge">fastq.gz</code> format is a compressed version of a <code class="language-plaintext highlighter-rouge">fastq</code> dataset.</li>       <li>The <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> format is a compressed version of the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype, etc.</li>     </ul>   </li>   <li>Compression saves space (and therefore your quota).</li>   <li>Tools can accept the compressed versions of input files</li>   <li>Make sure the datatype (compressed or uncompressed) is correct for your files, or it may cause tool errors.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Compressed FASTQ files, (`*.gz`) <a href="/training-material/faqs/galaxy/datatypes_fastq_compressed.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-compressed-fastq-files-gz-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-compressed-fastq-files-gz-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Compressed FASTQ files, (`*.gz`)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Files ending in <code class="language-plaintext highlighter-rouge">.gz</code> are compressed (zipped) files.     <ul>       <li>The <code class="language-plaintext highlighter-rouge">fastq.gz</code> format is a compressed version of a <code class="language-plaintext highlighter-rouge">fastq</code> dataset.</li>       <li>The <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> format is a compressed version of the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype, etc.</li>     </ul>   </li>   <li>Compression saves space (and therefore your quota).</li>   <li>Tools can accept the compressed versions of input files</li>   <li>Make sure the datatype (compressed or uncompressed) is correct for your files, or it may cause tool errors.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> FASTQ files: `fastq` vs `fastqsanger` vs .. <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_varieties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-fastq-files-fastq-vs-fastqsanger-vs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-fastq-files-fastq-vs-fastqsanger-vs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: FASTQ files: `fastq` vs `fastqsanger` vs ..</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>FASTQ files come in various flavours. They differ in the encoding scheme they use. See our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#inspect-a-raw-sequence-file">QC tutorial</a> for a more detailed explanation of encoding schemes.</p>   <p>Nowadays, the most commonly used encoding scheme is sanger. In Galaxy, this is the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype. If you are using older datasets, make sure to verify the FASTQ encoding scheme used in your data.</p>   <p><strong>Be Careful: choosing the wrong encoding scheme can lead to incorrect results!</strong></p>   <p><strong>Tip:</strong> There are 2 Galaxy datatypes that have similar names, but are not the same, please make sure you <code class="language-plaintext highlighter-rouge">fastqsanger</code> and <code class="language-plaintext highlighter-rouge">fastqcssanger</code> (not the additional <code class="language-plaintext highlighter-rouge">cs</code>).</p>   <p><strong>Tip:</strong> When in doubt, choose <code class="language-plaintext highlighter-rouge">fastqsanger</code></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> FASTQ files: `fastq` vs `fastqsanger` vs .. <a href="/training-material/faqs/galaxy/datatypes_fastq_varieties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-fastq-files-fastq-vs-fastqsanger-vs-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-fastq-files-fastq-vs-fastqsanger-vs-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: FASTQ files: `fastq` vs `fastqsanger` vs ..</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>FASTQ files come in various flavours. They differ in the encoding scheme they use. See our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#inspect-a-raw-sequence-file">QC tutorial</a> for a more detailed explanation of encoding schemes.</p>   <p>Nowadays, the most commonly used encoding scheme is sanger. In Galaxy, this is the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype. If you are using older datasets, make sure to verify the FASTQ encoding scheme used in your data.</p>   <p><strong>Be Careful: choosing the wrong encoding scheme can lead to incorrect results!</strong></p>   <p><strong>Tip:</strong> There are 2 Galaxy datatypes that have similar names, but are not the same, please make sure you <code class="language-plaintext highlighter-rouge">fastqsanger</code> and <code class="language-plaintext highlighter-rouge">fastqcssanger</code> (not the additional <code class="language-plaintext highlighter-rouge">cs</code>).</p>   <p><strong>Tip:</strong> When in doubt, choose <code class="language-plaintext highlighter-rouge">fastqsanger</code></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment? <a href="/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-fastq-gz-datasets-relate-to-the-fastqsanger-datatype-metadata-assignment"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-fastq-gz-datasets-relate-to-the-fastqsanger-datatype-metadata-assignment" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Before assigning <code class="language-plaintext highlighter-rouge">fastqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>, be sure to confirm the format.</p>   <p><strong>TIP:</strong></p>   <ul>   <li>Using <em>non-fastqsanger</em> scaled quality values will cause scientific problems with tools that expected <code class="language-plaintext highlighter-rouge">fastqsanger</code> formatted input.</li>   <li>Even if the tool does not fail, get the format right from the start to avoid problems. Incorrect format is still one of the most common reasons for tool errors or unexpected results (within Galaxy or not).</li>   <li>For more information on <a href="/training-material/faqs/galaxy/datatypes_fastq_format.html">How to format fastq data for tools that require .fastqsanger format?</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to format fastq data for tools that require .fastqsanger format? <a href="/training-material/faqs/galaxy/datatypes_fastq_format.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-format-fastq-data-for-tools-that-require-fastqsanger-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-format-fastq-data-for-tools-that-require-fastqsanger-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to format fastq data for tools that require .fastqsanger format?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Most tools that accept FASTQ data expect it to be in a specific FASTQ version: <code class="language-plaintext highlighter-rouge">.fastqsanger</code>. The <code class="language-plaintext highlighter-rouge">.fastqsanger</code> datatype must be assigned to each FASTQ dataset.</li> </ul>   <p>In order to do that:</p>   <ul>   <li>Watch the <strong><a href="http://vimeo.com/galaxyproject/fastqprep">FASTQ Prep Illumina</a></strong> video for a complete walk-through.</li>   <li>Run <strong>FastQC</strong> first to assess the type.     <ul>       <li>Run <strong>FASTQ Groomer</strong> if the data needs to have the quality scores rescaled.</li>       <li>If you are certain that the quality scores are already scaled to Sanger Phred+33 (the result of an Illumina 1.8+ pipeline), the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code> can be directly assigned. Click on the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <strong>Edit Attributes</strong> form. In the center panel, click on the “Datatype” tab, enter the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, and save.</li>     </ul>   </li>   <li>Run <strong>FastQC</strong> again on the entire dataset <em>if any changes were made to the quality scores</em> for QA.</li> </ul>   <p><strong>Other tips</strong></p>   <ul>   <li>If you are not sure what type of FASTQ data you have (maybe it is not Illumina?), see the help directly on the <strong>FASTQ Groomer</strong> tool for information about types.     <ul>       <li>For <em>Illumina</em>, first run <strong>FastQC</strong> on a sample of your data (<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">how to read the full report</a>). The output report will note the quality score type interpreted by the tool. If not <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, run <strong>FASTQ Groomer</strong> on the entire dataset. If <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, just assign the datatype.</li>       <li>For <em>SOLiD</em>, run <strong>NGS: Fastq manipulation → AB-SOLID DATA → Convert</strong>, to create a <code class="language-plaintext highlighter-rouge">.fastqcssanger</code> dataset. If you have uploaded a color space fastq sequence with quality scores already scaled to Sanger Phred+33 (<code class="language-plaintext highlighter-rouge">.fastqcssanger</code>), first confirm by running <strong>FastQC</strong> on a sample of the data. Then if you want to double-encode the color space into psuedo-nucleotide space (required by certain tools), see the instructions on the tool form <strong>Fastq Manipulation</strong> for the conversion.</li>       <li>If your data is <em>FASTA</em>, but you want to use tools that require FASTQ input, then using the tool <strong>NGS: QC and manipulation → Combine FASTA and QUAL</strong>. This tool will create “placeholder” quality scores that fit your data. On the output, click on the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <strong>Edit Attributes</strong> form. In the center panel, click on the “Datatype” tab, enter the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, and save.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Identifying and formatting Tabular Datasets <a href="/training-material/faqs/galaxy/datatypes_tabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Format help for Tabular/BED/Interval Datasets
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-identifying-and-formatting-tabular-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-identifying-and-formatting-tabular-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Identifying and formatting Tabular Datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A Tabular datatype is human readable and has tabs separating data columns. Please note that tabular data is different from comma separated data (.csv) and the common datatypes are: <code class="language-plaintext highlighter-rouge">.bed</code>, <code class="language-plaintext highlighter-rouge">.gtf</code>, <code class="language-plaintext highlighter-rouge">.interval</code>, or <code class="language-plaintext highlighter-rouge">.txt</code>.</p>   <ol>   <li>Click the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <strong><em>Edit Attributes</em></strong> form.     <ol>       <li>Change the datatype (3rd tab) and save.</li>       <li>Label columns (1st tab) and save.</li>       <li>Metadata will be assigned, then the dataset can be used.</li>     </ol>   </li>   <li>If the required input is a BED or Interval datatype, adjusting (<code class="language-plaintext highlighter-rouge">.tab</code> → <code class="language-plaintext highlighter-rouge">.bed</code>, <code class="language-plaintext highlighter-rouge">.tab</code> → <code class="language-plaintext highlighter-rouge">.interval</code>) maybe possible using a combination of <strong><em>Text Manipulation</em></strong> tools, to create a dataset that matches required specifications.</li>   <li>Some tools require that BED format be followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.     <ul>       <li>These tools will fail, if they are run with malformed BED datasets or non-specific column assignments.</li>       <li><strong>Solution</strong>: reorganize the data to be in BED format and rerun.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding Datatypes <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-datatypes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-datatypes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding Datatypes</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Allow Galaxy to detect the datatype during Upload, and adjust from there if needed.</li>   <li>Tool forms will filter for the appropriate datatypes it can use for each input.</li>   <li><a href="/training-material/faqs/galaxy/datasets_change_datatype.html">Directly changing</a> a datatype can lead to errors. Be intentional and consider <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">converting</a> instead when possible.</li>   <li>Dataset content can also be adjusted (tools: Data manipulation) and the expected <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">datatype detected</a>. Detected datatypes are the most reliable in most cases.</li>   <li>If a tool does not accept a dataset as valid input, it is not in the correct format with the correct datatype.</li>   <li>Once a dataset’s content matches the datatype, and that dataset is repeatedly used (example: Reference annotation) use that same dataset for all steps in an analysis or expect problems. This may mean rerunning prior tools if you need to make a correction.</li>   <li>Tip: Not sure what datatypes a tool is expecting for an input?     <ol>       <li>Create a new empty history</li>       <li>Click on a tool from the tool panel</li>       <li>The tool form will list the accepted datatypes per input</li>     </ol>   </li>   <li><em>Warning</em>: In some cases, tools will transform a dataset to a new datatype at runtime for you.     <ul>       <li>This is generally helpful, and best reserved for smaller datasets.</li>       <li>Why? This can also unexpectedly create hidden datasets that are near duplicates of your original data, only in a different format.</li>       <li>For large data, that can quickly consume working space (quota).</li>       <li>Deleting/purging any hidden datasets can lead to errors if you are still using the original datasets as an input.</li>       <li>Consider <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">converting</a> to the expected datatype yourself when data is large.</li>       <li>Then test the tool directly on converted data. If it works, purge the original to recover space.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using compressed fastq data as tool inputs <a href="/training-material/faqs/galaxy/datatypes_Using_compressed_fastq_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-compressed-fastq-data-as-tool-inputs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-compressed-fastq-data-as-tool-inputs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using compressed fastq data as tool inputs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>If the tool accepts <code class="language-plaintext highlighter-rouge">fastq</code> input, then <code class="language-plaintext highlighter-rouge">.gz</code> compressed data assigned to the datatype <code class="language-plaintext highlighter-rouge">fastq.gz</code> is appropriate.</li>   <li>If the tool accepts <code class="language-plaintext highlighter-rouge">fastqsanger</code> input, then <code class="language-plaintext highlighter-rouge">.gz</code> compressed data assigned to the datatype <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> is appropriate.</li>   <li>Using uncompressed <code class="language-plaintext highlighter-rouge">fastq</code> data is still an option with tools. The choice is yours.</li> </ul>   <p><strong>TIP:</strong> Avoid labeling compressed data with an uncompressed datatype, and the reverse. Jobs using mismatched datatype versus actual format will fail with an error.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="debugging"> <a href="#debugging"> Debugging </a></h3>

<hr/>



<h4 class="faq-area"> Why isn't my history updating? <a href="/training-material/topics/admin/faqs/why-isnt-my-history-updating.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-isn-t-my-history-updating"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-isn-t-my-history-updating" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why isn't my history updating?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Have you ever experienced that you would submit a job but your history wouldn’t update? Maybe it doesn’t scroll or the datasets stay permanently grey even when you know they should be complete, <em>until you refresh the webpage</em>?</p>   <p>One possible cause of this can be a difference in the clocks of your browser and the server. Check that your clocks match, and if not, reconfigure them! If you are following the Galaxy Admin Training, you will have setup <code class="language-plaintext highlighter-rouge">chrony</code>. Check that your <code class="language-plaintext highlighter-rouge">chrony</code> configuration is valid and requesting time from a local pool.</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># chronyc -n sources<br>210 Number of sources = 1<br>MS Name/IP address         Stratum Poll Reach LastRx Last sample               <br>===============================================================================<br>^? 169.254.169.123               0   7     0     -     +0ns[   +0ns] +/-    0ns<br><br></code></pre></div></div>   <p>This command should return some valid sources. THe above shows an example of a time source that isn’t working, 0ns is not a realistic office and LastRx is empty. Instead it should look more like::</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># chronyc -n sources<br>210 Number of sources = 5<br>MS Name/IP address         Stratum Poll Reach LastRx Last sample               <br>===============================================================================<br>^? 169.254.169.123               0   6     0     -     +0ns[   +0ns] +/-    0ns<br>^? 178.239.19.58                 0   6     0     -     +0ns[   +0ns] +/-    0ns<br>^? 194.104.0.153                 2   6     1     0   +138us[ +138us] +/-   30ms<br>^? 45.138.55.61                  1   6     1     1   -103us[ -103us] +/- 3158us<br>^? 178.239.19.57                 2   6     1     1   -301us[ -301us] +/- 3240us<br></code></pre></div></div>   <p>Here we see a number of sources, with more plausible offsets and non-empty LastRx.</p>   <p>If your time was misconfigured, you might now see something like:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code># chronyc -n tracking<br>Reference ID    : B950F724 (185.80.247.36)<br>Stratum         : 2<br>Ref time (UTC)  : Tue Oct 22 09:44:29 2024<br>System time     : 929.234680176 seconds slow of NTP time<br></code></pre></div></div>   <p>as chrony slowly adjusts the system clock to match NTP time.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="deployment"> <a href="#deployment"> Deployment </a></h3>

<hr/>



<h4 class="faq-area"> Blank page or no CSS/JavaScript <a href="/training-material/topics/admin/faqs/galaxy-blank-screen.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-blank-page-or-no-css-javascript"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-blank-page-or-no-css-javascript" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Blank page or no CSS/JavaScript</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This generally means that serving of static content is broken:</p>   <ul>   <li>Check browser console for 404 errors.</li>   <li>Check proxy error log for permission errors.</li>   <li>Verify that your proxy static configuration is correct.</li>   <li>If you have recently upgraded Galaxy or changed the GUI in some way, you will need to rebuild the client</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Database Issues <a href="/training-material/topics/admin/faqs/database-slow-query.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-database-issues"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-database-issues" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Database Issues</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>For slow queries, start with <code class="language-plaintext highlighter-rouge">EXPLAIN ANALYZE</code></p>   <ul>   <li><a href="https://gist.github.com/natefoo/da46bea8136ba67d65f616c86a27c454">Example of the “jobs ready to run” query</a></li> </ul>   <p>However it can be useful to dig into the queries with the <a href="https://tatiyants.com/pev/#/plans">Postgres EXPLAIN Visualizer (PEV)</a> to get a more visual and clear representation. (Try it with this <a href="https://gist.github.com/hexylena/467c7726b5ab9e18c47080893dbc072e">demo data</a>)</p>   <p>You can set some options in the Galaxy configuration or database that will help debugging this:</p>   <ul>   <li><code class="language-plaintext highlighter-rouge">database_engine_option_echo</code> (but warning, extremely verbose)</li>   <li><code class="language-plaintext highlighter-rouge">slow_query_log_threshold</code> logs to Galaxy log file</li>   <li><code class="language-plaintext highlighter-rouge">sentry_sloreq_threshold</code> if using Sentry</li> </ul>   <p>Additionally check that your database is running <code class="language-plaintext highlighter-rouge">VACUUM</code> regularly enough and look at <code class="language-plaintext highlighter-rouge">VACUUM ANALYZE</code></p>   <p>There are some <code class="language-plaintext highlighter-rouge">gxadmin query pg-*</code> commands which can help you monitor and track this information.</p>   <p>Lastly, check your database settings! It might not have enough resources allocated. Check <a href="https://pgtune.leopard.in.ua/">PGTune</a> for some suggestions of optimised parameters.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Debugging tool errors <a href="/training-material/topics/admin/faqs/galaxy-tool-error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-debugging-tool-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-debugging-tool-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Debugging tool errors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tool stdout/stderr is available in UI under “i” icon on history dataset</p>   <ol>   <li>Set <code class="language-plaintext highlighter-rouge">cleanup_job</code> to <code class="language-plaintext highlighter-rouge">onsuccess</code></li>   <li>Cause a job failure</li>   <li>Go to job working directory (find in logs or <code class="language-plaintext highlighter-rouge">/data/jobs/&lt;hash&gt;/&lt;job_id&gt;</code>)</li>   <li>Poke around, try running things (<code class="language-plaintext highlighter-rouge">srun --pty bash</code> considered useful)</li> </ol>   <p>Familiarize yourself with the places Galaxy keeps things</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Debugging tool memory errors <a href="/training-material/topics/admin/faqs/galaxy-tool-memory-error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-debugging-tool-memory-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-debugging-tool-memory-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Debugging tool memory errors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Often the tool output contains one of:</p>   <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="gp">MemoryError                 #</span><span class="w"> </span>Python<br><span class="gp">what():  std::bad_alloc     #</span><span class="w"> </span>C++<br><span class="gp">Segmentation Fault          #</span><span class="w"> </span>C - but could be other problems too<br><span class="gp">Killed                      #</span><span class="w"> </span>Linux OOM Killer<br></code></pre></div></div>   <p>Solutions:</p>   <ul>   <li>Change input sizes or params     <ul>       <li>Map/reduce?</li>     </ul>   </li>   <li>Decrease the amount of memory the tool needs</li>   <li>Increase the amount of memory available to the job     <ul>       <li>Request more memory from cluster scheduler</li>       <li>Use job resubmission to automatically rerun with a larger memory allocation</li>     </ul>   </li>   <li>Cross your fingers and rerun the job</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Galaxy UI is slow <a href="/training-material/topics/admin/faqs/galaxy-ui-slow.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-galaxy-ui-is-slow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-galaxy-ui-is-slow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Galaxy UI is slow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There is a great <a href="https://docs.galaxyproject.org/en/master/dev/finding_and_improving_slow_code.html#profiling">Tutorial from @mvdbeek</a> which we recommend you follow.</p>   <p>Additionally you can use <code class="language-plaintext highlighter-rouge">py-spy</code> to record the issue and generate a flame graph.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Tool missing from Galaxy <a href="/training-material/topics/admin/faqs/galaxy-tool-missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-tool-missing-from-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-tool-missing-from-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Tool missing from Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>First, restart Galaxy and watch the log for lines like:</p>   <div class="language-console highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="go">Loaded tool id: toolshed.g2.bx.psu.edu/repos/iuc/sickle/sickle/1.33, version: 1.33 into tool panel....<br></span></code></pre></div></div>   <p>After startup, check <code class="language-plaintext highlighter-rouge">integrated_tool_panel.xml</code> for a line like the following to be sure it was loaded properly and added to the toolbox (if not, check the logs further)</p>   <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;tool</span> <span class="na">id=</span><span class="s">"toolshed.g2.bx.psu.edu/repos/iuc/sickle/sickle/1.33"</span> <span class="nt">/&gt;</span><br></code></pre></div></div>   <p>If it is a toolshed tool, check <code class="language-plaintext highlighter-rouge">shed_tool_conf.xml</code> for</p>   <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;tool</span> <span class="na">file=</span><span class="s">"toolshed.g2.bx.psu.edu/repos/iuc/sickle/43e081d32f90/sickle/sickle.xml"</span> <span class="na">guid=</span><span class="s">"toolshed.g2.bx.psu.edu/repos/iuc/sickle/sickle/1.33"</span><span class="nt">&gt;</span><br>...<br><span class="nt">&lt;/tool&gt;</span><br></code></pre></div></div>   <p>Additionally if you have multiple job handlers, <em>sometimes</em>, rarely they don’t all get the update. Just restart them if that’s the case. Alternatively you can send an (authenticated) API requested:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>curl -X PUT https://galaxy.example.org/api/configuration<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using data source tools with Pulsar <a href="/training-material/topics/admin/faqs/data-source-with-pulsar.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-data-source-tools-with-pulsar"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-data-source-tools-with-pulsar" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using data source tools with Pulsar</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Data source tools such as UCSC Main will fail if Pulsar is the default destination.</p>   <p>To fix this issue you can force individual tools to run on a specific destination or handler by adding to your job_conf file:</p>   <p>For job_conf.xml</p>   <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;tools&gt;</span><br>    <span class="nt">&lt;tool</span> <span class="na">id=</span><span class="s">"ucsc_table_direct1"</span> <span class="na">destination=</span><span class="s">"my-local"</span> <span class="nt">/&gt;</span><br><span class="nt">&lt;/tools&gt;</span><br></code></pre></div></div>   <p>For job_conf.yml</p>   <div class="language-yml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">tools</span><span class="pi">:</span><br><span class="pi">-</span> <span class="na">id</span><span class="pi">:</span> <span class="s">ucsc_table_direct1</span><br>  <span class="na">handler</span><span class="pi">:</span> <span class="s">my-local</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="diffs"> <a href="#diffs"> Diffs </a></h3>

<hr/>



<h4 class="faq-area"> How to read a Diff <a href="/training-material/topics/admin/faqs/diffs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-read-a-diff"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-read-a-diff" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to read a Diff</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you haven’t worked with diffs before, this can be something quite new or different.</p>   <p>If we have two files, let’s say a grocery list, in two files. We’ll call them ‘a’ and ‘b’.</p>   <blockquote class="code-2col">   <blockquote class="code-in">     <div class="box-title code-in-title" id="code-in-old"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Old</div>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ cat old<br>🍎<br>🍐<br>🍊<br>🍋<br>🍒<br>🥑<br></code></pre></div>    </div>   </blockquote>   <blockquote class="code-out">     <div class="box-title code-out-title" id="code-out-new"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: New</div>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>$ cat new<br>🍎<br>🍐<br>🍊<br>🍋<br>🍍<br>🥑<br></code></pre></div>    </div>   </blockquote> </blockquote>   <p>We can see that they have some different entries. We’ve removed 🍒 because they’re awful, and replaced them with an 🍍</p>   <p>Diff lets us compare these files</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>diff old new<br>5c5<br>&lt; 🍒<br><span class="nt">---</span><br><span class="o">&gt;</span> 🍍<br></code></pre></div></div>   <p>Here we see that 🍒 is only in a, and 🍍 is only in b. But otherwise the files are identical.</p>   <p>There are a couple different formats to diffs, one is the ‘unified diff’</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>diff <span class="nt">-U2</span> old new<br><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ new	2022-02-16 14:06:36.340962616 +0100<br>@@ <span class="nt">-3</span>,4 +3,4 @@<br> 🍊<br> 🍋<br>-🍒<br>+🍍<br> 🥑<br></code></pre></div></div>   <p>This is basically what you see in the training materials which gives you a lot of context about the changes:</p>   <ul>   <li><code class="language-plaintext highlighter-rouge">--- old</code> is the ‘old’ file in our view</li>   <li><code class="language-plaintext highlighter-rouge">+++ new</code> is the ‘new’ file</li>   <li>@@ these lines tell us where the change occurs and how many lines are added or removed.</li>   <li>Lines starting with a - are removed from our ‘new’ file</li>   <li>Lines with a + have been added.</li> </ul>   <p>So when you go to apply these diffs to your files in the training:</p>   <ol>   <li>Ignore the header</li>   <li>Remove lines starting with - from your file</li>   <li>Add lines starting with + to your file</li> </ol>   <p>The other lines (🍊/🍋 and 🥑) above just provide “context”, they help you know where a change belongs in a file, but <strong>should not be edited</strong> when you’re making the above change. Given the above diff, you would find a line with a 🍒, and replace it with a 🍍</p>   <h4 id="added--removed-lines">Added &amp; Removed Lines</h4>   <p>Removals are very easy to spot, we just have removed lines</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ new	2022-02-16 14:10:14.370722802 +0100<br>@@ <span class="nt">-4</span>,3 +4,2 @@<br> 🍋<br> 🍒<br>-🥑<br></code></pre></div></div>   <p>And additions likewise are very easy, just add a new line, between the other lines in your file.</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ new	2022-02-16 14:11:11.422135393 +0100<br>@@ <span class="nt">-1</span>,3 +1,4 @@<br> 🍎<br>+🍍<br> 🍐<br> 🍊<br></code></pre></div></div>   <h4 id="completely-new-files">Completely new files</h4>   <p>Completely new files look a bit different, there the “old” file is <code class="language-plaintext highlighter-rouge">/dev/null</code>, the empty file in a Linux machine.</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span>diff <span class="nt">-U2</span> /dev/null old<br><span class="nt">---</span> /dev/null	2022-02-15 11:47:16.100000270 +0100<br>+++ old	2022-02-16 14:06:19.697132568 +0100<br>@@ <span class="nt">-0</span>,0 +1,6 @@<br>+🍎<br>+🍐<br>+🍊<br>+🍋<br>+🍒<br>+🥑<br></code></pre></div></div>   <p>And removed files are similar, except with the new file being /dev/null</p>   <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">---</span> old	2022-02-16 14:06:19.697132568 +0100<br>+++ /dev/null	2022-02-15 11:47:16.100000270 +0100<br>@@ <span class="nt">-1</span>,6 +0,0 @@<br>-🍎<br>-🍐<br>-🍊<br>-🍋<br>-🍒<br>-🥑<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="estimation of strandedness"> <a href="#estimation of strandedness"> Estimation of strandedness </a></h3>

<hr/>



<h4 class="faq-area"> In 'infer experiments' I get unequal numbers, but in the IGV it looks like it is unstranded. What does this mean? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-in-infer-experiments-i-get-unequal-numbers-but-in-the-igv-it-looks-like-it-is-unstranded-what-does-this-mean"><i class="far fa-question-circle" aria-hidden="true"></i> Question: In 'infer experiments' I get unequal numbers, but in the IGV it looks like it is unstranded. What does this mean?</div>   <p>It’s also often the case that elimination of the second strand is not perfect, and there are genuine cases of bidirectional transcription in the genome. 70 / 30 % as in your report is not a good result for a stranded library. You can treat this as a stranded library in your analysis, but for instance you couldn’t make the conclusion that a given gene is actually transcribed from the reverse strand. Likely that the library preparation didn’t work perfectly. This can depend on many factors, one is that you need to completely digest your DNA using a high quality DNase before doing the reverse transcription.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> When is the "infer experiment" tool used in practice? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-when-is-the-quot-infer-experiment-quot-tool-used-in-practice"><i class="far fa-question-circle" aria-hidden="true"></i> Question: When is the &quot;infer experiment&quot; tool used in practice?</div>   <p>Often you are already aware whether the RNA-seq data is stranded or not in the first place because you sequenced it yourself or ordered it from a company.</p>   <p>But it can happen in cases where you get the data from someone else, that this information is lost and you need to find out.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="features"> <a href="#features"> Features </a></h3>

<hr/>



<h4 class="faq-area"> Using the Window Manager to view multiple datasets <a href="/training-material/faqs/galaxy/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-the-window-manager-to-view-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-the-window-manager-to-view-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using the Window Manager to view multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you would like to view two or more datasets at once, you can use the <strong>Window Manager</strong> feature in Galaxy:</p>   <ol>   <li><strong>Click</strong> on the <em>Window Manager</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> on the top menu bar.     <ul>       <li>You should see a little checkmark on the icon now</li>     </ul>   </li>   <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> to view the output     <ul>       <li>You should see the output in a window overlayed over Galaxy</li>       <li>You can resize this window by dragging the bottom-right corner</li>     </ul>   </li>   <li><strong>Click</strong> outside the file to exit the Window Manager</li>   <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> a second dataset from your history     <ul>       <li>You should now see a second window with the new dataset</li>       <li>This makes it easier to compare the two outputs</li>     </ul>   </li>   <li>Repeat this for as many files as you would like to compare</li>   <li>You can turn off the <strong>Window Manager</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> by clicking on the icon again</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Uso del cuaderno de apuntes para ver varios conjuntos de datos <a href="/training-material/faqs/galaxy-es/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-uso-del-cuaderno-de-apuntes-para-ver-varios-conjuntos-de-datos"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-uso-del-cuaderno-de-apuntes-para-ver-varios-conjuntos-de-datos" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Uso del cuaderno de apuntes para ver varios conjuntos de datos</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Si deseas ver dos o más conjuntos de datos al mismo tiempo, puedes usar la función <strong>Scratchbook</strong> en Galaxy:     1. <strong>Haz clic</strong> en el icono <em>Scratchbook</em> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> en la barra de menú superior.        - Debería aparecer ver una pequeña marca de verificación en el icono     2. <strong>Ver</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> un conjunto de datos haciendo clic en el icono de ojo <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> para ver el resultado.        - Deberías ver la salida en una ventana emergente sobre Galaxy        - Puedes cambiar el tamaño de esta ventana arrastrando la esquina inferior derecha     3. <strong>Haz clic</strong> fuera del archivo para salir del Scratchbook     4. <strong>Ver</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> un segundo conjunto de datos de tu historial        - Ahora deberías poder ver una segunda ventana con el nuevo conjunto de datos        - Esto hace que sea más fácil comparar las dos salidas.     5. Repite estos pasos para todos los archivos que desees comparar.     6. Puedes desactivar <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> haciendo clic en el icono nuevamente.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why not use Excel? <a href="/training-material/faqs/galaxy/why_not_excel.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Excel is a fantastic tool and a great place to build simple analysis models, but when it comes to scaling, Galaxy wins every time.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-not-use-excel"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-not-use-excel" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why not use Excel?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You could just as easily use Excel to answer the same question, and if the goal is to learn how to use a tool, then either tool would be great! But what if you are working on a question where your analysis matters? Maybe you are working with human clinical data trying to diagnose a set of symptoms, or you are working on research that will eventually be published and maybe earn you a Nobel Prize?</p>   <p>In these cases your analysis, <em>and the ability to reproduce it exactly</em>, is vitally important, and Excel won’t help you here. It doesn’t track changes and it offers very little insight to others on how you got from your initial data to your conclusions.</p>   <p>Galaxy, on the other hand, <em>automatically records every step of your analysis.</em> And when you are done, you can share your analysis with anyone. You can even include a link to it in a paper (or your acceptance speech). In addition, you can create a reusable workflow from your analysis that others (or yourself) can use on other datasets.</p>   <p>Another challenge with spreadsheet programs is that they don’t scale to support <em>next generation sequencing (NGS)</em> datasets, a common type of data in genomics, and which often reach gigabytes or even terabytes in size. Excel has been used for large datasets, but you’ll often find that learning a new tool gives you significantly more ability to scale up, and scale out your analyses.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="format"> <a href="#format"> Format </a></h3>

<hr/>



<h4 class="faq-area"> FASTQ format <a href="/training-material/topics/sequence-analysis/faqs/fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="details">   <div class="box-title details-title" id="details-fastq-format"><button class="gtn-boxify-button details" type="button" aria-controls="details-fastq-format" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> <span>Details: FASTQ format</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Although it looks complicated (and maybe it is), the FASTQ format is easy to understand with a little decoding. Each read, representing a fragment of DNA, is encoded by 4 lines:</p>   <table>   <thead>     <tr>       <th>Line</th>       <th>Description</th>     </tr>   </thead>   <tbody>     <tr>       <td>1</td>       <td>Always begins with <code class="language-plaintext highlighter-rouge">@</code> followed by the information about the read</td>     </tr>     <tr>       <td>2</td>       <td>The actual nucleic sequence</td>     </tr>     <tr>       <td>3</td>       <td>Always begins with a <code class="language-plaintext highlighter-rouge">+</code> and contains sometimes the same info in line 1</td>     </tr>     <tr>       <td>4</td>       <td>Has a string of characters which represent the quality scores associated with each base of the nucleic sequence; must have the same number of characters as line 2</td>     </tr>   </tbody> </table>   <p>So for example, the first sequence in our file is:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@03dd2268-71ef-4635-8bce-a42a0439ba9a runid=8711537cc800b6622b9d76d9483ecb373c6544e5 read=252 ch=179 start_time=2019-12-08T11:54:28Z flow_cell_id=FAL10820 protocol_group_id=la_trappe sample_id=08_12_2019<br>AGTAAGTAGCGAACCGGTTTCGTTTGGGTGTTTAACCGTTTTCGCATTTATCGTGAAACGCTTTCGCGTTTTCGTGCGGAAGGCGCTTCACCCAGGGCCTCTCATGCTTTGTCTTCCTGTTTATTCAGGATCGCCCAAAGCGAGAATCATACCACTAGACCACACGCCCGAATTATTGTTGCGTTAATAAGAAAAGCAAATATTTAAGATAGGAAGTGATTAAAGGGAATCTTCTACCAACAATATCCATTCAAATTCAGGCA<br>+<br>$'())#$$%#$%%'-$&amp;$%'%#$%('+;&lt;&gt;&gt;&gt;18.?ACLJM7E:CFIMK&lt;=@0/.4&lt;9&lt;&amp;$007:,3&lt;IIN&lt;3%+&amp;$(+#$%'$#$.2@401/5=49IEE=CH.20355&gt;-@AC@:B?7;=C4419)*$$46211075.$%..#,529,''=CFF@:&lt;?9B522.(&amp;%%(9:3E99&lt;BIL?:&gt;RB--**5,3(/.-8B&gt;F@@=?,9'36;:87+/19BAD@=8*''&amp;''7752'$%&amp;,5)AM&lt;99$%;EE;BD:=9&lt;@=9+%$<br></code></pre></div></div>   <p>It means that the fragment named <code class="language-plaintext highlighter-rouge">@03dd2268-71ef-4635-8bce-a42a0439ba9a</code> (ID given in line1) corresponds to:</p>   <ul>   <li>the DNA sequence <code class="language-plaintext highlighter-rouge">AGTAAGTAGCGAACCGGTTTCGTTTGGGTGTTTAACCGTTTTCGCATTTATCGTGAAACGCTTTCGCGTTTTCGTGCGGAAGGCGCTTCACCCAGGGCCTCTCATGCTTTGTCTTCCTGTTTATTCAGGATCGCCCAAAGCGAGAATCATACCACTAGACCACACGCCCGAATTATTGTTGCGTTAATAAGAAAAGCAAATATTTAAGATAGGAAGTGATTAAAGGGAATCTTCTACCAACAATATCCATTCAAATTCAGGCA</code> (line2)</li>   <li>this sequence has been sequenced with a quality <code class="language-plaintext highlighter-rouge">$'())#$$%#$%%'-$&amp;$%'%#$%('+;&lt;&gt;&gt;&gt;18.?ACLJM7E:CFIMK&lt;=@0/.4&lt;9&lt;&amp;$007:,3&lt;IIN&lt;3%+&amp;$(+#$%'$#$.2@401/5=49IEE=CH.20355&gt;-@AC@:B?7;=C4419)*$$46211075.$%..#,529,''=CFF@:&lt;?9B522.(&amp;%%(9:3E99&lt;BIL?:&gt;RB--**5,3(/.-8B&gt;F@@=?,9'36;:87+/19BAD@=8*''&amp;''7752'$%&amp;,5)AM&lt;99$%;EE;BD:=9&lt;@=9+%$</code> (line 4).</li> </ul>   <p>But what does this quality score mean?</p>   <p>The quality score for each sequence is a string of characters, one for each base of the nucleotide sequence, used to characterize the probability of misidentification of each base. The score is encoded using the ASCII character table (with <a href="https://en.wikipedia.org/wiki/FASTQ_format#Encoding">some historical differences</a>):</p>   <p><img src="/training-material/topics/sequence-analysis/faqs/images/fastq-quality-encoding.png" alt="Encoding of the quality score with ASCII characters for different Phred encoding. The ascii code sequence is shown at the top with symbols for 33 to 64, upper case letters, more symbols, and then lowercase letters. Sanger maps from 33 to 73 while solexa is shifted, starting at 59 and going to 104. Illumina 1.3 starts at 54 and goes to 104, Illumina 1.5 is shifted three scores to the right but still ends at 104. Illumina 1.8+ goes back to the Sanger except one single score wider. Illumina" /></p>   <p>So there is an ASCII character associated with each nucleotide, representing its <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred quality score</a>, the probability of an incorrect base call:</p>   <table>   <thead>     <tr>       <th>Phred Quality Score</th>       <th>Probability of incorrect base call</th>       <th>Base call accuracy</th>     </tr>   </thead>   <tbody>     <tr>       <td>10</td>       <td>1 in 10</td>       <td>90%</td>     </tr>     <tr>       <td>20</td>       <td>1 in 100</td>       <td>99%</td>     </tr>     <tr>       <td>30</td>       <td>1 in 1000</td>       <td>99.9%</td>     </tr>     <tr>       <td>40</td>       <td>1 in 10,000</td>       <td>99.99%</td>     </tr>     <tr>       <td>50</td>       <td>1 in 100,000</td>       <td>99.999%</td>     </tr>     <tr>       <td>60</td>       <td>1 in 1,000,000</td>       <td>99.9999%</td>     </tr>   </tbody> </table> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Kraken2 and the k-mer approach for taxonomy classification <a href="/training-material/topics/microbiome/faqs/kraken.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="details">   <div class="box-title details-title" id="details-kraken2-and-the-k-mer-approach-for-taxonomy-classification"><button class="gtn-boxify-button details" type="button" aria-controls="details-kraken2-and-the-k-mer-approach-for-taxonomy-classification" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> <span>Details: Kraken2 and the k-mer approach for taxonomy classification</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In the \(k\)-mer approach for taxonomy classification, we use a database containing DNA sequences of genomes whose taxonomy we already know. On a computer, the genome sequences are broken into short pieces of length \(k\) (called \(k\)-mers), usually 30bp.</p>   <p><strong>Kraken</strong> examines the \(k\)-mers within the query sequence, searches for them in the database, looks for where these are placed within the taxonomy tree inside the database, makes the classification with the most probable position, then maps \(k\)-mers to the lowest common ancestor (LCA) of all genomes known to contain the given \(k\)-mer.</p>   <p><img src="/training-material/topics/microbiome/faqs/images/kmers-kraken.jpg" alt="Kraken2" title="Kraken sequence classification algorithm. To classify a sequence, each k-mer in the sequence is mapped to the lowest common ancestor (LCA, i.e. the lowest node) of the genomes that contain that k-mer in the database. The taxa associated with the sequence's k-mers, as well as the taxa's ancestors, form a pruned subtree of the general taxonomy tree, which is used for classification. In the classification tree, each node has a weight equal to the number of k-mers in the sequence associated with the node's taxon. Each root-to-leaf (RTL) path in the classification tree is scored by adding all weights in the path, and the maximal RTL path in the classification tree is the classification path (nodes highlighted in yellow). The leaf of this classification path (the orange, leftmost leaf in the classification tree) is the classification used for the query sequence. Source: &lt;span class=&quot;citation&quot;&gt;&lt;a href=&quot;#Wood2014&quot;&gt;Wood and Salzberg 2014&lt;/a&gt;&lt;/span&gt;" /></p>   <p><strong>Kraken2</strong> uses a compact hash table, a probabilistic data structure that allows for faster queries and lower memory requirements. It applies a spaced seed mask of <em>s</em> spaces to the minimizer and calculates a compact hash code, which is then used as a search query in its compact hash table; the lowest common ancestor (LCA) taxon associated with the compact hash code is then assigned to the k-mer.</p>   <p>You can find more information about the <strong>Kraken2</strong> algorithm in the paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0"><em>Improved metagenomic analysis with Kraken 2</em></a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Quality Scores <a href="/training-material/topics/sequence-analysis/faqs/quality_score.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="details">   <div class="box-title details-title" id="details-quality-scores"><button class="gtn-boxify-button details" type="button" aria-controls="details-quality-scores" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> <span>Details: Quality Scores</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>But what does this quality score mean?</p>   <p>The quality score for each sequence is a string of characters, one for each base of the nucleotide sequence, used to characterize the probability of misidentification of each base. The score is encoded using the ASCII character table (with <a href="https://en.wikipedia.org/wiki/FASTQ_format#Encoding">some historical differences</a>):</p>   <p>To save space, the sequencer records an <a href="http://drive5.com/usearch/manual/quality_score.html">ASCII character</a> to represent scores 0-42. For example 10 corresponds to “+” and 40 corresponds to “I”. FastQC knows how to translate this. This is often called “Phred” scoring.</p>   <p><img src="/training-material/topics/sequence-analysis/faqs/images/fastq-quality-encoding.png" alt="Encoding of the quality score with ASCII characters for different Phred encoding. The ascii code sequence is shown at the top with symbols for 33 to 64, upper case letters, more symbols, and then lowercase letters. Sanger maps from 33 to 73 while solexa is shifted, starting at 59 and going to 104. Illumina 1.3 starts at 54 and goes to 104, Illumina 1.5 is shifted three scores to the right but still ends at 104. Illumina 1.8+ goes back to the Sanger except one single score wider. Illumina" /></p>   <p>So there is an ASCII character associated with each nucleotide, representing its <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred quality score</a>, the probability of an incorrect base call:</p>   <table>   <thead>     <tr>       <th>Phred Quality Score</th>       <th>Probability of incorrect base call</th>       <th>Base call accuracy</th>     </tr>   </thead>   <tbody>     <tr>       <td>10</td>       <td>1 in 10</td>       <td>90%</td>     </tr>     <tr>       <td>20</td>       <td>1 in 100</td>       <td>99%</td>     </tr>     <tr>       <td>30</td>       <td>1 in 1000</td>       <td>99.9%</td>     </tr>     <tr>       <td>40</td>       <td>1 in 10,000</td>       <td>99.99%</td>     </tr>     <tr>       <td>50</td>       <td>1 in 100,000</td>       <td>99.999%</td>     </tr>     <tr>       <td>60</td>       <td>1 in 1,000,000</td>       <td>99.9999%</td>     </tr>   </tbody> </table>   <p>What does 0-42 represent? These numbers, when plugged into a formula, tell us the probability of an error for that base. This is the formula, where Q is our quality score (0-42) and P is the probability of an error:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Q = -10 log10(P)<br></code></pre></div></div>   <p>Using this formula, we can calculate that a quality score of 40 means only 0.00010 probability of an error!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is Taxonomy? <a href="/training-material/topics/microbiome/faqs/taxon.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="details">   <div class="box-title details-title" id="details-what-is-taxonomy"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-taxonomy" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true"></i> <span>Details: What is Taxonomy?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Taxonomy is the method used to naming, defining (circumscribing) and classifying groups of biological organisms based on shared characteristics such as morphological characteristics, phylogenetic characteristics, DNA data, etc. It is founded on the concept that the similarities descend from a common evolutionary ancestor.</p>   <p>Defined groups of organisms are known as <strong>taxa</strong>. Taxa are given a taxonomic rank and are aggregated into super groups of higher rank to create a taxonomic hierarchy. The taxonomic hierarchy includes eight levels: <strong>Domain, Kingdom, Phylum, Class, Order, Family, Genus and Species.</strong></p>   <p><img src="/training-material/topics/microbiome/faqs/images/taxonomy.png" alt="Example of taxonomy. It starts, top to bottom, with Kingdom &quot;Animalia&quot;, Phylum &quot;Chordata&quot;, Class &quot;Mammalia&quot;, and Order &quot;Carnivora&quot;. Then it splits in 3. On the left, Family &quot;Felidae&quot;, with 2 genus &quot;Felis&quot; and &quot;Panthera&quot; and below 3 species &quot;F. catus&quot; and &quot;F. pardalis&quot; below &quot;Felis&quot;, &quot;P. pardus&quot; below &quot;Panthera&quot;. In the middle, Family &quot;Canidae&quot;, genus &quot;Canis&quot; and 2 species &quot;C. familiaris&quot; and &quot;C. lupus&quot;. On the right, Family &quot;Ursidae&quot;, Genus &quot;Ursus&quot; and 2 species &quot;U. arctos&quot; and &quot;U. horribilus&quot;. Below each species is a illustration of the species" /></p>   <p>The classification system begins with <strong>3 domains</strong> that encompass all living and extinct forms of life</p>   <ul>   <li>The <strong>Bacteria</strong> and <strong>Archae</strong> are mostly microscopic, but quite widespread.</li>   <li>Domain <strong>Eukarya</strong> contains more complex organisms</li> </ul>   <p>When new species are found, they are assigned into taxa in the taxonomic hierarchy. For example for the cat:</p>   <table>   <thead>     <tr>       <th>Level</th>       <th>Classification</th>     </tr>   </thead>   <tbody>     <tr>       <td>Domain</td>       <td>Eukaryota</td>     </tr>     <tr>       <td>Kingdom</td>       <td>Animalia</td>     </tr>     <tr>       <td>Phylum</td>       <td>Chordata</td>     </tr>     <tr>       <td>Class</td>       <td>Mammalia</td>     </tr>     <tr>       <td>Order</td>       <td>Carnivora</td>     </tr>     <tr>       <td>Family</td>       <td>Felidae</td>     </tr>     <tr>       <td>Genus</td>       <td><em>Felis</em></td>     </tr>     <tr>       <td>Species</td>       <td><em>F. catus</em></td>     </tr>   </tbody> </table>   <p>From this classification, one can generate a tree of life, also known as a phylogenetic tree. It is a rooted tree that describes the relationship of all life on earth. At the root sits the “last universal common ancestor” and the three main branches (in taxonomy also called domains) are bacteria, archaea and eukaryotes. Most important for this is the idea that all life on earth is derived from a common ancestor and therefore when comparing two species, you will -sooner or later- find a common ancestor for all of them.</p>   <p>Let’s explore taxonomy in the Tree of Life, using <a href="https://lifemap.univ-lyon1.fr/">Lifemap</a></p>   <iframe id="Lifemap" src="https://lifemap.univ-lyon1.fr/explore.html" frameborder="0" width="100%" height="600px"></iframe> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="further reading"> <a href="#further reading"> Further reading </a></h3>

<hr/>



<h4 class="faq-area"> Where can I read more about this analysis? <a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop/faqs/more_info.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-where-can-i-read-more-about-this-analysis"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Where can I read more about this analysis?</div>   <p>This tutorial was adapted from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a> created by the Schloss lab. Here you can find more information about the mothur tools and file formats. Their <a href="https://mothur.org/wiki/frequently_asked_questions/">FAQ page</a> and <a href="https://forum.mothur.org/">Help Forum</a> are also quite useful!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where can I read more about this analysis? <a href="/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/more_info.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-where-can-i-read-more-about-this-analysis-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Where can I read more about this analysis?</div>   <p>This tutorial was adapted from the <a href="https://mothur.org/wiki/miseq_sop/">mothur MiSeq SOP</a> created by the Schloss lab. Here you can find more information about the mothur tools and file formats. Their <a href="https://mothur.org/wiki/frequently_asked_questions/">FAQ page</a> and <a href="https://forum.mothur.org/">Help Forum</a> are also quite useful!</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="galaxy"> <a href="#galaxy"> Galaxy </a></h3>

<hr/>



<h4 class="faq-area"> How can I get my container requiring jobs to run in a container? <a href="/training-material/topics/admin/faqs/galaxy-jobs-in-containers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-get-my-container-requiring-jobs-to-run-in-a-container"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-get-my-container-requiring-jobs-to-run-in-a-container" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I get my container requiring jobs to run in a container?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some tools will only run in a container, i.e. they have a container defined in the ‘requirements’ section of the tool’s XML file. Galaxy will not refuse to run these tools if the container isn’t available or if Galaxy isn’t configured use containers. Instead it’ll run in the host system and likely fail.</p>   <h2 id="job-configuration">Job Configuration</h2>   <p>You can resolve this by configuring your job conf to have destinations that support containers (or even require them.):</p>   <p>The destination must have <code class="language-plaintext highlighter-rouge">docker_enabled</code> (Or <code class="language-plaintext highlighter-rouge">singularity_enabled</code>), and you can consider adding <code class="language-plaintext highlighter-rouge">require_container</code> to make sure the job will fail if the container isn’t available. The <code class="language-plaintext highlighter-rouge">docker_volumes</code> string will allow you to control which volumes are attached to that container;</p>   <p>In TPV configuration (provided by @gtn:thanhleviet) this would look like:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  <span class="na">docker</span><span class="pi">:</span><br>    <span class="na">inherits</span><span class="pi">:</span> <span class="s">slurm</span><br>    <span class="na">scheduling</span><span class="pi">:</span><br>      <span class="na">require</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">docker</span><br>    <span class="na">params</span><span class="pi">:</span><br>      <span class="na">docker_enabled</span><span class="pi">:</span> <span class="kc">true</span><br>      <span class="na">require_container</span><span class="pi">:</span> <span class="kc">true</span><br><br>  <span class="na">singularity</span><span class="pi">:</span><br>    <span class="na">inherits</span><span class="pi">:</span> <span class="s">slurm</span><br>    <span class="na">scheduling</span><span class="pi">:</span><br>      <span class="na">require</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">singularity</span><br>    <span class="na">params</span><span class="pi">:</span><br>      <span class="na">singularity_enabled</span><span class="pi">:</span> <span class="kc">true</span><br><br>  <span class="na">podman</span><span class="pi">:</span><br>    <span class="na">inherits</span><span class="pi">:</span> <span class="s">slurm</span><br>    <span class="na">scheduling</span><span class="pi">:</span><br>      <span class="na">require</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">podman</span><br>    <span class="na">params</span><span class="pi">:</span><br>      <span class="na">docker_enabled</span><span class="pi">:</span> <span class="kc">true</span><br>      <span class="na">require_container</span><span class="pi">:</span> <span class="kc">true</span><br>      <span class="na">docker_volumes</span><span class="pi">:</span> <span class="s2">"</span><span class="s">$galaxy_root:ro,$tool_directory:ro,$job_directory:ro,$working_directory:z,$default_file_path:z"</span><br>      <span class="na">docker_sudo</span><span class="pi">:</span> <span class="kc">false</span><br>      <span class="na">docker_cmd</span><span class="pi">:</span> <span class="s">/usr/bin/podman</span><br>      <span class="na">docker_run_extra_arguments</span><span class="pi">:</span> <span class="s2">"</span><span class="s">--userns=keep-id"</span><br></code></pre></div></div>   <p>Or in XML:</p>   <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;destination</span> <span class="na">id=</span><span class="s">"docker"</span> <span class="na">runner=</span><span class="s">"local"</span><span class="nt">&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"docker_enabled"</span><span class="nt">&gt;</span>true<span class="nt">&lt;/param&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"require_container"</span><span class="nt">&gt;</span>true<span class="nt">&lt;/param&gt;</span><br><span class="nt">&lt;/destionation&gt;</span><br><br><span class="nt">&lt;destination</span> <span class="na">id=</span><span class="s">"podman"</span> <span class="na">runner=</span><span class="s">"local"</span><span class="nt">&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"docker_enabled"</span><span class="nt">&gt;</span>true<span class="nt">&lt;/param&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"require_container"</span><span class="nt">&gt;</span>true<span class="nt">&lt;/param&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"docker_sudo"</span><span class="nt">&gt;</span>false<span class="nt">&lt;/param&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"docker_cmd"</span><span class="nt">&gt;</span>/usr/bin/podman<span class="nt">&lt;/param&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"docker_run_extra_arguments"</span><span class="nt">&gt;</span>--userns=keep-id<span class="nt">&lt;/param&gt;</span><br>    <span class="c">&lt;!-- This will not work until https://github.com/galaxyproject/galaxy/pull/18998 is merged for SELinux users. For now you may want to patch it manually. --&gt;</span><br>    <span class="c">&lt;!-- &lt;param id="docker_volumes"&gt;$galaxy_root:ro,$tool_directory:ro,$job_directory:ro,$working_directory:z,$default_file_path:z&lt;/param&gt; --&gt;</span><br><span class="nt">&lt;/destination&gt;</span><br><br><span class="nt">&lt;destination</span> <span class="na">id=</span><span class="s">"singularity"</span> <span class="na">runner=</span><span class="s">"local"</span><span class="nt">&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"singularity_enabled"</span><span class="nt">&gt;</span>true<span class="nt">&lt;/param&gt;</span><br>    <span class="nt">&lt;param</span> <span class="na">id=</span><span class="s">"require_container"</span><span class="nt">&gt;</span>true<span class="nt">&lt;/param&gt;</span><br><span class="nt">&lt;/destionation&gt;</span><br></code></pre></div></div>   <p>Configuring a tool to use this destination would look like:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">toolshed.g2.bx.psu.edu/repos/thanhlv/metaphlan4/metaphlan4/4.0.3</span><span class="pi">:</span><br>    <span class="na">cores</span><span class="pi">:</span> <span class="m">12</span><br>    <span class="na">mem</span><span class="pi">:</span> <span class="s">cores * </span><span class="m">8</span><br>    <span class="na">params</span><span class="pi">:</span><br>      <span class="na">singularity_enabled</span><span class="pi">:</span> <span class="kc">true</span><br></code></pre></div></div>   <p>Or in XML:</p>   <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;tools&gt;</span><br>    <span class="nt">&lt;tool</span> <span class="na">id=</span><span class="s">"toolshed.g2.bx.psu.edu/repos/thanhlv/metaphlan4/metaphlan4/4.0.3"</span> <span class="na">destination=</span><span class="s">"docker"</span><span class="nt">/&gt;</span><br><span class="nt">&lt;/tools&gt;</span><br></code></pre></div></div>   <h2 id="container-resolvers-configuration">Container Resolvers Configuration</h2>   <p>If you’re using the default <code class="language-plaintext highlighter-rouge">container_resolvers_conf.yml</code> then there is nothing you need to do. Otherwise you may want to ensure that you have items in there such as <code class="language-plaintext highlighter-rouge">explicit</code> and <code class="language-plaintext highlighter-rouge">explicit_singularity</code> among others. See the <a href="https://docs.galaxyproject.org/en/master/admin/container_resolvers.html">galaxy documentation</a> on the topic.</p>   <h2 id="testing">Testing</h2>   <p>Here is an example of a tool that requires a container, that you can use to test your container configuration:</p>   <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;tool</span> <span class="na">name=</span><span class="s">"container-test"</span> <span class="na">id=</span><span class="s">"container"</span> <span class="na">version=</span><span class="s">"5.0"</span> <span class="na">profile=</span><span class="s">"21.09"</span><span class="nt">&gt;</span><br>        <span class="nt">&lt;requirements&gt;</span><br>                <span class="nt">&lt;container</span> <span class="na">type=</span><span class="s">"docker"</span><span class="nt">&gt;</span>ubuntu:22.04<span class="nt">&lt;/container&gt;</span><br>        <span class="nt">&lt;/requirements&gt;</span><br>        <span class="nt">&lt;command&gt;</span><span class="cp">&lt;![CDATA[<br>                pwd &gt;</span>&gt; '$output';<br>                hostname -f &gt;&gt; '$output';<br>                echo "" &gt;&gt; '$output';<br>                cat /etc/os-release &gt;&gt; '$output';<br>                echo "" &gt;&gt; '$output';<br>                env | sort &gt;&gt; '$output';<br>]]&gt;<span class="nt">&lt;/command&gt;</span><br>        <span class="nt">&lt;inputs&gt;</span><br>        <span class="nt">&lt;/inputs&gt;</span><br>        <span class="nt">&lt;outputs&gt;</span><br>                <span class="nt">&lt;data</span> <span class="na">name=</span><span class="s">"output"</span> <span class="na">format=</span><span class="s">"txt"</span> <span class="na">label=</span><span class="s">"log"</span> <span class="nt">/&gt;</span><br>        <span class="nt">&lt;/outputs&gt;</span><br>        <span class="nt">&lt;help&gt;</span><span class="cp">&lt;![CDATA[]]&gt;</span><span class="nt">&lt;/help&gt;</span><br><span class="nt">&lt;/tool&gt;</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How many mules? <a href="/training-material/topics/admin/faqs/galaxy-how-many-mules.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-many-mules"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-many-mules" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How many mules?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Start with 2 and add more as needed. If you notice that your jobs seem to inexplicably sit for a long time before being dispatched to the cluster, or after they have finished on the cluster, you may need additional handlers.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="galaxy admin interface"> <a href="#galaxy admin interface"> Galaxy admin interface </a></h3>

<hr/>



<h4 class="faq-area"> Install tools via the Admin UI <a href="/training-material/topics/admin/faqs/install_tool.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-install-tools-via-the-admin-ui"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-install-tools-via-the-admin-ui" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Install tools via the Admin UI</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open Galaxy in your browser and type `` in the tool search box on the left. If “” is among the search results, you can skip the following steps.</li>   <li>Access the Admin menu from the top bar (you need to be logged-in with an email specified in the <code class="language-plaintext highlighter-rouge">admin_users</code> setting)</li>   <li>Click “Install and Uninstall”, which can be found on the left, under “Tool Management”</li>   <li>Enter `` in the search interface</li>   <li>Click on the first hit, having <code class="language-plaintext highlighter-rouge">devteam</code> as owner</li>   <li>Click the “Install” button for the latest revision</li>   <li>Enter “” as the target section and click “OK”.</li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="gat"> <a href="#gat"> Gat </a></h3>

<hr/>



<h4 class="faq-area"> Time to git commit <a href="/training-material/topics/admin/faqs/git-commit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-time-to-git-commit"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Time to git commit</div>   <p>It’s time to commit your work! Check the status with</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git status<br></code></pre></div></div>   <p>Add your changed files with</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git add ... # any files you see that are changed<br></code></pre></div></div>   <p>And then commit it!</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>git commit -m 'Finished '<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using Git With Ansible Vaults <a href="/training-material/topics/admin/faqs/git-attributes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-using-git-with-ansible-vaults"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Using Git With Ansible Vaults</div>   <p>When looking at <code class="language-plaintext highlighter-rouge">git log</code> to see what you changed, you cannot easily look into Ansible Vault changes: you just see the changes in the encrypted versions which is unpleasant to read.</p>   <p>Instead we can use <a href="https://www.git-scm.com/docs/gitattributes"><code class="language-plaintext highlighter-rouge">.gitattributes</code></a> to tell <code class="language-plaintext highlighter-rouge">git</code> that we want to use a different program to visualise differences between two versions of a file, namely <code class="language-plaintext highlighter-rouge">ansible-vault</code>.</p>   <ol>   <li>Check your <code class="language-plaintext highlighter-rouge">git log -p</code> and see how the Vault changes look (you can type <code class="language-plaintext highlighter-rouge">/vault</code> to search). Notice that they’re just changed encoded content.</li>   <li>     <p>Create the file <code class="language-plaintext highlighter-rouge">.gitattributes</code> in the same folder as your <code class="language-plaintext highlighter-rouge">galaxy.yml</code> playbook, with the following contents:</p>     <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>group_vars/secret.yml diff=ansible-vault merge=binary<br></code></pre></div>    </div>   </li>   <li>Try again to <code class="language-plaintext highlighter-rouge">git log -p</code> and look for the vault changes. Note that you can now see the decrypted content! Very useful.</li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="github"> <a href="#github"> Github </a></h3>

<hr/>



<h4 class="faq-area"> Forking the GTN repository <a href="/training-material/topics/contributing/faqs/github-fork-gtn.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-forking-the-gtn-repository"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-forking-the-gtn-repository" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Forking the GTN repository</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Go on the GitHub repository: <a href="https://github.com/galaxyproject/training-material" width="50%">github.com/galaxyproject/training-material</a></li>   <li>Click on the <strong>Fork</strong> button (top-right corner of the page)</li> </ul>   <p><img src="/training-material/topics/contributing/images/PR_fork.jpg" alt="The fork button on GitHub" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Syncing your Fork of the GTN <a href="/training-material/topics/contributing/faqs/github-fork-sync.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-syncing-your-fork-of-the-gtn"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-syncing-your-fork-of-the-gtn" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Syncing your Fork of the GTN</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Whenever you want to contribute something new to the GTN, it is important to start with an up-to-date branch. To do this, you should always update the main branch of your fork, before creating a so-called <em>feature branch</em>, a branch where you make your changes.</p>   <ol>   <li>Point your browser to your fork of the GTN repository     <ul>       <li>The url will be <code class="language-plaintext highlighter-rouge">https://github.com/&lt;your username&gt;/training_material</code> (replacing ‘your username’ with your GitHub username)</li>     </ul>   </li>   <li>     <p>You might see a message like “This branch is 367 commits behind galaxyproject/training-material:main.” as in the screenshot below.</p>     <p><img src="/training-material/topics/contributing/images/github_update_fork.png" alt="Github with the top bar of a repository shown, including the button for 'Sync Fork'" /></p>   </li>   <li>     <p>Click the <strong>Sync Fork</strong> button on your fork to update it to the latest version.</p>   </li>   <li><strong>TIP:</strong> never work directly on your main branch, since that will make the sync process more difficult. Always create a new branch before committing your changes.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Updating the default branch from master to main <a href="/training-material/topics/contributing/faqs/github-fork-master-main.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-updating-the-default-branch-from-master-to-main"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-updating-the-default-branch-from-master-to-main" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Updating the default branch from master to main</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you created your fork a long time ago, the default branch on your fork may still be called <strong>master</strong> instead of <strong>main</strong></p>   <ol>   <li>Point your browser to your fork of the GTN repository     <ul>       <li>The url will be <code class="language-plaintext highlighter-rouge">https://github.com/&lt;your username&gt;/training_material</code> (replacing <your username=""> with your GitHub username)</your></li>     </ul>   </li>   <li>     <p>Check the default branch that is shown (at top left).</p>     <p><img src="/training-material/topics/contributing/images/github_update_fork.png" alt="Github with the top bar of a repository shown, including the button for 'Sync Fork'" /></p>   </li>   <li>Does it say <code class="language-plaintext highlighter-rouge">main</code>?     <ul>       <li>Congrats, nothing to do, <strong>you can skip the rest of these steps</strong></li>     </ul>   </li>   <li>     <p>Does it say <code class="language-plaintext highlighter-rouge">master</code>? Then you need to update it, following the instructions below</p>   </li>   <li>Go to your fork’s settings (Click on the gear icon called “Settings”)</li>   <li>Find “Branches” on the left</li>   <li>If it says master you can click on the ⇆  icon to switch branches.</li>   <li>Select <code class="language-plaintext highlighter-rouge">main</code> (it may not be present).</li>   <li>If it isn’t present, use the pencil icon to rename <code class="language-plaintext highlighter-rouge">master</code> to <code class="language-plaintext highlighter-rouge">main</code>.</li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="gtn"> <a href="#gtn"> Gtn </a></h3>

<hr/>



<h4 class="faq-area"> Annotating Pre-requisites <a href="/training-material/topics/contributing/faqs/why-prerequisites.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-annotating-pre-requisites"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-annotating-pre-requisites" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Annotating Pre-requisites</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you are adding a tutorial, annotating the pre-requisites is an important task! It will help ensure learners know what they need to know before starting the tutorial. They also let instructors plan a schedule optimally.</p>   <p>Internal requirements often include specific features of Galaxy you plan to use in your training material, and let learners know which tutorials to follow first, before starting your tutorial.</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">requirements</span><span class="pi">:</span><br>  <span class="pi">-</span> <span class="na">type</span><span class="pi">:</span> <span class="s2">"</span><span class="s">internal"</span><br>    <span class="na">topic_name</span><span class="pi">:</span> <span class="s">galaxy-interface</span><br>    <span class="na">tutorials</span><span class="pi">:</span><br>      <span class="pi">-</span> <span class="s">collections</span><br>      <span class="pi">-</span> <span class="s">upload-rules</span><br></code></pre></div></div>   <p>Or you can have external requirements, which link to another site.</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">requirements</span><span class="pi">:</span><br>  <span class="pi">-</span><br>    <span class="na">type</span><span class="pi">:</span> <span class="s2">"</span><span class="s">external"</span><br>    <span class="na">title</span><span class="pi">:</span> <span class="s2">"</span><span class="s">Trackster"</span><br>    <span class="na">link</span><span class="pi">:</span> <span class="s2">"</span><span class="s">https://wiki.galaxyproject.org/Learn/Visualization"</span><br></code></pre></div></div>   <p>Least commonly needed are software requirements. These are usually used in e.g. Galaxy Admin Training tutorials, but if you have specific software requirements, you can list them here:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">requirements</span><span class="pi">:</span><br><span class="pi">-</span> <span class="na">type</span><span class="pi">:</span> <span class="s">none</span><br>  <span class="na">title</span><span class="pi">:</span> <span class="s2">"</span><span class="s">Web</span><span class="nv"> </span><span class="s">browser"</span><br><span class="pi">-</span> <span class="na">type</span><span class="pi">:</span> <span class="s">none</span><br>  <span class="na">title</span><span class="pi">:</span> <span class="s2">"</span><span class="s">A</span><span class="nv"> </span><span class="s">linux-based</span><span class="nv"> </span><span class="s">machine</span><span class="nv"> </span><span class="s">or</span><span class="nv"> </span><span class="s">linux</span><span class="nv"> </span><span class="s">emulator"</span><br><span class="pi">-</span> <span class="na">type</span><span class="pi">:</span> <span class="s">none</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Input Histories & Answer Keys <a href="/training-material/topics/contributing/faqs/input-and-answer-keys.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-input-histories-answer-keys"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-input-histories-answer-keys" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Input Histories &amp; Answer Keys</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tutorials sometimes require significant amounts of data or data prepared in a very specific manner which often is shown to cause errors for learners that significantly affect downstream results. Input histories are an answer to that:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">input_histories</span><span class="pi">:</span><br>  <span class="pi">-</span> <span class="na">label</span><span class="pi">:</span> <span class="s2">"</span><span class="s">UseGalaxy.eu"</span><br>    <span class="na">history</span><span class="pi">:</span> <span class="s">https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/cs1pre-processing-with-alevin---input-1</span><br>    <span class="na">date</span><span class="pi">:</span> <span class="s2">"</span><span class="s">2021-09-01"</span><br></code></pre></div></div>   <p>Additionally once the learner has gotten started, tutorials sometimes feature tools which produce stochastic outputs, or have very long-running steps. In these cases, the tutorial authors may provide answer histories to help learners verify that they are on the right track, or to enable them to catch up if they fall behind or something goes wrong.</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">answer_histories</span><span class="pi">:</span><br>    <span class="pi">-</span> <span class="na">label</span><span class="pi">:</span> <span class="s2">"</span><span class="s">UseGalaxy.eu"</span><br>    <span class="na">history</span><span class="pi">:</span> <span class="s">https://humancellatlas.usegalaxy.eu/u/j.jakiela/h/generating-a-single-cell-matrix-using-alevin-3</span><br>    <span class="pi">-</span> <span class="na">label</span><span class="pi">:</span> <span class="s2">"</span><span class="s">Older</span><span class="nv"> </span><span class="s">Alevin</span><span class="nv"> </span><span class="s">version"</span><br>    <span class="na">history</span><span class="pi">:</span> <span class="s">https://humancellatlas.usegalaxy.eu/u/wendi.bacon.training/h/cs1pre-processing-with-alevin---answer-key</span><br>    <span class="na">date</span><span class="pi">:</span> <span class="s">2024-01-01</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using the new Contributions Annotation framework <a href="/training-material/topics/contributing/faqs/new-contributions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-the-new-contributions-annotation-framework"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-the-new-contributions-annotation-framework" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using the new Contributions Annotation framework</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you are writing a tutorial or slides, there are two ways to annotate contributions:</p>   <p>The old way, which doesn’t accurately track roles</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>contributors: [hexylena, shiltemann]<br></code></pre></div></div>   <p>And the new way which lets you annotate who has helped build a tutorial in a much richer way:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">contributions</span><span class="pi">:</span><br>    <span class="na">authorship</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">shiltemann</span><br>        <span class="pi">-</span> <span class="s">bebatut</span><br>    <span class="na">editing</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">hexylena</span><br>        <span class="pi">-</span> <span class="s">bebatut</span><br>        <span class="pi">-</span> <span class="s">natefoo</span><br>    <span class="na">testing</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">bebatut</span><br>    <span class="na">infrastructure</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">natefoo</span><br>    <span class="na">translation</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">shiltemann</span><br>    <span class="na">funding</span><span class="pi">:</span><br>        <span class="pi">-</span> <span class="s">gallantries</span><br></code></pre></div></div>   <p>This is especially important if you want to track funding or infrastructure contributions. The old way doesn’t allow for this, and thus we would <em>strongly</em> recommend you use the new format!</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="historias"> <a href="#historias"> Historias </a></h3>

<hr/>



<h4 class="faq-area"> Cambiando el nombre de un historial <a href="/training-material/faqs/galaxy-es/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-cambiando-el-nombre-de-un-historial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-cambiando-el-nombre-de-un-historial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Cambiando el nombre de un historial</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Haz clic sobre  <strong>Unnamed history</strong> (o el nombre que tenga el historial sobre el que estás trabajando) (<strong>Haz clic para cambiar el nombre del historial</strong>) en la parte superior de tu panel de historial</li>   <li>Escribe el nombre nuevo</li>   <li>Pulsa <kbd>Enter</kbd></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Para la creación de un historial nuevo <a href="/training-material/faqs/galaxy-es/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Los historiales son una parte importante de Galaxy, la mayoría de la gente utiliza un historial para cada análisis nuevo. Asegúrate siempre de darle buenos nombres a tus historiales, de tal forma que después puedas encontrar fácilmente tus resultados.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-para-la-creación-de-un-historial-nuevo"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-para-la-creación-de-un-historial-nuevo" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Para la creación de un historial nuevo</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Haz click sobre el icono <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> en la parte superior del panel de historiales.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="histories"> <a href="#histories"> Histories </a></h3>

<hr/>



<h4 class="faq-area"> Compartiendo un historial <a href="/training-material/faqs/galaxy-es/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Puedes compartir tu trabajo en Galaxy. Hay varias formas de dar acceso a tus historiales a otros usuarios.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-compartiendo-un-historial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-compartiendo-un-historial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Compartiendo un historial</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Compartir tu historial permite a otros importar y acceder a los conjuntos de datos, parámetros y pasos de tu historial.</p>   <ol>   <li><strong>Compartir a través de un enlace</strong>     <ul>       <li>Abre el menú <strong>Opciones de historial</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (icono de engranaje) en la parte superior del panel de historial         <ul>           <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Hacer que el historial sea accesible</strong></li>           <li>Aparecerá un <strong>Compartir enlace</strong> que puedes dar a otros usuarios.</li>         </ul>       </li>       <li>Cualquiera que tenga este enlace puede ver y copiar tu historial.</li>     </ul>   </li>   <li><strong>Publica tu historial</strong>     <ul>       <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Hacer que el historial esté disponible públicamente en Historias publicadas</strong></li>       <li>Cualquiera en este servidor Galaxy podrá ver tu historial en el menú <em>Datos compartidos</em></li>     </ul>   </li>   <li><strong>Comparte solo con otro usuario</strong>.     <ul>       <li>Haz clic en el botón <strong>Compartir con un usuario</strong> en la parte inferior</li>       <li>Ingresa una dirección de correo electrónico del usuario con el que deseas compartir</li>       <li>Tu historial se compartirá solo con este usuario.</li>     </ul>   </li>   <li><strong>Encontrar historiales que otros han compartido conmigo</strong>     <ul>       <li>Haz clic en el menú <strong>Usuario</strong> en la barra superior</li>       <li>Selecciona <strong>Historiales compartidos conmigo</strong></li>       <li>Aquí verás todos los historiales que otros han compartido contigo directamente ** Nota: ** Si deseas realizar cambios en tu historial sin afectar la versión compartida, crea una copia mediante al ícono <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <em>Opciones de historial</em> en tu historial y haciendo clic en <em>Copiar</em></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Copy a dataset between histories <a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Sometimes you may want to use a dataset in multiple histories. You do not need to re-upload the data, but you can copy datasets from one history to another.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-copy-a-dataset-between-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-copy-a-dataset-between-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Copy a dataset between histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There 3 ways to copy datasets between histories</p>   <ol>   <li>     <p>From the original history</p>     <ol>       <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon which is on the top of the list of datasets in the history panel</li>       <li>Click on <strong>Copy Datasets</strong></li>       <li>         <p>Select the desired files</p>       </li>       <li>         <p>Give a relevant name to the “New history”</p>       </li>       <li>Validate by ‘Copy History Items’</li>       <li>Click on the new history name in the green box that have just appear to switch to this history</li>     </ol>   </li>   <li>     <p>Using the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>Show Histories Side-by-Side</strong></p>     <ol>       <li>Click on the <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> dropdown arrow top right of the history panel (<strong>History options</strong>)</li>       <li>Click on <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>Show Histories Side-by-Side</strong></li>       <li>If your target history is not present         <ol>           <li>Click on ‘Select histories’</li>           <li>Click on your target history</li>           <li>Validate by ‘Change Selected’</li>         </ol>       </li>       <li>Drag the dataset to copy from its original history</li>       <li>Drop it in the target history</li>     </ol>   </li>   <li>     <p>From the target history</p>     <ol>       <li>Click on <strong>User</strong> in the top bar</li>       <li>Click on <strong>Datasets</strong></li>       <li>Search for the dataset to copy</li>       <li>Click on its name</li>       <li>Click on <strong>Copy to current History</strong></li>     </ol>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a new history <a href="/training-material/faqs/galaxy/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Histories are an important part of Galaxy, most people use a new history for every new analysis. Always make sure to give your histories good names, so you can easily find your results back later.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Créer un nouvel history <a href="/training-material/faqs/galaxy-fr/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Les historiques sont une partie importante de Galaxy, la plupart des gens utilisent un nouvel historique pour chaque nouvelle analyse. Assurez-vous toujours de donner de bons noms à vos historiques, afin de pouvoir retrouver facilement vos résultats plus tard.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-créer-un-nouvel-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-créer-un-nouvel-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Créer un nouvel history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Cliquez sur l’icone <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> en haut du panneau d’historique.</p>   <p>Si l’icone <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> est manquant :</p>   <ol>   <li>Cliquez sur l’icone <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (<strong>Options d’historique</strong>) en haut du panneau d’historique</li>   <li>Selectionner l’option <strong>Créer un nouveau</strong> depuis le menu</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Dataset colors <a href="/training-material/faqs/galaxy/histories_dataset_colors.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains meaning of dataset colors in Galaxy's history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-colors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-colors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset colors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are several different “states” a dataset can be in. These states are indicated by colors:</p>   <p><img src="/training-material/shared/images/galactic_colors.svg" alt="Colors indicating states of Galaxy datasets" /></p>   <ul>   <li><strong>ok</strong>: everything is fine, life is good;</li>   <li><strong>new</strong>: the dataset was just created. Galaxy does not yet know when it is;</li>   <li><strong>queued</strong>: indicates that the job generating this dataset is scheduled for execution but not running yet;</li>   <li><strong>running</strong>: job generating this dataset is running;</li>   <li><strong>setting metadata</strong>: when a new dataset is uploaded Galaxy examines it to understand what kind of data it is (e.g., BAM, FASTQ, fasta, BED, etc.). This is called “setting metadata”;</li>   <li><strong>deferred</strong>: sometimes it does not make sense to upload the dataset until it is needed for an analysis. Galaxy will download <strong>deferred</strong> datasets later during the job execution. Those datasets do not count toward your quota;</li>   <li><strong>paused</strong>: in some cases as, for example, workflow executions, upstream errors prevent subsequent jobs from starting creating datasets in “paused” state;</li>   <li><strong>discarded</strong>: something went wrong such as, for example, a job producing this dataset might have been cancelled;</li>   <li><strong>error</strong>: everything is not fine; life is bad!</li>   <li><strong>placeholder</strong>: similar to “new”; we know something will be there but are not yet sure what;</li>   <li><strong>failed populated state</strong>: this refers to collections (not individual datasets). Here a collection has failed to be populated with datasets;</li>   <li><strong>new populated state</strong>: this refers to collections (not individual datasets). A collection was created but not populated yet.</li> </ul>   <!-- original editable image = https://docs.google.com/drawings/d/1F2Lq1m3cMIckvCexXMzug-dqwgifkoMOzoyH4VcoVX0/edit?usp=sharing -->   <!-- TO DO  Needs to be linked to FAQs on: - how to report errors - explaining collections  --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Dataset snippet <a href="/training-material/faqs/galaxy/histories_dataset_item.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Describes features of a single dataset element in the history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-snippet"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-snippet" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset snippet</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A single Galaxy dataset can either be “collapsed” or “expanded”.</p>   <p><strong>Collapsed dataset view</strong></p>   <p>Datasets in the panel are initially shown in a “collapsed” view:</p>   <p><img src="/training-material/shared/images/history_item_collapsed.png" alt="Collapsed view of a single Galaxy dataset" /></p>   <p>It contains the following elements:</p>   <ul>   <li><strong>Dataset number</strong>: (“1”) order of dataset in the history;</li>   <li><strong>Dataset name</strong>: (“M117-bl_1.fq.gz”) its name;</li>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>: click this to view the dataset contents;</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span>: click this to edit dataset properties;</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span>: click this to delete the dataset from the history (<em>don’t worry</em>, you can undo this action!).</li> </ul>   <p>Clicking on a collapsed dataset will expand it.</p>   <blockquote class="details">   <div class="box-title details-title" id="details-some-buttons-can-be-disabled"><button class="gtn-boxify-button details" type="button" aria-controls="details-some-buttons-can-be-disabled" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Some buttons can be disabled.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some of the buttons above may be disabled if the dataset is in a state that doesn’t allow the action. For example, the ‘edit’ button is disabled for datasets that are still queued or running</p> </blockquote>   <p><strong>Expanded dataset view</strong></p>   <p>Expanded dataset view adds a preview element and many additional controls.</p>   <p><img src="/training-material/shared/images/history_item_expanded.png" alt="Expanded view of a single Galaxy dataset" /></p>   <p>In addition to the elements described above for the collapsed dataset, its expanded view contains:</p>   <ul>   <li><strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>: click on this to tag this dateset;</li>   <li><strong>Dataset size</strong>: (“2 variants, 18 comments”) lists the size of the dataset. When datasets are small (like in this example) the exact size is shown. For large datasets, Galaxy gives an approximate estimate.</li>   <li><strong>format</strong>: (“VCF”) lists the datatype;</li>   <li><strong>database</strong>: (“?”) lists which genome built this dataset corresponds to. This usually lists “?” unless the genome build is set explicitly or the dataset is derived from another dataset with defined genome build information;</li>   <li><strong>info field</strong>: (“INFO [2024-03-26 12:08:53,435]…”) displays information provided by the tool that generated this dataset. This varies widely and depends on the type of job that generated this dataset.</li>   <li><i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span>: Saves dataset to disk;</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">dataset-link</span>: Copies dataset link into clipboard;</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">dataset-info</span>: Displays additional details about the dataset in the center pane;</li>   <li><i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">dataset-rerun</span>: Reruns job that generated this dataset. This button is unavailable for datasets uploaded into history because they were not produced by a Galaxy tool;</li>   <li><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">dataset-visualize</span>: Displays visualization options for this dataset. The list of options is dependent on the datatype;</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">dataset-related-datasets</span>: Shows datasets related to this dataset. This is useful for tracking down parental datasets - those that were used as inputs into a job that produced this particular dataset.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading histories <a href="/training-material/faqs/galaxy/histories_download_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the gear icon <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> on the top of the history panel.</li>   <li>Select “Export History to File” from the History menu.</li>   <li>Click on the “Click here to generate a new archive for this history” text.</li>   <li>Wait for the Galaxy server to prepare history for download.</li>   <li>Click on the generated link to download the history.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Find all Histories and purge (aka permanently delete) <a href="/training-material/faqs/galaxy/histories_find_purge.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-find-all-histories-and-purge-aka-permanently-delete"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-find-all-histories-and-purge-aka-permanently-delete" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Find all Histories and purge (aka permanently delete)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Login to your Galaxy account.</li>   <li>On the top navigation bar Click on <strong>User</strong>.</li>   <li>On the drop down menu that appears Click on <strong>Histories</strong>.</li>   <li>Click on <strong>Advanced Search</strong>, additional fields will be displayed.</li>   <li>Next to the <strong>Status</strong> field, click <strong>All</strong>, a list of all histories will be displayed.</li>   <li>Check the box next to <strong>Name</strong> in the displayed list to select all histories.</li>   <li>Click <strong>Delete Permanently</strong> to purge all histories.</li>   <li>A pop up dialogue box will appear letting you know history contents will be removed and cannot be undone, then click <strong>OK</strong> to confirm.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding Histories <a href="/training-material/faqs/galaxy/histories_finding_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding Histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>To review all histories in your account, go to <strong>User &gt; Histories</strong> in the top menu bar.</li>   <li>At the top of the History listing, click on <strong>Advanced Search</strong>.</li>   <li>Set the status to all to view all of your active, deleted, and permanently deleted (purged) histories.</li>   <li>Histories in all states are listed for registered accounts. Meaning one will always find their data here if it ever appears to be “lost”.</li>   <li>Note: Permanently deleted (purged) Histories may be fully removed from the server at any time. The data content inside the History is always removed at the time of purging (by a double-confirmed user action), but the purged History artifact may still be in the listing. Purged data content cannot be restored, even by an administrator.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding and working with "Histories shared with me" <a href="/training-material/faqs/galaxy/histories_shared.html"><i class="fas fa-external-link-alt"></i></a> </h4>
How to find and work on histories shared with you
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-and-working-with-quot-histories-shared-with-me-quot"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-and-working-with-quot-histories-shared-with-me-quot" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding and working with &quot;Histories shared with me&quot;</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>To find histories shared with me:</strong></p>   <ol>   <li>Log into your account.</li>   <li>Select <strong>User</strong>, in the drop-down menu, select <strong>Histories shared with me</strong>.</li> </ol>   <p><strong>To work with shared histories:</strong></p>   <ul>   <li>Import the History into your account via copying it to work with it.</li>   <li>Unshare Histories that you no longer want shared with you or that you have already made a copy of.</li> </ul>   <p><strong>Note:</strong> Shared Histories (when copied into your account or not) do count in portion toward your total account data quota usage. More details on histories shared concerning account quota usage can be found in <a href="/training-material/faqs/galaxy/histories_unsharing_histories.html">this link</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> History annotation <a href="/training-material/faqs/galaxy/histories_annotation.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to annotate a history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-history-annotation"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-history-annotation" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: History annotation</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes tags and names are not enough to describe the work done within a history. Galaxy allows you to create history annotations: longer text entries that allow for more formatting options. The formatting of the text is preserved. Later, if you publish or share the history, the annotation will be displayed automatically - allowing you to share additional notes about the analysis. Multiple lines, spaces, and emoji! 😹🏳️‍⚧️🌈 can be used while writing annotations.</p>   <p>To annotate a history:</p>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name. A larger text section will appear displaying any   existing annotation or <code class="language-plaintext highlighter-rouge">Annotation (optional)</code> if empty.</li>   <li>Add your text. <kbd>Enter</kbd> will move the cursor to the next line. (Tabs cannot be   entered since the ‘Tab’ button is used to switch between controls on the page - tabs can be pasted in, however).</li>   <li>Click on <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>.</li>   <li>To cancel, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button.</li> </ol>   <p><img src="/training-material/shared/images/history_annotations.png" alt="UI for annotating histories" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> History options <a href="/training-material/faqs/galaxy/histories_options.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains different history options
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-history-options"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-history-options" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: History options</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Clicking the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> button will open a drop-down menu with several options:</p>   <ul>   <li><strong>Show histories side-by-side</strong> - brings up a view in which multiple histories can be viewed and manipulated simultaneously. Datasets can be dragged between histories in this view.</li>   <li><strong>Resume Paused Jobs</strong> - restarts paused jobs in history.</li>   <li><strong>Copy this history</strong> - creates an exact copy of the current history in the current account.</li>   <li><strong>Delete this history</strong> - deletes the current history.</li>   <li><strong>Export tool citations</strong> - export citations for tools that were used in the current history.</li>   <li><strong>Export history to File</strong> - creates a compressed archive containing data from the current history.</li>   <li><strong>Archive history</strong> - moves history to a non-active, archived, state.</li>   <li><strong>Extract workflow</strong> - converts the current history into a workflow</li>   <li><strong>Show invocations</strong> - shows a list of all workflows that were run in the current history</li>   <li><strong>Share or Publish</strong> - allows controlling access to history. It can be made public or shared with a specific user.</li>   <li><strong>Set Permissions</strong> - allows to set the rules on who can access daysets in the current history.</li>   <li><strong>Make Private</strong> - resets all permission and makes the current history private.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> History tagging <a href="/training-material/faqs/galaxy/histories_tagging.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to add tags to a history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-history-tagging"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-history-tagging" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: History tagging</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tags are short pieces of text used to describe the thing they’re attached to and many things in Galaxy can be tagged. Each item can have many tags and you can add new tags or remove them at any time. Tags can be another useful way to organize and search your data. For instance, you might tag a history with the type of analysis you did in it: <code class="language-plaintext highlighter-rouge">assembly</code> or <code class="language-plaintext highlighter-rouge">variants</code>. Or you may tag them according to data sources or some other metadata: <code class="language-plaintext highlighter-rouge">long-term-care-facility</code> or <code class="language-plaintext highlighter-rouge">yellowstone-park:2014</code>.</p>   <p><strong>To tag a history:</strong></p>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”).</li>   <li>Click on <strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> and start typing. Any tags that you’ve used previously will show below your partial entry -   allowing you to use this ‘autocomplete’ data to re-use your previous tags without typing them in full.</li>   <li>Click on <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>.</li>   <li>To cancel, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button.</li> </ol>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-do-not-use-spaces"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Do not use spaces</div>   <p>It is strongly recommended to replace spaces in tags with <code class="language-plaintext highlighter-rouge">_</code> or <code class="language-plaintext highlighter-rouge">-</code>, as spaces will automatically be removed when the tag is saved.</p> </blockquote>   <p><img src="/training-material/shared/images/history_tags.png" alt="UI for tagging histories" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to set Data Privacy Features? <a href="/training-material/faqs/galaxy/datasets_data_privacy_features.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-set-data-privacy-features"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-set-data-privacy-features" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to set Data Privacy Features?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Privacy controls are only enabled if desired. Otherwise, datasets by defaults remain private and unlisted in Galaxy. This means that a dataset you’ve created is virtually invisible until you publish a link to it.</p>   <p>Below are three optional steps to setting private Histories, a user can make use of any of the options below depending on what the user want to achieve:</p>   <ol>   <li>     <p>Changing the privacy settings of individual dataset.</p>     <ul>       <li>Click on the <strong>dataset</strong> name for a dropdown.</li>       <li>Clicking the ‘<strong>pencil - <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> icon</strong></li>       <li>Move on the <strong>Permissions</strong> tab.</li>       <li>On the permission tab is two input tab</li>       <li>On the second input with a label of <strong>access</strong></li>       <li>Search for the name of the user to grant permission</li>       <li>Click on <strong>save permission</strong></li>     </ul>     <p><img src="https://galaxyproject.org/learn/privacy-features/set-perm.gif" alt="gif of the process described above, in Galaxy" /></p>     <p><em>Note</em>: Adding additional roles to the ‘access’ permission along with your “private role” does not do what you may expect. Since roles are always logically added together, only you will be able to access the dataset, since only you are a member of your “private role”.</p>   </li>   <li>     <p>Make all datasets in the current history private.</p>     <ul>       <li>Open the History Options galaxy-gear menu <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> at the top of your history panel</li>       <li>Click the <strong>Make Private</strong> option in the dropdown menu available</li>       <li>Sets the default settings for all new datasets in this history to private.</li>     </ul>     <p><img src="https://galaxyproject.org/learn/privacy-features/this-hist-priv-perm.gif" alt="gif of the process described above, in Galaxy" /></p>   </li>   <li>     <p>Set the default privacy settings for new histories</p>     <ul>       <li>Click <strong>user</strong> button on top of the main channel for a dropdown <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span></li>       <li>Click on the <em>preferences</em> under the dropdown <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span></li>       <li>Select <strong>Set Dataset Permissions for New Histories</strong> icon <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">cofest</span></li>       <li>Add a permission and click <strong>save permission</strong></li>     </ul>     <p><img src="https://galaxyproject.org/learn/privacy-features/new-hist-perm.gif" alt="gif of the process described above, in Galaxy" /></p>     <p><em>Note</em>: Changes made here will only affect histories created after these settings have been stored.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing a history <a href="/training-material/faqs/galaxy/histories_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <em>Import this history</em> button on the top left</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Copy History</strong></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Manipulating multiple history datasets <a href="/training-material/faqs/galaxy/datasets_multiple.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to manipulate multiple history datasets at once
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-manipulating-multiple-history-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-manipulating-multiple-history-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Manipulating multiple history datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can also hide, delete, and purge multiple datasets at once by <strong>multi-selecting datasets</strong>:</p>   <ol>   <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> Click the multi-select button containing the checkbox just below the history size.</li>   <li>Checkboxes will appear inside each dataset in the history.</li>   <li>Scroll and click the checkboxes next to the datasets you want to manage.</li>   <li>Click the ‘n of N selected’ to choose the action. The action will be performed on all selected datasets, except for the ones that don’t support the action. That is, if an action doesn’t apply to a selected dataset, like deleting a deleted dataset, nothing will happen to that dataset, while all other selected datasets will be deleted.</li>   <li>You can click the multi-select button again to hide the checkboxes.</li> </ol>   <p><img src="/training-material/faqs/galaxy/images/multiselect.gif" alt="Operating on multiple datasets" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming a history <a href="/training-material/faqs/galaxy/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to rename a history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Searching your history <a href="/training-material/faqs/galaxy/histories_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-searching-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-searching-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Searching your history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To make it easier to find datasets in large histories, you can filter your history by keywords as follows:</p>   <ol>   <li>     <p><strong>Click</strong> on the <em>search datasets</em> box at the top of the history panel.</p>     <p><img src="/training-material/shared/images/history_search_datasets.png" alt="history search box" /></p>   </li>   <li><strong>Type</strong> a search term in this box     <ul>       <li>For example a tool name, or sample name</li>     </ul>   </li>   <li>To undo the filtering and show your full history again, press on the <em>clear search</em> button <i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-clear</span> next to the search box</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Sharing your History <a href="/training-material/faqs/galaxy/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can share your work in Galaxy. There are various ways you can give access to one of your histories to other users.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sharing-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sharing-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sharing your History</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sharing your history allows others to import and access the datasets, parameters, and steps of your history.</p>   <p>Access the history sharing menu via the History Options dropdown (<i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span>), and clicking “<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">history-share</span> Share or Publish”</p>   <ol>   <li><strong>Share via link</strong>     <ul>       <li>Open the <strong>History Options</strong> <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> menu at the top of your history panel and select “<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">history-share</span> Share or Publish”         <ul>           <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Make History accessible</strong></li>           <li>A <strong>Share Link</strong> will appear that you give to others</li>         </ul>       </li>       <li>Anybody who has this link can view and copy your history</li>     </ul>   </li>   <li><strong>Publish your history</strong>     <ul>       <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Make History publicly available in Published Histories</strong></li>       <li>Anybody on this Galaxy server will see your history listed under the <em>Published Histories</em> tab opened via the <i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">galaxy-histories-activity</span> <strong>Histories</strong> activity</li>     </ul>   </li>   <li><strong>Share only with another user</strong>.     <ul>       <li>Enter an email address for the user you want to share with in the <em>Please specify user email</em> input below <strong>Share History with Individual Users</strong></li>       <li>Your history will be shared only with this user.</li>     </ul>   </li>   <li><strong>Finding histories others have shared with me</strong>     <ul>       <li>Click on the <i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">galaxy-histories-activity</span> <strong>Histories</strong> activity in the activity bar on the left</li>       <li>Click the <strong>Shared with me</strong> tab</li>       <li>Here you will see all the histories others have shared with you directly</li>     </ul>   </li> </ol>   <p><strong>Note:</strong> If you want to make changes to your history without affecting the shared version, make a copy by going to <strong>History Options</strong> <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> icon in your history and clicking <em>Copy this History</em></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Switching to an existing history <a href="/training-material/faqs/galaxy/histories_switch.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Shows how to switch to another existing history in your account
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-switching-to-an-existing-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-switching-to-an-existing-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Switching to an existing history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To switch to an existing history simply click the <i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">switch-histories</span> icon at the top of the history panel. This opens a list of histories existing in a given Galaxy account in the middle part of the interface.</p>   <p><img src="/training-material/shared/images/list_of_histories.png" alt="UI for switching histories" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Top level history controls <a href="/training-material/faqs/galaxy/histories_top_control_buttons.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Description of three history buttons for creating a new histiory, switching histories, and opening history options dropdown
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-top-level-history-controls"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-top-level-history-controls" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Top level history controls</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Above the current history panel are three buttons:</p>   <ul>   <li>The <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> “<strong>Create new history</strong>” button will create an empty history.</li>   <li>The <i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">switch-histories</span> “<strong>Switch to history</strong>” will open a window letting you easily swap to any of your other histories.</li>   <li>The <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> “<strong>History options</strong>” (formerly the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> “Gear menu”) gives you access to advanced options to work with your history.</li> </ul>   <p><img src="/training-material/shared/images/history_top_level_buttons.svg" alt="The three history buttons" /></p>   <!-- history_top_level_buttons.svg = editable version = https://docs.google.com/drawings/d/11ZIUNYVHzrieHLVph9lhXgfzrO7MM-N4ehTehMr_eto/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Transfer entire histories from one Galaxy server to another <a href="/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-transfer-entire-histories-from-one-galaxy-server-to-another"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-transfer-entire-histories-from-one-galaxy-server-to-another" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Transfer entire histories from one Galaxy server to another</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Transfer a Single Dataset</strong></p>   <p>At the <strong>sender</strong> Galaxy server, <a href="/training-material/faqs/galaxy/histories_sharing.html">set the history to a shared state</a>, then directly capture the <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> link for a dataset and paste the URL into the <strong>Upload</strong> tool at the <strong>receiver</strong> Galaxy server.</p>   <p><strong>Transfer an Entire History</strong></p>   <p><a href="/training-material/faqs/galaxy/account_create.html">Have an account</a> at two different Galaxy servers, and be logged into both.</p>   <p>At the <strong>sender</strong> Galaxy server</p>   <ol>   <li>Navigate to the history you want to transfer, and <a href="/training-material/faqs/galaxy/histories_sharing.html">set the history to a shared state</a>.</li>   <li>Click into the <strong>History Options</strong> menu in the history panel.</li>   <li>Select from the menu <i class="fas fa-archive" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-archive</span> <strong>Export History to File</strong>.</li>   <li>Choose the option for <strong>How do you want to export this History?</strong> as <strong>to direct download</strong>.</li>   <li>Click on <strong>Generate direct download</strong>.</li>   <li>Allow the archive generation process to complete. *</li>   <li>Copy the <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> link for your new archive.</li> </ol>   <p>At the <strong>receiver</strong> Galaxy server</p>   <ol>   <li>Confirm that you are logged into your account.</li>   <li>Click on <strong>Data</strong> in the top menu, and choose <strong>Histories</strong> to reach your <strong>Saved Histories</strong>.</li>   <li>Click on <strong>Import history</strong> in the grey button on the top right.</li>   <li>Paste in your link’s URL from step 7.</li>   <li>Click on <strong>Import History</strong>.</li>   <li>Allow the archive import process to complete. *</li>   <li>The transfered history will be uncompressed and added to your <strong>Saved Histories</strong>.</li> </ol>   <p>* For steps 6 and 13: It is Ok to navigate away for other tasks during processing. If enabled, Galaxy will send you <a href="/training-material/faqs/galaxy/account_update_preference.html">status notifications</a>.</p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> If the history to transfer is large, you may <a href="/training-material/faqs/galaxy/histories_copy_dataset.html">copy just your important datasets into a new history</a>, and create the archive from that new smaller history. Clearing away deleted and purged datasets will make <em>all</em> histories smaller and faster to archive and transfer!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Undeleting history <a href="/training-material/faqs/galaxy/histories_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Undelete your deleted histories
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-undeleting-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-undeleting-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Undeleting history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Deleted histories can be <strong>undeleted</strong>:</p>   <ol>   <li>Select “Histories” from the activity bar on the left</li>   <li>Toggle “Advanced search”</li>   <li>Click “Deleted”</li>   <li>Click on the title of the history you want to un-delete and un-delete it!</li> </ol>   <p><img src="/training-material/faqs/galaxy/images/histories_undelete.gif" alt="Undeleting a deleted history" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Unsharing unwanted histories <a href="/training-material/faqs/galaxy/histories_unsharing_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-unsharing-unwanted-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-unsharing-unwanted-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Unsharing unwanted histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>All account Histories owned by others but shared with you can be reviewed under <strong>User &gt; Histories shared with me</strong>.</li>   <li>The other person does not need to unshare a history with you. <em>Unshare histories yourself</em> on this page using the pull-down menu per history.</li>   <li>Dataset and History privacy options, including sharing, can be set under <strong>User &gt; Preferences</strong>.</li> </ul>   <p>Three key features to work with shared data are:</p>   <ul>   <li><strong>View</strong> is a review feature. The data cannot be worked with, but many details, including tool and dataset metadata/parameters, are included.</li>   <li><strong>Copy</strong> those you want to work with. This will increase your quota usage. This will also allow you to manipulate the datasets or the history independently from the original owner. All History/Dataset functions are available if the other person granted full access to the datasets to you.</li>   <li><strong>Unshare</strong> any on the list not needed anymore. After a history is copied, you will still have your version of the history, even if later unshared or the other person who shared it with you changes their version later. Meaning, that <em>each account’s version of a History and the Datasets in it are distinct</em> (unless the Datasets were not shared, you will still only be able to “view” but not work with or download them).</li> </ul>   <p><strong>Note:</strong> “Histories shared with me” result in only a tiny part of your quota usage. Unsharing will not significantly reduce quota usage unless hundreds (or more!) or many significant histories are shared. If you share a History with someone else, that does not increase or decrease your quota usage.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> View a list of all histories <a href="/training-material/faqs/galaxy/histories_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This FAQ demonstrates how to list all histories for a given user
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-view-a-list-of-all-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-view-a-list-of-all-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: View a list of all histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are multiple ways in which you can view your histories:</p>   <ol>   <li>     <p>Viewing histories using <i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">switch-histories</span> “<strong>Switch to history</strong>” button. This is best for quickly switching between multiple histories.</p>     <p>Click the “<strong>Switch history</strong>” icon at the top of the history panel to bring up a list of all your histories: <img src="/training-material/shared/images/histories_list.svg" alt="Listing histories using the &quot;Switch history&quot; button" /></p>   </li>   <li>     <p>Using the “<strong>Activity Bar</strong>”:</p>     <p>Click the “<strong>Show all histories</strong>” button within the Activity Bar on the left: <img src="/training-material/shared/images/histories_list_via_activity_bar.svg" alt="Listing histories using Activity Bar" /></p>   </li>   <li>     <p>Using “<strong>Data</strong>” drop-down:</p>     <p>Click the “<strong>Data</strong>” link on the top bar of Galaxy interface and select “<strong>Histories</strong>”: <img src="/training-material/shared/images/histories_list_via_masthead.svg" alt="Listing histories using &quot;Data&quot; menu" /></p>   </li>   <li>     <p>Using the <strong>Multi-view</strong>, which is best for moving datasets between histories:</p>     <p>Click the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> menu, and select <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> <strong>Show histories side-by-side</strong></p>   </li> </ol>   <!-- Original drawing for histories_list.svg = https://docs.google.com/drawings/d/1fiMTFt4q1DwJllnxjddG0LA-XwKkxyXfPlOqUHteX_k/edit?usp=sharing -->   <!-- Original drawing for histories_list_via_activity_bar.svg = https://docs.google.com/drawings/d/1YyUMe6hhhejhRaqFEBmG_NwUS_9VXFZY2uYSoUNlFwY/edit?usp=sharing -->   <!-- Original drawing for histories_list_via_mesthead.svg = https://docs.google.com/drawings/d/1ReFUBwfVlbrKz0pJqY0xxnbb3EjRYJRzoq9bi1NqH-w/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> View histories side-by-side <a href="/training-material/faqs/galaxy/histories_side_by_side_view.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This FAQ demonstrates how to view histories side-by-sde
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-view-histories-side-by-side"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-view-histories-side-by-side" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: View histories side-by-side</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can view multiple Galaxy histories at once. This allows to better understand your analyses and also makes it possible to drag datasets between histories. This is called “<strong>History multiview</strong>”. The multiview can be enabled either view History menu or via the Activity Bar:</p>   <ol>   <li>     <p>Enabling Multiview via History menu is done by first clicking on the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> “<strong>History options</strong>” drop-down and selecting <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> “<strong>Show Histories Side-by-Side option</strong>”:</p>     <p><img src="/training-material/shared/images/histories_side_by_side_view.svg" alt="Enabling side-by-side view using History Options menu" /></p>   </li>   <li>     <p>Clicking the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> “<strong>History Multiview</strong>” button within the Activity Bar:</p>     <p><img src="/training-material/shared/images/histories_multiview_via_activity_bar.svg" alt="Enabling side-by-side view using Activity Bar" /></p>   </li> </ol>   <p><!-- Original drawings are accessible from https://docs.google.com/drawings/d/1coRlf1PxDKZYFt9j83MHGVYvwhBZhBkE-PcnhgIQ6gs/edit?usp=sharing and https://docs.google.com/drawings/d/1ksdR7wf2i5CJA1Pa-frCLMlgfITwwpOh2cKay8xDIi0/edit?usp=sharing --></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="history"> <a href="#history"> History </a></h3>

<hr/>



<h4 class="faq-area"> My jobs are not running / I cannot see the history overview menu <a href="/training-material/topics/introduction/faqs/login.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-my-jobs-are-not-running-i-cannot-see-the-history-overview-menu"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-my-jobs-are-not-running-i-cannot-see-the-history-overview-menu" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: My jobs are not running / I cannot see the history overview menu</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Please make sure you are logged in. At the top menu bar, you should see a section labeled “User”. If you see “Login/Register” here you are not logged in.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="igv"> <a href="#igv"> Igv </a></h3>

<hr/>



<h4 class="faq-area"> Add Mapped reads track to IGV from Galaxy <a href="/training-material/topics/assembly/faqs/visualizations_igv_view_mapping.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-add-mapped-reads-track-to-igv-from-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-mapped-reads-track-to-igv-from-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Add Mapped reads track to IGV from Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>   <li>Launch IGV on your computer</li>   <li>Check if the reference genome is available on the IGV instance</li>   <li>Expand the BAM dataset with the mapped reads in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the reads into the IGV browser</li>   <li>     <p>Switch to the IGV instance</p>     <p>The mapped reads track should appear. Be sure that all files have the same genome ID</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Add Mapped reads track to IGV from Galaxy <a href="/training-material/topics/visualisation/faqs/visualizations_igv_view_mapping.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-add-mapped-reads-track-to-igv-from-galaxy-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-mapped-reads-track-to-igv-from-galaxy-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Add Mapped reads track to IGV from Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>   <li>Launch IGV on your computer</li>   <li>Check if the reference genome is available on the IGV instance</li>   <li>Expand the BAM dataset with the mapped reads in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the reads into the IGV browser</li>   <li>     <p>Switch to the IGV instance</p>     <p>The mapped reads track should appear. Be sure that all files have the same genome ID</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Add genome and annotations to IGV from Galaxy <a href="/training-material/topics/assembly/faqs/visualizations_igv_add_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-add-genome-and-annotations-to-igv-from-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-genome-and-annotations-to-igv-from-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Add genome and annotations to IGV from Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>   <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>   <li>Launch IGV on your computer</li>   <li>Expand the FASTA dataset with the genome in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>   <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>   <li>     <p>Close the window</p>     <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>   </li>   <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>   <li>     <p>Switch to the IGV instance</p>     <p>The annotation track should appear. Be careful that all files have the same genome ID</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Add genome and annotations to IGV from Galaxy <a href="/training-material/topics/visualisation/faqs/visualizations_igv_add_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-add-genome-and-annotations-to-igv-from-galaxy-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-genome-and-annotations-to-igv-from-galaxy-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Add genome and annotations to IGV from Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Upload a FASTA file with the reference genome and a GFF3 file with its annotation in the history (if not already there)</li>   <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a> (if not already installed)</li>   <li>Launch IGV on your computer</li>   <li>Expand the FASTA dataset with the genome in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the genome into the IGV browser</li>   <li>Wait until all <strong>Dataset status</strong> are <code class="language-plaintext highlighter-rouge">ok</code></li>   <li>     <p>Close the window</p>     <p>An alert <code class="language-plaintext highlighter-rouge">ERROR Parameter "file" is required</code> may appear. Ignore it.</p>   </li>   <li>Expand the GFF3 dataset with the annotations of the genome in the history</li>   <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the annotation into the IGV browser</li>   <li>     <p>Switch to the IGV instance</p>     <p>The annotation track should appear. Be careful that all files have the same genome ID</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="inputs"> <a href="#inputs"> Inputs </a></h3>

<hr/>



<h4 class="faq-area"> Do I need to create collections to run MaxQuant analysis or can I use single sample inputs? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-i-need-to-create-collections-to-run-maxquant-analysis-or-can-i-use-single-sample-inputs"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do I need to create collections to run MaxQuant analysis or can I use single sample inputs?</div>   <p>Collections are not necessary to run MaxQuant but they make the history more clean and easier to navigate. The multiple datasets options allows to select multiple files that are not part of a collection and will give the same result as with a collection as input.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Do we need a contaminant FASTA for MQ in galaxy? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/contaminants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-we-need-a-contaminant-fasta-for-mq-in-galaxy"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do we need a contaminant FASTA for MQ in galaxy?</div>   <p>Normally MaxQuant has a default contaminant fasta that we don’t have to input ourselves. MaxQuant in galaxy comes with the option to add contaminants automatically (one does not need to add contaminants to the fasta file)</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Do you need to merge the databases? Because you can select multiple fasta files in MaxQuant. <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/merge_database.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-do-you-need-to-merge-the-databases-because-you-can-select-multiple-fasta-files-in-maxquant"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Do you need to merge the databases? Because you can select multiple fasta files in MaxQuant.</div>   <p>For MaxQuant one does not need to merge the databases, also MaxQuant offers the function to add common contaminants to the provided fasta.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="instructors"> <a href="#instructors"> Instructors </a></h3>

<hr/>



<h4 class="faq-area"> Finding a material's PURL or Short URL <a href="/training-material/faqs/gtn/short-url.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-a-material-s-purl-or-short-url"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-a-material-s-purl-or-short-url" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding a material's PURL or Short URL</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Every material in the GTN is automatically assigned two short URLs:</p>   <ul>   <li>a <a href="/training-material/news/2023/04/19/shortlinks.html">PURL</a> which will always point to the material, and looks like <code class="language-plaintext highlighter-rouge">https://gxy.io/GTN:T00001</code></li>   <li>a tutorial ID based short URL like <code class="language-plaintext highlighter-rouge">https://gxy.io/GTN:admin/ansible-galaxy</code>, which will redirect to topics/admin/tutorials/ansible-galaxy/tutorial.md</li> </ul>   <p>The PURLs, when available, are listed in the Metadata box of a given material, or on the first slide of a slide deck. Additionally any page with a PURL lists it in the footer of the page. PURLs are generated every monday, so it can take up to a week for your PURL to be available. If you need it sooner, please let us know.</p>   <p>The second short URL is not currently displayed anywhere but can be constructed manually based on the URL of the page.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I get help? <a href="/training-material/faqs/gtn/instructors_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-get-help"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-get-help" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I get help?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The support channel for instructors is the same as for individual learners.  We suggest you start by posting a question to the Galaxy Training Network <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter chat</a>. Anyone can view the discussion, but you’ll need to login (using your GitHub or Twitter account) to add to the discussion.</p>   <p>If you have questions about Galaxy in general (that are not training-centric) then there are <a href="https://galaxyproject.org/support/">several support options</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What Galaxy instance should I use for my training? <a href="/training-material/faqs/gtn/instructors_choosing_server.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-galaxy-instance-should-i-use-for-my-training"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-galaxy-instance-should-i-use-for-my-training" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What Galaxy instance should I use for my training?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To teach the hands-on tutorials you need a Galaxy server to run the examples on.</p>   <p>Each tutorial is annotated with the information on which <a href="https://galaxyproject.org/public-galaxy-servers/">public Galaxy servers</a> it can be run. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial. If you choose this option then you should work with that server’s admins to confirm that the server can handle the workload for a workshop. For example, the <a href="https://usegalaxy.eu/">usegalaxy.eu</a></p>   <p>If your organization/consortia/community has its own Galaxy server, then you may want to run tutorials on that. This can be ideal because then the instance you are teaching on is the same as your participants will be using after the training. They’ll also be able to revisit any analysis they did during the training. If you pursue this option you’ll need to work with your organization’s Galaxy Admins to confirm that</p>   <ul>   <li>the server can support a room full of people all doing the same analysis at the same time.</li>   <li>all tools and reference datasets needed in the tutorial are locally installed.  To learn how to setup a Galaxy instance for a tutorial, you can follow our <a href="/training-material/topics/teaching/tutorials/setup-galaxy-for-training/tutorial.html">dedicated tutorial</a>.</li>   <li>all participants will be able to create/use accounts on the system.</li> </ul>   <p>Some training topics have a Docker image that can be installed and run on all participants’ laptops.  These images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant.</p>   <p>Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is <a href="https://galaxyproject.org/galaxy-services/">available on many national research infrastructures</a> such as <a href="https://galaxyproject.org/cloud/jetstream/">Jetstream</a> (United States), <a href="https://www.genap.ca/">GenAP</a> (Canada), <a href="https://launch.genome.edu.au/launch">GVL</a> (Australia), <a href="http://www.climb.ac.uk/">CLIMB</a> (United Kingdom), and more.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What are the best practices for teaching with Galaxy? <a href="/training-material/faqs/gtn/instructors_best_practices.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-are-the-best-practices-for-teaching-with-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-are-the-best-practices-for-teaching-with-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What are the best practices for teaching with Galaxy?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>We started to collect some best practices for instructors inside our <a href="/training-material/topics/teaching/index.html">Good practices slides</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where do I start? <a href="/training-material/faqs/gtn/instructors_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-do-i-start"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-do-i-start" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where do I start?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Spend some time exploring the different tutorials and the different resources that are available. Become familiar with the structure of the tutorials and think about how you might use them in your teaching.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="interactive tools"> <a href="#interactive tools"> Interactive tools </a></h3>

<hr/>



<h4 class="faq-area"> Knitting RMarkdown documents in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-knitting-rmarkdown-documents-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Knitting RMarkdown documents in RStudio</div>   <p>One of the other nice features of RMarkdown documents is making lovely presentation-quality worthy documents. You can take, for example, a tutorial and produce a nice report like output as HTML, PDF, or <code class="language-plaintext highlighter-rouge">.doc</code> document that can easily be shared with colleagues or students.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-meta-output.png" alt="Screenshot of the metadata with html_notebook and word_document being visible and a number of options controlling their output. TOC, standing for table of contents, has been set to true for both." /></p>   <p>Now you’re <strong>ready to preview the document</strong>:</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of preview dropdown with options like preview, knit to html, knit to pdf, knit to word" /></p>   <p><strong>Click Preview</strong>. A window will popup with a preview of the rendered verison of this document.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of rendered document with the table of contents on left, title is in a large font, and there are coloured boxes similar to GTN tutorials offering tips and more information" /></p>   <p>The preview is really similar to the GTN rendering, no cells have been executed, and no output is embedded yet in the preview document. But if you have run cells (e.g. the first few loading a library and previewing the <code class="language-plaintext highlighter-rouge">msleep</code> dataset:</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-preview-output.png" alt="screenshot of the rendered document with a fancy table browser embedded as well as the output of each step" /></p>   <p><strong>When you’re ready to distribute</strong> the document, you can instead use the <code class="language-plaintext highlighter-rouge">Knit</code> button. This runs every cell in the entire document fresh, and then compiles the outputs together with the rendered markdown to produce a nice result file as HTML, PDF, or Word document.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-knit.png" alt="screenshot of the console with 'chunks' being knitted together" /></p>   <blockquote class="tip">   <h3 data-toc-skip id="tip-tip-pdf--word-require-a-latex-installation"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: PDF + Word require a LaTeX installation</h3>   <p>You might need to install additional packages to compile the PDF and Word document versions</p> </blockquote>   <p>And at the end you can see a pretty document rendered with all of the output of every step along the way. This is a fantastic way to e.g. distribute read-only lesson materials to students, if you feel they might struggle with using an RMarkdown document, or just want to read the output without <em>doing</em> it themselves.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-pdf.png" alt="screenshot of a PDF document showing the end of the tutorial where a pretty plot has been rendered and there is some text for conclusions and citations" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Launch JupyterLab <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch JupyterLab</div>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-jupyterlab-in-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-jupyterlab-in-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch JupyterLab in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently JupyterLab in Galaxy is available on <a href="https://live.usegalaxy.eu">Live.useGalaxy.eu</a>, <a href="https://usegalaxy.org">usegalaxy.org</a> and <a href="https://usegalaxy.eu">usegalaxy.eu</a>.</p>   <blockquote class="notranslate hands_on">     <div class="box-title hands-on-title" id="hands-on-run-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run JupyterLab</div>     <ol>       <li><span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Interactive Jupyter Notebook tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Interactive Jupyter Notebook</strong></span>. Note that on some Galaxies this is called <span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Interactive JupyTool and notebook tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Interactive JupyTool and notebook</strong></span>:</li>       <li>Click Run Tool</li>       <li>         <p>The tool will start running and will stay running permanently</p>         <p>This may take a moment, but once the <code class="language-plaintext highlighter-rouge">Executed notebook</code> in your history is <strong>orange</strong>, you are up and running!</p>       </li>       <li>Click on the <strong>User</strong> menu at the top and go to <strong>Active Interactive Tools</strong> and locate the JupyterLab instance you started.</li>       <li>Click on your JupyterLab instance</li>     </ol>   </blockquote> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-try-jupyterlab-if-not-available-on-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-try-jupyterlab-if-not-available-on-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch Try JupyterLab if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If JupyterLab is not available on the Galaxy instance:</p>   <ol>     <li>Start <a href="https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/try.jupyter.org?urlpath=lab">Try JupyterLab</a></li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Launch RStudio <a href="/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch RStudio</div>   <p>Depending on which server you are using, you may be able to run RStudio directly in Galaxy. If that is not available, RStudio Cloud can be an alternative.</p>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-in-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-in-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently RStudio in Galaxy is only available on <a href="https://usegalaxy.eu">UseGalaxy.eu</a> and <a href="https://usegalaxy.org">UseGalaxy.org</a></p>   <ol>     <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here to launch RStudio</a></li>     <li>Click Run Tool</li>     <li>The tool will start running and will stay running permanently</li>     <li>Click on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started.</li>   </ol> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-cloud-if-not-available-on-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-cloud-if-not-available-on-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio Cloud if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If RStudio is not available on the Galaxy instance:</p>   <ol>     <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>     <li>Create a new project</li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Launch RStudio <a href="/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-rstudio-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch RStudio</div>   <p>Depending on which server you are using, you may be able to run RStudio directly in Galaxy. If that is not available, RStudio Cloud can be an alternative.</p>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-in-galaxy-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-in-galaxy-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently RStudio in Galaxy is only available on <a href="https://usegalaxy.eu">UseGalaxy.eu</a> and <a href="https://usegalaxy.org">UseGalaxy.org</a></p>   <ol>     <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here to launch RStudio</a></li>     <li>Click Run Tool</li>     <li>The tool will start running and will stay running permanently</li>     <li>Click on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started.</li>   </ol> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-cloud-if-not-available-on-galaxy-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-cloud-if-not-available-on-galaxy-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio Cloud if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If RStudio is not available on the Galaxy instance:</p>   <ol>     <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>     <li>Create a new project</li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Launch RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-rstudio-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch RStudio</div>   <p>Depending on which server you are using, you may be able to run RStudio directly in Galaxy. If that is not available, RStudio Cloud can be an alternative.</p>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-in-galaxy-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-in-galaxy-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently RStudio in Galaxy is only available on <a href="https://usegalaxy.eu">UseGalaxy.eu</a> and <a href="https://usegalaxy.org">UseGalaxy.org</a></p>   <ol>     <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here to launch RStudio</a></li>     <li>Click Run Tool</li>     <li>The tool will start running and will stay running permanently</li>     <li>Click on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started.</li>   </ol> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-cloud-if-not-available-on-galaxy-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-cloud-if-not-available-on-galaxy-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio Cloud if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If RStudio is not available on the Galaxy instance:</p>   <ol>     <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>     <li>Create a new project</li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Learning with RMarkdown in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-learning-with-rmarkdown-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Learning with RMarkdown in RStudio</div>   <p>Learning with RMarkdown is a bit different than you might be used to. Instead of copying and pasting code from the GTN into a document you’ll instead be able to run the code directly as it was written, inside RStudio! You can now focus just on the code and reading within RStudio.</p>   <ol>   <li>     <p><strong>Load the notebook</strong> if you have not already, following the tip box at the top of the tutorial</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-download-file.png" alt="Screenshot of the Console in RStudio. There are three lines visible of not-yet-run R code with the download.file statements which were included in the setup tip box." /></p>   </li>   <li>     <p><strong>Open it</strong> by clicking on the <code class="language-plaintext highlighter-rouge">.Rmd</code> file in the file browser (bottom right)</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-file-menu.png" alt="Screenshot of Files tab in RStudio, here there are three files listed, a data-science-r-dplyr.Rmd file, a css and a bib file." /></p>   </li>   <li>     <p>The RMarkdown document will appear in the document viewer (top left)</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-document-view.png" alt="Screenshot of an open document in RStudio. There is some yaml metadata above the tutorial showing the title of the tutorial." /></p>   </li> </ol>   <p>You’re now ready to view the RMarkdown notebook! Each notebook starts with a lot of metadata about how to build the notebook for viewing, but you can ignore this for now and <strong>scroll down to the content of the tutorial</strong>.</p>   <p>You can switch to the visual mode which is way easier to read - just click on the gear icon and select <code class="language-plaintext highlighter-rouge">Use Visual Editor</code>.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-visual-mode.png" alt="Screenshot of dropdown menu after clicking on the gear icon. The first option is `Use Visual Editor`." /></p>   <p>You’ll see codeblocks scattered throughout the text, and these are all runnable snippets that appear like this in the document:</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-run-cell.png" alt="Screenshot of the RMarkdown document in the viewer, a cell is visible between markdown text reading library tidyverse. It is slightly more grey than the background region, and it has a run button at the right of the cell in a contextual menu." /></p>   <p>And you have a few options for how to run them:</p>   <ol>   <li>Click the green arrow</li>   <li><kbd>ctrl+enter</kbd></li>   <li>     <p>Using the menu at the top to run all</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-run-all.png" alt="Screenshot of the run dropdown menu in R, the first item is run selected lines showing the mentioned shortcut above, the second is run next chunk, and then it also mentions a 'run all chunks below' and 'restart r and run all chunks' option." /></p>   </li> </ol>   <p>When you run cells, the output will appear below in the Console. RStudio essentially copies the code from the RMarkdown document, to the console, and runs it, just as if you had typed it out yourself!</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-cell-output.png" alt="Screenshot of a run cell, its output is included below in the RMarkdown document and the same output is visible below in the console. It shows a log of loading the tidyverse library." /></p>   <p>One of the best features of RMarkdown documents is that they include a very nice table browser which makes previewing results a lot easier! Instead of needing to use <code class="language-plaintext highlighter-rouge">head</code> every time to preview the result, you get an interactive table browser for any step which outputs a table.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-table.png" alt="Screenshot of the table browser. Below a code chunk is a large white area with two images, the first reading 'r console' and the second reading 'tbl_df'. The tbl_df is highlighted like it is active. Below that is a pretty-printed table with bold column headers like name and genus and so on. At the right of the table is a small arrow indicating you can switch to seeing more columns than just the initial three. At the bottom of the table is 1-10 of 83 rows written, and buttons for switching between each page of results." /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Open a Terminal in Jupyter <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-open-a-terminal-in-jupyter"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Open a Terminal in Jupyter</div>   <p>This tutorial will let you accomplish almost everything from this view, running code in the cells below directly in the training material. You can choose between running the code here, or opening up a terminal tab in which to run it.Here are some instructions for how to do this on various environments.</p>   <p><strong>Jupyter</strong> on UseGalaxy.* and <strong>MyBinder.org</strong></p>   <ol>   <li>     <p>Use the File → New → Terminal menu to launch a terminal.</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-launch.png" alt="screenshot of jupyterlab showing the File menu expanded to show new and terminal option." /></p>   </li>   <li>     <p>Disable “Simple” mode in the bottom left hand corner, if it activated.</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-simple.png" alt="screenshot of jupyterlab showing a toggle labelled simple" /></p>   </li>   <li>     <p>Drag one of the terminal or notebook tabs to the side to have the training materials and terminal side-by-side</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-final.png" alt="screenshot of jupyterlab with notebook and terminal side-by-side." /></p>   </li> </ol>   <p><strong>CoCalc</strong></p>   <ol>   <li>     <p>Use the <em>Split View</em> functionality of cocalc to split your view into two portions.</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-split.png" alt="screenshot of cocalc button to split views" /></p>   </li>   <li>     <p>Change the view of one panel to a terminal</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-terminal.png" alt="screenshot of cocalc swapping view port to that of a terminal" /></p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Open interactive tool <a href="/training-material/faqs/galaxy/interactive_tools_open.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-open-interactive-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-open-interactive-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Open interactive tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Go to User &gt; Active InteractiveTools</li>   <li>Wait for the  to be running (Job Info)</li>   <li>Click on</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Stop RStudio <a href="/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Stop RStudio</div>   <p>When you have finished your R analysis, it’s time to stop RStudio.</p>   <ol>   <li>First, save your work into Galaxy, to ensure reproducibility:     <ol>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>     </ol>   </li>   <li>Once you have saved your data, you can proceed in 2 different ways:     <ul>       <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a “in progress state”, so yellow, OR</li>       <li>Clicking on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the “Stop” button at the bottom.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Stop RStudio <a href="/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-stop-rstudio-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Stop RStudio</div>   <p>When you have finished your R analysis, it’s time to stop RStudio.</p>   <ol>   <li>First, save your work into Galaxy, to ensure reproducibility:     <ol>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>     </ol>   </li>   <li>Once you have saved your data, you can proceed in 2 different ways:     <ul>       <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a “in progress state”, so yellow, OR</li>       <li>Clicking on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the “Stop” button at the bottom.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Stop RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-stop-rstudio-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Stop RStudio</div>   <p>When you have finished your R analysis, it’s time to stop RStudio.</p>   <ol>   <li>First, save your work into Galaxy, to ensure reproducibility:     <ol>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>     </ol>   </li>   <li>Once you have saved your data, you can proceed in 2 different ways:     <ul>       <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a “in progress state”, so yellow, OR</li>       <li>Clicking on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the “Stop” button at the bottom.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="introduction"> <a href="#introduction"> Introduction </a></h3>

<hr/>



<h4 class="faq-area"> How can I advertise the training materials on my posters? <a href="/training-material/faqs/gtn/gtn_advertise.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-advertise-the-training-materials-on-my-posters"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-advertise-the-training-materials-on-my-posters" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I advertise the training materials on my posters?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>We provide some QR codes and logos in the <a href="https://github.com/galaxyproject/training-material/tree/main/assets/images">images folder</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I cite the GTN? <a href="/training-material/faqs/gtn/gtn_citing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-cite-the-gtn"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-cite-the-gtn" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I cite the GTN?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>We wrote two articles about our efforts:</p>   <ul>   <li><a href="https://doi.org/10.1016/j.cels.2018.05.012">Batut, Bérénice, et al. “Community-driven data analysis training for biology.” Cell systems 6.6 (2018): 752-758</a></li>   <li><a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">Hiltemann, Saskia, Rasche, Helena et al. “Galaxy Training: A Powerful Framework for Teaching!.” PLOS Computational Biology (2023) 19(1): e1010752</a></li> </ul>   <p>To cite individual tutorials, please find citation information at the end of the tutorial.</p>   <p>Here is the BibTeX formatted version of those citations:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@article{Hiltemann_2023,<br>	title        = {Galaxy Training: A powerful framework for teaching!},<br>	author       = {Hiltemann, Saskia and Rasche, Helena and Gladman, Simon and Hotz, Hans-Rudolf and Larivi\`{e}re, Delphine and Blankenberg, Daniel and Jagtap, Pratik D. and Wollmann, Thomas and Bretaudeau, Anthony and Gou\'{e}, Nadia and Griffin, Timothy J. and Royaux, Coline and Le Bras, Yvan and Mehta, Subina and Syme, Anna and Coppens, Frederik and Droesbeke, Bert and Soranzo, Nicola and Bacon, Wendi and Psomopoulos, Fotis and Gallardo-Alba, Crist\'{o}bal and Davis, John and F\"{o}ll, Melanie Christine and Fahrner, Matthias and Doyle, Maria A. and Serrano-Solano, Beatriz and Fouilloux, Anne Claire and van Heusden, Peter and Maier, Wolfgang and Clements, Dave and Heyl, Florian and Gr\"{u}ning, Bj\"{o}rn and Batut, B\'{e}r\'{e}nice},<br>	year         = 2023,<br>	month        = jan,<br>	journal      = {PLOS Computational Biology},<br>	publisher    = {Public Library of Science (PLoS)},<br>	volume       = 19,<br>	number       = 1,<br>	pages        = {e1010752},<br>	doi          = {10.1371/journal.pcbi.1010752},<br>	issn         = {1553-7358},<br>	url          = {http://dx.doi.org/10.1371/journal.pcbi.1010752},<br>	editor       = {Ouellette, Francis},<br>}<br>@article{Batut_2018,<br>	title        = {Community-Driven Data Analysis Training for Biology},<br>	author       = {Batut, B\'{e}r\'{e}nice and Hiltemann, Saskia and Bagnacani, Andrea and Baker, Dannon and Bhardwaj, Vivek and Blank, Clemens and Bretaudeau, Anthony and Brillet-Gu\'{e}guen, Loraine and \v{C}ech, Martin and Chilton, John and Clements, Dave and Doppelt-Azeroual, Olivia and Erxleben, Anika and Freeberg, Mallory Ann and Gladman, Simon and Hoogstrate, Youri and Hotz, Hans-Rudolf and Houwaart, Torsten and Jagtap, Pratik and Larivi\`{e}re, Delphine and Le Corguill\'{e}, Gildas and Manke, Thomas and Mareuil, Fabien and Ram\'{\i}rez, Fidel and Ryan, Devon and Sigloch, Florian Christoph and Soranzo, Nicola and Wolff, Joachim and Videm, Pavankumar and Wolfien, Markus and Wubuli, Aisanjiang and Yusuf, Dilmurat and Taylor, James and Backofen, Rolf and Nekrutenko, Anton and Gr\"{u}ning, Bj\"{o}rn},<br>	year         = 2018,<br>	month        = jun,<br>	journal      = {Cell Systems},<br>	publisher    = {Elsevier BV},<br>	volume       = 6,<br>	number       = 6,<br>	pages        = {752--758.e1},<br>	doi          = {10.1016/j.cels.2018.05.012},<br>	issn         = {2405-4712},<br>	url          = {http://dx.doi.org/10.1016/j.cels.2018.05.012},<br>}<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I load data? <a href="/training-material/faqs/gtn/get_data_upload.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-load-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-load-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I load data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Load by “browsing” for a local file. Some servers will support load data that is 2 GB or larger. If you are having problems with this method, try FTP.</li>   <li>Load using an <strong>HTTP URL</strong> or <strong>FTP URL</strong>.</li>   <li>Load a few lines of plain text.</li>   <li>Load using FTP. Either line command or with a desktop client.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How is the content licensed? <a href="/training-material/faqs/gtn/gtn_license.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-is-the-content-licensed"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-is-the-content-licensed" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How is the content licensed?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The content of this website is licensed under the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 License</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using Answer Key Histories <a href="/training-material/faqs/gtn/gtn_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-answer-key-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-answer-key-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using Answer Key Histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you get stuck, you can first check your history against an exemplar history, from your tutorial.</p>   <p>First, import the target history.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <em>Import this history</em> button on the top left</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Copy History</strong></li> </ol> </blockquote>   <p><!--END_SNIPPET--></p>   <p>Next, compare the answer key history with your own history.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-view-histories-side-by-side-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-view-histories-side-by-side-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: View histories side-by-side</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can view multiple Galaxy histories at once. This allows to better understand your analyses and also makes it possible to drag datasets between histories. This is called “<strong>History multiview</strong>”. The multiview can be enabled either view History menu or via the Activity Bar:</p>   <ol>   <li>     <p>Enabling Multiview via History menu is done by first clicking on the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> “<strong>History options</strong>” drop-down and selecting <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> “<strong>Show Histories Side-by-Side option</strong>”:</p>     <p><img src="/training-material/shared/images/histories_side_by_side_view.svg" alt="Enabling side-by-side view using History Options menu" /></p>   </li>   <li>     <p>Clicking the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> “<strong>History Multiview</strong>” button within the Activity Bar:</p>     <p><img src="/training-material/shared/images/histories_multiview_via_activity_bar.svg" alt="Enabling side-by-side view using Activity Bar" /></p>   </li> </ol>   <p><!-- Original drawings are accessible from https://docs.google.com/drawings/d/1coRlf1PxDKZYFt9j83MHGVYvwhBZhBkE-PcnhgIQ6gs/edit?usp=sharing and https://docs.google.com/drawings/d/1ksdR7wf2i5CJA1Pa-frCLMlgfITwwpOh2cKay8xDIi0/edit?usp=sharing --></p> </blockquote>   <p><!--END_SNIPPET--></p>   <p>You can compare there, or if you’re really stuck, you can also click and drag a given dataset to your history to continue the tutorial from there.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-copy-a-dataset-between-histories-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-copy-a-dataset-between-histories-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Copy a dataset between histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There 3 ways to copy datasets between histories</p>   <ol>   <li>     <p>From the original history</p>     <ol>       <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon which is on the top of the list of datasets in the history panel</li>       <li>Click on <strong>Copy Datasets</strong></li>       <li>         <p>Select the desired files</p>       </li>       <li>         <p>Give a relevant name to the “New history”</p>       </li>       <li>Validate by ‘Copy History Items’</li>       <li>Click on the new history name in the green box that have just appear to switch to this history</li>     </ol>   </li>   <li>     <p>Using the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>Show Histories Side-by-Side</strong></p>     <ol>       <li>Click on the <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> dropdown arrow top right of the history panel (<strong>History options</strong>)</li>       <li>Click on <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>Show Histories Side-by-Side</strong></li>       <li>If your target history is not present         <ol>           <li>Click on ‘Select histories’</li>           <li>Click on your target history</li>           <li>Validate by ‘Change Selected’</li>         </ol>       </li>       <li>Drag the dataset to copy from its original history</li>       <li>Drop it in the target history</li>     </ol>   </li>   <li>     <p>From the target history</p>     <ol>       <li>Click on <strong>User</strong> in the top bar</li>       <li>Click on <strong>Datasets</strong></li>       <li>Search for the dataset to copy</li>       <li>Click on its name</li>       <li>Click on <strong>Copy to current History</strong></li>     </ol>   </li> </ol> </blockquote>   <p><!--END_SNIPPET--></p>   <p>You can also use our handy troubleshooting guide.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-errors-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-errors-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Troubleshooting errors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When something goes wrong in Galaxy, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your Galaxy errors.</p>   <ol>   <li><strong>Expand</strong> the red history dataset by clicking on it.     <ul>       <li>Sometimes you can already see an error message here</li>     </ul>   </li>   <li>     <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></p>   </li>   <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:     <ul>       <li>Expand the history item</li>       <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>       <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:         <ul>           <li>Tool Standard Output</li>           <li>Tool Standard Error</li>         </ul>       </li>       <li>For more information about specific tool errors, please see the <a href="/training-material/faqs/galaxy/index.html#troubleshooting">Troubleshooting section</a></li>     </ul>   </li>   <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.     <ul>       <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></li>       <li>Write down any information you think might help solve the problem         <ul>           <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>         </ul>       </li>       <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> Report</strong> button</li>     </ul>   </li>   <li><strong>Ask for help!</strong>     <ul>       <li>Where?         <ul>           <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Matrix Channel</a></li>           <li>In the <a href="https://gitter.im/galaxyproject/Lobby">Galaxy Matrix Channel</a></li>           <li>Browse the <a href="https://help.galaxyproject.org/">Galaxy Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>         </ul>       </li>       <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>     </ul>   </li> </ol> </blockquote>   <p><!--END_SNIPPET--></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Ways to use Galaxy <a href="/training-material/faqs/gtn/galaxy_usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-ways-to-use-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-ways-to-use-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Ways to use Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>All ways to use Galaxy are included in the <a href="https://galaxyproject.org/use">Galaxy Directory listing</a>.</p>   <p>Having one account at several public Galaxy servers expands your access to distinct data storage and computational resources, plus common and domain-specific analysis tools.</p>   <p>When running your own private Galaxy server for routine analysis, publishing results at a public Galaxy server allows for worldwide access by others when you share your data: Histories, Workflows, and related assets.</p>   <p>Tips:</p>   <ul>   <li><strong>Teaching with Galaxy</strong> We strongly recommend using Galaxy’s Training Infrastructure as a Service (<a href="https://usegalaxy-eu.github.io/tiaas">TIaaS</a>) for synchronous class work.</li>   <li><strong>Public Galaxy</strong> servers are appropriate for many analysis projects or for when sharing data or results publicly is a goal. These are also a great choice when learning on your own with <a href="https://training.galaxyproject.org/">GTN</a> tutorials.</li>   <li><strong>Private Galaxy</strong> servers are more appropriate when working with very large data, time sensitive projects, and ongoing research projects that require more resources than the public Galaxy servers can support. These two options are scientist friendly as they require very little to no server administration.     <ul>       <li><a href="https://galaxyproject.org/use/aws/">GVL Cloudman</a> is a single or multi-user choice and AWS <a href="https://aws.amazon.com/grants/">offers grants</a>.</li>       <li><a href="https://galaxyproject.org/use/anvil/">AnVIL</a> is a single-user choice sponsored by NHGRI and is a pay-for-use Google Cloud platform.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What are the tutorials for? <a href="/training-material/faqs/gtn/gtn_tutorials.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-are-the-tutorials-for"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-are-the-tutorials-for" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What are the tutorials for?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>These tutorials can be used for learning and teaching how to use Galaxy for general data analysis, and for learning/teaching specific domains such as assembly and differential gene expression analysis with RNA-Seq data.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What audiences are the tutorials for? <a href="/training-material/faqs/gtn/gtn_audience.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-audiences-are-the-tutorials-for"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-audiences-are-the-tutorials-for" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What audiences are the tutorials for?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are two distinct audiences for these materials.</p>   <ol>   <li><strong>Self-paced individual learners.</strong> These tutorials provide everything you need to learn a topic, from explanations of concepts to detailed hands-on exercises.</li>   <li><strong>Instructors.</strong> They are also designed to be used by instructors in teaching/training settings. Slides, and detailed tutorials are provided. Most tutorials also include computational support with the needed tools, data as well as Docker images that can be used to scale the lessons up to many participants.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is Galaxy? <a href="/training-material/faqs/gtn/galaxy_what_is_it.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-is-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-is-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What is Galaxy?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://galaxyproject.org/">Galaxy</a> is an open data integration and analysis platform for the life sciences, and it is particularly well-suited for data analysis training in life science research.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is a Learning Pathway? <a href="/training-material/faqs/gtn/what-is-a-learning-pathway.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-what-is-a-learning-pathway"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: What is a Learning Pathway?</div>   <div style="display:inline-block;vertical-align:top;width:10%"> <img src="/training-material/assets/images/learning-pathway.png" alt="A graphic depicting a winding path from a start symbol to a trophy, with tutorials along the way" class="invert-safe" /> </div>   <div style="display:inline-block; margin-left: 2em; width:70%"> We recommend you follow the tutorials in the order presented on this page. They have been selected to fit together and build up your knowledge step by step. If a lesson has both slides and a tutorial, we recommend you start with the slides, then proceed with the tutorial. </div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is my.galaxy.training <a href="/training-material/faqs/gtn/my-galaxy-training.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-is-my-galaxy-training"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-is-my-galaxy-training" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What is my.galaxy.training</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The <a href="https://my.galaxy.training"><code class="language-plaintext highlighter-rouge">my.galaxy.training</code></a> is part of the GTN. We found that often need to direct our learners to specific pages within Galaxy, but which Galaxy? Should we add three links, one for each of the current bigger UseGalaxy.* servers? That would be really annoying for users who aren’t using one of those servers.</p>   <p>E.g. how do we link to <a href="https://my.galaxy.training/?path=/user">/user</a>, the user preferences page which is available on every Galaxy Instance? This service handles that in a private and user-friendly manner.</p>   <h3 data-toc-skip id="learners-how-to-use-it">(Learners) How to Use It</h3>   <p>When you access a my.galaxy.training page you’ll be prompted to select a server, simply select one and you’re good to go!</p>   <p>If you want to enter a private Galaxy instance, perhaps, behind a firewall, that’s also an option! Just select the ‘other’ option and provide your domain. Since the redirection happens in your browser with no servers involved, as long as <em>you</em> can access the server, you’ll get redirected to the right location.</p>   <h3 data-toc-skip id="tutorial-authors-how-to-use-it">(Tutorial Authors) How to use it</h3>   <p>If you want to link to a specific page within Galaxy, simple construct the URL: <code class="language-plaintext highlighter-rouge">https://my.galaxy.training/?path=/user</code> where everything after <code class="language-plaintext highlighter-rouge">?path</code> is the location they should be redirected to on Galaxy. That example link will eventually redirect the learner to something like <code class="language-plaintext highlighter-rouge">https://usegalaxy.eu/user</code>.</p>   <h3 data-toc-skip id="technical-background">Technical Background</h3>   <p>So we took inspiration from <a href="https://my.home-assistant.io/">Home Assistant</a> which had the same problem, how to redirect users to pages on their own servers. The <code class="language-plaintext highlighter-rouge">my.galaxy.training</code> service is a very simple static page which looks in the user’s <a href="https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage"><code class="language-plaintext highlighter-rouge">localStorage</code></a> for their preferred server. If it’s not set, the user can click one of the common domains, and be redirected. When they access another link, they’ll be prompted to use a button that remembers which server they chose.</p>   <h3 data-toc-skip id="data-privacy">Data Privacy</h3>   <p>Any domain selected is not tracked nor communicated to any third party. Your preferred server is stored in your browser, and never transmitted to the GTN. That’s why we use localStorage instead of cookies.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What is this website? <a href="/training-material/faqs/gtn/gtn-what-is-it.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-is-this-website"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-is-this-website" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What is this website?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <!-- GTN:IGNORE:004 -->   <p>This website is a collection of hands-on tutorials that are designed to be interactive and are built around Galaxy:</p>   <p><img src="/training-material/shared/images/interactive_training.png" alt="Interactive training" title="The hands-on nature of our training material. It can be used with two web browser windows open side-by-side, one pointed at the current tutorial and the other at a Galaxy instance" /></p>   <p>This material is developed and maintained by the <a href="https://galaxyproject.org/">worldwide Galaxy community</a>. You can learn more about this effort by reading our <a href="https://doi.org/10.1016/j.cels.2018.05.012">first</a> and <a href="https://doi.org/10.1371/journal.pcbi.1010752">second articles</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What licenses are used in the GTN? <a href="/training-material/faqs/gtn/gtn_licenses.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-licenses-are-used-in-the-gtn"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-licenses-are-used-in-the-gtn" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What licenses are used in the GTN?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>We provide a listing of all licenses for code and things displayed to you in the GTN in <a href="/training-material/licenses.html">the licenses page</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why host your materials with the GTN? <a href="/training-material/faqs/gtn/why-the-gtn.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-host-your-materials-with-the-gtn"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-host-your-materials-with-the-gtn" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why host your materials with the GTN?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The short version is we’re a popular, FAIR training materials platform, and we want to be a home for your training materials.</p>   <h3 data-toc-skip id="your-content-in-front-of-the-world">Your content in front of the world</h3>   <p>As of June 2023 the GTN sees around 60k visitors per month. Please see <a href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=30d">our public page view monitoring</a> for more details.</p>   <h3 data-toc-skip id="fair">FAIR</h3>   <p>We go to great lengths to make sure our training platform is completely FAIR. See <a href="/training-material/faqs/gtn/fair_training.html">this FAQ for the details on how we achieve that.</a>. All of our materials have extensive <a href="https://bioschemas.org">BioSchemas</a> markup ensuring they’re easily accessible to search engines. Our materials are automatically indexed by <a href="https://tess.elixir-europe.org/materials?content_provider=Galaxy+Training">TeSS</a>, and we are working on a WorkflowHub integration.</p>   <h3 data-toc-skip id="accessible">Accessible</h3>   <p>We regularly test our pages with a thorough suite of accessibility tools, as well as via screen reader.</p>   <h3 data-toc-skip id="not-just-galaxy">Not Just Galaxy</h3>   <p>Our name can be a bit misleading! While a lot of our tutorials are focused on Galaxy, we have multiple growing topics which are unrelated to Galaxy.</p>   <ul>   <li><a href="/training-material/topics/ai4life/index.html">ai4life</a></li>   <li><a href="/training-material/topics/fair/index.html">FAIR</a></li>   <li><a href="/training-material/topics/data-science/index.html">Data Science</a></li> </ul>   <p>All of these topics are using the GTN as a platform to disseminate their materials far and wide.</p>   <h3 data-toc-skip id="features">Features</h3>   <p>Do you need</p>   <ul>   <li>Choose your own adventure tutorials</li>   <li>Automatic videos from your slides</li> </ul>   <p>Then choose the GTN.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="learners"> <a href="#learners"> Learners </a></h3>

<hr/>



<h4 class="faq-area"> How can I get help? <a href="/training-material/faqs/gtn/learners_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-get-help"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-get-help" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I get help?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you have questions about this training material, you can reach us using the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter chat</a>.  You’ll need a <a href="https://github.com/">GitHub</a> or <a href="https://twitter.com/">Twitter</a> account to post questions.  If you have questions about Galaxy outside the context of training, see the <a href="https://galaxyproject.org/support/">Galaxy Support page</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I use this material? <a href="/training-material/faqs/gtn/learners_using_materials.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-use-this-material"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-use-this-material" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I use this material?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Many topics include slide decks and if the topic you are interested in has slides then start there.  These will introduce the topic and important concepts.</p>   <p>Most of your learning will happen in the next step - the hands-on tutorials.  This is where you’ll become familiar with using the Galaxy interface and experiment with different ways to use Galaxy and the tools in Galaxy.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Is there a certification available for the FAIR-by-Design Methodology? <a href="/training-material/topics/contributing/tutorials/fair-by-design/faqs/FAQ-L1.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-is-there-a-certification-available-for-the-fair-by-design-methodology"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-is-there-a-certification-available-for-the-fair-by-design-methodology" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Is there a certification available for the FAIR-by-Design Methodology?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There is no specific certification regarding the implementation of the FAIR-by-Design Methodology when developing learning materials for GTN.</p>   <p>There is, however, the possibility to obtain FAIR-by-Design per stages and overall badge based on the Training of Trainers learning materials prepared by the Skills4EOSC project.</p>   <p>To obtain this certification follow the <a href="https://learning.skills4eosc.eu/course/view.php?id=19">FAIR-by-Design Learning Materials Methodology Course</a> and make sure you successfully pass all of the assessments.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using answer key histories <a href="/training-material/faqs/gtn/using-answer-key-histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-answer-key-histories-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-answer-key-histories-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using answer key histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you are struggling with a tutorial and you can’t figure out why, you can use the answer history (if it’s available) which is linked at the top of the tutorial.</p>   <ol>   <li>     <p>Select the <em>Answer history</em> and choose the appropriate link from the resulting drop-down menu.</p>   </li>   <li>     <p>Import the history (If you aren’t sure how, <a href="/training-material/faqs/galaxy/histories_import.html">go to the Import History FAQ</a>) 3. Select <em>View Here</em> or otherwise navigate to the imported history. 4. Find the dataset that was not outputting for you. You might be able to sport errors from looking at this dataset - such as, you may see that a different tool version was used, or different parameters were run, or different inputs were used. You may find that the input dataset itself is different, and perhaps something has gone wrong earlier in the tutorial for you.</p>   </li> </ol>   <p>If you can’t find what has gone wrong, you can then: 5. [Ask for help] (/training-material/faqs/gtn/instructors_getting_help.html).</p>   <ol>   <li><a href="/training-material/faqs/galaxy/histories_copy_dataset.html">Copy the necessary datasets over from the answer history into your history</a> so that you can continue your analysis from wherever you got stuck!</li> </ol>   <!-- It would be awesome if this was automatically added to any tutorial that has answer key histories --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where can I run the hands-on tutorials? <a href="/training-material/faqs/gtn/learners_running_tutorials.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-can-i-run-the-hands-on-tutorials"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-can-i-run-the-hands-on-tutorials" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where can I run the hands-on tutorials?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To run the hands-on tutorials you need a Galaxy server to run them on.</p>   <p>Each tutorial is annotated with information about which <a href="https://galaxyproject.org/public-galaxy-servers/">public Galaxy servers</a> it can be run on. These servers are available to anyone on the world wide web and some may have all the tools that are needed by a specific tutorial.</p>   <p>If your organization/consortia/community has its own Galaxy server, then you may  want to run tutorials on that. You will need to confirm that all necessary tools and reference genomes are available on your server and possible install missing tools and data. To learn how to do that, you can follow our <a href="/training-material/topics/teaching/tutorials/setup-galaxy-for-training/tutorial.html">dedicated tutorial</a>.</p>   <p>Some topics have a <a href="https://www.docker.com/">Docker</a> image that can be installed and run on participants’ laptops.  These Docker images contain Galaxy instances that include all tools and datasets used in a tutorial, as well as saved analyses and repeatable workflows that are relevant. You will need to <a href="https://docs.docker.com/install/">install Docker</a>.</p>   <p>Finally, you can also run your tutorials on cloud-based infrastructures.  Galaxy is <a href="https://galaxyproject.org/galaxy-services/">available on many national research infrastructures</a> such as <a href="https://galaxyproject.org/cloud/jetstream/">Jetstream</a> (United States), <a href="https://www.genap.ca/">GenAP</a> (Canada), <a href="https://www.gvl.org.au/">GVL</a> (Australia), <a href="http://www.climb.ac.uk/">CLIMB</a> (United Kingdom), and more.  These instances are typically easy to launch, and easy to shut down when you are done.</p>   <p>If you are already familiar with, and have an account on <a href="https://aws.amazon.com/">Amazon Web Services</a> then you can also launch a Galaxy server there using <a href="https://launch.usegalaxy.org/">CloudLaunch</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where do I start? <a href="/training-material/faqs/gtn/learners_getting_started.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-do-i-start-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-do-i-start-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where do I start?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you are new to Galaxy then start with one of the introductory topics.  These introduce you to concepts that are useful in Galaxy, no matter what domain you are doing analysis in.</p>   <p>If you are already familiar with Galaxy basics and want to learn how to use it in a particular domain (for example, ChIP-Seq), then start with one of those topics.</p>   <p>If you are already well informed about bioinformatics data analysis and you just want to get a feel for how it works in Galaxy, then many tutorials include <em>Instructions for the impatient</em> sections.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="mapping"> <a href="#mapping"> Mapping </a></h3>

<hr/>



<h4 class="faq-area"> Is it possible to visualize the RNA STAR bam file using the JBrowse tool? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-is-it-possible-to-visualize-the-rna-star-bam-file-using-the-jbrowse-tool"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Is it possible to visualize the RNA STAR bam file using the JBrowse tool?</div>   <p>Yes, that should work.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> RNAstar: Why do we set 36 for 'Length of the genomic sequence around annotated junctions'? <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-rnastar-why-do-we-set-36-for-length-of-the-genomic-sequence-around-annotated-junctions"><i class="far fa-question-circle" aria-hidden="true"></i> Question: RNAstar: Why do we set 36 for 'Length of the genomic sequence around annotated junctions'?</div>   <p>RNA STAR is using the gene model to create the database of splice junctions, and that these don’t “need” to have a length longer than the reads (37bp).</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="markdown"> <a href="#markdown"> Markdown </a></h3>

<hr/>



<h4 class="faq-area"> How can I create a tutorial skeleton from a Galaxy workflow? <a href="/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-create-a-tutorial-skeleton-from-a-galaxy-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-create-a-tutorial-skeleton-from-a-galaxy-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I create a tutorial skeleton from a Galaxy workflow?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are two ways to do this:</p>   <ol>   <li>Use <a href="https://planemo.readthedocs.io/en/latest/">planemo</a> on your local machine. Please see the tutorial named <a href="/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html">“Creating a new tutorial”</a> for detailed instructions.</li>   <li>Use our <a href="https://ptdk.apps.galaxyproject.eu/">web service</a></li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="notebooks"> <a href="#notebooks"> Notebooks </a></h3>

<hr/>



<h4 class="faq-area"> Contributing a Jupyter Notebook to the GTN <a href="/training-material/topics/contributing/faqs/contributing_a_notebook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-contributing-a-jupyter-notebook-to-the-gtn"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-contributing-a-jupyter-notebook-to-the-gtn" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Contributing a Jupyter Notebook to the GTN</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Problem: I have a notebook that I’d like to add to the GTN.</p>   <p>Solution: While we do not support directly adding notebooks to the GTN, as all of our notebooks are generated from the tutorial Markdown files, there is an alternative path! Instead you can:</p>   <ol>   <li>Install <a href="https://pypi.org/project/jupytext/"><code class="language-plaintext highlighter-rouge">jupytext</code></a></li>   <li>Use it to convert the ipynb file into a Markdown file (<code class="language-plaintext highlighter-rouge">jupytext notebook.ipynb --to markdown</code>)</li>   <li>Add this Markdown file to the GTN</li>   <li>Fix any missing header metadata</li> </ol>   <p>Then the GTN’s infrastructure will automatically convert that Markdown file directly to a notebook on deployment. This approach has the advantage that Markdown files are more diff-friendly than ipynb, making it much easier to review updates to a tutorial.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="other"> <a href="#other"> Other </a></h3>

<hr/>



<h4 class="faq-area"> Are there any upcoming events focused on Galaxy Training? <a href="/training-material/faqs/gtn/galaxy_events.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-are-there-any-upcoming-events-focused-on-galaxy-training"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-are-there-any-upcoming-events-focused-on-galaxy-training" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Are there any upcoming events focused on Galaxy Training?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Yes, always! Have a look at the <a href="https://galaxyproject.org/events/">Galaxy Community Events Calendar</a> for what coming up right now.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Compatible Versions of Galaxy <a href="/training-material/faqs/gtn/galaxy_tested_with.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="warning">   <div class="box-title warning-title" id="warning-compatible-versions-of-galaxy"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i> Warning: Compatible Versions of Galaxy</div>   <p>This tutorial may not be updated for the latest version of Galaxy.</p>   <ul>   <li>Galaxy’s Interface <strong>may</strong> be different to the Galaxy where you are following this tutorial.</li>   <li>✅ All tutorial steps will still be able to be followed (potentially with minor differences for moved buttons or changed icons.)</li>   <li>✅ Tools will all still work</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> GTN Stats <a href="/training-material/faqs/gtn/gtn_stats.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Statistics over the GTN
<!--SNIPPET--> <!-- tutorial stats --> <div class="row" style="color: var(--text-color-boxtitle)"> <div class="col-md-3">  <div class="gtn-card color-agenda">    <div class="card-title">33</div>    <div class="card-text">Topics</div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-tip">    <div class="card-title">435</div>    <div class="card-text">Tutorials</div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-handson">    <div class="card-title">20</div>    <div class="card-text">Learning Paths</div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-details">    <div class="card-title">462</div>    <div class="card-text"><abbr title="Frequently Asked Questions">FAQs</abbr></div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-comment">    <div class="card-title">433</div>    <div class="card-text">Contributors</div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-handson">    <div class="card-title">9.4</div>    <div class="card-text">Years</div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-handson">    <div class="card-title">98</div>    <div class="card-text">News Posts</div>  </div> </div> <div class="col-md-3">  <div class="gtn-card color-handson">    <div class="card-title">200</div>    <div class="card-text">Videos (135.0h)</div>  </div> </div> </div> <!--END_SNIPPET-->




<h4 class="faq-area"> Sustainability of the training-material and metadata <a href="/training-material/faqs/gtn/gtn_sustainability.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sustainability-of-the-training-material-and-metadata"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sustainability-of-the-training-material-and-metadata" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sustainability of the training-material and metadata</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This repository is hosted on <a href="https://github.com/">GitHub</a> using git as a <a href="https://en.wikipedia.org/wiki/Distributed_version_control">DVCS</a>. Therefore the community is hosting backups of this repository in a decentralised way. The repository is self-contained and contains all needed content and all metadata. In addition we mirror snapshops of this repo on <a href="http://zenodo.org">Zenodo</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Translations within the GTN <a href="/training-material/faqs/gtn/translations.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-translations-within-the-gtn"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-translations-within-the-gtn" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Translations within the GTN</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The GTN currently supports two forms of translation:</p>   <ul>   <li>Manual (<code class="language-plaintext highlighter-rouge">tutorial_ES.md</code> and <code class="language-plaintext highlighter-rouge">slides_ES.html</code> for example)</li>   <li>Automated (via linking through to Google Translate)</li> </ul>   <p>We accept manual translations if and only if there is a team that is able to commit to their maintenance. We need to ensure the trainings are kept up to date and high quality, but that requires native speakers of that language to maintain those translations.</p>   <p>Please contact us if you have any questions regarding translations.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="outputs"> <a href="#outputs"> Outputs </a></h3>

<hr/>



<h4 class="faq-area"> Does MaxQuant give as output possibility the PSMs and PEPs? <a href="/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/outputs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-does-maxquant-give-as-output-possibility-the-psms-and-peps"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Does MaxQuant give as output possibility the PSMs and PEPs?</div>   <p>Many output options, evidence &amp; msms contain e.g. PSM or feature level info</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="proteogenomics general"> <a href="#proteogenomics general"> Proteogenomics general </a></h3>

<hr/>



<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_own_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can I use these workflows on datasets generated from our laboratory?</div>   <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_own_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can I use these workflows on datasets generated from our laboratory?</div>   <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can I use these workflows on datasets generated from our laboratory? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_own_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-can-i-use-these-workflows-on-datasets-generated-from-our-laboratory-2"><i class="far fa-question-circle" aria-hidden="true"></i> Question: Can I use these workflows on datasets generated from our laboratory?</div>   <p>Yes, the workflows can be used on other datasets as well. However, you will need to consider data acquisition and sample preparation methods so that the tool parameters can be adjusted accordingly.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Example histories for the proteogenomics tutorials <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-example-histories-for-the-proteogenomics-tutorials"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-example-histories-for-the-proteogenomics-tutorials" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Example histories for the proteogenomics tutorials</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you get stuck or would like to see what the results should look like, you can have a look at one of the following public histories:</p>   <p>Galaxy EU (usegalaxy.eu):</p>   <ul>   <li>History1: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-1-database-creation">Part 1: database creation</a></li>   <li>History2: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-2-database-search">Part 2: database search</a></li>   <li>History3: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-3-novel-peptide-analysis">Part3: novel peptide analysis</a></li> </ul>   <p>Galaxy Main (usegalaxy.org):</p>   <ul>   <li>History 1: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-1-database-creation">Part 1: database creation</a></li>   <li>History 2: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-2-database-search">Part 2: database search</a></li>   <li>History 3: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-3-novel-peptide-analysis">Part 3: novel peptide analysis</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Example histories for the proteogenomics tutorials <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-example-histories-for-the-proteogenomics-tutorials-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-example-histories-for-the-proteogenomics-tutorials-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Example histories for the proteogenomics tutorials</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you get stuck or would like to see what the results should look like, you can have a look at one of the following public histories:</p>   <p>Galaxy EU (usegalaxy.eu):</p>   <ul>   <li>History1: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-1-database-creation">Part 1: database creation</a></li>   <li>History2: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-2-database-search">Part 2: database search</a></li>   <li>History3: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-3-novel-peptide-analysis">Part3: novel peptide analysis</a></li> </ul>   <p>Galaxy Main (usegalaxy.org):</p>   <ul>   <li>History 1: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-1-database-creation">Part 1: database creation</a></li>   <li>History 2: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-2-database-search">Part 2: database search</a></li>   <li>History 3: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-3-novel-peptide-analysis">Part 3: novel peptide analysis</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Example histories for the proteogenomics tutorials <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_example_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-example-histories-for-the-proteogenomics-tutorials-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-example-histories-for-the-proteogenomics-tutorials-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Example histories for the proteogenomics tutorials</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you get stuck or would like to see what the results should look like, you can have a look at one of the following public histories:</p>   <p>Galaxy EU (usegalaxy.eu):</p>   <ul>   <li>History1: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-1-database-creation">Part 1: database creation</a></li>   <li>History2: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-2-database-search">Part 2: database search</a></li>   <li>History3: <a href="https://usegalaxy.eu/u/galaxyp/h/proteogenomics-3-novel-peptide-analysis">Part3: novel peptide analysis</a></li> </ul>   <p>Galaxy Main (usegalaxy.org):</p>   <ul>   <li>History 1: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-1-database-creation">Part 1: database creation</a></li>   <li>History 2: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-2-database-search">Part 2: database search</a></li>   <li>History 3: <a href="https://usegalaxy.org/u/nate/h/imported-from-usegalaxyeu-proteogenomics-3-novel-peptide-analysis">Part 3: novel peptide analysis</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The workflows contain several Query tabular for text manipulation, is there a tutorial for that? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_querytabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-the-workflows-contain-several-query-tabular-for-text-manipulation-is-there-a-tutorial-for-that"><i class="far fa-question-circle" aria-hidden="true"></i> Question: The workflows contain several Query tabular for text manipulation, is there a tutorial for that?</div>   <p>Query tabular loads a tabular database and creates a sqlite database and tabular file. To learn more about SQL Queries - please look at <a href="https://www.sqlite.org/docs.html">this documentation</a>.</p>   <p>The help section on the Query Tabular tool provides simple examples of both filtering the input tabular datasets, as well as examples of SQL queries.  Query Tabular also incorporates regex functions that can be used queries.  The PSM report datasets in these tutorials have fields that are lists of protein IDs.</p>   <p>Query Tabular help shows how to normalize those protein list fields so that we can perform operations by protein ID.  See section:  <em>Normalizing by Line Filtering into 2 Tables</em> in the tool help (below the tool in Galaxy).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The workflows contain several Query tabular for text manipulation, is there a tutorial for that? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_querytabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-the-workflows-contain-several-query-tabular-for-text-manipulation-is-there-a-tutorial-for-that-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question: The workflows contain several Query tabular for text manipulation, is there a tutorial for that?</div>   <p>Query tabular loads a tabular database and creates a sqlite database and tabular file. To learn more about SQL Queries - please look at <a href="https://www.sqlite.org/docs.html">this documentation</a>.</p>   <p>The help section on the Query Tabular tool provides simple examples of both filtering the input tabular datasets, as well as examples of SQL queries.  Query Tabular also incorporates regex functions that can be used queries.  The PSM report datasets in these tutorials have fields that are lists of protein IDs.</p>   <p>Query Tabular help shows how to normalize those protein list fields so that we can perform operations by protein ID.  See section:  <em>Normalizing by Line Filtering into 2 Tables</em> in the tool help (below the tool in Galaxy).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> The workflows contain several Query tabular for text manipulation, is there a tutorial for that? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_querytabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-the-workflows-contain-several-query-tabular-for-text-manipulation-is-there-a-tutorial-for-that-2"><i class="far fa-question-circle" aria-hidden="true"></i> Question: The workflows contain several Query tabular for text manipulation, is there a tutorial for that?</div>   <p>Query tabular loads a tabular database and creates a sqlite database and tabular file. To learn more about SQL Queries - please look at <a href="https://www.sqlite.org/docs.html">this documentation</a>.</p>   <p>The help section on the Query Tabular tool provides simple examples of both filtering the input tabular datasets, as well as examples of SQL queries.  Query Tabular also incorporates regex functions that can be used queries.  The PSM report datasets in these tutorials have fields that are lists of protein IDs.</p>   <p>Query Tabular help shows how to normalize those protein list fields so that we can perform operations by protein ID.  See section:  <em>Normalizing by Line Filtering into 2 Tables</em> in the tool help (below the tool in Galaxy).</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What kind of variants are seen in the output? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_variants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-kind-of-variants-are-seen-in-the-output"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What kind of variants are seen in the output?</div>   <p>From this workflow we can see insertions, deletions, SNVs, or we will know whether it’s an intron, exon, splice junction etc.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What kind of variants are seen in the output? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_variants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-kind-of-variants-are-seen-in-the-output-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What kind of variants are seen in the output?</div>   <p>From this workflow we can see insertions, deletions, SNVs, or we will know whether it’s an intron, exon, splice junction etc.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What kind of variants are seen in the output? <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_variants.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-what-kind-of-variants-are-seen-in-the-output-2"><i class="far fa-question-circle" aria-hidden="true"></i> Question: What kind of variants are seen in the output?</div>   <p>From this workflow we can see insertions, deletions, SNVs, or we will know whether it’s an intron, exon, splice junction etc.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="recommended browser"> <a href="#recommended browser"> Recommended browser </a></h3>

<hr/>



<h4 class="faq-area"> UCSC - I fetched data from a remote website but now I’m logged out of Galaxy and my data is gone? <a href="/training-material/topics/introduction/faqs/fetched_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-ucsc-i-fetched-data-from-a-remote-website-but-now-i-m-logged-out-of-galaxy-and-my-data-is-gone"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-ucsc-i-fetched-data-from-a-remote-website-but-now-i-m-logged-out-of-galaxy-and-my-data-is-gone" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: UCSC - I fetched data from a remote website but now I’m logged out of Galaxy and my data is gone?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This is a known bug with Chrome + Galaxy, we’re working on it <a href="https://github.com/galaxyproject/galaxy/issues/11374">galaxyproject/galaxy#11374</a>. For now we can recommend using Firefox (known to work) or trying another browser.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="reference genomes"> <a href="#reference genomes"> Reference genomes </a></h3>

<hr/>



<h4 class="faq-area"> How to use Custom Reference Genomes? <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-use-custom-reference-genomes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-use-custom-reference-genomes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to use Custom Reference Genomes?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A <strong>reference genome</strong> contains the nucleotide sequence of the chromosomes, scaffolds, transcripts, or contigs for single species. It is representative of a specific genome assembly build or release.</p>   <p>There are two options for reference genomes in Galaxy.</p>   <ul>   <li><strong>Native</strong>     <ul>       <li>Index provided by the server administrators.</li>       <li>Found on tool forms in a drop down menu.</li>       <li>A database key is automatically assigned. See tip 1.</li>       <li>The database is what links your data to a FASTA index. Example: used with BAM data</li>     </ul>   </li>   <li><strong>Custom</strong>     <ul>       <li>FASTA file uploaded by users.</li>       <li>Input on tool forms then indexed at runtime by the tool.</li>       <li>An optional custom database key can be created and <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">assigned by the user</a>.</li>     </ul>   </li> </ul>   <p>There are five basic steps to use a <strong>Custom Reference Genome</strong>, plus one optional.</p>   <ol>   <li>Obtain a FASTA copy of the target genome. See tip 2.</li>   <li>Upload the genome to Galaxy and to add it as a dataset in your history.</li>   <li><a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">Clean up the format</a> with the tool <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace.</li>   <li>Make sure the <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html">chromosome identifiers</a> are a match for other inputs.</li>   <li>Set a tool form’s options to use a custom reference genome from the history and select the loaded genome FASTA.</li>   <li>(Optional) Create a <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">custom genome build’s database</a> that you can <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">assign to datasets</a>.</li> </ol>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> TIP 1: Avoid <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">assigning a native database</a> to uploaded data unless you confirmed the data are based on the <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html">same exact genome assembly</a> or you <a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">adjusted the data to be a match</a> <strong>first</strong>!</p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> TIP 2: When choosing your reference genome, consider <a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html">choosing your reference annotation</a> at the same time. Standardize the format of both as a preparation step. Put the files in a dedicated “reference data” history for easy reuse.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Sorting Reference Genome <a href="/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sorting-reference-genome"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sorting-reference-genome" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sorting Reference Genome</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Certain tools expect that reference genomes are sorted in <a href="https://en.wikipedia.org/wiki/Lexicographic_order">lexicographical order</a>. These tools are often downstream of the initial mapping tools, which means that a large investment in a project has already been made, before a problem with sorting pops up in conclusion layer tools. How to avoid? Always sort your FASTA reference genome dataset at the beginning of a project. Many sources only provide sorted genomes, but double checking is your own responsibility, and super easy in Galaxy!</p>   <ol>   <li>Convert Formats -&gt; <strong>FASTA-to-Tabular</strong></li>   <li>Filter and Sort -&gt; <strong>Sort</strong>     on column: c1      with flavor: Alphabetical     everything in: Ascending order</li>   <li>Convert Formats -&gt; <strong>Tabular-to-FASTA</strong></li> </ol>   <p>Note: The above sorting method is for most tools, but not all. In particular, GATK tools have a tool-specific sort order requirement.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Troubleshooting Custom Genome fasta <a href="/training-material/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-custom-genome-fasta"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-custom-genome-fasta" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Troubleshooting Custom Genome fasta</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If a custom genome/transcriptome/exome dataset is producing errors, double check the format and that the chromosome identifiers between <strong>ALL</strong> inputs. Clicking on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> will often provide a description of the problem. This does not automatically submit a bug report, and it is not always necessary to do so, but it is a good way to get some information about <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">why a job is failing</a>.</p>   <ul>   <li>     <p>Custom genome not assigned as FASTA format</p>     <ul>       <li><strong>Symptoms include</strong>: Dataset not included in custom genome “From history” pull down menu on tool forms.</li>       <li><strong>Solution</strong>: Check datatype assigned to dataset and assign <strong>fasta</strong> format.</li>       <li><strong>How</strong>: Click on the dataset’s pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the “Edit Attributes” form, and in the Datatypes tab &gt; <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">redetect the datatype</a>.</li>       <li>If <code class="language-plaintext highlighter-rouge">fasta</code> is not assigned, there is a format problem to correct.</li>     </ul>   </li>   <li>     <p>Incomplete Custom genome file load</p>     <ul>       <li><strong>Symptoms include</strong>: Tool errors result the first time you use the Custom genome.</li>       <li><strong>Solution</strong>: Use <strong>Text Manipulation → Select last lines from a dataset</strong> to check last 10 lines to see if file is truncated.</li>       <li><strong>How</strong>: Reload the dataset (switch to <a href="https://galaxyproject.org/ftp-upload/">FTP</a> if not using already). Check your FTP client logs to make sure the load is complete.</li>     </ul>   </li>   <li>     <p>Extra spaces, extra lines, inconsistent line wrapping, or any deviation from strict FASTA format</p>     <ul>       <li><strong>Symptoms include: RNA-seq tools (Cufflinks, Cuffcompare, Cuffmerge, Cuffdiff)</strong> fails with error <code class="language-plaintext highlighter-rouge">Error: sequence lines in a FASTA record must have the same length!.</code></li>       <li><strong>Solution</strong>: File tested and corrected locally then re-upload or test/fix within Galaxy, then re-run.</li>       <li><strong>How</strong>:         <ul>           <li><strong>Quick re-formatting</strong> Run the tool through the tool <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace.</li>           <li><strong>Optional Detailed re-formatting</strong> Start with <strong>FASTA manipulation → FASTA Width formatter</strong> with a value between 40-80 (60 is common) to reformat wrapping. Next, use Filter and Sort → Select with “&gt;” to examine identifiers. Use a combination of <strong>Convert Formats → FASTA-to-Tabular, Text Manipulation</strong> tools, then <strong>Tabular-to-FASTA</strong> to correct.</li>           <li><strong>With either of the above</strong>, finish by using <strong>Filter and Sort → Select</strong> with <code class="language-plaintext highlighter-rouge">^\w*$</code> to search for empty lines (use “NOT matching” to remove these lines and output a properly format fasta dataset).</li>         </ul>       </li>     </ul>   </li>   <li>     <p>Inconsistent line wrapping, common if merging chromosomes from various Genbank records (e.g. primary chroms with mito)</p>     <ul>       <li><strong>Symptoms include</strong>: Tools (<strong>SAMTools, Extract Genomic DNA</strong>, but rarely alignment tools) may complain about unexpected line lengths/missing identifiers. Or they may just fail for what appears to be a cluster error.</li>       <li><strong>Solution</strong>: File tested and corrected locally then re-upload or test/fix within Galaxy.</li>       <li><strong>How</strong>: Use <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace. Finish by using <strong>Filter and Sort → Select</strong> with <code class="language-plaintext highlighter-rouge">^\w*$</code> to search for empty lines (use “NOT matching” to remove these lines and output a properly format fasta dataset).</li>     </ul>   </li>   <li>     <p>Unsorted fasta genome file</p>     <ul>       <li><strong>Symptoms include</strong>: Tools such as <strong>Extract Genomic DNA</strong> report problems with sequence lengths.</li>       <li><strong>Solution</strong>: First try sorting and re-formatting in Galaxy then re-run.</li>       <li><strong>How</strong>: To sort, follow instructions for <a href="/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html">Sorting</a> a Custom Genome.</li>     </ul>   </li>   <li>     <p>Identifier and Description in “&gt;” title lines used inconsistently by tools in the same analysis</p>     <ul>       <li><strong>Symptoms include</strong>: Will generally manifest as a false genome-mismatch problem.</li>       <li><strong>Solution</strong>: Remove the description content and re-run all tools/workflows that used this input. Mapping tools will usually not fail, but downstream tools will. When this comes up, it usually means that an analysis needs to be started over from the mapping step to correct the problems. No one enjoys redoing this work. Avoid the problems by formatting the genome, by double checking that the same reference genome was used for all steps, and by making certain the ‘identifiers’ are a match between all planned inputs (including reference annotation such as GTF data) <strong>before using your custom genome</strong>.</li>       <li><strong>How</strong>: To drop the title line description content, use <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace. Next, double check that the chromosome identifiers are an exact match between all inputs.</li>     </ul>   </li>   <li>     <p>Unassigned database</p>     <ul>       <li><strong>Symptoms include</strong>: Tools report that no build is available for the assigned reference genome.</li>       <li><strong>Solution</strong>: This occurs with tools that require an assigned database metadata attribute. <strong>SAMTools and Picard</strong> often require this assignment.</li>       <li><strong>How</strong>: Create a <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">Custom Build</a> and assign it to the dataset.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="reports"> <a href="#reports"> Reports </a></h3>

<hr/>



<h4 class="faq-area"> Enhancing tabular dataset previews in reports/pages <a href="/training-material/faqs/galaxy/reports_tables.html"><i class="fas fa-external-link-alt"></i></a> </h4>
There are lots of fun advanced features!
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-enhancing-tabular-dataset-previews-in-reports-pages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-enhancing-tabular-dataset-previews-in-reports-pages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Enhancing tabular dataset previews in reports/pages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are a number of options, specifically for tabular data, that can allow it to render more nicely in your workflow reports and pages and anywhere that GalaxyMarkdown is used.</p>   <ul>   <li><code class="language-plaintext highlighter-rouge">title</code> to give your table a title</li>   <li><code class="language-plaintext highlighter-rouge">footer</code> allows you to caption your table</li>   <li><code class="language-plaintext highlighter-rouge">show_column_headers=false</code> to hide the column headers</li>   <li><code class="language-plaintext highlighter-rouge">compact=true</code> to make the table show up more inline, hiding that it was embedded from a Galaxy dataset.</li> </ul>   <p>The existing <code class="language-plaintext highlighter-rouge">history_dataset_display</code> directive displays the dataset name and some useful context at the expense of potentially breaking the flow of the document</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_display(history_dataset_id=1e8ab44153008be8) </span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_display.png" alt="a tabular dataset rendered, it has a title and a download button and sortable columns" /></p> </blockquote>   <p>The existing <code class="language-plaintext highlighter-rouge">history_dataset_embedded</code> directive was implemented to try to inline results more and make the results more readable within a more… curated document. It is dispatches on tabular types and puts the results in a table but the table doesn’t have a lot of options.</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-1"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_embedded(history_dataset_id=1e8ab44153008be8) </span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-1"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_embedded.png" alt="the same as before but no title nor download button. just a rendered table with sortable columns" /></p> </blockquote>   <p>The <code class="language-plaintext highlighter-rouge">history_dataset_as_table</code> directive mirrors the <code class="language-plaintext highlighter-rouge">history_dataset_as_image</code> directive: it tries harder to coerce the data into a table and provides new table—specific options. The first of these is “show_column_headers<code class="language-plaintext highlighter-rouge"> which defaults to </code>true`.</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-2"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_as_table(history_dataset_id=1e8ab44153008be8,show_column_headers=false)</span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-2"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_as_table.png" alt="the same as before but no title nor download button nor column headers" /></p> </blockquote>   <p>There is also a <code class="language-plaintext highlighter-rouge">compact</code> option. This provides a much more inline experience for tabular datasets:</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-3"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_as_table(history_dataset_id=1e8ab44153008be8,show_column_headers=false,compact=true)</span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-3"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_compact.png" alt="again the same screenshot, no table metadata, and now it lacks the small margin around it." /></p> </blockquote>   <p>Figures in general should have titles and legends — so there is the “title” and “footer” options also.</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-4"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_as_table(history_dataset_id=1e8ab44153008be8,show_column_headers=false,title='Binding Site Results',footer='Here is a very good figure caption for this table.')</span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-4"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_title.png" alt="the same table with now a tasteful title and small caption below it describing that the author would write a caption if he knew what a binding site was." /></p> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Making an element collapsible in a report <a href="/training-material/faqs/galaxy/reports_collapsible_box.html"><i class="fas fa-external-link-alt"></i></a> </h4>
If you have extraneous information you might want to let a user collapse it.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-making-an-element-collapsible-in-a-report"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-making-an-element-collapsible-in-a-report" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Making an element collapsible in a report</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This applies to any GalaxyMarkdown elements, i.e. the things you’ve clicked in the left panel to embed in your Workflow Report or Page</p>   <p>By adding a <code class="language-plaintext highlighter-rouge">collapse=""</code> attribute to a markdown element, you can make it collapsible. Whatever you put in the quotes will be the title of the collapsible box.</p>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl"><br></span>history_dataset_type(history_dataset_id=3108c91feeb505da, collapse="[TITLE]")<br><span class="p">```</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="rule-builder"> <a href="#rule-builder"> Rule-builder </a></h3>

<hr/>



<h4 class="faq-area"> Flatten a list of list of paired datasets into a list of paired datasets <a href="/training-material/topics/galaxy-interface/faqs/flatten-list-list-paired-to-list-paired.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-flatten-a-list-of-list-of-paired-datasets-into-a-list-of-paired-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-flatten-a-list-of-list-of-paired-datasets-into-a-list-of-paired-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Flatten a list of list of paired datasets into a list of paired datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes you find yourself with a <code class="language-plaintext highlighter-rouge">list:list:paired</code>, i.e. a collection of collection of paired end data, and you really want a <code class="language-plaintext highlighter-rouge">list:paired</code>, a flatter collection of paired end data. This is easy to resolve with <span class="tool" data-tool="__APPLY_RULES__" title="Apply rules tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Apply rules</strong></span>:</p>   <ol>   <li>Open <span class="tool" data-tool="__APPLY_RULES__" title="Apply rules tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Apply rules</strong></span></li>   <li>Select your collection</li>   <li>Click <strong>Edit</strong></li> </ol>   <p>You’ll now be in the Apply rules editing interface. There are three columns (if it’s a <code class="language-plaintext highlighter-rouge">list:list:paired</code>)</p>   <ol>   <li>The outermost <code class="language-plaintext highlighter-rouge">list</code> identifier(s)</li>   <li>The next <code class="language-plaintext highlighter-rouge">list</code> identifier(s)</li>   <li>The paired-end indicator</li> </ol>   <p>Flattening this top level list, so it’s just a <code class="language-plaintext highlighter-rouge">list:paired</code> requires a few changes:</p>   <ol>   <li>From <strong>Column</strong> menu select <code class="language-plaintext highlighter-rouge">Concatenate Columns</code>     <ul>       <li><em>“From Column”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>       <li><em>“From Column”</em>: <code class="language-plaintext highlighter-rouge">B</code> This creates a column with the top list identifier, and the inner list identifier, which will be our new list identifier for the flattened list.</li>     </ul>   </li>   <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>     <ul>       <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Paired-end Indicator</code>         <ul>           <li><em>“Paired-end Indicator”</em>: <code class="language-plaintext highlighter-rouge">C</code></li>         </ul>       </li>       <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>         <ul>           <li><em>“List Identifier(s)”</em>: <code class="language-plaintext highlighter-rouge">D</code></li>         </ul>       </li>       <li>Click Apply</li>     </ul>   </li>   <li>Click <strong>Save</strong></li>   <li>Click <strong>Run Tool</strong></li> </ol>   <p>The tool will execute and reshape your list, congratulations!</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="sequencing"> <a href="#sequencing"> Sequencing </a></h3>

<hr/>



<h4 class="faq-area"> Illumina MiSeq sequencing <a href="/training-material/faqs/galaxy/sequencing_illumina_miseq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-illumina-miseq-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Illumina MiSeq sequencing</div>   <p>Illumina MiSeq sequencing is based on sequencing by synthesis. As the name suggests, fluorescent labels are measured for every base that bind at a specific moment at a specific place on a flow cell. These flow cells are covered with oligos (small single strand DNA strands). In the library preparation the DNA strands are cut into small DNA fragments (differs per kit/device) and specific pieces of DNA (adapters) are added, which are complementary to the oligos. Using bridge amplification large amounts of clusters of these DNA fragments are made. The reverse string is washed away, making the clusters single stranded. Fluorescent bases are added one by one, which emit a specific light for different bases when added. This is happening for whole clusters, so this light can be detected and this data is basecalled (translation from light to a nucleotide) to a nucleotide sequence (Read). For every base a quality score is determined and also saved per read. This process is repeated for the reverse strand on the same place on the flow cell, so the forward and reverse reads are from the same DNA strand. The forward and reversed reads are linked together and should always be processed together!</p>   <p>For more information watch this <a href="https://youtu.be/fCd6B5HRaZ8">video from Illumina</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Nanopore sequencing <a href="/training-material/faqs/galaxy/sequencing_nanopore.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Nanopore sequencing</div>   <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>   <ol>   <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>   <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>   <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li> </ol>   <p><img src="/training-material/topics/microbiome/images/plasmid-metagenomics-nanopore/nanopore_sequencing.png" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://doi.org/10.7875/togopic.2020.01&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>   <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change in electrical current is measured over the membrane of a flow cell. When nucleotides pass the pores in the flow cell the current change is translated (basecalled) to nucleotides by a basecaller. A schematic overview is given in the picture above.</p>   <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software is present on the devices. With basecalling the electrical signals are translated to bases (A,T,G,C) with a quality score per base. The sequenced DNA strand will be basecalled and this will form one read. Multiple reads will be stored in a fastq file.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="support"> <a href="#support"> Support </a></h3>

<hr/>



<h4 class="faq-area"> Can I use a public Galaxy for my private data? <a href="/training-material/faqs/galaxy/can-i-use-a-public-galaxy-for-my-private-data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-i-use-a-public-galaxy-for-my-private-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-i-use-a-public-galaxy-for-my-private-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can I use a public Galaxy for my private data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Of course*!</p>   <p>If your data is not sensitive (i.e. human patient) but just private (sequencing from other animals/bacteria/etc), then it is absolutely ok to use a public galaxy server like usegalaxy.eu or usegalaxy.org!</p>   <p>Data uploaded is private to your account, it isn’t available to others publicly. No one will scoop your results, if you use a public galaxy server to analyse your data :)</p>   <p>A great benefit of this is then when your paper is being reviewed you can share that history or workflow with reviewers, and when it’s published you can click a button to share those results with the world as well, such that others can reproduce your analysis!</p>   <p>(of course system administrators can see the files on disk but they are not interested and will not be looking at your data. If you file a bug report they may see your data but they are system administrators, not bioinformatics experts that might be interested in your results.)</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Contacting Galaxy Administrators <a href="/training-material/faqs/galaxy/support_admins.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-contacting-galaxy-administrators"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-contacting-galaxy-administrators" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Contacting Galaxy Administrators</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you suspect there is something wrong with the server, or would like to request a tool to be installed, you should contact the server administrators for the Galaxy you are on.</p>   <ul>   <li><strong>Tool error?</strong> Please follow these <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">troubleshooting steps</a></li>   <li>Each Galaxy server has different contact procedures, here are the contact options for the 3 biggest servers:     <ul>       <li><strong>Galaxy US:</strong> <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a></li>       <li><strong>Galaxy EU:</strong> <a href="https://gitter.im/usegalaxy-eu/Lobby">Gitter channel</a>, <a href="https://usegalaxy.eu/tiaas/">Request TIaaS</a></li>       <li><strong>Galaxy AU:</strong> <a href="mail:help@genome.edu.au">Email</a>, <a href="https://request.usegalaxy.org.au/">Request a tool</a>, <a href="https://docs.google.com/forms/d/e/1FAIpQLSeiw6ajmkezLCwbXc3OFQEU3Ai9hGnBd967u9YbQ8ANPgvatA/viewform">Request Data Quota</a></li>       <li><strong>Galaxy FR:</strong> <a href="https://usegalaxy.fr/tiaas/">Request TIaaS</a></li>       <li>Other Galaxy servers? Check the homepage for more information.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why? <a href="/training-material/topics/transcriptomics/tutorials/de-novo/faqs/different_results.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="question">   <div class="box-title question-title" id="question-i-get-a-different-number-of-transcripts-with-a-significant-change-in-gene-expression-between-the-g1e-and-megakaryocyte-cellular-states-why"><i class="far fa-question-circle" aria-hidden="true"></i> Question: I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why?</div>   <p>This is okay! Many aspects of the tutorial can potentially affect the exact results you obtain. For example, the reference genome version used and versions of tools. It’s less important to get the exact results shown in the tutorial, and more important to understand the concepts so you can apply them to your own data.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where do I get more support? <a href="/training-material/faqs/galaxy/support.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-do-i-get-more-support"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-do-i-get-more-support" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where do I get more support?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you need support for using Galaxy, running your analysis or completing a tutorial, please try one of the following options:</p>   <ul>   <li><strong>Gitter Chat</strong>: You can get help on Gitter chat platform, on various channels.     <ul>       <li><a href="https://gitter.im/galaxyproject/Lobby">Galaxy General Support</a></li>       <li><a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Training Support</a></li>       <li><a href="https://gitter.im/usegalaxy-eu/Lobby">Galaxy EU Server</a></li>     </ul>   </li>   <li>     <p><strong>Galaxy Help Forum</strong>: You can also have a look at the <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>. Your question may already have been answered here before. If not, you can post your question here.</p>   </li>   <li><strong>Contact Server Admins:</strong> If you think there is a problem with the Galaxy server, or you would like to make a request, <a href="/training-material/faqs/galaxy/support_admins.html">contact the Galaxy server administrators</a>.</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="tips"> <a href="#tips"> Tips </a></h3>

<hr/>



<h4 class="faq-area"> Opening a split screen in byobu <a href="/training-material/topics/admin/faqs/byobu-shortcuts.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-opening-a-split-screen-in-byobu"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-opening-a-split-screen-in-byobu" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Opening a split screen in byobu</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <div class="box-title tip-title" id="tip-opening-a-split-screen-in-byobu-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-opening-a-split-screen-in-byobu-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Opening a split screen in byobu</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><kbd>Shift-F2</kbd>: Create a horizontal split</p>   <p><kbd>Shift-Left/Right/Up/Down</kbd>: Move focus among splits</p>   <p><kbd>Ctrl-F6</kbd>:  Close split in focus</p>   <p><kbd>Ctrl-D</kbd>:  (Linux, Mac users) Close split in focus</p>   <p>There are more byobu commands described in this <a href="https://gist.github.com/devhero/7b9a7281db0ac4ba683f">gist</a></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="tools"> <a href="#tools"> Tools </a></h3>

<hr/>



<h4 class="faq-area"> Add Toolshed category to a tool <a href="/training-material/faqs/galaxy/add-toolshed-category-to-a-tool.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-add-toolshed-category-to-a-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-toolshed-category-to-a-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Add Toolshed category to a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p>Find the target tool in the <a href="https://toolshed.g2.bx.psu.edu/repository">Galaxy Toolshed</a>.</p>     <p>Note: the easiest way to do this from the Galaxy interface is to (A) search for the tool, then (B) select the drop-down menu <strong>See tool in toolshed</strong>.</p>   </li>   <li>Follow the <strong>Development respository</strong> URL.</li>   <li>Go to the <code class="language-plaintext highlighter-rouge">.shed.yml</code> file.</li>   <li>In the <code class="language-plaintext highlighter-rouge">categories:</code> metadata section, add your Toolshed category (which must correspond to those already in the <a href="https://toolshed.g2.bx.psu.edu/repository">Galaxy Toolshed</a>.</li> </ol>   <p>Example format:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">categories</span><span class="pi">:</span><br> <span class="pi">-</span> <span class="s">Single Cell</span><br> <span class="pi">-</span> <span class="s">Spatial Omics</span><br> <span class="pi">-</span> <span class="s">Transcriptomics</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the tool version <a href="/training-material/faqs/galaxy/tools_change_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-tool-version-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-tool-version-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the tool version</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool.</p>   <p><strong>Switching to a different version</strong> of a tool:</p>   <ul>   <li>Open the tool</li>   <li>Click on the <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> versions logo at the <strong>top right</strong></li>   <li>Select the desired version from the dropdown list</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> If a Tool is Missing <a href="/training-material/faqs/galaxy/tools_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-if-a-tool-is-missing"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-if-a-tool-is-missing" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: If a Tool is Missing</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To use the tools installed and available on the Galaxy server:</p>   <ol>   <li>At the top of the left tool panel, type in a tool name or datatype into the tool search box.</li>   <li>Shorter keywords find more choices.</li>   <li>Tools can also be directly browsed by category in the tool panel.</li> </ol>   <p>If you can’t find a tool you need for a tutorial on Galaxy, please:</p>   <ol>   <li><strong>Check that you are using a compatible Galaxy server</strong>     <ul>       <li>Navigate to the overview box at the top of the tutorial</li>       <li>Find the <strong>“Supporting Materials”</strong> section</li>       <li>Check <strong>“Available on these Galaxies”</strong></li>       <li>If your server is not listed here, the tutorial is not supported on your Galaxy server</li>       <li>You can create an account on one of the supporting Galaxies   <img src="/training-material/faqs/galaxy/images/supporting_galaxies.png" alt="screenshot of overview box with available Galaxies section" /></li>     </ul>   </li>   <li>Use the <strong>Tutorial mode</strong> feature     <ul>       <li>Open your Galaxy server</li>       <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>       <li>Navigate to your tutorial</li>       <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>       <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>     </ul>   </li>   <li><strong>Still not finding the tool?</strong>     <ul>       <li>Ask help in <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a>.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Multipile similar tools available <a href="/training-material/faqs/galaxy/tools_similar.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-multipile-similar-tools-available"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-multipile-similar-tools-available" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Multipile similar tools available</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes there are multiple tools with very similar names. If the parameters in the tutorial don’t match with what you see in Galaxy, please try the following:</p>   <ol>   <li>     <p>Use <strong>Tutorial Mode</strong> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> in Galaxy, and click on the blue tool button in the tutorial to automatically open the correct tool and version (not available for all tutorials yet)</p>     <!--SNIPPET-->     <blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>     <p><!--END_SNIPPET--></p>   </li>   <li>     <p>Check that the <strong>entire tool name</strong> matches what you see in the tutorial.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Organizing the tool panel <a href="/training-material/faqs/galaxy/tools_favorite.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-organizing-the-tool-panel"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-organizing-the-tool-panel" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Organizing the tool panel</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy servers can have a lot of tools available, which can make it challenging to find the tool you are looking for. To help find your favourite tools, you can:</p>   <ul>   <li><strong>Keep a list of your favourite tools</strong> to find them back easily later.     <ul>       <li><strong>Adding tools to your favourites</strong>         <ul>           <li>Open a tool</li>           <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> next to the tool name to add it to your favourites</li>         </ul>       </li>       <li><strong>Viewing your favourite tools</strong>         <ul>           <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> at the top of the Galaxy tool panel (above the tool search bar)</li>           <li>This will filter the toolbox to show all your starred tools</li>         </ul>       </li>     </ul>   </li>   <li><strong>Change the tool panel view</strong>     <ul>       <li>Click on the <i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden">galaxy-panelview</span> icon at the top of the Galaxy tool panel (above the tool search bar)</li>       <li>Here you can view the tools by EDAM ontology terms         <ul>           <li>EDAM Topics (e.g. biology, ecology)</li>           <li>EDAM Operations (e.g. quality control, variant analysis)</li>           <li>You can always get back to the default view by choosing <strong>“Full Tool Panel”</strong></li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Para volver a ejecutar una herramienta <a href="/training-material/faqs/galaxy-es/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-para-volver-a-ejecutar-una-herramienta"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-para-volver-a-ejecutar-una-herramienta" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Para volver a ejecutar una herramienta</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expande</strong> uno de los conjuntos de datos de la salida de la herramienta haciendo clic sobre él</li>   <li><strong>Selecciona</strong> volver a ejecutar <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> de la herramienta</li> </ol>   <p>Esto es de utilidad si quieres volver a correr la herramienta variando ligeramente los valores de los parámetros, o si deseas verificar la configuración de parámetros que utilizaste.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Re-running a tool <a href="/training-material/faqs/galaxy/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Re-running a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Regular Expressions 101 <a href="/training-material/faqs/galaxy/analysis_regular_expressions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-regular-expressions-101"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-regular-expressions-101" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Regular Expressions 101</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Regular expressions are a standardized way of describing patterns in textual data. They can be extremely useful for tasks such as finding and replacing data. They can be a bit tricky to master, but learning even just a few of the basics can help you get the most out of Galaxy.</p>   <p><strong>Finding</strong></p>   <p>Below are just a few examples of basic expressions:</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">abc</code></td>       <td>an occurrence of <code class="language-plaintext highlighter-rouge">abc</code> within your data</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(abc|def)</code></td>       <td><code class="language-plaintext highlighter-rouge">abc</code> <em>or</em> <code class="language-plaintext highlighter-rouge">def</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[abc]</code></td>       <td>a single character which is either <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, or <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[^abc]</code></td>       <td>a character that is NOT <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, nor <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-z]</code></td>       <td>any lowercase letter</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-zA-Z]</code></td>       <td>any letter (upper or lower case)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[0-9]</code></td>       <td>numbers 0-9</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d</code></td>       <td>any digit (same as <code class="language-plaintext highlighter-rouge">[0-9]</code>)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\D</code></td>       <td>any non-digit character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\w</code></td>       <td>any alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\W</code></td>       <td>any non-alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\s</code></td>       <td>any whitespace</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\S</code></td>       <td>any non-whitespace character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.</code></td>       <td><em>any</em> character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\.</code></td>       <td> </td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">{x,y}</code></td>       <td>between <em>x</em> and <em>y</em> repetitions</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^</code></td>       <td>the beginning of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">$</code></td>       <td>the end of the line</td>     </tr>   </tbody> </table>   <p><strong>Note:</strong> you see that characters such as <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">?</code>, <code class="language-plaintext highlighter-rouge">.</code>, <code class="language-plaintext highlighter-rouge">+</code> etc have a special meaning in a regular expression. If you want to match on those characters, you can escape them with a backslash. So <code class="language-plaintext highlighter-rouge">\?</code> matches the question mark character exactly.</p>   <p><strong>Examples</strong></p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d{4}</code></td>       <td>4 digits (e.g. a year)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr\d{1,2}</code></td>       <td><code class="language-plaintext highlighter-rouge">chr</code> followed by 1 or 2 digits</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.*abc$</code></td>       <td>anything with <code class="language-plaintext highlighter-rouge">abc</code> at the end of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^$</code></td>       <td>empty line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^&gt;.*</code></td>       <td>Line starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta header)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^[^&gt;].*</code></td>       <td>Line not starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta sequence)</td>     </tr>   </tbody> </table>   <p><strong>Replacing</strong></p>   <p>Sometimes you need to capture the exact value you matched on, in order to use it in your replacement, we do this using capture groups <code class="language-plaintext highlighter-rouge">(...)</code>, which we can refer to using <code class="language-plaintext highlighter-rouge">\1</code>, <code class="language-plaintext highlighter-rouge">\2</code> etc for the first and second captured values. If you want to refer to the whole match, use <code class="language-plaintext highlighter-rouge">&amp;</code>.</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Input</th>       <th>Captures</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr(\d{1,2})</code></td>       <td><code class="language-plaintext highlighter-rouge">chr14</code></td>       <td><code class="language-plaintext highlighter-rouge">\1 = 14</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(\d{2}) July (\d{4})</code></td>       <td>24 July 1984</td>       <td><code class="language-plaintext highlighter-rouge">\1 = 24</code>, <code class="language-plaintext highlighter-rouge">\2 = 1984</code></td>     </tr>   </tbody> </table>   <p>An expression like <code class="language-plaintext highlighter-rouge">s/find/replacement/g</code> indicates a replacement expression, this will search (<code class="language-plaintext highlighter-rouge">s</code>) for any occurrence of <code class="language-plaintext highlighter-rouge">find</code>, and replace it with <code class="language-plaintext highlighter-rouge">replacement</code>. It will do this globally (<code class="language-plaintext highlighter-rouge">g</code>) which means it doesn’t stop after the first match.</p>   <p>Example: <code class="language-plaintext highlighter-rouge">s/chr(\d{1,2})/CHR\1/g</code> will replace <code class="language-plaintext highlighter-rouge">chr14</code> with <code class="language-plaintext highlighter-rouge">CHR14</code> etc.</p>   <p>You can also use replacement modifier such as convert to lower case <code class="language-plaintext highlighter-rouge">\L</code> or upper case <code class="language-plaintext highlighter-rouge">\U</code>. Example: <code class="language-plaintext highlighter-rouge">s/.*/\U&amp;/g</code> will convert the whole text to upper case.</p>   <p>Note: In Galaxy, you are often asked to provide the find and replacement expressions separately, so you don’t have to use the <code class="language-plaintext highlighter-rouge">s/../../g</code> structure.</p>   <p>There is a lot more you can do with regular expressions, and there are a few different flavours in different tools/programming languages, but these are the most important basics that will already allow you to do many of the tasks you might need in your analysis.</p>   <p><strong>Tip:</strong> <a href="https://regexone.com/">RegexOne</a> is a nice interactive tutorial to learn the basics of regular expressions.</p>   <p><strong>Tip:</strong> <a href="https://regex101.com/">Regex101.com</a> is a great resource for interactively testing and constructing your regular expressions, it even provides an explanation of a regular expression if you provide one.</p>   <p><strong>Tip:</strong> <a href="https://extendsclass.com/regex-tester.html">Cyrilex</a> is a visual regular expression tester.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Request Galaxy tools on a specific server <a href="/training-material/faqs/galaxy/request-galaxy-tools-on-a-specific-server.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-request-galaxy-tools-on-a-specific-server"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-request-galaxy-tools-on-a-specific-server" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Request Galaxy tools on a specific server</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To request tools that already exist in the <a href="https://toolshed.g2.bx.psu.edu">Galaxy toolshed</a>, but not in your server, please raise an issue at:</p>   <ul>   <li>     <p><em>Europe - usegalaxy.eu</em> | https://github.com/usegalaxy-eu/usegalaxy-eu-tools</p>   </li>   <li>     <p><em>USA - usegalaxy.org</em> | https://github.com/galaxyproject/usegalaxy-tools</p>   </li>   <li>     <p><em>Australia - usegaalxy.org.au</em> | https://github.com/usegalaxy-au/usegalaxy-au-tools/tree/master/usegalaxy.org.au</p>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Select multiple datasets <a href="/training-material/faqs/galaxy/tools_select_multiple_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Selecting a dataset collection as input <a href="/training-material/faqs/galaxy/tools_select_collection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-selecting-a-dataset-collection-as-input"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-selecting-a-dataset-collection-as-input" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Selecting a dataset collection as input</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in  front of the input parameter you want to supply the collection to.</li>   <li>Select the collection you want to use from the list</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Sorting Tools <a href="/training-material/faqs/galaxy/tools_sorting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sorting-tools"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sorting-tools" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sorting Tools</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes input errors are caused because of non-sorted inputs. Try using these:</p>   <ul>   <li><strong>Picard SortSam:</strong> Sort SAM/BAM by coordinate or queryname.</li>   <li><strong>Samtools Sort:</strong> Alternate for SAM/BAM, best when used for coordinate sorting only.</li>   <li><strong>SortBED order the intervals:</strong> Best choice for BED/Interval.</li>   <li><strong>Sort data in ascending or descending order:</strong> Alternate choice for Tabular/BED/Interval/GTF.</li>   <li><strong>VCFsort:</strong> Best choice for VFC.</li>   <li><strong>Tool Form Options for Sorting:</strong> Some tools have an option to sort inputs during job execution. Whenever possible, sort inputs before using tools, especially if jobs fail for not having enough memory resources.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Tool doesn't recognize input datasets <a href="/training-material/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-tool-doesn-t-recognize-input-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-tool-doesn-t-recognize-input-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Tool doesn't recognize input datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The expected input datatype assignment is explained on the tool form. Review the input select areas and the help section below the Run Tool button.</p>   <p>Understanding datatypes <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html">FAQ</a>.</p>   <p>No datasets or collections available? Solutions:</p>   <ol>   <li>Upload or Copy an appropriate dataset for the input into the active history.     <ul>       <li>To load new datasets, review the <strong>Upload</strong> tool and more choices under <strong>Get Data</strong> within Galaxy.</li>       <li>To copy datasets from a different history into the active history see <a href="/training-material/faqs/galaxy/histories_copy_dataset.html">this FAQ</a>.</li>       <li>To use datasets loaded into a shared Data Library see <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html">this FAQ</a>.</li>     </ul>   </li>   <li>Resolve a datatype assignment incompatibility between the dataset and the tool.     <ul>       <li>To redetect a datatype see <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">this FAQ</a>.</li>       <li>To convert a datatype see <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">this FAQ</a>.</li>       <li>To change a datatype see <a href="/training-material/faqs/galaxy/datasets_change_datatype.html">this FAQ</a>.</li>     </ul>   </li>   <li>Individual datasets and dataset collections are selected differently on tool forms.     <ul>       <li>To select a collection input on a tool form see <a href="/training-material/faqs/galaxy/tools_select_collection.html">this FAQ</a>.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using tutorial mode <a href="/training-material/faqs/galaxy/tutorial_mode.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tutorial mode saves you screen space, finds the tools you need, and ensures you use the correct versions for the tutorials to run.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work-2"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using tutorial mode and the Case Study suite <a href="/training-material/topics/single-cell/faqs/gtn-in-galaxy_mode-cs.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tutorial mode saves you screen space, finds the tools you need, and ensures you use the correct versions for the tutorials to run.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode-and-the-case-study-suite"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode-and-the-case-study-suite" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode and the Case Study suite</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial via <strong>Single-cell</strong> (Underneath the <em>Methodologies</em> section), then <strong>Case Study</strong>, then Select your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work-3"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Viewing tool logs (`stdout` and `stderr`) <a href="/training-material/faqs/galaxy/tools_logs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-viewing-tool-logs-stdout-and-stderr"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-viewing-tool-logs-stdout-and-stderr" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Viewing tool logs (`stdout` and `stderr`)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Most tools create log files as output, which can contain useful information about how the tool ran (<code class="language-plaintext highlighter-rouge">stdout</code>, or <em>standard output</em>), and what went wrong (<code class="language-plaintext highlighter-rouge">stderr</code>, or <em>standard error</em>).</p>   <p>To view these log files in Galaxy:</p>   <ul>   <li>Expand one of the outputs of the tool in your history</li>   <li>Click on <strong>View details</strong> <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span></li>   <li>Scroll to the <strong>Job Information</strong> section     <ul>       <li>Here you will find links to the log files (<code class="language-plaintext highlighter-rouge">stdout</code> and <code class="language-plaintext highlighter-rouge">stderr</code>).</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where is the tool help? <a href="/training-material/faqs/galaxy/tools_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Finding tool support
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-is-the-tool-help"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-is-the-tool-help" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where is the tool help?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There is documentation available on the tool form itself which mentions the following information:</p>   <ul>   <li>Parameters</li>   <li>Expected format for input dataset(s)</li>   <li>Links to publications and ToolShed source repositories</li>   <li>Tool and wrapper version(s)</li>   <li>3rd party author web sites and documentation</li> </ul>   <p><strong>Scroll down on the tool form</strong> to locate:</p>   <ul>   <li>Information about expected inputs/outputs</li>   <li>Expanded definitions</li>   <li>Sample data</li>   <li>Example use cases</li>   <li>Graphics</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="troubleshooting"> <a href="#troubleshooting"> Troubleshooting </a></h3>

<hr/>



<h4 class="faq-area"> How to find and correct tool errors related to Metadata? <a href="/training-material/faqs/galaxy/dataset_tool_errors_metadata.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Finding and Correcting Metadata
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-find-and-correct-tool-errors-related-to-metadata"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-find-and-correct-tool-errors-related-to-metadata" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to find and correct tool errors related to Metadata?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools can error when the wrong dataset attributes (metadata) are assigned. Some of these wrong assignments may be:</p>   <ul>   <li>Tool outputs, which are automatically assigned without user action.</li>   <li>Incorrect autodetection of datatypes, which need manual modification.</li>   <li>Undetected attributes, which require user action (example: assigning database to newly uploaded data).</li> </ul>   <p>How to notice missing Dataset Metadata:</p>   <ul>   <li>Dataset will not be downloaded when using the disk icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>.</li>   <li>Tools error when using a previously successfully used specific dataset.</li>   <li>Tools error with a message that ends with: <code class="language-plaintext highlighter-rouge">OSError: [Errno 2] No such file or directory</code>.</li> </ul>   <p>Solution:</p>   <p>Click on the dataset’s pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <em>Edit Attributes</em> forms and do one of the following as applies:</p>   <ul>   <li><strong>Directly reset metadata</strong>     <ul>       <li>Find the tab for the metadata you want to change, make the change, and save.</li>     </ul>   </li>   <li><strong>Autodetect metadata</strong>     <ul>       <li>Click on the <em>Auto-detect button</em>. The dataset will turn yellow in the history while the job is processing.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Incomplete Dataset Download <a href="/training-material/faqs/galaxy/dataset_incomplete_download.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-incomplete-dataset-download"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-incomplete-dataset-download" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Incomplete Dataset Download</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In case the dataset downloads incompletely:</p>   <ul>   <li>Use the <em>Google Chrome</em> web browser. Sometimes Chrome works better at supporting continuous data transfers.</li>   <li>Use the <a href="/training-material/faqs/galaxy/datasets_download_using_command_line.html">command-line option</a> instead. The data may really be too large to download OR your connection is slower. This can also be a faster way to download multiple datasets plus ensure a complete transfer (small or large data).</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding 'canceled by admin' or cluster failure error messages <a href="/training-material/faqs/galaxy/troubleshooting_cluster_failure.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-canceled-by-admin-or-cluster-failure-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-canceled-by-admin-or-cluster-failure-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding 'canceled by admin' or cluster failure error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The initial error message could be:</p>   <pre><br>This job failed because it was cancelled by an administrator.<br>Please click the bug icon to report this problem if you need help.<br></pre>   <p>Or</p>   <pre><br>job info:<br>Remote job server indicated a problem running or monitoring this job.<br></pre>   <ul>   <li>Causes:     <ul>       <li>Server or cluster error.</li>       <li>Less frequently, input problems are a factor.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun. Server/cluster errors like this are usually transient.</li>       <li>Review the Solutions section of the <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html">Understanding input error messages</a> FAQ.</li>       <li>If after any corrections, the job still fails, please report the technical issue <a href="/training-material/faqs/galaxy/analysis_reporting_issues.html">following the extended issue guidelines</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding 'exceeds memory allocation' error messages <a href="/training-material/faqs/galaxy/troubleshooting_excess_memory.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-exceeds-memory-allocation-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-exceeds-memory-allocation-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding 'exceeds memory allocation' error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The error message to be displayed are as follows:</p>   <pre><br>job info:<br>This job was terminated because it used more memory than it was allocated.<br>Please click the bug icon to report this problem if you need help.<br></pre>   <p>Or</p>   <pre><br>stderr:<br>Fatal error: Exit code 1 ()<br>slurmstepd: error: Detected 1 oom-kill event(s) in step XXXXXXX.batch cgroup.<br></pre>   <p>Sometimes this message may appear at the bottom</p>   <pre><br>job stderr:<br>slurmstepd: error: Detected 1 oom-kill event(s) in step XXXXXXX.batch cgroup.<br></pre>   <p>In rare cases when the memory quota is exceeded very quickly, an error message such as the following can appear</p>   <pre><br>job stderr:<br>Fatal error: Exit code 1 ()<br>Traceback (most recent call last):<br>(other lines)<br>Memory Error<br></pre>   <p><strong>Note:</strong> Job runtime memory is different from the amount of free storage space (quota) in an account.</p>   <ul>   <li>Causes:     <ul>       <li>The job ran out of memory while executing on the cluster node that ran the job.</li>       <li>The most common reasons for this error are input and tool parameters problems that must be adjusted/corrected.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun to execute the job on a different cluster node.</li>       <li>Review the Solutions section of the <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html">Understanding input error messages</a> FAQ.</li>       <li>Your data may actually be too large to process at a public Galaxy server. Alternatives include setting up a <a href="/training-material/faqs/gtn/galaxy_usage.html">private Galaxy server</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding ValueError error messages <a href="/training-material/faqs/galaxy/troubleshooting_value_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-valueerror-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-valueerror-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding ValueError error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The full error is usually a longer message seen only after clicking on the bug icon or by reviewing the job details <code class="language-plaintext highlighter-rouge">stderr</code>.</p>   <p>How to do both is covered in the Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">FAQ</a>.</p>   <pre><br>stderr<br>...<br>Many lines of text, may include parameters<br>...<br>...<br>ValueError: invalid literal for int() with base 10: some-sequence-read-name<br></pre>   <ul>   <li>Causes:     <ul>       <li>MACS2 produces this error the first time it is run. MACS is not the only tool that can produce this issue, but it is the most common.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun.</li>       <li>MACS/2 is not capable of interpreting sequence read names with spaces included. Try following these two:         <ul>           <li>Remove unmapped reads from the SAM dataset. There are several filtering tools in the groups <strong>SAMTools</strong> and <strong>Picard</strong> that can do this.</li>           <li>Convert the SAM input to BAM format with the tool <strong>SAMtools: SAM-to-BAM</strong>. When compressed input is given to MACS, the spaces are no longer an issue.</li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding input error messages <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-input-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-input-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding input error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Input problems are very common across any analysis that makes use of programmed tools.</p>   <ul>   <li>Causes:     <ul>       <li>No quality assurance or content/formatting checks were run on the first datasets of an analysis workflow.</li>       <li>Incomplete dataset Upload.</li>       <li>Incorrect or unassigned <a href="/training-material/faqs/galaxy/index.html#datasets">datatype</a> or <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">database</a>.</li>       <li>Tool-specific formatting requirements for inputs were not met.</li>       <li>Parameters set on a tool form are a mismatch for the input data content or format.</li>       <li>Inputs were in an error state (red) or were putatively successful (green) but are empty.</li>       <li>Inputs do not meet the datatype specification.</li>       <li>Inputs do not contain the exact content that a tool is expecting or that was input in the form.</li>       <li>Annotation files are a mismatch for the selected or assigned reference genome build.</li>       <li><strong>Special case:</strong> Some of the data were generated outside of Galaxy, but later a built-in indexed genome build was assigned in Galaxy for use with downstream tools. This scenario can work, but only if those two reference genomes are an exact match.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Review our <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting Tips</a> for what and where to check.</li>       <li>Review the <a href="https://training.galaxyproject.org/">GTN</a> for related tutorials on tools/analysis plus FAQs.</li>       <li>Review <a href="https://help.galaxyproject.org/">Galaxy Help</a> for prior discussion with extended solutions.</li>       <li>Review <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html">datatype FAQs</a>.</li>       <li>Review the tool form.         <ul>           <li>Input selection areas include usage help.</li>           <li>The help section at the bottom of a tool form often has examples. Does your own data match the format/content?</li>           <li>See the links to publications and related resources.</li>         </ul>       </li>       <li>Review the inputs.         <ul>           <li>All inputs must be in a success state (green) and actually contain content.</li>           <li>Did you <a href="/training-material/faqs/galaxy/datasets_change_datatype.html">directly assign the datatype</a> or <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">convert the datatype</a>? What results when the <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">datatype is detected</a> by Galaxy? If these differ, there is likely a content problem.</li>           <li>For most analysis, allowing Galaxy to detect the datatype during Upload is best and adjusting a datatype later should rarely be needed. If a datatype is modified, the change has a specific purpose/reason.</li>           <li>Does your data have headers? Is that in specification for the datatype? Does the tool form have an option to specify if the input has headers or not? Do you need to remove headers first for the correct datatype to be detected? Example <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">GTF</a>.</li>           <li>Large inputs? Consider modifying your inputs to be smaller. Examples: <a href="/training-material/faqs/galaxy/datasets_working_with_fastq.html">FASTQ</a> and <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">FASTA</a>.</li>         </ul>       </li>       <li>Run quality checks on your data.         <ul>           <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword “qa-qc” for examples.</li>           <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “qa-qc” and your datatype(s) for more help.</li>         </ul>       </li>       <li>Reference annotation tips.         <ul>           <li>In most cases, <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">GTF is preferred over GFF3</a>.</li>           <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “gtf” and “gff3” for more help.</li>         </ul>       </li>       <li>Input mismatch tips.         <ul>           <li>Do the chromosome/sequence identifiers exactly match between all inputs? Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> for more help about how to correct build/version identifier mismatches between inputs.</li>           <li>“Chr1” and “chr1” and “1” do not mean the same thing to a tool.</li>         </ul>       </li>       <li>Custom genome transcriptome exome tips. See <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">FASTA</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding walltime error messages <a href="/training-material/faqs/galaxy/troubleshooting_walltime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-walltime-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-walltime-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding walltime error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The full error message will be reported as below, and can be found by clicking on the bug icon for a failed job run (red dataset):</p>   <pre><br>job info:<br>This job was terminated because it ran longer than the maximum allowed job run time.<br>Please click the bug icon to report this problem if you need help.<br></pre>   <p>Or sometimes,</p>   <pre><br>job stderr:<br>slurmstepd: error: *** JOB XXXX ON XXXX CANCELLED AT 2019-XX-XXTXX:XX:XX DUE TO TIME LIMIT ***<br><br>job info:<br>Remote job server indicated a problem running or monitoring this job.<br></pre>   <ul>   <li>Causes:     <ul>       <li>The job execution time exceeded the “wall-time” on the cluster node that ran the job.</li>       <li>The server may be undergoing maintenance.</li>       <li>Very often input problems also cause this same error.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun.</li>       <li>Check the server homepage for banners or notices. Selected servers also post to the <a href="https://status.galaxyproject.org/">Galaxy status page</a>.</li>       <li>Review the Solutions section of the <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html">Understanding input error messages</a> FAQ.</li>       <li>Your data may actually be too large to process at a public Galaxy server. Alternatives include setting up a <a href="/training-material/faqs/gtn/galaxy_usage.html">private Galaxy server</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-information-should-i-include-when-reporting-a-problem"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-information-should-i-include-when-reporting-a-problem" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What information should I include when reporting a problem?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>   <p><strong>What to include</strong></p>   <ol>   <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>   <li>Which server(s) did you run those commands on?</li>   <li>What account/username did you use?</li>   <li>Where did it go wrong?</li>   <li>What were the stdout/stderr of the tool that failed? Include the text.</li>   <li>Did you try any workarounds? What results did those produce?</li>   <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>   <li>If there are job IDs, please include them as text so administrators don’t have to manually transcribe the job ID in your picture.</li> </ol>   <p>It makes the process of answering ‘bug reports’ much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>   <p><strong>What does a GOOD bug report look like?</strong></p>   <p>The people who provide support for Galaxy are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>   <blockquote class="quote">   <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My username is jane-doe.</p>   <p>Here is everything that I know:</p>   <ul>     <li>The dataset is green, the job did not fail</li>     <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>     <li>I have read it but I do not understand what X/Y means.</li>     <li>The job ID from the output information page is 123123abdef.</li>     <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>   </ul>   <p>Could you help me?</p> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/gtn/issues_galaxy.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-information-should-i-include-when-reporting-a-problem-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-information-should-i-include-when-reporting-a-problem-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What information should I include when reporting a problem?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>   <p><strong>What to include</strong></p>   <ol>   <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>   <li>Which server(s) did you run those commands on?</li>   <li>What account/username did you use?</li>   <li>Where did it go wrong?</li>   <li>What were the stdout/stderr of the tool that failed? Include the text.</li>   <li>Did you try any workarounds? What results did those produce?</li>   <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>   <li>If there are job IDs, please include them as text so administrators don’t have to manually transcribe the job ID in your picture.</li> </ol>   <p>It makes the process of answering ‘bug reports’ much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>   <p><strong>What does a GOOD bug report look like?</strong></p>   <p>The people who provide support for Galaxy are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>   <blockquote class="quote">   <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My username is jane-doe.</p>   <p>Here is everything that I know:</p>   <ul>     <li>The dataset is green, the job did not fail</li>     <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>     <li>I have read it but I do not understand what X/Y means.</li>     <li>The job ID from the output information page is 123123abdef.</li>     <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>   </ul>   <p>Could you help me?</p> </blockquote> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="user interface"> <a href="#user interface"> User interface </a></h3>

<hr/>



<h4 class="faq-area"> I can’t find the “Analyze Data” button <a href="/training-material/topics/introduction/faqs/analyze_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-i-can-t-find-the-analyze-data-button"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-i-can-t-find-the-analyze-data-button" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: I can’t find the “Analyze Data” button</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The Galaxy interface has changed a bit recently, “Analyze Data” was always the home button, and now looks like a home icon.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> My Galaxy looks different than in the tutorial/video <a href="/training-material/topics/introduction/faqs/galaxy.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-my-galaxy-looks-different-than-in-the-tutorial-video"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-my-galaxy-looks-different-than-in-the-tutorial-video" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: My Galaxy looks different than in the tutorial/video</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy gets frequent updates, different servers will be running different versions. This is nothing to worry about, just let us know if you can’t find how to perform a task in your Galaxy.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="user preferences"> <a href="#user preferences"> User preferences </a></h3>

<hr/>



<h4 class="faq-area"> Does your account usage quota seem incorrect? <a href="/training-material/faqs/galaxy/user_preferences_account_usage_quota.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-does-your-account-usage-quota-seem-incorrect"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-does-your-account-usage-quota-seem-incorrect" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Does your account usage quota seem incorrect?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Log out of Galaxy, then back in again. This refreshes the disk usage calculation displayed in the Masthead usage (summary) and under <em>User &gt; Preferences</em> (exact).</li> </ol>   <p>Note:</p>   <ul>   <li>Your account usage quota can be found at the bottom of your user preferences page.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Forgot Password <a href="/training-material/faqs/galaxy/user_preferences_forgot_password.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-forgot-password"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-forgot-password" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Forgot Password</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Go to the Galaxy server you are using.</li>   <li>Click on <strong>Login or Register</strong>.</li>   <li>Enter your email on the Public Name or Email Address entry box.</li>   <li>Click on the link under the password entry box titled <strong>Forgot password? Click here to reset your password.</strong></li>   <li>An email will be sent with a password reset link. This email may be in your email Spam or Trash folders, depending on your filters.</li>   <li>Click on the reset link in the email or copy and paste it into a web browser window.</li>   <li>Enter your new password and click on <strong>Save new password</strong>.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Getting your API key <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-getting-your-api-key"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-getting-your-api-key" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Getting your API key</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>In your browser, open your Galaxy homepage</li>   <li>Log in, or register a new account, if it’s the first time you’re logging in</li>   <li>Go to <code class="language-plaintext highlighter-rouge">User -&gt; Preferences</code> in the top menu bar, then click on <code class="language-plaintext highlighter-rouge">Manage API key</code></li>   <li>If there is no current API key available, click on <code class="language-plaintext highlighter-rouge">Create a new key</code> to generate it</li>   <li>Copy your API key to somewhere convenient, you will need it throughout this tutorial</li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="utilities"> <a href="#utilities"> Utilities </a></h3>

<hr/>



<h4 class="faq-area"> Got lost along the way? <a href="/training-material/topics/admin/faqs/missed-something.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-got-lost-along-the-way"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Got lost along the way?</div>   <p>If you missed any steps, you can compare against the <a href="https://github.com/hexylena/git-gat/tree/step-">reference files</a>, or see what changed since <a href="https://github.com/hexylena/git-gat/compare/step--1...step-#files_bucket">the previous tutorial</a>.</p>   <p>If you’re using <code class="language-plaintext highlighter-rouge">git</code> to track your progress, remember to add your changes and commit with a good commit message!</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="visualisation"> <a href="#visualisation"> Visualisation </a></h3>

<hr/>



<h4 class="faq-area"> Open History files in Integrated Genome Browser (IGB) <a href="/training-material/faqs/galaxy/visualisations_igb.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-open-history-files-in-integrated-genome-browser-igb"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-open-history-files-in-integrated-genome-browser-igb" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Open History files in Integrated Genome Browser (IGB)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can open some file types in Integrated Genome Browser (IGB), a desktop genome browser. (<a href="https://wiki.bioviz.org/confluence/display/igbman/File+Formats">Supported File Types</a>)</p>   <p>Here’s how:</p>   <ol>   <li>Install IGB on your computer (<a href="https://bioviz.org/download.html">download page</a>).</li>   <li>Start IGB.</li>   <li>In Galaxy, click the desired dataset’s name to expand it.     <ul>       <li>Check that the reference genome (dbkey) is set (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>).</li>     </ul>   </li>   <li>Click on the Charts icon <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span></li>   <li>In the central panel, next to <code class="language-plaintext highlighter-rouge">display in IGB</code>, choose <code class="language-plaintext highlighter-rouge">View</code>.</li> </ol>   <p>When you choose “View” in Galaxy, your browser opens a new tab showing a page from BioViz.org. Check the newly opened page for next steps.</p>   <p>Having trouble? Working with a custom genome assembly not yet available in Galaxy or IGB?</p>   <p>Contact <a href="https://bioviz.org/help.html">the IGB team</a> for help and advice!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/proteomics/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/assembly/tutorials/unicycler-assembly/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/assembly/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-4"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-4" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/visualisation/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-5"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-5" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-6"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-6" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-7"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-7" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/epigenetics/tutorials/atac-seq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-8"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-8" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-9"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-9" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/epigenetics/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-10"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-10" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/transcriptomics/tutorials/ref-based/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-11"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-11" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-12"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-12" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/transcriptomics/tutorials/rb-rnaseq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-13"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-13" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-14"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-14" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/transcriptomics/tutorials/clipseq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-15"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-15" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/transcriptomics/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-16"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-16" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/introduction/tutorials/galaxy-intro-strands/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-17"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-17" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-18"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-18" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/introduction/tutorials/igv-introduction/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-19"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-19" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/introduction/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-20"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-20" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/variant-analysis/tutorials/non-dip/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-21"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-21" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/variant-analysis/tutorials/dunovo/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-22"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-22" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/variant-analysis/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-23"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-23" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/galaxy-interface/tutorials/collections/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-24"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-24" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/galaxy-interface/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-25"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-25" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/genome-annotation/tutorials/tnseq/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-26"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-26" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/genome-annotation/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-27"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-27" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/sequence-analysis/tutorials/mapping/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-28"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-28" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/topics/sequence-analysis/faqs/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-29"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-29" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/faqs/galaxy/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy-30"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy-30" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="workflows"> <a href="#workflows"> Workflows </a></h3>

<hr/>



<h4 class="faq-area"> Annotate a workflow <a href="/training-material/faqs/galaxy/workflows_annotate.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-annotate-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-annotate-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Annotate a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Open the workflow editor for the workflow</li>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>Edit Attributes</strong> on the top right</li>   <li>Write a description of the workflow in the <strong>Annotation</strong> box</li>   <li>Add a tag (which will help to search for the workflow) in the <strong>Tags</strong> section</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a new workflow <a href="/training-material/faqs/galaxy/workflows_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can create a Galaxy workflow from scratch in the Galaxy workflow editor.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click <strong>Workflow</strong> on the top bar</li>   <li>Click the new workflow <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> button</li>   <li>Give it a clear and memorable name</li>   <li>Clicking <strong>Save</strong> will take you directly into the workflow editor for that workflow</li>   <li>Need more help? Please see the <em>How to make a workflow subsection</em> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">here</a></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Ensuring Workflows meet Best Practices <a href="/training-material/faqs/galaxy/workflows_best_practices.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-ensuring-workflows-meet-best-practices"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-ensuring-workflows-meet-best-practices" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Ensuring Workflows meet Best Practices</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When you are editing a workflow, there are a number of additional steps you can take to ensure that it is a Best Practice workflow and will be more reusable.</p>   <ol>   <li>Open a workflow for editing</li>   <li>In the workflow menu bar, you’ll find the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-options</span> <strong>Workflow Options</strong> dropdown menu.</li>   <li>     <p>Click on it and select <i class="fas fa-wand-magic-sparkles" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-best-practices</span> <strong>Best Practices</strong> from the dropdown menu.</p>     <p><img src="/training-material/faqs/galaxy/images/best-practices1.png" alt="screenshot showing the best practices menu item in the gear dropdown." /></p>   </li>   <li>     <p>This will take you to a new side panel, which allows you to investigate and correct any issues with your workflow.</p>     <p><img src="/training-material/faqs/galaxy/images/best-practices2.png" alt="screenshot showing the best practices side panel. several issues are raised like a missing annotation with a link to add that, and non-optional inputs that are unconnected. Additionally several items already have green checks like the workflow defining creator information and a license." /></p>   </li> </ol>   <p>The Galaxy community also has a <a href="https://planemo.readthedocs.io/en/latest/best_practices_workflows.html">guide on best practices for maintaining workflows</a>. This guide includes the best practices from the Galaxy workflow panel, plus:</p>   <ul>   <li>adding tests to the workflow</li>   <li>publishing the workflow on GitHub, a public GitLab server, or another public version-controlled repository</li>   <li>registering the workflow with a workflow registry such as WorkflowHub or Dockstore</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Extracting a workflow from your history <a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy can automatically create a workflow based on the analysis you have performed in a history. This means that once you have done an analysis manually once, you can easily extract a workflow to repeat it on different data.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-extracting-a-workflow-from-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-extracting-a-workflow-from-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Extracting a workflow from your history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> button.</p>     <p>This will make the creation of the workflow easier.</p>   </li>   <li>     <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract workflow</strong>.</p>     <p><img src="/training-material/topics/introduction/images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>     <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the workflow.</p>   </li>   <li>     <p>Replace the <strong>Workflow name</strong> to something more descriptive.</p>   </li>   <li>     <p><strong>Rename</strong> each workflow input in the boxes at the top of the second column.</p>   </li>   <li>     <p>If there are any steps that shouldn’t be included in the workflow, you can <strong>uncheck</strong> them in the first column of boxes.</p>   </li>   <li>     <p>Click on the <strong>Create Workflow</strong> button near the top.</p>     <p>You will get a message that the workflow was created.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Extraer un flujo de trabajo de tu historial <a href="/training-material/faqs/galaxy-es/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy puede crear automáticamente un flujo de trabajo basado en un análisis almacenado en tu historial. Esto significa que una vez que hayas realizado un análisis manualmente, puedes extraer fácilmente un flujo de trabajo para repetirlo con diferentes datos.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-extraer-un-flujo-de-trabajo-de-tu-historial"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-extraer-un-flujo-de-trabajo-de-tu-historial" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Extraer un flujo de trabajo de tu historial</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Elimina cualquier trabajo fallido o no deseado de tu historial.</li>   <li>Haz clic en <strong>Opciones de historial</strong> (icono de engranaje <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span>) en la parte superior del panel de historial.</li>   <li>Selecciona <strong>Extraer flujo de trabajo</strong></li>   <li>Verifica los pasos, ingresa un nombre para tu flujo de trabajo y presiona el botón <strong>Crear flujo de trabajo</strong>.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Hiding intermediate steps <a href="/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-hiding-intermediate-steps"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-hiding-intermediate-steps" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Hiding intermediate steps</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When a workflow is executed, the user is usually primarily interested in the final product and not in all intermediate steps. By default all the outputs of a workflow will be shown, but we can explicitly tell Galaxy which outputs to show and which to hide for a given workflow. This behaviour is controlled by the little checkbox in front of every output dataset:</p>   <p><img src="/training-material/shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Import workflows from DockStore <a href="/training-material/faqs/galaxy/workflows_import_from_dockstore.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-import-workflows-from-dockstore"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-import-workflows-from-dockstore" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Import workflows from DockStore</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://dockstore.org/">Dockstore</a> is a free and open source platform for sharing reusable and scalable analytical tools and workflows.</p>   <ol>   <li>Ensure that you are logged in to your Galaxy account.</li>   <li>Go to <a href="https://dockstore.org">DockStore</a>.</li>   <li>Select any Galaxy workflow you want to import.</li>   <li>Click on “Galaxy” dropdown within the “Launch with” panel located in the upper right corner.</li>   <li>Select a galaxy instance you want to launch this workflow with.</li>   <li>You will be redirected to Galaxy and presented with a list of workflow versions.</li>   <li>Click the version you want (usually the latest labelled as “main”)</li>   <li>You are done!</li> </ol>   <p>The following short video walks you through this uncomplicated procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/K2wFrSLFpa0" title="Importing from Dockstore" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing from Dockstore</figcaption> </figure> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Import workflows from WorkflowHub <a href="/training-material/faqs/galaxy/workflows_import_from_workflowhub.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-import-workflows-from-workflowhub"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-import-workflows-from-workflowhub" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Import workflows from WorkflowHub</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://workflowhub.eu/">WorkflowHub</a> is a workflow management system which allows workflows to be FAIR (Findable, Accessible, Interoperable, and Reusable), citable, have managed metadata profiles, and be openly available for review and analytics.</p>   <ol>   <li>Ensure that you are logged in to your Galaxy account.</li>   <li>Click on the <strong>Workflow</strong> menu, located in the top bar.</li>   <li>Click on the <strong>Import</strong> button, located in the right corner.</li>   <li>In the section “Import a Workflow from Configured GA4GH Tool Registry Servers (e.g. Dockstore)”, click on <strong>Search form</strong>.</li>   <li>In the <strong>TRS Server: <em>workflowhub.eu</em></strong> menu you should type your query. <img src="/training-material/topics/assembly/images/vgp_assembly/workflow_list.png" alt="galaxy TRS workflow search field, name:vgp is entered in the search bar, and five different workflows all labelled VGP are listed" /></li>   <li>Click on the desired workflow, and finally select the latest available version.</li> </ol>   <p>After that, the imported workflows will appear in the main workflow menu. In order to run the workflow, just need to click in the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> <strong>Run workflow</strong> icon.</p>   <p>Below is a short video showing this uncomplicated procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing from WorkflowHub" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing from WorkflowHub</figcaption> </figure> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing a workflow <a href="/training-material/faqs/galaxy/workflows_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing a workflow using the search <a href="/training-material/faqs/galaxy/workflows_import_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-using-the-search"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-using-the-search" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a workflow using the search</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <em>Workflow</em> in the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> icon at the top-right of the screen</li>   <li>On the new page, select the <strong>GA4GH servers</strong> tab, and configure the <strong>GA4GH Tool Registry Server (TRS) Workflow Search</strong> interface as follows:     <ol>       <li><em>“TRS Server”</em>: the TRS Server you want to search on (Dockstore or Workflowhub)</li>       <li>Type in the <em>search query</em></li>       <li>Expand the correct workflow by clicking on it</li>       <li>Select the version you would like to <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> import</li>     </ol>   </li> </ol>   <p>The workflow will be imported to your list of workflows. Note that it will also carry a little green check mark next to its name, which indicates that this is an original workflow version imported from a TRS server. If you ever modify the workflow with Galaxy’s workflow editor, it will lose this indicator.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing and Launching a Dockstore Workflow <a href="/training-material/faqs/galaxy/workflows_run_ds.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-dockstore-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and Launching a Dockstore Workflow</div>   <!-- GTN:IGNORE:011 _blank is used here due to iframeing sites that (can) set x-frame-options. The contexts in which _blank is set are only visible when iframe'd -->   <ol>   <li>Go to <a href="https://my.galaxy.training/?path=/workflows/import">Workflow → Import</a> in your Galaxy</li>   <li>Switch tabs to TRS ID</li>   <li>Ensure the TRS server is set to “dockstore.org”</li>   <li>Provide your workflow hub ID</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing and Launching a WorkflowHub.eu Workflow <a href="/training-material/faqs/galaxy/workflows_run_wfh.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-workflowhub-eu-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and Launching a WorkflowHub.eu Workflow</div>   <!-- GTN:IGNORE:011 _blank is used here due to iframeing sites that (can) set x-frame-options. The contexts in which _blank is set are only visible when iframe'd -->   <ol>   <li>Go to <a href="https://my.galaxy.training/?path=/workflows/import">Workflow → Import</a> in your Galaxy</li>   <li>Switch tabs to TRS ID</li>   <li>Ensure the TRS server is set to “workflowhub.eu”</li>   <li>Provide your workflow hub ID</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing and launching a GTN workflow <a href="/training-material/faqs/galaxy/workflows_run_trs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-gtn-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and launching a GTN workflow</div>   <ol>   <li>Find the material you are interested in</li>   <li>View it’s workflows, which can be found in the metadata box at the top of the tutorial</li>   <li>Click the button on any workflow to run it.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Make a workflow public <a href="/training-material/faqs/galaxy/workflows_publish.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-make-a-workflow-public"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-make-a-workflow-public" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Make a workflow public</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <strong>Workflow</strong> on the top menu bar of Galaxy. You will see a list of all your workflows</li>   <li>Click on the interesting workflow</li>   <li>Click on <strong>Share</strong></li>   <li>Click on <strong>Make Workflow accessible</strong>. This makes the workflow <em>publicly accessible</em> but <em>unlisted</em>.</li>   <li>To also list the workflow in the <em>Shared Data</em> section (in the top menu bar) of Galaxy, click <strong>Make Workflow publicly available in Published Workflows</strong></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Opening the workflow editor <a href="/training-material/faqs/galaxy/workflows_edit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-opening-the-workflow-editor"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-opening-the-workflow-editor" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Opening the workflow editor</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>In the top menu bar, click on <strong>Workflows</strong></li>   <li>Click on the name of the workflow you want to edit <img src="/training-material/topics/galaxy-interface/images/edit_workflow.png" alt="Workflow drop down menu showing Edit option" title="When you click on the name of a workflow it will open a drop down menu whose first option is Edit. Select the Edit option to edit the workflow." /></li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Edit</strong> from the dropdown menu to open the workflow in the workflow editor</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming workflow outputs <a href="/training-material/faqs/galaxy/workflows_rename_output.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-workflow-outputs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-workflow-outputs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming workflow outputs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the workflow editor</li>   <li>Click on the tool in the workflow to get the details of the tool on the right-hand side of the screen.</li>   <li>Scroll down to the <strong>Configure Output</strong> section of your desired parameter, and click it to expand it.     <ul>       <li>         <p>Under <strong>Rename dataset</strong>, give it a meaningful name</p>         <p><img src="/training-material/shared/images/workflow_rename_datasets.png" alt="Rename output datasets" /></p>       </li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Running a workflow <a href="/training-material/faqs/galaxy/workflows_run.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Running a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Setting parameters at run-time <a href="/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-setting-parameters-at-run-time"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-setting-parameters-at-run-time" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Setting parameters at run-time</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the workflow editor</li>   <li>Click on the tool in the workflow to get the details of the tool on the right-hand side of the screen.</li>   <li>Scroll down to the parameter you want users to provide every time they run the workflow</li>   <li>Click on the arrow in front of the name <i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden">workflow-runtime-toggle</span> to toggle to <em>set at runtime</em></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Viewing a workflow report <a href="/training-material/faqs/galaxy/workflows_report_view.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can find the workflow report from the workflow invocation
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-viewing-a-workflow-report"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-viewing-a-workflow-report" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Viewing a workflow report</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Go to <strong>User</strong> on the top menu bar of Galaxy.</li>   <li>Click on <strong>Workflow invocations</strong>     <ul>       <li>Here you will find a list of all the workflows you have run</li>     </ul>   </li>   <li>Click on the name of a workflow invocation to expand it <img src="/training-material/faqs/galaxy/images/workflow_invocations_menu.png" alt="workflow invocations list" /></li>   <li>Click on <strong>View Report</strong> to go to the workflow report page</li>   <li><strong>Note:</strong> The report can also be downloaded in PDF format by clicking on the <i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-report-download</span> icon.</li> </ul> </blockquote> <!--END_SNIPPET-->









<h2 id="bibliography">References</h2>
<ol class="bibliography"><li id="Wood2014">Wood, D. E., and S. L. Salzberg, 2014 <b>Kraken: ultrafast metagenomic sequence classification using exact alignments</b>. Genome Biology 15: R46. <a href="https://doi.org/10.1186/gb-2014-15-3-r46">10.1186/gb-2014-15-3-r46</a></li>
<li id="Devenyi_2018">Devenyi, G. A., R. Emonet, R. M. Harris, K. L. Hertweck, D. Irving <i>et al.</i>, 2018 <b>Ten simple rules for collaborative lesson development</b> (S. Markel, Ed.). PLOS Computational Biology 14: e1005963. <a href="https://doi.org/10.1371/journal.pcbi.1005963">10.1371/journal.pcbi.1005963</a></li>
<li id="Garcia2020">Garcia, L., B. Batut, M. L. Burke, M. Kuzak, F. Psomopoulos <i>et al.</i>, 2020 <b>Ten simple rules for making training materials FAIR</b> (S. Markel, Ed.). PLOS Computational Biology 16: e1007854. <a href="https://doi.org/10.1371/journal.pcbi.1007854">10.1371/journal.pcbi.1007854</a></li></ol>



<br><br><hr>
<h5>Still have questions?</h5>

<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://gitter.im/Galaxy-Training-Network/Lobby" class="btn btn-square-medium btn-yellow"><div>Gitter Chat Support</div></a>


<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://help.galaxyproject.org" class="btn btn-square-medium btn-purple"><div>Galaxy Help Forum</div></a>



            </div>
        </div>
    </div>
</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/all-faq.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/all-faq.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/715a5f0aefd15f73c5a3222b576a5f6c21ada9c5">715a5f0</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>