<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>Training-material by bgruening</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>Training-material</h1>
        <p>A collection of training material from offered Galaxy courses</p>

        <p class="view"><a href="https://github.com/bgruening/training-material">View the Project on GitHub <small>bgruening/training-material</small></a></p>


        <ul>
          <li><a href="https://github.com/bgruening/training-material/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/bgruening/training-material/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/bgruening/training-material">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h1>
<a id="training-material" class="anchor" href="#training-material" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Training material</h1>

<table>
<thead>
<tr>
<th align="left">topic</th>
<th align="center">features</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/Genome-Annotation/general-introduction/README.md">Genome Annotation</a></td>
<td align="center"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></td>
</tr>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/RNA-Seq/">RNA-seq</a></td>
<td align="center">
<g-emoji alias="whale" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f433.png">ğŸ³</g-emoji> <a href="https://vimeo.com/128268401"><g-emoji alias="movie_camera" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f3a5.png">ğŸ¥</g-emoji></a> <g-emoji alias="eyes" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f440.png">ğŸ‘€</g-emoji> <a href="https://usegalaxy.org/u/jeremy/p/galaxy-rna-seq-analysis-exercise"><g-emoji alias="page_facing_up" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4c4.png">ğŸ“„</g-emoji></a> <a href="https://github.com/bgruening/training-material/blob/master/rna-seq/rna-seq.md"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></a>
</td>
</tr>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/ChIP-Seq/ChIPseq.md">ChIP-seq</a></td>
<td align="center"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></td>
</tr>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/Exome-Seq/Exome-Seq.md">Exome-seq</a></td>
<td align="center"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></td>
</tr>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/Methylation-Seq/Methylation-Seq.md">MethylC-seq</a></td>
<td align="center"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></td>
</tr>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/Galaxy_Introduction/Galaxy_Introduction.md">Galaxy Introduction</a></td>
<td align="center"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></td>
</tr>
<tr>
<td align="left"><a href="https://github.com/bgruening/training-material/blob/master/Data_Sources/Data_Sources.md">Data Sources</a></td>
<td align="center"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></td>
</tr>
<tr>
<td align="left"><a href="http://planemo.readthedocs.io/en/latest/writing_appliance.html">Galaxy Tool Dev</a></td>
<td align="center"><a href="http://planemo.readthedocs.io/en/latest/writing_appliance.html"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></a></td>
</tr>
<tr>
<td align="left"><a href="https://slides.com/bgruening/the-galaxy-docker-project/live#/">Galaxy Docker Project</a></td>
<td align="center"><a href="https://slides.com/bgruening/the-galaxy-docker-project/live#/"><g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji></a></td>
</tr>
</tbody>
</table>

<p>The Galaxy community offers many different ways of training. The table above lists all available training features of a specific topic to give you a fast and comprehensive overview.</p>

<ul>
<li>
<g-emoji alias="whale" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f433.png">ğŸ³</g-emoji> The training course and material is available as Docker container.</li>
<li>
<g-emoji alias="movie_camera" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f3a5.png">ğŸ¥</g-emoji> The training course is available as video on our <a href="https://vimeo.com/galaxyproject">vimeo page</a>.</li>
<li>
<g-emoji alias="eyes" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f440.png">ğŸ‘€</g-emoji> The training course is available as <a href="https://github.com/galaxyproject/galaxy-tours">Galaxy tour</a>
</li>
<li>
<g-emoji alias="page_facing_up" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4c4.png">ğŸ“„</g-emoji> The training course and material is available as Galaxy page (<g-emoji alias="+1" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png">ğŸ‘</g-emoji> hands-on training).</li>
<li>
<g-emoji alias="book" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f4d6.png">ğŸ“–</g-emoji> The training course and material is available as annotated text (<g-emoji alias="+1" fallback-src="https://assets-cdn.github.com/images/icons/emoji/unicode/1f44d.png">ğŸ‘</g-emoji> hands-on training).</li>
</ul>

<h1>
<a id="literature" class="anchor" href="#literature" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Literature</h1>

<h2>
<a id="sequencing-biases" class="anchor" href="#sequencing-biases" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Sequencing biases</h2>

<p><strong>Meacham et al. (2011):</strong> <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-12-451">Identification and correction of systematic error in high-throughput sequence data</a>, (doi:10.1186/1471-2105-12-451) - Correcting for systematic base pair errors in deep sequencing; important paper if you want to look at any allele-specificy or if you're interested in SNPs</p>

<p><strong>Benjamini &amp; Speed (2012):</strong> <a href="http://nar.oxfordjournals.org/content/40/10/e72.long">Summarizing and correcting the GC content bias in high-throughput sequencing</a>, (doi: 10.1093/nar/gks001) - GC bias of deeply sequenced samples; very good paper that systematically assesses many possible sourced of GC bias for deeply sequenced samples and eventually pinpoints it to the DNA polymerase</p>

<p><strong>A collection of papers on quality controls for various NGS applications:</strong> <a href="http://journal.frontiersin.org/researchtopic/1683/quality-assessment-and-control-of-high-throughput-sequencing-data">Frontiers in Genetics (2014)</a></p>

<h2>
<a id="deep-sequencing" class="anchor" href="#deep-sequencing" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Deep sequencing</h2>

<p><strong>Zentner and Henikoff (2012):</strong> <a href="http://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-10-250">Surveying the epigenomic landscape, one base at a time</a>, (doi:10.1186/gb-2012-13-10-250) - Overview of popular *-seq techniques; very nice description of DNase-seq, MNase-seq, FAIRE-seq etc.</p>

<p><strong>Son and Taylor (2011):</strong> <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3076644/">Preparing DNA Libraries for Multiplexed Paired-End Deep Sequencing for Illumina GA Sequencers</a>, (doi:10.1002/9780471729259.mc01e04s20) - Paper on multiplexing; describes the individual steps of the Illumina deep sequencing protocols quite in detail</p>

<p><strong>Illumina's technical report</strong> - focuses on <a href="http://www.illumina.com/technology.html">Illumina's sequencing technology</a>; nice educative figures</p>

<h2>
<a id="mapping-of-short-ngs-reads" class="anchor" href="#mapping-of-short-ngs-reads" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Mapping of short NGS reads</h2>

<p><strong>informative slides</strong> <a href="http://people.binf.ku.dk/krogh/tmp/Mapping_Krogh_Monday.pdf">Mapping of sequencing reads</a> - Introduction to various aspects of NGS read mapping</p>

<p><strong>Fonseca et al. (2012):</strong> <a href="http://bioinformatics.oxfordjournals.org/content/28/24/3169.full">Tools for mapping high-throughput sequencing data</a>, (doi:10.1093/bioinformatics/bts605) - An excellent starting point despite its "old" age, you will learn a lot about the different philosophies behind the read alignment tools!</p>

<p><strong>Hatem et al. (2013):</strong> <a href="http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-184">Benchmarking short sequence mapping tools</a>, (doi:10.1186/1471-2105-14-184) (spoiler alert: bowtie wins)</p>

<p><strong>Engstrom et al. (2013):</strong> <a href="http://www.nature.com/nmeth/journal/v10/n12/full/nmeth.2722.html">Systematic evaluation of spliced alignment programs for RNA-seq data</a>, (doi:10.1038/nmeth.2722) </p>

<p><strong>Genome Mappability</strong></p>

<p><strong>Lee and Schatz (2012):</strong> <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3413383/?report=reader">The reliability of short read mapping</a>, (doi:10.1093/bioinformatics/bts330)  - Very detailed paper about genome mappability issues that presents a new suite of tools for taking the mappability into account</p>

<p><em>mappability maps</em> can be downloaded <a href="http://archive.gersteinlab.org/proj/PeakSeq/Mappability_Map/">here</a>  </p>

<h2>
<a id="ngs-data-formats" class="anchor" href="#ngs-data-formats" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>NGS data formats</h2>

<ul>
<li><p>UCSC has a very good overview with brief descriptions of BED, bedGraph, bigWig etc.: <a href="https://genome.ucsc.edu/FAQ/FAQformat.html">https://genome.ucsc.edu/FAQ/FAQformat.html</a></p></li>
<li><p><a href="http://gatkforums.broadinstitute.org/gatk/discussion/1268/how-should-i-interpret-vcf-files-produced-by-the-gatk">VCF format</a> (encoding SNPs, indels etc.): Very readable, albeit not exhausting description</p></li>
<li><p>Transcriptomes are often saved in <a href="http://www.sequenceontology.org/gff3.shtml">GFF3 format</a> (this is what TopHat needs, for example), but just to make things more complicated, GTF is another format used for transcriptome information, too (<a href="http://gmod.org/wiki/GFF2">here</a> are more information on GTF)</p></li>
</ul>

<h2>
<a id="bioinformatic-tools-linux-r-bedtools-etc---manuals-courses-original-papers" class="anchor" href="#bioinformatic-tools-linux-r-bedtools-etc---manuals-courses-original-papers" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Bioinformatic Tools (Linux, R, BEDTools etc.) - Manuals, courses, original papers</h2>

<ul>
<li>Why and how is bioinformatics software special? <strong>Altschul et a. (2013)</strong> <a href="http://www.ncbi.nlm.nih.gov/pubmed/24104757">The anatomy of successful computational biology software</a>, (doi:10.1038/nbt.2721) <strong>(Highly recommended to read!)</strong>
</li>
<li>
<strong>Bild et al. (2014)</strong> <a href="http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001744">A Field Guide to Genomics Research</a>, (doi:10.1371/journal.pbio.1001744) - Very readable introduction about the different caveats of genomics research (with cute cartoons!)</li>
</ul>

<p><strong>Linux Command Line</strong></p>

<ul>
<li>
<a href="http://korflab.ucdavis.edu/Unix_and_Perl/unix_and_perl_v3.1.1.html">Linux &amp; Perl Primer for Biologists</a> - Very entertaining introduction to command line commands and perl scripts with a focus on bioinformatic application, i.e. handling of DNA sequences</li>
<li>
<a href="http://www.ee.surrey.ac.uk/Teaching/Unix/">Linux Tutorial for Beginners</a> - Thorough, but concise online tutorial introducing the very basics of handling the Linux command line</li>
<li>
<a href="http://www.freeos.com/guides/lsst/index.html">Writing Linux shell scripts</a> - Useful for slightly more advanced Linux command line users</li>
</ul>

<p><strong>R</strong></p>

<ul>
<li>
<a href="http://www.uwyo.edu/mdillon/hor.html">Hands on R course</a> - For beginners - R is probably the most widely used open-source statistical software; through our epicenter website you can also access RStudio which provides are very nice interface to working and plotting with R. In fact, most of the plots generated within Galaxy are generated through R scripts, so if you're not happy with the default formats of the Galaxy graphs, definitely have a look at R yourself. The learning curve is steep, but it is worth it.</li>
</ul>

<p><strong>BEDTools</strong> </p>

<ul>
<li>
<a href="http://bedtools.readthedocs.org">BEDTools Manual</a> - When working with genomic intervals (e.g. genes, peaks, enriched regions...), BEDTools are invaluable! The manual is a very good read and we refer to it almost daily.</li>
</ul>

<h1>
<a id="license" class="anchor" href="#license" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>License</h1>

<p>This work is licensed under the <a href="http://creativecommons.org/licenses/by/3.0/">Creative Commons Attribution 3.0 Unported License</a>.</p>

<h1>
<a id="acknowledgment-and-funding" class="anchor" href="#acknowledgment-and-funding" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Acknowledgment and Funding</h1>

<p>We would like to thank all contributors to our Galaxy courses, especially those mentioned in the <a href="CONTRIBUTORS.md">Contributors list</a>, the Galaxy community for their constant support and our funding partners.</p>

<p><a href="http://www.denbi.de/"><img src="https://raw.githubusercontent.com/bgruening/rbc_docs/master/logo/deNBI_Logo_rgb.png" width="15%"></a>  â€ƒ<a href="http://www.sfb992.uni-freiburg.de/"><img src="https://raw.githubusercontent.com/bgruening/presentations/bce348bb606c312d531c479e63a66efc2bc38d44/shared/resources/img/MEDEP.jpg" width="15%"></a>    â€ƒ<a href="http://www.ie-freiburg.mpg.de"><img src="https://raw.githubusercontent.com/bgruening/presentations/master/shared/resources/img/14_MPI_IE_logo_mit_180.gif" width="15%"></a>  â€ƒ<a href="https://www.uni-freiburg.de/"><img src="https://raw.githubusercontent.com/bgruening/presentations/a2e38e4b007994af798320db3a0131c4bb891c0e/shared/resources/img/logo_freiburg.jpg" width="15%"></a></p>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/bgruening">bgruening</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
