<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Search Tutorials</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/fontawesome/css/all.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/search.html">
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/">

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Search Tutorials">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2022-03-14 18:11:59 +0000">

        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Search Tutorials">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
    </head>
    <body data-spy="scroll" data-target="#toc">
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/search.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/search.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                    <button class="btn" data-theme="default">Default</button>
                    <button class="btn" data-theme="night">Night</button>
                    <button class="btn" data-theme="midnight">Midnight</button>
                    <button class="btn" data-theme="rainbow">Rainbow</button>
                    <button class="btn" data-theme="blm">‚úäüèø</button>
                    <button class="btn" data-theme="progress" >üè≥Ô∏è‚Äçüåà</button>
                    <button class="btn" data-theme="halloween">üéÉ</button>
                    <button class="btn" data-theme="straya">üá¶üá∫</button>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-04-01/search.html" title="Version 2023-04-01">2023-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-03-01/search.html" title="Version 2023-03-01">2023-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-02-01/search.html" title="Version 2023-02-01">2023-02-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section>
  <h1>Search Tutorials</h1>

  <script src="https://cdnjs.cloudflare.com/ajax/libs/lunr.js/2.3.9/lunr.min.js" integrity="sha512-4xUl/d6D6THrAnXAwGajXkoWaeMNwEKK4iNfq5DotEbLPAfk6FSxSP3ydNxqDgCw1c/0Z1Jg6L8h2j+++9BZmg==" crossorigin="anonymous"></script>

<div id="search-container">
	<input type="text" id="search-input" placeholder=" search..." class="nicer" /><i class="fas fa fa-search" aria-hidden="true"></i><span class="visually-hidden">search</span>

	<input type="checkbox" id="search-faqs" name="search-faqs" checked="" />
	<label for="search-faqs">Search FAQs</label>

	<input type="checkbox" id="search-tutorials" name="search-tutorials" checked="" />
	<label for="search-tutorials">Search Tutorials</label>

	<div class="search-results row" id="results-container"></div>
</div>

<!-- Configuration -->
<script>

var resources = {
  "/topics/admin/tutorials/systemd-supervisor/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Controlling Galaxy with systemd or Supervisor",
    "contributors": "Nate Coraor, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/systemd-supervisor/slides.html"
  },
  "/topics/admin/tutorials/interactive-tools/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Interactive Tools",
    "contributors": "Nate Coraor, Simon Gladman, Helena Rasche, Anthony Bretaudeau",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/interactive-tools/tutorial.html"
  },
  "/topics/admin/tutorials/object-store/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Distributed Object Storage",
    "contributors": "Nate Coraor, Helena Rasche, Gianmauro Cuccuru",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged storage\" href=\"/training-material/search?query=storage\" style=\"--color-primary: #ffc7ec; --color-darker: #ffa1e2; --color-dimmed: #ffb4e7;\">storage</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/object-store/tutorial.html"
  },
  "/topics/admin/tutorials/troubleshooting/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Troubleshooting",
    "contributors": "Martin ƒåech, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/troubleshooting/slides.html"
  },
  "/topics/admin/tutorials/cloudbursting/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy on the Cloud",
    "contributors": "Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/cloudbursting/slides.html"
  },
  "/topics/admin/tutorials/stuff/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Server: Other",
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/stuff/slides.html"
  },
  "/topics/admin/tutorials/monitoring/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with Telegraf and Grafana",
    "contributors": "Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?query=monitoring\" style=\"--color-primary: #ffccbb; --color-darker: #ffab88; --color-dimmed: #ffbca3;\">monitoring</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/monitoring/tutorial.html"
  },
  "/topics/admin/tutorials/database/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galactic Database",
    "contributors": "Martin ƒåech, Nicola Soranzo, Nate Coraor, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/database/slides.html"
  },
  "/topics/admin/tutorials/production/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Gearing towards production",
    "contributors": "Nate Coraor, Daniel Blankenberg, Abdulrahman Azab, Martin ƒåech",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/production/slides.html"
  },
  "/topics/admin/tutorials/backup-cleanup/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Server Maintenance: Cleanup, Backup, and Restoration",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffcbc2; --color-darker: #ffaa98; --color-dimmed: #ffbbae;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/backup-cleanup/tutorial.html"
  },
  "/topics/admin/tutorials/empathy/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Empathy",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/empathy/slides.html"
  },
  "/topics/admin/tutorials/data-library/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Data Libraries",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged storage\" href=\"/training-material/search?query=storage\" style=\"--color-primary: #ffc7ec; --color-darker: #ffa1e2; --color-dimmed: #ffb4e7;\">storage</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/data-library/tutorial.html"
  },
  "/topics/admin/tutorials/gxadmin/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with gxadmin",
    "contributors": "Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?query=monitoring\" style=\"--color-primary: #ffccbb; --color-darker: #ffab88; --color-dimmed: #ffbca3;\">monitoring</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/gxadmin/tutorial.html"
  },
  "/topics/admin/tutorials/users-groups-quotas/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "User, Role, Group, Quota, and Authentication managment",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/users-groups-quotas/slides.html"
  },
  "/topics/admin/tutorials/advanced-galaxy-customisation/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Advanced customisation of a Galaxy instance",
    "contributors": "Bj√∂rn Gr√ºning, Helena Rasche, Martin ƒåech",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/advanced-galaxy-customisation/slides.html"
  },
  "/topics/admin/tutorials/celery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Setting up Celery Workers for Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/celery/tutorial.html"
  },
  "/topics/admin/tutorials/pulsar/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Running Jobs on Remote Resources with Pulsar",
    "contributors": "Nate Coraor, Simon Gladman, Marius van den Beek, Helena Rasche, Gianmauro Cuccuru, Mira Kuntz",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f86b; --color-darker: #00dc5e; --color-dimmed: #00ea64;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/pulsar/tutorial.html"
  },
  "/topics/admin/tutorials/ansible-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Installation with Ansible",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffcbc2; --color-darker: #ffaa98; --color-dimmed: #ffbbae;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/external-auth/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "External Authentication",
    "contributors": "Nate Coraor, Nicola Soranzo, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged authentication\" href=\"/training-material/search?query=authentication\" style=\"--color-primary: #ebccff; --color-darker: #e1abff; --color-dimmed: #e6bbff;\">authentication</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/external-auth/tutorial.html"
  },
  "/topics/admin/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy from an administrator's point of view",
    "contributors": "Valentin Marcon, Nate Coraor, Bj√∂rn Gr√ºning, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/introduction/slides.html"
  },
  "/topics/admin/tutorials/cvmfs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Reference Data with CVMFS",
    "contributors": "Simon Gladman, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/cvmfs/tutorial.html"
  },
  "/topics/admin/tutorials/ansible/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Ansible",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/ansible/tutorial.html"
  },
  "/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Connecting Galaxy to a compute cluster",
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f86b; --color-darker: #00dc5e; --color-dimmed: #00ea64;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html"
  },
  "/topics/admin/tutorials/reference-genomes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Reference Data with Data Managers",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/reference-genomes/tutorial.html"
  },
  "/topics/admin/tutorials/uwsgi/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "uWSGI",
    "contributors": "Nate Coraor, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/uwsgi/slides.html"
  },
  "/topics/admin/tutorials/general-monitoring/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring",
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/general-monitoring/slides.html"
  },
  "/topics/admin/tutorials/galaxy-docker/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Docker and Galaxy",
    "contributors": "B√©r√©nice Batut, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/galaxy-docker/slides.html"
  },
  "/topics/admin/tutorials/tool-management/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Tool Management with Ephemeris",
    "contributors": "Martin ƒåech, Helena Rasche, Nicola Soranzo",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tools\" href=\"/training-material/search?query=tools\" style=\"--color-primary: #ffc8de; --color-darker: #ffa4cb; --color-dimmed: #ffb7d4;\">tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/tool-management/tutorial.html"
  },
  "/topics/admin/tutorials/k8s-managing-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Managing Galaxy on Kubernetes",
    "contributors": "Nuwan Goonasekera, Enis Afgan, Alex Mahmoud, Pablo Moreno, John Davis",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged kubernetes\" href=\"/training-material/search?query=kubernetes\" style=\"--color-primary: #ffc9d7; --color-darker: #ffa6bf; --color-dimmed: #ffb8cb;\">kubernetes</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/k8s-managing-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/beacon/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying a Beacon v1 in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ga4gh\" href=\"/training-material/search?query=ga4gh\" style=\"--color-primary: #ffc9da; --color-darker: #ffa5c4; --color-dimmed: #ffb7cf;\">ga4gh</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged beacon\" href=\"/training-material/search?query=beacon\" style=\"--color-primary: #ffc9d8; --color-darker: #ffa5c2; --color-dimmed: #ffb8cd;\">beacon</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/beacon/tutorial.html"
  },
  "/topics/admin/tutorials/tiaas/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Training Infrastructure as a Service (TIaaS)",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged training\" href=\"/training-material/search?query=training\" style=\"--color-primary: #ffc5f6; --color-darker: #ff9ef1; --color-dimmed: #ffb2f4;\">training</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f86b; --color-darker: #00dc5e; --color-dimmed: #00ea64;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/tiaas/tutorial.html"
  },
  "/topics/admin/tutorials/stop-worrying-love-systemd/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "How I learned to stop worrying and love the systemd",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged systemd\" href=\"/training-material/search?query=systemd\" style=\"--color-primary: #99ee00; --color-darker: #88d400; --color-dimmed: #90e100;\">systemd</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/stop-worrying-love-systemd/tutorial.html"
  },
  "/topics/admin/tutorials/ftp/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Enable upload via FTP",
    "contributors": "Lucille Delisle",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data\" href=\"/training-material/search?query=data\" style=\"--color-primary: #00f850; --color-darker: #00dd47; --color-dimmed: #00eb4b;\">data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/ftp/tutorial.html"
  },
  "/topics/admin/tutorials/upgrading/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Upgrading Galaxy",
    "contributors": "Simon Gladman",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/upgrading/tutorial.html"
  },
  "/topics/admin/tutorials/wireguard-headscale/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying Tailscale/Headscale for private mesh networking",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged wireguard\" href=\"/training-material/search?query=wireguard\" style=\"--color-primary: #abdeff; --color-darker: #60caff; --color-dimmed: #8ad4ff;\">wireguard</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged networking\" href=\"/training-material/search?query=networking\" style=\"--color-primary: #ffc6f0; --color-darker: #ffa0e8; --color-dimmed: #ffb4ec;\">networking</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/wireguard-headscale/tutorial.html"
  },
  "/topics/admin/tutorials/database-schema/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Database schema",
    "contributors": "Hans-Rudolf Hotz, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/database-schema/tutorial.html"
  },
  "/topics/admin/tutorials/jenkins/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Automation with Jenkins",
    "contributors": "Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged automation\" href=\"/training-material/search?query=automation\" style=\"--color-primary: #dfd0ff; --color-darker: #ccb3ff; --color-dimmed: #d5c1ff;\">automation</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/jenkins/tutorial.html"
  },
  "/topics/admin/tutorials/reports/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with Reports",
    "contributors": "Nate Coraor, Bj√∂rn Gr√ºning, Simon Gladman, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?query=monitoring\" style=\"--color-primary: #ffccbb; --color-darker: #ffab88; --color-dimmed: #ffbca3;\">monitoring</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/reports/tutorial.html"
  },
  "/topics/admin/tutorials/wireguard/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying Wireguard for private mesh networking",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged wireguard\" href=\"/training-material/search?query=wireguard\" style=\"--color-primary: #abdeff; --color-darker: #60caff; --color-dimmed: #8ad4ff;\">wireguard</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged networking\" href=\"/training-material/search?query=networking\" style=\"--color-primary: #ffc6f0; --color-darker: #ffa0e8; --color-dimmed: #ffb4ec;\">networking</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/wireguard/tutorial.html"
  },
  "/topics/admin/tutorials/sentry/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Monitoring Galaxy and Pulsar with Sentry",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/sentry/tutorial.html"
  },
  "/topics/admin/tutorials/customization/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Customizing the look of Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/customization/tutorial.html"
  },
  "/topics/admin/tutorials/job-destinations/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Mapping Jobs to Destinations using TPV",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f86b; --color-darker: #00dc5e; --color-dimmed: #00ea64;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/job-destinations/tutorial.html"
  },
  "/topics/admin/tutorials/apptainer/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Use Apptainer containers for running Galaxy jobs",
    "contributors": "Torfinn Nome, Marius van den Beek, Matthias Bernt, Helena Rasche, Mira Kuntz",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f86b; --color-darker: #00dc5e; --color-dimmed: #00ea64;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2da; --color-dimmed: #ffb5e1;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffcacc; --color-darker: #ffa8ab; --color-dimmed: #ffb9bc;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/apptainer/tutorial.html"
  },
  "/topics/admin/tutorials/tus/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Performant Uploads with TUS",
    "contributors": "Marius van den Beek, Helena Rasche, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/tus/tutorial.html"
  },
  "/topics/admin/tutorials/customization-manual/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Customizing the look of Galaxy (Manual)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/customization-manual/tutorial.html"
  },
  "/topics/admin/tutorials/cvmfs-manual/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Reference Data with CVMFS without Ansible",
    "contributors": "Simon Gladman, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/cvmfs-manual/tutorial.html"
  },
  "/topics/admin/tutorials/k8s-deploying-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Installation on Kubernetes",
    "contributors": "Pablo Moreno, Enis Afgan, Nuwan Goonasekera, Alex Mahmoud, John Davis",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged kubernetes\" href=\"/training-material/search?query=kubernetes\" style=\"--color-primary: #ffc9d7; --color-darker: #ffa6bf; --color-dimmed: #ffb8cb;\">kubernetes</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/k8s-deploying-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/terraform/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying a compute cluster in OpenStack via Terraform",
    "contributors": "Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged terraform\" href=\"/training-material/search?query=terraform\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">terraform</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffcbc2; --color-darker: #ffaa98; --color-dimmed: #ffbbae;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cloud\" href=\"/training-material/search?query=cloud\" style=\"--color-primary: #ffcacb; --color-darker: #ffa8aa; --color-dimmed: #ffb9bb;\">cloud</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/terraform/tutorial.html"
  },
  "/topics/ai4life/tutorials/example-slides/slides.html": {
    "type": "Tutorial",
    "topic": "Image analysis using Deep Learning",
    "title": "Example Slide Deck",
    "contributors": "Beatriz Serrano-Solano",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged some\" href=\"/training-material/search?query=some\" style=\"--color-primary: #ffc5f5; --color-darker: #ff9ff0; --color-dimmed: #ffb3f2;\">some</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?query=tags\" style=\"--color-primary: #ffc9d8; --color-darker: #ffa5c1; --color-dimmed: #ffb8cc;\">tags</a>"
    ],
    "url": "/training-material/topics/ai4life/tutorials/example-slides/slides.html"
  },
  "/topics/assembly/tutorials/general-introduction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "An Introduction to Genome Assembly",
    "contributors": "Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/general-introduction/tutorial.html"
  },
  "/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "VGP assembly pipeline",
    "contributors": "Delphine Lariviere, Alex Ostrovsky, Crist√≥bal Gallardo, Anna Syme, Linelle Abueg, Brandon Pickett, Giulio Formenti, Marcella Sozzoni",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?query=pacbio\" style=\"--color-primary: #ffc8e3; --color-darker: #ffa3d3; --color-dimmed: #ffb6db;\">pacbio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged VGP\" href=\"/training-material/search?query=VGP\" style=\"--color-primary: #ffcea5; --color-darker: #ffae4d; --color-dimmed: #ffbe81;\">VGP</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/assembly-quality-control/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome Assembly Quality Control",
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, Erwan Corre, Laura Leroi, St√©phanie Robin",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged quality control\" href=\"/training-material/search?query=quality control\" style=\"--color-primary: #cfd5ff; --color-darker: #b0bbff; --color-dimmed: #c0c8ff;\">quality control</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html"
  },
  "/topics/assembly/tutorials/unicycler-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Unicycler Assembly",
    "contributors": "Anton Nekrutenko, Delphine Lariviere, Simon Gladman",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "De Bruijn Graph Assembly",
    "contributors": "Simon Gladman, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/ecoli_comparison/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Making sense of a newly assembled genome",
    "contributors": "Anton Nekrutenko, Delphine Lariviere, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html"
  },
  "/topics/assembly/tutorials/largegenome/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Large genome assembly and polishing",
    "contributors": "Anna Syme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged polishing\" href=\"/training-material/search?query=polishing\" style=\"--color-primary: #ffc8e2; --color-darker: #ffa3d2; --color-dimmed: #ffb6da;\">polishing</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">nanopore</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/largegenome/tutorial.html"
  },
  "/topics/assembly/tutorials/metagenomics-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Assembly of metagenomic sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #f3d700; --color-darker: #d8c000; --color-dimmed: #e6cb00;\">metagenomics</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/mrsa-illumina/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome Assembly of MRSA using Illumina MiSeq Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?query=illumina\" style=\"--color-primary: #ffcf95; --color-darker: #fbb200; --color-dimmed: #ffc062;\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?query=amr\" style=\"--color-primary: #c9d6ff; --color-darker: #a5beff; --color-dimmed: #b7caff;\">amr</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html"
  },
  "/topics/assembly/tutorials/vgp_workflow_training/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "VGP assembly pipeline - short version",
    "contributors": "Delphine Lariviere, Alex Ostrovsky, Crist√≥bal Gallardo, Brandon Pickett, Linelle Abueg",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?query=pacbio\" style=\"--color-primary: #ffc8e3; --color-darker: #ffa3d3; --color-dimmed: #ffb6db;\">pacbio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged VGP\" href=\"/training-material/search?query=VGP\" style=\"--color-primary: #ffcea5; --color-darker: #ffae4d; --color-dimmed: #ffbe81;\">VGP</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_workflow_training/tutorial.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Chloroplast genome assembly",
    "contributors": "Anna Syme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ff400; --color-darker: #62d900; --color-dimmed: #69e600;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">nanopore</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "SARS-CoV-2",
    "title": "Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads",
    "contributors": "Wolfgang Maier",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html"
  },
  "/topics/assembly/tutorials/flye-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome assembly using PacBio data",
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, Erwan Corre, Laura Leroi, St√©phanie Robin, Erasmus+ Programme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?query=pacbio\" style=\"--color-primary: #ffc8e3; --color-darker: #ffa3d3; --color-dimmed: #ffb6db;\">pacbio</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/mrsa-nanopore/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome Assembly of MRSA using Oxford Nanopore MinION Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?query=amr\" style=\"--color-primary: #c9d6ff; --color-darker: #a5beff; --color-dimmed: #b7caff;\">amr</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html"
  },
  "/topics/assembly/tutorials/algorithms-introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Deeper look into Genome Assembly algorithms",
    "contributors": "Anthony Bretaudeau, Simon Gladman, Anton Nekrutenko, Delphine Lariviere",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/algorithms-introduction/slides.html"
  },
  "/topics/assembly/tutorials/get-started-genome-assembly/slides.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "An introduction to get started in genome assembly and annotation",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/get-started-genome-assembly/slides.html"
  },
  "/topics/climate/tutorials/climate-101/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Getting your hands-on climate data",
    "contributors": "Anne Fouilloux",
    "tags": [

    ],
    "url": "/training-material/topics/climate/tutorials/climate-101/tutorial.html"
  },
  "/topics/climate/tutorials/pangeo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools",
    "contributors": "Anne Fouilloux",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00f5a9; --color-darker: #00da95; --color-dimmed: #00e79f;\">pangeo</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/pangeo/tutorial.html"
  },
  "/topics/climate/tutorials/pangeo-notebook/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Pangeo Notebook in Galaxy - Introduction to Xarray",
    "contributors": "Anne Fouilloux",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00f5a9; --color-darker: #00da95; --color-dimmed: #00e79f;\">pangeo</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/pangeo-notebook/tutorial.html"
  },
  "/topics/climate/tutorials/fates/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES)",
    "contributors": "Anne Fouilloux, Hui Tang",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/fates/tutorial.html"
  },
  "/topics/climate/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Introduction to climate data",
    "contributors": "Anne Fouilloux, Jean Iaquinta",
    "tags": [

    ],
    "url": "/training-material/topics/climate/tutorials/introduction/slides.html"
  },
  "/topics/climate/tutorials/panoply/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Visualize Climate data with Panoply netCDF viewer",
    "contributors": "Anne Fouilloux",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/panoply/tutorial.html"
  },
  "/topics/climate/tutorials/fates-jupyterlab/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab",
    "contributors": "Anne Fouilloux",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "High Throughput Molecular Dynamics and Analysis",
    "contributors": "Simon Bray, Tharindu Senapathi, Christopher Barnett, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Protein-ligand docking",
    "contributors": "Simon Bray",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Protein target prediction of a bioactive ligand with Align-it and ePharmaLib",
    "contributors": "Aur√©lien F. A. Moumbock, Simon Bray",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Running molecular dynamics simulations using GROMACS",
    "contributors": "Simon Bray",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Running molecular dynamics simulations using NAMD",
    "contributors": "Christopher Barnett, Tharindu Senapathi, Simon Bray",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Analysis of molecular dynamics simulations",
    "contributors": "Christopher Barnett, Tharindu Senapathi, Simon Bray, Nadia Gou√©",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html": {
    "type": "Tutorial",
    "topic": "SARS-CoV-2",
    "title": "Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring",
    "contributors": "Simon Bray",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>"
    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Setting up molecular systems",
    "contributors": "Christopher Barnett, Simon Bray, Nadia Gou√©",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis",
    "contributors": "Marius van den Beek, Dave Clements, Daniel Blankenberg, Anton Nekrutenko",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc8e3; --color-darker: #ffa3d4; --color-dimmed: #ffb6dc;\">virology</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data",
    "contributors": "Wolfgang Maier, B√©r√©nice Batut",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc8e3; --color-darker: #ffa3d4; --color-dimmed: #ffb6dc;\">virology</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Submitting sequence data to ENA",
    "contributors": "Miguel Roncoroni, Bert Droesbeke",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Automating Galaxy workflows using the command line",
    "contributors": "Simon Bray, Wolfgang Maier",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffcdac; --color-darker: #ffad64; --color-dimmed: #ffbd8d;\">workflows</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged variant-analysis\" href=\"/training-material/search?query=variant-analysis\" style=\"--color-primary: #00f3be; --color-darker: #00d8a9; --color-dimmed: #00e6b4;\">variant-analysis</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Removal of human reads from SARS-CoV-2 sequencing data",
    "contributors": "Wolfgang Maier",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>"
    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ncbi\" href=\"/training-material/search?query=ncbi\" style=\"--color-primary: #ffc9d9; --color-darker: #ffa5c3; --color-dimmed: #ffb7ce;\">ncbi</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">covid19</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html"
  },
  "/topics/dev/tutorials/containers/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool Dependencies and Containers",
    "contributors": "John Chilton, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/containers/slides.html"
  },
  "/topics/dev/tutorials/tool-generators/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "ToolFactory: Generating Tools From Simple Scripts",
    "contributors": "Ross Lazarus, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/tutorial.html"
  },
  "/topics/dev/tutorials/conda_sys/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Prerequisites for building software/conda packages",
    "contributors": "Anthony Bretaudeau, Cyril Monjeaud",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/conda_sys/slides.html"
  },
  "/topics/dev/tutorials/interactive-environments/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Environments",
    "contributors": "Saskia Hiltemann, Bj√∂rn Gr√ºning, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/dev/tutorials/interactive-environments/slides.html"
  },
  "/topics/dev/tutorials/webhooks/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Webhooks",
    "contributors": "Clemens Blank, Martin ƒåech, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/webhooks/tutorial.html"
  },
  "/topics/dev/tutorials/bioblend-api/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Scripting Galaxy using the API and BioBlend",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/bioblend-api/tutorial.html"
  },
  "/topics/dev/tutorials/conda/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool Dependencies and Conda",
    "contributors": "Nicola Soranzo, John Chilton, Bj√∂rn Gr√ºning, Herv√© M√©nager",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/conda/slides.html"
  },
  "/topics/dev/tutorials/architecture/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Code Architecture",
    "contributors": "John Chilton, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/architecture/slides.html"
  },
  "/topics/dev/tutorials/visualization-generic/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Generic plugins",
    "contributors": "Saskia Hiltemann, Youri Hoogstrate",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/visualization-generic/tutorial.html"
  },
  "/topics/dev/tutorials/tool-integration/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool development and integration into Galaxy",
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut, Anthony Bretaudeau, John Chilton, Nicola Soranzo, Bj√∂rn Gr√ºning, Gildas Le Corguill√©",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-integration/slides.html"
  },
  "/topics/dev/tutorials/visualization-charts/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "JavaScript plugins",
    "contributors": "Saskia Hiltemann, Youri Hoogstrate, Bj√∂rn Gr√ºning, Aysam Guerler, Dannon Baker",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/visualization-charts/tutorial.html"
  },
  "/topics/dev/tutorials/interactive-tours/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Tours",
    "contributors": "B√©r√©nice Batut, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/interactive-tours/slides.html"
  },
  "/topics/dev/tutorials/toolshed/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool Shed: sharing Galaxy tools",
    "contributors": "Anthony Bretaudeau, Loraine Gu√©guen, St√©phanie Legras, B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/toolshed/slides.html"
  },
  "/topics/dev/tutorials/interactive-tools/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Tools",
    "contributors": "Estelle Ancelet, Yvan Le Bras, Cameron Hyde, Lain Pavot, Anthony Bretaudeau, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/interactive-tools/tutorial.html"
  },
  "/topics/dev/tutorials/data-source-integration/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Data source integration",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Gianmauro Cuccuru, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/data-source-integration/tutorial.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Creating Galaxy tools from Conda Through Deployment",
    "contributors": "Alex Ostrovsky, Dave B., Matthias Bernt",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html"
  },
  "/topics/dev/tutorials/core-contributing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Contributing a New Feature to Galaxy Core",
    "contributors": "John Chilton, John Davis",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/core-contributing/tutorial.html"
  },
  "/topics/dev/tutorials/debugging/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Debugging Galaxy",
    "contributors": "Assunta DeSanto, John Davis, John Chilton",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/debugging/tutorial.html"
  },
  "/topics/dev/tutorials/bioblend-dev/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Contributing to BioBlend as a developer",
    "contributors": "Erik Schill, Simon Bray",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/bioblend-dev/tutorial.html"
  },
  "/topics/dev/tutorials/tool-generators-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "ToolFactory: Generating Tools From More Complex Scripts",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators-advanced/tutorial.html"
  },
  "/topics/dev/tutorials/writing_tests/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Writing Automated Tests for Galaxy",
    "contributors": "John Davis, John Chilton",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/writing_tests/tutorial.html"
  },
  "/topics/ecology/tutorials/champs-blocs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Champs blocs indicators",
    "contributors": "Marie Josse, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html"
  },
  "/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "RAD-Seq de-novo data analysis",
    "contributors": "Yvan Le Bras",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?query=RAD-seq\" style=\"--color-primary: #fff5d1; --color-darker: #f6da00; --color-dimmed: #ffe875;\">RAD-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species populations EBV class\" href=\"/training-material/search?query=Species populations EBV class\" style=\"--color-primary: #00f0df; --color-darker: #00d6c6; --color-dimmed: #00e3d3;\">Species populations EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eccbff; --color-darker: #e2aaff; --color-dimmed: #e7bbff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00f3c9; --color-darker: #00d8b3; --color-dimmed: #00e5be;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Biodiversity data exploration",
    "contributors": "Olivier Norvez, Marie Josse, Coline Royaux, Yvan Le Bras",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged taxonomic data\" href=\"/training-material/search?query=taxonomic data\" style=\"--color-primary: #76f300; --color-darker: #69d800; --color-dimmed: #6fe600;\">taxonomic data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data quality\" href=\"/training-material/search?query=data quality\" style=\"--color-primary: #ffc8e2; --color-darker: #ffa3d2; --color-dimmed: #ffb6da;\">data quality</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html"
  },
  "/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Compute and analyze biodiversity metrics with PAMPA toolsuite",
    "contributors": "Coline Royaux, Yvan Le Bras, Jean-Baptiste Mihoub",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species population EBV class\" href=\"/training-material/search?query=Species population EBV class\" style=\"--color-primary: #00f0df; --color-darker: #00d6c6; --color-dimmed: #00e3d3;\">Species population EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Community composition EBV class\" href=\"/training-material/search?query=Community composition EBV class\" style=\"--color-primary: #b9dbff; --color-darker: #84c5ff; --color-dimmed: #a0d0ff;\">Community composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eccbff; --color-darker: #e2aaff; --color-dimmed: #e7bbff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00f3c9; --color-darker: #00d8b3; --color-dimmed: #00e5be;\">EBV workflow</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged modeling\" href=\"/training-material/search?query=modeling\" style=\"--color-primary: #e1cfff; --color-darker: #cfb1ff; --color-dimmed: #d8c1ff;\">modeling</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html"
  },
  "/topics/ecology/tutorials/remote-sensing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Sentinel 2 biodiversity",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Ecology\" href=\"/training-material/search?query=Ecology\" style=\"--color-primary: #ffc4fd; --color-darker: #ff9cfc; --color-dimmed: #ffb1fd;\">Ecology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Remote sensing\" href=\"/training-material/search?query=Remote sensing\" style=\"--color-primary: #ffc9da; --color-darker: #ffa5c4; --color-dimmed: #ffb7cf;\">Remote sensing</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html"
  },
  "/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "RAD-Seq Reference-based data analysis",
    "contributors": "Yvan Le Bras",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?query=RAD-seq\" style=\"--color-primary: #fff5d1; --color-darker: #f6da00; --color-dimmed: #ffe875;\">RAD-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species population EBV class\" href=\"/training-material/search?query=Species population EBV class\" style=\"--color-primary: #00f0df; --color-darker: #00d6c6; --color-dimmed: #00e3d3;\">Species population EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eccbff; --color-darker: #e2aaff; --color-dimmed: #e7bbff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00f3c9; --color-darker: #00d8b3; --color-dimmed: #00e5be;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "RAD-Seq to construct genetic maps",
    "contributors": "Yvan Le Bras",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?query=RAD-seq\" style=\"--color-primary: #fff5d1; --color-darker: #f6da00; --color-dimmed: #ffe875;\">RAD-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eccbff; --color-darker: #e2aaff; --color-dimmed: #e7bbff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00f3c9; --color-darker: #00d8b3; --color-dimmed: #00e5be;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/x-array-map-plot/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Visualization of Climate Data using  NetCDF xarray Map Plotting",
    "contributors": "Soumya Jha",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00f5a9; --color-darker: #00da95; --color-dimmed: #00e79f;\">pangeo</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html"
  },
  "/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Preparing genomic data for phylogeny reconstruction",
    "contributors": "Miguel Roncoroni, Brigida Gallone",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged phylogeny\" href=\"/training-material/search?query=phylogeny\" style=\"--color-primary: #00f915; --color-darker: #00de11; --color-dimmed: #00ec13;\">phylogeny</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data handling\" href=\"/training-material/search?query=data handling\" style=\"--color-primary: #fff8a0; --color-darker: #e8df00; --color-dimmed: #f6ed00;\">data handling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged functional annotation\" href=\"/training-material/search?query=functional annotation\" style=\"--color-primary: #86e4ff; --color-darker: #00d1f4; --color-dimmed: #3cddff;\">functional annotation</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html"
  },
  "/topics/ecology/tutorials/regionalGAM/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Regional GAM",
    "contributors": "Clara Urfer, Elisa Michon, Yvan Le Bras, B√©r√©nice Batut, Simon Benateau",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species populations EBV class\" href=\"/training-material/search?query=Species populations EBV class\" style=\"--color-primary: #00f0df; --color-darker: #00d6c6; --color-dimmed: #00e3d3;\">Species populations EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species traits EBV class\" href=\"/training-material/search?query=Species traits EBV class\" style=\"--color-primary: #cfd5ff; --color-darker: #b0bbff; --color-dimmed: #c0c8ff;\">Species traits EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eccbff; --color-darker: #e2aaff; --color-dimmed: #e7bbff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00f3c9; --color-darker: #00d8b3; --color-dimmed: #00e5be;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html"
  },
  "/topics/ecology/tutorials/species-distribution-modeling/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Species distribution modeling",
    "contributors": "Elisa Michon, Yvan Le Bras, B√©r√©nice Batut",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged modeling\" href=\"/training-material/search?query=modeling\" style=\"--color-primary: #e1cfff; --color-darker: #cfb1ff; --color-dimmed: #d8c1ff;\">modeling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gbif\" href=\"/training-material/search?query=gbif\" style=\"--color-primary: #ffcacb; --color-darker: #ffa8aa; --color-dimmed: #ffb9bb;\">gbif</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged species populations EBV class\" href=\"/training-material/search?query=species populations EBV class\" style=\"--color-primary: #e7dc00; --color-darker: #cec300; --color-dimmed: #dacf00;\">species populations EBV class</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html"
  },
  "/topics/ecology/tutorials/obisindicators/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Obis marine indicators",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/ecology/tutorials/obisindicators/tutorial.html"
  },
  "/topics/ecology/tutorials/gbif_cleaning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Cleaning GBIF data for the use in Ecology",
    "contributors": "Yvan Le Bras, Simon Benateau",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gbif\" href=\"/training-material/search?query=gbif\" style=\"--color-primary: #ffcacb; --color-darker: #ffa8aa; --color-dimmed: #ffb9bb;\">gbif</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffd084; --color-darker: #f3b500; --color-dimmed: #ffc133;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data cleaning\" href=\"/training-material/search?query=data cleaning\" style=\"--color-primary: #00f863; --color-darker: #00dd57; --color-dimmed: #00ea5d;\">data cleaning</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html"
  },
  "/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Metabarcoding/eDNA through Obitools",
    "contributors": "Coline Royaux, Olivier Norvez, Eric Coissac, Fr√©d√©ric Boyer, Yvan Le Bras",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Community composition EBV class\" href=\"/training-material/search?query=Community composition EBV class\" style=\"--color-primary: #b9dbff; --color-darker: #84c5ff; --color-dimmed: #a0d0ff;\">Community composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eccbff; --color-darker: #e2aaff; --color-dimmed: #e7bbff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00f3c9; --color-darker: #00d8b3; --color-dimmed: #00e5be;\">EBV workflow</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eDNA\" href=\"/training-material/search?query=eDNA\" style=\"--color-primary: #fffc00; --color-darker: #e2e000; --color-dimmed: #f1ee00;\">eDNA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Metabarcoding\" href=\"/training-material/search?query=Metabarcoding\" style=\"--color-primary: #ffd175; --color-darker: #eeb700; --color-dimmed: #fcc300;\">Metabarcoding</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html"
  },
  "/topics/ecology/tutorials/panoply_ebv/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Visualize EBV cube data with Panoply netCDF viewer",
    "contributors": "Yvan Le Bras, Coline Royaux, Marie Josse, Anne Fouilloux",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV cube\" href=\"/training-material/search?query=EBV cube\" style=\"--color-primary: #ffc4fd; --color-darker: #ff9cfc; --color-dimmed: #ffb1fd;\">EBV cube</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Data visualization\" href=\"/training-material/search?query=Data visualization\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">Data visualization</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/panoply_ebv/tutorial.html"
  },
  "/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Formation of the Super-Structures on the Inactive X",
    "contributors": "Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez, Leily Rabbani",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?query=ChIP-seq\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">ChIP-seq</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html"
  },
  "/topics/epigenetics/tutorials/methylation-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "DNA Methylation data analysis",
    "contributors": "Joachim Wolff, Devon Ryan",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html"
  },
  "/topics/epigenetics/tutorials/hicexplorer/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer",
    "contributors": "Joachim Wolff, Fidel Ramirez, Vivek Bhardwaj, Ekaterina Polkh",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html"
  },
  "/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Identification of the binding sites of the Estrogen receptor",
    "contributors": "Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?query=ChIP-seq\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">ChIP-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "ATAC-Seq data analysis",
    "contributors": "Lucille Delisle, Maria Doyle, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html"
  },
  "/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)",
    "contributors": "Mallory Freeberg, Mo Heydarian, Vivek Bhardwaj, Joachim Wolff, Anika Erxleben",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?query=ChIP-seq\" style=\"--color-primary: #ffc7e7; --color-darker: #ffa2d9; --color-dimmed: #ffb5e0;\">ChIP-seq</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html"
  },
  "/topics/epigenetics/tutorials/ewas-suite/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Infinium Human Methylation BeadChip",
    "contributors": "Katarzyna Murat, Krzysztof Poterlowicz",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html"
  },
  "/topics/epigenetics/tutorials/introduction-dna-methylation/slides.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Introduction to DNA Methylation data analysis",
    "contributors": "Joachim Wolff, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/introduction-dna-methylation/slides.html"
  },
  "/topics/epigenetics/tutorials/introduction-chip-seq/slides.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Introduction to ChIP-Seq data analysis",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/introduction-chip-seq/slides.html"
  },
  "/topics/evolution/tutorials/mtb_phylogeny/tutorial.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Tree thinking for tuberculosis evolution and epidemiology",
    "contributors": "Christoph Stritt, Daniela Brites, Galo A. Goig",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryotes\" href=\"/training-material/search?query=prokaryotes\" style=\"--color-primary: #fff5d4; --color-darker: #f8d900; --color-dimmed: #ffe780;\">prokaryotes</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged phylogenetics\" href=\"/training-material/search?query=phylogenetics\" style=\"--color-primary: #00f863; --color-darker: #00dd57; --color-dimmed: #00ea5d;\">phylogenetics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged evolution\" href=\"/training-material/search?query=evolution\" style=\"--color-primary: #ffc8df; --color-darker: #ffa4cd; --color-dimmed: #ffb6d6;\">evolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/evolution/tutorials/mtb_phylogeny/tutorial.html"
  },
  "/topics/evolution/tutorials/mtb_transmission/tutorial.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Identifying tuberculosis transmission links: from SNPs to transmission clusters",
    "contributors": "Galo A. Goig, Daniela Brites, Christoph Stritt",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Using Workflow Parameters",
    "contributors": "Marius van den Beek, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffcdac; --color-darker: #ffad64; --color-dimmed: #ffbd8d;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/intermine/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "InterMine integration with Galaxy",
    "contributors": "Daniela Butano, Yo Yehudi",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/intermine/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Workflow Reports",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffcdac; --color-darker: #ffad64; --color-dimmed: #ffbd8d;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Creating, Editing and Importing Galaxy Workflows",
    "contributors": "Marius van den Beek",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffcdac; --color-darker: #ffad64; --color-dimmed: #ffbd8d;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/galaxy-on-dockstore/slides.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Galaxy workflows in Dockstore",
    "contributors": "Avani Khadilkar, Elizabeth Sheets, Charles Overbeck, Enis Afgan",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/galaxy-on-dockstore/slides.html"
  },
  "/topics/galaxy-interface/tutorials/get-data/slides.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Getting data into Galaxy",
    "contributors": "Anne Pajon, Dave Clements, Maria Doyle, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/get-data/slides.html"
  },
  "/topics/galaxy-interface/tutorials/name-tags/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Name tags for following complex histories",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Multisample Analysis",
    "contributors": "Anton Nekrutenko, Anne Pajon",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">collections</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Rule Based Uploader: Advanced",
    "contributors": "John Chilton, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?query=tags\" style=\"--color-primary: #ffc9d8; --color-darker: #ffa5c1; --color-dimmed: #ffb8cc;\">tags</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/history/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Understanding Galaxy history system",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/history/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/galaxy-intro-jupyter/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Use Jupyter notebooks in Galaxy",
    "contributors": "Delphine Lariviere",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/galaxy-intro-jupyter/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/rstudio/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "RStudio in Galaxy",
    "contributors": "B√©r√©nice Batut, Fotis E. Psomopoulos, Toby Hodges, Erasmus+ Programme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/rstudio/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "JupyterLab in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/collections/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Using dataset collections",
    "contributors": "Anton Nekrutenko",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">collections</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/group-tags/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Group tags for complex experimental designs",
    "contributors": "Marius van den Beek",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/group-tags/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-rules/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Rule Based Uploader",
    "contributors": "John Chilton, Helena Rasche",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?query=tags\" style=\"--color-primary: #ffc9d8; --color-darker: #ffa5c1; --color-dimmed: #ffb8cc;\">tags</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Extracting Workflows from Histories",
    "contributors": "Dave Clements",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffcdac; --color-darker: #ffad64; --color-dimmed: #ffbd8d;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/download-delete-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Downloading and Deleting Data in Galaxy",
    "contributors": "Simon Gladman, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/download-delete-data/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/search/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Searching Your History",
    "contributors": "Helena Rasche, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/search/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome annotation with Prokka",
    "contributors": "Anna Syme, Torsten Seemann, Simon Gladman",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/apollo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Refining Genome Annotations with Apollo (prokaryotes)",
    "contributors": "Anthony Bretaudeau, Helena Rasche, Nathan Dunn, Mateo Boudet, Erasmus+ Programme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/repeatmasker/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Masking repeats with RepeatMasker",
    "contributors": "Anthony Bretaudeau, Alexandre Cormier, Laura Leroi, Erwan Corre, St√©phanie Robin, Erasmus+ Programme, Jonathan Kreplak",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "From small to large-scale genome comparison",
    "contributors": "Esteban Perez-Wohlfeil",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Introduction to Genome Annotation",
    "contributors": "Anthony Bretaudeau, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/tutorials/introduction/slides.html"
  },
  "/topics/genome-annotation/tutorials/genome-annotation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome Annotation",
    "contributors": "Anika Erxleben, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/funannotate/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome annotation with Funannotate",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/gene-centric/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Comparative gene analysis in unannotated genomes",
    "contributors": "Anton Nekrutenko",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged evolution\" href=\"/training-material/search?query=evolution\" style=\"--color-primary: #ffc8df; --color-darker: #ffa4cd; --color-dimmed: #ffb6d6;\">evolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged annotation\" href=\"/training-material/search?query=annotation\" style=\"--color-primary: #76f300; --color-darker: #69d800; --color-dimmed: #6fe600;\">annotation</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged vgp\" href=\"/training-material/search?query=vgp\" style=\"--color-primary: #ffc6ef; --color-darker: #ffa0e7; --color-dimmed: #ffb4eb;\">vgp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cookbook\" href=\"/training-material/search?query=cookbook\" style=\"--color-primary: #00f3bd; --color-darker: #00d8a8; --color-dimmed: #00e6b2;\">cookbook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome annotation with Maker (short)",
    "contributors": "Anthony Bretaudeau",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/tnseq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Essential genes detection with Transposon insertion sequencing",
    "contributors": "Delphine Lariviere, B√©r√©nice Batut",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bacteria\" href=\"/training-material/search?query=bacteria\" style=\"--color-primary: #ffcf8c; --color-darker: #f6b400; --color-dimmed: #ffc14b;\">bacteria</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tnseq\" href=\"/training-material/search?query=tnseq\" style=\"--color-primary: #ffcbc5; --color-darker: #ffa99e; --color-dimmed: #ffbab2;\">tnseq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged essential genes\" href=\"/training-material/search?query=essential genes\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">essential genes</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome annotation with Maker",
    "contributors": "Anthony Bretaudeau",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/functional/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Functional annotation of protein sequences",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/functional/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/apollo-euk/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Refining Genome Annotations with Apollo (eukaryotes)",
    "contributors": "Anthony Bretaudeau, Erasmus+ Programme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffc6ef; --color-darker: #ffa0e7; --color-dimmed: #ffb4eb;\">cyoa</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/lncrna/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Long non-coding RNAs (lncRNAs) annotation with FEELnc",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #56f600; --color-darker: #4bdb00; --color-dimmed: #51e900;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/crispr-screen/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "CRISPR screen analysis",
    "contributors": "Maria Doyle, Kenji Fujihara, Twishi Gulati",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged genome editing\" href=\"/training-material/search?query=genome editing\" style=\"--color-primary: #83e5ff; --color-darker: #00d1f2; --color-dimmed: #2ddeff;\">genome editing</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged CRISPR\" href=\"/training-material/search?query=CRISPR\" style=\"--color-primary: #00eefa; --color-darker: #00d4df; --color-dimmed: #00e1ed;\">CRISPR</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged essential genes\" href=\"/training-material/search?query=essential genes\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">essential genes</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cell line\" href=\"/training-material/search?query=cell line\" style=\"--color-primary: #00f778; --color-darker: #00dc6a; --color-dimmed: #00ea70;\">cell line</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html"
  },
  "/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Object tracking using CellProfiler",
    "contributors": "Yi Sun, Beatriz Serrano-Solano, Jean-Karim H√©rich√©",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html"
  },
  "/topics/imaging/tutorials/tutorial-CP/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Nucleoli segmentation and feature extraction using CellProfiler",
    "contributors": "Beatriz Serrano-Solano, Jean-Karim H√©rich√©",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html"
  },
  "/topics/imaging/tutorials/hela-screen-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Analyse HeLa fluorescence siRNA screen",
    "contributors": "Thomas Wollmann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #cfd5ff; --color-darker: #b0bbff; --color-dimmed: #c0c8ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html"
  },
  "/topics/imaging/tutorials/imaging-introduction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Introduction to image analysis using Galaxy",
    "contributors": "Thomas Wollmann, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #cfd5ff; --color-darker: #b0bbff; --color-dimmed: #c0c8ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html"
  },
  "/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "End-to-End Tissue Microarray Image Analysis with Galaxy-ME",
    "contributors": "Cameron Watson, Allison Creason",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "From peaks to genes",
    "contributors": "Anne Pajon, Clemens Blank, B√©r√©nice Batut, Bj√∂rn Gr√ºning, Nicola Soranzo, Dilmurat Yusuf, Sarah Peter, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "A short introduction to Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?query=espa√±ol\" style=\"--color-primary: #00f0e4; --color-darker: #00d6cb; --color-dimmed: #00e3d8;\">espa√±ol</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Galaxy 101",
    "contributors": "Saskia Hiltemann, Nicola Soranzo, Clemens Blank, Anton Nekrutenko, Bj√∂rn Gr√ºning, Anne Pajon, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "NGS data logistics",
    "contributors": "Anton Nekrutenko, Marius van den Beek, Dave Clements, Daniel Blankenberg",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Introduction to Genomics and Galaxy",
    "contributors": "Dave Clements, Crist√≥bal Gallardo",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-reproduce/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "How to reproduce published Galaxy analyses",
    "contributors": "Melanie F√∂ll, Anne Fouilloux",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html"
  },
  "/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Data Manipulation Olympics",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffc6ef; --color-darker: #ffa0e7; --color-dimmed: #ffb4eb;\">cyoa</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Galaxy 101 for everyone",
    "contributors": "Anne Fouilloux, Nadia Gou√©, Christopher Barnett, Michele Maroni, Olha Nahorna, Dave Clements, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html"
  },
  "/topics/introduction/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Introduction to Galaxy",
    "contributors": "Andrea Bagnacani, B√©r√©nice Batut, Saskia Hiltemann, Anne Pajon, Nicola Soranzo, Helena Rasche, Christopher Barnett, Michele Maroni, Anne Fouilloux, Nadia Gou√©, Olha Nahorna, Dave Clements",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/introduction/slides.html"
  },
  "/topics/introduction/tutorials/options-for-using-galaxy/slides.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Options for using Galaxy",
    "contributors": "Dave Clements, Anthony Bretaudeau, Anne Pajon, Nicola Soranzo, Alaa Elghamry",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/options-for-using-galaxy/slides.html"
  },
  "/topics/introduction/tutorials/dummy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Upload data to Galaxy",
    "contributors": "B√©r√©nice Batut",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/dummy/tutorial.html"
  },
  "/topics/introduction/tutorials/igv-introduction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "IGV Introduction",
    "contributors": "Anton Nekrutenko, Joachim Wolff, Anne Pajon",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html"
  },
  "/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry imaging: Examining the spatial distribution of analytes",
    "contributors": "Melanie F√∂ll, Maren Stillger",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html"
  },
  "/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry imaging: Finding differential analytes",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS analysis",
    "contributors": "M√©lanie Petera, Gildas Le Corguill√©, Jean-Fran√ßois Martin, Yann Guitton, Workflow4Metabolomics core team",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/lcms/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS preprocessing with XCMS",
    "contributors": "M√©lanie Petera, Jean-Fran√ßois Martin, Gildas Le Corguill√©, Workflow4Metabolomics core team",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html"
  },
  "/topics/metabolomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Introduction to Metabolomics",
    "contributors": "Gildas Le Corguill√©, C√©cile Canlet",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/introduction/slides.html"
  },
  "/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS data processing",
    "contributors": "M√©lanie Petera, Workflow4Metabolomics core team",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html"
  },
  "/topics/metabolomics/tutorials/gcms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry : GC-MS analysis with metaMS package",
    "contributors": "Julien Saint-Vanne",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/metabolomics/tutorials/gcms/tutorial.html"
  },
  "/topics/metagenomics/tutorials/general-tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Analyses of metagenomics data - The global picture",
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/general-tutorial/tutorial.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Metatranscriptomics analysis using microbiome RNA-seq data (short)",
    "contributors": "Pratik Jagtap, Subina Mehta, Ray Sajulga, B√©r√©nice Batut, Emma Leith, Praveen Kumar, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?query=metatranscriptomics\" style=\"--color-primary: #eecaff; --color-darker: #e6a9ff; --color-dimmed: #eabaff;\">metatranscriptomics</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics-short/tutorial.html"
  },
  "/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "16S Microbial analysis with Nanopore data",
    "contributors": "Crist√≥bal Gallardo",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">nanopore</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html"
  },
  "/topics/metagenomics/tutorials/beer-data-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Identification of the micro-organisms in a beer using Nanopore sequencing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged beer\" href=\"/training-material/search?query=beer\" style=\"--color-primary: #00efed; --color-darker: #00d5d3; --color-dimmed: #00e2e0;\">beer</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged citizen science\" href=\"/training-material/search?query=citizen science\" style=\"--color-primary: #3bf800; --color-darker: #34dd00; --color-dimmed: #37ea00;\">citizen science</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #f3d700; --color-darker: #d8c000; --color-dimmed: #e6cb00;\">metagenomics</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/beer-data-analysis/tutorial.html"
  },
  "/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Assembly of metagenomic sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #f3d700; --color-darker: #d8c000; --color-dimmed: #e6cb00;\">metagenomics</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/metagenomics-assembly/tutorial.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "16S Microbial Analysis with mothur (extended)",
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut, Dave Clements",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/tutorial.html"
  },
  "/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Nanopore data analysis\" href=\"/training-material/search?query=Nanopore data analysis\" style=\"--color-primary: #00eff0; --color-darker: #00d5d6; --color-dimmed: #00e2e3;\">Nanopore data analysis</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Pathogens detection\" href=\"/training-material/search?query=Pathogens detection\" style=\"--color-primary: #abeb00; --color-darker: #97d100; --color-dimmed: #a1de00;\">Pathogens detection</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Phylogenetic tree\" href=\"/training-material/search?query=Phylogenetic tree\" style=\"--color-primary: #00f77d; --color-darker: #00dc6e; --color-dimmed: #00e976;\">Phylogenetic tree</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Heatmap\" href=\"/training-material/search?query=Heatmap\" style=\"--color-primary: #ffc7ea; --color-darker: #ffa1de; --color-dimmed: #ffb5e4;\">Heatmap</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffc6ef; --color-darker: #ffa0e7; --color-dimmed: #ffb4eb;\">cyoa</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Metatranscriptomics analysis using microbiome RNA-seq data",
    "contributors": "Pratik Jagtap, Subina Mehta, Ray Sajulga, B√©r√©nice Batut, Emma Leith, Praveen Kumar, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?query=metatranscriptomics\" style=\"--color-primary: #eecaff; --color-darker: #e6a9ff; --color-dimmed: #eabaff;\">metatranscriptomics</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics/tutorial.html"
  },
  "/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Antibiotic resistance detection",
    "contributors": "Willem de Koning, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plasmids\" href=\"/training-material/search?query=plasmids\" style=\"--color-primary: #00f4b9; --color-darker: #00d9a4; --color-dimmed: #00e6ae;\">plasmids</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffc7e5; --color-darker: #ffa3d7; --color-dimmed: #ffb6de;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?query=amr\" style=\"--color-primary: #c9d6ff; --color-darker: #a5beff; --color-dimmed: #b7caff;\">amr</a>"
    ],
    "url": "/training-material/topics/metagenomics/tutorials/plasmid-metagenomics-nanopore/tutorial.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "16S Microbial Analysis with mothur (short)",
    "contributors": "Saskia Hiltemann, B√©r√©nice Batut, Dave Clements",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/tutorial.html"
  },
  "/topics/metagenomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Metagenomics",
    "title": "Introduction to metagenomics",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/introduction/slides.html"
  },
  "/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Statistical analysis of DIA data",
    "contributors": "Matthias Fahrner, Melanie F√∂ll",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #7fe6ff; --color-darker: #00d2f1; --color-dimmed: #15deff;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "metaQuantome 1: Data creation",
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?query=microbiome\" style=\"--color-primary: #ffcbbf; --color-darker: #ffaa91; --color-dimmed: #ffbba9;\">microbiome</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html"
  },
  "/topics/proteomics/tutorials/ntails/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Detection and quantitation of N-termini (degradomics) via N-TAILS",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/ntails/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteome_annotation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Annotating a protein list identified by LC-MS/MS experiments",
    "contributors": "Valentin Loux, Florence Combes, David Christiany, Yves Vandenbrouck",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-id-oms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide and Protein ID using OpenMS tools",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #cfd5ff; --color-darker: #b0bbff; --color-dimmed: #c0c8ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Proteogenomics 1: Database Creation",
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?query=proteogenomics\" style=\"--color-primary: #00f68b; --color-darker: #00db7b; --color-dimmed: #00e983;\">proteogenomics</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-label-free/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Label-free data analysis using MaxQuant",
    "contributors": "Melanie F√∂ll, Matthias Fahrner",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-free\" href=\"/training-material/search?query=label-free\" style=\"--color-primary: #ced5ff; --color-darker: #afbcff; --color-dimmed: #bec8ff;\">label-free</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide and Protein ID using SearchGUI and PeptideShaker",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #cfd5ff; --color-darker: #b0bbff; --color-dimmed: #c0c8ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-quant-sil/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide and Protein Quantification via Stable Isotope Labelling (SIL)",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning, Matthias Fahrner",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SILAC\" href=\"/training-material/search?query=SILAC\" style=\"--color-primary: #d2e200; --color-darker: #bbc900; --color-dimmed: #c7d500;\">SILAC</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "metaQuantome 3: Taxonomy",
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane, Praveen Kumar",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?query=microbiome\" style=\"--color-primary: #ffcbbf; --color-darker: #ffaa91; --color-dimmed: #ffbba9;\">microbiome</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Biomarker candidate identification",
    "contributors": "Florence Combes, David Christiany, Valentin Loux, Yves Vandenbrouck",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Library Generation for DIA Analysis",
    "contributors": "Matthias Fahrner, Melanie F√∂ll",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #7fe6ff; --color-darker: #00d2f1; --color-dimmed: #15deff;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html"
  },
  "/topics/proteomics/tutorials/database-handling/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Protein FASTA Database Handling",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/database-handling/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "MaxQuant and MSstats for the analysis of label-free data",
    "contributors": "Melanie F√∂ll, Matthias Fahrner",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-free\" href=\"/training-material/search?query=label-free\" style=\"--color-primary: #ced5ff; --color-darker: #afbcff; --color-dimmed: #bec8ff;\">label-free</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "MaxQuant and MSstats for the analysis of TMT data",
    "contributors": "Klemens Fr√∂hlich, Melanie F√∂ll, Matthias Fahrner",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged TMT\" href=\"/training-material/search?query=TMT\" style=\"--color-primary: #e4dd00; --color-darker: #cac400; --color-dimmed: #d7d000;\">TMT</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "DIA Analysis using OpenSwathWorkflow",
    "contributors": "Matthias Fahrner, Melanie F√∂ll",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #7fe6ff; --color-darker: #00d2f1; --color-dimmed: #15deff;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-function/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "metaQuantome 2: Function",
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Emma Leith, Marie Crane, Praveen Kumar",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?query=microbiome\" style=\"--color-primary: #ffcbbf; --color-darker: #ffaa91; --color-dimmed: #ffbba9;\">microbiome</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html"
  },
  "/topics/proteomics/tutorials/secretome-prediction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Secretome Prediction",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html"
  },
  "/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Machine Learning Modeling of Anticancer Peptides",
    "contributors": "Jayadev Joshi, Daniel Blankenberg",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ML\" href=\"/training-material/search?query=ML\" style=\"--color-primary: #ded0ff; --color-darker: #cbb3ff; --color-dimmed: #d4c2ff;\">ML</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cancer\" href=\"/training-material/search?query=cancer\" style=\"--color-primary: #ffc8e4; --color-darker: #ffa3d5; --color-dimmed: #ffb6dd;\">cancer</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Proteogenomics 3: Novel peptide analysis",
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?query=proteogenomics\" style=\"--color-primary: #00f68b; --color-darker: #00db7b; --color-dimmed: #00e983;\">proteogenomics</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html"
  },
  "/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide Library Data Analysis",
    "contributors": "Jayadev Joshi, Daniel Blankenberg",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html"
  },
  "/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Mass spectrometry imaging: Loading and exploring MSI data",
    "contributors": "Melanie F√∂ll, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00f3c7; --color-darker: #00d8b1; --color-dimmed: #00e5bc;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged imaging\" href=\"/training-material/search?query=imaging\" style=\"--color-primary: #ffc7e8; --color-darker: #ffa2dc; --color-dimmed: #ffb5e2;\">imaging</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "EncyclopeDIA",
    "contributors": "Emma Leith, Subina Mehta, James Johnson, Pratik Jagtap, Timothy J. Griffin",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #7fe6ff; --color-darker: #00d2f1; --color-dimmed: #15deff;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Metaproteomics tutorial",
    "contributors": "Timothy J. Griffin, Pratik Jagtap, James Johnson, Clemens Blank, Subina Mehta",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microbiome\" href=\"/training-material/search?query=microbiome\" style=\"--color-primary: #ffcbbf; --color-darker: #ffaa91; --color-dimmed: #ffbba9;\">microbiome</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Proteogenomics 2: Database Search",
    "contributors": "Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?query=proteogenomics\" style=\"--color-primary: #00f68b; --color-darker: #00db7b; --color-dimmed: #00e983;\">proteogenomics</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html"
  },
  "/topics/proteomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Introduction to proteomics, protein identification, quantification and statistical modelling",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/introduction/slides.html"
  },
  "/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Label-free versus Labelled - How to Choose Your Quantitation Method",
    "contributors": "Florian Christoph Sigloch, Bj√∂rn Gr√ºning",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">DDA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Mapping",
    "contributors": "Joachim Wolff, B√©r√©nice Batut, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Quality Control",
    "contributors": "B√©r√©nice Batut, Maria Doyle, Alexandre Cormier, Anthony Bretaudeau, Laura Leroi, Erwan Corre, St√©phanie Robin, Erasmus+ Programme",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "NCBI BLAST+ against the MAdLand",
    "contributors": "Deepti Varshney",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Pre-processing of Single-Cell RNA Data",
    "contributors": "Mehmet Tekman, B√©r√©nice Batut, Anika Erxleben, Wolfgang Maier",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Single-cell quality control with scater",
    "contributors": "Graham Etherington, Nicola Soranzo",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Generating a single cell matrix using Alevin",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?query=espa√±ol\" style=\"--color-primary: #00f0e4; --color-darker: #00d6cb; --color-dimmed: #00e3d8;\">espa√±ol</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Bulk RNA Deconvolution with MuSiC",
    "contributors": "Mehmet Tekman, Wendi Bacon",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00f3c7; --color-darker: #00d8b1; --color-dimmed: #00e5bc;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deconvolution\" href=\"/training-material/search?query=deconvolution\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">deconvolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-intro/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "An introduction to scRNA-seq data analysis",
    "contributors": "Mehmet Tekman",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?query=espa√±ol\" style=\"--color-primary: #00f0e4; --color-darker: #00d6cb; --color-dimmed: #00e3d8;\">espa√±ol</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-intro/slides.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Combining datasets after pre-processing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?query=espa√±ol\" style=\"--color-primary: #00f0e4; --color-darker: #00d6cb; --color-dimmed: #00e3d8;\">espa√±ol</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Removing the effects of the cell cycle",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Creating the single-cell RNA-seq reference dataset for deconvolution",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deconvolution\" href=\"/training-material/search?query=deconvolution\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">deconvolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-raceid/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Downstream Single-cell RNA analysis with RaceID",
    "contributors": "Mehmet Tekman, Alex Ostrovsky",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-umis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Understanding Barcodes",
    "contributors": "Mehmet Tekman",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Creating the bulk RNA-seq dataset for deconvolution",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deconvolution\" href=\"/training-material/search?query=deconvolution\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">deconvolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Filter, Plot and Explore Single-cell RNA-seq Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged espa√±ol\" href=\"/training-material/search?query=espa√±ol\" style=\"--color-primary: #00f0e4; --color-darker: #00d6cb; --color-dimmed: #00e3d8;\">espa√±ol</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Pre-processing of 10X Single-Cell RNA Datasets",
    "contributors": "Mehmet Tekman, Hans-Rudolf Hotz, Daniel Blankenberg, Wendi Bacon",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Plates, Batches, and Barcodes",
    "contributors": "Wendi Bacon, Mehmet Tekman, Alex Ostrovsky",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html"
  },
  "/topics/single-cell/tutorials/bulk-music-4-compare/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Comparing inferred cell compositions using MuSiC deconvolution",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00f3c7; --color-darker: #00d8b1; --color-dimmed: #00e5bc;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffcbc7; --color-darker: #ffa9a3; --color-dimmed: #ffbab5;\">human</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deconvolution\" href=\"/training-material/search?query=deconvolution\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">deconvolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music-4-compare/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Clustering 3K PBMCs with Scanpy",
    "contributors": "B√©r√©nice Batut, Hans-Rudolf Hotz, Mehmet Tekman",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_JUPYTER-trajectories/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Inferring Trajectories using Scanpy (Python)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">jupyter-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_JUPYTER-trajectories/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-plant/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Analysis of plant scRNA-Seq Data with Scanpy",
    "contributors": "Mehmet Tekman, Beatriz Serrano-Solano, Crist√≥bal Gallardo, Pavankumar Videm",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ff400; --color-darker: #62d900; --color-dimmed: #69e600;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Trajectory Analysis using Monocle3",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffcad1; --color-darker: #ffa7b5; --color-dimmed: #ffb9c3;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #ecda00; --color-darker: #d2c200; --color-dimmed: #dfce00;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Trajectory Analysis: Monocle3 in R",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00f1de; --color-darker: #00d6c5; --color-dimmed: #00e3d1;\">single-cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged trajectory-analysis\" href=\"/training-material/search?query=trajectory-analysis\" style=\"--color-primary: #00f5ab; --color-darker: #00da98; --color-dimmed: #00e7a1;\">trajectory-analysis</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00efec; --color-darker: #00d5d2; --color-dimmed: #00e2df;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #23edff; --color-darker: #00d3e4; --color-dimmed: #00e0f2;\">jupyter-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dfd0ff; --color-darker: #ccb3ff; --color-dimmed: #d5c1ff;\">rmarkdown-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html"
  },
  "/topics/statistics/tutorials/classification_regression/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Machine learning: classification and regression",
    "contributors": "Anup Kumar, B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/classification_regression/tutorial.html"
  },
  "/topics/statistics/tutorials/clustering_machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Clustering in Machine Learning",
    "contributors": "Alireza Khanteymoori, Anup Kumar",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/clustering_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/fruit_360/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Image classification in Galaxy with fruit 360 dataset",
    "contributors": "Kaivan Kamali",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/fruit_360/tutorial.html"
  },
  "/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "A Docker-based interactive Jupyterlab powered by GPU for artificial intelligence in Galaxy",
    "contributors": "Anup Kumar",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged machine-learning\" href=\"/training-material/search?query=machine-learning\" style=\"--color-primary: #f7d600; --color-darker: #dcbe00; --color-dimmed: #e9ca00;\">machine-learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deep-learning\" href=\"/training-material/search?query=deep-learning\" style=\"--color-primary: #ffc7ec; --color-darker: #ffa1e2; --color-dimmed: #ffb4e7;\">deep-learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-lab\" href=\"/training-material/search?query=jupyter-lab\" style=\"--color-primary: #fff7bd; --color-darker: #eddd00; --color-dimmed: #fceb00;\">jupyter-lab</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged image-segmentation\" href=\"/training-material/search?query=image-segmentation\" style=\"--color-primary: #60f500; --color-darker: #54da00; --color-dimmed: #5ae800;\">image-segmentation</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged protein-3D-structure\" href=\"/training-material/search?query=protein-3D-structure\" style=\"--color-primary: #ffcea2; --color-darker: #ffaf3b; --color-dimmed: #ffbe7a;\">protein-3D-structure</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html"
  },
  "/topics/statistics/tutorials/text-mining_simtext/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Text-mining with the SimText toolset",
    "contributors": "Marie Gramm, Dennis Lal group, Daniel Blankenberg",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/text-mining_simtext/tutorial.html"
  },
  "/topics/statistics/tutorials/RNN/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 2) - Recurrent neural networks (RNN)",
    "contributors": "Kaivan Kamali",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/RNN/tutorial.html"
  },
  "/topics/statistics/tutorials/machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Basics of machine learning",
    "contributors": "Anup Kumar",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/iwtomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Interval-Wise Testing for omics data",
    "contributors": "Marzia A Cremona, Fabio Cumbo",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/iwtomics/tutorial.html"
  },
  "/topics/statistics/tutorials/intro_deep_learning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Introduction to deep learning",
    "contributors": "Anup Kumar, Alireza Khanteymoori",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html"
  },
  "/topics/statistics/tutorials/classification_machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Classification in Machine Learning",
    "contributors": "Alireza Khanteymoori, Anup Kumar, Simon Bray",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/FNN/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 1) - Feedforward neural networks (FNN)",
    "contributors": "Kaivan Kamali",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/FNN/tutorial.html"
  },
  "/topics/statistics/tutorials/regression_machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Regression in Machine Learning",
    "contributors": "Alireza Khanteymoori, Anup Kumar, Simon Bray",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Age prediction using machine learning",
    "contributors": "Ekaterina Polkh, Anup Kumar",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html"
  },
  "/topics/statistics/tutorials/CNN/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 3) - Convolutional neural networks (CNN)",
    "contributors": "Kaivan Kamali",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/CNN/tutorial.html"
  },
  "/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis",
    "contributors": "Vijay, Daniel Blankenberg",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Machine learning\" href=\"/training-material/search?query=Machine learning\" style=\"--color-primary: #d6d3ff; --color-darker: #beb7ff; --color-dimmed: #cac5ff;\">Machine learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Pan-cancer\" href=\"/training-material/search?query=Pan-cancer\" style=\"--color-primary: #00f0e6; --color-darker: #00d5cc; --color-dimmed: #00e3d9;\">Pan-cancer</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cancer biomarkers\" href=\"/training-material/search?query=cancer biomarkers\" style=\"--color-primary: #00f696; --color-darker: #00db85; --color-dimmed: #00e88e;\">cancer biomarkers</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged oncogenes and tumor suppressor genes\" href=\"/training-material/search?query=oncogenes and tumor suppressor genes\" style=\"--color-primary: #ffc9d4; --color-darker: #ffa6bb; --color-dimmed: #ffb8c7;\">oncogenes and tumor suppressor genes</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html"
  },
  "/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Introduction to Machine Learning using R",
    "contributors": "Fotis E. Psomopoulos, Erasmus+ Programme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html"
  },
  "/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Synthetic Biology",
    "title": "Designing plasmids encoding predicted pathways by using the BASIC assembly method",
    "contributors": "Kenza Bazi-Kabbaj, Thomas Duigou, Joan H√©risson, Guillaume Gricourt, Ioana Popescu, Jean-Loup Faulon",
    "tags": [

    ],
    "url": "/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html"
  },
  "/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Synthetic Biology",
    "title": "Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools",
    "contributors": "Guillaume Gricourt, Thomas Duigou, Kenza Bazi-Kabbaj, Joan H√©risson, Ioana Popescu, Jean-Loup Faulon",
    "tags": [

    ],
    "url": "/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html"
  },
  "/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Synthetic Biology",
    "title": "Evaluating and ranking a set of pathways based on multiple metrics",
    "contributors": "Kenza Bazi-Kabbaj, Thomas Duigou, Joan H√©risson, Guillaume Gricourt, Ioana Popescu, Jean-Loup Faulon",
    "tags": [

    ],
    "url": "/training-material/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/full-de-novo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "De novo transcriptome assembly, annotation, and differential expression analysis",
    "contributors": "Anthony Bretaudeau, Gildas Le Corguill√©, Erwan Corre, Xi Liu",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/srna/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Differential abundance testing of small RNAs",
    "contributors": "Mallory Freeberg",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/srna/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "3: RNA-seq genes to pathways",
    "contributors": "Maria Doyle, Belinda Phipson",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00f3c7; --color-darker: #00d8b1; --color-dimmed: #00e5bc;\">mouse</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/goenrichment/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "GO Enrichment Analysis",
    "contributors": "IGC Bioinformatics Unit, Maria Doyle",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/goenrichment/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "2: RNA-seq counts to genes",
    "contributors": "Maria Doyle, Belinda Phipson, Jovana Maksimovic, Anna Trigos, Matt Ritchie, Harriet Dashnow, Shian Su, Charity Law",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged limma-voom\" href=\"/training-material/search?query=limma-voom\" style=\"--color-primary: #75e7ff; --color-darker: #00d2ee; --color-dimmed: #00dffd;\">limma-voom</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00f3c7; --color-darker: #00d8b1; --color-dimmed: #00e5bc;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffcaca; --color-darker: #ffa8a9; --color-dimmed: #ffbaba;\">QC</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with Volcano Plot",
    "contributors": "Maria Doyle",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "1: RNA-Seq reads to counts",
    "contributors": "Maria Doyle, Belinda Phipson, Harriet Dashnow",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00f3c7; --color-darker: #00d8b1; --color-dimmed: #00e5bc;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffcaca; --color-darker: #ffa8a9; --color-dimmed: #ffbaba;\">QC</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with heatmap2",
    "contributors": "Maria Doyle",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Whole transcriptome analysis of Arabidopsis thaliana",
    "contributors": "Crist√≥bal Gallardo, Pavankumar Videm, Beatriz Serrano-Solano",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged miRNA\" href=\"/training-material/search?query=miRNA\" style=\"--color-primary: #75e7ff; --color-darker: #00d2ee; --color-dimmed: #00dffd;\">miRNA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ff400; --color-darker: #62d900; --color-dimmed: #69e600;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged stress tolerance\" href=\"/training-material/search?query=stress tolerance\" style=\"--color-primary: #00f77d; --color-darker: #00dc6e; --color-dimmed: #00e976;\">stress tolerance</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/clipseq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "CLIP-Seq data analysis from pre-processing to motif detection",
    "contributors": "Florian Heyl, Daniel Maticzka, B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Small Non-coding RNA Clustering using BlockClust",
    "contributors": "Pavankumar Videm",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-interactome/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA-RNA interactome data analysis",
    "contributors": "Pavankumar Videm",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Network analysis with Heinz",
    "contributors": "Chao Zhang",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?query=metatranscriptomics\" style=\"--color-primary: #eecaff; --color-darker: #e6a9ff; --color-dimmed: #eabaff;\">metatranscriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged network analysis\" href=\"/training-material/search?query=network analysis\" style=\"--color-primary: #00f2d1; --color-darker: #00d7ba; --color-dimmed: #00e4c5;\">network analysis</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA-Seq analysis with AskOmics Interactive Tool",
    "contributors": "Xavier Garnier, Anthony Bretaudeau, Anne Siegel, Olivier Dameron, Mateo Boudet",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Introduction to Transcriptomics",
    "contributors": "B√©r√©nice Batut, Anika Erxleben, Markus Wolfien",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/introduction/slides.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with CummeRbund",
    "contributors": "Andrea Bagnacani",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Reference-based RNAseq data analysis (long)",
    "contributors": "Anton Nekrutenko",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with Volcano Plot in R",
    "contributors": "Maria Doyle",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA Seq Counts to Viz in R",
    "contributors": "B√©r√©nice Batut, Fotis E. Psomopoulos, Toby Hodges, Erasmus+ Programme",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffccbc; --color-darker: #ffab8a; --color-dimmed: #ffbca4;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Reference-based RNA-Seq data analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #7cf200; --color-darker: #6ed700; --color-dimmed: #75e500;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rna-seq\" href=\"/training-material/search?query=rna-seq\" style=\"--color-primary: #ffcdb2; --color-darker: #ffac74; --color-dimmed: #ffbd96;\">rna-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffcacf; --color-darker: #ffa7b1; --color-dimmed: #ffb9c0;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged drosophila\" href=\"/training-material/search?query=drosophila\" style=\"--color-primary: #b6dbff; --color-darker: #7dc6ff; --color-dimmed: #9cd1ff;\">drosophila</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffcaca; --color-darker: #ffa8a9; --color-dimmed: #ffbaba;\">QC</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffc6ef; --color-darker: #ffa0e7; --color-dimmed: #ffb4eb;\">cyoa</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "De novo transcriptome reconstruction with RNA-Seq",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Introduction to Variant analysis",
    "contributors": "B√©r√©nice Batut, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/introduction/slides.html"
  },
  "/topics/variant-analysis/tutorials/non-dip/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Calling variants in non-diploid systems",
    "contributors": "Anton Nekrutenko, Alex Ostrovsky",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/dunovo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Calling very rare variants",
    "contributors": "Anton Nekrutenko, Nick Stoler",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #bfd9ff; --color-darker: #91c3ff; --color-dimmed: #a9ceff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/dip/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Calling variants in diploid systems",
    "contributors": "Anton Nekrutenko, Nicola Soranzo",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/dip/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/somatic-variants/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Identification of somatic and germline variants from tumor and normal sample pairs",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/trio-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Trio Analysis using Synthetic Datasets from RD-Connect GPAP",
    "contributors": "Jasper Ouwerkerk, Wolfgang Maier, Helena Rasche, Saskia Hiltemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffc6ef; --color-darker: #ffa0e7; --color-dimmed: #ffb4eb;\">cyoa</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/microbial-variants/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Microbial Variant Calling",
    "contributors": "Anna Syme, Simon Gladman, Torsten Seemann",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/microbial-variants/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "M. tuberculosis Variant Analysis",
    "contributors": "Peter van Heusden, Simon Gladman, Thoba Lose",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d8d2ff; --color-darker: #c1b6ff; --color-dimmed: #cdc4ff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Somatic Variant Discovery from WES Data Using Control-FREEC",
    "contributors": "Khaled Jum'ah, Katarzyna Murat, Wolfgang Maier, David Salgado, Krzysztof Poterlowicz",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Mapping and molecular identification of phenotype-causing mutations",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/exome-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Exome sequencing data analysis for diagnosing a genetic disease",
    "contributors": "Wolfgang Maier, B√©r√©nice Batut, Torsten Houwaart, Anika Erxleben, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Avian influenza viral strain analysis from gene segment sequencing data",
    "contributors": "Wolfgang Maier",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc8e3; --color-darker: #ffa3d4; --color-dimmed: #ffb6dc;\">virology</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html"
  },
  "/topics/visualisation/tutorials/circos/tutorial.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Visualisation with Circos",
    "contributors": "Saskia Hiltemann, Helena Rasche, Crist√≥bal Gallardo",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/circos/tutorial.html"
  },
  "/topics/visualisation/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Visualisations in Galaxy",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/introduction/slides.html"
  },
  "/topics/visualisation/tutorials/jbrowse/tutorial.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Genomic Data Visualisation with JBrowse",
    "contributors": "Helena Rasche, Saskia Hiltemann, Erasmus+ Programme",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do I need to create collections to run MaxQuant analysis or can I use single sample inputs?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Including custom modifications into MaxQuant in Galaxy?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which isobaric labeled quantification methods does MaxQuant in Galaxy support?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html"
  },
  "/faqs/gtn/a11y.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does the GTN ensure accessibility?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/a11y.html"
  },
  "/faqs/galaxy/account_create.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I create an account on a public Galaxy instance?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone, Samantha Anjei, Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_create.html"
  },
  "/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/account_exists.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "It says I already have an account when registering for ecology.usegalaxy.eu",
    "contributors": "Coline Royaux, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/account_exists.html"
  },
  "/faqs/galaxy/account_multiple.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I create multiple Galaxy accounts?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_multiple.html"
  },
  "/faqs/galaxy/account_reduce_quota_usage.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I reduce quota usage while still retaining prior work (data, tools, methods)?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_reduce_quota_usage.html"
  },
  "/faqs/galaxy/account_update_preference.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to update account preferences?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_update_preference.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/adapter_trimming.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Automatically trim adapters (without providing custom sequences)",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/adapter_trimming.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I adapt this tutorial to my own data?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I adapt this tutorial to my own data?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html"
  },
  "/topics/admin/faqs/admin-testing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Operating system compatibility",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/admin-testing.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/faqs/advantages.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What advantages does a Chromatogram Library have over a DDA-generated library or predicted spectral library?",
    "contributors": "Emma Leith",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/faqs/advantages.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/faqs/alevin_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is Alevin is not working?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/faqs/alevin_version.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin/faqs/alevin_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is Alevin is not working?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin/faqs/alevin_version.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/alignment_tools.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to replace the existing alignment tools such as HISAT and Freebayes with other tools?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/alignment_tools.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/faqs/alternative_tools.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to use alternative tools to those proposed in the tutorial?",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/alternative_tools.html"
  },
  "/topics/single-cell/faqs/amplification.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is amplification more of an issue in scRNA-seq than RNA-seq?",
    "contributors": "Wendi Bacon, Mehmet Tekman, Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/amplification.html"
  },
  "/faqs/galaxy/analysis_add_custom_build.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding a custom database/build (dbkey)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_add_custom_build.html"
  },
  "/faqs/galaxy/analysis_cut.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Beware of Cuts",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_cut.html"
  },
  "/faqs/galaxy/analysis_differential_expression_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Extended Help for Differential Expression Analysis Tools",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_differential_expression_help.html"
  },
  "/faqs/galaxy/analysis_jobs_keep_running.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Will my jobs keep running?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_jobs_keep_running.html"
  },
  "/faqs/galaxy/analysis_jobs_not_running.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My jobs aren't running!",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_jobs_not_running.html"
  },
  "/faqs/galaxy/analysis_regular_expressions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Regular Expressions 101",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_regular_expressions.html"
  },
  "/faqs/galaxy/analysis_reporting_issues.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Reporting usage problems, security issues, and bugs",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_reporting_issues.html"
  },
  "/faqs/galaxy/analysis_results_may_vary.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Results may vary",
    "contributors": "Saskia Hiltemann, Anne Fouilloux",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_results_may_vary.html"
  },
  "/faqs/galaxy/analysis_troubleshooting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Troubleshooting errors",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_troubleshooting.html"
  },
  "/faqs/galaxy/analysis_troubleshooting_reporting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What information should I include when reporting a problem?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html"
  },
  "/topics/introduction/faqs/analyze_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I can‚Äôt find the ‚ÄúAnalyze Data‚Äù button",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/analyze_data.html"
  },
  "/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/anndata_not_working.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "AnnData Import/ AnnData Manipulate not working?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/anndata_not_working.html"
  },
  "/topics/admin/faqs/ansible-connection.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Variable connection",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible-connection.html"
  },
  "/topics/admin/faqs/ansible_define-once.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Define once, reference many times",
    "contributors": "Nate Coraor, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_define-once.html"
  },
  "/topics/admin/faqs/ansible_diff.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What if you forget `--diff`?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_diff.html"
  },
  "/topics/admin/faqs/ansible_error-no-hosts-matched.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Error: \"skipping: no hosts matched\"",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_error-no-hosts-matched.html"
  },
  "/topics/admin/faqs/ansible_local.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Running Ansible on your remote machine",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_local.html"
  },
  "/topics/admin/faqs/ansible_role-prefix.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the difference between the roles with `role:` prefix and without?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_role-prefix.html"
  },
  "/topics/admin/faqs/ansible_role-properties.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I know what I can do with a role? What variables are available?",
    "contributors": "Helena Rasche, Nicola Soranzo, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_role-properties.html"
  },
  "/topics/admin/faqs/ansible_see_inventory.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I see what variables are set for a host?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_see_inventory.html"
  },
  "/topics/admin/faqs/ansible_yaml.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is YAML sensitive to True/true/False/false",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_yaml.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/faqs/bandage_different.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why does my assembly graph in Bandage look different to the one pictured in the tutorial?",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/bandage_different.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/faqs/bioconda_files_exist.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The folder `recipes/belerophon/` and the file `meta.yaml` already exist in bioconda?",
    "contributors": "Alex Ostrovsky, Dave B., Matthias Bernt",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/faqs/bioconda_files_exist.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/bowtie_parameters.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "In bowtie 2 parameters, in place of 1000 for other experiments, should we mention the median fragment length observed in our library?",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/bowtie_parameters.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/breakdown.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the advantage of breaking down protein to peptides before mass spec?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/breakdown.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/faqs/build_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The Build tissue-specific expression dataset tool (step one) exits with an error code.",
    "contributors": "Florence Combes",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/build_error.html"
  },
  "/topics/admin/faqs/byobu-shortcuts.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Opening a split screen in byobu",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/byobu-shortcuts.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/chromosome_names.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we change the chromosome names in the Ensembl GTF to match the UCSC genome reference?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/chromosome_names.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/cloud.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do you have resources to help me get started working in the cloud?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/cloud.html"
  },
  "/faqs/gtn/collaborative_development.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does the GTN implement the \"Ten simple rules for collaborative lesson development\"",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/collaborative_development.html"
  },
  "/faqs/galaxy/collections_add_tag.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding a tag to a collection",
    "contributors": "Saskia Hiltemann, Helena Rasche, Delphine Lariviere",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_add_tag.html"
  },
  "/faqs/galaxy/collections_build_list.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a dataset collection",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_build_list.html"
  },
  "/faqs/galaxy/collections_build_list_paired.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a paired collection",
    "contributors": "Saskia Hiltemann, Helena Rasche, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_build_list_paired.html"
  },
  "/faqs/galaxy/collections_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming a collection",
    "contributors": "Delphine Lariviere, Martin ƒåech, Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_rename.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/collections_test.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I want to use a collection for outputs but it always passes the test even when the script fails. Why?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/collections_test.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/comparing_measurement.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does one compare metaproteomics measurements from two experimental conditions?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/comparing_measurement.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/contaminants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do we need a contaminant FASTA for MQ in galaxy?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/contaminants.html"
  },
  "/topics/dev/faqs/contributing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to Contribute to Galaxy",
    "contributors": "John Chilton, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/dev/faqs/contributing.html"
  },
  "/topics/contributing/faqs/contributing_a_notebook.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Contributing a Jupyter Notebook to the GTN",
    "contributors": "Helena Rasche, Nicola Soranzo",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/contributing_a_notebook.html"
  },
  "/faqs/gtn/contributors_appreciated.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Thanks!",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_appreciated.html"
  },
  "/faqs/gtn/contributors_getting_started.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I get started with contributing?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_getting_started.html"
  },
  "/faqs/gtn/contributors_git_advanced.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I contribute in \"advanced\" mode?",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_git_advanced.html"
  },
  "/faqs/gtn/contributors_github_interface.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I fix mistakes or expand an existing tutorial using the GitHub interface?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_github_interface.html"
  },
  "/faqs/gtn/contributors_giving_feedback.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I give feedback?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_giving_feedback.html"
  },
  "/faqs/gtn/contributors_helping_project.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What can I do to help the project?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_helping_project.html"
  },
  "/faqs/gtn/contributors_reporting_issues.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I report mistakes or errors?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_reporting_issues.html"
  },
  "/faqs/gtn/contributors_tours.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I test an Interactive Tour?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_tours.html"
  },
  "/faqs/gtn/contributors_without_git.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I create new content without dealing with git?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_without_git.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/convert_to_MGF.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does one convert RAW files to MGF peak lists within Galaxy?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/convert_to_MGF.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/converting_FASTA.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/converting_FASTA.html"
  },
  "/topics/admin/tutorials/ansible-galaxy/faqs/custom-welcome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Customising the welcome page",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/ansible-galaxy/faqs/custom-welcome.html"
  },
  "/faqs/galaxy/dataset_incomplete_download.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Incomplete Dataset Download",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataset_incomplete_download.html"
  },
  "/faqs/galaxy/dataset_tool_errors_metadata.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to find and correct tool errors related to Metadata?",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataset_tool_errors_metadata.html"
  },
  "/faqs/galaxy/dataset_upload_fastqsanger_via_urls.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload fastqsanger datasets via links",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataset_upload_fastqsanger_via_urls.html"
  },
  "/faqs/galaxy/datasets_BAM_dataset_identifiers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding BAM dataset identifiers",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html"
  },
  "/faqs/galaxy/datasets_UCSC_sourced_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Directly obtaining UCSC sourced *genome* identifiers",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html"
  },
  "/faqs/galaxy/datasets_add_tag.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding a tag",
    "contributors": "B√©r√©nice Batut, Wolfgang Maier, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_add_tag.html"
  },
  "/faqs/galaxy/datasets_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the datatype",
    "contributors": "B√©r√©nice Batut, Nicola Soranzo, Helena Rasche, Saskia Hiltemann, Barakat Ajadi, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_change_datatype.html"
  },
  "/faqs/galaxy-es/datasets_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Cambiar el tipo de datos",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/datasets_change_datatype.html"
  },
  "/faqs/galaxy-fr/datasets_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changer le type de donn√©es",
    "contributors": "Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-fr/datasets_change_datatype.html"
  },
  "/faqs/galaxy/datasets_change_dbkey.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing database/build (dbkey)",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_change_dbkey.html"
  },
  "/faqs/galaxy/datasets_chromosome_identifiers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Mismatched Chromosome identifiers and how to avoid them",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_chromosome_identifiers.html"
  },
  "/faqs/galaxy/datasets_convert_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Converting the file format",
    "contributors": "Saskia Hiltemann, Helena Rasche, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_convert_datatype.html"
  },
  "/faqs/galaxy/datasets_create_new_file.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a new file",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Wolfgang Maier, Helena Rasche, Simon Bray",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_create_new_file.html"
  },
  "/faqs/galaxy/datasets_data_privacy_features.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to set Data Privacy Features?",
    "contributors": "Jennifer Hillman-Jackson, Desmond Jasper",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_data_privacy_features.html"
  },
  "/faqs/galaxy/datasets_detect_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Detecting the datatype (file format)",
    "contributors": "B√©r√©nice Batut, Nicola Soranzo, Helena Rasche, Anne Fouilloux, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_detect_datatype.html"
  },
  "/faqs/galaxy/datasets_download_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading datasets",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_download_datasets.html"
  },
  "/faqs/galaxy/datasets_download_using_command_line.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading datasets using command line",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_download_using_command_line.html"
  },
  "/faqs/galaxy/datasets_finding_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding Datasets",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_finding_datasets.html"
  },
  "/faqs/galaxy/datasets_hidden.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to unhide \"hidden datasets\"?",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_hidden.html"
  },
  "/faqs/galaxy/datasets_icons.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Different dataset icons and their usage",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_icons.html"
  },
  "/faqs/galaxy/datasets_import_from_data_library.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing data from a data library",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Nicola Soranzo, Helena Rasche, Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_import_from_data_library.html"
  },
  "/faqs/galaxy/datasets_import_from_remote_files.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing data from remote files",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Nicola Soranzo, Helena Rasche, Wolfgang Maier, Soumya Jha",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_import_from_remote_files.html"
  },
  "/faqs/galaxy/datasets_import_via_link.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing via links",
    "contributors": "B√©r√©nice Batut, Helena Rasche, Mehmet Tekman, Gildas Le Corguill√©, Saskia Hiltemann, Wendi Bacon, Anton Nekrutenko, Maria Doyle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_import_via_link.html"
  },
  "/faqs/galaxy-es/datasets_import_via_link.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importaci√≥n por medio de enlaces",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/datasets_import_via_link.html"
  },
  "/faqs/galaxy-fr/datasets_importa_via_link.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importer via un lien",
    "contributors": "Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-fr/datasets_importa_via_link.html"
  },
  "/faqs/galaxy/datasets_job_status.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding job statuses",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_job_status.html"
  },
  "/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Moving datasets between Galaxy servers",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html"
  },
  "/faqs/galaxy/datasets_not_downloading_at_all.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Datasets not downloading at all",
    "contributors": "Nurzhamalyrys, Melkeberhan Berhanu Degefa, Jennifer Hillman-Jackson",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_not_downloading_at_all.html"
  },
  "/faqs/galaxy/datasets_purging_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Purging datasets",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_purging_datasets.html"
  },
  "/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Quotas for datasets and histories",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html"
  },
  "/faqs/galaxy/datasets_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming a dataset",
    "contributors": "B√©r√©nice Batut, Nicola Soranzo, Saskia Hiltemann, Wolfgang Maier, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_rename.html"
  },
  "/faqs/galaxy/datasets_upload.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload few files (1-10)",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_upload.html"
  },
  "/faqs/galaxy/datasets_upload_ftp.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload many files (>10) via FTP",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_upload_ftp.html"
  },
  "/faqs/galaxy/datasets_working_with_deleted_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with deleted datasets",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_deleted_datasets.html"
  },
  "/faqs/galaxy/datasets_working_with_fasta.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with very large fasta datasets",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_fasta.html"
  },
  "/faqs/galaxy/datasets_working_with_fastq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with very large fastq datasets",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_fastq.html"
  },
  "/faqs/galaxy/datasets_working_with_reference_annotation.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with GFF GFT GTF2 GFF3 reference annotation",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html"
  },
  "/faqs/galaxy/datatypes_Using_compressed_fastq_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using compressed fastq data as tool inputs",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_Using_compressed_fastq_data.html"
  },
  "/faqs/galaxy/datatypes_best_practices_fastq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Best practices for loading fastq data into Galaxy",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_best_practices_fastq.html"
  },
  "/faqs/galaxy/datatypes_fastq_and_fastqsanger.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment?",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html"
  },
  "/faqs/galaxy/datatypes_fastq_compressed.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compressed FASTQ files, (`*.gz`)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_compressed.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_compressed.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compressed FASTQ files, (`*.gz`)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_compressed.html"
  },
  "/faqs/galaxy/datatypes_fastq_format.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to format fastq data for tools that require .fastqsanger format?",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_format.html"
  },
  "/faqs/galaxy/datatypes_fastq_varieties.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "FASTQ files: `fastq` vs `fastqsanger` vs ..",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_varieties.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_varieties.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "FASTQ files: `fastq` vs `fastqsanger` vs ..",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_varieties.html"
  },
  "/faqs/galaxy/datatypes_tabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Identifying and formatting Tabular Datasets",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_tabular.html"
  },
  "/faqs/galaxy/datatypes_understanding_datatypes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding Datatypes",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_understanding_datatypes.html"
  },
  "/faqs/galaxy/dataupload_EBI-SRA.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I  upload data using EBI-SRA?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataupload_EBI-SRA.html"
  },
  "/faqs/galaxy/dataupload_NCBI_SRA.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Data retrieval with ‚ÄúNCBI SRA Tools‚Äù (fastq-dump)",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataupload_NCBI_SRA.html"
  },
  "/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "NCBI SRA sourced fastq data",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The tutorial uses the normalised count table for visualisation. What about using VST normalised counts or rlog normalised counts?",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/determine_taxonomic_composition.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What software tools are available to determine taxonomic composition from mass spectrometry data?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/determine_taxonomic_composition.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/faqs/different_padj_value.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Could I use a different p-adj value for filtering differentially expressed genes?",
    "contributors": "Crist√≥bal Gallardo",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/different_padj_value.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/faqs/different_results.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why?",
    "contributors": "Mallory Freeberg",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/faqs/different_results.html"
  },
  "/topics/admin/faqs/diffs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to read a Diff",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/diffs.html"
  },
  "/topics/single-cell/faqs/dimension_reduction.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/dimension_reduction.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/faqs/enrichment.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the principle of an enrichment analysis?",
    "contributors": "Florence Combes",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/enrichment.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/error_closure_args.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My Rscript tool generates a strange R error on STDOUT about an invalid operation on a closure called 'args' ?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/error_closure_args.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/error_compose.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "`docker-compose up` fails with error `/usr/bin/start.sh: line 133: /galaxy/.venv/bin/uwsgi: No such file or directory`",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/error_compose.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/example-queries.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I find example queries for use in the cloud and elsewhere?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/example-queries.html"
  },
  "/faqs/gtn/fair_training.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does the GTN ensure our training materials are FAIR?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/fair_training.html"
  },
  "/topics/sequence-analysis/faqs/fastq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "FASTQ format",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/faqs/fastq.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics/faqs/fastq_interlacer_vs_joiner.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we use FASTQ interlacer and not the FASTQ joiner?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/fastq_interlacer_vs_joiner.html"
  },
  "/faqs/galaxy/features_scratchbook.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using the Scratchbook to view multiple datasets",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/features_scratchbook.html"
  },
  "/faqs/galaxy-es/features_scratchbook.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Uso del cuaderno de apuntes para ver varios conjuntos de datos",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/features_scratchbook.html"
  },
  "/topics/introduction/faqs/fetched_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "UCSC - I fetched data from a remote website but now I‚Äôm logged out of Galaxy and my data is gone?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/fetched_data.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/findmarkers_empty.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My Scanpy FindMarkers step is giving me an empty table",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/findmarkers_empty.html"
  },
  "/topics/galaxy-interface/faqs/flatten-list-list-paired-to-list-paired.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Flatten a list of list of paired datasets into a list of paired datasets",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/faqs/flatten-list-list-paired-to-list-paired.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/formats.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What file/data formats are defined for I/O in Galaxy?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/formats.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/functional_analysis_tools.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/functional_analysis_tools.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/faqs/further_learning.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I read more about Quality Control of data?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/further_learning.html"
  },
  "/topics/admin/faqs/galaxy-how-many-mules.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How many mules?",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-how-many-mules.html"
  },
  "/topics/introduction/faqs/galaxy.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My Galaxy looks different than in the tutorial/video",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/galaxy.html"
  },
  "/faqs/gtn/galaxy_events.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are there any upcoming events focused on Galaxy Training?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_events.html"
  },
  "/faqs/gtn/galaxy_tested_with.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compatible Versions of Galaxy",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_tested_with.html"
  },
  "/faqs/gtn/galaxy_usage.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Ways to use Galaxy",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_usage.html"
  },
  "/faqs/gtn/galaxy_what_is_it.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is Galaxy?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_what_is_it.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/faqs/gene_ontology.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is Gene Ontology (GO)?",
    "contributors": "Florence Combes",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/gene_ontology.html"
  },
  "/topics/single-cell/faqs/gene_profile.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What exactly is a ‚ÄòGene profile‚Äô?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/gene_profile.html"
  },
  "/faqs/gtn/get_data_upload.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I load data?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/get_data_upload.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/faqs/get_planemo.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I get planemo?",
    "contributors": "Alex Ostrovsky, Dave B., Matthias Bernt",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/faqs/get_planemo.html"
  },
  "/topics/admin/faqs/git-attributes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using Git With Ansible Vaults",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/git-attributes.html"
  },
  "/topics/admin/faqs/git-commit.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Time to git commit",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/git-commit.html"
  },
  "/topics/admin/faqs/git-gat-path.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Galaxy Admin Training Path",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/git-gat-path.html"
  },
  "/faqs/gtn/gtn-what-is-it.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is this website?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn-what-is-it.html"
  },
  "/faqs/gtn/gtn_advertise.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I advertise the training materials on my posters?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_advertise.html"
  },
  "/faqs/gtn/gtn_audience.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What audiences are the tutorials for?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_audience.html"
  },
  "/faqs/gtn/gtn_citing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I cite the GTN?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_citing.html"
  },
  "/faqs/gtn/gtn_license.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How is the content licensed?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_license.html"
  },
  "/faqs/gtn/gtn_news_create_post.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a GTN News post",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_news_create_post.html"
  },
  "/faqs/gtn/gtn_stats.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "GTN Stats",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_stats.html"
  },
  "/faqs/gtn/gtn_sustainability.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sustainability of the training-material and metadata",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_sustainability.html"
  },
  "/faqs/gtn/gtn_tutorials.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What are the tutorials for?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_tutorials.html"
  },
  "/faqs/gtn/gtn_workflow_testing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding workflow tests with Planemo",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_workflow_testing.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/heatmap_colours.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the heatmap colours",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/heatmap_colours.html"
  },
  "/faqs/galaxy/histories_copy_dataset.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Copy a dataset between histories",
    "contributors": "Gildas Le Corguill√©, Saskia Hiltemann, Helena Rasche, B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_copy_dataset.html"
  },
  "/faqs/galaxy/histories_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a new history",
    "contributors": "B√©r√©nice Batut, Wolfgang Maier, Saskia Hiltemann, Helena Rasche, Wendi Bacon, Nicola Soranzo",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_create_new.html"
  },
  "/faqs/galaxy-es/histories_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Para la creaci√≥n de un historial nuevo",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/histories_create_new.html"
  },
  "/faqs/galaxy-fr/histories_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Cr√©er un nouvel history",
    "contributors": "B√©r√©nice Batut, Wolfgang Maier, Saskia Hiltemann, Helena Rasche, Wendi Bacon, Nicola Soranzo, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-fr/histories_create_new.html"
  },
  "/faqs/galaxy/histories_download_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading histories",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_download_histories.html"
  },
  "/faqs/galaxy/histories_find_purge.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Find all Histories and purge (aka permanently delete)",
    "contributors": "Jennifer Hillman-Jackson, Samantha Anjei, Anne Fouilloux",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_find_purge.html"
  },
  "/faqs/galaxy/histories_finding_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding Histories",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_finding_histories.html"
  },
  "/faqs/galaxy/histories_import.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing a history",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_import.html"
  },
  "/faqs/galaxy/histories_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming a history",
    "contributors": "B√©r√©nice Batut, Wolfgang Maier, Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_rename.html"
  },
  "/faqs/galaxy-es/histories_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Cambiando el nombre de un historial",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/histories_rename.html"
  },
  "/faqs/galaxy/histories_search.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Searching your history",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_search.html"
  },
  "/faqs/galaxy/histories_shared.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding and working with \"Histories shared with me\"",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_shared.html"
  },
  "/faqs/galaxy/histories_sharing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sharing your History",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_sharing.html"
  },
  "/faqs/galaxy-es/histories_sharing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compartiendo un historial",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/histories_sharing.html"
  },
  "/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Transfer entire histories from one Galaxy server to another",
    "contributors": "Jennifer Hillman-Jackson, Adaora Okwo",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html"
  },
  "/faqs/galaxy/histories_undelete.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Undeleting history",
    "contributors": "Valentine Gatwiri",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_undelete.html"
  },
  "/faqs/galaxy/histories_unsharing_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Unsharing unwanted histories",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_unsharing_histories.html"
  },
  "/topics/assembly/faqs/hybrid_assembly.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Most tools seem to have options for assembly using long and short reads, what are the pros and cons of the different tools?",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/hybrid_assembly.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which icons are available to use in my tutorial?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When is the \"infer experiment\" tool used in practice?",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/input_not_found.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The input for a tool is not listed in the dropdown",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/input_not_found.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/input_not_found.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The input for a tool is not listed in the dropdown",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/input_not_found.html"
  },
  "/topics/admin/faqs/install_tool.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Install tools via the Admin UI",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/install_tool.html"
  },
  "/faqs/gtn/instructors_best_practices.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What are the best practices for teaching with Galaxy?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_best_practices.html"
  },
  "/faqs/gtn/instructors_choosing_server.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What Galaxy instance should I use for my training?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_choosing_server.html"
  },
  "/faqs/gtn/instructors_getting_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I get help?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_getting_help.html"
  },
  "/faqs/gtn/instructors_getting_started.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where do I start?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_getting_started.html"
  },
  "/faqs/galaxy/interactive_tools_jupyter_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch JupyterLab",
    "contributors": "Anne Fouilloux, Saskia Hiltemann, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"
  },
  "/faqs/galaxy/interactive_tools_jupyter_terminal.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Open a Terminal in Jupyter",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"
  },
  "/faqs/galaxy/interactive_tools_open.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Open interactive tool",
    "contributors": "Marie Gramm",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_open.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_knit.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Knitting RMarkdown documents in RStudio",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch RStudio",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Helena Rasche, Yvan Le Bras, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"
  },
  "/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch RStudio",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Helena Rasche, Yvan Le Bras, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_launch.html"
  },
  "/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch RStudio",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Helena Rasche, Yvan Le Bras, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_launch.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Learning with RMarkdown in RStudio",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_stop.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Stop RStudio",
    "contributors": "Yvan Le Bras, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"
  },
  "/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_stop.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Stop RStudio",
    "contributors": "Yvan Le Bras, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_stop.html"
  },
  "/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_stop.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Stop RStudio",
    "contributors": "Yvan Le Bras, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_stop.html"
  },
  "/faqs/gtn/issues_galaxy.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What information should I include when reporting a problem?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/issues_galaxy.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/faqs/jbrowse_runtime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "JBrowse is taking a long time to complete?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/faqs/jbrowse_runtime.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/job_grey.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "First job I submitted remains grey or running for a long time - is it broken?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/job_grey.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/key-error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "TB Variant Report crashes (with an error about KeyError: 'protein')",
    "contributors": "Peter van Heusden, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/key-error.html"
  },
  "/faqs/gtn/learners_getting_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I get help?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_getting_help.html"
  },
  "/faqs/gtn/learners_getting_started.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where do I start?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_getting_started.html"
  },
  "/faqs/gtn/learners_running_tutorials.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I run the hands-on tutorials?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_running_tutorials.html"
  },
  "/faqs/gtn/learners_using_materials.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I use this material?",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_using_materials.html"
  },
  "/topics/contributing/faqs/learning-pathways.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Defining a Learning Pathway",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/learning-pathways.html"
  },
  "/topics/admin/faqs/library-permissions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Library Permission Issues",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/library-permissions.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/local_install.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do I need that big (~5GB!) complicated Docker thing - can I just install the ToolFactory into our local galaxy server from the toolshed?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/local_install.html"
  },
  "/topics/introduction/faqs/login.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My jobs are not running / I cannot see the history overview menu",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/login.html"
  },
  "/topics/admin/faqs/maintain_fail-users-jobs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Failing all jobs from a specific user",
    "contributors": "Helena Rasche, Bj√∂rn Gr√ºning",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/maintain_fail-users-jobs.html"
  },
  "/topics/admin/faqs/mapping_by_user.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Mapping Jobs to Specific Storage By User",
    "contributors": "Helena Rasche, Bj√∂rn Gr√ºning, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/mapping_by_user.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/match.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When can you use (or cannot use) Match between runs in MaxQuant?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/match.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/merge_database.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do you need to merge the databases? Because you can select multiple fasta files in MaxQuant.",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/merge_database.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/metadata-description.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I find the full listing and description of the columns in each metadata table?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/metadata-description.html"
  },
  "/topics/admin/faqs/missed-something.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Got lost along the way?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/missed-something.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/more_info.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I read more about this analysis?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop/faqs/more_info.html"
  },
  "/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/more_info.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I read more about this analysis?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/mothur-miseq-sop-short/faqs/more_info.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/mqpar.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "If you use a mqpar file, can you include modifications that are not in the Galaxy version? For instance, propionamide (Cys alkylation by acrylamide).",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/mqpar.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics/faqs/multiple_inputs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can we use the ASaiM-MT workflow on multiple input files at the same time?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/multiple_inputs.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/faqs/multiqc_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "MultiQC error for your FastQC reports?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/multiqc_error.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/mvp.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "In the MVP platform, is it possible to view the genomic location of all the peptides?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/mvp.html"
  },
  "/topics/dev/tutorials/debugging/faqs/no_tests_no_yarn.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I cannot run client tests because yarn is not installed.",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/debugging/faqs/no_tests_no_yarn.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/normalize_lfq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What does it mean to normalize the LFQ intensities?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/normalize_lfq.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/other-datatypes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When will aligned read objects be available for other data types?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/other-datatypes.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/outputs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Does MaxQuant give as output possibility the PSMs and PEPs?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/outputs.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/faqs/perbase_warning.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When I get a warning for base per sequence content, what should I do?",
    "contributors": "Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/perbase_warning.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/faqs/pilon.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Running more than one round of Pilon polishing",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/pilon.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/planemo_multiple.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Only one Planemo test runs at a time. Why doesn't the server allow more than one at once?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/planemo_multiple.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_fails.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "On Scanpy PlotEmbed, the tool is failing",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_fails.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_results_missing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "On the Scanpy PlotEmbed step, my object doesn‚Äôt have Il2ra or Cd8b1 or Cd8a etc.",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_results_missing.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/faqs/polish_longreads.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can we polish the assembly with long reads too?",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/polish_longreads.html"
  },
  "/faqs/galaxy/preferences_admin_api_key.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Getting your API key",
    "contributors": "Helena Rasche, Saskia Hiltemann, B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/preferences_admin_api_key.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Example histories for the proteogenomics tutorials",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_example_histories.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Example histories for the proteogenomics tutorials",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_example_histories.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Example histories for the proteogenomics tutorials",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_example_histories.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_own_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_own_data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_own_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_own_data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_own_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_own_data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_querytabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The workflows contain several Query tabular for text manipulation, is there a tutorial for that?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_querytabular.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_querytabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The workflows contain several Query tabular for text manipulation, is there a tutorial for that?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_querytabular.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_querytabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The workflows contain several Query tabular for text manipulation, is there a tutorial for that?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_querytabular.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_variants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What kind of variants are seen in the output?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_variants.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_variants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What kind of variants are seen in the output?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_variants.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_variants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What kind of variants are seen in the output?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_variants.html"
  },
  "/topics/single-cell/tutorials/scrna-umis/faqs/protocol.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I know what protocol my data was sequenced with?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-umis/faqs/protocol.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/quality-scores.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why don't the aligned read files have quality scores?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/quality-scores.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/faqs/quantification_alternatives.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use alternative tools for the Quantification step?",
    "contributors": "Crist√≥bal Gallardo",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/quantification_alternatives.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/quantitation_default.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "For the ‚Äúquantitation method‚Äù what is the default if I just leave it as ‚ÄúNone‚Äù? Label free?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/quantitation_default.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/faqs/query_no_results.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why does the query `SRR11772204 OR SRR11597145 OR SRR11667145` in the Run Selector not return any results?",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/faqs/query_no_results.html"
  },
  "/topics/single-cell/tutorials/scrna-umis/faqs/r1r2.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are Barcodes always on R1 and Sequence data on R2?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-umis/faqs/r1r2.html"
  },
  "/faqs/galaxy/reference_genomes_custom_genomes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to use Custom Reference Genomes?",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reference_genomes_custom_genomes.html"
  },
  "/faqs/galaxy/reference_genomes_sorting_reference_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sorting Reference Genome",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html"
  },
  "/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Troubleshooting Custom Genome fasta",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/replace_manual_install.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is the ToolFactory a complete replacement for manual tool building?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/replace_manual_install.html"
  },
  "/topics/proteomics/faqs/resources.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Additional resources to learn more about proteomic data analysis",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/faqs/resources.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/reusing_workflows.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/reusing_workflows.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to visualize the RNA STAR bam file using the JBrowse tool?",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "RNAstar: Why do we set 36 for 'Length of the genomic sequence around annotated junctions'?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/runtime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My snippy is running for a very long time. Is this normal?",
    "contributors": "Peter van Heusden",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/runtime.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-automation/faqs/sars-bot-usegalaxy-be.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What should I do special if on usegalaxy.be?",
    "contributors": "Bert Droesbeke",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-automation/faqs/sars-bot-usegalaxy-be.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/search_algorithms.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which search algorithms are recommended for searching the metaproteomics data?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_algorithms.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/search_gui.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How many search engines can you use in SearchGUI?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/search_gui.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/search_strategies.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_strategies.html"
  },
  "/topics/dev/tutorials/debugging/faqs/selenium_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When I try to run a Selenium test, I get an error",
    "contributors": "Assunta DeSanto",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/debugging/faqs/selenium_error.html"
  },
  "/faqs/galaxy/sequencing_illumina_miseq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Illumina MiSeq sequencing",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/sequencing_illumina_miseq.html"
  },
  "/faqs/galaxy/sequencing_nanopore.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Nanopore sequencing",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/sequencing_nanopore.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/shotgun.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How many proteins can be identified and quantified in shotgun proteomics?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/shotgun.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics/faqs/single_end.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can this ASaiM workflow be used for single-end data?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/single_end.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/faqs/snp.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is a SNP?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/faqs/snp.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/stats.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "MSStats: what does ‚Äòcompare groups = yes‚Äô mean? And the comparison matrix to define the contrast between the 2 groups?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/stats.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "In 'infer experiments' I get unequal numbers, but in the IGV it looks like it is unstranded. What does this mean?",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/studying_microbiome_state.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What other methods are available to study the functional state of the microbiome within Galaxy?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/studying_microbiome_state.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/subsampling.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to subsample some samples if you have more reads?",
    "contributors": "Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/subsampling.html"
  },
  "/faqs/galaxy/support.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where do I get more support?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/support.html"
  },
  "/faqs/galaxy/support_admins.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Contacting Galaxy Administrators",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/support_admins.html"
  },
  "/topics/metagenomics/faqs/taxon.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is Taxonomy?",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/faqs/taxon.html"
  },
  "/topics/single-cell/faqs/techniques.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can RNA-seq techniques be applied to scRNA-seq?",
    "contributors": "Wendi Bacon, Mehmet Tekman, Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/techniques.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/terms-of-usage.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are these data free to use and download?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/terms-of-usage.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/tmt_itraq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Does MaxQuant in Galaxy support TMT, iTRAQ, etc.?",
    "contributors": "Melanie F√∂ll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/tmt_itraq.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/tool_order.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do I have to run the tools in the order of the tutorial?",
    "contributors": "Peter van Heusden",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/tool_order.html"
  },
  "/faqs/galaxy/tools_change_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the tool version",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_change_version.html"
  },
  "/faqs/galaxy/tools_favorite.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Organizing the tool panel",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_favorite.html"
  },
  "/faqs/galaxy/tools_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where is the tool help?",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_help.html"
  },
  "/faqs/galaxy/tools_logs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Viewing tool logs (`stdout` and `stderr`)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_logs.html"
  },
  "/faqs/galaxy/tools_missing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "If a Tool is Missing",
    "contributors": "Saskia Hiltemann, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_missing.html"
  },
  "/faqs/galaxy/tools_rerun.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Re-running a tool",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_rerun.html"
  },
  "/faqs/galaxy-es/tools_rerun.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Para volver a ejecutar una herramienta",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/tools_rerun.html"
  },
  "/faqs/galaxy/tools_select_collection.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Selecting a dataset collection as input",
    "contributors": "B√©r√©nice Batut, Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_select_collection.html"
  },
  "/faqs/galaxy/tools_select_multiple_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Select multiple datasets",
    "contributors": "B√©r√©nice Batut, Wolfgang Maier, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_select_multiple_datasets.html"
  },
  "/faqs/galaxy/tools_similar.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Multipile similar tools available",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_similar.html"
  },
  "/faqs/galaxy/tools_sorting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sorting Tools",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_sorting.html"
  },
  "/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Tool doesn't recognize input datasets",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/tools_versions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which version of SearchGUI and PeptideShaker shall I use for this tutorial?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/tools_versions.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/faqs/transcript.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I‚Äôm using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/faqs/transcript.html"
  },
  "/faqs/galaxy/troubleshooting_cluster_failure.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding 'canceled by admin' or cluster failure error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_cluster_failure.html"
  },
  "/faqs/galaxy/troubleshooting_excess_memory.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding 'exceeds memory allocation' error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_excess_memory.html"
  },
  "/faqs/galaxy/troubleshooting_input_problem.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding input error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_input_problem.html"
  },
  "/faqs/galaxy/troubleshooting_value_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding ValueError error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_value_error.html"
  },
  "/faqs/galaxy/troubleshooting_walltime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding walltime error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_walltime.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I create a tutorial skeleton from a Galaxy workflow?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html"
  },
  "/faqs/galaxy/tutorial_mode.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using tutorial mode",
    "contributors": "Saskia Hiltemann, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tutorial_mode.html"
  },
  "/topics/introduction/faqs/ucsc.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I got slightly different numbers than were in the tutorial",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/ucsc.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/ucsc_import.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "UCSC import: what should my file look like?",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/ucsc_import.html"
  },
  "/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/umap_plot_errors.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The UMAP Plots errors out sometimes?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/umap_plot_errors.html"
  },
  "/topics/single-cell/faqs/umi.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are UMIs not actually unique?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/umi.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/faqs/upload_slow.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading the files from the NCBI server fails or takes too long.",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/upload_slow.html"
  },
  "/faqs/galaxy/user_preferences_account_usage_quota.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Does your account usage quota seem incorrect?",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/user_preferences_account_usage_quota.html"
  },
  "/faqs/galaxy/user_preferences_forgot_password.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Forgot Password",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/user_preferences_forgot_password.html"
  },
  "/faqs/galaxy/user_preferences_manage_information.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing acount email or password",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/user_preferences_manage_information.html"
  },
  "/topics/single-cell/faqs/variable_genes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we only consider highly variable genes?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/variable_genes.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/variant_mapping_file.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we have a variant mapping file when it is not being used in the workflow?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/variant_mapping_file.html"
  },
  "/faqs/galaxy/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/visualisations_igv.html"
  },
  "/topics/assembly/tutorials/unicycler-assembly/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/unicycler-assembly/faqs/visualisations_igv.html"
  },
  "/topics/assembly/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/visualisations_igv.html"
  },
  "/topics/variant-analysis/tutorials/dunovo/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/dunovo/faqs/visualisations_igv.html"
  },
  "/topics/variant-analysis/tutorials/non-dip/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/non-dip/faqs/visualisations_igv.html"
  },
  "/topics/variant-analysis/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/faqs/visualisations_igv.html"
  },
  "/topics/galaxy-interface/tutorials/processing-many-samples-at-once/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/faqs/visualisations_igv.html"
  },
  "/topics/galaxy-interface/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/faqs/visualisations_igv.html"
  },
  "/topics/genome-annotation/tutorials/tnseq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/tutorials/tnseq/faqs/visualisations_igv.html"
  },
  "/topics/genome-annotation/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/faqs/visualisations_igv.html"
  },
  "/topics/visualisation/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/faqs/visualisations_igv.html"
  },
  "/topics/introduction/tutorials/igv-introduction/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/igv-introduction/faqs/visualisations_igv.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-strands/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-strands/faqs/visualisations_igv.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101/faqs/visualisations_igv.html"
  },
  "/topics/introduction/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/visualisations_igv.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/faqs/visualisations_igv.html"
  },
  "/topics/sequence-analysis/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/tal1-binding-site-identification/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/rb-rnaseq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rb-rnaseq/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/clipseq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/clipseq/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/small_ncrna_clustering/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/faqs/visualisations_igv.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/visualisations_igv.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/visualisations_igv.html"
  },
  "/topics/proteomics/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/faqs/visualisations_igv.html"
  },
  "/topics/assembly/faqs/visualizations_igv_add_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add genome and annotations to IGV from Galaxy",
    "contributors": "Delphine Lariviere, B√©r√©nice Batut, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/visualizations_igv_add_genome.html"
  },
  "/topics/visualisation/faqs/visualizations_igv_add_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add genome and annotations to IGV from Galaxy",
    "contributors": "Delphine Lariviere, B√©r√©nice Batut, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/faqs/visualizations_igv_add_genome.html"
  },
  "/topics/assembly/faqs/visualizations_igv_view_mapping.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add Mapped reads track to IGV from Galaxy",
    "contributors": "Delphine Lariviere, B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/visualizations_igv_view_mapping.html"
  },
  "/topics/visualisation/faqs/visualizations_igv_view_mapping.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add Mapped reads track to IGV from Galaxy",
    "contributors": "Delphine Lariviere, B√©r√©nice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/faqs/visualizations_igv_view_mapping.html"
  },
  "/topics/metagenomics/tutorials/metatranscriptomics/faqs/vsearch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why don't we perform the V-Search dereplication step of ASaiM for metatrascriptomic data?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/metagenomics/tutorials/metatranscriptomics/faqs/vsearch.html"
  },
  "/faqs/galaxy/why_not_excel.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why not use Excel?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/why_not_excel.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/faqs/without_spectral_library.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can EncyclopeDIA be run on a DIA-MS dataset without a spectral library?",
    "contributors": "Emma Leith",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/faqs/without_spectral_library.html"
  },
  "/faqs/galaxy/workflows_annotate.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Annotate a workflow",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_annotate.html"
  },
  "/faqs/galaxy/workflows_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a new workflow",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_create_new.html"
  },
  "/faqs/galaxy/workflows_edit.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Opening the workflow editor",
    "contributors": "Peter van Heusden, Saskia Hiltemann, Anne Fouilloux, Nicola Soranzo, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_edit.html"
  },
  "/faqs/galaxy/workflows_extract_from_history.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Extracting a workflow from your history",
    "contributors": "Saskia Hiltemann, Helena Rasche, Nicola Soranzo",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_extract_from_history.html"
  },
  "/faqs/galaxy-es/workflows_extract_from_history.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Extraer un flujo de trabajo de tu historial",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/workflows_extract_from_history.html"
  },
  "/faqs/galaxy/workflows_hide_intermediate_steps.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Hiding intermediate steps",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"
  },
  "/faqs/galaxy/workflows_import.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing a workflow",
    "contributors": "Saskia Hiltemann, Maria Doyle, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import.html"
  },
  "/faqs/galaxy/workflows_import_from_workflowhub.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Import workflows from WorkflowHub",
    "contributors": "Crist√≥bal Gallardo, Linelle Abueg",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import_from_workflowhub.html"
  },
  "/faqs/galaxy/workflows_import_search.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing a workflow using the search",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import_search.html"
  },
  "/faqs/galaxy/workflows_parameter_set_runtime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Setting parameters at run-time",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"
  },
  "/faqs/galaxy/workflows_publish.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Make a workflow public",
    "contributors": "B√©r√©nice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_publish.html"
  },
  "/faqs/galaxy/workflows_rename_output.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming workflow outputs",
    "contributors": "Saskia Hiltemann, Wolfgang Maier, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_rename_output.html"
  },
  "/faqs/galaxy/workflows_report_view.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Viewing a workflow report",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_report_view.html"
  },
  "/faqs/galaxy/workflows_run.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Running a workflow",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_run.html"
  },
  "/faqs/galaxy/workflows_run_trs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing and Launching a workflow",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_run_trs.html"
  }
};

function search(idx, q, includeFaqs, includeTutorials){
	if(q.length > 2){
        var results_partial = idx.search(`*${q}*`),
            results_exact = idx.search(`${q}`),
            results_fuzzy = idx.search(`${q}~3`);

        thereMap  = Object.assign({}, ...results_partial.map((x) => ({[x.ref]: x.score})));

        results_exact.forEach(x => {
            if(thereMap[x.ref] !== undefined){
                if(thereMap[x.ref] < x.score + 4){
                    thereMap[x.ref] = x.score + 4
                }
            } else {
                    thereMap[x.ref] = x.score + 4
            }
        })
        results_fuzzy.forEach(x => {
            if(thereMap[x.ref] !== undefined){
                if(thereMap[x.ref] < x.score - 2){
                    thereMap[x.ref] = x.score - 2
                }
            } else {
                    thereMap[x.ref] = x.score - 2
            }
        })

        combined_results = Object.getOwnPropertyNames(thereMap);
        combined_results.sort((a, b) => {
            if (thereMap[a] > thereMap[b]) {
                return -1;
            }
            if (thereMap[a] < thereMap[b]) {
                return 1;
            }
            return 0;
        });

		var results_final = combined_results.map(x => {
			return resources['/' + x.replaceAll(".md", ".html")];
		}).filter(x => x !== undefined);

		if(! includeFaqs) {
			results_final = results_final.filter(x => x.type != 'FAQ')
		}
		if(! includeTutorials) {
			results_final = results_final.filter(x => x.type != 'Tutorial')
		}

        $("#results-container").html(results_final.map(x => `
        <div class='col-sm-6'>
        <div class='card'>
        <div class='card-body'>
          <h5 class='card-title'>${x.title}</h5>
          <h6 class='card-subtitle text-muted'>${x.topic}</h6>
          <p>${x.tags.join(' ')}</p>
          <p>${x.contributors}</p>
          <a class='btn btn-primary' href='${x.url}'>View ${x.type}</a>
          </div>
          </div>
          </div>
                    `));
	}
}

function searchWrap(idx) {
	console.log(
	'search',
		$("#search-input").val(),
		$("input[name='search-faqs']").is(':checked'),
		$("input[name='search-tutorials']").is(':checked')
	)
	search(idx,
		$("#search-input").val(),
		$("input[name='search-faqs']").is(':checked'),
		$("input[name='search-tutorials']").is(':checked')
	);
}

fetch('/training-material/search.json')
	.then(response => response.json())
	.then(data => {
		var idx = lunr.Index.load(data);

		var  params = (new URL(document.location)).searchParams;
		paramQuery = params.get('query');
		if(paramQuery){
			document.getElementById('search-input').value = paramQuery;
			searchWrap(idx);
		}

		$("#search-input").on("change keyup paste", function(){
			searchWrap(idx);
		})

		$("input[name='search-faqs']:checkbox").change(
			function(){
				searchWrap(idx);
			}
		);

		$("input[name='search-tutorials']:checkbox").change(
			function(){
				searchWrap(idx);
			}
		);

});
</script>



</section>

        </div>
        
            <footer>
    <div class="container">
        <br><br>
        <p>
            The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
            provides researchers with online training materials, connects them with local trainers, and helps promoting open data analysis practices worldwide.
        </p>
        <p>
        The content of the tutorials is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>. The website and infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>.
        </p>

        <p>This is revision <a href="https://github.com/galaxyproject/training-material/commit/cd5bf7caa80b8a113d6cb39efd0dfeeef514cb5b">cd5bf7caa80b8a113d6cb39efd0dfeeef514cb5b</a>. </p>
    </div>
</footer>

        
        <script src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script src="/training-material/assets/js/popper.min.js"></script>
        <script src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script src="/training-material/assets/js/main.js"></script>
        <script src="/training-material/assets/js/clipboard.min.js"></script>
        <script>
        var snippets=document.querySelectorAll('div.highlight');
        [].forEach.call(snippets,function(snippet){
            snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet tabindex="0"><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });

        var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
            target:function(trigger){return trigger.nextElementSibling;
        }});
        </script>
        

        <script type="text/javascript">
            if(window.location.hostname === "galaxyproject.github.io") {
                // Redirect
                var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
                $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

                window.setTimeout(function(){
                    window.location.href = redirect;
                }, 5000)

            }
        </script>
    </body>
</html>
