<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>GTN Tutorial Search</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@8cb9a3a5113d0b6a5b4374ea1865adbe76b9da68",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/search.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="GTN Tutorial Search">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-12-14 16:07:10 +0000"><meta name="description" content="    {% icon search %}     Search FAQs     Search Tutorials         ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="GTN Tutorial Search">
        <meta property="og:description" content="    {% icon search %}     Search FAQs     Search Tutorials         ">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/search.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/search.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section>
  
  <h1>GTN Tutorial Search</h1>
  

  <script src="https://cdnjs.cloudflare.com/ajax/libs/lunr.js/2.3.9/lunr.min.js" integrity="sha512-4xUl/d6D6THrAnXAwGajXkoWaeMNwEKK4iNfq5DotEbLPAfk6FSxSP3ydNxqDgCw1c/0Z1Jg6L8h2j+++9BZmg==" crossorigin="anonymous"></script>

<div id="search-container">
	<input type="text" id="search-input" placeholder=" search..." class="nicer" /><i class="fas fa fa-search" aria-hidden="true"></i><span class="visually-hidden">search</span>

	<input type="checkbox" id="search-faqs" name="search-faqs" checked="" />
	<label for="search-faqs">Search FAQs</label>

	<input type="checkbox" id="search-tutorials" name="search-tutorials" checked="" />
	<label for="search-tutorials">Search Tutorials</label>

	<div class="search-results row" id="results-container"></div>
</div>

<!-- Configuration -->
<script>

var resources = {
  "/topics/admin/tutorials/celery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Setting up Celery Workers for Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/celery/tutorial.html"
  },
  "/topics/admin/tutorials/cvmfs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Reference Data with CVMFS",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/cvmfs/tutorial.html"
  },
  "/topics/admin/tutorials/database/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galactic Database",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/database/slides.html"
  },
  "/topics/admin/tutorials/gxadmin/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with gxadmin",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?query=monitoring\" style=\"--color-primary: #ffc4b0; --color-darker: #ffa47b; --color-dimmed: #ffb597;\">monitoring</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/gxadmin/tutorial.html"
  },
  "/topics/admin/tutorials/ansible/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Ansible",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/ansible/tutorial.html"
  },
  "/topics/admin/tutorials/advanced-galaxy-customisation/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Advanced customisation of a Galaxy instance",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/advanced-galaxy-customisation/slides.html"
  },
  "/topics/admin/tutorials/backup-cleanup/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Server Maintenance: Cleanup, Backup, and Restoration",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffc4b9; --color-darker: #ffa28f; --color-dimmed: #ffb3a5;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/backup-cleanup/tutorial.html"
  },
  "/topics/admin/tutorials/pulsar/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Running Jobs on Remote Resources with Pulsar",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/pulsar/tutorial.html"
  },
  "/topics/admin/tutorials/ansible-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Installation with Ansible",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffc4b9; --color-darker: #ffa28f; --color-dimmed: #ffb3a5;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/ansible-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/systemd-supervisor/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Controlling Galaxy with systemd or Supervisor",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/systemd-supervisor/slides.html"
  },
  "/topics/admin/tutorials/users-groups-quotas/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "User, Role, Group, Quota, and Authentication managment",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/users-groups-quotas/slides.html"
  },
  "/topics/admin/tutorials/object-store/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Distributed Object Storage",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged storage\" href=\"/training-material/search?query=storage\" style=\"--color-primary: #ffbeea; --color-darker: #ff98e0; --color-dimmed: #fface5;\">storage</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/object-store/tutorial.html"
  },
  "/topics/admin/tutorials/general-monitoring/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/general-monitoring/slides.html"
  },
  "/topics/admin/tutorials/external-auth/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "External Authentication",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged authentication\" href=\"/training-material/search?query=authentication\" style=\"--color-primary: #e9c4ff; --color-darker: #dfa3ff; --color-dimmed: #e3b4ff;\">authentication</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/external-auth/tutorial.html"
  },
  "/topics/admin/tutorials/data-library/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Data Libraries",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged storage\" href=\"/training-material/search?query=storage\" style=\"--color-primary: #ffbeea; --color-darker: #ff98e0; --color-dimmed: #fface5;\">storage</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/data-library/tutorial.html"
  },
  "/topics/admin/tutorials/poll-ssa/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Administrator Time Burden and Technology Usage",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffc4b9; --color-darker: #ffa28f; --color-dimmed: #ffb3a5;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged maintenance\" href=\"/training-material/search?query=maintenance\" style=\"--color-primary: #d1ccff; --color-darker: #b9b2ff; --color-dimmed: #c5bfff;\">maintenance</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged survey\" href=\"/training-material/search?query=survey\" style=\"--color-primary: #caceff; --color-darker: #aeb5ff; --color-dimmed: #bcc2ff;\">survey</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/poll-ssa/slides.html"
  },
  "/topics/admin/tutorials/monitoring/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with Telegraf and Grafana",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?query=monitoring\" style=\"--color-primary: #ffc4b0; --color-darker: #ffa47b; --color-dimmed: #ffb597;\">monitoring</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/monitoring/tutorial.html"
  },
  "/topics/admin/tutorials/production/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Gearing towards production",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/production/slides.html"
  },
  "/topics/admin/tutorials/troubleshooting/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Troubleshooting",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/troubleshooting/slides.html"
  },
  "/topics/admin/tutorials/interactive-tools/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Interactive Tools",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/interactive-tools/tutorial.html"
  },
  "/topics/admin/tutorials/tool-management/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Tool Management with Ephemeris",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tools\" href=\"/training-material/search?query=tools\" style=\"--color-primary: #ffc0d9; --color-darker: #ff9cc7; --color-dimmed: #ffafd0;\">tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/tool-management/tutorial.html"
  },
  "/topics/admin/tutorials/cloudbursting/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy on the Cloud",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/cloudbursting/slides.html"
  },
  "/topics/admin/tutorials/galaxy-docker/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Docker and Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/galaxy-docker/slides.html"
  },
  "/topics/admin/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy from an administrator's point of view",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/introduction/slides.html"
  },
  "/topics/admin/tutorials/stuff/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Server: Other",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/stuff/slides.html"
  },
  "/topics/admin/tutorials/uwsgi/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "uWSGI",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/uwsgi/slides.html"
  },
  "/topics/admin/tutorials/reference-genomes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Reference Data with Data Managers",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/reference-genomes/tutorial.html"
  },
  "/topics/admin/tutorials/empathy/slides.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Empathy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/empathy/slides.html"
  },
  "/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Connecting Galaxy to a compute cluster",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/connect-to-compute-cluster/tutorial.html"
  },
  "/topics/admin/tutorials/tus/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Performant Uploads with TUS",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/tus/tutorial.html"
  },
  "/topics/admin/tutorials/k8s-managing-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Managing Galaxy on Kubernetes",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged kubernetes\" href=\"/training-material/search?query=kubernetes\" style=\"--color-primary: #ffc1d2; --color-darker: #ff9eba; --color-dimmed: #ffb0c6;\">kubernetes</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/k8s-managing-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/sentry/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Monitoring Galaxy and Pulsar with Sentry",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/sentry/tutorial.html"
  },
  "/topics/admin/tutorials/celeryless/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Alternative Celery Deployment for Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/celeryless/tutorial.html"
  },
  "/topics/admin/tutorials/job-destinations/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Mapping Jobs to Destinations using TPV",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/job-destinations/tutorial.html"
  },
  "/topics/admin/tutorials/upgrading/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Upgrading Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/upgrading/tutorial.html"
  },
  "/topics/admin/tutorials/tiaas/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Training Infrastructure as a Service (TIaaS)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged training\" href=\"/training-material/search?query=training\" style=\"--color-primary: #ffbdf5; --color-darker: #ff95f0; --color-dimmed: #ffaaf3;\">training</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/tiaas/tutorial.html"
  },
  "/topics/admin/tutorials/ftp/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Enable upload via FTP",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data\" href=\"/training-material/search?query=data\" style=\"--color-primary: #00f24e; --color-darker: #00d745; --color-dimmed: #00e549;\">data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/ftp/tutorial.html"
  },
  "/topics/admin/tutorials/subdomain/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Create a subdomain for your community on UseGalaxy.eu",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged subdomain\" href=\"/training-material/search?query=subdomain\" style=\"--color-primary: #ffc78e; --color-darker: #fbaa00; --color-dimmed: #ffb85e;\">subdomain</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged community\" href=\"/training-material/search?query=community\" style=\"--color-primary: #ffc1d7; --color-darker: #ff9dc3; --color-dimmed: #ffafcd;\">community</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/subdomain/tutorial.html"
  },
  "/topics/admin/tutorials/stop-worrying-love-systemd/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "How I learned to stop worrying and love the systemd",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged systemd\" href=\"/training-material/search?query=systemd\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">systemd</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/stop-worrying-love-systemd/tutorial.html"
  },
  "/topics/admin/tutorials/beacon/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying a Beacon v1 in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ga4gh\" href=\"/training-material/search?query=ga4gh\" style=\"--color-primary: #ffc1d5; --color-darker: #ff9dc0; --color-dimmed: #ffafca;\">ga4gh</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged beacon\" href=\"/training-material/search?query=beacon\" style=\"--color-primary: #ffc1d3; --color-darker: #ff9ebd; --color-dimmed: #ffb0c8;\">beacon</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/beacon/tutorial.html"
  },
  "/topics/admin/tutorials/wireguard-headscale/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying Tailscale/Headscale for private mesh networking",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged wireguard\" href=\"/training-material/search?query=wireguard\" style=\"--color-primary: #9dd9ff; --color-darker: #46c7ff; --color-dimmed: #79d0ff;\">wireguard</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged networking\" href=\"/training-material/search?query=networking\" style=\"--color-primary: #ffbdee; --color-darker: #ff97e7; --color-dimmed: #ffabea;\">networking</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/wireguard-headscale/tutorial.html"
  },
  "/topics/admin/tutorials/cvmfs-manual/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Reference Data with CVMFS without Ansible",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/cvmfs-manual/tutorial.html"
  },
  "/topics/admin/tutorials/wireguard/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying Wireguard for private mesh networking",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged wireguard\" href=\"/training-material/search?query=wireguard\" style=\"--color-primary: #9dd9ff; --color-darker: #46c7ff; --color-dimmed: #79d0ff;\">wireguard</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged networking\" href=\"/training-material/search?query=networking\" style=\"--color-primary: #ffbdee; --color-darker: #ff97e7; --color-dimmed: #ffabea;\">networking</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/wireguard/tutorial.html"
  },
  "/topics/admin/tutorials/jenkins/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Automation with Jenkins",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged automation\" href=\"/training-material/search?query=automation\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">automation</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/jenkins/tutorial.html"
  },
  "/topics/admin/tutorials/customization-manual/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Customizing the look of Galaxy (Manual)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/customization-manual/tutorial.html"
  },
  "/topics/admin/tutorials/apptainer/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Use Apptainer containers for running Galaxy jobs",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jobs\" href=\"/training-material/search?query=jobs\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">jobs</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/apptainer/tutorial.html"
  },
  "/topics/admin/tutorials/reports/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Monitoring with Reports",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ansible\" href=\"/training-material/search?query=ansible\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">ansible</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged monitoring\" href=\"/training-material/search?query=monitoring\" style=\"--color-primary: #ffc4b0; --color-darker: #ffa47b; --color-dimmed: #ffb597;\">monitoring</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged broken\" href=\"/training-material/search?query=broken\" style=\"--color-primary: #ffc791; --color-darker: #fda900; --color-dimmed: #ffb863;\">broken</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/reports/tutorial.html"
  },
  "/topics/admin/tutorials/customization/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Customizing the look of Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged git-gat\" href=\"/training-material/search?query=git-gat\" style=\"--color-primary: #ffc3c5; --color-darker: #ffa0a4; --color-dimmed: #ffb2b5;\">git-gat</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/customization/tutorial.html"
  },
  "/topics/admin/tutorials/database-schema/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Database schema",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/database-schema/tutorial.html"
  },
  "/topics/admin/tutorials/k8s-deploying-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Galaxy Installation on Kubernetes",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged kubernetes\" href=\"/training-material/search?query=kubernetes\" style=\"--color-primary: #ffc1d2; --color-darker: #ff9eba; --color-dimmed: #ffb0c6;\">kubernetes</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/k8s-deploying-galaxy/tutorial.html"
  },
  "/topics/admin/tutorials/terraform/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Server administration",
    "title": "Deploying a compute cluster in OpenStack via Terraform",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged terraform\" href=\"/training-material/search?query=terraform\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">terraform</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deploying\" href=\"/training-material/search?query=deploying\" style=\"--color-primary: #ffc4b9; --color-darker: #ffa28f; --color-dimmed: #ffb3a5;\">deploying</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cloud\" href=\"/training-material/search?query=cloud\" style=\"--color-primary: #ffc3c4; --color-darker: #ffa1a3; --color-dimmed: #ffb2b4;\">cloud</a>"
    ],
    "url": "/training-material/topics/admin/tutorials/terraform/tutorial.html"
  },
  "/topics/ai4life/tutorials/contribute-materials/slides.html": {
    "type": "Tutorial",
    "topic": "Image analysis using Deep Learning",
    "title": "Contribute to the BioImage.IO models",
    "contributors": "Caterina Fuster-Barceló",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimageio\" href=\"/training-material/search?query=bioimageio\" style=\"--color-primary: #00eeae; --color-darker: #00d49a; --color-dimmed: #00e1a4;\">bioimageio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged image analysis\" href=\"/training-material/search?query=image analysis\" style=\"--color-primary: #00f08f; --color-darker: #00d57f; --color-dimmed: #00e287;\">image analysis</a>"
    ],
    "url": "/training-material/topics/ai4life/tutorials/contribute-materials/slides.html"
  },
  "/topics/ai4life/tutorials/getting-started/slides.html": {
    "type": "Tutorial",
    "topic": "Image analysis using Deep Learning",
    "title": "BioImage Model Zoo: Advanced models in one-click",
    "contributors": "Caterina Fuster-Barceló",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimageio\" href=\"/training-material/search?query=bioimageio\" style=\"--color-primary: #00eeae; --color-darker: #00d49a; --color-dimmed: #00e1a4;\">bioimageio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged image analysis\" href=\"/training-material/search?query=image analysis\" style=\"--color-primary: #00f08f; --color-darker: #00d57f; --color-dimmed: #00e287;\">image analysis</a>"
    ],
    "url": "/training-material/topics/ai4life/tutorials/getting-started/slides.html"
  },
  "/topics/ai4life/tutorials/welcome-bioimageio/slides.html": {
    "type": "Tutorial",
    "topic": "Image analysis using Deep Learning",
    "title": "Welcome to the BioImage Model Zoo",
    "contributors": "Caterina Fuster-Barceló",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimageio\" href=\"/training-material/search?query=bioimageio\" style=\"--color-primary: #00eeae; --color-darker: #00d49a; --color-dimmed: #00e1a4;\">bioimageio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged image analysis\" href=\"/training-material/search?query=image analysis\" style=\"--color-primary: #00f08f; --color-darker: #00d57f; --color-dimmed: #00e287;\">image analysis</a>"
    ],
    "url": "/training-material/topics/ai4life/tutorials/welcome-bioimageio/slides.html"
  },
  "/topics/assembly/tutorials/general-introduction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "An Introduction to Genome Assembly",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/general-introduction/tutorial.html"
  },
  "/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html": {
    "type": "Tutorial",
    "topic": "Plants",
    "title": "ERGA post-assembly QC",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged animals\" href=\"/training-material/search?query=animals\" style=\"--color-primary: #00eadd; --color-darker: #00d0c5; --color-dimmed: #00ddd1;\">animals</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged genome\" href=\"/training-material/search?query=genome\" style=\"--color-primary: #b8d3ff; --color-darker: #8cbdff; --color-dimmed: #a3c8ff;\">genome</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffc3c3; --color-darker: #ffa1a1; --color-dimmed: #ffb2b2;\">QC</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/ERGA-post-assembly-QC/tutorial.html"
  },
  "/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Vertebrate genome assembly using HiFi, Bionano and Hi-C data - Step by Step",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?query=pacbio\" style=\"--color-primary: #ffc0df; --color-darker: #ff9bd0; --color-dimmed: #ffaed7;\">pacbio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged VGP\" href=\"/training-material/search?query=VGP\" style=\"--color-primary: #ffc696; --color-darker: #ffa822; --color-dimmed: #ffb76d;\">VGP</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/assembly-quality-control/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome Assembly Quality Control",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged quality control\" href=\"/training-material/search?query=quality control\" style=\"--color-primary: #c8cfff; --color-darker: #aab6ff; --color-dimmed: #b9c3ff;\">quality control</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-quality-control/tutorial.html"
  },
  "/topics/assembly/tutorials/ecoli_comparison/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Making sense of a newly assembled genome",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/ecoli_comparison/tutorial.html"
  },
  "/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "De Bruijn Graph Assembly",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/assembly-decontamination/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Decontamination of a genome assembly",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-decontamination/tutorial.html"
  },
  "/topics/assembly/tutorials/mitochondrion-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Assembly of the mitochondrial genome from PacBio HiFi reads",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/mitochondrion-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/largegenome/tutorial.html": {
    "type": "Tutorial",
    "topic": "Plants",
    "title": "Large genome assembly and polishing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged polishing\" href=\"/training-material/search?query=polishing\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">polishing</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/largegenome/tutorial.html"
  },
  "/topics/assembly/tutorials/mrsa-nanopore/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome Assembly of MRSA from Oxford Nanopore MinION data (and optionally Illumina data)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?query=amr\" style=\"--color-primary: #c1d1ff; --color-darker: #9eb9ff; --color-dimmed: #afc5ff;\">amr</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/mrsa-nanopore/tutorial.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "SARS-CoV-2",
    "title": "Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html"
  },
  "/topics/assembly/tutorials/vgp_workflow_training/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Using the VGP workflows to assemble a vertebrate genome with HiFi and Hi-C data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?query=pacbio\" style=\"--color-primary: #ffc0df; --color-darker: #ff9bd0; --color-dimmed: #ffaed7;\">pacbio</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged VGP\" href=\"/training-material/search?query=VGP\" style=\"--color-primary: #ffc696; --color-darker: #ffa822; --color-dimmed: #ffb76d;\">VGP</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_workflow_training/tutorial.html"
  },
  "/topics/assembly/tutorials/unicycler-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Unicycler Assembly",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Plants",
    "title": "Chloroplast genome assembly",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/metagenomics-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Assembly of metagenomic sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/metagenomics-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/flye-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome assembly using PacBio data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pacbio\" href=\"/training-material/search?query=pacbio\" style=\"--color-primary: #ffc0df; --color-darker: #ff9bd0; --color-dimmed: #ffaed7;\">pacbio</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html"
  },
  "/topics/assembly/tutorials/mrsa-illumina/tutorial.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Genome Assembly of a bacterial genome (MRSA) sequenced using Illumina MiSeq Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?query=illumina\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/assembly/tutorials/mrsa-illumina/tutorial.html"
  },
  "/topics/assembly/tutorials/get-started-genome-assembly/slides.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "An introduction to get started in genome assembly and annotation",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/get-started-genome-assembly/slides.html"
  },
  "/topics/assembly/tutorials/algorithms-introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Assembly",
    "title": "Deeper look into Genome Assembly algorithms",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/algorithms-introduction/slides.html"
  },
  "/topics/climate/tutorials/fates/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/fates/tutorial.html"
  },
  "/topics/climate/tutorials/ocean-variables/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Ocean's variables study",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged netcdf data\" href=\"/training-material/search?query=netcdf data\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">netcdf data</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/ocean-variables/tutorial.html"
  },
  "/topics/climate/tutorials/argo_pangeo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Analyse Argo data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00eea4; --color-darker: #00d492; --color-dimmed: #00e19b;\">pangeo</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged argo\" href=\"/training-material/search?query=argo\" style=\"--color-primary: #ffc2c5; --color-darker: #ffa0a6; --color-dimmed: #ffb2b6;\">argo</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/argo_pangeo/tutorial.html"
  },
  "/topics/climate/tutorials/climate-101/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Getting your hands-on climate data",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/climate/tutorials/climate-101/tutorial.html"
  },
  "/topics/climate/tutorials/pangeo-notebook/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Pangeo Notebook in Galaxy - Introduction to Xarray",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00eea4; --color-darker: #00d492; --color-dimmed: #00e19b;\">pangeo</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/pangeo-notebook/tutorial.html"
  },
  "/topics/climate/tutorials/pangeo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Pangeo ecosystem 101 for everyone - Introduction to Xarray Galaxy Tools",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00eea4; --color-darker: #00d492; --color-dimmed: #00e19b;\">pangeo</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/pangeo/tutorial.html"
  },
  "/topics/climate/tutorials/sentinel5_data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Sentinel 5P data visualisation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged volcano\" href=\"/training-material/search?query=volcano\" style=\"--color-primary: #00e9e9; --color-darker: #00cfcf; --color-dimmed: #00dcdc;\">volcano</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged satellite data\" href=\"/training-material/search?query=satellite data\" style=\"--color-primary: #b4e200; --color-darker: #a0c900; --color-dimmed: #aad500;\">satellite data</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/sentinel5_data/tutorial.html"
  },
  "/topics/climate/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Introduction to climate data",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/climate/tutorials/introduction/slides.html"
  },
  "/topics/climate/tutorials/fates-jupyterlab/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html"
  },
  "/topics/climate/tutorials/ocean-data-view/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Ocean Data View (ODV)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged geographical information system\" href=\"/training-material/search?query=geographical information system\" style=\"--color-primary: #f5bfff; --color-darker: #f19bff; --color-dimmed: #f3adff;\">geographical information system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ODV\" href=\"/training-material/search?query=ODV\" style=\"--color-primary: #add6ff; --color-darker: #76c1ff; --color-dimmed: #94cbff;\">ODV</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged netcdf data\" href=\"/training-material/search?query=netcdf data\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">netcdf data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged maps\" href=\"/training-material/search?query=maps\" style=\"--color-primary: #98daff; --color-darker: #32c8ff; --color-dimmed: #72d1ff;\">maps</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged marine data\" href=\"/training-material/search?query=marine data\" style=\"--color-primary: #98daff; --color-darker: #32c8ff; --color-dimmed: #72d1ff;\">marine data</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/ocean-data-view/tutorial.html"
  },
  "/topics/climate/tutorials/earth_system/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Getting your hands-on earth data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged land\" href=\"/training-material/search?query=land\" style=\"--color-primary: #00e8f4; --color-darker: #00ced9; --color-dimmed: #00dbe7;\">land</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged atmosphere\" href=\"/training-material/search?query=atmosphere\" style=\"--color-primary: #ffbeeb; --color-darker: #ff98e2; --color-dimmed: #fface6;\">atmosphere</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged biodiversity\" href=\"/training-material/search?query=biodiversity\" style=\"--color-primary: #00eadb; --color-darker: #00d0c3; --color-dimmed: #00ddcf;\">biodiversity</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/earth_system/tutorial.html"
  },
  "/topics/climate/tutorials/panoply/tutorial.html": {
    "type": "Tutorial",
    "topic": "Climate",
    "title": "Visualize Climate data with Panoply netCDF viewer",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/climate/tutorials/panoply/tutorial.html"
  },
  "/topics/community/tutorials/tools_subdomains/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Community Building",
    "title": "Make your tools available on your subdomain",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged subdomain\" href=\"/training-material/search?query=subdomain\" style=\"--color-primary: #ffc78e; --color-darker: #fbaa00; --color-dimmed: #ffb85e;\">subdomain</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/community/tutorials/tools_subdomains/tutorial.html"
  },
  "/topics/community/tutorials/community-tool-table/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Community Building",
    "title": "Creation of an interactive Galaxy tools table for your community",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Community\" href=\"/training-material/search?query=Community\" style=\"--color-primary: #00ebd3; --color-darker: #00d1bb; --color-dimmed: #00dec7;\">Community</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SIG\" href=\"/training-material/search?query=SIG\" style=\"--color-primary: #00e9e4; --color-darker: #00d0ca; --color-dimmed: #00ddd7;\">SIG</a>"
    ],
    "url": "/training-material/topics/community/tutorials/community-tool-table/tutorial.html"
  },
  "/topics/community/tutorials/sig_define/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Community Building",
    "title": "What's a Special Interest Group?",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/community/tutorials/sig_define/tutorial.html"
  },
  "/topics/community/tutorials/sig_create/tutorial.html": {
    "type": "Tutorial",
    "topic": "Galaxy Community Building",
    "title": "Creating a Special Interest Group",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/community/tutorials/sig_create/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Protein-ligand docking",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "High Throughput Molecular Dynamics and Analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/htmd-analysis/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Data management in Medicinal Chemistry",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data-management\" href=\"/training-material/search?query=data-management\" style=\"--color-primary: #98daff; --color-darker: #32c8ff; --color-dimmed: #72d1ff;\">data-management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged medicinal-chemistry\" href=\"/training-material/search?query=medicinal-chemistry\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">medicinal-chemistry</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged computational-chemistry\" href=\"/training-material/search?query=computational-chemistry\" style=\"--color-primary: #00f07e; --color-darker: #00d670; --color-dimmed: #00e377;\">computational-chemistry</a>"
    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Protein target prediction of a bioactive ligand with Align-it and ePharmaLib",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/zauberkugel/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Running molecular dynamics simulations using GROMACS",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/md-simulation-gromacs/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html": {
    "type": "Tutorial",
    "topic": "One Health",
    "title": "Virtual screening of the SARS-CoV-2 main protease with rxDock and pose scoring",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>"
    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/covid19-docking/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Analysis of molecular dynamics simulations",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/analysis-md-simulations/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Running molecular dynamics simulations using NAMD",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/md-simulation-namd/tutorial.html"
  },
  "/topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.html": {
    "type": "Tutorial",
    "topic": "Computational chemistry",
    "title": "Setting up molecular systems",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/computational-chemistry/tutorials/setting-up-molecular-systems/tutorial.html"
  },
  "/topics/contributing/tutorials/github-command-line-contribution/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Contributing with GitHub via command-line",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/github-command-line-contribution/tutorial.html"
  },
  "/topics/contributing/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Overview of the Galaxy Training Material",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/introduction/slides.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-tours/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Creating Interactive Galaxy Tours",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-tours/tutorial.html"
  },
  "/topics/contributing/tutorials/github-interface-contribution/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Contributing with GitHub via its interface",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/github-interface-contribution/tutorial.html"
  },
  "/topics/contributing/tutorials/generating-pdf/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Generating PDF artefacts of the website",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/generating-pdf/tutorial.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Tools, Data, and Workflows for tutorials",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-technical/tutorial.html"
  },
  "/topics/contributing/tutorials/python/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Teaching Python",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/python/tutorial.html"
  },
  "/topics/contributing/tutorials/running-codespaces/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Running the GTN website online using GitHub CodeSpaces",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/running-codespaces/tutorial.html"
  },
  "/topics/contributing/tutorials/design/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Design and plan session, course, materials",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/design/tutorial.html"
  },
  "/topics/contributing/tutorials/schemas/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "GTN Metadata",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/schemas/tutorial.html"
  },
  "/topics/contributing/tutorials/gitpod/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Running the GTN website online using GitPod",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/gitpod/tutorial.html"
  },
  "/topics/contributing/tutorials/fair-by-design/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "FAIR-by-Design methodology",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged FAIR-by-Design Learning Materials\" href=\"/training-material/search?query=FAIR-by-Design Learning Materials\" style=\"--color-primary: #bcd2ff; --color-darker: #94bbff; --color-dimmed: #a9c7ff;\">FAIR-by-Design Learning Materials</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged FAIR Learning Objects\" href=\"/training-material/search?query=FAIR Learning Objects\" style=\"--color-primary: #ffbcf7; --color-darker: #ff95f4; --color-dimmed: #ffa9f6;\">FAIR Learning Objects</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged FAIR-by-Design Methodology\" href=\"/training-material/search?query=FAIR-by-Design Methodology\" style=\"--color-primary: #9fd9ff; --color-darker: #4ec6ff; --color-dimmed: #7dcfff;\">FAIR-by-Design Methodology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/contributing/tutorials/fair-by-design/tutorial.html"
  },
  "/topics/contributing/tutorials/slides-with-video/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Adding auto-generated video to your slides",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/slides-with-video/tutorial.html"
  },
  "/topics/contributing/tutorials/create-new-topic/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Including a new topic",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-topic/tutorial.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-quiz/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Adding Quizzes to your Tutorial",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-quiz/tutorial.html"
  },
  "/topics/contributing/tutorials/learning-principles/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Principles of learning and how they apply to training and teaching",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/learning-principles/tutorial.html"
  },
  "/topics/contributing/tutorials/admin-knitting/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Updating diffs in admin training",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/admin-knitting/tutorial.html"
  },
  "/topics/contributing/tutorials/running-jekyll/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Running the GTN website locally using the command line",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/running-jekyll/tutorial.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Creating a new tutorial",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial/tutorial.html"
  },
  "/topics/contributing/tutorials/meta-analysis-plot/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Single Cell Publication - Data Plotting",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">rmarkdown-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/contributing/tutorials/meta-analysis-plot/tutorial.html"
  },
  "/topics/contributing/tutorials/fair-gtn/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "FAIR Galaxy Training Material",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gtn\" href=\"/training-material/search?query=gtn\" style=\"--color-primary: #fff5be; --color-darker: #efdb00; --color-dimmed: #fee800;\">gtn</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged training\" href=\"/training-material/search?query=training\" style=\"--color-primary: #ffbdf5; --color-darker: #ff95f0; --color-dimmed: #ffaaf3;\">training</a>"
    ],
    "url": "/training-material/topics/contributing/tutorials/fair-gtn/tutorial.html"
  },
  "/topics/contributing/tutorials/meta-analysis-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Single Cell Publication - Data Analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Ruby\" href=\"/training-material/search?query=Ruby\" style=\"--color-primary: #00eadd; --color-darker: #00d0c5; --color-dimmed: #00ddd1;\">Ruby</a>"
    ],
    "url": "/training-material/topics/contributing/tutorials/meta-analysis-data/tutorial.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Creating content in Markdown",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-slides/slides.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Creating Slides",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-slides/slides.html"
  },
  "/topics/contributing/tutorials/updating_tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Contributing to the Galaxy Training Material",
    "title": "Updating tool versions in a tutorial",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/updating_tutorial/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Mutation calling, viral genome reconstruction and lineage/clade assignment from SARS-CoV-2 sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc0e0; --color-darker: #ff9bd1; --color-dimmed: #ffadd8;\">virology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "From NCBI's Sequence Read Archive (SRA) to Galaxy: SARS-CoV-2 variant analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc0e0; --color-darker: #ff9bd1; --color-dimmed: #ffadd8;\">virology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/sars-cov-2/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Automating Galaxy workflows using the command line",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged variant-analysis\" href=\"/training-material/search?query=variant-analysis\" style=\"--color-primary: #00edba; --color-darker: #00d3a5; --color-dimmed: #00e0af;\">variant-analysis</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-automation/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Submitting sequence data to ENA",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-data-to-ena/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Removal of human reads from SARS-CoV-2 sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>"
    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/human-reads-removal/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "SRA Aligned Read Format to Speed Up SARS-CoV-2 data Analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ncbi\" href=\"/training-material/search?query=ncbi\" style=\"--color-primary: #ffc1d4; --color-darker: #ff9dbe; --color-dimmed: #ffafc9;\">ncbi</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/sars-with-galaxy-on-anvil/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "SARS-CoV-2 Viral Sample Alignment and Variant Visualization",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged covid19\" href=\"/training-material/search?query=covid19\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">covid19</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/sars-with-galaxy-on-anvil/tutorial.html"
  },
  "/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Data Manipulation Olympics",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/data-manipulation-olympics/tutorial.html"
  },
  "/topics/data-science/tutorials/online-resources-gene/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Learning about one gene across biological resources and formats",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/online-resources-gene/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture1/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "A (very) brief history of genomics",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture1/tutorial.html"
  },
  "/topics/data-science/tutorials/sql-python/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "SQL with Python",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SQL\" href=\"/training-material/search?query=SQL\" style=\"--color-primary: #ffc1d7; --color-darker: #ff9dc3; --color-dimmed: #ffafcd;\">SQL</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Python\" href=\"/training-material/search?query=Python\" style=\"--color-primary: #d9caff; --color-darker: #c7adff; --color-dimmed: #d0bbff;\">Python</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/sql-python/tutorial.html"
  },
  "/topics/data-science/tutorials/online-resources-protein/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "One protein along the UniProt page",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/online-resources-protein/tutorial.html"
  },
  "/topics/data-science/tutorials/cli-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Advanced CLI in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bash\" href=\"/training-material/search?query=bash\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">bash</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/cli-advanced/tutorial.html"
  },
  "/topics/data-science/tutorials/cli-bashcrawl/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "CLI Educational Game - Bashcrawl",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged game\" href=\"/training-material/search?query=game\" style=\"--color-primary: #ffbdef; --color-darker: #ff97e8; --color-dimmed: #ffabeb;\">game</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bash\" href=\"/training-material/search?query=bash\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">bash</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/cli-bashcrawl/tutorial.html"
  },
  "/topics/data-science/tutorials/snakemake/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Make & Snakemake",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/snakemake/tutorial.html"
  },
  "/topics/data-science/tutorials/python-math/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Math",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-math/tutorial.html"
  },
  "/topics/data-science/tutorials/bash-variant-calling/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Variant Calling Workflow",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/bash-variant-calling/tutorial.html"
  },
  "/topics/data-science/tutorials/sql-basic/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Introduction to SQL",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SQL\" href=\"/training-material/search?query=SQL\" style=\"--color-primary: #ffc1d7; --color-darker: #ff9dc3; --color-dimmed: #ffafcd;\">SQL</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/sql-basic/tutorial.html"
  },
  "/topics/data-science/tutorials/r-basics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "R basics in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/r-basics/tutorial.html"
  },
  "/topics/data-science/tutorials/sql-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "SQL with R",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SQL\" href=\"/training-material/search?query=SQL\" style=\"--color-primary: #ffc1d7; --color-darker: #ff9dc3; --color-dimmed: #ffafcd;\">SQL</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">rmarkdown-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/sql-r/tutorial.html"
  },
  "/topics/data-science/tutorials/sql-game/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "SQL Educational Game - Murder Mystery",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged game\" href=\"/training-material/search?query=game\" style=\"--color-primary: #ffbdef; --color-darker: #ff97e8; --color-dimmed: #ffabeb;\">game</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SQL\" href=\"/training-material/search?query=SQL\" style=\"--color-primary: #ffc1d7; --color-darker: #ff9dc3; --color-dimmed: #ffafcd;\">SQL</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/sql-game/tutorial.html"
  },
  "/topics/data-science/tutorials/git-cli/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Basics of using Git from the Command Line",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/git-cli/tutorial.html"
  },
  "/topics/data-science/tutorials/python-basics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Introduction to Python",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-basics/tutorial.html"
  },
  "/topics/data-science/tutorials/sql-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Advanced SQL",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SQL\" href=\"/training-material/search?query=SQL\" style=\"--color-primary: #ffc1d7; --color-darker: #ff9dc3; --color-dimmed: #ffafcd;\">SQL</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/sql-advanced/tutorial.html"
  },
  "/topics/data-science/tutorials/bash-git/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Version Control with Git",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/bash-git/tutorial.html"
  },
  "/topics/data-science/tutorials/cli-basics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "CLI basics",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bash\" href=\"/training-material/search?query=bash\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">bash</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/cli-basics/tutorial.html"
  },
  "/topics/data-science/tutorials/data-manipulation-olympics-viz-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Data visualisation Olympics - Visualization in R",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">rmarkdown-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/data-manipulation-olympics-viz-r/tutorial.html"
  },
  "/topics/data-science/tutorials/python-advanced-np-pd/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Advanced Python",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-advanced-np-pd/tutorial.html"
  },
  "/topics/data-science/tutorials/r-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Advanced R in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/r-advanced/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture2/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Introduction to sequencing with Python (part one)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture2/tutorial.html"
  },
  "/topics/data-science/tutorials/python-functions/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Functions",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-functions/tutorial.html"
  },
  "/topics/data-science/tutorials/data-manipulation-olympics-sql/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Data Manipulation Olympics - SQL",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged sql\" href=\"/training-material/search?query=sql\" style=\"--color-primary: #ffbee8; --color-darker: #ff99dd; --color-dimmed: #fface2;\">sql</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/data-manipulation-olympics-sql/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture3/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Introduction to sequencing with Python (part two)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture3/tutorial.html"
  },
  "/topics/data-science/tutorials/r-dplyr/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "dplyr & tidyverse for data processing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">rmarkdown-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/r-dplyr/tutorial.html"
  },
  "/topics/data-science/tutorials/python-types/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Basic Types & Type Conversion",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-types/tutorial.html"
  },
  "/topics/data-science/tutorials/python-plotting/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Plotting in Python",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-plotting/tutorial.html"
  },
  "/topics/data-science/tutorials/data-manipulation-olympics-jq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Data Manipulation Olympics - JQ",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged json\" href=\"/training-material/search?query=json\" style=\"--color-primary: #ffc1d4; --color-darker: #ff9dbe; --color-dimmed: #ffafc9;\">json</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/data-manipulation-olympics-jq/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture4/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Introduction to sequencing with Python (part three)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture4/tutorial.html"
  },
  "/topics/data-science/tutorials/python-iterables/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Lists & Strings & Dictionaries",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-iterables/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture5/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Introduction to sequencing with Python (part four)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture5/tutorial.html"
  },
  "/topics/data-science/tutorials/python-flow/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Flow Control",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-flow/tutorial.html"
  },
  "/topics/data-science/tutorials/python-loops/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Loops",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-loops/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture6/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Data manipulation with Pandas",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html"
  },
  "/topics/data-science/tutorials/python-files/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Files & CSV",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-files/tutorial.html"
  },
  "/topics/data-science/tutorials/gnmx-lecture7/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Versioning your code and data with git",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/gnmx-lecture7/tutorial.html"
  },
  "/topics/data-science/tutorials/python-exceptions/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Try & Except",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-exceptions/tutorial.html"
  },
  "/topics/data-science/tutorials/python-basics-recap/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Introductory Graduation",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-basics-recap/tutorial.html"
  },
  "/topics/data-science/tutorials/python-testing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Testing",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-testing/tutorial.html"
  },
  "/topics/data-science/tutorials/python-venv/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Virtual Environments For Software Development",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-venv/tutorial.html"
  },
  "/topics/data-science/tutorials/python-conda/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Conda Environments For Software Development",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged conda\" href=\"/training-material/search?query=conda\" style=\"--color-primary: #00f07e; --color-darker: #00d670; --color-dimmed: #00e377;\">conda</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/data-science/tutorials/python-conda/tutorial.html"
  },
  "/topics/data-science/tutorials/python-typing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Type annotations",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-typing/tutorial.html"
  },
  "/topics/data-science/tutorials/python-argparse/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Argparse",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-argparse/tutorial.html"
  },
  "/topics/data-science/tutorials/python-glob/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Globbing",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-glob/tutorial.html"
  },
  "/topics/data-science/tutorials/python-subprocess/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Subprocess",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-subprocess/tutorial.html"
  },
  "/topics/data-science/tutorials/python-linting/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Coding Style",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-linting/tutorial.html"
  },
  "/topics/data-science/tutorials/python-multiprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Foundations of Data Science",
    "title": "Python - Multiprocessing",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/python-multiprocessing/tutorial.html"
  },
  "/topics/dev/tutorials/conda_sys/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Prerequisites for building software/conda packages",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/conda_sys/slides.html"
  },
  "/topics/dev/tutorials/interactive-tours/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Tours",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/interactive-tours/slides.html"
  },
  "/topics/dev/tutorials/conda/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool Dependencies and Conda",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/conda/slides.html"
  },
  "/topics/dev/tutorials/containers/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool Dependencies and Containers",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/containers/slides.html"
  },
  "/topics/dev/tutorials/toolshed/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool Shed: sharing Galaxy tools",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/toolshed/slides.html"
  },
  "/topics/dev/tutorials/webhooks/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Webhooks",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/webhooks/tutorial.html"
  },
  "/topics/dev/tutorials/bioblend-api/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Scripting Galaxy using the API and BioBlend",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/bioblend-api/tutorial.html"
  },
  "/topics/dev/tutorials/visualization-generic/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Generic plugins",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/visualization-generic/tutorial.html"
  },
  "/topics/dev/tutorials/tool-integration/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Tool development and integration into Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-integration/slides.html"
  },
  "/topics/dev/tutorials/architecture/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Code Architecture",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/architecture/slides.html"
  },
  "/topics/dev/tutorials/tool-generators/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "ToolFactory: Generating Tools From Simple Scripts",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/tutorial.html"
  },
  "/topics/dev/tutorials/interactive-environments/slides.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Environments",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/dev/tutorials/interactive-environments/slides.html"
  },
  "/topics/dev/tutorials/visualization-charts/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "JavaScript plugins",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/visualization-charts/tutorial.html"
  },
  "/topics/dev/tutorials/tool-annotation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Adding and updating best practice metadata for Galaxy tools using the bio.tools registry",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-annotation/tutorial.html"
  },
  "/topics/dev/tutorials/writing_tests/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Writing Automated Tests for Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/writing_tests/tutorial.html"
  },
  "/topics/dev/tutorials/bioblend-dev/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Contributing to BioBlend as a developer",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/bioblend-dev/tutorial.html"
  },
  "/topics/dev/tutorials/core-contributing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Contributing a New Feature to Galaxy Core",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/core-contributing/tutorial.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Creating Galaxy tools from Conda Through Deployment",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/tutorial.html"
  },
  "/topics/dev/tutorials/interactive-tools/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Galaxy Interactive Tools",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/interactive-tools/tutorial.html"
  },
  "/topics/dev/tutorials/data-source-integration/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Data source integration",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/data-source-integration/tutorial.html"
  },
  "/topics/dev/tutorials/debugging/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "Debugging Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/debugging/tutorial.html"
  },
  "/topics/dev/tutorials/tool-generators-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Development in Galaxy",
    "title": "ToolFactory: Generating Tools From More Complex Scripts",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators-advanced/tutorial.html"
  },
  "/topics/ecology/tutorials/x-array-map-plot/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Visualization of Climate Data using  NetCDF xarray Map Plotting",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged pangeo\" href=\"/training-material/search?query=pangeo\" style=\"--color-primary: #00eea4; --color-darker: #00d492; --color-dimmed: #00e19b;\">pangeo</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/x-array-map-plot/tutorial.html"
  },
  "/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Metabarcoding/eDNA through Obitools",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Community composition EBV class\" href=\"/training-material/search?query=Community composition EBV class\" style=\"--color-primary: #add6ff; --color-darker: #76c1ff; --color-dimmed: #94cbff;\">Community composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eDNA\" href=\"/training-material/search?query=eDNA\" style=\"--color-primary: #fdfb00; --color-darker: #e1df00; --color-dimmed: #efed00;\">eDNA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Metabarcoding\" href=\"/training-material/search?query=Metabarcoding\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">Metabarcoding</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html"
  },
  "/topics/ecology/tutorials/ndvi_openeo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "From NDVI data with OpenEO to time series visualisation with Holoviews",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged land degradation\" href=\"/training-material/search?query=land degradation\" style=\"--color-primary: #00ecc1; --color-darker: #00d2ab; --color-dimmed: #00dfb6;\">land degradation</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged NDVI\" href=\"/training-material/search?query=NDVI\" style=\"--color-primary: #ffc2ce; --color-darker: #ff9fb4; --color-dimmed: #ffb0c1;\">NDVI</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged copernicus\" href=\"/training-material/search?query=copernicus\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">copernicus</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged holoviews\" href=\"/training-material/search?query=holoviews\" style=\"--color-primary: #00e9eb; --color-darker: #00cfd1; --color-dimmed: #00dcde;\">holoviews</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/ndvi_openeo/tutorial.html"
  },
  "/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "RAD-Seq to construct genetic maps",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?query=RAD-seq\" style=\"--color-primary: #fff4ca; --color-darker: #f5d900; --color-dimmed: #ffe667;\">RAD-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/genetic-map-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "RAD-Seq de-novo data analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?query=RAD-seq\" style=\"--color-primary: #fff4ca; --color-darker: #f5d900; --color-dimmed: #ffe667;\">RAD-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species populations EBV class\" href=\"/training-material/search?query=Species populations EBV class\" style=\"--color-primary: #00ead9; --color-darker: #00d1c1; --color-dimmed: #00ddcd;\">Species populations EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/obisindicators/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Obis marine indicators",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged marine omics\" href=\"/training-material/search?query=marine omics\" style=\"--color-primary: #c7cfff; --color-darker: #a8b7ff; --color-dimmed: #b8c3ff;\">marine omics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged biodiversity\" href=\"/training-material/search?query=biodiversity\" style=\"--color-primary: #00eadb; --color-darker: #00d0c3; --color-dimmed: #00ddcf;\">biodiversity</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/obisindicators/tutorial.html"
  },
  "/topics/ecology/tutorials/gbif_cleaning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Cleaning GBIF data for the use in Ecology",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gbif\" href=\"/training-material/search?query=gbif\" style=\"--color-primary: #ffc3c4; --color-darker: #ffa1a3; --color-dimmed: #ffb2b4;\">gbif</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data cleaning\" href=\"/training-material/search?query=data cleaning\" style=\"--color-primary: #00f160; --color-darker: #00d755; --color-dimmed: #00e45b;\">data cleaning</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/gbif_cleaning/tutorial.html"
  },
  "/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Metadata\" href=\"/training-material/search?query=Metadata\" style=\"--color-primary: #00efa1; --color-darker: #00d48f; --color-dimmed: #00e198;\">Metadata</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EML\" href=\"/training-material/search?query=EML\" style=\"--color-primary: #c3d0ff; --color-darker: #a2b8ff; --color-dimmed: #b3c4ff;\">EML</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Ecology\" href=\"/training-material/search?query=Ecology\" style=\"--color-primary: #ffbbfd; --color-darker: #ff93fc; --color-dimmed: #ffa8fd;\">Ecology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Biodiversity\" href=\"/training-material/search?query=Biodiversity\" style=\"--color-primary: #91dbff; --color-darker: #00cafd; --color-dimmed: #64d3ff;\">Biodiversity</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged FAIR\" href=\"/training-material/search?query=FAIR\" style=\"--color-primary: #d4cbff; --color-darker: #beb0ff; --color-dimmed: #c9beff;\">FAIR</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Data Paper\" href=\"/training-material/search?query=Data Paper\" style=\"--color-primary: #ffc0db; --color-darker: #ff9cc9; --color-dimmed: #ffaed2;\">Data Paper</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html"
  },
  "/topics/ecology/tutorials/remote-sensing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Sentinel 2 biodiversity",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Remote sensing\" href=\"/training-material/search?query=Remote sensing\" style=\"--color-primary: #ffc1d5; --color-darker: #ff9dc0; --color-dimmed: #ffafca;\">Remote sensing</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/remote-sensing/tutorial.html"
  },
  "/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Preparing genomic data for phylogeny reconstruction",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged phylogeny\" href=\"/training-material/search?query=phylogeny\" style=\"--color-primary: #00f314; --color-darker: #00d811; --color-dimmed: #00e612;\">phylogeny</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data handling\" href=\"/training-material/search?query=data handling\" style=\"--color-primary: #fff88c; --color-darker: #e7dd00; --color-dimmed: #f5eb00;\">data handling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged functional annotation\" href=\"/training-material/search?query=functional annotation\" style=\"--color-primary: #6be1ff; --color-darker: #00cced; --color-dimmed: #00d8fc;\">functional annotation</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/phylogeny-data-prep/tutorial.html"
  },
  "/topics/ecology/tutorials/species-distribution-modeling/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Species distribution modeling",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged modeling\" href=\"/training-material/search?query=modeling\" style=\"--color-primary: #ddc8ff; --color-darker: #ccabff; --color-dimmed: #d4baff;\">modeling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gbif\" href=\"/training-material/search?query=gbif\" style=\"--color-primary: #ffc3c4; --color-darker: #ffa1a3; --color-dimmed: #ffb2b4;\">gbif</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged species populations EBV class\" href=\"/training-material/search?query=species populations EBV class\" style=\"--color-primary: #e1d600; --color-darker: #c8be00; --color-dimmed: #d5ca00;\">species populations EBV class</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/species-distribution-modeling/tutorial.html"
  },
  "/topics/ecology/tutorials/bacterial-isolate-species-contamination-checking/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Checking expected species and contamination in bacterial isolate",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?query=illumina\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bacteria\" href=\"/training-material/search?query=bacteria\" style=\"--color-primary: #ffc973; --color-darker: #f0af00; --color-dimmed: #feba00;\">bacteria</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/bacterial-isolate-species-contamination-checking/tutorial.html"
  },
  "/topics/ecology/tutorials/marine_omics_bgc/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Marine Omics identifying biosynthetic gene clusters",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged marine omics\" href=\"/training-material/search?query=marine omics\" style=\"--color-primary: #c7cfff; --color-darker: #a8b7ff; --color-dimmed: #b8c3ff;\">marine omics</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/marine_omics_bgc/tutorial.html"
  },
  "/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "RAD-Seq Reference-based data analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged RAD-seq\" href=\"/training-material/search?query=RAD-seq\" style=\"--color-primary: #fff4ca; --color-darker: #f5d900; --color-dimmed: #ffe667;\">RAD-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Genetic composition EBV class\" href=\"/training-material/search?query=Genetic composition EBV class\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">Genetic composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species population EBV class\" href=\"/training-material/search?query=Species population EBV class\" style=\"--color-primary: #00ead9; --color-darker: #00d1c1; --color-dimmed: #00ddcd;\">Species population EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html"
  },
  "/topics/ecology/tutorials/regionalGAM/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Regional GAM",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species populations EBV class\" href=\"/training-material/search?query=Species populations EBV class\" style=\"--color-primary: #00ead9; --color-darker: #00d1c1; --color-dimmed: #00ddcd;\">Species populations EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species traits EBV class\" href=\"/training-material/search?query=Species traits EBV class\" style=\"--color-primary: #c8cfff; --color-darker: #aab6ff; --color-dimmed: #b9c3ff;\">Species traits EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/regionalGAM/tutorial.html"
  },
  "/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Biodiversity data exploration",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged taxonomic data\" href=\"/training-material/search?query=taxonomic data\" style=\"--color-primary: #73ed00; --color-darker: #65d300; --color-dimmed: #6ce000;\">taxonomic data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data quality\" href=\"/training-material/search?query=data quality\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">data quality</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/biodiversity-data-exploration/tutorial.html"
  },
  "/topics/ecology/tutorials/Metashrimps_tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Creating FAIR Quality assessment reports and draft of Data Papers from EML metadata with MetaShRIMPS",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Metadata\" href=\"/training-material/search?query=Metadata\" style=\"--color-primary: #00efa1; --color-darker: #00d48f; --color-dimmed: #00e198;\">Metadata</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EML\" href=\"/training-material/search?query=EML\" style=\"--color-primary: #c3d0ff; --color-darker: #a2b8ff; --color-dimmed: #b3c4ff;\">EML</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged FAIR\" href=\"/training-material/search?query=FAIR\" style=\"--color-primary: #d4cbff; --color-darker: #beb0ff; --color-dimmed: #c9beff;\">FAIR</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Data Paper\" href=\"/training-material/search?query=Data Paper\" style=\"--color-primary: #ffc0db; --color-darker: #ff9cc9; --color-dimmed: #ffaed2;\">Data Paper</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/Metashrimps_tutorial/tutorial.html"
  },
  "/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Compute and analyze biodiversity metrics with PAMPA toolsuite",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Species population EBV class\" href=\"/training-material/search?query=Species population EBV class\" style=\"--color-primary: #00ead9; --color-darker: #00d1c1; --color-dimmed: #00ddcd;\">Species population EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Community composition EBV class\" href=\"/training-material/search?query=Community composition EBV class\" style=\"--color-primary: #add6ff; --color-darker: #76c1ff; --color-dimmed: #94cbff;\">Community composition EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged modeling\" href=\"/training-material/search?query=modeling\" style=\"--color-primary: #ddc8ff; --color-darker: #ccabff; --color-dimmed: #d4baff;\">modeling</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/tutorial.html"
  },
  "/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Ecoregionalization workflow tutorial",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ecology\" href=\"/training-material/search?query=ecology\" style=\"--color-primary: #ffbcf9; --color-darker: #ff94f6; --color-dimmed: #ffa9f7;\">ecology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged taxonomic data\" href=\"/training-material/search?query=taxonomic data\" style=\"--color-primary: #73ed00; --color-darker: #65d300; --color-dimmed: #6ce000;\">taxonomic data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged modeling\" href=\"/training-material/search?query=modeling\" style=\"--color-primary: #ddc8ff; --color-darker: #ccabff; --color-dimmed: #d4baff;\">modeling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gbif\" href=\"/training-material/search?query=gbif\" style=\"--color-primary: #ffc3c4; --color-darker: #ffa1a3; --color-dimmed: #ffb2b4;\">gbif</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/Ecoregionalization_tutorial/tutorial.html"
  },
  "/topics/ecology/tutorials/champs-blocs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Champs blocs indicators",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Ecosystem EBV class\" href=\"/training-material/search?query=Ecosystem EBV class\" style=\"--color-primary: #00e9e7; --color-darker: #00cfce; --color-dimmed: #00dcdb;\">Ecosystem EBV class</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV dataset\" href=\"/training-material/search?query=EBV dataset\" style=\"--color-primary: #eac4ff; --color-darker: #e0a3ff; --color-dimmed: #e5b4ff;\">EBV dataset</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV workflow\" href=\"/training-material/search?query=EBV workflow\" style=\"--color-primary: #00ecc4; --color-darker: #00d2ae; --color-dimmed: #00dfb9;\">EBV workflow</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Marine ecosystems\" href=\"/training-material/search?query=Marine ecosystems\" style=\"--color-primary: #ffbcf6; --color-darker: #ff95f2; --color-dimmed: #ffa9f4;\">Marine ecosystems</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/champs-blocs/tutorial.html"
  },
  "/topics/ecology/tutorials/QGIS_Web_Feature_Services/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "QGIS Web Feature Services",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged GIS\" href=\"/training-material/search?query=GIS\" style=\"--color-primary: #4fe3ff; --color-darker: #00cde8; --color-dimmed: #00d9f6;\">GIS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Geographical Information System\" href=\"/training-material/search?query=Geographical Information System\" style=\"--color-primary: #d1db00; --color-darker: #bac300; --color-dimmed: #c5cf00;\">Geographical Information System</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged WFS\" href=\"/training-material/search?query=WFS\" style=\"--color-primary: #ffc3c0; --color-darker: #ffa19c; --color-dimmed: #ffb3ae;\">WFS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Spatial data\" href=\"/training-material/search?query=Spatial data\" style=\"--color-primary: #d9caff; --color-darker: #c7adff; --color-dimmed: #d0bbff;\">Spatial data</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Maps\" href=\"/training-material/search?query=Maps\" style=\"--color-primary: #ffc1d2; --color-darker: #ff9ebc; --color-dimmed: #ffb0c7;\">Maps</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged OGC\" href=\"/training-material/search?query=OGC\" style=\"--color-primary: #ffc3be; --color-darker: #ffa298; --color-dimmed: #ffb3ac;\">OGC</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/QGIS_Web_Feature_Services/tutorial.html"
  },
  "/topics/ecology/tutorials/panoply_ebv/tutorial.html": {
    "type": "Tutorial",
    "topic": "Ecology",
    "title": "Visualize EBV cube data with Panoply netCDF viewer",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged EBV cube\" href=\"/training-material/search?query=EBV cube\" style=\"--color-primary: #ffbbfd; --color-darker: #ff93fc; --color-dimmed: #ffa8fd;\">EBV cube</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Data visualization\" href=\"/training-material/search?query=Data visualization\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">Data visualization</a>"
    ],
    "url": "/training-material/topics/ecology/tutorials/panoply_ebv/tutorial.html"
  },
  "/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Formation of the Super-Structures on the Inactive X",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?query=ChIP-seq\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">ChIP-seq</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html"
  },
  "/topics/epigenetics/tutorials/methylation-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "DNA Methylation data analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html"
  },
  "/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Identification of the binding sites of the T-cell acute lymphocytic leukemia protein 1 (TAL1)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?query=ChIP-seq\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">ChIP-seq</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/tutorial.html"
  },
  "/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Identification of the binding sites of the Estrogen receptor",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ChIP-seq\" href=\"/training-material/search?query=ChIP-seq\" style=\"--color-primary: #ffbfe3; --color-darker: #ff9ad6; --color-dimmed: #ffaddd;\">ChIP-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html"
  },
  "/topics/epigenetics/tutorials/ewas-suite/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Infinium Human Methylation BeadChip",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "ATAC-Seq data analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/tutorial.html"
  },
  "/topics/epigenetics/tutorials/hicexplorer/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Hi-C analysis of Drosophila melanogaster cells using HiCExplorer",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html"
  },
  "/topics/epigenetics/tutorials/cut_and_run/tutorial.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "CUT&RUN data analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/epigenetics/tutorials/cut_and_run/tutorial.html"
  },
  "/topics/epigenetics/tutorials/introduction-dna-methylation/slides.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Introduction to DNA Methylation data analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/introduction-dna-methylation/slides.html"
  },
  "/topics/epigenetics/tutorials/introduction-chip-seq/slides.html": {
    "type": "Tutorial",
    "topic": "Epigenetics",
    "title": "Introduction to ChIP-Seq data analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/introduction-chip-seq/slides.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to basics",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged phylogenetics\" href=\"/training-material/search?query=phylogenetics\" style=\"--color-primary: #00f160; --color-darker: #00d755; --color-dimmed: #00e45b;\">phylogenetics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged evolution\" href=\"/training-material/search?query=evolution\" style=\"--color-primary: #ffc0db; --color-darker: #ff9cc9; --color-dimmed: #ffaed2;\">evolution</a>"
    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo/tutorial.html"
  },
  "/topics/evolution/tutorials/mtb_phylogeny/tutorial.html": {
    "type": "Tutorial",
    "topic": "One Health",
    "title": "Tree thinking for tuberculosis evolution and epidemiology",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged phylogenetics\" href=\"/training-material/search?query=phylogenetics\" style=\"--color-primary: #00f160; --color-darker: #00d755; --color-dimmed: #00e45b;\">phylogenetics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/evolution/tutorials/mtb_phylogeny/tutorial.html"
  },
  "/topics/evolution/tutorials/mtb_transmission/tutorial.html": {
    "type": "Tutorial",
    "topic": "One Health",
    "title": "Identifying tuberculosis transmission links: from SNPs to transmission clusters",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/evolution/tutorials/mtb_transmission/tutorial.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo-intro/slides.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to Basics - Introduction",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo-intro/slides.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo-terminology/slides.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to Basics - Terminology",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo-terminology/slides.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo-msa/slides.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to Basics - Multiple Sequence Alignment",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo-msa/slides.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo-trees/slides.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to Basics - Building Trees",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo-trees/slides.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo-mle-trees/slides.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to Basics - Estimating trees from alignments",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo-mle-trees/slides.html"
  },
  "/topics/evolution/tutorials/abc_intro_phylo-networks/slides.html": {
    "type": "Tutorial",
    "topic": "Evolution",
    "title": "Phylogenetics - Back to Basics - Phylogenetic Networks",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/evolution/tutorials/abc_intro_phylo-networks/slides.html"
  },
  "/topics/fair/tutorials/med-chem-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Data management in Medicinal Chemistry",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data-management\" href=\"/training-material/search?query=data-management\" style=\"--color-primary: #98daff; --color-darker: #32c8ff; --color-dimmed: #72d1ff;\">data-management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged medicinal-chemistry\" href=\"/training-material/search?query=medicinal-chemistry\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">medicinal-chemistry</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged computational-chemistry\" href=\"/training-material/search?query=computational-chemistry\" style=\"--color-primary: #00f07e; --color-darker: #00d670; --color-dimmed: #00e377;\">computational-chemistry</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/med-chem-data/tutorial.html"
  },
  "/topics/fair/tutorials/earth_system_rocrate/slides.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "An earth-system RO-Crate",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged earth-system\" href=\"/training-material/search?query=earth-system\" style=\"--color-primary: #a5d7ff; --color-darker: #60c4ff; --color-dimmed: #87ceff;\">earth-system</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ro-crate\" href=\"/training-material/search?query=ro-crate\" style=\"--color-primary: #f7fd00; --color-darker: #dbe100; --color-dimmed: #e9ef00;\">ro-crate</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ocean\" href=\"/training-material/search?query=ocean\" style=\"--color-primary: #00e7ff; --color-darker: #00cde3; --color-dimmed: #00daf1;\">ocean</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged marine omics\" href=\"/training-material/search?query=marine omics\" style=\"--color-primary: #c7cfff; --color-darker: #a8b7ff; --color-dimmed: #b8c3ff;\">marine omics</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/earth_system_rocrate/slides.html"
  },
  "/topics/fair/tutorials/fair-clinical/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Making clinical datasets FAIR",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged open\" href=\"/training-material/search?query=open\" style=\"--color-primary: #ffbcf9; --color-darker: #ff94f6; --color-dimmed: #ffa9f7;\">open</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-clinical/tutorial.html"
  },
  "/topics/fair/tutorials/ro-crate-workflow-run-ro-crate/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Workflow Run RO-Crate Introduction",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ro-crate\" href=\"/training-material/search?query=ro-crate\" style=\"--color-primary: #f7fd00; --color-darker: #dbe100; --color-dimmed: #e9ef00;\">ro-crate</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/ro-crate-workflow-run-ro-crate/tutorial.html"
  },
  "/topics/fair/tutorials/fair-origin/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "FAIR and its Origins",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-origin/tutorial.html"
  },
  "/topics/fair/tutorials/fair-ena/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Sequence data submission to ENA",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged FAIR\" href=\"/training-material/search?query=FAIR\" style=\"--color-primary: #d4cbff; --color-darker: #beb0ff; --color-dimmed: #c9beff;\">FAIR</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ENA\" href=\"/training-material/search?query=ENA\" style=\"--color-primary: #00eae2; --color-darker: #00d0c9; --color-dimmed: #00ddd6;\">ENA</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-ena/tutorial.html"
  },
  "/topics/fair/tutorials/data-management-peatlands/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Introduction to Data Management Plan (DMP) for Peatland Research and PeatDataHub",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged open\" href=\"/training-material/search?query=open\" style=\"--color-primary: #ffbcf9; --color-darker: #ff94f6; --color-dimmed: #ffa9f7;\">open</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management plan\" href=\"/training-material/search?query=data management plan\" style=\"--color-primary: #00e9e7; --color-darker: #00cfce; --color-dimmed: #00dcdb;\">data management plan</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/data-management-peatlands/tutorial.html"
  },
  "/topics/fair/tutorials/fair-intro/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "FAIR in a nutshell",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged open\" href=\"/training-material/search?query=open\" style=\"--color-primary: #ffbcf9; --color-darker: #ff94f6; --color-dimmed: #ffa9f7;\">open</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-intro/tutorial.html"
  },
  "/topics/fair/tutorials/ro-crate-intro/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "RO-Crate - Introduction",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/fair/tutorials/ro-crate-intro/tutorial.html"
  },
  "/topics/fair/tutorials/dataplant-arcs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Plants",
    "title": "DataPLANT ARCs",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/dataplant-arcs/tutorial.html"
  },
  "/topics/fair/tutorials/ro-crate-in-python/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "RO-Crate in Python",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ro-crate\" href=\"/training-material/search?query=ro-crate\" style=\"--color-primary: #f7fd00; --color-darker: #dbe100; --color-dimmed: #e9ef00;\">ro-crate</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/ro-crate-in-python/tutorial.html"
  },
  "/topics/fair/tutorials/data-management/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "FAIR data management solutions",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/data-management/tutorial.html"
  },
  "/topics/fair/tutorials/fair-metadata/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Metadata",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-metadata/tutorial.html"
  },
  "/topics/fair/tutorials/ro-crate-in-galaxy/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Exporting Workflow Run RO-Crates from Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ro-crate\" href=\"/training-material/search?query=ro-crate\" style=\"--color-primary: #f7fd00; --color-darker: #dbe100; --color-dimmed: #e9ef00;\">ro-crate</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/ro-crate-in-galaxy/tutorial.html"
  },
  "/topics/fair/tutorials/ro-crate-galaxy-best-practices/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Best practices for workflows in GitHub repositories",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ro-crate\" href=\"/training-material/search?query=ro-crate\" style=\"--color-primary: #f7fd00; --color-darker: #dbe100; --color-dimmed: #e9ef00;\">ro-crate</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/ro-crate-galaxy-best-practices/tutorial.html"
  },
  "/topics/fair/tutorials/fair-data-registration/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Data Registration",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-data-registration/tutorial.html"
  },
  "/topics/fair/tutorials/fair-gtn/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "FAIR Galaxy Training Material",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gtn\" href=\"/training-material/search?query=gtn\" style=\"--color-primary: #fff5be; --color-darker: #efdb00; --color-dimmed: #fee800;\">gtn</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged training\" href=\"/training-material/search?query=training\" style=\"--color-primary: #ffbdf5; --color-darker: #ff95f0; --color-dimmed: #ffaaf3;\">training</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-gtn/tutorial.html"
  },
  "/topics/fair/tutorials/bioimage-metadata/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "FAIR Bioimage Metadata",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimaging\" href=\"/training-material/search?query=bioimaging\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">bioimaging</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/bioimage-metadata/tutorial.html"
  },
  "/topics/fair/tutorials/ro-crate-submitting-life-monitor/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Submitting workflows to LifeMonitor",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ro-crate\" href=\"/training-material/search?query=ro-crate\" style=\"--color-primary: #f7fd00; --color-darker: #dbe100; --color-dimmed: #e9ef00;\">ro-crate</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/ro-crate-submitting-life-monitor/tutorial.html"
  },
  "/topics/fair/tutorials/fair-access/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Access",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-access/tutorial.html"
  },
  "/topics/fair/tutorials/fair-persistent-identifiers/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "Persistent Identifiers",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-persistent-identifiers/tutorial.html"
  },
  "/topics/fair/tutorials/bioimage-REMBI/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "REMBI - Recommended Metadata for Biological Images – metadata guidelines for bioimaging data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimaging\" href=\"/training-material/search?query=bioimaging\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">bioimaging</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/bioimage-REMBI/tutorial.html"
  },
  "/topics/fair/tutorials/fair-rna/tutorial.html": {
    "type": "Tutorial",
    "topic": "FAIR Data, Workflows, and Research",
    "title": "FAIRification of an RNAseq dataset",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dmp\" href=\"/training-material/search?query=dmp\" style=\"--color-primary: #00e9ec; --color-darker: #00cfd2; --color-dimmed: #00dcdf;\">dmp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data stewardship\" href=\"/training-material/search?query=data stewardship\" style=\"--color-primary: #70e0ff; --color-darker: #00ccef; --color-dimmed: #00d8fd;\">data stewardship</a>"
    ],
    "url": "/training-material/topics/fair/tutorials/fair-rna/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/intermine/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "InterMine integration with Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged intermine\" href=\"/training-material/search?query=intermine\" style=\"--color-primary: #ffc3c0; --color-darker: #ffa19c; --color-dimmed: #ffb3ae;\">intermine</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/intermine/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Creating, Editing and Importing Galaxy Workflows",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Using Workflow Parameters",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-parameters/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Workflow Reports",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-reports/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/get-data/slides.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Getting data into Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/get-data/slides.html"
  },
  "/topics/galaxy-interface/tutorials/galaxy-on-dockstore/slides.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Galaxy workflows in Dockstore",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/galaxy-on-dockstore/slides.html"
  },
  "/topics/galaxy-interface/tutorials/download-delete-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Downloading and Deleting Data in Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/download-delete-data/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Extracting Workflows from Histories",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Rule Based Uploader: Advanced",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?query=tags\" style=\"--color-primary: #ffc1d2; --color-darker: #ff9ebc; --color-dimmed: #ffb0c7;\">tags</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-rules-advanced/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/history/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Understanding Galaxy history system",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/history/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/rstudio/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "RStudio in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/rstudio/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/group-tags/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Group tags for complex experimental designs",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/group-tags/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "JupyterLab in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/collections/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Using dataset collections",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">collections</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/search/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Searching Your History",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/search/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/galaxy-intro-jupyter/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Use Jupyter notebooks in Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/galaxy-intro-jupyter/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/upload-rules/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Rule Based Uploader",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tags\" href=\"/training-material/search?query=tags\" style=\"--color-primary: #ffc1d2; --color-darker: #ff9ebc; --color-dimmed: #ffb0c7;\">tags</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/upload-rules/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-posters/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Creating high resolution images of Galaxy Workflows",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged workflows\" href=\"/training-material/search?query=workflows\" style=\"--color-primary: #ffc69e; --color-darker: #ffa74c; --color-dimmed: #ffb67c;\">workflows</a>"
    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-posters/tutorial.html"
  },
  "/topics/galaxy-interface/tutorials/name-tags/tutorial.html": {
    "type": "Tutorial",
    "topic": "Using Galaxy and Managing your Data",
    "title": "Name tags for following complex histories",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Genome annotation with Prokka",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Bacterial Genome Annotation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?query=illumina\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bacteria\" href=\"/training-material/search?query=bacteria\" style=\"--color-primary: #ffc973; --color-darker: #f0af00; --color-dimmed: #feba00;\">bacteria</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/bacterial-genome-annotation/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/apollo/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Refining Genome Annotations with Apollo (prokaryotes)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged apollo2\" href=\"/training-material/search?query=apollo2\" style=\"--color-primary: #bcd2ff; --color-darker: #94bbff; --color-dimmed: #a9c7ff;\">apollo2</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/apollo/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Secondary metabolite discovery",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/secondary-metabolite-discovery/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/repeatmasker/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Masking repeats with RepeatMasker",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html": {
    "type": "Tutorial",
    "topic": "One Health",
    "title": "Identification of AMR genes in an assembled bacterial genome",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?query=illumina\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?query=amr\" style=\"--color-primary: #c1d1ff; --color-darker: #9eb9ff; --color-dimmed: #afc5ff;\">amr</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/amr-gene-detection/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/helixer/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome annotation with Helixer",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryota\" href=\"/training-material/search?query=eukaryota\" style=\"--color-primary: #a9e400; --color-darker: #96cb00; --color-dimmed: #a0d700;\">eukaryota</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged helixer\" href=\"/training-material/search?query=helixer\" style=\"--color-primary: #bcd2ff; --color-darker: #94bbff; --color-dimmed: #a9c7ff;\">helixer</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/helixer/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html": {
    "type": "Tutorial",
    "topic": "Plants",
    "title": "From small to large-scale genome comparison",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/hpc-for-lsgc/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Introduction to Genome Annotation",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/tutorials/introduction/slides.html"
  },
  "/topics/genome-annotation/tutorials/official-gene-set/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Creating an Official Gene Set",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged apollo2\" href=\"/training-material/search?query=apollo2\" style=\"--color-primary: #bcd2ff; --color-darker: #94bbff; --color-dimmed: #a9c7ff;\">apollo2</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/official-gene-set/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/genome-annotation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Genome Annotation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/genome-annotation/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/gene-centric/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Comparative gene analysis in unannotated genomes",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged evolution\" href=\"/training-material/search?query=evolution\" style=\"--color-primary: #ffc0db; --color-darker: #ff9cc9; --color-dimmed: #ffaed2;\">evolution</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged annotation\" href=\"/training-material/search?query=annotation\" style=\"--color-primary: #73ed00; --color-darker: #65d300; --color-dimmed: #6ce000;\">annotation</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged vgp\" href=\"/training-material/search?query=vgp\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">vgp</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cookbook\" href=\"/training-material/search?query=cookbook\" style=\"--color-primary: #00edb8; --color-darker: #00d3a3; --color-dimmed: #00e0ad;\">cookbook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/gene-centric/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/funannotate/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Genome annotation with Funannotate",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Genome annotation with Maker (short)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged maker\" href=\"/training-material/search?query=maker\" style=\"--color-primary: #ffbeeb; --color-darker: #ff98e2; --color-dimmed: #fface6;\">maker</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-maker-short/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/tnseq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Essential genes detection with Transposon insertion sequencing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bacteria\" href=\"/training-material/search?query=bacteria\" style=\"--color-primary: #ffc973; --color-darker: #f0af00; --color-dimmed: #feba00;\">bacteria</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tnseq\" href=\"/training-material/search?query=tnseq\" style=\"--color-primary: #ffc3bc; --color-darker: #ffa295; --color-dimmed: #ffb3a9;\">tnseq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged essential genes\" href=\"/training-material/search?query=essential genes\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">essential genes</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/tnseq/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Genome annotation with Maker",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged maker\" href=\"/training-material/search?query=maker\" style=\"--color-primary: #ffbeeb; --color-darker: #ff98e2; --color-dimmed: #fface6;\">maker</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/functional/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Functional annotation of protein sequences",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/functional/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/apollo-euk/tutorial.html": {
    "type": "Tutorial",
    "topic": "GMOD",
    "title": "Refining Genome Annotations with Apollo (eukaryotes)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged apollo2\" href=\"/training-material/search?query=apollo2\" style=\"--color-primary: #bcd2ff; --color-darker: #94bbff; --color-dimmed: #a9c7ff;\">apollo2</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/lncrna/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "Long non-coding RNAs (lncRNAs) annotation with FEELnc",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged eukaryote\" href=\"/training-material/search?query=eukaryote\" style=\"--color-primary: #53f000; --color-darker: #49d500; --color-dimmed: #4ee200;\">eukaryote</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/lncrna/tutorial.html"
  },
  "/topics/genome-annotation/tutorials/crispr-screen/tutorial.html": {
    "type": "Tutorial",
    "topic": "Genome Annotation",
    "title": "CRISPR screen analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged genome editing\" href=\"/training-material/search?query=genome editing\" style=\"--color-primary: #65e1ff; --color-darker: #00ccec; --color-dimmed: #00d8fa;\">genome editing</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged CRISPR\" href=\"/training-material/search?query=CRISPR\" style=\"--color-primary: #00e8f4; --color-darker: #00ced9; --color-dimmed: #00dbe7;\">CRISPR</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged essential genes\" href=\"/training-material/search?query=essential genes\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">essential genes</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cell line\" href=\"/training-material/search?query=cell line\" style=\"--color-primary: #00f174; --color-darker: #00d667; --color-dimmed: #00e36d;\">cell line</a>"
    ],
    "url": "/training-material/topics/genome-annotation/tutorials/crispr-screen/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/microbial-variants/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Microbial Variant Calling",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/microbial-variants/tutorial.html"
  },
  "/topics/visualisation/tutorials/jbrowse/tutorial.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Genomic Data Visualisation with JBrowse",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged gmod\" href=\"/training-material/search?query=gmod\" style=\"--color-primary: #ffca4e; --color-darker: #e8b300; --color-dimmed: #f6be00;\">gmod</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jbrowse1\" href=\"/training-material/search?query=jbrowse1\" style=\"--color-primary: #f1c1ff; --color-darker: #eb9eff; --color-dimmed: #eeb0ff;\">jbrowse1</a>"
    ],
    "url": "/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html"
  },
  "/topics/imaging/tutorials/detection-of-mitoflashes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Tracking of mitochondria and capturing mitoflashes",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimaging\" href=\"/training-material/search?query=bioimaging\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">bioimaging</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mitoflash\" href=\"/training-material/search?query=mitoflash\" style=\"--color-primary: #fecc00; --color-darker: #e2b500; --color-dimmed: #f0c000;\">mitoflash</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mitochondria\" href=\"/training-material/search?query=mitochondria\" style=\"--color-primary: #5def00; --color-darker: #52d500; --color-dimmed: #57e200;\">mitochondria</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/detection-of-mitoflashes/tutorial.html"
  },
  "/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "End-to-End Tissue Microarray Image Analysis with Galaxy-ME",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/multiplex-tissue-imaging-TMA/tutorial.html"
  },
  "/topics/imaging/tutorials/omero-suite/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Overview of the Galaxy OMERO-suite - Upload images and metadata in OMERO using Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/omero-suite/tutorial.html"
  },
  "/topics/imaging/tutorials/imaging-introduction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Introduction to Image Analysis using Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #c8cfff; --color-darker: #aab6ff; --color-dimmed: #b9c3ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/imaging-introduction/tutorial.html"
  },
  "/topics/imaging/tutorials/hela-screen-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Analyse HeLa fluorescence siRNA screen",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #c8cfff; --color-darker: #aab6ff; --color-dimmed: #b9c3ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/hela-screen-analysis/tutorial.html"
  },
  "/topics/imaging/tutorials/tutorial-CP/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Nucleoli segmentation and feature extraction using CellProfiler",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html"
  },
  "/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "Object tracking using CellProfiler",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/imaging/tutorials/object-tracking-using-cell-profiler/tutorial.html"
  },
  "/topics/imaging/tutorials/bioimage-metadata/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "FAIR Bioimage Metadata",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimaging\" href=\"/training-material/search?query=bioimaging\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">bioimaging</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/bioimage-metadata/tutorial.html"
  },
  "/topics/imaging/tutorials/bioimage-REMBI/tutorial.html": {
    "type": "Tutorial",
    "topic": "Imaging",
    "title": "REMBI - Recommended Metadata for Biological Images – metadata guidelines for bioimaging data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fair\" href=\"/training-material/search?query=fair\" style=\"--color-primary: #bee000; --color-darker: #a9c700; --color-dimmed: #b3d300;\">fair</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bioimaging\" href=\"/training-material/search?query=bioimaging\" style=\"--color-primary: #00f168; --color-darker: #00d75c; --color-dimmed: #00e462;\">bioimaging</a>"
    ],
    "url": "/training-material/topics/imaging/tutorials/bioimage-REMBI/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "From peaks to genes",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html"
  },
  "/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Data Manipulation Olympics",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-reproduce/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "How to reproduce published Galaxy analyses",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-reproduce/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Introduction to Genomics and Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-strands/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "NGS data logistics",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html"
  },
  "/topics/introduction/tutorials/options-for-using-galaxy/slides.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Options for using Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/options-for-using-galaxy/slides.html"
  },
  "/topics/introduction/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Introduction to Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/introduction/slides.html"
  },
  "/topics/introduction/tutorials/dummy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Upload data to Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/dummy/tutorial.html"
  },
  "/topics/introduction/tutorials/igv-introduction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "IGV Introduction",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "A short introduction to Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged español\" href=\"/training-material/search?query=español\" style=\"--color-primary: #00eadf; --color-darker: #00d0c6; --color-dimmed: #00ddd2;\">español</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-short/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Galaxy Basics for genomics",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Galaxy Basics for everyone",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html"
  },
  "/topics/introduction/tutorials/vsi_qc/tutorial.html": {
    "type": "Tutorial",
    "topic": "Introduction to Galaxy Analyses",
    "title": "Very Short Introductions: QC",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/introduction/tutorials/vsi_qc/tutorial.html"
  },
  "/topics/materials-science/tutorials/muon-stopping-sites-muairss-uep/tutorial.html": {
    "type": "Tutorial",
    "topic": "Materials Science",
    "title": "Finding the muon stopping site with pymuon-suite in Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/materials-science/tutorials/muon-stopping-sites-muairss-uep/tutorial.html"
  },
  "/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry imaging: Finding differential analytes",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/msi-finding-nglycans/tutorial.html"
  },
  "/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Predicting EI+ mass spectra with QCxMS",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/lcms/tutorial.html"
  },
  "/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html"
  },
  "/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry imaging: Examining the spatial distribution of analytes",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/msi-analyte-distribution/tutorial.html"
  },
  "/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS preprocessing with XCMS",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html"
  },
  "/topics/metabolomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Introduction to Metabolomics",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/introduction/slides.html"
  },
  "/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry: LC-MS data processing",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/metabolomics/tutorials/lcms-dataprocessing/tutorial.html"
  },
  "/topics/metabolomics/tutorials/gcms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Metabolomics",
    "title": "Mass spectrometry : GC-MS analysis with metaMS package",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/metabolomics/tutorials/gcms/tutorial.html"
  },
  "/topics/microbiome/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Introduction to Microbiome Analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/introduction/slides.html"
  },
  "/topics/microbiome/tutorials/general-tutorial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Analyses of metagenomics data - The global picture",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/general-tutorial/tutorial.html"
  },
  "/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Metatranscriptomics analysis using microbiome RNA-seq data (short)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?query=metatranscriptomics\" style=\"--color-primary: #ecc3ff; --color-darker: #e4a1ff; --color-dimmed: #e8b2ff;\">metatranscriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/metatranscriptomics-short/tutorial.html"
  },
  "/topics/microbiome/tutorials/beer-data-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Identification of the micro-organisms in a beer using Nanopore sequencing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged beer\" href=\"/training-material/search?query=beer\" style=\"--color-primary: #00e9e7; --color-darker: #00cfce; --color-dimmed: #00dcdb;\">beer</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged citizen science\" href=\"/training-material/search?query=citizen science\" style=\"--color-primary: #39f100; --color-darker: #32d700; --color-dimmed: #36e400;\">citizen science</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/beer-data-analysis/tutorial.html"
  },
  "/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Antibiotic resistance detection",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plasmids\" href=\"/training-material/search?query=plasmids\" style=\"--color-primary: #00edb4; --color-darker: #00d3a0; --color-dimmed: #00e0aa;\">plasmids</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged amr\" href=\"/training-material/search?query=amr\" style=\"--color-primary: #c1d1ff; --color-darker: #9eb9ff; --color-dimmed: #afc5ff;\">amr</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/plasmid-metagenomics-nanopore/tutorial.html"
  },
  "/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Plants",
    "title": "16S Microbial analysis with Nanopore data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metabarcoding\" href=\"/training-material/search?query=metabarcoding\" style=\"--color-primary: #c9cfff; --color-darker: #acb6ff; --color-dimmed: #bbc2ff;\">metabarcoding</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 16S\" href=\"/training-material/search?query=16S\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">16S</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged nanopore\" href=\"/training-material/search?query=nanopore\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">nanopore</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/nanopore-16S-metagenomics/tutorial.html"
  },
  "/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Taxonomic Profiling and Visualization of Metagenomic Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged taxonomic profiling\" href=\"/training-material/search?query=taxonomic profiling\" style=\"--color-primary: #ffbdf3; --color-darker: #ff96ed; --color-dimmed: #ffaaf0;\">taxonomic profiling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/taxonomic-profiling/tutorial.html"
  },
  "/topics/microbiome/tutorials/lotus2-identifying-fungi/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Identifying Mycorrhizal Fungi from ITS2 sequencing using LotuS2",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fungi\" href=\"/training-material/search?query=fungi\" style=\"--color-primary: #ffc0d8; --color-darker: #ff9cc6; --color-dimmed: #ffafcf;\">fungi</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ecology\" href=\"/training-material/search?query=ecology\" style=\"--color-primary: #ffbcf9; --color-darker: #ff94f6; --color-dimmed: #ffa9f7;\">ecology</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/lotus2-identifying-fungi/tutorial.html"
  },
  "/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Clinical Metaproteomics 2: Discovery",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/clinical-mp-2-discovery/tutorial.html"
  },
  "/topics/microbiome/tutorials/dada-16S/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Building an amplicon sequence variant (ASV) table from 16S data using DADA2",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metabarcoding\" href=\"/training-material/search?query=metabarcoding\" style=\"--color-primary: #c9cfff; --color-darker: #acb6ff; --color-dimmed: #bbc2ff;\">metabarcoding</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 16S\" href=\"/training-material/search?query=16S\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">16S</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/dada-16S/tutorial.html"
  },
  "/topics/microbiome/tutorials/metatranscriptomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Metatranscriptomics analysis using microbiome RNA-seq data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?query=metatranscriptomics\" style=\"--color-primary: #ecc3ff; --color-darker: #e4a1ff; --color-dimmed: #e8b2ff;\">metatranscriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/metatranscriptomics/tutorial.html"
  },
  "/topics/microbiome/tutorials/clinical-mp-5-data-interpretation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Clinical Metaproteomics 5: Data Interpretation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/clinical-mp-5-data-interpretation/tutorial.html"
  },
  "/topics/microbiome/tutorials/clinical-mp-1-database-generation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Clinical Metaproteiomics 1: Database-Generation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/clinical-mp-1-database-generation/tutorial.html"
  },
  "/topics/microbiome/tutorials/clinical-mp-4-quantitation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Clinical Metaproteomics 4: Quantitation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/clinical-mp-4-quantitation/tutorial.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Nanopore data analysis\" href=\"/training-material/search?query=Nanopore data analysis\" style=\"--color-primary: #00e9ea; --color-darker: #00cfd0; --color-dimmed: #00dcdd;\">Nanopore data analysis</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Pathogens detection\" href=\"/training-material/search?query=Pathogens detection\" style=\"--color-primary: #a6e500; --color-darker: #93cb00; --color-dimmed: #9dd800;\">Pathogens detection</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Phylogenetic tree\" href=\"/training-material/search?query=Phylogenetic tree\" style=\"--color-primary: #00f17a; --color-darker: #00d66c; --color-dimmed: #00e372;\">Phylogenetic tree</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Heatmap\" href=\"/training-material/search?query=Heatmap\" style=\"--color-primary: #ffbee7; --color-darker: #ff99dc; --color-dimmed: #fface1;\">Heatmap</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html"
  },
  "/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Clinical Metaproteomics 3: Verification",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/clinical-mp-3-verification/tutorial.html"
  },
  "/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Assembly of metagenomic sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged assembly\" href=\"/training-material/search?query=assembly\" style=\"--color-primary: #ffbfe2; --color-darker: #ff9ad4; --color-dimmed: #ffaddb;\">assembly</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/metagenomics-assembly/tutorial.html"
  },
  "/topics/microbiome/tutorials/diversity/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Calculating α and β diversity from microbiome taxonomic data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged diversity\" href=\"/training-material/search?query=diversity\" style=\"--color-primary: #00eadf; --color-darker: #00d0c6; --color-dimmed: #00ddd2;\">diversity</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/diversity/tutorial.html"
  },
  "/topics/microbiome/tutorials/metagenomics-binning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "Binning of metagenomic sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged binning\" href=\"/training-material/search?query=binning\" style=\"--color-primary: #ffbfe4; --color-darker: #ff9ad8; --color-dimmed: #ffadde;\">binning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/metagenomics-binning/tutorial.html"
  },
  "/topics/microbiome/tutorials/qiime2-moving-pictures/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "QIIME 2 Moving Pictures",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metagenomics\" href=\"/training-material/search?query=metagenomics\" style=\"--color-primary: #edd200; --color-darker: #d3bb00; --color-dimmed: #e0c600;\">metagenomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged taxonomic profiling\" href=\"/training-material/search?query=taxonomic profiling\" style=\"--color-primary: #ffbdf3; --color-darker: #ff96ed; --color-dimmed: #ffaaf0;\">taxonomic profiling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged diversity\" href=\"/training-material/search?query=diversity\" style=\"--color-primary: #00eadf; --color-darker: #00d0c6; --color-dimmed: #00ddd2;\">diversity</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/qiime2-moving-pictures/tutorial.html"
  },
  "/topics/microbiome/tutorials/qiime2-cancer-microbiome-intervention/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "QIIME 2 Cancer Microbiome Intervention",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metabarcoding\" href=\"/training-material/search?query=metabarcoding\" style=\"--color-primary: #c9cfff; --color-darker: #acb6ff; --color-dimmed: #bbc2ff;\">metabarcoding</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 16S\" href=\"/training-material/search?query=16S\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">16S</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged taxonomic profiling\" href=\"/training-material/search?query=taxonomic profiling\" style=\"--color-primary: #ffbdf3; --color-darker: #ff96ed; --color-dimmed: #ffaaf0;\">taxonomic profiling</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged diversity\" href=\"/training-material/search?query=diversity\" style=\"--color-primary: #00eadf; --color-darker: #00d0c6; --color-dimmed: #00ddd2;\">diversity</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/qiime2-cancer-microbiome-intervention/tutorial.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "16S Microbial Analysis with mothur (extended)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metabarcoding\" href=\"/training-material/search?query=metabarcoding\" style=\"--color-primary: #c9cfff; --color-darker: #acb6ff; --color-dimmed: #bbc2ff;\">metabarcoding</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 16S\" href=\"/training-material/search?query=16S\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">16S</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop/tutorial.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html": {
    "type": "Tutorial",
    "topic": "Microbiome",
    "title": "16S Microbial Analysis with mothur (short)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metabarcoding\" href=\"/training-material/search?query=metabarcoding\" style=\"--color-primary: #c9cfff; --color-darker: #acb6ff; --color-dimmed: #bbc2ff;\">metabarcoding</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 16S\" href=\"/training-material/search?query=16S\" style=\"--color-primary: #ffc0de; --color-darker: #ff9bcf; --color-dimmed: #ffaed6;\">16S</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Pox virus genome analysis from tiled-amplicon sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc0e0; --color-darker: #ff9bd1; --color-dimmed: #ffadd8;\">virology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/pox-tiled-amplicon/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "M. tuberculosis Variant Analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged tuberculosis\" href=\"/training-material/search?query=tuberculosis\" style=\"--color-primary: #65e1ff; --color-darker: #00ccec; --color-dimmed: #00d8fa;\">tuberculosis</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Avian influenza viral strain analysis from gene segment sequencing data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged virology\" href=\"/training-material/search?query=virology\" style=\"--color-primary: #ffc0e0; --color-darker: #ff9bd1; --color-dimmed: #ffadd8;\">virology</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged one-health\" href=\"/training-material/search?query=one-health\" style=\"--color-primary: #95e800; --color-darker: #84ce00; --color-dimmed: #8ddb00;\">one-health</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/aiv-analysis/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Whole transcriptome analysis of Arabidopsis thaliana",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged miRNA\" href=\"/training-material/search?query=miRNA\" style=\"--color-primary: #4fe3ff; --color-darker: #00cde8; --color-dimmed: #00d9f6;\">miRNA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged stress tolerance\" href=\"/training-material/search?query=stress tolerance\" style=\"--color-primary: #00f17a; --color-darker: #00d66c; --color-dimmed: #00e372;\">stress tolerance</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "NCBI BLAST+ against the MAdLand",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>"
    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/ncbi-blast-against-the-madland/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-plant/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Analysis of plant scRNA-Seq Data with Scanpy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged plants\" href=\"/training-material/search?query=plants\" style=\"--color-primary: #6ced00; --color-darker: #60d300; --color-dimmed: #66e000;\">plants</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-plant/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Statistical analysis of DIA data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #5fe2ff; --color-darker: #00cceb; --color-dimmed: #00d9f9;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/DIA_Analysis_MSstats/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Metaproteomics tutorial",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html"
  },
  "/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide Library Data Analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/peptide-library-data-analysis/tutorial.html"
  },
  "/topics/proteomics/tutorials/secretome-prediction/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Secretome Prediction",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffc3bf; --color-darker: #ffa19a; --color-dimmed: #ffb3ad;\">human</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/secretome-prediction/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "DIA Analysis using OpenSwathWorkflow",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #5fe2ff; --color-darker: #00cceb; --color-dimmed: #00d9f9;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-id-oms/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide and Protein ID using OpenMS tools",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #c8cfff; --color-darker: #aab6ff; --color-dimmed: #b9c3ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Biomarker candidate identification",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffc3bf; --color-darker: #ffa19a; --color-dimmed: #ffb3ad;\">human</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "metaQuantome 3: Taxonomy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-taxonomy/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-label-free/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Label-free data analysis using MaxQuant",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-free\" href=\"/training-material/search?query=label-free\" style=\"--color-primary: #c7cfff; --color-darker: #a8b7ff; --color-dimmed: #b8c3ff;\">label-free</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-label-free/tutorial.html"
  },
  "/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Library Generation for DIA Analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #5fe2ff; --color-darker: #00cceb; --color-dimmed: #00d9f9;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html"
  },
  "/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Clinical Metaproteomics 2: Discovery",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/clinical-mp-2-discovery/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Proteogenomics 3: Novel peptide analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?query=proteogenomics\" style=\"--color-primary: #00f087; --color-darker: #00d678; --color-dimmed: #00e380;\">proteogenomics</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html"
  },
  "/topics/proteomics/tutorials/database-handling/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Protein FASTA Database Handling",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/database-handling/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-function/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "metaQuantome 2: Function",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-function/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "MaxQuant and MSstats for the analysis of label-free data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-free\" href=\"/training-material/search?query=label-free\" style=\"--color-primary: #c7cfff; --color-darker: #a8b7ff; --color-dimmed: #b8c3ff;\">label-free</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/tutorial.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "MaxQuant and MSstats for the analysis of TMT data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged TMT\" href=\"/training-material/search?query=TMT\" style=\"--color-primary: #ded700; --color-darker: #c5bf00; --color-dimmed: #d1cb00;\">TMT</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-tmt/tutorial.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "EncyclopeDIA",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DIA\" href=\"/training-material/search?query=DIA\" style=\"--color-primary: #5fe2ff; --color-darker: #00cceb; --color-dimmed: #00d9f9;\">DIA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/tutorial.html"
  },
  "/topics/proteomics/tutorials/ntails/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Detection and quantitation of N-termini (degradomics) via N-TAILS",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/ntails/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide and Protein ID using SearchGUI and PeptideShaker",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged HeLa\" href=\"/training-material/search?query=HeLa\" style=\"--color-primary: #c8cfff; --color-darker: #aab6ff; --color-dimmed: #b9c3ff;\">HeLa</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html"
  },
  "/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Clinical Metaproteomics 5: Data Interpretation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/clinical-mp-5-data-interpretation/tutorial.html"
  },
  "/topics/proteomics/tutorials/protein-quant-sil/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Peptide and Protein Quantification via Stable Isotope Labelling (SIL)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged SILAC\" href=\"/training-material/search?query=SILAC\" style=\"--color-primary: #cddc00; --color-darker: #b6c400; --color-dimmed: #c2d000;\">SILAC</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Proteogenomics 2: Database Search",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?query=proteogenomics\" style=\"--color-primary: #00f087; --color-darker: #00d678; --color-dimmed: #00e380;\">proteogenomics</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Proteogenomics 1: Database Creation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged proteogenomics\" href=\"/training-material/search?query=proteogenomics\" style=\"--color-primary: #00f087; --color-darker: #00d678; --color-dimmed: #00e380;\">proteogenomics</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html"
  },
  "/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Mass spectrometry imaging: Loading and exploring MSI data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00ecc2; --color-darker: #00d2ad; --color-dimmed: #00dfb7;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged imaging\" href=\"/training-material/search?query=imaging\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">imaging</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/mass-spectrometry-imaging-loading-exploring-data/tutorial.html"
  },
  "/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Clinical Metaproteiomics 1: Database-Generation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/clinical-mp-1-database-generation/tutorial.html"
  },
  "/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Clinical Metaproteomics 4: Quantitation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/clinical-mp-4-quantitation/tutorial.html"
  },
  "/topics/proteomics/tutorials/proteome_annotation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Annotating a protein list identified by LC-MS/MS experiments",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged human\" href=\"/training-material/search?query=human\" style=\"--color-primary: #ffc3bf; --color-darker: #ffa19a; --color-dimmed: #ffb3ad;\">human</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/proteome_annotation/tutorial.html"
  },
  "/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Clinical Metaproteomics 3: Verification",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged label-TMT11\" href=\"/training-material/search?query=label-TMT11\" style=\"--color-primary: #ffbfe5; --color-darker: #ff9ad9; --color-dimmed: #ffaddf;\">label-TMT11</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/clinical-mp-3-verification/tutorial.html"
  },
  "/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "metaQuantome 1: Data creation",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/metaquantome-data-creation/tutorial.html"
  },
  "/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Machine Learning Modeling of Anticancer Peptides",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged ML\" href=\"/training-material/search?query=ML\" style=\"--color-primary: #d9caff; --color-darker: #c7adff; --color-dimmed: #d0bbff;\">ML</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cancer\" href=\"/training-material/search?query=cancer\" style=\"--color-primary: #ffbfe1; --color-darker: #ff9bd3; --color-dimmed: #ffadda;\">cancer</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/ml-modeling-of-anti-cancer-peptides/tutorial.html"
  },
  "/topics/proteomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Introduction to proteomics, protein identification, quantification and statistical modelling",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/introduction/slides.html"
  },
  "/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html": {
    "type": "Tutorial",
    "topic": "Proteomics",
    "title": "Label-free versus Labelled - How to Choose Your Quantitation Method",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged DDA\" href=\"/training-material/search?query=DDA\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">DDA</a>"
    ],
    "url": "/training-material/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Quality and contamination control in bacterial isolate using Illumina MiSeq Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged illumina\" href=\"/training-material/search?query=illumina\" style=\"--color-primary: #ffc880; --color-darker: #f5ad00; --color-dimmed: #ffb93e;\">illumina</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bacteria\" href=\"/training-material/search?query=bacteria\" style=\"--color-primary: #ffc973; --color-darker: #f0af00; --color-dimmed: #feba00;\">bacteria</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-contamination-control/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/ncbi-fcs/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Screening assembled genomes for contamination using NCBI FCS",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/ncbi-fcs/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Clean and manage Sanger sequences from raw files to aligned consensus",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Mapping",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/tutorial.html": {
    "type": "Tutorial",
    "topic": "Sequence analysis",
    "title": "Quality Control",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"
  },
  "/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Pre-processing of 10X Single-Cell ATAC-seq Datasets",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged epigenetics\" href=\"/training-material/search?query=epigenetics\" style=\"--color-primary: #9ce700; --color-darker: #8acd00; --color-dimmed: #93da00;\">epigenetics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Generating a single cell matrix using Alevin",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html"
  },
  "/topics/single-cell/tutorials/GO-enrichment/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "GO Enrichment Analysis on Single-Cell RNA-Seq Data",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single cell\" href=\"/training-material/search?query=single cell\" style=\"--color-primary: #b4d4ff; --color-darker: #84bfff; --color-dimmed: #9dc9ff;\">single cell</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged GO enrichment\" href=\"/training-material/search?query=GO enrichment\" style=\"--color-primary: #bce000; --color-darker: #a7c700; --color-dimmed: #b1d400;\">GO enrichment</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/GO-enrichment/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Converting between common single cell data formats",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data import\" href=\"/training-material/search?query=data import\" style=\"--color-primary: #ffc3bd; --color-darker: #ffa297; --color-dimmed: #ffb3ab;\">data import</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-data-ingest/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Pre-processing of 10X Single-Cell RNA Datasets",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Bulk RNA Deconvolution with MuSiC",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #e6d400; --color-darker: #cdbd00; --color-dimmed: #dac800;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Pre-processing of Single-Cell RNA Data",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Single-cell quality control with scater",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scater-qc/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-intro/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "An introduction to scRNA-seq data analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-intro/slides.html"
  },
  "/topics/single-cell/tutorials/alevin-commandline/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Generating a single cell matrix using Alevin and combining datasets (bash + R)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/alevin-commandline/tutorial.html"
  },
  "/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Single-cell ATAC-seq standard processing with SnapATAC2",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged epigenetics\" href=\"/training-material/search?query=epigenetics\" style=\"--color-primary: #9ce700; --color-darker: #8acd00; --color-dimmed: #93da00;\">epigenetics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged single-cell\" href=\"/training-material/search?query=single-cell\" style=\"--color-primary: #00ead8; --color-darker: #00d1c0; --color-dimmed: #00decc;\">single-cell</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Clustering 3K PBMCs with Scanpy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Removing the effects of the cell cycle",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-raceid/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Downstream Single-cell RNA analysis with RaceID",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-raceid/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_FilterPlotandExploreRStudio/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Filter, plot, and explore single cell RNA-seq data with Seurat (R)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">rmarkdown-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExploreRStudio/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Combining single cell datasets after pre-processing",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html"
  },
  "/topics/single-cell/tutorials/EBI-retrieval/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Importing files from public atlases",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data import\" href=\"/training-material/search?query=data import\" style=\"--color-primary: #ffc3bd; --color-darker: #ffa297; --color-dimmed: #ffb3ab;\">data import</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case-jupyter_basic-pipeline/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Filter, plot and explore single-cell RNA-seq data with Scanpy (Python)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Python\" href=\"/training-material/search?query=Python\" style=\"--color-primary: #d9caff; --color-darker: #c7adff; --color-dimmed: #d0bbff;\">Python</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case-jupyter_basic-pipeline/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-umis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Understanding Barcodes",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html"
  },
  "/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Scanpy Parameter Iterator",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Converting NCBI Data to the AnnData Format",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data import\" href=\"/training-material/search?query=data import\" style=\"--color-primary: #ffc3bd; --color-darker: #ffa297; --color-dimmed: #ffb3ab;\">data import</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-ncbi-anndata/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Filter, plot and explore single-cell RNA-seq data with Scanpy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Filter, plot, and explore single cell RNA-seq data with Seurat",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_FilterPlotandExplore_SeuratTools/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Plates, Batches, and Barcodes",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html"
  },
  "/topics/single-cell/tutorials/scrna-case_JUPYTER-trajectories/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Inferring single cell trajectories with Scanpy (Python)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Python\" href=\"/training-material/search?query=Python\" style=\"--color-primary: #d9caff; --color-darker: #c7adff; --color-dimmed: #d0bbff;\">Python</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_JUPYTER-trajectories/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Inferring single cell trajectories with Scanpy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_trajectories/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music-4-compare/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Comparing inferred cell compositions using MuSiC deconvolution",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #e6d400; --color-darker: #cdbd00; --color-dimmed: #dac800;\">transcriptomics</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music-4-compare/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Matrix Exchange Format to ESet | Creating a single-cell RNA-seq reference dataset for deconvolution",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music-2-preparescref/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-data-formats/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Single-cell Formats and Resources",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-data-formats/slides.html"
  },
  "/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Inferring single cell trajectories with Monocle3 (R)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged R\" href=\"/training-material/search?query=R\" style=\"--color-primary: #ace400; --color-darker: #99ca00; --color-dimmed: #a2d700;\">R</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rmarkdown-notebook\" href=\"/training-material/search?query=rmarkdown-notebook\" style=\"--color-primary: #dac9ff; --color-darker: #c9acff; --color-dimmed: #d1bbff;\">rmarkdown-notebook</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-notebook\" href=\"/training-material/search?query=jupyter-notebook\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">jupyter-notebook</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_monocle3-rstudio/tutorial.html"
  },
  "/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Bulk matrix to ESet | Creating the bulk RNA-seq dataset for deconvolution",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged transcriptomics\" href=\"/training-material/search?query=transcriptomics\" style=\"--color-primary: #e6d400; --color-darker: #cdbd00; --color-dimmed: #dac800;\">transcriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged data management\" href=\"/training-material/search?query=data management\" style=\"--color-primary: #ffc967; --color-darker: #edb100; --color-dimmed: #fbbc00;\">data management</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/bulk-music-3-preparebulk/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Inferring single cell trajectories with Monocle3",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged 10x\" href=\"/training-material/search?query=10x\" style=\"--color-primary: #ffc2ca; --color-darker: #ff9fae; --color-dimmed: #ffb1bc;\">10x</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged paper-replication\" href=\"/training-material/search?query=paper-replication\" style=\"--color-primary: #00e9e6; --color-darker: #00d0cd; --color-dimmed: #00dcd9;\">paper-replication</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MIGHTS\" href=\"/training-material/search?query=MIGHTS\" style=\"--color-primary: #88ea00; --color-darker: #79d000; --color-dimmed: #81dd00;\">MIGHTS</a>"
    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/tutorial.html"
  },
  "/topics/single-cell/tutorials/scrna-trajectories/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Trajectory analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-trajectories/slides.html"
  },
  "/topics/single-cell/tutorials/scrna-case-cell-annotation/slides.html": {
    "type": "Tutorial",
    "topic": "Single Cell",
    "title": "Automated Cell Annotation",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case-cell-annotation/slides.html"
  },
  "/topics/statistics/tutorials/classification_regression/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Machine learning: classification and regression",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/classification_regression/tutorial.html"
  },
  "/topics/statistics/tutorials/clustering_machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Clustering in Machine Learning",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/clustering_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/text-mining_simtext/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Text-mining with the SimText toolset",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/text-mining_simtext/tutorial.html"
  },
  "/topics/statistics/tutorials/galaxy-ludwig/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Train and Test a Deep learning image classifier with Galaxy-Ludwig",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged MNIST\" href=\"/training-material/search?query=MNIST\" style=\"--color-primary: #ffc1d3; --color-darker: #ff9ebd; --color-dimmed: #ffb0c8;\">MNIST</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Deep learning\" href=\"/training-material/search?query=Deep learning\" style=\"--color-primary: #00f16e; --color-darker: #00d661; --color-dimmed: #00e468;\">Deep learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Ludwig\" href=\"/training-material/search?query=Ludwig\" style=\"--color-primary: #00e7f9; --color-darker: #00cede; --color-dimmed: #00daec;\">Ludwig</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/galaxy-ludwig/tutorial.html"
  },
  "/topics/statistics/tutorials/regression_machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Regression in Machine Learning",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/regression_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/FNN/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 1) - Feedforward neural networks (FNN)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/FNN/tutorial.html"
  },
  "/topics/statistics/tutorials/fruit_360/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Image classification in Galaxy with fruit 360 dataset",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/fruit_360/tutorial.html"
  },
  "/topics/statistics/tutorials/machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Basics of machine learning",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/intro_deep_learning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Introduction to deep learning",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/intro_deep_learning/tutorial.html"
  },
  "/topics/statistics/tutorials/iwtomics/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Interval-Wise Testing for omics data",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/iwtomics/tutorial.html"
  },
  "/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "PAPAA PI3K_OG: PanCancer Aberrant Pathway Activity Analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Machine learning\" href=\"/training-material/search?query=Machine learning\" style=\"--color-primary: #d1ccff; --color-darker: #b9b2ff; --color-dimmed: #c5bfff;\">Machine learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged Pan-cancer\" href=\"/training-material/search?query=Pan-cancer\" style=\"--color-primary: #00eae0; --color-darker: #00d0c7; --color-dimmed: #00ddd3;\">Pan-cancer</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cancer biomarkers\" href=\"/training-material/search?query=cancer biomarkers\" style=\"--color-primary: #00ef92; --color-darker: #00d582; --color-dimmed: #00e28a;\">cancer biomarkers</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged oncogenes and tumor suppressor genes\" href=\"/training-material/search?query=oncogenes and tumor suppressor genes\" style=\"--color-primary: #ffc2ce; --color-darker: #ff9fb5; --color-dimmed: #ffb0c2;\">oncogenes and tumor suppressor genes</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/aberrant_pi3k_pathway_analysis/tutorial.html"
  },
  "/topics/statistics/tutorials/RNN/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 2) - Recurrent neural networks (RNN)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/RNN/tutorial.html"
  },
  "/topics/statistics/tutorials/classification_machinelearning/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Classification in Machine Learning",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/classification_machinelearning/tutorial.html"
  },
  "/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "A Docker-based interactive Jupyterlab powered by GPU for artificial intelligence in Galaxy",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged machine-learning\" href=\"/training-material/search?query=machine-learning\" style=\"--color-primary: #f0d100; --color-darker: #d6ba00; --color-dimmed: #e3c500;\">machine-learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deep-learning\" href=\"/training-material/search?query=deep-learning\" style=\"--color-primary: #ffbeea; --color-darker: #ff98e0; --color-dimmed: #fface5;\">deep-learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-lab\" href=\"/training-material/search?query=jupyter-lab\" style=\"--color-primary: #fff5b1; --color-darker: #ecdc00; --color-dimmed: #fbe900;\">jupyter-lab</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged image-segmentation\" href=\"/training-material/search?query=image-segmentation\" style=\"--color-primary: #5def00; --color-darker: #52d500; --color-dimmed: #57e200;\">image-segmentation</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged protein-3D-structure\" href=\"/training-material/search?query=protein-3D-structure\" style=\"--color-primary: #ffc791; --color-darker: #fda900; --color-dimmed: #ffb863;\">protein-3D-structure</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/gpu_jupyter_lab/tutorial.html"
  },
  "/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Age prediction using machine learning",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/age-prediction-with-ml/tutorial.html"
  },
  "/topics/statistics/tutorials/CNN/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Deep Learning (Part 3) - Convolutional neural networks (CNN)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/CNN/tutorial.html"
  },
  "/topics/statistics/tutorials/fine_tuning_protTrans/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Fine tune large protein model (ProtTrans) using HuggingFace",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged machine-learning\" href=\"/training-material/search?query=machine-learning\" style=\"--color-primary: #f0d100; --color-darker: #d6ba00; --color-dimmed: #e3c500;\">machine-learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged deep-learning\" href=\"/training-material/search?query=deep-learning\" style=\"--color-primary: #ffbeea; --color-darker: #ff98e0; --color-dimmed: #fface5;\">deep-learning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged jupyter-lab\" href=\"/training-material/search?query=jupyter-lab\" style=\"--color-primary: #fff5b1; --color-darker: #ecdc00; --color-dimmed: #fbe900;\">jupyter-lab</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged fine-tuning\" href=\"/training-material/search?query=fine-tuning\" style=\"--color-primary: #00ef92; --color-darker: #00d582; --color-dimmed: #00e28a;\">fine-tuning</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged dephosphorylation-site-prediction\" href=\"/training-material/search?query=dephosphorylation-site-prediction\" style=\"--color-primary: #d4cbff; --color-darker: #beb0ff; --color-dimmed: #c9beff;\">dephosphorylation-site-prediction</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/fine_tuning_protTrans/tutorial.html"
  },
  "/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Introduction to Machine Learning using R",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/statistics/tutorials/intro-to-ml-with-r/tutorial.html"
  },
  "/topics/statistics/tutorials/hyperdimensional_computing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Statistics and machine learning",
    "title": "Supervised Learning with Hyperdimensional Computing",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/statistics/tutorials/hyperdimensional_computing/tutorial.html"
  },
  "/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Synthetic Biology",
    "title": "Generating theoretical possible pathways for the production of Lycopene in E.Coli using Retrosynthesis tools",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/synthetic-biology/tutorials/retrosynthesis_analysis/tutorial.html"
  },
  "/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Synthetic Biology",
    "title": "Designing plasmids encoding predicted pathways by using the BASIC assembly method",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/synthetic-biology/tutorials/basic_assembly_analysis/tutorial.html"
  },
  "/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Synthetic Biology",
    "title": "Evaluating and ranking a set of pathways based on multiple metrics",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/synthetic-biology/tutorials/pathway_analysis/tutorial.html"
  },
  "/topics/teaching/tutorials/gtn-for-instructors/slides.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Overview of the Galaxy Training Material for Instructors",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/gtn-for-instructors/slides.html"
  },
  "/topics/teaching/tutorials/workshop-intro/slides.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Workshop Kickoff",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/workshop-intro/slides.html"
  },
  "/topics/teaching/tutorials/experiences/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Teaching experiences",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/experiences/tutorial.html"
  },
  "/topics/teaching/tutorials/hybrid-training/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Hybrid training",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/teaching/tutorials/hybrid-training/tutorial.html"
  },
  "/topics/teaching/tutorials/running-workshop/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Running a workshop as instructor",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/running-workshop/tutorial.html"
  },
  "/topics/teaching/tutorials/motivation_demotivation/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Motivation and Demotivation",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/motivation_demotivation/tutorial.html"
  },
  "/topics/teaching/tutorials/learner_participation_engagement/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Training techniques to enhance learner participation and engagement",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/learner_participation_engagement/tutorial.html"
  },
  "/topics/teaching/tutorials/live-coding/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Live Coding is a Skill",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/teaching/tutorials/live-coding/tutorial.html"
  },
  "/topics/teaching/tutorials/galaxy-admin-training/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Galaxy Admin Training",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/galaxy-admin-training/tutorial.html"
  },
  "/topics/teaching/tutorials/gallantries-async/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Asynchronous training",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/teaching/tutorials/gallantries-async/tutorial.html"
  },
  "/topics/teaching/tutorials/assessment/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Assessment and feedback in training and teachings",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/assessment/tutorial.html"
  },
  "/topics/teaching/tutorials/teaching-online/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Teaching online",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/teaching/tutorials/teaching-online/tutorial.html"
  },
  "/topics/teaching/tutorials/setup-galaxy-for-training/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Set up a Galaxy for Training",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/setup-galaxy-for-training/tutorial.html"
  },
  "/topics/teaching/tutorials/setup-tiaas-for-training/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Training Infrastructure as a Service",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/teaching/tutorials/setup-tiaas-for-training/tutorial.html"
  },
  "/topics/teaching/tutorials/organize-workshop/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Organizing a workshop",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/organize-workshop/tutorial.html"
  },
  "/topics/teaching/tutorials/workshop-final-exercise/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Train-the-Trainer: putting it all together",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/teaching/tutorials/workshop-final-exercise/tutorial.html"
  },
  "/topics/teaching/tutorials/gallantries-coursebuilder/tutorial.html": {
    "type": "Tutorial",
    "topic": "Teaching and Hosting Galaxy training",
    "title": "Course Builder",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/teaching/tutorials/gallantries-coursebuilder/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/full-de-novo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "De novo transcriptome assembly, annotation, and differential expression analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/full-de-novo/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Small Non-coding RNA Clustering using BlockClust",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/goenrichment/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "GO Enrichment Analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/goenrichment/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "2: RNA-seq counts to genes",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged limma-voom\" href=\"/training-material/search?query=limma-voom\" style=\"--color-primary: #4fe3ff; --color-darker: #00cde8; --color-dimmed: #00d9f6;\">limma-voom</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00ecc2; --color-darker: #00d2ad; --color-dimmed: #00dfb7;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffc3c3; --color-darker: #ffa1a1; --color-dimmed: #ffb2b2;\">QC</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-genes/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "1: RNA-Seq reads to counts",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00ecc2; --color-darker: #00d2ad; --color-dimmed: #00dfb7;\">mouse</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffc3c3; --color-darker: #ffa1a1; --color-dimmed: #ffb2b2;\">QC</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Network analysis with Heinz",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged metatranscriptomics\" href=\"/training-material/search?query=metatranscriptomics\" style=\"--color-primary: #ecc3ff; --color-darker: #e4a1ff; --color-dimmed: #e8b2ff;\">metatranscriptomics</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged network analysis\" href=\"/training-material/search?query=network analysis\" style=\"--color-primary: #00eccb; --color-darker: #00d2b5; --color-dimmed: #00dfc0;\">network analysis</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/clipseq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "CLIP-Seq data analysis from pre-processing to motif detection",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with heatmap2",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-heatmap2/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "3: RNA-seq genes to pathways",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged mouse\" href=\"/training-material/search?query=mouse\" style=\"--color-primary: #00ecc2; --color-darker: #00d2ad; --color-dimmed: #00dfb7;\">mouse</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-genes-to-pathways/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with Volcano Plot",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Genome-wide alternative splicing analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged alternative splicing\" href=\"/training-material/search?query=alternative splicing\" style=\"--color-primary: #96dbff; --color-darker: #23c9ff; --color-dimmed: #6dd1ff;\">alternative splicing</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged isoform switching\" href=\"/training-material/search?query=isoform switching\" style=\"--color-primary: #ffbbfd; --color-darker: #ff93fc; --color-dimmed: #ffa8fd;\">isoform switching</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/srna/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Differential abundance testing of small RNAs",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/srna/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-interactome/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA-RNA interactome data analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-interactome/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with CummeRbund",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-cummerbund/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA-Seq analysis with AskOmics Interactive Tool",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-analysis-with-askomics-it/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Introduction to Transcriptomics",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/introduction/slides.html"
  },
  "/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Reference-based RNAseq data analysis (long)",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-bash-star-align/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA-seq Alignment with STAR",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-bash-star-align/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Visualization of RNA-Seq results with Volcano Plot in R",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-viz-with-volcanoplot-r/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "RNA Seq Counts to Viz in R",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged interactive-tools\" href=\"/training-material/search?query=interactive-tools\" style=\"--color-primary: #ffc4b1; --color-darker: #ffa47e; --color-dimmed: #ffb499;\">interactive-tools</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Reference-based RNA-Seq data analysis",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #79ec00; --color-darker: #6bd200; --color-dimmed: #72df00;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rna-seq\" href=\"/training-material/search?query=rna-seq\" style=\"--color-primary: #ffc5a5; --color-darker: #ffa562; --color-dimmed: #ffb688;\">rna-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged collections\" href=\"/training-material/search?query=collections\" style=\"--color-primary: #ffc2c8; --color-darker: #ffa0aa; --color-dimmed: #ffb1b9;\">collections</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged drosophila\" href=\"/training-material/search?query=drosophila\" style=\"--color-primary: #aad6ff; --color-darker: #6ec2ff; --color-dimmed: #8fccff;\">drosophila</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged QC\" href=\"/training-material/search?query=QC\" style=\"--color-primary: #ffc3c3; --color-darker: #ffa1a1; --color-dimmed: #ffb2b2;\">QC</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "Pathway analysis with the MINERVA Platform",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged bulk\" href=\"/training-material/search?query=bulk\" style=\"--color-primary: #79ec00; --color-darker: #6bd200; --color-dimmed: #72df00;\">bulk</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged rna-seq\" href=\"/training-material/search?query=rna-seq\" style=\"--color-primary: #ffc5a5; --color-darker: #ffa562; --color-dimmed: #ffb688;\">rna-seq</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged viz\" href=\"/training-material/search?query=viz\" style=\"--color-primary: #91e900; --color-darker: #81cf00; --color-dimmed: #89dc00;\">viz</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/transcriptomics/tutorials/minerva-pathways/tutorial.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Transcriptomics",
    "title": "De novo transcriptome reconstruction with RNA-Seq",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Introduction to Variant analysis",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/introduction/slides.html"
  },
  "/topics/variant-analysis/tutorials/non-dip/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Calling variants in non-diploid systems",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged prokaryote\" href=\"/training-material/search?query=prokaryote\" style=\"--color-primary: #d3ccff; --color-darker: #bdb0ff; --color-dimmed: #c8beff;\">prokaryote</a>",
      "<a class=\"label label-default\" title=\"Show all tutorials tagged microgalaxy\" href=\"/training-material/search?query=microgalaxy\" style=\"--color-primary: #dfd600; --color-darker: #c7bf00; --color-dimmed: #d3ca00;\">microgalaxy</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/dunovo/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Calling very rare variants",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged work-in-progress\" href=\"/training-material/search?query=work-in-progress\" style=\"--color-primary: #b5d4ff; --color-darker: #87beff; --color-dimmed: #9fc9ff;\">work-in-progress</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/dunovo/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/exome-seq/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Exome sequencing data analysis for diagnosing a genetic disease",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/dip/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Calling variants in diploid systems",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/dip/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/trio-analysis/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Trio Analysis using Synthetic Datasets from RD-Connect GPAP",
    "contributors": "",
    "tags": [
      "<a class=\"label label-default\" title=\"Show all tutorials tagged cyoa\" href=\"/training-material/search?query=cyoa\" style=\"--color-primary: #ffbeed; --color-darker: #ff98e5; --color-dimmed: #ffabe9;\">cyoa</a>"
    ],
    "url": "/training-material/topics/variant-analysis/tutorials/trio-analysis/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/somatic-variants/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Identification of somatic and germline variants from tumor and normal sample pairs",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Somatic Variant Discovery from WES Data Using Control-FREEC",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/somatic-variant-discovery/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Mapping and molecular identification of phenotype-causing mutations",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/beacon_cnv_query/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Querying the University of Bradford GDC Beacon Database for Copy Number Variants (CNVs)",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/beacon_cnv_query/tutorial.html"
  },
  "/topics/variant-analysis/tutorials/beaconise_1000hg/tutorial.html": {
    "type": "Tutorial",
    "topic": "Variant Analysis",
    "title": "Working with Beacon V2: A Comprehensive Guide to Creating, Uploading, and Searching for Variants with Beacons",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/beaconise_1000hg/tutorial.html"
  },
  "/topics/visualisation/tutorials/circos-microbial/tutorial.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Ploting a Microbial Genome with Circos",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/circos-microbial/tutorial.html"
  },
  "/topics/visualisation/tutorials/circos/tutorial.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Visualisation with Circos",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/circos/tutorial.html"
  },
  "/topics/visualisation/tutorials/bad-graphs/slides.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Friends Don't Let Friends Make Bad Graphs",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/bad-graphs/slides.html"
  },
  "/topics/visualisation/tutorials/introduction/slides.html": {
    "type": "Tutorial",
    "topic": "Visualisation",
    "title": "Visualisations in Galaxy",
    "contributors": "",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/tutorials/introduction/slides.html"
  },
  "/topics/contributing/tutorials/fair-by-design/faqs/FAQ-C1.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can this tutorial be adapted to other instructional development platforms?",
    "contributors": "Sonja Filiposka",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/fair-by-design/faqs/FAQ-C1.html"
  },
  "/topics/contributing/tutorials/fair-by-design/faqs/FAQ-C2.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can the FAIR-by-Design Methodology be used for FAIR development of other types of resources?",
    "contributors": "Anastas Mishev",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/fair-by-design/faqs/FAQ-C2.html"
  },
  "/topics/contributing/tutorials/fair-by-design/faqs/FAQ-L1.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is there a certification available for the FAIR-by-Design Methodology?",
    "contributors": "Anastas Mishev",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/fair-by-design/faqs/FAQ-L1.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do I need to create collections to run MaxQuant analysis or can I use single sample inputs?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_collection.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Including custom modifications into MaxQuant in Galaxy?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_custom_modifications.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which isobaric labeled quantification methods does MaxQuant in Galaxy support?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/Maxquant_isobaric_quant.html"
  },
  "/faqs/gtn/a11y.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does the GTN ensure accessibility?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/a11y.html"
  },
  "/faqs/galaxy/account_create.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I create an account on a public Galaxy instance?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone, Samantha Anjei, Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_create.html"
  },
  "/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/account_exists.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "It says I already have an account when registering for ecology.usegalaxy.eu",
    "contributors": "Coline Royaux, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/topics/ecology/tutorials/PAMPA-toolsuite-tutorial/faqs/account_exists.html"
  },
  "/faqs/galaxy/account_multiple.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I create multiple Galaxy accounts?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_multiple.html"
  },
  "/faqs/galaxy/account_reduce_quota_usage.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I reduce quota usage while still retaining prior work (data, tools, methods)?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_reduce_quota_usage.html"
  },
  "/faqs/galaxy/account_update_preference.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to update account preferences?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/account_update_preference.html"
  },
  "/faqs/galaxy/activity_bar_how_to_enable.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to enable the Activity Bar",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/activity_bar_how_to_enable.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/adapter_trimming.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Automatically trim adapters (without providing custom sequences)",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/adapter_trimming.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I adapt this tutorial to my own data?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop/faqs/adapting_to_different_data.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I adapt this tutorial to my own data?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/adapting_to_different_data.html"
  },
  "/faqs/galaxy/add-toolshed-category-to-a-tool.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add Toolshed category to a tool",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/add-toolshed-category-to-a-tool.html"
  },
  "/topics/admin/faqs/admin-testing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Operating system compatibility",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/admin-testing.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/faqs/advantages.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What advantages does a Chromatogram Library have over a DDA-generated library or predicted spectral library?",
    "contributors": "Emma Leith",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/faqs/advantages.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin/faqs/alevin_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is Alevin is not working?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin/faqs/alevin_version.html"
  },
  "/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/faqs/alevin_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is Alevin is not working?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/faqs/alevin_version.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/alignment_tools.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to replace the existing alignment tools such as HISAT and Freebayes with other tools?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/alignment_tools.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/faqs/alternative_tools.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to use alternative tools to those proposed in the tutorial?",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/alternative_tools.html"
  },
  "/topics/single-cell/faqs/amplification.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is amplification more of an issue in scRNA-seq than RNA-seq?",
    "contributors": "Wendi Bacon, Mehmet Tekman, Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/amplification.html"
  },
  "/faqs/galaxy/analysis_add_custom_build.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding a custom database/build (dbkey)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_add_custom_build.html"
  },
  "/faqs/galaxy/analysis_concatenate.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Pick the right Concatenate tool",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_concatenate.html"
  },
  "/faqs/galaxy/analysis_cut.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Beware of Cuts",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_cut.html"
  },
  "/faqs/galaxy/analysis_differential_expression_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Extended Help for Differential Expression Analysis Tools",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_differential_expression_help.html"
  },
  "/faqs/galaxy/analysis_getting_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I do analysis X? - Getting help",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_getting_help.html"
  },
  "/faqs/galaxy/analysis_jobs_keep_running.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Will my jobs keep running?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_jobs_keep_running.html"
  },
  "/faqs/galaxy/analysis_jobs_not_running.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My jobs aren't running!",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_jobs_not_running.html"
  },
  "/faqs/galaxy/analysis_regular_expressions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Regular Expressions 101",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_regular_expressions.html"
  },
  "/faqs/galaxy/analysis_reporting_issues.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Reporting usage problems, security issues, and bugs",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_reporting_issues.html"
  },
  "/faqs/galaxy/analysis_results_may_vary.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Results may vary",
    "contributors": "Saskia Hiltemann, Anne Fouilloux",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_results_may_vary.html"
  },
  "/faqs/galaxy/analysis_troubleshooting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Troubleshooting errors",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_troubleshooting.html"
  },
  "/faqs/galaxy/analysis_troubleshooting_reporting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What information should I include when reporting a problem?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html"
  },
  "/topics/introduction/faqs/analyze_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I can’t find the “Analyze Data” button",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/analyze_data.html"
  },
  "/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/anndata_not_working.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "AnnData Import/ AnnData Manipulate not working?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/anndata_not_working.html"
  },
  "/topics/admin/faqs/ansible-connection.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Variable connection",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible-connection.html"
  },
  "/topics/admin/faqs/ansible_define-once.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Define once, reference many times",
    "contributors": "Nate Coraor, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_define-once.html"
  },
  "/topics/admin/faqs/ansible_diff.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What if you forget `--diff`?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_diff.html"
  },
  "/topics/admin/faqs/ansible_error-no-hosts-matched.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Error: \"skipping: no hosts matched\"",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_error-no-hosts-matched.html"
  },
  "/topics/admin/faqs/ansible_local.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Running Ansible on your remote machine",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_local.html"
  },
  "/topics/admin/faqs/ansible_role-prefix.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the difference between the roles with `role:` prefix and without?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_role-prefix.html"
  },
  "/topics/admin/faqs/ansible_role-properties.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I know what I can do with a role? What variables are available?",
    "contributors": "Helena Rasche, Nicola Soranzo, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_role-properties.html"
  },
  "/topics/admin/faqs/ansible_see_inventory.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I see what variables are set for a host?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_see_inventory.html"
  },
  "/topics/admin/faqs/ansible_yaml.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is YAML sensitive to True/true/False/false",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/ansible_yaml.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/faqs/bandage_different.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why does my assembly graph in Bandage look different to the one pictured in the tutorial?",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/bandage_different.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/faqs/bioconda_files_exist.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The folder `recipes/belerophon/` and the file `meta.yaml` already exist in bioconda?",
    "contributors": "Alex Ostrovsky, Dave B., Matthias Bernt",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/faqs/bioconda_files_exist.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/bowtie_parameters.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "In bowtie 2 parameters, in place of 1000 for other experiments, should we mention the median fragment length observed in our library?",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/bowtie_parameters.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/breakdown.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the advantage of breaking down protein to peptides before mass spec?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/breakdown.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/faqs/build_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The Build tissue-specific expression dataset tool (step one) exits with an error code.",
    "contributors": "Florence Combes",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/build_error.html"
  },
  "/topics/admin/faqs/byobu-shortcuts.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Opening a split screen in byobu",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/byobu-shortcuts.html"
  },
  "/faqs/galaxy/can-i-use-a-public-galaxy-for-my-private-data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use a public Galaxy for my private data?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/can-i-use-a-public-galaxy-for-my-private-data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/chromosome_names.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we change the chromosome names in the Ensembl GTF to match the UCSC genome reference?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/chromosome_names.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/cloud.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do you have resources to help me get started working in the cloud?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/cloud.html"
  },
  "/topics/community/faqs/codex.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I add my community to the Galaxy CoDex?",
    "contributors": "Bérénice Batut, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/community/faqs/codex.html"
  },
  "/faqs/gtn/collaborative_development.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does the GTN implement the \"Ten simple rules for collaborative lesson development\"",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/collaborative_development.html"
  },
  "/faqs/galaxy/collections_add_tag.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding a tag to a collection",
    "contributors": "Saskia Hiltemann, Helena Rasche, Delphine Lariviere",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_add_tag.html"
  },
  "/faqs/galaxy/collections_build_list.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a dataset collection",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_build_list.html"
  },
  "/faqs/galaxy/collections_build_list_paired.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a paired collection",
    "contributors": "Saskia Hiltemann, Helena Rasche, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_build_list_paired.html"
  },
  "/faqs/galaxy/collections_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the datatype of a collection",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_change_datatype.html"
  },
  "/faqs/galaxy/collections_convert_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Converting the datatype of a collection",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_convert_datatype.html"
  },
  "/faqs/galaxy/collections_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming a collection",
    "contributors": "Delphine Lariviere, Martin Čech, Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/collections_rename.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/collections_test.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I want to use a collection for outputs but it always passes the test even when the script fails. Why?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/collections_test.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/comparing_measurement.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does one compare metaproteomics measurements from two experimental conditions?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/comparing_measurement.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/compining_fastq_files_of_one_sample.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "After sequencing with MinKNOW software, we get many fastq files, do these files need to be combined into one file before uploading or is it possible to upload them all at once?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/compining_fastq_files_of_one_sample.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/contaminants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do we need a contaminant FASTA for MQ in galaxy?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/contaminants.html"
  },
  "/topics/dev/faqs/contributing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to Contribute to Galaxy",
    "contributors": "John Chilton, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/dev/faqs/contributing.html"
  },
  "/topics/contributing/faqs/contributing_a_notebook.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Contributing a Jupyter Notebook to the GTN",
    "contributors": "Helena Rasche, Nicola Soranzo",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/contributing_a_notebook.html"
  },
  "/faqs/gtn/contributors_appreciated.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Thanks!",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_appreciated.html"
  },
  "/faqs/gtn/contributors_getting_started.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I get started with contributing?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_getting_started.html"
  },
  "/faqs/gtn/contributors_git_advanced.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I contribute in \"advanced\" mode?",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_git_advanced.html"
  },
  "/faqs/gtn/contributors_github_interface.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I fix mistakes or expand an existing tutorial using the GitHub interface?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_github_interface.html"
  },
  "/faqs/gtn/contributors_giving_feedback.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I give feedback?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_giving_feedback.html"
  },
  "/faqs/gtn/contributors_helping_project.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What can I do to help the project?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_helping_project.html"
  },
  "/faqs/gtn/contributors_reporting_issues.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I report mistakes or errors?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_reporting_issues.html"
  },
  "/faqs/gtn/contributors_tours.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I test an Interactive Tour?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_tours.html"
  },
  "/faqs/gtn/contributors_without_git.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I create new content without dealing with git?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/contributors_without_git.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/convert_to_MGF.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does one convert RAW files to MGF peak lists within Galaxy?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/convert_to_MGF.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/converting_FASTA.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I have FASTQ files from metagenomics or metatranscriptomics datasets? How can I convert them into a protein FASTA file for metaproteomics searches?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/converting_FASTA.html"
  },
  "/topics/admin/tutorials/ansible-galaxy/faqs/custom-welcome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Customising the welcome page",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/tutorials/ansible-galaxy/faqs/custom-welcome.html"
  },
  "/topics/admin/faqs/data-source-with-pulsar.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using data source tools with Pulsar",
    "contributors": "Martin Wolstencroft",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/data-source-with-pulsar.html"
  },
  "/topics/admin/faqs/database-slow-query.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Database Issues",
    "contributors": "Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/database-slow-query.html"
  },
  "/faqs/galaxy/dataset_incomplete_download.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Incomplete Dataset Download",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataset_incomplete_download.html"
  },
  "/faqs/galaxy/dataset_tool_errors_metadata.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to find and correct tool errors related to Metadata?",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataset_tool_errors_metadata.html"
  },
  "/topics/assembly/tutorials/vgp_genome_assembly/faqs/dataset_upload_fasta_via_urls.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload fasta datasets via links",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_genome_assembly/faqs/dataset_upload_fasta_via_urls.html"
  },
  "/topics/assembly/tutorials/vgp_workflow_training/faqs/dataset_upload_fasta_via_urls.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload fasta datasets via links",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_workflow_training/faqs/dataset_upload_fasta_via_urls.html"
  },
  "/topics/assembly/tutorials/vgp_genome_assembly/faqs/dataset_upload_fastqsanger_via_urls.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload fastqsanger datasets via links",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_genome_assembly/faqs/dataset_upload_fastqsanger_via_urls.html"
  },
  "/topics/assembly/tutorials/vgp_workflow_training/faqs/dataset_upload_fastqsanger_via_urls.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload fastqsanger datasets via links",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/vgp_workflow_training/faqs/dataset_upload_fastqsanger_via_urls.html"
  },
  "/faqs/galaxy/datasets_BAM_dataset_identifiers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding BAM dataset identifiers",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html"
  },
  "/faqs/galaxy/datasets_UCSC_sourced_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Directly obtaining UCSC sourced *genome* identifiers",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html"
  },
  "/faqs/galaxy/datasets_add_tag.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding a tag",
    "contributors": "Bérénice Batut, Wolfgang Maier, Helena Rasche, Saskia Hiltemann, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_add_tag.html"
  },
  "/faqs/galaxy/datasets_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the datatype",
    "contributors": "Bérénice Batut, Nicola Soranzo, Helena Rasche, Saskia Hiltemann, Barakat Ajadi, Lucille Delisle, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_change_datatype.html"
  },
  "/faqs/galaxy-es/datasets_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Cambiar el tipo de datos",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/datasets_change_datatype.html"
  },
  "/faqs/galaxy-fr/datasets_change_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changer le type de données",
    "contributors": "Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-fr/datasets_change_datatype.html"
  },
  "/faqs/galaxy/datasets_change_dbkey.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing database/build (dbkey)",
    "contributors": "Saskia Hiltemann, Helena Rasche, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_change_dbkey.html"
  },
  "/faqs/galaxy/datasets_chromosome_identifiers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Mismatched Chromosome identifiers and how to avoid them",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_chromosome_identifiers.html"
  },
  "/faqs/galaxy/datasets_convert_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Converting the file format",
    "contributors": "Saskia Hiltemann, Helena Rasche, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_convert_datatype.html"
  },
  "/faqs/galaxy/datasets_create_new_file.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a new file",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Wolfgang Maier, Helena Rasche, Simon Bray",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_create_new_file.html"
  },
  "/faqs/galaxy/datasets_data_privacy_features.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to set Data Privacy Features?",
    "contributors": "Jennifer Hillman-Jackson, Desmond Jasper",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_data_privacy_features.html"
  },
  "/faqs/galaxy/datasets_deleting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to delete datasets?",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_deleting.html"
  },
  "/faqs/galaxy/datasets_detect_datatype.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Detecting the datatype (file format)",
    "contributors": "Bérénice Batut, Nicola Soranzo, Helena Rasche, Anne Fouilloux, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_detect_datatype.html"
  },
  "/faqs/galaxy/datasets_download_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading datasets",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_download_datasets.html"
  },
  "/faqs/galaxy/datasets_download_using_command_line.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading datasets using command line",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_download_using_command_line.html"
  },
  "/faqs/galaxy/datasets_finding_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding Datasets",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_finding_datasets.html"
  },
  "/faqs/galaxy/datasets_hidden.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to hide datasets?",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_hidden.html"
  },
  "/faqs/galaxy/datasets_icons.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Different dataset icons and their usage",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_icons.html"
  },
  "/faqs/galaxy/datasets_import_from_data_library.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing data from a data library",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Nicola Soranzo, Helena Rasche, Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_import_from_data_library.html"
  },
  "/faqs/galaxy/datasets_import_from_remote_files.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing data from remote files",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Nicola Soranzo, Helena Rasche, Wolfgang Maier, Soumya Jha",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_import_from_remote_files.html"
  },
  "/faqs/galaxy/datasets_import_via_link.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing via links",
    "contributors": "Bérénice Batut, Helena Rasche, Mehmet Tekman, Gildas Le Corguillé, Saskia Hiltemann, Wendi Bacon, Anton Nekrutenko, Maria Doyle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_import_via_link.html"
  },
  "/faqs/galaxy-es/datasets_import_via_link.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importación por medio de enlaces",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/datasets_import_via_link.html"
  },
  "/faqs/galaxy-fr/datasets_importa_via_link.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importer via un lien",
    "contributors": "Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-fr/datasets_importa_via_link.html"
  },
  "/faqs/galaxy/datasets_job_status.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding job statuses",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_job_status.html"
  },
  "/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Moving datasets between Galaxy servers",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html"
  },
  "/faqs/galaxy/datasets_multiple.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Manipulating multiple history datasets",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_multiple.html"
  },
  "/faqs/galaxy/datasets_not_downloading_at_all.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Datasets not downloading at all",
    "contributors": "Nurzhamalyrys, Melkeberhan Berhanu Degefa, Jennifer Hillman-Jackson",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_not_downloading_at_all.html"
  },
  "/faqs/galaxy/datasets_purging_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Purging datasets",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_purging_datasets.html"
  },
  "/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Quotas for datasets and histories",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html"
  },
  "/faqs/galaxy/datasets_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming a dataset",
    "contributors": "Bérénice Batut, Nicola Soranzo, Saskia Hiltemann, Wolfgang Maier, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_rename.html"
  },
  "/faqs/galaxy/datasets_undelete.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to un-delete datasets?",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_undelete.html"
  },
  "/faqs/galaxy/datasets_unhidden.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to un-hide datasets?",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_unhidden.html"
  },
  "/faqs/galaxy/datasets_upload.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload few files (1-10)",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_upload.html"
  },
  "/faqs/galaxy/datasets_upload_from_genomeark.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload datasets from GenomeArk",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_upload_from_genomeark.html"
  },
  "/faqs/galaxy/datasets_upload_ftp.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Upload many files (>10) via FTP",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_upload_ftp.html"
  },
  "/faqs/galaxy/datasets_working_with_deleted_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with deleted datasets",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_deleted_datasets.html"
  },
  "/faqs/galaxy/datasets_working_with_fasta.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with very large fasta datasets",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_fasta.html"
  },
  "/faqs/galaxy/datasets_working_with_fastq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with very large fastq datasets",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_fastq.html"
  },
  "/faqs/galaxy/datasets_working_with_reference_annotation.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Working with GFF GFT GTF2 GFF3 reference annotation",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html"
  },
  "/faqs/galaxy/datatypes_Using_compressed_fastq_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using compressed fastq data as tool inputs",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_Using_compressed_fastq_data.html"
  },
  "/faqs/galaxy/datatypes_best_practices_fastq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Best practices for loading fastq data into Galaxy",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_best_practices_fastq.html"
  },
  "/faqs/galaxy/datatypes_fastq_and_fastqsanger.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment?",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_compressed.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compressed FASTQ files, (`*.gz`)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_compressed.html"
  },
  "/faqs/galaxy/datatypes_fastq_compressed.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compressed FASTQ files, (`*.gz`)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_compressed.html"
  },
  "/faqs/galaxy/datatypes_fastq_format.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to format fastq data for tools that require .fastqsanger format?",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_format.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_varieties.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "FASTQ files: `fastq` vs `fastqsanger` vs ..",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/datatypes_fastq_varieties.html"
  },
  "/faqs/galaxy/datatypes_fastq_varieties.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "FASTQ files: `fastq` vs `fastqsanger` vs ..",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_fastq_varieties.html"
  },
  "/faqs/galaxy/datatypes_tabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Identifying and formatting Tabular Datasets",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_tabular.html"
  },
  "/faqs/galaxy/datatypes_understanding_datatypes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding Datatypes",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/datatypes_understanding_datatypes.html"
  },
  "/faqs/galaxy/dataupload_EBI-SRA.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I  upload data using EBI-SRA?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataupload_EBI-SRA.html"
  },
  "/faqs/galaxy/dataupload_NCBI_SRA.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Data retrieval with “NCBI SRA Tools” (fastq-dump)",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataupload_NCBI_SRA.html"
  },
  "/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "NCBI SRA sourced fastq data",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The tutorial uses the normalised count table for visualisation. What about using VST normalised counts or rlog normalised counts?",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/deseq2.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/determine_taxonomic_composition.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What software tools are available to determine taxonomic composition from mass spectrometry data?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/determine_taxonomic_composition.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/faqs/different_padj_value.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Could I use a different p-adj value for filtering differentially expressed genes?",
    "contributors": "Cristóbal Gallardo",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/different_padj_value.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/faqs/different_results.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why?",
    "contributors": "Mallory Freeberg",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/faqs/different_results.html"
  },
  "/topics/admin/faqs/diffs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to read a Diff",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/diffs.html"
  },
  "/topics/single-cell/faqs/dimension_reduction.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/dimension_reduction.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/faqs/enrichment.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the principle of an enrichment analysis?",
    "contributors": "Florence Combes",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/enrichment.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/error_closure_args.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My Rscript tool generates a strange R error on STDOUT about an invalid operation on a closure called 'args' ?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/error_closure_args.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/error_compose.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "`docker-compose up` fails with error `/usr/bin/start.sh: line 133: /galaxy/.venv/bin/uwsgi: No such file or directory`",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/error_compose.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/example-queries.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I find example queries for use in the cloud and elsewhere?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/example-queries.html"
  },
  "/faqs/gtn/fair_training.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How does the GTN ensure our training materials are FAIR?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/fair_training.html"
  },
  "/topics/sequence-analysis/faqs/fastq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "FASTQ format",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/faqs/fastq.html"
  },
  "/topics/microbiome/tutorials/metatranscriptomics/faqs/fastq_interlacer_vs_joiner.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we use FASTQ interlacer and not the FASTQ joiner?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/fastq_interlacer_vs_joiner.html"
  },
  "/faqs/galaxy/features_scratchbook.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using the Window Manager to view multiple datasets",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/features_scratchbook.html"
  },
  "/faqs/galaxy-es/features_scratchbook.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Uso del cuaderno de apuntes para ver varios conjuntos de datos",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/features_scratchbook.html"
  },
  "/topics/introduction/faqs/fetched_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "UCSC - I fetched data from a remote website but now I’m logged out of Galaxy and my data is gone?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/fetched_data.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/findmarkers_empty.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My Scanpy FindMarkers step is giving me an empty table",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/findmarkers_empty.html"
  },
  "/topics/galaxy-interface/faqs/flatten-list-list-paired-to-list-paired.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Flatten a list of list of paired datasets into a list of paired datasets",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/faqs/flatten-list-list-paired-to-list-paired.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/formats.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What file/data formats are defined for I/O in Galaxy?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/formats.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/functional_analysis_tools.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/functional_analysis_tools.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/faqs/further_learning.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I read more about Quality Control of data?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/further_learning.html"
  },
  "/topics/admin/faqs/galaxy-blank-screen.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Blank page or no CSS/JavaScript",
    "contributors": "Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-blank-screen.html"
  },
  "/topics/admin/faqs/galaxy-how-many-mules.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How many mules?",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-how-many-mules.html"
  },
  "/topics/admin/faqs/galaxy-jobs-in-containers.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I get my container requiring jobs to run in a container?",
    "contributors": "Helena Rasche, Thanh Le Viet",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-jobs-in-containers.html"
  },
  "/topics/admin/faqs/galaxy-tool-error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Debugging tool errors",
    "contributors": "Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-tool-error.html"
  },
  "/topics/admin/faqs/galaxy-tool-memory-error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Debugging tool memory errors",
    "contributors": "Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-tool-memory-error.html"
  },
  "/topics/admin/faqs/galaxy-tool-missing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Tool missing from Galaxy",
    "contributors": "Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-tool-missing.html"
  },
  "/topics/admin/faqs/galaxy-ui-slow.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Galaxy UI is slow",
    "contributors": "Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-ui-slow.html"
  },
  "/topics/admin/faqs/galaxy-update-22.05.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Updating from 22.01 to 23.0 with Ansible",
    "contributors": "Helena Rasche, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/galaxy-update-22.05.html"
  },
  "/topics/introduction/faqs/galaxy.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My Galaxy looks different than in the tutorial/video",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/galaxy.html"
  },
  "/faqs/gtn/galaxy_events.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are there any upcoming events focused on Galaxy Training?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_events.html"
  },
  "/faqs/gtn/galaxy_tested_with.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compatible Versions of Galaxy",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_tested_with.html"
  },
  "/faqs/gtn/galaxy_usage.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Ways to use Galaxy",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_usage.html"
  },
  "/faqs/gtn/galaxy_what_is_it.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is Galaxy?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/galaxy_what_is_it.html"
  },
  "/topics/proteomics/tutorials/biomarker_selection/faqs/gene_ontology.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is Gene Ontology (GO)?",
    "contributors": "Florence Combes",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/biomarker_selection/faqs/gene_ontology.html"
  },
  "/topics/single-cell/faqs/gene_profile.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What exactly is a ‘Gene profile’?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/gene_profile.html"
  },
  "/faqs/gtn/get_data_upload.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I load data?",
    "contributors": "Jennifer Hillman-Jackson, Alouzeh Brandone",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/get_data_upload.html"
  },
  "/topics/dev/tutorials/tool-from-scratch/faqs/get_planemo.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I get planemo?",
    "contributors": "Alex Ostrovsky, Dave B., Matthias Bernt",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-from-scratch/faqs/get_planemo.html"
  },
  "/topics/admin/faqs/git-attributes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using Git With Ansible Vaults",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/git-attributes.html"
  },
  "/topics/admin/faqs/git-commit.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Time to git commit",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/git-commit.html"
  },
  "/topics/admin/faqs/git-gat-path.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Galaxy Admin Training Path",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/git-gat-path.html"
  },
  "/topics/contributing/faqs/github-fork-gtn.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Forking the GTN repository",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/github-fork-gtn.html"
  },
  "/topics/contributing/faqs/github-fork-master-main.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Updating the default branch from master to main",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/github-fork-master-main.html"
  },
  "/topics/contributing/faqs/github-fork-sync.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Syncing your Fork of the GTN",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/github-fork-sync.html"
  },
  "/topics/community/faqs/governance_gcb.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What's the Galaxy Community Board?",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/community/faqs/governance_gcb.html"
  },
  "/topics/community/faqs/governance_gcb_join.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I join the Galaxy Community Board?",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/community/faqs/governance_gcb_join.html"
  },
  "/topics/community/faqs/governance_structure.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is the Galaxy Governance Structure?",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/community/faqs/governance_structure.html"
  },
  "/topics/single-cell/faqs/gtn-in-galaxy_mode-cs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using tutorial mode and the Case Study suite",
    "contributors": "Saskia Hiltemann, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/gtn-in-galaxy_mode-cs.html"
  },
  "/faqs/gtn/gtn-what-is-it.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is this website?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn-what-is-it.html"
  },
  "/faqs/gtn/gtn_advertise.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I advertise the training materials on my posters?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_advertise.html"
  },
  "/faqs/gtn/gtn_audience.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What audiences are the tutorials for?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_audience.html"
  },
  "/faqs/gtn/gtn_citing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I cite the GTN?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_citing.html"
  },
  "/faqs/gtn/gtn_event_create.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a GTN Event",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_event_create.html"
  },
  "/faqs/gtn/gtn_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using Answer Key Histories",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_example_histories.html"
  },
  "/faqs/gtn/gtn_faq_create.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a GTN FAQ",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_faq_create.html"
  },
  "/faqs/gtn/gtn_license.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How is the content licensed?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_license.html"
  },
  "/faqs/gtn/gtn_licenses.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What licenses are used in the GTN?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_licenses.html"
  },
  "/faqs/gtn/gtn_news_create_post.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a GTN News post",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_news_create_post.html"
  },
  "/faqs/gtn/gtn_stats.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "GTN Stats",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_stats.html"
  },
  "/faqs/gtn/gtn_sustainability.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sustainability of the training-material and metadata",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_sustainability.html"
  },
  "/faqs/gtn/gtn_tutorial_mode.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Supporting Tutorial Mode (GTN-in-Galaxy) in a tutorial",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_tutorial_mode.html"
  },
  "/faqs/gtn/gtn_tutorials.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What are the tutorials for?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_tutorials.html"
  },
  "/faqs/gtn/gtn_workflow_testing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding workflow tests with Planemo",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/gtn_workflow_testing.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/heatmap_colours.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the heatmap colours",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/heatmap_colours.html"
  },
  "/faqs/galaxy/histories_annotation.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "History annotation",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_annotation.html"
  },
  "/faqs/galaxy/histories_copy_dataset.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Copy a dataset between histories",
    "contributors": "Gildas Le Corguillé, Saskia Hiltemann, Helena Rasche, Bérénice Batut, Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_copy_dataset.html"
  },
  "/faqs/galaxy/histories_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a new history",
    "contributors": "Bérénice Batut, Wolfgang Maier, Saskia Hiltemann, Helena Rasche, Wendi Bacon, Nicola Soranzo, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_create_new.html"
  },
  "/faqs/galaxy-es/histories_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Para la creación de un historial nuevo",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/histories_create_new.html"
  },
  "/faqs/galaxy-fr/histories_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Créer un nouvel history",
    "contributors": "Bérénice Batut, Wolfgang Maier, Saskia Hiltemann, Helena Rasche, Wendi Bacon, Nicola Soranzo, Yvan Le Bras",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-fr/histories_create_new.html"
  },
  "/faqs/galaxy/histories_dataset_colors.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Dataset colors",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_dataset_colors.html"
  },
  "/faqs/galaxy/histories_dataset_item.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Dataset snippet",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_dataset_item.html"
  },
  "/faqs/galaxy/histories_datasets_vs_collections.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Datasets versus collections",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_datasets_vs_collections.html"
  },
  "/faqs/galaxy/histories_download_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading histories",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_download_histories.html"
  },
  "/faqs/galaxy/histories_find_purge.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Find all Histories and purge (aka permanently delete)",
    "contributors": "Jennifer Hillman-Jackson, Samantha Anjei, Anne Fouilloux",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_find_purge.html"
  },
  "/faqs/galaxy/histories_finding_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding Histories",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_finding_histories.html"
  },
  "/faqs/galaxy/histories_import.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing a history",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_import.html"
  },
  "/faqs/galaxy/histories_list.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "View a list of all histories",
    "contributors": "Anton Nekrutenko, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_list.html"
  },
  "/faqs/galaxy/histories_options.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "History options",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_options.html"
  },
  "/faqs/galaxy/histories_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming a history",
    "contributors": "Bérénice Batut, Wolfgang Maier, Saskia Hiltemann, Helena Rasche, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_rename.html"
  },
  "/faqs/galaxy-es/histories_rename.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Cambiando el nombre de un historial",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/histories_rename.html"
  },
  "/faqs/galaxy/histories_search.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Searching your history",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_search.html"
  },
  "/faqs/galaxy/histories_shared.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding and working with \"Histories shared with me\"",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_shared.html"
  },
  "/faqs/galaxy/histories_sharing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sharing your History",
    "contributors": "Saskia Hiltemann, Ahmed Hamid Awan",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_sharing.html"
  },
  "/faqs/galaxy-es/histories_sharing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Compartiendo un historial",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/histories_sharing.html"
  },
  "/faqs/galaxy/histories_side_by_side_view.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "View histories side-by-side",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_side_by_side_view.html"
  },
  "/faqs/galaxy/histories_switch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Switching to an existing history",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_switch.html"
  },
  "/faqs/galaxy/histories_tagging.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "History tagging",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_tagging.html"
  },
  "/faqs/galaxy/histories_top_control_buttons.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Top level history controls",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_top_control_buttons.html"
  },
  "/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Transfer entire histories from one Galaxy server to another",
    "contributors": "Jennifer Hillman-Jackson, Adaora Okwo",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html"
  },
  "/faqs/galaxy/histories_undelete.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Undeleting history",
    "contributors": "Valentine Gatwiri, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_undelete.html"
  },
  "/faqs/galaxy/histories_unsharing_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Unsharing unwanted histories",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/histories_unsharing_histories.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_filtering_filter.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What does `^.*Gallus|Homo|Bos.*$` mean?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_filtering_filter.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_removal.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "For preprocessing part with host removal: Where do you find the abbreviations for each host species available (e.g. bos is cow, homo is human..)?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/host_removal.html"
  },
  "/topics/assembly/faqs/hybrid_assembly.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Most tools seem to have options for assembly using long and short reads, what are the pros and cons of the different tools?",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/hybrid_assembly.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which icons are available to use in my tutorial?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/icons_list.html"
  },
  "/faqs/galaxy/importing-data-from-sierra-lims.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing data from Sierra LIMS",
    "contributors": "Morgan Howells",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/importing-data-from-sierra-lims.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/importing_genomes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "From where can I import other genomes?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/importing_genomes.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When is the \"infer experiment\" tool used in practice?",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/infer_experiments.html"
  },
  "/topics/contributing/faqs/input-and-answer-keys.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Input Histories & Answer Keys",
    "contributors": "Helena Rasche, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/input-and-answer-keys.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop/faqs/input_not_found.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The input for a tool is not listed in the dropdown",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop/faqs/input_not_found.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/input_not_found.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The input for a tool is not listed in the dropdown",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/input_not_found.html"
  },
  "/topics/admin/faqs/install_tool.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Install tools via the Admin UI",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/install_tool.html"
  },
  "/faqs/gtn/instructors_best_practices.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What are the best practices for teaching with Galaxy?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_best_practices.html"
  },
  "/faqs/gtn/instructors_choosing_server.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What Galaxy instance should I use for my training?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_choosing_server.html"
  },
  "/faqs/gtn/instructors_getting_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I get help?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_getting_help.html"
  },
  "/faqs/gtn/instructors_getting_started.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where do I start?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/instructors_getting_started.html"
  },
  "/faqs/galaxy/interactive_tools_jupyter_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch JupyterLab",
    "contributors": "Anne Fouilloux, Saskia Hiltemann, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"
  },
  "/faqs/galaxy/interactive_tools_jupyter_terminal.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Open a Terminal in Jupyter",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"
  },
  "/faqs/galaxy/interactive_tools_open.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Open interactive tool",
    "contributors": "Marie Gramm",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_open.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_knit.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Knitting RMarkdown documents in RStudio",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"
  },
  "/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch RStudio",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Helena Rasche, Yvan Le Bras, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_launch.html"
  },
  "/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch RStudio",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Helena Rasche, Yvan Le Bras, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_launch.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_launch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Launch RStudio",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Helena Rasche, Yvan Le Bras, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Learning with RMarkdown in RStudio",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"
  },
  "/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_stop.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Stop RStudio",
    "contributors": "Yvan Le Bras, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-advanced/faqs/interactive_tools_rstudio_stop.html"
  },
  "/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_stop.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Stop RStudio",
    "contributors": "Yvan Le Bras, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/data-science/tutorials/r-basics/faqs/interactive_tools_rstudio_stop.html"
  },
  "/faqs/galaxy/interactive_tools_rstudio_stop.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Stop RStudio",
    "contributors": "Yvan Le Bras, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"
  },
  "/faqs/gtn/issues_galaxy.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What information should I include when reporting a problem?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/issues_galaxy.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/faqs/jbrowse_runtime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "JBrowse is taking a long time to complete?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/faqs/jbrowse_runtime.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/job_grey.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "First job I submitted remains grey or running for a long time - is it broken?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/job_grey.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/kalamri_database_filtering.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is there a way to filter on the Kalimari database?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/kalamri_database_filtering.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/key-error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "TB Variant Report crashes (with an error about KeyError: 'protein')",
    "contributors": "Peter van Heusden, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/key-error.html"
  },
  "/topics/microbiome/faqs/kraken.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Kraken2 and the k-mer approach for taxonomy classification",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/faqs/kraken.html"
  },
  "/faqs/gtn/learners_getting_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I get help?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_getting_help.html"
  },
  "/faqs/gtn/learners_getting_started.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where do I start?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_getting_started.html"
  },
  "/faqs/gtn/learners_running_tutorials.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I run the hands-on tutorials?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_running_tutorials.html"
  },
  "/faqs/gtn/learners_using_materials.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I use this material?",
    "contributors": "Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/learners_using_materials.html"
  },
  "/topics/contributing/faqs/learning-pathways.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Defining a Learning Pathway",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/learning-pathways.html"
  },
  "/topics/admin/faqs/library-permissions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Library Permission Issues",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/library-permissions.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/local_install.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do I need that big (~5GB!) complicated Docker thing - can I just install the ToolFactory into our local galaxy server from the toolshed?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/local_install.html"
  },
  "/topics/introduction/faqs/login.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My jobs are not running / I cannot see the history overview menu",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/login.html"
  },
  "/topics/admin/faqs/maintain_fail-users-jobs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Failing all jobs from a specific user",
    "contributors": "Helena Rasche, Björn Grüning",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/maintain_fail-users-jobs.html"
  },
  "/topics/admin/faqs/mapping_by_user.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Mapping Jobs to Specific Storage By User",
    "contributors": "Helena Rasche, Björn Grüning, Nate Coraor",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/mapping_by_user.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/match.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When can you use (or cannot use) Match between runs in MaxQuant?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/match.html"
  },
  "/topics/community/faqs/meeting_times.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I plan meetings across timezones?",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/community/faqs/meeting_times.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/merge_database.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do you need to merge the databases? Because you can select multiple fasta files in MaxQuant.",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/merge_database.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/metadata-description.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I find the full listing and description of the columns in each metadata table?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/metadata-description.html"
  },
  "/topics/admin/faqs/missed-something.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Got lost along the way?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/missed-something.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop/faqs/more_info.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I read more about this analysis?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop/faqs/more_info.html"
  },
  "/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/more_info.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where can I read more about this analysis?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/mothur-miseq-sop-short/faqs/more_info.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/mqpar.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "If you use a mqpar file, can you include modifications that are not in the Galaxy version? For instance, propionamide (Cys alkylation by acrylamide).",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/mqpar.html"
  },
  "/topics/microbiome/tutorials/metatranscriptomics/faqs/multiple_inputs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can we use the ASaiM-MT workflow on multiple input files at the same time?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/multiple_inputs.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/faqs/multiqc_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "MultiQC error for your FastQC reports?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/multiqc_error.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/mvp.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "In the MVP platform, is it possible to view the genomic location of all the peptides?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/mvp.html"
  },
  "/faqs/gtn/my-galaxy-training.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is my.galaxy.training",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/my-galaxy-training.html"
  },
  "/topics/contributing/faqs/new-contributions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using the new Contributions Annotation framework",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/new-contributions.html"
  },
  "/topics/dev/tutorials/debugging/faqs/no_tests_no_yarn.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I cannot run client tests because yarn is not installed.",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/debugging/faqs/no_tests_no_yarn.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/normalize_lfq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What does it mean to normalize the LFQ intensities?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/normalize_lfq.html"
  },
  "/topics/single-cell/faqs/notebook_warning.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Notebook-based tutorials can give different outputs",
    "contributors": "Morgan Howells, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/notebook_warning.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/other-datatypes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When will aligned read objects be available for other data types?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/other-datatypes.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/outputs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Does MaxQuant give as output possibility the PSMs and PEPs?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/outputs.html"
  },
  "/topics/sequence-analysis/tutorials/quality-control/faqs/perbase_warning.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When I get a warning for base per sequence content, what should I do?",
    "contributors": "Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/quality-control/faqs/perbase_warning.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/faqs/pilon.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Running more than one round of Pilon polishing",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/pilon.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/planemo_multiple.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Only one Planemo test runs at a time. Why doesn't the server allow more than one at once?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/planemo_multiple.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_fails.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "On Scanpy PlotEmbed, the tool is failing",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_fails.html"
  },
  "/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_results_missing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "On the Scanpy PlotEmbed step, my object doesn’t have Il2ra or Cd8b1 or Cd8a etc.",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/faqs/plotembed_results_missing.html"
  },
  "/topics/assembly/tutorials/chloroplast-assembly/faqs/polish_longreads.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can we polish the assembly with long reads too?",
    "contributors": "Anna Syme, Simon Gladman",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/chloroplast-assembly/faqs/polish_longreads.html"
  },
  "/faqs/galaxy/preferences_admin_api_key.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Getting your API key",
    "contributors": "Helena Rasche, Saskia Hiltemann, Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/preferences_admin_api_key.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Example histories for the proteogenomics tutorials",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_example_histories.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Example histories for the proteogenomics tutorials",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_example_histories.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_example_histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Example histories for the proteogenomics tutorials",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_example_histories.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_own_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_own_data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_own_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_own_data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_own_data.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_own_data.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_querytabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The workflows contain several Query tabular for text manipulation, is there a tutorial for that?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_querytabular.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_querytabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The workflows contain several Query tabular for text manipulation, is there a tutorial for that?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_querytabular.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_querytabular.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The workflows contain several Query tabular for text manipulation, is there a tutorial for that?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_querytabular.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_variants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What kind of variants are seen in the output?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/proteogenomics_variants.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_variants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What kind of variants are seen in the output?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/proteogenomics_variants.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_variants.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What kind of variants are seen in the output?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/proteogenomics_variants.html"
  },
  "/topics/single-cell/tutorials/scrna-umis/faqs/protocol.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How do I know what protocol my data was sequenced with?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-umis/faqs/protocol.html"
  },
  "/topics/admin/faqs/python-memory-leaks.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Debugging Memory Leaks",
    "contributors": "Marius van den Beek, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/python-memory-leaks.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/quality-scores.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why don't the aligned read files have quality scores?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/quality-scores.html"
  },
  "/topics/sequence-analysis/faqs/quality_score.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Quality Scores",
    "contributors": "Bérénice Batut, Natalie Kucher, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/faqs/quality_score.html"
  },
  "/topics/transcriptomics/tutorials/mirna-target-finder/faqs/quantification_alternatives.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use alternative tools for the Quantification step?",
    "contributors": "Cristóbal Gallardo",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/quantification_alternatives.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/quantitation_default.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "For the “quantitation method” what is the default if I just leave it as “None”? Label free?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/quantitation_default.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/faqs/query_no_results.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why does the query `SRR11772204 OR SRR11597145 OR SRR11667145` in the Run Selector not return any results?",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/faqs/query_no_results.html"
  },
  "/topics/single-cell/tutorials/scrna-umis/faqs/r1r2.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are Barcodes always on R1 and Sequence data on R2?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-umis/faqs/r1r2.html"
  },
  "/faqs/gtn/recordings_add.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Adding your recording to a tutorial or slide deck",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/recordings_add.html"
  },
  "/faqs/gtn/recordings_create.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Recording a video tutorial",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/recordings_create.html"
  },
  "/faqs/galaxy/reference_genomes_custom_genomes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How to use Custom Reference Genomes?",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reference_genomes_custom_genomes.html"
  },
  "/faqs/galaxy/reference_genomes_sorting_reference_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sorting Reference Genome",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html"
  },
  "/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Troubleshooting Custom Genome fasta",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/remaining_human_sequences.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Isn't it awkward to find so many humans sequences there, since we filter for them before?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/remaining_human_sequences.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-cyoa.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Preparing materials for asynchronous learning: CYOA",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-cyoa.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-faqs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Preparing materials for asynchronous learning: FAQs",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-faqs.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-selfstudy.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Preparing materials for asynchronous learning: Self-Study",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching-selfstudy.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Preparing materials for asynchronous learning: Tips",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/remote-teaching.html"
  },
  "/topics/dev/tutorials/tool-generators/faqs/replace_manual_install.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is the ToolFactory a complete replacement for manual tool building?",
    "contributors": "Ross Lazarus",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/tool-generators/faqs/replace_manual_install.html"
  },
  "/faqs/galaxy/reports_collapsible_box.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Making an element collapsible in a report",
    "contributors": "Helena Rasche, Aysam Guerler",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reports_collapsible_box.html"
  },
  "/faqs/galaxy/reports_tables.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Enhancing tabular dataset previews in reports/pages",
    "contributors": "John Chilton, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/reports_tables.html"
  },
  "/faqs/galaxy/request-galaxy-tools-on-a-specific-server.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Request Galaxy tools on a specific server",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/request-galaxy-tools-on-a-specific-server.html"
  },
  "/topics/proteomics/faqs/resources.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Additional resources to learn more about proteomic data analysis",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/faqs/resources.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/reusing_workflows.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can I use these workflows on datasets generated from our laboratory?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/reusing_workflows.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to visualize the RNA STAR bam file using the JBrowse tool?",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_bam.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "RNAstar: Why do we set 36 for 'Length of the genomic sequence around annotated junctions'?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/rnastar_tool.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/runtime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "My snippy is running for a very long time. Is this normal?",
    "contributors": "Peter van Heusden",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/runtime.html"
  },
  "/topics/galaxy-interface/tutorials/workflow-automation/faqs/sars-bot-usegalaxy-be.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What should I do special if on usegalaxy.be?",
    "contributors": "Bert Droesbeke",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/workflow-automation/faqs/sars-bot-usegalaxy-be.html"
  },
  "/topics/single-cell/faqs/sc_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why is my tool erroring as 'Above error raised while reading key '/layers' of type <class 'h5py._hl.group.Group'> from /.'",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/sc_version.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/search_algorithms.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which search algorithms are recommended for searching the metaproteomics data?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_algorithms.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/search_gui.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How many search engines can you use in SearchGUI?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/faqs/search_gui.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/search_strategies.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I have a really large search database, what search strategies do you recommend for searching my mass spectrometry dataset?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/search_strategies.html"
  },
  "/topics/dev/tutorials/debugging/faqs/selenium_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "When I try to run a Selenium test, I get an error",
    "contributors": "Assunta DeSanto",
    "tags": [

    ],
    "url": "/training-material/topics/dev/tutorials/debugging/faqs/selenium_error.html"
  },
  "/faqs/galaxy/sequencing_illumina_miseq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Illumina MiSeq sequencing",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/sequencing_illumina_miseq.html"
  },
  "/faqs/galaxy/sequencing_nanopore.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Nanopore sequencing",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/sequencing_nanopore.html"
  },
  "/faqs/gtn/short-url.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Finding a material's PURL or Short URL",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/short-url.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/shotgun.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How many proteins can be identified and quantified in shotgun proteomics?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/shotgun.html"
  },
  "/topics/single-cell/faqs/single_cell_omics.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Use our Single Cell Omics Lab",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/single_cell_omics.html"
  },
  "/topics/microbiome/tutorials/metatranscriptomics/faqs/single_end.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can this ASaiM workflow be used for single-end data?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/single_end.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/faqs/snp.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is a SNP?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/faqs/snp.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/stats.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "MSStats: what does ‘compare groups = yes’ mean? And the comparison matrix to define the contrast between the 2 groups?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/stats.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "In 'infer experiments' I get unequal numbers, but in the IGV it looks like it is unstranded. What does this mean?",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/strandedness.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/studying_microbiome_state.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What other methods are available to study the functional state of the microbiome within Galaxy?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/studying_microbiome_state.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/subsampling.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Is it possible to subsample some samples if you have more reads?",
    "contributors": "Lucille Delisle",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/subsampling.html"
  },
  "/faqs/galaxy/support.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where do I get more support?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/support.html"
  },
  "/faqs/galaxy/support_admins.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Contacting Galaxy Administrators",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/support_admins.html"
  },
  "/topics/microbiome/faqs/taxon.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is Taxonomy?",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/faqs/taxon.html"
  },
  "/topics/single-cell/faqs/techniques.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can RNA-seq techniques be applied to scRNA-seq?",
    "contributors": "Wendi Bacon, Mehmet Tekman, Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/techniques.html"
  },
  "/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/terms-of-usage.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are these data free to use and download?",
    "contributors": "Jon Trow, Adelaide Rhodes",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/ncbi-sarf/faqs/terms-of-usage.html"
  },
  "/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/tmt_itraq.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Does MaxQuant in Galaxy support TMT, iTRAQ, etc.?",
    "contributors": "Melanie Föll",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/maxquant-msstats-dda-lfq/faqs/tmt_itraq.html"
  },
  "/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/tool_order.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do I have to run the tools in the order of the tutorial?",
    "contributors": "Peter van Heusden",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/tb-variant-analysis/faqs/tool_order.html"
  },
  "/faqs/galaxy/tools_change_version.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing the tool version",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_change_version.html"
  },
  "/faqs/galaxy/tools_favorite.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Organizing the tool panel",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_favorite.html"
  },
  "/faqs/galaxy/tools_help.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Where is the tool help?",
    "contributors": "Jennifer Hillman-Jackson, Eshaan Deshmukh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_help.html"
  },
  "/faqs/galaxy/tools_logs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Viewing tool logs (`stdout` and `stderr`)",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_logs.html"
  },
  "/faqs/galaxy/tools_missing.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "If a Tool is Missing",
    "contributors": "Saskia Hiltemann, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_missing.html"
  },
  "/faqs/galaxy/tools_rerun.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Re-running a tool",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_rerun.html"
  },
  "/faqs/galaxy-es/tools_rerun.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Para volver a ejecutar una herramienta",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/tools_rerun.html"
  },
  "/faqs/galaxy/tools_select_collection.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Selecting a dataset collection as input",
    "contributors": "Bérénice Batut, Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_select_collection.html"
  },
  "/faqs/galaxy/tools_select_multiple_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Select multiple datasets",
    "contributors": "Bérénice Batut, Wolfgang Maier, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_select_multiple_datasets.html"
  },
  "/faqs/galaxy/tools_similar.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Multipile similar tools available",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_similar.html"
  },
  "/faqs/galaxy/tools_sorting.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Sorting Tools",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_sorting.html"
  },
  "/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Tool doesn't recognize input datasets",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html"
  },
  "/topics/proteomics/tutorials/metaproteomics/faqs/tools_versions.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Which version of SearchGUI and PeptideShaker shall I use for this tutorial?",
    "contributors": "Milla Berezutskaia",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/metaproteomics/faqs/tools_versions.html"
  },
  "/topics/transcriptomics/tutorials/de-novo/faqs/transcript.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I’m using the same training data, tools, and parameters as the tutorial, but I get a different number of transcripts with a significant change in gene expression between the G1E and megakaryocyte cellular states. Why?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/de-novo/faqs/transcript.html"
  },
  "/faqs/gtn/translations.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Translations within the GTN",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/translations.html"
  },
  "/faqs/galaxy/troubleshooting_cluster_failure.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding 'canceled by admin' or cluster failure error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_cluster_failure.html"
  },
  "/faqs/galaxy/troubleshooting_excess_memory.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding 'exceeds memory allocation' error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_excess_memory.html"
  },
  "/faqs/galaxy/troubleshooting_input_problem.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding input error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_input_problem.html"
  },
  "/faqs/galaxy/troubleshooting_value_error.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding ValueError error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_value_error.html"
  },
  "/faqs/galaxy/troubleshooting_walltime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Understanding walltime error messages",
    "contributors": "Jennifer Hillman-Jackson, Garima Singh",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/troubleshooting_walltime.html"
  },
  "/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I create a tutorial skeleton from a Galaxy workflow?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/tutorials/create-new-tutorial-content/faqs/tutorial_from_workflows.html"
  },
  "/faqs/galaxy/tutorial_mode.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using tutorial mode",
    "contributors": "Saskia Hiltemann, Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/tutorial_mode.html"
  },
  "/topics/introduction/faqs/ucsc.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "I got slightly different numbers than were in the tutorial",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/ucsc.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/ucsc_import.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "UCSC import: what should my file look like?",
    "contributors": "Lucille Delisle, Hans-Rudolf Hotz, Florian Heyl",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/ucsc_import.html"
  },
  "/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/umap_plot_errors.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "The UMAP Plots errors out sometimes?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/faqs/umap_plot_errors.html"
  },
  "/topics/single-cell/faqs/umi.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Are UMIs not actually unique?",
    "contributors": "Wendi Bacon, Mehmet Tekman",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/umi.html"
  },
  "/topics/assembly/tutorials/assembly-with-preprocessing/faqs/upload_slow.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Downloading the files from the NCBI server fails or takes too long.",
    "contributors": "Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/upload_slow.html"
  },
  "/topics/single-cell/faqs/user_community_join.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "How can I talk with other users?",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/user_community_join.html"
  },
  "/faqs/galaxy/user_preferences_account_usage_quota.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Does your account usage quota seem incorrect?",
    "contributors": "Jennifer Hillman-Jackson, Nurzhamalyrys",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/user_preferences_account_usage_quota.html"
  },
  "/faqs/galaxy/user_preferences_forgot_password.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Forgot Password",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/user_preferences_forgot_password.html"
  },
  "/faqs/galaxy/user_preferences_manage_information.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Changing acount email or password",
    "contributors": "Jennifer Hillman-Jackson, Melkeberhan Berhanu Degefa",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/user_preferences_manage_information.html"
  },
  "/faqs/gtn/using-answer-key-histories.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using answer key histories",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/using-answer-key-histories.html"
  },
  "/topics/single-cell/faqs/variable_genes.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we only consider highly variable genes?",
    "contributors": "Rahmot Afolabi",
    "tags": [

    ],
    "url": "/training-material/topics/single-cell/faqs/variable_genes.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/variant_calling_quality.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can we use snippy pipeline instead for the phylogenetic analysis?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/variant_calling_quality.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/variant_mapping_file.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why do we have a variant mapping file when it is not being used in the workflow?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/variant_mapping_file.html"
  },
  "/faqs/galaxy/visualisations_igb.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Open History files in Integrated Genome Browser (IGB)",
    "contributors": "Paige Kulzer",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/visualisations_igb.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/faqs/visualisations_igv.html"
  },
  "/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/faqs/visualisations_igv.html"
  },
  "/topics/proteomics/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/faqs/visualisations_igv.html"
  },
  "/topics/assembly/tutorials/unicycler-assembly/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/tutorials/unicycler-assembly/faqs/visualisations_igv.html"
  },
  "/topics/assembly/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/visualisations_igv.html"
  },
  "/topics/visualisation/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/tal1-binding-site-identification/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/tal1-binding-site-identification/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/atac-seq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/atac-seq/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/visualisations_igv.html"
  },
  "/topics/epigenetics/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/epigenetics/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/ref-based/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/ref-based/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/small_ncrna_clustering/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/small_ncrna_clustering/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/rb-rnaseq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rb-rnaseq/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/rna-seq-reads-to-counts/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/tutorials/clipseq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/tutorials/clipseq/faqs/visualisations_igv.html"
  },
  "/topics/transcriptomics/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/transcriptomics/faqs/visualisations_igv.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-strands/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-strands/faqs/visualisations_igv.html"
  },
  "/topics/introduction/tutorials/galaxy-intro-101/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/galaxy-intro-101/faqs/visualisations_igv.html"
  },
  "/topics/introduction/tutorials/igv-introduction/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/tutorials/igv-introduction/faqs/visualisations_igv.html"
  },
  "/topics/introduction/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/introduction/faqs/visualisations_igv.html"
  },
  "/topics/variant-analysis/tutorials/non-dip/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/non-dip/faqs/visualisations_igv.html"
  },
  "/topics/variant-analysis/tutorials/dunovo/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/tutorials/dunovo/faqs/visualisations_igv.html"
  },
  "/topics/variant-analysis/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/variant-analysis/faqs/visualisations_igv.html"
  },
  "/topics/galaxy-interface/tutorials/collections/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/tutorials/collections/faqs/visualisations_igv.html"
  },
  "/topics/galaxy-interface/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/galaxy-interface/faqs/visualisations_igv.html"
  },
  "/topics/genome-annotation/tutorials/tnseq/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/tutorials/tnseq/faqs/visualisations_igv.html"
  },
  "/topics/genome-annotation/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/genome-annotation/faqs/visualisations_igv.html"
  },
  "/topics/sequence-analysis/tutorials/mapping/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/tutorials/mapping/faqs/visualisations_igv.html"
  },
  "/topics/sequence-analysis/faqs/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/sequence-analysis/faqs/visualisations_igv.html"
  },
  "/faqs/galaxy/visualisations_igv.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Using IGV with Galaxy",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/visualisations_igv.html"
  },
  "/topics/assembly/faqs/visualizations_igv_add_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add genome and annotations to IGV from Galaxy",
    "contributors": "Delphine Lariviere, Bérénice Batut, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/visualizations_igv_add_genome.html"
  },
  "/topics/visualisation/faqs/visualizations_igv_add_genome.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add genome and annotations to IGV from Galaxy",
    "contributors": "Delphine Lariviere, Bérénice Batut, Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/faqs/visualizations_igv_add_genome.html"
  },
  "/topics/assembly/faqs/visualizations_igv_view_mapping.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add Mapped reads track to IGV from Galaxy",
    "contributors": "Delphine Lariviere, Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/assembly/faqs/visualizations_igv_view_mapping.html"
  },
  "/topics/visualisation/faqs/visualizations_igv_view_mapping.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Add Mapped reads track to IGV from Galaxy",
    "contributors": "Delphine Lariviere, Bérénice Batut, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/topics/visualisation/faqs/visualizations_igv_view_mapping.html"
  },
  "/topics/microbiome/tutorials/metatranscriptomics/faqs/vsearch.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why don't we perform the V-Search dereplication step of ASaiM for metatrascriptomic data?",
    "contributors": "Subina Mehta",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/metatranscriptomics/faqs/vsearch.html"
  },
  "/faqs/gtn/what-is-a-learning-pathway.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "What is a Learning Pathway?",
    "contributors": "Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/what-is-a-learning-pathway.html"
  },
  "/topics/admin/faqs/why-isnt-my-history-updating.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why isn't my history updating?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/admin/faqs/why-isnt-my-history-updating.html"
  },
  "/topics/contributing/faqs/why-prerequisites.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Annotating Pre-requisites",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/topics/contributing/faqs/why-prerequisites.html"
  },
  "/faqs/gtn/why-the-gtn.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why host your materials with the GTN?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/gtn/why-the-gtn.html"
  },
  "/faqs/galaxy/why_not_excel.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Why not use Excel?",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/why_not_excel.html"
  },
  "/topics/proteomics/tutorials/encyclopedia/faqs/without_spectral_library.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can EncyclopeDIA be run on a DIA-MS dataset without a spectral library?",
    "contributors": "Emma Leith",
    "tags": [

    ],
    "url": "/training-material/topics/proteomics/tutorials/encyclopedia/faqs/without_spectral_library.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/worflow_and_sequencing_techniques.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Can we also use this workflow on Illumina raw reads?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/worflow_and_sequencing_techniques.html"
  },
  "/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/workflow_and_isolations.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Do the pipelines work with both isolates and direct from raw meat? or only isolate?",
    "contributors": "Engy Nasr",
    "tags": [

    ],
    "url": "/training-material/topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/faqs/workflow_and_isolations.html"
  },
  "/faqs/galaxy/workflows_annotate.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Annotate a workflow",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_annotate.html"
  },
  "/faqs/galaxy/workflows_best_practices.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Ensuring Workflows meet Best Practices",
    "contributors": "Helena Rasche, Eli Chadwick",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_best_practices.html"
  },
  "/faqs/galaxy/workflows_create_new.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Creating a new workflow",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_create_new.html"
  },
  "/faqs/galaxy/workflows_edit.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Opening the workflow editor",
    "contributors": "Peter van Heusden, Saskia Hiltemann, Anne Fouilloux, Nicola Soranzo, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_edit.html"
  },
  "/faqs/galaxy/workflows_extract_from_history.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Extracting a workflow from your history",
    "contributors": "Saskia Hiltemann, Helena Rasche, Nicola Soranzo",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_extract_from_history.html"
  },
  "/faqs/galaxy-es/workflows_extract_from_history.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Extraer un flujo de trabajo de tu historial",
    "contributors": "Wendi Bacon",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy-es/workflows_extract_from_history.html"
  },
  "/faqs/galaxy/workflows_hide_intermediate_steps.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Hiding intermediate steps",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"
  },
  "/faqs/galaxy/workflows_import.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing a workflow",
    "contributors": "Saskia Hiltemann, Maria Doyle, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import.html"
  },
  "/faqs/galaxy/workflows_import_from_dockstore.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Import workflows from DockStore",
    "contributors": "Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import_from_dockstore.html"
  },
  "/faqs/galaxy/workflows_import_from_workflowhub.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Import workflows from WorkflowHub",
    "contributors": "Cristóbal Gallardo, Linelle Abueg, Anton Nekrutenko",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import_from_workflowhub.html"
  },
  "/faqs/galaxy/workflows_import_search.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing a workflow using the search",
    "contributors": "Bérénice Batut, Wolfgang Maier",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_import_search.html"
  },
  "/faqs/galaxy/workflows_parameter_set_runtime.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Setting parameters at run-time",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"
  },
  "/faqs/galaxy/workflows_publish.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Make a workflow public",
    "contributors": "Bérénice Batut",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_publish.html"
  },
  "/faqs/galaxy/workflows_rename_output.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Renaming workflow outputs",
    "contributors": "Saskia Hiltemann, Wolfgang Maier, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_rename_output.html"
  },
  "/faqs/galaxy/workflows_report_view.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Viewing a workflow report",
    "contributors": "Helena Rasche, Saskia Hiltemann",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_report_view.html"
  },
  "/faqs/galaxy/workflows_run.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Running a workflow",
    "contributors": "Saskia Hiltemann, Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_run.html"
  },
  "/faqs/galaxy/workflows_run_ds.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing and Launching a Dockstore Workflow",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_run_ds.html"
  },
  "/faqs/galaxy/workflows_run_trs.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing and launching a GTN workflow",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_run_trs.html"
  },
  "/faqs/galaxy/workflows_run_wfh.html": {
    "type": "FAQ",
    "topic": "FAQ",
    "title": "Importing and Launching a WorkflowHub.eu Workflow",
    "contributors": "Helena Rasche",
    "tags": [

    ],
    "url": "/training-material/faqs/galaxy/workflows_run_wfh.html"
  }
};

function search(idx, q, includeFaqs, includeTutorials){
	if(q.length > 2){
        var results_partial = idx.search(`*${q}*`),
            results_exact = idx.search(`${q}`),
            results_fuzzy = idx.search(`${q}~3`);

	// Include search term in page title
		document.getElementsByTagName("title")[0].innerText = `${q} | GTN Tutorial Search`

        thereMap  = Object.assign({}, ...results_partial.map((x) => ({[x.ref]: x.score})));

        results_exact.forEach(x => {
            if(thereMap[x.ref] !== undefined){
                if(thereMap[x.ref] < x.score + 4){
                    thereMap[x.ref] = x.score + 4
                }
            } else {
                    thereMap[x.ref] = x.score + 4
            }
        })
        results_fuzzy.forEach(x => {
            if(thereMap[x.ref] !== undefined){
                if(thereMap[x.ref] < x.score - 2){
                    thereMap[x.ref] = x.score - 2
                }
            } else {
                    thereMap[x.ref] = x.score - 2
            }
        })

        combined_results = Object.getOwnPropertyNames(thereMap);
        combined_results.sort((a, b) => {
            if (thereMap[a] > thereMap[b]) {
                return -1;
            }
            if (thereMap[a] < thereMap[b]) {
                return 1;
            }
            return 0;
        });

		var results_final = combined_results.map(x => {
			return resources['/' + x.replaceAll(".md", ".html")];
		}).filter(x => x !== undefined);

		if(! includeFaqs) {
			results_final = results_final.filter(x => x.type != 'FAQ')
		}
		if(! includeTutorials) {
			results_final = results_final.filter(x => x.type != 'Tutorial')
		}

        $("#results-container").html(results_final.map(x => `
        <div class='col-sm-6'>
        <div class='card'>
        <div class='card-body'>
          <h5 class='card-title'>${x.title}</h5>
          <h6 class='card-subtitle text-muted'>${x.topic}</h6>
          <p>${x.tags.join(' ')}</p>
          <p>${x.contributors}</p>
          <a class='btn btn-primary' href='${x.url}'>View ${x.type}</a>
          </div>
          </div>
          </div>
                    `));
	}
}

function searchWrap(idx) {
	console.log(
	'search',
		$("#search-input").val(),
		$("input[name='search-faqs']").is(':checked'),
		$("input[name='search-tutorials']").is(':checked')
	)
	search(idx,
		$("#search-input").val(),
		$("input[name='search-faqs']").is(':checked'),
		$("input[name='search-tutorials']").is(':checked')
	);
}

fetch('/training-material/search.json')
	.then(response => response.json())
	.then(data => {
		var idx = lunr.Index.load(data);

		var  params = (new URL(document.location)).searchParams;
		paramQuery = params.get('query');
		if(paramQuery){
			document.getElementById('search-input').value = paramQuery;
			searchWrap(idx);
		}

		$("#search-input").on("change keyup paste", function(){
			searchWrap(idx);
		})

		$("input[name='search-faqs']:checkbox").change(
			function(){
				searchWrap(idx);
			}
		);

		$("input[name='search-tutorials']:checkbox").change(
			function(){
				searchWrap(idx);
			}
		);

});
</script>



</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/search.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/search.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/8cb9a3a5113d0b6a5b4374ea1865adbe76b9da68">8cb9a3a</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>
