---
layout: tutorial_hands_on
topic_name: epigenetics
tutorial_name: ewas-suite
---

# Introduction
{:.no_toc}

<!-- This is a comment. -->

The field of cancer genomics has demonstrated the power of massively parallel sequencing techniques to inform on the genes and specific alterations that drive tumor onset and progression. Although large comprehensive sequence data sets continue to be made increasingly available, data analysis remains an ongoing challenge, particularly for laboratories lacking dedicated resources and bioinformatics expertise. To address this, we have provide training based on  Galaxy tools ewas suite that represent many popular algorithms for detecting somatic genetic alterations from cancer genome and exome data.  

This exercise uses the dataset from the Cell publication by Hugo, Willy, et al., 2015 (DOI: 10.1016/j.cell.2015.07.061). The goal of this article was to identify differentially methylated regions and positions associated with melanoma MAPKi resistance.

To this end, Illumina 450K Methylation array  was performed in melanoma tumors from example patient pre and post MAPKi and BRAFi treatment with  different outcomes (sensitive and resistant). For each sample there is raw green and red colour arrays containing the summarized bead information generated bythe Illumina 450k scanner, i.e. there are 8 files in total:

Accession  | Sensitivity | Treatment
--- | ---  | ---
GSM1588704 | sensitive   | MAPKi
GSM1588705 | sensitive   | MAPKi
GSM1588706 | resistant   | BRAFi
GSM1588707 | resistant   | BRAFi

> ### Agenda
>
> 1. TOC
>{:toc}
> 
{: .agenda}

# Step 1: Illumina Array Data Loading

The first step of EWAS data anylalysis is raw methylation data loading (intensity information files  for each two color micro array)

> ### {% icon hands_on %} Hands-on: Data Loading
>
> 1. Create a new history for this tutorial and give it a proper name
> 2. Import Grn and Red .IDAT files from [Zenodo](https://zenodo.org/record/1251211#.WwREQ1Mvz-Y)
> 3. Run **minfi read450k**  {% icon tool %} with red and green .IDAT files i.e. `red files:` GSM1588704_8795207135_R01C02_Red.idat, GSM1588705_8795207119_R05C02_Red.idat, GSM1588706_8795207135_R02C02_Red.idat, GSM1588707_8795207119_R06C02_Red.idat and `green files:` GSM1588704_8795207135_R01C02_Grn.idat, GSM1588705_8795207119_R05C02_Grn.idat, GSM1588706_8795207135_R02C02_Grn.idat, GSM1588707_8795207119_R06C02_Grn.idat
> 4. Inspect generated set of data
{: .hands_on}
> ### {% icon question %} Questions
>
> 1. How are the Green and Red signals are stored?
>  <details>
>    <summary>Click to view answers</summary>
>    <ol type="1">
>    <li> Green and Red micro arrays are builded up into `RGChannelSet` class</li>
>    </ol>
>   </details>
>  {: .question}  

# Step 2: Preprocessing and  Quality Assessment
Preprocessing and data quality assurance is an important step in Infinium Methylation Assay analysis. 
`RGChannelSet` represents two color data with a green and a red channel and can be converted into methylated and unmethylated signals assigned to `MethylSet` or into Beta values build in `RatioSet`. User can convert from `RGChannelSet` into `MethylSet` using the **minfi mset** {% icon tool %} or compute Beta values using **minfi set** {% icon tool %}. The **minfi qc** {% icon tool %} tool extracts and plots the quality control data frame with two columns mMed and uMed which are the medians of `MethylSet` signals (Meth and Unmeth).Comparing them against one another allows user to detect and remove low-quality samples. 
> ### {% icon hands_on %} Hands-on: Preprocessing
> 1. Run **minfi mset** {% icon tool %} to create `MethylSet` object
> 2. Run **minfi qc** {% icon tool %} to estimate sample-specific quality control
> 3. Convert methylation data from the `MethylSet` , to ratios with **minfi rset** {% icon tool %}
> 4. Then map ratio data to the genome using **minfi maptogenome** {% icon tool %} tool  
> 
> ### {% icon tip %} Tip: Preprocess and Normalize data
> 
> If your files require normalisation, you might prefer to use other of preprocessing tools provided in EWAS suite i.e. **minfi ppfun** {% icon tool %} or **minfi ppquantile**  {% icon tool %} look for recomendation at (ref).
>    {: .tip}
> 
{: .hands_on}
# Step 3: Removing probes affected by genetic variation
Incomplete annotation of genetic variations such as single nucleotide polymorphism (SNP) may affect DNA measurements and interfere results from downstream analysis. 
> ### {% icon hands_on %} Hands-on: Removing probes affected by genetic variation
> 1. Run **minfi dropsnp** {% icon tool %} to remove the probes that contain either a SNP at the metylated loci interrogation or at the single nucleotide extension (highly recommended /ref)
> 
> 
# Step 4: DMPs and DMRs Identification
The main goal of the EWAS suite is to simplify the way differentially methylated loci sites are detected. The EWAS suite contains **minfi dmp** {% icon tool %} tool detecting differentially methylated positions (DMPs) with respect to a phenotype covariate, and more complex **minfi dmr** {% icon tool %} solution for finding differentially methylated regions (DMRs). Genomic regions that are differentially methylated between two conditions can be tracked using a bump hunting algorithm. The algorithm first implements a t-statistic at each methylated loci location, with optional smoothing, then groups probes into clusters with a maximum location gap and a cutoff size to refer the lowest possible value of genomic profile hunted by our tool.
> ### {% icon question %} Questions
>
> 1. How are we define phenotype covariate?
>  <details>
>    <summary>Click to view answers</summary>
>    <ol type="1">
>    <li>Phenotype covariate is the set of observable characteristics of an individual resulting from the gene-environment interactions.</li>
>    </ol>
>   </details>
>  {: .question}
> ### {% icon hands_on %} Hands-on: DMPs and DMRs Identification
> 1. Import phenotypeTable.txt from [Zenodo](https://zenodo.org/record/1251211#.WwREQ1Mvz-Y)
> 2. Run **minfi dmp** {% icon tool %} with 
    -  `GenomicRatioSet` from Step 3  and tab_factor.txt
    -  Phenotype Type = categorical
    -  qCutoff Size = 0.5 (DMPs with an FDR q-value greater than this will not be returned)
    -  Variance Shrinkage = TRUE (is recommended when sample sizes are small <10)
> 3. Run **minfi dmr** {% icon tool %} 
    -  `GenomicRatioSet` from Step 3 and tab_factor.txt
    -  factor1 = sensitive factor2 = resistant (factor1 vs factor2)
    -  maxGap Size = 250
    - coef Size = 2
    - Cutoff Size = 0.1
    - nullMethod = permutation
    - verbose = TRUE
> ### {% icon tip %} Tip: Preprocess and Normalize data
> 
> Phenotype table can be in diffrent size with diffrent arguments only second column is required to contain phenotype covariate information for each sample. 
>    {: .tip}
>
# Step 5: Annotation and Visualization
In addition to downstream analysis users can annotate the differentially methylated loci to the promoter regions of genes with gene function description, and relationships between these concepts.
> ### {% icon hands_on %} Hands-on:  Annotate Differentially Methylated Position
> 1. Run **chipeakanno annopeaks** {% icon tool %} with
  - Differentially Methylated Positions from Step 4
  - **bindingType: StartSite**
  - promoter region defined as upstream **bindingRegionStart:-5000** 
  - downstream 3000 from TSS (transcription start sites) **bindingRegionEnd:3000**
  - **Additional Column of Score:8** optional value if it is required 
> 2. **Cut** {% icon tool %}  "gene_name" Column from Table of Annotated Peaks to get List of Genes
  - Cut columns: c16
  - Delimited by: Tab
> 3. **Remove beginning** {% icon tool %} of Gene List
  - Remove first: 1
> 4. Convert List of Genes to List of  entrez ID using **clusterProfiler bitr** {% icon tool %}
  - Input Type Gene ID: SYMBOL
  - Output Type Gene ID: ENTREZID
 5. Run GO Enrichment Analysis of a gene set with **clusterProfiler go** {% icon tool %}
{: .hands_on}
> # {% icon tip %} Tip:Biological Id TRanslator
No matter what ID we need, this tool allows various conversions suitable for different databases and annotations.
>    {: .tip}
> ![GO Enrichment Analysis Visualization](../../images/functionalanno.png)

ID  | Description | pvalue | qvalue | geneID | Count
--- | ---  | --- | --- | --- | --- 
GO:0048732 | gland development  | 1.38E-58 | 4.23E-55 | PTGS2/KCNC1/FZD1/SLC22A18/SLC22A3(...) | 372
GO:1901652 | response to peptide | 3.99E-57 | 8.13E-54 | SULF1/LAMA5/MED1/CFLAR/MSX2(...) | 359
GO:0048545 | response to steroid hormone | 1.38EE-54 | 2.11E-51 | HDAC9/RAB10/CFLAR/WDTC1(...) | 394


# Conclusion
{:.no_toc}

Epigenetic aberrations which involve DNA modifications give researchers an interest to identify novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and  time consuming computational analysis. Here, the EWAS suite becomes a solution for this inconvenience and provides a simplified downstream analysis available as ready to run pipline in supplementary materials. 
