---
layout: tutorial_hands_on
topic_name: transcriptomics
tutorial_name: scrna
---

# Introduction
{:.no_toc}

The advent of single-cell RNA sequencing has provided the means to explore samples at the individual cell level, enabling a greater understanding of the development and function of such samples by the characteristics of their constituent cells. These cells can be classified into different cell subpopulations or "clusters" by the variation of gene expression in each cell, where cells in the same cluster exhibit the same levels of differential expression in the same set of related genes.

By identifying significant genes in each cluster, cell types can be inferred and similarity between clusters can be determined based on their proximity to one another. If the cells all come from the same sample tissue, the cell types will likely correspond to the different stages of cell differentiation expected of that tissue, wherein a lineage/heirarchy could potentially be derived.

This tutorial is in part inspired by aspects of the [Hemberg workflow](https://hemberg-lab.github.io/scRNA.seq.course/) at the Sangar insititute, as well as the [CGATOxford](https://github.com/CGATOxford/UMI-tools) workflow. The barcoding follows the CelSeq2 protocol and uses the lane configuration as that utilized by the Freiburg MPI GrÃ¼n lab.

> ### Agenda
>
> In this tutorial, we will deal with:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Analyse 

Introduction about this part

## Subpart 1

Short introduction about this subpart.

<!--
{% icon hands_on %} will render the hands_on icon as specified in
_config.yml in the root of this repository.
-->

> ### {% icon hands_on %} Hands-on: Data uplodae
>
> 1. Step1
> 2. Step2
>
>    > ### {% icon comment %} Comments
>    > A comment
>    {: .comment}
>
>    > ### {% icon tip %}Tip: A tip
>    >
>    > * Step1
>    > * Step2
>    {: .tip}
{: .hands_on}

## Subpart 2

Short introduction about this subpart.

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. Step2
>
>    > ### {% icon question %} Question
>    >
>    > Question?
>    >
>    > > ### {% icon solution %} Solution
>    > >
>    > > Answer to question
>    > >
>    > {: .solution}
>    >
>    {: .question}
{: .hands_on}

Some blabla
> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. **My Tool** {% icon tool %} with the following parameters
>   - *"param1"*: the file `myfile`
>   - *"param2"*: `42`
>   - *"param3"*: `Yes`
>
> 3. **My Tool** {% icon tool %} with the following parameters
>   - {% icon param-text %} *"My text parameter"*: `my value`
>   - {% icon param-file %} *"My input file"*: `my file`
>   - {% icon param-files %} *"My multiple file input or collection"*: `my collection`
>   - {% icon param-select %} *"My select menu"*: `my choice`
>   - {% icon param-check %} *"My check box"*: `yes`
>
>    > ### {% icon question %} Questions
>    >
>    > 1. Question1?
>    > 2. Question2?
>    >
>    > > ### {% icon solution %} Solution
>    > >
>    > > 1. Answer for question1
>    > > 2. Answer for question2
>    > >
>    > {: .solution}
>    >
>    {: .question}
>
> 3. Step3
{: .hands_on}

# Part 2

Short introduction about this subpart.

> ### {% icon comment %} Comment
>
> Do you want to learn more about the principles behind mapping? Follow our [training](../../NGS-mapping)
{: .comment}


> ### {% icon details %} More details on the ....
>
> Add more details in Markdown. By default the box is collapsed. And is expanded when clicked
>
{: .details}

# Conclusion
{:.no_toc}

Conclusion about the technical key points. And then relation between the techniques and the biological question to end with a global view.
