<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Metabarcoding/eDNA through Obitools</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Learn to analyse Ecological data through Galaxy." />
        <meta property="og:title" content="Galaxy Training: Metabarcoding/eDNA through Obitools" />
        <meta property="og:description" content="Learn to analyse Ecological data through Galaxy." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/ecology" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Ecology
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/ecology/faqs/" title="Check our FAQs for the Ecology topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-03-01/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html" title="Version 2022-03-01">2022-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-02-01/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html" title="Version 2022-02-01">2022-02-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-01-01/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html" title="Version 2022-01-01">2022-01-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:42 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Metabarcoding/eDNA through Obitools",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Ecology",
    "description": "Learn to analyse Ecological data through Galaxy.",
    "url": "https://training.galaxyproject.org//training-material/topics/ecology/"
  },
  "courseCode": "ecology / Obitools-metabarcoding / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Metabarcoding/eDNA through Obitools' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - how to analyze DNA metabarcoding / eDNA data produced on Illumina sequencers using the OBITools?\n\n\\nThe objectives are:\n - Deal with paired-end data to create consensus sequences\n - Clean, filter and anlayse data to obtain strong results\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Coline Royaux"
    },
    {
      "@type": "Person",
      "name": "Olivier Norvez"
    },
    {
      "@type": "Person",
      "name": "Eric Coissac"
    },
    {
      "@type": "Person",
      "name": "Fr√©d√©ric Boyer"
    },
    {
      "@type": "Person",
      "name": "Yvan Le Bras"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Coline Royaux"
    },
    {
      "@type": "Person",
      "name": "Olivier Norvez"
    },
    {
      "@type": "Person",
      "name": "Eric Coissac"
    },
    {
      "@type": "Person",
      "name": "Fr√©d√©ric Boyer"
    },
    {
      "@type": "Person",
      "name": "Yvan Le Bras"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Ecology",
      "description": "Learn to analyse Ecological data through Galaxy.",
      "url": "https://training.galaxyproject.org//training-material/topics/ecology/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0610",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0610",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0610"
    }
  ]
}
</script>

<section class="tutorial topic-ecology">
    <h1 data-toc-skip>Metabarcoding/eDNA through Obitools</h1>
    


    <div class="contributors-line">Authors: 
<a href="/training-material/hall-of-fame/colineroyaux/" class="contributor-badge contributor-colineroyaux"><img src="https://avatars.githubusercontent.com/colineroyaux?s=27" alt="Avatar">Coline Royaux</a>


<a href="/training-material/hall-of-fame/onorvez/" class="contributor-badge contributor-onorvez"><img src="https://avatars.githubusercontent.com/onorvez?s=27" alt="Avatar">Olivier Norvez</a>


<a href="/training-material/hall-of-fame/ecoissac/" class="contributor-badge contributor-ecoissac"><img src="https://avatars.githubusercontent.com/ecoissac?s=27" alt="Avatar">Eric Coissac</a>


<a href="/training-material/hall-of-fame/fboyer/" class="contributor-badge contributor-fboyer"><img src="https://avatars.githubusercontent.com/fboyer?s=27" alt="Avatar">Fr√©d√©ric Boyer</a>


<a href="/training-material/hall-of-fame/yvanlebras/" class="contributor-badge contributor-yvanlebras"><img src="https://avatars.githubusercontent.com/yvanlebras?s=27" alt="Avatar">Yvan Le Bras</a>


<a href="https://training.galaxyproject.org/training-material/topics/contributing/" class="contributor-badge contributor-newcontributors"><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span>Add Contributions!</a>

</div>


    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>how to analyze DNA metabarcoding / eDNA data produced on Illumina sequencers using the OBITools?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Deal with paired-end data to create consensus sequences</p>
</li>
        
        <li><p>Clean, filter and anlayse data to obtain strong results</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/5932108/files/wolf_tutorial.zip?download=1">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/ecology/tutorials/Obitools-metabarcoding/workflows/" title="Workflows" alt="Metabarcoding/eDNA through Obitools workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











  
  <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/ecology/Obitools-metabarcoding/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (March 2022)
            </a>
        </li>
        
        
        
    </ul>
   </li>
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/ecology/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://ecology.usegalaxy.eu/" title="Galaxy for Ecology">
            Galaxy for Ecology
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Feb 28, 2022 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Based on this <a href="https://pythonhosted.org/OBITools/wolves.html">OBITools official tutorial</a>, you will learn here how to analyze <span class="notranslate">DNA</span> metabarcoding data produced on Illumina sequencers using:</p>
<ul>
  <li>the OBITools on <span class="notranslate">Galaxy</span></li>
  <li>some classical <span class="notranslate">Galaxy</span> tools</li>
</ul>

<p>The data used in this tutorial correspond to the analysis of four wolf scats, using the protocol published in <span class="citation"><a href="#Shehzad2012">SHEHZAD <i>et al.</i> 2012</a></span> for assessing carnivore diet. After extracting <span class="notranslate">DNA</span> from the faeces, the <span class="notranslate">DNA</span> amplifications were carried out using the primers TTAGATACCCCACTATGC and TAGAACAGGCTCCTCTAG amplifying the 12S-V5 region (<span class="citation"><a href="#Riaz2011">Riaz <i>et al.</i> 2011</a></span>), together with a wolf blocking oligonucleotide.</p>

<p>It is always a good idea to have a look at the intermediate results or to evaluate the best parameter for each step. Some commands are designed for that purpose, for example you can use :</p>

<ul>
  <li>obicount to count the number of sequence records in a file</li>
  <li>obihead and obitail to view the first or last sequence records of a file</li>
  <li>obistat to get some basic statistics (count, mean, standard deviation) on the attributes (key=value combinations) in the header of each sequence record (see The <a href="https://pythonhosted.org/OBITools/fasta.html">extended OBITools fasta format in the fasta format description</a>)</li>
  <li>any <span class="notranslate">Galaxy</span> tools corresponding to classical unix command such as less, awk, sort, wc to check your files.</li>
</ul>

<p>The OBITools programs imitate Unix standard programs because they usually act as filters. The main difference with classical Unix programs is that text files are not analyzed line per line but sequence record per sequence record (see below for a detailed description of a sequence record). Compared to packages for similar purposes like mothur (<span class="citation"><a href="#Schloss2009">Schloss <i>et al.</i> 2009</a></span>) or QIIME (<span class="citation"><a href="#Caporaso2010">Caporaso <i>et al.</i> 2010</a></span>), the OBITools mainly rely on filtering and sorting algorithms. This allows users to set up versatile data analysis pipelines</p>

<p><img src="../../images/Obitools/fig-Pipeline.png" alt="OBITools pipeline. " loading="lazy" /></p>

<p>Most of the OBITools commands read sequence records from a file or from the stdin, make some computations on the sequence records and output annotated sequence records. As inputs, the OBITools are able to automatically recognize the most common sequence file formats (i.e. FASTA, FASTQ, EMBL, and GenBank). They are also able to read ecoPCR (<span class="citation"><a href="#Ficetola2010">Ficetola <i>et al.</i> 2010</a></span>) result files and ecoPCR/ecoPrimers formatted sequence databases (<span class="citation"><a href="#Riaz2011">Riaz <i>et al.</i> 2011</a></span>) as ordinary sequence files. File format outputs are more limited. By default, sequences without and with quality information are written in FASTA and FASTQ formats, respectively. However, dedicated options allow enforcing the output format, and the OBITools are also able to write sequences in the ecoPCR/ecoPrimers database format, to produce reference databases for these programs. In FASTA or FASTQ format, the attributes are written in the header line just after the id, following a key=value; format.</p>

<p><img src="../../images/Obitools/fig-Record.png" alt="OBITools attributes records. " loading="lazy" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#manage-input-data" id="markdown-toc-manage-input-data">Manage input data</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
      <li><a href="#unzip-the-downloaded-archive" id="markdown-toc-unzip-the-downloaded-archive"><strong>Unzip</strong> the downloaded archive</a></li>
    </ol>
  </li>
  <li><a href="#use-obitools" id="markdown-toc-use-obitools">Use OBITools</a>    <ol>
      <li><a href="#micro-assembly-of-paired-end-sequences-with-illuminapairedend" id="markdown-toc-micro-assembly-of-paired-end-sequences-with-illuminapairedend">Micro assembly of paired-end sequences with <strong>illuminapairedend</strong></a></li>
      <li><a href="#remove-unaligned-sequence-records-with-obigrep" id="markdown-toc-remove-unaligned-sequence-records-with-obigrep">Remove unaligned sequence records with <strong>obigrep</strong></a></li>
      <li><a href="#assign-each-sequence-record-to-the-corresponding-samplemarker-combination-with-ngsfilter" id="markdown-toc-assign-each-sequence-record-to-the-corresponding-samplemarker-combination-with-ngsfilter">Assign each sequence record to the corresponding sample/marker combination with <strong>NGSfilter</strong></a></li>
      <li><a href="#dereplicate-reads-into-uniq-sequences-with-obiuniq" id="markdown-toc-dereplicate-reads-into-uniq-sequences-with-obiuniq">Dereplicate <span class="notranslate">reads</span> into uniq sequences with <strong>obiuniq</strong></a></li>
      <li><a href="#limit-number-of-informations-with-obiannotate" id="markdown-toc-limit-number-of-informations-with-obiannotate">Limit number of informations with <strong>obiannotate</strong></a></li>
      <li><a href="#computes-basic-statistics-for-attribute-values-with-obistat" id="markdown-toc-computes-basic-statistics-for-attribute-values-with-obistat">Computes basic statistics for attribute values with <strong>obistat</strong></a></li>
      <li><a href="#filtering-sequances-by-count-and-length-with-obigrep" id="markdown-toc-filtering-sequances-by-count-and-length-with-obigrep">Filtering sequances by count and length with <strong>obigrep</strong></a></li>
      <li><a href="#clean-the-sequences-for-pcrsequencing-errors-sequence-variants-with-obiclean" id="markdown-toc-clean-the-sequences-for-pcrsequencing-errors-sequence-variants-with-obiclean">Clean the sequences for PCR/sequencing errors (sequence variants) with <strong>obiclean</strong></a></li>
      <li><a href="#taxonomic-assignment-of-sequences-with-ncbi-blast-blastn" id="markdown-toc-taxonomic-assignment-of-sequences-with-ncbi-blast-blastn">Taxonomic assignment of sequences with <strong>NCBI BLAST+ blastn</strong></a></li>
      <li><a href="#filter-database-and-query-sequences-by-id-to-re-associate-informations-with-filter-sequences-by-id" id="markdown-toc-filter-database-and-query-sequences-by-id-to-re-associate-informations-with-filter-sequences-by-id">Filter database and query sequences by ID to re associate informations with <strong>Filter sequences by ID</strong></a></li>
      <li><a href="#from-fasta-to-tabular-with-obitab" id="markdown-toc-from-fasta-to-tabular-with-obitab">From FASTA to tabular with <strong>Obitab</strong></a></li>
      <li><a href="#create-a-final-synthesis-as-a-tabular-file" id="markdown-toc-create-a-final-synthesis-as-a-tabular-file">create a final synthesis as a tabular file</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="manage-input-data">Manage input data</h1>
<p>The data needed to run the tutorial are the following:</p>

<ul>
  <li>FASTQ files resulting from a GA IIx (Illumina) paired-end (2 x 108 bp) sequencing assay of <span class="notranslate">DNA</span> extracted and amplified from four wolf faeces:
    <ul>
      <li>wolf_F.fastq</li>
      <li>wolf_R.fastq</li>
    </ul>
  </li>
  <li>the file describing the primers and tags used for all samples sequenced:
    <ul>
      <li>wolf_diet_ngsfilter.txt</li>
    </ul>
  </li>
  <li>the tags correspond to short and specific sequences added on the 5‚Äô end of each primer to distinguish the different samples</li>
  <li>the file containing the reference database in a fasta format:
  -db_v05_r117.fasta This reference database has been extracted from the release 117 of EMBL using ecoPCR</li>
</ul>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the zip archive containing input files from <a href="https://zenodo.org/record/5932108/files/wolf_tutorial.zip?download=1">Zenodo</a></p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/5932108/files/wolf_tutorial.zip
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
    <li>Rename the dataset, here a zip archive, if needed</li>
    <li>
      <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">zip</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select <code class="language-plaintext highlighter-rouge">datatypes</code></li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="unzip-the-downloaded-archive"><strong>Unzip</strong> the downloaded archive</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-unzip-the-downladed-zip-archive-and-prepare-unzipped-files-to-be-used-by-obitools"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Unzip the downladed .zip archive and prepare unzipped files to be used by OBITools</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/imgteam/unzip/unzip/0.2" title="Tested with toolshed.g2.bx.psu.edu/repos/imgteam/unzip/unzip/0.2"><strong>Unzip</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/imgteam/unzip/unzip/0.2</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúExtract single file‚Äù</em>: <code class="language-plaintext highlighter-rouge">All files</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>To work properly, this unzip <span class="notranslate">Galaxy</span> tool is waiting ‚Äúsimple‚Äù archive as input, this means without sub directory.</p>
      </blockquote>
    </li>
    <li>
      <p>Add to each datafile a tag and/or modify names (<em>optional</em>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>
    </li>
    <li>
      <p>Unhide all dataset from the resulting data collection so you can use these files independently.</p>
    </li>
    <li>Modify datatype from txt to tabular for the <code class="language-plaintext highlighter-rouge">wolf_diet_ngsfilter</code> dataset</li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Why do we need to unhide manually datasets from the data collection?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Data collection is a functionality often used to deal with multiple datasets on the same format who can be analysed in batch mode. Here, the data collection is populated with heterogenous datafiles, coming from an archive. We thus need to treat separately each dataset of the collection, and to do so, we need to unhide corresponding datasets from the history, as datasets inside collections are just like ‚Äúsymbolic link‚Äù to ‚Äúclassical‚Äù history datasets hidden by default.</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="use-obitools">Use OBITools</h1>

<p>OBITools (<span class="citation"><a href="#Boyer2015">Boyer <i>et al.</i> 2015</a></span>) is a set of programs specifically designed for analyzing NGS data in a <span class="notranslate">DNA</span> metabarcoding context, taking into account taxonomic information. It is distributed as an open source software available on the following website: <a href="http://metabarcoding.org/obitools">http://metabarcoding.org/obitools</a>.</p>

<p>The OBITools commands consider a sequence record as an entity composed of five distinct elements. Two of them are mandatory, the identifier (id) and the <span class="notranslate">DNA</span> or protein sequence itself. The id is a single word composed of characters, digits, and other symbols like dots or underscores excluding spaces. Formally, the ids should be unique within a dataset and should identify each sequence record unambiguously, but only a few OBITools actually rely on this property. The three other elements composing a sequence record are optional. They consist in a sequence definition, a quality vector, and a set of attributes. The last element is a set of attributes qualifying the sequence, each attribute being described by a key=value pair. The set of attributes is the central concept of the OBITools system. When an OBITools command is run on the sequence records included in a dataset, the result of the computation often consist in the addition of new attributes completing the annotation of each sequence record. This strategy of sequence annotation allows the OBITools to return their results as a new sequence record file that can be used as the input of another OBITools program, ultimately creating complex pipelines (source: <a href="https://pythonhosted.org/OBITools/welcome.html">OBITools Welcome</a>).</p>

<h2 id="micro-assembly-of-paired-end-sequences-with-illuminapairedend">Micro assembly of paired-end sequences with <strong>illuminapairedend</strong></h2>

<p>When using the result of a paired-end sequencing assay with supposedly overlapping forward and reverse <span class="notranslate">reads</span>, the first step is to recover the assembled sequence.</p>

<p>The forward and reverse <span class="notranslate">reads</span> of the same fragment are at the same line position in the two FASTQ files obtained after sequencing. Based on these two files, the assembly of the forward and reverse <span class="notranslate">reads</span> is done with the <strong>illuminapairedend</strong> utility that aligns the two <span class="notranslate">reads</span> and returns the reconstructed sequence.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-recover-consensus-sequences-from-overlapping-forward-and-reverse-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Recover consensus sequences from overlapping forward and reverse <span class="notranslate">reads</span>.</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_illuminapairedend/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_illuminapairedend/1.2.13"><strong>illuminapairedend</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_grep/obi_illuminapairedend/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúRead from file‚Äù</em>: <code class="language-plaintext highlighter-rouge">wolf_F</code> for the 3p file</li>
        <li><em>‚ÄúRead from file‚Äù</em>: <code class="language-plaintext highlighter-rouge">wolf_R</code> for the 5p file</li>
        <li><em>‚Äúminimum score for keeping aligment‚Äù</em>: <code class="language-plaintext highlighter-rouge">40.0</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Sequence records corresponding to the same read pair must be in the same order in the two files !</p>

        <p>If the alignment score is below the defined score, here 40, the forward and reverse <span class="notranslate">reads</span> are not aligned but concatenated, and the value of the mode attribute in the sequence header is set to joined instead of alignment</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="remove-unaligned-sequence-records-with-obigrep">Remove unaligned sequence records with <strong>obigrep</strong></h2>

<p>In this step we are going to use the value of the mode attribute in the sequence header of the illuminapairedend output file to discard sequences indicated as ‚Äújoined‚Äù, so not assembled (‚Äúalignment‚Äù) (see explanation about this mode on the previous step)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-unaligned-sequence-records"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove unaligned sequence records</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13"><strong>obigrep</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">ilumimnapairedend fastq groomer output file</code></li>
        <li><em>‚ÄúChoose the sequence record selection option‚Äù</em>: <code class="language-plaintext highlighter-rouge">predicat</code>
          <ul>
            <li><em>‚ÄúPython boolean <span class="notranslate">expression</span> to be evaluated for each sequence record.‚Äù</em>: <code class="language-plaintext highlighter-rouge">mode!="joined"</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-tip-verifying-fastq-format-and-converting-it"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Verifying FastQ format and converting it</h3>

        <ul>
          <li>You can use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+<span class="notranslate">galaxy</span>0</span></span>  to look at the format of the sequencing encoding score then <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/1.1.5" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/1.1.5"><strong>FASTQ GROOMER</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/1.1.5</span></span>  to specify the guessed sequencing encoding score and create a fastqsanger file.</li>
        </ul>

      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>The obigrep command is in some way analog to the standard Unix grep command. It selects a subset of sequence records from a sequence file.</p>

        <p>A sequence record is a complex object composed of an identifier, a set of attributes (key=value), a definition, and the sequence itself.</p>

        <p>Instead of working text line by text line as the standard Unix tool, selection is done sequence record by sequence record. A large set of options allows refining selection on any of the sequence record elements.</p>

        <p>Moreover obigrep allows specifying simultaneously several conditions (that take the value TRUE or FALSE) and only the sequence records that fulfill all the conditions (all conditions are TRUE) are selected. You can refer to <a href="https://pythonhosted.org/OBITools/scripts/obigrep.html">https://pythonhosted.org/OBITools/scripts/obigrep.html</a> for more details</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How do you verify the operation is successful?</li>
    <li>How many sequences are kept? Discarded?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>you can search in the input file content the presence of <code class="language-plaintext highlighter-rouge">mode=joined</code> and same on the output file (just clicking the eye to visualize the content of each file and typing CTRL+C for example to search <code class="language-plaintext highlighter-rouge">mode=joined</code> in the file, or using a regex <span class="notranslate">Galaxy</span> tool for example). You can also at least look at the size of the output file, if smaller than input file, this is a first good indication.</li>
      <li>You can use a <span class="notranslate">Galaxy</span> tool like <code class="language-plaintext highlighter-rouge">Line/Word/Character count of a dataset</code> to count the number of lines of each dataset (input and output of obigrep) and divided by 4 (as in a FastQ file, each sequence is represented by a block of 4 lines). 45 276 sequences for input file. 44 717 for output file. Thus 559 sequences discarded.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="assign-each-sequence-record-to-the-corresponding-samplemarker-combination-with-ngsfilter">Assign each sequence record to the corresponding sample/marker combination with <strong>NGSfilter</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assigns-sequence-records-to-the-corresponding-experimentsample-based-on-dna-tags-and-primers"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Assigns sequence records to the corresponding experiment/sample based on <span class="notranslate">DNA</span> tags and primers</h3>

  <blockquote class="comment">
    <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>Each sequence record is assigned to its corresponding sample and marker using the data provided in a text file (here wolf_diet_ngsfilter.txt). This text file contains one line per sample, with the name of the experiment (several experiments can be included in the same file), the name of the tags (for example: aattaac if the same tag has been used on each extremity of the PCR products, or aattaac:gaagtag if the tags were different), the sequence of the forward primer, the sequence of the reverse primer, the letter T or F for sample identification using the forward primer and tag only or using both primers and both tags, respectively.</p>
  </blockquote>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_ngsfilter/obi_ngsfilter/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_ngsfilter/obi_ngsfilter/1.2.13"><strong>NGSfilter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_ngsfilter/obi_ngsfilter/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúParameter file‚Äù</em>: <code class="language-plaintext highlighter-rouge">wolf_diet_ngsfilter</code></li>
        <li><em>‚ÄúRead from file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obigrep output</code></li>
        <li><em>‚ÄúNumber of errors allowed for matching primers‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
        <li><em>‚ÄúOutput data type‚Äù</em>: <code class="language-plaintext highlighter-rouge">fastq</code></li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-tip-be-sure-the-text-file-is-in-tabular-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Be sure the text file is in tabular datatype</h3>

        <ul>
          <li>If you are sure the format is compatible with a tabular datatype, as this is the case here ;), you can manually change it, clicking on the eye of the ‚Äúwolf_diet_ngsfilter.txt‚Äù dataset, then selecting the ‚ÄúDatatypes‚Äù section then affecting manually tabular and saving the operation</li>
        </ul>

      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many sequences are not assigned?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>1391</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="dereplicate-reads-into-uniq-sequences-with-obiuniq">Dereplicate <span class="notranslate">reads</span> into uniq sequences with <strong>obiuniq</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-groups-together-sequence-records"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Groups together sequence records</h3>

  <blockquote class="comment">
    <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>The same <span class="notranslate">DNA</span> molecule can be sequenced several times. In order to reduce both file size and computations time, and to get easier interpretable results, it is convenient to work with unique sequences instead of <span class="notranslate">reads</span>. To dereplicate such <span class="notranslate">reads</span> into unique sequences, we use the obiuniq command.
Definition: Dereplicate <span class="notranslate">reads</span> into unique sequences</p>
    <ul>
      <li>compare all the <span class="notranslate">reads</span> in a data set to each other</li>
      <li>group strictly identical <span class="notranslate">reads</span> together</li>
      <li>output the sequence for each group and its count in the original dataset (in this way, all duplicated <span class="notranslate">reads</span> are removed)
<em>Definition adapted from Seguritan and Rohwer (2001)</em></li>
    </ul>
  </blockquote>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_uniq/obi_uniq/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_uniq/obi_uniq/1.2.13"><strong>obiuniq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_uniq/obi_uniq/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Trimmed and annotated file by NGSfilter</code></li>
        <li><em>‚ÄúAttribute to merge‚Äù</em>: <code class="language-plaintext highlighter-rouge">sample</code></li>
        <li><em>‚ÄúUse specific option‚Äù</em>: <code class="language-plaintext highlighter-rouge">merge</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many sequences you had and how many you finally obtain?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>From 43 326 to 3 962</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="limit-number-of-informations-with-obiannotate">Limit number of informations with <strong>obiannotate</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-addsedits-sequence-record-annotations"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adds/Edits sequence record annotations</h3>

  <blockquote class="comment">
    <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>obiannotate is the command that allows adding/modifying/removing annotation attributes attached to sequence records.
Once such attributes are added, they can be used by the other OBITools commands for filtering purposes or for statistics computing.</p>

    <p>Here, the goal is to keep only <code class="language-plaintext highlighter-rouge">count</code> and <code class="language-plaintext highlighter-rouge">merged_sample</code> key=value attributes!</p>
  </blockquote>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_annotate/obi_annotate/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_annotate/obi_annotate/1.2.13"><strong>obiannotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_annotate/obi_annotate/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obiuniq output file</code></li>
        <li>In <em>‚ÄúKeep only attribute with key‚Äù</em>:
          <ul>
            <li><em>‚Äúkey‚Äù</em>: <code class="language-plaintext highlighter-rouge">count</code></li>
            <li><em>‚Äúif you want to specify a second key‚Äù</em>: <code class="language-plaintext highlighter-rouge">merged_sample</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="computes-basic-statistics-for-attribute-values-with-obistat">Computes basic statistics for attribute values with <strong>obistat</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-computes-basic-statistics-for-attribute-values"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Computes basic statistics for attribute values</h3>

  <blockquote class="comment">
    <h3 id="comment-comment-6"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>stats computes basic statistics for attribute values of sequence records. The sequence records can be categorized or not using one or several -c options. By default, only the number of sequence records and the total count are computed for each category. Additional statistics can be computed for attribute values in each category, like:</p>
    <ul>
      <li>minimum value (-m option)</li>
      <li>maximum value (-M option)</li>
      <li>mean value (-a option)</li>
      <li>variance (-v option)</li>
      <li>standard deviation (-s option)</li>
    </ul>

    <p>The result is a contingency table with the different categories in rows, and the computed statistics in columns.</p>
  </blockquote>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_stat/obi_stat/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_stat/obi_stat/1.2.13"><strong>obistat</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_stat/obi_stat/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obiannotate output file</code></li>
        <li>In <em>‚ÄúCategory attribute‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Category attribute‚Äù</em>
              <ul>
                <li><em>‚ÄúHow would you specify the category attribute key?‚Äù</em>: <code class="language-plaintext highlighter-rouge">simply by a key of an attribute</code>
                  <ul>
                    <li><em>‚ÄúAttribute used to categorize the sequence records‚Äù</em>: <code class="language-plaintext highlighter-rouge">count</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúUse a specific option‚Äù</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Can you use this result to say how many sequences occuring only once? You would need to use <span class="notranslate">Galaxy</span> tools like <code class="language-plaintext highlighter-rouge">Sort data in ascending or descending order</code> and ` Select first lines from a dataset` to answer the question</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>3131 sequences are occuring once.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="filtering-sequances-by-count-and-length-with-obigrep">Filtering sequances by count and length with <strong>obigrep</strong></h2>

<p>In this step, we are going to use <em>obigrep</em> in order to keep only the sequences having a count greater or equal to 10 and a length shorter than 80 bp.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-sequences-with-obigrep"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: filter sequences with <strong>obigrep</strong></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13"><strong>obigrep</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obiannotate output file</code></li>
        <li><em>‚ÄúChoose the sequence record selection option‚Äù</em>: <code class="language-plaintext highlighter-rouge">predicat</code>
          <ul>
            <li><em>‚ÄúPython boolean <span class="notranslate">expression</span> to be evaluated for each sequence record.‚Äù</em>: <code class="language-plaintext highlighter-rouge">count&gt;=10</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13"><strong>obigrep</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_grep/obi_grep/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obigrep output file</code></li>
        <li><em>‚ÄúChoose the sequence record selection option‚Äù</em>: <code class="language-plaintext highlighter-rouge">lmin</code>
          <ul>
            <li><em>‚Äúlmin‚Äù</em>: <code class="language-plaintext highlighter-rouge">80</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-7"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>Based on the previous observation, we set the cut-off for keeping sequences for further analysis to a count of 10
Based on previous knowledge we also remove sequences with a length shorter than 80 bp (option -l) as we know that the amplified 12S-V5 barcode for vertebrates must have a length around 100bp</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many sequences are kept following the ‚Äúcount‚Äù filter?</li>
    <li>How many sequences are kept following the ‚Äúlength‚Äù filter?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>178</li>
      <li>175</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="clean-the-sequences-for-pcrsequencing-errors-sequence-variants-with-obiclean">Clean the sequences for PCR/sequencing errors (sequence variants) with <strong>obiclean</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-clean-the-sequences-for-pcrsequencing-errors-sequence-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Clean the sequences for PCR/sequencing errors (sequence variants)</h3>

  <p>As a final denoising step, using the obiclean program, we keep the head sequences that are sequences with no variants with a count greater than 5% of their own count</p>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_clean/obi_clean/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_clean/obi_clean/1.2.13"><strong>obiclean</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_clean/obi_clean/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obigrep output file</code></li>
        <li><em>‚Äúattribute containing sample definition‚Äù</em>: <code class="language-plaintext highlighter-rouge">merged_sample</code></li>
        <li><em>‚ÄúMaximum numbers of differences between two variant sequences (default: 1)‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>‚ÄúThreshold ratio between counts (rare/abundant counts) of two sequence records so that the less abundant one is a variant of the more abundant (default: 1, i.e. all less abundant sequences are variants)‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.05</code></li>
        <li><em>‚ÄúDo you want to select only sequences with the head status in a least one sample?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="taxonomic-assignment-of-sequences-with-ncbi-blast-blastn">Taxonomic assignment of sequences with <strong>NCBI BLAST+ blastn</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-search-nucleotide-database-with-nucleotide-query-sequences-from-obitools-treatments"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Search nucleotide database with nucleotide query sequence(s) from OBITools treatments</h3>

  <p>Once denoising has been done, the next step in diet analysis is to assign the barcodes to the corresponding species in order to get the complete list of species associated to each sample. Taxonomic assignment of sequences requires a reference database compiling all possible species to be identified in the sample. Assignment is then done based on sequence comparison between sample sequences and reference sequences. We here propose to use BLAST+ blastn.</p>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.10.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.10.1+galaxy0"><strong>NCBI BLAST+ blastn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.10.1+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúNucleotide query sequence(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">obiclean output file</code></li>
        <li><em>‚ÄúSubject database/sequences‚Äù</em>: <code class="language-plaintext highlighter-rouge">FASTA file from your history</code></li>
        <li><em>‚ÄúNucleotide FASTA subject file to use instead of a database‚Äù</em>: <code class="language-plaintext highlighter-rouge">db_v05_r117</code></li>
        <li><em>‚ÄúSet expectation value cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.0001</code></li>
        <li><em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tabular (extended 25 columns)</code></li>
        <li><em>‚ÄúAdvanced Options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Show Advanced Options</code></li>
        <li>*‚ÄúMaximum hits to consider/show‚Äù: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-8"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>Here we directly use the <code class="language-plaintext highlighter-rouge">db_v05_r117</code> fasta file proposed on the <a href="https://pythonhosted.org/OBITools/wolves.html#step-by-step-analysis">original obitools tutorial</a>. One can mention you can create such a fasta file using same obitools <span class="notranslate">workflow</span> describe before (using obigrep/obiuniq/obigrep/obiannotate) on downloaded EMBL datrabases and taxonomy treated by obitools ecoPCR tool.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="filter-database-and-query-sequences-by-id-to-re-associate-informations-with-filter-sequences-by-id">Filter database and query sequences by ID to re associate informations with <strong>Filter sequences by ID</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-blast-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter Blast results</h3>

  <blockquote class="comment">
    <h3 id="comment-comment-9"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>This tool allows you to re-associate all the reference sequences information, notably the <code class="language-plaintext highlighter-rouge">species_name</code> one so you can see which species are potentially seen on the sample.
We will also use it to re-associate all the query sequences information, notably the <code class="language-plaintext highlighter-rouge">merged_sample</code> and <code class="language-plaintext highlighter-rouge">obiclean_count</code> attributes so we can better evaluate quality of the results.</p>
  </blockquote>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7" title="Tested with toolshed.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7"><strong>Filter sequences by ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSequence file to be filtered‚Äù</em>: <code class="language-plaintext highlighter-rouge">db_v05_r117</code></li>
        <li><em>‚ÄúFilter using the ID list from‚Äù</em>: <code class="language-plaintext highlighter-rouge">tabular file</code></li>
        <li><em>‚ÄúTabular file containing sequence identifiers‚Äù</em>: <code class="language-plaintext highlighter-rouge">megablast on obiclean output</code></li>
        <li><em>‚ÄúColumn(s) containing sequence identifiers‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 2</code></li>
        <li><em>‚ÄúOutput positive matches, negative matches, or both?‚Äù</em>: <code class="language-plaintext highlighter-rouge">just positive match</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7" title="Tested with toolshed.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7"><strong>Filter sequences by ID</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/peterjc/seq_filter_by_id/seq_filter_by_id/0.2.7</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSequence file to be filtered‚Äù</em>: <code class="language-plaintext highlighter-rouge">obiclean output data</code></li>
        <li><em>‚ÄúFilter using the ID list from‚Äù</em>: <code class="language-plaintext highlighter-rouge">tabular file</code></li>
        <li><em>‚ÄúTabular file containing sequence identifiers‚Äù</em>: <code class="language-plaintext highlighter-rouge">megablast on obiclean output</code></li>
        <li><em>‚ÄúColumn(s) containing sequence identifiers‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><em>‚ÄúOutput positive matches, negative matches, or both?‚Äù</em>: <code class="language-plaintext highlighter-rouge">just positive match</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="from-fasta-to-tabular-with-obitab">From FASTA to tabular with <strong>Obitab</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-convert-fasta-filtered-files-in-tabular-ones"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Convert fasta filtered files in tabular ones</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_tab/obi_tab/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_tab/obi_tab/1.2.13"><strong>obitab</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_tab/obi_tab/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">db_v05_r117 with matched ID</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-10"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>This tool allows you to convert a fasta file into a tabular one so it is easier to read sequences definitions.</p>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/obi_tab/obi_tab/1.2.13" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/obi_tab/obi_tab/1.2.13"><strong>obitab</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/obi_tab/obi_tab/1.2.13</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput sequences file‚Äù</em>: <code class="language-plaintext highlighter-rouge">obiclean on data 61 with matched ID</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="create-a-final-synthesis-as-a-tabular-file">create a final synthesis as a tabular file</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-join-blast-and-obitab-files-then-cut-relevant-column-and-apply-filters"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Join blast and obitab files then cut relevant column and apply filters</h3>

  <ol>
    <li><span class="tool" data-tool="join1" title="Tested with join1"><strong>Join two datasets side by side on a specified field</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: join1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúJoin‚Äù</em>: <code class="language-plaintext highlighter-rouge">obitab on obiclean output file</code></li>
        <li><em>‚Äúusing column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><em>‚Äúwith‚Äù</em>: <code class="language-plaintext highlighter-rouge">megablast output file</code></li>
        <li><em>‚Äúusing column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><em>‚ÄúFill empty columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFill Columns by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single fill value</code></li>
        <li><em>‚ÄúFill value‚Äù</em>: <code class="language-plaintext highlighter-rouge">NA</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="join1" title="Tested with join1"><strong>Join two datasets side by side on a specified field</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: join1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúJoin‚Äù</em>: last <code class="language-plaintext highlighter-rouge">Join two Datasets output file</code></li>
        <li><em>‚Äúusing column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 26</code></li>
        <li><em>‚Äúwith‚Äù</em>: <code class="language-plaintext highlighter-rouge">obitab on db_v05_r117 with matched ID output file</code></li>
        <li><em>‚Äúusing column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><em>‚ÄúFill empty columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFill Columns by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single fill value</code></li>
        <li><em>‚ÄúFill value‚Äù</em>: <code class="language-plaintext highlighter-rouge">NA</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-11"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>To have something easier to read and understand, we create a tabular file containing only columns with important informations (c1: query sequences names / c3-7: query counts / c50: reference sequences names / c54: family / c59: genus / c51: reference annotations).</p>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="Cut1" title="Tested with Cut1"><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Cut1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c1,c3,c4,c5,c6,c7,c50,c54,c59,c51</code></li>
        <li><em>‚ÄúFrom‚Äù</em>: last <code class="language-plaintext highlighter-rouge">Join two Datasets output file</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Filter1" title="Tested with Filter1"><strong>Filter data on any column using simple <span class="notranslate">expression</span>s</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Filter1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúFilter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cut output file</code></li>
        <li><em>‚ÄúWith following condition‚Äù</em>: <code class="language-plaintext highlighter-rouge">c3&gt;1000 or c4&gt;1000 or c5&gt;1000 or c6&gt;1000</code></li>
        <li><em>‚ÄúNumber of header lines to skip‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-12"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>To keep only data with significative counts.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many species are identified? You can use <code class="language-plaintext highlighter-rouge">Cut columns from a table</code> and <code class="language-plaintext highlighter-rouge">unique occurences of each record</code> to isolate the <code class="language-plaintext highlighter-rouge">species name</code> column of obitab results.</li>
    <li>Can you deduce the diet of each sample? You can use tools like <code class="language-plaintext highlighter-rouge">obitab</code> and <code class="language-plaintext highlighter-rouge">Join two Datasets side by side on a specified field</code> to join megablast results to obigrep one and <code class="language-plaintext highlighter-rouge">db_v05_r117 with matched ID</code></li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>1O</li>
      <li>If we remove human sequences ;), some squirrel (sample:26a_F040644), deer (sample:15a_F730814 + sample:29a_F260619), stag (sample:29a_F260619 + sample:13a_F730603), marmot (sample:26a_F040644), doe (sample:29a_F260619), Grimm‚Äôs duiker (sample:29a_F260619).</li>
    </ol>

  </blockquote>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>You just did a ecological analysis, finding diet from wolves faeces ! So now you know how to preprocess metabarcoding data on <span class="notranslate">Galaxy</span>, producing quantitative informations with quality checks and filtering results to interpret it and to have a synthesis table you can share broadly!</p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>From raw reads you can process, clean and filter data to obtain a list of species from environmental DNA (eDNA) samples.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/ecology/faqs/">FAQ page for the Ecology topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/ecology#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Schloss2009">Schloss, P. D., S. L. Westcott, T. Ryabin, J. R. Hall, M. Hartmann <i>et al.</i>, 2009 <b>Introducing mothur: open-source, platform-independent, community-supported software for describing and comparing microbial communities</b>. Appl. Environ. Microbiol. 75: 7537‚Äì7541.</li>
<li id="Caporaso2010">Caporaso, J. G., J. Kuczynski, J. Stombaugh, K. Bittinger, F. D. Bushman <i>et al.</i>, 2010 <b>QIIME allows analysis of high-throughput community sequencing data</b>. Nat. Methods 7: 335‚Äì336.</li>
<li id="Ficetola2010">Ficetola, G. F., E. Coissac, S. Zundel, T. Riaz, W. Shehzad <i>et al.</i>, 2010 <b>An In silico approach for the evaluation of DNA barcodes</b>. BMC Genomics 39: e145‚Äìe145. <a href="https://doi.org/10.1186/1471-2164-11-434">10.1186/1471-2164-11-434</a></li>
<li id="Riaz2011">Riaz, T., W. Shehzad, A. Viari, F. Pompanon, P. Taberlet <i>et al.</i>, 2011 <b>ecoPrimers: inference of new DNA barcode markers from whole genome sequence analysis</b>. Nucleic Acids Research 39: e145‚Äìe145. <a href="https://doi.org/10.1093/nar/gkr732">10.1093/nar/gkr732</a></li>
<li id="Shehzad2012">SHEHZAD, W. A. S. I. M., T. I. A. Y. Y. B. A. RIAZ, M. U. H. A. M. M. A. D. A. NAWAZ, C. H. R. I. S. T. I. A. N. MIQUEL, C. A. R. O. L. E. POILLOT <i>et al.</i>, 2012 <b>Carnivore diet analysis based on next-generation sequencing: application to the leopard cat (Prionailurus bengalensis) in Pakistan</b>. Molecular Ecology 21: 1951‚Äì1965. <a href="https://doi.org/10.1111/j.1365-294x.2011.05424.x">10.1111/j.1365-294x.2011.05424.x</a></li>
<li id="Boyer2015">Boyer, F., C. Mercier, A. Bonin, Y. L. Bras, P. Taberlet <i>et al.</i>, 2015 <b>obitools: aunix-inspired software package for DNA metabarcoding</b>. Molecular Ecology Resources 16: 176‚Äì182. <a href="https://doi.org/10.1111/1755-0998.12428">10.1111/1755-0998.12428</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Metabarcoding/eDNA through Obitools (Ecology)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Coline Royaux, Olivier Norvez, Eric Coissac, Fr√©d√©ric Boyer, Yvan Le Bras, 2022 <b>Metabarcoding/eDNA through Obitools (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html">https://training.galaxyproject.org/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{ecology-Obitools-metabarcoding,
author = "Coline Royaux and Olivier Norvez and Eric Coissac and Fr√©d√©ric Boyer and Yvan Le Bras",
title = "Metabarcoding/eDNA through Obitools (Galaxy Training Materials)",
year = "2022",
month = "02",
day = "28"
url = "\url{https://training.galaxyproject.org/training-material/topics/ecology/tutorials/Obitools-metabarcoding/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
