<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>FAQs / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@ae15e8218fd9438d728c72dee2577c3c01592390",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/faqs/">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2022-11-18 15:34:02 +0000"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / FAQs">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/faqs/index.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/faqs/">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<section class="faqs">

   <nav aria-label="breadcrumb">
    <ol class="breadcrumb">

     <li class="breadcrumb-item "><a href="/training-material/faqs">All FAQs</a></li>
     
     <li class="breadcrumb-item "><a href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
     
     
     
     
     <li class="breadcrumb-item active">Topic FAQs</li>
     

     
     </ol>
    </nav>

    <h1 data-toc-skip>Frequently Asked Questions</h1>
    
    <div class="container">
        <div class="row">

            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h3 id="Analysis"> <a href="#Analysis"> Analysis </a></h3>

<hr/>



<h4 class="faq-area"> Are UMIs not actually unique? <a href="/training-material/topics/single-cell/faqs/umi.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-are-umis-not-actually-unique"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-are-umis-not-actually-unique" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Are UMIs not actually unique?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Not strictly, but unique enough. The distribution of UMIs should ideally be uniform so that the chance of any two same UMIs capturing the same transcript (via different amplicons) is small. As barcodes have increased in size, the number of UMIs has also increased allowing for UMIs to reach more or less the same numbers of transcripts.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Can RNA-seq techniques be applied to scRNA-seq? <a href="/training-material/topics/single-cell/faqs/techniques.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-rna-seq-techniques-be-applied-to-scrna-seq"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-rna-seq-techniques-be-applied-to-scrna-seq" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can RNA-seq techniques be applied to scRNA-seq?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The short answer is ‘no, but yes’. At the beginning this was impossible due to the over-prevalence of dropout events (“zeroes”) in the data complicating the normalisation techniques, but this is not so much of a problem any more with newer methods.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Notebook-based tutorials can give different outputs <a href="/training-material/topics/single-cell/faqs/notebook_warning.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="warning">   <div class="box-title warning-title" id="warning-notebook-based-tutorials-can-give-different-outputs"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i> Warning: Notebook-based tutorials can give different outputs</div>   <p>The nature of coding pulls the most recent tools to perform tasks. This can - and often does - change the outputs of an analysis. Be prepared, as you are unlikely to get outputs identical to a tutorial if you are running it in a programming environment like a Jupyter Notebook or R-Studio. That’s ok! The outputs should still be pretty close.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we do dimension reduction and then clustering? Why not just cluster on the actual data? <a href="/training-material/topics/single-cell/faqs/dimension_reduction.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-do-we-do-dimension-reduction-and-then-clustering-why-not-just-cluster-on-the-actual-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why do we do dimension reduction and then clustering? Why not just cluster on the actual data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The actual data has tens of thousands of genes, and so tens of thousands of variables to consider. Even after selecting for the most variable genes and the most high quality genes, we can still be left with &gt; 1000 genes. Performing clustering on a dataset with 1000s of variables is possible, but computationally expensive. It is therefore better to perform dimension reduction to reduce the number of variables to a latent representation of these variables. These latent variables are ideally more than 10 but less than 50 to capture the variability in the data to perform clustering upon.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why do we only consider highly variable genes? <a href="/training-material/topics/single-cell/faqs/variable_genes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-do-we-only-consider-highly-variable-genes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-do-we-only-consider-highly-variable-genes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why do we only consider highly variable genes?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The non-variable genes are likely housekeeping genes, which are expressed everywhere and are not so useful for distinguishing one cell type from another. However background genes are important to the analysis and are used to generate a background baseline model for measuring the variability of the other genes.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why is my tool erroring as 'Above error raised while reading key '/layers' of type <class 'h5py._hl.group.Group'> from /.' <a href="/training-material/topics/single-cell/faqs/sc_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-is-my-tool-erroring-as-above-error-raised-while-reading-key-layers-of-type-class-h5py-hl-group-group-from"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-is-my-tool-erroring-as-above-error-raised-while-reading-key-layers-of-type-class-h5py-hl-group-group-from" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why is my tool erroring as 'Above error raised while reading key '/layers' of type &lt;class 'h5py._hl.group.Group'&gt; from /.'</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Are you getting the following error, or similar?</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Traceback (most recent call last):<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/utils.py", line 177, in func_wrapper<br>return func(elem, *args, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/h5ad.py", line 527, in read_group<br>EncodingVersions[encoding_type].check(<br>File "/usr/local/lib/python3.9/enum.py", line 432, in __getitem__<br>return cls._member_map_[name]<br>KeyError: 'dict'<br><br>During handling of the above exception, another exception occurred:<br><br>Traceback (most recent call last):<br>File "/usr/local/bin/scanpy-cli", line 10, in &lt;module&gt;<br>sys.exit(cli())<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 829, in __call__<br>return self.main(*args, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 782, in main<br>rv = self.invoke(ctx)<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1259, in invoke<br>return _process_result(sub_ctx.command.invoke(sub_ctx))<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1259, in invoke<br>return _process_result(sub_ctx.command.invoke(sub_ctx))<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1066, in invoke<br>return ctx.invoke(self.callback, **ctx.params)<br>File "/usr/local/lib/python3.9/site-packages/click/core.py", line 610, in invoke<br>return callback(*args, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/scanpy_scripts/cmd_utils.py", line 45, in cmd<br>adata = _read_obj(input_obj, input_format=input_format)<br>File "/usr/local/lib/python3.9/site-packages/scanpy_scripts/cmd_utils.py", line 87, in _read_obj<br>adata = sc.read(input_obj, **kwargs)<br>File "/usr/local/lib/python3.9/site-packages/scanpy/readwrite.py", line 112, in read<br>return _read(<br>File "/usr/local/lib/python3.9/site-packages/scanpy/readwrite.py", line 713, in _read<br>return read_h5ad(filename, backed=backed)<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/h5ad.py", line 421, in read_h5ad<br>d[k] = read_attribute(f[k])<br>File "/usr/local/lib/python3.9/functools.py", line 877, in wrapper<br>return dispatch(args[0].__class__)(*args, **kw)<br>File "/usr/local/lib/python3.9/site-packages/anndata/_io/utils.py", line 183, in func_wrapper<br>raise AnnDataReadError(<br>anndata._io.utils.AnnDataReadError: Above error raised while reading key '/layers' of type &lt;class 'h5py._hl.group.Group'&gt; from /.<br></code></pre></div></div>   <p>This is likely a Tool Version error. If you use a newer version of a tool with an AnnData object, and then try and use an older version of the tool or other tool in the same toolsuite (Scanpy) later, this will often fail with the above error message. The Scanpy toolsuite is not ‘backwards compatable’ - few toolsuites are. If this happened while performing a tutorial, we recommend Tutorial Mode as this embeds the correct tool version in each tool button.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>   <p><!--END_SNIPPET--></p>   <p>To fix this in your current history, try re-running the tool with the newer tool version. Or, re-run the prior dataset with an older version.</p>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-tool-version"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-tool-version" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the tool version</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool.</p>   <p><strong>Switching to a different version</strong> of a tool:</p>   <ul>   <li>Open the tool</li>   <li>Click on the <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> versions logo at the <strong>top right</strong></li>   <li>Select the desired version from the dropdown list</li> </ul> </blockquote>   <p><!--END_SNIPPET--></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Community"> <a href="#Community"> Community </a></h3>

<hr/>



<h4 class="faq-area"> How can I talk with other users? <a href="/training-material/topics/single-cell/faqs/user_community_join.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-talk-with-other-users"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-talk-with-other-users" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I talk with other users?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> To discuss with like-minded scientists, join our <a href="http://gxy.io/gtn-slack">Galaxy Training Network chatspace in Slack</a> and discuss with fellow users of Galaxy single cell analysis tools on <a href="https://gtnsmrgsbord.slack.com/archives/C06PBRR40D7">#single-cell-users</a></p>   <p>We also post new tutorials / workflows there from time to time, as well as any other news.</p>   <p><i class="far fa-hand-point-right" aria-hidden="true"></i><span class="visually-hidden">point-right</span> If you’d like to contribute ideas, requests or feedback as part of the wider community building single-cell and spatial resources within Galaxy, you can also join our <a href="https://galaxyproject.org/community/sig/singlecell/">Single cell &amp; sPatial Omics Community of Practice</a>.</p>   <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> You can request tools here on our <a href="https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing">Single Cell and Spatial Omics Community Tool Request Spreadsheet</a></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Interpretation"> <a href="#Interpretation"> Interpretation </a></h3>

<hr/>



<h4 class="faq-area"> What exactly is a ‘Gene profile’? <a href="/training-material/topics/single-cell/faqs/gene_profile.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-exactly-is-a-gene-profile"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-exactly-is-a-gene-profile" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What exactly is a ‘Gene profile’?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Think of it like a fingerprint that some cells exhibit and others don’t. It’s a small collection of genes which are up or down regulated in relation to one another. Their differences are not absolute, but relative. So if CellA has 100 counts of Gene1 and 50 counts of Gene2, this creates a relation of 2:1 between Gene1 and Gene2. If CellB has a 20 counts of Gene1 and 10 counts of Gene2, then they share the same relation. If CellA and CellB share other relations with other genes than this might be enough to say that they share a Gene profile, and will therefore likely cluster together as they describe the same cell type.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Resources"> <a href="#Resources"> Resources </a></h3>

<hr/>



<h4 class="faq-area"> Use our Single Cell Omics Lab <a href="/training-material/topics/single-cell/faqs/single_cell_omics.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-use-our-single-cell-omics-lab"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-use-our-single-cell-omics-lab" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Use our Single Cell Omics Lab</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Did you know we have a unique Single Cell Omics Lab with all our single cell tools highlighted to make it easier to use on Galaxy? We recommend this site for all your single cell analysis needs, particularly for newer users.</p>   <p>The Single Cell Omics Lab currently uses the main European Galaxy infrastructure and power, it’s just organised better for users of particular analyses…like single cell!</p>   <p>Try it out! All your histories/workflows/logins from the general <a href="https://usegalaxy.eu">European Galaxy server</a> will be there!</p>   <ul>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.eu">Single Cell Omics Lab</a></li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="Single-Cell RNA"> <a href="#Single-Cell RNA"> Single-cell rna </a></h3>

<hr/>



<h4 class="faq-area"> Why is amplification more of an issue in scRNA-seq than RNA-seq? <a href="/training-material/topics/single-cell/faqs/amplification.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-is-amplification-more-of-an-issue-in-scrna-seq-than-rna-seq"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-is-amplification-more-of-an-issue-in-scrna-seq-than-rna-seq" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why is amplification more of an issue in scRNA-seq than RNA-seq?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Due to the extremely small amount of starting material, the initial amplification is likely to be uneven due to the first cycle of amplified products being overrepresented in the second cycle of amplification leading to further bias. In Bulk RNA-seq, the larger selection of RNA molecules to amplify, evens out the odds that any one transcript will be amplified more than others.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="tools"> <a href="#tools"> Tools </a></h3>

<hr/>



<h4 class="faq-area"> Using tutorial mode and the Case Study suite <a href="/training-material/topics/single-cell/faqs/gtn-in-galaxy_mode-cs.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tutorial mode saves you screen space, finds the tools you need, and ensures you use the correct versions for the tutorials to run.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode-and-the-case-study-suite"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode-and-the-case-study-suite" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode and the Case Study suite</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial via <strong>Single-cell</strong> (Underneath the <em>Methodologies</em> section), then <strong>Case Study</strong>, then Select your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote> <!--END_SNIPPET-->











<br><br><hr>
<h5>Still have questions?</h5>

<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://gitter.im/Galaxy-Training-Network/Lobby" class="btn btn-square-medium btn-yellow"><div>Gitter Chat Support</div></a>


<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://help.galaxyproject.org" class="btn btn-square-medium btn-purple"><div>Galaxy Help Forum</div></a>



            </div>
        </div>
    </div>
</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/faqs/index.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/faqs/index.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/ae15e8218fd9438d728c72dee2577c3c01592390">ae15e82</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>