<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Single-cell ATAC-seq standard processing with SnapATAC2 / Single-cell ATAC-seq standard processing with SnapATAC2 / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@7f6ee502828a7660a00ce43e77e864d7cbada116",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Single-cell ATAC-seq standard processing with SnapATAC2">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-08-09 08:49:25 +0000">
<meta name="DC.creator" content="Timon Schlegel"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Single-cell ATAC-seq standard processing with SnapATAC2 / Hands-on: Single-cell ATAC-seq standard processing with SnapATAC2">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Single-cell ATAC-seq standard processing with SnapATAC2",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-08-09 08:49:25 +0000",
  "datePublished": "2024-07-11 14:34:09 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00445",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!\n\nWhat else do you want to see? You can submit tool and/or tutorial request on our <i class=\"fa fa-wrench\" aria-hidden=\"true\"></i> [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).\n {: .alert.alert-success}\n",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "abstract": "Single-cell Assay for Transposase-Accessible Chromatin using sequencing (scATAC-seq) analysis is a method to decipher the chromatin states of the analyzed cells. In general, genes are only expressed in accessible (i.e. \"open\") chromatin and not in closed chromatin.",
  "learningResourceType": "e-learning",
  "name": "Single-cell ATAC-seq standard processing with SnapATAC2",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html",
  "version": 2,
  "timeRequired": "PT4H",
  "teaches": "- Learn how single-cell ATAC-seq data is processed\n- Create a count-matrix from a 10X fragment file\n- Perform filtering, dimension reduction and clustering on AnnData matrices\n- Generate and filter a cell-by-gene matrix\n- Identify marker genes for the clusters and annotate the cell types",
  "keywords": [
    "Single Cell",
    "10x",
    "epigenetics",
    "single-cell"
  ],
  "description": "## Abstract\n\nSingle-cell Assay for Transposase-Accessible Chromatin using sequencing (scATAC-seq) analysis is a method to decipher the chromatin states of the analyzed cells. In general, genes are only expressed in accessible (i.e. \"open\") chromatin and not in closed chromatin.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - What does ATAC-seq data tell us about the cell?\n - Which steps are necessary to cluster the cells of single-cell ATAC-seq data?\n - Why is dimension reduction important for the analysis of single-cell data?\n\n\n## Learning Objectives\n\n- Learn how single-cell ATAC-seq data is processed\n- Create a count-matrix from a 10X fragment file\n- Perform filtering, dimension reduction and clustering on AnnData matrices\n- Generate and filter a cell-by-gene matrix\n- Identify marker genes for the clusters and annotate the cell types\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html",
      "name": "Pre-processing of 10X Single-Cell ATAC-seq Datasets",
      "description": "Hands-on for 'Pre-processing of 10X Single-Cell ATAC-seq Datasets' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/timonschlegel/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/timonschlegel/",
      "name": "Timon Schlegel",
      "image": "https://avatars.githubusercontent.com/timonschlegel",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0001-3228-105X",
      "orcid": "https://orcid.org/0009-0001-3228-105X"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "name": "Pavankumar Videm",
      "image": "https://avatars.githubusercontent.com/pavanvidem",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5192-126X",
      "orcid": "https://orcid.org/0000-0002-5192-126X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "name": "Björn Grüning",
      "image": "https://avatars.githubusercontent.com/bgruening",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eurosciencegateway/",
          "name": "EuroScienceGateway",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://galaxyproject.org/projects/esg/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eosc-life/",
          "name": "EOSC-Life",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.eosc-life.eu"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/deNBI/",
          "name": "de.NBI",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.denbi.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/sfb992/",
          "name": "Collaborative Research Centre 992",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.sfb992.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-converge/",
          "name": "ELIXIR-CONVERGE",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org/about-us/how-funded/eu-projects/converge"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-3079-6586",
      "orcid": "https://orcid.org/0000-0002-3079-6586"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!\n\nWhat else do you want to see? You can submit tool and/or tutorial request on our <i class=\"fa fa-wrench\" aria-hidden=\"true\"></i> [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).\n {: .alert.alert-success}\n",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/records/12707159",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Single-cell ATAC-seq standard processing with SnapATAC2</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/timonschlegel/" class="contributor-badge contributor-timonschlegel"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/timonschlegel?s=36" alt="Timon Schlegel avatar" width="36" class="avatar" />
    Timon Schlegel</a>
		</td>
	</tr>
	

	

	

	

	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" />
    Pavankumar Videm</a><a href="/training-material/hall-of-fame/bgruening/" class="contributor-badge contributor-bgruening"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bgruening?s=36" alt="Björn Grüning avatar" width="36" class="avatar" />
    Björn Grüning</a></td>
	</tr>
	

	
	<tr>
		<td><abbr title="These people tested the tutorial to ensure it works correctly for students, or reported issues with the tutorial.">Tester(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" />
    Pavankumar Videm</a></td>
	</tr>
	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What does ATAC-seq data tell us about the cell?</p>
</li>
        
        <li><p>Which steps are necessary to cluster the cells of single-cell ATAC-seq data?</p>
</li>
        
        <li><p>Why is dimension reduction important for the analysis of single-cell data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn how single-cell ATAC-seq data is processed</p>
</li>
        
        <li><p>Create a count-matrix from a 10X fragment file</p>
</li>
        
        <li><p>Perform filtering, dimension reduction and clustering on AnnData matrices</p>
</li>
        
        <li><p>Generate and filter a cell-by-gene matrix</p>
</li>
        
        <li><p>Identify marker genes for the clusters and annotate the cell types</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Pre-processing of 10X Single-Cell ATAC-seq Datasets</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 4 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/records/12707159">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/" title="Single-cell ATAC-seq standard processing with SnapATAC2 workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Jul 11, 2024 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Aug 9, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00445">https://gxy.io/GTN:T00445</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 2 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Single-cell Assay for Transposase-Accessible Chromatin using sequencing (scATAC-seq) analysis is a method to decipher the chromatin states of the analyzed cells. In general, genes are only expressed in accessible (i.e. “open”) chromatin and not in closed chromatin.
By analyzing which genomic sites have an <em>open</em> chromatin state, cell-type specific patterns of gene accessibility can be determined.
<abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> is particularly useful for analyzing tissue containing different cell populations, such as peripheral blood mononuclear cells (PBMCs).</p>

<p>In this tutorial we will analyze <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> data using the tool suites <a href="https://kzhang.org/SnapATAC2/version/2.5/index.html">SnapATAC2</a> (<span class="citation"><a href="#Zhang2024">Zhang <i>et al.</i> 2024</a></span>) and <a href="https://scanpy.readthedocs.io/en/stable/index.html">Scanpy</a> (<span class="citation"><a href="#Wolf2018">Wolf <i>et al.</i> 2018</a></span>).
With both of these tool suites we will perform preprocessing, clustering and identification of <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> datasets from <a href="https://www.10xgenomics.com/products/single-cell-atac">10x Genomics</a>.
The analysis will be performed using a dataset of <abbr title="peripheral blood mononuclear cells">PBMCs</abbr> containing ~4,620 single nuclei.</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-use-our-single-cell-omics-lab"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-use-our-single-cell-omics-lab" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Use our Single Cell Omics Lab</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Did you know we have a unique Single Cell Omics Lab with all our single cell tools highlighted to make it easier to use on Galaxy? We recommend this site for all your single cell analysis needs, particularly for newer users.</p>   <p>The Single Cell Omics Lab currently uses the main European Galaxy infrastructure and power, it’s just organised better for users of particular analyses…like single cell!</p>   <p>Try it out!</p>   <ul>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.eu">Europe: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org">USA: Single Cell Omics Lab</a></li>   <li><i class="fa fa-tv" aria-hidden="true"></i><span class="visually-hidden">subdomain</span> <a href="https://singlecell.usegalaxy.org.au">Australia: Single Cell Omics Lab</a></li> </ul> </blockquote>
<p><!--END_SNIPPET--></p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>
<p><!--END_SNIPPET--></p>

<!-- This is a comment. -->

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>This tutorial is significantly based on <a href="https://kzhang.org/SnapATAC2/version/2.5/tutorials/pbmc.html">“Standard pipeline” tutorial from SnapATAC2</a>. It can be seen as the <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> counterpart to the scRNA-seq tutorial <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Clustering 3K PBMCs with Scanpy</a>.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#scatac-seq-with-10x-genomics" id="markdown-toc-scatac-seq-with-10x-genomics"><abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> with 10X Genomics</a></li>
  <li><a href="#data" id="markdown-toc-data">Data</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get Data</a></li>
      <li><a href="#creating-a-fragment-file" id="markdown-toc-creating-a-fragment-file">Creating a fragment file</a></li>
    </ol>
  </li>
  <li><a href="#preprocessing" id="markdown-toc-preprocessing">Preprocessing</a>    <ol>
      <li><a href="#anndata" id="markdown-toc-anndata">AnnData</a></li>
      <li><a href="#import-files-to-snapatac2" id="markdown-toc-import-files-to-snapatac2">Import files to SnapATAC2</a></li>
      <li><a href="#calculate-and-visualize-qc-metrics" id="markdown-toc-calculate-and-visualize-qc-metrics">Calculate and visualize <abbr title="quality control">QC</abbr> metrics</a></li>
      <li><a href="#filtering" id="markdown-toc-filtering">Filtering</a></li>
      <li><a href="#feature-selection" id="markdown-toc-feature-selection">Feature selection</a></li>
      <li><a href="#doublet-removal" id="markdown-toc-doublet-removal">Doublet removal</a></li>
    </ol>
  </li>
  <li><a href="#dimension-reduction" id="markdown-toc-dimension-reduction">Dimension reduction</a>    <ol>
      <li><a href="#spectral-embedding" id="markdown-toc-spectral-embedding">Spectral embedding</a></li>
      <li><a href="#umap-embedding" id="markdown-toc-umap-embedding">UMAP embedding</a></li>
    </ol>
  </li>
  <li><a href="#clustering" id="markdown-toc-clustering">Clustering</a>    <ol>
      <li><a href="#community-clustering" id="markdown-toc-community-clustering">Community clustering</a></li>
      <li><a href="#plotting-of-clusters" id="markdown-toc-plotting-of-clusters">Plotting of clusters</a></li>
    </ol>
  </li>
  <li><a href="#cell-cluster-annotation" id="markdown-toc-cell-cluster-annotation">Cell cluster annotation</a>    <ol>
      <li><a href="#gene-matrix" id="markdown-toc-gene-matrix">Gene matrix</a></li>
      <li><a href="#imputation-with-scanpy-and-magic" id="markdown-toc-imputation-with-scanpy-and-magic">Imputation with Scanpy and MAGIC</a></li>
      <li><a href="#copy-over-embeddings" id="markdown-toc-copy-over-embeddings">Copy-over embeddings</a></li>
      <li><a href="#visualize-gene-activity-of-marker-genes" id="markdown-toc-visualize-gene-activity-of-marker-genes">Visualize gene activity of marker genes</a></li>
      <li><a href="#manual-cluster-annotation" id="markdown-toc-manual-cluster-annotation">Manual cluster annotation</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="scatac-seq-with-10x-genomics"><abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> with 10X Genomics</h1>
<p>ATAC-seq utilizes a hyperactive Tn5 transposase (<span class="citation"><a href="#Kia2017">Kia <i>et al.</i> 2017</a></span>) to ligate adaptors to genome fragments, created by the transposase. Performing ATAC-seq on individual cells used to be an expensive and time-consuming labour.
The 10X Chromium NextGEM system made <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> a cost-effective method for gaining high-resolution data with a simple protocol.
After the transposition of nuclei in bulk, individual nuclei are put into Gel beads in Emulsion (GEM), containing unique 10x cell barcodes and sequencing adaptors for Illumina sequencing.</p>
<figure id="figure-1" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/tenx_libprep_scatac.png" alt="Library Preparation. " width="890" height="343" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/tenx_libprep_scatac.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> An overview of the 10X single-nuclei ATAC-seq library preparation</figcaption></figure>

<h1 id="data">Data</h1>

<p>The 5k <abbr title="peripheral blood mononuclear cells">PBMCs</abbr> dataset for this tutorial is available for free from <a href="https://www.10xgenomics.com/datasets/5-k-peripheral-blood-mononuclear-cells-pbm-cs-from-a-healthy-donor-next-gem-v-1-1-1-1-standard-2-0-0">10X Genomics</a>. The blood samples were collected from a healthy donor and were prepared following the Chromium Next GEM scATAC-seq protocol. After sequencing on Illumina NovaSeq, the reads were processed by the <strong>Cell Ranger ATAC 2.0.0</strong> pipeline from 10X to generate a <a href="https://support.10xgenomics.com/single-cell-atac/software/pipelines/latest/output/fragments"><em>Fragments File</em></a>.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-fragments-file"><button class="gtn-boxify-button details" type="button" aria-controls="details-fragments-file" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Fragments File </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The Fragments File is a tabular file in a <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED</a>-like format, containing information about the position of the fragments on the chromosome and their corresponding 10x cell barcodes.</p>

</blockquote>
<p>SnapATAC2 requires 3 input files for the standard pathway of processing:</p>
<ul>
  <li><code class="language-plaintext highlighter-rouge">fragments_file.tsv</code>: A tabular file containing the chromosome positions of the fragments and their corresponding 10x cell barcodes.</li>
  <li><code class="language-plaintext highlighter-rouge">chrom_sizes.txt</code>: A tabular file of the number of bases of each chromosome of the reference genome</li>
  <li><code class="language-plaintext highlighter-rouge">gene_annotation.gtf.gz</code>: A tabular file listing genomic features of the reference genome (GENCODE GRCh38)</li>
</ul>

<blockquote class="details">
  <div class="box-title details-title" id="details-chromosome-sizes"><button class="gtn-boxify-button details" type="button" aria-controls="details-chromosome-sizes" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Chromosome sizes </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>A chromosome sizes file can be generated using the tool <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fasta_compute_length/fasta_compute_length/1.0.3" title="Compute sequence length tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute sequence length</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span>.</li>
    <li>The reference genome can either be selected from cached genomes or uploaded to the galaxy history.</li>
  </ul>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <ul>
    <li>This tutorial starts with a <code class="language-plaintext highlighter-rouge">fragment</code> file.</li>
    <li>SnapATAC2 also accepts mapped reads in a <code class="language-plaintext highlighter-rouge">BAM</code> file.</li>
    <li>To learn how to get a <code class="language-plaintext highlighter-rouge">fragment</code> file or <code class="language-plaintext highlighter-rouge">BAM</code> file from raw <code class="language-plaintext highlighter-rouge">.FASTQ</code>-reads, please check out the tutorial <a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">“Pre-processing of 10X Single-Cell ATAC-seq Datasets”</a></li>
    <li>If you would like to start the analysis with a <code class="language-plaintext highlighter-rouge">BAM</code> file, you can expand the details section <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html#creating-a-fragment-file">“Details: Creating a fragment file”</a>.</li>
  </ul>
</blockquote>

<h2 id="get-data">Get Data</h2>
<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the <code class="language-plaintext highlighter-rouge">fragments_file.tsv</code>, <code class="language-plaintext highlighter-rouge">chrom_sizes.txt</code> and <code class="language-plaintext highlighter-rouge">gene_annotation.gtf.gz</code> from <a href="https://zenodo.org/records/12707159">Zenodo</a> or from the shared data library</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/12707159/files/atac_pbmc_5k_nextgem_fragments.tsv.gz
https://zenodo.org/records/12707159/files/chrom_sizes.txt
https://zenodo.org/records/12707159/files/gencode.v46.annotation.gtf.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Rename the datasets
      <ul>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>Rename</strong> the file <code class="language-plaintext highlighter-rouge">atac_pbmc_5k_nextgem_fragments.tsv</code> to <code class="language-plaintext highlighter-rouge">fragments_file.tsv</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>Rename</strong> the file <code class="language-plaintext highlighter-rouge">gencode.v46.annotation.gtf.gz</code> to <code class="language-plaintext highlighter-rouge">gene_annotation.gtf.gz</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Inspect <code class="language-plaintext highlighter-rouge">chrom_sizes</code> and <code class="language-plaintext highlighter-rouge">fragments_file</code></li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many chromosomes are in <code class="language-plaintext highlighter-rouge">chrom_sizes</code>?</li>
    <li>In which column are the cell barcodes stored in the <code class="language-plaintext highlighter-rouge">fragments_file</code>?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>There are 25 chromosomes. The 22 autosomes (Chr. 1-22), both sex chromosomes (Chr. X and Y) and the small circular mitochondrial chromosome (Chr. M).</li>
      <li>The cell barcodes are unique 16 bp oligos, located in column 4.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="creating-a-fragment-file">Creating a fragment file</h2>
<blockquote class="details">
  <div class="box-title details-title" id="details-creating-a-fragment-file"><button class="gtn-boxify-button details" type="button" aria-controls="details-creating-a-fragment-file" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Creating a fragment file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-fragment-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: fragment file</div>
    <ol>
      <li>Import the file  <code class="language-plaintext highlighter-rouge">BAM_500-PBMC</code> from <a href="https://zenodo.org/records/12707159">Zenodo</a> or from the shared data library
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/12707159/files/BAM_500-PBMC.bam
</code></pre></div>        </div>
        <ul>
          <li>This dataset contains mapped reads in the <code class="language-plaintext highlighter-rouge">BAM</code> format.</li>
          <li>It was generated by following the tutorial <a href="/training-material/topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.html">“Pre-processing of 10X Single-Cell ATAC-seq Datasets”</a> until the output of <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.18" title="Map with BWA-MEM tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with BWA-MEM</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.7.18)</span></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
        <ul>
          <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Convert a BAM file to a fragment file, using 'pp.make_fragment_file'</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File name of the BAM file”</em>: <code class="language-plaintext highlighter-rouge">BAM_500-PBMC</code> (Input dataset)</li>
              <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <em>“Indicate whether the BAM file contain paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              <li><em>“How to extract barcodes from BAM records?”</em>: <code class="language-plaintext highlighter-rouge">From read names using regular expressions</code>
                <ul>
                  <li><em>“Extract barcodes from read names of BAM records using regular expressions”</em>: <code class="language-plaintext highlighter-rouge">(................):</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>

        <blockquote class="comment">
          <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
          <ul>
            <li>Not every regular expression type is supported.</li>
            <li>This expression selects 16 characters if they are followed by a colon. Only the cell barcodes of the <code class="language-plaintext highlighter-rouge">BAM</code> file will match.</li>
          </ul>
        </blockquote>
      </li>
      <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Fragments 500 PBMC</code></li>
      <li>Now you can continue with either the <code class="language-plaintext highlighter-rouge">fragments_file</code> from earlier or the new file <code class="language-plaintext highlighter-rouge">Fragments 500 PBMC</code>.
        <ul>
          <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> The tool <code class="language-plaintext highlighter-rouge">pp.make_fragment_file</code> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> has implemented additional quality control (QC) measures. These filter out larger fragments, which will be noticeable in the log-scale fragment size distribution.</li>
          <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> Please note that <code class="language-plaintext highlighter-rouge">Fragments 500 PBMC</code> only contains 500 <abbr title="peripheral blood mononuclear cells">PBMCs</abbr> and thus the clustering will produce different outputs compared to the outputs generated by <code class="language-plaintext highlighter-rouge">fragments_file</code> (with 5k PBMC).</li>
        </ul>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="preprocessing">Preprocessing</h1>

<p>Preprocessing of the scATAC-seq data contained in the <code class="language-plaintext highlighter-rouge">fragment</code> file with SnapATAC2 begins with importing the files and computing basic <abbr title="quality control">QC</abbr> metrics.</p>

<p>SnapATAC2 compresses and stores the fragments into an <code class="language-plaintext highlighter-rouge">AnnData</code> object.</p>

<h2 id="anndata">AnnData</h2>
<p>The <a href="https://anndata.readthedocs.io/en/latest/"><code class="language-plaintext highlighter-rouge">AnnData</code></a> format was initially developed for the <a href="https://scanpy.readthedocs.io/en/stable/index.html"><code class="language-plaintext highlighter-rouge">Scanpy</code></a> package and is now a widely accepted data format to
store annotated data matrices in a space-efficient manner.</p>

<p><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/anndata_schema.svg" alt="Anndata format" title="&lt;code&gt;AnnData&lt;/code&gt; format stores a count matrix &lt;code&gt;X&lt;/code&gt; together with annotations of
observations (i.e. cells) &lt;code&gt;obs&lt;/code&gt;, variables (i.e. genes) &lt;code&gt;var&lt;/code&gt; and unstructured annotations &lt;code&gt;uns&lt;/code&gt;." /></p>

<h2 id="import-files-to-snapatac2">Import files to SnapATAC2</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-an-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Create an AnnData object</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Import data fragment files and compute basic QC metrics, using 'pp.import_data'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Fragment file, optionally compressed with gzip or zstd”</em>: <code class="language-plaintext highlighter-rouge">fragments_file.tsv</code> (Input dataset)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“A tabular file containing chromosome names and sizes”</em>: <code class="language-plaintext highlighter-rouge">chrom_sizes.txt</code> (Input dataset)</li>
            <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <em>“Whether the fragment file has been sorted by cell barcodes”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="details">
        <div class="box-title details-title" id="details-sorted-by-barcodes"><button class="gtn-boxify-button details" type="button" aria-controls="details-sorted-by-barcodes" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Sorted by barcodes</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <ul>
          <li>This tool requires the fragment file to be sorted according to cell barcodes.</li>
          <li>If <strong>pp.make_fragment_file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> was used to generate the fragment file, this has automatically been done.
            <ul>
              <li>Otherwise, the setting <em>“sorted by cell barcodes”</em> should remain <code class="language-plaintext highlighter-rouge">No</code>.</li>
            </ul>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC</code></p>
    </li>
    <li>Check that the format is <code class="language-plaintext highlighter-rouge">h5ad</code></li>
  </ol>
</blockquote>

<p>Because the <code class="language-plaintext highlighter-rouge">AnnData</code> format is an extension of the HDF5 format, i.e. a binary format, an <code class="language-plaintext highlighter-rouge">AnnData</code> object can not be inspected directly in Galaxy by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (<strong>View data</strong>) icon.
Instead, we need to use a dedicated tool from the <strong>AnnData</strong> suite.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-an-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect an AnnData object</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0" title="Inspect AnnData tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Inspect AnnData</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC</code></li>
        <li><em>“What to inspect?”</em>: <code class="language-plaintext highlighter-rouge">General information about the object</code></li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the generated file</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 14232 × 0
 obs: 'n_fragment', 'frac_dup', 'frac_mito'
 uns: 'reference_sequences'
 obsm: 'fragment_paired'
</code></pre></div>        </div>

        <ol>
          <li>How many observations are there? What do they represent?</li>
          <li>How many variables are there?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>There are 14232 observations, representing the cells.</li>
            <li>There are 0 variables, representing genomic regions. This is because genome-wide 500-bp bins are only added after initial filtering.</li>
          </ol>

        </blockquote>

      </blockquote>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-faster-method-for-general-information"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-faster-method-for-general-information" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Faster Method for General Information</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <ul>
          <li>Many toolsets producing outputs in <em>AnnData</em> formats in Galaxy, provide the general information by default:
            <ul>
              <li>Click on the name of the dataset in the history to expand it.</li>
              <li>
                <p>General Anndata information will be given in the expanded box:</p>

                <p>e.g.</p>

                <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs x n_vars]
-    14232 × 0
</code></pre></div>                </div>
              </li>
              <li>
                <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> This feature isn’t the most reliable and might display:</p>

                <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs x n_vars]
-    1 × 1
</code></pre></div>                </div>
              </li>
            </ul>
          </li>
          <li>In such cases and for more specific queries, <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0" title="Inspect AnnData tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Inspect AnnData</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.3+galaxy0)</span> is required.</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="calculate-and-visualize-qc-metrics">Calculate and visualize <abbr title="quality control">QC</abbr> metrics</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-fragment-size-distribution"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Fragment-size distribution</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot fragment size distribution, using 'pl.frag_size_distr'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC</code> (output of <strong>pp.import_data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot fragment size distribution, using 'pl.frag_size_distr'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC</code> (output of <strong>pp.import_data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Change the y-axis (fragment counts) to log scale”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/pl.frag_size.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/pl.frag_size.png" alt="fragment_size_distribution. " width="600" height="400" loading="lazy" /></a></p>

  <p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/log_pl.frag_size.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/log_pl.frag_size.png" alt="log_fragment_size_distribution. " width="600" height="400" loading="lazy" /></a></p>
  <blockquote class="question">
    <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>What distinct features do the plots have? And what do they represent?</li>
      <li>Which fragments are generally from open chromatin?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>3 peaks are clearly visible (at &lt;100-bp, ~200-bp and ~400-bp). The smallest fragments are from nucleosome-free regions, while the larger peaks of 200- and 400-bp contain mono- and di-nucleosome fragments, respectively.</li>
        <li>The small fragments (&lt;100-bp) are from open chromatin reads, since the Tn5 transposase could easily access the loosely packed DNA (<span class="citation"><a href="#Yan2020">Yan <i>et al.</i> 2020</a></span>).</li>
      </ol>

    </blockquote>

  </blockquote>
</blockquote>

<p>The transcription start site enrichment (TSSe) is another important <abbr title="quality control">QC</abbr> metric. Nucleosome-free fragments are expected to be enriched at transcription start sites (TSS). TSSe shows increased fragmentation of chromatin around the TSS. This suggests open and accessible nucleosome-free chromatin.</p>

<p><abbr title="transcription start site enrichment">TSSe</abbr> is used as a QC metric, since an increased enrichment around TSS regions suggests that the experiment has captured biological meaningful genomic features.
TSSe scores of individual cells can be calculated using SnapATAC2’s <em>metrics.tsse</em> function.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-calculate-and-plot-tsse"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Calculate and Plot TSSe</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Compute the TSS enrichment score (TSSe) for each cell, using 'metrics.tsse'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC</code> (output of <strong>pp.import_data</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF/GFF file containing the gene annotation”</em>: <code class="language-plaintext highlighter-rouge">gene_annotation.gtf.gz</code> (Input dataset)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC TSSe</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">TSSe</code> to the dataset:</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot the TSS enrichment vs. number of fragments density figure, using 'pl.tsse'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC TSSe</code> (output of <strong>metrics.tsse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/pl.tsse.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/pl.tsse.png" alt="TSSe plot against number of unique fragments. " width="600" height="400" loading="lazy" /></a>
High-quality cells can be identified in the plot of <abbr title="transcription start site enrichment">TSSe</abbr> scores against a number of unique fragments for each cell.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>Where are high-quality cells located in the plot?</li>
      <li>Based on this plot, how should the filter be set?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>The cells in the upper right are high-quality cells, enriched for <abbr title="transcription start sites">TSS</abbr>. Fragments in the lower left represent low-quality cells or empty droplets and should be filtered out.</li>
        <li>Setting the minimum number of counts at 5,000 and the minimum TSS enrichment to 10.0 is an adequate filter.</li>
      </ol>

    </blockquote>

  </blockquote>
</blockquote>

<h2 id="filtering">Filtering</h2>
<p>Based on the <abbr title="transcription start site enrichment">TSSe</abbr> plot the cells can be filtered by TSSe and fragment counts.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-cells"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Filter cells</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Filter cell outliers based on counts and numbers of genes expressed, using 'pp.filter_cells'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC TSSe</code> (output of <strong>metrics.tsse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Minimum number of counts required for a cell to pass filtering”</em>: <code class="language-plaintext highlighter-rouge">5000</code></li>
            <li><em>“Minimum TSS enrichemnt score required for a cell to pass filtering”</em>: <code class="language-plaintext highlighter-rouge">10.0</code></li>
            <li><em>“Maximum number of counts required for a cell to pass filtering”</em>: <code class="language-plaintext highlighter-rouge">100000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC TSSe filtered</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">filtered</code> to the dataset</li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4564 × 0
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse'
 uns: 'reference_sequences'
 obsm: 'fragment_paired'
</code></pre></div>        </div>

        <ol>
          <li>How have the observations changed, compared to the first <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC</code> AnnData file?</li>
          <li>What does this tell us about the quality of the data?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>
              <p>There are only 4564 observations, compared to the initial 14232 observations.</p>

              <p>And the <code class="language-plaintext highlighter-rouge">obs: 'tsse'</code> has been added (but already during <strong>metrics.tsse</strong><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</p>
            </li>
            <li>
              <p>The empty droplets and low-quality cells have been filtered out, leaving us with 4564 high-quality cells.</p>
            </li>
          </ol>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="feature-selection">Feature selection</h2>
<p>Currently, our AnnData matrix does not contain any variables. The variables will be added in the following step with the function <em>pp.add_tile_matrix</em>.
This creates a cell-by-bin matrix containing insertion counts across genome-wide 500-bp bins.</p>

<p>After creating the variables, the most accessible features are selected.</p>
<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-select-features"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Select features</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Generate cell by bin count matrix, using 'pp.add_tile_matrix'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC TSSe filtered</code> (output of <strong>pp.filter_cells</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“The size of consecutive genomic regions used to record the counts”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
            <li><em>“The strategy to compute feature counts”</em>: <code class="language-plaintext highlighter-rouge">"insertion": based on the number of insertions that overlap with a region of interest</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Perform feature selection, using 'pp.select_features'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata tile_matrix</code> (output of <strong>pp.add_tile_matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Number of features to keep”</em>: <code class="language-plaintext highlighter-rouge">250000</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-select-features"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  Select features</div>

        <ul>
          <li>Including more features improves resolution and can reveal finer details, but it may also introduce noise.
            <ul>
              <li>To optimize results, experiment with the <code class="language-plaintext highlighter-rouge">n_features</code> parameter to find the most appropriate value for your dataset.</li>
            </ul>
          </li>
        </ul>

        <blockquote class="details">
          <div class="box-title details-title" id="details-different-number-of-features"><button class="gtn-boxify-button details" type="button" aria-controls="details-different-number-of-features" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  Different number of features </span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ul>
            <li>To demonstrate the differences when selecting features, the following UMAP plots are the outputs from processing with a number of features between 1,000 and 500,000.</li>
            <li>Fewer features result in fewer, but larger clusters. Selecting a lot of features will output more granular clusters and the compute time will increase.</li>
          </ul>
          <figure id="figure-2" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/number_features.png" alt="Different number of features UMAP. " width="2421" height="2004" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/number_features.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> UMAP plots with different selected features</figcaption></figure>

        </blockquote>
        <ul>
          <li>At this step you can provide a blacklist or whitelist to specifically select relevant features.
            <ul>
              <li>For example the <a href="https://github.com/Boyle-Lab/Blacklist"><strong>ENCODE Blacklist</strong></a> (<span class="citation"><a href="#Amemiya2019">Amemiya <i>et al.</i> 2019</a></span>) can be applied here.</li>
            </ul>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC select_features</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">select_features</code> to the dataset</li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4564 × 6062095
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse'
 var: 'count', 'selected'
 uns: 'reference_sequences'
 obsm: 'fragment_paired'
</code></pre></div>        </div>

        <ol>
          <li>How did <code class="language-plaintext highlighter-rouge">n_vars</code> change compared to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC TSSe filtered</code></li>
          <li>Where are the selected features stored in the count matrix?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>There are 6,062,095 variables, compared to 0 in <code class="language-plaintext highlighter-rouge">TSSe filtered</code>.</li>
            <li>Selected features are stored in <code class="language-plaintext highlighter-rouge">var: 'selected'</code>.</li>
          </ol>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="doublet-removal">Doublet removal</h2>

<p>Doublets are removed by calling a customized <a href="https://github.com/AllonKleinLab/scrublet"><strong>scrublet</strong></a> (<span class="citation"><a href="#Wolock2019">Wolock <i>et al.</i> 2019</a></span>) algorithm. <em>pp.scrublet</em> will identify potential doublets and the function <em>pp.filter_doublets</em> removes them.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-doublets-in-single-cell-datasets"><button class="gtn-boxify-button details" type="button" aria-controls="details-doublets-in-single-cell-datasets" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Doublets in Single-cell datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>During single-cell sequencing, multiple nuclei can be encapsulated into the same 10x gel bead. The resulting multiplets (&gt;97% of which are doublets) produce sequences from both cells.</li>
    <li>Doublets can confound the results by appearing as “new” clusters or artifactual intermediary cell states.
      <ul>
        <li>These problematic doublets are called <strong>neotypic</strong> doublets, since they appear as “new” cell types.</li>
      </ul>
    </li>
    <li><strong>Scrublet</strong> (Single-cell Remover of Doublets) is an algorithm which can detect neotypic doublets that produce false results.
      <ul>
        <li>The algorithm first simulates doublets by combining random pairs of observed cell features.</li>
        <li>The observed features of the “cells” are then compared to the simulated doublets and scored on their doublet probability.</li>
        <li>SnapATAC2’s <em>pp.filter_doublets</em> then removes all cells with a doublet probability &gt;50%.</li>
      </ul>
    </li>
  </ul>

  <figure id="figure-3" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/doublets-and-scrublet.png" alt="Doublet removal with scrublet. " width="1635" height="1635" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/doublets-and-scrublet.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Scrublet simulates expected doublets and produces doublet scores for each cell. (<span class="citation"><a href="#Wolock2019">Wolock <i>et al.</i> 2019</a></span>)</figcaption></figure>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-scrublet"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Scrublet</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Compute probability of being a doublet using the scrublet algorithm, using 'pp.scrublet'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC select_features</code> (output of <strong>pp.select_features</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Remove doublets according to the doublet probability or doublet score, using 'pp.filter_doublets'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata scrublet</code> (output of <strong>pp.scrublet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Threshold for doublet probability”</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC filter_doublets</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">filter_doublets</code> to the dataset</li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4430 × 6062095
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score'
 var: 'count', 'selected'
 uns: 'doublet_rate', 'reference_sequences', 'scrublet_sim_doublet_score'
 obsm: 'fragment_paired'
</code></pre></div>        </div>

        <ol>
          <li>What was removed by <strong>pp.filter_doublets</strong>?</li>
          <li>Where are the new annotations stored?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>Cell doublets were removed. <code class="language-plaintext highlighter-rouge">n_obs</code> was reduced from 4564 to 4430 cells.</li>
            <li>The outputs of <strong>pp.scrublet</strong> are stored in observations <code class="language-plaintext highlighter-rouge">obs: 'doublet_probability', 'doublet_score'</code> and in unstructured annotations <code class="language-plaintext highlighter-rouge">uns: 'scrublet_sim_doublet_score'</code>. The output of <strong>pp.filter_doublets</strong> is stored in <code class="language-plaintext highlighter-rouge">uns: 'doublet_rate'</code>.</li>
          </ol>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="dimension-reduction">Dimension reduction</h1>

<p>Dimension reduction is a very important step during the analysis of single cell data. During this, the complex multi-dimensional data is projected into lower-dimensional space, while the lower-dimensional embedding of the complex data retains as much information as possible. Dimension reduction enables batch correction, data visualization and quicker downstream analysis since the data is more simplified and the memory usage is reduced (<span class="citation"><a href="#Zhang2024">Zhang <i>et al.</i> 2024</a></span>).</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-dimension-reduction-with-snapatac2"><button class="gtn-boxify-button details" type="button" aria-controls="details-dimension-reduction-with-snapatac2" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Dimension reduction with SnapATAC2</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>Dimension reduction algorithms can be either linear or non-linear.</li>
    <li>
      <p>Linear methods are generally computationally efficient and well scalable.</p>

      <p>A popular linear dimension reduction algorithm is:</p>
      <ul>
        <li><strong>PCA</strong> (Principle Component Analysis), implemented in <strong>Scanpy</strong> (please check out our <a href="/training-material/topics/single-cell/tutorials/scrna-scanpy-pbmc3k/tutorial.html">Scanpy</a> tutorial for an explanation).</li>
      </ul>
    </li>
    <li>Nonlinear methods however are well suited for multimodal and complex datasets.
      <ul>
        <li>in contrast to linear methods, which often preserve global structures, non-linear methods have a locality-preserving character.</li>
        <li>This makes non-linear methods relatively insensitive to outliers and noise while emphasizing natural clusters in the data (<span class="citation"><a href="#Belkin2003">Belkin and Niyogi 2003</a></span>)</li>
        <li>As such, they are implemented in many algorithms to visualize the data in 2 dimensions (f.ex. <strong>UMAP</strong> embedding).</li>
      </ul>
    </li>
    <li>The nonlinear dimension reduction algorithm, through <em>matrix-free spectral embedding</em>, used in <strong>SnapATAC2</strong> is a very fast and memory efficient non-linear algorithm (<span class="citation"><a href="#Zhang2024">Zhang <i>et al.</i> 2024</a></span>).
      <ul>
        <li><strong>Spectral embedding</strong> utilizes an iterative algorithm to calculate the <strong>spectrum</strong> (<em>eigenvalues</em> and <em>eigenvectors</em>) of a matrix without computing the matrix itself.</li>
      </ul>
    </li>
    <li>For a simple introduction into <em>spectral embedding</em> and how it compares to <em>PCA</em>, please check out the blog post <a href="https://juanitorduz.github.io/laplacian_eigenmaps_dim_red/">“On Laplacian Eigenmaps for Dimensionality Reduction”</a> by Juan Orduz.</li>
  </ul>

</blockquote>

<h2 id="spectral-embedding">Spectral embedding</h2>
<p>The dimension reduction, produced by the algorithm <em>tl.spectral</em>, is required for later steps, such as plotting and clustering.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-spectral-embedding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Spectral embedding</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Perform dimension reduction using Laplacian Eigenmap, using 'tl.spectral'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC filter_doublets</code> (output of <strong>pp.filter_doublets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Distance metric”</em>: <code class="language-plaintext highlighter-rouge">cosine</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-distance-metric"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  Distance metric</div>

        <ul>
          <li>The fast and well scalable <em>“matrix-free spectral embedding”</em> algorithm depends on the distance metric: <code class="language-plaintext highlighter-rouge">cosine</code></li>
        </ul>
      </blockquote>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC spectral</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">spectral</code> to the dataset</li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4430 × 6062095
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score'
 var: 'count', 'selected'
 uns: 'doublet_rate', 'reference_sequences', 'scrublet_sim_doublet_score', 'spectral_eigenvalue'
 obsm: 'fragment_paired', 'X_spectral'
</code></pre></div>        </div>

        <p>Where are the new annotations stored?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <p>The outputs of <strong>tl.spectral</strong> are stored in unstructured annotations <code class="language-plaintext highlighter-rouge">uns: 'spectral_eigenvalue'</code> and as multidimensional observations <code class="language-plaintext highlighter-rouge">obsm: 'X_spectral'</code>.</p>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>
<h2 id="umap-embedding">UMAP embedding</h2>
<p>With the already reduced dimensionality of the data stored in <code class="language-plaintext highlighter-rouge">X_spectral</code>, the cells can be further embedded (i.e. transformed into lower dimensions) with Uniform Manifold Approximation and Projection (UMAP).
<strong>UMAP</strong> projects the cells and their relationship to each other into 2-dimensional space, which can be easily visualized (<span class="citation"><a href="#McInnes2018">McInnes <i>et al.</i> 2018</a></span>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-umap-embedding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  UMAP embedding</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Compute Umap, using 'tl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC spectral</code> (output of <strong>tl.spectral</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Use the indicated representation in ‘.obsm’“</em>: <code class="language-plaintext highlighter-rouge">X_spectral</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC UMAP</code> or add the tag  <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">UMAP</code> to the dataset</li>
  </ol>
</blockquote>

<h1 id="clustering">Clustering</h1>
<p>During clustering, cells that share similar accessibility profiles are organized into clusters. <strong>SnapATAC2</strong> utilizes graph-based community clustering with the <em>Leiden</em> algorithm (<span class="citation"><a href="#Traag2019">Traag <i>et al.</i> 2019</a></span>).
This method takes the k-nearest neighbor (KNN) graph as input data and produces well-connected communities.</p>

<h2 id="community-clustering">Community clustering</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-clustering-analysis"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Clustering analysis</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Compute a neighborhood graph of observations, using 'pp.knn'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC UMAP</code> (output of <strong>tl.umap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1" title="SnapATAC2 Clustering tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Clustering</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Dimension reduction and Clustering”</em>: <code class="language-plaintext highlighter-rouge">Cluster cells into subgroups, using 'tl.leiden'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata knn</code> (output of <strong>pp.knn</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Whether to use the Constant Potts Model (CPM) or modularity”</em>: <code class="language-plaintext highlighter-rouge">modularity</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-cpm-or-modularity"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  CPM or modularity</div>
        <ul>
          <li>make sure you selected <code class="language-plaintext highlighter-rouge">modularity</code></li>
          <li>the clusters produced by <code class="language-plaintext highlighter-rouge">CPM</code> are not represented well in the UMAP projections</li>
        </ul>

      </blockquote>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">leiden</code> to the dataset</li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4564 × 6062095
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score', 'leiden'
 var: 'count', 'selected'
 uns: 'doublet_rate', 'reference_sequences', 'scrublet_sim_doublet_score', 'spectral_eigenvalue'
 obsm: 'X_spectral', 'X_umap', 'fragment_paired'
 obsp: 'distances'
</code></pre></div>        </div>

        <p>Where are the <strong>leiden</strong> clusters stored in the AnnData?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <p>The clusters are stored in <code class="language-plaintext highlighter-rouge">obs: 'leiden'</code></p>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="plotting-of-clusters">Plotting of clusters</h2>
<p>Now that we have produced <strong>UMAP</strong> embeddings of our cells and have organized the cells into <strong>leiden</strong> clusters, we can now visualize this information with <em>pl.umap</em>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plotting-the-clusters"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Plotting the clusters</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1" title="SnapATAC2 Plotting tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Plotting</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Plot the UMAP embedding, using 'pl.umap'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code> (output of <strong>tl.leiden</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>“Color”</em>: <code class="language-plaintext highlighter-rouge">leiden</code></li>
            <li><em>“Height of the plot”</em>: <code class="language-plaintext highlighter-rouge">500</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/pl.umap.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/pl.umap.png" alt="umap_leiden_clustering. " width="600" height="500" loading="lazy" /></a></p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How many leiden clusters were discovered?</li>
      <li>What does the distance of clusters to each other tell us about their chromatin states?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>There are 13 leiden clusters.</li>
        <li>Clusters in close proximity (f.ex. clusters 0 and 5) share a similar chromatin accessibility profile (and very likely also a similar cell type), compared to a cluster further away (f.ex. cluster 9).</li>
      </ol>

    </blockquote>

  </blockquote>
</blockquote>

<h1 id="cell-cluster-annotation">Cell cluster annotation</h1>

<p>After clustering the cells, they must be annotated. This categorizes the clusters into known cell types. <strong>Manual Cell Annotation</strong> requires known marker genes and varying expression profiles of the marker genes among clusters.</p>

<p>Luckily, the marker genes for <abbr title="peripheral blood mononuclear cells">PBMCs</abbr> are known (<span class="citation"><a href="#Sun2019">Sun <i>et al.</i> 2019</a></span>).
Marker genes for other single cell datasets can also be found in databases such as <a href="https://panglaodb.se/markers.html">PanglaoDB</a> (<span class="citation"><a href="#Franzn2019">Franzén <i>et al.</i> 2019</a></span>). Using the known marker genes we can now annotate our clusters manually.</p>

<h2 id="gene-matrix">Gene matrix</h2>
<p>Since our data currently doesn’t contain gene information, we have to create a cell-by-gene activity matrix using the function <em>pp.make_gene_matrix</em>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Task description</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1" title="SnapATAC2 Preprocessing tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Preprocessing</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Method used for preprocessing”</em>: <code class="language-plaintext highlighter-rouge">Generate cell by gene activity matrix, using 'pp.make_gene_matrix'</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code> (output of <strong>tl.leiden</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF/GFF file containing the gene annotation”</em>: <code class="language-plaintext highlighter-rouge">gene_annotation.gtf.gz</code> (Input dataset)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">gene_matrix</code> to the dataset
      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-gene-matrix"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-gene-matrix" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip:  Gene matrix </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>Please note that <em>pp.make_gene_matrix</em> removes all annotations except those stored in <code class="language-plaintext highlighter-rouge">obs</code>.</li>
          <li>Therefore it might be necessary to remove propagating tags <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> (tags starting with <code class="language-plaintext highlighter-rouge">#</code>) from <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix</code>.
            <ul>
              <li>Tags can be removed by expanding the dataset with a tag and clicking <i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden">galaxy-cross</span> next to the tag.</li>
            </ul>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-10"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4430 × 60606
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score', 'leiden'
</code></pre></div>        </div>

        <p>What does <code class="language-plaintext highlighter-rouge">n_vars</code> represent in <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix</code> and what did it represent in <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code>?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ul>
            <li>The variables now represent accessible genes. There are 60606 accessible genes in our samples. In <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code> and all earlier AnnData the variables represented the 6062095 fixed-sized genomic bins.</li>
          </ul>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="imputation-with-scanpy-and-magic">Imputation with Scanpy and MAGIC</h2>
<p>Similar to scRNA-seq data, the cell-by-gene-activity matrix is very sparse. Additionally, high gene variance between cells, due to technical confounders, could impact the downstream analysis.
In scRNA-seq, filtering and normalization are therefore required to produce a high-quality gene matrix.</p>

<p>Since the <em>cell-by-gene-activity</em> matrix resembles the <em>cell-by-gene-expression</em> matrix of scRNA-seq, we can use the tools of the <a href="https://scanpy.readthedocs.io/en/stable/index.html">Scanpy</a> (<span class="citation"><a href="#Wolf2018">Wolf <i>et al.</i> 2018</a></span>) tool suite to continue with our data.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-imputation-with-magic"><button class="gtn-boxify-button details" type="button" aria-controls="details-imputation-with-magic" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Imputation with MAGIC</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>The count matrices of single-cell data are sparse and noisy.</li>
    <li>Confounding issues, such as “dropout” effects, where some mRNA or DNA-segments are not detected although they are present in the cell, also result in some cells missing important cell-type defining features.
      <ul>
        <li>These problems can obscure the data, as only the strongest gene-gene relationships are still detectable.</li>
      </ul>
    </li>
    <li>The <em>Markov Affinity-based Graph Imputation of Cells</em> (MAGIC) algorithm (<span class="citation"><a href="#vanDijk2018">van Dijk <i>et al.</i> 2018</a></span>) tries to solve these issues by filling in missing data from some cells with transcript information from similar cells.
      <ul>
        <li>The algorithm calculates the likely gene expression of a single cell, based on similar cells and fills in the missing data to produce the expected expression.
          <ul>
            <li><em>MAGIC</em> achieves this by building a graph from the data and using data diffusion to smooth out the noise.</li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>

  <figure id="figure-4" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/magic_method.png" alt="Imputation with the MAGIC algorithm. " width="1635" height="1635" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/magic_method.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> MAGIC restores noisy and sparse single-cell data using diffusion geometry (<span class="citation"><a href="#vanDijk2018">van Dijk <i>et al.</i> 2018</a></span>)</figcaption></figure>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-and-normalize"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Filter and normalize</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.9.6+galaxy3" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix</code> (output of <strong>pp.make_gene_matrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used for filtering”</em>: <code class="language-plaintext highlighter-rouge">Filter genes based on number of cells or counts, using 'pp.filter_genes'</code>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Minimum number of cells expressed</code>
              <ul>
                <li><em>“Minimum number of cells expressed required for a gene to pass filtering”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.9.6+galaxy3" title="Normalize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Normalize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata filter_genes</code> (output of <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used for normalization”</em>: <code class="language-plaintext highlighter-rouge">Normalize counts per cell, using 'pp.normalize_total'</code></li>
        <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <em>“Exclude (very) highly expressed genes for the computation of the normalization factor (size factor) for each cell”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.9.6+galaxy3" title="Inspect and manipulate tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Inspect and manipulate</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata normalize</code> (output of <strong>Normalize</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used for inspecting”</em>: <code class="language-plaintext highlighter-rouge">Logarithmize the data matrix, using 'pp.log1p'</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.9.6+galaxy3" title="Normalize tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Normalize</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata log1p</code> (output of <strong>log1p</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used for normalization”</em>: <code class="language-plaintext highlighter-rouge">Denoising using Markov Affinity-based Graph Imputation of Cells (MAGIC) API 'external.pp.magic'</code>
          <ul>
            <li><em>“Denoised genes to return”</em>: <code class="language-plaintext highlighter-rouge">All genes</code></li>
            <li><em>“Which solver to use”</em>: <code class="language-plaintext highlighter-rouge">"approximate", is faster that performs imputation in the PCA space and then projects back to the gene space</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:</div>
        <ul>
          <li>Choosing the setting <code class="language-plaintext highlighter-rouge">Which solver to use: 'exact'</code> will result in a output file with better resolution.</li>
          <li>This is not necessary for our purposes, since the compute time also increases with this setting.</li>
        </ul>

      </blockquote>
    </li>
    <li>
      <p>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC magic</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">magic</code> to the dataset</p>
    </li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-11"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs × n_vars = 4430 × 55106
 obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score', 'leiden', 'n_genes', 'n_counts'
 var: 'n_cells', 'n_counts'
 uns: 'log1p'
</code></pre></div>        </div>

        <ol>
          <li>How did <code class="language-plaintext highlighter-rouge">n_vars</code> change, compared to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix</code>?</li>
          <li>Which data was ‘lost’, compared to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code>, and must be added to the file in order to produce <strong>UMAP</strong> plots?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>The number of genes was reduced from 60606 to 55106 by the filtering. Additional annotations were added, such as: <code class="language-plaintext highlighter-rouge">obs: 'n_genes', 'n_counts'</code>, <code class="language-plaintext highlighter-rouge">var: 'n_cells', 'n_counts'</code> and <code class="language-plaintext highlighter-rouge">uns: 'log1p'</code>.</li>
            <li>The UMAP embeddings <code class="language-plaintext highlighter-rouge">obsm: 'X_umap'</code> are missing and should be added to the Anndata in the next step. Without <code class="language-plaintext highlighter-rouge">X_umap</code> it won’t be possible to visualize the plots.</li>
          </ol>

        </blockquote>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="copy-over-embeddings">Copy-over embeddings</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-copy-umap-embedding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Copy UMAP embedding</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.9.3+galaxy0" title="AnnData Operations tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>AnnData Operations</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input object in hdf5 AnnData format”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC magic</code> (output of <strong>external.pp.magic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Copy embeddings (such as UMAP, tSNE)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Keys from embeddings to copy”</em>: <code class="language-plaintext highlighter-rouge">X_umap</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“IAnnData objects with embeddings to copy”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC leiden</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-annotations-to-copy"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  Annotations to copy</div>

        <ul>
          <li>This tutorial only focuses on producing an <strong>UMAP</strong> plot with marker-genes.</li>
          <li>If further analysis, with tools requiring more annotations, is intended, these can be added in a similar way as shown above.
            <ul>
              <li>f.ex. <em>Peak and Motif Analysis</em> with <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_peaks_and_motif/snapatac2_peaks_and_motif/2.6.4+galaxy1" title="Snapatac2 peaks and motif tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Snapatac2 peaks and motif</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span> requires annotations from <code class="language-plaintext highlighter-rouge">uns</code>.</li>
            </ul>
          </li>
          <li>It is also possible to leave the input <em>“Keys from embeddings to copy”</em> empty, to copy all annotations of a given category such as <code class="language-plaintext highlighter-rouge">obsm</code>.</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC magic UMAP</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">UMAP</code> to the dataset</p>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the general information of the <code class="language-plaintext highlighter-rouge">.h5ad</code> output, to check if <code class="language-plaintext highlighter-rouge">obsm</code> contains <code class="language-plaintext highlighter-rouge">X_umap</code></li>
  </ol>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>  AnnData object with n_obs × n_vars = 4430 × 55106
   obs: obs: 'n_fragment', 'frac_dup', 'frac_mito', 'tsse', 'doublet_probability', 'doublet_score', 'leiden', 'n_genes', 'n_counts'
   var: 'n_cells', 'n_counts'
   uns: 'log1p'
   obsm: 'X_umap'
</code></pre></div>  </div>

</blockquote>

<h2 id="visualize-gene-activity-of-marker-genes">Visualize gene activity of marker genes</h2>
<p>The gene activity of selected marker genes can now be visualized with Scanpy.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plot-marker-genes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Plot marker genes</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.9.6+galaxy3" title="Plot with Scanpy tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Plot with Scanpy</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">output_h5ad</code> (output of <strong>AnnData Operations</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Embeddings: Scatter plot in UMAP basis, using 'pl.umap'</code>
          <ul>
            <li><em>“Keys for annotations of observations/cells or variables/genes”</em>: <code class="language-plaintext highlighter-rouge">leiden, CD3D, CD8A, CD4, MS4A1, NKG7, CD14, FCER1A</code></li>
            <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <em>“Show edges?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li>In <em>“Plot attributes”</em>
              <ul>
                <li><em>“Number of panels per row”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/umap_leiden_marker-genes.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/umap_leiden_marker-genes.png" alt="umap_leiden_marker_gene_clustering. " width="1177" height="1682" loading="lazy" /></a></p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-12"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>Are the marker genes selectively expressed in the clusters?</li>
      <li>Which marker genes have overlapping expression profiles? And what could that imply?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-12"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-12" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>Some marker genes, such as <code class="language-plaintext highlighter-rouge">MS4A1</code> or <code class="language-plaintext highlighter-rouge">CD8A</code>, are only expressed in a few clusters (clusters 6+11 and clusters 1+8, respectively).</li>
        <li>The marker gene <code class="language-plaintext highlighter-rouge">CD3D</code> is expressed in multiple clusters (1, 2, 4, 7 and 8). Overlapping expression profiles imply similar cell types since similar cell types have similar marker genes upregulated. In this case, <code class="language-plaintext highlighter-rouge">CD3D</code> expression classifies the cells in these clusters as T-cells.</li>
      </ol>

    </blockquote>

  </blockquote>
</blockquote>

<h2 id="manual-cluster-annotation">Manual cluster annotation</h2>
<p>Comparison of marker gene expression in our clusters with a table of canonical marker genes (<span class="citation"><a href="#Sun2019">Sun <i>et al.</i> 2019</a></span>), enables us to annotate the clusters manually.</p>

<table>
  <thead>
    <tr>
      <th>Cell type</th>
      <th>Marker genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>CD8+ T cells</td>
      <td>CD3D+, CD8A+, CD4-</td>
    </tr>
    <tr>
      <td>CD4+ T cells</td>
      <td>CD3D+, CD8A-, CD4+</td>
    </tr>
    <tr>
      <td>B cells</td>
      <td>CD3D-, MS4A1+</td>
    </tr>
    <tr>
      <td>Natural killer (NK) cells</td>
      <td>CD3D-, NKG7+</td>
    </tr>
    <tr>
      <td>Monocytes</td>
      <td>CD3D-, CD14+</td>
    </tr>
    <tr>
      <td>Dendritic cells</td>
      <td>CD3D-, FCER1A+</td>
    </tr>
  </tbody>
</table>

<p>These canonical marker genes can match the clusters to known cell types:</p>

<table>
  <thead>
    <tr>
      <th>Cluster</th>
      <th>Cell type</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>Monocytes</td>
    </tr>
    <tr>
      <td>1</td>
      <td>CD8+ T cells</td>
    </tr>
    <tr>
      <td>2</td>
      <td>CD4+ T cells</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Monocytes</td>
    </tr>
    <tr>
      <td>4</td>
      <td>CD4+ T cells</td>
    </tr>
    <tr>
      <td>5</td>
      <td>Monocytes</td>
    </tr>
    <tr>
      <td>6</td>
      <td>B cells</td>
    </tr>
    <tr>
      <td>7</td>
      <td>CD4+ T cells</td>
    </tr>
    <tr>
      <td>8</td>
      <td>CD8+ T cells</td>
    </tr>
    <tr>
      <td>9</td>
      <td>NK cells</td>
    </tr>
    <tr>
      <td>10</td>
      <td>Monocytes</td>
    </tr>
    <tr>
      <td>11</td>
      <td>B cells</td>
    </tr>
    <tr>
      <td>12</td>
      <td>Dendritic cells</td>
    </tr>
  </tbody>
</table>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>Note that some clusters contain subtypes (f.ex. the annotated B cell clusters contain both naive and memory B cells). The cell-type annotation can be refined by choosing more specific marker genes.</p>
</blockquote>

<p>To manually annotate the <em>Leiden</em> clusters, we will need to perform multiple steps:</p>

<ol>
  <li><strong>Inspect</strong> the key-indexed observations of <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix magic UMAP</code></li>
  <li><strong>Cut</strong> the <em>Leiden</em> annotations out of the table</li>
  <li><strong>Upload</strong> a <em>replace file</em> containing the new cell type annotations for the <em>Leiden</em> clusters</li>
  <li><strong>Replace</strong> the values of the cluster annotation with cell type annotation</li>
  <li><strong>Add</strong> the cell type annotation to the AnnData</li>
  <li><strong>Plot</strong> the annotated cell types</li>
</ol>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-manual-annotation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Manual annotation</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0" title="Inspect AnnData tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Inspect AnnData</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix magic UMAP</code></li>
        <li><em>“What to inspect?”</em>: <code class="language-plaintext highlighter-rouge">Key-indexed observations annotation</code></li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">5k PBMC observations</code></li>
    <li>
      <p><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the generated file</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-13"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
        <p>In which column is the <strong>Leiden</strong> annotation located?</p>
        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-13"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-13" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>The <strong>Leiden</strong> annotation is in column 8.</p>

          <table>
            <thead>
              <tr>
                <th>Column 1</th>
                <th>Column 2</th>
                <th>Column 3</th>
                <th>Column 4</th>
                <th>Column 5</th>
                <th>Column 6</th>
                <th>Column 7</th>
                <th>Column 8</th>
                <th>Column 9</th>
                <th>Column 10</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>””</td>
                <td>n_fragment</td>
                <td>frac_dup</td>
                <td>frac_mito</td>
                <td>tsse</td>
                <td>doublet_probability</td>
                <td>doublet_score</td>
                <td>leiden</td>
                <td>n_genes</td>
                <td>n_counts</td>
              </tr>
              <tr>
                <td>AAACGAAAGACGTCAG-1</td>
                <td>22070</td>
                <td>0.5219425551271499</td>
                <td>0.0</td>
                <td>30.43315066436454</td>
                <td>0.004634433324822066</td>
                <td>0.009276437847866418</td>
                <td>8</td>
                <td>52303</td>
                <td>16521.599844068267</td>
              </tr>
              <tr>
                <td>AAACGAAAGATTGACA-1</td>
                <td>10500</td>
                <td>0.5345125681606597</td>
                <td>0.0</td>
                <td>29.10551296093465</td>
                <td>0.004668403569267374</td>
                <td>0.001088139281828074</td>
                <td>1</td>
                <td>54501</td>
                <td>15020.42495602328</td>
              </tr>
              <tr>
                <td>AAACGAAAGGGTCCCT-1</td>
                <td>19201</td>
                <td>0.5101785714285714</td>
                <td>0.0</td>
                <td>19.90011850347046</td>
                <td>0.004634433324822066</td>
                <td>0.009276437847866418</td>
                <td>5</td>
                <td>54212</td>
                <td>16294.751533305309</td>
              </tr>
              <tr>
                <td>AAACGAACAATTGTGC-1</td>
                <td>13242</td>
                <td>0.487399837417257</td>
                <td>0.0</td>
                <td>29.060913705583758</td>
                <td>0.004660125753854076</td>
                <td>0.0022172949002217295</td>
                <td>7</td>
                <td>53530</td>
                <td>15456.629863655084</td>
              </tr>
            </tbody>
          </table>

        </blockquote>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/cut_columns/Cut1/1.0.2" title="Cut columns tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut columns</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.2)</span> with the following parameters:
      <ul>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c8</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">5k PBMC observations</code> (output of <strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>Create a new <strong>tabular</strong> file from the following
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>leiden cell_type
0 Monocytes
1 CD8_Tcells
2 CD4_Tcells
3 Monocytes
4 CD4_Tcells
5 Monocytes
6 Bcells
7 CD4_Tcells
8 CD8_Tcells
9 NKcells
10 Monocytes
11 Bcells
12 Dendritic_cells
</code></pre></div>      </div>
      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-file"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-file" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>   <li>     <p>Paste the file contents into the text field</p>   </li>   <li>Change the dataset name from “New File” to <code class="language-plaintext highlighter-rouge">replace_file</code></li>   <li>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code>* From the Settings menu (<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span>) select <strong>Convert spaces to tabs</strong>* Press <strong>Start</strong> and <strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="details">
        <div class="box-title details-title" id="details-replace-file"><button class="gtn-boxify-button details" type="button" aria-controls="details-replace-file" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Replace file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>The first column of the replace file contains the “old” annotations and the second column contains the “new” annotation.</li>
          <li><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Spaces in the new annotations can lead to errors. Please use underscores (<code class="language-plaintext highlighter-rouge">_</code>) instead.</li>
        </ul>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file/0.2" title="Replace column tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Replace column</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File in which you want to replace some values”</em>: <code class="language-plaintext highlighter-rouge">Cut columns leiden</code> (output of <strong>Cut columns</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Replace information file”</em>: <code class="language-plaintext highlighter-rouge">replace_file</code></li>
        <li><em>“Which column should be replaced?</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Cell type annotation</code> and <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> inspect the file to check if the replacement was successful
      <ul>
        <li>check if the datatype is set to <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li>
          <p>you may need to change the datatype manually</p>

          <!--SNIPPET-->
          <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">tabular</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
          <p><!--END_SNIPPET--></p>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.3+galaxy0" title="Manipulate AnnData tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Manipulate AnnData</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix magic UMAP</code></li>
        <li><em>“Function to manipulate the object”</em>: <code class="language-plaintext highlighter-rouge">Add new annotation(s) for observations or variables</code>
          <ul>
            <li><em>“What to annotate”</em>: <code class="language-plaintext highlighter-rouge">Observations (obs)</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Table with new annotations”</em>: <code class="language-plaintext highlighter-rouge">Cell type annotation</code>(output of <strong>Replace column</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file to <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix magic cell_type</code> or add the tag <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> <code class="language-plaintext highlighter-rouge">cell_type</code> to the dataset</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.9.6+galaxy3" title="Plot with Scanpy tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Plot with Scanpy</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Annotated data matrix”</em>: <code class="language-plaintext highlighter-rouge">Anndata 5k PBMC gene_matrix magic cell_type</code> (output of <strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method used for plotting”</em>: <code class="language-plaintext highlighter-rouge">Embeddings: Scatter plot in UMAP basis, using 'pl.umap'</code></li>
      </ul>
      <ul>
        <li><em>“Keys for annotations of observations/cells or variables/genes”</em>: <code class="language-plaintext highlighter-rouge">cell_type</code></li>
        <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <em>“Show edges?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>In <em>“Plot attributes”</em>
          <ul>
            <li><em>“Location of legend”</em>: <code class="language-plaintext highlighter-rouge">on data</code></li>
            <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <em>“Draw a frame around the scatter plot?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> Inspect the <code class="language-plaintext highlighter-rouge">.png</code> output</li>
  </ol>

  <p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/umap_cell-types.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/umap_cell-types.png" alt="UMAP annotated cell types. " width="535" height="429" loading="lazy" /></a></p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-14"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <ol>
      <li>Are clusters with the same assigned cell type located close to each other?</li>
      <li>
        <p>Are the <a href="https://www.akadeum.com/blog/what-are-pbmc-human-pbmc-cells/">expected cell type percentages of <abbr title="peripheral blood mononuclear cells">PBMCs</abbr></a> visible in the annotated plot?</p>

        <table>
          <thead>
            <tr>
              <th>Cell Type</th>
              <th>Expected Percentage</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>CD3+ T cells</td>
              <td>50-70%</td>
            </tr>
            <tr>
              <td>CD4+ T cells</td>
              <td>25-45%</td>
            </tr>
            <tr>
              <td>CD8+ T cells</td>
              <td>10-25%</td>
            </tr>
            <tr>
              <td>B cells</td>
              <td>5-15%</td>
            </tr>
            <tr>
              <td>NK cells</td>
              <td>&lt;5%</td>
            </tr>
            <tr>
              <td>Monocytes</td>
              <td>10-30%</td>
            </tr>
            <tr>
              <td>Dendritic cells</td>
              <td>&lt;2%</td>
            </tr>
          </tbody>
        </table>
      </li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-14"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-14" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>Yes. B-cells are far away from NK and T cells. Only the myeloid lineage of monocytes and dendritic cells are located close to each other. This might be due to the common progenitor cell lineage and thus a similar chromatin profile.</li>
        <li>Yes. The most common cell types are T cells, followed by Monocytes and B cells. NK cells and Dendritic cells only make up a small percentage of PBMCs.</li>
      </ol>
    </blockquote>

  </blockquote>

</blockquote>

<h1 id="conclusion">Conclusion</h1>
<p><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Well done, you’ve made it to the end! You might want to consult your results with this <a href="https://usegalaxy.eu/u/videmp/h/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">control history</a>, or check out the <a href="https://usegalaxy.eu/u/videmp/w/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">full workflow</a> for this tutorial.</p>

<p>In this tutorial, we produced a count matrix of <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> reads in the <code class="language-plaintext highlighter-rouge">AnnData</code> format and performed:</p>
<ol>
  <li>Preprocessing:
    <ol>
      <li>Plotting the fragment-size distributions</li>
      <li>Calculating and plotting <abbr title="transcription start site enrichment">TSSe</abbr> scores</li>
      <li>Filtering cells and selecting features (fixed-size genomic bins)</li>
    </ol>
  </li>
  <li>Dimension reduction through <strong>Spectral embedding</strong> and <strong><abbr title="Uniform Manifold Approximation and Projection">UMAP</abbr></strong></li>
  <li>Clustering of cells via the <em>Leiden</em> method</li>
  <li>Cluster annotation
    <ol>
      <li>Producing and filtering a cell by gene activity matrix</li>
      <li>Data normalization and imputation with <strong>Scanpy</strong> and the <em>MAGIC</em> algorithm</li>
      <li>Visualizing marker genes in the clusters</li>
      <li>Manually annotating the cell types with selected marker genes</li>
    </ol>
  </li>
</ol>

<p><a href="/training-material/topics/single-cell/images/scatac-standard-snapatac2/snapatac2-pipeline.png" rel="noopener noreferrer"><img src="/training-material/topics/single-cell/images/scatac-standard-snapatac2/snapatac2-pipeline.png" alt="SnapATAC2 processing pipeline. " width="4028" height="2208" loading="lazy" /></a></p>

<p>The <abbr title="Single-cell Assay for Transposase-Accessible Chromatin using sequencing">scATAC-seq</abbr> analysis can now continue with downstream analysis, for example <em>differential peak analysis</em>.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-differential-peak-analysis"><button class="gtn-boxify-button details" type="button" aria-controls="details-differential-peak-analysis" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Differential peak analysis</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The <strong>SnapATAC2</strong> tools for differential peak analysis are already accessible on Galaxy. However, there are no GTN trainings available yet. Until such a tutorial is uploaded, you can visit the <strong>SnapATAC2</strong> documentation for a <a href="https://kzhang.org/SnapATAC2/version/2.6/tutorials/diff.html">tutorial on differential peak analysis</a>. And check out our <a href="https://usegalaxy.eu/u/timonschlegel/h/differential-peak-analysis-with-snapatac2">example history</a> and the exemplary <a href="https://usegalaxy.eu/u/timonschlegel/w/copy-of-differential-peak-analysissnapatac2">workflow</a>.</p>

  <p>The tools are available in Galaxy under <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_peaks_and_motif/snapatac2_peaks_and_motif/2.6.4+galaxy1" title="SnapATAC2 Peaks and Motif tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnapATAC2 Peaks and Motif</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6.4+galaxy1)</span>.</p>
  <ul>
    <li>If you want to continue with differential peak analysis, please make sure that the AnnData object with the annotated cell types contains unspecified annotations for the reference sequences (<code class="language-plaintext highlighter-rouge">uns: 'reference_sequences'</code>).
      <ul>
        <li>The section <a href="/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html#copy-over-embeddings">Copy-over embeddings</a> explains how to copy annotations from one AnnData object to another.</li>
      </ul>
    </li>
  </ul>

</blockquote>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Single-cell ATAC-seq can identify open chromatin sites</p>
</li>
                        
                        <li><p>Dimension reduction is required to simplify the data while preserving important information about the relationships of cells to each other.</p>
</li>
                        
                        <li><p>Clusters of similar cells can be annotated to their respective cell-types</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/single-cell/faqs/">FAQ page for the Single Cell topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Belkin2003">Belkin, M., and P. Niyogi, 2003 <b>Laplacian Eigenmaps for Dimensionality Reduction and Data Representation</b>. Neural Computation 15: 1373–1396. <a href="https://doi.org/10.1162/089976603321780317">10.1162/089976603321780317</a></li>
<li id="Kia2017">Kia, A., C. Gloeckner, T. Osothprarop, N. Gormley, E. Bomati <i>et al.</i>, 2017 <b>Improved genome sequencing using an engineered transposase</b>. BMC Biotechnology 17: <a href="https://doi.org/10.1186/s12896-016-0326-1">10.1186/s12896-016-0326-1</a></li>
<li id="McInnes2018">McInnes, L., J. Healy, and J. Melville, 2018 <b>UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction</b>. <a href="https://doi.org/10.48550/ARXIV.1802.03426">10.48550/ARXIV.1802.03426</a> <a href="https://arxiv.org/abs/1802.03426">https://arxiv.org/abs/1802.03426</a></li>
<li id="Wolf2018">Wolf, F. A., P. Angerer, and F. J. Theis, 2018 <b>SCANPY: large-scale single-cell gene expression data analysis</b>. Genome Biology 19: <a href="https://doi.org/10.1186/s13059-017-1382-0">10.1186/s13059-017-1382-0</a></li>
<li id="vanDijk2018">Dijk, D. van, R. Sharma, J. Nainys, K. Yim, P. Kathail <i>et al.</i>, 2018 <b>Recovering Gene Interactions from Single-Cell Data Using Data Diffusion</b>. Cell 174: 716–729.e27. <a href="https://doi.org/10.1016/j.cell.2018.05.061">10.1016/j.cell.2018.05.061</a></li>
<li id="Amemiya2019">Amemiya, H. M., A. Kundaje, and A. P. Boyle, 2019 <b>The ENCODE Blacklist: Identification of Problematic Regions of the Genome</b>. Scientific Reports 9: <a href="https://doi.org/10.1038/s41598-019-45839-z">10.1038/s41598-019-45839-z</a></li>
<li id="Franzn2019">Franzén, O., L.-M. Gan, and J. L. M. Björkegren, 2019 <b>PanglaoDB: a web server for exploration of mouse and human single-cell RNA sequencing data</b>. Database 2019: <a href="https://doi.org/10.1093/database/baz046">10.1093/database/baz046</a></li>
<li id="Sun2019">Sun, Z., L. Chen, H. Xin, Y. Jiang, Q. Huang <i>et al.</i>, 2019 <b>A Bayesian mixture model for clustering droplet-based single-cell transcriptomic data from population studies</b>. Nature Communications 10: <a href="https://doi.org/10.1038/s41467-019-09639-3">10.1038/s41467-019-09639-3</a></li>
<li id="Traag2019">Traag, V. A., L. Waltman, and N. J. van Eck, 2019 <b>From Louvain to Leiden: guaranteeing well-connected communities</b>. Scientific Reports 9: <a href="https://doi.org/10.1038/s41598-019-41695-z">10.1038/s41598-019-41695-z</a></li>
<li id="Wolock2019">Wolock, S. L., R. Lopez, and A. M. Klein, 2019 <b>Scrublet: Computational Identification of Cell Doublets in Single-Cell Transcriptomic Data</b>. Cell Systems 8: 281–291.e9. <a href="https://doi.org/10.1016/j.cels.2018.11.005">10.1016/j.cels.2018.11.005</a></li>
<li id="Yan2020">Yan, F., D. R. Powell, D. J. Curtis, and N. C. Wong, 2020 <b>From reads to insight: a hitchhiker’s guide to ATAC-seq data analysis</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-1929-3">10.1186/s13059-020-1929-3</a></li>
<li id="Zhang2024">Zhang, K., N. R. Zemke, E. J. Armand, and B. Ren, 2024 <b>A fast,  scalable and versatile tool for analysis of single-cell omics data</b>. Nature Methods 21: 217–227. <a href="https://doi.org/10.1038/s41592-023-02139-9">10.1038/s41592-023-02139-9</a></li></ol>
                

                
                <h1 data-toc-skip>Glossary</h1>
                <dl>
                    
                    
                    <dt>PBMCs</dt>
                    <dd>peripheral blood mononuclear cells</dd>
                    
                    <dt>QC</dt>
                    <dd>quality control</dd>
                    
                    <dt>TSS</dt>
                    <dd>transcription start sites</dd>
                    
                    <dt>TSSe</dt>
                    <dd>transcription start site enrichment</dd>
                    
                    <dt>UMAP</dt>
                    <dd>Uniform Manifold Approximation and Projection</dd>
                    
                    <dt>scATAC-seq</dt>
                    <dd>Single-cell Assay for Transposase-Accessible Chromatin using sequencing</dd>
                    
                </dl>
                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scatac-standard-processing-snapatac2"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scatac-standard-processing-snapatac2">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Timon Schlegel,  <b>Single-cell ATAC-seq standard processing with SnapATAC2 (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-scatac-standard-processing-snapatac2,
author = "Timon Schlegel",
	title = "Single-cell ATAC-seq standard processing with SnapATAC2 (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: replace_column_by_key_value_file
  owner: bgruening
  revisions: d533e4b75800
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bwa
  owner: devteam
  revisions: 2477830927ec
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: cut_columns
  owner: devteam
  revisions: cec635fab700
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fasta_compute_length
  owner: devteam
  revisions: 7d37cfda8e00
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: anndata_ops
  owner: ebi-gxa
  revisions: a4774b7b2e85
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: anndata_inspect
  owner: iuc
  revisions: 6f0d0c784f09
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: anndata_manipulate
  owner: iuc
  revisions: ed4996a16f7f
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: scanpy_filter
  owner: iuc
  revisions: d636ce5cde16
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: scanpy_inspect
  owner: iuc
  revisions: f54f0f0598ad
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: scanpy_normalize
  owner: iuc
  revisions: d844935c906c
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: scanpy_plot
  owner: iuc
  revisions: f7446fd8d5e7
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_clustering
  owner: iuc
  revisions: d3ea0ba3d066
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_peaks_and_motif
  owner: iuc
  revisions: 64fa083411f1
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_plotting
  owner: iuc
  revisions: 05bd4db20227
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snapatac2_preprocessing
  owner: iuc
  revisions: c5c94e01a1b5
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00445">gxy.io/GTN:T00445</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/7f6ee502828a7660a00ce43e77e864d7cbada116">7f6ee50</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>