<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Multisample Analysis</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="A collection of microtutorials explaining various feature..." />
        <meta property="og:title" content="Galaxy Training: Multisample Analysis" />
        <meta property="og:description" content="A collection of microtutorials explaining various feature..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/galaxy-interface" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Using Galaxy and Managing your Data
                        </a>
                        
                    </li>
                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/galaxy-interface/faqs/" title="Check our FAQs for the Using Galaxy and Managing your Data topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Translate this page
            </div>

            <div id="lang-selector">
                
                <strong>Automatic Translations</strong>

<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html&edit-text=&act=url">
                Fran√ßais
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html&edit-text=&act=url">
                Êó•Êú¨Ë™û
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html&edit-text=&act=url">
                Espa√±ol
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html&edit-text=&act=url">
                Portugu√™s
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html&edit-text=&act=url">
                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                </a>
                
                <a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html&edit-text=&act=url" title="">
                And more!
                </a>
            </div>
        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-10-01/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html" title="Version 2021-10-01">2021-10-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-09-01/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html" title="Version 2021-09-01">2021-09-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-08-01/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html" title="Version 2021-08-01">2021-08-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:40 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Multisample Analysis",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Using Galaxy and Managing your Data",
    "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
    "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/"
  },
  "courseCode": "galaxy-interface / processing-many-samples-at-once / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Multisample Analysis' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html",
  "timeRequired": "PT1H",
  "keywords": "collections",
  "description": "",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Anne Pajon"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Anne Pajon"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Using Galaxy and Managing your Data",
      "description": "A collection of microtutorials explaining various features of the Galaxy user interface and manipulating data within Galaxy.",
      "url": "https://training.galaxyproject.org//training-material/topics/galaxy-interface/"
    }
  ],
  "skillLevel": "Advanced"
}
</script>

<section class="tutorial topic-galaxy-interface">
    <h1 data-toc-skip>Multisample Analysis</h1>
    

    <div class="contributors-line">Authors: <a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge contributor-nekrut"><img src="https://avatars.githubusercontent.com/nekrut?s=27" alt="Avatar">Anton Nekrutenko</a><a href="/training-material/hall-of-fame/pajanne/" class="contributor-badge contributor-pajanne"><img src="https://avatars.githubusercontent.com/pajanne?s=27" alt="Avatar">Anne Pajon</a></div>

    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        </ul>

        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Advanced <span class="visually-hidden">Advanced</span>
<span class="level advanced" title="Advanced Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 15, 2020 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                

                <blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#processing-many-samples-at-once" id="markdown-toc-processing-many-samples-at-once">Processing many samples at once</a></li>
  <li><a href="#getting-data" id="markdown-toc-getting-data">Getting data</a></li>
  <li><a href="#creating-a-list-of-paired-datasets" id="markdown-toc-creating-a-list-of-paired-datasets">Creating a list of paired datasets</a></li>
  <li><a href="#using-collections" id="markdown-toc-using-collections">Using collections</a></li>
  <li><a href="#processing-collection-as-a-single-entity" id="markdown-toc-processing-collection-as-a-single-entity">Processing collection as a single entity</a>    <ol>
      <li><a href="#ensuring-consistency-of-bam-dataset" id="markdown-toc-ensuring-consistency-of-bam-dataset">Ensuring consistency of BAM dataset</a></li>
      <li><a href="#retaining-proper-pairs" id="markdown-toc-retaining-proper-pairs">Retaining ‚Äòproper pairs‚Äô</a></li>
      <li><a href="#merging-collection-into-a-single-dataset" id="markdown-toc-merging-collection-into-a-single-dataset">Merging collection into a single dataset</a></li>
    </ol>
  </li>
  <li><a href="#lets-look-at-what-weve-got" id="markdown-toc-lets-look-at-what-weve-got">Let‚Äôs look at what we‚Äôve got!</a></li>
  <li><a href="#we-did-not-fake-this" id="markdown-toc-we-did-not-fake-this">We did not fake this:</a></li>
  <li><a href="#if-things-dont-work" id="markdown-toc-if-things-dont-work">If things don‚Äôt work‚Ä¶</a></li>
</ol>

</blockquote>

<h1 id="processing-many-samples-at-once">Processing many samples at once</h1>

<p>Here we will show <span class="notranslate">Galaxy</span> features designed to help with the analysis of large numbers of samples. When you have just a few samples - clicking through them is easy. But once you‚Äôve got hundreds - it becomes very annoying. In <span class="notranslate">Galaxy</span> we have introduced <strong>Dataset collections</strong> that allow you to combine numerous datasets in a single entity that can be easily manipulated.</p>

<h4 id="in-this-tutorial-we-assume-the-following">In this tutorial we assume the following:</h4>

<ul>
  <li>you already have basic understanding of how <span class="notranslate">Galaxy</span> works (if you don‚Äôt see <a href="https://usegalaxy.org/galaxy101"><span class="notranslate">Galaxy</span> 101</a> tutorial;</li>
  <li>you have an account in <span class="notranslate">Galaxy</span> (<a href="https://github.com/nekrut/galaxy/wiki/Galaxy101-1#01-setting-up-galaxy-account">see this</a> if you don‚Äôt);</li>
  <li>you have your browser configured as described <a href="https://github.com/nekrut/galaxy/wiki/Galaxy101-1#00-getting-your-display-sorted-out">here</a>.</li>
</ul>

<p><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> <em>At this time this tutorial is using <span class="notranslate">Galaxy</span>‚Äôs test server at https://test.<span class="notranslate">galaxy</span>project.org. Once the main site is updated this tutorial will be edited.</em></p>

<h1 id="getting-data">Getting data</h1>
<p><a href="https://test.galaxyproject.org/u/anton/h/collections-1">Here</a> is a history containing a few datasets we will be practicing with (as always with <span class="notranslate">Galaxy</span> tutorial you can upload your own data and play with it instead of the provided datasets):</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">M117-bl_1</code> - family 117, mother, 1-st (<strong>F</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117-bl_2</code> - family 117, mother, 2-nd (<strong>R</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117-ch_1</code> - family 117, mother, 1-st (<strong>F</strong>) read from <strong>cheek</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117-ch_1</code> - family 117, mother, 2-nd (<strong>R</strong>) read from <strong>cheek</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-bl_1</code>- family 117, child, 1-st (<strong>F</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-bl_2</code>- family 117, child, 2-nd (<strong>R</strong>) read from <strong>blood</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-ch_1</code>- family 117, child, 1-st (<strong>F</strong>) read from <strong>cheek</strong></li>
  <li><code class="language-plaintext highlighter-rouge">M117C1-ch_2</code>- family 117, child, 2-nd (<strong>R</strong>) read from <strong>cheek</strong></li>
</ul>

<p>These datasets represent genomic <span class="notranslate">DNA</span> (enriched for mitochondria via a long range PCR) isolated from blood and cheek (buccal swab) of mother (<code class="language-plaintext highlighter-rouge">M117</code>) and her child (<code class="language-plaintext highlighter-rouge">M117C1</code>) that was sequenced on an Illumina miSeq machine as paired-read library (250-bp <span class="notranslate">reads</span>; see our <a href="http://www.pnas.org/content/111/43/15474.abstract">2014</a> manuscript for <strong>Methods</strong>).</p>

<h1 id="creating-a-list-of-paired-datasets">Creating a list of paired datasets</h1>

<p>If you imported <a href="https://test.galaxyproject.org/u/anton/h/collections-1">history</a> as described <a href="https://github.com/nekrut/galaxy/wiki/Processing-many-samples-at-once#0-getting-data">above</a>, your screen will look something like this:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/manyDatasets.png" alt="manyDatasets. " loading="lazy" /></p>

<p>Now click the checkbox in <img src="https://galaxyproject.org/tutorials/collections/historyItemControls.png" alt="HistioryItemControls. " loading="lazy" /> and you will see your history changing like this:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/historyWithCheckboxes.png" alt="historyWithCheckboxes. " loading="lazy" /></p>

<p>Let‚Äôs click <code class="language-plaintext highlighter-rouge">All</code>, which will select all datasets in the history, then click <img src="https://galaxyproject.org/tutorials/collections/forAllSelected.png" alt="allSelected. " loading="lazy" /> and finally select <strong>Build List of Dataset Pairs</strong> from the following menu:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/buildPairs.png" alt="buildPairs. " loading="lazy" /></p>

<p>The following wizard will appear:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/collectionCreation.png" alt="collectionCreation. " loading="lazy" /></p>

<p>In this case <span class="notranslate">Galaxy</span> automatically assigned pairs using the <code class="language-plaintext highlighter-rouge">_1</code> and <code class="language-plaintext highlighter-rouge">_2</code> endings of dataset names. Let‚Äôs however pretend that this did not happen. Click on <strong>Unpair all</strong> (highlighted in red in the figure above) link and then on <strong>Clear</strong> link (highlighted in blue in the figure above). The interface will change into its virgin state:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/collectionCreationClean.png" alt="collectionCreationClean. " loading="lazy" /></p>

<p>Hopefully you remember that we have paired-end data in this scenario. Datasets containing the first (forward) and the second (reverse) read are differentiated by having <code class="language-plaintext highlighter-rouge">_1</code> and <code class="language-plaintext highlighter-rouge">_2</code> in the filename. We can use this feature in dataset collection wizard to pair our datasets.  Type <code class="language-plaintext highlighter-rouge">_1</code> in the left <strong>Filter this list</strong> text box and <code class="language-plaintext highlighter-rouge">_2</code> in the right:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/1and2.png" alt="1and2. " loading="lazy" /></p>

<p>You will see that the dataset collection wizard will automatically filter lists on each side of the interface:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/collectionPrefiltered.png" alt="collectionPrefiltered. " loading="lazy" /></p>

<p>Now you can either click <strong>Auto pair</strong> if pairs look good to you (proper combinations of datasets are listed in each line) or pair each forward/reverse group individually by pressing <strong>Pair these datasets</strong> button separating each pair:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/collectionCreation.png" alt="collectionCreation. " loading="lazy" /></p>

<p>Now it is time to name the collection:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/collectionNaming.png" alt="collectionNaming. " loading="lazy" /></p>

<p>and create the collection by clicking <strong>Create list</strong>. A new item will appear in the history as you can see on the panel <strong>A</strong> below. Clicking on collection will expand it to show four pairs it contains (panel <strong>B</strong>). Clicking individual pairs will expand them further to reveal <strong>forward</strong> and <strong>reverse</strong> datasets (panel <strong>C</strong>). Expanding these further will enable one to see individual datasets (panel <strong>D</strong>).</p>

<p><img src="https://galaxyproject.org/tutorials/collections/collection_ABCD.png" alt="collection_ABCD. " loading="lazy" /></p>

<h1 id="using-collections">Using collections</h1>

<p>By now we see that a collection can be used to bundle a large number of items into a single history item. This means that many <span class="notranslate">Galaxy</span> tools will be able to process all datasets in a collection transparently to you. Let‚Äôs try to map these datasets to human genome using <code class="language-plaintext highlighter-rouge">bwa-mem</code> mapper:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/bwa_mem_collection_readGroups.png" alt="bwa_mem_collection_readGroups. " loading="lazy" /></p>

<p>Here is what you need to do:</p>

<ul>
  <li>set <strong>Using reference genome</strong> to <code class="language-plaintext highlighter-rouge">hg38</code> (red outline);</li>
  <li>set <strong>Single or Paired-end <span class="notranslate">reads</span></strong> to <code class="language-plaintext highlighter-rouge">Paired collection</code> (blue outline);</li>
  <li>select <code class="language-plaintext highlighter-rouge">M177-collection</code> from <strong>Select a paired collection</strong> dropdown (magenta outline);</li>
  <li>In <strong>Set read groups information</strong> select <code class="language-plaintext highlighter-rouge">Automatically assign ID</code> (green outline);</li>
  <li>scroll down and click <strong>Execute</strong>.</li>
</ul>

<p>You will see jobs being submitted and new datasets appearing in the history. IN particular below you can see that <span class="notranslate">Galaxy</span> has started four jobs (two yellow and two gray). This is because we have eight paired datasets with each pair being processed separately by <code class="language-plaintext highlighter-rouge">bwa-mem</code>. As a result we have four <code class="language-plaintext highlighter-rouge">bwa-mem</code> runs:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/bwa_memCollectionRunning.png" alt="bwa_memCollectionRunning. " loading="lazy" /></p>

<p>Once these jobs are finished they will disappear from the history and all results will be represented as a new collection:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/bwa_memCollectionDone.png" alt="bwa_memCollectionDone. " loading="lazy" /></p>

<p>Let‚Äôs look at this collection by clicking on it (panel <strong>A</strong> in the figure below). You can see that now this collection is no longer paired (compared to the collection we created in the beginning of this tutorial). This is because <code class="language-plaintext highlighter-rouge">bwa-mem</code> takes forward and reverse data as input, but produces only a single BAM dataset as the output. So what we have in the result is a <em>list</em> of four dataset (BAM files; panels <strong>B</strong> and <strong>C</strong>).</p>

<p><img src="https://galaxyproject.org/tutorials/collections/bwa_memCollection_ABC.png" alt="bwa_memCollection_ABC. " loading="lazy" /></p>

<h1 id="processing-collection-as-a-single-entity">Processing collection as a single entity</h1>

<p>Now that <code class="language-plaintext highlighter-rouge">bwa-mem</code> has finished and generated a collection of BAM datasets we can continue to analyze the entire collection as a single <span class="notranslate">Galaxy</span> ‚Äò<em>item</em>‚Äô.</p>

<h2 id="ensuring-consistency-of-bam-dataset">Ensuring consistency of BAM dataset</h2>

<p>Let‚Äôs perform cleanup of our BAM files with <code class="language-plaintext highlighter-rouge">cleanSam</code> utility from the <strong>Picard</strong> package:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/cleanSam.png" alt="cleanSam. " loading="lazy" /></p>

<p>If you look at the picture above carefully, you will see that the <strong>Select SAM/BAM dataset or dataset collection</strong> parameter is empty (it says <code class="language-plaintext highlighter-rouge">No sam or bam datasets available.</code>). This is because we do not have single SAM or BAM datasets in the history. Instead we have a collection. So all you need to do is to click on the <strong>folder</strong> (<img src="https://galaxyproject.org/tutorials/collections/folder.png" alt="folder. " loading="lazy" />) button and you will our BAM collection selected:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/cleanSam_closeup.png" alt="cleanSam_closeup. " loading="lazy" /></p>

<p>Click <strong>Execute</strong>. As an output this tool will produce a collection contained cleaned data.</p>

<h2 id="retaining-proper-pairs">Retaining ‚Äòproper pairs‚Äô</h2>

<p>Now let‚Äôs clean the dataset further by only preserving truly paired <span class="notranslate">reads</span> (<span class="notranslate">reads</span> satisfying two requirements: (1) read is paired, and (2) it is mapped as a proper pair). For this we will use <code class="language-plaintext highlighter-rouge">Filter SAM or BAM</code> tools from <strong>SAMTools</strong> collection:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/filter.png" alt="filter. " loading="lazy" /></p>

<p>parameters should be set as shown below. By setting <span class="notranslate">mapping</span> quality to <code class="language-plaintext highlighter-rouge">20</code> we avoid <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to multiple locations and by using <strong>Filter on bitwise flag</strong> option we ensure that the resulting dataset will contain only properly paired <span class="notranslate">reads</span>. This operation will produce yet another collection containing now filtered datasets.</p>

<p><img src="https://galaxyproject.org/tutorials/collections/filter_closeup.png" alt="filter_closeup. " loading="lazy" /></p>

<h2 id="merging-collection-into-a-single-dataset">Merging collection into a single dataset</h2>

<p>The beauty of BAM datasets is that they can be combined in a single entity using so called <em>Read group</em> (<a href="https://wiki.galaxyproject.org/Learn/GalaxyNGS101#Understanding_and_manipulating_SAM.2FBAM_datasets">learn more</a> about Read Groups on old wiki, which will be migrated here shortly). This allows to bundle <span class="notranslate">reads</span> from multiple experiments into a single dataset where read identity is maintained by labelling every sequence with <em>read group</em> tags. So let‚Äôs finally reduce this collection to a single BAM dataset. For this we will use <code class="language-plaintext highlighter-rouge">MergeSamFiles</code> tool for the <code class="language-plaintext highlighter-rouge">Picard</code> suite:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/merge.png" alt="merge. " loading="lazy" /></p>

<p>Here we select the collection generated by the filtering tool described above in <a href="https://github.com/nekrut/galaxy/wiki/Processing-many-samples-at-once#31-retaining-proper-pairs">3.1</a>:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/merge_closeup.png" alt="merge_closeup. " loading="lazy" /></p>

<p>This operation will <strong>not</strong> generate a collection. Instead, it will generate a single BAM dataset containing mapped <span class="notranslate">reads</span> from our four samples (<code class="language-plaintext highlighter-rouge">M117-bl</code>, <code class="language-plaintext highlighter-rouge">M117-ch</code>, <code class="language-plaintext highlighter-rouge">M117C1-bl</code>, and <code class="language-plaintext highlighter-rouge">M117C1-ch</code>).</p>

<h1 id="lets-look-at-what-weve-got">Let‚Äôs look at what we‚Äôve got!</h1>

<p>So we have one BAM dataset combining everything we‚Äôve done so far. Let‚Äôs look at the contents of this dataset using a genome browser. First, we will need to downsample the dataset to avoiding overwhelming the browser. For this we will use <code class="language-plaintext highlighter-rouge">Downsample SAM/BAM</code> tool:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/downsample.png" alt="downsample. " loading="lazy" /></p>

<p>Set <strong>Probability (between 0 and 1) that any given read will be kept</strong> to roughly <code class="language-plaintext highlighter-rouge">5%</code> (or <code class="language-plaintext highlighter-rouge">0.05</code>) using the slider control:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/downsample_closeup.png" alt="downsample_closeup. " loading="lazy" /></p>

<p>This will generate another BAM dataset containing only 5% of the original <span class="notranslate">reads</span> and much smaller as a result. Click on this dataset and you will see links to various genome browsers:</p>

<p><img src="https://galaxyproject.org/tutorials/collections/browserLinks.png" alt="browserLinks. " loading="lazy" /></p>

<p>Click the <strong>Human hg38</strong> link in the <strong>display with IGV</strong> line as highlighted above (<a href="https://wiki.galaxyproject.org/Learn/GalaxyNGS101#Visualizing_multiple_datasets_in_Integrated_Genome_Viewer_.28IGV.29">learn</a> more about displaying <span class="notranslate">Galaxy</span> data in IGV with this <a href="https://vimeo.com/123442619#t=4m16s">movie</a>). Below is an example generated with IGV on these data. In this screenshot <span class="notranslate">reads</span> are colored by read group (four distinct colors). A yellow inset displays additional information about a single read. One can see that this read corresponds to read group <code class="language-plaintext highlighter-rouge">M117-bl</code>.</p>

<p><img src="https://galaxyproject.org/tutorials/collections/igv.png" alt="igv. " loading="lazy" /></p>

<h1 id="we-did-not-fake-this">We did not fake this:</h1>
<p>The two histories and the <span class="notranslate">workflow</span> described in this page are accessible directly from this page below:</p>

<ul>
  <li>History <a href="https://test.galaxyproject.org/u/anton/h/collections-1"><strong>Collections</strong></a></li>
  <li>History <a href="https://test.galaxyproject.org/u/anton/h/collections-full-analysis"><strong>Collections (full analysis)</strong></a></li>
</ul>

<p>From there you can import histories to make them your own.</p>

<h1 id="if-things-dont-work">If things don‚Äôt work‚Ä¶</h1>
<p>‚Ä¶you need to complain. Use <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span>‚Äôs Help Channel</a> to do this.</p>


                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/galaxy-interface/faqs/">FAQ page for the Using Galaxy and Managing your Data topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Multisample Analysis (Using Galaxy and Managing your Data)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Anton Nekrutenko, Anne Pajon, 2020 <b>Multisample Analysis (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html">https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{galaxy-interface-processing-many-samples-at-once,
author = "Anton Nekrutenko and Anne Pajon",
title = "Multisample Analysis (Galaxy Training Materials)",
year = "2020",
month = "06",
day = "15"
url = "\url{https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
