<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>RAD-Seq Reference-based data analysis</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Learn to analyse Ecological data through Galaxy." />
        <meta property="og:title" content="Galaxy Training: RAD-Seq Reference-based data analysis" />
        <meta property="og:description" content="Learn to analyse Ecological data through Galaxy." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/ecology" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Ecology
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fecology%2Ftutorials%2Fref-based-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fecology%2Ftutorials%2Fref-based-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fecology%2Ftutorials%2Fref-based-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fecology%2Ftutorials%2Fref-based-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fecology%2Ftutorials%2Fref-based-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fecology%2Ftutorials%2Fref-based-rad-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/ecology/tutorials/ref-based-rad-seq/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "RAD-Seq Reference-based data analysis",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Ecology",
    "description": "Learn to analyse Ecological data through Galaxy.",
    "url": "https://training.galaxyproject.org//training-material/topics/ecology/"
  },
  "courseCode": "ecology / ref-based-rad-seq / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'RAD-Seq Reference-based data analysis' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html",
  "timeRequired": "PT8H",
  "keywords": "RAD-seq",
  "description": "The questions this  addresses are:\n - How to analyze RAD sequencing data using a reference genome for a population genomics study?\n\n\\nThe objectives are:\n - Analysis of RAD sequencing data using a reference genome\n - SNP calling from RAD sequencing data\n - Calculate population genomics statistics from RAD sequencing data\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Yvan Le Bras"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Yvan Le Bras"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Ecology",
      "description": "Learn to analyse Ecological data through Galaxy.",
      "url": "https://training.galaxyproject.org//training-material/topics/ecology/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0610",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0610",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0610"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>RAD-Seq Reference-based data analysis</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/yvanlebras/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/yvanlebras" alt="Yvan Le Bras">Yvan Le Bras</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to analyze RAD sequencing data using a reference genome for a population genomics study?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Analysis of RAD sequencing data using a reference genome</p>
</li>
            
            <li><p>SNP calling from RAD sequencing data</p>
</li>
            
            <li><p>Calculate population genomics statistics from RAD sequencing data</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 8 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1134547">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/workflows/" title="Workflows" alt="RAD-Seq Reference-based data analysis workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/ecology/tutorials/ref-based-rad-seq/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/ecology/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In the study of <a href="http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000862">Hohenlohe <em>et al.</em> 2010</a>, a genome scan of nucleotide diversity and differentiation in natural populations of threespine stickleback <em>Gasterosteus aculeatus</em> was conducted. Authors used Illumina-sequenced RAD tags to identify and type over 45,000 single nucleotide polymorphisms (SNPs) in each of 100 individuals from two oceanic and three freshwater populations.</p>

<p><img src="../../images/RAD4_Population_Genomics/Hohenlohe_et_al_2010.png" alt="The abstract of the paper on Population Genomics" /></p>

<p>We here proposed to re-analyze these data at least until the population genomics statistics calculation step using STACKS pipeline. <em>Gasterosteus aculeatus</em> draft genome will be used here as reference genome. In a reference-based RAD-seq data analysis, the <span class="notranslate">reads</span> are aligned (or mapped) against a reference genome to constrain our analysis to focus on already discovered loci. A de-novo approach can also be conducted (see <a href="/training-material/topics/ecology/tutorials/de-novo-rad-seq/tutorial.html">de_novo tutorial</a>, enhancing discoverability of new loci of interest but also of false positive one).</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#demultiplexing-reads" id="markdown-toc-demultiplexing-reads">Demultiplexing <span class="notranslate">reads</span></a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
    </ol>
  </li>
  <li><a href="#mapping" id="markdown-toc-mapping"><span class="notranslate">Mapping</span></a></li>
  <li><a href="#snp-calling-from-radtags" id="markdown-toc-snp-calling-from-radtags">SNP calling from radtags</a></li>
  <li><a href="#calculate-population-genomics-statistics" id="markdown-toc-calculate-population-genomics-statistics">Calculate population genomics statistics</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>The original data is available at NCBI SRA ENA under accession number <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?run=SRR034310">SRR034310</a> as part of the NCBI SRA ENA study accession number <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP001747">SRP001747</a>.</p>

<p>We will look at the first run SRR034310 out of seven which includes 16 samples from 2 populations, 8 from Bear Paw (freshwater) and 8 from Rabbit Slough (oceanic). We will download the <span class="notranslate">reads</span> directly from SRA and the remaining data (i.e reference genome, population map file, and barcodes file) from <a href="https://zenodo.org/record/1134547#.WkugUjfjJXI">Zenodo</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this RAD-seq exercise. If you are not inspired, you can name it ‚ÄúSTACKS RAD: population genomics with reference genome‚Äù for example‚Ä¶
      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li><strong>EBI SRA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> import files from SRA:
      <ul>
        <li>Search for <code class="language-plaintext highlighter-rouge">SRR034310</code></li>
        <li>Under results for <em>Run</em>, click on <em>view all 1 result</em></li>
        <li>In the next menu, click on the result <code class="language-plaintext highlighter-rouge">SRR034310</code></li>
        <li>In the table at the bottom, click the link in the column <strong>FASTQ files (<span class="notranslate">Galaxy</span>)</strong></li>
      </ul>

      <p>This will redirect to the <span class="notranslate">Galaxy</span> website and start the download.</p>
    </li>
    <li>Upload remaining training data from Zenodo:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1134547/files/Barcode_SRR034310.txt
https://zenodo.org/record/1134547/files/Details_Barcode_Population_SRR034310.txt
https://zenodo.org/record/1134547/files/Reference_genome_11_chromosomes.fasta
</code></pre></div>      </div>
      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>
    </li>
    <li>
      <p>Make sure the fastq.gz files you imported are of datatype <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>

      <p>Next, we need to tell <span class="notranslate">Galaxy</span> about the reference genome associated with this.</p>
    </li>
    <li>
      <p>We will define a custom reference build** for this tutorial:</p>

      <blockquote class="tip">

        <h3 id="tip-tip-adding-a-custom-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Adding a custom Database/Build (dbkey)</h3>
        <ul>
          <li>In the top menu bar, go to the <em>User</em>, and select <em>Custom Builds</em></li>
          <li>Choose a name for your reference build <code class="language-plaintext highlighter-rouge">Stickleback</code></li>
          <li>Choose a dbkey for your reference build <code class="language-plaintext highlighter-rouge">stickleback</code></li>
          <li>Under <strong>Definition</strong>, select the option <code class="language-plaintext highlighter-rouge">FASTA-file from history</code></li>
          <li>Under <strong>FASTA-file</strong>, select your fasta file <code class="language-plaintext highlighter-rouge">Reference_genome_11_chromosomes.fasta</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Next, we set the <code class="language-plaintext highlighter-rouge">dbkey</code> of our imported fastq files to our custom build key</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing Database/Build (dbkey)</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Database/Build</strong> field</li>
          <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">stickleback</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li><strong>Rename datasets</strong> according to the sample names</li>
  </ol>

</blockquote>

<p>The sequences are raw sequences from the sequencing machine, without any pretreatments. They need to be demultiplexed. To do so, we can use the Process Radtags tool from STACKS.</p>

<h2 id="demultiplexing-reads">Demultiplexing <span class="notranslate">reads</span></h2>

<p>For demultiplexing, we use the Process Radtags tool from <a href="https://www.g3journal.org/content/1/3/171.full">STACKS</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-demultiplexing-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Demultiplexing <span class="notranslate">reads</span></h3>

  <ol>
    <li><strong>Process Radtags</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúSingle-end or paired-end <span class="notranslate">reads</span> files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end files</code>
          <ul>
            <li><em>‚ÄúSingles-end <span class="notranslate">reads</span> infile(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">SRR034310.fastq(.gz)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúBarcode file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Barcodes_SRR034310.tabular</code></li>
        <li><em>‚ÄúNumber of enzymes‚Äù</em>: <code class="language-plaintext highlighter-rouge">One</code></li>
        <li><em>‚ÄúEnzyme‚Äù</em>: <code class="language-plaintext highlighter-rouge">sbfI</code></li>
        <li><em>‚ÄúCapture discarded <span class="notranslate">reads</span> to a file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">fastq</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many <span class="notranslate">reads</span> where on the original dataset?</li>
          <li>How many are kept?</li>
          <li>Can you try to explain the reason why we loose a lot of <span class="notranslate">reads</span> here?</li>
          <li>What kind of information does this result give concerning the upcoming data analysis and the barcodes design in general?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>The informations can be found in the results log file:</p>

          <ol>
            <li>8895289 total <span class="notranslate">reads</span></li>
            <li>8139531 retained <span class="notranslate">reads</span></li>
            <li>There are no sequences filtered because of low quality. This is because radtags didn‚Äôt apply quality related filtering since the corresponding advanced option (Discard <span class="notranslate">reads</span> with low quality scores) has not been enabled. So here, all not retained sequences are removed because of an ambiguous barcode (626265) or an ambiguous RAD-Tag (129493). This means that some barcodes are not exactly what was specified on the barcode file and that sometimes, no SbfI restriction enzyme site was found. This can be due to some sequencing problems but here, this is also due to the addition, in the original sequencing library, of RAD-seq samples from another study. This strategy is often used to avoid having too much sequences beginning with the exact same nucleotide sequence which may cause Illumina related issues during sequencing and cluster analysis</li>
            <li>Sequencing quality is essential! Each time your sequencing quality decreases, you loose data and thus essential biological information!</li>
          </ol>

          <p>In addition to the overall statistics the numbers of retained and removed <span class="notranslate">reads</span> are also given for each barcode sequence.</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>Process Radtags</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Re-Run playing with parameters
      <ul>
        <li>Section <strong>Advanced</strong>
          <ul>
            <li><em>‚ÄúDiscard <span class="notranslate">reads</span> with low quality scores‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚Äúscore limit‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code> (for example; play with this)</li>
            <li><em>‚ÄúSet the size of the sliding window as a fraction of the read length, between 0 and 1‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.30</code> (for example; play with this)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <p>This sliding window parameter allows notably the user to deal with the declining quality at the 3‚Äô end of <span class="notranslate">reads</span>.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on</h3>

  <p>You can use the <code class="language-plaintext highlighter-rouge">Charts</code> functionality through the Visualize button reachable on the <code class="language-plaintext highlighter-rouge">Radtags logs</code> file you just generated.</p>

  <p>If like me you don‚Äôt have payed attention to the organization of you file for the graphical representation you obtain a non optimal bars diagram with a not intelligent X-axis ordering. There is a lot of different manner to fix this. You can use a copy/paste ‚Äúbidouille‚Äù or you can use <span class="notranslate">Galaxy</span> tools to manipulate the <code class="language-plaintext highlighter-rouge">radtags logs</code> file to generate a better graph. For example, you can use <code class="language-plaintext highlighter-rouge">Select lines that match an <span class="notranslate">expression</span></code> tool to select rows then use the <code class="language-plaintext highlighter-rouge">Concatenate datasets tail-to-head</code> tool to reorganize these lines in a new file.
Then we generate a graphical display of the changes:</p>

  <p>First we cut the interesting lines of each <code class="language-plaintext highlighter-rouge">result.log with Stacks: process radtags</code></p>

  <ol>
    <li><strong>Select lines that match an <span class="notranslate">expression</span></strong> applying <code class="language-plaintext highlighter-rouge">^R1.fq.gz</code> on the log files and then</li>
    <li><strong>Replace Text in entire line</strong> on the resulting data sets finding <code class="language-plaintext highlighter-rouge">R1.fq.gz</code> and replacing with <code class="language-plaintext highlighter-rouge">NoScoreLimit</code> or <code class="language-plaintext highlighter-rouge">Score10</code> or <code class="language-plaintext highlighter-rouge">Score20</code> depending of the case</li>
    <li><strong>Select lines that match an <span class="notranslate">expression</span></strong> applying <code class="language-plaintext highlighter-rouge">File\tRetained <span class="notranslate">Reads</span></code> on one of the log file to obtain you futur header</li>
    <li><strong>Replace Text in entire line</strong> on the resulting data set finding <code class="language-plaintext highlighter-rouge">File</code> and replacing with <code class="language-plaintext highlighter-rouge">#</code> to have a best display</li>
    <li><strong>Concatenate datasets tail-to-head</strong> on the resulting data sets stating from the header you just created</li>
    <li>Finally <strong>Convert delimiters to TAB</strong> on the resulting data set converting all Tabs to be sure having a well formatted tabular file at the end</li>
  </ol>

  <p>Alte<span class="notranslate">rna</span>tively just copy/paste these lines on the <span class="notranslate">Galaxy</span> upload tool using Paste/fetch data section and modifying the File header by sample and filename by Score 10 / Score 20 and Lowquality for example‚Ä¶ Before Starting the upload, you can select the <code class="language-plaintext highlighter-rouge">Convert spaces to tabs</code> option through the <code class="language-plaintext highlighter-rouge">Upload configuration</code> wheel. If you did not pay attention to the order you can just sort the file using the first column.</p>
</blockquote>

<p>And you obtain a file like this one, ready to generate a beautiful and smart bar stacked!</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#		Retained <span class="notranslate">Reads</span>	Low Quality	Ambiguous Barcodes	Ambiguous RAD-Tag	Total
NoScoreLimit	8139531		0		626265			129493			8895289
Score10		7373160		766371		626265			129493			8895289
Score20		2980543		5158988		626265			129493			8895289
</code></pre></div></div>

<p>You can use the <code class="language-plaintext highlighter-rouge">Charts</code> functionality through the Visualize button to plot the data.</p>

<p><img src="../../images/RAD4_Population_Genomics/Process_radtags_charts_end.PNG" alt="The chart on the sorted file" width="75%" /></p>

<p>You can further test using a filter like <code class="language-plaintext highlighter-rouge">clean data, remove any read with an uncalled base</code> and see that here, this has only little impact on the number of retained <span class="notranslate">reads</span>.</p>

<p>The demultiplexed sequences are raw sequences from the sequencing machine, without any pretreatments. They need to be controlled for their quality.</p>

<p>We propose to continue the tutorial using the dataset collection containing the demultiplexed <span class="notranslate">reads</span> obtained with Process Radtag execution made with a quality score of 10 and with the <code class="language-plaintext highlighter-rouge">Discard <span class="notranslate">reads</span> with low quality scores</code> parameter set to Yes (so containing 7373160 retained <span class="notranslate">reads</span>).</p>

<h2 id="quality-control">Quality control</h2>

<p>For quality control, we use similar tools as described in <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">NGS-QC tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to check the quality of the <span class="notranslate">reads</span>:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúShort read data from your current history‚Äù</em>: the fastq files (collection)</li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the read length?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>The read length is 32 bp</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
        <li><em>‚ÄúType of FastQC output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">raw data</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQC output‚Äù</em>: the <code class="language-plaintext highlighter-rouge">raw data</code> output of FastQC (collection)</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/RAD4_Population_Genomics/multiqc.PNG" alt="MultiQC output" /></p>

<p>Note the quality drop at bases 5-10 which are the cut site of the RADSeq
protocol (TGCAGG). This is caused by the extremely uneven distribution the
sequencer observes TGCAGG in lockstep on all spots/clusters, i.e. all clusters
light up at the same time. Therefore the sequencer can not be sure that each
spot/cluster is correct which is expressed by the lower quality score. Usually, to reduce this problem,
PhiX is added, OR, as it is the case here, samples with heterogeneous barcodes lengths are mixed.
Still, approximately ~99% of the cases match nucleotide expected for the cut
site sequence.</p>

<p>As there is a draft genome for <em>Gasterosteus aculeatus</em>, we can use this information and map the sequences on this genome to identify polymorphism.</p>

<h1 id="mapping"><span class="notranslate">Mapping</span></h1>

<p>To make sense of the <span class="notranslate">reads</span>, their positions within <em>Gasterosteus aculeatus</em> genome must be determined. This process is known as aligning or ‚Äò<span class="notranslate">mapping</span>‚Äô the <span class="notranslate">reads</span> to the reference genome.</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Do you want to learn more about the principles behind <span class="notranslate">mapping</span>? Follow our <a href="/training-material/topics/sequence-analysis/">training</a></p>
</blockquote>

<p>Here we will use BWA. BWA is a fast light-weighted tool that aligns relatively short sequences (queries) to a sequence database (large), such as the human reference genome. It is developed by Heng Li at the Sanger Institute.</p>

<p><em><a href="https://www.ncbi.nlm.nih.gov/pubmed/19451168">Li et Durbin, Bioinformatics, 2009</a></em></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-map-with-bwa"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Map with BWA</h3>

  <ol>
    <li><strong>Map with BWA - map short <span class="notranslate">reads</span> (&lt; 100 bp) against reference genome</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code></li>
        <li><em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Reference_genome_11_chromosomes.fasta</code></li>
        <li><em>‚ÄúSelect input type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single Fastq</code></li>
        <li><em>‚ÄúSelect fastq dataset‚Äù</em>: One dataset collection containing the demultiplexed <span class="notranslate">reads</span> obtained with Process Radtag execution made with a quality score of 10 and with the <code class="language-plaintext highlighter-rouge">Discard <span class="notranslate">reads</span> with low quality scores</code> parameter set to Yes (so containing 7373160 retained <span class="notranslate">reads</span>).</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><strong>BWA</strong> generates BAM files with the mapped <span class="notranslate">reads</span>.</p>

<p>We next want to run Stacks on the freshwater and oceanic populations.</p>

<h1 id="snp-calling-from-radtags">SNP calling from radtags</h1>

<p>Run <code class="language-plaintext highlighter-rouge">Stacks: Reference map</code> <span class="notranslate">Galaxy</span> tool. This program will run pstacks, cstacks, and sstacks using a perl scrip (ref_map.pl) on the members of the population, accounting for the alignments of each read.</p>

<blockquote class="comment">
  <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Information on ref_map.pl and its parameters can be found online: https://creskolab.uoregon.edu/stacks/comp/ref_map.php.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-stacks-reference-map"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Stacks: Reference map</h3>

  <ol>
    <li>
      <p>Create a new file with our population map and name it <code class="language-plaintext highlighter-rouge">Population_map.txt</code>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>sample_CCCC	1
sample_CCAA	1
sample_CCTT	1
sample_CCGG	1
sample_CACA	1
sample_CAAC	1
sample_CATG	1
sample_CAGT	1
sample_CTCT	2
sample_CTAG	2
sample_CTTC	2
sample_CTGA	2
sample_GGGG	2
sample_GGAA	2
sample_GGTT	2
sample_GGCC	2
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new file</h3>

        <ul>
          <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the file contents into the text field</p>
          </li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p><strong>Stacks: Reference map</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:</p>
      <ul>
        <li><em>‚ÄúSelect your usage‚Äù</em>: <code class="language-plaintext highlighter-rouge">Population</code></li>
        <li><em>‚ÄúFiles containing an individual sample from a population‚Äù</em>: Choose the mapped <span class="notranslate">reads</span> (data collection)</li>
        <li><em>‚Äúpecify a population map‚Äù</em>: <code class="language-plaintext highlighter-rouge">Population_map.txt</code> created above</li>
        <li><em>‚ÄúMinimum depth of coverage‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>If you are using a file presenting population information and individual name in a different manner than expected by STACKS, you can use <span class="notranslate">Galaxy</span> tools like <code class="language-plaintext highlighter-rouge">Replace Text</code> (for example to replace <code class="language-plaintext highlighter-rouge">Rabbit Slough</code> by a population number like <code class="language-plaintext highlighter-rouge">2</code>, <code class="language-plaintext highlighter-rouge">Add column</code> (for example to add <code class="language-plaintext highlighter-rouge">sample_</code>) or <code class="language-plaintext highlighter-rouge">Cut columns from a table</code> (to put the new <code class="language-plaintext highlighter-rouge">sample_</code> column af the first place) and finally <code class="language-plaintext highlighter-rouge">Regex replace</code> (replacing <code class="language-plaintext highlighter-rouge">(sample_)\t</code> by <code class="language-plaintext highlighter-rouge">\1</code>) to generate it‚Ä¶</p>
      </blockquote>
    </li>
  </ol>

  <p>Once Stacks has completed running, investigate the output files: <code class="language-plaintext highlighter-rouge">ref_map.log</code>, <code class="language-plaintext highlighter-rouge">Summary from Stacks: reference map on data...</code> and <code class="language-plaintext highlighter-rouge">Catalog (*)</code> (snps, alleles and tags).
Notice that each locus now has a chromosome/base pair specified in each of the <code class="language-plaintext highlighter-rouge">*tags.tsv</code> files and in the catalog files.</p>

</blockquote>

<h1 id="calculate-population-genomics-statistics">Calculate population genomics statistics</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-calculate-population-genomics-statistics"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Calculate population genomics statistics</h3>

  <ol>
    <li><strong>Stacks: populations</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúInput type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Stacks output</code></li>
        <li><em>‚ÄúOutput from previous Stacks pipeline steps (e.g. denovo_map or refmap)‚Äù</em>: Full output from ref_map</li>
        <li><em>‚ÄúSpecify a population map‚Äù</em>: <code class="language-plaintext highlighter-rouge">Population_map.txt</code></li>
        <li>Section <strong>Data filtering options</strong>
          <ul>
            <li><em>‚ÄúMinimum percentage of individuals in a population required to process a locus for that population‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.75</code></li>
          </ul>
        </li>
        <li>Section <strong>Output options</strong> (VCF and Structure) and enabling SNP and haplotype-based F statistics calculation.
          <ul>
            <li><em>‚ÄúOutput results in Variant Call Format (VCF)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúOutput results in Structure Format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>‚ÄúEnable SNP and haplotype-based F statistics‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Now look at the output in the file <code class="language-plaintext highlighter-rouge">batch_1.sumstats</code> named <code class="language-plaintext highlighter-rouge">SNP and Haplotype-based F statistics with Stacks: populations ...</code> on your history. This file is also reachable on the data collection nammed <code class="language-plaintext highlighter-rouge">Full output from ref_map .....</code> with his original name <code class="language-plaintext highlighter-rouge">batch_1.sumstats</code>. There are a large number of statistics calculated at each SNP, so use <span class="notranslate">Galaxy</span> tools like filter, cut, and sort to focus on some.</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

    <ol>
      <li>What is the maximum value of FST‚Äô at any SNP? Don‚Äôt hesitate to look at the <a href="http://catchenlab.life.illinois.edu/stacks/manual/#files">STACKS manual</a> to find where this parameter is</li>
      <li>What is the meaning of this FST‚Äô value compared a a classical FST one? Once again, the <a href="http://catchenlab.life.illinois.edu/stacks/manual/#files">STACKS manual</a> can help you ;)</li>
      <li>How many SNPs reach this FST‚Äô value?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>1</li>
        <li>FST‚Äô is a haplotype measure of FST that is scaled to the theoretical maximum FST value at this locus. Depending on how many haplotypes there are, it may not be possible to reach an FST of 1, so this method will scale the value to 1</li>
        <li>78</li>
      </ol>
    </blockquote>
  </blockquote>
  <p>You can now for example filter this dataset to only keep FST‚Äô=1 loci for further analysis‚Ä¶</p>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed real RAD sequencing data to extract useful information, such as which loci are candidate regarding the genetic differentiation between freshwater and oceanic Stickelback populations. To answer these questions, we analyzed RAD sequence datasets using a reference-based RAD-seq data analysis approach. This approach can be sum up with the following scheme:</p>

<p><img src="../../images/ref_based_workflow.PNG" alt="Ref based tutorial workflow" /></p>


                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/ecology#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=RAD-Seq Reference-based data analysis (Ecology)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Yvan Le Bras, 2021 <b>RAD-Seq Reference-based data analysis (Galaxy Training Materials)</b>. <a href="/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html">/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{ecology-ref-based-rad-seq,
    author = "Yvan Le Bras",
    title = "RAD-Seq Reference-based data analysis (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/ecology/tutorials/ref-based-rad-seq/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Yvan Le Bras)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/ecology/tutorials/ref-based-rad-seq/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
