<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Importing files from public atlases / Importing files from public atlases / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@55607e95d8efdacc58f4eb03501f1419ce408b25",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Importing files from public atlases">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-09-13 15:56:15 +0000">
<meta name="DC.creator" content="Julia Jakiela">
<meta name="DC.creator" content="Wendi Bacon"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Importing files from public atlases / Hands-on: Importing files from public atlases">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/EBI-retrieval/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Importing files from public atlases",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-09-13 15:56:15 +0000",
  "datePublished": "2023-11-14 17:16:58 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [
    {
      "@context": "https://schema.org",
      "@type": "Grant",
      "identifier": "MR/V038966/1",
      "url": "https://gtr.ukri.org/projects?ref=MR/V038966/1",
      "funder": {
        "@context": "https://schema.org",
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
          "@type": "CreativeWork"
        },
        "name": "ELIXIR-UK: FAIR Data Stewardship training",
        "description": "This work has been funded by the ELIXIR-UK FAIR Data Stewardship training UKRI award (MR/V038966/1)\n",
        "url": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-fair-data/",
        "logo": "https://github.com/elixir-fair-data.png"
      }
    }
  ],
  "identifier": "https://gxy.io/GTN:T00374",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!\n\nWhat else do you want to see? You can submit tool and/or tutorial request on our <i class=\"fa fa-wrench\" aria-hidden=\"true\"></i> [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).\n {: .alert.alert-success}\n",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "abstract": "# Introduction",
  "learningResourceType": "e-learning",
  "name": "Importing files from public atlases",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html",
  "version": 5,
  "timeRequired": "PT15M",
  "teaches": "- You will retrieve raw data from the EBI Single Cell Expression Atlas and Human Cell Atlas.\n- You will manipulate the metadata and matrix files.\n- You will combine the metadata and matrix files into an AnnData or Seurat object for downstream analysis.",
  "keywords": [
    "Single Cell",
    "data import",
    "data management"
  ],
  "description": "## Abstract\n\n# Introduction\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - How do I use the EBI Single Cell Expression Atlas and Human Cell Atlas?\n - How can I reformat and manipulate the downloaded files to create the correct input for downstream analysis?\n\n\n## Learning Objectives\n\n- You will retrieve raw data from the EBI Single Cell Expression Atlas and Human Cell Atlas.\n- You will manipulate the metadata and matrix files.\n- You will combine the metadata and matrix files into an AnnData or Seurat object for downstream analysis.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html",
      "name": "Generating a single cell matrix using Alevin",
      "description": "Hands-on for 'Generating a single cell matrix using Alevin' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html",
      "name": "Combining single cell datasets after pre-processing",
      "description": "Hands-on for 'Combining single cell datasets after pre-processing' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "name": "Julia Jakiela",
      "image": "https://avatars.githubusercontent.com/wee-snufkin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0001-2017-8805",
      "orcid": "https://orcid.org/0009-0001-2017-8805"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/nomadscientist/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/nomadscientist/",
      "name": "Wendi Bacon",
      "image": "https://avatars.githubusercontent.com/nomadscientist",
      "description": "Senior Lecturer at The Open University",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-8170-8806",
      "orcid": "https://orcid.org/0000-0002-8170-8806"
    }
  ],
  "contributor": [

  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!\n\nWhat else do you want to see? You can submit tool and/or tutorial request on our <i class=\"fa fa-wrench\" aria-hidden=\"true\"></i> [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).\n {: .alert.alert-success}\n",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/EBI-retrieval/workflows/",
      "name": "Associated Workflows"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/EBI-retrieval/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Importing files from public atlases</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/wee-snufkin/" class="contributor-badge contributor-wee-snufkin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wee-snufkin?s=36" alt="Julia Jakiela avatar" width="36" class="avatar" />
    Julia Jakiela</a><a href="/training-material/hall-of-fame/nomadscientist/" class="contributor-badge contributor-nomadscientist"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/nomadscientist?s=36" alt="Wendi Bacon avatar" width="36" class="avatar" />
    Wendi Bacon</a>
		</td>
	</tr>
	

	

	

	

	

	

	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How do I use the EBI Single Cell Expression Atlas and Human Cell Atlas?</p>
</li>
        
        <li><p>How can I reformat and manipulate the downloaded files to create the correct input for downstream analysis?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>You will retrieve raw data from the EBI Single Cell Expression Atlas and Human Cell Atlas.</p>
</li>
        
        <li><p>You will manipulate the metadata and matrix files.</p>
</li>
        
        <li><p>You will combine the metadata and matrix files into an AnnData or Seurat object for downstream analysis.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Generating a single cell matrix using Alevin</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin-combine-datasets/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Combining single cell datasets after pre-processing</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 15 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/single-cell/tutorials/EBI-retrieval/workflows/" title="Importing files from public atlases workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            
                <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Input Histories">
        <i class="fas fa-sign-in-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-input</span>&nbsp;Input Histories
    </a>
    <ul class="dropdown-menu">
      
        <li>
            <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/u/j.jakiela/h/input-history-ebi-scxa---anndata-scanpy-or-seurat-object">
              UseGalaxy.eu
              
            </a>
        </li>
      
      <li class="dropdown-divider"></li>
      <li>
        <a class="dropdown-item" href="/training-material/faqs/gtn/gtn_example_histories.html"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> How to Use This</a>
      </li>
    </ul>

</li>
            

            
                <li class="btn btn-default supporting_material">
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Answer Histories">
        <i class="fas fa-sign-out-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-answer</span>&nbsp;Answer Histories
    </a>
    <ul class="dropdown-menu">
      
        <li>
            <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/u/j.jakiela/h/ebi-scxa---anndata-scanpy-or-seurat-object-1">
              EBI SCXA retrieval tool (UseGalaxy.eu)
              
              <br/><span class="text-muted">2024-01-10</span>
              
            </a>
        </li>
      
        <li>
            <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/u/j.jakiela/h/hca-downloader---example-files-1">
              HCA Downloader (UseGalaxy.eu)
              
              <br/><span class="text-muted">2024-09-08</span>
              
            </a>
        </li>
      
      <li class="dropdown-divider"></li>
      <li>
        <a class="dropdown-item" href="/training-material/faqs/gtn/gtn_example_histories.html"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> How to Use This</a>
      </li>
    </ul>

</li>
            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











<li class="btn btn-default supporting_material">


  <!-- dropdown with all recordings -->
  <a href="/training-material/topics/single-cell/tutorials/EBI-retrieval/recordings/" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this material in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
  </a>


  <ul class="dropdown-menu">
    

    
      
      
      
    <li><a class="dropdown-item" href="/training-material/topics/single-cell/tutorials/EBI-retrieval/recordings/index.html#tutorial-recording-12-september-2024" title="View the recording for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (September 2024) - 18m54s</a>
    </li>
      
    
    <li><a class="dropdown-item" href="/training-material/topics/single-cell/tutorials/EBI-retrieval/recordings/" title="View all recordings for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> View All</a>
    </li>

  </ul>

  
</li>
  





            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    

    <!--
    
    -->
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Nov 14, 2023 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Sep 13, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00374">https://gxy.io/GTN:T00374</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 5 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>Public single cell datasets seem to accumulate by the second. Well annotated, quality datasets are slightly trickier to find, which is why projects like the <a href="https://www.ebi.ac.uk/gxa/sc/home">Single Cell Expression Atlas</a> (SCXA) exist - to curate datasets for public use. Here, we will guide you through transforming data imported from the SCXA repository into the input file required for the <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">Filter, Plot, Explore tutorial</a> and we will also show how to use the public atlases for your own research.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#getting-data-from-the-single-cell-expression-atlas" id="markdown-toc-getting-data-from-the-single-cell-expression-atlas">Getting data from the Single Cell Expression Atlas</a>    <ol>
      <li><a href="#examine-the-imported-files" id="markdown-toc-examine-the-imported-files">Examine the imported files</a></li>
      <li><a href="#metadata-manipulation" id="markdown-toc-metadata-manipulation">Metadata manipulation</a></li>
      <li><a href="#check-mitochondrial-gene-name-format" id="markdown-toc-check-mitochondrial-gene-name-format">Check mitochondrial gene name format</a></li>
      <li><a href="#creating-the-anndata-object" id="markdown-toc-creating-the-anndata-object">Creating the AnnData object</a></li>
      <li><a href="#anndata-manipulation" id="markdown-toc-anndata-manipulation">AnnData manipulation</a></li>
      <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
      <li><a href="#creating-the-seurat-object" id="markdown-toc-creating-the-seurat-object">Creating the Seurat Object</a></li>
      <li><a href="#conclusion-1" id="markdown-toc-conclusion-1">Conclusion</a></li>
    </ol>
  </li>
  <li><a href="#human-cell-atlas-matrix-downloader" id="markdown-toc-human-cell-atlas-matrix-downloader">Human Cell Atlas Matrix Downloader</a></li>
</ol>

</blockquote>

<h1 id="getting-data-from-the-single-cell-expression-atlas">Getting data from the Single Cell Expression Atlas</h1>

<p>Galaxy has a specific tool for importing data from the SCXA (<span class="citation"><a href="#Moreno2020.04.08.032698">Moreno <i>et al.</i> 2020</a></span>), which combines all the preprocessing steps shown in <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">the corresponding tutorial</a> into one! For this tutorial, the dataset can be seen <a href="https://www.ebi.ac.uk/gxa/sc/experiments/E-MTAB-6945/downloads">at the EBI</a> with experiment ID of <code class="language-plaintext highlighter-rouge">E-MTAB-6945</code>.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-how-to-access-other-experiments"><button class="gtn-boxify-button details" type="button" aria-controls="details-how-to-access-other-experiments" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  How to access other experiments? </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>You can search datasets according to various criteria either using search box in <strong>Home</strong> tab or choosing kingdom, experiment collection, technology type (and others) in <strong>Browse experiments</strong> tab. When you find the experiment you are interested in, just click on it and the experiment ID will be displayed in the website URL, as shown below.</p>

  <figure id="figure-1" style="max-width: 90%;"><img src="../../images/scrna-data/exp_id.jpg" alt="Arrow pointing to the website URL where you can find experiment ID." width="1757" height="627" loading="lazy" /><a target="_blank" href="../../images/scrna-data/exp_id.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Where to find experiment ID on the EBI Single Cell Expression Atlas website.</figcaption></figure>

  <p>Once you know the experiment ID, you can use EBI SCXA Data Retrieval tool in Galaxy!</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-retrieving-data-from-single-cell-expression-atlas"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Retrieving data from Single Cell Expression Atlas</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/retrieve_scxa/retrieve_scxa/v0.0.2+galaxy2" title="EBI SCXA Data Retrieval tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>EBI SCXA Data Retrieval</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version v0.0.2+galaxy2)</span> with the following parameters:
      <ul>
        <li><em>“SC-Atlas experiment accession”</em>: <code class="language-plaintext highlighter-rouge">E-MTAB-6945</code></li>
        <li><em>“Choose the type of matrix to download”</em>: <code class="language-plaintext highlighter-rouge">Raw filtered counts</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>It’s important to note that this matrix is processed somewhat through the SCXA pipeline, which is quite similar to the pre-processing that has been shown in this case study tutorial series. The resultant datasets contain any and all metadata provided by the SCXA pipeline as well as the metadata contributed by the original authors (for instance, more cell or gene annotations). So while the AnnData object generated at the end of this tutorial will be similar to that generated using the <a href="/training-material/topics/single-cell/tutorials/scrna-case_alevin/tutorial.html">Alevin workflows</a> on the original FASTQ files, some of the metadata will be slightly different. Relevant results and interpretation will not change, however!</p>

<h2 id="examine-the-imported-files">Examine the imported files</h2>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What format has this tool imported?</li>
  </ol>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip"><button class="gtn-boxify-button tip" type="button" aria-controls="tip" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Selecting the title of each resultant dataset will expand the dataset in the Galaxy history.</p>
  </blockquote>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Matrix Market Format! We can tell this because our first file helpfully says <code class="language-plaintext highlighter-rouge">MatrixMarket</code> in the first line.</p>
    <figure id="figure-2" style="max-width: 90%;"><img src="../../tutorials/EBI-retrieval/images/matrix-output.png" alt="Green box containing first output, the matrix.mtx file. Columns are labelled 14458, 5218, and 5308559. " width="273" height="410" loading="lazy" /><a target="_blank" href="../../tutorials/EBI-retrieval/images/matrix-output.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Matrix Market Output</figcaption></figure>

    <p>This <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">matrix.mtx</code> file, in Matrix Market format, contains a column referring to each gene (column 1), to each cell (column 2), and the expression values themselves in the final column. To be useful, then, we need to know which genes and cells the numbers are referring to. That’s why this format comes with two more files.</p>

    <figure id="figure-3" style="max-width: 90%;"><img src="../../tutorials/EBI-retrieval/images/genes-output.png" alt="Green box containing second output, the genes.tsv file. The first column contains EnsemblIDs such as ENSMUSG######, while the second column contains gene names. There are 14,457 lines." width="274" height="414" loading="lazy" /><a target="_blank" href="../../tutorials/EBI-retrieval/images/genes-output.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Genes Output</figcaption></figure>

    <p>The <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">genes.tsv</code> file lists each EnsemblID and its gene name. The lines (14,457) corresponds with the 14458 in the Matrix file…but the 14458 contains a header, so that’s why it has one more than the genes file!</p>

    <figure id="figure-4" style="max-width: 90%;"><img src="../../tutorials/EBI-retrieval/images/cells-output.png" alt="Green box containing third output, the barcodes.tsv file. The file consists of 5,217 lines and a single column containing the cell barcode, variations of ERR2704656-AAAACACTCTGA." width="275" height="414" loading="lazy" /><a target="_blank" href="../../tutorials/EBI-retrieval/images/cells-output.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Cells Output</figcaption></figure>

    <p>The <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">barcodes.tsv</code> file lists each barcode. The lines (5,217) again correspond with the 5,218 lines in the Matrix file…which adds in the header again!</p>

    <figure id="figure-5" style="max-width: 90%;"><img src="../../tutorials/EBI-retrieval/images/cell-metadata.png" alt="Green box containing fourth output, the exp_design.tsv file. The file consists of 5,218 lines and numerous columns starting with 'Assay' and 'Sample Characteristic'." width="281" height="410" loading="lazy" /><a target="_blank" href="../../tutorials/EBI-retrieval/images/cell-metadata.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Experimental Design</figcaption></figure>

    <p>Finally, and helpfully, the tool also includes cell metadata where the <code class="language-plaintext highlighter-rouge">Assay</code> column corresponds with the barcodes in the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">barcodes.tsv</code> file. While this is not a required file to create an AnnData object from the three matrix market files, it is extremely necessary for actually interpreting the data. Imagine not knowing which barcodes came from which sample!</p>

  </blockquote>
</blockquote>

<h2 id="metadata-manipulation">Metadata manipulation</h2>

<p>At this point you might want to do some modifications in the files before downstream analysis. That can include re-formating the cell metadata or changing the names of the column headers, it all depends on your dataset and how you want to perfrom your analysis. It’s also fine to transform those files straight away. Here, we will show an extended version of metadata manipulation which allows us to create an input file consistent with <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">the next tutorial</a> workflow.</p>

<p>Before creating an AnnData object, we need to make a small modification in experimental design table. The dataset contains information about the 7 experimental samples (N701 – N707). However, in the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">exp_design.tsv</code> dataset, which contains the cell metadata, these samples are just numbered from 1 to 7.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-how-did-we-know-this"><button class="gtn-boxify-button details" type="button" aria-controls="details-how-did-we-know-this" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: How did we know this?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>You can preview this column in the the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">exp_design.tsv</code> dataset by selecting the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> in the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-history</span> Galaxy history. If you scroll to the right, and move to the column <code class="language-plaintext highlighter-rouge">Sample Characteristic[individual]</code>, you will find the batch information. Don’t worry, we’re about to rename and reformat this whole dataset to more useful titles. Make a note of the number of that column - number 12 - as we will need it to change the batch number to a batch name shortly.</p>

</blockquote>

<p>The plotting tool that we will use later will fail if the entries are integers and not categorical values, so we will change <code class="language-plaintext highlighter-rouge">1</code> to <code class="language-plaintext highlighter-rouge">N01</code> and so on.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-change-batch-numbers-into-names"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Change batch numbers into names</div>

  <ol>
    <li>
      <p>Change the datatype of <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 exp_design.tsv</code> to <code class="language-plaintext highlighter-rouge">tabular</code>:</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">tabular</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.3" title="Column Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span> with the following parameters:</p>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select cells from”</em>: <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 exp_design.tsv</code></li>
        <li><em>“using column”</em>: <code class="language-plaintext highlighter-rouge">Column: 12</code></li>
        <li>In <em>“Check”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N01</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N02</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N03</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N04</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N05</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">6</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N06</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">7</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">N07</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>While we’re renaming things, let’s also fix our titles.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-change-cell-metadata-titles"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Change cell metadata titles</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Replace parts of text</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.4)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: output from <strong>Column Regex and Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>“Find and Replace”</em>:
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">"Sample Characteristic[genotype]"</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">genotype</code></li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">"Sample Characteristic[individual]"</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">batch</code></li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">"Sample Characteristic[sex]"</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">sex</code></li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Find and Replace”</em>
          <ul>
            <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">"Sample Characteristic[cell type]"</code></li>
            <li><em>“Replace with”</em>: <code class="language-plaintext highlighter-rouge">cell_type</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output <code class="language-plaintext highlighter-rouge">Cell metadata</code></li>
  </ol>
</blockquote>

<h2 id="check-mitochondrial-gene-name-format">Check mitochondrial gene name format</h2>

<p>We might like to flag mitochondrial genes. They can be identified quite easily since - depending on the species and formatting convention - their names often start with <code class="language-plaintext highlighter-rouge">mt</code>. Since tools for flagging mitochondrial genes are often case-sensitive, it might be a good idea to check the formatting of the mitochondrial genes in our dataset.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-check-the-format-of-mitochondrial-genes-names"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Check the format of mitochondrial genes names</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1" title="Search in textfiles tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search in textfiles</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span>  <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 genes.tsv (Raw filtered counts)</code></li>
        <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Match</code></li>
        <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">mt</code></li>
        <li><em>“Match type”</em>: <code class="language-plaintext highlighter-rouge">case insensitive</code></li>
        <li><em>“Output”</em>: <code class="language-plaintext highlighter-rouge">Highlighted HTML (for easier viewing)</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output <code class="language-plaintext highlighter-rouge">Mito genes check</code></li>
  </ol>

</blockquote>

<p>If you click on that dataset, you will see all the genes containing <code class="language-plaintext highlighter-rouge">mt</code> in their name. We can now clearly see that mitochondrial genes in our dataset start with <code class="language-plaintext highlighter-rouge">mt-</code>. Keep that in mind, we might use it in a moment!</p>

<p>Now we can create our single cell object!</p>

<blockquote class="notranslate hands_on gtn-cyoa" id="gtn-cyoa">
	<div class="box-title" aria-label="hands-on box: choose your own tutorial"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Choose Your Own Tutorial</div>
	<p>This is a "Choose Your Own Tutorial" section, where you can select between multiple paths. Click one of the buttons below to select how you want to follow the tutorial</p>
	
	<p>
		You can choose whether you want to create an AnnData object for Scanpy Analysis or an RDS object for Seurat Analysis. Galaxy has more resources for Scanpy analysis, but sometimes Seurat might have what you want. The two packages are constantly trying to outdo the other! It often depends on what is more 'standard' in your work environment!
	</p>
	
	
		<input id="cyoa-opt1" type="radio" name="cyoa" value="Scanpy" onclick="cyoaChoice('Scanpy', 'gtn-cyoa')" />
		<label for="cyoa-opt1" class="select btn btn-secondary">Scanpy</label>
	
	
		<input id="cyoa-opt2" type="radio" name="cyoa" value="Seurat" onclick="cyoaChoice('Seurat', 'gtn-cyoa')" />
		<label for="cyoa-opt2" class="select btn btn-secondary">Seurat</label>
	
	
	

	
	<script>
		function ready(fn) {
			if (document.readyState != 'loading'){
				fn();
			} else {
				document.addEventListener('DOMContentLoaded', fn);
			}
		}
		ready(function() {
			cyoaDefault("Scanpy", "gtn-cyoa");
		})
	</script>
	
</blockquote>

<div class="Scanpy">

  <h2 id="creating-the-anndata-object">Creating the AnnData object</h2>

  <p>We will do several modifications within the AnnData object so that you can follow <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">the next tutorial</a>.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-create-the-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Create the AnnData Object</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_read_10x/scanpy_read_10x/1.8.1+galaxy9" title="Scanpy Read10x tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Scanpy Read10x</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8.1+galaxy9)</span></li>
      <li>Make sure you are using version <strong>1.8.1+galaxy9</strong> of the tool (change by clicking on <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> Versions button):</li>
    </ol>
    <figure id="figure-6" style="max-width: 90%;"><img src="../../images/scrna-casestudy/version.png" alt="List of available tool versions shown when clicking on the 'Versions' button on the top of the page." width="444" height="562" loading="lazy" /><a target="_blank" href="../../images/scrna-casestudy/version.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> How to change the version of the tool</figcaption></figure>

    <ol>
      <li>Set the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Expression matrix in sparse matrix format (.mtx)”</em>: <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 matrix.mtx (Raw filtered counts)</code></li>
          <li><em>“Gene table”</em>:  <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 genes.tsv (Raw filtered counts)</code></li>
          <li><em>“Barcode/cell table”</em>: <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 barcodes.tsv (Raw filtered counts)</code></li>
          <li><em>“Cell metadata table”</em>: <code class="language-plaintext highlighter-rouge">Cell metadata</code></li>
        </ul>
      </li>
      <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output <code class="language-plaintext highlighter-rouge">AnnData object</code></li>
    </ol>

  </blockquote>

  <h2 id="anndata-manipulation">AnnData manipulation</h2>

  <p>We will now change the header of the column containing gene names from <code class="language-plaintext highlighter-rouge">gene_symbols</code> to <code class="language-plaintext highlighter-rouge">Symbol</code>. This edit is only needed to make our AnnData object compatible with this tutorial’s workflow. We will also flag the mitochondrial genes.</p>

  <p>And the good news is that we can do both those steps using only one tool!</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-modify-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Modify AnnData object</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.8.1+galaxy92" title="AnnData Operations tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>AnnData Operations</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8.1+galaxy92)</span></li>
      <li>Make sure you are using version <strong>1.8.1+galaxy92</strong> of the tool (change by clicking on <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> Versions button)</li>
      <li>Set the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> In <em>“Input object in hdf5 AnnData format”</em>: <code class="language-plaintext highlighter-rouge">AnnData object</code></li>
          <li>In <em>“Change field names in AnnData var”</em>:
            <ul>
              <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Change field names in AnnData var”</em>
                <ul>
                  <li><em>“Original name”</em>: <code class="language-plaintext highlighter-rouge">gene_symbols</code></li>
                  <li><em>“New name”</em>: <code class="language-plaintext highlighter-rouge">Symbol</code></li>
                </ul>
              </li>
            </ul>
          </li>
          <li><em>“Gene symbols field in AnnData”</em>: <code class="language-plaintext highlighter-rouge">Symbol</code></li>
          <li>In <em>“Flag genes that start with these names”</em>:
            <ul>
              <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Flag genes that start with these names”</em>
                <ul>
                  <li><em>“Starts with”</em>: <code class="language-plaintext highlighter-rouge">mt-</code></li>
                  <li><em>“Var name”</em>: <code class="language-plaintext highlighter-rouge">mito</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output <code class="language-plaintext highlighter-rouge">Mito-counted AnnData for downstream analysis</code></li>
    </ol>

  </blockquote>

  <p>And that’s all! What’s even more exciting about the <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>AnnData Operations</strong> tool is that it automatically calculates a bunch of metrics, such as <code class="language-plaintext highlighter-rouge">log1p_mean_counts</code>, <code class="language-plaintext highlighter-rouge">log1p_total_counts</code>, <code class="language-plaintext highlighter-rouge">mean_counts</code>, <code class="language-plaintext highlighter-rouge">n_cells</code>, <code class="language-plaintext highlighter-rouge">n_cells_by_counts</code>, <code class="language-plaintext highlighter-rouge">n_counts</code>, <code class="language-plaintext highlighter-rouge">pct_dropout_by_counts</code>, and <code class="language-plaintext highlighter-rouge">total_counts</code>. Amazing, isn’t it?</p>

  <h2 id="conclusion">Conclusion</h2>
  <p>Now you can use this object as input for the <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html">Filter, Plot, Explore tutorial</a> and its associated workflow!</p>

  <p>Even though this tutorial was designed specifically to modify the AnnData object to be compatible with the subsequent tutorial, it also shows useful tools that you can use for your own, independent data analysis. You can find the <a href="https://singlecell.usegalaxy.eu/u/j.jakiela/w/workflow-constructed-from-history-ebi-workflow">workflow</a> and the <a href="https://singlecell.usegalaxy.eu/u/j.jakiela/h/ebi-scxa---anndata-scanpy-or-seurat-object-1">answer key history</a>. However, if you want to use the workflow from this tutorial, you have to keep in mind that different datasets may have different column names. So you have to check them first, and only then you can modify them.</p>

</div>

<div class="Seurat">

  <h2 id="creating-the-seurat-object">Creating the Seurat Object</h2>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-create-the-seurat-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Create the Seurat Object</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/seurat_read10x/seurat_read10x/4.0.4+galaxy0" title="Seurat Read10x tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Seurat Read10x</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.0.4+galaxy0)</span></li>
      <li>Set the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Expression matrix in sparse matrix format (.mtx)”</em>: <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 matrix.mtx (Raw filtered counts)</code></li>
          <li><em>“Gene table”</em>:  <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 genes.tsv (Raw filtered counts)</code></li>
          <li><em>“Barcode/cell table”</em>: <code class="language-plaintext highlighter-rouge">EBI SCXA Data Retrieval on E-MTAB-6945 barcodes.tsv (Raw filtered counts)</code></li>
          <li><em>“Cell metadata”</em>: <code class="language-plaintext highlighter-rouge">Cell metadata</code></li>
        </ul>
      </li>
      <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> output <code class="language-plaintext highlighter-rouge">Seurat object</code></li>
    </ol>

    <p>You can also choose if you want to create Seurat object, Loom or Single Cell Experiment by selecting your option in <em>“Choose the format of the output”</em>.</p>

  </blockquote>

  <h2 id="conclusion-1">Conclusion</h2>
  <p>And you’re there! You now have a usable Seurat object for analysis with Seurat tools in your history! <i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congrats!</p>

</div>

<!---
HCA doesn't work well for other datasets for now...
https://github.com/galaxyproject/training-material/issues/4567 
-->

<h1 id="human-cell-atlas-matrix-downloader">Human Cell Atlas Matrix Downloader</h1>

<p>Another public atlas that you can use to access the datasets is <a href="https://data.humancellatlas.org/">Human Cell Atlas data portal</a>. We will show you the tool in Galaxy which allows to retrieve expression matrices and metadata for any public experiment available in that repository.</p>

<p>To use it, simply set the project title, project label or project UUID, which can be found at the <a href="https://data.humancellatlas.org/explore/projects">HCA data browser</a>, and select the desired matrix format (Matrix Market or Loom).</p>

<figure id="figure-7" style="max-width: 90%;"><img src="../../images/scrna-data/HCA.jpg" alt="Image showing project UUID as a final fragment of link address, project title (self-explanatory) and project label as an entry in the box on the right side of the page." width="1869" height="973" loading="lazy" /><a target="_blank" href="../../images/scrna-data/HCA.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Where to find project title, project label and project UUID</figcaption></figure>

<p>For projects that have more than one organism, one needs to be specified. Otherwise, there is no need to set the species.</p>

<p>Let’s use the suggested example of the project <em>Single cell transcriptome analysis of human pancreas</em>. If you check this project in HCA, you’ll find out that it’s actually its label. But it should work well if you enter the title or UUID!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create AnnData object</div>

  <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/hca_matrix_downloader/hca_matrix_downloader/v0.0.4+galaxy0" title="Human Cell Atlas Matrix Downloader tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Human Cell Atlas Matrix Downloader</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version v0.0.4+galaxy0)</span> with the following parameters:</p>
  <ul>
    <li><em>“Human Cell Atlas project name/label/UUID”</em>: <code class="language-plaintext highlighter-rouge">Single cell transcriptome analysis of human pancreas</code></li>
    <li><em>“Choose the format of matrix to download”</em>:  <code class="language-plaintext highlighter-rouge">Matrix Market</code></li>
  </ul>
</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-errors-that-you-might-encounter"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Errors that you might encounter</div>
  <p>If your dataset turns red, there might be several reasons for that, for example:</p>
  <ul>
    <li>“There are too many connected users” - please be patient and re-run the step later, as it is advised</li>
    <li>“Project identifier was not found in the database” - double check the spelling, try entering project title, project label or project UUID.</li>
  </ul>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-will-be-the-output"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-will-be-the-output" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What will be the output?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>When “Matrix Market” is seleted, outputs are in 10X-compatible Matrix Market format:</p>
  <ul>
    <li><strong>Matrix (txt)</strong>: Contains the expression values for genes (rows) and cells (columns) in raw counts. This text file is formatted as a Matrix Market file, and as such it is accompanied by separate files for the gene identifiers and the cells identifiers.</li>
    <li><strong>Genes (tsv)</strong>: Identifiers (column repeated) for the genes present in the matrix of expression, in the same order as the matrix rows.</li>
    <li><strong>Barcodes (tsv)</strong>: Identifiers for the cells of the data matrix. The file is ordered to match the columns of the matrix.</li>
    <li><strong>Experiment Design file (tsv)</strong>: Contains metadata for the different cells of the experiment.</li>
  </ul>

  <p>When “Loom” is selected, output is a single Loom HDF5 file:</p>
  <ul>
    <li><strong>Loom (h5)</strong>: Contains expression values for genes (rows) and cells (columns) in raw counts, cell metadata table and gene metadata table, in a <a href="http://linnarssonlab.org/loompy/format/index.html">single HDF5 file</a>.</li>
  </ul>

</blockquote>

<p>If you chose <strong>Loom</strong> format and you need to convert your file to other datatype, you can use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/sceasy_convert/sceasy_convert/0.0.7+galaxy1" title="SCEasy tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SCEasy</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.7+galaxy1)</span> (more details in the next section). If you chose <strong>Matrix Market</strong> format, you can then transform the output to AnnData or Seurat, as shown in the EBI SCXA example above. Below, you will find an example of transforming the output to AnnData object.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-anndata-object-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create AnnData object</div>

  <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_read_10x/scanpy_read_10x/1.8.1+galaxy9" title="Scanpy Read10x tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Scanpy Read10x</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8.1+galaxy9)</span> with the following parameters:
 Make sure you are using version <strong>1.8.1+galaxy9</strong> of the tool (change by clicking on <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> Versions button)</p>
  <ul>
    <li><em>“Expression matrix in sparse matrix format (.mtx)”</em>: <code class="language-plaintext highlighter-rouge">Human Cell Atlas Matrix Downloader on matrix.mtx</code></li>
    <li><em>“Gene table”</em>:  <code class="language-plaintext highlighter-rouge">Human Cell Atlas Matrix Downloader on genes.tsv</code></li>
    <li><em>“Barcode/cell table”</em>: <code class="language-plaintext highlighter-rouge">Human Cell Atlas Matrix Downloader on barcodes.tsv</code></li>
    <li><em>“Cell metadata table”</em>: <code class="language-plaintext highlighter-rouge">Human Cell Atlas Matrix Downloader on exp_design.tsv</code></li>
  </ul>
</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-flagging-genes-by-using-anndata-operations"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-flagging-genes-by-using-anndata-operations" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Flagging genes by using AnnData Operations</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>After you create AnnData file, you can additionally use the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.8.1+galaxy92" title="AnnData Operations tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>AnnData Operations</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8.1+galaxy92)</span> tool (note the version <strong>1.8.1+galaxy92</strong>) before downstream analysis. It’s quite a useful tool since not only does it flag mitochondrial genes, but also automatically calculates a bunch of metrics, such as <code class="language-plaintext highlighter-rouge">log1p_mean_counts</code>, <code class="language-plaintext highlighter-rouge">log1p_total_counts</code>, <code class="language-plaintext highlighter-rouge">mean_counts</code>, <code class="language-plaintext highlighter-rouge">n_cells</code>, <code class="language-plaintext highlighter-rouge">n_cells_by_counts</code>, <code class="language-plaintext highlighter-rouge">n_counts</code>, <code class="language-plaintext highlighter-rouge">pct_dropout_by_counts</code>, and <code class="language-plaintext highlighter-rouge">total_counts</code>.</p>

  <p>When you use it to flag mitochondrial genes, here are some formatting tips:</p>
  <ul>
    <li>Remember to check the name of the column with gene symbols</li>
    <li>This tool is case sensitive</li>
    <li>No parentheses needed when typing in the values</li>
    <li>Including a dash is important to identify mitochondrial genes (eg. <strong>MT-</strong>)</li>
  </ul>
</blockquote>

<p>You can have a look at the <a href="https://singlecell.usegalaxy.eu/u/j.jakiela/h/hca-downloader---example-files-1">answer history</a> of performing those three quick steps.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>The EMBL-EBI Single-cell Expression Atlas contains high quality datasets.</p>
</li>
                        
                        <li><p>Metadata manipulation is key for generating the correctly formatted files.</p>
</li>
                        
                        <li><p>To use Scanpy tools, you have to transform your metadata into an AnnData object.</p>
</li>
                        
                        <li><p>To use Seurat tools, you have to transform your metadata into a Seurat object.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/single-cell/faqs/">FAQ page for the Single Cell topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Moreno2020.04.08.032698">Moreno, P., N. Huang, J. R. Manning, S. Mohammed, A. Solovyev <i>et al.</i>, 2020 <b>User-friendly, scalable tools and workflows for single-cell analysis</b>. bioRxiv. <a href="https://doi.org/10.1101/2020.04.08.032698">10.1101/2020.04.08.032698</a> <a href="https://www.biorxiv.org/content/early/2020/04/09/2020.04.08.032698">https://www.biorxiv.org/content/early/2020/04/09/2020.04.08.032698</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/EBI-retrieval"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/EBI-retrieval">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Julia Jakiela, Wendi Bacon,  <b>Importing files from public atlases (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-EBI-retrieval,
author = "Julia Jakiela and Wendi Bacon",
	title = "Importing files from public atlases (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/EBI-retrieval/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        
		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>

                
<div class="d-flex flex-wrap">

	<a href="/training-material/hall-of-fame/elixir-fair-data/" class="funder-badge">
		
		<div class="avatar">
			
				
				<img class="funder-avatar" src="https://avatars.githubusercontent.com/elixir-fair-data" alt="Logo">
				
			
		</div>

		<div class="info">
			<div class="name">ELIXIR Fair Data</div>
			<div class="description">
			This work has been funded by the ELIXIR-UK FAIR Data Stewardship training UKRI award (MR/V038966/1)

            
			</div>
		</div>
	</a>

</div>


        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Filter, plot and explore single-cell RNA-seq data with Scanpy</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


                    </ul>
                </blockquote>
                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/EBI-retrieval/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: anndata_ops
  owner: ebi-gxa
  revisions: a0274bc43b7e
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hca_matrix_downloader
  owner: ebi-gxa
  revisions: c3c85b67d118
  tool_panel_section_label: Single Cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: retrieve_scxa
  owner: ebi-gxa
  revisions: 72b6e1747e37
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: scanpy_read_10x
  owner: ebi-gxa
  revisions: 9afcfcd1d215
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_read10x
  owner: ebi-gxa
  revisions: 301cf1b4b43d
  tool_panel_section_label: Seurat
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seurat_read10x
  owner: ebi-gxa
  revisions: f5db0e853f57
  tool_panel_section_label: Seurat
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: regex_find_replace
  owner: galaxyp
  revisions: 503bcd6ebe4b
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sceasy_convert
  owner: iuc
  revisions: d70139524747
  tool_panel_section_label: Single-cell
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00374">gxy.io/GTN:T00374</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/EBI-retrieval/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/EBI-retrieval/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/55607e95d8efdacc58f4eb03501f1419ce408b25">55607e9</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>