<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Peptide and Protein Quantification via Stable Isotope Labelling (SIL) / Proteomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@0fde6e77192bb0a1f0dac3e8539204bd58b7263c",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.f1f2de89.js?v=1708530376' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.80a5488c.js?v=1708530376' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Hands-on: Hands-on: Peptide and Protein Quantification via Stable Isotope Labelling (SIL)">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-03 14:30:27 +0000">
<meta name="DC.creator" content="Florian Christoph Sigloch">
<meta name="DC.creator" content="Björn Grüning">
<meta name="DC.creator" content="Matthias Fahrner"><meta name="description" content="Training material for proteomics workflows in Galaxy">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Proteomics / Hands-on: Peptide and Protein Quantification via Stable Isotope Labelling (SIL)">
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "dateModified": "2023-11-03 14:30:27 +0000",
  "datePublished": "2017-03-17 17:43:31 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Hands-on: Hands-on: Peptide and Protein Quantification via Stable Isotope Labelling (SIL)",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "identifier": "https://gxy.io/GTN:T00230",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on: Hands-on: Peptide and Protein Quantification via Stable Isotope Labelling (SIL)",
  "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html",
  "version": 66,
  "timeRequired": "PT1H",
  "teaches": "MS1 feature quantitation and mapping of quantitations to peptide and protein IDs.",
  "keywords": [
    "Proteomics",
    "DDA",
    "SILAC"
  ],
  "description": "The questions this  addresses are:\n - What are MS1 features?\n - How to quantify based on MS1 features?\n - How to map MS1 features to MS2 identifications?\n - How to evaluate and optimize the results?\n\n\nThe objectives are:\n - MS1 feature quantitation and mapping of quantitations to peptide and protein IDs.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/database-handling/tutorial.html",
      "name": "Protein FASTA Database Handling",
      "description": "Hands-on for 'Protein FASTA Database Handling' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html",
      "name": "Hands-on: Hands-on: Peptide and Protein ID using OpenMS tools",
      "description": "Hands-on for 'Hands-on: Hands-on: Peptide and Protein ID using OpenMS tools' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/stortebecker/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/stortebecker/",
      "name": "Florian Christoph Sigloch",
      "image": "https://avatars.githubusercontent.com/stortebecker",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "name": "Björn Grüning",
      "image": "https://avatars.githubusercontent.com/bgruening",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-3079-6586",
      "orcid": "https://orcid.org/0000-0002-3079-6586"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/matthias313/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/matthias313/",
      "name": "Matthias Fahrner",
      "image": "https://avatars.githubusercontent.com/matthias313",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-7955-2518",
      "orcid": "https://orcid.org/0000-0001-7955-2518"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ],
  "educationalLevel": "Advanced",
  "mentions": [
    {
      "@type": "Thing",
      "name": "DDA"
    },
    {
      "@type": "Thing",
      "name": "SILAC"
    }
  ],
  "abstract": "To compare protein amounts in different samples from MS/MS data, two different experiment setups exist. Firstly, unmodified proteins can be measured in separate runs at one sample per MS-run. Secondly, proteins of samples to compare can be labelled with small chemical tags, mixed, and measured side-by-side in a single MS-run."
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.f1f2de89.js?v=1708530376'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Proteomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/proteomics/faqs/" title="Check our FAQs for the Proteomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/protein-quant-sil/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/smorgasbord2023/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">smorgasbord2023</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2024-02-01/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">2024-02-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2024-01-01/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">2024-01-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-proteomics">
    <h1 data-toc-skip>Peptide and Protein Quantification via Stable Isotope Labelling (SIL)</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/stortebecker/" class="contributor-badge contributor-stortebecker"><img src="https://avatars.githubusercontent.com/stortebecker?s=36" alt="Florian Christoph Sigloch avatar" width="36" class="avatar" />
    Florian Christoph Sigloch</a><a href="/training-material/hall-of-fame/bgruening/" class="contributor-badge contributor-bgruening"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bgruening?s=36" alt="Björn Grüning avatar" width="36" class="avatar" />
    Björn Grüning</a><a href="/training-material/hall-of-fame/matthias313/" class="contributor-badge contributor-matthias313"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/matthias313?s=36" alt="Matthias Fahrner avatar" width="36" class="avatar" />
    Matthias Fahrner</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are MS1 features?</p>
</li>
        
        <li><p>How to quantify based on MS1 features?</p>
</li>
        
        <li><p>How to map MS1 features to MS2 identifications?</p>
</li>
        
        <li><p>How to evaluate and optimize the results?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>MS1 feature quantitation and mapping of quantitations to peptide and protein IDs.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Protein FASTA Database Handling:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Hands-on: Hands-on: Peptide and Protein ID using OpenMS tools:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Advanced <span class="visually-hidden">Advanced</span>
<span class="level advanced" title="Advanced Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/1051552">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/proteomics/tutorials/protein-quant-sil/workflows/" title="Peptide and Protein Quantification via Stable Isotope Labelling (SIL) workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/protein-quant-sil/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Mar 17, 2017 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 3, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00230">https://gxy.io/GTN:T00230</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 66 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>To compare protein amounts in different samples from MS/MS data, two different experiment setups exist. Firstly, unmodified proteins can be measured in separate runs at one sample per MS-run. Secondly, proteins of samples to compare can be labelled with small chemical tags, mixed, and measured side-by-side in a single MS-run.
There are two types of chemical tags:</p>
<ol>
  <li>Isobaric tags display the same mass on first hand, but fragment during the generation of the MS/MS spectra to yield reporter ions of different mass. The intensity of those reporter ions can be compared in MS/MS spectra. There are two types of isobaric tags commercially available: <em>tandem mass tags</em> (TMT) and <em>isobaric tags for relative and absolute quantitation</em> (iTRAQ).</li>
  <li>Isotopic tags are chemically identical, but differ in their mass due to incorporated stable isotopes. Examples of different isotopic tags for stable isotope labelling (SIL) are ICAT, SILAC, dimethylation, or heavy oxygen (<sup>18</sup>O).</li>
</ol>

<p>This tutorial deals with protein quantitation via stable isotope labelling (SIL). For isotopic tags, quantitation can be achieved by comparing the intensity of MS1 peptide mass traces. The whole MS1 profile of a peptide, i.e. the intensities of all its isotopic peaks over time, is called a <em>peptide feature</em> (Figure 1a).
Incorporation of stable isotopes results in different peptide masses on MS1 level, which give rise to coeluting ion traces in the TIC with a mass difference typical for each different chemical tag (Figure 1b). Figure originally published in <a href="http://www.ncbi.nlm.nih.gov/pubmed/25931027">Nilse et al, 2015</a>.</p>

<p>In this tutorial, we will use tools of the <a href="http://www.openms.org">OpenMS suite</a> to identify and quantify peptides and proteins.</p>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/protein-quant-sil_ms1feature.png" alt="ms1 feature. " width="3265" height="684" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_ms1feature.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> MS1 mass traces. A) Two peptide features of co-eluting SIL peptides. B) MS1 spectra at a given RT. C) XIC monoisotopic peak light peptide. D) XIC monoisotopic peak heavy peptide.</figcaption></figure>

<h2 id="prerequisites">Prerequisites</h2>

<p>If you are in the planning phase of your quantitative proteomics experiment, you may want to consider our tutorial on different <a href="/training-material/topics/proteomics/tutorials/labelfree-vs-labelled/tutorial.html">quantitation methods</a> first.</p>

<p>To learn about <em>protein identification</em> in Galaxy, please consider our OpenMS-based <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">peptide ID tutorial</a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-introduction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Introduction</div>
  <p>In the hands-on section of this tutorial, we will use a quantitative comparison of HEK cell lysate as a test dataset. In this experiment, HEK cells were once labelled with light, once with heavy SILAC. Both cultures were lysed simultaneously and the cell lysates were mixed in a certain ratio.
A detailed description of the full dataset is available in the <a href="https://www.ebi.ac.uk/pride/">PRIDE archive</a>.</p>

  <p>Your objective in this hands-on-tutorial is to find out the correct mixing ratio of the test sample.</p>

  <p>To speed up analysis, the input dataset was filtered to include only those data acquired in second 2000-3000 of the original LC gradient.</p>
</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#prerequisites" id="markdown-toc-prerequisites">Prerequisites</a></li>
  <li><a href="#peptide-and-protein-identification" id="markdown-toc-peptide-and-protein-identification">Peptide and Protein Identification</a></li>
  <li><a href="#ms1-feature-detection" id="markdown-toc-ms1-feature-detection">MS1 Feature Detection</a></li>
  <li><a href="#mapping-identifications-to-features" id="markdown-toc-mapping-identifications-to-features">Mapping Identifications to Features</a></li>
  <li><a href="#descriptive-statistics-and-plotting-of-protein-quantitations" id="markdown-toc-descriptive-statistics-and-plotting-of-protein-quantitations">Descriptive Statistics and Plotting of Protein Quantitations</a></li>
  <li><a href="#evaluation-and-optimization-of-quantitation-results" id="markdown-toc-evaluation-and-optimization-of-quantitation-results">Evaluation and Optimization of Quantitation Results</a>    <ol>
      <li><a href="#examples" id="markdown-toc-examples">Examples</a></li>
      <li><a href="#typical-problems" id="markdown-toc-typical-problems">Typical Problems</a></li>
      <li><a href="#optimization-of-quantitation-results" id="markdown-toc-optimization-of-quantitation-results">Optimization of Quantitation Results</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="peptide-and-protein-identification">Peptide and Protein Identification</h1>

<p>In this tutorial, peptide identification will be performed using the workflow of the previous <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">Peptide ID Tutorial</a>. Alternatively one can perform the protein identification step by step in the <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/tutorial.html">Peptide ID Tutorial</a> using the SILAC dataset from <a href="https://zenodo.org/record/1051552">zenodo</a> <strong>but</strong> beware to specify the labels in the <code class="language-plaintext highlighter-rouge">param_variable_modifications</code> of <strong>XTandemAdapter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<p>A common problem in mass spectrometry are misassigned mono-isotopic precursor peaks. Most search engines allow for some adaptation of the monoisotopic peak and we will use this by leaving <code class="language-plaintext highlighter-rouge">By default, misassignment to the first and second isotopic 13C peak are also considered</code> at <code class="language-plaintext highlighter-rouge">No</code>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this SILAC Proteome exercise</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the mzml file, containing the measured mass spectra from <a href="https://doi.org/10.5281/zenodo.1051552">Zenodo</a> or a data library:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1051552/files/HEK_SILAC-K6R6_ST905_part.mzml
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <p>The data have been preprocessed during the conversion from the machine raw file. We used background removal on MS1 and MS2 level, and MS2 deisotoping.</p>
      </blockquote>
    </li>
    <li>
      <p>Import the human protein database (including cRAP contaminants and decoys) from <a href="https://zenodo.org/record/892005/files/Human_database_including_decoys_%28cRAP_added%29.fasta">zenodo</a> or a data library:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/892005/files/Human_database_including_decoys_%28cRAP_added%29.fasta
</code></pre></div>      </div>
    </li>
    <li>
      <p>Import the <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/workflows/workflow.ga">workflow</a></p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/protein-id-oms/workflows/workflow.ga
</code></pre></div>      </div>
    </li>
    <li>
      <p>Modify the workflow</p>
    </li>
    <li>Connect the <code class="language-plaintext highlighter-rouge">mzML</code> input directly to the <strong>XTandemAdapter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> node and delete <strong>PeakPickerHiRes</strong>.</li>
    <li>
      <p>Change the <strong>XTandemAdapter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> parameters: Add the <strong>param_variable_modifications</strong> <code class="language-plaintext highlighter-rouge">Label:13C(6) (K)</code> and <code class="language-plaintext highlighter-rouge">Label:13C(6) (R)</code></p>
    </li>
    <li>Run the workflow with
      <ul>
        <li>the mzML dataset <code class="language-plaintext highlighter-rouge">1: mzML dataset</code></li>
        <li>the human FASTA database <code class="language-plaintext highlighter-rouge">2: protein FASTA database</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-using-galaxy-workflows"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-galaxy-workflows" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Using Galaxy Workflows</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>If you want to learn more about Galaxy workflows, please consult the <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101/tutorial.html#the-workflow-editor">Galaxy Introduction</a></p>
  </blockquote>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>How many peptides and proteins were successfully identified?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>2217 non-redundant peptides and 763 proteins were identified.
Numbers may slightly vary depending on the versions of the tools and the used FASTA file.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="ms1-feature-detection">MS1 Feature Detection</h1>

<p>MS1 feature detection is a critical step in quantitative workflows. In principle, there are two different ways to define features:</p>
<ol>
  <li>Feature detection solely based upon MS1 data (mzML/raw files) without prior knowledge of peptide identifications (IDs).
    <ul>
      <li>Advantage: Feature results can be used to assist in peptide identification.</li>
      <li>Drawback: Not all peptide identifications can be mapped to features, thus not every identified peptide can be quantified.</li>
    </ul>
  </li>
  <li>Feature detection based upon peptide IDs.
    <ul>
      <li>Advantage: Most peptide IDs trigger a feature detection.</li>
      <li>Drawback: Feature results cannot be used to improve peptide ID.</li>
    </ul>
  </li>
</ol>

<p>The OpenMS suite provides several tools (FeatureFinders) for MS1 feature detection. For SIL we have to use <strong>FeatureFinderMultiplex</strong>, which does not take peptide IDs as an input.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-ms1-feature-detection"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: MS1 Feature Detection</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/openms_featurefindermultiplex/FeatureFinderMultiplex/2.6+galaxy0" title="FeatureFinderMultiplex tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FeatureFinderMultiplex</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6+galaxy0)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“LC-MS dataset in centroid or profile mode”</em>: mzML file</li>
        <li>In <em>“algorithmic parameters”</em>
          <ul>
            <li><em>“Labels used for labelling the samples”</em>: <code class="language-plaintext highlighter-rouge">[ ][Arg6,Lys6]</code></li>
            <li><em>“m/z tolerance for search of peak patterns”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>“Maximum number of missed cleavages due to incomplete digestion”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
        <li><em>“Optional outputs”</em>: <code class="language-plaintext highlighter-rouge">out_multiplets (Optional output file containing all detected peptide groups</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-multiple-labels-per-peptide"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Multiple labels per peptide</div>
  <p>When using SILAC-KR or dimethyl-labelling and trypsin digestion, exactly one labelled amino acid per peptide is expected. The only labelled amino acids are lysine (K) and arginine (R) and trypsin cuts after each of them. However, a small percentage of missed cleavage normally occur also in those datasets. Setting <em>“Maximum number of missed cleavages due to incomplete digestion”</em> to <code class="language-plaintext highlighter-rouge">1</code> will be sufficient to deal with most missed cleavages.</p>

  <p>When using other enzymes (e.g. Lys-C) or other labels (e.g. \(^{18}O\)), several labelled amino acids per peptide are expected. You can search for those features by increasing the parameter <em>“Maximum number of missed cleavages due to incomplete digestion”</em>.</p>
</blockquote>

<h1 id="mapping-identifications-to-features">Mapping Identifications to Features</h1>

<p>We now have feature quantifications for MS1 elution peaks, peptide identifications for the MS2 spectra and protein identifications.
The next step is to map the MS2-based peptide identifications to the quantified MS1 precursor peaks (“peptide features”). This will enable the quantification of identified peptides.
For labelled data, it is necessary to map peptide identifications to <em>consensus</em> features (i.e. a pair of one light peptide feature with one matching heavy feature in the correct m/z distance).
For <code class="language-plaintext highlighter-rouge">consensusXML</code>, IDMapper uses the consensus centroids, not the feature boundaries for mapping. Therefore, the RT tolerance has to be set higher than for mapping to <code class="language-plaintext highlighter-rouge">featureXML</code>. A good starting value is half the expected typical retention time.</p>

<p>Sometimes several peptide identifications are mapped to a feature. The tool <a href="http://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/TOPP_IDConflictResolver.html">IDConflictResolver</a> filters the mapping so that only the identification with the best score is associated to each feature.
Another refinement of the quantitative result is obtained by removing falsely mapped identifications e.g. light identification mapped to heavy feature. This step is performed by the <strong>MultiplexResolver</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> that returns a first file with the correctly mapped peptides and as a second output the falsly mapped peptides.</p>

<p>Finally, the correctly mapped peptides will be combined into protein quantifications with the <strong>ProteinQuantifier</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quant-to-id-matching"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quant to ID matching</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/openms_idmapper/IDMapper/2.6+galaxy0" title="IDMapper tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>IDMapper</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6+galaxy0)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Protein/peptide identifications file”</em>: output of <strong>FalseDiscoveryRate</strong></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Feature map/consensus map file”</em>: output of <strong>FeatureFinderMultiplex</strong></li>
        <li><em>“RT tolerance (in seconds) for the matching of peptide identifications and (consensus) features”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
        <li><em>“m/z tolerance (in ppm or Da) for matching of peptide identifications and (consensus) features”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
        <li>In <em>“Additional options for consensusXML input”</em>
          <ul>
            <li><em>“Match using RT and m/z of sub-features instead of consensus RT and m/z”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>“Store the map index of the sub-feature in the peptide ID”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/openms_idconflictresolver/IDConflictResolver/2.6+galaxy0" title="IDConflictResolver tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>IDConflictResolver</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6+galaxy0)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: output of <strong>IDMapper</strong></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/openms_multiplexresolver/MultiplexResolver/2.5+galaxy0" title="MultiplexResolver tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiplexResolver</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.5+galaxy0)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Peptide multiplets with assigned sequence information”</em>: output of <strong>IDConflictResolver</strong></li>
        <li><em>“Labels used for labelling the samples”</em>: <code class="language-plaintext highlighter-rouge">[ ][Arg6,Lys6]</code></li>
        <li><em>“Maximum number of missed cleavages due to incomplete digestion”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/openms_proteinquantifier/ProteinQuantifier/2.6+galaxy0" title="ProteinQuantifier tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>ProteinQuantifier</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.6+galaxy0)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: first output of <strong>MultiplexResolver</strong></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Protein inference results”</em>: output of <strong>FalseDiscoveryRate</strong></li>
        <li><em>“Calculate protein abundance from this number of proteotypic peptides (most abundant first; ‘0’ for all)”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li><em>“Averaging method used to compute protein abundances from peptide abundances”</em>: <code class="language-plaintext highlighter-rouge">sum</code></li>
        <li><em>“Add the log2 ratios of the abundance values to the output”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Optional outputs”</em>: <code class="language-plaintext highlighter-rouge">out (Output file for protein abundances)</code> and <code class="language-plaintext highlighter-rouge">peptide_out (Output file for peptide abundances)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-proteinquantifier-parameters"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: ProteinQuantifier parameters</div>
  <p>Peptide quantitation algorithms are more precise for high abundant peptides. Therefore, it is recommended to base protein quantitations on those peptides. In ProteinQuantifier, you may restrict the calculation of protein abundances to the most abundant peptides by using the option “Calculate protein abundance from this number of proteotypic peptides”.
However, we recommend to use the averaging method <code class="language-plaintext highlighter-rouge">sum</code> instead. By using this option, protein ratios are based on the sum of all peptide abundances. Thus, highly abundant peptides thus have more influence on protein abundance calculation than low abundant peptides.
A simple sum-of-intensities algorithm provided the best estimates of true protein ratios in a comparison of several protein quantitation algorithms (<a href="https://www.ncbi.nlm.nih.gov/pubmed/19892804">Carrillo et al., Bioinformatics, 2009</a>).</p>
</blockquote>

<h1 id="descriptive-statistics-and-plotting-of-protein-quantitations">Descriptive Statistics and Plotting of Protein Quantitations</h1>

<p><strong>ProteinQuantifier</strong> produces two output tables: the first one gives information about the quantified proteins, the second one gives information about the quantified peptides.
For proteins, we added a log-transformed ratio to the output, which is saved in column 8 of the protein table. The ratio is calculated as <strong>log2 (abundance2/abundance1)</strong>, which is sometimes called the <em>fold change (FC)</em> ratio.</p>

<p>To get a quick overview of the results, you can calculate basic descriptive statistics and plot the data as a histogram.
Comment lines in the beginning of a <code class="language-plaintext highlighter-rouge">tabular</code> file may sometimes cause errors, therefore we will remove them with the tool <strong>Select last lines from a dataset (tail)</strong>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-descriptive-statistics"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Descriptive Statistics</div>
  <ol>
    <li>Run <span class="tool" data-tool="Summary_Statistics1" title="Summary Statistics tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Summary Statistics</strong></span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Summary statistics on”</em>: protein table output (first file) of <strong>ProteinQuantifier</strong></li>
        <li><em>“Column or expression”</em>: <code class="language-plaintext highlighter-rouge">c8</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_tail_tool/1.1.0" title="Select last lines from a dataset (tail) tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Select last lines from a dataset (tail)</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Text file”</em>: protein table output (first file) of <strong>ProteinQuantifier</strong></li>
        <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Keep everything from this line on</code></li>
        <li><em>“Number of lines”</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/histogram/histogram_rpy/1.0.4" title="Histogram tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Histogram</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.4)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset”</em>: output of <strong>Select last</strong></li>
        <li><em>“Numerical column for x axis”</em>: <code class="language-plaintext highlighter-rouge">Column: 8</code></li>
        <li><em>“Number of breaks (bars)”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
        <li><em>“Plot title”</em> and <em>“Label for x axis”</em> to something meaningful.</li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-calculating-descriptive-statistics-for-peptides"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Calculating descriptive statistics for peptides</div>
    <p>The peptide table output of <strong>ProteinQuantifier</strong> does not give the log-transformed ratio for each peptide. Nonetheless, you may calculate basic statistics of the FC values by running <strong>Summary Statistics</strong> with <em>“Column or expression”</em> set to <code class="language-plaintext highlighter-rouge">log(c6/c5,2)</code>.</p>
  </blockquote>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>How many peptides and proteins were successfully quantified?</li>
    <li>What might have been the mixing ratio of the input dataset?</li>
    <li>In the histogram, there is a second local maximum at about FC 0. What might that mean?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>With the above parameters, you should have quantified 870 peptides and 421 proteins (slight variations are expected depending on tool version or fasta file)</li>
      <li>In the histogram, you see that the peak of the density curve is between -1.1 and -1.2. In the summary statistics, you can see that the mean protein ratio was -0.98. An FC of -1 indicates that the unlabelled proteins were twice as abundant as their heavy-labelled counterparts. Indeed, the mixing ratio of the dataset was 2 parts light labelled HEK cell lysate and 1 part heavy labelled HEK cell lysate.</li>
      <li>Some proteins were quantified with an FC close to 0. These may stem from incomplete SILAC labelling. Even after two weeks of cell culture in SILAC medium, some proteins with a very low turnover rate may remain unlabelled.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="evaluation-and-optimization-of-quantitation-results">Evaluation and Optimization of Quantitation Results</h1>

<p>Protein quantitation is a multi-step procedure. Many parameters of different steps influence the final results. Therefore, it is recommended to optimize the tool parameters for each dataset and to carefully evaluate quantitation results. While the total number of quantified proteins is a first important parameter for optimization, it is also necessary to visualize the results and check for correct feature finding and ID mapping.</p>

<p>Galaxy does not provide a tool for proteomics visualization, we recommend to use the OpenMS Viewer <strong>TOPPView</strong>.
Basic TOPPView tutorials are available as <a href="https://www.openms.de/getting-started/command-line-and-visualisations/">videos</a> and a more comprehensive tutorial as <a href="http://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/html/TOPP_TOPPView.html">HTML</a> or <a href="http://ftp.mi.fu-berlin.de/pub/OpenMS/release-documentation/TOPP_tutorial.pdf">PDF</a>.</p>

<p>For the optimization of tool parameters, it is recommended not to work with a complete LC-MS/MS run. Instead, we will use <strong>FileFilter</strong> to extract a small <em>RT-slice</em> of our input dataset, i.e. a fraction of the original dataset that was measured during a short period of time. Reducing the test data reduces the time needed for analysis and facilitates visual examination of the data.</p>

<p>Using Galaxy Workflows enables us to quickly re-run a full analysis with changed parameters. To learn about Galaxy Workflows, consult this <a href="/training-material/topics/galaxy-interface/tutorials/history-to-workflow/tutorial.html">tutorial</a>.</p>

<p><strong>Cave:</strong> Be aware that only very small parts of your dataset can be checked by visual examination. To minimize biases, try to look at the same areas / features of each result file.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-reduction-and-visual-evaluation-with-toppview"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data reduction and visual evaluation with TOPPView</div>

  <ol>
    <li>Run <strong>FileFilter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">mzML</code> file</li>
        <li><em>“Retention time range to extract”</em>: <code class="language-plaintext highlighter-rouge">2000:2200</code></li>
      </ul>
    </li>
    <li>
      <p>Extract a workflow out of your history or import the <a href="/training-material/topics/proteomics/tutorials/protein-id-oms/workflows/workflow.ga">premade workflow</a></p>
    </li>
    <li>
      <p>Run the whole workflow again with default settings on the reduced <code class="language-plaintext highlighter-rouge">mzML</code> file.</p>
    </li>
    <li>Run <strong>FileFilter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <strong>IDConflictResolver</strong> output</li>
        <li><em>“Remove features without annotations”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the <strong>FileFilter</strong> output to <code class="language-plaintext highlighter-rouge">Annotated features</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Run <strong>FileFilter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <strong>IDConflictResolver</strong> output</li>
        <li><em>“Remove features without annotations”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the <strong>FileFilter</strong> output to <code class="language-plaintext highlighter-rouge">UNannotated features</code></p>
    </li>
    <li>Download locally the following files:
      <ul>
        <li>Spectra: <code class="language-plaintext highlighter-rouge">mzML</code> file</li>
        <li>peptide IDs: <strong>IDScoreSwitcher</strong> <code class="language-plaintext highlighter-rouge">idXML</code> output file</li>
        <li>features: <strong>FeatureFinderMultiplex</strong> <code class="language-plaintext highlighter-rouge">featureXML</code> output file</li>
        <li>consensus features: <strong>FileFilter</strong> <code class="language-plaintext highlighter-rouge">consensusXML</code> output files (“Annotated” and “UNannotated” features)</li>
      </ul>
    </li>
    <li>Open <strong>TOPPView</strong></li>
    <li>Open the <code class="language-plaintext highlighter-rouge">mzML</code> file in <strong>TOPPView</strong> with
      <ul>
        <li><strong>Open as</strong> set to <code class="language-plaintext highlighter-rouge">new window</code></li>
        <li><strong>Map view</strong> set to <code class="language-plaintext highlighter-rouge">2D</code></li>
        <li><strong>Low intensity cutoff</strong> set to <code class="language-plaintext highlighter-rouge">off</code></li>
      </ul>
    </li>
    <li>Open all other downloaded files in <strong>TOPPView</strong> with
      <ul>
        <li><strong>Open as</strong> set to <code class="language-plaintext highlighter-rouge">new layer</code></li>
      </ul>
    </li>
    <li>Activate the <code class="language-plaintext highlighter-rouge">mzML</code> layer and click on <code class="language-plaintext highlighter-rouge">Show projections</code></li>
  </ol>

  <figure id="figure-2" style="max-width: 90%;"><img src="../../images/protein-quant-sil_showProjections.png" alt="showProjections. " width="1265" height="692" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_showProjections.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> The projections display intensities plotted against RT (top panel) and plotted against m/z (right panel).</figcaption></figure>

  <ol>
    <li>Activate the <code class="language-plaintext highlighter-rouge">consensusXML</code> layers and click on <code class="language-plaintext highlighter-rouge">Show consensus feature element positions</code></li>
  </ol>

  <figure id="figure-3" style="max-width: 90%;"><img src="../../images/protein-quant-sil_showConsensus.png" alt="showConsensus. " width="1396" height="214" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_showConsensus.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Arrows between linked light and heavy features centroids are displayed.</figcaption></figure>

  <ol>
    <li>Evaluate your data analysis, by
      <ul>
        <li>zooming into a specific region (hold <code class="language-plaintext highlighter-rouge">Ctrl</code> and use the mouse to zoom)</li>
        <li>measuring m/z and RT distances (select the <code class="language-plaintext highlighter-rouge">mzML</code> layer, hold <code class="language-plaintext highlighter-rouge">Shift</code> and use the mouse to measure)</li>
        <li>displaying an area in 3D view (<code class="language-plaintext highlighter-rouge">right-click</code> into the 2D View and select <code class="language-plaintext highlighter-rouge">Switch to 3D view</code>)</li>
        <li>switching on and off the display of single layers (<code class="language-plaintext highlighter-rouge">left-click</code> at the tick-boxes in the window “Layers”)</li>
      </ul>
    </li>
  </ol>
</blockquote>

<h2 id="examples">Examples</h2>

<ol>
  <li>
    <p><strong>Displaying annotated vs. UNannotated features</strong>: visualize annotated (= mapped) and unannotated (= unmapped) features by switching between activating only the “Annotated_features.consensusxml” or only the “UNannotated_features.consensusxml” layer</p>

    <figure id="figure-4" style="max-width: 90%;"><img src="../../images/protein-quant-sil_example_featureAnnotation.png" alt="displaying feature annotations. " width="2496" height="565" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_example_featureAnnotation.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> The feature below was mapped to and annotated with a peptide identification, the feature above was not mapped. A) Annotated layer is active. B) UNannotated layer is active.</figcaption></figure>
  </li>
  <li>
    <p><strong>Correct mapping</strong>: a feature was detected, a peptide was identified and the two were mapped.</p>

    <figure id="figure-5" style="max-width: 90%;"><img src="../../images/protein-quant-sil_example_highAbundant.png" alt="Correct mapping of a high-abundant peptide. " width="1772" height="584" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_example_highAbundant.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> A high-abundant peptide. A) 2D View. B) 3D View.</figcaption></figure>
    <figure id="figure-6" style="max-width: 90%;"><img src="../../images/protein-quant-sil_example_lowAbundant.png" alt="Correct mapping of a low-abundant peptide. " width="1802" height="611" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_example_lowAbundant.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> A low-abundant peptide. A) 2D View. B) 3D View.</figcaption></figure>
  </li>
  <li>
    <p><strong>No feature detected for a contaminant.</strong> Contaminants are often not labelled, but occur only in their unlabelled isoform. Therefore, they do not give rise to a consensus feature in FeatureFinderMultiplex.</p>

    <figure id="figure-7" style="max-width: 90%;"><img src="../../images/protein-quant-sil_example_contaminant.png" alt="contaminant. " width="1893" height="591" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_example_contaminant.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> A contaminant. There are no elution peaks for the heavy labelled peptide isoform. A) 2D View. B) 3D View.</figcaption></figure>
  </li>
</ol>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-check-a-possible-contaminant"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Check a possible contaminant</div>
  <ol>
    <li>Run <strong>TextExporter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <strong>IDFilter</strong> output file</li>
      </ul>
    </li>
    <li>Run <strong>Search in textfiles (grep)</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span><em>“Select lines from”</em>: output of <strong>TextExporter</strong></li>
        <li><em>“Regular Expression”</em>: the peptide sequence (e.g. <code class="language-plaintext highlighter-rouge">MFLSFPTTK</code>)</li>
      </ul>
    </li>
    <li>Check if the peptide was mapped to a protein marked with “CONTAMINANT”.</li>
  </ol>
</blockquote>

<h2 id="typical-problems">Typical Problems</h2>

<p>Three problems typically hamper correct peptide mapping:</p>

<ol>
  <li>
    <p><strong>A feature is detected, but no peptide identification is nearby.</strong></p>

    <ul>
      <li><em>Possible cause</em>: This may be caused by imperfect peptide identification. However, it is never expected that every single MS2-spectrum leads to an identification. The protein might be missing in the database, or the peptide may carry a modification that was not included in the search.</li>
      <li><em>Possible solution</em>: Improve your search engine settings.</li>
    </ul>

    <figure id="figure-8" style="max-width: 90%;"><img src="../../images/protein-quant-sil_problem_noID.png" alt="noID. " width="1267" height="692" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_problem_noID.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span> Several features without peptide IDs. Several MS2 spectra were generated (fragment scan precursors marked with red circles), but did not lead to a peptide identification.</figcaption></figure>
  </li>
  <li>
    <p><strong>A peptide was identified, but no feature is nearby.</strong></p>

    <ul>
      <li><em>Possible cause</em>:
        <ol>
          <li>The elution peaks of the peptide may be distorted. This is typical for low intensity peptides. If a lot of peptides have distorted elution peaks this may be a sign of spray instability.</li>
          <li>The peptide is a contaminant.</li>
        </ol>
      </li>
      <li><em>Possible solution</em>:
        <ol>
          <li>Lower the FeatureFinderMultiplex parameters <strong>Two peptides in a multiplet are expected to have the same isotopic pattern</strong> and/or <strong>The isotopic pattern of a peptide should resemble the averagine model at this m/z position</strong> or broaden the <strong>Range of isotopes per peptide in the sample</strong> (in <strong>Advanced Options</strong>).</li>
          <li>No optimization of parameters is needed (see <a href="#examples">example above</a>)</li>
        </ol>
      </li>
    </ul>

    <figure id="figure-9" style="max-width: 90%;"><img src="../../images/protein-quant-sil_problem_noFeature.png" alt="noID. " width="1793" height="586" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_problem_noFeature.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> A peptide was identified in its unlabelled and in its labelled isoform, but no feature was detected. A) 2D View. No third isotopic peak was detected for the labelled peptide. B) 3D View.</figcaption></figure>
  </li>
  <li>
    <p><strong>A peptide was identified and a feature was detected nearby, but the two are not mapped to each other.</strong></p>

    <ul>
      <li><em>Possible cause</em>:
        <ol>
          <li>The MS2 event and the feature are too far apart to be mapped.</li>
          <li>The precursor of the MS2 was not correctly assigned to the mono-isotopic peak.</li>
          <li>The detected feature is too small in RT dimension and covers only a part of the peptide peaks.</li>
        </ol>
      </li>
      <li><em>Possible solution</em>:
        <ol>
          <li>Increase the IDMapper parameter <strong>RT tolerance (in seconds) for the matching of peptide identifications and (consensus) features</strong>.</li>
          <li>Feature size in RT dimension cannot be directly corrected, use solution 1 instead.</li>
        </ol>
      </li>
    </ul>

    <figure id="figure-10" style="max-width: 90%;"><img src="../../images/protein-quant-sil_problem_noMapping.png" alt="noMapping. " width="1833" height="589" loading="lazy" /><a target="_blank" href="../../images/protein-quant-sil_problem_noMapping.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> A feature was detected, but the RT dimension is shorter than the peptides's elution peak. Also, the 2nd isotopic peak was fragmented and was not corrected due to the short feature. A) 2D View. B) 3D View.</figcaption></figure>
  </li>
</ol>

<p>Two problems typically disturb correct peptide quantitation:</p>

<ol>
  <li><strong>A peptide is mapped to the wrong feature.</strong>
    <ul>
      <li><em>Possible cause</em>: Co-eluting peptides of a similar mass may be falsely mapped to a nearby feature, if the correct peptide did not lead to an identification or was identified only with a low score. In <em>high-resolution data</em>, this problem of is very limited. Co-eluting peptides can normally be distinguished by slightly different m/z values. In <em>low-resolution data</em>, wrong assignment may occur more often.</li>
      <li><em>Possible solution</em>: If a high value is used for the precursor mass tolerance, try to keep the RT tolerance low to avoid false mapping.</li>
    </ul>
  </li>
  <li><strong>Background noise (1) or co-eluting peptides (2) are incorporated in a feature.</strong>
    <ul>
      <li><em>Possible solution</em>:
        <ol>
          <li>Use noise-filtering either during pre-processing or by increasing the FeatureFinderMultiplex parameter <strong>Lower bound for the intensity of isotopic peaks</strong></li>
          <li>Reduce the FeatureFinderMultiplex parameter <strong>m/z tolerance for search of peak patterns</strong>.</li>
        </ol>
      </li>
    </ul>
  </li>
</ol>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <ol>
    <li>How many peptides could not be mapped to MS1 features? (Click on the <strong>IDMapper</strong> output and look at the tool’s infobox.)</li>
    <li>How many features could not be mapped to a peptide identification? (Click on the <strong>ProteinQuantifier</strong> output and look at the tool’s infobox.)</li>
    <li>Which problems are most prominent in the test dataset?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>1,395 peptide IDs could not be mapped to a feature.</li>
      <li>1,898 features, corresponding to 949 consensus features could not be mapped to a peptide identification.</li>
      <li>The mapping of peptide IDs to features seems to have worked mostly fine. The main problems seem to be (1) missing peptide identifications, (2) missing features where a peptide was identified and (3) features that span a shorter RT range than the corresponding peptide’s elution peak.</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="optimization-of-quantitation-results">Optimization of Quantitation Results</h2>

<p>For optimization, it is critical to modify <strong>only one parameter at a time</strong>.
Also, it is recommended to optimize the tools in the order of their position in the workflow.</p>

<p>In the test dataset, several peptides were identified, but not quantified. Some of the peptides were even identified in the unlabelled, as well as in the labelled form. To optimize the feature detection, we will relax the parameters of FeatureFinderMultiplex.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-optimize-feature-detection"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Optimize Feature Detection</div>

  <ol>
    <li>Run the whole workflow again:
      <ul>
        <li>Change the <strong>FeatureFinderMultiplex</strong> parameter <em>“Range of isotopes per peptide in the sample”</em> from <code class="language-plaintext highlighter-rouge">3:6</code> to <code class="language-plaintext highlighter-rouge">2:6</code>.</li>
      </ul>
    </li>
    <li>Integrate the <strong>HighResPrecursorMassCorrector</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> into the workflow.
      <ul>
        <li>Use the mzML file and the featureXML from the <strong>FeatureFinderMultiplex</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> as input</li>
        <li>Set <strong>Additional retention time tolerance added to feature boundaries</strong> between <code class="language-plaintext highlighter-rouge">0.0</code> and <code class="language-plaintext highlighter-rouge">10.0</code></li>
        <li>Connect the output with the search engine.</li>
      </ul>
    </li>
    <li>Compare the number of identified proteins, unmatched features and unmapped peptides for each parameter setting.</li>
    <li>Visualize the results with TOPPView to check for correct feature detection and feature-to-peptide mapping.</li>
  </ol>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-sending-workflow-results-to-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sending-workflow-results-to-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Sending workflow results to new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>When running a workflow, you may send the results to a new history. This helps keeping track of different parameter settings.</p>
  </blockquote>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Which parameter improved the number of quantified proteins?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>Both changes led to more quantified proteins. Increasing the isotope range led to 26 \% more protein quantitations, increasing the RT tolerance led to 7 \% more protein quantitations.</p>
  </blockquote>
</blockquote>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Peptides labelled with stable isotopes result in multiple parallel MS1 ion traces.</p>
</li>
                        
                        <li><p>MS1 features can be used for relative protein quantitation.</p>
</li>
                        
                        <li><p>Quantitations have to be mapped to PSMs.</p>
</li>
                        
                        <li><p>PSM quantitations can be used to calculate protein quantitations.</p>
</li>
                        
                        <li><p>Proper quantitation and mapping needs careful evaluation and optimization.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/proteomics/faqs/">FAQ page for the Proteomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

		<a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=proteomics/protein-quant-sil" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Florian Christoph Sigloch, Björn Grüning, Matthias Fahrner,  <b>Peptide and Protein Quantification via Stable Isotope Labelling (SIL) (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html">https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{proteomics-protein-quant-sil,
author = "Florian Christoph Sigloch and Björn Grüning and Matthias Fahrner",
	title = "Peptide and Protein Quantification via Stable Isotope Labelling (SIL) (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/protein-quant-sil/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/proteomics/tutorials/protein-quant-sil/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: histogram
  owner: devteam
  revisions: 6f134426c2b0
  tool_panel_section_label: Graph/Display Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_falsediscoveryrate
  owner: galaxyp
  revisions: c9eff69a7c62
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_featurefindermultiplex
  owner: galaxyp
  revisions: 5e948629bc73
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_fidoadapter
  owner: galaxyp
  revisions: 658494a2cb00
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_fileinfo
  owner: galaxyp
  revisions: 52ba9695e752
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_idconflictresolver
  owner: galaxyp
  revisions: c28f13cab67a
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_idmapper
  owner: galaxyp
  revisions: 39caaa778d2e
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_idposteriorerrorprobability
  owner: galaxyp
  revisions: 111c3a00136d
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_idscoreswitcher
  owner: galaxyp
  revisions: be74a8cd7d66
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_multiplexresolver
  owner: galaxyp
  revisions: fb647b7cf2c1
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_peptideindexer
  owner: galaxyp
  revisions: dd173849c3d8
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_proteinquantifier
  owner: galaxyp
  revisions: 16026cdb1330
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_textexporter
  owner: galaxyp
  revisions: 9020199dba0a
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openms_xtandemadapter
  owner: galaxyp
  revisions: 7f6a1f27553e
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   

		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
		    
		</blockquote>


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/0fde6e77192bb0a1f0dac3e8539204bd58b7263c">0fde6e7</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00230">https://gxy.io/GTN:T00230</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/protein-quant-sil/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


        <script src='/training-material/assets/js/bundle.80a5488c.js?v=1708530376'></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        
    </body>
</html>