<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Genome annotation with Funannotate</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/fontawesome/css/all.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/">

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Genome annotation with Funannotate">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-04-26 14:23:27 +0000">
<meta name="DC.creator" content="Anthony Bretaudeau">

        
        
        
        
        <meta name="description" content="Genome annotation is a multi-level process that includes ...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Genome annotation with Funannotate">
        <meta property="og:description" content="Genome annotation is a multi-level process that includes ...">
        
        <meta property="og:image" content="/training-material/assets/images/gga.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-04-26 14:23:27 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Genome annotation with Funannotate",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Genome Annotation",
    "description": "Genome annotation is a multi-level process that includes prediction of protein-coding\ngenes, as well as other functional genome units such as structural RNAs, tRNAs,\nsmall RNAs, pseudogenes, control regions, direct and inverted repeats, insertion\nsequences, transposons and other mobile elements.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Genome annotation with Funannotate' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html",
  "timeRequired": "PT8H",
  "keywords": "genome-annotation, eukaryote",
  "description": "The questions this  addresses are:\n - How to annotate an eukaryotic genome with Funannotate?\n - How to perform functional annotation?\n - How to evaluate and visualize annotated genomic features?\n - How to format the annotation for submission at NCBI?\n\n\nThe objectives are:\n - Load genome into Galaxy\n - Annotate genome with Funannotate\n - Perform functional annotation using EggNOG-mapper and InterProScan\n - Evaluate annotation quality with BUSCO\n - View annotations in JBrowse\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html",
      "name": "Masking repeats with RepeatMasker",
      "description": "Hands-on for 'Masking repeats with RepeatMasker' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Genome Annotation",
      "description": "Genome annotation is a multi-level process that includes prediction of protein-coding\ngenes, as well as other functional genome units such as structural RNAs, tRNAs,\nsmall RNAs, pseudogenes, control regions, direct and inverted repeats, insertion\nsequences, transposons and other mobile elements.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0362",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0362",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0362"
    }
  ],
  "educationalLevel": "Intermediate"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc">
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/genome-annotation" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Genome Annotation
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/genome-annotation/faqs/" title="Check our FAQs for the Genome Annotation topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/genome-annotation/tutorials/funannotate/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                    <button class="btn" data-theme="default">Default</button>
                    <button class="btn" data-theme="night">Night</button>
                    <button class="btn" data-theme="midnight">Midnight</button>
                    <button class="btn" data-theme="rainbow">Rainbow</button>
                    <button class="btn" data-theme="blm">‚úäüèø</button>
                    <button class="btn" data-theme="progress" >üè≥Ô∏è‚Äçüåà</button>
                    <button class="btn" data-theme="halloween">üéÉ</button>
                    <button class="btn" data-theme="straya">üá¶üá∫</button>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-05-02/topics/genome-annotation/tutorials/funannotate/tutorial.html" title="Version 2023-05-02">2023-05-02</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-05-01/topics/genome-annotation/tutorials/funannotate/tutorial.html" title="Version 2023-05-01">2023-05-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-04-01/topics/genome-annotation/tutorials/funannotate/tutorial.html" title="Version 2023-04-01">2023-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        















<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'galaxy-genome-annotation/Lobby'
  };
</script>


<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-04-26 14:23:27 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Genome annotation with Funannotate",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Genome Annotation",
    "description": "Genome annotation is a multi-level process that includes prediction of protein-coding\ngenes, as well as other functional genome units such as structural RNAs, tRNAs,\nsmall RNAs, pseudogenes, control regions, direct and inverted repeats, insertion\nsequences, transposons and other mobile elements.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Genome annotation with Funannotate' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html",
  "timeRequired": "PT8H",
  "keywords": "genome-annotation, eukaryote",
  "description": "The questions this  addresses are:\n - How to annotate an eukaryotic genome with Funannotate?\n - How to perform functional annotation?\n - How to evaluate and visualize annotated genomic features?\n - How to format the annotation for submission at NCBI?\n\n\nThe objectives are:\n - Load genome into Galaxy\n - Annotate genome with Funannotate\n - Perform functional annotation using EggNOG-mapper and InterProScan\n - Evaluate annotation quality with BUSCO\n - View annotations in JBrowse\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html",
      "name": "Masking repeats with RepeatMasker",
      "description": "Hands-on for 'Masking repeats with RepeatMasker' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Genome Annotation",
      "description": "Genome annotation is a multi-level process that includes prediction of protein-coding\ngenes, as well as other functional genome units such as structural RNAs, tRNAs,\nsmall RNAs, pseudogenes, control regions, direct and inverted repeats, insertion\nsequences, transposons and other mobile elements.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/genome-annotation/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0362",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0362",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0362"
    }
  ],
  "educationalLevel": "Intermediate"
}
</script>

<article class="tutorial topic-genome-annotation">
    <h1 data-toc-skip>Genome annotation with Funannotate</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/abretaud?s=27" alt="Avatar">Anthony Bretaudeau</a>
		</td>
	</tr>
	


	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/alexcorm/" class="contributor-badge contributor-alexcorm"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/alexcorm?s=27" alt="Avatar">Alexandre Cormier</a><a href="/training-material/hall-of-fame/lleroi/" class="contributor-badge contributor-lleroi"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/lleroi?s=27" alt="Avatar">Laura Leroi</a><a href="/training-material/hall-of-fame/r1corre/" class="contributor-badge contributor-r1corre"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/r1corre?s=27" alt="Avatar">Erwan Corre</a><a href="/training-material/hall-of-fame/stephanierobin/" class="contributor-badge contributor-stephanierobin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/stephanierobin?s=27" alt="Avatar">St√©phanie Robin</a></td>
	</tr>
	

	

	

	
</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to annotate an eukaryotic genome with Funannotate?</p>
</li>
        
        <li><p>How to perform functional annotation?</p>
</li>
        
        <li><p>How to evaluate and visualize annotated genomic features?</p>
</li>
        
        <li><p>How to format the annotation for submission at NCBI?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Load genome into Galaxy</p>
</li>
        
        <li><p>Annotate genome with Funannotate</p>
</li>
        
        <li><p>Perform functional annotation using EggNOG-mapper and InterProScan</p>
</li>
        
        <li><p>Evaluate annotation quality with BUSCO</p>
</li>
        
        <li><p>View annotations in JBrowse</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
    <li>
    
        
        
        <a href="/training-material/topics/genome-annotation">Genome Annotation</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                            
                            <li> Masking repeats with RepeatMasker:
                            
                            
                                
                                     <a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 8 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/7867921">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/genome-annotation/tutorials/funannotate/workflows/" title="Genome annotation with Funannotate workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/genome-annotation/tutorials/funannotate/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  
<!-- GTN Video Library true -->
  
<li class="btn btn-default supporting_material">
  
    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this tutorial in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
    </a>
    <ul class="dropdown-menu">
        
        

        
        
        

        <li>
            <a class="dropdown-item" href="https://gallantries.github.io/video-library/videos/genome-annotation/funannotate/tutorial" title="View GTN Video Library for this Tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (March 2022)
            </a>
        </li>
        
        
        
        
    </ul>
  
</li>
  
   



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/genome-annotation/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Apr 26, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
        
        <div><strong>Short Link:</strong> <a href="https://gxy.io/GTN:T00172">https://gxy.io/GTN:T00172</a> </div>
        
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<p>Genome annotation of eukaryotes is a little more complicated than for prokaryotes: eukaryotic genomes are usually larger than prokaryotes, with more genes. The sequences determining the beginning and the end of a gene are generally less conserved than the prokaryotic ones. Many genes also contain introns, and the limits of these introns (acceptor and donor sites) are not highly conserved.</p>

<p>In this tutorial we will use a software tool called Funannotate (<span class="citation"><a href="#jonathan_m_palmer_2020_4054262">Palmer and Stajich 2020</a></span>) to annotate the genome sequence of a small eukaryote: <a href="https://en.wikipedia.org/wiki/Mucor_mucedo"><em>Mucor mucedo</em></a> (a fungal plant pathogen).</p>

<p>As explained on <a href="https://funannotate.readthedocs.io/">Funannotate‚Äôs website</a>, ‚Äúit was originally written to annotate fungal genomes (small eukaryotes ~ 30 Mb genomes), but has evolved over time to accomodate larger genomes‚Äù. As other annotation tools like <a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">Maker</a> (<span class="citation"><a href="#Campbell2014">Campbell <i>et al.</i> 2014</a></span>) or Braker (<span class="citation"><a href="#braker">Br≈Øna <i>et al.</i> 2021</a></span>), it works by aligning as many evidences as possible along the genome sequence, and then reconciliating all these signals to determine probable gene structures.</p>

<p>The evidences can be transcript or protein sequences from the same (or closely related) organism. These sequences can come from public databases (like NR or GenBank) or from your own experimental data (transcriptome assembly from an RNASeq experiment for example). Funannotate is also able to take into account repeated elements.</p>

<p>Funannotate uses ab-initio predictors (<a href="http://bioinf.uni-greifswald.de/augustus/">Augustus</a>, <a href="https://github.com/KorfLab/SNAP">SNAP</a>, <a href="https://bio.tools/glimmer-hmm">glimmerHMM</a>, <a href="https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-015-1344-4">CodingQuarry</a> and <a href="http://exon.gatech.edu/GeneMark/">GeneMark-ES/ET</a> (optional due to licensing)) to improve its predictions: these software tools are able to make gene structure predictions by analysing only the genome sequence with a statistical model.</p>

<p>While for <a href="/training-material/topics/genome-annotation/tutorials/annotation-with-maker/tutorial.html">Maker</a> you need to perform training steps for the ab-initio predictors, Funannotate is able to take care of that for you, which makes it much easier to use.</p>

<p>In this tutorial, you will learn how to perform a structural genome annotation, and how to evaluate its quality. Then, you will learn how to run functional annotation, using EggNOG-mapper and InterProScan to automatically assign names and functions to the annotated genes. And you will also learn how Funannotate can prepare files ready for submission of your annotation to the NCBI. Finally, you will learn how to use the <a href="http://jbrowse.org/">JBrowse</a> genome browser to visualise your new annotation.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
  <li><a href="#preparing-rnaseq-data" id="markdown-toc-preparing-rnaseq-data">Preparing RNASeq data</a></li>
  <li><a href="#structural-annotation" id="markdown-toc-structural-annotation">Structural annotation</a></li>
  <li><a href="#functional-annotation" id="markdown-toc-functional-annotation">Functional annotation</a>    <ol>
      <li><a href="#eggnog-mapper" id="markdown-toc-eggnog-mapper">EggNOG Mapper</a></li>
      <li><a href="#interproscan" id="markdown-toc-interproscan">InterProScan</a></li>
    </ol>
  </li>
  <li><a href="#submission-to-ncbi" id="markdown-toc-submission-to-ncbi">Submission to NCBI</a></li>
  <li><a href="#integrating-structural-and-functional-annotation" id="markdown-toc-integrating-structural-and-functional-annotation">Integrating structural and functional annotation</a>    <ol>
      <li><a href="#evaluation-with-busco" id="markdown-toc-evaluation-with-busco">Evaluation with <strong>Busco</strong></a></li>
    </ol>
  </li>
  <li><a href="#visualisation-with-a-genome-browser" id="markdown-toc-visualisation-with-a-genome-browser">Visualisation with a genome browser</a></li>
  <li><a href="#comparing-annotations" id="markdown-toc-comparing-annotations">Comparing annotations</a>    <ol>
      <li><a href="#comparing-with-aegean-parseval" id="markdown-toc-comparing-with-aegean-parseval">Comparing with <strong>AEGeAn ParsEval</strong></a></li>
      <li><a href="#comparing-with-funannotate-compare" id="markdown-toc-comparing-with-funannotate-compare">Comparing with <strong>Funannotate compare</strong></a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-upload">Data upload</h1>

<p>To annotate our genome using Funannotate, we will use the following files:</p>

<ul>
  <li>The <strong>genome sequence</strong> in fasta format. For best results, the sequence should be soft-masked beforehand. You can learn how to do it by following the <a href="/training-material/topics/genome-annotation/tutorials/repeatmasker/tutorial.html">RepeatMasker tutorial</a>. For this tutorial, we will try to annotate the genome assembled in the <a href="/training-material/topics/assembly/tutorials/flye-assembly/tutorial.html">Flye assembly tutorial</a>.</li>
  <li>Some RNASeq data in fastq format. We will align them on the genome, and Funannotate will use it as evidence to annotate genes.</li>
  <li>A set of <strong>protein sequences</strong>, like UniProt/SwissProt. It is important to have good quality, curated sequences here, that‚Äôs why, by default, Funannotate will use the UniProt/SwissProt databank. In this tutorial, we have prepared a subset of this databank to speed up computing, but you should use UniProt/SwissProt for real life analysis.</li>
</ul>

<p>Funannotate will take into account the position of mapped RNASeq reads, and the alignment of protein sequences on the genome sequence to determine gene positions.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
    </li>
    <li>
      <p>Import the files from <a href="https://zenodo.org/record/7867921">Zenodo</a> or from
the shared data library (<code class="language-plaintext highlighter-rouge">GTN - Material</code> -&gt; <code class="language-plaintext highlighter-rouge">genome-annotation</code>
 -&gt; <code class="language-plaintext highlighter-rouge">Genome annotation with Funannotate</code>):</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/api/files/47406781-e8af-42e7-855d-d29e4a098f6f/genome_masked.fasta
https://zenodo.org/api/files/47406781-e8af-42e7-855d-d29e4a098f6f/rnaseq_R1.fq.gz
https://zenodo.org/api/files/47406781-e8af-42e7-855d-d29e4a098f6f/rnaseq_R2.fq.gz
https://zenodo.org/api/files/47406781-e8af-42e7-855d-d29e4a098f6f/SwissProt_subset.fasta
https://zenodo.org/api/files/47406781-e8af-42e7-855d-d29e4a098f6f/alternate_annotation.gbk
https://zenodo.org/api/files/47406781-e8af-42e7-855d-d29e4a098f6f/alternate_annotation.gff3
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Open the Galaxy Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>   </li>   <li>Select <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ul>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor</li>   <li>Select the desired files</li>   <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>   <li>Click on <strong>Import</strong></li> </ul> </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="preparing-rnaseq-data">Preparing RNASeq data</h1>

<p>When you sequence a new genome, you usually sequence a few libraries of RNASeq data, from different tissues and in different conditions, because this data will help you in annotating the genome. Here, we are using data from one RNASeq dataset that is available on <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive (SRA)</a>: <a href="https://www.ncbi.nlm.nih.gov/sra/?term=SRR8534859">SRR8534859</a>.</p>

<p>You would normally get the Fastq files directly from SRA and use them in the following step. To speed up the tutorial (without impairing too much the quality of the results), we have reduced the size of the dataset into a single pair of (smaller) fastq files, available from Zenodo (or the GTN Data Libraries).</p>

<p>To make use of this RNASeq data, we need to map it on the genome. We will use <strong>RNA star</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and get a result in the form of a BAM file, that we will use in the rest of the tutorial.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy0" title="RNA STAR tool"><strong>RNA STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.8a+galaxy0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSingle-end or paired-end reads‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end (as individual datasets)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRNA-Seq FASTQ/FASTA file, forward reads‚Äù</em>: <code class="language-plaintext highlighter-rouge">rnaseq_R1.fq.gz</code> (Input dataset)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRNA-Seq FASTQ/FASTA file, reverse reads‚Äù</em>: <code class="language-plaintext highlighter-rouge">rnaseq_R2.fq.gz</code> (Input dataset)</li>
          </ul>
        </li>
        <li><em>‚ÄúCustom or built-in reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use reference genome from history and create temporary index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect a reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome_masked.fasta</code> (Input dataset)</li>
            <li><em>‚ÄúLength of the SA pre-indexing string‚Äù</em>: <code class="language-plaintext highlighter-rouge">11</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We select <code class="language-plaintext highlighter-rouge">11</code> for the <em>‚ÄúLength of the SA pre-indexing string‚Äù</em> parameter as it is the recommended value for a small genome of this size. If you select 14 (the default value), STAR will advise you to use <code class="language-plaintext highlighter-rouge">11</code> instead in its logs.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-if-i-want-to-use-multiple-rnaseq-libraries"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: What if I want to use multiple RNASeq libraries</div>

  <p>To get the best possible annotation, it is adviced to use multiple RNASeq libraries, sequences from different tissues in different conditions. To use them, you can map each one individually using STAR, just as in this tutorial. You will get one BAM file per RNASeq library, and you can then easily merge them into a single BAM file by using the <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/2.18.2.1" title="MergeSamFiles tool"><strong>MergeSamFiles</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/2.18.2.1</span></span>  tool. This single BAM file can then be used by Funannotate like we do in the next steps.</p>
</blockquote>

<p>Before we move on to the next step, we need to make sure that the mapping went well. Have a look at the <code class="language-plaintext highlighter-rouge">log</code> output of <code class="language-plaintext highlighter-rouge">RNA STAR</code>.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>

  <p>What proportion of reads were correctly mapped to the genome? Do you think it is enough to continue with this tutorial?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Look for <code class="language-plaintext highlighter-rouge">Uniquely mapped reads %</code>: you should get ~96%. This is a very good score, because while reducing the size of the dataset for this tutorial, we have kept mostly reads properly mapping.
Anyway, with real data, you expect to have a vast majority of reads mapping uniquely on the genome. If it‚Äôs not the case, check that you‚Äôre using the correct RNASeq files, with the correct genome sequence.</p>

  </blockquote>

</blockquote>

<h1 id="structural-annotation">Structural annotation</h1>

<p>We can now run <strong>Funannotate predict annotation</strong> to perform the structural annotation of the genome.</p>

<p>We need to input the genome sequence, the mapped RNASeq data, and the proteins to align on the genome. We also specify the name of the species and strain (they can be used later for submission to NCBI).</p>

<p>There are other parameters to finely tune how Funannotate will run ab-initio predictors to predict genes, and to filter the final results based on various criteria. As Funannotate uses <a href="http://busco.ezlab.org/">BUSCO</a> (Benchmarking Universal Single-Copy Orthologs) for initial training of ab-initio predictors, we select datasets close to the species we are annotating: <code class="language-plaintext highlighter-rouge">mucorales (orthodb 10)</code> and <code class="language-plaintext highlighter-rouge">rhizopus_oryzae</code>.</p>

<p>Funannotate is also able to use GeneMark to predict new genes, but to due to licensing restrictions, this software is not available on every Galaxy instance. We will ignore this for this tutorial, it will not impact the results too much.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/funannotate_predict/funannotate_predict/1.8.15+galaxy1" title="Funannotate predict annotation tool"><strong>Funannotate predict annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/funannotate_predict/funannotate_predict/1.8.15+galaxy1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAssembly to annotate‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome_masked.fasta</code> (Input dataset)</li>
        <li><em>‚ÄúFunannotate database‚Äù</em>: select the latest version available</li>
        <li>In <em>‚ÄúOrganism‚Äù</em>:
          <ul>
            <li><em>‚ÄúName of the species to annotate‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mucor mucedo</code></li>
            <li><em>‚ÄúStrain name‚Äù</em>: <code class="language-plaintext highlighter-rouge">muc1</code></li>
            <li><em>‚ÄúIs it a fungus species?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúEvidences‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRNA-seq mapped to genome to train Augustus/GeneMark-ET‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> (output of <strong>RNA STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><em>‚ÄúSelect protein evidences‚Äù</em>: <code class="language-plaintext highlighter-rouge">Custom protein sequences</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProteins to map to genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">SwissProt_subset.fasta</code> (Input dataset)</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúBusco‚Äù</em>:
          <ul>
            <li><em>BUSCO models to align‚Äù</em>‚Äù: <code class="language-plaintext highlighter-rouge">mucorales (orthodb 10)</code></li>
            <li><em>‚ÄúInitial Augustus species training set for BUSCO alignment‚Äù</em>: <code class="language-plaintext highlighter-rouge">rhizopus_oryzae</code></li>
          </ul>
        </li>
        <li><em>‚ÄúWhich outputs should be generated‚Äù</em>: Select all</li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-on-parameters"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: on parameters</div>

  <ul>
    <li>For <em>‚ÄúSelect protein evidences‚Äù</em> we select <code class="language-plaintext highlighter-rouge">Custom protein sequences</code> to reduce the computing time, but for real data analysis, you should select the default value: <code class="language-plaintext highlighter-rouge">Use UniProtKb/SwissProt (from selected Funannotate database)</code>.</li>
    <li>It is possible to enable the <em>‚ÄúIs it a fungus species?‚Äù</em> option in Funannotate: it launches an additional ab initio predictor (CodingQuerry) dedicated to fungi genomes. However it has proved to be unstable on the genome studied in this tutorial, and it can create a lot of fragmented gene models depending on the RNASeq data available. For this tutorial we leave this option to <code class="language-plaintext highlighter-rouge">No</code>. You can test it with real data, but be sure to compare the result with and without this option.</li>
    <li>For real data analysis you can consider enabling the <em>‚ÄúAugustus settings (advanced)‚Äù</em> &gt; <em>‚ÄúRun ‚Äòoptimize_augustus.pl‚Äô to refine training (long runtime)‚Äù</em>. If you have enough data, you might get better results as there will be an additional training step for augustus (at the cost of a longer runtime).</li>
  </ul>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-don-t-wait"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Don't wait</div>

  <p>This step will take a bit of time to run. While it runs, we can already schedule the following functional annotation steps. Galaxy will run them automatically as soon as the structural annotation is ready.</p>
</blockquote>

<p>This tool produces several output dataset, in particular:</p>

<ul>
  <li>the full structural annotation in Genbank, GFF3 or NCBI tbl formats: these files contain the position of all the genes that were found on the genome.</li>
  <li>the CDS, transcript and protein sequences of all the genes predicted by Funannotate (fasta files)</li>
  <li>some statistics and reports</li>
</ul>

<p>Let‚Äôs have a closer look at the output of our annotation. First display the <code class="language-plaintext highlighter-rouge">stats</code> dataset: the first par of the file contains some information on how funannotate was launched. If you go to the bottom, you‚Äôll find a few interesting numbers in the <code class="language-plaintext highlighter-rouge">annotation</code> section:</p>

<ul>
  <li>the total number of genes and mRNA</li>
  <li>the average length of genes, exons, proteins</li>
  <li>the number of single/multiple exon transcripts</li>
</ul>

<p>These number alone are interesting, but not fully informative on the quality of the annotation. For example, for the number of genes, you want a ‚Äògood‚Äô number based on what you expect for this species (a too big number can mean that genes are fragmented, and a too small number can mean that some genes were not annotated at all). These numbers can help when comparing an annotation with other ones performed with other parameters or tools. You can also compare these numbers with annotation of closely related species.</p>

<p>To get a better picture of the quality of the result, we will run BUSCO in the next step.</p>

<p>Before moving on, have a quick look at the <code class="language-plaintext highlighter-rouge">tbl2asn error summary report</code> output: it lists a few potential problems that were identified by Funannotate in the results it generated. For example, Funannotate can tell you when it predicted genes that contain very short exons, or that use a rare splice site sequence. You can have a detailed list of identified potential problems in the <code class="language-plaintext highlighter-rouge">tbl2asn genome validation report</code> dataset. It does not mean that each listed gene is wrong, but it means that you might want to give a closer look at these ones. If you have time to manually check each gene, Apollo can help you in doing this, see the note in the conclusion for this.</p>

<h1 id="functional-annotation">Functional annotation</h1>

<h2 id="eggnog-mapper">EggNOG Mapper</h2>

<p><strong>EggNOG Mapper</strong> compares each protein sequence of the annotation to a huge set of ortholog groups from the <a href="http://eggnog5.embl.de">EggNOG database</a>. In this database, each ortholog group is associated with functional annotation like <a href="http://www.geneontology.org/">Gene Ontology (GO)</a> terms or <a href="https://www.genome.jp/kegg/pathway.html">KEGG pathways</a>. When the protein sequence of a new gene is found to be very similar to one of these ortholog groups, the corresponding functional annotation is transfered to this new gene.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-8"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/eggnog_mapper/eggnog_mapper/2.1.8+galaxy3" title="eggNOG Mapper tool"><strong>eggNOG Mapper</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/galaxyp/eggnog_mapper/eggnog_mapper/2.1.8+galaxy3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFasta sequences to annotate‚Äù</em>: <code class="language-plaintext highlighter-rouge">protein sequences</code> (output of <strong>Funannotate predict annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)‚Äù</li>
        <li><em>‚ÄúVersion of eggNOG Database‚Äù</em>: select the latest version available</li>
        <li>In <em>‚ÄúOutput Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúExclude header lines and stats from output files‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output of this tool is a tabular file, where each line represents a gene from our annotation, with the functional annotation that was found by EggNOG-mapper. It includes a predicted protein name, GO terms, EC numbers, KEGG identifiers, ‚Ä¶</p>

<p>Display the file and explore which kind of identifiers were found by EggNOG Mapper.</p>

<h2 id="interproscan">InterProScan</h2>

<p><a href="https://www.ebi.ac.uk/interpro/">InterPro</a> is a huge integrated database of protein families. Each family is characterized by one or muliple signatures (i.e. sequence motifs) that are specific to the protein family, and corresponding functional annotation like protein names or <a href="http://www.geneontology.org/">Gene Ontology (GO)</a>. A good proportion of the signatures are manually curated, which means they are of very good quality.</p>

<p><strong>InterProScan</strong> is a tool that analyses each protein sequence from our annotation to determine if they contain one or several of the signatures from InterPro. When a protein contains a known signature, the corresponding functional annotation will be assigned to it by <strong>InterProScan</strong>.</p>

<p><strong>InterProScan</strong> itself runs multiple applications to search for the signatures in the protein sequences. It is possible to select exactly which ones we want to use when launching the analysis (by default all will be run).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-9"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/interproscan/interproscan/5.59-91.0+galaxy3" title="InterProScan tool"><strong>InterProScan</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/bgruening/interproscan/interproscan/5.59-91.0+galaxy3</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProtein FASTA File‚Äù</em>: <code class="language-plaintext highlighter-rouge">protein sequences</code> (output of <strong>Funannotate predict annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)‚Äù</li>
        <li><em>‚ÄúInterProScan database‚Äù</em>: select the latest version available</li>
        <li><em>‚ÄúUse applications with restricted license, only for non-commercial use?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (set it to <code class="language-plaintext highlighter-rouge">No</code> if you run InterProScan for commercial use)</li>
        <li><em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab-separated values format (TSV)</code> and <code class="language-plaintext highlighter-rouge">XML</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

  <p>To speed up the processing by InterProScan during this tutorial, you can disable <code class="language-plaintext highlighter-rouge">Pfam</code> and <code class="language-plaintext highlighter-rouge">PANTHER</code> applications. When analysing real data, it is adviced to keep them enabled.</p>

  <p>When some applications are disabled, you will of course miss the corresponding results in the output of <strong>InterProScan</strong>.</p>
</blockquote>

<p>The output of this tool is both a tabular file and an XML file. Both contain the same information, but the tabular one is more readable for a Human: each line represents a gene from our annotation, with the different domains and motifs that were found by InterProScan.</p>

<p>If you display the TSV file you should see something like this:</p>

<p><img src="../../images/interproscan_output.png" alt="InterProScan TSV output" title="Extract of an InterProScan TSV output" /></p>

<p>Each line correspond to a motif found in one of the annotated proteins. The most interesting columns are:</p>

<ul>
  <li>Column 1: the protein identifier</li>
  <li>Column 5: the identifier of the signature that was found in the protein sequence</li>
  <li>Column 4: the databank where this signature comes from (InterProScan regroups several motifs databanks)</li>
  <li>Column 6: the human readable description of the motif</li>
  <li>Columns 7 and 8: the position where the motif was found</li>
  <li>Column 9: a score for the match (if available)</li>
  <li>Column 12 and 13: identifier of the signature integrated in InterPro (if available). Have a look an example webpage for <a href="https://www.ebi.ac.uk/interpro/entry/InterPro/IPR036859/">IPR036859</a> on InterPro.</li>
  <li>The following columns contains various identifiers that were assigned to the protein based on the match with the signature (Gene ontology term, Reactome, ‚Ä¶)</li>
</ul>

<p>The XML output file contains the same information in a computer-friendly format, we will use it in the next step.</p>

<h1 id="submission-to-ncbi">Submission to NCBI</h1>

<p>If you plan to submit the final genome sequence and annotation to NCBI, there are a few steps to follow.</p>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-please-do-not-submit-this-genome-to-ncbi"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i> Warning: Please do not submit this genome to NCBI!</div>
  <p>In this tutorial we will <strong>not</strong> perform a real submission, as we don‚Äôt want to duplicate the annotation in Genbank every time someone follows this tutorial!</p>

  <p>But here‚Äôs a description of the main steps, and how Funannotate can help you in this process. NCBI provides a complete documentation for <a href="https://www.ncbi.nlm.nih.gov/genbank/genomesubmit/">genome and annotation submission</a>.</p>
</blockquote>

<ul>
  <li>First, you should have created a BioProject and a BioSample <a href="https://submit.ncbi.nlm.nih.gov/subs/bioproject/">on the NCBI portal</a>, corresponding to your scientific project, and the sample(s) you have sequenced</li>
  <li>The raw reads used for the assembly, and the RNASeq ones should be deposited on SRA, and taggued with the BioProject and BioSample ids</li>
  <li>Ideally you should start submitting the assembly <em>before</em> performing the annotation. The reason is simple: NCBI performs some validation on the genome sequence before accepting it into GenBank. It means that you may be forced to make modifications to the genome sequence (e.g. remove contigs, split contigs where you have adapter contamination, etc). You will save time and trouble doing the annotation on a fully validated and frozen genome sequence.</li>
  <li>You should get a <code class="language-plaintext highlighter-rouge">locus_tag</code> for your genome: it is the unique prefix that is used at the beginning of each gene name. By default, Funannotate uses <code class="language-plaintext highlighter-rouge">FUN_</code> (e.g. <code class="language-plaintext highlighter-rouge">FUN_000001</code>), but to submit to NCBI you need to have a prefix specific to your genome. NCBI should provide it to you when you create your BioProject.</li>
  <li>You then need to prepare a file containing a few metadata. This can be done online on <a href="https://submit.ncbi.nlm.nih.gov/genbank/template/submission/">https://submit.ncbi.nlm.nih.gov/genbank/template/submission/</a>. You need to fill the form with some basic information, like that for example (for real data, you should of course write real information!):</li>
</ul>

<figure id="figure-1"><img src="../../images/ncbi_template.png" alt="NCBI submission template. " width="1142" height="1640" loading="lazy" /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Example of a filled form to generate an NCBI submission template.</figcaption></figure>

<p>When filled, you can click on the <strong>Create template</strong> button: you will be able to download a <code class="language-plaintext highlighter-rouge">.sbt</code> file that you should then upload to your Galaxy history.</p>

<p>The next step will show you how Funannotate can use this <code class="language-plaintext highlighter-rouge">.sbt</code> file, and the <code class="language-plaintext highlighter-rouge">locus_tag</code>, to generate the needed file for the submission of the annotation to the NCBI. These files will need to be submitted on the <a href="https://submit.ncbi.nlm.nih.gov/subs/genome/">NCBI genome submission portal</a>.</p>

<h1 id="integrating-structural-and-functional-annotation">Integrating structural and functional annotation</h1>

<p>Now we have a structural annotation, and the results of both <strong>EggNOG Mapper</strong> and <strong>InterProScan</strong>. Each one is in a separate file, so we will now combine all this data into a single file that will contain both the structural <em>and</em> the functional annotation. This will be the final output of our annotation pipeline, ready to be submitted to the NCBI reference database.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/funannotate_annotate/funannotate_annotate/1.8.15+galaxy1" title="Funannotate functional tool"><strong>Funannotate functional</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/funannotate_annotate/funannotate_annotate/1.8.15+galaxy1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">GenBank (from 'Funannotate predict annotation' tool)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGenome annotation in genbank format‚Äù</em>: <code class="language-plaintext highlighter-rouge">annotation (genbank)</code> (output of <strong>Funannotate predict annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>‚ÄúFunannotate database‚Äù</em>: select the latest version available</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúNCBI submission template file‚Äù</em>: <code class="language-plaintext highlighter-rouge">template.sbt</code> (the file downloaded from NCBI website, or leave empty if you didn‚Äôt generate it)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúEggnog-mapper annotations file‚Äù</em>: <code class="language-plaintext highlighter-rouge">annotations</code> (output of <strong>eggNOG Mapper</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInterProScan5 XML file‚Äù</em>: <code class="language-plaintext highlighter-rouge">InterProScan XML</code> (output of <strong>InterProScan</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBUSCO models‚Äù</em>: <code class="language-plaintext highlighter-rouge">mucorales (orthodb 10)</code></li>
        <li><em>‚ÄúStrain name‚Äù</em>: <code class="language-plaintext highlighter-rouge">muc1</code></li>
        <li><em>‚Äúlocus_tag from NCBI to rename GFF gene models with‚Äù</em>: <code class="language-plaintext highlighter-rouge">MMUCEDO_</code> (we consider NCBI has assigned us this <code class="language-plaintext highlighter-rouge">locus_tag</code>)</li>
        <li><em>‚ÄúWhich outputs should be generated‚Äù</em>: Select all</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This tool produces several output dataset, in particular:</p>

<ul>
  <li>The full structural and functional annotation in Genbank and GFF3 format</li>
  <li>NCBI tbl and sqn (<code class="language-plaintext highlighter-rouge">NCBI Sequin genome</code>) formats: these files can be used for submission to NCBI</li>
  <li>The CDS, transcript and protein sequences of all the genes predicted by Funannotate</li>
  <li>Some statistics and reports</li>
</ul>

<p>If you display the GFF3 output, you will notice that the functional information, including gene names, is now stored in this file.</p>

<h2 id="evaluation-with-busco">Evaluation with <strong>Busco</strong></h2>

<p>We now have a complete annotation, including functional annotation, but it‚Äôs time to evaluate the quality of this annotation. <a href="http://busco.ezlab.org/">BUSCO</a> (Benchmarking Universal Single-Copy Orthologs) is a tool allowing to evaluate the quality of a genome assembly or of a genome annotation. By comparing genomes from various more or less related species, the authors determined sets of ortholog genes that are present in single copy in (almost) all the species of a clade (Bacteria, Fungi, Plants, Insects, Mammalians, ‚Ä¶). Most of these genes are essential for the organism to live, and are expected to be found in any newly sequenced and annotated genome from the corresponding clade. Using this data, BUSCO is able to evaluate the proportion of these essential genes (also named BUSCOs) found in a set of (predicted) transcript or protein sequences. This is a good evaluation of the ‚Äúcompleteness‚Äù of the annotation.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-3"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0" title="Busco tool"><strong>Busco</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.4.6+galaxy0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSequences to analyse‚Äù</em>: <code class="language-plaintext highlighter-rouge">protein sequences</code> (output of <strong>Funannotate functional</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúMode‚Äù</em>: <code class="language-plaintext highlighter-rouge">annotated gene sets (protein)</code></li>
        <li><em>‚ÄúAuto-detect or select lineage?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Select lineage</code>
          <ul>
            <li><em>‚ÄúLineage‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mucorales</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúAdvanced Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúWhich outputs should be generated‚Äù</em>: <code class="language-plaintext highlighter-rouge">shortsummary text</code> and <code class="language-plaintext highlighter-rouge">summary image</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>

  <p>How many BUSCO genes were found complete in the annotation? Do you think the quality of the annotation is good?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>On a total of 2449, you should find ~2312 BUSCO genes identifed as complete in the annotation, with 2281 being in single copy, and 31 being duplicated.</p>

    <p>That‚Äôs a quite good result as running BUSCO on the genome itself gives a very close number (2327 Complete BUSCOs) (see Flye assembly tutorial). It means the annotation process was able to detect most of the genes it was supposed to find.</p>

    <p>To improve the result you can consider using more RNASeq data, and using the <em>‚ÄúAugustus settings (advanced)‚Äù</em> &gt; <em>‚ÄúRun ‚Äòoptimize_augustus.pl‚Äô to refine training (long runtime)‚Äù</em> option.</p>
  </blockquote>

</blockquote>

<h1 id="visualisation-with-a-genome-browser">Visualisation with a genome browser</h1>

<p>With Galaxy, you can visualize the annotation you have generated using JBrowse genome browser. This allows you to navigate along the chromosomes of the genome and see the structure of each predicted gene. We also add an RNASeq track, using the BAM file created with <strong>RNA STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-4"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1" title="JBrowse tool"><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúReference genome to display‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome_masked.fasta</code> (Input dataset)</li>
          </ul>
        </li>
        <li>In <em>‚ÄúTrack Group‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li><em>‚ÄúTrack Category‚Äù</em>: <code class="language-plaintext highlighter-rouge">Annotation</code></li>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">GFF/GFF3/BED Features</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGFF/GFF3/BED Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">gff3</code> (output of <strong>Funannotate functional</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li><em>‚ÄúTrack Category‚Äù</em>: <code class="language-plaintext highlighter-rouge">RNASeq</code></li>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM Pileups</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapped.bam</code> (output of <strong>RNA STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                            <li><em>‚ÄúAutogenerate SNP Track‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Click on the newly created dataset‚Äôs eye to display it. You will see a JBrowse genome browser. You can have a look at the <a href="/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html">JBrowse tutorial</a> for a more in-depth description of JBrowse.</p>

<p>Enable the annotation track, and the SNPs/Coverage tracks on the left side of JBrowse, then navigate along the genome. You will see the different gene models predicted by Funannotate.</p>

<p>If you click on a gene, a popup will appear with detailed information on the selected gene: position and sequence, but also the functional information retrieved from the <strong>EggNOG mapper</strong> and <strong>InterProScan</strong> results.</p>

<p>If you zoom to a specific gene, and look at the RNASeq tracks, you will see light grey regions corresponding to portions of the genomes where RNASeq were mapped, and darker grey regions corresponding to introns (= regions were some reads were found to match both the end of an exon, and the start of the next one). You can enable the other RNASeq track to display each individual read that was mapped on the genome.</p>

<p>If you navigate along the genome, you will find genes with very low RNASeq coverage: this demonstrates how Funannotate is able to predict genes not only with RNASeq, but also by comparing to protein sequences (from SwissProt) and using ab initio predictors, trained using RNASeq data.</p>

<h1 id="comparing-annotations">Comparing annotations</h1>

<p>Earlier, we have seen how general statistics and BUSCO results can help to evaluate the quality of an annotation. But when annnotating a new genome, you might want to try different annotation methods and parameters. We will see now how to compare multiple annotations between them. As an example, we will compare the annotation we have generated, with an alternate one, that you have uploaded from Zenodo at the beginning of the tutorial. Comparison is only possible for annotations performed on the same genome sequence.</p>

<h2 id="comparing-with-aegean-parseval">Comparing with <strong>AEGeAn ParsEval</strong></h2>

<p>Let‚Äôs run <strong>AEGeAn ParsEval</strong> first: it compares two annotations in GFF3 format.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-5"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/aegean_parseval/aegean_parseval/0.16.0" title="AEGeAn ParsEval tool"><strong>AEGeAn ParsEval</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/aegean_parseval/aegean_parseval/0.16.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúReference GFF3 file‚Äù</em>: <code class="language-plaintext highlighter-rouge">gff3</code> (output of <strong>Funannotate functional</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPrediction GFF3 file‚Äù</em>: <code class="language-plaintext highlighter-rouge">alternate_annotation.gff3</code> (Input dataset)</li>
        <li><em>‚ÄúSelect the output type‚Äù</em>: <code class="language-plaintext highlighter-rouge">HTML</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><strong>AEGeAn ParsEval</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> compares gene loci spread all along the genome sequence, each locus containing gene(s) on the ‚Äúreference‚Äù (=the annotation we have generated) and/or the ‚Äúprediction‚Äù (=the alternate one) annotation.</p>

<p>The output is a web page where you can see how many loci or genes are identical or different between the two annotations. By clicking on the <code class="language-plaintext highlighter-rouge">(+)</code> links, or on the scaffold names, you can see the results more in detail, even at the gene level.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-6"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li>Display the report</li>
    <li>Click on <code class="language-plaintext highlighter-rouge">scaffold_11</code></li>
    <li>Click on the first locus (between positions 637 and 7730)</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>

  <p>What can you tell from the comparison at this gene locus?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>You should see something like this:</p>

    <figure id="figure-2"><img src="../../images/aegean_scaffold_11_637-7730.png" alt="Scaffold 11 - 637..7730. " width="650" height="268" loading="lazy" /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Comparison of the two annotations on Scaffold 11 - 637..7730</figcaption></figure>

    <p>As you can see, at this position, the alternate annotation as found only one big gene, while the annotation you have generated has found 4 different genes, not all on the same strand.</p>

    <p>This can be explained by the way the alternate annotation was performed: it was done by running Funannotate without any RNASeq data, and choosing a wrong value (<code class="language-plaintext highlighter-rouge">insecta</code>) in the busco parameter. The annotation you have generated is probably the good one here.</p>
  </blockquote>

</blockquote>

<h2 id="comparing-with-funannotate-compare">Comparing with <strong>Funannotate compare</strong></h2>

<p><strong>Funannotate compare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is another tool to compare several annotations. It uses genbank files, as generated by <strong>Funannotate functional</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-7"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/funannotate_compare/funannotate_compare/1.8.15+galaxy1" title="Funannotate compare tool"><strong>Funannotate compare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: toolshed.g2.bx.psu.edu/repos/iuc/funannotate_compare/funannotate_compare/1.8.15+galaxy1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúGenome annotations in genbank format‚Äù</em>: <code class="language-plaintext highlighter-rouge">alternate_annotation.gbk</code> (Input dataset) and <code class="language-plaintext highlighter-rouge">gbk</code> (output of <strong>Funannotate functional</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúFunannotate database‚Äù</em>: select the latest version available</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output is a web page with different tabs:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">Stats</code>: general statistics on the two annotations.</li>
  <li><code class="language-plaintext highlighter-rouge">Orthologs</code>: orthology relations between genes of the two annotations.</li>
  <li><code class="language-plaintext highlighter-rouge">InterPro</code> and <code class="language-plaintext highlighter-rouge">PFAM</code>: count the number of times each InterPro/PFAM signature is identified in each annotation. You can download a PDF file at the top, showing an Non-metric multidimensional scaling (NMDS) analysis of this data.</li>
  <li><code class="language-plaintext highlighter-rouge">Merops</code> and <code class="language-plaintext highlighter-rouge">CAZymes</code>: count the number of members of each Merops/CAZymes family in each annotation. You can download PDF files at the top, showing colorful representations of this data.</li>
  <li><code class="language-plaintext highlighter-rouge">GO</code>: list Gene Ontology terms that were found to be over or under represented in each annotation.</li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>

  <p>What can you tell from this report?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The alternate annotation contains less genes than the one you generated, and misses a lot of functional annotations. It confirms that this alternate annotation have a lower quality than the one you performed.</p>
  </blockquote>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>Congratulations for reaching the end of this tutorial! Now you know how to perform a structural and functional annotation of a new eukaryotic genome, using Funannotate, EggNOG mapper and InterProScan. You also learned how Funannotate can help you in the submission process to NCBI. And you learned how to visualise your new annotation using JBrowse, and how to compare it with another annotation.</p>

<p>An automatic annotation of an eukaryotic genome is unfortunately rarely perfect. If you inspect some predicted genes (or look at the <code class="language-plaintext highlighter-rouge">tbl2asn genome validation report</code> output of Funannotate), you may find some mistakes made by Funannotate, or potential problems, e.g. wrong exon/intron limits, splitted genes, or merged genes. Setting up a manual curation project using <a href="http://genomearchitect.org/">Apollo</a> can help a lot to manually fix these errors. Check out the <a href="/training-material/topics/genome-annotation/tutorials/apollo-euk/tutorial.html">Apollo tutorial</a> for more details.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Funannotate allows to perform structural annotation of an eukaryotic genome.</p>
</li>
                        
                        <li><p>Functional annotation can be performed using EggNOG-mapper and InterProScan.</p>
</li>
                        
                        <li><p>BUSCO and JBrowse allow to inspect the quality of an annotation.</p>
</li>
                        
                        <li><p>Funannotate allows to format an annotation for sumission at NCBI.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/genome-annotation/faqs/">FAQ page for the Genome Annotation topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Campbell2014">Campbell, M. S., C. Holt, B. Moore, and M. Yandell, 2014 <b>Genome annotation and curation using MAKER and MAKER-P</b>. Current Protocols in Bioinformatics 48: 4‚Äì11. <a href="https://doi.org/10.1002/0471250953.bi0411s48">10.1002/0471250953.bi0411s48</a></li>
<li id="jonathan_m_palmer_2020_4054262">Palmer, J. M., and J. Stajich, 2020 <b>Funannotate v1.8.1: Eukaryotic genome annotation</b>. <a href="https://doi.org/10.5281/zenodo.4054262">10.5281/zenodo.4054262</a></li>
<li id="braker">Br≈Øna, T., K. J. Hoff, A. Lomsadze, M. Stanke, and M. Borodovsky, 2021 <b>BRAKER2: automatic eukaryotic genome annotation with GeneMark-EP+ and AUGUSTUS supported by a protein database</b>. NAR Genomics and Bioinformatics 3: lqaa108. <a href="https://doi.org/10.1093/nargab/lqaa108">10.1093/nargab/lqaa108</a></li></ol>
                

                
                <h1 data-toc-skip>Glossary</h1>
                <dl>
                    
                    
                    <dt>NMDS</dt>
                    <dd>Non-metric multidimensional scaling</dd>
                    
                </dl>
                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Genome annotation with Funannotate (Genome Annotation)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Anthony Bretaudeau,  <b>Genome annotation with Funannotate (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html">https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{genome-annotation-funannotate,
author = "Anthony Bretaudeau",
title = "Genome annotation with Funannotate (Galaxy Training Materials)",
year = "",
month = "",
day = ""
url = "\url{https://training.galaxyproject.org/training-material/topics/genome-annotation/tutorials/funannotate/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine F√∂ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Bj√∂rn Gr√ºning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                
                
<div markdown="1">
<h2 id="funding">Funding</h2>
<p>These individuals or organisations provided funding support for the development of this resource</p>

<div class="row">

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="https://www.erasmusplus.nl/assets/images/logo.png" alt="Logo">
		
		<a href="/training-material/hall-of-fame/erasmusplus/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		<p>This project (<a href="https://ec.europa.eu/programmes/erasmus-plus/projects/eplus-project-details/#project/2020-1-NL01-KA203-064717"><code class="language-plaintext highlighter-rouge">2020-1-NL01-KA203-064717</code></a>) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.
<img src="https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg" alt="eu flag with the text: with the support of the erasmus programme of the european union" /></p>

	</div>

</div>
</div>


                


<script>
// update the date on load, or leave fallback of 'today'
var d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        
        <script src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script src="/training-material/assets/js/popper.min.js"></script>
        <script src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script src="/training-material/assets/js/main.js"></script>
        <script src="/training-material/assets/js/clipboard.min.js"></script>
        <script>
        var snippets=document.querySelectorAll('div.highlight');
        [].forEach.call(snippets,function(snippet){
            snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet tabindex="0"><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });

        var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
            target:function(trigger){return trigger.nextElementSibling;
        }});
        </script>
        

        <script type="text/javascript">
            if(window.location.hostname === "galaxyproject.github.io") {
                // Redirect
                var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
                $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

                window.setTimeout(function(){
                    window.location.href = redirect;
                }, 5000)

            }
        </script>
    </body>
</html>