<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Data management in Medicinal Chemistry / Data management in Medicinal Chemistry / Computational chemistry</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@2ca11edfcefe84c8a2e2c70ca7ba57e45c3a0b9e",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.20cf1f17.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Data management in Medicinal Chemistry">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-03-27 09:25:48 +0000">
<meta name="DC.creator" content="Julia Jakiela">
<meta name="DC.creator" content="Katarzyna Kamieniecka">
<meta name="DC.creator" content="Krzysztof Poterlowicz"><meta name="description" content="Modelling, simulation and analysis of biomolecular systems">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Computational chemistry / Data management in Medicinal Chemistry / Hands-on: Data management in Medicinal Chemistry">
        <meta property="og:description" content="Modelling, simulation and analysis of biomolecular systems">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Data management in Medicinal Chemistry",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-03-27 09:25:48 +0000",
  "datePublished": "2024-01-08 09:55:32 +0000",
  "identifier": "https://gxy.io/GTN:T00381",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Computational chemistry",
    "description": "Modelling, simulation and analysis of biomolecular systems",
    "url": "https://training.galaxyproject.org/training-material/topics/computational-chemistry/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Data management in Medicinal Chemistry",
  "url": "https://training.galaxyproject.org/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html",
  "version": 3,
  "timeRequired": "PT1H",
  "teaches": "Learn some terminology from the field of medicinal chemistry\n - Understand the idea of data-driven medicinal chemistry\n - Explore the databases used for drug discovery and development\n - Use Galaxy tools for data management, format conversion and simple analyses",
  "keywords": [
    "Computational chemistry",
    "fair",
    "data-management",
    "medicinal-chemistry",
    "computational-chemistry"
  ],
  "description": "The questions this  addresses are:\n - Why does medicinal chemistry research produce so much data?\n - How can big data be useful for medicinal chemists?\n - What are the current publicly available databases that can be used for drug discovery and development?\n - How to use Galaxy for data management and analysis of chemical data?\n\n\nThe objectives are:\n - Learn some terminology from the field of medicinal chemistry\n - Understand the idea of data-driven medicinal chemistry\n - Explore the databases used for drug discovery and development\n - Use Galaxy tools for data management, format conversion and simple analyses\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/fair/tutorials/fair-intro/tutorial.html",
      "name": "FAIR in a nutshell",
      "description": "Hands-on for 'FAIR in a nutshell' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/fair/tutorials/data-management/tutorial.html",
      "name": "FAIR data management solutions",
      "description": "Hands-on for 'FAIR data management solutions' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "name": "Julia Jakiela",
      "image": "https://avatars.githubusercontent.com/wee-snufkin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/kkamieniecka/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/kkamieniecka/",
      "name": "Katarzyna Kamieniecka",
      "image": "https://avatars.githubusercontent.com/kkamieniecka",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0009-0004-2454-5950",
      "orcid": "https://orcid.org/0009-0004-2454-5950"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/poterlowicz-lab/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/poterlowicz-lab/",
      "name": "Krzysztof Poterlowicz",
      "image": "https://avatars.githubusercontent.com/poterlowicz-lab",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6173-5674",
      "orcid": "https://orcid.org/0000-0001-6173-5674"
    }
  ],
  "contributor": [

  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Computational chemistry",
      "description": "Modelling, simulation and analysis of biomolecular systems",
      "url": "https://training.galaxyproject.org/training-material/topics/computational-chemistry/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3332",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3332",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3332"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/computational-chemistry/tutorials/med-chem-data/workflows/",
      "name": "Associated Workflows"
    }
  ],
  "abstract": "The development of medicinal chemistry is advancing very rapidly. Big pharmaceutical companies, research institutes and universities are working on ground-breaking solutions to help patients combat all kinds of diseases. During that development process, tons of data are generated – not only from the lab environment but also from clinical trials. Given that the discovery of more potent, safer and cheaper drugs is the ultimate goal of all research bodies, we should all focus on making the data we gather FAIR: **F**indable, **A**ccessible, **I**nteroperable, and **R**eusable to push the boundaries of drug development even further. "
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/computational-chemistry" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Computational chemistry
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/computational-chemistry/faqs/" title="Check our FAQs for the Computational chemistry topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/computational-chemistry/tutorials/med-chem-data/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-computational-chemistry">
    <h1 data-toc-skip>Data management in Medicinal Chemistry</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/wee-snufkin/" class="contributor-badge contributor-wee-snufkin"><img src="https://avatars.githubusercontent.com/wee-snufkin?s=36" alt="Julia Jakiela avatar" width="36" class="avatar" />
    Julia Jakiela</a><a href="/training-material/hall-of-fame/kkamieniecka/" class="contributor-badge contributor-kkamieniecka"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/kkamieniecka?s=36" alt="Katarzyna Kamieniecka avatar" width="36" class="avatar" />
    Katarzyna Kamieniecka</a><a href="/training-material/hall-of-fame/poterlowicz-lab/" class="contributor-badge contributor-poterlowicz-lab"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/poterlowicz-lab?s=36" alt="Krzysztof Poterlowicz avatar" width="36" class="avatar" />
    Krzysztof Poterlowicz</a>
		</td>
	</tr>
	

	

	

	

	

	

	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Why does medicinal chemistry research produce so much data?</p>
</li>
        
        <li><p>How can big data be useful for medicinal chemists?</p>
</li>
        
        <li><p>What are the current publicly available databases that can be used for drug discovery and development?</p>
</li>
        
        <li><p>How to use Galaxy for data management and analysis of chemical data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn some terminology from the field of medicinal chemistry</p>
</li>
        
        <li><p>Understand the idea of data-driven medicinal chemistry</p>
</li>
        
        <li><p>Explore the databases used for drug discovery and development</p>
</li>
        
        <li><p>Use Galaxy tools for data management, format conversion and simple analyses</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/fair/tutorials/fair-intro/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: FAIR in a nutshell</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/fair/tutorials/data-management/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: FAIR data management solutions</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/workflows/" title="Data management in Medicinal Chemistry workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/computational-chemistry/tutorials/med-chem-data/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/computational-chemistry/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Jan 8, 2024 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 27, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00381">https://gxy.io/GTN:T00381</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 3 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>The development of medicinal chemistry is advancing very rapidly. Big pharmaceutical companies, research institutes and universities are working on ground-breaking solutions to help patients combat all kinds of diseases. During that development process, tons of data are generated – not only from the lab environment but also from clinical trials. Given that the discovery of more potent, safer and cheaper drugs is the ultimate goal of all research bodies, we should all focus on making the data we gather FAIR: <strong>F</strong>indable, <strong>A</strong>ccessible, <strong>I</strong>nteroperable, and <strong>R</strong>eusable to push the boundaries of drug development even further.</p>

<p>With the currently available methods such as artificial intelligence, machine learning, many toolkits, software and access to various databases, managing big data is now inherently linked to medicinal chemistry and helps to make this area as efficient as it can be.</p>

<p>In this tutorial, we will therefore explore some concepts related to medicinal chemistry, explore the available chemical and pharmacological databases, and perform some basic data analyses using the Galaxy interface.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#principles-of-medicinal-chemistry-research" id="markdown-toc-principles-of-medicinal-chemistry-research">Principles of medicinal chemistry research</a>    <ol>
      <li><a href="#drug-likeness" id="markdown-toc-drug-likeness">Drug-likeness</a></li>
      <li><a href="#the-lipinski-rule-of-5" id="markdown-toc-the-lipinski-rule-of-5">The Lipinski rule of 5</a></li>
      <li><a href="#vebers-rule" id="markdown-toc-vebers-rule">Veber’s rule</a></li>
    </ol>
  </li>
  <li><a href="#big-data-for-drug-discovery" id="markdown-toc-big-data-for-drug-discovery">Big data for drug discovery</a>    <ol>
      <li><a href="#ten-vs-of-big-data-for-drug-discovery" id="markdown-toc-ten-vs-of-big-data-for-drug-discovery">Ten Vs of big data for drug discovery</a></li>
      <li><a href="#fair-medchem" id="markdown-toc-fair-medchem">FAIR MedChem</a></li>
    </ol>
  </li>
  <li><a href="#chemical-and-pharmacological-databases" id="markdown-toc-chemical-and-pharmacological-databases">Chemical and pharmacological databases</a>    <ol>
      <li><a href="#chemical-collections" id="markdown-toc-chemical-collections">Chemical collections</a></li>
      <li><a href="#drug--drug-like-compounds" id="markdown-toc-drug--drug-like-compounds">Drug / drug-like compounds</a></li>
      <li><a href="#drug-targets-including-genomics-and-proteomics-data" id="markdown-toc-drug-targets-including-genomics-and-proteomics-data">Drug targets, including genomics and proteomics data</a></li>
      <li><a href="#biological-data-from-assay-screening-metabolism-and-efficacy-studies" id="markdown-toc-biological-data-from-assay-screening-metabolism-and-efficacy-studies">Biological data from assay screening, metabolism, and efficacy studies</a></li>
      <li><a href="#drug-liabilities-and-toxicities" id="markdown-toc-drug-liabilities-and-toxicities">Drug liabilities and toxicities</a></li>
      <li><a href="#clinical-databases" id="markdown-toc-clinical-databases">Clinical databases</a></li>
    </ol>
  </li>
  <li><a href="#data-management-in-chemistry-using-galaxy" id="markdown-toc-data-management-in-chemistry-using-galaxy">Data management in chemistry using Galaxy</a>    <ol>
      <li><a href="#chemical-file-formats" id="markdown-toc-chemical-file-formats">Chemical file formats</a></li>
      <li><a href="#galaxy-tools" id="markdown-toc-galaxy-tools">Galaxy tools</a></li>
      <li><a href="#galaxy-workflow-summary" id="markdown-toc-galaxy-workflow-summary">Galaxy workflow summary</a></li>
    </ol>
  </li>
  <li><a href="#data-driven-medicinal-chemistry---yesterday-today-and-tomorrow" id="markdown-toc-data-driven-medicinal-chemistry---yesterday-today-and-tomorrow">Data-driven medicinal chemistry - yesterday, today and tomorrow</a>    <ol>
      <li><a href="#why-do-we-need-big-data-in-med-chem" id="markdown-toc-why-do-we-need-big-data-in-med-chem">Why do we need big data in med-chem?</a></li>
      <li><a href="#data-challenges" id="markdown-toc-data-challenges">Data challenges</a></li>
      <li><a href="#machine-learning-in-med-chem" id="markdown-toc-machine-learning-in-med-chem">Machine learning in med-chem</a></li>
      <li><a href="#knowledge-sharing" id="markdown-toc-knowledge-sharing">Knowledge sharing</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="principles-of-medicinal-chemistry-research">Principles of medicinal chemistry research</h1>

<h2 id="drug-likeness">Drug-likeness</h2>
<p>There are many factors that medicinal chemists take into account while designing new drugs. It is essential to estimate the properties of the molecule before synthesising it in the lab, so the structures of drug candidates are usually compared to the existing drugs by so-called drug-likeness. This includes intrinsic properties of a compound that will lead to favourable ADMET (<strong>A</strong>bsorption, <strong>D</strong>istribution, <strong>M</strong>etabolism, <strong>E</strong>xcretion, <strong>T</strong>oxicity) characteristics. Below are some properties usually assessed:</p>
<ul>
  <li>Lipophilicity</li>
  <li>Size</li>
  <li>Polarity</li>
  <li>Insolubility</li>
  <li>Unsaturation</li>
  <li>Flexibility</li>
</ul>

<p>Additionally, there have been some rules developed to help estimate drug-likeness, amongst which:</p>
<ul>
  <li>Lipinski rule (<span class="citation"><a href="#Lipinski2001">Lipinski <i>et al.</i> 2001</a></span>)</li>
  <li>Ghose rule (<span class="citation"><a href="#Ghose1999">Ghose <i>et al.</i> 1999</a></span>)</li>
  <li>Veber rule (<span class="citation"><a href="#Veber2002">Veber <i>et al.</i> 2002</a></span>)</li>
  <li>Egan rule (<span class="citation"><a href="#Egan2000">Egan <i>et al.</i> 2000</a></span>)</li>
  <li>Muegge rule (<span class="citation"><a href="#Muegge2001">Muegge <i>et al.</i> 2001</a></span>)</li>
  <li>Abbott bioavailability score (<span class="citation"><a href="#Martin2005">Martin 2005</a></span>)</li>
</ul>

<p>As you see, there are a number of characteristics to consider before even synthesising the molecule in a lab. One of the online tools that help to summarise all the most important information about intrinsic properties and drug-likeness of compounds is called <a href="http://www.swissadme.ch/">SwissADME</a> and is a free web tool to evaluate pharmacokinetics, drug-likeness and medicinal chemistry friendliness of small molecules (<span class="citation"><a href="#Daina2017">Daina <i>et al.</i> 2017</a></span>).</p>

<p>Below we briefly discuss two probably the most well-known rules: Lipinski’s and Veber’s rules.</p>

<h2 id="the-lipinski-rule-of-5">The Lipinski rule of 5</h2>

<p>This rule of thumb developed back in 1997 by Christopher Lipinski and colleagues tries to predict the likelihood that a given small molecule can be orally active. The Lipinski rule of 5 favours molecules as potential oral drug candidates if:</p>
<ul>
  <li>the molecular mass is less than 500</li>
  <li>the calculated logarithm of the octanol−water partition coefficient (clogP) is less than 5</li>
  <li>they have up to 5 H-bond donors</li>
  <li>they have up to 10 H-bond acceptors</li>
</ul>

<p>However, nowadays there are more and more drugs being developed which don’t comply with those rules and regardless are still effective. There are voices from the scientific community, pointing out that “We are in danger of repeating our past mistakes if we assume these new modalities are not ‘drug-like’ and cannot be oral drugs because they are not [rule of 5] compliant” (Michael Shultz from Novartis, cited in <span class="citation"><a href="#Mullard2018">Mullard 2018</a></span> where the authors re-assess the rule of 5). Then, in <span class="citation"><a href="#OHagan2014">O′Hagan <i>et al.</i> 2014</a></span> we read “This famous “rule of 5” has been highly influential in this regard, but only about 50 % of orally administered new chemical entities actually obey it.”</p>

<h2 id="vebers-rule">Veber’s rule</h2>

<p>In (<span class="citation"><a href="#Veber2002">Veber <i>et al.</i> 2002</a></span>) we read that the commonly applied molecular weight cutoff at 500 does not itself significantly separate compounds with poor oral bioavailability from those with acceptable values. The authors explain that on average both the number of rotatable bonds and polar surface area or hydrogen bond count tend to increase with molecular weight which may in part explain the success of the molecular weight parameter in predicting oral bioavailability. Their observations suggest that compounds will have a high probability of good oral bioavailability if they meet the two criteria:</p>
<ul>
  <li>10 or fewer rotatable bonds - increased rotatable bond count has a negative effect on the permeation rate</li>
  <li>a polar surface area no greater than 140 Å <sup>2</sup> (or 12 or fewer H-bond donors and acceptors) - reduced polar surface area correlates better with increased permeation rate than does lipophilicity</li>
</ul>

<h1 id="big-data-for-drug-discovery">Big data for drug discovery</h1>

<p>Have you ever wondered how new drugs are developed? It is actually quite a long and complex pathway and not only takes lots of time but also resources. The scheme below will help you understand the scale of that process.</p>

<figure id="figure-1" style="max-width: 90%;"><img src="/training-material/topics/computational-chemistry/images/medchem.png" alt="A scheme showing drug discovery pipeline: starting from target identification, hit, lead, lead optimisation, pre-clinical testing, clinical evaluation in phases I - III, and finally regulatory approval. Lead optimisation also includes iterative steps of designing, making, testing and analysing the proposed compounds." width="3801" height="2205" loading="lazy" /><a target="_blank" href="/training-material/topics/computational-chemistry/images/medchem.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> A scheme showing drug discovery pipeline</figcaption></figure>

<p>As you can see, to get just one drug to the market, often thousands of structures are assessed to find the most effective and safe compound. That’s lots of information being generated in the whole process! The information can be stored in repositories and databases so that subsequent drug discovery process in a similar area is easier, faster and cheaper. Managing big data and using machine learning and computational chemistry methods is crucial in the lead optimisation step. By reviewing chemical databases and studying molecular docking simulations, we can save much time and resources to synthesise only those compounds that are the most promising based on <em>in-silico</em> methods. Even after synthesising the optimised ‘leads’, we still get lots of data from tests and analyses. By using appropriate cheminformatic tools and software, we can use the results to drive forward this iterative process of lead optimisation.</p>

<h2 id="ten-vs-of-big-data-for-drug-discovery">Ten Vs of big data for drug discovery</h2>

<p>Hopefully, now you have an idea of what the process of drug discovery looks like. To understand why managing data in medicinal chemistry is so challenging, let’s have a closer look at the “Ten Vs” of big data <span class="citation"><a href="#Zhao2020">Zhao <i>et al.</i> 2020</a></span>:</p>

<ul>
  <li><strong>V</strong>olume: size of data.</li>
  <li><strong>V</strong>elocity: speed of new data generation.</li>
  <li><strong>V</strong>ariety: various formats of data.</li>
  <li><strong>V</strong>eracity: quality of data.</li>
  <li><strong>V</strong>alidity: authenticity of data.</li>
  <li><strong>V</strong>ocabulary: terminology of data.</li>
  <li><strong>V</strong>enue: platform of data generation.</li>
  <li><strong>V</strong>isualization: view of data.</li>
  <li><strong>V</strong>olatility: duration of data usefulness.</li>
  <li><strong>V</strong>alue: the potential of data usefulness to reduce the cost of drug discovery and development.</li>
</ul>

<p>Medicinal chemists must keep in mind those features while both searching for data and publishing it. In this way we can focus on the main aspects of data management, try to improve the availability and normalisation of data, and be aware of limitations of repositories and inconsistencies in data quality.</p>

<h2 id="fair-medchem">FAIR MedChem</h2>
<p>As you saw before, there are many properties that the compounds can be searched by, such as molecular weight, H-bond donors or acceptors, polarity, etc. Sometimes there is a need to assess other molecular properties but it might happen that not all the information is included in the database. This is one of the reasons why new repositories are being developed – they are more specific and gather particular properties of interest. How much easier the life of scientists could be if the relevant data was publicly available, well-ordered and contained the needed metadata? By submitting the data with the necessary information to the repository is a good way to make the data FAIR. This will make it:</p>
<ul>
  <li><strong>F</strong>indable, as the data will be given specific identifiers</li>
  <li><strong>A</strong>ccessible, as the data will be available online, open and free where possible</li>
  <li><strong>I</strong>nteroperable, as the repository will often enforce the use of formalised, consistent language</li>
  <li><strong>R</strong>eusable, as the data will be released under a license with detailed provenance
In the repositories</li>
</ul>

<h1 id="chemical-and-pharmacological-databases">Chemical and pharmacological databases</h1>

<p>Currently, there are lots of publicly available databases storing information about hit molecules, chemical structures and properties, drug targets, pharmaceuticals… In the recent paper <span class="citation"><a href="#Zhao2020">Zhao <i>et al.</i> 2020</a></span> the authors collated relevant databases to advance computer-aided drug discovery and divided them into several main groups:</p>
<ul>
  <li>Chemical collections</li>
  <li>Drug / drug-like compounds</li>
  <li>Drug targets, including genomics and proteomics data</li>
  <li>Biological data from assay screening, metabolism, and efficacy studies</li>
  <li>Drug liabilities and toxicities</li>
  <li>Clinical databases</li>
</ul>

<p>Below you will find the databases listed under corresponding categories, all taken from <span class="citation"><a href="#Zhao2020">Zhao <i>et al.</i> 2020</a></span> paper.</p>

<p>There are many more databases available, and many are still being developed. They are often quite specific and contain certain types of compounds (eg. <a href="http://cadd.zju.edu.cn/protacdb/">PROTACs</a> or are aimed at a particular disease. If the listed databases are not specific enough for your research, you can try luck by searching smaller, more specific databases.</p>

<h2 id="chemical-collections">Chemical collections</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Database</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Size (as of 29 October 2019)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><a href="https://enamine.net/hit-finding/compound-collections/real-database">Enamine REAL Database</a></td>
      <td style="text-align: center">Tool used to find new hit molecules using large-scale virtual screening and for searching analogues to hit molecules</td>
      <td style="text-align: center">&gt;700 million compounds that comply with ‘rule of 5’ and Veber criteria</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://zinc.docking.org/">ZINC</a></td>
      <td style="text-align: center">Contains compound information including 2D/3D structure, purchasability,   target, and biology-related information</td>
      <td style="text-align: center">&gt;230 million compounds in 3D formats and &gt;750 million compounds for analogue searching</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://pubchem.ncbi.nlm.nih.gov">PubChem</a></td>
      <td style="text-align: center">Contains chemical molecule (mostly small molecule) information, including chemical structures, identifiers, chemical and physical properties,   biological activities, safety and toxicity data</td>
      <td style="text-align: center">97 million compounds, 236 million substances, 268 million bioactivities</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.chemspider.com/">ChemSpider</a></td>
      <td style="text-align: center">Free chemical structure database providing fast access to &gt;78 million   structures, properties, and associated information</td>
      <td style="text-align: center">&gt;78 million compound structures</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://kolblab.org/scubidoo/index.php">SCUBIDOO</a></td>
      <td style="text-align: center">Freely accessible database that currently holds 21 million virtual products originating from a small library of building blocks and collection of robust organic reactions</td>
      <td style="text-align: center">21 million virtual products</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.ebi.ac.uk/chembl/">ChEMBL</a></td>
      <td style="text-align: center">Manually curated database of bioactive molecules with drug-like properties; brings together chemical, bioactivity, and genomic data to aid translation of genomic information into effective new drugs.</td>
      <td style="text-align: center">&gt;1.9 million compounds, 1.1 million pieces of assay information</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://mesh.tcm.microbioinformatics.org/">TCM-Mesh</a></td>
      <td style="text-align: center">Integration of a database and a data-mining system for network pharmacology analysis of all respects of traditional Chinese medicine,   including herbs, herbal ingredients, targets, related diseases, adverse effect, and toxicity</td>
      <td style="text-align: center">383 840 compounds, 6235 herbs</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://bioinf-applied.charite.de/supernatural_new/index.php">Super Natural II</a></td>
      <td style="text-align: center">Contains natural compounds, including information about corresponding 2D   structures, physicochemical properties, predicted toxicity class and potential vendors</td>
      <td style="text-align: center">325 508 natural compounds</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://webs.iiitd.edu.in/raghava/biadb/index.html">BIAdb</a></td>
      <td style="text-align: center">Comprehensive database of benzylisoquinoline alkaloids, containing information about 846 unique benzylisoquinoline alkaloids</td>
      <td style="text-align: center">~846 unique benzylisoquinoline alkaloids</td>
    </tr>
  </tbody>
</table>

<h2 id="drug--drug-like-compounds">Drug / drug-like compounds</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Database</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Size (as of 29 October 2019)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><a href="http://956023.ichengyun.net/AICD/index.php">AICD</a></td>
      <td style="text-align: center">Anti-Inflammatory Compounds Database (AICD) deposits compounds with potential antiinflammation activities</td>
      <td style="text-align: center">79 781 small molecules</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://go.drugbank.com/">DrugBank</a></td>
      <td style="text-align: center">Unique bioinformatics and cheminformatics resource that combines detailed drug data with comprehensive drug target information</td>
      <td style="text-align: center">13 441 drug entries</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://reframedb.org/">ReFRAME</a></td>
      <td style="text-align: center">Screening library of 12 000 molecules assembled by combining three databases (Clarivate Integrity, GVK Excelra GoStar, and Citeline   Pharmaprojects) to facilitate drug repurposing</td>
      <td style="text-align: center">12 000 molecules</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://cheminfo.charite.de/superdrug2/">SuperDRUG2</a></td>
      <td style="text-align: center">Contains approved/marketed drugs with regulatory details, chemical structures (2D and 3D), dosage, biological targets, physicochemical properties, external identifiers, adverse effects, and PK data</td>
      <td style="text-align: center">&gt;4600 active pharmaceutical ingredients</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.fda.gov/drugs/drug-approvals-and-databases/drugsfda-data-files">Drugs@FDA database</a></td>
      <td style="text-align: center">Information about drugs from FDA</td>
      <td style="text-align: center">~23 391 drug application records</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://chemoinfo.ipmc.cnrs.fr/MOLDB/index.php">e-Drug3D</a></td>
      <td style="text-align: center">Contains 1930 molecular structures approved by FDA between 1939 and 2019   with a molecular weight &lt;2000</td>
      <td style="text-align: center">1930 drugs</td>
    </tr>
  </tbody>
</table>

<h2 id="drug-targets-including-genomics-and-proteomics-data">Drug targets, including genomics and proteomics data</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Database</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Size (as of 29 October 2019)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><a href="https://www.bindingdb.org/bind/index.jsp">BindingDB</a></td>
      <td style="text-align: center">Public, web-accessible database of measured binding affinities, focusing chiefly on interactions of proteins considered to be candidate drug targets with ligands that are small,   drug-like molecules</td>
      <td style="text-align: center">1 756 093 binding data, for 7371 protein targets and 780 240 small molecules</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://insilico.charite.de/supertarget/index.php?site=home">Supertarget</a></td>
      <td style="text-align: center">An extensive web resource for analyzing drugtarget interactions.</td>
      <td style="text-align: center">332 828 drug-target interactions</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://ligand-expo.rutgers.edu/index.html">Ligand Expo</a></td>
      <td style="text-align: center">Provides chemical and structural information about small molecules within structure entries of Protein Data   Bank.</td>
      <td style="text-align: center">30 440 entries of ligand</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.ebi.ac.uk/pdbe-srv/pdbechem/">PDBeChem</a></td>
      <td style="text-align: center">Consistent and enriched library of ligands, small molecules, and monomers referenced as residues and hetgroups in PDB entries</td>
      <td style="text-align: center">&gt;29 922 ligands</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://www.pdbbind-cn.org/">PDBbind-CN</a></td>
      <td style="text-align: center">Provides essential linkage between energetic and structural information of biomolecular complexes, which is helpful for various computational and statistical studies on molecular recognition in biological systems</td>
      <td style="text-align: center">19 588 biomolecular complexes</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://stitch.embl.de/">STITCH</a></td>
      <td style="text-align: center">Database integrating information about interactions from metabolic pathways, crystal structures, binding experiments, and drug–target relationships</td>
      <td style="text-align: center">Interactions between 300 000 small molecules and 2.6 million proteins from 1133 organisms</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://thebiogrid.org/">BioGRID</a></td>
      <td style="text-align: center">The Biological General Repository for Interaction Datasets is an open-access database on protein, genetic, and chemical interactions for humans and all major model organisms</td>
      <td style="text-align: center">1 753 686 protein and genetic interactions, 28 093 chemical associations and 874 796 posttranslational modifications from major model organisms</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://bindingmoad.org/">Binding MOAD</a></td>
      <td style="text-align: center">Created from a subset of Protein   Data Bank (PDB), containing every high-quality example of ligand-protein binding.</td>
      <td style="text-align: center">36 047 protein-ligand structures, and 13 353 binding data</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://gpcrdb.org/">GPCRdb</a></td>
      <td style="text-align: center">Contains data from GPCRs, including crystal structures, sequence alignments, and receptor mutations; can be visualized in interactive diagrams; provides online analysis tools</td>
      <td style="text-align: center">15 149 proteins, and 144 917 ligands</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.guidetopharmacology.org/">Guide to Pharmacology</a></td>
      <td style="text-align: center">IUPHAR/BPS guide to pharmacology is an open-access, expert-curated database of molecular interactions between ligands and their targets.</td>
      <td style="text-align: center">2937 targets, and 9859 ligands</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://zhanglab.ccmb.med.umich.edu/GLASS/">GLASS</a></td>
      <td style="text-align: center">GPCR-Ligand Association (GLASS)   database is a manually curated repository for experimentally validated   GPCR–ligand interactions; along with relevant GPCR and chemical information,   GPCRligand association data are extracted and integrated into GLASS from literature and public databases</td>
      <td style="text-align: center">~ 277 651 unique ligands and 3048 GPCRs</td>
    </tr>
  </tbody>
</table>

<h2 id="biological-data-from-assay-screening-metabolism-and-efficacy-studies">Biological data from assay screening, metabolism, and efficacy studies</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Database</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Size (as of 29 October 2019)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><a href="https://hmdb.ca/about">HMDB</a></td>
      <td style="text-align: center">Freely available electronic database containing detailed information about small-molecule metabolites found in the human body</td>
      <td style="text-align: center">114 162 metabolite entries</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://smpdb.ca/">SMPDB</a></td>
      <td style="text-align: center">Small Molecule Pathway Database (SMPDB) is an interactive, visual database containing &gt;30 000 small-molecule pathways found in humans only</td>
      <td style="text-align: center">&gt;30 000 small-molecule pathways</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://db.idrblab.net/ttd/">TTD</a></td>
      <td style="text-align: center">Therapeutic Target Database (TTD) is a database providing information about known and explored therapeutic protein and nucleic acid targets,   targeted disease, pathway information and corresponding drugs directed at each of these targets</td>
      <td style="text-align: center">2589 targets, and 31 614 drugs</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://biocyc.org/">BioCyc</a></td>
      <td style="text-align: center">Collection of 7615 pathway/genome databases; each database in BioCyc collection describes genome and metabolic pathways of a single organism</td>
      <td style="text-align: center">7615 pathway/genome databases</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://bigg.ucsd.edu/">BiGG</a></td>
      <td style="text-align: center">Metabolic reconstruction of human metabolism designed for systems biology simulation and metabolic flux balance modelling</td>
      <td style="text-align: center">2004 proteins, 2766 metabolites, and 3311 metabolic and transport reactions</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.brenda-enzymes.org/">BRENDA</a></td>
      <td style="text-align: center">Main collection of enzyme functional data available to scientific   community</td>
      <td style="text-align: center">At least 40 000 different enzymes from &gt;6900 different organisms</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://reactome.org/">Reactome</a></td>
      <td style="text-align: center">Curated, peer-reviewed knowledgebase of biological pathways, including metabolic pathways, and protein trafficking and signalling pathways</td>
      <td style="text-align: center">&gt;9600 proteins, 9800 reactions, and 2000 pathways for humans</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.ebi.ac.uk/biomodels/">BioModels Database</a></td>
      <td style="text-align: center">Repository of computational models of biological processes; models   described from literature are manually curated and enriched with   cross-references</td>
      <td style="text-align: center">6753 patient-derived genome-scale metabolic models, 112 898 metabolic   models etc.</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.genome.jp/kegg">KEGG</a></td>
      <td style="text-align: center">Database resource that integrates genomic, chemical, and systemic functional information</td>
      <td style="text-align: center">18 652 metabolites</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://carlsbad.health.unm.edu/carlsbad/?mode=home">CARLSBAD</a></td>
      <td style="text-align: center">Database and knowledge inference system that integrates multiple bioactivity data sets to provide researchers with novel capabilities for mining and exploration of available structure-activity relationships (SAR)   throughout chemical biology space.</td>
      <td style="text-align: center">932 852 CARLSBAD activities, 890 323 unique structure–target pairs, 3542   targets, 435 343 unique structures</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://dud.docking.org/wombat/">WOMBAT</a></td>
      <td style="text-align: center">Contains 331 872 entries, representing 1966 unique targets, with bioactivity annotations</td>
      <td style="text-align: center">268 246 unique structures</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://cactus.nci.nih.gov/ncidb2.2/">Open NCI Database</a></td>
      <td style="text-align: center">Maintained by the National Cancer Institute; contains small-molecule information such as names, biological activities, and structures; a useful resource for researchers working in cancer/AIDS fields</td>
      <td style="text-align: center">&gt;250 000 compounds</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://crdd.osdd.net/raghava/npact/">NPACT</a></td>
      <td style="text-align: center">Provides information on the plant-derived natural compounds, including structure, properties (physical, elemental, and topological), cancer type,   cell lines, inhibitory values (IC50, ED50, EC50, GI50), molecular targets,   commercial suppliers, and drug-likeness of compounds</td>
      <td style="text-align: center">1574 entries</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://cfpub.epa.gov/ncea/risk/recordisplay.cfm?deid=204443">PKPB_DB</a></td>
      <td style="text-align: center">Contains physiological parameter values for humans from early childhood through senescence; intended to be used in physiologically based (PB) PK modelling; also contains similar data for animals (primarily rodents)</td>
      <td style="text-align: center">N/A</td>
    </tr>
  </tbody>
</table>

<h2 id="drug-liabilities-and-toxicities">Drug liabilities and toxicities</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Database</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Size (as of 29 October 2019)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><a href="http://www.t3db.ca/">T3DB</a></td>
      <td style="text-align: center">Unique bioinformatics resource that combines detailed toxin data with comprehensive toxin target information</td>
      <td style="text-align: center">3678 toxins</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://ntp.niehs.nih.gov/data/drugmatrix/">DrugMatrix</a></td>
      <td style="text-align: center">One of world’s largest toxicogenomic reference resources</td>
      <td style="text-align: center">~ 600 drug molecules and 10 000 genes</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://actor.epa.gov/actor/home.xhtml">ACToR</a></td>
      <td style="text-align: center">Includes computational toxicology information about compounds, including   HTS, chemical exposure, sustainable chemistry (chemical structures and physicochemical properties) and virtual tissue data</td>
      <td style="text-align: center">&gt;500 000 chemicals</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://cwtung.kmu.edu.tw/skinsensdb/">SkinSensDB</a></td>
      <td style="text-align: center">Contains curated data from published AOP-related skin sensitization   assays</td>
      <td style="text-align: center">710 unique chemicals</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://sideeffects.embl.de/download/">SIDER</a></td>
      <td style="text-align: center">Contains information on marketed medicines and their recorded adverse drug reactions, including frequency, drug and adverse effect classifications</td>
      <td style="text-align: center">1430 drugs with 5868 side effect information</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.fda.gov/science-research">LTKB Benchmark Dataset</a></td>
      <td style="text-align: center">Contains drugs with potential to cause drug-induced liver injury in humans; established using FDA-approved prescription drug labels;   liver-toxicity-knowledge-base-ltkb/ltkb-benchmark-dataset</td>
      <td style="text-align: center">287 prescription drugs</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://ctdbase.org/">CTD</a></td>
      <td style="text-align: center">Comparative Toxicogenomics Database (CTD) is a premier public resource for literature-based, manually curated associations between chemicals, gene products, phenotypes, diseases, and environmental exposures</td>
      <td style="text-align: center">13 378 unique chemicals and related information</td>
    </tr>
  </tbody>
</table>

<h2 id="clinical-databases">Clinical databases</h2>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Database</th>
      <th style="text-align: center">Description</th>
      <th style="text-align: center">Size (as of 29 October 2019)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><a href="https://clinicaltrials.gov/">ClinicalTrials.gov</a></td>
      <td style="text-align: center">Database of privately and publicly funded clinical studies conducted around the world</td>
      <td style="text-align: center">~ 324 429 research studies in all 50 US states and 209 countries</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://aact.ctti-clinicaltrials.org/">AACT database</a></td>
      <td style="text-align: center">Publicly available relational database that contains all information   (protocol and result data elements) about every study registered in   ClinicalTrials.gov. Content is downloaded from ClinicalTrials.gov daily and loaded into AACT</td>
      <td style="text-align: center">~ 324 429 research studies in all 50 US states and 209 countries</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.eortc.org/clinical-trials-database/">EORTC Clinical Trials Database</a></td>
      <td style="text-align: center">Contains information about EORTC clinical trials and clinical trials from other organizations with EORTC participation</td>
      <td style="text-align: center">N/A</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="http://exposome-explorer.iarc.fr/">Exposome-Explorer</a></td>
      <td style="text-align: center">Contains detailed information on the nature of biomarkers, populations and subjects where measured, samples analyzed, methods used for biomarker analyses, concentrations in biospecimens, correlations with external exposure measurements, and biological reproducibility over time</td>
      <td style="text-align: center">908 biomarkers</td>
    </tr>
    <tr>
      <td style="text-align: center"><a href="https://www.pharmgkb.org/">PharmaGKB</a></td>
      <td style="text-align: center">A pharmacogenomics knowledge resource that encompasses clinical information about drug molecules</td>
      <td style="text-align: center">733 drugs with their clinical information</td>
    </tr>
  </tbody>
</table>

<h1 id="data-management-in-chemistry-using-galaxy">Data management in chemistry using Galaxy</h1>

<h2 id="chemical-file-formats">Chemical file formats</h2>

<p>Depending on the type of analysis you are going to perform, you will work with different file formats. Below are the most important data types, commonly used in cheminformatics, that you will likely see in Galaxy:</p>
<ul>
  <li>SDF (.sd, .sdf) - stands for Structure-Data Format, and SDF files wrap the molfile (MDL Molfile) format. Stores information about the chemical structure and associated data of compounds in plain text. Files in SDF format can encode single or multiple molecules that are then delimited by lines consisting of four dollar signs. SDF files are formatted ASCII files that store information about the positions of the individual atoms (either in 2D or 3D space) that make up the molecule. The data on connectivity and hybridization state are also encoded, although their use is less frequent and often inconsistent.</li>
  <li>MOL (.mol) - an MDL Molfile for holding information about the atoms, bonds, connectivity and coordinates of a molecule.</li>
  <li>MOL2 (.mol2) - a Tripos Mol2 file is a complete, portable representation of a SYBYL molecule. It is an ASCII file which contains all the information needed to
reconstruct a SYBYL molecule</li>
  <li>CML (.cml) - Chemical Markup Language (ChemML or CML) is an approach to managing molecular information using tools such as XML and Java. It supports a wide range of chemical concepts including molecules, reactions, spectra and analytical data, computational chemistry, chemical crystallography and materials</li>
  <li>InChI (IUPAC International Chemical Identifier) - a textual identifier for chemical substances, designed to provide a standard way to encode molecular information. The identifiers describe chemical substances in terms of layers of information — the atoms and their bond connectivity, tautomeric information, isotope information, stereochemistry, and electronic charge information (<span class="citation"><a href="#Heller2015">Heller <i>et al.</i> 2015</a></span>)</li>
  <li>SMILES (.smi) - the <strong>s</strong>implified <strong>m</strong>olecular-<strong>i</strong>nput <strong>l</strong>ine-<strong>e</strong>ntry <strong>s</strong>ystem (SMILES) is a specification in the form of a line notation for describing the structure of chemical species using short ASCII strings. A linear text format which can describe the connectivity and chirality of a molecule (<span class="citation"><a href="#Weininger1988">Weininger 1988</a></span>)</li>
  <li>PDB - the Protein Data Bank (PDB) file format is a textual file format describing the three-dimensional structures of molecules held in the <a href="https://www.rcsb.org/">Protein Data Bank</a>, now succeeded by the mmCIF format. It contains a description and annotation of protein and nucleic acid structures including atomic coordinates, secondary structure assignments, as well as atomic connectivity. In addition, experimental metadata is stored. (<span class="citation"><a href="#Berman2007">Berman 2007</a></span>)</li>
  <li>GRO (.gro) - a plain text file storing spatial coordinates and velocities (if available) of atoms during a molecular dynamics simulation, utilised by <a href="https://manual.gromacs.org/archive/5.0.3/online/gro.html">GROMACS</a></li>
</ul>

<p>To visualise the structures held by the files with the positions of the individual atoms, you can use (NGL Viewer)[https://nglviewer.org/ngl/].</p>

<p>If you plan to work with molecular dynamics simulations, there are also some MD trajectory file formats that you might want to get familiar with. In Galaxy, you can convert between xtc, trr, dcd and netcdf files using a tool called <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/chemteam/md_converter/md_converter/1.9.6+galaxy0" title="MDTraj file converter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MDTraj file converter</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.6+galaxy0)</span>.</p>

<h2 id="galaxy-tools">Galaxy tools</h2>

<p>In Galaxy Chemical Toolbox there are dozens of tools that can be used for various analyses. Below we will only show a few, mostly related to data import, format conversion and some functions linked to what was discussed previously.</p>

<p>Let’s start with importing publicly available data. <a href="https://www.rcsb.org/">Protein Data Bank</a> stores thousands of three-dimensional structural data of proteins nucleic acids and other biological molecules.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-pdb-file-from-protein-data-bank"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import PDB file from Protein Data Bank</div>

  <ol>
    <li>Open <a href="https://www.rcsb.org/">Protein Data Bank</a></li>
    <li>Type your search term in the box <em>“Enter search term(s), Entry ID(s), or sequence”</em>: <code class="language-plaintext highlighter-rouge">MAO B</code></li>
  </ol>
  <figure id="figure-2" style="max-width: 90%;"><img src="/training-material/topics/computational-chemistry/images/pdb_search.jpg" alt="A screenshot showing a search box in the upper right corner of the Protein Data Bank website." width="1482" height="290" loading="lazy" /><a target="_blank" href="/training-material/topics/computational-chemistry/images/pdb_search.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Search box of the Protein Data Bank</figcaption></figure>
  <ol>
    <li>Choose the structure of interest and copy its accession code: 2BK3</li>
  </ol>
  <figure id="figure-3" style="max-width: 90%;"><img src="/training-material/topics/computational-chemistry/images/pdb_results.jpg" alt="A screenshot showing some of the resulting structures with their accession codes." width="1165" height="629" loading="lazy" /><a target="_blank" href="/training-material/topics/computational-chemistry/images/pdb_results.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> How to get accession code of the resulting structures.</figcaption></figure>
  <ol>
    <li>Switch to Galaxy</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/get_pdb/get_pdb/0.1.0" title="Get PDB file tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Get PDB file</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.1.0)</span> with the following parameters:
      <ul>
        <li><em>“PDB accession code”</em>: <code class="language-plaintext highlighter-rouge">2BK3</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This is often the first step of docking studies that you can learn about in other tutorials, for example in <a href="/training-material/topics/computational-chemistry/tutorials/cheminformatics/tutorial.html">Protein-ligand docking tutorial</a>.</p>

<p>You can import any molecule to Galaxy using SMILES notation. Below we show an example with benzenesulfonyl chloride and ethylamine.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-smiles"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import SMILES</div>

  <ol>
    <li>Copy SMILES of your molecule(s) of interest. In this example, we use benzenesulfonyl chloride (SMILES: C1=CC=C(C=C1)S(=O)(=O)Cl) and ethylamine (SMILES: CCN).</li>
    <li>Click on the <strong><i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> Upload data</strong> button in the tools panel on the left-hand side.</li>
    <li>Click on the <strong><i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> Paste/Fetch data</strong> button twice. Two boxes will appear.</li>
    <li>In the first box, under <strong>“Name”</strong> section, enter <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride</code> and under <strong>“Type”</strong>: <code class="language-plaintext highlighter-rouge">smi</code></li>
    <li>Below paste the SMILES of benzenesulfonyl chloride: <code class="language-plaintext highlighter-rouge">C1=CC=C(C=C1)S(=O)(=O)Cl</code></li>
    <li>In the second box, under <strong>“Name”</strong> section, enter <code class="language-plaintext highlighter-rouge">Ethylamine</code> and under <strong>“Type”</strong>: <code class="language-plaintext highlighter-rouge">smi</code></li>
    <li>Below paste the SMILES of ethylamine: <code class="language-plaintext highlighter-rouge">CCN</code></li>
    <li>Click the <strong>“Start”</strong> button, then close the dialogue box. Your files are being added to your history!</li>
  </ol>

</blockquote>

<p>As you learned from the previous section, there are many different formats used in computational chemistry. Of course, Galaxy allows you to interconvert between them - just have a look below!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-convert-the-file-format"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Convert the file format</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0" title="Compound conversion tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compound conversion</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.1.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Molecular input file”</em>: click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>Multiple datasets</em> icon and choose both <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride</code> and <code class="language-plaintext highlighter-rouge">Ethylamine</code></li>
        <li><em>“Output format”</em> - you have many options to choose from! Not only the formats but also the associated parameters. Just scroll the list and pick the one that is relevant to your downstream analysis. Here we use <code class="language-plaintext highlighter-rouge">MDL MOL format (sdf, mol)</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the corresponding files <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride SDF</code> and <code class="language-plaintext highlighter-rouge">Ethylamine SDF</code>. Before renaming the files, make sure to check either the starting dataset numbers or the atoms in the file in order not to confuse the two files!</li>
  </ol>

</blockquote>

<p>We intentionally chose to work on benzenesulfonyl chloride and ethylamine. Do you know why? Well, if you think about the synthesis of sulfonamides, that’s exactly what you need - a sulfonyl chloride and an amine. So… let’s do some computational synthesis!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-reaction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run the reaction</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/ctb_im_rxn_maker/ctb_im_rxn_maker/1.1.4+galaxy0" title="Reaction maker tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Reaction maker</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.4+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride SDF</code></li>
        <li><em>“Reagent options”</em>: Sulfonamide</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reagent file”</em>: <code class="language-plaintext highlighter-rouge">Ethylamine SDF</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-reaction-maker-tool-works-also-with-smiles"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-reaction-maker-tool-works-also-with-smiles" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip:  Reaction maker tool works also with SMILES!</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Above we converted the <em>.smi</em> file into <em>.sdf</em> to show how <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Compound conversion</strong> tool works. However, it is worth pointing out that <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Reaction maker</strong> works also with SMILES files and the conversion happens automatically so that your <em>.smi</em> dataset is imported as <em>.sdf</em> input.</p>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-which-reactions-i-can-run-using-this-tool"><button class="gtn-boxify-button details" type="button" aria-controls="details-which-reactions-i-can-run-using-this-tool" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Which reactions I can run using this tool?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Reaction maker</strong> works with more reaction types, such as:</p>
  <ul>
    <li>Amides synthesis</li>
    <li>Nucleophilic aromatic substitution (SNAr mechanism)</li>
    <li>Urea</li>
    <li>Suzuki coupling</li>
    <li>Sulfonamide</li>
    <li>Reductive amination</li>
    <li>N-alkylation</li>
    <li>Ether coupling</li>
    <li>Ester coupling</li>
    <li>Benzimidazole</li>
    <li>Triazole</li>
    <li>Benzoxazole</li>
    <li>Sarah Cu</li>
    <li>Sarah Quat Am</li>
  </ul>

</blockquote>

<p>The output is the <em>.txt</em> log file and the SDF file with the product molecule. If you have a look at that file, you’ll see that there is just a list of atoms with their corresponding coordinates. To draw the resulting structure, we can use another tool.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualisation-of-compounds"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualisation of compounds</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_svg_depiction/openbabel_svg_depiction/3.1.1+galaxy0" title="Visualisation tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Visualisation</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.1.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecular input file”</em>: <code class="language-plaintext highlighter-rouge">SDF output for Reaction maker</code></li>
        <li><em>“Property to display under the molecule”</em> - there are many properties that you can choose from! Please note that sometimes if not enough information is provided, then the property might not be displayed. Let’s choose a basic but useful property: <code class="language-plaintext highlighter-rouge">Molecular weight</code></li>
        <li><em>“Format of the resulting picture”</em>: <code class="language-plaintext highlighter-rouge">SVG</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Since we spoke about the drug-likeness in the previous section, let’s see how it works in practice, on the example of our “synthesised” molecule.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-estimate-the-drug-likeness"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Estimate the drug-likeness</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/qed/ctb_silicos_qed/2021.03.4+galaxy0" title="Drug-likeness tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Drug-likeness</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2021.03.4+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecule data in SDF or SMILES format”</em>: <code class="language-plaintext highlighter-rouge">SDF output for Reaction maker</code></li>
        <li><em>“Method”</em> - two possible methods to weight the features are available - one based on max. weight (QEDw,max), and the other on mean weight (QEDw,mo). There is also an option to leave features unweighted (QEDw,u) and we’ll choose this one: <code class="language-plaintext highlighter-rouge">unweighted (QEDw,u)</code></li>
        <li><em>“Include the descriptor names as header”</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-are-the-assessed-properties"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-are-the-assessed-properties" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What are the assessed properties?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The eight properties used are molecular weight (MW), octanol–water partition coefficient (ALOGP), number of hydrogen bond donors (HBDs), number of hydrogen bond acceptors (HBAs), molecular polar surface area (PSA), number of rotatable bonds (ROTBs), number of aromatic rings (AROMs) and number of structural alerts (ALERTS).</p>

</blockquote>

<p>In medicinal chemistry, we often base the new structures on scaffolds of existing drugs. Therefore, it is quite useful to be able to find similar structures - below is an example of how to do it in Galaxy.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-search-chembl-database-for-similar-compounds"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Search ChEMBL database for similar compounds</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/chembl/chembl/0.10.1+galaxy4" title="Search ChEMBL database tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search ChEMBL database</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.1+galaxy4)</span> with the following parameters:
      <ul>
        <li><em>“SMILES input type”</em>: <code class="language-plaintext highlighter-rouge">File</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride</code></li>
        <li><em>“Search type”</em>: <code class="language-plaintext highlighter-rouge">Substructure</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the output file <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride substructures</code></li>
  </ol>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-search-chembl-database-tool-works-also-with-sdf"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-search-chembl-database-tool-works-also-with-sdf" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Search ChEMBL database tool works also with SDF!</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Even though the input format of the above tool is SMILES, it can automatically interconvert between <em>.sdf</em> and <em>.smi</em> formats, so you don’t have to use <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Compound conversion</strong> tool beforehand.</p>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-parameters-of-search-chembl-database-tool"><button class="gtn-boxify-button details" type="button" aria-controls="details-parameters-of-search-chembl-database-tool" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Parameters of Search ChEMBL database tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Search ChEMBL database</strong> tool allows for searching the ChEMBL database for compounds which resemble a SMILES string. Two search options are possible:</p>
  <ul>
    <li>similarity (searches for compounds which are similar to the input within a specified Tanimoto cutoff)</li>
    <li>substructure (searches for compounds which contain the input substructure)</li>
  </ul>

  <p>Results can be filtered for compounds which are</p>
  <ul>
    <li>approved drugs</li>
    <li>biotherapeutic</li>
    <li>natural products</li>
    <li>fulfil all of the Lipinski rule of five criteria</li>
  </ul>

</blockquote>

<p>Let’s repeat this step, but with Lipinski’s Rule of Five which you learned about in the previous section of this tutorial!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-search-chembl-database-for-similar-compounds-using-the-lipinski-rule-of-five"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Search ChEMBL database for similar compounds using the Lipinski rule of five</div>

  <ol>
    <li>Click the <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> <strong>Repeat</strong> button on the previous dataset.</li>
    <li>You will be redirected to <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/chembl/chembl/0.10.1+galaxy4" title="Search ChEMBL database tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search ChEMBL database</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.10.1+galaxy4)</span>. Leave all the parameters as before, except for one:
      <ul>
        <li><em>“Filter for Lipinski’s Rule of Five”</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">param-toggle</span> <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Rename <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the output file <code class="language-plaintext highlighter-rouge">Benzenesulfonyl chloride Lipinski substructures</code></li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many molecules have been filtered out?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>If you click on the dataset, you will see a short summary of how many molecules have been found.
When we didn’t apply any filters, the tool found 45 molecules. After applying Lipinski’s Rule of Five filter, 36 molecules satisfied the rules.</p>
    <figure id="figure-4" style="max-width: 90%;"><img src="/training-material/topics/computational-chemistry/images/substructures.png" alt="Image showing the preview of the output datasets from searching benzenesulfonyl chloride substructures in the ChEMBL database, with and without Lipinski's Rule of Five filter." width="705" height="438" loading="lazy" /><a target="_blank" href="/training-material/topics/computational-chemistry/images/substructures.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Preview showing the number of molecules found in the ChEMBL database, before and after applying Lipinski's Rule of Five filter</figcaption></figure>

  </blockquote>

</blockquote>

<h2 id="galaxy-workflow-summary">Galaxy workflow summary</h2>

<p>Those are just a few tools, but they offer many parameters that you can tune depending on your analysis. Additionally, we provide you with an <a href="https://usegalaxy.eu/u/j.jakiela/h/data-management-in-medicinal-chemistry-tutorial">example history</a> and the dedicated <a href="https://usegalaxy.eu/u/j.jakiela/w/data-management-in-medicinal-chemistry-workflow">workflow</a> of this small tutorial. You are more than welcome to explore other tools in the Galaxy Chemical Toolbox though!</p>

<h1 id="data-driven-medicinal-chemistry---yesterday-today-and-tomorrow">Data-driven medicinal chemistry - yesterday, today and tomorrow</h1>

<h2 id="why-do-we-need-big-data-in-med-chem">Why do we need big data in med-chem?</h2>
<p>In the article by <span class="citation"><a href="#Lusher2014">Lusher <i>et al.</i> 2014</a></span> the role of the medicinal chemist is to make decisions about which of the infinite possibilities of new compounds should be made next. As the amount and variety of data on which to base these decisions grows, so too must the data analysis skills of the medicinal chemist. The authors claim that modern medicinal chemists should be able to recognise sources of relevant information, prepare raw data, use statistical tools, extract meaningful information, interpret results, recognise potential problems and make visualisations to communicate their findings to improve the quality of compounds being produced in research. According to the authors, this situation will require improved education and increased access to data and information management tools. This is where Galaxy comes in, offering not only a platform for data analysis with a plethora of the most commonly used tools in the field but also educational materials allowing anyone to learn and excel in analysis.</p>

<p>The privilege of applying computational methods in medicinal chemistry pipelines allows to focus on the most promising compounds and to remove unsuitable ones before the stage of chemical synthesis. Not only is this approach more efficient but also more sustainable and can reduce the costs of the synthetic stage of drug discovery (<span class="citation"><a href="#Brown2018">Brown <i>et al.</i> 2018</a></span>).</p>

<p>Another important role that big data plays in drug discovery is target identification (and validation) which might be currently one the biggest challenges in medicinal chemistry. By introducing genomic data in drug discovery, we can produce more specific and effective medicines. Understanding the underlying causes of the disease and the biological targets are crucial in designing new drugs and hence should be inherently linked with this process, where possible.</p>

<h2 id="data-challenges">Data challenges</h2>
<p>It is important to understand the limitations of the databases we use since if data is not well understood, the processing and analysis drawn may ultimately be flawed, following the ‘garbage in, garbage out’ principle. Here are some challenges mentioned in <span class="citation"><a href="#Brown2018">Brown <i>et al.</i> 2018</a></span> that you can explore more in-depth by referring to that article.</p>
<ul>
  <li>Errors</li>
  <li>Reproducibility</li>
  <li>Standardisation</li>
  <li>Formatting of data</li>
  <li>Information held in silos</li>
  <li>Loss of contextual information</li>
  <li>Limited and biased data</li>
</ul>

<h2 id="machine-learning-in-med-chem">Machine learning in med-chem</h2>
<p>Given the huge number of molecules in the various databases, applying machine learning in cheminformatics and building predictive models of the physiochemical properties of molecules is more and more popular. It helps to predict absorption, distribution, metabolism, excretion, toxicity (ADMET), likelihood of interaction with the drug targets as well as off-target effects. It can be useful in elucidating complex protein-protein or drug-drug interaction networks, in understanding structure-activity relationships (SARs) and when combined with data from the ‘omics’ revolution. This approach also helps to identify bioisosteres and introduce the idea of scaffold hopping in molecular design (<span class="citation"><a href="#Brown2018">Brown <i>et al.</i> 2018</a></span>).</p>

<!---
TO FURTHER IMPROVE THE TUTORIAL:
expand this section, include examples
-->

<h2 id="knowledge-sharing">Knowledge sharing</h2>
<p>It is not only about gathering and managing the data but also building a culture of knowledge sharing. We work in big, international teams with well-established methods and well-equipped labs. There are so many opportunities for knowledge capture and exchange! The challenge emerges here though – how to efficiently connect and help each other? This is the question that many pharmaceutical companies try to answer, such as Merck &amp; Co. who describe their approach to knowledge management and report on the multiple enduring and complementary teams and initiatives to capture and share knowledge (<span class="citation"><a href="#Beshore2022">Beshore <i>et al.</i> 2022</a></span>).</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(plausible !== undefined){
                            plausible('TutorialComplete', {callback: tutorial_finish_finish, props: {path: document.location.pathname}})
                          } else {
                            tutorial_finish_finish();
                          }
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>The power of big data might be useful to shape innovations of the future in the pharmaceutical field.</p>
</li>
                        
                        <li><p>There are many publicly available databases used for drug discovery and development and they might look at the same medicines from different angles and classify them based on various factors.</p>
</li>
                        
                        <li><p>Galaxy provides tools and a platform for medicinal chemistry analyses.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/computational-chemistry/faqs/">FAQ page for the Computational chemistry topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/computational-chemistry#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Weininger1988">Weininger, D., 1988 <b>SMILES,  a chemical language and information system. 1. Introduction to methodology and encoding rules</b>. Journal of Chemical Information and Computer Sciences 28: 31–36. <a href="https://doi.org/10.1021/ci00057a005">10.1021/ci00057a005</a></li>
<li id="Ghose1999">Ghose, A. K., V. N. Viswanadhan, and J. J. Wendoloski, 1999 <b>A Knowledge-Based Approach in Designing Combinatorial or Medicinal Chemistry Libraries for Drug Discovery. 1. A Qualitative and Quantitative Characterization of Known Drug Databases</b>. Journal of Combinatorial Chemistry 1: 55–68. <a href="https://doi.org/10.1021/cc9800071">10.1021/cc9800071</a></li>
<li id="Egan2000">Egan, W. J., K. M. Merz, and J. J. Baldwin, 2000 <b>Prediction of Drug Absorption Using Multivariate Statistics</b>. Journal of Medicinal Chemistry 43: 3867–3877. <a href="https://doi.org/10.1021/jm000292e">10.1021/jm000292e</a></li>
<li id="Lipinski2001">Lipinski, C. A., F. Lombardo, B. W. Dominy, and P. J. Feeney, 2001 <b>Experimental and computational approaches to estimate solubility and permeability in drug discovery and development settings 1PII of original article: S0169-409X(96)00423-1. The article was originally published in Advanced Drug Delivery Reviews 23 (1997) 3–25. 1</b>. Advanced Drug Delivery Reviews 46: 3–26. <a href="https://doi.org/10.1016/s0169-409x(00)00129-0">10.1016/s0169-409x(00)00129-0</a></li>
<li id="Muegge2001">Muegge, I., S. L. Heald, and D. Brittelli, 2001 <b>Simple Selection Criteria for Drug-like Chemical Matter</b>. Journal of Medicinal Chemistry 44: 1841–1846. <a href="https://doi.org/10.1021/jm015507e">10.1021/jm015507e</a></li>
<li id="Veber2002">Veber, D. F., S. R. Johnson, H.-Y. Cheng, B. R. Smith, K. W. Ward <i>et al.</i>, 2002 <b>Molecular Properties That Influence the Oral Bioavailability of Drug Candidates</b>. Journal of Medicinal Chemistry 45: 2615–2623. <a href="https://doi.org/10.1021/jm020017n">10.1021/jm020017n</a></li>
<li id="Martin2005">Martin, Y. C., 2005 <b>A Bioavailability Score</b>. Journal of Medicinal Chemistry 48: 3164–3170. <a href="https://doi.org/10.1021/jm0492002">10.1021/jm0492002</a></li>
<li id="Berman2007">Berman, H. M., 2007 <b>The Protein Data Bank: a historical perspective</b>. Acta Crystallographica Section A Foundations of Crystallography 64: 88–95. <a href="https://doi.org/10.1107/s0108767307035623">10.1107/s0108767307035623</a></li>
<li id="Lusher2014">Lusher, S. J., R. McGuire, R. C. van Schaik, C. D. Nicholson, and J. de Vlieg, 2014 <b>Data-driven medicinal chemistry in the era of big data</b>. Drug Discovery Today 19: 859–868. <a href="https://doi.org/10.1016/j.drudis.2013.12.004">10.1016/j.drudis.2013.12.004</a></li>
<li id="OHagan2014">O′Hagan, S., N. Swainston, J. Handl, and D. B. Kell, 2014 <b>A ‘rule of 0.5’ for the metabolite-likeness of approved pharmaceutical drugs</b>. Metabolomics 11: 323–339. <a href="https://doi.org/10.1007/s11306-014-0733-z">10.1007/s11306-014-0733-z</a></li>
<li id="Heller2015">Heller, S. R., A. McNaught, I. Pletnev, S. Stein, and D. Tchekhovskoi, 2015 <b>InChI,  the IUPAC International Chemical Identifier</b>. Journal of Cheminformatics 7: <a href="https://doi.org/10.1186/s13321-015-0068-4">10.1186/s13321-015-0068-4</a></li>
<li id="Daina2017">Daina, A., O. Michielin, and V. Zoete, 2017 <b>SwissADME: a free web tool to evaluate pharmacokinetics,  drug-likeness and medicinal chemistry friendliness of small molecules</b>. Scientific Reports 7: <a href="https://doi.org/10.1038/srep42717">10.1038/srep42717</a></li>
<li id="Brown2018">Brown, N., J. Cambruzzi, P. J. Cox, M. Davies, J. Dunbar <i>et al.</i>, 2018 <b>Big Data in Drug Discovery</b>, pp. 277–356 <span style="font-style: normal">in</span> <i>Progress in Medicinal Chemistry</i>, <i>Elsevier</i>. <a href="https://doi.org/10.1016/bs.pmch.2017.12.003">10.1016/bs.pmch.2017.12.003</a></li>
<li id="Mullard2018">Mullard, A., 2018 <b>Re-assessing the rule of 5,  two decades on</b>. Nature Reviews Drug Discovery 17: 777–777. <a href="https://doi.org/10.1038/nrd.2018.197">10.1038/nrd.2018.197</a></li>
<li id="Zhao2020">Zhao, L., H. L. Ciallella, L. M. Aleksunes, and H. Zhu, 2020 <b>Advancing computer-aided drug discovery (CADD) by big data and data-driven machine learning modeling</b>. Drug Discovery Today 25: 1624–1638. <a href="https://doi.org/10.1016/j.drudis.2020.07.005">10.1016/j.drudis.2020.07.005</a></li>
<li id="Beshore2022">Beshore, D. C., A. M. Haidle, A. Arasappan, Y.-H. Lim, I. Raheem <i>et al.</i>, 2022 <b>Building a Culture of Medicinal Chemistry Knowledge Sharing</b>. Journal of Medicinal Chemistry 65: 3776–3785. <a href="https://doi.org/10.1021/acs.jmedchem.1c02144">10.1021/acs.jmedchem.1c02144</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=computational-chemistry/med-chem-data"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=computational-chemistry/med-chem-data">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Julia Jakiela, Katarzyna Kamieniecka, Krzysztof Poterlowicz,  <b>Data management in Medicinal Chemistry (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html">https://training.galaxyproject.org/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{computational-chemistry-med-chem-data,
author = "Julia Jakiela and Katarzyna Kamieniecka and Krzysztof Poterlowicz",
	title = "Data management in Medicinal Chemistry (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/computational-chemistry/tutorials/med-chem-data/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                
                
<div class="d-flex flex-wrap">

	<a href="/training-material/hall-of-fame/elixir-uk-dash/" class="funder-badge">
		
		<div class="avatar">
			
			<img class="funder-avatar" src="/training-material/shared/images/dash.png" alt="Logo">
			
		</div>

		<div class="info">
			<div class="name">DASH UK</div>
			<div class="description">
			This Fellowship was funded through the ELIXIR-UK DaSH project as part of the UKRI Innovation Scholars: Data Science Training in Health and Bioscience call (DaSH). (MR/V038966/1). The project aims to embed Research Data Management (RDM) know-how into UK universities and institutes by producing and delivering training in FAIR data stewardship using ELIXIR-UK knowledge and resources.

            
			</div>
		</div>
	</a>

</div>


                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    
        
        
        
        <li>
          <a href="/training-material/topics/computational-chemistry">Computational chemistry</a>
        </li>
        
    


                    </ul>
                </blockquote>
                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/computational-chemistry/tutorials/med-chem-data/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: chembl
  owner: bgruening
  revisions: a446ea7e2bc1
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ctb_im_rxn_maker
  owner: bgruening
  revisions: 8ef5428fb809
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: get_pdb
  owner: bgruening
  revisions: 538790c6c21b
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openbabel_compound_convert
  owner: bgruening
  revisions: e2c36f62e22f
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openbabel_svg_depiction
  owner: bgruening
  revisions: 7672039a0bb0
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qed
  owner: bgruening
  revisions: 52a8d34dd08f
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: md_converter
  owner: chemteam
  revisions: ba83f0923369
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00381">gxy.io/GTN:T00381</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/computational-chemistry/tutorials/med-chem-data/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/computational-chemistry/tutorials/med-chem-data/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/2ca11edfcefe84c8a2e2c70ca7ba57e45c3a0b9e">2ca11ed</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.20cf1f17.js'></script>

	
	

    </body>
</html>