<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Advanced R in Galaxy / Advanced R in Galaxy / Foundations of Data Science</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@9cc53bcb522fc6b7739affd8a51b9fe23f7defcf",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/r-advanced/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.20cf1f17.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Advanced R in Galaxy">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-06-14 09:56:50 +0000">
<meta name="DC.creator" content="The Carpentries">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Fotis E. Psomopoulos">
<meta name="DC.creator" content="Toby Hodges">
<meta name="DC.creator" content="Gallantries: Bridging Training Communities in Life Science, Environment and Health"><meta name="description" content="These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Foundations of Data Science / Advanced R in Galaxy / Hands-on: Advanced R in Galaxy">
        <meta property="og:description" content="These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Advanced R in Galaxy",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-06-14 09:56:50 +0000",
  "datePublished": "2019-12-09 13:25:16 +0000",
  "identifier": "https://gxy.io/GTN:T00102",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Foundations of Data Science",
    "description": "These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.",
    "url": "https://training.galaxyproject.org/training-material/topics/data-science/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Advanced R in Galaxy",
  "url": "https://training.galaxyproject.org/training-material/topics/data-science/tutorials/r-advanced/tutorial.html",
  "version": 10,
  "timeRequired": "PT2H",
  "teaches": "Be able to load and explore the shape and contents of a tabular dataset using base R functions.\n - Understand factors and how they can be used to store and work with categorical data.\n - Apply common `dplyr` functions to manipulate data in R.\n - Employ the ‘pipe’ operator to link together a sequence of functions.",
  "keywords": [
    "Foundations of Data Science",
    "R"
  ],
  "description": "The questions this  addresses are:\n - How can I manipulate data using R in Galaxy?\n - How do I get started with tabular data from Galaxy in R?\n\n\nThe objectives are:\n - Be able to load and explore the shape and contents of a tabular dataset using base R functions.\n - Understand factors and how they can be used to store and work with categorical data.\n - Apply common `dplyr` functions to manipulate data in R.\n - Employ the ‘pipe’ operator to link together a sequence of functions.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/rstudio/tutorial.html",
      "name": "RStudio in Galaxy",
      "description": "Hands-on for 'RStudio in Galaxy' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/data-science/tutorials/r-basics/tutorial.html",
      "name": "R basics in Galaxy",
      "description": "Hands-on for 'R basics in Galaxy' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/carpentries/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/carpentries/",
      "name": "The Carpentries",
      "image": "https://avatars.githubusercontent.com/carpentries",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "This project ([`2020-1-NL01-KA203-064717`](https://erasmus-plus.ec.europa.eu/projects/search/details/2020-1-NL01-KA203-064717)) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.\n![eu flag with the text: with the support of the erasmus programme of the european union](https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg)\n![erasmus plus logo](https://www.erasmusplus.nl/assets/images/logo.png)\n",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/ifb/",
          "name": "Institut Français de Bioinformatique",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.france-bioinformatique.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/fpsom/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/fpsom/",
      "name": "Fotis E. Psomopoulos",
      "image": "https://avatars.githubusercontent.com/fpsom",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "This project ([`2020-1-NL01-KA203-064717`](https://erasmus-plus.ec.europa.eu/projects/search/details/2020-1-NL01-KA203-064717)) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.\n![eu flag with the text: with the support of the erasmus programme of the european union](https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg)\n![erasmus plus logo](https://www.erasmusplus.nl/assets/images/logo.png)\n",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-0222-4273",
      "orcid": "https://orcid.org/0000-0002-0222-4273"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/tobyhodges/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/tobyhodges/",
      "name": "Toby Hodges",
      "image": "https://avatars.githubusercontent.com/tobyhodges",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-1766-456X",
      "orcid": "https://orcid.org/0000-0003-1766-456X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
      "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
      "image": "https://avatars.githubusercontent.com/gallantries",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Foundations of Data Science",
      "description": "These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.",
      "url": "https://training.galaxyproject.org/training-material/topics/data-science/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3372",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3372",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3372"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "url": "",
      "name": "Associated Training Datasets"
    }
  ],
  "abstract": "With HTS-Seq data analysis, we generated tables containing list of DE genes, their expression, some statistics, etc. We can manipulate these tables using Galaxy, as we saw in some tutorials, e.g. [\"Reference-based RNA-Seq data analysis\"]({% link topics/transcriptomics/tutorials/ref-based/tutorial.md %}), and create some visualisations."
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/data-science" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Foundations of Data Science
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/data-science/faqs/" title="Check our FAQs for the Foundations of Data Science topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/data-science/tutorials/r-advanced/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/data-science/tutorials/r-advanced/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-data-science">
    <h1 data-toc-skip>Advanced R in Galaxy</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/carpentries/" class="contributor-badge contributor-carpentries"><img src="https://avatars.githubusercontent.com/carpentries?s=36" alt="The Carpentries avatar" width="36" class="avatar" />
    The Carpentries</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Bérénice Batut avatar" width="36" class="avatar" />
    Bérénice Batut</a><a href="/training-material/hall-of-fame/fpsom/" class="contributor-badge contributor-fpsom"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/fpsom?s=36" alt="Fotis E. Psomopoulos avatar" width="36" class="avatar" />
    Fotis E. Psomopoulos</a><a href="/training-material/hall-of-fame/tobyhodges/" class="contributor-badge contributor-tobyhodges"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/tobyhodges?s=36" alt="Toby Hodges avatar" width="36" class="avatar" />
    Toby Hodges</a><a href="/training-material/hall-of-fame/gallantries/" class="contributor-badge contributor-gallantries"><img src="https://gallantries.github.io/assets/images/gallantries-logo.png" alt="Gallantries: Bridging Training Communities in Life Science, Environment and Health avatar" width="36" class="avatar" />
    Gallantries</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can I manipulate data using R in Galaxy?</p>
</li>
        
        <li><p>How do I get started with tabular data from Galaxy in R?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Be able to load and explore the shape and contents of a tabular dataset using base R functions.</p>
</li>
        
        <li><p>Understand factors and how they can be used to store and work with categorical data.</p>
</li>
        
        <li><p>Apply common <code class="language-plaintext highlighter-rouge">dplyr</code> functions to manipulate data in R.</p>
</li>
        
        <li><p>Employ the ‘pipe’ operator to link together a sequence of functions.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        

        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/galaxy-interface/tutorials/rstudio/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: RStudio in Galaxy</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/data-science/tutorials/r-basics/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: R basics in Galaxy</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/data-science/tutorials/r-advanced/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











<li class="btn btn-default supporting_material">


  <!-- dropdown with all recordings -->
  <a href="/training-material/topics/data-science/tutorials/r-advanced/recordings/" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this material in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
  </a>


  <ul class="dropdown-menu">
    

    
      
      
      
    <li><a class="dropdown-item" href="/training-material/topics/data-science/tutorials/r-advanced/recordings/index.html#tutorial-recording-15-february-2021" title="View the recording for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021) - 1h</a>
    </li>
      
    
    <li><a class="dropdown-item" href="/training-material/topics/data-science/tutorials/r-advanced/recordings/" title="View all recordings for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> View All</a>
    </li>

  </ul>

  
</li>
  





            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    

    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Dec 9, 2019 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 14, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00102">https://gxy.io/GTN:T00102</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">4.5</a> (0 recent ratings, 8 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 10 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>With HTS-Seq data analysis, we generated tables containing list of DE genes, their expression, some statistics, etc. We can manipulate these tables using Galaxy, as we saw in some tutorials, e.g. <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">“Reference-based RNA-Seq data analysis”</a>, and create some visualisations.</p>

<p>Sometimes we want to have some customizations on visualization, some complex table manipulations or some statistical analysis. If we can not find a Galaxy tools for that or the right parameters, we may need to use programming languages as R or Python.</p>

<!-- Keeping this here because people may be familiar enough with R, that they go directly to this tutorial, and it is still useful information. -->

<p>R is one of the most widely-used and powerful programming languages in bioinformatics. R especially shines where a variety of statistical tools are required (e.g. RNA-Seq, population genomics, etc.) and in the generation of publication-quality graphs and figures. Rather than get into an R vs. Python debate (both are useful), keep in mind that many of the concepts you will learn apply to Python and other programming languages.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-on-the-content-of-the-tutorial"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: On the content of the tutorial</div>

  <p>We believe that every learner can <strong>achieve competency with R</strong>. You have reached competency when you find that you are able to <strong>use R to handle common analysis challenges in a reasonable amount of time</strong> (which includes time needed to look at learning materials, search for answers online, and ask colleagues for help). As you spend more time using R (there is no substitute for regular use and practice) you will find yourself gaining competency and even expertise. The more familiar you get, the more complex the analyses you will be able to carry out, with less frustration, and in less time - the fantastic world of R awaits you!</p>

  <p>Nobody wants to learn how to use R. People want to learn how to use R to answer their own research questions! Ok, maybe some folks learn R for R’s sake, but these lessons assume that you want to start analyzing genomic data as soon as possible. Given this, there are many valuable pieces of information about R that we simply won’t have time to cover. Hopefully, we will clear the hurdle of giving you just enough knowledge to be dangerous, which can be a high bar in R! We suggest you look into the additional learning materials in the box below.</p>

  <p><strong>Some R skills we will <em>not</em> cover in these lessons</strong></p>

  <ul>
    <li>How to create and work with R matrices and R lists</li>
    <li>How to create and work with loops and conditional statements, and the “apply” of functions (which are super useful, read more <a href="https://www.r-bloggers.com/r-tutorial-on-the-apply-family-of-functions/">on this blog about apply functions</a>)</li>
    <li>How to do basic string manipulations (e.g. finding patterns in text using grep, replacing text)</li>
    <li>How to plot using the default R graphic tools (we <em>will</em> cover plot creation, but will do so using the popular plotting package <code class="language-plaintext highlighter-rouge">ggplot2</code>)</li>
    <li>How to use advanced R statistical functions</li>
  </ul>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-where-to-learn-more"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-to-learn-more" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Where to learn more</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The following are good resources for learning more about R. Some of them can be quite technical, but if you are a regular R user you may ultimately need this technical knowledge.</p>
    <ul>
      <li><a href="https://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf">R for Beginners</a>, by Emmanuel Paradis: a great starting point</li>
      <li><a href="https://cran.r-project.org/manuals.html">The R Manuals</a>, by the R project people</li>
      <li><a href="https://cran.r-project.org/other-docs.html">R contributed documentation</a>, also linked to the R project, with materials available in several languages</li>
      <li><a href="http://r4ds.had.co.nz/">R for Data Science</a>: a wonderful collection by noted R educators and developers Garrett Grolemund and Hadley Wickham</li>
      <li><a href="https://peerj.com/collections/50-practicaldatascistats/">Practical Data Science for Stats</a>: not exclusively about R usage, but a nice collection of pre-prints on data science and applications for R</li>
      <li><a href="https://software-carpentry.org/lessons/">Programming in R Software Carpentry lesson</a>: several Software Carpentry lessons in R to choose from</li>
      <li><a href="https://www.datacamp.com/courses/free-introduction-to-r">Data Camp Introduction to R</a>: a fun online learning platform for Data Science, including R.</li>
    </ul>
  </blockquote>
</blockquote>

<p>In this tutorial, we will take the list of DE genes extracted from DESEq2’s output that we generated in the <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">“Reference-based RNA-Seq data analysis” tutorial</a>, manipulate it and create some visualizations.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#get-familiar-with-the-annotated-de-genes-table-in-r" id="markdown-toc-get-familiar-with-the-annotated-de-genes-table-in-r">Get familiar with the annotated DE genes table in R</a>    <ol>
      <li><a href="#import-tabular-data-into-r" id="markdown-toc-import-tabular-data-into-r">Import tabular data into R</a></li>
      <li><a href="#summarize-and-determine-the-structure-of-a-data-frame" id="markdown-toc-summarize-and-determine-the-structure-of-a-data-frame">Summarize and determine the structure of a data frame</a></li>
      <li><a href="#factors" id="markdown-toc-factors">Factors</a></li>
      <li><a href="#subset-a-data-frame" id="markdown-toc-subset-a-data-frame">Subset a data frame</a></li>
      <li><a href="#change-type-of-columns" id="markdown-toc-change-type-of-columns">Change type of columns</a></li>
      <li><a href="#inspect-a-numerical-column-using-math-sorting-renaming" id="markdown-toc-inspect-a-numerical-column-using-math-sorting-renaming">Inspect a numerical column using math, sorting, renaming</a></li>
      <li><a href="#save-the-data-frame-to-a-file" id="markdown-toc-save-the-data-frame-to-a-file">Save the data frame to a file</a></li>
    </ol>
  </li>
  <li><a href="#aggregate-and-analyze-with-dplyr" id="markdown-toc-aggregate-and-analyze-with-dplyr">Aggregate and Analyze with dplyr</a>    <ol>
      <li><a href="#select-columns-and-filter-rows" id="markdown-toc-select-columns-and-filter-rows">Select columns and filter rows</a></li>
      <li><a href="#create-columns" id="markdown-toc-create-columns">Create columns</a></li>
      <li><a href="#group-and-summarize-data" id="markdown-toc-group-and-summarize-data">Group and summarize data</a></li>
      <li><a href="#reshape-data-frames" id="markdown-toc-reshape-data-frames">Reshape data frames</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="get-familiar-with-the-annotated-de-genes-table-in-r">Get familiar with the annotated DE genes table in R</h1>

<p>A substantial amount of the data we work with in science is tabular data, i.e. data arranged in rows and columns - also known as spreadsheets.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-few-principles-when-working-with-data"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Few principles when working with data</div>
  <p>We want to remind you of a few principles before we work with our first set of example data:</p>

  <ol>
    <li>
      <p><strong>Keep raw data separate from analyzed data</strong></p>

      <p>This is principle number one, because if you can’t tell which files are the original raw data, you risk making some serious mistakes (e.g. drawing conclusion from data which have been manipulated in some unknown way).</p>

      <p>When you work with data in R, you are not changing the original file which you loaded the data from. This is different than (for example) working with a spreadsheet program where changing the value of the cell leaves you one “save”-click away from overwriting the original file. You have to purposely use a writing function (e.g. <code class="language-plaintext highlighter-rouge">write.csv()</code>) to save data loaded into R. In that case, be sure to save the manipulated data into a new file. More on this later in the lesson.</p>
    </li>
    <li>
      <p><strong>Keep spreadsheet data Tidy</strong></p>

      <p>The simplest principle of <em>Tidy</em> data is that we have one row in our spreadsheet for each observation or sample, and one column for every variable that we measure or report on. As simple as this sounds, it’s very easily violated. Most data scientists agree that significant amounts of their time is spent tidying data for analysis.</p>
    </li>
    <li>
      <p><strong>Trust but verify</strong></p>

      <p>Finally, while you don’t need to be paranoid about data, you should have a plan for how you will prepare it for analysis. <strong>This a focus of this lesson</strong>. You probably already have a lot of intuition, expectations, assumptions about your data - the range of values you expect, how many values should have been recorded, etc. Of course, as the data get larger our human ability to keep track will start to fail (and yes, it can fail for small data sets too). R will help you to examine your data so that you can have greater confidence in your analysis, and its reproducibility.</p>
    </li>
  </ol>

</blockquote>

<h2 id="import-tabular-data-into-r">Import tabular data into R</h2>

<p>As in the “R Basics” tutorial, we will continue in <span class="tool" data-tool="interactive_tool_rstudio" title="RStudio tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>RStudio</strong></span>.</p>

<!--SNIPPET-->
<blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch RStudio</div>   <p>Depending on which server you are using, you may be able to run RStudio directly in Galaxy. If that is not available, RStudio Cloud can be an alternative.</p>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-in-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-in-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently RStudio in Galaxy is only available on <a href="https://usegalaxy.eu">UseGalaxy.eu</a> and <a href="https://usegalaxy.org">UseGalaxy.org</a></p>   <ol>     <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here to launch RStudio</a></li>     <li>Click Run Tool</li>     <li>The tool will start running and will stay running permanently</li>     <li>Click on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started.</li>   </ol> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-cloud-if-not-available-on-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-cloud-if-not-available-on-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio Cloud if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If RStudio is not available on the Galaxy instance:</p>   <ol>     <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>     <li>Create a new project</li>   </ol> </blockquote> </blockquote>
<p><!--END_SNIPPET--></p>

<p>There are several ways to import data into R. For our purpose here, we will focus on using the tools every R installation comes with (so called “base” R) to import a comma-delimited file containing the results of our variant calling workflow. We will need to load the sheet using a function called <code class="language-plaintext highlighter-rouge">read.csv()</code>.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-review-the-arguments-of-the-read-csv-function"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Review the arguments of the `read.csv()` function</div>

  <p>Before using the <code class="language-plaintext highlighter-rouge">read.csv()</code> function, use R’s help feature to answer the following questions.</p>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-getting-help-in-r"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-getting-help-in-r" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Getting help in R</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>Entering <code class="language-plaintext highlighter-rouge">?</code> before the function name and then running that line will bring up the help documentation. Also, when reading this particular help be careful to pay attention to the <code class="language-plaintext highlighter-rouge">read.csv</code> expression under the <strong>Usage</strong> heading. Other answers will be in the <strong>Arguments</strong> heading.</p>
  </blockquote>

  <ol>
    <li>What is the default parameter for <code class="language-plaintext highlighter-rouge">header</code> in the <code class="language-plaintext highlighter-rouge">read.csv()</code> function?</li>
    <li>What argument would you have to change to read a file that was delimited by semicolons (<code class="language-plaintext highlighter-rouge">;</code>) rather than commas?</li>
    <li>What argument would you have to change to read file in which numbers used commas for decimal separation (i.e. 1,00)?</li>
    <li>What argument would you have to change to read in only the first 10,000 rows of a very large file?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The <code class="language-plaintext highlighter-rouge">read.csv() </code>function has the argument <code class="language-plaintext highlighter-rouge">header</code> set to <code class="language-plaintext highlighter-rouge">TRUE</code> by default, this means the function always assumes the first row is header information, (i.e. column names)</li>
      <li>The <code class="language-plaintext highlighter-rouge">read.csv()</code> function has the argument ‘sep’ set to “,”. This means the function assumes commas are used as delimiters, as you would expect. Changing this parameter (e.g. <code class="language-plaintext highlighter-rouge">sep=";"</code>) would now interpret semicolons as delimiters.</li>
      <li>Although it is not listed in the <code class="language-plaintext highlighter-rouge">read.csv()</code> usage, <code class="language-plaintext highlighter-rouge">read.csv()</code> is a “version” of the function <code class="language-plaintext highlighter-rouge">read.table()</code> and accepts all its arguments. If you set <code class="language-plaintext highlighter-rouge">dec=","</code> you could change the decimal operator. We’d probably assume the delimiter is some other character.</li>
      <li>You can set <code class="language-plaintext highlighter-rouge">nrow</code> to a numeric value (e.g. <code class="language-plaintext highlighter-rouge">nrow=10000</code>) to choose how many rows of a file you read in. This may be useful for very large files where not all the data is needed to test some data cleaning steps you are applying.</li>
    </ol>

    <p>Hopefully, this exercise gets you thinking about using the provided help documentation in R. There are many arguments that exist, but which we wont have time to cover. Look here to get familiar with functions you use frequently, you may be surprised at what you find they can do.</p>

  </blockquote>
</blockquote>

<p>Now, let’s read the file with the annotated differentially expressed genes that was produced in the “<a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">Reference-based RNA-Seq data analysis</a>” tutorial.</p>

<!--
> <div class="box-title hands-on-title" id="hands-on-import-dataset-into-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import dataset into Galaxy</div>
> 1. In Galaxy, upload the following file from Zenodo:
>    ```
>    https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular
>    ```
> 2. Remember the dataset number (e.g. `2` in the example below)
>    <a href="../../images/rna-seq-counts-to-viz-in-r/dataset_number.png" rel="noopener noreferrer"><img src="../../images/rna-seq-counts-to-viz-in-r/dataset_number.png"  alt="todo. "   loading="lazy"></a>
>
{: .hands_on}
-->

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-the-annotated-differentially-expressed-genes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read the annotated differentially expressed genes</div>
  <ol>
    <li>Create a new script (if needed)</li>
    <li>
      <p>Read the tabular file in an object called <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code>. We can import directly by URL:</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## read in a CSV file and save it as 'annotatedDEgenes'</span><span class="w">
</span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <p>The first argument to pass to our <code class="language-plaintext highlighter-rouge">read.csv()</code> function is the file path for our data. The file path must be in quotes and now is a good time to remember to use tab autocompletion. If you use tab autocompletion you avoid typos and errors in file paths. Use it!</p>
    </li>
    <li>Inspect the <strong>Environment</strong> panel</li>
  </ol>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-upload-files-from-a-galaxy-history-to-rstudio"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-files-from-a-galaxy-history-to-rstudio" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Upload files from a Galaxy history to RStudio</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>Rstudio in Galaxy provides some special functions to import and export from your history.</p>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">gx_get</span><span class="p">(</span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="c1"># will import dataset number 2 from your history</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>In the <strong>Environment</strong> panel, you should have the <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> object, listed as 130 obs. (observations/rows) of 1 variable (column) - so the command worked (sort of)!</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-the-annotateddegenes-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect the annotatedDEgenes file</div>
  <ol>
    <li>Double-click on the name of the object in the <strong>Environment</strong> panel</li>
  </ol>

</blockquote>

<p>It will open a view of the data in a new tab on the top-left panel.</p>

<p>As you can see, there is a problem with how the data has been loaded. The table should contain 130 observations of 13 variables.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-identify-the-issue-with-data"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Identify the issue with data</div>

  <ol>
    <li>What is wrong?</li>
    <li>How should you adjust the parameters for <code class="language-plaintext highlighter-rouge">read.csv()</code> in order to produce the intended output?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The data file was not delimited by commas (<code class="language-plaintext highlighter-rouge">,</code>) which is the default expected delimiter for <code class="language-plaintext highlighter-rouge">read.csv()</code>. Instead it seems like it’s delimited by “white space”, i.e. spaces and/or tabs.</li>
      <li>We need to set the correct delimiter using the parameter <code class="language-plaintext highlighter-rouge">sep</code>. Given that we are not sure what the actual delimiter is, we could try both options, i.e. <code class="language-plaintext highlighter-rouge">sep=" "</code> (<em>space</em>) and <code class="language-plaintext highlighter-rouge">sep="\t"</code> (<em>tab</em>).</li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-correctly-tabular-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read correctly tabular file</div>
  <ol>
    <li>
      <p>Read the file given it’s a tabular file</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">read.csv</span><span class="p">(</span><span class="s2">"https://zenodo.org/record/3477564/files/annotatedDEgenes.tabular"</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"\t"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>Inspect the object in the <strong>Environment</strong> panel</li>
    <li>
      <p>Open the object</p>

      <p>You should have now a nice table</p>
    </li>
    <li>
      <p>Check the column names</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">colnames</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="s2">"GeneID"</span><span class="w">     </span><span class="s2">"Base.mean"</span><span class="w">  </span><span class="s2">"log2.FC."</span><span class="w">   </span><span class="s2">"StdErr"</span><span class="w">     </span><span class="s2">"Wald.Stats"</span><span class="w">
</span><span class="p">[</span><span class="m">6</span><span class="p">]</span><span class="w"> </span><span class="s2">"P.value"</span><span class="w">    </span><span class="s2">"P.adj"</span><span class="w">      </span><span class="s2">"Chromosome"</span><span class="w"> </span><span class="s2">"Start"</span><span class="w">      </span><span class="s2">"End"</span><span class="w">
</span><span class="p">[</span><span class="m">11</span><span class="p">]</span><span class="w"> </span><span class="s2">"Strand"</span><span class="w">     </span><span class="s2">"Feature"</span><span class="w">    </span><span class="s2">"Gene.name"</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>Congratulations! You’ve successfully loaded your data into RStudio!</p>

<h2 id="summarize-and-determine-the-structure-of-a-data-frame">Summarize and determine the structure of a data frame</h2>

<p>Tabular data are stored in R using <strong>data frame</strong>. A data frame could also be thought of as a collection of vectors, all of which have the same length.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-learn-more-about-our-data-frame"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Learn more about our data frame</div>
  <ol>
    <li>
      <p>Get summary statistics of <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> using <code class="language-plaintext highlighter-rouge">summary</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## get summary statistics on a data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">summary</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span><span class="n">GeneID</span><span class="w">            </span><span class="n">Base.mean</span><span class="w">          </span><span class="n">log2.FC.</span><span class="w">          </span><span class="n">StdErr</span><span class="w">            </span><span class="n">Wald.Stats</span><span class="w">
</span><span class="n">FBgn0000071</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">   </span><span class="m">19.15</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-4.1485</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.08433</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-30.741</span><span class="w">
</span><span class="n">FBgn0000079</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">100.29</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-1.3363</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.12849</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-10.005</span><span class="w">
</span><span class="n">FBgn0000116</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w">  </span><span class="m">237.99</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">-1.0272</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.16370</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">-4.982</span><span class="w">
</span><span class="n">FBgn0000406</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">1911.27</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">-0.2074</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">0.16432</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">-1.901</span><span class="w">
</span><span class="n">FBgn0000567</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">948.66</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">1.2203</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.19866</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">7.692</span><span class="w">
</span><span class="n">FBgn0001137</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">65114.84</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">2.6999</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">0.23292</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">27.566</span><span class="w">
</span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="w">    </span><span class="o">:</span><span class="m">124</span><span class="w">
</span><span class="n">P.value</span><span class="w">             </span><span class="n">P.adj</span><span class="w">               </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">              </span><span class="n">End</span><span class="w">                </span><span class="n">Strand</span><span class="w">
</span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2L</span><span class="o">:</span><span class="m">24</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">127448</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">140340</span><span class="w">   </span><span class="o">-:</span><span class="m">58</span><span class="w">
</span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2R</span><span class="o">:</span><span class="m">31</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7277516</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7279063</span><span class="w">   </span><span class="o">+:</span><span class="m">72</span><span class="w">
</span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr3L</span><span class="o">:</span><span class="m">27</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">13161546</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">13166253</span><span class="w">
</span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">4.201e-07</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">9.321e-06</span><span class="w">   </span><span class="n">chr3R</span><span class="o">:</span><span class="m">32</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">13436843</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">13446444</span><span class="w">
</span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">7.000e-12</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">3.700e-10</span><span class="w">   </span><span class="n">chrX</span><span class="w"> </span><span class="o">:</span><span class="m">16</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">19250429</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">19284637</span><span class="w">
</span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">9.418e-06</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">1.951e-04</span><span class="w">              </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31196915</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31203722</span><span class="w">

</span><span class="n">Feature</span><span class="w">              </span><span class="n">Gene.Name</span><span class="w">
</span><span class="n">lincRNA</span><span class="w">       </span><span class="o">:</span><span class="w">  </span><span class="m">3</span><span class="w">   </span><span class="n">Ama</span><span class="w">        </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
</span><span class="n">protein_coding</span><span class="o">:</span><span class="m">126</span><span class="w">   </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w">      </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
</span><span class="n">pseudogene</span><span class="w">    </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">   </span><span class="n">Ant2</span><span class="w">       </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="n">Argk</span><span class="w">       </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="n">bou</span><span class="w">        </span><span class="o">:</span><span class="w">  </span><span class="m">1</span><span class="w">
                     </span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="w">    </span><span class="o">:</span><span class="m">124</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Get the structure of <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> using <code class="language-plaintext highlighter-rouge">str</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## get the structure of a data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">str</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
</span><span class="s1">'data.frame'</span><span class="o">:</span><span class="w">	</span><span class="m">130</span><span class="w"> </span><span class="n">obs.</span><span class="w"> </span><span class="n">of</span><span class="w">  </span><span class="m">13</span><span class="w"> </span><span class="n">variables</span><span class="o">:</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">GeneID</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">130</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"FBgn0000071"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">87</span><span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="m">28</span><span class="w"> </span><span class="m">26</span><span class="w"> </span><span class="m">31</span><span class="w"> </span><span class="m">96</span><span class="w"> </span><span class="m">65</span><span class="w"> </span><span class="m">62</span><span class="w"> </span><span class="m">125</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Base.mean</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">1087</span><span class="w"> </span><span class="m">6410</span><span class="w"> </span><span class="m">65115</span><span class="w"> </span><span class="m">2192</span><span class="w"> </span><span class="m">5430</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w">  </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">-4.15</span><span class="w"> </span><span class="m">-3</span><span class="w"> </span><span class="m">-2.38</span><span class="w"> </span><span class="m">2.7</span><span class="w"> </span><span class="m">-2.11</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">StdErr</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">0.1349</span><span class="w"> </span><span class="m">0.1043</span><span class="w"> </span><span class="m">0.0843</span><span class="w"> </span><span class="m">0.0979</span><span class="w"> </span><span class="m">0.0925</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">-30.7</span><span class="w"> </span><span class="m">-28.7</span><span class="w"> </span><span class="m">-28.2</span><span class="w"> </span><span class="m">27.6</span><span class="w"> </span><span class="m">-22.7</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">P.value</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">1.62e-207</span><span class="w"> </span><span class="m">9.42e-182</span><span class="w"> </span><span class="m">2.85e-175</span><span class="w"> </span><span class="m">2.85e-167</span><span class="w"> </span><span class="m">1.57e-114</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">P.adj</span><span class="w">     </span><span class="o">:</span><span class="w"> </span><span class="n">num</span><span class="w">  </span><span class="m">1.39e-203</span><span class="w"> </span><span class="m">4.04e-178</span><span class="w"> </span><span class="m">8.15e-172</span><span class="w"> </span><span class="m">6.10e-164</span><span class="w"> </span><span class="m">2.70e-111</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Seqname</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"chr2L"</span><span class="p">,</span><span class="s2">"chr2R"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">4</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Start</span><span class="w">     </span><span class="o">:</span><span class="w"> </span><span class="n">int</span><span class="w">  </span><span class="m">24141394</span><span class="w"> </span><span class="m">10780892</span><span class="w"> </span><span class="m">26869237</span><span class="w"> </span><span class="m">10778953</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">820758</span><span class="w"> </span><span class="o">&gt;</span><span class="w">    </span><span class="m">6762592</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">End</span><span class="w">       </span><span class="o">:</span><span class="w"> </span><span class="n">int</span><span class="w">  </span><span class="m">24147490</span><span class="w"> </span><span class="m">10786958</span><span class="w"> </span><span class="m">26871995</span><span class="w"> </span><span class="m">10786907</span><span class="w"> </span><span class="m">13860001</span><span class="w"> </span><span class="m">31203722</span><span class="w"> </span><span class="m">24946636</span><span class="w"> </span><span class="m">22552113</span><span class="w"> </span><span class="m">821512</span><span class="w"> </span><span class="o">&gt;</span><span class="w">    </span><span class="m">6765261</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Strand</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"-"</span><span class="p">,</span><span class="s2">"+"</span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Feature</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"lincRNA"</span><span class="p">,</span><span class="s2">"protein_coding"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">...</span><span class="w">
 </span><span class="o">$</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">130</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"Ama"</span><span class="p">,</span><span class="s2">"Amy-p"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">88</span><span class="w"> </span><span class="m">113</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">84</span><span class="w"> </span><span class="m">26</span><span class="w"> </span><span class="m">41</span><span class="w"> </span><span class="m">53</span><span class="w"> </span><span class="m">69</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>Ok, thats a lot to unpack! Some things to notice:</p>

<ul>
  <li>the object type <code class="language-plaintext highlighter-rouge">data.frame</code> is displayed in the first row along with its dimensions, in this case 130 observations (rows) and 13 variables (columns)</li>
  <li>Each variable (column) has a name (e.g. <code class="language-plaintext highlighter-rouge">GeneID</code>). This is followed by the object mode (e.g. <code class="language-plaintext highlighter-rouge">factor</code>, <code class="language-plaintext highlighter-rouge">int</code>, <code class="language-plaintext highlighter-rouge">num</code>, etc.). Notice that before each variable name there is a <code class="language-plaintext highlighter-rouge">$</code> - this will be important later.</li>
</ul>

<p>So from both <code class="language-plaintext highlighter-rouge">summary</code> and <code class="language-plaintext highlighter-rouge">str</code>, we know that our data frame had 13 variables that summarize the data.</p>

<ul>
  <li>
    <p><code class="language-plaintext highlighter-rouge">Base.mean</code>, <code class="language-plaintext highlighter-rouge">log2.FC.</code> and <code class="language-plaintext highlighter-rouge">P.value</code> variables (and several others) are numerical data</p>

    <p>We get summary statistics on the min and max values for these columns, as well as mean, median, and interquartile ranges</p>
  </li>
  <li>
    <p>Many of the other variables (e.g. <code class="language-plaintext highlighter-rouge">Strand</code>) are treated as categorical data</p>

    <p>This type has special treatment in R - more on this in a bit. The most frequent 6 different categories and the number of times they appear (e.g. the <code class="language-plaintext highlighter-rouge">Strand</code> called <code class="language-plaintext highlighter-rouge">-</code> appeared 58 times) are displayed. Another example is the <code class="language-plaintext highlighter-rouge">protein_coding</code> value for <code class="language-plaintext highlighter-rouge">Feature</code> which appeared in 126 observations.</p>
  </li>
</ul>

<h2 id="factors">Factors</h2>

<p>Factors, the final major data structure we will introduce, can be thought of as vectors which are specialized for categorical data. Given R’s specialization for statistics, this make sense since categorial and continuous variables usually have different treatments. Sometimes you may want to have data treated as a factor, but in other cases, this may be undesirable.</p>

<p>Since some of the data in our data frame are factors, lets see how factors work.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-a-factor-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect a factor object</div>
  <ol>
    <li>
      <p>Extract the <code class="language-plaintext highlighter-rouge">feature</code> column to a new object called <code class="language-plaintext highlighter-rouge">feature</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## extract the "feature" column to a new object</span><span class="w">
</span><span class="n">feature</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Feature</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Inspect the first few items in our  using <code class="language-plaintext highlighter-rouge">head</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">
</span><span class="p">[</span><span class="m">5</span><span class="p">]</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">
</span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">lincRNA</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">pseudogene</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>What we get back are the items in our factor, and also something called “Levels”. <strong>Levels are the different categories contained in a factor</strong>. By default, R will organize the levels in a factor in alphabetical order. So the first level in this factor is “lincRNA”.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-get-the-structure-of-a-factor-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get the structure of a factor object</div>
  <ol>
    <li>
      <p>Get the structure of the <code class="language-plaintext highlighter-rouge">feature</code> object</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">str</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span><span class="w">
</span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"lincRNA"</span><span class="p">,</span><span class="s2">"protein_coding"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>For the sake of efficiency, R stores the content of a factor as a vector of integers, which an integer is assigned to each of the possible levels. Recall levels are assigned in alphabetical order. In this case, the first item in our “Feature” object is “protein_coding”, which happens to be the 2nd level of our factor, ordered alphabeticaly. The 9th item in the list is “lincRNA”, which is the 1st level of our factor.</p>

<p>One of the most common uses for factors is to plot categorical values. For example, suppose we want to know how many of our differentially expressed genes correspond to different features.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plot-a-factor-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Plot a factor object</div>
  <ol>
    <li>
      <p>Plot the number of levels in the <code class="language-plaintext highlighter-rouge">feature</code> object using <code class="language-plaintext highlighter-rouge">plot</code> function</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plot</span><span class="p">(</span><span class="n">feature</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><a href="../../../introduction/images/rna-seq-counts-to-viz-in-r/featurePlot.png" rel="noopener noreferrer"><img src="../../../introduction/images/rna-seq-counts-to-viz-in-r/featurePlot.png" alt="feature plot. " width="720" height="478" loading="lazy" /></a></p>

<p>This isn’t a particularly pretty example of a plot. We’ll be learning much more about creating nice, publication-quality graphics later.</p>

<h2 id="subset-a-data-frame">Subset a data frame</h2>

<p>Now we would like to know how to get specific values from data frames, and where necessary, change the mode of a column of values.</p>

<p>The first thing to remember is that a data frame is two-dimensional (rows and columns). Therefore, to select a specific value we will once again use <code class="language-plaintext highlighter-rouge">[]</code> (bracket) notation, but we will specify more than one value (except in some cases where we are taking a range).</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-subsetting-a-data-frame"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Subsetting a data frame</div>

  <p>Run the following indices and functions to figure out what they return</p>

  <ol>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[1, 1]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[2, 4]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[130, 13]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[2, ]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[-1, ]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[1:4, 1]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[1:10, c("Feature", "Gene.Name")]</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[, c("Gene.Name")]</code></li>
    <li><code class="language-plaintext highlighter-rouge">head(annotatedDEgenes)</code></li>
    <li><code class="language-plaintext highlighter-rouge">tail(annotatedDEgenes)</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes$GeneID</code></li>
    <li><code class="language-plaintext highlighter-rouge">annotatedDEgenes[annotatedDEgenes$Feature == "pseudogene", ]</code></li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[1, 1]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">1</span><span class="p">,</span><span class="m">1</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[2, 4]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="m">4</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">0.1043451</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[130, 13]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">130</span><span class="p">,</span><span class="m">13</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">CG10440</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">Ama</span><span class="w"> </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="n">Ant2</span><span class="w"> </span><span class="n">Argk</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w"> </span><span class="n">bou</span><span class="w"> </span><span class="n">bves</span><span class="w"> </span><span class="n">CAHbeta</span><span class="w"> </span><span class="n">cbt</span><span class="w"> </span><span class="n">CG10063</span><span class="w"> </span><span class="n">CG10365</span><span class="w"> </span><span class="n">CG10440</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">zye</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[2, ]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">2</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w">
        </span><span class="n">GeneID</span><span class="w"> </span><span class="n">Base.mean</span><span class="w">   </span><span class="n">log2.FC.</span><span class="w">   </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.577</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.1043451</span><span class="w">  </span><span class="m">-28.74863</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">    </span><span class="n">chrX</span><span class="w"> </span><span class="m">10780892</span><span class="w">
        </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">sesB</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[-1, ]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">-1</span><span class="p">,</span><span class="w"> </span><span class="p">]</span><span class="w">
         </span><span class="n">GeneID</span><span class="w">   </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">
 </span><span class="m">2</span><span class="w">  </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.57713</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w"> </span><span class="m">-28.748628</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">    </span><span class="n">chrX</span><span class="w">
 </span><span class="m">3</span><span class="w">  </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.84056</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w"> </span><span class="m">-28.225437</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">   </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">4</span><span class="w">  </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.32237</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">  </span><span class="m">27.565988</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">    </span><span class="n">chrX</span><span class="w">
 </span><span class="m">5</span><span class="w">  </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.06728</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w"> </span><span class="m">-22.745964</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">   </span><span class="n">chr3L</span><span class="w">
 </span><span class="m">6</span><span class="w">  </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.90178</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w"> </span><span class="m">-21.889786</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">   </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">7</span><span class="w">  </span><span class="n">FBgn0035085</span><span class="w">   </span><span class="m">928.26381</span><span class="w"> </span><span class="m">-2.414074</span><span class="w"> </span><span class="m">0.11518516</span><span class="w"> </span><span class="m">-20.958204</span><span class="w">  </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w">   </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">8</span><span class="w">  </span><span class="n">FBgn0034736</span><span class="w">   </span><span class="m">330.38302</span><span class="w"> </span><span class="m">-3.018179</span><span class="w"> </span><span class="m">0.15815418</span><span class="w"> </span><span class="m">-19.083774</span><span class="w">  </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w">   </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">9</span><span class="w">  </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">955.45445</span><span class="w"> </span><span class="m">-2.334486</span><span class="w"> </span><span class="m">0.12423003</span><span class="w"> </span><span class="m">-18.791643</span><span class="w">  </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">   </span><span class="n">chr3L</span><span class="w">
 </span><span class="m">10</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w">   </span><span class="m">468.05793</span><span class="w">  </span><span class="m">2.360017</span><span class="w"> </span><span class="m">0.13564397</span><span class="w">  </span><span class="m">17.398615</span><span class="w">  </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">   </span><span class="n">chr3R</span><span class="w">
 </span><span class="c1"># ... with 120 more rows</span><span class="w">

       </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">2</span><span class="w">  </span><span class="m">10780892</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w">
 </span><span class="m">3</span><span class="w">  </span><span class="m">26869237</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
 </span><span class="m">4</span><span class="w">  </span><span class="m">10778953</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w">
 </span><span class="m">5</span><span class="w">  </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w">
 </span><span class="m">6</span><span class="w">  </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w">
 </span><span class="m">7</span><span class="w">  </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG3770</span><span class="w">
 </span><span class="m">8</span><span class="w">  </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG6018</span><span class="w">
 </span><span class="m">9</span><span class="w">    </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">        </span><span class="n">lincRNA</span><span class="w">     </span><span class="n">CR43883</span><span class="w">
 </span><span class="m">10</span><span class="w">  </span><span class="m">6762592</span><span class="w">  </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Ama</span><span class="w">
 </span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[1:4,1]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,</span><span class="m">1</span><span class="p">]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p>annotatedDEgenes[1:10,c(“Feature”,”Gene.Name”)]</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">10</span><span class="p">,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Feature"</span><span class="p">,</span><span class="s2">"Gene.Name"</span><span class="p">)]</span><span class="w">
           </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">1</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Kal1</span><span class="w">
 </span><span class="m">2</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w">
 </span><span class="m">3</span><span class="w">  </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
 </span><span class="m">4</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w">
 </span><span class="m">5</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w">
 </span><span class="m">6</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w">
 </span><span class="m">7</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG3770</span><span class="w">
 </span><span class="m">8</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG6018</span><span class="w">
 </span><span class="m">9</span><span class="w">         </span><span class="n">lincRNA</span><span class="w">     </span><span class="n">CR43883</span><span class="w">
 </span><span class="m">10</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Ama</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[,c("Gene.Name")]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[,</span><span class="nf">c</span><span class="p">(</span><span class="s2">"Gene.Name"</span><span class="p">)]</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">Kal1</span><span class="w">         </span><span class="n">sesB</span><span class="w">         </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">  </span><span class="n">Ant2</span><span class="w">         </span><span class="n">Hml</span><span class="w">          </span><span class="n">CG1544</span><span class="w">       </span><span class="n">CG3770</span><span class="w">
 </span><span class="p">[</span><span class="m">8</span><span class="p">]</span><span class="w"> </span><span class="n">CG6018</span><span class="w">       </span><span class="n">CR43883</span><span class="w">      </span><span class="n">Ama</span><span class="w">          </span><span class="n">CG3168</span><span class="w">       </span><span class="n">CG15695</span><span class="w">      </span><span class="n">CG8500</span><span class="w">       </span><span class="n">Rgk1</span><span class="w">
 </span><span class="p">[</span><span class="m">15</span><span class="p">]</span><span class="w"> </span><span class="n">Sesn</span><span class="w">         </span><span class="n">CG9119</span><span class="w">       </span><span class="n">Sema</span><span class="m">-2</span><span class="n">a</span><span class="w">      </span><span class="n">l</span><span class="p">(</span><span class="m">1</span><span class="p">)</span><span class="n">G0196</span><span class="w">    </span><span class="n">lectin</span><span class="m">-28</span><span class="n">C</span><span class="w">   </span><span class="n">Sox100B</span><span class="w">      </span><span class="n">Hsp27</span><span class="w">
 </span><span class="p">[</span><span class="m">22</span><span class="p">]</span><span class="w"> </span><span class="n">CG34330</span><span class="w">      </span><span class="n">Ugt58Fa</span><span class="w">      </span><span class="n">Src64B</span><span class="w">       </span><span class="n">Tequila</span><span class="w">      </span><span class="n">CG32407</span><span class="w">      </span><span class="n">LpR2</span><span class="w">         </span><span class="n">CG12290</span><span class="w">
 </span><span class="p">[</span><span class="m">29</span><span class="p">]</span><span class="w"> </span><span class="n">ps</span><span class="w">           </span><span class="n">CG8157</span><span class="w">       </span><span class="n">nemy</span><span class="w">         </span><span class="n">bves</span><span class="w">         </span><span class="n">CG8468</span><span class="w">       </span><span class="n">CR43855</span><span class="w">      </span><span class="n">CG13315</span><span class="w">
 </span><span class="p">[</span><span class="m">36</span><span class="p">]</span><span class="w"> </span><span class="n">CG1124</span><span class="w">       </span><span class="n">Reg</span><span class="m">-2</span><span class="w">        </span><span class="n">CG5261</span><span class="w">       </span><span class="n">Treh</span><span class="w">         </span><span class="n">Hsp23</span><span class="w">        </span><span class="n">Gale</span><span class="w">         </span><span class="n">PPO1</span><span class="w">
 </span><span class="p">[</span><span class="m">43</span><span class="p">]</span><span class="w"> </span><span class="n">Picot</span><span class="w">        </span><span class="n">RpS24</span><span class="w">        </span><span class="n">CG14946</span><span class="w">      </span><span class="n">CG10365</span><span class="w">      </span><span class="n">NimC4</span><span class="w">        </span><span class="n">CG42694</span><span class="w">      </span><span class="n">CG33307</span><span class="w">
 </span><span class="p">[</span><span class="m">50</span><span class="p">]</span><span class="w"> </span><span class="n">CG6006</span><span class="w">       </span><span class="n">grk</span><span class="w">          </span><span class="n">CG42369</span><span class="w">      </span><span class="n">GILT1</span><span class="w">        </span><span class="n">CG5001</span><span class="w">       </span><span class="n">Hsp26</span><span class="w">        </span><span class="n">MtnA</span><span class="w">
 </span><span class="p">[</span><span class="m">57</span><span class="p">]</span><span class="w"> </span><span class="n">CG6330</span><span class="w">       </span><span class="n">l</span><span class="p">(</span><span class="m">3</span><span class="p">)</span><span class="n">neo38</span><span class="w">    </span><span class="n">Cyt</span><span class="o">-</span><span class="n">b5</span><span class="o">-</span><span class="n">r</span><span class="w">     </span><span class="n">CG12116</span><span class="w">      </span><span class="n">PPO2</span><span class="w">         </span><span class="n">zye</span><span class="w">          </span><span class="n">tnc</span><span class="w">
 </span><span class="p">[</span><span class="m">64</span><span class="p">]</span><span class="w"> </span><span class="n">CG14257</span><span class="w">      </span><span class="n">GstE2</span><span class="w">        </span><span class="n">olf413</span><span class="w">       </span><span class="n">CG31642</span><span class="w">      </span><span class="n">Nxf3</span><span class="w">         </span><span class="n">CG8292</span><span class="w">       </span><span class="n">CG5397</span><span class="w">
 </span><span class="p">[</span><span class="m">71</span><span class="p">]</span><span class="w"> </span><span class="n">Mctp</span><span class="w">         </span><span class="n">CG7777</span><span class="w">       </span><span class="n">CG4019</span><span class="w">       </span><span class="n">ChLD3</span><span class="w">        </span><span class="n">Argk</span><span class="w">         </span><span class="n">bou</span><span class="w">          </span><span class="n">Mct1</span><span class="w">
 </span><span class="p">[</span><span class="m">78</span><span class="p">]</span><span class="w"> </span><span class="n">CG9416</span><span class="w">       </span><span class="n">CG12512</span><span class="w">      </span><span class="n">CG32625</span><span class="w">      </span><span class="n">Npc2b</span><span class="w">        </span><span class="n">CG30463</span><span class="w">      </span><span class="n">Eip74EF</span><span class="w">      </span><span class="n">CAHbeta</span><span class="w">
 </span><span class="p">[</span><span class="m">85</span><span class="p">]</span><span class="w"> </span><span class="n">CG14326</span><span class="w">      </span><span class="n">Cyp6w1</span><span class="w">       </span><span class="n">CG3457</span><span class="w">       </span><span class="n">wdp</span><span class="w">          </span><span class="n">cbt</span><span class="w">          </span><span class="n">CrebA</span><span class="w">        </span><span class="n">Cyp4d20</span><span class="w">
 </span><span class="p">[</span><span class="m">92</span><span class="p">]</span><span class="w"> </span><span class="n">CR45973</span><span class="w">      </span><span class="n">Sr</span><span class="o">-</span><span class="n">CIV</span><span class="w">       </span><span class="n">SPH93</span><span class="w">        </span><span class="n">Cyp9h1</span><span class="w">       </span><span class="n">sug</span><span class="w">          </span><span class="n">CG14516</span><span class="w">      </span><span class="n">Ggamma30A</span><span class="w">
 </span><span class="p">[</span><span class="m">99</span><span class="p">]</span><span class="w"> </span><span class="n">CG4297</span><span class="w">       </span><span class="n">CG9503</span><span class="w">       </span><span class="n">CG10063</span><span class="w">      </span><span class="n">CG8501</span><span class="w">       </span><span class="n">CG43336</span><span class="w">      </span><span class="n">stet</span><span class="w">         </span><span class="n">CG8852</span><span class="w">
 </span><span class="p">[</span><span class="m">106</span><span class="p">]</span><span class="w"> </span><span class="n">CG15080</span><span class="w">      </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w">        </span><span class="n">sn</span><span class="w">           </span><span class="n">CG9989</span><span class="w">       </span><span class="n">mlt</span><span class="w">          </span><span class="n">CG2444</span><span class="w">       </span><span class="n">CG17189</span><span class="w">
 </span><span class="p">[</span><span class="m">113</span><span class="p">]</span><span class="w"> </span><span class="n">RhoGAP1A</span><span class="w">     </span><span class="n">CG13743</span><span class="w">      </span><span class="n">TwdlU</span><span class="w">        </span><span class="n">Galphaf</span><span class="w">      </span><span class="n">CG32335</span><span class="w">      </span><span class="n">E</span><span class="p">(</span><span class="n">spl</span><span class="p">)</span><span class="n">m3</span><span class="o">-</span><span class="n">HLH</span><span class="w"> </span><span class="n">CG16749</span><span class="w">
 </span><span class="p">[</span><span class="m">120</span><span class="p">]</span><span class="w"> </span><span class="n">CG34434</span><span class="w">      </span><span class="n">CG43799</span><span class="w">      </span><span class="n">Pde6</span><span class="w">         </span><span class="n">Sr</span><span class="o">-</span><span class="n">CIII</span><span class="w">      </span><span class="n">NimC2</span><span class="w">        </span><span class="n">CG5895</span><span class="w">       </span><span class="n">SP1173</span><span class="w">
 </span><span class="p">[</span><span class="m">127</span><span class="p">]</span><span class="w"> </span><span class="n">CG1311</span><span class="w">       </span><span class="n">CG14856</span><span class="w">      </span><span class="n">CG6356</span><span class="w">       </span><span class="n">CG10440</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">Ama</span><span class="w"> </span><span class="n">Amy</span><span class="o">-</span><span class="n">p</span><span class="w"> </span><span class="n">Ant2</span><span class="w"> </span><span class="n">Argk</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w"> </span><span class="n">bou</span><span class="w"> </span><span class="n">bves</span><span class="w"> </span><span class="n">CAHbeta</span><span class="w"> </span><span class="n">cbt</span><span class="w"> </span><span class="n">CG10063</span><span class="w"> </span><span class="n">CG10365</span><span class="w"> </span><span class="n">CG10440</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">zye</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">head(annotatedDEgenes)</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">  </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">      </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.9743</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w">  </span><span class="m">-30.74090</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24141394</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.5771</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w">  </span><span class="m">-28.74863</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">    </span><span class="n">chrX</span><span class="w"> </span><span class="m">10780892</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.8406</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w">  </span><span class="m">-28.22544</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">26869237</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.3224</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">   </span><span class="m">27.56599</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">    </span><span class="n">chrX</span><span class="w"> </span><span class="m">10778953</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.0673</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w">  </span><span class="m">-22.74596</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">   </span><span class="n">chr3L</span><span class="w"> </span><span class="m">13846053</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.9018</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w">  </span><span class="m">-21.88979</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">31196915</span><span class="w">

 </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Kal1</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">tail(annotatedDEgenes)</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">tail</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">)</span><span class="w">
          </span><span class="n">GeneID</span><span class="w"> </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">    </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">      </span><span class="n">P.value</span><span class="w">        </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">
 </span><span class="m">125</span><span class="w"> </span><span class="n">FBgn0036560</span><span class="w">  </span><span class="m">27.71424</span><span class="w">  </span><span class="m">1.089000</span><span class="w"> </span><span class="m">0.2327527</span><span class="w">   </span><span class="m">4.678786</span><span class="w"> </span><span class="m">2.885785e-06</span><span class="w"> </span><span class="m">6.671531e-05</span><span class="w">   </span><span class="n">chr3L</span><span class="w"> </span><span class="m">16035484</span><span class="w">
 </span><span class="m">126</span><span class="w"> </span><span class="n">FBgn0035710</span><span class="w">  </span><span class="m">26.16177</span><span class="w">  </span><span class="m">1.048979</span><span class="w"> </span><span class="m">0.2329221</span><span class="w">   </span><span class="m">4.503559</span><span class="w"> </span><span class="m">6.682472e-06</span><span class="w"> </span><span class="m">1.436480e-04</span><span class="w">   </span><span class="n">chr3L</span><span class="w">  </span><span class="m">6689326</span><span class="w">
 </span><span class="m">127</span><span class="w"> </span><span class="n">FBgn0035523</span><span class="w">  </span><span class="m">70.99820</span><span class="w">  </span><span class="m">1.004819</span><span class="w"> </span><span class="m">0.2237625</span><span class="w">   </span><span class="m">4.490560</span><span class="w"> </span><span class="m">7.103600e-06</span><span class="w"> </span><span class="m">1.523189e-04</span><span class="w">   </span><span class="n">chr3L</span><span class="w">  </span><span class="m">4277961</span><span class="w">
 </span><span class="m">128</span><span class="w"> </span><span class="n">FBgn0038261</span><span class="w">  </span><span class="m">44.27058</span><span class="w">  </span><span class="m">1.006264</span><span class="w"> </span><span class="m">0.2241243</span><span class="w">   </span><span class="m">4.489756</span><span class="w"> </span><span class="m">7.130482e-06</span><span class="w"> </span><span class="m">1.525141e-04</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">14798985</span><span class="w">
 </span><span class="m">129</span><span class="w"> </span><span class="n">FBgn0039178</span><span class="w">  </span><span class="m">23.55006</span><span class="w">  </span><span class="m">1.040917</span><span class="w"> </span><span class="m">0.2326264</span><span class="w">   </span><span class="m">4.474631</span><span class="w"> </span><span class="m">7.654326e-06</span><span class="w"> </span><span class="m">1.629061e-04</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24283954</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">FBgn0034636</span><span class="w">  </span><span class="m">24.77052</span><span class="w"> </span><span class="m">-1.028531</span><span class="w"> </span><span class="m">0.2321678</span><span class="w">  </span><span class="m">-4.430119</span><span class="w"> </span><span class="m">9.418101e-06</span><span class="w"> </span><span class="m">1.951185e-04</span><span class="w">   </span><span class="n">chr2R</span><span class="w"> </span><span class="m">21560245</span><span class="w">

 </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w">
 </span><span class="m">125</span><span class="w"> </span><span class="m">16037227</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">CG5895</span><span class="w">
 </span><span class="m">126</span><span class="w">  </span><span class="m">6703521</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">SP1173</span><span class="w">
 </span><span class="m">127</span><span class="w">  </span><span class="m">4281585</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">CG1311</span><span class="w">
 </span><span class="m">128</span><span class="w"> </span><span class="m">14801163</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">   </span><span class="n">CG14856</span><span class="w">
 </span><span class="m">129</span><span class="w"> </span><span class="m">24288617</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">    </span><span class="n">CG6356</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="m">21576035</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">   </span><span class="n">CG10440</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes$GeneID</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">GeneID</span><span class="w">
 </span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="n">FBgn0034736</span><span class="w">
 </span><span class="p">[</span><span class="m">9</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0264475</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0029896</span><span class="w"> </span><span class="n">FBgn0038832</span><span class="w"> </span><span class="n">FBgn0037754</span><span class="w"> </span><span class="n">FBgn0264753</span><span class="w"> </span><span class="n">FBgn0034897</span><span class="w"> </span><span class="n">FBgn0035189</span><span class="w">
 </span><span class="p">[</span><span class="m">17</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0011260</span><span class="w"> </span><span class="n">FBgn0027279</span><span class="w"> </span><span class="n">FBgn0040099</span><span class="w"> </span><span class="n">FBgn0024288</span><span class="w"> </span><span class="n">FBgn0001226</span><span class="w"> </span><span class="n">FBgn0085359</span><span class="w"> </span><span class="n">FBgn0040091</span><span class="w"> </span><span class="n">FBgn0262733</span><span class="w">
 </span><span class="p">[</span><span class="m">25</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0023479</span><span class="w"> </span><span class="n">FBgn0052407</span><span class="w"> </span><span class="n">FBgn0051092</span><span class="w"> </span><span class="n">FBgn0039419</span><span class="w"> </span><span class="n">FBgn0261552</span><span class="w"> </span><span class="n">FBgn0034010</span><span class="w"> </span><span class="n">FBgn0261673</span><span class="w"> </span><span class="n">FBgn0031150</span><span class="w">
 </span><span class="p">[</span><span class="m">33</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0033913</span><span class="w"> </span><span class="n">FBgn0264437</span><span class="w"> </span><span class="n">FBgn0040827</span><span class="w"> </span><span class="n">FBgn0037290</span><span class="w"> </span><span class="n">FBgn0016715</span><span class="w"> </span><span class="n">FBgn0031912</span><span class="w"> </span><span class="n">FBgn0003748</span><span class="w"> </span><span class="n">FBgn0001224</span><span class="w">
 </span><span class="p">[</span><span class="m">41</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0035147</span><span class="w"> </span><span class="n">FBgn0261362</span><span class="w"> </span><span class="n">FBgn0024315</span><span class="w"> </span><span class="n">FBgn0261596</span><span class="w"> </span><span class="n">FBgn0032405</span><span class="w"> </span><span class="n">FBgn0039109</span><span class="w"> </span><span class="n">FBgn0260011</span><span class="w"> </span><span class="n">FBgn0261584</span><span class="w">
 </span><span class="p">[</span><span class="m">49</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0053307</span><span class="w"> </span><span class="n">FBgn0063649</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">FBgn0259715</span><span class="w"> </span><span class="n">FBgn0038149</span><span class="w"> </span><span class="n">FBgn0031322</span><span class="w"> </span><span class="n">FBgn0001225</span><span class="w"> </span><span class="n">FBgn0002868</span><span class="w">
 </span><span class="p">[</span><span class="m">57</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039464</span><span class="w"> </span><span class="n">FBgn0265276</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0030041</span><span class="w"> </span><span class="n">FBgn0033367</span><span class="w"> </span><span class="n">FBgn0036985</span><span class="w"> </span><span class="n">FBgn0039257</span><span class="w"> </span><span class="n">FBgn0039479</span><span class="w">
 </span><span class="p">[</span><span class="m">65</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0063498</span><span class="w"> </span><span class="n">FBgn0037153</span><span class="w"> </span><span class="n">FBgn0051642</span><span class="w"> </span><span class="n">FBgn0263232</span><span class="w"> </span><span class="n">FBgn0032004</span><span class="w"> </span><span class="n">FBgn0031327</span><span class="w"> </span><span class="n">FBgn0034389</span><span class="w"> </span><span class="n">FBgn0033635</span><span class="w">
 </span><span class="p">[</span><span class="m">73</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0034885</span><span class="w"> </span><span class="n">FBgn0032598</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0261284</span><span class="w"> </span><span class="n">FBgn0023549</span><span class="w"> </span><span class="n">FBgn0034438</span><span class="w"> </span><span class="n">FBgn0031703</span><span class="w"> </span><span class="n">FBgn0052625</span><span class="w">
 </span><span class="p">[</span><span class="m">81</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0038198</span><span class="w"> </span><span class="n">FBgn0050463</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0037646</span><span class="w"> </span><span class="n">FBgn0038528</span><span class="w"> </span><span class="n">FBgn0033065</span><span class="w"> </span><span class="n">FBgn0024984</span><span class="w"> </span><span class="n">FBgn0034718</span><span class="w">
 </span><span class="p">[</span><span class="m">89</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0043364</span><span class="w"> </span><span class="n">FBgn0004396</span><span class="w"> </span><span class="n">FBgn0035344</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w"> </span><span class="n">FBgn0031547</span><span class="w"> </span><span class="n">FBgn0032638</span><span class="w"> </span><span class="n">FBgn0033775</span><span class="w"> </span><span class="n">FBgn0033782</span><span class="w">
 </span><span class="p">[</span><span class="m">97</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039640</span><span class="w"> </span><span class="n">FBgn0267252</span><span class="w"> </span><span class="n">FBgn0031258</span><span class="w"> </span><span class="n">FBgn0030598</span><span class="w"> </span><span class="n">FBgn0035727</span><span class="w"> </span><span class="n">FBgn0033724</span><span class="w"> </span><span class="n">FBgn0263041</span><span class="w"> </span><span class="n">FBgn0020248</span><span class="w">
 </span><span class="p">[</span><span class="m">105</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0031548</span><span class="w"> </span><span class="n">FBgn0034391</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0003447</span><span class="w"> </span><span class="n">FBgn0039593</span><span class="w"> </span><span class="n">FBgn0265512</span><span class="w"> </span><span class="n">FBgn0030326</span><span class="w"> </span><span class="n">FBgn0039485</span><span class="w">
 </span><span class="p">[</span><span class="m">113</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0025836</span><span class="w"> </span><span class="n">FBgn0033368</span><span class="w"> </span><span class="n">FBgn0037223</span><span class="w"> </span><span class="n">FBgn0010223</span><span class="w"> </span><span class="n">FBgn0063667</span><span class="w"> </span><span class="n">FBgn0002609</span><span class="w"> </span><span class="n">FBgn0037678</span><span class="w"> </span><span class="n">FBgn0250904</span><span class="w">
 </span><span class="p">[</span><span class="m">121</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0264343</span><span class="w"> </span><span class="n">FBgn0038237</span><span class="w"> </span><span class="n">FBgn0020376</span><span class="w"> </span><span class="n">FBgn0028939</span><span class="w"> </span><span class="n">FBgn0036560</span><span class="w"> </span><span class="n">FBgn0035710</span><span class="w"> </span><span class="n">FBgn0035523</span><span class="w"> </span><span class="n">FBgn0038261</span><span class="w">
 </span><span class="p">[</span><span class="m">129</span><span class="p">]</span><span class="w"> </span><span class="n">FBgn0039178</span><span class="w"> </span><span class="n">FBgn0034636</span><span class="w">
 </span><span class="m">130</span><span class="w"> </span><span class="n">Levels</span><span class="o">:</span><span class="w"> </span><span class="n">FBgn0000071</span><span class="w"> </span><span class="n">FBgn0000079</span><span class="w"> </span><span class="n">FBgn0000116</span><span class="w"> </span><span class="n">FBgn0000406</span><span class="w"> </span><span class="n">FBgn0000567</span><span class="w"> </span><span class="n">FBgn0001137</span><span class="w"> </span><span class="n">...</span><span class="w"> </span><span class="n">FBgn0267635</span><span class="w">
</span></code></pre></div>        </div>
      </li>
      <li>
        <p><code class="language-plaintext highlighter-rouge">annotatedDEgenes[annotatedDEgenes$Feature == "pseudogene",]</code></p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Feature</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"pseudogene"</span><span class="p">,]</span><span class="w">
</span><span class="n">GeneID</span><span class="w"> </span><span class="n">Base.mean</span><span class="w">   </span><span class="n">log2.FC.</span><span class="w">    </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">      </span><span class="n">P.value</span><span class="w">        </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Seqname</span><span class="w">   </span><span class="n">Start</span><span class="w">
</span><span class="m">115</span><span class="w"> </span><span class="n">FBgn0037223</span><span class="w">  </span><span class="m">37.00783</span><span class="w"> </span><span class="m">-1.161706</span><span class="w"> </span><span class="m">0.2297393</span><span class="w">  </span><span class="m">-5.056627</span><span class="w"> </span><span class="m">4.267378e-07</span><span class="w"> </span><span class="m">1.192225e-05</span><span class="w">   </span><span class="n">chr3R</span><span class="w"> </span><span class="m">4243605</span><span class="w">
</span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">    </span><span class="n">Feature</span><span class="w"> </span><span class="n">Gene.Name</span><span class="w">
</span><span class="m">115</span><span class="w"> </span><span class="m">4245511</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">pseudogene</span><span class="w">     </span><span class="n">TwdlU</span><span class="w">
</span></code></pre></div>        </div>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>The subsetting notation is very similar to what we learned for vectors. The key differences include:</p>
<ul>
  <li>2 values, separated by commas: <code class="language-plaintext highlighter-rouge">data.frame[row, column]</code></li>
  <li>Use of a colon between the numbers (<code class="language-plaintext highlighter-rouge">start:stop, inclusive</code>) for a continuous range of numbers</li>
  <li>Passing a vector using <code class="language-plaintext highlighter-rouge">c()</code> for a non continuous set of numbers, pass a vector</li>
  <li>Index using the name of a column(s) by passing them as vectors using <code class="language-plaintext highlighter-rouge">c()</code></li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-a-new-data-frame-by-subsetting-an-existing-one"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create a new data frame by subsetting an existing one</div>
  <ol>
    <li>
      <p>Create a new data frame containing only observations from the “+” strand</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># create a new data frame containing only observations from the "+" strand</span><span class="w">
</span><span class="n">annotatedDEgenes_plusStrand</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">,]</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Check the dimension of the new data frame</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># check the dimension of the data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="nf">dim</span><span class="p">(</span><span class="n">annotatedDEgenes_plusStrand</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">72</span><span class="w"> </span><span class="m">13</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Get a summary of the data frame</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># get a summary of the data frame</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">summary</span><span class="p">(</span><span class="n">annotatedDEgenes_plusStrand</span><span class="p">)</span><span class="w">
</span><span class="n">GeneID</span><span class="w">           </span><span class="n">Base.mean</span><span class="w">           </span><span class="n">log2.FC.</span><span class="w">         </span><span class="n">StdErr</span><span class="w">            </span><span class="n">Wald.Stats</span><span class="w">
</span><span class="n">FBgn0000071</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">   </span><span class="m">23.55</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-4.1485</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.08603</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">-30.741</span><span class="w">
</span><span class="n">FBgn0001224</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">   </span><span class="m">96.04</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-1.3664</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.12523</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">-11.223</span><span class="w">
</span><span class="n">FBgn0001226</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w">  </span><span class="m">275.91</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">-1.0739</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.15648</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">-5.290</span><span class="w">
</span><span class="n">FBgn0002609</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">1365.34</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">-0.3133</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">0.16087</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">-2.729</span><span class="w">
</span><span class="n">FBgn0003447</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">988.33</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">1.1856</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.19743</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w">  </span><span class="m">7.592</span><span class="w">
</span><span class="n">FBgn0003748</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">28915.90</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">2.3600</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">0.23275</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">17.399</span><span class="w">
</span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="w">    </span><span class="o">:</span><span class="m">66</span><span class="w">

</span><span class="n">P.value</span><span class="w">             </span><span class="n">P.adj</span><span class="w">              </span><span class="n">Seqname</span><span class="w">    </span><span class="n">Start</span><span class="w">              </span><span class="n">End</span><span class="w">               </span><span class="n">Strand</span><span class="w">
</span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2L</span><span class="o">:</span><span class="m">12</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">127448</span><span class="w">   </span><span class="n">Min.</span><span class="w">   </span><span class="o">:</span><span class="w">  </span><span class="m">140340</span><span class="w">   </span><span class="o">-:</span><span class="w"> </span><span class="m">0</span><span class="w">
</span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr2R</span><span class="o">:</span><span class="m">14</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7377082</span><span class="w">   </span><span class="m">1</span><span class="n">st</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="w"> </span><span class="m">7379907</span><span class="w">   </span><span class="o">+:</span><span class="m">72</span><span class="w">
</span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">0.000e+00</span><span class="w">   </span><span class="n">chr3L</span><span class="o">:</span><span class="m">15</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">14068548</span><span class="w">   </span><span class="n">Median</span><span class="w"> </span><span class="o">:</span><span class="m">14070644</span><span class="w">
</span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">4.556e-07</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">1.013e-05</span><span class="w">   </span><span class="n">chr3R</span><span class="o">:</span><span class="m">24</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">14422245</span><span class="w">   </span><span class="n">Mean</span><span class="w">   </span><span class="o">:</span><span class="m">14431849</span><span class="w">
</span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">6.000e-12</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">3.200e-10</span><span class="w">   </span><span class="n">chrX</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">7</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">21339535</span><span class="w">   </span><span class="m">3</span><span class="n">rd</span><span class="w"> </span><span class="n">Qu.</span><span class="o">:</span><span class="m">21378634</span><span class="w">
</span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">7.654e-06</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">1.629e-04</span><span class="w">              </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31196915</span><span class="w">   </span><span class="n">Max.</span><span class="w">   </span><span class="o">:</span><span class="m">31203722</span><span class="w">

</span><span class="n">Feature</span><span class="w">             </span><span class="n">Gene.Name</span><span class="w">
</span><span class="n">lincRNA</span><span class="w">       </span><span class="o">:</span><span class="w"> </span><span class="m">2</span><span class="w">   </span><span class="n">Ama</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">protein_coding</span><span class="o">:</span><span class="m">69</span><span class="w">   </span><span class="n">bou</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
</span><span class="n">pseudogene</span><span class="w">    </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">   </span><span class="n">bves</span><span class="w">   </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="n">CAHbeta</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="n">CG10365</span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="n">CG1124</span><span class="w"> </span><span class="o">:</span><span class="w"> </span><span class="m">1</span><span class="w">
                    </span><span class="p">(</span><span class="n">Other</span><span class="p">)</span><span class="o">:</span><span class="m">66</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<h2 id="change-type-of-columns">Change type of columns</h2>

<p>Sometimes, it is possible that R will misinterpret the type of data represented in a data frame, or store that data in a mode which prevents you from operating on the data the way you wish. For example, a long list of gene names isn’t usually thought of as a categorical variable, the way that your experimental condition (e.g. control, treatment) might be. More importantly, some R packages you use to analyze your data may expect characters as input, not factors. At other times (such as plotting or some statistical analyses) a factor may be more appropriate. Ultimately, you should know how to change the mode of an object.</p>

<p>First, its very important to recognize that coercion happens in R all the time. This can be a good thing when R gets it right, or a bad thing when the result is not what you expect.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-format-gene-names"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Format gene names</div>
  <ol>
    <li>
      <p>Check the structure of the <code class="language-plaintext highlighter-rouge">Gene.name</code> column</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">str</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span><span class="n">Factor</span><span class="w"> </span><span class="n">w</span><span class="o">/</span><span class="w"> </span><span class="m">130</span><span class="w"> </span><span class="n">levels</span><span class="w"> </span><span class="s2">"Ama"</span><span class="p">,</span><span class="s2">"Amy-p"</span><span class="p">,</span><span class="n">..</span><span class="o">:</span><span class="w"> </span><span class="m">88</span><span class="w"> </span><span class="m">113</span><span class="w"> </span><span class="m">5</span><span class="w"> </span><span class="m">3</span><span class="w"> </span><span class="m">84</span><span class="w"> </span><span class="m">26</span><span class="w"> </span><span class="m">41</span><span class="w"> </span><span class="m">53</span><span class="w"> </span><span class="m">69</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Try to coerce the column into numeric</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="nf">as.numeric</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w">  </span><span class="m">88</span><span class="w"> </span><span class="m">113</span><span class="w">   </span><span class="m">5</span><span class="w">   </span><span class="m">3</span><span class="w">  </span><span class="m">84</span><span class="w">  </span><span class="m">26</span><span class="w">  </span><span class="m">41</span><span class="w">  </span><span class="m">53</span><span class="w">  </span><span class="m">69</span><span class="w">   </span><span class="m">1</span><span class="w">  </span><span class="m">33</span><span class="w">  </span><span class="m">27</span><span class="w">  </span><span class="m">60</span><span class="w"> </span><span class="m">109</span><span class="w"> </span><span class="m">114</span><span class="w">  </span><span class="m">63</span><span class="w"> </span><span class="m">112</span><span class="w">
</span><span class="p">[</span><span class="m">18</span><span class="p">]</span><span class="w">  </span><span class="m">89</span><span class="w">  </span><span class="m">91</span><span class="w"> </span><span class="m">116</span><span class="w">  </span><span class="m">87</span><span class="w">  </span><span class="m">38</span><span class="w"> </span><span class="m">128</span><span class="w"> </span><span class="m">121</span><span class="w"> </span><span class="m">124</span><span class="w">  </span><span class="m">35</span><span class="w">  </span><span class="m">92</span><span class="w">  </span><span class="m">15</span><span class="w"> </span><span class="m">107</span><span class="w">  </span><span class="m">57</span><span class="w">  </span><span class="m">97</span><span class="w">   </span><span class="m">7</span><span class="w">  </span><span class="m">59</span><span class="w">  </span><span class="m">68</span><span class="w">
</span><span class="p">[</span><span class="m">35</span><span class="p">]</span><span class="w">  </span><span class="m">18</span><span class="w">  </span><span class="m">13</span><span class="w"> </span><span class="m">108</span><span class="w">  </span><span class="m">49</span><span class="w"> </span><span class="m">126</span><span class="w">  </span><span class="m">85</span><span class="w">  </span><span class="m">78</span><span class="w"> </span><span class="m">105</span><span class="w"> </span><span class="m">104</span><span class="w"> </span><span class="m">111</span><span class="w">  </span><span class="m">24</span><span class="w">  </span><span class="m">11</span><span class="w">  </span><span class="m">99</span><span class="w">  </span><span class="m">44</span><span class="w">  </span><span class="m">37</span><span class="w">  </span><span class="m">52</span><span class="w">  </span><span class="m">82</span><span class="w">
</span><span class="n">...</span><span class="w">
</span></code></pre></div>      </div>

      <p>Strangely, it works! Almost. Instead of giving an error message, R returns numeric values, which in this case are the integers assigned to the levels in this factor. This kind of behavior can lead to hard-to-find bugs, for example when we do have numbers in a factor, and we get numbers from a coercion. If we don’t look carefully, we may not notice a problem.</p>
    </li>
    <li>
      <p>Coerce the column into character (which makes sense for gene names)</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>R will sometimes coerce without you asking for it (implicit coercion). You can explicitly change it (explicit coercion).</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-avoiding-needless-coercion"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-avoiding-needless-coercion" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Avoiding needless coercion</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>One way to avoid needless coercion when importing a data frame using functions such as <code class="language-plaintext highlighter-rouge">read.csv()</code> is to set the argument <code class="language-plaintext highlighter-rouge">StringsAsFactors</code> to <code class="language-plaintext highlighter-rouge">FALSE</code>. By default, this argument is <code class="language-plaintext highlighter-rouge">TRUE</code>. Setting it to <code class="language-plaintext highlighter-rouge">FALSE</code> will treat any non-numeric column to a character type. Going through the <code class="language-plaintext highlighter-rouge">read.csv() </code>documentation, you will also see you can explicitly type your columns using the <code class="language-plaintext highlighter-rouge">colClasses</code> argument. Other R packages (such as the Tidyverse <code class="language-plaintext highlighter-rouge">readr</code>) don’t have this particular conversion issue, but many packages will still try to guess a data type.</p>
</blockquote>

<h2 id="inspect-a-numerical-column-using-math-sorting-renaming">Inspect a numerical column using math, sorting, renaming</h2>

<p>Here are a few operations that don’t need much explanation, but which are good to know. There are lots of arithmetic functions you may want to apply to your data frame. For example, you can use functions like <code class="language-plaintext highlighter-rouge">mean()</code>, <code class="language-plaintext highlighter-rouge">min()</code>, <code class="language-plaintext highlighter-rouge">max()</code> on an individual column.</p>

<p>Let’s look at the <code class="language-plaintext highlighter-rouge">Base.mean</code> column, i.e. the mean of normalized counts of all samples, normalizing for sequencing depth and library composition.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-the-base-mean-column-and-sort"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect the `Base.mean` column and sort</div>
  <ol>
    <li>
      <p>Get the maximum value of <code class="language-plaintext highlighter-rouge">Base.mean</code> column</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">max</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">65114.84</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Sort the data frame by values in the <code class="language-plaintext highlighter-rouge">Base.mean</code> column</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">sorted_by_BaseMean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">sorted_by_BaseMean</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">19.15076</span><span class="w"> </span><span class="m">23.55006</span><span class="w"> </span><span class="m">24.77052</span><span class="w"> </span><span class="m">26.16177</span><span class="w"> </span><span class="m">27.71424</span><span class="w"> </span><span class="m">31.58769</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="question">
        <div class="box-title question-title" id="question-changing-the-order"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Changing the order</div>
        <p>The <code class="language-plaintext highlighter-rouge">order()</code> function lists values in increasing order by default. Look at the documentation for this function and change <code class="language-plaintext highlighter-rouge">sorted_by_BaseMean</code> to start with differentially expressed means with the greatest base mean.</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">sorted_by_BaseMean</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">order</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">,</span><span class="w"> </span><span class="n">decreasing</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="kc">TRUE</span><span class="p">),</span><span class="w"> </span><span class="p">]</span><span class="w">
</span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">sorted_by_BaseMean</span><span class="o">$</span><span class="n">Base.mean</span><span class="p">)</span><span class="w">
</span><span class="p">[</span><span class="m">1</span><span class="p">]</span><span class="w"> </span><span class="m">65114.841</span><span class="w"> </span><span class="m">28915.903</span><span class="w"> </span><span class="m">24584.306</span><span class="w"> </span><span class="m">22435.886</span><span class="w">  </span><span class="m">8903.630</span><span class="w">  </span><span class="m">8573.815</span><span class="w">
</span></code></pre></div>          </div>

        </blockquote>
      </blockquote>
    </li>
    <li>
      <p>Coerce the column into character (which makes sense for gene names)</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">as.character</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Gene.name</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<h2 id="save-the-data-frame-to-a-file">Save the data frame to a file</h2>

<p>We can save data frame as table in our Galaxy history to use it as input for other tools.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-save-data-frame-in-galaxy-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Save data frame in Galaxy history</div>

  <ol>
    <li>
      <p>Save the <code class="language-plaintext highlighter-rouge">annotatedDEgenes_plusStrand</code> object to a <code class="language-plaintext highlighter-rouge">.csv</code> file using the <code class="language-plaintext highlighter-rouge">write.csv()</code> function:</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">write.csv</span><span class="p">(</span><span class="n">annotatedDEgenes_plusStrand</span><span class="p">,</span><span class="w"> </span><span class="n">file</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="s2">"annotatedDEgenes_plusStrand.csv"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p><strong><em>TODO</em></strong>: <em>Description on how to save a file from RStudio to Galaxy</em></p>
    </li>
  </ol>
</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">write.csv()</code> function has some additional arguments listed in the help, but at a minimum you need to tell it what data frame to write to file, and give a path to a file name in quotes (if you only provide a file name, the file will be written in the current working directory).</p>

<h1 id="aggregate-and-analyze-with-dplyr">Aggregate and Analyze with dplyr</h1>

<p>Selecting columns and rows using bracket subsetting is handy, but it can be cumbersome and difficult to read, especially for complicated operations.</p>

<p>The functions we’ve been using so far, like <code class="language-plaintext highlighter-rouge">str()</code>, come built into R. By using packages, we can access more functions. Here, we would like to use <a href="https://cran.r-project.org/package=dplyr"><code class="language-plaintext highlighter-rouge">dplyr</code></a> package provides a number of very useful functions for manipulating data frames in a way that will reduce repetition, reduce the probability of making errors, and probably even save you some typing. As an added bonus, you might even find the <code class="language-plaintext highlighter-rouge">dplyr</code> grammar easier to read.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-is-dplyr"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What is dplyr?</div>

  <p>The package <code class="language-plaintext highlighter-rouge">dplyr</code> is a fairly new (2014) package that tries to provide easy tools for the most common data manipulation tasks. It is built to work directly with data frames. The thinking behind it was largely inspired by the package <code class="language-plaintext highlighter-rouge">plyr</code> which has been in use for some time but suffered from being slow in some cases.<code class="language-plaintext highlighter-rouge"> dplyr</code> addresses this by porting much of the computation to C++. An additional feature is the ability to work with data stored directly in an external database. The benefits of doing this are that the data can be managed natively in a relational database, queries can be conducted on that database, and only the results of the query returned.</p>

  <p>This addresses a common problem with R in that all operations are conducted in memory and thus the amount of data you can work with is limited by available memory. The database connections essentially remove that limitation in that you can have a database of many 100s GB, conduct queries on it directly and pull back just what you need for analysis in R.</p>
</blockquote>

<p>To use <code class="language-plaintext highlighter-rouge">dplyr</code>, we need first to install the package and then load it to be able to use it.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-install-and-load-dyplr"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Install and load dyplr</div>

  <ol>
    <li>
      <p>Load dyplr</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## load dyplr package</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="s2">"dplyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-didn-t-work"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-didn-t-work" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Didn't work?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>You might need to install dyplr first</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"dplyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>        </div>

        <p>You might get asked to choose a CRAN mirror – this is asking you to choose a site to download the package from. The choice doesn’t matter too much.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We only need to install a package once per computer, but we need to load it every time we open a new R session and want to use that package.</p>

<h2 id="select-columns-and-filter-rows">Select columns and filter rows</h2>

<p>To select columns, instead with bracket, we could use the <code class="language-plaintext highlighter-rouge">select</code> function of <code class="language-plaintext highlighter-rouge">dplyr</code></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-select-column"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Select column</div>

  <ol>
    <li>
      <p>Select <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">Start</code>, <code class="language-plaintext highlighter-rouge">End</code>, <code class="language-plaintext highlighter-rouge">Strand</code> columns in <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="p">)</span><span class="w">
         </span><span class="n">GeneID</span><span class="w">    </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">
</span><span class="m">1</span><span class="w">   </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">2</span><span class="w">   </span><span class="n">FBgn0003360</span><span class="w"> </span><span class="m">10780892</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w">
</span><span class="m">3</span><span class="w">   </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">26869237</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w">
</span><span class="m">4</span><span class="w">   </span><span class="n">FBgn0025111</span><span class="w"> </span><span class="m">10778953</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w">
</span><span class="m">5</span><span class="w">   </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">6</span><span class="w">   </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">7</span><span class="w">   </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">8</span><span class="w">   </span><span class="n">FBgn0034736</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">9</span><span class="w">   </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="m">10</span><span class="w">  </span><span class="n">FBgn0000071</span><span class="w">  </span><span class="m">6762592</span><span class="w">  </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>

      <p>The first argument to this function is the data frame (<code class="language-plaintext highlighter-rouge">annotatedDEgenes</code>), and the subsequent arguments are the columns to keep.</p>
    </li>
    <li>
      <p>Select all columns <em>except</em> certain ones</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
         </span><span class="n">GeneID</span><span class="w">   </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w">    </span><span class="n">Start</span><span class="w"> </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">    </span><span class="n">Feature</span><span class="w">    </span><span class="n">Gene.Name</span><span class="w">
</span><span class="m">1</span><span class="w">   </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.97430</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w"> </span><span class="m">-30.740902</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w"> </span><span class="m">24141394</span><span class="w">  </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Kal1</span><span class="w">
</span><span class="m">2</span><span class="w">   </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.57713</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w"> </span><span class="m">-28.748628</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w"> </span><span class="m">10780892</span><span class="w">  </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">sesB</span><span class="w">
</span><span class="m">3</span><span class="w">   </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.84056</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w"> </span><span class="m">-28.225437</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w"> </span><span class="m">26869237</span><span class="w">  </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">  </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w">
</span><span class="m">4</span><span class="w">   </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.32237</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">  </span><span class="m">27.565988</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w"> </span><span class="m">10778953</span><span class="w">  </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Ant2</span><span class="w">
</span><span class="m">5</span><span class="w">   </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.06728</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w"> </span><span class="m">-22.745964</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w"> </span><span class="m">13846053</span><span class="w">  </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Hml</span><span class="w">
</span><span class="m">6</span><span class="w">   </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.90178</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w"> </span><span class="m">-21.889786</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w"> </span><span class="m">31196915</span><span class="w">  </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG1544</span><span class="w">
</span><span class="m">7</span><span class="w">   </span><span class="n">FBgn0035085</span><span class="w">   </span><span class="m">928.26381</span><span class="w"> </span><span class="m">-2.414074</span><span class="w"> </span><span class="m">0.11518516</span><span class="w"> </span><span class="m">-20.958204</span><span class="w">  </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w"> </span><span class="m">24945138</span><span class="w">   </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG3770</span><span class="w">
</span><span class="m">8</span><span class="w">   </span><span class="n">FBgn0034736</span><span class="w">   </span><span class="m">330.38302</span><span class="w"> </span><span class="m">-3.018179</span><span class="w"> </span><span class="m">0.15815418</span><span class="w"> </span><span class="m">-19.083774</span><span class="w">  </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w"> </span><span class="m">22550093</span><span class="w">   </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG6018</span><span class="w">
</span><span class="m">9</span><span class="w">   </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">955.45445</span><span class="w"> </span><span class="m">-2.334486</span><span class="w"> </span><span class="m">0.12423003</span><span class="w"> </span><span class="m">-18.791643</span><span class="w">  </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">   </span><span class="n">lincRNA</span><span class="w">      </span><span class="n">CR43883</span><span class="w">
</span><span class="m">10</span><span class="w">  </span><span class="n">FBgn0000071</span><span class="w">   </span><span class="m">468.05793</span><span class="w">  </span><span class="m">2.360017</span><span class="w"> </span><span class="m">0.13564397</span><span class="w">  </span><span class="m">17.398615</span><span class="w">  </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">  </span><span class="m">6762592</span><span class="w">   </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w">  </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Ama</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><code class="language-plaintext highlighter-rouge">dplyr</code> also provides useful functions to select columns based on their names. For instance, <code class="language-plaintext highlighter-rouge">starts_with()</code> allows you to select columns that starts with specific letters.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-select-columns-that-start-with-the-letter-quot-p-quot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Select columns that start with the letter &quot;P&quot;</div>

  <ol>
    <li>
      <p>Select columns that start with the letter “P”</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">starts_with</span><span class="p">(</span><span class="s2">"P."</span><span class="p">))</span><span class="w">

          </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w">
</span><span class="m">1</span><span class="w">   </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">
</span><span class="m">2</span><span class="w">   </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">
</span><span class="m">3</span><span class="w">   </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">
</span><span class="m">4</span><span class="w">   </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">
</span><span class="m">5</span><span class="w">   </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">
</span><span class="m">6</span><span class="w">   </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">
</span><span class="m">7</span><span class="w">    </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w">
</span><span class="m">8</span><span class="w">    </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w">
</span><span class="m">9</span><span class="w">    </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">
</span><span class="m">10</span><span class="w">   </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-selecting-on-multiple-conditions"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Selecting on multiple conditions</div>

  <p>Create a table that contains all the columns with the letter “s” in their name except for the column “Wald.Stats”, and that contains the column “End”.</p>

  <p>Hint: look at the help for the function <code class="language-plaintext highlighter-rouge">starts_with()</code> we’ve just covered.</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">select</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">contains</span><span class="p">(</span><span class="s2">"s"</span><span class="p">),</span><span class="w"> </span><span class="o">-</span><span class="n">Wald.Stats</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>Similarly to choose rows, we can use the <code class="language-plaintext highlighter-rouge">filter()</code> function.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-rows"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter rows</div>

  <ol>
    <li>
      <p>Filter for rows on the forward strand</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">)</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">   </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">    </span><span class="n">Gene.Name</span><span class="w">
</span><span class="m">1</span><span class="w">  </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.97430</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w"> </span><span class="m">-30.740902</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Kal1</span><span class="w">
</span><span class="m">2</span><span class="w">  </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.06728</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w"> </span><span class="m">-22.745964</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">      </span><span class="n">chr3L</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Hml</span><span class="w">
</span><span class="m">3</span><span class="w">  </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.90178</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w"> </span><span class="m">-21.889786</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG1544</span><span class="w">
</span><span class="m">4</span><span class="w">  </span><span class="n">FBgn0035085</span><span class="w">   </span><span class="m">928.26381</span><span class="w"> </span><span class="m">-2.414074</span><span class="w"> </span><span class="m">0.11518516</span><span class="w"> </span><span class="m">-20.958204</span><span class="w">  </span><span class="m">1.579343e-97</span><span class="w">  </span><span class="m">1.935147e-94</span><span class="w">      </span><span class="n">chr2R</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG3770</span><span class="w">
</span><span class="m">5</span><span class="w">  </span><span class="n">FBgn0034736</span><span class="w">   </span><span class="m">330.38302</span><span class="w"> </span><span class="m">-3.018179</span><span class="w"> </span><span class="m">0.15815418</span><span class="w"> </span><span class="m">-19.083774</span><span class="w">  </span><span class="m">3.444661e-81</span><span class="w">  </span><span class="m">3.693107e-78</span><span class="w">      </span><span class="n">chr2R</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG6018</span><span class="w">
</span><span class="m">6</span><span class="w">  </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">955.45445</span><span class="w"> </span><span class="m">-2.334486</span><span class="w"> </span><span class="m">0.12423003</span><span class="w"> </span><span class="m">-18.791643</span><span class="w">  </span><span class="m">8.840041e-79</span><span class="w">  </span><span class="m">8.424559e-76</span><span class="w">      </span><span class="n">chr3L</span><span class="w"> </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="o">+</span><span class="w">        </span><span class="n">lincRNA</span><span class="w">      </span><span class="n">CR43883</span><span class="w">
</span><span class="m">7</span><span class="w">  </span><span class="n">FBgn0000071</span><span class="w">   </span><span class="m">468.05793</span><span class="w">  </span><span class="m">2.360017</span><span class="w"> </span><span class="m">0.13564397</span><span class="w">  </span><span class="m">17.398615</span><span class="w">  </span><span class="m">8.452137e-68</span><span class="w">  </span><span class="m">7.249398e-65</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">6762592</span><span class="w">  </span><span class="m">6765261</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">          </span><span class="n">Ama</span><span class="w">
</span><span class="m">8</span><span class="w">  </span><span class="n">FBgn0038832</span><span class="w">   </span><span class="m">429.85033</span><span class="w"> </span><span class="m">-2.340711</span><span class="w"> </span><span class="m">0.15483651</span><span class="w"> </span><span class="m">-15.117308</span><span class="w">  </span><span class="m">1.245307e-51</span><span class="w">  </span><span class="m">8.900828e-49</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">20842139</span><span class="w"> </span><span class="m">20844981</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG15695</span><span class="w">
</span><span class="m">9</span><span class="w">  </span><span class="n">FBgn0037754</span><span class="w">   </span><span class="m">394.46410</span><span class="w"> </span><span class="m">-2.071994</span><span class="w"> </span><span class="m">0.14494069</span><span class="w"> </span><span class="m">-14.295459</span><span class="w">  </span><span class="m">2.335613e-46</span><span class="w">  </span><span class="m">1.540966e-43</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">9784652</span><span class="w">  </span><span class="m">9789323</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">       </span><span class="n">CG8500</span><span class="w">
</span><span class="m">10</span><span class="w"> </span><span class="n">FBgn0034897</span><span class="w">  </span><span class="m">1340.88978</span><span class="w"> </span><span class="m">-1.795107</span><span class="w"> </span><span class="m">0.12629203</span><span class="w"> </span><span class="m">-14.213934</span><span class="w">  </span><span class="m">7.508655e-46</span><span class="w">  </span><span class="m">4.293449e-43</span><span class="w">      </span><span class="n">chr2R</span><span class="w"> </span><span class="m">23713899</span><span class="w"> </span><span class="m">23734846</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Sesn</span><span class="w">
</span><span class="c1"># ... with 120 more rows</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-equivalent-in-base-r-code"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Equivalent in base R code</div>
        <p>Note that this is equivalent to the base R code below, but is easier to read!</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="p">[</span><span class="n">annotatedDEgenes</span><span class="o">$</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">,]</span><span class="w">
</span></code></pre></div>        </div>
      </blockquote>

      <p><code class="language-plaintext highlighter-rouge">filter()</code> keeps all the rows that match the conditions that are provided</p>
    </li>
    <li>
      <p>Filter for rows corresponding to genes in Chromosome X or 2R</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## rows for genes in Chromosome X or 2R</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s2">"chrX"</span><span class="p">,</span><span class="w"> </span><span class="s2">"chr2R"</span><span class="p">))</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Filter for rows where the log2 fold change is greater than 2</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## rows where the log2 fold change is greater than 2</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p><code class="language-plaintext highlighter-rouge">filter()</code> allows to combine multiple conditions. They need to be separated using a <code class="language-plaintext highlighter-rouge">,</code> as arguments to the function, they will be combined using the <code class="language-plaintext highlighter-rouge">&amp;</code> (AND) logical operator. If you need to use the <code class="language-plaintext highlighter-rouge">|</code> (OR) logical operator, you can specify it explicitly:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-rows-for-multiple-conditions"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter rows for multiple conditions</div>

  <ol>
    <li>
      <p>Filter for genes on chrX and with adjusted p-value &lt;= 1e-100</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## both are equivalent</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">P.adj</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">1e-100</span><span class="p">)</span><span class="w">
</span><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="p">,</span><span class="w"> </span><span class="n">P.adj</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">1e-100</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Filter for genes on chrX and with log2 Fold Change smaller than -2 or larger than 2</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chrX"</span><span class="p">,</span><span class="w"> </span><span class="p">(</span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-2</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span><span class="p">))</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-practising-with-conditionals"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Practising with conditionals</div>

  <p>Select all the rows for genes that:</p>
  <ul>
    <li>start after position \(1e^6\) (one million) and before position \(2e^6\) (included) in their chromosome</li>
    <li>have a log2 fold change greater than 1 or an adjusted p-value less than \(1e^{-75}\).</li>
  </ul>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">filter</span><span class="p">(</span><span class="n">annotatedDEgenes</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">1e6</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">End</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">2e6</span><span class="p">,</span><span class="w"> </span><span class="p">(</span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="m">1</span><span class="w"> </span><span class="o">|</span><span class="w"> </span><span class="n">P.adj</span><span class="w"> </span><span class="o">&lt;</span><span class="w"> </span><span class="m">1e-75</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>But what if you wanted to select and filter?</p>

<p>We can do this with <strong>pipes</strong>. Pipes, are a fairly recent addition to R. They let you take the output of one function and send it directly to the next, which is useful when you need to do many things to the same data set. It was possible to do this before pipes were added to R, but it was much messier and more difficult. Pipes in R look like <code class="language-plaintext highlighter-rouge">%&gt;%</code> and are made available via the <code class="language-plaintext highlighter-rouge">magrittr</code> package, which is installed as part of <code class="language-plaintext highlighter-rouge">dplyr</code>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-pipes-on-rstudio"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Pipes on RStudio</div>

  <p>With RStudio, you can type the pipe with</p>
  <ul>
    <li><kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> if you’re using a PC</li>
    <li><kbd>Cmd</kbd> + <kbd>Shift</kbd> + <kbd>M</kbd> if you’re using a Mac</li>
  </ul>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-select-and-filter"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Select and filter</div>

  <ol>
    <li>
      <p>Filter for genes on forward strand, select <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">Start</code>, <code class="language-plaintext highlighter-rouge">End</code>, <code class="language-plaintext highlighter-rouge">Chromosome</code> and display the first results using pipes</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">filter</span><span class="p">(</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">head</span><span class="p">()</span><span class="w">

       </span><span class="n">GeneID</span><span class="w">    </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Chromosome</span><span class="w">
</span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
</span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
</span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
</span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0034736</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
</span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>In the above code, we use the pipe to</p>
<ol>
  <li>Send the <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> dataset through <code class="language-plaintext highlighter-rouge">filter()</code></li>
  <li>Keep rows for genes on the ‘+’ strand of their chromosome</li>
  <li>Select through <code class="language-plaintext highlighter-rouge">select()</code> to keep only the <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">Start</code>, <code class="language-plaintext highlighter-rouge">End</code>, and <code class="language-plaintext highlighter-rouge">Chromosome</code> columns</li>
  <li>Show only the first 6 rows with <code class="language-plaintext highlighter-rouge">head</code></li>
</ol>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-no-need-to-give-1st-argument-to-the-functions"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: No need to give 1st argument to the functions</div>

  <p>Since <code class="language-plaintext highlighter-rouge">%&gt;%</code> takes the object on its left and passes it as the first argument to the function on its right, we don’t need to explicitly include the data frame as an argument to the <code class="language-plaintext highlighter-rouge">filter()</code> and <code class="language-plaintext highlighter-rouge">select()</code> functions any more.</p>

</blockquote>

<p>Some may find it helpful to read the pipe like the word “then”. For instance, in the above example, we took the data frame <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code>, <em>then</em> we <code class="language-plaintext highlighter-rouge">filter</code>ed for rows where <code class="language-plaintext highlighter-rouge">Strand</code> was +, <em>then</em> we <code class="language-plaintext highlighter-rouge">select</code>ed the <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">Start</code>, <code class="language-plaintext highlighter-rouge">End</code>, and <code class="language-plaintext highlighter-rouge">Chromosome</code> columns, <em>then</em> we showed only the first six rows.</p>

<p>The <code class="language-plaintext highlighter-rouge">dplyr</code> functions by themselves are somewhat simple, but by combining them into linear workflows with the pipe, we can accomplish complex manipulations of data frames.</p>

<p>If we want to create a new object with this smaller version of the data we can do so by assigning it to a new object.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-select-and-filter-to-create-new-object"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Select and filter to create new object</div>

  <ol>
    <li>
      <p>Create an <code class="language-plaintext highlighter-rouge">plus_strand_genes</code> from <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> after filter for genes on forward strand and selection of <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">Start</code>, <code class="language-plaintext highlighter-rouge">End</code>, <code class="language-plaintext highlighter-rouge">Chromosome</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">plus_strand_genes</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
 </span><span class="n">filter</span><span class="p">(</span><span class="n">Strand</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"+"</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
 </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>
    </li>
    <li>
      <p>Check the first 6 rows</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">head</span><span class="p">(</span><span class="n">plus_strand_genes</span><span class="p">)</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">    </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Chromosome</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="n">chr3R</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0035085</span><span class="w"> </span><span class="m">24945138</span><span class="w"> </span><span class="m">24946636</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0034736</span><span class="w"> </span><span class="m">22550093</span><span class="w"> </span><span class="m">22552113</span><span class="w">      </span><span class="n">chr2R</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0264475</span><span class="w">   </span><span class="m">820758</span><span class="w">   </span><span class="m">821512</span><span class="w">      </span><span class="n">chr3L</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>This new object includes all of the data from this sample.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-pipes-and-up-regulation"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Pipes and up-regulation</div>

  <p>Starting with the <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> data frame, use pipes to</p>
  <ul>
    <li>subset the data to include only observations from chromosome 3L,with the log2 fold change is at least 2</li>
    <li>retain only the columns <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">P.adj</code>, and <code class="language-plaintext highlighter-rouge">log2.FC.</code>.</li>
  </ul>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">filter</span><span class="p">(</span><span class="n">Chromosome</span><span class="w"> </span><span class="o">==</span><span class="w"> </span><span class="s2">"chr3L"</span><span class="w"> </span><span class="o">&amp;</span><span class="w"> </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w"> </span><span class="m">2</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">P.adj</span><span class="p">,</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<h2 id="create-columns">Create columns</h2>

<p>Frequently we want to create new columns based on the values in existing columns, e.g. to do unit conversions or find the ratio of values in two columns. For this we can use the <code class="language-plaintext highlighter-rouge">dplyr</code> function <code class="language-plaintext highlighter-rouge">mutate()</code>.</p>

<p>In the data frame, there is a column titled <code class="language-plaintext highlighter-rouge">log2.FC.</code>. This is the logarithmically-adjusted representation of the fold-change observed in expression of the genes in the transcriptomic experiment from which this data is derived. We can calculate the observed expression level relative to the reference according to the formula: \(FC = 2^{log2.FC.}\)</p>

<p>Let’s create a column <code class="language-plaintext highlighter-rouge">FC</code> to our <code class="language-plaintext highlighter-rouge">annotatedDEgenes</code> data frame that show the observed expression as a multiple of the reference level.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-select-and-filter-to-create-new-object-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Select and filter to create new object</div>

  <ol>
    <li>
      <p>Create a a column <code class="language-plaintext highlighter-rouge">FC</code> that show the observed expression as a multiple of the reference level</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">mutate</span><span class="p">(</span><span class="n">FC</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">2</span><span class="w"> </span><span class="o">**</span><span class="w"> </span><span class="n">log2.FC.</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">head</span><span class="p">()</span><span class="w">
        </span><span class="n">GeneID</span><span class="w">  </span><span class="n">Base.mean</span><span class="w">  </span><span class="n">log2.FC.</span><span class="w">     </span><span class="n">StdErr</span><span class="w"> </span><span class="n">Wald.Stats</span><span class="w">       </span><span class="n">P.value</span><span class="w">         </span><span class="n">P.adj</span><span class="w"> </span><span class="n">Chromosome</span><span class="w">  </span><span class="n">Start</span><span class="w">      </span><span class="n">End</span><span class="w"> </span><span class="n">Strand</span><span class="w">        </span><span class="n">Feature</span><span class="w">   </span><span class="n">Gene.Name</span><span class="w">   </span><span class="n">FC</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">FBgn0039155</span><span class="w">  </span><span class="m">1086.9743</span><span class="w"> </span><span class="m">-4.148450</span><span class="w"> </span><span class="m">0.13494887</span><span class="w">  </span><span class="m">-30.74090</span><span class="w"> </span><span class="m">1.617921e-207</span><span class="w"> </span><span class="m">1.387691e-203</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">24141394</span><span class="w"> </span><span class="m">24147490</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Kal1</span><span class="w"> </span><span class="m">0.05638871</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">FBgn0003360</span><span class="w">  </span><span class="m">6409.5771</span><span class="w"> </span><span class="m">-2.999777</span><span class="w"> </span><span class="m">0.10434506</span><span class="w">  </span><span class="m">-28.74863</span><span class="w"> </span><span class="m">9.422382e-182</span><span class="w"> </span><span class="m">4.040788e-178</span><span class="w">       </span><span class="n">chrX</span><span class="w"> </span><span class="m">10780892</span><span class="w"> </span><span class="m">10786958</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">sesB</span><span class="w"> </span><span class="m">0.12501930</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">FBgn0026562</span><span class="w"> </span><span class="m">65114.8406</span><span class="w"> </span><span class="m">-2.380164</span><span class="w"> </span><span class="m">0.08432692</span><span class="w">  </span><span class="m">-28.22544</span><span class="w"> </span><span class="m">2.850473e-175</span><span class="w"> </span><span class="m">8.149503e-172</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">26869237</span><span class="w"> </span><span class="m">26871995</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w"> </span><span class="n">BM</span><span class="m">-40</span><span class="o">-</span><span class="n">SPARC</span><span class="w"> </span><span class="m">0.19208755</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">FBgn0025111</span><span class="w">  </span><span class="m">2192.3224</span><span class="w">  </span><span class="m">2.699939</span><span class="w"> </span><span class="m">0.09794457</span><span class="w">   </span><span class="m">27.56599</span><span class="w"> </span><span class="m">2.846764e-167</span><span class="w"> </span><span class="m">6.104174e-164</span><span class="w">       </span><span class="n">chrX</span><span class="w"> </span><span class="m">10778953</span><span class="w"> </span><span class="m">10786907</span><span class="w">      </span><span class="o">-</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">        </span><span class="n">Ant2</span><span class="w"> </span><span class="m">6.49774366</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">FBgn0029167</span><span class="w">  </span><span class="m">5430.0673</span><span class="w"> </span><span class="m">-2.105062</span><span class="w"> </span><span class="m">0.09254660</span><span class="w">  </span><span class="m">-22.74596</span><span class="w"> </span><span class="m">1.573283e-114</span><span class="w"> </span><span class="m">2.698810e-111</span><span class="w">      </span><span class="n">chr3L</span><span class="w"> </span><span class="m">13846053</span><span class="w"> </span><span class="m">13860001</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">         </span><span class="n">Hml</span><span class="w"> </span><span class="m">0.23244132</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">FBgn0039827</span><span class="w">   </span><span class="m">390.9018</span><span class="w"> </span><span class="m">-3.503014</span><span class="w"> </span><span class="m">0.16002962</span><span class="w">  </span><span class="m">-21.88979</span><span class="w"> </span><span class="m">3.250384e-106</span><span class="w"> </span><span class="m">4.646424e-103</span><span class="w">      </span><span class="n">chr3R</span><span class="w"> </span><span class="m">31196915</span><span class="w"> </span><span class="m">31203722</span><span class="w">      </span><span class="o">+</span><span class="w"> </span><span class="n">protein_coding</span><span class="w">      </span><span class="n">CG1544</span><span class="w"> </span><span class="m">0.08820388</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-selected-mutation"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Selected mutation</div>

  <p>Create a column with the gene length and just look at the <code class="language-plaintext highlighter-rouge">GeneID</code>, <code class="language-plaintext highlighter-rouge">Chromosome</code>, <code class="language-plaintext highlighter-rouge">Start</code>, <code class="language-plaintext highlighter-rouge">End</code>, and <code class="language-plaintext highlighter-rouge">Length</code> columns</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">mutate</span><span class="p">(</span><span class="n">Length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">End</span><span class="o">-</span><span class="n">Start</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">select</span><span class="p">(</span><span class="n">GeneID</span><span class="p">,</span><span class="w"> </span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">Start</span><span class="p">,</span><span class="w"> </span><span class="n">End</span><span class="p">,</span><span class="w"> </span><span class="n">Length</span><span class="p">)</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<h2 id="group-and-summarize-data">Group and summarize data</h2>

<p>Many data analysis tasks can be approached using the “split-apply-combine” paradigm: split the data into groups, apply some analysis to each group, and
then combine the results.</p>

<p><code class="language-plaintext highlighter-rouge">dplyr</code> makes this very easy through the use of the <code class="language-plaintext highlighter-rouge">group_by()</code> function, which splits the data into groups. When the data is grouped in this way <code class="language-plaintext highlighter-rouge">summarize()</code> can be used to collapse each group into a single-row summary, by applying an aggregating or summary function to each group.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-group-and-summarize-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Group and summarize data</div>

  <ol>
    <li>
      <p>Group by <code class="language-plaintext highlighter-rouge">Chromosome</code> and extract the number rows of data for each chromosome</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">summarize</span><span class="p">(</span><span class="n">n</span><span class="p">())</span><span class="w">
 </span><span class="c1"># A tibble: 5 x 2</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">`n()`</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">         </span><span class="m">24</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">         </span><span class="m">31</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">         </span><span class="m">27</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">         </span><span class="m">32</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">          </span><span class="m">16</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>Here the summary function used was <code class="language-plaintext highlighter-rouge">n()</code> to find the count for each group. Because it’s a common operation, the <code class="language-plaintext highlighter-rouge">dplyr</code> verb, <code class="language-plaintext highlighter-rouge">count()</code> is a “shortcut” that combines both <code class="language-plaintext highlighter-rouge">group_by</code> and <code class="language-plaintext highlighter-rouge">summarize</code>.</p>

<p>We can also apply many other functions to individual columns to get other summary statistics.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-group-and-summarize-data-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Group and summarize data</div>

  <ol>
    <li>
      <p>View the highest fold change (<code class="language-plaintext highlighter-rouge">log2.FC.</code>) for each chromsome</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">summarize</span><span class="p">(</span><span class="nf">max</span><span class="p">(</span><span class="n">log2.FC.</span><span class="p">))</span><span class="w">
 </span><span class="c1"># A tibble: 5 x 2</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">`max(log2.FC.)`</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">               </span><span class="o">&lt;</span><span class="n">dbl</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">                </span><span class="m">2.15</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">                </span><span class="m">2.41</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">                </span><span class="m">2.43</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">                </span><span class="m">2.36</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">                 </span><span class="m">2.70</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-summarizing-groups"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Summarizing groups</div>

  <p>What are the longest genes in each chromosome?</p>

  <p>Hint: the function <code class="language-plaintext highlighter-rouge">abs()</code> returns the absolute value.</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">mutate</span><span class="p">(</span><span class="n">GeneLength</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">abs</span><span class="p">(</span><span class="n">End</span><span class="o">-</span><span class="n">Start</span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">summarize</span><span class="p">(</span><span class="n">max_length</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="nf">max</span><span class="p">(</span><span class="n">GeneLength</span><span class="p">)))</span><span class="w">
</span><span class="c1"># A tibble: 5 x 2</span><span class="w">
  </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">max_length</span><span class="w">
  </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">           </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
</span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">           </span><span class="m">45932</span><span class="w">
</span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">           </span><span class="m">59682</span><span class="w">
</span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">          </span><span class="m">111613</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">           </span><span class="m">44404</span><span class="w">
</span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">            </span><span class="m">28429</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-missing-data-and-base-functions"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Missing data and base functions</div>

  <p>R has many base functions like <code class="language-plaintext highlighter-rouge">mean()</code>, <code class="language-plaintext highlighter-rouge">median()</code>, <code class="language-plaintext highlighter-rouge">min()</code>, and <code class="language-plaintext highlighter-rouge">max()</code> that are useful to compute summary statistics. These are also called “built-in functions” because they come with R and don’t require that you install any additional packages.</p>

  <p>By default, all <strong>R functions operating on vectors that contains missing data will return NA</strong>. It’s a way to make sure that users know they have missing data, and make a conscious decision on how to deal with it. When dealing with simple statistics like the mean, the easiest way to ignore <code class="language-plaintext highlighter-rouge">NA</code> (the missing data) is to use <code class="language-plaintext highlighter-rouge">na.rm = TRUE</code> (<code class="language-plaintext highlighter-rouge">rm</code> stands for remove).</p>
</blockquote>

<p><code class="language-plaintext highlighter-rouge">group_by()</code> can also take multiple column names.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-counting"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Counting</div>

  <p>How many genes are found on each strand of each chromosome?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">count</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="p">)</span><span class="w">
</span><span class="c1"># A tibble: 10 x 3</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">Strand</span><span class="w">     </span><span class="n">n</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">  </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">12</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">17</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">14</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
 </span><span class="m">6</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">15</span><span class="w">
 </span><span class="m">7</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">-</span><span class="w">          </span><span class="m">8</span><span class="w">
 </span><span class="m">8</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">24</span><span class="w">
 </span><span class="m">9</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">-</span><span class="w">          </span><span class="m">9</span><span class="w">
</span><span class="m">10</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">+</span><span class="w">          </span><span class="m">7</span><span class="w">
</span></code></pre></div>    </div>
  </blockquote>
</blockquote>

<h2 id="reshape-data-frames">Reshape data frames</h2>

<p>While the tidy format is useful to analyze and plot data in R, it can sometimes be useful to transform the “long” tidy format, into the wide format. This transformation can be done with the <code class="language-plaintext highlighter-rouge">spread()</code> function provided by the <code class="language-plaintext highlighter-rouge">tidyr</code> package (also part of the <code class="language-plaintext highlighter-rouge">tidyverse</code>).</p>

<p><code class="language-plaintext highlighter-rouge">spread()</code> takes a data frame as the first argument, and two subsequent arguments:</p>
<ol>
  <li>the name of the column whose values will become the column names</li>
  <li>the name of the column whose values will fill the cells in the wide data</li>
</ol>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-spread-a-key-value-pair-across-multiple-columns"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Spread a key-value pair across multiple columns</div>

  <ol>
    <li>
      <p>Install and load <code class="language-plaintext highlighter-rouge">tidyr</code> library</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## load dyplr package</span><span class="w">
</span><span class="n">library</span><span class="p">(</span><span class="s2">"tidyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>      </div>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-didn-t-work-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-didn-t-work-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Didn't work?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>You might need to install tidyr first</p>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">install.packages</span><span class="p">(</span><span class="s2">"tidyr"</span><span class="p">)</span><span class="w">
</span></code></pre></div>        </div>
      </blockquote>
    </li>
    <li>
      <p>Create <code class="language-plaintext highlighter-rouge">annotatedDEgenes_wide</code> with chromosomes as row, strand as column and number of genes as values</p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes_wide</span><span class="w"> </span><span class="o">&lt;-</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">group_by</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">Strand</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">summarize</span><span class="p">(</span><span class="n">n</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">n</span><span class="p">())</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">spread</span><span class="p">(</span><span class="n">Strand</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">)</span><span class="w">
 </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes_wide</span><span class="w">
 </span><span class="c1"># A tibble: 5 x 3</span><span class="w">
 </span><span class="c1"># Groups:   Chromosome [5]</span><span class="w">
   </span><span class="n">Chromosome</span><span class="w">   </span><span class="n">`+`</span><span class="w">   </span><span class="n">`-`</span><span class="w">
   </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w"> </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
 </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">         </span><span class="m">12</span><span class="w">    </span><span class="m">12</span><span class="w">
 </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">         </span><span class="m">14</span><span class="w">    </span><span class="m">17</span><span class="w">
 </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">         </span><span class="m">15</span><span class="w">    </span><span class="m">12</span><span class="w">
 </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">         </span><span class="m">24</span><span class="w">     </span><span class="m">8</span><span class="w">
 </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">           </span><span class="m">7</span><span class="w">     </span><span class="m">9</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>The opposite operation of <code class="language-plaintext highlighter-rouge">spread()</code> is taken care by <code class="language-plaintext highlighter-rouge">gather()</code>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-gather-columns-into-key-value-pairs"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Gather columns into key-value pairs</div>

  <ol>
    <li>
      <p>Create a data frame with chromosomes, strand and number as columns from <code class="language-plaintext highlighter-rouge">annotatedDEgenes_wide</code></p>

      <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="w"> </span><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes_wide</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
    </span><span class="n">gather</span><span class="p">(</span><span class="n">Strand</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">,</span><span class="w"> </span><span class="o">-</span><span class="n">Chromosome</span><span class="p">)</span><span class="w">
 </span><span class="c1"># A tibble: 10 x 3</span><span class="w">
 </span><span class="c1"># Groups:   Chromosome [5]</span><span class="w">
    </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">Strand</span><span class="w">     </span><span class="n">n</span><span class="w">
    </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">      </span><span class="o">&lt;</span><span class="n">chr</span><span class="o">&gt;</span><span class="w">  </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
  </span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
  </span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">17</span><span class="w">
  </span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">-</span><span class="w">         </span><span class="m">12</span><span class="w">
  </span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">-</span><span class="w">          </span><span class="m">8</span><span class="w">
  </span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">-</span><span class="w">          </span><span class="m">9</span><span class="w">
  </span><span class="m">6</span><span class="w"> </span><span class="n">chr2L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">12</span><span class="w">
  </span><span class="m">7</span><span class="w"> </span><span class="n">chr2R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">14</span><span class="w">
  </span><span class="m">8</span><span class="w"> </span><span class="n">chr3L</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">15</span><span class="w">
  </span><span class="m">9</span><span class="w"> </span><span class="n">chr3R</span><span class="w">      </span><span class="o">+</span><span class="w">         </span><span class="m">24</span><span class="w">
 </span><span class="m">10</span><span class="w"> </span><span class="n">chrX</span><span class="w">       </span><span class="o">+</span><span class="w">          </span><span class="m">7</span><span class="w">
</span></code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>We specify the names of the new columns, and here add <code class="language-plaintext highlighter-rouge">-Chromosome</code> as this column shouldn’t be affected by the reshaping:</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-categorising-expression-levels"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: Categorising expression levels</div>

  <ol>
    <li>
      <p>Classify each gene as either “up-regulated” (fold change &gt; 1) or “down-regulated” (fold change &lt; 1) and create a table with <code class="language-plaintext highlighter-rouge">Chromosome</code> as rows, the two new labels as columns, and the number of genes in the cells.</p>

      <blockquote class="solution">
        <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <div class="language-R highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="o">&gt;</span><span class="w"> </span><span class="n">annotatedDEgenes</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">mutate</span><span class="p">(</span><span class="n">exp_cat</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">case_when</span><span class="p">(</span><span class="w">
     </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&gt;=</span><span class="w">  </span><span class="m">1</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"up-regulated"</span><span class="p">,</span><span class="w">
     </span><span class="n">log2.FC.</span><span class="w"> </span><span class="o">&lt;=</span><span class="w"> </span><span class="m">-1</span><span class="w"> </span><span class="o">~</span><span class="w"> </span><span class="s2">"down-regulated"</span><span class="w">
   </span><span class="p">))</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">count</span><span class="p">(</span><span class="n">Chromosome</span><span class="p">,</span><span class="w"> </span><span class="n">exp_cat</span><span class="p">)</span><span class="w"> </span><span class="o">%&gt;%</span><span class="w">
   </span><span class="n">spread</span><span class="p">(</span><span class="n">exp_cat</span><span class="p">,</span><span class="w"> </span><span class="n">n</span><span class="p">)</span><span class="w">
</span><span class="c1"># A tibble: 5 x 3</span><span class="w">
  </span><span class="n">Chromosome</span><span class="w"> </span><span class="n">`down-regulated`</span><span class="w"> </span><span class="n">`up-regulated`</span><span class="w">
  </span><span class="o">&lt;</span><span class="n">fct</span><span class="o">&gt;</span><span class="w">                 </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">          </span><span class="o">&lt;</span><span class="n">int</span><span class="o">&gt;</span><span class="w">
</span><span class="m">1</span><span class="w"> </span><span class="n">chr2L</span><span class="w">                    </span><span class="m">17</span><span class="w">              </span><span class="m">7</span><span class="w">
</span><span class="m">2</span><span class="w"> </span><span class="n">chr2R</span><span class="w">                    </span><span class="m">16</span><span class="w">             </span><span class="m">15</span><span class="w">
</span><span class="m">3</span><span class="w"> </span><span class="n">chr3L</span><span class="w">                    </span><span class="m">11</span><span class="w">             </span><span class="m">16</span><span class="w">
</span><span class="m">4</span><span class="w"> </span><span class="n">chr3R</span><span class="w">                    </span><span class="m">18</span><span class="w">             </span><span class="m">14</span><span class="w">
</span><span class="m">5</span><span class="w"> </span><span class="n">chrX</span><span class="w">                      </span><span class="m">8</span><span class="w">              </span><span class="m">8</span><span class="w">
</span></code></pre></div>        </div>

      </blockquote>
    </li>
    <li>
      <p>How could the code above be improved?</p>

      <p>Hint: think about the assumptions that we made about the data when writing this solution.</p>
    </li>
  </ol>
</blockquote>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(plausible !== undefined){
                            plausible('TutorialComplete', {callback: tutorial_finish_finish, props: {path: document.location.pathname}})
                          } else {
                            tutorial_finish_finish();
                          }
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>It is easy to import data into R from tabular formats. However, you still need to check that R has imported and interpreted your data correctly.</p>
</li>
                        
                        <li><p>Just like Galaxy, base R’s capabilities can be further enhanced by software packages developed by the user community.</p>
</li>
                        
                        <li><p>Use the <code class="language-plaintext highlighter-rouge">dplyr</code> package to manipulate data frames.</p>
</li>
                        
                        <li><p>Pipes can be used to combine simple operations into complex procedures.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/data-science/faqs/">FAQ page for the Foundations of Data Science topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=data-science/r-advanced"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=data-science/r-advanced">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            carpentries, Bérénice Batut, Fotis E. Psomopoulos, Toby Hodges, gallantries,  <b>Advanced R in Galaxy (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/r-advanced/tutorial.html">https://training.galaxyproject.org/training-material/topics/data-science/tutorials/r-advanced/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{data-science-r-advanced,
author = "carpentries and Bérénice Batut and Fotis E. Psomopoulos and Toby Hodges and gallantries",
	title = "Advanced R in Galaxy (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/data-science/tutorials/r-advanced/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/transcriptomics/tutorials/rna-seq-counts-to-viz-in-r/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: RNA Seq Counts to Viz in R</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


                    </ul>
                </blockquote>
                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/data-science/tutorials/r-advanced/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">5</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.4">2</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.2">1</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>December 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: generally excellent
					<b>Disliked</b>: the file used is different from the tutorial
				</li>
				
		    </ul>
		    
		    <b>February 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Very interesting libraries (tidyr and dplyr) for data analysis
					<b>Disliked</b>: More self training exercise
				</li>
				
		    </ul>
		    
		    <b>December 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: so easy and clear
					
				</li>
				
		    </ul>
		    
		    <b>July 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Easy to follow and very usefull examples
					
				</li>
				
		    </ul>
		    
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00102">gxy.io/GTN:T00102</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/data-science/tutorials/r-advanced/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/data-science/tutorials/r-advanced/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/9cc53bcb522fc6b7739affd8a51b9fe23f7defcf">9cc53bc</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.20cf1f17.js'></script>

	
	<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	
	

    </body>
</html>