<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities / Ecology</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@275c49ff3b6670060b65ab23356b29e3dd1b9f97",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.105e595f.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Hands-on: Hands-on: Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-03-07 15:47:52 +0000">
<meta name="DC.creator" content="Yvan Le Bras">
<meta name="DC.creator" content="Glinez Thibaud"><meta name="description" content="Learn to analyse Ecological data through Galaxy.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Ecology / Hands-on: Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities">
        <meta property="og:description" content="Learn to analyse Ecological data through Galaxy.">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "dateModified": "2024-03-07 15:47:52 +0000",
  "datePublished": "2024-03-04 15:18:02 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Hands-on: Hands-on: Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Draft",
  "identifier": "https://gxy.io/GTN:T00422",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Ecology",
    "description": "Learn to analyse Ecological data through Galaxy.",
    "url": "https://training.galaxyproject.org/training-material/topics/ecology/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on: Hands-on: Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities",
  "url": "https://training.galaxyproject.org/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html",
  "version": 45,
  "timeRequired": "PT30M",
  "teaches": "Explain the necessity of using such tools when producing ecological metadata\n - Learn how to create rich metadata using Ecological Metadata Language (EML) standard\n - Learn how to update EML metadata",
  "keywords": [
    "Ecology",
    "Metadata",
    "EML",
    "Ecology",
    "Biodiversity",
    "FAIR",
    "Data Paper",
    "work-in-progress"
  ],
  "description": "The questions this  addresses are:\n - How to generate detailled metadata easily from biodiversity datasets ?\n - How to use international metadata standard?\n - How to update metadata informations ?\n\n\nThe objectives are:\n - Explain the necessity of using such tools when producing ecological metadata\n - Learn how to create rich metadata using Ecological Metadata Language (EML) standard\n - Learn how to update EML metadata\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/yvanlebras/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/yvanlebras/",
      "name": "Yvan Le Bras",
      "image": "https://avatars.githubusercontent.com/yvanlebras",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-8504-068X",
      "orcid": "https://orcid.org/0000-0002-8504-068X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/ThibaudGlinez/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/ThibaudGlinez/",
      "name": "Glinez Thibaud",
      "image": "https://avatars.githubusercontent.com/ThibaudGlinez",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0006-8655-7505",
      "orcid": "https://orcid.org/0009-0006-8655-7505"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/yvanlebras/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/yvanlebras/",
      "name": "Yvan Le Bras",
      "image": "https://avatars.githubusercontent.com/yvanlebras",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-8504-068X",
      "orcid": "https://orcid.org/0000-0002-8504-068X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/ThibaudGlinez/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/ThibaudGlinez/",
      "name": "Glinez Thibaud",
      "image": "https://avatars.githubusercontent.com/ThibaudGlinez",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0006-8655-7505",
      "orcid": "https://orcid.org/0009-0006-8655-7505"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/hexylena/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/hexylena/",
      "name": "Helena Rasche",
      "image": "https://avatars.githubusercontent.com/hexylena",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9760-8992",
      "orcid": "https://orcid.org/0000-0001-9760-8992"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Ecology",
      "description": "Learn to analyse Ecological data through Galaxy.",
      "url": "https://training.galaxyproject.org/training-material/topics/ecology/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0610",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0610",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0610"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "Metadata"
    },
    {
      "@type": "Thing",
      "name": "EML"
    },
    {
      "@type": "Thing",
      "name": "Ecology"
    },
    {
      "@type": "Thing",
      "name": "Biodiversity"
    },
    {
      "@type": "Thing",
      "name": "FAIR"
    },
    {
      "@type": "Thing",
      "name": "Data Paper"
    },
    {
      "@type": "Thing",
      "name": "work-in-progress"
    }
  ],
  "abstract": "<p align=\"justify\">This tutorial aims to teach <b>how to use functionalities of the EML Assembly Line R package to produce rich metadata</b> using the Ecological Metadata Language (EML) international metadata standard. Here, we will notably propose a concrete example on how to use <b>Galaxy Ecology tools to create, evaluate and modify EML metadata</b> content using both EML Assemby Line metadata template tabular files, easily readable and editable by humans, and XML file, devoted to machine.</p>"
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/ecology" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Ecology
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/ecology/faqs/" title="Check our FAQs for the Ecology topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-ecology">
    <h1 data-toc-skip>Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/yvanlebras/" class="contributor-badge contributor-yvanlebras"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/yvanlebras?s=36" alt="Yvan Le Bras avatar" width="36" class="avatar" />
    Yvan Le Bras</a><a href="/training-material/hall-of-fame/ThibaudGlinez/" class="contributor-badge contributor-ThibaudGlinez"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/ThibaudGlinez?s=36" alt="Glinez Thibaud avatar" width="36" class="avatar" />
    Glinez Thibaud</a>
		</td>
	</tr>
	


	

	
	<tr>
		<td><abbr title="These people edited the text, either for spelling and grammar, flow, GTN-fit, or other similar editing categories">Editor(s)</abbr></td>
		<td>
			<a href="/training-material/hall-of-fame/yvanlebras/" class="contributor-badge contributor-yvanlebras"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/yvanlebras?s=36" alt="Yvan Le Bras avatar" width="36" class="avatar" />
    Yvan Le Bras</a><a href="/training-material/hall-of-fame/ThibaudGlinez/" class="contributor-badge contributor-ThibaudGlinez"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/ThibaudGlinez?s=36" alt="Glinez Thibaud avatar" width="36" class="avatar" />
    Glinez Thibaud</a><a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge contributor-hexylena"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hexylena?s=36" alt="Helena Rasche avatar" width="36" class="avatar" />
    Helena Rasche</a></td>
	</tr>
	

	

	

	
</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to generate detailled metadata easily from biodiversity datasets ?</p>
</li>
        
        <li><p>How to use international metadata standard?</p>
</li>
        
        <li><p>How to update metadata informations ?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Explain the necessity of using such tools when producing ecological metadata</p>
</li>
        
        <li><p>Learn how to create rich metadata using Ecological Metadata Language (EML) standard</p>
</li>
        
        <li><p>Learn how to update EML metadata</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 30 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/records/10663465">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">‚úÖ</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">‚≠êÔ∏è</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr <abbr title="This instance supports the precise tool versions used in this tutorial">‚úÖ</abbr>
		</a>
	</li>
    
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/ecology/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Mar 4, 2024 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 7, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00422">https://gxy.io/GTN:T00422</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 45 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p align="justify">This tutorial aims to teach <b>how to use functionalities of the EML Assembly Line R package to produce rich metadata</b> using the Ecological Metadata Language (EML) international metadata standard. Here, we will notably propose a concrete example on how to use <b>Galaxy Ecology tools to create, evaluate and modify EML metadata</b> content using both EML Assemby Line metadata template tabular files, easily readable and editable by humans, and XML file, devoted to machine.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#1-how-can-eml-assembly-line-functionalities-help-producing-rich-metadata-easily" id="markdown-toc-1-how-can-eml-assembly-line-functionalities-help-producing-rich-metadata-easily">1] How can EML Assembly Line functionalities help producing rich metadata easily?</a></li>
  <li><a href="#2-get-data-to-describe-" id="markdown-toc-2-get-data-to-describe-">2] Get data to describe üíæüìÇ</a></li>
  <li><a href="#3-metashark---write-rapidly-an-eml-document-through-a-work-in-progress-interactive-app" id="markdown-toc-3-metashark---write-rapidly-an-eml-document-through-a-work-in-progress-interactive-app">3] MetaShARK ü¶à : Write rapidly an EML document through a work in progress interactive app</a></li>
  <li><a href="#4-metashrimps---easily-fairness-assessment-and-data-paper-sketches-creation" id="markdown-toc-4-metashrimps---easily-fairness-assessment-and-data-paper-sketches-creation">4] MetaShRIMPS ü¶ê : Easily FAIRness assessment and data paper sketches creation</a></li>
  <li><a href="#5-draft-of-data-paper-" id="markdown-toc-5-draft-of-data-paper-">5] Draft of Data Paper üìù</a></li>
  <li><a href="#6-fair-quality-assessment-report-" id="markdown-toc-6-fair-quality-assessment-report-">6] FAIR Quality Assessment report üìä‚úÖ</a></li>
  <li><a href="#7-conclusion" id="markdown-toc-7-conclusion">7] Conclusion</a></li>
</ol>

</blockquote>

<h1 id="1-how-can-eml-assembly-line-functionalities-help-producing-rich-metadata-easily">1] How can EML Assembly Line functionalities help producing rich metadata easily?</h1>

<p align="justify"> A major gap when a researcher is writing metadata documents is the fact that metadata international <b>standards often use formats not really human readable</b> and/or editable as XML or JSON. To answer this issue, <b>[Environmental Data Initiative](https://edirepository.org/) (EDI)</b> through the <b>EML Assembly Line R package</b> propose to generate intermediate metadata template files using classical tabular text format.
Another major issue regarding <b>metadata fill in</b>, is the fact that one <b>need to take a lot of time to write</b>, and often rewrite, metadata elements who can be already filled using automatic inferences or use of webservices. Here again, Environmental Data Initiaitve (EDI) through the <b>EML Assembly Line R</b> package propose to <b>generate automatically information</b> related to data attributes, geographic coverage, taxonomic coverage, using the content of provided datafiles.
<br /><br />
Finally, through the MetaShARK R Shiny app, an R Shiny app in beta version for test, created by the french biodiversity data hub research infrastructure (P√¥le national de donn√©es de Biodiversit√© (<b>PNDB</b>)), user can use a graphical user interface to apply the EML Assembly Line workflow and benefit from some additionnal functionnalities as:</p>

<ul>
  <li>Capacity to associate terminological resources terms coming from Bioportal ontologies to data attributes as keywords using CEDAR API</li>
  <li>Automatic fill in of personal information using ORCID API</li>
  <li>Automatic production of a data paper draft</li>
</ul>

<p><br /></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-is-a-b-data-paper-b"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What is a <b>Data Paper</b> ?</div>
  <p>According to the <a href="https://www.gbif.org/data-papers">GBIF</a> (Global Biodiversity Information Facility), 
A data paper is a peer reviewed document describing a dataset, published in a peer reviewed journal. It takes effort to prepare, curate and describe data. 
Data papers provide recognition for this effort by means of a scholarly article.</p>
</blockquote>

<h1 id="2-get-data-to-describe-">2] Get data to describe üíæüìÇ</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Data Upload</div>

  <ol>
    <li>Create a new history for this tutorial. You can name it ‚ÄúEML assembly Line tutorial‚Äù for example</li>
    <li>Download all files on your local computer from Zenodo: https://zenodo.org/api/records/10663465/files-archive. It is neccessary as for now, MetaShARK is deployed from Galaxy but without having a possibility to directly populate MetaShARK app with datafiles from Galaxy. You will thus then have to upload manually data files from your local computer to MetaShARK.</li>
    <li>Unzip the donwloaded archive so you can access each file independently</li>
    <li>Import tsv, netcdf and geotiff data files directly from <a href="https://zenodo.org/records/10663465">Zenodo</a> so it can be used on some steps of the tutorial.
 -&gt; Training Data for ‚ÄúCreating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities‚Äù Galaxy tutorial:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/10663465/files/datafile_1.tsv
https://zenodo.org/records/10663465/files/datafile_2.tsv
https://zenodo.org/records/10663465/files/LakeGeneva_phytoplankton_1974-2004.nc
https://zenodo.org/records/10663465/files/Present.Surface.pH.tif
</code></pre></div>      </div>
      <p><a href="./Images/0_Upload_datafiles.png" rel="noopener noreferrer"><img src="./Images/0_Upload_datafiles.png" alt="Upload datafiles. " width="1916" height="1080" loading="lazy" /></a></p>
    </li>
    <li>Import shapefile related files into Galaxy using the Galaxy upload tool, then on the ‚Äúcomposite‚Äù tab, specifying ‚Äúshp‚Äù composite type, then uploading .dbf, .shp and .shx files on the dedicated spaces.
<a href="./Images/1_upload_shapefile.png" rel="noopener noreferrer"><img src="./Images/1_upload_shapefile.png" alt="Upload shapefile. " width="1437" height="652" loading="lazy" /></a></li>
  </ol>
</blockquote>

<h1 id="3-metashark---write-rapidly-an-eml-document-through-a-work-in-progress-interactive-app">3] MetaShARK ü¶à : Write rapidly an EML document through a work in progress interactive app</h1>

<p>To deploy a MetaShARK app, you can go to the Galaxy tool <span class="tool" data-tool="interactive_tool_metashark" title="MetaShARK tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MetaShARK</strong></span> and click ‚ÄúExecute‚Äù. Then, you have to wait the launch of the app, and when ready to be used, you will see the message ‚ÄúThere is an InteractiveTool result view available,‚Äù with an hyperlink on the ‚ÄúOpen‚Äù statement allowing you to reach the app clicking on it.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-warning"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: WARNING</div>
  <p>üöß Note that MetaShARK R shiny app is in beta version. You can thus encounter issues using it!</p>
</blockquote>

<p>When oppening MetaShARK, you will have an interface looking like this :</p>

<p><a href="./Images/2_MetaShARK.png" rel="noopener noreferrer"><img src="./Images/2_MetaShARK.png" alt="MetaShARK interface. " width="1786" height="763" loading="lazy" /></a></p>

<p>To start creating metadata, you need to reach the <strong>Fill in EML</strong> module, then specify or complete the automatic <code class="language-plaintext highlighter-rouge">Data package name</code> and mention the <code class="language-plaintext highlighter-rouge">Dataset title</code>. Here a title can be ‚ÄúManage heterogeneous data files through EML‚Äù. Finally, you can choose an open licence between CC-BY-4.0, default, or CC0 then click on ‚ÄúCreate‚Äù.</p>

<p><a href="./Images/3_MetaShARK_create.png" rel="noopener noreferrer"><img src="./Images/3_MetaShARK_create.png" alt="Create EML Assembly Line &quot;data package&quot;. " width="1922" height="809" loading="lazy" /></a></p>

<p>Then you can upload datafiles. Here, you can import these files from the downloaded Zenodo archive (link : https://zenodo.org/api/records/10663465/files-archive):</p>

<p>üìÅ List of datasets :</p>
<ul>
  <li>datafile_1.<strong>tsv</strong></li>
  <li>datafile_2.<strong>tsv</strong></li>
  <li>LakeGeneva_phytoplankton_1974-2004.<strong>nc</strong></li>
  <li>Present.Surface.pH.<strong>tif</strong></li>
  <li>02_Ref.<strong>shp</strong></li>
  <li>02_Ref.<strong>shx</strong></li>
  <li>02_Ref.<strong>dbf</strong></li>
</ul>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-note"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: NOTE</div>
  <p>The folder contains many files with different extensions but MetaShark will normally recognise several types of file extension (notably .tsv for tabulated text files, shapefiles, .geotiff and .geojson)!</p>
</blockquote>

<p><a href="./Images/4_MetaShARK_upload.png" rel="noopener noreferrer"><img src="./Images/4_MetaShARK_upload.png" alt="Upload datafiles on MetaShARK. " width="1915" height="1079" loading="lazy" /></a></p>

<p align="justify">MetaShARK will normally guess that the three `02_Ref` files are representing a uniq shapefile. MetaShARK will normally guess each data type and infer list of attributes for each file but the geotiff `Present.Surface.pH.tif` one. So now you need to select this datafile and upload the `attributes_Present.Surface.pH.txt` metadata template file so MetaShARK can fill attributes of this file (here the attribute is named "Present.Surface.pH").</p>

<p><a href="./Images/5_MetaShARK_geotiffattributes.png" rel="noopener noreferrer"><img src="./Images/5_MetaShARK_geotiffattributes.png" alt="Upload metadata template for geotiff attributes. " width="1894" height="1019" loading="lazy" /></a></p>

<p align="justify">Then you can provide a description for this attribute, for example "Present surface pH", then look at each attribute information of each data file so you can click on the "Next" button and go to the next step, to give informations on categorical variables!</p>

<p><a href="./Images/6_MetaShARK_catvars.png" rel="noopener noreferrer"><img src="./Images/6_MetaShARK_catvars.png" alt="Categorical variables. " width="1920" height="1080" loading="lazy" /></a></p>

<p align="justify">Clicking "Next" button will then allows you to fill spatial informations about all GIS recognized datafiles, here the `Present.Surface.pH.tif` geotiff raster file and the `02_Ref` shapefile vector file. Geotiff is in pixel, accuracy can be set to unknown and shapefile is in Point, both are in `GCS_WGS_1984`spatial reference.</p>

<p><a href="./Images/7_MetaShARK_spatialinfo.png" rel="noopener noreferrer"><img src="./Images/7_MetaShARK_spatialinfo.png" alt="Spatial information. " width="1922" height="1051" loading="lazy" /></a></p>

<p align="justify">Next step is devoted to specifying geographic coverage.<br /><br /> You can use a method between "columns" or "custom". "Custom" allows you to create one to several geographical sites using a map widget where you can draw limits of each site or enter directly latitude and longitude coordinates. "Columns" method, used here, allows you to specify an attribute containing site names then associated latitude and longitudes attributes.</p>

<p><a href="./Images/8_MetaShARK_geocov.png" rel="noopener noreferrer"><img src="./Images/8_MetaShARK_geocov.png" alt="Geographic coverage. " width="1917" height="1074" loading="lazy" /></a></p>

<p align="justify">Now geographic coverage is set, one can specific taxonomic coverage.<br /><br /> To do so, you can select a data attribute containing taxonomic information then select kind of notation you want to have and finally on which taxonomic authority (or authorities) information will be compared. Note that this can take a while if you have a lot of taxons and time is duplicated for each selected additional authority.</p>

<p><a href="./Images/9_MetaShARK_taxcov.png" rel="noopener noreferrer"><img src="./Images/9_MetaShARK_taxcov.png" alt="Taxonomic coverage. " width="1918" height="953" loading="lazy" /></a></p>

<p align="justify">Now we can fill personal informations.<br /><br /> To do so, the easiest way is to provide ORCID identifiers for each individual person involved as creator, contact and/or PI. Depending on the information filled in ORCID by each individual and on the level of accessibility of each, all field can be automatically filled. If "PI" is selected, you can specify a project name, funder name and related funding number.</p>

<p><a href="./Images/10_MetaShARK_personal.png" rel="noopener noreferrer"><img src="./Images/10_MetaShARK_personal.png" alt="Personal informations. " width="1920" height="887" loading="lazy" /></a></p>

<p>Then you can add final elements as:</p>

<ul>
  <li>An abstract (writing directly on the dedicated field or uploading a text file containing the abstract)</li>
</ul>

<p><a href="./Images/11_MetaShARK_abstract.png" rel="noopener noreferrer"><img src="./Images/11_MetaShARK_abstract.png" alt="Abstract. " width="1893" height="1075" loading="lazy" /></a></p>

<ul>
  <li>Methods (writing directly on the dedicated field or uploading a text file, can be in markdown, containing details of the methods used to create data files)</li>
</ul>

<p><a href="./Images/12_MetaShARK_methods.png" rel="noopener noreferrer"><img src="./Images/12_MetaShARK_methods.png" alt="Methods. " width="1891" height="1049" loading="lazy" /></a></p>

<ul>
  <li>Keywords, who can be linked to keyword thesaurus. This allows you to create ‚Äúgroup‚Äù of keywords and/or refer to existing terms classifications as we can find in terminological resources such as ontologies or thesaurus.</li>
</ul>

<p><a href="./Images/13_MetaShARK_keywords.png" rel="noopener noreferrer"><img src="./Images/13_MetaShARK_keywords.png" alt="Keywords. " width="1888" height="1078" loading="lazy" /></a></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-you-can-add-semantic-annotations"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: You can add Semantic Annotations</div>
  <p><strong>You can add <a href="https://eml.ecoinformatics.org/semantic-annotation-primer">semantic annotations</a>!</strong></p>

  <p align="justify"> To do so, you need to reach the MetaShARK parameters (upper right icon) then enter your CEDAR token. To create a CEDAR account, you can 1/ register here http://cedar.metadatacenter.org/ then 2/ go on the "profile" on http://cedar.metadatacenter.org/ and there 3/ you can find the API key.</p>

  <p>API key format to enter is something like:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>api 205b1e521f2eaf0ad4a361c438b63205b1e521f2eaf0ad4a361c438b63c438b63
</code></pre></div>  </div>

  <p align="justify">You then can  use the `+` button on the keyword space to **Add keyword with dataset annotation**. You will have to choose a "predicate", from IAO ontology, then an "object" from ontologies coming from Bioportal to add information concerning a "subject", the ‚Äòthing‚Äô being annotated, here, regarding keyword, "dataset", but you can also apply the same to datafiles "attributes".</p>

  <p><a href="./Images/17_MetaShARK_annotations_predicate.png" rel="noopener noreferrer"><img src="./Images/17_MetaShARK_annotations_predicate.png" alt="Add semantic annotation predicate. " width="1874" height="1063" loading="lazy" /></a></p>

  <p><a href="./Images/18_MetaShARK_annotations_object.png" rel="noopener noreferrer"><img src="./Images/18_MetaShARK_annotations_object.png" alt="Add semantic annotation object. " width="1836" height="1062" loading="lazy" /></a></p>

</blockquote>

<p align="justify">Finally, you can specify a temporal coverage and go to the last step of this MetaShARK workflow: Generat an EML metadata file! If everything is ok, you will have creation of an EML metadata file.</p>

<p><a href="./Images/14_MetaShARK_makeeml.png" rel="noopener noreferrer"><img src="./Images/14_MetaShARK_makeeml.png" alt="Produce EML metadata file. " width="1920" height="1080" loading="lazy" /></a></p>

<p align="justify">Once EML written, you can <b>download the data package through the button</b> "Download Data Package". This will allow you to <b>download a zip archive</b> you can unzip on your local computer. Resulting files are organized through 2 main folders : </p>

<ul>
  <li><strong>A main folder with data_objects</strong>
    <ul>
      <li><code class="language-plaintext highlighter-rouge">all datafiles </code> you uploaded into MetaShARK</li>
      <li><code class="language-plaintext highlighter-rouge">eml </code> which is the EML metadata file written in XML format</li>
      <li><code class="language-plaintext highlighter-rouge">metadata_templates </code> with all metadata files written in text format, column separated by tabulations</li>
    </ul>
  </li>
  <li>A <strong>second folder called ‚Äúemldown‚Äù</strong> where a <strong>draft of data paper</strong> written in html format can be accessed</li>
</ul>

<hr />

<p align="center"><b>üëè Congratulations! You've just produced your first EML yourself!üëè</b></p>
<hr />

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-metashark-can-freeze"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: MetaShARK can Freeze</div>
  <p><strong>At this final stage, MetaShARK can freeze and show like a ‚Äúfront grey filter‚Äù</strong></p>

  <p>If this is the case, you can refresh MetaShARK app pressing F5 button of your keyboard for example. To dowload the resutling data package, you can go to the ‚ÄúFill In‚Äù module, select the data package you created and click on ‚ÄúDownload.zip‚Äù.</p>
</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-returning-on-your-metashark-instance"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-returning-on-your-metashark-instance" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Returning on your MetaShARK instance</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>If you close MetaShARK navigator window, you can reopen it clicking on Menu &gt; User &gt; Active InteractiveTools.
You can here click on ‚Äúmetashark visualization‚Äù hyperlink. You can then go on the Fill In module, select 
the data package you created and click <strong>Load</strong>.</p>
</blockquote>

<h1 id="4-metashrimps---easily-fairness-assessment-and-data-paper-sketches-creation">4] MetaShRIMPS ü¶ê : Easily FAIRness assessment and data paper sketches creation</h1>

<p>To evaluate and modify metadata elements you have created, you can upload the EML xml file on the Galaxy history you created and all MetaShARK metadata templates files.</p>

<p><a href="./Images/15_MetaShRIMPS.png" rel="noopener noreferrer"><img src="./Images/15_MetaShRIMPS.png" alt="Use MetaShRIMPS. " width="1919" height="1080" loading="lazy" /></a></p>

<p>Open the <a href="https://ecology.usegalaxy.eu/root?tool_id=interactive_tool_metashrimps">MetaShRIMPS interactive tool form</a> and select the EML xml file you just generate with MetaShARK. Clicking on Execute will launch the MetaShRIMPS R Shiny app, the message <code class="language-plaintext highlighter-rouge">There is an InteractiveTool result view available, waiting for view to become active... </code> is displayed until the app will be ready to use. After some time, the message <code class="language-plaintext highlighter-rouge">There is an InteractiveTool result view available</code> confirm the app is deployed and you can access it clicking on <strong>Open</strong>. You will then have an interface looking like this:</p>

<p><a href="./Images/upload_1.png" rel="noopener noreferrer"><img src="./Images/upload_1.png" alt="MetaShRIMPS use. " width="1110" height="521" loading="lazy" /></a></p>

<p>Click on the <strong>Execute</strong> Button, 2 new tabs called ‚ÄúDraft of Data Paper‚Äù and ‚ÄúFair Assessment‚Äù  will appear.
You can access all of the tool outputs by clicking on each tab (it can take a little time for your results to be displayed).</p>

<h1 id="5-draft-of-data-paper-">5] Draft of Data Paper üìù</h1>

<p align="justify">By clicking on the <b>"Draft of Data Paper"</b> tab, you will have access to the draft of Data Paper presented in an HTML format. You can either navigate through the Data Paper with the tabs or with the scrollbar on the right and access different elements.</p>

<ul>
  <li>You can at the top of the page <strong>download the draft</strong> in either an <em>HTML format</em> ‚Ä¶. :</li>
</ul>

<p><a href="./Images/Download_HTML.png" rel="noopener noreferrer"><img src="./Images/Download_HTML.png" alt="MetaShRIMPS data paper download in HTLM format. " width="1920" height="1080" loading="lazy" /></a></p>

<ul>
  <li>‚Ä¶. or an editable <em>docx format</em> :</li>
</ul>

<p><a href="./Images/Download_docx.png" rel="noopener noreferrer"><img src="./Images/Download_docx.png" alt="MetaShRIMPS editable data paper download in docx format. " width="1267" height="989" loading="lazy" /></a></p>

<hr />

<p>Voir comment mettre les deux photos c√¥t√©s √† c√¥tes ou un truc mieux</p>
<hr />

<h1 id="6-fair-quality-assessment-report-">6] FAIR Quality Assessment report üìä‚úÖ</h1>

<p><a href="https://www.go-fair.org/fair-principles/">FAIR</a> stand for <strong>Findable, Accessible, Interoperable, Reusable</strong>. 
<br /> <br /></p>

<p><a href="./Images/FAIR_data_principles.jpg" rel="noopener noreferrer"><img src="./Images/FAIR_data_principles.jpg" alt="FAIR principles. " width="1200" height="407" loading="lazy" /></a></p>

<p><br /></p>

<p align="justify">These principles were to <b>improve the access and usabiliy of data</b> by the machine and to help <b>making data reusable and shareable for users</b>. It covers the whole concept of why it is necessary to produce a rich and described metadata in order to permit external users to <b>understand and reuse data</b> for their own studies.
  
There are several ways of computing the FAIR index, for each letter of the word is associated with a degree of FAIRitude of the data.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-metadata-quality"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Metadata Quality</div>
  <p>The aim is to propose a metadata quality index and to see how it is possible to improve, not to punish the user üëç</p>
</blockquote>

<p>By clicking on the <code class="language-plaintext highlighter-rouge">Fair Assessment</code> tab, you will access the FAIR Quality report of the metadata uploaded.</p>

<p>You will have access to different figures such as a <strong>table displaying the results of all checks tested for your metadata</strong>:</p>

<figure id="figure-1" style="max-width: 90%;"><img src="./Images/Fairscore_tab2.png" alt="Table of results. " width="1860" height="966" loading="lazy" /><a target="_blank" href="./Images/Fairscore_tab2.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Example of the table displaying the results of the Quality Checks</figcaption></figure>

<p>You will also have acces to a graph presenting scores of Quality for each of the FAIR principles tested (Findable,
Acessible, Interoperable, Reusable) on a 100 point scale.</p>

<figure id="figure-2" style="max-width: 90%;"><img src="./Images/Fairscore_tab.png" alt="FAIR scores. " width="1422" height="880" loading="lazy" /><a target="_blank" href="./Images/Fairscore_tab.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Overall FAIR score</figcaption></figure>

<hr />

<p>Image un peu laide, je l‚Äôa modifierai un peu sur Paint pour optimiser sa taille vis √† vis du vide/blanc occup√© par certaines parties</p>
<hr />

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-how-you-can-improve-your-fair-score"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: How you can improve your FAIR score?</div>

  <p>You can look at the lines of the table for FAILURE (red) and WARNING (yellow) status. Two WARNING 
status lines  are related to an abstract content too short and an attribute definition too small.</p>

  <figure id="figure-3" style="max-width: 90%;"><img src="./Images/WARNING.png" alt="Warning status lines. " width="1809" height="278" loading="lazy" /><a target="_blank" href="./Images/WARNING.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Overall FAIR score</figcaption></figure>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-update-metadata-content-to-elevate-fair-score"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Update metadata content to elevate FAIR score</div>

  <ol>
    <li>
      <p>Search ‚Äúabstract‚Äù and ‚Äúattributes_Present.Surface.pH.txt‚Äù metadarta files on your history</p>
    </li>
    <li>
      <p>Modify each file using the Galaxy included text editor</p>

      <p>To do so, you can go on the ‚Äúvisualize‚Äù functionnality of each datafile, clicking on the name of the dataset on your history then on the ‚Äúvisualize‚Äù button at the bottom of the dataset description. you can then select ‚ÄúEditor / Manually edit text‚Äù to update the content and generate a new version of the dataset.</p>
    </li>
    <li>
      <p>Rename each datafile as original names (‚Äúabstract‚Äù and ‚Äúattributes_Present.Surface.pH.txt‚Äù), you can add a ‚Äúmodified‚Äù tag / label so you can better remember in the future the modification state.</p>
    </li>
    <li>
      <p>Recreate a data collection with all metadata template files, taking the new ‚Äúabstract‚Äù and ‚Äúattributes_Present.Surface.pH.txt‚Äù files instead of old ones.</p>
    </li>
    <li>
      <p>You can recreate an EML metadata file with <a href="https://ecology.usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/ecology/makeeml/makeeml/0.1.1+galaxy1"><strong>Make EML</strong></a> EAL tool and then redeploy a metashrimps tool on the new EML. Before executing <a href="https://ecology.usegalaxy.eu/root?tool_id=toolshed.g2.bx.psu.edu/repos/ecology/makeeml/makeeml/0.1.1+galaxy1"><strong>Make EML</strong></a> EAL tool, one need to pay attention to the rename composite datafiles (as here the shapefile one) as named originally (so <code class="language-plaintext highlighter-rouge">02_Ref</code>) and then create a data collection gathering all ‚ÄúdataTable‚Äù datafiles (both .tsv files and .nc one), a data collection gathering all ‚ÄúspatialRaster‚Äù datafiles (here <code class="language-plaintext highlighter-rouge">Present.Surface.pH.tif</code>) and a data collection gathering all ‚ÄúspatialVector‚Äù datafiles (here <code class="language-plaintext highlighter-rouge">LakeGeneva_phytoplankton_1974-2004.nc</code>).</p>
    </li>
  </ol>

</blockquote>

<h1 id="7-conclusion">7] Conclusion</h1>

<p>Here is the end of this short tutorial aiming in explaining the purpose of MetaShARK and how to use it.</p>

<p>Don‚Äôt hesitate to contact us if you have any questions ‚ò∫Ô∏è</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>This tool aims to improve FAIR quality of metadata focusing on user exeprience and automatic inferences</p>
</li>
                        
                        <li><p>Creating metadata as FAIR as possible is a must</p>
</li>
                        
                        <li><p>Be carefull of the format and standard of metadata used only EML metadata will work</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/ecology/faqs/">FAQ page for the Ecology topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/ecology#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

		<a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=ecology/MetaShARK_tutorial" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Yvan Le Bras, Glinez Thibaud,  <b>Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html">https://training.galaxyproject.org/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{ecology-MetaShARK_tutorial,
author = "Yvan Le Bras and Glinez Thibaud",
	title = "Creating metadata using Ecological Metadata Language (EML) standard with EML Assembly Line functionalities (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine F√∂ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Bj√∂rn Gr√ºning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                
                
<div markdown="1">
<h2 id="funding">Funding</h2>
<p>These individuals or organisations provided funding support for the development of this resource</p>

<div class="row">

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="https://avatars.githubusercontent.com/fnso2019" alt="Logo">
		
		<a href="/training-material/hall-of-fame/fnso2019/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		

	</div>

	<div class="col-md-3 col-xs-12">
		
		<img class="funder-avatar" src="https://avatars.githubusercontent.com/pndb" alt="Logo">
		
		<a href="/training-material/hall-of-fame/pndb/" class="btn btn-secondary">See Funder Profile</a>
	</div>
	<div class="col-md-9 col-xs-12">
		

	</div>

</div>
</div>


                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   

		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
		    
		</blockquote>


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/275c49ff3b6670060b65ab23356b29e3dd1b9f97">275c49f</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00422">https://gxy.io/GTN:T00422</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/ecology/tutorials/MetaShARK_tutorial/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.105e595f.js'></script>

	
	
    </body>
</html>