<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Reference-based RNAseq data analysis (long)</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Reference-based RNAseq data analysis (long)" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frb-rnaseq%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frb-rnaseq%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frb-rnaseq%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frb-rnaseq%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frb-rnaseq%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Frb-rnaseq%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Reference-based RNAseq data analysis (long)",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / rb-rnaseq / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Reference-based RNAseq data analysis (long)' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - How to perform analysis of RNAseq data when reference genome is available?\n\n\\nThe objectives are:\n - Learns basic concepts of RNAseq analysis\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Reference-based RNAseq data analysis (long)</h1>
        
        <div class="alert alert-warning" role="alert">
            <h4 class="alert-heading">Under Development!</h4>
            <p>This tutorial is not in its final state. The content may change a lot in the next months.
            Because of this status, it is also not listed in the topic pages.</p>
        </div>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nekrut" alt="Anton Nekrutenko">Anton Nekrutenko</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to perform analysis of RNAseq data when reference genome is available?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Learns basic concepts of RNAseq analysis</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                

                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 id="rnaseq-reference-based"><span class="notranslate">RNA</span>seq: Reference-based</h1>

<p>This tutorial is inspired by an exceptional <a href="http://chagall.med.cornell.edu/RNASEQcourse/"><span class="notranslate">RNA</span>seq course</a> at the Weill Cornell Medical College compiled by Friederike D√ºndar, Luce Skrabanek, and Paul Zumbo and by tutorials produced by Bj√∂rn Gr√ºning (@bgruening) for Freiburg <span class="notranslate">Galaxy</span> instance. Much of <span class="notranslate">Galaxy</span>-related features described in this section have been developed by Bj√∂rn Gr√ºning (@bgruening) and configured by Dave Bouvier (@davebx).</p>

<p><span class="notranslate">RNA</span>seq can be roughly divided into two ‚Äútypes‚Äù:</p>

<ul>
  <li><strong>Reference genome-based</strong> - an assembled genome exists for a species for which an <span class="notranslate">RNA</span>seq experiment is performed. It allows <span class="notranslate">reads</span> to be aligned against the reference genome and significantly improves our ability to reconstruct transcripts. This category would obviously include humans and most model organisms but excludes the majority of truly biologically intereting species (e.g., <a href="https://en.wikipedia.org/wiki/Hyacinth_macaw">Hyacinth macaw</a>);</li>
  <li><strong>Reference genome-free</strong> - no genome assembly for the species of interest is available. In this case one would need to assemble the <span class="notranslate">reads</span> into transcripts using <em>de novo</em> approaches. This type of <span class="notranslate">RNA</span>seq is as much of an art as well as science because assembly is heavily parameter-dependent and difficult to do well.</li>
</ul>

<p>In this lesson we will focus on the <strong>Reference genome-based</strong> type of <span class="notranslate">RNA</span> seq.</p>

<h2 id="experimental-procedures-affect-downstream-analyses">Experimental procedures affect <span class="notranslate">downstream</span> analyses</h2>

<p>The <em>Everything‚Äôs connected</em> slide by D√ºndar et al. (2015) explains the overall idea:</p>

<p><img src="../../images/everything_connected.png" alt="Everything's connected" /></p>

<p>There is a variety of ways in which <span class="notranslate">RNA</span> is treated during its conversion to c<span class="notranslate">DNA</span> and eventual preparation of sequencing libraries. In general the experimental <span class="notranslate">workflow</span> includes the following steps:</p>

<ul>
  <li><span class="notranslate">RNA</span> purification;</li>
  <li>Reverse transcription using Reverse Transcriptase (RT), which produces the first strand of c<span class="notranslate">DNA</span> (‚Äúc‚Äù stands for <em>complimentary</em>);</li>
  <li>Second strand synthesis using <span class="notranslate">DNA</span> polymerase;</li>
  <li>Library preparation for sequencing.</li>
</ul>

<p>In listing these basic steps we are ignoring a vast amount of details such as, for example, normalization strategies and procedures needed to deal with rare <span class="notranslate">RNA</span>s or degraded samples (see <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3821180/">Adiconis:2013</a>). Yet, there are two important experimental considerations that would effect the ways in which one analyses data and interprets the results. These are:</p>

<ul>
  <li>Priming for the first c<span class="notranslate">DNA</span> strand synthesis;</li>
  <li>Stranded versus Non-stranded libraries.</li>
</ul>

<h3 id="priming-for-the-first-strand-synthesis">Priming for the first strand synthesis</h3>

<p>Reverse Transcriptase (RT) requires a primer. One can leverage the fact that the majority of processed m<span class="notranslate">RNA</span>s are polyadenylated and use oligo-dT primer to (mostly) restrict c<span class="notranslate">DNA</span> synthesis to fully processed m<span class="notranslate">RNA</span>s. Alte<span class="notranslate">rna</span>tively one can use a mix of random oligonucleotides to prime RT at a multitude of inte<span class="notranslate">rna</span>l sites irrespective of <span class="notranslate">RNA</span> type and maturation status:</p>

<p><img src="../../images/dT_random.png" alt="Oligo-dT vs. random priming" /></p>

<p><strong>Oligo-dT vs. random priming</strong><br />
Oligo-dT (<strong>A</strong>) and random priming (<strong>B</strong>)</p>

<p>Depending on the choice of the approach one would have different types of <span class="notranslate">RNA</span>s included in the final sequencing outcome. For example, if one attempts to study <span class="notranslate">RNA</span>s that are not polyadenylated or not fully processed, it would be unwise to use oligo-dT priming approach.</p>

<h3 id="strand-specific-rnaseq">Strand-specific <span class="notranslate">RNA</span>seq</h3>

<p><span class="notranslate">RNA</span>s that are typically targeted in <span class="notranslate">RNA</span>seq experiments are single stranded (e.g., m<span class="notranslate">RNA</span>s) and thus have polarity (5‚Äô and 3‚Äô ends that are functionally distinct):</p>

<p><img src="../../images/dna_rna.png" alt="Relationship between DNA and RNA orientation" /></p>

<p><strong>Relationship between <span class="notranslate">DNA</span> and <span class="notranslate">RNA</span> orientation</strong></p>

<p>During a typical <span class="notranslate">RNA</span>seq experiment the information about strandedness is lost after both strands of c<span class="notranslate">DNA</span> are synthesized, size selected, and converted into sequencing library. However, this information can be quite useful for various aspects of <span class="notranslate">RNA</span>seq analysis such as transcript reconstruction and quantification. There is a number of methods for creating so called <em>stranded</em> <span class="notranslate">RNA</span>seq libraries that preserve the strand information (an excellent overview in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/">Levin et al, Nat Meth, 2010</a>):</p>

<figure id="figure-1"><img src="../../images/stranded_protocols.png" alt="Generation of stranded RNAseq libraries" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Generation of stranded <span class="notranslate">RNA</span>seq libraries (Levin et al, Nat Meth, 2010)</figcaption></figure>

<p><em>**</em><br />
Different types of stranded library generation protocols from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/">Levin:2010</a></p>

<p>Depending on the approach and whether one performs single- or paired-end sequencing there are multiple possibilities on how to interpret the results of <span class="notranslate">mapping</span> of these <span class="notranslate">reads</span> onto genome/transcriptome:</p>

<p><a href="https://sailfish.readthedocs.org/en/master/library_type.html"><img src="../../images/lib_type.png" alt="Effects of RNAseq library types" /></a></p>

<p><strong>Effects of <span class="notranslate">RNA</span>seq library types</strong><br />
Image and description below is from <a href="https://sailfish.readthedocs.org/en/master/library_type.html">Sailfish documentation</a></p>

<p>The relative orientation of the <span class="notranslate">reads</span> is only relevant if the library is pair-ended. The possible options are:</p>

<ul>
  <li><strong>I</strong> = inward;</li>
  <li><strong>O</strong> = outward;</li>
  <li><strong>M</strong> = matching (co-directional).</li>
</ul>

<p>Library can be stranded (<strong>S</strong>) or unstranded (<strong>U</strong>). If this library is stranded than depending on the protocols <span class="notranslate">reads</span> (single <span class="notranslate">reads</span> or forward <span class="notranslate">reads</span> in a paired-end run) may originate from:</p>

<ul>
  <li><strong>F</strong> = read 1 in paired-end sequencing or single-end read is derived from the Forward strand;</li>
  <li><strong>R</strong> = read 1 in paired-end sequencing or single-end read is derived from the Reverse strand.</li>
</ul>

<p>So by combining the relative orientation of <span class="notranslate">reads</span> is I, O, or M (if <span class="notranslate">reads</span> are paired), strandedness or the library (S or U), and whether the <span class="notranslate">reads</span> originate from forward and reverse strand (F or R) there can be quite a number of possibilities:</p>

<ul>
  <li>IU  - (an unstranded paired-end library where the <span class="notranslate">reads</span> face each other)</li>
  <li>SF  - (a stranded single-end protocol where the <span class="notranslate">reads</span> come from the forward strand)</li>
  <li>OSR - (a stranded paired-end protocol where the <span class="notranslate">reads</span> face away from each other,
    read1 comes from reverse strand and read2 comes from the forward strand).
    and so on‚Ä¶</li>
</ul>

<p>However, in practice, if you use Illumina paired-end <span class="notranslate">RNA</span>seq protocols you are unlikely to uncover many of these possibilities. You will either deal with:</p>

<ul>
  <li>unstranded <span class="notranslate">RNA</span>seq data (<strong>IU</strong> type from above. Also called <strong>fr-unstranded</strong> in TopHat/Cufflinks jargon);</li>
  <li>stranded <span class="notranslate">RNA</span>seq data produced with Illumina TrueSeq <span class="notranslate">RNA</span>seq kits and <a href="https://nar.oxfordjournals.org/content/37/18/e123">dUTP tagging</a> (<strong>ISR</strong> type from above or <strong>fr-firststrand</strong> in TopHat/Cufflinks nomenclature).</li>
</ul>

<p>The implication of stranded <span class="notranslate">RNA</span>seq is that you can distinguish whether the <span class="notranslate">reads</span> are derived from forward- or reverse-encoded transcripts:</p>

<p><img src="../../images/stranded_result.png" alt="Stranded RNAseq data look like this" /></p>

<p><strong>Stranded <span class="notranslate">RNA</span>seq data look like this</strong><br />
This example contrasts unstranded and stranded <span class="notranslate">RNA</span>seq experiments. <font color="red">Red transcripts</font> are from + strand and <font color="blue">blue</font> are from - strand. In stranded example <span class="notranslate">reads</span> are clearly stratified between the two strands.  A small number of <span class="notranslate">reads</span> from opposite strand may represent anti-sense transcription. The image from GATC Biotech.</p>

<h3 id="replicates-biological-or-technical-and-how-many">Replicates: Biological or Technical and how many?</h3>

<p>An <span class="notranslate">RNA</span>seq experiment without a sufficient number of replicates will be a waste of money. Replicates are essential to be able to correct for variation due to differences within/among organisms, cells, sequencing machines, library preparation protocols and numerous other potential factors. There are two types of replicates (as described by <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">D√ºndar:2015</a>):</p>

<ul>
  <li>
    <p><strong>Technical replicates</strong> can be defined as <em>different library preparations from the same <span class="notranslate">RNA</span> sample</em>. They should account for batch effects from the library preparation such as reverse transcription and PCR amplification. To avoid possible lane effects (e.g., differences in the sample loading, cluster amplification, and efficiency of the sequencing reaction), it is good practice to multiplex the same sample over different lanes of the same flowcell. In most cases, technical variability introduced by the sequencing protocol is quite low and well controlled.</p>
  </li>
  <li>
    <p><strong>Biological replicates</strong>. There is an on-going debate over what kinds of samples represent true biological replicates. Obviously, the variability between different samples will be greater between <span class="notranslate">RNA</span> extracted from two unrelated humans than between <span class="notranslate">RNA</span> extracted from two different batches of the same cell line. In the latter case, most of the variation that will eventually be detected was probably introduced by the experimenter (e.g., slightly differing media and plating conditions). Nevertheless, this is variation the researcher is typically not interested in assessing, therefore the ENCODE consortium defines biological replicates as <span class="notranslate">RNA</span> from an independent growth of cells/tissue (ENCODE <a href="https://genome.ucsc.edu/ENCODE/protocols/dataStandards/ENCODE_RNAseq_Standards_V1.0.pdf">2011</a>).</p>
  </li>
</ul>

<p>The number of replicates should be as high as practically possible. Most <span class="notranslate">RNA</span>seq experiments include three replicates and some have as many as 12 (see Schurch et al. <a href="https://arxiv.org/abs/1505.02017">2015</a>).</p>

<h2 id="read-mapping">Read <span class="notranslate">mapping</span></h2>

<p>After sequencing is performed you have a collection of sequencing <span class="notranslate">reads</span> for each sample/replicate. In a reference-based <span class="notranslate">RNA</span>seq experiment these need to be mapped against the genome. Because in the case of eukaryotic transcriptome most <span class="notranslate">reads</span> originate from processed m<span class="notranslate">RNA</span>s lacking exons, they cannot be simply mapped back to the genome. Instead they can be separated into two categories:</p>

<ul>
  <li><span class="notranslate">Reads</span> that map entirely within exons</li>
  <li><span class="notranslate">Reads</span> that cannot be mapped within an exon across their entire length because they span two or more exons</li>
</ul>

<p><em>Spliced</em> mappers have been developed to efficiently map transcript-derived <span class="notranslate">reads</span> against genome.</p>

<h3 id="tophat-tophat2-and-hisat">TopHat, TopHat2, and HiSat</h3>

<p><a href="https://bioinformatics.oxfordjournals.org/content/25/9/1105.abstract">Tophat</a> was one of the first tools designed specifically to address this problem by identifying potential exons using <span class="notranslate">reads</span> that do map to the genome, generating possible splices between neighboring exons, and comparing <span class="notranslate">reads</span> that did not initially map to the genome agaisnt these <em>in silico</em> created junctions:</p>

<p><a href="https://bioinformatics.oxfordjournals.org/content/25/9/1105/F1.expansion.html"><img src="../../images/tophat.png" alt="TopHat and TopHat2: Mapping RNAseq regions to genome" /></a></p>

<p><strong>TopHat and TopHat2: <span class="notranslate">Mapping</span> <span class="notranslate">RNA</span>seq regions to genome</strong><br />
In TopHat <span class="notranslate">reads</span> are mapped against the genome and are separated into two categories: (1) those that map, and (2) those that initially unmapped (IUM). ‚ÄúPiles‚Äù of <span class="notranslate">reads</span> representing potential exons are extended in search of potential donor/acceptor splice sites and potential splice junctions are reconstructed. IUMs are then mapped to these junctions. Image from <a href="https://bioinformatics.oxfordjournals.org/content/25/9/1105.full">Trapnell:2009</a>.</p>

<p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36"><img src="../../images/tophat2.png" alt="TopHat has been subsequently improved with the development of TopHat2" /></a></p>

<p><strong>TopHat has been subsequently improved with the development of TopHat2</strong><br />
Image from <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2013-14-4-r36">Kim:2012</a> summarizes steps involved in aligning of <span class="notranslate">RNA</span>seq <span class="notranslate">reads</span> with TopHat2</p>

<p>To further optimize and speed up spliced read alignment Kim at al. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4655817/">2015</a> developed <a href="https://ccb.jhu.edu/software/hisat2/index.shtml">HISAT</a>. It uses a set of <a href="https://en.wikipedia.org/wiki/FM-index">FM-indices</a> consisting one global genome-wide index and a collection of ~48,000 local overlapping 42 kb indices (~55,000 56 kb indices in HiSat2). This allows to find initial seed locations for potential read alignments in the genome using global index and to rapidly refine these alignments using a corresponding local index:</p>

<p><img src="../../images/hisat.png" alt="Hierarchical Graph FM index in HiSat/HiSat2" /></p>

<p><strong>Hierarchical Graph FM index in HiSat/HiSat2</strong><br />
A part of the read (blue arrow) is first mapped to the genome using the global FM index. The HiSat then tries to extend the alignment directly utilizing the genome sequence (violet arrow). In (<strong>a</strong>) it succeeds and this read aligned as it completely resides within an exon. In (<strong>b</strong>) the extension hits a mismatch. Now HiSat takes advantage of the local FM index overlapping this location to find the appropriate matting for the remainder of this read (green arrow). The (<strong>c</strong>) shows a combination these two strategies: the beginning of the read is mapped using global FM index (blue arrow), extended until it reaches the end of the exon (violet arrow), mapped using local FM index (green arrow) and extended again (violet arrow). Image from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4655817/">Kim:2015</a></p>

<h3 id="star-mapper">STAR mapper</h3>

<p><a href="https://github.com/alexdobin/STAR">STAR aligner</a> is a fast alte<span class="notranslate">rna</span>tive for <span class="notranslate">mapping</span> <span class="notranslate">RNA</span>seq <span class="notranslate">reads</span> against genome utilizing uncompressed <a href="https://en.wikipedia.org/wiki/Suffix_array">suffix array</a>. It operates in <a href="https://academic.oup.com/bioinformatics/article/29/1/15/272537/STAR-ultrafast-universal-RNA-seq-aligner">two stages</a>. In the first stage it performs seed search:</p>

<p><img src="../../images/star.png" alt="STAR's seed search" /></p>

<p><strong>STAR‚Äôs seed search</strong><br />
Here a read is split between two consecutive exons. STAR starts to look for a <em>maximum mappable prefix</em> (MMP) from the beginning of the read until it can no longer match continuously. After this point it start to MMP for the unmatched portion of the read (<strong>a</strong>). In the case of mismatches (<strong>b</strong>) and unalignable regions (<strong>c</strong>) MMPs serve as anchors from which to extend alignments. Image from <a href="https://bioinformatics.oxfordjournals.org/content/early/2012/10/25/bioinformatics.bts635.full.pdf+html">Dobin:2013</a>.</p>

<p>At the second stage STAR stitches MMPs to generate read-level alignments that (contrary to MMPs) can contain mismatches and indels. A scoring scheme is used to evaluate and prioritize stitching combinations and to evaluate <span class="notranslate">reads</span> that map to multiple locations. STAR is extremely fast but requires a substantial amount of RAM to run efficiently.</p>

<h2 id="transcript-reconstruction">Transcript reconstruction</h2>

<p>The previous step - <span class="notranslate">mapping</span> - assigns <span class="notranslate">RNA</span>seq <span class="notranslate">reads</span> to genomic locations and identifies splice junctions from <span class="notranslate">reads</span> that originate from different exons. At transcript reconstruction step this information is taken further in attempt to build transcript models. There is a number of tools for performing this task. A benchmarking paper by <a href="https://bioinformatics.oxfordjournals.org/content/early/2015/09/03/bioinformatics.btv488.full.pdf+html">Hayer:2015</a> attempted to compare performance of existing approaches with one of the outcomes shown below:</p>

<p><a href="https://bioinformatics.oxfordjournals.org/content/early/2015/09/08/bioinformatics.btv488/F5.large.jpg"><img src="../../images/rnaseq_comparison.png" alt="Comparison of transcript reconsruction approaches" /></a></p>

<p><strong>Comparison of transcript reconsruction approaches</strong><br />
Here <em>recall</em> (the number of correctly constructed forms divided by the total number of real forms) versus <em>precision</em> (true positives divided by the sum of true positives and false positives) are plotted for seven transcript assemblers tested on two simulated datasets: <em>EnsemblPerfect</em> and <em>EnsemblRealistic</em>. The shaded region is indicating suboptimal performance (i.e., the white, unshaded region is ‚Äúgood‚Äù). The figure is from <a href="https://bioinformatics.oxfordjournals.org/content/early/2015/09/03/bioinformatics.btv488.full.pdf+html">Hayer:2015</a>.</p>

<p>Based on these results <a href="https://cole-trapnell-lab.github.io/cufflinks/">Cufflinks</a> and <a href="https://ccb.jhu.edu/software/stringtie/">StringTie</a> have satisfactory performence. The following discussion is based on inner workings of StringTie.</p>

<h3 id="transcriptome-assembly-with-stringtie">Transcriptome assembly with StringTie</h3>

<p><a href="https://ccb.jhu.edu/software/stringtie/">StringTie</a> assembles transcripts from spliced read alignemnts produced by tools such as STAR, TopHat, or HISAT and simultaneously estimates their abundances using counts of <span class="notranslate">reads</span> assigned to each transcript. The following images illustrates details of StringTie <span class="notranslate">workflow</span>:</p>

<p><img src="../../images/stringtie1.png" alt="StringTie algorithm" /></p>

<p><strong>StringTie <span class="notranslate">workflow</span></strong><br />
Image from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4643835/">Pertea:2015</a></p>

<p>In essence StringTie builds an alte<span class="notranslate">rna</span>tive splice graph from overlapping <span class="notranslate">reads</span> in a given locus. In such a graph nodes correspond to exons (or, rather, contiguous regions of genome covered by <span class="notranslate">reads</span>; colored regions on the figure above), while edges are represented by <span class="notranslate">reads</span> connecting these exons. Next, it identifies a path within the splice graph that has the highest weight (largest number of <span class="notranslate">reads</span> on edges). Such path would correspond to an assembled transcript at this iteration of the algorithm. Because the edge weight is equal to the number of the <span class="notranslate">reads</span> StringTie estimates the coverage level for this transcript (see below) which can be used to estimate the transcript‚Äôs abundance. <span class="notranslate">Reads</span> that are associated with the transcript that was just assembled are then removed and the graph is updated to perform the next iteration of the algorithm.</p>

<h2 id="transcript-quantification">Transcript quantification</h2>

<p>Transcriptome quantification attempts to estimate <span class="notranslate">expression</span> levels of individuals transcripts. This is performed by assigning <span class="notranslate">RNA</span>seq <span class="notranslate">reads</span> to transcripts, counting, and normalization.</p>

<h3 id="assigning-reads-to-transcripts">Assigning <span class="notranslate">reads</span> to transcripts</h3>

<p>To associate <span class="notranslate">reads</span> with transcripts they (the <span class="notranslate">reads</span>) need to be aligned to the transcriptome. Tools like Cufflinks and StringTie reconstruct transcripts from spliced read alignments generated by other programs (TopHat, HISAT, STAR), so they already have the information about which <span class="notranslate">reads</span> belong to each reconstructed transcript. Other tools such as <a href="https://www.cs.cmu.edu/~ckingsf/software/sailfish/">Sailfish</a>, <a href="http://pachterlab.github.io/kallisto/">Kallisto</a>, and <a href="http://combine-lab.github.io/salmon/">Salmon</a> perform <em>lightweight</em> alignment of <span class="notranslate">RNA</span>seq <span class="notranslate">reads</span> against existing transcriptome sequences. The goal of lightweight alignment is to quickly distribute the <span class="notranslate">reads</span> across transcripts they likely originate from without worrying too much about producing high quality alignments. The upside of this is that the entire procedure can be performed very quickly. The downside is that these tools require high quality transcriptome as input, which is not a problem if you work with humans or mice, but is a problem if you are studying Hyacinth macaw or any other brilliantly colored creatures.</p>

<h4 id="lightweight-alignment">Lightweight alignment</h4>

<p><a href="https://www.cs.cmu.edu/~ckingsf/software/sailfish/">Sailfish</a> has been initially designed to utilize <a href="https://en.wikipedia.org/wiki/K-mer"><em>k</em>-mer</a> matching for finding association between <span class="notranslate">reads</span> and corresponding transcripts:</p>

<p><img src="../../images/sailfish.png" alt="Assigning reads to transcripts: Sailfish" /></p>

<p><strong>Assigning <span class="notranslate">reads</span> to transcripts: Sailfish</strong><br />
Sailfish indexes input transcriptome for a fixed <em>k</em>-mer length and compares <em>k</em>-mers derived from <span class="notranslate">RNA</span>seq <span class="notranslate">reads</span> against this index. Image from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4077321/">Patro:2014</a></p>

<p>The current version of Sailfish uses <a href="http://biorxiv.org/content/biorxiv/early/2015/10/22/029652.full.pdf">quasi-alignment</a> to extend exact matches found with <em>k</em>-mers:</p>

<p><img src="../../images/quasi_aln.png" alt="quasi-alignment" /></p>

<p><strong>Quasi-alignment of <span class="notranslate">reads</span> in Sailfish</strong><br />
In Sailfish version <a href="https://github.com/kingsfordgroup/sailfish/releases/tag/v0.7.0">0.7.0</a> and up transcriptome is concatenated into a single sequence using <code class="language-plaintext highlighter-rouge">$</code> separators from which a <a href="https://en.wikipedia.org/wiki/Suffix_array">suffix array</a> and a <a href="https://en.wikipedia.org/wiki/Hash_table">hash table</a> are constructed. A <em>k</em>-mer from an <span class="notranslate">RNA</span>seq read (green) is looked up in the hash table, which immediately gives its position in the suffix array allowing to extend the march as described in the legend and the <a href="http://biorxiv.org/content/biorxiv/early/2015/10/22/029652.full.pdf">paper</a>. Image from  <a href="http://biorxiv.org/content/biorxiv/early/2015/10/22/029652.full.pdf">Srivastava:2015</a></p>

<p><a href="https://pachterlab.github.io/kallisto/">Kallisto</a> also utilizes <em>k</em>-mer matching but uses a different data structure. It constructs a <a href="https://en.wikipedia.org/wiki/De_Bruijn_graph">De Bruijn graph</a> from transcriptome input (pane <strong>b</strong> of the figure below). This graph is different from De Bruijn graphs used for genome assembly in that its nodes are <em>k</em>-mers and transcripts correspond to paths through the graph. To accommodate multiple transcripts that can lay along the same path (or sub-path) the paths are ‚Äúcolored‚Äù with each transcript given a distinct ‚Äúcolor‚Äù (in genome assembly the graph is built from the <span class="notranslate">reads</span> and nodes usually correspond to overlaps between <em>k</em>-mers forming incoming and outgoing edges). Non-branching sections of the graph that have identical coloring are ‚Äúglued‚Äù into contigs. Finally a <a href="https://en.wikipedia.org/wiki/Hash_table">hash table</a> is built that stores the position of each transcriptome <em>k</em>-mer within the graph:</p>

<p><img src="../../images/kallisto.png" alt="Kallisto" /></p>

<p><strong>Assigning <span class="notranslate">reads</span> to transcripts: Kallisto</strong><br />
Here a black read is being associated with a set consisting of red, blue, and green transcripts (<strong>a</strong>). First, a graph is built from transcriptome (<strong>b</strong>). Next, by finding common <em>k</em>-mers between the read and the graph the read is ‚Äúthreaded‚Äù along a path (<strong>c</strong> and <strong>d</strong>). The colors along that path would indicate which transcripts it is likely derived from. Specifically, this is done by taking intersection of ‚Äúcolors‚Äù (<strong>c</strong>). It this case the read is assigned to two transcripts: red and blue. Image from <a href="https://arxiv.org/pdf/1505.02710v2.pdf">Bray:2015</a></p>

<p><a href="https://combine-lab.github.io/salmon/about/">Salmon</a> does not use <em>k</em>-mer matching approach. Instead it creates <a href="https://github.com/lh3/bwa">bwa</a>-like <a href="https://en.wikipedia.org/wiki/FM-index">FM-index</a> and uses it to finds chains of <em>Maximal Exact Matches</em> (MEMs) and <em>Super Maximal Exact Matches</em> (SMEMs) between a read and the transcriptome.
<a href="http://biorxiv.org/content/biorxiv/early/2015/06/27/021592.full.pdf">Patro:2015</a> define a MEM as ‚Äú<em>a substring that is shared by the query (read) and reference (transcript) that cannot be extended in either direction without introducing a mismatch</em>‚Äù. Similraly, a SMEM is defined as a ‚Äú<em>MEM that is not contained within any other MEM on the query.</em>‚Äù One of the advantages of utilizing the FM-index is that a new index does not need to re-generated for a search with different set of parameters. In the case of Sailfish and Kallisto an index is dependent on <em>k</em>-mer length and has to be recomputed every time the <em>k</em> is changed. The overall schematics of Salmon operation is as follows:</p>

<p><img src="../../images/salmon.png" alt="Salmon's method" /></p>

<p><strong>Assigning <span class="notranslate">reads</span> to transcripts: Salmon</strong><br />
Image from <a href="http://biorxiv.org/content/biorxiv/early/2015/06/27/021592.full.pdf">Patro:2015</a></p>

<h3 id="estimating-transcript-levels">Estimating transcript levels</h3>

<p>Once <span class="notranslate">reads</span> are apportioned across individual transcripts they can be quantified. There are several approaches for quantification.</p>

<h4 id="flow-networks">Flow networks</h4>

<p>StringTie, which performs assembly and quantification simultaneously converts splice graph into a flow network for which it solves <a href="https://en.wikipedia.org/wiki/Maximum_flow_problem">the maximum flow problem</a>. The maximum flow is such network represents the <span class="notranslate">expression</span> level for a given transcript:</p>

<p><img src="../../images/stringtie2.png" alt="Maximum flow network" /></p>

<p><strong>StringTie flow network</strong><br />
Here each exon node from the splice graph is split into <em>in</em> and <em>out</em> nodes connected with an edge weighted by the number of <span class="notranslate">reads</span> corresponding to that exon. For example, the first exon is covered by seven <span class="notranslate">reads</span> and so the edge between 1-in and 1-out has a weight of 7. <span class="notranslate">Expression</span> level would correspond to the maximum flow through a path representing a given transcript. Image from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4643835/">Pertea:2015</a></p>

<h4 id="expectation-maximization">Expectation Maximization</h4>

<p>The Expectation/Maximization framework (EM) is utilized in a number of tools such as <a href="http://bio.math.berkeley.edu/eXpress/index.html">eXpress</a> and more recently <a href="http://www.cs.cmu.edu/~ckingsf/software/sailfish/">Sailfish</a>, <a href="http://pachterlab.github.io/kallisto/">Kallisto</a>, and <a href="https://combine-lab.github.io/salmon/about/">Salmon</a> (As an alte<span class="notranslate">rna</span>tive strategy Salmon also utilizes <a href="http://arxiv.org/pdf/1308.5953v1.pdf">variational Bayesian method</a>. The principle of EM is nicely illustrated by <a href="https://liorpachter.wordpress.com/">Lior Pachter</a> in his transcript quantification <a href="http://arxiv.org/pdf/1104.3889v2.pdf">review</a>. Suppose, as shown on the image below, there are three transcripts (green, red, and blue). There are five <span class="notranslate">reads</span> associated with these transcripts. One read (<em>d</em>) is unique to the red transcript, while others correspond to two (<em>b</em>, <em>c</em>, <em>e</em>) or three (<em>a</em>) transcripts. The EM is an iterative procedure. In the first round transcript abundances are initialized as equal (0.33 each as there are three transcripts) and during expectation <span class="notranslate">reads</span> are apportioned across transcripts based on these abundances. Next, during maximization step transcript abundances are re-calculated as follow. For red transcript we sum up fraction of each read as 0.33 + 0 + 0.5 + 1 + 0.5 for <span class="notranslate">reads</span> a, b, c, d, and e, respectively. We now divide this by the sum of read allocations for each transcript as 2.33 + 1.33 + 1.33 for red, green, and blue transcripts respectively. For all three transcript calculation will look like this:</p>

<div>
	$$

\color{red}{red}   =  \frac{0.33 + 0.0 + 0.5 + 1.0 + 0.5}{2.33 + 1.33 + 1.33} = 0.47\\

\color{green}{green} =  \frac{0.33 + 0.5 + 0.0 + 0.0 + 0.5}{2.33 + 1.33 + 1.33} = 0.27\\

\color{blue}{blue}  =  \frac{0.33 + 0.5 + 0.5 + 0.0 + 0.0}{2.33 + 1.33 + 1.33} = 0.27

    $$
</div>

<p>During next expectation stage read are re-apportioned across transcripts and the procedure is repeated until convergence:</p>

<p><img src="../../images/em.png" alt="EM algorithm" /></p>

<p><strong>Expectation Maximization (EM)</strong><br />
Image from <a href="https://arxiv.org/pdf/1104.3889v2.pdf">Pacher:2011</a></p>

<h3 id="understanding-quantification-metrics">Understanding quantification metrics</h3>

<p>As we‚Äôve seen above quantification for a transcript is estimated using the number of associated <span class="notranslate">reads</span>. Yet the count is not a very good measure as it will be severely biased by multiple factors such as, for example, transcript length. Thus these counts need to be normalized. Normalization strategies can be roughly divided into two groups:</p>

<ul>
  <li>Normalization for comparison within a <strong>single</strong> sample;</li>
  <li>Normalization for comparison among <strong>multiple</strong> samples/conditions.</li>
</ul>

<p>In their <a href="http://chagall.med.cornell.edu/RNASEQcourse/">tutorial</a> D√ºndar et al. have compiled a table summarizing various metrics. Below is description of normalization technique for within sample comparisons (between sample comparison can be found in the next section on differential <span class="notranslate">expression</span> analysis):</p>

<p><img src="../../images/within_norm.png" alt="RNAseq normalization metrics" /></p>

<p><strong><span class="notranslate">RNA</span>seq normalization metrics: Within sample comparisons</strong><br />
Table from D√ºndar et al. <a href="http://chagall.med.cornell.edu/RNASEQcourse/">2015</a></p>

<p>In addition, an excellent overview of these metrics can be found <a href="https://haroldpimentel.wordpress.com/2014/05/08/what-the-fpkm-a-review-rna-seq-expression-units/">here</a>.</p>

<h4 id="9888-you-should-never-ever-use-rpkm-fpkm-or-tpm-to-compare-expression-levels-across-samples-these-are-relative-measures-consider-yourself-warned"><font color="red">&#9888; You should NEVER EVER use RPKM, FPKM, or TPM to compare <span class="notranslate">expression</span> levels <em>across</em> samples. These are RELATIVE measures! Consider yourself warned!</font></h4>

<h2 id="finding-expression-differences">Finding <span class="notranslate">expression</span> differences</h2>

<p>The goal of differential <span class="notranslate">expression</span> analysis (DE) is to find gene (DGE) or transcript (DTE) differences between conditions, developmental stages, treatments etc. In particular DE has two goals:</p>

<ul>
  <li>Estimate the <em>magnitude</em> of <span class="notranslate">expression</span> differences;</li>
  <li>Estimate the <em>significance</em> of <span class="notranslate">expression</span> differences.</li>
</ul>

<p>For this <span class="notranslate">expression</span> is estimated from read counts and attempts are made to correct for variability in measurements using replicates that are absolutely essential accurate results (see below). We begin our short discussion on DE by reproducing a figure from <a href="https://www.nature.com/nbt/journal/v31/n1/abs/nbt.2450.html">Trapnell:2013</a> highlighting some of the challenges associated with judging <span class="notranslate">expression</span> differences from read counts:</p>

<p><img src="../../images/diff.png" alt="Differential expression: Read counts and Expression levels" /></p>

<p><strong>Differential <span class="notranslate">expression</span>: Read counts and <span class="notranslate">Expression</span> levels</strong><br />
<strong>Change in fragment count for a gene does not necessarily equal a change in <span class="notranslate">expression</span></strong>. (<strong>a</strong>) Simple read-counting schemes sum the fragments incident on a gene‚Äôs exons. The exon-union model counts <span class="notranslate">reads</span> falling on any of a gene‚Äôs exons, whereas the exon-intersection model counts only <span class="notranslate">reads</span> on constitutive exons. (<strong>b</strong>) Both of the exon-union and exon intersection counting schemes may incorrectly estimate a change in <span class="notranslate">expression</span> in genes with multiple isoforms. The true <span class="notranslate">expression</span> is estimated by the sum of the length-normalized isoform read counts. The discrepancy between a change in the union or intersection count and a change in gene <span class="notranslate">expression</span> is driven by a change in the abundance of the isoforms with respect to one another. In the top row, the gene generates the same number of <span class="notranslate">reads</span> in conditions A and B, but in condition B, all of the <span class="notranslate">reads</span> come from the shorter of the two isoforms, and thus the true <span class="notranslate">expression</span> for the gene is higher in condition B. The intersection count scheme underestimates the true change in gene <span class="notranslate">expression</span>, and the union scheme fails to detect the change entirely. In the middle row, the intersection count fails to detect a change driven by a shift in the dominant isoform for the gene. The union scheme detects a shift in the wrong direction. In the bottom row, the gene‚Äôs <span class="notranslate">expression</span> is constant, but the isoforms undergo a complete switch between conditions A and B. Both simplified counting schemes register a change in count that does not reflect a change in gene <span class="notranslate">expression</span>. Figure from [Trapnell:2013] (https://www.nature.com/nbt/jou<span class="notranslate">rna</span>l/v31/n1/abs/nbt.2450.html)</p>

<p>The following discussion of DGE logic is reproduced from <a href="http://chagall.med.cornell.edu/RNASEQcourse/">D√ºndar:2015</a>.</p>

<p>To determine the genes whose read count differences between two conditions are greater than expected by chance, DGE tools must make assumptions about the distribution of read counts. The null hypothesis ‚Äì that the mean read counts of the samples of condition <strong>A</strong> are equal to the mean read counts of the samples of condition <strong>B</strong> ‚Äì is tested for each gene individually. One of the most popular choices to model the read counts is the <a href="https://en.wikipedia.org/wiki/Poisson_distribution">Poisson distribution</a> because:</p>

<ul>
  <li>individual <span class="notranslate">reads</span> can be interpreted as binary data (Bernoulli trials): they either originate from gene <em>i</em> or not;</li>
  <li>we are trying to model the discrete probability distribution of the number of successes (success = read is present in the sequenced library);</li>
  <li>the pool of possible <span class="notranslate">reads</span> that could be present is large, while the proportion of <span class="notranslate">reads</span> belonging to
ene <strong>i</strong> is quite small.</li>
</ul>

<p>The nice feature of a Poisson distribution is that variance = mean. Thus, if the <span class="notranslate">RNA</span>-seq experiment gives us a precise estimate of the mean read counts per condition, we implicitly know what kind of variance to expect for read counts that are not truly changing between two conditions. This, in turn, then allows us to identify those genes that show greater differences between the two conditions than expected by chance. While read counts of the same library preparation (<em>technical replicates</em>) can indeed be well approximated by the Poisson distribution, it has been shown that biological replicates have greater variance (noise) than expected. This <a href="https://en.wikipedia.org/wiki/Overdispersion">overdispersion</a> can be captured with the <a href="https://en.wikipedia.org/wiki/Negative_binomial_distribution">negative binomial distribution</a>, which is a more general form of the Poisson distribution that allows the variance to exceed the mean. The square root of the dispersion is the coefficient of variation ‚Äì SD/mean ‚Äì after subtracting the variance we expect due to Poisson sampling.</p>

<p>In contrast to the Poisson distribution, we now need to estimate two parameters from the read counts: the mean as well as the dispersion. The precision of these estimates strongly depends on the number (and variation) of replicates ‚Äì the more replicates, the better the grasp on the underlying mean <span class="notranslate">expression</span> values of unchanged genes and the variance that is due to biological variation rather than the experimental treatment. For most <span class="notranslate">RNA</span>-seq experiments, only two to three replicates are available, which is not enough for reliable mean and variance estimates. Some tools therefore compensate for the lack of replication by borrowing information across genes with similar <span class="notranslate">expression</span> values and shrink a given gene‚Äôs variance towards the regressed values. These fitted values of the mean and dispersion are then used instead of the raw estimates
o test for differential gene <span class="notranslate">expression</span>.</p>

<p>The best performing tools tend to be <a href="https://bioconductor.org/packages/release/bioc/html/edgeR.html">edgeR</a>, <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq/DESeq2</a>, and <a href="https://www.bioconductor.org/packages/release/bioc/html/limma.html">limma-voom</a> (see Rapaport et al. (<a href="https://link.springer.com/article/10.1186/gb-2013-14-9-r95">2013</a>); Soneson and Delorenzi (<a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-91">2013</a>); Schurch et al. (<a href="http://arxiv.org/abs/1505.02017">2015</a>) for reviews of DGE tools). DESeq and limma-voom tend to be more conservative than edgeR (better control of false positives), but edgeR is recommended for experiments with fewer than 12 replicates (Schurch et al., <a href="http://arxiv.org/abs/1505.02017">2015</a>).</p>

<h2 id="lets-try-it">Let‚Äôs try it</h2>

<h3 id="the-data">The data</h3>

<p>In this example we will use a downsampled version of simulated <em>Drosophila melanogaster</em> <span class="notranslate">RNA</span>-seq data used by Trapnell et al. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3334321/">2012</a>. These include two conditions (<strong>C1</strong> and <strong>C2</strong>), each containing three replicates (<strong>R1</strong>, <strong>R2</strong>, and <strong>R3</strong>) sequenced as a paired end library. Thus in total there are 12 fastq datasets.</p>

<p>Here is what to do to load the data:</p>

<h4 id="loading-the-data-and-create-dataset-collections">Loading the data and create dataset collections</h4>

<p>Go to the data library (ask your instructor) and select all fastq files. Then Click <code class="language-plaintext highlighter-rouge">to History</code> button:</p>

<p><img src="../../images/rnaseq_library.png" alt="Data libraries" /></p>

<p>The datasets will appear in your history:</p>

<p><img src="../../images/rnaseq_data_in_history.png" alt="RNASeq data in history" /></p>

<p>Twelve datasets make a lot of clicking necessary. To avoid this annoyance we will combine them into two collections - <strong>c1</strong> and <strong>c2</strong> as shown in the video below. Also, see this <a href="/training-material/topics/galaxy-interface/tutorials/processing-many-samples-at-once/tutorial.html">tutorial</a> for yet another explanation of dataset collections.</p>
<div class="embed-responsive embed-responsive-16by9"><iframe src="https://player.vimeo.com/video/163625221" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe></div>

<h3 id="mapping-reads"><span class="notranslate">Mapping</span> <span class="notranslate">reads</span></h3>

<p>We will map the <span class="notranslate">reads</span> with TopHat2. Select <strong>TopHat</strong> from <strong>NGS: <span class="notranslate">RNA</span> Analysis</strong> section of the tool menu (left pane of <span class="notranslate">Galaxy</span>‚Äôs interface):</p>

<h4 id="mapping-with-tophat2"><span class="notranslate">Mapping</span> with TopHat2</h4>

<p>In this case the input to TopHat is <strong>not</strong> individual datasets, but a collection instead. The <a href="https://vimeo.com/163625221">video</a> above shows how to generate collection. Since we have created two collection as was described above, we will used them as inputs (note that <strong>Is this single-end or paired-end data?</strong> is set to <code class="language-plaintext highlighter-rouge">Paired-end (as collection)</code>). Make sure that the top part of TopHat interface looks like in the image below. Here the following parameters are set:</p>

<ul>
  <li><strong>Mean Inner Distance between Mate Pairs</strong> = <code class="language-plaintext highlighter-rouge">28</code> This is because paired <span class="notranslate">reads</span> are 100 bp and mean insert size is 228 bp so that 228 - (100 + 100) = 28</li>
  <li><strong>Use a built in reference genome or own from your history</strong> = <code class="language-plaintext highlighter-rouge">Use a build-in genome</code> and <code class="language-plaintext highlighter-rouge">dm3</code> is selected. This is because the <span class="notranslate">reads</span> are from <em>D. melanogaster</em>.</li>
  <li><strong>TopHat settings to use</strong> = <code class="language-plaintext highlighter-rouge">Full parameter list</code> This is done to be able to specify the strandedness of the library.</li>
  <li><strong>Library Type</strong> = <code class="language-plaintext highlighter-rouge">FR First Strand</code></li>
</ul>

<p><img src="../../images/tophat_interface.png" alt="TopHat interface" /></p>

<p>The same procedure is then repeated for collection <strong>c2</strong>. In the end it generates a lot of datasets in the history resulting in something resembling an image below. TopHat produces five types of output and because we started with dataset collections every one of the green boxes shown below is actually a collection of outputs for <strong>c1</strong> and <strong>c2</strong>, respectively.</p>

<p><img src="../../images/tophat_output.png" alt="TopHat outputs for both collections" /></p>

<p>Let‚Äôs now take a look at some of the alignments. We will use IGV for this purpose.</p>

<p>First, let‚Äôs drill down to actual alignments produced by TopHat. For example, in figure shown above simply click on <strong>TopHat on collection 14: accepted_hits</strong> and you will see a list of datasets corresponding to alignments of <span class="notranslate">reads</span> derived from each conditions:</p>

<p><img src="../../images/accepted_hits_1.png" alt="Accepted hits on collection 14" /></p>

<p>Now, click on <strong>c2-r1x</strong> and the following will appear:</p>

<p><img src="../../images/accepted_hits_2.png" alt="Accepted hits c2-r1x and D. melanogaster button" /></p>

<p>Finally, use <strong>D. melanogaster</strong> link (highlighted above) and follow the on-screen instructions. By focusing IGV on genomic position <code class="language-plaintext highlighter-rouge">chrX:11,897,111-11,920,446</code> you will be able to see spliced alignments produced by TopHat:</p>

<p><img src="../../images/igv_tophat.png" alt="IGV TopHat" /></p>

<p>and <a href="https://software.broadinstitute.org/software/igv/Sashimi">sashimi plots</a> highlighting potential splice junctions:</p>

<p><img src="../../images/sashimi.png" alt="Sashimi plots" /></p>

<h3 id="performing-differential-expression-analysis">Performing differential <span class="notranslate">expression</span> analysis</h3>

<p>Using mapped <span class="notranslate">reads</span> produced by TopHat we will perform analysis of differential gene <span class="notranslate">expression</span> using HTSeq/DESeq2 pipeline.</p>

<h4 id="gene-based-read-counting-with-htseq-count">Gene-based read counting with HTseq-count</h4>

<p><a href="https://www-huber.embl.de/users/anders/HTSeq/doc/count.html"><code class="language-plaintext highlighter-rouge">HTSeq-count</code></a> is one of the most popular tools for gene quantification. <code class="language-plaintext highlighter-rouge">HTseq-count</code> gives you multiple choices on how to handle read <span class="notranslate">mapping</span> to multiple locations, <span class="notranslate">reads</span> overlapping introns, or <span class="notranslate">reads</span> that overlap more than one genomic feature:</p>

<p><a href="https://www-huber.embl.de/users/anders/HTSeq/doc/count.html"><img src="../../images/htseq_count.png" alt="HTseq-count read/feature overlap modes" /></a></p>

<p><strong><code class="language-plaintext highlighter-rouge">HTseq-count</code> read/feature overlap modes</strong><br />
The <code class="language-plaintext highlighter-rouge">htseq-count</code> script of the HTSeq suite offers three different modes to handle details of read‚Äìfeature overlaps that are depicted here. The default of featureCounts is the behavior of the union option. Image is from <a href="https://www-huber.embl.de/users/anders/HTSeq/doc/count.html">HTseq documentation</a>; Caption by <a href="http://chagall.med.cornell.edu/RNASEQcourse/">D√ºndar:2015</a></p>

<p>Before we can use <code class="language-plaintext highlighter-rouge">HTseq-count</code> we need to download gene annotations for version <code class="language-plaintext highlighter-rouge">dm3</code> of the <em>Drosophila melanogaster</em> genome. We use version <code class="language-plaintext highlighter-rouge">dm3</code> because it is the same genome we have mapped <span class="notranslate">reads</span> against during the TopHat step.</p>

<h4 id="getting-drosophila-malanogaster-gene-annotation-from-ucsc">Getting <em>Drosophila malanogaster</em> gene annotation from UCSC</h4>

<p>Select <strong>UCSC Main</strong> from <strong>Get Data</strong> section of the menu. Within the UCSC Genome Browser interface set parameters as shown below. In particular make sure that <strong>assembly</strong> is set ti <code class="language-plaintext highlighter-rouge">dm3</code> and <strong>output format</strong> is set to <code class="language-plaintext highlighter-rouge">GTF</code>. Click <strong>get output</strong>.
<a href="../../images/ucsc_dm3.png"><img src="../../images/ucsc_dm3.png" alt="UCSC table browser" /></a></p>

<p>This <a href="https://www.ensembl.org/info/website/upload/gff.html">GTF</a> dataset will be used one of the input for HTseq-count.</p>

<p><code class="language-plaintext highlighter-rouge">HTseq-count</code> takes two inputs: (1) mapped <span class="notranslate">reads</span> in BAM format and (2) a GTF dataset containing annotation of genes. Using these inputs it will compute the number of <span class="notranslate">reads</span> per gene.</p>

<h4 id="using-htseq-count-in-galaxy">Using HTseq-count in <span class="notranslate">Galaxy</span></h4>

<p><code class="language-plaintext highlighter-rouge">htseq-count</code> needs strand information to proceed. The strand information is specified as <code class="language-plaintext highlighter-rouge">+</code>, <code class="language-plaintext highlighter-rouge">-</code>, or <code class="language-plaintext highlighter-rouge">.</code> (unknown) in a GTF dataset. <code class="language-plaintext highlighter-rouge">htseq-count</code> does not like <code class="language-plaintext highlighter-rouge">.</code> and will generate an error if such unstranded features appear in data. To prevent these errors from happening we will filter them out from GTF file using <strong>Filter</strong> tool from <strong>Filter and Sort</strong> section of tool menu. Here <code class="language-plaintext highlighter-rouge">c7 != "."</code> means that we need to filter all rows where strand column (column #7) contains a dot:</p>

<p><img src="../../images/filter_gtf.png" alt="Filter parameters" /></p>

<p>Select <strong>htseq-count</strong> from <strong>NGS: <span class="notranslate">RNA</span> analysis</strong> section on the left side of the menu. Set parameters as shown below. The red arrow points that to enable <code class="language-plaintext highlighter-rouge">htseq-count</code> to see collections, you need to select the ‚Äòfolder‚Äô button. In the case of this particular <span class="notranslate">Galaxy</span> history we will need to run <code class="language-plaintext highlighter-rouge">htseq-count</code> twice. Once on TopHat alignemnts for collection <strong>c1</strong> (dataset #37; shown below) and then on alignments for collection <strong>c2</strong> (dataset # 57).|
<img src="../../images/htseq_count_interface.png" alt="htseq-count parameters" /></p>

<p>This will generate read counts per gene.</p>

<h4 id="normalizing-read-counts-and-computing-differential-expression-with-deseq2">Normalizing read counts and computing differential <span class="notranslate">expression</span> with <code class="language-plaintext highlighter-rouge">DESeq2</code></h4>

<p><code class="language-plaintext highlighter-rouge">DESeq2</code> takes read counts produced by <code class="language-plaintext highlighter-rouge">HTseq-count</code> and apply size factor normalization. Specifically, <code class="language-plaintext highlighter-rouge">DESeq2</code> will:</p>

<ul>
  <li>For each gene, compute the geometric mean of read counts across all samples;</li>
  <li>Every gene count in then divided by the geometric mean;</li>
  <li>The median of these ratios is a sample‚Äôs size factor used for normalization.</li>
</ul>

<font color="red">&#10148;</font>
<p>Note: For a comprehensive overview of differential gene <span class="notranslate">expression</span> with DESeq2 see <a href="https://www.bioconductor.org/packages/3.3/bioc/vignettes/DESeq2/inst/doc/DESeq2.pdf">Love:2016</a>.</p>

<h4 id="deseq2-in-galaxy">DESeq2 in <span class="notranslate">Galaxy</span></h4>

<p>The <code class="language-plaintext highlighter-rouge">DESeq2</code> <span class="notranslate">Galaxy</span>‚Äôs interface is shown below. <code class="language-plaintext highlighter-rouge">DESeq2</code> allows to incorporate multiple <em>factors</em> in the analysis. In our case we only have one factor, which we call <strong>Conditions</strong>. This is because we are trying to find genes that are differentially expressed between two conditions. The first condition will the first <strong>factor level</strong>, while condition 2 will be the second <strong>factor level</strong>. Here the input for this first factor level is set to a collection <code class="language-plaintext highlighter-rouge">84: htseq-count on collection 37</code> and the input for the second input is set to <code class="language-plaintext highlighter-rouge">92: htseq-count on collection 57</code>. Make sure that <strong>Visualising the analysis results</strong> is set to <code class="language-plaintext highlighter-rouge">Yes</code>:</p>

<p><img src="../../images/deseq2_interface.png" alt="DESeq2 parameters" /></p>

<p>This will produce output as shown below. The columns are: (<strong>1</strong>) gene identifier, (<strong>2</strong>) mean normalised counts, averaged over all samples from both conditions, (<strong>3</strong>) logarithm (base 2) of the fold change, (<strong>4</strong>) the standard error estimate for the log2 fold change estimate, (<strong>5</strong>) <a href="https://en.wikipedia.org/wiki/Wald_test">Wald test</a> statistic, (<strong>6</strong>) p value for the statistical significance of this change, and (<strong>7</strong>) <em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate (<a href="https://en.wikipedia.org/wiki/False_discovery_rate">FDR</a>). There is only one gene with significant change in gene <span class="notranslate">expression</span> between conditions: <code class="language-plaintext highlighter-rouge">CG1803-RC</code> with <em>p</em>-value = 1.6x10<sup>-05</sup></p>

<p><img src="../../images/deseq2_output.png" alt="DESeq2 output" /></p>

<p>In addition to the list of genes DESeq2 outputs a graphical summary of the result. It includes a number of plots that should be used to evaluate the quality of the experiment. The histogram of <em>p</em>-values below shows that in our sample there is in fact just one instance of a significant <em>p</em>-value:</p>

<p><img src="../../images/p_val_hist.png" alt="p values histogram" /></p>

<p>The <a href="https://en.wikipedia.org/wiki/MA_plot">MA plot</a> below shows the relationship between the <span class="notranslate">expression</span> change (M) and average <span class="notranslate">expression</span> strength (A). Genes with adjusted <em>p</em>-value &lt; 0.1 are in red (there is only one such gene in thi sample at the bottom of the graph):</p>

<p><img src="../../images/MA_plot.png" alt="MA plot" /></p>

<p>The Principal Component Analysis (<a href="https://en.wikipedia.org/wiki/Principal_component_analysis">PCA</a>) shows the separation between Condition 1 and 2. This type of plot is useful for visualizing the overall effect of experimental covariates and batch effects (each replicate is plotted as an individual data point):</p>

<p><img src="../../images/pca.png" alt="The Principal Component Analysis output" /></p>

<p>A heatmap of sample-to-sample distance matrix gives us an overview over similarities and dissimilarities between samples:</p>

<p><img src="../../images/euc_dist.png" alt="A heatmap of sample-to-sample distance matrix" /></p>


                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Reference-based RNAseq data analysis (long) (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Anton Nekrutenko, 2021 <b>Reference-based RNAseq data analysis (long) (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html">/training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-rb-rnaseq,
    author = "Anton Nekrutenko",
    title = "Reference-based RNAseq data analysis (long) (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Anton Nekrutenko)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/rb-rnaseq/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
