<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Pre-processing of Single-Cell RNA Data / Pre-processing of Single-Cell RNA Data / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@d7df185d039afa3d9d09a81c704876f007224bbd",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Pre-processing of Single-Cell RNA Data">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-10-10 13:37:19 +0000">
<meta name="DC.creator" content="Mehmet Tekman">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Anika Erxleben">
<meta name="DC.creator" content="Wolfgang Maier"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Pre-processing of Single-Cell RNA Data / Hands-on: Pre-processing of Single-Cell RNA Data">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scrna-preprocessing/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Pre-processing of Single-Cell RNA Data",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2023-10-10 13:37:19 +0000",
  "datePublished": "2019-02-22 19:53:50 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00251",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Single Cell",
    "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!\n\nWhat else do you want to see? You can submit tool and/or tutorial request on our <i class=\"fa fa-wrench\" aria-hidden=\"true\"></i> [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).\n {: .alert.alert-success}\n",
    "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
  },
  "abstract": "# Introduction",
  "learningResourceType": "e-learning",
  "name": "Pre-processing of Single-Cell RNA Data",
  "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html",
  "version": 2,
  "timeRequired": "PT3H",
  "teaches": "- Demultiplex FASTQ data via UMI-tools\n- Understand and validate the extraction of barcodes\n- Obtain an overview of general alignment and quantification techniques\n- Generate a count matrix for downstream single-cell RNA analysis",
  "keywords": [
    "Single Cell"
  ],
  "description": "## Abstract\n\n# Introduction\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - What is single-cell?\n - How do we process a batch?\n - How do we process multiple batches?\n - How do we check for cross-contamination?\n\n\n## Learning Objectives\n\n- Demultiplex FASTQ data via UMI-tools\n- Understand and validate the extraction of barcodes\n- Obtain an overview of general alignment and quantification techniques\n- Generate a count matrix for downstream single-cell RNA analysis\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-intro/slides.html",
      "name": "An introduction to scRNA-seq data analysis",
      "description": "Slides for 'An introduction to scRNA-seq data analysis' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/mtekman/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/mtekman/",
      "name": "Mehmet Tekman",
      "image": "https://avatars.githubusercontent.com/mtekman",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/ifb/",
          "name": "Institut Français de Bioinformatique",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.france-bioinformatique.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "name": "Anika Erxleben",
      "image": "https://avatars.githubusercontent.com/erxleben",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eurosciencegateway/",
          "name": "EuroScienceGateway",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://galaxyproject.org/projects/esg/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7427-6478",
      "orcid": "https://orcid.org/0000-0002-7427-6478"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Single Cell",
      "description": "Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!\n\nWhat else do you want to see? You can submit tool and/or tutorial request on our <i class=\"fa fa-wrench\" aria-hidden=\"true\"></i> [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).\n {: .alert.alert-success}\n",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/3253142",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/single-cell" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Single Cell
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-preprocessing/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-single-cell">
    <h1 data-toc-skip>Pre-processing of Single-Cell RNA Data</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/mtekman/" class="contributor-badge contributor-mtekman"><img src="https://avatars.githubusercontent.com/mtekman?s=36" alt="Mehmet Tekman avatar" width="36" class="avatar" />
    Mehmet Tekman</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Bérénice Batut avatar" width="36" class="avatar" />
    Bérénice Batut</a><a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge contributor-erxleben"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/erxleben?s=36" alt="Anika Erxleben avatar" width="36" class="avatar" />
    Anika Erxleben</a><a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Wolfgang Maier avatar" width="36" class="avatar" />
    Wolfgang Maier</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What is single-cell?</p>
</li>
        
        <li><p>How do we process a batch?</p>
</li>
        
        <li><p>How do we process multiple batches?</p>
</li>
        
        <li><p>How do we check for cross-contamination?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Demultiplex FASTQ data via UMI-tools</p>
</li>
        
        <li><p>Understand and validate the extraction of barcodes</p>
</li>
        
        <li><p>Obtain an overview of general alignment and quantification techniques</p>
</li>
        
        <li><p>Generate a count matrix for downstream single-cell RNA analysis</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/single-cell/tutorials/scrna-intro/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: An introduction to scRNA-seq data analysis</a>
                            </li>
                        
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3253142">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/workflows/" title="Pre-processing of Single-Cell RNA Data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main)
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Feb 22, 2019 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Oct 10, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00251">https://gxy.io/GTN:T00251</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">4.2</a> (0 recent ratings, 10 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 2 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<h2 id="why-do-single-cell-sequencing">Why do Single Cell sequencing?</h2>

<p>Single-cell RNA (scRNA) sequencing is the technological successor to classical “bulk” RNA-seq, where samples are no longer defined at the tissue level but at the individual cell level. The bulk RNA-seq methods seen in <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html">previous hands-on material</a> would give the average expression of genes in a sample, whilst overlooking the distinct expression profiles given by the cell sub-populations due to their heterogeneity.</p>

<p>The rise of scRNA sequencing provides the means to explore this heterogeneity by examining samples at the individual cell level, enabling a greater understanding of the development and function of such samples, by the characteristics of their constituent cells.</p>

<p>Consider the heterogeneity of cells sampled from bone marrow, where hematopoietic stem cells can give rise to many different cell types within the same tissue:</p>

<figure id="figure-1" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_hematopoiesis.png" alt="Cell Differentiation of Hematopoietic Stem cells. " width="1100" height="492" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_hematopoiesis.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Cell Differentiation of Hematopoietic Stem cells. Image Source: https://commons.wikimedia.org/wiki/File:Hematopoiesis_simple.svg</figcaption></figure>

<p>The genes expressed by these cells at different developmental time points can be subtle, but generally can be classified into discrete cell sub-populations or under statistical clustering methods such as PCA or tSNE. Cells in the same cluster exhibit similar profiles of differential expression in the same set of related genes, compared to cells in other clusters. By identifying significant genes in each cluster, cell types and cell developmental hierarchies can be inferred based on the proximity of these clusters to one another.</p>

<p>Other than cell development, there are many more factors that can shape the level of gene expression exhibited by a given cell. Intercellular cell-signalling can block or enhance specific transcripts, the total amount of transcripts of a cell increases with the cell-cycle, or the proximity of a cell within a tissue to nutrients or oxygen.</p>

<figure id="figure-2" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_nbt3711.png" alt="Facets of Cellular Identity. " width="1167" height="596" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_nbt3711.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Revealing the vectors of cellular identity with single-cell genomics, Nature Biotechnology, 2016</figcaption></figure>

<p>This tutorial is in part inspired by aspects of the <a href="https://hemberg-lab.github.io/scRNA.seq.course/">Hemberg workflow</a> at the Sanger institute, as well as the <a href="https://github.com/CGATOxford/UMI-tools">CGATOxford workflow</a> which provides the <strong>UMI-tools</strong> suite that we make use of. The barcoding follows the CEL-Seq2 protocol <span class="citation"><a href="#Hashimshony2016">Hashimshony <i>et al.</i> 2016</a></span>,  mentioned in the <a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html"><em>Understanding Barcodes</em></a> hands-on, and uses the same lane configuration as utilised by the <a href="https://www.ie-freiburg.mpg.de/gruen">Freiburg MPI Grün lab</a>.</p>

<h1 id="analysis-strategy">Analysis Strategy</h1>

<p>Most scRNA sequencing techniques use pooled-sequencing approaches to generate a higher throughput of data by performing amplification and sequencing upon multiple cells in the same “pool”. From a bioinformatics standpoint, this means that the output FASTQ data from the sequencer is batch-specific and contains all the sequences from multiple cells, where one sample of cells is equal to one batch.</p>

<p>In this tutorial, we will perform pre-processing upon scRNA FASTQ batch data to generate an <em>N</em>-by-<em>M</em>  count matrix of <em>N</em> cells and <em>M</em> genes, with each element indicating the level of expression of that gene in a particular cell.</p>

<p>This count matrix is crucial for performing the downstream analysis, where differential gene analysis is performed between cells in order to cluster them into groups denoting their cell type and lineage.</p>

<p>The tutorial is structured in two parts:</p>

<ol>
  <li><strong>Single-Batch Processing</strong></li>
  <li><strong>Multi-Batch Processing</strong></li>
</ol>

<figure id="figure-3" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow.svg" alt="Overview of workflow. " width="933" height="978" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow.svg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> An overview of the entire workflow</figcaption></figure>

<p>The first part of this tutorial will deal with batches, and use example <em>FASTQ</em> data from a single batch, which we will perform <a href="#barcode-extraction">barcode extraction</a> and annotation upon. Alignment and quality control will also be performed, and we will see how to construct a rudimentary count matrix.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-10x-datasets"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: 10x Datasets</div>

  <p>10x Genomics datasets can be processed in a much easier way that is outlined in the <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html"><em>Pre-processing of 10X Single-Cell RNA Datasets</em></a> tutorial.</p>

  <p>However, much of the essential concepts for scRNA-seq pre-processing are not explained, so it is a good idea to familiarise yourself with them in this tutorial.</p>

</blockquote>

<p>The second part of this tutorial will deal with merging several output count matrices from multiple single batches generated in the first portion. Here, a set of example count matrices are <a href="#merging-count-matrices">merged together</a> and quality control performed. This produces a final count matrix valid for downstream analysis.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a>    <ol>
      <li><a href="#why-do-single-cell-sequencing" id="markdown-toc-why-do-single-cell-sequencing">Why do Single Cell sequencing?</a></li>
    </ol>
  </li>
  <li><a href="#analysis-strategy" id="markdown-toc-analysis-strategy">Analysis Strategy</a></li>
  <li><a href="#single-batch-processing" id="markdown-toc-single-batch-processing">Single-Batch Processing</a>    <ol>
      <li><a href="#data-upload-and-organization" id="markdown-toc-data-upload-and-organization">Data upload and organization</a></li>
      <li><a href="#barcode-extraction" id="markdown-toc-barcode-extraction">Barcode Extraction</a></li>
      <li><a href="#mapping--alignment" id="markdown-toc-mapping--alignment">Mapping / Alignment</a></li>
      <li><a href="#filtering" id="markdown-toc-filtering">Filtering</a></li>
      <li><a href="#quantification" id="markdown-toc-quantification">Quantification</a></li>
    </ol>
  </li>
  <li><a href="#multi-batch-processing" id="markdown-toc-multi-batch-processing">Multi-Batch Processing</a>    <ol>
      <li><a href="#data-upload-and-organisation" id="markdown-toc-data-upload-and-organisation">Data upload and organisation</a></li>
      <li><a href="#merging-count-matrices" id="markdown-toc-merging-count-matrices">Merging Count Matrices</a></li>
      <li><a href="#guarding-against-cross-contamination" id="markdown-toc-guarding-against-cross-contamination">Guarding against Cross-Contamination</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="single-batch-processing">Single-Batch Processing</h1>

<h2 id="data-upload-and-organization">Data upload and organization</h2>

<p>In this tutorial we will be analysing scRNA-seq data of bone marrow cells taken from a single C57 mouse by <em>Herman et al.</em> (<span class="citation"><a href="#Herman2018">Herman <i>et al.</i> 2018</a></span>) and producing a count matrix that we can use for later analysis.</p>

<p>The size of scRNA files (.fastq) are typically in the gigabyte range and are somewhat impractical for training purposes, so we will expedite the analysis by using a smaller subset of actual batch data. We will also be using <em>Mus Musculus</em> annotation data (.gtf) from the  <a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Mus_musculus/">NCBI RefSeq</a> track, as well as a barcodes file (.tsv).</p>

<figure id="figure-4" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_upper.png" alt="Workflow Upper. " width="2500" height="1124" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_upper.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Single batch processing</figcaption></figure>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-and-organization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload and organization</div>

  <ol>
    <li>
      <p>Create a new history and rename it (e.g. scRNA-seq single batch tutorial)</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the subset FASTQ paired data from <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3253142/files/SRR5683689_1.subset.fastq
https://zenodo.org/record/3253142/files/SRR5683689_2.subset.fastq
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>     <p>Click on <strong>Collection</strong> on the top</p>   </li>   <li>     <p>Click on <strong>Collection Type</strong> and select <code class="language-plaintext highlighter-rouge">Paired</code></p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Change <strong>Genome</strong> to <code class="language-plaintext highlighter-rouge">GRCm38/mm10</code></p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li>     <p>Click on <strong>Build</strong> when available</p>   </li>   <li>Ensure that the forward and reverse reads are set to <code class="language-plaintext highlighter-rouge">SRR5683689_1</code> and <code class="language-plaintext highlighter-rouge">SRR5683689_2</code>, respectively.     <ul>       <li>Click <strong>Swap</strong> otherwise</li>     </ul>   </li>   <li>     <p>Enter a name for the collection</p>     <ul>       <li>         <p>A useful naming convention is to use <code class="language-plaintext highlighter-rouge">&lt;name&gt;_&lt;plate&gt;_&lt;batch&gt;</code> to preserve the sample names, sequencing plate number and batch number.</p>       </li>       <li>         <p>Here we will write <code class="language-plaintext highlighter-rouge">C57_P1_B1</code></p>       </li>     </ul>   </li>   <li>Click on <strong>Create list</strong> (and wait a bit)</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the Gene Annotations and Barcodes from <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3253142/files/Mus_musculus.GRCm38.93.mm10.UCSC.ncbiRefSeq.gtf
https://zenodo.org/record/3253142/files/celseq_barcodes.192.tabular
</code></pre></div>      </div>
    </li>
    <li>
      <p>Set the datatype of the <code class="language-plaintext highlighter-rouge">celseq_barcodes.192.tabular</code> to <code class="language-plaintext highlighter-rouge">tabular</code></p>
    </li>
  </ol>

</blockquote>

<h2 id="barcode-extraction">Barcode Extraction</h2>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-note"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Note</div>

  <p>Before performing the barcode extraction process, it is recommended that you familiarise yourself with the concepts of designing cell barcodes as given by the <a href="/training-material/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html"><em>Plates, Batches, and Barcodes</em></a>, as well as the <a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html"><em>Understanding Barcodes</em></a> hands-on material for an introduction into transcript barcodes.</p>

</blockquote>

<p>We will be demultiplexing our FASTQ batch data by performing barcode extraction whilst also making use of the provided barcodes file to filter for <em>specific</em> cell barcodes.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-barcode-extraction"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Barcode Extraction</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/umi_tools_extract/umi_tools_extract/0.5.5.1" title="UMI-tools extract tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>UMI-tools extract</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.5.5.1)</span> with the following parameters:
      <ul>
        <li><em>“Library type”</em>: <code class="language-plaintext highlighter-rouge">Paired-end Dataset Collection</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Reads in FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">C57_P1_B1</code> (Our paired set)</li>
            <li><em>“Barcode on both reads?”</em>: <code class="language-plaintext highlighter-rouge">Barcode on first read only</code></li>
          </ul>
        </li>
        <li><em>“Use Known Barcodes?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Barcode File”</em>: <code class="language-plaintext highlighter-rouge">celseq_barcodes.192.tabular</code> (Input dataset)</li>
          </ul>
        </li>
        <li><em>“Barcode pattern for first read”</em>: <code class="language-plaintext highlighter-rouge">NNNNNNCCCCCC</code></li>
        <li><em>“Enable quality filter?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Verifying that the desired UMI and cell barcodes have been extracted from the sequence of the Forward reads and inserted into the header of the Reverse reads is encouraged, using the method outlined in the above hands-on material.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many reads were filtered out, and why?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The input FASTQ files contained reads from all barcodes, including those with sequencing errors, resulting in a larger pool of detected barcodes than those desired. (e.g. Cell barcode <code class="language-plaintext highlighter-rouge">AAATTT</code> could have single base-pair sequencing errors that could modify it into <code class="language-plaintext highlighter-rouge">ATATTT</code> or <code class="language-plaintext highlighter-rouge">AAACTT</code>, etc).</p>

    <p>This information is included in the Log file of <strong>UMI-tools extract</strong> which contains all the parameters used to run, as well as <em>INFO</em> lines that indicate how many reads were read, and how many output. In this case: 134431 reads were retained (&gt;90% of input reads).</p>
  </blockquote>
</blockquote>

<h2 id="mapping--alignment">Mapping / Alignment</h2>

<p>FASTQ files contain sequence information that we wish to map to genes in a genome. Mapping is a relatively straightforward process, and is covered more extensively in the <a href="/training-material/topics/sequence-analysis/index.html"><em>Sequence Analysis</em></a> tutorials:</p>

<ol>
  <li>Select your genome</li>
  <li>Select your gene annotation file</li>
  <li>Run the alignment program</li>
  <li>(Optional) Run MultiQC to assess the quality of the mapping</li>
</ol>

<p>The FASTQ data was sequenced from mouse data, so to perform the alignment we will need to gather all data relevant to that genome. We will use the latest version (mm10).</p>

<p>The annotation GTF file must match the genome version used, since both use physical coordinates. Each GTF contains all the gene, exon, intron, and other regions of interest that we will use to annotate our reads, should our reads align to any of the regions specified in this file.</p>

<p>For alignment, we will use RNA-STAR for performance and splice-awareness.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-performing-the-alignment"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Performing the Alignment</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/rgrnastar/rna_star/2.7.7a" title="RNA-STAR tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>RNA-STAR</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.7.7a)</span> with the following parameters:
      <ul>
        <li><em>“Single-end or paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“RNA-Seq FASTQ/FASTA file”</em>: <code class="language-plaintext highlighter-rouge">Reads2</code> (output of <strong>UMI-tools extract</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>“Custom or built-in reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in index</code>
          <ul>
            <li><em>“Reference genome with or without an annotation”</em>: <code class="language-plaintext highlighter-rouge">use genome reference without builtin gene-model</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select reference genome”</em>: <code class="language-plaintext highlighter-rouge">Mus Musculus (mm10)</code> (Mouse)</li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene model (gff3,gtf) file for splice junctions”</em>: <code class="language-plaintext highlighter-rouge">Mus_musculus.GRCm38.93.mm10.UCSC.ncbiRefSeq</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9)</span> with the following parameters:
      <ul>
        <li><em>“Results”</em>:
          <ul>
            <li><em>“1: Results”</em>:
              <ul>
                <li><em>“Which tool was used to generate logs?”</em>:<code class="language-plaintext highlighter-rouge">STAR</code></li>
                <li><em>“STAR output”</em>:
                  <ul>
                    <li><em>“1: STAR output”</em>
                      <ul>
                        <li><em>“Type of STAR output?”</em>:<code class="language-plaintext highlighter-rouge">Log</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li><em>“STAR log output”</em> :(Select the STAR output file that ends in “log”)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> symbol on the MultiQC Webpage</li>
  </ol>

</blockquote>

<p>The purpose of MultiQC is to observe how well our reads were mapped against the reference genome. Many reads are discarded due to being of too low quality, or having ambiguous sequence content that can map them to multiple locations.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What percentage of our reads are uniquely mapped? How many millions of reads is this percentage?</li>
    <li>What percentage of our reads are mapped to more than one locus?</li>
    <li>Is our overall mapping ‘good’ ?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li><code class="language-plaintext highlighter-rouge">59.5%</code> or ~80k reads were successfully mapped</li>
      <li><code class="language-plaintext highlighter-rouge">13.6%</code> are multiply mapped, and <code class="language-plaintext highlighter-rouge">3.7%</code> were mapped to too many loci
        <ul>
          <li>Multiply mapped means that a read was aligned to more than one gene</li>
          <li>Mapped to too many loci means that a read was aligned to 10 or more loci, and should be ignored.</li>
        </ul>
      </li>
      <li>It depends on how good the sequencing protocol is, and how many reads in total were mapped.
        <ul>
          <li><code class="language-plaintext highlighter-rouge">90%</code> is amazing, reserved for bulk RNA-seq which typically has high coverage</li>
          <li><code class="language-plaintext highlighter-rouge">70%</code> is weak for bulk RNA-seq, but good for single-cell RNA-seq</li>
          <li>This a small subset of a real dataset, but one would expect that 6 million mapped reads would be enough to generate a downstream analysis.</li>
        </ul>
      </li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="filtering">Filtering</h2>

<p>Before continuing let us first look back on some of the previous stages:</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-recap-of-previous-stages"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Recap of previous stages</div>

  <ol>
    <li>
      <p><em>Barcode Extraction</em>:</p>

      <p>Here we used <code class="language-plaintext highlighter-rouge">umi_tools extract</code> on our input forward and reverse FASTQ files, and extracted the UMI and cell barcode from the forward read <em>sequence</em>, and placed it into the <em>header</em> of both forward and reverse FASTQ files. i.e. FASTQ files → Modified FASTQ files</p>
    </li>
    <li>
      <p><em>Mapping</em>:</p>

      <p>We took the sequencing data from the reverse FASTQ file (with modified headers) and aligned it to the mouse genome, using annotations presented in the GTF file for that genome. i.e. Modified FASTQ file (reverse) → BAM file</p>
    </li>
  </ol>

</blockquote>

<h3 id="confirming-reads-in-the-bam-file">Confirming Reads in the BAM file</h3>

<p>We now have a BAM file of our aligned reads, with cell and UMI barcodes embedded in the read headers. We also have the chromosome and base-pair positions of where these reads are aligned. The can be confirmed by peeking into the BAM file:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-confirming-the-alignment-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Confirming the Alignment Data</div>

  <ol>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> symbol of the BAM output from STAR.</li>
    <li>There are many header lines that begin with <code class="language-plaintext highlighter-rouge">@</code> which we are not interested in.</li>
    <li>
      <p>Look at 10th read directly below the header lines:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR5683689.38437_GCATTC_CTTCGT	16	chr1	3439991	255	70M	*	0	0	CTTTGAATCTCTTCTTCCCAGCTAGTCATCTTCCTGCTTTTCTCTCTGTCTGTCTGTCTGTCTGTCTGTC	'0'&lt;B&lt;''B77&lt;BFBBBBB7'FBFB0F7FBB&lt;B'''&lt;IFFBF&lt;FBFB&lt;FBBFBB0&lt;BFFFBB0BBFFB&lt;&lt;	NH:i:1 HI:i:1 AS:i:66 nM:i:1
</code></pre></div>      </div>
    </li>
  </ol>

</blockquote>

<p>The fields of the BAM file can be better explained at section 1.4 of <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">the SAM specification</a>, but we will summarise the main fields of interest here:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">SRR568..._GCATTC_CTTCGT</code>: The <em>readname</em> appended by <code class="language-plaintext highlighter-rouge">_</code>, the cell barcode, another <code class="language-plaintext highlighter-rouge">_</code>, and then the UMI barcode.</li>
  <li><code class="language-plaintext highlighter-rouge">16</code>: The FLAG value</li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-what-does-the-alignment-flag-value-of-16-tell-us-about-this-read"><i class="far fa-question-circle" aria-hidden="true" ></i> Question: What does the alignment flag value of 16 tell us about this read?</div>

  <p><!-- TODO This information needs to be integrated into an actual tool -->
  We can interactively see what the different FLAG values mean by feeding values into the SAM specification to the <a href="https://broadinstitute.github.io/picard/explain-flags.html">Picard web tool</a></p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The read aligns to the reverse strand</p>

  </blockquote>

</blockquote>

<ul>
  <li><code class="language-plaintext highlighter-rouge">chr1</code> <code class="language-plaintext highlighter-rouge">3439991</code>: The position and base-pair of alignment of the first base of the sequence.</li>
  <li>A series of quality fields, with the main contributors being  the sequence and sequence quality strings.</li>
  <li><code class="language-plaintext highlighter-rouge">NH</code>: The number of hits for  this read. If it is multiply mapped, then the number of multiples will be shown (here <code class="language-plaintext highlighter-rouge">1</code>, so not multiply mapped).</li>
  <li><code class="language-plaintext highlighter-rouge">HI</code>: Which number this particular read is in the series of (potentially) multi-mapped reads (here <code class="language-plaintext highlighter-rouge">1</code>, not necessarily meaning the first or ‘better’).</li>
  <li><code class="language-plaintext highlighter-rouge">nM</code>: The number of base mismatches in the alignment of this read to the reference (here <code class="language-plaintext highlighter-rouge">1</code>).</li>
</ul>

<h3 id="filtering-the-bam-file">Filtering the BAM File</h3>

<p>If we perform counting on the current BAM file we will be counting all reads, even the undesirable ones such as those that did not align so optimally.</p>

<p>The main filtering steps performed on our reads so far have been relatively silent due to the ‘default’ parameters used.</p>
<ul>
  <li><em>UMI-tools Extract</em> - Filters reads for those only with matching barcodes given by our barcodes file.</li>
  <li><em>RNA-STAR</em> - As seen in the log, we lose 25% of our reads for being too short or being multiply mapped.</li>
</ul>

<p>Another filtering measure we can apply is to keep reads that we are confident about, e.g those with a minimum number of mismatches to the reference within an acceptable range. Specifically, we want to keep all reads that align to the forward or reverse strand that also have less that 3 mismatches to the reference, and are also mapped only once to the reference.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Task description</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Filter BAM datasets on a variety of attributes tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter BAM datasets on a variety of attributes</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.4.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BAM dataset(s) to filter”</em>: <code class="language-plaintext highlighter-rouge">output_bam</code> (output of <strong>RNA STAR</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Condition”</em>:
          <ul>
            <li>In <em>“1: Condition”</em>:
              <ul>
                <li>In <em>“Filter”</em>:
                  <ul>
                    <li>In <em>“1: Filter”</em>:
                      <ul>
                        <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">alignmentFlag</code>
                          <ul>
                            <li><em>“Filter on this alignment flag”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <em>“Insert Condition”</em>:</li>
            <li>In <em>“2: Condition”</em>:
              <ul>
                <li>In <em>“Filter”</em>:
                  <ul>
                    <li>In <em>“1: Filter”</em>:
                      <ul>
                        <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">alignmentFlag</code>
                          <ul>
                            <li><em>“Filter on this alignment flag”</em>: <code class="language-plaintext highlighter-rouge">16</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <em>“Insert Condition”</em>:</li>
            <li>In <em>“3: Condition”</em>:
              <ul>
                <li>In <em>“Filter”</em>:
                  <ul>
                    <li>In <em>“1: Filter”</em>:
                      <ul>
                        <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">tag</code>
                          <ul>
                            <li><em>“Filter on a particular tag”</em>: <code class="language-plaintext highlighter-rouge">nM:&lt;3</code>
<small>(<strong>Attention! please use a lowercase ‘n’ here!</strong>)</small></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Click on <em>“Insert Condition”</em>:</li>
            <li>In <em>“4: Condition”</em>:
              <ul>
                <li>In <em>“Filter”</em>:
                  <ul>
                    <li>In <em>“1: Filter”</em>:
                      <ul>
                        <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">tag</code>
                          <ul>
                            <li><em>“Filter on a particular tag”</em>: <code class="language-plaintext highlighter-rouge">NH:&lt;2</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Would you like to set rules?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Enter rules here”</em>: <code class="language-plaintext highlighter-rouge">(1 | 2) &amp; 3 &amp; 4</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Why are we filtering only for alignment flags <code class="language-plaintext highlighter-rouge">0</code> and <code class="language-plaintext highlighter-rouge">16</code>?</li>
    <li>What do the tag filters <code class="language-plaintext highlighter-rouge">nM:&lt;3</code> and <code class="language-plaintext highlighter-rouge">NH:&lt;2</code> do?</li>
    <li>What is happening at the rules stage?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Alignment flags <code class="language-plaintext highlighter-rouge">0</code> and <code class="language-plaintext highlighter-rouge">16</code> specify that we wish to only keep reads that align to the forward and reverse strands.</li>
      <li>We are keeping reads that have a number of mismatches (<code class="language-plaintext highlighter-rouge">nM</code>) to the reference of less than 3, and has a number of hits (<code class="language-plaintext highlighter-rouge">NH</code>) across the reference of less than 2 (i.e. it is not a multiply-mapped read).</li>
      <li><a href="https://en.wikipedia.org/wiki/Boolean_expression">Boolean expressions</a> are applied that state that <em>either</em> conditions 1 or 2 can happen, in conjunction with rules 3 and 4 happening.</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="quantification">Quantification</h2>

<p>Once we have the name of the gene for a specific read, we can count how many of those reads fall into that gene and generate a count matrix.</p>

<p>Counting reads is performed by two commonly-used tools:</p>
<ol>
  <li>RNA-STAR</li>
  <li>FeatureCounts</li>
</ol>

<p>The <strong>RNA-STAR</strong> tools has the ability to count reads as it maps them. <strong>FeatureCounts</strong> performs the same task, but is capable of counting not just at the Read level, but also at the UMI level too, such that 10 duplicate reads will be counted only once.</p>

<p>Unfortunately, both are currently limited to counting without being able to distinguish between different cells.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-example"><button class="gtn-boxify-button details" type="button" aria-controls="details-example" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Example</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>If we consider the number of reads that align to <em>GeneA</em>, the output given by these two tools is as follows:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>(reads)</th>
          <th>RNA STAR</th>
          <th>FeatureCounts</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GeneA</td>
          <td>12</td>
          <td>12</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

  <p>But what we actually require is:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>(reads)</th>
          <th>C1</th>
          <th>C2</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GeneA</td>
          <td>10</td>
          <td>2</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

  <p>or more specifically:</p>

  <blockquote class="matrix">
    <table>
      <thead>
        <tr>
          <th>(UMIs)</th>
          <th>C1</th>
          <th>C2</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>GeneA</td>
          <td>2</td>
          <td>1</td>
        </tr>
      </tbody>
    </table>
  </blockquote>

</blockquote>

<p>In order to obtain this desired format, we must use <strong>UMI-tools count</strong> to perform the counting. However, this tool is dependent on <strong>FeatureCounts</strong> to annotate our reads with the one crucial piece of information that is missing from our BAM file: the name of the gene.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-verifying-missing-gene-name"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Verifying missing gene name</div>
  <p>You can check this yourself by examining the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> of the BAM file <em>“STAR Alignment file”</em></p>
</blockquote>

<h3 id="annotating-gene-name-with-featurecounts">Annotating Gene name with FeatureCounts</h3>

<p>Let us annotate our BAM file with desired gene tags.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quantification-assist-via-featurecounts"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quantification assist via FeatureCounts</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/featurecounts/featurecounts/2.0.1" title="FeatureCounts tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FeatureCounts</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Alignment file”</em>: <code class="language-plaintext highlighter-rouge">mapped_reads</code> (output of <strong>Filter BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Specify strand information”</em>:<code class="language-plaintext highlighter-rouge">Stranded (Forward)</code></li>
        <li><em>“Gene annotation file”</em>: <code class="language-plaintext highlighter-rouge">in your history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Gene annotation file”</em>: <code class="language-plaintext highlighter-rouge">Mus_musculus.GRCm38.93.mm10.UCSC.ncbiRefSeq.gtf</code></li>
          </ul>
        </li>
        <li>In <em>“Advanced options”</em>:
          <ul>
            <li><em>“Count multi-mapping reads/fragments”</em>: <code class="language-plaintext highlighter-rouge">Disabled; multi-mapping reads are excluded (default)</code></li>
            <li><em>“Exon-exon junctions”</em>: <code class="language-plaintext highlighter-rouge">No (default)</code></li>
            <li><em>“Annotates the alignment file with ‘XS:Z:’-tags to described per read or read-pair the corresponding assigned feature(s).”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Examine the output BAM file
      <ul>
        <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> for the “Feature Counts: Alignment File”</li>
        <li>Scroll down past the header lines</li>
        <li>Scroll horizontally to the tags, observe the new <code class="language-plaintext highlighter-rouge">XT</code> tag.</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The <code class="language-plaintext highlighter-rouge">XS</code> and <code class="language-plaintext highlighter-rouge">XT</code> tags in the BAM file will now form the basis for counting reads.
With all the relevant data now in our BAM file, we can actually perform the counting via <code class="language-plaintext highlighter-rouge">UMI-tools count</code>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-verifying-added-gene-name"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Verifying added gene name</div>
  <p>You can once again check this yourself by examining the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> of the BAM file <em>“STAR Alignment file”</em></p>
</blockquote>

<h3 id="counting-genes--cell">Counting Genes / Cell</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-final-quantification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Final Quantification</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/umi_tools_count/umi_tools_count/0.5.5.1" title="UMI-tools count tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>UMI-tools count</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.5.5.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sorted BAM file”</em>: <code class="language-plaintext highlighter-rouge">out_file1</code> (output of <strong>FeatureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“UMI Extract Method”</em>: <code class="language-plaintext highlighter-rouge">Barcodes are contained at the end of the read separated by a delimiter</code></li>
        <li><em>“Bam is paired-end”</em>:<code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Method to identify group of reads”</em>:<code class="language-plaintext highlighter-rouge">Unique</code></li>
        <li><em>“Extra Parameters”</em>:
          <ul>
            <li><em>“Deduplicate per gene.”</em>:<code class="language-plaintext highlighter-rouge">XT</code></li>
            <li><em>“Group reads only if they have the same cell barcode.”</em>:<code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>“Prepend a label to all column headers”</em>:<code class="language-plaintext highlighter-rouge">No modifications</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The important parameters to take note of are those given in the <em>Extra Parameters</em> where we specify that each of the reads with a <code class="language-plaintext highlighter-rouge">XT:Z</code> tag in the BAM file will be counted on a per cell basis. Reads sharing the same UMI and cell Barcodes will be de-duplicated into a single count, reducing PCR duplicate bias from the analysis.</p>

<p>At this stage, we now have a tabular file containing genes/features as rows, and cell labels as headers.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many genes do we have in the matrix?</li>
    <li>How many cells?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>
        <p>2,140 lines
This information can be seen in the file preview window by clicking on the name of the file (<strong>NOT</strong> the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> symbol).</p>
      </li>
      <li>
        <p>180 columns (not including the first column of gene names)
The number of columns can be seen by scrolling the file preview window completely to the right. 192 cell barcodes were given, but due to the subsetted data, only 180 were detected.</p>
      </li>
    </ol>

  </blockquote>
</blockquote>

<p>The generation of a single count matrix is now complete, with the emphasis on the word <em>single</em> due to the fact that we often deal in multiple batches of sequencing data.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-recap-of-previous-stages-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Recap of previous stages</div>

  <ol>
    <li>
      <p><em>Barcode Extraction</em>:</p>

      <p>Here we used <code class="language-plaintext highlighter-rouge">UMI_tools extract</code> on our input forward and reverse FASTQ files, and extracted the UMI and cell barcode from the forward read <em>sequence</em>, and placed it into the <em>header</em> of both forward and reverse FASTQ files. i.e. FASTQ files → Modified FASTQ files</p>
    </li>
    <li>
      <p><em>Mapping</em>:</p>

      <p>We took the sequencing data from the reverse FASTQ file (with modified headers) and aligned it to the mouse genome, using annotations presented in the GTF file for that genome. i.e. Modified FASTQ file (reverse) → BAM file</p>
    </li>
    <li>
      <p><em>Quality Filtering</em>:</p>

      <p>Reads with alignment mismatches greater than 2 were discarded, and only non multi-mapped reads that mapped to the forward or reverse strand were kept</p>
    </li>
    <li>
      <p><em>Quantification</em>:</p>

      <p>Gene tags were added to our alignment file, and reads were grouped according those sharing the same gene tag, with further reduction performed by collapsing all reads sharing the same cell and UMI barcode to be counted only once.</p>
    </li>
  </ol>

</blockquote>

<p>This concludes the first part of the tutorial which focused on the transformation of raw FASTQ data from a single batch into a count matrix. The second part of this tutorial guides us through the process of merging multiple processed batches from the first stage, and performing qualitative filtering.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-workflows-for-all-steps"><button class="gtn-boxify-button details" type="button" aria-controls="details-workflows-for-all-steps" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Workflows For All Steps</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Galaxy provides a workflow that captures the process of all the above stages for a <a href="workflows/scrna_pp_celseq.ga">single pair of FASTQ data</a>:</p>

  <figure id="figure-5" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_single.png" alt="Single Batch Workflow. " width="1416" height="935" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_single.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Single Batch Workflow</figcaption></figure>

  <p>For multiple batch processing, Galaxy can make use of <em>Nested Workflows</em> that in this case can take in a <a href="workflows/scrna_mp_celseq.ga">list of input paired FASTQ data</a> and process them in parallel.</p>

  <figure id="figure-6" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_nested.png" alt="Multiple Batch Workflow. " width="971" height="715" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_nested.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Nested Multiple Batch Workflow</figcaption></figure>

</blockquote>

<p>For the processing of 10x Genomics datasets, please refer to the <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing-tenx/tutorial.html"><em>Pre-processing of 10X Single-Cell RNA Datasets</em></a> tutorial.</p>

<h1 id="multi-batch-processing">Multi-Batch Processing</h1>

<figure id="figure-7" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_lower.png" alt="Workflow Lower. " width="2500" height="1016" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow_lower.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Multiple batch processing</figcaption></figure>

<p>Handling more than one batch of sequencing data is rather trivial when we take into account our main goals and requirements:</p>

<ol>
  <li>For each batch, convert FASTQ reads from into a count matrix.</li>
  <li>Merge all count matrices into a single count matrix</li>
</ol>

<p>The first step merely requires us to run the same workflow on each of our batches, using the exact same inputs except for the FASTQ paired data. The second step requires a minimal level of interaction from us; namely using a merge tool and selecting our matrices.</p>

<h2 id="data-upload-and-organisation">Data upload and organisation</h2>

<p>The count matrix we have generated in the previous section is too sparse to perform any reasonable analysis upon, and constitutes data only of a single batch. Here we will use more populated count matrices from multiple batches, under the assumption that we now know how to generate each individual one of them using the steps provided in the previous section. This data is available at <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a>.</p>

<p>Once again, file naming is important, and so we will rename our matrix files appropriately to the plate and batch they are supposed to originate from.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-and-organisation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload and organisation</div>

  <ol>
    <li>
      <p>Create a new history and rename it (<em>e.g.</em> scRNA-seq multiple-batch tutorial)</p>
    </li>
    <li>Import the eight matrices (<code class="language-plaintext highlighter-rouge">P1_B1.tsv</code>, <code class="language-plaintext highlighter-rouge">P1_B2.tsv</code>, etc.) and the barcodes file from <a href="https://zenodo.org/record/3253142"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> or from the data library (ask your instructor)
      <ul>
        <li>Set the datatype of the files to <strong>tabular</strong></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3253142/files/P1_B1.tsv
https://zenodo.org/record/3253142/files/P1_B2.tsv
https://zenodo.org/record/3253142/files/P1_B3.tsv
https://zenodo.org/record/3253142/files/P1_B4.tsv
https://zenodo.org/record/3253142/files/P2_B5.tsv
https://zenodo.org/record/3253142/files/P2_B6.tsv
https://zenodo.org/record/3253142/files/P2_B7.tsv
https://zenodo.org/record/3253142/files/P2_B8.tsv
https://zenodo.org/record/3253142/files/celseq_barcodes.192.tabular
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Rename a matrix
      <ul>
        <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> of the <em><code class="language-plaintext highlighter-rouge">P1_B1.tsv</code></em> file</li>
        <li>Set the Name field such that it is affixed with “_P1_B1” (e.g. ‘multibatch_P1_B1’)</li>
        <li>Click <em>Save</em></li>
      </ul>
    </li>
    <li>Repeat for all matrices
<strong>Pay attention to the Plate number which changes after Batch 4</strong></li>
  </ol>

</blockquote>

<h2 id="merging-count-matrices">Merging Count Matrices</h2>

<p>Before we begin, we must consider that our matrices are not equal – e.g. Batch1 has 3 cells that describe <em>Genes{A,B,C,D}</em> whereas Batch2 has 4 cells that describe <em>Genes{C,D,E}</em>.</p>

<p>We have the problem that only <em>GeneC</em> and <em>GeneD</em> appear in both batches, whilst describing 7 different cells in total.</p>

<p>To resolve this we can perform a “Full Table Join” where the missing data for <em>GeneE</em> and <em>GeneA</em> in Batch1 and Batch2 respectively are replaced with zeroes:</p>

<figure id="figure-8" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-scanpy-pbmc3k/scrna_downstream.svg" alt="Table Join. " width="616" height="494" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-scanpy-pbmc3k/scrna_downstream.svg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span> Full Table Join</figcaption></figure>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>For more information on table joins, see <a href="http://www.sql-join.com/sql-join-types/">this guide</a></p>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Why is it required to change the column headers in the Full matrix?</li>
    <li>Why were the cell labels in B1 and B2 the same, if they were labelling completely different cells?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Although the cell headers in each batch matrix is the same, the cells they label are <em>not</em> the same and need to be relabelled in the final matrix to tell us which batch they originated from.</li>
      <li>The reason the cell headers are the same is because the cells use the same barcodes, due to fact that the <em>same</em> barcodes are sometimes used across <em>different</em> batches.</li>
    </ol>

  </blockquote>

</blockquote>

<p>Let us now merge our matrices from different batches. In order to ensure that our batches are merged in the order that we wish, we should first create a list of datasets so that our matrices are merged in the order given by the list.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-table-merge"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Table Merge</div>

  <ol>
    <li>
      <p><strong>Create a Dataset List</strong></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a dataset collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create collection</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p><strong>Column Join on Collections</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
      <ul>
        <li><em>“Tabular Files”</em>: (Select the Dataset Collection icon <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span>, and select the Collection from the previous step each of the matrices that you wish to join)</li>
        <li><em>“Identifier column”</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“Number of Header lines in each item”</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“Add column name to header”</em>:<code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Fill character”</em>:<code class="language-plaintext highlighter-rouge">0</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The identifier column refers to the column where the gene names are listed. A 1:1 correspondence between matrices is checked, so that the merge does not concatenate the wrong rows between matrices. The <em>Fill character</em> provides a default value of 0 for cases where a Gene appears only in one of the matrices as per our example earlier.</p>

<p>Once the merge is complete, we can now peek at our full combined matrix by once again clicking on the file name to see a small summary.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Each of these matrices/batches come from the same organism.</p>

  <ol>
    <li>How much overlap in their detected genes did you expect?</li>
    <li>How much overlap in their detected genes did you observe?</li>
    <li>Why is this?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>
        <p>Given that they come from the same sample, and each matrix has ~15,000 genes, we would have expected a high overlap between matrices, yielding ~18,000 genes in the combined matrix if we assume a difference of +/- 500 genes per batch.</p>
      </li>
      <li>
        <p>We observe 20,800 genes, giving less overlap between batches of the same organism than we originally thought.</p>
      </li>
      <li>
        <p>The batches were sequenced at different time points along the organisms development, and therefore different genes were expressed/detected at different time points. For early development data, the overlap can be very sparse.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<p>In the new combined matrix we see that we have 1536 cells, but this number is greatly overestimated.  This is because <em>not all batches use the same barcodes</em>, and yet we applied the full set of 192 barcodes against our FASTQ data during the <a href="#barcode-extraction"><em>Barcode Extraction</em></a> stage previously.</p>

<p>The reason we do this is to test for cross-contamination between batches, the details of which are better explained in the <a href="/training-material/topics/single-cell/tutorials/scrna-preprocessing/slides.html"><em>accompanying slides</em></a>.</p>

<h2 id="guarding-against-cross-contamination">Guarding against Cross-Contamination</h2>

<p>There are multiple possible ways to configure a plate for sequencing multiple batches. Thankfully, Galaxy provides a tool that caters for this, and checks for cross-contamination in any experimental setup. It requires only the following information:</p>

<ol>
  <li>A full list of barcodes</li>
  <li>Which barcodes apply to which batches</li>
  <li>Which batches apply to which plates</li>
</ol>

<p>Since the plating protocol we are using is that designed by the Freiburg MPI Grün lab, we will follow their structure.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-plating-protocol"><button class="gtn-boxify-button details" type="button" aria-controls="details-plating-protocol" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Plating protocol</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>
      <p>Barcodes:
  These are each 8bp long, with an edit distance of 2, and there 192 of them.</p>

      <blockquote class="matrix">
        <table>
          <tbody>
            <tr>
              <td>001-006</td>
              <td>AACACC AACCTC AACGAG AACTGG AAGCAC AAGCCA</td>
            </tr>
            <tr>
              <td>007-012</td>
              <td>AAGGTG AAGTGC ACAAGC ACAGAC ACAGGA ACAGTG</td>
            </tr>
            <tr>
              <td>.</td>
              <td>.</td>
            </tr>
            <tr>
              <td>.</td>
              <td>.</td>
            </tr>
            <tr>
              <td>180-186</td>
              <td>TTACGC TTCACC TTCCAG TTCGAC TTCTCG TTGCAC</td>
            </tr>
            <tr>
              <td>187-192</td>
              <td>TTGCGA TTGCTG TTGGAG TTGGCA TTGGTC TTGTGC</td>
            </tr>
          </tbody>
        </table>
      </blockquote>
    </li>
    <li>
      <p>Plates:
  Here we have 8 batches spread out over 2 plates, with alternate barcode striping.</p>

      <blockquote class="matrix">
        <table>
          <thead>
            <tr>
              <th> </th>
              <th>001-096</th>
              <th>097-192</th>
              <th>001-096</th>
              <th>097-192</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Plate 1</td>
              <td>B1</td>
              <td>B2</td>
              <td>B3</td>
              <td>B4</td>
            </tr>
            <tr>
              <td>Plate 2</td>
              <td>B5</td>
              <td>B6</td>
              <td>B7</td>
              <td>B8</td>
            </tr>
          </tbody>
        </table>
      </blockquote>
    </li>
  </ul>

</blockquote>

<p>This plating protocol can be converted into a more textual format, which allows for many variable setups (see <em>Help</em> section of <em>Cross-contamination Barcode Filter</em> tool).</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[Barcodes → Batches]
001-096: B1 , B3 , B5 , B7
097-192: B2 , B4 , B6 , B8

[Plates → Batches]
1: B1 , B2 , B3 , B4
2: B5 , B6 , B7 , B8
</code></pre></div></div>

<p>Let us now apply this protocol to our count matrix, and look for any cross-contamination.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-barcode-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Barcode Filtering</div>

  <p>Select <strong>Cross-contamination Barcode Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
  <ul>
    <li><em>“Input Matrix”</em>:<code class="language-plaintext highlighter-rouge">Column Join output</code> (merged matrices)</li>
    <li><em>“Complete Barcodes”</em>:<code class="language-plaintext highlighter-rouge">celseq_barcodes.192.tabular</code> (barcodes file)</li>
    <li><em>“Plate Protocol”</em>:<code class="language-plaintext highlighter-rouge">Custom</code>
      <ul>
        <li><em>“Under ‘Barcode Format’“</em>:
          <ul>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Barcode Format</code>:
              <ul>
                <li><em>“1: Barcode Format”</em>:
                  <ul>
                    <li><em>“Barcode Range: Start”</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
                    <li><em>“Barcode Range: End”</em>:<code class="language-plaintext highlighter-rouge">96</code></li>
                    <li><em>“Batches utilizing this Range”</em>:<code class="language-plaintext highlighter-rouge">1,3,5,7</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Barcode Format</code>:
              <ul>
                <li><em>“2: Barcode Format”</em>:
                  <ul>
                    <li><em>“Barcode Range: Start”</em>:<code class="language-plaintext highlighter-rouge">97</code></li>
                    <li><em>“Barcode Range: End”</em>:<code class="language-plaintext highlighter-rouge">192</code></li>
                    <li><em>“Batches utilizing this Range”</em>:<code class="language-plaintext highlighter-rouge">2,4,6,8</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Under ‘Plate Format’“</em>:
          <ul>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Plate Format</code>:
              <ul>
                <li><em>“1: Plate Format”</em>:
                  <ul>
                    <li><em>“Plate Number”</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
                    <li><em>“Batches within this Plate Number”</em>:<code class="language-plaintext highlighter-rouge">1,2,3,4</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>Select <code class="language-plaintext highlighter-rouge">+ Insert Plate Format</code>:
              <ul>
                <li><em>“2: Plate Format”</em>:
                  <ul>
                    <li><em>“Plate Number”</em>:<code class="language-plaintext highlighter-rouge">2</code></li>
                    <li><em>“Batches within this Plate Number”</em>:<code class="language-plaintext highlighter-rouge">5,6,7,8</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Expand the <em>“RegEx Parameters”</em> section:
      <ul>
        <li><em>“RegEx to extract Plate, Batch, and Barcodes from headers”</em>:<code class="language-plaintext highlighter-rouge">.*P(\\d)_B(\\d)_([ACTG]+)</code></li>
        <li><em>“RegEx to replace Plate, Batch, and Barcodes from headers”</em>:<code class="language-plaintext highlighter-rouge">P\\1_B\\2_\\3</code></li>
      </ul>
    </li>
  </ul>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-regular-expressions"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Regular Expressions</div>
  <p>The <a href="https://www.regular-expressions.info/quickstart.html">regular expression</a> (RegEx) used in the final steps of the above <em>Hands-On</em> is required in order to tell us how to capture the important information in the cell headers contained in brackets <code class="language-plaintext highlighter-rouge">(</code> <code class="language-plaintext highlighter-rouge">)</code>, where <code class="language-plaintext highlighter-rouge">\\d</code> denotes an expected digit, and <code class="language-plaintext highlighter-rouge">[ACTG]+</code> denotes 1 or more characters matching A or C or T or G.</p>

  <p>The information captured in the brackets <code class="language-plaintext highlighter-rouge">(</code> <code class="language-plaintext highlighter-rouge">)</code> can then be placed in the desired arrangement, where <code class="language-plaintext highlighter-rouge">Place \\1 Matches \\2 Here \\3</code> would place the first <code class="language-plaintext highlighter-rouge">\\d</code> after “Place “, the second after “Matches “, and so on.</p>

</blockquote>

<p>The plot that follows tells us everything we need to know about each of our batches. Each batch is essentially tested against the full set of barcodes in order to assert that only the desired or ‘Real’ barcodes have been sequenced.</p>

<h3 id="cross-contamination-plot">Cross-contamination Plot</h3>

<figure id="figure-9" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_crosscontamination.png" alt="Contamination Plots. " width="2348" height="1050" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_crosscontamination.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> The Pre-filter and Post-filter plots</figcaption></figure>

<p>Two things to take note of:</p>

<ol>
  <li>In the pre-filter plot, we can see how only half of the sequences in each batch map to half the barcodes. This shows very little cross-contamination, and proves that our data is real.</li>
  <li>The post-filter plot essentially removes the false barcodes from each batch and retains only the ‘Real’ barcodes.</li>
</ol>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>The count matrix that is output from this tool has only half the number of cells as the original input count matrix. Why is this?</li>
    <li>Which batches yield worrying levels of cross-contamination?</li>
    <li>Which batches should we remove from all further analysis?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Because only half the barcodes in each batch were real. The <em>UMI-tools extract</em> took the entire barcodes file to filter against each batch, and the <em>UMI-tools count</em> also took the entire barcodes file to count against each batch. Naturally, each batch produced 192 cells, even though 96 were real. As a result of joining each of these matrices we ended up with a count-matrix of \(8 * 192 = 1536\) cells. The cross-contamination tool removes the false barcodes (50% in each batch), resulting in \(768\) cells.</li>
      <li>Batch 4 and Batch 6 both appear to have a significant number of mid-to-high range counts in cells under the <em>False Positives</em> section in these batches. This means that the cell barcodes that we should <em>not</em> be detecting in those batches, did in fact detect cells.</li>
      <li>Batch 4 does still have a good number of <em>True Positives</em>, despite the many mid-level <em>False Positives</em> so we could still perhaps use the cells from that Batch. Batch 6 on the other hand appears to derive most of its counts from the <em>False Positives</em>, and therefore is likely not so suitable for further analysis.</li>
    </ol>
  </blockquote>
</blockquote>

<p>All <em>False Positives</em> from all batches are filtered out, leaving only the <em>True Positives</em> (or ‘Real’ barcodes) in the remaining filtered matrix. It is up to the user to filter further based on the contamination information that they derive from this step. With this, a QC filtered count-matrix is produced that can then be used for further downstream analysis.</p>

<!--
Factoid: We can convert the number of UMIs to the number of molecules using a transformation script.
-->

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial we have learned the importance of barcoding; namely how to define, extract, and annotate them from our reads and into the read headers, in order to preserve them during mapping. We have discovered how these barcoded reads are transformed into counts, where the cell barcode and UMI barcode are used to denote individual cells and to correct against reads that are PCR duplicates. Finally, we have learned how to combine separate batch data as well as being able to check and correct against cross-contamination.</p>

<figure id="figure-10" style="max-width: 90%;"><img src="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow.svg" alt="Recap of workflow. " width="933" height="978" loading="lazy" /><a target="_blank" href="/training-material/topics/single-cell/images/scrna-pre-processing/scrna_workflow.svg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> A recap of the entire workflow</figcaption></figure>

<p>This tutorial is part of the https://singlecell.usegalaxy.eu portal (<span class="citation"><a href="#tekman2020single">Tekman <i>et al.</i> 2020</a></span>).</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Verifying the distribution of barcodes via a FASTQC plot</p>
</li>
                        
                        <li><p>Relocating barcodes into headers</p>
</li>
                        
                        <li><p>Merging matrices from different batches together</p>
</li>
                        
                        <li><p>Removing unwanted barcodes</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/single-cell/faqs/">FAQ page for the Single Cell topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/single-cell#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Hashimshony2016">Hashimshony, T., N. Senderovich, G. Avital, A. Klochendler, Y. de Leeuw <i>et al.</i>, 2016 <b>CEL-Seq2: sensitive highly-multiplexed single-cell RNA-Seq</b>. Genome Biology 17: <a href="https://doi.org/10.1186/s13059-016-0938-8">10.1186/s13059-016-0938-8</a></li>
<li id="Herman2018">Herman, J. S., Sagar, and D. Grün, 2018 <b>FateID infers cell fate bias in multipotent progenitors from single-cell RNA-seq data</b>. Nature Methods 15: 379–386. <a href="https://doi.org/10.1038/nmeth.4662">10.1038/nmeth.4662</a></li>
<li id="tekman2020single">Tekman, M., B. Batut, A. Ostrovsky, C. Antoniewski, D. Clements <i>et al.</i>, 2020 <b>A single-cell RNA-sequencing training and analysis suite using the Galaxy framework</b>. GigaScience 9: giaa102. <a href="https://doi.org/10.1093/gigascience/giaa102">10.1093/gigascience/giaa102</a> <a href="https://academic.oup.com/gigascience/article/9/10/giaa102/5931798">https://academic.oup.com/gigascience/article/9/10/giaa102/5931798</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scrna-preprocessing"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=single-cell/scrna-preprocessing">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Mehmet Tekman, Bérénice Batut, Anika Erxleben, Wolfgang Maier,  <b>Pre-processing of Single-Cell RNA Data (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html">https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{single-cell-scrna-preprocessing,
author = "Mehmet Tekman and Bérénice Batut and Anika Erxleben and Wolfgang Maier",
	title = "Pre-processing of Single-Cell RNA Data (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-preprocessing/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                <blockquote class="agenda follow-up">
                    <div class="box-title">Go Further</div>
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/single-cell/tutorials/scrna-umis/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Understanding Barcodes</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


                    </ul>
                </blockquote>
                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/single-cell/tutorials/scrna-preprocessing/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bamtools_filter
  owner: devteam
  revisions: cb20f99fd45b
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_stats
  owner: devteam
  revisions: 24c5d43cb545
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collection_column_join
  owner: iuc
  revisions: 3ddd99c7efee
  tool_panel_section_label: Join, Subtract and Group
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: featurecounts
  owner: iuc
  revisions: f3a5f075498f
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: featurecounts
  owner: iuc
  revisions: ea04b737afa0
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: df99138d2776
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: c1a4b5f3b432
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: rgrnastar
  owner: iuc
  revisions: 2055c2667eb3
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: rgrnastar
  owner: iuc
  revisions: 7ed2edc1337f
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: umi_tools_count
  owner: iuc
  revisions: b557acca0b56
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: umi_tools_count
  owner: iuc
  revisions: 8250ea3a1501
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: umi_tools_extract
  owner: iuc
  revisions: 828dba98cdb4
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: umi_tools_extract
  owner: iuc
  revisions: 6a675c3aa610
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">6</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.5">3</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">0 stars</span><i class="far fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.16666666666666666">1</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>May 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: the work flow summary in a diagram format
					<b>Disliked</b>: You could explain the steps in more detail
				</li>
				
		    </ul>
		    
		    <b>February 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: It is very didactical
					<b>Disliked</b>: a better explanation of cross-contamination plots
				</li>
				
		    </ul>
		    
		    <b>August 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The variety of tools used. 
					<b>Disliked</b>: A clearer explanation about barcodes and batches. 
				</li>
				
		    </ul>
		    
		    <b>March 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">0 stars</span><i class="far fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: All steps were clearly described and easily understood. 
					<b>Disliked</b>: I could not find the tool"filter bam dataset on a variety of attributes" in tool panel of Galaxy-human cell atlas , so I could not follow.
				</li>
				
		    </ul>
		    
		    <b>July 2019</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Clear step-by-step instructions
					<b>Disliked</b>: Suggestions of what to do next once the count matrix has been constructed. Can one just use standard methods such as DESeq2 for differential expression? Some examples of PCA?
				</li>
				
		    </ul>
		    
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00251">gxy.io/GTN:T00251</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-preprocessing/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scrna-preprocessing/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/d7df185d039afa3d9d09a81c704876f007224bbd">d7df185</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	
	

    </body>
</html>