<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Slides: Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads / Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads / Assembly</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@5297fe084ae5d03ac62a0ecb0faa55aed5766f4d",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-04-11 12:54:58 +0000">
<meta name="DC.creator" content="CristÃ³bal Gallardo"><meta name="description" content=".enlarge120[  # **SARS-CoV-2**  ]   The severe acute respiratory syndrome coronavirus, known as SARS-CoV-2, is a betacoronavirus which belongs to the subfamily Coronavidinae, family Coronavidae.  * Genome characteristics:    * Positive-sense single-stranded RNA (+ssRNA) virus of 30 kb.   * Encode 9860 aminoacids.   * Includes 14 functional ORFs.   * Codify 4 structural proteins and 23 non-structural proteins.  ---  # **SARS-CoV-2 genome structure**  ![Structure of the SARS CoV 2 genome, a 5' utr, a polyprotein pp1ab/pp1b, and several structural and accessory proteins before the 3' utr. The pp1ab polyprotein is shown exapnded into a series of non-structural proteins labelled nsp1 to nsp16.](../../images/SARS-CoV-2_Genome.png)  - NSP1-NSP16 encodes the replicase-transcriptase complex - It includes four structural proteins: Spike (S), Envelope (E), Membrane (M) and Nucleocapsid (N).  ---  # **SARS-CoV-2 structure**  ![A graphic of the virus as a sphere with spikes (S1 and S2) coming ou...">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Assembly / Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads / Slides: Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads">
        <meta property="og:description" content=".enlarge120[  # **SARS-CoV-2**  ]   The severe acute respiratory syndrome coronavirus, known as SARS-CoV-2, is a betacoronavirus which belongs to the subfamily Coronavidinae, family Coronavidae.  * Genome characteristics:    * Positive-sense single-stranded RNA (+ssRNA) virus of 30 kb.   * Encode 9860 aminoacids.   * Includes 14 functional ORFs.   * Codify 4 structural proteins and 23 non-structural proteins.  ---  # **SARS-CoV-2 genome structure**  ![Structure of the SARS CoV 2 genome, a 5' utr, a polyprotein pp1ab/pp1b, and several structural and accessory proteins before the 3' utr. The pp1ab polyprotein is shown exapnded into a series of non-structural proteins labelled nsp1 to nsp16.](../../images/SARS-CoV-2_Genome.png)  - NSP1-NSP16 encodes the replicase-transcriptase complex - It includes four structural proteins: Spike (S), Envelope (E), Membrane (M) and Nucleocapsid (N).  ---  # **SARS-CoV-2 structure**  ![A graphic of the virus as a sphere with spikes (S1 and S2) coming ou...">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/assembly/tutorials/assembly-with-preprocessing/slides.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/assembly-with-preprocessing/slides-plain.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/assembly/tutorials/assembly-with-preprocessing/slides.html">View markdown source on GitHub</a>
	<h1>Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		
  <a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge contributor-gallardoalba"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/gallardoalba?s=36" alt="CristÃ³bal Gallardo avatar" width="36" class="avatar" />
    CristÃ³bal Gallardo</a>
  
	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>How can a genome of interest be assembled against a background of contaminating reads from other genomes?</p>
</li>
		
		<li><p>How can sequencing data used to obtain an assembled genome?</p>
</li>
		
		</ul>
		

		
		<h2>Objectives</h2>
		<ul>
		
		<li><p>Know basic characteristics of SARS-CoV-2</p>
</li>
		
		<li><p>Understand Nanopore and Illumina technologies</p>
</li>
		
		<li><p>Detect and remove human reads from SARS-CoV-2 sequencing reads</p>
</li>
		
		<li><p>Know main <em>de novo</em> genome assembly algorithms</p>
</li>
		
		<li><p>Perform a hybrid <em>de novo</em> genome assembly</p>
</li>
		
		</ul>
		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Published:</strong> Jan 7, 2021 </div>
		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last Updated:</strong> Apr 11, 2023 </div>

	<hr />


	<p>.enlarge120[</p>

<h1 id="sars-cov-2"><strong>SARS-CoV-2</strong></h1>

<p>]</p>

<p>The severe acute respiratory syndrome coronavirus, known as SARS-CoV-2, is a betacoronavirus which belongs to the subfamily Coronavidinae, family Coronavidae.</p>

<ul>
  <li>
    <p>Genome characteristics:</p>

    <ul>
      <li>Positive-sense single-stranded RNA (+ssRNA) virus of 30 kb.</li>
      <li>Encode 9860 aminoacids.</li>
      <li>Includes 14 functional ORFs.</li>
      <li>Codify 4 structural proteins and 23 non-structural proteins.</li>
    </ul>
  </li>
</ul>

<hr />

<h1 id="sars-cov-2-genome-structure"><strong>SARS-CoV-2 genome structure</strong></h1>

<p><img src="../../images/SARS-CoV-2_Genome.png" alt="Structure of the SARS CoV 2 genome, a 5' utr, a polyprotein pp1ab/pp1b, and several structural and accessory proteins before the 3' utr. The pp1ab polyprotein is shown exapnded into a series of non-structural proteins labelled nsp1 to nsp16." /></p>

<ul>
  <li>NSP1-NSP16 encodes the replicase-transcriptase complex</li>
  <li>It includes four structural proteins: Spike (S), Envelope (E), Membrane (M) and Nucleocapsid (N).</li>
</ul>

<hr />

<h1 id="sars-cov-2-structure"><strong>SARS-CoV-2 structure</strong></h1>

<p><img src="../../images/structure_SARS-CoV_2.png" alt="A graphic of the virus as a sphere with spikes (S1 and S2) coming out from the membrane, envelope proteins embedded within the membrane, and then a nucelocapsid inside. The 3d protein structures are shown in two styles of the spike proteins with the receptor binding domain highlighted. " /></p>

<hr />

<p>.enlarge200[</p>

<h1 id="hybrid-assembly"><strong>Hybrid assembly</strong></h1>

<p>]</p>

<p>.reduce70[
Hybrid assembly consists in using a combination of long and short reads to produce genome sequence.</p>

<p>Long reads are used to resolve ambiguities that exist in genomes previously assembled using the short reads. In addition, low rate-error short reads are used to correct errors that exist in the error-prone long reads.
]</p>

<p><img src="../../images/hybrid_assembly.png" alt="Cartoon of hybrid assembly. Step 1 shows short reads from Illumina and long reads from Nanopore. In step 2 these are assembled separately and there are ambiguities in sequence assembly. In step 3, hybrid assembly shows the assembly done with both sets of data and it helps resolve ambiguities with higher coverage." /></p>

<hr />

<p>.enlarge200[</p>

<h1 id="data-sources"><strong>Data sources</strong></h1>

<p>]</p>

<p>.pull-left[</p>

<p><strong>Illumina reads</strong></p>

<ul>
  <li>Second generation sequencing</li>
  <li>Short size: 200 bp</li>
  <li>Low error rate (~1%)</li>
</ul>

<p><img src="../../images/illumina_sequencing.png" alt="A picture of an illumina sequencer. It is a large white and grey box with a computer screen" /></p>

<p>]</p>

<p>.pull-right[</p>

<p><strong>Oxford Nanopore reads</strong></p>

<ul>
  <li>Third generation sequencing</li>
  <li>Long size:  &gt;40,000 bp</li>
  <li>High error rate (~10%)</li>
</ul>

<p><img src="../../images/nanopore_sequencing.png" alt="A picture of someone holding a nanopore device, approximately the size of an oversize usb stick with a chip visible." /></p>

<p>]</p>

<hr />

<p>.enlarge120[</p>

<h1 id="data-sources-illumina-sequencing"><strong>Data sources: Illumina sequencing</strong></h1>

<p>]</p>

<p><img src="../../images/illumina_sequencing_cluster.png" alt="Cartoon of illumina sequence, DNA is fragmented and adapters added. This binds to nanowells with oligonucleotides. Then the DNA bends and attaches to another binder. A primer attaches to the adapter and polymerase adds flourescently tagged dNTPs. Imaging happens while these are added. Then it is split, the DNA strand is denaturalised and now it there are two strands, bound to different adapters in the well. This process is repeated many times." /></p>

<hr />

<p>.enlarge200[</p>

<h1 id="data-sources-nanopore-sequencing"><strong>Data sources: Nanopore sequencing</strong></h1>

<p>]</p>

<p><img src="../../images/nanopore_sequencing_process.png" alt="Single nucleic acid molecules pass through a nanopore and changes in electrical field are measured. The magnitude of the current density depends on which nucleotide occupies the nanopore. This produces a graph which is then rad into individual bases." /></p>

<hr />

<p>.enlarge200[</p>

<h1 id="quality-control"><strong>Quality control</strong></h1>

<p>]</p>

<p>Quality control, read trimming and filtering are essential preprocessing steps required to garantee accurate results from RNA-seq datasets. Due to their very different nature, Illumina and Nanopore reads should be processed by using different tools.</p>

<p><img src="../../images/rna-seq_processing.png" alt="Schematic of a workflow, RNA-seq dataset is input which consists of illumina and nanopore reads. Those go through quality assessment with fastp and nanoplot, trimming, and filtering, before producing processed reads." /></p>

<p>.reduce70[</p>

<p>]</p>

<hr />

<p>.enlarge200[</p>

<h1 id="subtraction-of-reads-mapping-to-the-human-reference-genome"><strong>Subtraction of reads mapping to the human reference genome</strong></h1>

<p>]</p>

<p>Since the SARS-CoV-2 samples were obtained from human tissues, it is necessary to retain only the reads that donât map to the human genome, i.e those of potential viral origin.</p>

<hr />

<p>.enlarge200[</p>

<h1 id="subtraction-of-reads-mapping-to-the-human-reference-genome-1"><strong>Subtraction of reads mapping to the human reference genome</strong></h1>

<p>]</p>

<p>.image-75[<img src="../../images/mapping_human_genome.png" alt="A set of reads and a human genome are put together, mapping done with bowtie2 or minimap2 to identify reads which map to the human genome. (Then these are removed.)" />]</p>

<p>.reduce70[</p>

<p>As with quality control, differential characteristics of Illumina and Nanopore reads require different tools for mapping the reads to the human genome:</p>

<ul>
  <li>
    <p><strong>Bowtie2</strong>:  It is optimized for the read lengths and error modes yielded by typical Illumina sequencers.</p>
  </li>
  <li>
    <p><strong>Minimap2</strong>: It is particularly efficient for mapping Nanopore long reads.</p>
  </li>
</ul>

<p>]</p>

<hr />

<p>.enlarge200[</p>

<h1 id="genome-assembly"><strong>Genome assembly</strong></h1>

<p>]</p>

<p>Now everything is ready to perform genome assembly!</p>

<p><img src="../../images/jigsaw_genome.png" alt="A picture of a jigsaw puzzle with a DNA image, and several missing pieces scattered around." /></p>

<p>Genome assembly is a complex computational process whose objetive is to reconstruct a genome from the reads obtained by sequencing technologies.</p>

<hr />

<p>.enlarge200[</p>

<h1 id="de-novo-genome-assembly"><strong><em>De novo</em> genome assembly</strong></h1>

<p>]</p>

<p><em>De novo</em> assembly is a method for constructing genomes from a large number of  DNA fragments, with no a priori knowledge of the correct sequence or order of those fragments.</p>

<p>Two common types of <em>de novo</em> assemblers are greedy algorithm assemblers and graph method assemblers.</p>

<hr />

<p>.enlarge200[</p>

<h1 id="de-novo-genome-assembly-algorithms"><strong><em>De novo</em> genome assembly algorithms</strong></h1>

<p>]</p>

<p>.pull-left[</p>

<p><strong>Greedy algorithm assemblers</strong></p>

<p>It finds overlaps between reads, then builds a consensus sequence from the aligned overlapping reads.</p>

<p><br />
.reduce70[</p>

<ul>
  <li>Relative efficiency</li>
  <li>Do not work well for large read sets because only takes into account local information</li>
  <li>Do not perform well with repeat regions</li>
</ul>

<p>]</p>

<p>.image-75[<img src="../../images/greedy_algoritm.png" alt="Rainbow coloured reads are aligned locally to make small high quality overlaps. These are then built up into a larger consensus with the entire rainbow." />]</p>

<p>]</p>

<p>.pull-right[</p>

<p><strong>Graph method assemblers</strong></p>

<p>Basically it represent reads as a set of nodes, and overlaps between these reads as directed edges which connect these nodes to form a complete graph.</p>

<p>.reduce70[</p>

<ul>
  <li>Computationally more expensive</li>
  <li>Aim for global optima</li>
  <li>Perform well on large read sets, specially when they contain repeat regions.
]</li>
</ul>

<p>.image-75[<img src="../../images/graph_example.png" alt="A graph with many nodes connected by lines in a large tangle." />]</p>

<p>]</p>

<hr />

<p>.enlarge200[</p>

<h1 id="graph-methods-assemblers-de-brujin-graphs"><strong>Graph methods assemblers: de Brujin graphs</strong></h1>
<p>]</p>

<p>.reduce70[
De Bruijn graphs is the graph model used by most genome assemblers.</p>

<p>During the assembly process reads are broken into smaller fragments of a specified size, the k-mers, whichs are then used as nodes in the graph assembly. Nodes that overlap are then connected by an edge, which represents the reads. An ideal genome assembly corresponds to the path that visits every node exactly once.</p>

<p>]</p>

<p>.image-75[<img src="../../images/olc_pic.png" alt="Reads are provided to the algorithm, they are in the colours of the rainbow. Next overlaps are identified and the rainbow resolves itself. A subset of that is highlighted and points to reads connected by overlaps with many arrows going between the bluegreen fragments that are highlighted. This goes to the hamiltonian path identified with a re-run arrow between, indicating some mount of backtracking needed to find the best path. Finally the hamiltonian produces a consensus sequence with the correct final ordering." />]</p>

<hr />

<p>.enlarge200[</p>

<h1 id="graph-methods-assemblers-de-brujin-graphs-1"><strong>Graph methods assemblers: de Brujin graphs</strong></h1>

<p>]</p>

<p>The <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/debruijn-graph-assembly/slides.html#1">de Brujin graph assembly tutorial</a> provides a detailed explanation about this topic.</p>

<hr />

<p>.enlarge200[</p>

<h1 id="assembly-genome-with-unicycler"><strong>Assembly genome with Unicycler</strong></h1>

<p>]</p>

<p>.image-50[<img src="../../images/unicycler_logo.png" alt="unicycler logo" />]</p>

<p>Unicycler is a software tool designed specifically for hybrid assembly of small genomes.</p>

<hr />

<p>.enlarge200[</p>

<h1 id="assembly-genome-with-unicycler-1"><strong>Assembly genome with Unicycler</strong></h1>

<p>]</p>

<p>It employs a multi-step process that utilizes a set of software tools.</p>

<p><img src="../../images/unicycler_pipeline.png" alt="schematic of the unicycler pipeline, illumina short reads are assembled with spades into an assembly graph in one branch. Another branch brings nanopore reads through miniasm and racon to assemble and polish into long read contigs. Bridge application and contig merging combin the assembly graph and long reads, and this is sent to bowtie2 and pilon for polishing, producing the final assembly." /></p>


	<hr />


<h2>Key Points</h2>

<ul>

<li>Certain types of NGS samples can be heavily contaminated with sequences from other genomes.</li>

<li>Reads from known/expected contaminating sources can be identified by mapping to the respective genomes.</li>

<li>The different characteristics of Illumina and Nanopore sequencing technologies require processing by different tools.</li>

<li>Hybrid genome assembly allows to obtain high quality genome sequences.</li>

</ul>







<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


Tutorial Content is licensed under

  <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.

</section>






        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/assembly/tutorials/assembly-with-preprocessing/slides.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/./topics/assembly/tutorials/assembly-with-preprocessing/slides.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/5297fe084ae5d03ac62a0ecb0faa55aed5766f4d">5297fe0</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>
