<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads / Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads / Assembly</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@62278d71c6fca8d63a60269bfa0554a1883631ca",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-06-14 09:56:50 +0000">
<meta name="DC.creator" content="Wolfgang Maier"><meta name="description" content="DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Assembly / Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads / Hands-on: Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads">
        <meta property="og:description" content="DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-06-14 09:56:50 +0000",
  "datePublished": "2020-04-21 21:52:12 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00029",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
  },
  "abstract": "In some research or clinical contexts it is not possible, or very hard, to",
  "learningResourceType": "e-learning",
  "name": "Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads",
  "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html",
  "version": 13,
  "timeRequired": "PT4H",
  "teaches": "- Obtain viral (SARS-CoV-2) sequencing data with contaminating human reads from public sources\n- Organize the data into collections and check its quality\n- Detect and remove human reads\n- Assemble retained reads and explore the results",
  "keywords": [
    "Assembly",
    "covid19"
  ],
  "description": "## Abstract\n\nIn some research or clinical contexts it is not possible, or very hard, to\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - How can a genome of interest be assembled against a background of contaminating reads from other genomes?\n - How can sequencing data from public sources be turned into assembly-ready polished datasets?\n\n\n## Learning Objectives\n\n- Obtain viral (SARS-CoV-2) sequencing data with contaminating human reads from public sources\n- Organize the data into collections and check its quality\n- Detect and remove human reads\n- Assemble retained reads and explore the results\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/unicycler-assembly/slides.html",
      "name": "Unicycler Assembly",
      "description": "Slides for 'Unicycler Assembly' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html",
      "name": "Unicycler Assembly",
      "description": "Hands-on for 'Unicycler Assembly' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html",
      "name": "Using dataset collections",
      "description": "Hands-on for 'Using dataset collections' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "BY-COVID is an EC funded project that tackles the data challenges that can hinder effective pandemic response.\n\nThis project has received funding from the European Union’s Horizon Europe research and innovation programme under grant agreement № 101046203 (BY-COVID)\n",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0196",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0196",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0196"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/3732359",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/assembly" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Assembly
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/assembly/faqs/" title="Check our FAQs for the Assembly topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-assembly">
    <h1 data-toc-skip>Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Wolfgang Maier avatar" width="36" class="avatar" />
    Wolfgang Maier</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can a genome of interest be assembled against a background of contaminating reads from other genomes?</p>
</li>
        
        <li><p>How can sequencing data from public sources be turned into assembly-ready polished datasets?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Obtain viral (SARS-CoV-2) sequencing data with contaminating human reads from public sources</p>
</li>
        
        <li><p>Organize the data into collections and check its quality</p>
</li>
        
        <li><p>Detect and remove human reads</p>
</li>
        
        <li><p>Assemble retained reads and explore the results</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    

    
        
        
        
            
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Quality Control</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Quality Control</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        
    
        
        
        
            
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/assembly/tutorials/unicycler-assembly/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Unicycler Assembly</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/assembly/tutorials/unicycler-assembly/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Unicycler Assembly</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    

    
        
        
        
            
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mapping</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mapping</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Using dataset collections</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 4 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/3732359">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/workflows/" title="Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











<li class="btn btn-default supporting_material">


  <!-- dropdown with all recordings -->
  <a href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/recordings/" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this material in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
  </a>


  <ul class="dropdown-menu">
    

    
      
      
      
    <li><a class="dropdown-item" href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/recordings/index.html#tutorial-recording-15-february-2021" title="View the recording for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021) - 25m</a>
    </li>
      
    
    <li><a class="dropdown-item" href="/training-material/topics/assembly/tutorials/assembly-with-preprocessing/recordings/" title="View all recordings for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> View All</a>
    </li>

  </ul>

  
</li>
  





            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://mississippi.sorbonne-universite.fr" title="">
			MISSISSIPPI
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main)
		</a>
	</li>
    
    

    <!--
    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/assembly/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    -->
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Apr 21, 2020 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 14, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00029">https://gxy.io/GTN:T00029</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">5.0</a> (0 recent ratings, 7 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 13 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>In some research or clinical contexts it is not possible, or very hard, to
purify DNA/RNA for sequencing from just the specimen of interest.
Instead you will isolate DNA that is contaminated, sometimes heavily, with
DNA/RNA of a different origin.
This is the case for example with microbiome samples, which typically display
considerable contamination with host DNA, or with samples of body fluids for
pathogen detection. Such contamination can pose an issue with certain types of
analyses, in particular with genome assembly.</p>

<p>This tutorial guides you through the preprocessing of sequencing data of
bronchoalveolar lavage fluid (BALF) samples obtained from early COVID-19
patients in China. Since such samples are expected to be contaminated
signficantly with human sequenced reads, the goal is to enrich the data for
SARS-CoV-2 reads by identifying and discarding reads of human origin before
trying to assemble the viral genome sequence.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-the-usegalaxy-covid-19-analysis-project"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: The usegalaxy.* COVID-19 analysis project</div>
  <p>This tutorial uses the same data as, and recapitulates to a large extent, the
<a href="https://covid19.galaxyproject.org/genomics/1-PreProcessing/">Pre-processing</a>
and <a href="https://covid19.galaxyproject.org/genomics/2-Assembly/">Assembly</a> steps
of the <a href="https://covid19.galaxyproject.org/genomics/">Genomics</a> section of
<a href="https://covid19.galaxyproject.org/">covid19.galaxyproject.org</a>.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a>    <ol>
      <li><a href="#get-data-from-ncbi-sra" id="markdown-toc-get-data-from-ncbi-sra">Get data from NCBI SRA</a></li>
      <li><a href="#get-data-from-zenodo" id="markdown-toc-get-data-from-zenodo">Get data from Zenodo</a></li>
    </ol>
  </li>
  <li><a href="#read-trimming-and-quality-control" id="markdown-toc-read-trimming-and-quality-control">Read trimming and quality control</a>    <ol>
      <li><a href="#trimming-and-filtering-of-illumina-reads" id="markdown-toc-trimming-and-filtering-of-illumina-reads">Trimming and filtering of Illumina reads</a></li>
      <li><a href="#quality-control-of-nanopore-reads" id="markdown-toc-quality-control-of-nanopore-reads">Quality control of Nanopore reads</a></li>
    </ol>
  </li>
  <li><a href="#subtraction-of-reads-mapping-to-the-human-reference-genome" id="markdown-toc-subtraction-of-reads-mapping-to-the-human-reference-genome">Subtraction of reads mapping to the human reference genome</a>    <ol>
      <li><a href="#mapping-of-illumina-reads" id="markdown-toc-mapping-of-illumina-reads">Mapping of Illumina reads</a></li>
      <li><a href="#mapping-of-nanopore-reads" id="markdown-toc-mapping-of-nanopore-reads">Mapping of Nanopore reads</a></li>
      <li><a href="#human-reads-subtraction" id="markdown-toc-human-reads-subtraction">Human reads subtraction</a></li>
    </ol>
  </li>
  <li><a href="#format-conversion-of-remaining-reads" id="markdown-toc-format-conversion-of-remaining-reads">Format conversion of remaining reads</a>    <ol>
      <li><a href="#conversion-to-fastq-format" id="markdown-toc-conversion-to-fastq-format">Conversion to fastq format</a></li>
      <li><a href="#optional-rearrange-the-filtered-data-into-the-original-nested-data-structure" id="markdown-toc-optional-rearrange-the-filtered-data-into-the-original-nested-data-structure">Optional: Rearrange the filtered data into the original nested data structure</a></li>
      <li><a href="#merging-of-reads-with-collection-operations" id="markdown-toc-merging-of-reads-with-collection-operations">Merging of reads with collection operations</a></li>
    </ol>
  </li>
  <li><a href="#sars-cov-2-genome-assembly" id="markdown-toc-sars-cov-2-genome-assembly">SARS-CoV-2 genome assembly</a>    <ol>
      <li><a href="#optional-subsampling-of-reads" id="markdown-toc-optional-subsampling-of-reads">Optional: Subsampling of reads</a></li>
      <li><a href="#create-assembly" id="markdown-toc-create-assembly">Create assembly</a></li>
      <li><a href="#explore-assembly" id="markdown-toc-explore-assembly">Explore assembly</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="get-data">Get data</h1>

<p>We are going to create an assembly of the SARS-CoV-2 genome based on mixed
short-reads (Illumina) and long-reads (Nanopore) data from a total of six
different samples, all of which are publicly accessible through the NCBI
and EBI short-reads archives (SRAs).</p>

<p>Since automated data downloads from SRAs can be unreliable at times and could
get broken by changes to the download interface on the NCBI/EBI side, this
tutorial offers two ways to access the sequenced reads input data:</p>

<ol>
  <li>
    <p>Direct download from the NCBI SRA based on accession numbers and using the
dedicated <strong>Faster Download</strong> tool</p>

    <p>Use this method if it works and is fast enough for you, and if you are
interested in learning to obtain short-reads data directly from NCBI, in
general.</p>
  </li>
  <li>
    <p>Download of the same data deposited as a copy at <a href="https://zenodo.org/record/3732359">Zenodo</a>
This method uses Galaxy’s generic data import functionality, and should be
very reliable and faster than the download from NCBI.
It also showcases <strong>rule-based</strong> uploads and demonstrates how they can be
used to download several datasets and to arrange them into easy to handle
data structures at the same time.</p>

    <blockquote class="details">
      <div class="box-title details-title" id="details-rule-based-uploads"><button class="gtn-boxify-button details" type="button" aria-controls="details-rule-based-uploads" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Rule-based uploads</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>In this tutorial you will only use the features of Galaxy’s rule-based
uploader that are required to get the input data ready for our analysis,
and we will not explain those features in much detail.</p>

      <p>If, after this first taste, you are interested in a thorough introduction
we recommend the advanced tutorial
<a href="../../../galaxy-interface/tutorials/upload-rules/tutorial.html">Collections: Rule Based Uploader</a>.</p>
    </blockquote>

    <p>Use this method if the direct download from the NCBI SRA does not work, or
is too slow for your time frame, or if you are interested in advanced use
of Galaxy’s data import functionality.</p>
  </li>
</ol>

<p>The corresponding two step-by-step instructions below have been crafted to
produce identically arranged data structures in your history so all subsequent
steps are independent of the data source you choose.</p>

<h2 id="get-data-from-ncbi-sra">Get data from NCBI SRA</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-to-galaxy-from-ncbi-sra"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload to Galaxy from NCBI SRA</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Create a new dataset listing the SRA accession numbers of the Illumina paired-end input data for this tutorial:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR10903401
SRR10903402
SRR10971381
</code></pre></div>      </div>

      <p>call it, <em>e.g.</em>, <code class="language-plaintext highlighter-rouge">Illumina accessions</code> and set its datatype to <code class="language-plaintext highlighter-rouge">tabular</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-file"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-file" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>   <li>Paste the file contents into the text field</li>   <li>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code>* Press <strong>Start</strong> and <strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Create another new dataset listing the SRA accession numbers of the Nanopore input data for this tutorial:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR10948550
SRR10948474
SRR10902284
</code></pre></div>      </div>

      <p>call it, <em>e.g.</em>, <code class="language-plaintext highlighter-rouge">Nanopore accessions</code> and set its datatype to <code class="language-plaintext highlighter-rouge">tabular</code>.</p>
    </li>
    <li>
      <p>Add <code class="language-plaintext highlighter-rouge">#illumina</code>/<code class="language-plaintext highlighter-rouge">#nanopore</code> tags to the datasets</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-name-tags-in-the-analysis"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Name tags in the analysis</div>
        <p>We are going to treat the Illumina- and the Nanopore-sequenced data
separately in this tutorial up to the actual genome assembly step.</p>

        <p>To make it easier to keep track of the two branches of the analysis, we
recommend the use of Galaxy’s dataset <strong>name tags</strong>.
A name tag will automatically propagate to any new dataset derived
from the tagged dataset.</p>
      </blockquote>

      <p>You can create a name tag by attaching a tag starting with <code class="language-plaintext highlighter-rouge">#</code> to any
dataset.</p>

      <p>Name tags are meant to help you identify the origin of datasets quickly.
Feel free to either use the suggested names above or choose ones you like.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Retrieve the Illumina reads data from NCBI:</p>

      <p>Run <strong>Faster Download and Extract Reads in FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
the following parameter settings:</p>
      <ul>
        <li><em>“select input type”</em>: <code class="language-plaintext highlighter-rouge">List of SRA accession, one per line</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“sra accession list”</em>:
the <code class="language-plaintext highlighter-rouge">Illumina accessions</code> dataset created above</li>
          </ul>
        </li>
        <li>in <em>“Advanced Options”</em>
          <ul>
            <li><em>“Select how to split the spots”</em>: <code class="language-plaintext highlighter-rouge">--split-3</code></li>
          </ul>
        </li>
      </ul>

      <p>The tool run should generate four new items in your history - three
collections and one <em>log</em> dataset with a summary of what happened.</p>

      <p>Since all three datasets that we tried to retrieve contain only
paired-end reads, only the <code class="language-plaintext highlighter-rouge">Pair-end data</code> collection is expected to
contain downloaded data. Click on the other two collections to verify that
they are empty, then delete them from your history (since the collections
do not contain any datasets, it does not matter, which delete option -
“Collection Only”, “Delete Datasets” or “Permanently Delete Datasets” you
are choosing when prompted).</p>
    </li>
    <li>
      <p>Retrieve the Nanopore reads data from NCBI:</p>

      <p>Run <strong>Faster Download and Extract Reads in FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
the following parameter settings:</p>
      <ul>
        <li><em>“select input type”</em>: <code class="language-plaintext highlighter-rouge">List of SRA accession, one per line</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“sra accession list”</em>:
the <code class="language-plaintext highlighter-rouge">Nanopore accessions</code> dataset created above</li>
          </ul>
        </li>
        <li>in <em>“Advanced Options”</em>
          <ul>
            <li><em>“Select how to split the spots”</em>: <code class="language-plaintext highlighter-rouge">--split-3</code></li>
          </ul>
        </li>
      </ul>

      <p>As in the previous step, the tool run should generate four new items in
your history.</p>

      <p>Since all three datasets that we tried to retrieve in this run contain
only single-end reads, only the <code class="language-plaintext highlighter-rouge">Single-end data</code> collection is expected to
contain downloaded data this time. Click on the other two collections to
verify that they are empty, then delete them from your history.</p>
    </li>
  </ol>

</blockquote>

<h2 id="get-data-from-zenodo">Get data from Zenodo</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-to-galaxy-from-zenodo"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload to Galaxy from Zenodo</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import Illumina-sequenced reads data from <a href="https://zenodo.org/record/3732359">Zenodo</a></p>

      <p>The Zenodo links for the data are these:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3732359/files/SRR10903401_r1.fq.gz
https://zenodo.org/record/3732359/files/SRR10903401_r2.fq.gz
https://zenodo.org/record/3732359/files/SRR10903402_r1.fq.gz
https://zenodo.org/record/3732359/files/SRR10903402_r2.fq.gz
https://zenodo.org/record/3732359/files/SRR10971381_r1.fq.gz
https://zenodo.org/record/3732359/files/SRR10971381_r2.fq.gz
</code></pre></div>      </div>

      <p>To upload these data to your Galaxy history in structured form:</p>

      <ul>
        <li>Copy the above list of links</li>
        <li>Open the Galaxy Upload Manager
(<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top right of the tool panel)</li>
        <li>In the pop-up window, switch to the <strong>Rule-based</strong> tab and select
          <ul>
            <li><em>“Upload data as:”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
            <li><em>“Load tabular data from:”</em>: <code class="language-plaintext highlighter-rouge">Pasted Table</code></li>
            <li>Paste the copied links into the text field</li>
            <li>Click <strong>Build</strong></li>
          </ul>
        </li>
        <li>In the next screen, select
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Column”</em>: <code class="language-plaintext highlighter-rouge">Using a Regular expression</code>
              <ul>
                <li><em>“From Column”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Create columns matching expression groups.”</em></li>
                <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">.+/(SRR\d+)_r(\d).fq.gz</code></li>
                <li><em>“Number of Groups”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                <li>Click <strong>Apply</strong></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Rules”</em>: <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Add Definition”</em>: <code class="language-plaintext highlighter-rouge">URL</code>
                  <ul>
                    <li><em>“URL”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                  </ul>
                </li>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Add Definition”</em>: <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>
                  <ul>
                    <li><em>“List Identifier(s)”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
                  </ul>
                </li>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Add Definition”</em>: <code class="language-plaintext highlighter-rouge">Paired-end Indicator</code>
                  <ul>
                    <li><em>“Paired-end Indicator”</em>: <code class="language-plaintext highlighter-rouge">C</code></li>
                  </ul>
                </li>
                <li>Click <strong>Apply</strong></li>
              </ul>
            </li>
            <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
            <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">Illumina PE data</code> (or similar)</li>
            <li>
              <p><em>“Add nametag for name:”</em> <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span></p>

              <blockquote class="comment">
                <div class="box-title comment-title" id="comment-name-tags-in-the-analysis-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Name tags in the analysis</div>
                <p>We are going to treat the Illumina- and the Nanopore-sequenced data
separately in this tutorial up to the actual genome assembly step.</p>

                <p>To make it easier to keep track of the two branches of the analysis, we
recommend the use of Galaxy’s dataset <strong>name tags</strong>.
A name tag will automatically propagate to any new dataset derived
from the tagged dataset.</p>
              </blockquote>

              <p>Checking this option tells Galaxy to reuse the collection name above
as a name tag on the collection.</p>
            </li>
            <li>Click <strong>Upload</strong></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Import Nanopore-sequenced reads data from <a href="https://zenodo.org/record/3732359">Zenodo</a></p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3732359/files/SRR10902284_ONT.fq.gz
https://zenodo.org/record/3732359/files/SRR10948474_ONT.fq.gz
https://zenodo.org/record/3732359/files/SRR10948550_ONT.fq.gz
</code></pre></div>      </div>

      <p>Again, we want to upload this data to our history in structured form.
To do so:</p>

      <ul>
        <li>Copy the above list of links</li>
        <li>Open the Galaxy Upload Manager
(<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top right of the tool panel)</li>
        <li>In the pop-up window, switch to the <strong>Rule-based</strong> tab and select
          <ul>
            <li><em>“Upload data as:”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
            <li><em>“Load tabular data from:”</em>: <code class="language-plaintext highlighter-rouge">Pasted Table</code></li>
            <li>Paste the copied links into the text field</li>
            <li>Click <strong>Build</strong></li>
          </ul>
        </li>
        <li>In the next screen, select
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Column”</em>: <code class="language-plaintext highlighter-rouge">Using a Regular expression</code>
              <ul>
                <li><em>“From Column”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Create columns matching expression groups.”</em></li>
                <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">.+/(SRR\d+)_ONT.fq.gz</code></li>
                <li><em>“Number of Groups”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li>Click <strong>Apply</strong></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Rules”</em>: <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Add Definition”</em>: <code class="language-plaintext highlighter-rouge">URL</code>
                  <ul>
                    <li><em>“URL”</em>: <code class="language-plaintext highlighter-rouge">A</code></li>
                  </ul>
                </li>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Add Definition”</em>: <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>
                  <ul>
                    <li><em>“List Identifier(s)”</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
                  </ul>
                </li>
                <li>Click <strong>Apply</strong></li>
              </ul>
            </li>
            <li><em>“Type”</em>: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
            <li><em>“Name”</em>: <code class="language-plaintext highlighter-rouge">Nanopore data</code> (or similar)</li>
            <li><em>“Add nametag for name:”</em> <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span></li>
            <li>Click <strong>Upload</strong></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="read-trimming-and-quality-control">Read trimming and quality control</h1>

<p>In the end, we do not want to base our mapping on low-quality reads that may
cause misassembly of fragments or introduce sequencing artefacts into the final
assembled sequence. Since quality control, read filtering and read trimming are
all quite fast and computationally cheap operations compared to the read
mapping that we will use to identify and eliminate human reads, it is best to
perform these steps up front.</p>

<p>Due to their very different nature, however, Illumina- and Nanopore-sequenced
reads should be treated rather differently.</p>

<h2 id="trimming-and-filtering-of-illumina-reads">Trimming and filtering of Illumina reads</h2>

<p>Galaxy offers a panel of different NGS reads trimming/filtering tools. Here,
we use <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, which is straightforward to configure,
and when combined with <strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, enables nice and
easy-to-interpret visualizations of the effects of preprocessing, in particular
for multiple samples.</p>

<p>In the following, we configure the tool to retain reads only if at most 20% of
their bases have a Phred-scaled quality &gt;= 20 and if there length in bases after
trimming of adapter sequences (which the tool auto-detects for us) is at least
50.</p>

<p>The JSON-formatted report produced by the tool, can serve as input to
<strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for a direct visual comparison of key quality
metrics for all samples before and after preprocessing.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-reads-preprocessing-and-quality-reporting"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Reads preprocessing and quality reporting</div>
  <ol>
    <li><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Single-end or paired reads”</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code>
          <ul>
            <li><em>“Select paired collection(s)”</em>: the collection of Illumina-sequenced
reads as produced in the <em>Get Data</em> section</li>
          </ul>
        </li>
        <li>in <em>“Filter Options”</em>
          <ul>
            <li>in <em>“Quality filtering options”</em>
              <ul>
                <li><em>“Disable quality filtering”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>“Qualified quality phred”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
                <li><em>“Unqualified percent limit”</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
              </ul>
            </li>
            <li>in <em>“Length filtering options”</em>
              <ul>
                <li><em>“Disable length filtering”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>“Length required”</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>in <em>“Output Options”</em>
          <ul>
            <li><em>“Output HTML report”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Output JSON report”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>

      <p>The tool run produces two collections - one with the actual preprocessed
reads of all input samples, another one with a JSON-formatted report of
the processing for every sample.</p>
    </li>
    <li><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">fastp</code>
              <ul>
                <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Output of fastp”</em>: the collection of
JSON-formatted reports, second collection produced by <strong>fastp</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <p>This tool run generates a single webpage output with the combined quality reports
for all samples before and after processing with <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>
    </li>
  </ol>

</blockquote>

<h2 id="quality-control-of-nanopore-reads">Quality control of Nanopore reads</h2>

<p>Nanopore-sequenced reads differ greatly in length from one another and are, on
average, of relatively low quality (in particular when compared to
Illumina-sequenced reads). These properties make it challenging to preprocess
them with standard tools. Quality assessment tools calibrated to work well with
Illumina-sequenced reads are less useful for Nanopore-sequenced reads, too, for
the same reasons. Here we restrict ourselves to a simple quality check with
<strong>NanoPlot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, a dedicated QC tool for Nanopore-sequenced reads.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-checking-the-quality-of-nanopore-reads-with-nanoplot"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Checking the quality of Nanopore reads with NanoPlot</div>
  <ol>
    <li><strong>NanoPlot</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Select multifile mode”</em>: <code class="language-plaintext highlighter-rouge">batch</code>
          <ul>
            <li><em>“Type of the file(s) to work on”</em>: <code class="language-plaintext highlighter-rouge">fastq</code>
              <ul>
                <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“files”</em>: the collection of
Nanopore-sequenced reads as produced in the <em>Get Data</em> section</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>in <em>“Options for filtering or transforming input prior to plotting”</em>
          <ul>
            <li><em>“Logarithmic scaling of lengths in plots.”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>Looking at the three generated quality reports, which of the three
samples seems to be of better quality overall than the other two,
and what are some criteria that support this conclusion?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>
              <p>Sample <code class="language-plaintext highlighter-rouge">SRR10948474</code> has the best overall quality.</p>

              <p>It has both higher average read length and quality than the other
two (see the <em>Summary statistics</em> table), has a distribution of
read qualities that peaks around intermediate quality scores (not
low ones as for the other samples), and contains some extra-long
(though at least partly rather low-quality) reads. Those last two
points become most obvious when looking at the <em>Read lengths vs
Average read quality</em> plots.</p>
            </li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="subtraction-of-reads-mapping-to-the-human-reference-genome">Subtraction of reads mapping to the human reference genome</h1>

<h2 id="mapping-of-illumina-reads">Mapping of Illumina reads</h2>

<p>In this tutorial, we are using <strong>Bowtie2</strong> for mapping our short-reads data to
the human genome. <em>BWA-MEM</em> would be a good alternative for mapping the 150
nucleotides (see the QC report above) reads from our samples.</p>

<p>According to its authors, the <em>Minimap2</em> aligner, which we will be using for
mapping the Nanopore-sequenced data in the next step, is supposed to outcompete
<em>Bowtie2</em> and <em>BWA-MEM</em> in terms of speed even for Illumina-sequenced reads
of length &gt; 100 nts, but we opt for the conservative approach of using a
widely-used, well-tested tool here.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapping-with-bowtie2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapping with Bowtie2</div>
  <ol>
    <li><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Is this single or paired library”</em>: <code class="language-plaintext highlighter-rouge">Paired-end Dataset Collection</code>
          <ul>
            <li><em>“FASTQ Paired Dataset”</em>: the collection of preprocessed
Illumina-sequenced reads, output of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li>
              <p><em>“Write unaligned reads (in fastq format) to separate file(s)”</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>Activating this option may seem attractive since the unaligned reads
are what we are interested in, but filtering for those reads with
a dedicated tool in a separate step allows us to filter on the
properties of the read pairs instead of those of individual reads.</p>
            </li>
            <li><em>“Write aligned reads (in fastq format) to separate file(s)”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>“Do you want to set paired-end options?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>:
<code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Select reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg38 Full</code></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">Default setting only</code></li>
        <li><em>“Do you want to tweak SAM/BAM Options?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Save the bowtie2 mapping statistics to the history”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>The tool run should produce two collections of output datasets. One with
the actual mapped reads of the three samples and one with the corresponding
mapping statistics for each sample, which we want to have a brief look at
next.</p>
    </li>
    <li>
      <p>Inspect the <code class="language-plaintext highlighter-rouge">mapping stats</code> of each sample by clicking on the corresponding collection, then on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon of each individual sample data</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What percentage of reads of each sample has been aligned to the <code class="language-plaintext highlighter-rouge">hg38</code> reference genome?</li>
          <li>Which sample is the least contaminated with human reads?</li>
          <li>Which sample contains the highest amount of SARS-CoV2 reads?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>The samples have between 13% and 21% of reads aligned to <code class="language-plaintext highlighter-rouge">hg38</code>.
The information can be found on the last line of output for each
sample.</li>
            <li>Sample <code class="language-plaintext highlighter-rouge">SRR10971381</code> is the least contaminated with just above 13%
of human reads</li>
            <li>You cannot answer this question with this data. While <code class="language-plaintext highlighter-rouge">SRR10971381</code>
shows the lowest relative contamination with human reads, that
does not mean that all other reads are from SARS-CoV-2. They could
come from other species (<em>e.g.</em>, bacteria or other viruses)
contained in this BALF sample.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="mapping-of-nanopore-reads">Mapping of Nanopore reads</h2>

<p>For the mapping of the Nanopore-sequenced data we are using the <strong>Minimap2</strong>
aligner, which is particularly efficient for mapping long reads.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-nanopore-reads-mapping"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Nanopore reads mapping</div>

  <ol>
    <li><strong>Map with minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>:
<code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human Dec. 2013 (GRCh38/hg38) (hg38)</code></li>
          </ul>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select fastq dataset”</em>: the collection
of Nanopore-sequenced reads as set up in the <em>Get Data</em> section</li>
          </ul>
        </li>
        <li><em>“Select a profile of preset options”</em>: <code class="language-plaintext highlighter-rouge">Oxford Nanopore read to reference mapping. ...</code></li>
      </ul>

      <p>This tool run produces one collection with the actual mapped reads for
each Nanopore-sequenced sample. Unlike <strong>Bowtie2</strong> it does not have an
option to output mapping statistics directly. However, we can generate
that information through an extra step.</p>
    </li>
    <li><strong>Samtools stats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM file”</em>: the collection of mapped
Nanopore-sequenced reads, output of **Map with minimap2 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (step 1)</li>
        <li><em>“Output”</em>: <code class="language-plaintext highlighter-rouge">Separate datasets for each statistic</code>
          <ul>
            <li><em>“Desired output files”</em>
              <ul>
                <li>
                  <p><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Summary numbers”</em></p>

                  <p>These simple summary stats correspond approximately to the
statistics generated by <strong>Bowtie2</strong> and are enough for our purpose.</p>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-mapping-stats-for-nanopore-sequenced-long-reads"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Mapping stats for Nanopore-sequenced long reads</div>

        <p>Since, unlike Illumina-generated reads, Nanopore-sequenced reads can
have very different lengths, it makes limited sense to calculate a
ratio of mapped to overall reads for them.</p>

        <p>Instead, <code class="language-plaintext highlighter-rouge">bases mapped</code> / <code class="language-plaintext highlighter-rouge">total length</code> should give a more reliable
estimate of which fraction of the data is due to human genome sequence.</p>

        <p>Try to calculate this ratio for the three samples on your own!</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="human-reads-subtraction">Human reads subtraction</h2>

<p>At this point you should have two collections of mapped reads - one holding the
mapping results obtained with <strong>Bowtie2</strong> of the three Illumina-sequenced
samples, the other one holding the <strong>minimap2</strong> output for the three
Nanopore-sequenced samples.</p>

<p>Next, we are going to filter the data from both collections to retain only
those reads that were <em>not</em> mapped to the human genome, <em>i.e</em> those of
potential viral origin.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapped-reads-filtering"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapped reads filtering</div>

  <ol>
    <li><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to filter the Illumina-sequenced reads mapped with Bowtie2:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“SAM/BAM/CRAM data set”</em>: the collection of
mapped Illumina-sequenced reads, output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code>
          <ul>
            <li>in <em>“Configure filters”</em>
              <ul>
                <li><em>“Require that these flags are set”</em>: <code class="language-plaintext highlighter-rouge">Read is unmapped</code> and
<code class="language-plaintext highlighter-rouge">Mate is unmapped</code></li>
              </ul>
            </li>
            <li><em>“What would you like to have reported?”</em>: <code class="language-plaintext highlighter-rouge">All reads retained after filtering and subsampling</code>
              <ul>
                <li><em>“Produce extra dataset with dropped reads?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">BAM (-b)</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to filter the Nanopore-sequenced reads mapped with minimap2:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“SAM/BAM/CRAM data set”</em>: the collection of
mapped Nanopore-sequenced reads, output of <strong>minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code>
          <ul>
            <li>In <em>“Configure filters”</em>
              <ul>
                <li><em>“Require that these flags are set”</em>: <code class="language-plaintext highlighter-rouge">Read is unmapped</code></li>
              </ul>
            </li>
            <li><em>“What would you like to have reported?”</em>: <code class="language-plaintext highlighter-rouge">All reads retained after filtering and subsampling</code>
              <ul>
                <li><em>“Produce extra dataset with dropped reads?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">BAM (-b)</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>(Optional) Remove the database <code class="language-plaintext highlighter-rouge">hg38</code> attribute from the output files</p>

      <blockquote class="details">
        <div class="box-title details-title" id="details-why-do-this"><button class="gtn-boxify-button details" type="button" aria-controls="details-why-do-this" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Why do this?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>When we ran the <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>minimap2</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> mappers before, these tools set the <em>database</em> attribute
on their outputs to <code class="language-plaintext highlighter-rouge">hg38</code> to indicate that the mapped reads in these
outputs have been mapped against this version of the human reference
genome - an important piece of information for further work with those
mapped reads.</p>

        <p>In step 1 and 2 above, however, we have eliminated all mapped reads so
the <code class="language-plaintext highlighter-rouge">database: hg38</code> info is misleading from this point onward in the
analysis. While not directly harmful, it is best practice to remove this
metadata now.</p>
      </blockquote>

      <p>For the outputs of step 1 and step 2 above, reset the database/build
(dbkey) to <code class="language-plaintext highlighter-rouge">unspecified (?)</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click the desired dataset’s name to expand it.</li>   <li>     <p>Click on the “?” next to database indicator:</p>     <p><img src="/training-material/shared/images/datasets_dbkey.svg" alt="UI for changing dbkey" /></p>   </li>   <li>In the central panel, change the <strong>Database/Build</strong> field</li>   <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">unspecified (?)</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="format-conversion-of-remaining-reads">Format conversion of remaining reads</h1>

<h2 id="conversion-to-fastq-format">Conversion to fastq format</h2>

<p>Assembly tools, typically, expect their input data to be fastq-formatted, but
what we have after mapping and filtering is data in BAM format. Hence, we need
to convert the retained Illumina- and Nanopore-sequenced reads back into their
original format before proceeding to assembly.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-bam-to-fastq-format-conversion"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: BAM to fastq format conversion</div>

  <ol>
    <li><strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to convert the filtered Illumina-sequenced reads to fastq format
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM or SAM file to convert”</em>: the collection of
filtered Illumina-sequenced reads, output of first <strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">compressed FASTQ</code></li>
        <li><em>“outputs”</em>: <code class="language-plaintext highlighter-rouge">READ1</code> and <code class="language-plaintext highlighter-rouge">READ2</code></li>
      </ul>
    </li>
    <li><strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to convert the filtered Nanopore-sequenced reads to fastq format
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“BAM or SAM file to convert”</em>: the collection of
filtered Nanopore-sequenced reads, output of second <strong>Samtools view</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">compressed FASTQ</code></li>
        <li><em>“outputs”</em>: <code class="language-plaintext highlighter-rouge">unspecific</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="optional-rearrange-the-filtered-data-into-the-original-nested-data-structure">Optional: Rearrange the filtered data into the original nested data structure</h2>

<p>If you compare the outputs of the last step to the input data we started out
with, you will notice that the Illumina-sequenced data is arranged differently
now than initially. It is arranged now into separate collections of forward and
reverse reads, whereas we started with a single nested collection of the data.</p>

<p>We can easily cast the data back into its original structure with one of
Galaxy’s collection manipulation tools, but note that we will not use the
resulting nested collection for this tutorial because the <strong>Unicycler</strong> tool
for assembling the reads would not be able to handle the nested data correctly.</p>

<p>Thus, the following just serves as an illustration and is entirely optional.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-arrange-two-list-collections-into-a-list-of-pairs"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Arrange two list collections into a list of pairs</div>

  <ol>
    <li><strong>Zip Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Dataset (Forward)”</em>: the collection
of filtered Illumina-sequenced forward reads in fastq format,
first output of first <strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Dataset (Reverse)”</em>: the collection
of filtered Illumina-sequenced reverse reads in fastq format,
second output of first <strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="merging-of-reads-with-collection-operations">Merging of reads with collection operations</h2>

<p>To merge reads from several samples into a combined final assembly, we need to
pass the data to <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in partially merged form. The
forward and reverse reads of paired-end data should be kept separate, and so
should short and long reads. However, the tool has no option to combine data
from individual samples, so we need to merge the forward, reverse, and the long
reads data, respectively, across samples. Conveniently for us, the outputs of
the earlier <strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> runs have already returned the
data structured into three corresponding collections for us.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-collapsing-each-collection-into-a-single-dataset"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Collapsing each collection into a single dataset</div>

  <ol>
    <li><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of Illumina-sequenced <em>forward</em> reads
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>:
the collection of filtered Illumina-sequenced forward reads in fastq format,
first output of first <strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><em>“Keep one header line”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Prepend File name”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of Illumina-sequenced <em>reverse</em> reads
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>:
the collection of filtered Illumina-sequenced reverse reads in fastq format,
second output of first <strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><em>“Keep one header line”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Prepend File name”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of Nanopore-sequenced reads
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>:
the collection of filtered Nanopore-sequenced reads in fastq format,
output of second <strong>Samtools fastx</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><em>“Keep one header line”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Prepend File name”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="sars-cov-2-genome-assembly">SARS-CoV-2 genome assembly</h1>

<h2 id="optional-subsampling-of-reads">Optional: Subsampling of reads</h2>

<p>The actual assembly of the sequenced reads represents the real bottleneck in
this tutorial. Assembly of the full set of sequences can easily take 10 hours
and would best be conducted overnight.</p>

<p>If you do not have that much time, you should downsample the Illumina-sequenced
combined reads now. Which will reduce the time required to finish the
subsequent assembly step to approximately 1-2 hours.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-if-you-are-in-a-hurry"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: If you are in a hurry</div>
  <p>The downsampling parameters below have been chosen to have minimal impact on
the assembly results. Further speed-ups are certainly possible, but will
likely lead to poor assembly outcomes.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-subsampling-of-paired-end-short-reads-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Subsampling of paired-end short-reads data</div>

  <ol>
    <li><strong>seqtk_sample</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input FASTA/Q file”</em>: The two datasets with the
combined Illumina-sequenced forward and reverse reads, outputs of the
first and the second run of <strong>Collapse Collection</strong></li>
        <li><em>“RNG seed”</em>: 4</li>
        <li><em>“Subsample (decimal fraction or number)”</em>: 0.1</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="create-assembly">Create assembly</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-assembly-of-sars-cov2-genome"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assembly of SARS-CoV2 genome</div>

  <ol>
    <li><strong>Create assemblies with Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Paired or Single end data?”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the combined
Illumina-sequenced forward reads from all samples, output of the
first <strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run (or first output of
<strong>seqtk_sample</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for a subsample-based assembly)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the combined
Illumina-sequenced reverse reads from all samples, output of the
second <strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run  (or second output
of <strong>seqtk_sample</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for a subsample-based assembly)</li>
          </ul>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select long reads. If there are no long reads, leave this empty”</em>:
Nanpore-sequenced reads from all samples, output of the third
<strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><em>“Select Bridging mode”</em>: <code class="language-plaintext highlighter-rouge">Normal (moderate contig size and misassembly rate)</code></li>
        <li><em>“Exclude contigs from the FASTA file which are shorter than this length (bp)”</em>:
<code class="language-plaintext highlighter-rouge">100</code></li>
        <li><em>“The expected number of linear (i.e. non-circular) sequences in the assembly”</em>:
<code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="explore-assembly">Explore assembly</h2>

<p>The <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run above should produce two output datasets:</p>

<ul>
  <li>a final assembly in FASTA format</li>
  <li>an assembly graph</li>
</ul>

<p>Of these, the assembly graph is more information-rich because it not only
contains the sequences of <em>all</em> assembled fragments (including the ones shorter
than the threshold length defined for inclusion of the fragments into the FASTA
output), but also indicates the relative average coverage of the fragments by
sequenced reads and how some of the fragments could potentially be bridged
after resolving ambiguities manually.</p>

<h3 id="assembly-inspection-with-bandage">Assembly inspection with Bandage</h3>

<p>On the downside, the assembly graph format takes some getting used to before
you can make sense out of the information it provides.</p>

<p>This issue can be alleviated through the use of <strong>Bandage</strong>, a package for
exploring assembly graphs through summary reports and visualizations of their
contents.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-assembly-stats-and-visualization-with-bandage"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assembly stats and visualization with Bandage</div>

  <ol>
    <li><strong>Bandage Info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Graphical Fragment Assembly”</em>: the assembly graph dataset produced by
<strong>Unicycler</strong></li>
        <li><em>“Output the information in a single tab-delimited line starting with the graph file”</em>:
<code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li><strong>Bandage Image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Graphical Fragment Assembly”</em>: the assembly graph dataset produced by
<strong>Unicycler</strong></li>
        <li><em>“Node name labels?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Node length labels?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Let us inspect the summary report produced by <strong>Bandage Info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
first:</p>

<p>You may be rather disappointed by the large <em>percentage of of dead ends</em> in the
assembly graph (in general, lower is better here), and by the correspondingly
large <em>node count</em>. After all, should the viral sequence not be encoded on a
single small contig (a quick check at
<a href="https://en.wikipedia.org/wiki/Coronavirus#Genome">Wikipedia</a> reveals that
coronaviruses have genomes in the size range of 30kb)?</p>

<p>On the other hand, there is the <em>Longest node</em> of 29768 bp of assembled
sequence, which is suspiciously close to the expected genome size, but a much
larger <em>Estimated sequence length</em>.</p>

<p>Next, take a look at the assembly graph visualization generated by <strong>Bandage
Image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to see if that tells us more:</p>

<p>Indeed, this output shows that <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> managed to
assemble a good number of contigs of moderate size, then had trouble with a
number of really small fragments that it could only assemble with lots of
ambiguities (leading to that ugly clutter of nodes in the top row of the
image). Those small fragments will probably be hard to make sense of, but the
manageable list of moderate-size contigs (nodes 1-23, 25, 26) is encouraging.</p>

<p>Of these, node 1 is the longest node mentioned in the report with a size close
to our expectations.</p>

<p>Node 2 looks peculiar since <strong>Unicycler</strong> claims it is circular, while
Coronavirus genomes are known to be linear.</p>

<h3 id="check-origin-of-assembled-sequences-with-blast">Check origin of assembled sequences with BLAST</h3>

<p>While we could view the actual contents of the assembly graph output of
<strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and extract node sequences of interest from it
(the longest node and that circular one could be a start), things are much
easier if we work with the FASTA output of <strong>Unicycler</strong> instead.</p>

<p>From the visualization with <strong>Bandage Image</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> we know that the
separately assembled nodes are all longer than 1000 bp. We can extract those
sequences based on the length threshold in Galaxy, then BLAST all retained
sequences in one go.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-fasta-sequences-by-their-length"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter FASTA sequences by their length</div>

  <ol>
    <li><strong>Filter sequences by length</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Fasta file”</em>: the FASTA output produced by
<strong>Unicycler</strong></li>
        <li><em>“Minimal length”</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
      </ul>

      <p>This outputs a new FASTA datasets with only the sequences satisfying our
length threshold.</p>
    </li>
  </ol>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-apply-length-filters-after-instead-of-during-assembly"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-apply-length-filters-after-instead-of-during-assembly" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Apply length filters after instead of during assembly</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>You may have noted that in the <strong>Unicycler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run we kept
the tool’s <em>“Exclude contigs from the FASTA file which are shorter than
this length (bp)”</em> option at its default value of <code class="language-plaintext highlighter-rouge">100</code> instead of using
the 1,000 bp threshold there directly to save a step in the analysis.</p>

    <p>The reason we did this is that normally you will not know the exact length
threshold you want until <em>after</em> having explored the generated assembly.</p>

    <p>Length-filtering some FASTA sequences is a trivial process that takes very
little time, but you would not want to rerun an hours-long assembly job
just because you accidentally stripped some interesting assembled sequences
from the output.</p>
  </blockquote>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-ncbi-blast-of-multiple-contigs"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: NCBI BLAST of multiple contigs</div>

  <ol>
    <li>View the output of <strong>Filter sequences by length</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by
clicking the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon attached to that dataset.</li>
    <li>Click into the middle panel, which should now display the content of the
dataset, select all sequences by pressing <kbd>Ctrl</kbd>+<kbd>A</kbd>,
then copy the selection to the clipboard with <kbd>Ctrl</kbd>+<kbd>C</kbd></li>
    <li>Head over to the
<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=MegaBlast&amp;PROGRAM=blastn">NCBI BLAST</a>
service, paste the copied content into the <code class="language-plaintext highlighter-rouge">Enter Query Sequence</code> text
box and click <code class="language-plaintext highlighter-rouge">BLAST</code> at the bottom of the form.</li>
    <li>Wait for the <em>BLAST</em> run to finish.</li>
    <li>On the results page, look for the drop-down menu next to <code class="language-plaintext highlighter-rouge">Results for</code>.
It lets you toggle the BLAST hits list further down to include only the
matches to individual sequences from your multi-sequence query.</li>
  </ol>

</blockquote>

<p>Now take a bit of time to explore the BLAST hits uncovered for some of your
assembled nodes. Pay attention, specifically, to the node with the longest
sequence (node #1) and the circular node #2, but also investigate the results
for a few others.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Which genome is represented by node #1?</li>
    <li>Which genome corresponds to node #2? Does this finding remind you of
something you have learnt before?</li>
    <li>What do most other node sequences have in common? Do these additinal
findings make sense?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The sequence of node #1 is the assembled SARS-CoV-2 sequence we are
looking for. It is a perfect match to various SARS-CoV-2 genome
sequences found in Genbank over the entire assembled length, and we have
been able to assemble almost the entire genome even from the subsampled
sequencing data.</li>
      <li>The circular sequence of node #2 corresponds to the 5,386 bp genome of
bacteriophage phiX174. As explained in the more general
<a href="../unicycler-assembly/tutorial.html">Unicycler assembly</a> tutorial,
this genome is often used as a spike-in in Illumina sequencing.
Finding the complete sequence here is, thus,  another indication that
our analysis worked and produced meaningful results.</li>
      <li>
        <p>Almost all other assembled sequences appear to represent parts of
bacterial genomes. The only exceptions are the node #10 sequence, for
which no significant BLAST hits could be found, and the node #11
sequence, which represents a small stretch of left-over human genomic
DNA, which seems to have survived our subtraction approach.</p>

        <p>What all the bacterial genomes have in common is that they represent
genera of bacteria that are known to colonize the oral cavity and
mucosa. Since all samples in this analysis are BALF samples the presence
of DNA from such bacteria should not be surprising. In addition, some
members of the identified genera are known as opportunistic pathogens.
In particular, members of the genus <em>Prevotella</em> can infect the
respiratory tract and contribute to inflammation under anaerobic
conditions caused by primary infections. Hence, an alternative
explanation for the presence of some of these sequences in the samples
might be that the corresponding bacteria contributed to the clinical
picture of some of the Covid-19 patients they were obtained from.</p>
      </li>
    </ol>
  </blockquote>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>The power of modern genome assembly tools is remarkable, and so is their
robustness in the face of data of metagenomic nature. Assembling reads derived
from a virus and a good handful of copurified bacteria back into separate
contigs is a challenging task, which Unicycler solved without major issues!</p>

<p>Nevertheless, good quality assemblies still rely on proper preprocessing and
filtering to reduce the number of misassembly events, ambiguous assemblies and
of incorporation of sequencing errors into the final assembly.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT), 
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Certain types of NGS samples can be heavily contaminated with sequences from other genomes</p>
</li>
                        
                        <li><p>Reads from known/expected contaminating sources can be identified by mapping to the respective genomes</p>
</li>
                        
                        <li><p>After mapping, use filtering tools to remove identified contaminating reads, and use conversion tools to convert remaining mapped reads back into raw sequenced reads expected by most downstream tools</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/assembly/faqs/">FAQ page for the Assembly topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=assembly/assembly-with-preprocessing"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=assembly/assembly-with-preprocessing">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wolfgang Maier,  <b>Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html">https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{assembly-assembly-with-preprocessing,
author = "Wolfgang Maier",
	title = "Unicycler assembly of SARS-CoV-2 genome with preprocessing to remove human genome reads (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: bowtie2
  owner: devteam
  revisions: 749c918495f7
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fasta_filter_by_length
  owner: devteam
  revisions: 8cacfcf96a52
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_stats
  owner: devteam
  revisions: 145f6d74ff5e
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bandage
  owner: iuc
  revisions: b2860df42e16
  tool_panel_section_label: Graph/Display Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bandage
  owner: iuc
  revisions: 94fe43e75ddc
  tool_panel_section_label: Graph/Display Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastp
  owner: iuc
  revisions: 1d8fe9bc4cb0
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: minimap2
  owner: iuc
  revisions: 53c0b7a1a0c3
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: b2f1f75d49c4
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: nanoplot
  owner: iuc
  revisions: 645159bcee2d
  tool_panel_section_label: Nanopore
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_fastx
  owner: iuc
  revisions: a8d69aee190e
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_view
  owner: iuc
  revisions: b01db2684fa5
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seqtk
  owner: iuc
  revisions: 3da72230c066
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sra_tools
  owner: iuc
  revisions: aad3885b3216
  tool_panel_section_label: Get Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: unicycler
  owner: iuc
  revisions: 9e3e80cc4ad4
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collapse_collections
  owner: nml
  revisions: 830961c48e42
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">7</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>August 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The activity was very interactive with a good combination of theory and practical
					<b>Disliked</b>: The last step on NCBI BLAST search was not clear enough to a biginner like me
				</li>
				
		    </ul>
		    
		    <b>May 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The basics of each step explained very clearly.
					<b>Disliked</b>: Maybe the names of the files which need to be put in the field for the tool. Took a bit of time to understand in few places.
				</li>
				
		    </ul>
		    
		    <b>February 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: How in depth and clear it was.
					
				</li>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: it was very helpful.
					
				</li>
				
		    </ul>
		    
		    <b>January 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Très didactique
					<b>Disliked</b>: Peut être ajouter des copies d'écrans de l'historique ? (repérer les numéro d'actions) Mais peut être "anxiogène" si on a refait certaines opérations et que l'on perd ce repère de numéro...Donc non en fait :)
				</li>
				
		    </ul>
		    
		    <b>July 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: It was very detailed and easy to follow
					
				</li>
				
		    </ul>
		    
		    <b>April 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The format and outlining are fantastic
					<b>Disliked</b>: Add a screen shot or graphic to illustrate the major steps in this workflow
				</li>
				
		    </ul>
		    
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00029">gxy.io/GTN:T00029</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/assembly/tutorials/assembly-with-preprocessing/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/62278d71c6fca8d63a60269bfa0554a1883631ca">62278d7</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>