---
layout: tutorial_slides
logo: GTN

title: "Genome Browsers"
questions:
- What is a genome browser?
- How can I view alignments JBrowse?
- How can genome browsers help me evaluate mutations in cancer?
objectives:
- Gain familiarity with **JBrowse** (a genome browser)
- See examples of viewing alignments in JBrowse
- Identify variants from alignments in JBroswe
key_points:
- Genome browsers enable interactive visualisation of genomics data
- Viewing alignments can reveal both small and structural mutations

contributions:
  authorship: [hgibling]
  editing: [shiltemann]

subtopic: data-formats-databases

---

# What do we do with all these genomics text files?

![An image with pictograms signifying the various genomics file formats; FASTA, BAM, VCF, GFF, BED](images/lecture/fileformats.jpg)

???
In genomocs analysis you encounter a lot of different file types, many a which are heavily text-based.

But what do we do with them and how do we interpret them?


---

# Genome Browsers!


![Banner image with several screenshots of genome browsers arranged behind each other](images/lecture/genome-browsers-banner.png)

Genome browsers are interactive visualisation tools for exploring your large genomic datasets.

Examples: [IGV](https://software.broadinstitute.org/software/igv/), [JBrowse](https://jbrowse.org/jb2/) (this tutorial),[UCSC](https://genome.ucsc.edu/), many many more

???

Often these text-based files are hard to interpret, and it can be very useful to visualize this genomic information, for instance in a genome browser.

We will be using JBrowse for this tutorial.

It is one of many genome browsers available.

IGV is a popular desktop genome browser, and UCSC is an popular web-based one, but most genome browsers are pretty similar, and JBrowse works best with Galaxy.

It supports many different file types.

You can use it to view DNA and RNA alignments, DNA Methylation data, CHIP-Seq data, VCF files and many more.


---

# Launch JBrowse

![Screenshot of the JBrowse tool interface in Galaxy. With highlighted areas marking the reference genome selection parameter and the track configuration parameter.](images/lecture/jbrowse-launch.png)

JBrowse is an **interactive tool** in Galaxy, launches just like a regular tool

???

JBrowse is an interactive application within Galaxy, but launching JBrowse works the same as any other tool in Galaxy.

The tool form will ask you to select a reference genome, and configure other parameters.

Most importantly your data tracks.


---

# Launch JBrowse

.pull-left[

- Configure one or more data tracks
  - Alignment files (BAM)
  - Variant files (VCF)
  - Feature files (GTF/GFF/BED)
  - Other (BigWig, BLAST)
  - Remote data sources (SPARQL/REST)

]

.pull-right[

![Screenshot of the JBrowse interface for configuring a data track. First a data format must be selected, then the from history may be selected.](images/lecture/jbrowse-launch-tracks.png)

]

???

You can add as many data tracks as you like.

You first select the type of track you would like to add, and then provide the corresponding datasets from your history and optionally some other track settings.


---

# Screen layout

![Screenshot of the JBrowse interface after launch. Data tracks are visualized in the center. A variety of different tracks are visible. ](images/lecture/jbrowse-interface.png)


???

Once they tool has finished, you can click the eye icon to open the JBroswe genome browser.

You will see a screen that looks like this.

Data tracks are visualised in the center.

---

# Screen layout

![The same screenshot of the JBrowse interface with areas highlighted, including the location bar and zoom buttons at the top, the available tracks that can be enabled or disabled in the left-hand panel.](images/lecture/jbrowse-interface-explained.png)

???

You can use the location bar in the top menu to jump to a different location on the genome.

The zoom buttons can but used to zoom in and out, and you can also drag the screen sideways to move to neighbouring regions.

In the left-hand panel you can see the available data tracks you configured and can activate and deactivate them.

---

# Viewing Alignments


![Screenshot of JBrowse with an alignment track visible. Above it is the corresponding coverage track.](images/lecture/jbrowse-viewing-alignments.png)

Forward strand reads are **red**, reverse reads **blue**. Mismatches are coloured by base.

The **SNPs/Coverage** track can be autogenerated by selecting the option when you launch JBrowse.

???


---

# Viewing Alignments

![Screenshot of JBrowse with alignment track and zoomed in to clearly see individual SNVs](images/lecture/jbrowse-snps-zoomed.png)

Zoom in to see individual bases (of reference sequence and of mismatches)


---

# Settings

![Screenshot of JBrowse with many mismatches visible. The default color settings make it hard to distinguish real SNVs from sequencing errors.](images/lecture/jbrowse-snps-quick.png)

Quick! How many SNVs are in this screenshot?

---

# Settings: colouring options

![Screenshot of the settings menu where various](images/track-settings.png)
![](images/lecture/jbrowse-snps-colours.png)

JBrowse makes use of color coding to make viewing variants much easier

Settings can be accessed via the menu on the track name
---

# Settings: View as pairs


![](images/lecture/jbrowse-view-as-pairs.png)

**Track Visualisation options** allow you to view reads as pairs, as well as other settings

---

# Paired-end Sequencing: Orientation

![](images/lecture/paired-end-sequencing.png)

???
As a refresher, during paired-end sequencing library preparation, DNA is sheared into fragments, and only the ends of the fragment are actually sequenced.

From one end going in, and from the other end going the other direction.

So we expect the read pairs to be oriente left to right, when they are mapped back to the reference genome.

---

# Interpreting read pair orientation

![](images/lecture/paired-end-read-orientation.png)


???

And if we see other orientations that doesn't mean something has gone horribly wrong during sequencing, but rather it indicates that there is a stuctural variant in the sample that causes the read orientations to look a bit unexpected.

So these left-left or right-right pairs are seen in inversions

And if you have reads that go in the opposite direction, if they are going right-left, it might be indicative of a tandem duplication, or as a translocation in the same chromosome.


---

# Viewing by read pair orientation: Inversion

![](images/lecture/inversion.png)

Green = both reads forward, Blue = both reads reverse.
Often a drop in coverage around the breakpoints

???

In this example we see an inversion, and we've coloured the alignments by read pair orientation.

So we can see there are a bunch of reads with incorrect orientation with their mates.

The blue read pairs are both reverse facing, whereas the green read pairs are both forward facing

And we can look at the coverage track and see that there is a drop in coverage around the break points.

So you can use multiple tracks to confirm or validate what is going on in a particular sample.

---

# Paired-end sequencing: Insert Size

![](images/lecture/paired-end-insert-size.png)

???

During library prep, fragments generally undergo size selection, to try to select fragments that are roughly the same size.

That means the read pairs have an expected insert size.

The insert size is the distance between the end of the first read and the end of the second read.

When you map your reads back to the reference, a lot of aligners will use this expected insert size to find the best mapping, if there are multiple places to map.

But we can also use information about read pairs with insert size that are either smaller than expected or larger than expected in order to identify structural variants.

So if a fragment originates from a region with a large deletion or insertion, the inferred insert size, this is the insert size when it's mapped back to the reference genome, it changes.

It is either going to get larger if it's a deletion, or smaller if it's an insertion.


---


# Interpreting inferred insert size

![](images/lecture/paired-end-inferred-insert-size.png)

---

# Viewing by insert size: Deletion


![JBrowse screenshot of a large deletion. Read pairs with a larger than expected insert size are indicative of a deletion, and are colored red in JBrowse. Many concordant read pairs are also present so this is not a homozygous deletion](images/large-deletion.png)

Note drop in coverage. Pairs with larger-than-expected insert size are colored red

---


# What about viewing alignments from cancer samples?

---

# Considerations for viewing cancer alignments

.pull-left[

## Germline variants

- Have an expected frequency of 0.5 or 1

- Roughly **half** the reads should carry a variant if it is **heterozygous**

- Nearly **all** of the reads should carry a variant if it is **homozygous**
]

.pull-right[

## Somatic variants

- Have variable frequency

- Can be very low, very high, and anywhere in between due to **tumour heterogeneity**

]

---

# Deep coverage is necessary to detect mutations in low purity or high ploidy tumours

![](images/lecture/deep-coverage.png)


---

# Multiple types of cancer genome vatiation may be inferred from sequencing read alignments

![](images/lecture/variation-from-alignments.png)


