---
layout: tutorial_slides
logo: GTN

title: "Genome Browsers"
questions:
- What is a genome browser?
- How can I view alignments JBrowse?
- How can genome browsers help me evaluate mutations in cancer?
objectives:
- Gain familiarity with **JBrowse** (a genome browser)
- See examples of viewing alignments in JBrowse
- Identify variants from alignments in JBroswe
key_points:
- Genome browsers enable interactive visualisation of genomics data
- Viewing alignments can reveal both small and structural mutations

contributions:
  authorship: [hgibling]
  editing: [shiltemann]

subtopic: data-formats-databases
---

# What do we do with all these genomics text files?

![](images/lecture/fileformats.jpg)

---

# Genome Browsers!


![](images/lecture/genome-browsers-banner.png)

Genome browsers are interactive visualisation tools for exploring your large genomic datasets.

Examples: [IGV](https://software.broadinstitute.org/software/igv/), [JBrowse](https://jbrowse.org/jb2/) (this tutorial),[UCSC](https://genome.ucsc.edu/), many many more

---

# Launch JBrowse

![](images/lecture/jbrowse-launch.png)

JBrowse is an **interactive tool** in Galaxy, launches just like a regular tool

---

# Launch JBrowse

.pull-left[

- Configure one or more data tracks
  - Alignment files (BAM)
  - Variant files (VCF)
  - Feature files (GTF/GFF/BED)
  - Other (BigWig, BLAST)
  - Remote data sources (SPARQL/REST)

]

.pull-right[

![](images/lecture/jbrowse-launch-tracks.png)

]

---

# Screen layout

![](images/lecture/jbrowse-interface.png)

---

# Screen layout

![](images/lecture/jbrowse-interface-explained.png)



---

# Viewing Alignments


![](images/lecture/jbrowse-viewing-alignments.png)

Forward strand reads are **red**, reverse reads **blue**. Mismatches are coloured by base.

The **SNPs/Coverage** track can be autogenerated by selecting the option when you launch JBrowse.


---

# Viewing Alignments

![](images/lecture/jbrowse-snps-zoomed.png)

Zoom in to see individual bases (of reference sequence and of mismatches)


---

# Settings

![](images/lecture/jbrowse-snps-quick.png)

Quick! How many SNVs are in this screenshot?

---

# Settings

![](images/track-settings.png)
![](images/lecture/jbrowse-snps-colours.png)

JBrowse makes use of color coding to make viewing variants much easier

Settings can be accessed via the menu on the track name
---

# Settings


![](images/lecture/jbrowse-view-as-pairs.png)

**Track Visualisation options** allow you to view reads as pairs, as well as other settings

---

# Paired-end Sequencing

![](images/lecture/paired-end-sequencing.png)

???
As a refresher, during paired-end sequencing library preparation, DNA is sheared into fragments, and only the ends of the fragment are actually sequenced.

From one end going in, and from the other end going the other direction.

So we expect the read pairs to be oriente left to right, when they are mapped back to the reference genome.

---

# Interpreting read pair orientation

![](images/lecture/paired-end-read-orientation.png)


???

And if we see other orientations that doesn't mean something has gone horribly wrong during sequencing, but rather it indicates that there is a stuctural variant in the sample that causes the read orientations to look a bit unexpected.

So these left-left or right-right pairs are seen in inversions

And if you have reads that go in the opposite direction, if they are going right-left, it might be indicative of a tandem duplication, or as a translocation in the same chromosome.


---

# Viewing by read pair orientation: inversion

![](images/lecture/breakpoint-coverage.png)

Note drop in coverage at breakpoints

???

In this example we see an inversion, and we've coloured the alignments by read pair orientation.

So we can see there are a bunch of reads with incorrect orientation with their mates.

And we can look at the coverage track and see that there is a drop in coverage around the break points.

So you can use multiple tracks to confirm or validate what is going on in a particular sample.

---

# Paired-end sequencing

![](images/lecture/paired-end-insert-size.png)

???

---


# Interpreting inferred insert size

![](images/lecture/paired-end-inferred-insert-size.png)

---

# Viewing by insert size: Deletion


TODO

Note drop in coverage. Pairs with larger-than-expected insert size are colored red

---
# Viewing by insert size: Rearrangement

TODO

---

# What about viewing alignments from cancer samples?

---

# Considerations for viewing cancer alignments

.pull-left[

## Germline variants

- Have an expected frequency of 0.5 or 1

- Roughly **half** the reads should carry a variant if it is **heterozygous**

- Nearly **all** of the reads should carry a variant if it is **homozygous**
]

.pull-right[

## Somatic variants

- Have variable frequency

- Can be very low, very high, and anywhere in between due to **tumour heterogeneity**

]

---

# Deep coverage is necessary to detect mutations in low purity or high ploidy tumours

![](images/lecture/deep-coverage.png)


---

# Multiple types of cancer genome vatiation may be inferred from sequencing read alignments

![](images/lecture/variation-from-alignments.png)


