---
title: "Galaxy Single-cell Community: Year in Review"
contributions:
  authorship: [mtekman, nomadscientist, pavanvidem]
tags: [gtn, communications, single-cell]
layout: news
cover: "news/images/2023_dec_sc.png"
coveralt: "swirled cluster dots surround a circle of people all holding hands, looking towards the bright center (future)"
---

ðŸš€Embarking on a cosmic journey, the Galaxy Single-cell Community has clustered together to unveil a constellation of tools, making strides in RNA-stellar discoveries and creating out-of-this-world workflows. With a commitment to battling work duplication across the multiverse, this community is boldly charting a course for global domination, proving that when it comes to bioinformatics, the Galaxy is the limit!âœ¨

# Tools & Workflows

The past year has been a spectacular journey in advancing tools and workflows within the Galaxy Single-cell Community. Breaking free from the confines of scRNA-seq, we've expanded our horizons to include tools for analyzing data from other modalities such as single-cell ATAC-seq and CITE-Seq. Dive into our updated scRNA-seq analysis tools, now featuring more downstream analysis options for trajectory analysis. We've even thrown in advanced analysis tools for estimating cell type proportions from bulk RNA-seq data!

**RNA STARSolo** just got a makeover to version 2.7.10b, boasting a revamped UI, Velocyto-like UMI counting, and nifty alternatives for quantifying single nuclei data. Analyzing single-cell ATAC-seq data on Galaxy is now a breeze with the introduction of the **EpiScanpy** tool suite, built on the renowned **Scanpy** toolkit. Plus, we've added some powerful tools from the **Sinto** toolkit for preprocessing scATAC-seq data on Galaxy. And that's not all! The **MuSiC** tool suite in Galaxy now unveils deconvolution tools for discovering cell type composition in bulk RNA-seq using cell types within scRNA-seq. Multi-faceted comparisons are now at your fingertips with the **MuSiC Compare** tool.

Seurat fans, rejoice! Updates have rolled in, offering additional functionalities and optional CITE-seq capabilities at runtime.

# Community Infrastructure

Our efforts in community building have seen the unification of single-cell subdomains into [https://singlecell.usegalaxy.eu/](https://singlecell.usegalaxy.eu/), the creation of the Single Cell Community of Practice hub page ([https://galaxyproject.org/projects/singlecell/](https://galaxyproject.org/projects/singlecell/)), and a user-focused Matrix channel. Testing brilliant widgets, automated updates, and the latest addition of automated help request updates onto Matrix channels have made community collaboration a breeze.

# Global Connections

Down under, the Australian BioCommons launched their [Single Cell and Spatial Omics Analysis Infrastructure Roadmap for Australia](https://zenodo.org/records/10368976), complementing our endeavors to unite across timezones. Collaboration, including international tool adventures across three continents, proves that team work does, in fact, make the dream work.

## Training

Get ready for a thrilling journey through the [Single Cell Training]({% link topics/single-cell %})'s latest updates! The **Case Study Reloaded** tutorial, fueled by three remote-working undergrads, introduces parallel trajectory analysis in R, Python, and the Galaxy GUI.

Sustainability shines as we've revamped numerous tutorials with hands-on and video re-recordings. New tools bring new tutorials, such as [Pre-processing of 10X Single-Cell ATAC-seq Datasets]({% link topics/single-cell/tutorials/scatac-preprocessing-tenx/tutorial.md %}) and [Comparing inferred cell compositions using MuSiC deconvolution]({% link topics/single-cell/tutorials/bulk-music-4-compare/tutorial.md %}).

Tackling user hurdles, our data ingest and conversion tutorials in the **Changing data formats & preparing objects** section await exploration.

Dive into the unique **Tips, tricks & other hints** section, offering analysis options like [parameter iteration]({% link topics/single-cell/tutorials/scanpy_parameter_iterator/tutorial.md %}) for swift optimization or [Removing the effects of the cell cycle]({% link topics/single-cell/tutorials/scrna-case_cell-cycle/tutorial.md %}) for precise clustering analysis.

And don't miss our inaugural [Learning Pathway]({% link learning-pathways/intro_single_cell.md %}), guiding Galaxy novices to becoming single-cell analysis experts!


# Fighting Work Duplication

Our mission: No more work duplication! We've consolidated single-cell instances and are diligently preventing tool and infrastructure duplication, creating smoother pathways across servers. We value efficiency here at single-cell, so we'd like to thank ChatGPT (free version) for adding in Galaxy puns to this post, as per our request.

# What a Year!

What's next on our list? World domination or global collaboration? You decide. Stay tuned for another exciting year with the Galaxy Single-cell Community!
