---
title: "New Tutorial: Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition"
tags: [new tutorial]
contributions:
  authorship: [bebatut, EngyNasr]
cover: "topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/images/FoodBorne-Workflow-updated.png"
tutorial: "topics/metagenomics/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.html"
coveralt: "screenshot from the data manipulation tutorial"

layout: news
---

Food contamination with pathogens are a major burden on our society. Globally, they affect an estimated 600 million people a year and impact socioeconomic development at different levels. These outbreaks are mainly due to _Salmonella spp._ followed by _Campylobacter spp._ and Noroviruses.

The evolution of techniques in the last decades has made possible the development of methods to quickly identify responsible pathogen using their **DNA and without prior isolation**. Long-read sequencing tehcniques like ONT make these methods even **easier and more practical to identify strains quicker and with fewer reads**.

In this tutorial, using direct Nanopore sequencing data, we show how to
1. **agnostically detect pathogens** (what exactly is this pathogen and what is the degree of its severity (its virulence factors) from data extracted directly (without prior cultivation) from a potentially contaminated sample (e.g. food like chicken, cow, etc.,) and sequenced using Nanopore
2. **compare different samples to track** the possible source of contamination

This tutorial is offered in 2 different version:
- A short version, running already build workflows
- A long version, going step-by-step
