---
title: From GTN Intern to Tutorial Author to Bioinformatician
layout: news
tags:
- single-cell
- training
- education
- trajectory
- user
- contributor
from_google_form: true
contributions:
  authorship:
  - Camila-goclowski
  editing:
  - nomadscientist
tutorial: topics/single-cell/tutorials/scrna-case_FilterPlotandExploreRStudio/tutorial.html

---
With growing access and interest in sequencing data, Galaxy is a knight in shining armor for wet lab scientists hoping to analyze their own data. With long term intentions of increasing access to bioinformatic analyses, the Galaxy Training Network (GTN) creates a safe space where non-computer-scientists may analyze their own data and even learn to code: an invaluable skill in today’s scientific world. Galaxy introduced me to brand new skills as an undergraduate and ultimately changed the trajectory of my career. Here is my story as a biology undergraduate with no coding experience turned GTN contributor &, eventually, coding bioinformatician: thanks to Galaxy.

 My introduction to Galaxy came from Dr. Wendi Bacon of the Open University. Dr. Bacon has spearheaded efforts to expand upon the GTN’s currently available single-cell tools and tutorials. I was taken on as an intern in the fall of 2021 to support these efforts–mainly focused on the development of single-cell RNA sequencing (scRNA-seq) analysis tutorials using Seurat. Although I had officially become a bioinformatics intern, I still had little to no experience with coding and none with scRNA-seq data.

 Dr. Bacon started me out kindly with introductory tutorials already available on the GTN while I learned what this data looks like and how to analyze it. While waiting for tools to become available, I began using RStudio and learned how to translate the Scanpy buttons tutorial into a Seurat RStudio one. This task proved to be more difficult than initially believed: me, with no coding experience, trying to translate wrapped Python tools into raw R code. After many months of trial and error, I drafted the first Seurat based tutorial hosted by the GTN. I quickly realized I had taught myself to code–or at the very least, learned to teach myself to code via pattern recognition and repetition.

 The trajectory of my career thus far has been heavily driven by a love for bioinformatics that was founded thanks to Galaxy. I have since begun conducting research as a technician at the University of Utah (USA), and will soon begin a PhD program leveraging my computational skills and interests. I have spearheaded multiple scRNA-seq projects my current lab studying development of the diaphragm (an essential mammalian organ), a common birth defect associated with aberrant diaphragm development (CDH), and retinoic acid’s role in both. Leveraging what I learned via Galaxy, I am answering increasingly complex biological questions.

 The GTN contributes to the scientific world something that is crucial for successful integration of bench and computational biology. The ability for scientists to produce and analyze their own datasets allows for more scientific hypotheses to be formed and questions to be answered. These facts have been integral to my success as a scientist and for that I am grateful for Galaxy, the GTN, and the community at large.

