---
layout: tutorial_slides
topic_name: assembly
tutorial_name: "debruijn-graph-assembly"
logo: "GTN"
class: enlarge120
---

.enlarge120[

# ***De novo* Genome Assembly**

## **Part 2: De Bruijn Graph Assembly**

]

#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman
---
.enlarge120[
# **de Bruijn Graphs**
]
.pull-left[
* Named after Nicolaas Govert de Bruijn
* Directed graph representing overlaps between sequences of symbols
* Sequences can be reconstructed by moving between nodes in graph
]
.pull-right[
.image-50[![](../../images/debruijn.png)]
]
---
.enlarge120[# **de Bruijn Graphs**

* A directed graph of sequences of symbols
* Nodes in the graph are k-mers
* Edges represent consecutive k-mers (which overlap by k-1 symbols)
]
Consider the 2 symbol alphabet (0 & 1) de Bruijn Graph for k =3
![](../../images/debruijn01.png)
---
.enlarge120[# **Producing sequences**

* Sequences of symbols are produced by moving through the graph
]
![](../../images/debruijn02.png)

---
.enlarge120[
# **K-mers?**       .pull-right[.image-25[![](../../images/specialk.png)]]


<hr>
* To be able to use de Bruijn graphs, we need reads of length **L** to overlap by **L-1** bases.
* Not all reads will overlap another read perfectly.
  * Read errors
  * Coverage "holes"
* Not all reads are the same length (depending on technology and quality cleanup)

***To help us get around these problems, we use all k-length subsequences of the reads, these are the k-mers.***
]
---
.enlarge120[
# **What are K-mers?**       .pull-right[.image-25[![](../../images/specialk.png)]]]
![](../../images/kmers01.png)

---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-1.png)
---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-2.png)
---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-3.png)
---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-4.png)

---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-1.png)
---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-2.png)
---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-3.png)
---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-4.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-1.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-2.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-3.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-4.png)

---
.enlarge120[
# **Choose k wisely**       .pull-right[.image-25[![](../../images/wisely.png)]]]

.enlarge120[
* Lower k
  * More connections
  * Less chance of resolving small repeats
  * Higher k-mer coverage

* Higher k
  * Less connections
  * More chance of resolving small repeats
  * Lower k-mer coverage

***Optimum value for k will balance these effects.***

]
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/blank.png)
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/ex3-2.png)
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/ex3-3.png)
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/ex3-4.png)

---
.enlarge120[
# **More coverage**       .pull-right[.image-50[![](../../images/depthlogo.png)]]]

.enlarge120[
* Errors won't be duplicated in every read
* Most reads will be error free
* We can count the frequency of each k-mer
* Annotate the graph with the frequencies
* Use the frequency data to clean the de Bruijn graph


***More coverage depth will help overcome errors!***
]
---
.enlarge120[
# **Read errors revisited**       .pull-right[.image-50[![](../../images/hazardsymbol.png)]]]
![](../../images/ex3a.png)

---
.enlarge120[
# **Another parameter - coverage cutoff**]

.enlarge120[
* At what point is a low coverage indicative of an error?
* Can we ignore low coverage nodes and paths?
* This is a new assembly parameter

***Coverage cutoff***
]

---
.enlarge120[
# **de Bruijn graph assembly process**]

.enlarge120[
1. Select a value for k
2. "Hash" the reads (make the kmers)
3. Count the kmers
4. Make the de Bruijn graph
5. **Perform graph simplification steps** - use cov cutoff
6. Read off contigs from simplified graph

]

---
.enlarge120[
# **Graph simplification**]

## Step 1: Chain merging

![](../../images/tipclipping.png)

---
.enlarge120[
# **Graph simplification**]

## Step 2: Tip clipping

![](../../images/shortnodes.png)

---
.enlarge120[
# **Graph simplification**]


.pull-left[
![](../../images/bubblemerging.png)
]
.pull-right[
## Step 3: Bubble collapsing

* Detect redundant paths through graph
* Compare the paths using sequence alignment
* If similar, merge the paths

.reduce70[Image: Zerbino & Birney 2008]
]

---
.enlarge120[
# **Graph simplification**

## Step 4: Remove low coverage nodes

* Remove erroneous nodes and connections using the "coverage cutoff"
* Genuine short nodes will have a high coverage

]
---
.enlarge120[
# **Make contigs**

* Find an unbalanced node in the graph
* Follow the chain of nodes and "read off" the bases to produce the contigs
* Where there is an ambiguous divergence/convergence, stop the current contig and start a new one.
* Re-trace the reads through the contigs to help with repeat resolution

]

---
.enlarge120[
# **Velvet**

Velvet has two separate programs:


* Velveth
  * Makes the k-mers and
  * Efficiently counts (hashes) them
  * All in O(N) time


* Velvetg
  * Makes the graph - O(U) time. U = unique k-mers.
  * Simplifies it
  * Makes contigs - O(E) time. E = edges in graph

But: You need to choose **k** and **c** wisely!

]

---
.enlarge120[
# **Velvet - Paired end scaffolding**

* Breadcrumb algorithm
]

![](../../images/breadcrumb.png)

---
.enlarge120[
# **Assembly Exercise #2**

* We will perform an optimised assembly using the Velvet Optimiser in Galaxy.
* The optimiser will pick a good value for **k** and **c**, and perform graph simplification for us.
* We will compare the results with our earlier, simple assemblies.
]
---
.enlarge120[
# **Extensions of the idea**
]
---
.enlarge120[
# **SPAdes**     .pull-right[.image-50[![](../../images/spades.png)]]]

.enlarge120[
* de Bruijn graph assembler by Pavel Pevzner's group out of St. Petersburg


* Uses multiple k-mers to build the graph
  * Graph has connectivity **and** specificity
  * Usually use a low, medium and high k-mer size together.


* Performs error correction on the reads first

* Maps reads back to the contigs and scaffolds as a check

* Under active development

* Much slower than Velvet
* Should be used in preference to Velvet now.

]
---
.enlarge120[
# **A move back to OLC**]

.pull-left[
.enlarge120[
* New long read technologies
  * PacBio and MinIon


* Assemblers: HGap, CANU
  * Use overlap, layout consensus approach


* CANU can perform hybrid assemblies with long and short reads
]
]

.pull-right[
![](../../images/minionpacbio.png)
]
---
.enlarge120[# **Bandage**

* Assembly graph viewer and manipulator
* Written by Ryan Wick of Centre for Systems Genomics - Uni. Melbourne, Australia
]
![](../../images/bandage.png)

---
.enlarge120[# **Assembly Exercise #3**

* We will perform an assembly with **SPAdes** in **Galaxy**
* We will again look at the assembly metrics and compare them with our sample and optimised Velvet assemblies.
]
