<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>De Bruijn Graph Assembly</title>
    
        <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="DNA sequence data has become an indispensable tool for Mo..." />
    <meta property="og:title" content="Galaxy Training: De Bruijn Graph Assembly" />
    <meta property="og:description" content="DNA sequence data has become an indispensable tool for Mo..." />
    <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    <script type="application/ld+json">
      


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "De Bruijn Graph Assembly",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org//training-material/topics/assembly/"
  },
  "courseCode": "assembly / debruijn-graph-assembly / slides",
  "learningResourceType": "slides",
  "name": "Slides for 'De Bruijn Graph Assembly' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/assembly/tutorials/debruijn-graph-assembly/slides.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - What are the factors that affect genome assembly?\n - How does Genome assembly work?\n\n\\nThe objectives are:\n - Perform an optimised Velvet assembly with the Velvet Optimiser\n - Compare this assembly with those we did in the basic tutorial\n - Perform an assembly using the SPAdes assembler.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Simon Gladman"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Simon Gladman"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org//training-material/topics/assembly/"
    }
  ]
}
    </script>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/assembly" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/assembly/tutorials/debruijn-graph-assembly/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# De Bruijn Graph Assembly



---

## Requirements

Before diving into this slide deck, we recommend you to have a look at:



    
        
        
            
        
- [Introduction to Galaxy Analyses](/training-material/topics/introduction)
    

    
        
        
            
        
- [Sequence analysis](/training-material/topics/sequence-analysis)
    - Quality Control: [<i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides](/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html) - [<i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on](/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html)
    






.footnote[Tip: press `P` to view the presenter notes]

???
Presenter notes contain extra information which might be useful if you
intend to use these slides for teaching.

Press `P` again to switch presenter notes off

Press `C` to create a new window where the same presentation will be displayed.
This window is linked to the main window. Changing slides on one will cause the
slide to change on the other. Useful when presenting.

---






### &lt;i class=&quot;far fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- What are the factors that affect genome assembly?

- How does Genome assembly work?


---



### &lt;i class=&quot;fas fa-bullseye&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;objectives&lt;/span&gt; Objectives


- Perform an optimised Velvet assembly with the Velvet Optimiser

- Compare this assembly with those we did in the basic tutorial

- Perform an assembly using the SPAdes assembler.


---


.enlarge120[

# ***De novo* Genome Assembly**

## **Part 2: De Bruijn Graph Assembly**

]

#### With thanks to T Seemann, D Bulach, I Cooke and Simon Gladman
---
.enlarge120[
# **de Bruijn Graphs**
]
.pull-left[
* Named after Nicolaas Govert de Bruijn
* Directed graph representing overlaps between sequences of symbols
* Sequences can be reconstructed by moving between nodes in graph
]
.pull-right[
.image-50[![](../../images/debruijn.png)]
]
---
.enlarge120[# **de Bruijn Graphs**

* A directed graph of sequences of symbols
* Nodes in the graph are k-mers
* Edges represent consecutive k-mers (which overlap by k-1 symbols)
]
Consider the 2 symbol alphabet (0 &amp; 1) de Bruijn Graph for k =3
![](../../images/debruijn01.png)
---
.enlarge120[# **Producing sequences**

* Sequences of symbols are produced by moving through the graph
]
![](../../images/debruijn02.png)

---
.enlarge120[
# **K-mers?**       .pull-right[.image-25[![](../../images/specialk.png)]]


&lt;hr&gt;
* To be able to use de Bruijn graphs, we need reads of length **L** to overlap by **L-1** bases.
* Not all reads will overlap another read perfectly.
  * Read errors
  * Coverage &quot;holes&quot;
* Not all reads are the same length (depending on technology and quality cleanup)

***To help us get around these problems, we use all k-length subsequences of the reads, these are the k-mers.***
]
---
.enlarge120[
# **What are K-mers?**       .pull-right[.image-25[![](../../images/specialk.png)]]]
![](../../images/kmers01.png)

---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-1.png)
---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-2.png)
---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-3.png)
---
.enlarge120[
# **K-mers de Bruijn graph**       .pull-right[.image-25[![](../../images/abc.png)]]]
![](../../images/ex1-4.png)

---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-1.png)
---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-2.png)
---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-3.png)
---
.enlarge120[
# **The problem of repeats**       .pull-right[.image-25[![](../../images/repeatlogo.png)]]]
![](../../images/ex2-4.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-1.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-2.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-3.png)
---
.enlarge120[
# **Different k**       .pull-right[.image-25[![](../../images/bigk.png)]]]
![](../../images/ex2a-4.png)

---
.enlarge120[
# **Choose k wisely**       .pull-right[.image-25[![](../../images/wisely.png)]]]

.enlarge120[
* Lower k
  * More connections
  * Less chance of resolving small repeats
  * Higher k-mer coverage

* Higher k
  * Less connections
  * More chance of resolving small repeats
  * Lower k-mer coverage

***Optimum value for k will balance these effects.***

]
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/blank.png)
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/ex3-2.png)
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/ex3-3.png)
---
.enlarge120[
# **Read errors**       .pull-right[.image-25[![](../../images/hazardsymbol.png)]]]
.image-75[![](../../images/ex3-head.png)]
![](../../images/ex3-4.png)

---
.enlarge120[
# **More coverage**       .pull-right[.image-50[![](../../images/depthlogo.png)]]]

.enlarge120[
* Errors won't be duplicated in every read
* Most reads will be error free
* We can count the frequency of each k-mer
* Annotate the graph with the frequencies
* Use the frequency data to clean the de Bruijn graph


***More coverage depth will help overcome errors!***
]
---
.enlarge120[
# **Read errors revisited**       .pull-right[.image-50[![](../../images/hazardsymbol.png)]]]
![](../../images/ex3a.png)

---
.enlarge120[
# **Another parameter - coverage cutoff**]

.enlarge120[
* At what point is a low coverage indicative of an error?
* Can we ignore low coverage nodes and paths?
* This is a new assembly parameter

***Coverage cutoff***
]

---
.enlarge120[
# **de Bruijn graph assembly process**]

.enlarge120[
1. Select a value for k
2. &quot;Hash&quot; the reads (make the kmers)
3. Count the kmers
4. Make the de Bruijn graph
5. **Perform graph simplification steps** - use cov cutoff
6. Read off contigs from simplified graph

]

---
.enlarge120[
# **Graph simplification**]

## Step 1: Chain merging

![](../../images/tipclipping.png)

---
.enlarge120[
# **Graph simplification**]

## Step 2: Tip clipping

![](../../images/shortnodes.png)

---
.enlarge120[
# **Graph simplification**]


.pull-left[
![](../../images/bubblemerging.png)
]
.pull-right[
## Step 3: Bubble collapsing

* Detect redundant paths through graph
* Compare the paths using sequence alignment
* If similar, merge the paths

.reduce70[Image: Zerbino &amp; Birney 2008]
]

---
.enlarge120[
# **Graph simplification**

## Step 4: Remove low coverage nodes

* Remove erroneous nodes and connections using the &quot;coverage cutoff&quot;
* Genuine short nodes will have a high coverage

]
---
.enlarge120[
# **Make contigs**

* Find an unbalanced node in the graph
* Follow the chain of nodes and &quot;read off&quot; the bases to produce the contigs
* Where there is an ambiguous divergence/convergence, stop the current contig and start a new one.
* Re-trace the reads through the contigs to help with repeat resolution

]

---
.enlarge120[
# **Velvet**

Velvet has two separate programs:


* Velveth
  * Makes the k-mers and
  * Efficiently counts (hashes) them
  * All in O(N) time


* Velvetg
  * Makes the graph - O(U) time. U = unique k-mers.
  * Simplifies it
  * Makes contigs - O(E) time. E = edges in graph

But: You need to choose **k** and **c** wisely!

]

---
.enlarge120[
# **Velvet - Paired end scaffolding**

* Breadcrumb algorithm
]

![](../../images/breadcrumb.png)

---
.enlarge120[
# **Assembly Exercise #2**

* We will perform an optimised assembly using the Velvet Optimiser in Galaxy.
* The optimiser will pick a good value for **k** and **c**, and perform graph simplification for us.
* We will compare the results with our earlier, simple assemblies.
]
---
.enlarge120[
# **Extensions of the idea**
]
---
.enlarge120[
# **SPAdes**     .pull-right[.image-50[![](../../images/spades.png)]]]

.enlarge120[
* de Bruijn graph assembler by Pavel Pevzner's group out of St. Petersburg


* Uses multiple k-mers to build the graph
  * Graph has connectivity **and** specificity
  * Usually use a low, medium and high k-mer size together.


* Performs error correction on the reads first

* Maps reads back to the contigs and scaffolds as a check

* Under active development

* Much slower than Velvet
* Should be used in preference to Velvet now.

]
---
.enlarge120[
# **A move back to OLC**]

.pull-left[
.enlarge120[
* New long read technologies
  * PacBio and MinIon


* Assemblers: HGap, CANU
  * Use overlap, layout consensus approach


* CANU can perform hybrid assemblies with long and short reads
]
]

.pull-right[
![](../../images/minionpacbio.png)
]
---
.enlarge120[# **Bandage**

* Assembly graph viewer and manipulator
* Written by Ryan Wick of Centre for Systems Genomics - Uni. Melbourne, Australia
]
![](../../images/bandage.png)

---
.enlarge120[# **Assembly Exercise #3**

* We will perform an assembly with **SPAdes** in **Galaxy**
* We will again look at the assembly metrics and compare them with our sample and optimised Velvet assemblies.
]




---
### &lt;i class=&quot;fas fa-key&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;keypoints&lt;/span&gt; Key points


- We learned about how the choice of k-mer size will affect assembly outcomes

- We learned about the strategies that assemblers use to make reference genomes

- We performed a number of assemblies with Velvet and SPAdes.

- You should use SPAdes or another more modern assembler than Velvet for actual assemblies now.







---

## Thank you!

This material is the result of a collaborative work. Thanks to the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame/slugger70/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/slugger70" alt="Simon Gladman">Simon Gladman</a>


</div>



<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.


    </textarea>
	<script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
      var snippets = document.querySelectorAll('code.remark-code');
        [].forEach.call(snippets,function(snippet){
          snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });
      var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.parentElement;
      }});
    </script>

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('body').prepend("<div style='text-align: center'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>

  </body>
</html>
