<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        



<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/assembly" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Assembly
                        </a>
                    </li>

                    
                        
                        
                        
                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="https://doi.org/10.5281/zenodo.582600" title="Links to data">
                            <i class="fa fa-files-o" aria-hidden="true"></i> Input Dataset
                        </a>
                    </li>
                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/assembly#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>De Bruijn Graph Assembly</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>What are the factors that affect genome assembly?</li>
            
            <li>How does Genome assembly work?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>Perform an optimised Velvet assembly with the Velvet Optimiser</li>
            
            <li>Compare this assembly with those we did in the basic tutorial</li>
            
            <li>Perform an assembly using the SPAdes assembler.</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
                <li>
                    
                    <a href="/training-material/topics//sequence-analysis/">Quality Control</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1d/3h/6h</p>
        </blockquote>

        <h1 id="optimised-de-bruijn-graph-assemblies-using-the-velvet-optimiser-and-spades">Optimised de Bruijn Graph assemblies using the Velvet Optimiser and SPAdes</h1>
<p>In this activity, we will perform <em>de novo</em> assemblies of a short read set using the Velvet Optimiser and the SPAdes assemblers. We are using the Velvet Optimiser for illustrative purposes. For real assembly work, a more suitable assembler should be chosen - such as SPAdes.</p>

<p>The Velvet Optimiser is a script written by Simon Gladman to optimise the k-mer size and coverage cutoff parameters for Velvet. More information can be found <a href="https://github.com/slugger70/VelvetOptimiser">here</a></p>

<p>SPAdes is a de novo genome assembler written by Pavel Pevzner’s group in St. Petersburg. More details on it can be found <a href="http://cab.spbu.ru/software/spades/">here</a></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

  <ol>
    <li><a href="#get-the-data">Get the data</a></li>
    <li><a href="#assembly-with-the-velvet-optimiser">Assemble with the Velvet Optimiser</a></li>
    <li><a href="#assemble-with-spades">Assemble with SPAdes</a></li>
  </ol>
</blockquote>

<h1 id="get-the-data">Get the data</h1>

<p>We will be using the same data that we used in the introductory tutorial, so if you have already completed that and have the data, skip this section.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-getting-the-data"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Getting the data</h3>

  <ol>
    <li>Create and name a new history for this tutorial.</li>
    <li>
      <p>Import the sequence read raw data (*.fastq) from <a href="https://zenodo.org/record/582600">Zenodo</a></p>

      <blockquote class="tip">
        <h3 id="-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location (Right-click on the filename <i class="fa fa-long-arrow-right"></i> Copy Link Address)</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field</li>
          <li>Change the data-type to <strong>fastqsanger</strong></li>
          <li>Press <strong>Start</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>Once the files have been uploaded, change their names to Mutant_R1.fastq and Mutant_R2.fastq respectively by clicking on the <img src="../../images/edit_file.png" alt="Pencil icon" /> icon next to the relevant history entry.</li>
  </ol>

  <p>Click on the View Data button (the <img src="../../images/eye.png" alt="Eye icon" />) next to one of the FASTQ sequence files.</p>

  <blockquote class="question">
    <h3 id="-questions"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

    <ol>
      <li>What are four key features of a FASTQ file?</li>
      <li>What is the main difference between a FASTQ and a FASTA file?</li>
    </ol>
  </blockquote>

</blockquote>

<h1 id="assembly-with-the-velvet-optimiser">Assembly with the Velvet Optimiser</h1>

<p>We will perform an assembly with the Velvet Optimiser. It will automatically choose a suitable value for the k-mer size (<strong>k</strong>). It will then go on to optimise the coverage cutoff (<strong>cov_cutoff</strong>) which corrects for read errors. It will use the “<em>n50</em>” metric for optimising the k-mer size and the “<em>total number of bases in contigs</em>” for optimising the coverage cutoff.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-assemble-with-the-velvet-optimiser"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Assemble with the Velvet Optimiser</h3>

  <h4 id="use-the-velvet-optimiser--tool">Use the <strong>Velvet Optimiser</strong> <i class="fa fa-wrench" aria-hidden="true"></i> tool.</h4>

  <ul>
    <li>
      <p>Set the following parameters (leave other settings as they are):</p>

      <ul>
        <li><strong>Start k-mer size</strong>: 45</li>
        <li><strong>End k-mer size</strong>: 73</li>
        <li><strong>Input file type</strong>: Fastq</li>
        <li><strong>Single or paired end reads</strong>: Paired</li>
        <li><strong>Select first set of reads</strong>: <em>mutant_R1.fastq</em></li>
        <li><strong>Select second set of reads</strong>: <em>mutant_R2.fastq</em></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>Your history will now contain a number of new files:</p>

<ul>
  <li>Velvet optimiser contigs
    <ul>
      <li>A fasta file of the final assembled contigs</li>
    </ul>
  </li>
  <li>Velvet optimiser contig stats
    <ul>
      <li>A table of the lengths (in k-mer length) and coverages (k-mer coverages) for the final contigs.</li>
    </ul>
  </li>
</ul>

<p>Have a look at each file.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-get-contig-statistics-for-velvet-optimiser-contigs"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Get contig statistics for Velvet Optimiser contigs</h3>

  <h4 id="use-the-fasta-statistics--tool-to-produce-a-summary-of-the-velvet-optimiser-contigs">Use the <strong>Fasta Statistics</strong> <i class="fa fa-wrench" aria-hidden="true"></i> tool to produce a summary of the velvet optimiser contigs</h4>

  <p>Look at the output file.</p>

  <blockquote class="question">
    <h3 id="-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

    <p>Compare the output we got here with the output of the simple assemblies obtained in the introductory tutorial.</p>
    <ol>
      <li>What are the main differences between them?</li>
      <li>Which has a higher “n50”? What does this mean?</li>
    </ol>
  </blockquote>

</blockquote>

<p>Tables of results from <strong>(a)</strong> Simple assembly and <strong>(b)</strong> optimised assembly.</p>

<p><strong>(a)</strong> <img src="../../images/image12.png" alt="The results of the contigs from Simple assembly." /></p>

<p><strong>(b)</strong> <img src="../../images/optstats.png" alt="The results of the contigs from Optimised assembly. In contrast to simple assembly produced much higher n_50, while num_seq is lower." /></p>

<h1 id="assemble-with-spades">Assemble with SPAdes</h1>

<p>We will now perform an assembly with the much more modern SPAdes assembler. It goes through a similar process to Velvet in the fact that it uses and simplifies de Bruijn graphs but it uses multiple values for k-mer size and combines the resultant graphs. This combination produces very good assemblies. When using SPAdes it is typical to choose at least 3 k-mer sizes. One lowm one medium and one high. We will use 33, 55 and 91.</p>

<blockquote class="hands_on">
  <h3 id="-hands-on-assemble-with-spades"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Assemble with SPAdes</h3>

  <h4 id="use-the-spades--tool">Use the <strong>SPAdes</strong> <i class="fa fa-wrench" aria-hidden="true"></i> tool.</h4>

  <ul>
    <li>
      <p>Set the following parameters (leave other settings as they are):</p>

      <ul>
        <li><strong>Run only assembly</strong>: yes</li>
        <li><strong>K-mers to use separated by commas</strong>: 33,55,91 <em>[note: no spaces!]</em></li>
        <li><strong>Coverage cutoff</strong>: auto</li>
        <li><strong>Files -&gt; forward reads</strong>: <em>mutant_R1.fastq</em></li>
        <li><strong>Files -&gt; reverse reads</strong>: <em>mutant_R2.fastq</em></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>You will now have 5 new files in your history:</p>

<ul>
  <li>two Fasta files, one for contigs and one for scaffolds</li>
  <li>two statistics files, one for contigs and one for scaffolds</li>
  <li>the SPAdes log file.</li>
</ul>

<p>Examine each file, especially the stats files.</p>

<p><img src="../../images/contig_stats.png" alt="Contig stats file with NODE_5 being the shortest contig with the highest coverage and NODE_1 being the opposite." /></p>

<blockquote class="question">
  <h3 id="-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

  <ol>
    <li>Why would one of the contigs have much higher coverage than the others?</li>
    <li>What could this represent?</li>
  </ol>

</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-get-contig-statistics-for-spades-contigs"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Get contig statistics for SPAdes contigs</h3>

  <h4 id="use-the-fasta-statistics--tool-to-produce-a-summary-of-the-spades-contigs">Use the <strong>Fasta Statistics</strong> <i class="fa fa-wrench" aria-hidden="true"></i> tool to produce a summary of the SPAdes contigs</h4>

  <p>Look at the output file.</p>

  <blockquote class="question">
    <h3 id="-questions-3"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

    <p>Compare the output we got here with the output of the simple assemblies obtained in the introductory tutorial.</p>
    <ol>
      <li>What are the main differences between them?</li>
      <li>Did SPAdes produce a better assembly than the Velvet Optimiser?</li>
    </ol>
  </blockquote>

</blockquote>


        
        <blockquote class="key_points">
            <h3><i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>We learned about how the choice of k-mer size will affect assembly outcomes</li>
                
                <li>We learned about the strategies that assemblers use to make reference genomes</li>
                
                <li>We performed a number of assemblies with Velvet and SPAdes.</li>
                
                <li>You should use SPAdes or another more modern assembler than Velvet for actual assemblies now.</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Useful information regarding this type of analysis with descriptions and paper references for the tools used in this tutorial, and literature for this analysis techniques and interpretations can be found <a href="/training-material/topics/assembly#references">here</a>.</p>
        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Feedback</h3>
            Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial.
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Simon Gladman)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/assembly/tutorials/debruijn-graph-assembly/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
