<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Data manipulation with Pandas / Data manipulation with Pandas / Foundations of Data Science</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@43973919866b066986c8bceb7a497d54a19de378",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Data manipulation with Pandas">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-02-20 16:10:07 +0000">
<meta name="DC.creator" content="Anton Nekrutenko"><meta name="description" content="These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Foundations of Data Science / Data manipulation with Pandas / Hands-on: Data manipulation with Pandas">
        <meta property="og:description" content="These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/data-science/tutorials/gnmx-lecture6/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Data manipulation with Pandas",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Draft",
  "dateModified": "2024-02-20 16:10:07 +0000",
  "datePublished": "2024-02-20 06:18:29 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00419",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Foundations of Data Science",
    "description": "These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.",
    "url": "https://training.galaxyproject.org/training-material/topics/data-science/"
  },
  "abstract": "#  SARS-CoV-2 as a research problem",
  "learningResourceType": "e-learning",
  "name": "Data manipulation with Pandas",
  "url": "https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html",
  "version": 2,
  "timeRequired": "PT1H",
  "teaches": "- Understand data manipulation in Pandas",
  "keywords": [
    "Foundations of Data Science",
    "jupyter-notebook"
  ],
  "description": "## Abstract\n\n#  SARS-CoV-2 as a research problem\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - What is Pandas\n - What is a dataframe\n - How to access data in dataframes\n\n\n## Learning Objectives\n\n- Understand data manipulation in Pandas\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/nekrut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/nekrut/",
      "name": "Anton Nekrutenko",
      "image": "https://avatars.githubusercontent.com/nekrut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "name": "Björn Grüning",
      "image": "https://avatars.githubusercontent.com/bgruening",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eurosciencegateway/",
          "name": "EuroScienceGateway",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://galaxyproject.org/projects/esg/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eosc-life/",
          "name": "EOSC-Life",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.eosc-life.eu"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/deNBI/",
          "name": "de.NBI",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.denbi.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/sfb992/",
          "name": "Collaborative Research Centre 992",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.sfb992.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-converge/",
          "name": "ELIXIR-CONVERGE",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org/about-us/how-funded/eu-projects/converge"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-3079-6586",
      "orcid": "https://orcid.org/0000-0002-3079-6586"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Foundations of Data Science",
      "description": "These lessons will help you get your feet in data science and give you tools to help you slice and dice your data into results.",
      "url": "https://training.galaxyproject.org/training-material/topics/data-science/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3372",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3372",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3372"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/data-science-gnmx-lecture6.ipynb",
      "name": "Jupyter Notebook (with Solutions)"
    },
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/data-science-gnmx-lecture6-course.ipynb",
      "name": "Jupyter Notebook (without Solutions)"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/data-science" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Foundations of Data Science
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/data-science/faqs/" title="Check our FAQs for the Foundations of Data Science topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/data-science/tutorials/gnmx-lecture6/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-data-science">
    <h1 data-toc-skip>Data manipulation with Pandas</h1>
    
    <div class="alert alert-warning" role="alert">
        <h4 class="alert-heading">Under Development!</h4>
        <p>This tutorial is not in its final state. The content may change a lot in the next months.
        Because of this status, it is also not listed in the topic pages.</p>
    </div>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			
  <a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge contributor-nekrut"><img src="https://avatars.githubusercontent.com/nekrut?s=36" alt="Anton Nekrutenko avatar" width="36" class="avatar" />
    Anton Nekrutenko</a>
  
		</td>
	</tr>
	

	

	

	

	

	

	

	
	<tr class="reviewers">
		<td><abbr title="These people reviewed this material for accuracy and correctness">Reviewers</abbr></td>
		<td>
			
  <a href="/training-material/hall-of-fame/bgruening/" class="contributor-badge contributor-bgruening">
    <img src="https://avatars.githubusercontent.com/bgruening?s=36" alt="Björn Grüning avatar" width="36" class="avatar" />
  </a>
  </td>
	</tr>
	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What is Pandas</p>
</li>
        
        <li><p>What is a dataframe</p>
</li>
        
        <li><p>How to access data in dataframes</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Understand data manipulation in Pandas</p>
</li>
        
        </ul>

        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            

            

            

            

            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/data-science/tutorials/gnmx-lecture6/data-science-gnmx-lecture6.ipynb" title="Jupyter Notebook" alt="Data manipulation with Pandas Notebook">
        <i class="far fa-file-code" aria-hidden="true"></i> Jupyter Notebook
    </a>
    

</li>
            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr" title="">
			UseGalaxy.fr
		</a>
	</li>
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Feb 20, 2024 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Feb 20, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00419">https://gxy.io/GTN:T00419</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 2 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 
                <section aria-label="Tutorial Notebook access information" id="tutorial-notebook">
                     
                     
                     
                     
                    
                        <blockquote class="comment">
                            <div class="box-title">Best viewed in a Jupyter Notebook</div>
                            <p>
                                This tutorial is best viewed in a Jupyter notebook! You can load this notebook one of the following ways<!--, or you can click to run it in MyBinder-->
                            </p>
                            <!--
                            <p><b>Launching on MyBinder</b>
                                Click here and the notebook will launch on MyBinder.org
                                <a href="https://mybinder.org/v2/gh/galaxyproject/training-material/main?filepath=training-material%2Ftopics%2Fdata-science%2Ftutorials%2Fgnmx-lecture6%2Ftutorial.md.ipynb" title="Launch notebook on MyBinder"><img src="https://mybinder.org/badge_logo.svg" alt="My Binder Logo"/></a>
                            </p>
                            -->
                            
                            <p><b>Run on the GTN with JupyterLite (in-browser computations)</b>
                                <ol>
                                    <li><a href="/training-material/jupyter/lab/index.html?path=data-science-gnmx-lecture6.ipynb">Click to Launch JupyterLite</a></li>
                                </ol>
                            </p>
                            
                            <p><b>Launching the notebook in Jupyter in Galaxy</b>
                                <ol>
                                    <li><a href="/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html">Instructions to Launch JupyterLab</a></li>
                                    <li>Open a Terminal in JupyterLab with <b>File -> New -> Terminal</b></li>
                                    <li>Run <code>wget https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/data-science-gnmx-lecture6.ipynb</code></li>
                                    <li>Select the notebook that appears in the list of files on the left.</li>
                                </ol>
                            </p>
                            <p><b>Downloading the notebook</b>
                                <ol>
                                    <li>Right click one of these links: <a href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/data-science-gnmx-lecture6.ipynb">Jupyter Notebook (With Solutions)</a>, <a href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/data-science-gnmx-lecture6-course.ipynb">Jupyter Notebook (Without Solutions)</a></li>
                                    <li>Save Link As..</li>
                                </ol>
                            </p>
                        </blockquote>
                    
                </section>
                

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="sars-cov-2-as-a-research-problem">SARS-CoV-2 as a research problem</h1>

<p>To learn about pandas we will use SARS-CoV-2 data. Before actually jumping to Pandas let’s learn about the coronavirus molecular biology.</p>

<p>The following summary is based on these publications:</p>

<ul>
  <li><a href="http://dx.doi.org/10.1016/S0065-3527(06)66005-3">Masters:2006</a></li>
  <li><a href="http://dx.doi.org/10.1007/978-1-4939-2438-7_1">Fehr and Perlman:2015</a></li>
  <li><a href="https://www.annualreviews.org/doi/full/10.1146/annurev-virology-100114-055218">Sola:2015</a></li>
  <li><a href="http://dx.doi.org/10.1038/nature17200">Kirchdoerfer:2016</a></li>
  <li><a href="http://dx.doi.org/10.1016/j.cell.2020.02.058">Walls:2020</a></li>
  <li><a href="https://www.nature.com/articles/s41580-021-00418-x">Jackson:2022</a></li>
</ul>

<h2 id="genome-organization">Genome organization</h2>

<p>All coronaviruses contain non-segmented positive-strand RNA genome approx. 30 kb in length. It is invariably 5’-leader-UTR-replicase-S-E-M-N-3’UTR-poly(A). In addition,
it contains a variety of accessory proteins interspersed throughout the genome (see Fig. below; (From <a href="http://dx.doi.org/10.1007/978-1-4939-2438-7_1">Fehr and Perlman:2015</a>)).</p>

<figure id="figure-1" style="max-width: 90%;"><img src="https://media.springernature.com/original/springer-static/image/chp%3A10.1007%2F978-1-4939-2438-7_1/MediaObjects/317916_1_En_1_Fig1_HTML.gif" alt="Genome organozation. " loading="lazy" /><a target="_blank" href="https://media.springernature.com/original/springer-static/image/chp%3A10.1007%2F978-1-4939-2438-7_1/MediaObjects/317916_1_En_1_Fig1_HTML.gif" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Genomic organization of representative α, β, and γ CoVs. An illustration of the MHV genome is depicted at the top. The expanded regions below show the structural and accessory proteins in the 3′ regions of the HCoV-229E, MHV, SARS-CoV, MERS-CoV and IBV. Size of the genome and individual genes are approximated using the legend at the top of the diagram but are not drawn to scale. HCoV-229E human coronavirus 229E, MHV mouse hepatitis virus, SARS-CoV severe acute respiratory syndrome coronavirus, MERS-CoV Middle East respiratory syndrome coronavirus, IBV infectious bronchitis virus.</figcaption></figure>

<p>The long replicase gene encodes a number of non-structural proteins (nsps) and occupies 2/3 of the genome. Because of -1 ribosomal frameshifting within ORFab approx. in 25% of the cases
polyprotein pp1ab is produced in place of pp1a. pp1a encodes 11 nsps while pp1ab encodes 15 (Fig. below from <a href="https://www.annualreviews.org/doi/full/10.1146/annurev-virology-100114-055218">Sola:2015</a>).</p>

<figure id="figure-2" style="max-width: 90%;"><img src="https://www.annualreviews.org/na101/home/literatum/publisher/ar/journals/content/virology/2015/virology.2015.2.issue-1/annurev-virology-100114-055218/20151024/images/large/vi20265.f1.jpeg" alt="Discontinuous transcription. " loading="lazy" /><a target="_blank" href="https://www.annualreviews.org/na101/home/literatum/publisher/ar/journals/content/virology/2015/virology.2015.2.issue-1/annurev-virology-100114-055218/20151024/images/large/vi20265.f1.jpeg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Coronavirus genome structure and gene expression. (a) Coronavirus genome structure. The upper scheme represents the TGEV genome. Labels indicate gene names; L corresponds to the leader sequence. Also represented are the nsps derived from the processing of the pp1a and pp1ab polyproteins. PLP1, PLP2, and 3CL protease sites are depicted as inverted triangles with the corresponding color code of each protease. Dark gray rectangles represent transmembrane domains, and light gray rectangles indicate other functional domains. (b) Coronavirus genome strategy of sgmRNA expression. The upper scheme represents the TGEV genome. Short lines represent the nested set of sgmRNAs, each containing a common leader sequence (black) and a specific gene to be translated (dark gray). (c) Key elements in coronavirus transcription. A TRS precedes each gene (TRS-B) and includes the core sequence (CS-B) and variable 5′ and 3′ flanking sequences. The TRS of the leader (TRS-L), containing the core sequence (CS-L), is present at the 5′ end of the genome, in an exposed location (orange box in the TRS-L loop). Discontinuous transcription occurs during the synthesis of the negative-strand RNA (light blue), when the copy of the TRS-B hybridizes with the TRS-L. Dotted lines indicate the complementarity between positive-strand and negative-strand RNA sequences. Abbreviations: EndoU, endonuclease; ExoN, exonuclease; HEL, helicase; MTase, methyltransferase (green, N7-methyltransferase; dark purple, 2′-O-methyltransferase); nsp, nonstructural protein; PLP, papain-like protease; RdRp, RNA-dependent RNA polymerase; sgmRNA, subgenomic RNA; TGEV, transmissible gastroenteritis virus; TRS, transcription-regulating sequence; UTR, untranslated region.</figcaption></figure>

<h2 id="virion-structure">Virion structure</h2>

<p>Coronavirus is a spherical particle approx. 125 nm in diameter. It is covered with S-protein projections giving it an appearance of solar corona - hence the term coronavirus. Inside the envelope is a nucleocapsid that has helical symmetry far more typical of (-)-strand RNA viruses. There are four main structure proteins: spike (S), membrane (M), envelope (E), and nucleocapsid (N; Fig. below <a href="http://dx.doi.org/10.1016/S0065-3527(06)66005-3">Masters:2006</a>):</p>

<figure id="figure-3" style="max-width: 90%;"><img src="https://ars.els-cdn.com/content/image/1-s2.0-S0065352706660053-gr1.jpg" alt="Virion structure. " loading="lazy" /><a target="_blank" href="https://ars.els-cdn.com/content/image/1-s2.0-S0065352706660053-gr1.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Schematic of the coronavirus virion, with the minimal set of structural proteins</figcaption></figure>

<p>Mature S protein is a trimer of two subunits: S1 and S2. The two subunits are produced from a single S-precursor by host proteases (see <a href="http://dx.doi.org/10.1038/nature17200">Kirchdoerfer:2016</a>; this however is not the case for all coronaviruses such as SARS-CoV). S1 forms the receptor-binding domain, while S2 forms the stalk.</p>

<p>M protein is the most abundant structural component of the virion and determines its shape. It possibly exists as a dimer.</p>

<p>E protein is the least abundant protein of the capsid and possesses ion channel activity. It facilitates the assembly and release of the virus. In SARS-CoV it is not required for replication but is essential for pathogenesis.</p>

<p>N proteins form the nucleocapsid. Its N- and C-terminal domains are capable of RNA binding. Specifically, it binds to transcription regulatory sequences and the genomic packaging signal. It also binds to <code class="language-plaintext highlighter-rouge">nsp3</code>
and M protein possible tethering viral genome and replicase-transcriptase complex.</p>

<h2 id="entering-the-cell">Entering the cell</h2>

<p>Virion attaches to the cells as a result of interaction between S-protein and a cellular receptor. In the case of SARS-CoV-2 (as is in SARS-CoV) angiotensin converting enzyme 2 (ACE2) serves as the receptor binding to C-terminal portion of S1 domain. After receptor binding S protein is cleaved at two sites within the S2 subdomain. The first cleavage separates the receptor-binding domain of S from the membrane fusion domain. The second cleavage (at S2’) exposes the fusion peptide. The SARS-CoV-2 is different from SARS-CoV in that it contains a <a href="https://en.wikipedia.org/wiki/Furin">furin</a> cleavage site that is processed during viral maturation within endoplasmatic reticulum (ER; see <a href="http://dx.doi.org/10.1016/j.cell.2020.02.058">Walls:2020</a>). The fusion peptide inserts into the cellular membrane and triggers joining on the two heptad regions with S2 forming an antiparallel six-helix bundle resulting in fusion and release of the genome into the cytoplasm (Fig. below from <a href="https://www.nature.com/articles/s41580-021-00418-x">Jackson:2022</a>.</p>

<figure id="figure-4" style="max-width: 90%;"><img src="https://i.imgur.com/DD7EO3G.png" alt="Entering the cell. " loading="lazy" /><a target="_blank" href="https://i.imgur.com/DD7EO3G.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Two spike (S) protein cleavage events are typically necessary for the coronavirus entry process: one at the junction of the S1 and S2 subunits and the other at the S2′ site, internal to the S2 subunit. In the case of SARS-CoV-2, the polybasic sequence at the S1–S2 boundary is cleaved during virus maturation in an infected cell, but the S2′ site is cleaved at the target cell following angiotensin-converting enzyme 2 (ACE2) binding. Virus binding to ACE2 (step 1) induces conformational changes in the S1 subunit and exposes the S2′ cleavage site in the S2 subunit. Depending on the entry route taken by SARS-CoV-2, the S2′ site is cleaved by different proteases. Left: If the target cell expresses insufficient transmembrane protease, serine 2 (TMPRSS2) or if a virus–ACE2 complex does not encounter TMPRSS2, the virus–ACE2 complex is internalized via clathrin-mediated endocytosis (step 2) into the endolysosomes, where S2′ cleavage is performed by cathepsins, which require an acidic environment for their activity (steps 3 and 4). Right: In the presence of TMPRSS2, S2′ cleavage occurs at the cell surface (step 2). In both entry pathways, cleavage of the S2′ site exposes the fusion peptide (FP) and dissociation of S1 from S2 induces dramatic conformational changes in the S2 subunit, especially in heptad repeat 1, propelling the fusion peptide forward into the target membrane, initiating membrane fusion (step 5 on the left and step 3 on the right). Fusion between viral and cellular membranes forms a fusion pore through which viral RNA is released into the host cell cytoplasm for uncoating and replication (step 6 on the left and step 4 on the right). Several agents disrupt interaction between the S protein and ACE2: ACE2 mimetics, therapeutic monoclonal antibodies targeting the neutralizing epitopes on the S protein and antibodies elicited by vaccination block virus binding to ACE2 and thus inhibit both entry pathways. By contrast, strategies targeting post-receptor-binding steps differ between the two pathways. Being a serine protease inhibitor, camostat mesylate restricts the TMPRSS2-mediated entry pathway. Hydroxychloroquine and chloroquine block endosomal acidification, which is necessary for cathepsin activity, and thus restrict the cathepsin-mediated entry pathway</figcaption></figure>

<h2 id="replication">Replication</h2>

<p>The above figure shows that in addition to the full length (+)-strand genome there is a number of (+)-strand subgenomic RNAs corresponding to 3’-end of the complete viral sequence. All of these <em>subgenomic</em> RNAs (sgRNAs) share the same leader sequence that is present only once at the extreme 5’-end of the viral genome. These RNAs are produced via discontinuous RNA synthesis when the RNA-dependent RNA-polymerase (RdRp) switches template:</p>

<iframe title="vimeo-player" src="https://player.vimeo.com/video/133088346" width="640" height="480" frameborder="0" allowfullscreen=""></iframe>

<p>☝ Model for the formation of genome high-order structures regulating N gene transcription. The upper linear scheme represents the coronavirus genome. The red line indicates the leader sequence in the 5′ end of the genome. The hairpin indicates the TRS-L. The gray line with arrowheads represents the nascent negative-sense RNA. The curved blue arrow indicates the template switch to the leader sequence during discontinuous transcription. The orange line represents the copy of the leader added to the nascent RNA after the template switch. The RNA-RNA interactions between the pE (nucleotides 26894 to 26903) and dE (nucleotides 26454 to 26463) and between the B-M in the active domain (nucleotides 26412 to 26421) and the cB-M in the 5′ end of the genome (nucleotides 477 to 486) are represented by solid lines. Dotted lines indicate the complementarity between positive-strand and negative-strand RNA sequences. Abbreviations: AD, active domain secondary structure prediction; B-M, B motif; cB-M, complementary copy of the B-M; cCS-N, complementary copy of the CS-N; CS-L, conserved core sequence of the leader; CS-N, conserved core sequence of the N gene; dE, distal element; pE, proximal element; TRS-L, transcription-regulating sequence of the leader (From <a href="https://www.annualreviews.org/doi/full/10.1146/annurev-virology-100114-055218">Sola:2015</a>).</p>

<p>Furthermore, <a href="https://www.annualreviews.org/doi/full/10.1146/annurev-virology-100114-055218">Sola:2015</a> suggests the coronavirus transcription model in which transcription initiation complex forms at the 3’-end of (+)-strand genomic RNA:</p>

<figure id="figure-5" style="max-width: 90%;"><img src="https://www.annualreviews.org/na101/home/literatum/publisher/ar/journals/content/virology/2015/virology.2015.2.issue-1/annurev-virology-100114-055218/20151024/images/large/vi20265.f3.jpeg" alt="Coronavirus transcription. " loading="lazy" /><a target="_blank" href="https://www.annualreviews.org/na101/home/literatum/publisher/ar/journals/content/virology/2015/virology.2015.2.issue-1/annurev-virology-100114-055218/20151024/images/large/vi20265.f3.jpeg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Three-step model of coronavirus transcription. (1) Complex formation. Proteins binding transcription-regulating sequences are represented by ellipsoids, the leader sequence is indicated with a red bar, and core sequences are indicated with orange boxes. (2) Base-pairing scanning. Negative-strand RNA is shown in light blue; the transcription complex is represented by a hexagon. Vertical lines indicate complementarity between the genomic RNA and the nascent negative strand. (3) Template switch. Due to the complementarity between the newly synthesized negative-strand RNA and the transcription-regulating sequence of the leader, template switch to the leader is made by the transcription complex to complete the copy of the leader sequence</figcaption></figure>

<h2 id="how-virus-is-screened">How virus is screened</h2>

<p>Two approaches: (1) RNAseq and (2) amplicons</p>

<p>RNAseq approach requires a large sample quantity, but is suitable for detecting low-frequency variants. Ampliconic approaches are useful for samples containing a small fraction of viral RNA and are
by far the most widely used (From <a href="https://primer-monitor.neb.com/lineages">NEB</a>):</p>

<figure id="figure-6" style="max-width: 90%;"><img src="https://i.imgur.com/9wLeHnZ.png" alt="Primer schemes. " loading="lazy" /><a target="_blank" href="https://i.imgur.com/9wLeHnZ.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Nucleotide variants related to lineages of concern are indicated in yellow above a schematic SARS-CoV-2 genome. Primers that do not overlap with variants associated with these lineages are shown in blue. Overlapping primers are indicated in orange.</figcaption></figure>

<h1 id="pandas">Pandas!</h1>

<blockquote class="quote">
  <p>This is an aggregated tutorial relying on material from the following fantastic sources:</p>
  <ul>
    <li><a href="http://justinbois.github.io/bootcamp/2020/index.html">Justin Bois</a>. It contains modified training datasets and adopts content to Colab environment.</li>
    <li><a href="https://people.duke.edu/~ccc14/bios-821-2017/index.html">BIOS821 course at Duke</a></li>
    <li><a href="https://pandas.pydata.org/docs/user_guide/index.html/">Pandas documentation</a></li>
  </ul>
</blockquote>

<p>Pandas (from “Panel Data”) is an essential piece of scientific (and not only) data analysis infrastructure. It is, in essence, a highly optimized library for manipulating very large tables (or “Data Frames”).</p>

<p>Today we will be using a single notebook that explains the basics of this powerful tool:</p>

<h2 id="pandas-learning-resources">Pandas learning resources</h2>

<ul>
  <li><a href="https://pandas.pydata.org/docs/getting_started/index.html#getting-started">Getting started</a> - official introduction from Pandas.</li>
  <li><a href="http://people.duke.edu/~ccc14/bios-821-2017/index.html">Data Science Tools</a> - Data Science for Biologists from Duke University.</li>
  <li><a href="https://datacarpentry.org/">Data Carpentry</a> - a collection of lessons <em>à la</em> Software Carpentry.</li>
</ul>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Pandas, conventionally imported as pd
</span><span class="kn">import</span> <span class="n">pandas</span> <span class="k">as</span> <span class="n">pd</span>
</code></pre></div></div>

<p>Throughout your research career, you will undoubtedly need to handle data, possibly lots of data. The data comes in lots of formats, and you will spend much of your time <strong>wrangling</strong> the data to get it into a usable form.</p>

<p>Pandas is the primary tool in the Python ecosystem for handling data. Its primary object, the <code class="language-plaintext highlighter-rouge">DataFrame</code> is extremely useful in wrangling data. We will explore some of that functionality here and will put it to use in the next lesson.</p>

<h1 id="basics">Basics</h1>

<h2 id="the-data-set">The data set</h2>

<p>The dataset we will be using is a subset of metadata describing SARS-CoV-2 datasets from the <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive</a>.</p>

<p>It is obtained by going to https://www.ncbi.nlm.nih.gov/sra and performing a query with the following search terms: <code class="language-plaintext highlighter-rouge">txid2697049[Organism:noexp]</code>.</p>

<p>Results are downloaded using <code class="language-plaintext highlighter-rouge">Send to:</code> menu selecting <code class="language-plaintext highlighter-rouge">File</code> and then <code class="language-plaintext highlighter-rouge">RunInfo</code>. Let’s get these results into this notebook:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="err">!</span><span class="n">wget</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">zenodo</span><span class="p">.</span><span class="n">org</span><span class="o">/</span><span class="n">records</span><span class="o">/</span><span class="mi">10680001</span><span class="o">/</span><span class="n">files</span><span class="o">/</span><span class="n">sra_ncov</span><span class="p">.</span><span class="n">csv</span><span class="p">.</span><span class="n">gz</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="err">!</span><span class="n">gunzip</span> <span class="o">-</span><span class="n">c</span>  <span class="n">sra_ncov</span><span class="p">.</span><span class="n">csv</span><span class="p">.</span><span class="n">gz</span> <span class="o">|</span> <span class="n">head</span>
</code></pre></div></div>

<p>The first line contains the <strong>headers</strong> for each column. The data follow. Given the file I/O skills you recently learned, you could write some functions to parse this file and extract the data you want.
You can imagine that this might be kind of painful. However, if the file format is nice and clean like we more or less have here, we can use pre-built tools. Pandas has a very powerful function, <code class="language-plaintext highlighter-rouge">pd.read_csv()</code>
that can read in a CSV file and store the contents in a convenient data structure called a <strong>data frame</strong>. In Pandas, the data type for a data frame is <code class="language-plaintext highlighter-rouge">DataFrame</code>, and we will use “data frame” and “<code class="language-plaintext highlighter-rouge">DataFrame</code>” interchangeably.</p>

<h2 id="reading-in-data">Reading in data</h2>

<p>Let’s first look at the doc string of <code class="language-plaintext highlighter-rouge">pd.read_csv()</code>.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pd</span><span class="p">.</span><span class="n">read_csv</span><span class="err">?</span>
</code></pre></div></div>

<p>Holy cow! There are so many options we can specify for reading in a CSV file. You will likely find reasons to use many of these throughout your research. For now, however, we do not need most of them.
So, let’s load in the data set. Note that even though the dataset is compressed with <code class="language-plaintext highlighter-rouge">gzip</code> we do not need to do anything additional - pandas magically understands and uncompresses the data while loading it into the dataframe.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_csv</span><span class="p">(</span><span class="sh">'</span><span class="s">sra_ncov.csv.gz</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<p>The same result can be achieved directly without downloading the file first:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_csv</span><span class="p">(</span><span class="sh">'</span><span class="s">https://zenodo.org/records/10680001/files/sra_ncov.csv.gz</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<p>We now have the data stored in a data frame. We can look at it in the Jupyter Notebook since Jupyter will display it in a well-organized, pretty way. Note that because our dataframe is big,
we only display the first five rows using <code class="language-plaintext highlighter-rouge">head()</code> function:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="nf">head</span><span class="p">()</span>
</code></pre></div></div>

<h2 id="indexing-data-frames">Indexing data frames</h2>

<p>The data frame is a convenient data structure for many reasons that will become clear as we start exploring. Let’s start by looking at how data frames are indexed. Let’s try to look at the first row.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
</code></pre></div></div>

<p>Yikes! Lots of errors. The problem is that we tried to index numerically by row. <strong>We index DataFrames, by columns.</strong> And no column has the name <code class="language-plaintext highlighter-rouge">0</code> in this data frame, though there could be.
Instead, you might want to look at the column with the percentage of correct face-matching tasks.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">].</span><span class="nf">head</span><span class="p">()</span>
</code></pre></div></div>

<p>This gave us the numbers we were after. Notice that when it was printed, the index of the rows came along with it. If we wanted to pull out a single percentage correct, say corresponding to index <code class="language-plaintext highlighter-rouge">4</code>, we can do that.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">][</span><span class="mi">4</span><span class="p">]</span>
</code></pre></div></div>

<p>However, this is <strong>not</strong> the preferred way to do this. It is better to use <code class="language-plaintext highlighter-rouge">.loc</code>. This gives the location in the data frame we want.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-tt-loc-tt-versus-tt-iloc-tt-1"><button class="gtn-boxify-button details" type="button" aria-controls="details-tt-loc-tt-versus-tt-iloc-tt-1" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: <tt>loc</tt> versus <tt>iloc</tt></span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><code class="language-plaintext highlighter-rouge">loc</code> and <code class="language-plaintext highlighter-rouge">iloc</code> are both methods in the Pandas library for DataFrame indexing.</p>

  <p><code class="language-plaintext highlighter-rouge">loc</code>: It is label-based indexing, meaning that you use the actual row and column labels to make selections. This means that you specify rows and columns based on their index labels.
For example, df.loc[3, ‘column_name’] will select the value in the third row and the column labeled ‘column_name’.</p>

  <p><code class="language-plaintext highlighter-rouge">iloc</code>: It is integer-based indexing, where you use the integer position of the rows and columns to make selections. This means that you specify rows and columns based on their integer position.
For example, df.iloc[3, 2] will select the value in the fourth row and the third column (remembering that indexing starts at 0).</p>

  <p>In summary, <code class="language-plaintext highlighter-rouge">loc</code> uses labels for indexing, while <code class="language-plaintext highlighter-rouge">iloc</code> uses integer positions.</p>
</blockquote>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">4</span><span class="p">,</span> <span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">]</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">4</span><span class="p">:</span><span class="mi">6</span><span class="p">]</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">4</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="p">[</span><span class="mi">0</span><span class="p">,</span><span class="mi">2</span><span class="p">,</span><span class="mi">4</span><span class="p">]]</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">4</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="p">[</span><span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">LibraryStrategy</span><span class="sh">'</span><span class="p">]]</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">4</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">:]</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="mi">4</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">:</span><span class="sh">'</span><span class="s">LibrarySource</span><span class="sh">'</span><span class="p">]</span>
</code></pre></div></div>

<p>It is also important to note that <strong>row indices need not be integers</strong>. And you should not count on them being integers. In practice, you will almost never use row indices, but rather use <strong>Boolean indexing</strong>.</p>

<h2 id="filtering-boolean-indexing-of-data-frames">Filtering: Boolean indexing of data frames</h2>

<p>Let’s say I wanted to pull out accession numbers of runs produced by Pacific Biosciences machines (in this table such datasets are labeled as <code class="language-plaintext highlighter-rouge">PACBIO_SMRT</code>. I can use Boolean indexing to specify the row. Specifically, I want the row for which <code class="language-plaintext highlighter-rouge">df['Platform'] == 'PACBIO_SMRT'</code>. You can essentially plop this syntax directly when using <code class="language-plaintext highlighter-rouge">.loc</code>.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">PACBIO_SMRT</span><span class="sh">'</span><span class="p">,</span> <span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">]</span>
</code></pre></div></div>

<p>If I want to pull the whole record for that participant, I can use <code class="language-plaintext highlighter-rouge">:</code> for the column index.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">PACBIO_SMRT</span><span class="sh">'</span><span class="p">,</span> <span class="p">:].</span><span class="nf">head</span><span class="p">(</span><span class="mi">10</span><span class="p">)</span>
</code></pre></div></div>

<p>Now, let’s pull out all PacBio records that were obtained from Amplicon sequencing. We can again use Boolean indexing, but we need to use an <code class="language-plaintext highlighter-rouge">&amp;</code> operator.
We have not covered this bitwise operator before, but the syntax is self-explanatory in the example below. Note that it is important that each Boolean operation you are doing is in parentheses because of the precedence of the operators involved.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[(</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">PACBIO_SMRT</span><span class="sh">'</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">LibraryStrategy</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">AMPLICON</span><span class="sh">'</span><span class="p">),</span> <span class="p">:].</span><span class="nf">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</code></pre></div></div>

<p>What is going on will be clearer if we set up our Boolean indexing first, as follows.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">inds</span> <span class="o">=</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">PACBIO_SMRT</span><span class="sh">'</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">LibraryStrategy</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">AMPLICON</span><span class="sh">'</span><span class="p">)</span>
<span class="n">inds</span><span class="p">[</span><span class="mi">2</span><span class="p">:</span><span class="mi">6</span><span class="p">]</span>
</code></pre></div></div>

<p>Notice that <code class="language-plaintext highlighter-rouge">inds</code> is an array (actually a Pandas <code class="language-plaintext highlighter-rouge">Series</code>, essentially a <code class="language-plaintext highlighter-rouge">DataFrame</code> with one column) of <code class="language-plaintext highlighter-rouge">True</code>s and <code class="language-plaintext highlighter-rouge">False</code>s. We can apply the <code class="language-plaintext highlighter-rouge">unique</code> function from <a href="https://numpy.org/">NumPy</a> to see how
many <code class="language-plaintext highlighter-rouge">True</code> and <code class="language-plaintext highlighter-rouge">False</code> rows we have:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">numpy</span> <span class="k">as</span> <span class="n">np</span>
<span class="n">np</span><span class="p">.</span><span class="nf">unique</span><span class="p">(</span><span class="n">inds</span><span class="p">,</span> <span class="n">return_counts</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>
</code></pre></div></div>

<p>When we index with it using .loc, we get back rows where inds is True:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[</span><span class="n">inds</span><span class="p">,</span> <span class="p">:]</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="n">loc</span><span class="p">[(</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">PACBIO_SMRT</span><span class="sh">'</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">LibraryStrategy</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="sh">'</span><span class="s">AMPLICON</span><span class="sh">'</span><span class="p">),</span> <span class="p">:].</span><span class="nf">head</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
</code></pre></div></div>

<h2 id="calculating-with-data-frames">Calculating with data frames</h2>

<p>The SRA data contains (sometimes) size of the run in MB (<code class="language-plaintext highlighter-rouge">size_MB</code>). Let’s suppose we want to consider only those run where <code class="language-plaintext highlighter-rouge">size_MB</code> is above 100. We might like to add a column to the data frame that specifies whether or not the corresponding run is above 100Mb. We can conveniently compute with columns. This is done element-wise.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="mi">100</span>
</code></pre></div></div>

<p>This tells us which run is above 100 Mb. We can simply add this back to the data frame.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1"># Add the column to the DataFrame
</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Over100Mb</span><span class="sh">'</span><span class="p">]</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="mi">100</span>

<span class="c1"># Take a look
</span><span class="n">df</span><span class="p">.</span><span class="nf">head</span><span class="p">()</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="nf">assign</span><span class="p">(</span><span class="n">over100</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">]</span><span class="o">&gt;=</span><span class="mi">100</span><span class="p">)</span>
</code></pre></div></div>

<h2 id="a-note-of-assign">A note of <code class="language-plaintext highlighter-rouge">assign</code></h2>

<p>The <code class="language-plaintext highlighter-rouge">assign</code> method in the pandas library is used to add new columns to a pandas DataFrame. The method allows you to perform calculations or operations on existing columns to generate new columns without modifying the original DataFrame.</p>

<p>For example, you can use the <code class="language-plaintext highlighter-rouge">assign</code> method to add a new column that calculates the sum of two existing columns or to add a column based on a complex calculation that involves multiple columns. The method also supports adding columns based on external data sources, such as NumPy arrays or other pandas DataFrames.</p>

<p>Using <code class="language-plaintext highlighter-rouge">assign</code> method is useful because it allows you to create new columns and add them to a DataFrame in a way that is both readable and easy to maintain. The method is also chain-able, meaning you can add multiple columns in one call, making your code more concise and efficient.</p>

<h2 id="a-note-about-vectorization">A note about vectorization</h2>

<p>Notice how applying the <code class="language-plaintext highlighter-rouge">&lt;=</code> operator to a <code class="language-plaintext highlighter-rouge">Series</code> resulted in <strong>elementwise</strong> application. This is called <code class="language-plaintext highlighter-rouge">vectorization</code>. It means that we do not have to write a <code class="language-plaintext highlighter-rouge">for</code> loop to do operations on the elements of a <code class="language-plaintext highlighter-rouge">Series</code> or other array-like object. Imagine if we had to do that with a <code class="language-plaintext highlighter-rouge">for</code> loop.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">big_runs</span> <span class="o">=</span> <span class="p">[]</span>
<span class="k">for</span> <span class="n">size</span> <span class="ow">in</span> <span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">]:</span>
    <span class="n">big_runs</span><span class="p">.</span><span class="nf">append</span><span class="p">(</span><span class="n">size</span> <span class="o">&gt;=</span> <span class="mi">100</span><span class="p">)</span>
</code></pre></div></div>

<p>This is cumbersome. The vectorization allows for much more convenient calculation. Beyond that, the vectorized code is almost always faster when using Pandas and Numpy because the looping is done with compiled code under the hood. This can be done with many operators, including those you’ve already seen, like <code class="language-plaintext highlighter-rouge">+</code>, <code class="language-plaintext highlighter-rouge">-</code>, <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">/</code>, <code class="language-plaintext highlighter-rouge">**</code>, etc.</p>

<h2 id="outputting-a-new-csv-file">Outputting a new CSV file</h2>

<p>Now that we added the <code class="language-plaintext highlighter-rouge">Over100Mb</code> column, we might like to save our data frame as a new CSV that we can reload later. We use <code class="language-plaintext highlighter-rouge">df.to_csv()</code> for this with the <code class="language-plaintext highlighter-rouge">index</code> kwarg to ask Pandas not to explicitly write the indices to the file.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="nf">to_csv</span><span class="p">(</span><span class="sh">'</span><span class="s">over100Mb_data.csv</span><span class="sh">'</span><span class="p">,</span> <span class="n">index</span><span class="o">=</span><span class="bp">False</span><span class="p">)</span>
</code></pre></div></div>

<p>Let’s take a look at what this file looks like.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="err">!</span><span class="n">head</span> <span class="n">over100Mb_data</span><span class="p">.</span><span class="n">csv</span>
</code></pre></div></div>

<h1 id="tidy-data">Tidy data</h1>

<p><a href="https://en.wikipedia.org/wiki/Hadley_Wickham">Hadley Wickham</a> wrote a <a href="http://dx.doi.org/10.18637/jss.v059.i10">great article</a> in favor of “tidy data.” Tidy data frames follow the rules:</p>

<ol>
  <li>Each variable is a column.</li>
  <li>Each observation is a row.</li>
  <li>Each type of observation has its own separate data frame.</li>
</ol>

<p>This is less pretty to visualize as a table, but we rarely look at data in tables. Indeed, the representation of data that is convenient for visualization is different from that which is convenient for analysis.
A tidy data frame is almost always <strong>much</strong> easier to work with than non-tidy formats.</p>

<p>Also, let’s take a look at this <a href="https://dtkaplan.github.io/DataComputingEbook/chap-tidy-data.html#chap:tidy-data">article</a>.</p>

<h2 id="the-data-set-1">The data set</h2>

<p>The dataset we will be using is a list of all SARS-CoV-2 datasets in <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive</a> as of January 20, 2021.</p>

<p>It is obtained by going to https://www.ncbi.nlm.nih.gov/sra and performing a query with the following search terms: <code class="language-plaintext highlighter-rouge">txid2697049[Organism:noexp]</code>.</p>

<p>Results are downloaded using <code class="language-plaintext highlighter-rouge">Send to:</code> menu selecting <code class="language-plaintext highlighter-rouge">File</code> and then <code class="language-plaintext highlighter-rouge">RunInfo</code>. Let’s get these results into this notebook:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_csv</span><span class="p">(</span><span class="sh">'</span><span class="s">https://zenodo.org/records/10680001/files/sra_ncov.csv.gz</span><span class="sh">'</span><span class="p">)</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">df</span><span class="p">[</span><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">]</span><span class="o">&gt;</span> <span class="mi">0</span><span class="p">].</span><span class="nf">reset_index</span><span class="p">(</span><span class="n">drop</span><span class="o">=</span><span class="bp">True</span><span class="p">)</span>

<span class="c1"># Take a look
</span><span class="n">df</span>
</code></pre></div></div>

<p>This data set is in tidy format. Each row represents a single SRA dataset. The properties of each run are given in each column. We already saw the power of having the data in this format when we did Boolean indexing in the last lesson.</p>

<h2 id="finding-unique-values-and-counts">Finding unique values and counts</h2>

<p>How many unique sequencing platforms do we have?</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">].</span><span class="nf">unique</span><span class="p">()</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">[</span><span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">].</span><span class="nf">value_counts</span><span class="p">()</span>
</code></pre></div></div>

<h2 id="sorting">Sorting</h2>

<p>(and axes!)</p>

<p>Let’s start by sorting on index:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span> <span class="o">=</span> <span class="n">df</span><span class="p">.</span><span class="nf">sample</span><span class="p">(</span><span class="n">n</span><span class="o">=</span><span class="mi">10</span><span class="p">)</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span><span class="p">.</span><span class="nf">sort_index</span><span class="p">()</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span><span class="p">.</span><span class="nf">sort_index</span><span class="p">(</span><span class="n">axis</span> <span class="o">=</span> <span class="mi">1</span><span class="p">)</span>
</code></pre></div></div>

<p>Now let’s try sorting by values:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span><span class="p">.</span><span class="nf">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="sh">'</span><span class="s">LibraryLayout</span><span class="sh">'</span><span class="p">])</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span><span class="p">.</span><span class="nf">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="sh">'</span><span class="s">LibraryLayout</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">])</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_subset</span><span class="p">.</span><span class="nf">sort_values</span><span class="p">(</span><span class="n">by</span><span class="o">=</span><span class="p">[</span><span class="sh">'</span><span class="s">LibraryLayout</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">],</span><span class="n">ascending</span><span class="o">=</span><span class="p">[</span><span class="bp">True</span><span class="p">,</span><span class="bp">False</span><span class="p">])</span>
</code></pre></div></div>

<h1 id="split-apply-combine">Split-apply-combine</h1>

<p>Let’s say we want to compute the total size of SRA runs for each <code class="language-plaintext highlighter-rouge">BioProject</code>. Ignoring for the second the mechanics of how we would do this with Pandas, let’s think about it in English. What do we need to do?</p>

<ol>
  <li><strong>Split</strong> the data set up according to the <code class="language-plaintext highlighter-rouge">'BioProject'</code> field, i.e., split it up so we have a separate data set for each BioProject ID.</li>
  <li><strong>Apply</strong> a median function to the activity in these split data sets.</li>
  <li><strong>Combine</strong> the results of these averages on the split data set into a new, summary data set that contains classes for each platform and medians for each.</li>
</ol>

<p>We see that the strategy we want is a <strong>split-apply-combine</strong> strategy. This idea was put forward by Hadley Wickham in <a href="http://dx.doi.org/10.18637/jss.v040.i01">this paper</a>. It turns out that this is a strategy we want to use <em>very</em> often. Split the data in terms of some criterion. Apply some function to the split-up data. Combine the results into a new data frame.</p>

<p>Note that if the data are tidy, this procedure makes a lot of sense. Choose the column you want to use to split by. All rows with like entries in the splitting column are then grouped into a new data set. You can then apply any function you want into these new data sets. You can then combine the results into a new data frame.</p>

<p>Pandas’s split-apply-combine operations are achieved using the <code class="language-plaintext highlighter-rouge">groupby()</code> method. You can think of <code class="language-plaintext highlighter-rouge">groupby()</code> as the splitting part. You can then apply functions to the resulting <code class="language-plaintext highlighter-rouge">DataFrameGroupBy</code> object. The <a href="http://pandas.pydata.org/pandas-docs/stable/groupby.html">Pandas documentation on split-apply-combine</a> is excellent and worth reading through. It is extensive though, so don’t let yourself get intimidated by it.</p>

<h2 id="aggregation">Aggregation</h2>

<p>Let’s go ahead and do our first split-apply-combine on this tidy data set. First, we will split the data set up by <code class="language-plaintext highlighter-rouge">BioProject</code>.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">grouped</span> <span class="o">=</span> <span class="n">df</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">BioProject</span><span class="sh">'</span><span class="p">])</span>

<span class="c1"># Take a look
</span><span class="n">grouped</span>
</code></pre></div></div>

<p>There is not much to see in the <code class="language-plaintext highlighter-rouge">DataFrameGroupBy</code> object that resulted. But there is a lot we can do with this object. Typing <code class="language-plaintext highlighter-rouge">grouped.</code> and hitting tab will show you the many possibilities. For most of these possibilities, the apply and combine steps happen together and a new data frame is returned. The <code class="language-plaintext highlighter-rouge">grouped.sum()</code> method is exactly what we want.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_sum</span> <span class="o">=</span> <span class="n">grouped</span><span class="p">.</span><span class="nf">sum</span><span class="p">()</span>

<span class="c1"># Take a look
</span><span class="n">df_sum</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_sum</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nc">DataFrame</span><span class="p">(</span><span class="n">grouped</span><span class="p">[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">].</span><span class="nf">sum</span><span class="p">())</span>
<span class="n">df_sum</span>
</code></pre></div></div>

<p>The outputted data frame has the sums of numerical columns only, which we have only one: <code class="language-plaintext highlighter-rouge">size_MS</code>. Note that this data frame has <code class="language-plaintext highlighter-rouge">Platform</code> as the name of the row index.
If we want to instead keep <code class="language-plaintext highlighter-rouge">Platform</code> (which, remember, is what we used to split up the data set before we computed the summary statistics) as a column, we can use the <code class="language-plaintext highlighter-rouge">reset_index()</code> method.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_sum</span><span class="p">.</span><span class="nf">reset_index</span><span class="p">()</span>
</code></pre></div></div>

<p>Note, though, that this was not done in-place. If you want to update your data frame, you have to explicitly do so.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df_sum</span> <span class="o">=</span> <span class="n">df_sum</span><span class="p">.</span><span class="nf">reset_index</span><span class="p">()</span>
</code></pre></div></div>

<p>We can also use multiple columns in our <code class="language-plaintext highlighter-rouge">groupby()</code> operation. To do this, we simply pass in a list of columns into <code class="language-plaintext highlighter-rouge">df.groupby()</code>. We will <strong>chain the methods</strong>, performing a groupby,
applying a median, and then resetting the index of the result, all in one line.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">BioProject</span><span class="sh">'</span><span class="p">,</span> <span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]).</span><span class="nf">sum</span><span class="p">().</span><span class="nf">reset_index</span><span class="p">()</span>
</code></pre></div></div>

<p>This type of operation is called an <strong>aggregation</strong>. That is, we split the data set up into groups, and then computed a summary statistic for each group, in this case the median.</p>

<p>You now have tremendous power in your hands. When your data are tidy, you can rapidly accelerate the ubiquitous split-apply-combine methods. Importantly, you now have a logical
framework to think about how you slice and dice your data. As a final, simple example, I will show you how to go start to finish after loading the data set into a data frame,
splitting by <code class="language-plaintext highlighter-rouge">BioProject</code> and <code class="language-plaintext highlighter-rouge">Platform</code>, and then getting the quartiles and extrema, in addition to the mean and standard deviation.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">BioProject</span><span class="sh">'</span><span class="p">,</span> <span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">])[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">].</span><span class="nf">describe</span><span class="p">()</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">BioProject</span><span class="sh">'</span><span class="p">,</span> <span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">])[</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">].</span><span class="nf">describe</span><span class="p">().</span><span class="nf">reset_index</span><span class="p">()</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">numpy</span> <span class="k">as</span> <span class="n">np</span>
<span class="n">df</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">BioProject</span><span class="sh">'</span><span class="p">,</span> <span class="sh">'</span><span class="s">Platform</span><span class="sh">'</span><span class="p">]).</span><span class="nf">agg</span><span class="p">({</span><span class="sh">'</span><span class="s">size_MB</span><span class="sh">'</span><span class="p">:</span><span class="n">np</span><span class="p">.</span><span class="n">mean</span><span class="p">,</span> <span class="sh">'</span><span class="s">Run</span><span class="sh">'</span><span class="p">:</span><span class="sh">'</span><span class="s">nunique</span><span class="sh">'</span><span class="p">})</span>
</code></pre></div></div>

<p>Yes, that’s right. One single, clean, easy-to-read line of code. In the coming tutorials, we will see how to use tidy data to quickly generate plots.</p>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-quotes-versus-no-quotes-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-quotes-versus-no-quotes-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Quotes versus no quotes?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><a href="https://stackoverflow.com/questions/66443260/why-are-some-pandas-aggregation-functions-in-quotes-and-others-not">Why <code class="language-plaintext highlighter-rouge">np.mean</code> is without quotes and <code class="language-plaintext highlighter-rouge">nunique</code> is with quotes?</a></p>
</blockquote>

<h2 id="tidying-a-data-set">Tidying a data set</h2>

<p>You should always organize your data sets in a tidy format. However, this is sometimes just not possible, since your data sets can come from instruments that do not output the
data in tidy format (though most do, at least in my experience), and you often have collaborators that send data in untidy formats.</p>

<p>The most useful function for tidying data is <code class="language-plaintext highlighter-rouge">pd.melt()</code>. To demonstrate this we will use a dataset describing read coverage across SARS-CoV-2 genomes for several samples.</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_csv</span><span class="p">(</span><span class="sh">'</span><span class="s">https://zenodo.org/records/10680470/files/coverage.tsv.gz</span><span class="sh">'</span><span class="p">,</span><span class="n">sep</span><span class="o">=</span><span class="sh">'</span><span class="se">\t</span><span class="sh">'</span><span class="p">)</span>

<span class="n">df</span><span class="p">.</span><span class="nf">head</span><span class="p">()</span>
</code></pre></div></div>

<p>Clearly, these data are not tidy. When we melt the data frame, the data within it, called <strong>values</strong>, become a single column. The headers, called <strong>variables</strong>, also become new columns.
So, to melt it, we need to specify what we want to call the values and what we want to call the variable. <a href="https://pandas.pydata.org/docs/reference/api/pandas.melt.html#pandas.melt"><code class="language-plaintext highlighter-rouge">pd.melt()</code></a> does the rest!</p>

<p><a href="https://pandas.pydata.org/docs/_images/07_melt.svg" rel="noopener noreferrer"><img src="https://pandas.pydata.org/docs/_images/07_melt.svg" alt="Dataframe melt. " loading="lazy" /></a></p>

<blockquote>
  <p>Image from <a href="https://pandas.pydata.org/docs/getting_started/intro_tutorials/07_reshape_table_layout.html#wide-to-long-format">Pandas Docs</a>.</p>
</blockquote>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">melted</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">melt</span><span class="p">(</span><span class="n">df</span><span class="p">,</span> <span class="n">value_name</span><span class="o">=</span><span class="sh">'</span><span class="s">coverage</span><span class="sh">'</span><span class="p">,</span> <span class="n">var_name</span><span class="o">=</span><span class="p">[</span><span class="sh">'</span><span class="s">sample</span><span class="sh">'</span><span class="p">],</span><span class="n">value_vars</span><span class="o">=</span><span class="n">df</span><span class="p">.</span><span class="n">columns</span><span class="p">[</span><span class="mi">3</span><span class="p">:],</span><span class="n">id_vars</span><span class="o">=</span><span class="p">[</span><span class="sh">'</span><span class="s">start</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">end</span><span class="sh">'</span><span class="p">])</span>

<span class="n">melted</span><span class="p">.</span><span class="nf">head</span><span class="p">()</span>
</code></pre></div></div>

<p>…and we are good to do with a tidy DataFrame! Let’s take a look at the summary. This would allow us to easily plot coverage:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">seaborn</span> <span class="k">as</span> <span class="n">sns</span>
<span class="n">sns</span><span class="p">.</span><span class="nf">relplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">melted</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="sh">'</span><span class="s">start</span><span class="sh">'</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="sh">'</span><span class="s">coverage</span><span class="sh">'</span><span class="p">,</span><span class="n">kind</span><span class="o">=</span><span class="sh">'</span><span class="s">line</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sns</span><span class="p">.</span><span class="nf">relplot</span><span class="p">(</span><span class="n">data</span><span class="o">=</span><span class="n">melted</span><span class="p">,</span> <span class="n">x</span><span class="o">=</span><span class="sh">'</span><span class="s">start</span><span class="sh">'</span><span class="p">,</span><span class="n">y</span><span class="o">=</span><span class="sh">'</span><span class="s">coverage</span><span class="sh">'</span><span class="p">,</span><span class="n">kind</span><span class="o">=</span><span class="sh">'</span><span class="s">line</span><span class="sh">'</span><span class="p">,</span><span class="n">hue</span><span class="o">=</span><span class="sh">'</span><span class="s">sample</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">melted</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">sample</span><span class="sh">'</span><span class="p">]).</span><span class="nf">describe</span><span class="p">()</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">melted</span><span class="p">.</span><span class="nf">groupby</span><span class="p">([</span><span class="sh">'</span><span class="s">sample</span><span class="sh">'</span><span class="p">])[</span><span class="sh">'</span><span class="s">coverage</span><span class="sh">'</span><span class="p">].</span><span class="nf">describe</span><span class="p">()</span>
</code></pre></div></div>

<p>To get back from melted (narrow) format to wide format we can use <a href="https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.pivot.html#pandas.DataFrame.pivot"><code class="language-plaintext highlighter-rouge">pivot()</code></a> function.</p>

<p><a href="https://pandas.pydata.org/docs/_images/07_pivot.svg" rel="noopener noreferrer"><img src="https://pandas.pydata.org/docs/_images/07_pivot.svg" alt="Dataframe pivot. " loading="lazy" /></a></p>

<blockquote>
  <p>Image from <a href="https://pandas.pydata.org/docs/getting_started/intro_tutorials/07_reshape_table_layout.html#long-to-wide-table-format">Pandas Docs</a>.</p>
</blockquote>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">melted</span><span class="p">.</span><span class="nf">pivot</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="p">[</span><span class="sh">'</span><span class="s">start</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">end</span><span class="sh">'</span><span class="p">],</span><span class="n">columns</span><span class="o">=</span><span class="sh">'</span><span class="s">sample</span><span class="sh">'</span><span class="p">,</span><span class="n">values</span><span class="o">=</span><span class="sh">'</span><span class="s">coverage</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<h1 id="working-with-multiple-tables">Working with multiple tables</h1>

<p>Working with multiple tables often involves joining them on a common key:</p>

<p><a href="https://pandas.pydata.org/docs/_images/08_merge_left.svg" rel="noopener noreferrer"><img src="https://pandas.pydata.org/docs/_images/08_merge_left.svg" alt="Left join. " loading="lazy" /></a></p>

<p>In fact, this can be done in several different ways described below. But first, let’s define two simple dataframes:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="err">!</span><span class="n">pip</span> <span class="n">install</span> <span class="o">--</span><span class="n">upgrade</span> <span class="n">pandasql</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">from</span> <span class="n">pandasql</span> <span class="kn">import</span> <span class="n">sqldf</span>
<span class="n">pysqldf</span> <span class="o">=</span> <span class="k">lambda</span> <span class="n">q</span><span class="p">:</span> <span class="nf">sqldf</span><span class="p">(</span><span class="n">q</span><span class="p">,</span> <span class="nf">globals</span><span class="p">())</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df1</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nc">DataFrame</span><span class="p">({</span><span class="sh">"</span><span class="s">key</span><span class="sh">"</span><span class="p">:</span> <span class="p">[</span><span class="sh">"</span><span class="s">A</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">B</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">C</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">D</span><span class="sh">"</span><span class="p">],</span> <span class="sh">"</span><span class="s">value</span><span class="sh">"</span><span class="p">:</span> <span class="n">np</span><span class="p">.</span><span class="n">random</span><span class="p">.</span><span class="nf">randn</span><span class="p">(</span><span class="mi">4</span><span class="p">)})</span>
<span class="n">df2</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nc">DataFrame</span><span class="p">({</span><span class="sh">"</span><span class="s">key</span><span class="sh">"</span><span class="p">:</span> <span class="p">[</span><span class="sh">"</span><span class="s">B</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">D</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">D</span><span class="sh">"</span><span class="p">,</span> <span class="sh">"</span><span class="s">E</span><span class="sh">"</span><span class="p">],</span> <span class="sh">"</span><span class="s">value</span><span class="sh">"</span><span class="p">:</span> <span class="n">np</span><span class="p">.</span><span class="n">random</span><span class="p">.</span><span class="nf">randn</span><span class="p">(</span><span class="mi">4</span><span class="p">)})</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df1</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">df2</span>
</code></pre></div></div>

<h2 id="inner-join">Inner join</h2>

<p><a href="https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/SQL_Join_-_07_A_Inner_Join_B.svg/234px-SQL_Join_-_07_A_Inner_Join_B.svg.png?20170204165143" rel="noopener noreferrer"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/1/18/SQL_Join_-_07_A_Inner_Join_B.svg/234px-SQL_Join_-_07_A_Inner_Join_B.svg.png?20170204165143" alt="Inner join. " loading="lazy" /></a></p>

<blockquote>
  <p>Figure from Wikimedia Commons</p>
</blockquote>

<p>Using pandas <code class="language-plaintext highlighter-rouge">merge</code>:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pd</span><span class="p">.</span><span class="nf">merge</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="sh">"</span><span class="s">key</span><span class="sh">"</span><span class="p">)</span>
</code></pre></div></div>

<p>Using <code class="language-plaintext highlighter-rouge">pysqldf</code>:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">pysqldf</span><span class="p">(</span><span class="sh">'</span><span class="s">select * from df1 join df2 on df1.key=df2.key</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<h2 id="left-join">Left join</h2>

<p><a href="https://upload.wikimedia.org/wikipedia/commons/thumb/d/dc/SQL_Join_-_01b_A_Left_Join_B.svg/234px-SQL_Join_-_01b_A_Left_Join_B.svg.png?20170204144906" rel="noopener noreferrer"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/d/dc/SQL_Join_-_01b_A_Left_Join_B.svg/234px-SQL_Join_-_01b_A_Left_Join_B.svg.png?20170204144906" alt="Left join again. " loading="lazy" /></a></p>

<blockquote>
  <p>Figure from Wikimedia Commons</p>
</blockquote>

<p>Using pandas <code class="language-plaintext highlighter-rouge">merge</code>:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pd</span><span class="p">.</span><span class="nf">merge</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="sh">"</span><span class="s">key</span><span class="sh">"</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="sh">"</span><span class="s">left</span><span class="sh">"</span><span class="p">).</span><span class="nf">fillna</span><span class="p">(</span><span class="sh">'</span><span class="s">.</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<p>Using <code class="language-plaintext highlighter-rouge">pysqldf</code>:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">pysqldf</span><span class="p">(</span><span class="sh">'</span><span class="s">select * from df1 left join df2 on df1.key=df2.key</span><span class="sh">'</span><span class="p">).</span><span class="nf">fillna</span><span class="p">(</span><span class="sh">'</span><span class="s">.</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">pysqldf</span><span class="p">(</span><span class="sh">'</span><span class="s">select df1.key, df1.value as value_x, df2.value as value_y from df1 left join df2 on df1.key=df2.key</span><span class="sh">'</span><span class="p">).</span><span class="nf">fillna</span><span class="p">(</span><span class="sh">'</span><span class="s">.</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<h2 id="right-join">Right join</h2>

<p><a href="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5f/SQL_Join_-_03_A_Right_Join_B.svg/234px-SQL_Join_-_03_A_Right_Join_B.svg.png?20170130230641" rel="noopener noreferrer"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5f/SQL_Join_-_03_A_Right_Join_B.svg/234px-SQL_Join_-_03_A_Right_Join_B.svg.png?20170130230641" alt="Right join. " loading="lazy" /></a></p>

<blockquote>
  <p>Figure from Wikimedia Commons</p>
</blockquote>

<p>Using pandas <code class="language-plaintext highlighter-rouge">merge</code>:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pd</span><span class="p">.</span><span class="nf">merge</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="sh">"</span><span class="s">key</span><span class="sh">"</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="sh">"</span><span class="s">right</span><span class="sh">"</span><span class="p">).</span><span class="nf">fillna</span><span class="p">(</span><span class="sh">'</span><span class="s">.</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<h2 id="full-join">Full join</h2>

<p><a href="https://upload.wikimedia.org/wikipedia/commons/thumb/6/61/SQL_Join_-_05_A_Full_Join_B.svg/234px-SQL_Join_-_05_A_Full_Join_B.svg.png?20170130230643" rel="noopener noreferrer"><img src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/61/SQL_Join_-_05_A_Full_Join_B.svg/234px-SQL_Join_-_05_A_Full_Join_B.svg.png?20170130230643" alt="Full join. " loading="lazy" /></a></p>

<blockquote>
  <p>Figure from Wikimedia Commons</p>
</blockquote>

<p>Using pandas <code class="language-plaintext highlighter-rouge">merge</code>:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">pd</span><span class="p">.</span><span class="nf">merge</span><span class="p">(</span><span class="n">df1</span><span class="p">,</span> <span class="n">df2</span><span class="p">,</span> <span class="n">on</span><span class="o">=</span><span class="sh">"</span><span class="s">key</span><span class="sh">"</span><span class="p">,</span> <span class="n">how</span><span class="o">=</span><span class="sh">"</span><span class="s">outer</span><span class="sh">"</span><span class="p">).</span><span class="nf">fillna</span><span class="p">(</span><span class="sh">'</span><span class="s">.</span><span class="sh">'</span><span class="p">)</span>
</code></pre></div></div>

<h1 id="putting-it-all-together-pandas--altair">Putting it all together: Pandas + Altair</h1>

<h2 id="understanding-altair">Understanding <a href="https://altair-viz.github.io/">Altair</a></h2>

<p>Vega-Altair is a declarative statistical visualization library for Python, based on Vega and Vega-Lite. It offers a powerful and concise grammar that enables you to quickly build a wide range of statistical visualizations. <a href="https://uwdata.github.io/visualization-curriculum/intro.html">This site</a> contains a complete set of “how-to” tutorials explaining all aspects of this remarkable package.</p>

<h2 id="importing-data">Importing data</h2>

<p>First, import all the packages we need:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="kn">import</span> <span class="n">pandas</span> <span class="k">as</span> <span class="n">pd</span>
<span class="kn">import</span> <span class="n">altair</span> <span class="k">as</span> <span class="n">alt</span>
<span class="kn">from</span> <span class="n">datetime</span> <span class="kn">import</span> <span class="n">date</span>
<span class="n">today</span> <span class="o">=</span> <span class="n">date</span><span class="p">.</span><span class="nf">today</span><span class="p">()</span>
</code></pre></div></div>

<p>Next, read a gigantic dataset from DropBox:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sra</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">read_csv</span><span class="p">(</span>
    <span class="sh">"</span><span class="s">https://zenodo.org/records/10680776/files/ena.tsv.gz</span><span class="sh">"</span><span class="p">,</span>
    <span class="n">compression</span><span class="o">=</span><span class="sh">'</span><span class="s">gzip</span><span class="sh">'</span><span class="p">,</span>
    <span class="n">sep</span><span class="o">=</span><span class="sh">"</span><span class="se">\t</span><span class="sh">"</span><span class="p">,</span>
    <span class="n">low_memory</span><span class="o">=</span><span class="bp">False</span>
<span class="p">)</span>
</code></pre></div></div>

<p>This dataset contains <em>a lot</em> of rows:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">len</span><span class="p">(</span><span class="n">sra</span><span class="p">)</span>
</code></pre></div></div>
<p>Which would output:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>1000000
</code></pre></div></div>

<p>Let’s look at the five random lines from this table (scroll sideways):</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sra</span><span class="p">.</span><span class="nf">sample</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
</code></pre></div></div>

<table>
  <thead>
    <tr>
      <th style="text-align: right"> </th>
      <th style="text-align: left">study_accession</th>
      <th style="text-align: left">run_accession</th>
      <th style="text-align: left">collection_date</th>
      <th style="text-align: left">instrument_platform</th>
      <th style="text-align: left">library_strategy</th>
      <th style="text-align: left">library_construction_protocol</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: right">2470282</td>
      <td style="text-align: left">PRJEB37886</td>
      <td style="text-align: left">ERR8618165</td>
      <td style="text-align: left">2022-02-03 00:00:00</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: left">nan</td>
    </tr>
    <tr>
      <td style="text-align: right">5747352</td>
      <td style="text-align: left">PRJNA686984</td>
      <td style="text-align: left">SRR21294506</td>
      <td style="text-align: left">2022-07-10 00:00:00</td>
      <td style="text-align: left">OXFORD_NANOPORE</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: left">nan</td>
    </tr>
    <tr>
      <td style="text-align: right">2588443</td>
      <td style="text-align: left">PRJEB37886</td>
      <td style="text-align: left">ERR8997118</td>
      <td style="text-align: left">2022-02-15 00:00:00</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: left">nan</td>
    </tr>
    <tr>
      <td style="text-align: right">3662323</td>
      <td style="text-align: left">PRJNA716984</td>
      <td style="text-align: left">SRR16033134</td>
      <td style="text-align: left">2021-09-08 00:00:00</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: left">Freed primers</td>
    </tr>
    <tr>
      <td style="text-align: right">5298351</td>
      <td style="text-align: left">PRJNA716984</td>
      <td style="text-align: left">SRR20455932</td>
      <td style="text-align: left">2022-04-10 00:00:00</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: left">Freed primers</td>
    </tr>
  </tbody>
</table>

<p>This dataset also has <em>a lot</em> of columns:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="k">for</span> <span class="n">_</span> <span class="ow">in</span> <span class="n">sra</span><span class="p">.</span><span class="n">columns</span><span class="p">:</span> <span class="nf">print</span><span class="p">(</span><span class="n">_</span><span class="p">)</span>
</code></pre></div></div>

<p>We do not need all the columns, so let’s restrict the dataset only to columns we would need. This would also make it much smaller:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sra</span> <span class="o">=</span> <span class="n">sra</span><span class="p">[</span>
    <span class="p">[</span> 
        <span class="sh">'</span><span class="s">study_accession</span><span class="sh">'</span><span class="p">,</span>
        <span class="sh">'</span><span class="s">run_accession</span><span class="sh">'</span><span class="p">,</span>
        <span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">,</span>
        <span class="sh">'</span><span class="s">instrument_platform</span><span class="sh">'</span><span class="p">,</span>
        <span class="sh">'</span><span class="s">library_strategy</span><span class="sh">'</span><span class="p">,</span>
        <span class="sh">'</span><span class="s">library_construction_protocol</span><span class="sh">'</span> 
    <span class="p">]</span>
<span class="p">]</span>
</code></pre></div></div>

<h2 id="cleaning-the-data">Cleaning the data</h2>

<p>The <code class="language-plaintext highlighter-rouge">collection_date</code> field will be useful for us to be able to filter out nonsense as you will see below. But to use it properly, we need tell Pandas that it is not just a text, but actually dates:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sra</span> <span class="o">=</span> <span class="n">sra</span><span class="p">.</span><span class="nf">assign</span><span class="p">(</span><span class="n">collection_date</span> <span class="o">=</span> <span class="n">pd</span><span class="p">.</span><span class="nf">to_datetime</span><span class="p">(</span><span class="n">sra</span><span class="p">[</span><span class="sh">"</span><span class="s">collection_date</span><span class="sh">"</span><span class="p">]))</span>
</code></pre></div></div>

<p>Let’s see what are the earliest and latest entries:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nf">print</span><span class="p">(</span><span class="sh">'</span><span class="s">Earliest entry:</span><span class="sh">'</span><span class="p">,</span> <span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">].</span><span class="nf">min</span><span class="p">())</span>
<span class="nf">print</span><span class="p">(</span><span class="sh">'</span><span class="s">Latest entry:</span><span class="sh">'</span><span class="p">,</span> <span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">].</span><span class="nf">max</span><span class="p">())</span>
</code></pre></div></div>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-metadata-is-128169-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Metadata is &#128169;</div>
  <p>😱 | Don’t get surprised here - the metadata is only as good as the person who entered it. So, <strong>when you enter metadata for you sequencing data – pay attention!!!</strong></p>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Can you write a statement that would show how many rows contain these erroneous dates?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sra</span><span class="p">[</span><span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">].</span><span class="nf">min</span><span class="p">()][</span><span class="sh">'</span><span class="s">run_accession</span><span class="sh">'</span><span class="p">].</span><span class="nf">nunique</span><span class="p">()</span>
<span class="n">sra</span><span class="p">[</span><span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">]</span> <span class="o">==</span> <span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">].</span><span class="nf">max</span><span class="p">()][</span><span class="sh">'</span><span class="s">run_accession</span><span class="sh">'</span><span class="p">].</span><span class="nf">nunique</span><span class="p">()</span>
</code></pre></div>    </div>
  </blockquote>
</blockquote>

<p>The data will likely need a bit of cleaning:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">sra</span> <span class="o">=</span> <span class="n">sra</span><span class="p">[</span> 
    <span class="p">(</span> <span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">]</span> <span class="o">&gt;=</span> <span class="sh">'</span><span class="s">2020-01-01</span><span class="sh">'</span> <span class="p">)</span> 
    <span class="o">&amp;</span> 
    <span class="p">(</span> <span class="n">sra</span><span class="p">[</span><span class="sh">'</span><span class="s">collection_date</span><span class="sh">'</span><span class="p">]</span> <span class="o">&lt;=</span> <span class="sh">'</span><span class="s">2023-02-16</span><span class="sh">'</span> <span class="p">)</span> 
<span class="p">]</span>
</code></pre></div></div>

<p>Finally, in order to build the heatmap, we need to aggregate the data:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">heatmap_2d</span> <span class="o">=</span> <span class="n">sra</span><span class="p">.</span><span class="nf">groupby</span><span class="p">(</span>
    <span class="p">[</span><span class="sh">'</span><span class="s">instrument_platform</span><span class="sh">'</span><span class="p">,</span><span class="sh">'</span><span class="s">library_strategy</span><span class="sh">'</span><span class="p">]</span>
<span class="p">).</span><span class="nf">agg</span><span class="p">(</span>
    <span class="p">{</span><span class="sh">'</span><span class="s">run_accession</span><span class="sh">'</span><span class="p">:</span><span class="sh">'</span><span class="s">nunique</span><span class="sh">'</span><span class="p">}</span>
<span class="p">).</span><span class="nf">reset_index</span><span class="p">()</span>
</code></pre></div></div>

<p>This will look something like this:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: right"> </th>
      <th style="text-align: left">instrument_platform</th>
      <th style="text-align: left">library_strategy</th>
      <th style="text-align: right">run_accession</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: right">0</td>
      <td style="text-align: left">BGISEQ</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">21</td>
    </tr>
    <tr>
      <td style="text-align: right">1</td>
      <td style="text-align: left">BGISEQ</td>
      <td style="text-align: left">OTHER</td>
      <td style="text-align: right">1067</td>
    </tr>
    <tr>
      <td style="text-align: right">2</td>
      <td style="text-align: left">BGISEQ</td>
      <td style="text-align: left">RNA-Seq</td>
      <td style="text-align: right">64</td>
    </tr>
    <tr>
      <td style="text-align: right">3</td>
      <td style="text-align: left">BGISEQ</td>
      <td style="text-align: left">Targeted-Capture</td>
      <td style="text-align: right">38</td>
    </tr>
    <tr>
      <td style="text-align: right">4</td>
      <td style="text-align: left">BGISEQ</td>
      <td style="text-align: left">WGA</td>
      <td style="text-align: right">1</td>
    </tr>
    <tr>
      <td style="text-align: right">5</td>
      <td style="text-align: left">CAPILLARY</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">3</td>
    </tr>
    <tr>
      <td style="text-align: right">6</td>
      <td style="text-align: left">DNBSEQ</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">325</td>
    </tr>
    <tr>
      <td style="text-align: right">7</td>
      <td style="text-align: left">DNBSEQ</td>
      <td style="text-align: left">OTHER</td>
      <td style="text-align: right">5</td>
    </tr>
    <tr>
      <td style="text-align: right">8</td>
      <td style="text-align: left">DNBSEQ</td>
      <td style="text-align: left">RNA-Seq</td>
      <td style="text-align: right">64</td>
    </tr>
    <tr>
      <td style="text-align: right">9</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">4833110</td>
    </tr>
    <tr>
      <td style="text-align: right">10</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">OTHER</td>
      <td style="text-align: right">204</td>
    </tr>
    <tr>
      <td style="text-align: right">11</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">RNA-Seq</td>
      <td style="text-align: right">34826</td>
    </tr>
    <tr>
      <td style="text-align: right">12</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">Targeted-Capture</td>
      <td style="text-align: right">14843</td>
    </tr>
    <tr>
      <td style="text-align: right">13</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">WCS</td>
      <td style="text-align: right">74</td>
    </tr>
    <tr>
      <td style="text-align: right">14</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">WGA</td>
      <td style="text-align: right">89499</td>
    </tr>
    <tr>
      <td style="text-align: right">15</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">WGS</td>
      <td style="text-align: right">59618</td>
    </tr>
    <tr>
      <td style="text-align: right">16</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">WXS</td>
      <td style="text-align: right">2</td>
    </tr>
    <tr>
      <td style="text-align: right">17</td>
      <td style="text-align: left">ILLUMINA</td>
      <td style="text-align: left">miRNA-Seq</td>
      <td style="text-align: right">28</td>
    </tr>
    <tr>
      <td style="text-align: right">18</td>
      <td style="text-align: left">ION_TORRENT</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">108617</td>
    </tr>
    <tr>
      <td style="text-align: right">19</td>
      <td style="text-align: left">ION_TORRENT</td>
      <td style="text-align: left">RNA-Seq</td>
      <td style="text-align: right">75</td>
    </tr>
    <tr>
      <td style="text-align: right">20</td>
      <td style="text-align: left">ION_TORRENT</td>
      <td style="text-align: left">WGA</td>
      <td style="text-align: right">724</td>
    </tr>
    <tr>
      <td style="text-align: right">21</td>
      <td style="text-align: left">ION_TORRENT</td>
      <td style="text-align: left">WGS</td>
      <td style="text-align: right">1385</td>
    </tr>
    <tr>
      <td style="text-align: right">22</td>
      <td style="text-align: left">OXFORD_NANOPORE</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">445491</td>
    </tr>
    <tr>
      <td style="text-align: right">23</td>
      <td style="text-align: left">OXFORD_NANOPORE</td>
      <td style="text-align: left">OTHER</td>
      <td style="text-align: right">31</td>
    </tr>
    <tr>
      <td style="text-align: right">24</td>
      <td style="text-align: left">OXFORD_NANOPORE</td>
      <td style="text-align: left">RNA-Seq</td>
      <td style="text-align: right">18617</td>
    </tr>
    <tr>
      <td style="text-align: right">25</td>
      <td style="text-align: left">OXFORD_NANOPORE</td>
      <td style="text-align: left">WGA</td>
      <td style="text-align: right">7842</td>
    </tr>
    <tr>
      <td style="text-align: right">26</td>
      <td style="text-align: left">OXFORD_NANOPORE</td>
      <td style="text-align: left">WGS</td>
      <td style="text-align: right">13670</td>
    </tr>
    <tr>
      <td style="text-align: right">27</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">AMPLICON</td>
      <td style="text-align: right">533112</td>
    </tr>
    <tr>
      <td style="text-align: right">28</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">FL-cDNA</td>
      <td style="text-align: right">11</td>
    </tr>
    <tr>
      <td style="text-align: right">29</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">RNA-Seq</td>
      <td style="text-align: right">1013</td>
    </tr>
    <tr>
      <td style="text-align: right">30</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">Synthetic-Long-Read</td>
      <td style="text-align: right">29</td>
    </tr>
    <tr>
      <td style="text-align: right">31</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">Targeted-Capture</td>
      <td style="text-align: right">465</td>
    </tr>
    <tr>
      <td style="text-align: right">32</td>
      <td style="text-align: left">PACBIO_SMRT</td>
      <td style="text-align: left">WGS</td>
      <td style="text-align: right">48</td>
    </tr>
  </tbody>
</table>

<h2 id="plotting-the-data">Plotting the data</h2>

<p>Now let’s create a graph. This graph will be layered: the “back” will be the heatmap squares and the “front” will be the numbers (see heatmap at the beginning of this page):</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">back</span> <span class="o">=</span> <span class="n">alt</span><span class="p">.</span><span class="nc">Chart</span><span class="p">(</span><span class="n">heatmap_2d</span><span class="p">).</span><span class="nf">mark_rect</span><span class="p">(</span><span class="n">opacity</span><span class="o">=</span><span class="mi">1</span><span class="p">).</span><span class="nf">encode</span><span class="p">(</span>
    <span class="n">x</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nc">X</span><span class="p">(</span>
        <span class="sh">"</span><span class="s">instrument_platform:N</span><span class="sh">"</span><span class="p">,</span>
        <span class="n">title</span><span class="o">=</span><span class="sh">"</span><span class="s">Instrument</span><span class="sh">"</span>
    <span class="p">),</span>
    <span class="n">y</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nc">Y</span><span class="p">(</span>
        <span class="sh">"</span><span class="s">library_strategy:N</span><span class="sh">"</span><span class="p">,</span>
        <span class="n">title</span><span class="o">=</span><span class="sh">"</span><span class="s">Strategy</span><span class="sh">"</span><span class="p">,</span>
        <span class="n">axis</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nc">Axis</span><span class="p">(</span><span class="n">orient</span><span class="o">=</span><span class="sh">'</span><span class="s">right</span><span class="sh">'</span><span class="p">)</span>
    <span class="p">),</span>
    <span class="n">color</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nc">Color</span><span class="p">(</span>
        <span class="sh">"</span><span class="s">run_accession:Q</span><span class="sh">"</span><span class="p">,</span>
        <span class="n">title</span><span class="o">=</span><span class="sh">"</span><span class="s"># Samples</span><span class="sh">"</span><span class="p">,</span>
        <span class="n">scale</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nc">Scale</span><span class="p">(</span>
            <span class="n">scheme</span><span class="o">=</span><span class="sh">"</span><span class="s">goldred</span><span class="sh">"</span><span class="p">,</span>
            <span class="nb">type</span><span class="o">=</span><span class="sh">"</span><span class="s">log</span><span class="sh">"</span>
        <span class="p">),</span>
    <span class="p">),</span>
    <span class="n">tooltip</span><span class="o">=</span><span class="p">[</span>
        <span class="n">alt</span><span class="p">.</span><span class="nc">Tooltip</span><span class="p">(</span>
            <span class="sh">"</span><span class="s">instrument_platform:N</span><span class="sh">"</span><span class="p">,</span>
            <span class="n">title</span><span class="o">=</span><span class="sh">"</span><span class="s">Machine</span><span class="sh">"</span>
        <span class="p">),</span>
        <span class="n">alt</span><span class="p">.</span><span class="nc">Tooltip</span><span class="p">(</span>
            <span class="sh">"</span><span class="s">run_accession:Q</span><span class="sh">"</span><span class="p">,</span>
            <span class="n">title</span><span class="o">=</span><span class="sh">"</span><span class="s">Number of runs</span><span class="sh">"</span>
        <span class="p">),</span>
        <span class="n">alt</span><span class="p">.</span><span class="nc">Tooltip</span><span class="p">(</span>
            <span class="sh">"</span><span class="s">library_strategy:N</span><span class="sh">"</span><span class="p">,</span>
            <span class="n">title</span><span class="o">=</span><span class="sh">"</span><span class="s">Protocol</span><span class="sh">"</span>
        <span class="p">)</span>
    <span class="p">]</span>
<span class="p">).</span><span class="nf">properties</span><span class="p">(</span>
    <span class="n">width</span><span class="o">=</span><span class="mi">500</span><span class="p">,</span>
    <span class="n">height</span><span class="o">=</span><span class="mi">150</span><span class="p">,</span>
    <span class="n">title</span><span class="o">=</span><span class="p">{</span>
      <span class="sh">"</span><span class="s">text</span><span class="sh">"</span><span class="p">:</span> 
        <span class="p">[</span><span class="sh">"</span><span class="s">Breakdown of datasets (unique accessions) from ENA</span><span class="sh">"</span><span class="p">,</span>
         <span class="sh">"</span><span class="s">by Platform and Library Strategy</span><span class="sh">"</span><span class="p">],</span>
      <span class="sh">"</span><span class="s">subtitle</span><span class="sh">"</span><span class="p">:</span><span class="sh">"</span><span class="s">(Updated {})</span><span class="sh">"</span><span class="p">.</span><span class="nf">format</span><span class="p">(</span><span class="n">today</span><span class="p">.</span><span class="nf">strftime</span><span class="p">(</span><span class="sh">"</span><span class="s">%B %d, %Y</span><span class="sh">"</span><span class="p">))</span>
    <span class="p">}</span>
<span class="p">)</span>

<span class="n">back</span>
</code></pre></div></div>

<p>This would give us a grid:</p>

<p><a href="https://i.imgur.com/74itE5M.png" rel="noopener noreferrer"><img src="https://i.imgur.com/74itE5M.png" alt="Grid. " loading="lazy" /></a></p>

<p>Now, it would be nice to fill the rectangles with actual numbers:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">front</span> <span class="o">=</span> <span class="n">back</span><span class="p">.</span><span class="nf">mark_text</span><span class="p">(</span>
    <span class="n">align</span><span class="o">=</span><span class="sh">"</span><span class="s">center</span><span class="sh">"</span><span class="p">,</span>
    <span class="n">baseline</span><span class="o">=</span><span class="sh">"</span><span class="s">middle</span><span class="sh">"</span><span class="p">,</span>
    <span class="n">fontSize</span><span class="o">=</span><span class="mi">12</span><span class="p">,</span>
    <span class="n">fontWeight</span><span class="o">=</span><span class="sh">"</span><span class="s">bold</span><span class="sh">"</span><span class="p">,</span>
<span class="p">).</span><span class="nf">encode</span><span class="p">(</span>
    <span class="n">text</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nc">Text</span><span class="p">(</span><span class="sh">"</span><span class="s">run_accession:Q</span><span class="sh">"</span><span class="p">,</span><span class="nb">format</span><span class="o">=</span><span class="sh">"</span><span class="s">,.0f</span><span class="sh">"</span><span class="p">),</span>
    <span class="n">color</span><span class="o">=</span><span class="n">alt</span><span class="p">.</span><span class="nf">condition</span><span class="p">(</span>
        <span class="n">alt</span><span class="p">.</span><span class="n">datum</span><span class="p">.</span><span class="n">run_accession</span> <span class="o">&gt;</span> <span class="mi">200</span><span class="p">,</span>
        <span class="n">alt</span><span class="p">.</span><span class="nf">value</span><span class="p">(</span><span class="sh">"</span><span class="s">white</span><span class="sh">"</span><span class="p">),</span>
        <span class="n">alt</span><span class="p">.</span><span class="nf">value</span><span class="p">(</span><span class="sh">"</span><span class="s">black</span><span class="sh">"</span><span class="p">)</span>
    <span class="p">)</span>
<span class="p">)</span>

<span class="n">front</span>
</code></pre></div></div>

<p>This would give us the text:</p>

<p><a href="https://i.imgur.com/pKalDaC.png" rel="noopener noreferrer"><img src="https://i.imgur.com/pKalDaC.png" alt="Text for grid. " loading="lazy" /></a></p>

<p>To superimpose these on top of each other we should simply do this:</p>

<div class="language-py highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="n">back</span> <span class="o">+</span> <span class="n">front</span>
</code></pre></div></div>

<p><a href="https://i.imgur.com/SzhM1QW.png" rel="noopener noreferrer"><img src="https://i.imgur.com/SzhM1QW.png" alt="Full graph. " loading="lazy" /></a></p>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Pandas are an essential tool for data manipulation</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/data-science/faqs/">FAQ page for the Foundations of Data Science topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=data-science/gnmx-lecture6"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=data-science/gnmx-lecture6">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Anton Nekrutenko,  <b>Data manipulation with Pandas (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html">https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{data-science-gnmx-lecture6,
author = "Anton Nekrutenko",
	title = "Data manipulation with Pandas (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/data-science/tutorials/gnmx-lecture6/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/data-science/tutorials/gnmx-lecture6/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00419">gxy.io/GTN:T00419</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/data-science/tutorials/gnmx-lecture6/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/data-science/tutorials/gnmx-lecture6/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/43973919866b066986c8bceb7a497d54a19de378">4397391</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>