<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>16S Microbial analysis with Nanopore data</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:title" content="Galaxy Training: 16S Microbial analysis with Nanopore data" />
        <meta property="og:description" content="Metagenomics is a discipline that enables the genomic stu..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/metagenomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Metagenomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fmetagenomics%2Ftutorials%2Fnanopore-16S-metagenomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fmetagenomics%2Ftutorials%2Fnanopore-16S-metagenomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fmetagenomics%2Ftutorials%2Fnanopore-16S-metagenomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fmetagenomics%2Ftutorials%2Fnanopore-16S-metagenomics%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fmetagenomics%2Ftutorials%2Fnanopore-16S-metagenomics%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fmetagenomics%2Ftutorials%2Fnanopore-16S-metagenomics%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "16S Microbial analysis with Nanopore data",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metagenomics",
    "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
    "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/"
  },
  "courseCode": "metagenomics / nanopore-16S-metagenomics / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for '16S Microbial analysis with Nanopore data' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - How can we analyse the health status of the soil?\n - How do plants modify the composition of microbial communities?\n\n\\nThe objectives are:\n - Use Nanopore data for studying soil metagenomics\n - Analyze and preprocess Nanopore reads\n - Use Kraken2 to assign a taxonomic labels\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Crist√≥bal Gallardo"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Crist√≥bal Gallardo"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metagenomics",
      "description": "Metagenomics is a discipline that enables the genomic study of uncultured microorganisms",
      "url": "https://training.galaxyproject.org//training-material/topics/metagenomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3174",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3174",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3174"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>16S Microbial analysis with Nanopore data</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/gallardoalba" alt="Crist√≥bal Gallardo">Crist√≥bal Gallardo</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How can we analyse the health status of the soil?</p>
</li>
            
            <li><p>How do plants modify the composition of microbial communities?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Use Nanopore data for studying soil metagenomics</p>
</li>
            
            <li><p>Analyze and preprocess Nanopore reads</p>
</li>
            
            <li><p>Use Kraken2 to assign a taxonomic labels</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 2 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/metagenomics/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/4274812">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/workflows/" title="Workflows" alt="16S Microbial analysis with Nanopore data workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/metagenomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://nanopore.usegalaxy.eu/" title="NanoGalaxy">
            NanoGalaxy
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<!-- This is a comment. -->

<p><strong>Healthy soils</strong> are an essential element in maintaining the planet‚Äôs ecological balance. That is why their protection must be considered a priority in order to guarantee the well-being of humanity. The alteration of microbial populations often precedes changes in the physical and chemical properties of soils, so monitoring their condition can serve to predict their future evolution, allowing to develop strategies to <strong>mitigate ecosystem damage</strong>.</p>

<p>Advances in sequencing technologies have opened up the possibility of using the study of taxa present in bacterial communities as indicators of soil condition. For such an approach to be possible, variations in microbial populations need to be less affected by spatial factors than by human-derived alterations, which has been confirmed by various investigations (<a class="citation" href="#Hermans2016">Hermans <i>et al.</i> 2016</a>, <a class="citation" href="#Fierer2006">Fierer and Jackson 2006</a>).</p>

<p>Among the environmental variables that have shown the greatest correlation with alterations in the composition of bacterial communities are soil pH, the carbon-nitrogen ratio (C:N), and the co-concentrations of Olsen P (a measure of plant available phosphorus), aluminium and copper (Fig. 1).</p>

<figure id="figure-1"><img src="../../images/metagenomics-nanopore/bioindicators.jpg" alt="Bacterial bioindicators" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Relationship between bacterial community composition and soil parameters. The radius of each circle represents the taxon abundance that was accounted for by each soil variable, based on adjusted R-squared values from distance-based multivariate multiple regression analyses. The univariate relationship between the abundance of each taxon and soil variables, calculated using Pearson‚Äôs correlation coefficient, is represented by the color of the circle. Phyla or classes are ordered according to overall abundance in all the samples, from most abundant (top) to least abundant (bottom). Image published originally in <a class="citation" href="#Hermans2016">Hermans <i>et al.</i> 2016</a>.</figcaption></figure>

<p>On the other hand, many species of microorganisms establish complex <strong>symbiotic relationships</strong> with plant organisms. The fraction of the substrate that is directly influenced by root secretions and associated microorganism is known as <strong>rhizosphere</strong>. Thus, for example, bacteria of the genus <em>Bacillus</em>, <em>Pseudomonas</em> or <em>Burkholderia</em> appear associated with the plant roots, protecting them from pathogenic microorganisms.</p>

<figure id="figure-2"><img src="../../images/metagenomics-nanopore/bacteria_community.png" alt="Bacterial communities" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Representation of the relationships between root microbiomes and plant organisms.</figcaption></figure>

<p>In this tutorial, we will use sequencing data obtained through the <strong>MinION sequencer</strong> (Oxford Nanopore Technologies) with two objectives: 1) evaluate the health status of soil samples and 2) study how microbial populations are modified by their interaction with plant roots. This example was inspired by <a class="citation" href="#Brown2017">Brown <i>et al.</i> 2017</a>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#background-on-data" id="markdown-toc-background-on-data">Background on data</a>    <ol>
      <li><a href="#the-datasets" id="markdown-toc-the-datasets">The datasets</a></li>
    </ol>
  </li>
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#assess-datasets-quality" id="markdown-toc-assess-datasets-quality">Assess datasets quality</a>    <ol>
      <li><a href="#quality-control-using-fastqc-and-multiqc" id="markdown-toc-quality-control-using-fastqc-and-multiqc">Quality control using FastQC and MultiQC</a></li>
    </ol>
  </li>
  <li><a href="#improve-the-dataset-quality" id="markdown-toc-improve-the-dataset-quality">Improve the dataset quality</a>    <ol>
      <li><a href="#adapter-and-chimera-removal-with-porechop" id="markdown-toc-adapter-and-chimera-removal-with-porechop">Adapter and chimera removal with porechop</a></li>
      <li><a href="#filter-sequences-with-fastp" id="markdown-toc-filter-sequences-with-fastp">Filter sequences with fastp</a></li>
    </ol>
  </li>
  <li><a href="#re-evaluate-datasets-quality" id="markdown-toc-re-evaluate-datasets-quality">Re-evaluate datasets quality</a></li>
  <li><a href="#assign-taxonomic-classifications" id="markdown-toc-assign-taxonomic-classifications">Assign taxonomic classifications</a>    <ol>
      <li><a href="#taxonomic-classification-with-kraken2" id="markdown-toc-taxonomic-classification-with-kraken2">Taxonomic classification with Kraken2</a></li>
    </ol>
  </li>
  <li><a href="#analyze-taxonomic-assigment" id="markdown-toc-analyze-taxonomic-assigment">Analyze taxonomic assigment</a>    <ol>
      <li><a href="#visualize-the-taxonomical-classification-with-krona" id="markdown-toc-visualize-the-taxonomical-classification-with-krona">Visualize the taxonomical classification with Krona</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="background-on-data">Background on data</h1>

<p>Next, we will introduce some details about the datasets that we are going to use to perform the analysis.</p>

<h2 id="the-datasets">The datasets</h2>

<p>In this example, we will use a dataset originally hosted in the <strong>NCBI SRA database</strong>, with the accession number <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP194577">SRP194577</a>. To obtain the samples, the <span class="notranslate">DNA</span> was extracted using the Zymo Research Kit, followed by <strong>PCR amplification of 16S r<span class="notranslate">RNA</span> genes</strong>. Sequencing was carried out on a Nanopore Minion device. We are going to use four datasets, corresponding to two experimental conditions:</p>

<ul>
  <li>Soil:
    <ul>
      <li>Surface sample (0-5 cm): <code class="language-plaintext highlighter-rouge">bulk_top.fastq.gz</code></li>
      <li>Deep sample (10-15 cm): <code class="language-plaintext highlighter-rouge">bulk_bottom.fastq.gz</code></li>
    </ul>
  </li>
  <li>Rhizosphere:
    <ul>
      <li>Surface sample (0-5 cm): <code class="language-plaintext highlighter-rouge">rhizosphere_top.fastq.gz</code></li>
      <li>Deep sample (10-15 cm): <code class="language-plaintext highlighter-rouge">rhizosphere_bottom.fastq.gz</code></li>
    </ul>
  </li>
</ul>

<blockquote class="question">

  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Why do we sequence the 16S r<span class="notranslate">RNA</span> genes for analyzing microbial communities?</p>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>There are several reasons to use these genes as taxonomic markers. First, they are constituent parts of ribosomes, a piece of highly-conserved biological machinery, which makes them universally distributed. On the other hand, these genes present regions with varying degrees of sequence variability, which allows them to be used as a species-specific signature.</p>
  </blockquote>
</blockquote>

<h1 id="get-data">Get data</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from Zenodo:</p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Collection</strong> tab</li>
        <li>Click of the <strong>Paste/Fetch</strong> button</li>
        <li>Paste the Zenodo links and press <kbd>Start</kbd> and <kbd>Build</kbd></li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> https://zenodo.org/record/4274812/files/bulk_bottom.fastq.gz
 https://zenodo.org/record/4274812/files/bulk_top.fastq.gz
 https://zenodo.org/record/4274812/files/rhizosphere_bottom.fastq.gz
 https://zenodo.org/record/4274812/files/rhizosphere_top.fastq.gz
</code></pre></div>      </div>
      <ul>
        <li>Assign a name to the new collection: <code class="language-plaintext highlighter-rouge">soil collection</code></li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="assess-datasets-quality">Assess datasets quality</h1>

<p>Before starting to work on our datasets, it is necessary to assess their quality. This is an essential step if we aim to obtain a <strong>meaningful <span class="notranslate">downstream</span> analysis</strong>.</p>

<blockquote class="details">
  <h3 id="details-fastq-sequence-quality-format"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> FASTQ sequence quality format</h3>

  <p>FASTQ format is a text-based format for storing both a biological sequence and its corresponding quality scores. Both the sequence letter and the quality score are encoded with a single ASCII character for brevity. You can find more info in the <a href="https://en.wikipedia.org/wiki/FASTQ_format">Wikipedia article</a>.</p>

</blockquote>

<h2 id="quality-control-using-fastqc-and-multiqc">Quality control using FastQC and MultiQC</h2>

<p><strong>FastQC</strong> is one of the most widely used tools to <strong>check the quality</strong> of the samples generated by High Throughput Sequencing (HTS) technologies.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-check"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality check</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">soil collection</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC unprocessed: Raw</code> and <code class="language-plaintext highlighter-rouge">FastQC unprocessed: Web</code></li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li>In <em>‚ÄúResults‚Äù</em>:
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>
              <p><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>:</p>

              <ul>
                <li><code class="language-plaintext highlighter-rouge">FastQC unprocessed: Raw</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<p><strong>FastQC</strong> provides information on various parameters, such as the range of quality values across all bases at each position. <strong>MultiQC</strong> allows summarizing the output of different outputs from <strong>FastQC</strong>.</p>

<figure id="figure-3"><img src="../../images/metagenomics-nanopore/fastqc_sequence_length_distribution_plot.png" alt="FastQC plots length" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Sequence length distribution</figcaption></figure>

<p>In the first place, we are going to analyse the sequence length distribution of the different datasets.</p>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>Can you explain the sequence length distribution plot?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The main peak, around 1700 bp, corresponds approximately to the length of the gene coding for 16S r<span class="notranslate">RNA</span>. There is also a secondary peak around 200 bp, which may be due to <strong>truncated amplifications</strong>, or as a result of <strong>non-specific hybridization of primers</strong> used for PCR.</p>

  </blockquote>
</blockquote>

<figure id="figure-4"><img src="../../images/metagenomics-nanopore/fastqc_per_base_sequence_quality_plot.png" alt="FastQC plots quality" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Per base sequence quality</figcaption></figure>

<p>If we examine figure 3, we can see that up to 3000 bp the quality of our sequencing data is around a <strong>Phred score</strong> of 12, which is a relatively low value compared to other sequencing technologies. This is explained because Nanopore <span class="notranslate">reads</span> poses <strong>high error rates in the basecalled <span class="notranslate">reads</span></strong> (10% as compared to 1% for Illumina). However, Nanopore sequencing generates <strong>very long <span class="notranslate">reads</span></strong> (in theory only limited by the mechanisms of extraction of the genetic material), enabling the sequencing of the complete 16S r<span class="notranslate">RNA</span> gene, which makes it possible to identify bacterial taxa at higher resolution.</p>

<figure id="figure-5"><img src="../../images/metagenomics-nanopore/fastqc_per_sequence_gc_content_plot.png" alt="FasTQC plots CG" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Per sequence GC content</figcaption></figure>

<p>Let‚Äôs have a look at figure 4, which shows the content of the GC by sequence.</p>

<blockquote class="question">

  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What can cause GC content to show a bimodal peak?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>There are several possible causes for the presence of twin peaks in the GC content. One possible explanation is the presence of adapters in the sequence. Another possible cause is some kind of contamination, such as chimeras.</p>

  </blockquote>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>For more information on the topic of quality control, please see our training materials <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">here</a>.</p>
</blockquote>

<h1 id="improve-the-dataset-quality">Improve the dataset quality</h1>

<p>In order to improve the quality of our data, we will use two tools presented above, <strong>porechop</strong> and <strong>fastp</strong>.</p>

<h2 id="adapter-and-chimera-removal-with-porechop">Adapter and chimera removal with porechop</h2>

<p>Nanopore sequencing technology requires to ligate adapters to both ends of genomic material to facilitate the strand capture and loading of a processive enzyme at the 5‚Äôend, boosting the effectiveness of the sequencing process. Adapter sequences should be removed because they <strong>can interfere with aligment of <span class="notranslate">reads</span></strong> to 16S r<span class="notranslate">RNA</span> gene reference database, for which we will use the <strong>porechop</strong> tool (<a class="citation" href="#rrwick2017">Wick 2017</a>).</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
  <p>If you are interested in Nanopore sequecing technology, you can find more information in <a class="citation" href="#Jain2016">Jain <i>et al.</i> 2016</a>.</p>

</blockquote>

<p>On the other hand, <strong>chimeric sequences</strong> are considered a contaminant and should be removed because they can result in artificial inflation of the microbial diversity. With <strong>Porechop</strong> you can eliminate them.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-adapters-with-porechop"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove adapters with porechop</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3"><strong>Porechop</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/porechop/porechop/0.2.3</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúInput FASTA/FASTQ‚Äù</em>: <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <code class="language-plaintext highlighter-rouge">soil collection</code></li>
        <li><em>‚ÄúOutput format for the <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">fastq</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">soil collection trimmed</code></li>
  </ol>

</blockquote>

<h2 id="filter-sequences-with-fastp">Filter sequences with fastp</h2>

<p>To increase the specificity of the analysis, we will select the <span class="notranslate">reads</span> with lengths between 1000 bp and 2000 bp, which are more informative from a taxonomic point of view, because they include both preserved and hypervariable regions of the 16S r<span class="notranslate">RNA</span> gene. In addition, sequences will be filtered on a minimum average read quality score of 9, according to the recommendations from (missing reference). This stage will be carried out through the use of <strong>fastp</strong> (<a class="citation" href="#Chen_2018">Chen <i>et al.</i> 2018</a>), an open-source tool designed to process FASTQ files.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-sequence-with-fastp"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter sequence with fastp</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0"><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSingle-end or paired <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">soil collection trimmed</code>
              <ul>
                <li>In <em>‚ÄúAdapter Trimming Options‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúDisable adapter trimming‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúFilter Options‚Äù</em>:
          <ul>
            <li>In <em>‚ÄúQuality filtering options‚Äù</em>:
              <ul>
                <li><em>‚ÄúQualified quality phred‚Äù</em>: <code class="language-plaintext highlighter-rouge">9</code></li>
              </ul>
            </li>
            <li>In <em>‚ÄúLength filtering options‚Äù</em>:
              <ul>
                <li><em>‚ÄúLength required‚Äù</em>: <code class="language-plaintext highlighter-rouge">1000</code></li>
                <li><em>‚ÄúMaximum length‚Äù</em>: <code class="language-plaintext highlighter-rouge">2000</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúRead Modification Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúPolyG tail trimming‚Äù</em>: <code class="language-plaintext highlighter-rouge">Disable polyG tail trimming</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">soil collection processed</code></li>
  </ol>

</blockquote>

<h1 id="re-evaluate-datasets-quality">Re-evaluate datasets quality</h1>

<p>After processing the sequences, we are going to analyze them again using <strong>FastQC</strong> and <strong>MultiQC</strong> to see if we have managed to correct the anomalies that we had detected.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-check-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality check</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">soil collection processed</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC processed: Raw</code> and <code class="language-plaintext highlighter-rouge">FastQC processed: Web</code></li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li>In <em>‚ÄúResults‚Äù</em>:
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>
              <p><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>:</p>

              <ul>
                <li><code class="language-plaintext highlighter-rouge">FastQC processed: Raw</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<p>We can verify that after processing the samples, the GC content presents a unimodal distribution, which indicates that the anomalies in the sequences have been successfully eliminated (Figure 6).</p>

<figure id="figure-6"><img src="../../images/metagenomics-nanopore/fasqc_per_sequence_content_plot_clean.png" alt="FasTQC plots GC" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Per sequence GC content in processed samples</figcaption></figure>

<h1 id="assign-taxonomic-classifications">Assign taxonomic classifications</h1>

<p>One of the key steps in metagenomic data analysis is to identify the taxon to which the individual <span class="notranslate">reads</span> belong. Taxonomic classification tools are based on microbial genome databases to identify the origin of each sequence.</p>

<h2 id="taxonomic-classification-with-kraken2">Taxonomic classification with Kraken2</h2>

<p>To perform the taxonomic classification we will use <strong>Kraken2</strong> ((missing reference)). This tool uses the minimizer method to sample the k-mers (all the read‚Äôs subsequences of length <em>k</em>) in a deterministic fashion in order to reduce memory constumption and processing time. In addition, it masks low-complexity sequences from reference sequences by using <strong>dustmasker</strong>.</p>

<blockquote class="comment">
  <h3 id="comment-comments-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p><strong>Kraken2</strong> uses a compact hash table, a probabilistic data structure that allows for faster queries and lower memory requirements. It applies a spaced seed mask of <em>s</em> spaces to the minimizer and calculates a compact hash code, which is then used as a search query in its compact hash table; the lowest common ancestor (LCA) taxon associated with the compact hash code is then assigned to the k-mer.
You can find more information about the <strong>Kraken2</strong> algorithm in the paper <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1891-0"><em>Improved metagenomic analysis with Kraken 2</em></a>.</p>
</blockquote>

<figure id="figure-7"><img src="../../images/metagenomics-nanopore/kmers-kraken.jpg" alt="Taxonomic classification" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Kraken2 sequence classification algorithm. To classify a sequence, each l-mer is mapped to the lowest common ancestor (LCA) of the genomes that contain that l-mer in a database. In the classification tree, each node has a weight equal to the number of l-mers in the sequence associated with the node‚Äôs taxon. Image originally published in <a class="citation" href="#Wood2014">Wood and Salzberg 2014</a>.</figcaption></figure>

<p>For this tutorial, we will use the <strong>SILVA database</strong> (<a class="citation" href="#Quast2012">Quast <i>et al.</i> 2012</a>). It includes over 3.2 million 16S r<span class="notranslate">RNA</span> sequences from the <em>Bacteria</em>, <em>Archaea</em> and <em>Eukaryota</em> domains.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assign-taxonomic-labels-with-kraken2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Assign taxonomic labels with Kraken2</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+galaxy0"><strong>Kraken2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/kraken2/kraken2/2.0.8_beta+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúSingle or paired <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">soil collection processed</code></li>
        <li><em>‚ÄúPrint scientific names instead of just taxids‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúConfidence‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
        <li>In <em>‚ÄúCreate Report‚Äù</em>:
          <ul>
            <li><em>‚ÄúPrint a report with aggregrate counts/clade to file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúFormat report output like Kraken 1‚Äôs kraken-mpa-report‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSelect a Kraken2 database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Silva (Created: 2020-06-24T164526Z, kmer-len=35, minimizer-len=31, minimizer-spaces=6)</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>A confidence score of 0.1 means that at least 10% of the k-mers should match entries in the database. This value can be reduced if a less restrictive taxonomic assignation is desired.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="analyze-taxonomic-assigment">Analyze taxonomic assigment</h1>

<p>Once we have assigned the corresponding taxa to each sequence, the next step is to properly visualize the data, for which we will use the <strong>Krona pie chart</strong> tool (<a class="citation" href="#Ondov_2011">Ondov <i>et al.</i> 2011</a>). But before that, we need to adjust the format of the data output from <strong>Kraken2</strong>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adjust-dataset-format"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adjust dataset format</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/datamash_reverse/datamash_reverse/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/datamash_reverse/datamash_reverse/1.1.0"><strong>Reverse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/datamash_reverse/datamash_reverse/1.1.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">Report: Kraken2 on collection</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2"><strong>Replace Text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.1.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">Reverse on collection</code></li>
        <li>In <em>‚ÄúReplacement‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Replacement‚Äù</em>
              <ul>
                <li><em>‚ÄúFind pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">\|</code></li>
                <li><em>‚ÄúReplace with‚Äù</em>: <code class="language-plaintext highlighter-rouge">\t</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="Remove+beginning1" title="Tested with Remove+beginning1"><strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Remove+beginning1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">Replace Text on collection</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="visualize-the-taxonomical-classification-with-krona">Visualize the taxonomical classification with Krona</h2>

<p><strong>Krona</strong> allows hierarchical data to be explored with zooming, multi-layered pie charts. With this tool, we can easily visualize the composition of the bacterial communities and compare how the populations of microorganisms are modified according to the conditions of the environment.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualize-metagenomics-analysis-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualize metagenomics analysis results</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1" title="Tested with toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1"><strong>Krona pie chart</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.7.1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWhat is the type of your input data‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tabular</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúDataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">Remove beginning on collection</code></li>
        <li><em>‚ÄúProvide a name for the basal rank‚Äù</em>: <code class="language-plaintext highlighter-rouge">Bacteria</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Let‚Äôs take a look at the result. Using the search bar we can check if certain taxa are present.</p>

<iframe id="krona" src="krona_all.html" frameborder="0" width="100%" height="900px"> ![Krona](../../images/metagenomics-nanopore/krona.png) </iframe>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What can we say about the health status of the soil samples?</li>
    <li>Are there significant differences in the bacterial population structure between soil samples and rhizosphere samples?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The low presence of <em>Alphaproteobacteria</em> (including members of the order <em>Rhizobiales</em>), as well as the abundance of <em>Bacteroidetes</em> and <em>Gammaproteobacteria</em>, suggest that the soil is highly exposed to phosphorus. This mineral is strongly associated with the anthropogenic activity as it is an important component of agrochemicals.</li>
      <li>We can observe that there are important differences in the structure of the bacterial populations between the soil and rhizosphere samples. Particularly significant is the increase in phylum <em>Plactomycetes</em>, which are usually abundant in the rhizosphere.</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial we used MinION Nanopore sequencing data to study the health status of soil samples and the structure of bacterial populations. The results indicate that the soil suffers some degree of erosion as a result of their exposure to agrochemicals. We have also been able to study how the composition of microbial communities is modified in presence of plant organisms.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>We learned to use MinION Nanopore data for analyzing the health status of the soil</p>
</li>
                            
                            <li><p>We preprocessed Nanopore sequences in order to improve their quality</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/metagenomics#references">here</a>.</p>
                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Fierer2006">Fierer, N., and R. B. Jackson, 2006 <b>The diversity and biogeography of soil bacterial communities</b>. Proceedings of the National Academy of Sciences 103: 626‚Äì631.</span>
    
    
    <a href="https://doi.org/10.1073/pnas.0507535103">10.1073/pnas.0507535103</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Ondov_2011">Ondov, B. D., N. H. Bergman, and A. M. Phillippy, 2011 <b>Interactive metagenomic visualization in a Web browser</b>. BMC Bioinformatics 12:</span>
    
    
    <a href="https://doi.org/10.1186/1471-2105-12-385">10.1186/1471-2105-12-385</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Quast2012">Quast, C., E. Pruesse, P. Yilmaz, J. Gerken, T. Schweer <i>et al.</i>, 2012 <b>The SILVA ribosomal RNA gene database project: improved data processing and web-based tools</b>. Nucleic Acids Research 41: D590‚ÄìD596.</span>
    
    
    <a href="https://doi.org/10.1093/nar/gks1219">10.1093/nar/gks1219</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Wood2014">Wood, D. E., and S. L. Salzberg, 2014 <b>Kraken: ultrafast metagenomic sequence classification using exact alignments</b>. Genome Biology 15: R46.</span>
    
    
    <a href="https://doi.org/10.1186/gb-2014-15-3-r46">10.1186/gb-2014-15-3-r46</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Hermans2016">Hermans, S. M., H. L. Buckley, B. S. Case, F. Curran-Cournane, M. Taylor <i>et al.</i>, 2016 <b>Bacteria as Emerging Indicators of Soil Condition</b> (F. E. Loeffler, Ed.). Applied and Environmental Microbiology 83:</span>
    
    
    <a href="https://doi.org/10.1128/aem.02826-16">10.1128/aem.02826-16</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Jain2016">Jain, M., H. E. Olsen, B. Paten, and M. Akeson, 2016 <b>The Oxford Nanopore MinION: delivery of nanopore sequencing to the genomics community</b>. Genome Biology 17:</span>
    
    
    <a href="https://doi.org/10.1186/s13059-016-1103-0">10.1186/s13059-016-1103-0</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="rrwick2017">Wick, R., 2017 <b>Porechop</b>. <i>GitHub</i>.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://github.com/rrwick/Porechop">https://github.com/rrwick/Porechop</a>
      
    

</li>
<li><span id="Brown2017">Brown, B. L., M. Watson, S. S. Minot, M. C. Rivera, and R. B. Franklin, 2017 <b>MinION‚Ñ¢ nanopore sequencing of environmental metagenomes: a synthetic approach</b>. GigaScience 6:</span>
    
    
    <a href="https://doi.org/10.1093/gigascience/gix007">10.1093/gigascience/gix007</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Chen_2018">Chen, S., Y. Zhou, Y. Chen, and J. Gu, 2018 <b>fastp: an ultra-fast all-in-one FASTQ preprocessor</b>.</span>
    
    
    <a href="https://doi.org/10.1101/274100">10.1101/274100</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=16S Microbial analysis with Nanopore data (Metagenomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Crist√≥bal Gallardo, 2021 <b>16S Microbial analysis with Nanopore data (Galaxy Training Materials)</b>. <a href="/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html">/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{metagenomics-nanopore-16S-metagenomics,
    author = "Crist√≥bal Gallardo",
    title = "16S Microbial analysis with Nanopore data (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Crist√≥bal Gallardo)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/metagenomics/tutorials/nanopore-16S-metagenomics/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
