<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Quality Control</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Analyses of sequences" />
        <meta property="og:title" content="Galaxy Training: Quality Control" />
        <meta property="og:description" content="Analyses of sequences" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Sequence analysis
                        </a>
                        
                    </li>
                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/sequence-analysis/faqs/" title="Check our FAQs for the Sequence analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/sequence-analysis/tutorials/quality-control/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Translate this page
            </div>

            <div id="lang-selector">
                
                <strong>Automatic Translations</strong>

<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html&edit-text=&act=url">
                Fran√ßais
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html&edit-text=&act=url">
                Êó•Êú¨Ë™û
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html&edit-text=&act=url">
                Espa√±ol
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html&edit-text=&act=url">
                Portugu√™s
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html&edit-text=&act=url">
                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                </a>
                
                <a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/sequence-analysis/tutorials/quality-control/tutorial.html&edit-text=&act=url" title="">
                And more!
                </a>
            </div>
        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-10-01/topics/sequence-analysis/tutorials/quality-control/tutorial.html" title="Version 2021-10-01">2021-10-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-09-01/topics/sequence-analysis/tutorials/quality-control/tutorial.html" title="Version 2021-09-01">2021-09-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-08-01/topics/sequence-analysis/tutorials/quality-control/tutorial.html" title="Version 2021-08-01">2021-08-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Quality Control",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Sequence analysis",
    "description": "Analyses of sequences",
    "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/"
  },
  "courseCode": "sequence-analysis / quality-control / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Quality Control' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
  "timeRequired": "PT1H30M",
  "description": "The questions this  addresses are:\n - How to perform quality control of NGS raw data (FASTQ)?\n - What are the quality parameters to check for a dataset?\n - How to improve the quality of a dataset?\n\n\\nThe objectives are:\n - Assess FASTQ quality using FASTQE üß¨üòé and FastQC\n - Perform quality correction with Cutadapt\n - Summarise quality metrics MultiQC\n - Process single-end and paired-end data\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Maria Doyle"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Maria Doyle"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Sequence analysis",
      "description": "Analyses of sequences",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/"
    }
  ],
  "skillLevel": "Introductory"
}
</script>

<section class="tutorial topic-sequence-analysis">
    <h1 data-toc-skip>Quality Control</h1>
    

    <div class="contributors-line">Authors: <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="https://avatars.githubusercontent.com/bebatut?s=27" alt="Avatar">B√©r√©nice Batut</a><a href="/training-material/hall-of-fame/mblue9/" class="contributor-badge contributor-mblue9"><img src="https://avatars.githubusercontent.com/mblue9?s=27" alt="Avatar">Maria Doyle</a></div>

    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to perform quality control of NGS raw data (FASTQ)?</p>
</li>
        
        <li><p>What are the quality parameters to check for a dataset?</p>
</li>
        
        <li><p>How to improve the quality of a dataset?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Assess FASTQ quality using FASTQE üß¨üòé and FastQC</p>
</li>
        
        <li><p>Perform quality correction with Cutadapt</p>
</li>
        
        <li><p>Summarise quality metrics MultiQC</p>
</li>
        
        <li><p>Process single-end and paired-end data</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour 30 minutes</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="topic-icon" href="https://doi.org/10.5281/zenodo.61771">
    <i class="far fa-copy" aria-hidden="true"></i> Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/sequence-analysis/tutorials/quality-control/workflows/" title="Workflows" alt="Quality Control workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            
                <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/main/topics/sequence-analysis/tutorials/quality-control/tours/" title="Galaxy interactive tour" >
    <i class="fas fa-magic" aria-hidden="true"></i> Tours
</a>

</li>
            

            

            
            
            

            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/sequence-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://clipseq.usegalaxy.eu/" title="CLIP-Explorer">
            CLIP-Explorer
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
        <a class="dropdown-item" href="https://ecology.usegalaxy.eu/" title="Galaxy for Ecology">
            Galaxy for Ecology
        </a>
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://hicexplorer.usegalaxy.eu" title="HiCExplorer">
            HiCExplorer
        </a>
        
    
        
        <a class="dropdown-item" href="https://humancellatlas.usegalaxy.eu/" title="Human Cell Atlas">
            Human Cell Atlas
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://metagenomics.usegalaxy.eu/" title="Metagenomics">
            Metagenomics
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="http://galaxy.migale.inra.fr/" title="Migale">
            Migale
        </a>
        
    
        
        <a class="dropdown-item" href="https://nanopore.usegalaxy.eu/" title="NanoGalaxy">
            NanoGalaxy
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://plants.usegalaxy.eu/" title="Plant Anaylsis Workbench">
            Plant Anaylsis Workbench
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/" title="Single Cell Omics workbench">
            Single Cell Omics workbench
        </a>
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://virology.usegalaxy.eu/" title="Virology">
            Virology
        </a>
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Sep 1, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                

                <h1 class="no_toc" id="introduction">Introduction</h1>

<p>During sequencing, the nucleotide bases in a <span class="notranslate">DNA</span> or <span class="notranslate">RNA</span> sample (library) are determined by the sequencer. For each fragment in the library, a short sequence is generated, also called a <strong>read</strong>, which is simply a succession of nucleotides.</p>

<p>Modern sequencing technologies can generate a massive number of sequence <span class="notranslate">reads</span> in a single experiment. However, no sequencing technology is perfect, and each instrument will generate different types and amount of errors, such as incorrect nucleotides being called. These wrongly called bases are due to the technical limitations of each sequencing platform.</p>

<p>Therefore, it is necessary to understand, identify and exclude error-types that may impact the interpretation of <span class="notranslate">downstream</span> analysis.
Sequence quality control is therefore an essential first step in your analysis. Catching errors early saves time later on.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#inspect-a-raw-sequence-file" id="markdown-toc-inspect-a-raw-sequence-file">Inspect a raw sequence file</a></li>
  <li><a href="#assess-quality-with-fastqe-" id="markdown-toc-assess-quality-with-fastqe-">Assess quality with FASTQE üß¨üòé</a></li>
  <li><a href="#assess-quality-with-fastqc" id="markdown-toc-assess-quality-with-fastqc">Assess quality with FastQC</a>    <ol>
      <li><a href="#per-base-sequence-quality" id="markdown-toc-per-base-sequence-quality">Per base sequence quality</a></li>
      <li><a href="#per-sequence-quality-scores" id="markdown-toc-per-sequence-quality-scores">Per sequence quality scores</a></li>
      <li><a href="#per-base-sequence-content" id="markdown-toc-per-base-sequence-content">Per base sequence content</a></li>
      <li><a href="#per-sequence-gc-content" id="markdown-toc-per-sequence-gc-content">Per sequence GC content</a></li>
      <li><a href="#sequence-duplication-levels" id="markdown-toc-sequence-duplication-levels">Sequence Duplication Levels</a></li>
      <li><a href="#over-represented-sequences" id="markdown-toc-over-represented-sequences">Over-represented sequences</a></li>
    </ol>
  </li>
  <li><a href="#trim-and-filter" id="markdown-toc-trim-and-filter">Trim and filter</a></li>
  <li><a href="#processing-multiple-datasets" id="markdown-toc-processing-multiple-datasets">Processing multiple datasets</a>    <ol>
      <li><a href="#process-paired-end-data" id="markdown-toc-process-paired-end-data">Process paired-end data</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="inspect-a-raw-sequence-file">Inspect a raw sequence file</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>
    </li>
    <li>
      <p>Import the file <code class="language-plaintext highlighter-rouge">female_oral2.fastq-4143.gz</code> from <a href="https://zenodo.org/record/3977236">Zenodo</a> or from the data library (ask your instructor)
This is a microbiome sample from a snake <span class="citation"><a href="#StJacques2021">Jacques <i>et al.</i> 2021</a></span>.</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3977236/files/female_oral2.fastq-4143.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li>    <p><strong>Close</strong> the window</p>  </li>  <li>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</li></ul></blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<p>We just imported a file into <span class="notranslate">Galaxy</span>. This file is similar to the data we could get directly from a sequencing facility: a <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ file</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-the-fastq-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect the FASTQ file</h3>

  <ol>
    <li>Inspect the file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon</li>
  </ol>

</blockquote>

<p>Although it looks complicated (and maybe it is), the FASTQ format is easy to understand with a little decoding.</p>

<p>Each read, representing a fragment of the library, is encoded by 4 lines:</p>

<table>
  <thead>
    <tr>
      <th>Line</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>1</td>
      <td>Always begins with <code class="language-plaintext highlighter-rouge">@</code> followed by the information about the read</td>
    </tr>
    <tr>
      <td>2</td>
      <td>The actual nucleic sequence</td>
    </tr>
    <tr>
      <td>3</td>
      <td>Always begins with a <code class="language-plaintext highlighter-rouge">+</code> and contains sometimes the same info in line 1</td>
    </tr>
    <tr>
      <td>4</td>
      <td>Has a string of characters which represent the quality scores associated with each base of the nucleic sequence; must have the same number of characters as line 2</td>
    </tr>
  </tbody>
</table>

<p>So for example, the first sequence in our file is:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@M00970:337:000000000-BR5KF:1:1102:17745:1557 1:N:0:CGCAGAAC+ACAGAGTT
GTGCCAGCCGCCGCGGTAGTCCGACGTGGCTGTCTCTTATACACATCTCCGAGCCCACGAGACCGAAGAACATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAGAAGCAAATGACGATTCAAGAAAGAAAAAAACACAGAATACTAACAATAAGTCATAAACATCATCAACATAAAAAAGGAAATACACTTACAACACATATCAATATCTAAAATAAATGATCAGCACACAACATGACGATTACCACACATGTGTACTACAAGTCAACTA
+
GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGFGGGGGGAFFGGFGGGGGGGGFGGGGGGGGGGGGGGFGGG+38+35*311*6,,31=******441+++0+0++0+*1*2++2++0*+*2*02*/***1*+++0+0++38++00++++++++++0+0+2++*+*+*+*+*****+0**+0**+***+)*.***1**//*)***)/)*)))*)))*),)0(((-((((-.(4(,,))).,(())))))).)))))))-))-(
</code></pre></div></div>

<p>It means that the fragment named <code class="language-plaintext highlighter-rouge">@M00970</code> corresponds to the <span class="notranslate">DNA</span> sequence <code class="language-plaintext highlighter-rouge">GTGCCAGCCGCCGCGGTAGTCCGACGTGGCTGTCTCTTATACACATCTCCGAGCCCACGAGACCGAAGAACATCTCGTATGCCGTCTTCTGCTTGAAAAAAAAAAAAAAAAAAAACAAAAAAAAAAAAAGAAGCAAATGACGATTCAAGAAAGAAAAAAACACAGAATACTAACAATAAGTCATAAACATCATCAACATAAAAAAGGAAATACACTTACAACACATATCAATATCTAAAATAAATGATCAGCACACAACATGACGATTACCACACATGTGTACTACAAGTCAACTA</code> and this sequence has been sequenced with a quality <code class="language-plaintext highlighter-rouge">GGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGGFGGGFGGGGGGAFFGGFGGGGGGGGFGGGGGGGGGGGGGGFGGG+38+35*311*6,,31=******441+++0+0++0+*1*2++2++0*+*2*02*/***1*+++0+0++38++00++++++++++0+0+2++*+*+*+*+*****+0**+0**+***+)*.***1**//*)***)/)*)))*)))*),)0(((-((((-.(4(,,))).,(())))))).)))))))-))-(</code>.</p>

<p>But what does this quality score mean?</p>

<p>The quality score for each sequence is a string of characters, one for each base of the nucleic sequence, used to characterize the probability of mis-identification of each base. The score is encoded using the ASCII character table (with <a href="https://en.wikipedia.org/wiki/FASTQ_format#Encoding">some historical differences</a>):</p>

<p><img src="../../../sequence-analysis/images/quality_score_encoding.png" alt="Encoding of the quality score with ASCII characters for different Phred encoding. The ascii code sequence is shown at the top with symbols for 33 to 64, upper case letters, more symbols, and then lowercase letters. Sanger maps from 33 to 73 while solexa is shifted, starting at 59 and going to 104. Illumina 1.3 starts at 54 and goes to 104, Illumina 1.5 is shifted three scores to the right but still ends at 104. Illumina 1.8+ goes back to the Sanger except one single score wider. Illumina. " loading="lazy" /></p>

<p>So there is an ASCII character associated with each nucleotide, representing its <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred quality score</a>, the probability of an incorrect base call:</p>

<table>
  <thead>
    <tr>
      <th>Phred Quality Score</th>
      <th>Probability of incorrect base call</th>
      <th>Base call accuracy</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>10</td>
      <td>1 in 10</td>
      <td>90%</td>
    </tr>
    <tr>
      <td>20</td>
      <td>1 in 100</td>
      <td>99%</td>
    </tr>
    <tr>
      <td>30</td>
      <td>1 in 1000</td>
      <td>99.9%</td>
    </tr>
    <tr>
      <td>40</td>
      <td>1 in 10,000</td>
      <td>99.99%</td>
    </tr>
    <tr>
      <td>50</td>
      <td>1 in 100,000</td>
      <td>99.999%</td>
    </tr>
    <tr>
      <td>60</td>
      <td>1 in 1,000,000</td>
      <td>99.9999%</td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Which ASCII character corresponds to the worst Phred score for Illumina 1.8+?</li>
    <li>What is the Phred quality score of the 3rd nucleotide of the 1st sequence?</li>
    <li>What is the accuracy of this 3rd nucleotide?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>The worst Phred score is the smallest one, so 0. For Illumina 1.8+, it corresponds to the <code class="language-plaintext highlighter-rouge">!</code> character.</li>
      <li>The 3rd nucleotide of the 1st sequence has a ASCII character <code class="language-plaintext highlighter-rouge">G</code>, which correspond to a score of 38.</li>
      <li>The corresponding nucleotide <code class="language-plaintext highlighter-rouge">G</code> has an accuracy of almost 99.99%</li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
  <p>The current lllumina (1.8+) uses Sanger format (Phred+33). If you are working with older datasets you may encounter the older scoring schemes. <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, a tool we will use later in this tutorial, can be used to try to determine what type of quality encoding is used (through assessing the range of Phred values seen in the FASTQ).</p>
</blockquote>

<p>When looking at the file in <span class="notranslate">Galaxy</span>, it looks like most the nucleotides have a high score (<code class="language-plaintext highlighter-rouge">G</code> corresponding to a score 38). Is it true for all sequences? And along the full sequence length?</p>

<h1 id="assess-quality-with-fastqe-">Assess quality with FASTQE üß¨üòé</h1>

<p>To take a look at sequence quality along all sequences, we can use <a href="https://fastqe.com/">FASTQE</a>. It is an open-source tool that provides a simple and fun way to quality control raw sequence data and print them as emoji. You can use it to give a quick impression of whether your data has any problems of which you should be aware before doing any further analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-check"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality check</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2"><strong>FASTQE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúFastQ data‚Äù</em>: <code class="language-plaintext highlighter-rouge">female_oral2.fastq-4143.gz</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúScore types to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<p>Rather than looking at quality scores for each individual read, FASTQE looks at quality collectively across all <span class="notranslate">reads</span> within a sample and can calculate the mean for each nucleotide position along the length of the <span class="notranslate">reads</span>. Below shows the mean values for this dataset.</p>

<figure id="figure-1"><img src="../../images/quality-control/fastqe-mean-before.png" alt="FASTQE before. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 1:</span> FASTQE mean scores</figcaption></figure>

<p>You can see the score for each emoji <a href="https://github.com/fastqe/fastqe#scale">here</a>. The emojis below, with Phred scores less than 20, are the ones we hope we don‚Äôt see much.</p>

<table>
  <thead>
    <tr>
      <th>Phred Quality Score</th>
      <th>ASCII code</th>
      <th>Emoji</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>!</td>
      <td>üö´</td>
    </tr>
    <tr>
      <td>1</td>
      <td>‚Äù</td>
      <td>‚ùå</td>
    </tr>
    <tr>
      <td>2</td>
      <td>#</td>
      <td>üë∫</td>
    </tr>
    <tr>
      <td>3</td>
      <td>$</td>
      <td>üíî</td>
    </tr>
    <tr>
      <td>4</td>
      <td>%</td>
      <td>üôÖ</td>
    </tr>
    <tr>
      <td>5</td>
      <td>&amp;</td>
      <td>üëæ</td>
    </tr>
    <tr>
      <td>6</td>
      <td>‚Äô</td>
      <td>üëø</td>
    </tr>
    <tr>
      <td>7</td>
      <td>(</td>
      <td>üíÄ</td>
    </tr>
    <tr>
      <td>8</td>
      <td>)</td>
      <td>üëª</td>
    </tr>
    <tr>
      <td>9</td>
      <td>*</td>
      <td>üôà</td>
    </tr>
    <tr>
      <td>10</td>
      <td>+</td>
      <td>üôâ</td>
    </tr>
    <tr>
      <td>11</td>
      <td>,</td>
      <td>üôä</td>
    </tr>
    <tr>
      <td>12</td>
      <td>-</td>
      <td>üêµ</td>
    </tr>
    <tr>
      <td>13</td>
      <td>.</td>
      <td>üòø</td>
    </tr>
    <tr>
      <td>14</td>
      <td>/</td>
      <td>üòæ</td>
    </tr>
    <tr>
      <td>15</td>
      <td>0</td>
      <td>üôÄ</td>
    </tr>
    <tr>
      <td>16</td>
      <td>1</td>
      <td>üí£</td>
    </tr>
    <tr>
      <td>17</td>
      <td>2</td>
      <td>üî•</td>
    </tr>
    <tr>
      <td>18</td>
      <td>3</td>
      <td>üò°</td>
    </tr>
    <tr>
      <td>19</td>
      <td>4</td>
      <td>üí©</td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What is the lowest mean score in this dataset?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The lowest score in this dataset is üòø 13.</p>
  </blockquote>
</blockquote>

<h1 id="assess-quality-with-fastqc">Assess quality with FastQC</h1>

<p>An additional or alte<span class="notranslate">rna</span>tive way we can check sequence quality is with <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>. It provides a modular set of analyses which you can use to check whether your data has any problems of which you should be aware before doing any further analysis.  We can use it, for example, to assess whether there are known adapters present in the data. We‚Äôll run it on the FASTQ file.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-check-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality check</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúShort read data from your current history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cutadapt Read 1 Output</code></li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Which Phred encoding is used in the FASTQ file for these sequences?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The Phred scores are encoded using <code class="language-plaintext highlighter-rouge">Sanger / Illumina 1.9</code> (<code class="language-plaintext highlighter-rouge">Encoding</code> in the top table).</p>
  </blockquote>
</blockquote>

<h2 id="per-base-sequence-quality">Per base sequence quality</h2>

<p>With FastQC we can use the per base sequence quality plot to check the base quality of the <span class="notranslate">reads</span>, similar to what we did with FASTQE.</p>

<figure id="figure-2"><img src="../../images/quality-control/per_base_sequence_quality-before.png" alt="Per base sequence quality. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Per base sequence quality</figcaption></figure>

<p>On the x-axis are the base position in the read. In this example, the sample contains <span class="notranslate">reads</span> that are up to 296 bp long.</p>

<blockquote class="details">
  <h3 id="details-non-uniform-x-axis"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Non uniform x-axis</h3>

  <p>The x-axis is not always uniform. When you have long <span class="notranslate">reads</span>, some binning is applied to keep things compact. We can see that in our sample. It starts out with individual 1-10 bases. After that, bases are binned across a window a certain number of bases wide. Data binning means grouping and is a data pre-processing technique used to reduce the effects of minor observation errors. The number of base positions binned together depends on the length of the read. With <span class="notranslate">reads</span> &gt;50bp, the latter part of the plot will report aggregate statistics for 5bp windows. Shorter <span class="notranslate">reads</span> will have smaller windows and longer <span class="notranslate">reads</span> larger windows. Binning can be removed when running FastQC by setting the paramter ‚ÄúDisable grouping of bases for <span class="notranslate">reads</span> &gt;50bp‚Äù to Yes.</p>
</blockquote>

<p>For each position, a boxplot is drawn with:</p>

<ul>
  <li>the median value, represented by the central red line</li>
  <li>the inter-quartile range (25-75%), represented by the yellow box</li>
  <li>the 10% and 90% values in the upper and lower whiskers</li>
  <li>the mean quality, represented by the blue line</li>
</ul>

<p>The y-axis shows the quality scores. The higher the score, the better the base call. The background of the graph divides the y-axis into very good quality scores (green), scores of reasonable quality (orange), and <span class="notranslate">reads</span> of poor quality (red).</p>

<p>It is normal with all Illumina sequencers for the median quality score to start out lower over the first 5-7 bases and to then rise. The quality of <span class="notranslate">reads</span> on most platforms will drop at the end of the read. This is often due to signal decay or phasing during the sequencing run. The recent developments in chemistry applied to sequencing has improved this somewhat, but <span class="notranslate">reads</span> are now longer than ever.</p>

<blockquote class="details">
  <h3 id="details-signal-decay-and-phasing"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Signal decay and phasing</h3>

  <ul>
    <li>Signal decay</li>
  </ul>

  <p>The fluorescent signal intensity decays with each cycle of the sequencing process. Due to the degrading fluorophores, a proportion of the strands in the cluster are not being elongated. The proportion of the signal being emitted continues to decrease with each cycle, yielding to a decrease of quality scores at the 3‚Äô end of the read.</p>

  <ul>
    <li>Phasing</li>
  </ul>

  <p>The signal starts to blur with the increase of number of cycles because the cluster looses synchronicity. As the cycles progress, some strands get random failures of nucleotides to incorporate due to:</p>

  <ul>
    <li>Incomplete removal of the 3‚Äô terminators and fluorophores</li>
    <li>Incorporation of nucleotides without effective 3‚Äô terminators</li>
  </ul>

  <p>This leads to a decrease in quality scores at the 3‚Äô end of the read.</p>
</blockquote>

<blockquote class="details">
  <h3 id="details-other-sequence-quality-profiles"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Other sequence quality profiles</h3>

  <p>These are some per base sequence quality profiles that can indicate issues with the sequencing.</p>

  <ul>
    <li>
      <p>Overclustering</p>

      <p>Sequencing facilities can overcluster the flow cells. It results in small distances between clusters and an overlap in the signals. Two clusters can be interpreted as a single cluster with mixed fluorescent signals being detected, decreasing signal purity. It generates lower quality scores across the entire read.</p>
    </li>
    <li>
      <p>Instrumentation breakdown</p>

      <p>Some issues can occasionally happen with the sequencing instruments during a run. Any sudden drop in quality or a large percentage of low quality <span class="notranslate">reads</span> across the read could indicate a problem at the facility. Some examples of such issues:</p>

      <ul>
        <li>
          <p>Manifold burst</p>

          <p><img src="../../images/quality-control/per_base_sequence_quality_manifold_burst.png" alt="Manifold burst. " loading="lazy" /></p>
        </li>
        <li>
          <p>Cycles loss</p>

          <p><img src="../../images/quality-control/per_base_sequence_quality_cycle_loss.png" alt="Cycles loss. " loading="lazy" /></p>
        </li>
        <li>
          <p>Read 2 failure</p>

          <p><img src="../../images/quality-control/per_base_sequence_quality_read2_failure.png" alt="Cycles loss. " loading="lazy" /></p>
        </li>
      </ul>

      <p>With such data, the sequencing facility should be contacted for discussion. Often, a resequencing then is needed (and from our experience also offered by the company).</p>
    </li>
  </ul>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How does the mean quality score change along the sequence?</li>
    <li>Is this tendency seen in all sequences?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>The mean quality score (blue line) drops about midway though these sequences. It is common for the mean quality to drop towards the end of the sequences, as the sequencers are incorporating more incorrect nucleotides at the end. However, in this sample there is a very large drop in quality from the middle onwards.</li>
      <li>The box plots are getting wider from position ~100.  It means a lot of sequences have their score dropping from the middle of the sequence. After 100 nucleotides, more than 10% of the sequences have scores below 20.</li>
    </ol>

  </blockquote>
</blockquote>

<p>When the median quality is below a Phred score of ~20, we should consider trimming away bad quality bases from the sequence. We will explain that process in the Trim and filter section.</p>

<h4 id="adapter-content">Adapter Content</h4>

<figure id="figure-3"><img src="../../images/quality-control/adapter_content-before.png" alt="Adapter Content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Adapter Content</figcaption></figure>

<p>The plot shows the cumulative percentage of <span class="notranslate">reads</span> with the different adapter sequences at each position. Once an adapter sequence is seen in a read it  is counted as being present right through to the end of the read so the percentage increases with the read length. FastQC can detect some adapters by default (e.g. Illumina, Nextera), for others we could provide a contaminants file as an input to the FastQC tool.</p>

<p>Ideally Illumina sequence data should not have any adapter sequence present. But with long <span class="notranslate">reads</span>, some of the library inserts are shorter than the read length resulting in read-through to the adapter at the 3‚Äô end of the read. This microbiome sample has relatively long <span class="notranslate">reads</span> and we can see Nextera dapater has been detected.</p>

<blockquote class="details">
  <h3 id="details-other-adapter-content-profiles"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Other adapter content profiles</h3>

  <p>Adapter content may also be detected with <span class="notranslate">RNA</span>-Seq libraries where the distribution of library insert sizes is varied and likely to include some short inserts.</p>

  <p><img src="../../images/quality-control/adapter_content_rna_seq.png" alt="Adapter Content. " loading="lazy" /></p>

</blockquote>

<p>We can run an trimming tool such as Cutadapt to remove this adapter. We will explain that process in the filter and trim section.</p>

<blockquote class="comment">
  <h3 id="tip-take-a-shortcut"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Take a shortcut</h3>

  <p>The following sections go into detail about some of the other plots generated by FastQC. Note that some plots/modules may give warnings but be normal
for the type of data you‚Äôre working with, as discussed below and <a href="https://rtsf.natsci.msu.edu/genomics/tech-notes/fastqc-tutorial-and-faq/">here</a>.
The other plots give us information to more deeply understand the quality of the data, and to see if changes could be made in the lab to get higher-quality data in the future.
These sections are <strong>optional</strong>, and if you would like to skip these you can:</p>
  <ul>
    <li>Jump straight to the <a href="#trim-and-filter">next section</a> to learn about trimming paired-end data</li>
  </ul>
</blockquote>

<h4 id="per-tile-sequence-quality">Per tile sequence quality</h4>

<p>This plot enables you to look at the quality scores from each tile across all of your bases to see if there was a loss in quality associated with only one part of the flowcell. The plot shows the deviation from the average quality for each flowcell tile. The hotter colours indicate that <span class="notranslate">reads</span> in the given tile have worse qualities for that position than <span class="notranslate">reads</span> in other tiles. With this sample, you can see that certain tiles show consistently poor quality, especially from ~100bp onwards. A good plot should be blue all over.</p>

<figure id="figure-4"><img src="../../images/quality-control/per_tile_sequence_quality-before.png" alt="Per tile sequence quality. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Per tile sequence quality</figcaption></figure>

<p>This plot will only appear for Illumina library which retains its original sequence identifiers. Encoded in these is the flowcell tile from which each read came.</p>

<blockquote class="details">
  <h3 id="details-other-tile-quality-profiles"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Other tile quality profiles</h3>

  <p>In some cases, the chemicals used during sequencing becoming a bit exhausted over the time and the last tiles got worst chemicals which makes the sequencing reactions a bit error-prone. The ‚ÄúPer tile sequence quality‚Äù graph will then have some horizontal lines like this:</p>

  <p><img src="../../images/quality-control/per_tile_sequence_quality_horizontal_lines.png" alt="Per tile sequence quality with horizontal lines. " loading="lazy" /></p>

</blockquote>

<h2 id="per-sequence-quality-scores">Per sequence quality scores</h2>

<p>It plots the average quality score over the full length of all <span class="notranslate">reads</span> on the x-axis and gives the total number of <span class="notranslate">reads</span> with this score on the y-axis:</p>

<figure id="figure-5"><img src="../../images/quality-control/per_sequence_quality_scores-before.png" alt="Per sequence quality scores. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Per sequence quality scores</figcaption></figure>

<p>The distribution of average read quality should be tight peak in the upper range of the plot. It can also report if a subset of the sequences have universally low quality values: it can happen because some sequences are poorly imaged (on the edge of the field of view etc), however these should represent only a small percentage of the total sequences.</p>

<h2 id="per-base-sequence-content">Per base sequence content</h2>

<figure id="figure-6"><img src="../../images/quality-control/per_base_sequence_content-before.png" alt="Per base sequence content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Per base sequence content for a <span class="notranslate">DNA</span> library</figcaption></figure>

<p>‚ÄúPer Base Sequence Content‚Äù plots the percentage of each of the four nucleotides (T, C, A, G) at each position across all <span class="notranslate">reads</span> in the input sequence file. As for the per base sequence quality, the x-axis is non-uniform.</p>

<p>In a random library we would expect that there would be little to no difference between the four bases. The proportion of each of the four bases should remain relatively constant over the length of the read with <code class="language-plaintext highlighter-rouge">%A=%T</code> and <code class="language-plaintext highlighter-rouge">%G=%C</code>, and the lines in this plot should run parallel with each other. This is amplicon data, where 16S <span class="notranslate">DNA</span> is PCR amplified and sequenced, so we‚Äôd expect this plot to have some bias and not show a random distribution.</p>

<blockquote class="details">
  <h3 id="details-biases-by-library-type"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Biases by library type</h3>

  <p>It‚Äôs worth noting that some library types will always produce biased sequence composition, normally at the start of the read. Libraries produced by priming using random hexamers (including nearly all <span class="notranslate">RNA</span>-Seq libraries), and those which were fragmented using transposases, will contain an intrinsic bias in the positions at which <span class="notranslate">reads</span> start (the first 10-12 bases). This bias does not involve a specific sequence, but instead provides enrichment of a number of different K-mers at the 5‚Äô end of the <span class="notranslate">reads</span>. Whilst this is a true technical bias, it isn‚Äôt something which can be corrected by trimming and in most cases doesn‚Äôt seem to adversely affect the <span class="notranslate">downstream</span> analysis. It will, however, produce a warning or error in this module.</p>

  <p><img src="../../images/quality-control/per_base_sequence_content_rnaseq.png" alt="Per base sequence content for RNA-seq data. " loading="lazy" /></p>

  <p>ChIP-seq data can also encounter read start sequence biases in this plot if fragmenting with transposases. With bisulphite converted data, e.g. HiC data, a separation of G from C and A from T is expected:</p>

  <p><img src="../../images/quality-control/per_base_sequence_content_bisulphite.png" alt="Per base sequence content for Bisulphite data. " loading="lazy" /></p>

  <p>At the end, there is an overall shift in the sequence composition. If the shift correlates with a loss of sequencing quality, it can be suspected that miscalls are made with a more even sequence bias than bisulphite converted libraries.  Trimming the sequences fixed this problem, but if this hadn‚Äôt been done it would have had a dramatic effect on the methylation calls which were made.</p>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Why is there a warning for the per-base sequence content graphs?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>In the beginning of sequences, the sequence content per base is not really good and the percentages are not equal, as expected for 16S amplicon data.</p>

  </blockquote>
</blockquote>

<h2 id="per-sequence-gc-content">Per sequence GC content</h2>

<figure id="figure-7"><img src="../../images/quality-control/per_sequence_gc_content-before.png" alt="Per sequence GC content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Per sequence GC content</figcaption></figure>

<p>This plot displays the number of <span class="notranslate">reads</span> vs. percentage of bases G and C per read. It is compared to a theoretical distribution assuming an uniform GC content for all <span class="notranslate">reads</span>, expected for whole genome shotgun sequencing, where the central peak corresponds to the overall GC content of the underlying genome. Since the GC content of the genome is not known, the modal GC content is calculated from the observed data and used to build a reference distribution.</p>

<p>An unusually-shaped distribution could indicate a contaminated library or some other kind of biased subset. A shifted normal distribution indicates some systematic bias, which is independent of base position. If there is a systematic bias which creates a shifted normal distribution then this won‚Äôt be flagged as an error by the module since it doesn‚Äôt know what your genome‚Äôs GC content should be.</p>

<p>But there are also other situations in which an unusually-shaped distribution may occur. For example, with <span class="notranslate">RNA</span> sequencing there may be a greater or lesser distribution of mean GC content among transcripts causing the observed plot to be wider or narrower than an ideal normal distribution.</p>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Why is there a fail for the per sequence GC content graphs?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>There are multiple peaks. This can be indicative of unexpected contamination, such as adapter, r<span class="notranslate">RNA</span> or overrepresented sequences. Or it may be normal if it is amplicon data or you have highly abundant <span class="notranslate">RNA</span>-seq transcripts.</p>
  </blockquote>
</blockquote>

<h4 id="sequence-length-distribution">Sequence length distribution</h4>

<p>This plot shows the distribution of fragment sizes in the file which was analysed. In many cases this will produce a simple plot showing a peak only at one size, but for variable length FASTQ files this will show the relative amounts of each different size of sequence fragment. Our plot shows variable length as we trimmed the data. The biggest peak is at 296bp but there is a second large peak at ~100bp. So even though our sequences range up to 296bp in length, a lot of the good-quality sequences are shorter. This corresponds with the drop we saw in the sequence quality at ~100bp and the red stripes starting at this position in the per tile sequence quality plot.</p>

<figure id="figure-8"><img src="../../images/quality-control/sequence_length_distribution-before.png" alt="Sequence length distribution. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Sequence length distribution</figcaption></figure>

<p>Some high-throughput sequencers generate sequence fragments of uniform length, but others can contain <span class="notranslate">reads</span> of widely varying lengths. Even within uniform length libraries some pipelines will trim sequences to remove poor quality base calls from the end or the first $n$ bases if they match the first $n$ bases of the adapter up to 90% (by default), with sometimes $n = 1$.</p>

<h2 id="sequence-duplication-levels">Sequence Duplication Levels</h2>

<p>The graph shows in blue the percentage of <span class="notranslate">reads</span> of a given sequence in the file which are present a given number of times in the file:</p>

<figure id="figure-9"><img src="../../images/quality-control/sequence_duplication_levels-before.png" alt="Sequence Duplication Levels. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 9:</span> Sequence Duplication Levels</figcaption></figure>

<p>In a diverse library most sequences will occur only once in the final set. A low level of duplication may indicate a very high level of coverage of the target sequence, but a high level of duplication is more likely to indicate some kind of enrichment bias.</p>

<p>Two sources of duplicate <span class="notranslate">reads</span> can be found:</p>
<ul>
  <li>
    <p>PCR duplication in which library fragments have been over-represented due to biased PCR enrichment</p>

    <p>It is a concern because PCR duplicates misrepresent the true proportion of sequences in the input.</p>
  </li>
  <li>
    <p>Truly over-represented sequences such as very abundant transcripts in an <span class="notranslate">RNA</span>-Seq library or in amplicon data (like this sample)</p>

    <p>It is an expected case and not of concern because it does faithfully represent the input.</p>
  </li>
</ul>

<blockquote class="details">
  <h3 id="details-more-details-about-duplication"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More details about duplication</h3>

  <p>FastQC counts the degree of duplication for every sequence in a library and creates a plot showing the relative number of sequences with different degrees of duplication. There are two lines on the plot:</p>
  <ul>
    <li>Blue line: distribution of the duplication levels for the full sequence set</li>
    <li>Red line: distribution for the de-duplicated sequences with the proportions of the deduplicated set which come from different duplication levels in the original data.</li>
  </ul>

  <p>For whole genome shotgun data it is expected that nearly 100% of your <span class="notranslate">reads</span> will be unique (appearing only 1 time in the sequence data). Most sequences should fall into the far left of the plot in both the red and blue lines. This indicates a highly diverse library that was not over sequenced. If the sequencing depth is extremely high (e.g. &gt; 100x the size of the genome) some inevitable sequence duplication can appear: there are in theory only a finite number of completely unique sequence <span class="notranslate">reads</span> which can be obtained from any given input <span class="notranslate">DNA</span> sample.</p>

  <p>More specific enrichments of subsets, or the presence of low complexity contaminants will tend to produce spikes towards the right of the plot. These high duplication peaks will most often appear in the blue trace as they make up a high proportion of the original library, but usually disappear in the red trace as they make up an insignificant proportion of the deduplicated set. If peaks persist in the red trace then this suggests that there are a large number of different highly duplicated sequences which might indicate either a contaminant set or a very severe technical duplication.</p>

  <p>It is usually the case for <span class="notranslate">RNA</span> sequencing where there is some very highly abundant transcripts and some lowly abundant. It is expected that duplicate <span class="notranslate">reads</span> will be observed for high abundance transcripts:</p>

  <p><img src="../../images/quality-control/sequence_duplication_levels_rna_seq.png" alt="Sequence Duplication Levels for RNA-seq. " loading="lazy" /></p>

</blockquote>

<h2 id="over-represented-sequences">Over-represented sequences</h2>

<p>A normal high-throughput library will contain a diverse set of sequences, with no individual sequence making up a tiny fraction of the whole. Finding that a single sequence is very over-represented in the set either means that it is highly biologically significant, or indicates that the library is contaminated, or not as diverse as expected.</p>

<p>FastQC lists all of the sequence which make up more than 0.1% of the total. For each over-represented sequence FastQC will look for matches in a database of common contaminants and will report the best hit it finds. Hits must be at least 20bp in length and have no more than 1 mismatch. Finding a hit doesn‚Äôt necessarily mean that this is the source of the contamination, but may point you in the right direction. It‚Äôs also worth pointing out that many adapter sequences are very similar to each other so you may get a hit reported which isn‚Äôt technically correct, but which has a very similar sequence to the actual match.</p>

<p><span class="notranslate">RNA</span> sequencing data may have some transcripts that are so abundant that they register as over-represented sequence. With <span class="notranslate">DNA</span> sequencing data no single sequence should be present at a high enough frequency to be listed, but we can sometimes see a small percentage of adapter <span class="notranslate">reads</span>.</p>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How could we find out what the overrepreseented sequences are?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>We can BLAST overrepresented sequences to see what they are. In this case, if we take the top overrepresented sequence</p>
    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;overrep_seq1
GTGTCAGCCGCCGCGGTAGTCCGACGTGGCTGTCTCTTATACACATCTCC
</code></pre></div>    </div>
    <p>and use <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">blastn</a> against the default Nucleotide (nr/nt) database we don‚Äôt get any hits. But if we use <a href="https://www.ncbi.nlm.nih.gov/tools/vecscreen/">VecScreen</a> we see it is the Nextera adapter.</p>
    <figure id="figure-10"><img src="../../images/quality-control/vecscreen-nextera.png" alt="VecScreen. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 10:</span> Nextera adapter</figcaption></figure>
  </blockquote>
</blockquote>

<blockquote class="details">
  <h3 id="details-more-details-about-other-fastqc-plots"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More details about other FastQC plots</h3>

  <h4 id="per-base-n-content">Per base N content</h4>

  <figure id="figure-11"><img src="../../images/quality-control/per_base_n_content-before.png" alt="Per base N content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 11:</span> Per base N content</figcaption></figure>

  <p>If a sequencer is unable to make a base call with sufficient confidence, it will write an ‚ÄúN‚Äù instead of a conventional base call. This plot displays the percentage of base calls at each position or bin for which an N was called.</p>

  <p>It‚Äôs not unusual to see a very high proportion of Ns appearing in a sequence, especially near the end of a sequence. But this curve should never rises noticeably above zero. If it does this indicates a problem occurred during the sequencing run. In the example below, an error caused the instrument to be unable to call a base for approximately 20% of the <span class="notranslate">reads</span> at position 29:</p>

  <p><img src="../../images/quality-control/per_base_n_content_error.png" alt="Per base N content. " loading="lazy" /></p>

  <h4 id="kmer-content">Kmer Content</h4>

  <p>This plot not output by default. As stated in the tool form, if you want this module it needs to be enabled using a custom Submodule and limits file. With this module, FastQC does a generic analysis of all of the short nucleotide sequences of length k (kmer, with k = 7 by default) starting at each position along the read in the library to find those which do not have an even coverage through the length of your <span class="notranslate">reads</span>. Any given kmer should be evenly represented across the length of the read.</p>

  <p>FastQC will report the list of kmers which appear at specific positions with a greater frequency than expected. This can be due to different sources of bias in the library, including the presence of read-through adapter sequences building up on the end of the sequences. The presence of any overrepresented sequences in the library (such as adapter dimers) causes the kmer plot to be dominated by the kmer from these sequences. Any biased kmer due to other interesting biases may be then diluted and not easy to see.</p>

  <p>The following example is from a high-quality <span class="notranslate">DNA</span>-Seq library. The biased kmers nearby the start of the read likely are due to slight sequence dependent efficiency of <span class="notranslate">DNA</span> shearing or a result of random priming:</p>

  <figure id="figure-12"><img src="../../images/quality-control/kmer_content.png" alt="Kmer Content. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 12:</span> Kmer content</figcaption></figure>

  <p>This module can be very difficult to interpret. The adapter content plot and overrepesented sequences table are easier to interpret and may give you enough information without needing this plot. <span class="notranslate">RNA</span>-seq libraries may have highly represented kmers that are derived from highly expressed sequences. To learn more about this plot, please check the <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/11%20Kmer%20Content.html">FastQC Kmer Content documentation</a>.</p>

</blockquote>

<p>We tried to explain here there different FastQC reports and some use cases. More about this and also some common next-generation sequencing problems can be found on <a href="https://sequencing.qcfail.com/">QCFAIL.com</a></p>

<blockquote class="details">
  <h3 id="details-specific-problem-for-alternate-library-types"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Specific problem for alte<span class="notranslate">rna</span>te library types</h3>

  <h4 id="smallmicro-rna">Small/micro <span class="notranslate">RNA</span></h4>

  <p>In small <span class="notranslate">RNA</span> libraries, we typically have a relatively small set of unique, short sequences. Small <span class="notranslate">RNA</span> libraries are not randomly sheared before adding sequencing adapters to their ends: all the <span class="notranslate">reads</span> for specific classes of micro<span class="notranslate">RNA</span>s will be identical. It will result in:</p>

  <ul>
    <li>Extremely biased per base sequence content</li>
    <li>Extremely narrow distribution of GC content</li>
    <li>Very high sequence duplication levels</li>
    <li>Abundance of overrepresented sequences</li>
    <li>Read-through into adapters</li>
  </ul>

  <h4 id="amplicon">Amplicon</h4>

  <p>Amplicon libraries are prepared by PCR amplification of a specific target. For example, the V4 hypervariable region of the bacterial 16S r<span class="notranslate">RNA</span> gene. All <span class="notranslate">reads</span> from this type of library are expected to be nearly identical. It will result in:</p>

  <ul>
    <li>Extremely biased per base sequence content</li>
    <li>Extremely narrow distribution of GC content</li>
    <li>Very high sequence duplication levels</li>
    <li>Abundance of overrepresented sequences</li>
  </ul>

  <h4 id="bisulfite-or-methylation-sequencing">Bisulfite or Methylation sequencing</h4>

  <p>With Bisulfite or methylation sequencing, the majority of the cytosine (C) bases are converted to thymine (T). It will result in:</p>

  <ul>
    <li>Biased per base sequence content</li>
    <li>Biased per sequence GC content</li>
  </ul>

  <h4 id="adapter-dimer-contamination">Adapter dimer contamination</h4>

  <p>Any library type may contain a very small percentage of adapter dimer (i.e. no insert) fragments. They are more likely to be found in amplicon libraries constructed entirely by PCR (by formation of PCR primer-dimers) than in <span class="notranslate">DNA</span>-Seq or <span class="notranslate">RNA</span>-Seq libraries constructed by adapter ligation. If a sufficient fraction of the library is adapter dimer it will become noticeable in the FastQC report:</p>

  <ul>
    <li>Drop in per base sequence quality after base 60</li>
    <li>Possible bi-modal distribution of per sequence quality scores</li>
    <li>Distinct pattern observed in per bases sequence content up to base 60</li>
    <li>Spike in per sequence GC content</li>
    <li>Overrepresented sequence matching adapter</li>
    <li>Adapter content &gt; 0% starting at base 1</li>
  </ul>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-bad-quality-sequences"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Bad quality sequences</h3>
  <p>If the quality of the <span class="notranslate">reads</span> is not good, we should always first check what is wrong and think about it: it may come from the type of sequencing or what we sequenced (high quantity of overrepresented sequences in transcriptomics data, biased percentage of bases in HiC data).</p>

  <p>You can also ask the sequencing facility about it, especially if the quality is really bad: the quality treatments can not solve everything. If too many bad quality bases are cut away, the corresponding <span class="notranslate">reads</span> then will be filtered out and you lose them.</p>
</blockquote>

<h1 id="trim-and-filter">Trim and filter</h1>

<p>The quality drops in the middle of these sequences. This could cause bias in <span class="notranslate">downstream</span> analyses with these potentially incorrectly called nucleotides. Sequences must be treated to reduce bias in <span class="notranslate">downstream</span> analysis. Trimming can help to increase the number of <span class="notranslate">reads</span> the aligner or assembler are able to succesfully use, reducing the number of <span class="notranslate">reads</span> that are unmapped or unassembled. In general, quality treatments include:</p>

<ol>
  <li>Trimming/cutting/masking sequences
    <ul>
      <li>from low quality score regions</li>
      <li>beginning/end of sequence</li>
      <li>removing adapters</li>
    </ul>
  </li>
  <li>Filtering of sequences
    <ul>
      <li>with low mean quality score</li>
      <li>too short</li>
      <li>with too many ambiguous (N) bases</li>
    </ul>
  </li>
</ol>

<p>To accomplish this task we will use <a href="https://cutadapt.readthedocs.io/en/stable/guide.html">Cutadapt</a> <span class="citation"><a href="#marcel2011cutadapt">Marcel 2011</a></span>, a tool that enhances sequence quality by automating adapter trimming as well as quality control.  We will:</p>

<ul>
  <li>Trim low-quality bases from the ends. Quality trimming is done before any adapter trimming. We will set the quality threshold as 20, a commonly used threshold, see more <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531872-Phred-scaled-quality-scores">here</a>.</li>
  <li>Trim adapter with Cutadapt. For that we need to supply the sequence of the adapter. In this sample, Nextera is the adapter that was detected. We can find the sequence of the Nextera adapter on the <a href="https://support.illumina.com/bulletins/2016/12/what-sequences-do-i-use-for-adapter-trimming.html">Illumina website here</a> <code class="language-plaintext highlighter-rouge">CTGTCTCTTATACACATCT</code>. We will trim that sequence from the 3‚Äô end of the <span class="notranslate">reads</span>.</li>
  <li>Filter out sequences with length &lt; 20 after trimming</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-improvement-of-sequence-quality"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Improvement of sequence quality</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy1"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters
      <ul>
        <li><em>‚ÄúSingle-end or Paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li>
              <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">Reads</span> in FASTQ format‚Äù</em>: <code class="language-plaintext highlighter-rouge">female_oral2.fastq-4143.gz</code> (Input dataset)</p>

              <blockquote class="tip">
                <h3 id="tip-tip-files-not-selectable"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Files not selectable?</h3>
                <p>If your FASTQ file cannot be selected, you might check whether the format is FASTQ with Sanger-scaled quality values (<code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>). You can edit the data type by clicking on the pencil symbol.</p>
              </blockquote>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúRead 1 Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúInsert 3‚Äô (End) Adapters‚Äù</em>:
              <ul>
                <li><em>‚ÄúSource‚Äù</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code></li>
                <li><em>‚ÄúEnter custom 3‚Äô adapter sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">CTGTCTCTTATACACATCT</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúFilter Options‚Äù</em>
          <ul>
            <li><em>‚ÄúMinimum length‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúRead Modification Options‚Äù</em>
          <ul>
            <li><em>‚ÄúQuality cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúOutputs selector‚Äù</em>: <code class="language-plaintext highlighter-rouge">Report</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt file (<code class="language-plaintext highlighter-rouge">Report</code>)</p>

      <blockquote class="question">
        <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What % <span class="notranslate">reads</span> contain adapter?</li>
          <li>What % <span class="notranslate">reads</span> have been trimmed because of bad quality?</li>
          <li>What % <span class="notranslate">reads</span> have been removed because they were too short?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>58.6% <span class="notranslate">reads</span> contain adapter (<code class="language-plaintext highlighter-rouge"><span class="notranslate">Reads</span> with adapters:</code>)</li>
            <li>35.1% <span class="notranslate">reads</span> have been trimmed because of bad quality (<code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>)</li>
            <li>0 % <span class="notranslate">reads</span> were removed because they were too short</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="details">
  <h3 id="details-trimming-with-cutadapt"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Trimming with Cutadapt</h3>

  <p>One of the biggest advantage of Cutadapt compared to other trimming tools (e.g. TrimGalore!) is that it has a good <a href="https://cutadapt.readthedocs.io">documentation</a> explaining how the tool works in detail.</p>

  <p>Cutadapt quality trimming algorithm consists of three simple steps:</p>

  <ol>
    <li>Subtract the chosen threshold value from the quality value of each position</li>
    <li>Compute a partial sum of these differences from the end of the sequence to each position
(as long as the partial sum is negative)</li>
    <li>Cut at the minimum value of the partial sum</li>
  </ol>

  <p>In the following example, we assume that the 3‚Äô end is to be quality-trimmed with a threshold of 10 and we have the following quality values</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>42 40 26 27 8 7 11 4 2 3
</code></pre></div>  </div>

  <ol>
    <li>
      <p>Subtract the threshold</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> 32 30 16 17 -2 -3 1 -6 -8 -7
</code></pre></div>      </div>
    </li>
    <li>
      <p>Add up the numbers, starting from the 3‚Äô end (partial sums) and stop early if the sum is greater than zero</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> (70) (38) 8 -8 -25 -23 -20, -21 -15 -7
</code></pre></div>      </div>

      <p>The numbers in parentheses are not computed (because 8 is greater than zero), but shown here for completeness.</p>
    </li>
    <li>
      <p>Choose the position of the minimum (<code class="language-plaintext highlighter-rouge">-25</code>) as the trimming position</p>
    </li>
  </ol>

  <p>Therefore, the read is trimmed to the first four bases, which have quality values</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>42 40 26 27
</code></pre></div>  </div>

  <p>Note that therefore, positions with a quality value larger than the chosen threshold are also removed if they are embedded in regions with lower quality (the partial sum is decreasing if the quality values are smaller than the threshold). The advantage of this procedure is that it is robust against a small number of positions with a quality higher than the threshold.</p>

  <p>Alte<span class="notranslate">rna</span>tives to this procedure would be:</p>

  <ul>
    <li>Cut after the first position with a quality smaller than the threshold</li>
    <li>
      <p>Sliding window approach</p>

      <p>The sliding window approach checks that the average quality of each sequence window of specified length is larger than the threshold. Note that in contrast to cutadapt‚Äôs approach, this approach has one more parameter and the robustness depends of the length of the window (in combination with the quality threshold). Both approaches are implemented in Trimmomatic.</p>
    </li>
  </ul>
</blockquote>

<p>We can examine our trimmed data with FASTQE and/or FastQC.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-checking-quality-after-trimming"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Checking quality after trimming</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+galaxy2"><strong>FASTQE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/fastqe/fastqe/0.2.6+<span class="notranslate">galaxy</span>2</span></span> : Re-run <strong>FASTQE</strong> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúFastQ data‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cutadapt Read 1 Output</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúScore types to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mean</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the new FASTQE report</p>

      <blockquote class="question">
        <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>Compare the FASTQE output to the previous one before trimming above. Has sequence quality been improved?</p>

        <!--SNIPPET-->
        <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-using-the-scratchbook-to-view-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using the Scratchbook to view multiple datasets</h3>  <p>If you would like to view two or more datasets at once, you can use the <strong>Scratchbook</strong> feature in <span class="notranslate">Galaxy</span>:</p>  <ol>  <li><strong>Click</strong> on the <em>Scratchbook</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> on the top menu bar.    <ul>      <li>You should see a little checkmark on the icon now</li>    </ul>  </li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> to view the output    <ul>      <li>You should see the output in a window overlayed over <span class="notranslate">Galaxy</span></li>      <li>You can resize this window by dragging the bottom-right corner</li>    </ul>  </li>  <li><strong>Click</strong> outside the file to exit the Scratchbook</li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a second dataset from your history    <ul>      <li>You should now see a second window with the new dataset</li>      <li>This makes it easier to compare the two outputs</li>    </ul>  </li>  <li>Repeat this for as many files as you would like to compare</li>  <li>You can turn off the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> by clicking on the icon again</li></ol></blockquote>

        <blockquote class="solution">
          <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>Yes, the quality score emojis look better (happier) now.</p>

          <figure id="figure-13"><img src="../../images/quality-control/fastqe-mean-before.png" alt="FASTQE before. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 13:</span> Before trimming</figcaption></figure>

          <figure id="figure-14"><img src="../../images/quality-control/fastqe-mean-after.png" alt="FASTQE after. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 14:</span> After trimming</figcaption></figure>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>With FASTQE we can see we improved the quality of the bases in the dataset.</p>

<p>We can also, or instead, check the quality-controlled data with FastQC.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-checking-quality-after-trimming-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Checking quality after trimming</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúShort read data from your current history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cutadapt Read 1 Output</code></li>
      </ul>
    </li>
    <li>Inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <ol>
    <li>Does the per base sequence quality look better?</li>
    <li>Is the adapter gone?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Yes. The vast majority of the bases have a quality score above 20 now.</li>
    </ol>
    <figure id="figure-15"><img src="../../images/quality-control/per_base_sequence_quality-after.png" alt="Per base sequence quality. " loading="lazy" /><figcaption><span class="figcaption-prefix">Figure 15:</span> Per base sequence quality</figcaption></figure>

    <ol>
      <li>Yes. No adapter is detected now.
<img src="../../images/quality-control/adapter_content-after.png" alt="Adapter Content. " loading="lazy" /></li>
    </ol>

  </blockquote>
</blockquote>

<p>With FastQC we can see we improved the quality of the bases in the dataset and removed the adapter.</p>

<blockquote class="details">
  <h3 id="details-other-fastqc-plots-after-trimming"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Other FastQC plots after trimming</h3>

  <p><img src="../../images/quality-control/per_tile_sequence_quality-after.png" alt="Per tile sequence quality. " loading="lazy" />
We have some red stripes as we‚Äôve trimmed those regions from the <span class="notranslate">reads</span>.</p>

  <p><img src="../../images/quality-control/per_sequence_quality_scores-after.png" alt="Per sequence quality scores. " loading="lazy" />
We now have one peak of high quality instead of one high and one lower quality that we had previously.</p>

  <p><img src="../../images/quality-control/per_base_sequence_content-after.png" alt="Per base sequence content. " loading="lazy" />
We don‚Äôt have equal representation of the bases as before as this is amplicon data.</p>

  <p><img src="../../images/quality-control/per_sequence_gc_content-after.png" alt="Per sequence GC content. " loading="lazy" />
We now have a single main GC peak due to removing the adapter.</p>

  <p><img src="../../images/quality-control/per_base_n_content-after.png" alt="Per base N content. " loading="lazy" />
This is the same as before as we don‚Äôt have any Ns in these <span class="notranslate">reads</span>.</p>

  <p><img src="../../images/quality-control/sequence_length_distribution-after.png" alt="Sequence length distribution. " loading="lazy" />
We now have multiple peaks and a range of lengths, instead of the single peak with had before trimming when all sequences were the same length.</p>

  <p><img src="../../images/quality-control/sequence_duplication_levels-after.png" alt="Sequence Duplication Levels. " loading="lazy" /></p>
  <blockquote class="question">
    <h3 id="question-questions-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>What does the top overrepresented sequence <code class="language-plaintext highlighter-rouge">GTGTCAGCCGCCGCGGTAGTCCGACGTGG</code> correspond to?</p>

    <blockquote class="solution">
      <h3 id="solution-solution-10"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <p>If we take the top overrepresented sequence</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;overrep_seq1_after
GTGTCAGCCGCCGCGGTAGTCCGACGTGG
</code></pre></div>      </div>
      <p>and use <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">blastn</a> against the default Nucleotide (nr/nt) database we see the top hits are to 16S r<span class="notranslate">RNA</span> genes. This makes sense as this is 16S amplicon data, where the 16S gene is PCR amplified.</p>
    </blockquote>
  </blockquote>
</blockquote>

<h1 id="processing-multiple-datasets">Processing multiple datasets</h1>

<h2 id="process-paired-end-data">Process paired-end data</h2>

<p>With paired-end sequencing, the fragments are sequenced from both sides. This approach results in two <span class="notranslate">reads</span> per fragment, with the first read in forward orientation and the second read in reverse-complement orientation. With this technique, we have the advantage to get more information about each <span class="notranslate">DNA</span> fragment compared to <span class="notranslate">reads</span> sequenced by only single-end sequencing:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>    ------&gt;                       [single-end]

    ----------------------------- [fragment]

    ------&gt;               &lt;------ [paired-end]
</code></pre></div></div>
<p>The distance between both <span class="notranslate">reads</span> is known and therefore is additional information that can improve read <span class="notranslate">mapping</span>.</p>

<p>Paired-end sequencing generates 2 FASTQ files:</p>
<ul>
  <li>One file with the sequences corresponding to <strong>forward</strong> orientation of all the fragments</li>
  <li>One file with the sequences corresponding to <strong>reverse</strong> orientation of all the fragments</li>
</ul>

<p>Usually we recognize these two files which belong to one sample by the name which has the same identifier for the <span class="notranslate">reads</span> but a different extension, e.g. <code class="language-plaintext highlighter-rouge">sampleA_R1.fastq</code> for the forward <span class="notranslate">reads</span> and <code class="language-plaintext highlighter-rouge">sampleA_R2.fastq</code> for the reverse <span class="notranslate">reads</span>. It can also be <code class="language-plaintext highlighter-rouge">_f</code> or <code class="language-plaintext highlighter-rouge">_1</code> for the forward <span class="notranslate">reads</span> and <code class="language-plaintext highlighter-rouge">_r</code> or <code class="language-plaintext highlighter-rouge">_2</code> for the reverse <span class="notranslate">reads</span>.</p>

<p>The data we analyzed in the previous step was single-end data so we will import a paired-end <span class="notranslate">RNA</span>-seq dataset to use. We will run FastQC and aggregate the two reports with MultiQC <span class="citation"><a href="#ewels2016multiqc">Ewels <i>et al.</i> 2016</a></span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assessing-the-quality-of-paired-end-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Assessing the quality of paired-end <span class="notranslate">reads</span></h3>

  <ol>
    <li>
      <p>Import the paired-end <span class="notranslate">reads</span> <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_1.fastq</code> and <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_2.fastq</code> from <a href="https://zenodo.org/record/61771">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/61771/files/GSM461178_untreat_paired_subset_1.fastq
https://zenodo.org/record/61771/files/GSM461178_untreat_paired_subset_2.fastq
</code></pre></div>      </div>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FASTQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  with both datasets</p>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+galaxy1"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.9+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters to aggregate the FastQC reports of both forward and reverse <span class="notranslate">reads</span>
      <ul>
        <li>In <em>‚ÄúResults‚Äù</em>
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>‚ÄúFastQC output‚Äù</em>
              <ul>
                <li><em>‚ÄúType of FastQC output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúFastQC output‚Äù</em>: <code class="language-plaintext highlighter-rouge">Raw data</code> files (output of both <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the webpage output from MultiQC</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What do you think about the quality of the sequences?</li>
    <li>What should we do?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-11"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The quality of the sequences seems worse for the reverse <span class="notranslate">reads</span> than for the forward <span class="notranslate">reads</span>:
        <ul>
          <li>Per Sequence Quality Scores: distribution more on the left, i.e. a lower mean quality of the sequences</li>
          <li>Per base sequence quality: less smooth curve and stronger decrease at the end with a mean value below 28</li>
          <li>Per Base Sequence Content: stronger bias at the beginning and no clear distinction between C-G and A-T groups</li>
        </ul>

        <p>The other indicators (adapters, duplication levels, etc) are similar.</p>
      </li>
      <li>We should trim the end of the sequences and filter them with <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
    </ol>

  </blockquote>
</blockquote>

<p>With paired-end <span class="notranslate">reads</span> the average quality scores for forward <span class="notranslate">reads</span> will almost always be higher than for reverse <span class="notranslate">reads</span>.</p>

<p>After trimming, reverse <span class="notranslate">reads</span> will be shorter because of their quality and then will be eliminated during the filtering step. If one of the reverse <span class="notranslate">reads</span> is removed, its corresponding forward read should be removed too. Otherwise we will get different number of <span class="notranslate">reads</span> in both files and in different order, and order is important for the next steps. Therefore <strong>it is important to treat the forward and reverse <span class="notranslate">reads</span> together for trimming and filtering</strong>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-improving-the-quality-of-paired-end-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Improving the quality of paired-end data</h3>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+galaxy1"><strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters
      <ul>
        <li><em>‚ÄúSingle-end or Paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTQ/A file #1‚Äù</em>: <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_1.fastq</code> (Input dataset)</li>
            <li>
              <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTQ/A file #2‚Äù</em>: <code class="language-plaintext highlighter-rouge">GSM461178_untreat_paired_subset_2.fastq</code> (Input dataset)</p>

              <p>The order is important here!</p>
            </li>
            <li>
              <p>In <em>Read 1 Options</em> or <em>Read 2 Options</em></p>

              <p>No adapters were found in these datasets. When you process your own data and you know which adapter sequences were used during library preparation, you should provide their sequences here.</p>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúFilter Options‚Äù</em>
          <ul>
            <li><em>‚ÄúMinimum length‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúRead Modification Options‚Äù</em>
          <ul>
            <li><em>‚ÄúQuality cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúOutput Options‚Äù</em>
          <ul>
            <li><em>‚ÄúReport‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt file (<code class="language-plaintext highlighter-rouge">Report</code>)</p>

      <blockquote class="question">
        <h3 id="question-questions-12"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many basepairs has been removed from the <span class="notranslate">reads</span> because of bad quality?</li>
          <li>How many sequence pairs have been removed because they were too short?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-12"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>44,164 bp (<code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>) for the forward <span class="notranslate">reads</span> and 138,638 bp for the reverse <span class="notranslate">reads</span>.</li>
            <li>1,376 sequences have been removed because at least one read was shorter than the length cutoff (322 when only the forward <span class="notranslate">reads</span> were analyzed).</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>In addition to the report, Cutadapt generates 2 files:</p>
<ul>
  <li>Read 1 with the trimmed and filtered forward <span class="notranslate">reads</span></li>
  <li>Read 2 with the trimmed and filtered reverse <span class="notranslate">reads</span></li>
</ul>

<p>These datasets can be used for the <span class="notranslate">downstream</span> analysis, e.g. <span class="notranslate">mapping</span>.</p>

<blockquote class="question">
  <h3 id="question-questions-13"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What kind of alignment is used for finding adapters in <span class="notranslate">reads</span>?</li>
    <li>What is the criterion to choose the best adapter alignment?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-13"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Semi-global alignment, i.e., only the overlapping part of the read and the adapter sequence is used for scoring.</li>
      <li>An alignment with maximum overlap is computed that has the smallest number of mismatches and indels.</li>
    </ol>

  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial we checked the quality of FASTQ files to ensure that their data looks good before inferring any further information. This step is the usual first step for analyses such as <span class="notranslate">RNA</span>-Seq, ChIP-Seq, or any other OMIC analysis relying on NGS data. Quality control steps are similar for any type of sequencing data:</p>

<ul>
  <li>Quality assessment with tools like <strong>FASTQE</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  <li>Trimming and filtering with a tool like <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
</ul>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>Perform quality control on every dataset before running any other bioinformatics analysis</p>
</li>
                        
                        <li><p>Assess the quality metrics and improve quality if necessary</p>
</li>
                        
                        <li><p>Check the impact of the quality control</p>
</li>
                        
                        <li><p>Different tools are available to provide additional quality metrics</p>
</li>
                        
                        <li><p>For paired-end reads analyze the forward and reverse reads together</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/sequence-analysis/faqs/">FAQ page for the Sequence analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
                

                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="marcel2011cutadapt">Marcel, M., 2011 <b>Cutadapt removes adapter sequences from high-throughput sequencing reads</b>. EMBnet.journal 17: 10‚Äì12. <a href="http://journal.embnet.org/index.php/embnetjournal/article/view/200">http://journal.embnet.org/index.php/embnetjournal/article/view/200</a></li>
<li id="ewels2016multiqc">Ewels, P., M. Magnusson, S. Lundin, and M. K√§ller, 2016 <b>MultiQC: summarize analysis results for multiple tools and samples in a single report</b>. Bioinformatics 32: 3047‚Äì3048. <a href="https://academic.oup.com/bioinformatics/article/32/19/3047/2196507">https://academic.oup.com/bioinformatics/article/32/19/3047/2196507</a></li>
<li id="StJacques2021">Jacques, R. M. S., W. M. Maza, S. D. Robertson, A. Lonsdale, C. S. Murray <i>et al.</i>, 2021 <b>A Fun Introductory Command Line Lesson: Next Generation Sequencing Quality Analysis with Emoji!</b> CourseSource 8: <a href="https://doi.org/10.24918/cs.2021.17">10.24918/cs.2021.17</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Quality Control (Sequence analysis)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">B√©r√©nice Batut, Maria Doyle, 2021 <b>Quality Control (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{sequence-analysis-quality-control,
author = "B√©r√©nice Batut and Maria Doyle",
title = "Quality Control (Galaxy Training Materials)",
year = "2021",
month = "09",
day = "01"
url = "\url{https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                
                <blockquote class="agenda follow-up">
                    <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
                    <ul>
                        
    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

                    </ul>
                </blockquote>
                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
