<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Differential abundance testing of small RNAs</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Differential abundance testing of small RNAs" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fsrna%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/srna/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Differential abundance testing of small RNAs",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / srna / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Differential abundance testing of small RNAs' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/srna/tutorial.html",
  "timeRequired": "PT3H",
  "description": "The questions this  addresses are:\n - What small RNAs are expressed?\n - What RNA features have significantly different numbers of small RNAs targeting them between two conditions?\n\n\\nThe objectives are:\n - Process small RNA-seq datasets to determine quality and reproducibility.\n - Filter out contaminants (e.g. rRNA reads) in small RNA-seq datasets.\n - Differentiate between subclasses of small RNAs based on their characteristics.\n - Identify differently abundant small RNAs and their targets.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Differential abundance testing of small RNAs</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/malloryfreeberg/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/malloryfreeberg" alt="Mallory Freeberg">Mallory Freeberg</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What small RNAs are expressed?</p>
</li>
            
            <li><p>What RNA features have significantly different numbers of small RNAs targeting them between two conditions?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Process small RNA-seq datasets to determine quality and reproducibility.</p>
</li>
            
            <li><p>Filter out contaminants (e.g. rRNA reads) in small RNA-seq datasets.</p>
</li>
            
            <li><p>Differentiate between subclasses of small RNAs based on their characteristics.</p>
</li>
            
            <li><p>Identify differently abundant small RNAs and their targets.</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/826906">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/srna/workflows/" title="Workflows" alt="Differential abundance testing of small RNAs workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/srna/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Small, noncoding <span class="notranslate">RNA</span> (s<span class="notranslate">RNA</span>) molecules, typically 18-40nt in length, are key features of post-transcriptional regulatory mechanisms governing gene <span class="notranslate">expression</span>. Through interactions with protein cofactors, these tiny s<span class="notranslate">RNA</span>s typically function by perfectly or imperfectly basepairing with substrate <span class="notranslate">RNA</span> molecules, and then eliciting <span class="notranslate">downstream</span> effects such as translation inhibition or <span class="notranslate">RNA</span> degradation. Different subclasses of s<span class="notranslate">RNA</span>s - <em>e.g.</em> micro<span class="notranslate">RNA</span>s (mi<span class="notranslate">RNA</span>s), Piwi-interaction <span class="notranslate">RNA</span>s (pi<span class="notranslate">RNA</span>s), and endogenous short interferring <span class="notranslate">RNA</span>s (si<span class="notranslate">RNA</span>s) - exhibit unique characteristics, and their relative abundances in biological contexts can indicate whether they are active or not. In this tutorial, we will examine <span class="notranslate">expression</span> of the pi<span class="notranslate">RNA</span> subclass of s<span class="notranslate">RNA</span>s and their targets in <em>Drosophila melanogaster</em>.</p>

<p>The data used in this tutorial are from polyphosphatase-treated s<span class="notranslate">RNA</span> sequencing (s<span class="notranslate">RNA</span>-seq) experiments in <em>Drosophila</em>. The goal of this study was to determine how si<span class="notranslate">RNA</span> <span class="notranslate">expression</span> changes in flies treated with <span class="notranslate">RNA</span> interference (<span class="notranslate">RNA</span>i) to knock down Symplekin, which is a component of the core cleaveage completx. To that end, in addition to s<span class="notranslate">RNA</span>-seq, m<span class="notranslate">RNA</span>-seq experiments were performed to determine whether targets of differentially expressed si<span class="notranslate">RNA</span>s were also differentially expressed. The original published study can be found <a href="https://www.ncbi.nlm.nih.gov/pubmed/28415970">here</a>. Because of the long processing time for the large original files - which contained 7-22 million <span class="notranslate">reads</span> each - we have downsampled the original input data to include only a subset of usable <span class="notranslate">reads</span>.</p>

<h1 id="analysis-strategy">Analysis strategy</h1>

<p>In this exercise we will identify what small <span class="notranslate">RNA</span>s are present in <em>Drosophila</em> Dmel-2 tissue culture cells treated with either <span class="notranslate">RNA</span>i against  core cleavage complex component Symplekin or blank (control) <span class="notranslate">RNA</span>i (published data available in GEO at <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE82128">GSE82128</a>). In this study, biological triplicate s<span class="notranslate">RNA</span>- and m<span class="notranslate">RNA</span>-seq libraries were sequenced for both <span class="notranslate">RNA</span>i conditions. After removing contaminant ribosomal <span class="notranslate">RNA</span> (r<span class="notranslate">RNA</span>) and mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span>, we will identify and quantify endogenous si<span class="notranslate">RNA</span>s from sequenced <span class="notranslate">reads</span> and test for differential abundance. We will follow a popular small <span class="notranslate">RNA</span> analysis pipeline developed for pi<span class="notranslate">RNA</span>s by the Phillip Zamore Lab and the ZLab (Zhiping Weng) at UMass Med School called <a href="https://github.com/bowhan/piPipes">PiPipes</a>. Although PiPipes was developed for analysis of pi<span class="notranslate">RNA</span>s, many of the basical principles can be applied to other classes of small <span class="notranslate">RNA</span>s.</p>

<p>It is of note that this tutorial uses datasets that have been de-multiplexed so that the input data files are a single FASTQ formatted file for each sample. Because s<span class="notranslate">RNA</span>s are typically much smaller than fragments generated for <span class="notranslate">RNA</span>-seq or other types of deep sequencing experiments, single-end sequencing strategies are almost always used to sequence s<span class="notranslate">RNA</span>s. This tutorial uses the <em>Collections</em> feature of <span class="notranslate">Galaxy</span> to orgainze each set of replicates into a single group, making tool form submission easier and ensuring reproducibility.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#analysis-strategy" id="markdown-toc-analysis-strategy">Analysis strategy</a>    <ol>
      <li><a href="#data-upload-and-organization" id="markdown-toc-data-upload-and-organization">Data upload and organization</a></li>
      <li><a href="#read-quality-checking" id="markdown-toc-read-quality-checking">Read quality checking</a></li>
      <li><a href="#adaptor-trimming" id="markdown-toc-adaptor-trimming">Adaptor trimming</a></li>
      <li><a href="#hierarchical-read-alignment-to-remove-rrnamirna-reads" id="markdown-toc-hierarchical-read-alignment-to-remove-rrnamirna-reads">Hierarchical read alignment to remove r<span class="notranslate">RNA</span>/mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span></a></li>
      <li><a href="#small-rna-subclass-distinction" id="markdown-toc-small-rna-subclass-distinction">Small <span class="notranslate">RNA</span> subclass distinction</a></li>
      <li><a href="#sirna-abundance-estimation" id="markdown-toc-sirna-abundance-estimation">si<span class="notranslate">RNA</span> abundance estimation</a></li>
      <li><a href="#sirna-differential-abundance-testing" id="markdown-toc-sirna-differential-abundance-testing">si<span class="notranslate">RNA</span> differential abundance testing</a></li>
      <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a></li>
      <li><a href="#small-rna-analysis-general-pipeline" id="markdown-toc-small-rna-analysis-general-pipeline">Small <span class="notranslate">RNA</span> analysis (general) pipeline</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h2 id="data-upload-and-organization">Data upload and organization</h2>

<p>Due to the large size of the original s<span class="notranslate">RNA</span>-seq datasets, we have downsampled them to only inlcude a subset of <span class="notranslate">reads</span>. These datasets are available at <a href="https://doi.org/10.5281/zenodo.826906"><code class="language-plaintext highlighter-rouge">Zenodo</code></a> where you can find the FASTQ files corresponding to replicate s<span class="notranslate">RNA</span>-seq experiments and additional annotation files for the <em>Drosophila melanogaster</em> genome version dm3.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload and organization</h3>

  <ol>
    <li>
      <p>Create a new history and name it something meaningful (<em>e.g.</em> s<span class="notranslate">RNA</span>-seq tutorial)</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import the 3 <code class="language-plaintext highlighter-rouge">Blank_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq</code> and 3 <code class="language-plaintext highlighter-rouge">Symp_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq</code> FASTQ files from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/826906/files/Blank_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq_rep1_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Blank_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq_rep2_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Blank_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq_rep3_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Symp_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq_rep1_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Symp_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq_rep2_downsampled.fastqsanger.gz
https://zenodo.org/record/826906/files/Symp_<span class="notranslate">RNA</span>i_s<span class="notranslate">RNA</span>-seq_rep3_downsampled.fastqsanger.gz
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>

      <p>Set the datatype of the read (.fastqsanger) files to <strong>fastq</strong>
As default, <span class="notranslate">Galaxy</span> takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Build a dataset list for both sets (<code class="language-plaintext highlighter-rouge">Blank</code> and <code class="language-plaintext highlighter-rouge">Symp</code>) of replicate FASTQ files</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a dataset collection</h3>

        <ul>
          <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="../../../galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>
          <li>Check all the datasets in your history you would like to include</li>
          <li>
            <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>

            <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>
          </li>
          <li>Enter a name for your collection</li>
          <li>Click <strong>Create List</strong> to build your collection</li>
          <li>Click on the checkmark icon at the top of your history again</li>
        </ul>
      </blockquote>

      <p><img src="../../images/sRNA/Fig3_dataset_lists_made.png" alt="History section with all the loaded datasets" /></p>
    </li>
    <li>
      <p>Import the remaining files from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/826906/files/dm3_transcriptome_Tx2Gene_downsampled.tab.gz
https://zenodo.org/record/826906/files/dm3_transcriptome_sequences_downsampled.fa.gz
https://zenodo.org/record/826906/files/dm3_mi<span class="notranslate">RNA</span>_hairpin_sequences.fa.gz
https://zenodo.org/record/826906/files/dm3_r<span class="notranslate">RNA</span>_sequences.fa.gz
</code></pre></div>      </div>

      <ul>
        <li>Set the datatype of the annotation file to <strong>tab</strong> and assign the Genome as <strong>dm3</strong></li>
        <li>Set the datatype of the reference files to <strong>fasta</strong> and assign the Genome as <strong>dm3</strong></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="read-quality-checking">Read quality checking</h2>

<p>Read quality scores (phred scores) in FASTQ-formatted data can be encoded by one of a few different encoding schemes. Most <span class="notranslate">Galaxy</span> tools assume that input FASTQ files are using the Sanger/Illumina 1.9 encoding scheme. If the input FASTQ files are using an alte<span class="notranslate">rna</span>te encoding scheme, then some tools will not interpret the quality score encodings correctly. It is good practice to confirm the quality encoding scheme of your data and then convert to Sanger/Illumina 1.9, if necessary. We can check the quality encoding scheme using the <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> tool (further described in the <a href="/training-material/topics/sequence-analysis">NGS-QC tutorial</a>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-checking"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality checking</h3>

  <ol>
    <li>Run <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the <code class="language-plaintext highlighter-rouge">Blank</code> collection
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúShort read data from your current history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-selecting-a-dataset-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Selecting a dataset collection as input</h3>

        <ol>
          <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in
front of the input parameter you want to supply the collection to.</li>
          <li>Select the collection you want to use from the list</li>
        </ol>
      </blockquote>
    </li>
    <li>Run <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
    <li>
      <p>Inspect the <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> reports</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What quality score encoding scheme is being used for each sample?</li>
          <li>What is the read length for each sample?</li>
          <li>
            <p>What does the base/read quality look like for each sample?</p>

            <p><img src="../../images/sRNA/Fig5a_fastqc_result_stats_boxplot.png" alt="The output of FastQC as a boxplot" /></p>
          </li>
          <li>
            <p>Are there any adaptors present in these <span class="notranslate">reads</span>? Which one(s)?</p>

            <p><img src="../../images/sRNA/Fig5b_fastqc_result_stats_adapter.png" alt="The output of FastQC stats adapted" /></p>
          </li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>All samples use the Illumina 1.9 quality encoding scheme, so we do <strong>not</strong> need to convert.</li>
            <li>All samples have a read length of 50 nt.</li>
            <li>The base quality across the entire length of the <span class="notranslate">reads</span> is good (phred score &gt; 28 for the most part).</li>
            <li>Yes, the ‚ÄúIllumina Universal Adapter‚Äù is present.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-formatting"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Formatting</h3>

  <p>If your data are not in <strong>Sanger / Illumina 1.9 format</strong>:</p>
  <ol>
    <li><strong>FASTQ Groomer</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFile to groom‚Äù</em>: input collection</li>
        <li><em>‚ÄúInput FASTQ quality scores type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Illumina 1.3-1.7</code></li>
      </ul>
    </li>
    <li>Check that the format is <strong>fastqsanger</strong> instead of <strong>fastq</strong></li>
  </ol>

  <p>If your data are already in <strong>Sanger / Illumina 1.9 format</strong>:</p>
  <ol>
    <li>
      <p>Change the datatype to <code class="language-plaintext highlighter-rouge">fastqsanger</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>Go back to the <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and scroll down to the ‚ÄúAdapter Content‚Äù section. You can see that Illumina Universal Adapters are present in <strong>&gt;80%</strong> of the <span class="notranslate">reads</span>. The next step is to remove these artificial adaptors because they are not part of the biological s<span class="notranslate">RNA</span>s. If a different adapter is present, you can update the <strong>Adapter sequence to be trimmed off</strong> in the <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step described below.</p>

<h2 id="adaptor-trimming">Adaptor trimming</h2>

<p>s<span class="notranslate">RNA</span>-seq library preparation involves adding an artificial adaptor sequence to both the 5‚Äô and 3‚Äô ends of the small <span class="notranslate">RNA</span>s. While the 5‚Äô adaptor anchors <span class="notranslate">reads</span> to the sequencing surface and thus are not sequenced, the 3‚Äô adaptor is typically sequenced immediately following the s<span class="notranslate">RNA</span> sequence. In the example datasets here, the 3‚Äô adaptor sequence is identified as the Illumina Universal Adapter, and needs to be removed from each read before aligning to a reference. We will be using the <span class="notranslate">Galaxy</span> tool <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> which implements the <a href="https://cutadapt.readthedocs.io/en/stable/"><code class="language-plaintext highlighter-rouge">cutadapt</code></a> tool for adapter trimming.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adaptor-trimming"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adaptor trimming</h3>

  <ol>
    <li><strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of FASTQ read files to remove Illumina adapters from the 3‚Äô ends of <span class="notranslate">reads</span> with the following parameters:
      <ul>
        <li><em>‚ÄúIs this library paired- or single-end‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚Äú<span class="notranslate">Reads</span> in FASTQ format?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
            <li><em>‚ÄúAdapter sequence to be trimmed‚Äù</em>: <code class="language-plaintext highlighter-rouge">Illumina universal</code></li>
          </ul>
        </li>
        <li><em>‚ÄúTrim Galore! advanced settings‚Äù</em>: <code class="language-plaintext highlighter-rouge">Full parameter list</code>
          <ul>
            <li><em>‚ÄúTrim low-quality ends from <span class="notranslate">reads</span> in addition to adapter removal‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
            <li><em>‚ÄúOverlap with adapter sequence required to trim a sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">6</code></li>
            <li><em>‚ÄúDiscard <span class="notranslate">reads</span> that became shorter than length N‚Äù</em>: <code class="language-plaintext highlighter-rouge">12</code></li>
            <li><em>‚ÄúGenerate a report file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
  </ol>

</blockquote>

<p>We don‚Äôt want to trim for quality because the adapter-trimmed sequences represent a full small <span class="notranslate">RNA</span> molecule, and we want to maintain the integrity of the entire molecule. We increase the minimum read length required to keep a read because small <span class="notranslate">RNA</span>s can potentially be shorter than 20 nt (the default value). We can check out the generated report file for any sample and see the command for the tool, a summary of the total <span class="notranslate">reads</span> processed and number of <span class="notranslate">reads</span> with an adapter identified, and histogram data of the length of adaptor trimmed. We also see that a very small percentage of low-quality bases have been trimmed</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adaptor-trimming-check"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adaptor trimming check</h3>

  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of trimmed FASTQ read files to assess whether adapters were successfully removed.</li>
    <li>
      <p>Inspect the <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> reports</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the read length?</li>
        </ol>

        <p><img src="../../images/sRNA/Fig8a_fastqc_post_trimming_result.png" alt="FastQC output after trimming" /></p>

        <ol>
          <li>Are there any adaptors present in these <span class="notranslate">reads</span>? Which one(s)?</li>
        </ol>

        <p><img src="../../images/sRNA/Fig8b_fastqc_post_trimming_lengths.png" alt="FastQC output after trimming" /></p>
        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The read lengths range from 12 to 50 nt after trimming.</li>
            <li>No, Illumina Universal Adaptors are no longer present. No other adapters are present.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>An interesting thing to note from our <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> results is the <em>Sequence Length Distribution</em> results. While many <span class="notranslate">RNA</span>-seq experiments have normal distribution of read lengths, an unusual spike at 22nt is observed in our data. This spike represents the large set of endogenous si<span class="notranslate">RNA</span>s that occur in the cell line used in this study, and is actually confirmation that our dataset captures the biological molecule we are interested in.</p>

<p>Now that we have converted to <em>fastqsanger</em> format and trimmed our <span class="notranslate">reads</span> of the Illumina Universal Adaptors, we will align our trimmed <span class="notranslate">reads</span> to reference <em>Drosophila</em> r<span class="notranslate">RNA</span> and mi<span class="notranslate">RNA</span> sequences (dm3) to remove these artifacts. Interestingly, <em>Drosophila</em> have a short 2S r<span class="notranslate">RNA</span> sequence that is 30nt long and typically co-migrates with s<span class="notranslate">RNA</span> populations during gel electrophoresis. r<span class="notranslate">RNA</span>s make up a very large proportion of all non-coding <span class="notranslate">RNA</span>s, and thus need to be removed. Oftentimes, experimental approaches can be utilized to deplete or avoid capture of r<span class="notranslate">RNA</span>s, but these methods are not always 100% efficient. We also want to remove any mi<span class="notranslate">RNA</span> sequences, as these are not relevant to our analysis. After removing r<span class="notranslate">RNA</span> and mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span>, we will analyze the remaining <span class="notranslate">reads</span>, which may be si<span class="notranslate">RNA</span> or pi<span class="notranslate">RNA</span> sequences.</p>

<h2 id="hierarchical-read-alignment-to-remove-rrnamirna-reads">Hierarchical read alignment to remove r<span class="notranslate">RNA</span>/mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span></h2>

<p>To first identify r<span class="notranslate">RNA</span>-originating <span class="notranslate">reads</span> (which we are not interested in in this case), we will align the <span class="notranslate">reads</span> to a reference set of r<span class="notranslate">RNA</span> sequences using <a href="https://ccb.jhu.edu/software/hisat2/index.shtml"><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></a>, an accurate and fast tool for aligning <span class="notranslate">reads</span> to a reference.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hierarchical-alignment-to-rrna-and-mirna-reference-sequences"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Hierarchical alignment to r<span class="notranslate">RNA</span> and mi<span class="notranslate">RNA</span> reference sequences</h3>

  <ol>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of trimmed <span class="notranslate">reads</span> to align to reference r<span class="notranslate">RNA</span> sequences with the following parameters
      <ul>
        <li><em>‚ÄúSource for the reference genome to align against‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dmel_r<span class="notranslate">RNA</span>_sequences.fa</code></li>
          </ul>
        </li>
        <li><em>‚ÄúIs this a single or paired library?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTA/Q file‚Äù</em>: output of <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
            <li><em>‚ÄúSpecify strand-specific information‚Äù</em>: <code class="language-plaintext highlighter-rouge">Forward (F)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
  </ol>
</blockquote>

<p>We now need to extract the <em>unaligned</em> <span class="notranslate">reads</span> from the output BAM file for aligning to reference mi<span class="notranslate">RNA</span> sequences. We can do this by using the <code class="language-plaintext highlighter-rouge">Filter SAM or BAM, output SAM or BAM</code> tool to obtain <span class="notranslate">reads</span> with a bit flag = 4 (meaning the read is unaligned) and then converting the filtered BAM file to FASTQ format with the <code class="language-plaintext highlighter-rouge">Convert from BAM to FastQ</code> tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hierarchical-alignment-to-rrna-and-mirna-reference-sequences-2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Hierarchical alignment to r<span class="notranslate">RNA</span> and mi<span class="notranslate">RNA</span> reference sequences (2)</h3>
  <ol>
    <li><strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on collection of HISAT2 output BAM files with the following parameters
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúSAM or BAM file to filter‚Äù</em>: output of <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
        <li><em>‚ÄúFilter on bitwise flag‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúOnly output alignments with all of these flag bits set‚Äù</em>: <code class="language-plaintext highlighter-rouge">The read in unmapped</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
    <li><strong>Convert from BAM to FastQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of filtered HISAT2 output BAM files with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúConvert the following BAM file to FASTQ‚Äù</em>: output of <strong>Filter SAM or BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
      </ul>
    </li>
    <li><strong>Convert from BAM to FastQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
  </ol>
</blockquote>

<p>Next we will align the non-r<span class="notranslate">RNA</span> <span class="notranslate">reads</span> to a known set of mi<span class="notranslate">RNA</span> hairpin sequences to identify mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-hierarchical-alignment-to-rrna-and-mirna-reference-sequences-3"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Hierarchical alignment to r<span class="notranslate">RNA</span> and mi<span class="notranslate">RNA</span> reference sequences (3)</h3>
  <ol>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of filtered HISAT2 output FASTQ files to align non-r<span class="notranslate">RNA</span> <span class="notranslate">reads</span> to reference mi<span class="notranslate">RNA</span> hairpin sequences using the following parameters:
      <ul>
        <li><em>‚ÄúSource for the reference genome to align against‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dmel_mi<span class="notranslate">RNA</span>_sequences.fa</code></li>
          </ul>
        </li>
        <li><em>‚ÄúIs this a single or paired library?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTA/Q file‚Äù</em>: output of <strong>Convert from BAM to FastQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
            <li><em>‚ÄúSpecify strand-specific information‚Äù</em>: <code class="language-plaintext highlighter-rouge">Forward (F)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
  </ol>
</blockquote>

<p>For this tutorial we are not interested in mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span>, so we need to extract <em>unaligned</em> <span class="notranslate">reads</span> from the output BAM files. To do this, repeat the <strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> steps for each dataset collection. Finally, rename the converted FASTQ files something meaningful (<em>e.g.</em> ‚Äúnon-r/mi<span class="notranslate">RNA</span> control <span class="notranslate">RNA</span>i s<span class="notranslate">RNA</span>-seq‚Äù).</p>

<p>In some instances, mi<span class="notranslate">RNA</span>-aligned <span class="notranslate">reads</span> are the desired output for <span class="notranslate">downstream</span> analyses, for example, if we are investigating how loss of key mi<span class="notranslate">RNA</span> pathway components affect levels of pre-mi<span class="notranslate">RNA</span> and mature mi<span class="notranslate">RNA</span>s. In this case, the BAM output of HISAT2 can directly be used in subsequent tools as it contains the subset of <span class="notranslate">reads</span> aligned to mi<span class="notranslate">RNA</span> sequences.</p>

<h2 id="small-rna-subclass-distinction">Small <span class="notranslate">RNA</span> subclass distinction</h2>

<p>In <em>Drosophila</em>, non-mi<span class="notranslate">RNA</span> small <span class="notranslate">RNA</span>s are typically divided into two major groups: endogenous si<span class="notranslate">RNA</span>s which are 20-22nt long and pi<span class="notranslate">RNA</span>s which are 23-29nt long. We want to analyze these s<span class="notranslate">RNA</span> subclasses independently, so next we are going to filter the non-r/mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span> based on length using the <strong>Manipulate FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-subclasses"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract subclasses</h3>

  <ol>
    <li><strong>Manipulate FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of non-r/mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span> to identify si<span class="notranslate">RNA</span>s (20-22nt) using the following parameters.
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQ File‚Äù</em>: output of last <strong>HISAT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
        <li>Click on <em>‚ÄúInsert Match <span class="notranslate">Reads</span>‚Äù</em>
          <ul>
            <li>In ‚Äú1: Match <span class="notranslate">Reads</span>‚Äù
              <ul>
                <li><em>‚ÄúMatch <span class="notranslate">Reads</span> by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sequence Content</code>
                  <ul>
                    <li><em>‚ÄúMatch by‚Äù</em>: <code class="language-plaintext highlighter-rouge">^.{12,19}$|^.{23,50}$</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Click on <em>‚ÄúManipulate <span class="notranslate">Reads</span>‚Äù</em>
          <ul>
            <li>In ‚Äú1: Manipulate <span class="notranslate">Reads</span>‚Äù
              <ul>
                <li><em>‚ÄúManipulate <span class="notranslate">Reads</span> on‚Äù</em>: <code class="language-plaintext highlighter-rouge">Miscellaneous actions</code>
                  <ul>
                    <li><em>‚ÄúMiscellaneous Manipulation Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Remove Read</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Manipulate FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
    <li>Rename each resulting dataset collection something meaningful (<em>i.e.</em> ‚Äúblank <span class="notranslate">RNA</span>i - si<span class="notranslate">RNA</span> <span class="notranslate">reads</span> (20-22nt)‚Äù)</li>
  </ol>
</blockquote>

<p>The regular <span class="notranslate">expression</span> in the <strong>Match by</strong> parameter tells the tool to identify sequences that are length 12-19 or 23-50 (inclusive), and the <strong>Miscellaneous Manipulation Type</strong> parameter tells the tool to remove these sequences. What remains are sequences of length 20-22nt. We will now repeat these steps to identify sequences in the size-range of pi<span class="notranslate">RNA</span>s (23-29nt).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-subclasses-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract subclasses</h3>
  <ol>
    <li><strong>Manipulate FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">Manipulate FASTQ</code> on each collection of non-r/mi<span class="notranslate">RNA</span> <span class="notranslate">reads</span> to identify sequences in the size-range of pi<span class="notranslate">RNA</span>s (23-29nt)
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQ File‚Äù</em>: output of last <strong>HISAT</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">Blank</code> collection</li>
        <li>Click on <em>‚ÄúInsert Match <span class="notranslate">Reads</span>‚Äù</em>
          <ul>
            <li>In ‚Äú1: Match <span class="notranslate">Reads</span>‚Äù
              <ul>
                <li><em>‚ÄúMatch <span class="notranslate">Reads</span> by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sequence Content</code>
                  <ul>
                    <li><em>‚ÄúMatch by‚Äù</em>: <code class="language-plaintext highlighter-rouge">^.{12,22}$|^.{30,50}$</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Click on <em>‚ÄúManipulate <span class="notranslate">Reads</span>‚Äù</em>
          <ul>
            <li>In ‚Äú1: Manipulate <span class="notranslate">Reads</span>‚Äù
              <ul>
                <li><em>‚ÄúManipulate <span class="notranslate">Reads</span> on‚Äù</em>: <code class="language-plaintext highlighter-rouge">Miscellaneous actions</code>
                  <ul>
                    <li><em>‚ÄúMiscellaneous Manipulation Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Remove Read</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Manipulate FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i dataset collection</li>
    <li>Rename each resulting dataset collection something meaningful (<em>i.e.</em> ‚Äúblank <span class="notranslate">RNA</span>i - pi<span class="notranslate">RNA</span> <span class="notranslate">reads</span> (23-29nt)‚Äù)</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-check-the-read-length"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Check the read length</h3>
  <ol>
    <li><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">FastQC</code> on each collection of si<span class="notranslate">RNA</span> and pi<span class="notranslate">RNA</span> read files to confirm the correct read lengths.</li>
  </ol>
</blockquote>

<p><img src="../../images/sRNA/Fig13_FastQC_piRNA_siRNA_result_Blank_rep1.png" alt="FastQC result on both collections" /></p>

<p>We see in the above image (for Blank <span class="notranslate">RNA</span>i, replicate 3) that we have ~32k <span class="notranslate">reads</span> that are 20-22nt long in our <span class="notranslate">Reads</span>_20-22_nt_length_si<span class="notranslate">RNA</span>s file and ~16k <span class="notranslate">reads</span> that are 23-29nt long in our <span class="notranslate">Reads</span>_23-29_nt_length_pi<span class="notranslate">RNA</span>s file, as expected. Given that we had ~268k trimmed <span class="notranslate">reads</span> in the Blank <span class="notranslate">RNA</span>i replicate 3 file, si<span class="notranslate">RNA</span>s represent ~12.1% of the <span class="notranslate">reads</span> and pi<span class="notranslate">RNA</span>s represent ~6.2% of the <span class="notranslate">reads</span> in this experiment. There is no distinct peak in the pi<span class="notranslate">RNA</span> range (you can find the example of the real pi<span class="notranslate">RNA</span> profile in Figure 1 from <a href="http://dx.doi.org/10.1016/j.cell.2009.03.040">Colin D. Malone et al, 2009</a>.</p>

<p>The next step in our analysis pipeline is to identify which <span class="notranslate">RNA</span> features - e.g. protein-coding m<span class="notranslate">RNA</span>s, transposable elements - the si<span class="notranslate">RNA</span>s and pi<span class="notranslate">RNA</span>s align to. Most fly pi<span class="notranslate">RNA</span>s originate from and thus align to transposable elements (TEs) but some also originate from genome pi<span class="notranslate">RNA</span> clusters or protein-coding genes. si<span class="notranslate">RNA</span>s are thought to inhibit TE mobility and potentially target m<span class="notranslate">RNA</span>s for degradation via an <span class="notranslate">RNA</span>-interference (<span class="notranslate">RNA</span>i) mechanism. Ultimately, we want to know whether an <span class="notranslate">RNA</span> feature (TE, pi<span class="notranslate">RNA</span> cluster, m<span class="notranslate">RNA</span>, etc.) has significantly different numbers of si<span class="notranslate">RNA</span>s or pi<span class="notranslate">RNA</span>s targeting it. To determine this, we will use <strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to simultaneously align and quantify si<span class="notranslate">RNA</span> <span class="notranslate">reads</span> against known target sequences to estimate si<span class="notranslate">RNA</span> abundance per target. For the remainder of the tutorial we will be focusing on si<span class="notranslate">RNA</span>s, but similar approaches can be done for pi<span class="notranslate">RNA</span>s, mi<span class="notranslate">RNA</span>s, or any other subclass of small, non-coding <span class="notranslate">RNA</span>s.</p>

<h2 id="sirna-abundance-estimation">si<span class="notranslate">RNA</span> abundance estimation</h2>

<p>We want to identify which si<span class="notranslate">RNA</span>s are differentially abundance between the blank and <em>Symplekin</em> <span class="notranslate">RNA</span>i conditions. To do this we will implement a counting approach using <strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to quantify si<span class="notranslate">RNA</span>s per <span class="notranslate">RNA</span> feature. Specifically, we will analyzing m<span class="notranslate">RNA</span> and TE elements by counting si<span class="notranslate">RNA</span>s against a FASTA file of transcript sequences, not the reference genome. This approach is especially useful in this case because we are interested in TEs, which occur at many copies (100s - 1000s) in the genome, and thus will result in multiple alignments if we align to the genome with a tool like <strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Further, we will be counting abundance of si<span class="notranslate">RNA</span>s that align <em>sense</em> or <em>antisense</em> to an <span class="notranslate">RNA</span> feature independently, as opposite sense alignments are correlated with unique <span class="notranslate">downstream</span> silencing effects. Then, we will provide this information to <strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to generate normalized counts and significance testing for differential abundance of si<span class="notranslate">RNA</span>s per feature.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-sirna-abundance-estimation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: si<span class="notranslate">RNA</span> abundance estimation</h3>

  <ol>
    <li><strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each collection of si<span class="notranslate">RNA</span> <span class="notranslate">reads</span> (20-22nt) to quantify the abundance of <em>antisense</em> si<span class="notranslate">RNA</span>s at relevant targets
      <ul>
        <li><em>‚ÄúSelect a reference transcriptome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use one from the history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference transcriptome‚Äù</em>: <code class="language-plaintext highlighter-rouge">the reference m<span class="notranslate">RNA</span> and TE fasta file</code></li>
            <li><em>‚ÄúThe size should be odd number‚Äù</em>: <code class="language-plaintext highlighter-rouge">19</code></li>
          </ul>
        </li>
        <li><em>‚ÄúIs this library mate-paired?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúFASTQ/FASTA file‚Äù</em>: collection with the Blank <span class="notranslate">RNA</span>i si<span class="notranslate">RNA</span> (20-22nt) <span class="notranslate">reads</span></li>
            <li><em>‚ÄúSpecify the strandedness of the <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">read 1 (or single-end read) comes from the reverse strand (SR)</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúAdditional Options‚Äù</em>
          <ul>
            <li><em>‚ÄúIncompatible Prior‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat for the <em>Symplekin</em> <span class="notranslate">RNA</span>i si<span class="notranslate">RNA</span>s (20-22nt) <span class="notranslate">reads</span> dataset collection</li>
    <li><strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat on each collection of si<span class="notranslate">RNA</span> <span class="notranslate">reads</span> (20-22nt) to quantify the abundance of <em>sense</em> si<span class="notranslate">RNA</span>s at relevant targets by changing the following parameters:
      <ul>
        <li><em>‚ÄúSpecify the strandedness of the <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">read 1 (or single-end read) comes from the forward strand (SF)</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The output of <strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> includes a table of <span class="notranslate">RNA</span> features, estimated counts, transcripts per million, and feature length. We will be using the TPM quantification column as input to <strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in the next section.</p>

<h2 id="sirna-differential-abundance-testing">si<span class="notranslate">RNA</span> differential abundance testing</h2>

<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html"><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></a> is a great tool for differential <span class="notranslate">expression</span> analysis, but we also employ it here for estimation of abundance of <span class="notranslate">reads</span> targeting each of our <span class="notranslate">RNA</span> features. As input, <strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> can take <em>t</em>ranscripts <em>p</em>er <em>m</em>illion (TPM) counts produced by <strong>Salmon</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for each feature. TPMs are estimates of the relative abundance of a given transcript in units, and analogously can be used here to quantify si<span class="notranslate">RNA</span>s that align either sense or antisense to transcripts.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-sirna-differential-abundance-testing"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: si<span class="notranslate">RNA</span> differential abundance testing</h3>

  <ol>
    <li><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to test for differential abundance of <em>antisense</em> si<span class="notranslate">RNA</span>s at m<span class="notranslate">RNA</span> and TE features using the following parameters:
      <ul>
        <li>In <em>‚ÄúFactor‚Äù</em>:
          <ul>
            <li>In ‚Äú1: Factor‚Äù
              <ul>
                <li><em>‚ÄúSpecify a factor name‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>i</code></li>
                <li>In <em>‚ÄúFactor level‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Factor level‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSpecify a factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge">Symplekin</code></li>
                        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúCounts file(s)‚Äù</em>: the Symplekin <span class="notranslate">RNA</span>i si<span class="notranslate">RNA</span> counts from reverse strand (SR) collection</li>
                      </ul>
                    </li>
                    <li>In <em>‚Äú2: Factor level‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSpecify a factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>i</code></li>
                        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúCounts file(s)‚Äù</em>: the Symplekin <span class="notranslate">RNA</span>i si<span class="notranslate">RNA</span> counts from reverse strand (SR)</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúChoice of Input data‚Äù</em>: <code class="language-plaintext highlighter-rouge">TPM values (e.g. from sailfish and salmon)</code>
          <ul>
            <li><em>‚ÄúProgram used to generate TPMs‚Äù</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
            <li><em>‚ÄúGene <span class="notranslate">mapping</span> format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Transcript-ID and Gene-ID <span class="notranslate">mapping</span> file</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTabular file with Transcript - Gene <span class="notranslate">mapping</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference tx2g (transcript to gene) file</code> (available through zenodo link)</li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúOutput normalized counts table‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <em>sense</em> si<span class="notranslate">RNA</span>s at m<span class="notranslate">RNA</span> and TE features:
      <ul>
        <li>In <em>‚ÄúFactor‚Äù</em>:
          <ul>
            <li>In ‚Äú1: Factor‚Äù
              <ul>
                <li><em>‚ÄúSpecify a factor name‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>i</code></li>
                <li>In <em>‚ÄúFactor level‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Factor level‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSpecify a factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge">Symplekin</code></li>
                        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúCounts file(s)‚Äù</em>: the Symplekin <span class="notranslate">RNA</span>i si<span class="notranslate">RNA</span> counts from forward strand (SF)</li>
                      </ul>
                    </li>
                    <li>In <em>‚Äú2: Factor level‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSpecify a factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">RNA</span>i</code></li>
                        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúCounts file(s)‚Äù</em>: the Symplekin <span class="notranslate">RNA</span>i si<span class="notranslate">RNA</span> counts from forward strand (SF)</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúChoice of Input data‚Äù</em>: <code class="language-plaintext highlighter-rouge">TPM values (e.g. from sailfish and salmon)</code>
          <ul>
            <li><em>‚ÄúProgram used to generate TPMs‚Äù</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
            <li><em>‚ÄúGene <span class="notranslate">mapping</span> format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Transcript-ID and Gene-ID <span class="notranslate">mapping</span> file</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTabular file with Transcript - Gene <span class="notranslate">mapping</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference tx2g (transcript to gene) file</code> (available through zenodo link)</li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúOutput normalized counts table‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to extract features with a significantly different <em>antisense</em> si<span class="notranslate">RNA</span> abundance (adjusted <em>p</em>-value less than 0.05).
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFilter‚Äù</em>: Select the <code class="language-plaintext highlighter-rouge">DESeq2</code> result file from testing <em>antisense</em> (SR) si<span class="notranslate">RNA</span> abundances</li>
        <li><em>‚ÄúWith following condition‚Äù</em>: <code class="language-plaintext highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
    <li><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for the <em>sense</em> si<span class="notranslate">RNA</span>s at m<span class="notranslate">RNA</span> and TE features:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFilter‚Äù</em>: Select the <code class="language-plaintext highlighter-rouge">DESeq2</code> result file from testing <em>sense</em> (SF) si<span class="notranslate">RNA</span> abundances</li>
        <li><em>‚ÄúWith following condition‚Äù</em>: <code class="language-plaintext highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>How many features have a significant difference in <em>antisense</em> and <em>sense</em> si<span class="notranslate">RNA</span> abundance in the <em>Symplekin</em> <span class="notranslate">RNA</span>i condition?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>There are 87 <em>antisense</em> and 15 <em>sense</em> features that have significanlty different si<span class="notranslate">RNA</span> abundances.</p>
  </blockquote>
</blockquote>

<p>For more information about <code class="language-plaintext highlighter-rouge">DESeq2</code> and its outputs, have a look at <a href="https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf"><code class="language-plaintext highlighter-rouge">DESeq2</code> documentation</a>.</p>

<h2 id="visualization">Visualization</h2>

<p>Coming soon!</p>

<h2 class="no_toc" id="conclusion">Conclusion</h2>

<p>Analysis of small <span class="notranslate">RNA</span>s is a complicated and intricate process due to the diversity in characteristics, functionality, and nuances of small <span class="notranslate">RNA</span> subclasses. The goal of this tutorial is to introduce you to a common small <span class="notranslate">RNA</span> <span class="notranslate">workflow</span> that specifically identifies changes in endogenous si<span class="notranslate">RNA</span> abundances that target protein-coding m<span class="notranslate">RNA</span>s and transposable elements. The steps presented here can be rearranged and modified based on small <span class="notranslate">RNA</span> features of specific systems and the needs of the user.</p>

<h2 id="small-rna-analysis-general-pipeline">Small <span class="notranslate">RNA</span> analysis (general) pipeline</h2>

<p><img src="../../images/sRNA/schematic_for_sRNAseq_tutorial.png" alt="sRNA tutorial workflow" /></p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Analysis of small RNAs is complex due to the diversity of small RNA subclasses.</p>
</li>
                            
                            <li><p>Both alignment to references and transcript quantification approaches are useful for small RNA-seq analyses.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Differential abundance testing of small RNAs (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Mallory Freeberg, 2021 <b>Differential abundance testing of small RNAs (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/srna/tutorial.html">/training-material/topics/transcriptomics/tutorials/srna/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-srna,
    author = "Mallory Freeberg",
    title = "Differential abundance testing of small RNAs (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/srna/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Mallory Freeberg)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/srna/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
