---
layout: tutorial_hands_on

title: Metatranscriptomics analysis using microbiome RNA-seq data
zenodo_link: https://zenodo.org/record/4776250
questions:
- How to analyze metatranscriptomics data?
- What information can be extracted of metatranscriptomics data?
- How to assign taxa and function to the identified sequences?
objectives:
- Choose the best approach to analyze metatranscriptomics data
- Understand the functional microbiome characterization using metatranscriptomic results
- Understand where metatranscriptomics fits in 'multi-omic' analysis of microbiomes
- Visualise a community structure
level: Introductory
tags:
- metatranscriptomics
time_estimation: 5H
key_points:
- Metatranscriptomics data have the same QC profile that RNA-seq data
- A lot of metatranscriptomics sequences are identified as rRNA sequences
- With shotgun data, we can extract information about the studied community structure
  and also the functions realised by the community
- Metatranscriptomics data analyses are complex and must be careful done, specially
  when they are done without combination to metagenomics data analyses
contributors:
- pratikdjagtap
- subinamehta
- jraysajulga
- bebatut
- emmaleith
- pravs3683
- shiltemann

---




{% include {{ page.dir }}content.md short=false %}
