<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Proteogenomics 2: Database Search</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Proteogenomics 2: Database Search" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2Fproteogenomics-dbsearch%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Proteogenomics 2: Database Search",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
  },
  "courseCode": "proteomics / proteogenomics-dbsearch / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Proteogenomics 2: Database Search' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html",
  "timeRequired": "PT15M",
  "description": "The questions this  addresses are:\n - How to identify variant proteoforms in MS data by searching with the customized Protein database?\n\n\\nThe objectives are:\n - A proteogenomic data analysis of mass spectrometry data to identify and visualize variant peptides.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html",
      "name": "Proteogenomics 1: Database Creation",
      "description": "Hands-on for 'Proteogenomics 1: Database Creation' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Subina Mehta"
    },
    {
      "@type": "Person",
      "name": "Timothy J. Griffin"
    },
    {
      "@type": "Person",
      "name": "Pratik Jagtap"
    },
    {
      "@type": "Person",
      "name": "Ray Sajulga"
    },
    {
      "@type": "Person",
      "name": "James Johnson"
    },
    {
      "@type": "Person",
      "name": "Praveen Kumar"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ],
  "skillLevel": "Intermediate"
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Proteogenomics 2: Database Search</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/subinamehta/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/subinamehta" alt="Subina Mehta">Subina Mehta</a>, <a href="/training-material/hall-of-fame/timothygriffin/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/timothygriffin" alt="Timothy J. Griffin">Timothy J. Griffin</a>, <a href="/training-material/hall-of-fame/pratikdjagtap/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pratikdjagtap" alt="Pratik Jagtap">Pratik Jagtap</a>, <a href="/training-material/hall-of-fame/jraysajulga/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jraysajulga" alt="Ray Sajulga">Ray Sajulga</a>, <a href="/training-material/hall-of-fame/jj-umn/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jj-umn" alt="James Johnson">James Johnson</a>, <a href="/training-material/hall-of-fame/pravs3683/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/pravs3683" alt="Praveen Kumar">Praveen Kumar</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to identify variant proteoforms in MS data by searching with the customized Protein database?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>A proteogenomic data analysis of mass spectrometry data to identify and visualize variant peptides.</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 1: Database Creation:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbcreation/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 15 minutes</p>
            

            
            <p><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> Level: </strong>
             Intermediate 
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span>
</span>

</p>
            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1489208">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/workflows/" title="Workflows" alt="Proteogenomics 2: Database Search workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In this tutorial, we perform proteogenomic database searching using the Mass Spectrometry data. The inputs for performing the proteogenomic database searching are the peaklist MGF files and the FASTA database file. The FASTA database is obtained by running the first <span class="notranslate">workflow</span> ‚ÄúUniprot_cRAP_SAV_indel_translatedbed.FASTA‚Äù. The second <span class="notranslate">workflow</span> focuses on performing database search of the peak list files (MGFs).</p>

<p><img src="../../images/second_workflow.png" alt="Workflow" /></p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>
<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a></li>
  <li><a href="#match-peptide-sequences" id="markdown-toc-match-peptide-sequences">Match peptide sequences</a>    <ol>
      <li><a href="#searchgui" id="markdown-toc-searchgui">SearchGUI</a></li>
      <li><a href="#peptideshaker" id="markdown-toc-peptideshaker">PeptideShaker</a></li>
    </ol>
  </li>
  <li><a href="#create-database" id="markdown-toc-create-database">Create database</a></li>
  <li><a href="#remove-known-peptides" id="markdown-toc-remove-known-peptides">Remove known peptides</a>    <ol>
      <li><a href="#merged-uniprot-and-crap-database" id="markdown-toc-merged-uniprot-and-crap-database">Merged Uniprot and cRAP database</a></li>
      <li><a href="#text-manipulation-steps" id="markdown-toc-text-manipulation-steps">Text Manipulation steps</a></li>
      <li><a href="#query-tabular" id="markdown-toc-query-tabular">Query Tabular</a></li>
    </ol>
  </li>
  <li><a href="#blastp-basic-local-alignment-search-tool--proteins" id="markdown-toc-blastp-basic-local-alignment-search-tool--proteins">BLASTP (Basic Local Alignment Search Tool- proteins)</a></li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<blockquote class="no_toc hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-organization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: data upload and organization</h3>

  <ol>
    <li>Create a <strong>new history</strong> and name it something meaningful (e.g. <em>Proteogenomics DB search</em>)</li>
    <li>Import the four MGF MS/MS files and the Trimmed_ref_5000_uniprot_cRAP.FASTA sequence file from Zenodo.<a href="https://doi.org/10.5281/zenodo.1489208"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1489208.svg" alt="DOI" /></a>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f4.mgf
https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f5.mgf
https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f8.mgf
https://zenodo.org/record/1489208/files/Mo_Tai_Trimmed_mgfs__Mo_Tai_iTRAQ_f9.mgf
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>
    </li>
    <li>Rename the datasets to something more recognizable (strip the URL prefix)</li>
    <li>
      <p>Build a <strong>Dataset list</strong> for the three MGF files, name it as ‚Äú<strong>Mo_Tai_MGFs</strong>‚Äù</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a dataset collection</h3>

        <ul>
          <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="../../../galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>
          <li>Check all the datasets in your history you would like to include</li>
          <li>
            <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>

            <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>
          </li>
          <li>Enter a name for your collection</li>
          <li>Click <strong>Create List</strong> to build your collection</li>
          <li>Click on the checkmark icon at the top of your history again</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="match-peptide-sequences">Match peptide sequences</h1>

<p>The search database labelled <code class="language-plaintext highlighter-rouge">Uniprot_cRAP_SAV_indel_translatedbed.FASTA</code> is the input database that
will be used to match MS/MS to peptide sequences via a sequence database search.</p>

<p>For this, the sequence database-searching program called <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> will be used.The generated dataset collection of the three <em>MGF files</em> in the history is used as the MS/MS input. We will walk through a number of these settings in order to utilize SearchGUI on these example MGF files.</p>

<h2 id="searchgui">SearchGUI</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-searchgui"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: SearchGUI</h3>

  <ol>
    <li><strong>SearchGUI</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProtein Database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Uniprot_cRAP_SAV_indel_translatedbed.FASTA</code> (Or however you named the <code class="language-plaintext highlighter-rouge">FASTA</code> file)</li>
        <li>
          <p><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúInput Peak lists (mgf)‚Äù</em>: <code class="language-plaintext highlighter-rouge">MGF files</code> dataset collection.</p>

          <blockquote class="tip">
            <h3 id="tip-tip-select-dataset-collections-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select dataset collections as input</h3>

            <ul>
              <li>
                <p>Click the <strong>Dataset collection</strong> icon on the left of the input field:</p>
              </li>
              <li>
                <p>Select the appropriate dataset collection from the list</p>
              </li>
            </ul>
          </blockquote>
        </li>
        <li>Section <strong>Search Engine Options</strong>:
          <ul>
            <li>
              <p><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúDB-Search Engines‚Äù</em>: <code class="language-plaintext highlighter-rouge">X!Tandem</code></p>

              <blockquote class="comment">
                <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>
                <p>The section <strong>Search Engine Options</strong> contains a selection of sequence database searching
   programs that are available in SearchGUI. Any combination of these programs can be used for
   generating PSMs from MS/MS data. For the purpose of this tutorial, <strong>X!Tandem</strong> we will be
   used.</p>
              </blockquote>
            </li>
          </ul>
        </li>
        <li>Section <strong>Protein Digestion Options</strong>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúEnzyme‚Äù</em>: <code class="language-plaintext highlighter-rouge">Trypsin</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMaximum Missed Cleavages‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
          </ul>
        </li>
        <li>Section <strong>Precursor options</strong>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúPrecursor Ion Tolerance Units‚Äù</em>: <code class="language-plaintext highlighter-rouge">Parts per million (ppm)</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúPrecursor Ion Tolerance‚Äù</em>:<code class="language-plaintext highlighter-rouge"> 10</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúFragment Tolerance (Daltons)‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.05</code> (this is high resolution MS/MS data)</li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMinimum charge‚Äù</em>:<code class="language-plaintext highlighter-rouge">2</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMaximum charge‚Äù</em>:<code class="language-plaintext highlighter-rouge">6</code></li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúForward Ion‚Äù</em>: <code class="language-plaintext highlighter-rouge">b</code></li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúReverse Ion‚Äù</em>:<code class="language-plaintext highlighter-rouge"> y</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMinimum Precursor Isotope‚Äù</em> :<code class="language-plaintext highlighter-rouge">0</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMaximum Precursor Isotope‚Äù</em> :<code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
        <li>Section <strong>Protein Modification Options</strong>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúFixed Modifications‚Äù</em>: <code class="language-plaintext highlighter-rouge">Carbamidomethylation of C, ITRAQ-4Plex of K, ITRAQ-4Plex of Ntermini</code></li>
            <li>
              <p><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúVariable modifications‚Äù</em>: <code class="language-plaintext highlighter-rouge">Oxidation of M, ITRAQ-4Plex of Y</code></p>

              <blockquote class="tip">
                <h3 id="tip-tip-search-for-options"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Search for options</h3>
                <p>For selection lists, typing the first few letters in the window will filter the
available options.</p>
              </blockquote>
            </li>
          </ul>
        </li>
        <li>Section <strong>Advanced Options</strong>:
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúX!Tandem Options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Advanced</code>
              <ul>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúX!Tandem: Quick Acetyl‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúX!Tandem: Quick Pyrolidone‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúX!Tandem: Protein stP Bias‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúX!Tandem: Maximum Valid Expectation Value‚Äù</em>: <code class="language-plaintext highlighter-rouge">100</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <p>Once the database search is completed, the SearchGUI tool will output a file (called a
SearchGUI archive file) that will serve as an input for the next section, PeptideShaker.
Rename the output as ‚Äú<strong>Compressed SearchGUI results</strong>‚Äù</p>

  <blockquote class="comment">
    <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>
    <p>Note that sequence databases used for proteogenomics are usually much larger than
the excerpt used in this tutorial. When using large databases, the peptide identification
step can take much more time for computation. In proteogenomics, choosing the optimal
database is a crucial step of your <span class="notranslate">workflow</span>.</p>
  </blockquote>

</blockquote>

<h2 id="peptideshaker">PeptideShaker</h2>

<p><a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a> is a post-processing software tool that
processes data from the SearchGUI software tool. It serves to organize the Peptide-Spectral
Matches (PSMs) generated from SearchGUI processing and is contained in the SearchGUI archive.
It provides an assessment of confidence of the data, inferring proteins identified from the
matched peptide sequences and generates outputs that can be visualized by users to interpret
results. PeptideShaker has been wrapped in <span class="notranslate">Galaxy</span> to work in combination with SearchGUI
outputs.</p>

<blockquote class="comment">
  <h3 id="comment-comment-file-formats"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: File Formats</h3>
  <p>There are a number of choices for different data files that can be generated using
  PeptideShaker. A compressed file can be made containing all information needed to
  view them results in the standalone PeptideShaker viewer. A <code class="language-plaintext highlighter-rouge">mzidentML</code> file can
  be generated that contains all peptide sequence matching information and can be
  utilized by compatible <span class="notranslate">downstream</span> software. Other outputs are focused on the inferred
  proteins identified from the PSMs, as well as phosphorylation reports, relevant if
  a phosphoproteomics experiment has been undertaken.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peptideshaker"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: PeptideShaker</h3>

  <ol>
    <li><strong>PeptideShaker</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúCompressed SearchGUI results‚Äù</em>: The SearchGUI archive file</li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúSpecify Advanced PeptideShaker Processing Options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Default Processing Options</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúSpecify Advanced Filtering Options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Default Filtering Options</code></li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúInclude the protein sequences in mzIdentML‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúOutput options‚Äù</em>: Select the <code class="language-plaintext highlighter-rouge">PSM Report</code> (Peptide-Spectral Match) and the <code class="language-plaintext highlighter-rouge">Certificate of Analysis</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-certificate-of-analysis"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Certificate of Analysis</h3>

        <p>The ‚ÄúCertificate of Analysis‚Äù provides details on all the parameters
used by both SearchGUI and PeptideShaker in the analysis. This can be downloaded from the
<span class="notranslate">Galaxy</span> instance to your local computer in a text file if desired.</p>

      </blockquote>
    </li>
    <li>Inspect <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> the resulting files</li>
  </ol>

</blockquote>

<p>A number of new items will appear in your history, each corresponding to the outputs selected
in the PeptideShaker parameters. Most relevant for this tutorial is the PSM report:</p>

<p>Scrolling at the bottom to the left will show the sequence for the PSM that matched to these
peptide entries. Column 3 is the sequence matched for each PSM entry. Every PSM is a
new row in the tabular output.</p>

<p>A number of new items will appear in your History, each corresponding to the outputs selected in the PeptideShaker parameters. The Peptide Shaker‚Äôs PSM report is used as an input for the BlastP analysis. Before performing BlastP analysis. The Query Tabular tool and few test manipulation tools are used to remove spectra that belongs to the reference proteins. The output tabular file ‚ÄúPeptides_for_Blast-P_analysis‚Äù will contain only those spectra that did not belong to any known proteins.</p>

<h1 id="create-database">Create database</h1>

<p>The mzidentml output from the Peptide shaker is converted into an sqlite database file by using the mz to sqlite tool. This sqlite output is used to open the Multi-omics visualization platform, wherein you can view the spectra of the peptides using Lorikeet parameters. To open the MVP viewer, click on the ‚ÄúVisualize in MVP Application‚Äù icon ( this will pop-open the interactive multi-omics viewer in a new window/tab)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mz-to-sqlite"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: mz to sqlite</h3>

  <p>This tool extracts mzidentml and its associated proteomics datasets into a sqlite db</p>

  <ol>
    <li><strong>mz to sqlite</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProteomics identification files‚Äù</em>: <code class="language-plaintext highlighter-rouge">PeptideShaker_mzidentml</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProteomics Spectrum files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mo_Tai_MGFs</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProteomics Search Database Fasta‚Äù</em>: <code class="language-plaintext highlighter-rouge">Uniprot_cRAP_SAV_indel_translatedbed.FASTA</code></li>
      </ul>

      <p><img src="../../images/mz2sqlite.png" alt="mz2sqlite" width="20%" /></p>
    </li>
  </ol>

</blockquote>

<p>The next step is to remove known peptides from the list of PSM‚Äôs that we acquired from the Peptide Shaker results. For that we need to perform some text manipulation steps to extract list of known peptides from the UniProt and cRAP database.</p>

<h1 id="remove-known-peptides">Remove known peptides</h1>

<h2 id="merged-uniprot-and-crap-database">Merged Uniprot and cRAP database</h2>

<p>The file named ‚ÄúTrimmed_ref_500_Uniprot_cRAP.fasta‚Äù is the trimmed version of Uniprot and cRAP database merged Fasta files.</p>

<p>This Fasta file will be subjected to few text manipulation steps in order to get the tabular file for the known peptides. The known peptides are those which are already annotated in the reference database, in this case the Uniprot database. The first step is to convert this FASTA file to tabular in order to proceed with text manipulation.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on--hands-on-fasta-to-tabular"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span>  Hands-on: FASTA to Tabular</h3>

  <ol>
    <li>Run <strong>FASTA to Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúData input ‚Äòinput‚Äô (fasta)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Trimmed_ref_500_Uniprot_cRAP.fasta</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúHow many columns to divide title string into?‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúHow many title characters to keep?‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The resultant tabular file will go through a series of text manipulation steps to make it suitable for
input to the Query Tabular tool.</p>

<h2 id="text-manipulation-steps">Text Manipulation steps</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-text-manipulation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Text Manipulation</h3>

  <ol>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúCut Columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
      </ul>

      <p>Upon completion of this step you will have extracted C1 (column 1) from the input tabular file</p>
    </li>
    <li><strong>Convert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúConvert all‚Äù</em>: <code class="language-plaintext highlighter-rouge">Whitespaces</code></li>
        <li><em>‚Äúin Dataset‚Äù</em> : <code class="language-plaintext highlighter-rouge">Data input 'input' (txt)</code></li>
        <li><em>‚ÄúStrip leading and trailing whitespaces‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúCondense consecutive delimiters in one TAB‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>This step will convert all the white spaces into different tabular column.</p>
    </li>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúCut Columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c2</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Pipe</code></li>
      </ul>

      <p>This step will extract information in column2 separated by Pipe</p>
    </li>
    <li><strong>Convert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúConvert all‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dots</code></li>
        <li><em>‚Äúin Dataset‚Äù</em> : <code class="language-plaintext highlighter-rouge">Data input 'input' (txt)</code></li>
        <li><em>‚ÄúStrip leading and trailing whitespaces‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúCondense consecutive delimiters in one TAB‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>This step will convert all the dots into different tabular column.</p>
    </li>
    <li><strong>Group</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúSelect data‚Äù</em>: <code class="language-plaintext highlighter-rouge">input from above</code></li>
        <li><em>‚ÄúGroup by column‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Now that we have the list of known peptides, the query tabular tool is used to move these reference pepides from the PSM report.</p>

<h2 id="query-tabular">Query Tabular</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-query-tabular"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <ol>
    <li><strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</li>
  </ol>

  <ul>
    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong> (a): <code class="language-plaintext highlighter-rouge">output from group</code>
      <ul>
        <li>Section <strong>Table Options</strong>:
          <ul>
            <li><em>‚ÄúTabular Dataset for Table‚Äù</em>: Uniprot</li>
            <li><em>‚ÄúUse first line as column names‚Äù</em> : <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúSpecify Column Names (comma-separated list)‚Äù</em>:<code class="language-plaintext highlighter-rouge">prot</code></li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>:
              <ul>
                <li><em>‚ÄúTable Index‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>‚ÄúIndex on Columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">Prot</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong> (b): <code class="language-plaintext highlighter-rouge">PSM report</code>
      <ul>
        <li>Section <strong>Filter Dataset Input</strong>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>‚ÄúFilter by‚Äù</em>:  <code class="language-plaintext highlighter-rouge">skip leading lines</code></li>
                <li><em>‚ÄúSkip lines‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Section <strong>Table Options</strong>:
          <ul>
            <li><em>‚ÄúSpecify Name for Table‚Äù</em>: <code class="language-plaintext highlighter-rouge">psms</code></li>
            <li><em>‚ÄúUse first line as column names‚Äù</em> : <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúSpecify Column Names (comma-separated list)‚Äù</em>:<code class="language-plaintext highlighter-rouge">id,Proteins,Sequence</code></li>
            <li><em>‚ÄúOnly load the columns you have named into database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Table Index</strong>
              <ul>
                <li><em>‚ÄúTable Index‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>‚ÄúIndex on Columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">id</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Database Table</strong> (c):<code class="language-plaintext highlighter-rouge">PSM report</code></p>

      <ul>
        <li>Section <strong>Filter Dataset Input</strong>
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>‚ÄúFilter by‚Äù</em>:  <code class="language-plaintext highlighter-rouge">skip leading lines</code></li>
                <li><em>‚ÄúSkip lines‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>‚ÄúFilter by‚Äù</em>:  <code class="language-plaintext highlighter-rouge">select columns</code></li>
                <li><em>‚ÄúEnter column numbers to keep‚Äù</em>: <code class="language-plaintext highlighter-rouge">1,2</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Filter Tabular Input Lines</strong>
              <ul>
                <li><em>‚ÄúFilter by‚Äù</em>:  <code class="language-plaintext highlighter-rouge">normalize list columns,replicate rows for each item in the list</code></li>
                <li><em>‚ÄúEnter column numbers to normalize‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                <li><em>‚ÄúList item delimiter in column‚Äù</em>: <code class="language-plaintext highlighter-rouge">,</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Section <strong>Table Options</strong>:
          <ul>
            <li><em>‚ÄúSpecify Name for Table‚Äù</em>: <code class="language-plaintext highlighter-rouge">prots</code></li>
            <li><em>‚ÄúUse first line as column names‚Äù</em> : <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúSpecify Column Names (comma-separated list)‚Äù</em>:<code class="language-plaintext highlighter-rouge">id,prot</code></li>
            <li><em>‚ÄúOnly load the columns you have named into database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <strong>Insert Table Index</strong>:
              <ul>
                <li><em>‚ÄúTable Index‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
                <li><em>‚ÄúIndex on Columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">prot, id</code></li>
              </ul>
            </li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>

            <p>By default, table columns will be named: c1,c2,c3,‚Ä¶,cn (column names for a table must be unique).
You can override the default names by entering a comma separated list of names, e.g. <code class="language-plaintext highlighter-rouge">,name1,,,name2</code>
would rename the second and fifth columns.
Check your input file to find the settings which best fits your needs.</p>

          </blockquote>
        </li>
        <li>
          <p><em>‚ÄúSave the sqlite database in your history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

          <blockquote class="comment">
            <h3 id="comment-querying-an-sqlite-database"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Querying an SQLite Database</h3>
            <p><strong>Query Tabular</strong> can also use an existing SQLite database.
Activating <code class="language-plaintext highlighter-rouge">Save the sqlite database in your history</code>
will store the generated database in the history, allowing to reuse it directly.</p>
          </blockquote>
        </li>
        <li><em>‚ÄúSQL Query to generate tabular output‚Äù</em>:
          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT psms.*
FROM psms
WHERE psms.id NOT IN
(SELECT distinct prots.id
FROM prots JOIN uniprot ON prots.prot = uniprot.prot)
ORDER BY psms.id
</code></pre></div>          </div>
        </li>
        <li><em>‚Äúinclude query result column headers‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ul>

</blockquote>

<p>The output from this step is that the resultant peptides would be those which doesn‚Äôt belong in the Uniprot or cRAP database. The query tabular tool is used again to create a tabular output containing peptides ready for Blast P analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-query-tabular-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Query Tabular</h3>

  <p><strong>Query Tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters:</p>

  <ul>
    <li>Section <strong>Filter Dataset Input</strong>
      <ul>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Filter Tabular Input Lines‚Äù</em>
          <ul>
            <li><em>‚ÄúFilter by‚Äù</em>:  <code class="language-plaintext highlighter-rouge">skip leading lines</code></li>
            <li><em>‚ÄúSkip lines‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Section <strong>Table Options</strong>:
      <ul>
        <li><em>‚ÄúSpecify Name for Table‚Äù</em>: <code class="language-plaintext highlighter-rouge">psm</code></li>
        <li><em>‚ÄúUse first line as column names‚Äù</em> : <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúSpecify Column Names (comma-separated list)‚Äù</em>:<code class="language-plaintext highlighter-rouge">id,Proteins,Sequence</code></li>
        <li><em>‚ÄúOnly load the columns you have named into database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li><em>‚ÄúSQL Query to generate tabular output‚Äù</em>:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT Sequence || ' PSM=' || group_concat(id,',') || ' length='
|| length(Sequence) as "ID",Sequence
FROM  psm
WHERE length(Sequence) &gt;6
AND length(Sequence) &lt;= 30
GROUP BY Sequence
ORDER BY length(Sequence),Sequence
</code></pre></div>      </div>
    </li>
    <li>
      <p><em>‚Äúinclude query result column headers‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>
    </li>
    <li>Click <strong>Execute</strong> and inspect the query results file after it turned green.</li>
  </ul>

  <p><img src="../../images/QT_output.png" alt="QT" /></p>

</blockquote>

<h1 id="blastp-basic-local-alignment-search-tool--proteins">BLASTP (Basic Local Alignment Search Tool- proteins)</h1>

<p><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a> is a web based tool used to compare biological sequences. BlastP, matches protein sequences against a protein database. More specifically, it looks at the amino acid sequence of proteins and can detect and evaluate the amount of differences between say, an experimentally derived sequence and all known amino acid sequences from a database. It can then find the most similar sequences and allow for identification of known proteins or for identification of potential peptides associated with novel proteoforms.</p>

<p>BlastP search is carried out with the PSM report (output from PeptideShaker). Before, BlastP analysis the ‚ÄúPeptides_for_Blast-P_analysis‚Äù is first converted from Tabular format to FASTA file format which can be easily read by the BlastP algorithm. This is done with the help of ‚ÄúTabular to FASTA‚Äù conversion tool.
The short BlastP uses parameters for short peptide sequences (8-30 aas). Please use the rerun option to look at the parameters used.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-tabular-to-fasta"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Tabular to FASTA</h3>
  <ol>
    <li><strong>Tabular to FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: with the following parameters:
      <ul>
        <li><em>‚ÄúTitle column‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>‚ÄúSequence Column‚Äù</em>:<code class="language-plaintext highlighter-rouge">2</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The output FASTA file is going to be subjected to BLAST-P analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ncbi-blast-blastp"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: NCBI BLAST+ blastp</h3>

  <ol>
    <li><strong>NCBI BLAST+ blastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <strong>BLASTP</strong> with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúProtein query sequence(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Data input 'query' (fasta)</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúSubject database/sequences‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally installed BLAST database</code>
          <ul>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúProtein Blast database‚Äù</em>: Select <code class="language-plaintext highlighter-rouge">nr_mouse_current</code></li>
          </ul>
        </li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúType of BLAST‚Äù</em>: <code class="language-plaintext highlighter-rouge">blastp-short - BLASTP optimized for queries shorter than 30 residues</code></li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúSet expectation value cutoff‚Äù</em>:<code class="language-plaintext highlighter-rouge">200000.0</code></li>
        <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tabular(extended 25 columns)</code></li>
        <li><em>‚ÄúAdvanced Options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Show Advanced Options</code>
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúFilter out low complexity regions (with SEG)‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúScoring matrix and gap costs‚Äù</em>: <code class="language-plaintext highlighter-rouge">PAM30</code>
              <ul>
                <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúGap Costs‚Äù</em>: <code class="language-plaintext highlighter-rouge">Existence: 9 Extension: 1</code></li>
              </ul>
            </li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMaximum hits to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMaximum number of HSPs (alignments) to keep for any single query-subject pair‚Äù</em>:<code class="language-plaintext highlighter-rouge">1</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúWord size for wordfinder algorithm‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMultiple hits window size‚Äù</em>: <code class="language-plaintext highlighter-rouge">40</code></li>
            <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMinimum score to add a word to the BLAST lookup table‚Äù</em>: <code class="language-plaintext highlighter-rouge">11</code></li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúComposition-based statistics‚Äù</em>: <code class="language-plaintext highlighter-rouge">0 (No composition)</code></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúShould the query and subject defline(s) be parsed?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">param-select</span> <em>‚ÄúRestrict search of database to a given set of ID‚Äôs‚Äù</em>:<code class="language-plaintext highlighter-rouge">No restriction</code></li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment:</h3>

            <p>This feature provides a means to exclude ID‚Äôs from a BLAST database search.
The expectation values in the BLAST results are based upon the sequences actually
searched, and not on the underlying database. Note this cannot be used when
comparing against a FASTA file.</p>

          </blockquote>
        </li>
        <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">param-text</span> <em>‚ÄúMinimum query coverage per hsp (percentage, 0 to 100)‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúCompute locally optimal Smith-Waterman alignments‚Äù</em>:<code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Once BlastP search is performed, it provides with a tabular output containing peptides corresponding to novel proteoforms termed as ‚Äú<strong>Novel peptides</strong>‚Äù. Now this output is further processed by comparing the novel peptide output with the PSM report for selecting only distinct peptides which pass these criteria. This could be achieved by proceeding to the novel peptide analysis tutorial.</p>

<blockquote class="comment">
  <h3 id="comment-comment-tool-versions"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Tool Versions</h3>

  <p>The tools are subjected to changes while being upgraded.
Thus, running the <span class="notranslate">workflow</span> provided with the tutorial, the user might need to make sure they are using the latest version including updating the parameters.</p>

</blockquote>



                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>With SearchGUI and PeptideShaker you can gain access to multiple search engines</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Proteogenomics 2: Database Search (Proteomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar, 2021 <b>Proteogenomics 2: Database Search (Galaxy Training Materials)</b>. <a href="/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html">/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{proteomics-proteogenomics-dbsearch,
    author = "Subina Mehta and Timothy J. Griffin and Pratik Jagtap and Ray Sajulga and James Johnson and Praveen Kumar",
    title = "Proteogenomics 2: Database Search (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
        <blockquote class="agenda follow-up">
            <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
            <ul>
                
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Proteogenomics 3: Novel peptide analysis:
                            
                            
                                
                                     <a href="/training-material/topics/proteomics/tutorials/proteogenomics-novel-peptide-analysis/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
        </blockquote>
        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Subina Mehta, Timothy J. Griffin, Pratik Jagtap, Ray Sajulga, James Johnson, Praveen Kumar)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/proteogenomics-dbsearch/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
