<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>NGS data logistics</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:title" content="Galaxy Training: NGS data logistics" />
        <meta property="og:description" content="Galaxy is a scientific workflow, data integration, and da..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/introduction" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Introduction to Galaxy Analyses
                        </a>
                        
                    </li>
                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="/training-material/topics/introduction/faqs/" title="Check our FAQs for the Introduction to Galaxy Analyses topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Translate this page
            </div>

            <div id="lang-selector">
                
                <strong>Automatic Translations</strong>

<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html&edit-text=&act=url">
                Fran√ßais
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html&edit-text=&act=url">
                Êó•Êú¨Ë™û
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html&edit-text=&act=url">
                Espa√±ol
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html&edit-text=&act=url">
                Portugu√™s
                </a>
                
<a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html&edit-text=&act=url">
                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                </a>
                
                <a class="btn btn-info" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html&edit-text=&act=url" title="">
                And more!
                </a>
            </div>
        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-10-01/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html" title="Version 2021-10-01">2021-10-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-09-01/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html" title="Version 2021-09-01">2021-09-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2021-08-01/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html" title="Version 2021-08-01">2021-08-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "1970-01-01 00:33:41 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "NGS data logistics",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Introduction to Galaxy Analyses",
    "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
    "url": "https://training.galaxyproject.org//training-material/topics/introduction/"
  },
  "courseCode": "introduction / galaxy-intro-ngs-data-managment / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'NGS data logistics' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html",
  "timeRequired": "PT1H30M",
  "description": "The questions this  addresses are:\n - How to manipulate and process NGS data\n\n\\nThe objectives are:\n - Understand most common types of NGS-related datatypes\n - Learn about how Galaxy handles NGS data using Illumina data derived from patients infected with SARS-CoV-2\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Marius van den Beek"
    },
    {
      "@type": "Person",
      "name": "Dave Clements"
    },
    {
      "@type": "Person",
      "name": "Daniel Blankenberg"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Marius van den Beek"
    },
    {
      "@type": "Person",
      "name": "Dave Clements"
    },
    {
      "@type": "Person",
      "name": "Daniel Blankenberg"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Introduction to Galaxy Analyses",
      "description": "Galaxy is a scientific workflow, data integration, and data and analysis persistence and publishing platform that aims to make computational biology accessible to research scientists that do not have computer programming experience. ",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/"
    }
  ]
}
</script>

<section class="tutorial topic-introduction">
    <h1 data-toc-skip>NGS data logistics</h1>
    

    <div class="contributors-line">Authors: <a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge contributor-nekrut"><img src="https://avatars.githubusercontent.com/nekrut?s=27" alt="Avatar">Anton Nekrutenko</a><a href="/training-material/hall-of-fame/mvdbeek/" class="contributor-badge contributor-mvdbeek"><img src="https://avatars.githubusercontent.com/mvdbeek?s=27" alt="Avatar">Marius van den Beek</a><a href="/training-material/hall-of-fame/tnabtaf/" class="contributor-badge contributor-tnabtaf"><img src="https://avatars.githubusercontent.com/tnabtaf?s=27" alt="Avatar">Dave Clements</a><a href="/training-material/hall-of-fame/blankenberg/" class="contributor-badge contributor-blankenberg"><img src="https://avatars.githubusercontent.com/blankenberg?s=27" alt="Avatar">Daniel Blankenberg</a></div>

    <blockquote class="overview">
        <h3>Overview</h3>
        
        <img alt="Creative Commons License" class="float-right" style="border-width:0; display: inline-block; margin:0" src="https://i.creativecommons.org/l/by/4.0/88x31.png" aria-hidden="true" />
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to manipulate and process NGS data</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Understand most common types of NGS-related datatypes</p>
</li>
        
        <li><p>Learn about how Galaxy handles NGS data using Illumina data derived from patients infected with SARS-CoV-2</p>
</li>
        
        </ul>

        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour 30 minutes</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default supporting_material"><a href="/training-material/topics/introduction/slides/introduction.html" title="Topic Overview slides">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Topic Overview slides
                </a></li>
            

            

            
                <li class="btn btn-default supporting_material">


    <a class="topic-icon" href="/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/workflows/" title="Workflows" alt="NGS data logistics workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i> Workflows
    </a>

</li>
            

            

            

            
            
            

            
                <li class="btn btn-default supporting_material">


    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/introduction/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 23, 2021 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                

                <p>In this section we will look at practical aspects of manipulation of next-generation sequencing data. We will start with the FASTQ format produced by most sequencing machines and will finish with the SAM/BAM format representing mapped <span class="notranslate">reads</span>. The cover image above shows a screen dump of a SAM dataset.</p>

<h1 id="introduction-to-sequencing-data">Introduction to sequencing data</h1>

<h2 id="fastq-manipulation-and-quality-control">FASTQ manipulation and quality control</h2>

<p><a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ</a> is not a very well defined format. In the beginning various manufacturers of sequencing instruments were free to interpret FASTQ as they saw fit, resulting in a multitude of FASTQ flavors. This variation stemmed primarily from different ways of encoding quality values as described <a href="https://en.wikipedia.org/wiki/FASTQ_format">here</a> (below you will find an explanation of quality scores and their meaning). Today, the <a href="https://www.ncbi.nlm.nih.gov/pubmed/20015970">FASTQ Sanger</a> version of the format is considered to be the standard form of FASTQ. <span class="notranslate">Galaxy</span> is using FASTQ Sanger as the only legitimate input for <span class="notranslate">downstream</span> processing tools and provides <a href="https://www.ncbi.nlm.nih.gov/pubmed/20562416">a number of utilities for converting FASTQ files</a> into this form (see <strong>FASTQ Quality Control</strong> section of <span class="notranslate">Galaxy</span> tools).</p>

<p>The FASTQ format looks like this:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>
@M02286:19:000000000-AA549:1:1101:12677:1273 1:N:0:23
CCTACGGGTGGCAGCAGTGAGGAATATTGGTCAATGGACGGAAGTCTGAACCAGCCAAGTAGCGTGCAG
+
ABC8C,:@F:CE8,B-,C,-6-9-C,CE9-CC--C-&lt;-C++,,+;CE&lt;,,CD,CEFC,@E9&lt;FCFCF?9
@M02286:19:000000000-AA549:1:1101:15048:1299 1:N:0:23
CCTACGGGTGGCTGCAGTGAGGAATATTGGACAATGGTCGGAAGACTGATCCAGCCATGCCGCGTGCAG
+
ABC@CC77CFCEG;F9&lt;F89&lt;9--C,CE,--C-6C-,CE:++7:,CF&lt;,CEF,CFGGD8FFCFCFEGCF
@M02286:19:000000000-AA549:1:1101:11116:1322 1:N:0:23
CCTACGGGAGGCAGCAGTAGGGAATCTTCGGCAATGGACGGAAGTCTGACCGAGCAACGCCGCGTGAGT
+
AAC&lt;CCF+@@&gt;CC,C9,F9C9@9-CFFFE@7@:+CC8-C@:7,@EFE,6CF:+8F7EFEEF@EGGGEEE

</code></pre></div></div>

<p>Each sequencing read is represented by four lines:</p>

<ol>
  <li><code class="language-plaintext highlighter-rouge">@</code> followed by read ID and optional information about sequencing run</li>
  <li>sequenced bases</li>
  <li><code class="language-plaintext highlighter-rouge">+</code> (optionally followed by the read ID and some additional info)</li>
  <li>quality scores for each base of the sequence encoded as <a href="https://en.wikipedia.org/wiki/ASCII">ASCII symbols</a></li>
</ol>

<h2 id="paired-end-data">Paired end data</h2>

<p>It is common to prepare pair-end and mate-pair sequencing libraries. This is highly beneficial for a number of applications discussed in subsequent topics. For now let‚Äôs just briefly discuss what these are and how they manifest themselves in FASTQ form.</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/pe_mp.png" alt="Paired-end and mate-pair reads. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small><strong>Paired-end and mate-pair <span class="notranslate">reads</span></strong>. In paired end sequencing (left) the actual ends of rather short <span class="notranslate">DNA</span> molecules (less than 1kb) are determined, while for mate pair sequencing (right) the ends of long molecules are joined and prepared in special sequencing libraries. In these mate pair protocols, the ends of long, size-selected molecules are connected with an inte<span class="notranslate">rna</span>l adapter sequence (i.e. linker, yellow) in a circularization reaction. The circular molecule is then processed using restriction enzymes or fragmentation. Fragments are enriched for the linker and outer library adapters are added around the two combined molecule ends. The inte<span class="notranslate">rna</span>l adapter can then be used as a second priming site for an additional sequencing reaction in the same orientation or sequencing can be performed from the second adapter, from the reverse strand. (From ‚ÄúUnderstanding and improving high-throughput sequencing data production and analysis‚Äù, Ph.D. dissertation by <a href="https://ul.qucosa.de/api/qucosa%3A11231/attachment/ATT-0/">Martin Kircher</a>)</small></td>
    </tr>
  </tbody>
</table>

<p>Thus in both cases (paired-end and mate-pair) a single physical piece of <span class="notranslate">DNA</span> (or <span class="notranslate">RNA</span> in the case of <span class="notranslate">RNA</span>-seq) is sequenced from two ends and so generates two <span class="notranslate">reads</span>. These can be represented as separate files (two FASTQ files with first and second <span class="notranslate">reads</span>) or a single file were <span class="notranslate">reads</span> for each end are interleaved. Here are examples:</p>

<h4 id="two-single-files">Two single files</h4>

<p><strong>File 1</strong></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code> @M02286:19:000000000-AA549:1:1101:12677:1273 1:N:0:23
 CCTACGGGTGGCAGCAGTGAGGAATATTGGTCAATGGACGGAAGTCT
 +
 ABC8C,:@F:CE8,B-,C,-6-9-C,CE9-CC--C-&lt;-C++,,+;CE
 @M02286:19:000000000-AA549:1:1101:15048:1299 1:N:0:23
 CCTACGGGTGGCTGCAGTGAGGAATATTGGACAATGGTCGGAAGACT
 +
 ABC@CC77CFCEG;F9&lt;F89&lt;9--C,CE,--C-6C-,CE:++7:,CF
</code></pre></div></div>

<p><strong>File 2</strong></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@M02286:19:000000000-AA549:1:1101:12677:1273 2:N:0:23
CACTACCCGTGTATCTAATCCTGTTTGATACCCGCACCTTCGAGCTTA
+
--8A,CCE+,,;,&lt;CC,,&lt;CE@,CFD,,C,CFF+@+@CCEF,,,B+C,
@M02286:19:000000000-AA549:1:1101:15048:1299 2:N:0:23
CACTACCGGGGTATCTAATCCTGTTCGCTCCCCACGCTTTCGTCCATC
+
-6AC,EE@::CF7CFF&lt;&lt;FFGGDFFF,@FGGGG?F7FEGGGDEFF&gt;FF
</code></pre></div></div>

<blockquote class="comment">
  <h3 id="comment-read-order-is-important"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Read order is important</h3>
  <p>Note that read IDs are <strong>identical</strong> in two files and they are listed in <strong>the same</strong> order. In some cases read IDs in the first and second file may be appended with <code class="language-plaintext highlighter-rouge">/1</code> and <code class="language-plaintext highlighter-rouge">/2</code> tags, respectively.</p>
</blockquote>

<h4 id="interleaved-file">Interleaved file</h4>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@1/1
AGGGATGTGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTA
+
EGGEGGGDFGEEEAEECGDEGGFEEGEFGBEEDDECFEFDD@CDD&lt;ED
@1/2
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC
+
GHHHDFDFGFGEGFBGEGGEGEGGGHGFGHFHFHHHHHHHEF?EFEFF
@2/1
AGGGATGTGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTAGGGTTA
+
HHHHHHEGFHEEFEEHEEHHGGEGGGGEFGFGGGGHHHHFBEEEEEFG
@2/2
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAAC
+
HHHHHHHHHHHHHGHHHHHHGHHHHHHHHHHHFHHHFHHHHHHHHHHH
</code></pre></div></div>

<p>Here the first and the second <span class="notranslate">reads</span> are identified with <code class="language-plaintext highlighter-rouge">/1</code> and <code class="language-plaintext highlighter-rouge">/2</code> tags.</p>

<blockquote class="comment">
  <h3 id="comment-fastq-format-is-a-loose-standard"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> FASTQ format is a loose standard</h3>
  <p>FASTQ format is not strictly defined and its variations will always cause headache for you. See <a href="https://www.ncbi.nlm.nih.gov/sra/docs/submitformats/">this page</a> for more information.</p>
</blockquote>

<h2 id="what-are-base-qualities">What are base qualities?</h2>

<p>As we‚Äôve seen above, FASTQ datasets contain two types of information:</p>

<ul>
  <li><em>sequence of the read</em></li>
  <li><em>base qualities</em> for each nucleotide in the read.</li>
</ul>

<p>The base qualities allow us to judge how trustworthy each base in a sequencing read is. The following excerpt from an excellent <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D√ºndar, Luce Skrabanek, Paul Zumbo explains what base qualities are:</p>

<blockquote class="comment">
  <h3 id="comment-from-introduction-to-differential-gene-expression-analysis-using-rna-seq"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> From ‚ÄúIntroduction to differential gene <span class="notranslate">expression</span> analysis using <span class="notranslate">RNA</span>-seq‚Äù</h3>
  <p>Illumina sequencing is based on identifying the individual nucleotides by the fluorescence signal emitted upon their incorporation into the growing sequencing read. Once the fluorescence intensities are extracted and translated into the four letter code. The deduction of nucleotide sequences from the images acquired during sequencing is commonly referred to as base calling.
<br />
Due to the imperfect nature of the sequencing process and limitations of the optical instruments, base calling will always have inherent uncertainty. This is the reason why FASTQ files store the <span class="notranslate">DNA</span> sequence of each read together with a position-specific quality score that represents the error probability, i.e., how likely it is that an individual base call may be incorrect. The score is called <a href="http://www.phrap.com/phred/">Phred score</a>, \(Q\), which is proportional to the probability \(p\) that a base call is incorrect, where \(Q = ‚àí10lg(p)\). For example, a Phred score of 10 corresponds to one error in every ten base calls (\(Q = ‚àí10lg(0.1)\)), or 90% accuracy; a Phred score of 20 corresponds to one error in every 100 base calls, or 99% accuracy. A higher Phred score thus reflects higher confidence in the reported base.
<br />
To assign each base a unique score identifier (instead of numbers of varying character length), Phred scores are typically represented as ASCII characters. At http://ascii-code.com/ you can see which characters are assigned to what number.
<br />
For raw <span class="notranslate">reads</span>, the range of scores will depend on the sequencing technology and the base caller used (Illumina, for example, used a tool called Bustard, or, more recently, RTA). Unfortunately, Illumina has been anything but consistent in how they calculated and ASCII-encoded the Phred score (see below)! In addition, Illumina now allows Phred scores for base calls with as high as 45, while 41 used to be the maximum score until the HiSeq X. This may cause issues with <span class="notranslate">downstream</span> sapplications that expect an upper limit of 41.</p>
</blockquote>

<p><img src="../../images/illumina_qs.png" alt="Illumina quality score. " loading="lazy" /></p>

<p>Base call quality scores are represented with the Phred range. Different Illumina (formerly Solexa) versions
used different scores and ASCII offsets. Starting with Illumina format 1.8, the score now represents the standard
Sanger/Phred format that is also used by other sequencing platforms and the sequencing archives.</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/fastq_qs.png" alt="FASTQ quality score. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small>The ASCII interpretation and ranges of the different Phred score notations used by Illumina and the original Sanger interpretation. Although the Sanger format allows a theoretical score of 93, raw sequencing <span class="notranslate">reads</span> typically do not exceed a Phred score of 60. In fact, most Illumina-based sequencing will result in maximum scores of 41 to 45 (image from <a href="https://en.wikipedia.org/wiki/FASTQ_format">Wikipedia</a>)</small></td>
    </tr>
  </tbody>
</table>

<h2 id="assessing-data-quality">Assessing data quality</h2>

<p>One of the first steps in the analysis of NGS data is seeing how good the data actually is. <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastqQC</a> is a fantastic tool allowing you to assess the quality of FASTQ datasets (and deciding whether to blame or not to blame whoever has done sequencing for you).</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">¬†</th>
      <th style="text-align: left">¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left"><img src="../../images/good_fq.png" alt="Good quality in FastQC. " loading="lazy" /></td>
      <td style="text-align: left"><img src="../../images/bad_fq.png" alt="Bad quality in FastQC. " loading="lazy" /></td>
    </tr>
    <tr>
      <td style="text-align: left"><small><strong>A.</strong> Excellent quality</small></td>
      <td style="text-align: left"><small><strong>B.</strong> Hmmm‚Ä¶OK</small></td>
    </tr>
  </tbody>
</table>

<p>Here you can see FastQC base quality reports (the tools gives you many other types of data) for two datasets: <strong>A</strong> and <strong>B</strong>. The <strong>A</strong> dataset has long <span class="notranslate">reads</span> (250 bp) and very good quality profile with no qualities dropping below <a href="http://www.phrap.com/phred/">phred score</a> of 30. The <strong>B</strong> dataset is significantly worse with ends of the <span class="notranslate">reads</span> dipping below phred score of 20. The <strong>B</strong> <span class="notranslate">reads</span> may need to be trimmed for further processing.</p>

<!--
<div class="embed-responsive embed-responsive-16by9"><iframe src="https://player.vimeo.com/video/123453134?portrait=0" webkitallowfullscreen mozallowfullscreen allowfullscreen></iframe></div>
-->

<h2 id="mapping-your-data"><span class="notranslate">Mapping</span> your data</h2>

<p><span class="notranslate">Mapping</span> of NGS <span class="notranslate">reads</span> against reference sequences is one of the key steps of the analysis. Now it is time to see how this is done in practice. Below is a list of key publications highlighting mainstream <span class="notranslate">mapping</span> tools:</p>

<ul>
  <li>2009 Bowtie 1 - <a href="http://genomebiology.com/content/10/3/R25">Langmead et al.</a></li>
  <li>2012 Bowtie 2 - <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3322381/">Langmead and Salzberg</a></li>
  <li>2009 BWA - <a href="https://academic.oup.com/bioinformatics/article/25/14/1754/225615/Fast-and-accurate-short-read-alignment-with">Li and Durbin</a></li>
  <li>2010 BWA - <a href="https://academic.oup.com/bioinformatics/article/26/5/589/211735/Fast-and-accurate-long-read-alignment-with-Burrows">Li and Durbin</a></li>
  <li>2013 BWA-MEM - <a href="https://arxiv.org/abs/1303.3997">Li</a></li>
</ul>

<h3 id="mapping-against-a-pre-computed-genome-index"><span class="notranslate">Mapping</span> against a pre-computed genome index</h3>

<p>Mappers usually compare <span class="notranslate">reads</span> against a reference sequence that has been transformed into a highly accessible data structure called genome index. Such indexes should be generated before <span class="notranslate">mapping</span> begins. <span class="notranslate">Galaxy</span> instances typically store indexes for a number of publicly available genome builds.</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/cached_genome.png" alt="Cached genome. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small><span class="notranslate">Mapping</span> against a pre-computed index in <span class="notranslate">Galaxy</span>.</small></td>
    </tr>
  </tbody>
</table>

<p>For example, the image above shows indexes for <code class="language-plaintext highlighter-rouge">hg38</code> version of the human genome. You can see that there are actually three choices: (1) <code class="language-plaintext highlighter-rouge">hg38</code>, (2) <code class="language-plaintext highlighter-rouge">hg38 canonical</code> and (3) <code class="language-plaintext highlighter-rouge">hg38 canonical female</code>. The <code class="language-plaintext highlighter-rouge">hg38</code> contains all chromosomes as well as all unplaced contigs. The <code class="language-plaintext highlighter-rouge">hg38 canonical</code> does not contain unplaced sequences and only consists of chromosomes 1 through 22, X, Y, and mitochondria. The
<code class="language-plaintext highlighter-rouge">hg38 canonical female</code> contains everything from the canonical set with the exception of chromosome Y.</p>

<h3 id="what-if-pre-computed-index-does-not-exist">What if pre-computed index does not exist?</h3>

<p>If <span class="notranslate">Galaxy</span> does not have a genome you need to map against, you can upload your genome sequence as a FASTA file and use it in the mapper directly as shown below (<strong>Load reference genome</strong> is set to <code class="language-plaintext highlighter-rouge">History</code>).</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/uploaded_genome.png" alt="Uploaded genome. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small><span class="notranslate">Mapping</span> against a pre-computed index in <span class="notranslate">Galaxy</span> </small></td>
    </tr>
  </tbody>
</table>

<p>In this case <span class="notranslate">Galaxy</span> will first create an index from this dataset and then run <span class="notranslate">mapping</span> analysis against it.</p>

<h2 id="sambam-datasets">SAM/BAM datasets</h2>

<p>The <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM/BAM</a> format is an accepted standard for storing aligned <span class="notranslate">reads</span> (it can also store unaligned <span class="notranslate">reads</span> and some mappers such as BWA are accepting unaligned BAM as input). The binary form of the format (BAM) is compact and can be rapidly searched (if indexed). In <span class="notranslate">Galaxy</span> BAM datasets are always indexed (accompanies by a .bai file) and sorted in coordinate order. In the following duscussion I once again rely on <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D√ºndar, Luce Skrabanek, and Paul Zumbo.</p>

<p>The Sequence Alignment/Map (SAM) format is, in fact, a generic nucleotide alignment format that describes the alignment of sequencing <span class="notranslate">reads</span> (or query sequences) to a reference. The human readable, TABdelimited SAM files can be compressed into the Binary Alignment/Map format. These BAM files are bigger than simply gzipped SAM files, because they have been optimized for fast random access rather than size reduction. Position-sorted BAM files can be indexed so that all <span class="notranslate">reads</span> aligning to a locus can be efficiently retrieved without loading the entire file into memory.</p>

<p>As shown below, SAM files typically contain a short header section and a very long alignment section where each row represents a single read alignment. The following sections will explain the SAM format in a bit more detail. For the most comprehensive and updated information go to https://github.com/samtools/hts-specs.</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/bam_structure.png" alt="BAM structure. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small><strong>Schematic representation of a SAM file</strong>. Each line of the optional header section starts with ‚Äú@‚Äù, followed by the appropriate abbreviation (e.g., SQ for sequence dictionary which lists all chromosomes names (SN) and their lengths (LN)). The vast majority of lines within a SAM file typically correspond to read alignments where each read is described by the 11 mandatory entries (black font) and a variable number of optional fields (grey font). From <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D√ºndar, Luce Skrabanek, and Paul Zumbo.</small></td>
    </tr>
  </tbody>
</table>

<h3 id="sam-header">SAM Header</h3>

<p>The header section includes information about how the alignment was generated and stored. All lines in the header section are tab-delimited and begin with the ‚Äú@‚Äù character, followed by tag:value pairs, where tag is a two-letter string that defines the content and the format of value. For example, the ‚Äú@SQ‚Äù line in the header section contains the information about the names and lengths of the *reference sequences to which the <span class="notranslate">reads</span> were aligned. For a hypothetical organism with three chromosomes of length 1,000 bp, the SAM header should contain the following three lines:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@SQ SN:chr1 LN:1000
@SQ SN:chr2 LN:1000
@SQ SN:chr3 LN:1000
</code></pre></div></div>

<h3 id="sam-alignment-section">SAM alignment section</h3>

<p>The optional header section is followed by the alignment section where each line corresponds to one sequenced read. For each read, there are 11 mandatory fields that always appear in the same order:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&lt;QNAME&gt; &lt;FLAG&gt; &lt;<span class="notranslate">RNA</span>ME&gt; &lt;POS&gt; &lt;MAPQ&gt; &lt;CIGAR&gt; &lt;MRNM&gt; &lt;MPOS&gt; &lt;ISIZE&gt; &lt;SEQ&gt; &lt;QUAL&gt;
</code></pre></div></div>

<p>If the corresponding information is unavailable or irrelevant, field values can be ‚Äò0‚Äô or ‚Äò*‚Äô (depending on the field, see below), but they cannot be missing! After the 11 mandatory fields, a variable number of optional fields can be present. Here‚Äôs an example of one single line of a real-life SAM file (you may need to scroll sideways):</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ERR458493 .552967 16 chrI 140 255 12 M61232N37M2S * 0 0 CCACTCGTTCACCAGGGCCGGCGGGCTGATCACTTTATCGTGCATCTTGGC BB?HHJJIGHHJIGIIJJIJGIJIJJIIIGHBJJJJJJHHHHFFDDDA1+B NH:i:1 HI:i:1 AS:i:41 nM:i:2
</code></pre></div></div>

<p>The following table explains the format and content of each field. The <code class="language-plaintext highlighter-rouge">FLAG</code>, <code class="language-plaintext highlighter-rouge">CIGAR</code>, and the optional fields (marked in blue) are explained in more detail below. The number of optional fields can vary widely between different SAM files and even between <span class="notranslate">reads</span> within in the same file. The field types marked in blue are explained in more detail in the main text below.</p>

<p><img src="../../images/sam_fields.png" alt="SAM fields. " loading="lazy" /></p>

<h3 id="flag-field"><code class="language-plaintext highlighter-rouge">FLAG</code> field</h3>

<p>The FLAG field encodes various pieces of information about the individual read, which is particularly important for PE <span class="notranslate">reads</span>. It contains an integer that is generated from a sequence of bits (0, 1). This way, answers to multiple binary (Yes/No) questions can be compactly stored as a series of bits, where each of the single bits can be addressed and assigned separately.</p>

<p>The following table gives an overview of the different properties that can be encoded in the FLAG field. The developers of the SAM format and samtools tend to use the hexadecimal encoding as a means to refer to the different bits in their documentation. The value of the FLAG field in a given SAM file, however, will always be the decimal representation of the sum of the underlying binary values (as shown in Table below, row 2).</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/sam_flag.png" alt="SAM flag. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small>The <code class="language-plaintext highlighter-rouge">FLAG</code> field of SAM files stores information about the respective read alignment in one single decimal number. The decimal number is the sum of all the answers to the Yes/No questions associated with each binary bit. The hexadecimal representation is used to refer to the individual bits (questions). A bit is set if the corresponding state is true. For example, if a read is paired, <code class="language-plaintext highlighter-rouge">0x1</code> will be set, returning the decimal value of 1. Therefore, all <code class="language-plaintext highlighter-rouge">FLAG</code> values associated with paired <span class="notranslate">reads</span> must be uneven decimal numbers. Conversely, if the <code class="language-plaintext highlighter-rouge">0x1</code> bit is unset (= read is not paired), no assumptions can be made about <code class="language-plaintext highlighter-rouge">0x2</code>, <code class="language-plaintext highlighter-rouge">0x8</code>, <code class="language-plaintext highlighter-rouge">0x20</code>, <code class="language-plaintext highlighter-rouge">0x40</code> and <code class="language-plaintext highlighter-rouge">0x80</code> because they refer to paired <span class="notranslate">reads</span>. From <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D√ºndar, Luce Skrabanek, and Paul Zumbo</small></td>
    </tr>
  </tbody>
</table>

<p>In a run with single <span class="notranslate">reads</span>, the flags you most commonly see are:</p>

<ul>
  <li>0: This read has been mapped to the forward strand. (None of the bit-wise flags have been set.)</li>
  <li>4: The read is unmapped (<code class="language-plaintext highlighter-rouge">0x4</code> is set).</li>
  <li>16: The read is mapped to the reverse strand (<code class="language-plaintext highlighter-rouge">0x10</code> is set)</li>
</ul>

<p>(<code class="language-plaintext highlighter-rouge">0x100</code>, <code class="language-plaintext highlighter-rouge">0x200</code> and <code class="language-plaintext highlighter-rouge">0x400</code> are not used by most aligners/mappers, but could, in principle be set for single <span class="notranslate">reads</span>.) Some common <code class="language-plaintext highlighter-rouge">FLAG</code> values that you may see in a PE experiment include:</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><strong>69</strong> (= 1 + 4 + 64)</td>
      <td>The read is paired, is the first read in the pair, and is unmapped.</td>
    </tr>
    <tr>
      <td><strong>77</strong> (= 1 + 4 + 8 + 64)</td>
      <td>The read is paired, is the first read in the pair, both are unmapped.</td>
    </tr>
    <tr>
      <td><strong>83</strong> (= 1 + 2 + 16 + 64)</td>
      <td>The read is paired, mapped in a proper pair, is the first read in the pair, and it is mapped to the reverse strand.</td>
    </tr>
    <tr>
      <td><strong>99</strong> (= 1 + 2 + 32 + 64)</td>
      <td>The read is paired, mapped in a proper pair, is the first read in the pair, and its mate is mapped to the reverse strand.</td>
    </tr>
    <tr>
      <td><strong>133</strong> (= 1 + 4 + 128)</td>
      <td>The read is paired, is the second read in the pair, and it is unmapped.</td>
    </tr>
    <tr>
      <td><strong>137</strong> (= 1 + 8 + 128)</td>
      <td>The read is paired, is the second read in the pair, and it is mapped while its mate is not.</td>
    </tr>
    <tr>
      <td><strong>141</strong> (= 1 + 4 + 8 + 128)</td>
      <td>The read is paired, is the second read in the pair, but both are unmapped.</td>
    </tr>
    <tr>
      <td><strong>147</strong> (= 1 + 2 + 16 + 128)</td>
      <td>The read is paired, mapped in a proper pair, is the second read in the pair, and mapped to the reverse strand.</td>
    </tr>
    <tr>
      <td><strong>163</strong> (= 1 + 2 + 32 + 128)</td>
      <td>The read is paired, mapped in a proper pair, is the second read in the pair, and its mate is mapped to the reverse strand.</td>
    </tr>
  </tbody>
</table>

<p>A useful website for quickly translating the FLAG integers into plain English explanations like the ones shown above is: https://broadinstitute.github.io/picard/explain-flags.html</p>

<h3 id="cigar-string"><code class="language-plaintext highlighter-rouge">CIGAR</code> string</h3>

<p><code class="language-plaintext highlighter-rouge">CIGAR</code> stands for <em>Concise Idiosyncratic Gapped Alignment Report</em>. This sixth field of a SAM file
contains a so-called CIGAR string indicating which operations were necessary to map the read to the reference sequence at that particular locus.</p>

<p>The following operations are defined in CIGAR format (also see figure below):</p>

<ul>
  <li><strong>M</strong> - Alignment (can be a sequence match or mismatch!)</li>
  <li><strong>I</strong> - Insertion in the read compared to the reference</li>
  <li><strong>D</strong> - Deletion in the read compared to the reference</li>
  <li><strong>N</strong> - Skipped region from the reference. For m<span class="notranslate">RNA</span>-to-genome alignments, an N operation represents an intron. For other types of alignments, the interpretation of N is not defined.</li>
  <li><strong>S</strong> - Soft clipping (clipped sequences are present in read); S may only have H operations between them and the ends of the string</li>
  <li><strong>H</strong> - Hard clipping (clipped sequences are NOT present in the alignment record); can only be present as the first and/or last operation</li>
  <li><strong>P</strong> - Padding (silent deletion from padded reference)</li>
  <li><strong>=</strong> - Sequence match (not widely used)</li>
  <li><strong>X</strong> - Sequence mismatch (not widely used)</li>
</ul>

<p>The sum of lengths of the <strong>M</strong>, <strong>I</strong>, <strong>S</strong>, <strong>=</strong>, <strong>X</strong> operations must equal the length of the read. Here are some examples:</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/cigar.png" alt="CIGAR. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small>From <a href="http://chagall.med.cornell.edu/RNASEQcourse/Intro2RNAseq.pdf">tutorial</a> by Friederike D√ºndar, Luce Skrabanek, and Paul Zumbo.</small></td>
    </tr>
  </tbody>
</table>

<h3 id="optional-fields">Optional fields</h3>

<p>Following the eleven mandatory SAM file fields, the optional fields are presented as key-value
pairs in the format of <code class="language-plaintext highlighter-rouge">&lt;TAG&gt;:&lt;TYPE&gt;:&lt;VALUE&gt;</code>, where <code class="language-plaintext highlighter-rouge">TYPE</code> is one of:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">A</code> - Character</li>
  <li><code class="language-plaintext highlighter-rouge">i</code> - Integer</li>
  <li><code class="language-plaintext highlighter-rouge">f</code> - Float number</li>
  <li><code class="language-plaintext highlighter-rouge">Z</code> - String</li>
  <li><code class="language-plaintext highlighter-rouge">H</code> - Hex string</li>
</ul>

<p>The information stored in these optional fields will vary widely depending on the mapper and new tags can be added freely. In addition, <span class="notranslate">reads</span> within the same SAM file may have different numbers of optional fields, depending on the program that generated the SAM file. Commonly used optional tags include:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">AS:i</code> - Alignment score</li>
  <li><code class="language-plaintext highlighter-rouge">BC:Z</code> - Barcode sequence</li>
  <li><code class="language-plaintext highlighter-rouge">HI:i</code> - Match is i-th hit to the read</li>
  <li><code class="language-plaintext highlighter-rouge">NH:i</code> - Number of reported alignments for the query sequence</li>
  <li><code class="language-plaintext highlighter-rouge">NM:i</code> - Edit distance of the query to the reference</li>
  <li><code class="language-plaintext highlighter-rouge">MD:Z</code> - String that contains the exact positions of mismatches (should complement the CIGAR string)</li>
  <li><code class="language-plaintext highlighter-rouge">RG:Z</code> - Read group (should match the entry after ID if @RG is present in the header.</li>
</ul>

<p>Thus, for example, we can use the NM:i:0 tag to select only those <span class="notranslate">reads</span> which map perfectly to the reference(i.e., have no mismatches). While the optional fields listed above are fairly standardized, tags that begin with <code class="language-plaintext highlighter-rouge">X</code>, <code class="language-plaintext highlighter-rouge">Y</code>, and <code class="language-plaintext highlighter-rouge">Z</code> are reserved for particularly free usage and will never be part of the official SAM file format specifications. <code class="language-plaintext highlighter-rouge">XS</code>, for example, is used by TopHat (an <span class="notranslate">RNA</span>-seq analysis tool we will discuss later) to encode the strand information (e.g., <code class="language-plaintext highlighter-rouge">XS:A:+</code>) while Bowtie2 and BWA use <code class="language-plaintext highlighter-rouge">XS:i:</code> for <span class="notranslate">reads</span> with multiple alignments to store the alignment score for the next-best-scoring alignment (e.g., <code class="language-plaintext highlighter-rouge">XS:i:30</code>).</p>

<h3 id="read-groups">Read Groups</h3>

<p>One of the key features of SAM/BAM format is the ability to label individual <span class="notranslate">reads</span> with readgroup tags. This allows pooling results of multiple experiments into a single BAM dataset. This significantly simplifies <span class="notranslate">downstream</span> logistics: instead of dealing with multiple datasets one can handle just one. Many <span class="notranslate">downstream</span> analysis tools such as variant callers are designed to recognize readgroup data and output results on per-readgroup basis.</p>

<p>One of the best descriptions of BAM readgroups is on <a href="https://gatkforums.broadinstitute.org/discussion/1317/collected-faqs-about-bam-files">GATK support site</a>. We have gratefully stolen two tables describing the most important readgroup tags - <code class="language-plaintext highlighter-rouge">ID</code>, <code class="language-plaintext highlighter-rouge">SM</code>, <code class="language-plaintext highlighter-rouge">LB</code>, and <code class="language-plaintext highlighter-rouge">PL</code> - from GATK forum and provide them here:</p>

<p><img src="../../images/rg.png" alt="Read groups. " loading="lazy" /></p>

<p>GATK forum also provides the following example:</p>

<p><img src="../../images/rg_example.png" alt="Read group example. " loading="lazy" /></p>

<h3 id="manipulating-sambam-datasets">Manipulating SAM/BAM datasets</h3>

<p>We support four major toolsets for processing of SAM/BAM datasets:</p>

<ul>
  <li><a href="https://deeptools.readthedocs.io">DeepTools</a> - a suite of user-friendly tools for the visualization, quality control and normalization of data from deep-sequencing <span class="notranslate">DNA</span> sequencing experiments.</li>
  <li><a href="http://www.htslib.org/">SAMtools</a> - various utilities for manipulating alignments in the SAM/BAM format, including sorting, merging, indexing and generating alignments in a per-position format.</li>
  <li><a href="https://bedtools.readthedocs.io/en/latest/"><span class="notranslate">BED</span>tools</a> - a toolkit originally written for <span class="notranslate">BED</span> format was expanded for analysis of BAM and VCF datasets.</li>
  <li><a href="https://broadinstitute.github.io/picard/">Picard</a> - a set of Java tools for manipulating high-throughput sequencing data (HTS) data and formats.</li>
</ul>

<h2 id="the-challenge-of-read-duplicates">The challenge of read duplicates</h2>

<h3 id="pcr-duplicates">PCR duplicates</h3>

<p>Preparation of sequencing libraries (at least at the time of writing) for technologies such as Illumina (used in this example) involves PCR amplification. It is required to generate sufficient number of sequencing templates so that a reliable detection can be performed by base callers. Yet PCR has its biases, which are especially profound in cases of multitemplate PCR used for construction of sequencing libraries (Kanagawa et al. <a href="https://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Abstract&amp;list_uids=16233530">2003</a>).</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/pcr-duplicates.png" alt="PCR duplicates. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small>Analyzing molecules aligning with the same outer coordinates, a <span class="notranslate">mapping</span> quality of at least 30 and a length of at least 30nt, resulted in an average coverage of 12.9 per PCR duplicate and an empirical coverage distribution similar to an exponential/power law distribution (left upper panel). This indicates that many molecules are only observed for deeper sequencing while other molecules are available at higher frequencies. Analyzing length (left middle panel) and GC content (left lower panel) patterns as well as the combination (right panel) shows higher PCR duplicate counts for a GC content between 30% to 70% as well as for shorter molecules compared to longer molecules. This effect may be due to an amplification bias from the polymerase or the cluster generation process necessary for Illumina sequencing. From Ph.D. dissertation of <a href="https://ul.qucosa.de/api/qucosa%3A11231/attachment/ATT-0/">Martin Kircher</a>).</small></td>
    </tr>
  </tbody>
</table>

<p>Duplicates can be identified based on their outer alignment coordinates or using sequence-based clustering. One of the common ways for identification of duplicate <span class="notranslate">reads</span> is the <code class="language-plaintext highlighter-rouge">MarkDuplicates</code> utility from <a href="https://broadinstitute.github.io/picard/command-line-overview.html">Picard</a> package. It is designed to identify both PCR and optical duplicates:</p>

<div class="well well-lg">

Duplicates are identified as read pairs having identical 5' positions (coordinate and strand) for both <span class="notranslate">reads</span> in a mate pair (and optionally, matching unique molecular identifier <span class="notranslate">reads</span>; see BARCODE_TAG option). Optical, or more broadly Sequencing, duplicates are duplicates that appear clustered together spatially during sequencing and can arise from optical/imagine-processing artifacts or from bio-chemical processes during clonal amplification and sequencing; they are identified using the READ_NAME_REGEX and the OPTICAL_DUPLICATE_PIXEL_DISTANCE options. The tool's main output is a new SAM or BAM file in which duplicates have been identified in the SAM flags field, or optionally removed (see REMOVE_DUPLICATE and REMOVE_SEQUENCING_DUPLICATES), and optionally marked with a duplicate type in the 'DT' optional attribute. In addition, it also outputs a metrics file containing the numbers of READ_PAIRS_EXAMINED, UNMAPPED_<span class="notranslate">READS</span>, UNPAIRED_<span class="notranslate">READS</span>, UNPAIRED_READ DUPLICATES, READ_PAIR_DUPLICATES, and READ_PAIR_OPTICAL_DUPLICATES. Usage example: java -jar picard.jar MarkDuplicates I=input.bam \ O=marked_duplicates.bam M=marked_dup_metrics.txt.`

</div>

<h3 id="sampling-coincidence-duplicates">Sampling coincidence duplicates</h3>

<p>However, one has to be careful when removing duplicates in cases when the sequencing targets are small (e.g., sequencing of bacterial, viral, or organellar genomes as well as amplicons). This is because when sequencing target is small <span class="notranslate">reads</span> will have the same coordinates by chance and not because of PCR amplification issues. The figure below illustrates the fine balance between estimates allele frequency, coverage, and variation in insert size:</p>

<table>
  <thead>
    <tr>
      <th>¬†</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="../../images/sampling-bias.png" alt="Sampling bias. " loading="lazy" /></td>
    </tr>
    <tr>
      <td><small>The Variant Allele Frequency (VAF) bias determined by coverage and insert size variance. <span class="notranslate">Reads</span> are paired-end and read length is 76. The insert size distribution is modeled as a Gaussian distribution with mean at 200 and standard deviation shown on the x-axis. The true VAF is 0.05. The darkness at each position indicates the magnitude of the bias in the VAF. (From Zhou et al. <a href="https://bioinformatics.oxfordjournals.org/content/30/8/1073">2013</a>).</small></td>
    </tr>
  </tbody>
</table>

<h1 id="getting-ngs-data-to-galaxy">Getting NGS data to <span class="notranslate">Galaxy</span></h1>

<p>You can upload data in <span class="notranslate">Galaxy</span> using one of these ways:</p>

<h2 id="from-your-computer">From your computer</h2>

<p>This works well for small files because web browser do not like lengthy file transfers:</p>

<iframe width="560" height="315" src="https://www.youtube.com/embed/FFCDx1rMGAQ" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>

<h2 id="using-ftp">Using FTP</h2>

<p>FTP (<a href="https://en.wikipedia.org/wiki/File_Transfer_Protocol">file transfer protocol</a>) allows transferring large collection of files:</p>

<iframe width="560" height="315" src="https://www.youtube.com/embed/hC8KSuT_OP8" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>

<h2 id="from-ncbi-short-read-archive">From NCBI short read archive</h2>

<p>Finally, datasets can be uploaded directly from NCBI‚Äôs short read archive:</p>

<iframe width="560" height="315" src="https://www.youtube.com/embed/Q4t-beYZ-do" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>We will use this last approach, getting data from NCBI SRA, in this tutorial.</p>

</blockquote>

<h1 id="lets-do-it-from-reads-to-variants">Let‚Äôs do it: From <span class="notranslate">reads</span> to variants</h1>

<p>In primary analysis we start with raw sequencing data (e.g., fastq <span class="notranslate">reads</span>) and convert them into a dataset for secondary analysis. Such dataset can be a list of sequence variants, a collection of ChIP-seq peaks, a list of differentially expressed genes and so on.</p>

<p>SARS-CoV-2 is undoubtedly the most talked about biological system of the day. We will use SARS-CoV-2 sequencing data to demonstrate how <span class="notranslate">Galaxy</span> handles NGS data. In this case we will go from FASTQ data to a list of variants.</p>

<h2 id="find-necessary-data-in-sra">Find necessary data in SRA</h2>

<p>First we need to find a good dataset to play with. The <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive (SRA)</a> is the primary archive of <em>unassembled <span class="notranslate">reads</span></em>  operated by the <a href="https://www.ncbi.nlm.nih.gov/">US National Institutes of Health (NIH)</a>.  SRA is a great place to get the sequencing data that underlie publications and studies. Let‚Äôs do that:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-metadata-from-ncbi-sra"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get Metadata from NCBI SRA</h3>

  <ol>
    <li>Go to NCBI‚Äôs SRA page by pointing your browser to https://www.ncbi.nlm.nih.gov/sra</li>
    <li>In the search box enter <code class="language-plaintext highlighter-rouge">SARS-CoV-2 Patient Sequencing From Partners / MGH</code>:
<img src="../../images/find_mgh_data.png" alt="Find data. " loading="lazy" /> (Alte<span class="notranslate">rna</span>tively, you simply click on this <a href="https://www.ncbi.nlm.nih.gov/sra/?term=SARS-CoV-2+Patient+Sequencing+From+Partners+%2F+MGH">link</a>)</li>
    <li>The web page will show a large number of SRA datasets (at the time of writing there were 2,223). This is data from a <a href="https://science.sciencemag.org/content/early/2020/12/09/science.abe3261">study</a> describing analysis of SARS-CoV-2 in Boston area.</li>
    <li>Download metadata describing these datasets by:
      <ul>
        <li>clicking on <strong>Send to:</strong> dropdown</li>
        <li>Selecting <code class="language-plaintext highlighter-rouge">File</code></li>
        <li>Changing <strong>Format</strong> to <code class="language-plaintext highlighter-rouge">RunInfo</code></li>
        <li>Clicking <strong>Create file</strong>
Here is how it should look like:
<img src="../../images/get_runinfo.png" alt="Getting RunInfo from SRA. " loading="lazy" /></li>
      </ul>
    </li>
    <li>This would create a rather large <code class="language-plaintext highlighter-rouge">SraRunInfo.csv</code> file in your <code class="language-plaintext highlighter-rouge">Downloads</code> folder.</li>
  </ol>
</blockquote>

<p>Now that we have downloaded this file we can go to a <span class="notranslate">Galaxy</span> instance and start processing it.</p>

<blockquote class="comment">
  <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Note that the file we just downloaded is <strong>not</strong> sequencing data itself. Rather, it is <em>metadata</em> describing properties of sequencing <span class="notranslate">reads</span>. We will filter this list down to just a few accessions that will be used in the remainder of this tutorial.</p>

</blockquote>

<h2 id="process-and-filter-sraruninfocsv-file-in-galaxy">Process and filter <code class="language-plaintext highlighter-rouge">SraRunInfo.csv</code> file in <span class="notranslate">Galaxy</span></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-upload-sraruninfocsv-file-into-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Upload <code class="language-plaintext highlighter-rouge">SraRunInfo.csv</code> file into <span class="notranslate">Galaxy</span></h3>

  <ol>
    <li>Go to your <span class="notranslate">Galaxy</span> instance of choice such as one of the <a href="https://usegalaxy.org/">use<span class="notranslate">galaxy</span>.org</a>, <a href="https://usegalaxy.eu">use<span class="notranslate">galaxy</span>.eu</a>, <a href="https://usegalaxy.org.au">use<span class="notranslate">galaxy</span>.org.au</a> or any other. (This tutorial uses use<span class="notranslate">galaxy</span>.org).</li>
    <li>Click <em>Upload Data</em> button:
<img src="../../images/upload_button.png" alt="Data upload button. " loading="lazy" /></li>
    <li>In the dialog box that would appear click ‚Äú<em>Choose local files</em>‚Äù button:
<img src="../../images/choose_local_files_button.png" alt="Choose local files button. " loading="lazy" /></li>
    <li>Find and select <code class="language-plaintext highlighter-rouge">SraRunInfo.csv</code> file from your computer</li>
    <li>Click <em>Start</em> button</li>
    <li>Close dialog by pressing <strong>Close</strong> button</li>
    <li>You can now look at the content of this file by clicking <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon. You will see that this file contains a lot of information about individual SRA accessions. In this study every accession corresponds to an individual patient whose samples were sequenced.</li>
  </ol>
</blockquote>

<p><span class="notranslate">Galaxy</span> can process all 2,000+ datasets, but to make this tutorial bearable we need to selected a smaller subset. In particular our previous experience with this data shows two interesting datasets <code class="language-plaintext highlighter-rouge">SRR11954102</code> and <code class="language-plaintext highlighter-rouge">SRR12733957</code>. So, let‚Äôs pull them out.</p>

<blockquote class="warning">
  <h3 id="warning-beware-of-cuts"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Beware of <strong>Cut</strong>s</h3>
  <p>The Hands-on section below uses <strong>Cut</strong> tool. There are two <strong>cut</strong> tools in <span class="notranslate">Galaxy</span> due to historical reasons. This example uses tool with the full name <strong>Cut columns from a table (cut)</strong>. However, the same logic applies to the other tool. It simply has a slightly different interface.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-creating-a-subset-of-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Creating a subset of data</h3>

  <ol>
    <li>Find <span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  tool in <strong>Filter and Sort</strong> section of the tool panel.
      <blockquote class="tip">
        <h3 id="tip-tip-finding-tools"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Finding tools</h3>
        <p><span class="notranslate">Galaxy</span> may have an overwhelming amount of tools installed. To find a specific tool type the tool name in the tool panel search box to find the tool.</p>
      </blockquote>
    </li>
    <li>Make sure the <code class="language-plaintext highlighter-rouge">SraRunInfo.csv</code> dataset we just uploaded is listed in the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> ‚Äú<em>Select lines from</em>‚Äù field of the tool form.</li>
    <li>In ‚Äú<em>the pattern</em>‚Äù field enter the following <span class="notranslate">expression</span> ‚Üí <code class="language-plaintext highlighter-rouge">SRR12733957|SRR11954102</code>. These are two accession we want to find separated by the pipe symbol <code class="language-plaintext highlighter-rouge">|</code>. The <code class="language-plaintext highlighter-rouge">|</code> means <code class="language-plaintext highlighter-rouge">or</code>: find lines containing <code class="language-plaintext highlighter-rouge">SRR12733957</code> <strong>or</strong> <code class="language-plaintext highlighter-rouge">SRR11954102</code>.</li>
    <li>Click <code class="language-plaintext highlighter-rouge">Execute</code> button.</li>
    <li>This will generate a file containing two lines (well ‚Ä¶ one line is also used as the header, so it will appear the the file has three lines. It is OK.)</li>
    <li>Cut the first column from the file using <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0"><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</span></span>  tool, which you will find in <strong>Text Manipulation</strong> section of the tool pane.</li>
    <li>Make sure the dataset produced by the previous step is selected in the ‚Äú<em>File to cut</em>‚Äù field of the tool form.</li>
    <li>Change ‚Äú<em>Delimited by</em>‚Äù to <code class="language-plaintext highlighter-rouge">Comma</code></li>
    <li>In ‚Äú<em>List of fields</em>‚Äù select <code class="language-plaintext highlighter-rouge">Column: 1</code>.</li>
    <li>Hit <code class="language-plaintext highlighter-rouge">Execute</code>
This will produce a text file with just two lines:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR12733957
SRR11954102
</code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<p>Now that we have identifiers of datasets we want we need to download the actual sequencing data. You can also watch <a href="#from-ncbi-short-read-archive">this video</a>.</p>

<p>The remaining portion of this tutorial can also be viewed as a video:</p>

<iframe width="560" height="315" src="https://www.youtube.com/embed/9mIL0tIfZ_o" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>

<h2 id="download-sequencing-data">Download sequencing data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-data-from-sra"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get data from SRA</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.10.9+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.10.9+galaxy0"><strong>Faster Download and Extract <span class="notranslate">Reads</span> in FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/sra_tools/fasterq_dump/2.10.9+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><em>‚Äúselect input type‚Äù</em>: <code class="language-plaintext highlighter-rouge">List of SRA accession, one per line</code>
          <ul>
            <li>The parameter <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúsra accession list‚Äù</em> should point the output of the <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> ‚Äú<strong>Cut</strong>‚Äù from the previous step.</li>
          </ul>
        </li>
        <li><strong>Click</strong> the <code class="language-plaintext highlighter-rouge">Execute</code> button. This will run the tool, which retrieves the sequence read datasets for the runs that were listed in the <code class="language-plaintext highlighter-rouge">SRA</code> dataset. It may take some time. So this may be a good time to take a break.</li>
      </ul>
    </li>
    <li>Several entries are created in your history panel when you submit this job:
      <ul>
        <li><strong><code class="language-plaintext highlighter-rouge">Pair-end data (fasterq-dump)</code></strong>: Contains Paired-end datasets (if available)</li>
        <li><strong><code class="language-plaintext highlighter-rouge">Single-end data (fasterq-dump)</code></strong> Contains Single-end datasets (if available)</li>
        <li><strong><code class="language-plaintext highlighter-rouge">Other data (fasterq-dump)</code></strong> Contains Unpaired datasets (if available)</li>
        <li><strong><code class="language-plaintext highlighter-rouge">fasterq-dump log</code></strong> Contains Information about the tool execution</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The first three items are actually <em>collections</em> of datasets. <em>Collections</em> in <span class="notranslate">Galaxy</span> are logical groupings of datasets that reflect the semantic relationships between them in the experiment / analysis. In this case the tool creates separate collections for paired-end <span class="notranslate">reads</span>, single <span class="notranslate">reads</span>, and <em>other</em>. See the <a href="https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/collections/tutorial.html">Collections tutorial</a> and watch <a href="https://youtube.com/playlist?list=PLNFLKDpdM3B9UaxWEXgziHXO3k-003FzE">videos</a> (with names beginning with ‚ÄúDataset Collections‚Äù) for more information.</p>

<p>Explore the collections by first <strong>clicking</strong> on the collection name in the history panel. This takes you inside the collection and shows you the datasets in it.  You can then navigate back to the outer level of your history.</p>

<p>Once <code class="language-plaintext highlighter-rouge">fasterq</code> finishes transferring data (all boxes are green / done), we are ready to analyze it.</p>

<h2 id="now-what">Now what?</h2>

<p>You can now analyze the retrieved data using any sequence analysis tools and workflows in <span class="notranslate">Galaxy</span>.  SRA holds backing data for every imaginable type of *-seq experiment.</p>

<p>If you ran this tutorial, but retrieved datasets that you were interested in, then see the rest of the GTN library for ideas on how to analyze in <span class="notranslate">Galaxy</span>.</p>

<p>However, if you retrieved the datasets used in this tutorial‚Äôs examples above, then you are ready to run the SARS-CoV-2 variant analysis below.</p>

<p>In this part of the tutorial we will perform variant calling and basic analysis of the datasets downloaded above. We will start by downloading the Wuhan-Hu-1 SARS-CoV-2 reference sequence, then run adapter trimming, alignment and variant calling.</p>

<blockquote class="comment">
  <h3 id="comment-the-usegalaxy-covid-19-analysis-project"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> The use<span class="notranslate">galaxy</span>.* COVID-19 analysis project</h3>
  <p>This tutorial uses a subset of the data and runs through the
<a href="https://covid19.galaxyproject.org/genomics/4-variation/">Variation Analysis</a>
section of <a href="https://covid19.galaxyproject.org/">covid19.<span class="notranslate">galaxy</span>project.org</a>.
The data for <a href="https://covid19.galaxyproject.org/">covid19.<span class="notranslate">galaxy</span>project.org</a> is
being updated continuously as new datasets are made public.</p>
</blockquote>

<h2 id="get-the-reference-genome-data">Get the reference genome data</h2>

<p>The reference genome data for today is for SARS-CoV-2, ‚ÄúSevere acute respiratory syndrome coronavirus 2 isolate Wuhan-Hu-1, complete genome‚Äù, having the accession ID of NC_045512.2.</p>

<p>This data is available from directly from GenBank using the following <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/009/858/895/GCF_009858895.2_ASM985889v3/GCF_009858895.2_ASM985889v3_genomic.fna.gz">link</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-the-reference-genome-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get the reference genome data</h3>

  <ol>
    <li>
      <p>Import the following file into your history:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/009/858/895/GCF_009858895.2_ASM985889v3/GCF_009858895.2_ASM985889v3_genomic.fna.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">  <h3 data-toc-skip="" id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li>    <p><strong>Close</strong> the window</p>  </li>  <li>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</li></ul></blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="adapter-trimming-with-fastp">Adapter trimming with <strong>fastp</strong></h2>

<p>Removing sequencing adapters improves alignments and variant calling. <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> can automatically detect widely used sequencing adapters.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-running-fastp"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Running <code class="language-plaintext highlighter-rouge">fastp</code></h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+galaxy0"><strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/fastp/fastp/0.20.1+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:</p>
  <ul>
    <li><em>‚ÄúSingle-end or paired <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect paired collection(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">list_paired</code> (output of <strong>Faster Download and Extract <span class="notranslate">Reads</span> in FASTQ</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li>In <em>‚ÄúOutput Options‚Äù</em>:
      <ul>
        <li><em>‚ÄúOutput JSON report‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ul>
</blockquote>

<h2 id="alignment-with--map-with-bwa-mem">Alignment with  <strong>Map with BWA-MEM</strong></h2>

<p><strong>BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is a widely used sequence aligner for short-read sequencing datasets such as those we are analysing in this tutorial.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-map-sequencing-reads-to-reference-genome"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Map sequencing <span class="notranslate">reads</span> to reference genome</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1"><strong>BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1</span></span>  with the following parameters:</p>
  <ul>
    <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Input dataset)</li>
      </ul>
    </li>
    <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired Collection</code>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect a paired collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_paired_coll</code> (output of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
    <li><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
  </ul>

</blockquote>

<h2 id="remove-duplicates-with-markduplicates">Remove duplicates with <strong>MarkDuplicates</strong></h2>

<p><strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> removes duplicate sequences originating from library preparation artifacts and sequencing artifacts. It is important to remove these artefactual sequences to avoid artificial overrepresentation of single molecule.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove duplicates</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2"><strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect SAM/BAM dataset or dataset collection‚Äù</em>: <code class="language-plaintext highlighter-rouge">bam_output</code> (output of <strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúIf true do not write duplicates to the output file instead of writing them with appropriate flags set‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
  </ul>

</blockquote>

<!-- Not used in this version since this step was not included in video

## Generate alignment statistics with **Samtools stats**

After the duplicate marking step above we can generate statistic about the alignment we have generated.

> ### <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate alignment statistics
>
> 1. **Samtools stats** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
>    - <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> *"BAM file"*: `outFile` (output of **MarkDuplicates** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)
>    - *"Set coverage distribution"*: `No`
>    - *"Output"*: `One single summary file`
>    - *"Filter by SAM flags"*: `Do not filter`
>    - *"Use a reference sequence"*: `No`
>    - *"Filter by regions"*: `No`
{: .hands_on}

-->

<h2 id="realign-reads-with-lofreq-viterbi"><strong>Realign <span class="notranslate">reads</span></strong> with lofreq viterbi</h2>

<p><strong>Realign <span class="notranslate">reads</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> corrects misalignments around insertions and deletions. This is required in order to accurately detect variants.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-realign-reads-around-indels"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Realign <span class="notranslate">reads</span> around indels</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+galaxy0"><strong>Realign <span class="notranslate">reads</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/lofreq_viterbi/lofreq_viterbi/2.1.5+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">Reads</span> to realign‚Äù</em>: <code class="language-plaintext highlighter-rouge">outFile</code> (output of <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">History</code>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúReference‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Input dataset)</li>
      </ul>
    </li>
    <li>In <em>‚ÄúAdvanced options‚Äù</em>:
      <ul>
        <li><em>‚ÄúHow to handle base qualities of 2?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Keep unchanged</code></li>
      </ul>
    </li>
  </ul>
</blockquote>

<h2 id="add-indel-qualities-with-lofreq-insert-indel-qualities">Add indel qualities with lofreq <strong>Insert indel qualities</strong></h2>

<p>This step adds indel qualities into our alignment file. This is necessary in order to call variants using <strong>Call variants</strong> with lofreq <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-indel-qualities"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add indel qualities</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+galaxy0"><strong>Insert indel qualities</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/lofreq_indelqual/lofreq_indelqual/2.1.5+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">Reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">realigned</code> (output of <strong>Realign <span class="notranslate">reads</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúIndel calculation approach‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dindel</code>
      <ul>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">History</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúReference‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Input dataset)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>

</blockquote>

<h2 id="call-variants-using-lofreq-call-variants">Call Variants using lofreq <strong>Call variants</strong></h2>

<p>We are now ready to call variants.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-call-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Call variants</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+galaxy1"><strong>Call variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/lofreq_call/lofreq_call/2.1.5+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput <span class="notranslate">reads</span> in BAM format‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>Insert indel qualities</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">History</code>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúReference‚Äù</em>: <code class="language-plaintext highlighter-rouge">output</code> (Input dataset)</li>
      </ul>
    </li>
    <li><em>‚ÄúCall variants across‚Äù</em>: <code class="language-plaintext highlighter-rouge">Whole reference</code></li>
    <li><em>‚ÄúTypes of variants to call‚Äù</em>: <code class="language-plaintext highlighter-rouge">SNVs and indels</code></li>
    <li><em>‚ÄúVariant calling parameters‚Äù</em>: <code class="language-plaintext highlighter-rouge">Configure settings</code>
      <ul>
        <li>In <em>‚ÄúCoverage‚Äù</em>:
          <ul>
            <li><em>‚ÄúMinimal coverage‚Äù</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúBase-calling quality‚Äù</em>:
          <ul>
            <li><em>‚ÄúMinimum baseQ‚Äù</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
            <li><em>‚ÄúMinimum baseQ for alte<span class="notranslate">rna</span>te bases‚Äù</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
          </ul>
        </li>
        <li>In <em>‚Äú<span class="notranslate">Mapping</span> quality‚Äù</em>:
          <ul>
            <li><em>‚ÄúMinimum <span class="notranslate">mapping</span> quality‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><em>‚ÄúVariant filter parameters‚Äù</em>: <code class="language-plaintext highlighter-rouge">Preset filtering on QUAL score + coverage + strand bias (lofreq call default)</code></li>
  </ul>
</blockquote>

<p>The output of this step is a collection of VCF files that can be visualized in a genome browser.</p>

<h2 id="annotate-variant-effects-with-snpeff-eff">Annotate variant effects with <strong>SnpEff eff:</strong></h2>

<p>We will now annotate the variants we called in the previous step with the effect they have on the SARS-CoV-2 genome.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-annotate-variant-effects"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Annotate variant effects</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19"><strong>SnpEff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/snpeff_sars_cov_2/snpeff_sars_cov_2/4.5covid19</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSequence changes (SNPs, MNPs, InDels)‚Äù</em>: <code class="language-plaintext highlighter-rouge">variants</code> (output of <strong>Call variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">VCF (only if input is VCF)</code></li>
    <li><em>‚ÄúCreate CSV report, useful for <span class="notranslate">downstream</span> analysis (-csvStats)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
    <li><em>‚ÄúAnnotation options‚Äù</em>: ``</li>
    <li><em>‚ÄúFilter output‚Äù</em>: ``</li>
    <li><em>‚ÄúFilter out specific Effects‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
  </ul>

</blockquote>

<p>The output of this step is a VCF file with added variant effects.</p>

<h2 id="create-table-of-variants-using-snpsift-extract-fields">Create table of variants using <strong>SnpSift Extract Fields</strong></h2>

<p>We will now select various effects from the VCF and create a tabular file that is easier to understand for humans.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-create-table-of-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Create table of variants</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.galaxy0"><strong>SnpSift Extract Fields</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/snpsift/snpSift_extractFields/4.3+t.<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúVariant input file in VCF format‚Äù</em>: <code class="language-plaintext highlighter-rouge">snpeff_output</code> (output of <strong>SnpEff eff:</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúFields to extract‚Äù</em>: <code class="language-plaintext highlighter-rouge">CHROM POS REF ALT QUAL DP AF SB DP4 EFF[*].IMPACT EFF[*].FUNCLASS EFF[*].EFFECT EFF[*].GENE EFF[*].CODON</code></li>
    <li><em>‚ÄúOne effect per line‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
    <li><em>‚Äúempty field text‚Äù</em>: <code class="language-plaintext highlighter-rouge">.</code></li>
  </ul>

</blockquote>

<p>We can inspect the output files and see check if Variants in this file are also described in <a href="https://observablehq.com/@spond/distribution-of-sars-cov-2-sequences-that-have-a-particular">an observable notebook that shows the geographic distribution of SARS-CoV-2 variant sequences</a></p>

<p>Interesting variants include the C to T variant at position 14408 (14408C/T) in SRR11772204, 28144T/C in SRR11597145 and 25563G/T in SRR11667145.</p>

<h2 id="summarize-data-with-multiqc">Summarize data with <strong>MultiQC</strong></h2>

<p>We will now summarize our analysis with MultiQC, which generates a beautiful report for our data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-summarize-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Summarize data</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:</p>
  <ul>
    <li>In <em>‚ÄúResults‚Äù</em>:
      <ul>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Results‚Äù</em>
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">fastp</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúOutput of fastp‚Äù</em>: <code class="language-plaintext highlighter-rouge">report_json</code> (output of <strong>fastp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Results‚Äù</em>
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Picard</code>
              <ul>
                <li>In <em>‚ÄúPicard output‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Picard output‚Äù</em>
                      <ul>
                        <li><em>‚ÄúType of Picard output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Markdups</code></li>
                        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPicard output‚Äù</em>: <code class="language-plaintext highlighter-rouge">metrics_file</code> (output of <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</blockquote>

<p>The above state allows us to judge the quality of the data. In this particular case data is not bad as quality values never drop below 30:</p>

<p><img src="../../images/multiqc.png" alt="MultiQC plot. " loading="lazy" /></p>

<h2 id="collapse-data-into-a-single-dataset">Collapse data into a single dataset</h2>

<p>We now extracted meaningful fields from VCF datasets. But they still exist as a collection. To move towards secondary analysis we need to <strong>collapse</strong> this collection into a single dataset. For more information about collapsing collections see this video:</p>

<iframe width="560" height="315" src="https://www.youtube.com/embed/ypuFZ1RKMIY" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-collapse-a-collection"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Collapse a collection</h3>

  <p>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.0" title="Tested with toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.0"><strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.0</span></span>  with the following parameters:</p>
  <ul>
    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúCollection of files to collapse into single dataset‚Äù</em>: <code class="language-plaintext highlighter-rouge">snpsift extract fields</code> (output of <strong>SnpSift Extract Fields</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li>‚Äú<em>Keep one header line</em>‚Äù: <code class="language-plaintext highlighter-rouge">Yes</code></li>
    <li>‚Äú<em>Prepend File name</em>‚Äù: <code class="language-plaintext highlighter-rouge">Yes</code></li>
    <li>‚Äú<em>Where to add dataset name</em>‚Äù: <code class="language-plaintext highlighter-rouge">Same line and each line in dataset</code></li>
  </ul>
</blockquote>

<p>You can see that this tool takes lines from all collection elements (in our case we have two), add element name as the first column, and pastes everything together. So if we have a collection as an input:</p>

<blockquote class="code-in">
  <h3 id="code-in-input-a-collection-with-two-items"><i class="far fa-keyboard" aria-hidden="true"></i><span class="visually-hidden">code-in</span> Input: A collection with two items</h3>
  <p>A collection element named <code class="language-plaintext highlighter-rouge">SRR11954102</code></p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>NC_045512.2  84 PASS  C T  960.0  28  1.0       0 0,0,15,13 MODIFIER  NONE  INTERGENIC
NC_045512.2 241 PASS  C T 2394.0  69  0.971014  0 0,0,39,29 MODIFIER  NONE  INTERGENIC
</code></pre></div>  </div>

  <p>A collection element named <code class="language-plaintext highlighter-rouge">SRR12733957</code>:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>NC_045512.2 241 PASS  C T 1954.0  63  0.888889  0 0,0,42,21 MODIFIER  NONE  INTERGENIC
NC_045512.2 823 PASS  C T 1199.0  50  0.76      3 5,6,13,26 LOW       LOW   LOW
</code></pre></div>  </div>
</blockquote>

<p>We will have a single dataset as the output:</p>

<blockquote class="code-out">
  <h3 id="code-out-output-a-single-dataset"><i class="fas fa-laptop-code" aria-hidden="true"></i><span class="visually-hidden">code-out</span> Output: A single dataset</h3>

  <p>then the <strong>Collapse Collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> will produce this:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR11954102 NC_045512.2  84 PASS  C T  960.0  28  1.0       0 0,0,15,13 MODIFIER  NONE  INTERGENIC
SRR11954102 NC_045512.2 241 PASS  C T 2394.0  69  0.971014  0 0,0,39,29 MODIFIER  NONE  INTERGENIC
SRR12733957 NC_045512.2 241 PASS  C T 1954.0  63  0.888889  0 0,0,42,21 MODIFIER  NONE  INTERGENIC
SRR12733957 NC_045512.2 823 PASS  C T 1199.0  50  0.76      3 5,6,13,26 LOW       LOW   LOW
</code></pre></div>  </div>
</blockquote>

<p>you can see that added a column with dataset ID taken from collection element name.</p>

<h2 id="anything-interesting">Anything interesting?</h2>

<p>These data are now ready for <span class="notranslate">downstream</span> analysis. One of the interesting things about these data is that it contains some of the mutations identified in the <a href="https://cov-lineages.org/global_report_B.1.1.7.html">B.1.1.7</a> lineage. For example, dataset <code class="language-plaintext highlighter-rouge">SRR12733957</code> contains a nonsense mutation (change for legitimate codon specifying an amino acid to a stop codon) in ORF8a:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR12733957 NC_045512.2 27972 PASS  C T 56.0  39  0.076923  0 7,29,0,3  HIGH  NONSENSE  STOP_GAINED ORF8  Caa/Taa
</code></pre></div></div>

<p>This is interesting because these datasets were collected well before B.1.1.7 became widely spread. Can you find more mutations here in these data?</p>

<!--

# Let's do it: A peek at secondary analysis in Galaxy

After a dataset is converted into an intermediary result (such as variant table generated during the previous step) by primary analysis, this intermediary data needs to be processed further to produce biological insight. IN Galaxy this can be done by starting a Jupyter notebook directly on history datasets. Here is a short demo.

> ### <i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Jupyter is in Beta
> Jupyter notebook functionality in Galaxy is currently in Beta. It means that we are still working on it and its implementation is constantly changing. Be aware of this fact.
{: .warning}

> ### <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Analyze data in Jupyter from Galaxy
>
> 1. Open <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> **Interactive Jupyter Notebook** from section **Interactive Tools**
> 1. *"Do you already have a notebook?"*: `Start with a fresh notebook`
> 1. *"Include data into the environment"*: Select history dataset containing the output of the collection collapse step
> 1. Click `Execute`
> 1. It will create a dataset in the history that will be permanently yellow (running)
> 1. You will see the following banner:
> <img src="../../images/jupyter_start.png" alt="Jupyter startup banner. "  loading="lazy">
> 1. Click on `User menu`. You may need to wait a bit (go get coffee quickly). In the end you will a `Jupyter Interactive Tool` link. Click on it.
> 1. This will open a new tab with a fully functional Jupyter environment:
> <img src="../../images/jupyter_running.png" alt="Jupyter running. "  loading="lazy">
> 1. Start a new notebook by clicking on Python 3 image. This will start a fresh notebook.
> 1. Type `!ls data/` in the first cell and you will see a file from Galaxy history.
> 1. Upgrade [Pandas](https://pandas.pydata.org/) and [Seaborn](https://seaborn.pydata.org/index.html): `!pip install -U pandas seaborn`
> 1. Read data into a dataframe:
>```
> import pandas as pd
> df = pd.read_csv('data/Collapsed_collection',sep='\t')
> # Drop duplicates is required to get rid of SnpEff artifacts
> df = df.drop_duplicates(ignore_index=True)
```
> <i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Note that your file may be named differently! Use it exact name instead of `Collapsed_collection` above.
> 1. Filter variants with intermediate frequencies between 20% and 80%:
>```
>df = df[ (df['AF']>=0.05) & (df['AF']<=0.8) ]
>```
> 1. Plot relationship between variant position and alternative allele frequency:
>```
>import seaborn as sns
>%matplotlib inline
>sns.relplot(x='POS',y='AF',hue='EFF[*].FUNCLASS',data=df,height=5,aspect=2,s=100,style='Sample')
>```
> This plot shows how variants are distributed across the SARS-CoV-2 genome:
> <img src="../../images/sns_plot.png" alt="Seaborn plot. "  loading="lazy">
> 1. Play with it!
{: .hands_on}


# Let's do it: A peek at secondary analysis in Google Colab

We can also import data directly into Google Colab:

> ### <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Analyze data in Google Colab
>
> 1. Expand history dataset containing the output of the collection collapse step
> 1. Right click on <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span> (disk) icon and copy link to clipboard
> 1. Go to https://colab.research.google.com/ and create a new notebook
> 1. Type `!wget <link>` where you must replace `<link>` with the URL you copied into clipboard at the previous step
> 1. This will copy this dataset into your Colaboratory environment
> 1. See the dataset file by typing `!ls` in the Colab cell
> 1. You will see a file with a name that looks something like this: `'display?to_ext=tabular'`
> 1. Rename it into something palatable:
>```
>!mv 'display?to_ext=tabular' data.tsv`
>```
> 1. Read data into a [Pandas](https://pandas.pydata.org/) dataframe:
> ```
> import pandas as pd
> df = pd.read_csv('data.tsv',sep='\t')
> # Drop duplicates is required to get rid of SnpEff artifacts
> df = df.drop_duplicates(ignore_index=True)
>```
> 1. Filter variants with intermediate frequencies between 20% and 80%:
>```
>df = df[ (df['AF']>=0.05) & (df['AF']<=0.8) ]
>```
> 1. Plot relationship between variant position and alternative allele frequency:
>```
>import seaborn as sns
>sns.relplot(x='POS',y='AF',hue='EFF[*].FUNCLASS',data=df,height=5,aspect=2,s=100,style='Sample')
>```
> This plot shows how variants are distributed across the SARS-CoV-2 genome:
> <img src="../../images/sns_plot.png" alt="Seaborn plot. "  loading="lazy">
> 1. Play with it!
{: .hands_on}

-->

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Congratulations, you now know how to import sequence data from the SRA and how to run an example analysis on these datasets.</p>


                
                <blockquote class="key_points">
                    <h3><i class="fas fa-key" aria-hidden="true"></i> Key points</h3>
                    <ul>
                        
                        <li><p>FASTQ Sanger version of the format is considered to be the standard form of FASTQ.</p>
</li>
                        
                        <li><p>Paired end data can be provided as two files or as an interleaved one.</p>
</li>
                        
                        <li><p>FastqQC is a tool allowing to check the quality of FASTQ datasets.</p>
</li>
                        
                        <li><p>The most common tools for mapping are Bowtie, BWA, BWA-MEM. You can use in-built genome to map against or upload one if it is missing.</p>
</li>
                        
                        <li><p>The standard format for storing aligned reads is SAM/BAM. The major toolsets to process these datasets are DeepTools, SAMtools, BAMtools and Picard.</p>
</li>
                        
                        <li><p>Data can be uploaded directly from a computer, from EBI SRA and from NCBI SRA, also using FTP or URL.</p>
</li>
                        
                        <li><p>One can retrieve NGS data from Sequence Read Archive</p>
</li>
                        
                        <li><p>Galaxy can analyze massive amounts of data and make them suitable for secondary analysis</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/introduction/faqs/">FAQ page for the Introduction to Galaxy Analyses topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                <div id="feedback-button">
                    <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                </div>
                <div id="feedback-form">
                </div>
                <script type="text/javascript">
                    (function (window, document) {
                        function onDocumentReady(fn) {
                            if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                fn();
                            } else {
                                document.addEventListener('DOMContentLoaded', fn);
                            }
                        }

                        onDocumentReady(function () {
                            $("#feedback-button").click(function(evt){
                                var e = $(evt.target)
                                e.hide();

                                $("#feedback-form").html(`
                                    <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=NGS data logistics (Introduction to Galaxy Analyses)">Loading...</iframe>
                                `)
                            })
                        });
                    })(window, document);
                </script>



                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">Anton Nekrutenko, Marius van den Beek, Dave Clements, Daniel Blankenberg, 2021 <b>NGS data logistics (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html">https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>


                <blockquote class="details">
                  <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                  <p style="display: none;">

                <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{introduction-galaxy-intro-ngs-data-managment,
author = "Anton Nekrutenko and Marius van den Beek and Dave Clements and Daniel Blankenberg",
title = "NGS data logistics (Galaxy Training Materials)",
year = "2021",
month = "03",
day = "23"
url = "\url{https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-ngs-data-managment/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
    doi = {10.1016/j.cels.2018.05.012},
    url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
    year = 2018,
    month = {jun},
    publisher = {Elsevier {BV}},
    volume = {6},
    number = {6},
    pages = {752--758.e1},
    author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
    title = {Community-Driven Data Analysis Training for Biology},
    journal = {Cell Systems}
}</code>
                </pre></div></div>
                </p>
                </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <h3><i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

                

                

            </div>
        </div>
    </div>
</section>
<br/>
<br/>
<br/>

        </div>
        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
