<!doctype html>
<html lang="en">

    <head>
        <meta charset="utf-8">

        <title>Options for using Galaxy</title>

        <link rel="stylesheet" href="../../shared/reveal.js/css/reveal.css">
        <link rel="stylesheet" href="../../shared/reveal.js/css/theme/night.css" id="theme">

        <!-- Code syntax highlighting -->
        <link rel="stylesheet" href="../../shared/reveal.js/lib/css/zenburn.css">
        <link rel="stylesheet" href="../../shared/font-awesome/css/font-awesome.min.css">
        <link rel="stylesheet" href="../../shared/css/custom.css" id="theme">

        <!-- Printing and PDF exports -->
        <script>
          if( window.location.search.match( /print-pdf/gi ) ) {
            var link = document.createElement( 'link' );
            link.rel = 'stylesheet';
            link.type = 'text/css';
            link.href = '../../shared/reveal.js/css/print/pdf.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
          }
        </script>
        <!--<script>
            if (window.location.search.match(/print-pdf/gi)) {
                document.write('<link rel="stylesheet" href="css/pdf.css" type="text/css">');
            }
        </script>-->

        <!--[if lt IE 9]>
        <script src="../../shared/reveal.js/lib/js/html5shiv.js"></script>
        <![endif]-->
    </head>

    <body>
        <div class="reveal">
            <div class="slides">
                <section data-markdown>
                    <script type="text/template">
                        ## Options for using Galaxy

                        <img src="../../shared/images/GTNLogo1000.png" alt="GTNLogo1000" width=30%/>

                        The easiest way to **navigate** this slide deck
                        is **by hitting `[space]` on your keyboard**

                        You can also navigate with arrow keys, but be careful because some
                        slides can be nested inside of each other (vertically)

                        <aside class="notes">
                            Speaker notes:
                            
                            These slides are from/based on the [Galaxy Training Network's collection of slide decks](http://github.com/galaxy-project/training-material).
                        </aside>
                    </script>
                </section>

                <section data-markdown>
                    <script type="text/template">
                        ## Options for using Galaxy

                        Galaxy is available in many ways, each with different configurations and capabilities.

                        Here are some options.

                        <aside class="notes">
                            Many workshops use Galaxy instances that go away once the workshop is done.
                            This deck lists options for using Galaaxy outside of a training context.
                        </aside>
                    </script>
                </section>

                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## [usegalaxy.org](https://usegalaxy.org/)

                            - [usegalaxy.org](https://usegalaxy.org/) is the Galaxy Project's public server
                            - Hosted at the [Texas Advanced Computing Center (TACC)](https://www.tacc.utexas.edu/)
                            - Lots of
                              - tools
                              - reference genomes
                              - compute power
                              - storage

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ## [usegalaxy.org](https://usegalaxy.org/)

                            But also very busy<br />
                            Averages 250,000 jobs *per month* in 2016

                            A single resource cannot scale to meet the demand of the entire world, nor can it support all wanted tools or genomes or ...
                            <aside class="notes">
                            </aside>
                        </script>
                    </section>
                </section>

                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## Public Galaxy Servers

                            [<img src="../images/public_server_galaxeast.png" alt="GalaxyEast" />](https://wiki.galaxyproject.org/PublicGalaxyServers#GalaxyEast)
                            [<img src="../images/public_servers_odometer_count.png" alt="See the list of publicly accessible Galaxy servers" />](https://wiki.galaxyproject.org/PublicGalaxyServers)
                            [<img src="../images/public_server_gigagalaxy.png" alt="GigaGalaxy" />](https://wiki.galaxyproject.org/PublicGalaxyServers#GigaGalaxy)
                            [<img src="../images/public_server_genomichyperbrowser.png" alt="Genomic Hyperbowser" />](http://hyperbrowser.uio.no/hb/)
                            [<img src="../images/public_server_lappsgrid.png" alt="LAPPS Grid" />](http://galaxy.lappsgrid.org/)


                            Directory of public Galaxy servers:  [bit.ly/gxyServers](http://bit.ly/gxyServers)<br />

                            <aside class="notes">
                            </aside>
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ## Public Galaxy Servers

                            Classified as:

                            1. General purpose genomics servers
                            1. Domain specific servers.
                              - These can be specific to 
                                - *Research methods:* ChIP-Seq, RNA-Seq, repeats, ...
                                - *Organisms:* Pathogens, phage, rice, poplars (!), ...
                                - *Non-genomic domains:* image analysis, social science, natural language processing, ...
                            1. Tool servers
                              - Host specific tools, usually from one lab, to make the tools easy to access and run

                            [bit.ly/gxyServers](http://bit.ly/gxyServers)
                            <aside class="notes">
                            </aside>
                        </script>
                    </section>
                </section>

                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## Semi-public Galaxy services

                            Access based on geography or membership in a community

                            [bit.ly/gxysemipublic](http://bit.ly/gxysemipublic)
                            <aside class="notes">
                            </aside>
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ## Semi-public Galaxy services
                            | Geography  | Resource |
                            | ---------- | -------- |
                            | Australia  | [Genomics Virtual Lab (GVL)](https://launch.genome.edu.au/launch)  |
                            | Canada | [GenAP](https://www.genap.ca/) |
                            | Norway | [Norwegian e-Infrastructure for Life Sciences (NeLS)](https://nels.bioinfo.no/) |
                            | Poland | [PL-Grid](https://galaxy.plgrid.pl/) |
                            | US | [Jetstream](https://wiki.galaxyproject.org/Cloud/Jetstream) |

                            [bit.ly/gxysemipublic](http://bit.ly/gxysemipublic)
                            <aside class="notes">
                            </aside>
                        </script>
                    </section>
                </section>

                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ### On public clouds via CloudLaunch

                            [<img src="../images/cloudlaunch_empty.png" alt="CloudLaunch" width="60%" />](https://launch.usegalaxy.org/)

                            Currently supports Amazon Web Services 

                            [launch.usegalaxy.org](https://launch.usegalaxy.org/)
                            <aside class="notes">
                                CloudLaunch currently only supports AWS (and Jetstream, but that's covered above).
                                To use it with AWS you need to create an AWS account, which does require a
                                credit card, and then provide your public and secret keys to CloudLaunch.
                            </aside>
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### CloudLaunch / CloudMan instances

                            [<img src="../../shared/images/cloudman_based_server_landing_page.png" alt="Galaxy server launched from CloudMan" width="80%" />](http://wiki.galaxyproject.org/CloudMan)

                            Comes with 100's of tools & many pre-defined reference genomes.

                            <aside class="notes">
                                CloudLaunch starts a CloudMan based Galaxy instance.
                                CloudMan is a cloud management tool that moves much / all of the cloud management work into a Graphical user interface.
                            </aside>
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### CloudMan

                            [<img src="../../shared/images/cloudman_management_console_showing_autoscaled_w_1_worker.png" alt="CloudMan enables you to autoscale the size of your server, dependding on load" width="50%" />](http://wiki.galaxyproject.org/CloudMan)

                            CloudMan enables you to statically or dynamically scale compute power.

                            <aside class="notes">
                                - This example shows a Galaxy Server with a head node and one worker node, and that is configured to scale up to 5 worker nodes, if demand justifies it.
                                - Before leaving this cloud section stress the importance of shutting down and removing everything when you are done.
                                - Cloud instances can be incredibly cost effective, but only if you are actually using them when they are up.
                                - Paying to have a cloud instance up for two weeks that you only use for 8 hours is not cost effective.
                            </aside>
                        </script>
                    </section>
                </section>

                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ### Docker (and maybe other appliances?)

                            Stuff about docker...

                            <aside class="notes">
                            </aside>
                        </script>
                    </section>
                </section>


                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ### Run your own Galaxy locally

                            - Galaxy is [open source software](http://getgalaxy.org/) and can be installed on local compute infrastructure, from lab servers to insitutional compute clusters.
                            - Installing Galaxy locally is relatively easy, but
                              - the initial install does not include reference genomes and only has a few tools
                              - Installing tools and genomes, setting up authentication, and connecting to institutional compute resources all takes work.
                            - There are hundreds, if not thousands of local Galaxy installs around the world.
                            <aside class="notes">
                                - Installing tools and genomes has become much easier in recent years, and can now often be done with the Galaxy Admin GUI.
                                - Authentication and connecting to institutional compute resources is still heavy lifting.
                            </aside>
                        </script>
                    </section>
                </section>

                <section data-markdown>
                    <script type="text/template">
                        ### Commercial support

                        - You can buy a [preconfigured Galaxy server](http://www.bioteam.net/products/galaxy-appliance/) from [BioTeam](http://www.bioteam.net/)
                        - [Globus Genomics](https://www.globus.org/genomics) and [GenomeCloud](http://www.genome-cloud.com/) provide cloud-based Galaxy servers.
                        - (And [commercially provided consulting and training](https://wiki.galaxyproject.org/Support/Commercial) are also available) 

                    </script>
                </section>
            </div>

        </div>

        <script type="text/javascript" src="../../shared/reveal.js/lib/js/head.min.js"></script>
        <script type="text/javascript" src="../../shared/reveal.js/js/reveal.js"></script>
        <script type="text/javascript">
            Reveal.initialize({
                slideNumber: !window.location.search.match(/print-pdf/gi),
                history: true,

                theme: Reveal.getQueryHash().theme,
                transition: Reveal.getQueryHash().transition || 'concave',

                dependencies: [
                    { src: '../../shared/reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
                    { src: '../../shared/reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
                    { src: '../../shared/reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
                    { src: '../../shared/js/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                    { src: '../../shared/reveal.js/plugin/notes/notes.js', async: true },
                ]
            });
        </script>

        </body>
</html>
