<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Clean and manage Sanger sequences from raw files to aligned consensus / Clean and manage Sanger sequences from raw files to aligned consensus / Sequence analysis</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@82479b8220ed6ad45f0170c036bce307d7f50bd3",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Clean and manage Sanger sequences from raw files to aligned consensus">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-03-05 13:33:11 +0000">
<meta name="DC.creator" content="Coline Royaux"><meta name="description" content="Analyses of sequences">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Sequence analysis / Clean and manage Sanger sequences from raw files to aligned consensus / Hands-on: Clean and manage Sanger sequences from raw files to aligned consensus">
        <meta property="og:description" content="Analyses of sequences">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Clean and manage Sanger sequences from raw files to aligned consensus",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-03-05 13:33:11 +0000",
  "datePublished": "2024-01-08 11:21:03 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00383",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Sequence analysis",
    "description": "Analyses of sequences",
    "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/"
  },
  "abstract": "The objective of this tutorial is to learn how to clean and manage AB1 data files freshly obtained from Sanger sequencing. ",
  "learningResourceType": "e-learning",
  "name": "Clean and manage Sanger sequences from raw files to aligned consensus",
  "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html",
  "version": 2,
  "timeRequired": "PT1H",
  "teaches": "- Learn how to manage sequencing files (AB1, FASTQ, FASTA)\n- Learn how to clean your Sanger sequences in an automated and reproducible way",
  "keywords": [
    "Sequence analysis"
  ],
  "description": "## Abstract\n\nThe objective of this tutorial is to learn how to clean and manage AB1 data files freshly obtained from Sanger sequencing. \n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - How to clean Sanger sequencing files?\n\n\n## Learning Objectives\n\n- Learn how to manage sequencing files (AB1, FASTQ, FASTA)\n- Learn how to clean your Sanger sequences in an automated and reproducible way\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/colineroyaux/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/colineroyaux/",
      "name": "Coline Royaux",
      "image": "https://avatars.githubusercontent.com/colineroyaux",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/pndb/",
          "name": "Pôle National de Données de Biodiversité",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.pndb.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-4308-5617",
      "orcid": "https://orcid.org/0000-0003-4308-5617"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Sequence analysis",
      "description": "Analyses of sequences",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0080",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0080",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0080"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/records/7104640",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Sequence analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/sequence-analysis/faqs/" title="Check our FAQs for the Sequence analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-sequence-analysis">
    <h1 data-toc-skip>Clean and manage Sanger sequences from raw files to aligned consensus</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/colineroyaux/" class="contributor-badge contributor-colineroyaux"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/colineroyaux?s=36" alt="Coline Royaux avatar" width="36" class="avatar" />
    Coline Royaux</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How to clean Sanger sequencing files?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn how to manage sequencing files (AB1, FASTQ, FASTA)</p>
</li>
        
        <li><p>Learn how to clean your Sanger sequences in an automated and reproducible way</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/records/7104640">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/workflows/" title="Clean and manage Sanger sequences from raw files to aligned consensus workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Jan 8, 2024 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 5, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00383">https://gxy.io/GTN:T00383</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">2.0</a> (2 recent ratings, 2 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 2 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>The objective of this tutorial is to learn how to clean and manage AB1 data files freshly obtained from Sanger sequencing. 
This kind of sequencing is targeting a specific sequence with short single DNA strands called primers. These primers are delimiting ends of the targeted marker.
Usually, one gets two .ab1 files for each sample, representing the sense (forward) and the antisense (reverse) strands of DNA.</p>

<p>Here, we’ll be using raw data from <span class="citation"><a href="#Garavaglia2022">“<b>AOPEP variants as a novel cause of recessive dystonia: Generalized dystonia and dystonia-parkinsonism</b>” 2022</a></span>.
In this article, two DNA markers are investiguated CHD8 (Chromodomain-helicase-DNA-binding protein 8) and AOPEP (Aminopeptidase O Putative).
We’ll focus on CHD8 sequences but you can try to apply the same steps on the AOPEP sequences to practice after the tutorial !</p>

<p>In the first section of the tutorial, we’ll be preparing primer’s data by:</p>
<ul>
  <li>selecting the right primer sequences with the identifier;</li>
  <li>removing eventual gaps included in the sequences;</li>
  <li>and compute the reverse-complement sequence for the antisense primer only.</li>
</ul>

<p>In the second section of the tutorial, we’ll be preparing the Sanger sequences data by:</p>
<ul>
  <li>extracting ab1 files of the interest sequence (CHD8) and separating sense and antisense sequences in two distinct data collections;</li>
  <li>converting ab1 files to FASTQ to permit its use in the following tools;</li>
  <li>trimming low quality ends of the sequences;</li>
  <li>compute the reverse-complement for the antisense sequence only;</li>
  <li>align sense and antisense sequences;</li>
  <li>obtain a consensus sequence (which results the correspondance between nucleotides of the sense and the antisense sequences) for each three samples.</li>
</ul>

<p>In the third section of the tutorial, primers and all consensus sequences are finally merged into a single file to be aligned and verified.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-is-the-reverse-complement-sequence-and-why-computing-it-for-the-antisense-sequence"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-the-reverse-complement-sequence-and-why-computing-it-for-the-antisense-sequence" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What is the reverse-complement sequence and why computing it for the antisense sequence?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Consider a double-strand DNA molecule with the following sequences:</p>
  <figure id="figure-1" style="max-width: 90%;"><img src="../../images/Sanger_tuto/Double-strand_DNA.png" alt="Double-strand DNA. " width="1186" height="198" loading="lazy" /><a target="_blank" href="../../images/Sanger_tuto/Double-strand_DNA.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Double-strand DNA</figcaption></figure>

  <p>When sequencing, each strand of DNA are read separately in the 5’-3’ orientation. Hence, in the sequence files each strand are provided as:</p>
  <figure id="figure-2" style="max-width: 90%;"><img src="../../images/Sanger_tuto/two_single-strand_sequences.png" alt="Single-strand DNA sequences in output file. " width="1171" height="193" loading="lazy" /><a target="_blank" href="../../images/Sanger_tuto/two_single-strand_sequences.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Single-strand DNA sequences in output file</figcaption></figure>

  <p>To get the antisense sequence in its original orientation, the reverse sequence is computed:</p>
  <figure id="figure-3" style="max-width: 90%;"><img src="../../images/Sanger_tuto/Reversed-antisense-sequence.png" alt="Reversed antisense sequence. " width="1776" height="64" loading="lazy" /><a target="_blank" href="../../images/Sanger_tuto/Reversed-antisense-sequence.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Reversed antisense sequence</figcaption></figure>

  <p>To align sense and antisense sequence, the complement sequence of the reversed antisense sequence is computed:</p>
  <figure id="figure-4" style="max-width: 90%;"><img src="../../images/Sanger_tuto/Reversed-antisense-sequence.png" alt="Reverse-complement antisense sequence. " width="1776" height="64" loading="lazy" /><a target="_blank" href="../../images/Sanger_tuto/Reversed-antisense-sequence.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Reversed antisense sequence</figcaption></figure>

  <p>The two sequences can be aligned now:</p>
  <figure id="figure-5" style="max-width: 90%;"><img src="../../images/Sanger_tuto/Aligned_sense_antisense.png" alt="Aligned sense and antisense sequences. " width="1836" height="178" loading="lazy" /><a target="_blank" href="../../images/Sanger_tuto/Aligned_sense_antisense.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Aligned sense and antisense sequences</figcaption></figure>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#prepare-primer-data" id="markdown-toc-prepare-primer-data">Prepare primer data</a>    <ol>
      <li><a href="#separate-and-format-primers-files" id="markdown-toc-separate-and-format-primers-files">Separate and format primers files</a></li>
    </ol>
  </li>
  <li><a href="#prepare-sequence-data" id="markdown-toc-prepare-sequence-data">Prepare sequence data</a>    <ol>
      <li><a href="#unzip-data-files" id="markdown-toc-unzip-data-files">Unzip data files</a></li>
      <li><a href="#filter-collection-to-separate-sense-and-antisense-sequence-files" id="markdown-toc-filter-collection-to-separate-sense-and-antisense-sequence-files">Filter collection to separate sense and antisense sequence files</a></li>
      <li><a href="#convert-ab1-sequence-files-to-fastq-and-trim-low-quality-ends" id="markdown-toc-convert-ab1-sequence-files-to-fastq-and-trim-low-quality-ends">Convert AB1 sequence files to FASTQ and trim low-quality ends</a></li>
      <li><a href="#compute-reverse-complement-sequence-for-antisense-reverse-sequences-only" id="markdown-toc-compute-reverse-complement-sequence-for-antisense-reverse-sequences-only">Compute reverse complement sequence for antisense (reverse) sequences only</a></li>
      <li><a href="#merge-corresponding-sense-and-antisense-sequences-single-files" id="markdown-toc-merge-corresponding-sense-and-antisense-sequences-single-files">Merge corresponding sense and antisense sequences single files</a></li>
      <li><a href="#convert-fastq-files-to-fasta" id="markdown-toc-convert-fastq-files-to-fasta">Convert FASTQ files to FASTA</a></li>
      <li><a href="#align-sequences-and-retrieve-consensus-for-each-sequence" id="markdown-toc-align-sequences-and-retrieve-consensus-for-each-sequence">Align sequences and retrieve consensus for each sequence</a></li>
    </ol>
  </li>
  <li><a href="#manage-primers-and-sequences" id="markdown-toc-manage-primers-and-sequences">Manage primers and sequences</a>    <ol>
      <li><a href="#merge-and-align-consensus-sequences-file-and-primer-files" id="markdown-toc-merge-and-align-consensus-sequences-file-and-primer-files">Merge and align consensus sequences file and primer files</a></li>
      <li><a href="#check-your-sequences-belongs-to-the-right-taxonomic-group-by-computing-a-blast-on-the-ncbi-database" id="markdown-toc-check-your-sequences-belongs-to-the-right-taxonomic-group-by-computing-a-blast-on-the-ncbi-database">Check your sequences belongs to the right taxonomic group by computing a BLAST on the NCBI database</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
  <li><a href="#aopep-sanger-files" id="markdown-toc-aopep-sanger-files">AOPEP Sanger files</a></li>
</ol>

</blockquote>

<h1 id="get-data">Get data</h1>

<p>Authors of <span class="citation"><a href="#Garavaglia2022">“<b>AOPEP variants as a novel cause of recessive dystonia: Generalized dystonia and dystonia-parkinsonism</b>” 2022</a></span> have shared openly their raw AB1 files on Zenodo.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Data Upload</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from <a href="https://zenodo.org/records/7104640">Zenodo</a> :</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/7104640/files/AOPEP_and_CHD8_sequences_20220907.zip
</code></pre></div>      </div>

      <p>Change <strong>Type (set all):</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">zip</code> and click <strong>Start</strong></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>Create primer FASTA file, copy:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>&gt;Forward_CHD8
GAGGTGAAAGAATCATAAATTGG
&gt;Reverse_CHD8
CCCTGTGTACAAATAGCTTTTGT
&gt;Forward_AOPEP
TCATGGTTCCAGGCAGAGTTATT
&gt;Reverse_AOPEP
TGCTGTGACAAGCCAACCAATGG
</code></pre></div>      </div>
      <ul>
        <li>Open the Galaxy Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</li>
        <li>Select <strong>Paste/Fetch Data</strong></li>
        <li>Paste into the text field</li>
        <li>Change <strong>Type (set all):</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">fasta</code></li>
        <li>Change the name from “New File” to “Primer file”</li>
        <li>Click <strong>Start</strong></li>
      </ul>

      <p>Note these primer sequences were invented for the purpose of the tutorial, it is not the sequences used in the publication.</p>
    </li>
  </ol>

</blockquote>

<h1 id="prepare-primer-data">Prepare primer data</h1>

<h2 id="separate-and-format-primers-files">Separate and format primers files</h2>

<p>Primers must be separated in distinct files because sense (forward) and antisense (reverse) primers won’t be subjected to the same formating.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-separate-files-for-each-primer"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Create separate files for each primer</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/filter_by_fasta_ids/filter_by_fasta_ids/2.3" title="Filter FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA sequences”</em>: <code class="language-plaintext highlighter-rouge">Primer file</code></li>
        <li><em>“Criteria for filtering on the headers”</em>: <code class="language-plaintext highlighter-rouge">Regular expression on the headers</code>
          <ul>
            <li><em>“Regular expression pattern the header should match”</em>: <code class="language-plaintext highlighter-rouge">Reverse_CHD8</code></li>
          </ul>
        </li>
        <li>Add tags “#Primer” and “#Reverse”</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Re-running a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/filter_by_fasta_ids/filter_by_fasta_ids/2.3" title="Filter FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA sequences”</em>: <code class="language-plaintext highlighter-rouge">Primer file</code></li>
        <li><em>“Criteria for filtering on the headers”</em>: <code class="language-plaintext highlighter-rouge">Regular expression on the headers</code>
          <ul>
            <li><em>“Regular expression pattern the header should match”</em>: <code class="language-plaintext highlighter-rouge">Forward_CHD8</code></li>
          </ul>
        </li>
        <li>Add tags “#Primer” and “#Forward”</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Remove eventual gaps from primers <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/mothur_degap_seqs/mothur_degap_seqs/1.39.5.0" title="Degap.seqs tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Degap.seqs</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.39.5.0)</span> with the following parameters:</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“fasta - Dataset”</em>: <code class="language-plaintext highlighter-rouge">Two Filter FASTA outputs</code> (outputs of <strong>Filter FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>In this previous hands-on, the step of removing eventual gaps (<code class="language-plaintext highlighter-rouge">-</code> in the FASTA files) is a precaution, there are no gaps in our primers file. However, it is important to remove gaps at this point in case you are using different data, otherwise some steps of the tutorial could fail (e.g. alignment).</p>

<p>This following hands-on is to be applied only on the sequence of the antisense (reverse) primer.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-compute-reverse-complement-of-the-antisense-reverse-primer"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Compute Reverse-Complement of the antisense (reverse) primer</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastx_reverse_complement/cshl_fastx_reverse_complement/1.0.2+galaxy0" title="Reverse-Complement tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Reverse-Complement</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.2+galaxy0)</span> the sequence antisense (reverse) primer with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file in FASTA or FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">Degap.seqs #Reverse FASTA output</code> (output of <strong>Degap.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
  </ol>

  <p>See in the introduction for explanations on the Reverse-Complement.</p>

</blockquote>

<h1 id="prepare-sequence-data">Prepare sequence data</h1>

<h2 id="unzip-data-files">Unzip data files</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-unzip"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Unzip</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/imgteam/unzip/unzip/6.0+galaxy0" title="Unzip tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Unzip</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 6.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“input_file”</em>: <code class="language-plaintext highlighter-rouge">AOPEP_and_CHD8_sequences_20220907.zip?download=1</code></li>
        <li><em>“Extract single file”</em>: <code class="language-plaintext highlighter-rouge">All files</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <p>How many files is there in the ZIP archive ?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>12 (if you have a different number of files something likely went wrong)</p>

    </blockquote>

  </blockquote>

</blockquote>

<p>From now on, we’ll be working a lot on data collections:</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-selecting-a-dataset-collection-as-input"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-selecting-a-dataset-collection-as-input" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Selecting a dataset collection as input</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in  front of the input parameter you want to supply the collection to.</li>   <li>Select the collection you want to use from the list</li> </ol> </blockquote>
<p><!--END_SNIPPET--></p>

<h2 id="filter-collection-to-separate-sense-and-antisense-sequence-files">Filter collection to separate sense and antisense sequence files</h2>

<p>As for primers, sense and antisense sequences will be subjected to slightly different procedures so they must be separated in distinct data collections.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Filter</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_element_identifiers/collection_element_identifiers/0.0.2" title="Extract element identifiers tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Extract element identifiers</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">output collection</code> (output of <strong>Unzip</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3" title="Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>Extract element identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Check”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">^[A-Za-z0-9_-]+F$</code></li>
                <li><em>“Replacement”</em>: ``</li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">^[A-Za-z0-9_-]+AOPEP[A-Za-z0-9_-]+$</code></li>
                <li><em>“Replacement”</em>: ``</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Tag output with “#Reverse”</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Re-running a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3" title="Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>Extract element identifiers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Check”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">^[A-Za-z0-9_-]+R$</code></li>
                <li><em>“Replacement”</em>: ``</li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">^[A-Za-z0-9_-]+AOPEP[A-Za-z0-9_-]+$</code></li>
                <li><em>“Replacement”</em>: ``</li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Tag output with “#Forward”</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><span class="tool" data-tool="__FILTER_FROM_FILE__" title="Filter collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter collection</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Collection</em>: <code class="language-plaintext highlighter-rouge">output collection</code> (output of <strong>Unzip</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“How should the elements to remove be determined?”</em>: <code class="language-plaintext highlighter-rouge">Remove if identifiers are ABSENT from file</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Filter out identifiers absent from”</em>: <code class="language-plaintext highlighter-rouge">#Forward files list</code> &amp; <code class="language-plaintext highlighter-rouge">#Reverse files list</code> (output of <strong>Regex Find And Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li>Tag <code class="language-plaintext highlighter-rouge">(filtered)</code> outputs with “#Forward” and “#Reverse”</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-4"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-4" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-what-s-happening-in-this-section"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  What's happening in this section?</div>

        <p>First step: Extracting the list of file names in the data collection
Second step: Removing file names containing a “F” and “AOPEP” -&gt; creating a list of antisense (reverse) sequence files of the marker CHD8
Third step: Removing file names containing a “R” and “AOPEP” -&gt; creating a list of sense (forward) sequence files of the marker CHD8
Fourth step: Select files in the collection -&gt; creating two distinct collections with sense (forward) sequence files on one hand and antisense (reverse) sequence file on the other hand</p>

        <p>For the second and third step, we used regular expressions (Regex):</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-regular-expressions-101"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-regular-expressions-101" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Regular Expressions 101</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Regular expressions are a standardized way of describing patterns in textual data. They can be extremely useful for tasks such as finding and replacing data. They can be a bit tricky to master, but learning even just a few of the basics can help you get the most out of Galaxy.</p>   <p><strong>Finding</strong></p>   <p>Below are just a few examples of basic expressions:</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">abc</code></td>       <td>an occurrence of <code class="language-plaintext highlighter-rouge">abc</code> within your data</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(abc|def)</code></td>       <td><code class="language-plaintext highlighter-rouge">abc</code> <em>or</em> <code class="language-plaintext highlighter-rouge">def</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[abc]</code></td>       <td>a single character which is either <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, or <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[^abc]</code></td>       <td>a character that is NOT <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, nor <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-z]</code></td>       <td>any lowercase letter</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-zA-Z]</code></td>       <td>any letter (upper or lower case)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[0-9]</code></td>       <td>numbers 0-9</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d</code></td>       <td>any digit (same as <code class="language-plaintext highlighter-rouge">[0-9]</code>)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\D</code></td>       <td>any non-digit character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\w</code></td>       <td>any alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\W</code></td>       <td>any non-alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\s</code></td>       <td>any whitespace</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\S</code></td>       <td>any non-whitespace character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.</code></td>       <td><em>any</em> character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\.</code></td>       <td> </td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">{x,y}</code></td>       <td>between <em>x</em> and <em>y</em> repetitions</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^</code></td>       <td>the beginning of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">$</code></td>       <td>the end of the line</td>     </tr>   </tbody> </table>   <p><strong>Note:</strong> you see that characters such as <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">?</code>, <code class="language-plaintext highlighter-rouge">.</code>, <code class="language-plaintext highlighter-rouge">+</code> etc have a special meaning in a regular expression. If you want to match on those characters, you can escape them with a backslash. So <code class="language-plaintext highlighter-rouge">\?</code> matches the question mark character exactly.</p>   <p><strong>Examples</strong></p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d{4}</code></td>       <td>4 digits (e.g. a year)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr\d{1,2}</code></td>       <td><code class="language-plaintext highlighter-rouge">chr</code> followed by 1 or 2 digits</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.*abc$</code></td>       <td>anything with <code class="language-plaintext highlighter-rouge">abc</code> at the end of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^$</code></td>       <td>empty line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^&gt;.*</code></td>       <td>Line starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta header)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^[^&gt;].*</code></td>       <td>Line not starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta sequence)</td>     </tr>   </tbody> </table>   <p><strong>Replacing</strong></p>   <p>Sometimes you need to capture the exact value you matched on, in order to use it in your replacement, we do this using capture groups <code class="language-plaintext highlighter-rouge">(...)</code>, which we can refer to using <code class="language-plaintext highlighter-rouge">\1</code>, <code class="language-plaintext highlighter-rouge">\2</code> etc for the first and second captured values. If you want to refer to the whole match, use <code class="language-plaintext highlighter-rouge">&amp;</code>.</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Input</th>       <th>Captures</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr(\d{1,2})</code></td>       <td><code class="language-plaintext highlighter-rouge">chr14</code></td>       <td><code class="language-plaintext highlighter-rouge">\1 = 14</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(\d{2}) July (\d{4})</code></td>       <td>24 July 1984</td>       <td><code class="language-plaintext highlighter-rouge">\1 = 24</code>, <code class="language-plaintext highlighter-rouge">\2 = 1984</code></td>     </tr>   </tbody> </table>   <p>An expression like <code class="language-plaintext highlighter-rouge">s/find/replacement/g</code> indicates a replacement expression, this will search (<code class="language-plaintext highlighter-rouge">s</code>) for any occurrence of <code class="language-plaintext highlighter-rouge">find</code>, and replace it with <code class="language-plaintext highlighter-rouge">replacement</code>. It will do this globally (<code class="language-plaintext highlighter-rouge">g</code>) which means it doesn’t stop after the first match.</p>   <p>Example: <code class="language-plaintext highlighter-rouge">s/chr(\d{1,2})/CHR\1/g</code> will replace <code class="language-plaintext highlighter-rouge">chr14</code> with <code class="language-plaintext highlighter-rouge">CHR14</code> etc.</p>   <p>You can also use replacement modifier such as convert to lower case <code class="language-plaintext highlighter-rouge">\L</code> or upper case <code class="language-plaintext highlighter-rouge">\U</code>. Example: <code class="language-plaintext highlighter-rouge">s/.*/\U&amp;/g</code> will convert the whole text to upper case.</p>   <p>Note: In Galaxy, you are often asked to provide the find and replacement expressions separately, so you don’t have to use the <code class="language-plaintext highlighter-rouge">s/../../g</code> structure.</p>   <p>There is a lot more you can do with regular expressions, and there are a few different flavours in different tools/programming languages, but these are the most important basics that will already allow you to do many of the tasks you might need in your analysis.</p>   <p><strong>Tip:</strong> <a href="https://regexone.com/">RegexOne</a> is a nice interactive tutorial to learn the basics of regular expressions.</p>   <p><strong>Tip:</strong> <a href="https://regex101.com/">Regex101.com</a> is a great resource for interactively testing and constructing your regular expressions, it even provides an explanation of a regular expression if you provide one.</p>   <p><strong>Tip:</strong> <a href="https://extendsclass.com/regex-tester.html">Cyrilex</a> is a visual regular expression tester.</p> </blockquote>
        <p><!--END_SNIPPET--></p>

        <p>With <code class="language-plaintext highlighter-rouge">[A-Za-z0-9_-]</code> meaning any character between A to Z, a to z, 0 to 9 or _ or -, the following <code class="language-plaintext highlighter-rouge">+</code> meaning that any of these characters are found once or more.</p>

      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="convert-ab1-sequence-files-to-fastq-and-trim-low-quality-ends">Convert AB1 sequence files to FASTQ and trim low-quality ends</h2>

<p>In Sanger sequencing, ends tend to be of low trust levels (each nucleotide has a quality score reflecting this trust level), it is important to delete these sections of the sequences to ensure wrong nucleotides aren’t introduced in the sequences.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-ab1-to-fastq-files-and-trim-low-quality-ends"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  AB1 to FASTQ files and trim low quality ends</div>

  <p>Do these steps twice !! We have Froward and antisense (reverse) sequence data collections, do these steps starting with each “(filtered)” data collections, this could help:</p>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Re-running a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote>
  <p><!--END_SNIPPET--></p>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/ab1_fastq_converter/ab1_fastq_converter/1.20.0" title="ab1 to FASTQ converter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>ab1 to FASTQ converter</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.20.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input ab1 file”</em>: <code class="language-plaintext highlighter-rouge">(filtered) output collection</code> (output of <strong>Filter collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Do you want trim ends according to quality scores ?”</em>: <code class="language-plaintext highlighter-rouge">No, use full sequences.</code></li>
      </ul>
    </li>
  </ol>

  <p>In this tool, it is possible to trim low-quality ends along with the conversion of the file but parametrization is less precise.</p>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_trimfq/1.3.1" title="seqtk_trimfq tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>seqtk_trimfq</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input FASTA/Q file”</em>: <code class="language-plaintext highlighter-rouge">output collection</code> (output of <strong>ab1 to FASTQ converter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Mode for trimming FASTQ File”</em>: <code class="language-plaintext highlighter-rouge">Quality</code>
          <ul>
            <li><em>“Maximally trim down to INT bp”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="compute-reverse-complement-sequence-for-antisense-reverse-sequences-only">Compute reverse complement sequence for antisense (reverse) sequences only</h2>

<p>See in the introduction for explanations on the Reverse-Complement.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-reverse-complement"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Reverse complement</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/1.1.5" title="FASTQ Groomer tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FASTQ Groomer</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.5)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to groom”</em>: <code class="language-plaintext highlighter-rouge">#Reverse output collection</code> (output of <strong>seqtk_trimfq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Advanced Options”</em>: <code class="language-plaintext highlighter-rouge">Show Advanced Options</code>
          <ul>
            <li><em>“Summarize input data”</em>: <code class="language-plaintext highlighter-rouge">Do not Summarize Input (faster)</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-what-is-this-step"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  What is this step?</div>

        <p>It is a necessary step to get the right input format for the following step <strong>Reverse-Complement</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastx_reverse_complement/cshl_fastx_reverse_complement/1.0.2+galaxy0" title="Reverse-Complement tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Reverse-Complement</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.2+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input file in FASTA or FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">#Reverse output collection</code> (output of <strong>FASTQ Groomer</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="merge-corresponding-sense-and-antisense-sequences-single-files">Merge corresponding sense and antisense sequences single files</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-sort-collections"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Sort collections</div>

  <p>Do this step twice !! One has to make sure sense (forward) and antisense (reverse) sequences collections are in the same order to get the right sense and the right antisense sequence to be merged together</p>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Re-running a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote>
  <p><!--END_SNIPPET--></p>

  <ol>
    <li><span class="tool" data-tool="__SORTLIST__" title="Sort collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Sort collection</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Collection”</em>: <code class="language-plaintext highlighter-rouge">Collection</code> (output of <strong>seqtk_trimfq</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> &amp; output of <strong>Reverse-Complement</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Sort type”</em>: <code class="language-plaintext highlighter-rouge">alphabetical</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-merge-sense-forward-and-antisense-reverse-sequence-files"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Merge sense (forward) and antisense (reverse) sequence files</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/seqtk/seqtk_mergepe/1.3.1" title="seqtk_mergepe tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>seqtk_mergepe</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input FASTA/Q file #1”</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>Sort collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input FASTA/Q file #2”</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>Sort collection</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
  </ol>

  <p>Check there is two sequences in each three files of the newly-created collection.</p>

</blockquote>

<h2 id="convert-fastq-files-to-fasta">Convert FASTQ files to FASTA</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-fastq-to-fasta"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  FASTQ to FASTA</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/1.1.5" title="FASTQ Groomer tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FASTQ Groomer</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.5)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to groom”</em>: <code class="language-plaintext highlighter-rouge">default</code> (output of <strong>seqtk_mergepe</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Advanced Options”</em>: <code class="language-plaintext highlighter-rouge">Show Advanced Options</code>
          <ul>
            <li><em>“Summarize input data”</em>: <code class="language-plaintext highlighter-rouge">Do not Summarize Input (faster)</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-what-is-this-step-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  What is this step?</div>

        <p>It is a necessary step to get the right input format for the following step <strong>FASTQ to FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastq_to_fasta/cshl_fastq_to_fasta/1.0.2+galaxy2" title="FASTQ to FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FASTQ to FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.2+galaxy2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ file to convert”</em>: <code class="language-plaintext highlighter-rouge">output collection</code> (output of <strong>FASTQ Groomer</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Discard sequences with unknown (N) bases”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>“Rename sequence names in output file (reduces file size)”</em>: <code class="language-plaintext highlighter-rouge">no</code></li>
        <li><em>“Compress output FASTA”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-information"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  information</div>

        <p>If this step doesn’t work, one can try tools <strong>FASTQ to tabular</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>tabular to FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> instead</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="align-sequences-and-retrieve-consensus-for-each-sequence">Align sequences and retrieve consensus for each sequence</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-align-and-consensus"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Align and consensus</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/qiime_align_seqs/qiime_align_seqs/1.9.1.0" title="Align sequences tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Align sequences</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.1.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input fasta file”</em>: <code class="language-plaintext highlighter-rouge">output collection</code> (output of <strong>FASTQ-to-FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Method for aligning sequences”</em>: <code class="language-plaintext highlighter-rouge">clustalw</code></li>
        <li><em>“Minimum percent sequence identity to closest blast hit to include sequence in alignment”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/ecology/aligned_to_consensus/aligned_to_consensus/1.0.0" title="Consensus sequence from aligned FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Consensus sequence from aligned FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input fasta file with at least two sequences”</em>: <code class="language-plaintext highlighter-rouge">aligned_sequences</code> (output of <strong>Align sequences</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>Add tag “#Consensus”</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/mothur_merge_files/mothur_merge_files/1.39.5.0" title="Merge.files tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge.files</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.39.5.0)</span> with the following parameters:
      <ul>
        <li><em>“Merge”</em>: <code class="language-plaintext highlighter-rouge">fasta files</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“inputs - fasta”</em>: <code class="language-plaintext highlighter-rouge">output collection</code> (output of <strong>Consensus sequence from aligned FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="manage-primers-and-sequences">Manage primers and sequences</h1>

<h2 id="merge-and-align-consensus-sequences-file-and-primer-files">Merge and align consensus sequences file and primer files</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-merge-and-format-consensus-sequences-primers-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Merge and format consensus sequences + primers file</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/mothur_merge_files/mothur_merge_files/1.39.5.0" title="Merge.files tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge.files</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.39.5.0)</span> with the following parameters:
      <ul>
        <li><em>“Merge”</em>: <code class="language-plaintext highlighter-rouge">fasta files</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“inputs - fasta”</em>: <code class="language-plaintext highlighter-rouge">consensus sequences</code> (output of <strong>Merge.files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>), <code class="language-plaintext highlighter-rouge">Reverse primer</code> (output of <strong>Reverse-Complement</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>), <code class="language-plaintext highlighter-rouge">Forward primer</code> (output of <strong>Degap.seqs</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <ul>
        <li>Remove tags “#Forward” and “#Reverse”</li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3" title="Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">output</code> (output of <strong>Merge.files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Check”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">([A-Z-])&gt;</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">\1\n&gt;</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-what-s-going-on-in-this-second-step"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  What's going on in this second step?</div>

        <p>Sometimes, <strong>Merge.files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> doesn’t keep linefeed between the files, this step permits to correct it and get a FASTA file that is formatted properly.</p>

        <p>For the second step, we used regular expressions (Regex):</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-regular-expressions-101-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-regular-expressions-101-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Regular Expressions 101</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Regular expressions are a standardized way of describing patterns in textual data. They can be extremely useful for tasks such as finding and replacing data. They can be a bit tricky to master, but learning even just a few of the basics can help you get the most out of Galaxy.</p>   <p><strong>Finding</strong></p>   <p>Below are just a few examples of basic expressions:</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">abc</code></td>       <td>an occurrence of <code class="language-plaintext highlighter-rouge">abc</code> within your data</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(abc|def)</code></td>       <td><code class="language-plaintext highlighter-rouge">abc</code> <em>or</em> <code class="language-plaintext highlighter-rouge">def</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[abc]</code></td>       <td>a single character which is either <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, or <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[^abc]</code></td>       <td>a character that is NOT <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, nor <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-z]</code></td>       <td>any lowercase letter</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-zA-Z]</code></td>       <td>any letter (upper or lower case)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[0-9]</code></td>       <td>numbers 0-9</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d</code></td>       <td>any digit (same as <code class="language-plaintext highlighter-rouge">[0-9]</code>)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\D</code></td>       <td>any non-digit character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\w</code></td>       <td>any alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\W</code></td>       <td>any non-alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\s</code></td>       <td>any whitespace</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\S</code></td>       <td>any non-whitespace character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.</code></td>       <td><em>any</em> character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\.</code></td>       <td> </td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">{x,y}</code></td>       <td>between <em>x</em> and <em>y</em> repetitions</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^</code></td>       <td>the beginning of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">$</code></td>       <td>the end of the line</td>     </tr>   </tbody> </table>   <p><strong>Note:</strong> you see that characters such as <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">?</code>, <code class="language-plaintext highlighter-rouge">.</code>, <code class="language-plaintext highlighter-rouge">+</code> etc have a special meaning in a regular expression. If you want to match on those characters, you can escape them with a backslash. So <code class="language-plaintext highlighter-rouge">\?</code> matches the question mark character exactly.</p>   <p><strong>Examples</strong></p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d{4}</code></td>       <td>4 digits (e.g. a year)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr\d{1,2}</code></td>       <td><code class="language-plaintext highlighter-rouge">chr</code> followed by 1 or 2 digits</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.*abc$</code></td>       <td>anything with <code class="language-plaintext highlighter-rouge">abc</code> at the end of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^$</code></td>       <td>empty line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^&gt;.*</code></td>       <td>Line starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta header)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^[^&gt;].*</code></td>       <td>Line not starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta sequence)</td>     </tr>   </tbody> </table>   <p><strong>Replacing</strong></p>   <p>Sometimes you need to capture the exact value you matched on, in order to use it in your replacement, we do this using capture groups <code class="language-plaintext highlighter-rouge">(...)</code>, which we can refer to using <code class="language-plaintext highlighter-rouge">\1</code>, <code class="language-plaintext highlighter-rouge">\2</code> etc for the first and second captured values. If you want to refer to the whole match, use <code class="language-plaintext highlighter-rouge">&amp;</code>.</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Input</th>       <th>Captures</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr(\d{1,2})</code></td>       <td><code class="language-plaintext highlighter-rouge">chr14</code></td>       <td><code class="language-plaintext highlighter-rouge">\1 = 14</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(\d{2}) July (\d{4})</code></td>       <td>24 July 1984</td>       <td><code class="language-plaintext highlighter-rouge">\1 = 24</code>, <code class="language-plaintext highlighter-rouge">\2 = 1984</code></td>     </tr>   </tbody> </table>   <p>An expression like <code class="language-plaintext highlighter-rouge">s/find/replacement/g</code> indicates a replacement expression, this will search (<code class="language-plaintext highlighter-rouge">s</code>) for any occurrence of <code class="language-plaintext highlighter-rouge">find</code>, and replace it with <code class="language-plaintext highlighter-rouge">replacement</code>. It will do this globally (<code class="language-plaintext highlighter-rouge">g</code>) which means it doesn’t stop after the first match.</p>   <p>Example: <code class="language-plaintext highlighter-rouge">s/chr(\d{1,2})/CHR\1/g</code> will replace <code class="language-plaintext highlighter-rouge">chr14</code> with <code class="language-plaintext highlighter-rouge">CHR14</code> etc.</p>   <p>You can also use replacement modifier such as convert to lower case <code class="language-plaintext highlighter-rouge">\L</code> or upper case <code class="language-plaintext highlighter-rouge">\U</code>. Example: <code class="language-plaintext highlighter-rouge">s/.*/\U&amp;/g</code> will convert the whole text to upper case.</p>   <p>Note: In Galaxy, you are often asked to provide the find and replacement expressions separately, so you don’t have to use the <code class="language-plaintext highlighter-rouge">s/../../g</code> structure.</p>   <p>There is a lot more you can do with regular expressions, and there are a few different flavours in different tools/programming languages, but these are the most important basics that will already allow you to do many of the tasks you might need in your analysis.</p>   <p><strong>Tip:</strong> <a href="https://regexone.com/">RegexOne</a> is a nice interactive tutorial to learn the basics of regular expressions.</p>   <p><strong>Tip:</strong> <a href="https://regex101.com/">Regex101.com</a> is a great resource for interactively testing and constructing your regular expressions, it even provides an explanation of a regular expression if you provide one.</p>   <p><strong>Tip:</strong> <a href="https://extendsclass.com/regex-tester.html">Cyrilex</a> is a visual regular expression tester.</p> </blockquote>
        <p><!--END_SNIPPET--></p>

        <p>With <code class="language-plaintext highlighter-rouge">[A-Z-]</code> meaning any character between A to Z or -, <code class="language-plaintext highlighter-rouge">\1</code> repeat the character chain between brackets in the <em>“Find Regex”</em> section, <code class="language-plaintext highlighter-rouge">\n</code> meaning a line-feed.</p>

      </blockquote>
    </li>
  </ol>

</blockquote>

<p>When you have the consensus sequences, you can check if any ambiguous nucleotide is to be found in the sequences. If you find such nucleotides, it means different nucleotides were found in the sense and antisense sequence at the same position, some checks are needed.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-ambiguous-nucleotide-correspondance"><button class="gtn-boxify-button details" type="button" aria-controls="details-ambiguous-nucleotide-correspondance" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Ambiguous nucleotide correspondance</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <ul>
    <li>Y = C or T</li>
    <li>R = A or G</li>
    <li>W = A or T</li>
    <li>S = G or C</li>
    <li>K = T or G</li>
    <li>M = C or A</li>
  </ul>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-look-for-ambiguous-nucleotides"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Look for ambiguous nucleotides</div>

  <ol>
    <li>
      <p>Click on output of <strong>Regex Find and Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in the history to expand it</p>
    </li>
    <li>
      <p>Click on <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Visualize</p>
    </li>
    <li>
      <p>Select <strong>Multiple Sequence Alignment</strong></p>
    </li>
    <li>
      <p>Set color scheme to <code class="language-plaintext highlighter-rouge">Clustal</code>, ambiguous nucleotides are highlighted in dark blue</p>
    </li>
    <li>
      <p>There are two nucleotide positions to check, Y at 121 in sequence <code class="language-plaintext highlighter-rouge">consensus_B05_CHD8-III6brother-18</code> and W at 286 in sequence <code class="language-plaintext highlighter-rouge">consensus_05_CHD8-III6mother-18</code></p>
    </li>
    <li>
      <p>You need to go back to your FASTQ sequences to understand the origin of the ambiguity</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regex1/1.0.3" title="Regex Find And Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Regex Find And Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: <code class="language-plaintext highlighter-rouge">#Consensus #Primer output</code> (output of <strong>Regex Find and Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Check”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">^[ACTG]+([ACTG]{20}Y)[ACTG]+$</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">\1</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Check”</em>
              <ul>
                <li><em>“Find Regex”</em>: <code class="language-plaintext highlighter-rouge">^[ACTG]+([ACTG]{20}W)[ACTG]+$</code></li>
                <li><em>“Replacement”</em>: <code class="language-plaintext highlighter-rouge">\1</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-what-s-going-on-in-this-step"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment:  What's going on in this step?</div>

        <p>We want to retrieve the 20 nucleotides before the ambiguities.</p>

        <p>We use regular expressions (Regex):</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-regular-expressions-101-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-regular-expressions-101-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Regular Expressions 101</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Regular expressions are a standardized way of describing patterns in textual data. They can be extremely useful for tasks such as finding and replacing data. They can be a bit tricky to master, but learning even just a few of the basics can help you get the most out of Galaxy.</p>   <p><strong>Finding</strong></p>   <p>Below are just a few examples of basic expressions:</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">abc</code></td>       <td>an occurrence of <code class="language-plaintext highlighter-rouge">abc</code> within your data</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(abc|def)</code></td>       <td><code class="language-plaintext highlighter-rouge">abc</code> <em>or</em> <code class="language-plaintext highlighter-rouge">def</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[abc]</code></td>       <td>a single character which is either <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, or <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[^abc]</code></td>       <td>a character that is NOT <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, nor <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-z]</code></td>       <td>any lowercase letter</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-zA-Z]</code></td>       <td>any letter (upper or lower case)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[0-9]</code></td>       <td>numbers 0-9</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d</code></td>       <td>any digit (same as <code class="language-plaintext highlighter-rouge">[0-9]</code>)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\D</code></td>       <td>any non-digit character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\w</code></td>       <td>any alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\W</code></td>       <td>any non-alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\s</code></td>       <td>any whitespace</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\S</code></td>       <td>any non-whitespace character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.</code></td>       <td><em>any</em> character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\.</code></td>       <td> </td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">{x,y}</code></td>       <td>between <em>x</em> and <em>y</em> repetitions</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^</code></td>       <td>the beginning of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">$</code></td>       <td>the end of the line</td>     </tr>   </tbody> </table>   <p><strong>Note:</strong> you see that characters such as <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">?</code>, <code class="language-plaintext highlighter-rouge">.</code>, <code class="language-plaintext highlighter-rouge">+</code> etc have a special meaning in a regular expression. If you want to match on those characters, you can escape them with a backslash. So <code class="language-plaintext highlighter-rouge">\?</code> matches the question mark character exactly.</p>   <p><strong>Examples</strong></p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d{4}</code></td>       <td>4 digits (e.g. a year)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr\d{1,2}</code></td>       <td><code class="language-plaintext highlighter-rouge">chr</code> followed by 1 or 2 digits</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.*abc$</code></td>       <td>anything with <code class="language-plaintext highlighter-rouge">abc</code> at the end of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^$</code></td>       <td>empty line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^&gt;.*</code></td>       <td>Line starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta header)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^[^&gt;].*</code></td>       <td>Line not starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta sequence)</td>     </tr>   </tbody> </table>   <p><strong>Replacing</strong></p>   <p>Sometimes you need to capture the exact value you matched on, in order to use it in your replacement, we do this using capture groups <code class="language-plaintext highlighter-rouge">(...)</code>, which we can refer to using <code class="language-plaintext highlighter-rouge">\1</code>, <code class="language-plaintext highlighter-rouge">\2</code> etc for the first and second captured values. If you want to refer to the whole match, use <code class="language-plaintext highlighter-rouge">&amp;</code>.</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Input</th>       <th>Captures</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr(\d{1,2})</code></td>       <td><code class="language-plaintext highlighter-rouge">chr14</code></td>       <td><code class="language-plaintext highlighter-rouge">\1 = 14</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(\d{2}) July (\d{4})</code></td>       <td>24 July 1984</td>       <td><code class="language-plaintext highlighter-rouge">\1 = 24</code>, <code class="language-plaintext highlighter-rouge">\2 = 1984</code></td>     </tr>   </tbody> </table>   <p>An expression like <code class="language-plaintext highlighter-rouge">s/find/replacement/g</code> indicates a replacement expression, this will search (<code class="language-plaintext highlighter-rouge">s</code>) for any occurrence of <code class="language-plaintext highlighter-rouge">find</code>, and replace it with <code class="language-plaintext highlighter-rouge">replacement</code>. It will do this globally (<code class="language-plaintext highlighter-rouge">g</code>) which means it doesn’t stop after the first match.</p>   <p>Example: <code class="language-plaintext highlighter-rouge">s/chr(\d{1,2})/CHR\1/g</code> will replace <code class="language-plaintext highlighter-rouge">chr14</code> with <code class="language-plaintext highlighter-rouge">CHR14</code> etc.</p>   <p>You can also use replacement modifier such as convert to lower case <code class="language-plaintext highlighter-rouge">\L</code> or upper case <code class="language-plaintext highlighter-rouge">\U</code>. Example: <code class="language-plaintext highlighter-rouge">s/.*/\U&amp;/g</code> will convert the whole text to upper case.</p>   <p>Note: In Galaxy, you are often asked to provide the find and replacement expressions separately, so you don’t have to use the <code class="language-plaintext highlighter-rouge">s/../../g</code> structure.</p>   <p>There is a lot more you can do with regular expressions, and there are a few different flavours in different tools/programming languages, but these are the most important basics that will already allow you to do many of the tasks you might need in your analysis.</p>   <p><strong>Tip:</strong> <a href="https://regexone.com/">RegexOne</a> is a nice interactive tutorial to learn the basics of regular expressions.</p>   <p><strong>Tip:</strong> <a href="https://regex101.com/">Regex101.com</a> is a great resource for interactively testing and constructing your regular expressions, it even provides an explanation of a regular expression if you provide one.</p>   <p><strong>Tip:</strong> <a href="https://extendsclass.com/regex-tester.html">Cyrilex</a> is a visual regular expression tester.</p> </blockquote>
        <p><!--END_SNIPPET--></p>

        <p>With <code class="language-plaintext highlighter-rouge">[ACTG]</code> meaning any character of the four unambiguous nucleotides followed by <code class="language-plaintext highlighter-rouge">+</code> meaning “at least once in the character chain” or by {20} meaning “20 times”.</p>

      </blockquote>

      <p>In the output of this tool we get:
    - the 20 nucleotides before the Y at position 121 in sequence <code class="language-plaintext highlighter-rouge">consensus_B05_CHD8-III6brother-18</code>: <code class="language-plaintext highlighter-rouge">CAGGCACGATGTCATCGAAT</code>
    - and the 20 nuleotides before the W at position 286 in sequence <code class="language-plaintext highlighter-rouge">consensus_05_CHD8-III6mother-18</code>: <code class="language-plaintext highlighter-rouge">AGTCCTCTTAGTTTATAGAT</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastq_masker_by_quality/fastq_masker_by_quality/1.1.5" title="FASTQ masker tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FASTQ masker</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.5)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“File to mask”</em>: <code class="language-plaintext highlighter-rouge">#Forward #Reverse collection</code> (output of <strong>FASTQ groomer</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Mask input with”</em>: <code class="language-plaintext highlighter-rouge">Lowercase</code></li>
        <li><em>“Quality score”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
      </ul>
    </li>
  </ol>

  <p>This tool displays low-quality bases in lowercase to permit better detection of potential errors.</p>

  <ol>
    <li>
      <p>Open <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <code class="language-plaintext highlighter-rouge">B05_CHD8-III6brother-18</code> output of <strong>FASTQ masker</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and ctrl+f : <code class="language-plaintext highlighter-rouge">CAGGCACGATGTCATCGAAT</code>.
In the sense sequence (ID ending with 18F), this fragment is followed by a <code class="language-plaintext highlighter-rouge">c</code> in low-quality, whereas in the antisense sequence it is followed by a <code class="language-plaintext highlighter-rouge">T</code> in decent quality.
Additionally, when looking into the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <code class="language-plaintext highlighter-rouge">#Consensus #Primer</code> output of <strong>Regex Find and Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, we can see the two other consensus sequences (<code class="language-plaintext highlighter-rouge">consensus_05_CHD8-III6mother-18</code> and <code class="language-plaintext highlighter-rouge">consensus_07_CHD8-III6-18</code>) have a <code class="language-plaintext highlighter-rouge">T</code> at this same position.
It seems more likely that the nucleotide at position 121 in sequence <code class="language-plaintext highlighter-rouge">consensus_B05_CHD8-III6brother-18</code> is a <code class="language-plaintext highlighter-rouge">T</code>.</p>
    </li>
    <li>
      <p>Open <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <code class="language-plaintext highlighter-rouge">05_CHD8-III6mother-18</code> outputs of <strong>FASTQ masker</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and ctrl+f : <code class="language-plaintext highlighter-rouge">AGTCCTCTTAGTTTATAGAT</code>.
In the antisense sequence (ID ending with 18R), this fragment is followed by a <code class="language-plaintext highlighter-rouge">t</code> in low-quality, whereas in the sense sequence it is followed by a <code class="language-plaintext highlighter-rouge">A</code> in decent quality.
Additionally, when looking into the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <code class="language-plaintext highlighter-rouge">#Consensus #Primer</code> output of <strong>Regex Find and Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, we can see the two other consensus sequences (<code class="language-plaintext highlighter-rouge">consensus_B05_CHD8-III6brother-18</code> and <code class="language-plaintext highlighter-rouge">consensus_07_CHD8-III6-18</code>) have a <code class="language-plaintext highlighter-rouge">A</code> at this same position.
It seems more likely that the nucleotide at position 286 in sequence <code class="language-plaintext highlighter-rouge">consensus_05_CHD8-III6mother-18</code> is a <code class="language-plaintext highlighter-rouge">A</code>.</p>
    </li>
    <li>
      <p>You can now correct them by clicking on output of <strong>Regex Find and Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in the history to expand it</p>
    </li>
    <li>
      <p>Click on <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Visualize</p>
    </li>
    <li>
      <p>Select <strong>Editor</strong> and:</p>
      <ul>
        <li>replace manually the <code class="language-plaintext highlighter-rouge">Y</code> with <code class="language-plaintext highlighter-rouge">T</code> in <code class="language-plaintext highlighter-rouge">consensus_B05_CHD8-III6brother-18</code></li>
        <li>replace manually the <code class="language-plaintext highlighter-rouge">W</code> with <code class="language-plaintext highlighter-rouge">A</code> in <code class="language-plaintext highlighter-rouge">consensus_05_CHD8-III6mother-18</code>
and click on <strong>export</strong></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Now, one can align its sequences with primers. Ultimately, it is common to cut sequences between primers to get the right fragment for each sequence.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-align-sequences-and-primers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Align sequences and primers</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/qiime_align_seqs/qiime_align_seqs/1.9.1.0" title="Align sequences tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Align sequences</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.9.1.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input fasta file”</em>: <code class="language-plaintext highlighter-rouge">out_file1</code> <strong>Regex Find And Replace (modified)</strong></li>
        <li><em>“Method for aligning sequences”</em>: <code class="language-plaintext highlighter-rouge">mafft</code></li>
        <li><em>“Minimum percent sequence identity to closest blast hit to include sequence in alignment”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="check-your-sequences-belongs-to-the-right-taxonomic-group-by-computing-a-blast-on-the-ncbi-database">Check your sequences belongs to the right taxonomic group by computing a BLAST on the NCBI database</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-nvbi-blast"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  NVBI Blast</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/2.10.1+galaxy2" title="NCBI BLAST+ blastn tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>NCBI BLAST+ blastn</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.10.1+galaxy2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Nucleotide query sequence(s)”</em>: <code class="language-plaintext highlighter-rouge">out_file1</code> (output of <strong>Regex Find And Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Subject database/sequences”</em>: <code class="language-plaintext highlighter-rouge">Locally installed BLAST database</code>
          <ul>
            <li><em>“Nucleotide BLAST database”</em>: select most recent <code class="language-plaintext highlighter-rouge">nt_</code> database</li>
          </ul>
        </li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">Tabular (select which columns)</code>
          <ul>
            <li><em>“Standard columns”</em>: <code class="language-plaintext highlighter-rouge">qseqid</code>, <code class="language-plaintext highlighter-rouge">pident</code>, <code class="language-plaintext highlighter-rouge">mismatch</code> and <code class="language-plaintext highlighter-rouge">gapopen</code></li>
            <li><em>“Extended columns”</em>: <code class="language-plaintext highlighter-rouge">gaps</code> and <code class="language-plaintext highlighter-rouge">salltitles</code></li>
            <li><em>“Other identifier columns”</em>: <code class="language-plaintext highlighter-rouge">saccver</code></li>
          </ul>
        </li>
        <li><em>“Advanced Options”</em>: <code class="language-plaintext highlighter-rouge">Show Advanced Options</code>
          <ul>
            <li><em>“Maximum hits to consider/show”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>“Restrict search of database to a given set of ID’s”</em>: <code class="language-plaintext highlighter-rouge">No restriction, search the entire database</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
    <p>The sequences we cleaned belong to what species?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p><em>Homo sapiens</em></p>

    </blockquote>

  </blockquote>

</blockquote>

<p>It is a good practice to proceed to such checks, its permits to make sure the sequencing went as planned and your samples haven’t been contaminated.</p>

<h1 id="conclusion">Conclusion</h1>

<p>We successfully cleaned AB1 sequence files !</p>

<h1 id="aopep-sanger-files">AOPEP Sanger files</h1>

<p>The history following the same steps but for AOPEP marker files is available: <a href="https://usegalaxy.fr/u/coline_royaux/h/tuto-sanger-w-aopep">Clean AOPEP sequences</a></p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Check your data and results for mistakes afterward !!!! This procedure is useful but not perfect</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/sequence-analysis/faqs/">FAQ page for the Sequence analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Garavaglia2022"><b>AOPEP variants as a novel cause of recessive dystonia: Generalized dystonia and dystonia-parkinsonism</b>, 2022 Parkinsonism and related disorders 97: 52–56. <a href="https://doi.org/10.1016/j.parkreldis.2022.03.007">10.1016/j.parkreldis.2022.03.007</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=sequence-analysis/Manage_AB1_Sanger"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=sequence-analysis/Manage_AB1_Sanger">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Coline Royaux,  <b>Clean and manage Sanger sequences from raw files to aligned consensus (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html">https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{sequence-analysis-Manage_AB1_Sanger,
author = "Coline Royaux",
	title = "Clean and manage Sanger sequences from raw files to aligned consensus (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: fastq_groomer
  owner: devteam
  revisions: 47e5dbc3e790
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastq_masker_by_quality
  owner: devteam
  revisions: 9dfda4e310ed
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastq_to_fasta
  owner: devteam
  revisions: 77b41c89d856
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastq_to_tabular
  owner: devteam
  revisions: 4b347702c4aa
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastx_reverse_complement
  owner: devteam
  revisions: 6027ef51ef91
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ncbi_blast_plus
  owner: devteam
  revisions: 0e3cf9594bb7
  tool_panel_section_label: NCBI Blast
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: tabular_to_fasta
  owner: devteam
  revisions: 0a7799698fe5
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: ab1_fastq_converter
  owner: ecology
  revisions: 307518fb51af
  tool_panel_section_label: Convert Formats
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: aligned_to_consensus
  owner: ecology
  revisions: 0ccbe1c20fc3
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: filter_by_fasta_ids
  owner: galaxyp
  revisions: dff7df6fcab5
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: regex_find_replace
  owner: galaxyp
  revisions: 503bcd6ebe4b
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: unzip
  owner: imgteam
  revisions: 57f0914ddb7b
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collection_element_identifiers
  owner: iuc
  revisions: d3c07d270a50
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_degap_seqs
  owner: iuc
  revisions: 6f08af23888a
  tool_panel_section_label: Mothur
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mothur_merge_files
  owner: iuc
  revisions: bc20680d28d5
  tool_panel_section_label: Mothur
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qiime_align_seqs
  owner: iuc
  revisions: e8bb88f051ec
  tool_panel_section_label: Sequence analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seqtk
  owner: iuc
  revisions: 3da72230c066
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: seqtk
  owner: iuc
  revisions: 3da72230c066
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">1</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">1 stars</span><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">1</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>August 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">1 stars</span><i class="fa fa-star" aria-hidden="true"></i>:
					
					<b>Disliked</b>: the steps are diffucult to follo, it's not always clear what was done, workflow doesn't reflect the tutorial well 
				</li>
				
		    </ul>
		    
		    <b>June 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: good dataset
					<b>Disliked</b>: align sequences tool error
				</li>
				
		    </ul>
		    
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00383">gxy.io/GTN:T00383</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/sequence-analysis/tutorials/Manage_AB1_Sanger/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/82479b8220ed6ad45f0170c036bce307d7f50bd3">82479b8</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>