---
layout: tutorial_slides
logo: "GTN"

title: "Identification of non-canonical ORFs and their potential biological function"
contributors:
  - gallardoalba

priority: 4

video: no
zenodo_link: ""


---

## Index of contents

1. Introduction

2. Galaxy workflow

---

## Introduction

- What do we mean by non-canonical ORFs?

- Why study non-canonical ORFs?

- Why have not been yet characterized?

- How identify non-canonical ORFs?

---

## What do we mean by non-canonical ORFs?

![Definition of canonical]({% link topics/transcriptomics/images/differential_isoform/slides/cambridge_dictionary.png %})

.footnote[Source: [Cambridge dictionary](https://dictionary.cambridge.org/dictionary/english/canonical)]

---

## What do we mean by non-canonical ORFs?

![Mean Ribo-Seq expression and Ribo-Seq expression standard deviation (SD) have been plotted for human lymphoblastoid cells from RPFdbV2.]({% link topics/transcriptomics/images/differential_isoform/slides/expression_distribution.png %})

.footnote[Source: {% cite Erady2021 %}]

---

## What do we mean by non-canonical ORFs?

![The dark proteome: translation from noncanonical open reading frames.]({% link topics/transcriptomics/images/differential_isoform/slides/dark_proteome.png %})

---

## Why study non-canonical ORFs?

- Potentially novel <b>prognostic and diagnostic markers</b>

- The vast majority <b>have not been investigated</b>

- Particulary attractive as <b>allosteric celullar regulators</b>

---

## Why study non-canonical ORFs?

![Noncanonical open reading frames enconde functional proteins essential for cancer cell survival.]({% link topics/transcriptomics/images/differential_isoform/slides/non-canonical_nature.png %})

---

## Why study non-canonical ORFs?

![Translation and natural selection of micropeptides from lon non-canonical RNAs.]({% link topics/transcriptomics/images/differential_isoform/slides/non-canonical_nature2.png %})

---

## Why have not been characterized?


- Arbitrary thresholds on ORF lengths
    - Peptides smaller than 100 aminoacids are usually discarted
- Frequently annotated as non-coding RNAs
- Propensity for structural disorder
    -  Discarted as intrinsically disordered proteins (IDPs)

---

## Why study small peptides?

![The large unexplored biology of small proteins in pro and eukaryotes.]({% link topics/transcriptomics/images/differential_isoform/slides/small_peptides.png %})

---

## Why study small peptides?

![Short peptides regulate gene expression.]({% link topics/transcriptomics/images/differential_isoform/slides/short_peptides.png %})

---

## Why study small peptides?

![Example of small peptides biological function.]({% link topics/transcriptomics/images/differential_isoform/slides/small_peptides_examples.png %})

.footnote[Source: {% cite Steinberg2021 %}]

---

## Annotated as non-coding RNAs?

![Coding or non-coding? This is the question.]({% link topics/transcriptomics/images/differential_isoform/slides/question.jpg %})

---

## Annotated as non-coding RNAs?

![The small peptide world in long noncoding RNAs.]({% link topics/transcriptomics/images/differential_isoform/slides/small_non_coding.png %})

---

## Annotated as non-coding RNAs?

![SPENCER: a comprehensive database for small peptides encoded by noncanical RNAs in cancer patients.]({% link topics/transcriptomics/images/differential_isoform/slides/spencer_datase.png %})

---

## Why study intrinsically disordered proteins (IDP)?

![Why study intrinsically disordered proteins?]({% link topics/transcriptomics/images/differential_isoform/slides/disordered_regions.png %})

.footnote[Source: {% cite Babu2012 %}]

---

## Disorder-Function Paradigm

![Disorder-Function paradigm.]({% link topics/transcriptomics/images/differential_isoform/slides/disorder_article.png %})

---

## Disorder-Function Paradigm

![Expanding the paradigm: intrinsically disordered proteins and allosteric regulation.]({% link topics/transcriptomics/images/differential_isoform/slides/intrinsically_disordered.png %})

---

## Disorder-Function Paradigm

![Intrinsically disordered proteins/regions and insight into their biomolecular interaction.]({% link topics/transcriptomics/images/differential_isoform/slides/disorder_protein.jpg %})

.footnote[Source: {% cite Chakrabarti2022 %}]

---

## How identify non-canonical ORFs?

![How to identify non-canonical ORFs.]({% link topics/transcriptomics/images/differential_isoform/slides/scientist.jpg %})

---

## How identify non-canonical ORFs?

![IsoformSwitchAnalyzeR: analysis of changes in genome-wide patterns of alternative splicing and its functional consequences.]({% link topics/transcriptomics/images/differential_isoform/slides/isoform_switching.png %})

---

## How identify non-canonical ORFs?

![Illustration of isoform switch]({% link topics/transcriptomics/images/differential_isoform/slides/isoform_switching.jpg %})

---

## Galaxy Workflow

![Galaxy workflow.]({% link topics/transcriptomics/images/differential_isoform/slides/galaxy_training.png %})

Full detailed explanation in the [Genome-wide alternative splicing analysis]({% link topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.md %}) Galaxy training.

---

## Galaxy Workflow

![Full workflow image.]({% link topics/transcriptomics/images/differential_isoform/slides/full_workflow.png %})

Full detailed explanation in the [Genome-wide alternative splicing analysis]({% link topics/transcriptomics/tutorials/differential-isoform-expression/tutorial.md %}) Galaxy training.

---

## Galaxy Workflow

![Workflow summary.]({% link topics/transcriptomics/images/differential_isoform/slides/simple_workflow.png %})

---

## Initial QC assessment

![QC step.Identify potential artifacts that may impact the interpretation of downstream analysis.]({% link topics/transcriptomics/images/differential_isoform/slides/initial_QC.png %})

Identify potential artifacts that may impact the interpretation of downstream analysis.

---

## Mapping and identication of novel splicing sites with RNASTAR

![Mapping step with RNASTAR. Two-pass alignment enables sequence reads to span novel splice junctions by fewer nucleotides, conferring greater read depth and providing significantly more accurate quantification of novel splice junctions.]({% link topics/transcriptomics/images/differential_isoform/slides/STAR_pipeline.png  %})

Two-pass alignment enables sequence reads to span novel splice junctions by fewer nucleotides, conferring greater read depth and providing significantly more accurate quantification of novel splice junctions.

---

## Post-mapping QC assessment with RSeQC

![Post-mapping QC. RSeQC is a toolkit for generating RNA-seq-specific quality control metrics. The figure corresponds to RSeQC junction saturation of known (A) and novel (B) splicing sites.]({% link topics/transcriptomics/images/differential_isoform/slides/rseqc_junction_saturation_plot.png %})

RSeQC is a toolkit for generating RNA-seq-specific quality control metrics. The figure corresponds to RSeQC junction saturation of known (A) and novel (B) splicing sites.

---

## Reference-based transcriptome assembly and quantification with StringTie

![Transcriptome assembly and quantification with StringTie. StringTie is a fast and highly efficient assembler of RNA-seq alignments into potential transcripts.]({% link topics/transcriptomics/images/differential_isoform/slides/stringtie_pipeline.png %})

StringTie is a fast and highly efficient assembler of RNA-seq alignments into potential transcripts.

---

## Post-assembly QC assessment with rnaQUAST

![Post-assembly QC. rnaQUAST, which will provide us diverse completeness/correctness statistics very useful in order to identify and address potential errors or gaps in the assembly process. The figure is a rnaQUAST cummulative isoform plot.]({% link topics/transcriptomics/images/differential_isoform/slides/rnaQUAST_cumulative_isoform.png %})

rnaQUAST, which will provide us diverse completeness/correctness statistics very useful in order to identify and address potential errors or gaps in the assembly process. The figure is a rnaQUAST cummulative isoform plot.

---

## Isoform switching and functional analysis with IsoformSwitchAnalyzeR

![Isoform switching and functional analysis. IsoformSwitchAnalyzieR performs the differential isoform usage analysis by using DEXSeq.]({% link topics/transcriptomics/images/differential_isoform/slides/isoformswitch_pipeline.png %})

IsoformSwitchAnalyzieR performs the differential isoform usage analysis by using DEXSeq.

---

## Isoform switching and functional analysis with IsoformSwitchAnalyzeR

![IsoformSwitchAnalyzeR consequences plot.]({% link topics/transcriptomics/images/differential_isoform/slides/consequences_plot.png %})

To analyze large-scale patterns in predicted IS consequences, IsoformSwitchAnalyzeR computes all isoform switching events resulting in a gain/loss of a specific consequence (e.g. protein domain gain/loss)

---

## Candidate non-canonical ORF

![Candidate non-canonical ORF.]({% link topics/transcriptomics/images/differential_isoform/slides/candidate_gene.png %})

