<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Filter, Plot and Explore Single-cell RNA-seq Data updated / Filter, plot and explore single-cell RNA-seq data with Scanpy / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@3b04f69d0ed7c2b1f7b1522c736ccfcd38f375c3",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Filter, Plot and Explore Single-cell RNA-seq Data updated">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="1970-01-01 00:00:00 +0000"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Filter, plot and explore single-cell RNA-seq data with Scanpy / Filter, Plot and Explore Single-cell RNA-seq Data updated">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.png">
	</head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<style>
hgroup {margin-bottom: 2rem; border-bottom: 1px solid var(--border-light);}
hgroup p { font-family: monospace; font-size: 0.8rem;}
.meta ul { padding-left: 1em;}
</style>

<section>
	<hgroup>
		<h1 data-toc-skip>Filter, Plot and Explore Single-cell RNA-seq Data updated</h1>
		<p>single-cell-scrna-case_basic-pipeline/filter--plot-and-explore-single-cell-rna-seq-data-updated</p>
	</hgroup>
	
	<div class="row">
		<div class="col-md-3 meta">
			
			<b>Author(s)</b>
			<br/>
			
			Wendi Bacon, 
			
			Julia Jakiela
			
			
			<br/>

			<i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span>
			<b>Version</b>
			<br/>
			1

			<br/>

			<i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span>
			<b>Last updated</b>
			<br/>
			Jun 13, 2023

			<br/>

			<i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">license</span>
			<b>License</b>
			<br/>

			
			<a href="https://spdx.org/licenses/CC-BY-4.0">CC-BY-4.0</a>
			
			<br/>

			
			<i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>
			<b>Tags</b>
			<br/>

			
			<div class="label label-default tutorial_tag" style="--color-primary: #d7c600; --color-darker: #c0b000; --color-dimmed: #cbbb00;">transcriptomics</div>
			
			</br>
			


			<b>Features</b>
			<ul>
			
			
			
			
			</ul>

			<hr/>


			<b>Tutorial</b>
			<br/>
			<i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> <a href="../tutorial.html">Filter, plot and explore single-cell RNA-seq data with Scanpy</a>
			</br>

			
			
			<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> <a href="index.html">Other workflows associated with this material</a>
			


			<hr/>
			<b>Workflow Testing</b>
			<br/>

			Tests: ✅ <br/>
			Results: Not yet automated

			<hr/>

			<b>FAIRness</b>

			<i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span>
			<b><abbr title="Persistent URL">PURL</abbr></b>
			<br/>
			<a href="https://gxy.io/GTN:W00197">https://gxy.io/GTN:W00197</a>
			<br/>

			
			<a style="margin-bottom: 0.2rem" href="/training-material/api/workflows/single-cell/scrna-case_basic-pipeline/filter--plot-and-explore-single-cell-rna-seq-data-updated/rocrate.zip" class="btn btn-secondary"><img src="/training-material/assets/images/ro-crate.png" style="height: 32px" alt="RO-Crate logo with flask"/> Download Workflow RO-Crate</a>
			

			
			<a style="margin-bottom: 0.2rem" href="https://workflowhub.eu/workflows/1212" class="btn btn-secondary"><img src="/training-material/assets/images/workflowhub-icon.svg" style="height: 32px" alt="Workflowhub cloud with gears logo"/> View on WorkflowHub</a>
			

		</div>
		<div class="col-md-9">
			<span tabindex="0" role="button" class="workflow show-when-galaxy-proxy-active" data-workflow="https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_basic-pipeline/versions/filter--plot-and-explore-single-cell-rna-seq-data-updated">Launch in Tutorial Mode</span>
			<a href="/training-material/faqs/galaxy/tutorial_mode.html" class="show-when-galaxy-proxy-active"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span></a>
			<br/>

			<span class="dropdown">
				<button class="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown" aria-expanded="false">
					<img src="/training-material/assets/images/galaxylogo.png" style="height: 16px" alt="galaxy logo"/> 
					Run Workflow in Galaxy
				</button>
				<ul class="dropdown-menu">
					
					
					<li class="dropdown-item"><a href="https://usegalaxy.org/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_basic-pipeline/versions/filter--plot-and-explore-single-cell-rna-seq-data-updated">Galaxy Main</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.eu/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_basic-pipeline/versions/filter--plot-and-explore-single-cell-rna-seq-data-updated">Galaxy Europe</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.fr/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_basic-pipeline/versions/filter--plot-and-explore-single-cell-rna-seq-data-updated">Galaxy France</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.org.au/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_basic-pipeline/versions/filter--plot-and-explore-single-cell-rna-seq-data-updated">Galaxy Australia</a></li>
					
					<li class="dropdown-item hide-when-galaxy-proxy-active">
						<a href="https://my.galaxy.training/?path=/workflows/trs_import%3Frun_form=true%26trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_basic-pipeline/versions/filter--plot-and-explore-single-cell-rna-seq-data-updated">Import to another server (≥23.0+ only!)</a>
					</li>
				</ul>
			</span>
			<a class="btn btn-primary" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.ga">
				<i class="fas fa-download" aria-hidden="true"></i><span class="visually-hidden">galaxy-download</span>
				Download
			</a>
			<pre class="mermaid">flowchart TD
  0["ℹ️ Input Dataset\nMito-counted AnnData"];
  style 0 stroke:#2c3143,stroke-width:4px;
  1["Inspect AnnData"];
  0 -->|output| 1;
  2["Scanpy FilterCells"];
  0 -->|output| 2;
  6a00b14b-d3f8-4771-963a-76115c8eaf2f["Output\nGenes-filtered Object"];
  2 --> 6a00b14b-d3f8-4771-963a-76115c8eaf2f;
  style 6a00b14b-d3f8-4771-963a-76115c8eaf2f stroke:#2c3143,stroke-width:4px;
  3["Plot"];
  0 -->|output| 3;
  c123842e-b2d2-43ad-81f0-6ba3b45d4021["Output\nScatter - genes x UMIs"];
  3 --> c123842e-b2d2-43ad-81f0-6ba3b45d4021;
  style c123842e-b2d2-43ad-81f0-6ba3b45d4021 stroke:#2c3143,stroke-width:4px;
  4["Plot"];
  0 -->|output| 4;
  f1020d4d-555e-42fa-ba21-52a151b91a5b["Output\nScatter - mito x genes"];
  4 --> f1020d4d-555e-42fa-ba21-52a151b91a5b;
  style f1020d4d-555e-42fa-ba21-52a151b91a5b stroke:#2c3143,stroke-width:4px;
  5["Plot"];
  0 -->|output| 5;
  a7fe6020-8f1e-470a-9ed3-f2b5e95516e0["Output\nViolin - genotype - log"];
  5 --> a7fe6020-8f1e-470a-9ed3-f2b5e95516e0;
  style a7fe6020-8f1e-470a-9ed3-f2b5e95516e0 stroke:#2c3143,stroke-width:4px;
  6["Plot"];
  0 -->|output| 6;
  15e501bc-8ddb-4519-89eb-dde431ea96c1["Output\nViolin - batch - log"];
  6 --> 15e501bc-8ddb-4519-89eb-dde431ea96c1;
  style 15e501bc-8ddb-4519-89eb-dde431ea96c1 stroke:#2c3143,stroke-width:4px;
  7["Inspect AnnData"];
  0 -->|output| 7;
  8["Plot"];
  0 -->|output| 8;
  eff22f46-baa6-4e00-ba82-d5e12ce26ff0["Output\nScatter - mito x UMIs"];
  8 --> eff22f46-baa6-4e00-ba82-d5e12ce26ff0;
  style eff22f46-baa6-4e00-ba82-d5e12ce26ff0 stroke:#2c3143,stroke-width:4px;
  9["Inspect AnnData"];
  0 -->|output| 9;
  10["Plot"];
  0 -->|output| 10;
  56677ca4-129c-476c-85ec-69d1bb3d800d["Output\nViolin - sex - log"];
  10 --> 56677ca4-129c-476c-85ec-69d1bb3d800d;
  style 56677ca4-129c-476c-85ec-69d1bb3d800d stroke:#2c3143,stroke-width:4px;
  11["Plot"];
  2 -->|output_h5ad| 11;
  acb61ea4-bcb9-45db-beef-e2bf1a176701["Output\nViolin - Filterbygenes"];
  11 --> acb61ea4-bcb9-45db-beef-e2bf1a176701;
  style acb61ea4-bcb9-45db-beef-e2bf1a176701 stroke:#2c3143,stroke-width:4px;
  12["Scanpy FilterCells"];
  2 -->|output_h5ad| 12;
  51853662-4519-4229-a2ea-b22a53e7ef73["Output\nCounts-filtered Object"];
  12 --> 51853662-4519-4229-a2ea-b22a53e7ef73;
  style 51853662-4519-4229-a2ea-b22a53e7ef73 stroke:#2c3143,stroke-width:4px;
  13["Inspect AnnData"];
  2 -->|output_h5ad| 13;
  362a7fe6-24bb-4398-ae48-870f4b4bb774["Output\nGeneral - Filterbygenes"];
  13 --> 362a7fe6-24bb-4398-ae48-870f4b4bb774;
  style 362a7fe6-24bb-4398-ae48-870f4b4bb774 stroke:#2c3143,stroke-width:4px;
  14["Inspect AnnData"];
  12 -->|output_h5ad| 14;
  edf24149-9341-4fe7-b10c-3fcf092faaa5["Output\nGeneral - Filterbycounts"];
  14 --> edf24149-9341-4fe7-b10c-3fcf092faaa5;
  style edf24149-9341-4fe7-b10c-3fcf092faaa5 stroke:#2c3143,stroke-width:4px;
  15["Scanpy FilterCells"];
  12 -->|output_h5ad| 15;
  a88ec405-265f-4a59-a75e-34e3b05b0096["Output\nMito-filtered Object"];
  15 --> a88ec405-265f-4a59-a75e-34e3b05b0096;
  style a88ec405-265f-4a59-a75e-34e3b05b0096 stroke:#2c3143,stroke-width:4px;
  16["Plot"];
  12 -->|output_h5ad| 16;
  a7c8b0d9-82d3-4438-a212-b5f7c56d36b8["Output\nViolin - Filterbycounts"];
  16 --> a7c8b0d9-82d3-4438-a212-b5f7c56d36b8;
  style a7c8b0d9-82d3-4438-a212-b5f7c56d36b8 stroke:#2c3143,stroke-width:4px;
  17["Inspect AnnData"];
  15 -->|output_h5ad| 17;
  56882809-e19f-451a-8010-bc55dcee482f["Output\nGeneral - Filterbymito"];
  17 --> 56882809-e19f-451a-8010-bc55dcee482f;
  style 56882809-e19f-451a-8010-bc55dcee482f stroke:#2c3143,stroke-width:4px;
  18["Scanpy FilterGenes"];
  15 -->|output_h5ad| 18;
  00846477-dec5-408a-83b2-105fff7ce05b["Output\nFiltered Object"];
  18 --> 00846477-dec5-408a-83b2-105fff7ce05b;
  style 00846477-dec5-408a-83b2-105fff7ce05b stroke:#2c3143,stroke-width:4px;
  19["Plot"];
  15 -->|output_h5ad| 19;
  7582e113-2004-4255-a1f3-d3123373f342["Output\nViolin - Filterbymito"];
  19 --> 7582e113-2004-4255-a1f3-d3123373f342;
  style 7582e113-2004-4255-a1f3-d3123373f342 stroke:#2c3143,stroke-width:4px;
  20["Inspect AnnData"];
  18 -->|output_h5ad| 20;
  2d870a40-c602-4a1c-afef-450489354d39["Output\nGeneral - Filtered object"];
  20 --> 2d870a40-c602-4a1c-afef-450489354d39;
  style 2d870a40-c602-4a1c-afef-450489354d39 stroke:#2c3143,stroke-width:4px;
  21["Scanpy NormaliseData"];
  18 -->|output_h5ad| 21;
  22["Scanpy FindVariableGenes"];
  21 -->|output_h5ad| 22;
  a0eb92b1-0263-4179-b7af-4bd9bcc9c960["Output\nUse_me_FVG"];
  22 --> a0eb92b1-0263-4179-b7af-4bd9bcc9c960;
  style a0eb92b1-0263-4179-b7af-4bd9bcc9c960 stroke:#2c3143,stroke-width:4px;
  23["Scanpy ScaleData"];
  22 -->|output_h5ad| 23;
  5776dbb9-0cac-40c0-9bae-9accae16a7a0["Output\nUse_me_Scaled"];
  23 --> 5776dbb9-0cac-40c0-9bae-9accae16a7a0;
  style 5776dbb9-0cac-40c0-9bae-9accae16a7a0 stroke:#2c3143,stroke-width:4px;
  24["Scanpy RunPCA"];
  23 -->|output_h5ad| 24;
  25["Plot"];
  24 -->|output_h5ad| 25;
  f0b6f578-050f-4936-9ee7-9956b0760c6f["Output\nPCA Variance"];
  25 --> f0b6f578-050f-4936-9ee7-9956b0760c6f;
  style f0b6f578-050f-4936-9ee7-9956b0760c6f stroke:#2c3143,stroke-width:4px;
  26["Scanpy ComputeGraph"];
  24 -->|output_h5ad| 26;
  27["Scanpy RunTSNE"];
  26 -->|output_h5ad| 27;
  28["Scanpy RunUMAP"];
  27 -->|output_h5ad| 28;
  29["Scanpy FindCluster"];
  28 -->|output_h5ad| 29;
  30["Scanpy FindMarkers"];
  29 -->|output_h5ad| 30;
  7cfe5c9c-1c80-41b4-b669-633b1d7d40e3["Output\nMarkers - cluster"];
  30 --> 7cfe5c9c-1c80-41b4-b669-633b1d7d40e3;
  style 7cfe5c9c-1c80-41b4-b669-633b1d7d40e3 stroke:#2c3143,stroke-width:4px;
  98e98405-951e-4c6c-be01-3c925ae35449["Output\nFinal object"];
  30 --> 98e98405-951e-4c6c-be01-3c925ae35449;
  style 98e98405-951e-4c6c-be01-3c925ae35449 stroke:#2c3143,stroke-width:4px;
  31["Scanpy FindMarkers"];
  29 -->|output_h5ad| 31;
  1e9f229d-eb34-4a5b-a6d9-e70c7b0581f4["Output\nMarkers - genotype"];
  31 --> 1e9f229d-eb34-4a5b-a6d9-e70c7b0581f4;
  style 1e9f229d-eb34-4a5b-a6d9-e70c7b0581f4 stroke:#2c3143,stroke-width:4px;
  32["Scanpy PlotEmbed"];
  30 -->|output_h5ad| 32;
  33["Scanpy PlotEmbed"];
  30 -->|output_h5ad| 33;
  34["Manipulate AnnData"];
  30 -->|output_h5ad| 34;
  35["Scanpy PlotEmbed"];
  30 -->|output_h5ad| 35;
  36["Inspect AnnData"];
  30 -->|output_h5ad| 36;
  37["AnnData Operations"];
  34 -->|anndata| 37;
  30 -->|output_h5ad| 37;
  38["Join two Datasets"];
  30 -->|output_tsv| 38;
  36 -->|var| 38;
  39["Join two Datasets"];
  31 -->|output_tsv| 39;
  36 -->|var| 39;
  40["AnnData Operations"];
  37 -->|output_h5ad| 40;
  10bd70f8-ffcb-442b-9647-e5b947b6d35e["Output\nFinal cell annotated object"];
  40 --> 10bd70f8-ffcb-442b-9647-e5b947b6d35e;
  style 10bd70f8-ffcb-442b-9647-e5b947b6d35e stroke:#2c3143,stroke-width:4px;
  41["Cut"];
  38 -->|out_file1| 41;
  4f822c1c-91c5-4be4-8f9b-d5bdda0a037e["Output\nMarkers - cluster - named"];
  41 --> 4f822c1c-91c5-4be4-8f9b-d5bdda0a037e;
  style 4f822c1c-91c5-4be4-8f9b-d5bdda0a037e stroke:#2c3143,stroke-width:4px;
  42["Cut"];
  39 -->|out_file1| 42;
  3b471f3d-263d-4299-9b7d-8a8ae1aa556e["Output\nMarkers - genotype - named"];
  42 --> 3b471f3d-263d-4299-9b7d-8a8ae1aa556e;
  style 3b471f3d-263d-4299-9b7d-8a8ae1aa556e stroke:#2c3143,stroke-width:4px;
  43["Scanpy PlotEmbed"];
  40 -->|output_h5ad| 43;</pre>
			<h2>Inputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Input</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>Input dataset</td>
					<td>Mito-counted AnnData</td>
				</tr>
				
				</tbody>
			</table>

			<h2>Outputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>From</th>
						<th>Output</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_cells/scanpy_filter_cells/1.8.1+galaxy9</td>
					<td>Scanpy FilterCells</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_cells/scanpy_filter_cells/1.8.1+galaxy9</td>
					<td>Scanpy FilterCells</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_cells/scanpy_filter_cells/1.8.1+galaxy9</td>
					<td>Scanpy FilterCells</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_genes/scanpy_filter_genes/1.8.1+galaxy9</td>
					<td>Scanpy FilterGenes</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_normalise_data/scanpy_normalise_data/1.8.1+galaxy9</td>
					<td>Scanpy NormaliseData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_variable_genes/scanpy_find_variable_genes/1.8.1+galaxy9</td>
					<td>Scanpy FindVariableGenes</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_scale_data/scanpy_scale_data/1.8.1+galaxy9</td>
					<td>Scanpy ScaleData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_pca/scanpy_run_pca/1.8.1+galaxy9</td>
					<td>Scanpy RunPCA</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</td>
					<td>Plot</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_compute_graph/scanpy_compute_graph/1.8.1+galaxy9</td>
					<td>Scanpy ComputeGraph</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_tsne/scanpy_run_tsne/1.8.1+galaxy9</td>
					<td>Scanpy RunTSNE</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_umap/scanpy_run_umap/1.8.1+galaxy9</td>
					<td>Scanpy RunUMAP</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_cluster/scanpy_find_cluster/1.8.1+galaxy9</td>
					<td>Scanpy FindCluster</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_markers/scanpy_find_markers/1.8.1+galaxy9</td>
					<td>Scanpy FindMarkers</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_markers/scanpy_find_markers/1.8.1+galaxy9</td>
					<td>Scanpy FindMarkers</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed/1.8.1+galaxy9</td>
					<td>Scanpy PlotEmbed</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed/1.8.1+galaxy9</td>
					<td>Scanpy PlotEmbed</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.7.5+galaxy1</td>
					<td>Manipulate AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed/1.8.1+galaxy9</td>
					<td>Scanpy PlotEmbed</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.8.1+galaxy91</td>
					<td>AnnData Operations</td>
					<td></td>
				</tr>
				
				<tr>
					<td>join1</td>
					<td>Join two Datasets</td>
					<td></td>
				</tr>
				
				<tr>
					<td>join1</td>
					<td>Join two Datasets</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.8.1+galaxy91</td>
					<td>AnnData Operations</td>
					<td></td>
				</tr>
				
				<tr>
					<td>Cut1</td>
					<td>Cut</td>
					<td></td>
				</tr>
				
				<tr>
					<td>Cut1</td>
					<td>Cut</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed/1.8.1+galaxy9</td>
					<td>Scanpy PlotEmbed</td>
					<td></td>
				</tr>
				
				</tbody>
			</table>


			<h2>Tools</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Tool</th>
						<th>Links</th>
					</tr>
				</thead>
				<tbody>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Cut1">Cut1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=join1">join1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.8.1+galaxy91">toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.8.1+galaxy91</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/anndata_ops/7ebc22f77d86">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_compute_graph/scanpy_compute_graph/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_compute_graph/scanpy_compute_graph/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_compute_graph/6417cccad133">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_cells/scanpy_filter_cells/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_cells/scanpy_filter_cells/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_filter_cells/e9283529cba1">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_genes/scanpy_filter_genes/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_filter_genes/scanpy_filter_genes/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_filter_genes/474a69c47184">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_cluster/scanpy_find_cluster/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_cluster/scanpy_find_cluster/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_cluster/2ccd9f9e2cd0">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_markers/scanpy_find_markers/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_markers/scanpy_find_markers/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_markers/81c34e67f262">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_variable_genes/scanpy_find_variable_genes/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_find_variable_genes/scanpy_find_variable_genes/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_find_variable_genes/2861ca3bfe4d">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_normalise_data/scanpy_normalise_data/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_normalise_data/scanpy_normalise_data/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_normalise_data/6b97ffba31da">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_plot_embed/scanpy_plot_embed/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_plot_embed/35f6611fb8c3">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_pca/scanpy_run_pca/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_pca/scanpy_run_pca/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_pca/59028945c857">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_tsne/scanpy_run_tsne/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_tsne/scanpy_run_tsne/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_tsne/181d61d2f3cd">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_umap/scanpy_run_umap/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_run_umap/scanpy_run_umap/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_run_umap/bd4a244faeb8">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_scale_data/scanpy_scale_data/1.8.1+galaxy9">toolshed.g2.bx.psu.edu/repos/ebi-gxa/scanpy_scale_data/scanpy_scale_data/1.8.1+galaxy9</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/scanpy_scale_data/f86b2c3bc459">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/anndata_inspect/ee98d611afc6">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.7.5+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.7.5+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/anndata_manipulate/3d748954434b">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.7.1+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/scanpy_plot/aa0c474463c2">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
				</tbody>
			</table>


			<p>To use these workflows in Galaxy you can either click the links to download the workflows, or you can right-click and copy the link to the workflow which can be used in the Galaxy form to import workflows.</p>


			<h2>Importing into Galaxy</h2>
			Below are the instructions for importing these workflows directly into your Galaxy server of choice to start using them!

			<!--SNIPPET--><blockquote class="hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-a-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing a workflow</div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote> <!--END_SNIPPET-->

			<h2>Version History</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Version</th>
						<th>Commit</th>
						<th>Time</th>
						<th>Comments</th>
					</tr>
				</thead>
				<tbody>
					
					<tr>
						<td>1</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/9a3fc8a6dad6c8d1b7b9897056cc75af00a12c57">9a3fc8a6d</a></td>
						<td>2023-06-09 21:11:27</td>
						<td>new workflow tests</td>
					</tr>
					
				</tbody>
			</table>


			<h2>For Admins</h2>
			<h3>Installing the workflow tools</h3>

			<pre>
wget https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.ga -O workflow.ga
workflow-to-tools -w workflow.ga -o tools.yaml
shed-tools install -g GALAXY -a API_KEY -t tools.yaml
workflow-install -g GALAXY -a API_KEY -w workflow.ga --publish-workflows
			</pre>
		</div>
	</div>
</section>


</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:W00197">gxy.io/GTN:W00197</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scrna-case_basic-pipeline/workflows/Filter,-Plot-and-Explore-Single-cell-RNA-seq-Data-updated.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/3b04f69d0ed7c2b1f7b1522c736ccfcd38f375c3">3b04f69</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	
	<script src="/training-material/assets/js/mermaid@10.6.1.js" async></script>
	<link rel="preload" href="/training-material/assets/js/mermaid@10.6.1.js" as="script" crossorigin>
	

    </body>
</html>
