<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Workflow - Standard processing of 10X single cell ATAC-seq data with SnapATAC2 / Single-cell ATAC-seq standard processing with SnapATAC2 / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@9f4997722c8fd4cb7125acf3ba7c6fa375e367a0",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Workflow - Standard processing of 10X single cell ATAC-seq data with SnapATAC2">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="1970-01-01 00:00:00 +0000"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Single-cell ATAC-seq standard processing with SnapATAC2 / Workflow - Standard processing of 10X single cell ATAC-seq data with SnapATAC2">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!). When you generate your lovely gene lists for your cells, consider checking out our Transcriptomics tutorials for further network analysis!  What else do you want to see? You can submit tool and/or tutorial request on our  [Single Cell Community Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.png">
	</head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<style>
hgroup {margin-bottom: 2rem; border-bottom: 1px solid var(--border-light);}
hgroup p { font-family: monospace; font-size: 0.8rem;}
.meta ul { padding-left: 1em;}
</style>

<section>
	<hgroup>
		<h1 data-toc-skip>Workflow - Standard processing of 10X single cell ATAC-seq data with SnapATAC2</h1>
		<p>single-cell-scatac-standard-processing-snapatac2/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2</p>
	</hgroup>
	
	<div class="row">
		<div class="col-md-3 meta">
			
			<b>Author(s)</b>
			<br/>
			
			Timon Schlegel
			
			
			<br/>

			<i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span>
			<b>Version</b>
			<br/>
			2

			<br/>

			<i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span>
			<b>Last updated</b>
			<br/>
			Aug 9, 2024

			<br/>

			<i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">license</span>
			<b>License</b>
			<br/>

			
			<a href="https://spdx.org/licenses/CC-BY-4.0">CC-BY-4.0</a>
			
			<br/>

			
			<i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>
			<b>Tags</b>
			<br/>

			
			<div class="label label-default tutorial_tag" style="--color-primary: #e4ceff; --color-darker: #d6afff; --color-dimmed: #ddbfff;">scATAC-seq</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #a0ed00; --color-darker: #8ed300; --color-dimmed: #97e000;">epigenetics</div>
			
			</br>
			


			<b>Features</b>
			<ul>
			
			
				<li>Includes a <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/workflows_report_view.html">Galaxy Workflow Report</a></li>
			
			
				<li>Uses <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/workflows_comments.html">Galaxy Workflow Comments</a></li>
			
			
			</ul>

			<hr/>


			<b>Tutorial</b>
			<br/>
			<i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> <a href="../tutorial.html">Single-cell ATAC-seq standard processing with SnapATAC2</a>
			</br>

			
			


			<hr/>
			<b>Workflow Testing</b>
			<br/>

			Tests: ✅ <br/>
			Results: Not yet automated

			<hr/>

			<b>FAIRness</b>

			<i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span>
			<b><abbr title="Persistent URL">PURL</abbr></b>
			<br/>
			<a href="https://gxy.io/GTN:W00274">https://gxy.io/GTN:W00274</a>
			<br/>

			
			<a style="margin-bottom: 0.2rem" href="/training-material/api/workflows/single-cell/scatac-standard-processing-snapatac2/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2/rocrate.zip" class="btn btn-secondary"><img src="/training-material/assets/images/ro-crate.png" style="height: 32px" alt="RO-Crate logo with flask"/> Download Workflow RO-Crate</a>
			

			
			<a style="margin-bottom: 0.2rem" href="https://workflowhub.eu/workflows/1343" class="btn btn-secondary"><img src="/training-material/assets/images/workflowhub-icon.svg" style="height: 32px" alt="Workflowhub cloud with gears logo"/> View on WorkflowHub</a>
			

		</div>
		<div class="col-md-9">
			<span tabindex="0" role="button" class="workflow show-when-galaxy-proxy-active" data-workflow="https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scatac-standard-processing-snapatac2/versions/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">Launch in Tutorial Mode</span>
			<a href="/training-material/faqs/galaxy/tutorial_mode.html" class="show-when-galaxy-proxy-active"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span></a>
			<br/>

			<span class="dropdown">
				<button class="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown" aria-expanded="false">
					<img src="/training-material/assets/images/galaxylogo.png" style="height: 16px" alt="galaxy logo"/> 
					Run Workflow in Galaxy
				</button>
				<ul class="dropdown-menu">
					
					
					<li class="dropdown-item"><a href="https://usegalaxy.org.au/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scatac-standard-processing-snapatac2/versions/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">Galaxy Australia</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.fr/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scatac-standard-processing-snapatac2/versions/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">Galaxy France</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.eu/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scatac-standard-processing-snapatac2/versions/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">Galaxy Europe</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.org/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scatac-standard-processing-snapatac2/versions/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">Galaxy Main</a></li>
					
					<li class="dropdown-item hide-when-galaxy-proxy-active">
						<a href="https://my.galaxy.training/?path=/workflows/trs_import%3Frun_form=true%26trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scatac-standard-processing-snapatac2/versions/standard-processing-of-10x-single-cell-atac-seq-data-with-snapatac2">Import to another server (≥23.0+ only!)</a>
					</li>
				</ul>
			</span>
			<a class="btn btn-primary" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.ga">
				<i class="fas fa-download" aria-hidden="true"></i><span class="visually-hidden">galaxy-download</span>
				Download
			</a>
			<pre class="mermaid">flowchart TD
  0["ℹ️ Input Dataset\nFragment_file"];
  style 0 stroke:#2c3143,stroke-width:4px;
  1["ℹ️ Input Dataset\nchromosome_sizes.tabular"];
  style 1 stroke:#2c3143,stroke-width:4px;
  2["ℹ️ Input Dataset\ngene_annotation"];
  style 2 stroke:#2c3143,stroke-width:4px;
  3["ℹ️ Input Dataset\nBam-file"];
  style 3 stroke:#2c3143,stroke-width:4px;
  4["ℹ️ Input Parameter\nKeys for annotations of obs/cells or vars/genes"];
  style 4 fill:#ded,stroke:#393,stroke-width:4px;
  5["ℹ️ Input Dataset\nReplace_file"];
  style 5 stroke:#2c3143,stroke-width:4px;
  6["pp.import_data"];
  1 -->|output| 6;
  0 -->|output| 6;
  7["pp.make_fragment_file"];
  3 -->|output| 7;
  8["pl.frag_size_distr"];
  6 -->|anndata_out| 8;
  265d6020-44e2-408c-9eb6-02cbc9ac23a7["Output\nplot frag_size"];
  8 --> 265d6020-44e2-408c-9eb6-02cbc9ac23a7;
  style 265d6020-44e2-408c-9eb6-02cbc9ac23a7 stroke:#2c3143,stroke-width:4px;
  9["pl.frag_size_distr_log"];
  6 -->|anndata_out| 9;
  a37cd3c3-e07e-4bea-899f-ee44cee8370b["Output\nplot log frag_size"];
  9 --> a37cd3c3-e07e-4bea-899f-ee44cee8370b;
  style a37cd3c3-e07e-4bea-899f-ee44cee8370b stroke:#2c3143,stroke-width:4px;
  10["metrics.tsse"];
  6 -->|anndata_out| 10;
  2 -->|output| 10;
  233284e5-c184-4330-a1d3-2d03951c4ce7["Output\nanndata tsse"];
  10 --> 233284e5-c184-4330-a1d3-2d03951c4ce7;
  style 233284e5-c184-4330-a1d3-2d03951c4ce7 stroke:#2c3143,stroke-width:4px;
  11["pp.import_data-sorted_by_barcodes"];
  1 -->|output| 11;
  7 -->|fragments_out| 11;
  50ef9fe7-bb08-4957-bc45-079ed78f9ee8["Output\nanndata"];
  11 --> 50ef9fe7-bb08-4957-bc45-079ed78f9ee8;
  style 50ef9fe7-bb08-4957-bc45-079ed78f9ee8 stroke:#2c3143,stroke-width:4px;
  12["pl.tsse"];
  10 -->|anndata_out| 12;
  9a5c0f9c-0423-4822-871e-ba8317f2fa7b["Output\nplot tsse"];
  12 --> 9a5c0f9c-0423-4822-871e-ba8317f2fa7b;
  style 9a5c0f9c-0423-4822-871e-ba8317f2fa7b stroke:#2c3143,stroke-width:4px;
  13["pp.filter_cells"];
  10 -->|anndata_out| 13;
  b58dc4ad-6bf0-4f31-99f4-4f8deb13427c["Output\nanndata filter cells"];
  13 --> b58dc4ad-6bf0-4f31-99f4-4f8deb13427c;
  style b58dc4ad-6bf0-4f31-99f4-4f8deb13427c stroke:#2c3143,stroke-width:4px;
  14["pp.add_tile_matrix"];
  13 -->|anndata_out| 14;
  d4b64fc4-fd35-4781-b799-0d07fe8fe438["Output\nanndata tile matrix"];
  14 --> d4b64fc4-fd35-4781-b799-0d07fe8fe438;
  style d4b64fc4-fd35-4781-b799-0d07fe8fe438 stroke:#2c3143,stroke-width:4px;
  15["pp.select_features"];
  14 -->|anndata_out| 15;
  f531f6bd-3fc6-48b9-bb33-66de7928e670["Output\nanndata select features"];
  15 --> f531f6bd-3fc6-48b9-bb33-66de7928e670;
  style f531f6bd-3fc6-48b9-bb33-66de7928e670 stroke:#2c3143,stroke-width:4px;
  16["pp.scrublet"];
  15 -->|anndata_out| 16;
  5b17aa3a-3694-440f-9022-cf9761feb760["Output\nanndata scrublet"];
  16 --> 5b17aa3a-3694-440f-9022-cf9761feb760;
  style 5b17aa3a-3694-440f-9022-cf9761feb760 stroke:#2c3143,stroke-width:4px;
  17["pp.filter_doublets"];
  16 -->|anndata_out| 17;
  3838cf9d-a018-4aa7-8852-893b2f60ec2a["Output\nanndata filter doublets"];
  17 --> 3838cf9d-a018-4aa7-8852-893b2f60ec2a;
  style 3838cf9d-a018-4aa7-8852-893b2f60ec2a stroke:#2c3143,stroke-width:4px;
  18["tl.spectral"];
  17 -->|anndata_out| 18;
  77883516-8ac4-40b5-8b75-5750639d9fc4["Output\nanndata spectral"];
  18 --> 77883516-8ac4-40b5-8b75-5750639d9fc4;
  style 77883516-8ac4-40b5-8b75-5750639d9fc4 stroke:#2c3143,stroke-width:4px;
  19["tl.umap"];
  18 -->|anndata_out| 19;
  062a5471-7b7a-4d9b-bf6c-fe9bbbc5d2ad["Output\nanndata umap"];
  19 --> 062a5471-7b7a-4d9b-bf6c-fe9bbbc5d2ad;
  style 062a5471-7b7a-4d9b-bf6c-fe9bbbc5d2ad stroke:#2c3143,stroke-width:4px;
  20["pp.knn"];
  19 -->|anndata_out| 20;
  b7ebe580-9938-48e5-8718-4df197126615["Output\nanndata knn"];
  20 --> b7ebe580-9938-48e5-8718-4df197126615;
  style b7ebe580-9938-48e5-8718-4df197126615 stroke:#2c3143,stroke-width:4px;
  21["tl.leiden"];
  20 -->|anndata_out| 21;
  9970eb96-013b-4d97-bbbf-2d1f4663b73c["Output\nanndata_leiden_clustering"];
  21 --> 9970eb96-013b-4d97-bbbf-2d1f4663b73c;
  style 9970eb96-013b-4d97-bbbf-2d1f4663b73c stroke:#2c3143,stroke-width:4px;
  22["pl.umap"];
  21 -->|anndata_out| 22;
  4fff8ec8-7d6f-4871-9e42-d6c1791d6aa9["Output\numap_leiden-clusters"];
  22 --> 4fff8ec8-7d6f-4871-9e42-d6c1791d6aa9;
  style 4fff8ec8-7d6f-4871-9e42-d6c1791d6aa9 stroke:#2c3143,stroke-width:4px;
  23["make_gene_matrix"];
  21 -->|anndata_out| 23;
  2 -->|output| 23;
  9a5c2e23-5b59-41ae-aceb-0e47ce9e7b1a["Output\nanndata gene matrix"];
  23 --> 9a5c2e23-5b59-41ae-aceb-0e47ce9e7b1a;
  style 9a5c2e23-5b59-41ae-aceb-0e47ce9e7b1a stroke:#2c3143,stroke-width:4px;
  24["scanpy_filter_genes"];
  23 -->|anndata_out| 24;
  7a9b22df-e0e5-49d9-9fa1-2efe316d6d66["Output\nanndata filter genes"];
  24 --> 7a9b22df-e0e5-49d9-9fa1-2efe316d6d66;
  style 7a9b22df-e0e5-49d9-9fa1-2efe316d6d66 stroke:#2c3143,stroke-width:4px;
  25["Normalize"];
  24 -->|anndata_out| 25;
  7dddd4c8-1944-4f50-b331-857d03cc9631["Output\nanndata normalize"];
  25 --> 7dddd4c8-1944-4f50-b331-857d03cc9631;
  style 7dddd4c8-1944-4f50-b331-857d03cc9631 stroke:#2c3143,stroke-width:4px;
  26["pp.log1p"];
  25 -->|anndata_out| 26;
  40ace30a-21e1-4bf2-8e08-ad93b62fb2ee["Output\nanndata log1p"];
  26 --> 40ace30a-21e1-4bf2-8e08-ad93b62fb2ee;
  style 40ace30a-21e1-4bf2-8e08-ad93b62fb2ee stroke:#2c3143,stroke-width:4px;
  27["external.pp.magic"];
  26 -->|anndata_out| 27;
  8c1c7aa9-b9dd-467b-9715-73773aea093f["Output\nanndata_magic"];
  27 --> 8c1c7aa9-b9dd-467b-9715-73773aea093f;
  style 8c1c7aa9-b9dd-467b-9715-73773aea093f stroke:#2c3143,stroke-width:4px;
  28["Copy obsm"];
  21 -->|anndata_out| 28;
  27 -->|anndata_out| 28;
  2c3825d3-7151-4eef-97d0-c8a31ca6fd3e["Output\nanndata_gene-matrix_leiden"];
  28 --> 2c3825d3-7151-4eef-97d0-c8a31ca6fd3e;
  style 2c3825d3-7151-4eef-97d0-c8a31ca6fd3e stroke:#2c3143,stroke-width:4px;
  29["umap_plot_with_scanpy"];
  28 -->|output_h5ad| 29;
  4 -->|output| 29;
  4cfbbb70-156b-48d7-b14b-db21198d9b6d["Output\numap_marker-genes"];
  29 --> 4cfbbb70-156b-48d7-b14b-db21198d9b6d;
  style 4cfbbb70-156b-48d7-b14b-db21198d9b6d stroke:#2c3143,stroke-width:4px;
  30["Inspect observations"];
  28 -->|output_h5ad| 30;
  31["Cut leiden from table"];
  30 -->|obs| 31;
  11c22200-a54e-4585-82c4-8457dbbfb53c["Output\nleiden annotation"];
  31 --> 11c22200-a54e-4585-82c4-8457dbbfb53c;
  style 11c22200-a54e-4585-82c4-8457dbbfb53c stroke:#2c3143,stroke-width:4px;
  32["Replace leiden"];
  31 -->|out_file1| 32;
  5 -->|output| 32;
  467de061-79e6-4148-8a65-c06383a14012["Output\ncell type annotation"];
  32 --> 467de061-79e6-4148-8a65-c06383a14012;
  style 467de061-79e6-4148-8a65-c06383a14012 stroke:#2c3143,stroke-width:4px;
  33["Manipulate AnnData"];
  28 -->|output_h5ad| 33;
  32 -->|outfile_replace| 33;
  cbba6173-f439-4fd5-8423-946aa625112b["Output\nanndata_cell_type"];
  33 --> cbba6173-f439-4fd5-8423-946aa625112b;
  style cbba6173-f439-4fd5-8423-946aa625112b stroke:#2c3143,stroke-width:4px;
  34["Plot cell types"];
  33 -->|anndata| 34;
  c28c53ae-8ee7-47d7-8a00-352908bb7c1a["Output\numap_cell-type"];
  34 --> c28c53ae-8ee7-47d7-8a00-352908bb7c1a;
  style c28c53ae-8ee7-47d7-8a00-352908bb7c1a stroke:#2c3143,stroke-width:4px;
  35["Final Anndata general info"];
  33 -->|anndata| 35;
  379a85d1-a204-4b0b-8b33-86e5582fc51f["Output\ngeneral"];
  35 --> 379a85d1-a204-4b0b-8b33-86e5582fc51f;
  style 379a85d1-a204-4b0b-8b33-86e5582fc51f stroke:#2c3143,stroke-width:4px;</pre>
			<h2>Inputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Input</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>Input dataset</td>
					<td>Fragment_file</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>chromosome_sizes.tabular</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>gene_annotation</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>Bam-file</td>
				</tr>
				
				<tr>
					<td>Input parameter</td>
					<td>Keys for annotations of obs/cells or vars/genes</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>Replace_file</td>
				</tr>
				
				</tbody>
			</table>

			<h2>Outputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>From</th>
						<th>Output</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td></td>
					<td>Input parameter</td>
					<td>Keys for annotations of obs/cells or vars/genes</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1</td>
					<td>SnapATAC2 Plotting</td>
					<td>pl.frag_size_distr</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1</td>
					<td>SnapATAC2 Plotting</td>
					<td>pl.frag_size_distr_log</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>metrics.tsse</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>pp.import_data-sorted_by_barcodes</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1</td>
					<td>SnapATAC2 Plotting</td>
					<td>pl.tsse</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>pp.filter_cells</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>pp.add_tile_matrix</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>pp.select_features</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>pp.scrublet</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>pp.filter_doublets</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1</td>
					<td>SnapATAC2 Clustering</td>
					<td>tl.spectral</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1</td>
					<td>SnapATAC2 Clustering</td>
					<td>tl.umap</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1</td>
					<td>SnapATAC2 Clustering</td>
					<td>pp.knn</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1</td>
					<td>SnapATAC2 Clustering</td>
					<td>tl.leiden</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1</td>
					<td>SnapATAC2 Plotting</td>
					<td>pl.umap</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</td>
					<td>SnapATAC2 Preprocessing</td>
					<td>make_gene_matrix</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.9.6+galaxy3</td>
					<td>Filter</td>
					<td>scanpy_filter_genes</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.9.6+galaxy3</td>
					<td>Normalize</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.9.6+galaxy3</td>
					<td>Inspect and manipulate</td>
					<td>pp.log1p</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.9.6+galaxy3</td>
					<td>Normalize</td>
					<td>external.pp.magic</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.9.3+galaxy0</td>
					<td>AnnData Operations</td>
					<td>Copy obsm</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.9.6+galaxy3</td>
					<td>Plot</td>
					<td>umap_plot_with_scanpy</td>
				</tr>
				
				<tr>
					<td>Cut1</td>
					<td>Cut</td>
					<td>Cut leiden from table</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file/0.2</td>
					<td>Replace column</td>
					<td>Replace leiden</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.3+galaxy0</td>
					<td>Manipulate AnnData</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.9.6+galaxy3</td>
					<td>Plot</td>
					<td>Plot cell types</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0</td>
					<td>Inspect AnnData</td>
					<td>Final Anndata general info</td>
				</tr>
				
				</tbody>
			</table>


			<h2>Tools</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Tool</th>
						<th>Links</th>
					</tr>
				</thead>
				<tbody>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Cut1">Cut1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file/0.2">toolshed.g2.bx.psu.edu/repos/bgruening/replace_column_by_key_value_file/replace_column_with_key_value_file/0.2</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/bgruening/replace_column_by_key_value_file/d533e4b75800">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.9.3+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/anndata_ops/anndata_ops/1.9.3+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/anndata_ops/a4774b7b2e85">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0">toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.10.3+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/anndata_inspect/6f0d0c784f09">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.3+galaxy0">toolshed.g2.bx.psu.edu/repos/iuc/anndata_manipulate/anndata_manipulate/0.10.3+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/anndata_manipulate/ed4996a16f7f">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.9.6+galaxy3">toolshed.g2.bx.psu.edu/repos/iuc/scanpy_filter/scanpy_filter/1.9.6+galaxy3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/scanpy_filter/d636ce5cde16">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.9.6+galaxy3">toolshed.g2.bx.psu.edu/repos/iuc/scanpy_inspect/scanpy_inspect/1.9.6+galaxy3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/scanpy_inspect/f54f0f0598ad">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.9.6+galaxy3">toolshed.g2.bx.psu.edu/repos/iuc/scanpy_normalize/scanpy_normalize/1.9.6+galaxy3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/scanpy_normalize/d844935c906c">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.9.6+galaxy3">toolshed.g2.bx.psu.edu/repos/iuc/scanpy_plot/scanpy_plot/1.9.6+galaxy3</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/scanpy_plot/f7446fd8d5e7">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_clustering/snapatac2_clustering/2.6.4+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/snapatac2_clustering/afb90d3485df">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_plotting/snapatac2_plotting/2.6.4+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/snapatac2_plotting/370d45e0d1a8">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/snapatac2_preprocessing/snapatac2_preprocessing/2.6.4+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/snapatac2_preprocessing/fef8f61cb34c">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
				</tbody>
			</table>


			<p>To use these workflows in Galaxy you can either click the links to download the workflows, or you can right-click and copy the link to the workflow which can be used in the Galaxy form to import workflows.</p>


			<h2>Importing into Galaxy</h2>
			Below are the instructions for importing these workflows directly into your Galaxy server of choice to start using them!

			<!--SNIPPET--><blockquote class="hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-a-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing a workflow</div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote> <!--END_SNIPPET-->

			<h2>Version History</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Version</th>
						<th>Commit</th>
						<th>Time</th>
						<th>Comments</th>
					</tr>
				</thead>
				<tbody>
					
					<tr>
						<td>2</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/747cf76eb0ac7e1a046715c274e302d31ea720f5">747cf76eb</a></td>
						<td>2024-08-08 16:59:41</td>
						<td>updated snapatac2 version to 2.6.4</td>
					</tr>
					
					<tr>
						<td>1</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/9740cbbaddce4bd770c90ec91e8990d0cddf3bc8">9740cbbad</a></td>
						<td>2024-07-11 13:29:51</td>
						<td>Update workflow and add tests</td>
					</tr>
					
				</tbody>
			</table>


			<h2>For Admins</h2>
			<h3>Installing the workflow tools</h3>

			<pre>
wget https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.ga -O workflow.ga
workflow-to-tools -w workflow.ga -o tools.yaml
shed-tools install -g GALAXY -a API_KEY -t tools.yaml
workflow-install -g GALAXY -a API_KEY -w workflow.ga --publish-workflows
			</pre>
		</div>
	</div>
</section>


</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:W00274">gxy.io/GTN:W00274</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scatac-standard-processing-snapatac2/workflows/Standard-processing-of-10X-single-cell-ATAC-seq-data-with-SnapATAC2.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/9f4997722c8fd4cb7125acf3ba7c6fa375e367a0">9f49977</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	
	<script src="/training-material/assets/js/mermaid@10.6.1.js" async></script>
	<link rel="preload" href="/training-material/assets/js/mermaid@10.6.1.js" as="script" crossorigin>
	

    </body>
</html>
