<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Identification of the binding sites of the Estrogen receptor</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:title" content="Galaxy Training: Identification of the binding sites of the Estrogen receptor" />
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Epigenetics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Festrogen-receptor-binding-site-identification%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Festrogen-receptor-binding-site-identification%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Festrogen-receptor-binding-site-identification%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Festrogen-receptor-binding-site-identification%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Festrogen-receptor-binding-site-identification%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Festrogen-receptor-binding-site-identification%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Identification of the binding sites of the Estrogen receptor",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
  },
  "courseCode": "epigenetics / estrogen-receptor-binding-site-identification / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Identification of the binding sites of the Estrogen receptor' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html",
  "timeRequired": "PT3H",
  "keywords": "ChIP-seq",
  "description": "The questions this  addresses are:\n - How is raw ChIP-seq data processed and analyzed?\n - What are the binding sites of the Estrogen receptor?\n\n\\nThe objectives are:\n - Inspect read quality with FastQC\n - Map reads with Bowtie2\n - Assess the quality of an ChIP-seq experiments\n - Extract coverage files\n - Call enriched regions or peaks\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Friederike D√ºndar"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Vivek Bhardwaj"
    },
    {
      "@type": "Person",
      "name": "Fidel Ramirez"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Friederike D√ºndar"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Vivek Bhardwaj"
    },
    {
      "@type": "Person",
      "name": "Fidel Ramirez"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Identification of the binding sites of the Estrogen receptor</h1>
        
        <div class="alert alert-warning" role="alert">
            <h4 class="alert-heading">Under Development!</h4>
            <p>This tutorial is not in its final state. The content may change a lot in the next months.
            Because of this status, it is also not listed in the topic pages.</p>
        </div>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/friedue/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/friedue" alt="Friederike D√ºndar">Friederike D√ºndar</a>, <a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/erxleben" alt="Anika Erxleben">Anika Erxleben</a>, <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="B√©r√©nice Batut">B√©r√©nice Batut</a>, <a href="/training-material/hall-of-fame/vivekbhr/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/vivekbhr" alt="Vivek Bhardwaj">Vivek Bhardwaj</a>, <a href="/training-material/hall-of-fame/fidelram/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/fidelram" alt="Fidel Ramirez">Fidel Ramirez</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How is raw ChIP-seq data processed and analyzed?</p>
</li>
            
            <li><p>What are the binding sites of the Estrogen receptor?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Inspect read quality with FastQC</p>
</li>
            
            <li><p>Map reads with Bowtie2</p>
</li>
            
            <li><p>Assess the quality of an ChIP-seq experiments</p>
</li>
            
            <li><p>Extract coverage files</p>
</li>
            
            <li><p>Call enriched regions or peaks</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/epigenetics/slides/introduction-dna-methylation.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                

                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>This exercise uses the dataset from the Nature publication by <a href="https://www.ncbi.nlm.nih.gov/pubmed/22217937">Ross-Inness et al., 2012</a>.
The goal of this article was to identify the binding sites of the Estrogen receptor, a transcription factor known to be associated with different types of breast cancer.</p>

<p>To this end, ChIP-seq was performed in breast cancer cells from 4 patients of different outcomes (good and poor). For each ChIP-seq experiment there is a matching technical control, i.e. there are 8 samples in total:</p>

<table>
  <thead>
    <tr>
      <th>Patient</th>
      <th>Outcome</th>
      <th>Treatment</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Patient 1</td>
      <td>Good</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 1</td>
      <td>Good</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
    <tr>
      <td>Patient 2</td>
      <td>Good</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 2</td>
      <td>Good</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
    <tr>
      <td>Patient 3</td>
      <td>Poor</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 3</td>
      <td>Poor</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
    <tr>
      <td>Patient 4</td>
      <td>Poor</td>
      <td>ChIP ER</td>
    </tr>
    <tr>
      <td>Patient 4</td>
      <td>Poor</td>
      <td>input (no immunoprecipitation step)</td>
    </tr>
  </tbody>
</table>

<p>Half of which are the so-called ‚Äòinput‚Äô samples for which the same treatment as the ChIP-seq samples was done except for the immunoprecipitation step.
The input files are used to identify potential sequencing bias, like open chromatin or GC bias.</p>

<p>Because of the long processing time for the large original files, we have downsampled the original data and provide already processed data for subsequent steps.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

<ol id="markdown-toc">
  <li><a href="#step-1-quality-control-and-treatment-of-the-sequences" id="markdown-toc-step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</a></li>
  <li><a href="#step-2-mapping-of-the-reads" id="markdown-toc-step-2-mapping-of-the-reads">Step 2: <span class="notranslate">Mapping</span> of the <span class="notranslate">reads</span></a>    <ol>
      <li><a href="#visualization-using-a-genome-browser" id="markdown-toc-visualization-using-a-genome-browser">Visualization using a Genome Browser</a></li>
      <li><a href="#inspection-of-the-sam-format" id="markdown-toc-inspection-of-the-sam-format">Inspection of the SAM format</a></li>
    </ol>
  </li>
  <li><a href="#step-3-chip-seq-quality-control" id="markdown-toc-step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</a>    <ol>
      <li><a href="#correlation-between-samples" id="markdown-toc-correlation-between-samples">Correlation between samples</a></li>
      <li><a href="#ip-strength-estimation" id="markdown-toc-ip-strength-estimation">IP strength estimation</a></li>
    </ol>
  </li>
  <li><a href="#step-4-normalization" id="markdown-toc-step-4-normalization">Step 4: Normalization</a>    <ol>
      <li><a href="#generation-of-coverage-files-normalized-by-sequencing-depth" id="markdown-toc-generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</a></li>
      <li><a href="#generation-of-input-normalized-coverage-files-and-their-visualization" id="markdown-toc-generation-of-input-normalized-coverage-files-and-their-visualization">Generation of input-normalized coverage files and their visualization</a></li>
    </ol>
  </li>
  <li><a href="#step-5-detecting-enriched-regions-peak-calling" id="markdown-toc-step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</a></li>
  <li><a href="#step-6-plot-the-signal-on-the-peaks-between-samples" id="markdown-toc-step-6-plot-the-signal-on-the-peaks-between-samples">Step 6: Plot the signal on the peaks between samples</a>    <ol>
      <li><a href="#computematrix" id="markdown-toc-computematrix">computeMatrix</a></li>
      <li><a href="#plotheatmap" id="markdown-toc-plotheatmap">plotHeatmap</a></li>
    </ol>
  </li>
  <li><a href="#additional-exercise-if-you-have-finished-all-above" id="markdown-toc-additional-exercise-if-you-have-finished-all-above">Additional exercise (if you have finished all above)</a>    <ol>
      <li><a href="#additional-quality-control--gc-bias-assessment" id="markdown-toc-additional-quality-control--gc-bias-assessment">Additional Quality control : GC bias assessment</a></li>
      <li><a href="#plotting-heatmap-from-multiple-samples-with-clustering" id="markdown-toc-plotting-heatmap-from-multiple-samples-with-clustering">Plotting heatmap from multiple samples with clustering</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</h1>

<p>The first step of any ChIP-Seq data analysis is quality control of the raw sequencing data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> from <a href="https://zenodo.org/record/892432/files/patient1_input_good_outcome.fastq">Zenodo</a> or from the data library into the history</p>

      <blockquote class="tip">
        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field</li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
          <li>Click on the <code class="language-plaintext highlighter-rouge">pencil</code> icon once the file is imported</li>
          <li>Click on <strong>Datatype</strong> in the central panel</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger</code> as <strong>New Type</strong></li>
        </ul>
      </blockquote>

      <blockquote class="tip">
        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <ul>
          <li>Go into ‚ÄúShared data‚Äù (top panel) then ‚ÄúData libraries‚Äù</li>
          <li>Click on ‚ÄúTraining data‚Äù and then ‚ÄúAnalyses of ChIP-Seq data‚Äù</li>
          <li>Select interesting file</li>
          <li>Click on ‚ÄúImport selected datasets into history‚Äù</li>
          <li>Import in the history</li>
        </ul>
      </blockquote>

      <p>As default, <span class="notranslate">Galaxy</span> takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Inspect the file by clicking on the <code class="language-plaintext highlighter-rouge">eye</code> icon</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How are the <span class="notranslate">DNA</span> sequences stored?</li>
          <li>What are the other entries?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The <span class="notranslate">DNA</span> sequences are stored in the second line of every 4-line group</li>
            <li>This file is called a <a href="https://en.wikipedia.org/wiki/FASTQ_format">FastQ file</a>. It stores sequence information and quality information. Each sequence is represented by a group of 4 lines with the 1st line being the sequence id, the second the sequence of nucleotides, the third a transition line and the last one a sequence of quality score for each nucleotide.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p>Run <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with</p>
      <ul>
        <li>‚ÄúShort read data from your current history‚Äù to the imported file</li>
      </ul>

      <p>Inspect the generated files</p>
    </li>
  </ol>
</blockquote>

<p>It is often necessary to trim sequenced read, for example, to get rid of bases that were sequenced with high uncertainty (= low quality bases).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li>Run <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúIs this library paired- or single-end?‚Äù to <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li>‚Äú<span class="notranslate">Reads</span> in FASTQ format‚Äù to the imported file</li>
        <li>‚ÄúTrim Galore! advanced settings‚Äù to <code class="language-plaintext highlighter-rouge">Full parameter list</code></li>
        <li>‚ÄúTrim low-quality ends from <span class="notranslate">reads</span>‚Äù to <code class="language-plaintext highlighter-rouge">15</code></li>
        <li>‚ÄúOverlap with adapter sequence required to trim a sequence‚Äù to <code class="language-plaintext highlighter-rouge">3</code></li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-tip-importing-data-from-a-data-library-1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>If your FASTQ files cannot be selected, you might check whether their format is FASTQ with Sanger-scaled quality values (<code class="language-plaintext highlighter-rouge">fastqsanger</code>). You can edit the data type by clicking on the <code class="language-plaintext highlighter-rouge">pencil</code> symbol.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="step-2-mapping-of-the-reads">Step 2: <span class="notranslate">Mapping</span> of the <span class="notranslate">reads</span></h1>

<p>In order to figure where the sequenced <span class="notranslate">DNA</span> fragments originated from in the genome, the short <span class="notranslate">reads</span> must be aligned to the reference genome. This is equivalent to solving a jigsaw puzzles, but unfortunately, not all pieces are unique. In principle, you could do a BLAST analysis to figure out where the sequenced pieces fit best in the known genome. Aligning millions of short sequences this way may, however, take a couple of weeks.</p>

<h3 id="running-bowtie2">Running Bowtie2</h3>

<p>Nowadays, there are many read alignment programs for shotgun sequenced <span class="notranslate">DNA</span>, Bowtie2 being one of them.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span></h3>

  <ol>
    <li><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúIs this single or paired library‚Äù to <code class="language-plaintext highlighter-rouge">Single-end</code></li>
        <li>‚ÄúFASTA/Q file‚Äù to the Trim Galore! output with the trimmed <span class="notranslate">reads</span></li>
        <li>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù to <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code></li>
        <li>‚ÄúSelect reference genome‚Äù to <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg18</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many <span class="notranslate">reads</span> where mapped?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>This information can be accessed by clicking on the resulting history entry. You can see some basic <span class="notranslate">mapping</span> statistics once the alignment is completed. 16676 (66.96%) were mapped exactly 1 time and 7919 (31.80%) aligned &gt;1 times. The overall alignment rate is then 98.76%.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The read alignment step with bowtie2 resulted in a compressed, binary file (BAM) that is not human-readable. It‚Äôs like the zipped version of a text file.</p>

<p>We will show you two ways to inspect the file:</p>

<ol>
  <li>Visualization using a Genome Browser</li>
  <li>Converting the binary format into its text file equivalent</li>
</ol>

<h2 id="visualization-using-a-genome-browser">Visualization using a Genome Browser</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-of-the-reads-in-igv"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization of the <span class="notranslate">reads</span> in IGV</h3>

  <ol>
    <li>Click on the <code class="language-plaintext highlighter-rouge">display with IGV local</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>
    <li>Zoom at the start of chromosome 11 (or <code class="language-plaintext highlighter-rouge">chr11:1,562,200-1,591,483</code>)</li>
  </ol>
</blockquote>

<p>The <span class="notranslate">reads</span> have a direction: they are mapped to the forward or reverse strand, respectively. When hovering over a read, extra information is displayed</p>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Some <span class="notranslate">reads</span> have colored lines included. What is this?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Try to zoom in in one of those lines and you will see the answer!</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>Because the number of <span class="notranslate">reads</span> over a region can be quite large, the IGV browser by default only allows to see the <span class="notranslate">reads</span> that fall into a small window. This behaviour can be changed in the IGV preferences panel.</p>
</blockquote>

<h2 id="inspection-of-the-sam-format">Inspection of the SAM format</h2>

<p>As mentioned above, you can convert the binary BAM file into a simple (but large!) text file, which is called a <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">SAM</a> (Sequence Alignment Map) file.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-conversion-into-a-sam-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Conversion into a SAM file</h3>

  <ol>
    <li><strong>BAM-to-SAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúBAM File to Convert‚Äù to the file generated by Bowtie2</li>
        <li>‚ÄúHeader options‚Äù to <code class="language-plaintext highlighter-rouge">Include header in SAM output</code></li>
      </ul>
    </li>
    <li>Inspect the file by clicking on <code class="language-plaintext highlighter-rouge">eye</code> icon</li>
  </ol>

</blockquote>

<p>A SAM file is a file with</p>

<ul>
  <li>A header with the chromosome names and lengths</li>
  <li>A file content as a tabular file with the location and other information of each read found in the FASTQ file and the <span class="notranslate">mapping</span> information</li>
</ul>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Which information do you find in a SAM/BAM file? What is the additional information compared to a FASTQ file.</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Sequences and Quality information, like FASTQ</li>
      <li><span class="notranslate">Mapping</span> information; Location of the read on the chromosome; <span class="notranslate">Mapping</span> quality ‚Ä¶</li>
      <li>For more information please see <a href="https://en.wikipedia.org/wiki/SAM_(file_format)">https://en.wikipedia.org/wiki/SAM_(file_format)</a></li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</h1>

<p>We already checked the quality of the raw sequencing <span class="notranslate">reads</span> in the first step.
Now we would like to test the quality of the ChIP-seq preparation, to know if your ChIP-seq samples are more enriched than the control (input) samples.</p>

<h2 id="correlation-between-samples">Correlation between samples</h2>

<p>To assess the similarity between the replicates of the ChIP and the input, respectively, it is a common technique to calculate the correlation of
read counts on different regions for all different samples.
We expect that the replicates of the ChIP-seq experiments should be clustered more closely to each other than the replicates of the input sample.
That is, because the input samples should not have enriched regions included - remember the immuno-precipitation step was skiped during the sample preparation.</p>

<p>To compute the correlation between the samples we are going to to use the QC modules of deepTools (http://deeptools.readthedocs.io/), a software package for the QC, processing and analysis of NGS data. Before computing the correlation a time consuming step is required, which is to compute the read coverage over a large number of regions from each BAM file. For this we will use the tool <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

<p>To do that, we need at first to catch up for all our samples and re-run the previous steps (quality control and <span class="notranslate">mapping</span>) on each sample.
To save time, we already did that and we can now work directly on the BAM files of the 8 samples</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-correlation-between-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Correlation between samples</h3>

  <ol>
    <li>Create a new history</li>
    <li>Import the 8 BAM files from <a href="https://doi.org/10.5281/zenodo.892432">Zenodo</a> or from the data library into the history
      <ul>
        <li><a href="https://zenodo.org/record/892432/files/patient1_ChIP_ER_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient1_input_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient1_input_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient2_ChIP_ER_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient2_ChIP_ER_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient2_input_good_outcome.bam"><code class="language-plaintext highlighter-rouge">patient2_input_good_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient3_ChIP_ER_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient3_ChIP_ER_poor_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient3_input_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient3_input_poor_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient4_ChIP_ER_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient4_ChIP_ER_poor_outcome.bam</code></a></li>
        <li><a href="https://zenodo.org/record/892432/files/patient4_input_poor_outcome.bam"><code class="language-plaintext highlighter-rouge">patient4_input_poor_outcome.bam</code></a></li>
      </ul>
    </li>
    <li><strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúSample order matters‚Äù to <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>‚ÄúBam file‚Äù to the 8 imported BAM files</li>
        <li>‚ÄúChoose computation mode‚Äù to <code class="language-plaintext highlighter-rouge">Bins</code></li>
        <li>
          <p>‚ÄúBin size in bp‚Äù to <code class="language-plaintext highlighter-rouge">100</code></p>

          <p>This corresponds to the length of the fragments that were sequenced; it is not the read length!</p>
        </li>
        <li>‚ÄúDistance between bins‚Äù to <code class="language-plaintext highlighter-rouge">500000</code> (to reduce the computation time for the tutorial)</li>
        <li>‚ÄúRegion of the genome to limit the operation to‚Äù to <code class="language-plaintext highlighter-rouge">chr1</code> (to reduce the computation time for the tutorial)</li>
      </ul>

      <p>Using these parameters, the tool will take bins of 100 bp separated by 500,000. For each bin the overlapping <span class="notranslate">reads</span> in each sample will be computed</p>
    </li>
    <li><strong>plotCorrelation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúMatrix file from the multiBamSummary tool‚Äù to the generated multiBamSummary output</li>
      </ul>

      <p>To compute and visualize the sample correlation we use plotCorrelation from deepTools. This is a fast process that allows the user to quickly try different color combinations and outputs. Feel free to try different parameters.</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <figure id="figure-1"><img src="../../images/estrogen-receptor-binding-site-identification/plotCorrelation_output.png" alt="Output for plotCorrelation with the correlation scores between the 8 samples" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Correlation scores between the 8 samples</figcaption></figure>

  <ol>
    <li>How are your samples clustered? Does that correspond to your expectations?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Badly, this is due to the subsampling of our data. You should expect better results from your real data.</li>
      <li>Expected is that your inputs and your ChIP samples cluster together.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-comments-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>More information on these two tools can be found at the <a href="https://deeptools.readthedocs.io/en/latest/content/list_of_tools.html">deepTools documentation page</a>.</p>
</blockquote>

<h2 id="ip-strength-estimation">IP strength estimation</h2>

<p>To evaluate the quality of the immuno-precipitation step, we can compute the IP strength. It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of <span class="notranslate">reads</span> in the control sample. To do that we take the data for one Patient and compare the input sample and the ChIP-seq sample.</p>

<blockquote class="comment">
  <h3 id="comment-comments-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>For more information on how the IP strength is estimated, you can check the <a href="https://deeptools.readthedocs.io/en/latest/content/list_of_tools.html">deepTools documentation page</a>.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ip-strength-estimation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: IP strength estimation</h3>

  <ol>
    <li><strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúSample order matters‚Äù to <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>‚ÄúBam file‚Äù to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> and <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>‚ÄúRegion of the genome to limit the operation to‚Äù to <code class="language-plaintext highlighter-rouge">chr1</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The plotFingerprint tool generates a fingerprint plot. You need to intepret it to know the IP strength. The <a href="https://deeptools.readthedocs.io/en/latest/content/list_of_tools.html">deepTools documentation</a> explains it clearly:</p>

<figure id="figure-2"><img src="../../images/estrogen-receptor-binding-site-identification/QC_fingerprint.png" alt="A guide to interpret a fingerprint plot" /><figcaption><span class="figcaption-prefix">Figure 2:</span> How to interpret a fingerprint plot? Image extracted from the deepTools documentation</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <figure id="figure-3"><img src="../../images/estrogen-receptor-binding-site-identification/plotFingerprint_output.png" alt="Output for plotFingerprint with the fingerprint plot to estimate the IP strength" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Fingerprint plot for the Patient 1 to estimate the IP strength</figcaption></figure>

  <ol>
    <li>What do you think about the quality of the IP for this experiment?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The difference between input and ChIP signal is not totally clear. 20% of chromosome 1 are not sequenced
 at all and the ChIP signal is only slightly more enrichted then the input.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-optional-hands-on-ip-strength-estimation-other-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> (Optional) Hands-on: IP strength estimation (other samples)</h3>

  <ol>
    <li>Run the same analysis on data of the 3 other patients</li>
  </ol>
</blockquote>

<h1 id="step-4-normalization">Step 4: Normalization</h1>

<p>We would like to know where the binding sites of the estrogen receptor are located. For this we need
to extract which parts of the genome have been enriched (more <span class="notranslate">reads</span> mapped) within the samples that underwent immunoprecipitation.</p>

<p>For the normalization we have two options.</p>

<ol>
  <li>Normalization by sequencing depth</li>
  <li>Normalization by input file</li>
</ol>

<h2 id="generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</h2>

<p>We first need to make the samples comparable. Indeed, the different samples have usually a different sequencing depth, i.e. a different number of <span class="notranslate">reads</span>.
These differences can bias the interpretation of the number of <span class="notranslate">reads</span> mapped to a specific genome region.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-coverage-file-normalization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Coverage file normalization</h3>

  <ol>
    <li><strong>IdxStats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúBAM file‚Äù to ‚ÄúMultiple datasets‚Äù: <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> and <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the output of this tool?</li>
          <li>How many <span class="notranslate">reads</span> has been mapped on chr2 for the input and for the ChIP-seq samples?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>This tool estimates how many <span class="notranslate">reads</span> mapped to which chromosome. Furthermore, it tells the chromosome lengths and naming convention (with or without ‚Äòchr‚Äô in the beginning)</li>
            <li>1,089,370 for ChIP-seq samples and 1,467,480 for the input</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúBAM file‚Äù to ‚ÄúMultiple datasets‚Äù: <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code> and <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>‚ÄúBin size in bases‚Äù to <code class="language-plaintext highlighter-rouge">25</code></li>
        <li>‚ÄúScaling/Normalization method‚Äù to <code class="language-plaintext highlighter-rouge">Normalize coverage to 1x</code></li>
        <li>‚ÄúEffective genome size‚Äù to <code class="language-plaintext highlighter-rouge">hg19 (2451960000)</code></li>
        <li>‚ÄúCoverage file format‚Äù to <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What are the different columns of a <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code> file?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>chrom, chromStart, chromEnd and a data value</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but to generate a <code class="language-plaintext highlighter-rouge">bigWig</code> output file</li>
    <li>
      <p><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect both signal coverages (input and ChIP samples) in IGV</p>

      <blockquote class="question">
        <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is a bigWig file?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>A bigWig file is a compressed <span class="notranslate">bed</span>graph file. Similar in relation as BAm to SAM, but this time just for coverage data. This means bigWig and <span class="notranslate">bed</span>graph
files are much smaller than BAM or SAM files.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="generation-of-input-normalized-coverage-files-and-their-visualization">Generation of input-normalized coverage files and their visualization</h2>

<p>To extract only the information induced by the immunoprecipitation, we normalize for each patient the coverage file for the sample that underwent immunoprecipitation by the coverage file for the input sample. Here we use the tool bamCompare which compare 2 BAM files while caring for sequencing depth normalization.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generation-of-input-normalized-coverage-files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generation of input-normalized coverage files</h3>

  <ol>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúFirst BAM file (e.g. treated sample)‚Äù to <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>‚ÄúSecond BAM file (e.g. control sample)‚Äù to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code></li>
        <li>‚ÄúBin size in bases‚Äù to <code class="language-plaintext highlighter-rouge">50</code></li>
        <li>‚ÄúHow to compare the two files‚Äù to <code class="language-plaintext highlighter-rouge">Compute log2 of the number of <span class="notranslate">reads</span> ratio</code></li>
        <li>‚ÄúCoverage file format‚Äù to <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code></li>
        <li>
          <p>‚ÄúRegion of the genome to limit the operation to‚Äù to <code class="language-plaintext highlighter-rouge">chr11</code> (to reduce the computation time for the tutorial)</p>

          <blockquote class="question">
            <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

            <ol>
              <li>What does mean a positive or a negative value in the 4th column?</li>
            </ol>

            <blockquote class="solution">
              <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
              <ol>
                <li>The 4th column contains the log2 of the number of <span class="notranslate">reads</span> ratio between the ChIP-seq sample and the input sample. A positive value means that the coverage on the portion is more important in the ChIP-seq sample than in the input sample</li>
              </ol>
            </blockquote>
          </blockquote>
        </li>
      </ul>
    </li>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but to generate a <code class="language-plaintext highlighter-rouge">bigWig</code> output file</li>
    <li>
      <p><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect the log2 ratio</p>

      <p>Remember that the bigWig file contains only the signal on chromosome 11!</p>
    </li>
  </ol>

</blockquote>

<h1 id="step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</h1>

<p>We can also call the enriched regions, or peaks, found in the ChIP-seq samples.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peak-calling"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Peak calling</h3>

  <ol>
    <li><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúChIP-Seq Treatment File‚Äù to <code class="language-plaintext highlighter-rouge">patient1_ChIP_ER_good_outcome</code></li>
        <li>‚ÄúChIP-Seq Control File‚Äù to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code></li>
        <li>‚ÄúEffective genome size‚Äù to <code class="language-plaintext highlighter-rouge">H. sapiens (2.7e9)</code></li>
        <li>‚ÄúOutputs‚Äù to <code class="language-plaintext highlighter-rouge">Summary page (html)</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comments-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
        <p>The advanced options may be adjusted, depending of the samples.
If your ChIP-seq experiment targets regions of broad enrichment, <em>e.g.</em> non-punctuate histone modifications, select calling of broad regions.
If your sample has a low duplication rate (<em>e.g.</em> below 10%), you might keep all duplicate <span class="notranslate">reads</span> (tags). Otherwise, you might use the ‚Äòauto‚Äô option to estimate the maximal allowed number of duplicated <span class="notranslate">reads</span> per genomic location.</p>
      </blockquote>
    </li>
    <li><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect with the signal coverage and log2 ratio tracks</li>
  </ol>

</blockquote>

<p>The called peak regions can be filtered by, <em>e.g.</em> fold change, FDR and region length for further <span class="notranslate">downstream</span> analysis.</p>

<h1 id="step-6-plot-the-signal-on-the-peaks-between-samples">Step 6: Plot the signal on the peaks between samples</h1>

<p>Plotting your region of interest will involve using two tools from the <strong>deepTools</strong> suite.</p>
<ul>
  <li>computeMatrix : Computes the signal on given regions, using the bigwig coverage files from different samples.</li>
  <li>plotHeatmap : Plots heatMap of the signals using the computeMatrix output.</li>
</ul>

<p>ptionally, you can also use <code class="language-plaintext highlighter-rouge">plotProfile</code>to create a profile plot using to computeMatrix output.</p>

<h2 id="computematrix">computeMatrix</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-of-the-coverage"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization of the coverage</h3>

  <ol>
    <li><strong>UCSC Main</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚Äúassembly‚Äù to <code class="language-plaintext highlighter-rouge">hg18</code></li>
        <li>‚Äútrack‚Äù to <code class="language-plaintext highlighter-rouge">RefSeq genes</code></li>
        <li>‚Äúregion‚Äù to <code class="language-plaintext highlighter-rouge">position</code> with <code class="language-plaintext highlighter-rouge">chr11</code></li>
        <li>‚Äúoutput format‚Äù to <code class="language-plaintext highlighter-rouge"><span class="notranslate">BED</span></code></li>
        <li>‚ÄúSend output to‚Äù to <code class="language-plaintext highlighter-rouge"><span class="notranslate">Galaxy</span></code></li>
      </ul>
    </li>
    <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúRegions to plot‚Äù to the imported UCSC file</li>
        <li>‚ÄúScore file‚Äù to the bigwig file generated by bamCompare</li>
        <li>
          <p>‚ÄúcomputeMatrix has two main output options‚Äù to <code class="language-plaintext highlighter-rouge">scale-regions</code></p>

          <p>This option stretches or shrinks all regions in the <span class="notranslate">BED</span> file (here: genes) to the same length (bp) as indicated by the user</p>
        </li>
        <li>‚ÄúShow advanced options‚Äù to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>‚ÄúConvert missing values to 0?‚Äù to <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <p>This tool prepares a file with scores per genomic region, which is required as input for the next tool.</p>
    </li>
  </ol>

</blockquote>

<h2 id="plotheatmap">plotHeatmap</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-of-the-coverage-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization of the coverage</h3>
  <ol>
    <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúMatrix file from the computeMatrix tool‚Äù to the generated matrix</li>
        <li>‚ÄúShow advanced options‚Äù to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>‚ÄúDid you compute the matrix with more than one groups of regions?‚Äù to the correct setting</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<figure id="figure-4"><img src="../../images/estrogen-receptor-binding-site-identification/tutorial-scheme.png" alt="Summary of the different steps of the tutorial and the generated files" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Different steps of the tutorials with the generated files</figcaption></figure>

<h1 id="additional-exercise-if-you-have-finished-all-above">Additional exercise (if you have finished all above)</h1>

<h2 id="additional-quality-control--gc-bias-assessment">Additional Quality control : GC bias assessment</h2>

<p>A common problem of PCR-based protocols is the observation that GC-rich regions tend to be amplified more readily than GC-poor regions.
We need to check that our samples do not have more <span class="notranslate">reads</span> from regions of the genome with high GC.</p>

<blockquote class="comment">
  <h3 id="comment-comments-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
  <p>GC bias was for many years a big problem, but recent advances in sample preparation have solved this problem to a degree that you can skip this step more often.</p>
</blockquote>

<p>For practical reasons, we will focus here only on one of our BAM files. With real data you should repeat these steps for all your samples.</p>

<blockquote class="question">
  <h3 id="question-questions-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Can you guess why it makes more sense to check the input files for GC bias?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-10"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>We only want to assess the bias induced by the PCR-based protocols. This is not possible with the ChIP samples, as the enriched regions (binding sites) can have a potential GC enrichment on their own.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-gc-bias-assessment"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: GC bias assessment</h3>

  <ol>
    <li><strong>computeGCbias</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúBAM file‚Äù to <code class="language-plaintext highlighter-rouge">patient1_input_good_outcome</code></li>
        <li>‚ÄúReference genome‚Äù to <code class="language-plaintext highlighter-rouge">locally cached</code></li>
        <li>‚ÄúUsing reference genome‚Äù to <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg18</code></li>
        <li>‚ÄúEffective genome size‚Äù to <code class="language-plaintext highlighter-rouge">hg19 (2451960000)</code></li>
        <li>‚ÄúFragment length used for the sequencing‚Äù to <code class="language-plaintext highlighter-rouge">300</code></li>
        <li>‚ÄúRegion of the genome to limit the operation to‚Äù to <code class="language-plaintext highlighter-rouge">chr1</code> (to reduce the computation time for the tutorial)</li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <figure id="figure-5"><img src="../../images/estrogen-receptor-binding-site-identification/computeGCBias_output.png" alt="Output for computeGCbias with the GC bias estimation" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Estimation of the GC bias for the input sample for the Patient 1</figcaption></figure>

        <ol>
          <li>Does this dataset have a GC bias?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-11"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>There is no significantly more <span class="notranslate">reads</span> in the GC-rich regions.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="plotting-heatmap-from-multiple-samples-with-clustering">Plotting heatmap from multiple samples with clustering</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plotting-multiple-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: plotting multiple samples</h3>

  <ol>
    <li>Run <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with  same parameters as above, for all 4 patients:
      <ul>
        <li>‚ÄúFirst BAM file (e.g. treated sample)‚Äù to <code class="language-plaintext highlighter-rouge">patient2_ChIP_ER_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient3_ChIP_ER_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient4_ChIP_ER_good_outcome</code></li>
        <li>‚ÄúSecond BAM file (e.g. control sample)‚Äù to <code class="language-plaintext highlighter-rouge">patient2_input_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient3_input_good_outcome</code>, <code class="language-plaintext highlighter-rouge">patient4_input_good_outcome</code></li>
      </ul>
    </li>
    <li>
      <p>Perform peak calling again using treatment file : <code class="language-plaintext highlighter-rouge">patient2_ChIP_ER_good_outcome</code> and control <code class="language-plaintext highlighter-rouge">patient2_input_good_outcome</code>, using macs2 parameters same as above.</p>
    </li>
    <li>
      <p>Concatenate the outputs (summits in <span class="notranslate">BED</span>) from <code class="language-plaintext highlighter-rouge">patient1</code> and <code class="language-plaintext highlighter-rouge">patient2</code> using <code class="language-plaintext highlighter-rouge">Operate on Genomic Intervals</code> ‚Äì&gt; <code class="language-plaintext highlighter-rouge">Concatenate</code></p>
    </li>
    <li>
      <p>Sort the output <code class="language-plaintext highlighter-rouge">Operate on Genomic Intervals</code> ‚Äì&gt; <code class="language-plaintext highlighter-rouge">sortBED</code></p>
    </li>
    <li>
      <p>Merge the overlapping intervals using <code class="language-plaintext highlighter-rouge">Operate on Genomic Intervals</code> ‚Äì&gt; <code class="language-plaintext highlighter-rouge">MergeBED</code></p>
    </li>
    <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but:
      <ul>
        <li>Regions to plot : select the merged <span class="notranslate">bed</span> from above</li>
        <li>Output option : <code class="language-plaintext highlighter-rouge">reference-point</code></li>
        <li>The reference point for the plotting: <code class="language-plaintext highlighter-rouge">center of region</code></li>
        <li>Distance <span class="notranslate">upstream</span> of the start site of the regions defined in the region file : <code class="language-plaintext highlighter-rouge">3000</code></li>
        <li>Distance <span class="notranslate">downstream</span> of the end site of the given regions: <code class="language-plaintext highlighter-rouge">3000</code></li>
      </ul>

      <p>With this option, it considers only those genomic positions before (<span class="notranslate">downstream</span>) and/or after (<span class="notranslate">upstream</span>) a reference point (<em>e.g.</em> TSS, which corresponds to the annotated gene start in our case)</p>
    </li>
    <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li>‚ÄúMatrix file from the computeMatrix tool‚Äù to the generated matrix</li>
        <li>‚ÄúShow advanced options‚Äù to <code class="language-plaintext highlighter-rouge">yes</code></li>
        <li>‚ÄúDid you compute the matrix with more than one groups of regions?‚Äù to <code class="language-plaintext highlighter-rouge">No, I used only one group</code></li>
        <li>‚ÄúClustering algorithm‚Äù to <code class="language-plaintext highlighter-rouge">Kmeans clustering</code>
Inspect the output</li>
      </ul>
    </li>
  </ol>

</blockquote>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>ChIP-seq data requires multiple methods of quality assessment to ensure that the data is of high quality.</p>
</li>
                            
                            <li><p>Multiple normalization methods exists depending on the availability of input data.</p>
</li>
                            
                            <li><p>Heatmaps containing all genes of an organism can be easily plotted given a BED file and a coverage file.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Identification of the binding sites of the Estrogen receptor (Epigenetics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez, 2021 <b>Identification of the binding sites of the Estrogen receptor (Galaxy Training Materials)</b>. <a href="/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html">/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{epigenetics-estrogen-receptor-binding-site-identification,
    author = "Friederike D√ºndar and Anika Erxleben and B√©r√©nice Batut and Vivek Bhardwaj and Fidel Ramirez",
    title = "Identification of the binding sites of the Estrogen receptor (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/estrogen-receptor-binding-site-identification/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
