<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/css/font-awesome.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
    </head>
    <body>
        



<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="fa fa-folder-o" aria-hidden="true"></i> Sequence analysis
                        </a>
                    </li>

                    
                        
                        
                        
                    

                    

                    

                    

                    
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/sequence-analysis#references" title="References">
                            <i class="fa fa-book" aria-hidden="true"></i> Literature
                        </a>
                    </li>
                    

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="fa fa-life-ring" aria-hidden="true"></i> Help
    </a>
    <div class="dropdown-menu">
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Chat on Gitter">
            Gitter
        </a>
        <a class="dropdown-item" href="https://biostar.usegalaxy.org/" title="Ask on Biostars">
            Biostars
        </a>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/mapping/tutorial.md">
                            <i class="fa fa-github" aria-hidden="true"></i> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <section class="tutorial">
        <h1>Mapping</h1>

        <blockquote class="overview">
            <h3>Overview</h3>

            <strong><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</strong>
            <ul>
            
            <li>What two things are crucial for a correct mapping?</li>
            
            <li>What is BAM?</li>
            
            </ul>

            <strong><i class="fa fa-bullseye" aria-hidden="true"></i> Objectives</strong>
            <ul>
            
            <li>You will learn what mapping is</li>
            
            <li>A genome browser is shown that helps you to understand your data</li>
            
            </ul>

            
            <strong><i class="fa fa-check-circle" aria-hidden="true"></i> Requirements</strong>
            <ul>
            
                <li>
                    
                    <a href="/training-material/topics//introduction/">Galaxy introduction</a>
                    
                </li>
            
            
            </ul>
            

            <p><strong><i class="fa fa-hourglass-end" aria-hidden="true"></i> Time estimation:</strong> 1h</p>
        </blockquote>

        <h1 class="no_toc" id="introduction-to-next-generation-sequencing-data-mapping">Introduction to next generation sequencing data mapping</h1>

<p>To map DNA/RNA reads of an experiment to a reference genome is a key step in modern genomic data analysis. With the mapping the reads are assigned to a specific location in the genome and insights like the expression level of genes can be gained.
In the following we will process a dataset with a mapper, ‘Bowtie2’, and we will visualize the data with the software ‘IGV’.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#mapping" id="markdown-toc-mapping">Mapping</a></li>
  <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a></li>
</ol>

</blockquote>

<h1 id="mapping">Mapping</h1>
<blockquote class="hands_on">
  <h3 id="-hands-on-mapping-with-bowtie2"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Mapping with Bowtie2</h3>

  <p>We will process data inside Galaxy with Bowtie2 and have a look at the results.</p>

  <ol>
    <li>
      <p>Load the <a href="https://zenodo.org/record/61771/files/GSM461178_untreat_paired_subset_1.fastq">‘GSM461178_untreat_paired_subset_1.fastq’</a> dataset from <a href="https://zenodo.org/record/61771">Zenodo</a> into Galaxy.</p>

      <blockquote class="tip">
        <h3 id="-tip-importing-data-via-links"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location [right click on Zenodo -&gt; Copy link location]</li>
          <li>Open the Galaxy Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>Paste the link into the text field</li>
          <li>Press <strong>Start</strong></li>
        </ul>
      </blockquote>

      <blockquote class="tip">
        <h3 id="-tip-changing-the-file-type-fastq-to-fastqsanger-once-the-data-file-is-in-your-history"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Changing the file type <code class="highlighter-rouge">fastq</code> to <code class="highlighter-rouge">fastqsanger</code> once the data file is in your history</h3>

        <ul>
          <li>Click on the pencil button displayed in your dataset in the history</li>
          <li>Choose <strong>Datatype</strong> on the top</li>
          <li>Select <code class="highlighter-rouge">fastqsanger</code></li>
          <li>Press <strong>save</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p><strong>Bowtie2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Search in the tool bar on the left the mapper ‘bowtie2’ and run the mapper with the dataset.</p>

      <blockquote class="tip">
        <h3 id="-tip-search-for-tools"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Search for tools</h3>

        <ul>
          <li>Clink into the search field on the left</li>
          <li>Type <strong>bowtie2</strong></li>
          <li>Select <strong>bowtie2</strong></li>
          <li>Select the uploaded dataset as the fastq file</li>
          <li>Choose as a reference genome human hg19</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Click on the right history panel on the Bowtie2 output. Look careful at the information which is given here:</p>

      <div class="highlighter-rouge"><div class="highlight"><pre class="highlight"><code>   100000 reads; of these: 100000 (100.00%) were unpaired;
   of these: 99407 (99.41%) aligned 0 times
   149 (0.15%) aligned exactly 1 time
   444 (0.44%) aligned &gt;1 times
   0.59% overall alignment rate
</code></pre></div>      </div>

      <blockquote class="question">
        <h3 id="-questions"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ul>
          <li>What information is provided here?</li>
          <li>Is everything as expected?</li>
          <li>
            <p>Why is the mapping so bad?</p>

            <details>
 <summary>Click to view answers</summary>
 <ol type="1">
 <li>The information given here is a quantity one. We can see how many sequences are aligned. It does not tell us something about the quality.</li>
 <li>No, only 0.59% of all reads could be mapped.</li>
 <li>We mapped against the wrong reference genome! </li>
 </ol>
 </details>
          </li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p><strong>Bowtie2</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Rerun Bowtie2 with the correct reference genome ‘Drosophila melanogaster’ or short ‘dm3’.</p>

      <blockquote class="comment">
        <h3 id="-comments"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <ul>
          <li>It is a good practice to give the datasets names that are easy to understand.</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="visualization">Visualization</h1>

<p>A general description of the user interface of the IGV browser can be read here: <a href="/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html">IGV Browser description</a></p>

<blockquote class="hands_on">
  <h3 id="-hands-on-visualization-with-the-igv-browser"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Visualization with the IGV browser</h3>

  <p>The Integrative Genomics Viewer (IGV) is a high-performance visualization tool for interactive exploration of large, integrated genomic datasets. It supports a wide variety of data types, including array-based and next-generation sequence data, and genomic annotations. In the following we will use it to visualize the computed mapping.</p>

  <ol>
    <li><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: To display the result in IGV open the IGV browser local on your computer.</li>
    <li><strong>Galaxy</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Click on the right history panel on the Bowtie2 item.</li>
    <li><strong>Galaxy</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Choose in the history on the computed Bowtie2 results and click on ‘local’ at ‘display with IGV’.</li>
    <li>
      <p><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: The BAM file should be opened in the IGV browser and the genome should be loaded automatically.</p>

      <blockquote class="tip">
        <h3 id="-tip-get-access-to-more-genomes"><i class="fa fa-lightbulb-o" aria-hidden="true"></i> Tip: Get access to more genomes</h3>

        <p>If the genome of your interest is not there check if its
available via “More…”. Is this is not the case you can add it manually via the menu
“Genomes -&gt; Load Genome from…”</p>

        <figure id="figure-1"><img src="../../images/igv_select_genome.png" alt="alt text" title="Select genome" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Select genome</figcaption></figure>
      </blockquote>
    </li>
    <li><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: The interesting part is at chromosome 4, position 86,761 to 87,907. Navigate to it.</li>
    <li><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: There are two views:
      <ul>
        <li>the aligned reads view</li>
        <li>the coverage view</li>
      </ul>

      <blockquote class="question">
        <h3 id="-questions-1"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ul>
          <li>What could it mean if a bar in the coverage view is colored?</li>
          <li>What could be the reason that a read is white instead of grey?</li>
          <li>
            <p>How many reads are mapped at position chr4:87482?</p>

            <details>
 <summary>Click to view answers</summary>
 <ol type="1">
 <li>If a nucleotide differs from the reference sequence in greater than 20% of quality weighted reads, IGV colors the bar in proportion to the read count of each base.</li>
 <li>They have a mapping quality equal to zero. Interpretation of this mapping quality depends on the mapping aligner as some commonly used aligners use this convention to mark a read with multiple alignments. In such a case, the read also maps to another location with equally good placement. It is also possible the read could not be uniquely placed but the other placements do not necessarily give equally good quality hits.</li>
 <li>There are seven reads. Six have the correct 'T', one read a 'G'.</li>
 </ol>
 </details>
          </li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="hands_on">
  <h3 id="-hands-on-differences-between-a-good-and-a-bad-mapping"><i class="fa fa-pencil" aria-hidden="true"></i> Hands-on: Differences between a good and a bad mapping</h3>

  <ol>
    <li>
      <p><strong>Galaxy</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: To show you the difference between a good and a bad dataset we provided an additional dataset which will give you a bad mapping. Load the <a href="https://zenodo.org/record/61771/files/GSM461178_untreat_paired_subset_1.fastq">‘GSM461182_untreat_single_subset.fastq’</a> dataset from <a href="https://zenodo.org/record/61771">Zenodo</a> into Galaxy.</p>

      <blockquote class="comment">
        <h3 id="-comments-1"><i class="fa fa-commenting-o" aria-hidden="true"></i> Comments</h3>
        <p>If you are using the <a href="http://galaxy.uni-freiburg.de">Freiburg Galaxy instance</a>, you can load the dataset using ‘Shared Data’ -&gt; ‘Data Libraries’ -&gt; ‘Galaxy Courses’ -&gt; ‘RNA-Seq’ -&gt; ‘fastq’</p>
      </blockquote>
    </li>
    <li>
      <p><strong>IGV</strong> <i class="fa fa-wrench" aria-hidden="true"></i>: Load the two mappings into IGV and compare them!</p>

      <blockquote class="question">
        <h3 id="-questions-2"><i class="fa fa-question-circle" aria-hidden="true"></i> Questions</h3>

        <ul>
          <li>
            <p>How IGV helps you to see that the second dataset is bad?</p>

            <details>
 <summary>Click to view answers</summary>
 <ol type="1">
 <li>White/transparent reads indicate bad mapping quality. Second, IGV shows you the nucleotides in color if they are different in the reference genome.</li>
 </ol>
 </details>
          </li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>


        
        <blockquote class="key_points">
            <h3><i class="fa fa-key" aria-hidden="true"></i> Key points</h3>

            <ul>
                
                <li>Know your data!</li>
                
                <li>Mapping is not trivial</li>
                
                <li>There are many mapping algorithms, it depends on your data which one to choose</li>
                
            </ul>
        </blockquote>
        

        
        <h1>Useful literature</h1>
        <p>Useful information regarding this type of analysis with descriptions and paper references for the tools used in this tutorial, and literature for this analysis techniques and interpretations can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
        

        <h3><i class="fa fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!</h3>

        <hr>

        <blockquote class="overview">
            <h3><i class="fa fa-comments-o" aria-hidden="true"></i> Feedback</h3>
            Please take a moment and provide your feedback on this tutorial. Your feedback will help guide and improve future revisions to this tutorial.
            <a href="https://tinyurl.com/GTNfeedback">Feedback Form</a>
        </blockquote>
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Joachim Wolff)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/mapping/tutorial.md">GitHub</a>.
        </p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
</html>
