<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Mapping</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Analyses of sequences" />
        <meta property="og:title" content="Galaxy Training: Mapping" />
        <meta property="og:description" content="Analyses of sequences" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/sequence-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Sequence analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fmapping%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fmapping%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fmapping%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fmapping%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fmapping%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fsequence-analysis%2Ftutorials%2Fmapping%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/sequence-analysis/tutorials/mapping/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Mapping",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Sequence analysis",
    "description": "Analyses of sequences",
    "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/"
  },
  "courseCode": "sequence-analysis / mapping / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Mapping' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - What is mapping?\n - What two things are crucial for a correct mapping?\n - What is BAM?\n\n\\nThe objectives are:\n - You will learn what mapping is\n - A genome browser is shown that helps you to understand your data\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Joachim Wolff"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Joachim Wolff"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Helena Rasche"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Sequence analysis",
      "description": "Analyses of sequences",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Mapping</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/joachimwolff/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/joachimwolff" alt="Joachim Wolff">Joachim Wolff</a>, <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="B√©r√©nice Batut">B√©r√©nice Batut</a>, <a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/hexylena" alt="Helena Rasche">Helena Rasche</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What is mapping?</p>
</li>
            
            <li><p>What two things are crucial for a correct mapping?</p>
</li>
            
            <li><p>What is BAM?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>You will learn what mapping is</p>
</li>
            
            <li><p>A genome browser is shown that helps you to understand your data</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1324070">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/sequence-analysis/tutorials/mapping/workflows/" title="Workflows" alt="Mapping workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/mapping/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://www.galaxytrakr.org/" title="GalaxyTrakr">
            GalaxyTrakr
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Sequencing produces a collection of sequences without genomic context. We do not know to which part of the genome the sequences correspond to. <span class="notranslate">Mapping</span> the <span class="notranslate">reads</span> of an experiment to a reference genome is a key step in modern genomic data analysis. With the <span class="notranslate">mapping</span> the <span class="notranslate">reads</span> are assigned to a specific location in the genome and insights like the <span class="notranslate">expression</span> level of genes can be gained.</p>

<p>The short <span class="notranslate">reads</span> do not come with position information, so we do not know what part of the genome they came from. We need to use the sequence of the read itself to find the corresponding region in the reference sequence. But the reference sequence can be quite long (~3 billion bases for human), making it a daunting task to find a matching region. Since our <span class="notranslate">reads</span> are short, there may be several, equally likely places in the reference sequence from which they could have been read. This is especially true for repetitive regions.</p>

<p>In principle, we could do a BLAST analysis to figure out where the sequenced pieces fit best in the known genome. We would need to do that for each of the millions of <span class="notranslate">reads</span> in our sequencing data. Aligning millions of short sequences this way may, however, take a couple of weeks. And we do not care about the exact base to base correspondence (alignment). What we are interested in is ‚Äúwhere these <span class="notranslate">reads</span> came from‚Äù. This approach is called <strong><span class="notranslate">mapping</span></strong>.</p>

<p>In the following, we will process a dataset with the mapper <strong>Bowtie2</strong> and we will visualize the data with the program <strong>IGV</strong>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#prepare-the-data" id="markdown-toc-prepare-the-data">Prepare the data</a></li>
  <li><a href="#map-reads-on-a-reference-genome" id="markdown-toc-map-reads-on-a-reference-genome">Map <span class="notranslate">reads</span> on a reference genome</a></li>
  <li><a href="#inspection-of-a-bam-file" id="markdown-toc-inspection-of-a-bam-file">Inspection of a BAM file</a></li>
  <li><a href="#visualization-using-a-genome-browser-igv" id="markdown-toc-visualization-using-a-genome-browser-igv">Visualization using a Genome Browser (IGV)</a></li>
  <li><a href="#visualization-using-a-genome-browser-jbrowse" id="markdown-toc-visualization-using-a-genome-browser-jbrowse">Visualization using a Genome Browser (JBrowse)</a></li>
</ol>

</blockquote>

<h1 id="prepare-the-data">Prepare the data</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2.fastq.gz</code> from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1324070/files/wt_H3K4me3_read1.fastq.gz
https://zenodo.org/record/1324070/files/wt_H3K4me3_read2.fastq.gz
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>

      <p>As default, <span class="notranslate">Galaxy</span> takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Rename the files to <code class="language-plaintext highlighter-rouge"><span class="notranslate">reads</span>_1</code> and <code class="language-plaintext highlighter-rouge"><span class="notranslate">reads</span>_2</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>We just imported in <span class="notranslate">Galaxy</span> FASTQ files corresponding to paired-end data as we could get directly from a sequencing facility.</p>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data. The first step for any type of sequencing data is always to check their quality.</p>

<p>There is a dedicated tutorial for <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">quality control</a> of sequencing data. We will not repeat the steps there. You should follow the <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">tutorial</a> and apply it to your data before going further.</p>

<h1 id="map-reads-on-a-reference-genome">Map <span class="notranslate">reads</span> on a reference genome</h1>

<p>Read <span class="notranslate">mapping</span> is the process to align the <span class="notranslate">reads</span> on a reference genomes. A mapper takes as input a reference genome and a set of <span class="notranslate">reads</span>. Its aim is to align each read in the set of <span class="notranslate">reads</span> on the reference genome, allowing mismatches, indels and clipping of some short fragments on the two ends of the <span class="notranslate">reads</span>:</p>

<figure id="figure-1"><img src="../../images/mapping/mapping.png" alt="Explanation of mapping" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Illustration of the <span class="notranslate">mapping</span> process. The input consists of a set of <span class="notranslate">reads</span> and a reference genome. In the middle, it gives the results of <span class="notranslate">mapping</span>: the locations of the <span class="notranslate">reads</span> on the reference genome. The first read is aligned at position 100 and the alignment has two mismatches. The second read is aligned at position 114. It is a local alignment with clippings on the left and right. The third read is aligned at position 123. It consists of a 2-base insertion and a 1-base deletion.</figcaption></figure>

<p>We need a reference genome to map the <span class="notranslate">reads</span> on.</p>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What is a reference genome?</li>
    <li>For each model organism, several possible reference genomes may be available (e.g. <code class="language-plaintext highlighter-rouge">hg19</code> and <code class="language-plaintext highlighter-rouge">hg38</code> for human). What do they correspond to?</li>
    <li>Which reference genome should we use?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>A reference genome (or reference assembly) is a set of nucleic acid sequences assembled as a representative example of a species‚Äô genetic material. As they are often assembled from the sequencing of different individuals, they do not accurately represent the set of genes of any single organism, but a mosaic of different nucleic acid sequences from each individual.</li>
      <li>As the cost of <span class="notranslate">DNA</span> sequencing falls, and new full genome sequencing technologies emerge, more genome sequences continue to be generated. Using these new sequences, new alignments are builts and the reference genomes improved (fewer gaps, fixed misrepresentations in the sequence, etc). The different reference genomes correspond to the different released versions (called ‚Äúbuilds‚Äù).</li>
      <li>This data comes from the ChIP-seq of mice, so we will use mm10 (<em>Mus musculus</em>).</li>
    </ol>
  </blockquote>
</blockquote>

<p>Currently, there are over 60 different mappers, and their number is growing. In this tutorial, we will use <a href="http://bowtie-bio.sourceforge.net/bowtie2/">Bowtie2</a>, a fast and memory-efficient open-source tool particularly good at aligning sequencing <span class="notranslate">reads</span> of about 50 up to 1,000s of bases to relatively long genomes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping-with-bowtie2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span> with Bowtie2</h3>
  <ol>
    <li><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúIs this single or paired library‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA/Q file #1‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">reads</span>_1</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA/Q file #2‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">reads</span>_2</code></li>
            <li>
              <p><em>‚ÄúDo you want to set paired-end options?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>You should have a look at the parameters there, specially the mate orientation if you know it. They can improve the quality of the paired-end <span class="notranslate">mapping</span>.</p>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúSelect reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mouse (Mus musculus): mm10</code></li>
          </ul>
        </li>
        <li>
          <p><em>‚ÄúSelect analysis mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Default setting only</code></p>

          <p>You should have a look at the non default parameters and try to understand them. They can have an impact on the <span class="notranslate">mapping</span> and improving it.</p>
        </li>
        <li><em>‚ÄúSave the bowtie2 <span class="notranslate">mapping</span> statistics to the history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the <code class="language-plaintext highlighter-rouge"><span class="notranslate">mapping</span> stats</code> file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What information is provided here?</li>
          <li>How many <span class="notranslate">reads</span> have been mapped exactly 1 time?</li>
          <li>How many <span class="notranslate">reads</span> have been mapped more than 1 time? How is it possible? What should we do with them?</li>
          <li>How many pair of <span class="notranslate">reads</span> have not been mapped? What are the causes?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The information given here is a quantity one. We can see how many sequences are aligned. It does not tell us something about the quality.</li>
            <li>~90% <span class="notranslate">reads</span> have been aligned exactly 1 time</li>
            <li>~7% <span class="notranslate">reads</span> have been aligned concordantly &gt;1 times. These are called multi-mapped <span class="notranslate">reads</span>. It can happen because of repetitions in the reference genome (multiple copies of a gene for example), particularly when the <span class="notranslate">reads</span> are small. It is difficult to decide where these sequences come from and therefore most of the pipelines ignore them. Always check the statistics there to be sure of not discarding too much information in any <span class="notranslate">downstream</span> analyses.</li>
            <li>~3% pair of <span class="notranslate">reads</span> have not been mapped because
              <ul>
                <li>both <span class="notranslate">reads</span> in the pair aligned but their positions do not concord with pair of <span class="notranslate">reads</span> (<code class="language-plaintext highlighter-rouge">aligned discordantly 1 time</code>)</li>
                <li><span class="notranslate">reads</span> of these pairs are multi-mapped (<code class="language-plaintext highlighter-rouge">aligned &gt;1 times</code> in <code class="language-plaintext highlighter-rouge">pairs aligned 0 times concordantly or discordantly</code>)</li>
                <li>one read of these pairs are mapped but not the paired read (<code class="language-plaintext highlighter-rouge">aligned exactly 1 time</code> in <code class="language-plaintext highlighter-rouge">pairs aligned 0 times concordantly or discordantly</code>)</li>
                <li>the rest are not mapped at all</li>
              </ul>
            </li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>Checking the <span class="notranslate">mapping</span> statistics is an important step to do before continuing any analyses. There are several potential sources for errors in <span class="notranslate">mapping</span>, including (but not limited to):</p>

<ul>
  <li><strong>Polymerase Chain Reaction (PCR) artifacts</strong>: Many high-throughput sequencing (HTS) methods involve one or multiple PCR steps. PCR errors will show as mismatches in the alignment, and especially errors in early PCR rounds will show up in multiple <span class="notranslate">reads</span>, falsely suggesting genetic variation in the sample. A related error would be PCR duplicates, where the same read pair occurs multiple times, skewing coverage calculations in the alignment.</li>
  <li><strong>Sequencing errors</strong>: The sequencing machine can make an erroneous call either for physical reasons (e.g. oil on an Illumina slide) or due to properties of the sequenced <span class="notranslate">DNA</span> (e.g., homopolymers). As sequencing errors are often random, they can be filtered out as singleton <span class="notranslate">reads</span> during variant calling.</li>
  <li><strong><span class="notranslate">Mapping</span> errors</strong>: The <span class="notranslate">mapping</span> algorithm can map a read to the wrong location in the reference. This often happens around repeats or other low-complexity regions.</li>
</ul>

<p>So if the <span class="notranslate">mapping</span> statistics are not good, you should investigate the cause of these errors before going further in your analyses.</p>

<p>After that, you should have a look at the <span class="notranslate">reads</span> and inspect the BAM file where the read <span class="notranslate">mapping</span>s are stored.</p>

<h1 id="inspection-of-a-bam-file">Inspection of a BAM file</h1>

<p>A BAM (<a href="https://en.wikipedia.org/wiki/SAM_(file_format)">Binary Alignment Map</a>) file is a compressed binary file storing the read sequences, whether they have been aligned to a reference sequence (e.g. a chromosome), and if so, the position on the reference sequence at which they have been aligned.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-a-bamsam-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect a BAM/SAM file</h3>

  <ol>
    <li>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  </ol>

</blockquote>

<p>A BAM file (or a SAM file, the non-compressed version) consists of:</p>

<ul>
  <li>A header section (the lines starting with <code class="language-plaintext highlighter-rouge">@</code>) containing metadata particularly the chromosome names and lengths (lines starting with the <code class="language-plaintext highlighter-rouge">@SQ</code> symbol)</li>
  <li>
    <p>An alignment section consisting of a table with 11 mandatory fields, as well as a variable number of optional fields:</p>

    <table>
      <thead>
        <tr>
          <th>Col</th>
          <th>Field</th>
          <th>Type</th>
          <th>Brief Description</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td>
          <td>QNAME</td>
          <td>String</td>
          <td>Query template NAME</td>
        </tr>
        <tr>
          <td>2</td>
          <td>FLAG</td>
          <td>Integer</td>
          <td>Bitwise FLAG</td>
        </tr>
        <tr>
          <td>3</td>
          <td><span class="notranslate">RNA</span>ME</td>
          <td>String</td>
          <td>References sequence NAME</td>
        </tr>
        <tr>
          <td>4</td>
          <td>POS</td>
          <td>Integer</td>
          <td>1- based leftmost <span class="notranslate">mapping</span> POSition</td>
        </tr>
        <tr>
          <td>5</td>
          <td>MAPQ</td>
          <td>Integer</td>
          <td><span class="notranslate">MAPping</span> Quality</td>
        </tr>
        <tr>
          <td>6</td>
          <td>CIGAR</td>
          <td>String</td>
          <td>CIGAR String</td>
        </tr>
        <tr>
          <td>7</td>
          <td>RNEXT</td>
          <td>String</td>
          <td>Ref. name of the mate/next read</td>
        </tr>
        <tr>
          <td>8</td>
          <td>PNEXT</td>
          <td>Integer</td>
          <td>Position of the mate/next read</td>
        </tr>
        <tr>
          <td>9</td>
          <td>TLEN</td>
          <td>Integer</td>
          <td>Observed Template LENgth</td>
        </tr>
        <tr>
          <td>10</td>
          <td>SEQ</td>
          <td>String</td>
          <td>Segment SEQuence</td>
        </tr>
        <tr>
          <td>11</td>
          <td>QUAL</td>
          <td>String</td>
          <td>ASCII of Phred-scaled base QUALity+33</td>
        </tr>
      </tbody>
    </table>
  </li>
</ul>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Which information do you find in a SAM/BAM file?</li>
    <li>What is the additional information compared to a FASTQ file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Sequences and quality information, like a FASTQ</li>
      <li><span class="notranslate">Mapping</span> information, Location of the read on the chromosome, <span class="notranslate">Mapping</span> quality, etc</li>
    </ol>
  </blockquote>
</blockquote>

<p>The BAM file includes a lot of information about each read, particularly the quality of <span class="notranslate">mapping</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-summary-of-mapping-quality"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Summary of <span class="notranslate">mapping</span> quality</h3>
  <ol>
    <li><strong>Samtools stats - generate statistics for BAM dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM file‚Äù</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span></code> (output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúUse reference sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
          <ul>
            <li><em>‚ÄúUsing genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mouse (Mus musculus): mm10 Full</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">Stats</code> file</p>

      <blockquote class="question">
        <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the proportion of mismatches in the mapped <span class="notranslate">reads</span> when aligned to the reference genome?</li>
          <li>What does the error rate represent?</li>
          <li>What is the average quality? How is it represented?</li>
          <li>What is the insert size average?</li>
          <li>How many <span class="notranslate">reads</span> have a <span class="notranslate">mapping</span> quality score below 20?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>There are ~21,900 mismatches for ~4,753,900 bases mapped which on average produces ~0.005 mismatches per mapped bases.</li>
            <li>The error rate is the proportion of mismatches per mapped bases, so the ratio computed right before.</li>
            <li>The average quality is the mean quality score of the <span class="notranslate">mapping</span>. It is a Phred score like the one used in the FASTQ file for each nucleotide. But here the score is not per nucleotide, but per read and it represents the probability of <span class="notranslate">mapping</span> quality.</li>
            <li>The insert size is the distance between the two <span class="notranslate">reads</span> in the pairs.</li>
            <li>To get the info:
              <ol>
                <li><strong>Filter BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with a filter to keep only the <span class="notranslate">reads</span> with a <span class="notranslate">mapping</span> quality &gt;= 20</li>
                <li><strong>Stats generate statistics for BAM dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the output of <strong>Filter</strong></li>
              </ol>

              <p>Before filtering: 95,412 <span class="notranslate">reads</span> and after filtering: 89,664 <span class="notranslate">reads</span>.</p>
            </li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="visualization-using-a-genome-browser-igv">Visualization using a Genome Browser (IGV)</h1>

<p>The Integrative Genomics Viewer (IGV) is a high-performance visualization tool for interactive exploration of large, integrated genomic datasets. It supports a wide variety of data types, including array-based and next-generation sequence data, and genomic annotations. In the following, we will use it to visualize the mapped <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-of-the-reads-in-igv"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization of the <span class="notranslate">reads</span> in IGV</h3>

  <p>There are two ways to run it:</p>

  <ol>
    <li>If you have IGV installed (or want to install it):
      <ol>
        <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a></li>
        <li>Launch IGV on your computer</li>
        <li>Expand the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>
      </ol>
    </li>
    <li>If you do not have IGV
      <ol>
        <li>Click on the <code class="language-plaintext highlighter-rouge">Mouse mm10</code> (or correct organism) in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>
      </ol>
    </li>
    <li>Zoom on the <code class="language-plaintext highlighter-rouge">chr2:98,666,236-98,667,473</code></li>
  </ol>
</blockquote>

<p>The <span class="notranslate">reads</span> have a direction: they are mapped to the forward or reverse strand, respectively. When hovering over a read, extra information is displayed</p>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What could it mean if a bar in the coverage view is colored?</li>
    <li>What could be the reason why a read is white instead of grey?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>If a nucleotide differs from the reference sequence in more than 20% of quality weighted <span class="notranslate">reads</span>, IGV colors the bar in proportion to the read count of each base.</li>
      <li>They have a <span class="notranslate">mapping</span> quality equal to zero. Interpretation of this <span class="notranslate">mapping</span> quality depends on the <span class="notranslate">mapping</span> aligner as some commonly used aligners use this convention to mark a read with multiple alignments. In such a case, the read also maps to another location with equally good placement. It is also possible that the read could not be uniquely placed but the other placements do not necessarily give equally good quality hits.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-tips-for-igv"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tips for IGV</h3>
  <ol>
    <li>Because the number of <span class="notranslate">reads</span> over a region can be quite large, the IGV browser by default only displays the <span class="notranslate">reads</span> that fall into a small window. This behaviour can be changed in the IGV from <code class="language-plaintext highlighter-rouge">view &gt; Preferences &gt; Alignments</code>.</li>
    <li>
      <p>If the genome of your interest is not there check if it is available via <strong>More‚Ä¶</strong>. If this is not the case, you can add it manually via the menu <strong>Genomes</strong> -&gt; <strong>Load Genome from‚Ä¶</strong></p>

      <p><img src="../../images/igv_select_genome.png" alt="Select genome in IGV" title="Select genome in IGV" /></p>
    </li>
  </ol>

  <p>A general description of the user interface of the IGV browser is available here: <a href="/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html">IGV Browser description</a></p>
</blockquote>

<h1 id="visualization-using-a-genome-browser-jbrowse">Visualization using a Genome Browser (JBrowse)</h1>

<p>JBrowse is an alte<span class="notranslate">rna</span>tive, web-based genome browser. Whereas IGV is a piece of software you must download and run, JBrowse instances are websites hosted online that provide an interface to browse genomics data. We‚Äôll use it to visualise the mapped <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-of-the-reads-in-jbrowse"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization of the <span class="notranslate">reads</span> in JBrowse</h3>

  <ol>
    <li><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> genome browser, with the following parameters:
      <ul>
        <li><em>‚ÄúReference genome to display‚Äù</em>: Use a built-in genome
          <ul>
            <li><em>‚ÄúSelect a reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">mm10</code></li>
          </ul>
        </li>
        <li><em>‚ÄúProduce Standalone Instance‚Äù</em>: Yes</li>
        <li><em>‚ÄúInsert Track Group‚Äù</em>:
          <ul>
            <li><em>‚ÄúInsert Annotation Track‚Äù</em>:
              <ul>
                <li><em>‚ÄúTrack Type‚Äù</em>: BAM Pileups</li>
                <li><em>‚ÄúBAM Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">aligned <span class="notranslate">reads</span></code> (output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
                <li><em>‚ÄúAutogenerate SNP Track‚Äù</em>: Yes</li>
                <li><em>‚ÄúTrack Visibility‚Äù</em>: On for new users</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Visualise the dataset <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span></li>
    <li>Zoom on the <code class="language-plaintext highlighter-rouge">chr2:98,666,236-98,667,473</code></li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-slow"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Slow</h3>
  <p>This can take about a minute or two to run, depending on your training instance‚Äôs resources. It takes time as the server builds up a small website for you, and pre-processes the reference genome into a more efficient format. If you wanted to share this with your colleagues, you could download this dataset and directly place it on your webserver.</p>
</blockquote>

<p>The <span class="notranslate">reads</span> have a direction: they are mapped to the forward or reverse strand, respectively. When clicking on a read, extra information is displayed</p>

<blockquote class="question">
  <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What do the teardrop shape and line mean in the autogenerated SNP track?</li>
    <li>What do differently coloured <span class="notranslate">reads</span> mean?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>If enough <span class="notranslate">reads</span> have a different value, then it is marked with a teardrop icon. The coverage plot is marked in height with the percentage of <span class="notranslate">reads</span> with a different call at that position</li>
      <li>
        <p>Colour Codes:</p>

        <table>
          <thead>
            <tr>
              <th>Colour</th>
              <th>Meaning</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td><i style="background:#ec8b8b">¬†¬†¬†¬†¬†</i> Original red</td>
              <td>Forward strand</td>
            </tr>
            <tr>
              <td><i style="background:#8f8fd8">¬†¬†¬†¬†¬†</i> Original blue</td>
              <td>Reverse strand</td>
            </tr>
            <tr>
              <td><i style="background:#d11919">¬†¬†¬†¬†¬†</i> Hard red</td>
              <td>Forward strand, missing mate</td>
            </tr>
            <tr>
              <td><i style="background:#1919d1">¬†¬†¬†¬†¬†</i> Hard Blue</td>
              <td>Reverse strand, missing mate</td>
            </tr>
            <tr>
              <td><i style="background:#ecc8c8">¬†¬†¬†¬†¬†</i> Light red</td>
              <td>Forward strand not proper</td>
            </tr>
            <tr>
              <td><i style="background:#bebed8">¬†¬†¬†¬†¬†</i> Light blue</td>
              <td>Reverse strand, not proper</td>
            </tr>
            <tr>
              <td><i style="background:#000000">¬†¬†¬†¬†¬†</i> Black</td>
              <td>Forward, diff chr</td>
            </tr>
            <tr>
              <td><i style="background:#969696">¬†¬†¬†¬†¬†</i> Grey</td>
              <td>Reverse, diff chr</td>
            </tr>
            <tr>
              <td><i style="background:#999999">¬†¬†¬†¬†¬†</i> Grey</td>
              <td>No strand</td>
            </tr>
          </tbody>
        </table>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>After quality control, <span class="notranslate">mapping</span> is an important step of most analyses of sequencing data (<span class="notranslate">RNA</span>-Seq, ChIP-Seq, etc) to determine where in the genome our <span class="notranslate">reads</span> originated from and use this information for <span class="notranslate">downstream</span> analyses.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Know your data!</p>
</li>
                            
                            <li><p>Mapping is not trivial</p>
</li>
                            
                            <li><p>There are many mapping algorithms, it depends on your data which one to choose</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/sequence-analysis#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Mapping (Sequence analysis)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Joachim Wolff, B√©r√©nice Batut, Helena Rasche, 2021 <b>Mapping (Galaxy Training Materials)</b>. <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{sequence-analysis-mapping,
    author = "Joachim Wolff and B√©r√©nice Batut and Helena Rasche",
    title = "Mapping (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
        <blockquote class="agenda follow-up">
            <strong class="follow-up"><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Do you want to extend your knowledge? Follow one of our recommended follow-up trainings:</strong>
            <ul>
                
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Reference-based RNA-Seq data analysis:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/ref-based/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/epigenetics">Epigenetics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Formation of the Super-Structures on the Inactive X:
                            
                                <a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
        </blockquote>
        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Joachim Wolff, B√©r√©nice Batut, Helena Rasche)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/sequence-analysis/tutorials/mapping/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
