<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Formation of the Super-Structures on the Inactive X</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:title" content="Galaxy Training: Formation of the Super-Structures on the Inactive X" />
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Epigenetics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fformation_of_super-structures_on_xi%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fformation_of_super-structures_on_xi%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fformation_of_super-structures_on_xi%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fformation_of_super-structures_on_xi%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fformation_of_super-structures_on_xi%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fformation_of_super-structures_on_xi%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Formation of the Super-Structures on the Inactive X",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
  },
  "courseCode": "epigenetics / formation_of_super-structures_on_xi / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Formation of the Super-Structures on the Inactive X' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html",
  "timeRequired": "PT3H",
  "keywords": "ChIP-seq",
  "description": "The questions this  addresses are:\n - Histone modification?\n - How is a raw set of ChIP-seq data processed and analyzed?\n - Where are the enriched regions for H3K27me3, H3K4me3 and CTCF on the chrX?\n\n\\nThe objectives are:\n - Inspect the read quality\n - Trim low quality bases\n - Map reads on a reference genome\n - Assess the quality of a ChIP-seq experiment\n - Extract coverage files\n - Call enriched regions or peaks\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Friederike D√ºndar"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Vivek Bhardwaj"
    },
    {
      "@type": "Person",
      "name": "Fidel Ramirez"
    },
    {
      "@type": "Person",
      "name": "Leily Rabbani"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Friederike D√ºndar"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Vivek Bhardwaj"
    },
    {
      "@type": "Person",
      "name": "Fidel Ramirez"
    },
    {
      "@type": "Person",
      "name": "Leily Rabbani"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Formation of the Super-Structures on the Inactive X</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/friedue/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/friedue" alt="Friederike D√ºndar">Friederike D√ºndar</a>, <a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/erxleben" alt="Anika Erxleben">Anika Erxleben</a>, <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="B√©r√©nice Batut">B√©r√©nice Batut</a>, <a href="/training-material/hall-of-fame/vivekbhr/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/vivekbhr" alt="Vivek Bhardwaj">Vivek Bhardwaj</a>, <a href="/training-material/hall-of-fame/fidelram/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/fidelram" alt="Fidel Ramirez">Fidel Ramirez</a>, <a href="/training-material/hall-of-fame/LeilyR/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/LeilyR" alt="Leily Rabbani">Leily Rabbani</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>Histone modification?</p>
</li>
            
            <li><p>How is a raw set of ChIP-seq data processed and analyzed?</p>
</li>
            
            <li><p>Where are the enriched regions for H3K27me3, H3K4me3 and CTCF on the chrX?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Inspect the read quality</p>
</li>
            
            <li><p>Trim low quality bases</p>
</li>
            
            <li><p>Map reads on a reference genome</p>
</li>
            
            <li><p>Assess the quality of a ChIP-seq experiment</p>
</li>
            
            <li><p>Extract coverage files</p>
</li>
            
            <li><p>Call enriched regions or peaks</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/1324070">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/workflows/" title="Workflows" alt="Formation of the Super-Structures on the Inactive X workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>Within a cell nucleus, the <span class="notranslate">DNA</span> is tightly-packed and the chromatin is spatially distributed with different levels and scales of organizations.</p>

<p>At the smallest scale, <span class="notranslate">DNA</span> is packaged into units called nucleosomes, made of eight histone proteins.</p>

<p>On a larger scale than nucleosomes, <span class="notranslate">DNA</span> is forming loops. <span class="notranslate">DNA</span> elements that would be otherwise separated by large distances can interact. The corresponding self-interacting (or self-associating) domains are found in many organisms: they are called Topologically Associating Domains (TADs) in mammalian cells. Mammalian chromosomes are also partitioned into two spatial compartments, labeled ‚ÄúA‚Äù and ‚ÄúB‚Äù, where regions in compartment A tend to interact preferentially with A compartment-associated regions than B compartment-associated ones. Similarly, regions in compartment B tend to associate with other B compartment-associated regions.</p>

<blockquote class="comment">
  <h3 id="comment-tip-learn-more-about-chromosome-conformation"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tip: Learn more about chromosome conformation</h3>
  <p>To learn more about chromosome conformation and TADs, you can follow our <a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">HiC tutorial</a></p>
</blockquote>

<p>In mammals, the X chromosome inactivation (XCI) balances the dosage of X-linked genes between females and males. The genes on the inactive X (Xi) chromosome are not expressed.</p>

<p>Binding certain proteins to each of the eight histone proteins may modify the chromatin structure and may result in changes in transcription level. For example, the H3K4me3 is adding 3 methyl-group of the 4th Lysine in the histone 3 amino-acid. This modification is known to activate the transcription on nearby genes by opening the chromatin. The H3K27me3 on the other hand is inactivating the transcription of the nearby genes:</p>

<figure id="figure-1"><img src="../../images/formation_of_super-structures_on_xi/histone_modifications.jpg" alt="Fadloun et al, 2013" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Source: Fadloun et al, 2013</figcaption></figure>

<p>More H3K27me3 and less H3K4me3 on the Xi could explain a lower <span class="notranslate">expression</span> of the genes there.</p>

<p>It has been also observed that the Xi reconfigures uniquely into a specific chromosomal conformation. cohesins, condensins and CCCTC-binding factor (CTCF) play key roles in chromosomal architectures and TAD formation which are other potential cause of the repression of the <span class="notranslate">expression</span> of the genes on Xi.</p>

<p>The structural-maintenance-of-chromosomes hinge domain containing 1 (SMCHD1) has been found enriched on the Xi. It may be the potential actor in the shape of Xi and the change in gene <span class="notranslate">expression</span> there.</p>

<p><a class="citation" href="#Wang2018">Wang <i>et al.</i> 2018</a> investigates the mechanism by which the SMCHD1 gene shapes the Xi and represses the <span class="notranslate">expression</span> of the genes on Xi in mouse.</p>

<p>Their idea was to identify the differences which could be observed between the Xi and activated X chromosome, on both wild-type and SMCHD1 gene knockdown samples to study the SMCHD1 effect.
In different experiments, they targeted histones with H3K27me3 or H3K4me3 and CTCF using ChIP-seq experiments.</p>

<figure id="figure-2"><img src="../../images/formation_of_super-structures_on_xi/Chromatin_immunoprecipitation_sequencing.jpg" alt="ChIP-seq workflow" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Source: http://e.biohackers.net/ChIP-seq</figcaption></figure>

<p>They obtained sequences corresponding to a portion of <span class="notranslate">DNA</span> linked to histones with H3K27me3, H3K4me3 or CTCF. Using this information, they could identify if there are differences in the H3K27me3, H3K4me3 and CTCF between the X (active or inactive) chromosomes and the potentially influenced genes.</p>

<p>In the upcoming tutorial, we will only use the wild type data from <a class="citation" href="#Wang2018">Wang <i>et al.</i> 2018</a> and analyze the ChIP-seq data step by step:</p>

<ul>
  <li>CTCF with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_CTCF_rep2</code></li>
  <li>H3K4me3 with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep2</code></li>
  <li>H3K27me3 with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep2</code></li>
  <li>
    <p>‚Äòinput‚Äô with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></p>

    <p>In ‚Äòinput‚Äô samples, the same treatment as the ChIP-seq samples was done except for the immunoprecipitation step.
  They are also used along with the ‚ÄòChIP-seq‚Äô samples to identify the potential sequencing bias and help for differential analysis.</p>
  </li>
</ul>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

<ol id="markdown-toc">
  <li><a href="#step-1-quality-control-and-treatment-of-the-sequences" id="markdown-toc-step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</a></li>
  <li><a href="#step-2-mapping-of-the-reads" id="markdown-toc-step-2-mapping-of-the-reads">Step 2: <span class="notranslate">Mapping</span> of the <span class="notranslate">reads</span></a>    <ol>
      <li><a href="#running-bowtie2" id="markdown-toc-running-bowtie2">Running Bowtie2</a></li>
      <li><a href="#inspection-of-a-bam-file" id="markdown-toc-inspection-of-a-bam-file">Inspection of a BAM file</a></li>
      <li><a href="#visualization-using-a-genome-browser" id="markdown-toc-visualization-using-a-genome-browser">Visualization using a Genome Browser</a></li>
    </ol>
  </li>
  <li><a href="#step-3-chip-seq-quality-control" id="markdown-toc-step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</a>    <ol>
      <li><a href="#correlation-between-samples" id="markdown-toc-correlation-between-samples">Correlation between samples</a></li>
      <li><a href="#ip-strength-estimation" id="markdown-toc-ip-strength-estimation">IP strength estimation</a></li>
    </ol>
  </li>
  <li><a href="#step-4-normalization" id="markdown-toc-step-4-normalization">Step 4: Normalization</a>    <ol>
      <li><a href="#generation-of-coverage-files-normalized-by-sequencing-depth" id="markdown-toc-generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</a></li>
      <li><a href="#generation-of-input-normalized-coverage-files" id="markdown-toc-generation-of-input-normalized-coverage-files">Generation of input-normalized coverage files</a></li>
    </ol>
  </li>
  <li><a href="#step-5-detecting-enriched-regions-peak-calling" id="markdown-toc-step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</a></li>
  <li><a href="#step-6-plot-the-signal-between-samples" id="markdown-toc-step-6-plot-the-signal-between-samples">Step 6: Plot the signal between samples</a></li>
</ol>

</blockquote>

<h1 id="step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</h1>

<p>The first step of any ChIP-Seq data analysis is quality control of the raw sequencing data.</p>

<p>To save time, we will do it only on the data of one sample <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> which has been already down-sampled. keep in mind that with real data this should be done on each and every sample.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-import-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Import the data</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2.fastq.gz</code> from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1324070/files/wt_H3K4me3_read1.fastq.gz
https://zenodo.org/record/1324070/files/wt_H3K4me3_read2.fastq.gz
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>

      <p>As default, <span class="notranslate">Galaxy</span> takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Rename the files <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code></p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Inspect the first file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (eye) icon (<strong>View data</strong>)</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How are the <span class="notranslate">DNA</span> sequences stored?</li>
    <li>What are the other entries of the file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>The <span class="notranslate">DNA</span> sequences are stored in a FASTQ file, in the second line of every 4-line group.</li>
      <li>This file format is called <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format</a>. It stores sequence information and quality information. Each sequence is represented by a group of 4 lines with the 1st line being the sequence id, the second the sequence of nucleotides, the third a transition line and the last one a sequence of quality score for each nucleotide.</li>
    </ol>
  </blockquote>
</blockquote>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. These are due to the technical limitations of each sequencing platform. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data.</p>

<p>Sequence quality control is therefore an essential first step in your analysis. We use here similar tools as described in <a href="/training-material/topics/sequence-analysis">‚ÄúQuality control‚Äù tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> and <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li>Run <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúShort read data from your current history‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> (Input datasets selected with <strong>Multiple datasets</strong>)</li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select multiple datasets</h3>

        <ol>
          <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
          <li>Select several files by keeping the <kbd>Ctrl</kbd> (or
<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Inspect the generated HTML files</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How is the quality of the <span class="notranslate">reads</span> in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code>?</li>
          <li>And in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code>?</li>
          <li>What should we do if the quality of the <span class="notranslate">reads</span> is not good?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The <span class="notranslate">reads</span> in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> are of good quality:
              <ul>
                <li>There is 50,000 sequences, all of 51 bp</li>
                <li>
                  <p>The ‚ÄúPer base sequence quality‚Äù is not decreasing too much at the end of the sequences</p>

                  <figure id="figure-3"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_quality.png" alt="Per base sequence quality for read1" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Per base sequence quality</figcaption></figure>
                </li>
                <li>
                  <p>The mean quality score over the <span class="notranslate">reads</span> is quite high</p>

                  <figure id="figure-4"><img src="../../images/formation_of_super-structures_on_xi/read1_per_sequence_quality_scores.png" alt="Per sequence quality scores for read1" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Per sequence quality scores</figcaption></figure>
                </li>
                <li>
                  <p>Homogeneous percentage of the bases</p>

                  <figure id="figure-5"><img src="../../images/formation_of_super-structures_on_xi/read1_per base_sequence_content.png" alt="Per base sequence content for read1" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Per base sequence content</figcaption></figure>
                </li>
                <li>
                  <p>No N in the <span class="notranslate">reads</span></p>

                  <figure id="figure-6"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_n_content.png" alt="Per base N content for read1" /><figcaption><span class="figcaption-prefix">Figure 6:</span> Per base N content</figcaption></figure>
                </li>
                <li>
                  <p>No duplicated sequences</p>

                  <figure id="figure-7"><img src="../../images/formation_of_super-structures_on_xi/read1_sequence_duplication_levels.png" alt="Sequence Duplication Levels for read1" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Sequence Duplication Levels</figcaption></figure>
                </li>
                <li>
                  <p>No more known adapters</p>

                  <figure id="figure-8"><img src="../../images/formation_of_super-structures_on_xi/read1_adapter_content.png" alt="Adapter Content for read1" /><figcaption><span class="figcaption-prefix">Figure 8:</span> Adapter Content</figcaption></figure>
                </li>
              </ul>
            </li>
            <li>
              <p>The <span class="notranslate">reads</span> in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> are a bit worse:</p>

              <ul>
                <li>
                  <p>The ‚ÄúPer base sequence quality‚Äù is decreasing more at the end of the sequences, but it stays correct</p>

                  <figure id="figure-9"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_quality.png" alt="Per base sequence quality for read2" /><figcaption><span class="figcaption-prefix">Figure 9:</span> Per base sequence quality</figcaption></figure>
                </li>
                <li>
                  <p>The sequence <code class="language-plaintext highlighter-rouge">NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</code> is over represented</p>
                </li>
              </ul>

              <p>Despite these small things, the overall quality of the <span class="notranslate">reads</span> is really good.</p>
            </li>
            <li>If the quality of the <span class="notranslate">reads</span> is not good, we should:
              <ol>
                <li>Check what is wrong and think about it: it may come from the type of sequencing or what we sequenced (high quantity of overrepresented sequences in transcriptomics data, biaised percentage of bases in HiC data)</li>
                <li>Ask the sequencing facility about it</li>
                <li>Perform some quality treatment (in a reasonable way to not loose too much information) with some trimming or removal of bad <span class="notranslate">reads</span></li>
              </ol>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>It is often necessary to trim sequenced read, for example, to get rid of bases that were sequenced with high uncertainty (= low quality bases) at the read ends.</p>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>What is the relation between <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> ?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The data has been sequenced using paired-end sequencing.</p>

    <p>The paired-end sequencing is based on the idea that the initial <span class="notranslate">DNA</span> fragments (longer than the actual read length) is sequenced from both sides. This approach results in two <span class="notranslate">reads</span> per fragment, with the first read in forward orientation and the second read in reverse-complement orientation. The distance between both <span class="notranslate">reads</span> is known. Thus, it can be used as an additional piece of information to improve the read <span class="notranslate">mapping</span>.</p>

    <p>With paired-end sequencing, each fragment is more covered than with single-end sequencing (only forward orientation sequenced):</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>    ------&gt;
      ------&gt;
        ------&gt;
          ------&gt;

    ----------------------------- [fragment]

    ------&gt;         &lt;------
      ------&gt;         &lt;------
        ------&gt;         &lt;------
          ------&gt;         &lt;------
</code></pre></div>    </div>

    <p>The paired-end sequencing generates then 2 files:</p>
    <ul>
      <li>One file with the sequences corresponding to forward orientation of all the fragments</li>
      <li>One file with the sequences corresponding to reverse orientation of all the fragments</li>
    </ul>

    <p>Here <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> corresponds to the forward <span class="notranslate">reads</span> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> to the reverse <span class="notranslate">reads</span>.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-trimming-low-quality-bases"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Trimming low quality bases</h3>

  <ol>
    <li>Run <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúIs this library paired- or single-end?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">Reads</span> in FASTQ format‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> (Input dataset)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú<span class="notranslate">Reads</span> in FASTQ format‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> (Input dataset)</li>
          </ul>

          <p>The order is important here!</p>

          <blockquote class="tip">
            <h3 id="tip-tip-not-selectable-files"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Not selectable files?</h3>

            <p>If your FASTQ files cannot be selected, you might check whether their format is FASTQ with Sanger-scaled quality values (<code class="language-plaintext highlighter-rouge">fastqsanger</code>). You can edit the data type by clicking on the <code class="language-plaintext highlighter-rouge">pencil</code> symbol.</p>
          </blockquote>
        </li>
        <li><em>‚ÄúTrim Galore! advanced settings‚Äù</em>: <code class="language-plaintext highlighter-rouge">Full parameter list</code>
          <ul>
            <li><em>‚ÄúTrim low-quality ends from <span class="notranslate">reads</span> in addition to adapter removal‚Äù</em>: <code class="language-plaintext highlighter-rouge">15</code></li>
            <li><em>‚ÄúOverlap with adapter sequence required to trim a sequence‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>‚ÄúGenerate a report file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt file (<code class="language-plaintext highlighter-rouge">report file</code>)</p>

      <blockquote class="question">
        <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>How many basepairs has been removed from the forwards <span class="notranslate">reads</span> because of bad quality?</li>
          <li>And from the reverse <span class="notranslate">reads</span>?</li>
          <li>How many sequence pairs have been removed because at least one read was shorter than the length cutoff?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>32,198 bp (1.3%) (first <code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>)</li>
            <li>116,414 bp (4.6%) (second <code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>). It is not a surprise: we saw that the quality was dropping more at the end of the sequences for the reverse <span class="notranslate">reads</span> thant for the forward <span class="notranslate">reads</span>.</li>
            <li>1569 (3.14%) sequences (last line of the file)</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="step-2-mapping-of-the-reads">Step 2: <span class="notranslate">Mapping</span> of the <span class="notranslate">reads</span></h1>

<p>With ChiP sequencing, we obtain sequences corresponding to a portion of <span class="notranslate">DNA</span> linked to the histone mark of interest, H3K4me3 in this case. As H3K4me3 opens the chromatime, nearby genes are gioing to be more transcribed. It would be interesting to know if there is a difference in the quantity of <span class="notranslate">DNA</span> impacted by H3K4me3 and the impacted genes between active and inactive X chromosome.</p>

<p>We first need to figure out where the sequenced <span class="notranslate">DNA</span> fragments originated from in the genome. The <span class="notranslate">reads</span> must be aligned to a reference genome to identify the binding sites.</p>

<p>This is equivalent to solving a jigsaw puzzles, but unfortunately, not all pieces are unique.</p>

<p>In principle, we could do a BLAST analysis to figure out where the sequenced pieces fit best in the known genome. Aligning millions of short sequences this way may, however, take a couple of weeks. And we do not really care about the exact base to base correspondence (<strong>alignment</strong>). What we are really interested in is ‚Äúwhere these <span class="notranslate">reads</span> came from‚Äù. This approach is called <strong><span class="notranslate">mapping</span></strong>.</p>

<p>Nowadays, there are many <span class="notranslate">mapping</span> programs for shotgun sequenced <span class="notranslate">DNA</span>. We will use <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a>.</p>

<blockquote class="question">
  <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What is a reference genome?</li>
    <li>For each model organism, several possible reference genomes may be available (e.g. <code class="language-plaintext highlighter-rouge">hg19</code> and <code class="language-plaintext highlighter-rouge">hg38</code> for human). What do they correspond to?</li>
    <li>Which reference genome should we use?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>A reference genome (or reference assembly) is a set of nucleic acid sequences assembled as a representative example of a species‚Äô genetic material. As they are often assembled from the sequencing of different individuals, they do not accurately represent the set of genes of any single organism, but a mosaic of different nucleic acid sequences from each individual.</li>
      <li>As the cost of <span class="notranslate">DNA</span> sequencing falls, and new full genome sequencing technologies emerge, more genome sequences continue to be generated. Using these new sequences, new alignments are builts and the reference genomes improved (fewer gaps, fixed misrepresentations in the sequence, etc). The different reference genomes correspond to the different released versions (called ‚Äúbuilds‚Äù).</li>
      <li>Wang et al. (2018) did ChIP-seq on mouse. So we should use the mouse reference genome. We will use mm10 (the latest build)</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="running-bowtie2">Running Bowtie2</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mapping"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: <span class="notranslate">Mapping</span></h3>

  <ol>
    <li><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><em>‚ÄúIs this single or paired library‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA/Q file #1‚Äù</em>: <code class="language-plaintext highlighter-rouge">trimmed <span class="notranslate">reads</span> pair 1</code> (output of <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA/Q file #2‚Äù</em>: <code class="language-plaintext highlighter-rouge">trimmed <span class="notranslate">reads</span> pair 2</code> (output of <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúSelect reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mouse (Mus musculus): mm10</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSave the bowtie2 <span class="notranslate">mapping</span> statistics to the history‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the <span class="notranslate">mapping</span> stats</p>

      <blockquote class="question">
        <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How many <span class="notranslate">reads</span> where mapped? Uniquely or several times?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>The overall alignment rate is 98.64%. This score is quite high. If you have less than 70-80%, you should investigate the cause: contamination, etc.</p>

          <p>43719 (90.27%) <span class="notranslate">reads</span> have been aligned concordantly exactly 1 time and 3340 (6.90%) aligned concordantly &gt;1 times. The latter ones correspond to multiple mapped <span class="notranslate">reads</span>. Allowing for multiple  mapped <span class="notranslate">reads</span> increases the number of usable <span class="notranslate">reads</span> and the sensitivity of peak detection;
however, the number of false positives may also increase.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The output of Bowtie2 is a BAM file.</p>

<h2 id="inspection-of-a-bam-file">Inspection of a BAM file</h2>

<p>A BAM (<a href="https://en.wikipedia.org/wiki/SAM_(file_format)">Binary Alignment Map</a>) file is a compressed binary file storing the read sequences, whether they have been aligned to a reference sequence (e.g. a chromosome), and if so, the position on the reference sequence at which they have been aligned.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-a-bamsam-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect a BAM/SAM file</h3>

  <ol>
    <li>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  </ol>

</blockquote>

<p>A BAM file (or a SAM file, the non-compressed version) consists of:</p>

<ul>
  <li>A header section (the lines starting with <code class="language-plaintext highlighter-rouge">@</code>) containing metadata particularly the chromosome names and lengths (lines starting with the <code class="language-plaintext highlighter-rouge">@SQ</code> symbol)</li>
  <li>
    <p>An alignment section consisting of a table with 11 mandatory fields, as well as a variable number of optional fields:</p>

    <table>
      <thead>
        <tr>
          <th>Col</th>
          <th>Field</th>
          <th>Type</th>
          <th>Brief Description</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td>
          <td>QNAME</td>
          <td>String</td>
          <td>Query template NAME</td>
        </tr>
        <tr>
          <td>2</td>
          <td>FLAG</td>
          <td>Integer</td>
          <td>Bitwise FLAG</td>
        </tr>
        <tr>
          <td>3</td>
          <td><span class="notranslate">RNA</span>ME</td>
          <td>String</td>
          <td>References sequence NAME</td>
        </tr>
        <tr>
          <td>4</td>
          <td>POS</td>
          <td>Integer</td>
          <td>1- based leftmost <span class="notranslate">mapping</span> POSition</td>
        </tr>
        <tr>
          <td>5</td>
          <td>MAPQ</td>
          <td>Integer</td>
          <td><span class="notranslate">MAPping</span> Quality</td>
        </tr>
        <tr>
          <td>6</td>
          <td>CIGAR</td>
          <td>String</td>
          <td>CIGAR String</td>
        </tr>
        <tr>
          <td>7</td>
          <td>RNEXT</td>
          <td>String</td>
          <td>Ref. name of the mate/next read</td>
        </tr>
        <tr>
          <td>8</td>
          <td>PNEXT</td>
          <td>Integer</td>
          <td>Position of the mate/next read</td>
        </tr>
        <tr>
          <td>9</td>
          <td>TLEN</td>
          <td>Integer</td>
          <td>Observed Template LENgth</td>
        </tr>
        <tr>
          <td>10</td>
          <td>SEQ</td>
          <td>String</td>
          <td>Segment SEQuence</td>
        </tr>
        <tr>
          <td>11</td>
          <td>QUAL</td>
          <td>String</td>
          <td>ASCII of Phred-scaled base QUALity+33</td>
        </tr>
      </tbody>
    </table>
  </li>
</ul>

<blockquote class="question">
  <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Which information do you find in a SAM/BAM file?</li>
    <li>What is the additional information compared to a FASTQ file?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Sequences and quality information, like a FASTQ</li>
      <li><span class="notranslate">Mapping</span> information, Location of the read on the chromosome, <span class="notranslate">Mapping</span> quality, etc</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="visualization-using-a-genome-browser">Visualization using a Genome Browser</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualization-of-the-reads-in-igv"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualization of the <span class="notranslate">reads</span> in IGV</h3>

  <p>There are two ways to run it:</p>

  <ol>
    <li>If you have IGV installed (or want to install it):
      <ol>
        <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a></li>
        <li>Launch IGV on your computer</li>
        <li>Expand the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>
      </ol>
    </li>
    <li>If you do not have IGV
      <ol>
        <li>Click on the <code class="language-plaintext highlighter-rouge">Mouse mm10</code> (or correct organism) in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the <span class="notranslate">reads</span> into the IGV browser</li>
      </ol>
    </li>
    <li>Zoom on the <code class="language-plaintext highlighter-rouge">chr2:91,053,413-91,055,345</code></li>
  </ol>
</blockquote>

<p>The <span class="notranslate">reads</span> have a direction: they are mapped to the forward or reverse strand, respectively. When hovering over a read, extra information is displayed</p>

<blockquote class="question">
  <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>What could it mean if a bar in the coverage view is colored?</li>
    <li>What could be the reason why a read is white instead of grey?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>If a nucleotide differs from the reference sequence in more than 20% of quality weighted <span class="notranslate">reads</span>, IGV colors the bar in proportion to the read count of each base.</li>
      <li>They have a <span class="notranslate">mapping</span> quality equal to zero. Interpretation of this <span class="notranslate">mapping</span> quality depends on the <span class="notranslate">mapping</span> aligner as some commonly used aligners use this convention to mark a read with multiple alignments. In such a case, the read also maps to another location with equally good placement. It is also possible that the read could not be uniquely placed but the other placements do not necessarily give equally good quality hits.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-tips-for-igv"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Tips for IGV</h3>
  <ol>
    <li>Because the number of <span class="notranslate">reads</span> over a region can be quite large, the IGV browser by default only displays the <span class="notranslate">reads</span> that fall into a small window. This behaviour can be changed in the IGV from <code class="language-plaintext highlighter-rouge">view &gt; Preferences &gt; Alignments</code>.</li>
    <li>
      <p>If the genome of your interest is not there check if it is available via <strong>More‚Ä¶</strong>. If this is not the case, you can add it manually via the menu <strong>Genomes</strong> -&gt; <strong>Load Genome from‚Ä¶</strong></p>

      <p><img src="../../images/igv_select_genome.png" alt="Select genome in IGV" title="Select genome in IGV" /></p>
    </li>
  </ol>

  <p>A general description of the user interface of the IGV browser is available here: <a href="/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html">IGV Browser description</a></p>
</blockquote>

<h1 id="step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</h1>

<p>We already checked the quality of the raw sequencing <span class="notranslate">reads</span> in the first step.
Now we would like to test the quality of the ChIP-seq preparation, to know if our ChIP-seq samples are more enriched than the control (input) samples.</p>

<h2 id="correlation-between-samples">Correlation between samples</h2>

<p>To assess the similarity between the replicates of the ChIP and the input, respectively, it is a common technique to calculate the correlation of read counts on different regions for all different samples.
We expect that the replicates of the ChIP-seq experiments should be clustered more closely to each other than the replicates of the input sample. That is, because the input samples should not have enriched regions included - remember the immuno-precipitation step was skiped during the sample preparation.</p>

<p>To compute the correlation between the samples we are going to to use the QC modules of <a href="http://deeptools.readthedocs.io/">deepTools</a>, a software package for the QC, processing and analysis of NGS data. Before computing the correlation a time consuming step is required, which is to compute the read coverage (number of unique <span class="notranslate">reads</span> mapped at a given nucleotide) over a large number of regions from each of the inputed BAM files. For this we will use the tool <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Then, we use <strong>plotCorrelation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from deepTools to compute and visualize the sample correlation. This is a fast process that allows to quickly try different color combinations and outputs.</p>

<p>Since in this tutorial we are interested in assessing H3K4me3, H3K27me3 and CTCF ChIP samples, the previous steps (quality control and <span class="notranslate">mapping</span>) needs to be run on all the replicates of ChIP samples as well as the input samples. To save time, we have already done that and you can now work directly on the BAM files of the provided 8 samples. For simplicity, the files include only the ChrX.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-correlation-between-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Correlation between samples</h3>

  <ol>
    <li>Create a new history</li>
    <li>
      <p>Import the 8 BAM files from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library into the history</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1324070/files/wt_CTCF_rep1.bam
https://zenodo.org/record/1324070/files/wt_CTCF_rep2.bam
https://zenodo.org/record/1324070/files/wt_H3K4me3_rep1.bam
https://zenodo.org/record/1324070/files/wt_H3K4me3_rep2.bam
https://zenodo.org/record/1324070/files/wt_H3K27me3_rep1.bam
https://zenodo.org/record/1324070/files/wt_H3K27me3_rep2.bam
https://zenodo.org/record/1324070/files/wt_input_rep1.bam
https://zenodo.org/record/1324070/files/wt_input_rep2.bam
</code></pre></div>      </div>
    </li>
    <li>Rename the files</li>
    <li><strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúSample order matters‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM/CRAM file‚Äù</em>: the 8 imported BAM files</li>
          </ul>
        </li>
        <li><em>‚ÄúChoose computation mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Bins</code>
          <ul>
            <li>
              <p><em>‚ÄúBin size in bp‚Äù</em>: <code class="language-plaintext highlighter-rouge">1000</code></p>

              <p>This corresponds to the length of the fragments that were sequenced; it is not the read length!</p>
            </li>
            <li>
              <p><em>‚ÄúDistance between bins‚Äù</em>: <code class="language-plaintext highlighter-rouge">500</code></p>

              <p>It reduces the computation time for the tutorial</p>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúRegion of the genome to limit the operation to‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <p>Using these parameters, the tool will take bins of 1000 bp separated by 500 bp on the chromosome X. For each bin the overlapping <span class="notranslate">reads</span> in each sample will be computed and stored into a matrix.</p>
    </li>
    <li><strong>plotCorrelation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúMatrix file from the multiBamSummary tool‚Äù</em>: <code class="language-plaintext highlighter-rouge">correlation matrix</code>(output of <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúCorrelation method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Pearson</code></li>
      </ul>

      <p>Feel free to try different parameters for the configuration of the plot (colors, title, etc)</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <figure id="figure-10"><img src="../../images/formation_of_super-structures_on_xi/plotCorrelation.png" alt="Output for plotCorrelation with the correlation scores between the 8 samples" /><figcaption><span class="figcaption-prefix">Figure 10:</span> Correlation scores between the 8 samples</figcaption></figure>

  <p>How are your samples clustered? Does that correspond to your expectations?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>As one could expect, the input replicates cluster together and the ChIP replicates cluster together. It confirms that the immuno-precipitation step worked on our ChIP replicates.</p>

    <p>Moreover, for each sample, there is a high correlation between the two replicates which confirms the validity of the experiments.</p>

  </blockquote>
</blockquote>

<h2 id="ip-strength-estimation">IP strength estimation</h2>

<p>To evaluate the quality of the immuno-precipitation(IP) step, we can compute the IP strength. It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of <span class="notranslate">reads</span> in the control sample (‚Äòinput‚Äô). After all, around 90% of all <span class="notranslate">DNA</span> fragments in a ChIP experiment will represent the genomic background.</p>

<p>To do that we take the data from the <code class="language-plaintext highlighter-rouge">rep1</code> of the <code class="language-plaintext highlighter-rouge">wt_H3K4me3</code> ChIP-seq sample and compare it with its corresponding input sample, using <strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of deepTools.</p>

<p>Similar to <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, <strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> randomly samples genome regions of a specified length (bins) and sums the per-base coverage in the indexed BAM files that overlap with those regions. These coverage values are then sorted according to their rank and the cumulative sum of read counts is plotted.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-ip-strength-estimation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: IP strength estimation</h3>

  <ol>
    <li><strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúSample order matters‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM/CRAM file‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code></li>
          </ul>
        </li>
        <li><em>‚ÄúRegion of the genome to limit the operation to‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúNumber of samples‚Äù</em>: <code class="language-plaintext highlighter-rouge">10000</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The plotFingerprint tool generates a fingerprint plot. You need to intepret it to know the IP strength.</p>

<p>An ideal ‚Äòinput‚Äô with perfect uniform distribution of <span class="notranslate">reads</span> along the genome (i.e. without enrichments in open chromatin) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of <span class="notranslate">reads</span> from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.</p>

<figure id="figure-11"><img src="../../images/estrogen-receptor-binding-site-identification/QC_fingerprint.png" alt="A guide to interpret a fingerprint plot" /><figcaption><span class="figcaption-prefix">Figure 11:</span> How to interpret a fingerprint plot? Image extracted from the deepTools documentation</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <figure id="figure-12"><img src="../../images/formation_of_super-structures_on_xi/plotFingerPrint.png" alt="Output for plotFingerprint with the fingerprint plot to estimate the IP strength" /><figcaption><span class="figcaption-prefix">Figure 12:</span> Fingerprint plot for the first replicates to estimate the IP strength</figcaption></figure>

  <p>What do you think about the quality of the IP for this experiment?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>There is clear distinction between H3K4me3 and the input.</p>

    <p>A small percentage of the genome contain a very large fraction of the <span class="notranslate">reads</span> (&gt;70%, point of change in the blue curve)</p>

    <p>The curves start to rise around 0.25. It means that almost 25% of the chromosome X is not sequenced at all.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-optional-hands-on-ip-strength-estimation-other-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> (Optional) Hands-on: IP strength estimation (other samples)</h3>

  <p>Run the same analysis on the other ChIP-seq data along with their corresponding input and compare the output</p>
</blockquote>

<h1 id="step-4-normalization">Step 4: Normalization</h1>

<p>One of the goals in ChIP-seq data analysis is finding regions on the genome which are enriched for the ChIP data of interest (regions with significantly higher read coverage for the ChIP data comparing to its corresponding input). In the following exercise we would like to know where the H3K4me3 sites are. For this we need to extract which parts of the genome have been enriched (i.e. more <span class="notranslate">reads</span> mapped) within the samples that underwent immunoprecipitation. However, to reach a reliable comparison the data needs to be normalized to remove any technical bias. For the normalization we have two steps:</p>

<ol>
  <li>Normalization by sequencing depth</li>
  <li>Normalization by input file</li>
</ol>

<p>To learn how to do the normalization, we will take the <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> sample as ChIP data and <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> as input.</p>

<h2 id="generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-estimation-of-the-sequencing-depth"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Estimation of the sequencing depth</h3>

  <ol>
    <li><strong>IdxStats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM file‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code> and <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>What is the output of <strong>IdxStats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>?</li>
      <li>How many <span class="notranslate">reads</span> has been mapped on chrX for the input and for the ChIP-seq samples?</li>
      <li>Why are the number of <span class="notranslate">reads</span> different? And what could be the impact of this difference?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-10"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>This tool generates a table with 4 columns:  reference sequence identifier, reference sequence length, number of mapped <span class="notranslate">reads</span> and number of placed but unmapped <span class="notranslate">reads</span>. Here it estimates how many <span class="notranslate">reads</span> mapped to which chromosome. Furthermore, it tells the chromosome lengths and naming convention (with or without ‚Äòchr‚Äô in the beginning)</li>
        <li>1,204,821 for ChIP-seq samples and 1,893,595 for the input</li>
        <li>The number of <span class="notranslate">reads</span> can be different because of  different sequencing depth. It can bias the interpretation of the number of <span class="notranslate">reads</span> mapped to a specific genome region and the identification of the H3K4me3 sites. Specially here, as the number of <span class="notranslate">reads</span> for the input is higher than the ChIP data less regions could be identified having a significantly higher read coverage for the ChIP data comparing to the corresponding input.</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<p>The different samples have usually a different sequencing depth, i.e. a different number of <span class="notranslate">reads</span>. These differences can bias the interpretation of the number of <span class="notranslate">reads</span> mapped to a specific genome region. We first need to make the samples comparable by normalizing the coverage by the sequencing depth.</p>

<p>We are using <strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Given a BAM file, this tool generates coverages by first calculating all the number of <span class="notranslate">reads</span> (either extended to match the fragment length or not) that overlap each bin in the genome and then normalizing with various options. It produces a coverage file where for each bin the number of overlapping <span class="notranslate">reads</span> (possibly normalized) is noted.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-coverage-file-normalization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Coverage file normalization</h3>

  <ol>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM file‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code> and <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
        <li><em>‚ÄúBin size in bases‚Äù</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
        <li><em>‚ÄúScaling/Normalization method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Normalize coverage to 1x</code>
          <ul>
            <li><em>‚ÄúEffective genome size‚Äù</em>: <code class="language-plaintext highlighter-rouge">GRCm38/mm10 (2308125349)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúCoverage file format‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code></li>
        <li><em>‚ÄúRegion of the genome to limit the operation to‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is a <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code> file?</li>
          <li>Which regions have the highest coverage in ChIP data and in the input?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-11"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>It is a tabular file with 4 columns: chrom, chromStart, chromEnd and a data value (coverage)</li>
            <li>We can run <strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the 4th column in descending order to get the regions with the highest (normalized) coverage. For <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code>, the regions between 152,233,600	and 152,233,625	 are the most covered. For <code class="language-plaintext highlighter-rouge">wt_input_rep1</code>, between 143,483,000 and 143,483,100.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but
      <ul>
        <li><em>‚ÄúCoverage file format‚Äù</em>: <code class="language-plaintext highlighter-rouge">bigWig</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-12"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>What is a <code class="language-plaintext highlighter-rouge">bigWig</code> file?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-12"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>A <code class="language-plaintext highlighter-rouge">bigWig</code> file is a compressed <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code> file. Similar in relation as BAM to SAM, but this time just for coverage data. This means bigWig and <span class="notranslate">bed</span>graph files are much smaller than BAM or SAM files (no sequence or quality information).</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect both signal coverages (input and ChIP samples) in IGV</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-13"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>If you zoom to <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code>, what do you observe?</p>

  <figure id="figure-13"><img src="../../images/formation_of_super-structures_on_xi/bamcoverage_igv.png" alt="Output of bamCoverage" /><figcaption><span class="figcaption-prefix">Figure 13:</span> bamCoverage for wt_H3K4me3_rep1 and wt_input_rep1 on chrX:151,385,260-152,426,526</figcaption></figure>

  <blockquote class="solution">
    <h3 id="solution-solution-13"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The track with the coverage for the input (<code class="language-plaintext highlighter-rouge">wt_input_rep1</code>) seems quite homogeneous. On the other hand, for <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code>, we can observe some clear peaks at the beginning of the genes.</p>
  </blockquote>
</blockquote>

<h2 id="generation-of-input-normalized-coverage-files">Generation of input-normalized coverage files</h2>

<p>To extract only the information induced by the immunoprecipitation, we normalize the coverage file for the sample that underwent immunoprecipitation by the coverage file for the input sample. Here we use the tool <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> which compare 2 BAM files while caring for sequencing depth normalization.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generation-of-input-normalized-coverage-files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generation of input-normalized coverage files</h3>

  <ol>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFirst BAM file (e.g. treated sample)‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSecond BAM file (e.g. control sample)‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
        <li><em>‚ÄúBin size in bases‚Äù</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
        <li><em>‚ÄúHow to compare the two files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Compute log2 of the number of <span class="notranslate">reads</span> ratio</code></li>
        <li><em>‚ÄúCoverage file format‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span>graph</code></li>
        <li><em>‚ÄúRegion of the genome to limit the operation to‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-14"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What does a positive value or a negative value mean in the 4th column?</li>
          <li>Which regions have the highest coverage in the ChIP data? and the lowest?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-14"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The 4th column contains the log2 of the number of <span class="notranslate">reads</span> ratio between the ChIP-seq sample and the input sample. A positive value means that the coverage on this portion of genome is higher in the ChIP-seq sample than in the input sample</li>
            <li>The highest: 152,233,800-152,233,850 (consistent with the most covered regions in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> given by <strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>). The lowest: 169,916,600-169,916,650</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but:
      <ul>
        <li><em>‚ÄúCoverage file format‚Äù</em>: <code class="language-plaintext highlighter-rouge">bigWig</code></li>
      </ul>
    </li>
    <li><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect the log2 ratio</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-15"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How could you interpret the new track if you zoom to <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code>?</p>

  <figure id="figure-14"><img src="../../images/formation_of_super-structures_on_xi/bamcompare_igv.png" alt="Output of bamCoverage" /><figcaption><span class="figcaption-prefix">Figure 14:</span> bamCoverage for wt_H3K4me3_rep1 and wt_input_rep1 on chrX:151,385,260-152,426,526</figcaption></figure>

  <blockquote class="solution">
    <h3 id="solution-solution-15"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>The new track is the difference between the first track (<code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code>) and the second track (<code class="language-plaintext highlighter-rouge">wt_input_rep1</code>)</p>
  </blockquote>
</blockquote>

<h1 id="step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</h1>

<p>We could see in the ChIP data some enriched regions (peaks). We now would like to call these regions to obtain their coordinates, using <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-peak-calling"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Peak calling</h3>

  <ol>
    <li><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúAre you pooling Treatment Files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúChIP-Seq Treatment File‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code></li>
          </ul>
        </li>
        <li><em>‚ÄúDo you have a Control File?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúAre you pooling Treatment Files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúChIP-Seq Treatment File‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúFormat of Input Files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end BAM</code></li>
        <li><em>‚ÄúEffective genome size‚Äù</em>: <code class="language-plaintext highlighter-rouge">M.musculus (1.87e9)</code></li>
        <li><em>‚ÄúOutputs‚Äù</em>: <code class="language-plaintext highlighter-rouge">Summary page (html)</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
        <p>The advanced options may be adjusted, depending of the samples.
If your ChIP-seq experiment targets regions of broad enrichment, <em>e.g.</em> non-punctuate histone modifications, select calling of broad regions.
If your sample has a low duplication rate (<em>e.g.</em> below 10%), you might keep all duplicate <span class="notranslate">reads</span> (tags). Otherwise, you might use the ‚Äòauto‚Äô option to estimate the maximal allowed number of duplicated <span class="notranslate">reads</span> per genomic location.</p>
      </blockquote>
    </li>
    <li>
      <p>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">(narrow Peaks)</code> file (output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</p>

      <blockquote class="question">
        <h3 id="question-questions-16"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>Which type of files were generated? What do they include?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-16"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> has generated a <span class="notranslate">bed</span> file with the coordinates of the identified peaks: chromosome, start, stop, name, integer score, strand, fold-change, -log10pvalue, -log10qvalue and relative summit position to peak start, as well as a html report which contains links to additional <span class="notranslate">bed</span> and xls files.</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect with the signal coverage and log2 ratio tracks</li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-17"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>
      <p>How many peaks have been identified in <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code> based on IGV?</p>

      <figure id="figure-15"><img src="../../images/formation_of_super-structures_on_xi/macs2_igv.png" alt="Output of MACS2" /><figcaption><span class="figcaption-prefix">Figure 15:</span> Peaks for wt_H3K4me3_rep1 and wt_input_rep1 on chrX:151,385,260-152,426,526</figcaption></figure>
    </li>
    <li>What are the fold change of the peaks identified in <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code>? Hint: using the <span class="notranslate">BED</span> file</li>
    <li>How many peaks have been identified on the full chromosome X? How many peaks have a fold change &gt; 50?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-17"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>We can see 11 peaks (track below the genes).</li>
      <li>Using <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">c2&gt;151385260 and c3&lt;152426526</code>, we found that the 11 peaks with fold changes between 3.81927 and 162.06572</li>
      <li>On the 656 peaks on the full chromosome (number of lines of the original <span class="notranslate">BED</span> file) there are 252 peaks with FC&gt;50 (using <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">c7&gt;50</code>)</li>
    </ol>
  </blockquote>
</blockquote>

<p>The called peak regions can be filtered by, <em>e.g.</em> fold change, FDR and region length for further <span class="notranslate">downstream</span> analysis.</p>

<h1 id="step-6-plot-the-signal-between-samples">Step 6: Plot the signal between samples</h1>

<p>So far, we have normalized the data and identified peaks. Now, we would like to visualize scores associated with certain genomic regions, for example ChIP enrichment values around the TSS of genes. Moreover, we would like to compare the enrichment of several ChIP samples (e.g. CTCF and H3K4me3 )on the regions of interest.</p>

<p>Since we already generated the required files for the H3K4me3 sample, let‚Äôs make them only for the CTCF sample:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-prepare-the-peaks-and-data-for-ctcf"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Prepare the peaks and data for CTCF</h3>

  <ol>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFirst BAM file (e.g. treated sample)‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1.bam</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSecond BAM file (e.g. control sample)‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
        <li><em>‚ÄúBin size in bases‚Äù</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
        <li><em>‚ÄúHow to compare the two files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Compute log2 of the number of <span class="notranslate">reads</span> ratio</code></li>
        <li><em>‚ÄúCoverage file format‚Äù</em>: <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li><em>‚ÄúRegion of the genome to limit the operation to‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>
    </li>
    <li>Rename the output of <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the name of the sample</li>
    <li><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúAre you pooling Treatment Files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúChIP-Seq Treatment File‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1.bam</code></li>
          </ul>
        </li>
        <li><em>‚ÄúDo you have a Control File?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúAre you pooling Treatment Files?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúChIP-Seq Treatment File‚Äù</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúFormat of Input Files‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end BAM</code></li>
        <li><em>‚ÄúEffective genome size‚Äù</em>: <code class="language-plaintext highlighter-rouge">M.musculus (1.87e9)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We can now concatenate the MACS2 outputs with the location of the peaks (concatenate the files and merge the overlapping regions) to obtain one <span class="notranslate">BED</span> file corresponding to the coordinates of the interesting regions to plot.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-prepare-the-peak-coordinates"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Prepare the peak coordinates</h3>

  <ol>
    <li><strong>Concatenate two datasets into one dataset</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConcatenate‚Äù</em>: output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúwith‚Äù</em>: output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code></li>
      </ul>
    </li>
    <li><strong>SortBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSort the following <span class="notranslate">bed</span>,<span class="notranslate">bed</span>graph,gff,vcf file‚Äù</em>: output of <strong>Concatenate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>MergeBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSort the following <span class="notranslate">bed</span>,<span class="notranslate">bed</span>graph,gff,vcf file‚Äù</em>: output of <strong>SortBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>To plot the the peaks score on the region generated above (MergeBED output) two tools from the <a href="http://deeptools.readthedocs.io/">deepTools</a> package are used:</p>
<ul>
  <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: it computes the signal on given regions, using the <code class="language-plaintext highlighter-rouge">bigwig</code> coverage files from different samples.</li>
  <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: it plots heatMap of the signals using the <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output.</li>
</ul>

<p>Optionally, we can also use <strong>plotProfile</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to create a profile plot from <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-the-heatmap"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot the heatmap</h3>

  <ol>
    <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúSelect regions‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRegions to plot‚Äù</em>: output of <strong>MergeBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>‚ÄúSample order matters‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúScore file‚Äù</em>: the 2 <code class="language-plaintext highlighter-rouge">bigwig</code> files generated by <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and renamed</li>
          </ul>
        </li>
        <li><em>‚ÄúcomputeMatrix has two main output options‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference-point</code>
          <ul>
            <li><em>‚ÄúThe reference point for the plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">center of region</code></li>
            <li><em>‚ÄúDistance <span class="notranslate">upstream</span> of the start site of the regions defined in the region file‚Äù</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
            <li><em>‚ÄúDistance <span class="notranslate">downstream</span> of the end site of the given regions‚Äù</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMatrix file from the computeMatrix tool‚Äù</em>: <code class="language-plaintext highlighter-rouge">Matrix</code> (output of <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>‚ÄúReference point label‚Äù</em>: select the right label</li>
            <li><em>‚ÄúDid you compute the matrix with more than one groups of regions?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, I used only one group</code>
              <ul>
                <li><em>‚ÄúClustering algorithm‚Äù</em>: <code class="language-plaintext highlighter-rouge">Kmeans clustering</code></li>
                <li><em>‚ÄúNumber of clusters to compute‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>It should generate an heatmap similar to:</p>

<figure id="figure-16"><img src="../../images/formation_of_super-structures_on_xi/peak_heatmap_1.png" alt="Output of plotHeatmap for 2 samples" /><figcaption><span class="figcaption-prefix">Figure 16:</span> Scores around the peaks for wt_H3K4me3_rep1 and wt_input_rep1</figcaption></figure>

<p>When we look at this graph, it seems that less but larger peaks are found for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and that only few peaks are shared.</p>

<blockquote class="question">
  <h3 id="question-questions-18"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many peaks have been found for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> and for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code>?</li>
    <li>What are the mean width of the peaks for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> and for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code>?</li>
    <li>How many peaks are specific to <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> or <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code>?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-18"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>656 peaks for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and 2,688 for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> (number of lines in the <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <span class="notranslate">BED</span> file)</li>
      <li>1630.77 bp for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and 404.55 for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> (<strong>Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">c3-c2</code> and then <strong>Datamash</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">Mean</code> on <code class="language-plaintext highlighter-rouge">Column:11</code>)</li>
      <li>443 peaks (over 656) are specific to <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and 2,464 (over 2,688) to <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> (<strong>Intersect intervals</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>). Around 220 peaks are then overlapping.</li>
    </ol>
  </blockquote>
</blockquote>

<p>So far, we have only analyzed 2 samples, but we can do the same for all the 6 samples:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-optional-hands-on-plot-the-heatmap-for-all-the-samples"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> (Optional) Hands-on: Plot the heatmap for all the samples</h3>

  <ol>
    <li><strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for each combination input - ChIP data:
      <ol>
        <li><code class="language-plaintext highlighter-rouge">wt_CTCF_rep1</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> (already done)</li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> (already done)</li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep1</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep1</code></li>
        <li><code class="language-plaintext highlighter-rouge">wt_CTCF_rep2</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep2</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep2</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></li>
      </ol>
    </li>
    <li>Rename the outputs of <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the name of the ChIP data</li>
    <li><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for each combination input - ChIP data</li>
    <li><strong>Concatenate datasets tail-to-head</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúConcatenate Dataset‚Äù</em>: one output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Click <em>‚ÄúInsert Dataset‚Äù</em> and <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect‚Äù</em> one other output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Redo for the 6 outputs of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>SortBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSort the following <span class="notranslate">bed</span>,<span class="notranslate">bed</span>graph,gff,vcf file‚Äù</em>: output of <strong>Concatenate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>MergeBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSort the following <span class="notranslate">bed</span>,<span class="notranslate">bed</span>graph,gff,vcf file‚Äù</em>: output of <strong>SortBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the same parameters but:
      <ul>
        <li><em>‚ÄúSelect regions‚Äù</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúRegions to plot‚Äù</em>: output of <strong>MergeBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>‚ÄúSample order matters‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúScore file‚Äù</em>: the 6 <code class="language-plaintext highlighter-rouge">bigwig</code> files generated by <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and renamed</li>
          </ul>
        </li>
        <li><em>‚ÄúcomputeMatrix has two main output options‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference-point</code>
          <ul>
            <li><em>‚ÄúThe reference point for the plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">center of region</code></li>
            <li><em>‚ÄúDistance <span class="notranslate">upstream</span> of the start site of the regions defined in the region file‚Äù</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
            <li><em>‚ÄúDistance <span class="notranslate">downstream</span> of the end site of the given regions‚Äù</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMatrix file from the computeMatrix tool‚Äù</em>: <code class="language-plaintext highlighter-rouge">Matrix</code> (output of <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúShow advanced options‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>‚ÄúReference point label‚Äù</em>: select the right label</li>
            <li><em>‚ÄúDid you compute the matrix with more than one groups of regions?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, I used only one group</code>
              <ul>
                <li><em>‚ÄúClustering algorithm‚Äù</em>: <code class="language-plaintext highlighter-rouge">Kmeans clustering</code></li>
                <li><em>‚ÄúNumber of clusters to compute‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-19"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many peaks are found for the different samples?</li>
    <li>
      <p>How are the peaks?</p>

      <figure id="figure-17"><img src="../../images/formation_of_super-structures_on_xi/peak_heatmap_all.png" alt="Output of plotHeatmap for all samples" /><figcaption><span class="figcaption-prefix">Figure 17:</span> Scores around the peaks for all samples</figcaption></figure>
    </li>
    <li>
      <p>How could be interpreted the peaks and read coverage in the <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code> region?</p>

      <figure id="figure-18"><img src="../../images/formation_of_super-structures_on_xi/rep1_igv.png" alt="Coverage and peaks for the replicate 1" /><figcaption><span class="figcaption-prefix">Figure 18:</span> Coverage and peaks for the replicate 1</figcaption></figure>
    </li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-19"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>
        <p>Found peaks (number of lines in <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs):</p>

        <table>
          <thead>
            <tr>
              <th>Target</th>
              <th>Rep 1</th>
              <th>Rep 2</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>CTCF</td>
              <td>2,688</td>
              <td>2,062</td>
            </tr>
            <tr>
              <td>H3K4me3</td>
              <td>656</td>
              <td>717</td>
            </tr>
            <tr>
              <td>H3K27me3</td>
              <td>221</td>
              <td>76</td>
            </tr>
          </tbody>
        </table>

        <p>The tendencies are similar for both replicates: more peaks for CTCF, less for H3K4me3 and only few for H3K27me3.</p>
      </li>
      <li>As observed with the 2 samples, the peaks for H3K4me3 are wider than for CTCF. We also observe that the peaks found with one replicate are found with the other replicate.</li>
      <li>
        <p>The H3K4me3 sample has clear and large regions in which the read coverage are enriched. H3K4me3 is one of the least abundant histone modifications. It is highly enriched at active promoters near transcription start sites (TSS) and positively correlated with transcription.</p>

        <p>For H3K27me3, the coverage is more homogeneous. A gene is a broad domain of H3K27me3 enrichment across its body of genes corresponds to a gene with a transcription inhibited by H3K27me3. We can also identified some ‚Äúbivalent‚Äù genes: gene with a peak around the TSS for H3K27me3(e.g. region_208 for gene Gpr173) but also H3K4me3. We also observe some H3K27me3-depleted regions sharply demarcated, with boundaries coinciding with gene borders (e.g. Kdm5c). This is a chromatin signature reminiscent of genes that escape XCI.</p>

        <p>To reproduce, run <strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, <strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Along this tutorial, we learn how to extract peaks and coverage information from raw data of ChIP experiments:</p>

<figure id="figure-19"><img src="../../images/formation_of_super-structures_on_xi/tutorial-scheme.png" alt="Workflow to extract peaks and coverage information from raw data of ChIP experiments" /><figcaption><span class="figcaption-prefix">Figure 19:</span> <span class="notranslate">Workflow</span> to extract peaks and coverage information from raw data of ChIP experiments</figcaption></figure>

<p>This information can be then related to the biology to answer the original question. We tried then to relate the observed differences of peak and read coverage between H3K27me3 and H3K4me3 to the known biology. We could go even further in the analysis to reproduce the results of the original paper (e.g. by looking at the bivalent genes, identifying the differences between Xa and Xi).</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>ChIP-seq data requires multiple methods of quality assessment to ensure that the data is of high quality.</p>
</li>
                            
                            <li><p>Multiple normalization methods exists depending on the availability of input data.</p>
</li>
                            
                            <li><p>Heatmaps containing all genes of an organism can be easily plotted given a BED file and a coverage file.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Wang2018">Wang, C.-Y., T. J√©gu, H.-P. Chu, H. J. Oh, and J. T. Lee, 2018 <b>SMCHD1 Merges Chromosome Compartments and Assists Formation of Super-Structures on the Inactive X</b>. Cell 174: 406‚Äì421.e25.</span>
    
    
    <a href="https://doi.org/10.1016/j.cell.2018.05.007">10.1016/j.cell.2018.05.007</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Formation of the Super-Structures on the Inactive X (Epigenetics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez, Leily Rabbani, 2021 <b>Formation of the Super-Structures on the Inactive X (Galaxy Training Materials)</b>. <a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{epigenetics-formation_of_super-structures_on_xi,
    author = "Friederike D√ºndar and Anika Erxleben and B√©r√©nice Batut and Vivek Bhardwaj and Fidel Ramirez and Leily Rabbani",
    title = "Formation of the Super-Structures on the Inactive X (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Friederike D√ºndar, Anika Erxleben, B√©r√©nice Batut, Vivek Bhardwaj, Fidel Ramirez, Leily Rabbani)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
