<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Formation of the Super-Structures on the Inactive X / Formation of the Super-Structures on the Inactive X / Epigenetics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@5012353c3c3462ce860e77e8fb34393bb8bbd8fb",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Formation of the Super-Structures on the Inactive X">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-09 08:55:09 +0000">
<meta name="DC.creator" content="Friederike Dündar">
<meta name="DC.creator" content="Anika Erxleben">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Vivek Bhardwaj">
<meta name="DC.creator" content="Fidel Ramirez">
<meta name="DC.creator" content="Leily Rabbani">
<meta name="DC.creator" content="Pavankumar Videm"><meta name="description" content="DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Epigenetics / Formation of the Super-Structures on the Inactive X / Hands-on: Formation of the Super-Structures on the Inactive X">
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Formation of the Super-Structures on the Inactive X",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2023-11-09 08:55:09 +0000",
  "datePublished": "2018-09-17 07:13:05 +0000",
  "identifier": "https://gxy.io/GTN:T00140",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
  },
  "abstract": "Within a cell nucleus, the DNA is tightly-packed and the chromatin is spatially distributed with different levels and scales of organizations.",
  "learningResourceType": "e-learning",
  "name": "Formation of the Super-Structures on the Inactive X",
  "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html",
  "version": 12,
  "timeRequired": "PT3H",
  "teaches": "- Inspect the read quality\n- Trim low quality bases\n- Map reads on a reference genome\n- Assess the quality of a ChIP-seq experiment\n- Extract coverage files\n- Call enriched regions or peaks",
  "keywords": [
    "Epigenetics",
    "ChIP-seq"
  ],
  "description": "## Abstract\n\nWithin a cell nucleus, the DNA is tightly-packed and the chromatin is spatially distributed with different levels and scales of organizations.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - Histone modification?\n - How is a raw set of ChIP-seq data processed and analyzed?\n - Where are the enriched regions for H3K27me3, H3K4me3 and CTCF on the chrX?\n\n\n## Learning Objectives\n\n- Inspect the read quality\n- Trim low quality bases\n- Map reads on a reference genome\n- Assess the quality of a ChIP-seq experiment\n- Extract coverage files\n- Call enriched regions or peaks\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/friedue/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/friedue/",
      "name": "Friederike Dündar",
      "image": "https://avatars.githubusercontent.com/friedue",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "name": "Anika Erxleben",
      "image": "https://avatars.githubusercontent.com/erxleben",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eurosciencegateway/",
          "name": "EuroScienceGateway",
          "description": "EuroScienceGateway was funded by the European Union programme Horizon Europe (HORIZON-INFRA-2021-EOSC-01-04) under grant agreement number 101057388 and by UK Research and Innovation (UKRI) under the UK government’s Horizon Europe funding guarantee grant number 10038963.\n",
          "url": "https://galaxyproject.org/projects/esg/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7427-6478",
      "orcid": "https://orcid.org/0000-0002-7427-6478"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "This project ([`2020-1-NL01-KA203-064717`](https://erasmus-plus.ec.europa.eu/projects/search/details/2020-1-NL01-KA203-064717)) is funded with the support of the Erasmus+ programme of the European Union. Their funding has supported a large number of tutorials within the GTN across a wide array of topics.\n![eu flag with the text: with the support of the erasmus programme of the european union](https://gallantries.github.io/assets/images/logosbeneficaireserasmusright_en.jpg)\n![erasmus plus logo](https://www.erasmusplus.nl/assets/images/logo.png)\n",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/ifb/",
          "name": "Institut Français de Bioinformatique",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.france-bioinformatique.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/vivekbhr/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/vivekbhr/",
      "name": "Vivek Bhardwaj",
      "image": "https://avatars.githubusercontent.com/vivekbhr",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/fidelram/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/fidelram/",
      "name": "Fidel Ramirez",
      "image": "https://avatars.githubusercontent.com/fidelram",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/LeilyR/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/LeilyR/",
      "name": "Leily Rabbani",
      "image": "https://avatars.githubusercontent.com/LeilyR",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "name": "Pavankumar Videm",
      "image": "https://avatars.githubusercontent.com/pavanvidem",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5192-126X",
      "orcid": "https://orcid.org/0000-0002-5192-126X"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/1324070",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Epigenetics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/epigenetics/faqs/" title="Check our FAQs for the Epigenetics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-epigenetics">
    <h1 data-toc-skip>Formation of the Super-Structures on the Inactive X</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/friedue/" class="contributor-badge contributor-friedue"><img src="https://avatars.githubusercontent.com/friedue?s=36" alt="Friederike Dündar avatar" width="36" class="avatar" />
    Friederike Dündar</a><a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge contributor-erxleben"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/erxleben?s=36" alt="Anika Erxleben avatar" width="36" class="avatar" />
    Anika Erxleben</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Bérénice Batut avatar" width="36" class="avatar" />
    Bérénice Batut</a><a href="/training-material/hall-of-fame/vivekbhr/" class="contributor-badge contributor-vivekbhr"><img src="https://avatars.githubusercontent.com/vivekbhr?s=36" alt="Vivek Bhardwaj avatar" width="36" class="avatar" />
    Vivek Bhardwaj</a><a href="/training-material/hall-of-fame/fidelram/" class="contributor-badge contributor-fidelram"><img src="https://avatars.githubusercontent.com/fidelram?s=36" alt="Fidel Ramirez avatar" width="36" class="avatar" />
    Fidel Ramirez</a><a href="/training-material/hall-of-fame/LeilyR/" class="contributor-badge contributor-LeilyR"><img src="https://avatars.githubusercontent.com/LeilyR?s=36" alt="Leily Rabbani avatar" width="36" class="avatar" />
    Leily Rabbani</a><a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" />
    Pavankumar Videm</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Histone modification?</p>
</li>
        
        <li><p>How is a raw set of ChIP-seq data processed and analyzed?</p>
</li>
        
        <li><p>Where are the enriched regions for H3K27me3, H3K4me3 and CTCF on the chrX?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Inspect the read quality</p>
</li>
        
        <li><p>Trim low quality bases</p>
</li>
        
        <li><p>Map reads on a reference genome</p>
</li>
        
        <li><p>Assess the quality of a ChIP-seq experiment</p>
</li>
        
        <li><p>Extract coverage files</p>
</li>
        
        <li><p>Call enriched regions or peaks</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    

    
        
        
        
            
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Quality Control</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Quality Control</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mapping</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mapping</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 3 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/1324070">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/workflows/" title="Formation of the Super-Structures on the Inactive X workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://galaxy.mesocentre.uca.fr" title="">
			Galaxy@AuBi
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
    

    <!--
    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    -->
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Sep 17, 2018 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 9, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00140">https://gxy.io/GTN:T00140</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">4.0</a> (0 recent ratings, 8 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 12 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Within a cell nucleus, the DNA is tightly-packed and the chromatin is spatially distributed with different levels and scales of organizations.</p>

<p>At the smallest scale, DNA is packaged into units called nucleosomes, made of eight histone proteins.</p>

<p>On a larger scale than nucleosomes, DNA is forming loops. DNA elements that would be otherwise separated by large distances can interact. The corresponding self-interacting (or self-associating) domains are found in many organisms: they are called Topologically Associating Domains (TADs) in mammalian cells. Mammalian chromosomes are also partitioned into two spatial compartments, labeled “A” and “B”, where regions in compartment A tend to interact preferentially with A compartment-associated regions than B compartment-associated ones. Similarly, regions in compartment B tend to associate with other B compartment-associated regions.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-tip-learn-more-about-chromosome-conformation"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Tip: Learn more about chromosome conformation</div>
  <p>To learn more about chromosome conformation and TADs, you can follow our <a href="/training-material/topics/epigenetics/tutorials/hicexplorer/tutorial.html">HiC tutorial</a></p>
</blockquote>

<p>In mammals, the X chromosome inactivation (XCI) balances the dosage of X-linked genes between females and males. The genes on the inactive X (Xi) chromosome are not expressed.</p>

<p>Binding certain molecules to histone proteins can modify the chromatin structure and may result in changes in transcription level. For example, the H3K4me3 results from adding 3 methyl-group to the 4th lysine in the histone 3 protein. This modification is known to activate the transcription on nearby genes by opening the chromatin. The H3K27me3 on the other hand is inactivating the transcription of the nearby genes:</p>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/histone_modifications.jpg" alt="Fadloun et al, 2013. " width="478" height="254" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/histone_modifications.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Source: Fadloun et al, 2013</figcaption></figure>

<p>More H3K27me3 and less H3K4me3 on the Xi could explain a lower expression of the genes there.</p>

<p>It has been also observed that the Xi reconfigures uniquely into a specific chromosomal conformation. cohesins, condensins and CCCTC-binding factor (CTCF) play key roles in chromosomal architectures and TAD formation which are other potential cause of the repression of the expression of the genes on Xi.</p>

<p>The structural-maintenance-of-chromosomes hinge domain containing 1 (SMCHD1) has been found enriched on the Xi. It may be the potential actor in the shape of Xi and the change in gene expression there.</p>

<p><span class="citation"><a href="#Wang2018">Wang <i>et al.</i> 2018</a></span> investigates the mechanism by which the SMCHD1 gene shapes the Xi and represses the expression of the genes on Xi in mouse.</p>

<p>Their idea was to identify the differences which could be observed between the Xi and activated X chromosome, on both wild-type and SMCHD1 gene knockdown samples to study the SMCHD1 effect.
In different experiments, they targeted histones with H3K27me3 or H3K4me3 and CTCF using ChIP-seq experiments.</p>

<figure id="figure-2" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/Chromatin_immunoprecipitation_sequencing.jpg" alt="ChIP-seq workflow. " width="700" height="933" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/Chromatin_immunoprecipitation_sequencing.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Source: http://e.biohackers.net/ChIP-seq</figcaption></figure>

<p>They obtained sequences corresponding to a portion of DNA linked to histones with H3K27me3, H3K4me3 or CTCF. Using this information, they could identify if there are differences in the H3K27me3, H3K4me3 and CTCF between the X (active or inactive) chromosomes and the potentially influenced genes.</p>

<p>In the upcoming tutorial, we will only use the wild type data from <span class="citation"><a href="#Wang2018">Wang <i>et al.</i> 2018</a></span> and analyze the ChIP-seq data step by step:</p>

<ul>
  <li>CTCF with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_CTCF_rep2</code></li>
  <li>H3K4me3 with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep2</code></li>
  <li>H3K27me3 with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep2</code></li>
  <li>
    <p>‘input’ with 2 replicates: <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></p>

    <p>In ‘input’ samples, the same treatment as the ChIP-seq samples was done except for the immunoprecipitation step.
  They are also used along with the ‘ChIP-seq’ samples to identify the potential sequencing bias and help for differential analysis.</p>
  </li>
</ul>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

<ol id="markdown-toc">
  <li><a href="#step-1-quality-control-and-treatment-of-the-sequences" id="markdown-toc-step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</a></li>
  <li><a href="#step-2-mapping-of-the-reads" id="markdown-toc-step-2-mapping-of-the-reads">Step 2: Mapping of the reads</a>    <ol>
      <li><a href="#running-bowtie2" id="markdown-toc-running-bowtie2">Running Bowtie2</a></li>
      <li><a href="#inspection-of-a-bam-file" id="markdown-toc-inspection-of-a-bam-file">Inspection of a BAM file</a></li>
      <li><a href="#visualization-using-a-genome-browser" id="markdown-toc-visualization-using-a-genome-browser">Visualization using a Genome Browser</a></li>
    </ol>
  </li>
  <li><a href="#step-3-chip-seq-quality-control" id="markdown-toc-step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</a>    <ol>
      <li><a href="#correlation-between-samples" id="markdown-toc-correlation-between-samples">Correlation between samples</a></li>
      <li><a href="#ip-strength-estimation" id="markdown-toc-ip-strength-estimation">IP strength estimation</a></li>
    </ol>
  </li>
  <li><a href="#step-4-normalization" id="markdown-toc-step-4-normalization">Step 4: Normalization</a>    <ol>
      <li><a href="#generation-of-coverage-files-normalized-by-sequencing-depth" id="markdown-toc-generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</a></li>
      <li><a href="#generation-of-input-normalized-coverage-files" id="markdown-toc-generation-of-input-normalized-coverage-files">Generation of input-normalized coverage files</a></li>
    </ol>
  </li>
  <li><a href="#step-5-detecting-enriched-regions-peak-calling" id="markdown-toc-step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</a></li>
  <li><a href="#step-6-plot-the-signal-between-samples" id="markdown-toc-step-6-plot-the-signal-between-samples">Step 6: Plot the signal between samples</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="step-1-quality-control-and-treatment-of-the-sequences">Step 1: Quality control and treatment of the sequences</h1>

<p>The first step of any ChIP-Seq data analysis is quality control of the raw sequencing data.</p>

<p>To save time, we will do it only on the data of one sample <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> which has been already down-sampled. keep in mind that with real data this should be done on each and every sample.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-the-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import the data</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2.fastq.gz</code> from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library (ask your instructor)</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1324070/files/wt_H3K4me3_read1.fastq.gz
https://zenodo.org/record/1324070/files/wt_H3K4me3_read2.fastq.gz
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>As default, Galaxy takes the link as name, so rename them.</p>
    </li>
    <li>
      <p>Rename the files <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Inspect the first file by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon (<strong>View data</strong>)</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How are the DNA sequences stored?</li>
    <li>What are the other entries of the file?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>The DNA sequences are stored in a FASTQ file, in the second line of every 4-line group.</li>
      <li>This file format is called <a href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ format</a>. It stores sequence information and quality information. Each sequence is represented by a group of 4 lines with the 1st line being the sequence id, the second the sequence of nucleotides, the third a transition line and the last one a sequence of quality score for each nucleotide.</li>
    </ol>
  </blockquote>
</blockquote>

<p>During sequencing, errors are introduced, such as incorrect nucleotides being called. These are due to the technical limitations of each sequencing platform. Sequencing errors might bias the analysis and can lead to a misinterpretation of the data.</p>

<p>Sequence quality control is therefore an essential first step in your analysis. We use here similar tools as described in <a href="/training-material/topics/sequence-analysis">“Quality control” tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> and <a href="https://www.bioinformatics.babraham.ac.uk/projects/trim_galore/">Trim Galore</a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality control</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.73+galaxy0" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.73+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> (Input datasets selected with <strong>Multiple datasets</strong>)</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Inspect the generated HTML files</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How is the quality of the reads in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code>?</li>
          <li>And in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code>?</li>
          <li>What should we do if the quality of the reads is not good?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>The reads in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> are of good quality:
              <ul>
                <li>There is 50,000 sequences, all of 51 bp</li>
                <li>
                  <p>The “Per base sequence quality” is not decreasing too much at the end of the sequences</p>

                  <figure id="figure-3" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_quality.png" alt="Per base sequence quality for read1. " width="600" height="438" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_quality.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Per base sequence quality</figcaption></figure>
                </li>
                <li>
                  <p>The mean quality score over the reads is quite high</p>

                  <figure id="figure-4" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_per_sequence_quality_scores.png" alt="Per sequence quality scores for read1. " width="500" height="366" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_per_sequence_quality_scores.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Per sequence quality scores</figcaption></figure>
                </li>
                <li>
                  <p>Homogeneous percentage of the bases</p>

                  <figure id="figure-5" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_content.png" alt="Per base sequence content for read1. " width="500" height="362" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_content.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Per base sequence content</figcaption></figure>
                </li>
                <li>
                  <p>No N in the reads</p>

                  <figure id="figure-6" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_n_content.png" alt="Per base N content for read1. " width="500" height="364" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_per_base_n_content.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Per base N content</figcaption></figure>
                </li>
                <li>
                  <p>No duplicated sequences</p>

                  <figure id="figure-7" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_sequence_duplication_levels.png" alt="Sequence Duplication Levels for read1. " width="500" height="366" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_sequence_duplication_levels.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Sequence Duplication Levels</figcaption></figure>
                </li>
                <li>
                  <p>No more known adapters</p>

                  <figure id="figure-8" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_adapter_content.png" alt="Adapter Content for read1. " width="500" height="365" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_adapter_content.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span> Adapter Content</figcaption></figure>
                </li>
              </ul>
            </li>
            <li>
              <p>The reads in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> are a bit worse:</p>

              <ul>
                <li>
                  <p>The “Per base sequence quality” is decreasing more at the end of the sequences, but it stays correct</p>

                  <figure id="figure-9" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_quality.png" alt="Per base sequence quality for read2. " width="600" height="438" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/read1_per_base_sequence_quality.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> Per base sequence quality</figcaption></figure>
                </li>
                <li>
                  <p>The sequence <code class="language-plaintext highlighter-rouge">NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN</code> is over represented</p>
                </li>
              </ul>

              <p>Despite these small things, the overall quality of the reads is really good.</p>
            </li>
            <li>If the quality of the reads is not good, we should:
              <ol>
                <li>Check what is wrong and think about it: it may come from the type of sequencing or what we sequenced (high quantity of overrepresented sequences in transcriptomics data, biaised percentage of bases in HiC data)</li>
                <li>Ask the sequencing facility about it</li>
                <li>Perform some quality treatment (in a reasonable way to not loose too much information) with some trimming or removal of bad reads</li>
              </ol>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>It is often necessary to trim sequenced read, for example, to get rid of bases that were sequenced with high uncertainty (= low quality bases) at the read ends.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What is the relation between <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> ?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The data has been sequenced using paired-end sequencing.</p>

    <p>The paired-end sequencing is based on the idea that the initial DNA fragments (longer than the actual read length) is sequenced from both sides. This approach results in two reads per fragment, with the first read in forward orientation and the second read in reverse-complement orientation. The distance between both reads is known. Thus, it can be used as an additional piece of information to improve the read mapping.</p>

    <p>With paired-end sequencing, each fragment is more covered than with single-end sequencing (only forward orientation sequenced):</p>

    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>    ------&gt;
      ------&gt;
        ------&gt;
          ------&gt;

    ----------------------------- [fragment]

    ------&gt;         &lt;------
      ------&gt;         &lt;------
        ------&gt;         &lt;------
          ------&gt;         &lt;------
</code></pre></div>    </div>

    <p>The paired-end sequencing generates then 2 files:</p>
    <ul>
      <li>One file with the sequences corresponding to forward orientation of all the fragments</li>
      <li>One file with the sequences corresponding to reverse orientation of all the fragments</li>
    </ul>

    <p>Here <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> corresponds to the forward reads and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> to the reverse reads.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-trimming-low-quality-bases"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Trimming low quality bases</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/trim_galore/trim_galore/0.6.7+galaxy0" title="Trim Galore! tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Trim Galore!</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.6.7+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Is this library paired- or single-end?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reads in FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read1</code> (Input dataset)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Reads in FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_read2</code> (Input dataset)</li>
          </ul>

          <p>The order is important here!</p>

          <blockquote class="tip">
            <div class="box-title tip-title" id="tip-not-selectable-files"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-not-selectable-files" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Not selectable files?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

            <p>If your FASTQ files cannot be selected, you might check whether their format is FASTQ with Sanger-scaled quality values (<code class="language-plaintext highlighter-rouge">fastqsanger</code>). You can edit the data type by clicking on the <code class="language-plaintext highlighter-rouge">pencil</code> symbol.</p>
          </blockquote>
        </li>
        <li><em>“Trim Galore! advanced settings”</em>: <code class="language-plaintext highlighter-rouge">Full parameter list</code>
          <ul>
            <li><em>“Trim low-quality ends from reads in addition to adapter removal”</em>: <code class="language-plaintext highlighter-rouge">15</code></li>
            <li><em>“Overlap with adapter sequence required to trim a sequence”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>“Generate a report file”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated txt file (<code class="language-plaintext highlighter-rouge">report file</code>)</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How many basepairs has been removed from the forwards reads because of bad quality?</li>
          <li>And from the reverse reads?</li>
          <li>How many sequence pairs have been removed because at least one read was shorter than the length cutoff?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>32,198 bp (1.3%) (first <code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>)</li>
            <li>116,414 bp (4.6%) (second <code class="language-plaintext highlighter-rouge">Quality-trimmed:</code>). It is not a surprise: we saw that the quality was dropping more at the end of the sequences for the reverse reads thant for the forward reads.</li>
            <li>1569 (3.14%) sequences (last line of the file)</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="step-2-mapping-of-the-reads">Step 2: Mapping of the reads</h1>

<p>With ChiP sequencing, we obtain sequences corresponding to a portion of DNA linked to the histone mark of interest, H3K4me3 in this case. As H3K4me3 opens the chromatime, nearby genes are gioing to be more transcribed. It would be interesting to know if there is a difference in the quantity of DNA impacted by H3K4me3 and the impacted genes between active and inactive X chromosome.</p>

<p>We first need to figure out where the sequenced DNA fragments originated from in the genome. The reads must be aligned to a reference genome to identify the binding sites.</p>

<p>This is equivalent to solving a jigsaw puzzles, but unfortunately, not all pieces are unique.</p>

<p>In principle, we could do a BLAST analysis to figure out where the sequenced pieces fit best in the known genome. Aligning millions of short sequences this way may, however, take a couple of weeks. And we do not really care about the exact base to base correspondence (<strong>alignment</strong>). What we are really interested in is “where these reads came from”. This approach is called <strong>mapping</strong>.</p>

<p>Nowadays, there are many mapping programs for shotgun sequenced DNA. We will use <a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie2</a>.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What is a reference genome?</li>
    <li>For each model organism, several possible reference genomes may be available (e.g. <code class="language-plaintext highlighter-rouge">hg19</code> and <code class="language-plaintext highlighter-rouge">hg38</code> for human). What do they correspond to?</li>
    <li>Which reference genome should we use?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>A reference genome (or reference assembly) is a set of nucleic acid sequences assembled as a representative example of a species’ genetic material. As they are often assembled from the sequencing of different individuals, they do not accurately represent the set of genes of any single organism, but a mosaic of different nucleic acid sequences from each individual.</li>
      <li>As the cost of DNA sequencing falls, and new full genome sequencing technologies emerge, more genome sequences continue to be generated. Using these new sequences, new alignments are built and the reference genomes improved (fewer gaps, fixed misrepresentations in the sequence, etc). The different reference genomes correspond to the different released versions (called “builds”).</li>
      <li>Wang et al. (2018) did ChIP-seq on mouse. So we should use the mouse reference genome. We will use mm10 (the latest build)</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="running-bowtie2">Running Bowtie2</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapping"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapping</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0" title="Bowtie2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Bowtie2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.5.0+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Is this single or paired library”</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA/Q file #1”</em>: <code class="language-plaintext highlighter-rouge">trimmed reads pair 1</code> (output of <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA/Q file #2”</em>: <code class="language-plaintext highlighter-rouge">trimmed reads pair 2</code> (output of <strong>Trim Galore!</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          </ul>
        </li>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Select reference genome”</em>: <code class="language-plaintext highlighter-rouge">Mouse (Mus musculus): mm10</code></li>
          </ul>
        </li>
        <li><em>“Save the bowtie2 mapping statistics to the history”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the mapping stats</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>How many reads where mapped? Uniquely or several times?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>The overall alignment rate is 98.57%. This score is quite high. If you have less than 70-80%, you should investigate the cause: contamination, etc.</p>

          <p>41514 (85.72%) reads have been aligned concordantly exactly 1 time and 5190 (10.72%) aligned concordantly &gt;1 times. The latter ones correspond to multiple mapped reads. Allowing for multiple  mapped reads increases the number of usable reads and the sensitivity of peak detection;
however, the number of false positives may also increase.</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The output of Bowtie2 is a BAM file.</p>

<h2 id="inspection-of-a-bam-file">Inspection of a BAM file</h2>

<p>A BAM (<a href="https://en.wikipedia.org/wiki/SAM_(file_format)">Binary Alignment Map</a>) file is a compressed binary file storing the read sequences, whether they have been aligned to a reference sequence (e.g. a chromosome), and if so, the position on the reference sequence at which they have been aligned.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-inspect-a-bam-sam-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Inspect a BAM/SAM file</div>

  <ol>
    <li>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  </ol>

</blockquote>

<p>A BAM file (or a SAM file, the non-compressed version) consists of:</p>

<ul>
  <li>A header section (the lines starting with <code class="language-plaintext highlighter-rouge">@</code>) containing metadata particularly the chromosome names and lengths (lines starting with the <code class="language-plaintext highlighter-rouge">@SQ</code> symbol)</li>
  <li>
    <p>An alignment section consisting of a table with 11 mandatory fields, as well as a variable number of optional fields:</p>

    <table>
      <thead>
        <tr>
          <th>Col</th>
          <th>Field</th>
          <th>Type</th>
          <th>Brief Description</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>1</td>
          <td>QNAME</td>
          <td>String</td>
          <td>Query template NAME</td>
        </tr>
        <tr>
          <td>2</td>
          <td>FLAG</td>
          <td>Integer</td>
          <td>Bitwise FLAG</td>
        </tr>
        <tr>
          <td>3</td>
          <td>RNAME</td>
          <td>String</td>
          <td>References sequence NAME</td>
        </tr>
        <tr>
          <td>4</td>
          <td>POS</td>
          <td>Integer</td>
          <td>1- based leftmost mapping POSition</td>
        </tr>
        <tr>
          <td>5</td>
          <td>MAPQ</td>
          <td>Integer</td>
          <td>MAPping Quality</td>
        </tr>
        <tr>
          <td>6</td>
          <td>CIGAR</td>
          <td>String</td>
          <td>CIGAR String</td>
        </tr>
        <tr>
          <td>7</td>
          <td>RNEXT</td>
          <td>String</td>
          <td>Ref. name of the mate/next read</td>
        </tr>
        <tr>
          <td>8</td>
          <td>PNEXT</td>
          <td>Integer</td>
          <td>Position of the mate/next read</td>
        </tr>
        <tr>
          <td>9</td>
          <td>TLEN</td>
          <td>Integer</td>
          <td>Observed Template LENgth</td>
        </tr>
        <tr>
          <td>10</td>
          <td>SEQ</td>
          <td>String</td>
          <td>Segment SEQuence</td>
        </tr>
        <tr>
          <td>11</td>
          <td>QUAL</td>
          <td>String</td>
          <td>ASCII of Phred-scaled base QUALity+33</td>
        </tr>
      </tbody>
    </table>
  </li>
</ul>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>Which information do you find in a SAM/BAM file?</li>
    <li>What is the additional information compared to a FASTQ file?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>Sequences and quality information, like a FASTQ</li>
      <li>Mapping information, Location of the read on the chromosome, Mapping quality, etc</li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="visualization-using-a-genome-browser">Visualization using a Genome Browser</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualization-of-the-reads-in-igv"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Visualization of the reads in IGV</div>

  <p>There are two ways to run it:</p>

  <ol>
    <li>If you have IGV installed (or want to install it):
      <ol>
        <li>Install <a href="https://software.broadinstitute.org/software/igv/download">IGV</a></li>
        <li>Launch IGV on your computer</li>
        <li>Expand the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> output of <strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Click on the <code class="language-plaintext highlighter-rouge">local</code> in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the reads into the IGV browser</li>
      </ol>
    </li>
    <li>If you do not have IGV
      <ol>
        <li>Click on the <code class="language-plaintext highlighter-rouge">Mouse mm10</code> (or correct organism) in <code class="language-plaintext highlighter-rouge">display with IGV</code> to load the reads into the IGV browser</li>
      </ol>
    </li>
    <li>Zoom on the <code class="language-plaintext highlighter-rouge">chr2:91,053,413-91,055,345</code></li>
  </ol>
</blockquote>

<p>The reads have a direction: they are mapped to the forward or reverse strand, respectively. When hovering over a read, extra information is displayed</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>What could it mean if a bar in the coverage view is colored?</li>
    <li>What could be the reason why a read is white instead of grey?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>If a nucleotide differs from the reference sequence in more than 20% of quality weighted reads, IGV colors the bar in proportion to the read count of each base.</li>
      <li>They have a mapping quality equal to zero. Interpretation of this mapping quality depends on the mapping aligner as some commonly used aligners use this convention to mark a read with multiple alignments. In such a case, the read also maps to another location with equally good placement. It is also possible that the read could not be uniquely placed but the other placements do not necessarily give equally good quality hits.</li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-tips-for-igv"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Tips for IGV</div>
  <ol>
    <li>Because the number of reads over a region can be quite large, the IGV browser by default only displays the reads that fall into a small window. This behaviour can be changed in the IGV from <code class="language-plaintext highlighter-rouge">view &gt; Preferences &gt; Alignments</code>.</li>
    <li>
      <p>If the genome of your interest is not there check if it is available via <strong>More…</strong>. If this is not the case, you can add it manually via the menu <strong>Genomes</strong> -&gt; <strong>Load Genome from…</strong></p>

      <p><img src="../../images/igv_select_genome.png" alt="Select genome in IGV" title="Select genome in IGV" /></p>
    </li>
  </ol>

  <p>A general description of the user interface of the IGV browser is available here: <a href="/training-material/topics/introduction/tutorials/igv-introduction/tutorial.html">IGV Browser description</a></p>
</blockquote>

<h1 id="step-3-chip-seq-quality-control">Step 3: ChIP-seq Quality Control</h1>

<p>We already checked the quality of the raw sequencing reads in the first step.
Now we would like to test the quality of the ChIP-seq preparation, to know if our ChIP-seq samples are more enriched than the control (input) samples.</p>

<h2 id="correlation-between-samples">Correlation between samples</h2>

<p>To assess the similarity between the replicates of the ChIP and the input, respectively, it is a common technique to calculate the correlation of read counts on different regions for all different samples.
We expect that the replicates of the ChIP-seq experiments should be clustered more closely to each other than the replicates of the input sample. That is, because the input samples should not have enriched regions included - remember the immuno-precipitation step was skiped during the sample preparation.</p>

<p>To compute the correlation between the samples we are going to to use the QC modules of <a href="http://deeptools.readthedocs.io/">deepTools</a>, a software package for the QC, processing and analysis of NGS data. Before computing the correlation a time consuming step is required, which is to compute the read coverage (number of unique reads mapped at a given nucleotide) over a large number of regions from each of the inputed BAM files. For this we will use the tool <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Then, we use <strong>plotCorrelation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> from deepTools to compute and visualize the sample correlation. This is a fast process that allows to quickly try different color combinations and outputs.</p>

<p>Since in this tutorial we are interested in assessing H3K4me3, H3K27me3 and CTCF ChIP samples, the previous steps (quality control and mapping) needs to be run on all the replicates of ChIP samples as well as the input samples. To save time, we have already done that and you can now work directly on the BAM files of the provided 8 samples. For simplicity, the files include only the ChrX.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-correlation-between-samples"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Correlation between samples</div>

  <ol>
    <li>Create a new history</li>
    <li>
      <p>Import the 8 BAM files from <a href="https://zenodo.org/record/1324070">Zenodo</a> or from the data library into the history</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1324070/files/wt_CTCF_rep1.bam
https://zenodo.org/record/1324070/files/wt_CTCF_rep2.bam
https://zenodo.org/record/1324070/files/wt_H3K4me3_rep1.bam
https://zenodo.org/record/1324070/files/wt_H3K4me3_rep2.bam
https://zenodo.org/record/1324070/files/wt_H3K27me3_rep1.bam
https://zenodo.org/record/1324070/files/wt_H3K27me3_rep2.bam
https://zenodo.org/record/1324070/files/wt_input_rep1.bam
https://zenodo.org/record/1324070/files/wt_input_rep2.bam
</code></pre></div>      </div>
    </li>
    <li>Rename the files</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_multi_bam_summary/deeptools_multi_bam_summary/3.5.1.0.0" title="multiBamSummary tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>multiBamSummary</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><em>“Sample order matters”</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM/CRAM file”</em>: the 8 imported BAM files</li>
          </ul>
        </li>
        <li><em>“Choose computation mode”</em>: <code class="language-plaintext highlighter-rouge">Bins</code>
          <ul>
            <li>
              <p><em>“Bin size in bp”</em>: <code class="language-plaintext highlighter-rouge">1000</code></p>

              <p>This corresponds to the length of the fragments that were sequenced; it is not the read length!</p>
            </li>
            <li>
              <p><em>“Distance between bins”</em>: <code class="language-plaintext highlighter-rouge">500</code></p>

              <p>It reduces the computation time for the tutorial</p>
            </li>
          </ul>
        </li>
        <li><em>“Region of the genome to limit the operation to”</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <p>Using these parameters, the tool will take bins of 1000 bp separated by 500 bp on the chromosome X. For each bin the overlapping reads in each sample will be computed and stored into a matrix.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_correlation/deeptools_plot_correlation/3.5.1.0.0" title="plotCorrelation tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>plotCorrelation</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Matrix file from the multiBamSummary tool”</em>: <code class="language-plaintext highlighter-rouge">correlation matrix</code>(output of <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Correlation method”</em>: <code class="language-plaintext highlighter-rouge">Pearson</code></li>
      </ul>

      <p>Feel free to try different parameters for the configuration of the plot (colors, title, etc)</p>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-8"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <figure id="figure-10" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/plotCorrelation.png" alt="Output for plotCorrelation with the correlation scores between the 8 samples. " width="1100" height="950" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/plotCorrelation.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> Correlation scores between the 8 samples</figcaption></figure>

  <p>How are your samples clustered? Does that correspond to your expectations?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-8"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-8" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>As one could expect, the input replicates cluster together and the ChIP replicates cluster together. It confirms that the immuno-precipitation step worked on our ChIP replicates.</p>

    <p>Moreover, for each sample, there is a high correlation between the two replicates which confirms the validity of the experiments.</p>

  </blockquote>
</blockquote>

<h2 id="ip-strength-estimation">IP strength estimation</h2>

<p>To evaluate the quality of the immuno-precipitation(IP) step, we can compute the IP strength. It determines how well the signal in the ChIP-seq sample can be differentiated from the background distribution of reads in the control sample (‘input’). After all, around 90% of all DNA fragments in a ChIP experiment will represent the genomic background.</p>

<p>To do that we take the data from the <code class="language-plaintext highlighter-rouge">rep1</code> of the <code class="language-plaintext highlighter-rouge">wt_H3K4me3</code> ChIP-seq sample and compare it with its corresponding input sample, using <strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of deepTools.</p>

<p>Similar to <strong>multiBamSummary</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, <strong>plotFingerprint</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> randomly samples genome regions of a specified length (bins) and sums the per-base coverage in the indexed BAM files that overlap with those regions. These coverage values are then sorted according to their rank and the cumulative sum of read counts is plotted.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-ip-strength-estimation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: IP strength estimation</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_fingerprint/deeptools_plot_fingerprint/3.5.1.0.0" title="plotFingerprint tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>plotFingerprint</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><em>“Sample order matters”</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM/CRAM file”</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> and <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code></li>
          </ul>
        </li>
        <li><em>“Region of the genome to limit the operation to”</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
        <li><em>“Show advanced options”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Number of samples”</em>: <code class="language-plaintext highlighter-rouge">10000</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>The plotFingerprint tool generates a fingerprint plot. You need to intepret it to know the IP strength.</p>

<p>An ideal ‘input’ with perfect uniform distribution of reads along the genome (i.e. without enrichments in open chromatin) and infinite sequencing coverage should generate a straight diagonal line. A very specific and strong ChIP enrichment will be indicated by a prominent and steep rise of the cumulative sum towards the highest rank. This means that a big chunk of reads from the ChIP sample is located in few bins which corresponds to high, narrow enrichments typically seen for transcription factors.</p>

<figure id="figure-11" style="max-width: 90%;"><img src="../../images/estrogen-receptor-binding-site-identification/QC_fingerprint.png" alt="A guide to interpret a fingerprint plot. " width="1436" height="722" loading="lazy" /><a target="_blank" href="../../images/estrogen-receptor-binding-site-identification/QC_fingerprint.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 11</strong>:</span> How to interpret a fingerprint plot? Image extracted from the deepTools documentation</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-9"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <figure id="figure-12" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/plotFingerPrint.png" alt="Output for plotFingerprint with the fingerprint plot to estimate the IP strength. " width="640" height="480" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/plotFingerPrint.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 12</strong>:</span> Fingerprint plot for the first replicates to estimate the IP strength</figcaption></figure>

  <p>What do you think about the quality of the IP for this experiment?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-9"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-9" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>There is clear distinction between H3K4me3 and the input.</p>

    <p>A small percentage of the genome contain a very large fraction of the reads (&gt;70%, point of change in the blue curve)</p>

    <p>The curves start to rise around 0.25. It means that almost 25% of the chromosome X is not sequenced at all.</p>
  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-optional-ip-strength-estimation-other-samples"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: (Optional) IP strength estimation (other samples)</div>

  <p>Run the same analysis on the other ChIP-seq data along with their corresponding input and compare the output</p>
</blockquote>

<h1 id="step-4-normalization">Step 4: Normalization</h1>

<p>One of the goals in ChIP-seq data analysis is finding regions on the genome which are enriched for the ChIP data of interest (regions with significantly higher read coverage for the ChIP data comparing to its corresponding input). In the following exercise we would like to know where the H3K4me3 sites are. For this we need to extract which parts of the genome have been enriched (i.e. more reads mapped) within the samples that underwent immunoprecipitation. However, to reach a reliable comparison the data needs to be normalized to remove any technical bias. For the normalization we have two steps:</p>

<ol>
  <li>Normalization by sequencing depth</li>
  <li>Normalization by input file</li>
</ol>

<p>To learn how to do the normalization, we will take the <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> sample as ChIP data and <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> as input.</p>

<h2 id="generation-of-coverage-files-normalized-by-sequencing-depth">Generation of coverage files normalized by sequencing depth</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-estimation-of-the-sequencing-depth"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Estimation of the sequencing depth</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_idxstats/samtools_idxstats/2.0.4" title="Samtools idxstats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Samtools idxstats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.4)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM file”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code> and <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <div class="box-title question-title" id="question-10"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>What is the output of <strong>Samtools idxstats</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>?</li>
      <li>How many reads has been mapped on chrX for the input and for the ChIP-seq samples?</li>
      <li>Why are the number of reads different? And what could be the impact of this difference?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-10"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-10" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <ol>
        <li>This tool generates a table with 4 columns:  reference sequence identifier, reference sequence length, number of mapped reads and number of placed but unmapped reads. Here it estimates how many reads mapped to which chromosome. Furthermore, it tells the chromosome lengths and naming convention (with or without ‘chr’ in the beginning)</li>
        <li>1,204,821 for ChIP-seq samples and 1,893,595 for the input</li>
        <li>The number of reads can be different because of  different sequencing depth. It can bias the interpretation of the number of reads mapped to a specific genome region and the identification of the H3K4me3 sites. Specially here, as the number of reads for the input is higher than the ChIP data less regions could be identified having a significantly higher read coverage for the ChIP data comparing to the corresponding input.</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<p>The different samples have usually a different sequencing depth, i.e. a different number of reads. These differences can bias the interpretation of the number of reads mapped to a specific genome region. We first need to make the samples comparable by normalizing the coverage by the sequencing depth.</p>

<p>We are using <strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. Given a BAM file, this tool generates coverages by first calculating all the number of reads (either extended to match the fragment length or not) that overlap each bin in the genome and then normalizing with various options. It produces a coverage file where for each bin the number of overlapping reads (possibly normalized) is noted.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-coverage-file-normalization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Coverage file normalization</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bam_coverage/deeptools_bam_coverage/3.5.1.0.0" title="bamCoverage tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bamCoverage</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM file”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code> and <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
        <li><em>“Bin size in bases”</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
        <li><em>“Scaling/Normalization method”</em>: <code class="language-plaintext highlighter-rouge">Normalize coverage to 1x</code>
          <ul>
            <li><em>“Effective genome size”</em>: <code class="language-plaintext highlighter-rouge">GRCm38/mm10 (2308125349)</code></li>
          </ul>
        </li>
        <li><em>“Coverage file format”</em>: <code class="language-plaintext highlighter-rouge">bedgraph</code></li>
        <li><em>“Region of the genome to limit the operation to”</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-11"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What is a <code class="language-plaintext highlighter-rouge">bedgraph</code> file?</li>
          <li>Which regions have the highest coverage in ChIP data and in the input?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-11"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-11" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>It is a tabular file with 4 columns: chrom, chromStart, chromEnd and a data value (coverage)</li>
            <li>We can run <strong>Sort</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the 4th column in descending order to get the regions with the highest (normalized) coverage. For <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code>, the regions between 152,233,600	and 152,233,625	 are the most covered. For <code class="language-plaintext highlighter-rouge">wt_input_rep1</code>, between 143,483,000 and 143,483,100.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bam_coverage/deeptools_bam_coverage/3.5.1.0.0" title="bamCoverage tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bamCoverage</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the same parameters but
      <ul>
        <li><em>“Coverage file format”</em>: <code class="language-plaintext highlighter-rouge">bigWig</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-12"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>What is a <code class="language-plaintext highlighter-rouge">bigWig</code> file?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-12"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-12" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p>A <code class="language-plaintext highlighter-rouge">bigWig</code> file is a compressed <code class="language-plaintext highlighter-rouge">bedgraph</code> file. Similar in relation as BAM to SAM, but this time just for coverage data. This means bigWig and bedgraph files are much smaller than BAM or SAM files (no sequence or quality information).</p>
        </blockquote>
      </blockquote>
    </li>
    <li>Use <strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect both signal coverages (input and ChIP samples).</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-13"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>If you zoom to <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code>, what do you observe?</p>

  <figure id="figure-13" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/bamcoverage_igv.png" alt="Output of bamCoverage. " width="1265" height="337" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/bamcoverage_igv.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 13</strong>:</span> bamCoverage for wt_H3K4me3_rep1 and wt_input_rep1 on chrX:151,385,260-152,426,526</figcaption></figure>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-13"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-13" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The track with the coverage for the input (<code class="language-plaintext highlighter-rouge">wt_input_rep1</code>) seems quite homogeneous. On the other hand, for <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code>, we can observe some clear peaks at the beginning of the genes.</p>
  </blockquote>
</blockquote>

<h2 id="generation-of-input-normalized-coverage-files">Generation of input-normalized coverage files</h2>

<p>To extract only the information induced by the immunoprecipitation, we normalize the coverage file for the sample that underwent immunoprecipitation by the coverage file for the input sample. Here we use the tool <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> which compare 2 BAM files while caring for sequencing depth normalization.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generation-of-input-normalized-coverage-files"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generation of input-normalized coverage files</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bam_compare/deeptools_bam_compare/3.5.1.0.0" title="bamCompare tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bamCompare</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First BAM file (e.g. treated sample)”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second BAM file (e.g. control sample)”</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
        <li><em>“Bin size in bases”</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
        <li><em>“How to compare the two files”</em>: <code class="language-plaintext highlighter-rouge">Compute log2 of the number of reads ratio</code></li>
        <li><em>“Coverage file format”</em>: <code class="language-plaintext highlighter-rouge">bedgraph</code></li>
        <li><em>“Region of the genome to limit the operation to”</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-14"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What does a positive value or a negative value mean in the 4th column?</li>
          <li>Which regions have the highest coverage in the ChIP data? and the lowest?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-14"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-14" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>The 4th column contains the log2 of the number of reads ratio between the ChIP-seq sample and the input sample. A positive value means that the coverage on this portion of genome is higher in the ChIP-seq sample than in the input sample</li>
            <li>The highest: 152,233,800-152,233,850 (consistent with the most covered regions in <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> given by <strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>). The lowest: 169,916,600-169,916,650</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bam_compare/deeptools_bam_compare/3.5.1.0.0" title="bamCompare tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bamCompare</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the same parameters but:
      <ul>
        <li><em>“Coverage file format”</em>: <code class="language-plaintext highlighter-rouge">bigWig</code></li>
      </ul>
    </li>
    <li>Use <strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect the log2 ratio.</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-15"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How could you interpret the new track if you zoom to <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code>?</p>

  <figure id="figure-14" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/bamcompare_igv.png" alt="Output of bamCoverage. " width="1266" height="336" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/bamcompare_igv.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 14</strong>:</span> bamCoverage for wt_H3K4me3_rep1 and wt_input_rep1 on chrX:151,385,260-152,426,526</figcaption></figure>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-15"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-15" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>The new track is the difference between the first track (<code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code>) and the second track (<code class="language-plaintext highlighter-rouge">wt_input_rep1</code>)</p>
  </blockquote>
</blockquote>

<h1 id="step-5-detecting-enriched-regions-peak-calling">Step 5: Detecting enriched regions (peak calling)</h1>

<p>We could see in the ChIP data some enriched regions (peaks). We now would like to call these regions to obtain their coordinates, using <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-peak-calling"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Peak calling</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0" title="MACS2 callpeak tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MACS2 callpeak</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.7.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Are you pooling Treatment Files?”</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“ChIP-Seq Treatment File”</em>: <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1.bam</code></li>
          </ul>
        </li>
        <li><em>“Do you have a Control File?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Are you pooling Treatment Files?”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“ChIP-Seq Treatment File”</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Format of Input Files”</em>: <code class="language-plaintext highlighter-rouge">Paired-end BAM</code></li>
        <li><em>“Effective genome size”</em>: <code class="language-plaintext highlighter-rouge">M.musculus (1.87e9)</code></li>
        <li><em>“Outputs”</em>: <code class="language-plaintext highlighter-rouge">Summary page (html)</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <p>The advanced options may be adjusted, depending of the samples.
If your ChIP-seq experiment targets regions of broad enrichment, <em>e.g.</em> non-punctuate histone modifications, select calling of broad regions.
If your sample has a low duplication rate (<em>e.g.</em> below 10%), you might keep all duplicate reads (tags). Otherwise, you might use the ‘auto’ option to estimate the maximal allowed number of duplicated reads per genomic location.</p>
      </blockquote>
    </li>
    <li>
      <p>Inspect the <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <code class="language-plaintext highlighter-rouge">(narrow Peaks)</code> file (output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question-16"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>Which type of files were generated? What do they include?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-16"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-16" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <p><strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> has generated a bed file with the coordinates of the identified peaks: chromosome, start, stop, name, integer score, strand, fold-change, -log10pvalue, -log10qvalue and relative summit position to peak start, as well as a html report which contains links to additional bed and xls files.</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to inspect with the signal coverage and log2 ratio tracks</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-17"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>
      <p>How many peaks have been identified in <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code> based on IGV?</p>

      <figure id="figure-15" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/macs2_igv.png" alt="Output of MACS2. " width="1267" height="393" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/macs2_igv.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 15</strong>:</span> Peaks for wt_H3K4me3_rep1 and wt_input_rep1 on chrX:151,385,260-152,426,526</figcaption></figure>
    </li>
    <li>What are the fold change of the peaks identified in <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code>? Hint: using the BED file</li>
    <li>How many peaks have been identified on the full chromosome X? How many peaks have a fold change &gt; 50?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-17"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-17" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>We can see 11 peaks (track below the genes).</li>
      <li>Using <span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> with <code class="language-plaintext highlighter-rouge">c2&gt;151385260 and c3&lt;152426526</code>, we found that the 11 peaks with fold changes between 3.81927 and 162.06572</li>
      <li>On the 656 peaks on the full chromosome (number of lines of the original BED file) there are 252 peaks with FC&gt;50 (using <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">c7&gt;50</code>)</li>
    </ol>
  </blockquote>
</blockquote>

<p>The called peak regions can be filtered by, <em>e.g.</em> fold change, FDR and region length for further downstream analysis.</p>

<h1 id="step-6-plot-the-signal-between-samples">Step 6: Plot the signal between samples</h1>

<p>So far, we have normalized the data and identified peaks. Now, we would like to visualize scores associated with certain genomic regions, for example ChIP enrichment values around the TSS of genes. Moreover, we would like to compare the enrichment of several ChIP samples (e.g. CTCF and H3K4me3 )on the regions of interest.</p>

<p>Since we already generated the required files for the H3K4me3 sample, let’s make them only for the CTCF sample:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-prepare-the-peaks-and-data-for-ctcf"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Prepare the peaks and data for CTCF</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bam_compare/deeptools_bam_compare/3.5.1.0.0" title="bamCompare tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bamCompare</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First BAM file (e.g. treated sample)”</em>: <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1.bam</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second BAM file (e.g. control sample)”</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
        <li><em>“Bin size in bases”</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
        <li><em>“How to compare the two files”</em>: <code class="language-plaintext highlighter-rouge">Compute log2 of the number of reads ratio</code></li>
        <li><em>“Coverage file format”</em>: <code class="language-plaintext highlighter-rouge">bigwig</code></li>
        <li><em>“Region of the genome to limit the operation to”</em>: <code class="language-plaintext highlighter-rouge">chrX</code></li>
      </ul>
    </li>
    <li>Rename the output of <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the name of the sample</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0" title="MACS2 callpeak tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MACS2 callpeak</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.7.1+galaxy0)</span> with the following parameters
      <ul>
        <li><em>“Are you pooling Treatment Files?”</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“ChIP-Seq Treatment File”</em>: <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1.bam</code></li>
          </ul>
        </li>
        <li><em>“Do you have a Control File?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Are you pooling Treatment Files?”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“ChIP-Seq Treatment File”</em>: <code class="language-plaintext highlighter-rouge">wt_input_rep1.bam</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Format of Input Files”</em>: <code class="language-plaintext highlighter-rouge">Paired-end BAM</code></li>
        <li><em>“Effective genome size”</em>: <code class="language-plaintext highlighter-rouge">M.musculus (1.87e9)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>We can now concatenate the MACS2 outputs with the location of the peaks (concatenate the files and merge the overlapping regions) to obtain one BED file corresponding to the coordinates of the interesting regions to plot.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-prepare-the-peak-coordinates"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Prepare the peak coordinates</div>

  <ol>
    <li><span class="tool" data-tool="cat1" title="Concatenate datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Concatenate Dataset”</em>: output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">wt_CTCF_rep1</code></li>
        <li>Click on <em>“Insert Dataset”</em>:
          <ul>
            <li>For <em>“Dataset”</em>: select the output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for <code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_sortbed/2.30.0+galaxy2" title="bedtools SortBED tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bedtools SortBED</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.30.0+galaxy2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort the following BED/bedGraph/GFF/VCF/EncodePeak file”</em>: output of <strong>Concatenate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.30.0" title="bedtools MergeBED tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bedtools MergeBED</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.30.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort the following BAM/BED/bedGraph/GFF/VCF/EncodePeak file”</em>: output of <strong>SortBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>To plot the the peaks score on the region generated above (MergeBED output) two tools from the <a href="http://deeptools.readthedocs.io/">deepTools</a> package are used:</p>
<ul>
  <li><strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: it computes the signal on given regions, using the <code class="language-plaintext highlighter-rouge">bigwig</code> coverage files from different samples.</li>
  <li><strong>plotHeatmap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: it plots heatMap of the signals using the <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output.</li>
</ul>

<p>Optionally, we can also use <strong>plotProfile</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to create a profile plot from <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-plot-the-heatmap"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Plot the heatmap</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.5.1.0.0" title="computeMatrix tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>computeMatrix</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the following parameters:
      <ul>
        <li><em>“Select regions”</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Regions to plot”</em>: output of <strong>MergeBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>“Sample order matters”</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Score file”</em>: the 2 <code class="language-plaintext highlighter-rouge">bigwig</code> files generated by <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and renamed</li>
          </ul>
        </li>
        <li><em>“computeMatrix has two main output options”</em>: <code class="language-plaintext highlighter-rouge">reference-point</code>
          <ul>
            <li><em>“The reference point for the plotting”</em>: <code class="language-plaintext highlighter-rouge">center of region</code></li>
            <li><em>“Distance upstream of the start site of the regions defined in the region file”</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
            <li><em>“Distance downstream of the end site of the given regions”</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.5.1.0.1" title="plotHeatmap tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>plotHeatmap</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.1)</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Matrix file from the computeMatrix tool”</em>: <code class="language-plaintext highlighter-rouge">Matrix</code> (output of <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Show advanced options”</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>“Reference point label”</em>: select the right label</li>
            <li><em>“Did you compute the matrix with more than one groups of regions?”</em>: <code class="language-plaintext highlighter-rouge">No, I used only one group</code>
              <ul>
                <li><em>“Clustering algorithm”</em>: <code class="language-plaintext highlighter-rouge">Kmeans clustering</code></li>
                <li><em>“Number of clusters to compute”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>It should generate an heatmap similar to:</p>

<figure id="figure-16" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/peak_heatmap_1.png" alt="Output of plotHeatmap for 2 samples. " width="849" height="1500" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/peak_heatmap_1.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 16</strong>:</span> Scores around the peaks for wt_H3K4me3_rep1 and wt_input_rep1</figcaption></figure>

<p>When we look at this graph, it seems that less but larger peaks are found for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and that only few peaks are shared.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-18"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many peaks have been found for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> and for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code>?</li>
    <li>What are the mean width of the peaks for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> and for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code>?</li>
    <li>How many peaks are specific to <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> or <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code>?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-18"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-18" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>656 peaks for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and 2,688 for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> (number of lines in the <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> BED file)</li>
      <li>1630.77 bp for <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and 404.55 for <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> (<strong>Compute</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">c3-c2</code> and then <strong>Datamash</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with <code class="language-plaintext highlighter-rouge">Mean</code> on <code class="language-plaintext highlighter-rouge">Column:11</code>)</li>
      <li>443 peaks (over 656) are specific to <code class="language-plaintext highlighter-rouge">H3K4me3_rep1</code> and 2,464 (over 2,688) to <code class="language-plaintext highlighter-rouge">CTCF_rep1</code> (<strong>Intersect intervals</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>). Around 220 peaks are then overlapping.</li>
    </ol>
  </blockquote>
</blockquote>

<p>So far, we have only analyzed 2 samples, but we can do the same for all the 6 samples:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-optional-plot-the-heatmap-for-all-the-samples"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: (Optional) Plot the heatmap for all the samples</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_bam_compare/deeptools_bam_compare/3.5.1.0.0" title="bamCompare tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bamCompare</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> for each combination input - ChIP data:
      <ol>
        <li><code class="language-plaintext highlighter-rouge">wt_CTCF_rep1</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> (already done)</li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep1</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep1</code> (already done)</li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep1</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep1</code></li>
        <li><code class="language-plaintext highlighter-rouge">wt_CTCF_rep2</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K4me3_rep2</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></li>
        <li><code class="language-plaintext highlighter-rouge">wt_H3K27me3_rep2</code> - <code class="language-plaintext highlighter-rouge">wt_input_rep2</code></li>
      </ol>
    </li>
    <li>Rename the outputs of <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the name of the ChIP data</li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.2.7.1+galaxy0" title="MACS2 callpeak tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MACS2 callpeak</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.7.1+galaxy0)</span> for each combination input - ChIP data</li>
    <li><span class="tool" data-tool="cat1" title="Concatenate datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Concatenate Dataset”</em>: one output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>Click on <em>“Insert Dataset”</em>:
          <ul>
            <li>In <em>“Select”</em>: one other output of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li>Redo for the 6 outputs of <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_sortbed/2.30.0+galaxy2" title="bedtools SortBED tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bedtools SortBED</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.30.0+galaxy2)</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort the following bed,bedgraph,gff,vcf file”</em>: output of <strong>Concatenate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.30.0" title="bedtools MergeBED tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bedtools MergeBED</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.30.0)</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort the following bed,bedgraph,gff,vcf file”</em>: output of <strong>SortBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_compute_matrix/deeptools_compute_matrix/3.5.1.0.0" title="computeMatrix tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>computeMatrix</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.0)</span> with the same parameters but:
      <ul>
        <li><em>“Select regions”</em>:
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Regions to plot”</em>: output of <strong>MergeBED</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>“Sample order matters”</em>: <code class="language-plaintext highlighter-rouge">No</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Score file”</em>: the 6 <code class="language-plaintext highlighter-rouge">bigwig</code> files generated by <strong>bamCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and renamed</li>
          </ul>
        </li>
        <li><em>“computeMatrix has two main output options”</em>: <code class="language-plaintext highlighter-rouge">reference-point</code>
          <ul>
            <li><em>“The reference point for the plotting”</em>: <code class="language-plaintext highlighter-rouge">center of region</code></li>
            <li><em>“Distance upstream of the start site of the regions defined in the region file”</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
            <li><em>“Distance downstream of the end site of the given regions”</em>: <code class="language-plaintext highlighter-rouge">3000</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/deeptools_plot_heatmap/deeptools_plot_heatmap/3.5.1.0.1" title="plotHeatmap tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>plotHeatmap</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.5.1.0.1)</span> with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Matrix file from the computeMatrix tool”</em>: <code class="language-plaintext highlighter-rouge">Matrix</code> (output of <strong>computeMatrix</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>“Show advanced options”</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><em>“Reference point label”</em>: select the right label</li>
            <li><em>“Did you compute the matrix with more than one groups of regions?”</em>: <code class="language-plaintext highlighter-rouge">No, I used only one group</code>
              <ul>
                <li><em>“Clustering algorithm”</em>: <code class="language-plaintext highlighter-rouge">Kmeans clustering</code></li>
                <li><em>“Number of clusters to compute”</em>: <code class="language-plaintext highlighter-rouge">6</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-19"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many peaks are found for the different samples?</li>
    <li>
      <p>How are the peaks?</p>

      <figure id="figure-17" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/peak_heatmap_all.png" alt="Output of plotHeatmap for all samples. " width="1500" height="995" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/peak_heatmap_all.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 17</strong>:</span> Scores around the peaks for all samples</figcaption></figure>
    </li>
    <li>
      <p>How could be interpreted the peaks and read coverage in the <code class="language-plaintext highlighter-rouge">chrX:151,385,260-152,426,526</code> region?</p>

      <figure id="figure-18" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/rep1_igv.png" alt="Coverage and peaks for the replicate 1. " width="1270" height="445" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/rep1_igv.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 18</strong>:</span> Coverage and peaks for the replicate 1</figcaption></figure>
    </li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-19"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-19" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ol>
      <li>
        <p>Found peaks (number of lines in <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs):</p>

        <table>
          <thead>
            <tr>
              <th>Target</th>
              <th>Rep 1</th>
              <th>Rep 2</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>CTCF</td>
              <td>2,672</td>
              <td>2,061</td>
            </tr>
            <tr>
              <td>H3K4me3</td>
              <td>657</td>
              <td>718</td>
            </tr>
            <tr>
              <td>H3K27me3</td>
              <td>220</td>
              <td>73</td>
            </tr>
          </tbody>
        </table>

        <p>The tendencies are similar for both replicates: more peaks for CTCF, less for H3K4me3 and only few for H3K27me3.</p>
      </li>
      <li>As observed with the 2 samples, the peaks for H3K4me3 are wider than for CTCF. We also observe that the peaks found with one replicate are found with the other replicate.</li>
      <li>
        <p>The H3K4me3 sample has clear and large regions in which the read coverage are enriched. H3K4me3 is one of the least abundant histone modifications. It is highly enriched at active promoters near transcription start sites (TSS) and positively correlated with transcription.</p>

        <p>For H3K27me3, the coverage is more homogeneous. A gene is a broad domain of H3K27me3 enrichment across its body of genes corresponds to a gene with a transcription inhibited by H3K27me3. We can also identified some “bivalent” genes: gene with a peak around the TSS for H3K27me3(e.g. region_208 for gene Gpr173) but also H3K4me3. We also observe some H3K27me3-depleted regions sharply demarcated, with boundaries coinciding with gene borders (e.g. Kdm5c). This is a chromatin signature reminiscent of genes that escape XCI.</p>

        <p>To reproduce, run <strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>, <strong>IGV</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>MACS2 callpeak</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>Along this tutorial, we learn how to extract peaks and coverage information from raw data of ChIP experiments:</p>

<figure id="figure-19" style="max-width: 90%;"><img src="../../images/formation_of_super-structures_on_xi/tutorial-scheme.png" alt="Workflow to extract peaks and coverage information from raw data of ChIP experiments. " width="2672" height="2917" loading="lazy" /><a target="_blank" href="../../images/formation_of_super-structures_on_xi/tutorial-scheme.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 19</strong>:</span> Workflow to extract peaks and coverage information from raw data of ChIP experiments</figcaption></figure>

<p>This information can be then related to the biology to answer the original question. We tried then to relate the observed differences of peak and read coverage between H3K27me3 and H3K4me3 to the known biology. We could go even further in the analysis to reproduce the results of the original paper (e.g. by looking at the bivalent genes, identifying the differences between Xa and Xi).</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(plausible !== undefined){
                            plausible('TutorialComplete', {callback: tutorial_finish_finish, props: {path: document.location.pathname}})
                          } else {
                            tutorial_finish_finish();
                          }
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>ChIP-seq data requires multiple methods of quality assessment to ensure that the data is of high quality.</p>
</li>
                        
                        <li><p>Multiple normalization methods exists depending on the availability of input data.</p>
</li>
                        
                        <li><p>Heatmaps containing all genes of an organism can be easily plotted given a BED file and a coverage file.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/epigenetics/faqs/">FAQ page for the Epigenetics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Wang2018">Wang, C.-Y., T. Jégu, H.-P. Chu, H. J. Oh, and J. T. Lee, 2018 <b>SMCHD1 Merges Chromosome Compartments and Assists Formation of Super-Structures on the Inactive X</b>. Cell 174: 406–421.e25. <a href="https://doi.org/10.1016/j.cell.2018.05.007">10.1016/j.cell.2018.05.007</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=epigenetics/formation_of_super-structures_on_xi"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=epigenetics/formation_of_super-structures_on_xi">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Friederike Dündar, Anika Erxleben, Bérénice Batut, Vivek Bhardwaj, Fidel Ramirez, Leily Rabbani, Pavankumar Videm,  <b>Formation of the Super-Structures on the Inactive X (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html">https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{epigenetics-formation_of_super-structures_on_xi,
author = "Friederike Dündar and Anika Erxleben and Bérénice Batut and Vivek Bhardwaj and Fidel Ramirez and Leily Rabbani and Pavankumar Videm",
	title = "Formation of the Super-Structures on the Inactive X (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: deeptools_bam_compare
  owner: bgruening
  revisions: de72ba3ed6f2
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_bam_coverage
  owner: bgruening
  revisions: 12af64f93288
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_compute_matrix
  owner: bgruening
  revisions: 9470f29702f8
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_multi_bam_summary
  owner: bgruening
  revisions: 004aea905d4d
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_plot_correlation
  owner: bgruening
  revisions: fd408d755394
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_plot_fingerprint
  owner: bgruening
  revisions: 793665fff0f8
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deeptools_plot_heatmap
  owner: bgruening
  revisions: ba6efa77fb79
  tool_panel_section_label: deepTools
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trim_galore
  owner: bgruening
  revisions: cd7e644cae1d
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bowtie2
  owner: devteam
  revisions: 03e9b2fbc005
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: 3d0c7bdf12f5
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_idxstats
  owner: devteam
  revisions: 7def34e6cf4d
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bedtools
  owner: iuc
  revisions: a1a923cd89e8
  tool_panel_section_label: BED
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bedtools
  owner: iuc
  revisions: a1a923cd89e8
  tool_panel_section_label: BED
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: macs2
  owner: iuc
  revisions: 640d3af5d833
  tool_panel_section_label: Peak Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.75">3</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">4</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">1 stars</span><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.25">1</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>June 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: it was clear 
					<b>Disliked</b>: maybe you could explain the file formats and what the tools do in a more clean way 
				</li>
				
		    </ul>
		    
		    <b>December 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Well detailed method of ChIPseq and associated explanations
					<b>Disliked</b>: The title in itself doesn't really reflect the final results of the tutorial, with the data used one do not really identify TAD on the inactive X chromosome... 
				</li>
				
		    </ul>
		    
		    <b>January 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					
					<b>Disliked</b>: In step 6 it is explained to insert each of the datasets one after the other (with Concatenate datasets tail-to-head). However, one can insert more than one dataset at once with this tool, so why not do that? Also, it should read "Redo for the remaining four outputs of MACS2 callpeak" - it is six in total and in the first step you concatenate two and then add the remaining.  Why are the bedgraph files created if they are not used for anything?
				</li>
				
		    </ul>
		    
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00140">gxy.io/GTN:T00140</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/epigenetics/tutorials/formation_of_super-structures_on_xi/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/5012353c3c3462ce860e77e8fb34393bb8bbd8fb">5012353</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>