<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Genomic Data Visualisation with JBrowse</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for data visualisation in Galaxy" />
        <meta property="og:title" content="Galaxy Training: Genomic Data Visualisation with JBrowse" />
        <meta property="og:description" content="Training material for data visualisation in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/visualisation" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Visualisation
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvisualisation%2Ftutorials%2Fjbrowse%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvisualisation%2Ftutorials%2Fjbrowse%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvisualisation%2Ftutorials%2Fjbrowse%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvisualisation%2Ftutorials%2Fjbrowse%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvisualisation%2Ftutorials%2Fjbrowse%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvisualisation%2Ftutorials%2Fjbrowse%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/visualisation/tutorials/jbrowse/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Genomic Data Visualisation with JBrowse",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Visualisation",
    "description": "Training material for data visualisation in Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/visualisation/"
  },
  "courseCode": "visualisation / jbrowse / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Genomic Data Visualisation with JBrowse' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/visualisation/tutorials/jbrowse/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - How can I visualise features or blast data?\n - How can I visualise sequencing data in a workflow\n\n\\nThe objectives are:\n - Build several visualisations in JBrowse\n - Have basic familiarity with moving around JBrowse, and loading several data tracks\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Helena Rasche"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Helena Rasche"
    },
    {
      "@type": "Person",
      "name": "Saskia Hiltemann"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Visualisation",
      "description": "Training material for data visualisation in Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/visualisation/"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Genomic Data Visualisation with JBrowse</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/hexylena/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/hexylena" alt="Helena Rasche">Helena Rasche</a>, <a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/shiltemann" alt="Saskia Hiltemann">Saskia Hiltemann</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How can I visualise features or blast data?</p>
</li>
            
            <li><p>How can I visualise sequencing data in a workflow</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Build several visualisations in JBrowse</p>
</li>
            
            <li><p>Have basic familiarity with moving around JBrowse, and loading several data tracks</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/visualisation/tutorials/jbrowse/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.3591856">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                

                

                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<blockquote class="quote">
  <p>JBrowse (<a class="citation" href="#Buels_2016">Buels <i>et al.</i> 2016</a>) is a fast, embeddable genome browser built completely with JavaScript
and HTML5, with optional run-once data formatting tools written in Perl.</p>

  <p><em>from <a href="http://jbrowse.org/">http://jbrowse.org/</a></em></p>
</blockquote>

<p>The <span class="notranslate">Galaxy</span> tool accepts data in many formats:</p>

<ul>
  <li>Intervals/Annotation/Feature Tracks (GFF/GFF3, <span class="notranslate">BED</span>, GenBank)</li>
  <li>BAM Pileups</li>
  <li>Blast XML results</li>
  <li>Wig/BigWig</li>
  <li>VCF files</li>
</ul>

<p>and executes the ‚Äúrun-once data formatting tools‚Äù mentioned in its description. The JBrowse tool has an incredibly extensive number of options, more than anyone needs most of the time. We‚Äôll go through them in detail but feel free to skip the sections that don‚Äôt apply to the data types you use. Not everyone has Blast results to visualise. This tutorial also is very hands off, we will give some guidance but it is up to you to explore JBrowse and consider ways in which it can apply to your own data. Unlike other tutorials there is no biological story here, just some ways you can visualise common datatypes.</p>

<p>This tutorial covers version 1.16.5+ of the JBrowse tool, earlier versions will have different behaviour and tool layout.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#data-upload" id="markdown-toc-data-upload">Data Upload</a></li>
  <li><a href="#simple-gene-tracks" id="markdown-toc-simple-gene-tracks">Simple Gene Tracks</a></li>
  <li><a href="#complex-gene-tracks" id="markdown-toc-complex-gene-tracks">Complex Gene Tracks</a></li>
  <li><a href="#sequencing-coverage-and-variation" id="markdown-toc-sequencing-coverage-and-variation">Sequencing, Coverage, and Variation</a></li>
  <li><a href="#blast-results" id="markdown-toc-blast-results">Blast Results</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-upload">Data Upload</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-getting-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Getting the data</h3>

  <ol>
    <li>
      <p>Create and name a new history for this tutorial.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import the datasets we will visualize:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3591856/files/blastp%20genes.gff3
https://zenodo.org/record/3591856/files/blastp%20vs%20swissprot.xml
https://zenodo.org/record/3591856/files/<span class="notranslate">dna</span>%20sequencing.bam
https://zenodo.org/record/3591856/files/<span class="notranslate">dna</span>%20sequencing%20coverage.bw
https://zenodo.org/record/3591856/files/genes%20(de%20novo).gff3
https://zenodo.org/record/3591856/files/genes%20(NCBI).gff3
https://zenodo.org/record/3591856/files/genome.fa
https://zenodo.org/record/3591856/files/<span class="notranslate">RNA</span>-Seq%20coverage%201.bw
https://zenodo.org/record/3591856/files/<span class="notranslate">RNA</span>-Seq%20coverage%202.bw
https://zenodo.org/record/3591856/files/variants.vcf
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>
    </li>
  </ol>

</blockquote>

<p>The data for today is a subset of real datasets from <em>E. coli MG1655 strain K-12</em></p>

<h1 id="simple-gene-tracks">Simple Gene Tracks</h1>

<p>We will start by adding a couple of gene call tracks. In our case these are genes and gene predictions, but they don‚Äôt have to be. In the general case this can be any interesting region from an analysis, where a tool has pointed out some region for further inspection for some reason, then this data can be visualised with the ‚Äúgene‚Äù track type.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-build-the-jbrowse"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Build the JBrowse</h3>

  <ol>
    <li><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúReference genome to display‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome.fa</code></li>
          </ul>
        </li>
        <li><em>‚ÄúGenetic Code‚Äù</em>: <code class="language-plaintext highlighter-rouge">11. The Bacterial, Archaeal and Plant Plastid Code</code></li>
        <li>In <em>‚ÄúTrack Group‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li><em>‚ÄúTrack Category‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genes</code></li>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">GFF/GFF3/<span class="notranslate">BED</span> Features</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGFF/GFF3/<span class="notranslate">BED</span> Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">genes (de novo).gff3</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Execute</p>
    </li>
    <li>
      <p>View the contents of the file</p>

      <figure id="figure-1"><img src="../../images/jbrowse-gff-track.png" alt="Screenshot of JBrowse" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Screenshot of JBrowse</figcaption></figure>

      <blockquote class="tip">
        <h3 id="tip-error-reading-from-name-store"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Error reading from name store</h3>
        <p><img src="../../images/jbrowse/error.png" alt="error message" /></p>

        <p>If you see this error message, click OK, this is a known bug.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>If you are not familiar with the operation of JBrowse there are some important points:</p>

<ul>
  <li>‚ÄúTracks‚Äù are shown on the left. Clicking the checkboxes will make the tracks visible or invisible</li>
  <li>You can use your mouse scrollwheel to move around the genome view area, or you can click and drag to move.</li>
  <li>Double clicking will zoom in on the genome, or you can use the magnifying glass icons to zoom in our out</li>
</ul>

<blockquote class="tip">
  <h3 id="tip-tip-naming-tracks"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Naming Tracks</h3>

  <ul>
    <li>The JBrowse tool takes track names directly from file names</li>
    <li>If you want to rename tracks: <strong>Click</strong> on the <strong>pencil</strong> icon, edit the <strong>Name</strong> and click <strong>Save</strong>.</li>
    <li>You can now re-run the JBrowse tool and it will produce a new JBrowse instance with corrected names.</li>
  </ul>
</blockquote>

<h1 id="complex-gene-tracks">Complex Gene Tracks</h1>

<p>All of the track types in the JBrowse tool support a wide array of features. We‚Äôve only looked at a simple track with default options, however there are more tools available to us to help create user-friendly JBrowse instances that can embed rich data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-build-a-jbrowse-for-viewing-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Build a JBrowse for viewing Genes</h3>

  <ol>
    <li><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúReference genome to display‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome.fa</code></li>
          </ul>
        </li>
        <li><em>‚ÄúGenetic Code‚Äù</em>: <code class="language-plaintext highlighter-rouge">11. The Bacterial, Archaeal and Plant Plastid Code</code></li>
        <li>In <em>‚ÄúTrack Group‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li><em>‚ÄúTrack Category‚Äù</em>: <code class="language-plaintext highlighter-rouge">Genes</code></li>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">GFF/GFF3/<span class="notranslate">BED</span> Features</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGFF/GFF3/<span class="notranslate">BED</span> Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">genes (de novo).gff3</code></li>
                            <li><em>‚ÄúJBrowse Track Type [Advanced]‚Äù</em>: <code class="language-plaintext highlighter-rouge">Canvas Features</code></li>
                            <li>In <em>‚ÄúJBrowse Feature Score Scaling &amp; Coloring Options [Advanced]‚Äù</em>:
                              <ul>
                                <li><em>‚ÄúColor Score Algorithm‚Äù</em>: <code class="language-plaintext highlighter-rouge">Based on score</code>
                                  <ul>
                                    <li><em>‚ÄúHow should minimum and maximum values be determined for the scores of the features‚Äù</em>: <code class="language-plaintext highlighter-rouge">Manually specify minimum and maximum expected scores for the feature track</code>
                                      <ul>
                                        <li><em>‚ÄúMinimum expected score‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                                        <li><em>‚ÄúMaximum expected score‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                                      </ul>
                                    </li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">GFF/GFF3/<span class="notranslate">BED</span> Features</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGFF/GFF3/<span class="notranslate">BED</span> Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">genes (NCBI).gff3</code></li>
                            <li><em>‚ÄúJBrowse Track Type [Advanced]‚Äù</em>: <code class="language-plaintext highlighter-rouge">Canvas Features</code></li>
                            <li>
                              <p>In <em>‚ÄúJBrowse Contextual Menu options [Advanced]‚Äù</em>:</p>

                              <blockquote class="tip">
                                <h3 id="tip-tip-contextual-menus"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Contextual Menus</h3>

                                <p>We‚Äôre going to add a ‚ÄúContextual Menu‚Äù which deserves a little explanation first. In JBrowse
terminology the right click menu of some features is called the contextual menus. You can customize
this menu to add new links and options that will be useful to the user. These links can be
templated with variables based on metadata of the feature that the user clicked upon.</p>

                                <p>There are more valid values for templating than mentioned in the help. When
you click on a feature in the JBrowse instance, it will present all of the
properties of the feature. Any of the top level properties can be used
directly in your templating</p>
                              </blockquote>

                              <ul>
                                <li>In <em>‚ÄúTrack Menu‚Äù</em>:
                                  <ul>
                                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Menu‚Äù</em>
                                      <ul>
                                        <li><em>‚ÄúMenu label‚Äù</em>: <code class="language-plaintext highlighter-rouge">See protein at NCBI</code></li>
                                        <li><em>‚ÄúMenu url‚Äù</em>: <code class="language-plaintext highlighter-rouge">https://www.ncbi.nlm.nih.gov/gene?term={locus_tag}%5BGene%20Name%5D</code></li>
                                        <li><em>‚ÄúMenu icon‚Äù</em>: <code class="language-plaintext highlighter-rouge">Database</code></li>
                                      </ul>
                                    </li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>View the output</p>

      <figure id="figure-2"><img src="../../images/jbrowse/genes.png" alt="genes in JBrowse" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Genes visualised in JBrowse. The lighter coloured genes on the denovo track are coloured based on the confidence score given to the prediction by AUGUSTUS. NCBI genes do not have scores, these are more like an official gene set.</figcaption></figure>
    </li>
    <li>
      <p>Turn on both tracks of data.</p>
    </li>
    <li>
      <p>Navigate to <code class="language-plaintext highlighter-rouge">21,200</code>, either manually, or by copying and pasting the location block: <code class="language-plaintext highlighter-rouge">NC_000913.3:18351..24780</code></p>
    </li>
    <li>
      <p>Right click on the <code class="language-plaintext highlighter-rouge">yaaY</code> gene, and click the ‚ÄúSee protein at NCBI‚Äù menu option.</p>

      <p>This menu option is dynamic, try it with a few other features from the <code class="language-plaintext highlighter-rouge">genes (NCBI).gff3</code> track. These features have a <code class="language-plaintext highlighter-rouge">locus_tag</code> and the menu button we added will open a URL to an NCBI search page for the value of this <code class="language-plaintext highlighter-rouge">locus_tag</code> attribute.</p>

      <figure id="figure-3"><img src="../../images/jbrowse/ncbi.png" alt="ncbi view" /><figcaption><span class="figcaption-prefix">Figure 3:</span> The NCBI iframe within JBrowse, you can see how to link to exte<span class="notranslate">rna</span>l databases from JBrowse and help the viewer obtain as much information or context as they might need... but you have to build it yourself.</figcaption></figure>
    </li>
  </ol>
</blockquote>

<p>Contextual menus can be used to link to more than just NCBI.</p>

<ul>
  <li>The links can go anywhere such as web search services (e.g. Google) or genomics web services (e.g. EBI)</li>
  <li>Some sites use the IFrame action to link genes to local services where users are expected to submit annotation notes or data.</li>
</ul>

<h1 id="sequencing-coverage-and-variation">Sequencing, Coverage, and Variation</h1>

<p>This is the next major category of data that people wish to visualize: sequencing, coverage, and variation. The sequencing data can be of any type, and does not necessarily need to be sequencing data, as long as the results are formatted as BAM files. Here we will add several coverage tracks (essentially a line plot along the genome, associating a position with a value) with various visualisation options like scaling and display type. Each of these visualisation options can be useful in different situations, but it largely is a matter of preference or what you are used to seeing.</p>

<p>Next we will add a sequencing dataset, a BAM file which maps some sequencing <span class="notranslate">reads</span> against various locations along the genome. JBrowse helpfully highlights which <span class="notranslate">reads</span> have <span class="notranslate">mapping</span> issues, and any changes in bases between the <span class="notranslate">reads</span> and the genome. We ‚ÄúAutogenerate a SNP track‚Äù, which produces an extra track we can enable in JBrowse. This track <span class="notranslate">reads</span> the same BAM file used for visualising <span class="notranslate">reads</span>, and then produces a SNP and coverage visualisation. <strong>NB</strong>: This only works for small BAM files, if your files are large (&gt;200 Mb), then you should consider generating these coverage and SNP tracks by other means (e.g. <code class="language-plaintext highlighter-rouge">bamCoverage</code> and <code class="language-plaintext highlighter-rouge">FreeBayes</code> or similar tools) as it will be significantly faster. You can learn more about generating these files in <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">the <span class="notranslate">Mapping</span> tutorial</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-task-description"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Task description</h3>

  <ol>
    <li><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúReference genome to display‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome.fa</code></li>
          </ul>
        </li>
        <li><em>‚ÄúGenetic Code‚Äù</em>: <code class="language-plaintext highlighter-rouge">11. The Bacterial, Archaeal and Plant Plastid Code</code></li>
        <li>In <em>‚ÄúTrack Group‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li><em>‚ÄúTrack Category‚Äù</em>: <code class="language-plaintext highlighter-rouge">Coverage</code></li>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">BigWig XY</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBigWig Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">dna</span> sequencing coverage.bw</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">BigWig XY</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBigWig Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">rna</span>-seq coverage/1.bw</code></li>
                            <li><em>‚ÄúUse XYPlot‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">BigWig XY</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBigWig Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">rna</span>-seq coverage/2.bw</code></li>
                            <li><em>‚ÄúUse XYPlot‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                            <li><em>‚ÄúShow variance band‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                            <li><em>‚ÄúTrack Scaling‚Äù</em>: <code class="language-plaintext highlighter-rouge">Autoscale (global)</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li><em>‚ÄúTrack Category‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sequencing &amp; Variation</code></li>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM Pileups</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">sequencing.bam</code></li>
                            <li><em>‚ÄúAutogenerate SNP Track‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">VCF SNPs</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSNP Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">variants.vcf</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Execute and then explore the resulting data.</p>

      <figure id="figure-4"><img src="../../images/jbrowse/sequencing.png" alt="sequencing tracks in JBrowse" /><figcaption><span class="figcaption-prefix">Figure 4:</span> All of the sequencing tracks in JBrowse. Try exploring! There are many menu and configuration options which can help you filter and sort your data.</figcaption></figure>
    </li>
  </ol>
</blockquote>

<p>Try:</p>

<ul>
  <li>Clicking on individual variations in the dedicated <code class="language-plaintext highlighter-rouge">variants.vcf</code> track</li>
  <li>Hovering over the autogenerated SNPs/Coverage track</li>
  <li>Clicking on individual <span class="notranslate">reads</span> of the <code class="language-plaintext highlighter-rouge">sequencing.bam</code> track</li>
  <li>Changing the visualisation options of the BAM track</li>
</ul>

<h1 id="blast-results">Blast Results</h1>

<p>The Blast visualisation module requires that you have a gff3 formatted set of features which you then exported as <span class="notranslate">DNA</span> or protein, and blasted. The reason is easy to understand: when you extract <span class="notranslate">DNA</span>/protein sequences for Blasting, this process looses information about where these sequences were along the genome. The results from Blast retains the identifiers from the <span class="notranslate">DNA</span>/protein sequences, so we need to ‚Äúmap‚Äù these identifiers, to proper features with locations.</p>

<p>The best way to accomplish this is through the <code class="language-plaintext highlighter-rouge">gffread</code> tool which can cleanup a gff3 file, and export various features, optionally translating them. With these outputs, the cleaned features and <code class="language-plaintext highlighter-rouge">fasta</code> formatted sequences, you can Blast the sequences, and then supply the resulting Blast XML outputs in addition to the cleaned features, allowing a script to re-associate these Blast results to their original locations along the genome.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-building-a-jbrowse-for-blast-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Building a JBrowse for Blast results</h3>

  <ol>
    <li><strong>JBrowse</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúReference genome to display‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome.fa</code></li>
          </ul>
        </li>
        <li><em>‚ÄúGenetic Code‚Äù</em>: <code class="language-plaintext highlighter-rouge">11. The Bacterial, Archaeal and Plant Plastid Code</code></li>
        <li>In <em>‚ÄúTrack Group‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Track Group‚Äù</em>
              <ul>
                <li>In <em>‚ÄúAnnotation Track‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation Track‚Äù</em>
                      <ul>
                        <li><em>‚ÄúTrack Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Blast XML</code>
                          <ul>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBlastXML Track Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">blastp vs swissprot.xml</code></li>
                            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFeatures used in Blast Search‚Äù</em>: <code class="language-plaintext highlighter-rouge">blastp genes.gff3</code></li>
                            <li><em>‚ÄúMinimum Gap Size‚Äù</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                            <li><em>‚ÄúIs this a protein blast search?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                            <li>In <em>‚ÄúJBrowse Feature Score Scaling &amp; Coloring Options [Advanced]‚Äù</em>:
                              <ul>
                                <li><em>‚ÄúColor Score Algorithm‚Äù</em>: <code class="language-plaintext highlighter-rouge">Based on score</code>
                                  <ul>
                                    <li><em>‚ÄúJBrowse style.color function‚Äôs score scaling‚Äù</em>: <code class="language-plaintext highlighter-rouge">Blast scaling</code></li>
                                  </ul>
                                </li>
                              </ul>
                            </li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Execute and then explore the resulting data.</p>

      <figure id="figure-5"><img src="../../images/jbrowse/blast.png" alt="blast results in JBrowse" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Blast results, coloured according to their e-value. This sort of track is commonly used to help genome annotators have additional genomic context when they are annotating.</figcaption></figure>
    </li>
  </ol>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>This does not exhaustively cover JBrowse, and the tool is more extensible than can be easily documented, but hopefully these examples are illustrative and can give you some ideas about your next steps.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>This tutorial can not exhaustively cover every data type, but maybe it provides inspiration for your own analyses</p>
</li>
                            
                            <li><p>JBrowse is a great, workflow-compatible alternative to other genome browsers</p>
</li>
                            
                            <li><p>You can build visualisations that summarise dozens of analyses in one visualisation</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Buels_2016">Buels, R., E. Yao, C. M. Diesh, R. D. Hayes, M. Munoz-Torres <i>et al.</i>, 2016 <b>JBrowse: a dynamic web platform for genome visualization and analysis</b>. Genome Biology 17:</span>
    
    
    <a href="https://doi.org/10.1186/s13059-016-0924-1">10.1186/s13059-016-0924-1</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Genomic Data Visualisation with JBrowse (Visualisation)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Helena Rasche, Saskia Hiltemann, 2021 <b>Genomic Data Visualisation with JBrowse (Galaxy Training Materials)</b>. <a href="/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html">/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{visualisation-jbrowse,
    author = "Helena Rasche and Saskia Hiltemann",
    title = "Genomic Data Visualisation with JBrowse (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/visualisation/tutorials/jbrowse/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Helena Rasche, Saskia Hiltemann)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/visualisation/tutorials/jbrowse/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
