---
layout: introduction_slides
logo: "GTN"

title: "Introduction to ChIP-Seq data analysis"
type: "introduction"
contributors:
  - bebatut
---

# What is ChIP sequencing?

---

### Where my data comes from?

![Cartoon showing the histones forming part of the chromosome. Various insets zoom in from the DNase I hypersensitive sites to DNA with transcription factors and binding sites, which zooms into a block diagram of a gene.](../images/transcriptomics_images/ecker_2012.jpg)
<small>
[*Ecker et al, Nature, 2012*](https://www.ncbi.nlm.nih.gov/pubmed/22955614)
</small>

---

### Chromatin immuno-precipitation

![Cartoon of Chromatin immuno-precipitation. DNA interacts with proteins, this is cross-linked and sheared into DNA-protein complexes, immunoprecipitated, purified, and the bound DNA is sequenced. This is then mapped to a reference genome and the tag distribution is plotted along a chromosome.](../images/szalkowski_schmid_2011.jpg)

[*Szalkowski & Schmid, Brief Bioinform, 2011*](https://bib.oxfordjournals.org/content/12/6/626.long)

---

## How to analyze ChIP-seq data?

---

### Common procedures for ChIP-seq data analysis

![Complex graphic with the first step being Sequence alignment. Then Inpsect data quality with several substeps and graphics shown genome browser, motifs, and a chart of tag density vs tss. Lastly is peak calling, for broad peaks CCAT and SICER, for Sharp peaks MACS, and for both ZINBA.](../images/kidder_2011.jpg)

[*Kidder et al, Nature Immunology, 2011*](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3541830/)
