---
layout: tutorial_slides
logo: "GTN"

title: "Introduction to Cancer Analysis"
questions:
  - TODO
objectives:
  - Describe how cancer genomes differ from normal genomes and various sources of cancer genome variation
  - Understand different bioinformatic approaches to detecting cancer genome variation
  - Learn strengths and weaknesses of different approaches to cancer genome analysis (whole genome, exome, RNA-seq, targeted sequencing, etc.)
  - Inform how cancer genome analysis may be used to guide cancer patient management
key_points:
   - TODO

contributions:
  authorship:
   - trevor-pugh
  editing:
   - shiltemann
  funding:
   - bioinformatics-ca
---

# 1


![](images/CAN_Module1_Lecture-1_1.jpg)
![](images/CAN_Module1_Lecture-1_2.jpg)

Canadian Bioinformatics Workshops
www.bioinformatics.ca
bioinformaticsdotca.github.io

---
# 2

![](images/CAN_Module1_Lecture-2_1.jpg)
![](images/CAN_Module1_Lecture-2_2.jpg)

2 Module #: Title of Module

---
# 3

![](images/CAN_Module1_Lecture-3_1.jpg)
![](images/CAN_Module1_Lecture-3_2.jpg)
![](images/CAN_Module1_Lecture-3_3.jpg)
![](images/CAN_Module1_Lecture-3_4.jpg)
![](images/CAN_Module1_Lecture-3_5.jpg)

<b>Module 1</b>
<b>Introduction to Cancer Genomics</b>
Trevor Pugh, PhD, FACMG
Bioinformatics for Cancer Genomics
June 3-7, 2019

---
# 4


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Learning Objectives of Module</b>


---
# 5

![](images/CAN_Module1_Lecture-5_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>We are all made of cells</b>
https://genographic.nationalgeographic.com/science-behind/genetics-overview/

---
# 6

![](images/CAN_Module1_Lecture-6_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>My human genome</b>

---
# 7

![](images/CAN_Module1_Lecture-7_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>My human genome</b>

---
# 8

![](images/CAN_Module1_Lecture-8_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Cancer is a disease of the genome</b>
www.cityofhope.org/research/support/cytogenetics/

---
# 9

![](images/CAN_Module1_Lecture-9_1.jpg)
![](images/CAN_Module1_Lecture-9_2.png)
![](images/CAN_Module1_Lecture-9_3.jpg)
![](images/CAN_Module1_Lecture-9_4.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
A   Point mutations
B   Copy number alterations
C   Structural rearrangements
D   Genes & regulatory elements
E   Pathways
F   Homologs, epigenetic modifications, et al.
www.circos.ca
<i>Anna Karenina</i>
Leo Tolstoy
<b>&#8220;All happy families are alike; each unhappy </b>
<b>family is unhappy in its own way&#8221;</b>

---
# 10

![](images/CAN_Module1_Lecture-10_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Cancer cells accumulate</b>
<b>somatic alterations over time</b>
Stratton MR, Campbell PJ, Futreal PA.  <i>Nature</i> . 2009 Apr 9;458(7239):719-24. Review.
Mutation frequency depends on cancer type
External forces (e.g. drug treatment) can select for specific
clones (e.g. cells with resistance mutations)

---
# 11

![](images/CAN_Module1_Lecture-11_1.jpg)
![](images/CAN_Module1_Lecture-11_2.jpg)
![](images/CAN_Module1_Lecture-11_3.png)
![](images/CAN_Module1_Lecture-11_4.jpg)
![](images/CAN_Module1_Lecture-11_5.jpg)
![](images/CAN_Module1_Lecture-11_6.jpg)
![](images/CAN_Module1_Lecture-11_7.jpg)
![](images/CAN_Module1_Lecture-11_8.jpg)
![](images/CAN_Module1_Lecture-11_9.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<i>Lawrence et al. Nature 2013 Jul 11;499(7457):214-8. </i>
<b>Mutation burden varies by cancer type, </b>
<b>exposure, age of onset, &amp; DNA repair ability</b>

---
# 12

![](images/CAN_Module1_Lecture-12_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Tumour cells acquire abnormal abilities </b>
<b>by co-opting normal cell behaviour</b>
<i>Hanahan and Weinberg</i>
<i>Cell. 2011 Mar 4;144(5):646-74.</i>

---
# 13

![](images/CAN_Module1_Lecture-13_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Oncogenic somatic alterations target a </b>
<b>core set of biological functions</b>
Differentiating “driver” from “passenger” mutations
is a central challenge of cancer genome analysis

---
# 14

![](images/CAN_Module1_Lecture-14_1.jpg)
![](images/CAN_Module1_Lecture-14_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Targeted therapies exploit specific mutations</b>
Lung adenocarcinoma with
activating mutations of  <i>EGFR</i>
ê
Inhibiting EGFR shrinks tumors
Lynch et al. N Engl J Med. 2004 May 20;350(21):2129-39
Metastatic melanoma with
activating mutations of  <i>BRAF</i>
ê
Inhibiting BRAF shrinks tumors
Bollag et al. Nat Rev Drug Discov. 2012 Nov;11(11):873-86
<i>Resistance inevitably arises</i>

---
# 15

![](images/CAN_Module1_Lecture-15_1.jpg)
![](images/CAN_Module1_Lecture-15_2.jpg)
![](images/CAN_Module1_Lecture-15_3.png)
![](images/CAN_Module1_Lecture-15_4.png)
![](images/CAN_Module1_Lecture-15_5.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>&#8220;Actionable&#8221; mutations in 3,299 tumours</b>
Ciriello et al.
Nature Genetics 2013
37 genes from
4 pathways
linked to
treatments
12 genes
disrupted by
multiple
mechanisms
1) Druggable alterations
cut across cancer types
2) Combination therapies
may be effective in
tumors with compound
pathway alterations
3) 50% of tumors
have at least two
disrupted, druggable
pathways

---
# 16

![](images/CAN_Module1_Lecture-16_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Even within the same tumour mass,</b>
<b>different cells may have different mutations,</b>
<i>Burrell et al. Nature. 2013 Sep 19;501(7467):338-45. </i>

---
# 17

![](images/CAN_Module1_Lecture-17_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Cancers are a mix of subclones that </b>
<b>can respond differently to therapy</b>
Ding et al. Nature. 2012 Jan 11;481(7382):506-10.

---
# 18

![](images/CAN_Module1_Lecture-18_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Subclones and metastases are genetically </b>
<b>related, diverge to form subpopulations</b>
Campbell et al. Nature. 2010 Oct 28;467(7319):1109-13.

---
# 19


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Reasons for molecular testing for cancer</b>
Treatment
treatment susceptibility, predict adverse side-effects, watch-and-
wait versus aggressive treatment
Drug resistance and metabolism
pre-existing resistance mutations, drug dosage
Inherited cancer syndromes
additional primary tumors, at-risk family members
Prognosis
molecular subtyping, benign vs. malignant,
determine unknown primary, predict metastatic potential

---
# 20


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
What are the targets in  <i>my </i> cancer…
…and what can be done about them?
Sequence mutations, polymorphisms
Structure copy number variation,
translocation,
loss-of-heterozygosity
Function expression, exon-usage
External Viruses, bacteria

---
# 21


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Applications of </b>
<b>next-generation DNA sequencing</b>
<b>technology to cancer</b>

---
# 22

![](images/CAN_Module1_Lecture-22_1.jpg)
![](images/CAN_Module1_Lecture-22_2.jpg)
![](images/CAN_Module1_Lecture-22_3.jpg)
![](images/CAN_Module1_Lecture-22_4.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>General DNA sequencing workflow</b>
<i>Genomic</i>
<i>DNA or RNA</i>
<i>Library of DNA </i>
<i>fragments</i>
<i>Sequencing</i>
<i>device</i>
<i>Computational</i>
<i>analysis</i>

---
# 23

![](images/CAN_Module1_Lecture-23_1.png)
![](images/CAN_Module1_Lecture-23_2.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
AAATGTCAAGGAGTCCTTTCTGACTAGTCTGTGCCTATATGGTTGGATATTTTTATGATTACCCCAACCACCATC
AGCAGCATTGTCAGAAACTACTTTGTGATGTTTGCATTCAAGTCACAGAATTGAACACTCCCTTTCACAGAGCAG
TGCTGATCCTATTGGTAAGTGCTCATACCTGCAGCATCTTCACCTGTGTCTTCTATATTCACCTGTGTGACTAAA
GCCCTTCTCCCTGATTCACCCCTCAGCATCGTCTCCCTCACCATCTTCTACCTTTTCAGCCTTTCCTTCCCCTCT
AGAAAACTCTACCTGTCATTTTTTCAACATTATCGATCCACTGGATCTTCATGGACTCAAAATGTCCATAAGCCG
GCTTATAATCTAGCCATGCTGGCAGCTGATTAGATTGTGCCCACTCAGATTAAGAGTCTGCCTTTCCCAGCCCAC
CTGGGGGATGGGATTTAGGAAAACTTCCCTAAGGGGTTTTGAAATGCAGTCCCTCAACCTCTCTTAAGCATCTCC
TTTGTTCTGAGGATAGGTTACATAGATGATTCTAATCCATTTATTATTTTTACATTCCAGGCTACCTAAAAGAAG
GCACTTTGGGAGGCCAAGGTTGGTGGATCATGAAGTCAGGAGTTGGAGACCATCCTGACCAGCATGGTGAAACCC
GTGTGCTCACAAATCTTTGTACAGAAAAGGCCCGGAGATTCCTACCTTGTTAAGCAAAGGGGCTGTATCACTACA
ATGACCTCATCTTAATTAATTGCATCTGCAATGGCCCTGTTTCTAAATAGGTCACACTCTGAGGTACTGGAGATT
GATTCCATTTGATGTTGATTCCATTCGAGTCCATTCGATCATTATTACATTTGATTCTATTCGATGATTCCATTC
GCAATACGGACTCAGTAGGAAAGCAACTGAAGGGTGTCATAAATTTATGCTGCAGATTTGCGAATGAGCTACTGT
TTGGTCTCCTCAAATTACTTTTATTTCTCACCTTCAGGATTTTGCATATGTTTTTTACTTTTATATAATACTTAA
TAACTTTTGATACAATGAAGTAAGTTCTATTTTTTTTTTTTTACCACTCTTAAGATATAATAGACCTGGAAATAC
AGTGCCAGGAGCTACCAGGGCTGTGCCAAGGAGGAAAATGTATCAACACCTTTGGGAGTTTCCAGCGCCGCTGTC
AGGATAAAGCAAGATGTAAGCTTCAAATTAGTCCAACTTGCTACTTTTTAGTCACTTTATTAAAGCATTTTGGGA
ATGCTATGTTTATTGCGGCACTATTCACAATAGCACAGACTTGGAACCAACCCAAATGATAGACTGGACTAAGAA
AGCCCTGTCCCTCCTCCAGGTTCCTCAAGCATGTCTCCTGCCTCTGCACCTACCTCAGAGAGCCAACATTCGACT
AAATTCAAAAAACAAACTCTGGTCCAGTTCAAGTATAATCTAATTATCTAACAGGTCACAGCCTCGGACTGTCCA
TTGTTTGAGCTCCTTATATATTCTGGTAATTAATCCCTTGTCAAACGAGTAGTTTGCAAGTATTTCTTCCGCTTC
TGGGGACTAAATGTAATGCTGGAAAAATTTCTTTTGCAGAGGCATATCCACCAACCTTTCTCTCCAGACCTAAGC
AGAATTCCCTTTGGTTTTCCCTTTCAGGAAGTAGATTATGAAGAAATGAAGAATCTTGACTTCAGTGTTATTGTC
ATGAACATTTCCCTATCTATTTCAAAATTACATGAATCACAATAAAGAATAACACAATCCTGAACAATGTGTGAA
ATGAAAATAATAATATCTTTTAAAGTCCCAGAGGTGCCAAAGGTAGCTGTCCCAGAACAGAAGGTGCCTGAAGCT
75 bp
25
reads
On an Illumina HiSeq 2500, one lane of an eight-lane flow cell generates >600 million reads

---
# 24

![](images/CAN_Module1_Lecture-24_1.jpg)
![](images/CAN_Module1_Lecture-24_2.png)
![](images/CAN_Module1_Lecture-24_3.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>&#8220;Pipeline&#8221;</b>
Variants
Raw DNA sequence
<i>www.hickerphoto.com</i>

---
# 25

![](images/CAN_Module1_Lecture-25_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>&#8220;Pipeline&#8221;</b>
1) Alignment
2) Pre-processing
3) QC metrics
4) Variant calling
5) Interpretation
6) Clinical report
<i>www.vnf.com</i>

---
# 26

![](images/CAN_Module1_Lecture-26_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Adapting Genome Analysis Toolkit Best </b>
<b>Practices for analysis of cancer genomics data </b>
http://www.broadinstitute.org/gatk/guide/best-practices
http://picard.sourceforge.net/

---
# 27

![](images/CAN_Module1_Lecture-27_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
muTect, IndelGenotyper,VarScan2,
UnifiedGenotyper, CluMP, VisCapCancer
SNPs &
Indels
Copy number variants
Translocations
Tumour Purity & Ploidy
Germline vs Somatic
http://www.broadinstitute.org/gatk/guide/best-practices
http://picard.sourceforge.net/
<b>Adapting Genome Analysis Toolkit Best </b>
<b>Practices for analysis of cancer genomics data </b>

---
# 28

![](images/CAN_Module1_Lecture-28_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
An alignment of reads to a human genome reference
Bwa, Novoalign, Stampy, SOAP, and >50 more listed on Wikipedia under “Short-Read Sequence Alignment”

---
# 29

![](images/CAN_Module1_Lecture-29_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<i>Meyerson M, Gabriel S, Getz G. Nat Rev Genet. 2010 Oct;11(10):685-96. Review.</i>
BWA
Picard
GATK
muTect IndelGenotyper xHMM, CoNIFER, CODEX, EXCAVATOR
SegSeq, CapSeg, VisCap
DELLY, CREST
deRanger
PathSeq
VirusFinder
<b>Multiple types of cancer genome variation</b>
<b>may be inferred from sequencing read alignments</b>
VarScan2

---
# 30


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>DNA sequencing approaches to cancer</b>
Whole genome sequencing (WGS)
Whole exome sequencing (WES)
Targeted gene sequencing
Targeted variant genotyping
Epigenome modification (bisulphite)
Transcriptome sequencing (RNAseq)
miRNA sequencing
Protein/DNA interaction mapping
Epigenome mapping (histones)
DNA
RNA
Protein
(DNA footprint)

---
# 31

![](images/CAN_Module1_Lecture-31_1.png)
![](images/CAN_Module1_Lecture-31_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
RNAseq
0-106X
Exome
~150X
(0-307X)
Genome
~40X
(0-92X)
<i>KRAS, 49 kb</i>
3.4 kb
<b>Coverage of genome, exome, &amp; RNAseq data</b>
www.broadinstitute.org/igv

---
# 32

![](images/CAN_Module1_Lecture-32_1.jpg)
![](images/CAN_Module1_Lecture-32_2.png)
![](images/CAN_Module1_Lecture-32_3.png)
![](images/CAN_Module1_Lecture-32_4.jpg)
![](images/CAN_Module1_Lecture-32_5.jpg)
![](images/CAN_Module1_Lecture-32_6.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Ex
ons
GC content
0.2 0.8
1
0.9
0.8
0.7
0.6
0.5
0.4
0.3
0.2
0.1
0
Fraction of each exon “callable”
<b>Exomes &amp; Genomes don&#8217;t measure everything</b>
60X genomes     30X genomes         Native exome     WGA exome
0.5
Pugh et al.
Nature 2013

---
# 33

![](images/CAN_Module1_Lecture-33_1.jpg)
![](images/CAN_Module1_Lecture-33_2.jpg)
![](images/CAN_Module1_Lecture-33_3.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Transcriptome coverage depends on expression</b>
Ex
pr
es
si
on
lev
el
<b>IMP$01022(</b>
•
 
Melanoma(pa*ent,(complete(response(to(Tamoxifen((Dr.(Anthony(Joshua)(
•  
RNA?Seq(on(primary(tumour,(exome(from(blood(sample((BC)(
<b>IMP$01022((</b> <b>ER(Posi.ve(BRCA(</b> <b>ER(Nega.ve(BRCA(</b>
<b>Preliminary(Data</b> (
<b>Patient</b>
Genes ordered by rank expression

---
# 34

![](images/CAN_Module1_Lecture-34_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Epigenetic modifications that regulate gene expression</b>
<b>detectable by bisulphite sequencing, even in admixed tissues</b>
Normal tissue
Endometrial
carcinoma
Unmethylated
<i>MLH1</i> promoter
(C>T)
Methylated
MLH1 promoter
(C>C)
<i>MLH1</i> <i>EPCAM</i>

---
# 35


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Data types are confirmatory and complementary</b>
<b>Genome Exome</b> <b>RNAseq</b> <b>MethylSeq</b>
Sequence mutations
Copy number
Rearrangements
Pathogens
Purity/Ploidy
Low purity or
subclonal mutations
Gene expression
Allele- or exon-
specific expression

---
# 36


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Somatic Mutations</b>

---
# 37

![](images/CAN_Module1_Lecture-37_1.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
37
Heterozygous
Total coverage = 19
C = 10
T = 9
Allele balance = 0.47
<b>Germline variants are detectable at low coverage</b>

---
# 38

![](images/CAN_Module1_Lecture-38_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Pugh  <i>et al.</i> BMC Cancer. 2007 Jul 13;7:128.
<b>Variable DNA quality &amp; quantity from </b>
<b>tissues used for routine diagnosis</b>

---
# 39

![](images/CAN_Module1_Lecture-39_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Pugh  <i>et al.</i> BMC Cancer. 2007 Jul 13;7:128.
<b>Tumors are a mix of cancer &amp; normal cells </b>
<b>(&#8220;purity&#8221; or &#8220;tumour content&#8221;)</b>

---
# 40

![](images/CAN_Module1_Lecture-40_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
www.cityofhope.org/research/support/cytogenetics/
<b>Tumors can have multiple genome copies </b>
<b>(&#8220;ploidy&#8221;)</b>

---
# 41

![](images/CAN_Module1_Lecture-41_1.jpg)
![](images/CAN_Module1_Lecture-41_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Deep coverage is necessary to detect</b>
<b>mutations in low purity or high ploidy tumors </b>
41
Genome 0/15 reads Exome 6/139 reads

---
# 42

![](images/CAN_Module1_Lecture-42_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Pyclone
Sequenza
Roth et al. Nat Methods. 2014 Apr;11(4):396-8. Favero et al. Ann Oncol. 2015 Jan;26(1):64-70.
<b>Deep (e.g. 250X) DNA sequencing coverage enables</b>
<b>inference of tumour purity, ploidy, &amp; subclonal structure</b>

---
# 43

![](images/CAN_Module1_Lecture-43_1.png)
![](images/CAN_Module1_Lecture-43_2.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Multiple DNA sequencing technologies enable</b>
<b>mutation confirmation &amp; cross-validation</b>
PCR/
PacBio
Hybrid
capture/
Illumina
Pugh et al. Nature 2012.

---
# 44

![](images/CAN_Module1_Lecture-44_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Mutation detection sensitivity is dictated by sequencing </b>
<b>depth and is limited by sequencer &amp; polymerase error</b>
<b>0.05</b>
<b>25,544X </b>
Minimum fraction of tumour DNA
in detectable with 99% probability
Exome
Clinical lab
Subclonality
Circulating
tumour DNA
Kis et al.  <i>Nat Commun. </i> 2017 May 11;8:15086.

---
# 45

![](images/CAN_Module1_Lecture-45_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>The ultimate complex mixture:</b>
<b>Cell-free DNA dissolved in blood is derived </b>
<b>from many normal cells and a few tumour cells</b>
Crowley et al.,  <i>Nature Reviews Clinical Oncology</i> , 2013

---
# 46

![](images/CAN_Module1_Lecture-46_1.png)
![](images/CAN_Module1_Lecture-46_2.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Hybrid capture ~50,000X PCR >100,000X coverage
<b>Coverage dependent on laboratory method </b>
<b>used to target regions of interest</b>
Tam-Seq: Forshew et al.
Sci Transl Med. 2012 May 30;4(136):136ra68.
CAPP-seq: Newman, Bratman et al.
Nature Medicine. 2014 May;20(5):548-54.

---
# 47

![](images/CAN_Module1_Lecture-47_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Quantity of circulating tumour DNA</b>
<b>reflects cancer burden</b>
Newman, Bratman et al. Nature Medicine. 2014 May;20(5):548-54.

---
# 48

![](images/CAN_Module1_Lecture-48_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Quantity of circulating tumour DNA</b>
<b>reflects clinical course</b>
Newman, Bratman et al. Nature Medicine. 2014 May;20(5):548-54.

---
# 49


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Somatic copy number alterations</b>
<b>and rearrangements</b>
<b>(structural alterations)</b>

---
# 50

![](images/CAN_Module1_Lecture-50_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Tumors can have complex structural </b>
<b>alterations</b>
http://www.bclq.org/en/index.html

---
# 51

![](images/CAN_Module1_Lecture-51_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Gains and losses evident from sequence data
Ch
ro
m
oso
m
e
Re
la
tiv
e
co
py
 n
um
be
r
Normal exome Tumor exome Tumor
microarray
Normal
array

---
# 52

![](images/CAN_Module1_Lecture-52_1.jpg)
![](images/CAN_Module1_Lecture-52_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Re-analysis of existing clinical sequence data</b>
<b>provides additional diagnostic value</b>
<i>Homozygous PTEN deletion in a</i>
<i>breast invasive ductal carcinoma</i>
<i>EGFR amplification (6 copies)</i>
<i>in a lung adenocarcinoma</i>

---
# 53

![](images/CAN_Module1_Lecture-53_1.jpg)
![](images/CAN_Module1_Lecture-53_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Additional probes provide richer copy number profiles</b>
<b>(183 gene panel, all exons plus select introns)</b>
<i>FGFR1</i>
5 copies, no mutations
Isochromosome 3

---
# 54

![](images/CAN_Module1_Lecture-54_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Exact breakpoints can be detected by </b>
<b>examining reads that span the region</b>
Imielinski et al. Cell. 2012 Sep 14;150(6):1107-20.

---
# 55

![](images/CAN_Module1_Lecture-55_1.jpg)
![](images/CAN_Module1_Lecture-55_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Translocation detection from soft-clipped bases</b>
<b>plus large-insert or intrachromosomal read-pairs</b>
PML-RARA
EWS-FLI

---
# 56

![](images/CAN_Module1_Lecture-56_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Rearrangements can be highly complex</b>
<b>and detectable at base-pair resolution</b>
Berger  <i>et al.</i> Nature. 2011 Feb 10;470(7333):214-20.

---
# 57

![](images/CAN_Module1_Lecture-57_1.jpg)
![](images/CAN_Module1_Lecture-57_2.jpg)
![](images/CAN_Module1_Lecture-57_3.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Some tumor genomes are highly rearranged, </b>
<b>some are quiet, some have no rearrangements</b>
Imielinski et al. 2012
Lung adenocarcinoma
Pugh et al. 2012
Neuroblastoma
Molenaar et al. 2012.

---
# 58


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Transcriptome sequencing</b>

---
# 59

![](images/CAN_Module1_Lecture-59_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>RNAseq can detect differences in exon usage </b>
<b>that underlie resistance to treatment</b>
Griffith et al. Nat Methods. 2010 Oct;7(10):843-7.

---
# 60

![](images/CAN_Module1_Lecture-60_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>RNAseq enables expression profiling &amp; subtyping</b>

---
# 61

![](images/CAN_Module1_Lecture-61_1.jpg)
![](images/CAN_Module1_Lecture-61_2.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Primary tissue of origin evident from expression </b>
<b>(if normal samples are available for comparison)</b>
Brain
Lung
Thyroid
Skin
Heart
Muscle
Blood
Four lung tumors
of unknown tissue
of origin

---
# 62

![](images/CAN_Module1_Lecture-62_1.jpg)
![](images/CAN_Module1_Lecture-62_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
Unusual mapping of RNAseq reads to Epstein-Barr
virus in lung adenocarcinoma
0.8 mm
ISH shows EBV confined
to tumor cells, tumor is actually a
lymphoepithelioma-like carcinoma
<b>Pathogens seen in Genomes, RNAseq, </b>
<b>sometimes Exomes</b>

---
# 63

![](images/CAN_Module1_Lecture-63_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Transcriptomes from single cells can now </b>
<b>be amplified and sequenced </b> <i><b>en masse</b></i>
10X Genomics Chromium technology
www.10xgenomics.com/solutions/single-cell/
Princess Margaret Genomics Centre (a service provider)
www.pmgenomics.ca/pmgenomics/services/single_cell_genomics.html

---
# 64

![](images/CAN_Module1_Lecture-64_1.jpg)
![](images/CAN_Module1_Lecture-64_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Combining cancer &amp; healthy cell data </b>
<b>highlights shared &amp; private populations</b>
Labeled by Cluster  Labeled by Donor
<b>PRTN3: </b> serine protease enzyme, expressed mainly by neutrophil granulocytes
<b>MPO: </b> peroxidase enzyme, abundantly expressed in neutrophil granulocytes
<b>LYZ</b> : present in cytoplasmic granules of macrophages
<b>FCN1: </b> pattern recognition molecule secreted by monocytes/macrophages
<b>LTB</b> : induces inflammatory response
<b>CD3D: </b> part of  TCR/CD3 complex
<b>TMEM66 (SARAF): </b> cytoplasmic calcium signalling
<b>JUNB: </b> transcription factor, growth factor response
<b>CD3E: </b> part of  TCR/CD3 complex
Patient A2
Healthy 1
Healthy 2

---
# 65

![](images/CAN_Module1_Lecture-65_1.jpg)

<b>HumanCellAtlas.org: Scientific and Funding Opportunities</b>
Chan Zuckerberg Initiative funding pilot projects in areas of technology
development, tissue sourcing, and computational methods
chanzuckerberg.com/human-cell-atlas

---
# 66


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Germline cancer genetics</b>
<b>(Hereditary Cancer Syndromes)</b>

---
# 67

![](images/CAN_Module1_Lecture-67_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>&#8220;Second hit&#8221; can unmask a </b>
<b>pathogenic germline allele</b>

---
# 68

![](images/CAN_Module1_Lecture-68_1.jpg)
![](images/CAN_Module1_Lecture-68_2.jpg)
![](images/CAN_Module1_Lecture-68_3.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Carrier for a pathogenic </b> <i><b>SUFU</b></i> <b>mutation </b>
<b>developed medulloblastoma (10-)</b>
Tumor, hemizygous Normal, heterozygous

---
# 69

![](images/CAN_Module1_Lecture-69_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
BWA
Picard
GATK
muTect Indel Genotyper SegSeq, CapSeg, VisCap deRanger,
BreakDancer,
CluMP
PathSeq
<b>Once found, variants (of all types) require</b>
<b>annotation to guide interpretation</b>
<i>Annotation, Interpretation, &amp; Reporting</i>

---
# 70


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>How do we interpret variants en masse? </b>
Clinical indication & history
(family members, de novo)
Locus-specific and internal variant databases
Population frequency
Public databases
(disease-specific & general)
Publications
Amino acid & biochemistry conservation
Substitution prediction tools
(Polyphen2, SIFT, AlignGVGD, SarcomerePolyphen)
Splicing prediction tools
(SpliceSiteFinder-like, MaxEntScan, NNSPLICE, GeneSplicer, Human Splice Finder)
Druggable targets and pathways
Drug Gene Interaction Database (DGIdb), Drug Bank

---
# 71

![](images/CAN_Module1_Lecture-71_1.jpg)
![](images/CAN_Module1_Lecture-71_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Command-line versus Manual Annotation</b>
Oncotator – plain text table with >150 columns
Alamut – Interactive desktop software
<i>Varient Effect Predictor, ANNOVAR, SNPeff, VAT, &amp; many others</i>

---
# 72

![](images/CAN_Module1_Lecture-72_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Novel Variant Assessment is still a manual</b>
<b>enterprise, aided greatly by sharing interpretations</b>
<i>Richards et al. Genetics in Medicine. (2015) 17, 405&#8211;423</i>

---
# 73

![](images/CAN_Module1_Lecture-73_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>How do we report and share these results? </b>
<b>Overall result: Somatic variants identified:</b>
BRAF (NM_004333.4) Heterozygous, c.1406G>T (p.G469V)
RB1 (NM_000321.2) Heterozygous, c.607+1G>A (splice)
TP53 (NM_000546.4) Heterozygous, c.488A>G (p.Y163C)
<b>Class 4A mutation: </b> BRAF p.Gly469Val (58% of reads)
BRAF p.G469V is a rare mutation across cancer (76 of
>193,000 cases in the Catalogue of Somatic Mutations
in Cancer), and has not been previously reported in
over 4,300 ovarian cancers. BRAF is recurrently
mutated in ovarian cancers, but is rarely mutated in
high grade serous carcinoma (mycancergenome.org). The
most common variant found in ovarian tumours is
p.V600E (mycancergenome.org). Currently, the impact
of this mutation on patient prognosis or treatment in
this tumour site is unknown.

---
# 74

![](images/CAN_Module1_Lecture-74_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
www.aacr.org/genie

---
# 75

![](images/CAN_Module1_Lecture-75_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Molecular report of the future?</b>
<b>Concise, systematic (crowd-sourced?) genome</b>
<b>interpretation alongside clinical annotations</b>
www.cbioportal.org

---
# 76

![](images/CAN_Module1_Lecture-76_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Case study</b>
1 Genome Sciences Centre, British Columbia Cancer Agency
2 British Columbia Cancer Agency
3 Vancouver General Hospital
4 Centre for Translational and Applied Genomics of British
Columbia Cancer Agency and the Provincial Health Services
Authority Laboratories
5 Molecular Oncology, BC Cancer Research Centre

---
# 77


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Initial presentation</b>
• 78 year-old Caucasian man
• Fit and active
• Presented in August 2007 with throat discomfort
• Examination found a 2 cm mass at left base of tongue
• Minimal comorbidities
• No obvious risk factors for oropharyngeal malignancy

---
# 78

![](images/CAN_Module1_Lecture-78_1.jpg)
![](images/CAN_Module1_Lecture-78_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>PET-CT scan &amp; subsequent biopsy</b>
<i>Fukui et al. Radiographics.</i>
<i>2005 Jul-Aug;25(4):913-30.</i>
Primary tongue mass,
H&E stain, 20X objective
20X

---
# 79


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Surgery &amp; further pathology</b>
Laser resection of tumor & lymph nodes
<b>Primary:</b> 1.5 cm poorly differentiated adenocarcinoma with
micropapillary & mucinous features
<b>Lymph nodes:</b> 3 of 21 neck nodes contain metastatic
adenocarcinoma
60 Gy of adjuvant radiation therapy completed in Feb.
Good quality of life, returned to work for four months
Then…numerous small (< 1.2 cm) bilateral pulmonary metastases

---
# 80

![](images/CAN_Module1_Lecture-80_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>EGFR IHC positive expression</b>
+2 EGFR expression by Zymed
immunohistochemistry
protein expression assay
<i>Treatments?</i>
40X

---
# 81


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>6 week trial of erlotinib</b>
All pulmonary nodules grew while on erlotinib
Largest lesion grew from 1.5 to 2.1 cm
Erlotinib discontinued in August
<i>Palliative care? What next?</i>
August
Initial
presentation
August
Erlotinib
failure
November
Surgical
resection
February
Radiation
complete
June
Lung mets,
erlotinib begun

---
# 82


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>What are the targets in </b> <i><b>our</b></i> <b>case?</b>
Having exhausted standard of care, BC Cancer Agency
oncologist turned to the Genome Sciences Centre (a large
research group) for new leads
Special meeting of the BCCA Research Ethnics Board
approved a single-case analysis for this patient
Patient consented to full genomic sequencing and analysis
with the understanding that novel treatment options may
be  <i>suggested</i>

---
# 83

![](images/CAN_Module1_Lecture-83_1.jpg)
![](images/CAN_Module1_Lecture-83_2.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Fresh frozen biopsy taken for RNA-seq</b>
Fine-needle aspirates taken from large lung lesion
Pathologist reviewed, used samples with highest tumor
content (~80%) for RNA sequencing
FFPE DNA from surgical resection used for DNA
40X

---
# 84


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Somatic Mutations</b>
<i>TP53</i> p.D259Y Located in region of LOH
Known somatic mutation in 7 cancer types
(16 entries + 4 alternate mutations across 28
tumors in the Catalogue of Somatic Mutations
in Cancer)
<i>RB1</i> p.L234* Novel truncating mutation
Results in loss of 75% of protein sequence
Loss of  <i>RB1</i> and <i>PTEN</i> result in gefitinib
resistance
<i>(Albitar et al. </i> <i>Gynecol Oncol. 2007 Jul;106(1):94-104.</i> <i>)</i>
<i>ZNF187</i> p.G63C Novel mutation of unknown significance
Zinc-finger protein
<i>ZFPM2</i> p.K785E Novel mutation of unknown significance
Zinc-finger protein
Confirmed by secondary sequencing method (Sanger)

---
# 85

![](images/CAN_Module1_Lecture-85_1.jpg)
![](images/CAN_Module1_Lecture-85_2.jpg)
![](images/CAN_Module1_Lecture-85_3.png)
![](images/CAN_Module1_Lecture-85_4.png)
![](images/CAN_Module1_Lecture-85_5.png)
![](images/CAN_Module1_Lecture-85_6.png)
![](images/CAN_Module1_Lecture-85_7.png)
![](images/CAN_Module1_Lecture-85_8.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Copy number</b>
<b>alterations</b>
18q loss (SMAD4)
with focal amps.
<i>MAPK3</i>
amplification
10q loss ( <i>PTEN</i> )
Focal  <i>RET</i> amplification
<i>TP53</i> LOH
<i>EGFR</i>
amplification

---
# 86

![](images/CAN_Module1_Lecture-86_1.jpg)
![](images/CAN_Module1_Lecture-86_2.jpg)
![](images/CAN_Module1_Lecture-86_3.jpg)
![](images/CAN_Module1_Lecture-86_4.jpg)
![](images/CAN_Module1_Lecture-86_5.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>FISH confirmation of copy number alterations</b>
<i>RBBP8</i>
amplification
<i>PTEN</i> single
copy loss
Focal  <i>RET</i>
amplification
100X 40X 40X

---
# 87


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Tumour gene expression levels </b>
<b>compared against reference set of</b>
<b>50 tumors &amp; matched blood sample</b>
<i>SMAD4</i> expression 43X lower vs compendium
<i><b>deleted &amp; down-regulated, associated with </b></i>
<i><b>metastasis of colorectal cancer</b></i>
<i>RET</i> in top 5% of expressed genes, 34X vs compendium
<i><b>most highly expressed oncogene </b></i>
<i>PTEN</i> in bottom 5% of expressed genes,
<i><b>significantly under-expressed</b></i>

---
# 88

![](images/CAN_Module1_Lecture-88_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Tumor driven by RET up-regulation, </b>
<b>PTEN loss, not AKT</b>

---
# 89

![](images/CAN_Module1_Lecture-89_1.jpg)
![](images/CAN_Module1_Lecture-89_2.jpg)
![](images/CAN_Module1_Lecture-89_3.jpg)
![](images/CAN_Module1_Lecture-89_4.png)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Short list presented to oncologist</b>

---
# 90

![](images/CAN_Module1_Lecture-90_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>22% decrease after 4 weeks on sunitinib</b>
<b>(16% growth prior to sunitinib)</b>
Oct 1 st 2008
Biopsy taken
Oct 29 th 2008
Sunitinib begun
Dec 9 th 2008
4 wks 50mg daily Sunitinib
No
d
u
le
 1
No
d
u
le
 2

---
# 91


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Stabilization for 7 months</b>
Sunitinib dose reduced due to known side-
effects, otherwise excellent quality of life
Repeated scans showed  no new nodules and
disease stabilization and 4 months
…then, existing lung mets began to grow
Switched to sorafenib & sulindac
Disease again stabilized within 4 weeks and
continued for 3 months

---
# 92


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Recurrent disease after 7 months</b>
Recurrent disease at primary site on tongue
New neck skin nodule
Progressive & new metastases in lung
Deteriorating quality of life
<i>What changed?</i>

---
# 93


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Skin metastasis contains new mutations</b>
No evidence of these in the pre-treatment biopsy,
even at low frequency
<i>ZMYM4</i> p.Q317H Zinc-finger
<i>DNAH7</i> p.V2590L Force generating protein of
respiratory cilia
<i>CXCL13</i> p.R56H B-cell attracting chemokine
<i>HSD17B8</i> p.A141T hydroxysteroid (17-beta)
dehydrogenase 8
<i>PCLO</i> p.T2759A presynaptic cytomatrix protein
<i>GRIA4</i> p.R872C glutamate receptor
<i>OR4K2</i> p.L197F olfactory receptor
<i>SYNE2</i> p.A302G tethers the nucleus to the
cytoskeleton
<i>PTPRM</i> p.A929T cell-cell adhesion, possibly signal
transduction & growth

---
# 94

![](images/CAN_Module1_Lecture-94_1.jpg)

<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Resistance due to intensified RET</b>
<b>and new AKT signaling?</b>

---
# 95


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Overcoming mechanisms of resistance</b>
Cocktail of targeted drugs against multiple members of
activated and parallel pathways?
- RET, EGFR, mTOR, Akt, etc.
- untested, risk of adverse side-effects
Detection of resistance mechanisms pre-treatment
à none of the new mutations were evident pre-treatment
Can resistance be modeled & monitored?
- serial biopsies? blood test?

---
# 96


<b>Module  1 - Intro to Cancer Genomics</b> <b>bio</b> informatics
.ca
<b>Timeline</b>
August
Initial
presentation
August
Erlotinib
failure
November
Surgical
resection
February
Radiation
complete
June
Lung mets,
erlotinib begun
Oct
1 st biopsy for
sequencing
Nov
Sunitinib
begun
Feb
Lung mets grow, switched
to sorafinib & sulindac
May
Primary recurrence,
2 nd biopsy
Palliative care
until death

---
# 97

![](images/CAN_Module1_Lecture-97_1.jpg)
![](images/CAN_Module1_Lecture-97_2.jpg)
![](images/CAN_Module1_Lecture-97_3.jpg)

<b>We are on a Coffee Break &amp; </b>
<b>Networking Session</b>
Workshop Sponsors:
