<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Exome sequencing data analysis for diagnosing a genetic disease / Exome sequencing data analysis for diagnosing a genetic disease / Variant Analysis</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@be6a5055148f12d87e2a00fb270ceb96a709582c",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Exome sequencing data analysis for diagnosing a genetic disease">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-09 08:55:09 +0000">
<meta name="DC.creator" content="Wolfgang Maier">
<meta name="DC.creator" content="Bérénice Batut">
<meta name="DC.creator" content="Torsten Houwaart">
<meta name="DC.creator" content="Anika Erxleben">
<meta name="DC.creator" content="Björn Grüning"><meta name="description" content="Genetic differences (variants) between healthy and diseased tissue, between individuals of a population, or between strains of an organism can provide mechanistic insight into disease processes and the natural function of affected genes.  The tutorials in this section show how to detect evidence for genetic variants in next-generation sequencing data, a process termed variant calling.  Of equal importance, they also demonstrate how you can interpret, for a range of different organisms, the resulting sets of variants by predicting their molecular effects on genes and proteins, by annotating previously observed variants with published knowledge, and by trying to link phenotypes of the sequenced samples to their variant genotypes. ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Variant Analysis / Exome sequencing data analysis for diagnosing a genetic disease / Hands-on: Exome sequencing data analysis for diagnosing a genetic disease">
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseased tissue, between individuals of a population, or between strains of an organism can provide mechanistic insight into disease processes and the natural function of affected genes.  The tutorials in this section show how to detect evidence for genetic variants in next-generation sequencing data, a process termed variant calling.  Of equal importance, they also demonstrate how you can interpret, for a range of different organisms, the resulting sets of variants by predicting their molecular effects on genes and proteins, by annotating previously observed variants with published knowledge, and by trying to link phenotypes of the sequenced samples to their variant genotypes. ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/variant-analysis/tutorials/exome-seq/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Exome sequencing data analysis for diagnosing a genetic disease",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2023-11-09 08:55:09 +0000",
  "datePublished": "2016-08-19 14:05:48 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00311",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
  },
  "abstract": "Exome sequencing is a method that enables the selective sequencing of the",
  "learningResourceType": "e-learning",
  "name": "Exome sequencing data analysis for diagnosing a genetic disease",
  "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html",
  "version": 27,
  "timeRequired": "PT5H",
  "teaches": "- Jointly call variants and genotypes for a family trio from whole-exome sequencing data\n- Use variant annotation and the observed inheritance pattern of a phenotype to identify candidate causative variants and to prioritize them",
  "keywords": [
    "Variant Analysis"
  ],
  "description": "## Abstract\n\nExome sequencing is a method that enables the selective sequencing of the\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - How do you identify genetic variants in samples based on exome sequencing data?\n - How do you, among the set of detected variants, identify candidate causative variants for a given phenotype/disease?\n\n\n## Learning Objectives\n\n- Jointly call variants and genotypes for a family trio from whole-exome sequencing data\n- Use variant annotation and the observed inheritance pattern of a phenotype to identify candidate causative variants and to prioritize them\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bebatut/",
      "name": "Bérénice Batut",
      "image": "https://avatars.githubusercontent.com/bebatut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/gallantries/",
          "name": "Gallantries: Bridging Training Communities in Life Science, Environment and Health",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.erasmusplus.nl"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/ifb/",
          "name": "Institut Français de Bioinformatique",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.france-bioinformatique.fr/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9852-1987",
      "orcid": "https://orcid.org/0000-0001-9852-1987"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/torhou/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/torhou/",
      "name": "Torsten Houwaart",
      "image": "https://avatars.githubusercontent.com/torhou",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/erxleben/",
      "name": "Anika Erxleben",
      "image": "https://avatars.githubusercontent.com/erxleben",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eurosciencegateway/",
          "name": "EuroScienceGateway",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://galaxyproject.org/projects/esg/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7427-6478",
      "orcid": "https://orcid.org/0000-0002-7427-6478"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/bgruening/",
      "name": "Björn Grüning",
      "image": "https://avatars.githubusercontent.com/bgruening",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eurosciencegateway/",
          "name": "EuroScienceGateway",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://galaxyproject.org/projects/esg/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/eosc-life/",
          "name": "EOSC-Life",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.eosc-life.eu"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/deNBI/",
          "name": "de.NBI",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.denbi.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/sfb992/",
          "name": "Collaborative Research Centre 992",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.sfb992.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-converge/",
          "name": "ELIXIR-CONVERGE",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org/about-us/how-funded/eu-projects/converge"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-3079-6586",
      "orcid": "https://orcid.org/0000-0002-3079-6586"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.3054169",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Variant Analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/variant-analysis/faqs/" title="Check our FAQs for the Variant Analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/exome-seq/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-variant-analysis">
    <h1 data-toc-skip>Exome sequencing data analysis for diagnosing a genetic disease</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Wolfgang Maier avatar" width="36" class="avatar" />
    Wolfgang Maier</a><a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge contributor-bebatut"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bebatut?s=36" alt="Bérénice Batut avatar" width="36" class="avatar" />
    Bérénice Batut</a><a href="/training-material/hall-of-fame/torhou/" class="contributor-badge contributor-torhou"><img src="https://avatars.githubusercontent.com/torhou?s=36" alt="Torsten Houwaart avatar" width="36" class="avatar" />
    Torsten Houwaart</a><a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge contributor-erxleben"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/erxleben?s=36" alt="Anika Erxleben avatar" width="36" class="avatar" />
    Anika Erxleben</a><a href="/training-material/hall-of-fame/bgruening/" class="contributor-badge contributor-bgruening"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/bgruening?s=36" alt="Björn Grüning avatar" width="36" class="avatar" />
    Björn Grüning</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How do you identify genetic variants in samples based on exome sequencing data?</p>
</li>
        
        <li><p>How do you, among the set of detected variants, identify candidate causative variants for a given phenotype/disease?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Jointly call variants and genotypes for a family trio from whole-exome sequencing data</p>
</li>
        
        <li><p>Use variant annotation and the observed inheritance pattern of a phenotype to identify candidate causative variants and to prioritize them</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Quality Control</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Quality Control</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mapping</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mapping</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 5 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.3054169">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/variant-analysis/tutorials/exome-seq/workflows/" title="Exome sequencing data analysis for diagnosing a genetic disease workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/exome-seq/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Aug 19, 2016 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 9, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00311">https://gxy.io/GTN:T00311</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">4.8</a> (5 recent ratings, 32 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 27 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Exome sequencing is a method that enables the selective sequencing of the
exonic regions of a genome - that is the transcribed parts of the genome present
in mature mRNA, including protein-coding sequences, but also untranslated
regions (UTRs).</p>

<p>In humans, there are about 180,000 exons with a combined length of ~ 30 million
base pairs (30 Mb). Thus, the exome represents only 1% of the human genome, but
has been estimated to harbor up to 85% of all disease-causing variants (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2768590/">Choi
et al., 2009</a>).</p>

<p>Exome sequencing, thus, offers an affordable alternative to whole-genome
sequencing in the diagnosis of genetic disease, while still covering far more
potential disease-causing variant sites than genotyping arrays. This is of
special relevance in the case of rare genetic diseases, for which the causative
variants may occur at too low a frequency in the human population to be
included on genotyping arrays.</p>

<p>Of note, a recent study focusing on the area of clinical pediatric neurology
indicates that the costs of exome sequencing may actually not be higher even
today than the costs of conventional genetic testing (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5589982/">Vissers et al., 2017</a>).</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-exome-sequencing-i-vs-i-whole-genome-sequencing"><button class="gtn-boxify-button details" type="button" aria-controls="details-exome-sequencing-i-vs-i-whole-genome-sequencing" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Exome sequencing <i>vs</i> whole-genome sequencing</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>In principle, the steps illustrated in this tutorial are suitable also for
the analysis of whole-genome sequencing (WGS) data. At comparable mean
coverage, however, WGS datasets will be much larger than exome sequencing
ones and their analysis will take correspondingly more time.</p>

  <p>The obvious benefit of WGS compared to exome-sequencing, of course, is that
it will allow variant detection in even more regions of the genome.
As a less apparent advantage, the more complete information of WGS data can
make it easier to detect <em>copy number variation (CNV)</em> and
<em>structural variants</em> such as translocations and inversions (although such
detection will require more sophisticated analysis steps, which are not
covered by this tutorial).</p>

  <p>Very generally, one could argue that exome-sequencing captures most of the
information that can be analyzed with standard bioinformatical tools today at
reasonable costs. WGS, on the other hand, captures as much information as
today’s sequencing technology can provide, and it may be possible to
reanalyze such data with more powerful bioinformatical software in the
future to exploit aspects of the information that were not amenable to
analysis at the time of data acquisition.</p>
</blockquote>

<p>The identification of causative variants underlying any particular genetic
disease is, as we will see in this tutorial, not just dependent on the
successful detection of variants in the genome of the patient, but also on
variant comparison between the patient and selected relatives. Most often
family trio data, consisting of the genome sequences of the patient and their
parents, is used for this purpose. With multisample data like this it becomes
possible to search for variants following any kind of Mendelian inheritance
scheme compatible with the observed inheritance pattern of the disease, or to
detect possibly causative <em>de-novo</em> mutations or <em>loss-of-heterozygosity</em> (LOH) events.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-related-tutorials"><button class="gtn-boxify-button details" type="button" aria-controls="details-related-tutorials" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Related tutorials</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>This tutorial focuses on the practical aspects of analyzing real-world
patient data. If you are more interested in the theoretical aspects of
variant calling, you may want to have a look at the related tutorial on
<a href="../dip/tutorial.html">Calling variants in diploid systems</a>.</p>

  <p>The tutorial on <a href="../somatic-variants/tutorial.html">Somatic variant calling</a>
follows an analysis workflow that is rather similar to the one here, but
tries to identify tumor variants by comparing a tumor sample to healthy
tissue from the same patient. Together, the two tutorials are intended to get
you started with genomics medicine using Galaxy.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
  <li><a href="#read-mapping" id="markdown-toc-read-mapping">Read mapping</a></li>
  <li><a href="#mapped-reads-postprocessing" id="markdown-toc-mapped-reads-postprocessing">Mapped reads postprocessing</a>    <ol>
      <li><a href="#filtering-on-mapped-reads-properties" id="markdown-toc-filtering-on-mapped-reads-properties">Filtering on mapped reads properties</a></li>
      <li><a href="#removing-duplicate-reads" id="markdown-toc-removing-duplicate-reads">Removing duplicate reads</a></li>
    </ol>
  </li>
  <li><a href="#variant-calling" id="markdown-toc-variant-calling">Variant calling</a>    <ol>
      <li><a href="#generating-freebayes-calls" id="markdown-toc-generating-freebayes-calls">Generating FreeBayes calls</a></li>
      <li><a href="#post-processing-freebayes-calls" id="markdown-toc-post-processing-freebayes-calls">Post-processing FreeBayes calls</a></li>
    </ol>
  </li>
  <li><a href="#variant-annotation-and-reporting" id="markdown-toc-variant-annotation-and-reporting">Variant annotation and reporting</a>    <ol>
      <li><a href="#get-data-1" id="markdown-toc-get-data-1">Get data</a></li>
      <li><a href="#variant-annotation-with-functional-genomic-effects" id="markdown-toc-variant-annotation-with-functional-genomic-effects">Variant annotation with functional genomic effects</a></li>
      <li><a href="#generating-a-gemini-database-of-variants-for-further-annotation-and-efficient-variant-queries" id="markdown-toc-generating-a-gemini-database-of-variants-for-further-annotation-and-efficient-variant-queries">Generating a GEMINI database of variants for further annotation and efficient variant queries</a></li>
      <li><a href="#candidate-variant-detection" id="markdown-toc-candidate-variant-detection">Candidate variant detection</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>In this tutorial, we are going to analyze exome sequencing data from a family
trio, in which the boy child is affected by the disease
<a href="https://ghr.nlm.nih.gov/condition/osteopetrosis/">osteopetrosis</a>, while both
parents, who happen to be consanguineous, are unaffected. Our goal is to
identify the genetic variation that is responsible for the disease.</p>

<h2 id="get-data">Get data</h2>

<p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> This tutorial offers <strong>two alternative entry points</strong> allowing
you to</p>

<ul>
  <li>conduct a full analysis starting from original sequenced reads in <code class="language-plaintext highlighter-rouge">fastq</code>
format or to</li>
  <li>start your analysis with premapped reads in <code class="language-plaintext highlighter-rouge">bam</code> format that are (almost)
ready for variant calling.</li>
</ul>

<p>While the full analysis is probably closer to how you would analyze your own
data, the shortened analysis from premapped reads may suit your time frame
better, and will avoid redundancy if you have previously worked through other
tutorials demonstrating NGS quality control and read mapping, like the
dedicated
<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality control</a>
and <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">Mapping</a>
tutorials.</p>

<p>The following hands-on section will guide you through obtaining the right
data for either analysis.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Obtain the raw sequencing data</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-starting-from-raw-sequencing-data"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Starting from raw sequencing data</div>
        <p>In this and the following steps you will obtain the original unmapped
sequencing data and prepare for a full analysis including the mapping of
the sequenced reads.</p>

        <p>If you prefer <strong>to skip the mapping step</strong> and start the analysis from
premapped data <strong>you should proceed directly to step 4</strong> of this
section.</p>
      </blockquote>

      <p>Import the original sequenced reads datasets of the family trio from
<a href="https://zenodo.org/record/3054169">Zenodo</a>:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3243160/files/father_R1.fq.gz
https://zenodo.org/record/3243160/files/father_R2.fq.gz
https://zenodo.org/record/3243160/files/mother_R1.fq.gz
https://zenodo.org/record/3243160/files/mother_R2.fq.gz
https://zenodo.org/record/3243160/files/proband_R1.fq.gz
https://zenodo.org/record/3243160/files/proband_R2.fq.gz
</code></pre></div>      </div>

      <p>Alternatively, the same files may be available on your Galaxy server
through a shared data library (your instructor may tell you so), in
which case you may prefer to import the data directly from there.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Change <strong>Type (set all):</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Check that the newly created datasets in your history have their
datatypes assigned correctly to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>, and fix any missing or
wrong datatype assignment</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Congratulations for obtaining the datasets required for
an analysis including reads mapping. You should now
<strong>proceed with Step 7</strong> below.</p>
    </li>
    <li>Obtain the premapped sequencing data
      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-alternative-entry-point-premapped-data"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Alternative entry point: Premapped data</div>
        <p>Skip this and the following two steps if you already obtained and
prepared the original unmapped seuencing data and are planning to
perform the mapping step yourself.</p>
      </blockquote>

      <p>Import the premapped reads datasets of the family trio from
<a href="https://zenodo.org/record/3054169">Zenodo</a>:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3243160/files/mapped_reads_father.bam
https://zenodo.org/record/3243160/files/mapped_reads_mother.bam
https://zenodo.org/record/3243160/files/mapped_reads_proband.bam
</code></pre></div>      </div>

      <p>Alternatively, the same files may be available on your Galaxy server
through a shared data library (your instructor may tell you so), in
which case you may prefer to import the data directly from there.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Change <strong>Type (set all):</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">bam</code></p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Check that the newly created datasets in your history have their
datatypes assigned correctly to <code class="language-plaintext highlighter-rouge">bam</code>, and fix any missing or wrong
datatype assignment</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">bam</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Specify the genome version that was used for mapping</p>

      <p>Change the database/build (dbkey) for each of your bam datasets
to <code class="language-plaintext highlighter-rouge">hg19</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click the desired dataset’s name to expand it.</li>   <li>     <p>Click on the “?” next to database indicator:</p>     <p><img src="/training-material/shared/images/datasets_dbkey.svg" alt="UI for changing dbkey" /></p>   </li>   <li>In the central panel, change the <strong>Database/Build</strong> field</li>   <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="details">
        <div class="box-title details-title" id="details-why-specify-genome-versions"><button class="gtn-boxify-button details" type="button" aria-controls="details-why-specify-genome-versions" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Why specify genome versions</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
        <p>When you are starting with sequencing data that has already been mapped
to a particular genome version (human hg19 in this case), it is good
practice to attach this information as metadata to the datasets.</p>

        <p>Doing so helps prevent accidental use of a different version of the
reference genome in later steps like variant calling, which would
typically lead to nonsensical results because of base position changes
between the different versions.</p>
      </blockquote>

      <p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Congratulations for obtaining the premapped sequencing
datasets. Now, <strong>follow the remaining steps</strong> to set everything up for a
successful analysis.</p>
    </li>
    <li>
      <p>Rename the datasets</p>

      <p>For datasets that you upload via a link, Galaxy will pick the link
address as the dataset name, which you will likely want to shorten to
just the file names.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Add #father/#mother/#child tags to the datasets</p>

      <p>Parts of the analysis in this tutorial will consist of identical steps
performed on the data of each family member.</p>

      <p>To make it easier to keep track of which dataset represents which step in
the analysis of which sample, Galaxy supports dataset tags. In particular,
if you attach a tag starting with <code class="language-plaintext highlighter-rouge">#</code> to any dataset, that tag will
automatically propagate to any new dataset derived from the tagged
dataset.</p>

      <p>Tags are supposed to help you identify the origin of datasets quickly,
but you can choose them as you like.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Obtain the reference genome</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-shortcut"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Shortcut</div>
        <p>You can skip this step if the Galaxy server you are working on offers
a <code class="language-plaintext highlighter-rouge">hg19</code> version of the human reference genome with prebuilt indexes for
<em>bwa-mem</em> (only necessary if starting from unmapped original sequencing
reads) and <em>freebayes</em>. Ask your instructor, or check the tools
<strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
if they list a <code class="language-plaintext highlighter-rouge">hg19</code> version as an option under
<em>“Select a / Using reference genome”</em>).</p>
      </blockquote>

      <p>Import the <code class="language-plaintext highlighter-rouge">hg19</code> version of the human chromosome 8 sequence:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3243160/files/hg19_chr8.fa.gz
</code></pre></div>      </div>

      <p>In the upload dialog, make sure you specify:</p>

      <ul>
        <li><strong>Type</strong>: <code class="language-plaintext highlighter-rouge">fasta</code></li>
        <li><strong>Genome</strong>: <code class="language-plaintext highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code></li>
      </ul>

      <p>Alternatively, load the dataset from a shared data library.</p>
    </li>
    <li>
      <p>Rename the reference genome</p>

      <p>The reference genome you have imported above came as a compressed
file, but got unpacked by Galaxy to plain <code class="language-plaintext highlighter-rouge">fasta</code> format according to
your datatype selection. At a minimum, you may now wish to remove the
<code class="language-plaintext highlighter-rouge">.gz</code> suffix from the dataset name to avoid confusion.</p>
    </li>
  </ol>

</blockquote>

<p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> <strong>Congratulations!</strong>
You are all set for starting the analysis now.</p>

<p>If you have chosen to follow the complete analysis from the original sequenced
data, just proceed with the next section.</p>

<p>If, on the other hand, you have prepared to start from the premapped data, skip
the sections on <em>Quality control</em> and <em>Read mapping</em>, and conitnue with
<strong>Mapped reads postprocessing</strong>.</p>

<h1 id="quality-control">Quality control</h1>

<p>This step serves the purpose of identifying possible issues with the raw
sequenced reads input data before embarking on any “real” analysis steps.</p>

<p>Some of the typical problems with NGS data can be mitigated by preprocessing
affected sequencing reads before trying to map them to the reference genome.
Detecting some other, more severe problems early on may at least save you a lot
of time spent on analyzing low-quality data that is not worth the effort.</p>

<p>Here, we will perform a standard quality check on our input data and only point
out a few interesting aspects about that data. For a more thorough explanation
of NGS data quality control, you may want to have a look at the dedicated
tutorial on <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality control</a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-control-of-the-input-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality control of the input datasets</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.74+galaxy0" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.74+galaxy0)</span> on each of your six fastq datasets
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: all 6 FASTQ  datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>When you start this job, twelve new datasets (one with the calculated raw
data, another one with an html report of the findings for each input
dataset) will get added to your history.</p>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.11+galaxy1" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.11+galaxy1)</span> to aggregate the raw <strong>FastQC</strong> data of all input datasets into one report
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: all six <em>RawData</em>
outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the <em>Webpage</em> output produced by the tool</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>Based on the report, do you think preprocessing of the reads
(trimming and/or filtering) will be necessary before mapping?</li>
          <li>Why do all samples show a non-normal GC content distribution, and
should you be worried?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>Sequence quality is quite good overall. If anything you might
consider trimming the 3’ ends of reads (base qualities decline
slightly towards the 3’ ends) or to filter out the small fraction
of reads with a mean base quality &lt; 5.
Feel free to run, <em>e.g.</em>, <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.38.1" title="Trimmomatic tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Trimmomatic</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.38.1)</span> on the
fastq datasets if you want to, but don’t expect this to have a big
effect on the analysis given the high overall quality of the data
of all samples.</li>
            <li>
              <p>In whole-genome sequencing, a non-normal distribution of the GC
content of the reads from a sample is typically considered to hint
at possible contamination.
Here, however, we are dealing with sequencing data from captured
exomes, <em>i.e</em>, the reads are not representing random sequences from
a genome, but rather a biased selection.</p>

              <p>A bimodal GC content distribution, like for the samples at hand, is
a characteristic feature of many exome capture methods and has also
been observed with an Illumina Nextera Rapid Capture exome kit
before (compare Fig. 3D in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4522667/">Shigemizu et al., 2015</a>
).</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="read-mapping">Read mapping</h1>

<p>Now that you confirmed that the quality of the input data is good enough to
warrant further analysis, it is time to map the sequenced reads to the
reference genome.</p>

<p>We assume here that you are at least vaguely familiar with the concept of read
mapping and only illustrate the concrete steps necessary  to map our specific
NGS reads datasets to the human reference genome. We recommend you to follow
the dedicated <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">Mapping tutorial</a>, if you need a general introduction to read mapping.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-mapping"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read Mapping</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Map with BWA-MEM tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with BWA-MEM</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.7.17.2)</span> to map the reads from the <strong>father</strong> sample to the reference genome
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> chr8 sequence as a fasta dataset into
your history instead:</p>
            <ul>
              <li><em>“Will you select a reference genome from your history or use a
built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the
forward reads (R1) dataset of the <strong>father</strong> sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the
reverse reads (R2) dataset of the <strong>father</strong> sample</li>
          </ul>

          <blockquote class="tip">
            <div class="box-title tip-title" id="tip-no-fastq-datasets-selectable"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-no-fastq-datasets-selectable" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: No FASTQ datasets selectable?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
            <p>Please confirm that the problematic datasets declare <em>format</em>:
<code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>.</p>

            <p>Most Galaxy tools that accept FASTQ input expect the data to be
formatted as <em>FASTQ with Sanger-scaled quality values</em>, the most
widely spread version of the FASTQ format. To make this requirement
explicit (instead of generating possibly wrong results) these tools
require you to set the dataset type to <code class="language-plaintext highlighter-rouge">fastqsanger</code> (<code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>
for data compressed with gzip). You can do so either on data upload
or later from the <em>Edit dataset attributes</em> view (which you can reach
by clicking on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> pencil icon.</p>
          </blockquote>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">000</code></li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">father</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="warning">
        <div class="box-title warning-title" id="warning-read-group-ids-and-sample-names-choose-but-choose-wisely"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Read group IDs and sample names - choose, but choose wisely</div>
        <p>In general, you are free to choose ID and SM values to your liking, but
…</p>

        <p>The <strong>ID</strong> should <strong>unambiguously identify</strong> the sequencing run that
produced the reads. At the very least, no two input datasets in any
given analysis should define the same ID twice, or tools like
<em>FreeBayes</em>, which we are going to use in the next step, will refuse
to work with the data.</p>

        <p>The <strong>SM</strong> value, on the other hand, should identify the biological
sample represented by the data and is used by many tools (like <em>GEMINI</em>
which we will use later) to let you refer to one specifc sample in a
multisample analysis. Choose descriptive, but short and easy to
remember sample names since you will have to type them in again!</p>

      </blockquote>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Map with BWA-MEM tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with BWA-MEM</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.7.17.2)</span> to map the reads from the <strong>mother</strong> sample to the reference genome <strong>using the same parameters as before</strong> except</p>

      <ul>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the
forward reads (R1) dataset of the <strong>mother</strong> sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the
reverse reads (R2) dataset of the <strong>mother</strong> sample</li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">001</code></li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">mother</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.2" title="Map with BWA-MEM tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with BWA-MEM</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.7.17.2)</span> to map the reads from the <strong>child</strong> sample to the reference genome <strong>using the same parameters as before</strong> except</p>

      <ul>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the
forward reads (R1) dataset of the <strong>child</strong> sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the
reverse reads (R2) dataset of the <strong>child</strong> sample</li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">002</code></li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">proband</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="mapped-reads-postprocessing">Mapped reads postprocessing</h1>

<p>At this point in the analysis you should have obtained three mapped reads
datasets in <code class="language-plaintext highlighter-rouge">bam</code> format. Each of these datasets should:</p>

<ul>
  <li>
    <p>have its <em>database</em> set to the key <code class="language-plaintext highlighter-rouge">hg19</code></p>

    <p>Please correct any missing (<code class="language-plaintext highlighter-rouge">?</code>) or wrong keys now!</p>

    <!--SNIPPET-->
    <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-database-build-dbkey-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-database-build-dbkey-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click the desired dataset’s name to expand it.</li>   <li>     <p>Click on the “?” next to database indicator:</p>     <p><img src="/training-material/shared/images/datasets_dbkey.svg" alt="UI for changing dbkey" /></p>   </li>   <li>In the central panel, change the <strong>Database/Build</strong> field</li>   <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code></li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
    <p><!--END_SNIPPET--></p>
  </li>
  <li>
    <p>ideally, carry one of the <code class="language-plaintext highlighter-rouge">#father</code>, <code class="language-plaintext highlighter-rouge">#mother</code> or <code class="language-plaintext highlighter-rouge">#child</code> tags
for quick identification of the samples they provide data for.</p>
  </li>
</ul>

<p>In principle, you could use these datasets directly for variant calling, and in
many cases, including this one, this would be sufficient to identify the
sought-after variants.</p>

<p>To obtain an accurate picture of the variant spectrum found in your samples it
is good practice though to perform various postprocessing steps on the mapped
reads before passing them to a variant caller.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-shortcut-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Shortcut</div>
  <p>If you are in a hurry, you may skip the postprocessing steps and continue
with your mapped reads datasets and the <em>Variant Calling</em> section. Make sure
though you come back here, or read about mapped reads postprocessing
elsewhere, before attempting to perform any variant analysis on your own data
that may require high accuracy/sensitivity.</p>
</blockquote>

<p>The optimal set of postprocessing steps required depends on the variant calling
software used at the next step. The <strong>FreeBayes</strong> variant caller that we are
going to use in this tutorial is particularly well suited for use with minimal
mapped reads postprocessing pipelines, so all we are going to do here is:</p>

<ul>
  <li>
    <p>filter the paired-end reads of all samples to retain only those read pairs,
for which both the forward and the reverse read have been mapped to the
reference successfully</p>

    <p>For such pairs of reads, we can be extra confident that they don’t
come from some non-human contaminant DNA or represent a sequencing artefact
of some sort.</p>
  </li>
  <li>
    <p>deduplicate reads</p>

    <p>Duplicate reads, which typically arise from PCR-overamplification of genomic
fragments during sequencing library preparation, can, to some extent, lead to
wrong genotype assignments at variant sites (if, for example, a sample is
heterozygous for a variant, but fragments with one of the two alleles get
amplified more efficiently than the others).</p>
  </li>
</ul>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-more-postprocessing-steps"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More postprocessing steps</div>
  <p>The <a href="../somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
provides an example of a much more complex mapped reads postprocessing
pipeline that may be required for optimal performance of other variant
callers.</p>
</blockquote>

<h2 id="filtering-on-mapped-reads-properties">Filtering on mapped reads properties</h2>

<p>To produce new filtered BAM datasets with only mapped reads the mate of which is also mapped:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filtering-for-read-pair-mapping-status"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filtering for read pair mapping status</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0" title="Samtools view tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Samtools view</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.15.1+galaxy0)</span> with the following
parameters (leaving non-mentioned ones at their defaults):
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“SAM/BAM/CRAM data set”</em>: all 3 mapped reads
 datasets of the family trio, outputs of <strong>Map with BWA-MEM</strong>
 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code></li>
      </ul>
      <ul>
        <li>In <em>“Configure filters”</em>:
          <ul>
            <li><em>“Exclude reads with any of the following flags set”</em>:
<code class="language-plaintext highlighter-rouge">Read is unmapped</code> <strong>and</strong> <code class="language-plaintext highlighter-rouge">Mate is unmapped</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This will result in three new datasets, one for each sample in the analysis.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-than-one-way-to-filter"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-than-one-way-to-filter" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: More than one way to filter</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>Instead of the above filter conditions we could also have exploited the
<em>Read is mapped in a proper pair</em> flag bit.</p>

  <p>For a read to be flagged as being mapped in a proper pair its mate needs to
be mapped, but the mapped pair also needs to meet additional,
aligner-specific criteria. These may include (and do so for <strong>BWA-MEM</strong>):</p>

  <ul>
    <li>both reads need to map to the same chomosome</li>
    <li>the two reads need to map to the reference in the orientation expected by
the aligner</li>
    <li>the two read pairs need to map to the reference within an
aligner-determined distance</li>
  </ul>

  <p>Thus, filtering based on the flag has two consequences:</p>

  <ul>
    <li>filtering will be stricter than with just the <em>Read is unmapped</em> and <em>Mate
is unmapped</em> flags above</li>
    <li>
      <p>you will eliminate read pairs that could be informative with regard to
chromosomal rearrangements and insertion/deletion events</p>

      <p>=&gt; Do not filter for properly paired reads if you plan to detect such
structural variants!</p>
    </li>
  </ul>

  <p>In addition, the <em>proper pair</em> flag is considered undefined if the read
itself is unmapped, so a <em>proper pair</em> filter should eliminate unmapped reads
explicitly to be on the safe side.</p>

  <p>Thus, if you would like to use proper pair filtering (we have no intention to
detect structural variants in this tutorial) instead of just filtering for
mapped reads with a mapped mate, you could run the alternative:</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/samtools_view/samtools_view/1.15.1+galaxy0" title="Samtools view tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Samtools view</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.15.1+galaxy0)</span>:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“SAM/BAM/CRAM data set”</em>: all 3 mapped reads
 datasets of the family trio, outputs of <strong>Map with BWA-MEM</strong>
 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“What would you like to look at?”</em>: <code class="language-plaintext highlighter-rouge">A filtered/subsampled selection of reads</code></li>
        </ul>
        <ul>
          <li>In <em>“Configure filters”</em>:
            <ul>
              <li><em>“Require that these flags are set”</em>: <code class="language-plaintext highlighter-rouge">Read is mapped in a proper pair</code></li>
              <li><em>“Exclude reads with any of the following flags set”</em>: <code class="language-plaintext highlighter-rouge">Read is unmapped</code></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="removing-duplicate-reads">Removing duplicate reads</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Remove duplicates</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1" title="RmDup tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>RmDup</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM file”</em>: all 3 filtered reads datasets; the
 outputs of <strong>Samtools view</strong></li>
        <li><em>“Is this paired-end or single end data”</em>: <code class="language-plaintext highlighter-rouge">BAM is paired-end</code></li>
      </ul>
      <ul>
        <li><em>“Treat as single-end”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Again, this will produce three new datasets, one for each member of the family
trio.</p>

<h1 id="variant-calling">Variant calling</h1>

<p>With the sequenced reads of all samples mapped and postprocessed, we can start
looking for evidence of sequence deviations, <em>i.e.</em> variants, between the
sequenced genomic samples and the reference genome.</p>

<p>This task has been automated and optimized continuously over the last decade,
and modern variant calling software hides much of the complexity involved in
it. At least a basic understanding of the underlying concepts is still highly
recommended though and, if you are new to variant calling, the tutorial on
<a href="../dip/tutorial.html">Calling variants in diploid systems</a> may be a good
starting point for you.</p>

<h2 id="generating-freebayes-calls">Generating FreeBayes calls</h2>

<p>We will use <strong>FreeBayes</strong> to call our variants. <strong>FreeBayes</strong> is a Bayesian
genetic variant detector designed to find small polymorphisms, specifically
SNPs (single-nucleotide polymorphisms), indels (insertions and deletions), MNPs
(multi-nucleotide polymorphisms), and complex events (composite insertion and
substitution events) smaller than the length of a short-read sequencing
alignment.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generating-freebayes-calls"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generating FreeBayes calls</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/freebayes/freebayes/1.3.6+galaxy0" title="FreeBayes tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FreeBayes</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span>:
      <ul>
        <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
          <ul>
            <li><em>“Run in batch mode?”</em>: <code class="language-plaintext highlighter-rouge">Merge output VCFs</code>
              <ul>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM dataset(s)”</em>: all three mapped reads
datasets of the family trio; the outputs of <strong>RmDup</strong></li>
              </ul>
            </li>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> chr8 sequence as a fasta dataset
into your history instead:</p>
            <ul>
              <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">History</code>
                <ul>
                  <li><em>“Run in batch mode?”</em>: <code class="language-plaintext highlighter-rouge">Merge output VCFs</code>
                    <ul>
                      <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM or CRAM dataset(s)”</em>: all three
 mapped and fully post-processed reads datasets of the family trio;
 outputs of <strong>RmDup</strong></li>
                    </ul>
                  </li>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the
reference sequence”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“Limit variant calling to a set of regions?”</em>: <code class="language-plaintext highlighter-rouge">Do not limit</code></li>
        <li><em>“Choose parameter selection level”</em>: <code class="language-plaintext highlighter-rouge">1. Simple diploid calling</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> <strong>Congratulations!</strong>
You have created you first multisample VCF file, one of the most complicated
file formats in bioinformatics. For every variant detected in at least one of
your samples, this tab-separated format uses a single line to store all
information about the variant. This includes but is not limited to:</p>

<ul>
  <li>the position of the variant in the genome (with respect to the reference
genome used for the analysis)</li>
  <li>the nature of the variant (the actual sequence change associated with it)</li>
  <li>the detected genotype of every sample at the variant position</li>
  <li>measures of the reliability of the variant call and of all individual
genotype calls</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-optional-hands-on-inspect-the-vcf-output-produced-by-freebayes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Optional hands-on: Inspect the VCF output produced by FreeBayes</div>

  <ol>
    <li>Display the VCF dataset:
      <ul>
        <li>
          <p>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> icon next to the VCF dataset generated
by FreeBayes to display its contents.</p>

          <p>VCF is a tabular plain text format though its information density makes
it complicated to  understand.</p>
        </li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>Can you locate at least some of the above-listed information in the
dataset?</p>

        <p>Hints:</p>
        <ul>
          <li>Lines starting with <code class="language-plaintext highlighter-rouge">##</code> are comment lines explaining the content of
the file.</li>
          <li>Diploid genotypes at biallelic sites are encoded using <code class="language-plaintext highlighter-rouge">0/0</code>, <code class="language-plaintext highlighter-rouge">0/1</code>
and <code class="language-plaintext highlighter-rouge">1/1</code> to represent homozygous reference, heterozygous and
homozygous variant states, respectively.</li>
        </ul>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ul>
            <li>
              <p>The position of each variant with respect to the reference genome is
stored in the first two columns (labeled <code class="language-plaintext highlighter-rouge">CHROM</code> and <code class="language-plaintext highlighter-rouge">POS</code>).</p>
            </li>
            <li>
              <p>The sequence change associated with the variant is provided by
columns 4 and 5 (<code class="language-plaintext highlighter-rouge">REF</code> and <code class="language-plaintext highlighter-rouge">ALT</code>).</p>

              <p>The first of these lists the sequence found at the variant site in
the reference genome, the second the variant sequence.</p>
            </li>
            <li>
              <p>The genotypes of the samples are stored as part of columns 10 and
up.</p>

              <p>Column 9 (<code class="language-plaintext highlighter-rouge">FORMAT</code>) provides the legend for understanding the
sample-specific columns following it and tells us that the first
element (before the first <code class="language-plaintext highlighter-rouge">:</code> separator) of each sample column holds
the sample genotype (<code class="language-plaintext highlighter-rouge">GT</code>).</p>
            </li>
            <li>
              <p>The variant call quality (a measure of the reliability of the
variant existing in any of the samples, <em>i.e.</em> it being real) is
stored in column 6 (<code class="language-plaintext highlighter-rouge">QUAL</code>).
The values are <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred-scaled</a>
like sequenced reads base qualities.</p>

              <p>The (log10-scaled) likelihoods of individual genotype calls are
stored as the last element (<code class="language-plaintext highlighter-rouge">GL</code>) in every sample column.</p>
            </li>
          </ul>

          <p>If you are eager to learn all the details about the format, you could
consult the excellent (but also long) official
<a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf">VCF specification</a>.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="post-processing-freebayes-calls">Post-processing FreeBayes calls</h2>

<p>Before starting to analyze the detected variants, we need to post-process the
VCF dataset to fix a few incompatibilities between Freebayes and downstream
analysis tools. In particular, we want to:</p>

<ul>
  <li>
    <p>Split multiallelic variant records, <em>i.e.</em>, records that list more than one
alternate allele at a given genomic position, into separate record lines.</p>

    <p>This will allow us to annotate each record with information about the impact
of one specific variant allele further on.</p>
  </li>
  <li>
    <p>Make sure that indels are represented in left-aligned and normalized form
because this is how known indels are stored in public annotation databases.</p>

    <blockquote class="details">
      <div class="box-title details-title" id="details-indel-normalization"><button class="gtn-boxify-button details" type="button" aria-controls="details-indel-normalization" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Indel normalization</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>Indel normalization is a surprisingly complex topic, which is explained
really well and in detail in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481842/">Tan et al., 2015</a>.
The following figure illustrates the various possibilities to represent
an indel, only one of which is the <em>normalized</em> form:</p>
      <figure id="figure-1" style="max-width: 90%;"><img src="../../images/indel_normalization.png" alt="different ways to represent the same indel. " width="720" height="540" loading="lazy" /><a target="_blank" href="../../images/indel_normalization.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> The many ways to represent an indel (from: https://genome.sph.umich.edu/wiki/Variant_Normalization</figcaption></figure>
      <p>.</p>

    </blockquote>
  </li>
</ul>

<p>A tool that can do this and also ensures that a VCF dataset conforms to
standards in some other, less important respects is <strong>bcftools norm</strong>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-post-processing-freebayes-calls"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Post-processing FreeBayes calls</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bcftools_norm/bcftools_norm/1.15.1+galaxy3" title="bcftools norm tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bcftools norm</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.15.1+galaxy3)</span> with the following parameters:
      <ul>
        <li><em>“VCF/BCF Data”</em>: the VCF output of <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><em>“Reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> chr8 sequence as a fasta dataset
into your history instead:</p>
            <ul>
              <li><em>“Choose the source for the reference genome”</em>:
<code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><em>“Reference genome”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“When any REF allele does not match the reference genome base”</em>:
<code class="language-plaintext highlighter-rouge">ignore the problem (-w)</code></li>
        <li><em>“Atomize”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Left-align and normalize indels?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>
          <p><em>“Perform deduplication for the folowing types of variant records”</em>:
<code class="language-plaintext highlighter-rouge">do not deduplicate any records</code></p>

          <p>Freebayes is not producing any duplicate calls.</p>
        </li>
        <li><em>“~multiallelics”</em>: <code class="language-plaintext highlighter-rouge">split multiallelic sites into biallelic records (-)</code>
          <ul>
            <li>
              <p><em>“split the following variant types”</em>: <code class="language-plaintext highlighter-rouge">both</code></p>

              <p>We want to split both, multiallelic SNP and indel records.</p>
            </li>
          </ul>
        </li>
        <li>
          <p><em>“output_type”</em>: <code class="language-plaintext highlighter-rouge">uncompressed VCF</code></p>

          <p>We would like to keep the results human-readable. VCF is also what tools
like <em>SnpEff</em> and <em>GEMINI</em> expect as input. Compressed, binary BCF is
interesting for space-efficient long-term storage of large lists of
variants.</p>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>You could try to look for the differences between the original and the
normalized VCF dataset, but for convenience <strong>bcftools norm</strong> reports a brief
summary of the actions it performed. Expand the dataset in the history (by
clicking on its name) to see this output listing the total number of variant
lines processed, along with the number of split, realigned and skipped records.</p>

<h1 id="variant-annotation-and-reporting">Variant annotation and reporting</h1>

<p>A list of variants detected in a set of samples is a start, but to discover
biologically or clinically relevant information in it is almost impossible
without some additional tools and data. In particular, the
variants in the list need to be:</p>

<ul>
  <li>
    <p><strong>prioritized</strong> with respect to their potential relevance for the biological
/ clinical phenotype that is studied</p>

    <p>Even with exome sequencing, only a fraction of the detected variants will
have a clear impact on the function of a protein (many variants will
introduce silent mutations, or reside in intronic regions still covered by
the exome-enriched sequencing data). Of these, many will have been observed
before in healthy individuals arguing against them playing an important role
in an adverse phenotype.</p>
  </li>
  <li>
    <p><strong>filtered</strong> based on the inheritance pattern expected for a causative
variant</p>

    <p>A multisample VCF file records the most likely genotypes of all samples at
every variant site. Knowing which individuals (samples) are affected by a
phenotype we can exclude variants with inheritance patterns that are
incompatible with the observed inheritance of the phenotype.</p>
  </li>
  <li>
    <p><strong>reported</strong> in a more human-friendly form</p>

    <p>While the VCF format can be used to encode all relevant information about any
variant, it is hard for humans to parse that information.</p>

    <p>Ideally, one would like to generate simpler reports for any set of filtered
and prioritized variants.</p>
  </li>
</ul>

<h2 id="get-data-1">Get data</h2>

<p>Our workhorse for annotating and reporting variants and the genes affected by
them will be the <strong>GEMINI</strong> framework. GEMINI comes bundled with a wealth of
annotation data for human variants from many different sources. These can be
used to annotate any list of human variants conveniently, without the need for
separate downloads and conversion between different annotation data formats.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-custom-annotations-and-gemini"><button class="gtn-boxify-button details" type="button" aria-controls="details-custom-annotations-and-gemini" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Custom annotations and GEMINI</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>Beyond its bundled annotation data, GEMINI also provides (limited) support
for using custom annotations.
The <a href="../somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
demonstrates the use of <strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for this purpose.</p>
</blockquote>

<p>The only additional annotation tool we need, for the purpose of this
tutorial, is the tool <strong>SnpEff</strong>, which can annotate variants with their
calculated effects on known genomic features. Because SnpEff is a generic tool
that can be used on variants found in the genome of any organism we need to
provide it with a so-called <strong>SnpEff genome file</strong> that holds the annotated
features (genes, transcripts, translated regions, <em>etc.</em>) for our genome of
interest.</p>

<p>While annotated variants are all we need to <em>prioritize</em> them as described
above, <em>filtering based on inheritance patterns</em> requires a way to inform
GEMINI about the relationship between our samples and their observed
phenotypes. This is done through a so-called <strong>pedigree file</strong> in PED format,
which is rather simple to generate manually.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-obtain-snpeff-genome-and-gemini-pedigree-files"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Obtain SnpEff genome and GEMINI pedigree files</div>
  <ol>
    <li>
      <p>Download <strong>SnpEff functional genomic annotations</strong></p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-shortcut-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Shortcut</div>
        <p>You can skip this step if the Galaxy server you are working on offers
<code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> as a locally installed snpEff database. You can
check the <strong>Genome source</strong> select list of the
<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy2" title="SnpEff eff tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnpEff eff</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.3+T.galaxy2)</span>
tool to see if this is the case.</p>
      </blockquote>

      <p>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_download/4.3+T.galaxy2" title="SnpEff Download tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnpEff Download</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.3+T.galaxy2)</span> to download genome annotation
 database <code class="language-plaintext highlighter-rouge">hg19</code>.</p>
    </li>
    <li>
      <p>Create a PED-formatted pedigree dataset describing our single-family sample trio:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#family_id    name     paternal_id    maternal_id    sex    phenotype
FAM           father   0              0              1      1
FAM           mother   0              0              2      1
FAM           proband  father         mother         1      2
</code></pre></div>      </div>

      <p>and set its datatype to <code class="language-plaintext highlighter-rouge">tabular</code>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-file"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-file" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>   <li>Paste the file contents into the text field</li>   <li>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code>* Press <strong>Start</strong> and <strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <blockquote class="warning">
        <div class="box-title warning-title" id="warning-remember-those-sample-names"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Remember those sample names</div>

        <p>The above content of the pedigree dataset assumes you chose <code class="language-plaintext highlighter-rouge">father</code>,
<code class="language-plaintext highlighter-rouge">mother</code>, <code class="language-plaintext highlighter-rouge">proband</code> as the sample names at the read mapping step
(if you haven’t mapped the reads yourself, but started with the
premapped data, you can safely skip this warning section).
By now, these sample names will have been propagated through <em>BWA-MEM</em>
and <em>Freebayes</em> to the VCF dataset of variants. It is <strong>important</strong> that
you use matching sample names in the pedigree and in the VCF dataset, or
GEMINI will not be able to connect the information in them.</p>

        <p>If you have chosen different sample names before, you have to adjust
the pedigree dataset accordingly!</p>

      </blockquote>

      <blockquote class="details">
        <div class="box-title details-title" id="details-more-on-ped-files"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-on-ped-files" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: More on PED files</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>The PED format is explained in the help section of
<span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+galaxy2" title="GEMINI load tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI load</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span>.</p>

        <p>Take a moment and try to understand the information that is encoded in
the PED dataset we are using here.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="variant-annotation-with-functional-genomic-effects">Variant annotation with functional genomic effects</h2>

<p>We need to start annotating our variants with SnpEff simply because Gemini
knows how to parse SnpEff-annotated VCFs, while GEMINI output cannot be used
with SnpEff.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-adding-annotations-with-snpeff"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Adding annotations with SnpEff</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy2" title="SnpEff eff tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnpEff eff</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.3+T.galaxy2)</span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sequence changes (SNPs, MNPs, InDels)”</em>: the
output of <strong>bcftools norm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Input format”</em>: <code class="language-plaintext highlighter-rouge">VCF</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">VCF (only if input is VCF)</code></li>
        <li><em>“Genome source”</em>: <code class="language-plaintext highlighter-rouge">Locally installed reference genome</code>
          <ul>
            <li><em>“Genome”</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-snpeff-genome-database"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` SnpEff genome database</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff genome database into your
history instead:</p>
            <ul>
              <li><em>“Genome source”</em>: <code class="language-plaintext highlighter-rouge">Downloaded snpEff database in your history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“SnpEff4.3 Genome Data”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“Produce Summary Stats”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Running the above job will produce two datasets. One is a <em>Summary Stats</em> HTML
report, which contains some interesting general metrics such as a distribution
of variants across gene features. The other one is the main annotation result -
a VCF like the input, but with annotations of variant effects added to the INFO
column.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-optional-hands-on-inspect-the-summary-stats-output-produced-by-snpeff"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Optional hands-on: Inspect the Summary Stats output produced by SnpEff</div>

  <ol>
    <li>Display the dataset:
      <ul>
        <li>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> icon next to the HTML dataset generated
by SnpEff to display its contents.</li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>One section in the report is <strong>Number of effects by type and region</strong>.
Given that you are analyzing exome data, what is the most surprising
aspect in this section? Do you have an idea how to explain it?</p>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <p>According to the report, intronic variants make up 50% of all
variants detected!</p>

          <p>Exome capture kits are designed to capture exons plus a bit of
surrounding sequence to ensure proper coverage of the exon ends
including splice junction sites.</p>

          <p>Thus, even though intronic sequences are underrepresented in exome
sequencing data, not all of them are eliminated. Since mutations
in most intron bases are neutral, they can accumulate at higher
frequency than most mutations in exons and, thus, still represent a
relevant fraction of all detected variants.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="generating-a-gemini-database-of-variants-for-further-annotation-and-efficient-variant-queries">Generating a GEMINI database of variants for further annotation and efficient variant queries</h2>

<p>Next, we are going to use the SnpEff-annotated VCF as the basis for more
exhaustive annotation with GEMINI. Unlike SnpEff, GEMINI does not just add
annotations to a list of variants in VCF format. Instead the framework
extracts the variants from the VCF input and stores them, together with newly
added annotations, in an SQL database. It then lets you formulate queries for
retrieving and reporting subsets of variants. The combined variant
extraction/annotation/storage step is performed by the <strong>GEMINI load</strong> tool. In
addition, that same tool can be used to incorporate sample pedigree info into
the database.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-creating-a-gemini-database-from-a-variants-dataset"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Creating a GEMINI database from a variants dataset</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+galaxy2" title="GEMINI load tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI load</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“VCF dataset to be loaded in the GEMINI database”</em>:
the output of <strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“The variants in this input are”</em>: <code class="language-plaintext highlighter-rouge">annotated with snpEff</code></li>
        <li>
          <p><em>“This input comes with genotype calls for its samples”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>Sample genotypes were called by Freebayes for us.</p>
        </li>
        <li><em>“Choose a gemini annotation source”</em>: select the latest available annotations snapshot (most likely, there will be only one)</li>
        <li><em>“Sample and family information in PED format”</em>: the pedigree file
prepared above</li>
        <li><em>“Load the following optional content into the database”</em>
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“GERP scores”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“CADD scores”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Gene tables”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Sample genotypes”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“variant INFO field”</em></li>
          </ul>

          <p>Leave <strong>unchecked</strong> the following:</p>
          <ul>
            <li>
              <p><em>“Genotype likelihoods (sample PLs)”</em></p>

              <p>Freebayes does not generate these values</p>
            </li>
            <li>
              <p><em>“only variants that passed all filters”</em></p>

              <p>This setting is irrelevant for our input because Freebayes did not
apply any variant filters.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Running this job generates a GEMINI-specific database dataset, which can only
be processed with other GEMINI tools. The benefit, however, is that we now have
variants, rich annotations and pedigree info stored in a format that enables
flexible and highly efficient queries, which will greatly simplify our actual
task to identify the variant responsible for the child’s disease!</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-the-gemini-suite-of-tools"><button class="gtn-boxify-button details" type="button" aria-controls="details-the-gemini-suite-of-tools" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: The GEMINI suite of tools</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>For a beginner, the sheer number of GEMINI tools may be a bit daunting and
give the impression that this framework adds a lot of complexity.
In practice, however, you will likely only need a very limited number of
these tools for any given analysis. To help you get an overview, here is a
list of the most general-purpose tools and their function:</p>

  <ul>
    <li>Every analysis will always start with <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to
build the GEMINI database from a variant list in VCF format and the
default annotations bundled with GEMINI.</li>
    <li><strong>GEMINI amend</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
let you add additional information to a GEMINI database after it has been
created. Use <strong>GEMINI amend</strong> to overwrite the pedigree info in a database
with the one found in a given PED input, and <strong>GEMINI annotate</strong> to add
custom annotations based on a VCF or BED dataset.</li>
    <li><strong>GEMINI database info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> lets you peek into the content of
a GEMINI database to explore, <em>e.g</em>, what annotations are stored in it.</li>
    <li><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>GEMINI gene-wise</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
are the two generic and most flexible tools for querying, filtering and
reporting variants in a database. There is hardly anything you cannot do
with these tools, but you need to know or learn at least some SQL-like
syntax to use them.</li>
    <li><strong>GEMINI inheritance pattern</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> simplifies variant queries
based on their inheritance pattern. While you could use the <em>query</em> and
<em>gene-wise</em> tools for the same purpose, this tool allows you to perform a
range of standard queries without typing any SQL. <strong>This tool is what we
are going to use here to identify the causative variant.</strong></li>
    <li>The remaining tools serve more specialized purposes, which are beyond the
scope of this tutorial.</li>
  </ul>

  <p>The <a href="../somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
demonstrates the use of the <strong>GEMINI annotate</strong> and <strong>GEMINI query</strong> tools,
and tries to introduce some essential bits of GEMINI’s SQL-like syntax.</p>

  <p>For a thorough explanation of all tools and functionality you should consult
the <a href="https://gemini.readthedocs.io">GEMINI documentation</a>.</p>
</blockquote>

<h2 id="candidate-variant-detection">Candidate variant detection</h2>

<p>Let us now try to identify variants that have the potential to explain the
boy child’s osteopetrosis phenotype. Remember that the parents are
consanguineous, but both of them do not suffer from the disease. This
information allows us to make some assumptions about the inheritance pattern of the causative variant.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Which inheritance patterns are in line with the phenotypic observations
for the family trio?</p>

  <p>Hint: GEMINI easily lets you search for variants fitting any of the following
inheritance patterns:</p>
  <ul>
    <li>Autosomal recessive</li>
    <li>Autosomal dominant</li>
    <li>X-linked recessive</li>
    <li>X-linked dominant</li>
    <li>Autosomal de-novo</li>
    <li>X-linked de-novo</li>
    <li>Compound heterozygous</li>
    <li>Loss of heterozygosity (LOH) events</li>
  </ul>

  <p>Think about which of these might apply to the causative variant.</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ul>
      <li>Since both parents are unaffected the variant cannot be dominant and
inherited.</li>
      <li>A de-novo acquisition of a dominant (or an X-linked recessive) mutation
is, of course, possible.</li>
      <li>A recessive variant is a possibility, and a more likely one given
the parents’ consanguinity.</li>
      <li>For both the de-novo and the inherited recessive case, the variant could
reside on an autosome or on the X chromosome. Given that we provided you
with only the subset of sequencing reads mapping to chr8, an X-linked
variant would not be exactly instructive though ;)</li>
      <li>A compound heterozygous combination of variant alleles affecting the
same gene is possible, but less likely given the consanguinity of the
parents (as this would require two deleterious variant alleles in the
gene circulating in the same family).</li>
      <li>A loss of heterozygosity (LOH) turning a heterozygous recessive variant
into a homozygous one could be caused by uniparental disomy or by an LOH
event early in embryonic development, but both these possibilities have
an exceedingly small probability.</li>
    </ul>

    <p>Based on these considerations it makes sense to start looking for
inherited autosomal recessive variants first. Then, if there is no
convincing candidate mutation among them, you could extend the search to
de-novo variants, compund heterozygous variant pairs and LOH events -
probably in that order.</p>
  </blockquote>
</blockquote>

<p>Since our GEMINI database holds the variant and genotype calls for the
family trio and the relationship between the family members, we can make use
of <strong>GEMINI inheritance pattern</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to report all variants
fitting any specific inheritance model with ease.</p>

<p>Below is how you can perform the query for inherited autosomal recessive
variants. Feel free to run analogous queries for other types of variants that
you think could plausibly be causative for the child’s disease.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-finding-and-reporting-plausible-causative-variants"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Finding and reporting plausible causative variants</div>
  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_inheritance/gemini_inheritance/0.20.1" title="GEMINI inheritance pattern tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI inheritance pattern</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1)</span>
      <ul>
        <li><em>“GEMINI database”</em>: the GEMINI database of annotated variants; output
of <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Your assumption about the inheritance pattern of the phenotype of interest”</em>:
<code class="language-plaintext highlighter-rouge">Autosomal recessive</code>
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Additional constraints on variants”</em>
              <ul>
                <li>
                  <p><em>“Additional constraints expressed in SQL syntax”</em>:
<code class="language-plaintext highlighter-rouge">impact_severity != 'LOW'</code></p>

                  <p>This is a simple way to prioritize variants based on their
functional genomic impact. Variants with <em>low impact severity</em> would
be those with no obvious impact on protein function (<em>i.e.</em>, silent
mutations and variants outside coding regions)</p>
                </li>
              </ul>
            </li>
            <li>
              <p><em>“Include hits with less convincing inheritance patterns”</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>This option is only meaningful with larger family trees to account
for errors in phenotype assessment.</p>
            </li>
            <li>
              <p><em>“Report candidates shared by unaffected samples”</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>This option is only meaningful with larger family trees to account
for alleles with partial phenotypic penetrance.</p>
            </li>
          </ul>
        </li>
        <li>
          <p><em>“Family-wise criteria for variant selection”</em>: keep default settings</p>

          <p>This section is not useful when you have data from just one family.</p>
        </li>
        <li>In <em>“Output - included information”</em>
          <ul>
            <li><em>“Set of columns to include in the variant report table”</em>:
<code class="language-plaintext highlighter-rouge">Custom (report user-specified columns)</code>
              <ul>
                <li><em>“Choose columns to include in the report”</em>:
                  <ul>
                    <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“alternative allele frequency (max_aaf_all)”</em></li>
                  </ul>
                </li>
                <li><em>“Additional columns (comma-separated)”</em>:
<code class="language-plaintext highlighter-rouge">chrom, start, ref, alt, impact, gene, clinvar_sig,
clinvar_disease_name, clinvar_gene_phenotype, rs_ids</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-variant-i-vs-i-gene-centric-clinvar-information"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Variant- <i>vs</i> gene-centric ClinVar information</div>
        <p>Most annotation data offered by GEMINI is variant-centric, <em>i.e.</em>,
annotations reported for a variant are specific to this exact variant.
A few annotation sources, however, also provide gene-centric
information, which applies to the gene affected by a variant, not
necessarily the variant itself.</p>

        <p>In the case of <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>, the
annotation fields/columns <code class="language-plaintext highlighter-rouge">clinvar_sig</code> and <code class="language-plaintext highlighter-rouge">clinvar_disease_name</code> refer
to the particular variant, but <code class="language-plaintext highlighter-rouge">clinvar_gene_phenotype</code> provides
information about the affected gene.</p>

        <p>Including the gene phenotype in the report can be crucial because a
gene may be well known to be disease-relevant, while a particular
variant may not have been clinically observed or been reported before.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>From the GEMINI reports you generated, can you identify the most likely
candidate variant responsible for the child’s disease?</p>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-gemini-usage-examples"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-gemini-usage-examples" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: More GEMINI usage examples</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>While only demonstrating command line use of GEMINI, the following tutorial
slides may give you additional ideas for variant queries and filters:</p>

  <ul>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Intro-To-Gemini.pdf">Introduction to GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-DeNovo-Tutorial.pdf">Identifying <em>de novo</em> mutations with GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Recessive-Tutorial.pdf">Identifying recessive gene candidates with GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Dominant-Tutorial.pdf">Identifying dominant gene candidates with GEMINI</a></li>
  </ul>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>It was not hard to find the most likely causative mutation for the child’s
disease (you did find it, right?).</p>

<p>Even though it will not always provide as strong support for just one specific
causative variant, analysis of whole-exome sequencing data of family trios (or
other related samples) can often narrow down the search for the cause of a
genetic disease to just a very small, manageable set of candidate variants, the
relevance of which can then be addressed through standard methods.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Exome sequencing is an efficient way to identify disease-relevant genetic variants.</p>
</li>
                        
                        <li><p>Freebayes is a good variant and genotype caller for the joint analysis of multiple samples. It is straightforward to use and requires only minimal processing of mapped reads.</p>
</li>
                        
                        <li><p>Variant annotation and being able to exploit genotype information across family members is key to identifying candidate disease variants. SnpEff and GEMINI, in particular, are powerful tools offered by Galaxy for that purpose.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/variant-analysis/faqs/">FAQ page for the Variant Analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                


                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=variant-analysis/exome-seq"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=variant-analysis/exome-seq">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wolfgang Maier, Bérénice Batut, Torsten Houwaart, Anika Erxleben, Björn Grüning,  <b>Exome sequencing data analysis for diagnosing a genetic disease (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{variant-analysis-exome-seq,
author = "Wolfgang Maier and Bérénice Batut and Torsten Houwaart and Anika Erxleben and Björn Grüning",
	title = "Exome sequencing data analysis for diagnosing a genetic disease (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/variant-analysis/tutorials/exome-seq/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: bwa
  owner: devteam
  revisions: e188dc7a68e6
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: 5ec9f6bceaee
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: freebayes
  owner: devteam
  revisions: a5937157062f
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_rmdup
  owner: devteam
  revisions: 586f9e1cdb2b
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bcftools_norm
  owner: iuc
  revisions: aaac3ed89508
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gemini_inheritance
  owner: iuc
  revisions: 2c68e29c3527
  tool_panel_section_label: Gemini
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gemini_load
  owner: iuc
  revisions: 2270a8b83c12
  tool_panel_section_label: Gemini
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: abfd8a6544d7
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_view
  owner: iuc
  revisions: 5826298f6a73
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpeff
  owner: iuc
  revisions: 6322be79bd8e
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpeff
  owner: iuc
  revisions: 6322be79bd8e
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trimmomatic
  owner: pjbriggs
  revisions: d94aff5ee623
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">22</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.13636363636363635">3</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.18181818181818182">4</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">1 stars</span><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.13636363636363635">3</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>October 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Description of the case make the usage of the training material ideal for understanding the aim of the analysis and the results. I like that you also reference other protocols for the one who like to have more information.
					<b>Disliked</b>: In general the tutorial runs extremely well until the variant annotation and classification. I think the issue might be related to the versioning and using HG19.  The issue starts with snpeff in which no genome source exists in the latest release and in which uploading the hg19 version of it is making issues due to version 5.2 grnome database. We were not able to continue with Gemini when using SnpEff eff version 4.3+T.galaxy2 and the integrated HG19 genome, only worked with 5.2+galaxy0 and download on demand the genome hg19.   Doing so, we manage to run toolshed.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+galaxy2 and toolshed.g2.bx.psu.edu/repos/iuc/gemini_inheritance/gemini_inheritance/0.20.1 with the anticipated output.
				</li>
				
		    </ul>
		    
		    <b>July 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The whole tutorial
					
				</li>
				
		    </ul>
		    
		    <b>June 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: everything
					<b>Disliked</b>: yes
				</li>
				
		    </ul>
		    
		    <b>May 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: clear steps and links that take you straight to the tool
					
				</li>
				
		    </ul>
		    
		    <b>March 2024</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Instructions were very clear
					
				</li>
				
		    </ul>
		    
		    <b>May 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					
					<b>Disliked</b>: Gemini is no longer used
				</li>
				
		    </ul>
		    
		    <b>April 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Most part of the instructions were very clear and detailed.
					<b>Disliked</b>: There are still some part of the tutorial which can be a little more clear, for example at "Hands-on: Generating FreeBayes calls" I was confused and took a while to figure out.
				</li>
				
		    </ul>
		    
		    <b>March 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					
					<b>Disliked</b>: You should specify that you can't feed GEMINI tool more recent versions of hg19 at the outset. I used a more recent version hg38 throughout my analysis, only to get to the last step and realize that I had to start over using hg19.   Also, GEMINI is not available on usegalaxy.org, only on galaxy.eu, and I am having a lot of difficulty importing my datasets over since the tags for my PED file disappear on transfer.   Extremely frustrating end result for what I originally thought was a great tutorial! 
				</li>
				
		    </ul>
		    
		    <b>November 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: detailed explanation of variant calling and genotype calling
					
				</li>
				
				<li>
					<span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: This tutorial gives a comprehensive overview about all analysis steps and important parameters involved.
					<b>Disliked</b>: Where is the actual hands-on command code, which is generated after all parameter boxes have been checked, and the 'submit button' is clicked? Where are the actual tool commands for each step, a user would need to perform these analysis steps on the command line? I know these can be displayed in Galaxy, but it would be helpful to show them in the tutorials as well. 
				</li>
				
		    </ul>
		    
		    <b>March 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The step-by-step approach and the additional explanations for some input fields
					<b>Disliked</b>: Perhaps an additional indication of which patch of the hg19 is recognised as the database of the SNPeff eff output of GEMINI load. It took me several attempts to realise that it must be the unpatched version.
				</li>
				
		    </ul>
		    
		    <b>November 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">3 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Nice features and understandable handling for beginners
					<b>Disliked</b>: Some data and tools are not at the described places; specifically genome annotation fasta and snpTools... snpEff is not existing, so I used snpEff eff; snpSift Annotate exists twice with different features. Description on how to create the pedigree file.
				</li>
				
		    </ul>
		    
		    <b>June 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Oustanding material!
					<b>Disliked</b>: Nothing!
				</li>
				
		    </ul>
		    
		    <b>May 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: all of it
					<b>Disliked</b>: nothing - it's perfect
				</li>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Gareth, Melissa and Khalid (I was in breakout room 3) were all incredibly helpful and it went at a good pace. It was VERY helpful getting the course material ahead of the workshop, which allowed time to familiarise myself with the content. 
					<b>Disliked</b>: I think it was on the mark for me. 
				</li>
				
		    </ul>
		    
		    <b>March 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Going to use this in Australia soon (late May 2021)- excited about it. More feedback to come afterwards no doubt!
					<b>Disliked</b>: I can see in the intro that this tutorial is set up for raw (fastq) and BAM data entry points. This is great but I was hoping you could include a workflow for fastq to BAM. I realise it is in the full workflow so I took the liberty of comparing the two offered workflows and trimmed down to a QC to BAM workflow (https://usegalaxy.org.au/u/gareth-qfab/w/copy-of-exome-seq-training-full-w-cached-ref-imported-from-uploaded-file ). I understand we can do this elsewhere but good to have it consistent within this tutorial. Tested on Zenodo data and works fine. Thanks, Gareth
				</li>
				
		    </ul>
		    
		    <b>February 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Concentration on the essential aspects with the possibility to further deepen the knowledge about the individual tools
					
				</li>
				
		    </ul>
		    
		    <b>June 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: I liked that it was very specific and went step by step on how to do everything. nothing was left out.
					<b>Disliked</b>: I would like to know how to take the vcf file and identify polymorphic sites in all three individuals. 
				</li>
				
		    </ul>
		    
		    <b>March 2020</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Clear, stepwise
					<b>Disliked</b>: Gemini load dialog in usegalaxy.eu suggested the SnpEff eff output file was 'unavailable' to use. It made me go back and check things before pressing on regardless. Not a problem with the tutorial per se. Thanks
				</li>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Very accessible, especially if you compare the galaxy ui to something like bash.
					<b>Disliked</b>: -
				</li>
				
		    </ul>
		    
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00311">gxy.io/GTN:T00311</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/exome-seq/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/variant-analysis/tutorials/exome-seq/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/be6a5055148f12d87e2a00fb270ceb96a709582c">be6a505</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>