<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Exome sequencing data analysis for diagnosing a genetic disease</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:title" content="Galaxy Training: Exome sequencing data analysis for diagnosing a genetic d..." />
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Variant Analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fexome-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fexome-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fexome-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fexome-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fexome-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fexome-seq%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/variant-analysis/tutorials/exome-seq/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Exome sequencing data analysis for diagnosing a genetic disease",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
  },
  "courseCode": "variant-analysis / exome-seq / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Exome sequencing data analysis for diagnosing a genetic disease' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html",
  "timeRequired": "PT5H",
  "description": "The questions this  addresses are:\n - How do you identify genetic variants in samples based on exome sequencing data?\n - How do you, among the set of detected variants, identify candidate causative variants for a given phenotype/disease?\n\n\\nThe objectives are:\n - Jointly call variants and genotypes for a family trio from whole-exome sequencing data\n - Use variant annotation and the observed inheritance pattern of a phenotype to identify candidate causative variants and to prioritize them\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Torsten Houwaart"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "Bj√∂rn Gr√ºning"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    },
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Torsten Houwaart"
    },
    {
      "@type": "Person",
      "name": "Anika Erxleben"
    },
    {
      "@type": "Person",
      "name": "Bj√∂rn Gr√ºning"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Exome sequencing data analysis for diagnosing a genetic disease</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/wm75/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/wm75" alt="Wolfgang Maier">Wolfgang Maier</a>, <a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="B√©r√©nice Batut">B√©r√©nice Batut</a>, <a href="/training-material/hall-of-fame/torhou/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/torhou" alt="Torsten Houwaart">Torsten Houwaart</a>, <a href="/training-material/hall-of-fame/erxleben/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/erxleben" alt="Anika Erxleben">Anika Erxleben</a>, <a href="/training-material/hall-of-fame/bgruening/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bgruening" alt="Bj√∂rn Gr√ºning">Bj√∂rn Gr√ºning</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How do you identify genetic variants in samples based on exome sequencing data?</p>
</li>
            
            <li><p>How do you, among the set of detected variants, identify candidate causative variants for a given phenotype/disease?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Jointly call variants and genotypes for a family trio from whole-exome sequencing data</p>
</li>
            
            <li><p>Use variant annotation and the observed inheritance pattern of a phenotype to identify candidate causative variants and to prioritize them</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 5 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/variant-analysis/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.3054169">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/exome-seq/workflows/" title="Workflows" alt="Exome sequencing data analysis for diagnosing a genetic disease workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 id="introduction">Introduction</h1>

<p>Exome sequencing is a method that enables the selective sequencing of the
exonic regions of a genome - that is the transcribed parts of the genome present
in mature m<span class="notranslate">RNA</span>, including protein-coding sequences, but also untranslated
regions (UTRs).</p>

<p>In humans, there are about 180,000 exons with a combined length of ~ 30 million
base pairs (30 Mb). Thus, the exome represents only 1% of the human genome, but
has been estimated to harbor up to 85% of all disease-causing variants (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2768590/">Choi
et al., 2009</a>).</p>

<p>Exome sequencing, thus, offers an affordable alte<span class="notranslate">rna</span>tive to whole-genome
sequencing in the diagnosis of genetic disease, while still covering far more
potential disease-causing variant sites than genotyping arrays. This is of
special relevance in the case of rare genetic diseases, for which the causative
variants may occur at too low a frequency in the human population to be
included on genotyping arrays.</p>

<p>Of note, a recent study focusing on the area of clinical pediatric neurology
indicates that the costs of exome sequencing may actually not be higher even
today than the costs of conventional genetic testing (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5589982/">Vissers et al., 2017</a>).</p>

<blockquote class="details">
  <h3 id="details-exome-sequencing-vs-whole-genome-sequencing"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Exome sequencing <em>vs</em> whole-genome sequencing</h3>

  <p>In principle, the steps illustrated in this tutorial are suitable also for
the analysis of whole-genome sequencing (WGS) data. At comparable mean
coverage, however, WGS datasets will be much larger than exome sequencing
ones and their analysis will take correspondingly more time.</p>

  <p>The obvious benefit of WGS compared to exome-sequencing, of course, is that
it will allow variant detection in even more regions of the genome.
As a less apparent advantage, the more complete information of WGS data can
make it easier to detect <em>copy number variation (CNV)</em> and
<em>structural variants</em> such as translocations and inversions (although such
detection will require more sophisticated analysis steps, which are not
covered by this tutorial).</p>

  <p>Very generally, one could argue that exome-sequencing captures most of the
information that can be analyzed with standard bioinformatical tools today at
reasonable costs. WGS, on the other hand, captures as much information as
today‚Äôs sequencing technology can provide, and it may be possible to
reanalyze such data with more powerful bioinformatical software in the
future to exploit aspects of the information that were not amenable to
analysis at the time of data acquisition.</p>
</blockquote>

<p>The identification of causative variants underlying any particular genetic
disease is, as we will see in this tutorial, not just dependent on the
successful detection of variants in the genome of the patient, but also on
variant comparison between the patient and selected relatives. Most often
family trio data, consisting of the genome sequences of the patient and their
parents, is used for this purpose. With multisample data like this it becomes
possible to search for variants following any kind of Mendelian inheritance
scheme compatible with the observed inheritance pattern of the disease, or to
detect possibly causative <em>de-novo</em> mutations or <em>loss-of-heterozygosity</em> (LOH) events.</p>

<blockquote class="details">
  <h3 id="details-related-tutorials"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Related tutorials</h3>

  <p>This tutorial focuses on the practical aspects of analyzing real-world
patient data. If you are more interested in the theoretical aspects of
variant calling, you may want to have a look at the related tutorial on
<a href="../dip/tutorial.html">Calling variants in diploid systems</a>.</p>

  <p>The tutorial on <a href="../somatic-variants/tutorial.html">Somatic variant calling</a>
follows an analysis <span class="notranslate">workflow</span> that is rather similar to the one here, but
tries to identify tumor variants by comparing a tumor sample to healthy
tissue from the same patient. Together, the two tutorials are intended to get
you started with genomics medicine using <span class="notranslate">Galaxy</span>.</p>

</blockquote>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a></li>
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
  <li><a href="#read-mapping" id="markdown-toc-read-mapping">Read <span class="notranslate">mapping</span></a></li>
  <li><a href="#mapped-reads-postprocessing" id="markdown-toc-mapped-reads-postprocessing">Mapped <span class="notranslate">reads</span> postprocessing</a>    <ol>
      <li><a href="#filtering-on-mapped-reads-properties" id="markdown-toc-filtering-on-mapped-reads-properties">Filtering on mapped <span class="notranslate">reads</span> properties</a></li>
      <li><a href="#removing-duplicate-reads" id="markdown-toc-removing-duplicate-reads">Removing duplicate <span class="notranslate">reads</span></a></li>
    </ol>
  </li>
  <li><a href="#variant-calling" id="markdown-toc-variant-calling">Variant calling</a>    <ol>
      <li><a href="#generating-freebayes-calls" id="markdown-toc-generating-freebayes-calls">Generating FreeBayes calls</a></li>
      <li><a href="#post-processing-freebayes-calls" id="markdown-toc-post-processing-freebayes-calls">Post-processing FreeBayes calls</a></li>
    </ol>
  </li>
  <li><a href="#variant-annotation-and-reporting" id="markdown-toc-variant-annotation-and-reporting">Variant annotation and reporting</a>    <ol>
      <li><a href="#get-data-1" id="markdown-toc-get-data-1">Get data</a></li>
      <li><a href="#variant-annotation-with-functional-genomic-effects" id="markdown-toc-variant-annotation-with-functional-genomic-effects">Variant annotation with functional genomic effects</a></li>
      <li><a href="#generating-a-gemini-database-of-variants-for-further-annotation-and-efficient-variant-queries" id="markdown-toc-generating-a-gemini-database-of-variants-for-further-annotation-and-efficient-variant-queries">Generating a GEMINI database of variants for further annotation and efficient variant queries</a></li>
      <li><a href="#candidate-variant-detection" id="markdown-toc-candidate-variant-detection">Candidate variant detection</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>In this tutorial, we are going to analyze exome sequencing data from a family
trio, in which the boy child is affected by the disease
<a href="https://ghr.nlm.nih.gov/condition/osteopetrosis/">osteopetrosis</a>, while both
parents, who happen to be consanguineous, are unaffected. Our goal is to
identify the genetic variation that is responsible for the disease.</p>

<h2 id="get-data">Get data</h2>

<p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> This tutorial offers <strong>two alte<span class="notranslate">rna</span>tive entry points</strong> allowing
you to</p>

<ul>
  <li>conduct a full analysis starting from original sequenced <span class="notranslate">reads</span> in <code class="language-plaintext highlighter-rouge">fastq</code>
format or to</li>
  <li>start your analysis with premapped <span class="notranslate">reads</span> in <code class="language-plaintext highlighter-rouge">bam</code> format that are (almost)
ready for variant calling.</li>
</ul>

<p>While the full analysis is probably closer to how you would analyze your own
data, the shortened analysis from premapped <span class="notranslate">reads</span> may suit your time frame
better, and will avoid redundancy if you have previously worked through other
tutorials demonstrating NGS quality control and read <span class="notranslate">mapping</span>, like the
dedicated
<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality control</a>
and <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><span class="notranslate">Mapping</span></a>
tutorials.</p>

<p>The following hands-on section will guide you through obtaining the right
data for either analysis.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Obtain the raw sequencing data</p>

      <blockquote class="comment">
        <h3 id="comment-starting-from-raw-sequencing-data"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Starting from raw sequencing data</h3>
        <p>In this and the following steps you will obtain the original unmapped
sequencing data and prepare for a full analysis including the <span class="notranslate">mapping</span> of
the sequenced <span class="notranslate">reads</span>.</p>

        <p>If you prefer <strong>to skip the <span class="notranslate">mapping</span> step</strong> and start the analysis from
premapped data <strong>you should proceed directly to step 4</strong> of this
section.</p>
      </blockquote>

      <p>Import the original sequenced <span class="notranslate">reads</span> datasets of the family trio from
<a href="https://zenodo.org/record/3054169">Zenodo</a>:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3243160/files/father_R1.fq.gz
https://zenodo.org/record/3243160/files/father_R2.fq.gz
https://zenodo.org/record/3243160/files/mother_R1.fq.gz
https://zenodo.org/record/3243160/files/mother_R2.fq.gz
https://zenodo.org/record/3243160/files/proband_R1.fq.gz
https://zenodo.org/record/3243160/files/proband_R2.fq.gz
</code></pre></div>      </div>

      <p>Alte<span class="notranslate">rna</span>tively, the same files may be available on your <span class="notranslate">Galaxy</span> server
through a shared data library (your instructor may tell you so), in
which case you may prefer to import the data directly from there.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from ‚ÄúAuto-detect‚Äù to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Check that the newly created datasets in your history have their
datatypes assigned correctly to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>, and fix any missing or
wrong datatype assignment</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>

      <p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Congratulations for obtaining the datasets required for
an analysis including <span class="notranslate">reads</span> <span class="notranslate">mapping</span>. You should now
<strong>proceed with Step 7</strong> below.</p>
    </li>
    <li>Obtain the premapped sequencing data
      <blockquote class="comment">
        <h3 id="comment-alternative-entry-point-premapped-data"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Alte<span class="notranslate">rna</span>tive entry point: Premapped data</h3>
        <p>Skip this and the following two steps if you already obtained and
prepared the original unmapped seuencing data and are planning to
perform the <span class="notranslate">mapping</span> step yourself.</p>
      </blockquote>

      <p>Import the premapped <span class="notranslate">reads</span> datasets of the family trio from
<a href="https://zenodo.org/record/3054169">Zenodo</a>:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3243160/files/mapped_<span class="notranslate">reads</span>_father.bam
https://zenodo.org/record/3243160/files/mapped_<span class="notranslate">reads</span>_mother.bam
https://zenodo.org/record/3243160/files/mapped_<span class="notranslate">reads</span>_proband.bam
</code></pre></div>      </div>

      <p>Alte<span class="notranslate">rna</span>tively, the same files may be available on your <span class="notranslate">Galaxy</span> server
through a shared data library (your instructor may tell you so), in
which case you may prefer to import the data directly from there.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links-1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from ‚ÄúAuto-detect‚Äù to <code class="language-plaintext highlighter-rouge">bam</code></p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library-1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Check that the newly created datasets in your history have their
datatypes assigned correctly to <code class="language-plaintext highlighter-rouge">bam</code>, and fix any missing or wrong
datatype assignment</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype-1"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge">bam</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Specify the genome version that was used for <span class="notranslate">mapping</span></p>

      <p>Change the database/build (dbkey) for each of your bam datasets
to <code class="language-plaintext highlighter-rouge">hg19</code>.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing Database/Build (dbkey)</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Database/Build</strong> field</li>
          <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code></li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>

      <blockquote class="details">
        <h3 id="details-why-specify-genome-versions"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Why specify genome versions</h3>
        <p>When you are starting with sequencing data that has already been mapped
to a particular genome version (human hg19 in this case), it is good
practice to attach this information as metadata to the datasets.</p>

        <p>Doing so helps prevent accidental use of a different version of the
reference genome in later steps like variant calling, which would
typically lead to nonsensical results because of base position changes
between the different versions.</p>
      </blockquote>

      <p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Congratulations for obtaining the premapped sequencing
datasets. Now, <strong>follow the remaining steps</strong> to set everything up for a
successful analysis.</p>
    </li>
    <li>
      <p>Rename the datasets</p>

      <p>For datasets that you upload via a link, <span class="notranslate">Galaxy</span> will pick the link
address as the dataset name, which you will likely want to shorten to
just the file names.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Add #father/#mother/#child tags to the datasets</p>

      <p>Parts of the analysis in this tutorial will consist of identical steps
performed on the data of each family member.</p>

      <p>To make it easier to keep track of which dataset represents which step in
the analysis of which sample, <span class="notranslate">Galaxy</span> supports dataset tags. In particular,
if you attach a tag starting with <code class="language-plaintext highlighter-rouge">#</code> to any dataset, that tag will
automatically propagate to any new dataset derived from the tagged
dataset.</p>

      <p>Tags are supposed to help you identify the origin of datasets quickly,
but you can choose them as you like.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Adding a tag</h3>
        <ul>
          <li>Click on the dataset</li>
          <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>
          <li>
            <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>

            <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>
          </li>
          <li>Check that the tag is appearing below the dataset name</li>
        </ul>

      </blockquote>
    </li>
    <li>
      <p>Obtain the reference genome</p>

      <blockquote class="comment">
        <h3 id="comment-shortcut"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Shortcut</h3>
        <p>You can skip this step if the <span class="notranslate">Galaxy</span> server you are working on offers
a <code class="language-plaintext highlighter-rouge">hg19</code> version of the human reference genome with prebuilt indexes for
<em>bwa-mem</em> (only necessary if starting from unmapped original sequencing
<span class="notranslate">reads</span>) and <em>freebayes</em>. Ask your instructor, or check the tools
<strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
if they list a <code class="language-plaintext highlighter-rouge">hg19</code> version as an option under
<em>‚ÄúSelect a / Using reference genome‚Äù</em>).</p>
      </blockquote>

      <p>Import the <code class="language-plaintext highlighter-rouge">hg19</code> version of the human chromosome 8 sequence:</p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3243160/files/hg19_chr8.fa.gz
</code></pre></div>      </div>

      <p>Make sure you specify the datatype as <code class="language-plaintext highlighter-rouge">fasta</code> in the import dialog.</p>

      <p>Alte<span class="notranslate">rna</span>tively, load the dataset from a shared data library.</p>
    </li>
    <li>
      <p>Rename the reference genome</p>

      <p>The reference genome you have imported above came as a compressed
file, but got unpacked by <span class="notranslate">Galaxy</span> to plain <code class="language-plaintext highlighter-rouge">fasta</code> format according to
your datatype selection. You may now wish to remove the <code class="language-plaintext highlighter-rouge">.gz</code> suffix
from the dataset name.</p>
    </li>
  </ol>

</blockquote>

<p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> <strong>Congratulations!</strong>
You are all set for starting the analysis now.</p>

<p>If you have chosen to follow the complete analysis from the original sequenced
data, just proceed with the next section.</p>

<p>If, on the other hand, you have prepared to start from the premapped data, skip
the sections on <em>Quality control</em> and <em>Read <span class="notranslate">mapping</span></em>, and conitnue with
<strong>Mapped <span class="notranslate">reads</span> postprocessing</strong>.</p>

<h1 id="quality-control">Quality control</h1>

<p>This step serves the purpose of identifying possible issues with the raw
sequenced <span class="notranslate">reads</span> input data before embarking on any ‚Äúreal‚Äù analysis steps.</p>

<p>Some of the typical problems with NGS data can be mitigated by preprocessing
affected sequencing <span class="notranslate">reads</span> before trying to map them to the reference genome.
Detecting some other, more severe problems early on may at least save you a lot
of time spent on analyzing low-quality data that is not worth the effort.</p>

<p>Here, we will perform a standard quality check on our input data and only point
out a few interesting aspects about that data. For a more thorough explanation
of NGS data quality control, you may want to have a look at the dedicated
tutorial on <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality control</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control-of-the-input-datasets"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control of the input datasets</h3>
  <ol>
    <li>Run <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on each of your six fastq datasets
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúShort read data from your current history‚Äù</em>: all 6 FASTQ  datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select multiple datasets</h3>

        <ol>
          <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
          <li>Select several files by keeping the <kbd>Ctrl</kbd> (or
<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
        </ol>
      </blockquote>

      <p>When you start this job, twelve new datasets (one with the calculated raw
data, another one with an html report of the findings for each input
dataset) will get added to your history.</p>
    </li>
    <li>Use <strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to aggregate the raw <strong>FastQC</strong> data of all input datasets into one report
      <ul>
        <li>In <em>‚ÄúResults‚Äù</em>
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>‚ÄúFastQC output‚Äù</em>
              <ul>
                <li><em>‚ÄúType of FastQC output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúFastQC output‚Äù</em>: all six <em>RawData</em>
outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the <em>Webpage</em> output produced by the tool</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>Based on the report, do you think preprocessing of the <span class="notranslate">reads</span>
(trimming and/or filtering) will be necessary before <span class="notranslate">mapping</span>?</li>
          <li>Why do all samples show a non-normal GC content distribution, and
should you be worried?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>Sequence quality is quite good overall. If anything you might
consider trimming the 3‚Äô ends of <span class="notranslate">reads</span> (base qualities decline
slightly towards the 3‚Äô ends) or to filter out the small fraction
of <span class="notranslate">reads</span> with a mean base quality &lt; 5.
Feel free to run, <em>e.g.</em>, <strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the
fastq datasets if you want to, but don‚Äôt expect this to have a big
effect on the analysis given the high overall quality of the data
of all samples.</li>
            <li>
              <p>In whole-genome sequencing, a non-normal distribution of the GC
content of the <span class="notranslate">reads</span> from a sample is typically considered to hint
at possible contamination.
Here, however, we are dealing with sequencing data from captured
exomes, <em>i.e</em>, the <span class="notranslate">reads</span> are not representing random sequences from
a genome, but rather a biased selection.</p>

              <p>A bimodal GC content distribution, like for the samples at hand, is
a characteristic feature of many exome capture methods and has also
been observed with an Illumina Nextera Rapid Capture exome kit
before (compare Fig. 3D in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4522667/">Shigemizu et al., 2015</a>
).</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="read-mapping">Read <span class="notranslate">mapping</span></h1>

<p>Now that you confirmed that the quality of the input data is good enough to
warrant further analysis, it is time to map the sequenced <span class="notranslate">reads</span> to the
reference genome.</p>

<p>We assume here that you are at least vaguely familiar with the concept of read
<span class="notranslate">mapping</span> and only illustrate the concrete steps necessary  to map our specific
NGS <span class="notranslate">reads</span> datasets to the human reference genome. We recommend you to follow
the dedicated <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><span class="notranslate">Mapping</span> tutorial</a>, if you need a general introduction to read <span class="notranslate">mapping</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-mapping"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read <span class="notranslate">Mapping</span></h3>
  <ol>
    <li><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to map the <span class="notranslate">reads</span> from the <strong>father</strong> sample to the reference genome
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-sequence"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> chr8 sequence as a fasta dataset into
your history instead:</p>
            <ul>
              <li><em>‚ÄúWill you select a reference genome from your history or use a
built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect first set of <span class="notranslate">reads</span>‚Äù</em>: the
forward <span class="notranslate">reads</span> (R1) dataset of the <strong>father</strong> sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect second set of <span class="notranslate">reads</span>‚Äù</em>: the
reverse <span class="notranslate">reads</span> (R2) dataset of the <strong>father</strong> sample</li>
          </ul>

          <blockquote class="tip">
            <h3 id="tip-no-fastq-datasets-selectable"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> No FASTQ datasets selectable?</h3>
            <p>Please confirm that the problematic datasets declare <em>format</em>:
<code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>.</p>

            <p>Most <span class="notranslate">Galaxy</span> tools that accept FASTQ input expect the data to be
formatted as <em>FASTQ with Sanger-scaled quality values</em>, the most
widely spread version of the FASTQ format. To make this requirement
explicit (instead of generating possibly wrong results) these tools
require you to set the dataset type to <code class="language-plaintext highlighter-rouge">fastqsanger</code> (<code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>
for data compressed with gzip). You can do so either on data upload
or later from the <em>Edit dataset attributes</em> view (which you can reach
by clicking on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> pencil icon.</p>
          </blockquote>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group identifier (ID)‚Äù</em>: <code class="language-plaintext highlighter-rouge">000</code></li>
              </ul>
            </li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group sample name (SM)‚Äù</em>: <code class="language-plaintext highlighter-rouge">father</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="warning-read-group-ids-and-sample-names---choose-but-choose-wisely"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Read group IDs and sample names - choose, but choose wisely</h3>
        <p>In general, you are free to choose ID and SM values to your liking, but
‚Ä¶</p>

        <p>The <strong>ID</strong> should <strong>unambiguously identify</strong> the sequencing run that
produced the <span class="notranslate">reads</span>. At the very least, no two input datasets in any
given analysis should define the same ID twice, or tools like
<em>FreeBayes</em>, which we are going to use in the next step, will refuse
to work with the data.</p>

        <p>The <strong>SM</strong> value, on the other hand, should identify the biological
sample represented by the data and is used by many tools (like <em>GEMINI</em>
which we will use later) to let you refer to one specifc sample in a
multisample analysis. Choose descriptive, but short and easy to
remember sample names since you will have to type them in again!</p>

      </blockquote>
    </li>
    <li>
      <p><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to map the <span class="notranslate">reads</span> from the <strong>mother</strong> sample to the reference genome <strong>using the same parameters as before</strong> except</p>

      <ul>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect first set of <span class="notranslate">reads</span>‚Äù</em>: the
forward <span class="notranslate">reads</span> (R1) dataset of the <strong>mother</strong> sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect second set of <span class="notranslate">reads</span>‚Äù</em>: the
reverse <span class="notranslate">reads</span> (R2) dataset of the <strong>mother</strong> sample</li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group identifier (ID)‚Äù</em>: <code class="language-plaintext highlighter-rouge">001</code></li>
              </ul>
            </li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group sample name (SM)‚Äù</em>: <code class="language-plaintext highlighter-rouge">mother</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to map the <span class="notranslate">reads</span> from the <strong>child</strong> sample to the reference genome <strong>using the same parameters as before</strong> except</p>

      <ul>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect first set of <span class="notranslate">reads</span>‚Äù</em>: the
forward <span class="notranslate">reads</span> (R1) dataset of the <strong>child</strong> sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect second set of <span class="notranslate">reads</span>‚Äù</em>: the
reverse <span class="notranslate">reads</span> (R2) dataset of the <strong>child</strong> sample</li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group identifier (ID)‚Äù</em>: <code class="language-plaintext highlighter-rouge">002</code></li>
              </ul>
            </li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group sample name (SM)‚Äù</em>: <code class="language-plaintext highlighter-rouge">proband</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="mapped-reads-postprocessing">Mapped <span class="notranslate">reads</span> postprocessing</h1>

<p>At this point in the analysis you should have obtained three mapped <span class="notranslate">reads</span>
datasets in <code class="language-plaintext highlighter-rouge">bam</code> format. Ideally, these would carry <code class="language-plaintext highlighter-rouge">#father</code>, <code class="language-plaintext highlighter-rouge">#mother</code> and
<code class="language-plaintext highlighter-rouge">#child</code> tags for quick identification of the samples they provide data for.</p>

<p>In principle, you could use these datasets directly for variant calling, and in
many cases, including this one, this would be sufficient to identify the
sought-after variants.</p>

<p>To obtain an accurate picture of the variant spectrum found in your samples it
is good practice though to perform various postprocessing steps on the mapped
<span class="notranslate">reads</span> before passing them to a variant caller.</p>

<blockquote class="comment">
  <h3 id="comment-shortcut-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Shortcut</h3>
  <p>If you are in a hurry, you may skip the postprocessing steps and continue
with your mapped <span class="notranslate">reads</span> datasets and the <em>Variant Calling</em> section. Make sure
though you come back here, or read about mapped <span class="notranslate">reads</span> postprocessing
elsewhere, before attempting to perform any variant analysis on your own data
that may require high accuracy/sensitivity.</p>
</blockquote>

<p>The optimal set of postprocessing steps required depends on the variant calling
software used at the next step. The <strong>FreeBayes</strong> variant caller that we are
going to use in this tutorial is particularly well suited for use with minimal
mapped <span class="notranslate">reads</span> postprocessing pipelines, so all we are going to do here is:</p>

<ul>
  <li>
    <p>filter the paired-end <span class="notranslate">reads</span> of all samples to retain only those read pairs,
for which both the forward and the reverse read have been mapped to the
reference successfully</p>

    <p>For such pairs of <span class="notranslate">reads</span>, we can be extra confident that they don‚Äôt
come from some non-human contaminant <span class="notranslate">DNA</span> or represent a sequencing artefact
of some sort.</p>
  </li>
  <li>
    <p>deduplicate <span class="notranslate">reads</span></p>

    <p>Duplicate <span class="notranslate">reads</span>, which typically arise from PCR-overamplification of genomic
fragments during sequencing library preparation, can, to some extent, lead to
wrong genotype assignments at variant sites (if, for example, a sample is
heterozygous for a variant, but fragments with one of the two alleles get
amplified more efficiently than the others).</p>
  </li>
</ul>

<blockquote class="comment">
  <h3 id="comment-more-postprocessing-steps"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> More postprocessing steps</h3>
  <p>The <a href="../somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
provides an example of a much more complex mapped <span class="notranslate">reads</span> postprocessing
pipeline that may be required for optimal performance of other variant
callers.</p>
</blockquote>

<h2 id="filtering-on-mapped-reads-properties">Filtering on mapped <span class="notranslate">reads</span> properties</h2>

<p>To produce new filtered BAM datasets with only mapped <span class="notranslate">reads</span> the mate of which is also mapped:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filtering-for-read-pair-mapping-status"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filtering for read pair <span class="notranslate">mapping</span> status</h3>

  <ol>
    <li><strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following
parameters (leaving non-mentioned ones at their defaults):
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSAM or BAM file to filter‚Äù</em>: all 3 mapped <span class="notranslate">reads</span>
 datasets of the family trio, outputs of <strong>Map with BWA-MEM</strong>
 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúFilter on bitwise flag‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
      </ul>
      <ul>
        <li><em>‚ÄúOnly output alignments with all of these flag bits set‚Äù</em>:
Do not select anything here!</li>
        <li><em>‚ÄúSkip alignments with any of these flag bits set‚Äù</em>:
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúThe read is unmapped‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúThe mate is unmapped‚Äù</em></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This will result in three new datasets, one for each sample in the analysis.</p>

<blockquote class="details">
  <h3 id="details-more-than-one-way-to-filter"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More than one way to filter</h3>
  <p>Instead of the above filter conditions we could also have exploited the
<em>Read is mapped in a proper pair</em> flag bit.</p>

  <p>For a read to be flagged as being mapped in a proper pair its mate needs to
be mapped, but the mapped pair also needs to meet additional,
aligner-specific criteria. These may include (and do so for <strong>BWA-MEM</strong>):</p>

  <ul>
    <li>both <span class="notranslate">reads</span> need to map to the same chomosome</li>
    <li>the two <span class="notranslate">reads</span> need to map to the reference in the orientation expected by
the aligner</li>
    <li>the two read pairs need to map to the reference within an
aligner-determined distance</li>
  </ul>

  <p>Thus, filtering based on the flag has two consequences:</p>

  <ul>
    <li>filtering will be stricter than with just the <em>Read is unmapped</em> and <em>Mate
is unmapped</em> flags above</li>
    <li>
      <p>you will eliminate read pairs that could be informative with regard to
chromosomal rearrangements and insertion/deletion events</p>

      <p>=&gt; Do not filter for properly paired <span class="notranslate">reads</span> if you plan to detect such
structural variants!</p>
    </li>
  </ul>

  <p>In addition, the <em>proper pair</em> flag is considered undefined if the read
itself is unmapped, so a <em>proper pair</em> filter should eliminate unmapped <span class="notranslate">reads</span>
explicitly to be on the safe side.</p>

  <p>Thus, if you would like to use proper pair filtering (we have no intention to
detect structural variants in this tutorial) instead of just filtering for
mapped <span class="notranslate">reads</span> with a mapped mate, you could run the alte<span class="notranslate">rna</span>tive:</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on:</h3>

    <ol>
      <li><strong>Filter SAM or BAM, output SAM or BAM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSAM or BAM file to filter‚Äù</em>: all 3 mapped
 <span class="notranslate">reads</span> datasets of the family trio, outputs of <strong>Map with BWA-MEM</strong>
 <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>‚ÄúFilter on bitwise flag‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code></li>
        </ul>
        <ul>
          <li><em>‚ÄúOnly output alignments with all of these flag bits set‚Äù</em>:
            <ul>
              <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúRead is mapped in a proper pair‚Äù</em></li>
            </ul>
          </li>
          <li><em>‚ÄúSkip alignments with any of these flag bits set‚Äù</em>:
            <ul>
              <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúThe read is unmapped‚Äù</em></li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="removing-duplicate-reads">Removing duplicate <span class="notranslate">reads</span></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove duplicates</h3>

  <ol>
    <li><strong>RmDup</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM file‚Äù</em>: all 3 filtered <span class="notranslate">reads</span> datasets; the
 outputs of <strong>Filter SAM or BAM</strong></li>
        <li><em>‚ÄúIs this paired-end or single end data‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM is paired-end</code></li>
      </ul>
      <ul>
        <li><em>‚ÄúTreat as single-end‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Again, this will produce three new datasets, one for each member of the family
trio.</p>

<h1 id="variant-calling">Variant calling</h1>

<p>With the sequenced <span class="notranslate">reads</span> of all samples mapped and postprocessed, we can start
looking for evidence of sequence deviations, <em>i.e.</em> variants, between the
sequenced genomic samples and the reference genome.</p>

<p>This task has been automated and optimized continuously over the last decade,
and modern variant calling software hides much of the complexity involved in
it. At least a basic understanding of the underlying concepts is still highly
recommended though and, if you are new to variant calling, the tutorial on
<a href="../dip/tutorial.html">Calling variants in diploid systems</a> may be a good
starting point for you.</p>

<h2 id="generating-freebayes-calls">Generating FreeBayes calls</h2>

<p>We will use <strong>FreeBayes</strong> to call our variants. <strong>FreeBayes</strong> is a Bayesian
genetic variant detector designed to find small polymorphisms, specifically
SNPs (single-nucleotide polymorphisms), indels (insertions and deletions), MNPs
(multi-nucleotide polymorphisms), and complex events (composite insertion and
substitution events) smaller than the length of a short-read sequencing
alignment.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generating-freebayes-calls"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generating FreeBayes calls</h3>

  <ol>
    <li>Run <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>:
      <ul>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
          <ul>
            <li><em>‚ÄúRun in batch mode?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Merge output VCFs</code>
              <ul>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM dataset(s)‚Äù</em>: all three mapped <span class="notranslate">reads</span>
datasets of the family trio; the outputs of <strong>RmDup</strong></li>
              </ul>
            </li>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-sequence-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> chr8 sequence as a fasta dataset
into your history instead:</p>
            <ul>
              <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">History</code>
                <ul>
                  <li><em>‚ÄúRun in batch mode?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Merge output VCFs</code>
                    <ul>
                      <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM dataset(s)‚Äù</em>: all three mapped
<span class="notranslate">reads</span> datasets of the family trio</li>
                    </ul>
                  </li>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the
reference sequence‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>‚ÄúLimit variant calling to a set of regions?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not limit</code></li>
        <li><em>‚ÄúChoose parameter selection level‚Äù</em>: <code class="language-plaintext highlighter-rouge">1. Simple diploid calling</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> <strong>Congratulations!</strong>
You have created you first multisample VCF file, one of the most complicated
file formats in bioinformatics. For every variant detected in at least one of
your samples, this tab-separated format uses a single line to store all
information about the variant. This includes but is not limited to:</p>

<ul>
  <li>the position of the variant in the genome (with respect to the reference
genome used for the analysis)</li>
  <li>the nature of the variant (the actual sequence change associated with it)</li>
  <li>the detected genotype of every sample at the variant position</li>
  <li>measures of the reliability of the variant call and of all individual
genotype calls</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-optional-hands-on-inspect-the-vcf-output-produced-by-freebayes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Optional hands-on: Inspect the VCF output produced by FreeBayes</h3>

  <ol>
    <li>Display the VCF dataset:
      <ul>
        <li>
          <p>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> icon next to the VCF dataset generated
by FreeBayes to display its contents.</p>

          <p>VCF is a tabular plain text format though its information density makes
it complicated to  understand.</p>
        </li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>Can you locate at least some of the above-listed information in the
dataset?</p>

        <p>Hints:</p>
        <ul>
          <li>Lines starting with <code class="language-plaintext highlighter-rouge">##</code> are comment lines explaining the content of
the file.</li>
          <li>Diploid genotypes at biallelic sites are encoded using <code class="language-plaintext highlighter-rouge">0/0</code>, <code class="language-plaintext highlighter-rouge">0/1</code>
and <code class="language-plaintext highlighter-rouge">1/1</code> to represent homozygous reference, heterozygous and
homozygous variant states, respectively.</li>
        </ul>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ul>
            <li>
              <p>The position of each variant with respect to the reference genome is
stored in the first two columns (labeled <code class="language-plaintext highlighter-rouge">CHROM</code> and <code class="language-plaintext highlighter-rouge">POS</code>).</p>
            </li>
            <li>
              <p>The sequence change associated with the variant is provided by
columns 4 and 5 (<code class="language-plaintext highlighter-rouge">REF</code> and <code class="language-plaintext highlighter-rouge">ALT</code>).</p>

              <p>The first of these lists the sequence found at the variant site in
the reference genome, the second the variant sequence.</p>
            </li>
            <li>
              <p>The genotypes of the samples are stored as part of columns 10 and
up.</p>

              <p>Column 9 (<code class="language-plaintext highlighter-rouge">FORMAT</code>) provides the legend for understanding the
sample-specific columns following it and tells us that the first
element (before the first <code class="language-plaintext highlighter-rouge">:</code> separator) of each sample column holds
the sample genotype (<code class="language-plaintext highlighter-rouge">GT</code>).</p>
            </li>
            <li>
              <p>The variant call quality (a measure of the reliability of the
variant existing in any of the samples, <em>i.e.</em> it being real) is
stored in column 6 (<code class="language-plaintext highlighter-rouge">QUAL</code>).
The values are <a href="https://en.wikipedia.org/wiki/Phred_quality_score">Phred-scaled</a>
like sequenced <span class="notranslate">reads</span> base qualities.</p>

              <p>The (log10-scaled) likelihoods of individual genotype calls are
stored as the last element (<code class="language-plaintext highlighter-rouge">GL</code>) in every sample column.</p>
            </li>
          </ul>

          <p>If you are eager to learn all the details about the format, you could
consult the excellent (but also long) official
<a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf">VCF specification</a>.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="post-processing-freebayes-calls">Post-processing FreeBayes calls</h2>

<p>Before starting to analyze the detected variants, we need to post-process the
VCF dataset to fix a few incompatibilities between Freebayes and <span class="notranslate">downstream</span>
analysis tools. In particular, we want to:</p>

<ul>
  <li>
    <p>Split multiallelic variant records, <em>i.e.</em>, records that list more than one
alte<span class="notranslate">rna</span>te allele at a given genomic position, into separate record lines.</p>

    <p>This will allow us to annotate each record with information about the impact
of one specific variant allele further on.</p>
  </li>
  <li>
    <p>Make sure that indels are represented in left-aligned and normalized form
because this is how known indels are stored in public annotation databases.</p>

    <blockquote class="details">
      <h3 id="details-indel-normalization"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Indel normalization</h3>
      <p>Indel normalization is a surprisingly complex topic, which is explained
really well and in detail in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481842/">Tan et al., 2015</a>.
The following figure illustrates the various possibilities to represent
an indel, only one of which is the <em>normalized</em> form:</p>
      <figure id="figure-1"><img src="../../images/indel_normalization.png" alt="different ways to represent the same indel" /><figcaption><span class="figcaption-prefix">Figure 1:</span> The many ways to represent an indel (from: https://genome.sph.umich.edu/wiki/Variant_Normalization</figcaption></figure>
      <p>.</p>

    </blockquote>
  </li>
</ul>

<p>A tool that can do this and also ensures that a VCF dataset conforms to
standards in some other, less important respects is <strong>bcftools norm</strong>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-post-processing-freebayes-calls"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Post-processing FreeBayes calls</h3>

  <ol>
    <li><strong>bcftools norm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúVCF/BCF Data‚Äù</em>: the VCF output of <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><em>‚ÄúReference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-sequence-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> chr8 sequence as a fasta dataset
into your history instead:</p>
            <ul>
              <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>:
<code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><em>‚ÄúReference genome‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>‚ÄúWhen any REF allele does not match the reference genome base‚Äù</em>:
<code class="language-plaintext highlighter-rouge">ignore the problem (-w)</code></li>
        <li><em>‚ÄúLeft-align and normalize indels?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li>
          <p><em>‚ÄúPerform deduplication for the folowing types of variant records‚Äù</em>:
<code class="language-plaintext highlighter-rouge">do not deduplicate any records</code></p>

          <p>Freebayes is not producing any duplicate calls.</p>
        </li>
        <li><em>‚Äú~multiallelics‚Äù</em>: <code class="language-plaintext highlighter-rouge">split multiallelic sites into biallelic records (-)</code>
          <ul>
            <li>
              <p><em>‚Äúsplit the following variant types‚Äù</em>: <code class="language-plaintext highlighter-rouge">both</code></p>

              <p>We want to split both, multiallelic SNP and indel records.</p>
            </li>
          </ul>
        </li>
        <li>
          <p><em>‚Äúoutput_type‚Äù</em>: <code class="language-plaintext highlighter-rouge">uncompressed VCF</code></p>

          <p>We would like to keep the results human-readable. VCF is also what tools
like <em>SnpEff</em> and <em>GEMINI</em> expect as input. Compressed, binary BCF is
interesting for space-efficient long-term storage of large lists of
variants.</p>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>You could try to look for the differences between the original and the
normalized VCF dataset, but for convenience <strong>bcftools norm</strong> reports a brief
summary of the actions it performed. Expand the dataset in the history (by
clicking on its name) to see this output listing the total number of variant
lines processed, along with the number of split, realigned and skipped records.</p>

<h1 id="variant-annotation-and-reporting">Variant annotation and reporting</h1>

<p>A list of variants detected in a set of samples is a start, but to discover
biologically or clinically relevant information in it is almost impossible
without some additional tools and data. In particular, the
variants in the list need to be:</p>

<ul>
  <li>
    <p><strong>prioritized</strong> with respect to their potential relevance for the biological
/ clinical phenotype that is studied</p>

    <p>Even with exome sequencing, only a fraction of the detected variants will
have a clear impact on the function of a protein (many variants will
introduce silent mutations, or reside in intronic regions still covered by
the exome-enriched sequencing data). Of these, many will have been observed
before in healthy individuals arguing against them playing an important role
in an adverse phenotype.</p>
  </li>
  <li>
    <p><strong>filtered</strong> based on the inheritance pattern expected for a causative
variant</p>

    <p>A multisample VCF file records the most likely genotypes of all samples at
every variant site. Knowing which individuals (samples) are affected by a
phenotype we can exclude variants with inheritance patterns that are
incompatible with the observed inheritance of the phenotype.</p>
  </li>
  <li>
    <p><strong>reported</strong> in a more human-friendly form</p>

    <p>While the VCF format can be used to encode all relevant information about any
variant, it is hard for humans to parse that information.</p>

    <p>Ideally, one would like to generate simpler reports for any set of filtered
and prioritized variants.</p>
  </li>
</ul>

<h2 id="get-data-1">Get data</h2>

<p>Our workhorse for annotating and reporting variants and the genes affected by
them will be the <strong>GEMINI</strong> framework. GEMINI comes bundled with a wealth of
annotation data for human variants from many different sources. These can be
used to annotate any list of human variants conveniently, without the need for
separate downloads and conversion between different annotation data formats.</p>

<blockquote class="details">
  <h3 id="details-custom-annotations-and-gemini"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Custom annotations and GEMINI</h3>
  <p>Beyond its bundled annotation data, GEMINI also provides (limited) support
for using custom annotations.
The <a href="../somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
demonstrates the use of <strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> for this purpose.</p>
</blockquote>

<p>The only additional annotation tool we need, for the purpose of this
tutorial, is the tool <strong>SnpEff</strong>, which can annotate variants with their
calculated effects on known genomic features. Because SnpEff is a generic tool
that can be used on variants found in the genome of any organism we need to
provide it with a so-called <strong>SnpEff genome file</strong> that holds the annotated
features (genes, transcripts, translated regions, <em>etc.</em>) for our genome of
interest.</p>

<p>While annotated variants are all we need to <em>prioritize</em> them as described
above, <em>filtering based on inheritance patterns</em> requires a way to inform
GEMINI about the relationship between our samples and their observed
phenotypes. This is done through a so-called <strong>pedigree file</strong> in PED format,
which is rather simple to generate manually.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-obtain-snpeff-genome-and-gemini-pedigree-files"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Obtain SnpEff genome and GEMINI pedigree files</h3>
  <ol>
    <li>
      <p>Download <strong>SnpEff functional genomic annotations</strong></p>

      <blockquote class="comment">
        <h3 id="comment-shortcut-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Shortcut</h3>
        <p>You can skip this step if the <span class="notranslate">Galaxy</span> server you are working on offers
<code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> as a locally installed snpEff database. You can
check the <strong>Genome source</strong> select list of the <strong>SnpEff eff</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool to see if this is the case.</p>
      </blockquote>

      <p>Use <strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to download genome annotation
 database <code class="language-plaintext highlighter-rouge">hg19</code>.</p>
    </li>
    <li>
      <p>Create a PED-formatted pedigree dataset describing our single-family sample trio:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#family_id    name     pate<span class="notranslate">rna</span>l_id    mate<span class="notranslate">rna</span>l_id    sex    phenotype
FAM           father   0              0              1      1
FAM           mother   0              0              2      1
FAM           proband  father         mother         1      2
</code></pre></div>      </div>

      <p>and set its datatype to <code class="language-plaintext highlighter-rouge">tabular</code>.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new file</h3>

        <ul>
          <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the file contents into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from ‚ÄúAuto-detect‚Äù to <code class="language-plaintext highlighter-rouge">tabular</code></p>
          </li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </blockquote>

      <blockquote class="warning">
        <h3 id="warning-remember-those-sample-names"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> Remember those sample names</h3>

        <p>The above content of the pedigree dataset assumes you chose <code class="language-plaintext highlighter-rouge">father</code>,
<code class="language-plaintext highlighter-rouge">mother</code>, <code class="language-plaintext highlighter-rouge">proband</code> as the sample names at the read <span class="notranslate">mapping</span> step
(if you haven‚Äôt mapped the <span class="notranslate">reads</span> yourself, but started with the
premapped data, you can safely skip this warning section).
By now, these sample names will have been propagated through <em>BWA-MEM</em>
and <em>Freebayes</em> to the VCF dataset of variants. It is <strong>important</strong> that
you use matching sample names in the pedigree and in the VCF dataset, or
GEMINI will not be able to connect the information in them.</p>

        <p>If you have chosen different sample names before, you have to adjust
the pedigree dataset accordingly!</p>

      </blockquote>

      <blockquote class="details">
        <h3 id="details-more-on-ped-files"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More on PED files</h3>

        <p>The PED format is explained in the help section of <strong>GEMINI load</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

        <p>Take a moment and try to understand the information that is encoded in
the PED dataset we are using here.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="variant-annotation-with-functional-genomic-effects">Variant annotation with functional genomic effects</h2>

<p>We need to start annotating our variants with SnpEff simply because Gemini
knows how to parse SnpEff-annotated VCFs, while GEMINI output cannot be used
with SnpEff.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adding-annotations-with-snpeff"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adding annotations with SnpEff</h3>
  <ol>
    <li><strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSequence changes (SNPs, MNPs, InDels)‚Äù</em>: the
output of <strong>bcftools norm</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúInput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">VCF</code></li>
        <li><em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">VCF (only if input is VCF)</code></li>
        <li><em>‚ÄúGenome source‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally installed reference genome</code>
          <ul>
            <li><em>‚ÄúGenome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-snpeff-genome-database"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff genome database</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff genome database into your
history instead:</p>
            <ul>
              <li><em>‚ÄúGenome source‚Äù</em>: <code class="language-plaintext highlighter-rouge">Downloaded snpEff database in your history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSnpEff4.3 Genome Data‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>‚ÄúProduce Summary Stats‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Running the above job will produce two datasets. One is a <em>Summary Stats</em> HTML
report, which contains some interesting general metrics such as a distribution
of variants across gene features. The other one is the main annotation result -
a VCF like the input, but with annotations of variant effects added to the INFO
column.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-optional-hands-on-inspect-the-summary-stats-output-produced-by-snpeff"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Optional hands-on: Inspect the Summary Stats output produced by SnpEff</h3>

  <ol>
    <li>Display the dataset:
      <ul>
        <li>Click the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> icon next to the HTML dataset generated
by SnpEff to display its contents.</li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>One section in the report is <strong>Number of effects by type and region</strong>.
Given that you are analyzing exome data, what is the most surprising
aspect in this section? Do you have an idea how to explain it?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>According to the report, intronic variants make up 50% of all
variants detected!</p>

          <p>Exome capture kits are designed to capture exons plus a bit of
surrounding sequence to ensure proper coverage of the exon ends
including splice junction sites.</p>

          <p>Thus, even though intronic sequences are underrepresented in exome
sequencing data, not all of them are eliminated. Since mutations
in most intron bases are neutral, they can accumulate at higher
frequency than most mutations in exons and, thus, still represent a
relevant fraction of all detected variants.</p>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="generating-a-gemini-database-of-variants-for-further-annotation-and-efficient-variant-queries">Generating a GEMINI database of variants for further annotation and efficient variant queries</h2>

<p>Next, we are going to use the SnpEff-annotated VCF as the basis for more
exhaustive annotation with GEMINI. Unlike SnpEff, GEMINI does not just add
annotations to a list of variants in VCF format. Instead the framework
extracts the variants from the VCF input and stores them, together with newly
added annotations, in an SQL database. It then lets you formulate queries for
retrieving and reporting subsets of variants. The combined variant
extraction/annotation/storage step is performed by the <strong>GEMINI load</strong> tool. In
addition, that same tool can be used to incorporate sample pedigree info into
the database.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-creating-a-gemini-database-from-a-variants-dataset"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Creating a GEMINI database from a variants dataset</h3>

  <ol>
    <li><strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúVCF dataset to be loaded in the GEMINI database‚Äù</em>:
the output of <strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúThe variants in this input are‚Äù</em>: <code class="language-plaintext highlighter-rouge">annotated with snpEff</code></li>
        <li>
          <p><em>‚ÄúThis input comes with genotype calls for its samples‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>Sample genotypes were called by Freebayes for us.</p>
        </li>
        <li><em>‚ÄúChoose a gemini annotation source‚Äù</em>: select the latest available annotations snapshot (most likely, there will be only one)</li>
        <li><em>‚ÄúSample and family information in PED format‚Äù</em>: the pedigree file
prepared above</li>
        <li><em>‚ÄúLoad the following optional content into the database‚Äù</em>
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúGERP scores‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúCADD scores‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúGene tables‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúSample genotypes‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúvariant INFO field‚Äù</em></li>
          </ul>

          <p>Leave <strong>unchecked</strong> the following:</p>
          <ul>
            <li>
              <p><em>‚ÄúGenotype likelihoods (sample PLs)‚Äù</em></p>

              <p>Freebayes does not generate these values</p>
            </li>
            <li>
              <p><em>‚Äúonly variants that passed all filters‚Äù</em></p>

              <p>This setting is irrelevant for our input because Freebayes did not
apply any variant filters.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Running this job generates a GEMINI-specific database dataset, which can only
be processed with other GEMINI tools. The benefit, however, is that we now have
variants, rich annotations and pedigree info stored in a format that enables
flexible and highly efficient queries, which will greatly simplify our actual
task to identify the variant responsible for the child‚Äôs disease!</p>

<blockquote class="details">
  <h3 id="details-the-gemini-suite-of-tools"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> The GEMINI suite of tools</h3>

  <p>For a beginner, the sheer number of GEMINI tools may be a bit daunting and
give the impression that this framework adds a lot of complexity.
In practice, however, you will likely only need a very limited number of
these tools for any given analysis. To help you get an overview, here is a
list of the most general-purpose tools and their function:</p>

  <ul>
    <li>Every analysis will always start with <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to
build the GEMINI database from a variant list in VCF format and the
default annotations bundled with GEMINI.</li>
    <li><strong>GEMINI amend</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
let you add additional information to a GEMINI database after it has been
created. Use <strong>GEMINI amend</strong> to overwrite the pedigree info in a database
with the one found in a given PED input, and <strong>GEMINI annotate</strong> to add
custom annotations based on a VCF or <span class="notranslate">BED</span> dataset.</li>
    <li><strong>GEMINI database info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> lets you peek into the content of
a GEMINI database to explore, <em>e.g</em>, what annotations are stored in it.</li>
    <li><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>GEMINI gene-wise</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
are the two generic and most flexible tools for querying, filtering and
reporting variants in a database. There is hardly anything you cannot do
with these tools, but you need to know or learn at least some SQL-like
syntax to use them.</li>
    <li><strong>GEMINI inheritance pattern</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> simplifies variant queries
based on their inheritance pattern. While you could use the <em>query</em> and
<em>gene-wise</em> tools for the same purpose, this tool allows you to perform a
range of standard queries without typing any SQL. <strong>This tool is what we
are going to use here to identify the causative variant.</strong></li>
    <li>The remaining tools serve more specialized purposes, which are beyond the
scope of this tutorial.</li>
  </ul>

  <p>The <a href="../somatic-variants/tutorial.html">Somatic variant calling tutorial</a>
demonstrates the use of the <strong>GEMINI annotate</strong> and <strong>GEMINI query</strong> tools,
and tries to introduce some essential bits of GEMINI‚Äôs SQL-like syntax.</p>

  <p>For a thorough explanation of all tools and functionality you should consult
the <a href="https://gemini.readthedocs.io">GEMINI documentation</a>.</p>
</blockquote>

<h2 id="candidate-variant-detection">Candidate variant detection</h2>

<p>Let us now try to identify variants that have the potential to explain the
boy child‚Äôs osteopetrosis phenotype. Remember that the parents are
consanguineous, but both of them do not suffer from the disease. This
information allows us to make some assumptions about the inheritance pattern of the causative variant.</p>

<blockquote class="question">
  <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>Which inheritance patterns are in line with the phenotypic observations
for the family trio?</p>

  <p>Hint: GEMINI easily lets you search for variants fitting any of the following
inheritance patterns:</p>
  <ul>
    <li>Autosomal recessive</li>
    <li>Autosomal dominant</li>
    <li>X-linked recessive</li>
    <li>X-linked dominant</li>
    <li>Autosomal de-novo</li>
    <li>X-linked de-novo</li>
    <li>Compound heterozygous</li>
    <li>Loss of heterozygosity (LOH) events</li>
  </ul>

  <p>Think about which of these might apply to the causative variant.</p>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ul>
      <li>Since both parents are unaffected the variant cannot be dominant and
inherited.</li>
      <li>A de-novo acquisition of a dominant (or an X-linked recessive) mutation
is, of course, possible.</li>
      <li>A recessive variant is a possibility, and a more likely one given
the parents‚Äô consanguinity.</li>
      <li>For both the de-novo and the inherited recessive case, the variant could
reside on an autosome or on the X chromosome. Given that we provided you
with only the subset of sequencing <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to chr8, an X-linked
variant would not be exactly instructive though ;)</li>
      <li>A compound heterozygous combination of variant alleles affecting the
same gene is possible, but less likely given the consanguinity of the
parents (as this would require two deleterious variant alleles in the
gene circulating in the same family).</li>
      <li>A loss of heterozygosity (LOH) turning a heterozygous recessive variant
into a homozygous one could be caused by uniparental disomy or by an LOH
event early in embryonic development, but both these possibilities have
an exceedingly small probability.</li>
    </ul>

    <p>Based on these considerations it makes sense to start looking for
inherited autosomal recessive variants first. Then, if there is no
convincing candidate mutation among them, you could extend the search to
de-novo variants, compund heterozygous variant pairs and LOH events -
probably in that order.</p>
  </blockquote>
</blockquote>

<p>Since our GEMINI database holds the variant and genotype calls for the
family trio and the relationship between the family members, we can make use
of <strong>GEMINI inheritance pattern</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to report all variants
fitting any specific inheritance model with ease.</p>

<p>Below is how you can perform the query for inherited autosomal recessive
variants. Feel free to run analogous queries for other types of variants that
you think could plausibly be causative for the child‚Äôs disease.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-finding-and-reporting-plausible-causative-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Finding and reporting plausible causative variants</h3>
  <ol>
    <li><strong>GEMINI inheritance pattern</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
      <ul>
        <li><em>‚ÄúGEMINI database‚Äù</em>: the GEMINI database of annotated variants; output
of <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúYour assumption about the inheritance pattern of the phenotype of interest‚Äù</em>:
<code class="language-plaintext highlighter-rouge">Autosomal recessive</code>
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúAdditional constraints on variants‚Äù</em>
              <ul>
                <li>
                  <p><em>‚ÄúAdditional constraints expressed in SQL syntax‚Äù</em>:
<code class="language-plaintext highlighter-rouge">impact_severity != 'LOW'</code></p>

                  <p>This is a simple way to prioritize variants based on their
functional genomic impact. Variants with <em>low impact severity</em> would
be those with no obvious impact on protein function (<em>i.e.</em>, silent
mutations and variants outside coding regions)</p>
                </li>
              </ul>
            </li>
            <li>
              <p><em>‚ÄúInclude hits with less convincing inheritance patterns‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>This option is only meaningful with larger family trees to account
for errors in phenotype assessment.</p>
            </li>
            <li>
              <p><em>‚ÄúReport candidates shared by unaffected samples‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

              <p>This option is only meaningful with larger family trees to account
for alleles with partial phenotypic penetrance.</p>
            </li>
          </ul>
        </li>
        <li>
          <p><em>‚ÄúFamily-wise criteria for variant selection‚Äù</em>: keep default settings</p>

          <p>This section is not useful when you have data from just one family.</p>
        </li>
        <li>In <em>‚ÄúOutput - included information‚Äù</em>
          <ul>
            <li><em>‚ÄúSet of columns to include in the variant report table‚Äù</em>:
<code class="language-plaintext highlighter-rouge">Custom (report user-specified columns)</code>
              <ul>
                <li><em>‚ÄúChoose columns to include in the report‚Äù</em>:
                  <ul>
                    <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúalte<span class="notranslate">rna</span>tive allele frequency (max_aaf_all)‚Äù</em></li>
                  </ul>
                </li>
                <li><em>‚ÄúAdditional columns (comma-separated)‚Äù</em>:
<code class="language-plaintext highlighter-rouge">chrom, start, ref, alt, impact, gene, clinvar_sig,
clinvar_disease_name, clinvar_gene_phenotype, rs_ids</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-variant--vs-gene-centric-clinvar-information"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Variant- <em>vs</em> gene-centric ClinVar information</h3>
        <p>Most annotation data offered by GEMINI is variant-centric, <em>i.e.</em>,
annotations reported for a variant are specific to this exact variant.
A few annotation sources, however, also provide gene-centric
information, which applies to the gene affected by a variant, not
necessarily the variant itself.</p>

        <p>In the case of <a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar</a>, the
annotation fields/columns <code class="language-plaintext highlighter-rouge">clinvar_sig</code> and <code class="language-plaintext highlighter-rouge">clinvar_disease_name</code> refer
to the particular variant, but <code class="language-plaintext highlighter-rouge">clinvar_gene_phenotype</code> provides
information about the affected gene.</p>

        <p>Including the gene phenotype in the report can be crucial because a
gene may be well known to be disease-relevant, while a particular
variant may not have been clinically observed or been reported before.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-question-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>From the GEMINI reports you generated, can you identify the most likely
candidate variant responsible for the child‚Äôs disease?</p>
</blockquote>

<blockquote class="details">
  <h3 id="details-more-gemini-usage-examples"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More GEMINI usage examples</h3>

  <p>While only demonstrating command line use of GEMINI, the following tutorial
slides may give you additional ideas for variant queries and filters:</p>

  <ul>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Intro-To-Gemini.pdf">Introduction to GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-DeNovo-Tutorial.pdf">Identifying <em>de novo</em> mutations with GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Recessive-Tutorial.pdf">Identifying recessive gene candidates with GEMINI</a></li>
    <li><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Dominant-Tutorial.pdf">Identifying dominant gene candidates with GEMINI</a></li>
  </ul>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>It was not hard to find the most likely causative mutation for the child‚Äôs
disease (you did find it, right?).</p>

<p>Even though it will not always provide as strong support for just one specific
causative variant, analysis of whole-exome sequencing data of family trios (or
other related samples) can often narrow down the search for the cause of a
genetic disease to just a very small, manageable set of candidate variants, the
relevance of which can then be addressed through standard methods.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Exome sequencing is an efficient way to identify disease-relevant genetic variants.</p>
</li>
                            
                            <li><p>Freebayes is a good variant and genotype caller for the joint analysis of multiple samples. It is straightforward to use and requires only minimal processing of mapped reads.</p>
</li>
                            
                            <li><p>Variant annotation and being able to exploit genotype information across family members is key to identifying candidate disease variants. SnpEff and GEMINI, in particular, are powerful tools offered by Galaxy for that purpose.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Exome sequencing data analysis for diagnosing a genetic disease (Variant Analysis)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Wolfgang Maier, B√©r√©nice Batut, Torsten Houwaart, Anika Erxleben, Bj√∂rn Gr√ºning, 2021 <b>Exome sequencing data analysis for diagnosing a genetic disease (Galaxy Training Materials)</b>. <a href="/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html">/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{variant-analysis-exome-seq,
    author = "Wolfgang Maier and B√©r√©nice Batut and Torsten Houwaart and Anika Erxleben and Bj√∂rn Gr√ºning",
    title = "Exome sequencing data analysis for diagnosing a genetic disease (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/variant-analysis/tutorials/exome-seq/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Wolfgang Maier, B√©r√©nice Batut, Torsten Houwaart, Anika Erxleben, Bj√∂rn Gr√ºning)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/exome-seq/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
