---
layout: tutorial_slides
topic_name: "Dev-Corner"
tutorial_name: tool_integration
logo: "GTN"
---

## A tool in Galaxy

---

### A tool in Galaxy

![](../images/galaxy_instance_detailed_screenshot.png)

---

### Galaxy tool / wrapper

![](../images/graphlan_screenshot.png)

```bash
graphlan.py --format "png" --size 7 input_tree.txt png_image.png
```

---

### Wrapper

- Description of the user interface
- Link between the tool and the Galaxy instance
- How to invoke the tool
- Which options to pass
- Which files the tool will produce as output

---

### Wrapper

![](../images/wrapper_layers.png)

???

1. `<inputs>` (datasets and parameters) specified in the tool XML are exposed in the Galaxy tool UI
2. When the user fills the form and click the `Execute` button, Galaxy fills the `<command>` template in the XML with the inputs entered by the user and execute the Cheetah code, producing a script as output
3. Galaxy creates a job for the generated script and executes it somewhere (bowtie is run in this case)
4. Some (not necessarily all) output files become new history datasets, as specified in the `<outputs>` XML tag set

---

![](../images/planemo-logo.png)

> Command-line utilities to assist in building and publishing Galaxy tools.

- [Documentation](https://planemo.readthedocs.io/en/latest/)
- [Tutorial](https://planemo.readthedocs.io/en/latest/writing_standalone.html)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal

- Developing a wrapper for Seqtk

    A tool for processing sequence data in FASTA and FASTQ files

---

## Wrapper

---

### Wrapper

![](../images/wrapper_big_picture_1.png)

???

- CDATA tags are used to prevent the interpretation of ampersands and less-than signs as XML special characters
- The tool name and description are combined in the left panel (tool menu), keep them short!

---

### Wrapper

![](../images/wrapper_big_picture_2.png)

---

### Wrapper

![](../images/wrapper_big_picture_3.png)

---

### Wrapper

.image-10[![](../images/exercise.png)]

<small>Galaxy tool XML is formally defined in a XML Schema Definition (XSD), used to generate the [online documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html)</small>

---

### `command`
#### How to invoke the tool?

```xml
<command><![CDATA[
graphlan.py
--format $format
#if $dpi
  --dpi $dpi
#end if
--size $size
"$input_tree"
]]></command>
```

<small>If the script is provided with the wrapper xml</small>
```xml
<command><![CDATA[
python '$__tool_directory__/graphlan.py'
# command params...
]]></command>
```

???

- In the first case, `graphlan.py` is expected to be on the PATH when the job executes
- In the second case, `$__tool_directory__` is a special variable which is substituted by Galaxy with the directory where the tool XML is

---

### `command`
#### How to pass the parameters?

![](../images/graphlan_screenshot.png)

---

### `inputs` > `param` to `command`
#### How to pass the parameters?

```xml
<inputs>
  <param name="dpi" type="..." label="..." optional="True"/>

  <param argument="--size" type="..." value="7" label="..." help="..."/>
</inputs>
```

#### Directly linked to `<command>`

```xml
<command><![CDATA[
graphlan.py
...
#if $dpi
  --dpi $dpi
#end if
--size $size
...
]]></command>
```

---

### `inputs` > `param` > `data`

![](../images/input_data.png)

```xml
<param name="..." type="data" format="txt" label="..." help="..." />
```

.footnote[[List of possible formats](https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample)]

---

### `inputs` > `param` > `integer` | `float`

![](../images/input_integer.png)

```xml
<param name="..." type="integer" value="7" label="..." help="..."/>
```

![](../images/input_integer_range.png)

```xml
<param name="..." type="float" min="0" max="10" value="1" label="..."
help="..."/>
```

???

- In the first case, Galaxy creates a text box which accepts only integer values
- In the second case, since *both* `min` and `max` are specified, a slider is shown in addition

---

### `inputs` > `param` > `text`

![](../images/input_text.png)

```xml
<param name="..." type="text" value="" label="..." help="..."/>
```

---

### `inputs` > `param` > `select`

![](../images/input_select.png)

```xml
<param name="..." type="select" label="..." help="...">
  <option value="png" selected="true">PNG</option>
  <option value="pdf">PDF</option>
  <option value="ps">PS</option>
  <option value="eps">EPS</option>
  <option value="svg">SVG</option>
</param>
```

---

### `inputs` > `param` > `select`

![](../images/input_select_checkboxes.png)

```xml
<param name="..." type="select" display="radio" label="..." help="...">
  <option value="min" selected="true">Minimum</option>
  <option value="mean">Mean</option>
  <option value="max">Max</option>
  <option value="sum">Sum</option>
</param>
```

---

### `inputs` > `param` > `select`

![](../images/input_select_multiple.png)

```xml
<param name="..." type="select" multiple="true" label="..." help="...">
  <option value="ld" selected="true">Length distribution</option>
  <option value="gc" selected="true">GC content distribution</option>
</param>
```

---

### `inputs` > `param` > `boolean`

![](../images/input_boolean.png)

```xml
<param name="..." type="boolean" checked="false" truevalue="--log" falsevalue=""
  label="..."  help="..." />
```

---

class: packed

### `inputs` > `param` > `conditional`

![](../images/input_conditional.png)

```xml
<command><![CDATA[
#if $input_format.format == "fasta":
  phyml_fasta_id_encode_wrapper.sh $input_format.input $output_tree
#else if $input_format.format == "phylip":
  phyml.sh
#end if
]]></command>
<inputs>
  <conditional name="input_format">
    <param type="select" name="format" label="Input format">
      <option value="phylip">Phylip</option>
      <option value="fasta">Fasta</option>
    </param>
    <when value="phylip">
      <param format="phylip" name="input" type="data"  />
    </when>
    <when value="fasta">
      <param format="fasta" name="input" type="data"  />
    </when>
  </conditional>
</inputs>
```

---

class: packed

### `inputs` > `param` > `repeat`

![](../images/input_repeat.png)

```xml
<command><![CDATA[
#for $i, $s in enumerate( $series )
  rank_of_series=$i
  input_path=${s.input}
  x_colom=${s.xcol}
  y_colom=${s.ycol}
#end for
]]></command>

<inputs>
  <repeat name="series" title="Series">
    <param name="input" type="data" format="tabular" label="Dataset"/>
    <param name="xcol" type="data_column" data_ref="input" label="Column for x axis"/>
    <param name="ycol" type="data_column" data_ref="input" label="Column for y axis"/>
  </repeat>
</inputs>
```

---

### `outputs`
#### Which files the tool will produce as output?

![](../images/output.png)

---

### `outputs`

```xml
<outputs>
  <data format="txt" name="tree" label="${tool.name} on ${on_string}: Tree" />
  <data format="txt" name="annotation" label="${tool.name} on ${on_string}:
    Annotation" />
</outputs>
```

???

`${tool.name} on ${on_string}` is the default output label, need to modify this if the tool generates more than 1 output

---

### `outputs` > `filter`
#### Conditional

```xml
<inputs>
  <param type="select" name="format" label="Output format">
    <option value="png">PNG</option>
    <option value="pdf">PDF</option>
  </param>
</inputs>
<outputs>
  <data format="png" name="png_output" label="${tool.name} on ${on_string}: PNG">
    <filter>format=="png"</filter>
  </data>
  <data format="pdf" name="pdf_output" label="${tool.name} on ${on_string}: PDF">
    <filter>format=="pdf"</filter>
  </data>
</outputs>
```

---

### `stdio`

Error if tool exit code is not 0 (default):
```xml
<command detect_errors="exit_code"> ... </command>
```
Error if tool exit code is not 0 or either "Exception:" or "Error:" appears in standard error/output:
```xml
<command detect_errors="aggressive"> ... </command>
```
If you need more precision:
```xml
<stdio>
  <exit_code range=":-2" level="warning" description="Low disk space" />
  <exit_code range="1:" level="fatal"  />
  <regex match="Error:"  level="fatal" />
</stdio>
```

<small>"Warning" level allow to add information to stderr without marking the dataset as failed</small>

---

### `help`

![](../images/help.png)

```xml
<help><![CDATA[

**What it does**
GraPhlAn is a software tool for producing high-quality circular
representations of taxonomic and phylogenetic trees. GraPhlAn focuses
on concise, integrative, informative, and publication-ready
representations of phylogenetically- and taxonomically-driven
investigation.

For more information, check the `user manual
<https://bitbucket.org/nsegata/graphlan/overview>`_.

]]></help>
```

.footnote[[reStructuredText markup format](http://docutils.sourceforge.net/rst.html)]

---

### `citations`

![](../images/citations.png)

```xml
<citations>
  <citation type="doi">10.1093/bioinformatics/bts611</citation>
  <citation type="doi">10.1093/nar/gks1219</citation>
  <citation type="doi">10.1093/nar/gks1005</citation>
  <citation type="doi">10.1093/bioinformatics/btq461</citation>
  <citation type="doi">10.1038/nbt.2198</citation>
</citations>
```

<small>If no DOI is available, a BibTeX citation can be specified with `type="bibtex"`</small>

---

### Other tips

Always quote text/data parameters

```xml
<command><![CDATA[
graphlan.py
--format '$format'
#if $dpi
  --dpi $dpi
#end if
--size $size
'$input_tree'
]]></command>
```

- For security reason
- Prefer single quotes over double quotes

---

### Other tips

Use &amp;&amp; to run multiple commands

```xml
<command><![CDATA[
graphlan.py
--format '$format'
&&
echo "Yeah it worked!"
]]></command>
```

<small>The job will exit on the first error encountered.</small>

---

### Other tips

Use the `argument` tag when a param name reflects the command line argument

```xml
<param argument="--size" type="integer" value="7" label="..." help="..."/>
```

- It will be printed at the end of the param help section
- No need to write `name="size"`, it is implicit

---

### Other tips

Use sections to group related parameters

```xml
<section name="advanced" title="Advanced options" expanded="False">
    <param argument="--size" type="integer" value="7" label="..." help="..."/>
</section>
```

---

###.image-25[![](../images/planemo-logo.png)]

#### `planemo tool_init`

Creates a skeleton of xml file

```bash
$ mkdir new_tool
$ cd new_tool
$ planemo tool_init --id 'some_short_id' --name 'My super tool'
```

Complicated version:

```bash
$ planemo tool_init  --id "samtools_sort" --name "Samtools sort" \
                     --description "Sort alignments by leftmost coordinates, or by read nam" \
                     --requirement samtools \
                     --example_command "samtools sort -o 1.bam null > 1_sorted.bam" \
                     --example_input 1.bam \
                     --example_output 1_sorted.bam \
                     --test_case --version_command "samtools --version | head -1" \
                     --help_from_command "samtools sort" \
                     --doi 10.1093/bioinformatics/btp352'
```

---

class: packed

###.image-25[![](../images/planemo-logo.png)]

#### `planemo lint`

Checks the syntax of a wrapper

```bash
$ planemo lint
```

```bash
Linting tool /opt/galaxy/tools/seqtk_seq.xml
Applying linter tests... CHECK
.. CHECK: 1 test(s) found.
Applying linter output... CHECK
.. INFO: 1 outputs found.
Applying linter inputs... CHECK
.. INFO: Found 1 input parameters.
Applying linter help... CHECK
.. CHECK: Tool contains help section.
.. CHECK: Help contains valid reStructuredText.
Applying linter general... CHECK
.. CHECK: Tool defines a version [0.1.0].
.. CHECK: Tool defines a name [Convert to FASTA (seqtk)].
.. CHECK: Tool defines an id [seqtk_seq].
Applying linter command... CHECK
.. INFO: Tool contains a command.
Applying linter citations... CHECK
.. CHECK: Found 1 likely valid citations.
```

---

###.image-25[![](../images/planemo-logo.png)]

#### `planemo serve`

View your new tool in a local Galaxy instance

```bash
$ planemo serve
```

Open http://127.0.0.1:9090/ in your web browser to view your new tool

---

###.image-25[![](../images/planemo-logo.png)]

#### Building Galaxy Tools

![](../images/planemo_tool_building_lint.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Before: install Seqtk
*No brain needed*

```bash
$ . ~/.venv_planemo/bin/activate  # not mandatory
$ planemo conda_init
$ export PATH=~/miniconda3/bin:$PATH
$ conda config --add channels conda-forge
$ conda config --add channels defaults
$ conda config --add channels r
$ conda config --add channels bioconda
$ conda install seqtk
```

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Before: observe and test Seqtk

```bash
$ seqtk

$ seqtk seq

$ wget https://raw.githubusercontent.com/galaxyproject/galaxy-test-data/master/2.fastq

$ cat 2.fastq

$ seqtk seq -A 2.fastq > 2.fasta

$ cat 2.fasta
```

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Create a new directory for your tool, `cd` into it
2. Init the wrapper with `planemo tool_init`
3. Fill the template
4. Test the template with `planemo lint`
<small>(Don't worry about "No tests found" for now)</small>
5. Display the tool with `planemo serve`
6. and try it with a file

???

```bash
# 1.
planemo tool_init
# -> hands_on-tool_integration/seqtk/seqtk_seq_step1.xml
# 2. -> hands_on-tool_integration/seqtk/seqtk_seq_step2.xml
# 3.
planemo lint # seqtk_seq_step2.xml
# 4.
planemo serve # seqtk_seq_step2.xml
```

---

## Functional tests

---

### Functional tests

- Functional testing is a quality assurance (QA) process.
- The tests confort developers and users that the tools can run across different servers/architectures. And that the latest modifications don't break older features.
- Tools are tested by feeding them input(s) and parameter(s) and examining the output(s) (typically a diff)

---

### Functional tests
#### Good practices

1. Build tests using inputs/parameters/outputs
2. Run the tests --> Failed --> Coffee
3. Develop
4. Run the tests --> Failed  --> Coffee
5. Develop
6. Run the tests --> Succeed --> Beer

---

### `tests`

```xml
<tests>
  <test>
    <param name="input_tree" value="input_tree.txt"/>
    <param name="format" value="png"/>
    <param name="dpi" value="100"/>
    <param name="size" value="7"/>
    <param name="pad" value="2"/>
    <output name="png_output_image" file="png_image.png" />
  </test>
</tests>
```

<small>`input_tree.txt` and `png_image.png` must be in the `test-data` directory</small>

---

### Wrapper directory tree

```
graphlan/
├── graphlan.xml
├── graphlan.py
└── test-data/
    ├── input_tree.txt
    └── png_image.png

```

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
<output name="out_file1" file="cf_maf2fasta_concat.dat" ftype="fasta"/>
```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
<test>
  <output ...
    compare="diff|re_match|sim_size|contains|re_match_multiline"
    ... />

```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
<test>
  <output ...
    compare="diff"
    lines_diff="4"
    ... />

```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
<test>
  <output ...
    compare="sim_size"
    delta="1000"
    ... />

```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: comparing to an expected result

```xml
<test>
  <output ...
    md5="68b329da9893e34099c7d8ad5cb9c940"
    ... />

```

.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output)]

---

### `tests`
#### Comparison Methods: checking the content

```xml
<test>
  <output name="out_file1">
    <assert_contents>
      <has_text text="chr7" />
      <not_has_text text="chr8" />
      <has_text_matching expression="1274\d+53" />
      <has_line_matching expression=".*\s+127489808\s+127494553" />
      <!-- &#009; is XML escape code for tab -->
      <has_line line="chr7&#009;127471195&#009;127489808" />
      <has_n_columns n="3" />
    </assert_contents>
  </output>
</test>
```
.footnote[[Complete documentation](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-output-assert-contents)]

---

### `tests`
#### Tests with conditionals, repeats, sections

```xml
<tests>
  <test>
    <section name="advanced">
        <repeat name="names">
            <param name="first" value="Abraham"/>
            <param name="last" value="Lincoln"/>
        </repeat>
        <repeat name="names">
            <param name="first" value="Donald"/>
            <param name="last" value="Trump"/>
        </repeat>
        <conditional name="image">
          <param name="output_image" value="yes"/>
          <param name="format" value="png"/>
        </conditional>
    </section>
    (...)
  </test>
</tests>
```

---

###.image-25[![](../images/planemo-logo.png)]

#### `planemo test`

Runs all functional tests

```bash
$ planemo test
```

An HTML report is automatically created with logs in case of failing test

---

###.image-25[![](../images/planemo-logo.png)]

#### Test Galaxy Tools

![](../images/planemo_tool_building_test.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal

- Add functional test to the Seqtk wrapper for Seqtk

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Fill a `<test>` tag using the file you generated previously with `seqtk`
2. Use `planemo test` ... until everything is green

???

```bash
# 1. -> hands_on-tool_integration/seqtk/seqtk_seq_step3.xml
# 2.
planemo test # seqtk_seq_step3.xml
```

---

## Dependencies

---

### Dependencies
#### How Galaxy will deal with dependencies?

![](../images/galaxy_instance.png)

---

### Dependencies
#### Embedded tool

```
graphlan/
├── graphlan.xml
└── graphlan.py

```

```xml
<command><![CDATA[
$__tool_directory__/graphlan.py
...
]]></command>
```

---

### Dependencies
#### External tool

Definition of requirements in `.xml` file

```xml
<requirements>
  <requirement type="package" version="1.66">biopython</requirement>
  <requirement type="package" version="1.0.0">graphlan</requirement>
</requirements>
```

Local installation using ~~tool_dependencies.xml~~

---

### Dependencies
#### External tool

Definition of requirements in `.xml` file

```xml
<requirements>
  <requirement type="package" version="1.66">biopython</requirement>
  <requirement type="package" version="1.0.0">graphlan</requirement>
</requirements>
```

Local installation using `conda` package

---

![](../../shared/images/conda_logo.png)

Package, dependency and environment management

---

.image-50[![](../../shared/images/conda_logo.png)]

- Based on recipes describing how to install the software which are then built for their distribution
- No compilation at installation: binaries with their dependencies, libraries...
- Not restricted to Galaxy

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Conda distributions

```


```
![](../images/miniconda_vs_anaconda.png)

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Channels

- Packages through channels within Continuum.Anaconda
- Conda channel searched by Galaxy for packages:
    - iuc
    - bioconda
    - defaults
    - r
    - conda-forge

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Quickstart
#### Use Conda outside Planemo

- Install a package


```bash
conda install pyyaml
```

- Install some packages within an isolate environment

```bash
$ conda create -n yaml pyyaml
$ conda env list
yaml        *  ~/miniconda3/envs/yaml
root             ~/miniconda3
$ source activate yaml
(yaml) $
```

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Tips and tricks
#### Because you don't want Conda binaries all the time in your $PATH
<small>Conda masks some of your binaries like python (with all corresponding python modules) with its own versions</small>

```bash
$ vi ~/.bashrc
```
```bash
function conda3activate () {
  export PATH=~/miniconda3/bin:$PATH; }
function conda3deactivate () {
  export PATH=$(echo $PATH | sed "s|$HOME/miniconda3/bin:||g"); }
```

---

.image-25[![](../images/planemo-logo.png)]
#### Test Galaxy Tools using Conda packages

![](../images/planemo_tool_building.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal
- Use Conda package installation during the functional tests of the wrapper for Seqtk

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Before
1. `conda uninstall seqtk`

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Fill a `<requirement>` tag to target a `seqtk` conda package
2. Use `planemo conda_install .` to install tools and set an environment
3. Use `planemo test`

???

```bash
# 1. -> hands_on-tool_integration/seqtk/seqtk_seq_step4.xml
# 2.
planemo conda_install .
# 3.
planemo test # seqtk_seq_step4.xml
```

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe

You can create a new Conda package. This will be necessary if a specific tool or version is not available.

.footnote[[Documentation about how to write conda recipes from scratch](http://conda.pydata.org/docs/build_tutorials/pkgs2.html) <br/> [Bioconda guidelines](https://bioconda.github.io/guidelines.html)]

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### Files

- `meta.yaml`: contains all the metadata of the recipe
- `build.sh`: the Unix script that installs the files

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### `meta.yaml` basic information

```yaml
package:
  name: bwa
  version: "0.7.15"
about:
  home: http://bio-bwa.sourceforge.net
  license: MIT
  summary: The BWA read mapper.
build:
  number: 0
  skip: True # [osx]
source:
  fn: v0.7.15.tar.gz
  url: https://github.com/lh3/bwa/archive/v0.7.15.tar.gz
  md5: 54fdee953c5c256d36885a1c5c6b118c
```

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### `meta.yaml` requirements

```yaml
requirements:
  build:
    - gcc   # [not osx]
    - llvm  # [osx]
    - zlib
  run:
    - zlib
    - bowtie2
```
<small>
- `build`: requirements needed during the build step (here compilation)
- `run`: requirements needed at runtime

</small>

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### `meta.yaml` test

```yaml
test:
  commands:
    - bowtie2 --version
    - bwa 2>&1 | grep 'index sequences in the'
    - '$R -e "library(''xcms'')"'
```
<small>
Those tests are launched at the end of `conda build`. <br>
They need to return 0 on success.
</small>

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### `build.sh`

```bash
#!/bin/bash
./configure --prefix=$PREFIX
make
make install
```
```bash
#!/bin/bash
mkdir -p $PREFIX/bin
cp *.py $PREFIX/bin
```
<small>
- `$PREFIX` is a variable defined by conda when building the package
- `$PREFIX` usually contains `bin/` `lib/` `include/` that are filled with the package files
- `$PREFIX` will be loaded in the user environment at runtime: `PATH`, `LD_LIBRARY_PATH`, `PYTHONPATH`, ...
</small>

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### Building recipe skeletons

- `conda skeleton pypi <packagename>`
- `conda skeleton cran <packagename>`
- `bioconductor_skeleton.py <packagename>`
- `conda skeleton cpan <packagename>`

Generates pre-filled recipes <small>(not guaranteed to work out of the box)</small>

.footnote[[Building conda packages with conda skeleton](http://conda.pydata.org/docs/build_tutorials/pkgs.html)]

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/> Writing a Conda recipe
#### `conda build` and test the package

```bash
$ conda install conda-build

$ conda build .

```

(`planemo conda_init` now includes `conda-build` so that first command may not be needed.)

1. BUILD START: Builds/Compiles the package
2. BUILD START: Provides a .tar.bz2
3. TEST START: Installs the .tar.bz2 previously generated
4. TEST START: Launches the functional tests
5. (Provides the .tar.bz2 path)

---

###.image-25[![](../../shared/images/conda_logo.png)] <br/>  Install a Conda package for test
#### `conda install`

To continue your `planemo test`, you may want to use your new package before it's available through an official channel.


Get the path of your new package .tar.bz2
```bash
conda install ~/miniconda3/conda-bld/linux-64/bwa-0.7.15-0.tar.bz2
```

---

.image-75[![](../../shared/images/bioconda_logo.png)]


- A channel dedicated to bioinformatics packages https://anaconda.org/bioconda
- Open to [contribution](https://github.com/bioconda/bioconda-recipes/blob/master/CONTRIBUTING.md)
- GitHub repository: https://github.com/bioconda/bioconda-recipes

---

###.image-25[![](../../shared/images/bioconda_logo.png)]

1. Clone `bioconda/bioconda-recipes.git` or a fork
2. Create a new branch "package_new_login"
3. Fill two files `meta.yaml` and `build.sh`
4. Build your new package and test it using `conda build`
5. Push on GitHub
6. Wait for the functional tests to pass on your branch
7. Create a Pull Request on `bioconda/bioconda-recipes.git`
8. Wait for the functional tests to pass on the master branch
9. Enjoy your new Conda package installed from https://anaconda.org/bioconda

See [Contributing with GitHub](github_contribution.html)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

![](../images/exercise.png)

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### The Goal
Build a conda package for the suite seqtk

---

### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

#### Steps
1. Read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html)
2. Follow seqtk [installation instructions](https://github.com/lh3/seqtk/)
3. Write a `meta.yaml` and a `build.sh`
4. Test the recipe with `conda build`

???

```bash
# hands_on-conda/seqtk/meta.yaml
# hands_on-conda/seqtk/build.sh
```

---

![](../images/planemo-logo.png)

> Command-line utilities to assist in building and publishing Galaxy tools.

---

###.image-25[![](../images/planemo-logo.png)]

![](../images/big_picture.png)

---

###.image-25[![](../images/planemo-logo.png)]
#### Checklist (1/3)

- Create a dedicated repository
```bash
mkdir my_tool
cd my_tool
```
- Create the main `.xml` file with
```bash
planemo tool_init
```
- Fill the `.xml` file
- Test the wrapper structure with
```bash
planemo lint
```

---

###.image-25[![](../images/planemo-logo.png)]
#### Checklist (2/3)

- Check if the dependencies are available on Conda
- If not, create the missing Conda packages
- Add functional tests
- Run functional tests
```bash
planemo test
```

---

## Advanced features

---

### Other tags: `configfiles`

Like `command`, uses cheetah code to write:

- A config file
- A whole script or a module
- Any text file needed for the tool execution

---

class: packed

### `configfiles`

```xml
<command><![CDATA[ mb $script_nexus  ]]></command>

<configfiles>
  <configfile name="script_nexus"><![CDATA[
set autoclose = yes;
execute $input_data;
#if str( $data_type.type ) == "nuc“
  lset nst=$data_type.lset_params.lset_Nst;
#end if
mcmcp ngen=$mcmcp_ngen;
mcmc;
quit
  ]]></configfile>
</configfiles>

```

```bash
set autoclose = yes;
execute dataset_42.dat;
lset nst=2 ;
mcmcp ngen=100000;
mcmc;
quit
```

---

### `macros`

![](../images/macro.png)

.footnote[[Planemo documents about macros ](http://planemo.readthedocs.io/en/latest/writing_advanced.html?highlight=macros#macros-reusable-elements)]

---

class: packed

### `macros` > `xml`

macros.xml

```xml
<macros>
  <xml name="requirements">
    <requirements>
      <requirement type="package" version="2.5.0">blast</requirement>
    </requirements>
  </xml>
  <xml name="stdio">
    <stdio>
      <exit_code range="1" level="fatal" />
    </stdio>
  </xml>
</macros>
```

ncbi_blastn_wrapper.xml

```xml
<macros>
  <import>macros.xml</import>
</macros>
<expand macro="requirements"/>
<expand macro="stdio"/>
```

---

### `macros` > `token`

macros.xml

```xml
<macros>
  <token name="@THREADS@">-num_threads "\${GALAXY_SLOTS:-8}"</token>
</macros>
```

ncbi_blastn_wrapper.xml

```xml
<command>
  blastn -query "$query" @THREADS@ [...]
</command>
```

---

### `macros` > `yield`

macros.xml

```xml
<macros>
  <xml name="requirements">
    <requirements>
      <requirement type="package" version="2.2.0">trinity</requirement>
      <yield/>
    </requirements>
  </xml>
</macros>
```

trinity.xml

```xml
<expand macro="requirements">
  <requirement type="package" version="1.1.2">bowtie</requirement>
  <requirement type="set_environment">TRINITY_MEM_OPTIONS</requirement>
</expand>
```

---

### `command` > Reserved Variables

```xml
<command><![CDATA[
# Email’s numeric ID (id column of galaxy_user table in the database)
echo $__user_id__

# User’s email address
echo $__user_email__

# The galaxy.app.UniverseApplication instance, gives access to all other configuration file variables.
# Should be used as a last resort, may go away in future releases.
echo $__app__.config.user_library_import_dir
echo $__app__.tool_data_tables[ 'pfam_databases' ].get_fields() )[0][-1]

# Check a dataset type
#if $input1.is_of_type('gff'):
    echo "input1 type is ${input1.datatype}"
#end if
]]></command>
```
.footnote[[Reserved Variables List](https://docs.galaxyproject.org/en/latest/dev/schema.html#reserved-variables)]

---

### Multiple input

```xml
<param name="..." type="data" format="txt" label="..." help="..." />
```

![](../images/input_data.png)

Possible to select multiple dataset:

![](../images/input_data_multiple.png)

- Useful to launch the same tool on multiple datasets independently
- One job per dataset

---

### Multiple input

```xml
<param name="..." type="data" format="txt" multiple="true" label="..." help="..." />
```

![](../images/input_data_multiple2.png)

In the command:

```xml
<command><![CDATA[
...
#for $input in $inputs
    --input "$input"
#end for
]]></command>
```

One job for all selected dataset

---

### Multiple output

```xml
<outputs>
    <data format="txt" name="output">
        <discover_datasets pattern="__designation_and_ext__"
            directory="output_dir" visible="true" />
    </data>
</outputs>
```

- `__designation_and_ext__`: a predefined regexp,

- catches the dataset identifier + the datatype

If the output file extension is not present/usable:

```xml
<outputs>
    <data format="txt" name="output">
        <discover_datasets pattern="__designation__" format="txt"
            directory="output_dir" visible="true" />
    </data>
</outputs>
```

---

### Dataset collection

To group related datasets, and treat them as a single entity

- `list`: a simple list of datasets
- `paired`: a pair of datasets, `forward` and `reverse` for NGS
- composite: e.g. `list:paired` for a list of dataset pairs

Usage

- Useful to launch a workflow on many samples
- Sample names are kept along the workflow: `element_identifier`
- Galaxy tools are available to manipulate collections

---

.center[### Dataset collection]

.center[Creation from history]

.pull-left[
![](../images/historyWithCheckboxes.png)
]

.pull-right[
![](../images/buildPairs.png)
]

---

### Dataset collection

Creation when uploading (>=17.05)

![](../images/upload_collection.png)

---

### Dataset collection > input

```xml
<param type="data" multiple="true" name="inputs" label="Input BAM(s)"
  format="bam" />
```

![](../images/input_data_collection.png)

Accept only collections:

```xml
<param type="data_collection" collection_type="list|paired|list:paired|..."
  name="inputs" label="Input BAM(s)" format="bam" />
```

```xml
<command><![CDATA[
...
#for $input in $inputs
--input "$input"
--sample_name "$input.element_identifier"
#end for
]]></command>
```

---

### Dataset collection > output

A single paired collection:

```xml
<collection name="paired_output" type="paired" label="Split Pair">
  <data name="forward" format="txt" />
  <data name="reverse" format_source="input1" from_work_dir="reverse.txt" />
</collection>
```

Unknown number of files:

```xml
<collection name="output" type="list" label="Unknown number of files">
  <discover_datasets pattern="__name_and_ext__" directory="outputs" />
</collection>
```


- `__name_and_ext__`: a predefined regexp,
- catches the dataset identifier + the datatype

---

### Some Config

---

### `job_conf.xml`

![](../images/job_conf.xml_1.png)

8 is the default value if not set in destination

```bash
blastn -query foo_bar -num_threads 8
```

---

### `job_conf.xml`

![](../images/job_conf.xml_2.png)

```bash
blastn -query foo_bar -num_threads 4
```

---

### Data tables

- They list all reference data used by tools
- e.g.: Blast databanks, BWA indexes, Fasta files
- Stored in `.loc` files
- Populated by hand or using Data Managers
- Data Managers are dedicated kind of Galaxy tools

---

### Using a data table in a tool

![](../images/tool_data_table_conf.xml.png)

---

### Datatypes

- Every Galaxy dataset is associated with a datatype
- Datatype can be detected or user specified
- Gain of usability

.footnote[[Documentation: Adding Datatypes](https://galaxyproject.org/admin/datatypes/adding-datatypes/)]

---

.image-50[![](../images/github_logo.png)]

.image-50[![](../images/travis_ci_logo.png)]

.image-50[![](../images/planemo-logo.png)]

.image-50[![](../../shared/images/conda_logo.png)]

---

### GitHub side

![](../images/github_travis_github.png)

---

### Travis CI side

![](../images/github_travis_travis.png)

---

.image-50[![](../../shared/images/ansible_logo.png)]

.image-50[![](../../shared/images/conda_logo.png)]

.image-50[![](../../shared/images/docker_logo.png)]

---

### Docker Galaxy Flavors

![](../images/galaxydocker_logo.png)

.footnote[[Docker Galaxy Stable](https://github.com/bgruening/docker-galaxy-stable)]

---

### Docker Galaxy Flavors
#### Dockerfile

```yaml
# Galaxy - My own flavour
# VERSION       0.1

FROM quay.io/bgruening/galaxy:16.07

MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com

ENV GALAXY_CONFIG_BRAND deepTools

# Adding the tool definitions to the container
ADD my_tool_list.yml $GALAXY_ROOT/my_tool_list.yml

# Install deepTools
RUN install-tools $GALAXY_ROOT/my_tool_list.yml
```

---

### Docker Galaxy Flavors
#### tool_list.yml

```yaml
api_key: admin
galaxy_instance: http://127.0.0.1:8080/
tools:
- name: fastqc
  owner: devteam
  tool_panel_section_id: cshl_library_information
  tool_shed_url: https://toolshed.g2.bx.psu.edu
  revisions:
    - '8c650f7f76e9'  # v0.62
    - 'd2cf2c0c8a11'  # v0.63
- name: suite_deeptools
  owner: bgruening
  tool_panel_section_label: 'deepTools'
  tool_shed_url: https://toolshed.g2.bx.psu.edu
  revisions:
    - '7f5562625ae2'
```

---

### Docker Galaxy Flavors
#### RUN

```bash
docker run -d -p 8080:80 galaxy-deeptools
```

---

## Conclusion

---

### IUC: Intergalactic Utilities Commission

- Lobbyist :)
- Host orphan tools: [GitHub](https://github.com/galaxyproject/tools-iuc)
- [IUC-standards](http://galaxy-iuc-standards.readthedocs.io/)
