<!doctype html>
<html lang="en">

  <head>
    <meta charset="utf-8">

    <title>Tool development and integration into Galaxy</title>

    <link rel="stylesheet" href="../../assets/reveal.js/css/reveal.css">
    <link rel="stylesheet" href="../../assets/reveal.js/css/theme/simple.css" id="theme">

    <!-- Code syntax highlighting -->
    <link rel="stylesheet" href="../../assets/reveal.js/lib/css/zenburn.css">
        <link rel="stylesheet" href="../../assets/css/font-awesome.css">
    <link rel="stylesheet" href="../../assets/css/slide_custom.css" id="theme">

    <!-- Printing and PDF exports -->
        <script>
          if( window.location.search.match( /print-pdf/gi ) ) {
            var link = document.createElement( 'link' );
            link.rel = 'stylesheet';
            link.type = 'text/css';
            link.href = '../../assets/reveal.js/css/print/pdf.css';
            document.getElementsByTagName( 'head' )[0].appendChild( link );
          }
        </script>
    <!--<script>
            if (window.location.search.match(/print-pdf/gi)) {
                document.write('<link rel="stylesheet" href="css/pdf.css" type="text/css">');
            }
        </script>-->

    <!--[if lt IE 9]>
        <script src="../../assets/reveal.js/lib/js/html5shiv.js"></script>
        <![endif]-->
  </head>

  <body>
    <div class="reveal">
            <div class="slides">
                <section data-markdown>
                    <script type="text/template">
                        ## Welcome!

                        <img src="../../shared/images/GTNLogo1000.png" alt="GTNLogo1000"
                        class="no-border" width=30%/>

                        The easiest way to **navigate** this slide deck
                        is **by hitting `[space]` on your keyboard**

                        You can also navigate with arrow keys, but be careful because some
                        slides can be nested inside of each other (vertically)
                    </script>
                </section>

                <section data-markdown>
                    <script type="text/template">
                        # Tool development and integration into Galaxy
                    </script>
                </section>

                <!-- INTRODUCTION -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## A tool in Galaxy

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### A tool in Galaxy

                            <img src="../images/galaxy_instance_detailed_screenshot.png" alt="Galaxy instance"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Galaxy tool / wrapper

                            <img src="../images/graphlan_screenshot.png" alt="Tool Form" height=400/>

                            ```bash
                            graphlan.py --format "png" --size 7 input_tree.txt png_image.png
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Wrapper

                            - Description of the user interface
                            - Link between the tool and Galaxy instance
                                - How to invoke the tool
                                - What options to pass
                                - What files the tool will produce as output

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Wrapper

                            <img src="../images/wrapper_layers.png" height=400/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" width=500/>

                            > Command-line utilities to assist in building and publishing Galaxy tools.

                            - [Documentation](https://planemo.readthedocs.io/en/latest/)
                            - [Tutorial](https://planemo.readthedocs.io/en/latest/writing_standalone.html)
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-question-circle" aria-hidden="true"></i> Questions

                            - What is a tool for Galaxy?
                            - How to build a tool/wrapper with the good practices?
                            - How deal with the tool environment?
                            - How to publish a tool?

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-bullseye" aria-hidden="true"></i> Objectives

                            - Discover what is a wrapper and its structure
                            - Use the Planemo utilities to develop a good wrapper
                            - Deal with the dependencies
                            - Write functional tests
                            - Publish tools in the Toolshed

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            <img src="../images/exercise.png" alt="exercise"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### The Goal
                            Developing a wrapper for Seqtk

                            <small>A tool for processing sequence data in FASTA and FASTQ files</small>

                        </script>
                    </section>

                </section>

                <!-- WRAPPER -->
                <section>
                   <section data-markdown>
                       <script type="text/template">
                          ## Wrapper

                       </script>
                   </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Wrapper

                            <img src="../images/wrapper_big_picture_1.png" height=400/>

                            <small>[Galaxy Tool XML File](https://docs.galaxyproject.org/en/latest/dev/schema.html)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Wrapper

                            <img src="../images/wrapper_big_picture_2.png" height=400/>

                            <small>[Galaxy Tool XML File](https://docs.galaxyproject.org/en/latest/dev/schema.html)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Wrapper

                            <img src="../images/wrapper_big_picture_3.png" height=400/>

                            <small>[Galaxy Tool XML File](https://docs.galaxyproject.org/en/latest/dev/schema.html)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `command`
                            #### How to invoke the tool?

                            ```xml
                            <command><![CDATA[
                                graphlan.py
                                    --format $format
                                    #if $dpi
                                        --dpi $dpi
                                    #end if
                                    --size $size
                                    "$input_tree"
                            ]]></command>
                            ```

                            <small>If the script is provided with the wrapper xml</small>
                            ```xml
                            <command><![CDATA[
                                python $__tool_directory__/graphlan.py
                                    --format $format
                                    #if $dpi
                                        --dpi $dpi
                                    #end if
                                    --size $size
                                    "$input_tree"
                            ]]></command>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `command`
                            #### How to pass the parameters?

                            <img src="../images/graphlan_screenshot.png" height=500 alt="input"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` To `command`
                            #### How to pass the parameters?

                            ```xml
                            <inputs>
                                <param name="dpi" type="..." label="..." optional="True"/>

                                <param argument="--size" type="..." value="7" label="..." help="..."/>
                            </inputs>
                            ```

                            #### Directly linked to `<command>`

                            ```xml
                            <command><![CDATA[
                                graphlan.py
                                    ...
                                    #if $dpi
                                        --dpi $dpi
                                    #end if
                                    --size $size
                                    ...
                            ]]></command>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `data`

                            <img src="../images/input_data.png" alt="input data"/>

                            ```xml
                            <param name="..." type="data" format="txt" label="..." help="..." />
                            ```

                            <small>[List of possible formats](https://github.com/galaxyproject/galaxy/blob/dev/config/datatypes_conf.xml.sample)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `integer` | `float`

                            <img src="../images/input_integer.png" alt="input integer"/>

                            ```xml
                            <param name="..." type="integer" value="7" label="..." help="..."/>
                            ```

                            <img src="../images/input_integer_range.png" alt="input integer with range"/>

                            ```xml
                            <param name="..." type="float" min="0" max="10" value="1" label="..."
                            help="..."/>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `text`

                            <img src="../images/input_text.png" alt="input text"/>

                            ```xml
                            <param name="..." type="text" value="" label="..." help="..."/>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `select`

                            <img src="../images/input_select.png" alt="input select"/>

                            ```xml
                            <param name="..." type="select" label="..." help="...">
                                <option value="png" selected="true">PNG</option>
                                <option value="pdf">PDF</option>
                                <option value="ps">PS</option>
                                <option value="eps">EPS</option>
                                <option value="svg">SVG</option>
                            </param>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `select`

                            <img src="../images/input_select_checkboxes.png" alt="input select"/>

                            ```xml
                            <param name="..." type="select" display="radio" label="..." help="...">
                                <option value="min" selected="true">Mininum</option>
                                <option value="mean" >Mean</option>
                                <option value="max" >Max</option>
                                <option value="sum" >Sum</option>
                            </param>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `select`

                            <img src="../images/input_select_multiple.png" alt="input select"/>

                            ```xml
                            <param name="..." type="select" multiple="true" label="..." help="...">
                                <option value="ld" selected="true">Length distribution</option>
                                <option value="gc" selected="true">GC content distribution</option>
                                ...
                            </param>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `boolean`

                            <img src="../images/input_boolean.png" alt="input select"/>

                            ```xml
                            <param name="..." type="boolean" checked="false" truevalue="--log"
                            falsevalue="" label="..."  help="..." />
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `conditional`

                            <img src="../images/input_conditional.png" height="120" alt="input conditional" />

                            ```xml
                            <command interpreter="bash"><![CDATA[
                                #if $input_format.format == "fasta":
                                    phyml_fasta_id_encode_wrapper.sh $input_format.input $output_tree
                                #end if
                                #if $input_format.format == "phylip":
                                    phyml.sh
                                #end if
                            ]]></command>
                            <inputs>
                               <conditional name="input_format">
                                <param type="select" name="format" label="Input format">
                                  <option value="phylip">Phylip</option>
                                  <option value="fasta">Fasta</option>
                                </param>
                                <when value="phylip">
                                  <param format="phylip" name="input" type="data"  />
                                </when>
                                <when value="fasta">
                                  <param format="fasta" name="input" type="data"  />
                                </when>
                              </conditional>
                            </inputs>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `inputs` > `param` > `repeat`

                            <img src="../images/input_repeat.png" height="190" alt="input repeat"/>

                            ```xml
                            <command><![CDATA[
                              #for $i, $s in enumerate( $series )
                                rank_of_series=$i
                                input_path=${s.input}
                                x_colom=${s.xcol}
                                y_colom=${s.ycol}
                              #end for
                            ]]></command>

                            <inputs>
                              <repeat name="series" title="Series">
                                <param name="input" type="data" format="tabular" label="Dataset"/>
                                <param name="xcol" type="data_column" data_ref="input" label="Column for x axis"/>
                                <param name="ycol" type="data_column" data_ref="input" label="Column for y axis"/>
                              </repeat>
                            </inputs>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `outputs`
                            #### What files the tool will produce as output?

                            <img src="../images/output.png" alt="outputs" height=550/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `outputs`

                            ```xml
                            <outputs>
                                <data format="txt" name="tree" label="${tool.name} on ${on_string}:
                                Tree" />
                                <data format="txt" name="annotation" label="${tool.name} on
                                ${on_string}: Annotation" />
                            </outputs>
                            ```

                        </script>
                    </section>


                    <section data-markdown>
                        <script type="text/template">
                            ### `outputs` > `filter`
                            #### Conditional

                            ```xml
                            <outputs>
                                <data format="png" name="png_output_image" label="${tool.name} on
                                ${on_string}: Image">
                                    <filter>format=="png"</filter>
                                </data>
                            </outputs>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `stdio`


                            ```xml
                            <command detect_errors="aggressive"> ... </command>
                            ```
                            or if you need more precision
                            ```
                            <stdio>
                              <exit_code range=":-2" level="warning" description="Low disk space" />
                              <exit_code range="1:" level="fatal"  />
                              <regex match="Error:"  level="fatal" />
                            </stdio>

                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `help`

                            <img src="../images/help.png" alt="help"/>

                            ```xml
                            <help><![CDATA[

                            **What it does**
                            GraPhlAn is a software tool for producing high-quality circular
                            representations of taxonomic and phylogenetic trees. GraPhlAn focuses
                            on concise, integrative, informative, and publication-ready
                            representations of phylogenetically- and taxonomically-driven
                            investigation.

                            For more information, check the `user manual
                            <https://bitbucket.org/nsegata/graphlan/overview>`_.

                            ]]></help>
                            ```

                            <small>[reStructuredText markdown format](http://docutils.sourceforge.net/docs/user/rst/quickref.html)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `citations`

                            <img src="../images/citations.png" alt="citations"/>

                            ```
                            <citations>
                                <citation type="doi">10.1093/bioinformatics/bts611</citation>
                                <citation type="doi">10.1093/nar/gks1219</citation>
                                <citation type="doi">10.1093/nar/gks1005</citation>
                                <citation type="doi">10.1093/bioinformatics/btq461</citation>
                                <citation type="doi">10.1038/nbt.2198</citation>
                            </citations>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            #### Building Galaxy Tools

                            <img src="../images/planemo_tool_building_lint.png" alt="Building tools" class="no-border"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            <img src="../images/exercise.png" alt="exercise"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Before: install Seqtk
                            *No brain needed*
                            ```bash
                            $ planemo conda_init
                            $ export PATH=~/miniconda2/bin:$PATH
                            $ conda config --add channels conda-forge
                            $ conda config --add channels r
                            $ conda config --add channels bioconda
                            $ conda install seqtk
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Before: observe and test Seqtk

                            ```bash
                            $ seqtk

                            $ seqtk seq

                            $ wget https://raw.githubusercontent.com/galaxyproject/galaxy-test-data/master/2.fastq

                            $ cat 2.fastq

                            $ seqtk seq -A 2.fastq > 2.fasta

                            $ cat 2.fasta
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Steps
                            1. Init the wrapper with `planemo tool_init`
                            2. Fill the template
                            3. Test the template with `planemo lint`
                            <small>Don't worry now about "No tests found"</small>
                            4. Display the tool with `planemo serve`
                            5. and try it with a file

                            <aside class="notes">
                                ```bash
                                # 1.
                                planemo tool_init
                                # -> hands_on-tool_integration/seqtk/seqtk_seq_step1.xml
                                # 2. -> hands_on-tool_integration/seqtk/seqtk_seq_step2.xml
                                # 3.
                                planemo lint # seqtk_seq_step2.xml
                                # 4.
                                planemo serve # seqtk_seq_step2.xml
                                ```
                            </aside>
                        </script>
                    </section>

                </section>

                <!-- FUNCTIONAL TEST -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## Functional tests
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Functional tests

                            - Functional testing is a quality assurance (QA) process.
                            - The tests confort developers and users that the tools can run across different servers/architectures. And that the last modifications didn't break the different features.
                            - Tools are tested by feeding them input(s) and parameter(s) and examining the output(s) (typically a diff)

                          </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Functional tests
                            #### Good practices

                            1. Build tests using inputs/parameters/outputs
                            2. Run the tests --> Failed --> Coffee
                            3. Develop
                            4. Run the tests --> Failed  --> Coffee
                            5. Develop
                            6. Run the tests --> Succeed --> Beer

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `tests`

                            ```xml
                            <tests>
                                <test>
                                    <param name="input_tree" value="input_tree.txt"/>
                                    <param name="format" value="png"/>
                                    <param name="dpi" value="100"/>
                                    <param name="size" value="7"/>
                                    <param name="pad" value="2"/>
                                    <output name="png_output_image" file="png_image.png" />
                                </test>
                            </tests>
                            ```

                            <small>`input_tree.txt` and `png_image.png` in `test-data` repository</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `tests`
                            #### Comparison Methods

                            ```xml
                            <output name="out_file1" file="cf_maf2fasta_concat.dat" ftype="fasta"/>
                            ```

                            ```xml
                            <test>
                                <output name="out_file1">
                                    <assert_contents>
                                        <has_text text="chr7" />
                                        <not_has_text text="chr8" />
                                        <has_text_matching expression="1274\d+53" />
                                        <has_line_matching expression=".*\s+127489808\s+127494553" />
                                        <!-- &#009; is XML escape code for tab -->
                                        <has_line line="chr7&#009;127471195&#009;127489808" />
                                        <has_n_columns n="3" />
                                    </assert_contents>
                                </output>
                            </test>
                            ```

                            <small>[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `tests`
                            #### Comparison Methods on the output itself

                            ```xml
                            <test>
                                <output ...
                                    md5
                                    compare
                                        diff
                                        sim_size
                                    lines_diff
                                ... />

                            ```

                            <small>[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test|output)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                          ### `tests`
                          #### Comparison Methods within the output

                          ```xml
                          <test>
                              <output>
                                  <assert_contents>
                                      <has_text />
                                      <not_has_text />
                                      <has_text_matching />
                                      <has_text />
                                      <...>
                          ```
                          <small>[Complete information](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|tests|test|output|assert_contents)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Wrapper directory tree

                            ```
                            graphlan/
                            ├── graphlan.xml
                            ├── graphlan.py
                            ├── test-data/
                                ├── input_tree.txt
                                ├── png_image.png

                            ```

                        </script>
                    </section>


                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            #### Test Galaxy Tools

                            <img src="../images/planemo_tool_building_test.png" alt="Testing tools" class="no-border"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            <img src="../images/exercise.png" alt="exercise"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### The Goal
                            Add functional test to the Seqtk wrapper for Seqtk

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Steps
                            1. Fill a `<test>` tag using the file you generated previously with `seqtk`
                            2. Use `planemo test` ... until everything is green

                            <aside class="notes">
                                ```bash
                                # 1. -> hands_on-tool_integration/seqtk/seqtk_seq_step3.xml
                                # 2.
                                planemo test # seqtk_seq_step3.xml
                                ```
                            </aside>

                        </script>
                    </section>
                </section>

                <!-- DEPENDENCIES -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## Dependencies
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dependencies
                            #### How Galaxy will deal with dependencies?

                            <img src="../images/galaxy_instance.png" alt="Galaxy instance" height=500/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dependencies
                            #### Embedded tool

                            ```
                            graphlan/
                            ├── graphlan.xml
                            ├── graphlan.py

                            ```

                            ```
                            <command><![CDATA[
                                $__tool_directory__/graphlan.py
                                    ...
                            ]]></command>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dependencies
                            #### External tool

                            Definition of requirements in `.xml` file

                            ```
                            <requirements>
                                <requirement type="package" version="1.66">biopython</requirement>
                                <requirement type="package" version="1.0.0">graphlan</requirement>
                            </requirements>
                            ```

                            Local installation using ~~tool_dependencies.xml~~

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dependencies
                            #### External tool

                            Definition of requirements in `.xml` file

                            ```
                            <requirements>
                                <requirement type="package" version="1.66">biopython</requirement>
                                <requirement type="package" version="1.0.0">graphlan</requirement>
                            </requirements>
                            ```

                            Local installation using `conda` package

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border" />
                            > Package, dependency and environment management

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border"  height=50 />

                            - Based on recipes describing how to install the software which are then builded for their distribution
                            - At installation, no compilation, executables come with their dependencies, libraries...
                            - Not restricted to Galaxy

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border"  height=25 />
                            ### Miniconda

                            <img src="../images/miniconda_vs_anaconda.png" alt="Miniconda" class="no-border" height="450"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            ### Test Galaxy Tools using Conda packages

                            <img src="../images/planemo_tool_building.png" height="500" alt="Testing tools" class="no-border" />

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            <img src="../images/exercise.png" alt="exercise"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### The Goal
                            Use Conda package installation during the functional test of the wrapper for Seqtk

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Before
                            1. `conda uninstall seqtk`

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Steps
                            1. Fill a `<requirement>` tag to target a `seqtk` conda package
                            2. Use `planemo conda_install .` to install tools and set an environment
                            3. Use `planemo test --conda_dependency_resolution`

                            <aside class="notes">
                                ```bash
                                # 1. -> hands_on-tool_integration/seqtk/seqtk_seq_step4.xml
                                # 2.
                                planemo conda_install .
                                # 3.
                                planemo test --conda_dependency_resolution # seqtk_seq_step4.xml
                                ```
                            </aside>
                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border"  height=25 />
                            ### Write a Conda package
                            #### Files

                            - `meta.yaml`: contains all the metadata in the recipe.
                            - `build.sh`: the Unix script that installs the files

                            <small>[Documentation about how to build a conda packages from scratch](http://conda.pydata.org/docs/build_tutorials/pkgs2.html)</small>
                            <small>[Bioconda guidelines](https://github.com/bioconda/bioconda-recipes/blob/master/GUIDELINES.md)

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border"  height=25 />
                            ### Write a Conda package

                            <img src="../images/conda_files.png" alt="meta.yaml and build.sh" />

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border"  height=25 />
                            ### Write a Conda package
                            #### skeleton

                            - Python<br/>
                            `conda skeleton pypi <packagename>`
                            - R<br/>
                            `conda skeleton cran <packagename>`
                            - Bioconductor<br/>
                            `bioconductor_skeleton.py <packagename>`
                            - Perl<br/>
                            `conda skeleton cpan <packagename>`

                            <small>[Building conda packages with conda skeleton](http://conda.pydata.org/docs/build_tutorials/pkgs.html)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/conda_logo.png" class="no-border"  height=25 />
                            ### Write a Conda package
                            #### `conda build` and test the package

                            ```bash
                            $ conda install conda-build

                            $ conda build .

                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/bioconda_logo.png" class="no-border"  height=50 />

                            - A channel dedicated to bioinformatics packages
                            - Open to [contribution](https://github.com/bioconda/bioconda-recipes/blob/master/CONTRIBUTING.md)
                            - GitHub repository: https://github.com/bioconda/bioconda-recipes

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            <img src="../images/exercise.png" alt="exercise"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### The Goal
                            Build a conda package for the suite seqtk

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### <i class="fa fa-pencil" aria-hidden="true"></i> Hands-on

                            #### Steps
                            1. RTFM: https://github.com/bioconda/bioconda-recipes/blob/master/GUIDELINES.md#cc
                            2. https://github.com/lh3/seqtk/
                            3. Complete a `meta.yaml` and a `build.sh`
                            4. Test the build with `conda build .`

                            <aside class="notes">
                                meta.yaml
                                ```bash
                                package:
                                  name: seqtk
                                  version: 1.2

                                source:
                                  fn: v1.2.tar.gz
                                  url: https://github.com/lh3/seqtk/archive/v1.2.tar.gz

                                build:
                                  number: 0
                                  skip: False


                                requirements:
                                  build:
                                    - gcc   # [not osx]
                                    - llvm  # [osx]
                                    - zlib
                                  run:
                                    - zlib


                                about:
                                  home: https://github.com/lh3/seqtk
                                  license: MIT License
                                  summary: Seqtk is a fast and lightweight tool for processing sequences in the FASTA or FASTQ format

                                test:
                                  commands:
                                - seqtk seq
                                ```

                                build.sh
                                ```bash
                                #!/bin/bash

                                export C_INCLUDE_PATH=${PREFIX}/include
                                export LIBRARY_PATH=${PREFIX}/lib

                                make all
                                mkdir -p $PREFIX/bin
                                cp -f seqtk $PREFIX/bin/
                                ```
                            </aside>

                        </script>
                    </section>

                </section>


                <!-- PUBLISHING TOOLSHED -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## Publishing on Toolshed

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Publishing on ToolShed

                            Metadata description: `.shed.yml` file

                            ```
                            categories:
                              - Metagenomics
                              - Graphics
                              - Phylogenetics
                            description: "GraPhlAn is a software tool for producing high-quality
                            circular representations of taxonomic and phylogenetic trees"
                            homepage_url: "https://bitbucket.org/nsegata/graphlan/overview"
                            name: graphlan
                            owner: bebatut
                            remote_repository_url: "https://github.com/ASaiM/galaxytools/tree/master/tools/graphlan/"
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Publishing on ToolShed

                            Metadata description: `.shed.yml` file

                            ```
                            graphlan/
                            ├── graphlan.xml
                            ├── test_data/
                                ├── input_tree.txt
                                ├── png_image.png
                            ├── .shed.yml
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            #### Publishing to the ToolShed

                            <img src="../images/planemo_toolshed_publishing.png" alt="Publishing on ToolShed" class="no-border"/>

                        </script>
                    </section>
                </section>


                <!-- PLANEMO -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" width=750/>

                            > Command-line utilities to assist in building and publishing Galaxy tools.

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>

                            <img src="../images/big_picture.png" alt="The big picture" class="no-border" height="600" />

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            #### Checklist (1/3)

                            <p style="text-align: left;">
                              <input type="checkbox"/> Create a dedicated repository <br/>
                              <input type="checkbox"/> Create the main `.xml` file with <br/>
                              ```planemo create``` <br/>
                              <input type="checkbox"/> Fill the `.xml` file <br/>
                              <input type="checkbox"/> Test the wrapper structure with <br/>
                              ```planemo lint```<br/>
                            </p>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            #### Checklist (2/3)

                            <p style="text-align: left;">
                              <input type="checkbox"/> Check if the dependencies are available <br/>
                              <input type="checkbox"/> If not, create some conda packages<br/>
                              <input type="checkbox"/> Add functional tests <br/>
                              <input type="checkbox"/> Run functional tests <br/>
                              ```planemo test```<br/>
                            </p>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template" >
                            <img src="../images/planemo-logo.png" alt="Planemo" class="no-border" height=25/>
                            #### Checklist (3/3)

                            <p style="text-align: left;">
                              <input type="checkbox"/> Initiate the `.shed.yml` file with <br/>
                              ```planemo shed_init``` <br/>
                              <input type="checkbox"/> Create the repository on ToolShed with <br/>
                              ```planemo shed_create```<br/>
                              <input type="checkbox"/> Install and test the wrapper on a Galaxy instance <br/>
                              <input type="checkbox"/> Improve the wrapper and push modification on toolshed <br/>
                              ```planemo shed_update``` <br/>
                            </p>

                        </script>
                    </section>
                </section>


                <!-- GITHUB + TRAVIS.CI + PLANEMO + CONDA -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            <img src="../images/github_logo.png" alt="github" height="150"/><br/>
                            <img src="../images/travis_ci_logo.png" alt="travis ci" height="150"/><br/>
                            <img src="../images/planemo-logo.png" alt="planemo" height="100"/><br/>
                            <img src="../../shared/images/conda_logo.png" alt="conda" height="90"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### GitHub side

                            <img src="../images/github_travis_github.png" alt="Travis in GitHub"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Travis CI side

                            <img src="../images/github_travis_travis.png" alt="GitHub in Travis"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            <img src="../../shared/images/ansible_logo.png" alt="ansible" height="200"/><br/>
                            <img src="../../shared/images/conda_logo.png" alt="conda" height="90"/><br/>
                            <img src="../../shared/images/docker_logo.png" alt="docker" height="120"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Docker Galaxy Flavors

                            <img src="../images/galaxydocker_logo.png" alt="GitHub in Travis"/>

                            <small>[Docker Galaxy Stable](https://github.com/bgruening/docker-galaxy-stable)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Docker Galaxy Flavors
                            #### Dockerfile

                            ```
                            # Galaxy - My own flavour
                            # VERSION       0.1

                            FROM quay.io/bgruening/galaxy:16.07

                            MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com

                            ENV GALAXY_CONFIG_BRAND deepTools

                            # Adding the tool definitions to the container
                            ADD my_tool_list.yml $GALAXY_ROOT/my_tool_list.yml

                            # Install deepTools
                            RUN install-tools $GALAXY_ROOT/my_tool_list.yml
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Docker Galaxy Flavors
                            #### tool_list.yml

                            ```bash
                            api_key: admin
                            galaxy_instance: http://127.0.0.1:8080/
                            tools:
                            - name: fastqc
                              owner: devteam
                              tool_panel_section_id: cshl_library_information
                              tool_shed_url: https://toolshed.g2.bx.psu.edu
                              revisions:
                              - '8c650f7f76e9'  # v0.62
                              - 'd2cf2c0c8a11'  # v0.63
                            - name: suite_deeptools
                              owner: bgruening
                              tool_panel_section_label: 'deepTools'
                              tool_shed_url: https://toolshed.g2.bx.psu.edu
                              revisions:
                             - '7f5562625ae2'
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Docker Galaxy Flavors
                            #### RUN

                            ```bash
                            docker run -d -p 8080:80 galaxy-deeptools
                            ```

                        </script>
                    </section>

                </section>


                <!-- ADVANCED -->
                <section>
                    <section data-markdown>
                        <script type="text/template">
                            ## Advanced features

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Other tags

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `configfiles`

                            <div align="left">
                              Like `command` allow cheetah code<br/>
                              <br/>
                              <ul>
                                <li>Create a config file</li>
                                <li>Create a whole script or a module</li>
                              </ul>
                            </div>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `configfiles`

                            ```xml
                            <command><![CDATA[ mb $script_nexus  ]]></command>

                            <configfiles>
                              <configfile name="script_nexus"><![CDATA[
                            set autoclose = yes;
                            execute $input_data;
                            #if str( $data_type.type ) == "nuc“
                                      lset nst=$data_type.lset_params.lset_Nst;
                            #end if
                            mcmcp ngen=$mcmcp_ngen;
                            mcmc;
                            quit
                              ]]></configfile>
                            </configfiles>

                            ```

                            ```bash
                            set autoclose = yes;
                            execute dataset_42.dat;
                            lset nst=2 ;
                            mcmcp ngen=100000;
                            mcmc;
                            quit
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `command` > Reserved Variables

                            ```xml
                            <command><![CDATA[
                              ${input_foobar.ext}

                              --search_pfam "${ filter( lambda x: str( x[0] ) == str( $search.pfam_db ), $__app__.tool_data_tables[ 'pfam_databases' ].get_fields() )[0][-1] }"

                              library $__app__.config.user_library_import_dir/$__user_email__/$inputs.library

                              #if isinstance( $input1.datatype, $__app__.datatypes_registry.get_datatype_by_extension('gff').__class__):
                            ]]></command>
                            ```
                            <small>[Reserved Variables List](https://docs.galaxyproject.org/en/latest/dev/schema.html#tool|command)</small>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `macro`

                            <img src="../images/macro.png" alt="macro"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dataset collection

                            <div align="left">
                              To group related datasets, and treat them as a single unit<br/>
                              <br/>
                              <ul>
                                <li>`list`: a simple list of files</li>
                                <li>`paired`: a pair of files, `forward` and `reverse` for NGS</li>
                                <li>composite: e.g. `list:paired` for a list of dataset pairs</li>
                              </ul>
                            </div>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dataset collection > input

                            <div align="left">
                                Accept both datasets and collections:
                            </div>
                            ```xml
                            <param type="data" multiple="true" name="inputs" label="Input BAM(s)" format="bam" />
                            ```

                            <div align="left">
                                Accept only collections:
                            </div>
                            ```xml
                            <param type="data_collection" collection_type="list|paired|list:paired|..." name="inputs" label="Input BAM(s)" format="bam" />
                            ```

                            <div align="left">
                                In the command:
                            </div>
                            ```xml
                            <command><![CDATA[
                                ...
                                    #for $input in $inputs
                                        --input "$input"
                                        --sample_name "$input.element_identifier"
                                    #end for
                            ]]></command>
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Dataset collection > output

                            <p align="left">A single paired collection:</p>
                            ```xml
                            <collection name="paired_output" type="paired" label="Split Pair">
                                <data name="forward" format="txt" />
                                <data name="reverse" format_source="input1" from_work_dir="reverse.txt" />
                            </collection>
                            ```

                            <p align="left">Unknown number of files:</p>
                            ```xml
                            <collection name="output" type="list" label="Unknown number of files">
                                <discover_datasets pattern="__name_and_ext__" directory="outputs" />
                            </collection>
                            ```

                            <div align="left">
                                `__name_and_ext__`: a predefined regexp,
                                catches the dataset identifier + the format
                            </div>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Some Config

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `job_conf.xml`

                            <img src="../images/job_conf.xml_1.png" alt="job_conf.xml" height="500"/>
                            ```bash
                            blastn -query foo_bar -num_threads 8
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `job_conf.xml`

                            <img src="../images/job_conf.xml_2.png" alt="job_conf.xml" height="500"/>
                            ```bash
                            blastn -query foo_bar -num_threads 4
                            ```

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### `tool_data_table_conf.xml`

                            <img src="../images/tool_data_table_conf.xml.png" alt="tool_data_table_conf.xml"/>

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Datatypes

                        </script>
                    </section>

                    <section data-markdown>
                        <script type="text/template">
                            ### Datatypes

                            - Every Galaxy dataset is associated with a datatype
                            - Datatype can be detected or user specified
                            - Gain of usability

                            <small>[Documentation: Adding Datatypes](https://wiki.galaxyproject.org/Admin/Datatypes/Adding%20Datatypes)</small>
                        </script>
                    </section>

                </section>

                <!-- CONCLUSION -->
                <section>
                  <section data-markdown>
                      <script type="text/template">
                          ## Conclusion

                      </script>
                  </section>

                  <section data-markdown>
                      <script type="text/template">
                          ### IUC: Intergalactic Utilities Commission

                          - Lobbyist :)
                          - Host orphan tools: [GitHub](https://github.com/galaxyproject/tools-iuc)
                          - [IUC-standards](http://galaxy-iuc-standards.readthedocs.io/)

                      </script>
                  </section>

                  <section data-markdown>
                      <script type="text/template">
                          ### <i class="fa fa-exclamation-circle" aria-hidden="true"></i> Key points

                          - Use Planemo
                          - Use Conda
                          - Use GitHub
                          - Use Travis
                          - No more excuse to develop crappy Galaxy tools

                      </script>
                  </section>
                </section>


      </div>

    </div>

    <script type="text/javascript" src="../../assets/reveal.js/lib/js/head.min.js"></script>
        <script type="text/javascript" src="../../assets/reveal.js/js/reveal.js"></script>
        <script type="text/javascript">
            Reveal.initialize({
                slideNumber: !window.location.search.match(/print-pdf/gi),
                history: true,

                theme: Reveal.getQueryHash().theme,
                transition: Reveal.getQueryHash().transition || 'concave',

                dependencies: [
                    { src: '../../assets/reveal.js/lib/js/classList.js', condition: function() { return !document.body.classList; } },
                    { src: '../../assets/reveal.js/plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
                    { src: '../../assets/reveal.js/plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
                    { src: '../../assets/js/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
                    { src: '../../assets/reveal.js/plugin/notes/notes.js', async: true },
                ]
            });
        </script>

  </body>
</html>
