<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Calling variants in diploid systems</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:title" content="Galaxy Training: Calling variants in diploid systems" />
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Variant Analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fdip%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fdip%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fdip%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fdip%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fdip%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fdip%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/variant-analysis/tutorials/dip/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Calling variants in diploid systems",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
  },
  "courseCode": "variant-analysis / dip / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Calling variants in diploid systems' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/tutorials/dip/tutorial.html",
  "timeRequired": "PT3H",
  "description": "The questions this  addresses are:\n - How to find variable sites in diploid genomes?\n\n\\nThe objectives are:\n - Identification of genetic variations using variant calling\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Calling variants in diploid systems</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nekrut" alt="Anton Nekrutenko">Anton Nekrutenko</a>, <a href="/training-material/hall-of-fame/nsoranzo/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nsoranzo" alt="Nicola Soranzo">Nicola Soranzo</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to find variable sites in diploid genomes?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Identification of genetic variations using variant calling</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 3 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/variant-analysis/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.60520">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/dip/workflows/" title="Workflows" alt="Calling variants in diploid systems workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/dip/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <p>Today we hear a lot about personalized medicine. Yet the <em>personalization</em> is defined by the genetic make up of the individual. In this tutorial we will discuss how this information can be uncovered from the genomic sequencing data.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#calling-variants" id="markdown-toc-calling-variants">Calling variants</a>    <ol>
      <li><a href="#how-does-snp-calling-and-genotyping-work" id="markdown-toc-how-does-snp-calling-and-genotyping-work">How does SNP calling and genotyping work?</a></li>
      <li><a href="#calling-with-freebayes" id="markdown-toc-calling-with-freebayes">Calling with FreeBayes</a></li>
    </ol>
  </li>
  <li><a href="#annotating-variants-with-snpeff" id="markdown-toc-annotating-variants-with-snpeff">Annotating variants with SnpEff</a></li>
  <li><a href="#manipulating-variation-data-with-gemini" id="markdown-toc-manipulating-variation-data-with-gemini">Manipulating variation data with GEMINI</a>    <ol>
      <li><a href="#loading-data-into-gemini" id="markdown-toc-loading-data-into-gemini">Loading data into GEMINI</a></li>
      <li><a href="#querying-the-gemini-database" id="markdown-toc-querying-the-gemini-database">Querying the GEMINI database</a></li>
      <li><a href="#querying-genotypes" id="markdown-toc-querying-genotypes">Querying genotypes</a></li>
      <li><a href="#using-wildcards" id="markdown-toc-using-wildcards">Using wildcards</a></li>
      <li><a href="#going-further" id="markdown-toc-going-further">Going further</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="calling-variants">Calling variants</h1>

<p>Variant calling is a complex field that was significantly propelled by advances in <span class="notranslate">DNA</span> sequencing and efforts of large scientific consortia such as the <a href="http://www.1000genomes.org">1000 Genomes</a>. Here we summarize basic ideas central to Genotype and Variant calling. First, let‚Äôs contrast the two things although they often go together:</p>

<ul>
  <li><strong>Variant calling</strong> - identification of positions where the sequenced sample is different from the reference sequence (or <a href="https://github.com/vgteam/vg">reference genome graph</a>)</li>
  <li><strong>Genotype calling</strong> - identifying individual‚Äôs genotype at variable sites.</li>
</ul>

<p>A typical <span class="notranslate">workflow</span> for variation discovery involves the following steps (e.g., see Nielsen et al. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3593722/">2011</a>):</p>

<ol>
  <li><span class="notranslate">Mapping</span> <span class="notranslate">reads</span> against the reference genome</li>
  <li>Thresholding BAM datasets by, for example, retaining paired, properly mapped <span class="notranslate">reads</span></li>
  <li>Performing quality score recalibration</li>
  <li>Performing realignment</li>
  <li>Performing variant calling/genotype assignment</li>
  <li>Performing filtering and genotype quality score recalibration</li>
  <li>Annotating variants and performing <span class="notranslate">downstream</span> analyses</li>
</ol>

<p>However, continuing evolution of variant detection methods has made some of these steps obsolete. For instance, omitting quality score recalibration and re-alignment (steps 3 and 4 above) when using haplotype-aware variant callers such as <a href="https://github.com/ekg/freebayes">FreeBayes</a> does not have an effect on the resulting calls (see Brad Chapman‚Äôs methodological comparisons at <a href="https://bit.ly/1S9kFJN">bcbio</a>). Before going forward with an actual genotype calling in <span class="notranslate">Galaxy</span> let‚Äôs take a look as some basic ideas behind modern variant callers.</p>

<h2 id="how-does-snp-calling-and-genotyping-work">How does SNP calling and genotyping work?</h2>

<p>Consider a set of sequencing <span class="notranslate">reads</span> derived from a diploid individual:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>REFERENCE: atcatgacggcaGtagcatat
--------------------------------
READ1:     atcatgacggcaGtagcatat
READ2:         tgacggcaGtagcatat
READ3:     atcatgacggcaAtagca
READ4:            cggcaGtagcatat
READ5:     atcatgacggcaGtagc
</code></pre></div></div>

<p>The capitalized position contains a G ‚Üí A <a href="https://en.wikipedia.org/wiki/Transition_(genetics)">transition</a>. So, in principle this can be a heterozygous site with two alleles <strong>G</strong> and <strong>A</strong>. A commonly used na√Øve procedure would define a site as <em>heterozygous</em> if there is a non-reference allele with frequency between 20% and 80%. In this case <strong>A</strong> is present in 1/5 or 20% of the cases, so we can say that this is a heterozygous site. Yet it is only represented by a single read and thus is hardly reliable. Here are some of the possibilities that would explain this <em>variant</em>. It can be:</p>

<ul>
  <li>a true variant</li>
  <li>an experimental artifact, e.g. a library preparation error</li>
  <li>a base calling error</li>
  <li>an analysis error, e.g. a misalignment (though unlikely in the above example)</li>
</ul>

<p>The modern variant callers attempt to assign a reliability estimate for each genotype call. This is done using Bayes reasoning (for a great visual explanation see <a href="https://oscarbonilla.com/2009/05/visualizing-bayes-theorem/">blog</a> by Oscar Bonilla). Here we present a SNP-relevant ‚Äútranslation‚Äù on this explanation (with inspiration from <a href="https://github.com/ekg">Erik Garrison</a>).</p>

<p>Suppose in a population you have \(A\) individuals (not to be confused with nucleotide <strong>A</strong>; in this case \(A\) is a number of individuals) with a variant. You are performing re-sequencing and observe a variant in \(B\) (again, a number) of your sequencing <span class="notranslate">reads</span>. We want to estimate the probability of having the real polymorphism in the population given our observations in sequencing <span class="notranslate">reads</span>. The logic is as follows:</p>

<ul>
  <li>The probability of having polymorphism <em>A</em> in the population is \(P(A) = \lvert A \rvert / \lvert U \rvert\)</li>
  <li>The probability of seeing a variant given our identification approach (i.e., sequencing) is \(P(B) = \lvert B \rvert / \lvert U \rvert\)</li>
</ul>

<p>Now, the probability of having a variant and it being observed in our sequencing data is the overlap between \(A\) and \(B\) sets: \(P(AB) = \lvert AB \rvert / \lvert U \rvert\). This is presented graphically below:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: center">Polymorphisms</th>
      <th style="text-align: center">Variant Calls</th>
      <th style="text-align: center">Polymorphisms and Variant Calls</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: center"><img src="../../images/pA.png" alt="P(A) polymorphism" /></td>
      <td style="text-align: center"><img src="../../images/pB.png" alt="P(B) variant calls" /></td>
      <td style="text-align: center"><img src="../../images/pAB.png" alt="P(AB) polymorphism and variant calls" /></td>
    </tr>
    <tr>
      <td style="text-align: center">\(P(A)\)</td>
      <td style="text-align: center">\(P(B)\)</td>
      <td style="text-align: center">\(P(AB)\)</td>
    </tr>
  </tbody>
</table>

<p>Now we can ask the following question: <em>What is the probability of a having a real polymorphism \(A\) given our observation of variants in <span class="notranslate">reads</span> \(B\)?</em> In other words, <em>what is the probability of \(A\) given \(B\)?</em> Or, as stated in the original <a href="https://oscarbonilla.com/2009/05/visualizing-bayes-theorem/">blog</a>: ‚Äú<em>given that we are in region \(B\), what is the probability that we are in the region \(AB\)?</em>‚Äù</p>

\[P(A \vert B) = \frac{\lvert AB \rvert}{\lvert B \rvert}\]

<p>Dividing both numerator and denominator by \(\lvert U \rvert\):</p>

\[P(A \vert B) = \frac{\frac{\lvert AB \rvert}{\lvert U \rvert}}{\frac{\lvert B \rvert}{\lvert U \rvert}}\]

<p>Because we know that \(P(AB) = \frac{\lvert AB \rvert}{\lvert U \rvert}\) and \(P(B) = \frac{\lvert B \rvert}{\lvert U \rvert}\), we can rewrite the equation above as:</p>

\[P(A \vert B) = \frac{P(AB)}{P(B)}\]

<p>Now, let‚Äôs ask an opposite question. Given a true polymorphism \(A\), what are the chances that we do detect it (i.e. find ourselves in \(AB\))? It will be:</p>

\[P(B \vert A) = \frac{P(AB)}{P(A)}\]

<p>So, because we know that \(P(A \vert B) = \frac{P(AB)}{P(B)}\) and we just reasoned that \(P(B \vert A) = \frac{P(AB)}{P(A)}\), we can say that \(P(A \vert B)P(B) = P(B \vert A)P(A)\), leading us to the Bayes formula:</p>

\[P(A \vert B) = \frac{P(B \vert A)P(A)}{P(B)}\]

<p>Translating this into ‚Äúgenomics terms‚Äù, the probability of having a genotype \(G\) given sequencing <span class="notranslate">reads</span> \(S\) is: \(P(G \vert S) = \frac{P(S \vert G)P(G)}{P(S)}\). Because in a given calculation of \(P(G \vert S)\) the <span class="notranslate">reads</span> are fixed, we can re-write the Bayes formula in the following way:</p>

\[P(G \vert S) \sim P(S \vert G)P(G)\]

<p>with \(P(S)\) becoming a constant. This leaves us with the need to estimate two things:</p>

<ol>
  <li>\(P(S \vert G)\), the data likelihood</li>
  <li>\(P(G)\), the prior probability for the variant.</li>
</ol>

<p>In the simplest case we can estimate these as follows:</p>

<h3 id="ps-vert-g">\(P(S \vert G)\)</h3>

<p>Suppose \(S_i\) is a base in read \(i\) corresponding to a genome position with genotype \(G\). The probability of seeing \(S_i\) given \(G\), or \(P(S_i \vert G)\), is given by the quality score of \(S_i\) (the quality scores are given by base calling software and reported as <a href="https://en.wikipedia.org/wiki/Phred_quality_score">phred scores</a>). Thus the genotype likelihood \(P(S \vert G)\) is the product of \(P(S_i \vert G)\) over all \(i\). In reality however there are many other sources of uncertainty (in addition to base qualities) that are incorporated in the calculation of data likelihoods including NGS technology-related issues, dependency of error rates on substitution type (e.g., transitions versus transversions), sequencing context etc‚Ä¶</p>

<h3 id="pg---a-single-sample-case">\(P(G)\) - a single sample case</h3>

<p>One can assign an equal probability to all possible genotypes, or to source this information based on previously obtained knowledge containing in a database, such as <a href="https://www.ncbi.nlm.nih.gov/SNP/">dbSNP</a>. In this case (as exemplified in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3593722/">Nielsen et al. 2011</a>) we may, for instance, have a site with a <strong>G/T</strong> polymorphism and genotypes <strong>GG</strong>, <strong>TT</strong>, and <strong>GT</strong> having frequencies of 0.45, 0.45, 0.09, respectively. We will use these values as priors.</p>

<h3 id="pg---a-multi-sample-case">\(P(G)\) - a multi-sample case</h3>

<p>Genotype calling reliability can be significantly improved when analyzing multiple samples jointly. In this case genotype frequencies can be inferred from allele frequencies using Hardy-Weinberg equilibrium (<a href="https://en.wikipedia.org/wiki/Hardy%E2%80%93Weinberg_principle">HWE</a>). The following example (again from <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3593722/">Nielsen et al. 2011</a>) illustrates this idea: suppose you are calling genotypes for a single individual using a combination of multiple samples. There are two genotypes, <strong>AT</strong> and <strong>AA</strong>, with equally large genotype likelihoods. If, however, in our collection of multiple samples the frequency of <strong>A</strong> is 1% (\(p = 0.01\); \(q = 1 - p = 0.99\)), then from the HWE we have:</p>

<table>
  <thead>
    <tr>
      <th><strong>AA</strong> (\(p^2\))</th>
      <th><strong>AT</strong> (\(2pq\))</th>
      <th><strong>TT</strong> (\(q^2\))</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0.0001</td>
      <td>0.0198</td>
      <td>0.9801</td>
    </tr>
  </tbody>
</table>

<p>This makes it highly unlikely that <strong>AA</strong> is a true genotype of this individual.</p>

<h2 id="calling-with-freebayes">Calling with FreeBayes</h2>

<p><a href="https://github.com/ekg/freebayes">FreeBayes</a> is an open source variant caller that has been battle-tested by the 1000 Genomes community and is extensively used today (also see <a href="https://bcbio.wordpress.com/">bcbio</a>). It has a number of features that simplify variant discovery workflows. These include (from FreeBayes github page):</p>

<ul>
  <li><strong>Indel realignment is accomplished inte<span class="notranslate">rna</span>lly</strong> using a read-independent method, and issues resulting from discordant alignments are dramatically reduced through the direct detection of haplotypes</li>
  <li><strong>The need for base quality recalibration is avoided</strong> through the direct detection of haplotypes. Sequencing platform errors tend to cluster (e.g. at the ends of <span class="notranslate">reads</span>), and generate unique, non-repeating haplotypes at a given locus</li>
  <li><strong>Variant quality recalibration is avoided</strong> by incorporating a number of metrics, such as read placement bias and allele balance, directly into the Bayesian model</li>
  <li><strong>Ability to incorporate non-diploid cases</strong> such as pooled datasets or data from polyploid samples.</li>
</ul>

<p>Freebayes is a <em>haplotype-based</em> variant caller. This implies that instead of looking at an individual positions within an alignment of <span class="notranslate">reads</span> to the reference genome, it looks at a haplotype window, length of which is dynamically determined (see section 3.2. in <a href="https://arxiv.org/pdf/1207.3907v2.pdf">FreeBayes manuscript</a>):</p>

<figure id="figure-1"><img src="../../images/freebayes.png" alt="Haplotype-based calling" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Looking at a haplotype window makes misalignments tolerable. In this case a low complexity poly(A) stretch is misaligned. As a result looking at individual positions will result in calling multiple spurious variants. In the case of FreeBayes looking at a haplotype identifies two alleles (this is a diploid example) A(7) and A(6), while A(8) is likely an error. Image by Erik Garrison</figcaption></figure>

<h3 id="the-data">The data</h3>

<p>In this example we will perform variant calling and annotation using <a href="https://jimb.stanford.edu/giab/">genome in the bottle data</a>. Specifically, we will use Ashkenazim Father-Mother-Son trio data from the Personal Genome Project:</p>

<ul>
  <li>HG002 - NA24385 - huAA53E0 (son)</li>
  <li>HG003 - NA24149 - hu6E4515 (father)</li>
  <li>HG004 - NA24143 - hu8E87A9 (mother)</li>
</ul>

<p>Yet for a quick tutorial these datasets are way too big, so we created a <a href="https://doi.org/10.5281/zenodo.60520">downsampled dataset</a>. This dataset was produced by <span class="notranslate">mapping</span> the trio <span class="notranslate">reads</span> against the <code class="language-plaintext highlighter-rouge">hg19</code> version of the human genome, merging the resulting bam files together (we use readgroups to label individual <span class="notranslate">reads</span> so they can be traced to each of the original individuals), and restricting alignments to a small portion of chromosome 19 containing the <a href="https://www.ncbi.nlm.nih.gov/gene?cmd=Retrieve&amp;dopt=Graphics&amp;list_uids=5442"><em>POLRMT</em></a> gene.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this variant calling exercise</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>Import the files named <code class="language-plaintext highlighter-rouge">GIAB-Ashkenazim-Trio.txt</code> (tabular format) and <code class="language-plaintext highlighter-rouge">GIAB-Ashkenazim-Trio-hg19</code> (BAM format) from <a href="https://doi.org/10.5281/zenodo.60520">Zenodo</a> or a data library:</li>
    <li>Specify the used genome for <span class="notranslate">mapping</span>:
      <ol>
        <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the BAM dataset to edit its attributes</li>
        <li>Select <code class="language-plaintext highlighter-rouge">Human Feb 2009</code> on <strong>Database/Build</strong></li>
        <li>Click the <strong>Save</strong> button</li>
      </ol>
    </li>
  </ol>
</blockquote>

<h3 id="generating-and-post-processing-freebayes-calls">Generating and post-processing FreeBayes calls</h3>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generating-freebayes-calls"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generating FreeBayes calls</h3>

  <ol>
    <li><strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">locally cached</code></li>
        <li><em>‚ÄúBAM dataset‚Äù</em>: the uploaded <code class="language-plaintext highlighter-rouge">GIAB-Ashkenazim-Trio-hg19</code> BAM dataset</li>
        <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human (Homo sapiens): hg19</code></li>
        <li><em>‚ÄúChoose parameter selection level‚Äù</em>: <code class="language-plaintext highlighter-rouge">5. Full list of options</code></li>
        <li><em>‚ÄúAlgorithmic features‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set algorithmic features</code></li>
        <li><em>‚ÄúCalculate the marginal probability of genotypes and report as GQ in each sample field in the VCF output‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (This would help us evaluating the quality of genotype calls)</li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>This will produce a dataset in <a href="http://www.1000genomes.org/wiki/Analysis/variant-call-format">VCF</a> format containing 35 putative variants. Before we can continue, we need to post-process this dataset by breaking compound variants into multiple independent variants.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-simplify-variant-representation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Simplify variant representation</h3>

  <ol>
    <li><strong>VcfAllelicPrimitives</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>‚ÄúSelect VCF dataset‚Äù</em>: the VCF output of <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúMaintain site and allele-level annotations when decomposing‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúMaintain genotype-level annotations when decomposing‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><strong>VCFAllelicPrimitives</strong> generates a VCF files containing 37 records (the input VCF only contained 35). This is because a multiple nucleotide polymorphism (<code class="language-plaintext highlighter-rouge">TAGG|CAGA</code>) at position 618851 have been converted to two:</p>

<table>
  <thead>
    <tr>
      <th>Before</th>
      <th>After</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code class="language-plaintext highlighter-rouge">chr19 618851 . TAGG CAGA 81.7546</code></td>
      <td><code class="language-plaintext highlighter-rouge">chr19 618851 . T C 81.7546</code><br /><code class="language-plaintext highlighter-rouge">chr19 618854 . G A 81.7546</code></td>
    </tr>
  </tbody>
</table>

<h1 id="annotating-variants-with-snpeff">Annotating variants with SnpEff</h1>

<p>At this point we are ready to begin annotating variants using <a href="http://snpeff.sourceforge.net/SnpEff.html"><strong>SnpEff</strong></a>. SnpEff ‚Äú<em>‚Ä¶annotates and predicts the effects of variants on genes (such as amino acid changes)‚Ä¶</em>‚Äù and so is critical for functional interpretation of variation data.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-annotating-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Annotating variants</h3>
  <ol>
    <li><strong>SnpEff</strong> (Variant effect and annotation) <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>‚ÄúSequence changes (SNPs, MNPs, InDels)‚Äù</em>: the VCF output of <strong>VcfAllelicPrimitives</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúGenome source‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally installed reference genome</code></li>
        <li><em>‚ÄúGenome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>SnpEff will generate two outputs:</p>

<ol>
  <li>an annotated VCF file</li>
  <li>an HTML report</li>
</ol>

<p>The report contains a number of useful metrics such as distribution of variants across gene features:</p>

<p><img src="../../images/snpeff_chart.png" alt="SnpEff output chart" /></p>

<p>or changes to codons:</p>

<p><img src="../../images/snpeff_codons.png" alt="SnpEff output codon changes" /></p>

<h1 id="manipulating-variation-data-with-gemini">Manipulating variation data with GEMINI</h1>

<p>Now that we have an annotated VCF file it is time to peek inside our variation data. <a href="http://quinlanlab.org/">Aaron Quinlan</a>, creator of <a href="http://gemini.readthedocs.org/en/latest/index.html">GEMINI</a>, calls it <em>Detective work</em>.</p>

<h2 id="loading-data-into-gemini">Loading data into GEMINI</h2>

<p>The first step is to convert a VCF file we would like to analyze into a GEMINI database. For this we will use <strong>GEMINI Load</strong> tool. GEMINI takes as input a VCF file and a <a href="https://www.cog-genomics.org/plink2/formats#ped">PED</a> file describing the relationship between samples. In our case the PED file looks like this (second imported file):</p>

<table class="table table-responsive">
  <thead>
    <tr>
      <th>#family_id</th>
      <th>sample_id</th>
      <th>pate<span class="notranslate">rna</span>l_id</th>
      <th>mate<span class="notranslate">rna</span>l_id</th>
      <th>sex</th>
      <th>phenotype</th>
      <th>ethnicity</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>family1</td>
      <td>HG004_NA24143_mother</td>
      <td>-9</td>
      <td>-9</td>
      <td>2</td>
      <td>1</td>
      <td>CEU</td>
    </tr>
    <tr>
      <td>family1</td>
      <td>HG003_NA24149_father</td>
      <td>-9</td>
      <td>-9</td>
      <td>1</td>
      <td>1</td>
      <td>CEU</td>
    </tr>
    <tr>
      <td>family1</td>
      <td>HG002_NA24385_son</td>
      <td>HG003_NA24149_father</td>
      <td>HG004_NA24143_mother</td>
      <td>1</td>
      <td>2</td>
      <td>CEU</td>
    </tr>
  </tbody>
</table>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-loading-data-into-gemini"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Loading data into GEMINI</h3>

  <ol>
    <li><strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>‚ÄúVCF file to be loaded in the GEMINI database‚Äù</em>: the VCF output of <strong>SnpEff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúSample information file in PED+ format‚Äù</em>: the uploaded <code class="language-plaintext highlighter-rouge">GIAB-Ashkenazim-Trio.txt</code> tabular</li>
        <li><em>‚ÄúChoose a gemini annotation database‚Äù</em>: the most recent available release</li>
      </ul>

      <p>This will create an SQLite database in your history.</p>
    </li>
    <li>Run <strong>GEMINI db_info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to see the content of the database:
      <ul>
        <li><em>‚ÄúGEMINI database‚Äù</em>: the output of <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>

      <p>This produces a list of all database tables and their columns. The latest version of the GEMINI database schema can be found <a href="https://gemini.readthedocs.org/en/latest/content/database_schema.html">here</a>.</p>
    </li>
  </ol>
</blockquote>

<h2 id="querying-the-gemini-database">Querying the GEMINI database</h2>

<p>The GEMINI database can be queried using the versatile SQL language (more on SQL <a href="https://swcarpentry.github.io/sql-novice-survey">here</a>). In <span class="notranslate">Galaxy</span> this is done using the <strong>GEMINI query</strong> tool. Within this tool SQL commands are typed directly into the <strong>The query to be issued to the database</strong> text box. Let‚Äôs begin getting information from some of the tables we discovered using the <strong>GEMINI db_info</strong> tool above.</p>

<blockquote class="comment">
  <h3 id="comment-gemini-tutorials"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> GEMINI tutorials</h3>

  <p>The examples below are taken from ‚Äú<a href="https://s3.amazonaws.com/gemini-tutorials/Intro-To-Gemini.pdf">Introduction to GEMINI</a>‚Äù tutorial. For extensive documentation see ‚Äú<a href="https://gemini.readthedocs.org/en/latest/content/querying.html">Querying the GEMINI database</a>‚Äù.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-selecting-novel-variants-that-are-not-annotated-in-dbsnp-database"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Selecting ‚Äúnovel‚Äù variants that are not annotated in dbSNP database</h3>

  <ol>
    <li><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>‚ÄúGEMINI database‚Äù</em>: the output of <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT count(*) FROM variants WHERE in_dbsnp == 0</code></li>
      </ul>

      <p>As we can see in the output dataset, there are 21 variants that are not annotated in dbSNP.</p>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-find-variants-within-the-polrmt-gene"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Find variants within the POLRMT gene</h3>

  <ol>
    <li><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with:
      <ul>
        <li><em>‚ÄúGEMINI database‚Äù</em>: the output of <strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT rs_ids, aaf_esp_ea, impact, clinvar_disease_name, clinvar_sig FROM variants WHERE filter is NULL and gene = 'POLRMT'</code></li>
      </ul>

      <p>Since the <code class="language-plaintext highlighter-rouge">variants</code> table has a large number of columns, in the query above we had to select only the most interesting columns. The output shows the variants found within the <em>POLRMT</em> gene.</p>
    </li>
  </ol>
</blockquote>

<h2 id="querying-genotypes">Querying genotypes</h2>

<p>GEMINI provides access to genotype, sequencing depth, genotype quality, and genotype likelihoods for each individual (<code class="language-plaintext highlighter-rouge">subjectID</code>):</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">gt_types.subjectID</code> - three types of genotype types: <code class="language-plaintext highlighter-rouge">HOM_REF</code>, <code class="language-plaintext highlighter-rouge">HET</code>, <code class="language-plaintext highlighter-rouge">HOM_ALT</code></li>
  <li><code class="language-plaintext highlighter-rouge">gt_quals.subjectID</code> - genotype quality</li>
  <li><code class="language-plaintext highlighter-rouge">gt_depths.subjectID</code> - total number of <span class="notranslate">reads</span> in this subject at position</li>
  <li><code class="language-plaintext highlighter-rouge">gt_ref_depths.subjectID</code> -  number of reference allele <span class="notranslate">reads</span> in this subject at position</li>
  <li><code class="language-plaintext highlighter-rouge">gt_alt_depths.subjectID</code> - number of alte<span class="notranslate">rna</span>te allele <span class="notranslate">reads</span> in this subject at position</li>
</ul>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>At how many sites does child have a non-reference allele?</li>
    <li>At how many sites both father and son have non reference alleles?</li>
    <li>List genotypes for father and son where they have non-reference alleles.</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>To answer this question you can run the <strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool with:
        <ul>
          <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT * from variants</code></li>
          <li><em>‚ÄúRestrictions to apply to genotype values‚Äù</em>: <code class="language-plaintext highlighter-rouge">gt_types.HG002_NA24385_son &lt;&gt; HOM_REF</code></li>
        </ul>
      </li>
      <li>To answer this question you can run the <strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool with:
        <ul>
          <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT * from variants</code></li>
          <li><em>‚ÄúRestrictions to apply to genotype values‚Äù</em>: <code class="language-plaintext highlighter-rouge">(gt_types.HG002_NA24385_son &lt;&gt; HOM_REF AND gt_types.HG003_NA24149_father &lt;&gt; HOM_REF)</code></li>
        </ul>
      </li>
      <li>To answer this question you can run the <strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool with:
        <ul>
          <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT gts.HG002_NA24385_son, gts.HG003_NA24149_father from variants</code></li>
          <li><em>‚ÄúRestrictions to apply to genotype values‚Äù</em>: <code class="language-plaintext highlighter-rouge">(gt_types.HG002_NA24385_son &lt;&gt; HOM_REF AND gt_types.HG003_NA24149_father &lt;&gt; HOM_REF)</code></li>
        </ul>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="using-wildcards">Using wildcards</h2>

<p>Wildcards simply writing SQL <span class="notranslate">expression</span>s when searching across multiple terms. The syntax for genotype filter wildcards is:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>(COLUMN).(SAMPLE_WILDCARD).(SAMPLE_WILDCARD_RULE).(RULE_ENFORCEMENT)
</code></pre></div></div>

<p>Let‚Äôs look at some examples.</p>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>At which variants are every sample heterozygous?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>To answer this question you can run the <strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool with:</p>
    <ul>
      <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT chrom, start, end, ref, alt, gene, impact, (gts).(*) FROM variants</code></li>
      <li><em>‚ÄúRestrictions to apply to genotype values‚Äù</em>: <code class="language-plaintext highlighter-rouge">(gt_types).(*).(==HET).(all)</code></li>
    </ul>

    <p>Here we use wildcards for the query (<code class="language-plaintext highlighter-rouge">(gts.(*))</code> means get genotypes for <strong>all</strong> samples) and genotype filtering (<code class="language-plaintext highlighter-rouge">(gt_types).(*).(==HET).(all)</code>, the <a href="https://gemini.readthedocs.org/en/latest/content/querying.html#the-all-operator">all operator</a> implies that we want results for <strong>all</strong> affected individuals).</p>

  </blockquote>
</blockquote>

<h2 id="going-further">Going further</h2>

<p>This short tutorial should give you an overall idea on how generate variant data in <span class="notranslate">Galaxy</span> and process it with GEMINI. Yet there is much more to learn. Below we list GEMINI tutorials and links to <span class="notranslate">Galaxy</span> libraries with relevant data:</p>

<table>
  <thead>
    <tr>
      <th style="text-align: left">Tutorial</th>
      <th style="text-align: center">PDF</th>
      <th style="text-align: center"><span class="notranslate">Galaxy</span> library</th>
      <th style="text-align: center"><span class="notranslate">Galaxy</span> history</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td style="text-align: left">Introduction</td>
      <td style="text-align: center"><a href="https://s3.amazonaws.com/gemini-tutorials/Intro-To-Gemini.pdf"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">references</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/library/folders/F0283ca691a41c352"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-library</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/u/aun1/h/gemini-introduction"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-history</span></a></td>
    </tr>
    <tr>
      <td style="text-align: left">Identifying <em>de novo</em> mutations underlying Mendelian disease</td>
      <td style="text-align: center"><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-DeNovo-Tutorial.pdf"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">references</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/library/folders/F775008f45cbbf010"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-library</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/u/aun1/h/gemini-de-novo-mutations"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-history</span></a></td>
    </tr>
    <tr>
      <td style="text-align: left">Identifying autosomal recessive variants underlying Mendelian disease</td>
      <td style="text-align: center"><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Recessive-Tutorial.pdf"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">references</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/library/folders/F35b262f5ac8aa63a"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-library</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/u/aun1/h/gemini-autosomal-recessive"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-history</span></a></td>
    </tr>
    <tr>
      <td style="text-align: left">Identifying autosomal dominant variants underlying Mendelian disease</td>
      <td style="text-align: center"><a href="https://s3.amazonaws.com/gemini-tutorials/Gemini-Dominant-Tutorial.pdf"><i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">references</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/library/folders/F1c4722ad56892a31"><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-library</span></a></td>
      <td style="text-align: center"><a href="https://usegalaxy.org/u/aun1/h/gemini-autosomal-dominant"><i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-history</span></a></td>
    </tr>
  </tbody>
</table>

<h3 id="how-to-use-these-tutorials">How to use these tutorials?</h3>

<ul>
  <li>Right click on the <strong>PDF</strong> <i class="far fa-bookmark" aria-hidden="true"></i><span class="visually-hidden">references</span> icon and open tutorial in a new browser tab</li>
  <li>Right click on <strong><span class="notranslate">Galaxy</span> history</strong> <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-history</span> icon and open <span class="notranslate">Galaxy</span> history in another new browser tab</li>
  <li>
    <p>When the <span class="notranslate">Galaxy</span> history interface opens, you will need to click the <strong>Import history</strong> link highlighted with a red outline in the following figure:</p>

    <p><img src="../../images/import_history.png" alt="Import history" /></p>
  </li>
  <li>
    <p>If you have a wide screen arrange browsers tabs side by side:</p>

    <p><img src="../../images/side-by-side.png" alt="Arranging browsers side by side" /></p>
  </li>
  <li>
    <p>Proceed with the tutorial. For example, to repeat the following command from GEMINI tutorial:</p>

    <p><img src="../../images/gemini_command.png" alt="GEMINI query command" /></p>

    <p>use <span class="notranslate">Galaxy</span>‚Äôs <strong>GEMINI query</strong> tool as below:</p>

    <p><img src="../../images/galaxy_command.png" alt="GEMINI_query tool interface" /></p>
  </li>
  <li>and so on‚Ä¶.</li>
</ul>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Variants versus Genotypes</p>
</li>
                            
                            <li><p>Bayesian approach to variant calling</p>
</li>
                            
                            <li><p>Freebayes calls variant using haplotype windows</p>
</li>
                            
                            <li><p>SNPEff allows functional annotation of variants</p>
</li>
                            
                            <li><p>Gemini is a flexible system for analyzing variation patterns</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Calling variants in diploid systems (Variant Analysis)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Anton Nekrutenko, Nicola Soranzo, 2021 <b>Calling variants in diploid systems (Galaxy Training Materials)</b>. <a href="/training-material/topics/variant-analysis/tutorials/dip/tutorial.html">/training-material/topics/variant-analysis/tutorials/dip/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{variant-analysis-dip,
    author = "Anton Nekrutenko and Nicola Soranzo",
    title = "Calling variants in diploid systems (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/variant-analysis/tutorials/dip/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Anton Nekrutenko, Nicola Soranzo)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/dip/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
