---
layout: tutorial_hands_on

title: Normalization of Ribo-Seq reads
zenodo_link: ''
questions:
- How to get read counts from the bam file?
- How to normalize read counts for comparing gene expression between different conditions?
objectives:
- Get information of read counts from the bam files
- Learn to normalize read counts using DESeq2
time_estimation: ''
key_points:
- We can obtain a read count matrix through featureCounts and DESeq2
- The matrix could be used for differential expression analysis
contributors:
- ldyang14
- IceApink
---


# Introduction
{:.no_toc}

<!-- This is a comment. -->

After mapping was completed, we would obtain detailedly mapped results, which usually is SAM or BAM format.  Spontaneously, we would wonder the expression of interested genes to compare, explore, and analyze differences for a customized experiment condition. Hence, how should we carry out this process? Firstly, we should extract information of read counts from BAM/SAM file using tools like `featureCounts`. Is that it? Of course not. Since library depth and gene lengths are different in each sample, we need to normalize read counts mapped on each gene for comparisons between them (an example is shown below). It is important to execute this step because we would analyze accurately only after that read counts were normalized exactly.

![Normalization of reads]( https://izabelcavassim.files.wordpress.com/2015/03/screenshot-from-2015-03-08-2245511.png "Normalization of reads")

> ### Agenda
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Import data

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Create a new history for this tutorial
> 2. Import the files from [Figshare]() or from the shared data library
>
>    ```
>    KO_1_RPF	KO_1	KO	https://ndownloader.figshare.com/files/20088581?private_link=a210c49b717479c834d7
KO_2_RPF	KO_2	KO	https://ndownloader.figshare.com/files/20088584?private_link=a210c49b717479c834d7
KO_3_RPF	KO_3	KO	https://ndownloader.figshare.com/files/20088587?private_link=a210c49b717479c834d7
WT_1_RPF	WT_1	WT	https://ndownloader.figshare.com/files/20088590?private_link=a210c49b717479c834d7
WT_2_RPF	WT_2	WT	https://ndownloader.figshare.com/files/20088593?private_link=a210c49b717479c834d7
>    ```
>
>    {% include snippets/import_via_link.md %}
>    {% include snippets/import_from_data_library.md %}
>
> 3. Rename the datasets
> 4. Check the datatype of upload files if are bam, if not, then modify the datatype to bam
>
>    {% include snippets/change_datatype.md datatype="datatypes" %}
{: .hands_on}

# Get read counts with **featureCounts**

Firstly, we need to calculate read counts for each gene according to the bam file, thus lay the foundation for the subsequent differential expression analysis. Here we selected [featureCounts](http://bioinf.wehi.edu.au/featureCounts/), which is an ultrafast and accurate tool for calculating read counts.

> ### {% icon hands_on %} Hands-on: Get read counts with **featureCounts**
>
> 1. **featureCounts** {% icon tool %} with the following parameters:
> - {% icon param-files %} *"Alignment file"*: `KO_1_RPF`, `KO_2_RPF`, `KO_3_RPF`, `WT_1_RPF`, `WT_2_RPF`
>    - *"Gene annotation file"*: `in your history`
>      - {% icon param-file %} *"Gene annotation file"*: `hg38.gencode.v32.gtf` 
> 2. **MultiQC** {% icon tool %} with the following parameters:
>    - In *"Results"*:
>      - {% icon param-select %} *"Which tool was used generate logs?"*: `featureCounts`
>        - {% icon param-files %} *"Output of FeatureCounts"*: `KO_1_RPF`, `KO_2_RPF`, `KO_3_RPF`, `WT_1_RPF`, `WT_2_RPF`
> 
{: .hands_on}

The results generated by the featureCounts record the aligned and unaligned number of reads for each sample. Hence, we can browse the general statistical information of the read alignment utilizing MultiQC.

![Assignment of reads](../../images/normalization-of-reads/featureCounts_assignment_plot.png "Assignment of reads")



# Normalization with **DESeq2**

Secondly, we need to normalized read counts to compare gene expression between different groups.

> ### {% icon hands_on %} Hands-on: Task description
>
> 1. **DESeq2** {% icon tool %} with the following parameters:
>    - *" how"*: `Select datasets per level`
>        - In *"Factor"*:
>            - {% icon param-repeat %} *"Insert Factor"*
>                - *"Specify a factor name, e.g. effects_drug_x or cancer_markers"*: `Treatment`
>                - In *"Factor level"*:
>                    - {% icon param-repeat %} *"Insert Factor level"*
>                        - *"Specify a factor level, typical values could be 'tumor', 'normal', 'treated' or 'control'"*: `WT`
>                        - {% icon param-files %} *"Counts file(s)"*: `WT_1_RPF`, `WT_2_RPF`
>                    - {% icon param-repeat %} *"Insert Factor level"*
>                        - *"Specify a factor level, typical values could be 'tumor', 'normal', 'treated' or 'control'"*: `KO`
>                        - {% icon param-files %} *"Counts file(s)"*: `KO_1_RPF`, `KO_2_RPF`, `KO_3_RPF`
>    - *"Choice of Input data"*: `Count data (e.g. from HTSeq-count, featureCounts or StringTie)`
>    - *"Output normalized counts table"*: `Yes`
>    - *"Output rLog normalized table"*: `Yes`
> 
{: .hands_on}

Four files will be generated, their are 3 tabular-format files about normalized gene expression and 1 file plotted the quality of samples.

![Sample distance](../../images/normalization-of-reads/DESeq2_PCA_samples.png "Sample distance")


> ### {% icon question %} Questions
>
> 1. How is the quality of RPF samples?
>
> > ### {% icon solution %} Solution
> >
> > 1. It seems pretty good because WT (wildtype) groups and KO (knockout) groups are clearly separated.
> >
> {: .solution}
>
{: .question}

# Conclusion

{:.no_toc}

Results from mapping contain particular information about the distribution of reads on the genome. However, the statistical or visual information needs to be extracted and processed further from the mapping results, thus helping us to understand the variation of gene expression. One can understand roughly these steps by means of this tutorial to analyze differential gene expression, then to predict genes correlated with conditions or to determine the function of interested genes.
