<!DOCTYPE html>
<html>









  <head>
    <meta charset="utf-8">
    <title>Scripting Galaxy using the API and BioBlend</title>
    
        <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

    
    <link rel="stylesheet" href="/training-material/assets/css/slides.css">
    <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

    
    
    
    
    <meta name="description" content="Galaxy is an open-source project. Everyone can contribute..." />
    <meta property="og:title" content="Galaxy Training: Scripting Galaxy using the API and BioBlend" />
    <meta property="og:description" content="Galaxy is an open-source project. Everyone can contribute..." />
    <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    <script type="application/ld+json">
      


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Scripting Galaxy using the API and BioBlend",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Development in Galaxy",
    "description": "Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...",
    "url": "https://training.galaxyproject.org//training-material/topics/dev/"
  },
  "courseCode": "dev / bioblend-api / slides",
  "learningResourceType": "slides",
  "name": "Slides for 'Scripting Galaxy using the API and BioBlend' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/dev/tutorials/bioblend-api/slides.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - What is a RESTful API?\n - How to interact with Galaxy in a script?\n - Why and when should I use BioBlend?\n\n\\nThe objectives are:\n - Describe the principles of RESTful APIs\n - Select the most efficient way to work with Galaxy\n - Use the API and BioBlend to interact with Galaxy\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://ipython.org",
      "name": "IPython"
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Olivia Doppelt-Azeroual"
    },
    {
      "@type": "Person",
      "name": "Fabien Mareuil"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Dannon Baker"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Olivia Doppelt-Azeroual"
    },
    {
      "@type": "Person",
      "name": "Fabien Mareuil"
    },
    {
      "@type": "Person",
      "name": "Nicola Soranzo"
    },
    {
      "@type": "Person",
      "name": "Dannon Baker"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Development in Galaxy",
      "description": "Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...",
      "url": "https://training.galaxyproject.org//training-material/topics/dev/"
    }
  ]
}
    </script>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
  </head>
  <body>
    <textarea id="source">
name: inverse
layout: true
class: center, middle, inverse

<div class="my-header"><span>
<a href="/training-material/topics/dev" title="Return to topic page" ><i class="fa fa-level-up" aria-hidden="true"></i></a>

<a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/dev/tutorials/bioblend-api/slides.html"><i class="fa fa-pencil" aria-hidden="true"></i></a>

</span></div>

<div class="my-footer"><span>

<img src="/training-material/assets/images/GTN-60px.png" alt="Galaxy Training Network" style="height: 40px;"/>

</span></div>

---

# Scripting Galaxy using the API and BioBlend



---

## Requirements

Before diving into this slide deck, we recommend you to have a look at:





    
- [IPython](https://ipython.org)
    




.footnote[Tip: press `P` to view the presenter notes]

???
Presenter notes contain extra information which might be useful if you
intend to use these slides for teaching.

Press `P` again to switch presenter notes off

Press `C` to create a new window where the same presentation will be displayed.
This window is linked to the main window. Changing slides on one will cause the
slide to change on the other. Useful when presenting.

---






### &lt;i class=&quot;far fa-question-circle&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;question&lt;/span&gt; Questions


- What is a RESTful API?

- How to interact with Galaxy in a script?

- Why and when should I use BioBlend?


---



### &lt;i class=&quot;fas fa-bullseye&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;objectives&lt;/span&gt; Objectives


- Describe the principles of RESTful APIs

- Select the most efficient way to work with Galaxy

- Use the API and BioBlend to interact with Galaxy


---


## Galaxy API

- **Application Programming Interface (API)**: the protocol defined by a software for how it can be controlled by an external program
- Galaxy provides a rich API:
  - Over the HTTP protocol
  - The Galaxy UI is being migrated on top of the Galaxy API

???

The server providing the API becomes the program, and the URLs become the commands - all through the medium of the HTTP protocol.

---

## Interacting with Galaxy: UI vs. API

- The Galaxy UI is good for:
  - exploring and visualizing data
  - experimenting
  - graphically designing workflows
  - people not comfortable with command line

- The Galaxy API is good for:
  - interact programmatically with the server
    - complex control: **branching** and **looping** (not possible in workflows)
  - automate repetitive tasks
  - integration with external resources

???

The API allows you to use Galaxy's capabilities programmatically:
- run a workflow over each file in a directory
- run a second workflow only if the first passes some QA threshold
- upload a FASTQ file when the sequencer finishes writing it.

Additionally, the features that make Galaxy great are still applied when using the API:
- histories still capture the exact steps of your experiments and reproducibility is maintained
- jobs and compute resources are still managed for you
- datasets and metadata persist and are centralized
- your work is still sharable.

It's also worth noting that all the work you did via the API is still accessible and modifiable when you return to the UI.

---

## Galaxy API functionalities

- Users can:
  - manage histories and datasets
  - upload and download data
  - run tools and workflows, ...

- Admins can also manage:
  - data libraries
  - Tool Shed repositories
  - users, quotas, roles...

- Source code lives at https://github.com/galaxyproject/galaxy/tree/dev/lib/galaxy/webapps/galaxy/api/

???

Admin examples:
- upload files and start a workflow when new files are added to a directory on the server
- automatically create libraries for new users
- replicate the set of tools installed on another server
- automatically assign users to the right group according to email domain

---

## RESTful API

.left[REpresentational State Transfer (REST) is the architectural style of the World Wide Web:]
- client–server
- stateless: no client context stored on the server between requests
- cacheable by client
- Uniform Resource Identifiers (URIs)
- JSON media type
- standard HTTP request methods (GET, POST, PUT, DELETE, PATCH) and status codes

???

A common way to construct an API command is to view it as a simple (imperative) sentence: `&lt;verb&gt; &lt;a resource&gt;`
- Examples of resources for the Galaxy API are: datasets, tools, jobs, histories, libraries, users... - essentially anything in Galaxy that is recorded in the database.
- The HTTP verbs are GET (i.e. retrieve), POST (create), PUT (replace), DELETE, PATCH (partially modify). There are others but they don't apply (yet). These are [HTTP request methods](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol#Request_methods).
  - GET calls should produce no side effects.
  - PUT and DELETE should be idempotent, i.e. should produce the same result no matter how many times they are repeated.
- Often additional options and parameters must be passed to specify how the command should take place.

---

## Resources, URIs, and parameters

- API call: **HTTP method + URI [+ payload]**

  - `GET https://usegalaxy.org/api/histories?order=name` -&gt; ordered list of histories
  - `POST /api/histories {&quot;name&quot;: &quot;New&quot;}` -&gt; create a history named &quot;New&quot;
  - `PUT /api/histories/&lt;id&gt; {&quot;published&quot;: true}` -&gt; publish a history
  - `DELETE /api/histories/&lt;history_id&gt;/contents/&lt;id&gt;` -&gt; delete a history dataset

- URI parameters: ids in path, others in the *query* (`?name=value&amp;...`)
- POST/PUT payload as JSON

---

## JSON format

.left[JavaScript Object Notation https://www.json.org/] &lt;img style=&quot;float: right;&quot; height=&quot;80&quot; width=&quot;80&quot; alt=&quot;JSON logo&quot; src=&quot;../../images/json160.gif&quot; /&gt;
  - Lightweight data-interchange text format
  - Easy to read/write for both humans and machines
  - ECMA-404 open standard (2013), [RFC 8259](https://tools.ietf.org/html/rfc8259) (2017)

```json
{&quot;history_id&quot;: &quot;b5731bb49a17bf50&quot;,
 &quot;id&quot;: &quot;df06cc665d85b6ea&quot;,
 &quot;inputs&quot;: {&quot;0&quot;: {&quot;id&quot;: &quot;bbd44e69cb8906b51528b5d606d1fdd0&quot;,
                  &quot;src&quot;: &quot;hda&quot;}},
 &quot;model_class&quot;: &quot;WorkflowInvocation&quot;,
 &quot;outputs&quot;: [&quot;bbd44e69cb8906b528819eaaff340ecd&quot;,
             &quot;0ff30b4e2a4bed9e&quot;],
 &quot;state&quot;: &quot;scheduled&quot;,
 &quot;update_time&quot;: &quot;2015-07-03T19:28:39.544574&quot;,
 &quot;workflow_id&quot;: &quot;56482e194d798eb6&quot;}
```

???

Data returned from (and in some cases passed to) the Galaxy API is often described or formatted in JSON.

---

## Status codes and errors

- HTTP status codes:
  - 200 OK, 400 your error, 500 server error, ...&lt;br&gt;
    https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
- Galaxy error codes and messages: [`lib/galaxy/exceptions/`](https://github.com/galaxyproject/galaxy/tree/dev/lib/galaxy/exceptions)
  - Still a work in progress

---

## How to access a REST API

.left[With anything that can communicate over HTTP:]
- Command line:
  - `wget`: only GET
  - `curl`: all methods
- GUI:
  - Browsers: only GET
  - [RESTClient](https://addons.mozilla.org/firefox/addon/restclient/) add-on for Firefox
  - [Advanced REST Client](https://advancedrestclient.com/)
- Software libraries:
  - General HTTP libraries (e.g. *requests* for Python)
  - Service-specific libraries (e.g. *BioBlend* to access Galaxy using Python)

---

## Security

- Most API calls require authentication
  - When the UI accesses the API, session auth is used
  - Other callers needs an **API key**: alphanumeric string (32 chars) identifying a registered user&lt;br&gt;
    Keep it secure, it’s the same as a username+password!
- Always use HTTPS:
  - **https**://localhost?key=foo is safer due to the encryption of the transmitted data

---

## Advanced Galaxy API config

.left[Options in `config/galaxy.yml`:]
- User impersonation by adding `run_as` in the payload
  ```yaml
  # Optional list of email addresses of API users who can make calls on
  # behalf of other users.
  api_allow_run_as: foo@foo.com
  ```

- Bootstrapping Galaxy

  ```yaml
  # Master key that allows many API admin actions to be used without
  # actually having a defined admin user in the database/config.  Only
  # set this if you need to bootstrap Galaxy, you probably do not want
  # to set this on public servers.
  master_api_key: MASTERLOCK
  ```

---

## Galaxy API pros and cons

- Pros:
  - Distributed with Galaxy
  - Well tested
  - Language agnostic
- Cons:
  - Very low-level

---

## BioBlend

- BioBlend is a **Python library** that wraps the functionality of Galaxy and CloudMan APIs
- Started by Enis Afgan, Nuwan Goonasekera and Clare Sloggett in 2012. Contributions by the Galaxy Team and the community
- Open source (MIT license)
- Available via PyPI and from https://github.com/galaxyproject/bioblend

---

## BioBlend features

- Stable procedural API
- Supported under Python &gt;=3.5
- Wraps all main Galaxy API controllers
- Extensive TravisCI testing:
  - on Galaxy release_17.09 and later
  - \&gt;180 unit tests
- Well-documented on https://bioblend.readthedocs.io

---

## Tutorials - part 1

Now it's a good time to try BioBlend! We have prepared
[some tutorials](https://github.com/nsoranzo/bioblend-tutorial/) as Jupyter
notebooks, which can be executed online with Binder at
https://mybinder.org/v2/gh/nsoranzo/bioblend-tutorial/master

Walk-throughs:
- `galaxy_api_histories.ipynb`: interact with histories using the **Galaxy API**
- `bioblend_histories.ipynb`: interact with histories using **BioBlend**

Exercises:
- `ex1_galaxy_api.ipynb`: launch a workflow using the **Galaxy API**
- `ex1_bioblend.ipynb`: launch a workflow using **BioBlend**

---

## BioBlend limitations

- Python-only (but separate *blend4j* and *blend4php* exist)
- Its methods just deserialize the JSON response
  - No isolation from changes in the Galaxy API
  - Need to extract the entity id for further processing
- No explicit modeling of Galaxy entities and their relationships
- Complex operations still need many function calls
  - Need for higher-level functionality

---

## BioBlend.objects

- BioBlend.objects is an extra layer which adds an **object-oriented interface** for the Galaxy API
- Started by Simone Leo, Luca Pireddu and Nicola Soranzo at CRS4 in 2013
- Distributed with BioBlend
- Presently limited to datasets, histories, jobs, libraries, tools and workflows

---

## Tutorials - part 2

Walk-throughs:
- `bioblend.objects_histories.ipynb`: interact with histories using **BioBlend.objects**

Exercises:
- `ex1_bioblend.objects.ipynb`: launch a workflow using **BioBlend.objects**

---

## References

- Galaxy API docs: https://docs.galaxyproject.org/en/master/api_doc.html
- BioBlend docs: https://bioblend.readthedocs.io/
- BioBlend chat: https://gitter.im/galaxyproject/bioblend
- C. Sloggett, N. Goonasekera, E. Afgan. BioBlend: automating pipeline analyses within Galaxy and CloudMan. *Bioinformatics* 29(13), 1685-1686, 2013, doi:[10.1093/bioinformatics/btt199](https://doi.org/10.1093/bioinformatics/btt199)
- S. Leo, L. Pireddu, G. Cuccuru, L. Lianas, N. Soranzo, E. Afgan, G. Zanetti. BioBlend.objects: metacomputing with Galaxy. *Bioinformatics* 30 (19), 2816-2817, 2014, doi:[10.1093/bioinformatics/btu386](https://doi.org/10.1093/bioinformatics/btu386)

---

## Before we start
1. Get your Galaxy API key:
   1. In your browser, open your Galaxy homepage (e.g. https://usegalaxy.org/ )
   2. Log in, or create an account if you don't have one yet
   3. Go to `User -&gt; Preferences` in the top menu bar, then click on `Manage API key`
   4. If there is no current API key available, click on `Create a new key` to generate it
   5. Copy your API key to somewhere convenient, you will need it throughout this tutorial

2. Install the tools and workflow on your Galaxy:
	1. The tools:  Click on the Admin tab
		* In the Tools and Tool Shed category, click on the line **Search Tool Shed**
		* Select the **&quot;Galaxy Main Tool Shed&quot;**, and the **&quot;Browse valid repositories&quot;** line
		* Search and install *bam_to_sam* from devteam and *samtools_sort* from IUC owner
	2. Launch IPython on a terminal

---

## Connect with Galaxy using IPython:

* Import the GalaxyInstance object from BioBlend module:

  ```python
  from bioblend.galaxy import GalaxyInstance
  ```

* Create your GalaxyInstance instance object using your Galaxy URL and API key

  ```python
  gi = GalaxyInstance(url=&quot;http://localhost:8080&quot;, key=&quot;your key&quot;)
  ```

#### Why IPython:
* Automatic completion, e.g. type *gi.* and then press the `Tab` key

* To better understand BioBlend methods and classes, you can use `help(command), object?, object??`

---

## Access BioBlend documentation

* Let's look at the help for the **run_tool** method:
  ```python
  help(gi.tools.run_tool)
  ```
* The help should tell you that `run_tool` needs 3 inputs:
  1. A *history_id*, where the input data is and where the output data will be
  2. A *tool_id*, which will tell Galaxy which tool to execute
  3. *tool_inputs*, a dictionary storing the data used to run the tool

---

class: top

## Tools Object (1/4)

* Get the samtools sort tool id

  ```python
  list_tool = gi.tools.get_tools(name='sort')
  ```

---

## Tools Object (2/4)

* The dictionary `tool_inputs` is needed to run a tool

* It is a Python dictionary defined by specific methods from the `bioblend.galaxy.tools.inputs` class

  ```python
  from bioblend.galaxy.tools.inputs import inputs
  ```

* The `inputs` method instantiates a class called `InputsBuilder`:

  ```python
  help(inputs)
  ```

* Each input from the tool XML needs to be defined using the methods `set_param` or `set_dataset_param` from the `InputsBuilder` class
==&gt; If the input format is &quot;data&quot;, the method to use is `set_dataset_param`

* Here is an example:

  ```python
  myinputs = inputs().set_param(&quot;param1&quot;, 'value')
        .set_dataset_param(&quot;data1&quot;, 'dataset_id', src=&quot;hda&quot;)
  ```

**To run the tool samtools sort, we need more information on the tool itself**

---

class: top
## Tools Object (3/4)

* Get the details on the &quot;samtool sort&quot; tool

  ```python
  detail_tool = gi.tools.show_tool(list_tool[0]['id'], io_details=True)
  ```

--

```python
detail_tool['inputs']
[{u'argument': None,
  u'edam_formats': [u'format_2572'],
  u'extensions': [u'bam'],
  ...
  u'multiple': False,
  u'name': u'input1', &lt;---------------------------------------|
  u'optional': False,                                         |
  u'options': {u'hda': [], u'hdca': []},                      |
  u'type': u'data'},                                          | Critical
  {...                                                         | information
  u'name': u'sort_mode', &lt;------------------------------------|
  u'optional': False,                                         |
  u'options': [[u'Chromosomal coordinates', u'', True], &lt;-----|
    [u'Read names', u'-n', False]],
  u'type': u'select',
  u'value': u''}]
```

---

class: top
## Tools Object (4/4)

* Inputs information can be displayed with `detail_tool['inputs']`, use this to instantiate the inputs object

  ```python
  myinputs = inputs().set_dataset_param(&quot;input1&quot;, data_history['id'], src='hda') \
      .set_param(&quot;sort_mode&quot;, &quot;&quot;)
  ```

--

* Now you can launch `samtool sort`

  ```python
  gi.tools.run_tool(new_history['id'], detail_tool['id'], myinputs)
  ```




---
### &lt;i class=&quot;fas fa-key&quot; aria-hidden=&quot;true&quot;&gt;&lt;/i&gt;&lt;span class=&quot;visually-hidden&quot;&gt;keypoints&lt;/span&gt; Key points


- The API allows you to use Galaxy's capabilities programmatically

- BioBlend wraps the functionality of Galaxy and CloudMan APIs

- BioBlend.objects is an object-oriented interface for interacting with Galaxy







---

## Thank you!

This material is the result of a collaborative work. Thanks to the [Galaxy Training Network](https://wiki.galaxyproject.org/Teach/GTN) and all the contributors!


<div class="contributors-line">


<a href="/training-material/hall-of-fame/odoppelt/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/odoppelt" alt="Olivia Doppelt-Azeroual">Olivia Doppelt-Azeroual</a>, <a href="/training-material/hall-of-fame/fmareuil/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/fmareuil" alt="Fabien Mareuil">Fabien Mareuil</a>, <a href="/training-material/hall-of-fame/nsoranzo/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nsoranzo" alt="Nicola Soranzo">Nicola Soranzo</a>, <a href="/training-material/hall-of-fame/dannon/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/dannon" alt="Dannon Baker">Dannon Baker</a>


</div>



<img src="/training-material/assets/images/GTN.png" alt="Galaxy Training Network" style="height: 100px;"/>


This material is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.


    </textarea>
	<script src="/training-material/assets/js/remark-latest.min.js" type="text/javascript"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
      var slideshow = remark.create({navigation: {scroll: false,}});
      var hljs = remark.highlighter.engine;
      var snippets = document.querySelectorAll('code.remark-code');
        [].forEach.call(snippets,function(snippet){
          snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });
      var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.parentElement;
      }});
    </script>

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('body').prepend("<div style='text-align: center'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>

  </body>
</html>
