---
layout: tutorial_hands_on

title: Identification of AMR genes in an assembled bacterial genome
zenodo_link: 'https://zenodo.org/record/10572227'
questions:
- Which resistance genes are on a bacterial genome?
- Where are the genes located on the genome?
objectives:
- Run a series of tool to assess the presence of antimicrobial resistance genes (ARG)
- Get information about ARGs 
- Visualize the ARGs and plasmid genes in their genomic context
time_estimation: 2h
key_points:
- staramr is a powerful tool to predict ARGs and plasmid genes
- Visualization of the ARGs and plasmid genes in their genomic context helps to make sense of the data
tags:
- gmod
- illumina
- amr
- one-health
- jbrowse1
- microgalaxy
subtopic: prokaryote
edam_ontology:
- topic_3673 # Whole genome sequencing
- topic_3305 # Public health and epidemiology
- topic_0622 # Genomics
- topic_3301 # Microbiology
- topic_0080 # Sequence analysis
- topic_3324 # Infectious disease
- topic_4013 # Antimicrobial resistance
level: Introductory

contributions:
  authorship:
  - bazante1
  - bebatut
  editing:
  - hexylena
  - bazante1
  - shiltemann
  - miaomiaozhou88
  funding:
  - avans-atgm
  - abromics
  - elixir

follow_up_training:
- type: "internal"
  topic_name: visualisation
  tutorials:
  - jbrowse
- type: "internal"
  topic_name: galaxy-interface
  tutorials:
  - history-to-workflow
---

Antimicrobial resistance (AMR) is a global phenomenon with no geographical or species boundaries, which poses an important threat to human, animal and environmental health. It is a complex and growing problem that compromises our ability to treat bacterial infections.

AMR gene content can be assessed from whole genome sequencing to detect known resistance mechanisms and potentially identify novel mechanisms. 

To illustrate the process to identify AMR gene in a bacterial genome, we take an assembly of a bacterial genome (KUN1163 sample) generated by following a [bacterial genome assembly tutorial]({% link topics/assembly/tutorials/mrsa-illumina/tutorial.md %}) from data produced in "Complete Genome Sequences of Eight Methicillin-Resistant *Staphylococcus aureus* Strains Isolated from Patients in Japan" ({% cite Hikichi_2019 %}). 

> Methicillin-resistant *Staphylococcus aureus* (MRSA) is a major pathogen
> causing nosocomial infections, and the clinical manifestations of MRSA
> range from asymptomatic colonization of the nasal mucosa to soft tissue
> infection to fulminant invasive disease. Here, we report the complete
> genome sequences of eight MRSA strains isolated from patients in Japan.
{: .quote cite="{% cite_url Hikichi_2019 %}"}

> <agenda-title></agenda-title>
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Galaxy and data preparation

Any analysis should get its own Galaxy history. So let's start by creating a new one and get the data (contig file from the assembly) into it.

> <hands-on-title>Prepare Galaxy and data</hands-on-title>
>
> 1. Create a new history for this analysis
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Rename the history
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
> 3. {% tool [Import](upload1) %} the contig file from [Zenodo]({{ page.zenodo_link }}) or from Galaxy shared data libraries:
>
>    ```
>    {{ page.zenodo_link }}/files/DRR187559_contigs.fasta
>    ```
>
>    {% snippet faqs/galaxy/datasets_import_via_link.md %}
>
>    {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>
>
{: .hands_on}

# Identification of AMR genes

To identify AMR genes in contigs, tools like ABRicate or staramr ({% cite bharat2022correlation %}) can be used. In this tutorial, we use [**staramr**](https://github.com/phac-nml/staramr). **Staramr** scans bacterial genome contigs against the **ResFinder** ({% cite zankari2012identification %}), **PointFinder** ({% cite zankari2017pointfinder %}), and **PlasmidFinder** ({% cite carattoli2014plasmidfinder %}) databases (used by the ResFinder webservice) and compiles a summary report of detected antimicrobial resistance genes.

> <comment-title></comment-title>
>
> If you are interested in learning more about using ABRicate for AMR gene detection, you can check the sub-section on AMR in "Gene based pathogenic identification" of ["Pathogen detection from (direct Nanopore) sequencing data using Galaxy - Foodborne Edition" tutorial]({% link topics/microbiome/tutorials/pathogen-detection-from-nanopore-foodborne-data/tutorial.md %})
>
{: .comment}

> <hands-on-title>Run staramr</hands-on-title>
>
> 1. {% tool [staramr](toolshed.g2.bx.psu.edu/repos/nml/staramr/staramr_search/0.10.0+galaxy1) %} with the following parameters:
>    - {% icon param-file %} *"genomes"*: Contig file
>
{: .hands_on}

**staramr** generates 8 outputs:

1. `summary.tsv`:  A summary of all detected AMR genes/mutations/plasmids/sequence type in each genome, one genome per line and the following columns:

   * __Isolate ID__: The id of the isolate/genome file(s) passed to `staramr`.
   * __Quality Module__: The isolate/genome file(s) pass/fail result(s) for the quality metrics
   * __Genotype__: The AMR genotype of the isolate.
   * __Predicted Phenotype__: The predicted AMR phenotype (drug resistances) for the isolate.
   * __CGE Predicted Phenotype__: The CGE-predicted AMR phenotype (drug resistances) for the isolate (CGE = [Center for Genomic Epidemiology](http://www.genomicepidemiology.org/))

   * __Plasmid__: Plasmid types that were found for the isolate.
   * __Scheme__: The MLST scheme used

      MLST stands for MultiLocus Sequence Typing. It is a technique for the typing of multiple loci, using DNA sequences of internal fragments of multiple housekeeping genes to characterize isolates of microbial species. 

      Here, **starmr** uses [mlst](https://github.com/tseemann/mlst) to scan the contig files against traditional [PubMLST](https://pubmlst.org/) typing schemes. The correspondance between the scheme and the bacteria genus and species is accessible in the [map](https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab)

   * __Sequence Type__: The sequence type that's assigned when combining all allele types
   * __Genome Length__: The isolate/genome file(s) genome length(s)
   * __N50 value__: The isolate/genome file(s) N50 value(s) 
   * __Number of Contigs Greater Than Or Equal To 300 bp__: The number of contigs greater or equal to 300 base pair in the isolate/genome file(s)
   * __Quality Module Feedback__: The isolate/genome file(s) detailed feedback for the quality metrics

    > <hands-on-title>Inspect the staramr output</hands-on-title>
    >
    > 1. View {% icon galaxy-eye %} the `summary.tsv` file
    >
    {: .hands_on}

    > <question-title>Inspect the staramr output</question-title>
    >
    > 1. How many genomes are there?
    > 2. Has the genome failed or passed the quality? Why?
    > 3. What are the predicted AMR drug resistances? Are they similar to the one found for KUN1163 in [Table 1](https://journals.asm.org/doi/10.1128/mra.01212-19#tab1) in {% cite Hikichi_2019 %}?
    > 4. What is the MLST scheme? Does that correspond to the expected species and genus?
    >
    > > <solution-title>Inspect the staramr output</solution-title>
    > >
    > > 1. There is one genome (1 line)
    > > 2. The genome has failed the quality (column 2), because the genome length is not within the acceptable length range (last column).
    > > 3. We can summarize starmr output and Table 1 in {% cite Hikichi_2019 %}:
    > >    
    > >    Antibiotic name | Abbreviation | staramr | {% cite Hikichi_2019 %}
    > >    --- | --- | --- | ---
    > >    Amikacin | | Yes |
    > >    Azithromycin | | Yes |
    > >    Cefazolin | CEZ | | Yes
    > >    clindamycin | CLDM | | Yes
    > >    Erythromycin | EM | Yes | Yes
    > >    Gentamicin | GM | Yes | Yes
    > >    Imipenem | IPM | | Yes
    > >    Levofloxacin | LVFX | | Yes
    > >    Oxacillin | MPIPC | | Yes
    > >    Penicillin | | Yes |
    > >    Spectinomycin | | Yes |
    > >    Tetracycline | | Yes |
    > >    Tobramycin | | Yes |
    > >    
    > > 4. The scheme is saureus, so *Staphylococcus aureus* (given the [scheme genus map](https://github.com/tseemann/mlst/blob/master/db/scheme_species_map.tab)), which is coherent with MRSA
    > {: .solution}
    >
    {: .question}

2. `detailed_summary.tsv`: A detailed summary of all detected AMR genes/mutations/plasmids/sequence type in each genome, one gene per line and the following columns:

    * __Isolate ID__: The id of the isolate/genome file(s) passed to `staramr`.
    * __Data__: The particular gene detected from ResFinder, PlasmidFinder, PointFinder, or the sequence type.
    * __Data Type__: The type of gene (Resistance or Plasmid), or MLST.
    * __Predicted Phenotype__: The predicted AMR phenotype (drug resistances) found in ResFinder/PointFinder. Plasmids will be left blank by default.
    * __CGE Predicted Phenotype__: The CGE-predicted AMR phenotype (drug resistances) found in ResFinder/PointFinder. Plasmids will be left blank by default.
    * __%Identity__: The % identity of the top BLAST HSP to the gene.
    * __%Overlap__: THe % overlap of the top BLAST HSP to the gene (calculated as __hsp length/total length * 100__).
    * __HSP Length/Total Length__ The top BLAST HSP length over the gene total length (nucleotides).
    * __Contig__: The contig id containing this gene.
    * __Start__: The start of the gene (will be greater than __End__ if on minus strand).
    * __End__: The end of the gene.
    * __Accession__: The accession of the gene from either ResFinder or PlasmidFinder  database.

    > <hands-on-title>Inspect the staramr output</hands-on-title>
    >
    > 1. View {% icon galaxy-eye %} the `detailed_summary.tsv` file
    >
    {: .hands_on}

    > <question-title>Inspect the staramr output</question-title>
    >
    > 1. What are the different sequence types that have been identified?
    > 2. Where are located the plasmid genes?
    > 3. Where are located the resistances genes?
    > 4. By looking for the accession number (M13771) on NCBI ([https://www.ncbi.nlm.nih.gov/nuccore/](https://www.ncbi.nlm.nih.gov/nuccore/)), where does the first resistance come from?
    >
    > > <solution-title>Inspect the staramr output</solution-title>
    > >
    > > 1. There are:
    > >    - 1 MSLT 
    > >    - 5 plasmid *rep* genes 
    > >    - 7 resistance genes
    > > 2. The plasmid genes are on contig00019 (3 genes - coherent with {% cite Lozano_2012 %}), contig00024, and contig00002.
    > > 3. The resistance genes are:
    > >    - 4/7 on contigs with plasmid genes (contig00002, contig00019, contig00024)
    > >    - 3/7 on contigs without plasmid genes (contig00022, contig00021)
    > > 4. M13771 is a 6'-aminoglycoside acetyltransferase phosphotransferase (AAC(6')-APH(2')) bifunctional resistance protein from *Streptococcus faecalis*. It might come from horizontal transfer from *Streptococcus faecalis* on a plasmid of *Staphylococcus aureus*.
    > {: .solution}
    >
    {: .question}

3. `resfinder.tsv`: A tabular file of each AMR gene and additional BLAST information from the **ResFinder** database, one gene per line and the following columns:

   * __Isolate ID__: The id of the isolate/genome file(s) passed to `staramr`.
   * __Gene__: The particular AMR gene detected.
   * __Predicted Phenotype__: The predicted AMR phenotype (drug resistances) for this gene.
   * __CGE Predicted Phenotype__: The CGE-predicted AMR phenotype (drug resistances) for this gene.
   * __%Identity__: The % identity of the top BLAST HSP to the AMR gene.
   * __%Overlap__: THe % overlap of the top BLAST HSP to the AMR gene (calculated as __hsp length/total length * 100__).
   * __HSP Length/Total Length__ The top BLAST HSP length over the AMR gene total length (nucleotides).
   * __Contig__: The contig id containing this AMR gene.
   * __Start__: The start of the AMR gene (will be greater than __End__ if on minus strand).
   * __End__: The end of the AMR gene.
   * __Accession__: The accession of the AMR gene in the ResFinder database.
   * __Sequence__: The AMR Gene sequence
   * __CGE Notes__: Any CGE notes associated with the prediction.

4. `plasmidfinder.tsv`: A tabular file of each AMR plasmid type and additional BLAST information from the **PlasmidFinder** database, one plasmid type per line with the following columns:

   * __Isolate ID__: The id of the isolate/genome file(s) passed to `staramr`.
   * __Plasmid__: The particular plasmid type detected.
   * __%Identity__: The % identity of the top BLAST HSP to the plasmid type.
   * __%Overlap__: The % overlap of the top BLAST HSP to the plasmid type (calculated as    __hsp length/total length * 100__).
   * __HSP Length/Total Length__ The top BLAST HSP length over the plasmid type total length (nucleotides).
   * __Contig__: The contig id containing this plasmid type.
   * __Start__: The start of the plasmid type (will be greater than __End__ if on minus strand).
   * __End__: The end of the plasmid type.
   * __Accession__: The accession of the plasmid type in the PlasmidFinder database.

5. `mlst.tsv`: A tabular file of each multi-locus sequence type (MLST) and it's corresponding locus/alleles, one genome per line with the following columns:

   * __Isolate ID__: The id of the isolate/genome file(s) passed to `staramr`.
   * __Scheme__: The scheme that `MLST` has identified.
   * __Sequence Type__: The sequence type that's assigned when combining all allele types
   * __Locus #__: A particular locus in the specified MLST scheme.

6. `settings.txt`: The command-line, database versions, and other settings used to run `staramr`.
7. `results.xlsx`: An Excel spreadsheet containing the previous files as separate worksheets.
8. `hits`: A collection with fasta files of the BLAST HSP nucleotides for the entries listed in the **resfinder.tsv** and **pointfinder.tsv** files. There are up to two files per input genome, one for ResFinder and one for PointFinder.

# Getting more information about ARG via the CARD database

To get more information about the antibiotic resistant genes (ARG), we can check the [CARD database](https://card.mcmaster.ca) (**C**omprehensive **A**ntibiotic **R**esistance **D**atabase) ({% cite Jia2016 %})

CARD can be very helpful to check all the resistance genes and check if it is logical to find the resistance gene in a specific bacteria.

> <question-title></question-title>
>
> 1. To what family does [mecA](https://card.mcmaster.ca/ontology/36911) belong?
> 2. Do you expect to find this gene in this MRSA strain and why?
> 3. Is the accession number of the entry related to the accession reported by staramr?
>
> > <solution-title></solution-title>
> >
> > 1. [Methicillin resistant PBP2](https://card.mcmaster.ca/ontology/37589)
> > 2. The strain we use is a Methicillin(multi) resistant Staphylococcus aureus. As `mecA` has a perfect resistome mach with *S. aureus*, and the AMR Gene Family is methicillin resistant PBP2, we expect to see mecA in MRSA.
> > 3. No, these are completely unrelated. Unfortunately this is a **very** common issue in bioinformatics. Everyone builds their own numbering system for entries in their database (usually calling them 'accessions'), and then someone else needs to build a service to link these databases.
> >
> {: .solution}
{: .question}

# Visualization of the ARGs and plasmid genes in their genomic context

We would like to look at the ARGs and plasmid genes in their genomic context. To do that, we will usie [JBrowse](https://jbrowse.org/jb2/) ({% cite diesh2023jbrowse %}) with several information: 

1. Assembly as the reference
2. ARGs location
3. Contigs annotation (genes, etc)
4. Coverage of the contigs from the raw reads

## Extraction of the ARG and plasmid genes location

The first step is to extract the location of the ARGs and plasmid genes on the contigs. This information is available on the `detailed_summary.tsv` output of **starmar**. The genes and their location are on the lines with a decimal value on column 6 or 7. So to only get this information, we need to select lines with a decimal value (`###.##`) followed by a tab character, the column separator in Galaxy. As a result, any lines without an identity or overlap value will be filtered out.

> <hands-on-title>Extract ARGs and plasmid genes</hands-on-title>
> 1. {% tool [Select lines that match an expression](Grep1) %} with the following parameters:
>    - {% icon param-file %} *"Select lines from"*: `detailed_summary.tsv` **staramr** output
>    - *"that"*: Matching
>    - *"the pattern"*: `[0-9]+\.[0-9]+\t`
{: .hands_on}

> <question-title></question-title>
>
> How many genes have been kept?
>
> > <solution-title></solution-title>
> >
> > 12
> >
> {: .solution}
{: .question}

This table can not be used directly in JBrowse. It first needs to be transformed in a standard format: GFF3, a file format used for describing genes and other features of DNA, RNA and protein sequences. 


> <comment-title>GFF3 file format</comment-title>
> 
> A GFF is a tab delimited file with 9 fields per line:
> 1. **seqid**: The name of the sequence where the feature is located. 
> 2. **source**: The algorithm or procedure that generated the feature. This is typically the name of a software or database.
> 3. **type**: The feature type name, like "gene" or "exon". In a well structured GFF file, all the children features always follow their parents in a single block (so all exons of a transcript are put after their parent "transcript" feature line and before any other parent transcript line). In GFF3, all features and their relationships should be compatible with the standards released by the Sequence Ontology Project.
> 4. **start**: Genomic start of the feature, with a 1-base offset. This is in contrast with other 0-offset half-open sequence formats, like BED.
> 5. **end**: Genomic end of the feature, with a 1-base offset. This is the same end coordinate as it is in 0-offset half-open sequence formats, like BED.
> 6. **score**: Numeric value that generally indicates the confidence of the source in the annotated feature. A value of "." (a dot) is used to define a null value.
> 7. **strand**: Single character that indicates the strand of the feature. This can be "+" (positive, or 5'->3'), "-", (negative, or 3'->5'), "." (undetermined), or "?" for features with relevant but unknown strands.
> 8. **phase**: phase of CDS features; it can be either one of 0, 1, 2 (for CDS features) or "." (for everything else). See the section below for a detailed explanation.
> 9. **attributes**: A list of tag-value pairs separated by a semicolon with additional information about the feature. 
{: .comment}

> <hands-on-title>Create a GFF file</hands-on-title>
> 3. {% tool [Table to GFF3](toolshed.g2.bx.psu.edu/repos/iuc/tbl2gff3/tbl2gff3/1.2) %}
>    - {% icon param-file %} *"Table"*: output of the above **Select lines** {% icon tool %} step
>    - *"Record ID column or value"*: `9`
>    - *"Start column or value"*: `10`
>    - *"End column or value"*: `11`
>    - *"Type column or value"*: `3`
>    - *"Score column or value"*: `6`
>    - *"Source column or value"*: `3`
>    - {% icon param-repeat %} *"Insert Qualifiers"*
>        - *"Name"*: `name`
>        - *"Qualifier value column or raw text"*: `2`
>    - {% icon param-repeat %} *"Insert Qualifiers"*
>        - *"Name"*: `phenotype`
>        - *"Qualifier value column or raw text"*: `4`
>    - {% icon param-repeat %} *"Insert Qualifiers"*
>        - *"Name"*: `accession`
>        - *"Qualifier value column or raw text"*: `12`
>
{: .hands_on}

> <question-title></question-title>
>
> How many genes have been found per contigs?
>
> > <solution-title></solution-title>
> >
> > - Contig 19: 3 plasmid genes, 1 ARG
> > - Contig 24: 1 plasmid gene, 2 ARGs
> > - Contig 2: 1 plasmid gene, 1 ARG
> > - Contig 22: 2 ARGs
> > - Contig 21: 1 ARG
> >
> {: .solution}
{: .question}

## Annotation of the contigs

In addition to the ARGs and plasmid genes, it would be good to have extra information about other genes on the contigs. Several tools exists to do that: Prokka ({% cite seemann2014prokka %}), Bakta ({% cite schwengers2021bakta %}), etc. Here, we use **Bakta** as recommended by Torsten Seeman as the successor of Prokka.

{% include _includes/cyoa-choices.html option1="No" option2="Yes" default="No" text="Have you already run Bakta on this dataset before?"%}

<div class="No" markdown="1">

**Bakta**  is a tool for the rapid & standardized annotation of bacterial genomes and plasmids from both isolates and MAGs. 

> <hands-on-title>Annotate the contigs</hands-on-title>
>
> 1. {% tool [Bakta](toolshed.g2.bx.psu.edu/repos/iuc/bakta/bakta/1.8.2+galaxy0) %} with the following parameters:
>    - In *"Input/Output options"*:
>        - *"The bakta database"*: latest one
>        - *"The amrfinderplus database"*: latest one
>        - {% icon param-file %} *"Select genome in fasta format"*: Contig file
>    - In *"Optional annotation"*:
>        - *"Keep original contig header (--keep-contig-headers)"*: `Yes`
{: .hands_on}

> <question-title></question-title>
>
> How many features have been identified?
>
> > <solution-title></solution-title>
> >
> > 51k+ (number of lines in the GFF)
> >
> {: .solution}
{: .question}

</div>

<div class="Yes" markdown="1">

If you already ran **Bakta** on your contigs, you can just upload here the GFF generated by **Bakta**.

> <hands-on-title>Import annotation GFF</hands-on-title>
>
> 1. {% tool [Import](upload1) %} the GFF file generated by **Bakta**
>
{: .hands_on}

![Bakta - circular plot of the annotation, with the contigs on the outside ordered clockwise by length size.](./images/bakta.svg)

</div>

## Mapping the raw reads on the contigs

To get information about the coverage of the contigs and genes, we map the reads used to build the contigs to the contigs using **Bowtie2** ({% cite langmead2009ultrafast %}).

> <hands-on-title>Annotating the Genome</hands-on-title>
> 1. {% tool [Import](upload1) %} the reads (after quality control) from another history, from [Zenodo]({{ page.zenodo_link }}) or from Galaxy shared data libraries:
>
>    ```
>    {{ page.zenodo_link }}/files/DRR187559_after_fastp_1.fastq.gz
>    {{ page.zenodo_link }}/files/DRR187559_after_fastp_2.fastq.gz
>    ```
>
> 2. {% tool [Bowtie2](toolshed.g2.bx.psu.edu/repos/devteam/bowtie2/bowtie2/2.5.0+galaxy0) %} with the following parameters:
>    - *"Is this single or paired library"*: `Paired-end`
>        - {% icon param-file %} *"FASTA/Q file #1"*: `DRR187559_after_fastp_1.fastq.gz`
>        - {% icon param-file %} *"FASTA/Q file #2"*: `DRR187559_after_fastp_2.fastq.gz`
>    - *"Will you select a reference genome from your history or use a built-in index?"*: `Use a genome from the history and build index`
>        - {% icon param-file %} *"Select reference genome"*: Contig file
>    - *"Save the bowtie2 mapping statistics to the history"*: `Yes`
>
{: .hands_on}

> <question-title></question-title>
>
> What is the alignment rate? What does that mean for the assembly?
>
> > <solution-title></solution-title>
> >
> > The alignment rate is 99.77%. Most of the reads have been mapped on the contigs. They have been then used to build the contigs.
> >
> {: .solution}
{: .question}

## Visualisation of the ARGs

We can now visualize the contigs, the mapping coverage, and the genes, using JBrowse and different information track.

> <hands-on-title>Visualize the Genome</hands-on-title>
> 1. {% tool [JBrowse](toolshed.g2.bx.psu.edu/repos/iuc/jbrowse/jbrowse/1.16.11+galaxy1) %} with the following parameters:
>    - *"Reference genome to display"*: `Use a genome from history`
>        - {% icon param-file %} *"Select the reference genome"*: Contig file
>    - *"Genetic Code"*: `11. The Bacterial, Archaeal and Plant Plastid Code`
>    - In *"Track Group"*:
>        - {% icon param-repeat %} *"Insert Track Group"*
>            - *"Track Category"*: `Bakta`
>            - In *"Annotation Track"*:
>                - {% icon param-repeat %} *"Insert Annotation Track"*
>                    - *"Track Type"*: `GFF/GFF3/BED Features`
>                        - {% icon param-file %} *"GFF/GFF3/BED Track Data"*: `Annotation_and_sequences` (output of **Bakta** {% icon tool %})
>                        - *"JBrowse Track Type [Advanced]"*: `Neat Canvas Features`
>                        - *"Track Visibility"*: `On for new users`
>        - {% icon param-repeat %} *"Insert Track Group"*
>            - *"Track Category"*: `ARGs and plasmid genes`
>            - In *"Annotation Track"*:
>                - {% icon param-repeat %} *"Insert Annotation Track"*
>                    - *"Track Type"*: `GFF/GFF3/BED Features`
>                        - {% icon param-file %} *"GFF/GFF3/BED Track Data"*: Output of **Table to GFF3**
>                        - *"JBrowse Track Type [Advanced]"*: `Neat Canvas Features`
>                        - *"Track Visibility"*: `On for new users`
>        - {% icon param-repeat %} *"Insert Track Group"*
>            - *"Track Category"*: `Coverage`
>            - In *"Annotation Track"*:
>                - {% icon param-repeat %} *"Insert Annotation Track"*
>                    - *"Track Type"*: `BAM Pileups`
>                        - {% icon param-file %} *"BAM Track Data"*: Bowtie2's output
>                        - *"Autogenerate SNP Track"*: `Yes`
>
> 3. View the output of JBrowse
>
{: .hands_on}

In the output of the JBrowse you can view the mapped reads and the found genes against the reference genome. With the search tools you can easily find genes of interest. JBrowse can handle many inputs and can be very useful. Using the Bowtie2 mapping output, low coverage regions can be detected. This SNP detection can also give a clear view of where the data was less reliable or where variations were located.

If it takes too long to build the JBrowse instance, you can view an embedded one here. (**Warning**: feature name search will not work.)

{% snippet topics/visualisation/faqs/visualizations_jbrowse.html datadir="data" loc="contig00019:81..23760" tracks="DNA,878c908bab1edf92fb1577fad80f7bae_0,a7fe8304cddfe5fbececdc7a2bd7ed18_0," %}

> <question-title></question-title>
>
> 1. What is the name for the gene found by Bakta corresponding to the rep16 gene found by staramr?
> 2. What are the NCBI Protein id and UniRef id for aac(6')-aph(2'')?
>
> > <solution-title></solution-title>
> >
> > 1. By zooming at the beginning of the contig, the name for rep16 is DUF536 domain-containing protein
> > 2. aac(6')-aph(2'') is on contig00019:20311..21749. The corresponding gene found by Bakta is bifunctional aminoglycoside N-acetyltransferase AAC(6')-Ie/aminoglycoside O-phosphotransferase APH(2'')-Ia. When clicking on it, we can find the dbxref attributes:
> >   - NCBI Protein: WP_001028144.1
> >   - UniRef: UniRef50_P0A0C2
> {: .solution}
{: .question}

# Conclusion

In this tutorial, contigs were scaned for AMR and plasmid genes. The genes were then visualized in their genomic context after contig annotation.
