---
layout: introduction_slides
logo: "GTN"
title: "Introduction to Metabolomics"
type: "introduction"
contributors:
  - lecorguille
  - cecilecanlet
---

# Metabolomics

.pull-left[![An archaic image of a large wheel with substances.](../images/slide-intro-wheel.png)]
.pull-right[![A cartoon of genes pointing to a genome, mRNA to a transcriptome, proteins to a proteome, and metabolites to a metabolome. All of these are connected by various lines, and finally point to function at the bottom.](../images/slide-intro-omics.png)]

«Quantitative measurement of the dynamic multiparametric metabolic response of living systems to pathophysiological stimuli or genetic modification »
(Nicholson, 1999)

---

## Definitions

- Metabolomics
  - Newly emerging field of 'omics' research
  - Comprehensive and simultaneous systematic determination of metabolite levels  in the metabolome and their changes over time as a consequence of stimuli
- Metabolome
  - Refers to the complete set of small-molecule metabolites
  - Dynamic
- Metabolites
  - Intermediates and products of metabolism
  - Examples include antibiotics, pigments, carbohydrates, fatty acids and amino acids
  - Primary and secondary metabolites

---

## Definitions

.pull-left[
Metabolo **_l_** omics
.left[
Metabolic 'fingerprints' (unique for each type of cell or tissue) based on comprehensive quantification

i.e. comprehensive and simultaneous systematic determination of metabolite levels in a biological system (cell, tissue, fluid)
]]

.pull-right[
Metabolo **_n_** omics
.left[
Description of metabolic changes

i.e. detection of disrupted metabolic pathways
]]

---

## Metabolomics : the world of small molecules

.pull-left[
**Physico-chemical diversity**
- Comparison with
  - DNA/RNA: 4 bases
  - Proteins: 20 amino acids
  - Common physico-chemical properties
  - Extraction/Analysis: easy automation

- Metabolites
  - Number ≥ 150 000 in Nature; most not identified
  - Wide diversity of structures, functional groups, physico-chemical properties
  - e.g. lipids, sugars, amino acids, etc.
  - High turn-over rates (&lt;sec)
  - Range from 1 pM to 100 mM
]

.pull-right[
**Molecular mass range**
![Histogram comparing E. coli and S. cerevisiae and their molecular weight ranges and frequency.](../images/slide-intro-molecular_mass_range.png)
<small>80% of metabolites with mol. mass ≤ 600 (E. coli, S. cerevisiae)</small>
]

---

## Analytical challenge

![Large pathway map with text over it reading: targeted/global, sensitity coverage. Great number of metabolites, 2.5k to 15k. Chemical diversity of metabolites, mass, pKa, polarity, volatiity. Variable concentrations. NMR vs MS. Universality, Sensitivity, Can get both structural and quantitative information. And in red in the center: The metabolome analysis relies on combinations of approaches.](../images/slide-intro-metabolome_pathway.png)

---

## Complementary analytical tools
### For metabolome investigation

![MS produces m/z masses, and ms/ms structure. NMR produces other spectra. A stacked bar chart compares use of ms only, nmr only, or both across years of publications.](../images/slide-intro-ms_nmr_complementarity.png)

---

## Strategy: untargeted metabolomics

or metabolic fingerprinting

![A chart shows an exposed and control group, different tissues being extracted and analysed with NMR, MS, etc. The spectra is processed to reduce the data, and multivariate statistics are done to identify metabolites like glucose, taurine, glycine, etc.](../images/slide-intro-untargeted_metabolomics.png)

---

## Strategy: targeted metabolomics

Qualitative and quantitative analysis

![Samples are extracted, run through GC(-MS) and LC(-MS) to produce metabolome coverage and lists of classes like lipids, polar metabolites, xenobiotics, and other classes.](../images/slide-intro-targeted_metabolomics.png)
