<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab / Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab / Climate</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@4bbac78b4791e7e0a5f3fcee9e506ed0d1e4d691",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.20cf1f17.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-12-05 14:44:50 +0000">
<meta name="DC.creator" content="Anne Fouilloux"><meta name="description" content="Learn to analyze climate data through Galaxy.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Climate / Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab / Hands-on: Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab">
        <meta property="og:description" content="Learn to analyze climate data through Galaxy.">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2023-12-05 14:44:50 +0000",
  "datePublished": "2020-10-25 21:05:55 +0000",
  "identifier": "https://gxy.io/GTN:T00043",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Climate",
    "description": "Learn to analyze climate data through Galaxy.",
    "url": "https://training.galaxyproject.org/training-material/topics/climate/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab",
  "url": "https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html",
  "version": 15,
  "timeRequired": "PT6H",
  "teaches": "Motivation for using the Galaxy Climate JupyterLab for CLM-FATES.\n - Setting up CLM-FATES case with Galaxy Climate JupyterLab.\n - Running CLM-FATES in Galaxy for single-point locations where in-situ measurements are available.\n - Analyzing CLM-FATES results.\n - Sharing CLM-FATES simulations.\n - Composing, executing and publishing the corresponding Jupyter notebooks.",
  "keywords": [
    "Climate",
    "interactive-tools"
  ],
  "description": "The questions this  addresses are:\n - Why and when using Galaxy Climate JupyterLab for CLM-FATES?\n - How to start Galaxy Climate JupyterLab in Galaxy?\n - How to upload input data for running CLM-FATES?\n - How to create CLM-FATES case in Galaxy Climate JupyterLab?\n - How to customize your run?\n - How to analyze your model outputs?\n - How to save your model results into a Galaxy history?\n - How to share your results?\n\n\nThe objectives are:\n - Motivation for using the Galaxy Climate JupyterLab for CLM-FATES.\n - Setting up CLM-FATES case with Galaxy Climate JupyterLab.\n - Running CLM-FATES in Galaxy for single-point locations where in-situ measurements are available.\n - Analyzing CLM-FATES results.\n - Sharing CLM-FATES simulations.\n - Composing, executing and publishing the corresponding Jupyter notebooks.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates/slides.html",
      "name": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES)",
      "description": "Slides for 'Functionally Assembled Terrestrial Ecosystem Simulator (FATES)' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates/tutorial.html",
      "name": "Functionally Assembled Terrestrial Ecosystem Simulator (FATES)",
      "description": "Hands-on for 'Functionally Assembled Terrestrial Ecosystem Simulator (FATES)' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html",
      "name": "JupyterLab in Galaxy",
      "description": "Hands-on for 'JupyterLab in Galaxy' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@type": "CreativeWork",
      "url": "https://swcarpentry.github.io/python-novice-inflammation/",
      "name": "Programming with Python"
    },
    {
      "@type": "CreativeWork",
      "url": "http://swcarpentry.github.io/shell-novice/",
      "name": "The Unix Shell"
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annefou/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annefou/",
      "name": "Anne Fouilloux",
      "image": "https://avatars.githubusercontent.com/annefou",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-1784-2920",
      "orcid": "https://orcid.org/0000-0002-1784-2920"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Climate",
      "description": "Learn to analyze climate data through Galaxy.",
      "url": "https://training.galaxyproject.org/training-material/topics/climate/"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.4108341",
      "name": "Associated Training Datasets"
    }
  ],
  "abstract": "The practical aims at familiarizing you with running CLM-FATES within Galaxy Climate JupyterLab."
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/climate" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Climate
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/climate/faqs/" title="Check our FAQs for the Climate topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/climate/tutorials/fates-jupyterlab/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-climate">
    <h1 data-toc-skip>Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/annefou/" class="contributor-badge contributor-annefou"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/annefou?s=36" alt="Anne Fouilloux avatar" width="36" class="avatar" />
    Anne Fouilloux</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Why and when using Galaxy Climate JupyterLab for CLM-FATES?</p>
</li>
        
        <li><p>How to start Galaxy Climate JupyterLab in Galaxy?</p>
</li>
        
        <li><p>How to upload input data for running CLM-FATES?</p>
</li>
        
        <li><p>How to create CLM-FATES case in Galaxy Climate JupyterLab?</p>
</li>
        
        <li><p>How to customize your run?</p>
</li>
        
        <li><p>How to analyze your model outputs?</p>
</li>
        
        <li><p>How to save your model results into a Galaxy history?</p>
</li>
        
        <li><p>How to share your results?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Motivation for using the Galaxy Climate JupyterLab for CLM-FATES.</p>
</li>
        
        <li><p>Setting up CLM-FATES case with Galaxy Climate JupyterLab.</p>
</li>
        
        <li><p>Running CLM-FATES in Galaxy for single-point locations where in-situ measurements are available.</p>
</li>
        
        <li><p>Analyzing CLM-FATES results.</p>
</li>
        
        <li><p>Sharing CLM-FATES simulations.</p>
</li>
        
        <li><p>Composing, executing and publishing the corresponding Jupyter notebooks.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/climate/tutorials/fates/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Functionally Assembled Terrestrial Ecosystem Simulator (FATES)</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/climate/tutorials/fates/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Functionally Assembled Terrestrial Ecosystem Simulator (FATES)</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/galaxy-interface/tutorials/jupyterlab/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: JupyterLab in Galaxy</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    

    
      <li>
        <a href="https://swcarpentry.github.io/python-novice-inflammation/">Programming with Python</a>
      </li>
    

    
      <li>
        <a href="http://swcarpentry.github.io/shell-novice/">The Unix Shell</a>
      </li>
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 6 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.4108341">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/climate/tutorials/fates-jupyterlab/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/climate/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Oct 25, 2020 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Dec 5, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00043">https://gxy.io/GTN:T00043</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 15 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>The practical aims at familiarizing you with running CLM-FATES within Galaxy Climate JupyterLab.</p>

<blockquote class="agenda">
  <h2 id="agenda">Agenda</h2>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#agenda" id="markdown-toc-agenda">Agenda</a></li>
  <li><a href="#motivation" id="markdown-toc-motivation">Motivation</a></li>
  <li><a href="#step-1-get-data" id="markdown-toc-step-1-get-data">Step-1: Get data</a></li>
  <li><a href="#step-2-opening-up-climate-jupyterlab" id="markdown-toc-step-2-opening-up-climate-jupyterlab">Step-2: Opening up Climate JupyterLab</a></li>
  <li><a href="#step-3-create-a-new-session-clm-fates-in-jupyterlab" id="markdown-toc-step-3-create-a-new-session-clm-fates-in-jupyterlab">Step-3: Create a new session CLM-FATES in JupyterLab</a>    <ol>
      <li><a href="#import-input-data-to-jupyterlab" id="markdown-toc-import-input-data-to-jupyterlab">Import input data to JupyterLab</a></li>
      <li><a href="#get-clm-fates-emerald-release" id="markdown-toc-get-clm-fates-emerald-release">Get CLM-FATES EMERALD release</a></li>
      <li><a href="#create-clm-fates-new-case" id="markdown-toc-create-clm-fates-new-case">Create CLM-FATES new case</a></li>
      <li><a href="#setup-build-and-submit-your-first-simulation" id="markdown-toc-setup-build-and-submit-your-first-simulation">Setup, build and submit your first simulation</a></li>
      <li><a href="#check-your-run" id="markdown-toc-check-your-run">Check your run</a></li>
      <li><a href="#customize-your-run" id="markdown-toc-customize-your-run">Customize your run</a></li>
    </ol>
  </li>
  <li><a href="#analysis" id="markdown-toc-analysis">Analysis</a>    <ol>
      <li><a href="#analyzing-fates-clm-model-outputs" id="markdown-toc-analyzing-fates-clm-model-outputs">Analyzing FATES-CLM model outputs</a></li>
    </ol>
  </li>
  <li><a href="#save-your-results-to-your-galaxy-history" id="markdown-toc-save-your-results-to-your-galaxy-history">Save your results to your Galaxy history</a></li>
  <li><a href="#share-your-work" id="markdown-toc-share-your-work">Share your work</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-background"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Background</div>

  <p>FATES is the “Functionally Assembled Terrestrial Ecosystem Simulator”.
FATES needs what we call a “Host Land Model” (HLM) to run and in this tutorial
we will be using the <a href="http://www.cesm.ucar.edu/models/clm/">Community Land Model</a>
of the <a href="https://github.com/ESCOMP/CTSM">Community Terrestrial Systems Model</a> (CLM-CTSM).
FATES was derived from the CLM Ecosystem Demography model (CLM(ED)), which was documented in
<span class="citation"><a href="#Fisher2015"><b>Taking off the training wheels: the properties of a dynamic vegetation model without climate envelopes, CLM4.5(ED)</b> 2015</a></span>.
and this technical note was first published as an appendix to <a href="https://pdfs.semanticscholar.org/396c/b9f172cb681421ed78325a2237bfb428eece.pdf">that paper</a>.
The <a href="https://fates-docs.readthedocs.io/en/latest/index.html">FATES documentation</a> will provide some more insight on FATES too.</p>

</blockquote>

<h2 id="motivation">Motivation</h2>

<p>In this tutorial, we will be using a Galaxy interactive tool called Galaxy Climate JupyterLab for running CLM-FATES.
This interactive tool is only available from <a href="https://live.usegalaxy.eu/">LiveGalaxy.eu</a>.</p>

<p>This platform is meant to be used for:</p>
<ul>
  <li>preparing new input datasets;</li>
  <li>testing new versions of fates, including code changes;</li>
  <li>developing new notebooks for analyzing and showing the model results;</li>
  <li>creating interactive publications (notebooks);</li>
  <li>teaching purposes.</li>
</ul>

<p>The main advantage over the Galaxy CLM-FATES tool is that you can run any versions of FATES, including developments
that are not released yet. However, it is not recommended to run more than a few decades of simulation time.
In that particular case, we would suggest to use the Galaxy CLM-FATES Galaxy tool.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-cml-fates-in-jupyterlab-versus-clm-fates-galaxy-tool"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: CML-FATES in JupyterLab versus CLM-FATES Galaxy tool</div>
  <p>Do not use the interactive Galaxy Climate JupyterLab for running long and “operational” simulations and
do not forget that you need to save back your results to your Galaxy history or local machine before
stopping your JupyterLab.</p>

</blockquote>

<h1 id="step-1-get-data">Step-1: Get data</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial. If you are not inspired, you can name it <em>fates-jupyterlab</em>.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the files from <a href="https://doi.org/10.5281/zenodo.4108341">Zenodo</a> or from the shared data library</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4108341/files/inputdata_version2.0.0_ALP1.tar
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Check the datatype is <strong>tar</strong></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">datatypes</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename Datasets</p>

      <p>As “https://zenodo.org/record/4108341/files/inputdata_version2.0.0_ALP1.tar” is not a beautiful name and can give errors for some tools,
it is a good practice to change the dataset name by something more meaningfull. For example by removing <code class="language-plaintext highlighter-rouge">https://zenodo.org/record/4108341/files/</code> to obtain <code class="language-plaintext highlighter-rouge">inputdata_version2.0.0_ALP1.tar</code>, respectively.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Add a tag to the dataset corresponding to <code class="language-plaintext highlighter-rouge">fates-jupyterlab</code></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="step-2-opening-up-climate-jupyterlab">Step-2: Opening up Climate JupyterLab</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-launch-jupyterlab-for-ocean-atmosphere-land-climate-python-ecosystem-in-galaxy"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Launch JupyterLab for Ocean / Atmosphere / Land / Climate Python ecosystem in Galaxy</div>

  <p>Currently JupyterLab for Ocean / Atmosphere / Land / Climate Python ecosystem in Galaxy is available on <a href="https://live.usegalaxy.eu">Live.useGalaxy.eu</a> only. JupyterLab for Ocean / Atmosphere / Land / Climate Python ecosystem and not the default JupyterLab in Galaxy contains all the python packages and additional software we need for running Earth System Model, including Functionally Assembled Terrestrial Ecosystem Simulator (FATES). The default JupyterLab in Galaxy would not be sufficient for executing all the tasks in th
is tutorial.</p>

  <ol>
    <li>Open the <span class="tool" data-tool="interactive_tool_jupyter_notebook" title="JupyterLab tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>JupyterLab</strong></span> or opening directly on <a href="https://live.usegalaxy.eu/?tool_id=interactive_tool_climate_notebook">live.usegalaxy</a></li>
    <li>Click Run Tool</li>
    <li>The tool will start running and will stay running permanently</li>
    <li>Click on the “User” menu at the top and go to “Active Interactive Tools” and locate the JupyterLab instance you started.</li>
    <li>Click on your JupyterLab instance (please not that it may take a few minutes before you can click on the link to your jupyterLab instance).</li>
  </ol>

</blockquote>

<p>You should now be looking at a page with the JupyterLab interface:</p>

<p><a href="../../images/jupyterlab_climate_session.png" rel="noopener noreferrer"><img src="../../images/jupyterlab_climate_session.png" alt="Jupyterlab climate session interface. " width="1201" height="794" loading="lazy" /></a></p>

<h1 id="step-3-create-a-new-session-clm-fates-in-jupyterlab">Step-3: Create a new session CLM-FATES in JupyterLab</h1>

<h2 id="import-input-data-to-jupyterlab">Import input data to JupyterLab</h2>

<p>In this part of the tutorial, we will be using the existing Jupyter Notebook called <strong>ipython_galaxy_notebook.ipynb</strong></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-open-a-jupyterlab-terminal"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Open a JupyterLab Terminal</div>
  <p>To open <strong>ipython_galaxy_notebook.ipynb</strong>, double click on it.
More information on the JupyterLab interface can be found on the <a href="https://jupyterlab.readthedocs.io/en/stable/user/interface.html">JupyterLab documentation</a>.</p>

  <p>Import the FATES input dataset from your history:</p>
  <ul>
    <li>In a new code cell:</li>
  </ul>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash

get -i inputdata_version2.0.0_ALP1.tar -t name
</code></pre></div>  </div>

  <p>By default code cells execute Python 3 code (default kernel) so to execute the Shell command lines
we will use <code class="language-plaintext highlighter-rouge">%%bash</code>.  In that case the cell runs with bash in a subprocess.</p>

  <p>Then untar this file:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
mkdir $HOME/inputdata
tar xf /import/inputdata_version2.0.0_ALP1.tar --directory $HOME/inputdata
</code></pre></div>  </div>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-direct-download-in-jupyterlab-from-zenodo"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Direct download in JupyterLab from Zenodo</div>

  <p>You may also download the input dataset directly from Zenodo.</p>
  <ul>
    <li>Open a JupyterLab Terminal and enter the following command:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
cd /import
wget https://zenodo.org/record/4108341/files/inputdata_version2.0.0_ALP1.tar
</code></pre></div>      </div>
    </li>
  </ul>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-using-jupyterlab-terminal"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using JupyterLab Terminal</div>

  <p>Most of the tutorial (except visualization) can be executed from a JupyterLab Terminal. In that case,
you should not add <code class="language-plaintext highlighter-rouge">%%bash</code> to your commands. More on JupyterLab Terminal can be found on <a href="https://jupyterlab.readthedocs.io/en/stable/user/terminal.html">Read the Docs</a>.</p>
</blockquote>

<h2 id="get-clm-fates-emerald-release">Get CLM-FATES EMERALD release</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-clone-clm-fates-for-nordic-sites"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Clone CLM-FATES for Nordic sites</div>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
conda create --name fates -y fates-emerald=2.0.1
</code></pre></div>  </div>
  <p>The command above is required once only. It creates a new conda environment called <strong>fates</strong> and
install <strong>fates-emerald</strong> version 2.0.1 conda package. It is important to always specify the version of
CLM-FATES you would like to use as it needs to match your input dataset.
Now a new <strong>fates</strong> conda environment has been
created in your current JupyterLab session and can be use every time you activate it.</p>

  <p>Then to activate this new conda environment:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
source activate fates
</code></pre></div>  </div>
  <p>Please note that you would need to activate <strong>fates</strong> environment in every new code cell (because it starts a new Shell subprocess).</p>

</blockquote>

<h2 id="create-clm-fates-new-case">Create CLM-FATES new case</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-create-clm-fates-new-case-for-alp1-site"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Create CLM-FATES new case for ALP1 site</div>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
source activate fates

create_newcase --case $HOME/ctsm_cases/fates_alp1 --compset 2000_DATM%1PTGSWP3_CLM50%FATES_SICE_SOCN_MOSART_SGLC_SWAV --res 1x1_ALP1 --machine espresso --run-unsupported
</code></pre></div>  </div>

</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-command-not-found"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Command not found!</div>
  <p>If you get an error when invoking <code class="language-plaintext highlighter-rouge">create_newcase</code> make sure you have switch to fates conda environment:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
source acticate fates
create_newcase --help
</code></pre></div>  </div>
</blockquote>

<p>The 4 main arguments of create_newcase are explained on the figure below: <a href="../../images/newcase_fates.png" rel="noopener noreferrer"><img src="../../images/newcase_fates.png" alt="Snapshot of create_newcase main arguments showing: what is the case name? which resolution? which model configuration? which set of components? which machine are you running on?" width="805" height="305" loading="lazy" /></a>.</p>

<ul>
  <li><strong>case</strong>: specifies the name and location of the case being created. It creates a new case in <code class="language-plaintext highlighter-rouge">$HOME/ctsm_cases/</code> and its name is <code class="language-plaintext highlighter-rouge">fates_alp1</code>. make sure to give a meaningful name to your FATES experiments.</li>
  <li><strong>res</strong>: specifies the model resolution (resolution of the grid). Here <strong>1x1_ALP1</strong> corresponds to a single point resolution.</li>
  <li><strong>compset</strong>: specifies the component set, i.e., component models, forcing scenarios and physics options for those models.
    <ul>
      <li>The long name of the compset we have chosen is <code class="language-plaintext highlighter-rouge">2000_DATM%1PTGSWP3_CLM50%FATES_SICE_SOCN_MOSART_SGLC_SWAV</code></li>
      <li>The notation for the compset longname is: <code class="language-plaintext highlighter-rouge">TIME_ATM[%phys]_LND[%phys]_ICE[%phys]_OCN[%phys]_ROF[%phys]_GLC[%phys]_WAV[%phys][_BGC%phys]</code></li>
      <li>The compset longname has the specified order: <strong>atm, lnd, ice, ocn, river, glc wave cesm-options</strong> where:
        <ul>
          <li><strong>Initialization Time</strong>:2000</li>
          <li><strong>Atmosphere</strong>: Data atmosphere DATM%1PTGSWP3</li>
          <li><strong>Land</strong>: CLM50%FATES</li>
          <li><strong>Sea-Ice</strong>: SICE Stub ICE</li>
          <li><strong>Ocean</strong>: SOCN Stub ocean</li>
          <li><strong>River runoff</strong>:MOSART: MOdel for Scale Adaptive River Transport</li>
          <li><strong>Land Ice</strong>: SGLC Stub Glacier (land ice) component</li>
          <li><strong>Wave</strong>-   SWAV Stub wave component
 See also the <a href="http://www.cesm.ucar.edu/models/cesm2/config/compsets.html">list of available component sets</a>.</li>
        </ul>
      </li>
    </ul>
  </li>
  <li><strong>mach</strong>: specifies the machine where CLM-FATES will be compiled and run. We use <code class="language-plaintext highlighter-rouge">espresso</code> which is the local setup (see <code class="language-plaintext highlighter-rouge">$HOME/.cime/</code> folder).</li>
</ul>

<h2 id="setup-build-and-submit-your-first-simulation">Setup, build and submit your first simulation</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-setup-build-and-submit"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Setup, build and submit</div>

  <p>Check the content of the directory and browse the sub-directories:</p>
  <ul>
    <li>CaseDocs: namelists or similar</li>
    <li>SourceMods: this is where you can add local source code changes.</li>
    <li>Tools: a few utilities (we won’t use them directly)</li>
    <li>Buildconf: configuration for building each component
For this tutorial, we  wish to have a “cold” start as we are mostly interested in setting up our model.
When ready to run in production, the model needs to be spin-up (run for several centuries until it reaches some kind of equilibrium).</li>
  </ul>

  <p>We will first make a short simulation (6 months):</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
source activate fates

cd $HOME/ctsm_cases/fates_alp1
./case.setup
./case.build
./xmlchange STOP_OPTION=nmonths            # set the simulation periods to "nmonths"
./xmlchange STOP_N=6                       # set the length of simulation, i.e, how many months
./case.submit &gt; case_submit.out 2&gt;&amp;1
</code></pre></div>  </div>
  <p>The step above can take a lot of time because it needs to compile and run the FATES model.
Therefore we suggest you make a break and come back later (or the following day) before you continue the
tutorial.</p>
</blockquote>

<h2 id="check-your-run">Check your run</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-check-your-simulation"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: check your simulation</div>

  <ol>
    <li>From a new code cell:</li>
  </ol>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash

cd $HOME/work/fates_alp1
ls -la
</code></pre></div>  </div>
  <p>You should see two folders:</p>
  <ul>
    <li>bld: contains the object and CESM executable (called cesm.exe) for your configuration</li>
    <li>run: this directory will be used during your simulation run to generate output files, etc.</li>
  </ul>

  <p>The <strong>bld</strong> folder contains the model executable (called <code class="language-plaintext highlighter-rouge">cesm.exe</code>) while <strong>run</strong> contains all the files used for running CLM-FATES (and not already archived).
Once your run is terminated, many files are moved from the <strong>run</strong> folder to the <strong>archive</strong> folder:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
cd $HOME/archive/fates_alp1
ls lnd/hist
</code></pre></div>  </div>

  <p>We are interested in the “history” files from the CLM-FATES model and these files are all located in <code class="language-plaintext highlighter-rouge">lnd/hist</code> folder.
You can also check other model components in the archive directory (atm, etc.): in our case, it is not of a great interest as
we are running the CLM-FATES component.
We have run a very short simulation and get one file only, called <code class="language-plaintext highlighter-rouge">fates_alp1_t.clm2.h0.2000-01.nc</code>. The
CLM-FATES model outputs are stored in netCDF format.</p>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-what-is-a-netcdf-file"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What is a netCDF file?</div>

    <p>Netcdf stands for “network Common Data Form”. It is self-describing, portable, metadata friendly, supported by many languages
(including python, R, fortran, C/C++, Matlab, NCL, etc.), viewing tools (like panoply, ncview/ncdump) and tool suites of file operators (in particular NCO and CDO).</p>
  </blockquote>

  <ol>
    <li>Create a new Jupyter Notebook for analyzing your results:
      <ul>
        <li>From the <strong>File Menu</strong> –&gt; <strong>New</strong> –&gt; <strong>Notebook</strong>:</li>
      </ul>
      <ul>
        <li>Rename your notebook to <strong>check_analysis.ipynb</strong></li>
        <li>All the analysis of the 6 month FATES simulation will be done from this notebook</li>
      </ul>
    </li>
    <li>Get metadata
In a Code cell:</li>
  </ol>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>import os
import xarray as xr
xr.set_options(display_style="html")
%matplotlib inline

case = 'fates_alp1'
path = os.path.join(os.getenv('HOME'), 'archive', case, 'lnd', 'hist')
dset = xr.open_mfdataset(path + '/*.nc', combine='by_coords')
dset
</code></pre></div>  </div>
  <p>As shown above, we are now using Python 3 for analyzing the results and <a href="http://xarray.pydata.org/en/stable/">xarray</a> which
is a Python package that can easily handle <a href="https://en.wikipedia.org/wiki/NetCDF">netCDF</a> files.
we opened all the history files we have produced and print metadata.</p>

  <ol>
    <li>Plotting 1D variables (timeseries)</li>
  </ol>

  <p>You can select a variable by using its short name (see metadata above) and then calling the plot method:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>dset['AREA_TREES'].plot()
</code></pre></div>  </div>
  <p>As we ran 6 months only, we have very little points in our timeseries!</p>

  <p>To plot 2D variables such as <strong>CANOPY_AREA_BY_AGE</strong>, you can use the <em>col_wrap</em> option when plotting:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>dset['CANOPY_AREA_BY_AGE'].plot(aspect=3, size=6, col='fates_levage', col_wrap=1)
</code></pre></div>  </div>
  <p>In the plot above, we have one plot per row (<em>col_wrap=1</em>) and we will have a plot for each value of the <em>fates_levage</em> dimension.
We also changed the aspect of the plot (aspect=3, size=6).</p>
</blockquote>

<h2 id="customize-your-run">Customize your run</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-10-years"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run 10 years</div>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
source activate fates
cd $HOME/ctsm_cases/fates_alp1

./xmlchange RUN_STARTDATE=0001-01-01      # set up the starting date of your simulation
./xmlchange STOP_OPTION=nyears            # set the simulation periods to "years"
./xmlchange STOP_N=5                      # set the length of simulation, i.e, how many years
./xmlchange CONTINUE_RUN=TRUE             # if you want to continue your simulation from restart file, set it to TRUE
./xmlchange RESUBMIT=1                    # set up how many times you want to resubmit your simulation.
                                          # e.g, STOP_N=5, RESUBMIT=1, you will have simulation for 5+5*1=10
./xmlchange DATM_CLMNCEP_YR_START=1901    # set up the start year of the atmospheric forcing
./xmlchange DATM_CLMNCEP_YR_END=1910      # set up the end year of the atmospheric forcing
./case.submit &gt; case_submit_sontinue_run.out 2&gt;&amp;1
</code></pre></div>  </div>
  <p>This step will take several hours.</p>
</blockquote>

<h1 id="analysis">Analysis</h1>

<p>In this section, we will be able to analyze your 10 year simulation only when the run is terminated (note that data will be moved to the archive folder every 5 years).</p>

<h2 id="analyzing-fates-clm-model-outputs">Analyzing FATES-CLM model outputs</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-open-a-new-python-notebook"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Open a new Python notebook</div>
  <ul>
    <li>Create a notebook by clicking the <code class="language-plaintext highlighter-rouge">+</code> button in the file browser and then selecting a kernel in the new Launcher tab:</li>
    <li>Rename your notebook to <strong>analyse_case.ipynb</strong>
Get more information online at <a href="https://jupyterlab.readthedocs.io/en/stable/user/notebook.html">JupyterLab notebooks</a>.</li>
  </ul>
</blockquote>

<h3 id="use-xarray-to-read-and-plot">Use <code class="language-plaintext highlighter-rouge">xarray</code> to read and plot</h3>

<p>In this section, we give additional examples on how to visualize your results using <code class="language-plaintext highlighter-rouge">xarray</code>:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>import xarray as xr
xr.set_options(display_style="html")
%matplotlib inline

case = 'fates_alp1'
path = os.path.join(os.getenv('HOME'), 'archive', case, 'lnd', 'hist')
dset = xr.open_mfdataset(path + '/*.nc', combine='by_coords')
dset
</code></pre></div></div>
<p>As you can see, we are now using <code class="language-plaintext highlighter-rouge">open_mfdataset</code> to read all the netCDF files available in the history folder.
The option <code class="language-plaintext highlighter-rouge">combine='by_coords')</code> is used to tell the method <code class="language-plaintext highlighter-rouge">open_mfdataset</code> how to combine the different files
together.</p>

<p>You can use the same <code class="language-plaintext highlighter-rouge">plot</code> method as before for plotting any variable. For instance:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>dset['AREA_TREES'].plot(aspect=3, size=6)

</code></pre></div></div>

<p>For saving your plot, for instance in a <em>png</em> file format:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>import matplotlib.pyplot as plt
fig = plt.figure(1, figsize=[14,7])
ax = plt.subplot(1, 1, 1)
dset['AREA_TREES'].plot(ax=ax)
ax.set_title(dset['AREA_TREES'].long_name)
fig.savefig('AREA_TREES.png')
</code></pre></div></div>

<p><a href="../../images/AREA_TREES_10yrs.png" rel="noopener noreferrer"><img src="../../images/AREA_TREES_10yrs.png" alt="AREA TREES (10 years). " width="1008" height="504" loading="lazy" /></a></p>

<p>In the plot above, we create a figure (with specific dimension [14,7]) and one subplot with one row and one column.
The last argument of <em>subplot</em> is the index (1) of this particular subplot.</p>

<p>Finally, the resulting figure is saved in a file called ‘AREA_TREES.png’.</p>

<p>To plot 2D variables and save the resulting plot in a png file, for instance <strong>CANOPY_HEIGHT_DIST</strong>:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>p = dset['CANOPY_HEIGHT_DIST'].plot(aspect=3, size=6, col_wrap=1, col='fates_levheight')
p.fig.savefig('CANOPY_HEIGHT_DIST.png')
</code></pre></div></div>

<p><a href="../../images/CANOPY_HEIGHT_DIST_10yrs.png" rel="noopener noreferrer"><img src="../../images/CANOPY_HEIGHT_DIST_10yrs.png" alt="CANOPY HEIGHT DIST (10 years). " width="1368" height="2592" loading="lazy" /></a></p>

<h1 id="save-your-results-to-your-galaxy-history">Save your results to your Galaxy history</h1>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-put-your-data-to-your-galaxy-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Put your data to your Galaxy history</div>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
cd $HOME
tar cvf archive_emerald_fates_test.tar archive
</code></pre></div>  </div>
  <p>Then you are now ready to put your dataset into Galaxy. As it can be large, we recommend to use FTP:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>curl -T {"archive_emerald_fates_test.tar"} ftp://ftp.usegalaxy.eu --user USER:PASSWORD --ssl
</code></pre></div>  </div>
  <p>Where you replace <code class="language-plaintext highlighter-rouge">USER</code> by your galaxy username (what you used to login to Galaxy i.e. usually your email address
and <code class="language-plaintext highlighter-rouge">PASSWORD</code> by your Galaxy login password.</p>

  <p>To get <code class="language-plaintext highlighter-rouge">archive_emerald_fates_test.tar</code> in your history:</p>
  <ul>
    <li>Open the Galaxy Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</li>
    <li>Click on <strong>Choose FTP files</strong> and select <code class="language-plaintext highlighter-rouge">archive_emerald_fates_test.tar</code> to import it into your history.</li>
  </ul>

  <p>And make sure to save all your notebooks to your Galaxy history too:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
put -p ipython_galaxy_notebook.ipynb
put -p check_analysis.ipynb
put -p analyse_case.ipynb
put -p AREA_TREES.png
put -p CANOPY_HEIGHT_DIST.png
</code></pre></div>  </div>
</blockquote>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-danger-you-can-lose-data"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Danger: You can lose data!</div>
  <p>If you do not copy data (FATES model results, jupyter notebooks, plots, etc.) before you stop your Galaxy climate
JupyterLab tool, all your results will be lost!</p>
</blockquote>

<h1 id="share-your-work">Share your work</h1>

<p>One of the most important features of Galaxy comes at the end of an analysis. When you have published striking findings, it is important that other researchers are able to reproduce your in-silico experiment. Galaxy enables users to easily share their workflows and histories with others.</p>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-sharing-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sharing-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sharing your History</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sharing your history allows others to import and access the datasets, parameters, and steps of your history.</p>   <p>Access the history sharing menu via the History Options dropdown (<i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span>), and clicking “<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">history-share</span> Share or Publish”</p>   <ol>   <li><strong>Share via link</strong>     <ul>       <li>Open the <strong>History Options</strong> <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> menu at the top of your history panel and select “<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">history-share</span> Share or Publish”         <ul>           <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Make History accessible</strong></li>           <li>A <strong>Share Link</strong> will appear that you give to others</li>         </ul>       </li>       <li>Anybody who has this link can view and copy your history</li>     </ul>   </li>   <li><strong>Publish your history</strong>     <ul>       <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Make History publicly available in Published Histories</strong></li>       <li>Anybody on this Galaxy server will see your history listed under the <em>Published Histories</em> tab opened via the <i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">galaxy-histories-activity</span> <strong>Histories</strong> activity</li>     </ul>   </li>   <li><strong>Share only with another user</strong>.     <ul>       <li>Enter an email address for the user you want to share with in the <em>Please specify user email</em> input below <strong>Share History with Individual Users</strong></li>       <li>Your history will be shared only with this user.</li>     </ul>   </li>   <li><strong>Finding histories others have shared with me</strong>     <ul>       <li>Click on the <i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">galaxy-histories-activity</span> <strong>Histories</strong> activity in the activity bar on the left</li>       <li>Click the <strong>Shared with me</strong> tab</li>       <li>Here you will see all the histories others have shared with you directly</li>     </ul>   </li> </ol>   <p><strong>Note:</strong> If you want to make changes to your history without affecting the shared version, make a copy by going to <strong>History Options</strong> <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> icon in your history and clicking <em>Copy this History</em></p> </blockquote>
<p><!--END_SNIPPET--></p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-share-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Share history</div>

  <ol>
    <li>Share your history with your neighbour.</li>
    <li>Find the history shared by your neighbour. Histories shared with specific users can be accessed by those users under their top masthead “User” menu under <code class="language-plaintext highlighter-rouge">Histories shared with me</code>.</li>
  </ol>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-clone-clm-fates-release-for-nordic-site-from-github-advanced"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Clone CLM-FATES release for Nordic site from github (advanced)</div>

  <p>You may also get the CLM-FATES release 2.0.1 directly from github:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
cd $HOME
git clone -b release-emerald-platform2.0.1 https://github.com/NordicESMhub/ctsm.git
cd ctsm
./manage_externals/checkout_externals
</code></pre></div>  </div>
  <p>This approach may be interesting if you wish to run another release or development version of CLM-FATES.
All the tutorial shown can be done with your local version. In that case, you would need to use the
local command such as <code class="language-plaintext highlighter-rouge">create__newcase</code> which then require the following steps:</p>
  <ul>
    <li>locate the command on your local folder; for instance to locate <code class="language-plaintext highlighter-rouge">create_newcase</code>:</li>
  </ul>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>%%bash
cd $HOME/ctsm
find . -name create_newcase
</code></pre></div>  </div>
  <p>The command above will give you the location of <code class="language-plaintext highlighter-rouge">create_newcase</code>:</p>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>./cime/scripts/create_newcase
</code></pre></div>  </div>
  <p>Be aware that it is a relative path. Then to create a new case:</p>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>./cime/scripts/create_newcase --case $HOME/ctsm_cases/fates_alp1_local --compset 2000_DATM%1PTGSWP3_CLM50%FATES_SICE_SOCN_MOSART_SGLC_SWAV --res 1x1_ALP1 --machine espresso --run-unsupported

</code></pre></div>  </div>
  <p>Finally, if you wish to make changes to the source code, we recommend first to add your changes in different folder and use the option <code class="language-plaintext highlighter-rouge">--user-mods-dir</code> when
creating your case. In addition, you should make sure to use version control to save your changes.
If you are not familiar with <code class="language-plaintext highlighter-rouge">git</code>, you could also save your changes in the corresponding Galaxy history.</p>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>We have learnt to run single-point simulations with FATES-CLM through the Galaxy Climate JupyterLab.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(plausible !== undefined){
                            plausible('TutorialComplete', {callback: tutorial_finish_finish, props: {path: document.location.pathname}})
                          } else {
                            tutorial_finish_finish();
                          }
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Galaxy Climate JupyterLab</p>
</li>
                        
                        <li><p>CLM-FATES</p>
</li>
                        
                        <li><p>Model analysis</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/climate/faqs/">FAQ page for the Climate topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Fisher2015"><b>Taking off the training wheels: the properties of a dynamic vegetation model without climate envelopes, CLM4.5(ED)</b>, 2015. <a href="https://doi.org/10.5194/gmd-8-3593-2015">10.5194/gmd-8-3593-2015</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=climate/fates-jupyterlab"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=climate/fates-jupyterlab">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Anne Fouilloux,  <b>Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html">https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{climate-fates-jupyterlab,
author = "Anne Fouilloux",
	title = "Functionally Assembled Terrestrial Ecosystem Simulator (FATES) with Galaxy Climate JupyterLab (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/climate/tutorials/fates-jupyterlab/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/climate/tutorials/fates-jupyterlab/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00043">gxy.io/GTN:T00043</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/climate/tutorials/fates-jupyterlab/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/climate/tutorials/fates-jupyterlab/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/4bbac78b4791e7e0a5f3fcee9e506ed0d1e4d691">4bbac78</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.20cf1f17.js'></script>

	
	

    </body>
</html>