<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Large genome assembly and polishing</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <link rel="stylesheet" href="/training-material/assets/fontawesome/css/all.css">
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/largegenome/tutorial.html">
        <link rel="license" href="https://creativecommons.org/licenses/by/4.0/">

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Large genome assembly and polishing">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-03-06 12:23:54 +0000">
<meta name="DC.creator" content="Anna Syme">

        
        
        
        
        <meta name="description" content="DNA sequence data has become an indispensable tool for Mo...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Large genome assembly and polishing">
        <meta property="og:description" content="DNA sequence data has become an indispensable tool for Mo...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-03-06 12:23:54 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Large genome assembly and polishing",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Large genome assembly and polishing' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/largegenome/tutorial.html",
  "timeRequired": "PT2H",
  "keywords": "assembly, assembly, polishing, nanopore",
  "description": "The questions this  addresses are:\n - How can we assemble a large plant or animal genome?\n\n\nThe objectives are:\n - Assemble a large genome from long reads\n - Polish the assembly\n - Assess the quality of the assembly\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "name": "Anna Syme",
      "image": "https://avatars.githubusercontent.com/annasyme",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9906-0673",
      "orcid": "https://orcid.org/0000-0002-9906-0673"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "name": "Anna Syme",
      "image": "https://avatars.githubusercontent.com/annasyme",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9906-0673",
      "orcid": "https://orcid.org/0000-0002-9906-0673"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0196",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0196",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0196"
    }
  ]
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc">
        <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/assembly" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Assembly
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/assembly/faqs/" title="Check our FAQs for the Assembly topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/largegenome/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/largegenome/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                    <button class="btn" data-theme="default">Default</button>
                    <button class="btn" data-theme="night">Night</button>
                    <button class="btn" data-theme="midnight">Midnight</button>
                    <button class="btn" data-theme="rainbow">Rainbow</button>
                    <button class="btn" data-theme="blm">‚úäüèø</button>
                    <button class="btn" data-theme="progress" >üè≥Ô∏è‚Äçüåà</button>
                    <button class="btn" data-theme="halloween">üéÉ</button>
                    <button class="btn" data-theme="straya">üá¶üá∫</button>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-04-01/topics/assembly/tutorials/largegenome/tutorial.html" title="Version 2023-04-01">2023-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-03-01/topics/assembly/tutorials/largegenome/tutorial.html" title="Version 2023-03-01">2023-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-02-01/topics/assembly/tutorials/largegenome/tutorial.html" title="Version 2023-02-01">2023-02-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        











<!-- Gitter -->




<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>


<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<script type="application/ld+json">
    


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "dateModified": "2023-03-06 12:23:54 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Large genome assembly and polishing",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "identifier": "https://github.com/galaxyproject/training-material",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Large genome assembly and polishing' tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/largegenome/tutorial.html",
  "timeRequired": "PT2H",
  "keywords": "assembly, assembly, polishing, nanopore",
  "description": "The questions this  addresses are:\n - How can we assemble a large plant or animal genome?\n\n\nThe objectives are:\n - Assemble a large genome from long reads\n - Polish the assembly\n - Assess the quality of the assembly\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "name": "Anna Syme",
      "image": "https://avatars.githubusercontent.com/annasyme",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9906-0673",
      "orcid": "https://orcid.org/0000-0002-9906-0673"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.2-DRAFT-2019_07_19",
        "@type": "Person"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "name": "Anna Syme",
      "image": "https://avatars.githubusercontent.com/annasyme",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://galaxyproject.org/teach/gtn/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9906-0673",
      "orcid": "https://orcid.org/0000-0002-9906-0673"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0196",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0196",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0196"
    }
  ]
}
</script>

<article class="tutorial topic-assembly">
    <h1 data-toc-skip>Large genome assembly and polishing</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/annasyme/" class="contributor-badge contributor-annasyme"><img src="/training-material/assets/images/orcid.png" alt="orcid logo"/><img src="https://avatars.githubusercontent.com/annasyme?s=27" alt="Avatar">Anna Syme</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can we assemble a large plant or animal genome?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Assemble a large genome from long reads</p>
</li>
        
        <li><p>Polish the assembly</p>
</li>
        
        <li><p>Assess the quality of the assembly</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        
        

        
        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="http://doi.org/10.5281/zenodo.7055935">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/assembly/tutorials/largegenome/workflows/" title="Large genome assembly and polishing workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                <li class="btn btn-default supporting_material">


    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/assembly/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
        <a class="dropdown-item" href="https://india.usegalaxy.eu/" title="Galaxy India">
            Galaxy India
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
        <a class="dropdown-item" href="https://galaxytrakr.org/" title="GalaxyTrakr">
            GalaxyTrakr
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://streetscience.usegalaxy.eu/" title="Street Science">
            Street Science
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
            
        </ul>
        

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Mar 6, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
            
            <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Tutorial Content is licensed under Creative Commons Attribution 4.0 International License</a>
            
            <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">The GTN Framework is licensed under MIT</a>
        </div>
    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="introduction">Introduction</h1>

<h2 id="what-is-genome-assembly">What is genome assembly?</h2>

<p>A genome is a representation of the set of DNA in an organism, such as the set of chromosomes. When the DNA is extracted from the sample, it is broken up into fragments much smaller than the lengths of DNA in the chromosomes. After being sequenced, these fragments (and their copies) are called sequencing reads. To assemble the genome, we need to join the reads back into, ideally, chromosome-sized lengths.</p>

<p><img src="images/sampletogenome.png" alt="&quot;Image showing the concept of a genome assembly: a sample (a tree), with an arrow going to sequencing reads, with an arrow going to a genome (a set of diploid chromosomes)&quot;. " width="800" loading="lazy" /></p>

<h3 id="assembly-challenges">Assembly challenges</h3>

<p>In reality, we rarely get chromosome-length assemblies, due to many challenges. Here are some examples of particular challenges in a diploid genome assembly:</p>

<p><img src="images/challenges.png" alt="&quot;Image showing potential assembly challenges: a set of diploid chromosomes with descriptions of different problems which are: a contig may be a consensus from both chromosomes; telomeres and centromeres are difficult to assemble; contigs may be assembled separately as a copy in both chromosomes but it is unclear which parental chromosome they are from; some contigs are short or missing; some contigs are not placed in the assembly.&quot;. " width="800" loading="lazy" /></p>

<p>Even though most assemblies are not chromosome-length, the assembly in contigs is still valuable for many research questions. Lengths of assembled contigs are increasing as sequencing technology and assembly tools improve.</p>

<h3 id="can-i-use-my-own-data-with-these-workflows">Can I use my own data with these workflows?</h3>

<p>This tutorial has been tested on real-sized data sets and should work with your own data. However, there will most likely be some modifications required to tools and settings. Furthermore, as most species have never had their genome sequenced, it is not possible to guarantee existing workflows are optimal for new data.</p>

<p>It is most likely that any new genome assembly will have its own set of required workflow and analysis customisations to account for things such as ploidy and repeats. Usually, an assembly workflow will need testing and customising, in concert with reading the biological domain literature.</p>

<p>Before using these workflows on real datasets, we recommend completing all the relevant GTN tutorials about Data QC and Asssembly to give extra familiarity with the concepts. Then, we suggest creating subsampled data sets from your full-sized data, for testing. For best results, it is recommended to test each tool separately, and consider the parameters and their relevance and suitability for your own research questions.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>
  <p>In this tutorial we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#introduction" id="markdown-toc-introduction">Introduction</a>    <ol>
      <li><a href="#what-is-genome-assembly" id="markdown-toc-what-is-genome-assembly">What is genome assembly?</a></li>
    </ol>
  </li>
  <li><a href="#analysis-workflow" id="markdown-toc-analysis-workflow">Analysis workflow</a>    <ol>
      <li><a href="#how-to-run-a-workflow-in-galaxy" id="markdown-toc-how-to-run-a-workflow-in-galaxy">How to run a workflow in Galaxy</a></li>
    </ol>
  </li>
  <li><a href="#upload-data" id="markdown-toc-upload-data">Upload data</a>    <ol>
      <li><a href="#what-sequence-data-are-we-using-in-the-tutorial" id="markdown-toc-what-sequence-data-are-we-using-in-the-tutorial">What sequence data are we using in the tutorial?</a></li>
      <li><a href="#how-has-this-data-subset-been-prepared" id="markdown-toc-how-has-this-data-subset-been-prepared">How has this data subset been prepared?</a></li>
      <li><a href="#import-the-data" id="markdown-toc-import-the-data">Import the data</a></li>
    </ol>
  </li>
  <li><a href="#check-read-quality" id="markdown-toc-check-read-quality">Check read quality</a>    <ol>
      <li><a href="#data-qc-workflow" id="markdown-toc-data-qc-workflow">Data QC workflow</a></li>
      <li><a href="#data-qc-results" id="markdown-toc-data-qc-results">Data QC results</a></li>
    </ol>
  </li>
  <li><a href="#determine-genome-characteristics" id="markdown-toc-determine-genome-characteristics">Determine genome characteristics</a>    <ol>
      <li><a href="#kmer-counting-workflow" id="markdown-toc-kmer-counting-workflow">Kmer counting workflow</a></li>
      <li><a href="#kmer-counting-results" id="markdown-toc-kmer-counting-results">Kmer counting results</a></li>
    </ol>
  </li>
  <li><a href="#trim-and-filter-reads" id="markdown-toc-trim-and-filter-reads">Trim and filter reads</a>    <ol>
      <li><a href="#trimming-and-filtering-workflow" id="markdown-toc-trimming-and-filtering-workflow">Trimming and filtering workflow</a></li>
      <li><a href="#trim-and-filter-reads-results" id="markdown-toc-trim-and-filter-reads-results">Trim and filter reads: results</a></li>
      <li><a href="#summary-of-read-data-for-genome-assembly" id="markdown-toc-summary-of-read-data-for-genome-assembly">Summary of read data for genome assembly</a></li>
    </ol>
  </li>
  <li><a href="#genome-assembly" id="markdown-toc-genome-assembly">Genome Assembly</a></li>
  <li><a href="#assembly-polishing" id="markdown-toc-assembly-polishing">Assembly polishing</a>    <ol>
      <li><a href="#polishing-workflow" id="markdown-toc-polishing-workflow">Polishing workflow</a></li>
      <li><a href="#polishing-results" id="markdown-toc-polishing-results">Polishing results</a></li>
    </ol>
  </li>
  <li><a href="#genome-quality-assessment" id="markdown-toc-genome-quality-assessment">Genome quality assessment</a>    <ol>
      <li><a href="#genome-assessment-workflow" id="markdown-toc-genome-assessment-workflow">Genome assessment workflow</a></li>
    </ol>
  </li>
  <li><a href="#combining-workflows" id="markdown-toc-combining-workflows">Combining workflows</a>    <ol>
      <li><a href="#combined-assembly-and-polishing-workflow" id="markdown-toc-combined-assembly-and-polishing-workflow">Combined assembly and polishing workflow</a></li>
    </ol>
  </li>
  <li><a href="#summary" id="markdown-toc-summary">Summary</a></li>
</ol>

</blockquote>

<h1 id="analysis-workflow">Analysis workflow</h1>

<p>In this tutorial, we will follow these steps:</p>

<p><img src="images/steps.png" alt="&quot;An image showing the steps in the tutorial in a stylized form: quality control and kmer counting, represented by bold lines as sequencing reads; trimming and filtering reads, represented by truncated reads; assembly, represented by joined reads into one long piece; polishing, represented by corrections (ticks) in the assembly; the finished genome assembly, represented by a set of diploid chromosomes; quality assessment, represented by ticks and crosses at different chromosome locations.&quot;. " width="800" loading="lazy" /></p>

<ul>
  <li>Each of these steps is described in a section in this tutorial.</li>
  <li>For each step, we will run a workflow.</li>
  <li>We will stay in the same Galaxy history throughout.</li>
</ul>

<h2 id="how-to-run-a-workflow-in-galaxy">How to run a workflow in Galaxy</h2>

<ul>
  <li>All of these workflows have been uploaded to <a href="https://workflowhub.eu/">workflowhub.eu</a> and have been tagged with ‚ÄúLarge-genome-assembly‚Äù. There, each workflow is accompanied by an image of the workflow canvas and the tool connections.</li>
  <li>The workflows are also linked to this tutorial - see above and import them into your own Galaxy Account.</li>
  <li>If you are using Galaxy Australia, the workflows are published with the tag ‚Äúlg-wf‚Äù.</li>
  <li>A note: as of September 2022, these workflows have been tested on Galaxy Australia and we are now in the process of testing them on Galaxy Europe and Galaxy Main.</li>
</ul>

<p>Here is a list of the workflows:</p>

<table>
  <thead>
    <tr>
      <th>Tutorial section</th>
      <th>Workflow name</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Quality Control</td>
      <td>Data QC</td>
    </tr>
    <tr>
      <td>Kmer counting</td>
      <td>kmer counting - meryl</td>
    </tr>
    <tr>
      <td>Trim and filter reads</td>
      <td>Trim and filter reads - fastp</td>
    </tr>
    <tr>
      <td>Assembly</td>
      <td>Assembly with Flye</td>
    </tr>
    <tr>
      <td>Polish assembly</td>
      <td>Assembly polishing</td>
    </tr>
    <tr>
      <td>Assess quality</td>
      <td>Assess genome quality</td>
    </tr>
  </tbody>
</table>

<!--SNIPPET-->
<blockquote class="tip">   <div class="box-title tip-title" id="tip-import-workflows-from-workflowhub"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-import-workflows-from-workflowhub" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Import workflows from WorkflowHub<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://workflowhub.eu/">WorkflowHub</a> is a workflow management system which allows workflows to be FAIR (Findable, Accessible, Interoperable, and Reusable), citable, have managed metadata profiles, and be openly available for review and analytics.</p>   <blockquote class="notranslate hands_on">   <div class="box-title hands-on-title" id="hands-on-import-a-workflow"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Import a workflow</div>   <ol>     <li>Click on the <strong>Workflow</strong> menu, located in the top bar.</li>     <li>Click on the <strong>Import</strong> button, located in the right corner.</li>     <li>       <p>In the section ‚ÄúImport a Workflow from Configured GA4GH Tool Registry Servers (e.g. Dockstore)‚Äù, click on <strong>Search form</strong>.</p>     </li>     <li>In the <strong>TRS Server: <em>workflowhub.eu</em></strong> menu you should type <code class="language-plaintext highlighter-rouge">Large-genome-assembly</code> <img src="/training-material/topics/assembly/images/vgp_assembly/workflow_list.png" alt="galaxy TRS workflow search field, name:vgp is entered in the search bar, and five different workflows all labelled VGP are listed" /></li>     <li>Click on the desired workflow, and finally select the latest available version.</li>   </ol> </blockquote>   <p>After that, the imported workflows will appear in the main workflow menu. In order to initialize the workflow, we just need to click in the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> <strong>Run workflow</strong> icon.</p> </blockquote>

<p>After importing the workflows into your account:</p>

<ul>
  <li>Look at your of Workflows. (Galaxy top panel: Workflow)</li>
  <li>For the workflow you want to run, go to the right hand side and see the arrow button (a triangle), click</li>
  <li>This brings up the workflow in the centre Galaxy panel</li>
  <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
  <li>For ‚ÄúSend results to a new history‚Äù, leave it as ‚ÄúNo‚Äù.</li>
  <li>Each time you run a workflow, you need to specify the input data set (or sets). Galaxy will try to guess which file this is, but change if required using the drop-down arrow.</li>
  <li>At the top right, click ‚ÄúRun Workflow‚Äù.</li>
  <li>The result files will appear at the top of your current history</li>
</ul>

<p>Each workflow will be discussed in a separate section.</p>

<h1 id="upload-data">Upload data</h1>

<p>Let‚Äôs start with uploading the data.</p>

<h2 id="what-sequence-data-are-we-using-in-the-tutorial">What sequence data are we using in the tutorial?</h2>

<ul>
  <li>The data sets for genome projects can be very large and tools can take some time to run. It is a good idea to test that your planned tools and workflows will work on smaller-sized test data sets, as it is much quicker to find out about any problems.</li>
  <li>In this tutorial we will use a subset of real sequencing data from a plant genome, the snow gum, <em>Eucalyptus pauciflora</em>, from a genome project described in this paper: <span class="citation"><a href="#Wang2020">Wang <i>et al.</i> 2020</a></span>. Data is hosted at NCBI BioProject number: PRJNA450887.</li>
</ul>

<h2 id="how-has-this-data-subset-been-prepared">How has this data subset been prepared?</h2>

<ul>
  <li>From NCBI, three read files were imported into Galaxy for this tutorial: nanopore reads (SRR7153076), and paired Illumina reads (SRR7153045).</li>
  <li>These were randomly subsampled to 10% of the original file size.</li>
  <li>Plant genomes may contain an excess of reads from the chloroplast genome (of which there are many copies per cell). To ensure our test data sets are not swamped from excessive chloroplast-genome reads, reads that mapped to a set of known chloroplast gene sequences were discarded.</li>
</ul>

<p>We are also using a reference genome <em>Arabidopsis thaliana</em> for a later comparison step - file TAIR10_chr_all.fas downloaded from <a href="https://www.arabidopsis.org/download/index-auto.jsp?dir=%2Fdownload_files%2FGenes%2FTAIR10_genome_release%2FTAIR10_chromosome_files">The Arabidopsis Information Resource</a>.</p>

<h2 id="import-the-data">Import the data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Import the data</div>
  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a proper name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Creating a new history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Renaming a history<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is ‚ÄúUnnamed history‚Äù)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
    </li>
    <li>
      <p>Import from <a href="https://zenodo.org/record/7055935">Zenodo</a> or a data library (ask your instructor):</p>

      <ul>
        <li>2 FASTQ files with illumina reads: these files have a R1 or R2 in the name</li>
        <li>1 FASTQ file with nanopore reads: this file has nano in the name</li>
        <li>1 FASTQ file for the genome Arabidopsis</li>
      </ul>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7055935/files/Eucalyptus_subsample_ten_pc_SRR7153045_R1.fastq.gz
https://zenodo.org/record/7055935/files/Eucalyptus_subsample_ten_pc_SRR7153045_R2.fastq.gz
https://zenodo.org/record/7055935/files/Eucalyptus_subsample_ten_pc_nano_SRR7153076.fastq.gz
https://zenodo.org/record/7055935/files/Ref_genome_Arabidopsis.fasta
</code></pre></div>      </div>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing via links<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Open the Galaxy Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> on the top-right of the tool panel)</p>   </li>   <li>Select <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing data from a data library<span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ul>   <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor</li>   <li>Select the desired files</li>   <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>   <li>Click on <strong>Import</strong></li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<ul>
  <li>This tutorial uses these input files and gives some examples from the results.</li>
  <li>It is likely that your results will differ slightly (e.g. number of bases in the genome assembly). This is common, because many tools start from different random seeds.</li>
  <li>Also, tool versions are being constantly updated.  Newer versions may be available since this tutorial was written and could give slightly different results.</li>
</ul>

<h1 id="check-read-quality">Check read quality</h1>

<p>Let‚Äôs look at how many reads we have and their quality scores using the Data QC workflow.</p>

<h2 id="data-qc-workflow">Data QC workflow</h2>

<p>What it does: Reports statistics from sequencing reads</p>

<p>Inputs:</p>
<ul>
  <li>long reads (fastq.gz format)</li>
  <li>short reads (R1 and R2) (fastq.gz format)</li>
</ul>

<p>Outputs:</p>
<ul>
  <li>For long reads: a nanoplot report (the HTML report summarizes all the information).</li>
  <li>For short reads: a MultiQC report</li>
</ul>

<p>Tools used:</p>
<ul>
  <li>Nanoplot</li>
  <li>FastQC</li>
  <li>MultiQC</li>
</ul>

<p>Workflow steps:</p>
<ul>
  <li>Long reads are analysed by Nanoplot.</li>
  <li>Short reads (R1 and R2) are analysed by FastQC</li>
  <li>the resulting reports are processed by MultiQC</li>
</ul>

<p>Options:</p>
<ul>
  <li>See the tool settings options at runtime and change as required.</li>
  <li>Alternative tool option: fastp</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-data-qc-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Data QC workflow</div>

  <ol>
    <li><strong>Import the Data QC workflow</strong> into Galaxy:
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/assembly/tutorials/largegenome/workflows/Galaxy-Workflow-Data_QC.ga">this workflow</a> or download it to your computer.</li>
        <li>Import the workflow into Galaxy</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>‚ÄúArchived Workflow URL‚Äù</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>‚ÄúArchived Workflow File‚Äù</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>

      <ul>
        <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
      </ul>
    </li>
    <li>Run <strong>Data QC</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
      <ul>
        <li><em>‚ÄúSend results to a new history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1: Input file: long reads‚Äù</em>: the nanopore fastq.gz file</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú2: Input file: Illumina reads R1‚Äù</em>: the illumina R1 fastq.gz file</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú3: Input file: Illumina reads R2‚Äù</em>: the illumina R2 fastq.gz file</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="data-qc-results">Data QC results</h2>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>

  <ol>
    <li>What are the results from the two output files? Are the reads long enough and of high enough quality for our downstream analyses?  Will reads need any trimming or filtering?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Common things to check are average read length, average quality, and whether quality varies by position in the reads.
        <ul>
          <li>Look at the plot for MultiQC for the Sequence Quality Histograms. This shows how the read quality of Illumina reads (y axis) varies according to base position (x axis).You may see for Illumina reads that there is some drop-off in quality towards the end of the reads, which may benefit from trimming.</li>
          <li>Look at the plot from Nanoplot for ‚ÄúRead lengths vs Average read quality‚Äù. The nanopore reads have a mean read quality of 9.0. Depending on the size of our input read sets and the research question, we may filter out reads below a certain average quality. If we had a lot of reads, we may be able to set a higher threshold for filtering according to read quality.</li>
        </ul>
      </li>
    </ol>
  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-additional-information-about-the-tools"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Additional information about the tools</div>
  <p>More about interpreting nanoplot plots:</p>
  <ul>
    <li><a href="https://github.com/wdecoster/NanoPlot">Nanoplot on Github</a></li>
    <li><a href="https://gigabaseorgigabyte.wordpress.com/2017/06/01/example-gallery-of-nanoplot/">Example nanoplots</a></li>
  </ul>

  <p>More about FastQC results:</p>
  <ul>
    <li><a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC webpage</a></li>
    <li><a href="https://timkahlke.github.io/LongRead_tutorials/QC_F.html">FastQC tips</a></li>
  </ul>
</blockquote>

<h1 id="determine-genome-characteristics">Determine genome characteristics</h1>

<p>To prepare for genome assembly you might want to know things about your genome such as size, ploidy level (how many sets of chromosomes) and heterozygosity (how variable the sequence is between homologous chromosomes). A relatively fast way to estimate these things is to count small fragments of the sequencing reads (called kmers).</p>

<p>A read broken into kmers:</p>

<p><img src="images/kmers.png" alt="&quot;An image of the a read broken into kmers: the read is a bold bar with the letters ABCDEF; each kmer is a smaller bar with three letters in it: the kmers are ABC, BCD, CDE, and DEF.&quot;. " width="650" loading="lazy" /></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-is-kmer-counting"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: What is kmer counting?</div>

  <p>Kmer counting is usually done with high-accuracy short reads, not long reads which may have high error rates.  After counting how many times each kmer is seen in the reads, we can see what sorts of counts are common. For example, lots of kmers may have been found 24 or 25 times. A graph shows the number of different kmers (y axis) found at different counts, or depths (x axis).</p>

  <p>Many different kmers will be found the same number of times; e.g. X25. If kmer length approaches read length, this means the average depth of your sequencing is also ~X25, and there would be a peak in the graph at this position (smaller kmers = higher kmer depth). There may be smaller peaks of kmer counts at higher depths, e.g. X50 or X100, indicating repeats in the genome. There may be other smaller peaks of kmers found at half the average depth, indicating a diploid genome with a certain amount of difference between the homologous chromosomes - this is known as heterozygosity. Thus, the plot of how many different kmers are found at all the depths will help inform estimates of sequencing depth, ploidy level, heterozygosity, and genome size.</p>
</blockquote>

<h2 id="kmer-counting-workflow">Kmer counting workflow</h2>

<p>What it does: Estimates genome size and heterozygosity based on counts of kmers</p>

<p>Inputs: One set of short reads: e.g. R1.fq.gz</p>

<p>Outputs: GenomeScope graphs</p>

<p>Tools used:</p>
<ul>
  <li>Meryl</li>
  <li>GenomeScope</li>
</ul>

<p>Workflow steps:</p>
<ul>
  <li>The tool meryl counts kmers in the input reads (k=21), then converts this into a histogram.</li>
  <li>GenomeScope: runs a model on the histogram; reports estimates. k-mer size set to 21.</li>
</ul>

<p>Options:</p>
<ul>
  <li>Use a different kmer counting tool. e.g. khmer. If so, for the settings, advanced parameters:</li>
  <li>k-mer size: 21 (as per <a href="https://github.com/schatzlab/genomescope/issues/32">this recommendation</a>).</li>
  <li>n_tables: 4. tablesize: set at 8 billion (as per <a href="https://khmer.readthedocs.io/en/v1.0/choosing-table-sizes.html">this recommendation</a>).</li>
  <li>Will also need to run some formatting steps to convert khmer output to a two-column matrix, for the  Genomscope. See <a href="https://usegalaxy.org.au/u/anna/w/kmer-counting-khmer">this workflow</a>.</li>
  <li>Note: khmer: to use both R1 and R2 read sets, khmer needs these paired reads in interleaved format.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-kmer-counting-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Kmer counting workflow</div>

  <ol>
    <li><strong>Import the Kmer counting workflow</strong> into Galaxy:
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/assembly/tutorials/largegenome/workflows/Galaxy-Workflow-kmer_counting.ga">this workflow</a> or download it to your computer.</li>
        <li>Import the workflow into Galaxy</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>‚ÄúArchived Workflow URL‚Äù</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>‚ÄúArchived Workflow File‚Äù</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>

      <ul>
        <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
      </ul>
    </li>
    <li>Run <strong>Data QC</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
      <ul>
        <li><em>‚ÄúSend results to a new history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1: Input file: Illumina reads R1‚Äù</em>: the illumina R1 fastq.gz file</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="kmer-counting-results">Kmer counting results</h2>

<p>GenomeScope transformed linear plot:</p>

<p><img src="images/genomescope.png" alt="&quot;An image of the Genomescope plot, described fully in the text following this image.&quot;. " width="800" loading="lazy" /></p>

<p>Here we can see a central peak - showing that most of the different kmers were found at counts of ~ 120. These are kmers from single-copy homozygous alleles. To the left, a smaller peak at around half the coverage, showing kmers from heterozygous alleles (note that this peak gets higher than the main peak when heterozygosity is only ~ 1.2%). To the right, another smaller peak showing kmers at higher coverage, from repeat regions. Information from these three peaks provide a haploid genome length estimate of ~240,000 bp (note this is test data so smaller than whole plant genome size).</p>

<p>The output Summary file shows more detail:</p>
<ul>
  <li>Genome unique length: from single copy homozygous and heterozygous alleles (under the main and left peak).</li>
  <li>Genome repeat length: from repeat copies (under the graph to the right of the main peak).</li>
  <li>Genome haploid length: unique length + repeat length</li>
</ul>

<p>More about kmer counting: See <a href="https://bioinformatics.uconn.edu/genome-size-estimation-tutorial/#">this tutorial</a></p>

<p>For more about Meryl see <span class="citation"><a href="#Rhie2020">Rhie <i>et al.</i> 2020</a></span>.</p>

<p>For more about Genomescope see <span class="citation"><a href="#Vurture2017">Vurture <i>et al.</i> 2017</a></span>(Note: the supplementary information is very informative).</p>

<h1 id="trim-and-filter-reads">Trim and filter reads</h1>

<p>Using information from Data QC and kmer counting, we may want to trim and/or filter reads. The settings for trimming and filtering depend on many things, including:</p>
<ul>
  <li>your aim (accuracy; contiguity)</li>
  <li>your data: type, error rate, read depth, lengths, quality (average, variation by position)</li>
  <li>the ploidy and heterozygosity of your sample</li>
  <li>choice of assembly tool (e.g. it may automatically deal with adapters, low qualities, etc.)</li>
</ul>

<p>Because of all these factors, few specific recommendations are made here, but the workflow is provided and can be customised. If you are unsure how to start, use your test data to try different settings and see the effect on the resulting size and quality of the reads, and the downstream assembly contigs. Newer assemblers are often configured to work well with long-read data and in some cases, read trimming/filtering for long reads may be unnecessary.</p>

<p>Trimming and filtering reads:</p>

<p><img src="images/trimfilterreads.png" alt="&quot;An image of the concepts of trimming and filtering reads, with reads represented by bold bars; trimming represented by crosses over various locations in each read.&quot;. " width="800" loading="lazy" /></p>

<h2 id="trimming-and-filtering-workflow">Trimming and filtering workflow</h2>

<p>What it does: Trims and filters raw sequence reads according to specified settings.</p>

<p>Inputs:</p>
<ul>
  <li>Long reads (format fastq)</li>
  <li>Short reads R1 and R2 (format fastq)</li>
</ul>

<p>Outputs: Trimmed and filtered reads:</p>
<ul>
  <li>fastp_filtered_long_reads.fastq.gz (But note: no trimming or filtering is on by default)</li>
  <li>fastp_filtered_R1.fastq.gz</li>
  <li>fastp_filtered_R2.fastq.gz</li>
</ul>

<p>Tools used:  fastp (Note. The latest version (0.20.1) of fastp has an issue displaying plot results. Using version 0.19.5 here instead until this is rectified).</p>

<p>Input parameters: None required, but recommend removing the long reads from the workflow if not using any trimming/filtering settings.</p>

<p>Workflow parameter settings:</p>
<ul>
  <li>Long reads: fastp settings:
    <ul>
      <li>These settings have been changed from the defaults (so that all filtering and trimming settings are now disabled).</li>
      <li>Adapter trimming options: Disable adapter trimming: yes.</li>
      <li>Filter options: Quality filtering options: Disable quality filtering: yes.</li>
      <li>Filter options: Length filtering options: Disable length filtering: yes.</li>
      <li>Read modification options: PolyG tail trimming: Disable.</li>
      <li>Output options: output JSON report: yes</li>
    </ul>
  </li>
  <li>Short reads: fastp settings:
    <ul>
      <li>adapter trimming (default setting: adapters are auto-detected)</li>
      <li>quality filtering (default: phred quality 15), unqualified bases limit (default = 40%), number of Ns allowed in a read (default = 5)</li>
      <li>length filtering (default length = min 15)</li>
      <li>polyG tail trimming (default = on for NextSeq/NovaSeq data which is auto detected)</li>
    </ul>
  </li>
  <li>Output options: output JSON report: yes</li>
</ul>

<p>Options:</p>
<ul>
  <li>Change any settings in fastp for any of the input reads.</li>
  <li>Adapter trimming: input the actual adapter sequences. (Alternative tool for long read adapter trimming: Porechop.)</li>
  <li>Trimming n bases from ends of reads if quality less than value x  (Alternative tool for trimming long reads: NanoFilt.)</li>
  <li>Discard post-trimmed reads if length is &lt; x (e.g. for long reads, 1000 bp)</li>
  <li>Example filtering/trimming that you might do on long reads: remove adapters (can also be done with Porechop), trim bases from ends of the reads with low quality (can also be done with NanoFilt), after this can keep only reads of length x (e.g. 1000 bp)</li>
  <li>If not running any trimming/filtering on nanopore reads, could delete this step from the workflow entirely.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-trim-and-filter-reads-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Trim and Filter Reads workflow</div>

  <ol>
    <li><strong>Import the Trim and Filter reads workflow</strong> into Galaxy:
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/assembly/tutorials/largegenome/workflows/Galaxy-Workflow-Trim_and_filter_reads.ga">this workflow</a> or download it to your computer.</li>
        <li>Import the workflow into Galaxy</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>‚ÄúArchived Workflow URL‚Äù</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>‚ÄúArchived Workflow File‚Äù</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>

      <ul>
        <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
      </ul>
    </li>
    <li>Run <strong>Trim and Filter Reads</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
      <ul>
        <li><em>‚ÄúSend results to a new history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1: Input file: long reads‚Äù</em>: the nanopore fastq.gz file</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú2: Input file: Illumina reads R1‚Äù</em>: the illumina R1 fastq.gz file</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú3: Input file: Illumina reads R2‚Äù</em>: the illumina R2 fastq.gz file</li>
        <li><em>Note: the workflow parameter settings described above are already set, and don‚Äôt need to be changed when running the workflow, unless you wish to.</em></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="trim-and-filter-reads-results">Trim and filter reads: results</h2>

<p>There are two fastp reports - one for the illumina reads and one for the nanopore reads. We have only processed illumina reads in this example. Look at the fastp illumina report. (Note: the title in the report refers to only one of the input read sets but the report is for both read sets. This is a known issue under investigation.)</p>

<p>Filtering results from fastp on short reads:</p>

<p><img src="images/fastp.png" alt="&quot;Image of a table of results from fastp, showing the percentage of reads that were low quality, had too many Ns, or too short.&quot;. " width="800" loading="lazy" /></p>

<p>Here we can see that less than 0.5 % of the reads were discarded based on quality. If our read set had high enough coverage for downstream analyses, we might choose to apply a stricter quality filter.</p>

<h2 id="summary-of-read-data-for-genome-assembly">Summary of read data for genome assembly</h2>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>
  <ol>
    <li>How many reads do we have now for our genome assembly? Is the read coverage high enough?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>
    <ul>
      <li>Genome size: From kmer counting, the estimated genome size is ~ 240,000 bp  (this is only subsampled data; full data would likely suggest a size of 0.5 - 1 Gbp for a typical plant genome)</li>
      <li>Genome coverage (or depth):  total base pairs in the reads / base pairs in genome</li>
      <li>Short reads: From the fastp report after trimming and filtering short reads, there are 3.2 million reads, comprising 482 million base pairs</li>
      <li>Short read coverage:  = X2008</li>
      <li>Long reads: From the fastp report of the long reads (although no filtering and trimming performed) there are 85 thousand reads, comprising 761 million base pairs.  From the nanoplot tool we ran in the Data QC section , we know that the mean read length is almost 9,000 base pairs, and the longest read is &gt; 140,000 base pairs.</li>
      <li>Long read coverage:  = X3170
These coverages are very high but are ok to use with tutorial data. With a typical full data set, coverage would be more in the order of X40 to X200.</li>
    </ul>

  </blockquote>
</blockquote>

<h1 id="genome-assembly">Genome Assembly</h1>

<p>Genome assembly means joining the reads up to make contiguous sections of the genome. A simplified way to imagine this is overlapping all the different sequencing reads to make a single length or contig, ideally one for each original chromosome. The output is a set of contigs and a graph showing how contigs are connected.</p>

<p>Extreme simplification of genome assembly:</p>

<p><img src="images/assembly.png" alt="&quot;Image of the simplified process of genome assembly, from a tree, to reads (represented by bold bars), to an assembly (represented by one long bold bar), to contigs (various long bold bars) and to an assembly graph.&quot;. " width="800" loading="lazy" /></p>

<p>Genome assembly algorithms use different approaches to work with the complexities of large sequencing read data sets, large genomes, different sequencing error rates, and computational resources. Many use graph-based algorithms.  For more about genome assembly algorithms see <a href="https://langmead-lab.org/teaching-materials/">these tutorials by Ben Langmead</a>.</p>

<h3 id="which-assembly-tool-and-approach-to-use">Which assembly tool and approach to use?</h3>

<p>Here, we will use the assembly tool called Flye to assemble the long reads. This is fast and deals well with the high error rate. Then, we will polish (correct) the assembly using information from the long reads (in their unassembled state), as well as the more accurate short Illumina reads.</p>

<p>There are many other approaches and combinations of using short and long reads, and the polishing steps. For example, the long reads can be polished before assembly (with themselves, or with short reads). This may increase accuracy of the assembly, but it may also introduce errors if similar sequences are ‚Äúcorrected‚Äù into an artificial consensus. Long reads are usually used in the assembly, but it is possible to assemble short reads and then scaffold these into longer contigs using information from long reads.</p>

<p>For more about the differences between current assembly and polishing tools see <span class="citation"><a href="#Chen2021">Chen <i>et al.</i> 2021</a></span> and <span class="citation"><a href="#McCartney2021">McCartney <i>et al.</i> 2021</a></span>.</p>

<h4 id="assembly-with-flye-workflow">Assembly with Flye workflow</h4>

<p>What it does: Assembles long reads with the tool Flye</p>

<p>Inputs: long reads (may be raw, or filtered, and/or corrected); fastq.gz format</p>

<p>Outputs:</p>
<ul>
  <li>Flye assembly fasta.</li>
  <li>Fasta stats on assembly.fasta</li>
  <li>Assembly graph image from Bandage</li>
  <li>Bar chart of contig sizes</li>
  <li>Quast reports of genome assembly</li>
</ul>

<p>Tools used:</p>
<ul>
  <li>Flye</li>
  <li>Fasta statistics</li>
  <li>Bandage</li>
  <li>Bar chart</li>
  <li>Quast</li>
</ul>

<p>Input parameters: None required, but recommend setting assembly mode to match input sequence type</p>

<p>Workflow steps:</p>
<ul>
  <li>Long reads are assembled with Flye, using default tool settings. Note: the default setting for read type (‚Äúmode‚Äù) is nanopore raw. Change this at runtime if required.</li>
  <li>Statistics are computed from the assembly.fasta file output, using Fasta Statistics and Quast (is genome large: Yes; distinguish contigs with more that 50% unaligned bases: no)</li>
  <li>The graphical fragment assembly file is visualized with the tool Bandage.</li>
  <li>Assembly information sent to bar chart to visualize contig sizes</li>
</ul>

<p>Options</p>
<ul>
  <li>See other Flye options.</li>
  <li>Use a different assembler (in a different workflow).</li>
  <li>Bandage image options - change size (max size is 32767), labels - add (e.g. node lengths). You can also install Bandage on your own computer and download the ‚Äúgraphical fragment assembly‚Äù file to view in greater detail.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-assembly-with-flye-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Assembly with Flye workflow</div>

  <ol>
    <li><strong>Import the Assembly with Flye workflow</strong> into Galaxy:
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/assembly/tutorials/largegenome/workflows/Galaxy-Workflow-Assembly_with_Flye.ga">this workflow</a> or download it to your computer.</li>
        <li>Import the workflow into Galaxy</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>‚ÄúArchived Workflow URL‚Äù</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>‚ÄúArchived Workflow File‚Äù</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>

      <ul>
        <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
      </ul>
    </li>
    <li>Run <strong>Assembly with Flye</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
      <ul>
        <li><em>‚ÄúSend results to a new history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1: Input file: long reads‚Äù</em>: the fastp filtered long reads fastq.gz file</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<h4 id="assembly-results">Assembly results</h4>

<p>The assembled contigs are in the ‚ÄúFlye assembly on data X (consensus)‚Äù (X is a number that will vary depending on where it sits in your history).<br />
Open the Quast tabular report to see the assembly statistics:</p>

<p><img src="images/quast.png" alt="&quot;Image of the Quast tabular report, showing various assembly statistics.&quot;. " width="800" loading="lazy" /></p>

<p>There are 153 contigs, largest is ~246,000 bp, and total length almost 10 million bp. This is a fair bit longer than the estimated genome size from kmer counting (which was ~240,000 bp), but the difference is likely mainly due to idiosyncrasies of using a subsampled data set. The read coverage was likely &lt;1, causing many kmers to have frequency of &lt;1 and be classed as errors, rather than contributing to the genome size estimate.</p>

<p>Open the Quast HTML report, then click on ‚ÄúView in Icarus contig browser‚Äù.  This is a way to visualize the contigs and their sizes:</p>

<p><img src="images/icarus.png" alt="&quot;Image of the Icarus contig browser, showing the set of contigs arranged from larges to smallest, represented by rectangles of different widths.&quot;. " width="800" loading="lazy" /></p>

<p>View the Bandage image of the assembly graph:</p>

<p><img src="images/bandagegraphpartial.png" alt="&quot;Image of the bandage graph showing one multi-part set of joined contigs, some smaller one- or two- piece contigs, and many single contigs.&quot;. " width="800" loading="lazy" /></p>

<p>As this is a subsampled data set, it is not surprising that most of the contigs are unjoined. The joined contigs at the top left are likely to be part of the mitochondrial genome as these reads were probably over-represented in our subsampled data set.</p>

<h4 id="what-about-centromeres-and-telomeres">What about centromeres and telomeres?</h4>

<p>Some genomic areas such as centromeres, telomeres, and ribosomal DNA arrays, are much harder to assemble. These are long stretches of very similar repeats. With improved sequencing accuracy, length, and technologies (particularly long-range scaffolding), these may soon be much easier to assemble. The latest human genome assembly has a good demonstration of the techniques used for this.   See <span class="citation"><a href="#Nurk2021">Nurk <i>et al.</i> 2021</a></span>, and in particular, Figure 2: Bandage graphs of the human genome chromosomes, with the grey shading showing centromeric regions.</p>

<h4 id="what-about-haplotigs">What about haplotigs?</h4>

<p>Although our sample may be diploid, with pairs of chromosomes, the resulting assembly is often a haploid (or ‚Äúcollapsed‚Äù) assembly. This is not the sequence of one of the chromosomes, but a mix of the two.</p>

<p>Some assemblers will produce extra contigs called haplotigs. These are parts of the assembly from heterozygous regions (that is, the sequence is relatively different between the chromosome pair). There are tools to remove haplotigs from the assembly if that is preferred.</p>

<p>For more on differences between collapsed, primary/alternate and partially-phased assemblies, with a great visual representation: see <a href="http://lh3.github.io/2021/04/17/concepts-in-phased-assemblies">Heng Li‚Äôs Github page</a></p>

<p>For more on the phased assemblies, particularly for diploids or polyploids, see <span class="citation"><a href="#Garg2021">Garg 2021</a></span>.</p>

<h1 id="assembly-polishing">Assembly polishing</h1>

<p>We will polish the assembly using both the long reads and short reads. This process aligns the reads to the assembly contigs, and makes corrections to the contigs where warranted.  For more, see <span class="citation"><a href="#Aury2021">Aury and Istace 2021</a></span>, particularly for a discussion about polishing diploid genomes.</p>

<p>Assembly polishing:</p>

<p><img src="images/polish.png" alt="&quot;Image of the concept of assembly polishing, showing reads mapped to the assembly contigs.&quot;. " width="800" loading="lazy" /></p>

<h2 id="polishing-workflow">Polishing workflow</h2>

<p>Workflow structure: The workflow includes two subworkflows:</p>
<ul>
  <li>Racon polish with long reads, x 4</li>
  <li>Racon polish with illumina reads, x2</li>
</ul>

<p>What it does: Polishes (corrects) an assembly, using long reads (with the tools Racon and Medaka) and short reads (with the tool Racon). (Note: medaka is only for nanopore reads, not PacBio reads).</p>

<p>Inputs:</p>
<ul>
  <li>assembly to be polished:  assembly.fasta</li>
  <li>long reads - the same set used in the assembly (e.g. may be raw or filtered) fastq.gz format</li>
  <li>short reads, R1 only, in fastq.gz format</li>
</ul>

<p>Outputs:</p>
<ul>
  <li>Racon+Medaka+Racon polished_assembly. fasta</li>
  <li>Fasta statistics after each polishing tool</li>
</ul>

<p>Tools used:</p>
<ul>
  <li>Minimap2</li>
  <li>Racon</li>
  <li>Fasta statistics</li>
  <li>Medaka</li>
</ul>

<p>Input parameters: None required, but recommended to set the Medaka model correctly (default = r941_min_high_g360). See drop down list for options.</p>

<p>Workflow steps for Part 1, Polish with long reads: using Racon</p>
<ul>
  <li>Long reads and assembly contigs =&gt; Racon polishing (subworkflow):</li>
  <li>minimap2 : long reads are mapped to assembly =&gt; overlaps.paf.</li>
  <li>overaps, long reads, assembly =&gt; Racon =&gt; polished assembly 1</li>
  <li>using polished assembly 1 as input; repeat minimap2 + racon =&gt; polished assembly 2</li>
  <li>using polished assembly 2 as input, repeat minimap2 + racon =&gt; polished assembly 3</li>
  <li>using polished assembly 3 as input, repeat minimap2 + racon =&gt; polished assembly 4</li>
  <li>Racon long-read polished assembly =&gt; Fasta statistics</li>
  <li>Note: The Racon tool panel can be a bit confusing and is under review for improvement. Presently it requires sequences (= long reads), overlaps (= the paf file created by minimap2), and target sequences (= the contigs to be polished) as per ‚Äúusage‚Äù described at this link <a href="https://github.com/isovic/racon/blob/master/README.md">https://github.com/isovic/racon/blob/master/README.md</a></li>
  <li>Note: Racon: the default setting for ‚Äúoutput unpolished target sequences?‚Äù is No. This has been changed to Yes for all Racon steps in these polishing workflows.  This means that even if no polishes are made in some contigs, they will be part of the output fasta file.</li>
  <li>Note: the contigs output by Racon have new tags in their headers. For more on this see <a href="https://github.com/isovic/racon/issues/85">this issue</a>.</li>
</ul>

<p>Workflow steps for Part 2, Polish with long reads: using Medaka</p>
<ul>
  <li>Racon polished assembly + long reads =&gt; medaka polishing X1 =&gt; medaka polished assembly</li>
  <li>Medaka polished assembly =&gt; Fasta statistics</li>
</ul>

<p>Workflow steps for Part 3, Polish with short reads: using Racon</p>
<ul>
  <li>Short reads and Medaka polished assembly =&gt;Racon polish (subworkflow):</li>
  <li>minimap2: short reads (R1 only) are mapped to the assembly =&gt; overlaps.paf. Minimap2 setting is for short reads.</li>
  <li>overlaps + short reads + assembly =&gt; Racon =&gt; polished assembly 1</li>
  <li>using polished assembly 1 as input; repeat minimap2 + racon =&gt; polished assembly 2</li>
  <li>Racon short-read polished assembly =&gt; Fasta statistics</li>
</ul>

<p>Options:</p>
<ul>
  <li>Change settings for Racon long read polishing if using PacBio reads:  The default profile setting for Racon long read polishing: minimap2 read mapping is ‚ÄúOxford Nanopore read to reference mapping‚Äù, which is specified as an input parameter to the whole Assembly polishing workflow, as text: map-ont. If you are not using nanopore reads and/or need a different setting, change this input. To see the other available settings, open the minimap2 tool, find ‚ÄúSelect a profile of preset options‚Äù, and click on the drop down menu. For each described option, there is a short text in brackets at the end (e.g. map-pb). This is the text to enter into the assembly polishing workflow at runtime instead of the default (map-ont).</li>
  <li>Other options: change the number of polishes (in Racon and/or Medaka). There are ways to assess how much improvement in assembly quality has occurred per polishing round (for example, the number of corrections made; the change in Busco score - see section ‚ÄúGenome quality assessment‚Äù for more on Busco).</li>
  <li>Option: change polishing settings for any of these tools. Note: for Racon - these will have to be changed within those subworkflows first. Then, in the main workflow, update the</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-assembly-polishing-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Assembly polishing workflow</div>

  <ol>
    <li><strong>Import the Assembly polishing workflow</strong> into Galaxy:
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/assembly/tutorials/largegenome/workflows/Galaxy-Workflow-Assembly_polishing.ga">this workflow</a> or download it to your computer.</li>
        <li>Import the workflow into Galaxy</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-4"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-4" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>‚ÄúArchived Workflow URL‚Äù</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>‚ÄúArchived Workflow File‚Äù</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>

      <ul>
        <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
      </ul>
    </li>
    <li>Run <strong>Assembly polishing</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
      <ul>
        <li><em>‚ÄúSend results to a new history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1: Assembly to be polished‚Äù</em>: the assembly fasta file</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú2: long reads‚Äù</em>: the nanopore reads, the same set that was used for the assembly</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú3: Illumina reads R1‚Äù</em>: the illumina R1 fastq.gz file</li>
        <li>No other settings need to be changed, as they are set as described above, but recommended to set the Medaka model correctly (default = r941_min_high_g360). See drop down list for options.</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-4"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-4" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="polishing-results">Polishing results</h2>

<p>The polished assembly is the final Racon file.</p>

<p>Look at the Fasta Statistics output files for the status of the assemblies after each polishing tool. From these, we can see that some polishes decrease or increase the size of the assembly. The final size is approximately 300,000 bp shorter than the original Flye assembly. (These numbers may vary slightly even if the same input data is used).</p>

<h1 id="genome-quality-assessment">Genome quality assessment</h1>

<p>The polished genome is in 145 contigs with a total length of ~ 9.6 million base pairs.  We have some idea of how these contigs may be joined (or not) from the Bandage assembly graph.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true"></i> Question</div>

  <ol>
    <li>How good is the assembly?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true"></i> Solution<span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>One measure is the N50, a number that indicates how large the contigs are (although, have the contigs been joined correctly)? Another measure is to see if expected gene sequences are found in the assembly, using a tool called BUSCO. For a discussion on these and other methods, see <span class="citation"><a href="#Wang2020">Wang <i>et al.</i> 2020</a></span>.</li>
    </ol>

  </blockquote>
</blockquote>

<p>Here, we will use the BUSCO tool to annotate the genome and then assess whether expected genes are found. Note: this is a brief annotation only, not the full genome annotation that would typically be done following genome assembly.</p>

<p>For more about Busco see <span class="citation"><a href="#Simo2015">Sim√£o <i>et al.</i> 2015</a></span>.</p>

<p>Genome assessment:</p>

<p><img src="images/assess.png" alt="&quot;Image of genome assessment, showing contigs as bold bars, with one contig annotated (various sections labelled by coloured boxes), and then compared to an expected annotation.&quot;. " width="800" loading="lazy" /></p>

<p>We will also map the assembled contigs to a known reference genome using the tool Quast, to see how they align. For more about quast see <a href="http://quast.sourceforge.net/docs/manual.html">the manual</a>.</p>

<p>For more about the Icarus browser see <span class="citation"><a href="#Mikheenko2016">Mikheenko <i>et al.</i> 2016</a></span>.</p>

<h2 id="genome-assessment-workflow">Genome assessment workflow</h2>

<p>What it does:</p>
<ul>
  <li>Assesses the quality of the genome assembly: generate some statistics and determine if expected genes are present</li>
  <li>Align contigs to a reference genome.</li>
</ul>

<p>Inputs:</p>
<ul>
  <li>polished assembly</li>
  <li>reference_genome.fasta (e.g. of a closely-related species, if available).</li>
</ul>

<p>Outputs</p>
<ul>
  <li>Busco table of genes found</li>
  <li>Quast HTML report, and link to Icarus contigs browser, showing contigs aligned to a reference genome</li>
</ul>

<p>Tools used:</p>
<ul>
  <li>Busco</li>
  <li>Quast</li>
</ul>

<p>Input parameters: None required</p>

<p>Workflow steps:</p>
<ul>
  <li>Polished assembly =&gt; Busco:</li>
  <li>First: predict genes in the assembly: using Metaeuk</li>
  <li>Second: compare the set of predicted genes to the set of expected genes in a particular lineage. Default setting for lineage: Eukaryota</li>
  <li>Polished assembly and a reference genome =&gt; Quast:</li>
  <li>For the tutorial we will use the Arabidopsis genome. This is not closely related to our Eucalyptus species but will give an idea of how to use Quast.</li>
  <li>Contigs/scaffolds file: polished assembly</li>
  <li>Type of assembly: Genome</li>
  <li>Use a reference genome: Yes</li>
  <li>Reference genome: Arabidopsis genome</li>
  <li>Is the genome large (&gt; 100Mbp)? Yes. (Our test data set won‚Äôt be, but this will still give us some results).</li>
  <li>All other settings as defaults, except second last setting: Distinguish contigs with more than 50% unaligned bases as a separate group of contigs?: change to No</li>
</ul>

<p>Options:</p>
<ul>
  <li><strong>Gene prediction:</strong></li>
  <li>Change tool used by Busco to predict genes in the assembly: instead of Metaeuk, use Augustus (Metaeuk is meant to be faster; unsure which is better).</li>
  <li>select: Use Augustus; Use another predefined species model; then choose from the drop down list.</li>
  <li>select from a database of trained species models. Link to lists of species models: <a href="https://github.com/Gaius-Augustus/Augustus/tree/master/config/species">https://github.com/Gaius-Augustus/Augustus/tree/master/config/species</a></li>
  <li>Note: if using Augustus: it may fail if the input assembly is too small (e.g. a test-size data assembly). It can‚Äôt do the training part properly.</li>
  <li><strong>Compare genes found to other lineage:</strong></li>
  <li>Busco has databases of lineages and their expected genes. Option to change lineage. Not all lineages are available - there is a mix of broader and narrower lineages.</li>
  <li>link to list of lineages <a href="https://busco.ezlab.org/list_of_lineages.html">https://busco.ezlab.org/list_of_lineages.html</a></li>
  <li>link to groups in taxonomic hierarchies <a href="https://busco.ezlab.org/frames/euka.htm">https://busco.ezlab.org/frames/euka.htm</a></li>
  <li>For example,  if you have a plant species from Fabales, you could set that as the lineage.</li>
  <li>The narrower the taxonomic group, the more total genes are expected.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-assess-genome-quality-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Assess Genome Quality workflow</div>

  <ol>
    <li><strong>Import the Assess Genome Quality workflow</strong> into Galaxy:
      <ul>
        <li>Copy the URL (e.g. via right-click) of <a href="/training-material/topics/assembly/tutorials/largegenome/workflows/Galaxy-Workflow-Assess_genome_quality.ga">this workflow</a> or download it to your computer.</li>
        <li>Import the workflow into Galaxy</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-5"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-5" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Importing a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>‚ÄúArchived Workflow URL‚Äù</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>‚ÄúArchived Workflow File‚Äù</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul> </blockquote>

      <ul>
        <li>Click ‚ÄúExpand to full workflow form‚Äù</li>
      </ul>
    </li>
    <li>Run <strong>Assess Genome Quality</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:
      <ul>
        <li><em>‚ÄúSend results to a new history‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1: Input file: Polished assembly‚Äù</em>: the polished assembly fasta file from the previous polishing workflow</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú2: Input file: Reference genome‚Äù</em>: Arabidopsis.fasta</li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow-5"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow-5" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> Tip: Running a workflow<span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
    </li>
  </ol>
</blockquote>

<h3 id="assessment-results">Assessment results</h3>

<p>The output is a set of Quast and Busco reports.</p>

<p>Busco: As this is a test dataset, the assembly is small (~ 10 million base pairs, rather than ~ 1 billion base pairs), and likely missing most of the real genome (and genes). Thus, for this test case, we would expect most of the genes not to be found, but we can view the Busco results to see an example of how it works. Open the Busco short summary file: only 1 complete BUSCO has been found, out of 255 expected. You can re-run the workflow and change the lineage to ‚ÄúEmbryophyta‚Äù to see that more BUSCOs are found (8 out of 1614).</p>

<p>Open the Quast HTML report, and at the top of this, click on ‚ÄúView in Icarus contig browser‚Äù.  This shows how our assembly contigs have mapped to the reference genome. For this reference genome there are 5 chromosomes and two organelles. As this reference genome species is not closely related, not many contigs have mapped well. But we can see that some of them match the organelles (which is expected, as these reads are likely to be overrepresented in the test data). For the nuclear genome, there are some matches to parts of chromosome 2 and 3.</p>

<p><img src="images/quast2.png" alt="&quot;Image of quast results showing contig lengths.&quot;. " width="650" loading="lazy" /></p>

<p>Click on the ‚Äúmitochondria‚Äù to see how the assembly contigs align to the reference mitochondrial genome. Click the -5x button to zoom out to the full length.</p>

<p><img src="images/mitocontigs.png" alt="&quot;Image of mitochondrial contigs.&quot;. " width="650" loading="lazy" /></p>

<h1 id="combining-workflows">Combining workflows</h1>

<p>We can combine these galaxy workflows into a single workflow. (See <a href="/training-material/topics/galaxy-interface/tutorials/workflow-editor/tutorial.html">this GTN tutorial</a> for more information.)</p>

<h2 id="combined-assembly-and-polishing-workflow">Combined assembly and polishing workflow</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-run-the-combined-assembly-and-polishing-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Run the Combined Assembly and Polishing workflow</div>
  <ul>
    <li>Find this workflow, enter the correct input files, and run.</li>
    <li>What it does:  A workflow for genome assembly, containing subworkflows:
      <ul>
        <li>Data QC</li>
        <li>Kmer counting</li>
        <li>Trim and filter reads</li>
        <li>Assembly with Flye</li>
        <li>Assembly polishing</li>
        <li>Assess genome quality</li>
      </ul>
    </li>
    <li>Inputs
      <ul>
        <li>long reads and short reads in fastq format</li>
        <li>reference genome for Quast</li>
      </ul>
    </li>
    <li>Outputs
      <ul>
        <li>Data information - QC, kmers</li>
        <li>Filtered, trimmed reads</li>
        <li>Genome assembly, assembly graph, stats</li>
        <li>Polished assembly, stats</li>
        <li>Quality metrics - Busco, Quast</li>
      </ul>
    </li>
    <li>Tools used: Sum of tools in each of the subworkflows</li>
    <li>Input parameters: None required</li>
    <li>Workflow steps: For detail see each subworkflow</li>
    <li>Options:
      <ul>
        <li>Omit some steps - e.g. Data QC and kmer counting</li>
        <li>Replace a module with one using a different tool - e.g. change assembly tool</li>
      </ul>
    </li>
  </ul>
</blockquote>

<h1 id="summary">Summary</h1>

<p>In this tutorial we have assembled sequencing reads into contigs, using tools and workflows in Galaxy. Although we have used test data, these tools and workflows should work on real-sized eukaryotic data sets although please see our recommendations in the Introduction section.</p>

<p>A summary of the workflow steps and the main tools used:</p>

<p><img src="images/summary.png" alt="&quot;A graphical summary of the steps in the assembly: quality control, kmer counting, trimming/filtering reads, assembly, polishing and assessment. A composite of all the earlier images in the tutorial.&quot;. " width="650" loading="lazy" /></p>

<p>We hope this has been useful for both learning about genome assembly concepts and as a customisable example for your own assembly data.</p>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>We can assemble long reads from plant or animal species into large contigs</p>
</li>
                        
                        <li><p>These contigs can be polished with both long and short reads</p>
</li>
                        
                        <li><p>We can assess the quality of this assembly with various tools</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the  <a href="/training-material/topics/assembly/faqs/">FAQ page for the Assembly topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Simo2015">Sim√£o, F. A., R. M. Waterhouse, P. Ioannidis, E. V. Kriventseva, and E. M. Zdobnov, 2015 <b>BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs</b>. Bioinformatics 31: 3210‚Äì3212. <a href="https://doi.org/10.1093/bioinformatics/btv351">10.1093/bioinformatics/btv351</a></li>
<li id="Mikheenko2016">Mikheenko, A., G. Valin, A. Prjibelski, V. Saveliev, and A. Gurevich, 2016 <b>Icarus: visualizer for de novo assembly evaluation</b>. Bioinformatics 32: 3321‚Äì3323. <a href="https://doi.org/10.1093/bioinformatics/btw379">10.1093/bioinformatics/btw379</a></li>
<li id="Vurture2017">Vurture, G. W., F. J. Sedlazeck, M. Nattestad, C. J. Underwood, H. Fang <i>et al.</i>, 2017 <b>GenomeScope: fast reference-free genome profiling from short reads</b> (B. Berger, Ed.). Bioinformatics 33: 2202‚Äì2204. <a href="https://doi.org/10.1093/bioinformatics/btx153">10.1093/bioinformatics/btx153</a></li>
<li id="Rhie2020">Rhie, A., B. P. Walenz, S. Koren, and A. M. Phillippy, 2020 <b>Merqury: reference-free quality,  completeness,  and phasing assessment for genome assemblies</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-02134-9">10.1186/s13059-020-02134-9</a></li>
<li id="Wang2020">Wang, W., A. Das, D. Kainer, M. Schalamun, A. Morales-Suarez <i>et al.</i>, 2020 <b>The draft nuclear genome assembly of Eucalyptus pauciflora: a pipeline for comparing de novo assemblies</b>. GigaScience 9: <a href="https://doi.org/10.1093/gigascience/giz160">10.1093/gigascience/giz160</a></li>
<li id="Aury2021">Aury, J.-M., and B. Istace, 2021 <b>Hapo-G,  haplotype-aware polishing of genome assemblies with accurate reads</b>. NAR Genomics and Bioinformatics 3: <a href="https://doi.org/10.1093/nargab/lqab034">10.1093/nargab/lqab034</a></li>
<li id="Chen2021">Chen, Y., F. Nie, S.-Q. Xie, Y.-F. Zheng, Q. Dai <i>et al.</i>, 2021 <b>Efficient assembly of nanopore reads via highly accurate and intact error correction</b>. Nature Communications 12: <a href="https://doi.org/10.1038/s41467-020-20236-7">10.1038/s41467-020-20236-7</a></li>
<li id="Garg2021">Garg, S., 2021 <b>Computational methods for chromosome-scale haplotype reconstruction</b>. Genome Biology 22: <a href="https://doi.org/10.1186/s13059-021-02328-9">10.1186/s13059-021-02328-9</a></li>
<li id="McCartney2021">McCartney, A. M., E. Hilario, S.-S. Choi, J. Guhlin, J. M. Prebble <i>et al.</i>, 2021 <b>An exploration of assembly strategies and quality metrics on the accuracy of the rewarewa Knightia excelsa genome</b>. Molecular Ecology Resources. <a href="https://doi.org/10.1111/1755-0998.13406">10.1111/1755-0998.13406</a></li>
<li id="Nurk2021">Nurk, S., S. Koren, A. Rhie, M. Rautiainen, A. V. Bzikadze <i>et al.</i>, 2021 <b>The complete sequence of a human genome</b>. <a href="https://doi.org/10.1101/2021.05.26.445798">10.1101/2021.05.26.445798</a></li></ol>
                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Large genome assembly and polishing (Assembly)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Anna Syme,  <b>Large genome assembly and polishing (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/largegenome/tutorial.html">https://training.galaxyproject.org/training-material/topics/assembly/tutorials/largegenome/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{assembly-largegenome,
author = "Anna Syme",
title = "Large genome assembly and polishing (Galaxy Training Materials)",
year = "",
month = "",
day = ""
url = "\url{https://training.galaxyproject.org/training-material/topics/assembly/tutorials/largegenome/tutorial.html}",
note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine F√∂ll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Bj√∂rn Gr√ºning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
var d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                
                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        
        <script src="/training-material/assets/js/jquery.slim.min.js"></script>
        <script src="/training-material/assets/js/popper.min.js"></script>
        <script src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
        <script src="/training-material/assets/js/details-element-polyfill.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
        <script src="/training-material/assets/js/bootstrap-toc.min.js"></script>
        <script src="/training-material/assets/js/main.js"></script>
        <script src="/training-material/assets/js/clipboard.min.js"></script>
        <script>
        var snippets=document.querySelectorAll('div.highlight');
        [].forEach.call(snippets,function(snippet){
            snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet tabindex="0"><i class="fa fa-copy"></i>&nbsp;Copy</button>');
        });

        var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
            target:function(trigger){return trigger.nextElementSibling;
        }});
        </script>
        

        <script type="text/javascript">
            if(window.location.hostname === "galaxyproject.github.io") {
                // Redirect
                var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
                $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

                window.setTimeout(function(){
                    window.location.href = redirect;
                }, 5000)

            }
        </script>
    </body>
</html>