<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Slides: Deeper look into Genome Assembly algorithms / Deeper look into Genome Assembly algorithms / Assembly</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@715a5f0aefd15f73c5a3222b576a5f6c21ada9c5",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/algorithms-introduction/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Deeper look into Genome Assembly algorithms">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2021-01-12 13:46:54 +0000">
<meta name="DC.creator" content="Anthony Bretaudeau">
<meta name="DC.creator" content="Simon Gladman">
<meta name="DC.creator" content="Anton Nekrutenko">
<meta name="DC.creator" content="Delphine Lariviere"><meta name="description" content="# Different types of input data  * Short reads (Illumina): numerous 👍, high quality 👍, cheap 👍, short 👎 * Long reads (PacBio, Nanopore): longer 👍, fewer 👎, (many) more errors 👎  Genome Assembly can be done with: * only short reads * only long reads * both (hybrid assembly)  Specific algorithms for each  ---  # Genome Assembly algorithms  Detect overlaps between reads to build the longest possible sequences  Algorithms use graphs to represent overlapping reads/words  Two steps:  * Build a (huge) graph while reading the input data * Try to find the longest paths traversing the graph  Two main types of algorithms:  * Short reads: de Bruijn Graphs * Long reads: OLC (Overlap Layout Consensus)  ---  # OLC (Overlap Layout Consensus)  The older, first used for Sanger sequencing  Compare all reads, look for read overlaps  If a *suffix* of one read is similar to a *prefix* of another read...  ``` TCTATATCTCGGCTCTAGG    **Directed graph** representing overlapping reads. (Image from [Ben Langme...">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Assembly / Deeper look into Genome Assembly algorithms / Slides: Deeper look into Genome Assembly algorithms">
        <meta property="og:description" content="# Different types of input data  * Short reads (Illumina): numerous 👍, high quality 👍, cheap 👍, short 👎 * Long reads (PacBio, Nanopore): longer 👍, fewer 👎, (many) more errors 👎  Genome Assembly can be done with: * only short reads * only long reads * both (hybrid assembly)  Specific algorithms for each  ---  # Genome Assembly algorithms  Detect overlaps between reads to build the longest possible sequences  Algorithms use graphs to represent overlapping reads/words  Two steps:  * Build a (huge) graph while reading the input data * Try to find the longest paths traversing the graph  Two main types of algorithms:  * Short reads: de Bruijn Graphs * Long reads: OLC (Overlap Layout Consensus)  ---  # OLC (Overlap Layout Consensus)  The older, first used for Sanger sequencing  Compare all reads, look for read overlaps  If a *suffix* of one read is similar to a *prefix* of another read...  ``` TCTATATCTCGGCTCTAGG    **Directed graph** representing overlapping reads. (Image from [Ben Langme...">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/assembly/tutorials/algorithms-introduction/slides.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/algorithms-introduction/slides-plain.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/assembly/tutorials/algorithms-introduction/slides.html">View markdown source on GitHub</a>
	<h1>Deeper look into Genome Assembly algorithms</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		
  <a href="/training-material/hall-of-fame/abretaud/" class="contributor-badge contributor-abretaud"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/abretaud?s=36" alt="Anthony Bretaudeau avatar" width="36" class="avatar" />
    Anthony Bretaudeau</a>
  
  <a href="/training-material/hall-of-fame/slugger70/" class="contributor-badge contributor-slugger70"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/slugger70?s=36" alt="Simon Gladman avatar" width="36" class="avatar" />
    Simon Gladman</a>
  
  <a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge contributor-nekrut"><img src="https://avatars.githubusercontent.com/nekrut?s=36" alt="Anton Nekrutenko avatar" width="36" class="avatar" />
    Anton Nekrutenko</a>
  
  <a href="/training-material/hall-of-fame/delphine-l/" class="contributor-badge contributor-delphine-l"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/delphine-l?s=36" alt="Delphine Lariviere avatar" width="36" class="avatar" />
    Delphine Lariviere</a>
  
	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>What are the main types of assembly algorithms?</p>
</li>
		
		<li><p>How do they perform with short and long reads?</p>
</li>
		
		<li><p>How to improve assemblies using other technologies?</p>
</li>
		
		</ul>
		

		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Published:</strong> Jan 12, 2021 </div>
		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last Updated:</strong> Jan 12, 2021 </div>

	<hr />


	<h1 id="different-types-of-input-data">Different types of input data</h1>

<ul>
  <li>Short reads (Illumina): numerous 👍, high quality 👍, cheap 👍, short 👎</li>
  <li>Long reads (PacBio, Nanopore): longer 👍, fewer 👎, (many) more errors 👎</li>
</ul>

<p>Genome Assembly can be done with:</p>
<ul>
  <li>only short reads</li>
  <li>only long reads</li>
  <li>both (hybrid assembly)</li>
</ul>

<p>Specific algorithms for each</p>

<hr />

<h1 id="genome-assembly-algorithms">Genome Assembly algorithms</h1>

<p>Detect overlaps between reads to build the longest possible sequences</p>

<p>Algorithms use graphs to represent overlapping reads/words</p>

<p>Two steps:</p>

<ul>
  <li>Build a (huge) graph while reading the input data</li>
  <li>Try to find the longest paths traversing the graph</li>
</ul>

<p>Two main types of algorithms:</p>

<ul>
  <li>Short reads: de Bruijn Graphs</li>
  <li>Long reads: OLC (Overlap Layout Consensus)</li>
</ul>

<hr />

<h1 id="olc-overlap-layout-consensus">OLC (Overlap Layout Consensus)</h1>

<p>The older, first used for Sanger sequencing</p>

<p>Compare all reads, look for read overlaps</p>

<p>If a <em>suffix</em> of one read is similar to a <em>prefix</em> of another read…</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>TCTATATCTCGGCTCTAGG    &lt;- read 1
    ||||||| |||||||
    TATCTCGACTCTAGGCC  &lt;- read 2
</code></pre></div></div>

<p>…then they may overlap within the genome.</p>

<hr />

<h2 id="directed-graphs">Directed graphs</h2>

<p>We build a <em>directed graph</em> where directed <em>edges</em> connect <em>nodes</em> representing overlapping reads:
<img src="../../images/dag.png" alt="Image showing example reads represented as nodes, and an overlap between two reads as an edge" /></p>

<blockquote>
  <p><strong>Directed graph</strong> representing overlapping reads. (Image from <a href="http://www.cs.jhu.edu/~langmea/resources/lecture_notes/assembly_scs.pdf">Ben Langmead</a>).</p>
</blockquote>

<hr />

<h2 id="directed-graphs-1">Directed graphs</h2>

<p>For example, consider these 15 three letter reads:</p>

<p>TAA AAT ATG TGC GCC CCA CAT ATG TGG GGG GGA GAT ATG TGT GTT</p>

<p>We’re lucky, we know the corresponding genome sequence is TAATGCCATGGGATGTT</p>

<p>So the 15 reads can be represented as the following directed graph:</p>

<p><img src="../../images/4.6.png" alt="A graph representing the 15 overlapping example reads" /></p>

<hr />

<p><strong>However</strong>, in real life we do not know the actual genome! All we have is a collection of reads. Let’s first build an overlap graph by connecting two 3-mers if suffix of one is equal to the prefix of the other:</p>

<blockquote>
  <p><img src="../../images/4.7.png" alt="All possible overlap connections for our 3-mer collection" /></p>

  <p><strong>Overlap graph</strong>. All possible overlap connections for our 3-mer collection. (Fig. 4.7 from <a href="http://bioinformaticsalgorithms.com/">CP</a>)</p>
</blockquote>

<p>So to determine the sequence of the underlying genome we are looking for a path in this graph that visits every node (3-mer) once. Such path is called <a href="https://en.wikipedia.org/wiki/Hamiltonian_path">Hamiltonian path</a> and it may not be unique.</p>

<hr />

<p>For example for our 3-mer collection there are two possible Hamiltonian paths:</p>

<p><img src="../../images/4.9a.png" alt="First possible Hamiltonian path" />
<img src="../../images/4.9b.png" alt="Second possible Hamiltonian path" /></p>

<p><strong>Two Hamiltonian paths for the 15 3-mers</strong>. Edges spelling “genomes” <code>TAATGCCATGGGATGTT</code> and <code>TAATGGGATGCCATGTT</code> are highlighted in black. (Fig. 4.9. from <a href="http://bioinformaticsalgorithms.com/">[CP](http://bioinformaticsalgorithms.com/)</a>).</p>

<hr />

<p>The reason for this “duality” is the fact that we have a <em>repeat</em>: 3-mer <code>ATG</code> is present three times in our data (<font color="green">green</font>, <font color="red">red</font>, and <font color="blue">blue</font>). Repeats cause a lot of trouble in genome assembly.</p>

<hr />

<h1 id="limits-of-olc">Limits of OLC</h1>

<ul>
  <li>Computationally intensive</li>
  <li>Doesn’t perform well with Illumina (massive amount of short reads)</li>
</ul>

<p>=&gt; deBruijn Graphs</p>

<hr />

<h1 id="de-bruijn-graphs"><strong>de Bruijn Graphs</strong></h1>

<p>Based on a counter-intuitive idea</p>

<ul>
  <li>Hash all reads in overlapping fixed-length words</li>
  <li>Words = k-mers of size k</li>
  <li>Build a directed graph where nodes are k-mers and edges represent overlaps between k-mers</li>
</ul>

<hr />

<h1 id="what-are-k-mers"><strong>What are K-mers?</strong></h1>
<p><img src="../../images/kmers01.png" alt="K-mers are subwords of length k of a string" /></p>

<hr />

<h1 id="k-mers-de-bruijn-graph"><strong>K-mers de Bruijn graph</strong></h1>
<p><img src="../../images/ex1-1.png" alt="Example with 4 english words" /></p>

<hr />

<h1 id="k-mers-de-bruijn-graph-1"><strong>K-mers de Bruijn graph</strong></h1>
<p><img src="../../images/ex1-2.png" alt="Example with 4 english words" /></p>

<hr />

<h1 id="k-mers-de-bruijn-graph-2"><strong>K-mers de Bruijn graph</strong></h1>
<p><img src="../../images/ex1-3.png" alt="Example with 4 english words" /></p>

<hr />

<h1 id="k-mers-de-bruijn-graph-3"><strong>K-mers de Bruijn graph</strong></h1>
<p><img src="../../images/ex1-4.png" alt="Example with 4 english words" /></p>

<hr />

<h1 id="the-problem-of-repeats"><strong>The problem of repeats</strong></h1>
<p><img src="../../images/ex2-1.png" alt="Example with a word containing repeats: Mississippi" /></p>

<hr />

<h1 id="the-problem-of-repeats-1"><strong>The problem of repeats</strong></h1>
<p><img src="../../images/ex2-2.png" alt="Example with a word containing repeats: Mississippi" /></p>

<hr />

<h1 id="the-problem-of-repeats-2"><strong>The problem of repeats</strong></h1>
<p><img src="../../images/ex2-3.png" alt="Example with a word containing repeats: Mississippi" /></p>

<hr />

<h1 id="the-problem-of-repeats-3"><strong>The problem of repeats</strong></h1>
<p><img src="../../images/ex2-4.png" alt="Example with a word containing repeats: Mississippi" /></p>

<hr />

<h1 id="different-k"><strong>Different k</strong></h1>
<p><img src="../../images/ex2a-1.png" alt="Effect of the size of k" />
—</p>
<h1 id="different-k-1"><strong>Different k</strong></h1>
<p><img src="../../images/ex2a-2.png" alt="Effect of the size of k" />
—</p>
<h1 id="different-k-2"><strong>Different k</strong></h1>
<p><img src="../../images/ex2a-3.png" alt="Effect of the size of k" />
—</p>
<h1 id="different-k-3"><strong>Different k</strong></h1>
<p><img src="../../images/ex2a-4.png" alt="Effect of the size of k" /></p>

<hr />
<h1 id="choose-k-wisely"><strong>Choose k wisely</strong></h1>

<ul>
  <li>Lower k
    <ul>
      <li>More connections</li>
      <li>Less chance of resolving small repeats</li>
      <li>Higher k-mer coverage</li>
    </ul>
  </li>
  <li>Higher k
    <ul>
      <li>Less connections</li>
      <li>More chance of resolving small repeats</li>
      <li>Lower k-mer coverage</li>
    </ul>
  </li>
</ul>

<p><strong><em>Optimum value for k will balance these effects.</em></strong></p>

<hr />

<h1 id="read-errors"><strong>Read errors</strong></h1>
<p>.image-75[<img src="../../images/ex3-head.png" alt="Example of english words with errors" />]
<img src="../../images/blank.png" alt="Example of english words with errors" />
—</p>

<h1 id="read-errors-1"><strong>Read errors</strong></h1>
<p>.image-75[<img src="../../images/ex3-head.png" alt="Example of english words with errors" />]
<img src="../../images/ex3-2.png" alt="Example of english words with errors" />
—</p>
<h1 id="read-errors-2"><strong>Read errors</strong></h1>
<p>.image-75[<img src="../../images/ex3-head.png" alt="Example of english words with errors" />]
<img src="../../images/ex3-3.png" alt="Example of english words with errors" />
—</p>
<h1 id="read-errors-3"><strong>Read errors</strong></h1>
<p>.image-75[<img src="../../images/ex3-head.png" alt="Example of english words with errors" />]
<img src="../../images/ex3-4.png" alt="Example of english words with errors" /></p>

<hr />

<h1 id="coverage-or-sequencing-depth">Coverage (or sequencing depth)</h1>

<p>The number of unique reads that contain a given nucleotide in the reconstructed sequence.</p>

<p><img src="../../images/coverage.png" alt="Aligned reads showing the notion of coverage" /></p>

<p>Source: <a href="https://commons.wikimedia.org/wiki/File:Read,_read_length_and_read_depth_to_achieve_a_read_depth">Wikimedia</a></p>

<p><strong>Estimated coverage</strong></p>

<p><strong>(number of reads x read length)</strong> / <strong>estimated size of the genome</strong></p>

<p>1,000,000 reads x 150bp / 10Mb = <strong>15X</strong></p>

<hr />
<h1 id="more-coverage"><strong>More coverage</strong></h1>

<ul>
  <li>Errors won’t be duplicated in every read</li>
  <li>Most reads will be error free</li>
  <li>We can count the frequency of each k-mer</li>
  <li>Annotate the graph with the frequencies</li>
  <li>Use the frequency data to clean the de Bruijn graph</li>
</ul>

<h2 id="more-coverage-depth-will-help-overcome-errors"><strong><em>More coverage depth will help overcome errors!</em></strong></h2>
<h1 id="read-errors-revisited"><strong>Read errors revisited</strong></h1>
<p><img src="../../images/ex3a.png" alt="Graph annotated with coverage" /></p>

<hr />

<h1 id="de-bruijn-graph-assembly-process"><strong>de Bruijn graph assembly process</strong></h1>

<ol>
  <li>Select a value for k</li>
  <li>“Hash” the reads (make the kmers)</li>
  <li>Count the kmers</li>
  <li>Make the de Bruijn graph</li>
  <li>Perform graph simplification steps</li>
  <li>Read off contigs from simplified graph</li>
</ol>

<hr />
<h1 id="make-contigs"><strong>Make contigs</strong></h1>

<ul>
  <li>Find an unbalanced node in the graph</li>
  <li>Follow the chain of nodes and “read off” the bases to produce the contigs</li>
  <li>Where there is an ambiguous divergence/convergence, stop the current contig and start a new one.</li>
  <li>Re-trace the reads through the contigs to help with repeat resolution</li>
</ul>

<hr />

<h1 id="assembly-in-real-life">Assembly in real life</h1>

<p>In this topic we’ve learned about two ways of representing the relationship between reads derived from a genome we are trying to assemble:</p>

<ol>
  <li><strong>Overlap graphs</strong> - nodes are reads, edges are overlaps between reads.</li>
  <li><strong>De Bruijn graphs</strong> - nodes are overlaps, edges are reads.</li>
</ol>

<hr />

<h1 id="overlap-graph">Overlap graph</h1>

<p><img src="../../images/4.7.png" alt="An example overlap graph" /></p>

<hr />

<h1 id="de-bruijn-graph---same-data">de Bruijn Graph - same data</h1>

<p><img src="../../images/4.13f.png" alt="An example de Bruijn graph" /></p>

<hr />

<p>Whatever the representation will be it will be messy:</p>

<p><img src="../../images/t12_mess.png" alt="An example of a real-life, noisy graph" /></p>

<p>A fragment of a very large De Bruijn graph (Image from <a href="https://github.com/BenLangmead/ads1-slides/blob/master/0580_asm__practice.pdf">BL</a>).</p>

<hr />

<p>There are multiple reasons for such messiness:</p>

<p><strong>Sequence errors</strong></p>

<p>Sequencing machines do not give perfect data. This results in spurious deviations on the graph. Some sequencing technologies such as Oxford Nanopore have very high error rate of ~15%.</p>

<blockquote>
  <p><img src="../../images/t12_errors.png" alt="Zoom on graph region containing sequencing errors" /></p>

  <p>Graph components resulting from sequencing errors (Image from <a href="https://github.com/BenLangmead/ads1-slides/blob/master/0580_asm__practice.pdf">BL</a>).</p>
</blockquote>

<hr />

<p><strong>Ploidy</strong></p>

<p>As we discussed earlier humans are diploid and there are multiple differences between maternal and paternal genomes. This creates “bubbles” on assembly graphs:</p>

<blockquote>
  <p><img src="../../images/t12_bubble.png" alt="Impact of diploid genome sequencing on graphs" /></p>

  <p>Bubbles due to a heterozygous site  (Image from <a href="https://github.com/BenLangmead/ads1-slides/blob/master/0580_asm__practice.pdf">BL</a>).</p>
</blockquote>

<hr />

<p><strong>Repeats</strong></p>

<p>As we’ve seen the third law of assembly is unbeatable. As a result some regions of the genome simply cannot be resolved and are reported in segments called <em>contigs</em>:</p>

<blockquote>
  <p><img src="../../images/t12_contigs.png" alt="Impact of repeats on graphs" /></p>

  <p>The following “genomic” segment will be reported in three pieces corresponding to regions flanking the repeat and repeat itself (Image from <a href="https://github.com/BenLangmead/ads1-slides/blob/master/0580_asm__practice.pdf">BL</a>).</p>
</blockquote>

<hr />

<h1 id="what-to-do-with-my-reads">What to do with my reads?</h1>

<ul>
  <li>Short reads =&gt; DBG assemblers (e.g. Spades, ABySS, DISCOVAR, Velvet, …)</li>
  <li>Long reads =&gt; OLC assemblers (e.g Canu, Falcon, Hgap4, …)</li>
  <li>Short + Long reads =&gt; hybrid assemblers (e.g. Unicycler, …)</li>
</ul>

<p>Hybrid assembly: long reads to resolve repeats, short reads to correct errors</p>

<p>Other data and tools for polishing (scaffolding, gap filling, …)</p>

<hr />

<h1 id="other-data-optical-maps">Other data: Optical maps</h1>

<p><img src="../../images/Optical_mapping.jpg" alt="Optical mapping workflow" />
(Source: <a href="https://en.wikipedia.org/wiki/Optical_mapping#/media/File:Optical_mapping.jpg">Wikimedia</a>)</p>

<p>Use for scaffolding by comparing the restriction map with the predicted restriction sites in the contig sequences</p>

<hr />

<h1 id="other-data--hi-c">Other data:  Hi-C</h1>

<p><img src="../../images/hic.jpg" alt="Hi-C sequencing workflow" />
(Source: Dovetail genomics)</p>

<p>Sequence chunks of genome colocalized</p>

<hr />

<h1 id="other-data-linked-reads">Other data: Linked-Reads</h1>

<p><img src="../../images/linked_reads.png" alt="Linked-Reads workflow" />
(Source: 10X)</p>

<p>Reads grouped by physical regions on the genome thanks to barcodes</p>

<p>10X Genomics (discontinued in 2020)</p>


	<hr />








<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


Tutorial Content is licensed under

  <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.

</section>






        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/assembly/tutorials/algorithms-introduction/slides.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/./topics/assembly/tutorials/algorithms-introduction/slides.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/715a5f0aefd15f73c5a3222b576a5f6c21ada9c5">715a5f0</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>
