<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>VGP assembly pipeline</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@5c4f29500474814d08883ef5f8fb236fe4a6c378",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.f1f2de89.js?v=1702253434' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.03aefefa.js?v=1702253434' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="VGP assembly pipeline">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-27 16:39:55 +0000">
<meta name="DC.creator" content="Delphine Lariviere">
<meta name="DC.creator" content="Alex Ostrovsky">
<meta name="DC.creator" content="Cristóbal Gallardo">
<meta name="DC.creator" content="Anna Syme">
<meta name="DC.creator" content="Linelle Abueg">
<meta name="DC.creator" content="Brandon Pickett">
<meta name="DC.creator" content="Giulio Formenti">
<meta name="DC.creator" content="Marcella Sozzoni">
<meta name="DC.creator" content="Anton Nekrutenko">

        
        
        
        
        <meta name="description" content="DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these field...">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: VGP assembly pipeline">
        <meta property="og:description" content="DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these field...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "dateModified": "2023-11-27 16:39:55 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "VGP assembly pipeline",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "identifier": "https://gxy.io/GTN:T00039",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Assembly",
    "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
    "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "VGP assembly pipeline",
  "url": "https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html",
  "version": 155,
  "timeRequired": "PT5H",
  "teaches": "Learn the tools necessary to perform a de novo assembly of a vertebrate genome\n - Evaluate the quality of the assembly",
  "keywords": [
    "Assembly",
    "pacbio",
    "eukaryote",
    "VGP"
  ],
  "description": "The questions this  addresses are:\n - What combination of tools can produce the highest quality assembly of vertebrate genomes?\n - How can we evaluate the quality of the assembly in a reference-free way?\n\n\nThe objectives are:\n - Learn the tools necessary to perform a de novo assembly of a vertebrate genome\n - Evaluate the quality of the assembly\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/delphine-l/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/delphine-l/",
      "name": "Delphine Lariviere",
      "image": "https://avatars.githubusercontent.com/delphine-l",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6421-3484",
      "orcid": "https://orcid.org/0000-0001-6421-3484"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/astrovsky01/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/astrovsky01/",
      "name": "Alex Ostrovsky",
      "image": "https://avatars.githubusercontent.com/astrovsky01",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-7901-7109",
      "orcid": "https://orcid.org/0000-0002-7901-7109"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "name": "Cristóbal Gallardo",
      "image": "https://avatars.githubusercontent.com/gallardoalba",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5752-2155",
      "orcid": "https://orcid.org/0000-0002-5752-2155"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/annasyme/",
      "name": "Anna Syme",
      "image": "https://avatars.githubusercontent.com/annasyme",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9906-0673",
      "orcid": "https://orcid.org/0000-0002-9906-0673"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/abueg/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/abueg/",
      "name": "Linelle Abueg",
      "image": "https://avatars.githubusercontent.com/abueg",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pickettbd/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pickettbd/",
      "name": "Brandon Pickett",
      "image": "https://avatars.githubusercontent.com/pickettbd",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/gf777/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/gf777/",
      "name": "Giulio Formenti",
      "image": "https://avatars.githubusercontent.com/gf777",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/msozzoni/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/msozzoni/",
      "name": "Marcella Sozzoni",
      "image": "https://avatars.githubusercontent.com/msozzoni",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/nekrut/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/nekrut/",
      "name": "Anton Nekrutenko",
      "image": "https://avatars.githubusercontent.com/nekrut",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Assembly",
      "description": "DNA sequence data has become an indispensable tool for Molecular Biology & Evolutionary Biology. Study in these fields now require a genome sequence to work from. We call this a 'Reference Sequence.' We need to build a reference for each species. We do this by Genome Assembly. De novo Genome Assembly is the process of reconstructing the original DNA sequence from the fragment reads alone.",
      "url": "https://training.galaxyproject.org/training-material/topics/assembly/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0196",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0196",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0196"
    }
  ],
  "educationalLevel": "Intermediate",
  "mentions": [
    {
      "@type": "Thing",
      "name": "pacbio"
    },
    {
      "@type": "Thing",
      "name": "eukaryote"
    },
    {
      "@type": "Thing",
      "name": "VGP"
    }
  ],
  "abstract": "Advances in sequencing technologies over the last few decades have revolutionized the field of genomics, allowing for a reduction in both the time and resources required to *de novo* genome assembly. Until recently, second-generation sequencing technologies (also known as next generation sequencing (NGS)) produced highly accurate but short (up to 800bp) reads. Those read extension was not long enough to cope with the difficulties associated with repetitive regions. Today, so-called third-generation sequencing (TGS) technologies, usually known as single-molecule real-time (SMRT) sequencing, have become dominant in *de novo* assembly of large genomes. TGS can use native DNA without amplification, reducing sequencing error and bias ({% cite Hon2020 %}, {% cite Giani2020 %}). Very recently, Pacific Biosciences introduced high fidelity (HiFi) sequencing, which produces reads 10-25 kbp in length with a minimum accuracy of 99% (Q20). In this tutorial you will use HiFi reads in combination with data from additional sequencing technologies to generate a high-quality genome assembly."
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.f1f2de89.js?v=1702253434'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/assembly" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Assembly
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/assembly/faqs/" title="Check our FAQs for the Assembly topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/vgp_genome_assembly/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/smorgasbord2023/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">smorgasbord2023</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-12-01/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">2023-12-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-11-01/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">2023-11-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-assembly">
    <h1 data-toc-skip>VGP assembly pipeline</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/delphine-l/" class="contributor-badge contributor-delphine-l"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/delphine-l?s=36" alt="Delphine Lariviere avatar" width="36" class="avatar" />
    Delphine Lariviere</a><a href="/training-material/hall-of-fame/astrovsky01/" class="contributor-badge contributor-astrovsky01"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/astrovsky01?s=36" alt="Alex Ostrovsky avatar" width="36" class="avatar" />
    Alex Ostrovsky</a><a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge contributor-gallardoalba"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/gallardoalba?s=36" alt="Cristóbal Gallardo avatar" width="36" class="avatar" />
    Cristóbal Gallardo</a><a href="/training-material/hall-of-fame/annasyme/" class="contributor-badge contributor-annasyme"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/annasyme?s=36" alt="Anna Syme avatar" width="36" class="avatar" />
    Anna Syme</a><a href="/training-material/hall-of-fame/abueg/" class="contributor-badge contributor-abueg"><img src="https://avatars.githubusercontent.com/abueg?s=36" alt="Linelle Abueg avatar" width="36" class="avatar" />
    Linelle Abueg</a><a href="/training-material/hall-of-fame/pickettbd/" class="contributor-badge contributor-pickettbd"><img src="https://avatars.githubusercontent.com/pickettbd?s=36" alt="Brandon Pickett avatar" width="36" class="avatar" />
    Brandon Pickett</a><a href="/training-material/hall-of-fame/gf777/" class="contributor-badge contributor-gf777"><img src="https://avatars.githubusercontent.com/gf777?s=36" alt="Giulio Formenti avatar" width="36" class="avatar" />
    Giulio Formenti</a><a href="/training-material/hall-of-fame/msozzoni/" class="contributor-badge contributor-msozzoni"><img src="https://avatars.githubusercontent.com/msozzoni?s=36" alt="Marcella Sozzoni avatar" width="36" class="avatar" />
    Marcella Sozzoni</a><a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge contributor-nekrut"><img src="https://avatars.githubusercontent.com/nekrut?s=36" alt="Anton Nekrutenko avatar" width="36" class="avatar" />
    Anton Nekrutenko</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What combination of tools can produce the highest quality assembly of vertebrate genomes?</p>
</li>
        
        <li><p>How can we evaluate the quality of the assembly in a reference-free way?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Learn the tools necessary to perform a de novo assembly of a vertebrate genome</p>
</li>
        
        <li><p>Evaluate the quality of the assembly</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 5 hours</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Intermediate <span class="visually-hidden">Intermediate</span>
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/5887339">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/workflows/" title="VGP assembly pipeline workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/assembly/tutorials/vgp_genome_assembly/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 27, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00039">https://gxy.io/GTN:T00039</a> </div>
        

	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 155 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Advances in sequencing technologies over the last few decades have revolutionized the field of genomics, allowing for a reduction in both the time and resources required to <em>de novo</em> genome assembly. Until recently, second-generation sequencing technologies (also known as next generation sequencing (NGS)) produced highly accurate but short (up to 800bp) reads. Those read extension was not long enough to cope with the difficulties associated with repetitive regions. Today, so-called third-generation sequencing (TGS) technologies, usually known as single-molecule real-time (SMRT) sequencing, have become dominant in <em>de novo</em> assembly of large genomes. TGS can use native DNA without amplification, reducing sequencing error and bias (<span class="citation"><a href="#Hon2020">Hon <i>et al.</i> 2020</a></span>, <span class="citation"><a href="#Giani2020">Giani <i>et al.</i> 2020</a></span>). Very recently, Pacific Biosciences introduced high fidelity (HiFi) sequencing, which produces reads 10-25 kbp in length with a minimum accuracy of 99% (Q20). In this tutorial you will use HiFi reads in combination with data from additional sequencing technologies to generate a high-quality genome assembly.</p>

<p>Deciphering the structural organization of complex vertebrate genomes is currently one of the largest challenges in genomics (<span class="citation"><a href="#Frenkel2012">Frenkel <i>et al.</i> 2012</a></span>). Despite the significant progress made in recent years, a key question remains: what combination of data and tools can produce the highest quality assembly? In order to adequately answer this question, it is necessary to analyse two of the main factors that determine the difficulty of genome assembly processes: repetitive content and heterozygosity.</p>

<p>Repetitive elements can be grouped into two categories: interspersed repeats, such as transposable elements (TE) that occur at multiple loci throughout the genome, and tandem repeats (TR) that occur at a single locus (<span class="citation"><a href="#Trresen2019">Tørresen <i>et al.</i> 2019</a></span>). Repetitive elements are an important component of eukaryotic genomes, constituting over a third of the genome in the case of mammals (<span class="citation"><a href="#SoteroCaio2017">Sotero-Caio <i>et al.</i> 2017</a></span>, <span class="citation"><a href="#Chalopin2015">Chalopin <i>et al.</i> 2015</a></span>). In the case of tandem repeats, various estimates suggest that they are present in at least one third of human protein sequences (<span class="citation"><a href="#Marcotte1999">Marcotte <i>et al.</i> 1999</a></span>). TE content is among the main factors contributing to the lack of continuity in the reconstruction of genomes, especially in the case of large ones, as TE content is highly correlated with genome size (<span class="citation"><a href="#SoteroCaio2017">Sotero-Caio <i>et al.</i> 2017</a></span>). On the other hand, TR usually lead to local genome assembly collapse, especially when their length is close to that of the reads (<span class="citation"><a href="#Trresen2019">Tørresen <i>et al.</i> 2019</a></span>).</p>

<p>Heterozygosity is also an important factor impacting genome assembly. Haplotype phasing, the identification of alleles that are co-located on the same chromosome, has become a fundamental problem in heterozygous and polyploid genome assemblies (<span class="citation"><a href="#Zhang2020">Zhang <i>et al.</i> 2020</a></span>). When no reference sequence is available, the <em>state-of-the-art</em> strategy consists of constructing a string graph with vertices representing reads and edges representing consistent overlaps. In this kind of graph, after transitive reduction, heterozygous alleles in the string graph are represented by bubbles. When combined with all-versus-all chromatin conformation capture (Hi-C) data, this approach allows complete diploid reconstruction (<span class="citation"><a href="#DominguezDelAngel2018">Angel <i>et al.</i> 2018</a></span>, <span class="citation"><a href="#Zhang2020">Zhang <i>et al.</i> 2020</a></span>, <span class="citation"><a href="#Dida2021">Dida and Yi 2021</a></span>).</p>

<p>The Genome 10K consortium (G10K) launched the Vertebrate Genome Project (Vertebrate Genomes Project (VGP)), whose goal is generating high-quality, near-error-free, gap-free, chromosome-level, haplotype-phased, annotated reference genome assemblies for every vertebrate species (<span class="citation"><a href="#Rhie2021">Rhie <i>et al.</i> 2020</a></span>). This tutorial will guide you step by step to assemble a high-quality genome using the VGP assembly pipeline, including multiple Quality Control (QC) evaluations.</p>

<blockquote class="warning">
  <div class="box-title warning-title" id="warning-your-results-may-differ"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Your results may differ!</div>

  <p>Some of your results may slightly differ from results shown in this tutorial, depending on the versions of the tools used, since algorithms can change between versions.</p>

</blockquote>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#important-terms-to-know" id="markdown-toc-important-terms-to-know">Important terms to know</a></li>
  <li><a href="#vgp-assembly-pipeline-overview" id="markdown-toc-vgp-assembly-pipeline-overview">VGP assembly pipeline overview</a></li>
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#genome-profile-analysis" id="markdown-toc-genome-profile-analysis">Genome profile analysis</a>    <ol>
      <li><a href="#generation-of-k-mer-spectra-with-meryl" id="markdown-toc-generation-of-k-mer-spectra-with-meryl">Generation of <em>k</em>-mer spectra with <strong>Meryl</strong></a></li>
      <li><a href="#genome-profiling-with-genomescope2" id="markdown-toc-genome-profiling-with-genomescope2">Genome profiling with <strong>GenomeScope2</strong></a></li>
    </ol>
  </li>
  <li><a href="#assembly-with-hifiasm" id="markdown-toc-assembly-with-hifiasm">Assembly with <strong>hifiasm</strong></a>    <ol>
      <li><a href="#assembly-evaluation" id="markdown-toc-assembly-evaluation">Assembly evaluation</a></li>
      <li><a href="#hic-phased-assembly-with-hifiasm" id="markdown-toc-hic-phased-assembly-with-hifiasm">HiC-phased assembly with <strong>hifiasm</strong></a></li>
      <li><a href="#pseudohaplotype-assembly-with-hifiasm" id="markdown-toc-pseudohaplotype-assembly-with-hifiasm">Pseudohaplotype assembly with <strong>hifiasm</strong></a></li>
      <li><a href="#purging-the-primary-and-alternate-assemblies" id="markdown-toc-purging-the-primary-and-alternate-assemblies">Purging the primary and alternate assemblies</a></li>
      <li><a href="#purging-with-purgedups" id="markdown-toc-purging-with-purgedups">Purging with <strong>purge_dups</strong></a></li>
    </ol>
  </li>
  <li><a href="#scaffolding" id="markdown-toc-scaffolding">Scaffolding</a></li>
  <li><a href="#hybrid-scaffolding-with-bionano-optical-maps" id="markdown-toc-hybrid-scaffolding-with-bionano-optical-maps">Hybrid scaffolding with Bionano optical maps</a>    <ol>
      <li><a href="#evaluating-bionano-scaffolds" id="markdown-toc-evaluating-bionano-scaffolds">Evaluating Bionano scaffolds</a></li>
    </ol>
  </li>
  <li><a href="#hi-c-scaffolding" id="markdown-toc-hi-c-scaffolding">Hi-C scaffolding</a>    <ol>
      <li><a href="#pre-processing-hi-c-data" id="markdown-toc-pre-processing-hi-c-data">Pre-processing Hi-C data</a></li>
      <li><a href="#generate-initial-hi-c-contact-map" id="markdown-toc-generate-initial-hi-c-contact-map">Generate initial Hi-C contact map</a></li>
      <li><a href="#salsa2-scaffolding" id="markdown-toc-salsa2-scaffolding">SALSA2 scaffolding</a></li>
      <li><a href="#evaluate-the-final-genome-assembly-with-pretext" id="markdown-toc-evaluate-the-final-genome-assembly-with-pretext">Evaluate the final genome assembly with Pretext</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="important-terms-to-know">Important terms to know</h1>

<p>Before getting into the thick of things, let’s go over some terms you will often hear when learning about genome assembly. These concepts will be used often throughout this tutorial as well, so please refer to this section as necessary to help your understanding.</p>

<p><strong>Pseudohaplotype assembly</strong>: A genome assembly that consists of long phased haplotype blocks separated by regions where the haplotype cannot be distinguished (often homozygous regions). This can result in “switch errors”, when the parental haplotypes alternate along the same sequence. These types of assemblies are usually represented by a <em>primary assembly</em> and an <em>alternate assembly</em>. (This definition largely taken from the <a href="https://www.ncbi.nlm.nih.gov/assembly/model/#asmb_def">NCBI’s Genome Assembly Model</a>.)</p>

<p><strong>Primary assembly</strong>: The primary assembly is traditionally the more complete representation of an individual’s genome and consists of homozygous regions and one set of loci for heterozygous regions. Because the primary assembly contains both homo- and heterozygous regions, it is more complete than the <em>alternate assambly</em> which often reports only the other set of loci for heterozygous regions. Thus, the primary assembly is usually what one would use for downstream analyses.</p>

<p><strong>Alternate assembly</strong>: The alternate assembly consists of the alternate loci not represented in the <em>primary assembly</em> (heterozygous loci from the other haplotype). These types of sequences are often referred to as haplotigs. Traditionally, the alternate assembly is less complete compared to the primary assembly since homozygous regions are not represented.</p>

<p><strong>Phasing</strong>: Phasing aims to partition the contigs for an individual according to the haplotype they are derived from. When possible, this is done by identifying parental alleles using read data from the parents. Locally, this is achieved using linkage information in long read datasets. Recent approaches have managed to phase using long-range Hi-C linkage information from the same individual (<span class="citation"><a href="#Cheng2021">Cheng <i>et al.</i> 2021</a></span>).</p>

<p><strong>Assembly graph</strong>: A representation of the genome inferred from sequencing reads. Sequencing captures the genome as many fragmented pieces, instead of whole entire chromosomes at once (we eagerly await the day when this statement will be outdated!). The start of the assembly process pieces together these genome fragments to generate an assembly graph, which is a representation of the sequences and their overlaps. Visualizing assembly graphs can show where homozygous regions branch off into alternate paths on different haplotypes.</p>

<p><strong>Unitig</strong>: Usually the smallest unit of an assembly graph, consistent with all the available sequencing data. A unitig is often constructed from an unambiguous path in the assembly graph where all the vertices have exactly one incoming and one outgoing edge, except the first vertex can have any number of incoming edges, while the last vertex can have any number of outgoing edges (<span class="citation"><a href="#Rahman2022">Rahman and Medvedev 2022</a></span>). In other words, the internal vertices in the unitig path can only be walked one way, so unitigs represent a path of confident sequence. In the assembly graph, unitig nodes can then have overlap edges with other unitigs.</p>

<p><strong>Contig</strong>: A contiguous (<em>i.e.</em>, gapless) sequence in an assembly, usually inferred algorithmically from the unitig graph.</p>

<p><strong>False duplications</strong>: Assembly errors that result in one region of the genome being represented twice in the same assembly as two separate regions. Not to be confused with optical or technical duplicates from PCR from short read sequencing. False duplications can further be classified as either <em>haplotypic duplications</em> or <em>overlaps</em>.</p>

<p><strong>Haplotypic duplication</strong> can happen when a region that is heterozygous in the individual has the two haplotypes showing enough divergence that the assembler fails to interpret them as homologous. For example, say an individual is heterozygous in the region Chr1[1:100] and has Haplotype A from their mother and Haplotype B from their father; a false duplication can arise when the haplotypes are not recognized as being from the same region, and the assembler ends up placing both haplotypes in the same assembly, resulting in Chr1[1:100] being represented twice in one assembly. Ideally, a properly phased assembly would have Haplotype A in one assembly, <em>e.g.</em>, the primary, while Haplotype B is in the alternate.</p>

<p>False duplications via <strong>overlaps</strong> result from unresolved overlaps in the assembly graph. In this case, two different contigs end up representing the same sequence, effectively duplicating it. Overlaps happen when a branching point in an assembly graph is resolved such that the contig before the vertex and a contig after the vertex share the same overlapping sequence.</p>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/vgp_assembly/falseduplications.png" alt="Types of false duplication." width="750" height="369" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/falseduplications.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Schematic of types of false duplication. Image adapted from <span class="citation"><a href="#Rhie2021">Rhie <i>et al.</i> 2020</a></span>.</figcaption></figure>

<p><strong>Purging</strong>: Purging aims to remove false duplications, collapsed repeats, and very low support/coverage regions from an assembly. When performed on a primary assembly, the haplotigs are retained and typically placed in the alternate assembly.</p>

<p><strong>Scaffold</strong>: A scaffold refers to one or more contigs separated by gap (unknown) sequence. Contigs are usually generated with the aid of additional information, such as Bionano optical maps, linked reads, Hi-C chromatin information, etc. The regions between contigs are usually of unknown sequence, thus they are represented by sequences of <em>N</em>’s. Gaps length in the sequence can be sized or arbitrary, depending on the technology used for scaffolding (<em>e.g.</em>, optical maps can introduce sized gaps).</p>

<p>For more about the specific scaffolding technologies used in the VGP pipeline (currently Bionano optical maps and Hi-C chromatin conformation data), please refer to those specific sections within this tutorial.</p>

<p><strong>HiFi reads</strong>: PacBio <abbr title="high fidelity">HiFi</abbr> reads are the focus of this tutorial. First described in 2019, they have revolutionized genome assembly by combining long (about 10-20 kbp) read lengths with high accuracy (&gt;Q20) typically associated with short read sequencing (<span class="citation"><a href="#Wenger2019">Wenger <i>et al.</i> 2019</a></span>). These higher read lengths enable HiFi reads to traverse some repeat regions that are problematic to assemble with short reads.</p>

<p><strong>Ultra-long reads</strong>: Ultra-long reads are typically defined as reads of over 100 kbp, and are usually generated using Oxford Nanopore Technology. Read quality is often lower than HiFi or Illumina (<em>i.e.</em>, have a higher error rate), but they are often significantly longer than any other current sequencing technology, and can help assembly algorithms walk complex repeat regions in the assembly graphs.</p>

<p><strong>Manual curation</strong>: This term refers to manually evaluating and manipulating an assembly based on the raw supporting evidence (<em>e.g.</em>, using Hi-C contact map information). The user takes into account the original sequencing data to resolve potential <em>misassemblies</em> and <em>missed joins</em>.</p>

<p><strong>Misassembly</strong>: Misassemblies are a type of assembly error that usually refers to any structural error in the genome reconstruction, <em>.e.g.</em>, sequences that are not adjacent in the genome being placed next to each other in the sequence. Misassemblies can be potentially identified and remedied by manual curation.</p>

<p><strong>Missed join</strong>: A missed join happens when two sequences are adjacent to each other in the genome but are not represented contiguously in the final sequence. Missed joins can be identified and remedied in manual curation with Hi-C data.</p>

<p><strong>Telomere-to-telomere assembly</strong>: Often abbreviated as “T2T”, this term refers to an assembly where each chromosome is completely gapless from telomere to telomere. The term usually refers to the recently completed CHM13 human genome (<span class="citation"><a href="#Nurk2022">Nurk <i>et al.</i> 2022</a></span>), though there is an increasing number of efforts to generate T2T genomes for other species.</p>

<h1 id="vgp-assembly-pipeline-overview">VGP assembly pipeline overview</h1>

<p>This training is an adaptation of the VGP assembly pipeline 2.0 (Fig. 2).</p>

<figure id="figure-2" style="max-width: 90%;"><img src="../../images/vgp_assembly/vgp_pipeline_2.0_current.png" alt="VGP pipeline schematic showing steps from genome profiling, to hifiasm contigging, to bionano scaffolding, to HiC scaffolding." width="1121" height="445" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/vgp_pipeline_2.0_current.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> VGP Pipeline 2.0. The pipeline starts with reference-free genome profiling using k-mers. Then HiFi reads are assembled into contigs using hifiasm, along with additional phasing data if available. If false duplicates are detected in QC, then the contigs undergo purging. Afterwards, scaffolding takes place with optical maps (if available) and Hi-C data. A final decontamination steps is performed before curation.</figcaption></figure>

<p>This training has been organized into four main sections: genome profile analysis, assembly of <abbr title="high fidelity">HiFi</abbr> reads with hifiasm, scaffolding with Bionano optical maps, and scaffolding with <abbr title="all-versus-all chromatin conformation capture">Hi-C</abbr> data. Additionally, the <strong>assembly with hifiasm</strong> section has two possible paths in this tutorial: solo contigging or solo w/HiC contigging.</p>

<p>Throughout this tutorial, there will be <strong>detail boxes</strong> with additional background information on the science behind the sequencing technologies and software we use in the pipeline. These boxes are minimized by default, but please expand them to learn more about the data we utilize in this pipeline.</p>

<h1 id="get-data">Get data</h1>

<p>In order to reduce computation time, we will assemble samples from the yeast <em>Saccharomyces cerevisiae</em> S288C, a widely used laboratory strain isolated in the 1950s by Robert Mortimer. Using <em>S. cerevisae</em>, one of the most intensively studied eukaryotic model organisms, has the additional advantage of allowing us to evaluate the final result of our assembly with great precision. For this tutorial, we generated a set of synthetic HiFi reads corresponding to a theoretical diploid genome.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-generation-of-synthetic-reads"><button class="gtn-boxify-button details" type="button" aria-controls="details-generation-of-synthetic-reads" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Generation of synthetic reads</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>The synthetic HiFi reads were generated by using the <a href="https://www.ncbi.nlm.nih.gov/genome/15?genome_assembly_id=22535">S288C assembly</a> as reference genome. With this objective we used <a href="https://github.com/thegenemyers/HI.SIM">HIsim</a>, a HiFi shotgun simulator. The commands used are detailed below:</p>

  <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code>./FastK <span class="nt">-T8</span> <span class="nt">-Nsynthetic</span> <span class="nt">-k40</span> <span class="nt">-t4</span> ./hifi_reads.fastq.gz
./FastK <span class="nt">-T8</span> <span class="nt">-Nsynthetic</span> <span class="nt">-k40</span> <span class="nt">-p</span>:synthetic.ktab .hifi_reads.fastq.gz
./Histex <span class="nt">-G</span> synthetic <span class="o">&gt;</span> synthetic.histogram
./GeneScopeFK.R <span class="nt">-i</span> synthetic.histogram <span class="nt">-o</span> ./synthetic <span class="nt">-k</span> 40 <span class="nt">-p</span> 1
./HImodel <span class="nt">-v</span> <span class="nt">-osynthetic</span> <span class="nt">-g10</span>:55 <span class="nt">-e5</span> <span class="nt">-T8</span> ./synthetic
./HIsim ./GCF_000146045.2_R64_genomic.fasta ./synthetic.model <span class="nt">-p</span>.2,.2 <span class="nt">-c30</span> <span class="nt">-r1</span> <span class="nt">-U</span> <span class="nt">-f</span> <span class="nt">-h</span> <span class="nt">-e</span>
</code></pre></div>  </div>
  <p>The selected mutation rate was 2%. Note that HIsim generates the synthetic reads in FASTA format. This is perfectly fine for illustrating the workflow, but you should be aware that usually you will work with HiFi reads in FASTQ format.</p>

</blockquote>

<p>The first step is to get the datasets from Zenodo. The VGP assembly pipeline uses data generated by a variety of technologies, including PacBio HiFi reads, Bionano optical maps, and Hi-C chromatin interaction maps.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from <a href="https://zenodo.org/record/5887339">Zenodo</a></p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Rule-based</strong> tab</li>
        <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Datasets</code></li>
        <li>
          <p>Copy the tabular data, paste it into the textbox and press <kbd>Build</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Hi-C_dataset_F   https://zenodo.org/record/5550653/files/SRR7126301_1.fastq.gz?download=1   fastqsanger.gz    Hi-C
Hi-C_dataset_R   https://zenodo.org/record/5550653/files/SRR7126301_2.fastq.gz?download=1   fastqsanger.gz    Hi-C
Bionano_dataset    https://zenodo.org/record/5550653/files/bionano.cmap?download=1   cmap    Bionano
</code></pre></div>          </div>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Name</code>: column <code class="language-plaintext highlighter-rouge">A</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>: column <code class="language-plaintext highlighter-rouge">B</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Type</code>: column <code class="language-plaintext highlighter-rouge">C</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Name Tag</code>: column <code class="language-plaintext highlighter-rouge">D</code></li>
          </ul>
        </li>
        <li>Click <code class="language-plaintext highlighter-rouge">Apply</code> and press <kbd>Upload</kbd></li>
      </ul>
    </li>
    <li>
      <p>Import the remaining datasets from <a href="https://zenodo.org/record/5887339">Zenodo</a></p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Rule-based</strong> tab</li>
        <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Collections</code></li>
        <li>
          <p>Copy the tabular data, paste it into the textbox and press <kbd>Build</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>dataset_01    https://zenodo.org/record/6098306/files/HiFi_synthetic_50x_01.fasta?download=1  fasta    HiFi  HiFi_collection
dataset_02    https://zenodo.org/record/6098306/files/HiFi_synthetic_50x_02.fasta?download=1  fasta    HiFi  HiFi_collection
dataset_03    https://zenodo.org/record/6098306/files/HiFi_synthetic_50x_03.fasta?download=1  fasta    HiFi  HiFi_collection
</code></pre></div>          </div>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>: column <code class="language-plaintext highlighter-rouge">A</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>: column <code class="language-plaintext highlighter-rouge">B</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Type</code>: column <code class="language-plaintext highlighter-rouge">C</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Group Tag</code>: column <code class="language-plaintext highlighter-rouge">D</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Collection Name</code>: column <code class="language-plaintext highlighter-rouge">E</code></li>
          </ul>
        </li>
        <li>Click <code class="language-plaintext highlighter-rouge">Apply</code> and press <kbd>Upload</kbd></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="hifi-reads-preprocessing-with-cutadapt">HiFi reads preprocessing with <strong>cutadapt</strong></h3>

<p>Adapter trimming usually means trimming the adapter sequence off the ends of reads, which is where the adapter sequence is usually located in <abbr title="next generation sequencing">NGS</abbr> reads. However, due to the nature of <abbr title="single-molecule real-time">SMRT</abbr> sequencing technology, adapters do not have a specific, predictable location in  <abbr title="high fidelity">HiFi</abbr> reads. Additionally, the reads containing adapter sequence could be of generally lower quality compared to the rest of the reads. Thus, we will use <strong>cutadapt</strong> not to trim, but to remove the entire read if a read is found to have an adapter inside of it.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-background-on-pacbio-hifi-reads"><button class="gtn-boxify-button details" type="button" aria-controls="details-background-on-pacbio-hifi-reads" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Background on PacBio HiFi reads</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>PacBio HiFi reads rely on <abbr title="single-molecule real-time">SMRT</abbr> sequencing technology. SMRT is based on real-time imaging of fluorescently tagged nucleotides as they are added to a newly synthesized DNA strand. HiFi further uses multiple subreads from the same circular template to produce one highly accurate consensus sequence (fig. 2).</p>

  <figure id="figure-3" style="max-width: 90%;"><img src="../../images/vgp_assembly/pacbio_technology.png" alt="PacBio HiFi technology. " width="1278" height="768" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/pacbio_technology.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> HiFi reads are produced by calling consensus from subreads generated by multiple passes of the enzyme around a circularized template. This results in a HiFi read that is both long and accurate.</figcaption></figure>

  <p>This technology allows to generate long-read sequencing data with read lengths in the range of 10-25 kb and minimum read consensus accuracy greater than 99% (Q20).</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-primer-removal-with-cutadapt"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Primer removal with Cutadapt</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/lparsons/cutadapt/cutadapt/3.4" title="Cutadapt tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cutadapt</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.4)</span> with the following parameters:
      <ul>
        <li><em>“Single-end or Paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/A file”</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection</code></li>
            <li>In <em>“Read 1 Options”</em>:
              <ul>
                <li>In <em>“5’ or 3’ (Anywhere) Adapters”</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert 5’ or 3’ (Anywhere) Adapters”</em>
                      <ul>
                        <li><em>“Source”</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><em>“Enter custom 5’ or 3’ adapter name”</em>: <code class="language-plaintext highlighter-rouge">First adapter</code></li>
                            <li><em>“Enter custom 5’ or 3’ adapter sequence”</em>: <code class="language-plaintext highlighter-rouge">ATCTCTCTCAACAACAACAACGGAGGAGGAGGAAAAGAGAGAGAT</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert 5’ or 3’ (Anywhere) Adapters”</em>
                      <ul>
                        <li><em>“Source”</em>: <code class="language-plaintext highlighter-rouge">Enter custom sequence</code>
                          <ul>
                            <li><em>“Enter custom 5’ or 3’ adapter name”</em>: <code class="language-plaintext highlighter-rouge">Second adapter</code></li>
                            <li><em>“Enter custom 5’ or 3’ adapter sequence”</em>: <code class="language-plaintext highlighter-rouge">ATCTCTCTCTTTTCCTCCTCCTCCGTTGTTGTTGTTGAGAGAGAT</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>“Adapter Options”</em>:
          <ul>
            <li><em>“Maximum error rate”</em>: <code class="language-plaintext highlighter-rouge">0.1</code></li>
            <li><em>“Minimum overlap length”</em>: <code class="language-plaintext highlighter-rouge">35</code></li>
            <li><em>“Look for adapters in the reverse complement”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li>In <em>“Filter Options”</em>:
          <ul>
            <li><em>“Discard Trimmed Reads”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-select-collection-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-collection-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Select collection dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ol>
          <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>
          <li>Select the collection you want to use from the list</li>
        </ol>

      </blockquote>
    </li>
    <li>Rename the output file as <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code>.</li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
  <p><!--END_SNIPPET--></p>

</blockquote>

<h1 id="genome-profile-analysis">Genome profile analysis</h1>

<p>Before starting a <em>de novo</em> genome assembly project, it is useful to collect metrics on the properties of the genome under consideration, such as the expected genome size, so that you know what to expect from your assembly. Traditionally, DNA flow cytometry was considered the golden standard for estimating the genome size. Nowadays, experimental methods have been replaced by computational approaches (<span class="citation"><a href="#wang2020estimation">Wang <i>et al.</i> 2020</a></span>). One of the widely used genome profiling methods is based on the analysis of <em>k</em>-mer frequencies. It allows one to provide information not only about the genomic complexity, such as the genome size and levels of heterozygosity and repeat content, but also about the data quality.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-i-k-i-mer-size-sequencing-coverage-and-genome-size"><button class="gtn-boxify-button details" type="button" aria-controls="details-i-k-i-mer-size-sequencing-coverage-and-genome-size" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: <i>K</i>-mer size, sequencing coverage and genome size</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><em>K</em>-mers are unique substrings of length <em>k</em> contained within a DNA sequence. For example, the DNA sequence <em>TCGATCACA</em> can be decomposed into five unique <em>k</em>-mers that have five bases long: <em>TCGAT</em>, <em>CGATC</em>, <em>GATCA</em>, <em>ATCAC</em> and <em>TCACA</em>. A sequence of length L will have  L-k+1 <em>k</em>-mers. On the other hand, the number of possible <em>k</em>-mers can be calculated as  n<sup>k</sup>, where n is the number of possible monomers and k is the k-mer size.</p>

  <table>
    <thead>
      <tr>
        <th>Bases</th>
        <th>K-mer size</th>
        <th>Total possible k-mers</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>4</td>
        <td>1</td>
        <td>4</td>
      </tr>
      <tr>
        <td>4</td>
        <td>2</td>
        <td>16</td>
      </tr>
      <tr>
        <td>4</td>
        <td>3</td>
        <td>64</td>
      </tr>
      <tr>
        <td>4</td>
        <td>…</td>
        <td>…</td>
      </tr>
      <tr>
        <td>4</td>
        <td>10</td>
        <td>1.048.576</td>
      </tr>
    </tbody>
  </table>

  <p>Thus, the k-mer size is a key parameter, which must be large enough to map uniquely to the genome, but not too large, since it can lead to wasting computational resources. In the case of the human genome, <em>k</em>-mers of 31 bases in length lead to 96.96% of unique <em>k</em>-mers.</p>

  <p>Each unique k-mer can be assigned a value for coverage based on the number of times it occurs in a sequence, whose distribution will approximate a Poisson distribution, with the peak corresponding to the average genome sequencing depth. From the genome coverage, the genome size can be easily computed.</p>
</blockquote>

<h2 id="generation-of-k-mer-spectra-with-meryl">Generation of <em>k</em>-mer spectra with <strong>Meryl</strong></h2>

<p>Meryl will allow us to generate the <em>k</em>-mer profile by decomposing the sequencing data into <em>k</em>-length substrings, counting the occurrence of each <em>k</em>-mer and determining its frequency. The original version of Meryl was developed for the Celera Assembler. The current Meryl version comprises three main modules: one for generating <em>k</em>-mer databases, one for filtering and combining databases, and one for searching databases. <em>K</em>-mers are stored in lexicographical order in the database, similar to words in a dictionary (<span class="citation"><a href="#Rhie2020">Rhie <i>et al.</i> 2020</a></span>).</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-i-k-i-mer-size-estimation"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: <i>k</i>-mer size estimation</div>

  <p>Given an estimated genome size (G) and a tolerable collision rate (p), an appropriate k can be computed as k = log4 (G(1 − p)/p).</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-i-k-i-mers-count-distribution"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate <i>k</i>-mers count distribution</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6" title="Meryl tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Meryl</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3+galaxy6)</span> with the following parameters:
      <ul>
        <li><em>“Operation type selector”</em>: <code class="language-plaintext highlighter-rouge">Count operations</code>
          <ul>
            <li><em>“Count operations”</em>: <code class="language-plaintext highlighter-rouge">Count: count the occurrences of canonical k-mers</code></li>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input sequences”</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code></li>
            <li><em>“k-mer size selector”</em>: <code class="language-plaintext highlighter-rouge">Set a k-mer size</code>
              <ul>
                <li>“<em>k-mer size</em>”: <code class="language-plaintext highlighter-rouge">31</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-selection-of-i-k-i-mer-size"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Selection of <i>k</i>-mer size</div>

        <p>We used 31 as <em>k</em>-mer size, as this length has demonstrated to be sufficiently long that most <em>k</em>-mers are not repetitive and is short enough to be more robust to sequencing errors. For very large (haploid size &gt; 10 Gb) and/or very repetitive genomes, larger <em>k</em>-mer length is recommended to increase the number of unique <em>k</em>-mers.</p>
      </blockquote>
    </li>
    <li>
      <p>Rename it <code class="language-plaintext highlighter-rouge">Collection meryldb</code></p>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6" title="Meryl tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Meryl</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3+galaxy6)</span> again with the following parameters:
      <ul>
        <li><em>“Operation type selector”</em>: <code class="language-plaintext highlighter-rouge">Operations on sets of k-mers</code>
          <ul>
            <li><em>“Operations on sets of k-mers”</em>: <code class="language-plaintext highlighter-rouge">Union-sum: return k-mers that occur in any input, set the count to the sum of the counts</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input meryldb”</em>: <code class="language-plaintext highlighter-rouge">Collection meryldb</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename it as <code class="language-plaintext highlighter-rouge">Merged meryldb</code></p>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/meryl/meryl/1.3+galaxy6" title="Meryl tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Meryl</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3+galaxy6)</span> for the third time with the following parameters:
      <ul>
        <li><em>“Operation type selector”</em>: <code class="language-plaintext highlighter-rouge">Generate histogram dataset</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input meryldb”</em>: <code class="language-plaintext highlighter-rouge">Merged meryldb</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Finally, rename it as <code class="language-plaintext highlighter-rouge">Meryldb histogram</code>.</li>
  </ol>

</blockquote>

<h2 id="genome-profiling-with-genomescope2">Genome profiling with <strong>GenomeScope2</strong></h2>

<p>The next step is to infer the genome properties from the <em>k</em>-mer histogram generated by Meryl, for which we will use GenomeScope2. GenomeScope2 relies on a nonlinear least-squares optimization to fit a mixture of negative binomial distributions, generating estimated values for genome size, repetitiveness, and heterozygosity rates (<span class="citation"><a href="#RanalloBenavidez2020">Ranallo-Benavidez <i>et al.</i> 2020</a></span>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-estimate-genome-properties"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Estimate genome properties</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/genomescope/genomescope/2.0+galaxy2" title="GenomeScope tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GenomeScope</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0+galaxy2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input histogram file”</em>: <code class="language-plaintext highlighter-rouge">Meryldb histogram</code></li>
        <li><em>Ploidy for model to use</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
        <li><em>“k-mer length used to calculate k-mer spectra”</em>: <code class="language-plaintext highlighter-rouge">31</code></li>
      </ul>
    </li>
  </ol>

  <ul>
    <li>In “<em>Output options</em>”: mark <code class="language-plaintext highlighter-rouge">Summary of the analysis</code></li>
    <li>In “<em>Advanced options</em>”:
      <ul>
        <li><em>“Create testing.tsv file with model parameters”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>Genomescope will generate six outputs:</p>

<ul>
  <li>Plots
    <ul>
      <li>Linear plot: <em>k</em>-mer spectra and fitted models: frequency (y-axis) versus coverage.</li>
      <li>Log plot: logarithmic transformation of the previous plot.</li>
      <li>Transformed linear plot: <em>k</em>-mer spectra and fitted models: frequency times coverage (y-axis) versus coverage (x-axis). This transformation increases the heights of higher-order peaks, overcoming the effect of high heterozygosity.</li>
      <li>Transformed log plot: logarithmic transformation of the previous plot.</li>
    </ul>
  </li>
  <li>Model: this file includes a detailed report about the model fitting.</li>
  <li>Summary: it includes the properties inferred from the model, such as genome haploid length and the percentage of heterozygosity.</li>
</ul>

<p>Now, let’s analyze the <em>k</em>-mer profiles, fitted models and estimated parameters (fig. 4).</p>

<figure id="figure-4" style="max-width: 90%;"><img src="../../images/vgp_assembly/genomescope_plot.png" alt="Genomescope plot. " width="2000" height="2000" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/genomescope_plot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> GenomeScope2 31-mer profile. The first peak located at coverage 25x corresponds to the heterozygous peak. The second peak at coverage 50x, corresponds to the homozygous peak. Estimate of the heterozygous portion is 0.576%. The plot also includes information about the inferred total genome length (len), genome unique length percent ('uniq'), overall heterozygosity rate ('ab'), mean k-mer coverage for heterozygous bases ('kcov'), read error rate ('err'), and average rate of read duplications ('dup'). It also reports the user-given parameters of k-mer size ('k') and ploidy ('p').</figcaption></figure>

<p>This distribution is the result of the Poisson process underlying the generation of sequencing reads. As we can see, the <em>k</em>-mer profile follows a bimodal distribution, indicative of a diploid genome. The distribution is consistent with the theoretical diploid model (model fit &gt; 93%). Low frequency <em>k</em>-mers are the result of sequencing errors. GenomeScope2 estimated a haploid genome size is around 11.7 Mb, a value reasonably close to <em>Saccharomyces</em> genome size. Additionally, it revealed that the variation across the genomic sequences is 0.576%.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-are-you-expecting-to-purge-your-assembly"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Are you expecting to purge your assembly?</div>
  <p>This tutorial covers purging using the program <strong>purge_dups</strong>. purge_dups has some default options and can try to detect coverage-based cutoffs automatically, but the VGP pipeline prefers to define these cutoffs using parameters derived from the GenomeScope2 output.</p>

  <p><em>If you expect you need to purge your genome, please see the <strong>solo</strong> contigging section of the tutorial for details on parsing the GenomeScope2 output for purging cutoffs.</em></p>
</blockquote>

<h1 id="assembly-with-hifiasm">Assembly with <strong>hifiasm</strong></h1>

<p>Once we have finished the genome profiling stage, we can start the genome assembly with hifiasm,  a fast open-source <em>de novo</em> assembler specifically developed for PacBio HiFi reads. One of the key advantages of hifiasm is that it allows us to resolve near-identical, but not exactly identical, sequences, such as repeats and segmental duplications (<span class="citation"><a href="#Cheng2021">Cheng <i>et al.</i> 2021</a></span>).</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-hifiasm-algorithm-details"><button class="gtn-boxify-button details" type="button" aria-controls="details-hifiasm-algorithm-details" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Hifiasm algorithm details</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>By default hifiasm performs three rounds of haplotype-aware error correction to correct sequence errors but keeping heterozygous alleles. A position on the target read to be corrected is considered informative if there are two different nucleotides at that position in the alignment, and each allele is supported by at least three reads.</p>

  <figure id="figure-5" style="max-width: 90%;"><img src="../../images/vgp_assembly/hifiasm_algorithm.png" alt="Hifiasm algorithm overview. " width="999" height="753" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hifiasm_algorithm.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Hifiasm algorithm overview. Orange and blue bars represent the reads with heterozygous alleles carrying local phasing information, while green bars come from the homozygous regions without any heterozygous alleles.</figcaption></figure>

  <p>Then, hifiasm builds a phased assembly string graph with local phasing information from the corrected reads. Only the reads coming from the same haplotype are connected in the phased assembly graph. After transitive reduction, a pair of heterozygous alleles is represented by a <em>bubble</em> in the string graph. If there is no additional data, hifiasm arbitrarily selects one side of each bubble and outputs a primary assembly. In the case of a heterozygous genome, the primary assembly generated at this step may still retain haplotigs from the alternate allele.</p>

</blockquote>

<p>The output of hifiasm will be Graphical Fragment Assembly (GFA) files. These differ from FASTA files in that they are a representation of the assembly graph instead of just linear sequences, so the GFA contains information about sequences, nodes, and edges (<em>i.e.</em>, overlaps). This output preserves the most information about how the reads assemble in graph space, and is useful to visualize in tools such as Bandage; however, our QV tools will expect FASTA files, so we will cover the GFA to FASTA conversion step later.</p>

<p>Hifiasm can be run in multiple modes depending on data availability:</p>

<p><strong>Solo</strong>: generates a pseudohaplotype assembly, resulting in a primary &amp; an alternate assembly (fig. 5).</p>
<ul>
  <li><em>Input: only HiFi reads</em></li>
  <li><em>Output: scaffolded primary assembly, and alternate contigs</em></li>
</ul>
<figure id="figure-6" style="max-width: 90%;"><img src="../../images/vgp_assembly/hifiasm_solo_schematic.png" alt="Diagram for hifiasm solo mode." width="1855" height="504" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hifiasm_solo_schematic.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> The <b>solo</b> pipeline creates primary and alternate contigs, which then typically undergo purging with purge_dups to reconcile the haplotypes. During the purging process, haplotigs are removed from the primary assembly and added to the alternate assembly, which is then purged to generate the final alternate set of contigs. The purged primary contigs are then carried through scaffolding with Bionano and/or Hi-C data, resulting in one final draft primary assembly to be sent to manual curation.</figcaption></figure>

<p><strong>Hi-C-phased</strong>: generates a hap1 assembly and a hap2 assembly, which are phased using the <abbr title="all-versus-all chromatin conformation capture">Hi-C</abbr> reads from the same individual (fig. 6).</p>
<ul>
  <li><em>Input: HiFi &amp; HiC reads</em></li>
  <li><em>Output: scaffolded hap1 assembly, and scaffolded hap2 assembly (assuming you run the scaffolding on <strong>both</strong> haplotypes)</em></li>
</ul>
<figure id="figure-7" style="max-width: 90%;"><img src="../../images/vgp_assembly/hifiasm_hic_schematic.png" alt="Diagram for hifiasm hic mode." width="1543" height="492" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hifiasm_hic_schematic.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> The <b>Hi-C-phased</b> mode produces hap1 and hap2 contigs, which have been phased using the HiC information as described in <span class="citation"><a href="#Cheng2021">Cheng <i>et al.</i> 2021</a></span>. Typically, these assemblies do not need to undergo purging, but you should always look at your assemblies' QC to make sure. These contigs are then scaffolded <i>separately</i> using Bionano and/or Hi-C workflows, resulting in two scaffolded assemblies.</figcaption></figure>

<p><strong>Trio</strong>: generates a maternal assembly and a paternal assembly, which are phased using reads from the parents (fig. 7).</p>
<ul>
  <li><em>Input: HiFi reads from child, Illumina reads from both parents.</em></li>
  <li><em>Output: scaffolded maternal assembly, and scaffolded paternal assembly (assuming you run the scaffolding on <strong>both</strong> haplotypes)</em></li>
</ul>
<figure id="figure-8" style="max-width: 90%;"><img src="../../images/vgp_assembly/hifiasm_trio_schematic.png" alt="Diagram for hifiasm trio mode." width="1519" height="495" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hifiasm_trio_schematic.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span> The <b>trio</b> mode produces maternal and paternal contigs, which have been phased using paternal short read data. Typically, these assemblies do not need to undergo purging, but you should always look at your assemblies' QC to make sure. These contigs are then scaffolded <i>separately</i> using Bionano and/or Hi-C workflows, resulting in two scaffolded assemblies.</figcaption></figure>

<p>No matter which way you run hifiasm, you will have to evaluate the assemblies’ <abbr title="Quality Control">QC</abbr> to ensure your genome is in good shape. The VGP pipeline features several reference-free ways of evaluating assembly quality, all of which are automatically generated with our workflows; however, we will run them manually in this tutorial so we can familiarize ourselves with how each QC metric captures a different aspect of assembly quality.</p>

<h2 id="assembly-evaluation">Assembly evaluation</h2>
<ul>
  <li><strong>gfastats</strong>: manipulation &amp; evaluation of assembly graphs and FASTA files, particularly used for summary statistics (<em>e.g.</em>, contig count, N50, NG50, etc.) (<span class="citation"><a href="#Formenti2022">Formenti <i>et al.</i> 2022</a></span>).</li>
</ul>
<figure id="figure-9" style="max-width: 90%;"><img src="../../images/vgp_assembly/n50schematic.jpg" alt="Schematic of N50 calculation." width="699" height="141" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/n50schematic.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> <b>N50</b> is a commonly reported statistic used to represent genome contiguity. N50 is calculated by sorting contigs according to their lengths, and then taking the halfway point of the total genome length. The size of the contig at that halfway point is the N50 value. In the pictured example, the total genome length is 400 bp, so the N50 value is 60 because the contig at the halfway point is 60 bp long. N50 can be interpreted as the value where &gt;50% of an assembly's contigs are at that value or higher. Image adapted from <a href="https://www.molecularecologist.com/2017/03/29/whats-n50/">Elin Videvall at The Molecular Ecologist</a>.</figcaption></figure>
<ul>
  <li><strong>Benchmarking Universal Single-Copy Orthologs (BUSCO)</strong>: assesses completeness of a genome from an evolutionarily informed functional point of view. BUSCO genes are genes that are expected to be present at single-copy in one haplotype for a certain clade, so their presence, absence, or duplication can inform scientists about if an assembly is likely missing important regions, or if it has multiple copies of them, which can indicate a need for purging (<span class="citation"><a href="#Simo2015">Simão <i>et al.</i> 2015</a></span>).</li>
  <li><strong>Merqury</strong>: reference-free assessment of assembly completeness and phasing based on <em>k</em>-mers. Merqury compares <em>k</em>-mers in the reads to the <em>k</em>-mers found in the assemblies, as well as the copy number (CN) of each <em>k</em>-mer in the assemblies (<span class="citation"><a href="#Rhie_merqury">Rhie <i>et al.</i> 2020</a></span>).</li>
</ul>

<blockquote class="notranslate hands_on gtn-cyoa" id="gtn-cyoa">
	<div class="box-title" aria-label="hands-on box: choose your own tutorial"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Choose Your Own Tutorial</div>
	<p>This is a "Choose Your Own Tutorial" section, where you can select between multiple paths. Click one of the buttons below to select how you want to follow the tutorial</p>
	
	<p>
		Use the following buttons to switch between contigging approaches. If you are assembling with only HiFi reads for an individual, then click <b><i>solo</i></b>. If you have HiC reads for the same indiviudal, then click <b><i>hic</i></b>. <b>NOTE: If you want to learn more about purging, then <u>please check out the <i>solo</i> tutorial for details on purging false duplications.</u></b>
	</p>
	
	
		<input id="cyoa-opt1" type="radio" name="cyoa" value="hic" onclick="cyoaChoice('hic')" />
		<label for="cyoa-opt1" class="select btn btn-secondary">hic</label>
	
	
		<input id="cyoa-opt2" type="radio" name="cyoa" value="solo" onclick="cyoaChoice('solo')" />
		<label for="cyoa-opt2" class="select btn btn-secondary">solo</label>
	
	
	

	
	<script>
		function ready(fn) {
			if (document.readyState != 'loading'){
				fn();
			} else {
				document.addEventListener('DOMContentLoaded', fn);
			}
		}
		ready(function() {
			cyoaDefault("hic");
		})
	</script>
	
</blockquote>

<div class="hic">

  <h2 id="hic-phased-assembly-with-hifiasm">HiC-phased assembly with <strong>hifiasm</strong></h2>

  <p>If you have the <abbr title="all-versus-all chromatin conformation capture">Hi-C</abbr> data for the individual you are assembling with <abbr title="high fidelity">HiFi</abbr> reads, then you can use that information to phase the {contigs}.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-hi-c-phased-assembly-with-b-hifiasm-b"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Hi-C-phased assembly with <b>hifiasm</b></div>
    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.18.8+galaxy1" title="Hifiasm tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Hifiasm</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.18.8+galaxy1)</span> with the following parameters:
        <ul>
          <li><em>“Assembly mode”</em>: <code class="language-plaintext highlighter-rouge">Standard</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input reads”</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              <li><em>“Hi-C R1 reads”</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
              <li><em>“Hi-C R2 reads”</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_R</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>After the tool has finished running, rename its outputs as follows:
        <ul>
          <li>Rename the <code class="language-plaintext highlighter-rouge">Hi-C hap1 balanced contig graph</code> as <code class="language-plaintext highlighter-rouge">Hap1 contigs graph</code> and add a <code class="language-plaintext highlighter-rouge">#hap1</code> tag</li>
          <li>Rename the <code class="language-plaintext highlighter-rouge">Hi-C hap2 balanced contig graph</code> as <code class="language-plaintext highlighter-rouge">Hap2 contigs graph</code> and  add a <code class="language-plaintext highlighter-rouge">#hap2</code> tag</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>We have obtained the fully phased contig graphs (as <abbr title="Graphical Fragment Assembly">GFA</abbr> files) of hap1 and hap2, but these must be converted to FASTA format for subsequent steps. We will use a tool developed from the VGP: <strong>gfastats</strong>. gfastats is a tool suite that allows for manipulation and evaluation of FASTA and GFA files, but in this instance we will use it to convert our GFAs to FASTA files. Later on we will use it to generate standard summary statistics for our assemblies.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-convert-gfa-to-fasta"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Convert GFA to FASTA</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0" title="gfastats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>gfastats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input GFA file”</em>: select <code class="language-plaintext highlighter-rouge">Hap1 contigs graph</code> and the <code class="language-plaintext highlighter-rouge">Hap2 contigs graph</code> datasets</li>
        </ul>

        <blockquote class="tip">
          <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
            <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
          </ol>
        </blockquote>

        <ul>
          <li><em>“Tool mode”</em>: <code class="language-plaintext highlighter-rouge">Genome assembly manipulation</code></li>
          <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">FASTA</code></li>
          <li><em>“Generates the initial set of paths</em>”: toggle to <code class="language-plaintext highlighter-rouge">yes</code></li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Hap1 contigs FASTA</code> and <code class="language-plaintext highlighter-rouge">Hap2 contigs FASTA</code></li>
    </ol>

  </blockquote>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-summary-statistics"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Summary statistics</div>

    <p>gfastats will provide us with the following statistics:</p>

    <ul>
      <li>No. of contigs: The total number of contigs in the assembly.</li>
      <li>Largest contig: The length of the largest contig in the assembly.</li>
      <li>Total length: The total number of bases in the assembly.</li>
      <li>Nx: The largest contig length, <em>L</em>, such that contigs of length &gt;= <em>L</em> account for at least <em>x</em>% of the bases of the assembly.</li>
      <li>NGx: The contig length such that using equal or longer length contigs produces <em>x</em>% of the length of the reference genome, rather than <em>x</em>% of the assembly length.</li>
      <li>GC content: the percentage of nitrogenous bases which are either guanine or cytosine.</li>
    </ul>

  </blockquote>

  <p>Let’s use gfastats to get a basic idea of what our assembly looks like. We’ll run gfastats on the <abbr title="Graphical Fragment Assembly">GFA</abbr> files because gfastats can report graph-specific statistics as well. After generating the stats, we’ll be doing some text manipulations to get the stats side-by-side in a pretty fashion.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-assembly-evaluation-with-gfastats"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assembly evaluation with gfastats</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0" title="gfastats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>gfastats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Hap1 contigs graph</code> and the <code class="language-plaintext highlighter-rouge">Hap2 contigs graph</code> datasets</li>
          <li><em>“Expected genome size”</em>: <code class="language-plaintext highlighter-rouge">11747160</code> (remember we calculated this value earlier, so it should be in your history!)</li>
          <li><em>“Generates the initial set of paths</em>”: toggle to <code class="language-plaintext highlighter-rouge">yes</code></li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Hap1 stats</code> and <code class="language-plaintext highlighter-rouge">Hap2 stats</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3" title="Column join tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column join</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.3)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Hap1 stats</code> and the <code class="language-plaintext highlighter-rouge">Hap2 stats</code> datasets</li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">gfastats on hap1 and hap2 (full)</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1" title="Search in textfiles tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search in textfiles</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">gfastats on hap1 and hap2 (full)</code></li>
          <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Don't Match</code></li>
          <li><em>“Type of regex”</em>: <code class="language-plaintext highlighter-rouge">Basic</code></li>
          <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">[Ss]caffold</code></li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">gfastats on hap1 and hap2 contigs</code></li>
    </ol>

  </blockquote>

  <p>Take a look at the <em>gfastats on hap1 and hap2 contigs</em> output — it should have three columns: 1) name of statistic, 2) hap1 value, and 3) hap2 value. According to the report, both assemblies are quite similar; the hap1 assembly includes 16 contigs, totalling ~11.3Mbp of sequence (the <code class="language-plaintext highlighter-rouge">Total contig length</code> statistic), while the hap2 assembly includes 17 contigs, whose total length is ~12.2Mbp. (<strong>NB</strong>: Your values may differ slightly, or be reversed between the two haplotypes!)</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>What is the length of the longest contigs in the assemblies?</li>
      <li>What are the N50 values of the two assemblies? Are they very different from each other?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>One assembly’s longest contig is 1,532,843 bp, and the other one’s is 1,531,728 bp.</li>
        <li>One assembly has a N50 of 922,430 and the other’s is 923,452. These are pretty close to each other!</li>
      </ol>

    </blockquote>

  </blockquote>

  <p>Next, we will use <abbr title="Benchmarking Universal Single-Copy Orthologs">BUSCO</abbr>, which will provide quantitative assessment of the completeness of a genome assembly in terms of expected gene content. It relies on the analysis of genes that should be present only once in a complete assembly or gene set, while allowing for rare gene duplications or losses (<span class="citation"><a href="#Simo2015">Simão <i>et al.</i> 2015</a></span>).</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-assessing-assembly-completeness-with-busco"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assessing assembly completeness with BUSCO</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0" title="Busco tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Busco</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Sequences to analyze”</em>: <code class="language-plaintext highlighter-rouge">Hap1 contigs FASTA</code> and <code class="language-plaintext highlighter-rouge">Hap2 contigs FASTA</code></li>
          <li><em>“Mode”</em>: <code class="language-plaintext highlighter-rouge">Genome assemblies (DNA)</code>
            <ul>
              <li><em>“Use Augustus instead of Metaeuk”</em>: <code class="language-plaintext highlighter-rouge">Use Metaeuk</code></li>
            </ul>
          </li>
          <li><em>“Auto-detect or select lineage?”</em>: <code class="language-plaintext highlighter-rouge">Select lineage</code>
            <ul>
              <li><em>“Lineage”</em>: <code class="language-plaintext highlighter-rouge">Saccharomycetes</code></li>
            </ul>
          </li>
          <li><em>“Which outputs should be generated”</em>: <code class="language-plaintext highlighter-rouge">short summary text</code> and <code class="language-plaintext highlighter-rouge">summary image</code></li>
        </ul>

        <blockquote class="comment">
          <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

          <p>Remember to modify the lineage option if you are working with vertebrate genomes.</p>
        </blockquote>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">BUSCO hap1</code> and <code class="language-plaintext highlighter-rouge">BUSCO hap2</code>.</li>
    </ol>

  </blockquote>

  <p>We have asked <abbr title="Benchmarking Universal Single-Copy Orthologs">BUSCO</abbr> to generate two particular outputs: the short summary, and a summary image.</p>
  <figure id="figure-10" style="max-width: 90%;"><img src="../../images/vgp_assembly/busco_hap1hap2.png" alt="BUSCO for hap1 &amp; hap2." width="2043" height="1124" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/busco_hap1hap2.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> BUSCO results for hap1 (top) and hap2 (bottom). Each haplotype is showing the summary image output as well as the short summary output. The summary image gives a good overall idea of the status of BUSCO genes within the assembly, while the short summary lists these as percentages as well. In our case, neither assembly seems to have duplicated BUSCO genes (there is a very low amount of dark blue in the summary images), though hap1 seems to be marginally less complete (there is more red in the summary imaged compared to hap2).</figcaption></figure>

  <blockquote class="question">
    <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How many complete and single copy BUSCO genes have been identified in the hap1 assembly? What percentage of the total BUSCO gene set is that?</li>
      <li>How many BUSCO genes are absent in the hap1 assembly?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>Hap1 has 1864 complete and single-copy BUSCO genes, which is 87.2% of the gene set.</li>
        <li>181 BUSCO genes are missing.</li>
      </ol>

    </blockquote>

  </blockquote>

  <p>Despite BUSCO being robust for species that have been widely studied, it can be inaccurate when the newly assembled genome belongs to a taxonomic group that is not well represented in <a href="https://www.orthodb.org/">OrthoDB</a>. Merqury provides a complementary approach for assessing genome assembly quality metrics in a reference-free manner via <em>k</em>-mer copy number analysis.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-i-k-i-mer-based-evaluation-with-merqury"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: <i>k</i>-mer based evaluation with Merqury</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3" title="Merqury tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merqury</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3)</span> with the following parameters:
        <ul>
          <li><em>“Evaluation mode”</em>: <code class="language-plaintext highlighter-rouge">Default mode</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“k-mer counts database”</em>: <code class="language-plaintext highlighter-rouge">Merged meryldb</code></li>
              <li><em>“Number of assemblies”</em>: `Two assemblies
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First genome assembly”</em>: <code class="language-plaintext highlighter-rouge">Hap1 contigs FASTA</code></li>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second genome assembly”</em>: <code class="language-plaintext highlighter-rouge">Hap2 contigs FASTA</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>By default, Merqury generates three collections as output: stats, plots and consensus accuracy quality value (QV) stats. The “stats” collection contains the completeness statistics, while the “QV stats” collection contains the quality value statistics. Let’s have a look at the assembly <abbr title="copy number">CN</abbr> spectrum plot, known as the <em>spectra-cn</em> plot (fig. 7).</p>

  <figure id="figure-11" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_cn_plot.png" alt="Merqury spectra-cn plot for the hap1/hap2 assemblies." width="1800" height="1500" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_cn_plot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 11</strong>:</span> Merqury CN plot. This plot tracks the multiplicity of each k-mer found in the Hi-Fi read set and colors it by the number of times it is found in a given assembly. Merqury connects the midpoint of each histogram bin with a line, giving the illusion of a smooth curve.</figcaption></figure>

  <p>The black region in the left side corresponds to k-mers found only in the read set; it is usually indicative of sequencing error in the read set, although it can also be indicative of missing sequences in the assembly. The red area represents one-copy k-mers in the genome, while the blue area represents two-copy k-mers originating from homozygous sequence or haplotype-specific duplications. From this figure we can state that the diploid sequencing coverage is around 50x, which we also know from the GenomeScope2 plot we looked at earlier.</p>

  <p>To get an idea of how the <em>k</em>-mers have been distributed between our hap1 and hap2 assemblies, we should look at the <em>spectra-asm</em> output of Merqury.</p>

  <figure id="figure-12" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_hap1hap2_asm.png" alt="Merqury spectra-asm plot for the hap1/hap2 assemblies." width="1268" height="988" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_hap1hap2_asm.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 12</strong>:</span> Merqury ASM plot. This plot tracks the multiplicity of each k-mer found in the Hi-Fi read set and colors it according to which assemblies contain those k-mers. This can tell you which k-mers are found in only one assembly or shared between them.</figcaption></figure>

  <p>The large green peak is centered at 50x coverage (remember that’s our diploid coverage!), indicating that <em>k</em>-mers suggested by the reads to be from diploid regions are in fact shared between the two assemblies, as they should be if they are from homozygous regions. The haploid coverage <em>k</em>-mers (around ~25x coverage) are split between hap1 and hap2 assemblies, somewhat unevenly but still not as bad as it would be in an assembly without phasing data.</p>

</div>

<div class="solo">

  <h2 id="pseudohaplotype-assembly-with-hifiasm">Pseudohaplotype assembly with <strong>hifiasm</strong></h2>

  <p>When hifiasm is run without any additional phasing data, it will do its best to generate a pseudohaplotype primary/alternate set of assemblies. These assemblies will typically contain more contigs that switch between parental blocks. Because of this, the primary assembly generated with this method can have a higher N50 value than an assembly generated with haplotype-phasing, but the contigs will contain more switch errors.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-pseudohaplotype-assembly-with-b-hifiasm-b"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Pseudohaplotype assembly with <b>hifiasm</b></div>
    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/hifiasm/hifiasm/0.18.8+galaxy1" title="Hifiasm tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Hifiasm</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.18.8+galaxy1)</span> with the following parameters:
        <ul>
          <li><em>“Assembly mode”</em>: <code class="language-plaintext highlighter-rouge">Standard</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input reads”</em>: <code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code> (output of <strong>Cutadapt</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              <li><em>“Options for purging duplicates”</em>: <code class="language-plaintext highlighter-rouge">Specify</code></li>
              <li><em>“Purge level”</em>: <code class="language-plaintext highlighter-rouge">Light (1)</code></li>
            </ul>
          </li>
        </ul>

        <blockquote class="comment">
          <div class="box-title comment-title" id="comment-a-note-on-hifiasm-purging-level"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: A note on hifiasm purging level</div>
          <p>hifiasm has an internal purging function, which we have set to <code class="language-plaintext highlighter-rouge">Light</code> here. The VGP pipeline currently disables the hifiasm internal purging, in favor of using the <b>purge_dups</b> suite after the fact in order to have more control over the parameters used for purging.</p>
        </blockquote>
      </li>
      <li>After the tool has finished running, rename its outputs as follows:
        <ul>
          <li>Rename the <code class="language-plaintext highlighter-rouge">primary assembly contig graph for pseudohaplotype assembly</code> as <code class="language-plaintext highlighter-rouge">Primary contigs graph</code> and add a <code class="language-plaintext highlighter-rouge">#pri</code> tag</li>
          <li>Rename the <code class="language-plaintext highlighter-rouge">alternate assembly contig graph for pseudohaplotype assemblyh</code> as <code class="language-plaintext highlighter-rouge">Alternate contigs graph</code> and add a <code class="language-plaintext highlighter-rouge">#alt</code> tag</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>We have obtained the primary and alternate contig graphs (as <abbr title="Graphical Fragment Assembly">GFA</abbr> files), but these must be converted to FASTA format for subsequent steps. We will use a tool developed from the VGP: <strong>gfastats</strong>. gfastats is a tool suite that allows for manipulation and evaluation of FASTA and GFA files, but in this instance we will use it to convert our GFAs to FASTA files. Later on we will use it to generate standard summary statistics for our assemblies.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-convert-gfa-to-fasta-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: convert GFA to FASTA</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0" title="gfastats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>gfastats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input GFA file”</em>: select <code class="language-plaintext highlighter-rouge">Primary contigs graph</code> and the <code class="language-plaintext highlighter-rouge">Alternate contigs graph</code> datasets</li>
        </ul>

        <blockquote class="tip">
          <div class="box-title tip-title" id="tip-select-multiple-datasets-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
            <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
          </ol>
        </blockquote>

        <ul>
          <li><em>“Tool mode”</em>: <code class="language-plaintext highlighter-rouge">Genome assembly manipulation</code></li>
          <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">FASTA</code></li>
          <li><em>“Generates the initial set of paths</em>”: toggle to <code class="language-plaintext highlighter-rouge">yes</code></li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code> and <code class="language-plaintext highlighter-rouge">Alternate contigs FASTA</code></li>
    </ol>

  </blockquote>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-summary-statistics-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Summary statistics</div>

    <p>gfastats will provide us with the following statistics:</p>

    <ul>
      <li>No. of contigs: The total number of contigs in the assembly.</li>
      <li>Largest contig: The length of the largest contig in the assembly.</li>
      <li>Total length: The total number of bases in the assembly.</li>
      <li>Nx: The largest contig length, <em>L</em>, such that contigs of length &gt;= <em>L</em> account for at least <em>x</em>% of the bases of the assembly.</li>
      <li>NGx: The contig length such that using equal or longer length contigs produces <em>x</em>% of the length of the reference genome, rather than <em>x</em>% of the assembly length.</li>
      <li>GC content: the percentage of nitrogenous bases which are either guanine or cytosine.</li>
    </ul>

  </blockquote>

  <p>Let’s use gfastats to get a basic idea of what our assembly looks like. We’ll run gfastats on the <abbr title="Graphical Fragment Assembly">GFA</abbr> files because gfastats can report graph-specific statistics as well. After generating the stats, we’ll be doing some text manipulation to get the stats side-by-side in a pretty fashion.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-assembly-evaluation-with-gfastats-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assembly evaluation with gfastats</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0" title="gfastats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>gfastats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Primary contigs graph</code> and the <code class="language-plaintext highlighter-rouge">Alternate contigs graph</code> datasets</li>
          <li><em>“Expected genome size”</em>: <code class="language-plaintext highlighter-rouge">11747160</code> (remember we calculated this value earlier, so it should be in your history!)</li>
          <li><em>“Generates the initial set of paths</em>”: toggle to <code class="language-plaintext highlighter-rouge">yes</code></li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Primary stats</code> and <code class="language-plaintext highlighter-rouge">Alternate stats</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3" title="Column join tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column join</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.3)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Primary stats</code> and the <code class="language-plaintext highlighter-rouge">Alternate stats</code> datasets</li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">gfastats on pri and alt (full)</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1" title="Search in textfiles tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search in textfiles</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">gfastats on pri and alt (full)</code></li>
          <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Don't Match</code></li>
          <li><em>“Type of regex”</em>: <code class="language-plaintext highlighter-rouge">Basic</code></li>
          <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">[Ss]caffold</code></li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">gfastats on pri and alt contigs</code></li>
    </ol>

  </blockquote>

  <p>Take a look at the <em>gfastats on pri and alt contigs</em> output — it should have three columns: 1) name of statistic, 2) primary assembly value, and 3) alternate assembly value. The report makes it clear that the two assemblies are markedly uneven: the primary assembly has 25 contigs totalling ~18.5 Mbp, while the alternate assembly has 8 contigs totalling only about 4.95 Mbp. If you’ll remember that our estimated genome size is ~11.7 Mbp, then you’ll see that the primary assembly has almost 2/3 more sequence than expected for a haploid representation of the genome! This is because a lot of heterozygous regions have had <em>both</em> copies of those loci placed into the primary assembly, as a result of incomplete purging. The presence of false duplications can be confirmed by looking at <abbr title="Benchmarking Universal Single-Copy Orthologs">BUSCO</abbr> and Merqury results.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>What is the length of the longest contigs in the assemblies?</li>
      <li>What are the N50 values of the two assemblies? Are they very different from each other? What might explain the difference between the two?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>The primary’s longest contig is 1,532,843 bp while the alternate’s longest contig is 1,090,521.</li>
        <li>The primary assembly has a N50 of 813,311, while the alternate’s is 1,077,964. These are not that far apart. The alternate might have a bigger N50 due to having fewer contigs.</li>
      </ol>

    </blockquote>

  </blockquote>

  <p>Next, we will use <abbr title="Benchmarking Universal Single-Copy Orthologs">BUSCO</abbr>, which will provide quantitative assessment of the completeness of a genome assembly in terms of expected gene content. It relies on the analysis of genes that should be present only once in a complete assembly or gene set, while allowing for rare gene duplications or losses (<span class="citation"><a href="#Simo2015">Simão <i>et al.</i> 2015</a></span>).</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-assessing-assembly-completeness-with-busco-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Assessing assembly completeness with BUSCO</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0" title="Busco tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Busco</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Sequences to analyze”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
          <li><em>“Mode”</em>: <code class="language-plaintext highlighter-rouge">Genome assemblies (DNA)</code>
            <ul>
              <li><em>“Use Augustus instead of Metaeuk”</em>: <code class="language-plaintext highlighter-rouge">Use Metaeuk</code></li>
            </ul>
          </li>
          <li><em>“Auto-detect or select lineage?”</em>: <code class="language-plaintext highlighter-rouge">Select lineage</code>
            <ul>
              <li><em>“Lineage”</em>: <code class="language-plaintext highlighter-rouge">Saccharomycetes</code></li>
            </ul>
          </li>
          <li><em>“Which outputs should be generated”</em>: <code class="language-plaintext highlighter-rouge">short summary text</code> and <code class="language-plaintext highlighter-rouge">summary image</code></li>
        </ul>

        <blockquote class="comment">
          <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

          <p>Remember to modify the lineage option if you are working with vertebrate genomes.</p>
        </blockquote>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">BUSCO primary contigs</code>.</li>
    </ol>

  </blockquote>

  <p>We have asked <abbr title="Benchmarking Universal Single-Copy Orthologs">BUSCO</abbr> to generate two particular outputs: the short summary, and a summary image.</p>
  <figure id="figure-13" style="max-width: 90%;"><img src="../../images/vgp_assembly/busco_pri_unpurged.png" alt="BUSCO for primary contigs." width="2450" height="1024" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/busco_pri_unpurged.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 13</strong>:</span> BUSCO results for the primary contigs. The summary image (left) gives a good overall idea of the status of BUSCO genes within the assembly, while the short summary (right) lists these as percentages as well. In this case, this primary assembly seems to have a large amount of duplicated BUSCO genes, but is otherwise complete (<i>i.e.</i>, not much missing content).</figcaption></figure>

  <p>The BUSCO results support our hypothesis that the primary assembly is so much larger than expected due to improper purging, resulting in false duplications.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <ol>
      <li>How many complete and single copy BUSCO genes have been identified in the pri assembly? What percentage of the total BUSCO gene set is that?</li>
      <li>How many complete and duplicated BUSCO genes have been identified in the pri assembly? What percentage of the total BUSCO gene set is that?</li>
      <li>What is the percentage of complete BUSCO genes for the primary assembly?</li>
    </ol>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <ol>
        <li>The primary assembly has 1159 complete and single-copy BUSCO genes, which is 54.2% of the gene set.</li>
        <li>The primary assembly has 927 complete and duplicated BUSCO genes, which is 43.4% of the gene set.</li>
        <li>97.6% of BUSCOs are complete in the assembly.</li>
      </ol>

    </blockquote>

  </blockquote>

  <p>Despite BUSCO being robust for species that have been widely studied, it can be inaccurate when the newly assembled genome belongs to a taxonomic group that is not well represented in <a href="https://www.orthodb.org/">OrthoDB</a>. Merqury provides a complementary approach for assessing genome assembly quality metrics in a reference-free manner via <em>k</em>-mer copy number analysis.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-i-k-i-mer-based-evaluation-with-merqury-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: <i>k</i>-mer based evaluation with Merqury</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3" title="Merqury tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merqury</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3)</span> with the following parameters:
        <ul>
          <li><em>“Evaluation mode”</em>: <code class="language-plaintext highlighter-rouge">Default mode</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“k-mer counts database”</em>: <code class="language-plaintext highlighter-rouge">Merged meryldb</code></li>
              <li><em>“Number of assemblies”</em>: `Two assemblies
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First genome assembly”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second genome assembly”</em>: <code class="language-plaintext highlighter-rouge">Alternate contigs FASTA</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>By default, Merqury generates three collections as output: stats, plots and <abbr title="consensus accuracy quality value">QV</abbr> stats. The “stats” collection contains the completeness statistics, while the “QV stats” collection contains the quality value statistics. Let’s have a look at the assembly <abbr title="copy number">CN</abbr> spectrum plot, known as the <em>spectra-cn</em> plot (fig. 7).</p>

  <figure id="figure-14" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_cn_plot.png" alt="Merqury spectra-cn plot for the pri/alt assemblies." width="1800" height="1500" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_cn_plot.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 14</strong>:</span> Merqury CN plot. This plot tracks the multiplicity of each k-mer found in the Hi-Fi read set and colors it by the number of times it is found in a given assembly. Merqury connects the midpoint of each histogram bin with a line, giving the illusion of a smooth curve.</figcaption></figure>

  <p>The black region in the left side corresponds to k-mers found only in the read set; it is usually indicative of sequencing error in the read set, although it can also be indicative of missing sequences in the assembly. The red area represents one-copy k-mers in the genome, while the blue area represents two-copy k-mers originating from homozygous sequence or haplotype-specific duplications. From this figure we can state that the diploid sequencing coverage is around 50x, which we also know from the GenomeScope2 plot we looked at earlier.</p>

  <p>To get an idea of how the <em>k</em>-mers have been distributed between our hap1 and hap2 assemblies, we should look at the <em>spectra-asm</em> output of Merqury.</p>

  <figure id="figure-15" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_prialt_asm_prepurge.png" alt="Merqury spectra-asm plot for the hap1/hap2 assemblies." width="1234" height="982" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_prialt_asm_prepurge.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 15</strong>:</span> Merqury ASM plot. This plot tracks the multiplicity of each k-mer found in the Hi-Fi read set and colors it according to which assemblies contain those k-mers. This can tell you which k-mers are found in only one assembly or shared between them.</figcaption></figure>

  <p>For an idea of what a properly phased spectra-asm plot would look like, <strong>please click over to the Hi-C phasing version of this tutorial</strong>. A properly phased spectra-asm plot should have a large green peak centered around the point of diploid coverage (here ~50X), and the two assembly-specific peaks should be centered around the point of haploid coverage (here ~25X) and resembling each other in size.</p>

  <p>The spectra-asm plot we have for our primary &amp; alternate assemblies here does not resemble one that is properly phased. There is a peak of green (shared) <em>k</em>-mers around diploid coverage, indicating that some homozygous regions have been properly split between the primary and alternate assemblies; however, there is still a large red peak of primary-assembly-only <em>k</em>-mers at that coverage value, too, which means that some homozygous regions are being represented twice in the primary assembly, instead of once in the primary and once in the alternate. Additionally, for the haploid peaks, the primary-only peak (in red) is much larger than the alternate-only peak (in blue), indicating that a lot of heterozygous regions might have both their alternate alleles represented in the primary assembly, which is false duplication.</p>

  <p>For further confirmation, we can also look at the individual, assembly-specific <abbr title="copy number">CN</abbr> plots. In the Merqury outputs, the <code class="language-plaintext highlighter-rouge">output_merqury.assembly_01.spectra-cn.fl</code> is a <abbr title="copy number">CN</abbr> spectra with <em>k</em>-mers colored according to their copy number in the primary assembly.</p>

  <figure id="figure-16" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_prialt_priCN_prepurge.png" alt="Merqury spectra-cn plot for the pri assembly only." width="1234" height="982" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_prialt_priCN_prepurge.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 16</strong>:</span> Merqury CN plot <i>for the primary assembly only</i>. This plot colors k-mers according to their copy number in the primary assembly. K-mers that are present in the reads but not the primary assembly are labelled 'read-only'.</figcaption></figure>

  <p>In the primary-only <abbr title="copy number">CN</abbr> plot, we observe a large 2-copy (colored blue) peak at diploid coverage. Ideally, this would not be here, beacause these diploid regions would be <em>1-copy in both assemblies</em>. Purging this assembly should reconcile this by removing one copy of false duplicates, making these 2-copy <em>k</em>-mers 1-copy. You might notice the ‘read-only’ peak at haploid coverage — this is actually expected, because ‘read-only’ here just means that the <em>k</em>-mer in question is not seen in this specific assembly while it was in the original readset. <strong>Often, these ‘read-only’ <em>k</em>-mers are actually present as alternate loci in the other assembly.</strong></p>

  <p>Now that we have looked at our primary assembly with multiple <abbr title="Quality Control">QC</abbr> metrics, we know that it should undergo purging. The VGP pipeline uses <strong>purge_dups</strong> to remove false duplications from the primary assembly and put them in the alternate assembly to reconcile the haplotypes. Additionally, purge_dups can also find collapsed repeats and regions of suspiciously low coverage.</p>

  <h2 id="purging-the-primary-and-alternate-assemblies">Purging the primary and alternate assemblies</h2>

  <p>Before proceeding to purging, we need to carry out some text manipulation operations on the output generated by GenomeScope2 to make it compatible with downstream tools. The goal is to extract some parameters which at a later stage will be used by <strong>purge_dups</strong>.</p>

  <h3 id="parsing-purgedups-cutoffs-from-genomescope2-output">Parsing <strong>purge_dups</strong> cutoffs from <strong>GenomeScope2</strong> output</h3>

  <p>The first relevant parameter is the <code class="language-plaintext highlighter-rouge">estimated genome size</code>.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-get-estimated-genome-size"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get estimated genome size</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Replace parts of text</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.4)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: <code class="language-plaintext highlighter-rouge">summary</code> (output of <strong>GenomeScope</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">bp</code></li>
          <li><em>“Replace with</em>”: leave this field empty (as it is)</li>
          <li><em>“Replace all occurrences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.4" title="Replace parts of text tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Replace parts of text</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.4)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: output file of <strong>Replace</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">,</code></li>
          <li><em>“Replace with</em>”: leave this field empty (as it is)</li>
          <li><em>“Replace all occurrences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1" title="Search in textfiles (grep) tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search in textfiles (grep)</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select lines from”</em>: output file of the previous step.</li>
          <li><em>“Type of regex”</em>: <code class="language-plaintext highlighter-rouge">Basic</code></li>
          <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">Haploid</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="Convert characters1" title="Convert delimiters to TAB tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Convert delimiters to TAB</strong></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“in Dataset”</em>: output of <strong>Search in textfiles</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Advanced Cut tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Advanced Cut</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: output of <strong>Convert delimiters to TAB</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code>
            <ul>
              <li><em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 5</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Estimated genome size</code>.</li>
    </ol>

    <blockquote class="question">
      <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

      <p>What is the estimated genome size?</p>

      <blockquote class="solution">
        <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>The estimated genome size is 11,747,076 bp.</p>

      </blockquote>

    </blockquote>

  </blockquote>

  <p>Now let’s parse the <code class="language-plaintext highlighter-rouge">transition between haploid &amp; diploid</code> and <code class="language-plaintext highlighter-rouge">upper bound for the read depth estimation</code> parameters. The transition between haploid &amp; diploid represents the coverage value halfway between haploid and diploid coverage, and helps purger_dups identify <em>haplotigs</em>. The upper bound parameter will be used by purge_dups as high read depth cutoff to identify <em>collapsed repeats</em>. When repeats are collapsed in an assembly, they are not as long as they actually are in the genome. This results in a pileup of reads at the collapsed region when mapping the reads back to the assembly.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-get-maximum-read-depth"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Get maximum read depth</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/column_maker/Add_a_column1/2.0" title="Compute on rows tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compute on rows</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: <code class="language-plaintext highlighter-rouge">model_params</code> (output of <strong>GenomeScope</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li>For 1: Expressions:
            <ul>
              <li><em>“Add expression”</em>: <code class="language-plaintext highlighter-rouge">round(1.5*c3)</code></li>
              <li><em>“Mode of the operation”</em>: <code class="language-plaintext highlighter-rouge">Append</code></li>
            </ul>
          </li>
          <li>Click <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> Insert Expressions</li>
          <li>For 2: Expressions:
            <ul>
              <li><em>“Add expression”</em>: <code class="language-plaintext highlighter-rouge">3*c7</code></li>
              <li><em>“Mode of the operation”</em>: <code class="language-plaintext highlighter-rouge">Append</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename it as <code class="language-plaintext highlighter-rouge">Parsing purge parameters</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Advanced Cut tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Advanced Cut</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: <code class="language-plaintext highlighter-rouge">Parsing purge parameters</code></li>
          <li><em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code>
            <ul>
              <li><em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 8</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Maximum depth</code></li>
    </ol>

    <blockquote class="question">
      <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

      <ol>
        <li>What is the estimated maximum depth?</li>
        <li>What does this parameter represent?</li>
      </ol>

      <blockquote class="solution">
        <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ol>
          <li>The estimated maximum depth is 114 reads.</li>
          <li>The maximum depth indicates the maximum number of sequencing reads that align to specific positions in the genome.</li>
        </ol>

      </blockquote>

    </blockquote>

    <p>Now let’s get the transition parameter.</p>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0" title="Advanced Cut tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Advanced Cut</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: <code class="language-plaintext highlighter-rouge">Parsing purge parameters</code></li>
          <li><em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code>
            <ul>
              <li><em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 7</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Transition parameter</code></li>
    </ol>

    <blockquote class="question">
      <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

      <p>What is the estimated transition parameter?</p>

      <blockquote class="solution">
        <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>The estimated transition parameter is 38 reads.</p>

      </blockquote>

    </blockquote>

  </blockquote>

  <h2 id="purging-with-purgedups">Purging with <strong>purge_dups</strong></h2>

  <p>An ideal haploid representation would consist of one allelic copy of all heterozygous regions in the two haplomes, as well as all hemizygous regions from both haplomes (<span class="citation"><a href="#Guan2019">Guan <i>et al.</i> 2019</a></span>). However, in highly heterozygous genomes, assembly algorithms are frequently not able to identify the highly divergent allelic sequences as belonging to the same region, resulting in the assembly of those regions as separate contigs. This can lead to issues in downstream analysis, such as scaffolding, gene annotation and read mapping in general (<span class="citation"><a href="#Small2007">Small <i>et al.</i> 2007</a></span>, <span class="citation"><a href="#Guan2019">Guan <i>et al.</i> 2019</a></span>, <span class="citation"><a href="#Roach2018">Roach <i>et al.</i> 2018</a></span>). In order to solve this problem, we are going to use purge_dups; this tool will allow us to identify and reassign allelic contigs.</p>

  <p>This stage consists of three substages: read-depth analysis, generation of all versus all self-alignment and resolution of haplotigs and overlaps (fig. 8).</p>

  <figure id="figure-17" style="max-width: 90%;"><img src="../../images/vgp_assembly/purge_dupspipeline.png" alt="Post-processing with purge_dups. " width="2268" height="1160" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/purge_dupspipeline.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 17</strong>:</span> Purge_dups pipeline. Adapted from github.com/dfguan/purge_dups. Purge_dups is integrated in a multi-step pipeline consisting in three main substages. Red indicates the steps which require to use Minimap2.</figcaption></figure>

  <h3 id="read-depth-analysis">Read-depth analysis</h3>

  <p>Initially, we need to collapse our HiFi trimmed reads collection into a single dataset.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-collapse-the-collection"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Collapse the collection</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/nml/collapse_collections/collapse_dataset/4.2" title="Collapse Collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Collapse Collection</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.2)</span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collection of files to collapse into single dataset”</em>:<code class="language-plaintext highlighter-rouge">HiFi_collection (trim)</code></li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">HiFi reads collapsed</code></li>
    </ol>
  </blockquote>

  <p>Now, we will map the reads against the primary assembly by using Minimap2 (<span class="citation"><a href="#Li2018">Li 2018</a></span>), an alignment program designed to map long sequences.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-map-the-reads-to-contigs-with-b-minimap2-b"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Map the reads to contigs with <b>Minimap2</b></div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Map with minimap2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with minimap2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.17+galaxy4)</span> with the following parameters:
        <ul>
          <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
            </ul>
          </li>
          <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
            <ul>
              <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">HiFi reads collapsed</code></li>
              <li><em>“Select a profile of preset options”</em>: <code class="language-plaintext highlighter-rouge">Long assembly to reference mapping (-k19 -w19 -A1 -B19 -O39,81 -E3,1 -s200 -z200 --min-occ-floor=100). Typically, the alignment will not extend to regions with 5% or higher sequence divergence. Only use this preset if the average divergence is far below 5%. (asm5)</code></li>
            </ul>
          </li>
          <li>In <em>“Set advanced output options”</em>:
            <ul>
              <li><em>“Select an output format”</em>: <code class="language-plaintext highlighter-rouge">paf</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Reads mapped to contigs</code></li>
    </ol>
  </blockquote>

  <p>Finally, we will use the <code class="language-plaintext highlighter-rouge">Reads mapped to contigs</code> pairwise mapping format (PAF) file for calculating some statistics required in a later stage. In this step, purge_dups (listed as <strong>Purge overlaps</strong> in Galaxy tool panel) initially produces a read-depth histogram from base-level coverages. This information is used for estimating the coverage cutoffs, taking into account that collapsed haplotype contigs will lead to reads from both alleles mapping to those contigs, whereas if the alleles have assembled as separate contigs, then the reads will be split over the two contigs, resulting in half the read-depth (<span class="citation"><a href="#Roach2018">Roach <i>et al.</i> 2018</a></span>).</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-read-depth-analisys"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read-depth analisys</div>
    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy3" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy3)</span> with the following parameters:
        <ul>
          <li><em>“Function mode”</em>: <code class="language-plaintext highlighter-rouge">Calculate coverage cutoff, base-level read depth and create read depth histogram for PacBio data (calcuts+pbcstats)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“PAF input file”</em>: <code class="language-plaintext highlighter-rouge">Reads mapped to contigs</code></li>
              <li>In <em>“Calcuts options”</em>:
                <ul>
                  <li><em>“Transition between haploid and diploid”</em>: 38</li>
                  <li><em>“Upper bound for read depth”</em>: <code class="language-plaintext highlighter-rouge">114</code> (the previously estimated maximum depth)</li>
                  <li><em>“Ploidity”</em>: <code class="language-plaintext highlighter-rouge">Diploid</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage primary</code>, <code class="language-plaintext highlighter-rouge">Histogram plot primary</code> and <code class="language-plaintext highlighter-rouge">Calcuts cutoff primary</code>.</li>
    </ol>
  </blockquote>

  <p>Purge overlaps (purge_dups) generates three outputs:</p>

  <ul>
    <li>PBCSTAT base coverage: it contains the base-level coverage information.</li>
    <li>Calcuts-cutoff: it includes the thresholds calculated by purge_dups.</li>
    <li>Histogram plot.</li>
  </ul>

  <h3 id="generation-of-all-versus-all-self-alignment">Generation of all versus all self-alignment</h3>

  <p>Now, we will segment the draft assembly into contigs by cutting at blocks of <em>N</em>s, and use minimap2 to generate an all by all self-alignment.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-purge-dups-pipeline"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: purge_dups pipeline</div>
    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy2)</span> with the following parameters:
        <ul>
          <li><em>“Function mode”</em>: <code class="language-plaintext highlighter-rouge">split assembly FASTA file by 'N's (split_fa)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Assembly FASTA file”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">Split FASTA</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Map with minimap2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with minimap2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.17+galaxy4)</span> with the following parameters:
        <ul>
          <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Split FASTA</code></li>
            </ul>
          </li>
          <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">Split FASTA</code></li>
              <li><em>“Select a profile of preset options”</em>: <code class="language-plaintext highlighter-rouge">Construct a self-homology map - use the same genome as query and reference (-DP -k19 -w 19 -m200) (self-homology)</code></li>
            </ul>
          </li>
          <li>In <em>“Set advanced output options”</em>:
            <ul>
              <li><em>“Select an output format”</em>: <code class="language-plaintext highlighter-rouge">PAF</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Self-homology map primary</code></li>
    </ol>
  </blockquote>

  <h3 id="resolution-of-haplotigs-and-overlaps">Resolution of haplotigs and overlaps</h3>

  <p>During the final step of the purge_dups pipeline, it will use the self alignments and the cutoffs for identifying the haplotypic duplications.</p>

  <blockquote class="details">
    <div class="box-title details-title" id="details-purge-overlaps-purge-dups-algorithm-details"><button class="gtn-boxify-button details" type="button" aria-controls="details-purge-overlaps-purge-dups-algorithm-details" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Purge overlaps (purge_dups) algorithm details</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>In order to identify the haplotypic duplications, purge_dups uses the  base-level coverage information to flag the contigs according the following criteria:</p>
    <ul>
      <li>If more than 80% bases of a contig are above the high read depth cutoff or below the noise cutoff, it is discarded.</li>
      <li>If more than 80% bases are in the diploid depth interval, it is labeled as a primary contig, otherwise it is considered further as a possible haplotig.</li>
    </ul>

    <p>Contigs that were flagged for further analysis according to read-depth are then evaluated to attempt to identify synteny with its allelic companion contig. In this step, purge_dups uses the information contained in the self alignments:</p>
    <ul>
      <li>
        <p>If the alignment score is larger than the cutoff <em>s</em> (default 70), the contig is marked for reassignment as haplotig. Contigs marked for reassignment with a maximum match score greater than the cutoff <em>m</em> (default 200) are further flagged as repetitive regions.</p>
      </li>
      <li>
        <p>Otherwise contigs are considered as a candidate primary contig.</p>
      </li>
    </ul>

    <p>Once all matches associated with haplotigs have been removed from the self-alignment set, purge_dups ties consistent matches between the remaining candidates to find collinear matches, filtering all the matches whose score is less than the minimum chaining score <em>l</em>.</p>

    <p>Finally, purge_dups calculates the average coverage of the matching intervals for each overlap, and mark an unambiguous overlap as heterozygous when the average coverage on both contigs is less than the read-depth cutoff, removing the sequences corresponding to the matching interval in the shorter contig.</p>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-resolution-of-haplotigs-and-overlaps"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Resolution of haplotigs and overlaps</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy5" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy5)</span> with the following parameters:
        <ul>
          <li><em>“Select the purge_dups function”</em>: <code class="language-plaintext highlighter-rouge">Purge haplotigs and overlaps for an assembly (purge_dups)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“PAF input file”</em>: <code class="language-plaintext highlighter-rouge">Self-homology map primary</code></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Base-level coverage file”</em>: <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage primary</code></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Cutoffs file”</em>: <code class="language-plaintext highlighter-rouge">calcuts cutoff primary</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">purge_dups BED</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy2)</span> with the following parameters:
        <ul>
          <li><em>“Select the purge_dups function”</em>: <code class="language-plaintext highlighter-rouge">Obtain sequences after purging (get_seqs)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Assembly FASTA file”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs FASTA</code></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BED input file”</em>: <code class="language-plaintext highlighter-rouge">purge_dups BED</code> (output of the previous step)</li>
            </ul>
          </li>
        </ul>
      </li>
      <li>Rename the output <code class="language-plaintext highlighter-rouge">get_seq purged sequences</code> as <code class="language-plaintext highlighter-rouge">Primary contigs purged</code> and the <code class="language-plaintext highlighter-rouge">get_seq haplotype</code> file as <code class="language-plaintext highlighter-rouge">Alternate haplotype contigs</code></li>
    </ol>

  </blockquote>

  <h3 id="process-the-alternate-assembly">Process the alternate assembly</h3>

  <p>Now we should repeat the same procedure with the alternate contigs generated by hifiasm.  In that case, we should start by merging the <code class="language-plaintext highlighter-rouge">Alternate haplotype contigs</code> generated in the previous step and the <code class="language-plaintext highlighter-rouge">Alternate contigs FASTA</code> file.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-merge-the-purged-sequences-and-the-alternate-contigs"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Merge the purged sequences and the Alternate contigs</div>

    <ol>
      <li><span class="tool" data-tool="cat1" title="Concatenate datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets</strong></span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Concatenate Dataset”</em>: <code class="language-plaintext highlighter-rouge">Alternate contigs FASTA</code></li>
          <li>In <em>“Dataset”</em>:
            <ul>
              <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Dataset”</em>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select”</em>: <code class="language-plaintext highlighter-rouge">Alternate haplotype contigs</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>

        <blockquote class="comment">
          <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

          <p>Remember that the <code class="language-plaintext highlighter-rouge">Alternate haplotype contigs</code> file contains those contigs that were considered to be haplotypic duplications of the primary contigs.</p>
        </blockquote>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">Alternate contigs full</code></li>
    </ol>

  </blockquote>

  <p>Once we have merged the files, we should run the purge_dups pipeline again, but using the <code class="language-plaintext highlighter-rouge">Alternate contigs full</code> file as input.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-process-the-alternate-assembly-with-i-purge-dups-i"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Process the alternate assembly with <i>purge_dups</i></div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Map with minimap2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with minimap2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.17+galaxy4)</span> with the following parameters:
        <ul>
          <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Alternate contigs full</code></li>
            </ul>
          </li>
          <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
            <ul>
              <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">HiFi reads collapsed</code></li>
              <li><em>“Select a profile of preset options”</em>: <code class="language-plaintext highlighter-rouge">Long assembly to reference mapping (-k19 -w19 -A1 -B19 -O39,81 -E3,1 -s200 -z200 --min-occ-floor=100). Typically, the alignment will not extend to regions with 5% or higher sequence divergence. Only use this preset if the average divergence is far below 5%. (asm5)</code> (<strong>Note</strong> <code class="language-plaintext highlighter-rouge">asm5</code> at the end!)</li>
            </ul>
          </li>
          <li>In <em>“Set advanced output options”</em>:
            <ul>
              <li><em>“Select an output format”</em>: <code class="language-plaintext highlighter-rouge">paf</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">Reads mapped to contigs alternate</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy3" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy3)</span> with the following parameters:
        <ul>
          <li><em>“Function mode”</em>: <code class="language-plaintext highlighter-rouge">Calculate coverage cutoff, base-level read depth and create read depth histogram for PacBio data (calcuts+pbcstats)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“PAF input file”</em>: <code class="language-plaintext highlighter-rouge">Reads mapped to contigs alternate</code></li>
              <li>In <em>“Calcuts options”</em>:
                <ul>
                  <li><em>“Upper bound for read depth”</em>: <code class="language-plaintext highlighter-rouge">114</code></li>
                  <li><em>“Ploidity”</em>: <code class="language-plaintext highlighter-rouge">Diploid</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the outputs as <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage alternate</code>, <code class="language-plaintext highlighter-rouge">Histogram plot alternate</code> and <code class="language-plaintext highlighter-rouge">Calcuts cutoff alternate</code>.</p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy2)</span> with the following parameters:
        <ul>
          <li><em>“Function mode”</em>: <code class="language-plaintext highlighter-rouge">split assembly FASTA file by 'N's (split_fa)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Assembly FASTA file”</em>: <code class="language-plaintext highlighter-rouge">Alternate contigs full</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">Split FASTA alternate</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/minimap2/minimap2/2.17+galaxy4" title="Map with minimap2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with minimap2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.17+galaxy4)</span> with the following parameters:
        <ul>
          <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Split FASTA alternate</code></li>
            </ul>
          </li>
          <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">Split FASTA alternate</code></li>
              <li><em>“Select a profile of preset options”</em>: <code class="language-plaintext highlighter-rouge">Construct a self-homology map - use the same genome as query and reference (-DP -k19 -w 19 -m200) (self-homology)</code></li>
            </ul>
          </li>
          <li>In <em>“Set advanced output options”</em>:
            <ul>
              <li><em>“Select an output format”</em>: <code class="language-plaintext highlighter-rouge">PAF</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">Self-homology map alternate</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy5" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy5)</span> with the following parameters:
        <ul>
          <li><em>“Select the purge_dups function”</em>: <code class="language-plaintext highlighter-rouge">Purge haplotigs and overlaps for an assembly (purge_dups)</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“PAF input file”</em>: <code class="language-plaintext highlighter-rouge">Self-homology map alternate</code></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Base-level coverage file”</em>: <code class="language-plaintext highlighter-rouge">PBCSTAT base coverage alternate</code></li>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Cutoffs file”</em>: <code class="language-plaintext highlighter-rouge">calcuts cutoff alternate</code></li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">purge_dups BED alternate</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/purge_dups/purge_dups/1.2.5+galaxy2" title="Purge overlaps tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Purge overlaps</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.2.5+galaxy2)</span> with the following parameters:
        <ul>
          <li><em>“Select the purge_dups function”</em>: <code class="language-plaintext highlighter-rouge">Obtain sequences after purging (get_seqs)</code></li>
        </ul>
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Assembly FASTA file”</em>: <code class="language-plaintext highlighter-rouge">Alternate contigs full</code></li>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BED input file”</em>: <code class="language-plaintext highlighter-rouge">purge_dups BED alternate</code></li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Alternate contigs purged</code> and <code class="language-plaintext highlighter-rouge">Alternate haplotype contigs</code>.</li>
    </ol>

  </blockquote>

  <h3 id="post-purge-quality-control">Post-purge quality control</h3>

  <p>Recall that, prior to purging, our primary assembly showed it needed to be purged due to several quality metrics: there was a large amount of BUSCO duplicated genes, many 2-copy <em>k</em>-mers that were primary-only instead of shared, and the primary was ~2/3 times larger than the expected genome size. Let’s check if these metrics changed after purging.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-evaluating-the-purged-assemblies"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Evaluating the purged assemblies</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0" title="gfastats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>gfastats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Primary contigs purged</code> and the <code class="language-plaintext highlighter-rouge">Alternate contigs purged</code> datasets</li>
          <li><em>“Expected genome size”</em>: <code class="language-plaintext highlighter-rouge">11747160</code> (remember we calculated this value earlier, so it should be in your history!)</li>
        </ul>
      </li>
      <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Primary purged stats</code> and <code class="language-plaintext highlighter-rouge">Alternate purged stats</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/collection_column_join/collection_column_join/0.0.3" title="Column join tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column join</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.3)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Primary purged stats</code> and the <code class="language-plaintext highlighter-rouge">Alternate purged stats</code> datasets</li>
        </ul>
      </li>
      <li>Rename the output as <code class="language-plaintext highlighter-rouge">gfastats on purged pri and alt (full)</code></li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.1.1" title="Search in textfiles tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search in textfiles</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">gfastats on purged pri and alt (full)</code></li>
          <li><em>“that”</em>: <code class="language-plaintext highlighter-rouge">Don't Match</code></li>
          <li><em>“Type of regex”</em>: <code class="language-plaintext highlighter-rouge">Basic</code></li>
          <li><em>“Regular Expression”</em>: <code class="language-plaintext highlighter-rouge">[Ss]caffold</code></li>
        </ul>
      </li>
      <li>
        <p>Rename the output as <code class="language-plaintext highlighter-rouge">gfastats on purged pri and alt contigs</code></p>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/busco/busco/5.0.0+galaxy0" title="Busco tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Busco</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.0.0+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Sequences to analyze”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs purged</code></li>
          <li><em>“Mode”</em>: <code class="language-plaintext highlighter-rouge">Genome assemblies (DNA)</code>
            <ul>
              <li><em>“Use Augustus instead of Metaeuk”</em>: <code class="language-plaintext highlighter-rouge">Use Metaeuk</code></li>
            </ul>
          </li>
          <li><em>“Auto-detect or select lineage?”</em>: <code class="language-plaintext highlighter-rouge">Select lineage</code>
            <ul>
              <li><em>“Lineage”</em>: <code class="language-plaintext highlighter-rouge">Saccharomycetes</code></li>
            </ul>
          </li>
          <li><em>“Which outputs should be generated”</em>: <code class="language-plaintext highlighter-rouge">short summary text</code> and <code class="language-plaintext highlighter-rouge">summary image</code></li>
        </ul>
      </li>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/merqury/merqury/1.3" title="Merqury tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merqury</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3)</span> with the following parameters:
        <ul>
          <li><em>“Evaluation mode”</em>: <code class="language-plaintext highlighter-rouge">Default mode</code>
            <ul>
              <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“k-mer counts database”</em>: <code class="language-plaintext highlighter-rouge">Merged meryldb</code></li>
              <li><em>“Number of assemblies”</em>: `Two assemblies
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First genome assembly”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs purged</code></li>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second genome assembly”</em>: <code class="language-plaintext highlighter-rouge">Alternate contigs purged</code></li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>The summary statistics indicate that both assemblies are now of a similar size that is much closer to the expected genome size: the purged primary assembly is 17 contigs totalling ~12.1 Mbp, while the purged alternate assembly is 16 contigs that amount to ~11.3 Mbp. This makes sense since, during purging, we removed sequences from the primary assembly and added them to the alternate assembly. But did we remove the <em>right</em> sequences? Remember, the sequences we are aiming to re-distribute are the false duplications that were flagged by BUSCO.</p>

  <figure id="figure-18" style="max-width: 90%;"><img src="../../images/vgp_assembly/busco_pri_purged.png" alt="BUSCO for primary assembly after purging." width="2450" height="1024" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/busco_pri_purged.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 18</strong>:</span> BUSCO for the primary assembly after purging.</figcaption></figure>

  <p>The <abbr title="Benchmarking Universal Single-Copy Orthologs">BUSCO</abbr> results for the purged primary assembly look much better, since we no longer have the large amount of duplicate BUSCOs that we previously had. Additionally, there is no large increase in missing BUSCOs, indicating that we have <em>not</em> over-purged the primary assembly.</p>

  <p>The previous metrics tell us that the primary is likely fixed after purging, but what about the previously incomplete alternate assembly? Let’s see if the Merqury spectra plots show any change in how <em>k</em>-mers are split up between the two assemblies.</p>

  <figure id="figure-19" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_prialt_asm_postpurge.png" alt="Merqury spectra-asm plot after purging." width="1234" height="982" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_prialt_asm_postpurge.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 19</strong>:</span> Merqury ASM plot after purging.</figcaption></figure>

  <p>This looks a lot better! The diploid regions are all shared between the two assemblies (the large green peak centered at 50x, the diploid coverage value), and the haplotypic variation is shared between the primary and alternate assemblies (the red and blue peaks centered around 25x, the haploid coverage value).</p>

  <figure id="figure-20" style="max-width: 90%;"><img src="../../images/vgp_assembly/merqury_prialt_priCN_postpurge.png" alt="Merqury spectra-cn plot for primary assembly after purging." width="1234" height="982" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/merqury_prialt_priCN_postpurge.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 20</strong>:</span> Merqury CN plot <i>for the primary assembly only</i> after purging.</figcaption></figure>

  <p>Additionally, when we look at the primary-only <abbr title="copy number">CN</abbr> plot, we see that the large peak of 2-copy <em>k</em>-mers is now gone, since those regions are now represented at 1-copy only in the primary assembly, as they should be.</p>

</div>

<h1 id="scaffolding">Scaffolding</h1>

<p>At this point, we have a set of contigs, which may or may not be fully phased, depending on how we ran hifiasm. Next, the contigs will be assembled into scaffolds, <em>i.e.</em>, sequences of contigs interspaced with gaps. The VGP pipeline currently scaffolds using two additional technologies: Bionano optical maps and <abbr title="all-versus-all chromatin conformation capture">Hi-C</abbr> data.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-assembly-am-i-scaffolding"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What assembly am I scaffolding??</div>

  <p>For the purposes of this tutorial, the scaffolding hands-on exercises will be <b>referring to a primary assembly</b>. If you have hap1 contigs or hap2 contigs, then you can also follow along just using hap1 contigs or hap2 contigs. <b>Wherever the tutorial refers to primary contigs, just replace with whichever haplotype you are scaffolding.</b></p>

</blockquote>

<figure id="figure-21" style="max-width: 90%;"><img src="../../images/vgp_assembly/scaffoldingandcuration.png" alt="Schematic of scaffolding contigs to generate a draft assembly, and then curating the draft scaffolds to generate a curated assembly." width="1184" height="1394" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/scaffoldingandcuration.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 21</strong>:</span> Scaffolding uses additional information to join together contigs. Optical maps, such as those produced using the Bionano Saphyr, as well as Hi-C data such as those generated by Arima Hi-C or Dovetail OmniC kits, are used to re-order and orient contigs according to long-range information. This generates the draft scaffold-level genome. This draft genome then undergoes manual curation, which uses information such as coverage tracks or repeat annotations in order to further orient scaffolds and correct misassemblies such as mis-joins and missed joins. Image adapted from <span class="citation"><a href="#Rhie2022">Rhie <i>et al.</i> 2020</a></span>.</figcaption></figure>

<h1 id="hybrid-scaffolding-with-bionano-optical-maps">Hybrid scaffolding with Bionano optical maps</h1>

<p>In this step, the linkage information provided by optical maps is integrated with primary assembly sequences, and the overlaps are used to orient and order the contigs, resolve chimeric joins, and estimate the length of gaps between adjacent contigs. One of the advantages of optical maps is that they can easily span genomic regions that are difficult to resolve using DNA sequencing technologies (<span class="citation"><a href="#Savara2021">Savara <i>et al.</i> 2021</a></span>, <span class="citation"><a href="#Yuan2020">Yuan <i>et al.</i> 2020</a></span>).</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-are-bionano-optical-maps"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-are-bionano-optical-maps" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What are Bionano optical maps?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>Bionano technology relies on the isolation of kilobase-long DNA fragments, which are labeled at specific sequence motifs with a fluorescent dye, resulting in a unique fluorescent pattern for each genome. DNA molecules are stretched into nanoscale channels and imaged with a high-resolution camera, allowing us to build optical maps that include the physical locations of labels rather than base-level information (<span class="citation"><a href="#Lam2012">Lam <i>et al.</i> 2012</a></span>, <span class="citation"><a href="#Giani2020">Giani <i>et al.</i> 2020</a></span>, <span class="citation"><a href="#Savara2021">Savara <i>et al.</i> 2021</a></span>).</p>

  <figure id="figure-22" style="max-width: 90%;"><img src="../../images/vgp_assembly/bionano.png" alt="Bionano optical maps. " width="5751" height="3751" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/bionano.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 22</strong>:</span> Bionano optical maps. Source: https://bionanogenomics.com</figcaption></figure>

  <p>The average optical map molecule length, around 225 kbp, is substantially larger than the PacBio HiFi reads, with read lengths averaging 10-25 kbp.</p>

</blockquote>

<p>The <em>Bionano Hybrid Scaffold</em> tool automates the scaffolding process, which includes five main steps:</p>

<ol>
  <li>Generate <em>in silico</em> maps for sequence assembly.</li>
  <li>Align <em>in silico</em> sequence maps against Bionano genome maps to identify and resolve potential conflicts.</li>
  <li>Merge the non-conflicting maps into hybrid scaffolds.</li>
  <li>Align sequence maps to the hybrid scaffolds</li>
  <li>Generate AGP and FASTA files for the scaffolds.</li>
</ol>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-bionano-hybrid-scaffolding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Bionano hybrid scaffolding</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/bionano_scaffold/bionano_scaffold/3.6.1+galaxy2" title="Bionano Hybrid Scaffold tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Bionano Hybrid Scaffold</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.6.1+galaxy2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“NGS FASTA”</em>: <code class="language-plaintext highlighter-rouge">Primary contigs purged</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“BioNano CMAP”</em>: <code class="language-plaintext highlighter-rouge">Bionano_dataset</code></li>
        <li><em>“Configuration mode”</em>: <code class="language-plaintext highlighter-rouge">VGP mode</code></li>
        <li><em>“Genome maps conflict filter”</em>: <code class="language-plaintext highlighter-rouge">Cut contig at conflict</code></li>
        <li><em>“Sequences conflict filter”</em>: <code class="language-plaintext highlighter-rouge">Cut contig at conflict</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>If your data are not associated with VGP, make sure that the configuration mode fits with your samples.</p>
      </blockquote>
    </li>
    <li><span class="tool" data-tool="cat1" title="Concatenate datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Concatenate Dataset”</em>: <code class="language-plaintext highlighter-rouge">NGScontigs scaffold NCBI trimmed</code> (output of <strong>Bionano Hybrid Scaffold</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>“Dataset”</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Dataset”</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select”</em>: <code class="language-plaintext highlighter-rouge">NGScontigs not scaffolded trimmed</code> (output of <strong>Bionano Hybrid Scaffold</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
  </ol>
</blockquote>

<h2 id="evaluating-bionano-scaffolds">Evaluating Bionano scaffolds</h2>

<p>Let’s evaluate our scaffolds to see the impact of scaffolding on some key assembly statistics.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-bionano-assembly-evaluation-with-quast-and-busco"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Bionano assembly evaluation with QUAST and BUSCO</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/gfastats/gfastats/1.3.6+galaxy0" title="gfastats tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>gfastats</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.6+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Input file”</em>: select <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
        <li><em>“Expected genome size”</em>: <code class="language-plaintext highlighter-rouge">11747160</code> (remember we calculated this value earlier, so it should be in your history!)</li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Bionano stats</code>.</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many scaffolds are in the primary assembly after the hybrid scaffolding?</li>
    <li>What is the size of the largest scaffold? Has improved with respect to the previous evaluation?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The number of contigs is 17.</li>
      <li>The largest contig is 1.531.728 bp long. This value hasn’t changed.</li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="hi-c-scaffolding">Hi-C scaffolding</h1>

<p>Hi-C is a sequencing-based molecular assay designed to identify regions of frequent physical interaction in the genome by measuring the contact frequency between all pairs of loci, allowing us to provide an insight into the three-dimensional organization of a genome  (<span class="citation"><a href="#Dixon2012">Dixon <i>et al.</i> 2012</a></span>, <span class="citation"><a href="#LiebermanAiden2009">Lieberman-Aiden <i>et al.</i> 2009</a></span>). In this final stage, we will exploit the fact that the contact frequency between a pair of loci strongly correlates with the one-dimensional distance between them with the objective of linking the Bionano scaffolds to a chromosome scale.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-how-does-hi-c-sequencing-work"><button class="gtn-boxify-button details" type="button" aria-controls="details-how-does-hi-c-sequencing-work" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: How does Hi-C sequencing work?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The high-throughput chromosome conformation capture (Hi-C) technology is based on the capture of the chromatin in three-dimensional space. During Hi-C library preparation, DNA is crosslinked in its 3D conformation. Then, the DNA is digested using restriction enzymes, and the digested ends are filled with biotinylated nucleotides. The biotinylated nucleotides enable the specific purification of the ligation junctions, preventing the sequencing of DNA molecules that do not contain such junctions which are thus mostly uninformative (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

  <figure id="figure-23" style="max-width: 90%;"><img src="../../images/vgp_assembly/hi-c_protocol.png" alt="Hi-C protocol. " width="2107" height="869" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hi-c_protocol.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 23</strong>:</span> Hi-C protocol. Adapted from <span class="citation"><a href="#Rao2014">Rao <i>et al.</i> 2014</a></span></figcaption></figure>

  <p>Next, the blunt ends of the spatially proximal digested end are ligated. Each DNA fragment is then sequenced from each end of this artificial junction, generating read pairs. This provides contact information that can be used to reconstruct the proximity of genomic sequences belonging to the same chromosome (<span class="citation"><a href="#Giani2020">Giani <i>et al.</i> 2020</a></span>). Hi-C data are in the form of two-dimensional matrices (contact maps) whose entries quantify the intensity of the physical interaction between genome regions.</p>

  <figure id="figure-24" style="max-width: 90%;"><img src="../../images/vgp_assembly/hic_scaffolding_haplotyping.jpg" alt="Hi-C scaffolding and haplotyping logic." width="1395" height="1397" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hic_scaffolding_haplotyping.jpg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 24</strong>:</span> Schematic of how Hi-C information is used to scaffold and orient contigs along chromosomes using long-range information, as well as link separated haplotype blocks into chromosome-scale haplotypes <span class="citation"><a href="#Korbel2013">Korbel and Lee 2013</a></span>.</figcaption></figure>
</blockquote>

<h2 id="pre-processing-hi-c-data">Pre-processing Hi-C data</h2>

<p>Despite Hi-C generating paired-end reads, we need to map each read separately. This is because most aligners assume that the distance between paired-end reads fit a known distribution, but in Hi-C data the insert size of the ligation product can vary between one base pair to hundreds of megabases (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapping-hi-c-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapping Hi-C reads</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy0" title="BWA-MEM2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>BWA-MEM2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
          </ul>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>“BAM sorting mode”</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM forward</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy0" title="BWA-MEM2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>BWA-MEM2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
          </ul>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_R</code></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>“BAM sorting mode”</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM reverse</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0" title="Filter and merge tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter and merge</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0+galaxy0)</span> chimeric reads from Arima Genomics with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First set of reads”</em>: <code class="language-plaintext highlighter-rouge">BAM forward</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second set of reads”</em>: <code class="language-plaintext highlighter-rouge">BAM  reverse</code></li>
      </ul>
    </li>
    <li>Rename it as <code class="language-plaintext highlighter-rouge">BAM Hi-C reads</code></li>
  </ol>
</blockquote>

<p>Finally, we need to convert the BAM file to BED format and sort it.</p>

<h2 id="generate-initial-hi-c-contact-map">Generate initial Hi-C contact map</h2>

<p>After mapping the Hi-C reads, the next step is to generate an initial Hi-C contact map, which will allow us to compare the Hi-C contact maps before and after using the Hi-C for scaffolding.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-biological-basis-of-hi-c-contact-maps"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Biological basis of Hi-C contact maps</div>

  <p>Hi-C contact maps reflect the interaction frequency between genomic loci. In order to understand the Hi-C contact maps, it is necessary to take into account two factors: the higher interaction frequency between loci that reside in the same chromosome (<em>i.e.</em>, in cis), and the distance-dependent decay of interaction frequency (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

  <p>The higher interaction between cis regions can be explained, at least in part, by the territorial organization of chromosomes in interphase (chromosome territories), and in a genome-wide contact map, this pattern appears as blocks of high interaction centered along the diagonal and matching individual chromosomes (fig. 12) (<span class="citation"><a href="#Cremer2010">Cremer and Cremer 2010</a></span>, <span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

  <figure id="figure-25" style="max-width: 90%;" width="10%"><img src="../../images/vgp_assembly/hic_map.png" alt="Hi-C map. " width="800" height="642" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hic_map.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 25</strong>:</span> An example of a Hi-C map. Genomic regions are arranged along the x and y axes, and contacts are colored on the matrix like a heat map; here darker color indicates greater interaction frequency.</figcaption></figure>

  <p>On the other hand, the distance-dependent decay may be due to random movement of the chromosomes, and in the contact map appears as a gradual decrease of the interaction frequency the farther away from the diagonal it moves (<span class="citation"><a href="#Lajoie2015">Lajoie <i>et al.</i> 2015</a></span>).</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-generate-a-contact-map-with-b-pretextmap-b-and-b-pretext-snapshot-b"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Generate a contact map with <b>PretextMap</b> and <b>Pretext Snapshot</b></div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+galaxy0" title="PretextMap tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>PretextMap</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.1.6+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input dataset in SAM or BAM format”</em>: <code class="language-plaintext highlighter-rouge">BAM Hi-C reads</code></li>
        <li><em>“Sort by”</em>: <code class="language-plaintext highlighter-rouge">Don't sort</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">PretextMap output</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy0" title="Pretext Snapshot tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Pretext Snapshot</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input Pretext map file”</em>: <code class="language-plaintext highlighter-rouge">PretextMap output</code></li>
        <li><em>“Output image format”</em>: <code class="language-plaintext highlighter-rouge">png</code></li>
        <li><em>“Show grid?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Let’s have a look at the Hi-C contact maps generated by Pretext Snapshot.</p>

<figure id="figure-26" style="max-width: 90%;"><img src="../../images/vgp_assembly/hic_map_pretext.png" alt="Pretext optical map. " width="2310" height="1000" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hic_map_pretext.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 26</strong>:</span> Hi-C map generated by Pretext. Primary assembly full contact map generated in this training (a) Hi-C map representative of a typical missasembly (b).</figcaption></figure>

<p>In the contact generated from the Bionano-scaffolded assembly can be identified 17 scaffolds, representing each of the haploid chromosomes of our genome (fig. 13.a). The fact that all the contact signals are found around the diagonal suggest that the contigs were scaffolded in the right order. However, during the assembly of complex genomes, it is common to find in the contact maps indicators of errors during the scaffolding process, as shown in the figure 13b. In that case, a contig belonging to the second chromosome has been misplaced as part of the fourth chromosome. We can also note that the final portion of the second chromosome should be placed at the beginning, as the off-diagonal contact signal suggests.</p>

<p>Once we have evaluated the quality of the scaffolded genome assembly, the next step consists in integrating the information contained in the HiC reads into our assembly, so that any errors identified can be resolved. For this purpose we will use SALSA2 (<span class="citation"><a href="#Ghurye2019">Ghurye <i>et al.</i> 2019</a></span>).</p>

<h2 id="salsa2-scaffolding">SALSA2 scaffolding</h2>

<p>SALSA2 is an open source software that makes use of Hi-C to linearly orient and order assembled contigs along entire chromosomes (<span class="citation"><a href="#Ghurye2019">Ghurye <i>et al.</i> 2019</a></span>). One of the advantages of SALSA2 with respect to most existing Hi-C scaffolding tools is that it doesn’t require the estimated number of chromosomes.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-salsa2-algorithm-overview"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: SALSA2 algorithm overview</div>

  <p>Initially SALSA2 uses the physical coverage of Hi-C pairs to identify suspicious regions and break the sequence at the likely point of mis-assembly. Then, a hybrid scaffold graph is constructed using edges from the Hi-C reads, scoring the edges according to a <em>best buddy</em> scheme (fig. 14a).</p>

  <figure id="figure-27" style="max-width: 90%;"><img src="../../images/vgp_assembly/salsa2_algorithm.png" alt="Figure 14: SALSA2 algorithm. " width="892" height="335" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/salsa2_algorithm.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 27</strong>:</span> Overview of the SALSA2 algorithm. Solid edges indicate the linkages between different contigs and dotted edges indicate the links between the ends of the same contig. B and E denote the start and end of contigs, respectively. Adapted from Ghurye et al. 2019.</figcaption></figure>

  <p>From this graph scaffolds are iteratively constructed using a greedy weighted maximum matching. After each iteration, a mis-join detection step is performed to check if any of the joins made during this round are incorrect. Incorrect joins are broken and the edges blacklisted during subsequent iterations. This process continues until the majority of joins made in the prior iteration are incorrect. This provides a natural stopping condition, when accurate Hi-C links have been exhausted (<span class="citation"><a href="#Ghurye2019">Ghurye <i>et al.</i> 2019</a></span>).</p>

</blockquote>

<p>Before launching SALSA2, we need to carry out some modifications on our datasets.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-bam-to-bed-conversion"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: BAM to BED conversion</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.30.0+galaxy1" title="bedtools BAM to BED tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>bedtools BAM to BED</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.30.0+galaxy1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Convert the following BAM file to BED”</em>: <code class="language-plaintext highlighter-rouge">BAM Hi-C reads</code></li>
        <li><em>“What type of BED output would you like”</em>: <code class="language-plaintext highlighter-rouge">Create a full, 12-column "blocked" BED file</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BED unsorted</code></p>
    </li>
    <li><span class="tool" data-tool="sort1" title="Sort tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Sort</strong></span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort Dataset”</em>: <code class="language-plaintext highlighter-rouge">BED unsorted</code></li>
        <li><em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 4</code></li>
        <li><em>“with flavor”</em>: <code class="language-plaintext highlighter-rouge">Alphabetical sort</code></li>
        <li><em>“everything in”</em>: <code class="language-plaintext highlighter-rouge">Ascending order</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BED sorted</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.1.3" title="Replace tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Replace</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.3)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to process”</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano</code></li>
        <li><em>“Find pattern”</em>: <code class="language-plaintext highlighter-rouge">:</code></li>
        <li><em>“Replace all occurrences of the pattern”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Find and Replace text in”</em>: <code class="language-plaintext highlighter-rouge">entire line</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Primary assembly bionano edited</code></li>
  </ol>
</blockquote>

<p>Now we can launch SALSA2 in order to generate the hybrid scaffolding based on the Hi-C data.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-salsa-scaffolding"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Salsa scaffolding</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/salsa/salsa/2.3+galaxy0" title="SALSA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SALSA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Initial assembly file”</em>: <code class="language-plaintext highlighter-rouge">Primary assembly bionano edited</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Bed alignment”</em>: <code class="language-plaintext highlighter-rouge">BED sorted</code></li>
        <li><em>“Restriction enzyme sequence(s)”</em>: <code class="language-plaintext highlighter-rouge">CTTAAG</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code> and <code class="language-plaintext highlighter-rouge">SALSA2 scaffold AGP</code></li>
  </ol>

</blockquote>

<h2 id="evaluate-the-final-genome-assembly-with-pretext">Evaluate the final genome assembly with Pretext</h2>

<p>Finally, we should repeat the procedure described previously for generating the contact maps, but in that case, we will use the scaffold generated by SALSA2.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-mapping-reads-against-the-scaffold"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Mapping reads against the scaffold</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy0" title="BWA-MEM2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>BWA-MEM2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code></li>
          </ul>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_F</code></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>“BAM sorting mode”</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM forward SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bwa_mem2/bwa_mem2/2.2.1+galaxy0" title="BWA-MEM2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>BWA-MEM2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.2.1+galaxy0)</span> with the following parameters:
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: <code class="language-plaintext highlighter-rouge">SALSA2 scaffold FASTA</code></li>
          </ul>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Single</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select fastq dataset”</em>: <code class="language-plaintext highlighter-rouge">Hi-C_dataset_R</code></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Do not set</code></li>
        <li><em>“Select analysis mode”</em>: <code class="language-plaintext highlighter-rouge">1.Simple Illumina mode</code></li>
        <li><em>“BAM sorting mode”</em>: <code class="language-plaintext highlighter-rouge">Sort by read names  (i.e., the QNAME field) </code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM reverse SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/bellerophon/bellerophon/1.0+galaxy0" title="Filter and merge tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter and merge</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.0+galaxy0)</span> chimeric reads from Arima Genomics with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“First set of reads”</em>: <code class="language-plaintext highlighter-rouge">BAM forward SALSA2</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Second set of reads”</em>: <code class="language-plaintext highlighter-rouge">BAM reverse SALSA2</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">BAM Hi-C reads SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_map/pretext_map/0.1.6+galaxy0" title="PretextMap tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>PretextMap</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.1.6+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input dataset in SAM or BAM format”</em>: <code class="language-plaintext highlighter-rouge">BAM Hi-C reads SALSA2</code></li>
        <li><em>“Sort by”</em>: <code class="language-plaintext highlighter-rouge">Don't sort</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">PretextMap output SALSA2</code></p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/pretext_snapshot/pretext_snapshot/0.0.3+galaxy0" title="Pretext Snapshot tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Pretext Snapshot</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.0.3+galaxy0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input Pretext map file”</em>: <code class="language-plaintext highlighter-rouge">PretextMap output SALSA2</code></li>
        <li><em>“Output image format”</em>: <code class="language-plaintext highlighter-rouge">png</code></li>
        <li><em>“Show grid?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>In order to evaluate the Hi-C hybrid scaffolding, we are going to compare the contact maps before and after running SALSA2 (fig. 15).</p>

<figure id="figure-28" style="max-width: 90%;"><img src="../../images/vgp_assembly/hi-c_pretext_final.png" alt="Figure 15: Pretext final contact map. " width="2310" height="1000" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hi-c_pretext_final.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 28</strong>:</span> Hi-C map generated by Pretext after the hybrid scaffolding based on Hi-C data. The red circles indicate the  differences between the contact map generated after (a) and before (b) Hi-C hybrid scaffolding.</figcaption></figure>

<p>Among the most notable differences that can be identified between the contact maps, it can be highlighted the regions marked with red circles, where inversion can be identified.</p>

<h1 id="conclusion">Conclusion</h1>

<p>To sum up, it is worthwhile to compare the final assembly with the <a href="https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/146/045/GCF_000146045.2_R64/GCF_000146045.2_R64_assembly_stats.txt"><em>S. cerevisiae</em> S288C reference genome</a>.</p>

<figure id="figure-29" style="max-width: 90%;"><img src="../../images/vgp_assembly/stats_conclusion.png" alt="Table 1: Final stats. " width="1197" height="409" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/stats_conclusion.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 29</strong>:</span> Comparison between the final assembly generating in this training and the reference genome. Contiguity plot using the reference genome size (a). Assemby statistics (b).</figcaption></figure>

<p>With respect to the total sequence length, we can conclude that the size of our genome assembly is almost identical to the reference genome (fig.16a,b). Regarding the number of scaffolds, the obtained value is similar to the reference assembly, which consist in 16 chromosomes plus the mitochondrial DNA, which consists of 85,779 bp. The remaining statistics exhibit very similar values (fig. 16b).</p>

<figure id="figure-30" style="max-width: 90%;"><img src="../../images/vgp_assembly/hi-c_pretext_conclusion.png" alt="Comparison reference genome. " width="2250" height="1000" loading="lazy" /><a target="_blank" href="../../images/vgp_assembly/hi-c_pretext_conclusion.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 30</strong>:</span> Comparison between contact maps generated by using the final assembly (a) and the reference genome (b).</figcaption></figure>

<p>If we compare the contact map of our assembled genome (fig. 17a) with the reference assembly (fig. 17b), we can see that the two are essentially identical. This means that we have achieved an almost perfect assembly at the chromosome level.</p>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>The VGP pipeline allows users to generate error-free, near gapless reference-quality genome assemblies</p>
</li>
                        
                        <li><p>The assembly can be divided in four main stages: genome profile analysis, HiFi long read phased assembly with hifiasm, Bionano hybrid scaffolding and Hi-C scaffolding</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/assembly/faqs/">FAQ page for the Assembly topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Marcotte1999">Marcotte, E. M., M. Pellegrini, T. O. Yeates, and D. Eisenberg, 1999 <b>A census of protein repeats</b>. Journal of Molecular Biology 293: 151–160. <a href="https://doi.org/10.1006/jmbi.1999.3136">10.1006/jmbi.1999.3136</a></li>
<li id="Small2007">Small, K. S., M. Brudno, M. M. Hill, and A. Sidow, 2007 <b>A haplome alignment and reference sequence of the highly polymorphic Ciona savignyi genome</b>. Genome Biology 8: R41. <a href="https://doi.org/10.1186/gb-2007-8-3-r41">10.1186/gb-2007-8-3-r41</a></li>
<li id="LiebermanAiden2009">Lieberman-Aiden, E., N. L. van Berkum, L. Williams, M. Imakaev, T. Ragoczy <i>et al.</i>, 2009 <b>Comprehensive Mapping of Long-Range Interactions Reveals Folding Principles of the Human Genome</b>. Science 326: 289–293. <a href="https://doi.org/10.1126/science.1181369">10.1126/science.1181369</a></li>
<li id="Cremer2010">Cremer, T., and M. Cremer, 2010 <b>Chromosome Territories</b>. Cold Spring Harbor Perspectives in Biology 2: a003889–a003889. <a href="https://doi.org/10.1101/cshperspect.a003889">10.1101/cshperspect.a003889</a></li>
<li id="Dixon2012">Dixon, J. R., S. Selvaraj, F. Yue, A. Kim, Y. Li <i>et al.</i>, 2012 <b>Topological domains in mammalian genomes identified by analysis of chromatin interactions</b>. Nature 485: 376–380. <a href="https://doi.org/10.1038/nature11082">10.1038/nature11082</a></li>
<li id="Frenkel2012">Frenkel, S., V. Kirzhner, and A. Korol, 2012 <b>Organizational Heterogeneity of Vertebrate Genomes</b> (V. Laudet, Ed.). PLoS ONE 7: e32076. <a href="https://doi.org/10.1371/journal.pone.0032076">10.1371/journal.pone.0032076</a></li>
<li id="Lam2012">Lam, E. T., A. Hastie, C. Lin, D. Ehrlich, S. K. Das <i>et al.</i>, 2012 <b>Genome mapping on nanochannel arrays for structural variation analysis and sequence assembly</b>. Nature Biotechnology 30: 771–776. <a href="https://doi.org/10.1038/nbt.2303">10.1038/nbt.2303</a></li>
<li id="Korbel2013">Korbel, J. O., and C. Lee, 2013 <b>Genome assembly and haplotyping with Hi-C</b>. Nature Biotechnology 31: 1099–1101. <a href="https://doi.org/10.1038/nbt.2764">10.1038/nbt.2764</a></li>
<li id="Rao2014">Rao, S. S. P., M. H. Huntley, N. C. Durand, E. K. Stamenova, I. D. Bochkov <i>et al.</i>, 2014 <b>A 3D Map of the Human Genome at Kilobase Resolution Reveals Principles of Chromatin Looping</b>. Cell 159: 1665–1680. <a href="https://doi.org/10.1016/j.cell.2014.11.021">10.1016/j.cell.2014.11.021</a></li>
<li id="Chalopin2015">Chalopin, D., M. Naville, F. Plard, D. Galiana, and J.-N. Volff, 2015 <b>Comparative Analysis of Transposable Elements Highlights Mobilome Diversity and Evolution in Vertebrates</b>. Genome Biology and Evolution 7: 567–580. <a href="https://doi.org/10.1093/gbe/evv005">10.1093/gbe/evv005</a></li>
<li id="Lajoie2015">Lajoie, B. R., J. Dekker, and N. Kaplan, 2015 <b>The Hitchhiker’s guide to Hi-C analysis: Practical guidelines</b>. Methods 72: 65–75. <a href="https://doi.org/10.1016/j.ymeth.2014.10.031">10.1016/j.ymeth.2014.10.031</a></li>
<li id="Simo2015">Simão, F. A., R. M. Waterhouse, P. Ioannidis, E. V. Kriventseva, and E. M. Zdobnov, 2015 <b>BUSCO: assessing genome assembly and annotation completeness with single-copy orthologs</b>. Bioinformatics 31: 3210–3212. <a href="https://doi.org/10.1093/bioinformatics/btv351">10.1093/bioinformatics/btv351</a></li>
<li id="SoteroCaio2017">Sotero-Caio, C. G., R. N. Platt, A. Suh, and D. A. Ray, 2017 <b>Evolution and Diversity of Transposable Elements in Vertebrate Genomes</b>. Genome Biology and Evolution 9: 161–177. <a href="https://doi.org/10.1093/gbe/evw264">10.1093/gbe/evw264</a></li>
<li id="DominguezDelAngel2018">Angel, V. D. D., E. Hjerde, L. Sterck, S. Capella-Gutierrez, C. Notredame <i>et al.</i>, 2018 <b>Ten steps to get started in Genome Assembly and Annotation</b>. F1000Research 7: 148. <a href="https://doi.org/10.12688/f1000research.13598.1">10.12688/f1000research.13598.1</a></li>
<li id="Li2018">Li, H., 2018 <b>Minimap2: pairwise alignment for nucleotide sequences</b> (I. Birol, Ed.). Bioinformatics 34: 3094–3100. <a href="https://doi.org/10.1093/bioinformatics/bty191">10.1093/bioinformatics/bty191</a></li>
<li id="Roach2018">Roach, M. J., S. A. Schmidt, and A. R. Borneman, 2018 <b>Purge Haplotigs: allelic contig reassignment for third-gen diploid genome assemblies</b>. BMC Bioinformatics 19: <a href="https://doi.org/10.1186/s12859-018-2485-7">10.1186/s12859-018-2485-7</a></li>
<li id="Ghurye2019">Ghurye, J., A. Rhie, B. P. Walenz, A. Schmitt, S. Selvaraj <i>et al.</i>, 2019 <b>Integrating Hi-C links with assembly graphs for chromosome-scale assembly</b> (I. Ioshikhes, Ed.). PLOS Computational Biology 15: e1007273. <a href="https://doi.org/10.1371/journal.pcbi.1007273">10.1371/journal.pcbi.1007273</a></li>
<li id="Guan2019">Guan, D., S. A. McCarthy, J. Wood, K. Howe, Y. Wang <i>et al.</i>, 2019 <b>Identifying and removing haplotypic duplication in primary genome assemblies</b>. <a href="https://doi.org/10.1101/729962">10.1101/729962</a></li>
<li id="Trresen2019">Tørresen, O. K., B. Star, P. Mier, M. A. Andrade-Navarro, A. Bateman <i>et al.</i>, 2019 <b>Tandem repeats lead to sequence assembly errors and impose multi-level challenges for genome and protein databases</b>. Nucleic Acids Research 47: 10994–11006. <a href="https://doi.org/10.1093/nar/gkz841">10.1093/nar/gkz841</a></li>
<li id="Wenger2019">Wenger, A. M., P. Peluso, W. J. Rowell, P.-C. Chang, R. J. Hall <i>et al.</i>, 2019 <b>Accurate circular consensus long-read sequencing improves variant detection and assembly of a human genome</b>. Nature Biotechnology 37: 1155–1162. <a href="https://doi.org/10.1038/s41587-019-0217-9">10.1038/s41587-019-0217-9</a></li>
<li id="Giani2020">Giani, A. M., G. R. Gallo, L. Gianfranceschi, and G. Formenti, 2020 <b>Long walk to genomics: History and current approaches to genome sequencing and assembly</b>. Computational and Structural Biotechnology Journal 18: 9–19. <a href="https://doi.org/10.1016/j.csbj.2019.11.002">10.1016/j.csbj.2019.11.002</a></li>
<li id="Hon2020">Hon, T., K. Mars, G. Young, Y.-C. Tsai, J. W. Karalius <i>et al.</i>, 2020 <b>Highly accurate long-read HiFi sequencing data for five complex genomes</b>. Scientific Data 7: <a href="https://doi.org/10.1038/s41597-020-00743-4">10.1038/s41597-020-00743-4</a></li>
<li id="RanalloBenavidez2020">Ranallo-Benavidez, T. R., K. S. Jaron, and M. C. Schatz, 2020 <b>GenomeScope 2.0 and Smudgeplot for reference-free profiling of polyploid genomes</b>. Nature Communications 11: <a href="https://doi.org/10.1038/s41467-020-14998-3">10.1038/s41467-020-14998-3</a></li>
<li id="Rhie2021">Rhie, A., B. P. Walenz, S. Koren, and A. M. Phillippy, 2020 <b>Merqury: reference-free quality,  completeness,  and phasing assessment for genome assemblies</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-02134-9">10.1186/s13059-020-02134-9</a></li>
<li id="Rhie2020">Rhie, A., B. P. Walenz, S. Koren, and A. M. Phillippy, 2020 <b>Merqury: reference-free quality,  completeness,  and phasing assessment for genome assemblies</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-02134-9">10.1186/s13059-020-02134-9</a></li>
<li id="Rhie_merqury">Rhie, A., B. P. Walenz, S. Koren, and A. M. Phillippy, 2020 <b>Merqury: reference-free quality,  completeness,  and phasing assessment for genome assemblies</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-02134-9">10.1186/s13059-020-02134-9</a></li>
<li id="Rhie2022">Rhie, A., B. P. Walenz, S. Koren, and A. M. Phillippy, 2020 <b>Merqury: reference-free quality,  completeness,  and phasing assessment for genome assemblies</b>. Genome Biology 21: <a href="https://doi.org/10.1186/s13059-020-02134-9">10.1186/s13059-020-02134-9</a></li>
<li id="wang2020estimation">Wang, H., B. Liu, Y. Zhang, F. Jiang, Y. Ren <i>et al.</i>, 2020 <b>Estimation of genome size using k-mer frequencies from corrected long reads</b>. arXiv preprint arXiv:2003.11817. <a href="https://arxiv.org/abs/2003.11817">https://arxiv.org/abs/2003.11817</a></li>
<li id="Yuan2020">Yuan, Y., C. Y.-L. Chung, and T.-F. Chan, 2020 <b>Advances in optical mapping for genomic research</b>. Computational and Structural Biotechnology Journal 18: 2051–2062. <a href="https://doi.org/10.1016/j.csbj.2020.07.018">10.1016/j.csbj.2020.07.018</a></li>
<li id="Zhang2020">Zhang, X., R. Wu, Y. Wang, J. Yu, and H. Tang, 2020 <b>Unzipping haplotypes in diploid and polyploid genomes</b>. Computational and Structural Biotechnology Journal 18: 66–72. <a href="https://doi.org/10.1016/j.csbj.2019.11.011">10.1016/j.csbj.2019.11.011</a></li>
<li id="Cheng2021">Cheng, H., G. T. Concepcion, X. Feng, H. Zhang, and H. Li, 2021 <b>Haplotype-resolved de novo assembly using phased assembly graphs with hifiasm</b>. Nature Methods 18: 170–175. <a href="https://doi.org/10.1038/s41592-020-01056-5">10.1038/s41592-020-01056-5</a></li>
<li id="Dida2021">Dida, F., and G. Yi, 2021 <b>Empirical evaluation of methods for de novo genome assembly</b>. PeerJ Computer Science 7: e636. <a href="https://doi.org/10.7717/peerj-cs.636">10.7717/peerj-cs.636</a></li>
<li id="Savara2021">Savara, J., T. Novosád, P. Gajdoš, and E. Kriegová, 2021 <b>Comparison of structural variants detected by optical mapping with long-read next-generation sequencing</b> (C. Kendziorski, Ed.). Bioinformatics 37: 3398–3404. <a href="https://doi.org/10.1093/bioinformatics/btab359">10.1093/bioinformatics/btab359</a></li>
<li id="Formenti2022">Formenti, G., L. Abueg, A. Brajuka, N. Brajuka, C. Gallardo-Alba <i>et al.</i>, 2022 <b>Gfastats: conversion,  evaluation and manipulation of genome sequences using assembly graphs</b> (A. Valencia, Ed.). Bioinformatics 38: 4214–4216. <a href="https://doi.org/10.1093/bioinformatics/btac460">10.1093/bioinformatics/btac460</a></li>
<li id="Nurk2022">Nurk, S., S. Koren, A. Rhie, M. Rautiainen, A. V. Bzikadze <i>et al.</i>, 2022 <b>The complete sequence of a human genome</b>. Science 376: 44–53. <a href="https://doi.org/10.1126/science.abj6987">10.1126/science.abj6987</a></li>
<li id="Rahman2022">Rahman, A., and P. Medvedev, 2022 <b>Assembler artifacts include misassembly because of unsafe unitigs and underassembly because of bidirected graphs</b>. Genome Research 32: 1746–1753. <a href="https://doi.org/10.1101/gr.276601.122">10.1101/gr.276601.122</a></li></ol>
                

                
                <h1 data-toc-skip>Glossary</h1>
                <dl>
                    
                    
                    <dt>ASM</dt>
                    <dd>assembly</dd>
                    
                    <dt>BUSCO</dt>
                    <dd>Benchmarking Universal Single-Copy Orthologs</dd>
                    
                    <dt>CN</dt>
                    <dd>copy number</dd>
                    
                    <dt>G10K</dt>
                    <dd>Genome 10K consortium</dd>
                    
                    <dt>GFA</dt>
                    <dd>Graphical Fragment Assembly</dd>
                    
                    <dt>Hi-C</dt>
                    <dd>all-versus-all chromatin conformation capture</dd>
                    
                    <dt>HiFi</dt>
                    <dd>high fidelity</dd>
                    
                    <dt>NGS</dt>
                    <dd>next generation sequencing</dd>
                    
                    <dt>QC</dt>
                    <dd>Quality Control</dd>
                    
                    <dt>QV</dt>
                    <dd>consensus accuracy quality value</dd>
                    
                    <dt>SMRT</dt>
                    <dd>single-molecule real-time</dd>
                    
                    <dt>TGS</dt>
                    <dd>third-generation sequencing</dd>
                    
                    <dt>VGP</dt>
                    <dd>Vertebrate Genomes Project</dd>
                    
                </dl>
                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=VGP assembly pipeline (Assembly)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Delphine Lariviere, Alex Ostrovsky, Cristóbal Gallardo, Anna Syme, Linelle Abueg, Brandon Pickett, Giulio Formenti, Marcella Sozzoni, Anton Nekrutenko,  <b>VGP assembly pipeline (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html">https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{assembly-vgp_genome_assembly,
author = "Delphine Lariviere and Alex Ostrovsky and Cristóbal Gallardo and Anna Syme and Linelle Abueg and Brandon Pickett and Giulio Formenti and Marcella Sozzoni and Anton Nekrutenko",
	title = "VGP assembly pipeline (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/assembly/tutorials/vgp_genome_assembly/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/assembly/tutorials/vgp_genome_assembly/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: bionano_scaffold
  owner: bgruening
  revisions: 96cacb31d571
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bionano_scaffold
  owner: bgruening
  revisions: 5258e18bbe23
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gfastats
  owner: bgruening
  revisions: 3ef480892a9f
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: hifiasm
  owner: bgruening
  revisions: da9d8bf98802
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: f46f0e4f75c4
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: d698c222f354
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: column_maker
  owner: devteam
  revisions: 02026300aa45
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: column_maker
  owner: devteam
  revisions: 6595517c2dd8
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bedtools
  owner: iuc
  revisions: a68aa6c1204a
  tool_panel_section_label: BED
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bedtools
  owner: iuc
  revisions: a1a923cd89e8
  tool_panel_section_label: BED
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bellerophon
  owner: iuc
  revisions: 25ca5d73aedf
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: busco
  owner: iuc
  revisions: cf13a1e03e5b
  tool_panel_section_label: Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: busco
  owner: iuc
  revisions: 46ae58b1d792
  tool_panel_section_label: Annotation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bwa_mem2
  owner: iuc
  revisions: b4a22d90cce9
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collection_column_join
  owner: iuc
  revisions: 3ddd99c7efee
  tool_panel_section_label: Join, Subtract and Group
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: genomescope
  owner: iuc
  revisions: 3169a38c2656
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: genomescope
  owner: iuc
  revisions: 01210c4e9144
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gfa_to_fa
  owner: iuc
  revisions: e33c82b63727
  tool_panel_section_label: Convert Formats
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: merqury
  owner: iuc
  revisions: 9d79beb19ac3
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: merqury
  owner: iuc
  revisions: 39edec572bae
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: meryl
  owner: iuc
  revisions: eadfd71dde37
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: meryl
  owner: iuc
  revisions: 29dabd8db6f2
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: minimap2
  owner: iuc
  revisions: f54f5baedfdd
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: minimap2
  owner: iuc
  revisions: 11a0d50a54e6
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pretext_map
  owner: iuc
  revisions: '02831bf9e95f'
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pretext_map
  owner: iuc
  revisions: dfb8a4497339
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pretext_snapshot
  owner: iuc
  revisions: 7c9417381ad5
  tool_panel_section_label: Graph/Display Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: pretext_snapshot
  owner: iuc
  revisions: 44c66e8d21e6
  tool_panel_section_label: Graph/Display Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: purge_dups
  owner: iuc
  revisions: 17b378303f2d
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: purge_dups
  owner: iuc
  revisions: 76d4cbefff85
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: purge_dups
  owner: iuc
  revisions: a315c25dc813
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: quast
  owner: iuc
  revisions: 875d0f36d66f
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: salsa
  owner: iuc
  revisions: ba7b0f11e470
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: salsa
  owner: iuc
  revisions: f77f7a7f3b83
  tool_panel_section_label: Assembly
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: cutadapt
  owner: lparsons
  revisions: 48f587c13075
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collapse_collections
  owner: nml
  revisions: 830961c48e42
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: collapse_collections
  owner: nml
  revisions: 90981f86000f
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/5c4f29500474814d08883ef5f8fb236fe4a6c378">5c4f295</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00039">https://gxy.io/GTN:T00039</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/assembly/tutorials/vgp_genome_assembly/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


        <script src='/training-material/assets/js/bundle.03aefefa.js?v=1702253434'></script>
        
    </body>
</html>