<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Hands-on: Generic plugins / Development in Galaxy</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@fc3f11ce65e906601f63521f6553699c960500e1",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/dev/tutorials/visualization-generic/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.f1f2de89.js?v=1705578603' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.03aefefa.js?v=1705578603' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Hands-on: Hands-on: Generic plugins">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-03 14:30:27 +0000">
<meta name="DC.creator" content="Saskia Hiltemann">
<meta name="DC.creator" content="Youri Hoogstrate"><meta name="description" content="Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Development in Galaxy / Hands-on: Generic plugins">
        <meta property="og:description" content="Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Galaxy Administrators"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "dateModified": "2023-11-03 14:30:27 +0000",
  "datePublished": "2016-09-30 10:05:09 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Hands-on: Hands-on: Generic plugins",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "identifier": "https://gxy.io/GTN:T00121",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Development in Galaxy",
    "description": "Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...",
    "url": "https://training.galaxyproject.org/training-material/topics/dev/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on: Hands-on: Generic plugins",
  "url": "https://training.galaxyproject.org/training-material/topics/dev/tutorials/visualization-generic/tutorial.html",
  "version": 26,
  "timeRequired": "PT90M",
  "teaches": "Implement a first Galaxy visualization\n - Understand the client side vs. server side principle",
  "keywords": [
    "Development in Galaxy"
  ],
  "description": "The questions this  addresses are:\n - How can visualization plugins benefit science?\n\n\nThe objectives are:\n - Implement a first Galaxy visualization\n - Understand the client side vs. server side principle\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    "Javascript knowledge"
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "name": "Saskia Hiltemann",
      "image": "https://avatars.githubusercontent.com/shiltemann",
      "description": "Researcher at Erasmus Medical Center",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3803-468X",
      "orcid": "https://orcid.org/0000-0003-3803-468X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/yhoogstrate/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/yhoogstrate/",
      "name": "Youri Hoogstrate",
      "image": "https://avatars.githubusercontent.com/yhoogstrate",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Development in Galaxy",
      "description": "Galaxy is an open-source project. Everyone can contribute to its development with core Galaxy development, integration of softwares in Galaxy environment, ...",
      "url": "https://training.galaxyproject.org/training-material/topics/dev/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3372",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3372",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3372"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [

  ],
  "abstract": "Visualizations may be very helpful in understanding data better. There is a whole"
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.f1f2de89.js?v=1705578603'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/dev" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Development in Galaxy
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/dev/faqs/" title="Check our FAQs for the Development in Galaxy topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/dev/tutorials/visualization-generic/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/dev/tutorials/visualization-generic/tutorial.html">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/smorgasbord2023/topics/dev/tutorials/visualization-generic/tutorial.html">smorgasbord2023</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2024-01-01/topics/dev/tutorials/visualization-generic/tutorial.html">2024-01-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-12-01/topics/dev/tutorials/visualization-generic/tutorial.html">2023-12-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-dev">
    <h1 data-toc-skip>Generic plugins</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Saskia Hiltemann avatar" width="36" class="avatar" />
    Saskia Hiltemann</a><a href="/training-material/hall-of-fame/yhoogstrate/" class="contributor-badge contributor-yhoogstrate"><img src="https://avatars.githubusercontent.com/yhoogstrate?s=36" alt="Youri Hoogstrate avatar" width="36" class="avatar" />
    Youri Hoogstrate</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can visualization plugins benefit science?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Implement a first Galaxy visualization</p>
</li>
        
        <li><p>Understand the client side vs. server side principle</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
        
    <li>
    
        Javascript knowledge
    
    </li>

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 90 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/dev/tutorials/visualization-generic/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/dev/tutorials/visualization-generic/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Sep 30, 2016 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 3, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00121">https://gxy.io/GTN:T00121</a> </div>
        

	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 26 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Visualizations may be very helpful in understanding data better. There is a whole
range of visualizations, from rather simple scatter and barplots up to projections
of high dimensional data or even entire genomes. Many of these visualizations often
require a lot of tweaking and changes in settings like zooming in and assigning colors, etc.
Therefore, visualizations are ideally interactive, and changing settings is often
an initial step in exploring data. For this reason it may be inconvenient to make use
of static galaxy tools because it lacks these interactive features. For these situations Galaxy
offers the option to create <em>visualizations plugins</em>, file format specific javascripts
that integrate with the history menu, without making redundant copies of data.</p>

<p>In this tutorial we shall go through how this system works and create a simple visualization
plugin. The tool will create a visualization of the number of aligned reads per
chromosome of a BAM file, and we will discuss possible optimizations and advantages
and disadvantages of the proposed implementation.</p>

<p>If you want to make visualizations ready for production, it is essential to have a good
understanding of HTML5 and JavaScript as these are the basic languages in which they are written.
However, for this tutorial we will keep it basic.</p>

<p>Additional documentation about Galaxy visualizations can be found here:</p>

<ul>
  <li><a href="https://galaxyproject.org/data-providers">DataProviders</a></li>
  <li><a href="https://galaxyproject.org/data-providers/cookbook">DataProviders/Cookbook</a></li>
</ul>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#part-1" id="markdown-toc-part-1">Part 1</a>    <ol>
      <li><a href="#linking-the-plugin-with-galaxy" id="markdown-toc-linking-the-plugin-with-galaxy">Linking the plugin with Galaxy</a></li>
      <li><a href="#creating-the-visualization" id="markdown-toc-creating-the-visualization">Creating the visualization</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="part-1">Part 1</h1>

<p>The visualization we are going to create in this tutorial, is a tool that shows the number of aligned
reads per chromosome of a BAM file. The first thing we need to do is to come up with a name.
Let’s call it <em>alignment_rname_boxplot</em>. Note that the reference sequences (usually chromosomes)
to which we align are named <code class="language-plaintext highlighter-rouge">RNAME</code> in the BAM/SAM specification.</p>

<p>The development of a Galaxy visualization takes place within the Galaxy codebase.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Clone an instance of Galaxy in a path, further referred to as <code class="language-plaintext highlighter-rouge">$GALAXY_ROOT</code></li>
    <li>
      <p>Explore the plugin directory as follows:</p>

      <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> <span class="nv">$GALAXY_ROOT</span>/config/plugins/visualizations
</code></pre></div>      </div>
    </li>
    <li>
      <p>Create a new directory for our new plugin project</p>

      <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mkdir </span>alignment_rname_boxplot
<span class="nv">$ </span><span class="nb">cd </span>alignment_rname_boxplot
</code></pre></div>      </div>
    </li>
    <li>
      <p>Make three (sub-)directories to complete the structure of the project:</p>

      <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">mkdir </span>config
<span class="nv">$ </span><span class="nb">mkdir </span>static
<span class="nv">$ </span><span class="nb">mkdir </span>templates
</code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<h2 id="linking-the-plugin-with-galaxy">Linking the plugin with Galaxy</h2>

<p>To create a bridge between our not-yet-written plugin and Galaxy, we need to write a
configuration in XML format.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <p>Create the file  <code class="language-plaintext highlighter-rouge">config/alignment_rname_boxplot.xml</code> with the following contents:</p>

  <div class="language-xml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cp">&lt;?xml version="1.0" encoding="UTF-8"?&gt;</span>
<span class="cp">&lt;!DOCTYPE visualization SYSTEM "../../visualization.dtd"&gt;</span>
<span class="nt">&lt;visualization</span> <span class="na">name=</span><span class="s">"alignment_rname_boxplot"</span><span class="nt">&gt;</span>
    <span class="nt">&lt;data_sources&gt;</span>
        <span class="nt">&lt;data_source&gt;</span>
            <span class="nt">&lt;model_class&gt;</span>HistoryDatasetAssociation<span class="nt">&lt;/model_class&gt;</span>

            <span class="nt">&lt;test</span> <span class="na">type=</span><span class="s">"isinstance"</span> <span class="na">test_attr=</span><span class="s">"datatype"</span> <span class="na">result_type=</span><span class="s">"datatype"</span><span class="nt">&gt;</span>binary.Bam<span class="nt">&lt;/test&gt;</span>
            <span class="nt">&lt;test</span> <span class="na">type=</span><span class="s">"isinstance"</span> <span class="na">test_attr=</span><span class="s">"datatype"</span> <span class="na">result_type=</span><span class="s">"datatype"</span><span class="nt">&gt;</span>tabular.Sam<span class="nt">&lt;/test&gt;</span>

            <span class="nt">&lt;to_param</span> <span class="na">param_attr=</span><span class="s">"id"</span><span class="nt">&gt;</span>dataset_id<span class="nt">&lt;/to_param&gt;</span>
        <span class="nt">&lt;/data_source&gt;</span>
    <span class="nt">&lt;/data_sources&gt;</span>
    <span class="nt">&lt;params&gt;</span>
        <span class="nt">&lt;param</span> <span class="na">type=</span><span class="s">"dataset"</span> <span class="na">var_name_in_template=</span><span class="s">"hda"</span> <span class="na">required=</span><span class="s">"true"</span><span class="nt">&gt;</span>dataset_id<span class="nt">&lt;/param&gt;</span>
    <span class="nt">&lt;/params&gt;</span>
    <span class="nt">&lt;template&gt;</span>alignment_rname_boxplot.mako<span class="nt">&lt;/template&gt;</span>
<span class="nt">&lt;/visualization&gt;</span>
</code></pre></div>  </div>

</blockquote>

<p>This configures the plugin’s name, which shall appear on pressing the visualization button in
the history menu. It also links the plugin to two file formats: BAM and SAM, which means that
for any history item of these file formats the plugin will automatically become available.</p>

<p>It also includes a reference to a mako template file (HTML + Python syntax), to be found in the
<code class="language-plaintext highlighter-rouge">templates</code> directory (we will create this file in the next section). The <code class="language-plaintext highlighter-rouge">var_name_in_template</code>
parameter is set to the value <code class="language-plaintext highlighter-rouge">hda</code>, which will be the name of the variable in the mako template
corresponding to the dataset to be visualized.</p>

<h2 id="creating-the-visualization">Creating the visualization</h2>

<p>We have linked our visualization to a mako file (which we have not yet created). This file is a
blueprint for the visualization. This means that for every invocation of the visualization, the
mako file will be compiled to render an HTML file.</p>

<p>Beause we would like the visualization to load quickly, computationally intensive tasks should
not be done prior to loading. A bit of server-side rendering in itself is not a problem, but the
visualizations (written in HTML and/or JS) should do most of the actual calculations and
conversions on the client side (in the browser). Therefore, unlike regular Galaxy tools, parsing
files does not take place on the server, but instead data will be downloaded by the client via an
exposed Galaxy URL prior to client-side rendering.</p>

<p>The most basic part of the mako file are the variables used for further web development, given
below.</p>

<div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cp">&lt;!DOCTYPE HTML&gt;</span>
<span class="nt">&lt;</span><span class="err">%</span>
    <span class="na">import</span> <span class="na">os</span>

    <span class="na">##</span> <span class="na">Generates</span> <span class="na">hash</span> <span class="err">(</span><span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'])</span> <span class="na">of</span> <span class="na">history</span> <span class="na">item</span>
    <span class="na">hdadict = </span><span class="s">trans.security.encode_dict_ids(</span> <span class="na">hda.to_dict</span><span class="err">()</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Finds</span> <span class="na">the</span> <span class="na">parent</span> <span class="na">directory</span> <span class="na">of</span> <span class="na">galaxy</span> <span class="err">(/,</span> <span class="err">/</span><span class="na">galaxy</span><span class="err">,</span> <span class="na">etc.</span><span class="err">)</span>
    <span class="na">root     = </span><span class="s">h.url_for(</span> <span class="err">'/'</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Determines</span> <span class="na">the</span> <span class="na">exposed</span> <span class="na">URL</span> <span class="na">of</span> <span class="na">the</span> <span class="err">./</span><span class="na">static</span> <span class="na">directory</span>
    <span class="na">app_root = </span><span class="s">root</span> <span class="err">+</span> <span class="err">'</span><span class="na">plugins</span><span class="err">/</span><span class="na">visualizations</span><span class="err">/'+</span><span class="na">visualization_name</span><span class="err">+'/</span><span class="na">static</span><span class="err">/'</span>

    <span class="na">##</span> <span class="na">Actual</span> <span class="na">file</span> <span class="na">URL:</span>
    <span class="na">file_url = </span><span class="s">os.path.join(root,</span> <span class="err">'</span><span class="na">datasets</span><span class="err">',</span> <span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'],</span> <span class="err">"</span><span class="na">display</span><span class="err">?</span><span class="na">to_ext=</span><span class="s">"+hda.ext)
%&gt;
</span></code></pre></div></div>

<ul>
  <li><code class="language-plaintext highlighter-rouge">hdadict</code> is a variable that contains a file identifier that has been encoded to it’s
 exposed uid.</li>
  <li><code class="language-plaintext highlighter-rouge">root</code> indicates location of Galaxy on the webserver (e.g. <code class="language-plaintext highlighter-rouge">/</code>, <code class="language-plaintext highlighter-rouge">/galaxy/</code>, <code class="language-plaintext highlighter-rouge">/galaxy-pub/</code>, etc).</li>
  <li><code class="language-plaintext highlighter-rouge">app_root</code> contains the exposed url of the static files for this visualization.</li>
  <li><code class="language-plaintext highlighter-rouge">file_url</code> contains the exposed url of the dataset selected (by user) for visualization.</li>
</ul>

<p>We could obtain the BAM file client-side by downloading the BAM/SAM file in its entirety via
<em>file_url</em>. However, BAM files can become quite large and it is usually not desired to transfer
such datasets over the network. In our case it is also rather inconvenient to parse the BAM file
with Javascript, just to count the number of reads.</p>

<p>Fortunately, BAM files have indices. These indices are brief summaries describing the
number of entries per chromosome, in order to be able to access the data contained in them more
quickly. In the mako template we can access a BAM index as <em>hda.metadata.bam_index</em>.
(Note that this is a file path on the server, not an exposed URL).</p>

<p>Samtools has a command named <code class="language-plaintext highlighter-rouge">idxstats</code> which is able to leverage this BAM index you. However,
since visualizations do not have dependency management, it is very tricky to let the mako template
do a system call to samtools. Fortunately, the Galaxy ecosystem ships with a built-in <code class="language-plaintext highlighter-rouge">pysam</code>
dependency, a library that can do any native samtools command within python.</p>

<p>The <code class="language-plaintext highlighter-rouge">*.metadata.bam_index</code> is a special kind of file in the Galaxy ecosystem. It is actually an
invisible file in Galaxy, linked to another history item, but does have a unique filename.
So, for the BAM file <code class="language-plaintext highlighter-rouge">./database/files/000/dataset_001.dat</code>, our BAI file (index) is <strong>not</strong>
<code class="language-plaintext highlighter-rouge">./database/files/000/dataset_001.dat.bai</code> but could be <code class="language-plaintext highlighter-rouge">./database/files/000/dataset_002.dat</code> or
<code class="language-plaintext highlighter-rouge">./database/files/000/dataset_003.dat</code>. We can create a symlink to this index file to ensure the
bam file and its index share the same prefix, as expected by samtools and pysam.</p>

<p>We can create this symlink as follows in our mako template:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Ensure BAI index is symlinked
</span><span class="n">bai_target</span> <span class="o">=</span> <span class="n">hda</span><span class="p">.</span><span class="n">file_name</span><span class="o">+</span><span class="sh">'</span><span class="s">.bai</span><span class="sh">'</span>
<span class="kn">import</span> <span class="n">os</span>

<span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="p">.</span><span class="n">path</span><span class="p">.</span><span class="nf">isfile</span><span class="p">(</span><span class="n">bai_target</span><span class="p">):</span>
   <span class="n">os</span><span class="p">.</span><span class="nf">symlink</span><span class="p">(</span><span class="n">hda</span><span class="p">.</span><span class="n">metadata</span><span class="p">.</span><span class="n">bam_index</span><span class="p">.</span><span class="n">file_name</span><span class="p">,</span> <span class="n">bai_target</span><span class="p">)</span>
</code></pre></div></div>

<p>Now the BAM file has a <code class="language-plaintext highlighter-rouge">.bai</code> file with the same prefix, and we can run the <code class="language-plaintext highlighter-rouge">idxstats</code>
as follows:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="c1">## Extract idxstats
</span><span class="kn">import</span> <span class="n">pysam</span>
<span class="n">bam_idxstats_data</span> <span class="o">=</span> <span class="n">pysam</span><span class="p">.</span><span class="nf">idxstats</span><span class="p">(</span><span class="n">hda</span><span class="p">.</span><span class="n">file_name</span><span class="p">)</span>
</code></pre></div></div>

<p>With the lines of python code above, the idxstats data is parsed into the RAM of python
during compilation on the server, but is not yet exported into the HTML page nor parsed by JS.
To do that, we add an HTML section to the end of the mako file.</p>

<div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;html&gt;</span>
    <span class="nt">&lt;head&gt;</span>
        <span class="nt">&lt;title&gt;</span>${hda.name | h} | ${visualization_name | h}<span class="nt">&lt;/title&gt;</span>
    <span class="nt">&lt;/head&gt;</span>
    <span class="nt">&lt;body&gt;</span>
        ${bam_idxstats_data | h}
    <span class="nt">&lt;/body&gt;</span>
<span class="nt">&lt;/html&gt;</span>
</code></pre></div></div>

<p>Here you see <code class="language-plaintext highlighter-rouge">${bam_idxstats_data | h}</code>, which prints the python variable into
the HTML page and also does HTML escaping by providing the ` | h`-flag (for security reasons).</p>

<p>Let’s put this all together.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>Create the mako file <code class="language-plaintext highlighter-rouge">templates/alignment_rname_boxplot.mako</code></li>
    <li>
      <p>Fill it with the following code:</p>

      <div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cp">&lt;!DOCTYPE HTML&gt;</span>
<span class="nt">&lt;</span><span class="err">%</span>
    <span class="na">import</span> <span class="na">os</span>

    <span class="na">##</span> <span class="na">Generates</span> <span class="na">hash</span> <span class="err">(</span><span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'])</span> <span class="na">of</span> <span class="na">history</span> <span class="na">item</span>
    <span class="na">hdadict = </span><span class="s">trans.security.encode_dict_ids(</span> <span class="na">hda.to_dict</span><span class="err">()</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Finds</span> <span class="na">the</span> <span class="na">parent</span> <span class="na">directory</span> <span class="na">of</span> <span class="na">galaxy</span> <span class="err">(/,</span> <span class="err">/</span><span class="na">galaxy</span><span class="err">,</span> <span class="na">etc.</span><span class="err">)</span>
    <span class="na">root     = </span><span class="s">h.url_for(</span> <span class="err">'/'</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Determines</span> <span class="na">the</span> <span class="na">exposed</span> <span class="na">URL</span> <span class="na">of</span> <span class="na">the</span> <span class="err">./</span><span class="na">static</span> <span class="na">directory</span>
    <span class="na">app_root = </span><span class="s">root</span> <span class="err">+</span> <span class="err">'</span><span class="na">plugins</span><span class="err">/</span><span class="na">visualizations</span><span class="err">/'+</span><span class="na">visualization_name</span><span class="err">+'/</span><span class="na">static</span><span class="err">/'</span>

    <span class="na">##</span> <span class="na">Actual</span> <span class="na">file</span> <span class="na">URL:</span>
    <span class="na">file_url = </span><span class="s">os.path.join(root,</span> <span class="err">'</span><span class="na">datasets</span><span class="err">',</span> <span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'],</span> <span class="err">"</span><span class="na">display</span><span class="err">?</span><span class="na">to_ext=</span><span class="s">"+hda.ext)

    ## Ensure BAI index is symlinked
    bai_target = hda.file_name+'.bai'

    if not os.path.isfile(bai_target):
        os.symlink(hda.metadata.bam_index.file_name, bai_target)

    ## Extract idxstats
    import pysam
    bam_idxstats_data = pysam.idxstats(hda.file_name)
%&gt;
&lt;html&gt;
    &lt;head&gt;
        &lt;title&gt;${hda.name | h} | ${visualization_name | h}&lt;/title&gt;
    &lt;/head&gt;
    &lt;body&gt;
        ${bam_idxstats_data | h}
    &lt;/body&gt;
&lt;/html&gt;
</span></code></pre></div>      </div>

      <p>We are now ready to test this very basic visualization, we just need a (small) BAM file for it.</p>
    </li>
    <li>Download <a href="https://zenodo.org/record/248730/files/tutorial.bam">the example BAM file</a></li>
    <li>
      <p>Go the galaxy root directory and start Galaxy:</p>

      <div class="language-bash highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nv">$ </span><span class="nb">cd</span> <span class="nv">$GALAXY_ROOT</span>
<span class="nv">$ </span>./run.sh
</code></pre></div>      </div>
    </li>
    <li>
      <p>Upload the example BAM file to your history</p>

      <p>If everything went well, our plugin has appeared as a visualization option for the dataset</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <p>You must be logged in to be able to use visualizations</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>All the visualization does at the moment, is show the contents of idxstats, compiled to HTML:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>['chrA\t5386\t2\t0\n', 'chrB\t5386\t4\t0\n', 'chrC\t5386\t1\t0\n', 'chrD\t5386\t6\t1\n',
'chrE\t5386\t3\t0\n', 'chrF\t5386\t2\t0\n', 'chrG\t5386\t1\t0\n', '*\t0\t0\t0\n']
</code></pre></div></div>

<p>It contains eight entries, one for each of our (made-up) chromosomes and one to <code class="language-plaintext highlighter-rouge">*</code>, which represents
the unmapped reads. Entries are tab delimited (<code class="language-plaintext highlighter-rouge">\t</code>) and for the <code class="language-plaintext highlighter-rouge">chrA</code> entry it indicates that the
length of the RNAME (chromosome) is 5386 bases and 2 reads are aligned to it.</p>

<p>To make the data a bit more usable for Javascript , we convert it into a simple dictionary
of the following syntax:</p>

<div class="language-python highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">{</span><span class="sh">'</span><span class="s">chrA</span><span class="sh">'</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span> <span class="sh">'</span><span class="s">chrB</span><span class="sh">'</span><span class="p">:</span><span class="mi">4</span><span class="p">,</span> <span class="sh">'</span><span class="s">chrC</span><span class="sh">'</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span> <span class="sh">'</span><span class="s">chrD</span><span class="sh">'</span><span class="p">:</span><span class="mi">6</span><span class="p">,</span> <span class="sh">'</span><span class="s">chrE</span><span class="sh">'</span><span class="p">:</span><span class="mi">3</span><span class="p">,</span> <span class="sh">'</span><span class="s">chrF</span><span class="sh">'</span><span class="p">:</span><span class="mi">2</span><span class="p">,</span> <span class="sh">'</span><span class="s">chrG</span><span class="sh">'</span><span class="p">:</span><span class="mi">1</span><span class="p">,</span> <span class="sh">'</span><span class="s">*</span><span class="sh">'</span><span class="p">:</span> <span class="mi">0</span><span class="p">}</span>
</code></pre></div></div>

<p>Although it is possible to do this in python we recommend doing this in JS.
The var dump provided by python/pysam is actually a valid syntax for Javascript too,
so getting the raw data into Javascript is rather easy:</p>

<div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;script&gt;</span>
    <span class="nx">bam_idxstats_data</span> <span class="o">=</span> <span class="nx">$</span><span class="p">{</span><span class="nx">bam_idxstats_data</span><span class="p">};</span>
<span class="nt">&lt;/script&gt;</span>
</code></pre></div></div>

<p>Converting the data is not the scope of the tutorial, so here we provide such a function:</p>

<div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="nt">&lt;script&gt;</span>
  <span class="nx">bam_idxstats_data</span> <span class="o">=</span> <span class="nx">$</span><span class="p">{</span><span class="nx">bam_idxstats_data</span><span class="p">};</span>
  <span class="kd">function</span> <span class="nf">parse_data</span><span class="p">(</span><span class="nx">bam_idxstats_data</span><span class="p">)</span> <span class="p">{</span>
    <span class="kd">var</span> <span class="nx">output</span> <span class="o">=</span> <span class="p">{};</span>

    <span class="k">for</span><span class="p">(</span><span class="kd">var</span> <span class="nx">i</span> <span class="o">=</span> <span class="mi">0</span><span class="p">;</span> <span class="nx">i</span> <span class="o">&lt;</span> <span class="nx">data</span><span class="p">.</span><span class="nx">length</span> <span class="p">;</span> <span class="nx">i</span><span class="o">++</span><span class="p">)</span> <span class="p">{</span>
      <span class="kd">var</span> <span class="nx">line</span> <span class="o">=</span> <span class="nx">data</span><span class="p">[</span><span class="nx">i</span><span class="p">];</span>
      <span class="kd">var</span> <span class="nx">chunks</span> <span class="o">=</span> <span class="nx">line</span><span class="p">.</span><span class="nf">split</span><span class="p">(</span><span class="dl">"</span><span class="se">\t</span><span class="dl">"</span><span class="p">);</span>

      <span class="k">if</span><span class="p">(</span><span class="nx">chunks</span><span class="p">[</span><span class="mi">0</span><span class="p">].</span><span class="nf">split</span><span class="p">(</span><span class="dl">"</span><span class="s2">_</span><span class="dl">"</span><span class="p">).</span><span class="nx">length</span> <span class="o">==</span> <span class="mi">1</span><span class="p">)</span> <span class="p">{</span> <span class="c1">// only if it does not contain underscore</span>
        <span class="nx">output</span><span class="p">[</span><span class="nx">chunks</span><span class="p">[</span><span class="mi">0</span><span class="p">]]</span> <span class="o">=</span> <span class="nf">parseInt</span><span class="p">(</span><span class="nx">chunks</span><span class="p">[</span><span class="mi">2</span><span class="p">]);</span>
      <span class="p">}</span>
    <span class="p">}</span>

   <span class="k">return</span> <span class="nx">output</span><span class="p">;</span>
  <span class="p">}</span>
  <span class="nt">&lt;/script&gt;</span>
</code></pre></div></div>

<p>The great thing about the mako system is that it does not require to restart galaxy in order to make
functional changes to the mako files.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-3"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Change the mako file to the following:</p>

      <div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cp">&lt;!DOCTYPE HTML&gt;</span>
<span class="nt">&lt;</span><span class="err">%</span>
    <span class="na">import</span> <span class="na">os</span>

    <span class="na">##</span> <span class="na">Generates</span> <span class="na">hash</span> <span class="err">(</span><span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'])</span> <span class="na">of</span> <span class="na">history</span> <span class="na">item</span>
    <span class="na">hdadict = </span><span class="s">trans.security.encode_dict_ids(</span> <span class="na">hda.to_dict</span><span class="err">()</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Finds</span> <span class="na">the</span> <span class="na">parent</span> <span class="na">directory</span> <span class="na">of</span> <span class="na">galaxy</span> <span class="err">(/,</span> <span class="err">/</span><span class="na">galaxy</span><span class="err">,</span> <span class="na">etc.</span><span class="err">)</span>
    <span class="na">root     = </span><span class="s">h.url_for(</span> <span class="err">'/'</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Determines</span> <span class="na">the</span> <span class="na">exposed</span> <span class="na">URL</span> <span class="na">of</span> <span class="na">the</span> <span class="err">./</span><span class="na">static</span> <span class="na">directory</span>
    <span class="na">app_root = </span><span class="s">root</span> <span class="err">+</span> <span class="err">'</span><span class="na">plugins</span><span class="err">/</span><span class="na">visualizations</span><span class="err">/'+</span><span class="na">visualization_name</span><span class="err">+'/</span><span class="na">static</span><span class="err">/'</span>

    <span class="na">##</span> <span class="na">Actual</span> <span class="na">file</span> <span class="na">URL:</span>
    <span class="na">file_url = </span><span class="s">os.path.join(root,</span> <span class="err">'</span><span class="na">datasets</span><span class="err">',</span> <span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'],</span> <span class="err">"</span><span class="na">display</span><span class="err">?</span><span class="na">to_ext=</span><span class="s">"+hda.ext)

    ## Ensure BAI index is symlinked
    bai_target = hda.file_name+'.bai'

    if not os.path.isfile(bai_target):
        os.symlink(hda.metadata.bam_index.file_name, bai_target)

    ## Extract idxstats
    import pysam
    bam_idxstats_data = pysam.idxstats(hda.file_name)

%&gt;
&lt;html&gt;
    &lt;head&gt;
       &lt;title&gt;${hda.name | h} | ${visualization_name | h}&lt;/title&gt;
        &lt;script&gt;
            bam_idxstats_data = ${bam_idxstats_data};
            function parse_data(data) {
                var output = {};
                for(var i = 0; i &lt; data.length ; i++) {
                    var line = data[i];
                    var chunks = line.split("</span><span class="err">\</span><span class="na">t</span><span class="err">");</span>

                    <span class="na">if</span><span class="err">(</span><span class="na">chunks</span><span class="err">[0].</span><span class="na">split</span><span class="err">("</span><span class="na">_</span><span class="err">"</span><span class="na">).length =</span><span class="s">=</span> <span class="err">1)</span> <span class="err">{</span> <span class="err">//</span> <span class="na">only</span> <span class="na">if</span> <span class="na">it</span> <span class="na">does</span> <span class="na">not</span> <span class="na">contain</span> <span class="na">underscore</span>
                        <span class="na">output[chunks[0]] = </span><span class="s">parseInt(chunks[2]);</span>
                    <span class="err">}</span>
                <span class="err">}</span>
                <span class="na">return</span> <span class="na">output</span><span class="err">;</span>
            <span class="err">}</span>
        <span class="err">&lt;/</span><span class="na">script</span><span class="nt">&gt;</span>
    <span class="nt">&lt;/head&gt;</span>
    <span class="nt">&lt;body</span> <span class="na">onload=</span><span class="s">"bam_idxstats = parse_data(bam_idxstats_data);"</span><span class="nt">&gt;</span>
        ${bam_idxstats_data | h}
    <span class="nt">&lt;/body&gt;</span>
<span class="nt">&lt;/html&gt;</span>
</code></pre></div>      </div>
    </li>
    <li>
      <p>Retrigger the visualization and open the developers console of your browser: In the console, type: <code class="language-plaintext highlighter-rouge">bam_idxstats_data</code> and press <kbd>Enter</kbd>
  This should give the parsed contents as a dictionary, which can directly be used in Javascript.</p>
    </li>
  </ol>

</blockquote>

<p>From this point forward you are encouraged to continue on your own to see if you are able to create
a simple visualization from this dictionary. Think of tables, DIVs or even more complicated
solutions :).</p>

<p>Below is an example visualization, which creates a bar plot showing the number of reads per
chromosome.</p>

<p><a href="../../images/vis_plugins_example.png" rel="noopener noreferrer"><img src="../../images/vis_plugins_example.png" alt="Example visualization. " width="567" height="321" loading="lazy" /></a></p>

<p>The full contents of this plugin are provided in the <a href="https://github.com/galaxyproject/training-material/tree/main/topics/dev/files/hands_on-visualizations/alignment_rname_boxplot">GitHub repository related to this material in <code class="language-plaintext highlighter-rouge">tree/master/topics/dev/files/hands_on-visualizations/alignment_rname_boxplot</code></a>.
To try out this example, simply copy this folder to the <code class="language-plaintext highlighter-rouge">$GALAXY_ROOT/config/plugins/visualizations/</code> folder
on your (local) Galaxy and restart Galaxy.</p>

<p>The contents of the mako file for this example are given below.</p>

<div class="language-html highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="cp">&lt;!DOCTYPE HTML&gt;</span>
<span class="nt">&lt;</span><span class="err">%</span>
    <span class="na">import</span> <span class="na">os</span>

    <span class="na">##</span> <span class="na">Generates</span> <span class="na">hash</span> <span class="err">(</span><span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'])</span> <span class="na">of</span> <span class="na">history</span> <span class="na">item</span>
    <span class="na">hdadict = </span><span class="s">trans.security.encode_dict_ids(</span> <span class="na">hda.to_dict</span><span class="err">()</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Finds</span> <span class="na">the</span> <span class="na">parent</span> <span class="na">directory</span> <span class="na">of</span> <span class="na">galaxy</span> <span class="err">(/,</span> <span class="err">/</span><span class="na">galaxy</span><span class="err">,</span> <span class="na">etc.</span><span class="err">)</span>
    <span class="na">root     = </span><span class="s">h.url_for(</span> <span class="err">'/'</span> <span class="err">)</span>

    <span class="na">##</span> <span class="na">Determines</span> <span class="na">the</span> <span class="na">exposed</span> <span class="na">URL</span> <span class="na">of</span> <span class="na">the</span> <span class="err">./</span><span class="na">static</span> <span class="na">directory</span>
    <span class="na">app_root = </span><span class="s">root</span> <span class="err">+</span> <span class="err">'</span><span class="na">plugins</span><span class="err">/</span><span class="na">visualizations</span><span class="err">/'+</span><span class="na">visualization_name</span><span class="err">+'/</span><span class="na">static</span><span class="err">/'</span>

    <span class="na">##</span> <span class="na">Actual</span> <span class="na">file</span> <span class="na">URL:</span>
    <span class="na">file_url = </span><span class="s">os.path.join(root,</span> <span class="err">'</span><span class="na">datasets</span><span class="err">',</span> <span class="na">hdadict</span><span class="err">['</span><span class="na">id</span><span class="err">'],</span> <span class="err">"</span><span class="na">display</span><span class="err">?</span><span class="na">to_ext=</span><span class="s">"+hda.ext)

    ## Ensure BAI index is symlinked
    bai_target = hda.file_name+'.bai'

    if not os.path.isfile(bai_target):
        os.symlink(hda.metadata.bam_index.file_name, bai_target)

    ## Extract idxstats
    import pysam
    bam_idxstats_data = pysam.idxstats(hda.file_name)
%&gt;
&lt;html&gt;
     &lt;head&gt;
        &lt;title&gt;${hda.name | h} | ${visualization_name | h}&lt;/title&gt;

        &lt;style&gt;
             .chart div {
               font: 10px sans-serif;
               background-color: steelblue;
               text-align: right;
               padding: 3px;
               margin: 1px;
               color: white;
             }
        &lt;/style&gt;
        &lt;script&gt;
            bam_idxstats_data = ${bam_idxstats_data};
            function parse_data(data) {
                 /*
                  Data comes in as tuple of unsplit lines:
                  ["</span><span class="na">chr1</span><span class="err">\</span><span class="na">t1000</span><span class="err">\</span><span class="na">t0</span><span class="err">\</span><span class="na">t0</span><span class="err">",</span> <span class="err">"</span><span class="na">chr2</span><span class="err">\</span><span class="na">t2500</span><span class="err">\</span><span class="na">t0</span><span class="err">\</span><span class="na">t0</span><span class="err">"]</span>

                  <span class="na">We</span> <span class="na">need</span> <span class="na">to</span> <span class="na">split</span> <span class="na">it</span> <span class="na">up</span><span class="err">,</span> <span class="na">and</span> <span class="na">ideally</span> <span class="na">only</span> <span class="na">keep</span> <span class="na">reference</span> <span class="na">names</span> <span class="na">without</span> <span class="na">an</span> <span class="na">underscore</span>
                 <span class="na">*</span><span class="err">/</span>
                 <span class="na">var</span> <span class="na">output = </span><span class="s">{};</span>

                 <span class="na">for</span><span class="err">(</span><span class="na">var</span> <span class="na">i = </span><span class="s">0;</span> <span class="na">i</span> <span class="err">&lt;</span> <span class="na">data.length</span> <span class="err">;</span> <span class="na">i</span><span class="err">++)</span> <span class="err">{</span>
                     <span class="na">var</span> <span class="na">line = </span><span class="s">data[i];</span>
                     <span class="na">var</span> <span class="na">chunks = </span><span class="s">line.split("\t");</span>

                     <span class="na">if</span><span class="err">(</span><span class="na">chunks</span><span class="err">[0].</span><span class="na">split</span><span class="err">("</span><span class="na">_</span><span class="err">"</span><span class="na">).length =</span><span class="s">=</span> <span class="err">1)</span> <span class="err">{</span> <span class="err">//</span> <span class="na">only</span> <span class="na">if</span> <span class="na">it</span> <span class="na">does</span> <span class="na">not</span> <span class="na">contain</span> <span class="na">underscore</span>
                         <span class="na">output[chunks[0]] = </span><span class="s">parseInt(chunks[2]);</span>
                     <span class="err">}</span>
                 <span class="err">}</span>

                 <span class="na">return</span> <span class="na">output</span><span class="err">;</span>
            <span class="err">}</span>

            <span class="na">function</span> <span class="na">calc_stats</span><span class="err">(</span><span class="na">parsed</span><span class="err">)</span> <span class="err">{</span>
                <span class="na">max = </span><span class="s">0;</span>
                <span class="na">sum = </span><span class="s">0;</span>
                <span class="na">for</span> <span class="err">(</span><span class="na">var</span> <span class="na">key</span> <span class="na">in</span> <span class="na">parsed</span><span class="err">)</span> <span class="err">{</span>
                    <span class="na">if</span> <span class="err">(</span><span class="na">parsed</span><span class="err">[</span><span class="na">key</span><span class="err">]</span> <span class="nt">&gt;</span> max){
                        max = parsed[key];
                    }
                    sum += parsed[key]
                }
                return [max, sum];
            }

            function plot_data(parsed) {
                var max = calc_stats(parsed)[0];
                var sum = calc_stats(parsed)[1];

                for (var key in parsed) {
                     var value = parsed[key];
                     var ratio = 100.0 * value / sum;
                     var ratio2 = 100.0 * value / max;

                     var div = document.createElement("div");
                     div.innerHTML = '<span class="nt">&lt;nobr&gt;</span>'+key+'<span class="nt">&lt;/nobr&gt;</span>';
                     document.getElementById("chart_names").appendChild(div);

                     var div = document.createElement("div");
                     div.innerHTML = '<span class="nt">&lt;nobr&gt;</span>'+value+" ("+Math.round(ratio*100)/100+"%)<span class="nt">&lt;/nobr&gt;</span>";
                     div.title = key+': '+value+" ("+Math.round(ratio*100)/100+"%)";
                     div.style.width =  ratio2+'%';
                     document.getElementById("chart").appendChild(div);
                }
            }
        <span class="nt">&lt;/script&gt;</span>
     <span class="nt">&lt;/head&gt;</span>
     <span class="nt">&lt;body</span> <span class="na">onload=</span><span class="s">"plot_data(parse_data(bam_idxstats_data));"</span><span class="nt">&gt;</span>
         <span class="nt">&lt;center&gt;</span>
             <span class="nt">&lt;h1&gt;</span>Bam contents of ${hda.name | h}<span class="nt">&lt;/h1&gt;</span>

             <span class="nt">&lt;table</span> <span class="na">border=</span><span class="s">"0"</span> <span class="na">borderpadding=</span><span class="s">"0"</span> <span class="na">borderpanning=</span><span class="s">")"</span> <span class="na">style=</span><span class="s">"width: 500px;"</span><span class="nt">&gt;</span>
                 <span class="nt">&lt;tr&gt;</span>
                     <span class="nt">&lt;td</span> <span class="na">style=</span><span class="s">"width:50px;"</span><span class="nt">&gt;</span>
                         <span class="nt">&lt;div</span> <span class="na">id=</span><span class="s">"chart_names"</span> <span class="na">class=</span><span class="s">"chart"</span> <span class="na">style=</span><span class="s">"width: 100%; border: 1px dashed gray;text-align: left;"</span> <span class="nt">/&gt;</span>
                     <span class="nt">&lt;/td&gt;</span>
                     <span class="nt">&lt;td</span> <span class="na">style=</span><span class="s">"width:450px;"</span><span class="nt">&gt;</span>
                         <span class="nt">&lt;div</span> <span class="na">id=</span><span class="s">"chart"</span> <span class="na">class=</span><span class="s">"chart"</span> <span class="na">style=</span><span class="s">"width:100%; border: 1px dashed gray;text-align: left;"</span> <span class="nt">/&gt;</span>
                     <span class="nt">&lt;/td&gt;</span>
                 <span class="nt">&lt;/tr&gt;</span>
             <span class="nt">&lt;/table&gt;</span>
         <span class="nt">&lt;/center&gt;</span>
     <span class="nt">&lt;/body&gt;</span>
<span class="nt">&lt;/html&gt;</span>

</code></pre></div></div>

<p>In the given example, the <code class="language-plaintext highlighter-rouge">RNAME</code> queries containing an underscore were removed.
This is because there are many alternative chromosomes, making the list very large
for certain reference genomes. However, for certain studies it might be desired to
just look at those.</p>

<p>The current plot is a box plot, but one can imagine a pie-chart may be convenient too.
All of those additional settings can be implemented for interactive behaviour,
contributing to quicker understanding of the data which is generally not so convenient
using static Galaxy tools.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-static-files"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Static files</div>

  <p>In the example we included Javascript and CSS into the HTML website.
Remember that for every new invocation of the visualization the entire CSS en JS are copied
and transferred as well. This is a waste of (redundant) bandwidth as we could save the
files in the static directory and refer to them within the HTML. The browser shall check
it’s cache for the presence of libs and style sheets and only update them if they have changed.</p>
</blockquote>

<h3 id="improvements">Improvements</h3>

<p>Another thing you may realize is that we still do the calculation (pysam.idxstats) server side.
Although this is a marginal calculation, it is causing delay and the bigger the files, the worse
this delay becomes. The underlying problem is that we did not obtain and parse the BAI file via
Javascript. This would be a more elaborate solution, but requires more development time as we
need to develop a function able to parse the binary file.</p>

<p>Another thing we could do is create a static <code class="language-plaintext highlighter-rouge">samtools idxstats</code> tool, that creates a file of
datatype <code class="language-plaintext highlighter-rouge">tabular.Idxstats</code> and include that datatype into Galaxy. We then make the visualization
specific for that datatype, just plotting the results of the <code class="language-plaintext highlighter-rouge">idxstats</code>.</p>

<p>A fundamental and more complicated problem is that BAM files are simply too big to transfer for
these kind of applications. It would be ideal to have web server integration that allows querying
of specific locations or metadata within or from a BAM file where indexing operations are taken
care of at the server side. This is what has been done in Trackster.</p>

<h3 id="more-examples">More examples</h3>

<p>For more examples of visualization plugins, you can browse this
<a href="https://github.com/bgruening/galaxytools/tree/master/visualisations">GitHub repo</a></p>

<h1 id="conclusion">Conclusion</h1>

<p>We have just created a visualization plugin in Galaxy to visualize the number of alignments
per <code class="language-plaintext highlighter-rouge">RNAME</code> (chromosome) in a BAM file.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Visualizations require a different way of thinking: server and client side; downloading files rather than system level access</p>
</li>
                        
                        <li><p>Interactivity is what makes visualizations different from static tools</p>
</li>
                        
                        <li><p>Requires understanding of both the Galaxy ecosystem as well as HTML5/JS</p>
</li>
                        
                        <li><p>Performance is more important than for static Galaxy tools</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/dev/faqs/">FAQ page for the Development in Galaxy topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Generic plugins (Development in Galaxy)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Saskia Hiltemann, Youri Hoogstrate,  <b>Generic plugins (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/dev/tutorials/visualization-generic/tutorial.html">https://training.galaxyproject.org/training-material/topics/dev/tutorials/visualization-generic/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{dev-visualization-generic,
author = "Saskia Hiltemann and Youri Hoogstrate",
	title = "Generic plugins (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/dev/tutorials/visualization-generic/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/dev/tutorials/visualization-generic/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools: []
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/fc3f11ce65e906601f63521f6553699c960500e1">fc3f11c</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00121">https://gxy.io/GTN:T00121</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/dev/tutorials/visualization-generic/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


        <script src='/training-material/assets/js/bundle.03aefefa.js?v=1705578603'></script>
        
    </body>
</html>