### Consensus genome characterization

Our next step is using the trimmed alignment and the Wuhan-Hu-1 reference (i.e. MN908947.3) to build a consensus
sequence for each of our viral samples. 

> ### Call consensus from the aligned BAM file with `ivar consensus`
> 1. Find the {% tool [ivar consensus](toolshed.g2.bx.psu.edu/repos/iuc/ivar_consensus/ivar_consensus/1.3.1+galaxy0) %} tool and use the collection of BAM files created by {% icon param-collection %} *ivar trim* as input.
> 2. Enter **20** for Minimum quality score thethreshold
> 3. Enter **0.7** for Minimum frequency threshold
> 4. Enter **20** for Minimum depth
> 5. Select **No** for Exclude regions with smaller depth than the minimum threshold
> 6. Select **Yes** for Use N instead of - for regions with less than minimum coverage
>
{: .hands_on}


> ### Rename the FASTA header with `Text transformation with sed`
>
> The consensus sequences generated by `ivar consensus` have FASTA IDs starting with *Consensus_*, but for downstream analysis it is better to use the sample ID as the FASTA IDs. We can use `sed` to make this change.
>
> 1. Using the {% tool [Text transformation with sed](toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.1.1) %} tool, select the output dataset collection from {% icon param-collection %} `ivar trim` as input and
> 2.  Paste this `/^>/s/Consensus_(.*)_threshold_.*/\1/` sed code in the SED Program window.
{: .hands_on}
