<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Calling variants in non-diploid systems</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:title" content="Galaxy Training: Calling variants in non-diploid systems" />
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Variant Analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fnon-dip%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fnon-dip%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fnon-dip%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fnon-dip%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fnon-dip%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fnon-dip%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/variant-analysis/tutorials/non-dip/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Calling variants in non-diploid systems",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
  },
  "courseCode": "variant-analysis / non-dip / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Calling variants in non-diploid systems' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html",
  "timeRequired": "PT1H30M",
  "keywords": "prokaryote",
  "description": "The questions this  addresses are:\n - How does frequency of mitochondrial polymorphisms change from mother to child?\n\n\\nThe objectives are:\n - Using Galaxy's main site we will see how to call variants in bacteria, viruses, and organelles.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Alex Ostrovsky"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Anton Nekrutenko"
    },
    {
      "@type": "Person",
      "name": "Alex Ostrovsky"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Calling variants in non-diploid systems</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/nekrut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/nekrut" alt="Anton Nekrutenko">Anton Nekrutenko</a>, <a href="/training-material/hall-of-fame/astrovsky01/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/astrovsky01" alt="Alex Ostrovsky">Alex Ostrovsky</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How does frequency of mitochondrial polymorphisms change from mother to child?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Using Galaxy‚Äôs main site we will see how to call variants in bacteria, viruses, and organelles.</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour 30 minutes</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/variant-analysis/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1251112">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/non-dip/workflows/" title="Workflows" alt="Calling variants in non-diploid systems workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/non-dip/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>The majority of life on Earth is non-diploid and represented by prokaryotes, viruses, and their derivatives, such as our own mitochondria or plant‚Äôs chloroplasts. In non-diploid systems, allele frequencies can range anywhere between 0 and 100% and there could be multiple (not just two) alleles per locus. The main challenge associated with non-diploid variant calling is the difficulty in distinguishing between the sequencing noise (abundant in all NGS platforms) and true low frequency variants. Some of the early attempts to do this well have been accomplished on human mitochondrial <span class="notranslate">DNA</span> although the same approaches will work equally good on viral and bacterial genomes (<a class="citation" href="#Rebolledo-Jaramillo2014">Rebolledo-Jaramillo <i>et al.</i> 2014</a>, <a class="citation" href="#Li2015">Li <i>et al.</i> 2015</a>).</p>

<p>As an example of non-diploid systems, we will be using human mitochondrial genome. However, this approach will also work for most bacterial and viral genomes.</p>

<p>There are two ways one can call variants:</p>

<ol>
  <li>By comparing <span class="notranslate">reads</span> against an existing genome assembly</li>
  <li>By assembling a genome first and then <span class="notranslate">mapping</span> against that assembly</li>
</ol>

<figure id="figure-1"><img src="../../images/ref_vs_assembly.jpg" alt="2 approaches" /><figcaption><span class="figcaption-prefix">Figure 1:</span> This figure from <a class="citation" href="#Olson2015">Olson <i>et al.</i> 2015</a> contrasts the two approaches.</figcaption></figure>

<p>In this tutorial we will take the <em>first</em> path, in which we map <span class="notranslate">reads</span> against an existing assembly. Later in the course, after we learn about assembly approaches, we will try the second approach as well.</p>

<p>The goal of this example is to detect heteroplasmies - variants within mitochondrial <span class="notranslate">DNA</span>. Mitochondria are transmitted mate<span class="notranslate">rna</span>lly, and heteroplasmy frequencies may change dramatically and unpredictably during the transmission due to a germ-line bottleneck (<a class="citation" href="#Cree2008">Cree <i>et al.</i> 2008</a>). As we mentioned above, the procedure for finding variants in bacterial or viral genomes will be essentially the same.</p>

<p>Datasets representing a child and a mother are available in <a href="https://doi.org/10.5281/zenodo.1251112">Zenodo</a>. These datasets were obtained by paired-end Illumina sequencing of human genomic <span class="notranslate">DNA</span> enriched for mitochondria. The enrichment was performed using long-range PCR with two primer pairs that amplify the entire mitochondrial genome. Samples will therefore still contain a lot of <span class="notranslate">DNA</span> from the nuclear genome, which, in this case, is a contaminant.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#importing-data" id="markdown-toc-importing-data">Importing data</a>    <ol>
      <li><a href="#checking-data-quality" id="markdown-toc-checking-data-quality">Checking data quality</a></li>
    </ol>
  </li>
  <li><a href="#mapping-reads-to-a-reference" id="markdown-toc-mapping-reads-to-a-reference"><span class="notranslate">Mapping</span> <span class="notranslate">reads</span> to a reference</a></li>
  <li><a href="#postprocessing-mapped-reads" id="markdown-toc-postprocessing-mapped-reads">Postprocessing mapped <span class="notranslate">reads</span></a>    <ol>
      <li><a href="#merging-bam-datasets" id="markdown-toc-merging-bam-datasets">Merging BAM datasets</a></li>
      <li><a href="#removing-duplicates" id="markdown-toc-removing-duplicates">Removing duplicates</a></li>
      <li><a href="#left-aligning-indels" id="markdown-toc-left-aligning-indels">Left-aligning indels</a></li>
      <li><a href="#filtering-reads" id="markdown-toc-filtering-reads">Filtering <span class="notranslate">reads</span></a></li>
    </ol>
  </li>
  <li><a href="#calling-non-diploid-variants" id="markdown-toc-calling-non-diploid-variants">Calling non-diploid variants</a>    <ol>
      <li><a href="#filtering-variants" id="markdown-toc-filtering-variants">Filtering variants</a></li>
    </ol>
  </li>
  <li><a href="#examining-the-results" id="markdown-toc-examining-the-results">Examining the results</a>    <ol>
      <li><a href="#visualising-with-vcfiobio" id="markdown-toc-visualising-with-vcfiobio">Visualising with VCF.IOBIO</a></li>
      <li><a href="#visualising-with-igv" id="markdown-toc-visualising-with-igv">Visualising with IGV</a></li>
      <li><a href="#comparing-frequencies" id="markdown-toc-comparing-frequencies">Comparing frequencies</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="importing-data">Importing data</h1>

<p>For this tutorial we have prepared a subset of data previously by our group (<a class="citation" href="#Rebolledo-Jaramillo2014">Rebolledo-Jaramillo <i>et al.</i> 2014</a>). Let‚Äôs import these data into <span class="notranslate">Galaxy</span>. They are available from <a href="https://usegalaxy.org/library/folders/Fe4842bd0c37b03a7">this <span class="notranslate">Galaxy</span> Library</a> or via <a href="https://zenodo.org/record/582600">Zenodo</a></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-get-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Get the data</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import files from <a href="https://zenodo.org/record/1251112">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1251112/files/raw_child-ds-1.fq
https://zenodo.org/record/1251112/files/raw_child-ds-2.fq
https://zenodo.org/record/1251112/files/raw_mother-ds-1.fq
https://zenodo.org/record/1251112/files/raw_mother-ds-2.fq
</code></pre></div>      </div>
    </li>
    <li>
      <p>Check that all newly created datasets in your history are assigned datatype
<code class="language-plaintext highlighter-rouge">fastqsanger</code>, and fix any missing or wrong datatype assignment</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="checking-data-quality">Checking data quality</h2>

<p>Before proceeding with the analysis, we need to find out how good the data actually is. For this will use <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-assess-quality-of-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Assess quality of data</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  on each of the four FASTQ datasets with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúShort read data from your current history‚Äù</em>: all 4 FASTQ datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select multiple datasets</h3>

        <ol>
          <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
          <li>Select several files by keeping the <kbd>Ctrl</kbd> (or
<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
        </ol>
      </blockquote>
    </li>
  </ol>

</blockquote>

<p>Once the FastQC jobs runs, you will be able to look at the HTML reports generated by this tool. The data have generally high quality in this example:</p>

<figure id="figure-2"><img src="../../images/mt_qc_plot.png" alt="FastQC output plot" /><figcaption><span class="figcaption-prefix">Figure 2:</span> FastQC plot for one of the mitochondrial datasets shows that qualities are acceptable for 250 bp <span class="notranslate">reads</span> (mostly in the green, which is at or above [Phred score](https://en.wikipedia.org/wiki/Phred_quality_score) of 30).</figcaption></figure>

<h1 id="mapping-reads-to-a-reference"><span class="notranslate">Mapping</span> <span class="notranslate">reads</span> to a reference</h1>

<p>Our <span class="notranslate">reads</span> are long (250 bp) so we will use BWA-MEM (<a class="citation" href="#Li2013">Li 2013</a>) to align them against the reference genome as it has good <span class="notranslate">mapping</span> performance for longer <span class="notranslate">reads</span> (100bp and up).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-map-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Map <span class="notranslate">reads</span></h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1</span></span>  to map the <span class="notranslate">reads</span> to the reference genome with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg38</code> (or a similarly named option)</li>
          </ul>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSelect first set of <span class="notranslate">reads</span>‚Äù</em>: select both <code class="language-plaintext highlighter-rouge">-1</code> datasets selected with <strong>Multiple datasets</strong></li>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSelect second set of <span class="notranslate">reads</span>‚Äù</em>: select both <code class="language-plaintext highlighter-rouge">-2</code> datasets selected with <strong>Multiple datasets</strong></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúPlatform/technology used to produce the <span class="notranslate">reads</span> (PL)‚Äù</em>: <code class="language-plaintext highlighter-rouge">ILLUMINA</code></li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-more-about-selecting-datasets"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> More about selecting datasets</h3>

  <p>By selecting datasets 1 and 3 as <strong>Select the first set of <span class="notranslate">reads</span></strong> and datasets 2 and 4 as <strong>Select the second set of <span class="notranslate">reads</span></strong>, <span class="notranslate">Galaxy</span> will automatically launch two BWA-MEM jobs using datasets 1,2 and 3,4 generating two resulting BAM files. 
By setting <strong>Set read groups information</strong> to <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specifications)</code> and clicking <strong>Auto-assign</strong> we will ensure that the <span class="notranslate">reads</span> in the resulting BAM dataset are properly set.</p>
</blockquote>

<h1 id="postprocessing-mapped-reads">Postprocessing mapped <span class="notranslate">reads</span></h1>

<h2 id="merging-bam-datasets">Merging BAM datasets</h2>

<p>Because we have set read groups, we can now merge the two BAM dataset into one. This is because read groups label each read as belonging to either <em>mother</em> or <em>child</em>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-merge-multiple-datasets-into-one"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Merge multiple datasets into one</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/2.18.2.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/2.18.2.1"><strong>MergeSAMFiles</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/2.18.2.1</span></span>  to merge the BAM datasets with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúSelect SAM/BAM dataset or dataset collection‚Äù</em>: select both BAM datasets produced by <strong>BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúSelect validation stringency‚Äù</em>: <code class="language-plaintext highlighter-rouge">Lenient</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="removing-duplicates">Removing duplicates</h2>

<p>Preparation of sequencing libraries (at least at the time of writing) for technologies such as Illumina (used in this example) involves PCR amplification. It is required to generate sufficient number of sequencing templates so that a reliable detection can be performed by base callers. PCR has its own biases which are especially profound in cases of multi-template PCR used for construction of sequencing libraries (<a class="citation" href="#Kanagawa2003">Kanagawa 2003</a>).</p>

<p>Duplicates can be identified based on their outer alignment coordinates or using sequence-based clustering. One of the common ways for identification of duplicate <span class="notranslate">reads</span> is the <strong>MarkDuplicates</strong> utility from <a href="https://broadinstitute.github.io/picard/command-line-overview.html">Picard</a> package which is designed to identify both PCR and optical duplicates.</p>

<blockquote class="details">
  <h3 id="details-more-about-markduplicates"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More about MarkDuplicates</h3>

  <p>Duplicates are identified as read pairs having identical 5‚Äô positions (coordinate and strand) for both <span class="notranslate">reads</span> in a mate pair (and optionally, matching unique molecular identifier <span class="notranslate">reads</span>; see <code class="language-plaintext highlighter-rouge">BARCODE_TAG</code> option). Optical, or more broadly sequencing, duplicates are duplicates that appear clustered together spatially during sequencing and can arise from optical/imagine-processing artifacts or from biochemical processes during clonal amplification and sequencing; they are identified using the <code class="language-plaintext highlighter-rouge">READ_NAME_REGEX</code> and the <code class="language-plaintext highlighter-rouge">OPTICAL_DUPLICATE_PIXEL_DISTANCE</code> options. The tool‚Äôs main output is a new SAM or BAM file in which duplicates have been identified in the SAM flags field, or optionally removed (see <code class="language-plaintext highlighter-rouge">REMOVE_DUPLICATE</code> and <code class="language-plaintext highlighter-rouge">REMOVE_SEQUENCING_DUPLICATES</code>), and optionally marked with a duplicate type in the ‚ÄòDT‚Äô optional attribute. In addition, it also outputs a metrics file containing the numbers of <code class="language-plaintext highlighter-rouge">READ_PAIRS_EXAMINED</code>, <code class="language-plaintext highlighter-rouge">SECONDARY_OR_SUPPLEMENTARY_RDS</code>, <code class="language-plaintext highlighter-rouge">UNMAPPED_<span class="notranslate">READS</span></code>, <code class="language-plaintext highlighter-rouge">UNPAIRED_<span class="notranslate">READS</span></code>, <code class="language-plaintext highlighter-rouge">UNPAIRED_READ DUPLICATES</code>, <code class="language-plaintext highlighter-rouge">READ_PAIR_DUPLICATES</code>, and <code class="language-plaintext highlighter-rouge">READ_PAIR_OPTICAL_DUPLICATES</code>.</p>

  <p>Usage example: <code class="language-plaintext highlighter-rouge">java -jar picard.jar MarkDuplicates I=input.bam O=marked_duplicates.bam M=marked_dup_metrics.txt</code></p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-de-duplicate-mapped-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: De-duplicate mapped data</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2"><strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/2.18.2.2</span></span>  to de-duplicate the merged BAM datasets with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect SAM/BAM dataset or dataset collection‚Äù</em>: select the merged BAM dataset produced by <strong>MergeSAMFiles</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúThe scoring strategy for choosing the non-duplicate among candidates‚Äù</em>: <code class="language-plaintext highlighter-rouge">SUM_OF_BASE_QUALITIES</code></li>
        <li><em>‚ÄúThe maximum offset between two duplicate clusters in order to consider them optical duplicates‚Äù</em>: <code class="language-plaintext highlighter-rouge">100</code></li>
        <li><em>‚ÄúSelect validation stringency‚Äù</em>: <code class="language-plaintext highlighter-rouge">Lenient</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>MarkDuplicates produces a BAM dataset with duplicates removed and also a metrics file. Let‚Äôs take a look at the metrics data:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>raw_child-ds-	55	27551	849	55	50	1658	1	0.06103	219750
raw_mother-ds-	95	54973	1951	95	89	4712	2	0.08645	302188
</code></pre></div></div>

<p>Column headers are:</p>

<ul>
  <li>LIBRARY (read group in our case)</li>
  <li>UNPAIRED_<span class="notranslate">READS</span>_EXAMINED</li>
  <li>READ_PAIRS_EXAMINED-</li>
  <li>SECONDARY_OR_SUPPLEMENTARY_RDS</li>
  <li>UNMAPPED_<span class="notranslate">READS</span></li>
  <li>UNPAIRED_READ_DUPLICATES</li>
  <li>READ_PAIR_DUPLICATES</li>
  <li>READ_PAIR_OPTICAL_DUPLICATES</li>
  <li>PERCENT_DUPLICATION</li>
  <li>ESTIMATED_LIBRARY_SIZE</li>
</ul>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>What percent of read duplication are found in each read group (child and mother)?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The two datasets have ~6% and ~9% duplicates for child and mother, respectively.</li>
    </ol>

  </blockquote>
</blockquote>

<h2 id="left-aligning-indels">Left-aligning indels</h2>

<p>Left aligning of indels (a variant of re-aligning) is extremely important for obtaining accurate variant calls. For illustrating how left-aligning works, we expanded on an example provided by <a class="citation" href="#Tan2015">Tan <i>et al.</i> 2015</a>. Suppose you have a reference sequence and a sequencing read:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Reference GGGCACACACAGGG
Read      GGGCACACAGGG
</code></pre></div></div>

<p>If you look carefully you will see that the read is simply missing a <code class="language-plaintext highlighter-rouge">CA</code> repeat. But it is not apparent to a mapper, so some of possible alignments and corresponding variant calls include:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Alignment                 Variant Call

GGGCACACACAGGG            Ref: CAC
GGGCAC--ACAGGG            Alt: C

GGGCACACACAGGG            Ref: ACA
GGGCA--CACAGGG            Alt: A

GGGCACACACAGGG            Ref: GCA
GGG--CACACAGGG            Alt: G
</code></pre></div></div>

<p>The last of these is <em>left-aligned</em>. In this case gaps (represented by dashes) are moved as far left as possible. For a formal definition of left-alignment and variant normalization, see <a class="citation" href="#Tan2015">Tan <i>et al.</i> 2015</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-left-align-indels"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Left-align indels</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1"><strong>BamLeftAlign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1</span></span>  to perform left alignment with the following parameters:
      <ul>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect alignment file in BAM format‚Äù</em>: select the BAM dataset produced by <strong>MarkDuplicates</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">hg38</code></li>
        <li><em>‚ÄúMaximum number of iterations‚Äô</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="filtering-reads">Filtering <span class="notranslate">reads</span></h2>

<p>Remember that we are trying to call variants in mitochondrial genome. Let focus only on the <span class="notranslate">reads</span> derived from mitochondria genome by filtering everything else out.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filter-bam-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filter BAM data</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1"><strong>Filter BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM dataset(s) to filter‚Äù</em>: select the BAM dataset produced by <strong>BamLeftAlign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚ÄúCondition‚Äù</em>:
          <ul>
            <li>In <em>‚Äú1: Condition‚Äù</em>:
              <ul>
                <li>In <em>‚ÄúFilter‚Äù</em>:
                  <ul>
                    <li>In <em>‚Äú1: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapQuality</code></li>
                        <li><em>‚ÄúFilter on read <span class="notranslate">mapping</span> quality (phred scale)‚Äù</em>: <code class="language-plaintext highlighter-rouge">&gt;=20</code></li>
                      </ul>
                    </li>
                    <li>In <em>‚Äú2: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">isPaired</code></li>
                        <li><em>‚ÄúSelected mapped <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                    <li>Click on <em>‚ÄúInsert Filter‚Äù</em></li>
                    <li>In <em>‚Äú3: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">isProperPair</code></li>
                        <li><em>‚ÄúSelect <span class="notranslate">reads</span> with mapped mate‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                    <li>Click on <em>‚ÄúInsert Filter‚Äù</em></li>
                    <li>In <em>‚Äú4: Filter‚Äù</em>:
                      <ul>
                        <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">reference</code></li>
                        <li><em>‚ÄúSelect <span class="notranslate">reads</span> with mapped mate‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrM</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-filtering-reads"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Filtering <span class="notranslate">reads</span></h3>
        <p>Further explanation of the filters used:</p>
        <ul>
          <li><em>mapQuality</em> is set to ‚â•20. <span class="notranslate">Mapping</span> quality reflects the probability that the read is placed incorrectly using <a href="https://en.wikipedia.org/wiki/Phred_quality_score">phred scale</a>. Thus 20 is 1/100 or 1% chance that the read is incorrectly mapped. By setting this parameter to ‚â•20, we will keep all <span class="notranslate">reads</span> that have 1% or less probability of being mapped incorrectly.</li>
          <li><em>isPaired</em> will eliminate singleton (unpaired) <span class="notranslate">reads</span>.</li>
          <li><em>isProperPair</em> will only keep <span class="notranslate">reads</span> that map to the same chromosome and are properly placed.</li>
          <li><em>reference</em> is set to the mitochondrial chromosome, chrM.</li>
        </ul>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="calling-non-diploid-variants">Calling non-diploid variants</h1>

<p>FreeBayes is widely used for calling variants in diploid systems. However, it can also be used for calling variants in pooled samples where the number of samples is not known. This is the exact scenario we have here: in our sample we have multiple mitochondrial (or bacterial or viral) genomes, but we do not know exactly how many. Thus we will use the <code class="language-plaintext highlighter-rouge">--pooled-continuous</code> option of FreeBayes to generate frequency-based variant calls as well as some other options highlighted below.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-calling-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Calling variants</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/freebayes/freebayes/1.3.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/freebayes/freebayes/1.3.1"><strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/freebayes/freebayes/1.3.1</span></span>  to call variants with the following parameters:
      <ul>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code></li>
        <li><em>‚ÄúRun in batch mode?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Run individually</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBAM dataset‚Äù</em>: select the BAM dataset produced by last <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step</li>
        <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">hg38</code></li>
        <li><em>‚ÄúLimit variant calling to a set of regions?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Limit to region</code>
          <ul>
            <li><em>‚ÄúRegion Chromosome‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrM</code></li>
            <li><em>‚ÄúRegion Start‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
            <li><em>‚ÄúRegion End‚Äù</em>: <code class="language-plaintext highlighter-rouge">16000</code></li>
          </ul>
        </li>
        <li><em>‚ÄúChoose parameter selection level‚Äù</em>: <code class="language-plaintext highlighter-rouge">5: Full list of options</code>
          <ul>
            <li><em>‚ÄúPopulation model options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set population model options</code>
              <ul>
                <li><em>‚ÄúThe expected mutation rate or pairwise nucleotide diversity among the population under analysis‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.001</code></li>
                <li><em>‚ÄúSet ploidy for the analysis‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li><em>‚ÄúAssume that samples result from pooled sequencing‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                <li><em>‚ÄúOutput all alleles which pass input filters, regardless of genotyping outcome or model‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-population-model-options"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Population model options</h3>
        <p>The ‚ÄúPopulation model options‚Äù are one of the most important parameter choices to make when calling variants in non-diploid systems.</p>
      </blockquote>

      <ul>
        <li><em>‚ÄúAllelic scope options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set allelic scope options</code>
          <ul>
            <li><em>‚ÄúIgnore SNP alleles‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúIgnore indels alleles‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúIgnore multi-nucleotide polymorphisms, MNPs‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúIgnore complex events (composites of other classes)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-allelic-scope-options"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Allelic scope options</h3>
        <p>Mitochondria has a number of low complexity regions (mononucleotide repeats). Setting the allelic scope parameters as described above will decrease noise from these regions.</p>
      </blockquote>

      <ul>
        <li><em>‚ÄúInput filters‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set input filters</code>
          <ul>
            <li><em>‚ÄúExclude alignments from analysis if they have a <span class="notranslate">mapping</span> quality less than‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>‚ÄúExclude alleles from analysis if their supporting base quality less than‚Äù</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-filter-options"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Filter options</h3>
        <p>Setting <em>Exclude alignments from analysis if they have a <span class="notranslate">mapping</span> quality less than</em> to <code class="language-plaintext highlighter-rouge">20</code> (phred score of 20) will make FreeBayes only consider reliably aligned <span class="notranslate">reads</span>.
Setting <em>Exclude alleles from analysis if their supporting base quality less than</em> to <code class="language-plaintext highlighter-rouge">30</code> (phred score of 30) will make FreeBayes only consider high quality bases.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>FreeBayes will produce a <a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF</a> dataset similar to what is shown below (you may need to scroll sideways to see it in full). It lists 25 sites of interest. For brevity, the header lines have been removed:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	raw_child-ds-	raw_mother-ds-
chrM	73	.	A	G	30438.6	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=949;CIGAR=1X;DP=950;DPB=950;DPRA=0;EPP=144.879;EPPR=5.18177;GTI=0;LEN=1;MEANALT=1;MQM=55.8314;MQMR=60;NS=2;NUMALT=1;ODDS=2187.64;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=34948;QR=37;RO=1;RPL=322;RPP=215.867;RPPR=5.18177;RPR=627;RUN=1;SAF=421;SAP=29.2075;SAR=528;SRF=0;SRP=5.18177;SRR=1;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:226:0,226:0:0:226:8314:-739.462,0	1:724:1,723:1:37:723:26634:-2346.86,0
chrM	263	.	A	G	11603.3	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=364;CIGAR=1X;DP=364;DPB=364;DPRA=0;EPP=16.755;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=0;NS=2;NUMALT=1;ODDS=982.556;PAIRED=1;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=13201;QR=0;RO=0;RPL=276;RPP=213.858;RPPR=0;RPR=88;RUN=1;SAF=172;SAP=5.39653;SAR=192;SRF=0;SRP=0;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:108:0,108:0:0:108:3896:-350.812,0	1:256:0,256:0:0:256:9305:-837.344,0
chrM	310	.	TCC	CCC	4471	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=150;CIGAR=1X2M;DP=171;DPB=180.333;DPRA=0;EPP=7.70068;EPPR=4.45795;GTI=0;LEN=1;MEANALT=2.5;MQM=59.88;MQMR=60;NS=2;NUMALT=1;ODDS=407.373;PAIRED=1;PAIREDR=1;PAO=11;PQA=237;PQR=0;PRO=0;QA=5147;QR=194;RO=6;RPL=113;RPP=86.6265;RPPR=8.80089;RPR=37;RUN=1;SAF=35;SAP=95.6598;SAR=115;SRF=1;SRP=8.80089;SRR=5;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:54:0,49:0:0:49:1645:-150.867,0	1:117:6,101:6:194:101:3502:-315.963,0
chrM	513	.	GCACACACACAC	GCACACACACACAC	2095.84	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=109;CIGAR=1M2I11M;DP=150;DPB=172.167;DPRA=0;EPP=109.173;EPPR=6.05036;GTI=0;LEN=2;MEANALT=3.5;MQM=60;MQMR=60;NS=2;NUMALT=1;ODDS=225.991;PAIRED=1;PAIREDR=1;PAO=5;PQA=101.5;PQR=101.5;PRO=5;QA=3824;QR=1196;RO=35;RPL=19;RPP=103.436;RPPR=4.56135;RPR=90;RUN=1;SAF=80;SAP=54.8268;SAR=29;SRF=24;SRP=13.4954;SRR=11;TYPE=ins;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:34:5,28:5:163:28:996:-74.9521,0	1:116:30,81:30:1033:81:2828:-161.475,0
chrM	750	.	A	G	47254.6	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1459;CIGAR=1X;DP=1459;DPB=1459;DPRA=0;EPP=48.5904;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=59.8705;MQMR=0;NS=2;NUMALT=1;ODDS=3230.11;PAIRED=1;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=53335;QR=0;RO=0;RPL=594;RPP=112.315;RPPR=0;RPR=865;RUN=1;SAF=1007;SAP=461.453;SAR=452;SRF=0;SRP=0;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:331:0,331:0:0:331:11988:-1073.7,0	1:1128:0,1128:0:0:1128:41347:-3719.34,0
chrM	1438	.	A	G	75374.8	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=2237;CIGAR=1X;DP=2238;DPB=2238;DPRA=0;EPP=47.8812;EPPR=5.18177;GTI=0;LEN=1;MEANALT=1;MQM=59.8605;MQMR=60;NS=2;NUMALT=1;ODDS=5003.07;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=84494;QR=37;RO=1;RPL=1397;RPP=304.171;RPPR=5.18177;RPR=840;RUN=1;SAF=925;SAP=148.392;SAR=1312;SRF=0;SRP=5.18177;SRR=1;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:488:0,488:0:0:488:18416:-1656.39,0	1:1750:1,1749:1:37:1749:66078:-5935.03,0
chrM	2706	.	A	G	39098.5	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1222;CIGAR=1X;DP=1235;DPB=1235;DPRA=0;EPP=3.188;EPPR=3.17734;GTI=0;LEN=1;MEANALT=1;MQM=59.9926;MQMR=60;NS=2;NUMALT=1;ODDS=2496.46;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=45333;QR=474;RO=13;RPL=507;RPP=79.8897;RPPR=3.17734;RPR=715;RUN=1;SAF=225;SAP=1062.06;SAR=997;SRF=5;SRP=4.51363;SRR=8;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:246:1,245:1:32:245:9136:-818.843,0	1:989:12,977:12:442:977:36197:-3215.79,0
chrM	3197	.	T	C	130583	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=3879;CIGAR=1X;DP=3892;DPB=3892;DPRA=0;EPP=59.2643;EPPR=3.17734;GTI=0;LEN=1;MEANALT=1;MQM=59.9938;MQMR=60;NS=2;NUMALT=1;ODDS=12192.8;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=146498;QR=463;RO=13;RPL=2190;RPP=143.521;RPPR=7.18621;RPR=1689;RUN=1;SAF=1459;SAP=519.999;SAR=2420;SRF=7;SRP=3.17734;SRR=6;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:1358:4,1354:4:150:1354:50893:-4563.63,0	1:2534:9,2525:9:313:2525:95605:-8569.92,0
chrM	3243	.	A	G	10397.4	.	AB=0;ABP=0;AC=1;AF=0.5;AN=2;AO=1365;CIGAR=1X;DP=3092;DPB=3092;DPRA=0;EPP=116.418;EPPR=46.9792;GTI=0;LEN=1;MEANALT=1;MQM=59.956;MQMR=59.8917;NS=2;NUMALT=1;ODDS=2394.09;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=49004;QR=63273;RO=1727;RPL=849;RPP=179.415;RPPR=105.14;RPR=516;RUN=1;SAF=443;SAP=368.01;SAR=922;SRF=637;SRP=261.033;SRR=1090;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:1035:341,694:341:12439:694:24871:-1118.32,0	0:2057:1386,671:1386:50834:671:24133:0,-2399.65
chrM	4769	.	A	G	48890.4	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1545;CIGAR=1X;DP=1545;DPB=1545;DPRA=0;EPP=162.63;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=51.4796;MQMR=0;NS=2;NUMALT=1;ODDS=4468.76;PAIRED=1;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=57939;QR=0;RO=0;RPL=463;RPP=541.537;RPPR=0;RPR=1082;RUN=1;SAF=890;SAP=80.6281;SAR=655;SRF=0;SRP=0;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:517:0,517:0:0:517:19209:-1665.33,0	1:1028:0,1028:0:0:1028:38730:-3308.01,0
chrM	5539	.	A	G	3281.62	.	AB=0;ABP=0;AC=1;AF=0.5;AN=2;AO=379;CIGAR=1X;DP=787;DPB=787;DPRA=0;EPP=216.428;EPPR=224.5;GTI=0;LEN=1;MEANALT=1;MQM=54.1504;MQMR=53.777;NS=2;NUMALT=1;ODDS=755.62;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=13803;QR=15250;RO=408;RPL=74;RPP=308.741;RPPR=351.808;RPR=305;RUN=1;SAF=286;SAP=216.428;SAR=93;SRF=318;SRP=279.681;SRR=90;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	0:299:221,78:221:8252:78:2824:0,-485.016	1:488:187,301:187:6998:301:10979:-358.012,0
chrM	7028	.	C	T	70453.4	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=2109;CIGAR=1X;DP=2114;DPB=2114;DPRA=0;EPP=63.8084;EPPR=3.44459;GTI=0;LEN=1;MEANALT=1;MQM=55.9113;MQMR=59.2;NS=2;NUMALT=1;ODDS=6416.42;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=78988;QR=192;RO=5;RPL=1123;RPP=22.3353;RPPR=3.44459;RPR=986;RUN=1;SAF=969;SAP=33.1175;SAR=1140;SRF=3;SRP=3.44459;SRR=2;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:708:1,707:1:39:707:26465:-2367.64,0	1:1406:4,1402:4:153:1402:52523:-4692.5,0
chrM	7269	.	G	A	59443.2	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1773;CIGAR=1X;DP=1773;DPB=1773;DPRA=0;EPP=129.209;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=58.3328;MQMR=0;NS=2;NUMALT=1;ODDS=5563.82;PAIRED=1;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=66393;QR=0;RO=0;RPL=929;RPP=11.8591;RPPR=0;RPR=844;RUN=1;SAF=848;SAP=10.2718;SAR=925;SRF=0;SRP=0;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:619:0,619:0:0:619:23040:-2069.26,0	1:1154:0,1154:0:0:1154:43353:-3894.52,0
chrM	8860	.	A	G	48674.5	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1552;CIGAR=1X;DP=1556;DPB=1556;DPRA=0;EPP=36.1924;EPPR=5.18177;GTI=0;LEN=1;MEANALT=1;MQM=46.5528;MQMR=58.25;NS=2;NUMALT=1;ODDS=4986.6;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=56983;QR=146;RO=4;RPL=845;RPP=29.6556;RPPR=5.18177;RPR=707;RUN=1;SAF=844;SAP=28.8889;SAR=708;SRF=2;SRP=3.0103;SRR=2;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:598:0,598:0:0:598:21921:-1876.69,0	1:958:4,954:4:146:954:35062:-3002.8,0
chrM	9477	.	G	A	31596.7	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=944;CIGAR=1X;DP=946;DPB=946;DPRA=0;EPP=58.9901;EPPR=3.0103;GTI=0;LEN=1;MEANALT=1;MQM=59.3178;MQMR=60;NS=2;NUMALT=1;ODDS=2909.29;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=35363;QR=67;RO=2;RPL=521;RPP=25.1023;RPPR=7.35324;RPR=423;RUN=1;SAF=469;SAP=3.09311;SAR=475;SRF=1;SRP=3.0103;SRR=1;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:329:2,327:2:67:327:12108:-1080.66,0	1:617:0,617:0:0:617:23255:-2090.4,0
chrM	9548	.	G	A	23689.1	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=729;CIGAR=1X;DP=730;DPB=730;DPRA=0;EPP=65.6375;EPPR=5.18177;GTI=0;LEN=1;MEANALT=1;MQM=59.7133;MQMR=60;NS=2;NUMALT=1;ODDS=2252;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=26539;QR=38;RO=1;RPL=394;RPP=13.3792;RPPR=5.18177;RPR=335;RUN=1;SAF=339;SAP=10.7579;SAR=390;SRF=1;SRP=5.18177;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:261:1,260:1:38:260:9395:-838.704,0	1:469:0,469:0:0:469:17144:-1542,0
chrM	11467	.	A	G	157655	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=4755;CIGAR=1X;DP=4759;DPB=4759;DPRA=0;EPP=620.69;EPPR=3.0103;GTI=0;LEN=1;MEANALT=1;MQM=59.9394;MQMR=42.5;NS=2;NUMALT=1;ODDS=15848.9;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=179486;QR=151;RO=4;RPL=3558;RPP=2548.64;RPPR=3.0103;RPR=1197;RUN=1;SAF=1995;SAP=270.266;SAR=2760;SRF=2;SRP=3.0103;SRR=2;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:1820:0,1820:0:0:1820:68685:-6177.29,0	1:2939:4,2935:4:151:2935:110801:-9948.16,0
chrM	11719	.	G	A	86257.7	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=2676;CIGAR=1X;DP=2687;DPB=2687;DPRA=0;EPP=15.4873;EPPR=12.6832;GTI=0;LEN=1;MEANALT=1;MQM=59.6039;MQMR=60;NS=2;NUMALT=1;ODDS=6603.9;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=96531;QR=397;RO=11;RPL=1384;RPP=9.87851;RPPR=4.78696;RPR=1292;RUN=1;SAF=1292;SAP=9.87851;SAR=1384;SRF=2;SRP=12.6832;SRR=9;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:714:3,711:3:109:711:25643:-2282.19,0	1:1973:8,1965:8:288:1965:70888:-6347.71,0
chrM	12308	.	A	G	63434.1	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1900;CIGAR=1X;DP=1905;DPB=1905;DPRA=0;EPP=73.302;EPPR=3.44459;GTI=0;LEN=1;MEANALT=1;MQM=59.9621;MQMR=60;NS=2;NUMALT=1;ODDS=5056.51;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=71094;QR=184;RO=5;RPL=1169;RPP=222.265;RPPR=3.44459;RPR=731;RUN=1;SAF=907;SAP=11.463;SAR=993;SRF=3;SRP=3.44459;SRR=2;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:547:5,542:5:184:542:20196:-1799.85,0	1:1358:0,1358:0:0:1358:50898:-4577.25,0
chrM	12372	.	G	A	58077.1	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1737;CIGAR=1X;DP=1741;DPB=1741;DPRA=0;EPP=74.4189;EPPR=5.18177;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=2;NUMALT=1;ODDS=4910.27;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=64920;QR=149;RO=4;RPL=775;RPP=46.726;RPPR=3.0103;RPR=962;RUN=1;SAF=941;SAP=29.2942;SAR=796;SRF=3;SRP=5.18177;SRR=1;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:539:3,536:3:112:536:19922:-1781.72,0	1:1202:1,1201:1:37:1201:44998:-4043.68,0
chrM	13617	.	T	C	27663.7	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=846;CIGAR=1X;DP=848;DPB=848;DPRA=0;EPP=303.228;EPPR=3.0103;GTI=0;LEN=1;MEANALT=1;MQM=59.8972;MQMR=60;NS=2;NUMALT=1;ODDS=2549.91;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=31932;QR=78;RO=2;RPL=638;RPP=477.603;RPPR=7.35324;RPR=208;RUN=1;SAF=314;SAP=124.993;SAR=532;SRF=1;SRP=3.0103;SRR=1;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:284:0,284:0:0:284:10656:-956.335,0	1:564:2,562:2:78:562:21276:-1906.39,0
chrM	14766	.	C	T	54363.3	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1633;CIGAR=1X;DP=1633;DPB=1633;DPRA=0;EPP=21.2132;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=59.9853;MQMR=0;NS=2;NUMALT=1;ODDS=4408.47;PAIRED=1;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=61054;QR=0;RO=0;RPL=852;RPP=9.71354;RPPR=0;RPR=781;RUN=1;SAF=1113;SAP=470.614;SAR=520;SRF=0;SRP=0;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:473:0,473:0:0:473:17508:-1575.01,0	1:1160:0,1160:0:0:1160:43546:-3916.74,0
chrM	14793	.	A	G	52354.7	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=1605;CIGAR=1X;DP=1605;DPB=1605;DPRA=0;EPP=123.965;EPPR=0;GTI=0;LEN=1;MEANALT=1;MQM=59.9688;MQMR=0;NS=2;NUMALT=1;ODDS=4357.82;PAIRED=1;PAIREDR=0;PAO=0;PQA=0;PQR=0;PRO=0;QA=58638;QR=0;RO=0;RPL=950;RPP=120.75;RPPR=0;RPR=655;RUN=1;SAF=940;SAP=105.327;SAR=665;SRF=0;SRP=0;SRR=0;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:475:0,475:0:0:475:17313:-1557.59,0	1:1130:0,1130:0:0:1130:41325:-3717.26,0
chrM	15301	.	G	A	67202.5	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=2029;CIGAR=1X;DP=2036;DPB=2036;DPRA=0;EPP=73.6971;EPPR=4.45795;GTI=0;LEN=1;MEANALT=1.5;MQM=60;MQMR=60;NS=2;NUMALT=1;ODDS=5288.64;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=75132;QR=224;RO=6;RPL=962;RPP=14.8095;RPPR=3.0103;RPR=1067;RUN=1;SAF=1129;SAP=59.1336;SAR=900;SRF=1;SRP=8.80089;SRR=5;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:563:2,561:2:75:561:20714:-1856.3,0	1:1473:4,1468:4:149:1468:54418:-4880.92,0
chrM	15326	.	A	G	73749.6	.	AB=0;ABP=0;AC=2;AF=1;AN=2;AO=2200;CIGAR=1X;DP=2201;DPB=2201;DPRA=0;EPP=68.7112;EPPR=5.18177;GTI=0;LEN=1;MEANALT=1;MQM=60;MQMR=60;NS=2;NUMALT=1;ODDS=5718.37;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=82190;QR=38;RO=1;RPL=1013;RPP=32.8937;RPPR=5.18177;RPR=1187;RUN=1;SAF=1088;SAP=3.57883;SAR=1112;SRF=0;SRP=5.18177;SRR=1;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:597:1,596:1:38:596:22276:-2000.05,0	1:1604:0,1604:0:0:1604:59914:-5388.94,0
</code></pre></div></div>

<h2 id="filtering-variants">Filtering variants</h2>

<p>After filtering the data with stringent input parameters (restricting base quality to a minimum of 30 and <span class="notranslate">mapping</span> quality to a minimum of 20) a considerable amount of variants due to read-alignment bias exists. <a href="https://github.com/ekg">Erik Garrison</a> has a beautiful illustration of various biases potentially affecting called variants and making a locus sequence-able.</p>

<figure id="figure-3"><img src="../../images/mt_biases.png" alt="Various biases potentially affecting called variants" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Here you can see that in an ideal case (indicated with a green star) a variant is evenly represent by different areas of sequencing <span class="notranslate">reads</span> (cycle and placement biases) and is balanced across the two strands (strand bias). Allele imbalance is not applicable in our case as it reflects significant deviation from the diploid (50/50) expectation.</figcaption></figure>

<p>A robust tool set for processing VCF data is provided by <a href="https://github.com/vcflib/vcflib">vcflib</a> developed by Erik Garrison, the author of FreeBayes. One way to filter VCF is using <code class="language-plaintext highlighter-rouge">INFO</code> fields of the VCF dataset. If you look at the VCF dataset shown above you will see all comment lines beginning with <code class="language-plaintext highlighter-rouge">##INFO</code>.  These are <code class="language-plaintext highlighter-rouge">INFO</code> fields. Each VCF record contains a list of <code class="language-plaintext highlighter-rouge">INFO</code> tags describing a wide range of properties for each VCF record. You will see that FreeBayes and NVC differ significantly in the number and types of <code class="language-plaintext highlighter-rouge">INFO</code> fields each of these caller generates. This why the two require different filtering strategies.</p>

<p>Among numerous types of data generated by FreeBayes let‚Äôs consider the following variant properties:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">##INFO=&lt;ID=DP,Number=1,Type=Integer,Description="Total read depth at the locus"&gt;</code> - This is the number of <span class="notranslate">reads</span> covering a given site.</li>
  <li><code class="language-plaintext highlighter-rouge">##INFO=&lt;ID=SRP,Number=1,Type=Float,Description="Strand balance probability for the reference allele: Phred-scaled upper-bounds estimate of the probability of observing the deviation between SRF and SRR given E(SRF/SRR) ~ 0.5, derived using Hoeffding's inequality"&gt;</code> - The higher this quantity the better the site as it diminishes the chance of the site having significant strand bias.</li>
  <li><code class="language-plaintext highlighter-rouge">##INFO=&lt;ID=SAP,Number=A,Type=Float,Description="Strand balance probability for the alte<span class="notranslate">rna</span>te allele: Phred-scaled upper-bounds estimate of the probability of observing the deviation between SAF and SAR given E(SAF/SAR) ~ 0.5, derived using Hoeffding's inequality"&gt;</code> - The higher this quantity the better the site as it diminishes the chance of the site having significant strand bias  (also see <a href="https://groups.google.com/forum/#!topic/freebayes/fX4TOAqXJrA">here</a>).</li>
  <li><code class="language-plaintext highlighter-rouge">##INFO=&lt;ID=EPP,Number=A,Type=Float,Description="End Placement Probability: Phred-scaled upper-bounds estimate of the probability of observing the deviation between EL and ER given E(EL/ER) ~ 0.5, derived using Hoeffding's inequality"&gt;</code> - The higher this number the lower the chance of having significant placement bias.</li>
  <li><code class="language-plaintext highlighter-rouge">QUAL</code> - phred scaled variant quality.</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filtering-vcf-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filtering VCF data</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/vcffilter/vcffilter2/1.0.0_rc1+galaxy3" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/vcffilter/vcffilter2/1.0.0_rc1+galaxy3"><strong>VCFfilter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/vcffilter/vcffilter2/1.0.0_rc1+<span class="notranslate">galaxy</span>3</span></span>  to filter the variants with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúVCF dataset to filter‚Äù</em>: select the VCF dataset produced by <strong>FreeBayes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚Äúmore filters‚Äù</em>:
          <ul>
            <li>In <em>‚Äú1: more filters‚Äù</em>:
              <ul>
                <li><em>‚ÄúSelect the filter type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Info filter (-f)</code></li>
                <li><em>‚ÄúSpecify filtering value‚Äù</em>: <code class="language-plaintext highlighter-rouge">SRP &gt; 20</code></li>
              </ul>
            </li>
            <li>Insert more filters:</li>
            <li>In <em>‚Äú2: more filters‚Äù</em>:
              <ul>
                <li><em>‚ÄúSelect the filter type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Info filter (-f)</code></li>
                <li><em>‚ÄúSpecify filtering value‚Äù</em>: <code class="language-plaintext highlighter-rouge">SAP &gt; 20</code></li>
              </ul>
            </li>
            <li>Insert more filters:</li>
            <li>In <em>‚Äú3: more filters‚Äù</em>:
              <ul>
                <li><em>‚ÄúSelect the filter type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Info filter (-f)</code></li>
                <li><em>‚ÄúSpecify filtering value‚Äù</em>: <code class="language-plaintext highlighter-rouge">EPP &gt; 20</code></li>
              </ul>
            </li>
            <li>Insert more filters:</li>
            <li>In <em>‚Äú4: more filters‚Äù</em>:
              <ul>
                <li><em>‚ÄúSelect the filter type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Info filter (-f)</code></li>
                <li><em>‚ÄúSpecify filtering value‚Äù</em>: <code class="language-plaintext highlighter-rouge">QUAL &gt; 20</code></li>
              </ul>
            </li>
            <li>Insert more filters:</li>
            <li>In <em>‚Äú5: more filters‚Äù</em>:
              <ul>
                <li><em>‚ÄúSelect the filter type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Info filter (-f)</code></li>
                <li><em>‚ÄúSpecify filtering value‚Äù</em>: <code class="language-plaintext highlighter-rouge">DP &gt; 20</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-vcf-filter-options"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> VCF filter options</h3>
        <p>Filtering FreeBayes VCF for strand bias (<code class="language-plaintext highlighter-rouge">SPR</code> and <code class="language-plaintext highlighter-rouge">SAP</code>), placement bias (<code class="language-plaintext highlighter-rouge">EPP</code>), variant quality (<code class="language-plaintext highlighter-rouge">QUAL</code>), and depth of coverage (<code class="language-plaintext highlighter-rouge">DP</code>).</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>How many variants remain after filtering the VCF?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Two variants remain after filtering the VCF (your results might be different depending on the input and references you use):
        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO	FORMAT	raw_child-ds-	raw_mother-ds-
chrM	3243	.	A	G	10397.4	.	AB=0;ABP=0;AC=1;AF=0.5;AN=2;AO=1365;CIGAR=1X;DP=3092;DPB=3092;DPRA=0;EPP=116.418;EPPR=46.9792;GTI=0;LEN=1;MEANALT=1;MQM=59.956;MQMR=59.8917;NS=2;NUMALT=1;ODDS=2394.09;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=49004;QR=63273;RO=1727;RPL=849;RPP=179.415;RPPR=105.14;RPR=516;RUN=1;SAF=443;SAP=368.01;SAR=922;SRF=637;SRP=261.033;SRR=1090;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	1:1035:341,694:341:12439:694:24871:-1118.32,0	0:2057:1386,671:1386:50834:671:24133:0,-2399.65
chrM	5539	.	A	G	3281.62	.	AB=0;ABP=0;AC=1;AF=0.5;AN=2;AO=379;CIGAR=1X;DP=787;DPB=787;DPRA=0;EPP=216.428;EPPR=224.5;GTI=0;LEN=1;MEANALT=1;MQM=54.1504;MQMR=53.777;NS=2;NUMALT=1;ODDS=755.62;PAIRED=1;PAIREDR=1;PAO=0;PQA=0;PQR=0;PRO=0;QA=13803;QR=15250;RO=408;RPL=74;RPP=308.741;RPPR=351.808;RPR=305;RUN=1;SAF=286;SAP=216.428;SAR=93;SRF=318;SRP=279.681;SRR=90;TYPE=snp;technology.ILLUMINA=1	GT:DP:AD:RO:QR:AO:QA:GL	0:299:221,78:221:8252:78:2824:0,-485.016	1:488:187,301:187:6998:301:10979:-358.012,0
</code></pre></div>        </div>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="examining-the-results">Examining the results</h1>

<p>For visualizing VCFs <span class="notranslate">Galaxy</span> has two exte<span class="notranslate">rna</span>l tool options. The first is called <a href="http://vcf.iobio.io/">VCF.IOBIO</a> developed by <a href="http://marthlab.org/">Gabor Marth‚Äôs group</a> at the University of Utah. The second is called <a href="http://software.broadinstitute.org/software/igv/">IGV</a> developed by Broad Institute.</p>

<h2 id="visualising-with-vcfiobio">Visualising with VCF.IOBIO</h2>

<p>VCF.IOBIO can be invoked by expanding a VCF dataset in <span class="notranslate">Galaxy</span>‚Äôs history by clicking on it:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-display-data-in-vcfiobio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Display data in VCF.IOBIO</h3>

  <figure id="figure-4"><img src="../../images/mt_vcf_dataset_expanded.png" alt="Expanded output" /><figcaption><span class="figcaption-prefix">Figure 4:</span> At the bottom there is a link 'display at vcf.iobio'. Clicking on this link will start indexing of VCF datasets, which is required to display them.</figcaption></figure>

  <figure id="figure-5"><img src="../../images/mt_vcfiobio.png" alt="VCF.IOBIO overview" /><figcaption><span class="figcaption-prefix">Figure 5:</span> There are not that many variants to look at in this example. Nevertheless there are helpful statistics such as Transition/Transversion (Ts/Tn) ratio.</figcaption></figure>
</blockquote>

<h2 id="visualising-with-igv">Visualising with IGV</h2>

<p>Similarly to VCF.IOBIO expanding a history item representing a VCF dataset will reveal an IGV link:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-display-data-in-igv"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Display data in IGV</h3>

  <figure id="figure-6"><img src="../../images/mt_vcf_dataset_expanded.png" alt="VCFfilter output expanded" /><figcaption><span class="figcaption-prefix">Figure 6:</span> At the bottom there is a link 'display at IGV: local Human hg38'.</figcaption></figure>

  <p>The difference between ‚Äòlocal‚Äô and ‚ÄòHuman hg38‚Äô links is explained in the following video:</p>

  <div class="embed-responsive embed-responsive-16by9"><iframe src="https://player.vimeo.com/video/123414437?portrait=0" webkitallowfullscreen="" mozallowfullscreen="" allowfullscreen=""></iframe></div>

  <p>Visualizing our FreeBayes dataset will produce this:</p>

  <figure id="figure-7"><img src="../../images/mt_igv.png" alt="IGV interface" /><figcaption><span class="figcaption-prefix">Figure 7:</span> Here we focus on one particular variant at position 3243 for reasons that will become apparent in the next section.</figcaption></figure>
</blockquote>

<h2 id="comparing-frequencies">Comparing frequencies</h2>

<p>Visualizing VCF datasets is a good way to get an overall idea of the data, but it does not tell a lot of details. For example, above we have visualized site 3243 using IGV. It is interesting but we need to find out more. One thing we can do is to convert VCF dataset into a tab-delimited representation and with it play a bit more.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-convert-vcf-to-tab-delimited-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Convert VCF to tab-delimited data</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/vcf2tsv/1.0.0_rc1+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/vcf2tsv/1.0.0_rc1+galaxy0"><strong>VCFtoTab-delimited</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/vcf2tsv/vcf2tsv/1.0.0_rc1+<span class="notranslate">galaxy</span>0</span></span>  to filter the variants with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect VCF dataset to convert‚Äù</em>: select the VCF dataset produced by <strong>VCFfilter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúReport data per sample‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúFill empty fields with‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nothing</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>Why are there four rows in the VCFtoTab-delimited <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output when the input VCF had only two rows?</li>
    <li>How many columns are in the VCFtoTab-delimited <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> output?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Since <em>‚ÄúReport data per sample‚Äù</em> was set to <code class="language-plaintext highlighter-rouge">Yes</code>, VCFtoTab-delimited <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> produced two rows for each of the two variants, one row per sample (child and mother).</li>
      <li>There are 62 (!) columns in the output of VCFtoTab-delimited <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</li>
    </ol>
  </blockquote>
</blockquote>

<figure id="figure-8"><img src="../../images/mt_tab.png" alt="VCFtoTab-delimited output" /><figcaption><span class="figcaption-prefix">Figure 8:</span> The columns in the above dataset represent INFO and Genotype fields on the original VCF dataset.</figcaption></figure>

<p>Let‚Äôs restrict ourselves to a few key columns:</p>

<ul>
  <li>2 <code class="language-plaintext highlighter-rouge">POS</code> - position along mitochondrial genome</li>
  <li>4 <code class="language-plaintext highlighter-rouge">REF</code> - reference allele</li>
  <li>5 <code class="language-plaintext highlighter-rouge">ALT</code> - alte<span class="notranslate">rna</span>tive allele</li>
  <li>52 <code class="language-plaintext highlighter-rouge">SAMPLE</code> - name of the sample</li>
  <li>54 <code class="language-plaintext highlighter-rouge">AO</code> - number of alte<span class="notranslate">rna</span>tive observations (how many times do we see the alte<span class="notranslate">rna</span>tive allele at this position in this sample)</li>
  <li>55 <code class="language-plaintext highlighter-rouge">DP</code> - depth of coverage at this site for this sample</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-cut-columns-from-a-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Cut columns from a file</h3>

  <ol>
    <li>Use <span class="tool" data-tool="https://usegalaxy.eu/root?tool_id=Cut1" title="Tested with https://usegalaxy.eu/root?tool_id=Cut1"><strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: https://use<span class="notranslate">galaxy</span>.eu/root?tool_id=Cut1</span></span>  to select specific columns with the following parameters:
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c2,c4,c5,c52,c54,c55</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFrom‚Äù</em>: select the tabular dataset produced by <strong>VCFtoTab-delimited</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>Running Cut will generate the following dataset:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>POS  REF ALT     SAMPLE       AO    DP
--------------------------------------
3243	A	G	raw_child-ds-	694	1035
3243	A	G	raw_mother-ds-	671	2057
5539	A	G	raw_child-ds-	78	299
5539	A	G	raw_mother-ds-	301	488
</code></pre></div></div>

<p>Let‚Äôs look at position 3243. At this site, the mother sample has 671 <code class="language-plaintext highlighter-rouge">G</code>s (since <code class="language-plaintext highlighter-rouge">G</code> is the alte<span class="notranslate">rna</span>tive allele) and 2057-671=1386 <code class="language-plaintext highlighter-rouge">A</code>s. The child sample has 694 <code class="language-plaintext highlighter-rouge">G</code>s and 1035-694=341 <code class="language-plaintext highlighter-rouge">A</code>s:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Allele     A     G
-------------------
Mother  1386   691
Child   341    694
</code></pre></div></div>

<blockquote class="question">
  <h3 id="question-question-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>What do you notice about the relative frequencies of the <code class="language-plaintext highlighter-rouge">A</code> and <code class="language-plaintext highlighter-rouge">G</code> alleles between mother and child?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The <em>major</em> allele in the mother (<code class="language-plaintext highlighter-rouge">A</code>) becomes the <em>minor</em> allele in the child ‚Äì a remarkable frequency change due to mitochondrial bottleneck!</li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>This entire analysis is available as a <a href="https://usegalaxy.org/u/aun1/h/non-diploid-freebayes"><span class="notranslate">Galaxy</span> history</a> that you can import into your <span class="notranslate">Galaxy</span> account and play with.</p>

<p>Now you know how to call variants in non-diploid system, so try it on bacteria, viruses, etc.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Variation in human mitochondria is a proxy for other haploid systems.</p>
</li>
                            
                            <li><p>Nucleotide variants in haploid systems may have any frequency between 0 and 1.</p>
</li>
                            
                            <li><p>Distinguishing true variants from noise is not easy and requires accounting <br /> for strand bias and read placement bias.</p>
</li>
                            
                            <li><p>FreeBayes can be effectively used to call variants in haploid systems.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Kanagawa2003">Kanagawa, T., 2003 <b>Bias and artifacts in multitemplate polymerase chain reactions (PCR)</b>. Journal of Bioscience and Bioengineering 96: 317‚Äì323.</span>
    
    
    <a href="https://doi.org/https://doi.org/10.1016/S1389-1723(03)90130-7">https://doi.org/10.1016/S1389-1723(03)90130-7</a>
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="http://www.sciencedirect.com/science/article/pii/S1389172303901307">http://www.sciencedirect.com/science/article/pii/S1389172303901307</a>
      
    

</li>
<li><span id="Cree2008">Cree, L. M., D. C. Samuels, S. C. de Sousa Lopes, H. K. Rajasimha, P. Wonnapinij <i>et al.</i>, 2008 <b>A reduction of mitochondrial DNA molecules during embryogenesis explains the rapid segregation of genotypes</b>. Nature Genetics 40: 249‚Äì254.</span>
    
    
    <a href="https://doi.org/10.1038/ng.2007.63">10.1038/ng.2007.63</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Li2013">Li, H., 2013 <b>Aligning sequence reads, clone sequences and assembly contigs with BWA-MEM</b>.</span>
    
    
    

</li>
<li><span id="Rebolledo-Jaramillo2014">Rebolledo-Jaramillo, B., M. S.-W. Su, N. Stoler, J. A. McElhoe, B. Dickins <i>et al.</i>, 2014 <b>Maternal age effect and severe germ-line bottleneck in the inheritance of human mitochondrial DNA</b>. Proceedings of the National Academy of Sciences 111: 15474‚Äì15479.</span>
    
    
    <a href="https://doi.org/10.1073/pnas.1409328111">10.1073/pnas.1409328111</a>
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.pnas.org/content/111/43/15474">https://www.pnas.org/content/111/43/15474</a>
      
    

</li>
<li><span id="Olson2015">Olson, N. D., S. P. Lund, R. E. Colman, J. T. Foster, J. W. Sahl <i>et al.</i>, 2015 <b>Best practices for evaluating single nucleotide variant calling methods for microbial genomics</b>. Frontiers in Genetics 6: 235.</span>
    
    
    <a href="https://doi.org/10.3389/fgene.2015.00235">10.3389/fgene.2015.00235</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Tan2015">Tan, A., G. R. Abecasis, and H. M. Kang, 2015 <b>Unified representation of genetic variants</b>. Bioinformatics 31: 2202‚Äì2204.</span>
    
    
    <a href="https://doi.org/10.1093/bioinformatics/btv112">10.1093/bioinformatics/btv112</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Li2015">Li, M., R. Schr√∂der, S. Ni, B. Madea, and M. Stoneking, 2015 <b>Extensive tissue-related and allele-related mtDNA heteroplasmy suggests positive selection for somatic mutations</b>. Proceedings of the National Academy of Sciences 112: 2491‚Äì2496.</span>
    
    
    <a href="https://doi.org/10.1073/pnas.1419651112">10.1073/pnas.1419651112</a>
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.pnas.org/content/112/8/2491">https://www.pnas.org/content/112/8/2491</a>
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Calling variants in non-diploid systems (Variant Analysis)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Anton Nekrutenko, Alex Ostrovsky, 2021 <b>Calling variants in non-diploid systems (Galaxy Training Materials)</b>. <a href="/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html">/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{variant-analysis-non-dip,
    author = "Anton Nekrutenko and Alex Ostrovsky",
    title = "Calling variants in non-diploid systems (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/variant-analysis/tutorials/non-dip/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Anton Nekrutenko, Alex Ostrovsky)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/non-dip/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
