---
layout: tutorial_slides
logo: "GTN"
title: EWAS Epigenome-Wide Association Studies Introduction
contributors:
 - kpoterlowicz
 - kpbioteam
---


# What is the topic?
Epigenome-wide association studies (EWAS) analyse genome-wide activity of epigenetic marks in  cohorts of different individuals to find associations between epigenetic variation and phenotype.

---

## Overview

- Applications of EWAS
- Infinium Human Methylation BeadChip
- The Analysis
- Integrative Approach
---
## Applications
- **Population Studies**
   - EWAS studies are the solution to explore gene-environment interactions on big scale
- **Clinical**
   - Genetic epidemiology is widely used in many progressive and incurable diseases i.e. cancers, autoimmune and neurological disorders

.image-75[ ![Graphic showing chromosome being unwound into chromatin and dna. DNA methylation is highlighted on individual dna molecules. The graphic shows more and less accessible DNA between histones in a 'histone tail'](../../images/Epigenetic_mechanisms.jpg)]

<small>[*National Institutes of Health, 2005*](http://commonfund.nih.gov/epigenomics/figure.aspx)</small>

---
### Infinium Human Methylation BeadChip
<small>One of the most common techniques used in EWAS studies is the Infinium Human Methylation BeadChip array quantifying the DNA methylation level, providing high accuracy with low input DNA requirements.</small>

.image-80[![At a methylated DNA locus, a methylated bead allows for transcription elongation while an unmethylated one doesn't. At an unmethylated site, the opposite happens.](../../images/methylationassay.png)]

<small>[*Illumina, 2018*](https://emea.illumina.com/)</small>

<small>The Infinium Methylation Assay uses two different bead types to detect changes in DNA methylation levels. In the figure we can see M - methylated and U - unmethylated bead types. Depending on the probe design, the bead signals are reported in different colors green or red.</small>
---
## The Analysis
.image-90[![.IDAT files are eread, preprocessed and normalised, QC happens, then genetic variation annotation, then DMP DMR finding, and finall functional annotation and visualisation](../../images/ewaschart1.png)]

<small>The analysis combines 6 main steps:
 1. Raw intensity data loading (.idat files)
 2. Preprocessing and optional normalization
 3. Quality assessment and control step
 4. Differentially methylated positions (DMP)  and regions (DMR) finding with respect to a phenotype covariate
 5. Functional annotation and graphical representation
</small>
.image-30[![Screenshot of the above workflow but as a Galaxy workflow](../../images/ewaschart2.png)]
---
## Integrative Approach
Methylation patterns obtained from Infinium Human Methylation BeadChip data are compatibile with other platforms (gene expression, microRNA profiling) and thus become one of the most comprehensive solution to invastigate epigenome.

---
## Summary
Epigenetic aberrations which involve DNA modifications give researchers an interest to identify novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and time consuming computational analysis.

Here, the Infinium Human Methylation BeadChip pipeline becomes a solution  for this inconvenience and provides a simplified downstream analysis  including preprocessing, quality evaluation and differentially methylated regions detection in one complex set of tools developed and published under the Galaxy platform. Ready to be integrated with existing resources and reused in multiple ways.
