---
layout: introduction_slides
topic_name: ewas-suite
tutorial_name: EWAS data analysis
logo: "GTN"
---

## EWAS Epigenome-Wide Association Studies Introduction

---

# What is the topic?
Epigenome-wide association studies (EWAS) analyse genome-wide activity of epigenetic marks in  cohorts of different individuals to find associations between epigenetic variation and phenotype.

---

## Overview

- Applications of EWAS
- The Data (450k what is it?)
- The Analysis
- Integrative Approach
---
## Applications
- **Population Studies**
   - EWAS studies are the solution to explore gene-environment interactions on big scale
- **Clinical**
   - Genetic epidemiology is widely used in many progressive and incurable diseases i.e. cancers, autoimmune and neurological disorders

.image-75[ ![](../../images/Epigenetic_mechanisms.jpg)]

<small>[*National Institutes of Health, 2005*](http://commonfund.nih.gov/epigenomics/figure.aspx)</small>

---
### 450k Data
<small>One of the most common techniques used in EWAS studies is the Infinium Methylation Assay quantifying the DNA methylation level of over 450 thousand loci, providing high accuracy with low input DNA requirements.</small>

.image-40[![](../../images/methylationassay.png)]

<small>[*Illumina, 2018*](https://emea.illumina.com/)</small>

<small>The Infinium Methylation Assay 450k uses two different bead types to detect changes in  DNA  methylation levels. In the figure we can see M - methylated and U - unmethylated bead types. Depending on the probe design, the bead signals are reported in different colors green or red.</small>
---
## The Analysis
.image-90[![](../../images/ewaschart1.png)]

<small>The analysis combines 6 main steps: 
 1. Raw intensity data loading (.idat files) 
 2. Preprocessing and optional normalization 
 3. Quality assessment and control  step
 4. Single nucleotide polymorphism finding and removal
 5. Differentially methylated positions (DMP)  and regions (DMR) finding with respect to a phenotype covariate
 6. Functional annotation and graphical representation
</small>
.image-30[![](../../images/ewaschart2.png)]
---
## Integrative Approach
Methylation patterns obtained from 450k data are compatibile with other platforms (gene expression, microRNA profiling) and thus become one of the most comprehensive solution to invastigate epigenome.

---
## Summary
Epigenetic aberrations which involve DNA modifications give researchers an interest to identify novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and  time consuming computational analysis.

Here, the EWAS suite becomes a solution  for this inconvenience and provides a simplified downstream analysis  including preprocessing, quality evaluation and differentially methylated regions detection in one complex set of tools developed and published under the Galaxy platform. Ready to be integrated with existing resources and reused in multiple ways.   
