---
layout: tutorial_hands_on
topic_name: epigenetics
tutorial_name: ewas_suite_analysis
---

# Introduction
{:.no_toc}

<!-- This is a comment. -->

The field of cancer genomics has demonstrated the power of massively parallel sequencing techniques to inform on the genes and specific alterations that drive tumor onset and progression. Although large comprehensive sequence data sets continue to be made increasingly available, data analysis remains an ongoing challenge, particularly for laboratories lacking dedicated resources and bioinformatics expertise. To address this, we have provide training based on  Galaxy tools ewas suite that represent many popular algorithms for detecting somatic genetic alterations from cancer genome and exome data.  

This exercise uses the dataset from the Cell publication by Hugo, Willy, et al., 2015 (DOI: 10.1016/j.cell.2015.07.061). The goal of this article was to identify differentially methylated regions and positions associated with melanoma MAPKi resistance.

To this end, Illumina 450K Methylation array  was performed in melanoma tumors from example patient pre and post MAPKi and BRAFi treatment with  different outcomes (sensitive and resistant). For each sample there is raw green and red colour arrays containing the summarized bead information generated bythe Illumina 450k scanner, i.e. there are 8 files in total:

Accession  | Sensitivity | Treatment
--- | ---  | ---
GSM1588704 | sensitive   | MAPKi
GSM1588705 | sensitive   | MAPKi
GSM1588706 | resistant   | BRAFi
GSM1588707 | resistant   | BRAFi

> ### Agenda
>
> 1. TOC
>{:toc}
> 
{: .agenda}

# Step 1: Illumina Array Data Loading

The first step of EWAS data anylalysis is raw methylation data loading (intensity information files  for each two color micro array)

> ### {% icon hands_on %} Hands-on: Data Loading
>
> 1. Create a new history for this tutorial and give it a proper name
> 2. Import Grn and Red .IDAT files from [Zenodo](https://zenodo.org/record/1251211#.WwREQ1Mvz-Y) or from the data library into the history
>
>
>     ### {% icon tip %} Tip: Importing data via links
>    
>     * Copy the link location
>     * Open the Galaxy Upload Manager
>     * Select **Paste/Fetch Data**
>     * Paste the link into the text field
>     * Press **Start** and **Close** the window
>     * Click on the `pencil` icon once the file is imported
>     * Click on **Datatype** in the middle panel
>     * Select `idat` as **New Type**
>    {: .tip}
>
>    ### {% icon tip %} Tip: Importing data from a data library
>    
>    * Go into "Shared data" (top panel) then "Data libraries"
>    * Click on "Training data" and then "EWAS suite analysis"
>    * Select interesting file
>    * Click on "Import selected datasets into history"
>    * Import in the history
>    {: .tip}
>
>    As default, Galaxy takes the link as name, so rename them.
>
> 3. Run **minfi read450k**  {% icon tool %} with red and green .IDAT files i.e. red files: GSM1588704_8795207135_R01C02_Red.idat, GSM1588705_8795207119_R05C02_Red.idat, GSM1588706_8795207135_R02C02_Red.idat, GSM1588707_8795207119_R06C02_Red.idat and green files: GSM1588704_8795207135_R01C02_Grn.idat, GSM1588705_8795207119_R05C02_Grn.idat, GSM1588706_8795207135_R02C02_Grn.idat, GSM1588707_8795207119_R06C02_Grn.idat
> 4. Inspect generated RGChannelSet class
{: .hands_on}
  
## Subpart 1

Short introduction about this subpart.

<!--
{% icon hands_on %} will render the hands_on icon as specified in
_config.yml in the root of this repository.
-->

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. Step2
>
>    > ### {% icon comment %} Comments
>    > A comment
>    {: .comment}
>
>    > ### {% icon tip %}Tip: A tip
>    >
>    > * Step1
>    > * Step2
>    {: .tip}
{: .hands_on}

## Subpart 2

Short introduction about this subpart.

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. Step2
>
>    > ### {% icon question %} Question
>    >
>    > Question?
>    >
>    > <details>
>    > <summary>Click to view answers</summary>
>    > Answer to question
>    > </details>
>    {: .question}
{: .hands_on}

Some blabla
> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. **My Tool** {% icon tool %} with the following parameters
>   - "param1" to the file `myfile`
>   - "param2" to `42`
>   - "param3" to `Yes`
>
>    > ### {% icon question %} Questions
>    >
>    > 1. Question1?
>    > 2. Question2?
>    >
>    >    <details>
>    >    <summary>Click to view answers</summary>
>    >    <ol type="1">
>    >    <li>Answer for question1</li>
>    >    <li>Answer for question2</li>
>    >    </ol>
>    >    </details>
>    {: .question}
>
> 3. Step3
{: .hands_on}

# Part 2

Short introduction about this subpart.

> ### {% icon comment %} Comment
>
> Do you want to learn more about the principles behind mapping? Follow our [training](../../NGS-mapping)
{: .comment}

# Conclusion
{:.no_toc}

Conclusion about the technical key points. And then relation between the technics and the biological question to end with a global view.
