<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Infinium Human Methylation BeadChip</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:title" content="Galaxy Training: Infinium Human Methylation BeadChip" />
        <meta property="og:description" content="DNA methylation is an epigenetic mechanism used by higher..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/epigenetics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Epigenetics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fepigenetics%2Ftutorials%2Fewas-suite%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/epigenetics/tutorials/ewas-suite/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Infinium Human Methylation BeadChip",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Epigenetics",
    "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
    "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
  },
  "courseCode": "epigenetics / ewas-suite / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Infinium Human Methylation BeadChip' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html",
  "timeRequired": "PT1H",
  "description": "The questions this  addresses are:\n - Which DNA regions and positions are diffrentialy methylated in pre MAPKi treatment and post MAPKi resistance Melanomas GSE65183?\n - How to analyse and visualise Infinium Human Methylation BeadChip's?\n\n\\nThe objectives are:\n - Learn how to perform reproducible Infinium Human Methylation BeadChip analysis\n - Visualise differentially methylated positions using UCSC browser\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Katarzyna Murat"
    },
    {
      "@type": "Person",
      "name": "Krzysztof Poterlowicz"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Katarzyna Murat"
    },
    {
      "@type": "Person",
      "name": "Krzysztof Poterlowicz"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Epigenetics",
      "description": "DNA methylation is an epigenetic mechanism used by higher eukaryotes and involved in e.g. gene expression, X-Chromosome inactivating, imprinting, and gene silencing of germline specific gene and repetitive elements.",
      "url": "https://training.galaxyproject.org//training-material/topics/epigenetics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3173",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3173",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3173"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Infinium Human Methylation BeadChip</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/kpbioteam/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/kpbioteam" alt="Katarzyna Murat">Katarzyna Murat</a>, <a href="/training-material/hall-of-fame/kpoterlowicz/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/kpoterlowicz" alt="Krzysztof Poterlowicz">Krzysztof Poterlowicz</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>Which DNA regions and positions are diffrentialy methylated in pre MAPKi treatment and post MAPKi resistance Melanomas GSE65183?</p>
</li>
            
            <li><p>How to analyse and visualise Infinium Human Methylation BeadChip‚Äôs?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Learn how to perform reproducible Infinium Human Methylation BeadChip analysis</p>
</li>
            
            <li><p>Visualise differentially methylated positions using UCSC browser</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/epigenetics/tutorials/ewas-suite/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/ewas-suite/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>
  <p>In this tutorial we will do:</p>
<ol id="markdown-toc">
  <li><a href="#raw-intensity-data-loading" id="markdown-toc-raw-intensity-data-loading">Raw intensity data loading</a></li>
  <li><a href="#idat-preprocessing" id="markdown-toc-idat-preprocessing">.idat preprocessing</a></li>
  <li><a href="#differentially-methylated-regions-and-positions-analysis" id="markdown-toc-differentially-methylated-regions-and-positions-analysis">Differentially methylated regions and positions analysis</a></li>
  <li><a href="#annotation-and-visualization" id="markdown-toc-annotation-and-visualization">Annotation and visualization</a></li>
</ol>
  <p>We will use a small subset of the original data. If we run the tutorial on the orginal dataset, analysis will be time consuming and not reproducible Infinium Human Methylation BeadChip computation on the orginal data can be found at <a href="https://github.com/kpbioteam/case_study">case study</a>.</p>

</blockquote>

<p>This tutorial is based on <a href="https://doi.org/10.1016/j.cell.2015.07.061">Hugo W, Shi H, Sun L, Piva M et al.: Non-genomic and Immune Evolution of Melanoma Acquiring MAPKi Resistance.</a>.</p>

<p>The data we use in this tutorial are available at <a href="https://zenodo.org/record/1251211">Zenodo</a>.</p>

<hr />

<h1 class="no_toc" id="introduction">Introduction</h1>

<p>The field of cancer genomics has demonstrated the power of massively parallel sequencing techniques to inform on genes and specific alterations that drive tumor onset and progression. Although large comprehensive sequence data sets continue to be made increasingly available, data analysis remains an ongoing challenge, particularly for laboratories lacking dedicated resources and bioinformatics expertise. To address this, we have provided training based on <span class="notranslate">Galaxy</span> Infinium Human Methylation BeadChip tool that represents many popular algorithms for detecting somatic genetic alterations from genome and exome data.</p>

<figure id="figure-1"><img src="../../images/Epigenetic_mechanisms.jpg" alt="epimechanism" /><figcaption><span class="figcaption-prefix">Figure 1:</span> How epigenetics mechanism can effect health (adapted from https://commonfund.nih.gov/epigenomics/figure)</figcaption></figure>

<p>This exercise uses datasets from the Cell publication by <a class="citation" href="#Hugo2015">Hugo <i>et al.</i> 2015</a>. with the goal being the identification of differentially methylated regions and positions associated with treatment resistant melanomas.
Datasets include the Infinium Human Methylation BeadChip array performed in melanoma tumors in a sample of patients pre and post MAPKi and BRAFi treatment with different outcomes (sensitive and resistant). For each sample there is raw green (methylated) and red (unmethylated) colour arrays containing the summarised bead information generated by the Infinium Human Methylation BeadChip scanner.</p>

<p>The Infinium Human Methylation BeadChip uses two different bead types to detect changes in <span class="notranslate">DNA</span> methylation levels. In the figure we can see M - methylated and U - unmethylated bead types. In our study unmethylated and methylated bead signals are reported as green and red colors respectively.</p>

<figure id="figure-2"><img src="../../images/methylationassay.png" alt="methassay" /><figcaption><span class="figcaption-prefix">Figure 2:</span> Infinium Methylation Assay Overview (adapted from <a class="citation" href="#Illumina">‚Äú<b>Illumina Infinium Methylation Assay Overview</b>‚Äù</a>)</figcaption></figure>

<table>
  <thead>
    <tr>
      <th>Accession</th>
      <th>Sensitivity</th>
      <th>Treatment</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GSM1588704</td>
      <td>baseline</td>
      <td>pre-treatment</td>
    </tr>
    <tr>
      <td>GSM1588705</td>
      <td>baseline</td>
      <td>pre-treatment</td>
    </tr>
    <tr>
      <td>GSM1588706</td>
      <td>resistant</td>
      <td>BRAFi</td>
    </tr>
    <tr>
      <td>GSM1588707</td>
      <td>resistant</td>
      <td>BRAFi</td>
    </tr>
  </tbody>
</table>

<p>The <span class="notranslate">workflow</span> combines 5 main steps, starting with raw intensity data loading (.idat) and then optional preprocessing and normalisation of the data. The next quality control step performs an additional sample check to remove low-quality data, which normalisation cannot detect. The <span class="notranslate">workflow</span> gives the user the opportunity to perform any of these preparation and data cleaning steps, including a highly recommended genetic variation annotation step resulting in single nucleotide polymorphism identification and removal. Finally, the dataset generated through all of these steps can be used to hunt (find) differentially-methylated positions (DMP) and regions (DMR) with respect to a phenotype covariate.</p>

<h1 id="raw-intensity-data-loading">Raw intensity data loading</h1>

<p>The first step of the Infinium Human Methylation BeadChip array analysis is raw methylation data loading (intensity information files for each two colour micro array)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-loading"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data Loading</h3>

  <p>Create a new history for this tutorial and give it a proper name</p>

  <blockquote class="tip">

    <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

    <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

    <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
    <ol>
      <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
      <li>Select the option <strong>Create New</strong> from the menu</li>
    </ol>
  </blockquote>

  <p>Import the following IDAT files from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> or from the data library (ask your instructor)</p>
  <ul>
    <li><code class="language-plaintext highlighter-rouge">GSM1588704_8795207135_R01C02_Red.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588705_8795207119_R05C02_Red.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588706_8795207135_R02C02_Red.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588707_8795207119_R06C02_Red.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588704_8795207135_R01C02_Grn.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588705_8795207119_R05C02_Grn.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588706_8795207135_R02C02_Grn.idat</code></li>
    <li><code class="language-plaintext highlighter-rouge">GSM1588707_8795207119_R06C02_Grn.idat</code></li>
  </ul>

  <blockquote class="details">
    <h3 id="details-list-of-zenodo-urls"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> List of Zenodo URLs</h3>
    <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Grn.idat
</code></pre></div>    </div>
  </blockquote>

  <blockquote class="tip">

    <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

    <ul>
      <li>Copy the link location</li>
      <li>
        <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
      </li>
      <li>Select <strong>Paste/Fetch Data</strong></li>
      <li>
        <p>Paste the link into the text field</p>
      </li>
      <li>
        <p>Press <strong>Start</strong></p>
      </li>
      <li><strong>Close</strong> the window</li>
    </ul>

    <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

  </blockquote>

  <blockquote class="tip">

    <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

    <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

    <ul>
      <li>
        <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
      </li>
      <li>
        <p>Find the correct folder (ask your instructor)</p>
      </li>
      <li>Select the desired files</li>
      <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
      <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
      <li>Click on <strong>Import</strong></li>
    </ul>
  </blockquote>

  <p>Run <strong>Infinium Human Methylation BeadChip</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
  <ul>
    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚Äúred channel files‚Äù</em>: all files ending in <code class="language-plaintext highlighter-rouge">_Red</code></li>
    <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚Äúgreen channel files‚Äù</em>: all files ending in <code class="language-plaintext highlighter-rouge">Grn</code>
<img src="../../images/1Raw_intensity_data_loading.png" alt="Raw_intensity_data_loading" /></li>
  </ul>
</blockquote>

<h1 id="idat-preprocessing">.idat preprocessing</h1>

<p>Preprocessing and data quality assurance is an important step in Infinium Methylation Assay analysis. Idat dataset represents two colour data with a green and a red channel and can be converted into methylated and unmethylated signals or into Beta values. The Infinium Human Methylation BeadChip tool extracts and plots the quality control data frame with two columns mMed and uMed which are the medians of methylation signals (Meth and Unmeth). Comparing them against one another allows users to detect and remove low-quality samples.</p>

<p><img src="../../images/quality_control.png" alt="quality_control" /></p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-preprocessing"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Preprocessing</h3>
  <p>Ilumina methylation array data can be mapped to the genome with or without additional preprocessing methods. Incomplete annotation of genetic variations such as single nucleotide polymorphism (SNP) may affect <span class="notranslate">DNA</span> measurements and disrupt <span class="notranslate">downstream</span> analysis of results.  <a class="citation" href="#Hansen">Aryee <i>et al.</i> 2014</a> It is highly recommended to remove the probes that contain either an SNP at the methylated loci interrogation or at the single nucleotide extension. In this tutorial we will remove probes affected by genetic variation by selecting <strong>(Optional) Preprocessing Method</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.
<img src="../../images/2idat_preprocessing.png" alt="idat_preprocessing" /></p>

</blockquote>

<blockquote class="comment">
  <h3 id="commentoptional-normalisation-of-the-data"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span>(optional) Normalisation of the data</h3>
  <p>If your files require normalisation, you might prefer to use one of the other preprocessing tools provided in Infinium Human Methylation BeadChip tool i.e. Preprocess Funnorm or Preprocess Quantile look for recommendation at <a class="citation" href="#Hansen">Aryee <i>et al.</i> 2014</a>.</p>

</blockquote>

<h1 id="differentially-methylated-regions-and-positions-analysis">Differentially methylated regions and positions analysis</h1>

<p>The main goal of the <strong>Infinium Human Methylation BeadChip</strong> analysis is to simplify the way differentially methylated loci sites are detected. The <strong>Infinium Human Methylation BeadChip</strong> pipeline contains differentially methylated positions (DMPs) detection with respect to a phenotype covariate, and more complex solutions for finding differentially methylated regions (DMRs). Genomic regions that are differentially methylated between two conditions can be tracked using a bumphunting algorithm. The algorithm first implements a t-statistic at each methylated loci location, with optional smoothing, then groups probes into clusters with a maximum location gap and a cutoff size to refer the lowest possible value of genomic profile hunted by our tool.</p>
<blockquote class="comment">
  <h3 id="comment-phenotype-table"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Phenotype table</h3>
  <p>Phenotype table can be in different sizes with different arguments, however the second column is required to contain phenotype covariate information for each sample.</p>
</blockquote>
<p>However, for the purpose of this tutorial we would like you to upload phenotype table from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> repository.</p>
<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-import-phenotypetabletxt-from-zenodo-or-data-library"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Import <code class="language-plaintext highlighter-rouge">phenotypeTable.txt</code> from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y">Zenodo</a> or data library:</h3>
  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>   https://zenodo.org/record/1251211/files/phenotypeTable.txt
</code></pre></div>  </div>
  <p><img src="../../images/3parameters.png" alt="pheno table" />
Set the following parameters:</p>
  <ul>
    <li><em>‚ÄúmaxGap Size‚Äù</em>:<code class="language-plaintext highlighter-rouge">250</code>
   We will use the default gap of 250 base pairs (bps), i.e. any two points more than 250 bps away are put in a new cluster.</li>
    <li><em>‚ÄúCutoff Size‚Äù</em>:<code class="language-plaintext highlighter-rouge">0.1</code>
   In order to find segments that are positive, near zero, and negative. We need a cutoff which is one number in which case ‚Äúnear zero‚Äù default 0.1</li>
  </ul>

  <p><img src="../../images/4parameters.png" alt="params" /></p>
  <ul>
    <li><em>‚ÄúNumber of Resamples‚Äù</em>:<code class="language-plaintext highlighter-rouge">0</code>
Default value 0 for permutation method apply selection of randomized cases with replacement from the original data while using ‚Äòbootstrap‚Äô method.</li>
    <li><em>‚ÄúnullMethod‚Äù</em>:<code class="language-plaintext highlighter-rouge">permutation</code>
Method used to generate null candidate regions, must be one of ‚Äòbootstrap‚Äô or
‚Äòpermutation‚Äô (defaults to ‚Äòpermutation‚Äô).</li>
    <li><em>‚ÄúPhenotype Type‚Äù</em>:<code class="language-plaintext highlighter-rouge">categorical</code>
Identify regions where methylation is associated with a continuous or categorical phenotype.</li>
  </ul>

  <p>Search for <code class="language-plaintext highlighter-rouge">UCSC Main</code> in the tool search bar (top left)
<img src="../../images/5parameters.png" alt="params" /></p>
  <ul>
    <li><em>‚ÄúqCutoff Size‚Äù</em>:<code class="language-plaintext highlighter-rouge">0.5</code>
Diffrentialy methylated positions with an FDR q-value greater than this value will not be returned.</li>
    <li><em>‚ÄúVariance Shrinkage‚Äù</em>:<code class="language-plaintext highlighter-rouge"> TRUE</code>
Default TRUE as it is recommended when sample sizes are small &lt;10</li>
    <li><em>‚ÄúGenome Table‚Äù</em>: <code class="language-plaintext highlighter-rouge">wgEncodeHaibMethyl450 ... </code>
Click on <code class="language-plaintext highlighter-rouge">UCSC Main</code> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>. You will be taken to the <strong>UCSC table browser</strong></li>
  </ul>
  <figure id="figure-3"><img src="../../images/5UCSC.png" alt="UCSC" /><figcaption><span class="figcaption-prefix">Figure 3:</span> UCSC</figcaption></figure>
  <p>Set the following options:
    - <em>‚Äúclade‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mammal</code>
    - <em>‚Äúgenome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human</code>
    - <em>‚Äúassembly‚Äù</em>: <code class="language-plaintext highlighter-rouge">Feb. 2009 (GRCh37/hg19)</code>
    - <em>‚Äúgroup‚Äù</em>: <code class="language-plaintext highlighter-rouge">Regulation</code>
    - <em>‚Äútrack‚Äù</em>: <code class="language-plaintext highlighter-rouge">HAIB Methyl450</code>
    - <em>‚Äútable‚Äù</em>: <code class="language-plaintext highlighter-rouge">GM12878 (wgEncodeHaibMethyl450Gm12878SitesRep1)</code>
    - <em>‚Äúregion‚Äù</em>: <code class="language-plaintext highlighter-rouge">genome</code>
    - <em>‚Äúoutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">GTF - gene transfer (limited)</code>
    - <em>‚ÄúSend output to‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">Galaxy</span></code> (only)
Click on the <strong>get output</strong> button at the bottom of the screen
On the next page, click on the <strong>Send Query to <span class="notranslate">Galaxy</span></strong> button
Wait for the upload to finish
We will now map the imported datasets against phenotype covariate and reference genome obtained from UCSC.
Click on the <strong>Differentially_Methylated_Positions.<span class="notranslate">bed</span></strong> output in your history to expand it.
Set the database build of your dataset to <code class="language-plaintext highlighter-rouge">Human Feb. 2009 (GRCh37/hg19) (hg19)</code>(if it is not set automatically)</p>
  <blockquote class="tip">

    <h3 id="tip-tip-changing-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing Database/Build (dbkey)</h3>
    <ul>
      <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
      <li>In the central panel, change the <strong>Database/Build</strong> field</li>
      <li>Select your desired database key from the dropdown list: <code class="language-plaintext highlighter-rouge">hg19</code></li>
      <li>Click the <strong>Save</strong> button</li>
    </ul>
  </blockquote>

  <p>Click on <code class="language-plaintext highlighter-rouge">display at UCSC</code> towards the bottom of the history item.
This will launch UCSC Genome Browser with your Custom Track</p>
</blockquote>

<figure id="figure-4"><img src="../../images/ucsc.png" alt="Display at UCSC" /><figcaption><span class="figcaption-prefix">Figure 4:</span> UCSC genome track showing differentialy methylated regions located on chromosome 6</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>How do we define phenotype covariate?</p>
  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <p>Phenotype covariate is the set of observable characteristics of an individual resulting from the gene-environment interactions</p>
  </blockquote>
</blockquote>

<h1 id="annotation-and-visualization">Annotation and visualization</h1>

<p>In addition to <span class="notranslate">downstream</span> analysis users can annotate the differentially methylated loci at the promoter regions of genes with gene function descriptions, and relationships between these concepts.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on--annotate-differentially-methylated-position"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on:  Annotate Differentially Methylated Position</h3>
  <ol>
    <li>Run <strong>chipeakanno annopeaks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>on the output of minfi_dmp with the following parameters
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDifferentialy methylated data‚Äù</em>: output of <strong>minfi dmp</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúbindingType‚Äù</em>: <code class="language-plaintext highlighter-rouge">StartSite</code></li>
        <li><em>‚ÄúbindingRegionStart‚Äù</em>:<code class="language-plaintext highlighter-rouge">-5000</code></li>
        <li><em>‚ÄúbindingRegionEnd‚Äù</em>:<code class="language-plaintext highlighter-rouge">3000</code></li>
        <li>
          <p><em>‚ÄúAdditional Column of Score‚Äù</em>:<code class="language-plaintext highlighter-rouge">8</code></p>

          <p>Position of column of score optional value if it is required</p>
        </li>
      </ul>
    </li>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the previous output adjusting the following parameters to cut ‚Äúgene_name‚Äù column from table of annotated peaks and then get a list of genes
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c16</code>hhh</li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFrom‚Äù</em>: output of <strong>chipeakanno annopeaks</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li><strong>Remove beginning</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> of <code class="language-plaintext highlighter-rouge">Gene List</code> with the following parameters
      <ul>
        <li><em>‚ÄúRemove first‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúfrom‚Äù</em>: output of <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>Run  <strong>clusterProfiler bitr</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on the previous output adjusting the following parameters to convert the list of genes to list of entrez ID
      <ul>
        <li><em>‚ÄúInput Type Gene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">SYMBOL</code></li>
        <li><em>‚ÄúOutput Type Gene ID‚Äù</em>: <code class="language-plaintext highlighter-rouge">ENTREZID</code></li>
      </ul>
    </li>
    <li>Use the output of the  clusterProfiler bitr <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to run a GO Enrichment Analysis using <strong>clusterProfiler go</strong></li>
  </ol>
</blockquote>

<figure id="figure-5"><img src="../../images/funcann.jpg" alt="Functional annotations" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Results of GO enrichments analysis for DMPs</figcaption></figure>

<table>
  <thead>
    <tr>
      <th>ID</th>
      <th>Description</th>
      <th>pvalue</th>
      <th>qvalue</th>
      <th>geneID</th>
      <th>Count</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>GO:0048732</td>
      <td>gland development</td>
      <td>1.38E-58</td>
      <td>4.23E-55</td>
      <td>PTGS2 / KCNC1 / FZD1 /SLC22A18 /SLC22A3 (‚Ä¶)</td>
      <td>372</td>
    </tr>
    <tr>
      <td>GO:1901652</td>
      <td>response to peptide</td>
      <td>3.99E-57</td>
      <td>8.13E-54</td>
      <td>SULF1/ LAMA5/ MED1 /CFLAR/ MSX2 (‚Ä¶)</td>
      <td>359</td>
    </tr>
    <tr>
      <td>GO:0048545</td>
      <td>response to steroid hormone</td>
      <td>1.38EE-54</td>
      <td>2.11E-51</td>
      <td>HDAC9/ RAB10/ CFLAR/ WDTC1 (‚Ä¶)</td>
      <td>394</td>
    </tr>
  </tbody>
</table>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>Epigenetic aberrations which involve <span class="notranslate">DNA</span> modifications give researchers an interest in identifying novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and time consuming computational analysis. Here, the EWAS suite becomes a solution for this inconvenience and provides a simplified <span class="notranslate">downstream</span> analysis available as a ready to run pipline in supplementary materials.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Infinium Human Methylation BeadChip is an array based technology to generate DNA methylation profiling at individual CpG loci in the human genome based on Illumina‚Äôs bead technology.</p>
</li>
                            
                            <li><p>Time and cost efficiency followed by high sample output, and overall quantitative accuracy and reproducibility made Infinium Human Methylation BeadChip one of the most widely used arrays on the market.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Illumina"><b>Illumina Infinium Methylation Assay Overview</b>.</span>
    
    Accessed: 2020-02-22.
    
    
    

</li>
<li><span id="Hansen">Aryee, M. J., A. E. Jaffe, H. Corrada-Bravo, C. Ladd-Acosta, A. P. Feinberg <i>et al.</i>, 2014 <b>Minfi: a flexible and comprehensive Bioconductor package for the analysis of Infinium DNA methylation microarrays</b>. Bioinformatics 30: 1363‚Äì1369.</span>
    
    
    <a href="https://doi.org/10.1093/bioinformatics/btu049">10.1093/bioinformatics/btu049</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li>
<li><span id="Hugo2015">Hugo, W., H. Shi, L. Sun, M. Piva, C. Song <i>et al.</i>, 2015 <b>Non-genomic and Immune Evolution of Melanoma Acquiring MAPKi Resistance</b>. Cell 162: 1271‚Äì1285.</span>
    
    
    <a href="https://doi.org/10.1016/j.cell.2015.07.061">10.1016/j.cell.2015.07.061</a>
    
    
      <!-- prevent duplicate doi links -->
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Infinium Human Methylation BeadChip (Epigenetics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Katarzyna Murat, Krzysztof Poterlowicz, 2021 <b>Infinium Human Methylation BeadChip (Galaxy Training Materials)</b>. <a href="/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html">/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{epigenetics-ewas-suite,
    author = "Katarzyna Murat and Krzysztof Poterlowicz",
    title = "Infinium Human Methylation BeadChip (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/epigenetics/tutorials/ewas-suite/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Katarzyna Murat, Krzysztof Poterlowicz)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/epigenetics/tutorials/ewas-suite/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
