<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Metaproteomics tutorial / Metaproteomics tutorial / Proteomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@b71986380d4874b98c2b7cd3c9a2fe9500f99672",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Metaproteomics tutorial">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-06-14 09:56:50 +0000">
<meta name="DC.creator" content="Timothy J. Griffin">
<meta name="DC.creator" content="Pratik Jagtap">
<meta name="DC.creator" content="James Johnson">
<meta name="DC.creator" content="Clemens Blank">
<meta name="DC.creator" content="Subina Mehta"><meta name="description" content="Training material for proteomics workflows in Galaxy">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Proteomics / Metaproteomics tutorial / Hands-on: Metaproteomics tutorial">
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/proteomics/tutorials/metaproteomics/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Metaproteomics tutorial",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-06-14 09:56:50 +0000",
  "datePublished": "2017-06-28 13:33:16 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00221",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
  },
  "abstract": "In this metaproteomics tutorial we will identify expressed proteins from a complex bacterial community sample.",
  "learningResourceType": "e-learning",
  "name": "Metaproteomics tutorial",
  "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html",
  "version": 41,
  "timeRequired": "PT2H",
  "teaches": "- A taxonomy and functional analysis of metaproteomic mass spectrometry data.",
  "keywords": [
    "Proteomics",
    "microgalaxy"
  ],
  "description": "## Abstract\n\nIn this metaproteomics tutorial we will identify expressed proteins from a complex bacterial community sample.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - How can I match metaproteomic mass spectrometry data to peptide sequences derived from shotgun metagenomic data?\n - How can I perform taxonomy analysis and visualize metaproteomics data?\n - How can I perform functional analysis on this metaproteomics data?\n\n\n## Learning Objectives\n\n- A taxonomy and functional analysis of metaproteomic mass spectrometry data.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/timothygriffin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/timothygriffin/",
      "name": "Timothy J. Griffin",
      "image": "https://avatars.githubusercontent.com/timothygriffin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6801-2559",
      "orcid": "https://orcid.org/0000-0001-6801-2559"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pratikdjagtap/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pratikdjagtap/",
      "name": "Pratik Jagtap",
      "image": "https://avatars.githubusercontent.com/pratikdjagtap",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-0984-0973",
      "orcid": "https://orcid.org/0000-0003-0984-0973"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/jj-umn/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/jj-umn/",
      "name": "James Johnson",
      "image": "https://avatars.githubusercontent.com/jj-umn",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ]
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/blankclemens/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/blankclemens/",
      "name": "Clemens Blank",
      "image": "https://avatars.githubusercontent.com/blankclemens",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-1726-2256",
      "orcid": "https://orcid.org/0000-0002-1726-2256"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/subinamehta/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/subinamehta/",
      "name": "Subina Mehta",
      "image": "https://avatars.githubusercontent.com/subinamehta",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-9818-0537",
      "orcid": "https://orcid.org/0000-0001-9818-0537"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3922",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3922",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3922"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3050",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3050",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3050"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0637",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0637",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0637"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.839701",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Proteomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/proteomics/faqs/" title="Check our FAQs for the Proteomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/metaproteomics/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-proteomics">
    <h1 data-toc-skip>Metaproteomics tutorial</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/timothygriffin/" class="contributor-badge contributor-timothygriffin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="/training-material/assets/images/avatar.png" alt="Timothy J. Griffin avatar" width="36" class="avatar" />
    Timothy J. Griffin</a><a href="/training-material/hall-of-fame/pratikdjagtap/" class="contributor-badge contributor-pratikdjagtap"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pratikdjagtap?s=36" alt="Pratik Jagtap avatar" width="36" class="avatar" />
    Pratik Jagtap</a><a href="/training-material/hall-of-fame/jj-umn/" class="contributor-badge contributor-jj-umn"><img src="https://avatars.githubusercontent.com/jj-umn?s=36" alt="James Johnson avatar" width="36" class="avatar" />
    James Johnson</a><a href="/training-material/hall-of-fame/blankclemens/" class="contributor-badge contributor-blankclemens"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/blankclemens?s=36" alt="Clemens Blank avatar" width="36" class="avatar" />
    Clemens Blank</a><a href="/training-material/hall-of-fame/subinamehta/" class="contributor-badge contributor-subinamehta"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/subinamehta?s=36" alt="Subina Mehta avatar" width="36" class="avatar" />
    Subina Mehta</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>How can I match metaproteomic mass spectrometry data to peptide sequences derived from shotgun metagenomic data?</p>
</li>
        
        <li><p>How can I perform taxonomy analysis and visualize metaproteomics data?</p>
</li>
        
        <li><p>How can I perform functional analysis on this metaproteomics data?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>A taxonomy and functional analysis of metaproteomic mass spectrometry data.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.839701">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/proteomics/tutorials/metaproteomics/workflows/" title="Metaproteomics tutorial workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/proteomics/tutorials/metaproteomics/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











<li class="btn btn-default supporting_material">


  <!-- dropdown with all recordings -->
  <a href="/training-material/topics/proteomics/tutorials/metaproteomics/recordings/" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this material in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
  </a>


  <ul class="dropdown-menu">
    

    
      
      
      
    <li><a class="dropdown-item" href="/training-material/topics/proteomics/tutorials/metaproteomics/recordings/index.html#tutorial-recording-15-february-2021" title="View the recording for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (February 2021) - 1h</a>
    </li>
      
    
    <li><a class="dropdown-item" href="/training-material/topics/proteomics/tutorials/metaproteomics/recordings/" title="View all recordings for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> View All</a>
    </li>

  </ul>

  
</li>
  





            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
    

    <!--
    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    -->
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Jun 28, 2017 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Jun 14, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00221">https://gxy.io/GTN:T00221</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">5.0</a> (0 recent ratings, 1 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 41 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>In this metaproteomics tutorial we will identify expressed proteins from a complex bacterial community sample.
For this MS/MS data will be matched to peptide sequences provided through a FASTA file.</p>

<p>Metaproteomics is the large-scale characterization of the entire protein complement of environmental microbiota
at a given point in time. It has the potential to unravel the mechanistic details of microbial interactions with
the host / environment by analyzing the functional dynamics of the microbiome.</p>

<p>In this tutorial, we will analyze a sample of sea water that was collected in August of 2013 from the Bering
Strait chlorophyll maximum layer (7m depth, 65° 43.44″ N, 168° 57.42″ W). The data were originally published in <span class="citation"><a href="#May_2016">May <i>et al.</i> 2016</a></span>.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#pretreatments" id="markdown-toc-pretreatments">Pretreatments</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
    </ol>
  </li>
  <li><a href="#analysis" id="markdown-toc-analysis">Analysis</a>    <ol>
      <li><a href="#match-peptide-sequences" id="markdown-toc-match-peptide-sequences">Match peptide sequences</a></li>
      <li><a href="#taxonomy-analysis" id="markdown-toc-taxonomy-analysis">Taxonomy analysis</a></li>
      <li><a href="#genus-taxonomy-level-summary" id="markdown-toc-genus-taxonomy-level-summary">Genus taxonomy level summary</a></li>
      <li><a href="#functional-analysis" id="markdown-toc-functional-analysis">Functional Analysis</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="pretreatments">Pretreatments</h1>

<h2 id="data-upload">Data upload</h2>

<p>There are three ways to upload your data.</p>

<ul>
  <li>Upload/Import the files from your computer</li>
  <li>Using a direct link</li>
  <li>Import from the data library if your instance provides the files</li>
</ul>

<p>In this tutorial, we will get the data from Zenodo: <a href="https://doi.org/10.5281/zenodo.839701"><a href="https://zenodo.org/badge/DOI/10.5281/zenodo.839701.svg" rel="noopener noreferrer"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.839701.svg" alt="DOI. " loading="lazy" /></a></a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-and-organization"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload and organization</div>

  <ol>
    <li>
      <p>Create a new history and name it something meaningful (e.g. <em>Metaproteomics tutorial</em>)</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the three MGF MS/MS files and the FASTA sequence file from Zenodo.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/839701/files/2016_Jan_12_QE2_45.mgf
https://zenodo.org/record/839701/files/2016_Jan_12_QE2_46.mgf
https://zenodo.org/record/839701/files/2016_Jan_12_QE2_47.mgf
https://zenodo.org/record/839701/files/FASTA_Bering_Strait_Trimmed_metapeptides_cRAP.fasta
https://zenodo.org/record/839701/files/Gene_Ontology_Terms.tabular
</code></pre></div>      </div>

      <p>As default, Galaxy takes the link as name.</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <ul>
          <li>Rename the datasets to a more descriptive name</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Build a <strong>Dataset list</strong> for the three MGF files</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a dataset collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create collection</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<p>We have a choice to run all these steps using a single workflow, then discuss each step and the results in more detail.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-pretreatments"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Pretreatments</div>

  <ol>
    <li>
      <p><strong>Import the workflow</strong> into Galaxy:</p>

      <!--SNIPPET-->
      <blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-gtn-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and launching a GTN workflow</div>   <div class="show-when-galaxy-proxy-active"> <span class="workflow" data-workflow="https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/proteomics-metaproteomics/versions/workflow">   Launch <strong>Pretreatments</strong> <i class="fas fa-share-alt" aria-hidden="true"></i> </span> (<a href="https://github.com/galaxyproject/training-material/blob/main/topics/proteomics/tutorials/metaproteomics/workflows/workflow.ga">View on GitHub</a>, <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/workflows/workflow.ga">Download workflow</a>) workflow. </div>   <div class="hide-when-galaxy-proxy-active"> Click to  <a href="https://my.galaxy.training/?path=/workflows/trs_import%3Frun_form=true%26trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/proteomics-metaproteomics/versions/workflow">     Launch <strong>Pretreatments</strong> <i class="fas fa-share-alt" aria-hidden="true"></i> </a> (<a href="https://github.com/galaxyproject/training-material/blob/main/topics/proteomics/tutorials/metaproteomics/workflows/workflow.ga">View on GitHub</a>, <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/workflows/workflow.ga">Download workflow</a>) </div>   <!--SNIPPET-->   <blockquote class="tip">   <div class="box-title tip-title" id="tip-if-this-does-not-work"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-if-this-does-not-work" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: If this does not work</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Paste the following URL into the box labelled <em>“Archived Workflow URL”</em>: <code class="language-plaintext highlighter-rouge">https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/workflows/workflow.ga</code></li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote>   <p><!--END_SNIPPET--></p> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Run <strong>Workflow</strong> <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> using the following parameters:</p>
      <ul>
        <li><em>“Send results to a new history”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“1: SixGill generated protein fasta file”</em>: <code class="language-plaintext highlighter-rouge">FASTA_Bering_Strait_Trimmed_metapeptides_cRAP.fasta</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“2: Dataset collection of Bering Strait MGF files”</em>: <code class="language-plaintext highlighter-rouge">Dataset collection of bering MGF</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“3: GeneOntology terms (selected)”</em>: <code class="language-plaintext highlighter-rouge">Gene_Ontology_terms.tabular</code></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Running a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
  </ol>

</blockquote>

<h1 id="analysis">Analysis</h1>

<h2 id="match-peptide-sequences">Match peptide sequences</h2>

<p>The search database labelled <code class="language-plaintext highlighter-rouge">FASTA_Bering_Strait_Trimmed_metapeptides_cRAP.FASTA</code> is the input database that
will be used to match MS/MS to peptide sequences via a sequence database search. It is a small excerpt of the original database, which was constructed based on a metagenomic screening of the sea water samples (see <span class="citation"><a href="#May_2016">May <i>et al.</i> 2016</a></span>). The full original database <a href="https://noble.gs.washington.edu/proj/metapeptide/data/metapeptides_BSt.fasta">is available online</a>. The contaminant database (cRAP) was merged with the original database.</p>

<p>For this, the sequence database-searching program called <a href="https://compomics.github.io/projects/searchgui.html">SearchGUI</a> will be used.
The created dataset collection of the three <em>MGF files</em> in the history is used as the MS/MS input.</p>

<h3 id="searchgui">SearchGUI</h3>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-searchgui"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: SearchGUI</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/search_gui/3.3.10.1" title="Search GUI tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Search GUI</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.3.10.1)</span> with the following parameters:
      <ul>
        <li><strong>Protein Database</strong>: <code class="language-plaintext highlighter-rouge">FASTA_Bering_Strait_Trimmed_metapeptides_cRAP.FASTA</code>(or however you named the <code class="language-plaintext highlighter-rouge">FASTA</code> file)</li>
        <li><strong>Input Peak lists (mgf)</strong>: <code class="language-plaintext highlighter-rouge">MGF files</code> dataset collection.</li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-select-dataset-collections-as-input"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-dataset-collections-as-input" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Select dataset collections as input</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>
            <p>Click the <strong>Dataset collection</strong> icon on the left of the input field:</p>

            <p><a href="../../images/dataset_button.png" rel="noopener noreferrer"><img src="../../images/dataset_button.png" alt="Dataset collection button. " width="168" height="86" loading="lazy" /></a></p>
          </li>
          <li>
            <p>Select the appropriate dataset collection from the list</p>
          </li>
        </ul>
      </blockquote>

      <p>Section <strong>Search Engine Options</strong>:</p>

      <ul>
        <li><strong>Search Engines</strong>: <code class="language-plaintext highlighter-rouge">X!Tandem</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>The section <strong>Search Engine Options</strong> contains a selection of sequence database searching
algorithms that are available in SearchGUI. Any combination of these programs can be used for
generating PSMs from MS/MS data. For the purpose of this tutorial, <strong>X!Tandem</strong> we will be used.</p>
      </blockquote>

      <p>Section <strong>Precursor Options</strong>:</p>

      <ul>
        <li><strong>Fragment Tolerance Units</strong>: <code class="language-plaintext highlighter-rouge">Daltons</code></li>
        <li><strong>Fragment Tolerance</strong>: <code class="language-plaintext highlighter-rouge">0.2</code>- this is high resolution MS/MS data</li>
        <li><strong>Maximum Charge</strong>: <code class="language-plaintext highlighter-rouge">6</code></li>
      </ul>

      <p>Section <strong>Protein Modification Options</strong>:</p>

      <ul>
        <li><strong>Fixed Modifications</strong>: <code class="language-plaintext highlighter-rouge">Carbamidomethylation of C</code></li>
        <li><strong>Variable modifications</strong>: <code class="language-plaintext highlighter-rouge">Oxidation of M</code></li>
      </ul>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-search-for-options"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-search-for-options" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Search for options</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <ul>
          <li>For selection lists, typing the first few letters in the window will filter the available options.</li>
        </ul>
      </blockquote>

      <p>Section <strong>Advanced Options</strong>:</p>
      <ul>
        <li><strong>X!Tandem Options</strong>: <code class="language-plaintext highlighter-rouge">Advanced</code></li>
        <li><strong>X!Tandem: Quick Acetyl</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><strong>X!Tandem: Quick Pyrolidone</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><strong>X!Tandem: Protein stP Bias</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>
          <p><strong>X!Tandem: Maximum Valid Expectation Value</strong>: <code class="language-plaintext highlighter-rouge">100</code></p>
        </li>
        <li>leave everything else as default</li>
      </ul>
    </li>
    <li>Click <strong>Run Tool</strong>.</li>
  </ol>

</blockquote>

<p>Once the database search is completed, the SearchGUI tool will output a file (called a
SearchGUI archive file) that will serve as an input for the next section, PeptideShaker.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>Note that sequence databases used for metaproteomics are usually much larger than the excerpt used in this tutorial. When using large databases, the peptide identification step can take much more time for computation. In metaproteomics, choosing the optimal database is a crucial step of your workflow, for further reading see <a href="https://www.ncbi.nlm.nih.gov/pubmed/27824341">Timmins-Schiffman et al (2017)</a>.</p>

  <p>To learn more about database construction in general, like integrating contaminant databases or using a decoy strategy for FDR searching, please consult our tutorial on <a href="/training-material/topics/proteomics/tutorials/database-handling/tutorial.html">Database Handling</a>.</p>

</blockquote>

<h3 id="peptideshaker">PeptideShaker</h3>

<p><a href="https://compomics.github.io/projects/peptide-shaker.html">PeptideShaker</a> is a post-processing software tool that
processes data from the SearchGUI software tool. It serves to organize the Peptide-Spectral
Matches (PSMs) generated from SearchGUI processing and is contained in the SearchGUI archive.
It provides an assessment of confidence of the data, inferring proteins identified from the
matched peptide sequences and generates outputs that can be visualized by users to interpret
results. PeptideShaker has been wrapped in Galaxy to work in combination with SearchGUI
outputs.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>There are a number of choices for different data files that can be generated using
PeptideShaker. A compressed file can be made containing all information needed to view the
results in the standalone PeptideShaker viewer. A <code class="language-plaintext highlighter-rouge">mzidentML</code> file can be created that contains
all peptide sequence matching information and can be utilized by compatible downstream
software. Other outputs are focused on the inferred proteins identified from the PSMs, as well
as phosphorylation reports, relevant if a phosphoproteomics experiment has been undertaken.
More detailed information on peptide inference using SearchGUI and PeptideShaker can be found in our tutorial on <a href="/training-material/topics/proteomics/tutorials/protein-id-sg-ps/tutorial.html">Peptide and Protein ID</a>.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-peptideshaker"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: PeptideShaker</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/peptideshaker/peptide_shaker/1.16.36.3" title="Peptide Shaker tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Peptide Shaker</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.16.36.3)</span> with the following parameters:
      <ul>
        <li><strong>Compressed SearchGUI results</strong>: The SearchGUI archive file</li>
        <li><strong>Specify Advanced PeptideShaker Processing Options</strong>: <code class="language-plaintext highlighter-rouge">Default Processing Options</code></li>
        <li><strong>Specify Advanced Filtering Options</strong>: <code class="language-plaintext highlighter-rouge">Advanced Filtering Options</code></li>
        <li><strong>Maximum Precursor Error Type</strong>: <code class="language-plaintext highlighter-rouge">Daltons</code></li>
        <li><strong>Specify Contact Information for mzIdendML</strong>: You can leave the default dummy options for now, but feel free to enter custom contact information.</li>
        <li><strong>Include the protein sequences in mzIdentML</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><strong>Output options</strong>: Select the <code class="language-plaintext highlighter-rouge">PSM Report</code> (Peptide-Spectral Match) and the <code class="language-plaintext highlighter-rouge">Certificate of Analysis</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-4"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>The <strong>Certificate of Analysis</strong> provides details on all the parameters
used by both SearchGUI and PeptideShaker in the analysis. This can be downloaded from the
Galaxy instance to your local computer in a text file if desired.</p>
      </blockquote>
    </li>
    <li>Click <strong>Run Tool</strong> and inspect the resulting files after they turned green with the <strong>View data</strong> icon:
 <a href="../../images/view_data_icon.png" rel="noopener noreferrer"><img src="../../images/view_data_icon.png" alt="View data button. " width="281" height="78" loading="lazy" /></a></li>
  </ol>

</blockquote>

<p>A number of new items will appear in your history, each corresponding to the outputs selected
in the PeptideShaker parameters. Most relevant for this tutorial is the PSM report:</p>

<figure id="figure-1" style="max-width: 90%;"><img src="../../images/psm_report.png" alt="Display of the PSM report tabular file. " width="1268" height="690" loading="lazy" /><a target="_blank" href="../../images/psm_report.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> The PSM report</figcaption></figure>

<p>Scrolling towards left will show the sequence for the PSM that matched to these
metapeptide entries. Column 3 is the sequence matched for each PSM entry. Every identified PSM is a
new row in the tabular output.</p>

<p>In the following steps of this tutorial, selected portions of this output will be extracted and used for
analysis of the taxonomic make-up of the sample as well as the biochemical functions
represented by the proteins identified.</p>

<h2 id="taxonomy-analysis">Taxonomy analysis</h2>

<p>In the previous section, the genome sequencing and mass spectrometry data from
processing of biological samples was used to identify peptides present in those samples.
Now those peptides are used as evidence to infer which organisms are represented in the sample,
and what biological functions those peptides and associated proteins suggest are occurring.</p>

<p>The UniProt organization collects and annotates all known proteins for organisms. A UniProt
entry includes the protein amino acid sequence, the NCBI taxonomy, and any annotations
about structure and function of the protein. The UniPept web resource developed
by Ghent University will be used to match the sample peptides to proteins. UniPept indexes all Uniprot
proteins and provides a fast matching algorithm for peptides.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-unipept"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Unipept</div>

  <p>Users can access UniPept via a <a href="https://unipept.ugent.be">web page</a> and paste peptide
sequences into the search form to retrieve protein information. But we’ll use the Galaxy
<em>Unipept</em> tool to automate the process. The <em>Unipept</em> tool sends the peptide list to the
UniPept REST API service, then transforms the results into datasets that can be further analyzed
or operated on within Galaxy.</p>
</blockquote>

<h3 id="recieving-the-list-of-peptides-query-tabular">Recieving the list of peptides: Query Tabular</h3>

<p>In order to use <em>Unipept</em>, a list containing the peptide sequences has to be generated.
The tool <strong>Query Tabular</strong> can load tabular data (the PSM report in this case) into a SQLite data base.
As a tabular file is being read, line filters may be applied and an SQL query can be performed.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-query-tabular"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Query Tabular</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0" title="Query Tabular tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Query Tabular</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.0.0)</span> with the following parameters:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code>:</li>
        <li><strong>Tabular Dataset for Table</strong>: The PSM report</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="language-plaintext highlighter-rouge">^\d</code></li>
            <li><strong>action for regex match</strong>: <code class="language-plaintext highlighter-rouge">include line on pattern match</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">psm</code></li>
        <li>
          <p><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">id,,sequence,,,,,,,,,,,,,,,,,,,,confidence,validation</code></p>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-5"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

            <p>By default, table columns will be named: c1,c2,c3,…,cn (column names for a table must be unique).
You can override the default names by entering a comma separated list of names, e.g. <code class="language-plaintext highlighter-rouge">,name1,,,name2</code>
would rename the second and fifth columns.</p>

            <p>Check your input file to find the settings which best fits your needs.</p>
          </blockquote>
        </li>
        <li>
          <p><strong>Only load the columns you have named into database</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>Save the sqlite database in your history</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-querying-sqlite-databases"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Querying SQLite Databases</div>

            <ul>
              <li><strong>Query Tabular</strong> can also use an existing SQLite database. Activating <code class="language-plaintext highlighter-rouge">Save the sqlite database in your history</code>
will store the created database in the history, allowing to reuse it directly.</li>
            </ul>

          </blockquote>
        </li>
        <li>
          <p><strong>SQL Query to generate tabular output</strong>:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT distinct sequence

FROM psm

WHERE confidence &gt;= 95

ORDER BY sequence
</code></pre></div>          </div>
        </li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <p>The SQL query might look confusing at first, but having a closer look should clarify a lot.</p>

        <ol>
          <li>What does <code class="language-plaintext highlighter-rouge">FROM psm</code> mean?</li>
          <li>What need to be changed if we only want peptides with a confidence higher then 98%?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
          <ol>
            <li>We want to read from table “psm”. We defined the name before in the “Specify Name for Table” option.</li>
            <li>We need to change the value in line 3: “WHERE validation IS NOT ‘Confident’ AND confidence &gt;= 98”</li>
          </ol>
        </blockquote>
      </blockquote>

      <ul>
        <li><strong>include query result column headers</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong> and inspect the query results file after it turned green. If everything went well, it should look similiar:</p>

      <figure id="figure-2" style="max-width: 90%;"><img src="../../images/query_tabular_1.png" alt="Query Tabular output showing the peptides. " width="590" height="460" loading="lazy" /><a target="_blank" href="../../images/query_tabular_1.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Query Tabular output</figcaption></figure>
    </li>
  </ol>

</blockquote>

<p>While we can proceed with this list of peptides, let’s practice using the created SQLite database for further queries.
We might not only be interested in all the distinct peptides, but also on how many PSMs a single peptide had.
Therefore we can search the database for the peptides and count the occurrence without configuring the tables and columns again:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-sqlite-to-tabular"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: SQLite to tabular</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/sqlite_to_tabular/sqlite_to_tabular/2.0.0" title="SQLite to tabular tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SQLite to tabular</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.0)</span> with the following parameters:</p>

      <ul>
        <li>
          <p><strong>SQL Query</strong>:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT sequence as "peptide", count(id) as "PSMs"

FROM psm

WHERE confidence &gt;= 95

GROUP BY sequence

ORDER BY sequence
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong>. The resulting file should have two columns, one with the distinct peptides, the other with the count number of PSMs.</p>
    </li>
  </ol>

</blockquote>

<h3 id="retrieve-taxonomy-for-peptides-unipept">Retrieve taxonomy for peptides: Unipept</h3>

<p>The generated list of peptides can now be used to search via <em>Unipept</em>.
We do a taxonomy analysis using the UniPept pept2lca function to return the taxonomic lowest common ancestor for each peptide:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-unipept"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Unipept</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.3.0" title="Unipept tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Unipept</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.3.0)</span> with the following parameters:</p>

      <ul>
        <li><strong>Unipept application</strong>: <code class="language-plaintext highlighter-rouge">pept2lca: lowest common ancestor</code></li>
        <li><strong>Peptides input format</strong>: <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li><strong>Tabular Input Containing Peptide column</strong>: The query results file.</li>
        <li><strong>Select column with peptides</strong>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><strong>Choose outputs</strong>: Select <code class="language-plaintext highlighter-rouge">tabular</code> and <code class="language-plaintext highlighter-rouge">JSON taxonomy tree</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong>. The history should grow by two files. View each to see the difference.</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-6"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

        <p>The JSON (JavaScript Object Notation) file contains the same information as the tabular file but is not comfortably human readable.
Instead, we can use it to use JavaScript libraries to visualize this data.</p>
      </blockquote>
    </li>
    <li>
      <p>Visualize the data:</p>

      <ul>
        <li>
          <p>Click on the JSON output file from the <em>Unipept</em> tool to expand it. Click on the <strong>Visualize</strong> button and select <strong>Unipept Tree viewer</strong>:</p>

          <p><a href="../../images/visualize_button.png" rel="noopener noreferrer"><img src="../../images/visualize_button.png" alt="Visualize button. " width="289" height="223" loading="lazy" /></a></p>
        </li>
        <li>
          <p>A new window should appear with a visualization of the taxonomy tree of your data. Use the mouse wheel to scroll in and out and click on nodes to expand or collapse them:</p>

          <figure id="figure-3" style="max-width: 90%;"><img src="../../images/unipept_tree_viewer.png" alt="Unipept Tree viewer visual output. " width="1263" height="665" loading="lazy" /><a target="_blank" href="../../images/unipept_tree_viewer.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Interactive visualization from the Unipept Tree viever plugin</figcaption></figure>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="genus-taxonomy-level-summary">Genus taxonomy level summary</h2>

<p>The tabular <em>Unipept</em> output lists the taxonomy assignments for each peptide. To create a meaningful summary, the <strong>Query Tabular</strong> tool is
once again used, aggregating the number of peptides and PSMs for each genus level taxonomy assignment:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-query-tabular-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Query Tabular</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0" title="Query Tabular tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Query Tabular</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.0.0)</span> with the following parameters:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The PSM report</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="language-plaintext highlighter-rouge">^\d</code></li>
            <li><strong>action for regex match</strong>: <code class="language-plaintext highlighter-rouge">include line on pattern match</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">psm</code></li>
        <li>
          <p><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">,,sequence,,,,,,,,,,,,,,,,,,,,confidence,validation</code></p>
        </li>
        <li><strong>Only load the columns you have named into database</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have a second <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <strong>Unipept</strong> <code class="language-plaintext highlighter-rouge">tabular</code>/<code class="language-plaintext highlighter-rouge">tsv</code> output</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="language-plaintext highlighter-rouge">#peptide</code></li>
            <li><strong>action for regex match</strong>: <code class="language-plaintext highlighter-rouge">exclude line on pattern match</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">lca</code></li>
        <li>
          <p><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">peptide,,,,,,,,,,,,,,,,,,,,,genus</code></p>
        </li>
        <li>
          <p><strong>Only load the columns you have named into database</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>Save the sqlite database in your history</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>SQL Query to generate tabular output</strong>:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT lca.genus,count(psm.sequence) as "PSMs",count(distinct psm.sequence) as "DISTINCT PEPTIDES"

FROM psm LEFT JOIN lca ON psm.sequence = lca.peptide

WHERE confidence &gt;= 95

GROUP BY lca.genus

ORDER BY PSMs desc, 'DISTINCT PEPTIDES' desc
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong> and inspect the query results file after it turned green:</p>

      <figure id="figure-4" style="max-width: 90%;"><img src="../../images/metaproteomics_summary.png" alt="Query Tabular output showing gene, PSMs and distinct peptides. " width="399" height="495" loading="lazy" /><a target="_blank" href="../../images/metaproteomics_summary.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Query Tabular output</figcaption></figure>
    </li>
  </ol>

</blockquote>

<h2 id="functional-analysis">Functional Analysis</h2>

<p>Recent advances in microbiome research indicate that functional characterization via metaproteomics analysis has the potential to accurately
measure the microbial response to perturbations. In particular, metaproteomics enables the estimation of the function of the microbial
community based on expressed microbial proteome.</p>

<p>In the following chapter, a functional analysis will be performed using the <strong>UniPept</strong> application <code class="language-plaintext highlighter-rouge">pept2prot</code> in order to match the list of peptides with the correlated Gene Ontology terms.
This allows to get an insight of the <strong>biological process</strong>, the <strong>molecular function</strong> and the <strong>cellular component</strong> related to the sample data.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-gene-ontology-go-consortium"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Gene Ontology (GO) Consortium</div>

  <p>The <a href="http://www.geneontology.org/">Gene Ontology Consortium</a> provides with its Ontology a framework for the model of biology.
The GO defines concepts/classes used to describe gene function, and relationships between these concepts. It classifies functions along three aspects:</p>

  <ul>
    <li>
      <p><strong>molecular function</strong></p>

      <ul>
        <li>molecular activities of gene products</li>
      </ul>
    </li>
    <li>
      <p><strong>cellular component</strong></p>

      <ul>
        <li>where gene products are active</li>
      </ul>
    </li>
    <li>
      <p><strong>biological process</strong></p>

      <ul>
        <li>pathways and larger processes made up of the activities of multiple gene products.</li>
      </ul>
    </li>
  </ul>

  <p><a href="http://geneontology.org/page/ontology-documentation">more information</a></p>

</blockquote>

<h3 id="data-upload-1">Data upload</h3>

<p>For this tutorial, a tabular file containing the relevant GO terms has been created. It contains the GO aspect, the ID and the name.
It is available at Zenodo: <a href="https://doi.org/10.5281/zenodo.839701"><a href="https://zenodo.org/badge/DOI/10.5281/zenodo.839701.svg" rel="noopener noreferrer"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.839701.svg" alt="DOI. " loading="lazy" /></a></a>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Import the file <code class="language-plaintext highlighter-rouge">Gene_Ontology_Terms.tabular</code> from Zenodo.</p>

      <blockquote class="tip">
        <div class="box-title tip-title" id="tip-setting-file-metadata-on-upload"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-setting-file-metadata-on-upload" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Setting file metadata on upload</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

        <p>In the upload window of Galaxy you can set the filetype and related genome of the file you’re uploading in the corresponding columns beforehand.
This might be handy if the automatic detection of the filetype didn’t work out perfectly or if you want to avoid setting the genome later on, especially for multiple files.</p>

      </blockquote>

      <p>As default, Galaxy takes the link as name.</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-7"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <ul>
          <li>Rename the datasets to a more descriptive name, e.g. <code class="language-plaintext highlighter-rouge">Gene Ontology Terms</code></li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-creating-your-own-gene-ontology-list"><button class="gtn-boxify-button details" type="button" aria-controls="details-creating-your-own-gene-ontology-list" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Creating your own Gene Ontology list</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The latest Gene Ontology can be downloaded <a href="http://geneontology.org/page/download-ontology">the GO website</a> as a text file in the <code class="language-plaintext highlighter-rouge">OBO</code> format.
<code class="language-plaintext highlighter-rouge">OBO</code> files are human-readable (in addition to machine-readable) and can be opened in any text editor. They contain more information than just the name and aspect.</p>

  <p>In order to receive a file like we use in the tutorial for your own analysis, different tools are available to extract information from <code class="language-plaintext highlighter-rouge">OBO</code> files,
one of them being ONTO-PERL (<span class="citation"><a href="#Antezana_2008">Antezana <i>et al.</i> 2008</a></span>).
An example file with all GO terms from 08.07.2017 named <code class="language-plaintext highlighter-rouge">Gene_Ontology_Terms_full_07.08.2017.tabular</code> can be found on the <a href="https://doi.org/10.5281/zenodo.839701">Zenodo repository</a> of this tutorial as well.
You could also upload the Gene Ontology Terms by copying this link on to the Upload Data - Paste/Fetch data <code class="language-plaintext highlighter-rouge">https://zenodo.org/record/839701/files/Gene_Ontology_Terms_full_07.08.2017.tabular</code></p>

</blockquote>

<h3 id="retrieve-go-ids-for-peptides-unipept">Retrieve GO IDs for peptides: Unipept</h3>

<p>The <strong>UniPept</strong> application <code class="language-plaintext highlighter-rouge">pept2prot</code> can be used to return the list of proteins containing each peptide.
The option <code class="language-plaintext highlighter-rouge">retrieve extra information</code> option is set to <code class="language-plaintext highlighter-rouge">yes</code> so that we retrieve Gene Ontology assignments (<code class="language-plaintext highlighter-rouge">go_references</code>)
for each protein.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-unipept-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Unipept</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/unipept/unipept/4.3.0" title="Unipept tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Unipept</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.3.0)</span> with the following parameters:</p>

      <ul>
        <li><strong>Unipept application</strong>: <code class="language-plaintext highlighter-rouge">pept2prot: UniProt entries containing a given tryptic peptide</code></li>
        <li><strong>retrieve extra information</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><strong>Peptides input format</strong>: <code class="language-plaintext highlighter-rouge">tabular</code></li>
        <li><strong>Tabular Input Containing Peptide column</strong>: The first query results file.</li>
        <li><strong>Select column with peptides</strong>: <code class="language-plaintext highlighter-rouge">Column 1</code></li>
        <li><strong>Choose outputs</strong>: Select <code class="language-plaintext highlighter-rouge">tabular</code></li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong>.</p>
    </li>
    <li>
      <p>inspect the result:</p>

      <ul>
        <li>The output should be a tabular file containing a column labeled <code class="language-plaintext highlighter-rouge">go_references</code>. This is what we’re looking for.</li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="combine-all-information-to-quantify-the-go-results">Combine all information to quantify the GO results</h3>

<p>As a final step we will use <strong>Query Tabular</strong> in a more sophisticated way to combine all information to quantify the GO analysis. The three used file and the extracted information are:</p>

<ul>
  <li><strong>Gene Ontology Terms</strong>:
    <ul>
      <li><code class="language-plaintext highlighter-rouge">go_id</code> to match with <strong>Normalized UniPept output</strong></li>
      <li>The GO <code class="language-plaintext highlighter-rouge">aspect</code> to group the results in three separate files</li>
      <li>The GO <code class="language-plaintext highlighter-rouge">description</code> to annotate the results</li>
    </ul>
  </li>
  <li><strong>Normalized UniPept output</strong>:
    <ul>
      <li><code class="language-plaintext highlighter-rouge">peptide</code> to match with <strong>PSM Report</strong> and to count distinct peptides per GO term</li>
      <li><code class="language-plaintext highlighter-rouge">go_reference</code> to match with <strong>Gene Ontology Terms</strong></li>
    </ul>
  </li>
  <li><strong>PSM Report</strong>:
    <ul>
      <li><code class="language-plaintext highlighter-rouge">sequence</code> to match with <strong>Normalized UniPept output</strong></li>
      <li><code class="language-plaintext highlighter-rouge">id</code> to count distinct PSM’s per GO term</li>
    </ul>
  </li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-query-tabular-2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Query Tabular</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/query_tabular/query_tabular/3.0.0" title="Query Tabular tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Query Tabular</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.0.0)</span> with the following parameters:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <code class="language-plaintext highlighter-rouge">Gene Ontology Terms</code> file</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">skip leading lines</code>
          <ul>
            <li><strong>Skip lines</strong>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">go</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">aspect,go_id,description</code></li>
        <li><strong>Table Index</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="language-plaintext highlighter-rouge">aspect,go_id</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have a second <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <strong>Unipept</strong> <code class="language-plaintext highlighter-rouge">tabluar</code>/<code class="language-plaintext highlighter-rouge">tsv</code> output</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">skip leading lines</code>
          <ul>
            <li><strong>Skip lines</strong>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">prepend a line number column</code></li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">bering_prot</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">id,peptide,uniprot_id,taxon_id,taxon_name,ec_references,go_references,refseq_ids,refseq_protein_ids,insdc_ids,insdc_protein_ids</code></li>
        <li><strong>Table Index</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="language-plaintext highlighter-rouge">id,peptide</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have another <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The same <strong>Unipept</strong> <code class="language-plaintext highlighter-rouge">tabluar</code>/<code class="language-plaintext highlighter-rouge">tsv</code> output</li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">skip leading lines</code>
          <ul>
            <li><strong>Skip lines</strong>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">prepend a line number column</code></li>
        <li>Add another Filter: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">select columns</code>
          <ul>
            <li><strong>enter column numbers to keep</strong>: <code class="language-plaintext highlighter-rouge">1,7</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">normalize list columns, replicates row for each item in list</code>
          <ul>
            <li><strong>enter column numbers to normalize</strong>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><strong>List item delimiter in column</strong>: ` ` (a single blank character)</li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-8"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
        <ul>
          <li>The UniPept result file can contain multiple GO IDs in a single row. In order to create a normalized table of this data, these rows will be split so each record contains only one GO ID.</li>
        </ul>
      </blockquote>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">bering_prot_go</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">id,go_reference</code></li>
        <li><strong>Table Index</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="language-plaintext highlighter-rouge">go_reference,id</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Repeat this step to have another <strong>Database Table</strong>:</p>

      <ul>
        <li><strong>Database Table</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Database Table</code></li>
        <li><strong>Tabular Dataset for Table</strong>: The <code class="language-plaintext highlighter-rouge">PSM Report</code></li>
      </ul>

      <p>Section <strong>Filter Dataset Input</strong>:</p>

      <ul>
        <li><strong>Filter Tabular Input Lines</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">by regex expression matching</code>
          <ul>
            <li><strong>regex pattern</strong>: <code class="language-plaintext highlighter-rouge">^\d</code></li>
            <li><strong>action for regex match</strong>: <code class="language-plaintext highlighter-rouge">include line on pattern match</code></li>
          </ul>
        </li>
        <li>Add another Filter: Click on <code class="language-plaintext highlighter-rouge">+ Insert Filter Tabular Input Lines</code>:</li>
        <li><strong>Filter By</strong>: Select <code class="language-plaintext highlighter-rouge">select columns</code>
          <ul>
            <li><strong>enter column numbers to keep</strong>: <code class="language-plaintext highlighter-rouge">1,3,23,24</code></li>
          </ul>
        </li>
      </ul>

      <p>Section <strong>Table Options</strong>:</p>

      <ul>
        <li><strong>Specify Name for Table</strong>: <code class="language-plaintext highlighter-rouge">bering_psms</code></li>
        <li><strong>Specify Column Names (comma-separated list)</strong>: <code class="language-plaintext highlighter-rouge">id,sequence,confidence,validation</code></li>
        <li><strong>Only load the columns you have named into database</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><strong>Table Index</strong>: Click on <code class="language-plaintext highlighter-rouge">+ Insert Table Index</code>:
          <ul>
            <li><strong>This is a unique index</strong>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><strong>Index on Columns</strong>: <code class="language-plaintext highlighter-rouge">sequence,id</code></li>
          </ul>
        </li>
        <li>
          <p><strong>Save the sqlite database in your history</strong>: <code class="language-plaintext highlighter-rouge">Yes</code></p>
        </li>
        <li>
          <p><strong>SQL Query to generate tabular output</strong>:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT sequence as "peptide", count(id) as "PSMs"

FROM bering_psms

WHERE confidence &gt;= 95

GROUP BY sequence

ORDER BY sequence
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>
      <p>Click <strong>Run Tool</strong>.</p>
    </li>
  </ol>

</blockquote>

<p>With this we have combined all the data into a single database which we can now query to extract the desired information with <strong>SQLite to tabular</strong>:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-sqlite-to-tabular-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: SQLite to tabular</div>

  <ol>
    <li>
      <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/sqlite_to_tabular/sqlite_to_tabular/2.0.0" title="SQLite to tabular tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SQLite to tabular</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.0)</span> with the following parameters:</p>

      <ul>
        <li><strong>SQLite Database</strong>: The created SQLite database from the former step</li>
        <li>
          <p><strong>SQL Query</strong>:</p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SELECT go.description,

count(distinct bering_psms.sequence) as "bering_peptides", count(distinct bering_psms.id) as "bering_psms"

FROM go JOIN bering_prot_go ON go.go_id = bering_prot_go.go_reference JOIN bering_prot on bering_prot_go.id = bering_prot.id JOIN

bering_psms ON bering_prot.peptide = bering_psms.sequence

WHERE go.aspect = 'molecular_function'

GROUP BY go.description

ORDER BY  bering_peptides desc,bering_psms desc
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>Click <strong>Run Tool</strong>.</li>
    <li>Repeat these steps two times by replacing <code class="language-plaintext highlighter-rouge">molecular_function</code> in the fifth row of the SQL query by <code class="language-plaintext highlighter-rouge">biological_process</code> and <code class="language-plaintext highlighter-rouge">cellular_component</code>.</li>
  </ol>

</blockquote>

<p>With these three output files the functional analysis of this tutorial is finished. Each record contains the name of a GO term, the amount of peptides related to it and the amount of PSMs for these peptides.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-references"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: References</div>

  <ul>
    <li>
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/27824341">Dataset</a> and <a href="https://www.ncbi.nlm.nih.gov/pubmed/27396978">SixGill software</a></p>
    </li>
    <li>
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/26058579">Galaxy workflows for metaproteomics</a></p>
    </li>
    <li>
      <p><a href="https://z.umn.edu/gcc2017mporal">Metaproteomics community effort</a></p>
    </li>
    <li>
      <p><a href="https://www.ncbi.nlm.nih.gov/pubmed/28552653">Unipept</a></p>
    </li>
    <li>
      <p><a href="http://z.umn.edu/mpvideo2018">Metaproteomics video</a></p>
    </li>
  </ul>
</blockquote>


                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT), 
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Use dataset collections</p>
</li>
                        
                        <li><p>With SearchGUI and PeptideShaker you can gain access to multiple search engines</p>
</li>
                        
                        <li><p>Learning the basics of SQL queries can pay off</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/proteomics/faqs/">FAQ page for the Proteomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Antezana_2008">Antezana, E., M. Egana, B. D. Baets, M. Kuiper, and V. Mironov, 2008 <b>ONTO-PERL: An API for supporting the development and analysis of bio-ontologies</b>. Bioinformatics 24: 885–887. <a href="https://doi.org/10.1093/bioinformatics/btn042">10.1093/bioinformatics/btn042</a></li>
<li id="May_2016">May, D. H., E. Timmins-Schiffman, M. P. Mikan, H. R. Harvey, E. Borenstein <i>et al.</i>, 2016 <b>An Alignment-Free “Metapeptide” Strategy for Metaproteomic Characterization of Microbiome Samples Using Shotgun Metagenomic Sequencing</b>. Journal of Proteome Research 15: 2697–2705. <a href="https://doi.org/10.1021/acs.jproteome.6b00239">10.1021/acs.jproteome.6b00239</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=proteomics/metaproteomics"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=proteomics/metaproteomics">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Timothy J. Griffin, Pratik Jagtap, James Johnson, Clemens Blank, Subina Mehta,  <b>Metaproteomics tutorial (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html">https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{proteomics-metaproteomics,
author = "Timothy J. Griffin and Pratik Jagtap and James Johnson and Clemens Blank and Subina Mehta",
	title = "Metaproteomics tutorial (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/proteomics/tutorials/metaproteomics/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/proteomics/tutorials/metaproteomics/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: peptideshaker
  owner: galaxyp
  revisions: 7fdd9119cc4f
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: peptideshaker
  owner: galaxyp
  revisions: 7fdd9119cc4f
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: unipept
  owner: galaxyp
  revisions: 9aaa46d45472
  tool_panel_section_label: Get Data
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: query_tabular
  owner: iuc
  revisions: 33d61c89fb8d
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sqlite_to_tabular
  owner: iuc
  revisions: b722161a845a
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">1</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00221">gxy.io/GTN:T00221</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/proteomics/tutorials/metaproteomics/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/proteomics/tutorials/metaproteomics/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/b71986380d4874b98c2b7cd3c9a2fe9500f99672">b719863</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>