<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Clustering 3K PBMCs with Scanpy</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Clustering 3K PBMCs with Scanpy" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/galaxy-single-cell'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-scanpy-pbmc3k%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-scanpy-pbmc3k%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-scanpy-pbmc3k%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-scanpy-pbmc3k%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-scanpy-pbmc3k%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fscrna-scanpy-pbmc3k%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Clustering 3K PBMCs with Scanpy",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / scrna-scanpy-pbmc3k / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Clustering 3K PBMCs with Scanpy' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html",
  "timeRequired": "PT5H",
  "keywords": "single-cell, 10x",
  "description": "The questions this  addresses are:\n - What are the steps to prepare single-cell RNA-Seq data for clustering?\n - How to cluster cells in single-cell RNA-Seq data?\n - How cell type annotation can be assigned to cell clusters?\n\n\\nThe objectives are:\n - Describe an AnnData object to store single-cell data\n - Explain the preprocessing steps for single-cell data\n - Evaluate quality of single-cell data and apply steps to select and filter cells and genes based on QC\n - Execute data normalization and scaling\n - Identify highly variable genes\n - Construct and run a dimensionality reduction using Principal Component Analysis\n - Perform a graph-based clustering for cells\n - Identify marker genes for the clusters\n - Construct and run a cell type annotation for the clusters\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-preprocessing/slides.html",
      "name": "Dealing with Cross-Contamination in Fixed Barcode Protocols",
      "description": "Slides for 'Dealing with Cross-Contamination in Fixed Barcode Protocols' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html",
      "name": "Pre-processing of Single-Cell RNA Data",
      "description": "Hands-on for 'Pre-processing of Single-Cell RNA Data' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html",
      "name": "Pre-processing of 10X Single-Cell RNA Datasets",
      "description": "Hands-on for 'Pre-processing of 10X Single-Cell RNA Datasets' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Hans-Rudolf Hotz"
    },
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "B√©r√©nice Batut"
    },
    {
      "@type": "Person",
      "name": "Hans-Rudolf Hotz"
    },
    {
      "@type": "Person",
      "name": "Mehmet Tekman"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Clustering 3K PBMCs with Scanpy</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/bebatut/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/bebatut" alt="B√©r√©nice Batut">B√©r√©nice Batut</a>, <a href="/training-material/hall-of-fame/hrhotz/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/hrhotz" alt="Hans-Rudolf Hotz">Hans-Rudolf Hotz</a>, <a href="/training-material/hall-of-fame/mtekman/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/mtekman" alt="Mehmet Tekman">Mehmet Tekman</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What are the steps to prepare single-cell RNA-Seq data for clustering?</p>
</li>
            
            <li><p>How to cluster cells in single-cell RNA-Seq data?</p>
</li>
            
            <li><p>How cell type annotation can be assigned to cell clusters?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Describe an AnnData object to store single-cell data</p>
</li>
            
            <li><p>Explain the preprocessing steps for single-cell data</p>
</li>
            
            <li><p>Evaluate quality of single-cell data and apply steps to select and filter cells and genes based on QC</p>
</li>
            
            <li><p>Execute data normalization and scaling</p>
</li>
            
            <li><p>Identify highly variable genes</p>
</li>
            
            <li><p>Construct and run a dimensionality reduction using Principal Component Analysis</p>
</li>
            
            <li><p>Perform a graph-based clustering for cells</p>
</li>
            
            <li><p>Identify marker genes for the clusters</p>
</li>
            
            <li><p>Construct and run a cell type annotation for the clusters</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/transcriptomics">Transcriptomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Pre-processing of Single-Cell RNA Data:
                            
                                <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                            
                            <li> Pre-processing of 10X Single-Cell RNA Datasets:
                            
                            
                                
                                     <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 5 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/3581213">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/workflows/" title="Workflows" alt="Clustering 3K PBMCs with Scanpy workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://humancellatlas.usegalaxy.eu/" title="Human Cell Atlas">
            Human Cell Atlas
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/" title="Single Cell Omics workbench">
            Single Cell Omics workbench
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>This tutorial is significantly based on <a href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html#Clustering-3K-PBMCs">‚ÄúClustering 3K PBMCs‚Äù tutorial from Scanpy</a>, <a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html">‚ÄúSeurat - Guided Clustering Tutorial‚Äù</a> and <a href="https://osca.bioconductor.org/">‚ÄúOrchestrating <span class="notranslate">Single-Cell</span> Analysis with Bioconductor‚Äù</a> <a class="citation" href="#amezquita2019orchestrating">Amezquita <i>et al.</i> 2019</a></p>

</blockquote>

<p><span class="notranslate">Single-cell</span> <span class="notranslate">RNA</span>-seq analysis is a rapidly evolving field at the forefront of transcriptomic research, used in high-throughput developmental studies and rare transcript studies to examine cell heterogeneity within a populations of cells. The cellular resolution and genome wide scope make it possible to draw new conclusions that are not otherwise possible with bulk <span class="notranslate">RNA</span>-seq.</p>

<p>In this tutorial, we will investigate clustering of <span class="notranslate">single-cell</span> data from 10x Genomics, including preprocessing, clustering and the identification of cell types via known marker genes, using <a href="https://scanpy.readthedocs.io/en/stable/index.html">Scanpy</a> (<a class="citation" href="#wolf2018scanpy">Wolf <i>et al.</i> 2018</a>). It will be illustrated using a dataset of Peripheral Blood Mononuclear Cells (PBMC), containing 2,700 <span class="notranslate">single cell</span>s.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#data" id="markdown-toc-data">Data</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#anndata" id="markdown-toc-anndata">AnnData</a></li>
    </ol>
  </li>
  <li><a href="#preprocessing" id="markdown-toc-preprocessing">Preprocessing</a>    <ol>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
      <li><a href="#normalization-and-scaling" id="markdown-toc-normalization-and-scaling">Normalization and scaling</a></li>
      <li><a href="#selection-of-features" id="markdown-toc-selection-of-features">Selection of features</a></li>
      <li><a href="#scaling-the-data" id="markdown-toc-scaling-the-data">Scaling the data</a></li>
    </ol>
  </li>
  <li><a href="#dimensionality-reduction" id="markdown-toc-dimensionality-reduction">Dimensionality reduction</a>    <ol>
      <li><a href="#principal-component-analysis" id="markdown-toc-principal-component-analysis">Principal Component Analysis</a></li>
      <li><a href="#visualization-of-pca" id="markdown-toc-visualization-of-pca">Visualization of PCA</a></li>
      <li><a href="#determination-of-the-number-of-pcs-to-keep" id="markdown-toc-determination-of-the-number-of-pcs-to-keep">Determination of the number of PCs to keep</a></li>
    </ol>
  </li>
  <li><a href="#clustering-of-the-cells" id="markdown-toc-clustering-of-the-cells">Clustering of the cells</a>    <ol>
      <li><a href="#computation-of-a-neighborhood-graph" id="markdown-toc-computation-of-a-neighborhood-graph">Computation of a neighborhood graph</a></li>
      <li><a href="#visualization-of-the-neighborhood-graph" id="markdown-toc-visualization-of-the-neighborhood-graph">Visualization of the neighborhood graph</a></li>
      <li><a href="#clustering-of-the-neighborhood-graph" id="markdown-toc-clustering-of-the-neighborhood-graph">Clustering of the neighborhood graph</a></li>
    </ol>
  </li>
  <li><a href="#finding-marker-genes" id="markdown-toc-finding-marker-genes">Finding marker genes</a>    <ol>
      <li><a href="#using-the-t-test" id="markdown-toc-using-the-t-test">Using the <em>t</em>-test</a></li>
      <li><a href="#using-wilcoxon-rank-sum-test" id="markdown-toc-using-wilcoxon-rank-sum-test">Using Wilcoxon rank sum test</a></li>
      <li><a href="#visualization-of-expression-of-the-marker-genes" id="markdown-toc-visualization-of-expression-of-the-marker-genes">Visualization of <span class="notranslate">expression</span> of the marker genes</a></li>
      <li><a href="#comparison-of-the-marker-genes-between-clusters" id="markdown-toc-comparison-of-the-marker-genes-between-clusters">Comparison of the marker genes between clusters</a></li>
    </ol>
  </li>
  <li><a href="#cell-type-annotation" id="markdown-toc-cell-type-annotation">Cell type annotation</a></li>
</ol>

</blockquote>

<h1 id="data">Data</h1>

<p>For this tutorial, we analyze a dataset of Peripheral Blood Mononuclear Cells (PBMC) extracted from a healthy donor, freely available from 10X Genomics. The dataset contains 2,700 <span class="notranslate">single cell</span>s sequenced using Illumina NextSeq 500. The raw sequences have been processed by the <a href="https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger"><strong>cellranger</strong></a> pipeline from 10X to extract a unique molecular identifier (UMI) count matrix, in a similar way to that as explained in <a href="/training-material/topics/transcriptomics/tutorials/scrna-preprocessing-tenx/tutorial.html">‚ÄúPre-processing of 10X <span class="notranslate">Single-Cell</span> <span class="notranslate">RNA</span> Datasets‚Äù tutorial</a>.</p>

<p>In this matrix, the values represent the number for each feature (i.e. gene; row) that are detected in each cell (column). Such matrices can be quite large, where here there are 2,700 columns with 32,738 lines, with mostly zero values, i.e. an extremely sparse matrix. To optimize the storage of such a table and the information about the genes and cells, <strong>cellranger</strong> creates 3 files:</p>

<ul>
  <li><code class="language-plaintext highlighter-rouge">genes.tsv</code>: a tabular file with information about the 32,738 genes in 2 columns (Ensembl gene id and the gene symbol)</li>
  <li><code class="language-plaintext highlighter-rouge">barcodes.tsv</code>: a tabular file with the barcode for each of the 2,700 cells</li>
  <li>
    <p><code class="language-plaintext highlighter-rouge">matrix.mtx</code>: a condensed version of the count matrix</p>

    <p>The count matrix is therefore represented by its non-zero values. Each non-zero value is indicated by its line number (1st column), its column number (2nd column) and its value (3rd column). The first row gives indication about the number of lines, column and non-zero values. More information on the Matrix Market Exchange (mtx) format can be found <a href="https://math.nist.gov/MatrixMarket/formats.html">here</a></p>
  </li>
</ul>

<h2 id="data-upload">Data upload</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the <code class="language-plaintext highlighter-rouge">genes.tsv</code>, <code class="language-plaintext highlighter-rouge">barcodes.tsv</code> and <code class="language-plaintext highlighter-rouge">matrix.mtx</code> from <a href="https://zenodo.org/record/3581213">Zenodo</a> or from the shared data library</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/3581213/files/genes.tsv
https://zenodo.org/record/3581213/files/barcodes.tsv
https://zenodo.org/record/3581213/files/matrix.mtx
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>Rename the datasets</li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">matrix</code> file</li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>32738	2700	2286884
32709	1	4
32707	1	1
32706	1	10
32704	1	1
</code></pre></div>  </div>

  <ol>
    <li>How many non-zero values are in the matrix?</li>
    <li>How many counts are found for the 32,706th gene in the 1st cell?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>There are 2,286,884 (2.6%) non-zero values for the 88,392,600 possible counts of the 32,738 genes (rows) and 2,700 cells (columns).</li>
      <li>10 counts are found for the 32,706th row and 1st column.</li>
    </ol>

  </blockquote>

</blockquote>

<p>The representation of the matrix with 3 files is convenient to share the data but not to process them. <span class="notranslate">Single-cell</span> analysis packages have attempted to solve the problem of storage and analysis by inventing their own standard, which has led to the proliferation of many different ‚Äústandards‚Äù in the sc<span class="notranslate">RNA</span>-seq package ecosystem.</p>

<h2 id="anndata">AnnData</h2>

<p>The most common format, called <a href="https://anndata.readthedocs.io/en/stable/"><code class="language-plaintext highlighter-rouge">AnnData</code></a>, stores the matrix as well as gene and cell annotations in a concise, compressed and extremely readable manner:</p>

<figure id="figure-1"><img src="../../images/tenx_anndata.svg" alt="Anndata format" /><figcaption><span class="figcaption-prefix">Figure 1:</span> <code>AnnData</code> format stores a count matrix <code>X</code> together with annotations of observations (i.e. cells) <code>obs</code>, variables (i.e. genes) <code>var</code> and unstructured annotations <code>uns</code>.</figcaption></figure>

<p>This format is used by <a href="https://scanpy.readthedocs.io/en/stable/index.html">Scanpy</a> (<a class="citation" href="#wolf2018scanpy">Wolf <i>et al.</i> 2018</a>), the tool suite for analyzing <span class="notranslate">single-cell</span> gene <span class="notranslate">expression</span> data that we will use in this tutorial. So we need first to import the matrix and annotations of genes and cells into an <code class="language-plaintext highlighter-rouge">AnnData</code> object.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-transform-matrix-and-all-into-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Transform matrix and all into AnnData object</h3>

  <ol>
    <li><strong>Import AnnData and Loom</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚Äúhd5 format to be created‚Äù</em>: <code class="language-plaintext highlighter-rouge">Anndata file</code></li>
        <li><em>‚ÄúFormat for the annotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">Matrix Market (mtx), from Cell ranger or not</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúMatrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">matrix.mtx</code></li>
            <li><em>‚ÄúUse 10x Genomics formatted mtx‚Äù</em>: <code class="language-plaintext highlighter-rouge">Output from Cell Ranger v2 or earlier versions</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGenes‚Äù</em>: <code class="language-plaintext highlighter-rouge">genes.tsv</code></li>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúBarcodes‚Äù</em>: <code class="language-plaintext highlighter-rouge">barcodes.tsv</code></li>
                <li><em>‚ÄúVariables index‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene_symbols</code></li>
                <li><em>‚ÄúMake the variable index unique by appending ‚Äò-1‚Äô, ‚Äò-2‚Äô?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated file to <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></li>
    <li>Check that the format is <code class="language-plaintext highlighter-rouge">h5ad</code></li>
  </ol>
</blockquote>

<p>Because the <code class="language-plaintext highlighter-rouge">AnnData</code> format is an extension of the HDF5 format, i.e. a binary format, an <code class="language-plaintext highlighter-rouge">AnnData</code> object can not be inspected directly in <span class="notranslate">Galaxy</span> by clicking on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> (<strong>View data</strong>) icon. Instead we need to use a dedicated tool from the <strong>AnnData</strong> suite.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-inspect-an-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect an AnnData object</h3>

  <ol>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">General information about the object</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs √ó n_vars = 2700 √ó 32738
    var: 'gene_ids'
</code></pre></div>        </div>

        <ol>
          <li>How many observations are there? What do they represent?</li>
          <li>How many variables are there? What do they represent?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>There are 2,700 observations, representing the cells.</li>
            <li>There are 32,738 variables, representing the genes.</li>
          </ol>

        </blockquote>

      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment-faster-method-for-general-information"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Faster Method for General Information</h3>

        <ul>
          <li>The <strong>ScanPy</strong> toolset in <span class="notranslate">Galaxy</span> produces <em>AnnData</em> formats as
output, where general information is provided for each dataset:
            <ul>
              <li>Click on the name of the dataset in the history to expand it.</li>
              <li>
                <p>General Anndata information would be given in the expanded box:</p>

                <p>e.g.</p>

                <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs x n_vars]
-    2700 x 32738
</code></pre></div>                </div>
              </li>
            </ul>
          </li>
          <li>For more specific queries, <strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is required.</li>
        </ul>
      </blockquote>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">The full data matrix</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>index	MIR1302-10	FAM138A	OR4F5	RP11-34P13.7	RP11-34P13.8 ...
AAACATACAACCAC-1	0.0	0.0	0.0	0.0
AAACATTGAGCTAC-1	0.0	0.0	0.0	0.0
AAACATTGATCAGC-1	0.0	0.0	0.0	0.0
AAACCGTGCTTCCG-1	0.0	0.0	0.0	0.0
AAACCGTGTATGCG-1	0.0	0.0	0.0	0.0
</code></pre></div>        </div>

        <p>What is stored in the generated file?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The file is a table with 2,700 lines (observations or cells) and 32,738 columns (variables or genes): the count matrix for each of the 32,738 genes and 2,700 cells. The 1st row contains the gene symbol as annotation of the columns and the 1st column the barcodes of the cells as annotation of the rows.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Key-indexed observations annotation (obs)</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>index
AAACATACAACCAC-1
AAACATTGAGCTAC-1
AAACATTGATCAGC-1
AAACCGTGCTTCCG-1
AAACCGTGTATGCG-1
</code></pre></div>        </div>

        <p>What is stored in the generated file?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The file is a tabular with annotations of the observations. i.e. the cells. Here there are only the barcodes as annotation, so only one column, being also the index for the count matrix.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Key-indexed annotation of variables/features (var)</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-4"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>index	gene_ids
MIR1302-10	ENSG00000243485
FAM138A	ENSG00000237613
OR4F5	ENSG00000186092
RP11-34P13.7	ENSG00000238009
RP11-34P13.8	ENSG00000239945
</code></pre></div>        </div>

        <p>What is stored in the generated file?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The file is a tabular with annotations of the variables, i.e. the genes. Here the files has 2 columns: the gene symbols (also the index in the count matrix), <code class="language-plaintext highlighter-rouge">gene_ids</code> or the Ensemble gene ids.</p>

        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<h1 id="preprocessing">Preprocessing</h1>

<p>The first step when we get our counts is to prepare them before we perform the clustering. This includes (<a class="citation" href="#amezquita2019orchestrating">Amezquita <i>et al.</i> 2019</a>):</p>

<ol>
  <li>
    <p><strong>Selection and filtration of cells and genes based on quality metrics</strong></p>

    <p>Low-quality cells would interfere with <span class="notranslate">downstream</span> analyses. These cells may have been damaged during processing and may not have been fully captured by the sequencing protocol.</p>
  </li>
  <li>
    <p><strong>Data normalization and scaling</strong></p>

    <p>It will eliminate cell-specific biases (e.g., in capture efficiency), allowing us to perform explicit comparisons across cells afterwards. Some transformations, typically log, are also applied to adjust for the mean-variance relationship.</p>
  </li>
  <li>
    <p><strong>Selection of features</strong></p>

    <p>This selection picks a subset of interesting features for <span class="notranslate">downstream</span> analysis, by modelling the variance across cells for each gene and retaining genes that are highly variable. This step is done to reduce computational overhead and noise from uninteresting genes.</p>
  </li>
</ol>

<h2 id="quality-control">Quality control</h2>

<p>Genes that appear in less than a few cells can be considered noise and thus removed.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-genes-found-in-less-than-3-cells"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove genes found in less than 3 cells</h3>

  <ol>
    <li><strong>Filter</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">Input 3k PBMC</code></li>
        <li><em>‚ÄúMethod used for filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">Filter genes based on number of cells or counts, using 'pp.filter_genes'</code>
          <ul>
            <li><em>‚ÄúFilter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Minimum number of cells expressed</code>
              <ul>
                <li><em>‚ÄúMinimum number of cells expressed required for a gene to pass filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated file <code class="language-plaintext highlighter-rouge">3k PBMC</code></li>
    <li>
      <p>Inspect the dataset.</p>

      <blockquote class="question">
        <h3 id="question-questions-5"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs √ó n_vars]
-    2700 √ó 13714
[var]
-    gene_ids
-    n_cells
</code></pre></div>        </div>

        <p>How many genes have been removed because they are expressed in less than 3 expressed cells?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-5"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>There are now 13,714 cells. So 19,024 (32,738 - 13,714) genes have been removed.</p>

        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<p>Low-quality cells may be due to a variety of sources such as cell damage during dissociation or failure in library preparation (e.g., inefficient reverse transcription or PCR amplification, <a class="citation" href="#ilicic2016classification">Ilicic <i>et al.</i> 2016</a>). Usually these low-quality cells have low total counts, few expressed genes and high mitochondrial or spike-in proportions. Low-quality cells are problematic for <span class="notranslate">downstream</span> analyses and may contribute to misleading results.</p>

<blockquote class="details">
  <h3 id="details-impact-of-low-quality-cells-on-the-downstream-analyses"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Impact of low-quality cells on the <span class="notranslate">downstream</span> analyses</h3>

  <h4 id="formation-of-their-own-distinct-clusters">Formation of their own distinct cluster(s)</h4>

  <p>Increased mitochondrial proportions or enrichment for nuclear <span class="notranslate">RNA</span>s after cell damage are the most obvious causes for this phenomenon. Low-quality cells in different cell types can, in the worst case, cluster together (because of similarities in the damage-induced <span class="notranslate">expression</span> profiles) leading to artificial intermediate states or trajectories between subpopulations that should be otherwise distinct. This phenomenon makes the results difficult to interpret.</p>

  <h4 id="distortion-of-population-heterogeneity-during-variance-estimation-or-principal-components-analysis">Distortion of population heterogeneity during variance estimation or principal components analysis</h4>

  <p>With low-quality cells, the main differences captured by the first few principal components will be based on quality rather than biology, reducing then the effectiveness of dimensionality reduction. Similarly, differences between low- and high-quality cells will create genes with the largest variances. For example, in low-quality cells with very low counts, the scaling normalization increases the variance of genes that happen to have a non-zero count in those cells.</p>

  <h4 id="misidentification-of-upregulated-genes">Misidentification of upregulated genes</h4>

  <p>Due to aggressive scaling to normalize for small cell sizes, low-quality cells shows genes that appear to be strongly ‚Äúupregulated‚Äù. For example, contaminating transcripts may be present in all cells with a low but constant levels. With the increased scaling normalization in low-quality cells, the small counts for these transcripts may become large normalized <span class="notranslate">expression</span> values, i.e. an apparent upregulation compared to other cells.</p>

</blockquote>

<p>To mitigate these problems, we need to remove these low-quality cells at the start of the analysis. Several common QC metrics can be used to identify these cells based on their <span class="notranslate">expression</span> profile:</p>

<ul>
  <li>
    <p><strong>Cell size</strong>, i.e. the total sum of counts across all genes for each cells</p>

    <p>When cells are very degraded or absent from the library preparation, the number of <span class="notranslate">reads</span> sequenced from that library will be very low. Cells with small sizes are then considered to be of low quality: the <span class="notranslate">RNA</span> has been lost during the library preparation (cell lysis, inefficient c<span class="notranslate">DNA</span> capture and amplification, etc.)</p>
  </li>
  <li>
    <p><strong>Number of expressed genes</strong>, i.e. the number of genes with non-zero counts for each cells</p>

    <p>A low number of expressed genes may be a result of poor-quality cells (e.g. dying, degraded, damaged, etc.), followed by high PCR amplification of the remaining <span class="notranslate">RNA</span>. The diverse transcript population has not been successfully captured so the cell is considered of low quality. Cell doublets or multiplets may also exhibit an aberrantly high gene count.</p>
  </li>
  <li>
    <p><strong>Proportion of <span class="notranslate">reads</span> mapped to genes in the mitochondrial genome</strong></p>

    <p>High concentrations of mitochondrial genes is often a result of damaged cells (<a class="citation" href="#islam2014quantitative">Islam <i>et al.</i> 2014</a>, <a class="citation" href="#ilicic2016classification">Ilicic <i>et al.</i> 2016</a>) where the endogenous <span class="notranslate">RNA</span> escapes or degrades. As mitochondria has its own cell membranes, it is often the last <span class="notranslate">DNA</span>/<span class="notranslate">RNA</span> in damaged cells to degrade and hence occurs in high quantities during sequencing.</p>
  </li>
</ul>

<p>We will make the key assumption that these metrics are independent of the biological state of each cell. Technical factors rather than biological processes are presumed to drive poor values (e.g., low cell sizes, high mitochondrial proportions). The subsequent removal of impacted cells do not misrepresent the biology in <span class="notranslate">downstream</span> analyses.</p>

<p>To identify low-quality cells, the simplest approach is to apply thresholds on the QC metrics. This strategy, while simple, is required to determine appropriate thresholds that will change given the experimental protocol and the biological system.</p>

<h3 id="computation-of-qc-metrics">Computation of QC metrics</h3>

<p>To estimate the thresholds, we first need to look at the distribution of QC metrics for our dataset.</p>

<p>The first 2 QC metrics (cell size and number of expressed genes) can be easily estimated from the count table. For the third metric (proportion of <span class="notranslate">reads</span> mapped to genes in the mitochondrial genome), we need the information about which genes are mitochondrial or not.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-gene-annotation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract gene annotation</h3>

  <ol>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Key-indexed annotation of variables/features (var)</code></li>
      </ul>
    </li>
    <li>Rename the generated file <code class="language-plaintext highlighter-rouge">Gene annotation</code></li>
  </ol>
</blockquote>

<p>In the gene annotation, we currently have the gene symbol the Ensembl gene ids and the number of cells in which the genes are expressed, but have no information on whether a gene is mitochondrial. This can be extracted from the gene symbol as all mitochondrial genes have a name starting with <code class="language-plaintext highlighter-rouge">MT-</code> and added to the <code class="language-plaintext highlighter-rouge">var</code> of the <code class="language-plaintext highlighter-rouge">AnnData</code> object using <strong>Manipulate AnnData</strong> tool.</p>

<p>A new annotation for <code class="language-plaintext highlighter-rouge">var</code> should be a table with the same number of lines as <code class="language-plaintext highlighter-rouge">var</code> (i.e. one line per genes) and as many columns as new annotations, with their names on the 1st line. In our case, we would need to create a 1 column table with <code class="language-plaintext highlighter-rouge">mito</code> on the 1st line and then <code class="language-plaintext highlighter-rouge">True</code> for mitochondrial gene or <code class="language-plaintext highlighter-rouge">False</code> for non mitochondrial genes.</p>

<p>To create this table, we need to:</p>

<ol>
  <li>
    <p>Create a file with gene symbols from <code class="language-plaintext highlighter-rouge">Gene annotation</code> and a column with <code class="language-plaintext highlighter-rouge">1</code> for mitochondrial genes and <code class="language-plaintext highlighter-rouge">0</code> for other genes</p>

    <p>Using the <code class="language-plaintext highlighter-rouge">awk</code> program, we would like to print for every line the 1st column and a 2nd column with <code class="language-plaintext highlighter-rouge">1</code> if the 1st column starts with <code class="language-plaintext highlighter-rouge">MT-</code> or <code class="language-plaintext highlighter-rouge">0</code> if not, using the regular <span class="notranslate">expression</span> <code class="language-plaintext highlighter-rouge">$1 ~ /^MT-/</code></p>
  </li>
  <li>
    <p>Format it</p>

    <ol>
      <li>Replace the <code class="language-plaintext highlighter-rouge">0</code> by <code class="language-plaintext highlighter-rouge">False</code> and <code class="language-plaintext highlighter-rouge">1</code> by <code class="language-plaintext highlighter-rouge">True</code></li>
      <li>Remove the first line</li>
      <li>Keep only the 2nd column</li>
      <li>Add <code class="language-plaintext highlighter-rouge">mito</code> on the 1st line</li>
    </ol>
  </li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-mitochondrial-gene-annotation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add mitochondrial gene annotation</h3>

  <ol>
    <li><strong>Text reformatting</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to process‚Äù</em>: <code class="language-plaintext highlighter-rouge">Gene annotation</code></li>
        <li><em>‚ÄúAWK Program‚Äù</em>: <code class="language-plaintext highlighter-rouge">{print $1, $1 ~ /^MT-/}</code></li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-6"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How many genes are found as mitochondrial?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-6"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>To evaluate the number of non mitochondrial, we need to extract all lines for which the 2nd column is 1. This can be done using the <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool:</p>

          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFilter‚Äù</em>: output of <strong>Text reformatting</strong></li>
            <li><em>‚ÄúWith following condition‚Äù</em>: <code class="language-plaintext highlighter-rouge">c2!=0</code></li>
            <li><em>‚ÄúNumber of header lines to skip‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          </ul>

          <p>There are 13 genes found as mitochondrial (0.10% of the 13,714 genes).</p>
        </blockquote>

      </blockquote>
    </li>
    <li><strong>Replace Text in a specific column</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFile to process‚Äù</em>: output of <strong>Text reformatting</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚ÄúReplacement‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Replacement‚Äù</em>
              <ul>
                <li><em>‚Äúin column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
                <li><em>‚ÄúFind pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                <li><em>‚ÄúReplace with‚Äù</em>: <code class="language-plaintext highlighter-rouge">False</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Replacement‚Äù</em>
              <ul>
                <li><em>‚Äúin column‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
                <li><em>‚ÄúFind pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                <li><em>‚ÄúReplace with‚Äù</em>: <code class="language-plaintext highlighter-rouge">True</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <p><small><strong>Note</strong>: <em>Ensure that you have selected the same column for both replacements.</em></small></p>
    </li>
    <li><strong>Select last</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúText file‚Äù</em>: output of <strong>Replace Text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúOperation‚Äù</em>: <code class="language-plaintext highlighter-rouge">Keep everything from this line on</code></li>
        <li><em>‚ÄúNumber of lines‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
      </ul>
    </li>
    <li><strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c2</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFrom‚Äù</em>: output of <strong>Select last</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      </ul>
    </li>
    <li>
      <p>Create a new <strong>tabular</strong> file from the following</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>mito
</code></pre></div>      </div>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new file</h3>

        <ul>
          <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the file contents into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from ‚ÄúAuto-detect‚Äù to <code class="language-plaintext highlighter-rouge">tabular</code></p>
          </li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </blockquote>
    </li>
    <li><strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDatasets to concatenate‚Äù</em>: <code class="language-plaintext highlighter-rouge">Pasted entry</code> dataset</li>
        <li>In <em>‚ÄúDataset‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Dataset‚Äù</em>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect‚Äù</em>: output of <strong>Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file <code class="language-plaintext highlighter-rouge">Mitochondrial annotation</code> and ensure that the datatype is <code class="language-plaintext highlighter-rouge">tabular</code></p>
    </li>
    <li><strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC</code></li>
        <li><em>‚ÄúFunction to manipulate the object‚Äù</em>: <code class="language-plaintext highlighter-rouge">Add new annotation(s) for observations or variables</code>
          <ul>
            <li><em>‚ÄúWhat to annotate?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Variables (var)</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTable with new annotations‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mitochondrial annotation</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation</code></p>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Key-indexed annotation of variables/features (var)</code></li>
      </ul>
    </li>
    <li>Inspect the generated file and check if the mitochondrial annotation has been added</li>
  </ol>
</blockquote>

<p>We can now compute QC metrics on the <code class="language-plaintext highlighter-rouge">AnnData</code> object.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-compute-qc-metrics"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Compute QC metrics</h3>

  <ol>
    <li><strong>Inspect and manipulate</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation</code></li>
        <li><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Calculate quality control metrics, using 'pp.calculate_qc_metrics'</code>
          <ul>
            <li><em>‚ÄúName of kind of values in X‚Äù</em>: <code class="language-plaintext highlighter-rouge">counts</code></li>
            <li><em>‚ÄúThe kind of thing the variables are‚Äù</em>: <code class="language-plaintext highlighter-rouge">genes</code></li>
            <li><em>‚ÄúKeys for boolean columns of <code class="language-plaintext highlighter-rouge">.var</code> which identify variables you could want to control for‚Äù</em>: <code class="language-plaintext highlighter-rouge">mito</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation and qc metrics</code></p>
    </li>
    <li>
      <p>Inspect the dataset</p>

      <blockquote class="question">
        <h3 id="question-questions-7"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs √ó n_vars]
-    2700 √ó 13714
[obs]
-    n_genes_by_counts
-    log1p_n_genes_by_counts
-    total_counts
-    log1p_total_counts
-    pct_counts_in_top_50_genes
-    pct_counts_in_top_100_genes
-    pct_counts_in_top_200_genes
-    pct_counts_in_top_500_genes
-    total_counts_mito
-    log1p_total_counts_mito
-    pct_counts_mito
[var]
-    gene_ids
-    n_cells
-    mito
-    n_cells_by_counts
-    mean_counts
-    log1p_mean_counts
-    pct_dropout_by_counts
-    total_counts
-    log1p_total_counts
</code></pre></div>        </div>

        <p>Which QC metrics have been computed?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-7"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The tool computed several QC metrics at both cell and gene levels. These metrics are added to the annotation tables.</p>

          <p>At gene levels (stored in <code class="language-plaintext highlighter-rouge">var</code>):</p>
          <ul>
            <li><code class="language-plaintext highlighter-rouge">total_counts</code>: sum of counts for a gene</li>
            <li><code class="language-plaintext highlighter-rouge">mean_counts</code>: mean <span class="notranslate">expression</span> for a gene over all cells</li>
            <li><code class="language-plaintext highlighter-rouge">n_cells_by_counts</code>: number of cells with non-zero counts for a gene</li>
            <li><code class="language-plaintext highlighter-rouge">pct_dropout_by_counts</code>: percentage of cells this gene does not appear in</li>
          </ul>

          <p>At cell levels (stored in <code class="language-plaintext highlighter-rouge">obs</code>):</p>
          <ul>
            <li><code class="language-plaintext highlighter-rouge">total_counts</code>: total number of counts for a cell</li>
            <li><code class="language-plaintext highlighter-rouge">total_counts_mito</code>: total number of counts for the mitochondrial genes in a cell</li>
            <li><code class="language-plaintext highlighter-rouge">n_genes_by_counts</code>: number of genes with non-zero counts</li>
            <li><code class="language-plaintext highlighter-rouge">pct_counts_mito</code>: proportion of total counts for a cell which are from mitochondrial genes</li>
            <li><code class="language-plaintext highlighter-rouge">pct_counts_in_top_50_genes</code>: cumulative percentage of counts for 50 most expressed genes in a cell</li>
            <li><code class="language-plaintext highlighter-rouge">pct_counts_in_top_100_genes</code>: cumulative percentage of counts for 100 most expressed genes in a cell</li>
            <li><code class="language-plaintext highlighter-rouge">pct_counts_in_top_200_genes</code>: cumulative percentage of counts for 200 most expressed genes in a cell</li>
            <li><code class="language-plaintext highlighter-rouge">pct_counts_in_top_500_genes</code>: cumulative percentage of counts for 500 most expressed genes in a cell</li>
          </ul>
        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<p>We would like to visualize 3 of the more informative QC metrics:</p>

<ul>
  <li>the cell size, i.e. <code class="language-plaintext highlighter-rouge">total_counts</code></li>
  <li>the number of expressed genes, i.e. <code class="language-plaintext highlighter-rouge">n_genes_by_counts</code></li>
  <li>the proportion of <span class="notranslate">reads</span> mapped to mitochondrial genes, i.e. <code class="language-plaintext highlighter-rouge">pct_counts_mito</code></li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualize-qc-metrics"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualize QC metrics</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation and qc metrics</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generic: Violin plot, using 'pl.violin'</code>
          <ul>
            <li><em>‚ÄúKeys for accessing variables‚Äù</em>: <code class="language-plaintext highlighter-rouge">Subset of variables in 'adata.var_names' or fields of '.obs'</code>
              <ul>
                <li><em>‚ÄúKeys for accessing variables‚Äù</em>: <code class="language-plaintext highlighter-rouge">n_genes_by_counts, total_counts, pct_counts_mito</code></li>
              </ul>
            </li>
            <li>In <em>‚ÄúViolin plot attributes‚Äù</em>:
              <ul>
                <li><em>‚ÄúAdd a stripplot on top of the violin plot‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
                  <ul>
                    <li><em>‚ÄúAdd a jitter to the stripplot‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
                      <ul>
                        <li><em>‚ÄúSize of the jitter points‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.4</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li><em>‚ÄúDisplay keys in multiple panels‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-8"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p><img src="../../images/scrna-scanpy-pbmc3k/qc_violin_plot.png" alt="QC violin plot" />
<!-- To update... --></p>

        <p>How do the distributions of the 3 QC metrics look?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-8"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>For the cell size, i.e. <code class="language-plaintext highlighter-rouge">total_counts</code>, most of the values are between 1,000 <span class="notranslate">reads</span> and 4,000 <span class="notranslate">reads</span>, with some extremely high values skewing the distribution.</p>

          <p>The numbers of expressed genes, i.e. <code class="language-plaintext highlighter-rouge">n_genes_by_counts</code>, are mostly between 500 genes and 1,200 genes, with also some extremely high values skewing the distribution.</p>

          <p>The distribution of the proportions of <span class="notranslate">reads</span> mapped to mitochondrial genes, i.e. <code class="language-plaintext highlighter-rouge">pct_counts_mito</code>, is even more narrow with some cells having no counts from mitochondrial genes but also having some really extreme values (above 5%).</p>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation and qc metrics</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generic: Scatter plot along observations or variables axes, using 'pl.scatter'</code>
          <ul>
            <li><em>‚ÄúPlotting tool that computed coordinates‚Äù</em>: <code class="language-plaintext highlighter-rouge">Using coordinates</code>
              <ul>
                <li><em>‚Äúx coordinate‚Äù</em>: <code class="language-plaintext highlighter-rouge">total_counts</code></li>
                <li><em>‚Äúy coordinate‚Äù</em>: <code class="language-plaintext highlighter-rouge">n_genes_by_counts</code></li>
                <li><em>‚ÄúUse the layers attribute?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-9"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p><img src="../../images/scrna-scanpy-pbmc3k/qc_scatter_total_counts_n_genes_by_counts.png" alt="QC total_counts n_genes_by_counts" /></p>

        <p>Is there any relationship between the cell size and the number of expressed genes?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-9"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>On the plot, we can see a strong correlation between the total number of counts for a cell and the number of genes with positive counts.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation and qc metrics</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generic: Scatter plot along observations or variables axes, using 'pl.scatter'</code>
          <ul>
            <li><em>‚ÄúPlotting tool that computed coordinates‚Äù</em>: <code class="language-plaintext highlighter-rouge">Using coordinates</code>
              <ul>
                <li><em>‚Äúx coordinate‚Äù</em>: <code class="language-plaintext highlighter-rouge">n_genes_by_counts</code></li>
                <li><em>‚Äúy coordinate‚Äù</em>: <code class="language-plaintext highlighter-rouge">pct_counts_mito</code></li>
                <li><em>‚ÄúUse the layers attribute?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the generated file</p>

      <blockquote class="question">
        <h3 id="question-questions-10"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p><img src="../../images/scrna-scanpy-pbmc3k/qc_scatter_n_genes_by_counts_pct_counts_mito.png" alt="QC total_counts pct_counts_mito" /></p>

        <ol>
          <li>Is there any relationship between the number of expressed genes and the proportion of <span class="notranslate">reads</span> mapped to mitochondrial genes?</li>
          <li>What could be a good threshold to filter for cells with high concentrations of mitochondrial genes?</li>
          <li>What could be good thresholds to filter for cells based on the number of expressed genes?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-10"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>Cells with a high proportion of mitochondrial genes are not also cells that have many expressed genes. There is no visible correlation.</li>
            <li>The cells with a percentage of mitochondrial counts above 5% have also few genes. So 5% may be a good threshold.</li>
            <li>As discussed before, a low number of expressed genes may be a sign of poor-quality cells. Any cells with less than 200 genes is definitely out of the main distribution so this could be a good threshold to use. High gene count may also be a sign for cell multiplets. We will choose here a threshold of 2,500 genes.</li>
          </ol>

        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<h3 id="filtering-of-low-quality-cells">Filtering of low-quality cells</h3>

<p>As explained before, we would like now to filter for low-quality cells based on the 3 previous metrics (cell size, number of expressed genes, and proportion of <span class="notranslate">reads</span> mapped to mitochondrial genes). As the cell size is highly correlated with the number expressed genes, we can focus only on the number expressed genes and the proportion of <span class="notranslate">reads</span> mapped to mitochondrial genes.</p>

<p>Based on the previous plot, we would like to remove cells that have:</p>

<ul>
  <li>a number of expressed genes below 200 or above 2,500</li>
  <li>a percentage of <span class="notranslate">reads</span> mapped to mitochondrial genes above 5%</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-low-quality-cells"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove low-quality cells</h3>

  <ol>
    <li><strong>Filter</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with mito annotation and qc metrics</code></li>
        <li><em>‚ÄúMethod used for filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">Filter cell outliers based on counts and numbers of genes expressed, using 'pp.filter_cells'</code>
          <ul>
            <li><em>‚ÄúFilter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Minimum number of genes expressed</code>
              <ul>
                <li><em>‚ÄúMinimum number of genes expressed required for a cell to pass filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the dataset</p>

      <blockquote class="question">
        <h3 id="question-questions-11"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs √ó n_vars]
-    2700 √ó 13714
</code></pre></div>        </div>

        <p>How many cells have been removed because they have less than 200 expressed genes?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-11"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>There are still 2,700 cells. So no cells have been removed because they have less than 200 expressed genes.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Filter</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúMethod used for filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">Filter cell outliers based on counts and numbers of genes expressed, using 'pp.filter_cells'</code>
          <ul>
            <li><em>‚ÄúFilter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Maximum number of genes expressed</code>
              <ul>
                <li><em>‚ÄúMaximum number of genes expressed required for a cell to pass filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">2500</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the dataset
      <blockquote class="question">

        <h3 id="question-questions-12"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs √ó n_vars]
-    2695 √ó 13714
</code></pre></div>        </div>

        <p>How many cells have been removed because they have more than 2,500 expressed genes?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-12"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>There are now 2,695 cells. So 5 cells have been removed because they have more than 2,500 expressed genes.</p>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúFunction to manipulate the object‚Äù</em>: <code class="language-plaintext highlighter-rouge">Filter observations or variables</code>
          <ul>
            <li><em>‚ÄúWhat to filter?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Observations (obs)</code></li>
            <li><em>‚ÄúType of filtering?‚Äù</em>: <code class="language-plaintext highlighter-rouge">By key (column) values</code>
              <ul>
                <li><em>‚ÄúKey to filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">pct_counts_mito</code></li>
                <li><em>‚ÄúType of value to filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Number</code>
                  <ul>
                    <li><em>‚ÄúFilter‚Äù</em>: <code class="language-plaintext highlighter-rouge">less than</code></li>
                    <li><em>‚ÄúValue‚Äù</em>: <code class="language-plaintext highlighter-rouge">5.0</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated file <code class="language-plaintext highlighter-rouge">3k PBMC after QC filtering</code></p>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC after QC filtering</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">General information about the object</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-13"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs √ó n_vars = 2638 √ó 13714
</code></pre></div>        </div>

        <p>How many cells have been removed because they have more than 5% of <span class="notranslate">reads</span> mapped to mitochondrial genes?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-13"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>There are now 2,638 cells. So 57 cells have been removed because they have more than 5% of <span class="notranslate">reads</span> mapped to mitochondrial genes.</p>

        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="normalization-and-scaling">Normalization and scaling</h2>

<p>In sc<span class="notranslate">RNA</span>-seq, we can observe systematic differences in sequencing coverage between cells (<a class="citation" href="#stegle2015computational">Stegle <i>et al.</i> 2015</a>), because of technical differences in c<span class="notranslate">DNA</span> capture or PCR amplification efficiency across cells, attributable to the difficulty of achieving consistent library preparation with minimal starting material. After removing low-quality cells, normalization of the counts removes these differences to avoid that they interfere with comparisons of the <span class="notranslate">expression</span> profiles between cells. Any observed heterogeneity or differential <span class="notranslate">expression</span> within the cell population after normalization are then driven by biology and not technical biases.</p>

<p>Scaling normalization is the simplest and most commonly-used class of normalization strategies. All counts for each cell are divided by a cell-specific scaling factor, often called a ‚Äúsize factor‚Äù. The assumption behind this process is that any technical biases tend to affect genes in a similar manner. The relative bias for each cell is represented by the size factor for that cell. Dividing the cell counts by its size factor should then remove the bias.</p>

<p>The simplest strategy for scaling normalization is the cell size normalization, i.e. similar total sum of counts across all genes for each cell. The ‚Äúcell size factor‚Äù for each cell is computed directly for its cell size and transformed such that the mean size factor across all cells is equal to 1. The normalized <span class="notranslate">expression</span> values are then kept on the same scale as the original counts.</p>

<p>Here we would to normalize our count table such that each cell have 10,000 <span class="notranslate">reads</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-normalize-for-cell-size"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Normalize for cell size</h3>

  <ol>
    <li><strong>Normalize</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC after QC filtering</code></li>
        <li><em>‚ÄúMethod used for normalization‚Äù</em>: <code class="language-plaintext highlighter-rouge">Normalize counts per cell, using 'pp.normalize_total'</code>
          <ul>
            <li><em>‚ÄúTarget sum‚Äù</em>: <code class="language-plaintext highlighter-rouge">10000.0</code></li>
            <li><em>‚ÄúExclude (very) highly expressed genes for the computation of the normalization factor (size factor) for each cell‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúName of the field in ‚Äòadata.obs‚Äô where the normalization factor is stored‚Äù</em>: <code class="language-plaintext highlighter-rouge">norm</code></li>
            <li><em>‚ÄúList of layers to normalize‚Äù</em>: <code class="language-plaintext highlighter-rouge">all</code></li>
            <li><em>‚ÄúHow to normalize layers?‚Äù</em>: <code class="language-plaintext highlighter-rouge">After: for each layer in layers each cell has a total count equal to target_sum.</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The normalized counts should be log-transformed afterwards to adjust for the mean-variance relationship.</p>

<p>With log-transformation, the differences in the log-values represent log-fold changes in <span class="notranslate">expression</span>. Log-transformation focuses on promoting contributions from genes with strong relative differences (e.g. a gene that is expressed at an average count of 50 in cell type A and 10 in cell type B rather than a gene that is expressed at an average count of 1100 in A and 1000 in B).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-log-transform-the-counts"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Log-transform the counts</h3>

  <ol>
    <li><strong>Inspect and manipulate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of <strong>Normalize</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Logarithmize the data matrix, using 'pp.log1p'</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>We will freeze the current state of the AnnData object, i.e. the logarithmized raw gene <span class="notranslate">expression</span>, in the a <code class="language-plaintext highlighter-rouge">raw</code> attribute. This information will be used later in differential testing and visualizations of gene <span class="notranslate">expression</span>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-freeze-the-state-of-the-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Freeze the state of the AnnData object</h3>

  <ol>
    <li><strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of <strong>Inspect and manipulate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúFunction to manipulate the object‚Äù</em>: <code class="language-plaintext highlighter-rouge">Freeze the current state into the 'raw' attribute</code></li>
      </ul>
    </li>
    <li>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC after QC filtering and normalization</code></li>
  </ol>
</blockquote>

<h2 id="selection-of-features">Selection of features</h2>

<p>Data from sc<span class="notranslate">RNA</span>-Seq is often used in exploratory analyses to characterize heterogeneity across cells. With clustering and dimensionality reduction, cells are compared based on their gene <span class="notranslate">expression</span> profiles. The choice of genes to use may have a major impact on the behaviour of the clustering and the dimensionality reduction. We need then to remove genes with random noise while keeping genes containing useful information about the biology of the system. This reduces the data size but still highlights any interesting biological signal without the noise that obscures that structure.</p>

<p>Selecting the most variable genes based on their <span class="notranslate">expression</span> across the cells (i.e. genes highly expressed in some cells, and lowly expressed in others) is the simplest approach for feature selection. With this approach, we assume that increased variation in some genes compared to other genes are genuine biological differences and not technical noise or a baseline level of ‚Äúuninteresting‚Äù biological variation.</p>

<p>To quantify the per-gene variation, the simplest approach consists of computing the variance of the log-normalized <span class="notranslate">expression</span> values for each gene across all cells in the population. With this approach, the feature selection is based on the same log-values as the ones used in clustering, ensuring then that the quantitative definition of heterogeneity is consistent throughout the entire analysis.</p>

<p>Calculation of the per-gene variance is simple but feature selection requires modelling of the mean-variance relationship, as done in the Seurat procedure (<a class="citation" href="#stuart2019comprehensive">Stuart <i>et al.</i> 2019</a>) we will use.</p>

<p>Once the per-gene variation has been quantified, we need to select the subset of highly variable genes that we will use in <span class="notranslate">downstream</span> analyses. A large subset reduces the risk of discarding any interesting biological signal, but increases the noise from irrelevant genes on the signal. The optimal trade-off is difficult to determine but there are several common strategies routinely used. You can read the <a href="https://osca.bioconductor.org/feature-selection.html#hvg-selection">Chapter 8 from ‚ÄúOrchestrating <span class="notranslate">Single-Cell</span> Analysis with Bioconductor‚Äù</a> for a nice presentation of the different strategies. Here, we will define the set of highly variable genes as those which, after normalization, have a normalized dispersion amount higher than 0.5.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-identify-the-highly-variable-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Identify the highly variable genes</h3>

  <ol>
    <li><strong>Filter</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC after QC filtering and normalization</code></li>
        <li><em>‚ÄúMethod used for filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">Annotate (and filter) highly variable genes, using 'pp.highly_variable_genes'</code>
          <ul>
            <li><em>‚ÄúFlavor for computing normalized dispersion‚Äù</em>: <code class="language-plaintext highlighter-rouge">seurat</code>
              <ul>
                <li><em>‚ÄúMinimal mean cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.0125</code></li>
                <li><em>‚ÄúMaximal mean cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                <li><em>‚ÄúMinimal normalized dispersion cutoff‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
              </ul>
            </li>
            <li><em>‚ÄúInplace subset to highly-variable genes?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of the last <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Preprocessing: Plot dispersions versus means for genes, using 'pl.highly_variable_genes'</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/feature_selection_highly_variable_genes.png" alt="Highly variable genes" /></p>

<p>Both highly variable genes and other genes are still in the <code class="language-plaintext highlighter-rouge">AnnData</code> object. We can now actually keep only the highly variable genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-keep-the-highly-variable-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Keep the highly variable genes</h3>

  <ol>
    <li>
      <p>Inspect the output of the last <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></p>

      <blockquote class="question">
        <h3 id="question-questions-14"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs √ó n_vars]
-    2638 √ó 13714
[obs]
-    n_genes_by_counts
-    log1p_n_genes_by_counts
-    total_counts
-    log1p_total_counts
-    pct_counts_in_top_50_genes
-    pct_counts_in_top_100_genes
-    pct_counts_in_top_200_genes
-    pct_counts_in_top_500_genes
-    total_counts_mito
-    log1p_total_counts_mito
-    pct_counts_mito
-    n_genes
-    norm
[var]
-    gene_ids
-    n_genes
-    mito
-    n_cells_by_counts
-    mean_counts
-    log1p_mean_counts
-    pct_dropout_by_counts
-    total_counts
-    log1p_total_counts
-    highly_variable
-    means
-    dispersions
-    dispersions_norm
</code></pre></div>        </div>

        <ol>
          <li>How many genes are in the <code class="language-plaintext highlighter-rouge">AnnData</code> object?</li>
          <li>Where is the stored the information about the genes and if they are highly variable or not?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-14"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>There are now 13,714 genes, as before.</li>
            <li>Extra annotations have been added to <code class="language-plaintext highlighter-rouge">var</code>, whose a boolean annotation <code class="language-plaintext highlighter-rouge">highly_variable</code> for highly variable genes.</li>
          </ol>

        </blockquote>

      </blockquote>
    </li>
    <li><strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of the last <strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúFunction to manipulate the object‚Äù</em>: <code class="language-plaintext highlighter-rouge">Filter observations or variables</code>
          <ul>
            <li><em>‚ÄúWhat to filter?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Variables (var)</code>
              <ul>
                <li><em>‚ÄúType of filtering?‚Äù</em>: <code class="language-plaintext highlighter-rouge">By key (column) values</code>
                  <ul>
                    <li><em>‚ÄúKey to filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">highly_variable</code></li>
                    <li><em>‚ÄúType of value to filter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Boolean</code></li>
                    <li><em>‚ÄúValue to keep‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG</code></p>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">General information about the object</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-15"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AnnData object with n_obs √ó n_vars = 2638 √ó 1838
</code></pre></div>        </div>

        <p>How many genes have been removed?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-15"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>Only 1,838 (over the 13,714) genes are kept.</p>

        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="scaling-the-data">Scaling the data</h2>

<p>Prior to any <span class="notranslate">downstream</span> analysis like dimensional reduction, we need to apply a linear transformation or scaling to:</p>

<ol>
  <li>Regress out unwanted sources of variation in the total counts per cell and the percentage of mitochondrial genes expressed.</li>
  <li>Scale data to unit variance and zero mean, i.e. the variance across cells is 1 and the mean <span class="notranslate">expression</span> is 0, in order to give equal weight in <span class="notranslate">downstream</span> analyses and ensure that highly-expressed genes do not dominate.</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-scale-the-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Scale the data</h3>

  <ol>
    <li><strong>Remove confounders</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Regress out unwanted sources of variation, using 'pp.regress_out'</code>
          <ul>
            <li><em>‚ÄúKeys for observation annotation on which to regress on‚Äù</em>: <code class="language-plaintext highlighter-rouge">total_counts, pct_counts_mito</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Inspect and manipulate</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: output of <strong>Remove confounders</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Scale data to unit variance and zero mean, using 'pp.scale'</code>
          <ul>
            <li><em>‚ÄúZero center?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li>
              <p><em>‚ÄúMaximum value‚Äù</em>: <code class="language-plaintext highlighter-rouge">10.0</code></p>

              <p>It clips values exceeding a standard deviation of 10.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling</code></li>
  </ol>
</blockquote>

<h1 id="dimensionality-reduction">Dimensionality reduction</h1>

<p>We aim in sc<span class="notranslate">RNA</span>-seq to compare cells based on their <span class="notranslate">expression</span>s across genes, e.g. to identify similar transcriptomic profiles. Each gene represents then a dimension of the data.</p>

<p>With a dataset of 2 genes, we could make a 2-dimensional plot where each point is a cell and each axis is the <span class="notranslate">expression</span> of one gene. For datasets with thousands of genes, the concept is the same: each cell‚Äôs <span class="notranslate">expression</span> profile defines its location in the high-dimensional <span class="notranslate">expression</span> space.</p>

<p><span class="notranslate">Expression</span>s of different genes are correlated if they are affected by the same biological process. The separate information for these individual genes do not need to be stored, but can instead be compressed into a single dimension, e.g. an ‚Äúeigengene‚Äù. Dimensionality reduction aims then to reduce the number of separate dimensions in the data and then:</p>

<ul>
  <li>reduces the computational work in <span class="notranslate">downstream</span> analyses to only a few dimensions</li>
  <li>reduces noise by averaging across multiple genes to obtain a more precise representation of the patterns in the data</li>
  <li>enables effective plotting of the data.</li>
</ul>

<h2 id="principal-component-analysis">Principal Component Analysis</h2>

<p>Principal Component Analysis (PCA) is a dimensionality reduction technique consisting in the identification of axes in high-dimensional space that capture the largest amount of variation. This simple, highly effective strategy is widely used in data science.</p>

<p>In PCA, the first axis (or Principal Component (PC)) is chosen such that it captures the greatest variance across cells. The next PC should be orthogonal to the first and capture the greatest remaining amount of variation, and so on.</p>

<p>By applying PCA to sc<span class="notranslate">RNA</span>-Seq, we assume that multiple genes are affected by the same biological processes in a coordinated way and random technical or biological noise affects each gene independently. As more variation can be captured by considering the correlated behaviour of many genes, the top PCs are likely to represent the biological signal and the noise are concentrated into the later PCs. The dominant factors of heterogeneity are then captured by the top PCs. Restricting <span class="notranslate">downstream</span> analyses to the top PCs will then reduce the dimensionality of the data whilst focusing on the biological signal and removing the noise.</p>

<p>Here we perform the PCA on the log-normalized <span class="notranslate">expression</span> values and compute the first 50 PCs.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-perform-the-pca"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Perform the PCA</h3>

  <ol>
    <li><strong>Cluster, infer trajectories and embed</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Computes PCA (principal component analysis) coordinates, loadings and variance decomposition, using 'tl.pca'</code>
          <ul>
            <li><em>‚ÄúNumber of principal components to compute‚Äù</em>: <code class="language-plaintext highlighter-rouge">50</code></li>
            <li><em>‚ÄúType of PCA?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Full PCA</code>
              <ul>
                <li><em>‚ÄúCompute standard PCA from covariance matrix?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                <li><em>‚ÄúSVD solver to use‚Äù</em>: <code class="language-plaintext highlighter-rouge">ARPACK wrapper in SciPy</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
  </ol>

</blockquote>

<blockquote class="details">
  <h3 id="details-storage-of-pca-information-in-anndata-objects"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Storage of PCA information in <code class="language-plaintext highlighter-rouge">AnnData</code> objects</h3>

  <p>PCA information are stored in the <code class="language-plaintext highlighter-rouge">AnnData</code> object in a quite complex way.</p>

  <blockquote class="notranslate hands_on">
    <h3 id="hands_on-hands-on-inspect-the-pca-inside-an-anndata-object"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Inspect the PCA inside an <code class="language-plaintext highlighter-rouge">AnnData</code> object</h3>

    <ol>
      <li>
        <p>Inspect the <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code> dataset</p>

        <blockquote class="question">
          <h3 id="question-questions-16"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>[n_obs √ó n_vars]
-    2638 √ó 1838
[obs]
-    n_genes_by_counts
-    log1p_n_genes_by_counts
-    total_counts
-    log1p_total_counts
-    pct_counts_in_top_50_genes
-    pct_counts_in_top_100_genes
-    pct_counts_in_top_200_genes
-    pct_counts_in_top_500_genes
-    total_counts_mito
-    log1p_total_counts_mito
-    pct_counts_mito
-    n_genes
-    norm
[var]
-    gene_ids
-    mito
-    n_cells_by_counts
-    mean_counts
-    log1p_mean_counts
-    pct_dropout_by_counts
-    total_counts
-    log1p_total_counts
-    highly_variable
-    means
-    dispersions
-    dispersions_norm
[uns]
-    pca
[obsm]
-    X_pca
[varm]
-    PCs
</code></pre></div>          </div>

          <p>How is the PCA stored in the <code class="language-plaintext highlighter-rouge">AnnData</code> object?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-16"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <p>3 new objects (<code class="language-plaintext highlighter-rouge">uns</code>, <code class="language-plaintext highlighter-rouge">obsm</code>, <code class="language-plaintext highlighter-rouge">varm</code>) have been added to the <code class="language-plaintext highlighter-rouge">AnnData</code> object with information that seem related to PCA.</p>

            <p><code class="language-plaintext highlighter-rouge">uns</code> is an unstructured annotation, <code class="language-plaintext highlighter-rouge">obsm</code> multi-dimensional annotation of the observations (i.e. genes) and <code class="language-plaintext highlighter-rouge">varm</code> multi-dimensional annotation of the variables (i.e. cells).</p>
          </blockquote>

        </blockquote>
      </li>
      <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
          <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Unstructured annotation (uns)</code>
            <ul>
              <li><em>‚ÄúWhat to inspect in uns?‚Äù</em>: <code class="language-plaintext highlighter-rouge">PCA</code></li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-questions-17"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <p>What information is stored in <code class="language-plaintext highlighter-rouge">uns</code> regarding the PCA?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-17"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <p><code class="language-plaintext highlighter-rouge">uns</code> is storing:</p>

            <ul>
              <li>the ratio of explained variance by the PCs, sorted by PC, as a one-column table</li>
              <li>the explained variance for each PCs, equivalent to the eigenvalues of the covariance matrix, as a one-column table</li>
            </ul>
          </blockquote>

        </blockquote>
      </li>
      <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
          <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Multi-dimensional observations annotation (obsm)</code>
            <ul>
              <li><em>‚ÄúWhich annotation to inspect for the observations?‚Äù</em>: <code class="language-plaintext highlighter-rouge">PCA coordinates (X_pca)</code></li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-questions-18"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <p>What are the information stored in <code class="language-plaintext highlighter-rouge">obsm</code> regarding the PCA?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-18"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <p><code class="language-plaintext highlighter-rouge">obsm</code> is storing the PCA coordinates for the cells as a table with the rows being the cells, the columns the PCs and the values the PCs coordinate for cell and PC.</p>
          </blockquote>

        </blockquote>
      </li>
      <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
          <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Multi-dimensional variables annotation (varm)</code>
            <ul>
              <li><em>‚ÄúWhich annotation to inspect for the variables?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Principal components containing the loadings</code></li>
            </ul>
          </li>
        </ul>

        <blockquote class="question">
          <h3 id="question-questions-19"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

          <p>What information is stored in <code class="language-plaintext highlighter-rouge">varm</code> regarding the PCA?</p>

          <blockquote class="solution">
            <h3 id="solution-solution-19"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

            <p><code class="language-plaintext highlighter-rouge">varm</code> is storing the PCA coordinates for the genes as a table with the rows being the genes, the columns the PCs and the values the PCs coordinate for gene and PC.</p>
          </blockquote>

        </blockquote>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h2 id="visualization-of-pca">Visualization of PCA</h2>

<p>One application of dimensionality reduction and PCA is to compress the data for plotting into 2 (sometimes 3) dimensions with the most salient features of the data.</p>

<p>Scanpy provides several useful ways of visualizing both cells and genes that define the PCA. The simplest approach for visualization is to plot the top 3 PCs on 2D plots (PC1 vs PC2 and PC2 vs PC3)</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-the-top-2-pcs-the-pca"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot the top 2 PCs the PCA</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">PCA: Plot PCA results, using 'pl.pca_overview'</code>
          <ul>
            <li>In <em>‚ÄúPlot attributes‚Äù</em>
              <ul>
                <li>In <em>‚ÄúComponent‚Äù</em>
                  <ul>
                    <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Component‚Äù</em></li>
                    <li>In <em>‚Äú1: Component‚Äù</em>
                      <ul>
                        <li><em>‚ÄúX-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                        <li><em>‚ÄúY-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                      </ul>
                    </li>
                    <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Component‚Äù</em></li>
                    <li>In <em>‚Äú1: Component‚Äù</em>
                      <ul>
                        <li><em>‚ÄúX-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                        <li><em>‚ÄúY-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li><em>‚ÄúNumber of panels per row‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/pca_overview.png" alt="PCA overview" /></p>

<p>On these plots we see the different cells projected onto the first 3 PCs. We can already see subpopulations of cells, but only 3 PCs are represented there and plot like these are not so informative. It may be more interesting to project also the values for the genes, since perhaps these are the genes most involved in the 3 PCs.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualize-the-top-genes-associated-with-pcs"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Visualize the top genes associated with PCs</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">PCA: Rank genes according to contributions to PCs, using 'pl.pca_loadings'</code>
          <ul>
            <li><em>‚ÄúList of comma-separated components‚Äù</em>: <code class="language-plaintext highlighter-rouge">1,2,3</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="question">
        <h3 id="question-questions-20"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p><img src="../../images/scrna-scanpy-pbmc3k/pca_loadings.png" alt="PCA loadings" /></p>

        <p>What are the top genes for each of the 3 first PCs? What do they represent?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-20"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>CST3 is the gene the most associated with the 1st PC, NKG7 the one for the 2nd PC, and PF4 and PPBP for the 3rd PC (for consistency with the published <a href="https://scanpy-tutorials.readthedocs.io/en/latest/pbmc3k.html">scanpy</a> and <a href="https://satijalab.org/seurat/v3.1/pbmc3k_tutorial.html">Seurat</a> documentation, we will use PPBP).</p>

        </blockquote>
      </blockquote>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">PCA: Plot PCA results, using 'pl.pca_overview'</code>
          <ul>
            <li><em>‚ÄúKeys for annotations of observations/cells or variables/genes‚Äù</em>: <code class="language-plaintext highlighter-rouge">CST3, NKG7, PPBP</code></li>
            <li>In <em>‚ÄúPlot attributes‚Äù</em>
              <ul>
                <li>In <em>‚ÄúComponent‚Äù</em>
                  <ul>
                    <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Component‚Äù</em></li>
                    <li>In <em>‚Äú1: Component‚Äù</em>
                      <ul>
                        <li><em>‚ÄúX-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                        <li><em>‚ÄúY-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                      </ul>
                    </li>
                    <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Component‚Äù</em></li>
                    <li>In <em>‚Äú1: Component‚Äù</em>
                      <ul>
                        <li><em>‚ÄúX-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                        <li><em>‚ÄúY-axis‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li><em>‚ÄúNumber of panels per row‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/pca_scatter.png" alt="PCA scatter for CST3, NKG7 and PPBP" /></p>

<blockquote class="question">
  <h3 id="question-questions-21"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Where are the differences in <span class="notranslate">expression</span> for CST3, NKG7, and PPBP?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-21"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ul>
      <li>For CST3, the differences are mostly projected on PC1 (expected as CST3 is the top gene for PC1), and not visible on the PC3 vs PC2 plot.</li>
      <li>For NKG7, the differences in <span class="notranslate">expression</span> are seen on PC2.</li>
      <li>For PPBP, the differences in <span class="notranslate">expression</span> are seen on PC3.</li>
    </ul>

  </blockquote>
</blockquote>

<h2 id="determination-of-the-number-of-pcs-to-keep">Determination of the number of PCs to keep</h2>

<p>We performed the PCA analyses using 50 PCS. They represent a robust compression of the dataset, but we may not need to keep all of them. How many components should we choose to include?</p>

<p>The choice of the number of PCs is a similar question to the choice of the number of highly variable genes to keep. More PCs means more noise but also more biological signal.</p>

<p>A simple heuristic for choosing the number of PCs generates an ‚ÄúElbow plot‚Äù: a ranking of the PCs based on the percentage of variance they explain.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-generate-an-elbow-plot"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Generate an Elbow plot</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">PCA: Scatter plot in PCA coordinates, using 'pl.pca_variance_ratio'</code>
          <ul>
            <li><em>‚ÄúUse the log of the values?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/pca_variance_ratio.png" alt="PCA variance ratio" /></p>

<p>To determine the elbow point, we assume that each of the PCs should explain much more variance than the remaining PCS. So after the last PCs we choose, the percentage of variance explained should not drop much.</p>

<blockquote class="question">
  <h3 id="question-questions-22"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How many PCs should we keep given the previous plot?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-22"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>We choose here to keep 10 PCs.</p>

  </blockquote>
</blockquote>

<p>We encourage users to repeat <span class="notranslate">downstream</span> analyses with different number of PCs.</p>

<h1 id="clustering-of-the-cells">Clustering of the cells</h1>

<p>In the PC projection, we saw some subpopulations of cells emerging. We would like now to empirically define these subpopulation of cells with similar <span class="notranslate">expression</span> profiles using unsupervised clustering.</p>

<p>Clustering summarizes the data and allows us to describe the population heterogeneity in terms of discrete and easily understandable labels. The subpopulations can be afterwards treated as proxies for biological objects like cell types or states.</p>

<p>Graph-based clustering has been popularized for clustering large sc<span class="notranslate">RNA</span>-Seq datasets by its use in Seurat (<a class="citation" href="#butler2018integrating">Butler <i>et al.</i> 2018</a>, <a class="citation" href="#stuart2019comprehensive">Stuart <i>et al.</i> 2019</a>). Such approaches like the K-nearest neighbor (KNN) graph works in 2 steps:</p>

<ol>
  <li>
    <p><strong>Computation of a neighborhood graph</strong></p>

    <p>A graph is first built with each node being a cell connected to its nearest neighbours having similar <span class="notranslate">expression</span> patterns. Edges are weighted based on the similarity between the cells: higher weight is given to cells that are more closely related, i.e. have similar <span class="notranslate">expression</span> profiles.</p>
  </li>
  <li>
    <p><strong>Clustering of the neighborhood graph</strong></p>

    <p>The graph is then partitioned into highly interconnected ‚Äúquasi-cliques‚Äù or ‚Äúcommunities‚Äù, i.e. cells that are more connected to cells in the same community than they are to cells of different communities. Each community represents a cluster that we can use for <span class="notranslate">downstream</span> interpretation.</p>
  </li>
</ol>

<h2 id="computation-of-a-neighborhood-graph">Computation of a neighborhood graph</h2>

<p>When running graph-based clustering, we need to think about:</p>

<ul>
  <li>How many neighbors are considered when constructing the graph.</li>
  <li>What scheme is used to weight the edges.</li>
</ul>

<p>Here, to reproduce original results, we choose 10 neighbors for a KNN graph, the Euclidian distance metrics and the UMAP method (<a class="citation" href="#mcinnes2018umap">McInnes <i>et al.</i> 2018</a>) to compute the connectivities. The values will change depending on the data and can not easily predefined: testing different values is the only solution.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-compute-the-neighborhood-graph"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Compute the neighborhood graph</h3>

  <ol>
    <li><strong>Inspect and manipulate</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling and PCA</code></li>
        <li><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Compute a neighborhood graph of observations, using 'pp.neighbors'</code>
          <ul>
            <li><em>‚ÄúThe size of local neighborhood (in terms of number of neighboring data points) used for manifold approximation‚Äù</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>‚ÄúNumber of PCs to use‚Äù</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>‚ÄúUse a hard threshold to restrict the number of neighbors to n_neighbors?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúMethod for computing connectivities‚Äù</em>: <code class="language-plaintext highlighter-rouge">umap (McInnes et al, 2018)</code></li>
            <li><em>‚ÄúDistance metric‚Äù</em>: <code class="language-plaintext highlighter-rouge">euclidean</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA and KNN graph</code></p>
    </li>
    <li>
      <p>Inspect the dataset</p>

      <blockquote class="question">
        <h3 id="question-questions-23"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How is the neighborhood graph stored in the <code class="language-plaintext highlighter-rouge">AnnData</code> object?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-23"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>An extra object <code class="language-plaintext highlighter-rouge">neighbors</code> has been added to <code class="language-plaintext highlighter-rouge">uns</code> with 2 mtx objects (similar format to the original count table):</p>

          <ul>
            <li>Distance between each cells</li>
            <li>Weighted adjacency matrix between cells</li>
          </ul>

          <p>This information can be accessed using <strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA and KNN graph</code></li>
            <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Unstructured annotation (uns)</code>
              <ul>
                <li><em>‚ÄúWhat to inspect in uns?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Neighbors</code></li>
              </ul>
            </li>
          </ul>
        </blockquote>

      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="visualization-of-the-neighborhood-graph">Visualization of the neighborhood graph</h2>

<p>To visualize and explore the neighborhood graph, we can apply an extra step of non-linear dimensional reduction techniques to learn the underlying manifold of the data in order to place similar cells together in low-dimensional space. Cells that will be in the same clusters, i.e. cells with similar local neighborhoods in high-dimensional space, should co-localize on these low-dimensional plots.</p>

<p>Two techniques are commonly used for the non-linear dimensional reduction: <em>t</em>-SNE (t-distributed stochastic neighbor embedding) and UMAP. Scanpy authors recommend to use here UMAP as it better preserves trajectories (check <a class="citation" href="#becht2019dimensionality">Becht <i>et al.</i> 2019</a> for a review) and easily accommodates new data.</p>

<p>Here, we will reduce the neighborhood to 2 UMAP components and then we will check to see how the cells are projected on them given the top genes</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-embed-and-plot-the-neighborhood-graph"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Embed and plot the neighborhood graph</h3>

  <ol>
    <li><strong>Cluster, infer trajectories and embed</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA and KNN graph</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Embed the neighborhood graph using UMAP, using 'tl.umap'</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP</code></p>

      <blockquote class="question">
        <h3 id="question-questions-24"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How is the UMAP reduction stored in the <code class="language-plaintext highlighter-rouge">AnnData</code> object?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-24"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>An extra object <code class="language-plaintext highlighter-rouge">X_umap</code> has been added to <code class="language-plaintext highlighter-rouge">obsm</code> with the 2 UMAP coordinates for each cell, as a table of 2 columns and 2,638 lines.</p>

          <p>This information can be accessed using:</p>
          <ol>
            <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP</code></li>
                <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generalinformation about the object</code></li>
              </ul>
            </li>
            <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA and KNN graph, UMAP</code></li>
                <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Multi-dimensional observations annotation (obsm)</code></li>
                <li><em>‚ÄúWhat to inspect in for the observations?‚Äù</em>: <code class="language-plaintext highlighter-rouge">UMAP coordinates (X_umap)</code></li>
              </ul>
            </li>
          </ol>
        </blockquote>

      </blockquote>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Embeddings: Scatter plot in UMAP basis, using 'pl.umap'</code>
          <ul>
            <li><em>‚ÄúKeys for annotations of observations/cells or variables/genes‚Äù</em>: <code class="language-plaintext highlighter-rouge">CST3, NKG7, PPBP</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/umap_before_clustering.png" alt="UMAP coordinates before clustering" /></p>

<blockquote class="question">
  <h3 id="question-questions-25"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Are clusters identifiable on these graphs? Might they be linked to PCs?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-25"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>There seem to be at least 3 big clusters (the 3 blobs) before looking at the <span class="notranslate">expression</span> of some representative genes.</p>

    <p>With the plot colored with NKG7 <span class="notranslate">expression</span>, we clearly see that the blob on the bottom right could be divided in 3 sub-clusters (light green, blue, purple). Given the <span class="notranslate">expression</span> of PPBP, the ‚Äúarm‚Äù on the blob on the left should be a different cluster.</p>

  </blockquote>
</blockquote>

<h2 id="clustering-of-the-neighborhood-graph">Clustering of the neighborhood graph</h2>

<p>Given the first visualization, we can now cluster the cells within a neighborhood graph.</p>

<p>Which community detection algorithm should we use to define the clusters? Several modularity optimization techniques such as the SLM (<a class="citation" href="#blondel2008fast">Blondel <i>et al.</i> 2008</a>), Louvain algorithm (<a class="citation" href="#levine2015data">Levine <i>et al.</i> 2015</a>) or the Leiden algorithm (<a class="citation" href="#traag2019louvain">Traag <i>et al.</i> 2019</a>) are available to iteratively group cells together while optimizing the standard modularity function.</p>

<p>Currently, the Louvain graph-clustering method (community detection based on optimizing modularity) is the one recommended. We need to define a value for the resolution parameter, i.e. the ‚Äògranularity‚Äô of the <span class="notranslate">downstream</span> clustering. High values lead to a greater number of clusters. For <span class="notranslate">single-cell</span> datasets of around 3K cells, we recommend to use a value between 0.4 and 1.2. For larger datasets, the optimal resolution will be higher.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-cluster-the-neighborhood-graph"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Cluster the neighborhood graph</h3>

  <ol>
    <li><strong>Cluster, infer trajectories and embed</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cluster cells into subgroups, using 'tl.louvain'</code>
          <ul>
            <li><em>‚ÄúFlavor for the clustering‚Äù</em>: <code class="language-plaintext highlighter-rouge">vtraag (much more powerful)</code>
              <ul>
                <li><em>‚ÄúResolution‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering</code></p>

      <blockquote class="question">
        <h3 id="question-questions-26"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How is the clustering information stored in the <code class="language-plaintext highlighter-rouge">AnnData</code> object?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-26"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>An extra column <code class="language-plaintext highlighter-rouge">louvain</code> has been added to the <code class="language-plaintext highlighter-rouge">obs</code> object with the cluster id for each cell.</p>

          <p>This information can be accessed using:</p>
          <ol>
            <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering</code></li>
                <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generalinformation about the object</code></li>
              </ul>
            </li>
            <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering</code></li>
                <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Key-indexed observations annotation (obs)</code></li>
              </ul>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>The cells in the same clusters should be co-localized in the UMAP coordinate plots.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-the-neighborhood-graph-and-the-clusters"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot the neighborhood graph and the clusters</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Embeddings: Scatter plot in UMAP basis, using 'pl.umap'</code>
          <ul>
            <li><em>‚ÄúKeys for annotations of observations/cells or variables/genes‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain, CST3, NKG7, PPBP</code></li>
            <li>In <em>‚ÄúPlot attributes‚Äù</em>
              <ul>
                <li><em>‚ÄúNumber of panels per row‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/umap_after_clustering.png" alt="UMAP coordinates after clustering" /></p>

<blockquote class="question">
  <h3 id="question-questions-27"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many clusters have been identified? Do they fit with the ones quickly identified with UMAP plots?</li>
    <li>In which clusters do you expected to have CST3, NKG7 and PPBP as representative?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-27"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>8 clusters are identified, more or less corresponding to the ones we could see on the UMAP plots.</li>
      <li>We expect that:
        <ul>
          <li>CST3 should be representative of clusters 1, 3, 4, 6</li>
          <li>NKG7 for clusters 0, 3 and 5</li>
          <li>PPBP for cluster 7</li>
        </ul>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="finding-marker-genes">Finding marker genes</h1>

<p>To give sense to the clusters, we need to identify the genes that drive separation between clusters. These marker genes can then be used to assign biological sense (e.g. cell type) to each cluster based on their functional annotation, but also to identify subtle differences between clusters (e.g., changes in activation or differentiation state) based on the behaviour of genes in the affected pathways.</p>

<p>Marker genes are usually detected by their differential <span class="notranslate">expression</span> between clusters, as the more strongly DE genes are more likely to have caused separate clustering of cells. To quantify the differences in <span class="notranslate">expression</span> profiles, several different statistical tests can be used.</p>

<p>The identification of the marker genes for each cluster is made not only on the highly variable genes, but on the whole set (currently stored in the <code class="language-plaintext highlighter-rouge">raw</code> attribute).</p>

<h2 id="using-the-t-test">Using the <em>t</em>-test</h2>

<p>The simplest and fastest method is the Welch <em>t</em>-test. It has good statistical properties for large numbers of cells (<a class="citation" href="#soneson2018bias">Soneson and Robinson 2018</a>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-rank-the-highly-differential-genes-using-t-test"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Rank the highly differential genes using t-test</h3>

  <ol>
    <li><strong>Inspect and manipulate</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering</code></li>
        <li><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rank genes for characterizing groups, using 'tl.rank_genes_groups'</code>
          <ul>
            <li><em>‚ÄúThe key of the observations grouping to consider‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúComparison‚Äù</em>: <code class="language-plaintext highlighter-rouge">Compare each group to the union of the rest of the group</code></li>
            <li><em>‚ÄúThe number of genes that appear in the returned tables‚Äù</em>: <code class="language-plaintext highlighter-rouge">100</code></li>
            <li><em>‚ÄúMethod‚Äù</em>: <code class="language-plaintext highlighter-rouge">t-test</code>
              <ul>
                <li><em>‚ÄúP-value correction method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Benjamini-Hochberg</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with t-test</code></p>

      <blockquote class="question">
        <h3 id="question-questions-28"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How is the marker gene information stored in the <code class="language-plaintext highlighter-rouge">AnnData</code> object?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-28"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>An extra object <code class="language-plaintext highlighter-rouge">rank_genes_groups</code> has been added to the <code class="language-plaintext highlighter-rouge">uns</code> object with 5 tables for the 100 top ranked genes (rows) for each cluster (column):</p>

          <ul>
            <li>Names of the rank genes</li>
            <li>Z-scores for the rank genes</li>
            <li>Log2 Fold changes</li>
            <li>P-values</li>
            <li>Adjusted p-values</li>
          </ul>

          <p>This information can be accessed using:</p>
          <ol>
            <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with t-test</code></li>
                <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generalinformation about the object</code></li>
              </ul>
            </li>
            <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with t-test</code></li>
                <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Unstructured annotation (uns)</code>
                  <ul>
                    <li><em>‚ÄúWhat to inspect in uns?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rank gene groups (rank_genes_groups)</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with t-test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Marker genes: Plot ranking of genes using dotplot plot, using 'pl.rank_genes_groups'</code>
          <ul>
            <li><em>‚ÄúNumber of genes to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>‚ÄúNumber of panels per row‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>‚ÄúShould the y-axis of each panels be shared?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with t-test</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Unstructured annotation (uns)</code>
          <ul>
            <li><em>‚ÄúWhat to inspect in uns?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rank gene groups (rank_genes_groups)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Inspect the <code class="language-plaintext highlighter-rouge">Names for rank genes</code> file</li>
  </ol>
</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/rank_genes_groups_t-test.png" alt="Rank genes with t-test" /></p>

<p>We can quickly look at the 5 top ranked genes per cluster:</p>

<table>
  <thead>
    <tr>
      <th>0</th>
      <th>1</th>
      <th>2</th>
      <th>3</th>
      <th>4</th>
      <th>5</th>
      <th>6</th>
      <th>7</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>LDHB</td>
      <td>LYZ</td>
      <td>CD74</td>
      <td>CCL5</td>
      <td>LST1</td>
      <td>NKG7</td>
      <td>CD74</td>
      <td>GPX1</td>
    </tr>
    <tr>
      <td>CD3D</td>
      <td>FTL</td>
      <td>HLA-DRA</td>
      <td>NKG7</td>
      <td>AIF1</td>
      <td>GZMB</td>
      <td>HLA-DRA</td>
      <td>PF4</td>
    </tr>
    <tr>
      <td>RPS12</td>
      <td>TYROBP</td>
      <td>HLA-DPB1</td>
      <td>B2M</td>
      <td>FCER1G</td>
      <td>PRF1</td>
      <td>CST3</td>
      <td>PPBP</td>
    </tr>
    <tr>
      <td>RPS27</td>
      <td>CST3</td>
      <td>CD79A</td>
      <td>IL32</td>
      <td>FTL</td>
      <td>CTSW</td>
      <td>HLA-DPA1</td>
      <td>CALM3</td>
    </tr>
    <tr>
      <td>RPS25</td>
      <td>S100A9</td>
      <td>HLA-DRB1</td>
      <td>CST7</td>
      <td>FTH1</td>
      <td>GNLY</td>
      <td>HLA-DRB1</td>
      <td>NRGN</td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <h3 id="question-questions-29"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Are CST3, NKG7 and PPBP in the set of marker genes? If yes, are they assigned to the clusters we guessed before?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-29"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ul>
      <li>CST3 is a marker gene for clusters 1, 4, 6 (not 3 as guessed previously)</li>
      <li>NKG7 for clusters 3 and 5 (not 0 as guessed previously)</li>
      <li>PPBP for cluster 7, as guessed previously</li>
    </ul>

  </blockquote>
</blockquote>

<h2 id="using-wilcoxon-rank-sum-test">Using Wilcoxon rank sum test</h2>

<p>Another widely used method for pairwise comparisons between groups of observations is the Wilcoxon rank sum test (also known as the Wilcoxon-Mann-Whitney test). This test directly assesses separation between the <span class="notranslate">expression</span> distributions of different clusters. It is the recommended approach to use in publication (<a class="citation" href="#soneson2018bias">Soneson and Robinson 2018</a>).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-rank-the-highly-differential-genes-using-wilcoxon-rank-sum"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Rank the highly differential genes using Wilcoxon rank sum</h3>

  <ol>
    <li><strong>Inspect and manipulate</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>
          <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering</code></p>

          <p><small><strong>Note:</strong> <em>Please pay attention to the dataset name.</em></small></p>
        </li>
        <li>
          <p><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rank genes for characterizing groups, using 'tl.rank_genes_groups'</code></p>
          <ul>
            <li><em>‚ÄúThe key of the observations grouping to consider‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúComparison‚Äù</em>: <code class="language-plaintext highlighter-rouge">Compare each group to the union of the rest of the group</code></li>
            <li><em>‚ÄúThe number of genes that appear in the returned tables‚Äù</em>: <code class="language-plaintext highlighter-rouge">100</code></li>
            <li><em>‚ÄúMethod‚Äù</em>: <code class="language-plaintext highlighter-rouge">Wilcoxon-Rank-Sum</code>
              <ul>
                <li><em>‚ÄúP-value correction method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Benjamini-Hochberg</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></p>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Marker genes: Plot ranking of genes using dotplot plot, using 'pl.rank_genes_groups'</code>
          <ul>
            <li><em>‚ÄúNumber of genes to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>‚ÄúNumber of panels per row‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
            <li><em>‚ÄúShould the y-axis of each panels be shared?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Unstructured annotation (uns)</code>
          <ul>
            <li><em>‚ÄúWhat to inspect in uns?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rank gene groups (rank_genes_groups)</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the <code class="language-plaintext highlighter-rouge">Names for rank genes</code> file to <code class="language-plaintext highlighter-rouge">Ranked genes with Wilcoxon test</code></li>
    <li>Inspect <code class="language-plaintext highlighter-rouge">Ranked genes with Wilcoxon test</code> file</li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/rank_genes_groups_wilcoxon.png" alt="Rank genes with Wilcoxon" /></p>

<p>We can quickly look at the 5 top ranked genes per cluster:</p>

<table>
  <thead>
    <tr>
      <th>0</th>
      <th>1</th>
      <th>2</th>
      <th>3</th>
      <th>4</th>
      <th>5</th>
      <th>6</th>
      <th>7</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>LDHB</td>
      <td>LYZ</td>
      <td>CD74</td>
      <td>CCL5</td>
      <td>LST1</td>
      <td>NKG7</td>
      <td>HLA-DPA1</td>
      <td>PF4</td>
    </tr>
    <tr>
      <td>RPS12</td>
      <td>S100A9</td>
      <td>CD79A</td>
      <td>NKG7</td>
      <td>FCER1G</td>
      <td>GZMB</td>
      <td>HLA-DPB1</td>
      <td>GNG11</td>
    </tr>
    <tr>
      <td>RPS25</td>
      <td>S100A8</td>
      <td>HLA-DRA</td>
      <td>CST7</td>
      <td>AIF1</td>
      <td>PRF1</td>
      <td>HLA-DRB1</td>
      <td>SDPR</td>
    </tr>
    <tr>
      <td>RPS27</td>
      <td>TYROBP</td>
      <td>CD79B</td>
      <td>CTSW</td>
      <td>COTL1</td>
      <td>GNLY</td>
      <td>HLA-DRA</td>
      <td>PPBP</td>
    </tr>
    <tr>
      <td>RPS6</td>
      <td>FTL</td>
      <td>HLA-DPB1</td>
      <td>B2M</td>
      <td>FCGR3A</td>
      <td>CTSW</td>
      <td>CD74</td>
      <td>NRGN</td>
    </tr>
    <tr>
      <td>RPS3</td>
      <td>CST3</td>
      <td>HLA-DQA1</td>
      <td>GZMA</td>
      <td>IFITM2</td>
      <td>GZMA</td>
      <td>CST3</td>
      <td>SPARC</td>
    </tr>
    <tr>
      <td>CD3D</td>
      <td>FCN1</td>
      <td>MS4A1</td>
      <td>HLA-C</td>
      <td>FTH1</td>
      <td>CST7</td>
      <td>HLA-DQA1</td>
      <td>GPX1</td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <h3 id="question-questions-30"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <ol>
    <li>Are the 5 top ranked genes different than the one for the <em>t</em>-test?</li>
    <li>Are CST3, NKG7 and PPBP in the marker genes? If yes, are they assigned to the clusters we guessed before?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-30"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The 5 top ranked genes are slightly different, at least in their order.</li>
      <li>We see that:
        <ul>
          <li>CST3 is a ranked genes for clusters 1, 4, 6 (not 3 as guessed previously)</li>
          <li>NKG7 for clusters 3 and 5 (not 0 as guessed previously)</li>
          <li>PPBP for cluster 7, as we guessed previously.</li>
        </ul>
      </li>
    </ol>

  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-compare-differential-expression-for-cst3-nkg7-and-ppbp-in-the-different-clusters"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Compare differential <span class="notranslate">expression</span> for CST3, NKG7 and PPBP in the different clusters</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generic: Violin plot, using 'pl.violin'</code>
          <ul>
            <li><em>‚ÄúKeys for accessing variables‚Äù</em>: <code class="language-plaintext highlighter-rouge">Subset of variables in 'adata.var_names' or fields in '.obs'</code>
              <ul>
                <li><em>‚ÄúKeys for accessing variables‚Äù</em>: <code class="language-plaintext highlighter-rouge">CST3, NKG7, PPBP</code></li>
              </ul>
            </li>
            <li><em>‚ÄúThe key of the observation grouping to consider‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/violin_plot_rank_genes_groups_CST3_NKG7_PPBP.png" alt="Violin plot for CST3, NKG7 and PPBP after clustering" /></p>

<blockquote class="question">
  <h3 id="question-questions-31"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>Are CST3, NKG7 and PPBP more expressed in the clusters for which they are marker genes?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-31"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ul>
      <li>CST3 is more expressed in clusters 1, 4, 6, the ones for which it was previously found as a marker gene, but also in cluster 7, which is unexpected.</li>
      <li>NKG7 is more expressed in clusters 3 and 5, the ones for which it was previously found as a marker gene.</li>
      <li>PPBP is very pronounced in cluster 7, for which it was previously found as a marker gene.</li>
    </ul>

  </blockquote>
</blockquote>

<h2 id="visualization-of-expression-of-the-marker-genes">Visualization of <span class="notranslate">expression</span> of the marker genes</h2>

<p>The marker genes should be more expressed in the clusters for which they are markers. We would like to confirm this idea using some visualization.</p>

<h3 id="expression-of-the-top-marker-genes-in-each-cluster"><span class="notranslate">Expression</span> of the top marker genes in each cluster</h3>

<p>The assumption should be even more true for the top marker genes. The first way to visualize the <span class="notranslate">expression</span> of the top marker genes is to look at the distribution of the <span class="notranslate">expression</span> of each marker gene in cells for each cluster.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-expression-probability-distributions-across-clusters-of-top-marker-genes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot <span class="notranslate">expression</span> probability distributions across clusters of top marker genes</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generic: Stacked violin plot, using 'pl.stacked_violin'</code>
          <ul>
            <li><em>‚ÄúVariables to plot (columns of the heatmaps)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Subset of variables in 'adata.var_names'</code>
              <ul>
                <li><em>‚ÄúList of variables to plot‚Äù</em>: <code class="language-plaintext highlighter-rouge">LDHB, LYZ, CD74, CCL5, LST1, NKG7, HLA-DPA1, PF4</code></li>
              </ul>
            </li>
            <li><em>‚ÄúThe key of the observation grouping to consider‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúNumber of categories‚Äù</em>: <code class="language-plaintext highlighter-rouge">8</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúCustom figure size</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúSwap axes?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li>In <em>‚ÄúViolin plot attributes‚Äù</em>:
              <ul>
                <li><em>‚ÄúAdd a stripplot on top of the violin plot‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/violin_plot_marker_genes.png" alt="Violin plot for top marker genes" /></p>

<blockquote class="question">
  <h3 id="question-questions-32"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>Are the top marker genes only expressed in the cluster for which they are markers?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-32"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ul>
      <li>
        <p>LDHB, LYZ and CD74, even if they are top markers genes for the cluster 0, 1, 2 respectively, are also expressed in all other clusters (and also found in the top 100 marker genes for other clusters), but with higher level in the cluster for they are markers.</p>
      </li>
      <li>
        <p>CCL5, LST1, NKG7 and HLA-DPA1 are not expressed in all clusters but also not only in the one they are markers.</p>
      </li>
      <li>
        <p>PF4 is only expressed in cluster 7.</p>
      </li>
    </ul>

  </blockquote>
</blockquote>

<p>Another approach consists of displaying the mean <span class="notranslate">expression</span> of the marker genes for the cells on the neighborhood graph.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-top-marker-gene-expression-on-an-umap-plot"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot top marker gene <span class="notranslate">expression</span> on an UMAP plot</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Embeddings: Scatter plot in UMAP basis, using 'pl.umap'</code>
          <ul>
            <li><em>‚ÄúKeys for annotations of observations/cells or variables/genes‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain, LDHB, LYZ, CD74, CCL5, LST1, NKG7, HLA-DPA1, PF4</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li>In <em>‚ÄúPlot attributes‚Äù</em>
              <ul>
                <li><em>‚ÄúNumber of panels per row‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/umap_plot_marker_genes.png" alt="Top marker gene expression on a UMAP plot" /></p>

<blockquote class="question">
  <h3 id="question-questions-33"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>Are the top marker genes clearly associated to their clusters?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-33"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>For most genes, we can clearly see for which cluster they are marker genes. But for physically close clusters, the differences are not so obvious.</p>

  </blockquote>
</blockquote>

<h3 id="expression-of-the-top-20-markers-genes-in-cells-for-each-cluster"><span class="notranslate">Expression</span> of the top 20 markers genes in cells for each cluster</h3>

<p>We would like now to have a look at the <span class="notranslate">expression</span> of the top 20 marker genes in the different cells for each cluster</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-heatmap-of-the-gene-expression-in-cells"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot heatmap of the gene <span class="notranslate">expression</span> in cells</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Marker genes: Plot ranking of genes as heatmap plot, using 'pl.rank_genes_groups_heatmap'</code>
          <ul>
            <li><em>‚ÄúNumber of genes to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúCompute and plot a dendrogram?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/plot_rank_genes_groups_heatmap.png" alt="Heatmap of the gene expression in cells" /></p>

<p>Each cells is shown in a row and in columns are the marker genes for each cluster.</p>

<blockquote class="question">
  <h3 id="question-questions-34"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <p>How are clusters close to each other in terms of <span class="notranslate">expression</span> of the top 20 marker genes?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-34"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ul>
      <li>
        <p>Clusters 0, 3, and 5 are similar in term of <span class="notranslate">expression</span>. This was expected as they are physically close on the neighborhood graph.</p>
      </li>
      <li>
        <p>Clusters 1 and 4 are together and then 2 and 6 are together after 7. These observations are less expected given the neighborhood graph: 1 and 4 are physically close, but 2 is far from 7 and 6.</p>
      </li>
    </ul>

  </blockquote>
</blockquote>

<h2 id="comparison-of-the-marker-genes-between-clusters">Comparison of the marker genes between clusters</h2>

<p>Previously we identified marker genes by taking gene <span class="notranslate">expression</span> in cells for each cluster individually and comparing them to all remaining cells.</p>

<p>In some cases, it may also be interesting to find marker genes distinguishing one given cluster from one or several clusters. For example, the marker genes distinguishing cluster 0 from cluster 1.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-identify-the-marker-genes-distinguishing-cluster-0-from-cluster-1-using-wilcoxon-rank-sum"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Identify the marker genes distinguishing cluster 0 from cluster 1 using Wilcoxon rank sum</h3>

  <ol>
    <li><strong>Inspect and manipulate</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for inspecting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rank genes for characterizing groups, using 'tl.rank_genes_groups'</code>
          <ul>
            <li><em>‚ÄúThe key of the observations grouping to consider‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúSubset of groups to which comparison shall be restricted‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
            <li><em>‚ÄúComparison‚Äù</em>: <code class="language-plaintext highlighter-rouge">Compare with respect to a specific group</code>
              <ul>
                <li><em>‚ÄúGroup identifier with respect to which compare‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
              </ul>
            </li>
            <li><em>‚ÄúThe number of genes that appear in the returned tables‚Äù</em>: <code class="language-plaintext highlighter-rouge">100</code></li>
            <li><em>‚ÄúMethod‚Äù</em>: <code class="language-plaintext highlighter-rouge">Wilcoxon-Rank-Sum</code>
              <ul>
                <li><em>‚ÄúP-value correction method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Benjamini-Hochberg</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes for 0 vs 1 with Wilcoxon test</code></p>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes for 0 vs 1 with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Marker genes: Plot ranking of genes using dotplot plot, using 'pl.rank_genes_groups'</code>
          <ul>
            <li><em>‚ÄúNumber of genes to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">20</code></li>
            <li><em>‚ÄúShould the y-axis of each panels be shared?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/rank_genes_groups_wilcoxon_0_vs_1.png" alt="Rank genes 0 vs 1 with Wilcoxon" /></p>

<blockquote class="question">
  <h3 id="question-questions-35"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How can we interpret this plot?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-35"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>In this graph are the marker genes distinguishing cluster 0 from cluster 1, ranked based on their difference of <span class="notranslate">expression</span> of the genes between cells in both clusters. So MALAT1 is the most differentially expressed gene between cells in cluster 0 and cells in cluster 1, even though it was not in the set of  top marker genes for cluster 0.</p>

  </blockquote>
</blockquote>

<p>The marker genes distinguishing cluster 0 from cluster 1 are extracted based on their differences in <span class="notranslate">expression</span>, which can be easily visualized.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-expression-difference-for-the-marker-genes-distinguishing-cluster-0-from-cluster-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot <span class="notranslate">expression</span> difference for the marker genes distinguishing cluster 0 from cluster 1</h3>
  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes for 0 vs 1 with Wilcoxon test</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Marker genes: Plot ranking of genes as violin plot, using 'pl.rank_genes_groups_violin'</code>
          <ul>
            <li><em>‚ÄúWhich genes to plot?‚Äù</em>: <code class="language-plaintext highlighter-rouge">A number of genes</code>
              <ul>
                <li><em>‚ÄúNumber of genes to show‚Äù</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
              </ul>
            </li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/rank_genes_groups_violin_wilcoxon_0_vs_1.png" alt="Violin plot for marker genes for clusters 0 vs 1 with Wilcoxon" /></p>

<p>Previous visualizations like the heatmap can also be used to represent the differential <span class="notranslate">expression</span> of the marker genes between both clusters.</p>

<p>In the next steps, we are mostly interested in the marker genes for each cluster individually by comparison one cluster to the rest, instead of a 1-to-1 comparison. So we will use again the AnnData object called <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code>.</p>

<h1 id="cell-type-annotation">Cell type annotation</h1>

<p>Obtaining clusters of cells is quite straightforward. Determining what biological state is represented by each of those clusters is likely the most challenging task in sc<span class="notranslate">RNA</span>-Seq data analysis. To do so, we need to bridge the gap between our current dataset and prior biological knowledge.</p>

<p>This biological knowledge is not always available in a consistent and quantitative manner. For example, the concept of ‚Äúcell type‚Äù is not clearly defined. The interpretation of sc<span class="notranslate">RNA</span>-seq data is often then quite manual.</p>

<p>Fortunately in the case of our dataset, we can use canonical markers to known cell types:</p>

<table>
  <thead>
    <tr>
      <th>Cell type</th>
      <th>Marker genes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>CD4+ T cells</td>
      <td>IL7R, CCR7</td>
    </tr>
    <tr>
      <td>CD8+ T cells</td>
      <td>CD8A</td>
    </tr>
    <tr>
      <td>CD14+ Monocytes</td>
      <td>CD14, LYZ</td>
    </tr>
    <tr>
      <td>B cells</td>
      <td>MS4A1, CD79A</td>
    </tr>
    <tr>
      <td>Natural killer (NK) cells</td>
      <td>GNLY, NKG7, KLRB1</td>
    </tr>
    <tr>
      <td>Dendritic Cells</td>
      <td>FCER1A, CST3</td>
    </tr>
    <tr>
      <td>Megakaryocytes</td>
      <td>PPBP</td>
    </tr>
    <tr>
      <td>FCGR3A+ Monocytes</td>
      <td>FCGR3A, MS4A7</td>
    </tr>
  </tbody>
</table>

<blockquote class="tip">
  <h3 id="tip-how-to-find-canonical-markers"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> How to find canonical markers?</h3>

  <p>Canonical markers are usually found in the literature and are also aggregated into dedicated database like the <a href="https://panglaodb.se/markers.html">PanglaoDB</a> <a class="citation" href="#franzen2019panglaodb">Franz√©n <i>et al.</i> 2019</a></p>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions-36"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Is it possible to match the clusters to cell types using the previous table?</p>

  <p><em>Hint</em>: search for the marker genes in the table in the <code class="language-plaintext highlighter-rouge">Ranked genes with Wilcoxon test</code> dataset</p>

  <blockquote class="solution">
    <h3 id="solution-solution-36"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>By searching the marker genes in the the <code class="language-plaintext highlighter-rouge">Ranked genes with Wilcoxon test</code> dataset:</p>

    <table>
      <thead>
        <tr>
          <th>Marker genes</th>
          <th>Cell type</th>
          <th>Cluster</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>IL7R</td>
          <td>CD4+ T cells</td>
          <td>0</td>
        </tr>
        <tr>
          <td>CCR7</td>
          <td>CD4+ T cells</td>
          <td>0</td>
        </tr>
        <tr>
          <td>CD8A</td>
          <td>CD8+ T cells</td>
          <td>3</td>
        </tr>
        <tr>
          <td>CD14</td>
          <td>CD14+ Monocytes</td>
          <td>1</td>
        </tr>
        <tr>
          <td>LYZ</td>
          <td>CD14+ Monocytes</td>
          <td>1 (6)</td>
        </tr>
        <tr>
          <td>MS4A1</td>
          <td>B cells</td>
          <td>2</td>
        </tr>
        <tr>
          <td>CD79A</td>
          <td>B cells</td>
          <td>2</td>
        </tr>
        <tr>
          <td>GNLY</td>
          <td>Natural killer (NK) cells</td>
          <td>5</td>
        </tr>
        <tr>
          <td>NKG7</td>
          <td>Natural killer (NK) cells</td>
          <td>5 (3)</td>
        </tr>
        <tr>
          <td>KLRB1</td>
          <td>Natural killer (NK) cells</td>
          <td>-</td>
        </tr>
        <tr>
          <td>FCER1A</td>
          <td>Dendritic cells</td>
          <td>6</td>
        </tr>
        <tr>
          <td>CST3</td>
          <td>Dendritic cells</td>
          <td>6 (4)</td>
        </tr>
        <tr>
          <td>PPBP</td>
          <td>Megakaryocytes</td>
          <td>7</td>
        </tr>
        <tr>
          <td>FCGR3A</td>
          <td>FCGR3A+ Monocytes</td>
          <td>4 (5)</td>
        </tr>
      </tbody>
    </table>

    <p>We can then aggregate the results by clusters</p>

    <table>
      <thead>
        <tr>
          <th>Cell type</th>
          <th>Marker genes</th>
          <th>Cluster</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>CD4+ T cells</td>
          <td>IL7R, CCR7</td>
          <td>0</td>
        </tr>
        <tr>
          <td>CD8+ T cells</td>
          <td>CD8A</td>
          <td>3</td>
        </tr>
        <tr>
          <td>CD14+ Monocytes</td>
          <td>CD14, LYZ</td>
          <td>1</td>
        </tr>
        <tr>
          <td>B cells</td>
          <td>MS4A1, CD79A</td>
          <td>2</td>
        </tr>
        <tr>
          <td>Natural killer (NK) cells</td>
          <td>GNLY, NKG7, KLRB1</td>
          <td>5</td>
        </tr>
        <tr>
          <td>Dendritic Cells</td>
          <td>FCER1A, CST3</td>
          <td>6</td>
        </tr>
        <tr>
          <td>Megakaryocytes</td>
          <td>PPBP</td>
          <td>7</td>
        </tr>
        <tr>
          <td>FCGR3A+ Monocytes</td>
          <td>FCGR3A, MS4A7</td>
          <td>4</td>
        </tr>
      </tbody>
    </table>

  </blockquote>
</blockquote>

<p>These canonical marker genes can be easily match the clusters to known cell types:</p>

<table>
  <thead>
    <tr>
      <th>Cluster</th>
      <th>Cell type</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>0</td>
      <td>CD4+ T cells</td>
    </tr>
    <tr>
      <td>1</td>
      <td>CD14+ Monocytes</td>
    </tr>
    <tr>
      <td>2</td>
      <td>B cells</td>
    </tr>
    <tr>
      <td>3</td>
      <td>CD8+ T cells</td>
    </tr>
    <tr>
      <td>4</td>
      <td>FCGR3A+ Monocytes</td>
    </tr>
    <tr>
      <td>5</td>
      <td>Natural killer (NK) cells</td>
    </tr>
    <tr>
      <td>6</td>
      <td>Dendritic Cells</td>
    </tr>
    <tr>
      <td>7</td>
      <td>Megakaryocytes</td>
    </tr>
  </tbody>
</table>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-add-the-cell-type-as-cluster-names"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Add the cell type as cluster names</h3>

  <ol>
    <li><strong>Manipulate AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>
          <p><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test</code></p>

          <p><small><strong>Note</strong>: <em>Take note that this is not the ‚Äú0 vs 1 Wilcoxon‚Äù dataset</em></small></p>
        </li>
        <li>
          <p><em>‚ÄúFunction to manipulate the object‚Äù</em>: <code class="language-plaintext highlighter-rouge">Rename categories of annotation</code></p>
          <ul>
            <li><em>‚ÄúKey for observations or variables annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúComma-separated list of new categories‚Äù</em>: <code class="language-plaintext highlighter-rouge">CD4+ T, CD14+, B, CD8+ T, FCGR3A+, NK, Dendritic, Megakaryocytes</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Rename the generated output <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test, annotation</code></p>

      <blockquote class="question">
        <h3 id="question-questions-37"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <p>How can we check that the cell type has been correctly added?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-37"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <p>The cluster information is available in the <code class="language-plaintext highlighter-rouge">obs</code> attribute of the AnnData object. So to check, we should run <strong>Inspect AnnData</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA and KNN graph</code></li>
            <li><em>‚ÄúWhat to inspect?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Key-indexed observations annotation (obs)</code></li>
          </ul>

          <p>In the last column (called <code class="language-plaintext highlighter-rouge">louvain</code>), we should now have cell type.</p>
        </blockquote>
      </blockquote>
    </li>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test, annotation</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Embeddings: Scatter plot in UMAP basis, using 'pl.umap'</code>
          <ul>
            <li><em>‚ÄúKeys for annotations of observations/cells or variables/genes‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li>In <em>‚ÄúPlot attributes‚Äù</em>
              <ul>
                <li><em>‚ÄúLocation of legend‚Äù</em>: <code class="language-plaintext highlighter-rouge">on data</code></li>
                <li><em>‚ÄúDraw a frame around the scatter plot?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/umap_annotated_clusters.png" alt="UMAP plot with annotated clusters" /></p>

<blockquote class="question">
  <h3 id="question-questions-38"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>How well are the cell types clustered in the neighborhood graph?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-38"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>T cells (CD4+ and CD8+) are clustered together with NK cells. Monocytes cells (CD14+ and FCGR3A+) are close to each others, with Dendritic and Megakaryocytes. B cells are physically independent.</p>

  </blockquote>
</blockquote>

<p>With the annotated cell types, we can also visualize the <span class="notranslate">expression</span> of their canonical marker genes.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-plot-expression-of-canonical-marker-genes-for-the-annotated-cell-types"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Plot <span class="notranslate">expression</span> of canonical marker genes for the annotated cell types</h3>

  <ol>
    <li><strong>Plot</strong> with scanpy <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAnnotated data matrix‚Äù</em>: <code class="language-plaintext highlighter-rouge">3k PBMC with only HVG, after scaling, PCA, KNN graph, UMAP, clustering, marker genes with Wilcoxon test, annotation</code></li>
        <li><em>‚ÄúMethod used for plotting‚Äù</em>: <code class="language-plaintext highlighter-rouge">Generic: Makes a dot plot of the <span class="notranslate">expression</span> values, using 'pl.dotplot'</code>
          <ul>
            <li><em>‚ÄúVariables to plot (columns of the heatmaps)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Subset of variables in 'adata.var_names'</code>
              <ul>
                <li><em>‚ÄúList of variables to plot‚Äù</em>: <code class="language-plaintext highlighter-rouge">IL7R, CCR7, CD8A, CD14, LYZ, MS4A1, CD79A, GNLY, NKG7, KLRB1, FCER1A, CST3, PPBP, FCGR3A</code></li>
              </ul>
            </li>
            <li><em>‚ÄúThe key of the observation grouping to consider‚Äù</em>: <code class="language-plaintext highlighter-rouge">louvain</code></li>
            <li><em>‚ÄúNumber of categories‚Äù</em>: <code class="language-plaintext highlighter-rouge">8</code></li>
            <li><em>‚ÄúUse ‚Äòraw‚Äô attribute of input if present‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúCompute and plot a dendrogram?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li>In <em>‚ÄúGroup of variables to highlight‚Äù</em>
              <ul>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú1: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">CD4+ T</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú2: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">CD8+</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú3: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">4</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">CD14+</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú4: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">6</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">B</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú5: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">7</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">9</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">NK</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú6: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">11</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dendritic</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú7: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">12</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">12</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">Megakaryocytes</code></li>
                  </ul>
                </li>
                <li>Click on <i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúGroup of variables to highlight‚Äù</em></li>
                <li>In <em>‚Äú8: Group of variables to highlight‚Äù</em>
                  <ul>
                    <li><em>‚ÄúStart‚Äù</em>: <code class="language-plaintext highlighter-rouge">13</code></li>
                    <li><em>‚ÄúEnd‚Äù</em>: <code class="language-plaintext highlighter-rouge">13</code></li>
                    <li><em>‚ÄúLabel‚Äù</em>: <code class="language-plaintext highlighter-rouge">FCGR3A+</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/scrna-scanpy-pbmc3k/dotplot_annotated_clusters.png" alt="Dotplot plot with annotated clusters" /></p>

<blockquote class="question">
  <h3 id="question-questions-39"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Are the canonical marker genes only expressed in their cell type?</li>
    <li>Are the previous physical cluster (UMAP graph) confirmed?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-39"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Some canonical marker genes like LYZ or CST3 are not only highly expressed in their cell type but also on closely related other cell types.</li>
      <li>As seen on the UMAP graph, T cells (CD4+ and CD8+) are clustered together with NK cells, but NK cells and CD8+ closer to each other. Monocytes cells (CD14+ and FCGR3A+) are close to each others, but B cells are close to Dendritic and then Megakaryocytes despite B cells being physically independent on the neighborhood graph.</li>
    </ol>

  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we investigated clustering and annotation of <span class="notranslate">single-cell</span> data from 10x Genomics using Scanpy. This <span class="notranslate">workflow</span> used here was typical for sc<span class="notranslate">RNA</span>-seq data analysis:</p>

<ol>
  <li>Preprocessing with
    <ol>
      <li>Selection and filtration of cells and genes based on quality metrics</li>
      <li>Data normalization and scaling</li>
      <li>Selection of features, i.e. marker genes</li>
    </ol>
  </li>
  <li>Reduction of the dimensionality via a Principal Component Analysis</li>
  <li>Clustering of the cells by
    <ol>
      <li>Computation of a neighborhood graph</li>
      <li>Clustering of the neighborhood graph into 8 clusters of cells</li>
    </ol>
  </li>
  <li>Identification of marker genes for the clusters</li>
  <li>Annotation of the clusters with cell types</li>
</ol>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>scRNA-seq data analysis is complex and exploratory process, still in development</p>
</li>
                            
                            <li><p>Different tools and parameters should be tested for each step of the process</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="blondel2008fast">Blondel, V. D., J.-L. Guillaume, R. Lambiotte, and E. Lefebvre, 2008 <b>Fast unfolding of communities in large networks</b>. Journal of statistical mechanics: theory and experiment 2008: P10008.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://iopscience.iop.org/article/10.1088/1742-5468/2008/10/P10008/meta">https://iopscience.iop.org/article/10.1088/1742-5468/2008/10/P10008/meta</a>
      
    

</li>
<li><span id="islam2014quantitative">Islam, S., A. Zeisel, S. Joost, G. La Manno, P. Zajac <i>et al.</i>, 2014 <b>Quantitative single-cell RNA-seq with unique molecular identifiers</b>. Nature methods 11: 163.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.nature.com/nmeth/journal/v11/n2/abs/nmeth.2772.html">https://www.nature.com/nmeth/journal/v11/n2/abs/nmeth.2772.html</a>
      
    

</li>
<li><span id="levine2015data">Levine, J. H., E. F. Simonds, S. C. Bendall, K. L. Davis, D. A. El-ad <i>et al.</i>, 2015 <b>Data-driven phenotypic dissection of AML reveals progenitor-like cells that correlate with prognosis</b>. Cell 162: 184‚Äì197.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.sciencedirect.com/science/article/pii/S0092867415006376">https://www.sciencedirect.com/science/article/pii/S0092867415006376</a>
      
    

</li>
<li><span id="stegle2015computational">Stegle, O., S. A. Teichmann, and J. C. Marioni, 2015 <b>Computational and analytical challenges in single-cell transcriptomics</b>. Nature Reviews Genetics 16: 133‚Äì145.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.nature.com/articles/nrg3833">https://www.nature.com/articles/nrg3833</a>
      
    

</li>
<li><span id="ilicic2016classification">Ilicic, T., J. K. Kim, A. A. Kolodziejczyk, F. O. Bagger, D. J. McCarthy <i>et al.</i>, 2016 <b>Classification of low quality cells from single-cell RNA-seq data</b>. Genome biology 17: 29.</span>
    
    
    

</li>
<li><span id="butler2018integrating">Butler, A., P. Hoffman, P. Smibert, E. Papalexi, and R. Satija, 2018 <b>Integrating single-cell transcriptomic data across different conditions, technologies, and species</b>. Nature biotechnology 36: 411.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.nature.com/articles/nbt.4096?draft=journal">https://www.nature.com/articles/nbt.4096?draft=journal</a>
      
    

</li>
<li><span id="mcinnes2018umap">McInnes, L., J. Healy, and J. Melville, 2018 <b>Umap: Uniform manifold approximation and projection for dimension reduction</b>. arXiv preprint arXiv:1802.03426.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://arxiv.org/abs/1802.03426">https://arxiv.org/abs/1802.03426</a>
      
    

</li>
<li><span id="soneson2018bias">Soneson, C., and M. D. Robinson, 2018 <b>Bias, robustness and scalability in single-cell differential expression analysis</b>. Nature methods 15: 255.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.nature.com/nmeth/journal/v15/n4/abs/nmeth.4612.html">https://www.nature.com/nmeth/journal/v15/n4/abs/nmeth.4612.html</a>
      
    

</li>
<li><span id="wolf2018scanpy">Wolf, F. A., P. Angerer, and F. J. Theis, 2018 <b>SCANPY: large-scale single-cell gene expression data analysis</b>. Genome biology 19: 15.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-017-1382-0</a>
      
    

</li>
<li><span id="amezquita2019orchestrating">Amezquita, R. A., V. J. Carey, L. N. Carpp, L. Geistlinger, A. T. L. Lun <i>et al.</i>, 2019 <b>Orchestrating Single-Cell Analysis with Bioconductor</b>. bioRxiv 590562.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.biorxiv.org/content/10.1101/590562v1">https://www.biorxiv.org/content/10.1101/590562v1</a>
      
    

</li>
<li><span id="becht2019dimensionality">Becht, E., L. McInnes, J. Healy, C.-A. Dutertre, I. W. H. Kwok <i>et al.</i>, 2019 <b>Dimensionality reduction for visualizing single-cell data using UMAP</b>. Nature biotechnology 37: 38.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.nature.com/articles/nbt.4314">https://www.nature.com/articles/nbt.4314</a>
      
    

</li>
<li><span id="franzen2019panglaodb">Franz√©n, O., L.-M. Gan, and J. L. M. Bj√∂rkegren, 2019 <b>PanglaoDB: a web server for exploration of mouse and human single-cell RNA sequencing data</b>. Database 2019:</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://academic.oup.com/database/article-abstract/doi/10.1093/database/baz046/5427041">https://academic.oup.com/database/article-abstract/doi/10.1093/database/baz046/5427041</a>
      
    

</li>
<li><span id="stuart2019comprehensive">Stuart, T., A. Butler, P. Hoffman, C. Hafemeister, E. Papalexi <i>et al.</i>, 2019 <b>Comprehensive Integration of Single-Cell Data</b>. Cell.</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.sciencedirect.com/science/article/pii/S0092867419305598">https://www.sciencedirect.com/science/article/pii/S0092867419305598</a>
      
    

</li>
<li><span id="traag2019louvain">Traag, V. A., L. Waltman, and N. J. van Eck, 2019 <b>From Louvain to Leiden: guaranteeing well-connected communities</b>. Scientific reports 9:</span>
    
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6435756/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6435756/</a>
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Clustering 3K PBMCs with Scanpy (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">B√©r√©nice Batut, Hans-Rudolf Hotz, Mehmet Tekman, 2021 <b>Clustering 3K PBMCs with Scanpy (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html">/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-scrna-scanpy-pbmc3k,
    author = "B√©r√©nice Batut and Hans-Rudolf Hotz and Mehmet Tekman",
    title = "Clustering 3K PBMCs with Scanpy (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (B√©r√©nice Batut, Hans-Rudolf Hotz, Mehmet Tekman)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/scrna-scanpy-pbmc3k/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
