<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Whole transcriptome analysis of Arabidopsis thaliana / Whole transcriptome analysis of Arabidopsis thaliana / Transcriptomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@8a9b0dca294ebaf945f5b32f9e7854b7eca4b918",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Whole transcriptome analysis of Arabidopsis thaliana">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-08-09 08:51:50 +0000">
<meta name="DC.creator" content="Cristóbal Gallardo">
<meta name="DC.creator" content="Pavankumar Videm">
<meta name="DC.creator" content="Beatriz Serrano-Solano"><meta name="description" content="Training material for all kinds of transcriptomics analysis.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Transcriptomics / Whole transcriptome analysis of Arabidopsis thaliana / Hands-on: Whole transcriptome analysis of Arabidopsis thaliana">
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis.">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Whole transcriptome analysis of Arabidopsis thaliana",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-08-09 08:51:50 +0000",
  "datePublished": "2021-04-08 16:29:49 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00292",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
  },
  "abstract": "As sessile organisms, the survival of plants under adverse environmental conditions depends, to a large extent, on their ability to perceive stress stimuli and respond appropriately to counteract the potentially damaging effects. Coordination of phytohormones and reactive oxygen species are considered a key element for enhancing stress resistance, allowing fine-tuning of gene expression in response to environmental changes,. These molecules constitute complex signalling networks, endowing with the ability to respond to a variable natural environment.",
  "learningResourceType": "e-learning",
  "name": "Whole transcriptome analysis of Arabidopsis thaliana",
  "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html",
  "version": 17,
  "timeRequired": "PT2H",
  "teaches": "- Perform miRNA differential expression analysis\n- Understand the quasi-mapping-based Salmon method for quantifying the expression of transcripts using RNA-Seq data\n- Idenfity potential miRNAs involved in brassinosteroid-mediated regulation networks",
  "keywords": [
    "Transcriptomics",
    "miRNA",
    "plants",
    "stress tolerance"
  ],
  "description": "## Abstract\n\nAs sessile organisms, the survival of plants under adverse environmental conditions depends, to a large extent, on their ability to perceive stress stimuli and respond appropriately to counteract the potentially damaging effects. Coordination of phytohormones and reactive oxygen species are considered a key element for enhancing stress resistance, allowing fine-tuning of gene expression in response to environmental changes,. These molecules constitute complex signalling networks, endowing with the ability to respond to a variable natural environment.\n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - Which miRNAs are upregulated in response to brassinosteroids?\n - Which genes are potential target of brassinosteroid-induced miRNAs?\n\n\n## Learning Objectives\n\n- Perform miRNA differential expression analysis\n- Understand the quasi-mapping-based Salmon method for quantifying the expression of transcripts using RNA-Seq data\n- Idenfity potential miRNAs involved in brassinosteroid-mediated regulation networks\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/gallardoalba/",
      "name": "Cristóbal Gallardo",
      "image": "https://avatars.githubusercontent.com/gallardoalba",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5752-2155",
      "orcid": "https://orcid.org/0000-0002-5752-2155"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/pavanvidem/",
      "name": "Pavankumar Videm",
      "image": "https://avatars.githubusercontent.com/pavanvidem",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5192-126X",
      "orcid": "https://orcid.org/0000-0002-5192-126X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/beatrizserrano/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/beatrizserrano/",
      "name": "Beatriz Serrano-Solano",
      "image": "https://avatars.githubusercontent.com/beatrizserrano",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-5862-6132",
      "orcid": "https://orcid.org/0000-0002-5862-6132"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/mirna-target-finder/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/4710649",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>Whole transcriptome analysis of Arabidopsis thaliana</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/gallardoalba/" class="contributor-badge contributor-gallardoalba"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/gallardoalba?s=36" alt="Cristóbal Gallardo avatar" width="36" class="avatar" />
    Cristóbal Gallardo</a><a href="/training-material/hall-of-fame/pavanvidem/" class="contributor-badge contributor-pavanvidem"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/pavanvidem?s=36" alt="Pavankumar Videm avatar" width="36" class="avatar" />
    Pavankumar Videm</a><a href="/training-material/hall-of-fame/beatrizserrano/" class="contributor-badge contributor-beatrizserrano"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/beatrizserrano?s=36" alt="Beatriz Serrano-Solano avatar" width="36" class="avatar" />
    Beatriz Serrano-Solano</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which miRNAs are upregulated in response to brassinosteroids?</p>
</li>
        
        <li><p>Which genes are potential target of brassinosteroid-induced miRNAs?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Perform miRNA differential expression analysis</p>
</li>
        
        <li><p>Understand the quasi-mapping-based Salmon method for quantifying the expression of transcripts using RNA-Seq data</p>
</li>
        
        <li><p>Idenfity potential miRNAs involved in brassinosteroid-mediated regulation networks</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Quality Control</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Quality Control</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mapping</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mapping</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 2 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/4710649">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/workflows/" title="Whole transcriptome analysis of Arabidopsis thaliana workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            











<li class="btn btn-default supporting_material">


  <!-- dropdown with all recordings -->
  <a href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/recordings/" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Latest recordings of this material in the GTN Video Library">
        <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span>&nbsp;Recordings
  </a>


  <ul class="dropdown-menu">
    

    
      
      
      
    <li><a class="dropdown-item" href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/recordings/index.html#tutorial-recording-25-july-2021" title="View the recording for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> Tutorial (July 2021) - 1h10m</a>
    </li>
      
    
    <li><a class="dropdown-item" href="/training-material/topics/transcriptomics/tutorials/mirna-target-finder/recordings/" title="View all recordings for this tutorial">
                <i class="fas fa-video" aria-hidden="true"></i><span class="visually-hidden">video</span> View All</a>
    </li>

  </ul>

  
</li>
  





            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
    

    <!--
    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    -->
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Apr 8, 2021 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Aug 9, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00292">https://gxy.io/GTN:T00292</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">4.0</a> (0 recent ratings, 4 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 17 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>As sessile organisms, the survival of plants under adverse environmental conditions depends, to a large extent, on their ability to perceive stress stimuli and respond appropriately to counteract the potentially damaging effects. Coordination of phytohormones and reactive oxygen species are considered a key element for enhancing stress resistance, allowing fine-tuning of gene expression in response to environmental changes (<span class="citation"><a href="#PlanasRiverola2019">Planas-Riverola <i>et al.</i> 2019</a></span>, <span class="citation"><a href="#Ivashuta2011">Ivashuta <i>et al.</i> 2011</a></span>). These molecules constitute complex signalling networks, endowing with the ability to respond to a variable natural environment.</p>

<p>Brassinosteroids (BRs) are a group of plant steroid hormones essential for plant growth and development, as well as for controlling abiotic and biotic stress. Structurally, BRs are polyhydroxylated sterol derivatives with close similarity to animal hormones (Figure 1). This group of phytohormones is comprised of around 60 different compounds, of which brassinolide (BL), 24-epibrassinolide (EBR), and 28-homobrassinolide (HBR) are considered the most bioactive.</p>

<figure id="figure-1" style="max-width: 90%;"><img src="/training-material/topics/transcriptomics/images/transcriptomics_images/brassinosteroids.png" alt="fig1:Esteroid hormones. " width="712" height="252" loading="lazy" /><a target="_blank" href="/training-material/topics/transcriptomics/images/transcriptomics_images/brassinosteroids.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> Structure of various plant and animal steroid hormones.</figcaption></figure>

<p>Several recent studies suggest that the BR-mediated gene regulatory networks have the potential to reshape the future of agriculture, not only by alleviating the antagonistic effect diverse abiotic stress conditios, such as drought, but also by enhancing plant growth and yield. For instance, in tomato (<em>Solanum lycopersicum</em>), EBR treatment enhances drought tolerance, improving photosynthetic capacity, leaf water status, and antioxidant defense (<span class="citation"><a href="#Wang2018">Wang <i>et al.</i> 2018</a></span>). In pepper (<em>Capsicum annuum</em>), BL treatment increased the efficiency of light utilization under drought (<span class="citation"><a href="#Hu2013">Hu <i>et al.</i> 2013</a></span>). Gram (<em>Cicer arietinum</em>) plants exposed to drought stress and treated with BL showed significant increases in weight (<span class="citation"><a href="#Anwar2018">Anwar <i>et al.</i> 2018</a></span>). However, the mechanisms of BRs action in enhancing plant tolerance to abiotic stresses still remain largely unknown.</p>

<p>MicroRNAs (miRNAs), mainly 20–22 nucleotide small RNAs (sRNAs), are characterized for regulating gene expression at the post-transcriptional level. miRNAs are distinguished from other sRNAs by being generated from precursor harboring an imperfect stem‐loop structure. Unlike in animals, the pre-processing of plant miRNA occurs in the nucleus (Figure 2). The pre-miRNAs are then exported to the cytoplasm after methylation and incorporated into the Argonaute 1 protein to form RISC (RNA-induced silencing complex). The miRNA itself does not have the ability to cleave mRNAs or interfere with translation, but it plays a role in scanning the appropriate target.</p>

<figure id="figure-2" style="max-width: 90%;"><img src="../../images/transcriptomics_images/plant_mirna_synthesis.png" alt="fig2:miRNA biosynthesis. " width="917" height="916" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/plant_mirna_synthesis.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Plant miRNA biosynthesis, homeostasis and mechanisms of action (<span class="citation"><a href="#Wang2019">Wang <i>et al.</i> 2018</a></span>).</figcaption></figure>

<p>miRNAs have been found to be important regulators of many physiological processes, such as stress and hormonal responses. Four factors justify the miRNAs to be considered as master regulators of the plant response to the surrounding environment:</p>

<ul>
  <li>Multiple miRNA genes are regulated under given environmental conditions</li>
  <li>Computational predictions estimate that each miRNA regulates hundreds of genes</li>
  <li>The majority of plant miRNAs regulate genes encoding for transcription factors (TFs)</li>
  <li>Targets include not only mRNAs but also long noncoding RNAs (lncRNAs)</li>
</ul>

<p>In plants, miRNAs can silence targets through RNA degradation as well as translational repression pathways, and unlike animals, a large proportion of miRNA and their targets have less than four mismatches. This feature has been exploited for developing miRNAs target prediction tools, providing an efficient approach to elucidate the miRNA-mediated regulatory networks, which can contribute to biotechnological solutions to improve crops productivity.</p>

<p>In this tutorial, inspired by <span class="citation"><a href="#Park2020">Park <i>et al.</i> 2020</a></span>, we aim to explore the interplay between brassinosteroids and the miRNA-gene silencing pathway, considered one of the most versatile regulatory mechanisms in response to stressful situations in plants.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#experimental-design" id="markdown-toc-experimental-design">Experimental design</a></li>
  <li><a href="#background-on-data" id="markdown-toc-background-on-data">Background on data</a></li>
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#mirna-data-analysis" id="markdown-toc-mirna-data-analysis">miRNA data analysis</a>    <ol>
      <li><a href="#quality-assessment-of-mirna-reads" id="markdown-toc-quality-assessment-of-mirna-reads">Quality assessment of miRNA reads</a></li>
      <li><a href="#mirna-quantification-mirdeep2" id="markdown-toc-mirna-quantification-mirdeep2">miRNA quantification: <strong>MiRDeep2</strong></a></li>
      <li><a href="#differential-expression-analysis-of-mirnas-deseq2" id="markdown-toc-differential-expression-analysis-of-mirnas-deseq2">Differential expression analysis of miRNAs: <strong>DESeq2</strong></a></li>
      <li><a href="#filter-significantly-differentially-expressed-mirnas" id="markdown-toc-filter-significantly-differentially-expressed-mirnas">Filter significantly differentially expressed miRNAs</a></li>
    </ol>
  </li>
  <li><a href="#mrna-data-analysis" id="markdown-toc-mrna-data-analysis">mRNA data analysis</a>    <ol>
      <li><a href="#quality-assessment-of-mrna-reads" id="markdown-toc-quality-assessment-of-mrna-reads">Quality assessment of mRNA reads</a></li>
      <li><a href="#quantification-of-gene-expression-salmon" id="markdown-toc-quantification-of-gene-expression-salmon">Quantification of gene expression: <strong>Salmon</strong></a></li>
      <li><a href="#differential-expression-analysis-of-mrnas-deseq2" id="markdown-toc-differential-expression-analysis-of-mrnas-deseq2">Differential expression analysis of mRNAs: <strong>DESeq2</strong></a></li>
      <li><a href="#filter-significantly-differentially-expressed-mrnas" id="markdown-toc-filter-significantly-differentially-expressed-mrnas">Filter significantly differentially expressed mRNAs</a></li>
    </ol>
  </li>
  <li><a href="#identification-of-mirna-targets" id="markdown-toc-identification-of-mirna-targets">Identification of miRNA targets</a>    <ol>
      <li><a href="#mirna-target-prediction-using-targetfinder" id="markdown-toc-mirna-target-prediction-using-targetfinder">miRNA target prediction using <strong>TargetFinder</strong></a></li>
    </ol>
  </li>
  <li><a href="#optional-exercise" id="markdown-toc-optional-exercise">Optional exercise</a></li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="experimental-design">Experimental design</h1>

<p>The main objective of this training is to identify potential targets of miRNAs whose expression is induced by brassinosteroids. Our starting hypothesis is that there must be brassinosteroid-induced miRNAs that have high sequence complementarity with mRNAs whose expression is inhibited in the presence of these hormones (Figure 3).</p>

<figure id="figure-3" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_pipeline.png" alt="fig3:Experimental design. " width="1619" height="750" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_pipeline.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Experimental design</figcaption></figure>

<h1 id="background-on-data">Background on data</h1>

<p>The datasets to be used in this training can be classified into three groups: miRNA reads, mRNA reads and additional datasets.</p>

<h3 id="mirna-reads">miRNA reads</h3>

<p>The miRNA datasets consist of six FASTQ files, obtained by using the Illumina GAxII sequencing platform. The plant samples were obtained from wild-type Ws-2 seedlings treated with mock or 1 μM EBR for 90 min before harvest. The original datasets are available in the NCBI SRA database, with the accession number <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP258575">SRP258575</a>. As in the previous case, for this tutorial, we will use a reduced version of the data.</p>

<h3 id="mrna-reads">mRNA reads</h3>

<p>The mRNA datasets consist of four FASTQ files, generated through the Illumina HiSeq 2000 sequencing system. The samples were obtained from wild-type Columbia (Col-0) seedlings treated with mock or 100 nM BL for 4 hours. The original datasets are available in the NCBI SRA database, with the accession number <a href="https://www.ncbi.nlm.nih.gov/sra?term=SRP032274">SRP032274</a>. For this tutorial, subsets from the original data were generated in order to reduce the analysis run time.</p>

<h3 id="additional-datasets">Additional datasets</h3>

<p>In addition to the RNA-Seq reads obtained from the NCBI database, we will use datasets from two sources:</p>

<ul>
  <li><a href="https://ics.hutton.ac.uk/atRTD/">AtRTD2</a> is a high-quality transcript reference dataset developed to exploit the accuracy of transcript quantification tools such as <strong>Salmon</strong> and <strong>Kallisto</strong> in analyzing <em>Arabidopsis</em> RNA-Seq data.</li>
  <li><a href="http://pmiren.com/">PmiREN</a> is a comprehensive functional plant miRNA database that includes more than 20,000 annotated miRNAs diverse plant species.</li>
</ul>

<h1 id="get-data">Get data</h1>

<p>The first step of our analysis consists of retrieving the miRNA-Seq datasets from Zenodo and organizing them into collections.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-retrieve-mirna-seq-and-mrna-seq-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Retrieve miRNA-Seq and mRNA-Seq datasets</div>

  <ol>
    <li>Create a new history for this tutorial</li>
    <li>
      <p>Import the files from Zenodo:</p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li>Click on <strong>Rule-based</strong> tab</li>
        <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Collection(s)</code></li>
        <li>
          <p>Copy the following tabular data, paste it into the textbox and press <kbd>Build</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>SRR11611349	Control miRNA	https://zenodo.org/record/4710649/files/SRR11611349_MIRNASEQ_CTL.fastqsanger.gz	fastqsanger.gz
SRR11611350	Control miRNA	https://zenodo.org/record/4710649/files/SRR11611350_MIRNASEQ_CTL.fastqsanger.gz	fastqsanger.gz
SRR11611351	Control miRNA	https://zenodo.org/record/4710649/files/SRR11611351.MIRNASEQ_CTLfastqsanger.gz	fastqsanger.gz
SRR11611352	BR treated miRNA	https://zenodo.org/record/4710649/files/SRR11611352_MIRNASEQ_BL.fastqsanger.gz	fastqsanger.gz
SRR11611353	BR treated miRNA	https://zenodo.org/record/4710649/files/SRR11611353_MIRNASEQ_BL.fastqsanger.gz	fastqsanger.gz
SRR11611354	BR treated miRNA	https://zenodo.org/record/4710649/files/SRR11611354_MIRNASEQ_BL.fastqsanger.gz	fastqsanger.gz
SRR1019436	Control mRNA	https://zenodo.org/record/4710649/files/SRR1019436_RNASEQ_CTL.fastqsanger.gz	fastqsanger.gz
SRR1019437	Control mRNA	https://zenodo.org/record/4710649/files/SRR1019437_RNASEQ_CTL.fastqsanger.gz	fastqsanger.gz
SRR1019438	BR treated mRNA	https://zenodo.org/record/4710649/files/SRR1019438_RNASEQ_BL.fastqsanger.gz	fastqsanger.gz
SRR1019439	BR treated mRNA	https://zenodo.org/record/4710649/files/SRR1019439_RNASEQ_BL.fastqsanger.gz	fastqsanger.gz
</code></pre></div>          </div>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>
              <p>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">List Identifier(s)</code>: column <code class="language-plaintext highlighter-rouge">A</code></p>

              <blockquote class="tip">
                <div class="box-title tip-title" id="tip-can-t-find-i-list-identifier-i"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-t-find-i-list-identifier-i" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Can't find <i>List Identifier</i>?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
                <p>Then you’ve chosen to upload as a ‘dataset’ and not a ‘collection’. Close the upload menu, and restart the process, making sure you check <em>Upload data as</em>: <strong>Collection(s)</strong></p>
              </blockquote>
            </li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Collection Name</code>: column <code class="language-plaintext highlighter-rouge">B</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>: column <code class="language-plaintext highlighter-rouge">C</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Type</code>: column <code class="language-plaintext highlighter-rouge">D</code></li>
          </ul>
        </li>
        <li>Click <code class="language-plaintext highlighter-rouge">Apply</code> and press <kbd>Upload</kbd></li>
      </ul>
    </li>
  </ol>

  <!--SNIPPET-->
  <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
  <p><!--END_SNIPPET--></p>
</blockquote>

<p>Next we will retrieve the remaining datasets.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-retrieve-the-additional-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Retrieve the additional datasets</div>

  <ol>
    <li>
      <p>Import the files from Zenodo:</p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li><em>“Upload data as”</em>: <code class="language-plaintext highlighter-rouge">Datasets</code></li>
        <li>
          <p>Once again, copy the tabular data, paste it into the textbox and press <kbd>Build</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>annotation_AtRTD2.gtf	https://zenodo.org/record/4710649/files/annotation_AtRTD2_19April2016.gtf.gz
transcriptome.fasta	https://zenodo.org/record/4710649/files/transcriptome_AtRTD2_12April2016.fasta.gz
star_miRNA_seq.fasta	https://zenodo.org/record/4710649/files/star_miRNA_seq.fasta
mature_miRNA_AT.fasta	https://zenodo.org/record/4710649/files/mature_miRNA_AT.fasta
miRNA_stem-loop_seq.fasta	https://zenodo.org/record/4710649/files/miRNA_stem-loop_seq.fasta
</code></pre></div>          </div>
        </li>
        <li>From <strong>Rules</strong> menu select <code class="language-plaintext highlighter-rouge">Add / Modify Column Definitions</code>
          <ul>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">Name</code>: column <code class="language-plaintext highlighter-rouge">A</code></li>
            <li>Click <code class="language-plaintext highlighter-rouge">Add Definition</code> button and select <code class="language-plaintext highlighter-rouge">URL</code>: column <code class="language-plaintext highlighter-rouge">B</code></li>
          </ul>
        </li>
        <li>Click <code class="language-plaintext highlighter-rouge">Apply</code> and press <kbd>Upload</kbd></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-dataset-subsampling"><button class="gtn-boxify-button details" type="button" aria-controls="details-dataset-subsampling" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Dataset subsampling</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>As indicated above, for this tutorial the depth of the samples was reduced in order to speed up the time needed to carry out the analysis. This was done as follows:</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-dataset-subsampling"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Dataset subsampling</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/peterjc/sample_seqs/sample_seqs/0.2.5" title="Sub-sample sequences files  tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Sub-sample sequences files </strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.2.5)</span> with the following parameters:
        <ul>
          <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Multiple datasets”</em>: Each of the fastq files</li>
          <li><em>“Subsampling approach”</em>: <code class="language-plaintext highlighter-rouge">Take every N-th sequence (or pair e.g. every fifth sequence)</code></li>
          <li><em>“N”</em>: <code class="language-plaintext highlighter-rouge">100</code></li>
        </ul>
      </li>
    </ol>
  </blockquote>

  <p>In this way, we will only take 1% of reads at a random sampling rate.</p>
</blockquote>

<h1 id="mirna-data-analysis">miRNA data analysis</h1>

<p>Once we have imported the data, we can begin to study how brassinosteroid exposure alters gene expression patterns.</p>

<h2 id="quality-assessment-of-mirna-reads">Quality assessment of miRNA reads</h2>

<p>Due to technical limitations, sequencing is considered an error-prone process. In Illumina sequencing platforms, substitution type miscalls are the dominant source of errors, which can lead to inconsistent results. Another factor that can interfere with our analyses is the presence of adapter contaminants, which can result in an increased number of unaligned reads, since the adapter sequences are synthetic and do not occur in the genomic sequence.</p>

<p>Sequence quality control is therefore an essential first step in your analysis. We will use two popular tools for evaluating the quality of our raw reads: <strong>FastQC</strong> and <strong>MultiQC</strong>.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>In order to visualize the data from both collections together in the <strong>MultiQC</strong> tool, it will be necessary to combine the results generated by <strong>FastQC</strong>.
For more information on the topic of quality control, please see our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">dedicated training materials</a>.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-initial-quality-check"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Initial quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.72+galaxy1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Control miRNA</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC unprocessed control miRNA: RawData</code> and <code class="language-plaintext highlighter-rouge">FastQC unprocessed control miRNA: Webpage</code></li>
    <li>Repeat the previous steps with the <code class="language-plaintext highlighter-rouge">BR treated miRNA</code> collection.</li>
    <li><span class="tool" data-tool="__MERGE_COLLECTION__" title="Merge Collections tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge Collections</strong></span> with the following parameters:
      <ul>
        <li>In <em>“Input collections”</em>:
          <ul>
            <li><em>“1.Input Collections”</em>: <code class="language-plaintext highlighter-rouge">FastQC unprocessed control miRNA: RawData</code></li>
            <li><em>“2.Input Collections”</em>: <code class="language-plaintext highlighter-rouge">FastQC unprocessed BR treated mRiNA: RawData</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8+galaxy1)</span> with the following parameters:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: select the output generated in the previous step.</li>
          </ul>
        </li>
        <li>In <em>“Report title”</em>: <code class="language-plaintext highlighter-rouge">miRNA initial quality check</code></li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Based on the information provided by <strong>MultiQC</strong>, is it necessary to trimming/filtering the reads?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The report generated by <strong>MultiQC</strong> indicates that three quality parameters show values outside the recommended limits: per sequence G/C content, overrepresented sequences and adapter content.</p>

    <figure id="figure-4" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_GC_content.png" alt="fig4:GC content. " width="1200" height="800" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_GC_content.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Per sequence GC content of the miRNA samples.</figcaption></figure>
    <figure id="figure-5" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_overepresented.png" alt="fig5:Overrepresented sequences. " width="1200" height="800" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_overepresented.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Overrepresented sequences in miRNA samples.</figcaption></figure>
    <figure id="figure-6" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_adaptors.png" alt="fig6:Adaptor content. " width="1200" height="800" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_adaptors.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 6</strong>:</span> Adaptor content of miRNA samples.</figcaption></figure>

    <p>Particularly remarkable is the content of adapters, which are up to 80% of the reads in some samples. Given the abundance of adapters, one would expect that if we eliminate this source of contamination, the rest of the parameters would show a noticeable improvement.</p>

  </blockquote>
</blockquote>

<p>To remove the adapters contamination, we will employ the <strong>Trim Galore</strong> tool, a wrapper script around <a href="https://journal.embnet.org/index.php/embnetjournal/article/view/200"><strong>Cutadapt</strong></a> and <strong>FastQC</strong> which automates quality and adapter trimming.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-trimming-of-adapter-sequences"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Trimming of adapter sequences</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/trim_galore/trim_galore/0.4.3.1" title="Trim Galore! tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Trim Galore!</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.4.3.1)</span> with the following parameters:
      <ul>
        <li><em>“Is this library paired- or single-end?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
          <ul>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Reads in FASTQ format”</em>: <code class="language-plaintext highlighter-rouge">Control miRNA</code></li>
            <li><em>“Adapter sequence to be trimmed”</em>: <code class="language-plaintext highlighter-rouge">Illumina small RNA adapters</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the output collection as <code class="language-plaintext highlighter-rouge">Control miRNA trimmed</code></li>
    <li>Repeat the previous steps with the <code class="language-plaintext highlighter-rouge">BR treated miRNA</code> collection.</li>
  </ol>

</blockquote>

<p>Next, we will reassess the quality of the sequences to check if the adapters have been removed.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-post-processing-quality-check"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Post-processing quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.72+galaxy1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Control miRNA trimmed</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC processed control miRNA: RawData</code> and <code class="language-plaintext highlighter-rouge">FastQC processed control miRNA: Webpage</code></li>
    <li>Repeat the previous steps with the <code class="language-plaintext highlighter-rouge">BR treated miRNA trimmed</code> collection.</li>
    <li><span class="tool" data-tool="__MERGE_COLLECTION__" title="Merge Collections tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge Collections</strong></span> with the following parameters:
      <ul>
        <li>In <em>“Input collections”</em>:
          <ul>
            <li><em>“1.Input Collections”</em>: <code class="language-plaintext highlighter-rouge">FastQC processed control miRNA: RawData</code></li>
            <li><em>“2.Input Collections”</em>: <code class="language-plaintext highlighter-rouge">FastQC processed BR treated miRNA: RawData</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8+galaxy1)</span> with the following parameters:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: select the output generated in the previous step.</li>
          </ul>
        </li>
        <li>In <em>“Report title”</em>: <code class="language-plaintext highlighter-rouge">miRNA trimmed quality check</code></li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<p>The evaluation of the report generated by <strong>MultiQC</strong> after having processed the samples by <strong>Trim Galore</strong> indicates that the G/C content has been successfully corrected, and that the adapter contamination has been eliminated. However, the samples still show a high degree of over-represented sequences (Figure 7).</p>

<figure id="figure-7" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_overepresented_processed.png" alt="fig7:Overexpressed sequences. " width="1200" height="800" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_overepresented_processed.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 7</strong>:</span> Report of overexpressed sequences in the miRNA reads</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What do you think could be the cause of the high number of over-represented sequences?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Two of the factors that may determine the abundance of overrepresented sequences are the existence of highly expressed miRNAs (Seco-Cervera et al. 2018), the existence of highly conserved sequence motifs within the miRNA (<span class="citation"><a href="#Glazov2008">Glazov <i>et al.</i> 2008</a></span>), and contamination with foreign sequences.</p>

  </blockquote>
</blockquote>

<h2 id="mirna-quantification-mirdeep2">miRNA quantification: <strong>MiRDeep2</strong></h2>

<p>Quantification of miRNAs requires to use two different tools:</p>
<ul>
  <li>The <strong>MiRDeep2 Mapper</strong> tool for preprocessing the reads.</li>
  <li>The <strong>MiRDeep2 Quantifier</strong>  tool for mapping the deep sequencing reads to predefined miRNA precursors and determining the expression of the corresponding miRNAs. It is carried out in two steps: firstly, the predefined mature miRNA sequences are mapped to the predefined precursors (optionally, predefined star sequences can be mapped to the precursors too). And second, the deep sequencing reads are mapped to the precursors.</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quantification-of-mirnas"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quantification of miRNAs</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/mirdeep2_mapper/rbc_mirdeep2_mapper/2.0.0" title="MiRDeep2 Mapper tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MiRDeep2 Mapper</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Deep sequencing reads”</em>: <code class="language-plaintext highlighter-rouge">Control miRNA trimmed</code></li>
        <li><em>“Remove reads with non-standard nucleotides”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Clip 3’ Adapter Sequence”</em>: <code class="language-plaintext highlighter-rouge">Don't Clip</code></li>
        <li><em>“Collapse reads and/or Map”</em>: <code class="language-plaintext highlighter-rouge">Collapse</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Collapsed control miRNA</code></li>
    <li>
      <p>Repeat the previous stages by providing <code class="language-plaintext highlighter-rouge">BR treated miRNA trimmed</code>as input, and rename it as <code class="language-plaintext highlighter-rouge">Collapsed BR treated miRNA</code>.</p>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/mirdeep2_quantifier/rbc_mirdeep2_quantifier/2.0.0" title="MiRDeep2 Quantifier tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MiRDeep2 Quantifier</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.0)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Collapsed deep sequencing reads”</em>: <code class="language-plaintext highlighter-rouge">Collapsed control miRNA</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Precursor sequences”</em>: <code class="language-plaintext highlighter-rouge">miRNA_stem-loop_seq.fasta</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Mature miRNA sequences”</em>: <code class="language-plaintext highlighter-rouge">mature_miRNA_AT.fasta</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Star sequences”</em>: <code class="language-plaintext highlighter-rouge">star_miRNA_seq.fasta</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">MiRDeep2 control miRNA</code> and <code class="language-plaintext highlighter-rouge">MiRDeep2 control miRNA (html)</code>.</li>
    <li>Repeat the fourth step by providing <code class="language-plaintext highlighter-rouge">Collapsed BR treated miRNA</code> as input, and rename the outputs as <code class="language-plaintext highlighter-rouge">MiRDeep2 BR treated miRNA</code> and <code class="language-plaintext highlighter-rouge">MiRDeep2 BR treated miRNA (html)</code></li>
  </ol>

</blockquote>

<p>To use the outputs generated by <strong>MiRDeep2 Quantifier</strong> in the differential expression analysis, it is necessary to modify the datasets.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-edition-of-mirdeep2-quantifier-outputs"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Edition of MiRDeep2 Quantifier outputs</div>

  <ol>
    <li><span class="tool" data-tool="Cut1" title="Cut columns from a table tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut columns from a table</strong></span> with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c2</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">MiRDeep2 control miRNA</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">control miRNA counts</code></li>
    <li><span class="tool" data-tool="Cut1" title="Cut columns from a table tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut columns from a table</strong></span>  with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c2</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">MiRDeep2 BR treated miRNA</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">BR treated miRNA counts</code></li>
  </ol>

</blockquote>

<h2 id="differential-expression-analysis-of-mirnas-deseq2">Differential expression analysis of miRNAs: <strong>DESeq2</strong></h2>

<p><strong>DESeq2</strong> is a tool for differential gene expression analysis based on a negative binomial generalized linear model. <strong>DESeq2</strong> internally corrects the differences in library size, due to which no preliminary normalization of input datasets is required.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>It is desirable to use at least three replicates of each experimental condition to ensure sufficient statistical power.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-differential-expression-analysis-using-deseq2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Differential expression analysis using DESeq2</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.6+galaxy1" title="DESeq2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>DESeq2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.11.40.6+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“How”</em>: <code class="language-plaintext highlighter-rouge">Select datasets per level</code>
          <ul>
            <li>In <em>“Factor”</em>:
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor”</em>
                  <ul>
                    <li><em>“Specify a factor name, e.g. effects_drug_x or cancer_markers”</em>: <code class="language-plaintext highlighter-rouge">effects_brassinolide</code></li>
                    <li>In <em>“Factor level”</em>:
                      <ul>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor level”</em>
                          <ul>
                            <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">brassinolide</code></li>
                            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: <code class="language-plaintext highlighter-rouge">BR treated miRNA counts</code></li>
                          </ul>
                        </li>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor level”</em>
                          <ul>
                            <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">control</code></li>
                            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: <code class="language-plaintext highlighter-rouge">control miRNA counts</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Files have header?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Choice of Input data”</em>: <code class="language-plaintext highlighter-rouge">Count data (e.g. from HTSeq-count, featureCounts or StringTie)</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">DESeq2 results miRNA</code> and <code class="language-plaintext highlighter-rouge">DESeq2 plots miRNA</code></li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the <code class="language-plaintext highlighter-rouge">DESeq2 plots miRNA</code> file</li>
  </ol>
</blockquote>

<p>DESeq2 generated 2 outputs: a table with the normalized counts and a graphical summary of the results. To evaluate the similarity of our samples, we are going to inspect the Principal Component Analysis (PCA) plot. PCA allows evaluating the dominant directions of the highest variability in the data. Thus, the samples subjected to the same conditions should cluster together.</p>
<figure id="figure-8" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_PCA.png" alt="fig8:PCA miRNA. " width="1106" height="522" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_PCA.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 8</strong>:</span>  PCA plot of expression data from control and BR treated miRNA samples.</figcaption></figure>

<p>As can be seen, the main axes account for only 47% and 19% of the total variation. This suggests that the effect of brassinosteroids on miRNA regulation is limited (Figure 8).</p>

<h2 id="filter-significantly-differentially-expressed-mirnas">Filter significantly differentially expressed miRNAs</h2>

<p>Next, we will extract those genes whose expression is statistically significantly differentially expressed (DE) due to BR treatment by selecting those whose adjusted p-value is less than or 0.05. A cut-off value of 0.05 indicates that the probability of a false positive result is less than 5%.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-p-value-vs-p-adj"><button class="gtn-boxify-button details" type="button" aria-controls="details-p-value-vs-p-adj" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: p-value vs p-adj</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>The p-value is a measure of the probability that an observed difference could have occurred just by random chance. A small p-value indicates that there is a small chance of getting this data if no real difference existed. A p-value threshold of 0.05 indicates that there is a 5% chance that the result is a false positive. The p-adj (also known as q-value) is an adjusted p-value which taking in to account the false discovery rate (FDR). Applying a FDR becomes necessary when we’re measuring thousands of variables from a small sample set.</p>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-differentially-expressed-mirnas"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter differentially expressed miRNAs</div>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> data on any column using simple expressions (Galaxy Version 1.1.1) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">DESeq2 results miRNA</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>How many miRNAs show statistically significant differential expression?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Unfortunately, we have not detected any differentially expressed miRNAs. In this case, this is caused by the fact that the subsampled datasets don’t have sufficient read depth to test for differential expression.</p>
  </blockquote>

</blockquote>

<p>To get any sensible results, it is worth analyzing the full dataset. You can analyze the full datasets following the above tutorial. Otherwise, you can import the DESeq2 analysis results that we generated from full dataset into your history.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-retrieve-the-deseq2-analysis-results-on-full-mirna-dataset"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Retrieve the DESeq2 analysis results on full miRNA dataset</div>

  <ol>
    <li>
      <p>Import the files from Zenodo:</p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li>Click of the <strong>Paste/Fetch</strong> button</li>
        <li>
          <p>Copy the Zenodo links and press <kbd>Start</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4663389/files/miRNA_DESeq2_results_complete_dataset.tabular
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>Rename each dataset according to the sample id (e.g. <code class="language-plaintext highlighter-rouge">miRNA_DESeq2_results_complete_dataset</code>)</li>
    <li>Add all miRNA data analysis related tags: #miRNA #BR #control</li>
  </ol>
</blockquote>

<p>Repeat the filtering step on the imported DESeq2 result.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filter-and-sort-differentially-expressed-mirnas-from-the-full-dataset"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filter and sort differentially expressed miRNAs from the full dataset</div>

  <ol class="comment">
    <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> data on any column using simple expressions (Galaxy Version 1.1.1) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">miRNA_DESeq2_results_complete_dataset</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Differentially expressed miRNAs</code></li>
    <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> data on any column using simple expressions (Galaxy Version 1.1.1) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Differentially expressed miRNAs</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c3&gt;0</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Upregulated miRNAs</code></li>
    <li><span class="tool" data-tool="sort1" title="Sort tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Sort</strong></span> data in ascending or descending order (Galaxy Version 1.1.0) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sort Dataset”</em>: <code class="language-plaintext highlighter-rouge">Upregulated miRNAs</code></li>
        <li><em>“on column”</em>: <code class="language-plaintext highlighter-rouge">Column: 3</code></li>
        <li><em>“everything in”</em>: <code class="language-plaintext highlighter-rouge">Descending order</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-3"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many miRNAs are differentially expressed?</li>
    <li>How many miRNAs show statistically significant upregulation and what is the most upregulated miRNA?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-3"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-3" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>Out of 442 miRNAs, 39 show significant differential expression.</li>
      <li>There are 16 upregulated miRNAs and <code class="language-plaintext highlighter-rouge">Ath-miR156g</code> is the most upregulated one.</li>
    </ol>
  </blockquote>

</blockquote>

<h1 id="mrna-data-analysis">mRNA data analysis</h1>

<p>Once the differential expression analysis of miRNAs has been carried out, the next stage is to analyze how mRNA expression is altered in response to brassinosteroids.</p>

<h2 id="quality-assessment-of-mrna-reads">Quality assessment of mRNA reads</h2>

<p>As in the previous section, we shall begin by assessing the quality of our sequences.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-initial-quality-check-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Initial quality check</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.72+galaxy1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: <code class="language-plaintext highlighter-rouge">Control mRNA</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">FastQC unprocessed control mRNA: RawData</code> and <code class="language-plaintext highlighter-rouge">FastQC unprocessed control mRNA: Webpage</code></li>
    <li>Repeat the previous step with the <code class="language-plaintext highlighter-rouge">BR treated mRNA</code> collection.</li>
    <li><span class="tool" data-tool="__MERGE_COLLECTION__" title="Merge Collections tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Merge Collections</strong></span> with the following parameters:
      <ul>
        <li>In <em>“Input collections”</em>:
          <ul>
            <li><em>“1.Input Collections”</em>: <code class="language-plaintext highlighter-rouge">FastQC unprocessed control mRNA: RawData</code></li>
            <li><em>“2.Input Collections”</em>: <code class="language-plaintext highlighter-rouge">FastQC unprocessed BR treated mRNA: RawData</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy1" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8+galaxy1)</span> with the following parameters:
      <ul>
        <li>In <em>“Results”</em>:
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Dataset collection”</em>: select the output generated in the previous step.</li>
          </ul>
        </li>
        <li><em>“Report title”</em>: <code class="language-plaintext highlighter-rouge">mRNA initial quality check</code></li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the generated HTML file</li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-4"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Are there any stats that indicate the need to process the samples in order to improve their quality?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-4"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-4" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>All the stats are within acceptable limits. However, the adapter content shows the presence of Illumina universal adapters in our reads, which can be removed to avoid possible interferences at later stages (Figure 10).</p>

    <figure id="figure-9" style="max-width: 90%;"><img src="../../images/transcriptomics_images/miRNA_adapter_01.png" alt="fig10:Adapter content of mRNA samples. " width="994" height="500" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/miRNA_adapter_01.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 9</strong>:</span> Quality assessment of mRNA samples</figcaption></figure>

  </blockquote>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>
  <p>Although our samples have adapters, we are not going to trim them in this case. We will explain the reason in the next section.</p>
</blockquote>

<h2 id="quantification-of-gene-expression-salmon">Quantification of gene expression: <strong>Salmon</strong></h2>

<p>After performing the quality assessment of the reads, we can move on to quantifying the gene expression. The aim of this step is to identify which transcript each read comes from and the total number of reads associated with each transcript. In this tutorial, we will use the Salmon tool (<span class="citation"><a href="#Patro2017">Patro <i>et al.</i> 2017</a></span>) for the quantification of mRNA transcripts.</p>

<p><a href="../../images/transcriptomics_images/salmon_logo.png" rel="noopener noreferrer"><img src="../../images/transcriptomics_images/salmon_logo.png" alt="fig11:Salmon logo. " width="1513" height="250" loading="lazy" /></a></p>

<p>One of the characteristics of <strong>Salmon</strong> is that it doesn’t require performing a base-to-base alignment, which is the time-consuming step of tools such as <strong>STAR</strong> and <strong>HISAT2</strong>. Salmon relies on the quasi-mapping concept, a new mapping technique that allows the rapid and accurate mapping of RNA-Seq reads to a target transcriptome. Rather than a standard alignment, quasi-mapping seeks to find the best mappings for each read, and does so by finding minimal collections of dynamically sized, right-maximal, matching contexts between target and query positions (<span class="citation"><a href="#Srivastava2016">Srivastava <i>et al.</i> 2016</a></span>)</p>

<p>The quasi-mapping approach by <strong>Salmon</strong> requires a reference index to determine the position and orientation information for accurate mapping prior to quantification. It allows providing the transcriptome in a format that optimizes its use in transcript identification and quantification.</p>

<p>After determining the best mapping for each read, <strong>Salmon</strong>  generates the final transcript abundance estimation after modeling sample-specific parameters and biases. Reads that map equally well to more than one transcript will have the count divided between all of the mappings, thus avoiding the loss of information on the different gene isoforms.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-quasi-mapping-algorithm"><button class="gtn-boxify-button details" type="button" aria-controls="details-quasi-mapping-algorithm" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Quasi-mapping algorithm</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>The quasi-mapping algorithm makes use of two main data structures, the generalized suffix array (SA) of the transcriptome T, and a hash table (h) that maps each k-mer in T to its SA interval (by default k = 31). During the quasi-mapping procedure, a read is scanned from left to right until a k-mer (k<sub>i</sub>, starting at position i on the read) is encountered that appears in h. The k<sub>i</sub> is looked up in the hash table and the SA intervals are retrieved, giving all suffixes containing the k-mer k<sub>i</sub>. Then, the Maximal Mappable Prefix (MMP<sub>i</sub>) is computed by finding the longest substring of the read that matches the reference suffixes. Owing to sequencing errors, the MMPs may not span the complete read. In this case, the next informative position (NIP) is determined based on the longest common prefix of the SA intervals of MMP<sub>i</sub>. Suffix array search continues from k-bases before NIP until the final NIP is less than k-bases before the read end. Finally, for each read, the algorithm reports the transcripts it mapped to, location and strand information (<span class="citation"><a href="#Srivastava2016">Srivastava <i>et al.</i> 2016</a></span>).</p>
  <figure id="figure-10" style="max-width: 90%;"><img src="../../images/transcriptomics_images/salmon_quasialignment.png" alt="fig12:Quasi-mapping algorithm. " width="1542" height="852" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/salmon_quasialignment.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 10</strong>:</span> Illustration of the quasi-mapping of a read using k=3. Hash table lookup of k<sub>i</sub> returns the suffix array interval [b, e). The base 'G' at position 6 in the read is a result of sequencing error. Hence, the MMP<sub>i</sub> is 'ATTGA' and the SA interval of MMP<sub>i</sub> is [b', e'). The next k-mer starts at k-bases prior to the NIP which is base after the longest common prefix of the interval [b',e'). In the end, the read in the above example most likely map to the suffix array at [e'-1, e').</figcaption></figure>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quantify-gene-expression-with-salmon"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quantify gene expression with Salmon</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/salmon/salmon/0.14.1.2+galaxy1" title="Salmon quant tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Salmon quant</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.14.1.2+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“Select salmon quantification mode:”</em>: <code class="language-plaintext highlighter-rouge">Reads</code>
          <ul>
            <li><em>“Select a reference transcriptome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use one from the history</code>
              <ul>
                <li>In <em>“Salmon index”</em>:
                  <ul>
                    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Transcripts fasta file”</em>: <code class="language-plaintext highlighter-rouge">transcriptome.fasta</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li>In <em>“Data input”</em>:
              <ul>
                <li><em>“Is this library mate-paired?”</em>: <code class="language-plaintext highlighter-rouge">Single-end</code>
                  <ul>
                    <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“FASTQ/FASTA file”</em>: <code class="language-plaintext highlighter-rouge">Control mRNA</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>“Validate mappings”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File containing a mapping of transcripts to genes”</em>: <code class="language-plaintext highlighter-rouge">annotation_AtRTD2.gtf</code></li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">Salmon control mRNA (Quantification)</code> and <code class="language-plaintext highlighter-rouge">Salmon control mRNA (Gene Quantification)</code></li>
    <li>Repeat the previous procedure by using the <code class="language-plaintext highlighter-rouge">BR treated mRNA</code> dataset</li>
  </ol>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-quasi-mapping-sequence-requirements"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Quasi-mapping sequence requirements</div>
    <p>Trimming the reads is not required  when using this method, since if there are k-mers in the reads that are not in the hash table, they are not counted. Quantification of the reads is only as good as the quality of the reference transcriptome.</p>
  </blockquote>

</blockquote>

<p><strong>Salmon</strong> generates two outputs for each input sample:</p>

<ul>
  <li>Quantification: summarizes the quantifications by transcript</li>
  <li>Gene quantification: summarizes the quantification by gene</li>
</ul>

<p>Each output consists of a tabular dataset with five columns:</p>

<table>
  <thead>
    <tr>
      <th>Field</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Name</td>
      <td>The name of the target transcript provided in the input transcriptome.</td>
    </tr>
    <tr>
      <td>Length</td>
      <td>The length of the target transcript.</td>
    </tr>
    <tr>
      <td>EffectiveLength</td>
      <td>The computed effective length.</td>
    </tr>
    <tr>
      <td>TPM</td>
      <td>The relative abundance of this transcript in units of Transcripts Per Million.</td>
    </tr>
    <tr>
      <td>NumReads</td>
      <td>The number of reads mapping to each transcript that was quantified.</td>
    </tr>
  </tbody>
</table>

<blockquote class="question">
  <div class="box-title question-title" id="question-5"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>Could you explain why we did not trim the reads before?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-5"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-5" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>The reason is that, since the k-mers which contain the adaptor sequences are not present in the transcriptome (from which the hash table is generated), they are omitted.</p>

  </blockquote>

</blockquote>

<h2 id="differential-expression-analysis-of-mrnas-deseq2">Differential expression analysis of mRNAs: <strong>DESeq2</strong></h2>

<p>Now, let’s analyze which genes show statistically significant differential expression in response to brassinosteoids.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-differential-expression-analysis-using-deseq2-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Differential expression analysis using DEseq2</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/deseq2/deseq2/2.11.40.6+galaxy1" title="DESeq2 tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>DESeq2</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.11.40.6+galaxy1)</span> with the following parameters:
      <ul>
        <li><em>“How”</em>: <code class="language-plaintext highlighter-rouge">Select datasets per level</code>
          <ul>
            <li>In <em>“Factor”</em>:
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor”</em>
                  <ul>
                    <li><em>“Specify a factor name, e.g. effects_drug_x or cancer_markers”</em>: <code class="language-plaintext highlighter-rouge">effects_brassinolide</code></li>
                    <li>In <em>“Factor level”</em>:
                      <ul>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor level”</em>
                          <ul>
                            <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">brassinolide</code></li>
                            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: <code class="language-plaintext highlighter-rouge">Salmon BR treated mRNA (Gene Quantification)</code></li>
                          </ul>
                        </li>
                        <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Factor level”</em>
                          <ul>
                            <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">control</code></li>
                            <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Counts file(s)”</em>: <code class="language-plaintext highlighter-rouge">Salmon control mRNA (Gene Quantification)</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Choice of Input data”</em>: <code class="language-plaintext highlighter-rouge">TPM values (e.g. from kallisto, sailfish or salmon)</code>
          <ul>
            <li><em>“Program used to generate TPMs”</em>: <code class="language-plaintext highlighter-rouge">Salmon</code></li>
            <li><em>“Gene mapping format”</em>: <code class="language-plaintext highlighter-rouge">GTF/GFF3</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GTF/GFF3 annotation file”</em>: <code class="language-plaintext highlighter-rouge">annotation_AtRTD2.gtf</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename the outputs as <code class="language-plaintext highlighter-rouge">DESeq2 results mRNA</code> and <code class="language-plaintext highlighter-rouge">DESeq2 plots mRNA</code></li>
  </ol>
</blockquote>

<figure id="figure-11" style="max-width: 90%;"><img src="../../images/transcriptomics_images/mRNA_PCA.png" alt="fig13:PCA plot of mRNA samples. " width="1159" height="532" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/mRNA_PCA.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 11</strong>:</span>  PCA plot of differential expression data from control and BR treated samples.</figcaption></figure>

<blockquote class="question">
  <div class="box-title question-title" id="question-6"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What conclusions about the similarity of the samples can be derived from the PCA plot?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-6"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-6" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>From the information provided by the plot, it is possible to state that there is a high similarity between the samples belonging to the same experimental conditions, since the first dimension (x-axis) allows to explain 81% of the variability, and the samples are located at opposite ends of the x-axis.</p>

  </blockquote>

</blockquote>

<h2 id="filter-significantly-differentially-expressed-mrnas">Filter significantly differentially expressed mRNAs</h2>

<p>To conclude the analysis of the differential expression of mRNAs, we will extract those transcripts that show a significant induction of expression in response to brassinosteroids. Before continuing with further analysis, similar to miRNA data analysis, import the DESeq2 results generated from full mRNA datasets.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-retrieve-the-deseq2-analysis-results-on-full-mrna-dataset"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Retrieve the DESeq2 analysis results on full mRNA dataset</div>

  <ol>
    <li>
      <p>Import the files from Zenodo:</p>

      <ul>
        <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>
        <li>Click of the <strong>Paste/Fetch</strong> button</li>
        <li>
          <p>Copy the Zenodo links and press <kbd>Start</kbd></p>

          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4663389/files/mRNA_DESeq2_results_complete_dataset.tabular
</code></pre></div>          </div>
        </li>
      </ul>
    </li>
    <li>Rename each dataset according to the sample id (e.g. <code class="language-plaintext highlighter-rouge">mRNA_DESeq2_results_complete_dataset</code>)</li>
    <li>Add all mRNA data analysis related tags: #mRNA #BR #control</li>
  </ol>
</blockquote>

<p>Now we continue with the DE genes analysis.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-extract-significantly-de-genes"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Extract significantly DE genes</div>

  <ol>
    <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> data on any column using simple expressions (Galaxy Version 1.1.1) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">mRNA_DESeq2_results_complete_dataset</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c7&lt;0.05</code></li>
      </ul>
    </li>
    <li>
      <p>Rename the output as <code class="language-plaintext highlighter-rouge">Differentially expressed mRNAs</code></p>
    </li>
    <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span>data on any column using simple expressions (Galaxy Version 1.1.1) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Differentially expressed mRNAs</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c3&gt;1</code></li>
      </ul>
    </li>
    <li>
      <p>Rename it as <code class="language-plaintext highlighter-rouge">Upregulated mRNAs</code></p>
    </li>
    <li><span class="tool" data-tool="Filter1" title="Filter tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter</strong></span> data on any column using simple expressions (Galaxy Version 1.1.1) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Filter”</em>: <code class="language-plaintext highlighter-rouge">Differentially expressed mRNAs</code></li>
        <li><em>“With following condition”</em>: <code class="language-plaintext highlighter-rouge">c3&lt;-1</code></li>
      </ul>
    </li>
    <li>Rename it as <code class="language-plaintext highlighter-rouge">Downregulated mRNAs</code></li>
  </ol>

</blockquote>

<blockquote class="question">
  <div class="box-title question-title" id="question-7"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <ol>
    <li>How many genes show statistically significant differential expression?</li>
    <li>How many of them are significantly upregulated (at least two-fold change)? And downregulated?</li>
    <li>What is the most significantly DE downregulated gene and what is it biological function?</li>
  </ol>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-7"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-7" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <ol>
      <li>The total number of genes whose expression differential between the two experimental conditions is 4176.</li>
      <li>Of them, 328 are significantly downregulated by the BR treatment and 778 are upregulated.</li>
      <li>The most significantly DE gene is AT3G30180 (BR60X2). BR60X2 encodes a cytochrome p450 enzyme that catalyzes the last reaction in the production of brassinolide. It is capable of converting 6-deoxocastasterone into castasterone, a C-6 oxidation, as well as the further conversion of castasterone into brassinolide (source: <a href="https://www.arabidopsis.org/servlets/TairObject?name=at3g30180&amp;type=locus">TAIR database</a>).</li>
    </ol>
  </blockquote>

</blockquote>

<h1 id="identification-of-mirna-targets">Identification of miRNA targets</h1>

<p>To predict which miRNAs target which mRNAs, first we need their transcriptomic sequences in FASTA format. Now we will obtain the sequences of miRNAs whose expression is induced by brassinosteroids.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-obtaining-the-sequences-of-upregulated-mirnas"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Obtaining the sequences of upregulated miRNAs</div>
  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment</div>

    <p>The following tools can be found in <em>Text Manipulation</em> and <em>Filter and Sort</em> sections.</p>

  </blockquote>

  <ol>
    <li><span class="tool" data-tool="Cut1" title="Cut columns from a table tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut columns from a table</strong></span> with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">Upregulated miRNAs</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Upregulated miRNA ids</code></li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/filter_by_fasta_ids/filter_by_fasta_ids/2.1" title="Filter FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA sequences”</em>: <code class="language-plaintext highlighter-rouge">mature_miRNA_AT.fasta</code></li>
        <li><em>“Criteria for filtering on the headers”</em>: <code class="language-plaintext highlighter-rouge">List of IDs</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“List of IDs to extract sequences for”</em>: <code class="language-plaintext highlighter-rouge">Upregulated miRNA ids</code></li>
            <li><em>“Match IDs by”</em>: <code class="language-plaintext highlighter-rouge">Default: ID is expected at the beginning: &gt;ID</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/filter_by_fasta_ids/filter_by_fasta_ids/2.1" title="Filter FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA sequences”</em>: <code class="language-plaintext highlighter-rouge">star_miRNA_seq.fasta</code></li>
        <li><em>“Criteria for filtering on the headers”</em>: <code class="language-plaintext highlighter-rouge">List of IDs</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“List of IDs to extract sequences for”</em>: <code class="language-plaintext highlighter-rouge">Upregulated miRNA ids</code></li>
            <li><em>“Match IDs by”</em>: <code class="language-plaintext highlighter-rouge">Default: ID is expected at the beginning: &gt;ID</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><span class="tool" data-tool="cat1" title="Concatenate datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets</strong></span> tail-to-head (Galaxy Version 1.0.0) with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Concatenate Dataset”</em>: output of <strong>Filter FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on <code class="language-plaintext highlighter-rouge">mature_miRNA_AT.fasta</code></li>
        <li>
          <ul>
            <li>Insert Dataset
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select”</em>: output of <strong>Filter FASTA</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on <code class="language-plaintext highlighter-rouge">star_miRNA_seq.fasta</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename it as <code class="language-plaintext highlighter-rouge">Upregulated miRNA sequences</code></li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the <code class="language-plaintext highlighter-rouge">Upregulated miRNA sequences</code> file</li>
  </ol>
</blockquote>

<p>To identify putative targets of upregulated miRNAs, it is necessary to obtain the sequences of all downregulated mRNAs in FASTA format.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-obtaining-the-gene-sequences-of-downregulated-mrnas"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Obtaining the gene sequences of downregulated mRNAs</div>
  <ol>
    <li><span class="tool" data-tool="Cut1" title="Cut columns from a table tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Cut columns from a table</strong></span> with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“From”</em>: <code class="language-plaintext highlighter-rouge">Downregulated mRNAs</code></li>
      </ul>
    </li>
    <li>Rename the output as <code class="language-plaintext highlighter-rouge">Downregulated mRNA ids</code></li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/filter_by_fasta_ids/filter_by_fasta_ids/2.1" title="Filter FASTA tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter FASTA</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“FASTA sequences”</em>: <code class="language-plaintext highlighter-rouge">transcriptome.fasta</code> (Input dataset)</li>
        <li><em>“Criteria for filtering on the headers”</em>: <code class="language-plaintext highlighter-rouge">List of IDs</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> *“List of IDs to extract sequences for: <code class="language-plaintext highlighter-rouge">Downregulated mRNA ids</code></li>
            <li><em>“Match IDs by”</em>: <code class="language-plaintext highlighter-rouge">Custom regex pattern</code>
              <ul>
                <li><em>“Regex search pattern for ID”</em>: <code class="language-plaintext highlighter-rouge">GENE=(AT.{7})</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Rename it as <code class="language-plaintext highlighter-rouge">Downregulated mRNA sequences</code></li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the <code class="language-plaintext highlighter-rouge">Downregulated mRNA sequences</code>  file</li>
  </ol>
</blockquote>

<h2 id="mirna-target-prediction-using-targetfinder">miRNA target prediction using <strong>TargetFinder</strong></h2>

<p>We are now ready to launch the search for miRNA target genes. For this we will use the <strong>TargetFinder</strong> tool that is implemented and used for miRNA target prediction in plants (<span class="citation"><a href="#Srivastava2014">Srivastava <i>et al.</i> 2014</a></span>, <span class="citation"><a href="#Ma2017">Ma <i>et al.</i> 2017</a></span>).</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-identification-of-potential-mirna-targets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: identification of potential miRNA targets</div>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/rnateam/targetfinder/targetfinder/1.7.0+galaxy1" title="TargetFinder tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>TargetFinder</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.7.0+galaxy1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input small RNA sequences file”</em>: <code class="language-plaintext highlighter-rouge">Upreguled miRNA sequences</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Target sequence database file</em>”: <code class="language-plaintext highlighter-rouge">Downregulated mRNA sequences</code></li>
        <li><em>“Prediction score cutoff value”</em>: <code class="language-plaintext highlighter-rouge">5.0</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">Tab-delimited format</code></li>
      </ul>
    </li>
    <li>Click on the <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> (eye) icon and inspect the output of <strong>TargetFinder</strong>.</li>
  </ol>
</blockquote>

<p>Congratulations! We have identified the following 5 potential genes involved in the brassinosteroid-miRNA regulatory network.</p>

<figure id="figure-12" style="max-width: 90%;"><img src="../../images/transcriptomics_images/targetFinder_results.png" alt="fig15:TargetFinder results. " width="1173" height="564" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/targetFinder_results.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 12</strong>:</span> TargetFinder results.</figcaption></figure>

<p>Finally, we can access all the information available on the genes identified in the TAIR database:</p>

<ul>
  <li><a href="https://www.arabidopsis.org/servlets/TairObject?type=locus&amp;name=AT5G10180">AT5G10180</a>: ARABIDOPSIS SULFATE TRANSPORTER 68, AST6</li>
  <li><a href="https://www.arabidopsis.org/servlets/TairObject?type=locus&amp;name=AT3G09220">AT3G09220</a>: LAC7, LACCASE 7</li>
  <li><a href="https://www.arabidopsis.org/servlets/TairObject?type=locus&amp;name=AT2G46850">AT2G46850</a>: Protein kinase superfamily protein</li>
  <li><a href="https://www.arabidopsis.org/servlets/TairObject?type=locus&amp;name=AT5G64260">AT5G64260</a>: EXL2, EXORDIUM LIKE 2</li>
  <li><a href="https://www.arabidopsis.org/servlets/TairObject?type=locus&amp;name=AT3G63200">AT3G63200</a>: PATATIN-LIKE PROTEIN 9, PLA IIIB</li>
</ul>

<p>Both AT4G14365 and AT1G26890 are not well characterized genes. In the case of AT5G50570, experimental research have demonstrated that this gene is involved in  flooding tolerance in <em>Medicago sativa</em>, through a signaling path mediated by miR156 (<span class="citation"><a href="#Feyissa2021">Feyissa <i>et al.</i> 2021</a></span>). On the other hand, according to <span class="citation"><a href="#Gao2017">Gao <i>et al.</i> 2017</a></span>, SPL13 is likely a negative regulator of plant vegetative growth. The interaction between miR156 and SPL transcription factors has been suggested for diverse Poaceae family members (<span class="citation"><a href="#Yue2021">Yue <i>et al.</i> 2021</a></span>).</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-hypothesis-testing"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Hypothesis testing</div>
  <p>One of the hypotheses that we could propose from our results is: the inhibition of the AT2G46850 gene can result in plants with improved resistance to drought conditions. Is it possible to validate it? Yes! We propose this approach: to acquire <a href="https://abrc.osu.edu/stocks/392113">AT2G46850 mutant seeds</a> and <a href="http://arabidopsis.info/StockInfo?NASC_id=N1093">wild type seeds</a>, grow them under two controlled conditions: watered and drought stress, and analyze plant weight after 33 days (Figure 13).</p>

  <figure id="figure-13" style="max-width: 90%;"><img src="../../images/transcriptomics_images/arabidopis_growth_protocol.png" alt="fig16:Plant growth. " width="1565" height="2481" loading="lazy" /><a target="_blank" href="../../images/transcriptomics_images/arabidopis_growth_protocol.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 13</strong>:</span> Arabidopsis growth conditions protocol (<span class="citation"><a href="#deOllas2019">de Ollas <i>et al.</i> 2019</a></span>).</figcaption></figure>

</blockquote>

<h1 id="optional-exercise">Optional exercise</h1>

<p>As additional activity, you can try to repeat the workflow by using the sequences stored in the NCBI GEO database with the accession number <code class="language-plaintext highlighter-rouge">GSE119382</code>. In that case, we will compare gene expression patterns of mutants overexpressing the brassinosteroid receptor BRL3 under two experimental conditions: control and drought-stress. The required datasets are available in the data library:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-import-data-from-the-data-libraries"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Import data from the Data Libraries</div>
  <ol>
    <li>Go into <strong>Data</strong> (top panel) and click on <strong>Data Libraries</strong></li>
    <li>In the search box enter the following identifier: <code class="language-plaintext highlighter-rouge">4710649</code></li>
    <li>Select the following files:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4710649/files/SRR7779222_BRL3_mRNA_drought.fastqsanger.gz
https://zenodo.org/record/4710649/files/SRR7779223_BRL3_mRNA_drought.fastqsanger.gz
https://zenodo.org/record/4710649/files/SRR7779224_BRL3_mRNA_drought.fastqsanger.gz
</code></pre></div>      </div>
    </li>
    <li>Click on <strong>Export to History</strong> and <strong>as a Collection</strong></li>
    <li>In the pop-up window press <kbd>Continue</kbd></li>
    <li>Provide it the name <code class="language-plaintext highlighter-rouge">BRL3 mRNA drought</code> and push <kbd>Create list</kbd></li>
    <li>Repeat the previous procedure with the remaining files:
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4710649/files/SRR7779228_BRL3_mRNA_watered.fastqsanger.gz
https://zenodo.org/record/4710649/files/SRR7779229_BRL3_mRNA_watered.fastqsanger.gz
</code></pre></div>      </div>
    </li>
    <li>Finally provide it the name <code class="language-plaintext highlighter-rouge">BRL3 mRNA control</code> and push <kbd>Create list</kbd></li>
  </ol>
</blockquote>

<p>We will use the upregulated miRNAs obtained in the previous analysis in order to identify potential targets.</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>Upregulated miRNA   https://zenodo.org/record/4710649/files/upregulated_miRNA_BR_complete_dataset.fasta
</code></pre></div></div>

<h1 id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed RNA sequencing data to extract information about potential genes regulated by brassinosteroids. For this purpose, the approach chosen was the identification of genes complementary to miRNAs upregulated in response by brassinosteroids. The final result has been the identification of five potential miRNA targets.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>MiRDeep2 in combination with the PmiREN database allows to quantify miRNA expression in plants</p>
</li>
                        
                        <li><p>The use of the Salmon tool and the AtRTD2 transcriptome allows rapid and accurate quantification of transcripts in <em>Arabidopsis</em></p>
</li>
                        
                        <li><p>The combination of differential expression analysis and the TargetFinder tool is an effective strategy for the identification of miRNA target genes</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Glazov2008">Glazov, E. A., S. McWilliam, W. C. Barris, and B. P. Dalrymple, 2008 <b>Origin,  Evolution,  and Biological Role of miRNA Cluster in DLK-DIO3 Genomic Region in Placental Mammals</b>. Molecular Biology and Evolution 25: 939–948. <a href="https://doi.org/10.1093/molbev/msn045">10.1093/molbev/msn045</a></li>
<li id="Ivashuta2011">Ivashuta, S., I. R. Banks, B. E. Wiggins, Y. Zhang, T. E. Ziegler <i>et al.</i>, 2011 <b>Regulation of Gene Expression in Plants through miRNA Inactivation</b> (G. Dalmasso, Ed.). PLoS ONE 6: e21330. <a href="https://doi.org/10.1371/journal.pone.0021330">10.1371/journal.pone.0021330</a></li>
<li id="Hu2013">Hu, W.-hai, X.-hong Yan, Y.-an Xiao, J.-jun Zeng, H.-juan Qi <i>et al.</i>, 2013 <b>24-Epibrassinosteroid alleviate drought-induced inhibition of photosynthesis in Capsicum annuum</b>. Scientia Horticulturae 150: 232–237. <a href="https://doi.org/10.1016/j.scienta.2012.11.012">10.1016/j.scienta.2012.11.012</a></li>
<li id="Srivastava2014">Srivastava, P. K., T. Moturu, P. Pandey, I. T. Baldwin, and S. P. Pandey, 2014 <b>A comparison of performance of plant miRNA target prediction tools and the characterization of features for genome-wide target prediction</b>. BMC Genomics 15: 348. <a href="https://doi.org/10.1186/1471-2164-15-348">10.1186/1471-2164-15-348</a></li>
<li id="Srivastava2016">Srivastava, A., H. Sarkar, N. Gupta, and R. Patro, 2016 <b>RapMap: a rapid,  sensitive and accurate tool for mapping RNA-seq reads to transcriptomes</b>. Bioinformatics 32: i192–i200. <a href="https://doi.org/10.1093/bioinformatics/btw277">10.1093/bioinformatics/btw277</a></li>
<li id="Gao2017">Gao, R., M. Y. Gruber, L. Amyot, and A. Hannoufa, 2017 <b>SPL13 regulates shoot branching and flowering time in Medicago sativa</b>. Plant Molecular Biology 96: 119–133. <a href="https://doi.org/10.1007/s11103-017-0683-8">10.1007/s11103-017-0683-8</a></li>
<li id="Ma2017">Ma, X., C. Liu, L. Gu, B. Mo, X. Cao <i>et al.</i>, 2017 <b>TarHunter,  a tool for predicting conserved microRNA targets and target mimics in plants</b> (I. Hofacker, Ed.). Bioinformatics 34: 1574–1576. <a href="https://doi.org/10.1093/bioinformatics/btx797">10.1093/bioinformatics/btx797</a></li>
<li id="Patro2017">Patro, R., G. Duggal, M. I. Love, R. A. Irizarry, and C. Kingsford, 2017 <b>Salmon provides fast and bias-aware quantification of transcript expression</b>. Nature Methods 14: 417–419. <a href="https://doi.org/10.1038/nmeth.4197">10.1038/nmeth.4197</a></li>
<li id="Anwar2018">Anwar, A., Y. Liu, R. Dong, L. Bai, X. Yu <i>et al.</i>, 2018 <b>The physiological and molecular mechanism of brassinosteroid in response to stress: a review</b>. Biological Research 51: <a href="https://doi.org/10.1186/s40659-018-0195-2">10.1186/s40659-018-0195-2</a></li>
<li id="Wang2018">Wang, C.-Y., T. Jégu, H.-P. Chu, H. J. Oh, and J. T. Lee, 2018 <b>SMCHD1 Merges Chromosome Compartments and Assists Formation of Super-Structures on the Inactive X</b>. Cell 174: 406–421.e25. <a href="https://doi.org/10.1016/j.cell.2018.05.007">10.1016/j.cell.2018.05.007</a></li>
<li id="Wang2019">Wang, Y., J.-J. Cao, K.-X. Wang, X.-J. Xia, K. Shi <i>et al.</i>, 2018 <b>BZR1 Mediates Brassinosteroid-Induced Autophagy and Nitrogen Starvation in Tomato</b>. Plant Physiology 179: 671–685. <a href="https://doi.org/10.1104/pp.18.01028">10.1104/pp.18.01028</a></li>
<li id="PlanasRiverola2019">Planas-Riverola, A., A. Gupta, I. Betegón-Putze, N. Bosch, M. Ibañes <i>et al.</i>, 2019 <b>Brassinosteroid signaling in plant development and adaptation to stress</b>. Development 146: dev151894. <a href="https://doi.org/10.1242/dev.151894">10.1242/dev.151894</a></li>
<li id="deOllas2019">Ollas, C. de, C. Segarra-Medina, M. González-Guzmán, J. Puertolas, and A. Gómez-Cadenas, 2019 <b>A customizable method to characterize Arabidopsis thaliana transpiration under drought conditions</b>. Plant Methods 15: <a href="https://doi.org/10.1186/s13007-019-0474-0">10.1186/s13007-019-0474-0</a></li>
<li id="Park2020">Park, S. Y., J.-H. Choi, D.-H. Oh, J. C. Johnson, M. Dassanayake <i>et al.</i>, 2020 <b>Genome-wide analysis of brassinosteroid responsive small RNAs in Arabidopsis thaliana</b>. Genes &amp; Genomics 42: 957–969. <a href="https://doi.org/10.1007/s13258-020-00964-2">10.1007/s13258-020-00964-2</a></li>
<li id="Feyissa2021">Feyissa, B. A., L. Amyot, V. Nasrollahi, Y. Papadopoulos, S. E. Kohalmi <i>et al.</i>, 2021 <b>Involvement of the miR156/SPL module in flooding response in Medicago sativa</b>. Scientific Reports 11: <a href="https://doi.org/10.1038/s41598-021-82450-7">10.1038/s41598-021-82450-7</a></li>
<li id="Yue2021">Yue, E., H. Tao, and J. Xu, 2021 <b>Genome-wide analysis of microRNA156 and its targets,  the genes encoding SQUAMOSA promoter-binding protein-like (SPL) transcription factors,  in the grass family Poaceae</b>. Journal of Zhejiang University-SCIENCE B 22: 366–382. <a href="https://doi.org/10.1631/jzus.b2000519">10.1631/jzus.b2000519</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=transcriptomics/mirna-target-finder"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=transcriptomics/mirna-target-finder">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Cristóbal Gallardo, Pavankumar Videm, Beatriz Serrano-Solano,  <b>Whole transcriptome analysis of Arabidopsis thaliana (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{transcriptomics-mirna-target-finder,
author = "Cristóbal Gallardo and Pavankumar Videm and Beatriz Serrano-Solano",
	title = "Whole transcriptome analysis of Arabidopsis thaliana (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: salmon
  owner: bgruening
  revisions: e3d32471da11
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trim_galore
  owner: bgruening
  revisions: 949f01671246
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: e7b2202befea
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: filter_by_fasta_ids
  owner: galaxyp
  revisions: 3c623e81be77
  tool_panel_section_label: Proteomics
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: deseq2
  owner: iuc
  revisions: 6a3a025714d3
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: b2f1f75d49c4
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: 5e33b465d8d5
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: sample_seqs
  owner: peterjc
  revisions: 7c40a1fbc82e
  tool_panel_section_label: Filter and Sort
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mirdeep2_mapper
  owner: rnateam
  revisions: a8d24f4b6d95
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: mirdeep2_quantifier
  owner: rnateam
  revisions: d5ea61ff12eb
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: targetfinder
  owner: rnateam
  revisions: c9d56748fbde
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">4</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>September 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: Data made small enough to process it together in class
					<b>Disliked</b>: Could you please correct the following typos: "Differentially expressed miRNAs" into "Differentially expressed mRNAs" in the section "mRNA data analysis > Filter.. Thanks!
				</li>
				
		    </ul>
		    
		    <b>April 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: I like all the indications step by step. Very clear. I would like to have some information about others parameters that are setted by default. 
					<b>Disliked</b>: I only correct one part where there are 2 mistakes. In "Filter significantly differentially expressed mRNAs" section, Hands-on: Extract significantly DE genes in point 3) Filter “Filter”: Differentially expressed miRNAs (replace by mRNAs)  And in point 5) Filter “Filter”: Differentially expressed miRNAs (replace by mRNAs). This is about the text. 
				</li>
				
		    </ul>
		    
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00292">gxy.io/GTN:T00292</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/transcriptomics/tutorials/mirna-target-finder/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/8a9b0dca294ebaf945f5b32f9e7854b7eca4b918">8a9b0dc</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>