<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Network analysis with Heinz</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: Network analysis with Heinz" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fnetwork-analysis-with-heinz%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fnetwork-analysis-with-heinz%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fnetwork-analysis-with-heinz%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fnetwork-analysis-with-heinz%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fnetwork-analysis-with-heinz%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fnetwork-analysis-with-heinz%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Network analysis with Heinz",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / network-analysis-with-heinz / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Network analysis with Heinz' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html",
  "timeRequired": "PT1H30M",
  "keywords": "metatranscriptomics, network analysis",
  "description": "The questions this  addresses are:\n - Which pathways are potentially contributing to dental caries?\n\n\\nThe objectives are:\n - Network Analysis of metatranscriptomics data using Heinz in Galaxy to pinpoint the optimal scoring subnetwork.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Chao (Cico) Zhang"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Chao (Cico) Zhang"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Network analysis with Heinz</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/cicozhang/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/cicozhang" alt="Chao (Cico) Zhang">Chao (Cico) Zhang</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>Which pathways are potentially contributing to dental caries?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Network Analysis of metatranscriptomics data using Heinz in Galaxy to pinpoint the optimal scoring subnetwork.</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 1 hour 30 minutes</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1344105">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/workflows/" title="Workflows" alt="Network analysis with Heinz workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/network-analysis-with-heinz/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://clipseq.usegalaxy.eu/" title="CLIP-Explorer">
            CLIP-Explorer
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://humancellatlas.usegalaxy.eu/" title="Human Cell Atlas">
            Human Cell Atlas
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/" title="Single Cell Omics workbench">
            Single Cell Omics workbench
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="overview">Overview</h1>

<p>The human microbiome plays a key role in health and disease. Thanks to comparative metatranscriptomics,
the cellular functions that are deregulated by the microbiome in disease can now be computationally
explored. Unlike gene-centric approaches, pathway-based methods provide a systemic view of such
functions; however, they typically consider each pathway in isolation and in its entirety.
They can therefore overlook the key differences that (i) span multiple pathways, (ii) contain
bidirectionally deregulated components, (iii) are confined to a pathway region. To capture these
properties, computational methods that reach beyond the scope of predefined pathways are needed.</p>

<p>In this tutorial, we will perform a network analysis using Heinz (<a href="https://github.com/ls-cwi/heinz">GitHub</a>,
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2718639/">publication</a>) in <span class="notranslate">Galaxy</span>. The data comes from
the study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a>,
and we will reproduce some of the computational steps from this study with simplified data and
parameters to speed up the analysis for the purposes of this tutorial.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#obtaining-and-preparing-data--" id="markdown-toc-obtaining-and-preparing-data--">Obtaining and preparing data  <img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1344105.svg" alt="DOI" /></a>    <ol>
      <li><a href="#understanding-our-input-data" id="markdown-toc-understanding-our-input-data">Understanding our input data</a></li>
      <li><a href="#importing-the-data-into-galaxy" id="markdown-toc-importing-the-data-into-galaxy">Importing the data into <span class="notranslate">Galaxy</span></a></li>
    </ol>
  </li>
  <li><a href="#differential-expression-analysis-dea-by-deseq2" id="markdown-toc-differential-expression-analysis-dea-by-deseq2">Differential <span class="notranslate">Expression</span> Analysis (DEA) by DESeq2</a>    <ol>
      <li><a href="#what-is-differential-expression-analysis" id="markdown-toc-what-is-differential-expression-analysis">What is differential <span class="notranslate">expression</span> analysis?</a></li>
      <li><a href="#which-tools-are-available-for-dea" id="markdown-toc-which-tools-are-available-for-dea">Which tools are available for DEA?</a></li>
      <li><a href="#perform-differential-expression-analysis-dea" id="markdown-toc-perform-differential-expression-analysis-dea">Perform Differential <span class="notranslate">Expression</span> Analysis (DEA)</a></li>
    </ol>
  </li>
  <li><a href="#fit-a-bum-model-a-mixture-model" id="markdown-toc-fit-a-bum-model-a-mixture-model">Fit a BUM model (a mixture model)</a></li>
  <li><a href="#pinpoint-the-key-pathways-with-heinz" id="markdown-toc-pinpoint-the-key-pathways-with-heinz">Pinpoint the key pathways with Heinz</a>    <ol>
      <li><a href="#calculate-heinz-scores" id="markdown-toc-calculate-heinz-scores">Calculate Heinz scores</a></li>
      <li><a href="#run-heinz-pinpoint-the-optimal-subnetwork" id="markdown-toc-run-heinz-pinpoint-the-optimal-subnetwork">Run Heinz: pinpoint the optimal subnetwork</a></li>
      <li><a href="#visualize-the-output-visualize-the-optimal-subnetwork" id="markdown-toc-visualize-the-output-visualize-the-optimal-subnetwork">Visualize the output: visualize the optimal subnetwork</a></li>
    </ol>
  </li>
  <li><a href="#save-the-history-into-a-workflow" id="markdown-toc-save-the-history-into-a-workflow">Save the history into a <span class="notranslate">workflow</span></a></li>
</ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-note-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: results may vary</h3>

  <p>Your results may be slightly different from the ones presented in this tutorial
due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or
because of stochastic processes in the algorithms.</p>

</blockquote>

<p>In this tutorial, we will run the Heinz <span class="notranslate">workflow</span> step by step to understand how each step works. To get an overview of what this <span class="notranslate">workflow</span> does
and what kind of calculation is involved, view the flowchart below.</p>

<p><img src="../../images/flowchart-heinz.png" alt="Heinz flowchart" /></p>

<h1 id="obtaining-and-preparing-data--">Obtaining and preparing data  <a href="https://doi.org/10.5281/zenodo.1344105"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1344105.svg" alt="DOI" /></a></h1>

<p>The study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a> includes the computation
steps starting from the raw <span class="notranslate">RNA</span>seq datasets. The operations that processed raw data into the interpreted data are
beyond the scope of this tutorial. To learn that, please refer to the relevant topics in the <span class="notranslate">Galaxy</span> training
material. In this tutorial, we start with the interpreted data, which are KO
(<a href="https://www.genome.jp/kegg/ko.html">KEGG Orthology</a>) count data. All the data needed for this tutorial are
available from Zenodo.</p>

<blockquote class="details">
  <h3 id="details-background-what-is-ko-count"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Background: What is KO count?</h3>

  <p>KOs are organism-independent identifiers that group together proteins of similar biochemical functions.
It is a term specific to <a href="https://www.genome.jp/kegg/">KEGG</a> database. It is a group concept, similar to the concept ‚Äúpathway‚Äù, which
includes a bunch of molecules and their interactions. Take a random line in our dataset, ‚ÄúK01369 7‚Äù as an example, the first column represents the KO ID,
and the second column ‚Äî the numeric value is the KO counts. If we understand it by the face value, it is 7 protein molecules that are counted from our dataset as this KO ‚ÄúK01369‚Äù; but we usually mingle proteins with genes because we often assume that each protein corresponds to a gene, so it can be understood as 7 genes. Now we are working on the microbial <span class="notranslate">RNA</span>seq data, and the assumption is also used subconsciously.</p>

</blockquote>

<h2 id="understanding-our-input-data">Understanding our input data</h2>

<p>According to the study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a>,
dental caries (DC) dataset in this tutorial came from an experiment that comprised of supragingival plaque
samples collected from all dental surfaces of 36 individuals who had either a caries-positive (disease) or
a caries-negative (health) oral profile. Each of the 36 samples was sequenced, pre-processed and transformed
into KO counts. We will use these count data as the starting point to perform the network analysis.</p>

<blockquote class="comment">
  <h3 id="comment-dataset-details"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Dataset details</h3>
  <p>The count data of the 36 samples are separated into 36 files, organized into two groups:
‚ÄòCP‚Äô (caries-positive) and ‚ÄòCN‚Äô (caries-negative).</p>
</blockquote>

<h2 id="importing-the-data-into-galaxy">Importing the data into <span class="notranslate">Galaxy</span></h2>

<p>After knowing what our input data are like, let‚Äôs get them into <span class="notranslate">Galaxy</span> history:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-obtaining-our-data"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Obtaining our data</h3>

  <ol>
    <li>
      <p>Make sure we have an empty <span class="notranslate">Galaxy</span> history. Give it a sensible name.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li><strong>Upload Disease Dataset</strong>
      <ul>
        <li>Open the file upload menu</li>
        <li>Click on  <strong>Collection</strong> tab</li>
        <li>Click on the <strong>Paste/Fetch data</strong> button</li>
        <li>Copy the Zenodo links for the Disease Datasets
          <blockquote class="details">
            <h3 id="details-view-list-of-zenodo-urls-for-dental-caries-dataset-disease-cp"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> View list of Zenodo URLs for Dental Caries Dataset (Disease, CP)</h3>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1344105/files/2241_CP_DZ_PairTo_2242.txt
https://zenodo.org/record/1344105/files/2126_CP_MZ_PairTo_2125.txt
https://zenodo.org/record/1344105/files/2991_CP_DZ_PairTo_2992.txt
https://zenodo.org/record/1344105/files/2931_CP_DZ_PairTo_2930.txt
https://zenodo.org/record/1344105/files/2284_CP_DZ_PairTo_2283.txt
https://zenodo.org/record/1344105/files/2125_CP_MZ_PairTo_2126.txt
https://zenodo.org/record/1344105/files/4131_CP_DZ_PairTo_4132.txt
https://zenodo.org/record/1344105/files/2954_CP_DZ_PairTo_2955.txt
https://zenodo.org/record/1344105/files/2170_CP_MZ_PairTo_2169.txt
https://zenodo.org/record/1344105/files/2955_CP_DZ_PairTo_2954.txt
https://zenodo.org/record/1344105/files/2011_CP_DZ_PairTo_2012.txt
https://zenodo.org/record/1344105/files/2012_CP_DZ_PairTo_2011.txt
https://zenodo.org/record/1344105/files/2269_CP_DZ_PairTo_2270.txt
https://zenodo.org/record/1344105/files/3215_CP_MZ_PairTo_3214.txt
https://zenodo.org/record/1344105/files/2354_CP_DZ_PairTo_2355.txt
https://zenodo.org/record/1344105/files/3306_CP_DZ_PairTo_3307.txt
https://zenodo.org/record/1344105/files/2061_CP_DZ_PairTo_2062.txt
https://zenodo.org/record/1344105/files/2355_CP_DZ_PairTo_2354.txt
https://zenodo.org/record/1344105/files/2242_CP_DZ_PairTo_2241.txt
</code></pre></div>            </div>
          </blockquote>
        </li>
        <li>Click on <strong>Start</strong> to Upload the files</li>
        <li>Click <strong>Build</strong> once upload has completed</li>
        <li>Enter <strong>Name</strong>: <code class="language-plaintext highlighter-rouge">CP</code></li>
        <li>Click <strong>Create list</strong> to make the collection</li>
      </ul>
    </li>
    <li><strong>Upload control (healthy) datasets</strong>
      <ul>
        <li>Repeat the previous steps with the samples from healthy individuals:
          <blockquote class="details">
            <h3 id="details-view-list-of-zenodo-urls-for-dental-caries-dataset-healthy-cn"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> View list of Zenodo URLs for Dental Caries Dataset (Healthy, CN)</h3>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1344105/files/2310_CN_DZ_PairTo_2309.txt
https://zenodo.org/record/1344105/files/2062_CN_DZ_PairTo_2061.txt
https://zenodo.org/record/1344105/files/2191_CN_MZ_PairTo_2192.txt
https://zenodo.org/record/1344105/files/2052_CN_MZ_PairTo_2051.txt
https://zenodo.org/record/1344105/files/2051_CN_MZ_PairTo_2052.txt
https://zenodo.org/record/1344105/files/2192_CN_MZ_PairTo_2191.txt
https://zenodo.org/record/1344105/files/2234_CN_DZ_PairTo_2233.txt
https://zenodo.org/record/1344105/files/2233_CN_DZ_PairTo_2234.txt
https://zenodo.org/record/1344105/files/2270_CN_DZ_PairTo_2269.txt
https://zenodo.org/record/1344105/files/2225_CN_MZ_PairTo_2226.txt
https://zenodo.org/record/1344105/files/4132_CN_DZ_PairTo_4131.txt
https://zenodo.org/record/1344105/files/2309_CN_DZ_PairTo_2310.txt
https://zenodo.org/record/1344105/files/2992_CN_DZ_PairTo_2991.txt
https://zenodo.org/record/1344105/files/3214_CN_MZ_PairTo_3215.txt
https://zenodo.org/record/1344105/files/2169_CN_MZ_PairTo_2170.txt
https://zenodo.org/record/1344105/files/2930_CN_DZ_PairTo_2931.txt
https://zenodo.org/record/1344105/files/3307_CN_DZ_PairTo_3306.txt
</code></pre></div>            </div>
          </blockquote>
        </li>
        <li>Name this collection: <code class="language-plaintext highlighter-rouge">CN</code></li>
      </ul>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <ol>
          <li>How many samples do you have in our disease collection (CP)? How many healthy samples (CN)?</li>
          <li>How many columns in each file? What are these columns?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>You should have 19 samples in the disease collection (CP), and 17 in the negative collection (CN).</li>
            <li>There are two columns, one is the KO IDs, the other is the count.</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <h3 id="tip-tip-creating-a-collection-from-files-already-in-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a collection from files already in your history</h3>

  <p><em>Dataset collections</em> enables us to easily run tools on multiple datasets at once, you probably
have done that in the file upload menu using the <code class="language-plaintext highlighter-rouge">Collection</code> tab. If not, you can still create
<em>dataset collections</em> manually.</p>

  <blockquote class="details">
    <h3 id="hands_on-manually-organizing-our-data-into-a-collection"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Manually organizing our data into a collection</h3>

    <p>If you have all the datasets in the history, but they are not organized into a collection yet,
you can follow these steps to create a collection:</p>

    <ol>
      <li>
        <p>Click on the <strong>checkmark icon</strong> at top of your history.
  <img src="../../../../shared/images/history_menu_buttons2.png" alt="Checkmark icon in history menu" /></p>
      </li>
      <li>
        <p>Select all the files whose name contains <code class="language-plaintext highlighter-rouge">CP</code>, then click on <strong>for all selected..</strong> and select
  <strong>Build Dataset List</strong> from the dropdown menu.</p>
      </li>
      <li>
        <p>In the next dialog window, you need to give a name, here we just set it to <code class="language-plaintext highlighter-rouge">CP</code>, then click <strong>Create list</strong>.</p>
      </li>
      <li>
        <p>Hidden these selected files by clicking on <strong>for all selected..</strong> and selecting <strong>Hidden datasets</strong>.
  <strong>Note:</strong> This step is optional, we do it here to keep <span class="notranslate">Galaxy</span> history clean.</p>
      </li>
      <li>
        <p>Redo the Step 2, 3, 4 for CN, set the name of the data list as ‚ÄòCN‚Äô.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="differential-expression-analysis-dea-by-deseq2">Differential <span class="notranslate">Expression</span> Analysis (DEA) by DESeq2</h1>

<h2 id="what-is-differential-expression-analysis">What is differential <span class="notranslate">expression</span> analysis?</h2>

<blockquote class="comment">
  <h3 id="comment-a-defintion"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> A defintion</h3>

  <p>The definition of differential <span class="notranslate">expression</span> analysis given by
<a href="https://www.ebi.ac.uk/training/online/course/functional-genomics-ii-common-technologies-and-data-analysis-methods/differential-gene">EBI</a>
means taking the normalised read count data and performing statistical analysis to discover quantitative
changes in <span class="notranslate">expression</span> levels between experimental groups. For example, we use statistical testing to decide
whether, for a given gene, an observed difference in read counts is significant, that is, whether it is greater
than what would be expected just due to natural random variation.</p>

</blockquote>

<p>In principle, DEA is a causal analysis; but in reality, it is hampered by the complexity of the experimental
situation and measurement. Back to our datasets, CP and CN, they are from two experimental groups. By DEA,
we hope to pinpoint the candidate genes relevant to dental caries first, then we will use Heinz to infer the
related pathways.</p>

<h2 id="which-tools-are-available-for-dea">Which tools are available for DEA?</h2>

<p>There are a few canned tools commonly used for DEA, like <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">Limma</a>
and <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>. If you are interested, you
may look up the pros and cons of each tool. Here we use DESeq2.</p>

<h2 id="perform-differential-expression-analysis-dea">Perform Differential <span class="notranslate">Expression</span> Analysis (DEA)</h2>

<p>After learning about differential <span class="notranslate">expression</span> analysis, let‚Äôs get some hands-on experience.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-dea-via-deseq2"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: DEA via DESeq2</h3>

  <ul>
    <li><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúSpecify a factor name‚Äù</em>: <code class="language-plaintext highlighter-rouge">dental_caries</code> (under <code class="language-plaintext highlighter-rouge">1: Factor</code>)</li>
        <li><em>‚ÄúSpecify a factor level‚Äù</em>: to <code class="language-plaintext highlighter-rouge">CP</code> (under <code class="language-plaintext highlighter-rouge">1: Factor level</code>)
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúCounts file(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">CP</code> (collection)</li>
          </ul>
        </li>
        <li><em>‚ÄúSpecify a factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge">CN</code> (under <code class="language-plaintext highlighter-rouge">2: Factor level</code>)
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúCounts file(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">CN</code> (collection)</li>
          </ul>
        </li>
        <li><em>‚ÄúFiles have header‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúVisualising the analysis results‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>Leave all other parameters to the default settings <br /><br /></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>It takes a few minutes to finish DESseq2. After the analysis is finished, have a look at the file, it should look like something this:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>GeneID  Base mean   log2(FC)	        StdErr	            Wald-Stats	        P-value	                P-adj
K15661  55.0733128361562    2.49892393773464	0.508475451930939	4.91454194739384	8.89902710599613e-07	0.00491938218419466
K15666  48.4532561772761    2.22805029428311	0.493391718412049	4.51578372951607	6.30830206230236e-06	0.00820667379798327
K10213  25.1966742274619    -2.45624670858868	0.550895251183889	-4.45864563782342	8.24791475064012e-06	0.00820667379798327
K03732  563.63634258472 -1.41961721984814	0.316992240900184	-4.47839737596338	7.52055094378821e-06	0.00820667379798327
K01792  112.923146882195    -1.26925892659617	0.285732234964578	-4.44212717810268	8.90738834802815e-06	0.00820667379798327
</code></pre></div></div>

<h1 id="fit-a-bum-model-a-mixture-model">Fit a BUM model (a mixture model)</h1>

<p>From a statistical point of view, p-values are uniformly distributed under null hypothesis; in other words,
under alte<span class="notranslate">rna</span>tive hypothesis, the noise component (which holds under null hypothesis) will be adequately
modeled by a uniform distribution. With this knowledge, we can fit these p-values to a mixture model
(BUM model), as the figure below shows.</p>

<p><img src="../../images/bum.jpeg" alt="p-values are fitted to a mixture model" width="50%" /></p>

<blockquote class="comment">
  <h3 id="comment-why-do-we-need-a-mixture-model"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Why do we need a mixture model?</h3>

  <p>To avoid making this tutorial sound like a math class, let‚Äôs focus on the philosophy of using a statistical model.
In the graph above, we have visualized the p-values; those bins have some shapes; by viewing the figure, it
is possible for us to cherrypick the data points (KO) with mesh eyes, but it must be tedious and boring, nobody wants to
spend a few days on doing this. In this situation, computers come into play; however, unlike humans, computers cannot understand the graph but
mathematical formulas, that‚Äôs why we need to fix the data into a statistical model (denoted by the red curve and blue line in the graph),
which computers can understand easily.</p>

</blockquote>

<p>Before fitting to BUM model in <span class="notranslate">Galaxy</span>, we need to prepare the input data for the tool
<strong>Fit a BUM model</strong>, that‚Äôs a file that only contains p-values.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-p-values-from-deseq2-output"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: extract p-values from DESeq2 output</h3>

  <ul>
    <li><strong>cut</strong> columns from a table <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: to <code class="language-plaintext highlighter-rouge">TAB</code></li>
        <li><em>‚ÄúFrom‚Äù</em>: to the output of <strong>DESeq2</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>Now we can <strong>Fit a BUM model</strong>:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-fit-the-bum-model"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: fit the BUM model</h3>

  <ul>
    <li><strong>Fit a BUM model</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúInput file‚Äù</em>: the output of <strong>cut</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<h1 id="pinpoint-the-key-pathways-with-heinz">Pinpoint the key pathways with Heinz</h1>

<p>After getting the parameters of the BUM model from the last step, we will use Heinz to pinpoint
the optimal subnetwork. Then we could look for the key pathways relevant to the subnetwork.
Before we continue, let‚Äôs figure out what Heinz is actually doing.</p>

<p>Heinz is an algorithm for searching an optimal subnetwork from a bigger network. You may wonder what
the network is here. Through the previous steps, we have got a list of identities, that is a list
of gene IDs with p-values, which form the nodes  of ‚Äòthe bigger network‚Äô, the relations between the
nodes, that is the edges, needs to be obtained from a background network, which represents a pathway
relation databases, such as <a href="https://reactome.org/">Reactome</a> and <a href="https://string-db.org/">STRING</a>.
In this tutorial, we only use a small sample background network for demonstration purposes. The
background network is represented as edges in a TXT file where each line denotes an edge as follows:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ACTR1B	ACVR2B
ZSWIM9	FOXP3
LGALS4	PRKX
NPTX1	CIAO1
</code></pre></div></div>

<p>Upload this edge file (hereafter we call it edge file) into the <span class="notranslate">Galaxy</span> instance.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-upload-edge-file"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Upload edge file</h3>

  <ol>
    <li><strong>Upload the edge file from Zenodo</strong>
      <ul>
        <li>Open the upload menu</li>
        <li>Choose <strong>Paste/Fetch data</strong> option</li>
        <li>Paste the URL into the box:
          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1344105/files/edge.txt
</code></pre></div>          </div>
        </li>
        <li>Click <strong>Start</strong></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h2 id="calculate-heinz-scores">Calculate Heinz scores</h2>

<p>As the first step, we need to calculate a Heinz score for each node, using the BUM model parameters
we obtained; meanwhile, we also need to specify an FDR value as input.</p>

<blockquote class="comment">
  <h3 id="comment-what-is-an-fdr-value"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> What is an FDR value?</h3>

  <p>FDR is short for false discovery rate, which is a method of conceptualizing the rate of type I errors
in null hypothesis testing when conducting multiple comparisons, if you are interested, view the detail
in <a href="https://en.wikipedia.org/wiki/False_discovery_rate">Wikipedia</a>.</p>

</blockquote>

<p>For different datasets and problems, we probably need to pick up an FDR value separately. Here we set FDR to 0.11.</p>

<p>Similar to <strong>Fit a BUM model</strong>, we also need to prepare the input data for the tool
<strong>Calculate a Heinz score</strong>.</p>

<blockquote class="question">
  <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>What is the requirement of the input data format for <strong>Calculate a Heinz score</strong>?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>In the user interface of the tool ‚ÄúCalculate a Heinz score‚Äù, we see that ‚ÄúA node file with p-values‚Äù is needed.
It should contain two columns delimited by tab, one is KO ID; the other, p-value.</p>

  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-geneid-and-p-values-from-deseq2-output"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: extract geneID and p-values from DESeq2 output</h3>

  <ul>
    <li><strong>cut</strong> columns from a table <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúCut columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">c1,c6</code></li>
        <li><em>‚ÄúDelimited by‚Äù</em>: <code class="language-plaintext highlighter-rouge">TAB</code></li>
        <li><em>‚ÄúFrom‚Äù</em>: the output of <strong>DESeq2</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-calculate-heinz-scores"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: calculate Heinz scores</h3>

  <ul>
    <li><strong>Calculate a Heinz score</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúA node file with p-values‚Äù</em>: the output of the previous <strong>cut</strong> step</li>
        <li><em>‚ÄúFDR value‚Äù</em>: to 0.11</li>
        <li><em>‚ÄúChoose your input type for BUM parameters‚Äù</em>: <code class="language-plaintext highlighter-rouge">The output file of BUM model</code></li>
        <li><em>‚ÄúOutput file of BUM model as input: lambda on the first line and alpha, the second‚Äù</em>: the output of <strong>Fit a BUM model</strong></li>
      </ul>
    </li>
  </ul>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-what-is-the-heinz-score"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> What is the Heinz score?</h3>

  <p>To figure out this score without reading the formula, we can understand Heinz score in this way. FDR is involved in calculating a p-value threshold
and any KO whose p-value is below this threshold is regarded as significant, which means the Heinz score is positive (another calculation in the formula). If we pick a higher FDR value, then we will have a higher p-value threshold, and more KOs are probably regarded as significant. In this situation, we probably have many false positive (those regarded as significant are actually not) on the one hand; on the other hand, Heinz will deliver a bigger subnetwork, which might be exhausting to analyze. Therefore, we need to pick up an FDR value properly.</p>

</blockquote>

<h2 id="run-heinz-pinpoint-the-optimal-subnetwork">Run Heinz: pinpoint the optimal subnetwork</h2>

<p>After getting Heinz scores, let‚Äôs run Heinz program to find the optimal subnetwork from the background network which we mentioned earlier.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-pinpoint-the-optimal-subnetwork"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: pinpoint the optimal subnetwork</h3>

  <ul>
    <li><strong>Identify optimal scoring subnetwork</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúFile containing Heinz scores‚Äù</em>: the output of <strong>Calculate a Heinz score</strong></li>
        <li><em>‚ÄúEdge file‚Äù</em>: the edge file uploaded</li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>It usually takes a few minutes to get the result, but mind you, for some tasks, it might take a few hours to get a result in practice.</p>

<blockquote class="comment">
  <h3 id="comment-the-running-time-of-the-program"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> The running time of the program</h3>

  <ul>
    <li>Graph problem is way more complicated than we thought.</li>
    <li>It might take a much longer time for some complicated datasets.</li>
    <li>We can use multiple CPUs to accelerate the computation (for now, this function is not available in <span class="notranslate">Galaxy</span> yet),
but, to use that, you can install Heinz directly via <a href="https://anaconda.org/bioconda/heinz">Bioconda</a> in a Linux environment.</li>
  </ul>
</blockquote>

<h2 id="visualize-the-output-visualize-the-optimal-subnetwork">Visualize the output: visualize the optimal subnetwork</h2>

<p>The result we got from the last step is not very human-readable, is it? It is a little painful to understand the Heinz result directly. Therefore we need to visualize the output by making it into graphs.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-visualize-the-optimal-subnetwork"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: visualize the optimal subnetwork</h3>

  <ul>
    <li><strong>Visualize</strong> the optimal scoring subnetwork <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>‚ÄúHeinz output file‚Äù</em>: the output of <strong>Identify optimal scoring subnetwork</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>In this tutorial, you probably get a similar graph to the following:</p>

<p><img src="../../images/Heinz_visualisation.png" alt="p-values are fitted to a mixture model" width="30%" /></p>

<blockquote class="question">
  <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <p>According to the figure above, Why are some shapes are round, others square?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>Round shape corresponds to positive Heinz score; square shape corresponds to negative Heinz score.</p>

  </blockquote>
</blockquote>

<p>In addition to the tool we will use in <span class="notranslate">Galaxy</span>, you may consider using eXamine plugin in Cytoscape for a richer visualization, if we use the original data from the study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a>, perform all of these steps and visualize the result by eXamine, we can get a visualization as follows:</p>

<p><img src="../../images/metaModules.png" alt="metaModules Result" width="60%" /></p>

<p>If you are interested, you may try to make sense of the result yourself. To answer the question ‚Äúwhich pathways are potentially contributing to dental caries?‚Äù, we need to find the key pathways relevant to the subnetwork. To achieve that, we probably need to combine information from other bioinformatics services and literature.</p>

<h1 id="save-the-history-into-a-workflow">Save the history into a <span class="notranslate">workflow</span></h1>

<p>Let‚Äôs assume that these are the operations we always need for different datasets, to make life easier, we can save all of these operations into a <span class="notranslate">workflow</span> in <span class="notranslate">Galaxy</span>, which you can reuse for different datasets next time.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-extract-the-heinz-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Extract the Heinz <span class="notranslate">workflow</span></h3>

  <ol>
    <li>
      <p>Extract the <span class="notranslate">workflow</span> from your history</p>

      <blockquote class="tip">

        <h3 id="tip-tip-extracting-a-workflow-from-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Extracting a <span class="notranslate">workflow</span> from history</h3>

        <ol>
          <li>Remove any failed or unwanted jobs from your history.</li>
          <li>Click on <strong>History options</strong> (gear icon <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span>) at the top of your history panel.</li>
          <li>Select <strong>Extract <span class="notranslate">workflow</span></strong></li>
          <li>Check the steps, enter a name for your <span class="notranslate">workflow</span>, and press the <strong>Create <span class="notranslate">Workflow</span></strong> button.</li>
        </ol>

      </blockquote>
    </li>
    <li>
      <p>Run the <span class="notranslate">workflow</span> on the dataset in this tutorial</p>
    </li>
  </ol>

</blockquote>

<p><img src="../../images/heinz-workflow.png" alt="Heinz workflow" /></p>

<p>As a self practice, you ran the newly created <span class="notranslate">workflow</span> with the dataset in this tutorial. Do you feel the convenience of running a <span class="notranslate">workflow</span> directly?</p>

<p>Congrats! You have finished all the tools in Heinz <span class="notranslate">workflow</span>! You have successfully run each tool and understood how it works. In real research, running these tools are only part of the effort, we still need to invest a huge amount of intelligence in making sense of the results and converting them to knowledge, which is fraught with uncertainties and confounders, where perhaps luck will come into play. So good luck!</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Using Heinz do the network analysis based on the metatranscriptomics or transcriptomics data.</p>
</li>
                            
                            <li><p>Extracting the signals hidden in the p-value distribution of the differential expression analysis.</p>
</li>
                            
                            <li><p>Visualizing the pinpointed subnetwork helps us understand the result better</p>
</li>
                            
                            <li><p>Heinz helps narrow down the scope to find the key pathways</p>
</li>
                            
                            <li><p>Multiple CPUs will accelerate the computation especially for the complex datasets</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Network analysis with Heinz (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Chao (Cico) Zhang, 2021 <b>Network analysis with Heinz (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-network-analysis-with-heinz,
    author = "Chao (Cico) Zhang",
    title = "Network analysis with Heinz (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Chao (Cico) Zhang)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
