<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Network analysis with Heinz</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@9de96f24bf701758b2f5e169c4da60878ecefe7d",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.f1f2de89.js?v=1701303507' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.03aefefa.js?v=1701303507' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Network analysis with Heinz">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-05-18 09:34:29 +0000">
<meta name="DC.creator" content="Chao Zhang">

        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis.">
        <meta property="og:site_name" content="Galaxy Training Network">
        <meta property="og:title" content="Galaxy Training: Network analysis with Heinz">
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis.">
        
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png">

        
        <script type="application/ld+json">
            


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "dateModified": "2023-05-18 09:34:29 +0000",
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Network analysis with Heinz",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "identifier": "https://gxy.io/GTN:T00293",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Network analysis with Heinz",
  "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html",
  "version": 23,
  "timeRequired": "PT1H30M",
  "teaches": "Network Analysis of metatranscriptomics data using Heinz in Galaxy to pinpoint the optimal scoring subnetwork.",
  "keywords": [
    "Transcriptomics",
    "metatranscriptomics",
    "network analysis"
  ],
  "description": "The questions this  addresses are:\n - Which pathways are potentially contributing to dental caries?\n\n\nThe objectives are:\n - Network Analysis of metatranscriptomics data using Heinz in Galaxy to pinpoint the optimal scoring subnetwork.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/ChaoZhang123/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/ChaoZhang123/",
      "name": "Chao Zhang",
      "image": "https://avatars.githubusercontent.com/ChaoZhang123",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"
        }
      ]
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org/training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ],
  "educationalLevel": "Introductory",
  "mentions": [
    {
      "@type": "Thing",
      "name": "metatranscriptomics"
    },
    {
      "@type": "Thing",
      "name": "network analysis"
    }
  ],
  "abstract": "# Overview"
}
        </script>
        
    </head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script src='/training-material/assets/js/bundle.f1f2de89.js?v=1701303507'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Transcriptomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        <a class="dropdown-item" href="/training-material/topics/transcriptomics/faqs/" title="Check our FAQs for the Transcriptomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.md" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                    <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

		<a href="/training-material/preferences.html" class="dropdown-item">
			<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> User Preferences <span class="badge badge-secondary">New</span>
		</a>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/smorgasbord2023/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">smorgasbord2023</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-11-01/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">2023-11-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2023-10-01/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">2023-10-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-transcriptomics">
    <h1 data-toc-skip>Network analysis with Heinz</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/ChaoZhang123/" class="contributor-badge contributor-ChaoZhang123"><img src="https://avatars.githubusercontent.com/ChaoZhang123?s=36" alt="Chao Zhang avatar" width="36" class="avatar" />
    Chao Zhang</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Which pathways are potentially contributing to dental caries?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Network Analysis of metatranscriptomics data using Heinz in Galaxy to pinpoint the optimal scoring subnetwork.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                        
                    
                
                    
                        
                    
                        
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

        
        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour 30 minutes</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.1344105">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/workflows/" title="Network analysis with Heinz workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            













  



            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.be/" title="">
			UseGalaxy.be <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> ⭐️
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main)
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics//docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> May 18, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00293">https://gxy.io/GTN:T00293</a> </div>
        

	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 23 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <h1 id="overview">Overview</h1>

<p>The human microbiome plays a key role in health and disease. Thanks to comparative metatranscriptomics,
the cellular functions that are deregulated by the microbiome in disease can now be computationally
explored. Unlike gene-centric approaches, pathway-based methods provide a systemic view of such
functions; however, they typically consider each pathway in isolation and in its entirety.
They can therefore overlook the key differences that (i) span multiple pathways, (ii) contain
bidirectionally deregulated components, (iii) are confined to a pathway region. To capture these
properties, computational methods that reach beyond the scope of predefined pathways are needed.</p>

<p>In this tutorial, we will perform a network analysis using Heinz (<a href="https://github.com/ls-cwi/heinz">GitHub</a>,
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2718639/">publication</a>) in Galaxy. The data comes from
the study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a>,
and we will reproduce some of the computational steps from this study with simplified data and
parameters to speed up the analysis for the purposes of this tutorial.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#overview" id="markdown-toc-overview">Overview</a></li>
  <li><a href="#obtaining-and-preparing-data--" id="markdown-toc-obtaining-and-preparing-data--">Obtaining and preparing data  <a href="https://zenodo.org/badge/DOI/10.5281/zenodo.1344105.svg" rel="noopener noreferrer"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1344105.svg" alt="DOI. " loading="lazy" /></a></a>    <ol>
      <li><a href="#understanding-our-input-data" id="markdown-toc-understanding-our-input-data">Understanding our input data</a></li>
      <li><a href="#importing-the-data-into-galaxy" id="markdown-toc-importing-the-data-into-galaxy">Importing the data into Galaxy</a></li>
    </ol>
  </li>
  <li><a href="#differential-expression-analysis-dea-by-deseq2" id="markdown-toc-differential-expression-analysis-dea-by-deseq2">Differential Expression Analysis (DEA) by DESeq2</a>    <ol>
      <li><a href="#what-is-differential-expression-analysis" id="markdown-toc-what-is-differential-expression-analysis">What is differential expression analysis?</a></li>
      <li><a href="#which-tools-are-available-for-dea" id="markdown-toc-which-tools-are-available-for-dea">Which tools are available for DEA?</a></li>
      <li><a href="#perform-differential-expression-analysis-dea" id="markdown-toc-perform-differential-expression-analysis-dea">Perform Differential Expression Analysis (DEA)</a></li>
    </ol>
  </li>
  <li><a href="#fit-a-bum-model-a-mixture-model" id="markdown-toc-fit-a-bum-model-a-mixture-model">Fit a BUM model (a mixture model)</a></li>
  <li><a href="#pinpoint-the-key-pathways-with-heinz" id="markdown-toc-pinpoint-the-key-pathways-with-heinz">Pinpoint the key pathways with Heinz</a>    <ol>
      <li><a href="#calculate-heinz-scores" id="markdown-toc-calculate-heinz-scores">Calculate Heinz scores</a></li>
      <li><a href="#run-heinz-pinpoint-the-optimal-subnetwork" id="markdown-toc-run-heinz-pinpoint-the-optimal-subnetwork">Run Heinz: pinpoint the optimal subnetwork</a></li>
      <li><a href="#visualize-the-output-visualize-the-optimal-subnetwork" id="markdown-toc-visualize-the-output-visualize-the-optimal-subnetwork">Visualize the output: visualize the optimal subnetwork</a></li>
    </ol>
  </li>
  <li><a href="#save-the-history-into-a-workflow" id="markdown-toc-save-the-history-into-a-workflow">Save the history into a workflow</a></li>
</ol>

</blockquote>

<!--SNIPPET-->
<blockquote class="comment">   <div class="box-title comment-title" id="comment-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Results may vary</div>   <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, external databases, or because of stochastic processes in the algorithms.</p> </blockquote>
<p><!--END_SNIPPET--></p>

<p>In this tutorial, we will run the Heinz workflow step by step to understand how each step works. To get an overview of what this workflow does
and what kind of calculation is involved, view the flowchart below.</p>

<p><a href="../../images/transcriptomics_images/flowchart-heinz.png" rel="noopener noreferrer"><img src="../../images/transcriptomics_images/flowchart-heinz.png" alt="Heinz flowchart. " width="3112" height="558" loading="lazy" /></a></p>

<h1 id="obtaining-and-preparing-data--">Obtaining and preparing data  <a href="https://doi.org/10.5281/zenodo.1344105"><a href="https://zenodo.org/badge/DOI/10.5281/zenodo.1344105.svg" rel="noopener noreferrer"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.1344105.svg" alt="DOI. " loading="lazy" /></a></a></h1>

<p>The study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a> includes the computation
steps starting from the raw RNAseq datasets. The operations that processed raw data into the interpreted data are
beyond the scope of this tutorial. To learn that, please refer to the relevant topics in the Galaxy training
material. In this tutorial, we start with the interpreted data, which are KO
(<a href="https://www.genome.jp/kegg/ko.html">KEGG Orthology</a>) count data. All the data needed for this tutorial are
available from Zenodo.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-background-what-is-ko-count"><button class="gtn-boxify-button details" type="button" aria-controls="details-background-what-is-ko-count" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Background: What is KO count?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>KOs are organism-independent identifiers that group together proteins of similar biochemical functions.
It is a term specific to <a href="https://www.genome.jp/kegg/">KEGG</a> database. It is a group concept, similar to the concept “pathway”, which
includes a bunch of molecules and their interactions. Take a random line in our dataset, “K01369 7” as an example, the first column represents the KO ID,
and the second column — the numeric value is the KO counts. If we understand it by the face value, it is 7 protein molecules that are counted from our dataset as this KO “K01369”; but we usually mingle proteins with genes because we often assume that each protein corresponds to a gene, so it can be understood as 7 genes. Now we are working on the microbial RNAseq data, and the assumption is also used subconsciously.</p>

</blockquote>

<h2 id="understanding-our-input-data">Understanding our input data</h2>

<p>According to the study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a>,
dental caries (DC) dataset in this tutorial came from an experiment that comprised of supragingival plaque
samples collected from all dental surfaces of 36 individuals who had either a caries-positive (disease) or
a caries-negative (health) oral profile. Each of the 36 samples was sequenced, pre-processed and transformed
into KO counts. We will use these count data as the starting point to perform the network analysis.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-dataset-details"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Dataset details</div>
  <p>The count data of the 36 samples are separated into 36 files, organized into two groups:
‘CP’ (caries-positive) and ‘CN’ (caries-negative).</p>
</blockquote>

<h2 id="importing-the-data-into-galaxy">Importing the data into Galaxy</h2>

<p>After knowing what our input data are like, let’s get them into Galaxy history:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-obtaining-our-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Obtaining our data</div>

  <ol>
    <li>
      <p>Make sure we have an empty Galaxy history. Give it a sensible name.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>   <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>   <ol>   <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>   <li>Select the option <strong>Create New</strong> from the menu</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li><strong>Upload Disease Dataset</strong>
      <ul>
        <li>Open the file upload menu</li>
        <li>Click on  <strong>Collection</strong> tab</li>
        <li>Click on the <strong>Paste/Fetch data</strong> button</li>
        <li>Copy the Zenodo links for the Disease Datasets
          <blockquote class="details">
            <div class="box-title details-title" id="details-view-list-of-zenodo-urls-for-dental-caries-dataset-disease-cp"><button class="gtn-boxify-button details" type="button" aria-controls="details-view-list-of-zenodo-urls-for-dental-caries-dataset-disease-cp" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: View list of Zenodo URLs for Dental Caries Dataset (Disease, CP)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1344105/files/2241_CP_DZ_PairTo_2242.txt
https://zenodo.org/record/1344105/files/2126_CP_MZ_PairTo_2125.txt
https://zenodo.org/record/1344105/files/2991_CP_DZ_PairTo_2992.txt
https://zenodo.org/record/1344105/files/2931_CP_DZ_PairTo_2930.txt
https://zenodo.org/record/1344105/files/2284_CP_DZ_PairTo_2283.txt
https://zenodo.org/record/1344105/files/2125_CP_MZ_PairTo_2126.txt
https://zenodo.org/record/1344105/files/4131_CP_DZ_PairTo_4132.txt
https://zenodo.org/record/1344105/files/2954_CP_DZ_PairTo_2955.txt
https://zenodo.org/record/1344105/files/2170_CP_MZ_PairTo_2169.txt
https://zenodo.org/record/1344105/files/2955_CP_DZ_PairTo_2954.txt
https://zenodo.org/record/1344105/files/2011_CP_DZ_PairTo_2012.txt
https://zenodo.org/record/1344105/files/2012_CP_DZ_PairTo_2011.txt
https://zenodo.org/record/1344105/files/2269_CP_DZ_PairTo_2270.txt
https://zenodo.org/record/1344105/files/3215_CP_MZ_PairTo_3214.txt
https://zenodo.org/record/1344105/files/2354_CP_DZ_PairTo_2355.txt
https://zenodo.org/record/1344105/files/3306_CP_DZ_PairTo_3307.txt
https://zenodo.org/record/1344105/files/2061_CP_DZ_PairTo_2062.txt
https://zenodo.org/record/1344105/files/2355_CP_DZ_PairTo_2354.txt
https://zenodo.org/record/1344105/files/2242_CP_DZ_PairTo_2241.txt
</code></pre></div>            </div>
          </blockquote>
        </li>
        <li>Click on <strong>Start</strong> to Upload the files</li>
        <li>Click <strong>Build</strong> once upload has completed</li>
        <li>Enter <strong>Name</strong>: <code class="language-plaintext highlighter-rouge">CP</code></li>
        <li>Click <strong>Create list</strong> to make the collection</li>
      </ul>
    </li>
    <li><strong>Upload control (healthy) datasets</strong>
      <ul>
        <li>Repeat the previous steps with the samples from healthy individuals:
          <blockquote class="details">
            <div class="box-title details-title" id="details-view-list-of-zenodo-urls-for-dental-caries-dataset-healthy-cn"><button class="gtn-boxify-button details" type="button" aria-controls="details-view-list-of-zenodo-urls-for-dental-caries-dataset-healthy-cn" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: View list of Zenodo URLs for Dental Caries Dataset (Healthy, CN)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1344105/files/2310_CN_DZ_PairTo_2309.txt
https://zenodo.org/record/1344105/files/2062_CN_DZ_PairTo_2061.txt
https://zenodo.org/record/1344105/files/2191_CN_MZ_PairTo_2192.txt
https://zenodo.org/record/1344105/files/2052_CN_MZ_PairTo_2051.txt
https://zenodo.org/record/1344105/files/2051_CN_MZ_PairTo_2052.txt
https://zenodo.org/record/1344105/files/2192_CN_MZ_PairTo_2191.txt
https://zenodo.org/record/1344105/files/2234_CN_DZ_PairTo_2233.txt
https://zenodo.org/record/1344105/files/2233_CN_DZ_PairTo_2234.txt
https://zenodo.org/record/1344105/files/2270_CN_DZ_PairTo_2269.txt
https://zenodo.org/record/1344105/files/2225_CN_MZ_PairTo_2226.txt
https://zenodo.org/record/1344105/files/4132_CN_DZ_PairTo_4131.txt
https://zenodo.org/record/1344105/files/2309_CN_DZ_PairTo_2310.txt
https://zenodo.org/record/1344105/files/2992_CN_DZ_PairTo_2991.txt
https://zenodo.org/record/1344105/files/3214_CN_MZ_PairTo_3215.txt
https://zenodo.org/record/1344105/files/2169_CN_MZ_PairTo_2170.txt
https://zenodo.org/record/1344105/files/2930_CN_DZ_PairTo_2931.txt
https://zenodo.org/record/1344105/files/3307_CN_DZ_PairTo_3306.txt
</code></pre></div>            </div>
          </blockquote>
        </li>
        <li>Name this collection: <code class="language-plaintext highlighter-rouge">CN</code></li>
      </ul>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>How many samples do you have in our disease collection (CP)? How many healthy samples (CN)?</li>
          <li>How many columns in each file? What are these columns?</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>You should have 19 samples in the disease collection (CP), and 17 in the negative collection (CN).</li>
            <li>There are two columns, one is the KO IDs, the other is the count.</li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <div class="box-title tip-title" id="tip-creating-a-collection-from-files-already-in-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-collection-from-files-already-in-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Creating a collection from files already in your history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p><em>Dataset collections</em> enables us to easily run tools on multiple datasets at once, you probably
have done that in the file upload menu using the <code class="language-plaintext highlighter-rouge">Collection</code> tab. If not, you can still create
<em>dataset collections</em> manually.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-manually-organizing-our-data-into-a-collection"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Manually organizing our data into a collection</div>

    <p>If you have all the datasets in the history, but they are not organized into a collection yet,
you can follow these steps to create a collection:</p>

    <ol>
      <li>
        <p>Click on the <strong>checkmark icon</strong> at top of your history.
  <a href="../../../../shared/images/history_menu_buttons2.png" rel="noopener noreferrer"><img src="../../../../shared/images/history_menu_buttons2.png" alt="Checkmark icon in history menu. " width="75" height="33" loading="lazy" /></a></p>
      </li>
      <li>
        <p>Select all the files whose name contains <code class="language-plaintext highlighter-rouge">CP</code>, then click on <strong>n of N selected</strong> and select
  <strong>Build Dataset List</strong> from the dropdown menu.</p>
      </li>
      <li>
        <p>In the next dialog window, you need to give a name, here we just set it to <code class="language-plaintext highlighter-rouge">CP</code>, then click <strong>Create list</strong>.</p>
      </li>
      <li>
        <p>Redo the Step 2, 3 for CN, set the name of the data list as ‘CN’.</p>
      </li>
    </ol>
  </blockquote>
</blockquote>

<h1 id="differential-expression-analysis-dea-by-deseq2">Differential Expression Analysis (DEA) by DESeq2</h1>

<h2 id="what-is-differential-expression-analysis">What is differential expression analysis?</h2>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-a-definition"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: A definition</div>

  <p>The definition of differential expression analysis given by
<a href="https://www.ebi.ac.uk/training/online/course/functional-genomics-ii-common-technologies-and-data-analysis-methods/differential-gene">EBI</a>
means taking the normalised read count data and performing statistical analysis to discover quantitative
changes in expression levels between experimental groups. For example, we use statistical testing to decide
whether, for a given gene, an observed difference in read counts is significant, that is, whether it is greater
than what would be expected just due to natural random variation.</p>

</blockquote>

<p>In principle, DEA is a causal analysis; but in reality, it is hampered by the complexity of the experimental
situation and measurement. Back to our datasets, CP and CN, they are from two experimental groups. By DEA,
we hope to pinpoint the candidate genes relevant to dental caries first, then we will use Heinz to infer the
related pathways.</p>

<h2 id="which-tools-are-available-for-dea">Which tools are available for DEA?</h2>

<p>There are a few canned tools commonly used for DEA, like <a href="https://bioconductor.org/packages/release/bioc/html/limma.html">Limma</a>
and <a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">DESeq2</a>. If you are interested, you
may look up the pros and cons of each tool. Here we use DESeq2.</p>

<h2 id="perform-differential-expression-analysis-dea">Perform Differential Expression Analysis (DEA)</h2>

<p>After learning about differential expression analysis, let’s get some hands-on experience.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-dea-via-deseq2"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: DEA via DESeq2</div>

  <ul>
    <li><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Specify a factor name”</em>: <code class="language-plaintext highlighter-rouge">dental_caries</code> (under <code class="language-plaintext highlighter-rouge">1: Factor</code>)</li>
        <li><em>“Specify a factor level”</em>: to <code class="language-plaintext highlighter-rouge">CP</code> (under <code class="language-plaintext highlighter-rouge">1: Factor level</code>)
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Counts file(s)”</em>: <code class="language-plaintext highlighter-rouge">CP</code> (collection)</li>
          </ul>
        </li>
        <li><em>“Specify a factor level”</em>: <code class="language-plaintext highlighter-rouge">CN</code> (under <code class="language-plaintext highlighter-rouge">2: Factor level</code>)
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Counts file(s)”</em>: <code class="language-plaintext highlighter-rouge">CN</code> (collection)</li>
          </ul>
        </li>
        <li><em>“Files have header”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Visualising the analysis results”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>Leave all other parameters to the default settings <br /><br /></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>It takes a few minutes to finish DESseq2. After the analysis is finished, have a look at the file, it should look like something this:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>GeneID  Base mean   log2(FC)	        StdErr	            Wald-Stats	        P-value	                P-adj
K15661  55.0733128361562    2.49892393773464	0.508475451930939	4.91454194739384	8.89902710599613e-07	0.00491938218419466
K15666  48.4532561772761    2.22805029428311	0.493391718412049	4.51578372951607	6.30830206230236e-06	0.00820667379798327
K10213  25.1966742274619    -2.45624670858868	0.550895251183889	-4.45864563782342	8.24791475064012e-06	0.00820667379798327
K03732  563.63634258472 -1.41961721984814	0.316992240900184	-4.47839737596338	7.52055094378821e-06	0.00820667379798327
K01792  112.923146882195    -1.26925892659617	0.285732234964578	-4.44212717810268	8.90738834802815e-06	0.00820667379798327
</code></pre></div></div>

<h1 id="fit-a-bum-model-a-mixture-model">Fit a BUM model (a mixture model)</h1>

<p>From a statistical point of view, p-values are uniformly distributed under null hypothesis; in other words,
under alternative hypothesis, the noise component (which holds under null hypothesis) will be adequately
modeled by a uniform distribution. With this knowledge, we can fit these p-values to a mixture model
(BUM model), as the figure below shows.</p>

<p><a href="../../images/transcriptomics_images/bum.jpeg" rel="noopener noreferrer"><img src="../../images/transcriptomics_images/bum.jpeg" alt="p-values are fitted to a mixture model. " width="1280" height="1138" loading="lazy" /></a></p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-why-do-we-need-a-mixture-model"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Why do we need a mixture model?</div>

  <p>To avoid making this tutorial sound like a math class, let’s focus on the philosophy of using a statistical model.
In the graph above, we have visualized the p-values; those bins have some shapes; by viewing the figure, it
is possible for us to cherrypick the data points (KO) with mesh eyes, but it must be tedious and boring, nobody wants to
spend a few days on doing this. In this situation, computers come into play; however, unlike humans, computers cannot understand the graph but
mathematical formulas, that’s why we need to fix the data into a statistical model (denoted by the red curve and blue line in the graph),
which computers can understand easily.</p>

</blockquote>

<p>Before fitting to BUM model in Galaxy, we need to prepare the input data for the tool
<strong>Fit a BUM model</strong>, that’s a file that only contains p-values.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-extract-p-values-from-deseq2-output"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: extract p-values from DESeq2 output</div>

  <ul>
    <li><strong>cut</strong> columns from a table <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c6</code></li>
        <li><em>“Delimited by”</em>: to <code class="language-plaintext highlighter-rouge">TAB</code></li>
        <li><em>“From”</em>: to the output of <strong>DESeq2</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>Now we can <strong>Fit a BUM model</strong>:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-fit-the-bum-model"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: fit the BUM model</div>

  <ul>
    <li><strong>Fit a BUM model</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Input file”</em>: the output of <strong>cut</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<h1 id="pinpoint-the-key-pathways-with-heinz">Pinpoint the key pathways with Heinz</h1>

<p>After getting the parameters of the BUM model from the last step, we will use Heinz to pinpoint
the optimal subnetwork. Then we could look for the key pathways relevant to the subnetwork.
Before we continue, let’s figure out what Heinz is actually doing.</p>

<p>Heinz is an algorithm for searching an optimal subnetwork from a bigger network. You may wonder what
the network is here. Through the previous steps, we have got a list of identities, that is a list
of gene IDs with p-values, which form the nodes  of ‘the bigger network’, the relations between the
nodes, that is the edges, needs to be obtained from a background network, which represents a pathway
relation databases, such as <a href="https://reactome.org/">Reactome</a> and <a href="https://string-db.org/">STRING</a>.
In this tutorial, we only use a small sample background network for demonstration purposes. The
background network is represented as edges in a TXT file where each line denotes an edge as follows:</p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ACTR1B	ACVR2B
ZSWIM9	FOXP3
LGALS4	PRKX
NPTX1	CIAO1
</code></pre></div></div>

<p>Upload this edge file (hereafter we call it edge file) into the Galaxy instance.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-upload-edge-file"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Upload edge file</div>

  <ol>
    <li><strong>Upload the edge file from Zenodo</strong>
      <ul>
        <li>Open the upload menu</li>
        <li>Choose <strong>Paste/Fetch data</strong> option</li>
        <li>Paste the URL into the box:
          <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/1344105/files/edge.txt
</code></pre></div>          </div>
        </li>
        <li>Click <strong>Start</strong></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h2 id="calculate-heinz-scores">Calculate Heinz scores</h2>

<p>As the first step, we need to calculate a Heinz score for each node, using the BUM model parameters
we obtained; meanwhile, we also need to specify an FDR value as input.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-is-an-fdr-value"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What is an FDR value?</div>

  <p>FDR is short for false discovery rate, which is a method of conceptualizing the rate of type I errors
in null hypothesis testing when conducting multiple comparisons, if you are interested, view the detail
in <a href="https://en.wikipedia.org/wiki/False_discovery_rate">Wikipedia</a>.</p>

</blockquote>

<p>For different datasets and problems, we probably need to pick up an FDR value separately. Here we set FDR to 0.11.</p>

<p>Similar to <strong>Fit a BUM model</strong>, we also need to prepare the input data for the tool
<strong>Calculate a Heinz score</strong>.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What is the requirement of the input data format for <strong>Calculate a Heinz score</strong>?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>In the user interface of the tool “Calculate a Heinz score”, we see that “A node file with p-values” is needed.
It should contain two columns delimited by tab, one is KO ID; the other, p-value.</p>

  </blockquote>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-extract-geneid-and-p-values-from-deseq2-output"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: extract geneID and p-values from DESeq2 output</div>

  <ul>
    <li><strong>cut</strong> columns from a table <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Cut columns”</em>: <code class="language-plaintext highlighter-rouge">c1,c6</code></li>
        <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">TAB</code></li>
        <li><em>“From”</em>: the output of <strong>DESeq2</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-calculate-heinz-scores"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: calculate Heinz scores</div>

  <ul>
    <li><strong>Calculate a Heinz score</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“A node file with p-values”</em>: the output of the previous <strong>cut</strong> step</li>
        <li><em>“FDR value”</em>: to 0.11</li>
        <li><em>“Choose your input type for BUM parameters”</em>: <code class="language-plaintext highlighter-rouge">The output file of BUM model</code></li>
        <li><em>“Output file of BUM model as input: lambda on the first line and alpha, the second”</em>: the output of <strong>Fit a BUM model</strong></li>
      </ul>
    </li>
  </ul>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-what-is-the-heinz-score"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: What is the Heinz score?</div>

  <p>To figure out this score without reading the formula, we can understand Heinz score in this way. FDR is involved in calculating a p-value threshold
and any KO whose p-value is below this threshold is regarded as significant, which means the Heinz score is positive (another calculation in the formula). If we pick a higher FDR value, then we will have a higher p-value threshold, and more KOs are probably regarded as significant. In this situation, we probably have many false positive (those regarded as significant are actually not) on the one hand; on the other hand, Heinz will deliver a bigger subnetwork, which might be exhausting to analyze. Therefore, we need to pick up an FDR value properly.</p>

</blockquote>

<h2 id="run-heinz-pinpoint-the-optimal-subnetwork">Run Heinz: pinpoint the optimal subnetwork</h2>

<p>After getting Heinz scores, let’s run Heinz program to find the optimal subnetwork from the background network which we mentioned earlier.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-pinpoint-the-optimal-subnetwork"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: pinpoint the optimal subnetwork</div>

  <ul>
    <li><strong>Identify optimal scoring subnetwork</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“File containing Heinz scores”</em>: the output of <strong>Calculate a Heinz score</strong></li>
        <li><em>“Edge file”</em>: the edge file uploaded</li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>It usually takes a few minutes to get the result, but mind you, for some tasks, it might take a few hours to get a result in practice.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-the-running-time-of-the-program"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: The running time of the program</div>

  <ul>
    <li>Graph problem is way more complicated than we thought.</li>
    <li>It might take a much longer time for some complicated datasets.</li>
    <li>We can use multiple CPUs to accelerate the computation (for now, this function is not available in Galaxy yet),
but, to use that, you can install Heinz directly via <a href="https://anaconda.org/bioconda/heinz">Bioconda</a> in a Linux environment.</li>
  </ul>
</blockquote>

<h2 id="visualize-the-output-visualize-the-optimal-subnetwork">Visualize the output: visualize the optimal subnetwork</h2>

<p>The result we got from the last step is not very human-readable, is it? It is a little painful to understand the Heinz result directly. Therefore we need to visualize the output by making it into graphs.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-visualize-the-optimal-subnetwork"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: visualize the optimal subnetwork</div>

  <ul>
    <li><strong>Visualize</strong> the optimal scoring subnetwork <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters
      <ul>
        <li><em>“Heinz output file”</em>: the output of <strong>Identify optimal scoring subnetwork</strong></li>
      </ul>
    </li>
  </ul>

</blockquote>

<p>In this tutorial, you probably get a similar graph to the following:</p>

<p><a href="../../images/transcriptomics_images/Heinz_visualisation.png" rel="noopener noreferrer"><img src="../../images/transcriptomics_images/Heinz_visualisation.png" alt="p-values are fitted to a mixture model. " width="860" height="1414" loading="lazy" /></a></p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>According to the figure above, Why are some shapes are round, others square?</p>

  <blockquote class="solution">
    <div class="box-title solution-title" id="solution-2"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-2" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Round shape corresponds to positive Heinz score; square shape corresponds to negative Heinz score.</p>

  </blockquote>
</blockquote>

<p>In addition to the tool we will use in Galaxy, you may consider using eXamine plugin in Cytoscape for a richer visualization, if we use the original data from the study <a href="https://academic.oup.com/bioinformatics/article/32/11/1678/2240171">May et al.</a>, perform all of these steps and visualize the result by eXamine, we can get a visualization as follows:</p>

<p><a href="../../images/transcriptomics_images/metaModules.png" rel="noopener noreferrer"><img src="../../images/transcriptomics_images/metaModules.png" alt="metaModules Result. " width="2250" height="1742" loading="lazy" /></a></p>

<p>If you are interested, you may try to make sense of the result yourself. To answer the question “which pathways are potentially contributing to dental caries?”, we need to find the key pathways relevant to the subnetwork. To achieve that, we probably need to combine information from other bioinformatics services and literature.</p>

<h1 id="save-the-history-into-a-workflow">Save the history into a workflow</h1>

<p>Let’s assume that these are the operations we always need for different datasets, to make life easier, we can save all of these operations into a workflow in Galaxy, which you can reuse for different datasets next time.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-extract-the-heinz-workflow"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Extract the Heinz workflow</div>

  <ol>
    <li>
      <p>Extract the workflow from your history</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-extracting-a-workflow-from-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-extracting-a-workflow-from-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Extracting a workflow from your history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> button.</p>     <p>This will make the creation of the workflow easier.</p>   </li>   <li>     <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract workflow</strong>.</p>     <p><img src="/training-material/topics/introduction/images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>     <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the workflow.</p>   </li>   <li>     <p>Replace the <strong>Workflow name</strong> to something more descriptive.</p>   </li>   <li>     <p><strong>Rename</strong> each workflow input in the boxes at the top of the second column.</p>   </li>   <li>     <p>If there are any steps that shouldn’t be included in the workflow, you can <strong>uncheck</strong> them in the first column of boxes.</p>   </li>   <li>     <p>Click on the <strong>Create Workflow</strong> button near the top.</p>     <p>You will get a message that the workflow was created.</p>   </li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Run the workflow on the dataset in this tutorial</p>
    </li>
  </ol>

</blockquote>

<p><a href="../../images/transcriptomics_images/heinz-workflow.png" rel="noopener noreferrer"><img src="../../images/transcriptomics_images/heinz-workflow.png" alt="Heinz workflow. " width="2630" height="866" loading="lazy" /></a></p>

<p>As a self practice, you ran the newly created workflow with the dataset in this tutorial. Do you feel the convenience of running a workflow directly?</p>

<p>Congrats! You have finished all the tools in Heinz workflow! You have successfully run each tool and understood how it works. In real research, running these tools are only part of the effort, we still need to invest a huge amount of intelligence in making sense of the results and converting them to knowledge, which is fraught with uncertainties and confounders, where perhaps luck will come into play. So good luck!</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Using Heinz do the network analysis based on the metatranscriptomics or transcriptomics data.</p>
</li>
                        
                        <li><p>Extracting the signals hidden in the p-value distribution of the differential expression analysis.</p>
</li>
                        
                        <li><p>Visualizing the pinpointed subnetwork helps us understand the result better</p>
</li>
                        
                        <li><p>Heinz helps narrow down the scope to find the key pathways</p>
</li>
                        
                        <li><p>Multiple CPUs will accelerate the computation especially for the complex datasets</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/transcriptomics/faqs/">FAQ page for the Transcriptomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                


                

                

                <h1>Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Network analysis with Heinz (Transcriptomics)" alt="Feedback form link">
                    <img src="/training-material/shared/images/feedback.png" alt="Preview of the google form" />
                </a>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Chao Zhang,  <b>Network analysis with Heinz (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html">https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{transcriptomics-network-analysis-with-heinz,
author = "Chao Zhang",
	title = "Network analysis with Heinz (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="agenda follow-up" id="admins-install-missing-tools">
                    <div class="box-title">Galaxy Administrators: Install the missing tools</div>
			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: deseq2
  owner: iuc
  revisions: d9e5cadc7f0b
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: heinz
  owner: iuc
  revisions: 2b80a2596064
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: heinz
  owner: iuc
  revisions: 2b80a2596064
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: heinz
  owner: iuc
  revisions: 2b80a2596064
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: heinz
  owner: iuc
  revisions: 5f589c91566e
  tool_panel_section_label: RNA Analysis
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		


                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-6" style="text-align: left">
				<p>
					The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
					provides researchers with online training materials, connects them with local trainers, and helps promoting FAIR and Open Science practices worldwide. All contributions are subject to the <a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Galaxy Project Code of Conduct</a>.
				</p>
				<p>
				The GTN infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
				</p>
				<p>
				Revision <a href="https://github.com/galaxyproject/training-material/commit/9de96f24bf701758b2f5e169c4da60878ecefe7d">9de96f2</a>, built with Jekyll (4.3.2 | production)
				</p>
			</div>
			<div class="col-sm-6" style="text-align: right">
				
				<p>
					Resource <i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00293">https://gxy.io/GTN:T00293</a>
				</p>
				
				<p>
					<i>This Material</i>:
					
					is licensed under
					<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
						Creative Commons Attribution 4.0 International License
					</a>
				</p>
				<p>
					<a href="https://github.com/galaxyproject/training-material/edit/main/topics/transcriptomics/tutorials/network-analysis-with-heinz/tutorial.md" title="Edit on GitHub">
					<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
					</a>
				</p>
			</div>
		</div>
	</div>
</footer>


        <script src='/training-material/assets/js/bundle.03aefefa.js?v=1701303507'></script>
        
    </body>
</html>