---
layout: event
title: "Mycobacterium tuberculosis complex NGS made easy"

description: |
  These series of webinars and tutorials aim at improving basic and
  applied knowledge associated with next-generation sequencing (NGS)
  technologies and their applications in the field of Tuberculosis (TB).

cover-image: /events/images/TBCAPT_NGS_Training_banner.jpg
cover-image-alt: banner for the course

date_start: 2024-06-10
date_end: 2024-06-14 # optional, if event is more than one day
tags: [one-health, tuberculosis, NGS, Galaxy, training]

contributions:
  organisers: # GTN contributors or funders, must be defined in CONTRIBUTORS.yaml
  - dbrites
  - cstritt
  - andreacabibbe
  - arashghodousi
  - pvanheus
  - lilianarutaihwa
  - andrea-spitaleri
  - GaloGS

  instructors:
  - cstritt
  - dbrites
  - andreacabibbe
  - arashghodousi
  - Allen13x
  - pvanheus
  - lilianarutaihwa
  - linzyelton
  - bethlehem-adnew

  funding:
  - swiss-tph
  - san-raffaele-uni
  - sanbi
  - pangens
  - irccs
  - linq
  - find
  - panacea
  - edctp
  - edctp-eu
  - eu


registration:
  link:
  deadline: 2024-01-01

contact_email: d.brites@swisstph.ch

location:
  name: Online

cost: free
audience:  Open for all, but target audience is clinicians and researchers using MTBC sequencing data.

async: false
mode: online

infrastructure:
  tiaas: true
  servers:
    - server: https://usegalaxy.eu
      name: Galaxy EU
      tiaas_link: https://usegalaxy.eu/join-training/mtb-ngs-2024

  support:
    platform: Slack
    channel: event-mtb-ngs

program:
  - section: "Start the Course"
    description: |
      There were unusually high rates of TB cases in your country this year. To characterize the underlying
      bacterial strains driving the epidemic, isolates have been sent for whole-genome sequencing.  Doctors and public
      health authorities request information in order to take decisions. In this course it will be demonstrated how
      you would make use of NGS to answer several questions relevant for patient and public health system management
      such as:
        - Are there cases of drug resistant bacteria?
        - Is there transmission of drug resistance?
        - Is there evidence of de novo emergence of resistance?
        - Are there multiple infections per patient?
        - Do we have on-going transmission?

      We hope that at the end of the different training sessions you can answer this question on your own and can apply what you have learnt to your own data!

  - section: "Day 1: Overview of NGS technologies & TB specific NGS solutions"
    custom: true
    description: |
      Please watch the following pre-recorded webinars to know more about how NGS is being applied to TB and to know about what the WHO recomendations.
    async: true
    tutorials:
    - type: custom
      name: "**Icebreaker**"
      description: |
        **Introduce yourself on Slack and tell us one fun fact about yourself!**

        Post your answer to the [#event-mtb-ngs](https://gtnsmrgsbord.slack.com/archives/event-mtb-ngs)

        Please feel free to respond to each other here, this channel is for troubleshooting but also for getting to know each other! :)

  - section:
    custom: true
    tutorials:
    - type: custom
      name: "**Welcome & set up live session**"
      time: "**10:30-11:00** Central European Time"
      description: |
        We present the team and give general guidance on the training content and the set up.
        Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09)


    - type: custom
      name: "**Webinar**: Overview of NGS technologies & TB specific NGS solutions"
      description: |
        Introduction to different sequencing technologiesand what applies best to what kind of problem.

        [Watch webinar](https://www.youtube.com/watch?v=BcfBjCJRmUs) (1h)

        **Assessement**: Share your thoughts with us
        [here](https://docs.google.com/document/d/1Rd72luHh1Z33O-Le-FpEuPWi7X5yaBPTuVyB-oCSMQw/edit)

    - type: custom
      name: "**Webinar**: Implementation of NGS for TB- WHO documents and other considerations"
      description: |
        Summary of the recommendations and considerations available from the WHO documents on the use of NGS for TB.

        [Watch webinar](https://youtu.be/eSWnPange8w) (1h)

        **Assessement**: Share your thoughts with us
        [here](https://docs.google.com/document/d/1M6lsXVHBipxJO84gRO0Lx-xGTrkDYOQmzaDuM19rAwY/edit#heading=h.bs9dcv9mioqd)

    - type: custom
      name: "**Webinar**: Updated WHO catalogue version 2"
      description: |
       Description of main changes introduced in the second version of the WHO catalogue on mutations associated with drug resistance released in November 2023.

        [Watch webinar](https://www.youtube.com/watch?v=vxggV1wMLdk) (6 min)

    - type: custom
      name: "**Webinar**: WHO recommendations on targeted NGS (tNGS)"
      description: |
        Summary of the recommendations and considerations available from the WHO documents on the use of **targeted** NGS for TB

        [Watch webinar](https://www.youtube.com/watch?v=UdolTwTnsz8) (15 min)

  - section:
    custom: true
    tutorials:
    - type: custom
      name: "**Meet the experts: Q&A session**"
      time: "**14:00-16:00** Central European Time"
      description: |
        We would like to hear your opinion on the following questions
        and promote discussion in this Q&A session. Please let us know your thoughts on one or more of the following
        questions in the shared notes (link below);

        1. How do I choose the right sequencing technology for my samples?
        2. What is needed for NGS?
        3. Why is NGS better for drug resistance and outbreak analysis?

        Also take the chance to ask us or write down in the shared notes other questions you might have.

        - Zoom: [link](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09)
        - Shared Notes: [link](https://docs.google.com/document/d/1VtGho4Zz0qFp5tzqD8rl7NU83xtYb4tCkLDitC3DrhQ/edit#heading=h.e69aqv1nq3gw)

        *The completion of the assessments is a requirement for the certificate of attendance.*

  - section:
    description: |
        Optional webinars: These are produced outside this training by [tbnet](https://www.tbnet.eu/) and [Stop TB Partnership](https://www.stoptb.org/) and [FIND](https://www.finddx.org/), but might be of interest to you.

        [Targeted next generation sequencing for the detection of drug resistant TB: intended use, challenges and research priorities](https://www.tbnet.eu/webinars)

        [Implementing tNGS for detection of drug-resistant TB in high burden countries](https://www.stoptb.org/meetings-events/implementing-tngs-detection-of-drug-resistant-tb-high-burden-countries)


  - section: "Day 2: Mapping and variant calling"
    custom: true
    async: true
    description: |
      The 20 strains isolated in your country have been sequenced with
      Illumina technology to obtain whole-genome sequences. In this part
      of the workshop you will learn how to analyse those sequences.

      In a typical bioinformatic pipeline you would store your sequences
      in a computer server where all necessary software would be
      installed. This would be a server running the operating system
      LINUX, which is the most efficient way to run bioinformatics
      pipelines (more on this on Day 4). You will be running your
      analysis in a LINUX server from Galaxy, but instead of writing
      directly commands to execute operations in the server, you will be
      executing operations through a Galaxy graphical interface. This
      allows you to have access to a LINUX server and to run workflows
      without knowing LINUX. Importantly, for training purposes it also
      allows you to dedicate more attention in trying to understand what
      is being done in each of the steps without having to understand the
      programing behind. This being said, working directly on a LINUX
      cluster provides you always more flexibility, but if you donâ€™t have
      access to one, Galaxy is a very good alternative for data analysis.

    tutorials:
      - name: "**Icebreaker**"
        type: custom
        description: |
           Come say Hi in Slack! Let us know you are joining today and are getting started!
           Today's icebreaker question:
           - **What is your favorite dish (food or drink)? Bonus points for recipes!**

           Post your answers in [#event-mtb-ngs](https://gtnsmrgsbord.slack.com/archives/C035Q01UA2F) on Slack!

           Please feel free to respond to each other here, this channel is for socializing and getting to know each other! :)


  - section: "Session 1: Learning Galaxy"
    description: |
        You will need to understand how to use Galaxy to run all the hands-
        on tutorials and therefore is highly recommended that you follow
        the next webinar and hands-on on Galaxy. The good thing about this
        is that once you know how it works, you can use it to run your own
        analysis with your own data.

    subsection: true
    async: true
    tutorials:
      - topic: introduction
        name: galaxy-intro-short
      - topic: introduction
        name: galaxy-intro-101

  - section: "Session 2: Mapping and Variant calling of short MTBC reads "
    subsection: true
    custom: true
    async: true
    description: |
      Let us imagine that you have received the sequences of the 20 strains, the first step is to assess the quality of sequencing. Once we are sure that the sequencing worked well, we typically compare our sequencing results to a reference genome (re-sequencing approach) by using a bioinformatics procedure usually called mapping. After, we will identify the genomic variants in our sequences with respect to the reference genome, a bioinformatics procedure called, variant calling. Once we are certain of the variants we have identified, usually we are interested in determining to what genes they belong, to what pathways, or for instance if they are likely to disrupt protein function. This procedure is called annotation. Once we have gone through each of these steps we are ready to analyse drug resistant patterns, draw phylogenetic relationships or identify clusters of transmission of M. tuberculosis.

      You are now ready for performing bioinformatic analysis in Galaxy. Before we start we would like you to watch a short video on how Illumina sequencing works. Following that video we have prepared a webinar on mapping and variant calling of Illumina applied to MTBC. After watching it you will be hopefully able to know; how a reference genome is chosen, why we typically ignore some regions of the MTBC genomes or what is the difference between a fixed and a variable SNP and why do we care about it (among other things).
    tutorials:
      - name: "**Video:** Illumina sequencing"
        type: custom
        description: Please watch [this 5-minute video](https://youtu.be/fCd6B5HRaZ8) about the principles behind Illumina sequencing
      - name: "**Webinar**: Mapping and Variant calling"
        type: custom
        description: |
          Main bioinformatics steps involved in mapping and variant calling from Illumina short reads applied to MTBC.

          [Watch Webinar](https://youtu.be/38GUBKwWXv8) (45 minutes)

          **Assessment**: Share your thoughts with us [here](https://docs.google.com/document/d/18y0PScnbA0GuLKjITzIr01wyLVWH6RYPAE5RWkwlfNA/edit#heading=h.bs9dcv9mioqd)


          *The completion of the assessments is a requirement for the certificate of attendance.*

  - section:
    custom: true
    tutorials:
      - type: custom
        name: "**Meet the experts: Q&A session**"
        time: "**11:30-12:30**  Central European Time"
        description: |
          Meet the experts! Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).

  - section:
    async: true
    tutorials:
      - name: tb-variant-analysis
        topic: variant-analysis
  - section:
    custom: true
    tutorials:
      - type: custom
        name: "**Meet the experts: Q&A session**"
        time: |
          **16:30-17:30**
          Central European Time
        description: |
          Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).




  - section: "Day 3: Evolutionary epidemiology: using phylogenetics to understand DR emergence and Mtb transmission"
    description: |
      We are ready to analyse drug resistant patterns, draw phylogenetic relationships or
      identify recent transmission among the isolates we have sampled in our population. Before
      delving into the analysis of the genomes we would like to share with you some notions
      important to the inference of direct transmission and to the interpretation of drug
      resistant patterns.
  - section: "Evolutionary epidemiology: using phylogenetics to understand DR emergence and Mtb transmission"
    subsection: true
    async: true
    custom: true
    tutorials:
      - name: "**Icebreaker**"
        type: custom
        description: |
           Come say Hi in Slack! Let us know you are joining today and are getting started!
           Today's icebreaker question:
           - **What is the coolest, most mind blowing fact (nature/people/animal etc.) you know?**

           Post your answers in [#event-mtb-ngs](https://gtnsmrgsbord.slack.com/archives/C035Q01UA2F) on Slack!

           Please feel free to respond to each other here, this channel is for socializing and getting to know each other! :)

      - name: "**Webinar**: Drug resistance prediction"
        type: custom
        description: |
          Principles of drug resistance detection from genomic data

          [Watch Webinar](https://youtu.be/dL06GTNFRgc) (20 minutes)

          **Assessment**: Share your thoughts with us [here](https://docs.google.com/document/d/1zkZaq-cCqRaibep_npiiq7gvpQtlkOozixVC5KOc49I/edit)

          *The completion of the assessments is a requirement for the certificate of attendance.*

      - name: '**Webinar**: "Phylogenetic" mutations'
        type: custom
        description: |
          This video will introduce one special type of mutations to take into account when studying
          drug resistance patterns

          [Watch Webinar](https://youtu.be/i-JK-UdMscI) (15 minutes)

          **Assessment**: Share your thoughts with us [here](https://docs.google.com/document/d/167DXUopX_EtuMDXpvafsRmB49oJqjjsNIACvFU_bPYc/edit)

          *The completion of the assessments is a requirement for the certificate of attendance.*

      - name: "**Webinar:** The concept of clustering"
        type: custom
        description: |
          Main aspects of clustering analysis to infer transmission in MTBC

          [Watch Webinar](https://youtu.be/l4cPUECJ7VU) (15 minutes)


      - name: "**Webinar**: Genetic distance thresholds"
        type: custom
        description: |
          Clustering as an approximation to infer transmission

          [Watch Webinar](https://youtu.be/kKNgmpy1N94) (15 minutes)

          **Assessement**: Share your thoughts with us [here](https://docs.google.com/document/d/1Q4mw-y3FZrGvmz3VNQrLkLslsi5QoTgDxwhAYYYdvjI/edit#heading=h.bs9dcv9mioqd)

          *The completion of the assessment is a requirement for the certificate of attendance.*

  - section:
    async: true
    tutorials:
      - name: mtb_transmission
        topic: evolution

  - section:
    custom: true
    tutorials:
      - type: custom
        name: "**Meet the experts: Q&A session**"
        time: "**11:30-12:30** Central European time"
        description: |
          Discussion with the experts! Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).

      - type: custom
        name: "**Hands-on**:Introduction to phylogenetics"
        description: |
           Recommended tutorial from [EMBL-EBI](https://www.ebi.ac.uk/training/online/courses/introduction-to-phylogenetics/) for those who want to learn more about phylogenetics.

  - section:
    async: true
    tutorials:
      - name:  mtb_phylogeny
        topic: evolution

  - section:
    custom: true
    async: true
    tutorials:
      - type: custom
        name: "Check what you have learnt!"
        description: |
          We hope that you are enjoying the training, and that many things that you are learning will be useful for your research! We would like you to answer some questions, so both you and us, can assess whether the main concepts covered in the hands-on tutorials on Mtb NGS data analysis were understood. For that please follow the link bellow. If you are interested in knowing what we think about these questions join us on Day 5

          [**Assessment**](https://docs.google.com/document/d/1mT6z84iA2-E2D834lH1B6fJrWwfXYBKDkPrKCFAdwqk/edit#)

          *The completion of the assessment is a requirement for the certificate of attendance.*

  - section:
    custom: true
    tutorials:
      - type: custom
        name: "**Meet the experts: Q&A session**"
        time: "**16:30-17:30** Central European Time"
        description: |
          Discussion with the experts! Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).



  - section: "Day 4: Webtools dedicated to MTBC bioinformatics & Be a bioinformatician in the jungle (optional)"
  - section: "Session 1: Webtools dedicated to MTBC bioinformatics"
    subsection: true
    custom: true
    async: true
    description: |
      The use of whole-genome sequencing (WGS) for antibiotic resistance prediction and routine
      typing of bacterial isolates has increased substantially in recent years. To date a
      multitude of solutions for analyzing WGS data of the Mycobacterium tuberculosis complex
      (MTBC) data have been developed. In the first part of the 4th day of this workshop, we
      introduce some freely available webtools and open source pipelines designed to analyze
      MTBC sequence data and weâ€™ll provide some examples of how these tools work and how to
      interpret the results.
    tutorials:
      - name: "**Icebreaker**"
        type: custom
        description: |
           Come say Hi in Slack! Let us know you are joining today and are getting started!
           Today's icebreaker question:

           - **"What is a book, film, tv show or game that you've enjoyed recently?"**

           Post your answers in [#event-mtb-ngs](https://gtnsmrgsbord.slack.com/archives/C035Q01UA2F) on Slack!

           Please feel free to respond to each other here, this channel is for socializing and getting to know each other! :)

      - name: "**Webinar**: Use of web-tools & software for MTBC sequence analysis"
        type: custom
        description: |
          Introduction to most common web tools for fast identification of bacterial species from raw sequencing reads.

          [Watch Webinar](https://www.youtube.com/watch?v=FzXfea6q89c&t=2246s) (50 minutes)

          **Assessment**: Share your thoughts with us [here](https://docs.google.com/document/d/1QRnuHYXWGQnvf8fX5v6EIaynJh6AxkoOktS3VgL0Y8Q/edit)

      - name: "**Webinar:** Brief overview of MTBseq and MAGMA pipelines"
        type: custom
        description: |
          MAGMA and MTBseq pipelines are automated pipelines for mapping, variant calling and detection of resistance mediating and phylogenetic variants from whole genome sequence data of MTBC.
          This webinar presents the most important features between them.

          [Watch Webinar](https://www.youtube.com/watch?v=8uuaToa40h4) (10 min)

  - section:
    custom: true
    tutorials:
      - type: custom
        name: "**Meet the experts: Q&A session**"
        time: "**11:30-12:30** Central European Time"
        description: |
          Discussion with the experts! Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).


  - section:  "Session 2: Be a bioinformatician in the jungle (optional)"
    subsection: true
    custom: true
    async: true
    description: |
      On Day 2 and 3 you have learned how you could use galaxy for analysing your own data.
      Establishing your own workflows in galaxy would allow you combining different tools and
      build your own pipeline without having to know how to program. If you are not so
      interested in having your own pipeline, webtools for WGS analysis can be very useful, as
      we have shown in the previous session.

      However, in the last part of the training we would like to convey to you what it would
      take if would want to run Linux via the command line.  The Linux operating system will be
      introduced, how to perform basic tasks using the Unix shell and how to install and run
      pipelines on the command line. You will learn the power of the Unix shell in performing
      complex and powerful tasks, often with just a few keystrokes or lines of code. In fact,
      Unix shell helps users automate repetitive tasks and easily combine smaller tasks into
      larger, more powerful workflows (i.e. pipelines). Use of the shell is fundamental to a
      wide range of advanced computing tasks, including high-performance computing. These
      webinars will introduce you to this powerful tool. Which approach to choose, Galaxy
      workflows, Webtools or native Linux depends on your needs, your interests and what
      computer resources you have available.

    tutorials:
      - name: "**Webinar**: Introduction to Linux"
        type: custom
        description: |
          Introduction to Linux OS: installation and usage

          [Watch Webinar](https://youtu.be/N8rW07ByQOE) (35 minutes)

      - name: "**Webinar**: How to run programs (Python, Docker, Nextflow)"
        type: custom
        description: |
          Learning how to install and use programs to analyze data

          [Watch Webinar](https://youtu.be/feYweAQNGS8) (35 minutes)


      - name: "**Webinar**: Demo on how to run the Linux command line"
        type: custom
        description: |
          Demo video on how to use the shell commands

          [Watch Webinar](https://youtu.be/7V7LCjUEg1c) (20 minutes)

      - name: "**Hands-on**: The Unix Shell"
        type: custom
        description: |
          Recommended tutorial from software carpentries to those wanting to learn Linux.

          [View Carpentries Tutorial](https://swcarpentry.github.io/shell-novice/) (4 hours)

  - section:
    custom: true
    tutorials:
      - type: custom
        name: "**Meet the experts: Q&A session**"
        time: "**16:30-17:30** Central European time"
        description: |
          Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).

          If you have questions about web tools or what is the best way to become a bioinformatian, or if there are aspects of the webinars and tutorial that you would like to discuss in with the experts, please join the Q&A session. Fill free to write down in the [shared notes](https://docs.google.com/document/d/1FbjtNZOkND1ziRPr7KCumqryeQJ76ZZx-ZAgjydCZho/edit) those questions as well as that can help the experts to struture the discussion.

  - section: "Day 5: Live Discussion"
    description: |
      Today all experts will  be available to answer your questions and discuss any
      of the tutorials, webinars, or questions related to your own data. Meet with us at the
      zoom link!
    tutorials:
      - type: custom
        name: "**Meet the experts:Q&A session**"
        time: "**10:00-12:30** Central European Time"
        description: |
          Discussion with the experts! Join us on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09).



---

![banner for course]({% link events/images/TBCAPT_NGS_Training_banner.jpg %})

These series of webinars and tutorials aim at improving basic and applied knowledge associated with next-generation sequencing (NGS) technologies and their applications in the field of Tuberculosis (TB).

The trainings will introduce scalable and reproducible data analysis with Galaxy of *Mycobacterium tuberculosis* complex (MTBC) genomes.

A series of pre-recorded sessions and hands-on tutorials will show:
  - How to differentiate sequencing technologies and which ones are most commonly applied in TB and how **(Day 1)**
  - How NGS can be implement into TB laboratories **(Day 1)**
  - How to do mapping and variant calling **(Day 2)**
  - How to detect drug resistance conferring mutations, build phylogenetic trees and infer tuberculosis transmission **(Day 3)**
  - How to use different web tools dedicated to targeted analysis and what it takes to do bioinformatics **(Day 4)**.

After the trainings all participants are expected to:
  - Describe how NGS is being used in TB research and clinical pratice
  - Extract MTBC genomic variants from short sequencing reads
  - Identify drug resistant mutations
  - Identify genetic relationships and interpret a phylogenetic trees
  - Use web applications designed for M. tuberculosis

## Communication

- Chat on Slack channel [#event-mtb-ngs](https://gtnsmrgsbord.slack.com/?redir=%2Farchives%2FC01MNM98X9V%3Fname%3DC01MNM98X9V) (see "Setup" for instructions)
- Scheduled Questions & Answer sessions (Q&A) on [Zoom](https://swisstph.zoom.us/j/88614333103?pwd=OVVjeG04a2lVMXB0THUxMXoxNUNkdz09)
- Check your time correspondence with [this time zone converter](https://time.is/compare)

A Slack channel will allow participants to communicate in real-time via chat. The channel should be used to troubleshoot and to get to know each other.
There will be scheduled zoom sessions (Q&A) every day where you can meet the experts, discuss and ask questions (check the program to know the time and be aware of the difference in time zone).
If you are in Asia you can attend the zoom sessions in the morning, if you are in America you can attend the sessions in the afternoon, no need to be there all the time.

## Certificates

Requirements for acquiring a certificate:
- Being registered in the training
- Join us on Slack! (see "Setup" for instructions)
- Answer the assessments of days 1,2 and 3
- Fill the [Feedback Survey](https://forms.gle/6Xn8WudBvr3NMC3W7	)
- As you finish the hands-on tutorials share your Galaxy history with us. See [how to do this here](https://training.galaxyproject.org/training-material/faqs/galaxy/histories_sharing.html) and copy the url link in [this table](https://docs.google.com/document/d/1K2k_yzk14VIHjsedfHaUFJnLgq6JTweG2tqxmIKUM2M/edit?usp=sharing)

## Start the Course

On the day of the course, just go to the **Program** tab and follow the instructions! Make sure to follow the **Setup** instructions before starting!

<button id="setup-button" class="btn btn-info" onClick="$('#setup-tab').tab('show');">View Setup Instructions</button>

<button id="setup-button" class="btn btn-info" onClick="$('#program-tab').tab('show');">Start the course!</button>

