<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Predicting EI+ mass spectra with QCxMS / Predicting EI+ mass spectra with QCxMS / Metabolomics</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@43bed0043c7b744f5a6074fbdafe6dd7d31c4834",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Predicting EI+ mass spectra with QCxMS">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2024-10-01 20:12:43 +0000">
<meta name="DC.creator" content="Julia Jakiela">
<meta name="DC.creator" content="Helge Hecht"><meta name="description" content="Training material to analyse Mass spectrometry data in Galaxy: Metabolomics (LCMS, FIAMS, GCMS, NMR) and imaging.">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Metabolomics / Predicting EI+ mass spectra with QCxMS / Hands-on: Predicting EI+ mass spectra with QCxMS">
        <meta property="og:description" content="Training material to analyse Mass spectrometry data in Galaxy: Metabolomics (LCMS, FIAMS, GCMS, NMR) and imaging.">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/metabolomics/tutorials/qcxms-predictions/tutorial.png">
	<script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": [
    {
      "@type": "CreativeWork",
      "name": "Galaxy Training: A Powerful Framework for Teaching!",
      "url": "https://doi.org/10.1371/journal.pcbi.1010752"
    },
    {
      "@type": "CreativeWork",
      "name": "Community-Driven Data Analysis Training for Biology",
      "url": "https://doi.org/10.1016/j.cels.2018.05.012"
    }
  ],
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Predicting EI+ mass spectra with QCxMS",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2024-10-01 20:12:43 +0000",
  "datePublished": "2024-10-01 20:12:43 +0000",
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "funder": [

  ],
  "funding": [

  ],
  "identifier": "https://gxy.io/GTN:T00458",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Metabolomics",
    "description": "Training material to analyse Mass spectrometry data in Galaxy: Metabolomics (LCMS, FIAMS, GCMS, NMR) and imaging.",
    "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/"
  },
  "abstract": "Mass spectrometry (MS) is a powerful analytical technique used in many fields, including proteomics, metabolomics, drug discovery and many more areas relying on compound identifications. Even though nowadays MS is a standard and popular method, there are many compounds which lack experimental spectra. In those cases, predicting mass spectra from the chemical structure can reveal useful information, help in compound identification and expand the spectral databases, improving the accuracy and efficiency of database search. [,]. ",
  "learningResourceType": "e-learning",
  "name": "Predicting EI+ mass spectra with QCxMS",
  "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html",
  "version": 1,
  "timeRequired": "PT1H",
  "teaches": "- To prepare structure-data format (SDF) files for further operations analysis, starting from chemical structure descriptors in simplified molecular-input line-entry system (SMILES) format.\n- To generate 3D conformers and optimise them using semi-empirical quantum mechanical (SQM) methods.\n- To produce simulated mass spectra for a given molecule in MSP (text based) format.",
  "keywords": [
    "Metabolomics"
  ],
  "description": "## Abstract\n\nMass spectrometry (MS) is a powerful analytical technique used in many fields, including proteomics, metabolomics, drug discovery and many more areas relying on compound identifications. Even though nowadays MS is a standard and popular method, there are many compounds which lack experimental spectra. In those cases, predicting mass spectra from the chemical structure can reveal useful information, help in compound identification and expand the spectral databases, improving the accuracy and efficiency of database search. [,]. \n\n\n## About This Material\n\nThis is a Hands-on Tutorial from the GTN which is usable either for individual self-study, or as a teaching material in a classroom.\n\n\n## Questions this  will address\n\n - Can I predict QC-based mass spectra starting from SMILES?\n - How can I run those computationally heavy predictions?\n - Can I take into account different conformers?\n\n\n## Learning Objectives\n\n- To prepare structure-data format (SDF) files for further operations analysis, starting from chemical structure descriptors in simplified molecular-input line-entry system (SMILES) format.\n- To generate 3D conformers and optimise them using semi-empirical quantum mechanical (SQM) methods.\n- To produce simulated mass spectra for a given molecule in MSP (text based) format.\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/slides.html",
      "name": "Mass spectrometry: LC-MS preprocessing - advanced",
      "description": "Slides for 'Mass spectrometry: LC-MS preprocessing - advanced' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html",
      "name": "Mass spectrometry: LC-MS preprocessing with XCMS",
      "description": "Hands-on for 'Mass spectrometry: LC-MS preprocessing with XCMS' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html",
      "name": "Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)",
      "description": "Hands-on for 'Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)' tutorial",
      "learningResourceType": "e-learning",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "name": "Julia Jakiela",
      "image": "https://avatars.githubusercontent.com/wee-snufkin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0001-2017-8805",
      "orcid": "https://orcid.org/0009-0001-2017-8805"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/hechth/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/hechth/",
      "name": "Helge Hecht",
      "image": "https://avatars.githubusercontent.com/hechth",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0000-0001-6744-996X",
      "orcid": "https://orcid.org/0000-0001-6744-996X"
    }
  ],
  "contributor": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/shiltemann/",
      "name": "Saskia Hiltemann",
      "image": "https://avatars.githubusercontent.com/shiltemann",
      "description": "Researcher at Erasmus Medical Center",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/nfdi4plants/",
          "name": "DataPLANT (NFDI4Plants)",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://nfdi4plants.org"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0003-3803-468X",
      "orcid": "https://orcid.org/0000-0003-3803-468X"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/foellmelanie/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/foellmelanie/",
      "name": "Melanie Föll",
      "image": "https://avatars.githubusercontent.com/foellmelanie",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-1887-7543",
      "orcid": "https://orcid.org/0000-0002-1887-7543"
    },
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wee-snufkin/",
      "name": "Julia Jakiela",
      "image": "https://avatars.githubusercontent.com/wee-snufkin",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        }
      ],
      "identifier": "https://orcid.org/0009-0001-2017-8805",
      "orcid": "https://orcid.org/0009-0001-2017-8805"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Metabolomics",
      "description": "Training material to analyse Mass spectrometry data in Galaxy: Metabolomics (LCMS, FIAMS, GCMS, NMR) and imaging.",
      "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3172",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3172",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3172"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/qcxms-predictions/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://zenodo.org/record/13327051",
      "name": "Associated Training Datasets"
    }
  ]
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/metabolomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Metabolomics
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/metabolomics/faqs/" title="Check our FAQs for the Metabolomics topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/metabolomics/tutorials/qcxms-predictions/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-metabolomics">
    <h1 data-toc-skip>Predicting EI+ mass spectra with QCxMS</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		
<table class="contributions">
	
	<tr>
		<td><abbr title="These people wrote the bulk of the tutorial, they may have done the analysis, built the workflow, and wrote the text themselves.">Author(s)</abbr></td>
		<td>
			
  <a href="/training-material/hall-of-fame/wee-snufkin/" class="contributor-badge contributor-wee-snufkin"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wee-snufkin?s=36" alt="Julia Jakiela avatar" width="36" class="avatar" />
    Julia Jakiela</a>
  
  <a href="/training-material/hall-of-fame/hechth/" class="contributor-badge contributor-hechth"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/hechth?s=36" alt="Helge Hecht avatar" width="36" class="avatar" />
    Helge Hecht</a>
  
		</td>
	</tr>
	

	

	

	

	

	

	

	
	<tr class="reviewers">
		<td><abbr title="These people reviewed this material for accuracy and correctness">Reviewers</abbr></td>
		<td>
			
  <a href="/training-material/hall-of-fame/shiltemann/" class="contributor-badge contributor-shiltemann">
    <img src="https://avatars.githubusercontent.com/shiltemann?s=36" alt="Saskia Hiltemann avatar" width="36" class="avatar" />
  </a>
  
  <a href="/training-material/hall-of-fame/foellmelanie/" class="contributor-badge contributor-foellmelanie">
    <img src="https://avatars.githubusercontent.com/foellmelanie?s=36" alt="Melanie Föll avatar" width="36" class="avatar" />
  </a>
  
  <a href="/training-material/hall-of-fame/wee-snufkin/" class="contributor-badge contributor-wee-snufkin">
    <img src="https://avatars.githubusercontent.com/wee-snufkin?s=36" alt="Julia Jakiela avatar" width="36" class="avatar" />
  </a>
  </td>
	</tr>
	

	

	

</table>


	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>Can I predict QC-based mass spectra starting from SMILES?</p>
</li>
        
        <li><p>How can I run those computationally heavy predictions?</p>
</li>
        
        <li><p>Can I take into account different conformers?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>To prepare structure-data format (SDF) files for further operations analysis, starting from chemical structure descriptors in simplified molecular-input line-entry system (SMILES) format.</p>
</li>
        
        <li><p>To generate 3D conformers and optimise them using semi-empirical quantum mechanical (SQM) methods.</p>
</li>
        
        <li><p>To produce simulated mass spectra for a given molecule in MSP (text based) format.</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    


        
    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mass spectrometry: LC-MS preprocessing with XCMS</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/metabolomics/tutorials/lcms-preprocessing/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mass spectrometry: LC-MS preprocessing with XCMS</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/metabolomics/tutorials/gc_ms_with_xcms/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mass spectrometry: GC-MS data processing (with XCMS, RAMClustR, RIAssigner, and matchms)</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                
            
        
    


        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 1 hour</div>
        

        
        <div><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i> Level: </strong>
        


 Introductory <span class="visually-hidden">Introductory</span>
<span class="level introductory" title="Introductory Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i> <i class="fas fa-graduation-cap" aria-hidden="true"></i>
</span>

</div>
        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            
                <li class="btn btn-default"><a href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/slides.html" title="Slides for this tutorial">
                    <i class="fab fa-slideshare" aria-hidden="true"></i> Slides
                </a></li>
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://zenodo.org/record/13327051">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/workflows/" title="Predicting EI+ mass spectra with QCxMS workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/metabolomics/tutorials/qcxms-predictions/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au
		</a>
	</li>
    
    

    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Oct 1, 2024 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Oct 1, 2024 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00458">https://gxy.io/GTN:T00458</a> </div>
        

	
	

	
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 1 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>Mass spectrometry (MS) is a powerful analytical technique used in many fields, including proteomics, metabolomics, drug discovery and many more areas relying on compound identifications. Even though nowadays MS is a standard and popular method, there are many compounds which lack experimental spectra. In those cases, predicting mass spectra from the chemical structure can reveal useful information, help in compound identification and expand the spectral databases, improving the accuracy and efficiency of database search. [<span class="citation"><a href="#Zhu2023">Zhu and Jonas 2023</a></span>, <span class="citation"><a href="#Allen2016Computational">Allen <i>et al.</i> 2016</a></span>]. 
There have been several methods developed to predict mass spectra, which can be classified as either first-principles physical-based simulation or data-driven statistical methods [<span class="citation"><a href="#Zhu2023">Zhu and Jonas 2023</a></span>]. To the first category we can assign purely statistical theories (quasi-equilibrium theory (QET) or Rice–Ramsperger–Kassel–Marcus (RRKM) theories) [<span class="citation"><a href="#Vetter1994">Vetter 1994</a></span>], as well as QCxMS [<span class="citation"><a href="#Koopman2021">Koopman and Grimme 2021</a></span>] and semiempirical GFNn-xTB [<span class="citation"><a href="#Koopman2019">Koopman and Grimme 2019</a></span>] which use Born–Oppenheimer molecular dynamics (MD) combined with fragmentation pathways. Data-driven statistical methods - forming the second category - reach back to 1960s when the DENDRAL project (using rule-based heuristic programming) was started by early artificial intelligence (AI) scientists [<span class="citation"><a href="#Lindsay1980ApplicationsOA">Lindsay <i>et al.</i> 1980</a></span>]. More recently, CFM-ID has been introduced [<span class="citation"><a href="#Allen2014">Allen <i>et al.</i> 2014</a></span>, <span class="citation"><a href="#Allen2014metabolomics">Allen <i>et al.</i> 2014</a></span>, <span class="citation"><a href="#Allen2016Computational">Allen <i>et al.</i> 2016</a></span>, <span class="citation"><a href="#DjoumbouFeunang2019">Djoumbou-Feunang <i>et al.</i> 2019</a></span>, <span class="citation"><a href="#Wang2021">Wang <i>et al.</i> 2020</a></span>], which uses rule-based fragmentation and employs machine learning methods. Current advancements in machine learning led to recent work using deep neural networks that allow predicting spectra from molecular graphs or fingerprints [<span class="citation"><a href="#Wei2019">Wei <i>et al.</i> 2019</a></span>].</p>

<p>You will be able to check out how QCxMS works in practice since we are going to use Galaxy tool suite based on this method [<span class="citation"><a href="#Grimme2013">Grimme 2013</a></span>, <span class="citation"><a href="#Bauer2014">Bauer and Grimme 2014</a></span>, <span class="citation"><a href="#Bauer2016">Bauer and Grimme 2016</a></span>, <span class="citation"><a href="#Koopman2021">Koopman and Grimme 2021</a></span>]. Beforehand, we will generate conformers of the query molecule with <a href="http://www.rdkit.org">RDKit</a> and we will use xTB for molecular optimisation [<span class="citation"><a href="#Bannwarth2020">Bannwarth <i>et al.</i> 2020</a></span>].
But first things first, let’s get some toy data to play with and crack on!</p>

<blockquote class="question">
  <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

  <p>What does QCEIMS stand for?</p>

  <blockquote class="tip">
    <div class="box-title tip-title" id="tip-hint"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-hint" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Hint</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>With a little knowledge of chemistry, you’ll be able to work it out yourself!</p>

    <p>Look at the acronym in the following way: <strong>QC-EI-MS</strong>.</p>
  </blockquote>
  <blockquote class="solution">
    <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>QC = quantum chemical</p>

    <p>EI = electron ionisation</p>

    <p>MS = mass spectrometry</p>

    <p>Hence QCEIMS = quantum-chemical electron ionisation mass spectrometry</p>

  </blockquote>

</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-is-qcxms"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-qcxms" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What is QCxMS?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>QCxMS is a successor of QCEIMS, where the <em>EI</em> part is replaced by <em>x</em> to take into account other ionisation methods and improve the applicability of the program. In QCEIMS, <em>EI</em> stands for <em>electron ionisation</em>, while in QCxMS, <em>x</em> refers to <em>EI</em> or <em>CID (collision-induced dissociation)</em> [<span class="citation"><a href="#Koopman2021">Koopman and Grimme 2021</a></span>]. Currently, only <em>EI</em> simulations are supported - using <em>CID</em> with the Galaxy tool wrappers is still under development.</p>
</blockquote>

<h1 id="importing-data-and-pre-processing">Importing data and pre-processing</h1>

<p>In this tutorial, you can choose whether you want to predict the mass spectrum for one molecule only, or if you want to do it for multiple molecules at once. The pre-processing steps will slightly differ depending on your choice. If you are completing this tutorial just to see how the QCxMS tools work, feel free to follow the instructions for one molecule to skip some pre-processing steps.</p>

<p>In both cases, we start from molecule’s SMILES, and then we convert it to SDF, so if you already have SDF files to work with, simply jump in the relevant place in the workflow and carry on from there.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-what-is-smiles"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-is-smiles" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: What is SMILES?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

  <p>SMILES (.smi) - the simplified molecular-input line-entry system (SMILES) is a specification in the form of a line notation for describing the structure of a chemical species using short ASCII strings. It is a linear text format which can describe the connectivity and chirality of a molecule [<span class="citation"><a href="#Weininger1988">Weininger 1988</a></span>]. Even though many different forms of SMILES exist, the differences are not relevant for us in this application.</p>
  <figure id="figure-1" style="max-width: 90%;"><img src="../../images/qcxms_smiles_explanation.png" alt="Depiction of a molecular structure and the corresponding SMILES string. Starts of rings are denoted with a number after the corresponding element (i.e. C1). Non-organic elements need to be written in brackets and isotopic forms need to be denoted before the atom and in brackets as well. Atoms with charges need to be in brackets as well and the charge needs to be denoted behind the atom or the functional group. Hydrogens inside brackets need to be stated explicitly. Double bonds are denoted with a `=` and triple bonds with a `#`." width="1865" height="761" loading="lazy" /><a target="_blank" href="../../images/qcxms_smiles_explanation.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 1</strong>:</span> A quick explanation of the SMILES system.</figcaption></figure>
  <p>Image credit: Helge Hecht, License: MIT</p>

</blockquote>

<!---
consider the figure with SMILES and the corresponding structure
-->

<blockquote class="notranslate hands_on gtn-cyoa" id="gtn-cyoasingle_vs_multiple">
	<div class="box-title" aria-label="hands-on box: choose your own tutorial"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Choose Your Own Tutorial</div>
	<p>This is a "Choose Your Own Tutorial" section, where you can select between multiple paths. Click one of the buttons below to select how you want to follow the tutorial</p>
	
	<p>
		Choose below if you just want to follow the pipeline for prediting the spectrum for only one molecule or multiple molecules at once!
	</p>
	
	
		<input id="cyoa-opt1single_vs_multiple" type="radio" name="cyoasingle_vs_multiple" value="Predict-MS-for-a-single-molecule" onclick="cyoaChoice('Predict-MS-for-a-single-molecule', 'gtn-cyoasingle_vs_multiple')" />
		<label for="cyoa-opt1single_vs_multiple" class="select btn btn-secondary">Predict MS for a single molecule</label>
	
	
		<input id="cyoa-opt2single_vs_multiple" type="radio" name="cyoasingle_vs_multiple" value="Predict-MS-for-multiple-molecules-at-once" onclick="cyoaChoice('Predict-MS-for-multiple-molecules-at-once', 'gtn-cyoasingle_vs_multiple')" />
		<label for="cyoa-opt2single_vs_multiple" class="select btn btn-secondary">Predict MS for multiple molecules at once</label>
	
	
	

	
	<script>
		function ready(fn) {
			if (document.readyState != 'loading'){
				fn();
			} else {
				document.addEventListener('DOMContentLoaded', fn);
			}
		}
		ready(function() {
			cyoaDefault("Predict-MS-for-a-single-molecule", "gtn-cyoasingle_vs_multiple");
		})
	</script>
	
</blockquote>

<div class="Predict-MS-for-a-single-molecule">

  <h2 id="upload-data-onto-galaxy">Upload data onto Galaxy</h2>

  <p>Working on a single molecule means that we will work on a dataset and not on a collection of datasets. To simulate the spectrum, we will use ethanol (C<sub>2</sub>H<sub>5</sub> OH) as an example, but you can choose any other molecule that you want, but be aware that the more complex structure you choose, the more time it will take to complete the analysis since the workflow involves generating conformers, semiempirical methods and molecular optimisation.
We will simply start with molecule’s SMILES. 
You have three options for uploading the data. The first two - importing via history and Zenodo link will give a file specific to this tutorial, while the last one – “Paste data uploader” gives you more flexibility in terms of the compounds you would like to test with this workflow.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-option-1-data-upload-import-history"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 1: Data upload - Import history</div>

    <ol>
      <li>
        <p>You can simply import <a href="https://usegalaxy.eu/u/j.jakiela/h/input-file-ei-mass-spectra-prediction-using-qcxms-1">this history</a> with the input file.</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <em>Import this history</em> button on the top left</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Copy History</strong></li> </ol> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
      <li>
        <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the history to your name of choice.</p>
      </li>
    </ol>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-option-2-data-upload-add-to-history-via-zenodo"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 2: Data upload - Add to history via Zenodo</div>

    <ol>
      <li>Create a new history for this tutorial</li>
      <li>
        <p>Import the input table from <a href="https://zenodo.org/record/13327051">Zenodo</a></p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/13327051/files/ethanol_SMILES.smi
</code></pre></div>        </div>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-option-3-data-upload-paste-data"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Option 3: Data Upload  - paste data</div>

    <ol>
      <li>Create a new history for this tutorial</li>
      <li>
        <ul>
          <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>
          <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>
          <li>Paste the SMILES into the text field:
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>CCO
</code></pre></div>            </div>
          </li>
          <li>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">smi</code></li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </li>
      <li>You can then rename the dataset as you wish (here we use <code class="language-plaintext highlighter-rouge">ethanol_SMILES</code>)</li>
      <li>
        <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">smi</code>.</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">smi</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <h2 id="smiles-to-sdf">SMILES to SDF</h2>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-convert-smiles-to-sdf"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Convert SMILES to SDF</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0" title="Compound conversion tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compound conversion</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.1.1+galaxy0)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecular input file”</em>: <code class="language-plaintext highlighter-rouge">ethanol_SMILES</code> (your SMILES dataset from your history)</li>
      <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">MDL MOL format (sdf, mol)</code></li>
      <li><em>“Append the specified text after each molecule title”</em>: <code class="language-plaintext highlighter-rouge">ethanol</code></li>
    </ul>

  </blockquote>

  <p>We now have an SDF file, containing the atoms’ coordinates and the investigated molecule’s name.</p>

  <h1 id="d-conformer-generation--optimization">3D Conformer generation &amp; optimization</h1>

  <h2 id="generate-conformers">Generate conformers</h2>
  <p>The next step involves generating three-dimensional (3D) conformers for our molecule. It crteaes the actual 3D topology of the molecule based on electromagnetic forces. This process might seem trivial for very small and simplistic (meaning no complex structure) molecules, but this can be more challenging for larger molecules with a more flexible geometry. This concerns for example P containing biomolecules, where P often forms a rotational center of the molecule. The number of conformers to generate can be specified as an input parameter, with a default value of 1 if not provided. This process is crucial for exploring the possible shapes and energies that a molecule can adopt. The output of this step is a file containing the generated 3D conformers.</p>

  <blockquote class="details">
    <div class="box-title details-title" id="details-what-are-conformers"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-are-conformers" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  What are conformers? </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Conformers are different spatial arrangements of a molecule that result from rotations around single bonds. They have different potential energies and hence some are more favourable (local minima on the potential energy surface) than others.</p>
    <figure id="figure-2" style="max-width: 90%;"><div style="overflow-x: auto"><object data="../../images/qcxms_predictions_conformers.svg" type="image/svg+xml" alt="Newman projections of butane conformations &amp; their relative energy differences (not total energies). Conformations form when butane rotates about one of its single covalent bond. Torsional/dihedral angle is shown on x-axis.">Newman projections of butane conformations &amp; their relative energy differences (not total energies). Conformations form when butane rotates about one of its single covalent bond. Torsional/dihedral angle is shown on x-axis.</object></div><a target="_blank" href="../../images/qcxms_predictions_conformers.svg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 2</strong>:</span> Conformers of butane and their relative energy differences.</figcaption></figure>
    <p>Image credit: <a href="https://commons.wikimedia.org/wiki/File:Butane_conformations_and_relative_energies.svg">Keministi</a>, License: Creative Commons CC0 1.0.</p>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-generate-conformers"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Generate conformers</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/ctb_im_conformers/ctb_im_conformers/1.1.4+galaxy0" title="Generate conformers tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Generate conformers</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.4+galaxy0)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: output of <strong>Convert to sdf</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      <li><em>“Number of conformers to generate”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
    </ul>

  </blockquote>

  <p>Now - once again format conversion! This time we will convert the generated conformers from the SDF format to Cartesian coordinate (XYZ) format. The XYZ format lists the atoms in a molecule and their respective 3D coordinates, which is a common format used in computational chemistry for further processing and analysis.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-molecular-format-conversion"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Molecular Format Conversion</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0" title="Compound conversion tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compound conversion</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.1.1+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecular input file”</em>: output (sdf) of <strong>Generate conformers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">XYZ cartesian coordinates format</code></li>
          <li><em>“Add hydrogens appropriate for pH”</em>: <code class="language-plaintext highlighter-rouge">7.0</code></li>
        </ul>
      </li>
      <li>
        <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">xyz</code>. If it’s not, just change it – below is the tip how to do it.</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">xyz</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <h2 id="molecular-optimization">Molecular optimization</h2>

  <p>As shown in the image in the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> <em>Details box</em> above, different conformers have different energies. Therefore, our next step will optimize the geometry of the molecules to find the lowest energy conformation. This is crucial to achieve convergence in the next steps. If the input geometry for the QCxMS method is too crude, the ground state neutral run will not converge and we won’t be able to sample the geometry to calculate individual trajectories. We will perform semi-empirical optimization on the molecules using the <a href="https://github.com/grimme-lab/xtb">Extended Tight-Binding (xTB)</a> method. The level of optimization accuracy to be used can be specified as an input parameter, <em>“Optimization Levels”</em>. The default quantum chemical method is GFN2-xTB.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-molecular-optimisation-with-xtb"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Molecular optimisation with xTB</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/xtb_molecular_optimization/xtb_molecular_optimization/6.6.1+galaxy1" title="xtb molecular optimization tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>xtb molecular optimization</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 6.6.1+galaxy1)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Atomic coordinates file”</em>: output of <strong>Convert to xyz</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      <li><em>“Optimization Levels”</em>: <code class="language-plaintext highlighter-rouge">tight</code></li>
      <li><em>“Keep molecule name”</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span>  <code class="language-plaintext highlighter-rouge">Yes</code></li>
    </ul>

  </blockquote>

  <h1 id="qcxms-spectra-prediction">QCxMS Spectra Prediction</h1>

  <h2 id="neutral-and-production-runs">Neutral and production runs</h2>

  <p>Finally, let’s predict the spectra for our molecule. As mentioned, we will use <a href="https://github.com/qcxms/QCxMS">QCxMS</a> for this purpose. First, we need to prepare the necessary input files for the QCxMS production runs. These files are required for running the QCxMS simulations, which will predict the mass spectrum of the molecule. This step typically formats the optimized molecular data into a format that can be used for the production simulations. This step performs the ground state calculations. The resulting geometry trajectory is then sampled and one representation is used for each trajectory.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-qcxms-neutral-run"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  QCxMS neutral run</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/qcxms_neutral_run/qcxms_neutral_run/5.2.1+galaxy3" title="QCxMS neutral run tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>QCxMS neutral run</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.2.1+galaxy3)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecule 3D structure [.xyz]”</em>: output of <strong>xtb molecular optimization</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      <li><em>“QC Method”</em>: <code class="language-plaintext highlighter-rouge">GFN2-xTB</code></li>
    </ul>

  </blockquote>

  <p>The outputs of the above step are as follows:</p>
  <ul>
    <li>.in output: Input file for the QCxMS production run</li>
    <li>.start output: Start file for the QCxMS production run</li>
    <li>.xyz output: Cartesian coordinate file for the QCxMS production run</li>
  </ul>

  <p>We can now use those files as input for the next tool which calculates the mass spectra using QCxMS. This simulation generates .res files, which contain the raw results of the mass spectra calculations. These results are essential for predicting how the molecules will appear in mass spectrometry experiments.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-qcxms-production-run"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  QCxMS production run</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/qcxms_production_run/qcxms_production_run/5.2.1+galaxy3" title="QCxMS production run tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>QCxMS production run</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.2.1+galaxy3)</span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> <em>“in files [.in]”</em>: <code class="language-plaintext highlighter-rouge">input in files</code> generated by <strong>QCxMS neutral run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> <em>“start files [.start]”</em>: <code class="language-plaintext highlighter-rouge">input start files</code> generated by <strong>QCxMS neutral run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> <em>“xyz files [.xyz]”</em>: <code class="language-plaintext highlighter-rouge">input xyz files</code> generated by <strong>QCxMS neutral run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <h2 id="filter-failed-datasets">Filter failed datasets</h2>

  <p>It might be the case that some runs might have failed, therefore it is crucial to filter out any failed runs from the dataset to ensure only successful results are processed further. This step is important to maintain the integrity and quality of the data being analyzed in subsequent steps. The output is a file containing only the successful mass spectra results.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-filter-failed-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Filter failed datasets</div>

    <ol>
      <li><span class="tool" data-tool="__FILTER_FAILED_DATASETS__" title="Filter failed datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter failed datasets</strong></span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Collection”</em>: <code class="language-plaintext highlighter-rouge">res files generated by QCxMS</code> (output of <strong>QCxMS production run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> )</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <h2 id="get-msp-spectra">Get MSP spectra</h2>

  <p>The filtered collection contains .res files from the QCxMS production run. This final step converts the .res files into simulated mass spectra in MSP (Mass Spectrum Peak) file format. The MSP format is widely used for storing and sharing mass spectrometry data, enabling easy comparison and analysis of the results, for example by comparing the spectra using the <a href="https://github.com/matchms/matchms">matchms</a> package.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-qcxms-get-msp-results"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  QCxMS get MSP results</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/qcxms_getres/qcxms_getres/5.2.1+galaxy2" title="QCxMS get results tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>QCxMS get results</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.2.1+galaxy2)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecule 3D structure [.xyz]”</em>: <code class="language-plaintext highlighter-rouge">Convert to xyz</code> (output of <strong>Compound conversion</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><em>“res files [.res]”</em>: output of <strong>Filter failed datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (if the collection doesn’t appear in the drop-down list, simply drag and drop it from the history panel to the input box)</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <blockquote class="details">
    <div class="box-title details-title" id="details-msp-files"><button class="gtn-boxify-button details" type="button" aria-controls="details-msp-files" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: MSP files </span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>MSP (Mass Spectrum Peak) file is a text file structured according to the NIST MSSearch spectra format. MSP is one of the generally accepted formats for mass spectral libraries (or collections of unidentified spectra, so called spectral archives), and it is compatible with lots of spectra processing programmes (MS-DIAL, NIST MS Search, AMDIS, matchms, etc.). It can contain one or more mass spectra, these are split by an empty line. The individual spectra essentially consist of two sections: metadata (such as name, spectrum type, ion mode, retention time, and the number of m/z peaks) and peaks, consisting of m/z and intensity tuples.</p>
  </blockquote>

  <p>You can now <i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span> download the MSP file and open it in your spectra processing software for further investigation!</p>

  <p>To give you some insight into how well QCxMS can perform, below is the mass spectrum of ethanol resulting from our workflow compared with an <a href="https://hmdb.ca/spectra/c_ms/28442">experimental spectrum</a>. Both spectra were compiled using an <a href="https://www.sisweb.com/mstools/spectrum.htm">online mass spectrum generator</a> which requires only m/z values and intensities – so the values that you can get from our MSP file! As you can see, the predicted peaks nicely correspond to experimental ones. But be careful - there might be slight deviations for molecules with more structural complexity!</p>

  <figure id="figure-3" style="max-width: 90%;"><img src="../../images/qcxms_predictions_ms_exp_pred.png" alt="Upper panel shows the experimental spectrum of ethanol, while the lower panel shows analogical spectrum but predicted with the current workflow. The predicted peaks correspond well to the experimental ones. Intensities of simulated peaks have not been predicted perfectly, but the most important trends are preserved." width="635" height="453" loading="lazy" /><a target="_blank" href="../../images/qcxms_predictions_ms_exp_pred.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 3</strong>:</span> Comparison between experimental (upper panel) and predicted (lower panel) mass spectra of ethanol.</figcaption></figure>

  <h1 id="conclusion">Conclusion</h1>
  <p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Well done, you’ve simulated the mass spectrum! You might want to consult your results with the <a href="https://usegalaxy.eu/u/j.jakiela/h/ms-prediction-qcxms-single-molecule">key history</a>. If you would like to process multiple molecules at once, you can simply use <a href="https://usegalaxy.eu/u/hechth/h/end-to-end-ei-mass-spectra-prediction-workflow-using-qcxms-1">the workflow</a> or move to “Predict MS for multiple molecules at once” tab of this tutorial to learn how the pipeline differs from the one that we’ve just covered.</p>

</div>

<div class="Predict-MS-for-multiple-molecules-at-once">

  <h2 id="upload-data-onto-galaxy-1">Upload data onto Galaxy</h2>

  <p>We will work on two simple molecules – ethanol (C<sub>2</sub>H<sub>5</sub> OH) and ethylene (C<sub>2</sub>H<sub>4</sub>). Of course, you might add more and choose any other molecules that you want, but be aware that the more complex structure you choose, the more time it will take to complete the analysis since the workflow involves generating conformers, semiempirical methods and molecular optimisation.
We will start with a table with the first column being molecule names and the second one – corresponding SMILES. 
You have three options for uploading the data. The first two - importing via history and Zenodo link will give a file specific to this tutorial, while the last one – “Paste data uploader” gives you more flexibility in terms of the compounds you would like to test with this workflow.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-option-1-data-upload-import-history-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 1: Data upload - Import history</div>

    <ol>
      <li>
        <p>You can simply import <a href="https://usegalaxy.eu/u/j.jakiela/h/input-file-end-to-end-ei-mass-spectra-prediction-workflow-using-qcxms">this history</a> with the input table.</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <em>Import this history</em> button on the top left</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Copy History</strong></li> </ol> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
      <li>
        <p><strong>Rename</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> the history to your name of choice.</p>
      </li>
    </ol>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-option-2-data-upload-add-to-history-via-zenodo-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Option 2: Data upload - Add to history via Zenodo</div>

    <ol>
      <li>Create a new history for this tutorial</li>
      <li>
        <p>Import the input table from <a href="https://zenodo.org/record/13327051">Zenodo</a></p>

        <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/records/13327051/files/qcxms_prediction_input.tabular
</code></pre></div>        </div>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-option-3-data-upload-paste-data-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Option 3: Data Upload  - paste data</div>

    <ol>
      <li>Create a new history for this tutorial</li>
      <li>
        <ul>
          <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>
          <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>
          <li>Paste the contents into the text field, separated by space. First, enter the name of the molecule, then its SMILES. Please note that we are not using headers here. For this tutorial, we’ll use the example of ethanol and ethylene, but feel free to use your own examples.
            <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>ethanol CCO
ethylene C=C
</code></pre></div>            </div>
          </li>
          <li>Change <strong>Type</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">tabular</code></li>
          <li>Find the gear symbol (<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span>), deselect any ticked options and select only (<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span>) <strong>Convert spaces to tabs</strong></li>
          <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li>
        </ul>
      </li>
      <li>You can then rename the dataset as you wish.</li>
      <li>
        <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">tabular</code>.</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-2"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-2" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">tabular</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <h2 id="input-pre-processing">Input pre-processing</h2>

  <p>Once your dataset is uploaded, we can do some simple pre-processing to prepare the file for downstream analysis. Let’s start with ‘cutting’ the table into two columns – one with SMILES, the other with the molecule name – and then separating each entry to create a dataset collection, followed by parsing out the text information.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-cutting-out-name-column"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Cutting out name column</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1" title="Advanced Cut tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Advanced Cut</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 9.3+galaxy1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: the tabular file in your history with the compound name and SMILES</li>
          <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Keep</code></li>
          <li><em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code></li>
          <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
          <li><em>“Is there a header for the data’s columns”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          <li><em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        </ul>
      </li>
      <li>You can now rename the resulting dataset or just add a tag in order not to confuse it with subsequent outputs:
        <ul>
          <li><i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> Add tag: #names</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-creating-dataset-collection-molecule-name"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Creating dataset collection (molecule name)</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2" title="Split file to dataset collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Split file to dataset collection</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.5.2)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the file type to split”</em>: <code class="language-plaintext highlighter-rouge">Tabular</code></li>
      <li><em>“Tabular file to split”</em>: output of <strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (with #names tag)</li>
      <li><em>“Number of header lines to transfer to new files”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
      <li><em>“Split by row or by a column?”</em>: <code class="language-plaintext highlighter-rouge">By row</code></li>
      <li><em>“Specify number of output files or number of records per file?”</em>: <code class="language-plaintext highlighter-rouge">Number of records per file (‘chunk mode’)</code></li>
      <li><em>“Chunk size”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
      <li><em>“Base name for new files in collection”</em>: <code class="language-plaintext highlighter-rouge">split_file</code></li>
      <li><em>“Method to allocate records to new files”</em>: <code class="language-plaintext highlighter-rouge">Maintain record order</code></li>
    </ul>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-parsing-out-name-info"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Parsing out name info</div>

    <p><span class="tool" data-tool="param_value_from_file" title="Parse parameter value tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Parse parameter value</strong></span> with the following parameters:</p>
    <ul>
      <li><em>“Input file containing parameter to parse out of”</em>: * click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> and select output of <strong>Split file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      <li><em>“Select type of parameter to parse”</em>: <code class="language-plaintext highlighter-rouge">Text</code></li>
      <li><em>“Remove newlines ?”</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span>  <code class="language-plaintext highlighter-rouge">Yes</code></li>
    </ul>

    <blockquote class="tip">
      <div class="box-title tip-title" id="tip-can-t-find-quot-parse-parameter-value-quot-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-t-find-quot-parse-parameter-value-quot-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Can't find &quot;Parse parameter value&quot; tool? </span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>If you have any problems with accessing <strong>Parse parameter value</strong> tool, you can open <a href="https://usegalaxy.eu/root?tool_id=param_value_from_file">the tool</a> directly using a given link.</p>
    </blockquote>

  </blockquote>

  <p>We will repeat the first two steps, but processing SMILES this time.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-cutting-out-smiles-column"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Cutting out SMILES column</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/9.3+galaxy1" title="Advanced Cut tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Advanced Cut</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 9.3+galaxy1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“File to cut”</em>: the tabular file in your history with the compound name and SMILES</li>
          <li><em>“Operation”</em>: <code class="language-plaintext highlighter-rouge">Keep</code></li>
          <li><em>“Cut by”</em>: <code class="language-plaintext highlighter-rouge">fields</code></li>
          <li><em>“Delimited by”</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
          <li><em>“Is there a header for the data’s columns”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          <li><em>“List of Fields”</em>: <code class="language-plaintext highlighter-rouge">Column: 2</code></li>
        </ul>
      </li>
      <li>You can now rename the resulting dataset or just add a tag in order not to confuse it with subsequent outputs:
        <ul>
          <li><i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> Add tag: #SMILES</li>
        </ul>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-creating-dataset-collection-smiles"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:   Creating dataset collection (SMILES)</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/split_file_to_collection/split_file_to_collection/0.5.2" title="Split file to dataset collection tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Split file to dataset collection</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.5.2)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select the file type to split”</em>: <code class="language-plaintext highlighter-rouge">Tabular</code></li>
          <li><em>“Tabular file to split”</em>: output of the latest <strong>Advanced Cut</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (higher number in your history, with #SMILES tag)</li>
          <li><em>“Number of header lines to transfer to new files”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
          <li><em>“Split by row or by a column?”</em>: <code class="language-plaintext highlighter-rouge">By row</code></li>
          <li><em>“Specify number of output files or number of records per file?”</em>: <code class="language-plaintext highlighter-rouge">Number of records per file (‘chunk mode’)</code></li>
          <li><em>“Chunk size”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
          <li><em>“Base name for new files in collection”</em>: <code class="language-plaintext highlighter-rouge">split_file</code></li>
          <li><em>“Method to allocate records to new files”</em>: <code class="language-plaintext highlighter-rouge">Maintain record order</code></li>
        </ul>
      </li>
      <li>
        <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">smi</code>. If it’s not, just change it!</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-3"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-3" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">smi</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <p>Now, onto format conversion. Let’s convert our SMILES to SDF (Structure Data File) and append the molecule’s name that we have already extracted.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-convert-smiles-to-sdf-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Convert SMILES to SDF</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0" title="Compound conversion tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compound conversion</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.1.1+galaxy0)</span> with the following parameters:</p>
    <ul>
      <li><em>“Molecular input file”</em>: click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> and select output of <strong>Split file</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> on SMILES</li>
      <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">MDL MOL format (sdf, mol)</code></li>
    </ul>

    <blockquote class="tip">
      <div class="box-title tip-title" id="tip-add-name-of-the-molecule"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-name-of-the-molecule" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Add name of the molecule</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
      <p>Another parameter of this tool, <em>“Append the specified text after each molecule title”</em>, allows you to add name of the molecule to the title of the generated file. 
If you are working on a single molecule file, you can simply type the name of that compound into the parameter box.
However, if your input is a collection of SMILES (like in this case), then you have to add the names (output of <strong>Parse parameter value</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>) at the level of the workflow editor.</p>
    </blockquote>

  </blockquote>

  <!---
- *"Append the specified text after each molecule title"*: click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> *Dataset collection* and select output of **Parse parameter value** <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> which contains the extracted molecules' names
-->

  <p>We now have two SDF files, each containing the coordinates of the atoms and the name of the investigated molecule. Let’s combine them to make life easier and work on just one file.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-concatenating-the-files"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Concatenating the files</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1" title="Concatenate datasets tail-to-head (cat) tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets tail-to-head (cat)</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 9.3+galaxy1)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Datasets to concatenate”</em>:</li>
      <li>if nothing pops up when you click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em>, then click on “switch to column select”</li>
      <li>from the datasets list, find the output of <strong>Compound conversion</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (should have the highest number in your history) and select it</li>
      <li><i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> Run tool!</li>
    </ul>

  </blockquote>

  <h1 id="d-conformer-generation--optimization-1">3D Conformer generation &amp; optimization</h1>

  <h2 id="generate-conformers-1">Generate conformers</h2>
  <p>The next step involves generating three-dimensional (3D) conformers for each molecule from the generated SDF. It crteaes the actual 3D topology of the molecules based on electromagnetic forces. This process might seem trivial for very small and simplistic (meaning no complex structure) molecules, but this can be more challenging for larger molecules with a more flexible geometry. This concerns for example P containing biomolecules, where P often forms a rotational center of the molecule. The number of conformers to generate can be specified as an input parameter, with a default value of 1 if not provided. This process is crucial for exploring the possible shapes and energies that a molecule can adopt. The output of this step is a file containing the generated 3D conformers.</p>

  <blockquote class="details">
    <div class="box-title details-title" id="details-what-are-conformers-1"><button class="gtn-boxify-button details" type="button" aria-controls="details-what-are-conformers-1" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details:  What are conformers? </span><span class="fold-unfold fa fa-minus-square"></span></button></div>

    <p>Conformers are different spatial arrangements of a molecule that result from rotations around single bonds. They have different potential energies and hence some are more favourable (local minima on the potential energy surface) than others.</p>
    <figure id="figure-4" style="max-width: 90%;"><div style="overflow-x: auto"><object data="../../images/qcxms_predictions_conformers.svg" type="image/svg+xml" alt="Newman projections of butane conformations &amp; their relative energy differences (not total energies). Conformations form when butane rotates about one of its single covalent bond. Torsional/dihedral angle is shown on x-axis.">Newman projections of butane conformations &amp; their relative energy differences (not total energies). Conformations form when butane rotates about one of its single covalent bond. Torsional/dihedral angle is shown on x-axis.</object></div><a target="_blank" href="../../images/qcxms_predictions_conformers.svg" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 4</strong>:</span> Conformers of butane and their relative energy differences.</figcaption></figure>
    <p>Image credit: <a href="https://commons.wikimedia.org/wiki/File:Butane_conformations_and_relative_energies.svg">Keministi</a>, License: Creative Commons CC0 1.0.</p>

  </blockquote>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-generate-conformers-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Generate conformers</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/ctb_im_conformers/ctb_im_conformers/1.1.4+galaxy0" title="Generate conformers tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Generate conformers</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.4+galaxy0)</span> with the following parameters:</p>
    <ul>
      <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input file”</em>: output of <strong>Concatenate datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      <li><em>“Number of conformers to generate”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
    </ul>

  </blockquote>

  <p>Now - once again format conversion! This time we will convert the generated conformers from the SDF format to Cartesian coordinate (XYZ) format. The XYZ format lists the atoms in a molecule and their respective 3D coordinates, which is a common format used in computational chemistry for further processing and analysis.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-molecular-format-conversion-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Molecular Format Conversion</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/openbabel_compound_convert/openbabel_compound_convert/3.1.1+galaxy0" title="Compound conversion tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Compound conversion</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 3.1.1+galaxy0)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Molecular input file”</em>: output of <strong>Generate conformers</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">XYZ cartesian coordinates format</code></li>
          <li><em>“Split multi-molecule files into a collection”</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <code class="language-plaintext highlighter-rouge">Yes</code></li>
          <li><em>“Add hydrogens appropriate for pH”</em>: <code class="language-plaintext highlighter-rouge">7.0</code></li>
        </ul>
      </li>
      <li>
        <p>Check that the datatype is <code class="language-plaintext highlighter-rouge">xyz</code>. If it’s not, just change it!</p>

        <!--SNIPPET-->
        <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-4"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-4" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">xyz</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
        <p><!--END_SNIPPET--></p>
      </li>
    </ol>

  </blockquote>

  <h2 id="molecular-optimization-1">Molecular optimization</h2>

  <p>As shown in the image in the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> <em>Details box</em> above, different conformers have different energies. Therefore, our next step will optimize the geometry of the molecules to find the lowest energy conformation. This is crucial to achieve convergence in the next steps. If the input geometry for the QCxMS method is too crude, the ground state neutral run will not converge and we won’t be able to sample the geometry to calculate individual trajectories. We will perform semi-empirical optimization on the molecules using the <a href="https://github.com/grimme-lab/xtb">Extended Tight-Binding (xTB)</a> method. The level of optimization accuracy to be used can be specified as an input parameter, <em>“Optimization Levels”</em>. The default quantum chemical method is GFN2-xTB.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-molecular-optimisation-with-xtb-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Molecular optimisation with xTB</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/xtb_molecular_optimization/xtb_molecular_optimization/6.6.1+galaxy1" title="xtb molecular optimization tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>xtb molecular optimization</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 6.6.1+galaxy1)</span> with the following parameters:</p>
    <ul>
      <li><em>“Atomic coordinates file”</em>: click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> and select <code class="language-plaintext highlighter-rouge">Prepared ligands</code> (output of <strong>Compound conversion</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      <li><em>“Optimization Levels”</em>: <code class="language-plaintext highlighter-rouge">tight</code></li>
      <li><em>“Keep molecule name”</em>: <i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span>  <code class="language-plaintext highlighter-rouge">Yes</code></li>
    </ul>

  </blockquote>

  <h1 id="qcxms-spectra-prediction-1">QCxMS Spectra Prediction</h1>

  <h2 id="neutral-and-production-runs-1">Neutral and production runs</h2>

  <p>Finally, let’s predict the spectra for our molecules. As mentioned, we will use <a href="https://github.com/qcxms/QCxMS">QCxMS</a> for this purpose. First, we need to prepare the necessary input files for the QCxMS production runs. These files are required for running the QCxMS simulations, which will predict the mass spectra of the molecules. This step typically formats the optimized molecular data into a format that can be used for the production simulations. This step performs the ground state calculations. The resulting geometry trajectory is then sampled and one representation is used for each trajectory.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-qcxms-neutral-run-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  QCxMS neutral run</div>

    <p><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/qcxms_neutral_run/qcxms_neutral_run/5.2.1+galaxy3" title="QCxMS neutral run tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>QCxMS neutral run</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.2.1+galaxy3)</span> with the following parameters:</p>
    <ul>
      <li><em>“Molecule 3D structure [.xyz]”</em>: click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> and select output of <strong>xtb molecular optimization</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
      <li><em>“QC Method”</em>: <code class="language-plaintext highlighter-rouge">GFN2-xTB</code></li>
    </ul>

  </blockquote>

  <p>The outputs of the above step are as follows:</p>
  <ul>
    <li>.in output: Input file for the QCxMS production run</li>
    <li>.start output: Start file for the QCxMS production run</li>
    <li>.xyz output: Cartesian coordinate file for the QCxMS production run</li>
  </ul>

  <p>We can now use those files as input for the next tool which calculates the mass spectra for each molecule using QCxMS (Quantum Chemistry and Mass Spectrometry). This simulation generates .res files, which contain the raw results of the mass spectra calculations. These results are essential for predicting how the molecules will appear in mass spectrometry experiments.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-qcxms-production-run-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  QCxMS production run</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/qcxms_production_run/qcxms_production_run/5.2.1+galaxy3" title="QCxMS production run tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>QCxMS production run</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.2.1+galaxy3)</span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> <em>“in files [.in]”</em>: <code class="language-plaintext highlighter-rouge">input in files</code> generated by <strong>QCxMS neutral run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> <em>“start files [.start]”</em>: <code class="language-plaintext highlighter-rouge">input start files</code> generated by <strong>QCxMS neutral run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> <em>“xyz files [.xyz]”</em>: <code class="language-plaintext highlighter-rouge">input xyz files</code> generated by <strong>QCxMS neutral run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <h2 id="filter-failed-datasets-1">Filter failed datasets</h2>

  <p>It might be the case that some runs might have failed, therefore it is crucial to filter out any failed runs from the dataset to ensure only successful results are processed further. This step is important to maintain the integrity and quality of the data being analyzed in subsequent steps. The output is a file containing only the successful mass spectra results.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-filter-failed-datasets-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Filter failed datasets</div>

    <ol>
      <li><span class="tool" data-tool="__FILTER_FAILED_DATASETS__" title="Filter failed datasets tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter failed datasets</strong></span> with the following parameters:
        <ul>
          <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>“Input Collection”</em>: output of <strong>QCxMS production run</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <h2 id="get-msp-spectra-1">Get MSP spectra</h2>

  <p>The filtered collection contains .res files from the QCxMS production run. This final step converts the .res files into simulated mass spectra in MSP (Mass Spectrum Peak) file format. The MSP format is widely used for storing and sharing mass spectrometry data, enabling easy comparison and analysis of the results, for example by comparing the spectra using the <a href="https://github.com/matchms/matchms">matchms</a> package.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-qcxms-get-msp-results-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  QCxMS get MSP results</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/recetox/qcxms_getres/qcxms_getres/5.2.1+galaxy2" title="QCxMS get results tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>QCxMS get results</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 5.2.1+galaxy2)</span> with the following parameters:
        <ul>
          <li><em>“Molecule 3D structure [.xyz]”</em>: click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em> and select <code class="language-plaintext highlighter-rouge">Prepared ligands</code> (output of <strong>Compound conversion</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“res files [.res]”</em>: output of <strong>Filter failed datasets</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (if the collection doesn’t appear in the drop-down list, simply drag and drop it from the history panel to the input box)</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <p>The output of this step is a collection of two MSP files - one per each molecule. However, if you want, you can combine those two into one MSP file.</p>

  <blockquote class="notranslate hands_on">
    <div class="box-title hands-on-title" id="hands-on-combine-msp-files-into-one"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on:  Combine MSP files into one</div>

    <ol>
      <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/9.3+galaxy1" title="Concatenate datasets tail-to-head (cat) tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Concatenate datasets tail-to-head (cat)</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 9.3+galaxy1)</span> with the following parameters:
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Datasets to concatenate”</em>:</li>
          <li>if nothing pops up when you click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>Dataset collection</em>, then click on “switch to column select”</li>
          <li>from the datasets list, find the output of <strong>QCxMS get results</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (should be the latest output dataset) and select it</li>
          <li><i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> Run tool!</li>
        </ul>
      </li>
    </ol>

  </blockquote>

  <blockquote class="details">
    <div class="box-title details-title" id="details-msp-files-1"><button class="gtn-boxify-button details" type="button" aria-controls="details-msp-files-1" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: MSP files </span><span class="fold-unfold fa fa-minus-square"></span></button></div>
    <p>MSP (Mass Spectrum Peak) file is a text file structured according to the NIST MSSearch spectra format. MSP is one of the generally accepted formats for mass spectral libraries (or collections of unidentified spectra, so called spectral archives), and it is compatible with lots of spectra processing programmes (MS-DIAL, NIST MS Search, AMDIS, matchms, etc.). It can contain one or more mass spectra, these are split by an empty line. The individual spectra essentially consist of two sections: metadata (such as name, spectrum type, ion mode, retention time, and the number of m/z peaks) and peaks, consisting of m/z and intensity tuples.</p>
  </blockquote>

  <p>You can now <i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span> download the MSP file and open it in your spectra processing software for further investigation!</p>

  <p>To give you some insight into how well QCxMS can perform, below is the mass spectrum of ethanol resulting from our workflow compared with <a href="https://hmdb.ca/spectra/c_ms/28442">experimental spectrum</a>. Both spectra were compiled using an <a href="https://www.sisweb.com/mstools/spectrum.htm">online mass spectrum generator</a> which requires only m/z values and intensities – so the values that you can get from our MSP file! As you can see, the predicted peaks nicely correspond to experimental ones. But be careful - there might be slight deviations for molecules with more structural complexity!</p>

  <figure id="figure-5" style="max-width: 90%;"><img src="../../images/qcxms_predictions_ms_exp_pred.png" alt="Upper panel shows the experimental spectrum of ethanol, while the lower panel shows analogical spectrum but predicted with the current workflow. The predicted peaks correspond well to the experimental ones. Intensities of simulated peaks have not been predicted perfectly, but the most important trends are preserved." width="635" height="453" loading="lazy" /><a target="_blank" href="../../images/qcxms_predictions_ms_exp_pred.png" rel="noopener noreferrer"><small>Open image in new tab</small></a><br /><br /><figcaption><span class="figcaption-prefix"><strong>Figure 5</strong>:</span> Comparison between experimental (upper panel) and predicted (lower panel) mass spectra of ethanol.</figcaption></figure>

  <h1 id="conclusion-1">Conclusion</h1>
  <p><i class="fas fa-trophy" aria-hidden="true"></i><span class="visually-hidden">trophy</span> Well done, you’ve simulated mass spectra! You might want to consult your results with the <a href="https://usegalaxy.eu/u/hechth/h/end-to-end-ei-mass-spectra-prediction-workflow-using-qcxms-1">key history</a> or use <a href="https://usegalaxy.eu/u/hechth/h/end-to-end-ei-mass-spectra-prediction-workflow-using-qcxms-1">the workflow</a> associated with this tutorial.</p>

</div>

<p>The prediction of mass spectra might be very useful, particularly for compounds that lack experimental data. Simulating the spectra can also save time and resources. This field has been developing quite rapidly, and recent advancements in new algorithms and packages have led to more and more accurate results. However, one cannot forget that this kind of software should be used to deepen our chemical understanding of the structures of studied compounds and not as a replacement for practical experiments.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(typeof plausible !== 'undefined'){
                            // Plausible may be undefined (script blocked)
                            // or it may be defined, but opted-out (select box/DNT),
                            // which means `plausible()` will work but not send data, *nor* execute the callback.
                            plausible('TutorialComplete', {props: {path: document.location.pathname}})
                          }
                          // since the callback is completely cosmetic, we'll just issue it optimistically.
                          tutorial_finish_finish();
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Galaxy provides access to high-performance computing (HPC) resources and hence allows the use of semi-empirical quantum mechanical (SQM) methods and <em>in silico</em> prediction of QC-based mass spectra.</p>
</li>
                        
                        <li><p>The shown workflow allows to simulate mass spectra of a molecule starting from its SMILES.</p>
</li>
                        
                        <li><p>The xTB package is based on a SQM method to optimise the geometry of the generated conformers.</p>
</li>
                        
                        <li><p>The QCxMS tool suite performs quantum chemistry calculations to simulate mass spectra.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/metabolomics/faqs/">FAQ page for the Metabolomics topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                


                
                <h1 id="bibliography">References</h1>
                <ol class="bibliography"><li id="Lindsay1980ApplicationsOA">Lindsay, R. K., B. G. Buchanan, E. A. Feigenbaum, and J. Lederberg, 1980 <b>Applications of Artificial Intelligence for Organic Chemistry: The DENDRAL Project</b>. <a href="https://api.semanticscholar.org/CorpusID:60662239">https://api.semanticscholar.org/CorpusID:60662239</a></li>
<li id="Weininger1988">Weininger, D., 1988 <b>SMILES,  a chemical language and information system. 1. Introduction to methodology and encoding rules</b>. Journal of Chemical Information and Computer Sciences 28: 31–36. <a href="https://doi.org/10.1021/ci00057a005">10.1021/ci00057a005</a></li>
<li id="Vetter1994">Vetter, W., 1994 <b>F. W. McLafferty,  F. Turecek. Interpretation of mass spectra. Fourth edition (1993). University Science Books,  Mill Valley,  California</b>. Biological Mass Spectrometry 23: 379–379. <a href="https://doi.org/10.1002/bms.1200230614">10.1002/bms.1200230614</a></li>
<li id="Grimme2013">Grimme, S., 2013 <b>Towards First Principles Calculation of Electron Impact Mass Spectra of Molecules</b>. Angewandte Chemie International Edition 52: 6306–6312. <a href="https://doi.org/10.1002/anie.201300158">10.1002/anie.201300158</a></li>
<li id="Allen2014metabolomics">Allen, F., R. Greiner, and D. Wishart, 2014 <b>Competitive fragmentation modeling of ESI-MS/MS spectra for putative metabolite identification</b>. Metabolomics 11: 98–110. <a href="https://doi.org/10.1007/s11306-014-0676-4">10.1007/s11306-014-0676-4</a></li>
<li id="Allen2014">Allen, F., A. Pon, M. Wilson, R. Greiner, and D. Wishart, 2014 <b>CFM-ID: a web server for annotation,  spectrum prediction and metabolite identification from tandem mass spectra</b>. Nucleic Acids Research 42: W94–W99. <a href="https://doi.org/10.1093/nar/gku436">10.1093/nar/gku436</a></li>
<li id="Bauer2014">Bauer, C. A., and S. Grimme, 2014 <b>First principles calculation of electron ionization mass spectra for selected organic drug molecules</b>. Org. Biomol. Chem. 12: 8737–8744. <a href="https://doi.org/10.1039/c4ob01668h">10.1039/c4ob01668h</a></li>
<li id="Allen2016Computational">Allen, F., A. Pon, R. Greiner, and D. Wishart, 2016 <b>Computational Prediction of Electron Ionization Mass Spectra to Assist in GC/MS Compound Identification.</b> Analytical chemistry 88 15: 7689–97 . <a href="https://doi.org/10.1021/acs.analchem.6b01622">10.1021/acs.analchem.6b01622</a></li>
<li id="Bauer2016">Bauer, C. A., and S. Grimme, 2016 <b>How to Compute Electron Ionization Mass Spectra from First Principles</b>. The Journal of Physical Chemistry A 120: 3755–3766. <a href="https://doi.org/10.1021/acs.jpca.6b02907">10.1021/acs.jpca.6b02907</a></li>
<li id="DjoumbouFeunang2019">Djoumbou-Feunang, Y., A. Pon, N. Karu, J. Zheng, C. Li <i>et al.</i>, 2019 <b>CFM-ID 3.0: Significantly Improved ESI-MS/MS Prediction and Compound Identification</b>. Metabolites 9: 72. <a href="https://doi.org/10.3390/metabo9040072">10.3390/metabo9040072</a></li>
<li id="Koopman2019">Koopman, J., and S. Grimme, 2019 <b>Calculation of Electron Ionization Mass Spectra with Semiempirical GFNn-xTB Methods</b>. ACS Omega 4: 15120–15133. <a href="https://doi.org/10.1021/acsomega.9b02011">10.1021/acsomega.9b02011</a></li>
<li id="Wei2019">Wei, J. N., D. Belanger, R. P. Adams, and D. Sculley, 2019 <b>Rapid Prediction of Electron–Ionization Mass Spectrometry Using Neural Networks</b>. ACS Central Science 5: 700–708. <a href="https://doi.org/10.1021/acscentsci.9b00085">10.1021/acscentsci.9b00085</a></li>
<li id="Bannwarth2020">Bannwarth, C., E. Caldeweyher, S. Ehlert, A. Hansen, P. Pracht <i>et al.</i>, 2020 <b>Extended <scp>tight‐binding</scp> quantum chemistry methods</b>. WIREs Computational Molecular Science 11: <a href="https://doi.org/10.1002/wcms.1493">10.1002/wcms.1493</a></li>
<li id="Wang2021">Wang, S., T. Kind, D. J. Tantillo, and O. Fiehn, 2020 <b>Predicting in silico electron ionization mass spectra using quantum chemistry</b>. Journal of Cheminformatics 12: <a href="https://doi.org/10.1186/s13321-020-00470-3">10.1186/s13321-020-00470-3</a></li>
<li id="Koopman2021">Koopman, J., and S. Grimme, 2021 <b>From QCEIMS to QCxMS: A Tool to Routinely Calculate CID Mass Spectra Using Molecular Dynamics</b>. Journal of the American Society for Mass Spectrometry 32: 1735–1751. <a href="https://doi.org/10.1021/jasms.1c00098">10.1021/jasms.1c00098</a></li>
<li id="Zhu2023">Zhu, R. L., and E. Jonas, 2023 <b>Rapid Approximate Subset-Based Spectra Prediction for Electron Ionization–Mass Spectrometry</b>. Analytical Chemistry 95: 2653–2663. <a href="https://doi.org/10.1021/acs.analchem.2c02093">10.1021/acs.analchem.2c02093</a></li></ol>
                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=metabolomics/qcxms-predictions"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=metabolomics/qcxms-predictions">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Julia Jakiela, Helge Hecht,  <b>Predicting EI+ mass spectra with QCxMS (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html">https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{metabolomics-qcxms-predictions,
author = "Julia Jakiela and Helge Hecht",
	title = "Predicting EI+ mass spectra with QCxMS (Galaxy Training Materials)",
	year = "",
	month = "",
	day = "",
	url = "\url{https://training.galaxyproject.org/training-material/topics/metabolomics/tutorials/qcxms-predictions/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

        


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/metabolomics/tutorials/qcxms-predictions/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: ctb_im_conformers
  owner: bgruening
  revisions: a4b1474efada
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: openbabel_compound_convert
  owner: bgruening
  revisions: e2c36f62e22f
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: split_file_to_collection
  owner: bgruening
  revisions: 2dae863c8f42
  tool_panel_section_label: Collection Operations
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: 86755160afbf
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: fbf99087e067
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qcxms_getres
  owner: recetox
  revisions: 89b026d67e9f
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qcxms_neutral_run
  owner: recetox
  revisions: c1f752721827
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qcxms_neutral_run
  owner: recetox
  revisions: 8f465c0c3b6c
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: qcxms_production_run
  owner: recetox
  revisions: 356f7916ac07
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: xtb_molecular_optimization
  owner: recetox
  revisions: d5633eaf3552
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: xtb_molecular_optimization
  owner: recetox
  revisions: 6e1ef071fffc
  tool_panel_section_label: ChemicalToolBox
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
		   </table>
		   </p>
		
        
        
            <p>No feedback has been recieved yet for this training. Be the first one by filling in the feedback form from <a href="#gtn-feedback">above</a>. </p>
        
		</blockquote>




		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00458">gxy.io/GTN:T00458</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/metabolomics/tutorials/qcxms-predictions/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/metabolomics/tutorials/qcxms-predictions/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/43bed0043c7b744f5a6074fbdafe6dd7d31c4834">43bed00</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>