<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Galaxy FAQs</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@43973919866b066986c8bceb7a497d54a19de378",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/faqs/galaxy/aaaa_dontquestionthislinkitisthegluethatholdstogetherthegalaxy.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Galaxy FAQs">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2022-10-19 13:12:43 +0000"><meta name="description" content="Collection of tutorials developed and maintained by the worldwide Galaxy community">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Galaxy FAQs">
        <meta property="og:description" content="Collection of tutorials developed and maintained by the worldwide Galaxy community">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/faqs/galaxy/aaaa_dontquestionthislinkitisthegluethatholdstogetherthegalaxy.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/faqs/galaxy/aaaa_dontquestionthislinkitisthegluethatholdstogetherthegalaxy.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<section class="faqs">

   <nav aria-label="breadcrumb">
    <ol class="breadcrumb">

     <li class="breadcrumb-item "><a href="/training-material/faqs">All FAQs</a></li>
     
     <li class="breadcrumb-item "><a href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
     
     

     
     </ol>
    </nav>

    <h1 data-toc-skip>Frequently Asked Questions</h1>
    
    <div class="container">
        <div class="row">

            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h3 id="tutorial-faqs"> <a href="#tutorial-faqs"> Tutorial Questions </a></h3>

<hr/>



<h4 class="faq-area"> How to enable the Activity Bar <a href="/training-material/faqs/galaxy/activity_bar_how_to_enable.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This FAQ demonstrates how to enable the activity bar within the Galaxy interface
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-enable-the-activity-bar"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-enable-the-activity-bar" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to enable the Activity Bar</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you do not see the Activity Bar it can be enabled as follows:</p>   <ol>   <li>Click on the “<strong>User</strong>” link at the top of the Galaxy interface</li>   <li>Select “<strong>Preferences</strong>”</li>   <li>Scroll down and click on “<strong>Manage Activity Bar</strong>”</li>   <li>     <p>Toggle the “<strong>Enable Activity Bar</strong>” switch and voila!</p>     <p><img src="/training-material/shared/images/activity_bar_how_to_enable.svg" alt="The four steps described above are shown visually" /></p>   </li> </ol>   <!-- Original drawing = https://docs.google.com/drawings/d/1yynqEiV66F8idKH5JwTG-4ASuqK62thIwHeCMHRuupg/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="account"> <a href="#account"> Account </a></h3>

<hr/>



<h4 class="faq-area"> Can I create multiple Galaxy accounts? <a href="/training-material/faqs/galaxy/account_multiple.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-i-create-multiple-galaxy-accounts"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-i-create-multiple-galaxy-accounts" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can I create multiple Galaxy accounts?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The account registration form and activation email include a terms of service statement.</p>   <ul>   <li>You ARE NOT allowed to create more than 1 account per Galaxy server.</li>   <li>You ARE allowed to have accounts on different servers.</li> </ul>   <p>For example, you are allowed to have 1 account on Galaxy US, and another account on Galaxy EU, but never 2 accounts on the same Galaxy.</p>   <p><strong>WARNING: Having multiple accounts is a violation of the terms of service, and may result in deletion of your accounts.</strong></p>   <p><br /> Need more disk space?</p>   <ul>   <li>Review your <a href="https://my.galaxy.training/?path=/storage"><strong>User -&gt; Preferences -&gt; Storage Dashboard</strong></a> to find and manage all of your data.</li>   <li>Read about <a href="/training-material/faqs/galaxy/account_reduce_quota_usage.html">more ways to free up space in your account</a></li>   <li><a href="/training-material/faqs/galaxy/support_admins.html">Contact the admins of your Galaxy server</a> to ask about possibilities for temporarily increasing your quota.</li> </ul>   <p><br /> Other tips:</p>   <ul>   <li>Forgot your password? You can request a reset link in on the login page.</li>   <li>If you want to associate your account with a different email address, you can do so under <strong>User -&gt; Preferences</strong> in the top menu bar.</li>   <li>To start over with a new account, delete your existing account(s) first before creating your new account. This can be done in <strong>User -&gt; Preferences</strong> menu in the top bar.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing acount email or password <a href="/training-material/faqs/galaxy/user_preferences_manage_information.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-acount-email-or-password"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-acount-email-or-password" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing acount email or password</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Make sure you are logged in to Galaxy.</li>   <li>Go to <strong>User &gt; Preferences</strong> in the top menu bar.</li>   <li>To change email and public name, click on <strong>Manage Information</strong> and to change password, click on <strong>Change Password</strong>.</li>   <li>Make the changes and click on the <strong>Save</strong> button at the bottom.</li>   <li>To change email successfully, verify your account by email through the activation link sent by Galaxy.</li> </ol>   <p>Note: Don’t open another account if your email changes, update the existing account email instead. Creating a new account will be detected as a duplicate and will get your account disabled and deleted.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I reduce quota usage while still retaining prior work (data, tools, methods)? <a href="/training-material/faqs/galaxy/account_reduce_quota_usage.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-reduce-quota-usage-while-still-retaining-prior-work-data-tools-methods"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-reduce-quota-usage-while-still-retaining-prior-work-data-tools-methods" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I reduce quota usage while still retaining prior work (data, tools, methods)?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li><a href="/training-material/faqs/galaxy/datasets_download_datasets.html"><strong>Download</strong></a> Datasets as individual files or entire Histories as an archive. Then <em>purge</em> them from the public server.</li>   <li><strong>Transfer/Move</strong> <a href="/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html">Datasets</a> or <a href="/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html">Histories</a> to another Galaxy server, including your own Galaxy. Then purge.</li>   <li><a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><strong>Copy</strong></a> your most important Datasets into a new/other History (inputs, results), then purge the original full History.</li>   <li><a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><strong>Extract</strong></a> a <strong>Workflow</strong> from the History, then purge it.</li>   <li><strong>Back-up your work</strong>. It is a best practice to download an archive of your FULL original Histories periodically, even those still in use, as a backup.</li> </ul>   <p><strong>Resources</strong> Much discussion about all of the above options can be found at the <a href="https://help.galaxyproject.org/">Galaxy Help forum</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do I create an account on a public Galaxy instance? <a href="/training-material/faqs/galaxy/account_create.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-i-create-an-account-on-a-public-galaxy-instance"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-i-create-an-account-on-a-public-galaxy-instance" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do I create an account on a public Galaxy instance?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p>To create an account at any public Galaxy instance, choose your server from the available <a href="https://galaxyproject.org/use/">list of Galaxy Platforms</a>.</p>     <p>There are several UseGalaxy servers:</p>     <ul>       <li>         <p><a href="https://usegalaxy.fr">UseGalaxy.fr</a> (<strong>FR</strong>)</p>       </li>       <li>         <p><a href="https://usegalaxy.eu">UseGalaxy.eu</a> (<strong>EU</strong>)</p>       </li>       <li>         <p><a href="https://usegalaxy.org">UseGalaxy.org</a> (<strong>US</strong>)</p>       </li>       <li>         <p><a href="https://usegalaxy.org.au">UseGalaxy.org.au</a> (<strong>AU</strong>)</p>       </li>     </ul>   </li>   <li>     <p>Click on <strong>“Login or Register”</strong> in the masthead on the server.</p>     <p><img src="/training-material/faqs/galaxy/images/login_register.png" alt="Login or Register on the top panel" /></p>   </li>   <li>     <p>On the login page, find the <strong>Register here</strong> link and click on it.</p>   </li>   <li>     <p>Fill in the the registration form, then click on <strong>Create</strong>.</p>     <p>Your account should now get created, but will remain inactive until you verify the email address you provided in the registration form.</p>     <p><img src="/training-material/faqs/galaxy/images/unverified_account.png" alt="Banner warning about account with unverified email address" /></p>   </li>   <li>     <p>Check for a <strong>Confirmation Email</strong> in the email you used for account creation.</p>     <p>Missing? Check your <strong>Trash</strong> and <strong>Spam</strong> folders.</p>   </li>   <li>     <p>Click on the <strong>Email confirmation link</strong> to fully activate your account.</p>     <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> Delivery of the confimation email is blocked by your email provider or you mistyped the email address in the registration form?</p>     <p>Please do <strong>not</strong> register again, but follow the <a href="/training-material/faqs/galaxy/user_preferences_manage_information.html">instructions to change the email address</a> registered with your account! The confirmation email will be resent to your new address once you have changed it.</p>     <p>Trouble logging in later? Account email addresses and public names are caSe-sensiTive. Check your activation email for formats.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to update account preferences? <a href="/training-material/faqs/galaxy/account_update_preference.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-update-account-preferences"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-update-account-preferences" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to update account preferences?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Log in to Galaxy</li>   <li>Navigate to <a href="https://my.galaxy.training/?path=/user"><strong>User -&gt; Preferences</strong></a> on the top menu bar</li>   <li>Here you can update various preferences, such as:     <ul>       <li><i class="fas fa-user" aria-hidden="true"></i><span class="visually-hidden">pref-info</span> Manage Information (change your registered email addresses or public name)</li>       <li><i class="fas fa-unlock-alt" aria-hidden="true"></i><span class="visually-hidden">pref-password</span> Change Password (change your login credentials)</li>       <li><i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">pref-permissions</span> Set Dataset Permissions for New Histories (grant others default access to newly created histories)</li>       <li><i class="fas fa-filter" aria-hidden="true"></i><span class="visually-hidden">pref-toolboxfilters</span> Manage Toolbox Filters (customize your Toolbox by displaying or omitting sets of Tools)</li>       <li><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">pref-apikey</span> Manage API Key (access your current API key or create a new one)</li>       <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">pref-notifications</span> Manage Notifications (allow push and tab notifcations on job completion)</li>       <li><i class="fas fa-cloud" aria-hidden="true"></i><span class="visually-hidden">pref-cloud</span> Manage Cloud Authorization (grants Galaxy to access your cloud-based resources)</li>       <li><i class="far fa-id-card" aria-hidden="true"></i><span class="visually-hidden">pref-identities</span> Manage Third-Party Identities (connect or disconnect access to your third-party identities)</li>       <li><i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">pref-custombuilds</span> Manage Custom Builds (custom databases based on fasta datasets)</li>       <li><i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden">pref-list</span> Manage Activity Bar (a bonus navigation bar)</li>       <li><i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">pref-palette</span> Pick a Color Theme (interface color theme)</li>       <li><i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Make All Data Private (disable all data sharing)</li>       <li><i class="fas fa-radiation" aria-hidden="true"></i><span class="visually-hidden">pref-delete</span> Delete Account (on this Galaxy server)</li>       <li><i class="fa fa-sign-out" aria-hidden="true"></i><span class="visually-hidden">pref-signout</span> Sign out of Galaxy (signs you out of all sessions)</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="analysis"> <a href="#analysis"> Analysis </a></h3>

<hr/>



<h4 class="faq-area"> Adding a custom database/build (dbkey) <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy may have several reference genomes built-in, but you can also create your own.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-custom-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-custom-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a custom database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Navigate to the History that contains your fasta for the reference genome</li>   <li>Standarize the <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html">fasta format</a></li>   <li>In the top menu bar, go to <strong>User -&gt; Preferences -&gt; Manage Custom Builds</strong></li>   <li>Create a unique <strong>Name</strong> for your reference build</li>   <li>Create a unique <strong>Database</strong> (dbkey) for your reference build</li>   <li>Under <strong>Definition</strong>, select the option <code class="language-plaintext highlighter-rouge">FASTA-file from history</code></li>   <li>Under <strong>FASTA-file</strong>, select your fasta file</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Beware of Cuts <a href="/training-material/faqs/galaxy/analysis_cut.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy has several different cut tools
<!--SNIPPET--><blockquote class="warning">   <div class="box-title warning-title" id="warning-beware-of-cuts"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i> Warning: Beware of Cuts</div>   <p>The section below uses <strong>Cut</strong> tool. There are two <strong>cut</strong> tools in Galaxy due to historical reasons. This example uses tool with the full name <strong>Cut columns from a table (cut)</strong>. However, the same logic applies to the other tool. It simply has a slightly different interface.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Extended Help for Differential Expression Analysis Tools <a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-extended-help-for-differential-expression-analysis-tools"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-extended-help-for-differential-expression-analysis-tools" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Extended Help for Differential Expression Analysis Tools</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The error and usage help in this FAQ applies to most if not all Bioconductor tools.</p>   <ul>   <li>DEseq2</li>   <li>Limma</li>   <li>edgeR</li>   <li>goseq</li>   <li>Diffbind</li>   <li>StringTie</li>   <li>Featurecounts</li>   <li>HTSeq-count</li>   <li>HTseq-clip</li>   <li>Kalisto</li>   <li>Salmon</li>   <li>Sailfish</li>   <li>DEXSeq</li>   <li>DEXSeq-count</li>   <li>IsoformSwitchAnalyzeR</li> </ul>   <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span>  Review your error messages and you’ll find some clues about what may be going wrong and what needs to be adjusted in your rerun. If you are getting a message from <code class="language-plaintext highlighter-rouge">R</code>, that usually means the underlying tool could not read in or understand your inputs. This can be a labeling problem (what was typed on the form) or a content problem (data within the files).</p>   <p>Expect odd errors or content problems if any of the usage requirements below are not met.</p>   <p>General</p>   <ul>   <li>Are your reference genome, reference transcriptome, and reference annotation all based on the same genome assembly?     <ul>       <li>Check the identifiers in all inputs and adjust as needed.</li>       <li>These all may mean the same thing to a person but not to a computer or tool: chr1, Chr1, 1, chr1.1</li>     </ul>   </li>   <li>Differential expression tools all require sample count replicates. <a href="https://www.seqanswers.com/forum/bioinformatics/bioinformatics-aa/26388-deseq2-without-biol-replicates">Rationale from two of the DEseq tool authors</a>.     <ul>       <li>At least two factor levels/groups/conditions with two samples each.</li>       <li>All must all contain unique content for valid scientific results.</li>     </ul>   </li>   <li>Factor/Factor level names should only contain alphanumeric characters and optionally underscores.     <ul>       <li>Avoid starting these with a number and do not include spaces.</li>       <li>Galaxy <em>may</em> be able to normalize these values for you, but if you are getting an error: standardize the format yourself.</li>     </ul>   </li>   <li><strong>DEXSeq</strong> additionally requires that the first Condition is labeled as <code class="language-plaintext highlighter-rouge">Condition</code>.</li>   <li>If your count inputs have a header, the option <strong>Files have header?</strong> is set to <strong>Yes</strong>. If no headers, set to <strong>No</strong>.     <ul>       <li>If your files have more than one header line: keep the sample header line, remove all extra line(s).</li>     </ul>   </li>   <li>Make sure that tool form settings match your annotation content or the tool cannot match up the inputs!     <ul>       <li>If you are counting by <strong>gene_id</strong>, your annotation should contain gene_id attributes (9th column)</li>       <li>If you are summarizing by <strong>exon</strong>, your annotation should contain exon features (3rd column)</li>     </ul>   </li>   <li>Sometimes these tools do not understand <code class="language-plaintext highlighter-rouge">transcript_id.N</code> and <code class="language-plaintext highlighter-rouge">gene_id.N</code> notation (where N is a version number).     <ul>       <li>This notation could be in fasta or tabular inputs.</li>       <li>Try <a href="/training-material/search2.html?query=olympics">removing <code class="language-plaintext highlighter-rouge">.N</code> from all inputs</a>, and check for the accidential creation of new duplicates!</li>     </ul>   </li>   <li>Errors? <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Understanding the job log messages</a> can be confusing! But are accessible and worth reviewing.     <ul>       <li>The good news is that usage in Galaxy produces the same error messages as direct usage.</li>       <li>This means that a search at the <a href="https://support.bioconductor.org/">Bioconductor Support</a> website can provide useful clues! Come back to the <a href="https://help.galaxyproject.org/">Galaxy Help</a> forum with any remaining questions.</li>     </ul>   </li> </ul>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Remember, for any value in your inputs that is not a number, using only alphanumeric characters and optionally underscores <code class="language-plaintext highlighter-rouge">_</code> with no spaces is what the authors recommend. Check your factor names, sample names, gene identifiers, transcript identifiers, and header lines in files.</p>   <p>Reference genome (fasta)</p>   <ul>   <li>Can be a server reference genome (hosted index in the pull down menu) or a custom reference genome (fasta from the history).</li>   <li>Custom reference genomes must be <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html">formatted correctly</a>.</li>   <li>If you are using <strong>Salmon</strong> or <strong>Kalisto</strong>, you probably don’t need a reference genome but a reference transcriptome instead!</li>   <li>More about understanding and <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">working with large fasta datasets</a>.</li> </ul>   <p>Reference transcriptome (fasta)</p>   <ul>   <li>Fasta file containing assembled transcripts.</li>   <li>Unassembled short or long reads will not work as a substitute.</li>   <li>The transcript identifiers on the <code class="language-plaintext highlighter-rouge">&gt;seq</code> fasta lines must exactly match the <code class="language-plaintext highlighter-rouge">transcript_id</code> values in your annotation or tabular mapping file.</li> </ul>   <p>Reference annotation (tabular, GTF, GFF3)</p>   <ul>   <li>Reference annotation <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">in GTF format</a> works best.</li>   <li>If a GTF dataset is not available for your genome, a two-column tabular dataset containing <code class="language-plaintext highlighter-rouge">transcript &lt;tab&gt; gene</code> can be used instead with most of these tools.</li>   <li><strong>HTseq-count</strong> requires GTF attributes. Featurecounts is an alternative tool choice.</li>   <li>Sometimes the tool <strong>gffread</strong> is used to transform GFF3 data to GTF.</li>   <li>DO use UCSC’s reference annotation (GTF) and reference transcriptome (fasta) data from their <a href="https://hgdownload.soe.ucsc.edu/downloads.html">Downloads</a> area.     <ul>       <li>These are a match for the UCSC genomes indexed at public Galaxy servers.</li>       <li>Links can be directly copy/pasted into the Upload tool.</li>       <li>Allow Galaxy to <em>autodetect the datatype</em> to produce an uncompressed dataset in your history ready to use with tools.</li>     </ul>   </li>   <li>Avoid GTF data from the UCSC Table Browser: this leads to scientific problems. GTFs will have the same content populated for both the transcript_id and gene_id values. See the note at UCSC for more about why.</li>   <li>Still have problems? Try removing all GTF header lines with the tool <strong>Remove beginning of a file</strong>.</li>   <li>More about understanding and <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">working with GTF/GFF/GFF3 reference annotation</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I do analysis X? - Getting help <a href="/training-material/faqs/galaxy/analysis_getting_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-do-analysis-x-getting-help"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-do-analysis-x-getting-help" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I do analysis X? - Getting help</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you don’t know how to perform a certain analysis, you can ask the Galaxy community for help.</p>   <h4 id="where-to-ask">Where to ask</h4>   <p>The best places to ask your analysis questions are:</p>   <ul>   <li><a href="https://help.galaxyproject.org">Galaxy Help forum</a></li>   <li><a href="https://matrix.to/#/#Galaxy-Training-Network_Lobby:gitter.im">GTN Matrix chat</a></li> </ul>   <p><strong>Note:</strong> For questions about errors you’ve encountered in Galaxy, please see our <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">troubleshooting page</a>.</p>   <h4 id="how-to-ask">How to ask</h4>   <p>The more detail you provide, the better we can help you. Please provide information about:</p>   <ul>   <li><strong>Your data and experiment</strong> e.g. “paired-end RNASeq, mouse, 16 triplicates, 2 timepoints”, etc</li>   <li><strong>Your goal and research question</strong> e.g. “I want to detect diffentially expressed genes between these two groups and generate a volcano plot”</li>   <li><strong>What you have already tried?</strong> Do you already know which tools you want to use? Did you already try some but they didn’t work? Why not? Did you find good papers describing something similiar to what you want to do? etc.</li>   <li><strong>Which Galaxy are you using?</strong> And if you have already tried some steps, please <a href="/training-material/faqs/galaxy/histories_sharing.html">share your Galaxy history via URL</a> and provide this along with your question.</li>   <li><strong>Examples</strong>     <ul>       <li><strong>Bad Question:</strong> <em>“Help!!! How to perform metagenomics analysis. I need it urgent!”</em></li>       <li><strong>Good Question:</strong> <em>“Hello everybody, I have 16S rRNA sequencing data from Illumina, it was paired-end with 150bp reads. I want to perform a taxonomy analysis similar to this paper (provide link). I have followed this GTN tutorial (provide link), but my data is different because (reason) . How can I adapt this step of the analysis for my data? I read about a tool called X, but I cannot find it in Galaxy. I am using Galaxy EU, and here is a link to my history. Any help would be greatly appreciated!”</em></li>     </ul>   </li> </ul>   <h4 id="before-you-ask">Before you ask</h4>   <ul>   <li>Check the <a href="https://help.galaxyproject.org/">Galaxy Help forum</a> to see if others have already asked a similar question before.</li>   <li><a href="/training-material/search2.html">Search</a> the <a href="https://training.galaxyproject.org">GTN website</a> for a tutorial that matches what you want to do, and work your way through that. Even if it doesn’t doe exactly what you need, you usually learn a lot along the way that will help you adapt it to your own data or research question.</li> </ul>   <h4 id="be-patient">Be patient</h4>   <p>Please remember that most of the people answering questions on Matrix chat and the help forum are <strong>volunteers</strong> from the community. They take time out of their busy days to help you. They may also be in a different time zone, so it may take some time to get answers. Please always be patient and kind to each other, and adhere to our <a href="https://galaxyproject.org/community/coc/">code of conduct</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> My jobs aren't running! <a href="/training-material/faqs/galaxy/analysis_jobs_not_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-my-jobs-aren-t-running"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-my-jobs-aren-t-running" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: My jobs aren't running!</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p><strong>Please make sure you are logged in</strong>. At the top menu bar, you should see a section labeled “User”. If you see “Login/Register” here you are not logged in.</p>   </li>   <li><strong>Activate your account</strong>. If you have recently registered your account, you may first have to activate it. You will receive an e-mail with an activation link.     <ul>       <li>Make sure to check your spam folder!</li>     </ul>   </li>   <li>     <p><strong>Be patient</strong>. Galaxy is a free service, when a lot of people are using it, you may have to wait longer than usual (especially for ‘big’ jobs, e.g. alignments).</p>   </li>   <li><strong>Contact Support</strong>. If you really think something is wrong with the server, you can <a href="/training-material/faqs/galaxy/support.html">ask for support</a></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Pick the right Concatenate tool <a href="/training-material/faqs/galaxy/analysis_concatenate.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Most Galaxy servers will have two Concatenate tools installed - know which one to pick!
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-pick-the-right-concatenate-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-pick-the-right-concatenate-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Pick the right Concatenate tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>On most Galaxy servers you will find two <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> <strong>Concatenate datasets</strong> tools installed:</p>   <ol>   <li><strong>Concatenate datasets</strong> tail-to-head</li>   <li><strong>Concatenate datasets</strong> tail-to-head (cat)</li> </ol>   <p>The two tools have nearly identical interfaces, but behave differently in certain situations, specifically:</p>   <ul>   <li>     <p>The second tool, the one with “(cat)” in its name, simply concatenates everything you give to it into a single output dataset.</p>     <p>Whether you give it multiple datasets or a collection as the first parameter, or some datasets as the first and some others as the second parameter, it will always concatenate them all. In fact, the only reason for having multiple parameters for this tool is that by providing inputs through multiple parameters, you can make sure they are concatenated in the order you pass them in.</p>   </li>   <li>     <p>The first tool, on the other hand, will only ever concatenate inputs provided through <em>different</em> parameters.</p>     <p>This tool allows you to specify an arbitrary number of <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, but if you also want to use <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> multiple datasets or <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> a collection for some of the Dataset parameters, then all of these need to be of the same type (multiple datasets <em>or</em> collections) and have the same number of inputs.</p>     <p>Now depending on the inputs, one of the following behaviors will occur:</p>     <ul>       <li>If all the different inputs are <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, the tool will concatenate them all and produce a single output dataset.</li>       <li>If all the different inputs are specified <em>either</em> as <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> multiple datasets <em>or</em> as <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span>, and all have the same number of datasets, then the tool will concatenate the first datasets of each input parameter, the second datasets of each input parameter, the third, etc., and produce an output collection with as many elements as there are inputs per Dataset parameter.</li>       <li>In extension of the above, if <em>some</em> additional inputs are provided as <i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> single datasets, the content of these will be recycled and be reused in the concatenation of all the nth elements of the other parameters.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Reporting usage problems, security issues, and bugs <a href="/training-material/faqs/galaxy/analysis_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-reporting-usage-problems-security-issues-and-bugs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-reporting-usage-problems-security-issues-and-bugs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Reporting usage problems, security issues, and bugs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>For reporting <strong>Usage Problems</strong>, related to tools and functions, head to the <a href="https://help.galaxyproject.org/">Galaxy Help</a> site.     <ul>       <li><strong>Red Error Datasets</strong>:         <ul>           <li>Refer to the <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting errors</a> FAQ for red error in datasets.</li>         </ul>       </li>       <li><strong>Unexpected results in Green Success Dataset</strong>:         <ul>           <li>To resolve it you may be asked to send in a shared history link and possibly a shared workflow link. For sharing your history, refer to this <a href="/training-material/faqs/galaxy/histories_sharing.html">these instructions</a>.</li>           <li>To reach our support team, visit <a href="/training-material/faqs/galaxy/#support">Support FAQs</a>.</li>         </ul>       </li>       <li><strong>Functionality problems</strong>:         <ul>           <li>Using <a href="https://help.galaxyproject.org/">Galaxy Help</a> is the best way to get help in most cases.</li>           <li>If the problem is more complex, email a description of the problem and how to reproduce it.</li>         </ul>       </li>       <li><strong>Administrative problems</strong>:         <ul>           <li>If the problem is present in your own Galaxy, the administrative configuration may be a factor.</li>           <li>For the fastest help directly from the development community, admin issues can be alternatively reported to the mailing list or the GalaxyProject <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a>.</li>         </ul>       </li>     </ul>   </li>   <li>For <strong>Security Issues</strong>, do not report them via GitHub. Kindly disclose these as explained in this <a href="https://github.com/galaxyproject/galaxy/blob/dev/SECURITY.md">document</a>.</li>   <li>For <strong>Bug Reporting</strong>, create a Github issue. Include the steps mentioned <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">in these instructions</a>.</li>   <li>Search the <a href="https://training.galaxyproject.org/search">GTN Search</a> to find prior Q &amp; A, FAQs, tutorials, and other documentation across all Galaxy resources, to verify in case your issue was already faced by someone.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Results may vary <a href="/training-material/faqs/galaxy/analysis_results_may_vary.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Results may vary</div>   <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, external databases, or because of stochastic processes in the algorithms.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html"><i class="fas fa-external-link-alt"></i></a> </h4>
When you get a red dataset in your history, it means something went wrong. But how can you find out what it was? And how can you report errors?
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-errors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-errors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Troubleshooting errors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When something goes wrong in Galaxy, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your Galaxy errors.</p>   <ol>   <li><strong>Expand</strong> the red history dataset by clicking on it.     <ul>       <li>Sometimes you can already see an error message here</li>     </ul>   </li>   <li>     <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></p>   </li>   <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:     <ul>       <li>Expand the history item</li>       <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>       <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:         <ul>           <li>Tool Standard Output</li>           <li>Tool Standard Error</li>         </ul>       </li>       <li>For more information about specific tool errors, please see the <a href="/training-material/faqs/galaxy/index.html#troubleshooting">Troubleshooting section</a></li>     </ul>   </li>   <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.     <ul>       <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span></li>       <li>Write down any information you think might help solve the problem         <ul>           <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>         </ul>       </li>       <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> Report</strong> button</li>     </ul>   </li>   <li><strong>Ask for help!</strong>     <ul>       <li>Where?         <ul>           <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Matrix Channel</a></li>           <li>In the <a href="https://gitter.im/galaxyproject/Lobby">Galaxy Matrix Channel</a></li>           <li>Browse the <a href="https://help.galaxyproject.org/">Galaxy Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>         </ul>       </li>       <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Will my jobs keep running? <a href="/training-material/faqs/galaxy/analysis_jobs_keep_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-will-my-jobs-keep-running"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-will-my-jobs-keep-running" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Will my jobs keep running?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy is a fantastic system, but some users find themselves wondering:</p>   <blockquote>   <p>Will my jobs keep running once I’ve closed the tab? Do I need to keep my browser open?</p> </blockquote>   <p>No, you don’t! You can safely:</p>   <ol>   <li>Start jobs</li>   <li>Shut down your computer</li> </ol>   <p>and your jobs will keep running in the background! Whenever you next visit Galaxy, you can check if your jobs are still running or completed.</p>   <p><strong>However</strong>, this is not true for uploading data from your computer. You must wait for uploading a dataset <em>from your computer</em> to finish. (Uploading via URL is not affected by this, if you’re uploading from URL you can close your computer.)</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="collections"> <a href="#collections"> Collections </a></h3>

<hr/>



<h4 class="faq-area"> Adding a tag to a collection <a href="/training-material/faqs/galaxy/collections_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag-to-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag-to-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag to a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection in your history to view it</li>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name at the top of the history panel</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add a tag  starting with <code class="language-plaintext highlighter-rouge">#</code>     <ul>       <li>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs any tools using this dataset.</li>     </ul>   </li>   <li>Click <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span></li>   <li>Check that the tag appears below the collection name</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the datatype of a collection <a href="/training-material/faqs/galaxy/collections_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This will set the datatype for all files in your collection. Does not change the files themselves.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype-of-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype-of-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype of a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name in your history</li>   <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>Under <strong>new type</strong>, select  your desired datatype     <ul>       <li>tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ol>   <p><strong>Cannot find the feature?</strong></p>   <p>If you are on a smaller Galaxy server, i.e. not one of the large (multi)national public servers, you may not be able to find this operation, and there is no indication it is missing or why it is disabled.</p>   <p>Galaxy has recently started putting <a href="https://docs.galaxyproject.org/en/master/admin/production.html#use-celery-for-asynchronous-tasks">more features behind a setting and deployment configuration</a> that needs to be enabled by the server administrator. Your administrator will need to deploy Celery and potentially additionally flower and redis to their stack to enable changing the datatype of a collection. Consider sending your Galaxy administrator the link to the <a href="/training-material/topics/admin/tutorials/celeryless/tutorial.html">simpler deployment option</a> or more complex <a href="/training-material/topics/admin/tutorials/celery/tutorial.html">GTN tutorial for setting up redis and flower</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Converting the datatype of a collection <a href="/training-material/faqs/galaxy/collections_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This will convert all files in your collection to a different format. This will change the files themselves and create a new collection.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-converting-the-datatype-of-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-converting-the-datatype-of-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Converting the datatype of a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <strong>Edit</strong> <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> next to the collection name in your history</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>Under <strong>Converter Tool</strong>, select  your desired conversion</li>   <li>Click the <strong>Convert Collection</strong> button</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a dataset collection <a href="/training-material/faqs/galaxy/collections_build_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-dataset-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-dataset-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a dataset collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build Dataset List</strong></p>     <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>   </li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create collection</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a paired collection <a href="/training-material/faqs/galaxy/collections_build_list_paired.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-paired-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-paired-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a paired collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> <strong>Select Items</strong> at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Select Items button" /></li>   <li>Check all the datasets in your history you would like to include</li>   <li>     <p>Click <strong>n of N selected</strong> and choose <strong>Build List of Dataset Pairs</strong></p>   </li>   <li>Change the text of <em>unpaired forward</em> to a common selector for the forward reads</li>   <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse reads</li>   <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>   <li>Enter a name for your collection</li>   <li>Click <strong>Create List</strong> to build your collection</li>   <li>Click on the checkmark icon at the top of your history again</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming a collection <a href="/training-material/faqs/galaxy/collections_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-collection"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-collection" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a collection</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the collection</li>   <li>Click on the name of the collection at the top</li>   <li>Change the name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="collections, histories"> <a href="#collections, histories"> Collections, histories </a></h3>

<hr/>



<h4 class="faq-area"> Datasets versus collections <a href="/training-material/faqs/galaxy/histories_datasets_vs_collections.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explanation of why collections are needed and what they are
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-datasets-versus-collections"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-datasets-versus-collections" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Datasets versus collections</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Datasets versus collections</strong></p>   <p>In Galaxy’s history datasets can be present as individual entries or they can be combined into <em>Collections</em>. Why do we need collections? Collections combine multiple individual datasets into a single entity which is easy to manage. Galaxy tools can use collections directly as inputs. Collection can be <strong>simple</strong> or <strong>nested</strong>.</p>   <p><strong>Simple collections</strong></p>   <p>Imagine that you’ve uploaded a hundred FASTQ files corresponding to a hundred samples. These will appear as a hundred individual datasets in your history making it very long. But the chances are that when you analyze these data you will do the same thing on each dataset.</p>   <p>To simplify this process you can combine all hundred datasets into a single entity called a <em>dataset collection</em> (or simply a <em>collection</em> or a <em>list</em>). It will appear as a single box in your history making it much easier to understand. Galaxy tools are designed to take collections as inputs. So, for example, if you want to map each of these datasets against a reference genome using, say, <span><strong>Minimap2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span>, you will need to provide <code class="language-plaintext highlighter-rouge">minmap2</code> with just one input, the collection, and it will automatically start 100 jobs behind the scenes and will combine all outputs into a single collection containing BAM files.</p>   <p><img src="/training-material/shared/images/simple_collection.svg" alt="A simple collection is a container containing individual datasets" /></p>   <p>There is a number of situations when simple collections are not sufficient to reflect the complexity of the data. To deal with this situation Galaxy allows for <strong>nested</strong> collections.</p>   <p><strong>Nested collections</strong></p>   <p>Probably the most common example of this is pared end data when each sample is represented by two files: one containing forward reads and another containing reverse reads. In Galaxy you can create <strong>nested</strong> collection that reflects the hierarchy of the data. In the case of paired data Galaxy supports <strong>paired</strong> collections.</p>   <p><img src="/training-material/shared/images/paired_collection.svg" alt="A paired collection is a container containing individual datasets and preserving their hierarchy" /></p>   <!-- Original editable image for simple collections = https://docs.google.com/drawings/d/1A-tRerNLzC4FJfShUFT327wMvSSX4Y8AAdaxD_Fwaa0/edit?usp=sharing -->   <!-- Original editable image for paired collection = https://docs.google.com/drawings/d/1Bbx4UmIYdDAqK3KSm6LtLQ8zwEXxDglHmVbb0MK-mbQ/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="data upload"> <a href="#data upload"> Data upload </a></h3>

<hr/>



<h4 class="faq-area"> Data retrieval with “NCBI SRA Tools” (fastq-dump) <a href="/training-material/faqs/galaxy/dataupload_NCBI_SRA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-data-retrieval-with-ncbi-sra-tools-fastq-dump"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-data-retrieval-with-ncbi-sra-tools-fastq-dump" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Data retrieval with “NCBI SRA Tools” (fastq-dump)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This section will guide you through downloading experimental metadata, organizing the metadata to short lists corresponding to conditions and replicates, and finally importing the data from NCBI SRA in collections reflecting the experimental design.</p>   <p><strong>Downloading metadata</strong></p>   <ul>   <li>It is critical to understand the condition/replicate structure of an experiment before working with the data so that it can be imported as collections ready for analysis. Direct your browser to <a href="https://www.ncbi.nlm.nih.gov/Traces/study/">SRA Run Selector</a> and in the search box enter GEO data set identifier (for example: GSE72018). Once the study appears, click the box to download the “RunInfo Table”.</li> </ul>   <p><strong>Organizing metadata</strong></p>   <ul>   <li>The “RunInfo Table” provides the experimental condition and replicate structure of all of the samples. Prior to importing the data, we need to parse this file into individual files that contain the sample IDs of the replicates in each condition. This can be achieved by using a combination of the ‘group’, ‘compare two datasets’, ‘filter’, and ‘cut’ tools <strong>to end up with single column lists of sample IDs (SRRxxxxx) corresponding to each condition.</strong></li> </ul>   <p><strong>Importing data</strong></p>   <ul>   <li>Provide the files with SRR IDs to NCBI SRA Tools (fastq-dump) to import the data from SRA to Galaxy. By organizing the replicates of each condition in separate lists, <strong>the data will be imported as “collections” that can be directly loaded to a workflow or analysis pipeline.</strong></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Directly obtaining UCSC sourced *genome* identifiers <a href="/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-directly-obtaining-ucsc-sourced-genome-identifiers"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-directly-obtaining-ucsc-sourced-genome-identifiers" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Directly obtaining UCSC sourced *genome* identifiers</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Option 1</strong></p>   <ol>   <li>Go to <a href="http://genome.ucsc.edu/">UCSC Genome Browser</a>, navigate to “genomes”, then the species of interest.</li>   <li>On the home page for the genome build, immediately under the top navigation box, in the blue bar next to the full genome build name, you will find <strong>View sequences</strong> button.</li>   <li>Click on the <strong>View sequences</strong> button and it will take you to a detail page with a table listing out the contents.</li> </ol>   <p><strong>Option 2</strong></p>   <ol>   <li>Use the tool <strong>Get Data -&gt; UCSC Main</strong>.</li>   <li>In the Table Browser, choose the target genome and build.</li>   <li>For “group” choose the last option “All Tables”.</li>   <li>For “table” choose “chromInfo”.</li>   <li>Leave all other options at default and send the output to Galaxy.</li>   <li>This new dataset will load as a tabular dataset into your history.</li>   <li>It will list out the contents of the genome build, including the chromosome identifiers (in the first column).</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How can I  upload data using EBI-SRA? <a href="/training-material/faqs/galaxy/dataupload_EBI-SRA.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-can-i-upload-data-using-ebi-sra"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-can-i-upload-data-using-ebi-sra" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How can I  upload data using EBI-SRA?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Search for your data directly in the tool and use the Galaxy links.</li>   <li>Be sure to check your sequence data for correct quality score formats and the metadata “datatype” assignment.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing data from Sierra LIMS <a href="/training-material/faqs/galaxy/importing-data-from-sierra-lims.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-sierra-lims"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-sierra-lims" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from Sierra LIMS</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This section will guide you through generating external links to your data stored in the Sierra LIMS system to be downloaded directly into Galaxy.</p>   <ol>   <li>Go to the <a href="https://www.bioinformatics.babraham.ac.uk/sierra/sierra.pl">Sierra portal</a> and login to your account.</li>   <li>Click on the <strong>Sample ID</strong> of the sample you want to download data from.</li>   <li>Click on the <strong>Edit Sample Details</strong> button.</li>   <li>At the bottom of the page there will be an input box for creating a link, enter a description for the link in the <strong>Reason for link</strong> section, and click <strong>Create link</strong>. This will reload the page and add a new link to the sample under <strong>Authorised links to this sample</strong>.</li>   <li>Go back to the sample page or click on the hyperlink called <strong>link</strong> to take you back.</li>   <li>In the <strong>Results</strong> section select the lane you want to access your data from.</li>   <li>The bottom of the page, under the <strong>Links</strong> section, will now contain a list of <code class="language-plaintext highlighter-rouge">wget</code> commands with links for accessing all the files within that sample/lane.</li>   <li>Since this list is for <code class="language-plaintext highlighter-rouge">wget</code> commands, you need to extract out the links from the command. You can copy the link in the first set of double quotes for each line and <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> them directly into Galaxy to download the files.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing data from a data library <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing data from remote files <a href="/training-material/faqs/galaxy/datasets_import_from_remote_files.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-remote-files"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-remote-files" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from remote files</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>Choose remote files</em>:</p>   <ol>   <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>   <li>     <p>Click on <strong>Choose remote files</strong> and scroll down to find your data folder or type the folder name in the <code class="language-plaintext highlighter-rouge">search box</code> on the top.</p>   </li>   <li>click on <strong>OK</strong></li>   <li>Click on <strong>Start</strong></li>   <li>Click on <strong>Close</strong></li>   <li>You can find the dataset has begun loading in you history.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing via links <a href="/training-material/faqs/galaxy/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> NCBI SRA sourced fastq data <a href="/training-material/faqs/galaxy/dataupload_NCBI_SRA_sourced_fastq_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-ncbi-sra-sourced-fastq-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-ncbi-sra-sourced-fastq-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: NCBI SRA sourced fastq data</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In these FASTQ data:</p>   <ul>   <li>The quality score identifier (+) is sometimes not a match for the sequence identifier (@).</li>   <li>The forward and reverse reads may be interlaced and need to be separated into distinct datasets.</li>   <li>Both may be present in a dataset. Correct the first, then the second, as explained below.</li>   <li>Format problems of any kind can cause tool failures and/or unexpected results.</li>   <li>Fix the problems before running any other tools (including <strong>FastQC</strong>, <strong>Fastq Groomer</strong>, or other QA tools)</li> </ul>   <p><strong>For inconsistent sequence (@) and quality (+) identifiers</strong></p>   <ul>   <li>     <p>Correct the format by running the tool <strong>Replace Text in entire line</strong> with these options:</p>     <ul>       <li>Find pattern: <code class="language-plaintext highlighter-rouge">^\+SRR.+</code></li>       <li>Replace with: <code class="language-plaintext highlighter-rouge">+</code></li>     </ul>   </li> </ul>   <p>Note: If the quality score line is named like “+ERR” instead (or other valid options), modify the pattern search to match.</p>   <p><strong>For interlaced forward and reverse reads</strong></p>   <p><strong>Solution 1 (reads named /1 and /2)</strong></p>   <ul>   <li>Use the tool <strong>FASTQ de-interlacer on paired end reads</strong></li> </ul>   <p><strong>Solution 2 (reads named /1 and /2)</strong></p>   <ul>   <li>Create distinct datasets from an interlaced fastq dataset by running the tool <strong>Manipulate FASTQ reads on various attributes</strong> on the original dataset. It will run twice.</li> </ul>   <p>Note: The solution does NOT use the <strong>FASTQ Splitter</strong> tool. The data to be manipulated are interlaced sequences. This is different in format from data that are joined into a single sequence.</p>   <ul>   <li>     <p>Use the Manipulate FASTQ settings to produce a dataset that contains the <code class="language-plaintext highlighter-rouge">/1</code> reads**</p>     <p><em>Match Reads</em></p>     <ul>       <li>Match Reads by <code class="language-plaintext highlighter-rouge">Name/Identifier</code></li>       <li>Identifier Match Type <code class="language-plaintext highlighter-rouge">Regular Expression</code></li>       <li>Match by <code class="language-plaintext highlighter-rouge">.+/2</code></li>     </ul>     <p><em>Manipulate Reads</em></p>     <ul>       <li>Manipulate Reads by <code class="language-plaintext highlighter-rouge">Miscellaneous Actions</code></li>       <li>Miscellaneous Manipulation Type <code class="language-plaintext highlighter-rouge">Remove Read</code></li>     </ul>   </li>   <li>     <p>Use these Manipulate FASTQ settings to produce a dataset that contains the <code class="language-plaintext highlighter-rouge">/2</code> reads**</p>     <ul>       <li>Exact same settings as above except for this change: Match by <code class="language-plaintext highlighter-rouge">.+/1</code></li>     </ul>   </li> </ul>   <p><strong>Solution 3 (reads named /1 and /3)</strong></p>   <ul>   <li>Use the same operations as in Solution 2 above, except change the first <strong>Manipulate FASTQ</strong> query term to be:</li>   <li>Match by <code class="language-plaintext highlighter-rouge">.+/3</code></li> </ul>   <p><strong>Solution 4 (reads named without /N)</strong></p>   <ul>   <li>If your data has differently formatted sequence identifiers, the “Match by” expression from <em>Solution 2</em> above can be modified to suit your identifiers.</li> </ul>   <p>Alternative identifiers such as:</p>   <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>@M00946:180:000000000-ANFB2:1:1107:14919:14410 1:N:0:1<br>@M00946:180:000000000-ANFB2:1:1107:14919:14410 2:N:0:1<br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload datasets from GenomeArk <a href="/training-material/faqs/galaxy/datasets_upload_from_genomeark.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-datasets-from-genomeark"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-datasets-from-genomeark" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload datasets from GenomeArk</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the file <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>upload</strong> menu</li>   <li>Click on <strong>Choose remote files</strong> tab</li>   <li>Click on the <strong>Genome Ark</strong> button and then click on <strong>species</strong></li> </ol>   <p>You can find the data by following this path: <code class="language-plaintext highlighter-rouge">/species/${Genus}_${species}/${specimen_code}/genomic_data</code>. Inside a given datatype directory (<em>e.g.</em> <code class="language-plaintext highlighter-rouge">pacbio</code>), select all the relevant files individually until all the desired files are highlighted and click the <kbd>Ok</kbd> button. Note that there may be multiple pages of files listed. Also note that you may not want every file listed.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload few files (1-10) <a href="/training-material/faqs/galaxy/datasets_upload.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-few-files-1-10"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-few-files-1-10" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload few files (1-10)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>   <li>Click on <strong>Choose local file</strong> and select the files or drop the files in the <strong>Drop files here</strong> part</li>   <li>Click on <strong>Start</strong></li>   <li>Click on <strong>Close</strong></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Upload many files (>10) via FTP <a href="/training-material/faqs/galaxy/datasets_upload_ftp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-upload-many-files-10-via-ftp"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-upload-many-files-10-via-ftp" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Upload many files (&gt;10) via FTP</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some Galaxies offer FTP upload for very large datasets.</p>   <p><strong>Note:</strong> the <em>“Big Three”</em> Galaxies (Galaxy Main, Galaxy EU, and Galaxy Australia) no longer support FTP upload, due to the recent improvements of the default web upload, which should now support large file uploads and almost all use cases. For situations where uploading via the web interface is too tedious, the <a href="https://github.com/galaxyproject/galaxy-upload">galaxy-upload commandline utility</a> is also available as an alternative to FTP.</p>   <p>To upload files via FTP, please</p>   <ol>   <li>     <p>Check that your Galaxy supports FTP upload and look up the FTP settings.</p>   </li>   <li>     <p>Make sure to have an FTP client installed</p>     <p>There are many options. We can recommend <a href="https://filezilla-project.org/">FileZilla</a>, a free FTP client that is available on Windows, MacOS, and Linux.</p>   </li>   <li>Establish FTP connection to the Galaxy server     <ol>       <li>Provide the Galaxy server’s FTP server name (e.g. <code class="language-plaintext highlighter-rouge">ftp.mygalaxy.com</code>)</li>       <li>Provide the <strong>username</strong> (usually the e-mail address) and the <strong>password</strong> on the Galaxy server</li>       <li>Connect</li>     </ol>   </li>   <li>     <p>Add the files to the FTP server by dragging/dropping them or right clicking on them and uploading them</p>     <p>The FTP transfer will start. We need to wait until they are done.</p>   </li>   <li>Open the Upload menu on the Galaxy server</li>   <li>Click on <strong>Choose FTP file</strong> on the bottom</li>   <li>Select files to import into the history</li>   <li>Click on <strong>Start</strong></li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="datasets"> <a href="#datasets"> Datasets </a></h3>

<hr/>



<h4 class="faq-area"> Adding a tag <a href="/training-material/faqs/galaxy/datasets_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tags can help you to better organize your history and track datasets.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing database/build (dbkey) <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can tell Galaxy which dbkey (e.g. reference genome) your dataset is associated with. This may be used by tools to automatically use the correct settings.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-database-build-dbkey"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-database-build-dbkey" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing database/build (dbkey)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click the desired dataset’s name to expand it.</li>   <li>     <p>Click on the “?” next to database indicator:</p>     <p><img src="/training-material/shared/images/datasets_dbkey.svg" alt="UI for changing dbkey" /></p>   </li>   <li>In the central panel, change the <strong>Database/Build</strong> field</li>   <li>Select your desired database key from the dropdown list</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the datatype <a href="/training-material/faqs/galaxy/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select  your desired datatype  from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Converting the file format <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Some datasets can be transformed into a different format. Galaxy has some built-in file conversion options depending on the type of data you have.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-converting-the-file-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-converting-the-file-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Converting the file format</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong> tab on the top</li>   <li>In the upper part <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> <strong>Convert</strong>, select the appropriate datatype from the list</li>   <li>Click the <strong>Create dataset</strong> button to start the conversion.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a new file <a href="/training-material/faqs/galaxy/datasets_create_new_file.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy allows you to create new files from the upload menu. You can supply the contents of the file.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-file"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-file" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new file</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong> at the bottom</li>   <li>Paste the file contents into the text field</li>   <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Datasets not downloading at all <a href="/training-material/faqs/galaxy/datasets_not_downloading_at_all.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-datasets-not-downloading-at-all"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-datasets-not-downloading-at-all" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Datasets not downloading at all</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Check to see if pop-ups are blocked by your web browser. Where to check can vary by browser and extensions.</li>   <li>Double check your API key, if used. Go to <strong>User &gt; Preferences &gt; Manage API key.</strong></li>   <li>Check the sharing/permission status of the Datasets. Go to <strong>Dataset &gt; Pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> &gt; Edit attributes &gt; Permissions.</strong> If you do not see a “Permissions” tab, then you are not the owner of the data.</li> </ol>   <p>Notes:</p>   <ul>   <li>If the data was shared with you by someone else from a Shared History, or was copied from a Published History, be aware that there are multiple levels of data sharing permissions.</li>   <li>All data are set to <strong>not shared</strong> by default.</li>   <li>Datasets sharing permissions for a new history can be set before creating a new history. Go to <strong>User &gt; Preferences &gt; Set Dataset Permissions for New Histories.</strong></li>   <li><strong>User &gt; Preferences &gt; Make all data private is a “one click” option to unshare ALL data (Datasets, Histories).</strong> Note that once confirmed and all data is unshared, the action cannot be “undone” in batch, even by an administrator. You will need to re-share data again and/or reset your global sharing preferences as wanted.</li>   <li>Only the data owner has control over sharing/permissions.</li>   <li>Any data you upload or create yourself is automatically owned by you with full access.</li>   <li>You may not have been granted full access if the data were shared or imported, and someone else is the data owner (your copy could be “view only”).</li>   <li>After you have a fully shared copy of any shared/published data from someone else, then you become the owner of that data copy. If the other person or you make changes, it applies to each person’s copy of the data, individually and only.</li>   <li>Histories can be shared with included Datasets. Datasets can be downloaded/manipulated by others or viewed by others.</li>   <li>Share access to Datasets is distinct but it relates to Histories’ access.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Detecting the datatype (file format) <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-detecting-the-datatype-file-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-detecting-the-datatype-file-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Detecting the datatype (file format)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>Click the <strong>Auto-detect</strong> button to have Galaxy try to autodetect it.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Different dataset icons and their usage <a href="/training-material/faqs/galaxy/datasets_icons.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Icons provide a visual experience for objects, actions, and ideas
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-different-dataset-icons-and-their-usage"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-different-dataset-icons-and-their-usage" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Different dataset icons and their usage</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Dataset icons and their usage:</p>   <ul>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> <strong>“Eye icon”</strong>: Display dataset contents.</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>“Pencil icon”</strong>: Edit attributes of dataset metadata: labels, datatype, database.</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> <strong>“Trash icon”</strong>: Delete the dataset.</li>   <li><i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> <strong>“Disc icon”</strong>: Download the dataset.</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> <strong>“Copy link”</strong>: Copy link URL to the dataset.</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span> <strong>“Info icon”</strong>: Dataset details and job runtime information: inputs, parameters, logs.</li>   <li><i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> <strong>“Refresh/Rerun icon”</strong>: Run this (selected) job again or examine original submitted form.</li>   <li><i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> <strong>“Visualize icon”</strong>: External display links (UCSC, IGV, NPL, PV); Charts and graphing; Editor (manually edit text).</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">galaxy-dataset-map</span> <strong>“Dataset Map icon”</strong>: Filter the history for related Input/Output Datasets. Click again to clear the filter.</li>   <li><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> <strong>“Bug icon”</strong>: Review subset of logs (review all under <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span>), and optionally submit a bug report.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading datasets <a href="/training-material/faqs/galaxy/datasets_download_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the dataset in your history to expand it</li>   <li>Click on the Download icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span> to save the dataset to your computer.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading datasets using command line <a href="/training-material/faqs/galaxy/datasets_download_using_command_line.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-datasets-using-command-line"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-datasets-using-command-line" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading datasets using command line</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>From the terminal window on your computer, you can use <strong>wget</strong> or <strong>curl</strong>.</p>   <ol>   <li>Make sure you have <strong>wget</strong> or <strong>curl</strong> installed.</li>   <li>Click on the Dataset name, then click on the copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span>. This is the direct-downloadable dataset link.</li>   <li>Once you have the link, use any of the following commands:     <ul>       <li>For <strong>wget</strong>         <blockquote>           <p><strong><code class="language-plaintext highlighter-rouge">wget '&lt;link&gt;'</code> <br /> <code class="language-plaintext highlighter-rouge">wget -O '&lt;link&gt;'</code><br /> <code class="language-plaintext highlighter-rouge">wget -O --no-check-certificate '&lt;link&gt;'  # ignore SSL certificate warnings</code> <br /> <code class="language-plaintext highlighter-rouge">wget -c '&lt;link&gt;' # continue an interrupted download</code></strong></p>         </blockquote>       </li>       <li>For <strong>curl</strong>         <blockquote>           <p><strong><code class="language-plaintext highlighter-rouge">curl -o outfile '&lt;link&gt;'</code><br /> <code class="language-plaintext highlighter-rouge">curl -o outfile --insecure '&lt;link&gt;'     # ignore SSL certificate warnings</code><br /> <code class="language-plaintext highlighter-rouge">curl -C - -o outfile '&lt;link&gt;'           # continue an interrupted download</code></strong></p>         </blockquote>       </li>     </ul>   </li>   <li>For dataset collections and datasets within collections you have to supply your <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html">API key</a> with the request     <ul>       <li>Sample commands for <strong>wget</strong> and <strong>curl</strong> respectively are:         <blockquote>           <p><strong><code class="language-plaintext highlighter-rouge">wget https://usegalaxy.org/api/dataset_collections/d20ad3e1ccd4595de/download?key=MYSECRETAPIKEY</code></strong></p>           <p><strong><code class="language-plaintext highlighter-rouge">curl -o myfile.txt https://usegalaxy.org/api/dataset_collections/d20ad3e1ccd4595de/download?key=MYSECRETAPIKEY</code></strong></p>         </blockquote>       </li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding BAM dataset identifiers <a href="/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Quickly learn what the identifiers are in any **BAM** dataset that is the result from mapping
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-bam-dataset-identifiers"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-bam-dataset-identifiers" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding BAM dataset identifiers</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Run <strong>Samtools: IdxStats</strong> on the aligned data (<code class="language-plaintext highlighter-rouge">bam</code> dataset).</li>   <li>The “index header” chromosome names and lengths will be listed in the output (along with read counts).</li>   <li>Compare the chromosome identifiers to the chromosome (aka “chrom”) field in all other inputs: VCF, GTF, GFF(3), BED, Interval, etc.</li> </ol>   <p><strong>Note:</strong></p>   <ul>   <li>The original mapping target may have been a built-in genome index, custom genome (transcriptome, exome, other) – the same <code class="language-plaintext highlighter-rouge">bam</code> data will still be summarized.</li>   <li>This method will <em>not</em> work for “sequence-only” <code class="language-plaintext highlighter-rouge">bam</code> datasets, as these usually have no header.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding Datasets <a href="/training-material/faqs/galaxy/datasets_finding_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding Datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>To review all active Datasets in your account, go to <strong>User &gt; Datasets</strong>.</li> </ul>   <p>Notes:</p>   <ul>   <li>Logging out of Galaxy while the Upload tool is still loading data can cause uploads to abort. This is most likely to occur when a dataset is loaded by browsing local files.</li>   <li>If you have more than one browser window open, each with a different Galaxy History loaded, the Upload tool will load data into the most recently used history.</li>   <li>Click on refresh icon <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> at the top of the History panel to display the current active History with the datasets.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to delete datasets? <a href="/training-material/faqs/galaxy/datasets_deleting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-delete-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-delete-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to delete datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Deleting datasets individually</strong></p>   <p>To delete datasets individually simply click the <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> button with dataset’s box. That’s it! This action is reversible: datasets can be undeleted.</p>   <p><strong>Deleting datasets in bulk</strong></p>   <p>To delete multiple datasets at once:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to delete;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Delete”</strong> option.</li> </ul>   <p>This action is also reversible: datasets can be undeleted.</p>   <p><img src="/training-material/shared/images/datasets_deleting.gif" alt="An animated gif showing how to delete datasets" /></p>   <p><strong>Deleting datasets permanently</strong> <font color="red"><i class="fas fa-exclamation-triangle" aria-hidden="true"></i><span class="visually-hidden">warning</span> <b>Danger zone!</b></font></p>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-permanent-is-permanent"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Permanent is ... PERMANENT!</div>   <p>Datasets deleted in this fashion CANNOT be undeleted!</p> </blockquote>   <p>To delete multiple datasets <font color="red">PERMANENTLY</font>:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to delete;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Delete (permanently)”</strong> option.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to hide datasets? <a href="/training-material/faqs/galaxy/datasets_hidden.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-hide-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-hide-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to hide datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To hide datasets:</p>   <ul>   <li>Click <i class="fas fa-check-square" aria-hidden="true"></i><span class="visually-hidden">history-select-multiple</span> icon at the top of the history pane;</li>   <li>Select datasets you want to hide;</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Hide”</strong> option.</li> </ul>   <p><img src="/training-material/shared/images/datasets_hide.gif" alt="An animated gif showing how to hide datasets" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to un-delete datasets? <a href="/training-material/faqs/galaxy/datasets_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-un-delete-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-un-delete-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to un-delete datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If your history contains deleted datasets you will see <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> <strong>“Include deleted”</strong> button directly above dataset display.</p>   <p>To un-delete datasets:</p>   <ul>   <li>Type <code class="language-plaintext highlighter-rouge">deleted:true</code> in the search box</li>   <li>Select datasets you want to un-delete</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Undelete”</strong> option.</li> </ul>   <p><img src="/training-material/shared/images/datasets_undeleting.gif" alt="An animated gif showing how to undelete datasets" /></p>   <p>Alternatively, you can:</p>   <ul>   <li>click <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> <strong>“Include deleted”</strong> button directly above dataset display. This will cause deleted datasets to appear in history along with normal (un-deleted) datasets;</li>   <li>deleted datasets are distinguished by having <i class="fas fa-trash-can-arrow-up" aria-hidden="true"></i><span class="visually-hidden">dataset-undelete</span> within dataset box. Clicking on this icon will un-delete a given dataset;</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to un-hide datasets? <a href="/training-material/faqs/galaxy/datasets_unhidden.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-un-hide-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-un-hide-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to un-hide datasets?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If your history contains hidden datasets you will see <i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">galaxy-show-hidden</span> <strong>“Include hidden”</strong> button directly above the dataset display.</p>   <p>To un-hide datasets:</p>   <ul>   <li>Type <code class="language-plaintext highlighter-rouge">visible:hidden</code> in the search box</li>   <li>Select datasets you want to un-hide</li>   <li>Click the dropdown that would appear at the top of the history;</li>   <li>Select <strong>“Unhide”</strong> option.</li> </ul>   <p><img src="/training-material/shared/images/datasets_unhide.gif" alt="An animated gif showing how to unhide datasets" /></p>   <p>Alternatively, you can:</p>   <ul>   <li>click <i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">galaxy-show-hidden</span> <strong>“Include hidden”</strong> button directly above dataset display. This will cause hidden datasets to appear in history along with normal (un-hidden) datasets;</li>   <li>hidden datasets are distinguished by having <i class="fas fa-eye-slash" aria-hidden="true"></i><span class="visually-hidden">galaxy-show-hidden</span> within dataset box. Clicking on this icon will un-hide a given dataset;</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Mismatched Chromosome identifiers and how to avoid them <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-mismatched-chromosome-identifiers-and-how-to-avoid-them"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-mismatched-chromosome-identifiers-and-how-to-avoid-them" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Mismatched Chromosome identifiers and how to avoid them</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Reference data mismatches are similiar to bad reagents in a wet lab experiment: all sorts of odd problems can come up!</p>   <p>You inputs must be all based on an identical genome assembly build to achieve correct scientific results.</p>   <p>There are two areas to review for data to be considered <strong>identical</strong>.</p>   <ol>   <li>The data are based on the same exact genome <strong>assembly</strong> (or “assembly release”).     <ul>       <li>The “assembly” refers to the nucleotide sequence of the genome.</li>       <li>If the base order and length of the chromosomes are not the same, then your coordinates will have scientific problems.</li>       <li>Converting coordinates between assemblies may be possible. Search tool panel with <code class="language-plaintext highlighter-rouge">CrossMap</code>.</li>     </ul>   </li>   <li>The data are based on the same exact genome assembly <strong>build</strong>.     <ul>       <li>The “build” refers to the labels used inside the file. In this context, pay attention to the chromosome identifiers.</li>       <li>These all may mean the same thing to a person but not to a computer or tool: chr1, Chr1, 1, chr1.1</li>       <li>Converting identifiers between builds may be possible. Search tool panel with <code class="language-plaintext highlighter-rouge">Replace</code>.</li>     </ul>   </li> </ol>   <p>The methods listed below help to identify and correct errors or unexpected results when the underlying genome assembly build for all inputs are <strong>not identical</strong>.</p>   <p><strong>Method 1</strong>: <a href="/training-material/faqs/galaxy/datasets_BAM_dataset_identifiers.html">Finding BAM dataset identifiers</a></p>   <p><strong>Method 2</strong>: <a href="/training-material/faqs/galaxy/datasets_UCSC_sourced_genome.html">Directly obtaining UCSC sourced <em>genome</em> identifiers</a></p>   <p><strong>Method 3</strong>: <a href="https://galaxyproject.org/support/chrom-identifiers/#adjusting-identifiers-or-input-source">Adjusting identifiers for UCSC sourced data used with other sourced data</a></p>   <p><strong>Method 4</strong>: <a href="https://galaxyproject.org/support/chrom-identifiers/#any-mixed-sourced-data">Adjusting identifiers or input source for any mixed sourced data</a></p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Reference data is self referential. <a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html">More help for your genome, transcriptome, and annotation</a></p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Genome not available as a native index? <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html">Use a custom genome fasta</a> and <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">create a custom build database</a> instead.</p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> More notes on Native Reference Genomes</p>   <ul>   <li>Native <strong>reference genomes</strong> (FASTA) are built as pre-computed indexes on the Galaxy server where you are working.</li>   <li>Different servers host both common <em>and</em> different reference genome data.</li>   <li>Most <strong>reference annotation</strong> (tabular, GTF, GFF3) is supplied from the history by the user, even when the genome is indexed.</li>   <li>Public Galaxy servers source reference genomes preferentially from <a href="https://hgdownload.soe.ucsc.edu/downloads.html">UCSC</a>.</li>   <li>A <strong>reference transcriptome</strong> (FASTA) is supplied from the history by the user.</li>   <li>Many experiements use a combination of all three types of reference data. Consider pre-preparing your files at the start!</li>   <li>The default variant for a native genome index is “Full”. Defined as: all primary chromosomes (or scaffolds/contigs) including mitochondrial plus associated unmapped, plasmid, and other segments.</li>   <li>When only one version of a genome is available for a tool, it represents the default “Full” variant.</li>   <li>Some genomes will have more than one variant available.</li>   <li>The “Canonical Male” or sometimes simply “Canonical” variant contains the primary chromosomes for a genome. For example a human “Canonical” variant contains chr1-chr22, chrX, chrY, and chrM.</li>   <li>The “Canonical Female” variant contains the primary chromosomes excluding chrY.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Moving datasets between Galaxy servers <a href="/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-moving-datasets-between-galaxy-servers"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-moving-datasets-between-galaxy-servers" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Moving datasets between Galaxy servers</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>On the origin Galaxy server:</p>   <ol>   <li>Click on the name of the dataset to expand the info.</li>   <li>Click on the <strong>Copy link icon</strong> <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span>.</li> </ol>   <p>On the destination Galaxy server:</p>   <ol>   <li>Click on <strong>Upload data &gt; Paste / Fetch Data</strong> and paste the link. Select attributes, such as genome assembly, if required. Hit the <strong>Start</strong> button.</li> </ol>   <p>Note: The copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> cannot be used to move HTML datasets (but this can be downloaded using the download button <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>) and SQLite datasets.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Purging datasets <a href="/training-material/faqs/galaxy/datasets_purging_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-purging-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-purging-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Purging datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>All account Datasets can be reviewed under <strong>User &gt; Datasets</strong>.</li>   <li>To permanently delete: use the link from within the dataset, or use the Operations on Multiple Datasets functions, or use the Purge Deleted Datasets option in the History menu.</li> </ol>   <p>Notes:</p>   <ul>   <li>Within a History, deleted/permanently deleted Datasets can be reviewed by toggling the deleted link at the top of the History panel, found immediately under the History name.</li>   <li>Both active (shown by default) and hidden (the other toggle link, next to the deleted link) datasets can be reviewed the same way.</li>   <li>Click on the far right “X” to delete a dataset.</li>   <li>Datasets in a deleted state are still part of your quota usage.</li>   <li>Datasets must be purged (permanently deleted) to not count toward quota.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Quotas for datasets and histories <a href="/training-material/faqs/galaxy/datasets_quotas_for_datasets_and_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-quotas-for-datasets-and-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-quotas-for-datasets-and-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Quotas for datasets and histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li><em>Deleted datasets</em> and <em>deleted histories</em> containing datasets <strong>are considered when calculating quotas</strong>.</li>   <li><em>Permanently deleted datasets</em> and <em>permanently deleted histories</em> containing datasets <strong>are not considered</strong>.</li>   <li><em>Histories/datasets that are shared with you</em> are only <strong>partially considered</strong> unless you import them.</li> </ul>   <p><strong>Note:</strong> To reduce quota usage, refer to <a href="/training-material/faqs/galaxy/account_reduce_quota_usage.html">How can I reduce quota usage while still retaining prior work (data, tools, methods)?</a> FAQ.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming a dataset <a href="/training-material/faqs/galaxy/datasets_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding job statuses <a href="/training-material/faqs/galaxy/datasets_job_status.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Job statuses will help you understand the stages of your work.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-job-statuses"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-job-statuses" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding job statuses</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Compare the color of your datasets to these job processing stages.</p>   <ul>   <li><strong>Grey</strong>: The job is queued. Allow this to complete!</li>   <li><strong>Yellow</strong>: The job is executing. Allow this to complete!</li>   <li><strong>Green</strong>: The job has completed successfully.</li>   <li><strong>Red</strong>: The job has failed. Check your inputs and parameters with Help examples and GTN tutorials. Scroll to the bottom of the tool form to find these.</li>   <li><strong>Light Blue</strong>: The job is paused. This indicates either an input has a problem or that you have exceeded the disk quota set by the administrator of the Galaxy instance you are working on.</li>   <li><strong>Grey, Yellow, Grey again</strong>: The job is waiting to run due to admin re-run or an automatic fail-over to a longer-running cluster.</li> </ul>   <p><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-info</span>  Don’t lose your queue placement! <strong>It is <em>essential</em> to allow queued jobs to remain queued, and to never interrupt an executing job.</strong> If you delete/re-run jobs, they are added back to the <em>end of the queue again</em>.</p>   <p>Related FAQs</p>   <ul>   <li><a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting errors</a></li>   <li><a href="/training-material/faqs/galaxy/analysis_jobs_not_running.html">My jobs aren’t running!</a></li>   <li><a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html">Extended Help for Differential Expression Analysis Tools</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with GFF GFT GTF2 GFF3 reference annotation <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-gff-gft-gtf2-gff3-reference-annotation"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-gff-gft-gtf2-gff3-reference-annotation" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with GFF GFT GTF2 GFF3 reference annotation</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>All annotation datatypes have a distinct format and content specification.     <ul>       <li>Data providers may release variations of any, and tools may produce variations.</li>       <li>GFF3 data may be labeled as GFF.</li>       <li>Content can overlap but is generally not understood by tools that are expecting just one of these specific formats.</li>     </ul>   </li>   <li>Best practices     <ul>       <li>The sequence identifiers must exactly match between reference annotation and reference genomes transcriptomes exomes.</li>       <li>Most tools expect GFT format unless the tool form specifically notes otherwise.         <ul>           <li>Get the GTF version from the data providers if it is available.</li>           <li>If only GFF3 is available, you can attempt to transform it with the tool <strong>gffread</strong>.</li>         </ul>       </li>       <li>Was GTF data detected as GFF during Upload? It probably has headers.   -Remove the headers (lines that start with a “#”) with the <strong>Select</strong> tool using the option “NOT Matching” with the regular expression: ^#         <ul>           <li><a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">Redetect the datatype</a>. It should be GTF once corrected.</li>         </ul>       </li>       <li>UCSC annotation         <ul>           <li>Find annotation under their Downloads area. The path will be similar to: <code class="language-plaintext highlighter-rouge">https://hgdownload.soe.ucsc.edu/goldenPath/&lt;database&gt;/bigZips/genes/</code></li>           <li>Copy the URL from UCSC and paste it into the Upload tool, allowing Galaxy to detect the datatype.</li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with deleted datasets <a href="/training-material/faqs/galaxy/datasets_working_with_deleted_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-deleted-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-deleted-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with deleted datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Deleted datasets and histories can be recovered by users as they are retained in Galaxy for a time period set by the instance administrator. Deleted datasets can be undeleted or permanently deleted within a History. <em>Links to show/hide deleted (and hidden) datasets are at the top of the History panel.</em></p>   <ul>   <li>To review or adjust an individual dataset:     <ol>       <li>Click on the name to expand it.</li>       <li>If it is only deleted, but not permanently deleted, you’ll see a message with links to recover or to purge.         <ul>           <li>Click on <em>Undelete</em> it to recover the dataset, making it active and accessible to tools again.</li>           <li>Click on <em>Permanently remove it from disk</em> to purge the dataset and remove it from the account quota calculation.</li>         </ul>       </li>     </ol>   </li>   <li>To review or adjust multiple datasets in batch:     <ol>       <li>Click on the checked box icon <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> near the top left of the history panel (Select Items) to switch into “Operations on Multiple Datasets” mode.</li>       <li>Accordingly for each individual dataset, choose the selection box. Check the datasets you want to modify and choose your option (show, hide, delete, undelete, purge, and group datasets).</li>     </ol>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with very large fasta datasets <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-very-large-fasta-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-very-large-fasta-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with very large fasta datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Run <strong>FastQC</strong> on your data to make sure the format/content is what you expect. Run more QA as needed.     <ul>       <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword “qa-qc” for examples.</li>       <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “qa-qc” and “fasta” for more help.</li>     </ul>   </li>   <li>Assembly result?     <ul>       <li>Consider filtering by length to remove reads that did not assemble.</li>       <li>Formatting criteria:         <ul>           <li>All sequence identifiers must be unique.</li>           <li>Some tools will require that there is no description line content, only identifiers, in the fasta title line (“&gt;” line). Use <strong>NormalizeFasta</strong> to remove the description (all content after the first whitespace) and wrap the sequences to 80 bases.</li>         </ul>       </li>     </ul>   </li>   <li><a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">Custom genome</a>, transcriptome exome?     <ul>       <li>Only appropriate for smaller genomes (bacterial, viral, most insects).</li>       <li>Not appropriate for any mammalian genomes, or some plants/fungi.</li>       <li>Sequence identifiers must be an exact match with all other inputs or expect problems. See <strong>GFF GFT GFF3</strong>.</li>       <li>Formatting criteria:         <ul>           <li>All sequence identifiers must be unique.</li>           <li>ALL tools will require that there is no description content, only identifiers, in the fasta title line (“&gt;” line). Use <strong>NormalizeFasta</strong> to remove the description (all content after the first whitespace) and wrap the sequences to 80 bases.</li>           <li>The only exception is when executing the <strong>MakeBLASTdb</strong> tool and when the input fasta is in NCBI BLAST format (see the tool form).</li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Working with very large fastq datasets <a href="/training-material/faqs/galaxy/datasets_working_with_fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-working-with-very-large-fastq-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-working-with-very-large-fastq-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Working with very large fastq datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Run <strong>FastQC</strong> on your data to make sure the format/content is what you expect. Run more QA as needed.     <ul>       <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword “qa-qc” for examples.</li>       <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “qa-qc” and “fastq” for more help.</li>     </ul>   </li>   <li>How to create a single smaller input. Search the tool panel with the keyword “subsample” for tool choices.</li>   <li>How to create multiple smaller inputs. Start with <strong>Split file to dataset collection</strong>, then merge the results back together using a tool specific for the datatype. Example: BAM results? Use <strong>MergeSamFiles</strong>.</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="datatypes"> <a href="#datatypes"> Datatypes </a></h3>

<hr/>



<h4 class="faq-area"> Best practices for loading fastq data into Galaxy <a href="/training-material/faqs/galaxy/datatypes_best_practices_fastq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-best-practices-for-loading-fastq-data-into-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-best-practices-for-loading-fastq-data-into-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Best practices for loading fastq data into Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>As of release <code class="language-plaintext highlighter-rouge">17.09</code>, <code class="language-plaintext highlighter-rouge">fastq</code> data will have the datatype <code class="language-plaintext highlighter-rouge">fastqsanger</code> auto-detected when that quality score scaling is detected and “autodetect” is used within the Upload tool. Compressed <code class="language-plaintext highlighter-rouge">fastq</code> data will be converted to uncompressed in the history.</li>   <li>To preserve <code class="language-plaintext highlighter-rouge">fastq</code> compression, directly assign the appropriate datatype (eg: <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>).</li>   <li>If the data is close to or over 2 GB in size, be sure to use FTP.</li>   <li>If the data was already loaded as <code class="language-plaintext highlighter-rouge">fastq.gz</code>, don’t worry! Just test the data for correct format (as needed) and assign the <a href="/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html">metadata type</a>.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Compressed FASTQ files, (`*.gz`) <a href="/training-material/faqs/galaxy/datatypes_fastq_compressed.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-compressed-fastq-files-gz"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-compressed-fastq-files-gz" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Compressed FASTQ files, (`*.gz`)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Files ending in <code class="language-plaintext highlighter-rouge">.gz</code> are compressed (zipped) files.     <ul>       <li>The <code class="language-plaintext highlighter-rouge">fastq.gz</code> format is a compressed version of a <code class="language-plaintext highlighter-rouge">fastq</code> dataset.</li>       <li>The <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> format is a compressed version of the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype, etc.</li>     </ul>   </li>   <li>Compression saves space (and therefore your quota).</li>   <li>Tools can accept the compressed versions of input files</li>   <li>Make sure the datatype (compressed or uncompressed) is correct for your files, or it may cause tool errors.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> FASTQ files: `fastq` vs `fastqsanger` vs .. <a href="/training-material/faqs/galaxy/datatypes_fastq_varieties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-fastq-files-fastq-vs-fastqsanger-vs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-fastq-files-fastq-vs-fastqsanger-vs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: FASTQ files: `fastq` vs `fastqsanger` vs ..</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>FASTQ files come in various flavours. They differ in the encoding scheme they use. See our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#inspect-a-raw-sequence-file">QC tutorial</a> for a more detailed explanation of encoding schemes.</p>   <p>Nowadays, the most commonly used encoding scheme is sanger. In Galaxy, this is the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype. If you are using older datasets, make sure to verify the FASTQ encoding scheme used in your data.</p>   <p><strong>Be Careful: choosing the wrong encoding scheme can lead to incorrect results!</strong></p>   <p><strong>Tip:</strong> There are 2 Galaxy datatypes that have similar names, but are not the same, please make sure you <code class="language-plaintext highlighter-rouge">fastqsanger</code> and <code class="language-plaintext highlighter-rouge">fastqcssanger</code> (not the additional <code class="language-plaintext highlighter-rouge">cs</code>).</p>   <p><strong>Tip:</strong> When in doubt, choose <code class="language-plaintext highlighter-rouge">fastqsanger</code></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment? <a href="/training-material/faqs/galaxy/datatypes_fastq_and_fastqsanger.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-do-fastq-gz-datasets-relate-to-the-fastqsanger-datatype-metadata-assignment"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-do-fastq-gz-datasets-relate-to-the-fastqsanger-datatype-metadata-assignment" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How do `fastq.gz` datasets relate to the `.fastqsanger` datatype metadata assignment?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Before assigning <code class="language-plaintext highlighter-rouge">fastqsanger</code> or <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code>, be sure to confirm the format.</p>   <p><strong>TIP:</strong></p>   <ul>   <li>Using <em>non-fastqsanger</em> scaled quality values will cause scientific problems with tools that expected <code class="language-plaintext highlighter-rouge">fastqsanger</code> formatted input.</li>   <li>Even if the tool does not fail, get the format right from the start to avoid problems. Incorrect format is still one of the most common reasons for tool errors or unexpected results (within Galaxy or not).</li>   <li>For more information on <a href="/training-material/faqs/galaxy/datatypes_fastq_format.html">How to format fastq data for tools that require .fastqsanger format?</a></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to format fastq data for tools that require .fastqsanger format? <a href="/training-material/faqs/galaxy/datatypes_fastq_format.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-format-fastq-data-for-tools-that-require-fastqsanger-format"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-format-fastq-data-for-tools-that-require-fastqsanger-format" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to format fastq data for tools that require .fastqsanger format?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Most tools that accept FASTQ data expect it to be in a specific FASTQ version: <code class="language-plaintext highlighter-rouge">.fastqsanger</code>. The <code class="language-plaintext highlighter-rouge">.fastqsanger</code> datatype must be assigned to each FASTQ dataset.</li> </ul>   <p>In order to do that:</p>   <ul>   <li>Watch the <strong><a href="http://vimeo.com/galaxyproject/fastqprep">FASTQ Prep Illumina</a></strong> video for a complete walk-through.</li>   <li>Run <strong>FastQC</strong> first to assess the type.     <ul>       <li>Run <strong>FASTQ Groomer</strong> if the data needs to have the quality scores rescaled.</li>       <li>If you are certain that the quality scores are already scaled to Sanger Phred+33 (the result of an Illumina 1.8+ pipeline), the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code> can be directly assigned. Click on the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <strong>Edit Attributes</strong> form. In the center panel, click on the “Datatype” tab, enter the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, and save.</li>     </ul>   </li>   <li>Run <strong>FastQC</strong> again on the entire dataset <em>if any changes were made to the quality scores</em> for QA.</li> </ul>   <p><strong>Other tips</strong></p>   <ul>   <li>If you are not sure what type of FASTQ data you have (maybe it is not Illumina?), see the help directly on the <strong>FASTQ Groomer</strong> tool for information about types.     <ul>       <li>For <em>Illumina</em>, first run <strong>FastQC</strong> on a sample of your data (<a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/">how to read the full report</a>). The output report will note the quality score type interpreted by the tool. If not <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, run <strong>FASTQ Groomer</strong> on the entire dataset. If <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, just assign the datatype.</li>       <li>For <em>SOLiD</em>, run <strong>NGS: Fastq manipulation → AB-SOLID DATA → Convert</strong>, to create a <code class="language-plaintext highlighter-rouge">.fastqcssanger</code> dataset. If you have uploaded a color space fastq sequence with quality scores already scaled to Sanger Phred+33 (<code class="language-plaintext highlighter-rouge">.fastqcssanger</code>), first confirm by running <strong>FastQC</strong> on a sample of the data. Then if you want to double-encode the color space into psuedo-nucleotide space (required by certain tools), see the instructions on the tool form <strong>Fastq Manipulation</strong> for the conversion.</li>       <li>If your data is <em>FASTA</em>, but you want to use tools that require FASTQ input, then using the tool <strong>NGS: QC and manipulation → Combine FASTA and QUAL</strong>. This tool will create “placeholder” quality scores that fit your data. On the output, click on the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <strong>Edit Attributes</strong> form. In the center panel, click on the “Datatype” tab, enter the datatype <code class="language-plaintext highlighter-rouge">.fastqsanger</code>, and save.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Identifying and formatting Tabular Datasets <a href="/training-material/faqs/galaxy/datatypes_tabular.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Format help for Tabular/BED/Interval Datasets
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-identifying-and-formatting-tabular-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-identifying-and-formatting-tabular-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Identifying and formatting Tabular Datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A Tabular datatype is human readable and has tabs separating data columns. Please note that tabular data is different from comma separated data (.csv) and the common datatypes are: <code class="language-plaintext highlighter-rouge">.bed</code>, <code class="language-plaintext highlighter-rouge">.gtf</code>, <code class="language-plaintext highlighter-rouge">.interval</code>, or <code class="language-plaintext highlighter-rouge">.txt</code>.</p>   <ol>   <li>Click the pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <strong><em>Edit Attributes</em></strong> form.     <ol>       <li>Change the datatype (3rd tab) and save.</li>       <li>Label columns (1st tab) and save.</li>       <li>Metadata will be assigned, then the dataset can be used.</li>     </ol>   </li>   <li>If the required input is a BED or Interval datatype, adjusting (<code class="language-plaintext highlighter-rouge">.tab</code> → <code class="language-plaintext highlighter-rouge">.bed</code>, <code class="language-plaintext highlighter-rouge">.tab</code> → <code class="language-plaintext highlighter-rouge">.interval</code>) maybe possible using a combination of <strong><em>Text Manipulation</em></strong> tools, to create a dataset that matches required specifications.</li>   <li>Some tools require that BED format be followed, even if the datatype Interval (with less strict column ordering) is accepted on the tool form.     <ul>       <li>These tools will fail, if they are run with malformed BED datasets or non-specific column assignments.</li>       <li><strong>Solution</strong>: reorganize the data to be in BED format and rerun.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding Datatypes <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-datatypes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-datatypes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding Datatypes</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Allow Galaxy to detect the datatype during Upload, and adjust from there if needed.</li>   <li>Tool forms will filter for the appropriate datatypes it can use for each input.</li>   <li><a href="/training-material/faqs/galaxy/datasets_change_datatype.html">Directly changing</a> a datatype can lead to errors. Be intentional and consider <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">converting</a> instead when possible.</li>   <li>Dataset content can also be adjusted (tools: Data manipulation) and the expected <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">datatype detected</a>. Detected datatypes are the most reliable in most cases.</li>   <li>If a tool does not accept a dataset as valid input, it is not in the correct format with the correct datatype.</li>   <li>Once a dataset’s content matches the datatype, and that dataset is repeatedly used (example: Reference annotation) use that same dataset for all steps in an analysis or expect problems. This may mean rerunning prior tools if you need to make a correction.</li>   <li>Tip: Not sure what datatypes a tool is expecting for an input?     <ol>       <li>Create a new empty history</li>       <li>Click on a tool from the tool panel</li>       <li>The tool form will list the accepted datatypes per input</li>     </ol>   </li>   <li><em>Warning</em>: In some cases, tools will transform a dataset to a new datatype at runtime for you.     <ul>       <li>This is generally helpful, and best reserved for smaller datasets.</li>       <li>Why? This can also unexpectedly create hidden datasets that are near duplicates of your original data, only in a different format.</li>       <li>For large data, that can quickly consume working space (quota).</li>       <li>Deleting/purging any hidden datasets can lead to errors if you are still using the original datasets as an input.</li>       <li>Consider <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">converting</a> to the expected datatype yourself when data is large.</li>       <li>Then test the tool directly on converted data. If it works, purge the original to recover space.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using compressed fastq data as tool inputs <a href="/training-material/faqs/galaxy/datatypes_Using_compressed_fastq_data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-compressed-fastq-data-as-tool-inputs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-compressed-fastq-data-as-tool-inputs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using compressed fastq data as tool inputs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>If the tool accepts <code class="language-plaintext highlighter-rouge">fastq</code> input, then <code class="language-plaintext highlighter-rouge">.gz</code> compressed data assigned to the datatype <code class="language-plaintext highlighter-rouge">fastq.gz</code> is appropriate.</li>   <li>If the tool accepts <code class="language-plaintext highlighter-rouge">fastqsanger</code> input, then <code class="language-plaintext highlighter-rouge">.gz</code> compressed data assigned to the datatype <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> is appropriate.</li>   <li>Using uncompressed <code class="language-plaintext highlighter-rouge">fastq</code> data is still an option with tools. The choice is yours.</li> </ul>   <p><strong>TIP:</strong> Avoid labeling compressed data with an uncompressed datatype, and the reverse. Jobs using mismatched datatype versus actual format will fail with an error.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="features"> <a href="#features"> Features </a></h3>

<hr/>



<h4 class="faq-area"> Using the Window Manager to view multiple datasets <a href="/training-material/faqs/galaxy/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-the-window-manager-to-view-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-the-window-manager-to-view-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using the Window Manager to view multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you would like to view two or more datasets at once, you can use the <strong>Window Manager</strong> feature in Galaxy:</p>   <ol>   <li><strong>Click</strong> on the <em>Window Manager</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> on the top menu bar.     <ul>       <li>You should see a little checkmark on the icon now</li>     </ul>   </li>   <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> to view the output     <ul>       <li>You should see the output in a window overlayed over Galaxy</li>       <li>You can resize this window by dragging the bottom-right corner</li>     </ul>   </li>   <li><strong>Click</strong> outside the file to exit the Window Manager</li>   <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span> a second dataset from your history     <ul>       <li>You should now see a second window with the new dataset</li>       <li>This makes it easier to compare the two outputs</li>     </ul>   </li>   <li>Repeat this for as many files as you would like to compare</li>   <li>You can turn off the <strong>Window Manager</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden">galaxy-scratchbook</span> by clicking on the icon again</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Why not use Excel? <a href="/training-material/faqs/galaxy/why_not_excel.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Excel is a fantastic tool and a great place to build simple analysis models, but when it comes to scaling, Galaxy wins every time.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-why-not-use-excel"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-why-not-use-excel" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Why not use Excel?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You could just as easily use Excel to answer the same question, and if the goal is to learn how to use a tool, then either tool would be great! But what if you are working on a question where your analysis matters? Maybe you are working with human clinical data trying to diagnose a set of symptoms, or you are working on research that will eventually be published and maybe earn you a Nobel Prize?</p>   <p>In these cases your analysis, <em>and the ability to reproduce it exactly</em>, is vitally important, and Excel won’t help you here. It doesn’t track changes and it offers very little insight to others on how you got from your initial data to your conclusions.</p>   <p>Galaxy, on the other hand, <em>automatically records every step of your analysis.</em> And when you are done, you can share your analysis with anyone. You can even include a link to it in a paper (or your acceptance speech). In addition, you can create a reusable workflow from your analysis that others (or yourself) can use on other datasets.</p>   <p>Another challenge with spreadsheet programs is that they don’t scale to support <em>next generation sequencing (NGS)</em> datasets, a common type of data in genomics, and which often reach gigabytes or even terabytes in size. Excel has been used for large datasets, but you’ll often find that learning a new tool gives you significantly more ability to scale up, and scale out your analyses.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="histories"> <a href="#histories"> Histories </a></h3>

<hr/>



<h4 class="faq-area"> Copy a dataset between histories <a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Sometimes you may want to use a dataset in multiple histories. You do not need to re-upload the data, but you can copy datasets from one history to another.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-copy-a-dataset-between-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-copy-a-dataset-between-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Copy a dataset between histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There 3 ways to copy datasets between histories</p>   <ol>   <li>     <p>From the original history</p>     <ol>       <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> icon which is on the top of the list of datasets in the history panel</li>       <li>Click on <strong>Copy Datasets</strong></li>       <li>         <p>Select the desired files</p>       </li>       <li>         <p>Give a relevant name to the “New history”</p>       </li>       <li>Validate by ‘Copy History Items’</li>       <li>Click on the new history name in the green box that have just appear to switch to this history</li>     </ol>   </li>   <li>     <p>Using the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>Show Histories Side-by-Side</strong></p>     <ol>       <li>Click on the <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> dropdown arrow top right of the history panel (<strong>History options</strong>)</li>       <li>Click on <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-columns</span> <strong>Show Histories Side-by-Side</strong></li>       <li>If your target history is not present         <ol>           <li>Click on ‘Select histories’</li>           <li>Click on your target history</li>           <li>Validate by ‘Change Selected’</li>         </ol>       </li>       <li>Drag the dataset to copy from its original history</li>       <li>Drop it in the target history</li>     </ol>   </li>   <li>     <p>From the target history</p>     <ol>       <li>Click on <strong>User</strong> in the top bar</li>       <li>Click on <strong>Datasets</strong></li>       <li>Search for the dataset to copy</li>       <li>Click on its name</li>       <li>Click on <strong>Copy to current History</strong></li>     </ol>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a new history <a href="/training-material/faqs/galaxy/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Histories are an important part of Galaxy, most people use a new history for every new analysis. Always make sure to give your histories good names, so you can easily find your results back later.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Dataset colors <a href="/training-material/faqs/galaxy/histories_dataset_colors.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains meaning of dataset colors in Galaxy's history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-colors"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-colors" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset colors</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are several different “states” a dataset can be in. These states are indicated by colors:</p>   <p><img src="/training-material/shared/images/galactic_colors.svg" alt="Colors indicating states of Galaxy datasets" /></p>   <ul>   <li><strong>ok</strong>: everything is fine, life is good;</li>   <li><strong>new</strong>: the dataset was just created. Galaxy does not yet know when it is;</li>   <li><strong>queued</strong>: indicates that the job generating this dataset is scheduled for execution but not running yet;</li>   <li><strong>running</strong>: job generating this dataset is running;</li>   <li><strong>setting metadata</strong>: when a new dataset is uploaded Galaxy examines it to understand what kind of data it is (e.g., BAM, FASTQ, fasta, BED, etc.). This is called “setting metadata”;</li>   <li><strong>deferred</strong>: sometimes it does not make sense to upload the dataset until it is needed for an analysis. Galaxy will download <strong>deferred</strong> datasets later during the job execution. Those datasets do not count toward your quota;</li>   <li><strong>paused</strong>: in some cases as, for example, workflow executions, upstream errors prevent subsequent jobs from starting creating datasets in “paused” state;</li>   <li><strong>discarded</strong>: something went wrong such as, for example, a job producing this dataset might have been cancelled;</li>   <li><strong>error</strong>: everything is not fine; life is bad!</li>   <li><strong>placeholder</strong>: similar to “new”; we know something will be there but are not yet sure what;</li>   <li><strong>failed populated state</strong>: this refers to collections (not individual datasets). Here a collection has failed to be populated with datasets;</li>   <li><strong>new populated state</strong>: this refers to collections (not individual datasets). A collection was created but not populated yet.</li> </ul>   <!-- original editable image = https://docs.google.com/drawings/d/1F2Lq1m3cMIckvCexXMzug-dqwgifkoMOzoyH4VcoVX0/edit?usp=sharing -->   <!-- TO DO  Needs to be linked to FAQs on: - how to report errors - explaining collections  --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Dataset snippet <a href="/training-material/faqs/galaxy/histories_dataset_item.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Describes features of a single dataset element in the history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-dataset-snippet"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-dataset-snippet" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Dataset snippet</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A single Galaxy dataset can either be “collapsed” or “expanded”.</p>   <p><strong>Collapsed dataset view</strong></p>   <p>Datasets in the panel are initially shown in a “collapsed” view:</p>   <p><img src="/training-material/shared/images/history_item_collapsed.png" alt="Collapsed view of a single Galaxy dataset" /></p>   <p>It contains the following elements:</p>   <ul>   <li><strong>Dataset number</strong>: (“1”) order of dataset in the history;</li>   <li><strong>Dataset name</strong>: (“M117-bl_1.fq.gz”) its name;</li>   <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">galaxy-eye</span>: click this to view the dataset contents;</li>   <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span>: click this to edit dataset properties;</li>   <li><i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span>: click this to delete the dataset from the history (<em>don’t worry</em>, you can undo this action!).</li> </ul>   <p>Clicking on a collapsed dataset will expand it.</p>   <blockquote class="details">   <div class="box-title details-title" id="details-some-buttons-can-be-disabled"><button class="gtn-boxify-button details" type="button" aria-controls="details-some-buttons-can-be-disabled" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Some buttons can be disabled.</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Some of the buttons above may be disabled if the dataset is in a state that doesn’t allow the action. For example, the ‘edit’ button is disabled for datasets that are still queued or running</p> </blockquote>   <p><strong>Expanded dataset view</strong></p>   <p>Expanded dataset view adds a preview element and many additional controls.</p>   <p><img src="/training-material/shared/images/history_item_expanded.png" alt="Expanded view of a single Galaxy dataset" /></p>   <p>In addition to the elements described above for the collapsed dataset, its expanded view contains:</p>   <ul>   <li><strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>: click on this to tag this dateset;</li>   <li><strong>Dataset size</strong>: (“2 variants, 18 comments”) lists the size of the dataset. When datasets are small (like in this example) the exact size is shown. For large datasets, Galaxy gives an approximate estimate.</li>   <li><strong>format</strong>: (“VCF”) lists the datatype;</li>   <li><strong>database</strong>: (“?”) lists which genome built this dataset corresponds to. This usually lists “?” unless the genome build is set explicitly or the dataset is derived from another dataset with defined genome build information;</li>   <li><strong>info field</strong>: (“INFO [2024-03-26 12:08:53,435]…”) displays information provided by the tool that generated this dataset. This varies widely and depends on the type of job that generated this dataset.</li>   <li><i class="fas fa-save" aria-hidden="true"></i><span class="visually-hidden">dataset-save</span>: Saves dataset to disk;</li>   <li><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">dataset-link</span>: Copies dataset link into clipboard;</li>   <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">dataset-info</span>: Displays additional details about the dataset in the center pane;</li>   <li><i class="fas fa-arrow-rotate-right" aria-hidden="true"></i><span class="visually-hidden">dataset-rerun</span>: Reruns job that generated this dataset. This button is unavailable for datasets uploaded into history because they were not produced by a Galaxy tool;</li>   <li><i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">dataset-visualize</span>: Displays visualization options for this dataset. The list of options is dependent on the datatype;</li>   <li><i class="fas fa-sitemap" aria-hidden="true"></i><span class="visually-hidden">dataset-related-datasets</span>: Shows datasets related to this dataset. This is useful for tracking down parental datasets - those that were used as inputs into a job that produced this particular dataset.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Downloading histories <a href="/training-material/faqs/galaxy/histories_download_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-downloading-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-downloading-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Downloading histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on the gear icon <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> on the top of the history panel.</li>   <li>Select “Export History to File” from the History menu.</li>   <li>Click on the “Click here to generate a new archive for this history” text.</li>   <li>Wait for the Galaxy server to prepare history for download.</li>   <li>Click on the generated link to download the history.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Find all Histories and purge (aka permanently delete) <a href="/training-material/faqs/galaxy/histories_find_purge.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-find-all-histories-and-purge-aka-permanently-delete"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-find-all-histories-and-purge-aka-permanently-delete" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Find all Histories and purge (aka permanently delete)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Login to your Galaxy account.</li>   <li>On the top navigation bar Click on <strong>User</strong>.</li>   <li>On the drop down menu that appears Click on <strong>Histories</strong>.</li>   <li>Click on <strong>Advanced Search</strong>, additional fields will be displayed.</li>   <li>Next to the <strong>Status</strong> field, click <strong>All</strong>, a list of all histories will be displayed.</li>   <li>Check the box next to <strong>Name</strong> in the displayed list to select all histories.</li>   <li>Click <strong>Delete Permanently</strong> to purge all histories.</li>   <li>A pop up dialogue box will appear letting you know history contents will be removed and cannot be undone, then click <strong>OK</strong> to confirm.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding Histories <a href="/training-material/faqs/galaxy/histories_finding_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding Histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>To review all histories in your account, go to <strong>User &gt; Histories</strong> in the top menu bar.</li>   <li>At the top of the History listing, click on <strong>Advanced Search</strong>.</li>   <li>Set the status to all to view all of your active, deleted, and permanently deleted (purged) histories.</li>   <li>Histories in all states are listed for registered accounts. Meaning one will always find their data here if it ever appears to be “lost”.</li>   <li>Note: Permanently deleted (purged) Histories may be fully removed from the server at any time. The data content inside the History is always removed at the time of purging (by a double-confirmed user action), but the purged History artifact may still be in the listing. Purged data content cannot be restored, even by an administrator.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Finding and working with "Histories shared with me" <a href="/training-material/faqs/galaxy/histories_shared.html"><i class="fas fa-external-link-alt"></i></a> </h4>
How to find and work on histories shared with you
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-finding-and-working-with-quot-histories-shared-with-me-quot"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-finding-and-working-with-quot-histories-shared-with-me-quot" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Finding and working with &quot;Histories shared with me&quot;</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>To find histories shared with me:</strong></p>   <ol>   <li>Log into your account.</li>   <li>Select <strong>User</strong>, in the drop-down menu, select <strong>Histories shared with me</strong>.</li> </ol>   <p><strong>To work with shared histories:</strong></p>   <ul>   <li>Import the History into your account via copying it to work with it.</li>   <li>Unshare Histories that you no longer want shared with you or that you have already made a copy of.</li> </ul>   <p><strong>Note:</strong> Shared Histories (when copied into your account or not) do count in portion toward your total account data quota usage. More details on histories shared concerning account quota usage can be found in <a href="/training-material/faqs/galaxy/histories_unsharing_histories.html">this link</a>.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> History annotation <a href="/training-material/faqs/galaxy/histories_annotation.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to annotate a history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-history-annotation"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-history-annotation" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: History annotation</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes tags and names are not enough to describe the work done within a history. Galaxy allows you to create history annotations: longer text entries that allow for more formatting options. The formatting of the text is preserved. Later, if you publish or share the history, the annotation will be displayed automatically - allowing you to share additional notes about the analysis. Multiple lines, spaces, and emoji! 😹🏳️‍⚧️🌈 can be used while writing annotations.</p>   <p>To annotate a history:</p>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name. A larger text section will appear displaying any   existing annotation or <code class="language-plaintext highlighter-rouge">Annotation (optional)</code> if empty.</li>   <li>Add your text. <kbd>Enter</kbd> will move the cursor to the next line. (Tabs cannot be   entered since the ‘Tab’ button is used to switch between controls on the page - tabs can be pasted in, however).</li>   <li>Click on <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>.</li>   <li>To cancel, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button.</li> </ol>   <p><img src="/training-material/shared/images/history_annotations.png" alt="UI for annotating histories" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> History options <a href="/training-material/faqs/galaxy/histories_options.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains different history options
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-history-options"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-history-options" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: History options</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Clicking the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> button will open a drop-down menu with several options:</p>   <ul>   <li><strong>Show histories side-by-side</strong> - brings up a view in which multiple histories can be viewed and manipulated simultaneously. Datasets can be dragged between histories in this view.</li>   <li><strong>Resume Paused Jobs</strong> - restarts paused jobs in history.</li>   <li><strong>Copy this history</strong> - creates an exact copy of the current history in the current account.</li>   <li><strong>Delete this history</strong> - deletes the current history.</li>   <li><strong>Export tool citations</strong> - export citations for tools that were used in the current history.</li>   <li><strong>Export history to File</strong> - creates a compressed archive containing data from the current history.</li>   <li><strong>Archive history</strong> - moves history to a non-active, archived, state.</li>   <li><strong>Extract workflow</strong> - converts the current history into a workflow</li>   <li><strong>Show invocations</strong> - shows a list of all workflows that were run in the current history</li>   <li><strong>Share or Publish</strong> - allows controlling access to history. It can be made public or shared with a specific user.</li>   <li><strong>Set Permissions</strong> - allows to set the rules on who can access daysets in the current history.</li>   <li><strong>Make Private</strong> - resets all permission and makes the current history private.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> History tagging <a href="/training-material/faqs/galaxy/histories_tagging.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to add tags to a history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-history-tagging"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-history-tagging" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: History tagging</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tags are short pieces of text used to describe the thing they’re attached to and many things in Galaxy can be tagged. Each item can have many tags and you can add new tags or remove them at any time. Tags can be another useful way to organize and search your data. For instance, you might tag a history with the type of analysis you did in it: <code class="language-plaintext highlighter-rouge">assembly</code> or <code class="language-plaintext highlighter-rouge">variants</code>. Or you may tag them according to data sources or some other metadata: <code class="language-plaintext highlighter-rouge">long-term-care-facility</code> or <code class="language-plaintext highlighter-rouge">yellowstone-park:2014</code>.</p>   <p><strong>To tag a history:</strong></p>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”).</li>   <li>Click on <strong>Add tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span> and start typing. Any tags that you’ve used previously will show below your partial entry -   allowing you to use this ‘autocomplete’ data to re-use your previous tags without typing them in full.</li>   <li>Click on <strong>Save</strong> <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>.</li>   <li>To cancel, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button.</li> </ol>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-do-not-use-spaces"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Do not use spaces</div>   <p>It is strongly recommended to replace spaces in tags with <code class="language-plaintext highlighter-rouge">_</code> or <code class="language-plaintext highlighter-rouge">-</code>, as spaces will automatically be removed when the tag is saved.</p> </blockquote>   <p><img src="/training-material/shared/images/history_tags.png" alt="UI for tagging histories" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> How to set Data Privacy Features? <a href="/training-material/faqs/galaxy/datasets_data_privacy_features.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-set-data-privacy-features"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-set-data-privacy-features" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to set Data Privacy Features?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Privacy controls are only enabled if desired. Otherwise, datasets by defaults remain private and unlisted in Galaxy. This means that a dataset you’ve created is virtually invisible until you publish a link to it.</p>   <p>Below are three optional steps to setting private Histories, a user can make use of any of the options below depending on what the user want to achieve:</p>   <ol>   <li>     <p>Changing the privacy settings of individual dataset.</p>     <ul>       <li>Click on the <strong>dataset</strong> name for a dropdown.</li>       <li>Clicking the ‘<strong>pencil - <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> icon</strong></li>       <li>Move on the <strong>Permissions</strong> tab.</li>       <li>On the permission tab is two input tab</li>       <li>On the second input with a label of <strong>access</strong></li>       <li>Search for the name of the user to grant permission</li>       <li>Click on <strong>save permission</strong></li>     </ul>     <p><img src="https://galaxyproject.org/learn/privacy-features/set-perm.gif" alt="gif of the process described above, in Galaxy" /></p>     <p><em>Note</em>: Adding additional roles to the ‘access’ permission along with your “private role” does not do what you may expect. Since roles are always logically added together, only you will be able to access the dataset, since only you are a member of your “private role”.</p>   </li>   <li>     <p>Make all datasets in the current history private.</p>     <ul>       <li>Open the History Options galaxy-gear menu <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> at the top of your history panel</li>       <li>Click the <strong>Make Private</strong> option in the dropdown menu available</li>       <li>Sets the default settings for all new datasets in this history to private.</li>     </ul>     <p><img src="https://galaxyproject.org/learn/privacy-features/this-hist-priv-perm.gif" alt="gif of the process described above, in Galaxy" /></p>   </li>   <li>     <p>Set the default privacy settings for new histories</p>     <ul>       <li>Click <strong>user</strong> button on top of the main channel for a dropdown <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span></li>       <li>Click on the <em>preferences</em> under the dropdown <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span></li>       <li>Select <strong>Set Dataset Permissions for New Histories</strong> icon <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">cofest</span></li>       <li>Add a permission and click <strong>save permission</strong></li>     </ul>     <p><img src="https://galaxyproject.org/learn/privacy-features/new-hist-perm.gif" alt="gif of the process described above, in Galaxy" /></p>     <p><em>Note</em>: Changes made here will only affect histories created after these settings have been stored.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing a history <a href="/training-material/faqs/galaxy/histories_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the link to the shared history</li>   <li>Click on the <em>Import this history</em> button on the top left</li>   <li>Enter a title for the new history</li>   <li>Click on <strong>Copy History</strong></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Manipulating multiple history datasets <a href="/training-material/faqs/galaxy/datasets_multiple.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to manipulate multiple history datasets at once
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-manipulating-multiple-history-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-manipulating-multiple-history-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Manipulating multiple history datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can also hide, delete, and purge multiple datasets at once by <strong>multi-selecting datasets</strong>:</p>   <ol>   <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">galaxy-selector</span> Click the multi-select button containing the checkbox just below the history size.</li>   <li>Checkboxes will appear inside each dataset in the history.</li>   <li>Scroll and click the checkboxes next to the datasets you want to manage.</li>   <li>Click the ‘n of N selected’ to choose the action. The action will be performed on all selected datasets, except for the ones that don’t support the action. That is, if an action doesn’t apply to a selected dataset, like deleting a deleted dataset, nothing will happen to that dataset, while all other selected datasets will be deleted.</li>   <li>You can click the multi-select button again to hide the checkboxes.</li> </ol>   <p><img src="/training-material/faqs/galaxy/images/multiselect.gif" alt="Operating on multiple datasets" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming a history <a href="/training-material/faqs/galaxy/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Explains how to rename a history
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Searching your history <a href="/training-material/faqs/galaxy/histories_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-searching-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-searching-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Searching your history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To make it easier to find datasets in large histories, you can filter your history by keywords as follows:</p>   <ol>   <li>     <p><strong>Click</strong> on the <em>search datasets</em> box at the top of the history panel.</p>     <p><img src="/training-material/shared/images/history_search_datasets.png" alt="history search box" /></p>   </li>   <li><strong>Type</strong> a search term in this box     <ul>       <li>For example a tool name, or sample name</li>     </ul>   </li>   <li>To undo the filtering and show your full history again, press on the <em>clear search</em> button <i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden">galaxy-clear</span> next to the search box</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Sharing your History <a href="/training-material/faqs/galaxy/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can share your work in Galaxy. There are various ways you can give access to one of your histories to other users.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sharing-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sharing-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sharing your History</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sharing your history allows others to import and access the datasets, parameters, and steps of your history.</p>   <p>Access the history sharing menu via the History Options dropdown (<i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span>), and clicking “<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">history-share</span> Share or Publish”</p>   <ol>   <li><strong>Share via link</strong>     <ul>       <li>Open the <strong>History Options</strong> <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> menu at the top of your history panel and select “<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">history-share</span> Share or Publish”         <ul>           <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Make History accessible</strong></li>           <li>A <strong>Share Link</strong> will appear that you give to others</li>         </ul>       </li>       <li>Anybody who has this link can view and copy your history</li>     </ul>   </li>   <li><strong>Publish your history</strong>     <ul>       <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden">galaxy-toggle</span> <strong>Make History publicly available in Published Histories</strong></li>       <li>Anybody on this Galaxy server will see your history listed under the <em>Published Histories</em> tab opened via the <i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">galaxy-histories-activity</span> <strong>Histories</strong> activity</li>     </ul>   </li>   <li><strong>Share only with another user</strong>.     <ul>       <li>Enter an email address for the user you want to share with in the <em>Please specify user email</em> input below <strong>Share History with Individual Users</strong></li>       <li>Your history will be shared only with this user.</li>     </ul>   </li>   <li><strong>Finding histories others have shared with me</strong>     <ul>       <li>Click on the <i class="fas fa-hdd" aria-hidden="true"></i><span class="visually-hidden">galaxy-histories-activity</span> <strong>Histories</strong> activity in the activity bar on the left</li>       <li>Click the <strong>Shared with me</strong> tab</li>       <li>Here you will see all the histories others have shared with you directly</li>     </ul>   </li> </ol>   <p><strong>Note:</strong> If you want to make changes to your history without affecting the shared version, make a copy by going to <strong>History Options</strong> <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> icon in your history and clicking <em>Copy this History</em></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Switching to an existing history <a href="/training-material/faqs/galaxy/histories_switch.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Shows how to switch to another existing history in your account
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-switching-to-an-existing-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-switching-to-an-existing-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Switching to an existing history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To switch to an existing history simply click the <i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">switch-histories</span> icon at the top of the history panel. This opens a list of histories existing in a given Galaxy account in the middle part of the interface.</p>   <p><img src="/training-material/shared/images/list_of_histories.png" alt="UI for switching histories" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Top level history controls <a href="/training-material/faqs/galaxy/histories_top_control_buttons.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Description of three history buttons for creating a new histiory, switching histories, and opening history options dropdown
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-top-level-history-controls"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-top-level-history-controls" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Top level history controls</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Above the current history panel are three buttons:</p>   <ul>   <li>The <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> “<strong>Create new history</strong>” button will create an empty history.</li>   <li>The <i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">switch-histories</span> “<strong>Switch to history</strong>” will open a window letting you easily swap to any of your other histories.</li>   <li>The <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> “<strong>History options</strong>” (formerly the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> “Gear menu”) gives you access to advanced options to work with your history.</li> </ul>   <p><img src="/training-material/shared/images/history_top_level_buttons.svg" alt="The three history buttons" /></p>   <!-- history_top_level_buttons.svg = editable version = https://docs.google.com/drawings/d/11ZIUNYVHzrieHLVph9lhXgfzrO7MM-N4ehTehMr_eto/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Transfer entire histories from one Galaxy server to another <a href="/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-transfer-entire-histories-from-one-galaxy-server-to-another"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-transfer-entire-histories-from-one-galaxy-server-to-another" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Transfer entire histories from one Galaxy server to another</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><strong>Transfer a Single Dataset</strong></p>   <p>At the <strong>sender</strong> Galaxy server, <a href="/training-material/faqs/galaxy/histories_sharing.html">set the history to a shared state</a>, then directly capture the <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> link for a dataset and paste the URL into the <strong>Upload</strong> tool at the <strong>receiver</strong> Galaxy server.</p>   <p><strong>Transfer an Entire History</strong></p>   <p><a href="/training-material/faqs/galaxy/account_create.html">Have an account</a> at two different Galaxy servers, and be logged into both.</p>   <p>At the <strong>sender</strong> Galaxy server</p>   <ol>   <li>Navigate to the history you want to transfer, and <a href="/training-material/faqs/galaxy/histories_sharing.html">set the history to a shared state</a>.</li>   <li>Click into the <strong>History Options</strong> menu in the history panel.</li>   <li>Select from the menu <i class="fas fa-archive" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-archive</span> <strong>Export History to File</strong>.</li>   <li>Choose the option for <strong>How do you want to export this History?</strong> as <strong>to direct download</strong>.</li>   <li>Click on <strong>Generate direct download</strong>.</li>   <li>Allow the archive generation process to complete. *</li>   <li>Copy the <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden">galaxy-link</span> link for your new archive.</li> </ol>   <p>At the <strong>receiver</strong> Galaxy server</p>   <ol>   <li>Confirm that you are logged into your account.</li>   <li>Click on <strong>Data</strong> in the top menu, and choose <strong>Histories</strong> to reach your <strong>Saved Histories</strong>.</li>   <li>Click on <strong>Import history</strong> in the grey button on the top right.</li>   <li>Paste in your link’s URL from step 7.</li>   <li>Click on <strong>Import History</strong>.</li>   <li>Allow the archive import process to complete. *</li>   <li>The transfered history will be uncompressed and added to your <strong>Saved Histories</strong>.</li> </ol>   <p>* For steps 6 and 13: It is Ok to navigate away for other tasks during processing. If enabled, Galaxy will send you <a href="/training-material/faqs/galaxy/account_update_preference.html">status notifications</a>.</p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> If the history to transfer is large, you may <a href="/training-material/faqs/galaxy/histories_copy_dataset.html">copy just your important datasets into a new history</a>, and create the archive from that new smaller history. Clearing away deleted and purged datasets will make <em>all</em> histories smaller and faster to archive and transfer!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Undeleting history <a href="/training-material/faqs/galaxy/histories_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Undelete your deleted histories
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-undeleting-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-undeleting-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Undeleting history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Deleted histories can be <strong>undeleted</strong>:</p>   <ol>   <li>Select “Histories” from the activity bar on the left</li>   <li>Toggle “Advanced search”</li>   <li>Click “Deleted”</li>   <li>Click on the title of the history you want to un-delete and un-delete it!</li> </ol>   <p><img src="/training-material/faqs/galaxy/images/histories_undelete.gif" alt="Undeleting a deleted history" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Unsharing unwanted histories <a href="/training-material/faqs/galaxy/histories_unsharing_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-unsharing-unwanted-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-unsharing-unwanted-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Unsharing unwanted histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>All account Histories owned by others but shared with you can be reviewed under <strong>User &gt; Histories shared with me</strong>.</li>   <li>The other person does not need to unshare a history with you. <em>Unshare histories yourself</em> on this page using the pull-down menu per history.</li>   <li>Dataset and History privacy options, including sharing, can be set under <strong>User &gt; Preferences</strong>.</li> </ul>   <p>Three key features to work with shared data are:</p>   <ul>   <li><strong>View</strong> is a review feature. The data cannot be worked with, but many details, including tool and dataset metadata/parameters, are included.</li>   <li><strong>Copy</strong> those you want to work with. This will increase your quota usage. This will also allow you to manipulate the datasets or the history independently from the original owner. All History/Dataset functions are available if the other person granted full access to the datasets to you.</li>   <li><strong>Unshare</strong> any on the list not needed anymore. After a history is copied, you will still have your version of the history, even if later unshared or the other person who shared it with you changes their version later. Meaning, that <em>each account’s version of a History and the Datasets in it are distinct</em> (unless the Datasets were not shared, you will still only be able to “view” but not work with or download them).</li> </ul>   <p><strong>Note:</strong> “Histories shared with me” result in only a tiny part of your quota usage. Unsharing will not significantly reduce quota usage unless hundreds (or more!) or many significant histories are shared. If you share a History with someone else, that does not increase or decrease your quota usage.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> View a list of all histories <a href="/training-material/faqs/galaxy/histories_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This FAQ demonstrates how to list all histories for a given user
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-view-a-list-of-all-histories"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-view-a-list-of-all-histories" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: View a list of all histories</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are multiple ways in which you can view your histories:</p>   <ol>   <li>     <p>Viewing histories using <i class="fas fa-exchange-alt" aria-hidden="true"></i><span class="visually-hidden">switch-histories</span> “<strong>Switch to history</strong>” button. This is best for quickly switching between multiple histories.</p>     <p>Click the “<strong>Switch history</strong>” icon at the top of the history panel to bring up a list of all your histories: <img src="/training-material/shared/images/histories_list.svg" alt="Listing histories using the &quot;Switch history&quot; button" /></p>   </li>   <li>     <p>Using the “<strong>Activity Bar</strong>”:</p>     <p>Click the “<strong>Show all histories</strong>” button within the Activity Bar on the left: <img src="/training-material/shared/images/histories_list_via_activity_bar.svg" alt="Listing histories using Activity Bar" /></p>   </li>   <li>     <p>Using “<strong>Data</strong>” drop-down:</p>     <p>Click the “<strong>Data</strong>” link on the top bar of Galaxy interface and select “<strong>Histories</strong>”: <img src="/training-material/shared/images/histories_list_via_masthead.svg" alt="Listing histories using &quot;Data&quot; menu" /></p>   </li>   <li>     <p>Using the <strong>Multi-view</strong>, which is best for moving datasets between histories:</p>     <p>Click the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> menu, and select <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> <strong>Show histories side-by-side</strong></p>   </li> </ol>   <!-- Original drawing for histories_list.svg = https://docs.google.com/drawings/d/1fiMTFt4q1DwJllnxjddG0LA-XwKkxyXfPlOqUHteX_k/edit?usp=sharing -->   <!-- Original drawing for histories_list_via_activity_bar.svg = https://docs.google.com/drawings/d/1YyUMe6hhhejhRaqFEBmG_NwUS_9VXFZY2uYSoUNlFwY/edit?usp=sharing -->   <!-- Original drawing for histories_list_via_mesthead.svg = https://docs.google.com/drawings/d/1ReFUBwfVlbrKz0pJqY0xxnbb3EjRYJRzoq9bi1NqH-w/edit?usp=sharing --> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> View histories side-by-side <a href="/training-material/faqs/galaxy/histories_side_by_side_view.html"><i class="fas fa-external-link-alt"></i></a> </h4>
This FAQ demonstrates how to view histories side-by-sde
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-view-histories-side-by-side"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-view-histories-side-by-side" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: View histories side-by-side</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can view multiple Galaxy histories at once. This allows to better understand your analyses and also makes it possible to drag datasets between histories. This is called “<strong>History multiview</strong>”. The multiview can be enabled either view History menu or via the Activity Bar:</p>   <ol>   <li>     <p>Enabling Multiview via History menu is done by first clicking on the <i class="fas fa-bars" aria-hidden="true"></i><span class="visually-hidden">galaxy-history-options</span> “<strong>History options</strong>” drop-down and selecting <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> “<strong>Show Histories Side-by-Side option</strong>”:</p>     <p><img src="/training-material/shared/images/histories_side_by_side_view.svg" alt="Enabling side-by-side view using History Options menu" /></p>   </li>   <li>     <p>Clicking the <i class="fas fa-columns" aria-hidden="true"></i><span class="visually-hidden">galaxy-multihistory</span> “<strong>History Multiview</strong>” button within the Activity Bar:</p>     <p><img src="/training-material/shared/images/histories_multiview_via_activity_bar.svg" alt="Enabling side-by-side view using Activity Bar" /></p>   </li> </ol>   <p><!-- Original drawings are accessible from https://docs.google.com/drawings/d/1coRlf1PxDKZYFt9j83MHGVYvwhBZhBkE-PcnhgIQ6gs/edit?usp=sharing and https://docs.google.com/drawings/d/1ksdR7wf2i5CJA1Pa-frCLMlgfITwwpOh2cKay8xDIi0/edit?usp=sharing --></p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="interactive tools"> <a href="#interactive tools"> Interactive tools </a></h3>

<hr/>



<h4 class="faq-area"> Knitting RMarkdown documents in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-knitting-rmarkdown-documents-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Knitting RMarkdown documents in RStudio</div>   <p>One of the other nice features of RMarkdown documents is making lovely presentation-quality worthy documents. You can take, for example, a tutorial and produce a nice report like output as HTML, PDF, or <code class="language-plaintext highlighter-rouge">.doc</code> document that can easily be shared with colleagues or students.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-meta-output.png" alt="Screenshot of the metadata with html_notebook and word_document being visible and a number of options controlling their output. TOC, standing for table of contents, has been set to true for both." /></p>   <p>Now you’re <strong>ready to preview the document</strong>:</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of preview dropdown with options like preview, knit to html, knit to pdf, knit to word" /></p>   <p><strong>Click Preview</strong>. A window will popup with a preview of the rendered verison of this document.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of rendered document with the table of contents on left, title is in a large font, and there are coloured boxes similar to GTN tutorials offering tips and more information" /></p>   <p>The preview is really similar to the GTN rendering, no cells have been executed, and no output is embedded yet in the preview document. But if you have run cells (e.g. the first few loading a library and previewing the <code class="language-plaintext highlighter-rouge">msleep</code> dataset:</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-preview-output.png" alt="screenshot of the rendered document with a fancy table browser embedded as well as the output of each step" /></p>   <p><strong>When you’re ready to distribute</strong> the document, you can instead use the <code class="language-plaintext highlighter-rouge">Knit</code> button. This runs every cell in the entire document fresh, and then compiles the outputs together with the rendered markdown to produce a nice result file as HTML, PDF, or Word document.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-knit.png" alt="screenshot of the console with 'chunks' being knitted together" /></p>   <blockquote class="tip">   <h3 data-toc-skip id="tip-tip-pdf--word-require-a-latex-installation"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: PDF + Word require a LaTeX installation</h3>   <p>You might need to install additional packages to compile the PDF and Word document versions</p> </blockquote>   <p>And at the end you can see a pretty document rendered with all of the output of every step along the way. This is a fantastic way to e.g. distribute read-only lesson materials to students, if you feel they might struggle with using an RMarkdown document, or just want to read the output without <em>doing</em> it themselves.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-pdf.png" alt="screenshot of a PDF document showing the end of the tutorial where a pretty plot has been rendered and there is some text for conclusions and citations" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Launch JupyterLab <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch JupyterLab</div>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-jupyterlab-in-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-jupyterlab-in-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch JupyterLab in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently JupyterLab in Galaxy is available on <a href="https://live.usegalaxy.eu">Live.useGalaxy.eu</a>, <a href="https://usegalaxy.org">usegalaxy.org</a> and <a href="https://usegalaxy.eu">usegalaxy.eu</a>.</p>   <blockquote class="notranslate hands_on">     <div class="box-title hands-on-title" id="hands-on-run-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Run JupyterLab</div>     <ol>       <li><span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Interactive Jupyter Notebook tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Interactive Jupyter Notebook</strong></span>. Note that on some Galaxies this is called <span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Interactive JupyTool and notebook tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Interactive JupyTool and notebook</strong></span>:</li>       <li>Click Run Tool</li>       <li>         <p>The tool will start running and will stay running permanently</p>         <p>This may take a moment, but once the <code class="language-plaintext highlighter-rouge">Executed notebook</code> in your history is <strong>orange</strong>, you are up and running!</p>       </li>       <li>Click on the <strong>User</strong> menu at the top and go to <strong>Active Interactive Tools</strong> and locate the JupyterLab instance you started.</li>       <li>Click on your JupyterLab instance</li>     </ol>   </blockquote> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-try-jupyterlab-if-not-available-on-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-try-jupyterlab-if-not-available-on-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch Try JupyterLab if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If JupyterLab is not available on the Galaxy instance:</p>   <ol>     <li>Start <a href="https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/try.jupyter.org?urlpath=lab">Try JupyterLab</a></li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Launch RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Launch RStudio</div>   <p>Depending on which server you are using, you may be able to run RStudio directly in Galaxy. If that is not available, RStudio Cloud can be an alternative.</p>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-in-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-in-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio in Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Currently RStudio in Galaxy is only available on <a href="https://usegalaxy.eu">UseGalaxy.eu</a> and <a href="https://usegalaxy.org">UseGalaxy.org</a></p>   <ol>     <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here to launch RStudio</a></li>     <li>Click Run Tool</li>     <li>The tool will start running and will stay running permanently</li>     <li>Click on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started.</li>   </ol> </blockquote>   <blockquote class="tip">   <div class="box-title tip-title" id="tip-launch-rstudio-cloud-if-not-available-on-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-launch-rstudio-cloud-if-not-available-on-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true" ></i> <span>Tip: Launch RStudio Cloud if not available on Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If RStudio is not available on the Galaxy instance:</p>   <ol>     <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>     <li>Create a new project</li>   </ol> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Learning with RMarkdown in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-learning-with-rmarkdown-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Learning with RMarkdown in RStudio</div>   <p>Learning with RMarkdown is a bit different than you might be used to. Instead of copying and pasting code from the GTN into a document you’ll instead be able to run the code directly as it was written, inside RStudio! You can now focus just on the code and reading within RStudio.</p>   <ol>   <li>     <p><strong>Load the notebook</strong> if you have not already, following the tip box at the top of the tutorial</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-download-file.png" alt="Screenshot of the Console in RStudio. There are three lines visible of not-yet-run R code with the download.file statements which were included in the setup tip box." /></p>   </li>   <li>     <p><strong>Open it</strong> by clicking on the <code class="language-plaintext highlighter-rouge">.Rmd</code> file in the file browser (bottom right)</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-file-menu.png" alt="Screenshot of Files tab in RStudio, here there are three files listed, a data-science-r-dplyr.Rmd file, a css and a bib file." /></p>   </li>   <li>     <p>The RMarkdown document will appear in the document viewer (top left)</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-document-view.png" alt="Screenshot of an open document in RStudio. There is some yaml metadata above the tutorial showing the title of the tutorial." /></p>   </li> </ol>   <p>You’re now ready to view the RMarkdown notebook! Each notebook starts with a lot of metadata about how to build the notebook for viewing, but you can ignore this for now and <strong>scroll down to the content of the tutorial</strong>.</p>   <p>You can switch to the visual mode which is way easier to read - just click on the gear icon and select <code class="language-plaintext highlighter-rouge">Use Visual Editor</code>.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-visual-mode.png" alt="Screenshot of dropdown menu after clicking on the gear icon. The first option is `Use Visual Editor`." /></p>   <p>You’ll see codeblocks scattered throughout the text, and these are all runnable snippets that appear like this in the document:</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-run-cell.png" alt="Screenshot of the RMarkdown document in the viewer, a cell is visible between markdown text reading library tidyverse. It is slightly more grey than the background region, and it has a run button at the right of the cell in a contextual menu." /></p>   <p>And you have a few options for how to run them:</p>   <ol>   <li>Click the green arrow</li>   <li><kbd>ctrl+enter</kbd></li>   <li>     <p>Using the menu at the top to run all</p>     <p><img src="/training-material/topics/data-science/images/rstudio/r-run-all.png" alt="Screenshot of the run dropdown menu in R, the first item is run selected lines showing the mentioned shortcut above, the second is run next chunk, and then it also mentions a 'run all chunks below' and 'restart r and run all chunks' option." /></p>   </li> </ol>   <p>When you run cells, the output will appear below in the Console. RStudio essentially copies the code from the RMarkdown document, to the console, and runs it, just as if you had typed it out yourself!</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-cell-output.png" alt="Screenshot of a run cell, its output is included below in the RMarkdown document and the same output is visible below in the console. It shows a log of loading the tidyverse library." /></p>   <p>One of the best features of RMarkdown documents is that they include a very nice table browser which makes previewing results a lot easier! Instead of needing to use <code class="language-plaintext highlighter-rouge">head</code> every time to preview the result, you get an interactive table browser for any step which outputs a table.</p>   <p><img src="/training-material/topics/data-science/images/rstudio/r-table.png" alt="Screenshot of the table browser. Below a code chunk is a large white area with two images, the first reading 'r console' and the second reading 'tbl_df'. The tbl_df is highlighted like it is active. Below that is a pretty-printed table with bold column headers like name and genus and so on. At the right of the table is a small arrow indicating you can switch to seeing more columns than just the initial three. At the bottom of the table is 1-10 of 83 rows written, and buttons for switching between each page of results." /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Open a Terminal in Jupyter <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-open-a-terminal-in-jupyter"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Open a Terminal in Jupyter</div>   <p>This tutorial will let you accomplish almost everything from this view, running code in the cells below directly in the training material. You can choose between running the code here, or opening up a terminal tab in which to run it.Here are some instructions for how to do this on various environments.</p>   <p><strong>Jupyter</strong> on UseGalaxy.* and <strong>MyBinder.org</strong></p>   <ol>   <li>     <p>Use the File → New → Terminal menu to launch a terminal.</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-launch.png" alt="screenshot of jupyterlab showing the File menu expanded to show new and terminal option." /></p>   </li>   <li>     <p>Disable “Simple” mode in the bottom left hand corner, if it activated.</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-simple.png" alt="screenshot of jupyterlab showing a toggle labelled simple" /></p>   </li>   <li>     <p>Drag one of the terminal or notebook tabs to the side to have the training materials and terminal side-by-side</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-final.png" alt="screenshot of jupyterlab with notebook and terminal side-by-side." /></p>   </li> </ol>   <p><strong>CoCalc</strong></p>   <ol>   <li>     <p>Use the <em>Split View</em> functionality of cocalc to split your view into two portions.</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-split.png" alt="screenshot of cocalc button to split views" /></p>   </li>   <li>     <p>Change the view of one panel to a terminal</p>     <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-terminal.png" alt="screenshot of cocalc swapping view port to that of a terminal" /></p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Open interactive tool <a href="/training-material/faqs/galaxy/interactive_tools_open.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-open-interactive-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-open-interactive-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Open interactive tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Go to User &gt; Active InteractiveTools</li>   <li>Wait for the  to be running (Job Info)</li>   <li>Click on</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Stop RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Stop RStudio</div>   <p>When you have finished your R analysis, it’s time to stop RStudio.</p>   <ol>   <li>First, save your work into Galaxy, to ensure reproducibility:     <ol>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>       <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>     </ol>   </li>   <li>Once you have saved your data, you can proceed in 2 different ways:     <ul>       <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a “in progress state”, so yellow, OR</li>       <li>Clicking on the “User” menu at the top and go to “Active InteractiveTools” and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the “Stop” button at the bottom.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="reference genomes"> <a href="#reference genomes"> Reference genomes </a></h3>

<hr/>



<h4 class="faq-area"> How to use Custom Reference Genomes? <a href="/training-material/faqs/galaxy/reference_genomes_custom_genomes.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-use-custom-reference-genomes"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-use-custom-reference-genomes" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to use Custom Reference Genomes?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>A <strong>reference genome</strong> contains the nucleotide sequence of the chromosomes, scaffolds, transcripts, or contigs for single species. It is representative of a specific genome assembly build or release.</p>   <p>There are two options for reference genomes in Galaxy.</p>   <ul>   <li><strong>Native</strong>     <ul>       <li>Index provided by the server administrators.</li>       <li>Found on tool forms in a drop down menu.</li>       <li>A database key is automatically assigned. See tip 1.</li>       <li>The database is what links your data to a FASTA index. Example: used with BAM data</li>     </ul>   </li>   <li><strong>Custom</strong>     <ul>       <li>FASTA file uploaded by users.</li>       <li>Input on tool forms then indexed at runtime by the tool.</li>       <li>An optional custom database key can be created and <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">assigned by the user</a>.</li>     </ul>   </li> </ul>   <p>There are five basic steps to use a <strong>Custom Reference Genome</strong>, plus one optional.</p>   <ol>   <li>Obtain a FASTA copy of the target genome. See tip 2.</li>   <li>Upload the genome to Galaxy and to add it as a dataset in your history.</li>   <li><a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">Clean up the format</a> with the tool <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace.</li>   <li>Make sure the <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html">chromosome identifiers</a> are a match for other inputs.</li>   <li>Set a tool form’s options to use a custom reference genome from the history and select the loaded genome FASTA.</li>   <li>(Optional) Create a <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">custom genome build’s database</a> that you can <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">assign to datasets</a>.</li> </ol>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> TIP 1: Avoid <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">assigning a native database</a> to uploaded data unless you confirmed the data are based on the <a href="/training-material/faqs/galaxy/datasets_chromosome_identifiers.html">same exact genome assembly</a> or you <a href="/training-material/topics/introduction/tutorials/data-manipulation-olympics/tutorial.html">adjusted the data to be a match</a> <strong>first</strong>!</p>   <p><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> TIP 2: When choosing your reference genome, consider <a href="/training-material/faqs/galaxy/analysis_differential_expression_help.html">choosing your reference annotation</a> at the same time. Standardize the format of both as a preparation step. Put the files in a dedicated “reference data” history for easy reuse.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Sorting Reference Genome <a href="/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sorting-reference-genome"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sorting-reference-genome" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sorting Reference Genome</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Certain tools expect that reference genomes are sorted in <a href="https://en.wikipedia.org/wiki/Lexicographic_order">lexicographical order</a>. These tools are often downstream of the initial mapping tools, which means that a large investment in a project has already been made, before a problem with sorting pops up in conclusion layer tools. How to avoid? Always sort your FASTA reference genome dataset at the beginning of a project. Many sources only provide sorted genomes, but double checking is your own responsibility, and super easy in Galaxy!</p>   <ol>   <li>Convert Formats -&gt; <strong>FASTA-to-Tabular</strong></li>   <li>Filter and Sort -&gt; <strong>Sort</strong>     on column: c1      with flavor: Alphabetical     everything in: Ascending order</li>   <li>Convert Formats -&gt; <strong>Tabular-to-FASTA</strong></li> </ol>   <p>Note: The above sorting method is for most tools, but not all. In particular, GATK tools have a tool-specific sort order requirement.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Troubleshooting Custom Genome fasta <a href="/training-material/faqs/galaxy/reference_genomes_troubleshooting_custom_genome.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-troubleshooting-custom-genome-fasta"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-troubleshooting-custom-genome-fasta" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Troubleshooting Custom Genome fasta</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If a custom genome/transcriptome/exome dataset is producing errors, double check the format and that the chromosome identifiers between <strong>ALL</strong> inputs. Clicking on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden">galaxy-bug</span> will often provide a description of the problem. This does not automatically submit a bug report, and it is not always necessary to do so, but it is a good way to get some information about <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">why a job is failing</a>.</p>   <ul>   <li>     <p>Custom genome not assigned as FASTA format</p>     <ul>       <li><strong>Symptoms include</strong>: Dataset not included in custom genome “From history” pull down menu on tool forms.</li>       <li><strong>Solution</strong>: Check datatype assigned to dataset and assign <strong>fasta</strong> format.</li>       <li><strong>How</strong>: Click on the dataset’s pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the “Edit Attributes” form, and in the Datatypes tab &gt; <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">redetect the datatype</a>.</li>       <li>If <code class="language-plaintext highlighter-rouge">fasta</code> is not assigned, there is a format problem to correct.</li>     </ul>   </li>   <li>     <p>Incomplete Custom genome file load</p>     <ul>       <li><strong>Symptoms include</strong>: Tool errors result the first time you use the Custom genome.</li>       <li><strong>Solution</strong>: Use <strong>Text Manipulation → Select last lines from a dataset</strong> to check last 10 lines to see if file is truncated.</li>       <li><strong>How</strong>: Reload the dataset (switch to <a href="https://galaxyproject.org/ftp-upload/">FTP</a> if not using already). Check your FTP client logs to make sure the load is complete.</li>     </ul>   </li>   <li>     <p>Extra spaces, extra lines, inconsistent line wrapping, or any deviation from strict FASTA format</p>     <ul>       <li><strong>Symptoms include: RNA-seq tools (Cufflinks, Cuffcompare, Cuffmerge, Cuffdiff)</strong> fails with error <code class="language-plaintext highlighter-rouge">Error: sequence lines in a FASTA record must have the same length!.</code></li>       <li><strong>Solution</strong>: File tested and corrected locally then re-upload or test/fix within Galaxy, then re-run.</li>       <li><strong>How</strong>:         <ul>           <li><strong>Quick re-formatting</strong> Run the tool through the tool <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace.</li>           <li><strong>Optional Detailed re-formatting</strong> Start with <strong>FASTA manipulation → FASTA Width formatter</strong> with a value between 40-80 (60 is common) to reformat wrapping. Next, use Filter and Sort → Select with “&gt;” to examine identifiers. Use a combination of <strong>Convert Formats → FASTA-to-Tabular, Text Manipulation</strong> tools, then <strong>Tabular-to-FASTA</strong> to correct.</li>           <li><strong>With either of the above</strong>, finish by using <strong>Filter and Sort → Select</strong> with <code class="language-plaintext highlighter-rouge">^\w*$</code> to search for empty lines (use “NOT matching” to remove these lines and output a properly format fasta dataset).</li>         </ul>       </li>     </ul>   </li>   <li>     <p>Inconsistent line wrapping, common if merging chromosomes from various Genbank records (e.g. primary chroms with mito)</p>     <ul>       <li><strong>Symptoms include</strong>: Tools (<strong>SAMTools, Extract Genomic DNA</strong>, but rarely alignment tools) may complain about unexpected line lengths/missing identifiers. Or they may just fail for what appears to be a cluster error.</li>       <li><strong>Solution</strong>: File tested and corrected locally then re-upload or test/fix within Galaxy.</li>       <li><strong>How</strong>: Use <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace. Finish by using <strong>Filter and Sort → Select</strong> with <code class="language-plaintext highlighter-rouge">^\w*$</code> to search for empty lines (use “NOT matching” to remove these lines and output a properly format fasta dataset).</li>     </ul>   </li>   <li>     <p>Unsorted fasta genome file</p>     <ul>       <li><strong>Symptoms include</strong>: Tools such as <strong>Extract Genomic DNA</strong> report problems with sequence lengths.</li>       <li><strong>Solution</strong>: First try sorting and re-formatting in Galaxy then re-run.</li>       <li><strong>How</strong>: To sort, follow instructions for <a href="/training-material/faqs/galaxy/reference_genomes_sorting_reference_genome.html">Sorting</a> a Custom Genome.</li>     </ul>   </li>   <li>     <p>Identifier and Description in “&gt;” title lines used inconsistently by tools in the same analysis</p>     <ul>       <li><strong>Symptoms include</strong>: Will generally manifest as a false genome-mismatch problem.</li>       <li><strong>Solution</strong>: Remove the description content and re-run all tools/workflows that used this input. Mapping tools will usually not fail, but downstream tools will. When this comes up, it usually means that an analysis needs to be started over from the mapping step to correct the problems. No one enjoys redoing this work. Avoid the problems by formatting the genome, by double checking that the same reference genome was used for all steps, and by making certain the ‘identifiers’ are a match between all planned inputs (including reference annotation such as GTF data) <strong>before using your custom genome</strong>.</li>       <li><strong>How</strong>: To drop the title line description content, use <strong>NormalizeFasta</strong> using the options to wrap sequence lines at 80 bases and to trim the title line at the first whitespace. Next, double check that the chromosome identifiers are an exact match between all inputs.</li>     </ul>   </li>   <li>     <p>Unassigned database</p>     <ul>       <li><strong>Symptoms include</strong>: Tools report that no build is available for the assigned reference genome.</li>       <li><strong>Solution</strong>: This occurs with tools that require an assigned database metadata attribute. <strong>SAMTools and Picard</strong> often require this assignment.</li>       <li><strong>How</strong>: Create a <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html">Custom Build</a> and assign it to the dataset.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="reports"> <a href="#reports"> Reports </a></h3>

<hr/>



<h4 class="faq-area"> Enhancing tabular dataset previews in reports/pages <a href="/training-material/faqs/galaxy/reports_tables.html"><i class="fas fa-external-link-alt"></i></a> </h4>
There are lots of fun advanced features!
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-enhancing-tabular-dataset-previews-in-reports-pages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-enhancing-tabular-dataset-previews-in-reports-pages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Enhancing tabular dataset previews in reports/pages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There are a number of options, specifically for tabular data, that can allow it to render more nicely in your workflow reports and pages and anywhere that GalaxyMarkdown is used.</p>   <ul>   <li><code class="language-plaintext highlighter-rouge">title</code> to give your table a title</li>   <li><code class="language-plaintext highlighter-rouge">footer</code> allows you to caption your table</li>   <li><code class="language-plaintext highlighter-rouge">show_column_headers=false</code> to hide the column headers</li>   <li><code class="language-plaintext highlighter-rouge">compact=true</code> to make the table show up more inline, hiding that it was embedded from a Galaxy dataset.</li> </ul>   <p>The existing <code class="language-plaintext highlighter-rouge">history_dataset_display</code> directive displays the dataset name and some useful context at the expense of potentially breaking the flow of the document</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_display(history_dataset_id=1e8ab44153008be8) </span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_display.png" alt="a tabular dataset rendered, it has a title and a download button and sortable columns" /></p> </blockquote>   <p>The existing <code class="language-plaintext highlighter-rouge">history_dataset_embedded</code> directive was implemented to try to inline results more and make the results more readable within a more… curated document. It is dispatches on tabular types and puts the results in a table but the table doesn’t have a lot of options.</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-1"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_embedded(history_dataset_id=1e8ab44153008be8) </span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-1"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_embedded.png" alt="the same as before but no title nor download button. just a rendered table with sortable columns" /></p> </blockquote>   <p>The <code class="language-plaintext highlighter-rouge">history_dataset_as_table</code> directive mirrors the <code class="language-plaintext highlighter-rouge">history_dataset_as_image</code> directive: it tries harder to coerce the data into a table and provides new table—specific options. The first of these is “show_column_headers<code class="language-plaintext highlighter-rouge"> which defaults to </code>true`.</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-2"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_as_table(history_dataset_id=1e8ab44153008be8,show_column_headers=false)</span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-2"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_as_table.png" alt="the same as before but no title nor download button nor column headers" /></p> </blockquote>   <p>There is also a <code class="language-plaintext highlighter-rouge">compact</code> option. This provides a much more inline experience for tabular datasets:</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-3"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_as_table(history_dataset_id=1e8ab44153008be8,show_column_headers=false,compact=true)</span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-3"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_compact.png" alt="again the same screenshot, no table metadata, and now it lacks the small margin around it." /></p> </blockquote>   <p>Figures in general should have titles and legends — so there is the “title” and “footer” options also.</p>   <blockquote class="code-in">   <div class="box-title code-in-title" id="code-in-galaxy-markdown-4"><i class="far fa-keyboard" aria-hidden="true" ></i> Input: Galaxy Markdown</div>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl">galaxy<br></span><span class="sb">history_dataset_as_table(history_dataset_id=1e8ab44153008be8,show_column_headers=false,title='Binding Site Results',footer='Here is a very good figure caption for this table.')</span><br><span class="p">```</span><br></code></pre></div>  </div> </blockquote>   <blockquote class="code-out">   <div class="box-title code-out-title" id="code-out-example-screenshot-4"><i class="fas fa-laptop-code" aria-hidden="true" ></i> Output: Example Screenshot</div>   <p><img src="/training-material/faqs/galaxy/images/report_table_history_dataset_title.png" alt="the same table with now a tasteful title and small caption below it describing that the author would write a caption if he knew what a binding site was." /></p> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Making an element collapsible in a report <a href="/training-material/faqs/galaxy/reports_collapsible_box.html"><i class="fas fa-external-link-alt"></i></a> </h4>
If you have extraneous information you might want to let a user collapse it.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-making-an-element-collapsible-in-a-report"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-making-an-element-collapsible-in-a-report" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Making an element collapsible in a report</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>This applies to any GalaxyMarkdown elements, i.e. the things you’ve clicked in the left panel to embed in your Workflow Report or Page</p>   <p>By adding a <code class="language-plaintext highlighter-rouge">collapse=""</code> attribute to a markdown element, you can make it collapsible. Whatever you put in the quotes will be the title of the collapsible box.</p>   <div class="language-markdown highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="p">```</span><span class="nl"><br></span>history_dataset_type(history_dataset_id=3108c91feeb505da, collapse="[TITLE]")<br><span class="p">```</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="sequencing"> <a href="#sequencing"> Sequencing </a></h3>

<hr/>



<h4 class="faq-area"> Illumina MiSeq sequencing <a href="/training-material/faqs/galaxy/sequencing_illumina_miseq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-illumina-miseq-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Illumina MiSeq sequencing</div>   <p>Illumina MiSeq sequencing is based on sequencing by synthesis. As the name suggests, fluorescent labels are measured for every base that bind at a specific moment at a specific place on a flow cell. These flow cells are covered with oligos (small single strand DNA strands). In the library preparation the DNA strands are cut into small DNA fragments (differs per kit/device) and specific pieces of DNA (adapters) are added, which are complementary to the oligos. Using bridge amplification large amounts of clusters of these DNA fragments are made. The reverse string is washed away, making the clusters single stranded. Fluorescent bases are added one by one, which emit a specific light for different bases when added. This is happening for whole clusters, so this light can be detected and this data is basecalled (translation from light to a nucleotide) to a nucleotide sequence (Read). For every base a quality score is determined and also saved per read. This process is repeated for the reverse strand on the same place on the flow cell, so the forward and reverse reads are from the same DNA strand. The forward and reversed reads are linked together and should always be processed together!</p>   <p>For more information watch this <a href="https://youtu.be/fCd6B5HRaZ8">video from Illumina</a></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Nanopore sequencing <a href="/training-material/faqs/galaxy/sequencing_nanopore.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">   <div class="box-title comment-title" id="comment-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i> Comment: Nanopore sequencing</div>   <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>   <ol>   <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>   <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>   <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li> </ol>   <p><img src="/training-material/topics/microbiome/images/plasmid-metagenomics-nanopore/nanopore_sequencing.png" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://doi.org/10.7875/togopic.2020.01&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>   <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change in electrical current is measured over the membrane of a flow cell. When nucleotides pass the pores in the flow cell the current change is translated (basecalled) to nucleotides by a basecaller. A schematic overview is given in the picture above.</p>   <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software is present on the devices. With basecalling the electrical signals are translated to bases (A,T,G,C) with a quality score per base. The sequenced DNA strand will be basecalled and this will form one read. Multiple reads will be stored in a fastq file.</p> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="support"> <a href="#support"> Support </a></h3>

<hr/>



<h4 class="faq-area"> Can I use a public Galaxy for my private data? <a href="/training-material/faqs/galaxy/can-i-use-a-public-galaxy-for-my-private-data.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-can-i-use-a-public-galaxy-for-my-private-data"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-can-i-use-a-public-galaxy-for-my-private-data" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Can I use a public Galaxy for my private data?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Of course*!</p>   <p>If your data is not sensitive (i.e. human patient) but just private (sequencing from other animals/bacteria/etc), then it is absolutely ok to use a public galaxy server like usegalaxy.eu or usegalaxy.org!</p>   <p>Data uploaded is private to your account, it isn’t available to others publicly. No one will scoop your results, if you use a public galaxy server to analyse your data :)</p>   <p>A great benefit of this is then when your paper is being reviewed you can share that history or workflow with reviewers, and when it’s published you can click a button to share those results with the world as well, such that others can reproduce your analysis!</p>   <p>(of course system administrators can see the files on disk but they are not interested and will not be looking at your data. If you file a bug report they may see your data but they are system administrators, not bioinformatics experts that might be interested in your results.)</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Contacting Galaxy Administrators <a href="/training-material/faqs/galaxy/support_admins.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-contacting-galaxy-administrators"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-contacting-galaxy-administrators" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Contacting Galaxy Administrators</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you suspect there is something wrong with the server, or would like to request a tool to be installed, you should contact the server administrators for the Galaxy you are on.</p>   <ul>   <li><strong>Tool error?</strong> Please follow these <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">troubleshooting steps</a></li>   <li>Each Galaxy server has different contact procedures, here are the contact options for the 3 biggest servers:     <ul>       <li><strong>Galaxy US:</strong> <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a></li>       <li><strong>Galaxy EU:</strong> <a href="https://gitter.im/usegalaxy-eu/Lobby">Gitter channel</a>, <a href="https://usegalaxy.eu/tiaas/">Request TIaaS</a></li>       <li><strong>Galaxy AU:</strong> <a href="mail:help@genome.edu.au">Email</a>, <a href="https://request.usegalaxy.org.au/">Request a tool</a>, <a href="https://docs.google.com/forms/d/e/1FAIpQLSeiw6ajmkezLCwbXc3OFQEU3Ai9hGnBd967u9YbQ8ANPgvatA/viewform">Request Data Quota</a></li>       <li><strong>Galaxy FR:</strong> <a href="https://usegalaxy.fr/tiaas/">Request TIaaS</a></li>       <li>Other Galaxy servers? Check the homepage for more information.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where do I get more support? <a href="/training-material/faqs/galaxy/support.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-do-i-get-more-support"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-do-i-get-more-support" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where do I get more support?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>If you need support for using Galaxy, running your analysis or completing a tutorial, please try one of the following options:</p>   <ul>   <li><strong>Gitter Chat</strong>: You can get help on Gitter chat platform, on various channels.     <ul>       <li><a href="https://gitter.im/galaxyproject/Lobby">Galaxy General Support</a></li>       <li><a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Training Support</a></li>       <li><a href="https://gitter.im/usegalaxy-eu/Lobby">Galaxy EU Server</a></li>     </ul>   </li>   <li>     <p><strong>Galaxy Help Forum</strong>: You can also have a look at the <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>. Your question may already have been answered here before. If not, you can post your question here.</p>   </li>   <li><strong>Contact Server Admins:</strong> If you think there is a problem with the Galaxy server, or you would like to make a request, <a href="/training-material/faqs/galaxy/support_admins.html">contact the Galaxy server administrators</a>.</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="tools"> <a href="#tools"> Tools </a></h3>

<hr/>



<h4 class="faq-area"> Add Toolshed category to a tool <a href="/training-material/faqs/galaxy/add-toolshed-category-to-a-tool.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-add-toolshed-category-to-a-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-add-toolshed-category-to-a-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Add Toolshed category to a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p>Find the target tool in the <a href="https://toolshed.g2.bx.psu.edu/repository">Galaxy Toolshed</a>.</p>     <p>Note: the easiest way to do this from the Galaxy interface is to (A) search for the tool, then (B) select the drop-down menu <strong>See tool in toolshed</strong>.</p>   </li>   <li>Follow the <strong>Development respository</strong> URL.</li>   <li>Go to the <code class="language-plaintext highlighter-rouge">.shed.yml</code> file.</li>   <li>In the <code class="language-plaintext highlighter-rouge">categories:</code> metadata section, add your Toolshed category (which must correspond to those already in the <a href="https://toolshed.g2.bx.psu.edu/repository">Galaxy Toolshed</a>.</li> </ol>   <p>Example format:</p>   <div class="language-yaml highlighter-rouge"><div class="highlight"><pre class="highlight"><code><span class="na">categories</span><span class="pi">:</span><br> <span class="pi">-</span> <span class="s">Single Cell</span><br> <span class="pi">-</span> <span class="s">Spatial Omics</span><br> <span class="pi">-</span> <span class="s">Transcriptomics</span><br></code></pre></div></div> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Changing the tool version <a href="/training-material/faqs/galaxy/tools_change_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-tool-version"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-tool-version" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the tool version</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool.</p>   <p><strong>Switching to a different version</strong> of a tool:</p>   <ul>   <li>Open the tool</li>   <li>Click on the <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> versions logo at the <strong>top right</strong></li>   <li>Select the desired version from the dropdown list</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> If a Tool is Missing <a href="/training-material/faqs/galaxy/tools_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-if-a-tool-is-missing"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-if-a-tool-is-missing" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: If a Tool is Missing</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To use the tools installed and available on the Galaxy server:</p>   <ol>   <li>At the top of the left tool panel, type in a tool name or datatype into the tool search box.</li>   <li>Shorter keywords find more choices.</li>   <li>Tools can also be directly browsed by category in the tool panel.</li> </ol>   <p>If you can’t find a tool you need for a tutorial on Galaxy, please:</p>   <ol>   <li><strong>Check that you are using a compatible Galaxy server</strong>     <ul>       <li>Navigate to the overview box at the top of the tutorial</li>       <li>Find the <strong>“Supporting Materials”</strong> section</li>       <li>Check <strong>“Available on these Galaxies”</strong></li>       <li>If your server is not listed here, the tutorial is not supported on your Galaxy server</li>       <li>You can create an account on one of the supporting Galaxies   <img src="/training-material/faqs/galaxy/images/supporting_galaxies.png" alt="screenshot of overview box with available Galaxies section" /></li>     </ul>   </li>   <li>Use the <strong>Tutorial mode</strong> feature     <ul>       <li>Open your Galaxy server</li>       <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>       <li>Navigate to your tutorial</li>       <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>       <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>     </ul>   </li>   <li><strong>Still not finding the tool?</strong>     <ul>       <li>Ask help in <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a>.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Multipile similar tools available <a href="/training-material/faqs/galaxy/tools_similar.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-multipile-similar-tools-available"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-multipile-similar-tools-available" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Multipile similar tools available</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes there are multiple tools with very similar names. If the parameters in the tutorial don’t match with what you see in Galaxy, please try the following:</p>   <ol>   <li>     <p>Use <strong>Tutorial Mode</strong> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> in Galaxy, and click on the blue tool button in the tutorial to automatically open the correct tool and version (not available for all tutorials yet)</p>     <!--SNIPPET-->     <blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote>     <p><!--END_SNIPPET--></p>   </li>   <li>     <p>Check that the <strong>entire tool name</strong> matches what you see in the tutorial.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Organizing the tool panel <a href="/training-material/faqs/galaxy/tools_favorite.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-organizing-the-tool-panel"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-organizing-the-tool-panel" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Organizing the tool panel</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Galaxy servers can have a lot of tools available, which can make it challenging to find the tool you are looking for. To help find your favourite tools, you can:</p>   <ul>   <li><strong>Keep a list of your favourite tools</strong> to find them back easily later.     <ul>       <li><strong>Adding tools to your favourites</strong>         <ul>           <li>Open a tool</li>           <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> next to the tool name to add it to your favourites</li>         </ul>       </li>       <li><strong>Viewing your favourite tools</strong>         <ul>           <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> at the top of the Galaxy tool panel (above the tool search bar)</li>           <li>This will filter the toolbox to show all your starred tools</li>         </ul>       </li>     </ul>   </li>   <li><strong>Change the tool panel view</strong>     <ul>       <li>Click on the <i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden">galaxy-panelview</span> icon at the top of the Galaxy tool panel (above the tool search bar)</li>       <li>Here you can view the tools by EDAM ontology terms         <ul>           <li>EDAM Topics (e.g. biology, ecology)</li>           <li>EDAM Operations (e.g. quality control, variant analysis)</li>           <li>You can always get back to the default view by choosing <strong>“Full Tool Panel”</strong></li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Re-running a tool <a href="/training-material/faqs/galaxy/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-re-running-a-tool"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-re-running-a-tool" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Re-running a tool</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>   <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden">galaxy-refresh</span> the tool</li> </ol>   <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Regular Expressions 101 <a href="/training-material/faqs/galaxy/analysis_regular_expressions.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-regular-expressions-101"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-regular-expressions-101" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Regular Expressions 101</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Regular expressions are a standardized way of describing patterns in textual data. They can be extremely useful for tasks such as finding and replacing data. They can be a bit tricky to master, but learning even just a few of the basics can help you get the most out of Galaxy.</p>   <p><strong>Finding</strong></p>   <p>Below are just a few examples of basic expressions:</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">abc</code></td>       <td>an occurrence of <code class="language-plaintext highlighter-rouge">abc</code> within your data</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(abc|def)</code></td>       <td><code class="language-plaintext highlighter-rouge">abc</code> <em>or</em> <code class="language-plaintext highlighter-rouge">def</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[abc]</code></td>       <td>a single character which is either <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, or <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[^abc]</code></td>       <td>a character that is NOT <code class="language-plaintext highlighter-rouge">a</code>, <code class="language-plaintext highlighter-rouge">b</code>, nor <code class="language-plaintext highlighter-rouge">c</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-z]</code></td>       <td>any lowercase letter</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[a-zA-Z]</code></td>       <td>any letter (upper or lower case)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">[0-9]</code></td>       <td>numbers 0-9</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d</code></td>       <td>any digit (same as <code class="language-plaintext highlighter-rouge">[0-9]</code>)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\D</code></td>       <td>any non-digit character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\w</code></td>       <td>any alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\W</code></td>       <td>any non-alphanumeric character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\s</code></td>       <td>any whitespace</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\S</code></td>       <td>any non-whitespace character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.</code></td>       <td><em>any</em> character</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">\.</code></td>       <td> </td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">{x,y}</code></td>       <td>between <em>x</em> and <em>y</em> repetitions</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^</code></td>       <td>the beginning of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">$</code></td>       <td>the end of the line</td>     </tr>   </tbody> </table>   <p><strong>Note:</strong> you see that characters such as <code class="language-plaintext highlighter-rouge">*</code>, <code class="language-plaintext highlighter-rouge">?</code>, <code class="language-plaintext highlighter-rouge">.</code>, <code class="language-plaintext highlighter-rouge">+</code> etc have a special meaning in a regular expression. If you want to match on those characters, you can escape them with a backslash. So <code class="language-plaintext highlighter-rouge">\?</code> matches the question mark character exactly.</p>   <p><strong>Examples</strong></p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>matches</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">\d{4}</code></td>       <td>4 digits (e.g. a year)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr\d{1,2}</code></td>       <td><code class="language-plaintext highlighter-rouge">chr</code> followed by 1 or 2 digits</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">.*abc$</code></td>       <td>anything with <code class="language-plaintext highlighter-rouge">abc</code> at the end of the line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^$</code></td>       <td>empty line</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^&gt;.*</code></td>       <td>Line starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta header)</td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">^[^&gt;].*</code></td>       <td>Line not starting with <code class="language-plaintext highlighter-rouge">&gt;</code> (e.g. Fasta sequence)</td>     </tr>   </tbody> </table>   <p><strong>Replacing</strong></p>   <p>Sometimes you need to capture the exact value you matched on, in order to use it in your replacement, we do this using capture groups <code class="language-plaintext highlighter-rouge">(...)</code>, which we can refer to using <code class="language-plaintext highlighter-rouge">\1</code>, <code class="language-plaintext highlighter-rouge">\2</code> etc for the first and second captured values. If you want to refer to the whole match, use <code class="language-plaintext highlighter-rouge">&amp;</code>.</p>   <table>   <thead>     <tr>       <th>Regular expression</th>       <th>Input</th>       <th>Captures</th>     </tr>   </thead>   <tbody>     <tr>       <td><code class="language-plaintext highlighter-rouge">chr(\d{1,2})</code></td>       <td><code class="language-plaintext highlighter-rouge">chr14</code></td>       <td><code class="language-plaintext highlighter-rouge">\1 = 14</code></td>     </tr>     <tr>       <td><code class="language-plaintext highlighter-rouge">(\d{2}) July (\d{4})</code></td>       <td>24 July 1984</td>       <td><code class="language-plaintext highlighter-rouge">\1 = 24</code>, <code class="language-plaintext highlighter-rouge">\2 = 1984</code></td>     </tr>   </tbody> </table>   <p>An expression like <code class="language-plaintext highlighter-rouge">s/find/replacement/g</code> indicates a replacement expression, this will search (<code class="language-plaintext highlighter-rouge">s</code>) for any occurrence of <code class="language-plaintext highlighter-rouge">find</code>, and replace it with <code class="language-plaintext highlighter-rouge">replacement</code>. It will do this globally (<code class="language-plaintext highlighter-rouge">g</code>) which means it doesn’t stop after the first match.</p>   <p>Example: <code class="language-plaintext highlighter-rouge">s/chr(\d{1,2})/CHR\1/g</code> will replace <code class="language-plaintext highlighter-rouge">chr14</code> with <code class="language-plaintext highlighter-rouge">CHR14</code> etc.</p>   <p>You can also use replacement modifier such as convert to lower case <code class="language-plaintext highlighter-rouge">\L</code> or upper case <code class="language-plaintext highlighter-rouge">\U</code>. Example: <code class="language-plaintext highlighter-rouge">s/.*/\U&amp;/g</code> will convert the whole text to upper case.</p>   <p>Note: In Galaxy, you are often asked to provide the find and replacement expressions separately, so you don’t have to use the <code class="language-plaintext highlighter-rouge">s/../../g</code> structure.</p>   <p>There is a lot more you can do with regular expressions, and there are a few different flavours in different tools/programming languages, but these are the most important basics that will already allow you to do many of the tasks you might need in your analysis.</p>   <p><strong>Tip:</strong> <a href="https://regexone.com/">RegexOne</a> is a nice interactive tutorial to learn the basics of regular expressions.</p>   <p><strong>Tip:</strong> <a href="https://regex101.com/">Regex101.com</a> is a great resource for interactively testing and constructing your regular expressions, it even provides an explanation of a regular expression if you provide one.</p>   <p><strong>Tip:</strong> <a href="https://extendsclass.com/regex-tester.html">Cyrilex</a> is a visual regular expression tester.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Request Galaxy tools on a specific server <a href="/training-material/faqs/galaxy/request-galaxy-tools-on-a-specific-server.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-request-galaxy-tools-on-a-specific-server"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-request-galaxy-tools-on-a-specific-server" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Request Galaxy tools on a specific server</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To request tools that already exist in the <a href="https://toolshed.g2.bx.psu.edu">Galaxy toolshed</a>, but not in your server, please raise an issue at:</p>   <ul>   <li>     <p><em>Europe - usegalaxy.eu</em> | https://github.com/usegalaxy-eu/usegalaxy-eu-tools</p>   </li>   <li>     <p><em>USA - usegalaxy.org</em> | https://github.com/galaxyproject/usegalaxy-tools</p>   </li>   <li>     <p><em>Australia - usegaalxy.org.au</em> | https://github.com/usegalaxy-au/usegalaxy-au-tools/tree/master/usegalaxy.org.au</p>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Select multiple datasets <a href="/training-material/faqs/galaxy/tools_select_multiple_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Selecting a dataset collection as input <a href="/training-material/faqs/galaxy/tools_select_collection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-selecting-a-dataset-collection-as-input"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-selecting-a-dataset-collection-as-input" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Selecting a dataset collection as input</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in  front of the input parameter you want to supply the collection to.</li>   <li>Select the collection you want to use from the list</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Sorting Tools <a href="/training-material/faqs/galaxy/tools_sorting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-sorting-tools"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-sorting-tools" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Sorting Tools</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Sometimes input errors are caused because of non-sorted inputs. Try using these:</p>   <ul>   <li><strong>Picard SortSam:</strong> Sort SAM/BAM by coordinate or queryname.</li>   <li><strong>Samtools Sort:</strong> Alternate for SAM/BAM, best when used for coordinate sorting only.</li>   <li><strong>SortBED order the intervals:</strong> Best choice for BED/Interval.</li>   <li><strong>Sort data in ascending or descending order:</strong> Alternate choice for Tabular/BED/Interval/GTF.</li>   <li><strong>VCFsort:</strong> Best choice for VFC.</li>   <li><strong>Tool Form Options for Sorting:</strong> Some tools have an option to sort inputs during job execution. Whenever possible, sort inputs before using tools, especially if jobs fail for not having enough memory resources.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Tool doesn't recognize input datasets <a href="/training-material/faqs/galaxy/tools_tool_doesnt_recognize_any_input_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-tool-doesn-t-recognize-input-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-tool-doesn-t-recognize-input-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Tool doesn't recognize input datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The expected input datatype assignment is explained on the tool form. Review the input select areas and the help section below the Run Tool button.</p>   <p>Understanding datatypes <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html">FAQ</a>.</p>   <p>No datasets or collections available? Solutions:</p>   <ol>   <li>Upload or Copy an appropriate dataset for the input into the active history.     <ul>       <li>To load new datasets, review the <strong>Upload</strong> tool and more choices under <strong>Get Data</strong> within Galaxy.</li>       <li>To copy datasets from a different history into the active history see <a href="/training-material/faqs/galaxy/histories_copy_dataset.html">this FAQ</a>.</li>       <li>To use datasets loaded into a shared Data Library see <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html">this FAQ</a>.</li>     </ul>   </li>   <li>Resolve a datatype assignment incompatibility between the dataset and the tool.     <ul>       <li>To redetect a datatype see <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">this FAQ</a>.</li>       <li>To convert a datatype see <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">this FAQ</a>.</li>       <li>To change a datatype see <a href="/training-material/faqs/galaxy/datasets_change_datatype.html">this FAQ</a>.</li>     </ul>   </li>   <li>Individual datasets and dataset collections are selected differently on tool forms.     <ul>       <li>To select a collection input on a tool form see <a href="/training-material/faqs/galaxy/tools_select_collection.html">this FAQ</a>.</li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using tutorial mode <a href="/training-material/faqs/galaxy/tutorial_mode.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tutorial mode saves you screen space, finds the tools you need, and ensures you use the correct versions for the tutorials to run.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-tutorial-mode-1"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-tutorial-mode-1" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using tutorial mode</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools are frequently updated to new versions. Your Galaxy may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step… this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>   <ul>   <li>Open your Galaxy server</li>   <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside Galaxy.</li>   <li>Navigate to your tutorial</li>   <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>   <li><strong>Note:</strong> this does not work for all tutorials (yet)   <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>   <li>You can click anywhere in the grey-ed out area outside of the tutorial box to return back to the Galaxy analytical interface</li> </ul>   <blockquote class="warning">   <div class="box-title warning-title" id="warning-not-all-browsers-work-1"><i class="fas fa-exclamation-triangle" aria-hidden="true" ></i> Warning: Not all browsers work!</div>   <ul>     <li>We’ve had some issues with Tutorial mode on Safari for Mac users.</li>     <li>Try a <strong>different browser</strong> if you aren’t seeing the button.</li>   </ul> </blockquote> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Viewing tool logs (`stdout` and `stderr`) <a href="/training-material/faqs/galaxy/tools_logs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-viewing-tool-logs-stdout-and-stderr"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-viewing-tool-logs-stdout-and-stderr" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Viewing tool logs (`stdout` and `stderr`)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Most tools create log files as output, which can contain useful information about how the tool ran (<code class="language-plaintext highlighter-rouge">stdout</code>, or <em>standard output</em>), and what went wrong (<code class="language-plaintext highlighter-rouge">stderr</code>, or <em>standard error</em>).</p>   <p>To view these log files in Galaxy:</p>   <ul>   <li>Expand one of the outputs of the tool in your history</li>   <li>Click on <strong>View details</strong> <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span></li>   <li>Scroll to the <strong>Job Information</strong> section     <ul>       <li>Here you will find links to the log files (<code class="language-plaintext highlighter-rouge">stdout</code> and <code class="language-plaintext highlighter-rouge">stderr</code>).</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Where is the tool help? <a href="/training-material/faqs/galaxy/tools_help.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Finding tool support
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-where-is-the-tool-help"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-where-is-the-tool-help" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Where is the tool help?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>There is documentation available on the tool form itself which mentions the following information:</p>   <ul>   <li>Parameters</li>   <li>Expected format for input dataset(s)</li>   <li>Links to publications and ToolShed source repositories</li>   <li>Tool and wrapper version(s)</li>   <li>3rd party author web sites and documentation</li> </ul>   <p><strong>Scroll down on the tool form</strong> to locate:</p>   <ul>   <li>Information about expected inputs/outputs</li>   <li>Expanded definitions</li>   <li>Sample data</li>   <li>Example use cases</li>   <li>Graphics</li> </ul> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="troubleshooting"> <a href="#troubleshooting"> Troubleshooting </a></h3>

<hr/>



<h4 class="faq-area"> How to find and correct tool errors related to Metadata? <a href="/training-material/faqs/galaxy/dataset_tool_errors_metadata.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Finding and Correcting Metadata
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-how-to-find-and-correct-tool-errors-related-to-metadata"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-how-to-find-and-correct-tool-errors-related-to-metadata" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: How to find and correct tool errors related to Metadata?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Tools can error when the wrong dataset attributes (metadata) are assigned. Some of these wrong assignments may be:</p>   <ul>   <li>Tool outputs, which are automatically assigned without user action.</li>   <li>Incorrect autodetection of datatypes, which need manual modification.</li>   <li>Undetected attributes, which require user action (example: assigning database to newly uploaded data).</li> </ul>   <p>How to notice missing Dataset Metadata:</p>   <ul>   <li>Dataset will not be downloaded when using the disk icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden">galaxy-save</span>.</li>   <li>Tools error when using a previously successfully used specific dataset.</li>   <li>Tools error with a message that ends with: <code class="language-plaintext highlighter-rouge">OSError: [Errno 2] No such file or directory</code>.</li> </ul>   <p>Solution:</p>   <p>Click on the dataset’s pencil icon <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> to reach the <em>Edit Attributes</em> forms and do one of the following as applies:</p>   <ul>   <li><strong>Directly reset metadata</strong>     <ul>       <li>Find the tab for the metadata you want to change, make the change, and save.</li>     </ul>   </li>   <li><strong>Autodetect metadata</strong>     <ul>       <li>Click on the <em>Auto-detect button</em>. The dataset will turn yellow in the history while the job is processing.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Incomplete Dataset Download <a href="/training-material/faqs/galaxy/dataset_incomplete_download.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-incomplete-dataset-download"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-incomplete-dataset-download" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Incomplete Dataset Download</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>In case the dataset downloads incompletely:</p>   <ul>   <li>Use the <em>Google Chrome</em> web browser. Sometimes Chrome works better at supporting continuous data transfers.</li>   <li>Use the <a href="/training-material/faqs/galaxy/datasets_download_using_command_line.html">command-line option</a> instead. The data may really be too large to download OR your connection is slower. This can also be a faster way to download multiple datasets plus ensure a complete transfer (small or large data).</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding 'canceled by admin' or cluster failure error messages <a href="/training-material/faqs/galaxy/troubleshooting_cluster_failure.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-canceled-by-admin-or-cluster-failure-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-canceled-by-admin-or-cluster-failure-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding 'canceled by admin' or cluster failure error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The initial error message could be:</p>   <pre><br>This job failed because it was cancelled by an administrator.<br>Please click the bug icon to report this problem if you need help.<br></pre>   <p>Or</p>   <pre><br>job info:<br>Remote job server indicated a problem running or monitoring this job.<br></pre>   <ul>   <li>Causes:     <ul>       <li>Server or cluster error.</li>       <li>Less frequently, input problems are a factor.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun. Server/cluster errors like this are usually transient.</li>       <li>Review the Solutions section of the <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html">Understanding input error messages</a> FAQ.</li>       <li>If after any corrections, the job still fails, please report the technical issue <a href="/training-material/faqs/galaxy/analysis_reporting_issues.html">following the extended issue guidelines</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding 'exceeds memory allocation' error messages <a href="/training-material/faqs/galaxy/troubleshooting_excess_memory.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-exceeds-memory-allocation-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-exceeds-memory-allocation-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding 'exceeds memory allocation' error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The error message to be displayed are as follows:</p>   <pre><br>job info:<br>This job was terminated because it used more memory than it was allocated.<br>Please click the bug icon to report this problem if you need help.<br></pre>   <p>Or</p>   <pre><br>stderr:<br>Fatal error: Exit code 1 ()<br>slurmstepd: error: Detected 1 oom-kill event(s) in step XXXXXXX.batch cgroup.<br></pre>   <p>Sometimes this message may appear at the bottom</p>   <pre><br>job stderr:<br>slurmstepd: error: Detected 1 oom-kill event(s) in step XXXXXXX.batch cgroup.<br></pre>   <p>In rare cases when the memory quota is exceeded very quickly, an error message such as the following can appear</p>   <pre><br>job stderr:<br>Fatal error: Exit code 1 ()<br>Traceback (most recent call last):<br>(other lines)<br>Memory Error<br></pre>   <p><strong>Note:</strong> Job runtime memory is different from the amount of free storage space (quota) in an account.</p>   <ul>   <li>Causes:     <ul>       <li>The job ran out of memory while executing on the cluster node that ran the job.</li>       <li>The most common reasons for this error are input and tool parameters problems that must be adjusted/corrected.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun to execute the job on a different cluster node.</li>       <li>Review the Solutions section of the <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html">Understanding input error messages</a> FAQ.</li>       <li>Your data may actually be too large to process at a public Galaxy server. Alternatives include setting up a <a href="/training-material/faqs/gtn/galaxy_usage.html">private Galaxy server</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding ValueError error messages <a href="/training-material/faqs/galaxy/troubleshooting_value_error.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-valueerror-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-valueerror-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding ValueError error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The full error is usually a longer message seen only after clicking on the bug icon or by reviewing the job details <code class="language-plaintext highlighter-rouge">stderr</code>.</p>   <p>How to do both is covered in the Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">FAQ</a>.</p>   <pre><br>stderr<br>...<br>Many lines of text, may include parameters<br>...<br>...<br>ValueError: invalid literal for int() with base 10: some-sequence-read-name<br></pre>   <ul>   <li>Causes:     <ul>       <li>MACS2 produces this error the first time it is run. MACS is not the only tool that can produce this issue, but it is the most common.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun.</li>       <li>MACS/2 is not capable of interpreting sequence read names with spaces included. Try following these two:         <ul>           <li>Remove unmapped reads from the SAM dataset. There are several filtering tools in the groups <strong>SAMTools</strong> and <strong>Picard</strong> that can do this.</li>           <li>Convert the SAM input to BAM format with the tool <strong>SAMtools: SAM-to-BAM</strong>. When compressed input is given to MACS, the spaces are no longer an issue.</li>         </ul>       </li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding input error messages <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-input-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-input-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding input error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Input problems are very common across any analysis that makes use of programmed tools.</p>   <ul>   <li>Causes:     <ul>       <li>No quality assurance or content/formatting checks were run on the first datasets of an analysis workflow.</li>       <li>Incomplete dataset Upload.</li>       <li>Incorrect or unassigned <a href="/training-material/faqs/galaxy/index.html#datasets">datatype</a> or <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">database</a>.</li>       <li>Tool-specific formatting requirements for inputs were not met.</li>       <li>Parameters set on a tool form are a mismatch for the input data content or format.</li>       <li>Inputs were in an error state (red) or were putatively successful (green) but are empty.</li>       <li>Inputs do not meet the datatype specification.</li>       <li>Inputs do not contain the exact content that a tool is expecting or that was input in the form.</li>       <li>Annotation files are a mismatch for the selected or assigned reference genome build.</li>       <li><strong>Special case:</strong> Some of the data were generated outside of Galaxy, but later a built-in indexed genome build was assigned in Galaxy for use with downstream tools. This scenario can work, but only if those two reference genomes are an exact match.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Review our <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting Tips</a> for what and where to check.</li>       <li>Review the <a href="https://training.galaxyproject.org/">GTN</a> for related tutorials on tools/analysis plus FAQs.</li>       <li>Review <a href="https://help.galaxyproject.org/">Galaxy Help</a> for prior discussion with extended solutions.</li>       <li>Review <a href="/training-material/faqs/galaxy/datatypes_understanding_datatypes.html">datatype FAQs</a>.</li>       <li>Review the tool form.         <ul>           <li>Input selection areas include usage help.</li>           <li>The help section at the bottom of a tool form often has examples. Does your own data match the format/content?</li>           <li>See the links to publications and related resources.</li>         </ul>       </li>       <li>Review the inputs.         <ul>           <li>All inputs must be in a success state (green) and actually contain content.</li>           <li>Did you <a href="/training-material/faqs/galaxy/datasets_change_datatype.html">directly assign the datatype</a> or <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html">convert the datatype</a>? What results when the <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html">datatype is detected</a> by Galaxy? If these differ, there is likely a content problem.</li>           <li>For most analysis, allowing Galaxy to detect the datatype during Upload is best and adjusting a datatype later should rarely be needed. If a datatype is modified, the change has a specific purpose/reason.</li>           <li>Does your data have headers? Is that in specification for the datatype? Does the tool form have an option to specify if the input has headers or not? Do you need to remove headers first for the correct datatype to be detected? Example <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">GTF</a>.</li>           <li>Large inputs? Consider modifying your inputs to be smaller. Examples: <a href="/training-material/faqs/galaxy/datasets_working_with_fastq.html">FASTQ</a> and <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">FASTA</a>.</li>         </ul>       </li>       <li>Run quality checks on your data.         <ul>           <li>Search <a href="https://training.galaxyproject.org/">GTN</a> tutorials with the keyword “qa-qc” for examples.</li>           <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “qa-qc” and your datatype(s) for more help.</li>         </ul>       </li>       <li>Reference annotation tips.         <ul>           <li>In most cases, <a href="/training-material/faqs/galaxy/datasets_working_with_reference_annotation.html">GTF is preferred over GFF3</a>.</li>           <li>Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> with the keywords “gtf” and “gff3” for more help.</li>         </ul>       </li>       <li>Input mismatch tips.         <ul>           <li>Do the chromosome/sequence identifiers exactly match between all inputs? Search <a href="https://help.galaxyproject.org/">Galaxy Help</a> for more help about how to correct build/version identifier mismatches between inputs.</li>           <li>“Chr1” and “chr1” and “1” do not mean the same thing to a tool.</li>         </ul>       </li>       <li>Custom genome transcriptome exome tips. See <a href="/training-material/faqs/galaxy/datasets_working_with_fasta.html">FASTA</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Understanding walltime error messages <a href="/training-material/faqs/galaxy/troubleshooting_walltime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-understanding-walltime-error-messages"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-understanding-walltime-error-messages" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Understanding walltime error messages</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>The full error message will be reported as below, and can be found by clicking on the bug icon for a failed job run (red dataset):</p>   <pre><br>job info:<br>This job was terminated because it ran longer than the maximum allowed job run time.<br>Please click the bug icon to report this problem if you need help.<br></pre>   <p>Or sometimes,</p>   <pre><br>job stderr:<br>slurmstepd: error: *** JOB XXXX ON XXXX CANCELLED AT 2019-XX-XXTXX:XX:XX DUE TO TIME LIMIT ***<br><br>job info:<br>Remote job server indicated a problem running or monitoring this job.<br></pre>   <ul>   <li>Causes:     <ul>       <li>The job execution time exceeded the “wall-time” on the cluster node that ran the job.</li>       <li>The server may be undergoing maintenance.</li>       <li>Very often input problems also cause this same error.</li>     </ul>   </li>   <li>Solutions:     <ul>       <li>Try at least one rerun.</li>       <li>Check the server homepage for banners or notices. Selected servers also post to the <a href="https://status.galaxyproject.org/">Galaxy status page</a>.</li>       <li>Review the Solutions section of the <a href="/training-material/faqs/galaxy/troubleshooting_input_problem.html">Understanding input error messages</a> FAQ.</li>       <li>Your data may actually be too large to process at a public Galaxy server. Alternatives include setting up a <a href="/training-material/faqs/gtn/galaxy_usage.html">private Galaxy server</a>.</li>     </ul>   </li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-what-information-should-i-include-when-reporting-a-problem"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-what-information-should-i-include-when-reporting-a-problem" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: What information should I include when reporting a problem?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>   <p><strong>What to include</strong></p>   <ol>   <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>   <li>Which server(s) did you run those commands on?</li>   <li>What account/username did you use?</li>   <li>Where did it go wrong?</li>   <li>What were the stdout/stderr of the tool that failed? Include the text.</li>   <li>Did you try any workarounds? What results did those produce?</li>   <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>   <li>If there are job IDs, please include them as text so administrators don’t have to manually transcribe the job ID in your picture.</li> </ol>   <p>It makes the process of answering ‘bug reports’ much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>   <p><strong>What does a GOOD bug report look like?</strong></p>   <p>The people who provide support for Galaxy are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>   <blockquote class="quote">   <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My username is jane-doe.</p>   <p>Here is everything that I know:</p>   <ul>     <li>The dataset is green, the job did not fail</li>     <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>     <li>I have read it but I do not understand what X/Y means.</li>     <li>The job ID from the output information page is 123123abdef.</li>     <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>   </ul>   <p>Could you help me?</p> </blockquote> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="user preferences"> <a href="#user preferences"> User preferences </a></h3>

<hr/>



<h4 class="faq-area"> Does your account usage quota seem incorrect? <a href="/training-material/faqs/galaxy/user_preferences_account_usage_quota.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-does-your-account-usage-quota-seem-incorrect"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-does-your-account-usage-quota-seem-incorrect" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Does your account usage quota seem incorrect?</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Log out of Galaxy, then back in again. This refreshes the disk usage calculation displayed in the Masthead usage (summary) and under <em>User &gt; Preferences</em> (exact).</li> </ol>   <p>Note:</p>   <ul>   <li>Your account usage quota can be found at the bottom of your user preferences page.</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Forgot Password <a href="/training-material/faqs/galaxy/user_preferences_forgot_password.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-forgot-password"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-forgot-password" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Forgot Password</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Go to the Galaxy server you are using.</li>   <li>Click on <strong>Login or Register</strong>.</li>   <li>Enter your email on the Public Name or Email Address entry box.</li>   <li>Click on the link under the password entry box titled <strong>Forgot password? Click here to reset your password.</strong></li>   <li>An email will be sent with a password reset link. This email may be in your email Spam or Trash folders, depending on your filters.</li>   <li>Click on the reset link in the email or copy and paste it into a web browser window.</li>   <li>Enter your new password and click on <strong>Save new password</strong>.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Getting your API key <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-getting-your-api-key"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-getting-your-api-key" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Getting your API key</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>In your browser, open your Galaxy homepage</li>   <li>Log in, or register a new account, if it’s the first time you’re logging in</li>   <li>Go to <code class="language-plaintext highlighter-rouge">User -&gt; Preferences</code> in the top menu bar, then click on <code class="language-plaintext highlighter-rouge">Manage API key</code></li>   <li>If there is no current API key available, click on <code class="language-plaintext highlighter-rouge">Create a new key</code> to generate it</li>   <li>Copy your API key to somewhere convenient, you will need it throughout this tutorial</li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="visualisation"> <a href="#visualisation"> Visualisation </a></h3>

<hr/>



<h4 class="faq-area"> Open History files in Integrated Genome Browser (IGB) <a href="/training-material/faqs/galaxy/visualisations_igb.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-open-history-files-in-integrated-genome-browser-igb"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-open-history-files-in-integrated-genome-browser-igb" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Open History files in Integrated Genome Browser (IGB)</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can open some file types in Integrated Genome Browser (IGB), a desktop genome browser. (<a href="https://wiki.bioviz.org/confluence/display/igbman/File+Formats">Supported File Types</a>)</p>   <p>Here’s how:</p>   <ol>   <li>Install IGB on your computer (<a href="https://bioviz.org/download.html">download page</a>).</li>   <li>Start IGB.</li>   <li>In Galaxy, click the desired dataset’s name to expand it.     <ul>       <li>Check that the reference genome (dbkey) is set (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>).</li>     </ul>   </li>   <li>Click on the Charts icon <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span></li>   <li>In the central panel, next to <code class="language-plaintext highlighter-rouge">display in IGB</code>, choose <code class="language-plaintext highlighter-rouge">View</code>.</li> </ol>   <p>When you choose “View” in Galaxy, your browser opens a new tab showing a page from BioViz.org. Check the newly opened page for next steps.</p>   <p>Having trouble? Working with a custom genome assembly not yet available in Galaxy or IGB?</p>   <p>Contact <a href="https://bioviz.org/help.html">the IGB team</a> for help and advice!</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Using IGV with Galaxy <a href="/training-material/faqs/galaxy/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-using-igv-with-galaxy"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-using-igv-with-galaxy" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Using IGV with Galaxy</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>You can send data from your Galaxy history to IGV for viewing as follows:</p>   <ol>   <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>   <li>Start IGV</li>   <li>In recent versions of IGV, you will have to enable the port:     <ul>       <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>       <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>     </ul>   </li>   <li>In Galaxy, expand the dataset you would like to view in IGV     <ul>       <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>       <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->



 <br> 

<h3 id="workflows"> <a href="#workflows"> Workflows </a></h3>

<hr/>



<h4 class="faq-area"> Annotate a workflow <a href="/training-material/faqs/galaxy/workflows_annotate.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-annotate-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-annotate-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Annotate a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Open the workflow editor for the workflow</li>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>Edit Attributes</strong> on the top right</li>   <li>Write a description of the workflow in the <strong>Annotation</strong> box</li>   <li>Add a tag (which will help to search for the workflow) in the <strong>Tags</strong> section</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Creating a new workflow <a href="/training-material/faqs/galaxy/workflows_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can create a Galaxy workflow from scratch in the Galaxy workflow editor.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click <strong>Workflow</strong> on the top bar</li>   <li>Click the new workflow <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-new</span> button</li>   <li>Give it a clear and memorable name</li>   <li>Clicking <strong>Save</strong> will take you directly into the workflow editor for that workflow</li>   <li>Need more help? Please see the <em>How to make a workflow subsection</em> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">here</a></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Ensuring Workflows meet Best Practices <a href="/training-material/faqs/galaxy/workflows_best_practices.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-ensuring-workflows-meet-best-practices"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-ensuring-workflows-meet-best-practices" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Ensuring Workflows meet Best Practices</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When you are editing a workflow, there are a number of additional steps you can take to ensure that it is a Best Practice workflow and will be more reusable.</p>   <ol>   <li>Open a workflow for editing</li>   <li>In the workflow menu bar, you’ll find the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-options</span> <strong>Workflow Options</strong> dropdown menu.</li>   <li>     <p>Click on it and select <i class="fas fa-wand-magic-sparkles" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-best-practices</span> <strong>Best Practices</strong> from the dropdown menu.</p>     <p><img src="/training-material/faqs/galaxy/images/best-practices1.png" alt="screenshot showing the best practices menu item in the gear dropdown." /></p>   </li>   <li>     <p>This will take you to a new side panel, which allows you to investigate and correct any issues with your workflow.</p>     <p><img src="/training-material/faqs/galaxy/images/best-practices2.png" alt="screenshot showing the best practices side panel. several issues are raised like a missing annotation with a link to add that, and non-optional inputs that are unconnected. Additionally several items already have green checks like the workflow defining creator information and a license." /></p>   </li> </ol>   <p>The Galaxy community also has a <a href="https://planemo.readthedocs.io/en/latest/best_practices_workflows.html">guide on best practices for maintaining workflows</a>. This guide includes the best practices from the Galaxy workflow panel, plus:</p>   <ul>   <li>adding tests to the workflow</li>   <li>publishing the workflow on GitHub, a public GitLab server, or another public version-controlled repository</li>   <li>registering the workflow with a workflow registry such as WorkflowHub or Dockstore</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Extracting a workflow from your history <a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Galaxy can automatically create a workflow based on the analysis you have performed in a history. This means that once you have done an analysis manually once, you can easily extract a workflow to repeat it on different data.
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-extracting-a-workflow-from-your-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-extracting-a-workflow-from-your-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Extracting a workflow from your history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>     <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-trash" aria-hidden="true"></i><span class="visually-hidden">galaxy-delete</span> button.</p>     <p>This will make the creation of the workflow easier.</p>   </li>   <li>     <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract workflow</strong>.</p>     <p><img src="/training-material/topics/introduction/images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>     <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the workflow.</p>   </li>   <li>     <p>Replace the <strong>Workflow name</strong> to something more descriptive.</p>   </li>   <li>     <p><strong>Rename</strong> each workflow input in the boxes at the top of the second column.</p>   </li>   <li>     <p>If there are any steps that shouldn’t be included in the workflow, you can <strong>uncheck</strong> them in the first column of boxes.</p>   </li>   <li>     <p>Click on the <strong>Create Workflow</strong> button near the top.</p>     <p>You will get a message that the workflow was created.</p>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Hiding intermediate steps <a href="/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-hiding-intermediate-steps"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-hiding-intermediate-steps" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Hiding intermediate steps</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>When a workflow is executed, the user is usually primarily interested in the final product and not in all intermediate steps. By default all the outputs of a workflow will be shown, but we can explicitly tell Galaxy which outputs to show and which to hide for a given workflow. This behaviour is controlled by the little checkbox in front of every output dataset:</p>   <p><img src="/training-material/shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Import workflows from DockStore <a href="/training-material/faqs/galaxy/workflows_import_from_dockstore.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-import-workflows-from-dockstore"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-import-workflows-from-dockstore" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Import workflows from DockStore</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://dockstore.org/">Dockstore</a> is a free and open source platform for sharing reusable and scalable analytical tools and workflows.</p>   <ol>   <li>Ensure that you are logged in to your Galaxy account.</li>   <li>Go to <a href="https://dockstore.org">DockStore</a>.</li>   <li>Select any Galaxy workflow you want to import.</li>   <li>Click on “Galaxy” dropdown within the “Launch with” panel located in the upper right corner.</li>   <li>Select a galaxy instance you want to launch this workflow with.</li>   <li>You will be redirected to Galaxy and presented with a list of workflow versions.</li>   <li>Click the version you want (usually the latest labelled as “main”)</li>   <li>You are done!</li> </ol>   <p>The following short video walks you through this uncomplicated procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/K2wFrSLFpa0" title="Importing from Dockstore" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing from Dockstore</figcaption> </figure> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Import workflows from WorkflowHub <a href="/training-material/faqs/galaxy/workflows_import_from_workflowhub.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-import-workflows-from-workflowhub"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-import-workflows-from-workflowhub" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Import workflows from WorkflowHub</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p><a href="https://workflowhub.eu/">WorkflowHub</a> is a workflow management system which allows workflows to be FAIR (Findable, Accessible, Interoperable, and Reusable), citable, have managed metadata profiles, and be openly available for review and analytics.</p>   <ol>   <li>Ensure that you are logged in to your Galaxy account.</li>   <li>Click on the <strong>Workflow</strong> menu, located in the top bar.</li>   <li>Click on the <strong>Import</strong> button, located in the right corner.</li>   <li>In the section “Import a Workflow from Configured GA4GH Tool Registry Servers (e.g. Dockstore)”, click on <strong>Search form</strong>.</li>   <li>In the <strong>TRS Server: <em>workflowhub.eu</em></strong> menu you should type your query. <img src="/training-material/topics/assembly/images/vgp_assembly/workflow_list.png" alt="galaxy TRS workflow search field, name:vgp is entered in the search bar, and five different workflows all labelled VGP are listed" /></li>   <li>Click on the desired workflow, and finally select the latest available version.</li> </ol>   <p>After that, the imported workflows will appear in the main workflow menu. In order to run the workflow, just need to click in the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> <strong>Run workflow</strong> icon.</p>   <p>Below is a short video showing this uncomplicated procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing from WorkflowHub" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing from WorkflowHub</figcaption> </figure> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing a workflow <a href="/training-material/faqs/galaxy/workflows_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing a workflow using the search <a href="/training-material/faqs/galaxy/workflows_import_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-a-workflow-using-the-search"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-a-workflow-using-the-search" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing a workflow using the search</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <em>Workflow</em> in the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> icon at the top-right of the screen</li>   <li>On the new page, select the <strong>GA4GH servers</strong> tab, and configure the <strong>GA4GH Tool Registry Server (TRS) Workflow Search</strong> interface as follows:     <ol>       <li><em>“TRS Server”</em>: the TRS Server you want to search on (Dockstore or Workflowhub)</li>       <li>Type in the <em>search query</em></li>       <li>Expand the correct workflow by clicking on it</li>       <li>Select the version you would like to <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> import</li>     </ol>   </li> </ol>   <p>The workflow will be imported to your list of workflows. Note that it will also carry a little green check mark next to its name, which indicates that this is an original workflow version imported from a TRS server. If you ever modify the workflow with Galaxy’s workflow editor, it will lose this indicator.</p> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing and Launching a Dockstore Workflow <a href="/training-material/faqs/galaxy/workflows_run_ds.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-dockstore-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and Launching a Dockstore Workflow</div>   <!-- GTN:IGNORE:011 _blank is used here due to iframeing sites that (can) set x-frame-options. The contexts in which _blank is set are only visible when iframe'd -->   <ol>   <li>Go to <a href="https://my.galaxy.training/?path=/workflows/import">Workflow → Import</a> in your Galaxy</li>   <li>Switch tabs to TRS ID</li>   <li>Ensure the TRS server is set to “dockstore.org”</li>   <li>Provide your workflow hub ID</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing and Launching a WorkflowHub.eu Workflow <a href="/training-material/faqs/galaxy/workflows_run_wfh.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-workflowhub-eu-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and Launching a WorkflowHub.eu Workflow</div>   <!-- GTN:IGNORE:011 _blank is used here due to iframeing sites that (can) set x-frame-options. The contexts in which _blank is set are only visible when iframe'd -->   <ol>   <li>Go to <a href="https://my.galaxy.training/?path=/workflows/import">Workflow → Import</a> in your Galaxy</li>   <li>Switch tabs to TRS ID</li>   <li>Ensure the TRS server is set to “workflowhub.eu”</li>   <li>Provide your workflow hub ID</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Importing and launching a GTN workflow <a href="/training-material/faqs/galaxy/workflows_run_trs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-and-launching-a-gtn-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing and launching a GTN workflow</div>   <ol>   <li>Find the material you are interested in</li>   <li>View it’s workflows, which can be found in the metadata box at the top of the tutorial</li>   <li>Click the button on any workflow to run it.</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Make a workflow public <a href="/training-material/faqs/galaxy/workflows_publish.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-make-a-workflow-public"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-make-a-workflow-public" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Make a workflow public</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <strong>Workflow</strong> on the top menu bar of Galaxy. You will see a list of all your workflows</li>   <li>Click on the interesting workflow</li>   <li>Click on <strong>Share</strong></li>   <li>Click on <strong>Make Workflow accessible</strong>. This makes the workflow <em>publicly accessible</em> but <em>unlisted</em>.</li>   <li>To also list the workflow in the <em>Shared Data</em> section (in the top menu bar) of Galaxy, click <strong>Make Workflow publicly available in Published Workflows</strong></li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Opening the workflow editor <a href="/training-material/faqs/galaxy/workflows_edit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-opening-the-workflow-editor"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-opening-the-workflow-editor" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Opening the workflow editor</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>In the top menu bar, click on <strong>Workflows</strong></li>   <li>Click on the name of the workflow you want to edit <img src="/training-material/topics/galaxy-interface/images/edit_workflow.png" alt="Workflow drop down menu showing Edit option" title="When you click on the name of a workflow it will open a drop down menu whose first option is Edit. Select the Edit option to edit the workflow." /></li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Edit</strong> from the dropdown menu to open the workflow in the workflow editor</li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Renaming workflow outputs <a href="/training-material/faqs/galaxy/workflows_rename_output.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-workflow-outputs"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-workflow-outputs" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming workflow outputs</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the workflow editor</li>   <li>Click on the tool in the workflow to get the details of the tool on the right-hand side of the screen.</li>   <li>Scroll down to the <strong>Configure Output</strong> section of your desired parameter, and click it to expand it.     <ul>       <li>         <p>Under <strong>Rename dataset</strong>, give it a meaningful name</p>         <p><img src="/training-material/shared/images/workflow_rename_datasets.png" alt="Rename output datasets" /></p>       </li>     </ul>   </li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Running a workflow <a href="/training-material/faqs/galaxy/workflows_run.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-running-a-workflow"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-running-a-workflow" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Running a workflow</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden">workflow-run</span> (<em>Run workflow</em>) button next to your workflow</li>   <li>Configure the workflow as needed</li>   <li>Click the <strong>Run Workflow</strong> button at the top-right of the screen</li>   <li>You may have to refresh your history to see the queued jobs</li> </ul> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Setting parameters at run-time <a href="/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-setting-parameters-at-run-time"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-setting-parameters-at-run-time" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Setting parameters at run-time</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Open the workflow editor</li>   <li>Click on the tool in the workflow to get the details of the tool on the right-hand side of the screen.</li>   <li>Scroll down to the parameter you want users to provide every time they run the workflow</li>   <li>Click on the arrow in front of the name <i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden">workflow-runtime-toggle</span> to toggle to <em>set at runtime</em></li> </ol> </blockquote> <!--END_SNIPPET-->




<h4 class="faq-area"> Viewing a workflow report <a href="/training-material/faqs/galaxy/workflows_report_view.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can find the workflow report from the workflow invocation
<!--SNIPPET--><blockquote class="tip">   <div class="box-title tip-title" id="tip-viewing-a-workflow-report"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-viewing-a-workflow-report" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Viewing a workflow report</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Go to <strong>User</strong> on the top menu bar of Galaxy.</li>   <li>Click on <strong>Workflow invocations</strong>     <ul>       <li>Here you will find a list of all the workflows you have run</li>     </ul>   </li>   <li>Click on the name of a workflow invocation to expand it <img src="/training-material/faqs/galaxy/images/workflow_invocations_menu.png" alt="workflow invocations list" /></li>   <li>Click on <strong>View Report</strong> to go to the workflow report page</li>   <li><strong>Note:</strong> The report can also be downloaded in PDF format by clicking on the <i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-report-download</span> icon.</li> </ul> </blockquote> <!--END_SNIPPET-->











<br><br><hr>
<h5>Still have questions?</h5>

<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://gitter.im/Galaxy-Training-Network/Lobby" class="btn btn-square-medium btn-yellow"><div>Gitter Chat Support</div></a>


<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://help.galaxyproject.org" class="btn btn-square-medium btn-purple"><div>Galaxy Help Forum</div></a>



            </div>
        </div>
    </div>
</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:F00001">gxy.io/GTN:F00001</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/faqs/galaxy/aaaa_dontquestionthislinkitisthegluethatholdstogetherthegalaxy.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/faqs/galaxy/aaaa_dontquestionthislinkitisthegluethatholdstogetherthegalaxy.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/43973919866b066986c8bceb7a497d54a19de378">4397391</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	

    </body>
</html>