<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Galaxy Training!</title>
        
            <script async defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>

        
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml" />

        
        
        
        
        
        <meta name="description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:title" content="Galaxy Training" />
        <meta property="og:description" content="Collection of tutorials developed and maintained by the w..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/hall-of-fame" title="Hall of Fame">
                           <i class="fas fa-users" aria-hidden="true"></i><span class="visually-hidden">hall-of-fame</span> Contributors
                        </a>

                        
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Languages">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Languages
                        </a>
                        <div class="dropdown-menu dropdown-menu-right">
                            
                              <strong class="dropdown-header">Automatic Translations</strong>
            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org/faqs/galaxy/&edit-text=&act=url">
                            Fran√ßais
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org/faqs/galaxy/&edit-text=&act=url">
                            Êó•Êú¨Ë™û
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org/faqs/galaxy/&edit-text=&act=url">
                            Espa√±ol
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org/faqs/galaxy/&edit-text=&act=url">
                            Portugu√™s
                            </a>
                            
            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org/faqs/galaxy/&edit-text=&act=url">
                            ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org/faqs/galaxy/&edit-text=&act=url" title="">
                            And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <!-- disable Tess for now
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>
        -->

        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Gitter
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Extras">
        <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">galaxy-star</span> Extras
    </a>
    <div class="dropdown-menu dropdown-menu-right">

        
        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/faqs/galaxy/" title="Edit on GitHub">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
        </a>

        <a class="dropdown-item" href="/training-material/stats.html" title="Show view statistics about this repository">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/faqs/galaxy/" title="Show view statistics of this page">
            <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page Metrics
        </a>

        
            <a class="dropdown-item" href="/training-material/feedback.html" title="Show feedback statistics about this repository">
                <i class="fas fa-chart-bar" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="progress" class="btn btn-secondary">
                    <input type="radio" name="options" id="progress" autocomplete="off">üè≥Ô∏è‚Äçüåà
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
                <label data-value="straya" class="btn btn-secondary">
                    <input type="radio" name="options" id="downunder" autocomplete="off"> üá¶üá∫
                </label>
            </div>

        </div>

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            
            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-04-01/faqs/galaxy/" title="Version 2022-04-01">2022-04-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-03-01/faqs/galaxy/" title="Version 2022-03-01">2022-03-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/2022-02-01/faqs/galaxy/" title="Version 2022-02-01">2022-02-01</a>
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/" title="Older Versions">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        <div class="container main-content" role="main">
        



<section class="faqs">

   <nav aria-label="breadcrumb">
    <ol class="breadcrumb">

     <li class="breadcrumb-item "><a href="/training-material/faqs">All FAQs</a></li>
     
     <li class="breadcrumb-item active">Galaxy FAQs</li>
     
     

     
     </ol>
    </nav>

    <h1 data-toc-skip>Frequently Asked Questions</h1>
    
<!--<div class="contributors-line">Authors: 
    </div>-->
    <div class="container">
        <div class="row">

            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2">
                <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
            </div>
            <div class="col-sm-10">
                













<h3 id="analysis"> <a href="#analysis"> Analysis </a></h3>

<hr/>



<h4 class="faq-area"> Adding a custom database/build (dbkey) <a href="/training-material/faqs/galaxy/analysis_add_custom_build.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> may have several reference genomes built-in, but you can also create your own.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-custom-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a custom database/build (dbkey)</h3>  <ul>  <li>In the top menu bar, go to the <em>User</em>, and select <em>Custom Builds</em></li>  <li>Choose a name for your reference build</li>  <li>Choose a dbkey for your reference build</li>  <li>Under <strong>Definition</strong>, select the option <code class="language-plaintext highlighter-rouge">FASTA-file from history</code></li>  <li>Under <strong>FASTA-file</strong>, select your fasta file</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Will my jobs keep running? <a href="/training-material/faqs/galaxy/analysis_jobs_keep_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-will-my-jobs-keep-running"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Will my jobs keep running?</h3>  <p><span class="notranslate">Galaxy</span> is a fantastic system, but some users find themselves wondering:</p>  <blockquote>  <p>Will my jobs keep running once I‚Äôve closed the tab? Do I need to keep my browser open?</p></blockquote>  <p>No, you don‚Äôt! You can safely:</p>  <ol>  <li>Start jobs</li>  <li>Shut down your computer</li></ol>  <p>and your jobs will keep running in the background! Whenever you next visit <span class="notranslate">Galaxy</span>, you can check if your jobs are still running or completed.</p>  <p><strong>However</strong>, this is not true for uploading data from your computer. You must wait for uploading a dataset <em>from your computer</em> to finish. (Uploading via URL is not affected by this, if you‚Äôre uploading from URL you can close your computer.)</p></blockquote>




<h4 class="faq-area"> My jobs aren't running! <a href="/training-material/faqs/galaxy/analysis_jobs_not_running.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-my-jobs-arent-running"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: My jobs aren‚Äôt running!</h3>  <ol>  <li>    <p><strong>Please make sure you are logged in</strong>. At the top menu bar, you should see a section labeled ‚ÄúUser‚Äù. If you see ‚ÄúLogin/Register‚Äù here you are not logged in.</p>  </li>  <li><strong>Activate your account</strong>. If you have recently registered your account, you may first have to activate it. You will receive an e-mail with an activation link.    <ul>      <li>Make sure to check your spam folder!</li>    </ul>  </li>  <li>    <p><strong>Be patient</strong>. <span class="notranslate">Galaxy</span> is a free service, when a lot of people are using it, you may have to wait longer than usual (especially for ‚Äòbig‚Äô jobs, e.g. alignments).</p>  </li>  <li><strong>Contact Support</strong>. If you really think something is wrong with the server, you can <a href="/training-material/faqs/galaxy/support.html">ask for support</a></li></ol></blockquote>




<h4 class="faq-area"> Reporting usage problems, security issues, and bugs <a href="/training-material/faqs/galaxy/analysis_reporting_issues.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-reporting-usage-problems-security-issues-and-bugs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Reporting usage problems, security issues, and bugs</h3>  <ul>  <li>For reporting <strong>Usage Problems</strong>, related to tools and functions, head to the <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> site.    <ul>      <li><strong>Red Error Datasets</strong>:        <ul>          <li>Refer to the <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_troubleshooting.html">Troubleshooting errors</a> FAQ for red error in datasets.</li>        </ul>      </li>      <li><strong>Unexpected results in Green Success Dataset</strong>:        <ul>          <li>To resolve it you may be asked to send in a shared history link and possibly a shared <span class="notranslate">workflow</span> link. For sharing your history, refer to this <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/histories_sharing.html">link</a>.</li>          <li>To reach our support team, visit <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/#support">Support FAQs</a>.</li>        </ul>      </li>      <li><strong>Functionality problems</strong>:        <ul>          <li>Using <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help</a> is the best way to get help in most cases.</li>          <li>If the problem is more complex, email a description of the problem and how to reproduce it.</li>        </ul>      </li>      <li><strong>Administrative problems</strong>:        <ul>          <li>If the problem is present in your own <span class="notranslate">Galaxy</span>, the administrative configuration may be a factor.</li>          <li>For the fastest help directly from the development community, admin issues can be alte<span class="notranslate">rna</span>tively reported to the mailing list or the <span class="notranslate">Galaxy</span>Project <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a>.</li>        </ul>      </li>    </ul>  </li>  <li>For <strong>Security Issues</strong>, do not report them via GitHub. Kindly disclose these as explained in this <a href="https://github.com/galaxyproject/galaxy/blob/dev/SECURITY_POLICY.md">document</a>.</li>  <li>For <strong>Bug Reporting</strong>, create a Github issue. Include the steps mentioned <a href="https://training.galaxyproject.org/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">here</a>.</li>  <li>Search the <a href="https://training.galaxyproject.org/search">GTN Search</a> to find prior Q &amp; A, FAQs, tutorials, and other documentation across all <span class="notranslate">Galaxy</span> resources, to verify in case your issue was already faced by someone.</li></ul></blockquote>




<h4 class="faq-area"> Results may vary <a href="/training-material/faqs/galaxy/analysis_results_may_vary.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-results-may-vary"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Results may vary</h3>  <p>Your results may be slightly different from the ones presented in this tutorial due to differing versions of tools, reference data, exte<span class="notranslate">rna</span>l databases, or because of stochastic processes in the algorithms.</p></blockquote>




<h4 class="faq-area"> Troubleshooting errors <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html"><i class="fas fa-external-link-alt"></i></a> </h4>
When you get a red dataset in your history, it means something went wrong. But how can you find out what it was? And how can you report errors?
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-troubleshooting-errors"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Troubleshooting errors</h3>  <p>When someting goes wrong in <span class="notranslate">Galaxy</span>, there are a number of things you can do to find out what it was. Error messages can help you figure out whether it was a problem with one of the settings of the tool, or with the input data, or maybe there is a bug in the tool itself and the problem should be reported. Below are the steps you can follow to troubleshoot your <span class="notranslate">Galaxy</span> errors.</p>  <ol>  <li><strong>Expand</strong> the red history dataset by clicking on it.    <ul>      <li>Sometime you can already see an error message here</li>    </ul>  </li>  <li>    <p><strong>View the error message</strong> by clicking on the <strong>bug icon</strong> <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span></p>  </li>  <li><strong>Check the logs.</strong> Output (stdout) and error logs (stderr) of the tool are available:    <ul>      <li>Expand the history item</li>      <li>Click on the <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> icon</li>      <li>Scroll down to the <strong>Job Information</strong> section to view the 2 logs:        <ul>          <li>Tool Standard Output</li>          <li>Tool Standard Error</li>        </ul>      </li>    </ul>  </li>  <li><strong>Submit a bug report!</strong> If you are still unsure what the problem is.    <ul>      <li>Click on the bug icon <i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span></li>      <li>Write down any information you think might help solve the problem        <ul>          <li>See <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html">this FAQ</a> on how to write good bug reports</li>        </ul>      </li>      <li>Click <strong><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span> Report</strong> button</li>    </ul>  </li>  <li><strong>Ask for help!</strong>    <ul>      <li>Where?        <ul>          <li>In the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a></li>          <li>In the <a href="https://gitter.im/galaxyproject/Lobby"><span class="notranslate">Galaxy</span> Gitter Channel</a></li>          <li>Browse the <a href="https://help.galaxyproject.org/"><span class="notranslate">Galaxy</span> Help Forum</a> to see if others have encountered the same problem before (or post your question).</li>        </ul>      </li>      <li>When asking for help, it is useful to <a href="/training-material/faqs/galaxy/histories_sharing.html">share a link to your history</a></li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Creating an account <a href="/training-material/faqs/galaxy/creating_account.html"><i class="fas fa-external-link-alt"></i></a> </h4>
To get access to all features of a <span class="notranslate">Galaxy</span> instance, you need to create an account.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-an-account"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating an account</h3>  <p>To create an account:</p>  <ol>  <li>Click <strong>Login or Register</strong></li>  <li>At the bottom of the form click <strong>Register here</strong></li>  <li>Fill the form and click <strong>Create</strong></li></ol></blockquote>



 <br> 

<h3 id="collections"> <a href="#collections"> Collections </a></h3>

<hr/>



<h4 class="faq-area"> Adding a tag to a collection <a href="/training-material/faqs/galaxy/collections_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-tag-to-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag to a collection</h3>  <ul>  <li>Click on the collection</li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code> in the <code class="language-plaintext highlighter-rouge">Add tags</code> field</p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Press <kbd>Enter</kbd></li>  <li>Check that the tag is appearing below the collection name</li></ul></blockquote>




<h4 class="faq-area"> Creating a dataset collection <a href="/training-material/faqs/galaxy/collections_build_list.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a dataset collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>    <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>  </li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>




<h4 class="faq-area"> Creating a paired collection <a href="/training-material/faqs/galaxy/collections_build_list_paired.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-paired-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a paired collection</h3>  <ul>  <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="/training-material/topics/galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>  <li>Check all the datasets in your history you would like to include</li>  <li>    <p>Click <strong>For all selected..</strong> and choose <strong>Build List of Dataset Pairs</strong></p>  </li>  <li>Change the text of <em>unpaired forward</em> to a common selector for the forward <span class="notranslate">reads</span></li>  <li>Change the text of <em>unpaired reverse</em> to a common selector for the reverse <span class="notranslate">reads</span></li>  <li>Click <strong>Pair these datasets</strong> for each valid <em>forward</em> and <em>reverse</em> pair.</li>  <li>Enter a name for your collection</li>  <li>Click <strong>Create List</strong> to build your collection</li>  <li>Click on the checkmark icon at the top of your history again</li></ul></blockquote>




<h4 class="faq-area"> Renaming a collection <a href="/training-material/faqs/galaxy/collections_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a collection</h3>  <ol>  <li>Click on the collection</li>  <li>Click on the name of the collection at the top</li>  <li>Change the name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>



 <br> 

<h3 id="contributors"> <a href="#contributors"> Contributors </a></h3>

<hr/>



<h4 class="faq-area"> What information should I include when reporting a problem? <a href="/training-material/faqs/galaxy/analysis_troubleshooting_reporting.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-what-information-should-i-include-when-reporting-a-problem"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: What information should I include when reporting a problem?</h3>  <p>Writing bug reports is a good skill to have as bioinformaticians, and a key point is that you should include enough information from the first message to help the process of resolving your issue more efficient and a better experience for everyone.</p>  <p><strong>What to include</strong></p>  <ol>  <li>Which commands did you run, precisely, we want details. Which flags did you set?</li>  <li>Which server(s) did you run those commands on?</li>  <li>What account/use<span class="notranslate">rna</span>me did you use?</li>  <li>Where did it go wrong?</li>  <li>What were the stdout/stderr of the tool that failed? Include the text.</li>  <li>Did you try any workarounds? What results did those produce?</li>  <li>(If relevant) screenshot(s) that show exactly the problem, if it cannot be described in text. Is there a details panel you could include too?</li>  <li>If there are job IDs, please include them as text so administrators don‚Äôt have to manually transcribe the job ID in your picture.</li></ol>  <p>It makes the process of answering ‚Äòbug reports‚Äô much smoother for us, as we will have to ask you these questions anyway. If you provide this information from the start, we can get straight to answering your question!</p>  <p><strong>What does a GOOD bug report look like?</strong></p>  <p>The people who provide support for <span class="notranslate">Galaxy</span> are largely volunteers in this community, so try and provide as much information up front to avoid wasting their time:</p>  <blockquote class="quote">  <p>I encountered an issue: I was working on (this server&gt; and trying to run (tool)+(version number) but all of the output files were empty. My use<span class="notranslate">rna</span>me is jane-doe.</p>  <p>Here is everything that I know:</p>  <ul>    <li>The dataset is green, the job did not fail</li>    <li>This is the standard output/error of the tool that I found in the information page (insert it here)</li>    <li>I have read it but I do not understand what X/Y means.</li>    <li>The job ID from the output information page is 123123abdef.</li>    <li>I tried re-running the job and changing parameter Z but it did not change the result.</li>  </ul>  <p>Could you help me?</p></blockquote></blockquote>



 <br> 

<h3 id="data upload"> <a href="#data upload"> Data upload </a></h3>

<hr/>



<h4 class="faq-area"> Importing data from a data library <a href="/training-material/faqs/galaxy/datasets_import_from_data_library.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing data from a data library</h3>  <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>  <ul>  <li>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></li>  <li>Navigate to the correct folder as indicated by your instructor</li>  <li>Select the desired files</li>  <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>  <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>  <li>Click on <strong>Import</strong></li></ul></blockquote>




<h4 class="faq-area"> Importing via links <a href="/training-material/faqs/galaxy/datasets_import_via_link.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing via links</h3>  <ul>  <li>Copy the link location</li>  <li>    <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>  </li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the link into the text field</p>  </li>  <li>    <p>Press <strong>Start</strong></p>  </li>  <li><strong>Close</strong> the window</li></ul></blockquote>



 <br> 

<h3 id="datasets"> <a href="#datasets"> Datasets </a></h3>

<hr/>



<h4 class="faq-area"> Upload fastqsanger datasets via links <a href="/training-material/faqs/galaxy/dataset_upload_fastqsanger_via_urls.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-fastqsanger-datasets-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload fastqsanger datasets via links</h3>  <ol>  <li>    <p>Click on <strong>Upload Data</strong> on the top of the left panel:</p>    <p><img src="/training-material/faqs/galaxy/images/upload_data_button.png" alt="UploadDataButton" /></p>  </li>  <li>    <p>Click on <strong>Paste/Fetch</strong>:</p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_button.png" alt="PasteFetchButton" /></p>  </li>  <li>    <p>Paste URL into text box that would appear:</p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_data_modal.png" alt="PasteFetchModal" /></p>  </li>  <li>    <p>Set <strong>Type (set all)</strong> to <code class="language-plaintext highlighter-rouge">fastqsanger</code> or, if your data is compressed as in URLs above (they have <code class="language-plaintext highlighter-rouge">.gz</code> extensions), to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>    <p><img src="/training-material/faqs/galaxy/images/paste_fetch_set_data_type.png" alt="ChangeTypeDropDown" />:</p>  </li></ol></blockquote>




<h4 class="faq-area"> Adding a tag <a href="/training-material/faqs/galaxy/datasets_add_tag.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tags can help you to better organize your history and track datasets.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Adding a tag</h3>  <ul>  <li>Click on the dataset</li>  <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>  <li>    <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>    <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>  </li>  <li>Check that the tag is appearing below the dataset name</li></ul></blockquote>




<h4 class="faq-area"> Changing the datatype <a href="/training-material/faqs/galaxy/datasets_change_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> will try to autodetect the datatype of your files, but you may need to manually set this occasionally.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the datatype</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Select  your desired datatype</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Changing database/build (dbkey) <a href="/training-material/faqs/galaxy/datasets_change_dbkey.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can tell <span class="notranslate">Galaxy</span> which dbkey (e.g. reference genome) your dataset is associated with. This may be used by tools to automatically use the correct settings.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-databasebuild-dbkey"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing database/build (dbkey)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Database/Build</strong> field</li>  <li>Select your desired database key from the dropdown list</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Converting the file format <a href="/training-material/faqs/galaxy/datasets_convert_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Some datasets can be transformed into a different format. <span class="notranslate">Galaxy</span> has some built-in file conversion options depending on the type of data you have.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-converting-the-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Converting the file format</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <strong>Convert</strong> tab on the top</li>  <li>Select the appropriate datatype from the list</li>  <li>Click the <strong>Convert datatype</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Creating a new file <a href="/training-material/faqs/galaxy/datasets_create_new_file.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> allows you to create new files from the upload menu. You can supply the contents of the file.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-file"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new file</h3>  <ul>  <li>Open the <span class="notranslate">Galaxy</span> Upload Manager</li>  <li>Select <strong>Paste/Fetch Data</strong></li>  <li>    <p>Paste the file contents into the text field</p>  </li>  <li>Press <strong>Start</strong> and <strong>Close</strong> the window</li></ul></blockquote>




<h4 class="faq-area"> Detecting the datatype (file format) <a href="/training-material/faqs/galaxy/datasets_detect_datatype.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-detecting-the-datatype-file-format"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Detecting the datatype (file format)</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>  <li>Click the <strong>Detect datatype</strong> button to have <span class="notranslate">Galaxy</span> try to autodetect it.</li></ul></blockquote>




<h4 class="faq-area"> Downloading datasets <a href="/training-material/faqs/galaxy/datasets_download_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-downloading-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Downloading datasets</h3>  <ol>  <li>Click on the dataset in your history to expand it</li>  <li>Click on the Download icon <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span> to save the dataset to your computer.</li></ol></blockquote>




<h4 class="faq-area"> Finding Datasets <a href="/training-material/faqs/galaxy/datasets_finding_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-finding-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Finding Datasets</h3>  <ul>  <li>To review all active Datasets in your account, go to <strong>User &gt; Datasets</strong>.</li></ul>  <p>Notes:</p>  <ul>  <li>Logging out of <span class="notranslate">Galaxy</span> while the Upload tool is still loading data can cause uploads to abort. This is most likely to occur when a dataset is loaded by browsing local files.</li>  <li>If you have more than one browser window open, each with a different <span class="notranslate">Galaxy</span> History loaded, the Upload tool will load data into the most recently used history.</li>  <li>Click on refresh icon <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> at the top of the History panel to display the current active History with the datasets.</li></ul></blockquote>




<h4 class="faq-area"> Different dataset icons and their usage <a href="/training-material/faqs/galaxy/datasets_icons.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Icons provide a visual experience for objects, actions, and ideas
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-different-dataset-icons-and-their-usage"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Different dataset icons and their usage</h3>  <p>Dataset icons and their usage:</p>  <ul>  <li><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> <strong>‚ÄúEye icon‚Äù</strong>: Display data of the job in the browser.</li>  <li><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>‚ÄúPencil icon‚Äù</strong>: Edit attributes of the job.</li>  <li><i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-cross</span> <strong>‚Äú‚ÄòX‚Äô icon‚Äù</strong>: Delete the job.</li>  <li><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-info</span> <strong>‚ÄúInfo icon‚Äù</strong>: Job details and run information.</li>  <li><i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> <strong>‚ÄúRefresh/Rerun icon‚Äù</strong>: Run this (selected) job again or examine original submitted form (filled in).</li>  <li><i class="fas fa-bug" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-bug</span> <strong>‚ÄúGreen bug icon‚Äù</strong>: Review and optionally submit a bug report.</li></ul></blockquote>




<h4 class="faq-area"> Understanding job statuses <a href="/training-material/faqs/galaxy/datasets_job_status.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Job statuses will help you understand the stage of your working.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-understanding-job-statuses"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Understanding job statuses</h3>  <p>The following job statuses will help you better understand the working stage of the process.</p>  <ul>  <li><strong>Green</strong>: The job was completed successfully.</li>  <li><strong>Yellow</strong>: The job is executing. Allow this to complete! Should they run longer, they will fail with a ‚Äúwall-time‚Äù error and turn <em>red</em>.</li>  <li><strong>Grey</strong>: The job is being evaluated to run (new dataset) or is queued. Allow this to complete.</li>  <li><strong>Red</strong>: The job has failed.</li>  <li><strong>Light Blue</strong>: The job is paused. This indicates either an input has a problem or that you have exceeded the disk quota set by the administrator of the <span class="notranslate">Galaxy</span> instance you are working on.</li>  <li><strong>Grey, Yellow, Grey again</strong>: The job is waiting to run due to admin re-run or an automatic fail-over to a longer-running cluster.</li>  <li><strong>Bright blue with moving arrow</strong>: May be found in earlier <span class="notranslate">Galaxy</span> versions. Applies to the ‚ÄúGet Data ‚Üí Upload File‚Äù tool only - the upload job is queuing or running.</li></ul>  <p><strong>It is <em>essential</em> to allow queued jobs to remain queued and not delete/re-run them.</strong></p></blockquote>




<h4 class="faq-area"> Moving datasets between <span class="notranslate">Galaxy</span> servers <a href="/training-material/faqs/galaxy/datasets_moving_datasets_between_galaxy_servers.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-moving-datasets-between-galaxy-servers"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Moving datasets between <span class="notranslate">Galaxy</span> servers</h3>  <p>On the origin <span class="notranslate">Galaxy</span> server:</p>  <ol>  <li>Click on name of the dataset to expand the info.</li>  <li>Click on the <strong>Copy link icon</strong><i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-link</span>.On the destination <span class="notranslate">Galaxy</span> server:</li>  <li>Click on <strong>Upload data &gt; Paste / Fetch Data</strong> and paste the link. Select attributes, such as genome assembly, if required. Hit the <strong>Start</strong> button.</li></ol>  <p>Note: The copy link icon <i class="fas fa-link" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-link</span> cannot be used to move HTML datasets (but this can be downloaded using the download button <i class="far fa-save" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-save</span>) and SQLite datasets.</p></blockquote>




<h4 class="faq-area"> Renaming a dataset <a href="/training-material/faqs/galaxy/datasets_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a dataset</h3>  <ul>  <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>  <li>In the central panel, change the <strong>Name</strong> field</li>  <li>Click the <strong>Save</strong> button</li></ul></blockquote>




<h4 class="faq-area"> Upload few files (1-10) <a href="/training-material/faqs/galaxy/datasets_upload.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-few-files-1-10"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload few files (1-10)</h3>  <ol>  <li>Click on <strong>Upload Data</strong> on the top of the left panel</li>  <li>Click on <strong>Choose local file</strong> and select the files or drop the files in the <strong>Drop files here</strong> part</li>  <li>Click on <strong>Start</strong></li>  <li>Click on <strong>Close</strong></li></ol></blockquote>




<h4 class="faq-area"> Upload many files (>10) via FTP <a href="/training-material/faqs/galaxy/datasets_upload_ftp.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-upload-many-files-10-via-ftp"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Upload many files (&gt;10) via FTP</h3>  <ol>  <li>    <p>Make sure to have an FTP client installed</p>    <p>There are many options. We can recommend <a href="https://filezilla-project.org/">FileZilla</a>, a free FTP client that is available on Windows, MacOS, and Linux.</p>  </li>  <li>Establish FTP connection to the <span class="notranslate">Galaxy</span> server    <ol>      <li>Provide the <span class="notranslate">Galaxy</span> server‚Äôs FTP server name (e.g. <code class="language-plaintext highlighter-rouge">use<span class="notranslate">galaxy</span>.org</code>, <code class="language-plaintext highlighter-rouge">ftp.use<span class="notranslate">galaxy</span>.eu</code>)</li>      <li>Provide the <strong>use<span class="notranslate">rna</span>me</strong> (usually the email address) and the <strong>password</strong> on the <span class="notranslate">Galaxy</span> server</li>      <li>Connect</li>    </ol>  </li>  <li>    <p>Add the files to the FTP server by dragging/dropping them or right clicking on them and uploading them</p>    <p>The FTP transfer will start. We need to wait until they are done.</p>  </li>  <li>Open the Upload menu on the <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Choose FTP file</strong> on the bottom</li>  <li>Select files to import into the history</li>  <li>Click on <strong>Start</strong></li></ol></blockquote>



 <br> 

<h3 id="datatypes"> <a href="#datatypes"> Datatypes </a></h3>

<hr/>



<h4 class="faq-area"> Compressed FASTQ files, (`*.gz`) <a href="/training-material/faqs/galaxy/datatypes_fastq_compressed.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-compressed-fastq-files-gz"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Compressed FASTQ files, (<code class="language-plaintext highlighter-rouge">*.gz</code>)</h3>  <ul>  <li>Files ending in <code class="language-plaintext highlighter-rouge">.gz</code> are compressed (zipped) files.    <ul>      <li>The <code class="language-plaintext highlighter-rouge">fastq.gz</code> format is a compressed version of a <code class="language-plaintext highlighter-rouge">fastq</code> dataset.</li>      <li>The <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> format is a compressed version of the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype, etc.</li>    </ul>  </li>  <li>Compression saves space (and therefore your quota).</li>  <li>Tools can accept the compressed versions of input files</li>  <li>Make sure the datatype (compressed or uncompressed) is correct for your files, or it may cause tool errors.</li></ul></blockquote>




<h4 class="faq-area"> FASTQ files: `fastq` vs `fastqsanger` vs .. <a href="/training-material/faqs/galaxy/datatypes_fastq_varieties.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-fastq-files-fastq-vs-fastqsanger-vs-"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: FASTQ files: <code class="language-plaintext highlighter-rouge">fastq</code> vs <code class="language-plaintext highlighter-rouge">fastqsanger</code> vs ..</h3>  <p>FASTQ files come in various flavours. They differ in the encoding scheme they use. See our <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html#inspect-a-raw-sequence-file">QC tutorial</a> for an more detailed explanation of encoding schemes.</p>  <p>Nowadays, the most commonly used encoding scheme is sanger. In <span class="notranslate">Galaxy</span>, this is the <code class="language-plaintext highlighter-rouge">fastqsanger</code> datatype. If you are using older datasets, make sure to verify the FASTQ encoding scheme used in your data.</p>  <p><strong>Be Careful: choosing the wrong encoding scheme can lead to incorrect results!</strong></p>  <p><strong>Tip:</strong> There are 2 <span class="notranslate">Galaxy</span> datatypes that have similar names, but are not the same, please make sure you <code class="language-plaintext highlighter-rouge">fastqsanger</code> and <code class="language-plaintext highlighter-rouge">fastqcssanger</code> (not the additional <code class="language-plaintext highlighter-rouge">cs</code>).</p>  <p><strong>Tip:</strong> When in doubt, choose <code class="language-plaintext highlighter-rouge">fastqsanger</code></p></blockquote>



 <br> 

<h3 id="features"> <a href="#features"> Features </a></h3>

<hr/>



<h4 class="faq-area"> Using the Scratchbook to view multiple datasets <a href="/training-material/faqs/galaxy/features_scratchbook.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-the-scratchbook-to-view-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using the Scratchbook to view multiple datasets</h3>  <p>If you would like to view two or more datasets at once, you can use the <strong>Scratchbook</strong> feature in <span class="notranslate">Galaxy</span>:</p>  <ol>  <li><strong>Click</strong> on the <em>Scratchbook</em> icon <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> on the top menu bar.    <ul>      <li>You should see a little checkmark on the icon now</li>    </ul>  </li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a dataset by clicking on the eye icon <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> to view the output    <ul>      <li>You should see the output in a window overlayed over <span class="notranslate">Galaxy</span></li>      <li>You can resize this window by dragging the bottom-right corner</li>    </ul>  </li>  <li><strong>Click</strong> outside the file to exit the Scratchbook</li>  <li><strong>View</strong> <i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-eye</span> a second dataset from your history    <ul>      <li>You should now see a second window with the new dataset</li>      <li>This makes it easier to compare the two outputs</li>    </ul>  </li>  <li>Repeat this for as many files as you would like to compare</li>  <li>You can turn off the <strong>Scratchbook</strong> <i class="fas fa-th" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-scratchbook</span> by clicking on the icon again</li></ol></blockquote>




<h4 class="faq-area"> Why not use Excel? <a href="/training-material/faqs/galaxy/why_not_excel.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Excel is a fantastic tool and a great place to build simple analysis models, but when it comes to scaling, <span class="notranslate">Galaxy</span> wins every time.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-why-not-use-excel"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Why not use Excel?</h3>  <p>You could just as easily use Excel to answer the same question, and if the goal is to learn how to use a tool, then either tool would be great! But what if you are working on a question where your analysis matters? Maybe you are working with human clinical data trying to diagnose a set of symptoms, or you are working on research that will eventually be published and maybe earn you a Nobel Prize?</p>  <p>In these cases your analysis, <em>and the ability to reproduce it exactly</em>, is vitally important, and Excel won‚Äôt help you here. It doesn‚Äôt track changes and it offers very little insight to others on how you got from your initial data to your conclusions.</p>  <p><span class="notranslate">Galaxy</span>, on the other hand, <em>automatically records every step of your analysis.</em> And when you are done, you can share your analysis with anyone. You can even include a link to it in a paper (or your acceptance speech). In addition, you can create a reusable <span class="notranslate">workflow</span> from your analysis that others (or yourself) can use on other datasets.</p>  <p>Another challenge with sp<span class="notranslate">reads</span>heet programs is that they don‚Äôt scale to support <em>next generation sequencing (NGS)</em> datasets, a common type of data in genomics, and which often reach gigabytes or even terabytes in size. Excel has been used for large datasets, but you‚Äôll often find that learning a new tool gives you significantly more ability to scale up, and scale out your analyses.</p></blockquote>



 <br> 

<h3 id="histories"> <a href="#histories"> Histories </a></h3>

<hr/>



<h4 class="faq-area"> Copy a dataset to a new history <a href="/training-material/faqs/galaxy/histories_copy_dataset.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Sometimes you may want to use a dataset in multiple histories. You do not need to re-upload the data, but you can copy datasets from one history to another.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-copy-a-dataset-to-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Copy a dataset to a new history</h3>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Click on <strong>Copy Dataset</strong></li>  <li>    <p>Select the desired files</p>  </li>  <li>    <p>Give a relevant name to the ‚ÄúNew history‚Äù</p>  </li>  <li>Click on the new history name in the green box that have just appear to switch to this history</li></ol></blockquote>




<h4 class="faq-area"> Creating a new history <a href="/training-material/faqs/galaxy/histories_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Histories are an important part of <span class="notranslate">Galaxy</span>, most people use a new history for every new analysis. Always make sure to give your histories good names, so you can easily find your results back later.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new history</h3>  <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel.</p>  <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>  <ol>  <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>  <li>Select the option <strong>Create New</strong> from the menu</li></ol></blockquote>




<h4 class="faq-area"> Downloading histories <a href="/training-material/faqs/galaxy/histories_download_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-downloading-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Downloading histories</h3>  <ol>  <li>Click on the gear icon <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> on the top of the history panel.</li>  <li>Select ‚ÄúExport History to File‚Äù from the History menu.</li>  <li>Click on the ‚ÄúClick here to generate a new archive for this history‚Äù text.</li>  <li>Wait for the <span class="notranslate">Galaxy</span> server to prepare history for download.</li>  <li>Click on the generated link to download the history.</li></ol></blockquote>




<h4 class="faq-area"> Finding Histories <a href="/training-material/faqs/galaxy/histories_finding_histories.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-finding-histories"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Finding Histories</h3>  <ol>  <li>To review all histories in your account, go to <strong>User &gt; Histories</strong> in the top menu bar.</li>  <li>At the top of the History listing, click on <strong>Advanced Search</strong>.</li>  <li>Set the status to all to view all of your active, deleted, and permanently deleted (purged) histories.</li>  <li>Histories in all states are listed for registered accounts. Meaning one will always find their data here if it ever appears to be ‚Äúlost‚Äù.</li>  <li>Note: Permanently deleted (purged) Histories may be fully removed from the server at any time. The data content inside the History is always removed at the time of purging (by a double-confirmed user action), but the purged History artifact may still be in the listing. Purged data content cannot be restored, even by an administrator.</li></ol></blockquote>




<h4 class="faq-area"> Importing a history <a href="/training-material/faqs/galaxy/histories_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a history</h3>  <ol>  <li>Open the link to the shared history</li>  <li>Click on the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> <em>Import history</em> button on the top right</li>  <li>Enter a title for the new history</li>  <li>Click on <strong>Import</strong></li></ol></blockquote>




<h4 class="faq-area"> Renaming a history <a href="/training-material/faqs/galaxy/histories_rename.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming a history</h3>  <ol>  <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>  <li>Type the new name</li>  <li>Press <kbd>Enter</kbd></li></ol></blockquote>




<h4 class="faq-area"> Searching your history <a href="/training-material/faqs/galaxy/histories_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-searching-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Searching your history</h3>  <p>To make it easier to find datasets in large histories, you can filter your history by keywords as follows:</p>  <ol>  <li>    <p><strong>Click</strong> on the <em>search datasets</em> box at the top of the history panel.</p>    <p><img src="/training-material/shared/images/history_search_datasets.png" alt="history search box" /></p>  </li>  <li><strong>Type</strong> a search term in this box    <ul>      <li>For example a tool name, or sample name</li>    </ul>  </li>  <li>To undo the filtering and show your full history again, press on the <em>clear search</em> button <i class="fas fa-times-circle" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-clear</span> next to the search box</li></ol></blockquote>




<h4 class="faq-area"> Sharing your History <a href="/training-material/faqs/galaxy/histories_sharing.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can share your work in <span class="notranslate">Galaxy</span>. There are various ways you can give access one of your histories to other users.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-sharing-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Sharing your History</h3>  <p>Sharing your history allows others to import and access the datasets, parameters, and steps of your history.</p>  <ol>  <li><strong>Share via link</strong>    <ul>      <li>Open the <strong>History Options</strong> <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> menu (gear icon) at the top of your history panel        <ul>          <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Make History accessible</strong></li>          <li>A <strong>Share Link</strong> will appear that you give to others</li>        </ul>      </li>      <li>Anybody who has this link can view and copy your history</li>    </ul>  </li>  <li><strong>Publish your history</strong>    <ul>      <li><i class="fas fa-toggle-on" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-toggle</span> <strong>Make History publicly available in Published Histories</strong></li>      <li>Anybody on this <span class="notranslate">Galaxy</span> server will see your history listed under the <em>Shared Data</em> menu</li>    </ul>  </li>  <li><strong>Share only with another user</strong>.    <ul>      <li>Click the <strong>Share with a user</strong> button at the bottom</li>      <li>Enter an email address for the user you want to share with</li>      <li>Your history will be shared only with this user.</li>    </ul>  </li>  <li><strong>Finding histories others have shared with me</strong>    <ul>      <li>Click on <strong>User</strong> menu on the top bar</li>      <li>Select <strong>Histories shared with me</strong></li>      <li>Here you will see all the histories others have shared with you directly</li>    </ul>  </li></ol>  <p><strong>Note:</strong> If you want to make changes to your history without affecting the shared version, make a copy by going to <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> <em>History options</em> icon in your history and clicking <em>Copy</em></p></blockquote>




<h4 class="faq-area"> Transfer entire histories from one <span class="notranslate">Galaxy</span> server to another <a href="/training-material/faqs/galaxy/histories_transfer_entire_histories_from_one_galaxy_server_to_another.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-transfer-entire-histories-from-one-galaxy-server-to-another"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Transfer entire histories from one <span class="notranslate">Galaxy</span> server to another</h3>  <ol>  <li>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> in the history panel of the <em>sender</em> <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Export to File</strong></li>  <li>Select either exporting history <strong>to a link</strong> or <strong>to a remote file</strong></li>  <li>Click on the link text to generate a new archive for the history <em>if</em> exporting to a link</li>  <li>Wait for the link to generate</li>  <li>Copy the link address or click on the generated link to download the history archive</li>  <li>Click on <strong>User</strong> on the top menu of the <em>receiver</em> <span class="notranslate">Galaxy</span> server</li>  <li>Click on <strong>Histories</strong> to view saved histories</li>  <li>Click on <strong>Import history</strong> in the grey button on the top right</li>  <li>Select the appropriate importing method based on the choices made in steps 3 and 6    <ul>      <li>Choose <strong>Export URL from another <span class="notranslate">galaxy</span> instance</strong> if link address was copied in step 6</li>      <li>Select <strong>Upload local file from your computer</strong> if history archive was downloaded in step 6</li>      <li>Choose <strong>Select a remote file</strong> if history was exported to a remote file in step 3</li>    </ul>  </li>  <li>Click the link text to check out your histories if import is successful</li></ol>  <p>If history being transferred is too large, you may:</p>  <ol>  <li>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> in the history panel of the <em>sender</em> <span class="notranslate">Galaxy</span> server</li>  <li>Click <strong>Copy Datasets</strong> to move just the important datasets into a new history</li>  <li>Create the archive from that smaller history</li></ol></blockquote>




<h4 class="faq-area"> Undeleting history <a href="/training-material/faqs/galaxy/histories_undelete.html"><i class="fas fa-external-link-alt"></i></a> </h4>
undelete your deleted histories
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-undeleting-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Undeleting history</h3>  <ul>  <li>Click on <strong>User</strong> then select <strong>Histories</strong></li>  <li>Click on <strong>Advanced search</strong> on the top left side below  <strong>Saved Histories</strong></li>  <li>On <strong>Status</strong> click <strong>Deleted</strong></li>  <li>Select the history you want to undelete using the <strong>checkbox</strong> on the left side</li>  <li>Click <strong>Undelete</strong> button below the deleted histories</li></ul></blockquote>



 <br> 

<h3 id="interactive tools"> <a href="#interactive tools"> Interactive tools </a></h3>

<hr/>



<h4 class="faq-area"> Launch JupyterLab <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch JupyterLab</h3>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-jupyterlab-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch JupyterLab in <span class="notranslate">Galaxy</span></h3>  <p>Currently JupyterLab in <span class="notranslate">Galaxy</span> is available on <a href="https://live.usegalaxy.eu">Live.use<span class="notranslate">Galaxy</span>.eu</a>, <a href="https://usegalaxy.org">use<span class="notranslate">galaxy</span>.org</a> and <a href="https://usegalaxy.eu">use<span class="notranslate">galaxy</span>.eu</a>.</p>  <blockquote class="notranslate hands_on">    <h3 data-toc-skip id="hands_on-hands-on-run-jupyterlab"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run JupyterLab</h3>    <ol>      <li><span class="tool" data-tool="interactive_tool_jupyter_notebook" title="Tested with interactive_tool_jupyter_notebook"><strong>Interactive Jupyter Notebook</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: interactive_tool_jupyter_notebook</span></span> :</li>      <li>Click Execute</li>      <li>The tool will start running and will stay running permanently</li>      <li>Click on the <strong>User</strong> menu at the top and go to <strong>Active Interactive Tools</strong> and locate the JupyterLab instance you started.</li>      <li>Click on your JupyterLab instance</li>    </ol>  </blockquote></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-try-jupyterlab-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch Try JupyterLab if not available on <span class="notranslate">Galaxy</span></h3>  <p>If JupyterLab is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Start <a href="https://mybinder.org/v2/gh/jupyterlab/jupyterlab-demo/try.jupyter.org?urlpath=lab">Try JupyterLab</a></li>  </ol></blockquote></blockquote>




<h4 class="faq-area"> Open a Terminal in Jupyter <a href="/training-material/faqs/galaxy/interactive_tools_jupyter_terminal.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-open-a-terminal-in-jupyter"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Open a Terminal in Jupyter</h3>  <p>This tutorial will let you accomplish almost everything from this view, running code in the cells below directly in the training material. You can choose between running the code here, or opening up a terminal tab in which to run it.Here are some instructions for how to do this on various environments.</p>  <p><strong>Jupyter</strong> on Use<span class="notranslate">Galaxy</span>.* and <strong>MyBinder.org</strong></p>  <ol>  <li>    <p>Use the File ‚Üí New ‚Üí Terminal menu to launch a terminal.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-launch.png" alt="screenshot of jupyterlab showing the File menu expanded to show new and terminal option." /></p>  </li>  <li>    <p>Disable ‚ÄúSimple‚Äù mode in the bottom left hand corner, if it activated.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-simple.png" alt="screenshot of jupyterlab showing a toggle labelled simple" /></p>  </li>  <li>    <p>Drag one of the terminal or notebook tabs to the side to have the training materials and terminal side-by-side</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/mybinder-final.png" alt="screenshot of jupyterlab with notebook and terminal side-by-side." /></p>  </li></ol>  <p><strong>CoCalc</strong></p>  <ol>  <li>    <p>Use the <em>Split View</em> functionality of cocalc to split your view into two portions.</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-split.png" alt="screenshot of cocalc button to split views" /></p>  </li>  <li>    <p>Change the view of one panel to a terminal</p>    <p><img src="/training-material/topics/data-science/images/carpentries-cli/cocalc-terminal.png" alt="screenshot of cocalc swapping view port to that of a terminal" /></p>  </li></ol></blockquote>




<h4 class="faq-area"> Open interactive tool <a href="/training-material/faqs/galaxy/interactive_tools_open.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-open-interactive-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Open interactive tool</h3>  <ol>  <li>Go to User &gt; Active InteractiveTools</li>  <li>Wait for the  to be running (Job Info)</li>  <li>Click on</li></ol></blockquote>




<h4 class="faq-area"> Knitting RMarkdown documents in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_knit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-knitting-rmarkdown-documents-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Knitting RMarkdown documents in RStudio</h3>  <p>One of the other nice features of RMarkdown documents is making lovely presentation-quality worthy documents. You can take, for example, a tutorial and produce a nice report like output as HTML, PDF, or <code class="language-plaintext highlighter-rouge">.doc</code> document that can easily be shared with colleagues or students.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-meta-output.png" alt="Screenshot of the metadata with html_notebook and word_document being visible and a number of options controlling their output. TOC, standing for table of contents, has been set to true for both." /></p>  <p>Now you‚Äôre <strong>ready to preview the document</strong>:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of preview dropdown with options like preview, knit to html, knit to pdf, knit to word" /></p>  <p><strong>Click Preview</strong>. A window will popup with a preview of the rendered verison of this document.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview.png" alt="screenshot of rendered document with the table of contents on left, title is in a large font, and there are coloured boxes similar to GTN tutorials offering tips and more information" /></p>  <p>The preview is really similar to the GTN rendering, no cells have been executed, and no output is embedded yet in the preview document. But if you have run cells (e.g. the first few loading a library and previewing the <code class="language-plaintext highlighter-rouge">msleep</code> dataset:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-preview-output.png" alt="screenshot of the rendered document with a fancy table browser embedded as well as the output of each step" /></p>  <p><strong>When you‚Äôre ready to distribute</strong> the document, you can instead use the <code class="language-plaintext highlighter-rouge">Knit</code> button. This runs every cell in the entire document fresh, and then compiles the outputs together with the rendered markdown to produce a nice result file as HTML, PDF, or Word document.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-knit.png" alt="screenshot of the console with 'chunks' being knitted together" /></p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-pdf--word-require-a-latex-installation"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: PDF + Word require a LaTeX installation</h3>  <p>You might need to install additional packages to compile the PDF and Word document versions</p></blockquote>  <p>And at the end you can see a pretty document rendered with all of the output of every step along the way. This is a fantastic way to e.g. distribute read-only lesson materials to students, if you feel they might struggle with using an RMarkdown document, or just want to read the output without <em>doing</em> it themselves.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-pdf.png" alt="screenshot of a PDF document showing the end of the tutorial where a pretty plot has been rendered and there is some text for conclusions and citations" /></p></blockquote>




<h4 class="faq-area"> Launch RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_launch.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-launch-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Launch RStudio</h3>  <p>Depending on which server you are using, you may be able to run RStudio directly in <span class="notranslate">Galaxy</span>. If that is not available, RStudio Cloud can be an alte<span class="notranslate">rna</span>tive.</p>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-in-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio in <span class="notranslate">Galaxy</span></h3>  <p>Currently RStudio in <span class="notranslate">Galaxy</span> is only available on <a href="https://usegalaxy.eu">Use<span class="notranslate">Galaxy</span>.eu</a> and <a href="https://usegalaxy.org">Use<span class="notranslate">Galaxy</span>.org</a></p>  <ol>    <li>Open the Rstudio tool <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> by clicking <a href="https://usegalaxy.eu/?tool_id=interactive_tool_rstudio">here</a></li>    <li>Click Execute</li>    <li>The tool will start running and will stay running permanently</li>    <li>Click on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started.</li>  </ol></blockquote>  <blockquote class="tip">  <h3 data-toc-skip id="tip-tip-launch-rstudio-cloud-if-not-available-on-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Launch RStudio Cloud if not available on <span class="notranslate">Galaxy</span></h3>  <p>If RStudio is not available on the <span class="notranslate">Galaxy</span> instance:</p>  <ol>    <li>Register for <a href="https://client.login.rstudio.cloud/oauth/login?show_auth=0&amp;show_login=1&amp;show_setup=1">RStudio Cloud</a>, or login if you already have an account</li>    <li>Create a new project</li>  </ol></blockquote></blockquote>




<h4 class="faq-area"> Learning with RMarkdown in RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_rmarkdown.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-learning-with-rmarkdown-in-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Learning with RMarkdown in RStudio</h3>  <p>Learning with RMarkdown is a bit different than you might be used to. Instead of copying and pasting code from the GTN into a document you‚Äôll instead be able to run the code directly as it was written, inside RStudio! You can now focus just on the code and reading within RStudio.</p>  <ol>  <li>    <p><strong>Load the notebook</strong> if you have not already, following the tip box at the top of the tutorial</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-download-file.png" alt="Screenshot of the Console in RStudio. There are three lines visible of not-yet-run R code with the download.file statements which were included in the setup tip box." /></p>  </li>  <li>    <p><strong>Open it</strong> by clicking on the <code class="language-plaintext highlighter-rouge">.Rmd</code> file in the file browser (bottom right)</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-file-menu.png" alt="Screenshot of Files tab in RStudio, here there are three files listed, a data-science-r-dplyr.Rmd file, a css and a bib file." /></p>  </li>  <li>    <p>The RMarkdown document will appear in the document viewer (top left)</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-document-view.png" alt="Screenshot of an open document in RStudio. There is some yaml metadata above the tutorial showing the title of the tutorial." /></p>  </li></ol>  <p>You‚Äôre now ready to view the RMarkdown notebook! Each notebook starts with a lot of metadata about how to build the notebook for viewing, but you can ignore this for now and <strong>scroll down to the content of the tutorial</strong>.</p>  <p>You‚Äôll see codeblocks scattered throughout the text, and these are all runnable snippets that appear like this in the document:</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-run-cell.png" alt="Screenshot of the RMarkdown document in the viewer, a cell is visible between markdown text reading library tidyverse. It is slightly more grey than the background region, and it has a run button at the right of the cell in a contextual menu." /></p>  <p>And you have a few options for how to run them:</p>  <ol>  <li>Click the green arrow</li>  <li><kbd>ctrl+enter</kbd></li>  <li>    <p>Using the menu at the top to run all</p>    <p><img src="/training-material/topics/data-science/images/rstudio/r-run-all.png" alt="Screenshot of the run dropdown menu in R, the first item is run selected lines showing the mentioned shortcut above, the second is run next chunk, and then it also mentions a 'run all chunks below' and 'restart r and run all chunks' option." /></p>  </li></ol>  <p>When you run cells, the output will appear below in the Console. RStudio essentially copies the code from the RMarkdown document, to the console, and runs it, just as if you had typed it out yourself!</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-cell-output.png" alt="Screenshot of a run cell, its output is included below in the RMarkdown document and the same output is visible below in the console. It shows a log of loading the tidyverse library." /></p>  <p>One of the best features of RMarkdown documents is that they include a very nice table browser which makes previewing results a lot easier! Instead of needing to use <code class="language-plaintext highlighter-rouge">head</code> every time to preview the result, you get an interactive table browser for any step which outputs a table.</p>  <p><img src="/training-material/topics/data-science/images/rstudio/r-table.png" alt="Screenshot of the table browser. Below a code chunk is a large white area with two images, the first reading 'r console' and the second reading 'tbl_df'. The tbl_df is highlighted like it is active. Below that is a pretty-printed table with bold column headers like name and genus and so on. At the right of the table is a small arrow indicating you can switch to seeing more columns than just the initial three. At the bottom of the table is 1-10 of 83 rows written, and buttons for switching between each page of results." /></p></blockquote>




<h4 class="faq-area"> Stop RStudio <a href="/training-material/faqs/galaxy/interactive_tools_rstudio_stop.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="notranslate hands_on">  <h3 data-toc-skip id="-hands-on-stop-rstudio"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"></span> Hands-on: Stop RStudio</h3>  <p>When you have finished your R analysis, it‚Äôs time to stop RStudio.</p>  <ol>  <li>First, save your work into <span class="notranslate">Galaxy</span>, to ensure reproducibility:    <ol>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_put(filename)</code> to save individual files by supplying the filename</li>      <li>You can use <code class="language-plaintext highlighter-rouge">gx_save()</code> to save the entire analysis transcript and any data objects loaded into your environment.</li>    </ol>  </li>  <li>Once you have saved your data, you can proceed in 2 different ways:    <ul>      <li>Deleting the corresponding history dataset named <code class="language-plaintext highlighter-rouge">RStudio</code> and showing a ‚Äúin progress state‚Äù, so yellow, OR</li>      <li>Clicking on the ‚ÄúUser‚Äù menu at the top and go to ‚ÄúActive InteractiveTools‚Äù and locate the RStudio instance you started, selecting the corresponding box, and finally clicking on the ‚ÄúStop‚Äù button at the bottom.</li>    </ul>  </li></ol></blockquote>



 <br> 

<h3 id="sequencing"> <a href="#sequencing"> Sequencing </a></h3>

<hr/>



<h4 class="faq-area"> Illumina MiSeq sequencing <a href="/training-material/faqs/galaxy/sequencing_illumina_miseq.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-illumina-miseq-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Illumina MiSeq sequencing</h3>  <p>Illumina MiSeq sequencing is based on sequencing by synthesis. As the namesuggests, fluorescent labels are measured for every base that bind at aspecific moment at a specific place on a flow cell. These flow cells arecovered with oligos (small single strand <span class="notranslate">DNA</span> strands). In the librarypreparation the <span class="notranslate">DNA</span> strands are cut into small <span class="notranslate">DNA</span> fragments (differs perkit/device) and specific pieces of <span class="notranslate">DNA</span> (adapters) are added, which arecomplementary to the oligos. Using bridge amplification large amounts ofclusters of these <span class="notranslate">DNA</span> fragments are made. The reverse string is washed away,making the clusters single stranded. Fluorescent bases are added one by one,which emit a specific light for different bases when added. This is happeningfor whole clusters, so this light can be detected and this data is basecalled(translation from light to a nucleotide) to a nucleotide sequence (Read). Forevery base a quality score is determined and also saved per read. Thisprocess is repeated for the reverse strand on the same place on the flowcell, so the forward and reverse <span class="notranslate">reads</span> are from the same <span class="notranslate">DNA</span> strand. Theforward and reversed <span class="notranslate">reads</span> are linked together and should always be processedtogether!</p>  <p>For more information watch this <a href="https://youtu.be/fCd6B5HRaZ8">video from Illumina</a></p></blockquote>




<h4 class="faq-area"> Nanopore sequencing <a href="/training-material/faqs/galaxy/sequencing_nanopore.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="comment">  <h3 data-toc-skip id="-nanopore-sequencing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden"></span> Nanopore sequencing</h3>  <p>Nanopore sequencing has several properties that make it well-suited for our purposes</p>  <ol>  <li>Long-read sequencing technology offers <strong>simplified</strong> and less ambiguous genome <strong>assembly</strong></li>  <li>Long-read sequencing gives the ability to <strong>span repetitive genomic regions</strong></li>  <li>Long-read sequencing makes it possible to <strong>identify large structural variations</strong></li></ol>  <p><img src="/training-material/topics/metagenomics/images/plasmid-metagenomics-nanopore/sequence_method.jpg" alt="How nanopore sequencing works" title="Using nanopore sequencing, a single molecule of DNA or RNA can be sequenced without the need for PCR amplification or chemical labeling of the sample. (Image from: &lt;a href=&quot;https://nanoporetech.com/sites/default/files/s3/white-papers/WGS_Assembly_white_paper.pdf?submissionGuid=40a7546b-9e51-42e7-bde9-b5ddef3c3512&quot;&gt;Nanopore sequencing: The advantages of long reads for genome assembly&lt;/a&gt;)" /> <br /><br /></p>  <p>When using Oxford Nanopore Technologies (ONT) sequencing, the change inelectrical current is measured over the membrane of a flow cell. Whennucleotides pass the pores in the flow cell the current change is translated(basecalled) to nucleotides by a basecaller. A schematic overview is given inthe picture above.</p>  <p>When sequencing using a MinIT or MinION Mk1C, the basecalling software ispresent on the devices. With basecalling the electrical signals are translatedto bases (A,T,G,C) with a quality score per base. The sequenced <span class="notranslate">DNA</span> strand willbe basecalled and this will form one read. Multiple <span class="notranslate">reads</span> will be stored in afastq file.</p></blockquote>



 <br> 

<h3 id="support"> <a href="#support"> Support </a></h3>

<hr/>



<h4 class="faq-area"> Where do I get more support? <a href="/training-material/faqs/galaxy/support.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-where-do-i-get-more-support"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Where do I get more support?</h3>  <p>If you need support for using <span class="notranslate">Galaxy</span>, running your analysis or completing a tutorial, please try one of the following options:</p>  <ul>  <li><strong>Gitter Chat</strong>: You can get help on Gitter chat platform, on various channels.    <ul>      <li><a href="https://gitter.im/galaxyproject/Lobby"><span class="notranslate">Galaxy</span> General Support</a></li>      <li><a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Training Support</a></li>      <li><a href="https://gitter.im/usegalaxy-eu/Lobby"><span class="notranslate">Galaxy</span> EU Server</a></li>    </ul>  </li>  <li>    <p><strong><span class="notranslate">Galaxy</span> Help Forum</strong>: You can also have a look at the <a href="https://help.galaxyproject.org"><span class="notranslate">Galaxy</span> Help Forum</a>. Your question may already have been answered here before. If not, you can post your question here.</p>  </li>  <li><strong>Contact Server Admins:</strong> If you think there is a problem with the <span class="notranslate">Galaxy</span> server, or you would like to make a request, <a href="/training-material/faqs/galaxy/support_admins.html">contact the <span class="notranslate">Galaxy</span> server administrators</a>.</li></ul></blockquote>




<h4 class="faq-area"> Contacting <span class="notranslate">Galaxy</span> Administrators <a href="/training-material/faqs/galaxy/support_admins.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-contacting-galaxy-administrators"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Contacting <span class="notranslate">Galaxy</span> Administrators</h3>  <p>If you suspect there is something wrong with the server, or would like to request a tool to be installed, you should contact the server administrators for the <span class="notranslate">Galaxy</span> you are on.</p>  <ul>  <li><strong>Tool error?</strong> Please follow these <a href="/training-material/faqs/galaxy/analysis_troubleshooting.html">troubleshooting steps</a></li>  <li>Each <span class="notranslate">Galaxy</span> server has different contact procedures, here are the contact options for the 3 biggest servers:    <ul>      <li><strong><span class="notranslate">Galaxy</span> US:</strong> <a href="mailto:help@galaxyproject.org">Email</a>, <a href="https://gitter.im/galaxyproject/Lobby">Gitter channel</a></li>      <li><strong><span class="notranslate">Galaxy</span> EU:</strong> <a href="https://gitter.im/usegalaxy-eu/Lobby">Gitter channel</a>, <a href="https://usegalaxy.eu/tiaas/">Request TIaaS</a></li>      <li><strong><span class="notranslate">Galaxy</span> AU:</strong> <a href="mail:help@genome.edu.au">Email</a>, <a href="https://request.usegalaxy.org.au/">Request a tool</a>, <a href="https://docs.google.com/forms/d/e/1FAIpQLSeiw6ajmkezLCwbXc3OFQEU3Ai9hGnBd967u9YbQ8ANPgvatA/viewform">Request Data Quota</a></li>      <li>Other <span class="notranslate">Galaxy</span> servers? Check the homepage for more information.</li>    </ul>  </li></ul></blockquote>



 <br> 

<h3 id="tools"> <a href="#tools"> Tools </a></h3>

<hr/>



<h4 class="faq-area"> Changing the tool version <a href="/training-material/faqs/galaxy/tools_change_version.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-the-tool-version"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing the tool version</h3>  <p>Tools are frequently updated to new versions. Your <span class="notranslate">Galaxy</span> may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool.</p>  <p><strong>Switching to a different version</strong> of a tool:</p>  <ul>  <li>Open the tool</li>  <li>Click on the <i class="fas fa-cubes" aria-hidden="true"></i><span class="visually-hidden">tool-versions</span> versions logo at the <strong>top right</strong></li>  <li>Select the desired version from the dropdown list</li></ul></blockquote>




<h4 class="faq-area"> Organizing the tool panel <a href="/training-material/faqs/galaxy/tools_favorite.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-organizing-the-tool-panel"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Organizing the tool panel</h3>  <p><span class="notranslate">Galaxy</span> servers can have a lot of tools available, which can make it challenging to find the tool you are looking for. To help find your favourite tools, you can:</p>  <ul>  <li><strong>Keep a list of your favourite tools</strong> to find them back easily later.    <ul>      <li><strong>Adding tools to your favourites</strong>        <ul>          <li>Open a tool</li>          <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-star</span> next to the tool name to add it to your favourites</li>        </ul>      </li>      <li><strong>Viewing your favourite tools</strong>        <ul>          <li>Click on the <strong>star icon</strong> <i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-star</span> at the top of the <span class="notranslate">Galaxy</span> tool panel (above the tool search bar)</li>          <li>This will filter the toolbox to show all your starred tools</li>        </ul>      </li>    </ul>  </li>  <li><strong>Change the tool panel view</strong>    <ul>      <li>Click on the <i class="fas fa-list" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-panelview</span> icon at the top of the <span class="notranslate">Galaxy</span> tool panel (above the tool search bar)</li>      <li>Here you can view the tools by EDAM ontology terms        <ul>          <li>EDAM Topics (e.g. biology, ecology)</li>          <li>EDAM Operations (e.g. quality control, variant analysis)</li>          <li>You can always get back to the default view by choosing <strong>‚ÄúFull Tool Panel‚Äù</strong></li>        </ul>      </li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> Viewing tool logs (`stdout` and `stderr`) <a href="/training-material/faqs/galaxy/tools_logs.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-viewing-tool-logs-stdout-and-stderr"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Viewing tool logs (<code class="language-plaintext highlighter-rouge">stdout</code> and <code class="language-plaintext highlighter-rouge">stderr</code>)</h3>  <p>Most tools create log files as output, which can contain useful information about how the tool ran (<code class="language-plaintext highlighter-rouge">stdout</code>, or <em>standard output</em>), and what went wrong (<code class="language-plaintext highlighter-rouge">stderr</code>, or <em>standard error</em>).</p>  <p>To view these log files in <span class="notranslate">Galaxy</span>:</p>  <ul>  <li>Expand one of the outputs of the tool in your history</li>  <li>Click on <strong>View details</strong> <i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span></li>  <li>Scroll to the <strong>Job Information</strong> section    <ul>      <li>Here you will find links to the log files (<code class="language-plaintext highlighter-rouge">stdout</code> and <code class="language-plaintext highlighter-rouge">stderr</code>).</li>    </ul>  </li></ul></blockquote>




<h4 class="faq-area"> If a Tool is Missing <a href="/training-material/faqs/galaxy/tools_missing.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-if-a-tool-is-missing"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: If a Tool is Missing</h3>  <p>To use the tools installed and available on the <span class="notranslate">Galaxy</span> server:</p>  <ol>  <li>At the top of the left tool panel, type in a tool name or datatype into the tool search box.</li>  <li>Shorter keywords find more choices.</li>  <li>Tools can also be directly browsed by category in the tool panel.</li></ol>  <p>If you can‚Äôt find a tool you need for a tutorial on <span class="notranslate">Galaxy</span>, please:</p>  <ol>  <li><strong>Check that you are using a compatible <span class="notranslate">Galaxy</span> server</strong>    <ul>      <li>Navigate to the overview box at the top of the tutorial</li>      <li>Find the <strong>‚ÄúSupporting Materials‚Äù</strong> section</li>      <li>Check <strong>‚ÄúAvailable on these Galaxies‚Äù</strong></li>      <li>If your server is not listed here, the tutorial is not supported on your <span class="notranslate">Galaxy</span> server</li>      <li>You can create an account on one of the supporting Galaxies  <img src="/training-material/faqs/galaxy/images/supporting_galaxies.png" alt="screenshot of overview box with available Galaxies section" /></li>    </ul>  </li>  <li>Use the <strong>Tutorial mode</strong> feature    <ul>      <li>Open your <span class="notranslate">Galaxy</span> server</li>      <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside <span class="notranslate">Galaxy</span>.</li>      <li>Navigate to your tutorial</li>      <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>      <li><strong>Note:</strong> this does not work for all tutorials (yet)  <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li>    </ul>  </li>  <li><strong>Still not finding the tool?</strong>    <ul>      <li>Ask help in <a href="https://gitter.im/Galaxy-Training-Network/Lobby">Gitter</a>.</li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Re-running a tool <a href="/training-material/faqs/galaxy/tools_rerun.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-re-running-a-tool"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Re-running a tool</h3>  <ol>  <li><strong>Expand</strong> one of the output datasets of the tool (by clicking on it)</li>  <li><strong>Click</strong> re-run <i class="fas fa-redo" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-refresh</span> the tool</li></ol>  <p>This is useful if you want to run the tool again but with slightly different paramters, or if you just want to check which parameter setting you used.</p></blockquote>




<h4 class="faq-area"> Selecting a dataset collection as input <a href="/training-material/faqs/galaxy/tools_select_collection.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-selecting-a-dataset-collection-as-input"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Selecting a dataset collection as input</h3>  <ol>  <li>Click on <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <strong>Dataset collection</strong> in front of the input parameter you want to supply the collection to.</li>  <li>Select the collection you want to use from the list</li></ol></blockquote>




<h4 class="faq-area"> Select multiple datasets <a href="/training-material/faqs/galaxy/tools_select_multiple_datasets.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Select multiple datasets</h3>  <ol>  <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>  <li>Select several files by keeping the <kbd>Ctrl</kbd> (or<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li></ol></blockquote>




<h4 class="faq-area"> Using tutorial mode <a href="/training-material/faqs/galaxy/tutorial_mode.html"><i class="fas fa-external-link-alt"></i></a> </h4>
Tutorial mode saves you screen space, finds the tools you need, and ensures you use the correct versions for the tutorials to run.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-tutorial-mode"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using tutorial mode</h3>  <p>Tools are frequently updated to new versions. Your <span class="notranslate">Galaxy</span> may have multiple versions of the same tool available. By default, you will be shown the latest version of the tool. This may NOT be the same tool used in the tutorial you are accessing. Furthermore, if you use a newer tool in one step, and try using an older tool in the next step‚Ä¶ this may fail! To ensure you use the same tool versions of a given tutorial, use the <strong>Tutorial mode</strong> feature.</p>  <ul>  <li>Open your <span class="notranslate">Galaxy</span> server</li>  <li>Click on the <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> icon on the top menu, this will open the GTN inside <span class="notranslate">Galaxy</span>.</li>  <li>Navigate to your tutorial</li>  <li>Tool names in tutorials will be blue buttons that open the correct tool for you</li>  <li><strong>Note:</strong> this does not work for all tutorials (yet)  <img src="/training-material/topics/contributing/images/88277962-ddda4a80-cce1-11ea-92cd-41b1df063db0.gif" alt="gif showing how GTN-in-Galaxy works" /></li></ul></blockquote>



 <br> 

<h3 id="user preferences"> <a href="#user preferences"> User preferences </a></h3>

<hr/>



<h4 class="faq-area"> Getting your API key <a href="/training-material/faqs/galaxy/preferences_admin_api_key.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-getting-your-api-key"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Getting your API key</h3>  <ol>  <li>In your browser, open your <span class="notranslate">Galaxy</span> homepage</li>  <li>Log in, or register a new account, if it‚Äôs the first time you‚Äôre logging in</li>  <li>Go to <code class="language-plaintext highlighter-rouge">User -&gt; Preferences</code> in the top menu bar, then click on <code class="language-plaintext highlighter-rouge">Manage API key</code></li>  <li>If there is no current API key available, click on <code class="language-plaintext highlighter-rouge">Create a new key</code> to generate it</li>  <li>Copy your API key to somewhere convenient, you will need it throughout this tutorial</li></ol></blockquote>




<h4 class="faq-area"> Forgot Password <a href="/training-material/faqs/galaxy/user_preferences_forgot_password.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-forgot-password"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Forgot Password</h3>  <ol>  <li>Go to the <span class="notranslate">Galaxy</span> server you are using.</li>  <li>Click on <strong>Login or Register</strong>.</li>  <li>Enter your email on the Public Name or Email Address entry box.</li>  <li>Click on the link under the password entry box titled <strong>Forgot password? Click here to reset your password.</strong></li>  <li>An email will be sent with a password reset link. This email may be in your email Spam or Trash folders, depending on your filters.</li>  <li>Click on the reset link in the email or copy and paste it into a web browser window.</li>  <li>Enter your new password and click on <strong>Save new password</strong>.</li></ol></blockquote>




<h4 class="faq-area"> Changing user preferences <a href="/training-material/faqs/galaxy/user_preferences_manage_information.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-changing-user-preferences"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Changing user preferences</h3>  <ol>  <li>Make sure you are logged in to <span class="notranslate">Galaxy</span>.</li>  <li>Go to <strong>User &gt; Preferences</strong> in the top menu bar.</li>  <li>To change email and public name, click on <strong>Manage Information</strong> and to change password, click on <strong>Change Password</strong>.</li>  <li>Make the changes and click on the <strong>Save</strong> button at the bottom.</li>  <li>To change email successfully, verify your account by email through the activation link sent by <span class="notranslate">Galaxy</span>.</li></ol>  <p>Note: Don‚Äôt open another account if your email changes, update the existing account email instead. Creating a new account will be detected as a duplicate and will get your account disabled and deleted.</p></blockquote>



 <br> 

<h3 id="visualisation"> <a href="#visualisation"> Visualisation </a></h3>

<hr/>



<h4 class="faq-area"> Using IGV with <span class="notranslate">Galaxy</span> <a href="/training-material/faqs/galaxy/visualisations_igv.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-using-igv-with-galaxy"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Using IGV with <span class="notranslate">Galaxy</span></h3>  <p>You can send data from your <span class="notranslate">Galaxy</span> history to IGV for viewing as follows:</p>  <ol>  <li>Install IGV on your computer (<a href="https://software.broadinstitute.org/software/igv/download">IGV download page</a>)</li>  <li>Start IGV</li>  <li>In recent verions of IGV, you will have to enable the port:    <ul>      <li>In IGV, go to <code class="language-plaintext highlighter-rouge">View &gt; Preferences &gt; Advanced</code></li>      <li>Check the box <code class="language-plaintext highlighter-rouge">Enable Port</code></li>    </ul>  </li>  <li>In <span class="notranslate">Galaxy</span>, expand the dataset you would like to view in IGV    <ul>      <li>Make sure you have set a reference genome/database correctly (dbkey) (<a href="/training-material/faqs/galaxy/datasets_change_dbkey.html">instructions</a>)</li>      <li>Under <code class="language-plaintext highlighter-rouge">display in IGV</code>, click on <code class="language-plaintext highlighter-rouge">local</code></li>    </ul>  </li></ol></blockquote>



 <br> 

<h3 id="workflows"> <a href="#workflows"> Workflows </a></h3>

<hr/>



<h4 class="faq-area"> Annotate a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_annotate.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-annotate-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Annotate a <span class="notranslate">workflow</span></h3>  <ul>  <li>Open the <span class="notranslate">workflow</span> editor for the <span class="notranslate">workflow</span></li>  <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>Edit Attributes</strong> on the top right</li>  <li>Write a description of the worklow in the <strong>Annotation</strong> box</li>  <li>Add a tag (which will help to search for the <span class="notranslate">workflow</span>) in the <strong>Tags</strong> section</li></ul></blockquote>




<h4 class="faq-area"> Creating a new <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_create_new.html"><i class="fas fa-external-link-alt"></i></a> </h4>
You can create a <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> from scratch in the <span class="notranslate">Galaxy</span> <span class="notranslate">workflow</span> editor.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-creating-a-new-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Creating a new <span class="notranslate">workflow</span></h3>  <ol>  <li>Click <strong><span class="notranslate">Workflow</span></strong> on the top bar</li>  <li>Click the new <span class="notranslate">workflow</span> <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-new</span> button</li>  <li>Give it a clear and memorable name</li>  <li>Clicking <strong>Save</strong> will take you directly into the <span class="notranslate">workflow</span> editor for that <span class="notranslate">workflow</span></li>  <li>Need more help? Please see the <em>How to make a <span class="notranslate">workflow</span> subsection</em> <a href="/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/tutorial.html">here</a></li></ol></blockquote>




<h4 class="faq-area"> Opening the <span class="notranslate">workflow</span> editor <a href="/training-material/faqs/galaxy/workflows_edit.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-opening-the-workflow-editor"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Opening the <span class="notranslate">workflow</span> editor</h3>  <ol>  <li>Click on the name of the imported <span class="notranslate">workflow</span><img src="/training-material/topics/galaxy-interface/images/edit_workflow.png" alt="Workflow drop down menu showing Edit option" title="When you click on the name of a workflow it will open a drop down menu whose first option is Edit. Select the Edit option to edit the workflow." /></li>  <li>Select the Edit <span class="notranslate">workflow</span> to open the <span class="notranslate">workflow</span> in the <span class="notranslate">workflow</span> editor</li></ol></blockquote>




<h4 class="faq-area"> Extracting a <span class="notranslate">workflow</span> from your history <a href="/training-material/faqs/galaxy/workflows_extract_from_history.html"><i class="fas fa-external-link-alt"></i></a> </h4>
<span class="notranslate">Galaxy</span> can automatically create a <span class="notranslate">workflow</span> based on the analysis you have performed in a history. This means that once you have done an analysis manually once, you can easily extract a <span class="notranslate">workflow</span> to repeat it on different data.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-extracting-a-workflow-from-your-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Extracting a <span class="notranslate">workflow</span> from your history</h3>  <ol>  <li>    <p><strong>Clean up</strong> your history: remove any failed (red) jobs from your history by clicking on the <i class="fas fa-times" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-cross</span> button.</p>    <p>This will make the creation of the <span class="notranslate">workflow</span> easier.</p>  </li>  <li>    <p>Click on <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> (<strong>History options</strong>) at the top of your history panel and select <strong>Extract <span class="notranslate">workflow</span></strong>.</p>    <p><img src="/training-material/topics/introduction/images/history_menu_extract_workflow.png" alt="`Extract Workflow` entry in the history options menu" /></p>    <p>The central panel will show the content of the history in reverse order (oldest on top), and you will be able to choose which steps to include in the <span class="notranslate">workflow</span>.</p>  </li>  <li>    <p>Replace the <strong><span class="notranslate">Workflow</span> name</strong> to something more descriptive.</p>  </li>  <li>    <p><strong>Rename</strong> each <span class="notranslate">workflow</span> input in the boxes at the top of the second column.</p>  </li>  <li>    <p>If there are any steps that shouldn‚Äôt be included in the <span class="notranslate">workflow</span>, you can <strong>uncheck</strong> them in the first column of boxes.</p>  </li>  <li>    <p>Click on the <strong>Create <span class="notranslate">Workflow</span></strong> button near the top.</p>    <p>You will get a message that the <span class="notranslate">workflow</span> was created.</p>  </li></ol></blockquote>




<h4 class="faq-area"> Hiding intermediate steps <a href="/training-material/faqs/galaxy/workflows_hide_intermediate_steps.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-hiding-intermediate-steps"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Hiding intermediate steps</h3>  <p>When a <span class="notranslate">workflow</span> is executed, the user is usually primarily interested in the final product and not in all intermediate steps.By default all the outputs of a <span class="notranslate">workflow</span> will be shown, but we can explicitly tell <span class="notranslate">Galaxy</span> which outputs to show and which to hide for a given <span class="notranslate">workflow</span>.This behaviour is controlled by the little checkbox in front of every output dataset:</p>  <p><img src="/training-material/shared/images/workflow_editor_mark_output.png" alt="Asterisk for `out_file1` in the `Select First` tool" /></p></blockquote>




<h4 class="faq-area"> Importing a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_import.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the upload icon <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> at the top-right of the screen</li>  <li>Provide your <span class="notranslate">workflow</span>    <ul>      <li>Option 1: Paste the URL of the <span class="notranslate">workflow</span> into the box labelled <em>‚ÄúArchived <span class="notranslate">Workflow</span> URL‚Äù</em></li>      <li>Option 2: Upload the <span class="notranslate">workflow</span> file in the box labelled <em>‚ÄúArchived <span class="notranslate">Workflow</span> File‚Äù</em></li>    </ul>  </li>  <li>Click the <strong>Import <span class="notranslate">workflow</span></strong> button</li></ul></blockquote>




<h4 class="faq-area"> Importing a <span class="notranslate">workflow</span> using the search <a href="/training-material/faqs/galaxy/workflows_import_search.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-importing-a-workflow-using-the-search"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Importing a <span class="notranslate">workflow</span> using the search</h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> <strong>Import</strong> icon at the top-right of the screen</li>  <li>    <p>Click on <strong>search form</strong> in <strong>Import a <span class="notranslate">Workflow</span> from Configured GA4GH Tool Registry Servers (e.g. Dockstore)</strong></p>  </li>  <li>    <p>Select the relevant TRS Server</p>  </li>  <li>    <p>Type the query</p>  </li>  <li>    <p>Expand the correct <span class="notranslate">workflow</span></p>  </li>  <li>    <p>Click on the wanted version</p>    <p>The <span class="notranslate">workflow</span> will be imported in your workflows</p>  </li></ul></blockquote>




<h4 class="faq-area"> Setting parameters at run-time <a href="/training-material/faqs/galaxy/workflows_parameter_set_runtime.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-setting-parameters-at-run-time"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Setting parameters at run-time</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the parameter you want users to provide every time they run the <span class="notranslate">workflow</span></li>  <li>Click on the arrow in front of the name <i class="far fa-caret-square-up" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-runtime-toggle</span> to toggle to <em>set at runtime</em></li></ol></blockquote>




<h4 class="faq-area"> Make a <span class="notranslate">workflow</span> public <a href="/training-material/faqs/galaxy/workflows_publish.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-make-a-workflow-public"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Make a <span class="notranslate">workflow</span> public</h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows</li>  <li>Click on the interesting <span class="notranslate">workflow</span></li>  <li>Click on <strong>Share</strong></li>  <li>Clik on *Make <span class="notranslate">Workflow</span> Accessible and Publish**</li></ul></blockquote>




<h4 class="faq-area"> Renaming <span class="notranslate">workflow</span> outputs <a href="/training-material/faqs/galaxy/workflows_rename_output.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-renaming-workflow-outputs"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Renaming <span class="notranslate">workflow</span> outputs</h3>  <ol>  <li>Open the <span class="notranslate">workflow</span> editor</li>  <li>Click on the tool in the <span class="notranslate">workflow</span> to get the details of the tool on the right-hand side of the screen.</li>  <li>Scroll down to the <strong>Configure Output</strong> section of your desired parameter, and click it to expand it.    <ul>      <li>        <p>Under <strong>Rename dataset</strong>, give it a meaningful name</p>        <p><img src="/training-material/shared/images/workflow_rename_datasets.png" alt="Rename output datasets" /></p>      </li>    </ul>  </li></ol></blockquote>




<h4 class="faq-area"> Viewing a <span class="notranslate">workflow</span> report <a href="/training-material/faqs/galaxy/workflows_report_view.html"><i class="fas fa-external-link-alt"></i></a> </h4>
When creating a <span class="notranslate">workflow</span> in <span class="notranslate">Galaxy</span>, you can also define an output report page that should be created. Here you can display certain outputs of the pipeline (e.g. output files, tables, images, etc.) and other information about the run.
<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-viewing-a-workflow-report"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Viewing a <span class="notranslate">workflow</span> report</h3>  <ul>  <li>Go to <strong>User</strong> on the top menu bar of <span class="notranslate">Galaxy</span>.</li>  <li>Click on <strong><span class="notranslate">Workflow</span> invocations</strong>    <ul>      <li>Here you will find a list of all the workflows you have run</li>    </ul>  </li>  <li>Click on the name of a <span class="notranslate">workflow</span> invocation to expand it<img src="/training-material/faqs/galaxy/images/workflow_invocations_menu.png" alt="workflow invocations list" /></li>  <li>Click on <strong>View Report</strong> to go to the <span class="notranslate">workflow</span> report page</li>  <li><strong>Note:</strong> The report can also be downloaded in PDF format by clicking on the <i class="fas fa-print" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-wf-report-download</span> icon.</li></ul></blockquote>




<h4 class="faq-area"> Running a <span class="notranslate">workflow</span> <a href="/training-material/faqs/galaxy/workflows_run.html"><i class="fas fa-external-link-alt"></i></a> </h4>

<!--SNIPPET--><blockquote class="tip">  <h3 data-toc-skip id="-tip-running-a-workflow"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden"></span> Tip: Running a <span class="notranslate">workflow</span></h3>  <ul>  <li>Click on <em><span class="notranslate">Workflow</span></em> on the top menu bar of <span class="notranslate">Galaxy</span>. You will see a list of all your workflows.</li>  <li>Click on the <i class="fas fa-play" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">workflow</span>-run</span> (<em>Run <span class="notranslate">workflow</span></em>) button next to your <span class="notranslate">workflow</span></li>  <li>Configure the <span class="notranslate">workflow</span> as needed</li>  <li>Click the <strong>Run <span class="notranslate">Workflow</span></strong> button at the top-right of the screen</li>  <li>You may have to refresh your history to see the queued jobs</li></ul></blockquote>











<br><br><hr>
<h5>Still have questions?</h5>

<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://gitter.im/Galaxy-Training-Network/Lobby" class="btn btn-square-medium btn-yellow"><div>Gitter Chat Support</div></a>


<!--
add a button, include configuration options:

 link: "https://example.com"
 label: this text goes on button
 size: small/medium/large/extralarge
 colour:  blue/purple/green/red/yellow/lightblue/white/black
 square: false , if you want a regular rectangular button
-->




<a href="https://help.galaxyproject.org" class="btn btn-square-medium btn-purple"><div><span class="notranslate">Galaxy</span> Help Forum</div></a>



            </div>
        </div>
    </div>
</section>

        </div>
        
            <footer>
    <div class="container">
        <br><br>
        <p>
            The <a href="https://galaxyproject.org/teach/gtn/">Galaxy Training Network</a>
            provides researchers with online training materials, connects them with local trainers, and helps promoting open data analysis practices worldwide.
        </p>
        <p>
The content of the tutorials is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>. The website and infrastructure is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>.
</p>
        <p>This is revision <a href="https://github.com/galaxyproject/training-material/commit/1fdfcf7b2f9ea897c951d39fd200966b67a20ac2">1fdfcf7b2f9ea897c951d39fd200966b67a20ac2</a></p>
    </div>
</footer>

        
    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
