<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Identification of somatic and germline variants from tumor and normal sample pairs</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:title" content="Galaxy Training: Identification of somatic and germline variants from tumo..." />
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Variant Analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fsomatic-variants%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fsomatic-variants%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fsomatic-variants%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fsomatic-variants%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fsomatic-variants%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fsomatic-variants%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/variant-analysis/tutorials/somatic-variants/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Identification of somatic and germline variants from tumor and normal sample pairs",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
  },
  "courseCode": "variant-analysis / somatic-variants / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Identification of somatic and germline variants from tumor and normal sample pairs' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html",
  "timeRequired": "PT7H",
  "description": "The questions this  addresses are:\n - What are the specific challenges in somatic variant calling that set it apart from regular diploid variant calling?\n - How can you call variants and classify them according to their presence/absence in/from tumor and normal tissue of the same individual?\n - How can you annotate variants and affected genes with prior knowledge from human genetic and cancer-specific databases to generate clinically relevant reports?\n\n\\nThe objectives are:\n - Call variants and their somatic status from whole-exome sequencing data\n - Annotate variants with a wealth of human genetic and cancer-specific information extracted from public databases\n - Add gene-level annotations and generate reports of annotated somatic and germline variants, loss-of-heterozygosity (LOH) events, and affected genes, ready for interpretation by clinicians\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Identification of somatic and germline variants from tumor and normal sample pairs</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/wm75/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/wm75" alt="Wolfgang Maier">Wolfgang Maier</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What are the specific challenges in somatic variant calling that set it apart from regular diploid variant calling?</p>
</li>
            
            <li><p>How can you call variants and classify them according to their presence/absence in/from tumor and normal tissue of the same individual?</p>
</li>
            
            <li><p>How can you annotate variants and affected genes with prior knowledge from human genetic and cancer-specific databases to generate clinically relevant reports?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Call variants and their somatic status from whole-exome sequencing data</p>
</li>
            
            <li><p>Annotate variants with a wealth of human genetic and cancer-specific information extracted from public databases</p>
</li>
            
            <li><p>Add gene-level annotations and generate reports of annotated somatic and germline variants, loss-of-heterozygosity (LOH) events, and affected genes, ready for interpretation by clinicians</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 7 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/variant-analysis/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.2582555">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/somatic-variants/workflows/" title="Workflows" alt="Identification of somatic and germline variants from tumor and normal sample pairs workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>When sequencing genomic material from a human tumor, the underlying clinical or
research question typically is <strong>what spectrum of mutations distinguishes this
tumor from healthy tissue</strong>.</p>

<p>This question cannot be answered adequately just by comparing the tumor tissue
to the human reference genome because even in healthy tissue there will be many
thousands of variants compared to the reference genome. This is because every
individual inherits a unique pattern of that many variants from her parents. A
fundamental difference between these variants and the tumor-specific mutations
is that the former are present in the carrier‚Äôs germline, while the latter have
been acquired somatically and will, thus, not be transmitted to offspring.
Therefor, we talk of germline variants to refer to variants present in healthy
and tumor tissue alike, and of somatic variants to refer to tumor-specific
variants. To be able to distinguish between these two types of variants always
requires a direct comparison of data from tumor and normal tissue samples.</p>

<p>In addition to acquiring new variants, tumors can also lose or gain chromosomal
copies of variants found heterozygously in an individual‚Äôs germline. This
phenomenon is termed <em>loss of heterozygosity</em> (LOH) because only one of the two
original alleles persists in the tumor (either in a hemizygous state if the
other allele is simply dropped, or in a homozygous state in the case of a
duplication of one allelic copy accompanied by loss of the other).
The detection of LOH events, again, is dependent on a comparison of tumor and
normal tissue data.</p>

<p>In this tutorial we are going to identify somatic and germline variants, as
well as variants affected by LOH, from a tumor and a normal sample of the same
patient. Our goal is to report the variant sites, and the genes affected by
them, annotated with the content of general human genetic and cancer-specific
databases. Ideally, this may provide insight into the genetic events driving
tumor formation and growth in the patient, and might be of prognostic and even
therapeutic value by revealing variants known to affect drug
resistance/sensitivity, tumor aggressiveness, <em>etc</em>.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control-and-mapping-of-ngs-reads" id="markdown-toc-quality-control-and-mapping-of-ngs-reads">Quality control and <span class="notranslate">mapping</span> of NGS <span class="notranslate">reads</span></a>    <ol>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
      <li><a href="#read-trimming-and-filtering" id="markdown-toc-read-trimming-and-filtering">Read trimming and filtering</a></li>
      <li><a href="#read-mapping" id="markdown-toc-read-mapping">Read <span class="notranslate">Mapping</span></a></li>
    </ol>
  </li>
  <li><a href="#mapped-reads-postprocessing" id="markdown-toc-mapped-reads-postprocessing">Mapped <span class="notranslate">reads</span> postprocessing</a>    <ol>
      <li><a href="#filtering-on-mapped-reads-properties" id="markdown-toc-filtering-on-mapped-reads-properties">Filtering on mapped <span class="notranslate">reads</span> properties</a></li>
      <li><a href="#removing-duplicate-reads" id="markdown-toc-removing-duplicate-reads">Removing duplicate <span class="notranslate">reads</span></a></li>
      <li><a href="#left-align-reads-around-indels" id="markdown-toc-left-align-reads-around-indels">Left-align <span class="notranslate">reads</span> around indels</a></li>
      <li><a href="#recalibrate-read-mapping-qualities" id="markdown-toc-recalibrate-read-mapping-qualities">Recalibrate read <span class="notranslate">mapping</span> qualities</a></li>
      <li><a href="#refilter-reads-based-on-mapping-quality" id="markdown-toc-refilter-reads-based-on-mapping-quality">Refilter <span class="notranslate">reads</span> based on <span class="notranslate">mapping</span> quality</a></li>
    </ol>
  </li>
  <li><a href="#variant-calling-and-classification" id="markdown-toc-variant-calling-and-classification">Variant calling and classification</a></li>
  <li><a href="#variant-annotation-and-reporting" id="markdown-toc-variant-annotation-and-reporting">Variant annotation and reporting</a>    <ol>
      <li><a href="#get-data-1" id="markdown-toc-get-data-1">Get data</a></li>
      <li><a href="#adding-annotations-to-the-called-variants" id="markdown-toc-adding-annotations-to-the-called-variants">Adding annotations to the called variants</a></li>
      <li><a href="#reporting-selected-subsets-of-variants" id="markdown-toc-reporting-selected-subsets-of-variants">Reporting selected subsets of variants</a></li>
      <li><a href="#generating-reports-of-genes-affected-by-variants" id="markdown-toc-generating-reports-of-genes-affected-by-variants">Generating reports of genes affected by variants</a></li>
      <li><a href="#adding-additional-annotations-to-the-gene-centered-report" id="markdown-toc-adding-additional-annotations-to-the-gene-centered-report">Adding additional annotations to the gene-centered report</a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>First we need to upload and prepare some input data to analyze. The sequencing
<span class="notranslate">reads</span> we are going to analyze are from real-world data from a cancer patient‚Äôs
tumor and normal tissue samples.
For the sake of an acceptable speed of the analysis, the original data has been
downsampled though to include only the <span class="notranslate">reads</span> from human chromosomes 5, 12 and
17.</p>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p>Import the following four files from
<a href="https://zenodo.org/record/2582555">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz
</code></pre></div>      </div>
      <p>where the first two files represent the forward and reverse <span class="notranslate">reads</span> sequence
data from a patient‚Äôs <strong>normal</strong> tissue, and the last two represent the
data of the same patient‚Äôs <strong>tumor</strong> tissue.</p>

      <p>Alte<span class="notranslate">rna</span>tively, the same files may be available on your <span class="notranslate">Galaxy</span> server
through a shared data library (your instructor may tell you so), in which
case you may prefer to import the data directly from there.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Change <strong>Type</strong> from ‚ÄúAuto-detect‚Äù to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-from-a-data-library"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data from a data library</h3>

        <p>As an alte<span class="notranslate">rna</span>tive to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>

        <ul>
          <li>
            <p>Go into <strong>Shared data</strong> (top panel) then <strong>Data libraries</strong></p>
          </li>
          <li>
            <p>Find the correct folder (ask your instructor)</p>
          </li>
          <li>Select the desired files</li>
          <li>Click on the <strong>To History</strong> button near the top and select <strong>as Datasets</strong> from the dropdown menu</li>
          <li>In the pop-up window, select the history you want to import the files to (or create a new one)</li>
          <li>Click on <strong>Import</strong></li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Check that all newly created datasets in your history have their datatypes assigned 
correctly, and fix any missing or wrong datatype assignment</p>

      <blockquote class="tip">

        <h3 id="tip-tip-changing-the-datatype"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Changing the datatype</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, click on the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>
          <li>Select <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></li>
          <li>Click the <strong>Change datatype</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>
      <p>Rename the datasets and add appropriate tags to them</p>

      <p>For datasets that you upload via a link, <span class="notranslate">Galaxy</span> will pick the link
address as the dataset name, which you will likely want to shorten to
just the file names.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>

      <p>Large parts of the analysis in this tutorial will consist of
identical steps performed on the normal and on the tumor tissue data
in parallel.</p>

      <p>To make it easier to keep track of which dataset represents which
branch of an analysis in a linear history, <span class="notranslate">Galaxy</span> supports dataset
tags. In particular, if you attach a tag starting with <code class="language-plaintext highlighter-rouge">#</code> to any
dataset, that tag will automatically propagate to any new dataset
derived from the tagged dataset.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-adding-a-tag"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Adding a tag</h3>
        <ul>
          <li>Click on the dataset</li>
          <li>Click on <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-tags</span> <strong>Edit dataset tags</strong></li>
          <li>
            <p>Add a tag starting with <code class="language-plaintext highlighter-rouge">#</code></p>

            <p>Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset.</p>
          </li>
          <li>Check that the tag is appearing below the dataset name</li>
        </ul>

      </blockquote>

      <p>Before starting our analysis it is, thus, a good idea to tag the two
fastq datasets representing the normal tissue with, <em>e.g.</em>, <code class="language-plaintext highlighter-rouge">#normal</code>
and the two datasets representing the tumor tissue with, <em>e.g.</em>,
<code class="language-plaintext highlighter-rouge">#tumor</code>.</p>
    </li>
    <li>
      <p>Import the reference genome with the <code class="language-plaintext highlighter-rouge">hg19</code> version of the sequences of
human chromosomes 5, 12 and 17:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2582555/files/hg19.chr5_12_17.fa.gz
</code></pre></div>      </div>

      <p>Make sure you specify the datatype as <code class="language-plaintext highlighter-rouge">fasta</code> in the import dialog.</p>

      <blockquote class="comment">
        <h3 id="comment-shortcut"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Shortcut</h3>
        <p>You can skip this step if the <span class="notranslate">Galaxy</span> server you are working on offers
a <code class="language-plaintext highlighter-rouge">hg19</code> version of the human reference genome with prebuilt indexes for
<em>bwa-mem</em> <strong>and</strong> <em>samtools</em> (ask your instructor, or check the tools
<strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>VarScan Somatic</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> if they list a <code class="language-plaintext highlighter-rouge">hg19</code> version as an option under
<em>‚Äú(Using) reference genome‚Äù</em>).</p>
      </blockquote>

      <p>Alte<span class="notranslate">rna</span>tively, load the dataset from a shared data library.</p>
    </li>
    <li>
      <p>Rename the reference genome</p>

      <p>The reference genome you have imported above came as a compressed
file, but got unpacked by <span class="notranslate">Galaxy</span> to plain <code class="language-plaintext highlighter-rouge">fasta</code> format according to
your datatype selection. You may now wish to remove the <code class="language-plaintext highlighter-rouge">.gz</code> suffix
from the dataset name.</p>
    </li>
  </ol>

</blockquote>

<h1 id="quality-control-and-mapping-of-ngs-reads">Quality control and <span class="notranslate">mapping</span> of NGS <span class="notranslate">reads</span></h1>

<p>Before starting our analysis, we would like to make sure that the input data
is of good quality, <em>i.e.</em>, that there haven‚Äôt been any major issues during
<span class="notranslate">DNA</span> preparation, exon capture, or during actual sequencing. To avoid spurious
variant calls due to low input quality, we can ensure that all sequencing <span class="notranslate">reads</span>
used in the analysis meet some minimal quality criteria by trimming low-quality
parts off of the ends of <span class="notranslate">reads</span> and/or discarding <span class="notranslate">reads</span> of poor quality
altogether. The resulting set of polished <span class="notranslate">reads</span> then needs to be mapped to the
human reference genome because knowing the genomic positions that the bases of
a read provide evidence for is, of course, a prerequisite for variant calling.</p>

<blockquote class="comment">
  <h3 id="comment-more-on-quality-control-and-mapping"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> More on quality control and <span class="notranslate">mapping</span></h3>
  <p>If you would like to explore the topics of quality control and read <span class="notranslate">mapping</span>
in detail, you should take a look at the separate
<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control</a>
and <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><span class="notranslate">Mapping</span></a>
tutorials.
Here, we will only illustrate the concrete steps necessary for quality
control and read <span class="notranslate">mapping</span> of our particular datasets.</p>
</blockquote>

<h2 id="quality-control">Quality control</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control-of-the-input-datasets"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control of the input datasets</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  on each of your four fastq datasets
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúShort read data from your current history‚Äù</em>: all 4 FASTQ  datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <blockquote class="tip">

        <h3 id="tip-tip-select-multiple-datasets"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Select multiple datasets</h3>

        <ol>
          <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>
          <li>Select several files by keeping the <kbd>Ctrl</kbd> (or
<kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li>
        </ol>
      </blockquote>

      <p>When you start this job, eight new datasets (one with the calculated raw
data, another one with an html report of the findings for each input
dataset) will get added to your history.</p>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+<span class="notranslate">galaxy</span>0</span></span>  to aggregate the raw <strong>FastQC</strong> data of all four input datasets into one report
      <ul>
        <li>In <em>‚ÄúResults‚Äù</em>
          <ul>
            <li><em>‚ÄúWhich tool was used generate logs?‚Äù</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>‚ÄúFastQC output‚Äù</em>
              <ul>
                <li><em>‚ÄúType of FastQC output?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúFastQC output‚Äù</em>: all four <em>RawData</em>
outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the <em>Webpage</em> output produced by the tool</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What do you think of the base qualities of the sequences?</li>
          <li>Which aspect of the quality report is most puzzling to you?
(Hint: Have a look at the GC content of the <span class="notranslate">reads</span>)</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

          <ol>
            <li>
              <p>The quality of the sequences looks promising. There are no
discernible systematic problems with it.</p>

              <p>Even the reverse <span class="notranslate">reads</span>, which are typically of somewhat poorer
quality than the corresponding forward <span class="notranslate">reads</span>, look good on average.</p>
            </li>
            <li>
              <p>The GC content plots of the forward and the reverse <span class="notranslate">reads</span> from both
samples reveal a very peculiar bimodal distribution.</p>

              <p>Typically, a non-normal distribution of the GC content of the <span class="notranslate">reads</span>
from a sample is considered to hint at possible contamination.
Here, however, we are dealing with sequencing data from captured
exomes, <em>i.e</em>, the <span class="notranslate">reads</span> are not representing random sequences from
a genome, but rather an arbitrary selection.
In fact, the samples at hand were prepared using Agilent‚Äôs
SureSelect V5 technology for exome enrichment, and bimodal GC
content distributions can be seen as a hallmark of that capture
method in several publications (see, for example, Fig. 4C in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4477645/">Meienberg et al., 2015</a>
).</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="read-trimming-and-filtering">Read trimming and filtering</h2>
<p>Although the raw <span class="notranslate">reads</span> used in this tutorial are of relatively good overall
quality already, we will apply read trimming and filtering to see if we can
improve things still a bit, but also to demonstrate the general concept.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read trimming and filtering of the normal tissue <span class="notranslate">reads</span></h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.36.5" title="Tested with toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.36.5"><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.36.5</span></span>  to trim and filter the normal tissue <span class="notranslate">reads</span>
      <ul>
        <li>
          <p><em>‚ÄúSingle-end or paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end (two separate
input files)</code></p>

          <p>This makes the tool treat the forward and reverse <span class="notranslate">reads</span> simultaneously.</p>

          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput FASTQ file (R1/first of pair)‚Äù</em>: the
forward <span class="notranslate">reads</span> (r1) dataset of the normal tissue sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput FASTQ file (R2/second of pair)‚Äù</em>: the
reverse <span class="notranslate">reads</span> (r2) dataset of the normal tissue sample</li>
          </ul>
        </li>
        <li><em>‚ÄúPerform initial ILLUMINACLIP step?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúSelect standard adapter sequences or provide custom?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Standard</code>
              <ul>
                <li><em>‚ÄúAdapter sequences to use‚Äù</em>: <code class="language-plaintext highlighter-rouge">TruSeq3 (paired-ended, for MiSeq and
 HiSeq)</code></li>
              </ul>
            </li>
            <li><em>‚ÄúMaximum mismatch count which will still allow a full match to be
  performed‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><em>‚ÄúHow accurate the match between the two ‚Äòadapter ligated‚Äô <span class="notranslate">reads</span> must
  be for PE palindrome read alignment‚Äù</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
            <li><em>‚ÄúHow accurate the match between any adapter etc. sequence must be
  against a read‚Äù</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>‚ÄúMinimum length of adapter that needs to be detected (PE specific/
  palindrome mode)‚Äù</em>: <code class="language-plaintext highlighter-rouge">8</code></li>
            <li><em>‚ÄúAlways keep both <span class="notranslate">reads</span> (PE specific/palindrome mode)?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>

          <p>These parameters are used to cut ILLUMINA-specific adapter sequences
 from the <span class="notranslate">reads</span>.</p>
        </li>
        <li>In <em>‚ÄúTrimmomatic Operation‚Äù</em>
          <ul>
            <li>In <em>‚Äú1: Trimmomatic Operation‚Äù</em>
              <ul>
                <li><em>‚ÄúSelect Trimmomatic operation to perform‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cut the specified number
 of bases from the start of the read (HEADCROP)</code>
                  <ul>
                    <li><em>‚ÄúNumber of bases to remove from the start of the read‚Äù</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> ‚ÄúInsert Trimmomatic Operation‚Äù*</li>
            <li>In <em>‚Äú2: Trimmomatic Operation‚Äù</em>
              <ul>
                <li><em>‚ÄúSelect Trimmomatic operation to perform‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cut bases off the end of
 a read, if below a threshold quality (TRAILING)</code>
                  <ul>
                    <li><em>‚ÄúMinimum quality required to keep a base‚Äù</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> ‚ÄúInsert Trimmomatic Operation‚Äù*</li>
            <li>In <em>‚Äú3: Trimmomatic Operation‚Äù</em>
              <ul>
                <li><em>‚ÄúSelect Trimmomatic operation to perform‚Äù</em>: <code class="language-plaintext highlighter-rouge">Drop <span class="notranslate">reads</span> below a
 specified length (MINLEN)</code>
                  <ul>
                    <li><em>‚ÄúMinimum quality required to keep a base‚Äù</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>

          <p>These three trimming and filtering operations will be applied to the
 <span class="notranslate">reads</span> in the given order after adapter trimming.</p>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Running this job will generate four output datasets:</p>
<ul>
  <li>two for the trimmed forward and reverse <span class="notranslate">reads</span> that still have a proper
mate in the other dataset</li>
  <li>two more datasets of orphaned forward and reverse <span class="notranslate">reads</span>, for which the
corresponding mate got dropped because of insufficient length after
trimming; when you inspect these two files, however, you should find
that they are empty because none of our relatively high quality <span class="notranslate">reads</span>
got trimmed that excessively. You can delete the two datasets to keep
your history more compact.</li>
</ul>

<blockquote class="details">
  <h3 id="details-more-on-handling-unpaired-reads-in-paired-end-data"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> More on handling unpaired <span class="notranslate">reads</span> in paired-end data</h3>
  <p>Splitting out potential unpaired <span class="notranslate">reads</span> into separate datasets like this
is important because read mappers, typically, expect <span class="notranslate">reads</span> in forward
and reverse input datasets to be arranged in proper pairs, and <span class="notranslate">reads</span>
in one of the datasets without a counterpart in the other would destroy
that expected structure. Therefor, when your data is paired-end data
always make sure you use Trimmomatic in paired-end mode, too!</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-trimming-and-filtering-of-the-tumor-tissue-reads"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read trimming and filtering of the tumor tissue <span class="notranslate">reads</span></h3>
  <ol>
    <li>
      <p>Trim and filter the <strong>tumor tissue</strong> <span class="notranslate">reads</span> following the same steps as above, just change the two input datasets to treat the tumor tissue <span class="notranslate">reads</span> with identical settings.</p>
    </li>
    <li>
      <p>Check that the two unpaired <span class="notranslate">reads</span> datasets are empty, and delete them.</p>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-exercise-quality-control-of-the-polished-datasets"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Exercise: Quality control of the polished datasets</h3>
  <p>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1"><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+<span class="notranslate">galaxy</span>1</span></span>  and <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0"><strong>MultiQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+<span class="notranslate">galaxy</span>0</span></span>  like before,
but using the four trimmed datasets produced by Trimmomatic as input.</p>

  <blockquote class="question">
    <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>How did read trimming affect the quality reports?</p>

    <blockquote class="solution">
      <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

      <p>As expected, trimming the relatively high-quality raw <span class="notranslate">reads</span> did not
have any substantial impact on average. A small fraction of successful
adapter removal is visible though.</p>

    </blockquote>
  </blockquote>
</blockquote>

<h2 id="read-mapping">Read <span class="notranslate">Mapping</span></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-read-mapping"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Read <span class="notranslate">Mapping</span></h3>
  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1</span></span>  to map the <span class="notranslate">reads</span> from the <strong>normal tissue</strong> sample to the reference genome
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-sequence"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>‚ÄúWill you select a reference genome from your history or use a
built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUse the following dataset as the reference sequence‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect first set of <span class="notranslate">reads</span>‚Äù</em>: the trimmed 
forward <span class="notranslate">reads</span> (r1) dataset of the <strong>normal tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect second set of <span class="notranslate">reads</span>‚Äù</em>: the trimmed
reverse <span class="notranslate">reads</span> (r2) dataset of the <strong>normal tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group identifier (ID)‚Äù</em>: <code class="language-plaintext highlighter-rouge">231335</code> (this value being taken
from the original name of the normal tissue input files)</li>
              </ul>
            </li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group sample name (SM)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Normal</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-more-on-read-group-identifiers-and-sample-names"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> More on read group identifiers and sample names</h3>
        <p>In general, you are free to choose ID and SM values to your liking,
but the ID should unambiguously identify the sequencing run that
produced the <span class="notranslate">reads</span>, while the SM value should identify the
biological sample.</p>
      </blockquote>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1"><strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1</span></span>  to map the <span class="notranslate">reads</span> from the <strong>tumor tissue</strong> sample,
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in index?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <p>Adjust these settings as before if you are using the imported reference
genome.</p>
        </li>
        <li><em>‚ÄúSingle or Paired-end <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect first set of <span class="notranslate">reads</span>‚Äù</em>: the trimmed
forward <span class="notranslate">reads</span> (r1) dataset of the <strong>tumor tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect second set of <span class="notranslate">reads</span>‚Äù</em>: the reverse
<span class="notranslate">reads</span> (r2) dataset of the <strong>tumor tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>‚ÄúSet read groups information?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group identifier (ID)‚Äù</em>: <code class="language-plaintext highlighter-rouge">231336</code> (this value, again, being
taken from the original name of the tumor tissue input files)</li>
              </ul>
            </li>
            <li><em>‚ÄúAuto-assign‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>‚ÄúRead group sample name (SM)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tumor</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="mapped-reads-postprocessing">Mapped <span class="notranslate">reads</span> postprocessing</h1>

<p>To ensure that we base our variant analysis only on unambiguous, high-quality
read <span class="notranslate">mapping</span>s we will do some postprocessing next.</p>

<h2 id="filtering-on-mapped-reads-properties">Filtering on mapped <span class="notranslate">reads</span> properties</h2>

<p>To produce new filtered BAM datasets with only those <span class="notranslate">reads</span> retained that have
been mapped to the reference successfully, have a minimal <span class="notranslate">mapping</span> quality of 1,
and for which the mate read has also been mapped:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-filtering-for-mapping-status-and-quality"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Filtering for <span class="notranslate">mapping</span> status and quality</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1"><strong>Filter BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM dataset(s) to filter‚Äù</em>: mapped <span class="notranslate">reads</span>
 datasets from the normal <em>and</em> the tumor tissue data, outputs of
 <strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚ÄúCondition‚Äù</em>:</li>
      </ul>
      <ul>
        <li>In <em>‚Äú1: Condition‚Äù</em>:
          <ul>
            <li>In <em>‚ÄúFilter‚Äù</em>:
              <ul>
                <li>In <em>‚Äú1: Filter‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapQuality</code>
                      <ul>
                        <li><em>‚ÄúFilter on read <span class="notranslate">mapping</span> quality (phred scale)‚Äù</em>: <code class="language-plaintext highlighter-rouge">&gt;=1</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>In <em>‚Äú2: Filter‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">isMapped</code>
                      <ul>
                        <li><em>‚ÄúSelected mapped <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>Click on <em>‚ÄúInsert Filter‚Äù</em></li>
                <li>In <em>‚Äú3: Filter‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">isMateMapped</code>
                      <ul>
                        <li><em>‚ÄúSelect <span class="notranslate">reads</span> with mapped mate‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>

          <p>When you configure multiple filters within one condition, <span class="notranslate">reads</span> have
to pass <em>all</em> the filters to be retained in the output. The above
settings, thus, retain only read pairs, for which both mates are
mapped.</p>

          <p>Note that filtering for a minimal <span class="notranslate">mapping</span> quality is not strictly
necessary. Most variant callers (including **VarScan somatic, which we
will be using later) have an option for using only <span class="notranslate">reads</span> above a
certain <span class="notranslate">mapping</span> quality. In this section, however, we are going to
process the retained <span class="notranslate">reads</span> further rather extensively so it pays off
in terms of performance to eliminate <span class="notranslate">reads</span> we do not plan to use at an
early step.</p>
        </li>
      </ul>
    </li>
  </ol>

  <ul>
    <li><em>‚ÄúWould you like to set rules?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
  </ul>

</blockquote>

<p>This will result in two new datasets, one for each of the normal and tumor data.</p>

<blockquote class="details">
  <h3 id="details-tools-for-filtering-bam-datasets"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Tools for filtering BAM datasets</h3>
  <p>The related tutorial on variant detection from exome-seq data, demonstrates
nearly identical filtering of BAM datasets using the tool <strong>Filter BAM
datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

  <p>These two tools offer very similar functionality and can often be used
interchangeably. Under the hood, <strong>Filter SAM or BAM, output SAM or BAM</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> uses the <code class="language-plaintext highlighter-rouge">samtools view</code> command line tool, while <strong>Filter
BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> uses <code class="language-plaintext highlighter-rouge">bamtools
filter</code>. Whatever the BAM filtering task, you should be able to perform it in
<span class="notranslate">Galaxy</span> with one of these two tools.</p>

</blockquote>

<h2 id="removing-duplicate-reads">Removing duplicate <span class="notranslate">reads</span></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Remove duplicates</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1"><strong>RmDup</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM file‚Äù</em>: filtered <span class="notranslate">reads</span> datasets from
 the normal <em>and</em> the tumor tissue data; the outputs of
 <strong>Filter SAM or BAM</strong></li>
        <li><em>‚ÄúIs this paired-end or single end data‚Äù</em>: <code class="language-plaintext highlighter-rouge">BAM is paired-end</code></li>
      </ul>
      <ul>
        <li><em>‚ÄúTreat as single-end‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Again, this will produce two new datasets, one for each of the normal and
tumor data.</p>

<h2 id="left-align-reads-around-indels">Left-align <span class="notranslate">reads</span> around indels</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-left-align"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Left-align</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1"><strong>BamLeftAlign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM dataset to re-align‚Äù</em>: your
filtered and deduplicated <span class="notranslate">reads</span> datasets from the normal <em>and</em> the tumor
tissue data; the outputs of <strong>RmDup</strong></li>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly
named choice)</li>
          </ul>
        </li>
        <li><em>‚ÄúMaximum number of iterations‚Äù</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-using-the-imported-hg19-sequence-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
    <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
    <ul>
      <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">History</code>
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUsing reference file‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset</li>
        </ul>
      </li>
    </ul>
  </blockquote>
</blockquote>

<p>As before, this will generate two new datasets, one for each of the normal and tumor data.</p>

<h2 id="recalibrate-read-mapping-qualities">Recalibrate read <span class="notranslate">mapping</span> qualities</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-recalibrate-read-quality-scores"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Recalibrate read quality scores</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_calmd/samtools_calmd/2.0.2" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/samtools_calmd/samtools_calmd/2.0.2"><strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/samtools_calmd/samtools_calmd/2.0.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM file to recalculate‚Äù</em>: the left-aligned
datasets from the normal and the tumor tissue data; the outputs of
<strong>BamLeftAlign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><em>‚ÄúUsing reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named choice)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-sequence-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>‚ÄúChoose the source for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúUsing reference file‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li>
          <p><em>‚ÄúDo you also want BAQ (Base Alignment Quality) scores to be calculated?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

          <p>The <em>VarScan somatic</em> tool that we are going to use for calling variants at
the next step is typically used in combination with unadjusted base quality
scores because the general opinion is that the base quality downgrades
performed by <em>CalMD</em> and other tools from the <em>samtools</em> suite of tools
are too severe for <em>VarScan</em> to retain good sensitivity. We are sticking
with this practice in this tutorial.</p>

          <blockquote class="comment">
            <h3 id="comment-using-adjusted-base-quality-scores"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using adjusted base quality scores</h3>
            <p>If, for your own data, you would like to experiment with adjusted base
quality scores, it is important to understand that <em>VarScan somatic</em> will
only make use of the adjusted scores if they are incorporated directly into
the read base qualities of a BAM input dataset, but not if they are written
to the dataset separately.</p>

            <p>Hence, should you ever decide to use:</p>
            <ul>
              <li>
                <p><em>‚ÄúDo you also want BAQ (Base Alignment Quality) scores to be calculated?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, run BAQ calculation</code></p>

                <p>and you want this setting to affect <span class="notranslate">downstream</span> variant calling with
<em>VarScan somatic</em>, make sure you also set then:</p>
                <ul>
                  <li><em>‚ÄúUse BAQ to cap read base qualities‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                </ul>
              </li>
            </ul>

            <p>Please also note that BAQ scores are quite expensive to calculate so
be prepared to see a substantial (up to 10x!) increase in job run time 
when enabling it.</p>
          </blockquote>
        </li>
        <li><em>‚ÄúAdditional options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Advanced options</code>
          <ul>
            <li><em>‚ÄúChange identical bases to ‚Äò=‚Äô‚Äú</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li>
              <p><em>‚ÄúCoefficient to cap <span class="notranslate">mapping</span> quality of poorly mapped <span class="notranslate">reads</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">50</code></p>

              <p>This last setting is the real reason why we use CalMD at this point.
It is an empirical, but well-established finding that the <span class="notranslate">mapping</span>
quality of <span class="notranslate">reads</span> mapped with <em>bwa</em> should be capped this way before
variant calling.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This will, once more, produce two new datasets, one for each of the normal
and tumor data.</p>

<h2 id="refilter-reads-based-on-mapping-quality">Refilter <span class="notranslate">reads</span> based on <span class="notranslate">mapping</span> quality</h2>

<p>During recalibration of read <span class="notranslate">mapping</span> qualities <strong>CalMD</strong> may have set some
<span class="notranslate">mapping</span> quality scores to 255. This special value is reserved for <em>undefined</em>
<span class="notranslate">mapping</span> qualities and is used by the tool when a recalibrated <span class="notranslate">mapping</span> quality
would drop below zero. In other words, a value of 255 does not indicate a
particularly good <span class="notranslate">mapping</span> score, but a really poor one.
To remove such <span class="notranslate">reads</span> from the data:</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-eliminating-reads-with-undefined-mapping-quality"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Eliminating <span class="notranslate">reads</span> with undefined <span class="notranslate">mapping</span> quality</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1"><strong>Filter BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>‚ÄúBAM dataset(s) to filter‚Äù</em>: the recalibrated
 datasets from the normal and the tumor tissue data; the outputs of
 <strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>‚ÄúCondition‚Äù</em>:</li>
      </ul>
      <ul>
        <li>In <em>‚Äú1: Condition‚Äù</em>:
          <ul>
            <li>In <em>‚ÄúFilter‚Äù</em>:
              <ul>
                <li>In <em>‚Äú1: Filter‚Äù</em>:
                  <ul>
                    <li><em>‚ÄúSelect BAM property to filter on‚Äù</em>: <code class="language-plaintext highlighter-rouge">mapQuality</code>
                      <ul>
                        <li><em>‚ÄúFilter on read <span class="notranslate">mapping</span> quality (phred scale)‚Äù</em>: <code class="language-plaintext highlighter-rouge">&lt;=254</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="variant-calling-and-classification">Variant calling and classification</h1>

<p>Having generated a high-quality set of mapped read pairs, we can proceed to
variant calling. The tool <strong>VarScan somatic</strong> is a dedicated solution for
somatic variant calling that:</p>

<ul>
  <li>detects variant alleles in tumor/normal sample pairs</li>
  <li>calls sample genotypes at variant sites</li>
  <li>classifies variants into germline, somatic and LOH event variants using
solid classical statistics even in the presence of non-pure samples like
those obtained from biopsies</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-calling-and-classification"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant calling and classification</h3>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/varscan_somatic/varscan_somatic/2.4.3.6" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/varscan_somatic/varscan_somatic/2.4.3.6"><strong>VarScan somatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/varscan_somatic/varscan_somatic/2.4.3.6</span></span>  with the following parameters:
      <ul>
        <li><em>‚ÄúWill you select a reference genome from your history or use a built-in
genome?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><em>‚Äúreference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named choice)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-sequence-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> sequence</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>‚ÄúWill you select a reference genome from your history or use a
built-in genome?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúreference genome‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúaligned <span class="notranslate">reads</span> from normal sample‚Äù</em>: the mapped
and fully post-processed normal tissue dataset; one of the two outputs
of filtering the <strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span><em>‚Äúaligned <span class="notranslate">reads</span> from tumor sample‚Äù</em>: the mapped
and fully post-processed tumor tissue dataset; the other output of
filtering the <strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs</li>
        <li><em>‚ÄúEstimated purity (non-tumor content) of normal sample‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>‚ÄúEstimated purity (tumor content) of tumor sample‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
        <li><em>‚ÄúGenerate separate output datasets for SNP and indel calls?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúSettings for Variant Calling‚Äù</em>: <code class="language-plaintext highlighter-rouge">Customize settings</code>
          <ul>
            <li>
              <p><em>‚ÄúMinimum base quality‚Äù</em>: <code class="language-plaintext highlighter-rouge">28</code></p>

              <p>We have seen, at the quality control step, that our sequencing data is of
really good quality, and we have chosen not to downgrade base qualities at
the quality scores recalibration step above, so we can increase the base
quality required at any given position without throwing away too much of
our data.</p>
            </li>
            <li>
              <p><em>‚ÄúMinimum <span class="notranslate">mapping</span> quality‚Äù</em>: <code class="language-plaintext highlighter-rouge">1</code></p>

              <p>During postprocessing, we have filtered our <span class="notranslate">reads</span> for ones with a
<span class="notranslate">mapping</span> quality of at least one, but <strong>CalMD</strong> may have lowered some
<span class="notranslate">mapping</span> qualities to zero afterwards.</p>
            </li>
          </ul>

          <p>Leave all other settings in this section at their default values.</p>
        </li>
        <li><em>‚ÄúSettings for Posterior Variant Filtering‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use default values</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h1 id="variant-annotation-and-reporting">Variant annotation and reporting</h1>

<p>For this tutorial we are going to use variant and gene annotations from many
different sources. Most of these are handled for us by the tools we are going
to use in this section, but we need to take care of importing the data from
four sources into <span class="notranslate">Galaxy</span> separately:</p>

<ul>
  <li>variant annotations from <a href="https://www.cancerhotspots.org">Cancer Hotspots</a></li>
  <li>variant and gene information from the
<a href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer Biomarkers database</a>
of the Cancer Genome Interpreter (CGI) project</li>
  <li>variant and gene information from the <a href="https://civicdb.org">CIViC</a> database</li>
  <li>variant annotations from <a href="https://www.ncbi.nlm.nih.gov/snp">dbSNP</a></li>
  <li>lists of genes annotated with the keywords <em>proto-oncogene</em> or <em>tumor
suppressor</em> at <a href="https://www.uniprot.org/">UniProt</a></li>
</ul>

<p>Each of these annotation sets has been released either in the public domain or
under a free data license, which allows you to use them as part of this
tutorial, but also for other purposes.</p>

<p>Starting from the data downloaded from each of these sites, we have generated
a set of new data files tailored to the requirements of the <span class="notranslate">workflow</span> of this
tutorial and have made them available through Zenodo, again under a free data
license.</p>

<h2 id="get-data-1">Get data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Import the following <strong>variant annotation files</strong> from
<a href="https://zenodo.org/record/2581873">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2581873/files/hotspots.<span class="notranslate">bed</span>
https://zenodo.org/record/2581873/files/cgi_variant_positions.<span class="notranslate">bed</span>
https://zenodo.org/record/2581873/files/01-Feb-2019-CIVic.<span class="notranslate">bed</span>
https://zenodo.org/record/2582555/files/dbsnp.b147.chr5_12_17.vcf.gz
</code></pre></div>      </div>

      <p>Make sure you select <code class="language-plaintext highlighter-rouge"><span class="notranslate">bed</span></code> as the datatype for the first three files and
<code class="language-plaintext highlighter-rouge">vcf</code> for the last file.</p>

      <p>Alte<span class="notranslate">rna</span>tively, add the files from a shared data library on your <span class="notranslate">Galaxy</span>
server instance.</p>
    </li>
    <li>
      <p>Import some <strong>gene-level annotation files</strong> from
<a href="https://zenodo.org/record/2581881">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2581881/files/Uniprot_Cancer_Genes.13Feb2019.txt
https://zenodo.org/record/2581881/files/cgi_genes.txt
https://zenodo.org/record/2581881/files/01-Feb-2019-GeneSummaries.tsv
</code></pre></div>      </div>

      <p>and make sure you select <code class="language-plaintext highlighter-rouge">tabular</code> as their datatype, or add them from
the shared data library.</p>
    </li>
    <li>
      <p>Download <strong>SnpEff functional genomic annotations</strong></p>

      <blockquote class="comment">
        <h3 id="comment-shortcut-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Shortcut</h3>
        <p>You can skip this step if the <span class="notranslate">Galaxy</span> server you are working on offers
<code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> as a locally installed snpEff database. You can
check the <strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool under <strong>Genome source</strong> to
see if this is the case.</p>
      </blockquote>

      <p>Use <strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to download genome annotation
database <code class="language-plaintext highlighter-rouge">hg19</code>.</p>
    </li>
    <li>
      <p>Check that the datatypes of all new datasets have been set correctly, and
change them if necessary. You may also want to shorten some of the
dataset names.</p>
    </li>
  </ol>

</blockquote>

<h2 id="adding-annotations-to-the-called-variants">Adding annotations to the called variants</h2>

<h3 id="adding-functional-genomic-annotations">Adding functional genomic annotations</h3>

<p>Certainly, not all variants are equal. Many may just be silent mutations with
no effect at the amino acid level, while a few others may be disrupting the
coding sequence of a protein by introducing a premature stop codon or a
frameshift. Of course, it is also important to know <em>which</em> gene is affected
by a variant.
Such functional genomic annotations can be added to a VCF dataset of variants
with <em>SnpEff</em>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adding-annotations-with-snpeff"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adding annotations with SnpEff</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy1"><strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.<span class="notranslate">galaxy</span>1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSequence changes (SNPs, MNPs, InDels)‚Äù</em>: the
output of <strong>VarScan somatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúInput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">VCF</code></li>
        <li><em>‚ÄúOutput format‚Äù</em>: <code class="language-plaintext highlighter-rouge">VCF (only if input is VCF)</code></li>
        <li><em>‚ÄúGenome source‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally installed reference genome</code>
          <ul>
            <li><em>‚ÄúGenome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-using-the-imported-hg19-snpeff-genome-database"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Using the imported <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff genome database</h3>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff genome database into your
history instead:</p>
            <ul>
              <li><em>‚ÄúGenome source‚Äù</em>: <code class="language-plaintext highlighter-rouge">Downloaded snpEff database in your history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSnpEff4.3 Genome Data‚Äù</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>‚ÄúProduce Summary Stats‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="adding-genetic-and-clinical-evidence-based-annotations">Adding genetic and clinical evidence-based annotations</h3>

<p>Other interesting pieces of information about a variant include aspects like
whether the variant has been observed before in the human population and,
if so, at which frequency. If a variant is known to be associated with specific
diseases, we would also very much like to know that.
To proceed with this kind of genetic and clinical evidence-based annotations,
we are going to convert our list of variants into a database that can be
handled more efficiently than a VCF dataset. We will use the <em>GEMINI</em> tool
suite for this task and for all further work with the variants.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-creating-a-gemini-database-from-a-variants-dataset"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Creating a GEMINI database from a variants dataset</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+galaxy2"><strong>GEMINI load</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+<span class="notranslate">galaxy</span>2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúVCF dataset to be loaded in the GEMINI database‚Äù</em>:
the output of <strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>‚ÄúThe variants in this input are‚Äù</em>: <code class="language-plaintext highlighter-rouge">annotated with snpEff</code></li>
        <li>
          <p><em>‚ÄúThis input comes with genotype calls for its samples‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>Calling sample genotypes was part of what we used VarScan somatic for.</p>
        </li>
        <li><em>‚ÄúChoose a gemini annotation source‚Äù</em>: select the latest available annotations snapshot (most likely, there will be only one)</li>
        <li><em>‚ÄúSample and family information in PED format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nothing selected</code></li>
        <li><em>‚ÄúLoad the following optional content into the database‚Äù</em>
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúGERP scores‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúCADD scores‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúGene tables‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚ÄúSample genotypes‚Äù</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúvariant INFO field‚Äù</em></li>
          </ul>

          <p>Be careful to leave <strong>unchecked</strong>:</p>
          <ul>
            <li>
              <p><em>‚ÄúGenotype likelihoods (sample PLs)‚Äù</em></p>

              <p>VarScan somatic does not generate these values</p>
            </li>
            <li>
              <p><em>‚Äúonly variants that passed all filters‚Äù</em></p>

              <p>It is simple and more flexible to filter for this property later</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>During the creation of the database <em>GEMINI</em> already (silently) adds an
impressive amount of annotations it knows about to our variants (including,
<em>e.g.</em>, the frequency at which every variant has been observed in large human
genome sequencing projects). We have gotten all of these for free, just by converting the variants to a GEMINI database!</p>

<p><em>GEMINI</em> also extracts a lot of the information stored in the VCF input dataset for us (such as the functional genomic annotations that we added with
<em>SnpEff</em>).</p>

<p>However, there are typically additional annotations from other sources (not
incorporated into <em>GEMINI</em>) that one would like to add. In addition, <em>GEMINI</em>
is not prepared to extract some non-standard information from VCF datasets,
including some important bits added by <em>VarScan somatic</em>.</p>

<p><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is the tool that is designed to help you
with these rather common issues. It lets you add further annotations to the
variants in an already loaded GEMINI database.</p>

<p>As a first step, we are going to use the tool to add some crucial information
generated by <em>VarScan somatic</em>, but not recognized by <em>GEMINI load</em>, to the
database. Specifically, we are interested in three values calculated by
<em>VarScan somatic</em> for each variant it detected:</p>

<ul>
  <li>
    <p><em>Somatic status (SS)</em></p>

    <p>This is a simple numeric code, in which a value of <code class="language-plaintext highlighter-rouge">1</code> indicates a germline
variant, <code class="language-plaintext highlighter-rouge">2</code> a somatic variant and <code class="language-plaintext highlighter-rouge">3</code> an LOH event.</p>
  </li>
  <li>
    <p><em>Germline p-value (GPV)</em></p>

    <p>For variants with a somatic status of 1, this is the error probability
associated with that status call.</p>
  </li>
  <li>
    <p><em>Somatic p-value (SPV)</em></p>

    <p>This is the error probability associated with status calls of 2 and 3
(somatic and LOH calls).</p>
  </li>
</ul>

<p>These values are encoded in the <em>INFO</em> column of the VarScan-generated VCF
dataset and we are going to extract them from there and add them to the
<em>GEMINI</em> database.</p>

<blockquote class="details">
  <h3 id="details-handling-of-the-vcf-info-field-by-gemini-load"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Handling of the VCF INFO field by GEMINI load</h3>
  <p>If you paid close attention to how we generated the GEMINI database, you
might remember that, under <em>‚ÄúLoad the following optional content into the
database‚Äù</em>, we checked the option <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúvariant INFO
field‚Äù</em>. Did the tool not do what it was supposed to do?</p>

  <p>The answer is that it did, but not in the way you may expect.</p>

  <p>In fact, <em>GEMINI load</em> always extracts INFO field elements it knows about and
stores them into predefined columns of the variants table in the database.
In addition, it can store the <em>whole</em> INFO field of each variant into a
separate <em>info</em> column so that no information gets lost on the way from the
VCF input to the GEMINI database, and that‚Äôs the optional behavior we have
been asking for. To keep things compact, however, this column stores the INFO
content in compressed form, which is not readily accessible.</p>

  <p>That‚Äôs why it is necessary to extract non-standard INFO field elements
explicitly if they are supposed to be used for filtering and querying.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-making-variant-call-statistics-accessible"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Making variant call statistics accessible</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2"><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+<span class="notranslate">galaxy</span>2</span></span> :
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGEMINI database‚Äù</em>: output of <strong>GEMINI load</strong></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDataset to use as the annotation source‚Äù</em>: output of <strong>VarScan somatic</strong></li>
        <li>
          <p><em>‚ÄúStrict variant-identity matching of database and annotation records
(VCF format only)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>This setting does not really matter here since you have built the GEMINI
database from the exact same list of variants that we are now retrieving
annotations from and because VarScan somatic does not call multiple alleles
at single sites. Matching on variant-identity is the behavior we would
like to see though, so we may as well be explicit about it.</p>
        </li>
        <li>
          <p><em>‚ÄúType of information to add to the database variants‚Äù</em>: <code class="language-plaintext highlighter-rouge">Specific values
extracted from matching records in the annotation source (extract)</code></p>

          <ul>
            <li>In <em>‚Äú1: Annotation extraction recipe‚Äù</em>:
              <ul>
                <li><em>‚ÄúElements to extract from the annotation source‚Äù</em>: <code class="language-plaintext highlighter-rouge">SS</code></li>
                <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>:
<code class="language-plaintext highlighter-rouge">somatic_status</code></li>
                <li><em>‚ÄúWhat type of data are you trying to extract?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Integer numbers</code></li>
                <li><em>‚ÄúIf multiple annotations are found for the same variant,
store ‚Ä¶‚Äú</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This is the recipe for extracting the VarScan-generated <em>SS</em> field and
adding it as a new column <em>somatic_status</em> to the GEMINI database.</p>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation extraction recipe‚Äù</em></li>
            <li>In <em>‚Äú2: Annotation extraction recipe‚Äù</em>:
              <ul>
                <li><em>‚ÄúElements to extract from the annotation source‚Äù</em>: <code class="language-plaintext highlighter-rouge">GPV</code></li>
                <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>:
 <code class="language-plaintext highlighter-rouge">germline_p</code></li>
                <li><em>‚ÄúWhat type of data are you trying to extract?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Numbers with
decimal precision</code></li>
                <li><em>‚ÄúIf multiple annotations are found for the same variant,
store ‚Ä¶‚Äú</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This is the recipe for extracting the VarScan-generated <em>GPV</em> field
and adding it as a new column <em>germline_p</em> to the GEMINI database.</p>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Annotation extraction recipe‚Äù</em></li>
            <li>In <em>‚Äú3: Annotation extraction recipe‚Äù</em>:
              <ul>
                <li><em>‚ÄúElements to extract from the annotation source‚Äù</em>: <code class="language-plaintext highlighter-rouge">SPV</code></li>
                <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>:
 <code class="language-plaintext highlighter-rouge">somatic_p</code></li>
                <li><em>‚ÄúWhat type of data are you trying to extract?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Numbers with
decimal precision</code></li>
                <li><em>‚ÄúIf multiple annotations are found for the same variant,
store ‚Ä¶‚Äú</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This is the recipe for extracting the VarScan-generated <em>SPV</em> field
and adding it as a new column <em>somatic_p</em> to the GEMINI database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Next, we are going to add additional annotations beyond the ones directly
obtainable through <em>GEMINI</em> or from the input VCF dataset. Specifically we
want to add:</p>

<ul>
  <li>
    <p>more information from <strong>dbSNP</strong></p>

    <p>As part of the database creation process, GEMINI already checks all
variants whether they occur in dbSNP and, if so, stores their dbSNP IDs.
Hence, we only need to extract some additional information of interest.</p>
  </li>
  <li>information from <strong>Cancer Hotspots</strong></li>
  <li>links to the <strong>CIViC</strong> database</li>
  <li>information from the <strong>Cancer Genome Interpreter (CGI)</strong></li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-adding-further-annotations"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Adding further annotations</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2"><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+<span class="notranslate">galaxy</span>2</span></span>  to add further annotations from <strong>dbSNP</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGEMINI database‚Äù</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDataset to use as the annotation source‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">dbsnp.b147.chr5_12_17.vcf</code></li>
        <li>
          <p><em>‚ÄúStrict variant-identity matching of database and annotation records
(VCF format only)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>dbSNP stores information about specific SNPs observed in human
populations and we would like to record if any exact same SNPs are among
our variants.</p>
        </li>
        <li><em>‚ÄúType of information to add to the database variants‚Äù</em>: <code class="language-plaintext highlighter-rouge">Specific
values extracted from matching records in the annotation source
(extract)</code>
          <ul>
            <li>In <em>‚Äú1: Annotation extraction recipe‚Äù</em>:
              <ul>
                <li><em>‚ÄúElements to extract from the annotation source‚Äù</em>: <code class="language-plaintext highlighter-rouge">SAO</code></li>
                <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>:
<code class="language-plaintext highlighter-rouge">rs_ss</code></li>
                <li><em>‚ÄúWhat type of data are you trying to extract?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Integer numbers</code></li>
                <li><em>‚ÄúIf multiple annotations are found for the same variant,
store ‚Ä¶‚Äú</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This recipe extracts the dbSNP <em>SAO</em> field and adds it as <em>rs_ss</em> to
the GEMINI database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2"><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+<span class="notranslate">galaxy</span>2</span></span>  to add further annotations from <strong>cancerhotspots</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGEMINI database‚Äù</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDataset to use as the annotation source‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">cancerhotspots_v2.<span class="notranslate">bed</span></code></li>
        <li>
          <p><em>‚ÄúStrict variant-identity matching of database and annotation records
(VCF format only)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (with input in <span class="notranslate">BED</span> format this setting will
be ignored)</p>

          <p>For the cancerhotspots data, we are simply going to record the best
q-value associated with any cancerhotspots variant overlapping one
of our variant sites.</p>
        </li>
        <li><em>‚ÄúType of information to add to the database variants‚Äù</em>: <code class="language-plaintext highlighter-rouge">Specific
values extracted from matching records in the annotation source
(extract)</code>
          <ul>
            <li>In <em>‚Äú1: Annotation extraction recipe‚Äù</em>:
              <ul>
                <li>
                  <p><em>‚ÄúElements to extract from the annotation source‚Äù</em>: <code class="language-plaintext highlighter-rouge">5</code></p>

                  <p>The q-values are stored in the fifth column of the <span class="notranslate">BED</span> dataset.</p>
                </li>
                <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>:
<code class="language-plaintext highlighter-rouge">hs_qvalue</code></li>
                <li><em>‚ÄúWhat type of data are you trying to extract?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Numbers with
 decimal precision</code></li>
                <li><em>‚ÄúIf multiple annotations are found for the same variant,
store ‚Ä¶‚Äú</em>: <code class="language-plaintext highlighter-rouge">the smallest of the (numeric) values</code></li>
              </ul>

              <p>This is the recipe for extracting the <em>q-values</em> of overlapping
cancerhotspots sites and adding them as <em>hs_qvalue</em> to the GEMINI
database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2"><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+<span class="notranslate">galaxy</span>2</span></span>  to add links to <strong>CIViC</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGEMINI database‚Äù</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDataset to use as the annotation source‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">CIViC.<span class="notranslate">bed</span></code></li>
        <li>
          <p><em>‚ÄúStrict variant-identity matching of database and annotation records
(VCF format only)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (with input in <span class="notranslate">BED</span> format this setting will
be ignored)</p>

          <p>For the CIViC data, we are going to record the hyperlink to any variant
found in the CIViC database that overlaps one of our variant sites.</p>
        </li>
        <li><em>‚ÄúType of information to add to the database variants‚Äù</em>: <code class="language-plaintext highlighter-rouge">Specific
values extracted from matching records in the annotation source
(extract)</code>
          <ul>
            <li>In <em>‚Äú1: Annotation extraction recipe‚Äù</em>:
              <ul>
                <li>
                  <p><em>‚ÄúElements to extract from the annotation source‚Äù</em>: <code class="language-plaintext highlighter-rouge">4</code></p>

                  <p>The hyperlinks are stored in the fourth column of the <span class="notranslate">BED</span> dataset.</p>
                </li>
                <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>:
<code class="language-plaintext highlighter-rouge">overlapping_civic_url</code></li>
                <li><em>‚ÄúWhat type of data are you trying to extract?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Text (text)</code></li>
                <li><em>‚ÄúIf multiple annotations are found for the same variant,
store ‚Ä¶‚Äú</em>: <code class="language-plaintext highlighter-rouge">a comma-separated list of non-redundant (text) values</code></li>
              </ul>

              <p>This is the recipe for extracting the hyperlinks of overlapping CIViC
sites and adding them as a list of <em>overlapping_civic_urls</em> to the
GEMINI database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2"><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+<span class="notranslate">galaxy</span>2</span></span>  to add information from the <strong>Cancer Genome Interpreter (CGI)</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGEMINI database‚Äù</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúDataset to use as the annotation source‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">cgi_variant_positions.<span class="notranslate">bed</span></code></li>
        <li>
          <p><em>‚ÄúStrict variant-identity matching of database and annotation records
(VCF format only)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (with input in <span class="notranslate">BED</span> format this setting will
be ignored)</p>

          <p>For the CGI data, we are going to record if any of our variant sites
is overlapped by a variant in the CGI biomarkers database.</p>
        </li>
        <li><em>‚ÄúType of information to add to the database variants‚Äù</em>: <code class="language-plaintext highlighter-rouge">Binary
indicator (1=found, 0=not found) of whether the variant had any match in
the annotation source (boolean)</code>
          <ul>
            <li><em>‚ÄúDatabase column name to use for recording annotations‚Äù</em>: <code class="language-plaintext highlighter-rouge">in_cgidb</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="reporting-selected-subsets-of-variants">Reporting selected subsets of variants</h2>

<p>Now that we have built our GEMINI database and enriched it with additional
annotations, it is time that we explore the wealth of information stored in it.
The goal is to produce filtered variant reports that list specific classes
(somatic, germline, LOH) of high-quality variants together with their most
relevant annotations. The way to achieve this is through <strong>GEMINI queries</strong>
that specify:</p>

<ol>
  <li>filters we want to apply to the variant list stored in the database</li>
  <li>the pieces of information about the filtered variants that we would like to
retrieve</li>
</ol>

<blockquote class="comment">
  <h3 id="comment-the-gemini-query-language"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> The GEMINI query language</h3>
  <p>GEMINI queries are extremely flexible, enabling users to express many
different ideas to explore the variant space, and as such, the complete
query syntax can be a bit overwhelming for beginners.</p>

  <p>In this section, we will start off with a rather simple query, then build on
it stepwise before trying a really complex query in the next section.</p>

  <p>For a more detailed explanation of the query syntax, you should consult the
<a href="https://gemini.readthedocs.io/en/latest/content/querying.html">query section of the GEMINI documentation</a>.
Since the GEMINI query syntax is built on the <em>SQLite</em> dialect of <em>SQL</em>, the
<em>SQLite</em> documentation, in particular, its chapter on
<a href="https://sqlite.org/lang_corefunc.html">SQLite core functions</a>, is another
really helpful resource for understanding more sophisticated queries.</p>
</blockquote>

<p>Lets start by configuring a simple query to obtain a report of <em>bona fide</em>
somatic variants. Our strategy for retrieving them is to:</p>

<ol>
  <li>rely on the <em>somatic status</em> of the variants called by <strong>VarScan somatic</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  <li>disregard questionable variants, for which either a non-negligible amount of
supporting sequencing <span class="notranslate">reads</span> is also found in the normal tissue data, or
which are only supported by a very small fraction of the <span class="notranslate">reads</span> from the
tumor sample</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-querying-the-gemini-database-for-somatic-variants"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Querying the GEMINI database for somatic variants</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1"><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+<span class="notranslate">galaxy</span>1</span></span>  with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGEMINI database‚Äù</em>: the fully annotated database
created in the last <strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step</li>
        <li><em>‚ÄúBuild GEMINI query using‚Äù</em>: <code class="language-plaintext highlighter-rouge">Basic variant query constructor</code>
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert Genotype filter <span class="notranslate">expression</span>‚Äù</em>
              <ul>
                <li><em>‚ÄúRestrictions to apply to genotype values‚Äù</em>:
<code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL &lt;= 0.05 AND gt_alt_freqs.TUMOR &gt;= 0.10</code></li>
              </ul>

              <p>With this genotype-based filter, we retain only those variants that
are supported by less than 5% of the <span class="notranslate">reads</span> of the normal sample, but
by more than 10% of the <span class="notranslate">reads</span> of the tumor sample.</p>
            </li>
            <li>
              <p><em>‚ÄúAdditional constraints expressed in SQL syntax‚Äù</em>: <code class="language-plaintext highlighter-rouge">somatic_status = 2</code></p>

              <p>Among the info stored in the GEMINI database is the somatic status
VarScan somatic has called for every variant (remember we
used GEMINI annotate to add it). With the condition
<code class="language-plaintext highlighter-rouge">somatic_status = 2</code> we retain only those variants passing the
genotype filter above <strong>and</strong> considered somatic variants by the
variant caller.</p>
            </li>
            <li>In <em>‚ÄúOutput format options‚Äù</em>
              <ul>
                <li><em>‚ÄúType of report to generate‚Äù</em>: <code class="language-plaintext highlighter-rouge">tabular (GEMINI default)</code>
                  <ul>
                    <li><em>‚ÄúAdd a header of column names to the output‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>‚ÄúSet of columns to include in the variant report table‚Äù</em>: <code class="language-plaintext highlighter-rouge">Custom
(report user-specified columns)</code>
                      <ul>
                        <li>In <em>‚ÄúChoose columns to include in the report‚Äù</em>:
                          <ul>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúchrom‚Äù</em></li>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äústart‚Äù</em></li>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúref‚Äù</em></li>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>‚Äúalt‚Äù</em></li>
                          </ul>
                        </li>
                        <li><em>‚ÄúAdditional columns (comma-separated)‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene, aa_change, rs_ids,
hs_qvalue, cosmic_ids</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>

              <p>Here we specify, which columns (from the <em>variants</em> table of the
GEMINI database) we want to have included, in the specified order, in
a <code class="language-plaintext highlighter-rouge">tabular</code> variant report.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-how-am-i-supposed-to-know-these-column-names"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> How am I supposed to know these column names?</h3>
  <p>Obviously, you need to know the names of the columns (in the tables of the
GEMINI database) to include them in the report, but how are you supposed to
know them?</p>

  <p>The standard ones (added by <em>GEMINI load</em> when building the database) are
listed in the
<a href="https://gemini.readthedocs.io/en/latest/content/database_schema.html">GEMINI documentation</a>.
The non-standard columns (the ones you added with <em>GEMINI annotate</em>) have
the names you gave them, when you added them.</p>

  <p>Alte<span class="notranslate">rna</span>tively, to get the tables and column names of a specific database
listed, you can use <strong>GEMINI database info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> like so:</p>

  <ul>
    <li><em>‚ÄúGEMINI database‚Äù</em>: the database you want to explore</li>
    <li><em>‚ÄúInformation to retrieve from the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">Names of database tables
and their columns</code></li>
  </ul>
</blockquote>

<p>What about more sophisticated filtering?</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-more-complex-filter-criteria"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: More complex filter criteria</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1"><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+<span class="notranslate">galaxy</span>1</span></span>  with the exact same settings as before, but:
      <ul>
        <li><em>‚ÄúAdditional constraints expressed in SQL syntax‚Äù</em>: <code class="language-plaintext highlighter-rouge">somatic_status = 2 AND somatic_p &lt;= 0.05 AND filter IS NULL</code></li>
      </ul>

      <p>This translates into ‚Äúvariants classified as somatic with a p-value &lt;=
 0.05, which haven‚Äôt been flagged as likely false-positives‚Äù.</p>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <h3 id="comment-sql-keywords"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> SQL keywords</h3>
  <p>In the condition above, SQL keywords are given in uppercase. This is not a
requirement, but it makes it easier to understand the syntax.</p>

  <p>You can check whether any cell in a data table is empty with <code class="language-plaintext highlighter-rouge">IS NULL</code>, and
whether it contains <em>any</em> value with <code class="language-plaintext highlighter-rouge">IS NOT NULL</code>. To combine different
filter criteria logically, you can use <code class="language-plaintext highlighter-rouge">AND</code> and <code class="language-plaintext highlighter-rouge">OR</code>, and parentheses to
group conditions if required.</p>
</blockquote>

<p>If you have followed all steps up to here exactly, running this job should give
you a tabular dataset of 43 variants, and with the annotations in the report
it is relatively easy to pick out a few interesting ones.
Before we focus on the content of the report, however, we could enhance the
report format a bit more.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-sql-based-output-formatting"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: SQL-based output formatting</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1"><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+<span class="notranslate">galaxy</span>1</span></span>  with the exact same settings as in the
last example, but:
      <ul>
        <li>In <em>‚ÄúOutput format options‚Äù</em>
          <ul>
            <li><em>‚ÄúAdditional columns (comma-separated)‚Äù</em>: <code class="language-plaintext highlighter-rouge">type, gt_alt_freqs.TUMOR, gt_alt_freqs.NORMAL,
ifnull(nullif(round(max_aaf_all,2),-1.0),0) AS MAF, gene, impact_so,
aa_change, ifnull(round(cadd_scaled,2),'.') AS cadd_scaled,
round(gerp_bp_score,2) AS gerp_bp, ifnull(round(gerp_element_pval,2),'.')
AS gerp_element_pval, ifnull(round(hs_qvalue,2), '.') AS hs_qvalue,
in_omim, ifnull(clinvar_sig,'.') AS clinvar_sig,
ifnull(clinvar_disease_name,'.') AS clinvar_disease_name,
ifnull(rs_ids,'.') AS dbsnp_ids, rs_ss, ifnull(cosmic_ids,'.') AS
cosmic_ids, ifnull(overlapping_civic_url,'.') AS overlapping_civic_url,
in_cgidb</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>This last query adds a lot more annotations to the report, and it also
demonstrates the use of the <code class="language-plaintext highlighter-rouge">AS</code> keyword to rename columns and of some
<a href="https://sqlite.org/lang_corefunc.html">SQLite functions</a> to clean up the
output.</p>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
  <p>Compare the new report to the previous one to see what has changed.</p>
</blockquote>

<blockquote class="details">
  <h3 id="details-limitations-of-genotype-column-queries"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Limitations of genotype column queries</h3>
  <p>In case you are wondering why the above query does not use rounding on
the alte<span class="notranslate">rna</span>te allele frequencies of the samples, <em>i.e.</em>, on
<code class="language-plaintext highlighter-rouge">gt_alt_freqs.TUMOR</code> and <code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL</code>, or why it does not rename
these columns, that is because it would break the query.</p>

  <p>As a general rule, note that all columns in the variants table starting
with <code class="language-plaintext highlighter-rouge">gt</code> (the genotype columns, calculated from the genotype fields of a
VCF dataset) cannot be used like regular SQLite columns, but are parsed by
GEMINI separately. That is why you cannot mix them with regular SQLite
elements like functions and alias specifications with <code class="language-plaintext highlighter-rouge">AS</code>.
You may also have noticed that <code class="language-plaintext highlighter-rouge">gt_alt_freqs.TUMOR</code> and
<code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL</code> do not obey the column order specification of the
query, but end up as the last columns of the tabular report. This is
another artefact of GEMINI‚Äôs special treatment of <code class="language-plaintext highlighter-rouge">gt</code> columns.</p>
</blockquote>

<h2 id="generating-reports-of-genes-affected-by-variants">Generating reports of genes affected by variants</h2>

<p>As a final step, let us now try to generate a gene-centered report based on the
same somatic variants we just selected above.</p>

<p>Such a gene-centered report would include annotations that apply to a whole
gene affected by a variant rather than to the variant itself. Examples of such
annotations include known synonyms of an affected gene, its NCBI entrez number,
the ClinVar phenotype, if any, associated with the gene, a hyperlink to the
gene‚Äôs page at CIViC.org, <em>etc.</em>.</p>

<p>Some of this information comes built-in into every GEMINI database, but it is
stored in a separate table called <code class="language-plaintext highlighter-rouge">gene_detailed</code>, while all information we
used and queried so far was from the <code class="language-plaintext highlighter-rouge">variants</code> table.</p>

<p>To access information from the <code class="language-plaintext highlighter-rouge">variants</code> and the <code class="language-plaintext highlighter-rouge">gene_detailed</code> table in the
same query we need to join the two tables. Such an operation is not possible
with the <code class="language-plaintext highlighter-rouge">Basic query constructor</code> we have used so far, but requires an
advanced mode for composing the query.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-turning-query-results-into-gene-centered-reports"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Turning query results into gene-centered reports</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="Tested with toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1"><strong>GEMINI query</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+<span class="notranslate">galaxy</span>1</span></span>  in advanced mode by choosing
      <ul>
        <li><em>‚ÄúBuild GEMINI query using‚Äù</em>: <code class="language-plaintext highlighter-rouge">Advanced query constructor</code></li>
        <li><em>‚ÄúThe query to be issued to the database‚Äù</em>: <code class="language-plaintext highlighter-rouge">SELECT v.gene, v.chrom,
 g.synonym, g.hgnc_id, g.entrez_id, g.rvis_pct, v.clinvar_gene_phenotype
 FROM variants v, gene_detailed g WHERE v.chrom = g.chrom AND
 v.gene = g.gene AND v.somatic_status = 2 AND v.somatic_p &lt;= 0.05 AND
 v.filter IS NULL GROUP BY g.gene</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-elements-of-the-sql-query"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Elements of the SQL query</h3>
        <p>In this full SQL query, the part between <code class="language-plaintext highlighter-rouge">SELECT</code> and <code class="language-plaintext highlighter-rouge">FROM</code> states which
columns from which database tables we wish to retrieve, while the part
between <code class="language-plaintext highlighter-rouge">FROM</code> and <code class="language-plaintext highlighter-rouge">WHERE</code> specifies the database tables that need to be
consulted and gives them simpler aliases (<code class="language-plaintext highlighter-rouge">v</code> becomes an alias for
the <code class="language-plaintext highlighter-rouge">variants</code> table, <code class="language-plaintext highlighter-rouge">g</code> for the <code class="language-plaintext highlighter-rouge">gene_detailed</code> table), which we can then
use everywhere else in the query.</p>

        <p>The part following <code class="language-plaintext highlighter-rouge">WHERE</code> are the filter criteria we want to apply. Note
that these criteria are almost the same as those we have used in our
earlier somatic variants query, but since we are working with two tables
instead of just one, we need to state which table the filter columns come
from through table prefixes. Thus, <code class="language-plaintext highlighter-rouge">somatic_status</code> becomes
<code class="language-plaintext highlighter-rouge">v.somatic_status</code>, <em>etc.</em>. In addition, we want to report, of course,
corresponding information from the two tables, which is ensured by the
additional criteria <code class="language-plaintext highlighter-rouge">v.chrom = g.chrom</code> and <code class="language-plaintext highlighter-rouge">v.gene = g.gene</code> (the SQL
terminology for this is: we want to join the <code class="language-plaintext highlighter-rouge">variants</code> and the
<code class="language-plaintext highlighter-rouge">gene_detailed</code> tables on their <code class="language-plaintext highlighter-rouge">chrom</code> and <code class="language-plaintext highlighter-rouge">gene</code> columns).</p>

        <p>The <code class="language-plaintext highlighter-rouge">GROUP BY</code> part, finally, specifies that we want to collapse records
affecting the same gene into one.</p>
      </blockquote>
    </li>
  </ol>

  <ul>
    <li>
      <p><em>‚ÄúGenotype filter <span class="notranslate">expression</span>‚Äù</em>: <code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL &lt;= 0.05 AND
gt_alt_freqs.TUMOR &gt;= 0.10</code></p>

      <p>This remains the same as in the previous somatic variants query.</p>
    </li>
  </ul>
</blockquote>

<h2 id="adding-additional-annotations-to-the-gene-centered-report">Adding additional annotations to the gene-centered report</h2>

<p>Unfortunately, <em>GEMINI annotate</em> lets you add columns only to the variants
table of a GEMINI database, but there is no simple way to enhance the
<code class="language-plaintext highlighter-rouge">gene_detailed</code> table with additional annotations. That‚Äôs why we are going to
add such extra annotations now to the tabular gene-centered report using more
general-purpose <span class="notranslate">Galaxy</span> tools.</p>

<p>Annotating the tabular gene report produced with GEMINI is a two-step
process, in which we first <em>join</em> the report and tabular annotation sources
into a larger tabular dataset, from which we then eliminate redundant and
unwanted columns, while <em>rearranging</em> the remaining ones.</p>

<p><strong>Step 1</strong> consists of three separate <em>join</em> operations that sequentially pull in the annotations found in the three gene-based tabular datasets that you imported in the <em>Get Data</em> step of this section.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-join"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Join</h3>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1"><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1</span></span>  to add <strong>UniProt cancer genes</strong> information
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1st file‚Äù</em>: the GEMINI-generated gene report from the previous step</li>
        <li><em>‚ÄúColumn to use from 1st file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span>  <em>‚Äú2nd file‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">Uniprot_Cancer_Genes</code> dataset</li>
        <li><em>‚ÄúColumn to use from 2nd file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li>
          <p><em>‚ÄúOutput lines appearing in‚Äù</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable
lines from 1st file. (-a 1)</code></p>

          <p>If a variant-affected gene is not listed as a Uniprot cancer gene, then,
obviously, we still want to have it included in the final report.</p>
        </li>
        <li><em>‚ÄúFirst line is a header line‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúIgnore case‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>
          <p><em>‚ÄúValue to put in unpaired (empty) fields‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></p>

          <p>If you inspect the <code class="language-plaintext highlighter-rouge">Uniprot_Cancer_Genes</code> dataset, you will see that it
has two annotation columns - one indicating, using <code class="language-plaintext highlighter-rouge">1</code> and <code class="language-plaintext highlighter-rouge">0</code>, whether
a given gene is a <em>proto-oncogene</em> or not, the other one indicating
<em>tumor suppressor</em> genes the same way. For genes missing from this
annotation dataset, we want to fill the corresponding two columns of the
join result with <code class="language-plaintext highlighter-rouge">0</code> to indicate the common case that a gene affected by
a variant is neither a known proto-oncogene, nor a tumor suppressor
gene.</p>
        </li>
      </ul>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1"><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1</span></span>  to add <strong>CGI biomarkers</strong> information
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1st file‚Äù</em>: the partially annotated dataset from the previous</li>
        <li><em>‚ÄúColumn to use from 1st file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú2nd file‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">cgi_genes</code> dataset</li>
        <li><em>‚ÄúColumn to use from 2nd file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><em>‚ÄúOutput lines appearing in‚Äù</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable
lines from 1st file. (-a 1)</code></li>
        <li><em>‚ÄúFirst line is a header line‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúIgnore case‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúValue to put in unpaired (empty) fields‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
      </ul>

      <p>Inspect the input and the result dataset to make sure you understand what
happened at this step.</p>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1"><strong>Join two files</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1</span></span>  to add gene information from <strong>CIViC</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú1st file‚Äù</em>: the partially annotated dataset from step 2</li>
        <li><em>‚ÄúColumn to use from 1st file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äú2nd file‚Äù</em>: the imported <code class="language-plaintext highlighter-rouge">GeneSummaries</code> dataset</li>
        <li>
          <p><em>‚ÄúColumn to use from 2nd file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Column: 3</code></p>

          <p>The gene column in the CIViC gene summaries annotation dataset is <em>not</em> the first one!</p>
        </li>
        <li><em>‚ÄúOutput lines appearing in‚Äù</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable
lines from 1st file. (-a 1)</code></li>
        <li><em>‚ÄúFirst line is a header line‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúIgnore case‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>‚ÄúValue to put in unpaired (empty) fields‚Äù</em>: <code class="language-plaintext highlighter-rouge">.</code></li>
      </ul>

      <p>Inspect the input and the result dataset to make sure you understand what
happened at this step.</p>
    </li>
  </ol>

</blockquote>

<p>If you took a look at all output datasets as suggested, you will have noticed
that each of the <em>join</em> operations kept the gene columns from both input
datasets. In addition, we had no control over the order, in which columns got
added to the report, nor could we exclude columns.</p>

<p>In <strong>Step 2</strong> of the report preparation we are going to address all of these
issues and rearrange to get a fully annotated gene report.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-rearrange-to-get-a-fully-annotated-gene-report"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Rearrange to get a fully annotated gene report</h3>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/column_arrange_by_header/bg_column_arrange_by_header/0.2" title="Tested with toolshed.g2.bx.psu.edu/repos/bgruening/column_arrange_by_header/bg_column_arrange_by_header/0.2"><strong>Column arrange by header name</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/bgruening/column_arrange_by_header/bg_column_arrange_by_header/0.2</span></span>  configured like
this:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äúfile to rearrange‚Äù</em>: the final <strong>Join</strong> result dataset from step 3</li>
        <li>In <em>‚ÄúSpecify the first few columns by name‚Äù</em>
          <ul>
            <li>In <em>‚Äú1: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú2: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">chrom</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú3: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">synonym</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú4: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">hgnc_id</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú5: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">entrez_id</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú6: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">rvis_pct</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú7: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">is_OG</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú8: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">is_TS</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú9: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">in_cgi_biomarkers</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú10: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">clinvar_gene_phenotype</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú11: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">gene_civic_url</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúSpecify the first few columns by name‚Äù</em></li>
            <li>In <em>‚Äú12: Specify the first few columns by name‚Äù</em>
              <ul>
                <li><em>‚Äúcolumn‚Äù</em>: <code class="language-plaintext highlighter-rouge">description</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúDiscard unspecified columns‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <h3 id="comment-alternative-tool-suggestion"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Alte<span class="notranslate">rna</span>tive tool suggestion</h3>
    <p>If your <span class="notranslate">Galaxy</span> server does not offer the <em>Column arrange</em> tool, it will
almost certainly offer <strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>,
which can be used as a drop-in replacement.
Instead of column names, however, this tool expects a comma-separated list
of column indexes, like <code class="language-plaintext highlighter-rouge">c1,c2</code> for the first and second column, so you
will have to first figure out the column numbers of your columns of
interest.</p>

    <p><strong>Note</strong>: Be sure not to confuse the suggested tool with <em>Cut
columns from a table (cut)</em>, which does <em>not</em> let you change the order of
columns!</p>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In addition to merely calling variants, <em>somatic variant calling</em> tries to
distinguish <em>somatic mutations</em>, which are private to tumor tissue, from
<em>germline</em> mutations, that are shared by tumor and healthy tissue, and
<em>loss-of-heterozygosity</em> events, which involve the loss, from tumor tissue, of
one of two alleles found at a biallelic site in healthy tissue.</p>

<p>Dedicated somatic variant callers can perform this classification on
statistical grounds, but the interpretation of any list of variants (somatic,
germline or LOH) also depends crucially on rich genetic and cancer-specific
variant and gene annotations.</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Follow best practices for read mapping, quality control and mapped reads postprocessing to minimize false-positive variant calls.</p>
</li>
                            
                            <li><p>Use a dedicated somatic variant caller to call variants and to classify them into somatic, germline and LOH event variants on statistical grounds.</p>
</li>
                            
                            <li><p>Annotations and queries based on variant properties add relevance to variant and gene reports.</p>
</li>
                            
                            <li><p>A framework like GEMINI is very helpful for managing, annotating and querying lists of variants in a flexible way.</p>
</li>
                            
                            <li><p>Prefer public, free annotation sources to foster reproducibility and information sharing.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Identification of somatic and germline variants from tumor and normal sample pairs (Variant Analysis)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Wolfgang Maier, 2021 <b>Identification of somatic and germline variants from tumor and normal sample pairs (Galaxy Training Materials)</b>. <a href="/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{variant-analysis-somatic-variants,
    author = "Wolfgang Maier",
    title = "Identification of somatic and germline variants from tumor and normal sample pairs (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Wolfgang Maier)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/somatic-variants/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
