<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Hands-on: Identification of somatic and germline variants from tumor and normal sample pairs / Identification of somatic and germline variants from tumor and normal sample pairs / Variant Analysis</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@4bbac78b4791e7e0a5f3fcee9e506ed0d1e4d691",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.20cf1f17.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Identification of somatic and germline variants from tumor and normal sample pairs">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-11-09 08:55:09 +0000">
<meta name="DC.creator" content="Wolfgang Maier"><meta name="description" content="Genetic differences (variants) between healthy and diseased tissue, between individuals of a population, or between strains of an organism can provide mechanistic insight into disease processes and the natural function of affected genes.  The tutorials in this section show how to detect evidence for genetic variants in next-generation sequencing data, a process termed variant calling.  Of equal importance, they also demonstrate how you can interpret, for a range of different organisms, the resulting sets of variants by predicting their molecular effects on genes and proteins, by annotating previously observed variants with published knowledge, and by trying to link phenotypes of the sequenced samples to their variant genotypes. ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Variant Analysis / Identification of somatic and germline variants from tumor and normal sample pairs / Hands-on: Identification of somatic and germline variants from tumor and normal sample pairs">
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseased tissue, between individuals of a population, or between strains of an organism can provide mechanistic insight into disease processes and the natural function of affected genes.  The tutorials in this section show how to detect evidence for genetic variants in next-generation sequencing data, a process termed variant calling.  Of equal importance, they also demonstrate how you can interpret, for a range of different organisms, the resulting sets of variants by predicting their molecular effects on genes and proteins, by annotating previously observed variants with published knowledge, and by trying to link phenotypes of the sequenced samples to their variant genotypes. ">
        
	<meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"><script type="application/ld+json">


{
  "@context": "http://schema.org",
  "@type": "LearningResource",
  "http://purl.org/dc/terms/conformsTo": {
    "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE",
    "@type": "CreativeWork"
  },
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "Students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Identification of somatic and germline variants from tumor and normal sample pairs",
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "isFamilyFriendly": true,
  "license": "https://spdx.org/licenses/CC-BY-4.0.html",
  "producer": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "provider": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "http://purl.org/dc/terms/conformsTo": {
      "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
      "@type": "Organization"
    },
    "id": "https://training.galaxyproject.org",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "legalName": "Galaxy Training Network",
    "alternateName": "GTN",
    "url": "https://training.galaxyproject.org",
    "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
    "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
    "keywords": [
      "galaxy",
      "bioinformatics",
      "training",
      "fair",
      "accessible"
    ],
    "status": "active",
    "foundingDate": "2015-06-29",
    "socialMedia": "https://mstdn.science/@gtn",
    "type": "project"
  },
  "workTranslation": [

  ],
  "creativeWorkStatus": "Active",
  "dateModified": "2023-11-09 08:55:09 +0000",
  "datePublished": "2019-03-13 10:31:31 +0000",
  "identifier": "https://gxy.io/GTN:T00318",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "The text aims to be as accessible as possible. Image descriptions will vary per tutorial, from images being completely inaccessible, to images with good descriptions for non-visual users.",
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
  },
  "learningResourceType": "hands-on tutorial",
  "name": "Identification of somatic and germline variants from tumor and normal sample pairs",
  "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html",
  "version": 15,
  "timeRequired": "PT7H",
  "teaches": "Call variants and their somatic status from whole-exome sequencing data\n - Annotate variants with a wealth of human genetic and cancer-specific information extracted from public databases\n - Add gene-level annotations and generate reports of annotated somatic and germline variants, loss-of-heterozygosity (LOH) events, and affected genes, ready for interpretation by clinicians",
  "keywords": [
    "Variant Analysis"
  ],
  "description": "The questions this  addresses are:\n - What are the specific challenges in somatic variant calling that set it apart from regular diploid variant calling?\n - How can you call variants and classify them according to their presence/absence in/from tumor and normal tissue of the same individual?\n - How can you annotate variants and affected genes with prior knowledge from human genetic and cancer-specific databases to generate clinically relevant reports?\n\n\nThe objectives are:\n - Call variants and their somatic status from whole-exome sequencing data\n - Annotate variants with a wealth of human genetic and cancer-specific information extracted from public databases\n - Add gene-level annotations and generate reports of annotated somatic and germline variants, loss-of-heterozygosity (LOH) events, and affected genes, ready for interpretation by clinicians\n\n",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "competencyRequired": [
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    },
    {
      "@context": "http://schema.org",
      "@type": "LearningResource",
      "url": "https://training.galaxyproject.org/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "http://purl.org/dc/terms/conformsTo": {
          "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
          "@type": "Organization"
        },
        "id": "https://training.galaxyproject.org",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "legalName": "Galaxy Training Network",
        "alternateName": "GTN",
        "url": "https://training.galaxyproject.org",
        "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
        "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
        "keywords": [
          "galaxy",
          "bioinformatics",
          "training",
          "fair",
          "accessible"
        ],
        "status": "active",
        "foundingDate": "2015-06-29",
        "socialMedia": "https://mstdn.science/@gtn",
        "type": "project"
      }
    }
  ],
  "author": [
    {
      "@context": "https://schema.org",
      "@type": "Person",
      "http://purl.org/dc/terms/conformsTo": {
        "@id": "https://bioschemas.org/profiles/Person/0.3-DRAFT",
        "@type": "CreativeWork"
      },
      "url": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "mainEntityOfPage": "https://training.galaxyproject.org/training-material/hall-of-fame/wm75/",
      "name": "Wolfgang Maier",
      "image": "https://avatars.githubusercontent.com/wm75",
      "description": "A contributor to the GTN project.",
      "memberOf": [
        {
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.2-DRAFT-2019_07_19",
            "@type": "Organization"
          },
          "id": "https://training.galaxyproject.org",
          "email": "galaxytrainingnetwork@gmail.com",
          "name": "Galaxy Training Network",
          "legalName": "Galaxy Training Network",
          "alternateName": "GTN",
          "url": "https://training.galaxyproject.org",
          "logo": "https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png",
          "fundingModel": "The GTN's infrastructure relies on GitHub and the Galaxy Project for hosting costs. There are no full time paid staff members of the GTN. Individuals are occasionally funded on GTN-adjacent projects.",
          "keywords": [
            "galaxy",
            "bioinformatics",
            "training",
            "fair",
            "accessible"
          ],
          "status": "active",
          "foundingDate": "2015-06-29",
          "socialMedia": "https://mstdn.science/@gtn",
          "type": "project"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/by-covid/",
          "name": "BeYond-COVID",
          "description": "BY-COVID is an EC funded project that tackles the data challenges that can hinder effective pandemic response.\n\nThis project has received funding from the European Union’s Horizon Europe research and innovation programme under grant agreement № 101046203 (BY-COVID)\n",
          "url": "https://by-covid.org/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/uni-freiburg/",
          "name": "University of Freiburg",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://www.uni-freiburg.de/"
        },
        {
          "@context": "https://schema.org",
          "@type": "Organization",
          "http://purl.org/dc/terms/conformsTo": {
            "@id": "https://bioschemas.org/profiles/Organization/0.3-DRAFT",
            "@type": "CreativeWork"
          },
          "id": "https://training.galaxyproject.org/training-material/hall-of-fame/elixir-europe/",
          "name": "ELIXIR Europe",
          "description": "An organization supporting the Galaxy Training Network",
          "url": "https://elixir-europe.org"
        }
      ],
      "identifier": "https://orcid.org/0000-0002-9464-6640",
      "orcid": "https://orcid.org/0000-0002-9464-6640"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ],
  "educationalLevel": "Beginner",
  "mentions": [
    {
      "@type": "Thing",
      "url": "https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/workflows/",
      "name": "Associated Workflows"
    },
    {
      "@type": "Thing",
      "url": "https://doi.org/10.5281/zenodo.2582555",
      "name": "Associated Training Datasets"
    }
  ],
  "abstract": "When sequencing genomic material from a human tumor, the underlying clinical or"
}</script></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i> Variant Analysis
                        </a>
                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/variant-analysis/faqs/" title="Check our FAQs for the Variant Analysis topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/somatic-variants/tutorial.md">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            
            
                <a class="dropdown-item" href="/training-material/feedback.html">
                    <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
                </a>
            
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        














<!-- Gitter -->






<article class="tutorial topic-variant-analysis">
    <h1 data-toc-skip>Identification of somatic and germline variants from tumor and normal sample pairs</h1>
    

    <section aria-labelledby="overview-box" id="tutorial-metadata">
    <div markdown="0">

	<div class="contributors-line">
		Authors: <a href="/training-material/hall-of-fame/wm75/" class="contributor-badge contributor-wm75"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/wm75?s=36" alt="Wolfgang Maier avatar" width="36" class="avatar" />
    Wolfgang Maier</a>
	</div>

</div>


    <blockquote class="overview">
        <div id="overview-box" class="box-title">Overview</div>
        
        <img alt="Creative Commons License: CC-BY" class="float-right" style="border-width:0; display: inline-block; margin:0" src="/training-material/assets/images/cc-by.png" width="88" height="31"/>
        
        <strong><i class="far fa-question-circle" aria-hidden="true"></i> Questions:</strong>
        <ul>
        
        <li><p>What are the specific challenges in somatic variant calling that set it apart from regular diploid variant calling?</p>
</li>
        
        <li><p>How can you call variants and classify them according to their presence/absence in/from tumor and normal tissue of the same individual?</p>
</li>
        
        <li><p>How can you annotate variants and affected genes with prior knowledge from human genetic and cancer-specific databases to generate clinically relevant reports?</p>
</li>
        
        </ul>

        <strong><i class="fas fa-bullseye" aria-hidden="true"></i> Objectives: </strong>
        <ul>
        
        <li><p>Call variants and their somatic status from whole-exome sequencing data</p>
</li>
        
        <li><p>Annotate variants with a wealth of human genetic and cancer-specific information extracted from public databases</p>
</li>
        
        <li><p>Add gene-level annotations and generate reports of annotated somatic and germline variants, loss-of-heterozygosity (LOH) events, and affected genes, ready for interpretation by clinicians</p>
</li>
        
        </ul>

        
        <strong><i class="fas fa-check-circle" aria-hidden="true"></i> Requirements:</strong>
        <ul>
        
    
        
        
        
        <li>
          <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        </li>
        
    

    
        
        
        
            
                
                    
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Quality Control</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Quality Control</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
            
                
                    
                
                    
                
                    
                
                    
                
                    
                        
                            <li>
                              <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides: Mapping</a>
                            </li>
                        
                        
                            
                                <li>
                                  <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> Hands-on: Mapping</a>
                                </li>
                            
                        
                    
                
                    
                
                    
                
                    
                
            
        
    


        

        </ul>
        

        
        <div><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i> Time estimation:</strong> 7 hours</div>
        

        

        

        

        <div id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials:</strong></div>
        <ul class="supporting_material">
            

            
                <li class="btn btn-default supporting_material">


<a class="btn btn-default topic-icon" title="Zenodo datasets used in this tutorial" href="https://doi.org/10.5281/zenodo.2582555">
    <i class="far fa-copy" aria-hidden="true"></i>&nbsp;Datasets
</a>

</li>
            

            
                <li class="btn btn-default supporting_material">


    <a class="btn btn-default topic-icon" href="/training-material/topics/variant-analysis/tutorials/somatic-variants/workflows/" title="Identification of somatic and germline variants from tumor and normal sample pairs workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i>&nbsp;Workflows
    </a>

</li>
            

            

            

            

            

            
            
            
                <li class="btn btn-default supporting_material">    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/somatic-variants/faqs/" title="Frequently Asked Questions">
        <i class="far fa-question-circle" aria-hidden="true"></i> FAQs
    </a>
</li>
            

            <!-- Check the GTN Video Library for recordings of this tutorial or associated slides -->
            












            
                
                <li class="btn btn-default supporting_material">






    <a href="#" class="dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span>&nbsp;Available on these Galaxies 
    </a>
    <ul class="dropdown-menu">
        
	<li class="dropdown-header">
		<b>Known Working</b>
	</li>
	
	

    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.eu" title="">
			UseGalaxy.eu <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org" title="">
			UseGalaxy.org (Main) <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.org.au" title="">
			UseGalaxy.org.au <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr> <abbr title="This is a UseGalaxy.* server which meets minimum requirements for a public Galaxy">⭐️</abbr>
		</a>
	</li>
    
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.no/" title="">
			UseGalaxy.no <abbr title="This instance supports the precise tool versions used in this tutorial">✅</abbr>
		</a>
	</li>
    
    
    
	<li class="dropdown-header">
		<b>Possibly Working</b>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.cz/" title="">
			UseGalaxy.cz
		</a>
	</li>
    
	<li>
		<a class="dropdown-item" href="https://usegalaxy.fr/" title="">
			UseGalaxy.fr
		</a>
	</li>
    
    

    
	<li class="dropdown-header">
		Containers
	</li>
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/main/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    </ul>

</li>
                
            
        </ul>

        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Published:</strong> Mar 13, 2019 </div>
        <div><strong><i class="far fa-calendar" aria-hidden="true"></i> Last modification:</strong> Nov 9, 2023 </div>
        <div><strong><i class="fas fa-balance-scale" aria-hidden="true"></i> License:</strong>
		
            Tutorial Content is licensed under
            
              <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
            
            The GTN Framework is licensed under <a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a>
        </div>
        
        <div><strong><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span> <abbr title="Persistent URL">PURL</abbr>:</strong> <a href="https://gxy.io/GTN:T00318">https://gxy.io/GTN:T00318</a> </div>
        

	
	

	
	
	<div><strong><i class="far fa-star" aria-hidden="true"></i><span class="visually-hidden">rating</span> Rating:</strong> <a href="#feedback-responses">5.0</a> (1 recent ratings, 13 all time)</div>
	
	<div><strong><i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span> Revision:</strong> 15 </div>

    </blockquote>
    </section>

    <div class="container">
        <div class="row">
            <!-- sidebar, which will move to the top on a small screen -->
            <div class="col-sm-2 hide-when-printing">
                <nav id="toc" data-toggle="toc" class="sticky-top" aria-label="Table of Contents"></nav>
            </div>
            <div class="col-sm-10">
                 

                <section aria-label="Tutorial Content" id="tutorial-content">
                <p>When sequencing genomic material from a human tumor, the underlying clinical or
research question typically is <strong>what spectrum of mutations distinguishes this
tumor from healthy tissue</strong>.</p>

<p>This question cannot be answered adequately just by comparing the tumor tissue
to the human reference genome because even in healthy tissue there will be many
thousands of variants compared to the reference genome. This is because every
individual inherits a unique pattern of that many variants from her parents. A
fundamental difference between these variants and the tumor-specific mutations
is that the former are present in the carrier’s germline, while the latter have
been acquired somatically and will, thus, not be transmitted to offspring.
Therefor, we talk of germline variants to refer to variants present in healthy
and tumor tissue alike, and of somatic variants to refer to tumor-specific
variants. To be able to distinguish between these two types of variants always
requires a direct comparison of data from tumor and normal tissue samples.</p>

<p>In addition to acquiring new variants, tumors can also lose or gain chromosomal
copies of variants found heterozygously in an individual’s germline. This
phenomenon is termed <em>loss of heterozygosity</em> (LOH) because only one of the two
original alleles persists in the tumor (either in a hemizygous state if the
other allele is simply dropped, or in a homozygous state in the case of a
duplication of one allelic copy accompanied by loss of the other).
The detection of LOH events, again, is dependent on a comparison of tumor and
normal tissue data.</p>

<p>In this tutorial we are going to identify somatic and germline variants, as
well as variants affected by LOH, from a tumor and a normal sample of the same
patient. Our goal is to report the variant sites, and the genes affected by
them, annotated with the content of general human genetic and cancer-specific
databases. Ideally, this may provide insight into the genetic events driving
tumor formation and growth in the patient, and might be of prognostic and even
therapeutic value by revealing variants known to affect drug
resistance/sensitivity, tumor aggressiveness, <em>etc</em>.</p>

<blockquote class="agenda">
  <div class="box-title agenda-title" id="agenda">Agenda</div>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a>    <ol>
      <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
    </ol>
  </li>
  <li><a href="#quality-control-and-mapping-of-ngs-reads" id="markdown-toc-quality-control-and-mapping-of-ngs-reads">Quality control and mapping of NGS reads</a>    <ol>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
      <li><a href="#read-trimming-and-filtering" id="markdown-toc-read-trimming-and-filtering">Read trimming and filtering</a></li>
      <li><a href="#read-mapping" id="markdown-toc-read-mapping">Read Mapping</a></li>
    </ol>
  </li>
  <li><a href="#mapped-reads-postprocessing" id="markdown-toc-mapped-reads-postprocessing">Mapped reads postprocessing</a>    <ol>
      <li><a href="#filtering-on-mapped-reads-properties" id="markdown-toc-filtering-on-mapped-reads-properties">Filtering on mapped reads properties</a></li>
      <li><a href="#removing-duplicate-reads" id="markdown-toc-removing-duplicate-reads">Removing duplicate reads</a></li>
      <li><a href="#left-align-reads-around-indels" id="markdown-toc-left-align-reads-around-indels">Left-align reads around indels</a></li>
      <li><a href="#recalibrate-read-mapping-qualities" id="markdown-toc-recalibrate-read-mapping-qualities">Recalibrate read mapping qualities</a></li>
      <li><a href="#refilter-reads-based-on-mapping-quality" id="markdown-toc-refilter-reads-based-on-mapping-quality">Refilter reads based on mapping quality</a></li>
    </ol>
  </li>
  <li><a href="#variant-calling-and-classification" id="markdown-toc-variant-calling-and-classification">Variant calling and classification</a></li>
  <li><a href="#variant-annotation-and-reporting" id="markdown-toc-variant-annotation-and-reporting">Variant annotation and reporting</a>    <ol>
      <li><a href="#get-data-1" id="markdown-toc-get-data-1">Get data</a></li>
      <li><a href="#adding-annotations-to-the-called-variants" id="markdown-toc-adding-annotations-to-the-called-variants">Adding annotations to the called variants</a></li>
      <li><a href="#reporting-selected-subsets-of-variants" id="markdown-toc-reporting-selected-subsets-of-variants">Reporting selected subsets of variants</a></li>
      <li><a href="#generating-reports-of-genes-affected-by-variants" id="markdown-toc-generating-reports-of-genes-affected-by-variants">Generating reports of genes affected by variants</a></li>
      <li><a href="#adding-additional-annotations-to-the-gene-centered-report" id="markdown-toc-adding-additional-annotations-to-the-gene-centered-report">Adding additional annotations to the gene-centered report</a></li>
    </ol>
  </li>
  <li><a href="#conclusion" id="markdown-toc-conclusion">Conclusion</a></li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>First we need to upload and prepare some input data to analyze. The sequencing
reads we are going to analyze are from real-world data from a cancer patient’s
tumor and normal tissue samples.
For the sake of an acceptable speed of the analysis, the original data has been
downsampled though to include only the reads from human chromosomes 5, 12 and
17.</p>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-creating-a-new-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-creating-a-new-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Creating a new history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>To create a new history simply click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel:</p>   <p><img src="/training-material/shared/images/history_create_new.svg" alt="UI for creating new history" /></p>   <!-- the original drawing can be found here https://docs.google.com/drawings/d/1cCBrLAo4kDGic5QyB70rRiWJAKTenTU8STsKDaLcVU8/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-history"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-history" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a history</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which by default is “Unnamed history”)</li>   <li>Type the new name</li>   <li>Click on <strong>Save</strong></li>   <li>To cancel renaming, click the <i class="fas fa-undo" aria-hidden="true"></i><span class="visually-hidden">galaxy-undo</span> “Cancel” button</li> </ol>   <p>If you do not have the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> (<strong>Edit</strong>) next to the history name (which can be the case if you are using an older version of Galaxy) do the following:</p>   <ol>   <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>   <li>Type the new name</li>   <li>Press <kbd>Enter</kbd></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Import the following four files from
<a href="https://zenodo.org/record/2582555">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-N_231335_r2_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r1_chr5_12_17.fastq.gz
https://zenodo.org/record/2582555/files/SLGFSK-T_231336_r2_chr5_12_17.fastq.gz
</code></pre></div>      </div>
      <p>where the first two files represent the forward and reverse reads sequence
data from a patient’s <strong>normal</strong> tissue, and the last two represent the
data of the same patient’s <strong>tumor</strong> tissue.</p>

      <p>Alternatively, the same files may be available on your Galaxy server
through a shared data library (your instructor may tell you so), in which
case you may prefer to import the data directly from there.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-via-links"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-via-links" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing via links</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Copy the link location</li>   <li>     <p>Click <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Upload Data</strong> at the top of the tool panel</p>   </li>   <li>Select <i class="fa fa-edit" aria-hidden="true"></i><span class="visually-hidden">galaxy-wf-edit</span> <strong>Paste/Fetch Data</strong></li>   <li>     <p>Paste the link(s) into the text field</p>   </li>   <li>     <p>Change <strong>Type (set all):</strong> from “Auto-detect” to <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code></p>   </li>   <li>     <p>Press <strong>Start</strong></p>   </li>   <li><strong>Close</strong> the window</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-importing-data-from-a-data-library"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-importing-data-from-a-data-library" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Importing data from a data library</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>As an alternative to uploading the data from a URL or your computer, the files may also have been made available from a <em>shared data library</em>:</p>   <ol>   <li>Go into <strong>Data</strong> (top panel) then <strong>Data libraries</strong></li>   <li>Navigate to  the correct folder as indicated by your instructor.     <ul>       <li>On most Galaxies tutorial data will be provided in a folder named <strong>GTN - Material –&gt; Topic Name -&gt; Tutorial Name</strong>.</li>     </ul>   </li>   <li>Select the desired files</li>   <li>Click on <strong>Add to History</strong> <i class="fas fa-caret-down" aria-hidden="true"></i><span class="visually-hidden">galaxy-dropdown</span> near the top and select <strong>as Datasets</strong> from the dropdown menu</li>   <li>     <p>In the pop-up window, choose</p>     <ul>       <li><em>“Select history”</em>: the history you want to import the data to (or create a new one)</li>     </ul>   </li>   <li>Click on <strong>Import</strong></li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Check that all newly created datasets in your history have their datatypes assigned 
correctly, and fix any missing or wrong datatype assignment</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-changing-the-datatype"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-changing-the-datatype" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Changing the datatype</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, click <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Datatypes</strong> tab on the top</li>   <li>In the <i class="fas fa-database" aria-hidden="true"></i><span class="visually-hidden">galaxy-chart-select-data</span> <strong>Assign Datatype</strong>, select <code class="language-plaintext highlighter-rouge">fastqsanger.gz</code> from “<em>New type</em>” dropdown     <ul>       <li>Tip: you can start typing the datatype into the field to filter the dropdown menu</li>     </ul>   </li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>
    </li>
    <li>
      <p>Rename the datasets and add appropriate tags to them</p>

      <p>For datasets that you upload via a link, Galaxy will pick the link
address as the dataset name, which you will likely want to shorten to
just the file names.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-renaming-a-dataset"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-renaming-a-dataset" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Renaming a dataset</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ul>   <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">galaxy-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>   <li>In the central panel, change the <strong>Name</strong> field</li>   <li>Click the <strong>Save</strong> button</li> </ul> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>Large parts of the analysis in this tutorial will consist of
identical steps performed on the normal and on the tumor tissue data
in parallel.</p>

      <p>To make it easier to keep track of which dataset represents which
branch of an analysis in a linear history, Galaxy supports dataset
tags. In particular, if you attach a tag starting with <code class="language-plaintext highlighter-rouge">#</code> to any
dataset, that tag will automatically propagate to any new dataset
derived from the tagged dataset.</p>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-adding-a-tag"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-adding-a-tag" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Adding a tag</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <p>Datasets can be tagged. This simplifies the tracking of datasets across the Galaxy interface. Tags can contain any combination of letters or numbers but cannot contain spaces.</p>   <p><strong>To tag a dataset</strong>:</p>   <ol>   <li>Click on the dataset to expand it</li>   <li>Click on <strong>Add Tags</strong> <i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span></li>   <li>Add  tag text. Tags starting with <code class="language-plaintext highlighter-rouge">#</code> will be automatically propagated to the outputs of tools using this dataset (see below).</li>   <li>Press <kbd>Enter</kbd></li>   <li>Check that the tag appears below the dataset name</li> </ol>   <p><strong>Tags beginning with <code class="language-plaintext highlighter-rouge">#</code> are special!</strong></p>   <p>They are called <strong>Name tags</strong>. The unique feature of these tags is that they <em>propagate</em>: if a dataset is labelled with a name tag, all derivatives (children) of this dataset will automatically inherit this tag (see below). The figure below explains why this is so useful. Consider the following analysis (numbers in parenthesis correspond to dataset numbers in the figure below):</p>   <ol>   <li>a set of forward and reverse reads (datasets 1 and 2) is mapped against a reference using <span><strong>Bowtie2</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating dataset 3;</li>   <li>dataset 3 is used to calculate read coverage using <span><strong>BedTools Genome Coverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> <em>separately</em> for <code class="language-plaintext highlighter-rouge">+</code> and <code class="language-plaintext highlighter-rouge">-</code> strands. This generates two datasets (4 and 5 for plus and minus, respectively);</li>   <li>datasets 4 and 5 are used as inputs to <span><strong>Macs2 broadCall</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> datasets generating datasets 6 and 8;</li>   <li>datasets 6 and 8 are intersected with coordinates of genes (dataset 9) using <span><strong>BedTools Intersect</strong> <i class="fas fa-wrench" aria-hidden="true"></i></span> generating datasets 10 and 11.</li> </ol>   <p><img src="/training-material/shared/images/histories_why_nametags.svg" alt="A history without name tags versus history with name tags" /></p>   <p>Now consider that this analysis is done without name tags. This is shown on the left side of the figure. It is hard to trace which datasets contain “plus” data versus “minus” data. For example, does dataset 10 contain “plus” data or “minus” data? Probably “minus” but are you sure? In the case of a small history like the one shown here, it is possible to trace this manually but as the size of a history grows it will become very challenging.</p>   <p>The right side of the figure shows exactly the same analysis, but using name tags. When the analysis was conducted datasets 4 and 5 were tagged with <code class="language-plaintext highlighter-rouge">#plus</code> and <code class="language-plaintext highlighter-rouge">#minus</code>, respectively. When they were used as inputs to Macs2 resulting datasets 6 and 8 automatically inherited them and so on… As a result it is straightforward to trace both branches (plus and minus) of this analysis.</p>   <p>More information is in a <a href="/training-material/topics/galaxy-interface/tutorials/name-tags/tutorial.html">dedicated #nametag tutorial</a>.</p>   <!-- Image is here = https://docs.google.com/drawings/d/1iiNsau6ddiE2MV9qMyekUq2mrpDHHcc02bXtcFEAnhY/edit?usp=sharing --> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>Before starting our analysis it is, thus, a good idea to tag the two
fastq datasets representing the normal tissue with, <em>e.g.</em>, <code class="language-plaintext highlighter-rouge">#normal</code>
and the two datasets representing the tumor tissue with, <em>e.g.</em>,
<code class="language-plaintext highlighter-rouge">#tumor</code>.</p>
    </li>
    <li>
      <p>Import the reference genome with the <code class="language-plaintext highlighter-rouge">hg19</code> version of the sequences of
human chromosomes 5, 12 and 17:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2582555/files/hg19.chr5_12_17.fa.gz
</code></pre></div>      </div>

      <p>Make sure you specify the datatype as <code class="language-plaintext highlighter-rouge">fasta</code> in the import dialog.</p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-shortcut"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Shortcut</div>
        <p>You can skip this step if the Galaxy server you are working on offers
a <code class="language-plaintext highlighter-rouge">hg19</code> version of the human reference genome with prebuilt indexes for
<em>bwa-mem</em> <strong>and</strong> <em>samtools</em> (ask your instructor, or check the tools
<strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and <strong>VarScan Somatic</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> if they list a <code class="language-plaintext highlighter-rouge">hg19</code> version as an option under
<em>“(Using) reference genome”</em>).</p>
      </blockquote>

      <p>Alternatively, load the dataset from a shared data library.</p>
    </li>
    <li>
      <p>Rename the reference genome</p>

      <p>The reference genome you have imported above came as a compressed
file, but got unpacked by Galaxy to plain <code class="language-plaintext highlighter-rouge">fasta</code> format according to
your datatype selection. You may now wish to remove the <code class="language-plaintext highlighter-rouge">.gz</code> suffix
from the dataset name.</p>
    </li>
  </ol>

</blockquote>

<h1 id="quality-control-and-mapping-of-ngs-reads">Quality control and mapping of NGS reads</h1>

<p>Before starting our analysis, we would like to make sure that the input data
is of good quality, <em>i.e.</em>, that there haven’t been any major issues during
DNA preparation, exon capture, or during actual sequencing. To avoid spurious
variant calls due to low input quality, we can ensure that all sequencing reads
used in the analysis meet some minimal quality criteria by trimming low-quality
parts off of the ends of reads and/or discarding reads of poor quality
altogether. The resulting set of polished reads then needs to be mapped to the
human reference genome because knowing the genomic positions that the bases of
a read provide evidence for is, of course, a prerequisite for variant calling.</p>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-more-on-quality-control-and-mapping"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More on quality control and mapping</div>
  <p>If you would like to explore the topics of quality control and read mapping
in detail, you should take a look at the separate
<a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html">Quality Control</a>
and <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html">Mapping</a>
tutorials.
Here, we will only illustrate the concrete steps necessary for quality
control and read mapping of our particular datasets.</p>
</blockquote>

<h2 id="quality-control">Quality control</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-quality-control-of-the-input-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Quality control of the input datasets</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.72+galaxy1)</span> on each of your four fastq datasets
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“Short read data from your current history”</em>: all 4 FASTQ  datasets selected with <strong>Multiple datasets</strong></li>
      </ul>

      <!--SNIPPET-->
      <blockquote class="tip">   <div class="box-title tip-title" id="tip-select-multiple-datasets"><button class="gtn-boxify-button tip" type="button" aria-controls="tip-select-multiple-datasets" aria-expanded="true"><i class="far fa-lightbulb" aria-hidden="true"></i> <span>Tip: Select multiple datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>   <ol>   <li>Click on <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <strong>Multiple datasets</strong></li>   <li>Select several files by keeping the <kbd>Ctrl</kbd> (or <kbd>COMMAND</kbd>) key pressed and clicking on the files of interest</li> </ol> </blockquote>
      <p><!--END_SNIPPET--></p>

      <p>When you start this job, eight new datasets (one with the calculated raw
data, another one with an html report of the findings for each input
dataset) will get added to your history.</p>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8+galaxy0)</span> to aggregate the raw <strong>FastQC</strong> data of all four input datasets into one report
      <ul>
        <li>In <em>“Results”</em>
          <ul>
            <li><em>“Which tool was used generate logs?”</em>: <code class="language-plaintext highlighter-rouge">FastQC</code></li>
            <li>In <em>“FastQC output”</em>
              <ul>
                <li><em>“Type of FastQC output?”</em>: <code class="language-plaintext highlighter-rouge">Raw data</code></li>
                <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“FastQC output”</em>: all four <em>RawData</em>
outputs of <strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p>Inspect the <em>Webpage</em> output produced by the tool</p>

      <blockquote class="question">
        <div class="box-title question-title" id="question"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

        <ol>
          <li>What do you think of the base qualities of the sequences?</li>
          <li>Which aspect of the quality report is most puzzling to you?
(Hint: Have a look at the GC content of the reads)</li>
        </ol>

        <blockquote class="solution">
          <div class="box-title solution-title" id="solution"><button class="gtn-boxify-button solution" type="button" aria-controls="solution" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

          <ol>
            <li>
              <p>The quality of the sequences looks promising. There are no
discernible systematic problems with it.</p>

              <p>Even the reverse reads, which are typically of somewhat poorer
quality than the corresponding forward reads, look good on average.</p>
            </li>
            <li>
              <p>The GC content plots of the forward and the reverse reads from both
samples reveal a very peculiar bimodal distribution.</p>

              <p>Typically, a non-normal distribution of the GC content of the reads
from a sample is considered to hint at possible contamination.
Here, however, we are dealing with sequencing data from captured
exomes, <em>i.e</em>, the reads are not representing random sequences from
a genome, but rather an arbitrary selection.
In fact, the samples at hand were prepared using Agilent’s
SureSelect V5 technology for exome enrichment, and bimodal GC
content distributions can be seen as a hallmark of that capture
method in several publications (see, for example, Fig. 4C in
<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4477645/">Meienberg et al., 2015</a>
).</p>
            </li>
          </ol>

        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="read-trimming-and-filtering">Read trimming and filtering</h2>
<p>Although the raw reads used in this tutorial are of relatively good overall
quality already, we will apply read trimming and filtering to see if we can
improve things still a bit, but also to demonstrate the general concept.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-trimming-and-filtering-of-the-normal-tissue-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read trimming and filtering of the normal tissue reads</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.36.5" title="Trimmomatic tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Trimmomatic</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.36.5)</span> to trim and filter the normal tissue reads
      <ul>
        <li>
          <p><em>“Single-end or paired-end reads?”</em>: <code class="language-plaintext highlighter-rouge">Paired-end (two separate
input files)</code></p>

          <p>This makes the tool treat the forward and reverse reads simultaneously.</p>

          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ file (R1/first of pair)”</em>: the
forward reads (r1) dataset of the normal tissue sample</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Input FASTQ file (R2/second of pair)”</em>: the
reverse reads (r2) dataset of the normal tissue sample</li>
          </ul>
        </li>
        <li><em>“Perform initial ILLUMINACLIP step?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>“Select standard adapter sequences or provide custom?”</em>: <code class="language-plaintext highlighter-rouge">Standard</code>
              <ul>
                <li><em>“Adapter sequences to use”</em>: <code class="language-plaintext highlighter-rouge">TruSeq3 (paired-ended, for MiSeq and
 HiSeq)</code></li>
              </ul>
            </li>
            <li><em>“Maximum mismatch count which will still allow a full match to be
  performed”</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><em>“How accurate the match between the two ‘adapter ligated’ reads must
  be for PE palindrome read alignment”</em>: <code class="language-plaintext highlighter-rouge">30</code></li>
            <li><em>“How accurate the match between any adapter etc. sequence must be
  against a read”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
            <li><em>“Minimum length of adapter that needs to be detected (PE specific/
  palindrome mode)”</em>: <code class="language-plaintext highlighter-rouge">8</code></li>
            <li><em>“Always keep both reads (PE specific/palindrome mode)?”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>

          <p>These parameters are used to cut ILLUMINA-specific adapter sequences
 from the reads.</p>
        </li>
        <li>In <em>“Trimmomatic Operation”</em>
          <ul>
            <li>In <em>“1: Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut the specified number
 of bases from the start of the read (HEADCROP)</code>
                  <ul>
                    <li><em>“Number of bases to remove from the start of the read”</em>: <code class="language-plaintext highlighter-rouge">3</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> “Insert Trimmomatic Operation”*</li>
            <li>In <em>“2: Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Cut bases off the end of
 a read, if below a threshold quality (TRAILING)</code>
                  <ul>
                    <li><em>“Minimum quality required to keep a base”</em>: <code class="language-plaintext highlighter-rouge">10</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> “Insert Trimmomatic Operation”*</li>
            <li>In <em>“3: Trimmomatic Operation”</em>
              <ul>
                <li><em>“Select Trimmomatic operation to perform”</em>: <code class="language-plaintext highlighter-rouge">Drop reads below a
 specified length (MINLEN)</code>
                  <ul>
                    <li><em>“Minimum quality required to keep a base”</em>: <code class="language-plaintext highlighter-rouge">25</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>

          <p>These three trimming and filtering operations will be applied to the
 reads in the given order after adapter trimming.</p>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Running this job will generate four output datasets:</p>
<ul>
  <li>two for the trimmed forward and reverse reads that still have a proper
mate in the other dataset</li>
  <li>two more datasets of orphaned forward and reverse reads, for which the
corresponding mate got dropped because of insufficient length after
trimming; when you inspect these two files, however, you should find
that they are empty because none of our relatively high quality reads
got trimmed that excessively. You can delete the two datasets to keep
your history more compact.</li>
</ul>

<blockquote class="details">
  <div class="box-title details-title" id="details-more-on-handling-unpaired-reads-in-paired-end-data"><button class="gtn-boxify-button details" type="button" aria-controls="details-more-on-handling-unpaired-reads-in-paired-end-data" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: More on handling unpaired reads in paired-end data</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>Splitting out potential unpaired reads into separate datasets like this
is important because read mappers, typically, expect reads in forward
and reverse input datasets to be arranged in proper pairs, and reads
in one of the datasets without a counterpart in the other would destroy
that expected structure. Therefor, when your data is paired-end data
always make sure you use Trimmomatic in paired-end mode, too!</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-trimming-and-filtering-of-the-tumor-tissue-reads"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read trimming and filtering of the tumor tissue reads</div>
  <ol>
    <li>
      <p>Trim and filter the <strong>tumor tissue</strong> reads following the same steps as above, just change the two input datasets to treat the tumor tissue reads with identical settings.</p>
    </li>
    <li>
      <p>Check that the two unpaired reads datasets are empty, and delete them.</p>
    </li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-exercise-quality-control-of-the-polished-datasets"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Exercise: Quality control of the polished datasets</div>
  <p>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.72+galaxy1" title="FastQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>FastQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.72+galaxy1)</span> and <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/multiqc/multiqc/1.8+galaxy0" title="MultiQC tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>MultiQC</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.8+galaxy0)</span> like before,
but using the four trimmed datasets produced by Trimmomatic as input.</p>

  <blockquote class="question">
    <div class="box-title question-title" id="question-1"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>

    <p>How did read trimming affect the quality reports?</p>

    <blockquote class="solution">
      <div class="box-title solution-title" id="solution-1"><button class="gtn-boxify-button solution" type="button" aria-controls="solution-1" aria-expanded="true"><i class="far fa-eye" aria-hidden="true" ></i> <span>Solution</span><span class="fold-unfold fa fa-minus-square"></span></button></div>

      <p>As expected, trimming the relatively high-quality raw reads did not
have any substantial impact on average. A small fraction of successful
adapter removal is visible though.</p>

    </blockquote>
  </blockquote>
</blockquote>

<h2 id="read-mapping">Read Mapping</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-read-mapping"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Read Mapping</div>
  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Map with BWA-MEM tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with BWA-MEM</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.7.17.1)</span> to map the reads from the <strong>normal tissue</strong> sample to the reference genome
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>“Will you select a reference genome from your history or use a
built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from history and build index</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Use the following dataset as the reference sequence”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the trimmed 
forward reads (r1) dataset of the <strong>normal tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the trimmed
reverse reads (r2) dataset of the <strong>normal tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">231335</code> (this value being taken
from the original name of the normal tissue input files)</li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">Normal</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-more-on-read-group-identifiers-and-sample-names"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: More on read group identifiers and sample names</div>
        <p>In general, you are free to choose ID and SM values to your liking,
but the ID should unambiguously identify the sequencing run that
produced the reads, while the SM value should identify the
biological sample.</p>
      </blockquote>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.17.1" title="Map with BWA-MEM tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Map with BWA-MEM</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.7.17.1)</span> to map the reads from the <strong>tumor tissue</strong> sample,
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in index?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome index</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named option)</li>
          </ul>

          <p>Adjust these settings as before if you are using the imported reference
genome.</p>
        </li>
        <li><em>“Single or Paired-end reads”</em>: <code class="language-plaintext highlighter-rouge">Paired</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select first set of reads”</em>: the trimmed
forward reads (r1) dataset of the <strong>tumor tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Select second set of reads”</em>: the reverse
reads (r2) dataset of the <strong>tumor tissue</strong> sample; output of
<strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
          </ul>
        </li>
        <li><em>“Set read groups information?”</em>: <code class="language-plaintext highlighter-rouge">Set read groups (SAM/BAM specification)</code>
          <ul>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group identifier (ID)”</em>: <code class="language-plaintext highlighter-rouge">231336</code> (this value, again, being
taken from the original name of the tumor tissue input files)</li>
              </ul>
            </li>
            <li><em>“Auto-assign”</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li><em>“Read group sample name (SM)”</em>: <code class="language-plaintext highlighter-rouge">Tumor</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="mapped-reads-postprocessing">Mapped reads postprocessing</h1>

<p>To ensure that we base our variant analysis only on unambiguous, high-quality
read mappings we will do some postprocessing next.</p>

<h2 id="filtering-on-mapped-reads-properties">Filtering on mapped reads properties</h2>

<p>To produce new filtered BAM datasets with only those reads retained that have
been mapped to the reference successfully, have a minimal mapping quality of 1,
and for which the mate read has also been mapped:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-filtering-for-mapping-status-and-quality"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Filtering for mapping status and quality</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Filter BAM datasets on a variety of attributes tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter BAM datasets on a variety of attributes</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.4.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM dataset(s) to filter”</em>: mapped reads
 datasets from the normal <em>and</em> the tumor tissue data, outputs of
 <strong>Map with BWA-MEM</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>“Condition”</em>:</li>
      </ul>
      <ul>
        <li>In <em>“1: Condition”</em>:
          <ul>
            <li>In <em>“Filter”</em>:
              <ul>
                <li>In <em>“1: Filter”</em>:
                  <ul>
                    <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">mapQuality</code>
                      <ul>
                        <li><em>“Filter on read mapping quality (phred scale)”</em>: <code class="language-plaintext highlighter-rouge">&gt;=1</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>In <em>“2: Filter”</em>:
                  <ul>
                    <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">isMapped</code>
                      <ul>
                        <li><em>“Selected mapped reads”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li>Click on <em>“Insert Filter”</em></li>
                <li>In <em>“3: Filter”</em>:
                  <ul>
                    <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">isMateMapped</code>
                      <ul>
                        <li><em>“Select reads with mapped mate”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>

          <p>When you configure multiple filters within one condition, reads have
to pass <em>all</em> the filters to be retained in the output. The above
settings, thus, retain only read pairs, for which both mates are
mapped.</p>

          <p>Note that filtering for a minimal mapping quality is not strictly
necessary. Most variant callers (including **VarScan somatic, which we
will be using later) have an option for using only reads above a
certain mapping quality. In this section, however, we are going to
process the retained reads further rather extensively so it pays off
in terms of performance to eliminate reads we do not plan to use at an
early step.</p>
        </li>
      </ul>
    </li>
  </ol>

  <ul>
    <li><em>“Would you like to set rules?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
  </ul>

</blockquote>

<p>This will result in two new datasets, one for each of the normal and tumor data.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-tools-for-filtering-bam-datasets"><button class="gtn-boxify-button details" type="button" aria-controls="details-tools-for-filtering-bam-datasets" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Tools for filtering BAM datasets</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>The related tutorial on variant detection from exome-seq data, demonstrates
nearly identical filtering of BAM datasets using the tool <strong>Filter BAM
datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>

  <p>These two tools offer very similar functionality and can often be used
interchangeably. Under the hood, <strong>Filter SAM or BAM, output SAM or BAM</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> uses the <code class="language-plaintext highlighter-rouge">samtools view</code> command line tool, while <strong>Filter
BAM datasets on a variety of attributes</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> uses <code class="language-plaintext highlighter-rouge">bamtools
filter</code>. Whatever the BAM filtering task, you should be able to perform it in
Galaxy with one of these two tools.</p>

</blockquote>

<h2 id="removing-duplicate-reads">Removing duplicate reads</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-remove-duplicates"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Remove duplicates</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_rmdup/samtools_rmdup/2.0.1" title="RmDup tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>RmDup</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM file”</em>: filtered reads datasets from
 the normal <em>and</em> the tumor tissue data; the outputs of
 <strong>Filter SAM or BAM</strong></li>
        <li><em>“Is this paired-end or single end data”</em>: <code class="language-plaintext highlighter-rouge">BAM is paired-end</code></li>
      </ul>
      <ul>
        <li><em>“Treat as single-end”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Again, this will produce two new datasets, one for each of the normal and
tumor data.</p>

<h2 id="left-align-reads-around-indels">Left-align reads around indels</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-left-align"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Left-align</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/1.3.1" title="BamLeftAlign tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>BamLeftAlign</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.3.1)</span> with the following parameters:
      <ul>
        <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
          <ul>
            <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM dataset to re-align”</em>: your
filtered and deduplicated reads datasets from the normal <em>and</em> the tumor
tissue data; the outputs of <strong>RmDup</strong></li>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly
named choice)</li>
          </ul>
        </li>
        <li><em>“Maximum number of iterations”</em>: <code class="language-plaintext highlighter-rouge">5</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
    <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
    <ul>
      <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">History</code>
        <ul>
          <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Using reference file”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset</li>
        </ul>
      </li>
    </ul>
  </blockquote>
</blockquote>

<p>As before, this will generate two new datasets, one for each of the normal and tumor data.</p>

<h2 id="recalibrate-read-mapping-qualities">Recalibrate read mapping qualities</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-recalibrate-read-quality-scores"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Recalibrate read quality scores</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/samtools_calmd/samtools_calmd/2.0.2" title="CalMD tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>CalMD</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.0.2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM file to recalculate”</em>: the left-aligned
datasets from the normal and the tumor tissue data; the outputs of
<strong>BamLeftAlign</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><em>“Using reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named choice)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence-2"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>“Choose the source for the reference genome”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Using reference file”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li>
          <p><em>“Do you also want BAQ (Base Alignment Quality) scores to be calculated?”</em>: <code class="language-plaintext highlighter-rouge">No</code></p>

          <p>The <em>VarScan somatic</em> tool that we are going to use for calling variants at
the next step is typically used in combination with unadjusted base quality
scores because the general opinion is that the base quality downgrades
performed by <em>CalMD</em> and other tools from the <em>samtools</em> suite of tools
are too severe for <em>VarScan</em> to retain good sensitivity. We are sticking
with this practice in this tutorial.</p>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-adjusted-base-quality-scores"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using adjusted base quality scores</div>
            <p>If, for your own data, you would like to experiment with adjusted base
quality scores, it is important to understand that <em>VarScan somatic</em> will
only make use of the adjusted scores if they are incorporated directly into
the read base qualities of a BAM input dataset, but not if they are written
to the dataset separately.</p>

            <p>Hence, should you ever decide to use:</p>
            <ul>
              <li>
                <p><em>“Do you also want BAQ (Base Alignment Quality) scores to be calculated?”</em>: <code class="language-plaintext highlighter-rouge">Yes, run BAQ calculation</code></p>

                <p>and you want this setting to affect downstream variant calling with
<em>VarScan somatic</em>, make sure you also set then:</p>
                <ul>
                  <li><em>“Use BAQ to cap read base qualities”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                </ul>
              </li>
            </ul>

            <p>Please also note that BAQ scores are quite expensive to calculate so
be prepared to see a substantial (up to 10x!) increase in job run time 
when enabling it.</p>
          </blockquote>
        </li>
        <li><em>“Additional options”</em>: <code class="language-plaintext highlighter-rouge">Advanced options</code>
          <ul>
            <li><em>“Change identical bases to ‘=’“</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li>
              <p><em>“Coefficient to cap mapping quality of poorly mapped reads”</em>: <code class="language-plaintext highlighter-rouge">50</code></p>

              <p>This last setting is the real reason why we use CalMD at this point.
It is an empirical, but well-established finding that the mapping
quality of reads mapped with <em>bwa</em> should be capped this way before
variant calling.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>This will, once more, produce two new datasets, one for each of the normal
and tumor data.</p>

<h2 id="refilter-reads-based-on-mapping-quality">Refilter reads based on mapping quality</h2>

<p>During recalibration of read mapping qualities <strong>CalMD</strong> may have set some
mapping quality scores to 255. This special value is reserved for <em>undefined</em>
mapping qualities and is used by the tool when a recalibrated mapping quality
would drop below zero. In other words, a value of 255 does not indicate a
particularly good mapping score, but a really poor one.
To remove such reads from the data:</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-eliminating-reads-with-undefined-mapping-quality"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Eliminating reads with undefined mapping quality</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/devteam/bamtools_filter/bamFilter/2.4.1" title="Filter BAM datasets on a variety of attributes tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Filter BAM datasets on a variety of attributes</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.4.1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">param-files</span> <em>“BAM dataset(s) to filter”</em>: the recalibrated
 datasets from the normal and the tumor tissue data; the outputs of
 <strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li>In <em>“Condition”</em>:</li>
      </ul>
      <ul>
        <li>In <em>“1: Condition”</em>:
          <ul>
            <li>In <em>“Filter”</em>:
              <ul>
                <li>In <em>“1: Filter”</em>:
                  <ul>
                    <li><em>“Select BAM property to filter on”</em>: <code class="language-plaintext highlighter-rouge">mapQuality</code>
                      <ul>
                        <li><em>“Filter on read mapping quality (phred scale)”</em>: <code class="language-plaintext highlighter-rouge">&lt;=254</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="variant-calling-and-classification">Variant calling and classification</h1>

<p>Having generated a high-quality set of mapped read pairs, we can proceed to
variant calling. The tool <strong>VarScan somatic</strong> is a dedicated solution for
somatic variant calling that:</p>

<ul>
  <li>detects variant alleles in tumor/normal sample pairs</li>
  <li>calls sample genotypes at variant sites</li>
  <li>classifies variants into germline, somatic and LOH event variants using
solid classical statistics even in the presence of non-pure samples like
those obtained from biopsies</li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-variant-calling-and-classification"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Variant calling and classification</div>

  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/varscan_somatic/varscan_somatic/2.4.3.6" title="VarScan somatic tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>VarScan somatic</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 2.4.3.6)</span> with the following parameters:
      <ul>
        <li><em>“Will you select a reference genome from your history or use a built-in
genome?”</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><em>“reference genome”</em>: <code class="language-plaintext highlighter-rouge">Human: hg19</code> (or a similarly named choice)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-sequence-3"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` sequence</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> sequence as a fasta dataset into your
history instead:</p>
            <ul>
              <li><em>“Will you select a reference genome from your history or use a
built-in genome?”</em>: <code class="language-plaintext highlighter-rouge">Use a genome from the history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“reference genome”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> fasta dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“aligned reads from normal sample”</em>: the mapped
and fully post-processed normal tissue dataset; one of the two outputs
of filtering the <strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span><em>“aligned reads from tumor sample”</em>: the mapped
and fully post-processed tumor tissue dataset; the other output of
filtering the <strong>CalMD</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> outputs</li>
        <li><em>“Estimated purity (non-tumor content) of normal sample”</em>: <code class="language-plaintext highlighter-rouge">1</code></li>
        <li><em>“Estimated purity (tumor content) of tumor sample”</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
        <li><em>“Generate separate output datasets for SNP and indel calls?”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Settings for Variant Calling”</em>: <code class="language-plaintext highlighter-rouge">Customize settings</code>
          <ul>
            <li>
              <p><em>“Minimum base quality”</em>: <code class="language-plaintext highlighter-rouge">28</code></p>

              <p>We have seen, at the quality control step, that our sequencing data is of
really good quality, and we have chosen not to downgrade base qualities at
the quality scores recalibration step above, so we can increase the base
quality required at any given position without throwing away too much of
our data.</p>
            </li>
            <li>
              <p><em>“Minimum mapping quality”</em>: <code class="language-plaintext highlighter-rouge">1</code></p>

              <p>During postprocessing, we have filtered our reads for ones with a
mapping quality of at least one, but <strong>CalMD</strong> may have lowered some
mapping qualities to zero afterwards.</p>
            </li>
          </ul>

          <p>Leave all other settings in this section at their default values.</p>
        </li>
        <li><em>“Settings for Posterior Variant Filtering”</em>: <code class="language-plaintext highlighter-rouge">Use default values</code></li>
      </ul>
    </li>
  </ol>
</blockquote>

<h1 id="variant-annotation-and-reporting">Variant annotation and reporting</h1>

<p>For this tutorial we are going to use variant and gene annotations from many
different sources. Most of these are handled for us by the tools we are going
to use in this section, but we need to take care of importing the data from
four sources into Galaxy separately:</p>

<ul>
  <li>variant annotations from <a href="https://www.cancerhotspots.org">Cancer Hotspots</a></li>
  <li>variant and gene information from the
<a href="https://www.cancergenomeinterpreter.org/biomarkers">Cancer Biomarkers database</a>
of the Cancer Genome Interpreter (CGI) project</li>
  <li>variant and gene information from the <a href="https://civicdb.org">CIViC</a> database</li>
  <li>variant annotations from <a href="https://www.ncbi.nlm.nih.gov/snp">dbSNP</a></li>
  <li>lists of genes annotated with the keywords <em>proto-oncogene</em> or <em>tumor
suppressor</em> at <a href="https://www.uniprot.org/">UniProt</a></li>
</ul>

<p>Each of these annotation sets has been released either in the public domain or
under a free data license, which allows you to use them as part of this
tutorial, but also for other purposes.</p>

<p>Starting from the data downloaded from each of these sites, we have generated
a set of new data files tailored to the requirements of the workflow of this
tutorial and have made them available through Zenodo, again under a free data
license.</p>

<h2 id="get-data-1">Get data</h2>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-data-upload-1"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Data upload</div>

  <ol>
    <li>
      <p>Import the following <strong>variant annotation files</strong> from
<a href="https://zenodo.org/record/7962928">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/7962928/files/hotspots.bed
https://zenodo.org/record/7962928/files/cgi_variant_positions.bed
https://zenodo.org/record/7962928/files/01-Feb-2019-CIVic.bed
https://zenodo.org/record/2582555/files/dbsnp.b147.chr5_12_17.vcf.gz
</code></pre></div>      </div>

      <p>Make sure you select <code class="language-plaintext highlighter-rouge">bed</code> as the datatype for the first three files and
<code class="language-plaintext highlighter-rouge">vcf</code> for the last file.</p>

      <p>Alternatively, add the files from a shared data library on your Galaxy
server instance.</p>
    </li>
    <li>
      <p>Import some <strong>gene-level annotation files</strong> from
<a href="https://zenodo.org/record/2581881">Zenodo</a>:</p>

      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/2581881/files/Uniprot_Cancer_Genes.13Feb2019.txt
https://zenodo.org/record/2581881/files/cgi_genes.txt
https://zenodo.org/record/2581881/files/01-Feb-2019-GeneSummaries.tsv
</code></pre></div>      </div>

      <p>and make sure you select <code class="language-plaintext highlighter-rouge">tabular</code> as their datatype, or add them from
the shared data library.</p>
    </li>
    <li>
      <p>Download <strong>SnpEff functional genomic annotations</strong></p>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-shortcut-1"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Shortcut</div>
        <p>You can skip this step if the Galaxy server you are working on offers
<code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> as a locally installed snpEff database. You can
check the <strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool under <strong>Genome source</strong> to
see if this is the case.</p>
      </blockquote>

      <p>Use <strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to download genome annotation
database <code class="language-plaintext highlighter-rouge">hg19</code>.</p>
    </li>
    <li>
      <p>Check that the datatypes of all new datasets have been set correctly, and
change them if necessary. You may also want to shorten some of the
dataset names.</p>
    </li>
  </ol>

</blockquote>

<h2 id="adding-annotations-to-the-called-variants">Adding annotations to the called variants</h2>

<h3 id="adding-functional-genomic-annotations">Adding functional genomic annotations</h3>

<p>Certainly, not all variants are equal. Many may just be silent mutations with
no effect at the amino acid level, while a few others may be disrupting the
coding sequence of a protein by introducing a premature stop codon or a
frameshift. Of course, it is also important to know <em>which</em> gene is affected
by a variant.
Such functional genomic annotations can be added to a VCF dataset of variants
with <em>SnpEff</em>.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-adding-annotations-with-snpeff"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Adding annotations with SnpEff</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.3+T.galaxy1" title="SnpEff eff tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>SnpEff eff</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 4.3+T.galaxy1)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Sequence changes (SNPs, MNPs, InDels)”</em>: the
output of <strong>VarScan somatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“Input format”</em>: <code class="language-plaintext highlighter-rouge">VCF</code></li>
        <li><em>“Output format”</em>: <code class="language-plaintext highlighter-rouge">VCF (only if input is VCF)</code></li>
        <li><em>“Genome source”</em>: <code class="language-plaintext highlighter-rouge">Locally installed reference genome</code>
          <ul>
            <li><em>“Genome”</em>: <code class="language-plaintext highlighter-rouge">Homo sapiens: hg19</code> (or a similarly named option)</li>
          </ul>

          <blockquote class="comment">
            <div class="box-title comment-title" id="comment-using-the-imported-hg19-snpeff-genome-database"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Using the imported `hg19` SnpEff genome database</div>
            <p>If you have imported the <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff genome database into your
history instead:</p>
            <ul>
              <li><em>“Genome source”</em>: <code class="language-plaintext highlighter-rouge">Downloaded snpEff database in your history</code>
                <ul>
                  <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“SnpEff4.3 Genome Data”</em>: your imported <code class="language-plaintext highlighter-rouge">hg19</code> SnpEff dataset.</li>
                </ul>
              </li>
            </ul>
          </blockquote>
        </li>
        <li><em>“Produce Summary Stats”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h3 id="adding-genetic-and-clinical-evidence-based-annotations">Adding genetic and clinical evidence-based annotations</h3>

<p>Other interesting pieces of information about a variant include aspects like
whether the variant has been observed before in the human population and,
if so, at which frequency. If a variant is known to be associated with specific
diseases, we would also very much like to know that.
To proceed with this kind of genetic and clinical evidence-based annotations,
we are going to convert our list of variants into a database that can be
handled more efficiently than a VCF dataset. We will use the <em>GEMINI</em> tool
suite for this task and for all further work with the variants.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-creating-a-gemini-database-from-a-variants-dataset"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Creating a GEMINI database from a variants dataset</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_load/gemini_load/0.20.1+galaxy2" title="GEMINI load tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI load</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“VCF dataset to be loaded in the GEMINI database”</em>:
the output of <strong>SnpEff eff</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
        <li><em>“The variants in this input are”</em>: <code class="language-plaintext highlighter-rouge">annotated with snpEff</code></li>
        <li>
          <p><em>“This input comes with genotype calls for its samples”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>Calling sample genotypes was part of what we used VarScan somatic for.</p>
        </li>
        <li><em>“Choose a gemini annotation source”</em>: select the latest available annotations snapshot (most likely, there will be only one)</li>
        <li><em>“Sample and family information in PED format”</em>: <code class="language-plaintext highlighter-rouge">Nothing selected</code></li>
        <li><em>“Load the following optional content into the database”</em>
          <ul>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“GERP scores”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“CADD scores”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Gene tables”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“Sample genotypes”</em></li>
            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“variant INFO field”</em></li>
          </ul>

          <p>Be careful to leave <strong>unchecked</strong>:</p>
          <ul>
            <li>
              <p><em>“Genotype likelihoods (sample PLs)”</em></p>

              <p>VarScan somatic does not generate these values</p>
            </li>
            <li>
              <p><em>“only variants that passed all filters”</em></p>

              <p>It is simple and more flexible to filter for this property later</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>During the creation of the database <em>GEMINI</em> already (silently) adds an
impressive amount of annotations it knows about to our variants (including,
<em>e.g.</em>, the frequency at which every variant has been observed in large human
genome sequencing projects). We have gotten all of these for free, just by converting the variants to a GEMINI database!</p>

<p><em>GEMINI</em> also extracts a lot of the information stored in the VCF input dataset for us (such as the functional genomic annotations that we added with
<em>SnpEff</em>).</p>

<p>However, there are typically additional annotations from other sources (not
incorporated into <em>GEMINI</em>) that one would like to add. In addition, <em>GEMINI</em>
is not prepared to extract some non-standard information from VCF datasets,
including some important bits added by <em>VarScan somatic</em>.</p>

<p><strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> is the tool that is designed to help you
with these rather common issues. It lets you add further annotations to the
variants in an already loaded GEMINI database.</p>

<p>As a first step, we are going to use the tool to add some crucial information
generated by <em>VarScan somatic</em>, but not recognized by <em>GEMINI load</em>, to the
database. Specifically, we are interested in three values calculated by
<em>VarScan somatic</em> for each variant it detected:</p>

<ul>
  <li>
    <p><em>Somatic status (SS)</em></p>

    <p>This is a simple numeric code, in which a value of <code class="language-plaintext highlighter-rouge">1</code> indicates a germline
variant, <code class="language-plaintext highlighter-rouge">2</code> a somatic variant and <code class="language-plaintext highlighter-rouge">3</code> an LOH event.</p>
  </li>
  <li>
    <p><em>Germline p-value (GPV)</em></p>

    <p>For variants with a somatic status of 1, this is the error probability
associated with that status call.</p>
  </li>
  <li>
    <p><em>Somatic p-value (SPV)</em></p>

    <p>This is the error probability associated with status calls of 2 and 3
(somatic and LOH calls).</p>
  </li>
</ul>

<p>These values are encoded in the <em>INFO</em> column of the VarScan-generated VCF
dataset and we are going to extract them from there and add them to the
<em>GEMINI</em> database.</p>

<blockquote class="details">
  <div class="box-title details-title" id="details-handling-of-the-vcf-info-field-by-gemini-load"><button class="gtn-boxify-button details" type="button" aria-controls="details-handling-of-the-vcf-info-field-by-gemini-load" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Handling of the VCF INFO field by GEMINI load</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>If you paid close attention to how we generated the GEMINI database, you
might remember that, under <em>“Load the following optional content into the
database”</em>, we checked the option <i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“variant INFO
field”</em>. Did the tool not do what it was supposed to do?</p>

  <p>The answer is that it did, but not in the way you may expect.</p>

  <p>In fact, <em>GEMINI load</em> always extracts INFO field elements it knows about and
stores them into predefined columns of the variants table in the database.
In addition, it can store the <em>whole</em> INFO field of each variant into a
separate <em>info</em> column so that no information gets lost on the way from the
VCF input to the GEMINI database, and that’s the optional behavior we have
been asking for. To keep things compact, however, this column stores the INFO
content in compressed form, which is not readily accessible.</p>

  <p>That’s why it is necessary to extract non-standard INFO field elements
explicitly if they are supposed to be used for filtering and querying.</p>
</blockquote>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-making-variant-call-statistics-accessible"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Making variant call statistics accessible</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="GEMINI annotate tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI annotate</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span>:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GEMINI database”</em>: output of <strong>GEMINI load</strong></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset to use as the annotation source”</em>: output of <strong>VarScan somatic</strong></li>
        <li>
          <p><em>“Strict variant-identity matching of database and annotation records
(VCF format only)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>This setting does not really matter here since you have built the GEMINI
database from the exact same list of variants that we are now retrieving
annotations from and because VarScan somatic does not call multiple alleles
at single sites. Matching on variant-identity is the behavior we would
like to see though, so we may as well be explicit about it.</p>
        </li>
        <li>
          <p><em>“Type of information to add to the database variants”</em>: <code class="language-plaintext highlighter-rouge">Specific values
extracted from matching records in the annotation source (extract)</code></p>

          <ul>
            <li>In <em>“1: Annotation extraction recipe”</em>:
              <ul>
                <li><em>“Elements to extract from the annotation source”</em>: <code class="language-plaintext highlighter-rouge">SS</code></li>
                <li><em>“Database column name to use for recording annotations”</em>:
<code class="language-plaintext highlighter-rouge">somatic_status</code></li>
                <li><em>“What type of data are you trying to extract?”</em>: <code class="language-plaintext highlighter-rouge">Integer numbers</code></li>
                <li><em>“If multiple annotations are found for the same variant,
store …“</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This is the recipe for extracting the VarScan-generated <em>SS</em> field and
adding it as a new column <em>somatic_status</em> to the GEMINI database.</p>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Annotation extraction recipe”</em></li>
            <li>In <em>“2: Annotation extraction recipe”</em>:
              <ul>
                <li><em>“Elements to extract from the annotation source”</em>: <code class="language-plaintext highlighter-rouge">GPV</code></li>
                <li><em>“Database column name to use for recording annotations”</em>:
 <code class="language-plaintext highlighter-rouge">germline_p</code></li>
                <li><em>“What type of data are you trying to extract?”</em>: <code class="language-plaintext highlighter-rouge">Numbers with
decimal precision</code></li>
                <li><em>“If multiple annotations are found for the same variant,
store …“</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This is the recipe for extracting the VarScan-generated <em>GPV</em> field
and adding it as a new column <em>germline_p</em> to the GEMINI database.</p>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Annotation extraction recipe”</em></li>
            <li>In <em>“3: Annotation extraction recipe”</em>:
              <ul>
                <li><em>“Elements to extract from the annotation source”</em>: <code class="language-plaintext highlighter-rouge">SPV</code></li>
                <li><em>“Database column name to use for recording annotations”</em>:
 <code class="language-plaintext highlighter-rouge">somatic_p</code></li>
                <li><em>“What type of data are you trying to extract?”</em>: <code class="language-plaintext highlighter-rouge">Numbers with
decimal precision</code></li>
                <li><em>“If multiple annotations are found for the same variant,
store …“</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This is the recipe for extracting the VarScan-generated <em>SPV</em> field
and adding it as a new column <em>somatic_p</em> to the GEMINI database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Next, we are going to add additional annotations beyond the ones directly
obtainable through <em>GEMINI</em> or from the input VCF dataset. Specifically we
want to add:</p>

<ul>
  <li>
    <p>more information from <strong>dbSNP</strong></p>

    <p>As part of the database creation process, GEMINI already checks all
variants whether they occur in dbSNP and, if so, stores their dbSNP IDs.
Hence, we only need to extract some additional information of interest.</p>
  </li>
  <li>information from <strong>Cancer Hotspots</strong></li>
  <li>links to the <strong>CIViC</strong> database</li>
  <li>information from the <strong>Cancer Genome Interpreter (CGI)</strong></li>
</ul>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-adding-further-annotations"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Adding further annotations</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="GEMINI annotate tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI annotate</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span> to add further annotations from <strong>dbSNP</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GEMINI database”</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset to use as the annotation source”</em>: the imported <code class="language-plaintext highlighter-rouge">dbsnp.b147.chr5_12_17.vcf</code></li>
        <li>
          <p><em>“Strict variant-identity matching of database and annotation records
(VCF format only)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></p>

          <p>dbSNP stores information about specific SNPs observed in human
populations and we would like to record if any exact same SNPs are among
our variants.</p>
        </li>
        <li><em>“Type of information to add to the database variants”</em>: <code class="language-plaintext highlighter-rouge">Specific
values extracted from matching records in the annotation source
(extract)</code>
          <ul>
            <li>In <em>“1: Annotation extraction recipe”</em>:
              <ul>
                <li><em>“Elements to extract from the annotation source”</em>: <code class="language-plaintext highlighter-rouge">SAO</code></li>
                <li><em>“Database column name to use for recording annotations”</em>:
<code class="language-plaintext highlighter-rouge">rs_ss</code></li>
                <li><em>“What type of data are you trying to extract?”</em>: <code class="language-plaintext highlighter-rouge">Integer numbers</code></li>
                <li><em>“If multiple annotations are found for the same variant,
store …“</em>: <code class="language-plaintext highlighter-rouge">the first annotation found</code></li>
              </ul>

              <p>This recipe extracts the dbSNP <em>SAO</em> field and adds it as <em>rs_ss</em> to
the GEMINI database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="GEMINI annotate tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI annotate</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span> to add further annotations from <strong>cancerhotspots</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GEMINI database”</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset to use as the annotation source”</em>: the imported <code class="language-plaintext highlighter-rouge">cancerhotspots_v2.bed</code></li>
        <li>
          <p><em>“Strict variant-identity matching of database and annotation records
(VCF format only)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (with input in BED format this setting will
be ignored)</p>

          <p>For the cancerhotspots data, we are simply going to record the best
q-value associated with any cancerhotspots variant overlapping one
of our variant sites.</p>
        </li>
        <li><em>“Type of information to add to the database variants”</em>: <code class="language-plaintext highlighter-rouge">Specific
values extracted from matching records in the annotation source
(extract)</code>
          <ul>
            <li>In <em>“1: Annotation extraction recipe”</em>:
              <ul>
                <li>
                  <p><em>“Elements to extract from the annotation source”</em>: <code class="language-plaintext highlighter-rouge">5</code></p>

                  <p>The q-values are stored in the fifth column of the BED dataset.</p>
                </li>
                <li><em>“Database column name to use for recording annotations”</em>:
<code class="language-plaintext highlighter-rouge">hs_qvalue</code></li>
                <li><em>“What type of data are you trying to extract?”</em>: <code class="language-plaintext highlighter-rouge">Numbers with
 decimal precision</code></li>
                <li><em>“If multiple annotations are found for the same variant,
store …“</em>: <code class="language-plaintext highlighter-rouge">the smallest of the (numeric) values</code></li>
              </ul>

              <p>This is the recipe for extracting the <em>q-values</em> of overlapping
cancerhotspots sites and adding them as <em>hs_qvalue</em> to the GEMINI
database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="GEMINI annotate tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI annotate</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span> to add links to <strong>CIViC</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GEMINI database”</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset to use as the annotation source”</em>: the imported <code class="language-plaintext highlighter-rouge">CIViC.bed</code></li>
        <li>
          <p><em>“Strict variant-identity matching of database and annotation records
(VCF format only)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (with input in BED format this setting will
be ignored)</p>

          <p>For the CIViC data, we are going to record the hyperlink to any variant
found in the CIViC database that overlaps one of our variant sites.</p>
        </li>
        <li><em>“Type of information to add to the database variants”</em>: <code class="language-plaintext highlighter-rouge">Specific
values extracted from matching records in the annotation source
(extract)</code>
          <ul>
            <li>In <em>“1: Annotation extraction recipe”</em>:
              <ul>
                <li>
                  <p><em>“Elements to extract from the annotation source”</em>: <code class="language-plaintext highlighter-rouge">4</code></p>

                  <p>The hyperlinks are stored in the fourth column of the BED dataset.</p>
                </li>
                <li><em>“Database column name to use for recording annotations”</em>:
<code class="language-plaintext highlighter-rouge">overlapping_civic_url</code></li>
                <li><em>“What type of data are you trying to extract?”</em>: <code class="language-plaintext highlighter-rouge">Text (text)</code></li>
                <li><em>“If multiple annotations are found for the same variant,
store …“</em>: <code class="language-plaintext highlighter-rouge">a comma-separated list of non-redundant (text) values</code></li>
              </ul>

              <p>This is the recipe for extracting the hyperlinks of overlapping CIViC
sites and adding them as a list of <em>overlapping_civic_urls</em> to the
GEMINI database.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_annotate/gemini_annotate/0.20.1+galaxy2" title="GEMINI annotate tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI annotate</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy2)</span> to add information from the <strong>Cancer Genome Interpreter (CGI)</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GEMINI database”</em>: the output of the last
<strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> run</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“Dataset to use as the annotation source”</em>: the imported <code class="language-plaintext highlighter-rouge">cgi_variant_positions.bed</code></li>
        <li>
          <p><em>“Strict variant-identity matching of database and annotation records
(VCF format only)”</em>: <code class="language-plaintext highlighter-rouge">Yes</code> (with input in BED format this setting will
be ignored)</p>

          <p>For the CGI data, we are going to record if any of our variant sites
is overlapped by a variant in the CGI biomarkers database.</p>
        </li>
        <li><em>“Type of information to add to the database variants”</em>: <code class="language-plaintext highlighter-rouge">Binary
indicator (1=found, 0=not found) of whether the variant had any match in
the annotation source (boolean)</code>
          <ul>
            <li><em>“Database column name to use for recording annotations”</em>: <code class="language-plaintext highlighter-rouge">in_cgidb</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="reporting-selected-subsets-of-variants">Reporting selected subsets of variants</h2>

<p>Now that we have built our GEMINI database and enriched it with additional
annotations, it is time that we explore the wealth of information stored in it.
The goal is to produce filtered variant reports that list specific classes
(somatic, germline, LOH) of high-quality variants together with their most
relevant annotations. The way to achieve this is through <strong>GEMINI queries</strong>
that specify:</p>

<ol>
  <li>filters we want to apply to the variant list stored in the database</li>
  <li>the pieces of information about the filtered variants that we would like to
retrieve</li>
</ol>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-the-gemini-query-language"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: The GEMINI query language</div>
  <p>GEMINI queries are extremely flexible, enabling users to express many
different ideas to explore the variant space, and as such, the complete
query syntax can be a bit overwhelming for beginners.</p>

  <p>In this section, we will start off with a rather simple query, then build on
it stepwise before trying a really complex query in the next section.</p>

  <p>For a more detailed explanation of the query syntax, you should consult the
<a href="https://gemini.readthedocs.io/en/latest/content/querying.html">query section of the GEMINI documentation</a>.
Since the GEMINI query syntax is built on the <em>SQLite</em> dialect of <em>SQL</em>, the
<em>SQLite</em> documentation, in particular, its chapter on
<a href="https://sqlite.org/lang_corefunc.html">SQLite core functions</a>, is another
really helpful resource for understanding more sophisticated queries.</p>
</blockquote>

<p>Lets start by configuring a simple query to obtain a report of <em>bona fide</em>
somatic variants. Our strategy for retrieving them is to:</p>

<ol>
  <li>rely on the <em>somatic status</em> of the variants called by <strong>VarScan somatic</strong>
<i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span></li>
  <li>disregard questionable variants, for which either a non-negligible amount of
supporting sequencing reads is also found in the normal tissue data, or
which are only supported by a very small fraction of the reads from the
tumor sample</li>
</ol>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-querying-the-gemini-database-for-somatic-variants"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Querying the GEMINI database for somatic variants</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="GEMINI query tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI query</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy1)</span> with:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“GEMINI database”</em>: the fully annotated database
created in the last <strong>GEMINI annotate</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> step</li>
        <li><em>“Build GEMINI query using”</em>: <code class="language-plaintext highlighter-rouge">Basic variant query constructor</code>
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Insert Genotype filter expression”</em>
              <ul>
                <li><em>“Restrictions to apply to genotype values”</em>:
<code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL &lt;= 0.05 AND gt_alt_freqs.TUMOR &gt;= 0.10</code></li>
              </ul>

              <p>With this genotype-based filter, we retain only those variants that
are supported by less than 5% of the reads of the normal sample, but
by more than 10% of the reads of the tumor sample.</p>
            </li>
            <li>
              <p><em>“Additional constraints expressed in SQL syntax”</em>: <code class="language-plaintext highlighter-rouge">somatic_status = 2</code></p>

              <p>Among the info stored in the GEMINI database is the somatic status
VarScan somatic has called for every variant (remember we
used GEMINI annotate to add it). With the condition
<code class="language-plaintext highlighter-rouge">somatic_status = 2</code> we retain only those variants passing the
genotype filter above <strong>and</strong> considered somatic variants by the
variant caller.</p>
            </li>
            <li>In <em>“Output format options”</em>
              <ul>
                <li><em>“Type of report to generate”</em>: <code class="language-plaintext highlighter-rouge">tabular (GEMINI default)</code>
                  <ul>
                    <li><em>“Add a header of column names to the output”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>“Set of columns to include in the variant report table”</em>: <code class="language-plaintext highlighter-rouge">Custom
(report user-specified columns)</code>
                      <ul>
                        <li>In <em>“Choose columns to include in the report”</em>:
                          <ul>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“chrom”</em></li>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“start”</em></li>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“ref”</em></li>
                            <li><i class="far fa-check-square" aria-hidden="true"></i><span class="visually-hidden">param-check</span> <em>“alt”</em></li>
                          </ul>
                        </li>
                        <li><em>“Additional columns (comma-separated)”</em>: <code class="language-plaintext highlighter-rouge">gene, aa_change, rs_ids,
hs_qvalue, cosmic_ids</code></li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>

              <p>Here we specify, which columns (from the <em>variants</em> table of the
GEMINI database) we want to have included, in the specified order, in
a <code class="language-plaintext highlighter-rouge">tabular</code> variant report.</p>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-how-am-i-supposed-to-know-these-column-names"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: How am I supposed to know these column names?</div>
  <p>Obviously, you need to know the names of the columns (in the tables of the
GEMINI database) to include them in the report, but how are you supposed to
know them?</p>

  <p>The standard ones (added by <em>GEMINI load</em> when building the database) are
listed in the
<a href="https://gemini.readthedocs.io/en/latest/content/database_schema.html">GEMINI documentation</a>.
The non-standard columns (the ones you added with <em>GEMINI annotate</em>) have
the names you gave them, when you added them.</p>

  <p>Alternatively, to get the tables and column names of a specific database
listed, you can use <strong>GEMINI database info</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> like so:</p>

  <ul>
    <li><em>“GEMINI database”</em>: the database you want to explore</li>
    <li><em>“Information to retrieve from the database”</em>: <code class="language-plaintext highlighter-rouge">Names of database tables
and their columns</code></li>
  </ul>
</blockquote>

<p>What about more sophisticated filtering?</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-more-complex-filter-criteria"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: More complex filter criteria</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="GEMINI query tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI query</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy1)</span> with the exact same settings as before, but:
      <ul>
        <li><em>“Additional constraints expressed in SQL syntax”</em>: <code class="language-plaintext highlighter-rouge">somatic_status = 2 AND somatic_p &lt;= 0.05 AND filter IS NULL</code></li>
      </ul>

      <p>This translates into “variants classified as somatic with a p-value &lt;=
 0.05, which haven’t been flagged as likely false-positives”.</p>
    </li>
  </ol>
</blockquote>

<blockquote class="comment">
  <div class="box-title comment-title" id="comment-sql-keywords"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: SQL keywords</div>
  <p>In the condition above, SQL keywords are given in uppercase. This is not a
requirement, but it makes it easier to understand the syntax.</p>

  <p>You can check whether any cell in a data table is empty with <code class="language-plaintext highlighter-rouge">IS NULL</code>, and
whether it contains <em>any</em> value with <code class="language-plaintext highlighter-rouge">IS NOT NULL</code>. To combine different
filter criteria logically, you can use <code class="language-plaintext highlighter-rouge">AND</code> and <code class="language-plaintext highlighter-rouge">OR</code>, and parentheses to
group conditions if required.</p>
</blockquote>

<p>If you have followed all steps up to here exactly, running this job should give
you a tabular dataset of 43 variants, and with the annotations in the report
it is relatively easy to pick out a few interesting ones.
Before we focus on the content of the report, however, we could enhance the
report format a bit more.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-sql-based-output-formatting"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: SQL-based output formatting</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="GEMINI query tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI query</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy1)</span> with the exact same settings as in the
last example, but:
      <ul>
        <li>In <em>“Output format options”</em>
          <ul>
            <li><em>“Additional columns (comma-separated)”</em>: <code class="language-plaintext highlighter-rouge">type, gt_alt_freqs.TUMOR, gt_alt_freqs.NORMAL,
ifnull(nullif(round(max_aaf_all,2),-1.0),0) AS MAF, gene, impact_so,
aa_change, ifnull(round(cadd_scaled,2),'.') AS cadd_scaled,
round(gerp_bp_score,2) AS gerp_bp, ifnull(round(gerp_element_pval,2),'.')
AS gerp_element_pval, ifnull(round(hs_qvalue,2), '.') AS hs_qvalue,
in_omim, ifnull(clinvar_sig,'.') AS clinvar_sig,
ifnull(clinvar_disease_name,'.') AS clinvar_disease_name,
ifnull(rs_ids,'.') AS dbsnp_ids, rs_ss, ifnull(cosmic_ids,'.') AS
cosmic_ids, ifnull(overlapping_civic_url,'.') AS overlapping_civic_url,
in_cgidb</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>
</blockquote>

<p>This last query adds a lot more annotations to the report, and it also
demonstrates the use of the <code class="language-plaintext highlighter-rouge">AS</code> keyword to rename columns and of some
<a href="https://sqlite.org/lang_corefunc.html">SQLite functions</a> to clean up the
output.</p>

<blockquote class="question">
  <div class="box-title question-title" id="question-2"><i class="far fa-question-circle" aria-hidden="true" ></i> Question</div>
  <p>Compare the new report to the previous one to see what has changed.</p>
</blockquote>

<blockquote class="details">
  <div class="box-title details-title" id="details-limitations-of-genotype-column-queries"><button class="gtn-boxify-button details" type="button" aria-controls="details-limitations-of-genotype-column-queries" aria-expanded="true"><i class="fas fa-info-circle" aria-hidden="true" ></i> <span>Details: Limitations of genotype column queries</span><span class="fold-unfold fa fa-minus-square"></span></button></div>
  <p>In case you are wondering why the above query does not use rounding on
the alternate allele frequencies of the samples, <em>i.e.</em>, on
<code class="language-plaintext highlighter-rouge">gt_alt_freqs.TUMOR</code> and <code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL</code>, or why it does not rename
these columns, that is because it would break the query.</p>

  <p>As a general rule, note that all columns in the variants table starting
with <code class="language-plaintext highlighter-rouge">gt</code> (the genotype columns, calculated from the genotype fields of a
VCF dataset) cannot be used like regular SQLite columns, but are parsed by
GEMINI separately. That is why you cannot mix them with regular SQLite
elements like functions and alias specifications with <code class="language-plaintext highlighter-rouge">AS</code>.
You may also have noticed that <code class="language-plaintext highlighter-rouge">gt_alt_freqs.TUMOR</code> and
<code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL</code> do not obey the column order specification of the
query, but end up as the last columns of the tabular report. This is
another artefact of GEMINI’s special treatment of <code class="language-plaintext highlighter-rouge">gt</code> columns.</p>
</blockquote>

<h2 id="generating-reports-of-genes-affected-by-variants">Generating reports of genes affected by variants</h2>

<p>As a final step, let us now try to generate a gene-centered report based on the
same somatic variants we just selected above.</p>

<p>Such a gene-centered report would include annotations that apply to a whole
gene affected by a variant rather than to the variant itself. Examples of such
annotations include known synonyms of an affected gene, its NCBI entrez number,
the ClinVar phenotype, if any, associated with the gene, a hyperlink to the
gene’s page at CIViC.org, <em>etc.</em>.</p>

<p>Some of this information comes built-in into every GEMINI database, but it is
stored in a separate table called <code class="language-plaintext highlighter-rouge">gene_detailed</code>, while all information we
used and queried so far was from the <code class="language-plaintext highlighter-rouge">variants</code> table.</p>

<p>To access information from the <code class="language-plaintext highlighter-rouge">variants</code> and the <code class="language-plaintext highlighter-rouge">gene_detailed</code> table in the
same query we need to join the two tables. Such an operation is not possible
with the <code class="language-plaintext highlighter-rouge">Basic query constructor</code> we have used so far, but requires an
advanced mode for composing the query.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-turning-query-results-into-gene-centered-reports"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Turning query results into gene-centered reports</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/iuc/gemini_query/gemini_query/0.20.1+galaxy1" title="GEMINI query tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>GEMINI query</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.20.1+galaxy1)</span> in advanced mode by choosing
      <ul>
        <li><em>“Build GEMINI query using”</em>: <code class="language-plaintext highlighter-rouge">Advanced query constructor</code></li>
        <li><em>“The query to be issued to the database”</em>: <code class="language-plaintext highlighter-rouge">SELECT v.gene, v.chrom,
 g.synonym, g.hgnc_id, g.entrez_id, g.rvis_pct, v.clinvar_gene_phenotype
 FROM variants v, gene_detailed g WHERE v.chrom = g.chrom AND
 v.gene = g.gene AND v.somatic_status = 2 AND v.somatic_p &lt;= 0.05 AND
 v.filter IS NULL GROUP BY g.gene</code></li>
      </ul>

      <blockquote class="comment">
        <div class="box-title comment-title" id="comment-elements-of-the-sql-query"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Elements of the SQL query</div>
        <p>In this full SQL query, the part between <code class="language-plaintext highlighter-rouge">SELECT</code> and <code class="language-plaintext highlighter-rouge">FROM</code> states which
columns from which database tables we wish to retrieve, while the part
between <code class="language-plaintext highlighter-rouge">FROM</code> and <code class="language-plaintext highlighter-rouge">WHERE</code> specifies the database tables that need to be
consulted and gives them simpler aliases (<code class="language-plaintext highlighter-rouge">v</code> becomes an alias for
the <code class="language-plaintext highlighter-rouge">variants</code> table, <code class="language-plaintext highlighter-rouge">g</code> for the <code class="language-plaintext highlighter-rouge">gene_detailed</code> table), which we can then
use everywhere else in the query.</p>

        <p>The part following <code class="language-plaintext highlighter-rouge">WHERE</code> are the filter criteria we want to apply. Note
that these criteria are almost the same as those we have used in our
earlier somatic variants query, but since we are working with two tables
instead of just one, we need to state which table the filter columns come
from through table prefixes. Thus, <code class="language-plaintext highlighter-rouge">somatic_status</code> becomes
<code class="language-plaintext highlighter-rouge">v.somatic_status</code>, <em>etc.</em>. In addition, we want to report, of course,
corresponding information from the two tables, which is ensured by the
additional criteria <code class="language-plaintext highlighter-rouge">v.chrom = g.chrom</code> and <code class="language-plaintext highlighter-rouge">v.gene = g.gene</code> (the SQL
terminology for this is: we want to join the <code class="language-plaintext highlighter-rouge">variants</code> and the
<code class="language-plaintext highlighter-rouge">gene_detailed</code> tables on their <code class="language-plaintext highlighter-rouge">chrom</code> and <code class="language-plaintext highlighter-rouge">gene</code> columns).</p>

        <p>The <code class="language-plaintext highlighter-rouge">GROUP BY</code> part, finally, specifies that we want to collapse records
affecting the same gene into one.</p>
      </blockquote>
    </li>
  </ol>

  <ul>
    <li>
      <p><em>“Genotype filter expression”</em>: <code class="language-plaintext highlighter-rouge">gt_alt_freqs.NORMAL &lt;= 0.05 AND
gt_alt_freqs.TUMOR &gt;= 0.10</code></p>

      <p>This remains the same as in the previous somatic variants query.</p>
    </li>
  </ul>
</blockquote>

<h2 id="adding-additional-annotations-to-the-gene-centered-report">Adding additional annotations to the gene-centered report</h2>

<p>Unfortunately, <em>GEMINI annotate</em> lets you add columns only to the variants
table of a GEMINI database, but there is no simple way to enhance the
<code class="language-plaintext highlighter-rouge">gene_detailed</code> table with additional annotations. That’s why we are going to
add such extra annotations now to the tabular gene-centered report using more
general-purpose Galaxy tools.</p>

<p>Annotating the tabular gene report produced with GEMINI is a two-step
process, in which we first <em>join</em> the report and tabular annotation sources
into a larger tabular dataset, from which we then eliminate redundant and
unwanted columns, while <em>rearranging</em> the remaining ones.</p>

<p><strong>Step 1</strong> consists of three separate <em>join</em> operations that sequentially pull in the annotations found in the three gene-based tabular datasets that you imported in the <em>Get Data</em> step of this section.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-join"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Join</div>

  <ol>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1" title="Join two files tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Join two files</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> to add <strong>UniProt cancer genes</strong> information
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“1st file”</em>: the GEMINI-generated gene report from the previous step</li>
        <li><em>“Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span>  <em>“2nd file”</em>: the imported <code class="language-plaintext highlighter-rouge">Uniprot_Cancer_Genes</code> dataset</li>
        <li><em>“Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li>
          <p><em>“Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable
lines from 1st file. (-a 1)</code></p>

          <p>If a variant-affected gene is not listed as a Uniprot cancer gene, then,
obviously, we still want to have it included in the final report.</p>
        </li>
        <li><em>“First line is a header line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Ignore case”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li>
          <p><em>“Value to put in unpaired (empty) fields”</em>: <code class="language-plaintext highlighter-rouge">0</code></p>

          <p>If you inspect the <code class="language-plaintext highlighter-rouge">Uniprot_Cancer_Genes</code> dataset, you will see that it
has two annotation columns - one indicating, using <code class="language-plaintext highlighter-rouge">1</code> and <code class="language-plaintext highlighter-rouge">0</code>, whether
a given gene is a <em>proto-oncogene</em> or not, the other one indicating
<em>tumor suppressor</em> genes the same way. For genes missing from this
annotation dataset, we want to fill the corresponding two columns of the
join result with <code class="language-plaintext highlighter-rouge">0</code> to indicate the common case that a gene affected by
a variant is neither a known proto-oncogene, nor a tumor suppressor
gene.</p>
        </li>
      </ul>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1" title="Join two files tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Join two files</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> to add <strong>CGI biomarkers</strong> information
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“1st file”</em>: the partially annotated dataset from the previous</li>
        <li><em>“Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“2nd file”</em>: the imported <code class="language-plaintext highlighter-rouge">cgi_genes</code> dataset</li>
        <li><em>“Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><em>“Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable
lines from 1st file. (-a 1)</code></li>
        <li><em>“First line is a header line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Ignore case”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Value to put in unpaired (empty) fields”</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
      </ul>

      <p>Inspect the input and the result dataset to make sure you understand what
happened at this step.</p>
    </li>
    <li>Use <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.1.1" title="Join two files tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Join two files</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 1.1.1)</span> to add gene information from <strong>CIViC</strong>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“1st file”</em>: the partially annotated dataset from step 2</li>
        <li><em>“Column to use from 1st file”</em>: <code class="language-plaintext highlighter-rouge">Column: 1</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“2nd file”</em>: the imported <code class="language-plaintext highlighter-rouge">GeneSummaries</code> dataset</li>
        <li>
          <p><em>“Column to use from 2nd file”</em>: <code class="language-plaintext highlighter-rouge">Column: 3</code></p>

          <p>The gene column in the CIViC gene summaries annotation dataset is <em>not</em> the first one!</p>
        </li>
        <li><em>“Output lines appearing in”</em>: <code class="language-plaintext highlighter-rouge">Both 1st &amp; 2nd file, plus unpairable
lines from 1st file. (-a 1)</code></li>
        <li><em>“First line is a header line”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>“Ignore case”</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
        <li><em>“Value to put in unpaired (empty) fields”</em>: <code class="language-plaintext highlighter-rouge">.</code></li>
      </ul>

      <p>Inspect the input and the result dataset to make sure you understand what
happened at this step.</p>
    </li>
  </ol>

</blockquote>

<p>If you took a look at all output datasets as suggested, you will have noticed
that each of the <em>join</em> operations kept the gene columns from both input
datasets. In addition, we had no control over the order, in which columns got
added to the report, nor could we exclude columns.</p>

<p>In <strong>Step 2</strong> of the report preparation we are going to address all of these
issues and rearrange to get a fully annotated gene report.</p>

<blockquote class="notranslate hands_on">
  <div class="box-title hands-on-title" id="hands-on-rearrange-to-get-a-fully-annotated-gene-report"><i class="fas fa-pencil-alt" aria-hidden="true" ></i> Hands-on: Rearrange to get a fully annotated gene report</div>
  <ol>
    <li>Run <span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/bgruening/column_arrange_by_header/bg_column_arrange_by_header/0.2" title="Column arrange by header name tool" aria-role="button"><i class="fas fa-wrench" aria-hidden="true"></i> <strong>Column arrange by header name</strong> (<i class="fas fa-cubes" aria-hidden="true"></i> Galaxy version 0.2)</span> configured like
this:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>“file to rearrange”</em>: the final <strong>Join</strong> result dataset from step 3</li>
        <li>In <em>“Specify the first few columns by name”</em>
          <ul>
            <li>In <em>“1: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">gene</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“2: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">chrom</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“3: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">synonym</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“4: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">hgnc_id</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“5: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">entrez_id</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“6: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">rvis_pct</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“7: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">is_OG</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“8: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">is_TS</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“9: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">in_cgi_biomarkers</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“10: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">clinvar_gene_phenotype</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“11: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">gene_civic_url</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>“Specify the first few columns by name”</em></li>
            <li>In <em>“12: Specify the first few columns by name”</em>
              <ul>
                <li><em>“column”</em>: <code class="language-plaintext highlighter-rouge">description</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>“Discard unspecified columns”</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

  <blockquote class="comment">
    <div class="box-title comment-title" id="comment-alternative-tool-suggestion"><i class="far fa-comment-dots" aria-hidden="true" ></i> Comment: Alternative tool suggestion</div>
    <p>If your Galaxy server does not offer the <em>Column arrange</em> tool, it will
almost certainly offer <strong>Cut columns from a table</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>,
which can be used as a drop-in replacement.
Instead of column names, however, this tool expects a comma-separated list
of column indexes, like <code class="language-plaintext highlighter-rouge">c1,c2</code> for the first and second column, so you
will have to first figure out the column numbers of your columns of
interest.</p>

    <p><strong>Note</strong>: Be sure not to confuse the suggested tool with <em>Cut
columns from a table (cut)</em>, which does <em>not</em> let you change the order of
columns!</p>
  </blockquote>
</blockquote>

<h1 id="conclusion">Conclusion</h1>

<p>In addition to merely calling variants, <em>somatic variant calling</em> tries to
distinguish <em>somatic mutations</em>, which are private to tumor tissue, from
<em>germline</em> mutations, that are shared by tumor and healthy tissue, and
<em>loss-of-heterozygosity</em> events, which involve the loss, from tumor tissue, of
one of two alleles found at a biallelic site in healthy tissue.</p>

<p>Dedicated somatic variant callers can perform this classification on
statistical grounds, but the interpretation of any list of variants (somatic,
germline or LOH) also depends crucially on rich genetic and cancer-specific
variant and gene annotations.</p>

                </section>

                <section aria-label="Tutorial Footer, Feedback, Citation" id="tutorial-footer">
                        <h3>You've Finished the Tutorial</h3>
                        <button id="tutorial-finish-button" class="btn btn-primary" onclick="tutorial_finish()">I finished this tutorial 👍</button>
                        <p style="display: none" id="tutorial-finish-text">Please also consider filling out the <a href="#gtn-feedback">Feedback Form</a> as well!</p>
                        <script>
                        function tutorial_finish() {
                          if(plausible !== undefined){
                            plausible('TutorialComplete', {callback: tutorial_finish_finish, props: {path: document.location.pathname}})
                          } else {
                            tutorial_finish_finish();
                          }
                        }
                        function tutorial_finish_finish() {
                          document.getElementById("tutorial-finish-button").innerText = 'Congrats! Thanks for letting us know! 🎉'
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-button").disabled = true
                          document.getElementById("tutorial-finish-text").style.display = 'block'
                        }
                        </script>

                
                <blockquote class="key_points">
                    <div class="box-title"><i class="fas fa-key" aria-hidden="true"></i> Key points</div>
                    <ul>
                        
                        <li><p>Follow best practices for read mapping, quality control and mapped reads postprocessing to minimize false-positive variant calls.</p>
</li>
                        
                        <li><p>Use a dedicated somatic variant caller to call variants and to classify them into somatic, germline and LOH event variants on statistical grounds.</p>
</li>
                        
                        <li><p>Annotations and queries based on variant properties add relevance to variant and gene reports.</p>
</li>
                        
                        <li><p>A framework like GEMINI is very helpful for managing, annotating and querying lists of variants in a flexible way.</p>
</li>
                        
                        <li><p>Prefer public, free annotation sources to foster reproducibility and information sharing.</p>
</li>
                        
                    </ul>
                </blockquote>
                

                <h1>Frequently Asked Questions</h1>
                Have questions about this tutorial? Check out the <a href="faqs/">tutorial FAQ page</a> or the  <a href="/training-material/topics/variant-analysis/faqs/">FAQ page for the Variant Analysis topic</a> to see if your question is listed there.
                If not, please ask your question on the <a href="https://gitter.im/Galaxy-Training-Network/Lobby">GTN Gitter Channel</a> or the
                <a href="https://help.galaxyproject.org">Galaxy Help Forum</a>

                

                
                <h1 data-toc-skip>Useful literature</h1>
                <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                


                

                

                <h1 id="gtn-feedback">Feedback</h1>
                <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452">how it went</a>.
                <br>Did you use this material as a learner or student? Click the form below to leave feedback.<i class="fas fa-hand-point-down"></i>
                </p>

                <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=variant-analysis/somatic-variants"><a href="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=variant-analysis/somatic-variants">Feedback Form</a></iframe>

                <h1>Citing this Tutorial</h1>
                <p>
                    <ol>
                        <li id="citation-text">
                            Wolfgang Maier,  <b>Identification of somatic and germline variants from tumor and normal sample pairs (Galaxy Training Materials)</b>. <a href="https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html">https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html</a> Online; accessed TODAY
                        </li>
                        <li>
                        Hiltemann, Saskia, Rasche, Helena et al., 2023 <b>Galaxy Training: A Powerful Framework for Teaching!</b> PLOS Computational Biology <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1010752">10.1371/journal.pcbi.1010752</a>
                        </li>
                        <li>
                        Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                        </li>
                    </ol>
                </p>

                <!-- collapsible boxcontaining the BibTeX-formatted citation -->
                <blockquote class="details">

                  <div id="citation-bibtex" class="box-title">
                    <button type="button" aria-controls="citation-bibtex" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> BibTeX<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>
                   <p style="display: none;">

                   <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">


<code id="citation-code">@misc{variant-analysis-somatic-variants,
author = "Wolfgang Maier",
	title = "Identification of somatic and germline variants from tumor and normal sample pairs (Galaxy Training Materials)",
	year = "",
	month = "",
	day = ""
	url = "\url{https://training.galaxyproject.org/training-material/topics/variant-analysis/tutorials/somatic-variants/tutorial.html}",
	note = "[Online; accessed TODAY]"
}
@article{Hiltemann_2023,
	doi = {10.1371/journal.pcbi.1010752},
	url = {https://doi.org/10.1371%2Fjournal.pcbi.1010752},
	year = 2023,
	month = {jan},
	publisher = {Public Library of Science ({PLoS})},
	volume = {19},
	number = {1},
	pages = {e1010752},
	author = {Saskia Hiltemann and Helena Rasche and Simon Gladman and Hans-Rudolf Hotz and Delphine Larivi{\`{e}}re and Daniel Blankenberg and Pratik D. Jagtap and Thomas Wollmann and Anthony Bretaudeau and Nadia Gou{\'{e}} and Timothy J. Griffin and Coline Royaux and Yvan Le Bras and Subina Mehta and Anna Syme and Frederik Coppens and Bert Droesbeke and Nicola Soranzo and Wendi Bacon and Fotis Psomopoulos and Crist{\'{o}}bal Gallardo-Alba and John Davis and Melanie Christine Föll and Matthias Fahrner and Maria A. Doyle and Beatriz Serrano-Solano and Anne Claire Fouilloux and Peter van Heusden and Wolfgang Maier and Dave Clements and Florian Heyl and Björn Grüning and B{\'{e}}r{\'{e}}nice Batut and},
	editor = {Francis Ouellette},
	title = {Galaxy Training: A powerful framework for teaching!},
	journal = {PLoS Comput Biol} Computational Biology}
}
</code>
                   </pre></div></div>
                   </p>
                </blockquote>

		<h2 id="funding">Funding</h2>
		<p>These individuals or organisations provided funding support for the development of this resource</p>
                


<script>
// update the date on load, or leave fallback of 'today'
const citationTodaysDate = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", citationTodaysDate.toDateString());
</script>

                <i class="far fa-thumbs-up" aria-hidden="true"></i> Congratulations on successfully completing this tutorial!

                

                

		
                <blockquote class="details follow-up" id="admins-install-missing-tools">
                  <div id="admin-missing-tools" class="box-title">
                    <button type="button" aria-controls="admin-missing-tools" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Galaxy Administrators: Install the missing tools<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>


			<p>You can use Ephemeris's <code>shed-tools install</code> command to install the tools used in this tutorial.</p>
<div class="highlight"><pre class="highlight"><code>shed-tools install [-g GALAXY] [-a API_KEY] -t &lt;(curl https://training.galaxyproject.org/training-material/api/topics/variant-analysis/tutorials/somatic-variants/tutorial.json | jq .admin_install_yaml -r)</code></pre></div>
<p>Alternatively you can copy and paste the following YAML</p>
<div class="highlight"><pre class="highlight"><code>---
install_tool_dependencies: true
install_repository_dependencies: true
install_resolver_dependencies: true
tools:
- name: column_arrange_by_header
  owner: bgruening
  revisions: 6c6d26ff01ff
  tool_panel_section_label: Filter and Sort
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: text_processing
  owner: bgruening
  revisions: 0a8c6b61f0f4
  tool_panel_section_label: Text Manipulation
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bamtools_filter
  owner: devteam
  revisions: cb20f99fd45b
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: bwa
  owner: devteam
  revisions: 3fe632431b68
  tool_panel_section_label: Mapping
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: fastqc
  owner: devteam
  revisions: e7b2202befea
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: freebayes
  owner: devteam
  revisions: 57def2d7c093
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_calmd
  owner: devteam
  revisions: 06dc50b7b711
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: samtools_rmdup
  owner: devteam
  revisions: 586f9e1cdb2b
  tool_panel_section_label: SAM/BAM
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gemini_annotate
  owner: iuc
  revisions: cf0f0f05ba9f
  tool_panel_section_label: Gemini
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gemini_load
  owner: iuc
  revisions: 2270a8b83c12
  tool_panel_section_label: Gemini
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: gemini_query
  owner: iuc
  revisions: da74170c55c7
  tool_panel_section_label: Gemini
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: multiqc
  owner: iuc
  revisions: bf675f34b056
  tool_panel_section_label: Quality Control
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: snpeff
  owner: iuc
  revisions: 5c7b70713fb5
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: varscan_somatic
  owner: iuc
  revisions: cf8ffc79db67
  tool_panel_section_label: Variant Calling
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
- name: trimmomatic
  owner: pjbriggs
  revisions: dfa082f84068
  tool_panel_section_label: FASTA/FASTQ
  tool_shed_url: https://toolshed.g2.bx.psu.edu/
</code></pre></div>
                </blockquote>
		

		<blockquote class="details hide-when-printing" id="feedback-responses">
                  <div id="feedback-response-c" class="box-title">
                    <button type="button" aria-controls="feedback-response-c" aria-expanded="false">
                      <i class="fas fa-info-circle" aria-hidden="true"></i>
                      <span class="visually-hidden"></span> Feedback<span role="button" class="fold-unfold fa fa-minus-square" aria-hidden="true"></span>
                    </button>
                   </div>

		   <p>
		   
		   <table class="charts-css bar show-labels" style="--labels-size: 8rem; overflow-y: hidden">
		   
			<tr>
				<th scope="row"><span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 1.0">10</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.1">1</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">2 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.1">1</td>
			</tr>
		   
			<tr>
				<th scope="row"><span class="sr-only">1 stars</span><i class="fa fa-star" aria-hidden="true"></i></th>
				<td style="--size: 0.1">1</td>
			</tr>
		   
		   </table>
		   </p>
		
        
        
		    
		    <b>November 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: it was easy to follow
					
				</li>
				
		    </ul>
		    
		    <b>January 2023</b>
		    <ul>
				
				<li>
					<span class="sr-only">4 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: In galaxy training,  the tools recommendation are quite good!
					<b>Disliked</b>: 1.  The workflow file should be updated.
				</li>
				
		    </ul>
		    
		    <b>September 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">2 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: its step by step
					<b>Disliked</b>: steps are not explained - for example: why do i need align left, why do i need to refilter quality levels...? Way to hard to understand. More Explanation why things are done a certain way. Its not explained why certain values are set and how to choose the right values. This tutorial is way to complicated.
				</li>
				
		    </ul>
		    
		    <b>March 2022</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The additional explanations of the individual configuration options within the tools, especially the SQLite Syntax.
					
				</li>
				
		    </ul>
		    
		    <b>November 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: It is structured well and well explained.
					<b>Disliked</b>: It might be helpful to mention the alternative tools and solutions
				</li>
				
		    </ul>
		    
		    <b>July 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: The detailed steps were very useful to get started and this would help me do advance analysis as well.
					
				</li>
				
		    </ul>
		    
		    <b>February 2021</b>
		    <ul>
				
				<li>
					<span class="sr-only">5 stars</span><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i><i class="fa fa-star" aria-hidden="true"></i>:
					<b>Liked</b>: the work flow and associated explanations. Perfect for my students to get acquainted with the workflow, inputs and outputs, and then move to HPC learning
					<b>Disliked</b>: incorporate UMI-based workflows (e.g., duplex consensus)
				</li>
				
		    </ul>
		    
        
		</blockquote>

		


		
		
		

                </section>

            </div>
        </div>
    </div>
</article>
<br/>
<br/>
<br/>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides researchers with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/faqs/gtn/gtn-what-is-it.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:T00318">gxy.io/GTN:T00318</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/variant-analysis/tutorials/somatic-variants/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/variant-analysis/tutorials/somatic-variants/tutorial.md">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/4bbac78b4791e7e0a5f3fcee9e506ed0d1e4d691">4bbac78</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><?xml version="1.0" encoding="UTF-8" standalone="no"?><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.20cf1f17.js'></script>

	
	

    </body>
</html>