<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>DIA Analysis using OpenSwathWorkflow</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:title" content="Galaxy Training: DIA Analysis using OpenSwathWorkflow" />
        <meta property="og:description" content="Training material for proteomics workflows in Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/proteomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Proteomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2FDIA_Analysis_OSW%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2FDIA_Analysis_OSW%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2FDIA_Analysis_OSW%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2FDIA_Analysis_OSW%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2FDIA_Analysis_OSW%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fproteomics%2Ftutorials%2FDIA_Analysis_OSW%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "DIA Analysis using OpenSwathWorkflow",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Proteomics",
    "description": "Training material for proteomics workflows in Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
  },
  "courseCode": "proteomics / DIA_Analysis_OSW / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'DIA Analysis using OpenSwathWorkflow' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - How to analyze data independent acquisition (DIA) mass spectrometry (MS) data using OpenSwathWorkflow?\n - How to detect different Spike-in amounts of Ecoli in a HEK-Ecoli Benchmark DIA dataset?\n\n\\nThe objectives are:\n - Analysis of HEK-Ecoli Spike-in DIA data in Galaxy\n - Understanding DIA data principles and characteristics\n - Using OpenSwathworkflow to analyze HEK-Ecoli Spike-in DIA data\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html",
      "name": "Library Generation for DIA Analysis",
      "description": "Hands-on for 'Library Generation for DIA Analysis' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Matthias Fahrner"
    },
    {
      "@type": "Person",
      "name": "Melanie F√∂ll"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Matthias Fahrner"
    },
    {
      "@type": "Person",
      "name": "Melanie F√∂ll"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Proteomics",
      "description": "Training material for proteomics workflows in Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/proteomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0121",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0121",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0121"
    }
  ],
  "skillLevel": "Intermediate"
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>DIA Analysis using OpenSwathWorkflow</h1>
        
        <div class="alert alert-warning" role="alert">
            <h4 class="alert-heading">Under Development!</h4>
            <p>This tutorial is not in its final state. The content may change a lot in the next months.
            Because of this status, it is also not listed in the topic pages.</p>
        </div>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/matthias313/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/matthias313" alt="Matthias Fahrner">Matthias Fahrner</a>, <a href="/training-material/hall-of-fame/foellmelanie/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/foellmelanie" alt="Melanie F√∂ll">Melanie F√∂ll</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How to analyze data independent acquisition (DIA) mass spectrometry (MS) data using OpenSwathWorkflow?</p>
</li>
            
            <li><p>How to detect different Spike-in amounts of Ecoli in a HEK-Ecoli Benchmark DIA dataset?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Analysis of HEK-Ecoli Spike-in DIA data in Galaxy</p>
</li>
            
            <li><p>Understanding DIA data principles and characteristics</p>
</li>
            
            <li><p>Using OpenSwathworkflow to analyze HEK-Ecoli Spike-in DIA data</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
    <li>
    
        
        
        <a href="/training-material/topics/proteomics">Proteomics</a>
        
            <ul>
                
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 2 hours</p>
            

            
            <p><strong><i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> Level: </strong>
             Intermediate 
<span class="level intermediate" title="Intermediate Tutorial">
  <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span> <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">level</span>
</span>

</p>
            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                

                

                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>This training covers data independent acquisition (DIA) mass spectrometry (MS) applying spectral libraries for peptide identification and quantification. 
You can learn how to prepare and optimize a spectral library for the analysis of DIA data in the<a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">DIA library generation tutorial</a>.</p>

<p>The proteome refers to the entirety of proteins in a biological system (e.g cell, tissue, organism). Proteomics is the large-scale experimental analysis of proteins and proteomes, most often performed by mass spectrometry that enables great sensitivity and throughput. Especially for complex protein mixtures, bottom-up mass spectrometry is the standard approach. In bottom-up proteomics, proteins are digested with a specific protease into peptides and the measured peptides are in silico reassembled into the corresponding proteins. Inside the mass spectrometer, not only the peptides are measured (MS1 level), but the peptides are also fragmented into smaller peptides which are measured again (MS2 level). This is referred to as tandem-mass spectrometry (MS/MS). Identification of peptides is performed by peptide spectrum matching of the theoretical spectra generated from the input protein database (fasta file) with the measured MS2 spectra. Peptide quantification is most often performed by measuring the area under the curve of the MS1 level peptide peaks, but special techniques such as TMT allow to quantify peptides on MS2 level. Nowadays, bottom-up tandem-mass spectrometry approaches allow for the identification and quantification of several thousand proteins.</p>

<p>In clinical proteomic studies often two or more conditions should be compared against each other, thus focusing on the proteins which were found in all conditions. Using the data dependent acquisition (DDA) approach could lead to limited numbers of cumulative proteins, due to method instrinsic dependencies (e.g. if the same peptides is selected for fragmentation in all measurements). 
Over the last decade another acquisition method has been developed addressing the urgent need for increased cumulatively identified and quantified peptides and proteins across multiple measurements (<a class="citation" href="#Ludwig2018">Ludwig <i>et al.</i> 2018</a>). The so called data independent acquisition (DIA) circumvents the time and abundance dependent selection for fragmentation by using predefined fragmentation windows (m/z windows) going through the whole m/z range of the previous MS1 scan.</p>
<figure id="figure-1"><img src="../../images/DIA_analysis_window.png" alt="DIA_window_schema" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Measurement principles of DIA and m/z window schema of staggered DIA windows.</figcaption></figure>

<p>Therefore, all peptides which are present in the same m/z window at the same time are fragmented simultaneously and a MS2 spectra containing fragments from multiple peptides is acquired. Using the same m/z windows for all measurements, results in more reproducible fragmentation and potential identification across multiple measurements. 
However, the resulting MS2 spectra contain fragments from multiple peptides and are often more complex and do not allow to directly link a specific (m/z) mass from the MS1 to a single MS2 fragment spectra.</p>
<figure id="figure-2"><img src="../../images/DIA_analysis_MS2.png" alt="DIA_vs_DDA" /><figcaption><span class="figcaption-prefix">Figure 2:</span> The MS2 scans in the DIA approach contain fragment ions from multiple precursers and are therefore more complex than the precursor-specific MS2 scans in DDA.</figcaption></figure>

<p>To allow for the identification of peptides in those ambiguous MS2 spectra, a spectral library can be used. The spectral library contains experimentally measured  MS2 spectra, which are specific for one precursor (from previous DDA measurements). In more recent approaches the MS2 spectra can be predicted based on theoretical peptide sequences (e.g. from a protein database).</p>
<figure id="figure-3"><img src="../../images/DIA_analysis_basic.png" alt="DIA_basics" /><figcaption><span class="figcaption-prefix">Figure 3:</span> Spectral libraries are necesseary for the identification of peptides in DIA MS2 scans. In this example the spectral library is generated based on DDA data from the same samples.</figcaption></figure>

<p>One of the open-source software options for DIA data analysis is <em>OpenSwath</em> (<a class="citation" href="#Rost2014">R√∂st <i>et al.</i> 2014</a>), which requires a spectral library, retention time alignment peptides as well as the DIA MS data.</p>

<p>The dataset in this tutorial consists of two different Spike-in mixtures of human and Ecoli peptides, with an increase of the Ecoli peptides in one of the Spike-ins. For each Spike-in there a four replicate measurements to allow for statistical analysis later on. Here we will go through the data analysis of DIA mass spectrometry data and will get first impressions on which of the Spike-ins might contain higher amounts of Ecoli peptides.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#raw-file-conversion-with-msconvert" id="markdown-toc-raw-file-conversion-with-msconvert">raw File conversion with <strong>msconvert</strong></a></li>
  <li><a href="#dia-analysis-using-openswathworkflow" id="markdown-toc-dia-analysis-using-openswathworkflow">DIA analysis using <strong>OpenSwath<span class="notranslate">Workflow</span></strong></a>    <ol>
      <li><a href="#combining-osw-results-using-pyprophet-merge" id="markdown-toc-combining-osw-results-using-pyprophet-merge">Combining osw results using <strong>PyProphet merge</strong></a></li>
    </ol>
  </li>
  <li><a href="#fdr-estimation-using-pyprophet-score" id="markdown-toc-fdr-estimation-using-pyprophet-score">FDR estimation using <strong>PyProphet score</strong></a>    <ol>
      <li><a href="#apply-scores-with-pyprophet-peptide" id="markdown-toc-apply-scores-with-pyprophet-peptide">Apply scores with <strong>PyProphet peptide</strong></a></li>
      <li><a href="#apply-scores-with-pyprophet-protein" id="markdown-toc-apply-scores-with-pyprophet-protein">Apply scores with <strong>PyProphet protein</strong></a></li>
      <li><a href="#exporting-the-results-with-pyprophet-export" id="markdown-toc-exporting-the-results-with-pyprophet-export">Exporting the results with <strong>PyProphet export</strong></a></li>
    </ol>
  </li>
</ol>

</blockquote>

<h2 id="get-data">Get data</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this tutorial and give it a meaningful name</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>Import the fasta and raw files as well as the sample annotation and the iRT Transition file from <a href="https://zenodo.org/record/4301690">Zenodo</a>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4301690/files/HEK_Ecoli_lib.pqp
https://zenodo.org/record/4301690/files/iRTassays.tsv
https://zenodo.org/record/4301690/files/Sample_annotation.txt
https://zenodo.org/record/4301690/files/Sample1.raw
https://zenodo.org/record/4301690/files/Sample2.raw
https://zenodo.org/record/4301690/files/Sample3.raw
https://zenodo.org/record/4301690/files/Sample4.raw
https://zenodo.org/record/4301690/files/Sample5.raw
https://zenodo.org/record/4301690/files/Sample6.raw
https://zenodo.org/record/4301690/files/Sample7.raw
https://zenodo.org/record/4301690/files/Sample8.raw
</code></pre></div>      </div>
      <blockquote class="tip">

        <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

        <ul>
          <li>Copy the link location</li>
          <li>
            <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
          </li>
          <li>Select <strong>Paste/Fetch Data</strong></li>
          <li>
            <p>Paste the link into the text field</p>
          </li>
          <li>
            <p>Press <strong>Start</strong></p>
          </li>
          <li><strong>Close</strong> the window</li>
        </ul>

        <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

      </blockquote>
    </li>
    <li>Once the files are green, rename the sample annotation file in ‚ÄòSample_annotation‚Äô, the spectral library in ‚ÄòHEK_Ecoli_lib‚Äô, the iRT transition file in ‚ÄòiRTassays‚Äô and the raw files in ‚ÄòSample1.raw‚Äô, ‚ÄòSample2.raw‚Äô, ‚ÄòSample3.raw‚Äô, ‚ÄòSample4.raw‚Äô, ‚ÄòSample5.raw‚Äô, ‚ÄòSample6.raw‚Äô, ‚ÄòSample7.raw‚Äô and ‚ÄòSample8.raw‚Äô
      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-dataset"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a dataset</h3>
        <ul>
          <li>Click on the <i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-pencil</span> <strong>pencil icon</strong> for the dataset to edit its attributes</li>
          <li>In the central panel, change the <strong>Name</strong> field</li>
          <li>Click the <strong>Save</strong> button</li>
        </ul>
      </blockquote>
    </li>
    <li>Generate a collection for all .raw files (and name it DIA_data)
      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-dataset-collection"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a dataset collection</h3>

        <ul>
          <li>Click on <strong>Operations on multiple datasets</strong> (check box icon) at the top of the history panel <img src="../../../galaxy-interface/images/historyItemControls.png" alt="Operations on multiple datasets button" /></li>
          <li>Check all the datasets in your history you would like to include</li>
          <li>
            <p>Click <strong>For all selected..</strong> and choose <strong>Build dataset list</strong></p>

            <p><img src="/training-material/topics/galaxy-interface/images/buildList.png" alt="build list collection menu item" width="15%" /></p>
          </li>
          <li>Enter a name for your collection</li>
          <li>Click <strong>Create List</strong> to build your collection</li>
          <li>Click on the checkmark icon at the top of your history again</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="raw-file-conversion-with-msconvert">raw File conversion with <strong>msconvert</strong></h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-converting-vendor-specific-raw-to-open-mzml-format"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Converting vendor specific raw to open mzML format</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/3.0.19052.1" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/msconvert/msconvert/3.0.19052.1"><strong>msconvert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/msconvert/msconvert/3.0.19052.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúInput unrefined MS data‚Äù</em>: <code class="language-plaintext highlighter-rouge">DIA_data</code></li>
        <li><em>‚ÄúDo you agree to the vendor licenses?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúOutput Type‚Äù</em>: <code class="language-plaintext highlighter-rouge">mzML</code></li>
        <li>In <em>‚ÄúData Processing Filters‚Äù</em>:
          <ul>
            <li><em>‚ÄúApply peak picking?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúDemultiplex overlapping or MSX spectra‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>‚ÄúOptimization‚Äù</em>: <code class="language-plaintext highlighter-rouge">Overlap only</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúGeneral Options‚Äù</em>:
          <ul>
            <li><em>‚ÄúSIM as Spectra‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúIgnore unknown instrument error‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúOutput Encoding Settings‚Äù</em>:
          <ul>
            <li><em>‚ÄúIntensity Encoding Precision‚Äù</em>: <code class="language-plaintext highlighter-rouge">64</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-demultiplexing"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Demultiplexing</h3>
        <p>Demultiplexing is needed when the MS2 m/z windows of consecutive scan cycles are overlapping e.g. for the staggered approach (50% shift between the cycles).</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="dia-analysis-using-openswathworkflow">DIA analysis using <strong>OpenSwath<span class="notranslate">Workflow</span></strong></h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-dia-analysis-using-openswathworkflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: DIA analysis using OpenSwath<span class="notranslate">Workflow</span></h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/openms_openswathworkflow/OpenSwathWorkflow/2.6+galaxy0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/openms_openswathworkflow/OpenSwathWorkflow/2.6+galaxy0"><strong>OpenSwath<span class="notranslate">Workflow</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/openms_openswath<span class="notranslate">workflow</span>/OpenSwath<span class="notranslate">Workflow</span>/2.6+<span class="notranslate">galaxy</span>0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúInput files separated by blank‚Äù</em>: <code class="language-plaintext highlighter-rouge">DIA_data</code> (output of <strong>msconvert</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äútransition file (‚ÄòTraML‚Äô,‚Äôtsv‚Äô,‚Äôpqp‚Äô)‚Äù</em>: <code class="language-plaintext highlighter-rouge">HEK_Ecoli_lib</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚Äútransition file (‚ÄòTraML‚Äô)‚Äù</em>: <code class="language-plaintext highlighter-rouge">iRTassays</code></li>
        <li><em>‚ÄúExtraction window in Thomson or ppm (see mz_extraction_window_unit)‚Äù</em>: <code class="language-plaintext highlighter-rouge">10.0</code></li>
        <li><em>‚ÄúExtraction window used in MS1 in Thomson or ppm (see mz_extraction_window_ms1_unit)‚Äù</em>: <code class="language-plaintext highlighter-rouge">10.0</code></li>
        <li>In <em>‚ÄúParameters for the RTNormalization for iRT petides‚Äù</em>:
          <ul>
            <li><em>‚ÄúWhich outlier detection method to use (valid: ‚Äòiter_residual‚Äô, ‚Äòiter_jackknife‚Äô, ‚Äòransac‚Äô, ‚Äònone‚Äô)‚Äù</em>: <code class="language-plaintext highlighter-rouge">none</code></li>
            <li><em>‚ÄúWhether the algorithms should try to choose the best peptides based on their peak shape for normalization‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúMinimal number of bins required to be covered‚Äù</em>: <code class="language-plaintext highlighter-rouge">7</code></li>
          </ul>
        </li>
        <li>In <em>‚ÄúScoring parameters section‚Äù</em>:
          <ul>
            <li>In <em>‚ÄúTransitionGroupPicker‚Äù</em>:
              <ul>
                <li><em>‚ÄúMinimal peak width (s), discard all peaks below this value (-1 means no action)‚Äù</em>: <code class="language-plaintext highlighter-rouge">5.0</code></li>
                <li><em>‚ÄúTries to compute a quality value for each peakgroup and detect outlier transitions‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
              </ul>
            </li>
            <li>In <em>‚ÄúScores‚Äù</em>:
              <ul>
                <li><em>‚ÄúUse the MI (mutual information) score‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúAdvanced Options‚Äù</em>: <code class="language-plaintext highlighter-rouge">Show Advanced Options</code>
          <ul>
            <li><em>‚ÄúOutput an XIC with a RT-window by this much large‚Äù</em>: <code class="language-plaintext highlighter-rouge">100.0</code></li>
            <li><em>‚ÄúExtraction window used for iRT and m/z correction in Thomson or ppm (see irt_mz_extraction_window_unit)‚Äù</em>: <code class="language-plaintext highlighter-rouge">10.0</code></li>
            <li><em>‚ÄúWhether to run OpenSWATH directly on the input data, cache data to disk first or to perform a datareduction step first‚Äù</em>: <code class="language-plaintext highlighter-rouge">cacheWorkingInMemory</code></li>
            <li><em>‚ÄúUse the retention time normalization peptide MS2 masses to perform a mass correction (linear, weighted by intensity linear or quadratic) of all spectra‚Äù</em>: <code class="language-plaintext highlighter-rouge">regression_delta_ppm</code></li>
          </ul>
        </li>
        <li><em>‚ÄúOptional outputs‚Äù</em>: <code class="language-plaintext highlighter-rouge">out_osw</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-mass-tolerances-and-minimal-number-of-bins-required-to-be-covered"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: Mass tolerances and <em>‚ÄúMinimal number of bins required to be covered‚Äù</em></h3>
        <p>Here we analyze data acquired on a QExactive Plus MS instrument which uses an Orbitrap and generates high resolution data. Other instrumentation (such as TOF devices) might require larger mass tolerances for improved peptide identification. Furthermore, here we require at least 7 of the iRT peptides to be found in each of the DIA measurements. This number can be set to lower values if for some reasons fewer iRT peptides were found in some of the measurements.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h2 id="combining-osw-results-using-pyprophet-merge">Combining osw results using <strong>PyProphet merge</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-combining-the-individual-osw-results-with-pyprophet-merge"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Combining the individual osw results with pyprophet merge</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_merge/pyprophet_merge/2.1.4.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_merge/pyprophet_merge/2.1.4.0"><strong>PyProphet merge</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_merge/pyprophet_merge/2.1.4.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">param-collection</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">out_osw</code> (output of <strong>OpenSwath<span class="notranslate">Workflow</span></strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTemplate osw file‚Äù</em>: <code class="language-plaintext highlighter-rouge">HEK_Ecoli_lib</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="fdr-estimation-using-pyprophet-score">FDR estimation using <strong>PyProphet score</strong></h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-semi-supervised-learning-and-scoring-of-openswathworkflow-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Semi-supervised learning and scoring of OpenSwath<span class="notranslate">Workflow</span> results</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_score/pyprophet_score/2.1.4.2" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_score/pyprophet_score/2.1.4.2"><strong>PyProphet score</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_score/pyprophet_score/2.1.4.2</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">merged.osw</code> (output of <strong>PyProphet merge</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEither a ‚ÄòLDA‚Äô or ‚ÄòXGBoost‚Äô classifier is used for semi-supervised learning‚Äù</em>: <code class="language-plaintext highlighter-rouge">XGBoost</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-fdr-scoring-using-pyprophet-score"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: FDR scoring using pyprophet score</h3>
        <p>During this step q-values corresponding to the FDR of peak identification is estimated with pyprophet. Typically this is the most time consuming step due to the involved maschine learning processes. To decrease the input size one can use <strong>PyProphet subsample</strong> to randomly select subsets of the identifications from each run in the merged.osw (<strong>PyProphet merge</strong> output). In this case, the FDR estimation needs to be applied on the full merged.osw afterwards using the scored subsample.osw in the <em>‚ÄúApply PyProphet score weights file (osw format) instead of semi-supervised learning.‚Äù</em> section of <strong>PyProphet score</strong>. The generated report.pdf is helpful to identify potential errors as well as get first insights on the quality of the identifications.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="tip">
  <h3 id="tip-tip-continue-with-results-from-zenodo"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Continue with results from Zenodo</h3>

  <p>In case the pyprophet score run is not yet finished, the results can be downloaded from Zenodo to be able to continue the tutorial</p>
  <ol>
    <li>Import the files from <a href="https://zenodo.org/record/4301690">Zenodo</a>
      <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>https://zenodo.org/record/4301690/files/PyProphet_score_report.pdf
https://zenodo.org/record/4301690/files/PyProphet_score.osw
</code></pre></div>      </div>
    </li>
  </ol>
</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>Does the distribution of target identifications differ from the decoy distribution?</li>
    <li>Is the observed distribution of decoy and target identifications expected?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>Yes, we can see a clearly different distribution of the target identification and the decoys. Both, target and decoy distribution were highest around 0. However, the target distribution shows a second peak at positiv d-score values.</li>
      <li>The decoy identifications show a Gaussian distribution around 0 which could be explained by the fact that the decoy sequences were randomly generated alterations from the target sequences in the spectral library (see <a href="/training-material/topics/proteomics/tutorials/DIA_lib_OSW/tutorial.html">DIA library generation tutorial</a>). Most target identifications show also d-scores around 0, thus reflect potential false positive identifications. Only the distribution of target identifications shows a second increase in higher d-score values, representing more confident identifications.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="apply-scores-with-pyprophet-peptide">Apply scores with <strong>PyProphet peptide</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-conduct-peptide-inference-in-experiment-wide-and-global-context"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Conduct peptide inference in experiment-wide and global context</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_peptide/pyprophet_peptide/2.1.4.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_peptide/pyprophet_peptide/2.1.4.0"><strong>PyProphet peptide</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_peptide/pyprophet_peptide/2.1.4.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">score.osw</code> (output of <strong>PyProphet score</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúContext to estimate protein-level FDR control‚Äù</em>: <code class="language-plaintext highlighter-rouge">experiment-wide</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_peptide/pyprophet_peptide/2.1.4.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_peptide/pyprophet_peptide/2.1.4.0"><strong>PyProphet peptide</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_peptide/pyprophet_peptide/2.1.4.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">peptide.osw</code> (output of <strong>PyProphet peptide</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúContext to estimate protein-level FDR control‚Äù</em>: <code class="language-plaintext highlighter-rouge">global</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="apply-scores-with-pyprophet-protein">Apply scores with <strong>PyProphet protein</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-conduct-protein-inference-in-experiment-wide-and-global-context"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Conduct protein inference in experiment-wide and global context</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_protein/pyprophet_protein/2.1.4.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_protein/pyprophet_protein/2.1.4.0"><strong>PyProphet protein</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_protein/pyprophet_protein/2.1.4.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">peptide.osw</code> (output of the second <strong>PyProphet peptide</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúContext to estimate protein-level FDR control‚Äù</em>: <code class="language-plaintext highlighter-rouge">experiment-wide</code></li>
      </ul>
    </li>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_protein/pyprophet_protein/2.1.4.0" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_protein/pyprophet_protein/2.1.4.0"><strong>PyProphet protein</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_protein/pyprophet_protein/2.1.4.0</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">protein.osw</code> (output of <strong>PyProphet protein</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúContext to estimate protein-level FDR control‚Äù</em>: <code class="language-plaintext highlighter-rouge">global</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How does the score distribution differ between the peptide and the protein inference (in global context)?</li>
    <li>What could be the reason for this difference?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>The d-score distribution and the ROC curve for the protein inference shows better target decoy discriminatory ability than for the peptide inference.</li>
      <li>The probabilty of multiple decoy peptides from the same protein having a high score is lower than for multiple target peptides from the same protein. Thus, the discriminatory scores on protein level are higher.</li>
    </ol>

  </blockquote>

</blockquote>

<h2 id="exporting-the-results-with-pyprophet-export">Exporting the results with <strong>PyProphet export</strong></h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-exporting-pyprophet-scored-osw-results"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Exporting pyprophet scored OSW results</h3>

  <ol>
    <li><span class="tool" data-tool="toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_export/pyprophet_export/2.1.4.1" title="Tested with toolshed.g2.bx.psu.edu/repos/galaxyp/pyprophet_export/pyprophet_export/2.1.4.1"><strong>PyProphet export</strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: <span class="notranslate">toolshed</span>.g2.bx.psu.edu/repos/<span class="notranslate">galaxy</span>p/pyprophet_export/pyprophet_export/2.1.4.1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput file‚Äù</em>: <code class="language-plaintext highlighter-rouge">protein.osw</code> (output of <strong>PyProphet protein</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúExport format, either matrix, legacy_split, legacy_merged (mProphet/PyProphet) or score_plots format‚Äù</em>: <code class="language-plaintext highlighter-rouge">legacy_merged</code></li>
        <li><em>‚ÄúUse swath2stats to export file for statsics‚Äù</em>: <code class="language-plaintext highlighter-rouge">yes</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúStudy design tabular file‚Äù</em>: <code class="language-plaintext highlighter-rouge">Sample_annotation</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-swath2stats-functionality"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment: swath2stats functionality</h3>
        <p>Utilizing the swath2stats functionality generates a summary as well as a peptide and protein <span class="notranslate">expression</span> matrix. In addition to the non-processed pyprophet tabular output a refined tabular is generated using the specified criteria. The refined tabular <code class="language-plaintext highlighter-rouge">msstats_input.tabular</code> is dircetly compatable with <strong>MSstats</strong> for statistical analysis.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <ol>
    <li>How many different peptides and proteins were identified and quatified?</li>
    <li>Could you already tell from the summary which Spike-in contained higher amounts of Ecoli peptides?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>In total, over 27.300 peptides and over 5.100 proteins were identified and quantified in the DIA measurements.</li>
      <li>No, the summary mainly provides an overview of the identifications in each individual DIA measurement as well as some descriptive statistics such as CVs and correlations.</li>
    </ol>

  </blockquote>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-analysis-of-ecoli-spike-in"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-On: Analysis of Ecoli Spike-in</h3>

  <ol>
    <li><span class="tool" data-tool="Grep1" title="Tested with Grep1"><strong>Select lines that match an <span class="notranslate">expression</span> </strong> <i class="fas fa-wrench" aria-hidden="true"></i><i aria-hidden="true" class="fas fa-cog"></i><span class="visually-hidden">Tool: Grep1</span></span>  with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect lines from‚Äù</em>: <code class="language-plaintext highlighter-rouge">protein_signal.tabular</code> (output of <strong>PyProphet export</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚Äúthat‚Äù</em>: <code class="language-plaintext highlighter-rouge">Matching</code></li>
        <li><em>‚Äúthe pattern‚Äù</em>: <code class="language-plaintext highlighter-rouge">(ECOLI)|(Spike_in)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>
  <ol>
    <li>How many Ecoli proteins were identified and quantified in the six DIA measurements?</li>
    <li>Can you guess which Spike-in contains higher amounts of Ecoli peptides?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
    <ol>
      <li>Over 800 Ecoli proteins were identified and quantified in the six DIA measurements.</li>
      <li>It seems that the samples in Spike_in_2 contained higher amounts of Ecoli peptides than the samples in Spike_in_1.</li>
    </ol>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<figure id="figure-4"><img src="../../images/DIA_analysis_sum.png" alt="DIA_analysis_pipe" /><figcaption><span class="figcaption-prefix">Figure 4:</span> All-in one <span class="notranslate">workflow</span> for DIA analysis in <span class="notranslate">Galaxy</span>. The DIA data analysis using OpenSwath<span class="notranslate">Workflow</span> is highlighted in yellow.</figcaption></figure>

<p>Using <strong>OpenSwath<span class="notranslate">Workflow</span></strong> and a spectral library (from DDA data) we were able to analyze data independent acquisition (DIA) data. Furthermore, we were able to process the FDR scored results and take a guess on which Spike-in contained higher amounts of Ecoli. To further investigate the two different Spike-in amounts as well as getting a significant result we would need to perform statistical analysis e.g. using <strong>MSstats</strong>.</p>

<figure id="figure-5"><img src="../../images/DIA_analysis_workflow.png" alt="DIA_OSW_galaxy" /><figcaption><span class="figcaption-prefix">Figure 5:</span> <span class="notranslate">Galaxy</span> worfklow containing all necessary steps for the analysis of DIA data using a spectral library and OpenSwath<span class="notranslate">Workflow</span></figcaption></figure>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>OpenSwathWorkflow enables analysis of data independent acquisition mass spectrometry data</p>
</li>
                            
                            <li><p>DIA data analysis of HEK-Ecoli Spike-in dataset</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/proteomics#references">here</a>.</p>
                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Rost2014">R√∂st, H. L., G. Rosenberger, P. Navarro, L. Gillet, S. M. Miladinoviƒá <i>et al.</i>, 2014 <b>OpenSWATH enables automated, targeted analysis of data-independent acquisition MS data.</b> Nature biotechnology 32: 219‚Äì23.</span>
    
    
    <a href="https://doi.org/10.1038/nbt.2841">10.1038/nbt.2841</a>
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="http://www.ncbi.nlm.nih.gov/pubmed/24727770">http://www.ncbi.nlm.nih.gov/pubmed/24727770</a>
      
    

</li>
<li><span id="Ludwig2018">Ludwig, C., L. Gillet, G. Rosenberger, S. Amon, B. C. Collins <i>et al.</i>, 2018 <b> Data‚Äêindependent acquisition‚Äêbased SWATH ‚Äê MS for quantitative proteomics: a tutorial </b>. Molecular Systems Biology 14: 1‚Äì23.</span>
    
    
    <a href="https://doi.org/10.15252/msb.20178126">10.15252/msb.20178126</a>
    
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=DIA Analysis using OpenSwathWorkflow (Proteomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Matthias Fahrner, Melanie F√∂ll, 2021 <b>DIA Analysis using OpenSwathWorkflow (Galaxy Training Materials)</b>. <a href="/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html">/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{proteomics-DIA_Analysis_OSW,
    author = "Matthias Fahrner and Melanie F√∂ll",
    title = "DIA Analysis using OpenSwathWorkflow (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Matthias Fahrner, Melanie F√∂ll)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/proteomics/tutorials/DIA_Analysis_OSW/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
