<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Slides: Plates, Batches, and Barcodes / Plates, Batches, and Barcodes / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@4dcf4c69d322d6ae99952dbf87a02edf19169088",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides-plain.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML" as="script" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Plates, Batches, and Barcodes">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="2023-10-10 13:37:19 +0000">
<meta name="DC.creator" content="Wendi Bacon">
<meta name="DC.creator" content="Mehmet Tekman">
<meta name="DC.creator" content="Alex Ostrovsky"><meta name="description" content="### Sorting Plates   .image-100[![slide5](../../images/scrna-intro/wab96wellplate.svg)]  .left[Plates are *N x M* arrays of wells that cells are sorted, to then be individually amplified and sequenced.]  ??? - Sorting plates are 2 dimensional arrays of wells that individual cells are placed into. - Each cell in each well is tagged with a unique barcode and then primed to then be amplified and sequenced - The method in which this is performed can yield significant pros and cons as we will see.  ---  ### Sorting Plates  .image-80[![slide6](../../images/scrna-intro/wabexampleplates.png)]  ??? - Here we see 3 different sorting strategies with different advantages for same treatment sample. - In the top image: We see all wells are filled, to maximise the number of cells on a plate - In the middle image: We see only the inner wells are filled, to maintain a protective border during library preparation - In the bottom image: We see a striping pattern, which mitigates contamination of mater...">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Plates, Batches, and Barcodes / Slides: Plates, Batches, and Barcodes">
        <meta property="og:description" content="### Sorting Plates   .image-100[![slide5](../../images/scrna-intro/wab96wellplate.svg)]  .left[Plates are *N x M* arrays of wells that cells are sorted, to then be individually amplified and sequenced.]  ??? - Sorting plates are 2 dimensional arrays of wells that individual cells are placed into. - Each cell in each well is tagged with a unique barcode and then primed to then be amplified and sequenced - The method in which this is performed can yield significant pros and cons as we will see.  ---  ### Sorting Plates  .image-80[![slide6](../../images/scrna-intro/wabexampleplates.png)]  ??? - Here we see 3 different sorting strategies with different advantages for same treatment sample. - In the top image: We see all wells are filled, to maximise the number of cells on a plate - In the middle image: We see only the inner wells are filled, to maintain a protective border during library preparation - In the bottom image: We see a striping pattern, which mitigates contamination of mater...">
        <meta property="og:image" content="https://training.galaxyproject.org/training-material/assets/images/GTNLogo1000.png"></head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/./topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides-plain.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        <section class="tutorial">
	<a href="https://github.com/galaxyproject/training-material/blob/main//topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html">View markdown source on GitHub</a>
	<h1>Plates, Batches, and Barcodes</h1>
		<h2>Contributors</h2>
<div markdown="0">

	<div class="contributors-line">
		
  <a href="/training-material/hall-of-fame/nomadscientist/" class="contributor-badge contributor-nomadscientist"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/nomadscientist?s=36" alt="Wendi Bacon avatar" width="36" class="avatar" />
    Wendi Bacon</a>
  
  <a href="/training-material/hall-of-fame/mtekman/" class="contributor-badge contributor-mtekman"><img src="https://avatars.githubusercontent.com/mtekman?s=36" alt="Mehmet Tekman avatar" width="36" class="avatar" />
    Mehmet Tekman</a>
  
  <a href="/training-material/hall-of-fame/astrovsky01/" class="contributor-badge contributor-astrovsky01"><img src="/training-material/assets/images/orcid.png" alt="orcid logo" width="36" height="36"/><img src="https://avatars.githubusercontent.com/astrovsky01?s=36" alt="Alex Ostrovsky avatar" width="36" class="avatar" />
    Alex Ostrovsky</a>
  
	</div>

</div>


		
		<h2>Questions</h2>
		<ul>
		
		<li><p>What is a batch when it comes to single cells?</p>
</li>
		
		<li><p>What’s the difference between a barcode and an index?</p>
</li>
		
		<li><p>What is batch effect and how do you prevent it?</p>
</li>
		
		<li><p>What is a lane?</p>
</li>
		
		</ul>
		

		
		<h2>Objectives</h2>
		<ul>
		
		<li><p>How to set up plates to prevent batch effect</p>
</li>
		
		<li><p>Proper naming conventions when dealing with scRNA-seq samples</p>
</li>
		
		</ul>
		

		

		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Published:</strong> Feb 16, 2019 </div>
		<div><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last Updated:</strong> Oct 10, 2023 </div>

	<hr />


	<h3 id="sorting-plates">Sorting Plates</h3>

<p>.image-100[<img src="../../images/scrna-intro/wab96wellplate.svg" alt="slide5" />]</p>

<p>.left[Plates are <em>N x M</em> arrays of wells that cells are sorted, to then be individually amplified and sequenced.]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Sorting plates are 2 dimensional arrays of wells that individual cells are placed into.</li>
  <li>Each cell in each well is tagged with a unique barcode and then primed to then be amplified and sequenced</li>
  <li>The method in which this is performed can yield significant pros and cons as we will see.</li>
</ul>

<hr />

<h3 id="sorting-plates-1">Sorting Plates</h3>

<p>.image-80[<img src="../../images/scrna-intro/wabexampleplates.png" alt="slide6" />]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Here we see 3 different sorting strategies with different advantages for same treatment sample.</li>
  <li>In the top image: We see all wells are filled, to maximise the number of cells on a plate</li>
  <li>In the middle image: We see only the inner wells are filled, to maintain a protective border during library preparation</li>
  <li>In the bottom image: We see a striping pattern, which mitigates contamination of material from adjoining wells</li>
</ul>

<hr />

<h3 id="setting-up-plates">Setting up Plates</h3>

<p>.image-75[<img src="../../images/scrna-intro/wabbatchproblems.png" alt="slide7" />]</p>

<p>.center[What is the problem with this plate setup?]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>However, if we consider different treatment samples, then we must consider different challenges.</li>
  <li>What is the advantage of the top example, i.e. Several different treatments on one plate?</li>
  <li>What is the advantage of the bottom example, i.e. One treatment on one plate?</li>
</ul>

<hr />

<h3 id="setting-up-plates-1">Setting up Plates</h3>

<p>.image-50[<img src="../../images/scrna-intro/wabbatchproblems.png" alt="slide7" />]</p>

<p>.center[Batch effect (plate vs plate) cannot be separated from treatment effect in either scenario.]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Trick question!</li>
  <li>In both examples, the treatment effect cannot be separated from any technical effect from the plate itself.</li>
  <li>For example, the biological effect of the yellow cells might just be the technical confounding effect from Batch 1.</li>
  <li>What can we do instead to reduce the effect of these technical confounding effects across treatments?</li>
</ul>

<hr />
<h3 id="setting-up-plates-2">Setting up Plates</h3>

<p>.image-50[<img src="../../images/scrna-intro/wabbalancedbatches.png" alt="slide7" />]</p>

<p>.left[Either of these are better set-ups. Mixing columns is good, but not required. Ultimately, batch effect can now be separated from variable effect.]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>The answer is that we can balance the treatments across our plates better.</li>
  <li>By ensuring the same treatment occurs in comparable amounts in different plates, we can reduce the effect that each plate would have on the treatment.</li>
</ul>

<hr />

<h3 id="setting-up-plates-3">Setting up Plates</h3>

<p>.image-90[<img src="../../images/scrna-intro/wabreplicates.png" alt="slide7" />]</p>

<p>.center[Can’t mix samples on plates? Separate replicates evenly and process together.]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>If putting multiple treatments on the same plate is not an option, then having enough separated replicates will also allow for batch correction.</li>
  <li>Here, you can assess the variation between replicates, and thus batches, as well as between treatments.</li>
</ul>

<hr />

<h3 id="what-about-sequencing-lanes">What about sequencing lanes?</h3>

<p>.center[So now it’s time to sequence our samples! How do we combine samples into sequencing lanes?]</p>

<p>.image-90[<img src="../../images/scrna-intro/wabseqlane.png" alt="slide7" />]</p>

<p>.center[This works well, but what if you have too many samples for one lane?]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>So now it’s time to sequence our samples! How do we combine samples into sequencing lanes?</li>
  <li>In many cases, it is enough to sequence 1 or 2 plates together in a single sequencing lane.</li>
  <li>But what if we wish to sequence 8 plates?</li>
</ul>

<hr />

<h3 id="what-about-sequencing-lanes-1">What about sequencing lanes?</h3>

<p>.center[So now it’s time to sequence our samples! How do we combine samples into sequencing lanes?]</p>

<p>.image-90[<img src="../../images/scrna-intro/wablanesbad.png" alt="slide7" />]</p>

<p>.center[Does this look ok?]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>In the example show, Treatment A would go to the first sequencing lane.</li>
  <li>Treatment B would go to the second sequencing lane.</li>
  <li>Is it okay to sequence different treatments across different sequencing lanes?</li>
</ul>

<p>–</p>

<p>.center[No! You’ve turned each treatment (A &amp; B) into a batch!]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>No, because you’ve turned Treatment A and Treatment B into separate batches.</li>
  <li>This gives us the same unbalanced design we saw with the sorting plates.</li>
</ul>

<hr />

<h3 id="what-about-sequencing-lanes-2">What about sequencing lanes?</h3>

<p>.image-75[<img src="../../images/scrna-intro/wablanesgood.png" alt="slide7" />]</p>

<p>.center[This is the way to balance your batches at the lane-level.]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Instead we should balance the availability of each treatment across different sequencing lanes</li>
  <li>This way, each treatment is sequenced at the same time as another treatment, reducing the technical effect from the sequencing on the treatment.</li>
</ul>

<hr />

<h3 id="distinguishing-cells-in-a-plate">Distinguishing cells in a plate</h3>

<p>.image-60[<img src="../../images/scrna-intro/wabplate.png" alt="slide11" />]</p>

<ul>
  <li>Cells are selected from a plate by their <em>barcodes</em></li>
  <li>Barcodes must be unique
    <ul>
      <li>e.g. 96 wells in a plate, need 96 barcodes to sequence them together</li>
    </ul>
  </li>
</ul>

<p><span>Speaker Notes</span></p>
<ul>
  <li>How do we distinguish each cell in each well?</li>
  <li>Do we remember the physical location of that cell on that well, or do we tag it somehow?</li>
  <li>The answer is that each cell is given a unique barcode that distinguishes from other cells which have different cell barcodes.</li>
</ul>

<hr />

<h3 id="what-are-cell-barcodes">What are Cell Barcodes?</h3>

<p>.image-100[<img src="../../images/scrna-intro/scrna_pbb_barcodes_add.svg" alt="Add Barcodes" />]</p>

<ul>
  <li>Barcodes are added to <em>all</em> transcripts of a <em>specific</em> cell</li>
  <li>Transcripts with different cell barcodes originate from different cells</li>
</ul>

<p><span>Speaker Notes</span></p>
<ul>
  <li>What exactly are cell barcodes?</li>
  <li>These are usually short nucleotide barcodes that are added to all transcripts of a specific cell</li>
  <li>The idea is that transcripts with different cell barcodes originate from different cells.</li>
</ul>

<hr />

<h3 id="what-are-cell-barcodes-1">What are Cell Barcodes?</h3>

<p>.image-100[<img src="../../images/scrna-intro/scrna_pbb_barcodes_overview.svg" alt="Barcodes in Plate" />]</p>

<ul>
  <li>Many different cell barcodes are used across many different cells</li>
  <li>Each well in a plate contains a cell, indexed by its cell barcode</li>
</ul>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Within a plate, many different barcodes are used, and each cell is indexed by the cell barcode assigned to that well.</li>
</ul>

<hr />

<h3 id="questions-about-cell-barcodes">Questions about Cell Barcodes</h3>

<p>.image-50[<img src="../../images/scrna-intro/wabplate.png" alt="slide14" />]</p>

<p>.left[Assuming you sequence one 96-well plate:]</p>

<ol>
  <li>
    <p>How many cell barcodes are needed for a single lane?</p>
  </li>
  <li>
    <p>How many cell barcodes are needed if you combine 10 plates into a single sequencing lane?</p>
  </li>
  <li>
    <p>What would be the minimum length of the barcodes for each of the previous questions?</p>
  </li>
</ol>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Consider a 12 by 8 plate with 96 wells.</li>
  <li>How many cell barcodes are needed if you sequence a single plate?</li>
  <li>How many cell barcodes are needed if you sequence 10 plates into a single sequencing lane?</li>
  <li>What would be the minimum length of the barcodes for each of the previous questions?</li>
</ul>

<p>–</p>

<p>.footnote[</p>
<ol>
  <li><em>96 unique barcodes per lane</em></li>
  <li><em>96 x 10 = 960 unique barcodes per plate</em>
]</li>
</ol>

<p><span>Speaker Notes</span></p>
<ul>
  <li>For one plate you would just need 96 unique barcodes</li>
  <li>For 10 plates you would need 960 unique barcodes</li>
</ul>

<hr />

<h3 id="questions-about-cell-barcodes-1">Questions about Cell Barcodes</h3>

<p>.left[<small>3) What would be the minimum length of the barcodes for each of the previous questions?</small>]</p>

<p>.pull-left[</p>

<p>A single lane?</p>

<ul>
  <li>96 barcodes, 4 nucleotides for each base of a barcode</li>
</ul>

<table>
  <thead>
    <tr>
      <th>Barcodes</th>
      <th>Result</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>\(4^2 =  16\)</td>
      <td><small>No, 2 bases is not enough</small></td>
    </tr>
    <tr>
      <td>\(4^3  = 64\)</td>
      <td><small>No, 3 bases is not enough</small></td>
    </tr>
    <tr>
      <td>\(4^4 = 256\)</td>
      <td><small>Yes, 4 bases is enough to cover 96 barcodes (and more!)</small></td>
    </tr>
  </tbody>
</table>

<p>]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>To answer the third question, we need to make some assumptions about the barcodes.</li>
  <li>If we assume a barcode is made up of a chain of 4 possible nucleotides, then we can solve the problem as a power of 4.</li>
  <li>For a single lane with only 96 barcodes, we require our barcodes to be of at least 4 base pairs in length.</li>
</ul>

<p>–
.pull-right[</p>

<p>10 plates in a single lane?</p>

<ul>
  <li>960 barcodes, 4 nucleotides for each base of a barcode</li>
</ul>

<table>
  <thead>
    <tr>
      <th>Barcodes</th>
      <th>Result</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>\(4^4 = 256\)</td>
      <td><small>No, 4 bases is not enough</small></td>
    </tr>
    <tr>
      <td>\(4^5 = 1024\)</td>
      <td><small>Yes, 5 bases is enough to cover 960 barcodes (just barely!)</small></td>
    </tr>
  </tbody>
</table>

<p>]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>For 10 lanes with 960 barcodes, we require our barcodes to be of at least 5 base pairs in length.</li>
  <li>However these calculations also make one serious assumption about the barcodes.</li>
</ul>

<hr />

<h3 id="barcode-safeguarding">Barcode Safeguarding</h3>

<p>.pull-left[</p>
<ul>
  <li>
    <p>Is 5 nucleotides really enough to capture 960 cells?</p>
  </li>
  <li>
    <p>What could go wrong if all barcodes are separated by 1 bp, as shown?
]</p>
  </li>
</ul>

<p>.pull-right[
<small><em>Edit distance = 1bp</em></small></p>

<div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>AAAAA AAAAC AAAAG AAAAT AAACA AAAGA AAATA ····
CCCCC CCCCA CCCCG CCCCT CCCAC CCCGC CCCTC ····
                    ·
                    ·
                    · ]
</code></pre></div></div>

<p><span>Speaker Notes</span></p>
<ul>
  <li>For example, we assume that all our barcodes are at minimum separated by a single base pair.</li>
  <li>This is known as an edit distance of 1.</li>
  <li>If we assume that every barcode is separated from every other barcode by 1 base pair, then 5 nucleotides is indeed enough to delineate 960 cells.</li>
  <li>However if there was even just one small sequencing error of 1 base pair on any barcode, it would immediately change that erronous barcode into another real barcode.</li>
  <li>That is to say, we have no way of detecting sequencing errors if we use an edit distance of 1.</li>
</ul>

<hr />

<h3 id="guarding-against-sequencing-errors">Guarding against Sequencing Errors</h3>

<p>.pull-left[</p>
<ul>
  <li>
    <p>Sequencing errors can change the cell barcode</p>
  </li>
  <li>
    <p>To guard against a 1 bp sequencing error, need to space barcodes by 2 bp</p>
  </li>
  <li>
    <p>How many barcodes for a cell barcode of length 5 bp (ED = 2)?
]
.pull-right[
<small><em>Edit Distance = 2 bp</em></small></p>

    <p>AAAAA AAACC AAAGG AAATT AACCA AAGGA AATTA ····
  CCCCC CCCAA CCCGG CCCTT CCCAA CCGGC CCTTC ····
                      ·
                      ·
                      ·</p>
  </li>
</ul>

<p>]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Let us try to rectify this by increasing the edit distance to 2.</li>
  <li>In the example barcodes shown we see barcodes of length 5, separated by 2 base pairs.</li>
  <li>How many unique cell barcodes can we make given these restrictions?</li>
</ul>

<p>–</p>

\[4^{5-1} = 512\]

<p><span>Speaker Notes</span></p>
<ul>
  <li>The answer is half the amount that we need.</li>
</ul>

<hr />

<h3 id="edit-distance--general-principle">Edit Distance : General Principle</h3>

<p>.pull-left[
e.g. For barcodes of length N=3:</p>

<ul>
  <li>
    <p>Edit Distance of E=1:
.center[<code class="language-plaintext highlighter-rouge">AAA AAC AAG AAT ACA ACC</code> …]
.center[<small>64 barcodes</small>]</p>
  </li>
  <li>
    <p>Edit Distance of E=2:
.center[<code class="language-plaintext highlighter-rouge">AAA ACC AGG ATT CAA CCC</code> …]
.center[<small>16 barcodes</small>]</p>
  </li>
  <li>
    <p>Edit Distance of E=3:
.center[<code class="language-plaintext highlighter-rouge">AAA CCC GGG TTT</code>]
.center[<small>4 barcodes</small>]
]</p>
  </li>
</ul>

<p><span>Speaker Notes</span></p>
<ul>
  <li>Let us explore this more explicitly.</li>
  <li>Here we will be using barcodes only of length 3, and we will explore different edit distances of 1, 2, and 3.</li>
  <li>These yield 64, 16, and 4 barcodes respectively.</li>
</ul>

<p>–
.pull-right[
Number of Barcodes :</p>

\[4^{N-(E-1)}\]

<p>.center[For barcodes of length <em>N</em>,
and Edit Distance of <em>E</em>.]</p>

<p>]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>This can be summarized by the formula given, where the length of the barcodes, minus one minus the edit distance, gives the number of available barcodes.</li>
</ul>

<hr />

<h3 id="how-many-available-barcodes-are-there">How many available barcodes are there?</h3>

<p>.pull-left[</p>
<ul>
  <li>
    <p>Barcodes typically limited to 4 main bases {A,C,T,G}.</p>
  </li>
  <li>
    <p>Number of available barcodes depends on:</p>

    <ol>
      <li>
        <p>The length of the barcode</p>
      </li>
      <li>
        <p>The edit distance between adjacent barcodes</p>
      </li>
    </ol>
  </li>
  <li>
    <p>Must take sequencing errors into account</p>
  </li>
</ul>

<p>]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>To summarize, barcodes are typically limited to 4 nucleotides, and the number of available barcodes depends on the length and edit distance.</li>
  <li>This is to ensure that sequencing errors are taken into account.</li>
</ul>

<p>–
.pull-right[</p>
<ul>
  <li>
    <p>Availability is balance between <em>barcode size</em> vs. <em>sequencing errors</em></p>
  </li>
  <li>
    <p>Typically between 4-10 bases, range of cells that can be labelled:</p>

\[[4^4, 4^{10}] = [256, 1048576]\]
  </li>
  <li>
    <p>True range is lower due to edit distance.
]</p>
  </li>
</ul>

<p><span>Speaker Notes</span></p>
<ul>
  <li>The availability of barcodes and the number of cells they can label is a design choice, where the technician must balance two opposing forces: barcode size vs sequencing errors.</li>
  <li>For the size, this means that for a barcode N bases long, there will 4 to the N barcodes available.</li>
  <li>Typically barcodes tend to span 4 to 10 bases, since longer barcodes tend to be more subjectable to sequencing errors.</li>
  <li>The true number of barcodes used is smaller than 4 to the N, due to the countermeasures used to space barcodes apart from one another, in order to reduce sequencing errors.</li>
</ul>

<hr />

<h3 id="design-factors-in-barcodes">Design Factors in Barcodes</h3>

<p>Is it better to use Longer or Shorter Barcodes?</p>

<table>
  <thead>
    <tr>
      <th>Longer</th>
      <th>Shorter</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Offer larger range of barcodes</td>
      <td>Offer smaller range of barcodes</td>
    </tr>
    <tr>
      <td>Prone to more sequencing errors</td>
      <td>More robust against sequencing errors</td>
    </tr>
    <tr>
      <td>Must increase edit distance significantly</td>
      <td>A smaller edit distance is more acceptable</td>
    </tr>
    <tr>
      <td>Can accommodate large edit distances</td>
      <td><em>Cannot</em> accommodate large edit distances</td>
    </tr>
  </tbody>
</table>

<p><span>Speaker Notes</span></p>
<ul>
  <li>To weigh the pros and cons of longer and shorter barcodes, we need to take into account the likelihood of sequencing errors for a given size, and the resulting barcode address space given by the edit distance.</li>
</ul>

<hr />

<h3 id="cell-barcodes-summary">Cell Barcodes: Summary</h3>

<p>.pull-left[</p>

<ul>
  <li>
    <p>A single barcode sequence indexes a single cell</p>
  </li>
  <li>
    <p>Every transcript in a specific cell has the same cell barcode</p>
  </li>
  <li>
    <p>Barcodes are <em>designed</em> for smaller plate-based protocols, while for split-pool and similar techniques they are randomised.</p>
  </li>
  <li>
    <p>Barcode use is <em>limited</em> by length and read depth</p>
  </li>
</ul>

<p>]</p>

<p><span>Speaker Notes</span></p>
<ul>
  <li>To summarize cell barcodes, we should note the following</li>
  <li>A single barcode sequence indexes a single cell</li>
  <li>Every transcript in a specific cell has the same cell barcode</li>
  <li>Barcodes are designed for smaller plate-based protocols, while for split-pool and similar techniques they are randomised</li>
  <li>Barcode use is limited by length and read depth</li>
</ul>

<p>–</p>
<hline>
.pull-right[#### Need to balance:

<small>

  | | | |
|-|-|-|
| Edit Distance | vs. | Barcode Length |
| Number of Barcodes | vs. | Plate and Lane Size |

</small>
]

<span>Speaker Notes</span>
- The number of reads wanted per cell determines how many cells you run in a sequencing lane, which in turn tells you how many barcodes you need.

---

# Summary

.pull-left[

* Cell Barcodes are sequences attached to transcripts to indicate what cell a transcript came from

* Cell Barcodes are *designed* to the Plate/Lane setup

]

<span>Speaker Notes</span>
- From the content shown here, you have learned the following.
- Cell Barcodes are sequences attached to transcripts to indicate what cell a transcript came from
- Cell Barcodes are designed for the Plate and Lane setup.


--
.pull-right[

* Indicates what cell a transcript came from.

* Reduce sequencing errors

]

<span>Speaker Notes</span>
- Cell barcodes indicate what cell a transcript came from
- They reduce sequencing errors when spaced appropriately
</hline>


	<hr />


<h2>Key Points</h2>

<ul>

<li>Balanced batches and replicates allow bioinformatic batch correction</li>

<li>A sequencing lane often contains multiple batches, and is itself a batch effect!</li>

</ul>







<h2>Thank you!</h2>

This material is the result of a collaborative work. Thanks to the <a href="https://training.galaxyproject.org" aria-label="Visit the GTN">Galaxy Training Network</a> and all the contributors!


<img src="/training-material/assets/images/GTNLogo1000.png" alt="Galaxy Training Network" style="height: 100px;"/>


Tutorial Content is licensed under

  <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.

</section>






        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/./topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/./topics/single-cell/tutorials/scrna-plates-batches-barcodes/slides.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/4dcf4c69d322d6ae99952dbf87a02edf19169088">4dcf4c6</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
	
	

    </body>
</html>
