---
layout: tutorial_slides_ai4life
logo: topics/ai4life/images/AI4Life-logo_giraffe-nodes.png
video: true

title: "Example Slide Deck"
tags:
  - some
  - tags
questions:
  - How to visualize your genome after automated annotations have been performed?
  - How to manually annotate genome after automated annotations have been performed?
objectives:
  - Load a genome into Galaxy
  - View annotations in JBrowse
time_estimation: "5m"
key_points:
  - Apollo is the Google Docs of the genome annotation world, real-time collaborative genome annotation.
contributors:
  - beatrizserrano
---

## Example slide

Example contents!

???

Speaker notes go here

---

### Genome annotation

- Classifies and describes parts of the genome sequence
- Focusing on genes: their positions, names and functions
- Automatic methods to annotate a newly sequenced and assembled genome

    - Prokaryotes: [Prokka]({% link topics/genome-annotation/tutorials/annotation-with-prokka/tutorial.md %}), Augustus
    - Eukaryotes: [Maker]({% link topics/genome-annotation/tutorials/annotation-with-maker/tutorial.md %}), Braker2, Augustus

.image-50[![JBrowse screenshot showing several genes aligned along a genome](../../images/intro-structural-annotation.png)]

???

- Annotating a genome means positioning features along the sequence of a genome.
- Here we focus on genes.
- We are interested in their positions, and functional annotation like their name of their function.
- Many software tools and pipelines exist to automatically perform the annotation of a new genome sequence.
- For example Prokka or Maker are widely used, and tutorials are available on the GTN to learn how to use them.

