<!DOCTYPE html>

<html>
<head>
<meta charset="UTF-8">

<title>schema-topic.yaml - RDoc Documentation</title>

<script type="text/javascript">
  var rdoc_rel_prefix = "../";
  var index_rel_prefix = "../";
</script>

<script src="../js/navigation.js" defer></script>
<script src="../js/search.js" defer></script>
<script src="../js/search_index.js" defer></script>
<script src="../js/searcher.js" defer></script>
<script src="../js/darkfish.js" defer></script>

<link href="../css/fonts.css" rel="stylesheet">
<link href="../css/rdoc.css" rel="stylesheet">




<body id="top" role="document" class="file">
<nav role="navigation">
  <div id="project-navigation">
    <div id="home-section" role="region" title="Quick navigation" class="nav-section">
  <h2>
    <a href="../index.html" rel="home">Home</a>
  </h2>

  <div id="table-of-contents-navigation">
    <a href="../table_of_contents.html#pages">Pages</a>
    <a href="../table_of_contents.html#classes">Classes</a>
    <a href="../table_of_contents.html#methods">Methods</a>
  </div>
</div>

    <div id="search-section" role="search" class="project-section initially-hidden">
  <form action="#" method="get" accept-charset="utf-8">
    <div id="search-field-wrapper">
      <input id="search-field" role="combobox" aria-label="Search"
             aria-autocomplete="list" aria-controls="search-results"
             type="text" name="search" placeholder="Search" spellcheck="false"
             title="Type to search, Up and Down to navigate, Enter to load">
    </div>

    <ul id="search-results" aria-label="Search Results"
        aria-busy="false" aria-expanded="false"
        aria-atomic="false" class="initially-hidden"></ul>
  </form>
</div>

  </div>

  

  <div id="project-metadata">
    <div id="fileindex-section" class="nav-section">
  <h3>Pages</h3>

  <ul class="link-list">
  
    <li><a href="../bin/add_galaxy_instance_annotations_py.html">add_galaxy_instance_annotations.py</a>
  
    <li><a href="../bin/add_galaxy_instance_badges_py.html">add_galaxy_instance_badges.py</a>
  
    <li><a href="../bin/ari-make_sh.html">ari-make.sh</a>
  
    <li><a href="../bin/ari-quick_sh.html">ari-quick.sh</a>
  
    <li><a href="../bin/ari_sh.html">ari.sh</a>
  
    <li><a href="../bin/check-broken-boxes_py.html">check-broken-boxes.py</a>
  
    <li><a href="../bin/check_instance_py.html">check_instance.py</a>
  
    <li><a href="../bin/data-library-update_sh.html">data-library-update.sh</a>
  
    <li><a href="../bin/data_libarary_download_sh.html">data_libarary_download.sh</a>
  
    <li><a href="../bin/docker-install-tutorials_sh.html">docker-install-tutorials.sh</a>
  
    <li><a href="../bin/gtn-fix_sh.html">gtn-fix.sh</a>
  
    <li><a href="../bin/install_topic_requirements_sh.html">install_topic_requirements.sh</a>
  
    <li><a href="../bin/install_tutorial_requirements_sh.html">install_tutorial_requirements.sh</a>
  
    <li><a href="../bin/knit-automated_sh.html">knit-automated.sh</a>
  
    <li><a href="../bin/knit-frog_py.html">knit-frog.py</a>
  
    <li><a href="../bin/knit_py.html">knit.py</a>
  
    <li><a href="../bin/knittingneedles_py.html">knittingneedles.py</a>
  
    <li><a href="../bin/lint-diffs_py.html">lint-diffs.py</a>
  
    <li><a href="../bin/lunr-index_js.html">lunr-index.js</a>
  
    <li><a href="../bin/lunr-search_js.html">lunr-search.js</a>
  
    <li><a href="../bin/mergeyaml_py.html">mergeyaml.py</a>
  
    <li><a href="../bin/prepare_docker_checks_py.html">prepare_docker_checks.py</a>
  
    <li><a href="../bin/publish-archive.html">publish-archive</a>
  
    <li><a href="../bin/schema-contributors_yaml.html">schema-contributors.yaml</a>
  
    <li><a href="../bin/schema-faq_yaml.html">schema-faq.yaml</a>
  
    <li><a href="../bin/schema-quiz_yaml.html">schema-quiz.yaml</a>
  
    <li><a href="../bin/schema-requirement-external_yaml.html">schema-requirement-external.yaml</a>
  
    <li><a href="../bin/schema-requirement-internal_yaml.html">schema-requirement-internal.yaml</a>
  
    <li><a href="../bin/schema-slides_yaml.html">schema-slides.yaml</a>
  
    <li><a href="../bin/schema-topic_yaml.html">schema-topic.yaml</a>
  
    <li><a href="../bin/schema-tutorial_yaml.html">schema-tutorial.yaml</a>
  
    <li><a href="../bin/setup_training_content_py.html">setup_training_content.py</a>
  
    <li><a href="../bin/slides-fix_css.html">slides-fix.css</a>
  
    <li><a href="../bin/validate-has-workflow_sh.html">validate-has-workflow.sh</a>
  
    <li><a href="../bin/video-browser-recorder_js.html">video-browser-recorder.js</a>
  
    <li><a href="../bin/video-builder_py.html">video-builder.py</a>
  
    <li><a href="../bin/video-diffplayer_py.html">video-diffplayer.py</a>
  
    <li><a href="../bin/video-extract-script_py.html">video-extract-script.py</a>
  
    <li><a href="../bin/video-term-demo-magic_sh.html">video-term-demo-magic.sh</a>
  
    <li><a href="../bin/video-term-recorder_sh.html">video-term-recorder.sh</a>
  
    <li><a href="../bin/workflow_to_tool_yaml_sh.html">workflow_to_tool_yaml.sh</a>
  
    <li><a href="../bin/yaml2json_py.html">yaml2json.py</a>
  
  </ul>
</div>

  </div>
</nav>

<main role="main" aria-label="Page bin/schema-topic.yaml">
<hr>

<p>type: map mapping:</p>

<pre>name:
    type: str
    required: true
    pattern: /^[a-z0-9_-]+$/
    description: |
        The internal identifier for a topic, it should be the same as the folder name.
    examples:
        - epigenetics
        - sequence-analysis
        - admin
type:
    type: str
    required: true
    enum:
        - admin-dev
        - basics
        - data-science
        - use
        - instructors
    description: |
        The type of topic, some have subtly different behaviours.

        `admin-dev`
        :  should be used for admin and developer topics that are not scientifically focused.

        `basics`
        :  Only used for galaxy-interface type topics

        `data-science`
        :  Topics which are not necessarily Galaxy focused but expand into broader communities

        `use`
        :  These topics use galaxy for some analysis

        `instructors`
        :  Specific to topics related to instruction of Galaxy
title:
    type: str
    required: true
    description: |
        Title of the topic, this is displayed for users to see.
    examples:
        - Proteomics
        - Variant Analysis
summary:
    type: str
    required: true
    description: |
        A longer description of the contents of this topic
    examples:
        - Statistical Analyses for omics data and machine learning using Galaxy tools
docker_image:
    type: str
    description: |
        The image ID for an image which contains all of the tools and data for this topic.
    examples:
        - quay.io/galaxy/sequence-analysis-training
subtopics:
    type: seq
    required: false
    description: |
        For large topics, we can define subtopics and create multiple tutorial lists, which separates the tutorials to help users find content that interests them more quickly.
    sequence:
        - type: map
          mapping:
              id:
                  type: str
                  required: true
                  pattern: /^[A-Za-z0-9_-]+$/
                  description: |
                      Subtopic ID, this should match what is used in tutorials.
                  examples:
                      - single-cell
              title:
                  type: str
                  required: true
                  description: |
                      Subtopic title, which is displayed for users to see.
                  examples:
                      - Maintaining a Production Galaxy
                      - Single-cell RNA-seq
              description:
                  type: str
                  required: true
                  description: |
                      A human readable textual description of a subtopic.
                  examples: |
                    - &quot;Start here if you are new to RNA-Seq analysis in Galaxy&quot;
                    - &quot;These tutorials take you from raw sequencing reads to pathway analysis&quot;
                    - &quot;Tutorials about analysis of single-cell RNA-seq data&quot;
                    - &quot;Tutorials using a single published single-cell RNA-seq dataset for a variety of analyses&quot;
              enable:
                  type: bool
                  description: |
                    `false` to hide your topic from the production GTN. This is useful if you need a topic for a workshop, but have not finished making it up to GTN standards.

maintainers:
    type: seq
    required: true
    sequence:
        - type: str
          required: true
          description: A contributor&#39;s ID in the CONTRIBUTORS.yaml file.
          enum:
          - CONTRIBUTORS
edam_ontology:
    type: str
    pattern: /^topic_[0-9]+$/
    description: |
        An edam ontology id that describes the topic.
    examples:
        - topic_3173
requirements:
    type: seq
    description: List of resources that the reader of the material should be familiar with before starting this training. The structure is identical to `follow_up_training`.
    examples:
        - |
            type: internal
            topic_name: statistics
            tutorials:
                - age-prediction-with-ml
    sequence:
        - type: map
          required: true
          mapping:
              type:
                  type: str
                  required: true
                  enum:
                      - internal
                  description: |
                    the type of link
              topic_name:
                  type: str
                  description: |
                    The name of the topic
              tutorials:
                  type: seq
                  sequence:
                      - type: str
                  description: |
                    List of required tutorials inside that topic

references:
    type: seq
    examples: |
      - |
        authors: &quot;Vaudel M, et al.&quot;
        title: &quot;Shedding light on black boxes in protein identification.&quot;
        link: &quot;https://www.ncbi.nlm.nih.gov/pubmed/24678044&quot;
        summary: &quot;An extensive tutorial for peptide and protein identification, available at http://compomics.com/bioinformatics-for-proteomics. The material is completely based on freely available and open-source tools.&quot;
    sequence:
        - type: map
          required: true
          mapping:
              authors:
                  type: str
                  required: true
              title:
                  type: str
                  required: true
              link:
                  type: str
                  required: true
              summary:
                  type: str
gitter:
    type: str
    description: |
        Link to a gitter channel that is more relevant for this topic than the default. E.g. a single cell topic, you could use `Galaxy-Training-Network/galaxy-single-cell` to link to their specific chat room in all of the child tutorials by default.
    examples:
        - Galaxy-Training-Network/galaxy-single-cell
        - galaxy-genome-annotation/Lobby
draft:
    type: bool
    description: |
        `true` to hide your tutorial from the topic page (optional). This is useful if you need a tutorial for a workshop, but have not finished making it up to GTN standards.
tag_based:
    type: bool
    required: false
    description: |
        If this is a synthetic tag based topic (no tutorials of it&#39;s own)
        then you must set this flag to indicate it as such, otherwise you
        will find yourself with an empty topic page.
og_image:
    type: str
    description: |
        A custom image to show on the link preview in external applications (e.g. when the URL is pasted into Twitter)
    pattern: /^\/.*/
    examples:
        - /assets/images/gat.png
custom_css:
    type: str
    description: |
      Any custom CSS you might want to apply to a specific topic. This could be overriding e.g. the masthead colour (beware, will potentially conflict with themes.)

      This is generally not a recommended property to set on topics, and something that is only used as part of outreach efforts to import content into the GTN.
    examples: |
      - :root { --brand-color: red; }

logo:
    type: str
    description: |
        A logo identifier (e.g. GTN) should be used by default, but may be swapped out for special logos from the assets folder.
logo_alt:
    type: str
    description: |
      The alt text for the logo (MANDATORY).</pre>

</main>



<footer id="validator-badges" role="contentinfo">
  <p><a href="https://validator.w3.org/check/referer">Validate</a>
  <p>Generated by <a href="https://ruby.github.io/rdoc/">RDoc</a> 6.2.1.1.
  <p>Based on <a href="http://deveiate.org/projects/Darkfish-RDoc/">Darkfish</a> by <a href="http://deveiate.org">Michael Granger</a>.
</footer>

