<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>De novo transcriptome reconstruction with RNA-Seq</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:title" content="Galaxy Training: De novo transcriptome reconstruction with RNA-Seq" />
        <meta property="og:description" content="Training material for all kinds of transcriptomics analysis." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/transcriptomics" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Transcriptomics
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fde-novo%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fde-novo%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fde-novo%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fde-novo%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fde-novo%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Ftranscriptomics%2Ftutorials%2Fde-novo%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/transcriptomics/tutorials/de-novo/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "De novo transcriptome reconstruction with RNA-Seq",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Transcriptomics",
    "description": "Training material for all kinds of transcriptomics analysis.",
    "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
  },
  "courseCode": "transcriptomics / de-novo / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'De novo transcriptome reconstruction with RNA-Seq' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html",
  "timeRequired": "PT6H",
  "description": "The questions this  addresses are:\n - What genes are differentially expressed between G1E cells and megakaryocytes?\n - How can we generate a transcriptome de novo from RNA sequencing data?\n\n\\nThe objectives are:\n - Analysis of RNA sequencing data using a reference genome\n - Reconstruction of transcripts without reference transcriptome (de novo)\n - Analysis of differentially expressed genes\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    },
    {
      "@type": "Person",
      "name": "Mo Heydarian"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Mallory Freeberg"
    },
    {
      "@type": "Person",
      "name": "Mo Heydarian"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Transcriptomics",
      "description": "Training material for all kinds of transcriptomics analysis.",
      "url": "https://training.galaxyproject.org//training-material/topics/transcriptomics/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_3308",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_3308",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_3308"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>De novo transcriptome reconstruction with RNA-Seq</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/malloryfreeberg/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/malloryfreeberg" alt="Mallory Freeberg">Mallory Freeberg</a>, <a href="/training-material/hall-of-fame/moheydarian/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/moheydarian" alt="Mo Heydarian">Mo Heydarian</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What genes are differentially expressed between G1E cells and megakaryocytes?</p>
</li>
            
            <li><p>How can we generate a transcriptome de novo from RNA sequencing data?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Analysis of RNA sequencing data using a reference genome</p>
</li>
            
            <li><p>Reconstruction of transcripts without reference transcriptome (de novo)</p>
</li>
            
            <li><p>Analysis of differentially expressed genes</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 6 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/transcriptomics/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://zenodo.org/record/254485#.WKODmRIrKRu">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/transcriptomics/tutorials/de-novo/workflows/" title="Workflows" alt="De novo transcriptome reconstruction with RNA-Seq workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/de-novo/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://clipseq.usegalaxy.eu/" title="CLIP-Explorer">
            CLIP-Explorer
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://humancellatlas.usegalaxy.eu/" title="Human Cell Atlas">
            Human Cell Atlas
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://singlecell.usegalaxy.eu/" title="Single Cell Omics workbench">
            Single Cell Omics workbench
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.be/" title="UseGalaxy.be">
            UseGalaxy.be
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.fr/" title="UseGalaxy.fr">
            UseGalaxy.fr
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>The data provided here are part of a <span class="notranslate">Galaxy</span> tutorial that analyzes <span class="notranslate">RNA</span>-seq data from a study published by <em>Wu et al.</em> in 2014 <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4248312/">DOI:10.1101/gr.164830.113</a>. The goal of this study was to investigate ‚Äúthe dynamics of occupancy and the role in gene regulation of the transcription factor Tal1, a critical regulator of hematopoiesis, at multiple stages of hematopoietic differentiation.‚Äù To this end, <span class="notranslate">RNA</span>-seq libraries were constructed from multiple mouse cell types including G1E - a GATA-null immortalized cell line derived from targeted disruption of GATA-1 in mouse embryonic stem cells - and megakaryocytes. This <span class="notranslate">RNA</span>-seq data was used to determine differential gene <span class="notranslate">expression</span> between G1E and megakaryocytes and later correlated with Tal1 occupancy. This dataset (GEO Accession: GSE51338) consists of biological replicate, paired-end, poly(A) selected <span class="notranslate">RNA</span>-seq libraries. Because of the long processing time for the large original files, we have downsampled the original raw data files to include only <span class="notranslate">reads</span> that align to chromosome 19 and a subset of interesting genomic loci identified by Wu <em>et al</em>.</p>

<h1 id="analysis-strategy">Analysis strategy</h1>

<p>The goal of this exercise is to identify what transcripts are present in the G1E and megakaryocyte cellular states and which transcripts are differentially expressed between the two states. We will use a <em>de novo</em> transcript reconstruction strategy to infer transcript structures from the mapped <span class="notranslate">reads</span> in the absence of the actual annotated transcript structures. This will allow us to identify novel transcripts and novel isoforms of known transcripts, as well as identify differentially expressed transcripts.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will deal with:</p>

<ol id="markdown-toc">
  <li><a href="#analysis-strategy" id="markdown-toc-analysis-strategy">Analysis strategy</a>    <ol>
      <li><a href="#data-upload" id="markdown-toc-data-upload">Data upload</a></li>
      <li><a href="#quality-control" id="markdown-toc-quality-control">Quality control</a></li>
    </ol>
  </li>
  <li><a href="#mapping" id="markdown-toc-mapping"><span class="notranslate">Mapping</span></a></li>
  <li><a href="#de-novo-transcript-reconstruction" id="markdown-toc-de-novo-transcript-reconstruction">De novo transcript reconstruction</a></li>
  <li><a href="#transcriptome-assembly" id="markdown-toc-transcriptome-assembly">Transcriptome assembly</a></li>
  <li><a href="#analysis-of-the-differential-gene-expression" id="markdown-toc-analysis-of-the-differential-gene-expression">Analysis of the differential gene <span class="notranslate">expression</span></a>    <ol>
      <li><a href="#count-the-number-of-reads-per-transcript" id="markdown-toc-count-the-number-of-reads-per-transcript">Count the number of <span class="notranslate">reads</span> per transcript</a></li>
      <li><a href="#perform-differential-gene-expression-testing" id="markdown-toc-perform-differential-gene-expression-testing">Perform differential gene <span class="notranslate">expression</span> testing</a></li>
    </ol>
  </li>
  <li><a href="#visualization" id="markdown-toc-visualization">Visualization</a></li>
</ol>

</blockquote>

<h2 id="data-upload">Data upload</h2>

<p>Due to the large size of this dataset, we have downsampled it to only include <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to chromosome 19 and certain loci with relevance to hematopoeisis. This data is available at <a href="https://zenodo.org/record/583140#.WSW3NhPyub8"><code class="language-plaintext highlighter-rouge">Zenodo</code></a>, where you can find the forward and reverse <span class="notranslate">reads</span> corresponding to replicate <span class="notranslate">RNA</span>-seq libraries from G1E and megakaryocyte cells and an annotation file of RefSeq transcripts we will use to generate our transcriptome database.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload</h3>

  <ol>
    <li>
      <p>Create a new history for this <span class="notranslate">RNA</span>-seq exercise</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>

      <blockquote class="tip">

        <h3 id="tip-tip-renaming-a-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Renaming a history</h3>

        <ol>
          <li>Click on <strong>Unnamed history</strong> (or the current name of the history) (<strong>Click to rename history</strong>) at the top of your history panel</li>
          <li>Type the new name</li>
          <li>Press <kbd>Enter</kbd></li>
        </ol>
      </blockquote>
    </li>
    <li>Open the data upload manager (Get Data -&gt; Upload file)</li>
    <li>Copy and paste the links for the <span class="notranslate">reads</span> and annotation file</li>
    <li>Select <strong>Paste/Fetch Data</strong></li>
    <li>Paste the link(s) into the text field</li>
    <li>Change the datatype of the read files to <strong>fastqsanger</strong></li>
    <li>Change the datatype of the annotation file to <strong>gtf</strong> and assign the Genome as <strong>mm10</strong></li>
    <li>Press <strong>Start</strong></li>
    <li>
      <p>Rename the files in your history to retain just the necessary information (<em>e.g.</em> ‚ÄúG1E R1 forward <span class="notranslate">reads</span>‚Äù)</p>
    </li>
    <li>Import the data from <a href="https://zenodo.org/record/583140">Zenodo</a> <a href="https://zenodo.org/record/583140"><img src="https://zenodo.org/badge/DOI/10.123/GTNdenovoRNAseq.svg" alt="DOI" /></a></li>
  </ol>

  <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>   https://zenodo.org/record/583140/files/G1E_rep1_forward_read_%28SRR549355_1%29
   https://zenodo.org/record/583140/files/G1E_rep1_reverse_read_%28SRR549355_2%29
   https://zenodo.org/record/583140/files/G1E_rep2_forward_read_%28SRR549356_1%29
   https://zenodo.org/record/583140/files/G1E_rep2_reverse_read_%28SRR549356_2%29
   https://zenodo.org/record/583140/files/Megakaryocyte_rep1_forward_read_%28SRR549357_1%29
   https://zenodo.org/record/583140/files/Megakaryocyte_rep1_reverse_read_%28SRR549357_2%29
   https://zenodo.org/record/583140/files/Megakaryocyte_rep2_forward_read_%28SRR549358_1%29
   https://zenodo.org/record/583140/files/Megakaryocyte_rep2_reverse_read_%28SRR549358_2%29
   https://zenodo.org/record/583140/files/RefSeq_reference_GTF_%28DSv2%29
</code></pre></div>  </div>

  <p>You will need to fetch the link to the annotation file yourself ;)</p>

  <blockquote class="tip">

    <h3 id="tip-tip-importing-data-via-links"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Importing data via links</h3>

    <ul>
      <li>Copy the link location</li>
      <li>
        <p>Open the <span class="notranslate">Galaxy</span> Upload Manager (<i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-upload</span> on the top-right of the tool panel)</p>
      </li>
      <li>Select <strong>Paste/Fetch Data</strong></li>
      <li>
        <p>Paste the link into the text field</p>
      </li>
      <li>
        <p>Press <strong>Start</strong></p>
      </li>
      <li><strong>Close</strong> the window</li>
    </ul>

    <p>By default, <span class="notranslate">Galaxy</span> uses the URL as the name, so rename the files with a more useful name.</p>

  </blockquote>

</blockquote>

<h2 id="quality-control">Quality control</h2>

<p>For quality control, we use similar tools as described in <a href="/training-material/topics/sequence-analysis/">NGS-QC tutorial</a>: <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> and <a href="http://www.usadellab.org/cms/?page=trimmomatic">Trimmomatic</a>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-quality-control"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Quality control</h3>

  <ol>
    <li>
      <p><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">FastQC</code> on the forward and reverse read files to assess the quality of the <span class="notranslate">reads</span>.</p>

      <blockquote class="question">
        <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the read length?</li>
          <li>Is there anything interesting about the quality of the base calls based on the position in the <span class="notranslate">reads</span>?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The read length is 99 bp</li>
            <li>The quality of base calls declines throughout a sequencing run.</li>
          </ol>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Trim off the low quality bases from the ends of the <span class="notranslate">reads</span> to increase <span class="notranslate">mapping</span> efficiency. Run <code class="language-plaintext highlighter-rouge">Trimmomatic</code> on each pair of forward and reverse <span class="notranslate">reads</span> with the following settings:</p>

      <ul>
        <li><em>‚ÄúSingle-end or paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end (two separate input files)</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput FASTQ file (R1/first of pair)‚Äù</em>: <code class="language-plaintext highlighter-rouge">G1E_rep1 forward read</code></li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúInput FASTQ file (R2/second of pair)‚Äù</em>: <code class="language-plaintext highlighter-rouge">G1E_rep1 reverse read</code></li>
          </ul>
        </li>
        <li><em>‚ÄúPerform initial ILLUMINACLIP step?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
    <li>
      <p><strong>FastQC</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Re-run <code class="language-plaintext highlighter-rouge">FastQC</code> on trimmed <span class="notranslate">reads</span> and inspect the differences.</p>

      <blockquote class="question">
        <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

        <ol>
          <li>What is the read length?</li>
          <li>Is there anything interesting about the quality of the base calls based on the position in the <span class="notranslate">reads</span>?</li>
        </ol>

        <blockquote class="solution">
          <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <ol>
            <li>The read lengths range from 1 to 99 bp after trimming</li>
            <li>The average quality of base calls does not drop off as sharply at the 3‚Äô ends of <span class="notranslate">reads</span>.</li>
          </ol>
        </blockquote>
      </blockquote>
      <p><img src="../../images/BeforeAndAfterTrimming.png" alt="Before and after trimming comparison" /></p>
    </li>
    <li>
      <p><strong>Trimmomatic</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">Trimmomatic</code> on the remaining forward/reverse read pairs with the same parameters.</p>
    </li>
  </ol>

</blockquote>

<p>Now that we have trimmed our <span class="notranslate">reads</span> and are fortunate that there is a reference genome assembly for mouse, we will align our trimmed <span class="notranslate">reads</span> to the genome.</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Instead of running a single tool multiple times on all your data, would you rather run a single tool on multiple datasets at once? Check out the <a href="https://galaxyproject.org/tutorials/collections/">dataset collections</a> feature of <span class="notranslate">Galaxy</span>!</p>
</blockquote>

<h1 id="mapping"><span class="notranslate">Mapping</span></h1>

<p>To make sense of the <span class="notranslate">reads</span>, their positions within mouse genome must be determined. This process is known as aligning or ‚Äò<span class="notranslate">mapping</span>‚Äô the <span class="notranslate">reads</span> to the reference genome.</p>

<blockquote class="comment">
  <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>Do you want to learn more about the principles behind <span class="notranslate">mapping</span>? Follow our <a href="/training-material/topics/sequence-analysis/">training</a></p>
</blockquote>

<p>In the case of a eukaryotic transcriptome, most <span class="notranslate">reads</span> originate from processed m<span class="notranslate">RNA</span>s lacking introns. Therefore, they cannot be simply mapped back to the genome as we normally do for <span class="notranslate">reads</span> derived from <span class="notranslate">DNA</span> sequences. Instead, the <span class="notranslate">reads</span> must be separated into two categories:</p>

<ul>
  <li><span class="notranslate">Reads</span> contained within mature exons - these align perfectly to the reference genome</li>
  <li><span class="notranslate">Reads</span> that span splice junctions in the mature m<span class="notranslate">RNA</span> - these align with gaps to the reference genome</li>
</ul>

<p>Spliced mappers have been developed to efficiently map transcript-derived <span class="notranslate">reads</span> against genomes. <a href="https://ccb.jhu.edu/software/hisat2/index.shtml"><code class="language-plaintext highlighter-rouge">HISAT</code></a> is an accurate and fast tool for <span class="notranslate">mapping</span> spliced <span class="notranslate">reads</span> to a genome. Another popular spliced aligner is <a href="https://ccb.jhu.edu/software/tophat/index.shtml"><code class="language-plaintext highlighter-rouge">TopHat</code></a>, but we will be using <code class="language-plaintext highlighter-rouge">HISAT</code> in this tutorial.</p>

<blockquote class="comment">
  <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
  <p>As it is sometimes quite difficult to determine which settings correspond to those of other programs, the following table might be helpful to identify the library type:</p>

  <table>
    <thead>
      <tr>
        <th>Library type</th>
        <th><strong>Infer Experiment</strong></th>
        <th><strong>TopHat</strong></th>
        <th><strong>HISAT</strong></th>
        <th><strong>htseq-count</strong></th>
        <th><strong>featureCounts</strong></th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>PE</td>
        <td>1++,1‚Äì,2+-,2-+</td>
        <td>FR Second Strand</td>
        <td>FR</td>
        <td>yes</td>
        <td>1</td>
      </tr>
      <tr>
        <td>PE</td>
        <td>1+-,1-+,2++,2‚Äì</td>
        <td>FR First Strand</td>
        <td>RF</td>
        <td>reverse</td>
        <td>2</td>
      </tr>
      <tr>
        <td>SE</td>
        <td>++,‚Äì</td>
        <td>FR Second Strand</td>
        <td>F</td>
        <td>yes</td>
        <td>1</td>
      </tr>
      <tr>
        <td>SE</td>
        <td>+-,-+</td>
        <td>FR First Strand</td>
        <td>R</td>
        <td>reverse</td>
        <td>2</td>
      </tr>
      <tr>
        <td>SE,PE</td>
        <td>undecided</td>
        <td>FR Unstranded</td>
        <td>default</td>
        <td>no</td>
        <td>0</td>
      </tr>
    </tbody>
  </table>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-spliced-mapping"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Spliced <span class="notranslate">mapping</span></h3>

  <ol>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">HISAT2</code> on one forward/reverse read pair and modify the following settings:
      <ul>
        <li><em>‚ÄúSource for the reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Use a built-in genome</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect a reference genome‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mouse (Mus Musculus): mm10</code></li>
          </ul>
        </li>
        <li><em>‚ÄúSingle-end or paired-end <span class="notranslate">reads</span>?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Paired-end</code>
          <ul>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA/Q file #1‚Äù</em>: Trimmomatic on G1E_rep1 forward read (R1 paired)</li>
            <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúFASTA/Q file #2‚Äù</em>: Trimmomatic on G1E_rep1 reverse read (R2 paired)</li>
            <li><em>‚ÄúSpecify strand information‚Äù</em>: <code class="language-plaintext highlighter-rouge">Forward(FR)</code></li>
          </ul>
        </li>
        <li><em>‚ÄúAdvanced options‚Äù</em>
          <ul>
            <li><em>‚ÄúSpliced alignment options‚Äù</em>
              <ul>
                <li><em>‚ÄúPenalty for non-canonical splice sites‚Äù</em>: ‚Äò3‚Äô</li>
                <li><em>‚ÄúPenalty function for long introns with canonical splice sites‚Äù</em>: ‚ÄòConstant [f(x) = B]‚Äô‚Äô</li>
                <li><em>‚ÄúConstant term (B)‚Äù</em>: ‚Äò0.0‚Äô</li>
                <li><em>‚ÄúPenalty function for long introns with non-canonical splice sites‚Äù</em>: ‚ÄòConstant [f(x) = B]‚Äô‚Äô</li>
                <li><em>‚ÄúTranscriptome assembly reporting‚Äù</em>: ‚ÄòReport alignments tailored for transcript assemblers including StringTie‚Äô</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>HISAT2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">HISAT2</code> on the remaining forward/reverse read pairs with the same parameters.</li>
  </ol>

</blockquote>

<h1 id="de-novo-transcript-reconstruction">De novo transcript reconstruction</h1>
<p>Now that we have mapped our <span class="notranslate">reads</span> to the mouse genome with <code class="language-plaintext highlighter-rouge">HISAT</code>, we want to determine transcript structures that are represented by the aligned <span class="notranslate">reads</span>. This is called <em>de novo</em> transcriptome reconstruction. This unbiased approach permits the comprehensive identification of all transcripts present in a sample, including annotated genes, novel isoforms of annotated genes, and novel genes. While common gene/transcript databases are quite large, they are not comprehensive, and the <em>de novo</em> transcriptome reconstruction approach ensures complete transcriptome(s) identification from the experimental samples. The leading tool for transcript reconstruction is <code class="language-plaintext highlighter-rouge">Stringtie</code>. Here, we will use <code class="language-plaintext highlighter-rouge">Stringtie</code> to predict transcript structures based on the <span class="notranslate">reads</span> aligned by <code class="language-plaintext highlighter-rouge">HISAT</code>.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-transcriptome-reconstruction"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Transcriptome reconstruction</h3>

  <ol>
    <li><strong>Stringtie</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">Stringtie</code> on the <code class="language-plaintext highlighter-rouge">HISAT2</code> alignments using the default parameters.
      <ul>
        <li>Use batch mode to run all four samples from one tool form.</li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSpecify strand information‚Äù</em>: <code class="language-plaintext highlighter-rouge">Forward (FR)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h1 id="transcriptome-assembly">Transcriptome assembly</h1>

<p>We just generated four transcriptomes with <code class="language-plaintext highlighter-rouge">Stringtie</code> representing each of the four <span class="notranslate">RNA</span>-seq libraries we are analyzing. Since these were generated in the absence of a reference transcriptome, and we ultimately would like to know what transcript structure corresponds to which annotated transcript (if any), we have to make a <strong>transcriptome database</strong>. We will use the tool <code class="language-plaintext highlighter-rouge">Stringtie - Merge</code> to combine redundant transcript structures across the four samples and the RefSeq reference. Once we have merged our transcript structures, we will use <code class="language-plaintext highlighter-rouge">GFFcompare</code> to annotate the transcripts of our newly created transcriptome so we know the relationship of each transcript to the RefSeq reference.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-transcriptome-assembly"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Transcriptome assembly</h3>

  <ol>
    <li><strong>Stringtie-merge</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">Stringtie-merge</code> on the <code class="language-plaintext highlighter-rouge">Stringtie</code> assembled transcripts along with the RefSeq annotation file we imported earlier.
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúTranscripts‚Äù</em>: <code class="language-plaintext highlighter-rouge">all four </code>Stringtie<code class="language-plaintext highlighter-rouge"> assemblies</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúReference annotation to include in the merging‚Äù</em>: <code class="language-plaintext highlighter-rouge">RefSeq_reference_GTF</code></li>
      </ul>
    </li>
    <li><strong>GFFCompare</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">GFFCompare</code> on the <code class="language-plaintext highlighter-rouge">Stringtie-merge</code> generated transcriptome along with the RefSeq annotation file.
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGTF inputs for comparison‚Äù</em>: <code class="language-plaintext highlighter-rouge">output of Stringtie-merge</code></li>
        <li><em>‚ÄúUse Reference Annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúChoose the source for the reference annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">History</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúReference Annotation‚Äù</em>: <code class="language-plaintext highlighter-rouge">RefSeq_reference_GTF</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúUse Sequence Data‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúChoose the source for the reference list‚Äù</em>: <code class="language-plaintext highlighter-rouge">Locally cached</code>
              <ul>
                <li><em>‚ÄúUsing reference genome‚Äù</em>: ‚ÄòMouse (Mus Musculus): mm10‚Äô</li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="comment">
  <h3 id="comment-note-transcript-categorization-used-by-gffcompare"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Note: Transcript categorization used by <code class="language-plaintext highlighter-rouge">GFFcompare</code></h3>

  <table>
    <thead>
      <tr>
        <th style="text-align: center"><strong>Class code</strong></th>
        <th style="text-align: left"><strong>Transcript category</strong></th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td style="text-align: center">=</td>
        <td style="text-align: left">Annotated in reference</td>
      </tr>
      <tr>
        <td style="text-align: center">j</td>
        <td style="text-align: left">Novel isoform of reference</td>
      </tr>
      <tr>
        <td style="text-align: center">u</td>
        <td style="text-align: left">Intergenic</td>
      </tr>
      <tr>
        <td style="text-align: center">x</td>
        <td style="text-align: left">Anti-sense</td>
      </tr>
      <tr>
        <td style="text-align: center">r</td>
        <td style="text-align: left">Repetitive</td>
      </tr>
      <tr>
        <td style="text-align: center">c</td>
        <td style="text-align: left">Contained in exon of reference</td>
      </tr>
      <tr>
        <td style="text-align: center">s</td>
        <td style="text-align: left">Anti-sense spliced intronic</td>
      </tr>
      <tr>
        <td style="text-align: center">e</td>
        <td style="text-align: left">Single exon transfrag overlapping a reference exon and at least 10 bp of a reference intron, indicating a possible pre-m<span class="notranslate">RNA</span> fragment.</td>
      </tr>
      <tr>
        <td style="text-align: center">i</td>
        <td style="text-align: left">A transfrag falling entirely within a reference intron</td>
      </tr>
      <tr>
        <td style="text-align: center">o</td>
        <td style="text-align: left">Generic exonic overlap with a reference transcript</td>
      </tr>
      <tr>
        <td style="text-align: center">p</td>
        <td style="text-align: left">Possible polymerase run-on fragment (within 2Kbases of a reference transcript)</td>
      </tr>
    </tbody>
  </table>
</blockquote>

<h1 id="analysis-of-the-differential-gene-expression">Analysis of the differential gene <span class="notranslate">expression</span></h1>

<p>We just generated a transriptome database that represents the transcripts present in the G1E and megakaryocytes samples. This database provides the location of our transcripts with non-redundant identifiers, as well as information regarding the origin of the transcript.</p>

<p>We now want to identify which transcripts are differentially expressed between the G1E and megakaryocyte cellular states. To do this we will implement a counting approach using <code class="language-plaintext highlighter-rouge">FeatureCounts</code> to count <span class="notranslate">reads</span> per transcript. Then we will provide this information to <code class="language-plaintext highlighter-rouge">DESeq2</code> to generate normalized transcript counts (abundance estimates) and significance testing for differential <span class="notranslate">expression</span>.</p>

<h2 id="count-the-number-of-reads-per-transcript">Count the number of <span class="notranslate">reads</span> per transcript</h2>

<p>To compare the abundance of transcripts between different cellular states, the first essential step is to quantify the number of <span class="notranslate">reads</span> per transcript. <a href="http://bioinf.wehi.edu.au/featureCounts/"><code class="language-plaintext highlighter-rouge">FeatureCounts</code></a> is one of the most popular tools for counting <span class="notranslate">reads</span> in genomic features. In our case, we‚Äôll be using <code class="language-plaintext highlighter-rouge">FeatureCounts</code> to count <span class="notranslate">reads</span> aligning in exons of our <code class="language-plaintext highlighter-rouge">GFFCompare</code> generated transcriptome database.</p>

<p>The recommended mode is ‚Äúunion‚Äù, which counts overlaps even if a read only shares parts of its sequence with a genomic feature and disregards <span class="notranslate">reads</span> that overlap more than one feature.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-counting-the-number-of-reads-per-transcript"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Counting the number of <span class="notranslate">reads</span> per transcript</h3>

  <ol>
    <li>
      <p><strong>FeatureCounts</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">FeatureCounts</code> on the aligned <span class="notranslate">reads</span> (<code class="language-plaintext highlighter-rouge">HISAT2</code> output) using the <code class="language-plaintext highlighter-rouge">GFFCompare</code> transcriptome database as the annotation file.</p>

      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúAlignment file‚Äù</em>: <code class="language-plaintext highlighter-rouge">four HISAT2 aligned read files</code></li>
        <li><em>‚ÄúSpecify strand information‚Äù</em>: <code class="language-plaintext highlighter-rouge">Stranded (Forward)</code></li>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúGene annotation file‚Äù</em>: <code class="language-plaintext highlighter-rouge">in your history</code>
          <ul>
            <li><em>‚ÄúGene annotation file‚Äù</em>: <code class="language-plaintext highlighter-rouge">the annotated transcripts GTF file output by GFFCompare</code></li>
          </ul>
        </li>
        <li><em>‚ÄúAdvanced options‚Äù</em>
          <ul>
            <li><em>‚ÄúGFF gene identifier‚Äù</em>: <code class="language-plaintext highlighter-rouge">transcript_id</code></li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="perform-differential-gene-expression-testing">Perform differential gene <span class="notranslate">expression</span> testing</h2>

<p>Transcript <span class="notranslate">expression</span> is estimated from read counts, and attempts are made to correct for variability in measurements using replicates. This is absolutely essential to obtaining accurate results. We recommend having at least two biological replicates.</p>

<p><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html"><code class="language-plaintext highlighter-rouge">DESeq2</code></a> is a great tool for differential gene <span class="notranslate">expression</span> analysis. It accepts read counts produced by <code class="language-plaintext highlighter-rouge">FeatureCounts</code> and applies size factor normalization:</p>

<ul>
  <li>Computation for each gene of the geometric mean of read counts across all samples</li>
  <li>Division of every gene count by the geometric mean</li>
  <li>Use of the median of these ratios as sample‚Äôs size factor for normalization</li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on:</h3>

  <ol>
    <li><strong>DESeq2</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">DESeq2</code> with the following parameters:
      <ul>
        <li><em>‚Äú1: Factor‚Äù</em>
          <ul>
            <li><em>‚Äú1: Factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge">G1E</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúCounts file(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">featureCount files corresponding to the two G1E replicates</code></li>
              </ul>
            </li>
            <li><em>‚Äú2: Factor level‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mega</code>
              <ul>
                <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúCounts file(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">featureCount files corresponding to the two Mega replicates</code></li>
              </ul>
            </li>
          </ul>

          <blockquote class="comment">
            <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

            <p>You can select several files by holding down the CTRL (or COMMAND) key and clicking on the desired files</p>
          </blockquote>
        </li>
        <li><em>‚ÄúVisualising the analysis results‚Äù</em>: Yes</li>
        <li><em>‚ÄúOutput normalized counts table‚Äù</em>: Yes</li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>The first output of <code class="language-plaintext highlighter-rouge">DESeq2</code> is a tabular file. The columns are:</p>

<ol>
  <li>Gene identifiers</li>
  <li>Mean normalized counts, averaged over all samples from both conditions</li>
  <li>Logarithm (base 2) of the fold change (the values correspond to up- or downregulation relative to the condition listed as Factor level 1)</li>
  <li>Standard error estimate for the log2 fold change estimate</li>
  <li><a href="http://data.princeton.edu/wws509/notes/c2s3.html">Wald</a> statistic</li>
  <li><em>p</em>-value for the statistical significance of this change</li>
  <li><em>p</em>-value adjusted for multiple testing with the Benjamini-Hochberg procedure which controls false discovery rate (<a href="http://www.biostathandbook.com/multiplecomparisons.html">FDR</a>)</li>
</ol>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-1"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on:</h3>

  <ol>
    <li>
      <p><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">Filter</code> to extract genes with a significant change in gene <span class="notranslate">expression</span> (adjusted <em>p</em>-value less than 0.05) between treated and untreated samples</p>

      <blockquote class="question">
        <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>How many transcripts have a significant change in <span class="notranslate">expression</span> between these conditions?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>To filter, use ‚Äúc7&lt;0.05‚Äù. And we get 249 transcripts with a significant change in gene <span class="notranslate">expression</span> between the G1E and megakaryocyte cellular states.</p>
        </blockquote>
      </blockquote>
    </li>
    <li>
      <p><strong>Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Determine how many transcripts are up or down regulated in the G1E state.</p>

      <blockquote class="comment">
        <h3 id="comment-comments"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comments</h3>
        <p>Rename your datasets for the <span class="notranslate">downstream</span> analyses</p>
      </blockquote>

      <blockquote class="question">
        <h3 id="question-question-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

        <p>Are there more upregulated or downregulated genes in the treated samples?</p>

        <blockquote class="solution">
          <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
          <p>To obtain the up-regulated genes in the G1E state, we filter the previously generated file (with the significant change in transcript <span class="notranslate">expression</span>) with the <span class="notranslate">expression</span> ‚Äúc3&gt;0‚Äù (the log2 fold changes must be greater than 0). We obtain 102  genes (40.9% of the genes with a significant change in gene <span class="notranslate">expression</span>). For the down-regulated genes in the G1E state, we did the inverse and we find 149 transcripts (59% of the genes with a significant change in transcript <span class="notranslate">expression</span>).</p>
        </blockquote>
      </blockquote>
    </li>
  </ol>
</blockquote>

<p>In addition to the list of genes, <code class="language-plaintext highlighter-rouge">DESeq2</code> outputs a graphical summary of the results, useful to evaluate the quality of the experiment:</p>

<ol>
  <li>
    <p>Histogram of <em>p</em>-values for all tests</p>

    <p><img src="../../images/Deseq2_histogram3.png" alt="DESeq2 histogram" /></p>
  </li>
  <li>
    <p><a href="https://en.wikipedia.org/wiki/MA_plot">MA plot</a>: global view of the relationship between the <span class="notranslate">expression</span> change of conditions (log ratios, M), the average <span class="notranslate">expression</span> strength of the genes (average mean, A), and the ability of the algorithm to detect differential gene <span class="notranslate">expression</span>. The genes that passed the significance threshold (adjusted p-value &lt; 0.1) are colored in red.</p>

    <p><img src="../../images/Deseq2_MAplot3.png" alt="MA plot" /></p>
  </li>
  <li>
    <p>Principal Component Analysis (<a href="https://en.wikipedia.org/wiki/Principal_component_analysis">PCA</a>) and the first two axes</p>

    <p><img src="../../images/Deseq2_PCA3.png" alt="Principal component analysis output" /></p>

    <p>Each replicate is plotted as an individual data point. This type of plot is useful for visualizing the overall effect of experimental covariates and batch effects.</p>
  </li>
  <li>
    <p>Heatmap of sample-to-sample distance matrix: overview over similarities and dissimilarities between samples</p>

    <p><img src="../../images/Deseq2_heatmap3.png" alt="Heatmap of sample-to-sample distance matrix" /></p>
  </li>
  <li>
    <p>Dispersion estimates: gene-wise estimates (black), the fitted values (red), and the final maximum a posteriori estimates used in testing (blue)</p>

    <p><img src="../../images/Deseq2_dispersion3.png" alt="Dispersion estimates" /></p>

    <p>This dispersion plot is typical, with the final estimates shrunk from the gene-wise estimates towards the fitted estimates. Some gene-wise estimates are flagged as outliers and not shrunk towards the fitted value. The amount of shrinkage can be more or less than seen here, depending on the sample size, the number of coefficients, the row mean and the variability of the gene-wise estimates.</p>
  </li>
</ol>

<p>For more information about <code class="language-plaintext highlighter-rouge">DESeq2</code> and its outputs, you can have a look at <a href="https://www.bioconductor.org/packages/release/bioc/manuals/DESeq2/man/DESeq2.pdf"><code class="language-plaintext highlighter-rouge">DESeq2</code> documentation</a>.</p>

<h1 id="visualization">Visualization</h1>
<p>Now that we have a list of transcript <span class="notranslate">expression</span> levels and their differential <span class="notranslate">expression</span> levels, it is time to visually inspect our transcript structures and the <span class="notranslate">reads</span> they were predicted from. It is a good practice to visually inspect (and present) loci with transcripts of interest. Fortunately, there is a built-in genome browser in <span class="notranslate">Galaxy</span>, <strong>Trackster</strong>, that make this task simple (and even fun!).</p>

<p>In this last section, we will convert our aligned read data from BAM format to bigWig format to simplify observing where our stranded <span class="notranslate">RNA</span>-seq data aligned to. We‚Äôll then initiate a session on Trackster, load it with our data, and visually inspect our interesting loci.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-converting-aligned-read-files-to-bigwig-format"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Converting aligned read files to bigWig format</h3>

  <ol>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Run <code class="language-plaintext highlighter-rouge">bamCoverage</code> on all four aligned read files (<code class="language-plaintext highlighter-rouge">HISAT2</code> output) with the following parameters:
      <ul>
        <li><em>‚ÄúBin size in bases‚Äù</em>: ‚Äò1‚Äô</li>
        <li><em>‚ÄúEffective genome size‚Äù</em>: ‚Äòmm9 (2150570000)‚Äô‚Äô</li>
        <li><em>‚ÄúAdvanced options‚Äù</em>
          <ul>
            <li><em>‚ÄúOnly include <span class="notranslate">reads</span> originating from fragments from the forward or reverse strand‚Äù</em>: ‚Äòforward‚Äô</li>
          </ul>
        </li>
      </ul>
    </li>
    <li>
      <p><strong>Rename</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Rename the outputs to reflect the origin of the <span class="notranslate">reads</span> and that they represent the <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the PLUS strand.</p>
    </li>
    <li><strong>bamCoverage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat Step 1 except changing the following parameter:
      <ul>
        <li><em>‚ÄúOnly include <span class="notranslate">reads</span> originating from fragments from the forward or reverse strand‚Äù</em>: ‚Äòreverse‚Äô</li>
      </ul>
    </li>
    <li><strong>Rename</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Rename the outputs to reflect the origin of the <span class="notranslate">reads</span> and that they represent the <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to the MINUS strand.</li>
  </ol>
</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-trackster-based-visualization"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Trackster based visualization</h3>

  <ol>
    <li><strong>Viz</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: On the center console at the top of the <span class="notranslate">Galaxy</span> interface, choose ‚Äú Visualization‚Äù -&gt; ‚ÄúNew track browser‚Äù
      <ul>
        <li>Name your visualization someting descriptive under ‚ÄúBrowser name:‚Äù</li>
        <li>Choose ‚ÄúMouse Dec. 2011 (GRCm38/mm10) (mm10)‚Äù as the ‚ÄúReference genome build (dbkey)</li>
        <li>Click ‚ÄúCreate‚Äù to initiate your Trackster session
<img src="../../images/Trackster_opening_window.png" alt="Trackster opening window" /></li>
      </ul>
    </li>
    <li><strong>Viz</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Click ‚ÄúAdd datasets to visualization‚Äù
      <ul>
        <li>Select the ‚ÄúRefSeq GTF mm10‚Äù file</li>
        <li>Select the output files from <code class="language-plaintext highlighter-rouge">Stringtie</code></li>
        <li>Select the output file from <code class="language-plaintext highlighter-rouge">GFFCompare</code></li>
        <li>Select the output files from <code class="language-plaintext highlighter-rouge">bamCoverage</code></li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Using the grey labels on the left side of each track, drag and arrange the track order to your preference.</p>
    </li>
    <li><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Hover over the grey label on the left side of the ‚ÄúRefSeq GTF mm10‚Äù track and click the ‚ÄúEdit settings‚Äù icon.
      <ul>
        <li>Adjust the block color to blue (#0000ff) and antisense strand color to red (#ff0000)</li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat the previous step on the output files from <code class="language-plaintext highlighter-rouge">StringTie</code> and <code class="language-plaintext highlighter-rouge">GFFCompare</code>.</p>
    </li>
    <li><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Hover over the grey label on the left side of the ‚ÄúG1E R1 plus‚Äù track and click the ‚ÄúEdit settings‚Äù icon.
      <ul>
        <li>Adjust the color to blue (#0000ff)</li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat the previous step on the other three bigWig files representing the plus strand.</p>
    </li>
    <li><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Hover over the grey label on the left side of the ‚ÄúG1E R1 minus‚Äù track and click the ‚ÄúEdit settings‚Äù icon.
      <ul>
        <li>Adjust the color to red (#ff0000)</li>
      </ul>
    </li>
    <li>
      <p><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Repeat the previous step on the other three bigWig files representing the minus strand.</p>
    </li>
    <li><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Adjust the track height of the bigWig files to be consistent for each set of plus strand and minus strand tracks.
<img src="../../images/Hoxb13_locus_screenshot.png" alt="Adjusting the track height" /></li>
    <li><i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>: Direct Trackster to the coordinates: chr11:96191452-96206029, what do you see?</li>
  </ol>

  <blockquote class="question">
    <h3 id="question-question-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>
    <p>what do you see?</p>

    <blockquote class="solution">
      <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>There are two clusters of transcripts that are exclusively expressed in the G1E background</li>
        <li>The left-most transcript is the Hoxb13 transcript</li>
        <li>The center cluster of transcripts are not present in the RefSeq annotation and are determined by <code class="language-plaintext highlighter-rouge">GFFCompare</code> to be ‚Äúu‚Äù and ‚Äúx‚Äù</li>
      </ol>
    </blockquote>
  </blockquote>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, we have analyzed <span class="notranslate">RNA</span> sequencing data to extract useful information, such as which genes are expressed in the G1E and megakaryocyte cellular states and which of these genes are differentially expressed between the two cellular states. In addition, we identified unannotated genes that are expressed in a cell-state dependent manner and at a locus with relevance to differentiation and development. To identify these transcripts, we analyzed <span class="notranslate">RNA</span> sequence datasets using a de novo transcriptome reconstruction <span class="notranslate">RNA</span>-seq data analysis approach. This approach can be summed up with the following scheme:</p>

<p><img src="../../images/schematic_for_RNAseq_de_novo_tutorial.png" alt="RNAseq de novo tutorial workflow" /></p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>De novo transcriptome reconstruction is the ideal approach for identifying differentially expressed known and novel transcripts.</p>
</li>
                            
                            <li><p>Differential gene expression testing is improved with the use of replicate experiments and deep sequence coverage.</p>
</li>
                            
                            <li><p>Visualizing data on a genome browser is a great way to display interesting patterns of differential expression.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/transcriptomics#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=De novo transcriptome reconstruction with RNA-Seq (Transcriptomics)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Mallory Freeberg, Mo Heydarian, 2021 <b>De novo transcriptome reconstruction with RNA-Seq (Galaxy Training Materials)</b>. <a href="/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html">/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{transcriptomics-de-novo,
    author = "Mallory Freeberg and Mo Heydarian",
    title = "De novo transcriptome reconstruction with RNA-Seq (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/transcriptomics/tutorials/de-novo/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Mallory Freeberg, Mo Heydarian)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/transcriptomics/tutorials/de-novo/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
