<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Mapping and molecular identification of phenotype-causing mutations</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:title" content="Galaxy Training: Mapping and molecular identification of phenotype-causing..." />
        <meta property="og:description" content="Genetic differences (variants) between healthy and diseas..." />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/variant-analysis" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Variant Analysis
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fmapping-by-sequencing%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fmapping-by-sequencing%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fmapping-by-sequencing%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fmapping-by-sequencing%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fmapping-by-sequencing%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fvariant-analysis%2Ftutorials%2Fmapping-by-sequencing%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Mapping and molecular identification of phenotype-causing mutations",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Variant Analysis",
    "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
    "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
  },
  "courseCode": "variant-analysis / mapping-by-sequencing / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Mapping and molecular identification of phenotype-causing mutations' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html",
  "timeRequired": "PT2H",
  "description": "The questions this  addresses are:\n - What is mapping-by-sequencing?\n - How can it help you identify the causative mutation in phenotypic mutants isolated from genetic screens?\n\n\\nThe objectives are:\n - Use joint variant calling and extraction to facilitate variant comparison across samples\n - Perform variant linkage analyses for phenotypically selected recombinant progeny\n - Filter, annotate and report lists of variants\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/slides.html",
      "name": "Quality Control",
      "description": "Slides for 'Quality Control' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html",
      "name": "Quality Control",
      "description": "Hands-on for 'Quality Control' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/slides.html",
      "name": "Mapping",
      "description": "Slides for 'Mapping' tutorial",
      "learningResourceType": "slides",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    },
    {
      "@type": "Course",
      "url": "https://training.galaxyproject.org//training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html",
      "name": "Mapping",
      "description": "Hands-on for 'Mapping' tutorial",
      "learningResourceType": "hands-on tutorial",
      "interactivityType": "expositive",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Wolfgang Maier"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Variant Analysis",
      "description": "Genetic differences (variants) between healthy and diseased tissue,\nbetween individuals of a population, or between strains of an organism\ncan provide mechanistic insight into disease processes and the natural\nfunction of affected genes.\n\nThe tutorials in this section show how to detect evidence for\ngenetic variants in next-generation sequencing data, a process termed\nvariant calling.\n\nOf equal importance, they also demonstrate how you can interpret, for\na range of different organisms, the resulting sets of variants by\npredicting their molecular effects on genes and proteins, by\nannotating previously observed variants with published knowledge, and\nby trying to link phenotypes of the sequenced samples to their variant\ngenotypes.\n",
      "url": "https://training.galaxyproject.org//training-material/topics/variant-analysis/"
    },
    {
      "@type": "DefinedTerm",
      "@id": "http://edamontology.org/topic_0199",
      "inDefinedTermSet": "http://edamontology.org",
      "termCode": "topic_0199",
      "url": "https://bioportal.bioontology.org/ontologies/EDAM/?p=classes&conceptid=http%3A%2F%2Fedamontology.org%2Ftopic_0199"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Mapping and molecular identification of phenotype-causing mutations</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/wm75/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/wm75" alt="Wolfgang Maier">Wolfgang Maier</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>What is mapping-by-sequencing?</p>
</li>
            
            <li><p>How can it help you identify the causative mutation in phenotypic mutants isolated from genetic screens?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>Use joint variant calling and extraction to facilitate variant comparison across samples</p>
</li>
            
            <li><p>Perform variant linkage analyses for phenotypically selected recombinant progeny</p>
</li>
            
            <li><p>Filter, annotate and report lists of variants</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

    <li>
    
        
        
        <a href="/training-material/topics/sequence-analysis">Sequence analysis</a>
        
            <ul>
                
                    
                        
                    
                        
                            
                            <li> Quality Control:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/quality-control/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                
                    
                        
                            
                            <li> Mapping:
                            
                                <a href="/training-material/topics/sequence-analysis/tutorials/mapping/slides.html"><i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> slides</a>
                                
                            
                            
                                
                                    - <a href="/training-material/topics/sequence-analysis/tutorials/mapping/tutorial.html"><i class="fas fa-laptop" aria-hidden="true"></i><span class="visually-hidden">tutorial</span> hands-on</a>
                                
                            
                            </li>
                        
                    
                        
                    
                
            </ul>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 2 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default supporting_material"><a href="/training-material/topics/variant-analysis/slides/introduction.html" title="Topic Overview slides">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Topic Overview slides
                    </a></li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://doi.org/10.5281/zenodo.1098033">
    <i class="far fa-copy" aria-hidden="true"></i><span class="visually-hidden">zenodo_link</span> Datasets
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/workflows/" title="Workflows" alt="Mapping and molecular identification of phenotype-causing mutations workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                
                    <li class="btn btn-default supporting_material">
<a class="topic-icon" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/mapping-by-sequencing/tours/">
    <i class="fas fa-magic" aria-hidden="true"></i><span class="visually-hidden">interactive_tour</span> Tours
</a>

</li>
                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://test.galaxyproject.org/" title="Galaxy Test">
            Galaxy Test
        </a>
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://galaxy.genouest.org" title="Galaxy@GenOuest">
            Galaxy@GenOuest
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org" title="UseGalaxy.org (Main)">
            UseGalaxy.org (Main)
        </a>
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.org.au" title="UseGalaxy.org.au">
            UseGalaxy.org.au
        </a>
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>In order to map and identify phenotype-causing mutations efficiently from a
single experiment, modern genetic research aims to combine classical genetic
<span class="notranslate">mapping</span> concepts with the power of next-generation-sequencing.</p>

<p>For example, after having obtained a mutant strain of an organism with a particular phenotype from a forward genetic screen, the classical approach towards
identification of the underlying causative mutation would be to:</p>

<ol>
  <li>perform <span class="notranslate">mapping</span> crosses to probe for linkage between the unknown mutation
and selected markers with known location on the genome, in order to determine
the approximate genomic region that the mutation resides in, then</li>
  <li>sequence candidate <span class="notranslate">DNA</span> stretches in this region to identify the precise
nature of the mutation.</li>
</ol>

<p>Modern approaches, in contrast, use just one set of genome-wide sequencing data
obtained from <span class="notranslate">mapping</span>-cross progeny and suitable parent strains to
simultaneously:</p>

<ul>
  <li>discover available marker mutations for linkage analysis,</li>
  <li>map the causative mutation using these markers, and</li>
  <li>identify candidate mutations.</li>
</ul>

<p>Since it uses sequencing data already at the <span class="notranslate">mapping</span> step, this process is
called <strong><span class="notranslate">mapping</span>-by-sequencing</strong>.</p>

<p>Since genome-wide sequencing will often reveal a few hundreds to many thousands
of marker variants that should be considered together in the linkage analysis,
<span class="notranslate">mapping</span>-by-sequencing data is typically analyzed with dedicated computational
tools. For <span class="notranslate">Galaxy</span>, the <strong>MiModD</strong> suite of tools offers efficient and flexible
analysis workflows compatible with a variety of <span class="notranslate">mapping</span>-by-sequencing
approaches.</p>

<blockquote class="details">
  <h3 id="details-further-reading"><i class="fas fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> Further reading</h3>
  <p>The MiModD documentation has its own chapter on supported
<a href="http://mimodd.readthedocs.io/en/latest/nacreousmap.html"><span class="notranslate">mapping</span>-by-sequencing schemes</a>
not covered here.</p>
</blockquote>

<p>In this tutorial we are going to use several of these tools to map and identify
a point mutation in an <em>Arabidopsis thaliana</em> strain from whole-genome
sequencing data. This will demonstrate one particular kind of linkage analysis
supported by <strong>MiModD</strong> called <em>Variant Allele Frequency <span class="notranslate">Mapping</span></em>.</p>

<h3 id="variant-allele-frequency-mapping">Variant Allele Frequency <span class="notranslate">Mapping</span></h3>

<figure id="figure-1"><img src="../../images/mbs_scheme.png" alt="Principal of variant allele frequency mapping" /><figcaption><span class="figcaption-prefix">Figure 1:</span> Principle of variant allele frequency <span class="notranslate">mapping</span></figcaption></figure>

<p>As illustrated in Figure 1, variant allele frequency <span class="notranslate">mapping</span> analyzes marker
variant inheritance patterns in pooled F2 recombinants selected for the
mutation to be mapped.</p>

<p>In the figure we assume a diploid
organism with just two pairs (dark and light grey) of chromosomes.</p>

<p>A mutant line established for any organism will harbor
two types of variants (diamonds):</p>
<ul>
  <li>variants induced during mutagenesis (orange), among them the causative
variant responsible for the mutant phenotype (yellow)</li>
  <li>variants (with respect to the species‚Äô reference genome) that were
present in the premutagenesis strain (green open diamonds)</li>
</ul>

<p>For <span class="notranslate">mapping</span> the causative mutation, additional variants (blue open diamonds)
are introduced through a cross to a <span class="notranslate">mapping</span> strain and F2 progeny is selected
for the mutant phenotype. While each individual in the F2 generation carries
its uniquely recombined variant pattern (3 possible outcomes for the diploid
genome are shown in the figure), this phenotypic selection will work against
<span class="notranslate">mapping</span> strain variants in the vicinity (yellow boxes) of the causative
variant. Hence, when the recombined <span class="notranslate">DNA</span> from phenotypic F2 progeny gets pooled
and sequenced, the causative variant can be found by looking for a region for
which all sequenced <span class="notranslate">reads</span> support variant alleles found in the original mutant
line, rather than <span class="notranslate">mapping</span> strain variant alleles.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

<ol id="markdown-toc">
  <li><a href="#data-preparation" id="markdown-toc-data-preparation">Data Preparation</a></li>
  <li><a href="#joint-variant-calling-and-extraction" id="markdown-toc-joint-variant-calling-and-extraction">Joint Variant Calling and Extraction</a></li>
  <li><a href="#linkage-analysis" id="markdown-toc-linkage-analysis">Linkage Analysis</a></li>
  <li><a href="#identifying-candidate-mutations" id="markdown-toc-identifying-candidate-mutations">Identifying Candidate Mutations</a></li>
</ol>

</blockquote>

<h1 id="data-preparation">Data Preparation</h1>

<p>In this tutorial we are going to (re-)identify the recessive causative mutation
in a late-flowering mutant line of <em>Arabidopsis thaliana</em>.</p>

<blockquote class="comment">
  <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
  <p><strong>MiModD</strong> was developed as a successor of the older and now unmaintained
<em>CloudMap</em> suite of tools, but also uses some key concepts first implemented
in the command line package <em>SHOREmap</em>.
The data used in this tutorial represent a subsample of the
<a href="http://bioinfo.mpipz.mpg.de/shoremap/examples.html">SHOREmap proof-of-principle data</a>
(specifically of the <code class="language-plaintext highlighter-rouge">OCF2</code> and <code class="language-plaintext highlighter-rouge">Ler</code> datasets).</p>
</blockquote>

<p>The mutant line was obtained from a mutagenesis screen on the <strong>Col</strong>umbia
reference background of <em>A. thaliana</em>. For <span class="notranslate">mapping</span>-by-sequencing, the line was
outcrossed to the highly polymorphic <strong>L</strong>andsberg <em><strong>er</strong>ecta</em> background as a
source of marker variants. From this outcross, 119 F2 siblings showing the
original late-flowering phenotype were recovered and their genomic <span class="notranslate">DNA</span> was
pooled and subjected to whole-genome sequencing.
Together with whole-genome sequencing data of a pure L<em>er</em> background strain,
this outcrossed pool data forms the starting point of our analysis.
The L<em>er</em> strain sample, thus, can inform us about the markers that the
outcrossed F2 pool <em>could</em> have inherited and exclusion of L<em>er</em> markers from
a particular genomic region from the F2 pool provides evidence that the
phenotype causing variant lies close to this region. With respect to
<a href="#figure-1">Figure 1</a>, this means that we will only use ‚Äúblue diamond‚Äù marker
variants introduced through the <span class="notranslate">mapping</span> cross for the linkage analysis and
ignore potentially informative (green and orange) variants found in the
original mutant line.</p>

<p>For your convenience, the sequenced <span class="notranslate">reads</span> from both datasets have already been
mapped (using <strong>Bowtie2</strong>) to the <em>A. thaliana</em> Col reference genome (version
TAIR10) so the two datasets take the form of BAM files with the <span class="notranslate">mapping</span> results
(see the tutorial on
<a href="../../../sequence-analysis/tutorials/mapping/tutorial.html">NGS data <span class="notranslate">mapping</span></a>
if you do not know what this means).</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-data-upload-and-preprocessing"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Data upload and preprocessing</h3>

  <blockquote class="tip">
    <h3 id="tip-tip"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip</h3>
    <p>This section assumes that you already know</p>

    <ul>
      <li>how to upload data to <span class="notranslate">Galaxy</span> via links</li>
      <li>edit dataset attributes</li>
      <li>do some basic text manipulation on datasets</li>
    </ul>

    <p>An excellent introduction into these topics can be found in the
<a href="../../../introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html">From peaks to genes</a>
tutorial (specifically, in the sections on
<a href="../../../introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html#data-upload">Data upload</a>
and
<a href="../../../introduction/tutorials/galaxy-intro-peaks2genes/tutorial.html#file-preparation">File preparation</a>).</p>
  </blockquote>

  <ol>
    <li>Import the two BAM datasets representing the:
      <ul>
        <li><a href="https://zenodo.org/record/1098034/files/Ler_mapping_strain.bam">L<em>er</em> <span class="notranslate">mapping</span> strain</a></li>
        <li><a href="https://zenodo.org/record/1098034/files/outcrossed_F2.bam">outcrossed F2 pool</a></li>
      </ul>
    </li>
    <li>Specify the genome that was used at the <span class="notranslate">reads</span> <span class="notranslate">mapping</span> step
      <ol>
        <li>Click on <strong>Edit attributes</strong> (the pencil icon displayed with the dataset)</li>
        <li>As the <strong>Database/Build</strong> select <code class="language-plaintext highlighter-rouge">Arabidopsis thaliana TAIR10</code></li>
        <li><strong>Save</strong> the edited attributes</li>
      </ol>
    </li>
    <li>
      <p>Import the reference genome</p>

      <blockquote class="comment">
        <h3 id="comment-tair10-version"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> TAIR10 Version</h3>
        <p>This step is only necessary if the TAIR10 version of the <em>A. thaliana</em>
genome is not available as a built-in genome on your <span class="notranslate">Galaxy</span> server.</p>
      </blockquote>

      <ol>
        <li>Upload the reference genome using the URL:
<code class="language-plaintext highlighter-rouge">https://www.arabidopsis.org/download_files/Genes/TAIR10_genome_release/TAIR10_chromosome_files/TAIR10_chr_all.fas</code></li>
        <li>Use <strong>Replace parts of text</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to truncate the title
lines in the fasta dataset to just the bare chromosome names. Set:
          <ul>
            <li>
              <p>the <strong>Find pattern</strong> to <code class="language-plaintext highlighter-rouge">^&gt;(\w+).*</code></p>

              <p>This will find entire fasta title lines, or more specifically, any
lines starting with <code class="language-plaintext highlighter-rouge">&gt;</code> followed by at least one letter, digit or
underscore.</p>
            </li>
            <li>
              <p>the <strong>Replace with</strong> pattern to <code class="language-plaintext highlighter-rouge">&gt;chr$1</code></p>

              <p>The <code class="language-plaintext highlighter-rouge">$1</code> in it refers to the parenthesized part of the find pattern,
<em>i.e.</em>, just the substring of letters/digits/underscores directly
following the <code class="language-plaintext highlighter-rouge">&gt;</code>.</p>
            </li>
            <li>
              <p><strong>Find-Pattern is a regular <span class="notranslate">expression</span></strong> to <code class="language-plaintext highlighter-rouge">Yes</code></p>
            </li>
          </ul>
        </li>
        <li>Make sure the datatype of the resulting dataset is set to <code class="language-plaintext highlighter-rouge">fasta</code> and
give the dataset a nice name (<em>e.g.</em>, TAIR10 reference).</li>
        <li>At this point you can delete the originally downloaded dataset.</li>
      </ol>
    </li>
  </ol>
</blockquote>

<h1 id="joint-variant-calling-and-extraction">Joint Variant Calling and Extraction</h1>

<p>A hallmark of all <span class="notranslate">mapping</span>-by-sequencing analyses is that one would like to
follow the fate of variants through genetic crosses. In this tutorial, for
example, we would like to determine, for every marker variant provided by the
L<em>er</em> background, at what frequency it got inherited by the outcrossed F2 pool.
For such analyses to work reliably, it is best to call variants jointly for all
relevant samples because this allows the variant caller to give us per-sample
variant call statistics for all sites at which <strong>any</strong> sample is found to have
a non-wt genotype. In our case, this means that the variant caller will report
any variants it finds for the L<em>er</em> background sample along with the call
statistics for the outcrossed F2 pool at that same site - even if the variant
allele contributes very little to the pool and would, thus, have been
disregarded in a standard single-sample analysis.</p>

<p>Here, we are using the <strong>MiModD Variant Calling</strong> tool for joint variant
calling for our two samples. This tool outputs variant call statistics across
all sites and all samples in the compact binary BCF format. We can then use
the <strong>MiModD Extract Variant Sites</strong> tool to extract sites, for which at least
one sample appears to have a non-wt genotype, and report them in VCF format.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-calling"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant Calling</h3>

  <ol>
    <li>Use <strong>MiModD Variant Calling</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to generate combined
variant call statistics:
      <ul>
        <li>
          <p>Select the <strong>reference genome</strong> to call variants against.</p>

          <p>If your <span class="notranslate">Galaxy</span> server offers version TAIR10 of the <em>A. thaliana</em>
reference as a <strong>built-in genome</strong>, you can go ahead and use it.
Otherwise, download and preprocess the reference as described under
Data Preparation and use it as a <strong>genome from my history</strong>.</p>
        </li>
        <li>As the <strong>Aligned <span class="notranslate">reads</span> input datasets</strong> for the analysis select the two
BAM datasets representing the two samples (use the <code class="language-plaintext highlighter-rouge">Control</code> key on the
keyboard to select multiple datasets from the list.</li>
        <li><strong>Execute</strong> the job</li>
      </ul>
    </li>
    <li>Run <strong>MiModD Extract Variant Sites</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with
      <ul>
        <li><strong>BCF input file</strong> set to the BCF dataset generated in the previous
step.</li>
      </ul>
    </li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>Approximately how many variants were extracted?</li>
      <li>Does this number surprise you?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>You can click on the extracted variants dataset in your history to
get more details about it. You should see that the file has an
estimated 130,000 lines. Since one variant gets reported per line in VCF format (apart from a number
of header lines), this roughly
equals the number of variants in the dataset.</li>
        <li>
          <p>The huge number of variants reflects the fact that the L<em>er</em>
<span class="notranslate">mapping</span> strain is really quite diverged from the Col reference
background that the mutant was generated in and, thus, offers many
marker variants for <span class="notranslate">mapping</span>.</p>

          <p>In total, the L<em>er</em> genome harbors &gt;400,000 single nucleotide
variants compared to the Col reference, but to keep the tutorial
datasets small we only provided <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to 2 out of 5
chromosomes. In addition, we provided really low-coverage data
even for those two chromosomes so our analysis will miss many true
variants and contain many questionable variant calls that may represent
false-positives. Regardless, an analysis based on such a large overall number
of variants is robust enough to provide useful results.</p>
        </li>
      </ol>
    </blockquote>

  </blockquote>
</blockquote>

<h1 id="linkage-analysis">Linkage Analysis</h1>

<p>The extracted variants dataset obtained in the previous step contains
information about thousands of sites that can be used as markers to map the
causative mutation in the F2 pool. We will now analyze the data with the
<strong>MiModD NacreousMap</strong> tool in <strong>Variant Allele Frequency</strong> <span class="notranslate">mapping</span> mode to
identify these markers and look for linkage between them and the causative
mutation.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-linkage-analysis-with-one-set-of-parentally-contributed-markers"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Linkage Analysis with one set of parentally contributed markers</h3>

  <p>In the <strong>MiModD NacreousMap</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> interface, set</p>
  <ul>
    <li><strong>type of <span class="notranslate">mapping</span> analysis to perform</strong> to <code class="language-plaintext highlighter-rouge">Variant Allele Frequency
  <span class="notranslate">Mapping</span></code> and</li>
    <li><strong>data source to use</strong> to <code class="language-plaintext highlighter-rouge">VCF file of variants</code>.</li>
    <li>As the <strong>input file with variants to analyze</strong> select your extracted
variants VCF dataset obtained in the previous step.</li>
    <li>As the <strong><span class="notranslate">mapping</span> sample name</strong> use <code class="language-plaintext highlighter-rouge">outcrossed F2</code>.</li>
    <li>Leave the <strong>name of the related parent sample</strong> input field empty because
in this two-sample there is no such sample.</li>
    <li>As the <strong>name of the unrelated parent sample</strong> specify <code class="language-plaintext highlighter-rouge">Ler</code>.
      <blockquote class="comment">
        <h3 id="comment-sample-roles"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Sample roles</h3>
        <p>In <code class="language-plaintext highlighter-rouge">Variant Allele Frequency <span class="notranslate">Mapping</span></code> mode the tool requires:</p>
        <ul>
          <li>a <strong><span class="notranslate">mapping</span> sample</strong>, which corresponds to the recombinant pool, from
which allele frequencies should be calculated</li>
          <li>either a <strong>related</strong> or an <strong>unrelated parent sample</strong> (or both),
from which the marker variant sites to use in the linkage analysis
will be determined. A related parent sample is related (or identical)
to the original mutant line by descent and provides information about
(some of) the variants present in that line, while an unrelated parent
sample provides information about the <span class="notranslate">mapping</span> strain variants
introduced in the <span class="notranslate">mapping</span> cross. If both a related and an unrelated
parent sample are available, the tool can also use the marker sets
from both sides for the linkage analysis.</li>
        </ul>

      </blockquote>
      <p>With these settings, we base the linkage analysis on crossed-in variants
only, <em>i.e.</em>, the tool will look for variants, for which the L<em>er</em> sample
appears homozygous mutant, and determine the contribution of each such
variant allele to the recombinant pool highlighting regions of low
contribution, <em>i.e.</em>, with suspected linkage to the causative variant, in its
output.</p>
    </li>
  </ul>

  <p>Running this job will generate two datasets, one with linkage tables for each
chromosome in the <em>A. thaliana</em> genome, one with graphical representations
of the data.</p>

  <blockquote class="question">
    <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>Have a look at the linkage plots dataset generated by the tool run. In
which interval on which chromosome would you expect the causative
mutation?</li>
      <li>Are the per-marker scatter plots and the histogram plots in agreement?</li>
      <li>What about those <code class="language-plaintext highlighter-rouge">No markers found!</code> messages (you saw them, right?)
that the tool printed?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>The plots suggest that the causative variant is likely to be found
in the region from 18,000,000 to 19,000,000 on chr2.</li>
        <li>
          <p>Look at the y axis labels! The scatter plots show the rates at
which marker sites appear to recombine with the causative mutation
in the pool of F2 recombinants, while the histograms show linkage
evidence. Hence, a dip of the scatter plot regression line towards zero
corresponds to a peak in the histograms and the two kinds of plots are,
thus, in very good agreement for this analysis.</p>

          <p>In general, the histogram plots are easier to digest and often
provide more precise <span class="notranslate">mapping</span> intervals, and are often all you need for
<span class="notranslate">mapping</span> a causative variant. The per-marker scatter plots, on the other
hand, are closer to the raw data and can sometimes reveal additional
irregularities in the data that may indicate problems with the <span class="notranslate">mapping</span>
cross or unexpected factors influencing the inheritance pattern.</p>
        </li>
        <li>Since we included only <span class="notranslate">reads</span> <span class="notranslate">mapping</span> to chromosome 1 or 2 in
the original BAM files, the tool will not be able to detect any markers
for the other chromosomes.</li>
      </ol>
    </blockquote>

  </blockquote>
</blockquote>

<h1 id="identifying-candidate-mutations">Identifying Candidate Mutations</h1>

<p>With the <span class="notranslate">mapping</span> results obtained above we can now try to get a list of
candidates for the causative mutation. Here is what we know about this mutation
so far:</p>

<ul>
  <li>
    <p>The F2 recombinant pool should appear homozygous mutant for it.</p>

    <p>This is because the F2 recombinants got selected based on the phenotype of
the recessive mutation so we can assume that all individuals in the pool were
homozygous for it.</p>
  </li>
  <li>
    <p>The L<em>er</em> strain should not carry it.</p>
  </li>
  <li>
    <p>It should reside in the genomic region identified in the previous step.</p>
  </li>
</ul>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-filtering"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant Filtering</h3>

  <p>With the above information, we can run <strong>MiModD VCF Filter</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>
to obtain a new VCF dataset with only those variants from the originally
extracted ones that fulfill all three criteria. In the tool interface,</p>

  <ul>
    <li>select as <strong>VCF input file</strong> the extracted variants dataset from your
history</li>
    <li>add a <strong>Sample-specific Filter</strong> and set its
      <ul>
        <li><strong>sample</strong> to <code class="language-plaintext highlighter-rouge">outcrossed F2</code></li>
        <li><strong>genotype pattern(s) for the inclusion of variants</strong> to <code class="language-plaintext highlighter-rouge">1/1</code></li>
      </ul>

      <p>These settings will make the filter retain only variants for which the F2
pool has a <code class="language-plaintext highlighter-rouge">1/1</code>, <em>i.e.</em>, homozygous mutant genotype.</p>
    </li>
    <li>add another <strong>Sample-specific Filter</strong> and set its
      <ul>
        <li><strong>sample</strong> to <code class="language-plaintext highlighter-rouge">Ler</code></li>
        <li><strong>genotype pattern(s) for the inclusion of variants</strong> to <code class="language-plaintext highlighter-rouge">0/0</code></li>
      </ul>

      <p>This filter will retain only variants for which the <span class="notranslate">mapping</span> strain sample
has a homozygous wt genotype.</p>
    </li>
    <li>add a <strong>Region Filter</strong> and set its
      <ul>
        <li><strong>Chromosome</strong> to <code class="language-plaintext highlighter-rouge">chr2</code></li>
        <li><strong>Region Start</strong> to 18000000</li>
        <li><strong>Region End</strong> to 19000000</li>
      </ul>

      <p>to retain only variants falling into the previously determined <span class="notranslate">mapping</span>
interval.</p>

      <blockquote class="comment">
        <h3 id="comment-sample-and-chromosome-names"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Sample and chromosome names</h3>
        <p>This is the second time in this tutorial that we ask you to type sample
or chromosome names exactly as they appear in a dataset, and so you
may have started wondering how you are supposed to keep track of which
names are used where in your own analyses.</p>

        <p>Because this is a valid concern, the MiModD tool suite features the
<strong>MiModD File Information</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool that can report the
sample and sequence names stored in almost any dataset that you may
encounter in a <span class="notranslate">mapping</span>-by-sequencing analysis (specifically, the
tool can handle datasets in vcf, bcf, sam, bam and fasta format).</p>

      </blockquote>
    </li>
  </ul>

  <p>Running the job yields a new VCF dataset with variants that passed all
filters.</p>

  <blockquote class="question">
    <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <ol>
      <li>How many variants passed the filters?</li>
      <li>How many variants would you have to consider as candidates if you did
not have any <span class="notranslate">mapping</span> information?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>If you used the suggested <span class="notranslate">mapping</span> interval of 1Mb on chr2, you
should have retained 5 variants. Filtering is a really powerful way to
reduce the number of variants to consider!</li>
        <li>Re-running the job with the region filter removed shows that across
the two chromosomes, for which the input datasets provided sequencing
<span class="notranslate">reads</span>, there are 50 variants for which the pool appears homozygous
mutant and for which the L<em>er</em> strain is homozygous wt.</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<p>Not all of the retained variants are equally likely candidates for being the
causative mutation. Since the input datasets consist of <span class="notranslate">reads</span> from whole-genome
sequencing, as opposed to exome sequencing, some of the variants may fall in
intergenic regions or introns where they are unlikely to cause a phenotype.
Other variants may fall in coding regions, but may be silent on the protein
level. Hence, we can prioritize our candidate variants after annotating them
with predicted functional effects.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-variant-annotation"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Variant Annotation</h3>

  <p>To annotate our variants we are going to use <strong>SnpEff</strong> with the
<code class="language-plaintext highlighter-rouge">athalianaTair10</code> genome database as the source of functional genomic
annotations, then generate a human-friendly variant report with the
<strong>MiModD Report Variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> tool.</p>

  <ol>
    <li>
      <p>Use <strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> (<span class="notranslate">Galaxy</span> tool version 4.1.0) to
download genome database <code class="language-plaintext highlighter-rouge">athalianaTair10</code>.</p>

      <blockquote class="comment">
        <h3 id="comment-snpeff-versions"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> SnpEff versions</h3>
        <p>SnpEff genome databases can only be used with the specific version of
SnpEff they were built for. Make sure you are using version <code class="language-plaintext highlighter-rouge">4.1.0</code> of
<strong>SnpEff Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in this and of <strong>SnpEff Variant
effect and annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> in the next step.
If your <span class="notranslate">Galaxy</span> server offers newer versions of these tools by default,
you may be able to request version <code class="language-plaintext highlighter-rouge">4.1.0</code> explicitly using the
<strong>Versions</strong> button at the top right of the tool interface.</p>
      </blockquote>
    </li>
    <li>In the <strong>SnpEff Variant effect and annotation</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> interface,
set:
      <ul>
        <li><strong>Sequence changes (SNPs, MNPs, InDels)</strong> to your filtered VCF dataset</li>
        <li><strong>Genome source</strong> to <code class="language-plaintext highlighter-rouge">Reference genome from your history</code></li>
        <li><strong>SnpEff4.1 Genome Data</strong> to your just downloaded genome dataset</li>
        <li>
          <p>under <strong>Annotation options</strong> check <code class="language-plaintext highlighter-rouge">Use 'EFF' field compatible with
older versions (instead of 'ANN')</code></p>

          <p>This last setting is required for compatibility of the resulting dataset
with the current version of <strong>MiModD Report Variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>.</p>
        </li>
      </ul>

      <p>Running this job will produce another VCF dataset with the functional
annotation data added to its INFO column. Inspecting this dataset is not
particularly convenient, but you can turn it into a more user-friendly
report by</p>
    </li>
    <li>using <strong>MiModD Report Variants</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> and setting:
      <ul>
        <li><strong>The VCF input with the variants to be reported</strong> to the annotated
variants dataset generated by <strong>SnpEff</strong></li>
        <li>the <strong>Format to use for the report</strong> to <code class="language-plaintext highlighter-rouge">HTML</code></li>
        <li>
          <p><strong>Species</strong> to <code class="language-plaintext highlighter-rouge">A. thaliana</code>.</p>

          <p>The species information can be used by the tool to enrich the output
with species-specific database and genome browser links.</p>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>
        <p>Variant reports in html format are meant to be used on relatively small
lists of candidate variants like in this example. If the input VCF
contains &gt; 1000 variants, the relatively large size of the reports and
slow loading in the browser may become annoying. In this case, or if you
experience formatting problems with the generated html in your browser,
it may be better to choose <code class="language-plaintext highlighter-rouge">Tab-separated plain text</code> as the report
format although the output will then lack those convenient hyperlinks.</p>
      </blockquote>
    </li>
  </ol>

  <blockquote class="question">
    <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

    <p>The published confirmed mutations in the mutant line are <em>chr2:18,774,111
C‚ÜíT</em> and <em>chr2:18,808,927 C‚ÜíT</em>. The latter was shown through follow-up
experiments to be the causative variant.</p>

    <ol>
      <li>What are the functional effects of these two mutations according to the
report?</li>
      <li>What biological functions are described for the two genes? How do these
relate to the phenotype of our mutant line?</li>
    </ol>

    <blockquote class="solution">
      <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>
      <ol>
        <li>The C‚ÜíT transition at 18,774,111 results in a premature
translational stop for the transcript AT2G45550.1. The C‚ÜíT transition
at 18,808,927 affects a splice donor site and, thus, is likely to
interfere with correct splicing of the transcript AT2G45660.1.</li>
        <li>From the linked gene pages at <em>TAIR</em> you can learn that the
gene AT2G45550 encodes a cytochrome P450, while AT2G45660 encodes a
protein called AGL20, which is known to control flowering. Thus, there
is a direct connection between the late-flowering phenotype of the
mutant line studied here and the known role of the gene affected by
chr2:18,808,927 C‚ÜíT.</li>
      </ol>
    </blockquote>
  </blockquote>
</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p><span class="notranslate">Mapping</span>-by-sequencing can greatly speed up and facilitate the molecular
identification of mutations recovered from mutagenesis screens. The method
makes extensive use of cross-sample comparison of variants and benefits strongly
from joint variant calling to generate reliable and manageable variant
information. A combination of variant linkage analysis and filtering often
results in very small lists of candidate variants that can then be confirmed
through further experimental work. Importantly, the selection of biological
samples for sequencing determines the meaningful variant comparisons that can
be made in the bioinformatics analysis so it is important to understand the
essence of the analysis method <strong>before</strong> the preparation of any biological
samples.</p>



                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Mapping-by-sequencing is a powerful method for the molecular identification of mutations</p>
</li>
                            
                            <li><p>The MiModD suite of tools bundles most of the functionality required to perform mapping-by-sequencing analyses with Galaxy</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/variant-analysis#references">here</a>.</p>
                    

                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Mapping and molecular identification of phenotype-causing mutations (Variant Analysis)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Wolfgang Maier, 2021 <b>Mapping and molecular identification of phenotype-causing mutations (Galaxy Training Materials)</b>. <a href="/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html">/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{variant-analysis-mapping-by-sequencing,
    author = "Wolfgang Maier",
    title = "Mapping and molecular identification of phenotype-causing mutations (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Wolfgang Maier)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/variant-analysis/tutorials/mapping-by-sequencing/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
