<!DOCTYPE html>
<html lang="en" dir="auto">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Trajectory analysis using Monocle3 - full tutorial workflow / Inferring single cell trajectories with Monocle3 / Single Cell</title>
        
            <meta name="google-site-verification" content="9mOXn2JL833-i7-aioCCEuIdG4_tb6qjwUozB5GJnPQ" />

<!-- JavaScript Error Monitoring, and performance tracking. -->
<script
  src="https://browser.sentry-cdn.com/7.52.1/bundle.tracing.min.js"
  integrity="sha384-muuFXKS3752PNA4rPm9Uq6BLvOfV4CXyr9MHDBPvozOJJUWLKkogEFWOIRoVps43"
  crossorigin="anonymous"
></script>
<script type="text/javascript">
if(localStorage.getItem('sentry-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	console.log("Sentry: opt-in");
	Sentry.init({
		dsn: "https://45e0ec6e4373462b92969505df37cf40@sentry.galaxyproject.org/10",
		release: "galaxy-training-network@e4931cc9823eac7a91f1584c8c92e46cf35d9346",
		integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()],
		sampleRate: 0.1,
		tracesSampleRate: 0.1,
		// Capture Replay for no sessions by default
		replaysSessionSampleRate: 0.01,
		// plus for 1% of sessions with an error
		replaysOnErrorSampleRate: 0.01,
		// PII OFF
		sendDefaultPii: false, // Off by default but just in case.
		environment: "production",
	});
}
</script>

<!-- Page view tracking -->
<script defer data-domain="training.galaxyproject.org" src="https://plausible.galaxyproject.eu/js/plausible.js"></script>
<script>
if(localStorage.getItem('plausible-opt-out') !== 'opt-out' && navigator.doNotTrack !== "1") {
	localStorage.removeItem("plausible_ignore")
	console.log("Plausible: opt-in");
	window.plausible = window.plausible || function() { (window.plausible.q = window.plausible.q || []).push(arguments) }
} else {
	// if they're opting-out, or DNT
	// we might get one page by accident but we won't get future ones.
	localStorage.setItem("plausible_ignore", "true")
}
</script>

        
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon">
        <link rel="alternate" type="application/atom+xml" href="/training-material/feed.xml">
        <link rel="canonical" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.html">
        <link rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Regular-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Bold-102a.woff2" as="font" type="font/woff2" crossorigin>
        <link rel="preload" href="/training-material/assets/fonts/AtkinsonHyperlegible/Atkinson-Hyperlegible-Italic-102a.woff2" as="font" type="font/woff2" crossorigin>
        
        <link rel="preload" href="/training-material/assets/css/main.css?v=3" as="style">
        <link rel='preload' href='/training-material/assets/js/bundle.theme.f1f2de89.js' as='script'>
<link rel='preload' href='/training-material/assets/js/bundle.main.40d4e218.js' as='script'>
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=3">
        <link rel="manifest" href="/training-material/manifest.json">
        <meta name="theme-color" content="#2c3143"/>
	

        <meta name="DC.identifier" content="https://github.com/galaxyproject/training-material">
<meta name="DC.type" content="text">
<meta name="DC.title" content="Trajectory analysis using Monocle3 - full tutorial workflow">
<meta name="DC.publisher" content="Galaxy Training Network">
<meta name="DC.date" content="1970-01-01 00:00:00 +0000"><meta name="description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!).  What else do you want to see? You can submit tool or tutorial requests, or upvote other requests, on our  [Single-Cell & Spatial Omics Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:site_name" content="Galaxy Training Network">
	<meta property="og:title" content="Single Cell / Inferring single cell trajectories with Monocle3 / Trajectory analysis using Monocle3 - full tutorial workflow">
        <meta property="og:description" content="Training material and practicals for all kinds of single cell analysis (particularly scRNA-seq!).  What else do you want to see? You can submit tool or tutorial requests, or upvote other requests, on our  [Single-Cell & Spatial Omics Tool Request Spreadsheet](https://docs.google.com/spreadsheets/d/15hqgqA-RMDhXR-ylKhRF-Dab9Ij2arYSKiEVoPl2df4/edit?usp=sharing).  {: .alert.alert-success} ">
        <meta property="og:image" content="https://galaxy-training.s3.amazonaws.com/social/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.png">
	</head>
    <body data-spy="scroll" data-target="#toc" data-brightness="auto" data-contrast="auto">
        <script  src='/training-material/assets/js/bundle.theme.f1f2de89.js'></script>
        <header>
    <nav class="navbar navbar-expand-md navbar-dark" aria-label="Site Navigation">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                
                    Galaxy Training!
                
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        
			

                        
                    </li>

                    <li class="nav-item">
                        
                        <a class="nav-link" href="/training-material/learning-pathways" title="Learning Pathways">
                           <i class="fas fa-graduation-cap" aria-hidden="true"></i><span class="visually-hidden">curriculum</span> Learning Pathways
                        </a>
                        
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
	<a class="dropdown-item" href="/training-material/faqs/index.html" title="Check our FAQs">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="/training-material/topics/single-cell/faqs/" title="Check our FAQs for the Single Cell topic">
           <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Topic FAQs
        </a>
        
        
        
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            <i class="far fa-comments" aria-hidden="true"></i><span class="visually-hidden">feedback</span> Galaxy Help Forum
        </a>
        <a class="dropdown-item" href="https://gitter.im/Galaxy-Training-Network/Lobby" title="Discuss on gitter">
           <i class="fab fa-gitter" aria-hidden="true"></i><span class="visually-hidden">gitter</span> Discuss on Matrix
        </a>
    </div>
</li>


                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Settings">
	<i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden">galaxy-gear</span> Settings
    </a>
    <div class="dropdown-menu dropdown-menu-right">

	<h6 class="dropdown-header">Preferences</h6>

	<a href="/training-material/user/theme.html" class="dropdown-item">
		<i class="fas fa-palette" aria-hidden="true"></i><span class="visually-hidden">gtn-theme</span> Theme
	</a>

	<a href="/training-material/user/privacy.html" class="dropdown-item">
		<i class="fas fa-lock" aria-hidden="true"></i><span class="visually-hidden">pref-dataprivate</span> Data Privacy
	</a>

	<div class="dropdown-divider"></div>

	<h6 class="dropdown-header">For Everyone</h6>

        <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.html">
          <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Propose a change or correction
        </a>

	<h6 class="dropdown-header">Instructor Utilities</h6>

        <a class="dropdown-item" href="/training-material/stats.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN statistics
        </a>

        <a class="dropdown-item" href="https://plausible.galaxyproject.eu/training.galaxyproject.org?period=12mo&page=/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.html">
            <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> Page View Metrics
        </a>

        <!-- link to feedback -->
        
            <a class="dropdown-item" href="/training-material/feedback.html">
                <i class="fas fa-chart-column" aria-hidden="true"></i><span class="visually-hidden">galaxy-barchart</span> GTN feedback
            </a>
        

        <div class="dropdown-item">
            <div>
                <i class="fas fa-history" aria-hidden="true"></i><span class="visually-hidden">galaxy-rulebuilder-history</span> Previous Versions
            </div>

            <div id="archive-selector">
            
                <a class="btn btn-warning" href="https://training.galaxyproject.org/archive/">Older Versions</a>
            </div>

        </div>

    </div>
</li>


                    <!-- Search bar-->
                    <li class="nav-item">
                      <div id="navbarSupportedContent" role="search">
                        <!-- Search form -->
                        <form class="form-inline mr-auto" method="GET" action="/training-material/search2">
                          <i class="fas fa-search nav-link" aria-hidden="true"></i>
                          <div class="md-form mb-2">
                            <input name="query" class="form-control nicer" type="text" placeholder="Search Tutorials" aria-label="Search">
                          </div>
                        </form>
                      </div>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

        
        <div class="container main-content" role="main">
        


<style>
hgroup {margin-bottom: 2rem; border-bottom: 1px solid var(--border-light);}
hgroup p { font-family: monospace; font-size: 0.8rem;}
.meta ul { padding-left: 1em;}
</style>

<section>
	<hgroup>
		<h1 data-toc-skip>Trajectory analysis using Monocle3 - full tutorial workflow</h1>
		<p>single-cell-scrna-case_monocle3-trajectories/trajectory-analysis-using-monocle3---full-tutorial-workflow</p>
	</hgroup>
	
	<div class="row">
		<div class="col-md-3 meta">
			
			<b>Author(s)</b>
			<br/>
			
			Julia Jakiela
			
			
			<br/>

			<i class="fas fa-code-commit" aria-hidden="true"></i><span class="visually-hidden">version</span>
			<b>Version</b>
			<br/>
			1

			<br/>

			<i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span>
			<b>Last updated</b>
			<br/>
			May 16, 2023

			<br/>

			<i class="fas fa-balance-scale" aria-hidden="true"></i><span class="visually-hidden">license</span>
			<b>License</b>
			<br/>

			
			<a href="https://spdx.org/licenses/CC-BY-4.0">CC-BY-4.0</a>
			
			<br/>

			
			<i class="fas fa-tags" aria-hidden="true"></i><span class="visually-hidden">galaxy-tags</span>
			<b>Tags</b>
			<br/>

			
			<div class="label label-default tutorial_tag" style="--color-primary: #b1c1ff; --color-darker: #92abff; --color-dimmed: #a2b6ff;">name:trajectory_analysis</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #00dcc0; --color-darker: #00c4aa; --color-dimmed: #00d0b5;">name:transcriptomics</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #00d9de; --color-darker: #00c1c6; --color-dimmed: #00cdd2;">name:scRNA-seq</div>
			
			<div class="label label-default tutorial_tag" style="--color-primary: #21e200; --color-darker: #1cca00; --color-dimmed: #1ed600;">name:single_cell</div>
			
			</br>
			


			<b>Features</b>
			<ul>
			
			
			
			
			</ul>

			<hr/>


			<b>Tutorial</b>
			<br/>
			<i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> <a href="../tutorial.html">Inferring single cell trajectories with Monocle3</a>
			</br>

			
			
			<i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> <a href="index.html">Other workflows associated with this material</a>
			


			<hr/>
			<b>Workflow Testing</b>
			<br/>

			Tests: ✅ <br/>
			Results: Not yet automated

			<hr/>

			<b>FAIRness</b>

			<i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span>
			<b><abbr title="Persistent URL">PURL</abbr></b>
			<br/>
			<a href="https://gxy.io/GTN:W00201">https://gxy.io/GTN:W00201</a>
			<br/>

			
			<a style="margin-bottom: 0.2rem" href="/training-material/api/workflows/single-cell/scrna-case_monocle3-trajectories/trajectory-analysis-using-monocle3---full-tutorial-workflow/rocrate.zip" class="btn btn-secondary"><img src="/training-material/assets/images/ro-crate.png" style="height: 32px" alt="RO-Crate logo with flask"/> Download Workflow RO-Crate</a>
			

			
			<a style="margin-bottom: 0.2rem" href="https://dev.workflowhub.eu/workflows/1228" class="btn btn-secondary"><img src="/training-material/assets/images/workflowhub-icon.svg" style="height: 32px" alt="Workflowhub cloud with gears logo"/> View on (Dev) WorkflowHub</a>
			

		</div>
		<div class="col-md-9">
			<span tabindex="0" role="button" class="workflow show-when-galaxy-proxy-active" data-workflow="https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_monocle3-trajectories/versions/trajectory-analysis-using-monocle3---full-tutorial-workflow">Launch in Tutorial Mode</span>
			<a href="/training-material/faqs/galaxy/tutorial_mode.html" class="show-when-galaxy-proxy-active"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span></a>
			<br/>

			<span class="dropdown">
				<button class="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown" aria-expanded="false">
					<img src="/training-material/assets/images/galaxylogo.png" style="height: 16px" alt="galaxy logo"/> 
					Run Workflow in Galaxy
				</button>
				<ul class="dropdown-menu">
					
					
					<li class="dropdown-item"><a href="https://usegalaxy.eu/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_monocle3-trajectories/versions/trajectory-analysis-using-monocle3---full-tutorial-workflow">Galaxy Europe</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.fr/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_monocle3-trajectories/versions/trajectory-analysis-using-monocle3---full-tutorial-workflow">Galaxy France</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.org.au/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_monocle3-trajectories/versions/trajectory-analysis-using-monocle3---full-tutorial-workflow">Galaxy Australia</a></li>
					
					<li class="dropdown-item"><a href="https://usegalaxy.org/workflows/trs_import?run_form=true&trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_monocle3-trajectories/versions/trajectory-analysis-using-monocle3---full-tutorial-workflow">Galaxy Main</a></li>
					
					<li class="dropdown-item hide-when-galaxy-proxy-active">
						<a href="https://my.galaxy.training/?path=/workflows/trs_import%3Frun_form=true%26trs_url=https://training.galaxyproject.org/training-material/api/ga4gh/trs/v2/tools/single-cell-scrna-case_monocle3-trajectories/versions/trajectory-analysis-using-monocle3---full-tutorial-workflow">Import to another server (≥23.0+ only!)</a>
					</li>
				</ul>
			</span>
			<a class="btn btn-primary" href="https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.ga">
				<i class="fas fa-download" aria-hidden="true"></i><span class="visually-hidden">galaxy-download</span>
				Download
			</a>
			<pre class="mermaid">flowchart TD
  0["ℹ️ Input Dataset\nAnnData to extract genes & cells annotations"];
  style 0 stroke:#2c3143,stroke-width:4px;
  1["ℹ️ Input Dataset\nAnnData before processing to extract clean expression matrix"];
  style 1 stroke:#2c3143,stroke-width:4px;
  2["Extract cell annotations"];
  0 -->|output| 2;
  0646798d-3a5d-420a-a7cf-cde6cec722d5["Output\nExtracted cell annotations (obs)"];
  2 --> 0646798d-3a5d-420a-a7cf-cde6cec722d5;
  style 0646798d-3a5d-420a-a7cf-cde6cec722d5 stroke:#2c3143,stroke-width:4px;
  3["Extract gene annotations"];
  0 -->|output| 3;
  84162745-68b0-41a5-b06e-a285440ba095["Output\nExtracted gene annotations (var)"];
  3 --> 84162745-68b0-41a5-b06e-a285440ba095;
  style 84162745-68b0-41a5-b06e-a285440ba095 stroke:#2c3143,stroke-width:4px;
  4["Extract clean expression matrix"];
  1 -->|output| 4;
  3f03af0c-ac41-4450-b5ba-639a79ea543b["Output\nUnprocessed expression matrix"];
  4 --> 3f03af0c-ac41-4450-b5ba-639a79ea543b;
  style 3f03af0c-ac41-4450-b5ba-639a79ea543b stroke:#2c3143,stroke-width:4px;
  5["Filter out macrophages"];
  2 -->|obs| 5;
  fb81e9a1-0603-419a-83f5-0c9f94939b16["Output\nCells without macrophages"];
  5 --> fb81e9a1-0603-419a-83f5-0c9f94939b16;
  style fb81e9a1-0603-419a-83f5-0c9f94939b16 stroke:#2c3143,stroke-width:4px;
  6["gene_short_name annotation"];
  3 -->|var| 6;
  94a60e6c-cdf7-48e5-8b55-d221cef22688["Output\nGenes table with gene_short_name colname"];
  6 --> 94a60e6c-cdf7-48e5-8b55-d221cef22688;
  style 94a60e6c-cdf7-48e5-8b55-d221cef22688 stroke:#2c3143,stroke-width:4px;
  7["Cut cells IDs"];
  5 -->|out_file1| 7;
  34146c30-2666-481f-8d49-a849d5d3aa91["Output\nFiltered cells IDs "];
  7 --> 34146c30-2666-481f-8d49-a849d5d3aa91;
  style 34146c30-2666-481f-8d49-a849d5d3aa91 stroke:#2c3143,stroke-width:4px;
  8["Cut genes IDs"];
  6 -->|out_file1| 8;
  8f38fb93-cb30-4fbc-90c5-7f6bd023feea["Output\nGenes IDs"];
  8 --> 8f38fb93-cb30-4fbc-90c5-7f6bd023feea;
  style 8f38fb93-cb30-4fbc-90c5-7f6bd023feea stroke:#2c3143,stroke-width:4px;
  9["Filter matrix by cells"];
  7 -->|out_file1| 9;
  4 -->|X| 9;
  f654c045-5fea-43f9-96d6-492c810593d2["Output\nPre-filtered matrix (by cells)"];
  9 --> f654c045-5fea-43f9-96d6-492c810593d2;
  style f654c045-5fea-43f9-96d6-492c810593d2 stroke:#2c3143,stroke-width:4px;
  10["Remove duplicate column cells IDs"];
  9 -->|out_file1| 10;
  f633c125-a4d0-4fa7-ab87-022349f4ac70["Output\nFiltered matrix (by cells) "];
  10 --> f633c125-a4d0-4fa7-ab87-022349f4ac70;
  style f633c125-a4d0-4fa7-ab87-022349f4ac70 stroke:#2c3143,stroke-width:4px;
  11["Transpose matrix"];
  10 -->|output| 11;
  3202091e-1afc-4f50-8922-1f9e50c490fc["Output\nfiltered matrix (by cells) transposed"];
  11 --> 3202091e-1afc-4f50-8922-1f9e50c490fc;
  style 3202091e-1afc-4f50-8922-1f9e50c490fc stroke:#2c3143,stroke-width:4px;
  12["Filter matrix by genes"];
  8 -->|out_file1| 12;
  11 -->|out_file| 12;
  6c23cbd5-a1b5-428e-8ff9-ebac4893ad88["Output\nPre-filtered matrix (by cells & genes)"];
  12 --> 6c23cbd5-a1b5-428e-8ff9-ebac4893ad88;
  style 6c23cbd5-a1b5-428e-8ff9-ebac4893ad88 stroke:#2c3143,stroke-width:4px;
  13["Remove duplicate column genes IDs"];
  12 -->|out_file1| 13;
  25839326-0b5d-4e74-91ec-24e22c08f7c5["Output\nFiltered matrix (by cells & genes)"];
  13 --> 25839326-0b5d-4e74-91ec-24e22c08f7c5;
  style 25839326-0b5d-4e74-91ec-24e22c08f7c5 stroke:#2c3143,stroke-width:4px;
  14["Monocle3 create"];
  5 -->|out_file1| 14;
  13 -->|output| 14;
  6 -->|out_file1| 14;
  f8c5589c-a70d-416e-a9de-ebc987f61fc3["Output\nMonocle3 create on input dataset(s): cds3"];
  14 --> f8c5589c-a70d-416e-a9de-ebc987f61fc3;
  style f8c5589c-a70d-416e-a9de-ebc987f61fc3 stroke:#2c3143,stroke-width:4px;
  15["Monocle3 preprocess"];
  14 -->|output_rds| 15;
  c134942f-60e7-4cd0-8c32-da4a303b879d["Output\nMonocle3 preprocess on input dataset(s): cds3"];
  15 --> c134942f-60e7-4cd0-8c32-da4a303b879d;
  style c134942f-60e7-4cd0-8c32-da4a303b879d stroke:#2c3143,stroke-width:4px;
  16["Monocle3 reduceDim"];
  15 -->|output_rds| 16;
  2c9fb108-2a1c-41df-a8d8-81a18b66d2a1["Output\nMonocle3 reduceDim on input dataset(s): cds3"];
  16 --> 2c9fb108-2a1c-41df-a8d8-81a18b66d2a1;
  style 2c9fb108-2a1c-41df-a8d8-81a18b66d2a1 stroke:#2c3143,stroke-width:4px;
  17["Monocle3 partition"];
  16 -->|output_rds| 17;
  93371c10-6583-47fc-830f-9a90dc980766["Output\nMonocle3 partition on input dataset(s): cds3"];
  17 --> 93371c10-6583-47fc-830f-9a90dc980766;
  style 93371c10-6583-47fc-830f-9a90dc980766 stroke:#2c3143,stroke-width:4px;
  18["Plot cell type"];
  16 -->|output_rds| 18;
  f096d060-82e8-4847-9e8f-b22c9288f565["Output\nCell type plot"];
  18 --> f096d060-82e8-4847-9e8f-b22c9288f565;
  style f096d060-82e8-4847-9e8f-b22c9288f565 stroke:#2c3143,stroke-width:4px;
  19["Plot genotype"];
  16 -->|output_rds| 19;
  aab28bb1-e2e7-4b7d-9b2b-1cab64bea287["Output\nGenotype plot"];
  19 --> aab28bb1-e2e7-4b7d-9b2b-1cab64bea287;
  style aab28bb1-e2e7-4b7d-9b2b-1cab64bea287 stroke:#2c3143,stroke-width:4px;
  20["Plot batch"];
  16 -->|output_rds| 20;
  9b362eee-9238-47c5-a4c2-85f3995430f4["Output\nBatch plot"];
  20 --> 9b362eee-9238-47c5-a4c2-85f3995430f4;
  style 9b362eee-9238-47c5-a4c2-85f3995430f4 stroke:#2c3143,stroke-width:4px;
  21["Plot sex"];
  16 -->|output_rds| 21;
  c2a1e104-ef2d-4b66-ac2b-bf0251714434["Output\nSex plot"];
  21 --> c2a1e104-ef2d-4b66-ac2b-bf0251714434;
  style c2a1e104-ef2d-4b66-ac2b-bf0251714434 stroke:#2c3143,stroke-width:4px;
  22["Monocle3 top markers"];
  17 -->|output_rds| 22;
  23["Plot genes"];
  17 -->|output_rds| 23;
  e8867379-82c6-4c32-a2d3-0b24670269ae["Output\nGene expression plot"];
  23 --> e8867379-82c6-4c32-a2d3-0b24670269ae;
  style e8867379-82c6-4c32-a2d3-0b24670269ae stroke:#2c3143,stroke-width:4px;
  24["Plot partition"];
  17 -->|output_rds| 24;
  3f7b0a7c-85d6-430b-8385-fa9a76385c0e["Output\nPartition plot"];
  24 --> 3f7b0a7c-85d6-430b-8385-fa9a76385c0e;
  style 3f7b0a7c-85d6-430b-8385-fa9a76385c0e stroke:#2c3143,stroke-width:4px;
  25["Plot cluster"];
  17 -->|output_rds| 25;
  40b65826-f73e-4054-a56d-b241985d6996["Output\nCluster plot"];
  25 --> 40b65826-f73e-4054-a56d-b241985d6996;
  style 40b65826-f73e-4054-a56d-b241985d6996 stroke:#2c3143,stroke-width:4px;
  26["Monocle3 learnGraph"];
  17 -->|output_rds| 26;
  294b1a49-cec6-49d0-bcd4-3a954da67bec["Output\nMonocle3 learnGraph on input dataset(s): cds3"];
  26 --> 294b1a49-cec6-49d0-bcd4-3a954da67bec;
  style 294b1a49-cec6-49d0-bcd4-3a954da67bec stroke:#2c3143,stroke-width:4px;
  27["Plot learned trajectory"];
  26 -->|output_rds| 27;
  c85ddefe-8eaf-4d21-93ed-ca0859286bf0["Output\nCell types & learned trajectory path plot"];
  27 --> c85ddefe-8eaf-4d21-93ed-ca0859286bf0;
  style c85ddefe-8eaf-4d21-93ed-ca0859286bf0 stroke:#2c3143,stroke-width:4px;
  28["Monocle3 orderCells"];
  26 -->|output_rds| 28;
  5aea5752-b767-4fc8-8741-1f4d1c659a41["Output\nMonocle3 orderCells on input dataset(s): cds3"];
  28 --> 5aea5752-b767-4fc8-8741-1f4d1c659a41;
  style 5aea5752-b767-4fc8-8741-1f4d1c659a41 stroke:#2c3143,stroke-width:4px;
  29["Monocle3 diffExp"];
  28 -->|output_rds| 29;
  3c298624-9f06-4e52-b2a8-fb834b87c296["Output\nDifferential expression of genes - table"];
  29 --> 3c298624-9f06-4e52-b2a8-fb834b87c296;
  style 3c298624-9f06-4e52-b2a8-fb834b87c296 stroke:#2c3143,stroke-width:4px;
  30["Plot pseudotime"];
  28 -->|output_rds| 30;
  813d15bb-0771-4aed-9226-4874c1ddba59["Output\nPseudotime plot"];
  30 --> 813d15bb-0771-4aed-9226-4874c1ddba59;
  style 813d15bb-0771-4aed-9226-4874c1ddba59 stroke:#2c3143,stroke-width:4px;</pre>
			<h2>Inputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Input</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>Input dataset</td>
					<td>AnnData to extract genes & cells annotations</td>
				</tr>
				
				<tr>
					<td>Input dataset</td>
					<td>AnnData before processing to extract clean expression matrix</td>
				</tr>
				
				</tbody>
			</table>

			<h2>Outputs</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>From</th>
						<th>Output</th>
						<th>Label</th>
					</tr>
				</thead>
				<tbody>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td>Extract cell annotations</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td>Extract gene annotations</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</td>
					<td>Inspect AnnData</td>
					<td>Extract clean expression matrix</td>
				</tr>
				
				<tr>
					<td>Filter1</td>
					<td>Filter</td>
					<td>Filter out macrophages</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.2</td>
					<td>Column Regex Find And Replace</td>
					<td>gene_short_name annotation</td>
				</tr>
				
				<tr>
					<td>Cut1</td>
					<td>Cut</td>
					<td>Cut cells IDs</td>
				</tr>
				
				<tr>
					<td>Cut1</td>
					<td>Cut</td>
					<td>Cut genes IDs</td>
				</tr>
				
				<tr>
					<td>join1</td>
					<td>Join two Datasets</td>
					<td>Filter matrix (by cells)</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</td>
					<td>Advanced Cut</td>
					<td>Remove duplicate column (cells IDs)</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0+galaxy2</td>
					<td>Transpose</td>
					<td>Transpose matrix</td>
				</tr>
				
				<tr>
					<td>join1</td>
					<td>Join two Datasets</td>
					<td>Filter matrix (by genes)</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</td>
					<td>Advanced Cut</td>
					<td>Remove duplicate column (genes IDs)</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_create/monocle3_create/0.1.4+galaxy2</td>
					<td>Monocle3 create</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_preprocess/monocle3_preprocess/0.1.4+galaxy0</td>
					<td>Monocle3 preprocess</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_reducedim/monocle3_reduceDim/0.1.4+galaxy0</td>
					<td>Monocle3 reduceDim</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_partition/monocle3_partition/0.1.4+galaxy0</td>
					<td>Monocle3 partition</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot cell type</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot genotype</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot batch</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot sex</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_topmarkers/monocle3_topmarkers/0.1.5+galaxy0</td>
					<td>Monocle3 top markers</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot genes</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot partition</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot cluster</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_learngraph/monocle3_learnGraph/0.1.4+galaxy0</td>
					<td>Monocle3 learnGraph</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot learned trajectory</td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_ordercells/monocle3_orderCells/0.1.4+galaxy0</td>
					<td>Monocle3 orderCells</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_diffexp/monocle3_diffExp/0.1.4+galaxy1</td>
					<td>Monocle3 diffExp</td>
					<td></td>
				</tr>
				
				<tr>
					<td>toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</td>
					<td>Monocle3 plotCells</td>
					<td>Plot pseudotime</td>
				</tr>
				
				</tbody>
			</table>


			<h2>Tools</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Tool</th>
						<th>Links</th>
					</tr>
				</thead>
				<tbody>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Cut1">Cut1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=Filter1">Filter1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=join1">join1</a></td>
				<td>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0">toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.1.0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/bgruening/text_processing/d698c222f354">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_create/monocle3_create/0.1.4+galaxy2">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_create/monocle3_create/0.1.4+galaxy2</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_create/ada68a9d0aec">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_diffexp/monocle3_diffExp/0.1.4+galaxy1">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_diffexp/monocle3_diffExp/0.1.4+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_diffexp/d43a608d14ac">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_learngraph/monocle3_learnGraph/0.1.4+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_learngraph/monocle3_learnGraph/0.1.4+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_learngraph/5f877012f7af">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_ordercells/monocle3_orderCells/0.1.4+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_ordercells/monocle3_orderCells/0.1.4+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_ordercells/3e77dc421569">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_partition/monocle3_partition/0.1.4+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_partition/monocle3_partition/0.1.4+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_partition/d98b4705eb4c">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_plotcells/monocle3_plotCells/0.1.5+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_plotcells/91980addce5e">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_preprocess/monocle3_preprocess/0.1.4+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_preprocess/monocle3_preprocess/0.1.4+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_preprocess/c150370b514c">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_reducedim/monocle3_reduceDim/0.1.4+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_reducedim/monocle3_reduceDim/0.1.4+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_reducedim/8900e2a18603">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_topmarkers/monocle3_topmarkers/0.1.5+galaxy0">toolshed.g2.bx.psu.edu/repos/ebi-gxa/monocle3_topmarkers/monocle3_topmarkers/0.1.5+galaxy0</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/ebi-gxa/monocle3_topmarkers/a4ab4af54b1a">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.2">toolshed.g2.bx.psu.edu/repos/galaxyp/regex_find_replace/regexColumn1/1.0.2</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/galaxyp/regex_find_replace/399da6b5ec21">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1">toolshed.g2.bx.psu.edu/repos/iuc/anndata_inspect/anndata_inspect/0.7.5+galaxy1</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/anndata_inspect/ee98d611afc6">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
			<tr>
				<td><a href="https://my.galaxy.training/?path=?tool_id=toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0+galaxy2">toolshed.g2.bx.psu.edu/repos/iuc/datamash_transpose/datamash_transpose/1.1.0+galaxy2</a></td>
				<td>
					
					
					<a href="https://toolshed.g2.bx.psu.edu/view/iuc/datamash_transpose/ac092723240d">
						View in ToolShed
					</a>
					
				</td>
			</tr>
			
				</tbody>
			</table>


			<p>To use these workflows in Galaxy you can either click the links to download the workflows, or you can right-click and copy the link to the workflow which can be used in the Galaxy form to import workflows.</p>


			<h2>Importing into Galaxy</h2>
			Below are the instructions for importing these workflows directly into your Galaxy server of choice to start using them!

			<!--SNIPPET--><blockquote class="hands_on">   <div class="box-title hands_on-title" id="hands-on-importing-a-workflow"><i class="fas fa-pencil-alt" aria-hidden="true"></i> Hands-on: Importing a workflow</div>   <ul>   <li>Click on <em>Workflow</em> on the top menu bar of Galaxy. You will see a list of all your workflows.</li>   <li>Click on <i class="fas fa-upload" aria-hidden="true"></i><span class="visually-hidden">galaxy-upload</span> <strong>Import</strong> at the top-right of the screen</li>   <li>Provide your workflow     <ul>       <li>Option 1: Paste the URL of the workflow into the box labelled <em>“Archived Workflow URL”</em></li>       <li>Option 2: Upload the workflow file in the box labelled <em>“Archived Workflow File”</em></li>     </ul>   </li>   <li>Click the <strong>Import workflow</strong> button</li> </ul>   <p>Below is a short video demonstrating how to import a workflow from GitHub using this procedure:</p>   <figure> <iframe credentialless="" allowfullscreen="" referrerpolicy="no-referrer" sandbox="allow-scripts allow-same-origin allow-popups" allow="accelerometer 'none'; ambient-light-sensor 'none'; autoplay 'none'; battery 'none'; bluetooth 'none'; browsing-topics 'none'; camera 'none'; ch-ua 'none'; display-capture 'none'; domain-agent 'none'; document-domain 'none'; encrypted-media 'none'; execution-while-not-rendered 'none'; execution-while-out-of-viewport 'none'; gamepad 'none'; geolocation 'none'; gyroscope 'none'; hid 'none'; identity-credentials-get 'none'; idle-detection 'none'; keyboard-map 'none'; local-fonts 'none'; magnetometer 'none'; microphone 'none'; midi 'none'; navigation-override 'none'; otp-credentials 'none'; payment 'none'; picture-in-picture 'none'; publickey-credentials-create 'none'; publickey-credentials-get 'none'; screen-wake-lock 'none'; serial 'none'; speaker-selection 'none'; sync-xhr 'none'; usb 'none'; web-share 'none'; window-management 'none'; xr-spatial-tracking 'none'" csp="sandbox allow-scripts allow-same-origin;" width="560" height="315" src="https://www.youtube-nocookie.com/embed/hoP36Te5wko" title="Importing a workflow from URL" frameborder="0" loading="lazy"></iframe> <figcaption>   <span class="figcaption-prefix"><strong>Video</strong>:</span> Importing a workflow from URL</figcaption> </figure> </blockquote> <!--END_SNIPPET-->

			<h2>Version History</h2>
			<table class="table table-striped">
				<thead>
					<tr>
						<th>Version</th>
						<th>Commit</th>
						<th>Time</th>
						<th>Comments</th>
					</tr>
				</thead>
				<tbody>
					
					<tr>
						<td>1</td>
						<td><a href="https://github.com/galaxyproject/training-material/commit/aa56146e23f13b8cde4e0fdf6cea9c5fd1869126">aa56146e2</a></td>
						<td>2023-05-15 19:11:50</td>
						<td>add workflow tests</td>
					</tr>
					
				</tbody>
			</table>


			<h2>For Admins</h2>
			<h3>Installing the workflow tools</h3>

			<pre>
wget https://training.galaxyproject.org/training-material/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.ga -O workflow.ga
workflow-to-tools -w workflow.ga -o tools.yaml
shed-tools install -g GALAXY -a API_KEY -t tools.yaml
workflow-install -g GALAXY -a API_KEY -w workflow.ga --publish-workflows
			</pre>
		</div>
	</div>
</section>


</section>

        </div>
        <footer>
	<hr />
	<div class="container">
		<div class="row">
			<div class="col-sm-3">
				<span style="font-size: 2em">GTN</span>
				<p>
					The GTN provides learners with a free, open repository of online training
					materials, with a focus on hands-on training that aims to be directly applicable for learners.
					We aim to connect researchers and learners with local trainers, and events worldwide.
				</p>
				<p>
					We promote FAIR and Open Science practices worldwide, are committed to the accessibility of this platform and training for everyone.
				</p>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">About Us</span>
				<ul class="no-bullets">
					<li><a href="/training-material/about.html">About</a></li>
					<li><a rel="code-of-conduct" href="https://galaxyproject.org/community/coc/">Code of Conduct</a></li>
					<li><a href="/training-material/accessibility.html">Accessibility</a></li>
					<li><a href="/training-material/faqs/gtn/fair_training.html">100% FAIR Training</a></li>
					<li><a href="/training-material/faqs/gtn/collaborative_development.html">Collaborative Development</a></li>
				</ul>
				<span style="font-size: 1.3em">Page</span>
				<ul class="no-bullets">
					
					<li><i class="fas fa-fingerprint" aria-hidden="true"></i><span class="visually-hidden">purl</span><abbr title="Persistent URL">PURL</abbr>: <a href="https://gxy.io/GTN:W00201">gxy.io/GTN:W00201</a></li>
					

					

					<li>
						<a rel="license" href="https://spdx.org/licenses/CC-BY-4.0">
							Content licensed under Creative Commons Attribution 4.0 International License
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/edit/main/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit on GitHub
						</a>
					</li>
					<li>
						<a href="https://github.com/galaxyproject/training-material/commits/main/topics/single-cell/tutorials/scrna-case_monocle3-trajectories/workflows/Trajectory-analysis-using-Monocle3---full-tutorial-workflow.html">
						<i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> View Changes on GitHub
						</a>
					</li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Support</span>
				<ul class="no-bullets">
					<li><a rel="me" href="/training-material/faqs/galaxy/">Galaxy FAQs</a></li>
					<li><a rel="me" href="https://help.galaxyproject.org">Galaxy Help Forum</a></li>
					<li><a rel="me" href="http://gxy.io/gtn-slack">GTN Slack Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/%23Galaxy-Training-Network_Lobby%3Agitter.im">GTN Matrix Chat</a></li>
					<li><a rel="me" href="https://matrix.to/#/#galaxyproject_Lobby:gitter.im">Galaxy Matrix Chat</a></li>
				</ul>
				<span style="font-size: 1.3em">Framework</span>
				<ul class="no-bullets">
					<li>Revision <a href="https://github.com/galaxyproject/training-material/commit/e4931cc9823eac7a91f1584c8c92e46cf35d9346">e4931cc</a></li>
					<li><a rel="license" href="https://github.com/galaxyproject/training-material/blob/main/LICENSE.md">MIT</a> Licensed</li>
					<li><a href="https://jekyllrb.com/">Jekyll(4.3.2 | production)</a></li>
				</ul>
			</div>
			<div class="col-sm-3">
				<span style="font-size: 1.3em">Follow Us!</span>
				<ul class="no-bullets">
					<li><span style="fill: var(--hyperlink);"><svg   width="1em"   height="1em"   viewBox="0 0 8.4937906 9.1084023"   version="1.1"   id="svg356"   xmlns="http://www.w3.org/2000/svg"   xmlns:svg="http://www.w3.org/2000/svg">  <g     id="layer1"     transform="translate(-70.566217,-144.26757)">    <path       style="fill-opacity:1;stroke:none;stroke-width:0.0179182"       d="m 76.39081,152.24155 c -0.737138,0.20763 -1.554999,0.29101 -2.311453,0.14333 -0.475335,-0.0928 -0.891898,-0.32923 -1.031589,-0.82423 -0.04356,-0.15434 -0.06132,-0.32388 -0.06142,-0.48378 0.353724,0.0457 0.702251,0.1304 1.057176,0.17407 0.701338,0.0864 1.394702,0.0784 2.096434,0.008 0.744056,-0.0745 1.433711,-0.21546 2.060598,-0.64854 0.243974,-0.16855 0.474672,-0.39133 0.603487,-0.66252 0.181421,-0.38195 0.175886,-0.89336 0.204447,-1.30803 0.0923,-1.34029 0.20588,-2.98599 -1.076708,-3.846 -0.499561,-0.33497 -1.208891,-0.39913 -1.791824,-0.45742 -0.987026,-0.0987 -1.971078,-0.0946 -2.956509,0.0338 -0.841146,0.10961 -1.595223,0.31468 -2.1065,1.0443 -0.493296,0.70396 -0.509564,1.52563 -0.509564,2.34729 0,1.37831 -0.05534,2.87744 0.595934,4.13911 0.504703,0.97774 1.498709,1.29589 2.52184,1.41832 0.473239,0.0566 0.96049,0.0849 1.434158,0.0172 0.328853,-0.0471 0.650325,-0.0999 0.966886,-0.20511 0.08957,-0.0298 0.266911,-0.0614 0.322027,-0.14486 0.04089,-0.0618 0.0099,-0.15812 0.0035,-0.22545 -0.01611,-0.16924 -0.02094,-0.34967 -0.02096,-0.51963 m -1.594723,-5.48298 c 0.214822,-0.25951 0.315898,-0.56088 0.60922,-0.75705 0.687899,-0.46006 1.692038,-0.11202 1.992096,0.63161 0.214571,0.5317 0.140174,1.15913 0.140174,1.72017 v 1.03925 c 0,0.0911 0.04009,0.30954 -0.01842,0.38339 -0.04193,0.053 -0.173018,0.0287 -0.232436,0.0287 h -0.698809 v -1.88142 c 0,-0.28413 0.04813,-0.63823 -0.09912,-0.89591 -0.234746,-0.4108 -0.875019,-0.36105 -1.092116,0.0358 -0.123368,0.22555 -0.116792,0.50369 -0.116792,0.75257 v 1.0751 h -0.931726 v -1.05718 c 0,-0.2555 0.0024,-0.53932 -0.121773,-0.77049 -0.21432,-0.39919 -0.857782,-0.44403 -1.090217,-0.0358 -0.147324,0.25871 -0.09604,0.61056 -0.09604,0.89591 v 1.88142 H 72.09042 v -1.98893 c 0,-0.4711 -0.01604,-0.95902 0.233201,-1.3797 0.585269,-0.98786 2.133584,-0.74836 2.472454,0.32253 z"       id="path2318" />  </g></svg></span> <a rel="me" href="https://mstdn.science/@gtn">Mastodon</a></li>
					<li><span style="fill: var(--hyperlink);"><svg  viewBox="0 0 64 57" width="1em" ><path style="fill-opacity:1;stroke:none;stroke-width:0.0179182" d="M13.873 3.805C21.21 9.332 29.103 20.537 32 26.55v15.882c0-.338-.13.044-.41.867-1.512 4.456-7.418 21.847-20.923 7.944-7.111-7.32-3.819-14.64 9.125-16.85-7.405 1.264-15.73-.825-18.014-9.015C1.12 23.022 0 8.51 0 6.55 0-3.268 8.579-.182 13.873 3.805ZM50.127 3.805C42.79 9.332 34.897 20.537 32 26.55v15.882c0-.338.13.044.41.867 1.512 4.456 7.418 21.847 20.923 7.944 7.111-7.32 3.819-14.64-9.125-16.85 7.405 1.264 15.73-.825 18.014-9.015C62.88 23.022 64 8.51 64 6.55c0-9.818-8.578-6.732-13.873-2.745Z"></path></svg></span><a rel="me" href="https://bsky.app/profile/galaxytraining.bsky.social"> Bluesky</a></li>
				</ul>

				<span style="font-size: 1.3em">Publications</span>
				<ul class="no-bullets">
					<li><a href="https://doi.org/10.1371/journal.pcbi.1010752">Hiltemann et al. 2023</a></li>
					<li><a href="https://doi.org/10.1016/j.cels.2018.05.012"> Batut et al. 2018</a></li>
					<li><a href="/training-material/faqs/gtn/gtn_citing.html">Citing Us</a></li>
				</ul>
			</div>
		</div>
	</div>
</footer>


        <script  async defer src='/training-material/assets/js/bundle.main.40d4e218.js'></script>

	
	
	<script src="/training-material/assets/js/mermaid@10.6.1.js" async></script>
	<link rel="preload" href="/training-material/assets/js/mermaid@10.6.1.js" as="script" crossorigin>
	

    </body>
</html>
