<!doctype html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="x-ua-compatible" content="ie=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
        <title>Nucleoli segmentation and feature extraction using CellProfiler</title>
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap.min.css?v=3">
        <link rel="stylesheet" href="/training-material/assets/css/bootstrap-toc.min.css">
        <link rel="stylesheet" href="/training-material/assets/css/main.css?v=2">
        <script src="https://kit.fontawesome.com/67b3f98409.js" crossorigin="anonymous"></script>
        <link rel="stylesheet" href="/training-material/assets/css/academicons.css">
        <link rel="stylesheet" href="/training-material/assets/css/syntax_highlighting.css">
        <link rel="shortcut icon" href="/training-material/favicon.ico" type="image/x-icon" />

        
        
        
        
        
        <meta name="description" content="Image analysis using Galaxy" />
        <meta property="og:title" content="Galaxy Training: Nucleoli segmentation and feature extraction using CellPr..." />
        <meta property="og:description" content="Image analysis using Galaxy" />
        <meta property="og:image" content="/training-material/assets/images/GTNLogo1000.png" />
    </head>
    <body data-spy="scroll" data-target="#toc">
        











<!-- Gitter -->


<script>
  ((window.gitter = {}).chat = {}).options = {
  room: 'Galaxy-Training-Network/Lobby'
  };
</script>
<script src="https://sidecar.gitter.im/dist/sidecar.v1.js" async defer></script>

<header>
    <nav class="navbar navbar-expand-lg navbar-dark">
        <div class="container">
            <a class="navbar-brand" href="/training-material/">
                <img src="/training-material/assets/images/GTN-60px.png" height="30" alt="Galaxy Training Network logo">
                Galaxy Training!
            </a>

            <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#top-navbar" aria-controls="top-navbar" aria-expanded="false" aria-label="Toggle navigation">
                <span class="navbar-toggler-icon"></span>
            </button>
            <div class="collapse navbar-collapse" id="top-navbar">
                <ul class="navbar-nav">
                    <li class="nav-item">
                        <a class="nav-link" href="/training-material/topics/imaging" title="Go back to list of tutorials">
                            <i class="far fa-folder" aria-hidden="true"></i><span class="visually-hidden">topic</span> Imaging
                        </a>
                    </li>

                    <li class="nav-item dropdown">
                        <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Language Selector">
                            <i class="fas fa-language" aria-hidden="true"></i><span class="visually-hidden">language</span> Language
                        </a>
                        <div class="dropdown-menu">
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=fr&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fimaging%2Ftutorials%2Ftutorial-CP%2Ftutorial.html&edit-text=&act=url" title="">
                                Fran√ßais
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ja&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fimaging%2Ftutorials%2Ftutorial-CP%2Ftutorial.html&edit-text=&act=url" title="">
                                Êó•Êú¨Ë™û
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=es&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fimaging%2Ftutorials%2Ftutorial-CP%2Ftutorial.html&edit-text=&act=url" title="">
                                Espa√±ol
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=pt&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fimaging%2Ftutorials%2Ftutorial-CP%2Ftutorial.html&edit-text=&act=url" title="">
                                Portugu√™s
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=ar&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fimaging%2Ftutorials%2Ftutorial-CP%2Ftutorial.html&edit-text=&act=url" title="">
                                ÿßŸÑÿπÿ±ÿ®Ÿäÿ©
                            </a>
                            
                            <a class="dropdown-item" href="https://translate.google.com/translate?hl=jp&sl=en&tl=&u=https%3A%2F%2Ftraining.galaxyproject.org%2Ftopics%2Fimaging%2Ftutorials%2Ftutorial-CP%2Ftutorial.html&edit-text=&act=url" title="">
                                And more!
                            </a>
                        </div>
                    </li>

                    <li class="nav-item dropdown">
    <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-expanded="false" title="Help">
        <i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">help</span> Help
    </a>
    <div class="dropdown-menu dropdown-menu-right">
        <form method="get" action="https://tess.elixir-europe.org/materials">
            <input type="text" id="search" name="q" value="" style="margin-left: 0.5em;/*! border-radius: 0px; */">
            <input type="hidden" value="Galaxy Training" name="content_provider">
            <input type="submit" value="Search on TeSS" style="width: 92%;border-radius: 0px;margin: 0.5em;background: #f47d20;border: 0px;padding: 0.25em;" class="">
        </form>

        <div class="dropdown-divider"></div>
        <a class="dropdown-item" href="/training-material/faq" title="Check our FAQ">
            FAQ
        </a>
        <a class="dropdown-item" href="https://help.galaxyproject.org/" title="Discuss on Galaxy Help">
            Discuss on Galaxy Help
        </a>

        <div class="dropdown-item">
            <div>
                Theme
            </div>

            <div id="theme-selector" data-toggle="buttons">
                <label data-value="default" class="btn btn-secondary">
                    <input type="radio" name="options" id="default" autocomplete="off"> Default
                </label>
                <label data-value="night" class="btn btn-secondary">
                    <input type="radio" name="options" id="night" autocomplete="off"> Night
                </label>
                <label data-value="midnight" class="btn btn-secondary">
                    <input type="radio" name="options" id="midnight" autocomplete="off"> Midnight
                </label>
                <label data-value="rainbow" class="btn btn-secondary">
                    <input type="radio" name="options" id="rainbow" autocomplete="off"> Rainbow
                </label>
                <label data-value="halloween" class="btn btn-secondary">
                    <input type="radio" name="options" id="halloween" autocomplete="off"> üéÉ
                </label>
            </div>

        </div>
    </div>
</li>


                    <li class="nav-item">
                        <a class="nav-link" href="https://github.com/galaxyproject/training-material/edit/master/topics/imaging/tutorials/tutorial-CP/tutorial.md">
                            <i class="fab fa-github" aria-hidden="true"></i><span class="visually-hidden">github</span> Edit
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
</header>

<div class="container main-content">
    <script type="application/ld+json">
        


{
  "@context": "http://schema.org",
  "@type": "Course",
  "accessMode": [
    "textual",
    "visual"
  ],
  "accessModeSufficient": [
    "textual",
    "visual"
  ],
  "accessibilityControl": [
    "fullKeyboardControl",
    "fullMouseControl"
  ],
  "accessibilityFeature": [
    "alternativeText",
    "tableOfContents"
  ],
  "accessibilitySummary": "Short descriptions are present but long descriptions will be needed for non-visual users",
  "audience": {
    "@type": "EducationalAudience",
    "educationalRole": "students"
  },
  "citation": {
    "@type": "CreativeWork",
    "name": "Community-Driven Data Analysis Training for Biology",
    "url": "https://doi.org/10.1016/j.cels.2018.05.012"
  },
  "copyrightHolder": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "discussionUrl": "https://gitter.im/Galaxy-Training-Network/Lobby",
  "headline": "Nucleoli segmentation and feature extraction using CellProfiler",
  "inLanguage": {
    "@type": "Language",
    "name": "English",
    "alternateName": "en"
  },
  "interactivityType": "mixed",
  "isAccessibleForFree": true,
  "license": "https://github.com/galaxyproject/training-material/blob/master/LICENSE.md",
  "producer": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "provider": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "sourceOrganization": {
    "@type": "Organization",
    "email": "galaxytrainingnetwork@gmail.com",
    "name": "Galaxy Training Network",
    "url": "https://galaxyproject.org/teach/gtn/"
  },
  "isPartOf": {
    "@type": "CreativeWork",
    "name": "Imaging",
    "description": "Image analysis using Galaxy",
    "url": "https://training.galaxyproject.org//training-material/topics/imaging/"
  },
  "courseCode": "imaging / tutorial-CP / hands-on",
  "learningResourceType": "hands-on tutorial",
  "name": "Hands-on for 'Nucleoli segmentation and feature extraction using CellProfiler' tutorial",
  "url": "https://training.galaxyproject.org//training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html",
  "timeRequired": "PT4H",
  "description": "The questions this  addresses are:\n - How do I run an image analysis pipeline on public data using CellProfiler?\n - How do I analyse the DNA channel of fluorescence siRNA screens?\n - How do I download public image data into my history?\n - How do I segment and label cell nuclei?\n - How do I segment nucleoli (as the absence of DNA)?\n - How do I combine nuclei and nucleoli into one segmentation mask?\n - How do I extract the background of an image?\n - How do I relate the nucleoli to their parent nucleus?\n - How do I measure the image and object features?\n - How do I measure the image quality?\n\n\\nThe objectives are:\n - How to download images from a public image repository.\n - How to segment cell nuclei using CellProfiler in Galaxy.\n - How to segment cell nucleoli using CellProfiler in Galaxy.\n - How to extract features for images, nuclei and nucleoli.\n\n",
  "coursePrerequisites": [
    {
      "@type": "CreativeWork",
      "url": "https://training.galaxyproject.org//training-material/topics/introduction/",
      "name": "Introduction to Galaxy Analyses",
      "description": "Introduction to Galaxy Analyses",
      "provider": {
        "@type": "Organization",
        "email": "galaxytrainingnetwork@gmail.com",
        "name": "Galaxy Training Network",
        "url": "https://galaxyproject.org/teach/gtn/"
      }
    }
  ],
  "hasPart": [

  ],
  "author": [
    {
      "@type": "Person",
      "name": "Beatriz Serrano-Solano"
    },
    {
      "@type": "Person",
      "name": "Jean-Karim H√©rich√©"
    }
  ],
  "contributor": [
    {
      "@type": "Person",
      "name": "Beatriz Serrano-Solano"
    },
    {
      "@type": "Person",
      "name": "Jean-Karim H√©rich√©"
    }
  ],
  "about": [
    {
      "@type": "CreativeWork",
      "name": "Imaging",
      "description": "Image analysis using Galaxy",
      "url": "https://training.galaxyproject.org//training-material/topics/imaging/"
    }
  ]
}
    </script>

    <section class="tutorial">
        <h1 data-toc-skip>Nucleoli segmentation and feature extraction using CellProfiler</h1>
        

        <div class="contributors-line">By: 

<a href="/training-material/hall-of-fame/beatrizserrano/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/beatrizserrano" alt="Beatriz Serrano-Solano">Beatriz Serrano-Solano</a>, <a href="/training-material/hall-of-fame/jkh1/" class="contributor-badge"><img src="https://avatars.githubusercontent.com/jkh1" alt="Jean-Karim H√©rich√©">Jean-Karim H√©rich√©</a>

</div>

        <blockquote class="overview">
            <h3>Overview</h3>
            <strong><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</strong>
            <ul>
            
            <li><p>How do I run an image analysis pipeline on public data using CellProfiler?</p>
</li>
            
            <li><p>How do I analyse the DNA channel of fluorescence siRNA screens?</p>
</li>
            
            <li><p>How do I download public image data into my history?</p>
</li>
            
            <li><p>How do I segment and label cell nuclei?</p>
</li>
            
            <li><p>How do I segment nucleoli (as the absence of DNA)?</p>
</li>
            
            <li><p>How do I combine nuclei and nucleoli into one segmentation mask?</p>
</li>
            
            <li><p>How do I extract the background of an image?</p>
</li>
            
            <li><p>How do I relate the nucleoli to their parent nucleus?</p>
</li>
            
            <li><p>How do I measure the image and object features?</p>
</li>
            
            <li><p>How do I measure the image quality?</p>
</li>
            
            </ul>

            <strong><i class="fas fa-bullseye" aria-hidden="true"></i><span class="visually-hidden">objectives</span> Objectives</strong>
            <ul>
            
            <li><p>How to download images from a public image repository.</p>
</li>
            
            <li><p>How to segment cell nuclei using CellProfiler in Galaxy.</p>
</li>
            
            <li><p>How to segment cell nucleoli using CellProfiler in Galaxy.</p>
</li>
            
            <li><p>How to extract features for images, nuclei and nucleoli.</p>
</li>
            
            </ul>

            
            <strong><i class="fas fa-check-circle" aria-hidden="true"></i><span class="visually-hidden">requirements</span> Requirements</strong>
            <ul>
            
    <li>
    
        
        
        <a href="/training-material/topics/introduction">Introduction to Galaxy Analyses</a>
        
    
    </li>

            
            </ul>
            

            
            <p><strong><i class="fas fa-hourglass-half" aria-hidden="true"></i><span class="visually-hidden">time</span> Time estimation:</strong> 4 hours</p>
            

            

            
            

            
            <p id="supporting-materials"><strong><i class="fa fa-external-link" aria-hidden="true"></i> Supporting Materials</strong></p>
            <ul>
                <div class="supporting_material">
                
                    <li class="btn btn-default"><a href="/training-material/topics/imaging/tutorials/tutorial-CP/slides.html" title="Slides for this tutorial">
                        <i class="fab fa-slideshare" aria-hidden="true"></i><span class="visually-hidden">slides</span> Slides
                    </a></li>
                

                

                
                    <li class="btn btn-default supporting_material">
    <a class="topic-icon" href="/training-material/topics/imaging/tutorials/tutorial-CP/workflows/" title="Workflows" alt="Nucleoli segmentation and feature extraction using CellProfiler workflows">
        <i class="fas fa-share-alt" aria-hidden="true"></i><span class="visually-hidden">workflow</span> Workflows
    </a>

</li>
                

                

                
                    <li class="btn btn-default supporting_material">

    <a href="#" class="btn btn-default dropdown-toggle topic-icon" data-toggle="dropdown" aria-expanded="false" title="Where to run the tutorial">
        <i class="fas fa-globe" aria-hidden="true"></i><span class="visually-hidden">instances</span> Available on these Galaxies
    </a>
    <ul class="dropdown-menu">
    
        <li>
            <a class="dropdown-item" href="https://github.com/galaxyproject/training-material/tree/master/topics/imaging/docker" title="Docker image for this tutorial">
                <i class="fab fa-docker" aria-hidden="true"></i><span class="visually-hidden">docker_image</span> Docker image
            </a>
        </li>
    
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://africa.usegalaxy.eu/" title="Galaxy Africa">
            Galaxy Africa
        </a>
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.eu" title="UseGalaxy.eu">
            UseGalaxy.eu
        </a>
        
    
        
    
        
        <a class="dropdown-item" href="https://usegalaxy.no/" title="UseGalaxy.no">
            UseGalaxy.no
        </a>
        
    
        
    
        
    
        
    
    </ul>


</li>
                
                </div>
            </ul>
            

            <p><strong> <i class="far fa-calendar" aria-hidden="true"></i><span class="visually-hidden">last_modification</span> Last modification:</strong> Jan 6, 2021 </p>
        </blockquote>

        <div class="container">
            <div class="row">
                <!-- sidebar, which will move to the top on a small screen -->
                <div class="col-sm-2">
                    <nav id="toc" data-toggle="toc" class="sticky-top"></nav>
                </div>
                <div class="col-sm-10">
                    <h1 class="no_toc" id="introduction">Introduction</h1>

<p>The nucleolus is a prominent structure of the nucleus of eukaryotic cells and is involved in ribosome biogenesis and cell cycle regulation. In <span class="notranslate">DNA</span> staining of cells, nucleoli can be identified as the absence of <span class="notranslate">DNA</span> in nuclei (<a href="#nucleoli">Fig. 1</a>).<br />
Phenotypes caused by reduced gene function are widely used to elucidate gene function and image-based <span class="notranslate">RNA</span> interference (<span class="notranslate">RNA</span>i) screens are routinely used to find and characterize genes involved in a particular biological process. While screens typically focus on one biological process of interest, the molecular markers used can also inform on other processes. Re-using published screens image data can then be a cost-effective alte<span class="notranslate">rna</span>tive to performing new experiments.<br />
In particular, regardless of the targeted biological process, many screens include a <span class="notranslate">DNA</span> label and therefore can also reveal the effect of gene knock-downs on nucleoli.</p>

<p><a name="nucleoli"></a></p>
<figure id="figure-1"><img src="../../images/tutorial-CP/img_dna_channel.png" alt="DNA channel" width="50%" /><figcaption><span class="figcaption-prefix">Figure 1:</span> <span class="notranslate">DNA</span> channel from the screen described in <a class="citation" href="#Heriche_2014">‚Äú<b>Integration of biological data by kernels on graph nodes allows prediction of new genes involved in mitotic chromosome condensation</b>‚Äù 2014</a>. The red arrows point at nucleoli.</figcaption></figure>

<p>In this tutorial, we will analyse <span class="notranslate">DNA</span> channel images of publicly available <span class="notranslate">RNA</span>i screens to extract numerical descriptors (i.e. features) of nucleoli.
The images and associated metadata will be retrieved from the <a href="http://idr.openmicroscopy.org/" target="_blank">Image Data Resource (IDR)</a>, a repository that collects image datasets of tissues and cells.</p>

<p>To process and analyse the images, we will use <a href="http://cellprofiler-manual.s3.amazonaws.com/CellProfiler-3.1.9/index.html">CellProfiler</a> (<a class="citation" href="#McQuin_2018">‚Äú<b>CellProfiler 3.0: Next-generation image processing for biology</b>‚Äù 2018</a>), a popular image analysis software. CellProfiler normally comes as a desktop application in which users can compose image analysis workflows from a series of modules. Many of these modules are now also available as tools in <span class="notranslate">Galaxy</span>.</p>

<p>To fully emulate the behaviour of the standalone CellProfiler in <span class="notranslate">Galaxy</span>, each image analysis <span class="notranslate">workflow</span> needs to have three parts:</p>

<p>1) <strong>StartingModules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to initialise the pipeline,</p>

<p>2) tools performing the analysis (<a href="#high_level_view">Fig. 2</a>): identification of the nuclei, nucleoli and background, together with the feature extraction,</p>

<p>3) <strong>CellProfiler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> to actually run the pipeline.</p>

<p><a name="high_level_view"></a></p>
<figure id="figure-2"><img src="../../images/tutorial-CP/wf.jpg" alt="High-level view of the workflow" /><figcaption><span class="figcaption-prefix">Figure 2:</span> High-level view of the <span class="notranslate">workflow</span></figcaption></figure>

<p>In this tutorial, you will learn how to create a <span class="notranslate">workflow</span> that downloads a selection of images from the IDR, and uses CellProfiler to segment the nuclei and nucleoli. You will also learn how to extract and export features at three different levels: image, nucleus, nucleolus.</p>

<blockquote class="agenda">
  <h3 id="agenda">Agenda</h3>

  <p>In this tutorial, we will cover:</p>

<ol id="markdown-toc">
  <li><a href="#get-data" id="markdown-toc-get-data">Get data</a></li>
  <li><a href="#start-cellprofiler-pipeline" id="markdown-toc-start-cellprofiler-pipeline">Start CellProfiler pipeline</a></li>
  <li><a href="#object-segmentation" id="markdown-toc-object-segmentation">Object segmentation</a>    <ol>
      <li><a href="#segment-nuclei" id="markdown-toc-segment-nuclei">Segment nuclei</a></li>
      <li><a href="#segment-nucleoli" id="markdown-toc-segment-nucleoli">Segment nucleoli</a></li>
      <li><a href="#combine-segmentation-masks" id="markdown-toc-combine-segmentation-masks">Combine segmentation masks</a></li>
    </ol>
  </li>
  <li><a href="#background-extraction" id="markdown-toc-background-extraction">Background extraction</a>    <ol>
      <li><a href="#identify-the-foreground" id="markdown-toc-identify-the-foreground">Identify the foreground</a></li>
      <li><a href="#remove-the-foreground-from-the-original-image" id="markdown-toc-remove-the-foreground-from-the-original-image">Remove the foreground from the original image</a></li>
    </ol>
  </li>
  <li><a href="#feature-extraction" id="markdown-toc-feature-extraction">Feature extraction</a>    <ol>
      <li><a href="#measure-the-granularity-texture-intensity-size-and-shape" id="markdown-toc-measure-the-granularity-texture-intensity-size-and-shape">Measure the granularity, texture, intensity, size and shape</a></li>
      <li><a href="#relate-nucleoli-to-their-parent-nucleus" id="markdown-toc-relate-nucleoli-to-their-parent-nucleus">Relate nucleoli to their parent nucleus</a></li>
      <li><a href="#measure-the-image-quality-related-parameters" id="markdown-toc-measure-the-image-quality-related-parameters">Measure the image quality-related parameters</a></li>
      <li><a href="#export-the-features-to-a-csv" id="markdown-toc-export-the-features-to-a-csv">Export the features to a CSV</a></li>
    </ol>
  </li>
  <li><a href="#run-the-cellprofiler-pipeline" id="markdown-toc-run-the-cellprofiler-pipeline">Run the CellProfiler pipeline</a></li>
</ol>

</blockquote>

<h1 id="get-data">Get data</h1>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-download-images-from-the-idr"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Download images from the IDR</h3>

  <ol>
    <li>
      <p>If you are logged in, create a new history for this tutorial.</p>

      <blockquote class="tip">

        <h3 id="tip-tip-creating-a-new-history"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Creating a new history</h3>

        <p>Click the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> icon at the top of the history panel</p>

        <p>If the <i class="fas fa-plus" aria-hidden="true"></i><span class="visually-hidden">new-history</span> is missing:</p>
        <ol>
          <li>Click on the <i class="fas fa-cog" aria-hidden="true"></i><span class="visually-hidden"><span class="notranslate">galaxy</span>-gear</span> icon (<strong>History options</strong>) on the top of the history panel</li>
          <li>Select the option <strong>Create New</strong> from the menu</li>
        </ol>
      </blockquote>
    </li>
    <li>
      <p><strong>IDR Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
      <ul>
        <li><em>‚ÄúHow would you like to specify the IDs of images to download?‚Äù</em>: <code class="language-plaintext highlighter-rouge">As text (comma-separated list of IDs or a valid IDR link)</code></li>
        <li><em>‚ÄúImage IDs to download‚Äù</em>: <code class="language-plaintext highlighter-rouge">http://idr.openmicroscopy.org/webclient/?show=image-295900|image-295905|image-295910|image-295918|image-295928|image-295934</code></li>
        <li><em>‚ÄúName of the channel to download‚Äù</em>: <code class="language-plaintext highlighter-rouge">Cy3</code></li>
        <li><em>‚Äúz-plane of images to download‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li><em>‚ÄúImage frame to download‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
        <li><em>‚ÄúLimit the download to a selected region of the image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, download the entire image plane</code></li>
        <li><em>‚ÄúSkip failed retrievals?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúDownload images in a tarball?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <blockquote class="tip">
        <h3 id="tip-tip-get-the-idr-link-from-a-manual-selection-of-images"><i class="far fa-lightbulb" aria-hidden="true"></i><span class="visually-hidden">tip</span> Tip: Get the IDR link from a manual selection of images</h3>

        <p>To get the valid IDR link, go to the <a href="http://idr.openmicroscopy.org/webclient/?show=screen-102" target="_blank">dataset of interest in the IDR</a> and select in the preview of a plate a few images ((figure <a href="#IDR_interface">Fig. 3</a> - 1)). Once you see them at the bottom of the page (figure <a href="#IDR_interface">Fig. 3</a> - 2), select them again and click the link button in the top-right corner of the right panel (figure <a href="#IDR_interface">Fig. 3</a> - 3).
<a name="IDR_interface"></a></p>
        <figure id="figure-3"><img src="../../images/tutorial-CP/IDR_interface.jpg" alt="IDR interface" /><figcaption><span class="figcaption-prefix">Figure 3:</span> IDR interface</figcaption></figure>
      </blockquote>

      <blockquote class="comment">
        <h3 id="comment-comment"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p><strong>IMPORTANT:</strong> When the number of images to download is high, it is strongly recommended to enable the option <em>‚ÄúDownload images in a tarball?‚Äù</em> in order to improve the performance.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-question"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Question</h3>

  <ol>
    <li>Why are we taking the <code class="language-plaintext highlighter-rouge">Cy3</code> channel in the <a href="http://idr.openmicroscopy.org/webclient/img_detail/295900/" target="_blank">example data</a>?</li>
    <li>How could we download 100,000 images in one go?</li>
  </ol>

  <blockquote class="solution">
    <h3 id="solution-solution"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <ol>
      <li>
        <p>The <code class="language-plaintext highlighter-rouge">Cy3</code> dye was used in the study to stain <span class="notranslate">DNA</span>. Since we want to segment the abscence of <span class="notranslate">DNA</span>, <code class="language-plaintext highlighter-rouge">Cy3</code> is the only channel that we need to download from the IDR.</p>
      </li>
      <li>
        <p>We could upload a text file with the image ids of interest.</p>
      </li>
    </ol>

  </blockquote>

</blockquote>

<h1 id="start-cellprofiler-pipeline">Start CellProfiler pipeline</h1>

<p>The tool <strong>Starting Modules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> comprises the first 4 modules of the standalone CellProfiler. It needs to be the first tool of a <span class="notranslate">workflow</span> because it sets the naming and metadata handling for the rest of tools.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-specify-metadata-to-cellprofiler"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Specify metadata to CellProfiler</h3>

  <ol>
    <li><strong>Starting Modules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li>Images
          <ul>
            <li><em>‚ÄúDo you want to filter only the images? ‚Äú</em>: <code class="language-plaintext highlighter-rouge">Select the images only</code></li>
          </ul>
        </li>
        <li>Metadata
          <ul>
            <li><em>‚ÄúDo you want to extract the metadata?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, specify metadata</code>
              <ul>
                <li><em>‚ÄúMetadata extraction method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Extract from file/folder names</code></li>
              </ul>
            </li>
            <li><em>‚ÄúMetadata source‚Äù</em>: <code class="language-plaintext highlighter-rouge">File name</code></li>
            <li><em>‚ÄúSelect the pattern to extract metadata from the file name‚Äù</em>: <code class="language-plaintext highlighter-rouge">field1__field2__field3__field4__field5__field6</code></li>
            <li><em>‚ÄúExtract metadata from‚Äù</em>: <code class="language-plaintext highlighter-rouge">All images</code></li>
          </ul>
        </li>
        <li>NamesAndTypes
          <ul>
            <li><em>‚ÄúProcess 3D‚Äù</em>: <code class="language-plaintext highlighter-rouge">No, do not process 3D data</code></li>
            <li><em>‚ÄúAssign a name to‚Äù</em>: <code class="language-plaintext highlighter-rouge">Give every image the same name</code></li>
            <li><em>‚ÄúName to assign these images‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
            <li><em>‚ÄúSelect the image type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Grayscale image</code>
              <ul>
                <li><em>‚ÄúSet intensity range from‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image metadata</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>Groups
          <ul>
            <li><em>‚ÄúDo you want to group your images?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, group the images</code></li>
            <li><em>‚Äúparam‚Äù</em>: <code class="language-plaintext highlighter-rouge">field1</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-1"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>The images downloaded from the IDR are named following the pattern: <code class="language-plaintext highlighter-rouge">plateName__imageID__cropX__cropY__cropWidth__cropHeight</code>. These fields indicate to which plate the image belongs, what is the identifier of the image in the IDR, and the 4 cropping parameters selected. In our case, the upper-left corner (X, Y) and the width and height from there. We have a total of 6 metadata values encoded in the name of the file, separated by <code class="language-plaintext highlighter-rouge">__</code>. The pattern to extract our metadata from the file name properly is, therefore: <code class="language-plaintext highlighter-rouge">field1__field2__field3__field4__field5__field6</code>. It is important to keep in mind that, for CellProfiler, our <code class="language-plaintext highlighter-rouge">plateName</code> will be called <code class="language-plaintext highlighter-rouge">field1</code>, <code class="language-plaintext highlighter-rouge">imageID</code> will be <code class="language-plaintext highlighter-rouge">field2</code>, <code class="language-plaintext highlighter-rouge">cropX</code> will be <code class="language-plaintext highlighter-rouge">field3</code>, etc. These matches are relevant for a meaningful interpretation of the features.</p>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="object-segmentation">Object segmentation</h1>

<h2 id="segment-nuclei">Segment nuclei</h2>

<p>Since we are interested in segmenting the nucleoli, you may wonder why we need to segment nuclei first. There are several reasons for that:</p>

<ul>
  <li>
    <p>Get the nuclei features. The intensity, size, shape, number of nucleoli per nucleus, etc. can be informative to study the nucleoli.</p>
  </li>
  <li>
    <p>Avoid the detection of wrong spots. In a first segmentation, we detect the nuclei, while in the second pass we will segment the holes (nucleoli). The holes need to fall inside the nuclei, and hence the importance of having them segmented too.</p>
  </li>
</ul>

<p>In the first step, we will identify the nuclei that are complete, meaning that they are not touching the borders of the image.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-segment-nuclei-that-are-complete-within-the-boundaries-of-the-image"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Segment nuclei that are complete within the boundaries of the image</h3>

  <ol>
    <li><strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>Starting Modules</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúUse advanced settings?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, use advanced settings</code>
          <ul>
            <li><em>‚ÄúEnter the name of the input image (from NamesAndTypes)‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
            <li><em>‚ÄúEnter the name of the primary objects to be identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
            <li><em>‚ÄúTypical minimum diameter of objects, in pixel units (Min)‚Äù</em>: <code class="language-plaintext highlighter-rouge">15</code></li>
            <li><em>‚ÄúTypical maximum diameter of objects, in pixel units (Max)‚Äù</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
            <li><em>‚ÄúDiscard objects outside the diameter range?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúDiscard objects touching the border of the image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúThreshold strategy‚Äù</em>: <code class="language-plaintext highlighter-rouge">Global</code>
              <ul>
                <li><em>‚ÄúThresholding method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Otsu</code>
                  <ul>
                    <li><em>‚ÄúTwo-class or three-class thresholding?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Two classes</code></li>
                    <li><em>‚ÄúThreshold correction factor‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.9</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúMethod to distinguish clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Shape</code>
              <ul>
                <li><em>‚ÄúMethod to draw dividing lines between clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Shape</code>
                  <ul>
                    <li><em>‚ÄúAutomatically calculate size of smoothing filter for declumping?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>‚ÄúAutomatically calculate minimum allowed distance between local maxima?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúHandling of objects if excessive number of objects identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Continue</code></li>
          </ul>
        </li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-2"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <ul>
          <li>The names entered to the input image and objects have to be consistent (case sensitive) with the ones in <em>NamesAndTypes</em> and the tools to follow.</li>
          <li>The min and max diameter of the objects (<code class="language-plaintext highlighter-rouge">Typical minimum diameter of objects, in pixel units (Min)</code> and <code class="language-plaintext highlighter-rouge">Typical minimum diameter of objects, in pixel units (Max)</code>) will have to be adjusted to the resolution of the images.</li>
          <li>The <code class="language-plaintext highlighter-rouge">Threshold correction factor</code> is used to adjust the threshold in Otsu‚Äôs method. The value depends on the images and it might be useful to test several values and pick the one that works best for a particular kind of image.</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>We are using here Otsu‚Äôs thresholding method for segmentation. What other segmentation options are available? What is the difference between them?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-1"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>In the <em>global</em> methods we have <code class="language-plaintext highlighter-rouge">Manual</code>, <code class="language-plaintext highlighter-rouge">Measurement</code>, <code class="language-plaintext highlighter-rouge">Minimum cross entropy</code>, <code class="language-plaintext highlighter-rouge">Otsu</code> and <code class="language-plaintext highlighter-rouge">Robust background</code>. For the <em>adaptive</em> ones we only have <code class="language-plaintext highlighter-rouge">Otsu</code>. Check the parameters‚Äô help to get more information on each one.</p>

  </blockquote>

</blockquote>

<p>From the previous tool, we got a group of objects (nuclei). Now, we want to export the segmentation masks as a single image to check how well the segmentation algorithm is performing. We also want to label the nuclei with their identifiers for future visual inspection of the results. The output of this step will look like (you will only get this image after running the whole <span class="notranslate">workflow</span>):</p>

<figure id="figure-4"><img src="../../images/tutorial-CP/img_nuclei_labels.png" alt="Identified nuclei with labels" /><figcaption><span class="figcaption-prefix">Figure 4:</span> Identified nuclei with labels.</figcaption></figure>

<blockquote class="question">
  <h3 id="question-questions-1"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Why are some nuclei not labeled in the image above?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-2"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>We have indicated to the tool <strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> that the nuclei that are either outside the diameter range or touching the border should be discarded.</p>

  </blockquote>

</blockquote>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-mask-the-nuclei-detected"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Mask the nuclei detected</h3>

  <ol>
    <li><strong>ConvertObjectsToImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEnter the name of the input objects you want to convert to an image‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
        <li><em>‚ÄúEnter the name of the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge">MaskNuclei</code></li>
        <li><em>‚ÄúSelect the color format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Binary (black &amp; white)</code></li>
      </ul>
    </li>
    <li><strong>DisplayDataOnImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>ConvertObjectsToImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúDisplay object or image measurements?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Object</code>
          <ul>
            <li><em>‚ÄúEnter the name of the input objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
            <li><em>‚ÄúMeasurement category‚Äù</em>: <code class="language-plaintext highlighter-rouge">Number</code></li>
          </ul>
        </li>
        <li><em>‚ÄúEnter the name of the image on which to display the measurements‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
        <li><em>‚ÄúDisplay mode‚Äù</em>: <code class="language-plaintext highlighter-rouge">Text</code>
          <ul>
            <li><em>‚ÄúText color‚Äù</em>: <code class="language-plaintext highlighter-rouge">#ff0000</code></li>
            <li><em>‚ÄúNumber of decimals‚Äù</em>: <code class="language-plaintext highlighter-rouge">0</code></li>
          </ul>
        </li>
        <li><em>‚ÄúName the output image that has the measurements displayed‚Äù</em>: <code class="language-plaintext highlighter-rouge">ImageDisplay</code></li>
      </ul>
    </li>
    <li><strong>SaveImages</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>DisplayDataOnImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúSelect the type of image to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image</code>
          <ul>
            <li><em>‚ÄúSaved the format to save the image(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">tiff</code></li>
          </ul>
        </li>
        <li><em>‚ÄúEnter the name of the image to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">ImageDisplay</code></li>
        <li><em>‚ÄúSelect method for constructing file names‚Äù</em>: <code class="language-plaintext highlighter-rouge">From image filename</code>
          <ul>
            <li><em>‚ÄúEnter the image name (from NamesAndTypes) to be used as file prefix‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
            <li><em>‚ÄúAppend a suffix to the image file name?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>‚ÄúText to append to the image name‚Äù</em>: <code class="language-plaintext highlighter-rouge">_nucleiNumbers</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúOverwrite existing files without warning?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-3"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <p>The <code class="language-plaintext highlighter-rouge">Text color</code> parameter can be any of your choice, the hexa code is not really relevant. The only consideration is that it needs to be visible on top of the nuclei.</p>
      </blockquote>
    </li>
  </ol>
</blockquote>

<h2 id="segment-nucleoli">Segment nucleoli</h2>

<p>The nucleoli are lacking intensity in the <span class="notranslate">DNA</span> staining and therefore, we need to enhance the black holes before masking.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-detect-and-mask-dark-holes"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Detect and mask dark holes</h3>

  <ol>
    <li><strong>EnhanceOrSuppressFeatures</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>SaveImages</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEnter the name of the input image‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
        <li><em>‚ÄúEnter a name for the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span>darkholes</code></li>
        <li><em>‚ÄúSelect the operation‚Äù</em>: <code class="language-plaintext highlighter-rouge">Enhance</code>
          <ul>
            <li><em>‚ÄúFeature type‚Äù</em>: <code class="language-plaintext highlighter-rouge">Dark holes</code>
              <ul>
                <li><em>‚ÄúMaximum hole size‚Äù</em>: <code class="language-plaintext highlighter-rouge">15</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MaskImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>EnhanceOrSuppressFeatures</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEnter the name of the input image‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span>darkholes</code></li>
        <li><em>‚ÄúEnter the name of the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mask<span class="notranslate">DNA</span>darkholes</code></li>
        <li><em>‚ÄúUse objects or an image as a mask?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Objects</code>
          <ul>
            <li><em>‚ÄúEnter the name objects to mask the input image‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
          </ul>
        </li>
        <li><em>‚ÄúInvert the mask?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<p>Now that we have all the holes in one mask, we can segment the nucleoli as individual objects in the same way as we did with the nuclei. All the nucleoli can be then combined into one single image.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-segment-nucleoli-as-individual-objects"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Segment nucleoli as individual objects</h3>

  <ol>
    <li><strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MaskImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúUse advanced settings?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, use advanced settings</code>
          <ul>
            <li><em>‚ÄúEnter the name of the input image (from NamesAndTypes)‚Äù</em>: <code class="language-plaintext highlighter-rouge">Mask<span class="notranslate">DNA</span>darkholes</code></li>
            <li><em>‚ÄúEnter the name of the primary objects to be identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nucleoli</code></li>
            <li><em>‚ÄúTypical minimum diameter of objects, in pixel units (Min)‚Äù</em>: <code class="language-plaintext highlighter-rouge">2</code></li>
            <li><em>‚ÄúTypical maximum diameter of objects, in pixel units (Max)‚Äù</em>: <code class="language-plaintext highlighter-rouge">15</code></li>
            <li><em>‚ÄúDiscard objects touching the border of the image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
            <li><em>‚ÄúThreshold strategy‚Äù</em>: <code class="language-plaintext highlighter-rouge">Global</code>
              <ul>
                <li><em>‚ÄúThresholding method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Otsu</code>
                  <ul>
                    <li><em>‚ÄúTwo-class or three-class thresholding?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Two classes</code></li>
                    <li><em>‚ÄúThreshold correction factor‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.9</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúMethod to distinguish clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Shape</code>
              <ul>
                <li><em>‚ÄúMethod to draw dividing lines between clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Shape</code>
                  <ul>
                    <li><em>‚ÄúAutomatically calculate size of smoothing filter for declumping?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>‚ÄúAutomatically calculate minimum allowed distance between local maxima?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúHandling of objects if excessive number of objects identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Continue</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>ConvertObjectsToImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEnter the name of the input objects you want to convert to an image‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nucleoli</code></li>
        <li><em>‚ÄúEnter the name of the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge">MaskNucleoli</code></li>
        <li><em>‚ÄúSelect the color format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Binary (black &amp; white)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="combine-segmentation-masks">Combine segmentation masks</h2>

<p>We have now one segmentation mask per image with all the nuclei detected, <code class="language-plaintext highlighter-rouge">MaskNuclei</code>, and another one for the nucleoli, <code class="language-plaintext highlighter-rouge">MaskNucleoli</code>. These are binary masks in which the background is black and the objects detected are white. We would like to check whether both segmentation steps went well. That could be achieved by combining both of them (using different colors) into one image. Here, we are converting the nucleus mask to blue and the nucleoli to magenta. The outcome that you will obtain after the execution of the <span class="notranslate">workflow</span> will look like:</p>

<figure id="figure-5"><img src="../../images/tutorial-CP/img_combined_masks.png" alt="Combined mask for nuclei and nucleoli" /><figcaption><span class="figcaption-prefix">Figure 5:</span> Combined segmentation masks for nuclei (blue) and nucleoli (magenta).</figcaption></figure>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-convert-and-save-the-nuclei-and-nucleoli-masks"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Convert and save the nuclei and nucleoli masks</h3>

  <ol>
    <li><strong>GrayToColor</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>ConvertObjectsToImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEnter the name of the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge">CombinedMask</code></li>
        <li><em>‚ÄúSelect a color scheme‚Äù</em>: <code class="language-plaintext highlighter-rouge">RGB</code>
          <ul>
            <li><em>‚ÄúEnter the name of the image to be colored red‚Äù</em>: <code class="language-plaintext highlighter-rouge">MaskNucleoli</code></li>
            <li><em>‚ÄúRelative weight for the red image‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.8</code></li>
            <li><em>‚ÄúEnter the name of the image to be colored blue‚Äù</em>: <code class="language-plaintext highlighter-rouge">MaskNuclei</code></li>
            <li><em>‚ÄúRelative weight for the blue image‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.5</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>SaveImages</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>GrayToColor</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúSelect the type of image to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image</code>
          <ul>
            <li><em>‚ÄúSaved the format to save the image(s)‚Äù</em>: <code class="language-plaintext highlighter-rouge">tiff</code></li>
          </ul>
        </li>
        <li><em>‚ÄúEnter the name of the image to save‚Äù</em>: <code class="language-plaintext highlighter-rouge">CombinedMask</code></li>
        <li><em>‚ÄúSelect method for constructing file names‚Äù</em>: <code class="language-plaintext highlighter-rouge">From image filename</code>
          <ul>
            <li><em>‚ÄúEnter the image name (from NamesAndTypes) to be used as file prefix‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
            <li><em>‚ÄúAppend a suffix to the image file name?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
              <ul>
                <li><em>‚ÄúText to append to the image name‚Äù</em>: <code class="language-plaintext highlighter-rouge">_combinedMask</code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúOverwrite existing files without warning?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>

      <blockquote class="comment">
        <h3 id="comment-comment-4"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

        <ul>
          <li>You can pick any other color of your choice, as long as the contrast is good enough to distinguish both objects.</li>
          <li>We are saving here a tiff image but any other format of your choice would work too.</li>
        </ul>
      </blockquote>
    </li>
  </ol>

</blockquote>

<h1 id="background-extraction">Background extraction</h1>

<p>The background extraction is useful for quality control. For instance, in a high-exposed or low-contrast image, the nuclei will not be very different from the background and that may lead to the wrong segmentation.</p>

<p>To extract the background, we first need to get the foreground and subtract it from the original image. We already have the nuclei mask, however, we excluded incomplete nuclei, i.e., those touching the borders or those which sizes are outside the specified range. This means that the mask is not covering all the nuclei and we need to get rid of the constraints to get the complete foreground. At this stage, we want to detect everything with a certain intensity (foreground) and subtract it from the complete image to get the background.</p>

<h2 id="identify-the-foreground">Identify the foreground</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-segment-all-nuclei"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Segment all nuclei</h3>

  <ol>
    <li><strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>SaveImages</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúUse advanced settings?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes, use advanced settings</code>
          <ul>
            <li><em>‚ÄúEnter the name of the input image (from NamesAndTypes)‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
            <li><em>‚ÄúEnter the name of the primary objects to be identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">NucleiIncludingTouchingBorders</code></li>
            <li><em>‚ÄúTypical minimum diameter of objects, in pixel units (Min)‚Äù</em>: <code class="language-plaintext highlighter-rouge">15</code></li>
            <li><em>‚ÄúTypical maximum diameter of objects, in pixel units (Max)‚Äù</em>: <code class="language-plaintext highlighter-rouge">200</code></li>
            <li><em>‚ÄúDiscard objects outside the diameter range?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúDiscard objects touching the border of the image?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
            <li><em>‚ÄúThreshold strategy‚Äù</em>: <code class="language-plaintext highlighter-rouge">Global</code>
              <ul>
                <li><em>‚ÄúThresholding method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Otsu</code>
                  <ul>
                    <li><em>‚ÄúTwo-class or three-class thresholding?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Two classes</code></li>
                    <li><em>‚ÄúThreshold correction factor‚Äù</em>: <code class="language-plaintext highlighter-rouge">0.9</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúMethod to distinguish clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Shape</code>
              <ul>
                <li><em>‚ÄúMethod to draw dividing lines between clumped objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Shape</code>
                  <ul>
                    <li><em>‚ÄúAutomatically calculate size of smoothing filter for declumping?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                    <li><em>‚ÄúAutomatically calculate minimum allowed distance between local maxima?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><em>‚ÄúHandling of objects if excessive number of objects identified‚Äù</em>: <code class="language-plaintext highlighter-rouge">Continue</code></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>ConvertObjectsToImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>IdentifyPrimaryObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúEnter the name of the input objects you want to convert to an image‚Äù</em>: <code class="language-plaintext highlighter-rouge">NucleiIncludingTouchingBorders</code></li>
        <li><em>‚ÄúEnter the name of the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image_NucleiIncludingTouchingBorders</code></li>
        <li><em>‚ÄúSelect the color format‚Äù</em>: <code class="language-plaintext highlighter-rouge">Binary (black &amp; white)</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="remove-the-foreground-from-the-original-image">Remove the foreground from the original image</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-subtract-the-foreground-from-the-original-image"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Subtract the foreground from the original image</h3>

  <p><strong>ImageMath</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>ConvertObjectsToImage</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúEnter a name for the resulting image‚Äù</em>: <code class="language-plaintext highlighter-rouge">BG</code></li>
    <li><em>‚ÄúOperation‚Äù</em>: <code class="language-plaintext highlighter-rouge">Subtract</code>
      <ul>
        <li>In <em>‚ÄúFirst Image‚Äù</em>:
          <ul>
            <li><em>‚ÄúImage or measurement?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image</code>
              <ul>
                <li><em>‚ÄúEnter the name of the first image‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚ÄúSecond Image‚Äù</em>:
          <ul>
            <li><em>‚ÄúImage or measurement?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image</code>
              <ul>
                <li><em>‚ÄúEnter the name of the second image‚Äù</em>: <code class="language-plaintext highlighter-rouge">Image_NucleiIncludingTouchingBorders</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><em>‚ÄúIgnore the image masks?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
  </ul>

</blockquote>

<h1 id="feature-extraction">Feature extraction</h1>

<p>Now that we have the objects of interest segmented and the background extracted, we can start measuring parameters on them. In particular it is relevant to:</p>

<ul>
  <li>measure the granularity, texture and intensity of the nuclei and original image,</li>
  <li>measure the intensity of the original image, the background and foreground,</li>
  <li>measure the size and shape of nuclei and nucleoli,</li>
  <li>assess the quality of the original image,</li>
  <li>measure the area ocuppied by nuclei and nucleoli.</li>
</ul>

<p>A step that requires special attention is the relationship nucleolus-nucleus. This is useful to compute statistics on the number of nucleoli by nucleus.</p>

<blockquote class="comment">
  <h3 id="comment-comment-5"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

  <p>The order in which the tools in this section are chained is not relevant for the outcome.</p>
</blockquote>

<h2 id="measure-the-granularity-texture-intensity-size-and-shape">Measure the granularity, texture, intensity, size and shape</h2>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-measure-the-granularity-texture-intensity-size-and-shape"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Measure the granularity, texture, intensity, size and shape</h3>

  <ol>
    <li><strong>MeasureGranularity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>ImageMath</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚Äúnew image‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of a greyscale image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MeasureTexture</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureGranularity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚Äúnew image‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of an image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><em>‚ÄúMeasure images or objects?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Objects</code>
          <ul>
            <li>In <em>‚Äúnew object‚Äù</em>:
              <ul>
                <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new object‚Äù</em>
                  <ul>
                    <li><em>‚ÄúEnter the names of the objects to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MeasureObjectIntensity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureTexture</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚Äúnew image‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of an image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
              </ul>
            </li>
          </ul>
        </li>
        <li>In <em>‚Äúnew object‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new object‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the objects to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MeasureObjectSizeShape</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureObjectIntensity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚Äúnew object‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new object‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the object to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new object‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the object to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nucleoli</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-2"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>Why are we measuring the granularity, texture and intensity of the original image and the nuclei only?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-3"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>The nucleoli was not stained in the <span class="notranslate">DNA</span> channel and hence, the granularity, texture and intensity are constant values.</p>

  </blockquote>

</blockquote>

<h2 id="relate-nucleoli-to-their-parent-nucleus">Relate nucleoli to their parent nucleus</h2>

<p>It might be relevant to compute some statistics on the number of nucleoli inside each nucleus. CellProfiler has a very interesting module to relate both objects, in which each one of the nucleoli is assigned an identifier and linked to the identifier of its parent nucleus.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-relate-nucleoli-to-their-parent-nucleus"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Relate nucleoli to their parent nucleus</h3>

  <ol>
    <li><strong>RelateObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureObjectSizeShape</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúParent objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
        <li><em>‚ÄúChild objects‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nucleoli</code></li>
        <li><em>‚ÄúCalculate child-parent distances?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Both</code>
          <ul>
            <li><em>‚ÄúCalculate distances to other parents?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
          </ul>
        </li>
        <li><em>‚ÄúDo you want to save the children with parents as a new object set?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
      </ul>
    </li>
  </ol>

</blockquote>

<h2 id="measure-the-image-quality-related-parameters">Measure the image quality-related parameters</h2>

<p>In this section, we will measure the image quality, the area occupied by the nuclei and nucleoli in the original image and the intensity of the foreground and background.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-measure-the-image-quality"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Measure the image quality</h3>

  <ol>
    <li><strong>MeasureImageQuality</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>RelateObjects</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li><em>‚ÄúCalculate blur metrics?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
        <li><em>‚ÄúCalculate thresholds?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
          <ul>
            <li><em>‚ÄúUse all thresholding methods?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code>
              <ul>
                <li>In <em>‚Äúnew threshold method‚Äù</em>:
                  <ul>
                    <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new threshold method‚Äù</em>
                      <ul>
                        <li><em>‚ÄúSelect a thresholding method‚Äù</em>: <code class="language-plaintext highlighter-rouge">Otsu</code>
                          <ul>
                            <li><em>‚ÄúTwo-class or three-class thresholding?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Two classes</code></li>
                          </ul>
                        </li>
                      </ul>
                    </li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MeasureImageAreaOccupied</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureImageQuality</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚Äúnew area‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new area‚Äù</em>
              <ul>
                <li><em>‚ÄúMeasure the area occupied in a binary image, or in objects?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Objects</code>
                  <ul>
                    <li><em>‚ÄúEnter the name of the objects to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new area‚Äù</em>
              <ul>
                <li><em>‚ÄúMeasure the area occupied in a binary image, or in objects?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Objects</code>
                  <ul>
                    <li><em>‚ÄúEnter the name of the objects to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nucleoli</code></li>
                  </ul>
                </li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
    <li><strong>MeasureImageIntensity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureImageAreaOccupied</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
        <li>In <em>‚Äúnew image‚Äù</em>:
          <ul>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
                <li><em>‚ÄúMeasure the intensity only from areas enclosed by objects?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge"><span class="notranslate">DNA</span></code></li>
                <li><em>‚ÄúMeasure the intensity only from areas enclosed by objects?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
                  <ul>
                    <li><em>‚ÄúEnter the name of the objects that the intensity will be aggregated within‚Äù</em>: <code class="language-plaintext highlighter-rouge">Nuclei</code></li>
                  </ul>
                </li>
              </ul>
            </li>
            <li><i class="far fa-plus-square" aria-hidden="true"></i><span class="visually-hidden">param-repeat</span> <em>‚ÄúInsert new image‚Äù</em>
              <ul>
                <li><em>‚ÄúEnter the name of the image to measure‚Äù</em>: <code class="language-plaintext highlighter-rouge">BG</code></li>
                <li><em>‚ÄúMeasure the intensity only from areas enclosed by objects?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
              </ul>
            </li>
          </ul>
        </li>
      </ul>
    </li>
  </ol>

</blockquote>

<blockquote class="question">
  <h3 id="question-questions-3"><i class="far fa-question-circle" aria-hidden="true"></i><span class="visually-hidden">question</span> Questions</h3>

  <p>In step 3, we are measuring the area occupied by objects for the <span class="notranslate">DNA</span> twice. Why is that?</p>

  <blockquote class="solution">
    <h3 id="solution-solution-4"><i class="far fa-eye" aria-hidden="true"></i><span class="visually-hidden">solution</span> Solution</h3>

    <p>We want to have the global intensity of the image, so the parameter <em>‚ÄúMeasure the intensity only from areas enclosed by objects?‚Äù</em> has to be set to <code class="language-plaintext highlighter-rouge">No</code>. However, we also want to obtain the intensity of the foreground, so only the parts of the images that are enclosed by nuclei.</p>

  </blockquote>

</blockquote>

<h2 id="export-the-features-to-a-csv">Export the features to a CSV</h2>

<p>All the parameters that we have measured related to the images and objects need to be exported to a file for each one of 6 example images analysed.</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-export-features"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Export features</h3>

  <p><strong>ExportToSp<span class="notranslate">reads</span>heet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúSelect the input CellProfiler pipeline‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>MeasureImageIntensity</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúSelect the column delimiter‚Äù</em>: <code class="language-plaintext highlighter-rouge">Tab</code></li>
    <li><em>‚ÄúAdd a prefix to file names?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Do not add prefix to the file name</code></li>
    <li><em>‚ÄúCreate a GenePattern GCT file?‚Äù</em>: <code class="language-plaintext highlighter-rouge">No</code></li>
    <li><em>‚ÄúExport all measurement types?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
  </ul>

</blockquote>

<h1 id="run-the-cellprofiler-pipeline">Run the CellProfiler pipeline</h1>

<p>All the steps in our <span class="notranslate">workflow</span> (except for the <strong>IDR download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>) have been passing through an <code class="language-plaintext highlighter-rouge">output_pipeline</code> as a parameter. This was the way to assemble all the modules from CellProfiler, now we can run all of them together!</p>

<blockquote class="notranslate hands_on">
  <h3 id="hands_on-hands-on-run-cellprofiler-pipeline"><i class="fas fa-pencil-alt" aria-hidden="true"></i><span class="visually-hidden">hands_on</span> Hands-on: Run CellProfiler pipeline</h3>

  <p><strong>CellProfiler</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span> with the following parameters:</p>
  <ul>
    <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúPipeline file‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_pipeline</code> (output of <strong>ExportToSp<span class="notranslate">reads</span>heet</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
    <li><em>‚ÄúAre the input images packed into a tar archive?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code>
      <ul>
        <li><i class="far fa-file" aria-hidden="true"></i><span class="visually-hidden">param-file</span> <em>‚ÄúA tarball of images‚Äù</em>: <code class="language-plaintext highlighter-rouge">output_tar</code> (output of <strong>IDR Download</strong> <i class="fas fa-wrench" aria-hidden="true"></i><span class="visually-hidden">tool</span>)</li>
      </ul>
    </li>
    <li><em>‚ÄúDetailed logging file?‚Äù</em>: <code class="language-plaintext highlighter-rouge">Yes</code></li>
  </ul>

  <blockquote class="comment">
    <h3 id="comment-comment-6"><i class="far fa-comment-dots" aria-hidden="true"></i><span class="visually-hidden">comment</span> Comment</h3>

    <p>This is the only time-consuming step of the <span class="notranslate">workflow</span>, as it needs to perform the whole analysis in the input dataset.</p>
  </blockquote>

</blockquote>

<h1 class="no_toc" id="conclusion">Conclusion</h1>

<p>In this tutorial, you have downloaded images from a public image repository into your <span class="notranslate">Galaxy</span> history. After that, you have built and run a typical image analysis pipeline, composed of segmentation of several objects and feature extraction. As an outcome, you got plenty of features to analyse! And some masks to check that the segmentation algorithms worked as expected. Now you are ready to perform your biological data analysis!</p>


                    
                    <blockquote class="key_points">
                        <h3><i class="fas fa-key" aria-hidden="true"></i><span class="visually-hidden">keypoints</span> Key points</h3>
                        <ul>
                            
                            <li><p>Galaxy workflows can download images from the IDR, selecting specific channels, time points, z-stack positions and crop the image in different ways.</p>
</li>
                            
                            <li><p>CellProfiler in Galaxy can segment and extract features of any object of interest.</p>
</li>
                            
                            <li><p>The features and masks can be exported for further analysis.</p>
</li>
                            
                        </ul>
                    </blockquote>
                    

                    
                    <h1 data-toc-skip>Useful literature</h1>
                    <p>Further information, including links to documentation and original publications, regarding the tools, analysis techniques and the interpretation of results described in this tutorial can be found <a href="/training-material/topics/imaging#references">here</a>.</p>
                    

                    
                    <h1 id="bibliography">References</h1>
                    <ol class="bibliography"><li><span id="Heriche_2014"><b>Integration of biological data by kernels on graph nodes allows prediction of new genes involved in mitotic chromosome condensation</b>, 2014 Mol Biol Cell. 2014;25(16):2522-2536:</span>
    
    
    <a href="https://doi.org/10.1091/mbc.E13-04-0221">10.1091/mbc.E13-04-0221</a>
    
    

</li>
<li><span id="McQuin_2018"><b>CellProfiler 3.0: Next-generation image processing for biology</b>, 2018 PLoS Biol. 16(7):e2005970:</span>
    
    
    <a href="https://doi.org/10.1371/journal.pbio.2005970">10.1371/journal.pbio.2005970</a>
    
    
      <!-- prevent duplicate doi links -->
      
      <a href="https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2005970">https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.2005970</a>
      
    

</li></ol>
                    

                    <h1>Feedback</h1>
                    <p class="text-muted">Did you use this material as an instructor? Feel free to give us feedback on <a href="https://github.com/galaxyproject/training-material/issues/1452" target="_blank">how it went</a>.</p>

                    <div id="feedback-button">
                        <img src="/training-material/shared/images/feedback.png" title="Click to activate" alt="Click here to load Google feedback frame" />
                    </div>
                    <div id="feedback-form">
                    </div>
                    <script type="text/javascript">
                        (function (window, document) {
                            function onDocumentReady(fn) {
                                if (document.attachEvent ? document.readyState === "complete" : document.readyState !== "loading") {
                                    fn();
                                } else {
                                    document.addEventListener('DOMContentLoaded', fn);
                                }
                            }

                            onDocumentReady(function () {
                                $("#feedback-button").click(function(evt){
                                    var e = $(evt.target)
                                    e.hide();

                                    $("#feedback-form").html(`
                                        <iframe id="feedback-google" class="google-form" src="https://docs.google.com/forms/d/e/1FAIpQLSd4VZptFTQ03kHkMz0JyW9b6_S8geU5KjNE_tLM0dixT3ZQmA/viewform?embedded=true&entry.1235803833=Nucleoli segmentation and feature extraction using CellProfiler (Imaging)">Loading...</iframe>
                                    `)
                                })
                            });
                        })(window, document);
                    </script>



                    <h1>Citing this Tutorial</h1>
                    <p>
                        <ol>
                            <li id="citation-text">Beatriz Serrano-Solano, Jean-Karim H√©rich√©, 2021 <b>Nucleoli segmentation and feature extraction using CellProfiler (Galaxy Training Materials)</b>. <a href="/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html">/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html</a> Online; accessed TODAY
                            </li>
                            <li>
                            Batut et al., 2018 <b>Community-Driven Data Analysis Training for Biology</b> Cell Systems <a href="https://doi.org/10.1016%2Fj.cels.2018.05.012">10.1016/j.cels.2018.05.012</a>
                            </li>
                        </ol>
                    </p>


                    <blockquote class="details">
                      <h3><i class="fa fa-info-circle" aria-hidden="true"></i><span class="visually-hidden">details</span> BibTeX</h3>
                      <p style="display: none;">

                    <div class="highlighter-rouge"><div class="highlight"><pre class="highlight">
<code id="citation-code">@misc{imaging-tutorial-CP,
    author = "Beatriz Serrano-Solano and Jean-Karim H√©rich√©",
    title = "Nucleoli segmentation and feature extraction using CellProfiler (Galaxy Training Materials)",
    year = "2021",
    month = "01",
    day = "06"
    url = "\url{/training-material/topics/imaging/tutorials/tutorial-CP/tutorial.html}",
    note = "[Online; accessed TODAY]"
}
@article{Batut_2018,
        doi = {10.1016/j.cels.2018.05.012},
        url = {https://doi.org/10.1016%2Fj.cels.2018.05.012},
        year = 2018,
        month = {jun},
        publisher = {Elsevier {BV}},
        volume = {6},
        number = {6},
        pages = {752--758.e1},
        author = {B{\'{e}}r{\'{e}}nice Batut and Saskia Hiltemann and Andrea Bagnacani and Dannon Baker and Vivek Bhardwaj and Clemens Blank and Anthony Bretaudeau and Loraine Brillet-Gu{\'{e}}guen and Martin {\v{C}}ech and John Chilton and Dave Clements and Olivia Doppelt-Azeroual and Anika Erxleben and Mallory Ann Freeberg and Simon Gladman and Youri Hoogstrate and Hans-Rudolf Hotz and Torsten Houwaart and Pratik Jagtap and Delphine Larivi{\`{e}}re and Gildas Le Corguill{\'{e}} and Thomas Manke and Fabien Mareuil and Fidel Ram{\'{\i}}rez and Devon Ryan and Florian Christoph Sigloch and Nicola Soranzo and Joachim Wolff and Pavankumar Videm and Markus Wolfien and Aisanjiang Wubuli and Dilmurat Yusuf and James Taylor and Rolf Backofen and Anton Nekrutenko and Bj√∂rn Gr√ºning},
        title = {Community-Driven Data Analysis Training for Biology},
        journal = {Cell Systems}
}</code>
                    </pre></div></div>
                    </p>
                    </blockquote>


<script type="text/javascript">
// update the date on load, or leave fallback of 'today'
d = new Date();
document.getElementById("citation-code").innerHTML = document.getElementById("citation-code").innerHTML.replace("TODAY", d.toDateString());
document.getElementById("citation-text").innerHTML = document.getElementById("citation-text").innerHTML.replace("TODAY", d.toDateString());
</script>

                </div>
            </div>
        </div>

        <h3><i class="far fa-thumbs-up" aria-hidden="true"></i><span class="visually-hidden">congratulations</span> Congratulations on successfully completing this tutorial!</h3>

        

        
    </section>
</div>


<footer>
    <div class="container">
        <p>
            This material is the result of a collaborative work. Thanks to the
            <a href="https://wiki.galaxyproject.org/Teach/GTN">Galaxy Training Network</a>
            and all the <a href="/training-material/hall-of-fame">contributors</a> (Beatriz Serrano-Solano, Jean-Karim H√©rich√©)!
        </p>
        <p>
            Found a typo? Something is wrong in this tutorial? Edit it on
            <a href="https://github.com/galaxyproject/training-material/tree/master/topics/imaging/tutorials/tutorial-CP/tutorial.md">GitHub</a>.
        </p>
        <p>
    The content of the tutorials and website is licensed under the <a rel="license" href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>.
</p>
    </div>
</footer>

    </body>
    <script type="text/javascript" src="/training-material/assets/js/jquery.slim.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/popper.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap.min.js?v=3"></script>
    <script type="text/javascript" src="/training-material/assets/js/details-element-polyfill.js"></script>
    <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="/training-material/assets/js/bootstrap-toc.min.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/main.js"></script>
    <script type="text/javascript" src="/training-material/assets/js/theme.js"></script>

    <script type="text/javascript" src="/training-material/assets/js/clipboard.min.js"></script>
    <script type="text/javascript">
    var snippets=document.querySelectorAll('div.highlight');
    [].forEach.call(snippets,function(snippet){
        snippet.firstChild.insertAdjacentHTML('beforebegin','<button class="btn btn-light" data-clipboard-snippet><i class="fa fa-copy"></i>&nbsp;Copy</button>');
    });

    var clipboardSnippets=new ClipboardJS('[data-clipboard-snippet]',{
        target:function(trigger){return trigger.nextElementSibling;
    }});
    </script>
    

    <script type="text/javascript">
        if(window.location.hostname === "galaxyproject.github.io") {
            // Redirect
            var redirect = "https://training.galaxyproject.org" + window.location.pathname + window.location.search;
            $('div.container.main-content').prepend("<div class='alert alert-warning'><strong>Note: </strong>This content has a new home at <a href=\"" + redirect + "\">" + redirect + "</a>, which you will be redirected to in 5 seconds.</div>");

            window.setTimeout(function(){
                window.location.href = redirect;
            }, 5000)

        }
    </script>
</html>
