---
layout: tutorial_slides
logo: "GTN"

title: "Module 10: Clinical Integration"
questions: []
objectives:
- Clinical data overview
- Some single data type analysis (quick review)
- Some data integration methods
- Concatenate and Cluster (iCluster, SNF)
- Advantages and drawbacks of different integration methods
- Survival analysis

key_points: []
time_estimation: 2H
contributions:
  authorship: [larunerdman]
  editing: [hexylena]
  funding: [bioinformatics-ca,erasmusplusgsickkids,vector-institute,uni-toronto,mcgill]

requirements:
- type: "internal"
  topic_name: data-science
  tutorials:
      - r-basics
      - r-advanced

priority: 11
---

# Canadian Bioinformatics Workshops

[www.bioinformatics.ca](www.bioinformatics.ca)

[bioinformaticsdotca.github.io](bioinformaticsdotca.github.io)

Bioinformatics for Cancer Genomics

June 2021

---

# 5

.pull-left[
.image-35[![DNA strands](images/CBW_CG2021_Erdman_v3-5_1.jpg)] genetic
.image-35[![schematic of gene expression](images/CBW_CG2021_Erdman_v3-5_2.jpg)] epigenetic
.image-35[![survey diagram](images/CBW_CG2021_Erdman_v3-5_3.jpg)] questionnaire
.image-35[![another tiny gene schematic](images/CBW_CG2021_Erdman_v3-5_4.jpg)] mRNA expression
.image-35[![cell nucleus with arrows coming out to proteins](images/CBW_CG2021_Erdman_v3-5_5.jpg)] miRNA
]
.pull-right[
.image-35[![photo of many fruits and vegetables](images/CBW_CG2021_Erdman_v3-5_6.jpg)] diet
.image-35[![cropped photo of white man wearing blue scrubs with hexagonal health graphics](images/CBW_CG2021_Erdman_v3-5_7.jpg)] clinical 
.image-35[![brain scans in rainbow colours](images/CBW_CG2021_Erdman_v3-5_8.jpg)] imaging
.image-35[![3d protein model](images/CBW_CG2021_Erdman_v3-5_9.jpg)] protein
.image-35[![cartoon of a person in bed](images/CBW_CG2021_Erdman_v3-5_10.jpg)] **available patient data**
]


---

# Why integrate patient data

- To identify more homogeneous subsets of patients that
- might respond similarly to a given drug
- may have a similar prognosis
- may respond better to similar clinical management

---

# Single data type analysis

20 GBM, gene expression for ~18,000 genes

.pull-right[
![hierarchically clustered gene expression table. It's mostly black except for some green squares in a line at the right](images/CBW_CG2021_Erdman_v3-7_1.jpg)
]

.pull-left[

- Collect gene expression
- Select most varied genes
- Perform hierarchical clustering using selected genes
- Identify clusters (here, 2 clusters)
- Identify genes associated that differ between clusters using t-test (if 2 clusters) or ANOVA (if more)
- Correct for multiple hypothesis testing

{% cite Liang_2005 %}
]

---

# Single data type analysis

20 GBM, gene expression for ~18,000 genes

.pull-left[
![](images/CBW_CG2021_Erdman_v3-8_2.jpg)
{% cite Liang_2005 %}
]

.pull-right[
Becomes this:
![](images/CBW_CG2021_Erdman_v3-8_1.jpg)
]

---

# Kaplan-Meier Curve

![](images/CBW_CG2021_Erdman_v3-8_1.jpg)

--

The probability of surviving for 1 year is 20% for Group 2
The probability of surviving for 1 year is 80% for Group 1
Black dots represent censored observations

---

# Single data type driven integration

200 GBMs

![](images/200gbms.png)

--

Lead to identification of proneural, neural, classical and mesenchymal groups

{% cite Verhaak_2010 %}

--

**What about methylation data?**

---

# Integration approaches

1. Concatenate and cluster (commonly used in TCGA analysis)
2. iCluster ({% cite Shen_2010 %})
3. SNF ({% cite Wang_2014 %})

**Not an exhaustive list!**

---

# Concatenate and Cluster

![](images/before.png)

---

# Concatenate and Cluster

![](images/after.png)


---

# Concatenate and Cluster

## Hierarchical Clustering

![](images/hierarchical.png)


.footnote[
[source](http://people.revoledu.com/kardi/tutorial/Clustering/Numerical%20Example.htm)
]

---

# Concatenate and Cluster

## Hierarchical Clustering


![](images/CBW_CG2021_Erdman_v3-20_1.jpg)
![](images/CBW_CG2021_Erdman_v3-20_2.jpg)
![](images/CBW_CG2021_Erdman_v3-20_3.jpg)


.footnote[
[source](http://people.revoledu.com/kardi/tutorial/Clustering/Numerical%20Example.htm)
]


---

# Deciding on the number of clusters

- Arbitrarily cutting the dendrogram (by eye)
- Silhouette statistic
- Eigen gap (for spectral clustering)
- Many more!

---

# Silhouette statistic

- First presented by Rousseeuw (1987) to show graphically how well each pattern (observation) is classified to a cluster
- For each pattern *i* in Class Cr:

<math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><mi>S</mi><mi>i</mi><msub><mi>l</mi><mi>j</mi></msub><mo>=</mo><mfrac><mrow><mi>b</mi><mo>(</mo><mi>i</mi><mo>)</mo><mo>-</mo><mi>a</mi><mo>(</mo><mi>i</mi><mo>)</mo></mrow><mrow><mtext>max</mtext><mo>{</mo><mi>b</mi><mo>(</mo><mi>i</mi><mo>)</mo><mo>,</mo><mi>a</mi><mo>(</mo><mi>i</mi><mo>)</mo><mo>}</mo></mrow></mfrac></math>

- a(i) is the average distance to all other patterns in Cr
- b(i) is the average distance to all other patterns in other clusters

.pull-left[
- -1≤ <i>sil</i><sub>i</sub>≤1
- <i>Sil = 1: </i> good assignment
- <i>Sil = -1: </i> bad assignment
- <i>Sil = 0: </i> borderline
]
.pull-right[
.image-30[![](images/CBW_CG2021_Erdman_v3-22_2.jpg)]
]

---

# iCluster ({% cite Shen_2010 %})

- Gaussian latent variable model
- Sparsity regularization (Lasso-type)
- Latent variables Z (embedding is shared)

![](images/CBW_CG2021_Erdman_v3-23_1.jpg)

Package on CRAN: iCluster

---

# Drawbacks of existing methods

- A lot of manual processing (e.g. need to pre-filter genes for iCluster – takes ~1500 genes max)
- Many steps in the pipeline
- Integration mostly done in the feature space – if there is signal in a combination of features, it’ll be lost
- Focusing on similarity across data types – what if there is complementary information?

---

# Similarity Network Fusion ({% cite Wang_2014 %})

Integrate data in the patient space

1. Construct patient similarity matrix
2. Fuse multiple matrices

---

# 1. Create Data Type-Specific

![](images/data-specific.png)

Similarity Networks

---

# 1. Create Data Type-Specific

![](images/data-specific2.png)

---

# 2. Combine networks

![](images/combine1.png)

---

# 2. Combine networks

![](images/combine2.png)

---

# 2. Combine networks

![](images/combine3.png)

---

# Case study: Glioblastoma

.pull-left[
![](images/case.png)
]

--

.pull-left[
![](images/case2.png)
]

---
# 32

![](images/CBW_CG2021_Erdman_v3-32_1.jpg)
![](images/CBW_CG2021_Erdman_v3-32_2.jpg)
![](images/CBW_CG2021_Erdman_v3-32_3.jpg)

---
# 33

![](images/CBW_CG2021_Erdman_v3-33_1.jpg)
![](images/CBW_CG2021_Erdman_v3-33_2.jpg)
![](images/CBW_CG2021_Erdman_v3-33_3.jpg)
![](images/CBW_CG2021_Erdman_v3-33_4.jpg)
![](images/CBW_CG2021_Erdman_v3-33_5.jpg)

# Clinical properties of the subtypes
Subtype1     Subtype2     Subtype3
80
60
40
20
A
ge
  (
ye
ar
s)
Treated
Untreated
Survival time (months)
0        20       40      60        80      100
1.0
0.8
0.6
0.4
0.2
0.0
S
ur
vi
va
l p
ro
ba
bi
lit
y
Subtype 1
p-value = 3x10 -5
p = 0.001
Subtype1
Subtype2
Subtype3
1.0
0.8
0.6
0.4
0.2
0.0
S
ur
vi
va
l p
ro
ba
bi
lit
y
0              40            80           120
Survival time (months)
p-value = 2x10 -4

---
# 34


# Patient networks: 
# advantages and disadvantages
✓ Creates a unified view of patients based on multiple
heterogeneous sources
✓ Integrates gene and non-gene based data
✓ No need to do gene pre-selection
✓ Robust to different types of noise
✓ Scalable
Package on CRAN :   <b>SNFtool</b>
- Integrative feature selection
- Growing the network requires extra work
- Unsupervised – hard to turn into a supervised problem

---
# 35

![](images/CBW_CG2021_Erdman_v3-35_1.jpg)
![](images/CBW_CG2021_Erdman_v3-35_2.jpg)
![](images/CBW_CG2021_Erdman_v3-35_3.jpg)
![](images/CBW_CG2021_Erdman_v3-35_4.jpg)

We are on a Short Break
Back at: 11:01 AM
Workshop Sponsors:

---
# 36


# Survival Analysis
• Survival data
– Hazard Rates
– Survival Functions
• Kaplan-Meier Estimator
• Log-rank test
• Cox proportional hazards ratio model

---
# 37


# Survival data
• Time to a single event (e.g. time to death or time to treatment
failure)
• Some data on patients maybe missing, e.g. lost due to the end of
the study – such data is called  <i>right censored (i.e. death occurred </i>
<i>to the right of the last observation)</i>
– Uncensored data:  we observe death time
– Censored data: we know that it’s beyond the reported time
<i>Note:</i>  Assume non-informative (independent) censoring (e.g. moved
out of town and lost track of the patient not related to the disease)

---
# 38

![](images/CBW_CG2021_Erdman_v3-38_1.jpg)
![](images/CBW_CG2021_Erdman_v3-38_2.jpg)

# Survival data example
days to last follow-up

---
# 39


# Two important statistics
• Event time: t
• Survival function
<i>S(t) = </i> P( <i>T &gt; t</i> ) <i> </i> – probability of a person being alive at  <i>t</i>
• Hazard rate
H(t) = lim
<i>&#916;t</i> ->0
 P( <i>t &#8804; T &#8804; t + &#916;t</i> | <i>T &gt; t</i> )
Probability of a person dying (failing) in the next instant

---
# 40


# Some examples of the hazard rate
• Constant hazard rate – no aging
• Positive hazard rate – the older you are the more
likely you are to die
• Negative hazard rate – e.g. dying risk is the highest
at birth (infant mortality)

---
# 41

![](images/CBW_CG2021_Erdman_v3-41_1.jpg)

# Kaplan Meier Estimator
• Basic estimator Kaplan-Meier (1958)
• S(t) – probability that a member from a given population
will have a lifetime exceeding t
• <i>n</i>
<i>i</i>
 – number of people at risk of dying at time t
• <i>d</i>
<i>i</i>
 – number of actual deaths at time t

---
# 42

![](images/CBW_CG2021_Erdman_v3-42_1.png)

# Kaplan-Meier Curve
https://www.graphpad.com/guides/prism/latest/statistics/stat_determining_the_median_followu.htm

---
# 43

![](images/CBW_CG2021_Erdman_v3-43_1.jpg)
![](images/CBW_CG2021_Erdman_v3-43_2.jpg)
![](images/CBW_CG2021_Erdman_v3-43_3.jpg)

# Are Kaplan Meier curves significantly 
# different?  Log-rank test
• Comparing 2 groups
• j = 1..J be the distinct events in either group
• For each j, N
1j
 and N
2j
 – numbers of people at risk in groups 1 and 2 respectively
at the start of period j (N
j
 = N
1j
+N
2j
)
• O
1J
 and O
2J
 - observed number of events in each group (corresponds to variable
d
i
 above). O
j
 = O
1J
 + O
2J
• Null hypothesis: the distributions (2 curves) are the same. Given that O
j

happened across both groups at time j, O
1J
 is hypergeometric with expected
value and variance:
get p-value from
look up table

---
# 44

![](images/CBW_CG2021_Erdman_v3-44_1.jpg)

# Hazard ratio

---
# 45

![](images/CBW_CG2021_Erdman_v3-45_1.jpg)
![](images/CBW_CG2021_Erdman_v3-45_2.png)

# Cox proportional hazard (Cox PH) model
• Captures how well multiple variables x
1
…x
p
(e.g. genes,
clinical variables, etc) affect survival
• Estimates the ratio of the risks
Cox regression:
h
0
(t) – baseline hazard
other predictors are exponentially increasing the hazard
prognostic index

---
# 46

![](images/CBW_CG2021_Erdman_v3-46_1.jpg)
![](images/CBW_CG2021_Erdman_v3-46_2.jpg)
![](images/CBW_CG2021_Erdman_v3-46_3.jpg)
![](images/CBW_CG2021_Erdman_v3-46_4.jpg)

# Some intuition behind the model
- represents log hazard ratio increase for one unit
increase of the predictor ( <i>x</i>
<i>j</i>
) holding all other
predictors constant
- Hazard ratio increase for one unit increase of x
j
- Means increasing x
j
 is associated with decreased
hazard and longer survival times

---
# 47

![](images/CBW_CG2021_Erdman_v3-47_1.jpg)

# Using and interpreting Cox PH
1. Hazard ratio for a subject i with a set of predictors X
compared to a subject j with a set of predictors X*:
2. HR can be interpreted as a percentage change in risk
Example:  Let x = 1 when treatment is active and x=0 when
treatment is placebo. If HR = e β =0.8, it means 20% decrease
in mortality risk if using treatment compared to placebo

---
# 48


# Concordance index (C-index)
• Captures the ability of the model to order individuals
correctly with respect to their survival time

C-index = (#concordant pairs of indiv. + 0.5 * ties)/ all pairs
• Important – no other metric captures the ordering of
individuals
